diff -Naur urlgrabber-3.10.2/setup.py urlgrabber-3.10.2-p/setup.py
--- urlgrabber-3.10.2/setup.py	2017-02-02 12:21:04.000000000 +0100
+++ urlgrabber-3.10.2-p/setup.py	2018-02-17 14:52:35.965691227 +0100
@@ -33,7 +33,7 @@
 scripts = ['scripts/urlgrabber']
 data_files = [
     ('share/doc/' + name + '-' + version, ['README','LICENSE', 'TODO', 'ChangeLog']),
-    ('libexec', ['scripts/urlgrabber-ext-down']),
+    ('lib/urlgrabber', ['scripts/urlgrabber-ext-down']),
 ]
 options = { 'clean' : { 'all' : 1 } }
 classifiers = [
diff -Naur urlgrabber-3.10.2/urlgrabber/grabber.py urlgrabber-3.10.2-p/urlgrabber/grabber.py
--- urlgrabber-3.10.2/urlgrabber/grabber.py	2017-02-02 12:21:04.000000000 +0100
+++ urlgrabber-3.10.2-p/urlgrabber/grabber.py	2018-02-17 14:52:54.133731969 +0100
@@ -2072,7 +2072,7 @@
 class _ExternalDownloader:
     def __init__(self):
         self.popen = subprocess.Popen(
-            '/usr/libexec/urlgrabber-ext-down',
+            '/usr/lib/urlgrabber/urlgrabber-ext-down',
             stdin = subprocess.PIPE,
             stdout = subprocess.PIPE,
         )
