diff -Naur mod_wsgi-4.9.4/configure mod_wsgi-4.9.4.new/configure
--- mod_wsgi-4.9.4/configure	2022-09-12 03:56:13.000000000 +0200
+++ mod_wsgi-4.9.4.new/configure	2023-10-05 19:08:10.172423213 +0200
@@ -2850,11 +2850,11 @@
 
 
 PYTHON_VERSION=`${PYTHON} -c 'from sys import stdout; \
-    from distutils import sysconfig; \
+    import sysconfig; \
     stdout.write(sysconfig.get_config_var("VERSION"))'`
 
 PYTHON_LDVERSION=`${PYTHON} -c 'from sys import stdout; \
-    from distutils import sysconfig; \
+    import sysconfig; \
     stdout.write(sysconfig.get_config_var("LDVERSION") or "")'`
 
 if test x"${PYTHON_LDVERSION}" = x""; then
@@ -2862,11 +2862,11 @@
 fi
 
 CPPFLAGS1=`${PYTHON} -c 'from sys import stdout; \
-    from distutils import sysconfig; \
+    import sysconfig; \
     stdout.write("-I" + sysconfig.get_config_var("INCLUDEPY"))'`
 
 CPPFLAGS2=`${PYTHON} -c 'from sys import stdout; \
-    from distutils import sysconfig; \
+    import sysconfig; \
     stdout.write(" ".join(filter(lambda x: x.startswith("-D"), \
     sysconfig.get_config_var("CFLAGS").split())))'`
 
@@ -2881,20 +2881,19 @@
 
 
 PYTHONLIBDIR=`${PYTHON} -c 'from sys import stdout; \
-    from distutils import sysconfig; \
+    import sysconfig; \
     stdout.write(sysconfig.get_config_var("LIBDIR"))'`
 PYTHONCFGDIR=`${PYTHON} -c 'from sys import stdout; \
-    import distutils.sysconfig; \
-    stdout.write(distutils.sysconfig.get_python_lib(plat_specific=1, \
-    standard_lib=1) +"/config")'`
+    import sysconfig; \
+    stdout.write(sysconfig.get_path("platstdlib") +"/config")'`
 PYTHONFRAMEWORKDIR=`${PYTHON} -c 'from sys import stdout; \
-    from distutils import sysconfig; \
+    import sysconfig; \
     stdout.write(sysconfig.get_config_var("PYTHONFRAMEWORKDIR"))'`
 PYTHONFRAMEWORKPREFIX=`${PYTHON} -c 'from sys import stdout; \
-    from distutils import sysconfig; \
+    import sysconfig; \
     stdout.write(sysconfig.get_config_var("PYTHONFRAMEWORKPREFIX"))'`
 PYTHONFRAMEWORK=`${PYTHON} -c 'from sys import stdout; \
-    from distutils import sysconfig; \
+    import sysconfig; \
     stdout.write(sysconfig.get_config_var("PYTHONFRAMEWORK"))'`
 
 if test "${PYTHON_LDVERSION}" != "${PYTHON_VERSION}"; then
@@ -2920,10 +2919,10 @@
     fi
 
     LDLIBS2=`${PYTHON} -c 'from sys import stdout; \
-        from distutils import sysconfig; \
+        import sysconfig; \
         stdout.write(sysconfig.get_config_var("LIBS"))'`
     LDLIBS3=`${PYTHON} -c 'from sys import stdout; \
-        from distutils import sysconfig; \
+        import sysconfig; \
         stdout.write(sysconfig.get_config_var("SYSLIBS"))'`
 else
     LDFLAGS1="-Wl,-F${PYTHONFRAMEWORKPREFIX} -framework ${PYTHONFRAMEWORK}"
@@ -2931,13 +2930,13 @@
     VERSION="${PYTHON_VERSION}"
     STRING="${PYTHONFRAMEWORKDIR}/Versions/${VERSION}/${PYTHONFRAMEWORK}"
     LDFLAGS2=`${PYTHON} -c "from sys import stdout; \
-        from distutils import sysconfig; \
+        import sysconfig; \
         stdout.write(sysconfig.get_config_var(
         \"LINKFORSHARED\").replace(\"${STRING}\", ''))" | \
         sed -e 's/-Wl,-stack_size,[0-9]*//'`
 
     LDLIBS1=`${PYTHON} -c 'from sys import stdout; \
-        from distutils import sysconfig; \
+        import sysconfig; \
         stdout.write(sysconfig.get_config_var("LIBS"))'`
 fi
 
diff -Naur mod_wsgi-4.9.4/configure.ac mod_wsgi-4.9.4.new/configure.ac
--- mod_wsgi-4.9.4/configure.ac	2022-09-12 03:56:13.000000000 +0200
+++ mod_wsgi-4.9.4.new/configure.ac	2023-10-05 19:08:10.172423213 +0200
@@ -141,9 +141,8 @@
     import sysconfig; \
     stdout.write(sysconfig.get_config_var("LIBDIR"))'`
 PYTHONCFGDIR=`${PYTHON} -c 'from sys import stdout; \
-    import distutils.sysconfig; \
-    stdout.write(distutils.sysconfig.get_python_lib(plat_specific=1, \
-    standard_lib=1) +"/config")'`
+    import sysconfig; \
+    stdout.write(sysconfig.get_path("platstdlib") +"/config")'`
 PYTHONFRAMEWORKDIR=`${PYTHON} -c 'from sys import stdout; \
     import sysconfig; \
     stdout.write(sysconfig.get_config_var("PYTHONFRAMEWORKDIR"))'`
diff -Naur mod_wsgi-4.9.4/.github/workflows/main.yml mod_wsgi-4.9.4.new/.github/workflows/main.yml
--- mod_wsgi-4.9.4/.github/workflows/main.yml	2022-09-12 03:56:13.000000000 +0200
+++ mod_wsgi-4.9.4.new/.github/workflows/main.yml	2023-10-05 19:08:10.175423235 +0200
@@ -11,11 +11,15 @@
   build:
     name: "Build mod_wsgi packages"
     runs-on: "ubuntu-20.04"
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12-dev"]
     steps:
-      - uses: "actions/checkout@v2"
-      - uses: "actions/setup-python@v2"
+      - uses: "actions/checkout@v4"
+      - uses: "actions/setup-python@v4"
         with:
-          python-version: "3.9"
+          python-version: "${{ matrix.python-version }}"
       - name: "Update package details"
         run: sudo apt --fix-missing update
       - name: "Install Apache package"
@@ -23,9 +27,9 @@
       - name: "Build mod_wsgi packages"
         run: ./package.sh && ls -las dist
       - name: "Store built packages"
-        uses: actions/upload-artifact@v2
+        uses: actions/upload-artifact@v3
         with:
-          name: dist
+          name: dist ${{ matrix.python-version }}
           path: dist/*
 
   tests:
@@ -38,14 +42,14 @@
       matrix:
         python-version: ["2.7", "3.5", "3.6", "3.7", "3.8", "3.9", "3.10", "3.11-dev"]
     steps:
-      - uses: "actions/checkout@v2"
-      - uses: "actions/setup-python@v2"
+      - uses: "actions/checkout@v4"
+      - uses: "actions/setup-python@v4"
         with:
           python-version: "${{ matrix.python-version }}"
       - name: "Download built packages"
-        uses: actions/download-artifact@v2
+        uses: actions/download-artifact@v3
         with:
-            name: dist
+            name: dist ${{ matrix.python-version }}
             path: dist
       - name: "Update package details"
         run: sudo apt --fix-missing update
diff -Naur mod_wsgi-4.9.4/package.sh mod_wsgi-4.9.4.new/package.sh
--- mod_wsgi-4.9.4/package.sh	2022-09-12 03:56:13.000000000 +0200
+++ mod_wsgi-4.9.4.new/package.sh	2023-10-05 19:08:10.173423220 +0200
@@ -6,6 +6,8 @@
 
 rm -f pyproject.toml
 
+pip install setuptools
+
 python setup.py sdist
 
 ln -s pyproject.toml.in pyproject.toml
diff -Naur mod_wsgi-4.9.4/setup.py mod_wsgi-4.9.4.new/setup.py
--- mod_wsgi-4.9.4/setup.py	2022-09-12 03:56:13.000000000 +0200
+++ mod_wsgi-4.9.4.new/setup.py	2023-10-05 19:08:10.174423228 +0200
@@ -15,9 +15,9 @@
     from urllib import urlretrieve
 
 from setuptools import setup
-from distutils.core import Extension
-from distutils.sysconfig import get_config_var as get_python_config
-from distutils.sysconfig import get_python_lib
+from setuptools.extension import Extension
+from sysconfig import get_config_var as get_python_config
+from sysconfig import get_path as get_python_lib
 
 # First work out what all the available source code files are that should
 # be compiled.
@@ -293,7 +293,7 @@
 
     PYTHON_LIBDIR = get_python_config('LIBDIR')
     APXS_LIBDIR = get_apxs_config('LIBDIR')
-    PYTHON_CFGDIR =  get_python_lib(plat_specific=1, standard_lib=1) + '/config'
+    PYTHON_CFGDIR =  get_python_lib('platstdlib') + '/config'
 
     if PYTHON_LDVERSION and PYTHON_LDVERSION != PYTHON_VERSION:
         PYTHON_CFGDIR = '%s-%s' % (PYTHON_CFGDIR, PYTHON_LDVERSION)
@@ -402,7 +402,7 @@
             not get_python_config('PYTHONFRAMEWORK')):
         print(SHARED_LIBRARY_WARNING)
 
-# Now finally run distutils.
+# Now finally run setuptools.
 
 package_name = 'mod_wsgi'
 long_description = open('README.rst').read()
@@ -440,17 +440,11 @@
         'Operating System :: POSIX :: SunOS/Solaris',
         'Programming Language :: Python',
         'Programming Language :: Python :: Implementation :: CPython',
-        'Programming Language :: Python :: 2.6',
-        'Programming Language :: Python :: 2.7',
-        'Programming Language :: Python :: 3.3',
-        'Programming Language :: Python :: 3.4',
-        'Programming Language :: Python :: 3.5',
-        'Programming Language :: Python :: 3.6',
-        'Programming Language :: Python :: 3.7',
         'Programming Language :: Python :: 3.8',
         'Programming Language :: Python :: 3.9',
         'Programming Language :: Python :: 3.10',
         'Programming Language :: Python :: 3.11',
+        'Programming Language :: Python :: 3.12',
         'Topic :: Internet :: WWW/HTTP :: WSGI',
         'Topic :: Internet :: WWW/HTTP :: WSGI :: Server'
     ],
diff -Naur mod_wsgi-4.9.4/src/server/__init__.py mod_wsgi-4.9.4.new/src/server/__init__.py
--- mod_wsgi-4.9.4/src/server/__init__.py	2022-09-12 03:56:13.000000000 +0200
+++ mod_wsgi-4.9.4.new/src/server/__init__.py	2023-10-05 19:08:10.173423220 +0200
@@ -35,7 +35,6 @@
 
 try:
     import sysconfig
-    import distutils.sysconfig
 
     _py_soabi = sysconfig.get_config_var('SOABI')
 
@@ -63,7 +62,7 @@
     MOD_WSGI_SO = posixpath.join(posixpath.dirname(__file__), MOD_WSGI_SO)
 
 if not os.path.exists(MOD_WSGI_SO) and os.name == 'nt':
-    MOD_WSGI_SO = 'mod_wsgi%s' % distutils.sysconfig.get_config_var('EXT_SUFFIX')
+    MOD_WSGI_SO = 'mod_wsgi%s' % sysconfig.get_config_var('EXT_SUFFIX')
     MOD_WSGI_SO = os.path.join(os.path.dirname(__file__), MOD_WSGI_SO)
     MOD_WSGI_SO = MOD_WSGI_SO.replace('\\', '/')
 
diff -Naur mod_wsgi-4.9.4/tox.ini mod_wsgi-4.9.4.new/tox.ini
--- mod_wsgi-4.9.4/tox.ini	2022-09-12 03:56:13.000000000 +0200
+++ mod_wsgi-4.9.4.new/tox.ini	2023-10-05 19:08:10.174423228 +0200
@@ -1,13 +1,10 @@
 [tox]
-envlist = py26,py27,py33,py34,py35,py36,py37,py38,py39,py310,py311
+envlist = py38,py39,py310,py311,py312
 
 [gh-actions]
 python =
-    2.7: py27
-    3.5: py35
-    3.6: py36
-    3.7: py37
     3.8: py38
     3.9: py39
     3.10: py310
     3.11: py311
+    3.12: py312
