diff -Naur libssh-0.11.orig/Makefile.in libssh-0.11/Makefile.in
--- libssh-0.11.orig/Makefile.in	2005-03-04 20:54:59.000000000 +0100
+++ libssh-0.11/Makefile.in	2007-11-23 23:13:51.000000000 +0100
@@ -56,11 +56,11 @@
 	  || case "$(MFLAGS)" in *k*) fail=yes;; *) exit 1;; esac; \
 	done && test -z "$$fail"
 	$(top_srcdir)/mkinstalldirs $(incldir)/libssh
-	$(INSTALL) include/libssh/libssh.h $(incldir)/libssh/
-	$(INSTALL) include/libssh/config.h $(incldir)/libssh/
-	$(INSTALL) include/libssh/sftp.h $(incldir)/libssh/
-	$(INSTALL) include/libssh/crypto.h $(incldir)/libssh/
-	$(INSTALL) include/libssh/server.h $(incldir)/libssh/
+	$(INSTALL) -m644 include/libssh/libssh.h $(incldir)/libssh/
+	$(INSTALL) -m644 include/libssh/config.h $(incldir)/libssh/
+	$(INSTALL) -m644 include/libssh/sftp.h $(incldir)/libssh/
+	$(INSTALL) -m644 include/libssh/crypto.h $(incldir)/libssh/
+	$(INSTALL) -m644 include/libssh/server.h $(incldir)/libssh/
 clean:
 	/bin/rm -f *~ *.o ssh sftp
 	@for dir in ${subdirs}; do \
