--- mkinitcpio-ykfde-0.7.4/bin/Makefile	2018-01-22 14:27:00.000000000 +0100
+++ mkinitcpio-ykfde-0.7.4.bak/bin/Makefile	2018-01-28 17:52:00.567599040 +0100
@@ -11,10 +11,10 @@
 all: worker ykfde ykfde-cpio
 
 worker: worker.c ../config.h
-	$(CC) $(CFLAGS) $(CFLAGS_YUBIKEY) $(CFLAGS_SYSTEMD) $(LDFLAGS) -o worker worker.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o worker worker.c $(CFLAGS_YUBIKEY) $(CFLAGS_SYSTEMD)
 
 ykfde: ykfde.c ../config.h ../version.h
-	$(CC) $(CFLAGS) $(CFLAGS_YUBIKEY) $(CFLAGS_SYSTEMD) -lcryptsetup $(LDFLAGS) -o ykfde ykfde.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o ykfde ykfde.c $(CFLAGS_YUBIKEY) $(CFLAGS_SYSTEMD) -lcryptsetup
 
 ykfde-cpio: ykfde-cpio.c ../config.h ../version.h
 	$(CC) $(CFLAGS) -larchive $(LDFLAGS) -o ykfde-cpio ykfde-cpio.c
