From 63c166b19549ea41a015d4c2cb3b32d4517d6377 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Sat, 25 Nov 2017 22:19:27 +0100
Subject: [PATCH] link required libraries only

---
 bin/Makefile | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/bin/Makefile b/bin/Makefile
index 53e14bc..1af9e16 100644
--- a/bin/Makefile
+++ b/bin/Makefile
@@ -3,18 +3,18 @@ CC	:= gcc
 INSTALL	:= install
 RM	:= rm
 # flags
-CFLAGS	+= -std=gnu11 -O2 -fPIC -Wall -Werror
-CFLAGS	+= -liniparser -lkeyutils -lykpers-1 -lyubikey
-CFLAGS	+= $(shell pkg-config --cflags --libs libsystemd 2>/dev/null)
-LDFLAGS	+= -Wl,-z,now -Wl,-z,relro -pie
+CFLAGS		+= -std=gnu11 -O2 -fPIC -Wall -Werror
+CFLAGS_YUBIKEY	+= -liniparser -lkeyutils -lykpers-1 -lyubikey
+CFLAGS_SYSTEMD	+= $(shell pkg-config --cflags --libs libsystemd 2>/dev/null)
+LDFLAGS		+= -Wl,-z,now -Wl,-z,relro -pie
 
 all: worker ykfde ykfde-cpio
 
 worker: worker.c ../config.h
-	$(CC) $(CFLAGS) $(LDFLAGS) -o worker worker.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o worker worker.c $(CFLAGS_SYSTEMD) $(CFLAGS_YUBIKEY)
 
 ykfde: ykfde.c ../config.h ../version.h
-	$(CC) $(CFLAGS) -lcryptsetup $(LDFLAGS) -o ykfde ykfde.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o ykfde ykfde.c $(CFLAGS_SYSTEMD) $(CFLAGS_YUBIKEY) -lcryptsetup
 
 ykfde-cpio: ykfde-cpio.c ../config.h ../version.h
 	$(CC) $(CFLAGS) -larchive $(LDFLAGS) -o ykfde-cpio ykfde-cpio.c
