diff -Naur mkinitcpio-ykfde-0.6.4/bin/Makefile mkinitcpio-ykfde-0.6.4.new/bin/Makefile
--- mkinitcpio-ykfde-0.6.4/bin/Makefile	2016-07-09 21:57:21.000000000 +0200
+++ mkinitcpio-ykfde-0.6.4.new/bin/Makefile	2017-01-21 22:47:42.000000000 +0100
@@ -9,10 +9,10 @@
 all: ykfde ykfde-cpio
 
 ykfde: ykfde.c ../config.h ../version.h
-	$(CC) $(CFLAGS) -lcryptsetup -liniparser -lkeyutils -lykpers-1 -lyubikey $(LDFLAGS) -o ykfde ykfde.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o ykfde ykfde.c -lcryptsetup -liniparser -lkeyutils -lykpers-1 -lyubikey
 
 ykfde-cpio: ykfde-cpio.c ../config.h ../version.h
-	$(CC) $(CFLAGS) -larchive $(LDFLAGS) -o ykfde-cpio ykfde-cpio.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o ykfde-cpio ykfde-cpio.c -larchive
 
 install: ykfde ykfde-cpio
 	$(INSTALL) -D -m0755 ykfde $(DESTDIR)/usr/bin/ykfde
diff -Naur mkinitcpio-ykfde-0.6.4/udev/Makefile mkinitcpio-ykfde-0.6.4.new/udev/Makefile
--- mkinitcpio-ykfde-0.6.4/udev/Makefile	2016-07-09 21:57:21.000000000 +0200
+++ mkinitcpio-ykfde-0.6.4.new/udev/Makefile	2017-01-21 22:48:16.000000000 +0100
@@ -9,7 +9,7 @@
 all: ykfde
 
 ykfde: ykfde.c ../config.h
-	$(CC) $(CFLAGS) -liniparser -lkeyutils -lykpers-1 -lyubikey $(LDFLAGS) -o ykfde ykfde.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o ykfde ykfde.c -liniparser -lkeyutils -lykpers-1 -lyubikey
 
 install: ykfde
 	$(INSTALL) -D -m0755 ykfde $(DESTDIR)/usr/lib/udev/ykfde
