diff -ur a/libs/libmyth/mythcontext.cpp b/libs/libmyth/mythcontext.cpp
--- a/libs/libmyth/mythcontext.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/libs/libmyth/mythcontext.cpp	2012-07-11 13:59:01.978233942 -0500
@@ -10,7 +10,7 @@
 #include <QMutex>
 
 #include <cmath>
-
+#include <unistd.h>
 #include <queue>
 #include <algorithm>
 using namespace std;
diff -ur a/libs/libmyth/programinfoupdater.cpp b/libs/libmyth/programinfoupdater.cpp
--- a/libs/libmyth/programinfoupdater.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/libs/libmyth/programinfoupdater.cpp	2012-07-11 13:59:14.917924280 -0500
@@ -1,3 +1,4 @@
+#include <unistd.h>
 #include "programinfoupdater.h"
 #include "remoteutil.h"
 #include "compat.h"
diff -ur a/libs/libmythdb/httpcomms.cpp b/libs/libmythdb/httpcomms.cpp
--- a/libs/libmythdb/httpcomms.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/libs/libmythdb/httpcomms.cpp	2012-07-11 13:58:45.223634881 -0500
@@ -5,6 +5,7 @@
 #include <QRegExp>
 #include <QTimer>
 #include <QFile>
+#include <unistd.h>
 
 #include "mythverbose.h"
 #include "compat.h"
diff -ur a/libs/libmythdb/lcddevice.cpp b/libs/libmythdb/lcddevice.cpp
--- a/libs/libmythdb/lcddevice.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/libs/libmythdb/lcddevice.cpp	2012-07-11 13:58:45.223634881 -0500
@@ -36,6 +36,7 @@
 #include <QTextStream>
 #include <QTextCodec>
 #include <QByteArray>
+#include <unistd.h>
 
 // MythTV headers
 #include "lcddevice.h"
diff -ur a/libs/libmythdb/mythdbcon.cpp b/libs/libmythdb/mythdbcon.cpp
--- a/libs/libmythdb/mythdbcon.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/libs/libmythdb/mythdbcon.cpp	2012-07-11 13:58:45.224634858 -0500
@@ -8,6 +8,7 @@
 #include <QSqlError>
 #include <QSqlField>
 #include <QSqlRecord>
+#include <unistd.h>
 
 // MythTV
 #include "compat.h"
diff -ur a/libs/libmythdb/mythsocket.cpp b/libs/libmythdb/mythsocket.cpp
--- a/libs/libmythdb/mythsocket.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/libs/libmythdb/mythsocket.cpp	2012-07-11 13:58:45.225634834 -0500
@@ -8,6 +8,7 @@
 // POSIX
 #ifndef USING_MINGW
 #include <sys/select.h> // for select
+#include <unistd.h>
 #endif
 
 // Qt
diff -ur a/libs/libmythdb/mythsocketthread.cpp b/libs/libmythdb/mythsocketthread.cpp
--- a/libs/libmythdb/mythsocketthread.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/libs/libmythdb/mythsocketthread.cpp	2012-07-11 13:58:45.225634834 -0500
@@ -3,6 +3,7 @@
 
 // C++
 #include <algorithm> // for min/max
+#include <unistd.h>
 //using namespace std;
 
 #include "compat.h"
diff -ur a/libs/libmythdb/remotefile.cpp b/libs/libmythdb/remotefile.cpp
--- a/libs/libmythdb/remotefile.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/libs/libmythdb/remotefile.cpp	2012-07-11 13:58:45.225634834 -0500
@@ -2,6 +2,7 @@
 using namespace std;
 
 #include <QUrl>
+#include <unistd.h>
 
 #include "mythdb.h"
 #include "remotefile.h"
diff -ur a/libs/libmythtv/audioinputoss.cpp b/libs/libmythtv/audioinputoss.cpp
--- a/libs/libmythtv/audioinputoss.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/libs/libmythtv/audioinputoss.cpp	2012-07-11 13:59:44.401218758 -0500
@@ -31,6 +31,7 @@
 #include "mythcontext.h"
 #include <fcntl.h>
 #include <sys/ioctl.h>
+#include <unistd.h>
 
 #define LOC     QString("AudioInOSS: ")
 #define LOC_DEV QString("AudioInOSS(%1): ").arg(m_device_name.constData())
diff -ur a/libs/libmythtv/previewgenerator.cpp b/libs/libmythtv/previewgenerator.cpp
--- a/libs/libmythtv/previewgenerator.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/libs/libmythtv/previewgenerator.cpp	2012-07-11 13:59:28.859590672 -0500
@@ -4,6 +4,7 @@
 // POSIX headers
 #include <sys/time.h>
 #include <fcntl.h>
+#include <unistd.h>
 
 // Qt headers
 #include <QFileInfo>
diff -ur a/libs/libmythupnp/upnpdevice.cpp b/libs/libmythupnp/upnpdevice.cpp
--- a/libs/libmythupnp/upnpdevice.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/libs/libmythupnp/upnpdevice.cpp	2012-07-11 13:58:45.225634834 -0500
@@ -29,6 +29,7 @@
 
 #include <QFile>
 #include <QTextStream>
+#include <unistd.h>
 
 int DeviceLocation::g_nAllocated   = 0;       // Debugging only
 
diff -ur a/libs/libmythupnp/upnptasknotify.cpp b/libs/libmythupnp/upnptasknotify.cpp
--- a/libs/libmythupnp/upnptasknotify.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/libs/libmythupnp/upnptasknotify.cpp	2012-07-11 13:58:45.225634834 -0500
@@ -28,6 +28,7 @@
 #include <QStringList>
 #include <QUuid> 
 #include <QFile>
+#include <unistd.h>
 
 // MythTV headers
 #include "upnp.h"
diff -ur a/libs/libmythupnp/upnptasksearch.cpp b/libs/libmythupnp/upnptasksearch.cpp
--- a/libs/libmythupnp/upnptasksearch.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/libs/libmythupnp/upnptasksearch.cpp	2012-07-11 13:58:45.226634809 -0500
@@ -23,6 +23,7 @@
 
 #include <compat.h>
 #include <stdlib.h>
+#include <unistd.h>
 
 #include <QStringList>
 #include <QFile>
diff -ur a/programs/mythfrontend/backendconnectionmanager.cpp b/programs/mythfrontend/backendconnectionmanager.cpp
--- a/programs/mythfrontend/backendconnectionmanager.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/programs/mythfrontend/backendconnectionmanager.cpp	2012-07-11 13:59:58.456882410 -0500
@@ -4,6 +4,7 @@
 #include <QString>
 #include <QEvent>
 #include <QTimer>
+#include <unistd.h>
 
 #include "mythcontext.h"
 #include "exitcodes.h"
diff -ur a/programs/mythtranscode/transcode.cpp b/programs/mythtranscode/transcode.cpp
--- a/programs/mythtranscode/transcode.cpp	2012-07-11 12:08:38.000000000 -0500
+++ b/programs/mythtranscode/transcode.cpp	2012-07-11 14:00:10.472594873 -0500
@@ -1,5 +1,6 @@
 #include <fcntl.h>
 #include <math.h>
+#include <unistd.h>
 
 using namespace std;
 
