--- poppler-0.24.1/configure.old	2014-08-14 19:44:17.015262710 +0200
+++ poppler-0.24.1/configure	2014-08-14 19:46:56.487530815 +0200
@@ -23296,9 +23296,9 @@
 
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking for qtchooser" >&5
 $as_echo_n "checking for qtchooser... " >&6; }
-      qt5tooldir=`QT_SELECT=qt5 qtchooser -print-env | grep QTTOOLDIR | cut -d '=' -f 2 | cut -d \" -f 2`
+      qt5tooldir="/usr/lib/qt5/bin"
       mocversion=`$qt5tooldir/moc -v 2>&1`
-      mocversiongrep=`echo $mocversion | grep "Qt 5"`
+      mocversiongrep=`echo $mocversion | grep "moc"`
       if test x"$mocversiongrep" != x"$mocversion"; then
         # no valid moc found
         enable_poppler_qt5=no;
