diff -ru squid-2.5.STABLE9/helpers/basic_auth/SMB/Makefile.in squid-2.5.STABLE9.arch/helpers/basic_auth/SMB/Makefile.in
--- squid-2.5.STABLE9/helpers/basic_auth/SMB/Makefile.in	2004-07-10 08:11:41.000000000 -0400
+++ squid-2.5.STABLE9.arch/helpers/basic_auth/SMB/Makefile.in	2005-04-15 06:08:13.000000000 -0400
@@ -128,7 +128,7 @@
 makesnmplib = @makesnmplib@
 
 SMB_AUTH_HELPER = smb_auth.sh
-SAMBAPREFIX = /usr/local/samba
+SAMBAPREFIX = /usr
 SMB_AUTH_HELPER_PATH = $(libexecdir)/$(SMB_AUTH_HELPER)
 
 libexec_SCRIPTS = $(SMB_AUTH_HELPER)
