diff -Naur opencv-3.1.0/cmake/OpenCVPCHSupport.cmake opencv-3.1.0-p/cmake/OpenCVPCHSupport.cmake
--- opencv-3.1.0/cmake/OpenCVPCHSupport.cmake	2015-12-18 16:02:16.000000000 +0100
+++ opencv-3.1.0-p/cmake/OpenCVPCHSupport.cmake	2016-09-24 20:36:38.212533750 +0200
@@ -19,7 +19,7 @@
         ARGS ${CMAKE_CXX_COMPILER_ARG1} -dumpversion
         OUTPUT_VARIABLE gcc_compiler_version)
     #MESSAGE("GCC Version: ${gcc_compiler_version}")
-    IF(gcc_compiler_version VERSION_GREATER "4.2.-1")
+    IF(gcc_compiler_version VERSION_GREATER "4.2.-1" AND gcc_compiler_version VERSION_LESS "6.0.0")
         SET(PCHSupport_FOUND TRUE)
     ENDIF()
 
