diff -Naur opencv-3.2.0/cmake/OpenCVCompilerOptions.cmake opencv-3.2.0-p/cmake/OpenCVCompilerOptions.cmake
--- opencv-3.2.0/cmake/OpenCVCompilerOptions.cmake	2016-12-23 13:54:44.000000000 +0100
+++ opencv-3.2.0-p/cmake/OpenCVCompilerOptions.cmake	2017-10-17 13:02:25.410817394 +0200
@@ -18,12 +18,10 @@
         message(STATUS "Unable to compile program with enabled ccache, reverting...")
         set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "${__OLD_RULE_LAUNCH_COMPILE}")
       endif()
-    else()
       message(STATUS "Looking for ccache - not found")
     endif()
   endif()
 endif()
-
 if((CMAKE_COMPILER_IS_CLANGCXX OR CMAKE_COMPILER_IS_CLANGCC OR CMAKE_COMPILER_IS_CCACHE) AND NOT CMAKE_GENERATOR MATCHES "Xcode")
   set(ENABLE_PRECOMPILED_HEADERS OFF CACHE BOOL "" FORCE)
 endif()
