diff -Naur opencv-2.4.10/modules/core/src/system.cpp opencv-2.4.10.new/modules/core/src/system.cpp
--- opencv-2.4.10/modules/core/src/system.cpp	2014-10-01 09:33:36.000000000 +0200
+++ opencv-2.4.10.new/modules/core/src/system.cpp	2014-10-23 09:13:25.099724441 +0200
@@ -267,14 +267,17 @@
          : "cc"
         );
         #else
+        // We need to preserve ebx since we are compiling PIC code.
+        // This means we cannot use "=b" for the 2nd output register.
         asm volatile
         (
          "pushl %%ebx\n\t"
          "movl $7,%%eax\n\t"
          "movl $0,%%ecx\n\t"
          "cpuid\n\t"
+	 "movl %%ebx,%1\n\t"
          "popl %%ebx\n\t"
-         : "=a"(cpuid_data[0]), "=b"(cpuid_data[1]), "=c"(cpuid_data[2]), "=d"(cpuid_data[3])
+         : "=a"(cpuid_data[0]), "=r"(cpuid_data[1]), "=c"(cpuid_data[2]), "=d"(cpuid_data[3])
          :
          : "cc"
         );
