diff -Naur SVK-v2.0.1/Makefile.PL SVK-v2.0.1-p/Makefile.PL
--- SVK-v2.0.1/Makefile.PL	2007-03-26 19:17:42.000000000 +0200
+++ SVK-v2.0.1-p/Makefile.PL	2007-04-20 13:29:31.000000000 +0200
@@ -84,7 +84,7 @@
     'UNIVERSAL::require'       => '0',
     'Term::ReadKey'            => '0',
     'Time::HiRes'              => '0',       # in core since 5.7.3
-    'File::Temp'               => '0.17',
+    'File::Temp'               => '0.16',
     'Encode'                   => '2.10',    # in core since 5.8.7
     'Getopt::Long'             => '2.35',    # in core since 5.8.8
     'Pod::Escapes'             => '0',       # in core since 5.9.3
