diff -Naur weechat-1.9/src/plugins/python/CMakeLists.txt weechat-1.9-p/src/plugins/python/CMakeLists.txt
--- weechat-1.9/src/plugins/python/CMakeLists.txt	2017-06-25 10:20:52.000000000 +0200
+++ weechat-1.9-p/src/plugins/python/CMakeLists.txt	2017-07-24 15:58:51.811594269 +0200
@@ -24,8 +24,9 @@
 
 if(PYTHON_FOUND)
   include_directories(${PYTHON_INCLUDE_PATH})
-  set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} ${PYTHON_LFLAGS}")
+  set(CMAKE_MODULE_LINKER_FLAGS "${PYTHON_LFLAGS} -ldl -lutil -Wl,--hash-style=both -Wl,-O1 -Wl,--no-as-needed")
   target_link_libraries(python ${PYTHON_LIBRARY} weechat_plugins_scripts)
 endif()
 
+
 install(TARGETS python LIBRARY DESTINATION ${LIBDIR}/plugins)
