diff -aur weechat-0.2.6/src/common/weeconfig.c fw_weechat/src/common/weeconfig.c
--- weechat-0.2.6/src/common/weeconfig.c	2007-08-29 14:47:50.000000000 +0200
+++ fw_weechat/src/common/weeconfig.c	2007-09-07 08:28:22.000000000 +0200
@@ -70,6 +70,7 @@
 int cfg_look_set_title;
 int cfg_look_startup_logo;
 int cfg_look_startup_version;
+int cfg_look_highlight_beep;
 char *cfg_look_weechat_slogan;
 int cfg_look_one_server_buffer;
 int cfg_look_scroll_amount;
@@ -135,6 +136,10 @@
     N_("display WeeChat version at startup"),
     OPTION_TYPE_BOOLEAN, BOOL_FALSE, BOOL_TRUE, BOOL_TRUE,
     NULL, NULL, &cfg_look_startup_version, NULL, config_change_noop },
+  { "look_highlight_beep", N_("console beep on highlight"),
+    N_("console beep on highlight"),
+    OPTION_TYPE_BOOLEAN, BOOL_FALSE, BOOL_TRUE, BOOL_TRUE,
+    NULL, NULL, &cfg_look_highlight_beep, NULL, config_change_noop },
   { "look_weechat_slogan", N_("WeeChat slogan"),
     N_("WeeChat slogan (if empty, slogan is not used)"),
     OPTION_TYPE_STRING, 0, 0, 0,
diff -aur weechat-0.2.6/src/common/weeconfig.h fw_weechat/src/common/weeconfig.h
--- weechat-0.2.6/src/common/weeconfig.h	2007-08-27 09:59:09.000000000 +0200
+++ fw_weechat/src/common/weeconfig.h	2007-09-07 08:28:50.000000000 +0200
@@ -97,6 +97,7 @@
 extern int cfg_look_set_title;
 extern int cfg_look_startup_logo;
 extern int cfg_look_startup_version;
+extern int cfg_look_highlight_beep;
 extern char *cfg_look_weechat_slogan;
 extern int cfg_look_one_server_buffer;
 extern int cfg_look_open_near_server;
diff -aur weechat-0.2.6/src/gui/gui-common.c fw_weechat/src/gui/gui-common.c
--- weechat-0.2.6/src/gui/gui-common.c	2007-08-31 15:25:30.000000000 +0200
+++ fw_weechat/src/gui/gui-common.c	2007-09-07 08:31:43.000000000 +0200
@@ -163,8 +163,11 @@
     buffer->last_line->length += length;
     if (type & GUI_MSG_TYPE_MSG)
         buffer->last_line->line_with_message = 1;
-    if (type & GUI_MSG_TYPE_HIGHLIGHT)
+    if (type & GUI_MSG_TYPE_HIGHLIGHT) {
         buffer->last_line->line_with_highlight = 1;
+	if (cfg_look_highlight_beep)
+		printf("\a");
+    }
     if ((type & GUI_MSG_TYPE_TIME) || (type & GUI_MSG_TYPE_NICK) || (type & GUI_MSG_TYPE_PREFIX))
         buffer->last_line->length_align += length;
     if (type & GUI_MSG_TYPE_NOLOG)
