diff -Naur xf86-input-mouse-1.0.3.1.orig/src/mouse.c xf86-input-mouse-1.0.3.1/src/mouse.c
--- xf86-input-mouse-1.0.3.1.orig/src/mouse.c	2005-12-15 01:21:33.000000000 +0100
+++ xf86-input-mouse-1.0.3.1/src/mouse.c	2006-04-20 14:40:34.000000000 +0200
@@ -1497,7 +1497,14 @@
 		      (pBuf[3] & 0x20) >> 1;        /* button 5 */
 	    dx = (pBuf[0] & 0x10) ?    pBuf[1]-256  :  pBuf[1];
 	    dy = (pBuf[0] & 0x20) ?  -(pBuf[2]-256) : -pBuf[2];
-	    dz = (pBuf[3] & 0x08) ? (pBuf[3] & 0x0f) - 16 : (pBuf[3] & 0x0f);
+            dz = (pBuf[3] & 0x01) ? ((pBuf[3] & 0x08) ? -1 : 1) : 0;
+            dw = (pBuf[3] & 0x02) ? ((pBuf[3] & 0x08) ? -1 : 1) : 0;
+            switch (pBuf[3] & 0x0f) {
+	          case 0x01: dz =  1; break;
+	          case 0x02: dw =  1; break;
+	          case 0x0e: dw = -1; break;
+	          case 0x0f: dz = -1; break;
+	        }
 	    break;
 
 	case PROT_MMPS2:	/* MouseMan+ PS/2 */
