diff -Naur synaptics-0.14.5.orig/eventcomm.c synaptics-0.14.5/eventcomm.c
--- synaptics-0.14.5.orig/eventcomm.c	2006-06-04 01:30:00.000000000 +0200
+++ synaptics-0.14.5/eventcomm.c	2006-06-10 19:53:54.000000000 +0200
@@ -21,6 +21,10 @@
 #include "synproto.h"
 #include "synaptics.h"
 #include <xf86.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
 
 
 #define SYSCALL(call) while (((call) == -1) && (errno == EINTR))
diff -Naur synaptics-0.14.5.orig/ps2comm.h synaptics-0.14.5/ps2comm.h
--- synaptics-0.14.5.orig/ps2comm.h	2006-04-16 21:31:43.000000000 +0200
+++ synaptics-0.14.5/ps2comm.h	2006-06-10 19:40:25.000000000 +0200
@@ -1,6 +1,7 @@
 #ifndef _PS2COMM_H_
 #define _PS2COMM_H_
 
+#include <unistd.h>
 #include <sys/ioctl.h>
 #include "xf86_OSproc.h"
 
diff -Naur synaptics-0.14.5.orig/psmcomm.c synaptics-0.14.5/psmcomm.c
--- synaptics-0.14.5.orig/psmcomm.c	2006-04-16 21:31:43.000000000 +0200
+++ synaptics-0.14.5/psmcomm.c	2006-06-10 19:53:49.000000000 +0200
@@ -33,6 +33,7 @@
 #include "synaptics.h"
 #include "ps2comm.h"			    /* ps2_print_ident() */
 #include <xf86.h>
+#include <errno.h>
 
 #define SYSCALL(call) while (((call) == -1) && (errno == EINTR))
 
diff -Naur synaptics-0.14.5.orig/psmcomm.h synaptics-0.14.5/psmcomm.h
--- synaptics-0.14.5.orig/psmcomm.h	2006-04-16 21:31:43.000000000 +0200
+++ synaptics-0.14.5/psmcomm.h	2006-06-10 19:40:46.000000000 +0200
@@ -1,6 +1,7 @@
 #ifndef _PSMCOMM_H_
 #define _PSMCOMM_H_
 
+#include <unistd.h>
 #include <sys/ioctl.h>
 #include "freebsd_mouse.h"
 
diff -Naur synaptics-0.14.5.orig/synaptics.c synaptics-0.14.5/synaptics.c
--- synaptics-0.14.5.orig/synaptics.c	2006-06-04 01:30:00.000000000 +0200
+++ synaptics-0.14.5/synaptics.c	2006-06-10 19:41:12.000000000 +0200
@@ -59,6 +59,7 @@
  *	Standard Headers
  ****************************************************************************/
 
+#include <unistd.h>
 #include <sys/ioctl.h>
 #include <misc.h>
 #include <xf86.h>
diff -Naur synaptics-0.14.5.orig/synproto.h synaptics-0.14.5/synproto.h
--- synaptics-0.14.5.orig/synproto.h	2006-04-16 21:31:43.000000000 +0200
+++ synaptics-0.14.5/synproto.h	2006-06-10 19:41:35.000000000 +0200
@@ -19,6 +19,7 @@
 #ifndef _SYNPROTO_H_
 #define _SYNPROTO_H_
 
+#include <unistd.h>
 #include <sys/ioctl.h>
 #include <xf86Xinput.h>
 #include <xisb.h>
