diff -up epiphany-2.91.2.1/data/org.gnome.epiphany.gschema.xml.in.homepage epiphany-2.91.2.1/data/org.gnome.epiphany.gschema.xml.in
--- epiphany-2.91.2.1/data/org.gnome.epiphany.gschema.xml.in.homepage	2010-11-11 19:05:59.789142001 -0500
+++ epiphany-2.91.2.1/data/org.gnome.epiphany.gschema.xml.in	2010-11-11 19:06:19.026142001 -0500
@@ -10,7 +10,7 @@
 			<summary>Browse with caret</summary>
 		</key>
 		<key type="s" name="homepage-url">
-			<default>'http://www.google.com'</default>
+			<default>'http://www.frugalware.org/'</default>
 			<summary>Home page</summary>
 			<description>Address of the user's home page.</description>
 		</key>

