diff -Naur kadu-0.12.2/plugins/jabber_protocol/3rdparty/CMakeLists.txt kadu-0.12.2.new/plugins/jabber_protocol/3rdparty/CMakeLists.txt
--- kadu-0.12.2/plugins/jabber_protocol/3rdparty/CMakeLists.txt	2012-08-30 16:13:17.000000000 +0200
+++ kadu-0.12.2.new/plugins/jabber_protocol/3rdparty/CMakeLists.txt	2013-12-13 09:34:20.110926728 +0100
@@ -26,7 +26,7 @@
 					get_filename_component (_basename ${_current_MOC} NAME_WE)
 					set (_header ${_abs_FILE})
 					set (_moc ${_abs_PATH}/${_current_MOC})
-					QT4_CREATE_MOC_COMMAND (${_header} ${_moc} "${_moc_INCS}" "")
+					QT4_CREATE_MOC_COMMAND (${_header} ${_moc} "${_moc_INCS}" "" "${moc_target}")
 					MACRO_ADD_FILE_DEPENDENCIES (${_abs_FILE} ${_moc})
 				endforeach (_current_MOC_INC)
 			endif (_match)
