diff -Naur enlightenment-0.21.6/src/modules/wl_drm/e_mod_main.c enlightenment-0.21.6-p/src/modules/wl_drm/e_mod_main.c
--- enlightenment-0.21.6/src/modules/wl_drm/e_mod_main.c	2017-02-15 04:09:40.000000000 +0100
+++ enlightenment-0.21.6-p/src/modules/wl_drm/e_mod_main.c	2017-02-21 21:22:04.153089646 +0100
@@ -502,26 +502,30 @@
                          s->config.geom.w, s->config.geom.h);
                }
 
+             /* TODO: cannot support rotations until we support planes
+              * and we cannot support planes until Atomic support is in */
+
              s->info.can_rot_0 = EINA_FALSE;
              s->info.can_rot_90 = EINA_FALSE;
              s->info.can_rot_180 = EINA_FALSE;
              s->info.can_rot_270 = EINA_FALSE;
 
-# if (EFL_VERSION_MAJOR > 1) || (EFL_VERSION_MINOR >= 18)
-             int rotations;
-
-             rotations =
-               ecore_drm2_output_supported_rotations_get(output);
+/* # if (EFL_VERSION_MAJOR > 1) || (EFL_VERSION_MINOR >= 18) */
+/*              unsigned int rotations; */
 
-             if (rotations & ECORE_DRM2_ROTATION_NORMAL)
-               s->info.can_rot_0 = EINA_TRUE;
-             if (rotations & ECORE_DRM2_ROTATION_90)
-               s->info.can_rot_90 = EINA_TRUE;
-             if (rotations & ECORE_DRM2_ROTATION_180)
-               s->info.can_rot_180 = EINA_TRUE;
-             if (rotations & ECORE_DRM2_ROTATION_270)
-               s->info.can_rot_270 = EINA_TRUE;
-# endif
+/*              rotations = */
+/*                ecore_drm_output_supported_rotations_get(output, */
+/*                                                         ECORE_DRM_PLANE_TYPE_PRIMARY); */
+
+/*              if (rotations & ECORE_DRM_PLANE_ROTATION_NORMAL) */
+/*                s->info.can_rot_0 = EINA_TRUE; */
+/*              if (rotations & ECORE_DRM_PLANE_ROTATION_90) */
+/*                s->info.can_rot_90 = EINA_TRUE; */
+/*              if (rotations & ECORE_DRM_PLANE_ROTATION_180) */
+/*                s->info.can_rot_180 = EINA_TRUE; */
+/*              if (rotations & ECORE_DRM_PLANE_ROTATION_270) */
+/*                s->info.can_rot_270 = EINA_TRUE; */
+/* # endif */
 
              if (cs)
                {
@@ -679,20 +683,6 @@
 
         /* TODO: cannot support rotations until we support planes
          * and we cannot support planes until Atomic support is in */
-# if (EFL_VERSION_MAJOR > 1) || (EFL_VERSION_MINOR >= 18)
-        int orient;
-
-        if (s->config.rotation == 0)
-          orient = ECORE_DRM2_ROTATION_NORMAL;
-        else if (s->config.rotation == 90)
-          orient = ECORE_DRM2_ROTATION_90;
-        else if (s->config.rotation == 180)
-          orient = ECORE_DRM2_ROTATION_180;
-        else if (s->config.rotation == 270)
-          orient = ECORE_DRM2_ROTATION_270;
-
-        ecore_drm2_output_rotation_set(output, orient);
-# endif
 
         if (s->config.priority == top_priority)
           _drm2_output_primary_set(outputs, output);
