diff -Naur enlightenment-0.21.6/src/bin/e_alert_main.c enlightenment-0.21.6-p2/src/bin/e_alert_main.c
--- enlightenment-0.21.6/src/bin/e_alert_main.c	2017-02-15 04:09:40.000000000 +0100
+++ enlightenment-0.21.6-p2/src/bin/e_alert_main.c	2017-02-21 21:30:11.000968887 +0100
@@ -488,17 +488,7 @@
    updates = evas_render_updates(canvas);
    evas_render_updates_free(updates);
 
-#ifdef EFL_VERSION_1_19
-   ecore_drm2_fb_flip(buffer, output);
-#else
-   if (E_EFL_VERSION_MINIMUM(1, 18, 99))
-     {
-        void (*fn)(void*, void*) = dlsym(NULL, "ecore_drm2_fb_flip");
-        if (fn) fn(buffer, output);
-     }
-   else
-     ecore_drm2_fb_flip(buffer, output, NULL);
-#endif
+   ecore_drm2_fb_flip(buffer, output, NULL);
 }
 
 static void
diff -Naur enlightenment-0.21.6/src/modules/wl_drm/e_mod_main.c enlightenment-0.21.6-p2/src/modules/wl_drm/e_mod_main.c
--- enlightenment-0.21.6/src/modules/wl_drm/e_mod_main.c	2017-02-15 04:09:40.000000000 +0100
+++ enlightenment-0.21.6-p2/src/modules/wl_drm/e_mod_main.c	2017-02-21 21:30:11.000968887 +0100
@@ -798,27 +798,12 @@
    out = ecore_drm2_output_find(dev, output->x, output->y);
    if (!out) return;
 
-#ifdef EFL_VERSION_1_19
-   fb = ecore_drm2_output_latest_fb_get(out);
-   if (!fb) return;
-#else
-   if (E_EFL_VERSION_MINIMUM(1, 18, 99))
+   fb = ecore_drm2_output_next_fb_get(out);
+   if (!fb)
      {
-        void *(*fn)(void*) = dlsym(NULL, "ecore_drm2_output_latest_fb_get");
-        if (!fn) return;
-        fb = fn(out);
+        fb = ecore_drm2_output_current_fb_get(out);
         if (!fb) return;
      }
-   else
-     {
-        fb = ecore_drm2_output_next_fb_get(out);
-        if (!fb)
-          {
-             fb = ecore_drm2_output_current_fb_get(out);
-             if (!fb) return;
-          }
-     }
-#endif
 
    data = ecore_drm2_fb_data_get(fb);
    fstride = ecore_drm2_fb_stride_get(fb);
