diff -Naur libtunepimp-0.4.2/lib/lookuptools.cpp libtunepimp-0.4.2-buffer-overflow/lib/lookuptools.cpp
--- libtunepimp-0.4.2/lib/lookuptools.cpp	2006-01-28 21:35:42.000000000 +0100
+++ libtunepimp-0.4.2-buffer-overflow/lib/lookuptools.cpp	2006-07-13 22:40:34.972957272 +0200
@@ -51,7 +51,7 @@
     musicbrainz_t  o;
     char          *args[7];
     int            ret, trackNum;
-    char           error[255], data[255], trackURI[256],
+    char           error[256], data[256], trackURI[256],
                    artistURI[256], albumURI[256];
     char           temp[100], duration[100], status[100];
 
@@ -160,7 +160,7 @@
             if (mb_Select1(o, MBS_SelectReleaseDate, j))
             {
                 // Pull back the release date and release country
-                if (mb_GetResultData(o, MBE_ReleaseGetDate, temp, 256))
+                if (mb_GetResultData(o, MBE_ReleaseGetDate, temp, 100))
                 {
                     int month = 0, day = 0, year = 0;
 
