diff -Naur libtunepimp-0.3.0/lib/threads/posix/thread.cpp libtunepimp-0.3.0-patched/lib/threads/posix/thread.cpp
--- libtunepimp-0.3.0/lib/threads/posix/thread.cpp	2004-01-11 10:27:52.000000000 +0100
+++ libtunepimp-0.3.0-patched/lib/threads/posix/thread.cpp	2005-10-26 06:01:38.000000000 +0200
@@ -29,6 +29,7 @@
 #include <string.h>
 #include <sched.h>                                                                                              
 #include "thread.h"
+#include <signal.h>
 
 #define DB printf("%s:%d\n", __FILE__, __LINE__);
 
