diff -Naur libmpeg3-1.5.4.orig/Makefile.am libmpeg3-1.5.4/Makefile.am
--- libmpeg3-1.5.4.orig/Makefile.am	1970-01-01 01:00:00.000000000 +0100
+++ libmpeg3-1.5.4/Makefile.am	2005-07-29 15:56:05.000000000 +0200
@@ -0,0 +1,102 @@
+NASM = sh ./strip_fPIC.sh nasm
+
+if USE_MMX
+ASMSRC = video/mmxidct.S
+NASMSRC = video/reconmmx.s
+NASMOBJ = reconmmx.lo
+else
+ASMSRC =
+NASMSRC =
+NASMOBJ =
+endif
+
+SRC = \
+	audio/ac3.c \
+	audio/dct.c \
+	audio/huffman.c \
+	audio/layer2.c \
+	audio/layer3.c \
+	audio/mpeg3audio.c \
+	audio/pcm.c \
+	audio/synthesizers.c \
+	audio/tables.c \
+	bitstream.c \
+	libmpeg3.c \
+	mpeg3atrack.c \
+	mpeg3css.c \
+	mpeg3demux.c \
+	mpeg3ifo.c \
+	mpeg3io.c \
+	mpeg3title.c \
+	mpeg3vtrack.c \
+	video/getpicture.c \
+	video/headers.c \
+	video/idct.c \
+	video/macroblocks.c \
+	video/mmxtest.c \
+	video/motion.c \
+	video/mpeg3video.c \
+	video/output.c \
+	video/reconstruct.c \
+	video/seek.c \
+	video/slice.c \
+	video/vlc.c \
+	workarounds.c
+
+mincdir = $(includedir)/libmpeg3
+maincdir = $(mincdir)/audio
+mvincdir = $(mincdir)/video
+
+minc_HEADERS = \
+	bitstream.h \
+	ifo.h \
+	libmpeg3.h \
+	mpeg3atrack.h \
+	mpeg3css.h \
+	mpeg3demux.h \
+	mpeg3io.h \
+	mpeg3private.h \
+	mpeg3protos.h \
+	mpeg3title.h \
+	mpeg3vtrack.h \
+	timecode.h \
+	workarounds.h
+
+mainc_HEADERS = \
+	audio/ac3.h \
+	audio/huffman.h \
+	audio/mpeg3audio.h \
+	audio/tables.h
+
+mvinc_HEADERS = \
+	video/idct.h \
+	video/layerdata.h \
+	video/mpeg3video.h \
+	video/mpeg3videoprotos.h \
+	video/slice.h \
+	video/vlc.h
+
+lib_LTLIBRARIES = libmpeg3.la
+
+bin_PROGRAMS = mpeg3dump mpeg3cat mpeg3toc mpeg3split
+
+LIBS = -lm -lpthread
+
+libmpeg3_la_SOURCES = $(ASMSRC) $(SRC)
+libmpeg3_la_LIBADD = $(NASMOBJ) -la52 -lmpeg2
+libmpeg3_la_LDFLAGS = -version-info 1:5:0
+
+mpeg3dump_SOURCES = mpeg3dump.c
+mpeg3dump_LDADD = libmpeg3.la
+
+mpeg3cat_SOURCES = mpeg3cat.c
+mpeg3cat_LDADD = libmpeg3.la
+
+mpeg3toc_SOURCES = mpeg3toc.c
+mpeg3toc_LDADD = libmpeg3.la
+
+mpeg3split_SOURCES = mpeg3split.c
+mpeg3split_LDADD = libmpeg3.la
+
+reconmmx.lo: $(NASMSRC)
+	$(LIBTOOL) --mode=compile --tag=$(NASM) -f elf $(NASMSRC) -o $(NASMOBJ)
diff -Naur libmpeg3-1.5.4.orig/bitstream.h libmpeg3-1.5.4/bitstream.h
--- libmpeg3-1.5.4.orig/bitstream.h	2005-07-29 15:55:57.000000000 +0200
+++ libmpeg3-1.5.4/bitstream.h	2005-07-29 15:56:05.000000000 +0200
@@ -1,7 +1,7 @@
 #ifndef BITSTREAM_H
 #define BITSTREAM_H
 
-#include "mpeg3demux.h"
+#include <libmpeg3/mpeg3demux.h>
 #include <sys/types.h>
 
 #endif
diff -Naur libmpeg3-1.5.4.orig/configure.in libmpeg3-1.5.4/configure.in
--- libmpeg3-1.5.4.orig/configure.in	1970-01-01 01:00:00.000000000 +0100
+++ libmpeg3-1.5.4/configure.in	2005-07-29 15:56:05.000000000 +0200
@@ -0,0 +1,29 @@
+AC_INIT(libmpeg3.c)
+AM_INIT_AUTOMAKE(libmpeg3, 1.5.0)
+AC_CONFIG_HEADER([config.h])
+
+AC_PROG_CC
+AC_PROG_CPP
+AC_PROG_INSTALL
+AC_PROG_LIBTOOL
+AM_PROG_AS
+
+AC_ARG_ENABLE(mmx, AC_HELP_STRING([--enable-mmx],[use MMX]),
+	[use_mmx=$enableval], [use_mmx=no])
+
+AM_CONDITIONAL(USE_MMX, [test "$use_mmx" != "no"])
+if test "$use_mmx" != "no"; then
+	AC_DEFINE(HAVE_MMX,,[use MMX])
+fi
+
+AC_ARG_ENABLE(css, AC_HELP_STRING([--disable-css],[disable CSS]),
+	[use_css=$enableval], [use_css=yes])
+
+if test "$use_css" != "no"; then
+	AC_DEFINE(HAVE_CSS,,[enable CSS])
+fi
+
+CFLAGS="${CFLAGS} -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -I/usr/include/a52dec -I/usr/include/mpeg2dec"
+
+AC_CONFIG_FILES([Makefile])
+AC_OUTPUT
diff -Naur libmpeg3-1.5.4.orig/libmpeg3.h libmpeg3-1.5.4/libmpeg3.h
--- libmpeg3-1.5.4.orig/libmpeg3.h	2005-07-29 15:55:57.000000000 +0200
+++ libmpeg3-1.5.4/libmpeg3.h	2005-07-29 15:56:05.000000000 +0200
@@ -5,7 +5,7 @@
 extern "C" {
 #endif
 
-#include "mpeg3private.h"
+#include <libmpeg3/mpeg3private.h>
 
 
 /* Supported color models for mpeg3_read_frame */
diff -Naur libmpeg3-1.5.4.orig/mpeg3atrack.h libmpeg3-1.5.4/mpeg3atrack.h
--- libmpeg3-1.5.4.orig/mpeg3atrack.h	2005-07-29 15:55:57.000000000 +0200
+++ libmpeg3-1.5.4/mpeg3atrack.h	2005-07-29 15:56:05.000000000 +0200
@@ -1,7 +1,7 @@
 #ifndef MPEG3ATRACK_H
 #define MPEG3ATRACK_H
 
-#include "mpeg3demux.h"
-#include "audio/mpeg3audio.h"
+#include <libmpeg3/mpeg3demux.h>
+#include <libmpeg3/audio/mpeg3audio.h>
 
 #endif
diff -Naur libmpeg3-1.5.4.orig/mpeg3css.h libmpeg3-1.5.4/mpeg3css.h
--- libmpeg3-1.5.4.orig/mpeg3css.h	2005-07-29 15:55:57.000000000 +0200
+++ libmpeg3-1.5.4/mpeg3css.h	2005-07-29 15:56:05.000000000 +0200
@@ -2,6 +2,6 @@
 #define MPEG3CSS_H
 
 
-#include "mpeg3private.inc"
+#include <libmpeg3/mpeg3private.inc>
 
 #endif
diff -Naur libmpeg3-1.5.4.orig/mpeg3demux.h libmpeg3-1.5.4/mpeg3demux.h
--- libmpeg3-1.5.4.orig/mpeg3demux.h	2005-07-29 15:55:57.000000000 +0200
+++ libmpeg3-1.5.4/mpeg3demux.h	2005-07-29 15:56:05.000000000 +0200
@@ -1,7 +1,7 @@
 #ifndef MPEG3DEMUX_H
 #define MPEG3DEMUX_H
 
-#include "mpeg3title.h"
+#include <libmpeg3/mpeg3title.h>
 #include <stdio.h>
 
 #endif
diff -Naur libmpeg3-1.5.4.orig/mpeg3io.h libmpeg3-1.5.4/mpeg3io.h
--- libmpeg3-1.5.4.orig/mpeg3io.h	2005-07-29 15:55:57.000000000 +0200
+++ libmpeg3-1.5.4/mpeg3io.h	2005-07-29 15:56:05.000000000 +0200
@@ -4,7 +4,7 @@
 
 #include <stdio.h>
 #include <stdint.h>
-#include "mpeg3css.h"
-#include "mpeg3private.inc"
+#include <libmpeg3/mpeg3css.h>
+#include <libmpeg3/mpeg3private.inc>
 
 #endif
diff -Naur libmpeg3-1.5.4.orig/mpeg3split.c libmpeg3-1.5.4/mpeg3split.c
--- libmpeg3-1.5.4.orig/mpeg3split.c	2005-07-29 15:55:57.000000000 +0200
+++ libmpeg3-1.5.4/mpeg3split.c	2005-07-29 15:56:05.000000000 +0200
@@ -1,6 +1,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include "mpeg3private.h"
 
 #include "mpeg3private.inc"
 
diff -Naur libmpeg3-1.5.4.orig/mpeg3title.h libmpeg3-1.5.4/mpeg3title.h
--- libmpeg3-1.5.4.orig/mpeg3title.h	2005-07-29 15:55:57.000000000 +0200
+++ libmpeg3-1.5.4/mpeg3title.h	2005-07-29 15:56:05.000000000 +0200
@@ -1,6 +1,6 @@
 #ifndef MPEG3TITLE_H
 #define MPEG3TITLE_H
 
-#include "mpeg3io.h"
+#include <libmpeg3/mpeg3io.h>
 
 #endif
diff -Naur libmpeg3-1.5.4.orig/mpeg3vtrack.h libmpeg3-1.5.4/mpeg3vtrack.h
--- libmpeg3-1.5.4.orig/mpeg3vtrack.h	2005-07-29 15:55:57.000000000 +0200
+++ libmpeg3-1.5.4/mpeg3vtrack.h	2005-07-29 15:56:05.000000000 +0200
@@ -1,7 +1,7 @@
 #ifndef MPEG3_VTRACK_H
 #define MPEG3_VTRACK_H
 
-#include "mpeg3demux.h"
-#include "video/mpeg3video.h"
+#include <libmpeg3/mpeg3demux.h>
+#include <libmpeg3/video/mpeg3video.h>
 
 #endif
diff -Naur libmpeg3-1.5.4.orig/strip_fPIC.sh libmpeg3-1.5.4/strip_fPIC.sh
--- libmpeg3-1.5.4.orig/strip_fPIC.sh	1970-01-01 01:00:00.000000000 +0100
+++ libmpeg3-1.5.4/strip_fPIC.sh	2005-07-29 15:56:05.000000000 +0200
@@ -0,0 +1,17 @@
+#!/bin/sh
+#
+# taken from flac sources  --qboosh
+#
+# libtool assumes that the compiler can handle the -fPIC flag
+# This isn't always true (for example, nasm can't handle it)
+command=""
+while [ $1 ]; do
+	if [ "$1" != "-fPIC" ]; then
+		if [ "$1" != "-DPIC" ]; then
+			command="$command $1"
+		fi
+	fi
+	shift
+done
+echo $command
+exec $command
diff -Naur libmpeg3-1.5.4.orig/video/mpeg3video.h libmpeg3-1.5.4/video/mpeg3video.h
--- libmpeg3-1.5.4.orig/video/mpeg3video.h	2005-07-29 15:55:57.000000000 +0200
+++ libmpeg3-1.5.4/video/mpeg3video.h	2005-07-29 15:56:05.000000000 +0200
@@ -1,9 +1,9 @@
 #ifndef MPEGVIDEO_H
 #define MPEGVIDEO_H
 
-#include "../bitstream.h"
-#include "../mpeg3private.inc"
-#include "idct.h"
-#include "slice.h"
-#include "../timecode.h"
+#include <libmpeg3/bitstream.h>
+#include <libmpeg3/mpeg3private.inc>
+#include <libmpeg3/video/idct.h>
+#include <libmpeg3/video/slice.h>
+#include <libmpeg3/timecode.h>
 #endif
diff -Naur libmpeg3-1.5.4.orig/video/output.c libmpeg3-1.5.4/video/output.c
--- libmpeg3-1.5.4.orig/video/output.c	2005-07-29 15:55:57.000000000 +0200
+++ libmpeg3-1.5.4/video/output.c	2005-07-29 15:56:37.000000000 +0200
@@ -6,21 +6,21 @@
 
 static long long mpeg3_MMX_0 = 0L;
 static unsigned long  mpeg3_MMX_10w[]         = {0x00100010, 0x00100010};                     /*dd    00010 0010h, 000100010h */
-static unsigned long  mpeg3_MMX_80w[]         = {0x00800080, 0x00800080};                     /*dd    00080 0080h, 000800080h */
+static unsigned long __attribute__((used)) mpeg3_MMX_80w[]         = {0x00800080, 0x00800080};                     /*dd    00080 0080h, 000800080h */
 
-static unsigned long  mpeg3_MMX_00FFw[]       = {0x00ff00ff, 0x00ff00ff};                     /*dd    000FF 00FFh, 000FF00FFh */
+static unsigned long __attribute__((used)) mpeg3_MMX_00FFw[]       = {0x00ff00ff, 0x00ff00ff};                     /*dd    000FF 00FFh, 000FF00FFh */
 
-static unsigned short mpeg3_MMX_Ublucoeff[]   = {0x81, 0x81, 0x81, 0x81};                     /*dd    00081 0081h, 000810081h */
-static unsigned short mpeg3_MMX_Vredcoeff[]   = {0x66, 0x66, 0x66, 0x66};                     /*dd    00066 0066h, 000660066h */
+static unsigned short __attribute__((used)) mpeg3_MMX_Ublucoeff[]   = {0x81, 0x81, 0x81, 0x81};                     /*dd    00081 0081h, 000810081h */
+static unsigned short __attribute__((used)) mpeg3_MMX_Vredcoeff[]   = {0x66, 0x66, 0x66, 0x66};                     /*dd    00066 0066h, 000660066h */
 
-static unsigned short mpeg3_MMX_Ugrncoeff[]   = {0xffe8, 0xffe8, 0xffe8, 0xffe8};             /*dd    0FFE7 FFE7h, 0FFE7FFE7h */
-static unsigned short mpeg3_MMX_Vgrncoeff[]   = {0xffcd, 0xffcd, 0xffcd, 0xffcd};             /*dd    0FFCC FFCCh, 0FFCCFFCCh */
+static unsigned short __attribute__((used)) mpeg3_MMX_Ugrncoeff[]   = {0xffe8, 0xffe8, 0xffe8, 0xffe8};             /*dd    0FFE7 FFE7h, 0FFE7FFE7h */
+static unsigned short __attribute__((used)) mpeg3_MMX_Vgrncoeff[]   = {0xffcd, 0xffcd, 0xffcd, 0xffcd};             /*dd    0FFCC FFCCh, 0FFCCFFCCh */
 
-static unsigned short mpeg3_MMX_Ycoeff[]      = {0x4a, 0x4a, 0x4a, 0x4a};                     /*dd    0004A 004Ah, 0004A004Ah */
+static unsigned short __attribute__((used)) mpeg3_MMX_Ycoeff[]      = {0x4a, 0x4a, 0x4a, 0x4a};                     /*dd    0004A 004Ah, 0004A004Ah */
 
-static unsigned short mpeg3_MMX_redmask[]     = {0xf800, 0xf800, 0xf800, 0xf800};             /*dd    07c00 7c00h, 07c007c00h */
+static unsigned short __attribute__((used)) mpeg3_MMX_redmask[]     = {0xf800, 0xf800, 0xf800, 0xf800};             /*dd    07c00 7c00h, 07c007c00h */
 
-static unsigned short mpeg3_MMX_grnmask[]     = {0x7e0, 0x7e0, 0x7e0, 0x7e0};                 /*dd    003e0 03e0h, 003e003e0h */
+static unsigned short __attribute__((used)) mpeg3_MMX_grnmask[]     = {0x7e0, 0x7e0, 0x7e0, 0x7e0};                 /*dd    003e0 03e0h, 003e003e0h */
 
 static unsigned char mpeg3_601_to_rgb[256];
 
@@ -207,12 +207,12 @@
 		);
 }
 
-static unsigned long long  mpeg3_MMX_U_80 = 0x0000008000800000LL;
-static unsigned long long  mpeg3_MMX_V_80 = 0x0000000000800080LL;
-static long long  mpeg3_MMX_U_COEF        = 0x00000058ffd30000LL;
-static long long  mpeg3_MMX_V_COEF        = 0x00000000ffea006fLL;
-static long long  mpeg3_MMX_601_Y_COEF    = 0x0000004800480048LL;
-static long long  mpeg3_MMX_601_Y_DIFF    = 0x0000000000000010LL;
+static unsigned long long __attribute__((used)) mpeg3_MMX_U_80 = 0x0000008000800000LL;
+static unsigned long long __attribute__((used)) mpeg3_MMX_V_80 = 0x0000000000800080LL;
+static long long __attribute__((used)) mpeg3_MMX_U_COEF        = 0x00000058ffd30000LL;
+static long long __attribute__((used)) mpeg3_MMX_V_COEF        = 0x00000000ffea006fLL;
+static long long __attribute__((used)) mpeg3_MMX_601_Y_COEF    = 0x0000004800480048LL;
+static long long __attribute__((used)) mpeg3_MMX_601_Y_DIFF    = 0x0000000000000010LL;
 
 inline void mpeg3_bgra32_mmx(unsigned long y, 
 		unsigned long u, 
@@ -297,10 +297,10 @@
 : "r" (&y), "r" (&u), "r" (&v), "r" (output));
 }
 
-static unsigned long long  mpeg3_MMX_U_80_RGB    = 0x0000000000800080LL;
-static unsigned long long  mpeg3_MMX_V_80_RGB    = 0x0000008000800000LL;
-static long long  mpeg3_MMX_U_COEF_RGB    = 0x00000000ffd30058LL;
-static long long  mpeg3_MMX_V_COEF_RGB    = 0x0000006fffea0000LL;
+static unsigned long long __attribute__((used)) mpeg3_MMX_U_80_RGB    = 0x0000000000800080LL;
+static unsigned long long __attribute__((used)) mpeg3_MMX_V_80_RGB    = 0x0000008000800000LL;
+static long long __attribute__((used)) mpeg3_MMX_U_COEF_RGB    = 0x00000000ffd30058LL;
+static long long __attribute__((used)) mpeg3_MMX_V_COEF_RGB    = 0x0000006fffea0000LL;
 
 inline void mpeg3_rgba32_mmx(unsigned long y, 
 		unsigned long u, 
@@ -497,10 +497,11 @@
 	*data++ = 0;
 
 #define STORE_PIXEL_RGB565 \
-	*((unsigned short*)data)++ = \
+	*((unsigned short*)data) = \
 		((CLIP(r_l) & 0xf8) << 8) | \
 		((CLIP(g_l) & 0xfc) << 3) | \
-		((CLIP(b_l) & 0xf8) >> 3);
+		((CLIP(b_l) & 0xf8) >> 3); \
+	*((unsigned short*)data) += 1;
 
 #define STORE_PIXEL_RGB888 \
 	*data++ = CLIP(r_l); \
diff -Naur libmpeg3-1.5.4.orig/video/reconstruct.c libmpeg3-1.5.4/video/reconstruct.c
--- libmpeg3-1.5.4.orig/video/reconstruct.c	2005-07-29 15:55:57.000000000 +0200
+++ libmpeg3-1.5.4/video/reconstruct.c	2005-07-29 15:56:05.000000000 +0200
@@ -271,8 +271,8 @@
 }
 
 #else  // HAVE_3DNOW
-	static long long ADD_1	=	0x0101010101010101LL;
-	static long long MASK_AND = 0x7f7f7f7f7f7f7f7fLL;
+	static long long __attribute__((used)) ADD_1	= 0x0101010101010101LL;
+	static long long __attribute__((used)) MASK_AND = 0x7f7f7f7f7f7f7f7fLL;
 #endif
 
 static inline void rec_mmx(unsigned char *s, unsigned char *d, int lx2, int h)
diff -Naur libmpeg3-1.5.4.orig/video/slice.c libmpeg3-1.5.4/video/slice.c
--- libmpeg3-1.5.4.orig/video/slice.c	2005-07-29 15:55:57.000000000 +0200
+++ libmpeg3-1.5.4/video/slice.c	2005-07-29 15:56:05.000000000 +0200
@@ -6,7 +6,7 @@
 
 #define CLIP(x)  ((x) >= 0 ? ((x) < 255 ? (x) : 255) : 0)
 
-static unsigned long long MMX_128 = 0x80008000800080LL;
+static unsigned long long __attribute__((used)) MMX_128 = 0x80008000800080LL;
 
 int mpeg3_new_slice_buffer(mpeg3_slice_buffer_t *slice_buffer)
 {
