diff -Naur xfsprogs-4.14.0/io/copy_file_range.c xfsprogs-4.14.0-p/io/copy_file_range.c
--- xfsprogs-4.14.0/io/copy_file_range.c	2017-11-20 01:14:14.000000000 +0100
+++ xfsprogs-4.14.0-p/io/copy_file_range.c	2018-01-28 19:35:51.268118456 +0100
@@ -43,7 +43,7 @@
 }
 
 static loff_t
-copy_file_range(int fd, loff_t *src, loff_t *dst, size_t len)
+copy_file_range_xfs(int fd, loff_t *src, loff_t *dst, size_t len)
 {
 	loff_t ret;
 
@@ -130,7 +130,7 @@
 		copy_dst_truncate();
 	}
 
-	ret = copy_file_range(fd, &src, &dst, len);
+	ret = copy_file_range_xfs(fd, &src, &dst, len);
 	close(fd);
 	return ret;
 }
