diff -Naur xfsprogs-4.7.0/include/buildmacros xfsprogs-4.7.0-p/include/buildmacros
--- xfsprogs-4.7.0/include/buildmacros	2016-02-29 00:39:26.000000000 +0100
+++ xfsprogs-4.7.0-p/include/buildmacros	2016-08-06 01:05:11.412205778 +0200
@@ -70,18 +70,9 @@
 # /usr/lib.
 ifeq ($(ENABLE_SHARED),yes)
 INSTALL_LTLIB_DEV = \
-	cd $(TOPDIR)/$(LIBNAME)/.libs; \
-	../$(INSTALL) -m 755 -d $(PKG_LIB_DIR); \
-	../$(INSTALL) -m 644 -T old_lib $(LIBNAME).lai $(PKG_LIB_DIR); \
-	../$(INSTALL) -m 644 $(LIBNAME).lai $(PKG_LIB_DIR)/$(LIBNAME).la ; \
-	../$(INSTALL) -m 755 -d $(PKG_ROOT_LIB_DIR); \
-	../$(INSTALL) -T so_base $(LIBNAME).lai $(PKG_ROOT_LIB_DIR); \
-	if [ "x$(shell readlink -f $(PKG_LIB_DIR))" != \
-	     "x$(shell readlink -f $(PKG_ROOT_LIB_DIR))" ]; then \
-		../$(INSTALL) -S $(PKG_LIB_DIR)/$(LIBNAME).a $(PKG_ROOT_LIB_DIR)/$(LIBNAME).a; \
-		../$(INSTALL) -S $(PKG_LIB_DIR)/$(LIBNAME).la $(PKG_ROOT_LIB_DIR)/$(LIBNAME).la; \
-		../$(INSTALL) -S $(PKG_ROOT_LIB_DIR)/$(LIBNAME).so $(PKG_LIB_DIR)/$(LIBNAME).so; \
-	fi
+	set -e; cd $(TOPDIR)/$(LIBNAME); \
+	$(INSTALL) -m 755 -d $(PKG_LIB_DIR); \
+	env -uDIST_ROOT $(LTINSTALL) $(TOPDIR)/$(LIBNAME)/$(LIBNAME).la $(DIST_ROOT)$(PKG_LIB_DIR)/$(LIBNAME).la
 else
 INSTALL_LTLIB_DEV = $(INSTALL_LTLIB_STATIC)
 endif
diff -Naur xfsprogs-4.7.0/libxcmd/Makefile xfsprogs-4.7.0-p/libxcmd/Makefile
--- xfsprogs-4.7.0/libxcmd/Makefile	2016-07-22 00:20:50.000000000 +0200
+++ xfsprogs-4.7.0-p/libxcmd/Makefile	2016-08-06 01:05:11.412205778 +0200
@@ -34,6 +34,9 @@
 
 include $(BUILDRULES)
 
-install install-dev: default
+install: default
+
+install-dev: default
+	$(INSTALL_LTLIB_DEV)
 
 -include .ltdep
diff -Naur xfsprogs-4.7.0/libxfs/Makefile xfsprogs-4.7.0-p/libxfs/Makefile
--- xfsprogs-4.7.0/libxfs/Makefile	2016-07-22 00:20:33.000000000 +0200
+++ xfsprogs-4.7.0-p/libxfs/Makefile	2016-08-06 01:05:26.209205467 +0200
@@ -138,6 +138,7 @@
 
 install-dev: install
 	$(INSTALL) -m 644 $(PKGHFILES) $(PKG_INC_DIR)
+	$(INSTALL_LTLIB_DEV)
 
 # We need to install the headers before building the dependencies.  If we
 # include the .ltdep file, the makefile decides that it needs to build the
diff -Naur xfsprogs-4.7.0/libxlog/Makefile xfsprogs-4.7.0-p/libxlog/Makefile
--- xfsprogs-4.7.0/libxlog/Makefile	2016-02-29 00:39:26.000000000 +0100
+++ xfsprogs-4.7.0-p/libxlog/Makefile	2016-08-06 01:05:11.412205778 +0200
@@ -12,6 +12,8 @@
 
 CFILES = xfs_log_recover.c util.c
 
+LTLIBS = $(LIBUUID) $(LIBXFS)
+
 # don't want to link xfs_repair with a debug libxlog.
 DEBUG = -DNDEBUG
 
@@ -19,6 +21,9 @@
 
 include $(BUILDRULES)
 
-install install-dev: default
+install: default
+
+install-dev: default
+	$(INSTALL_LTLIB_DEV)
 
 -include .ltdep
diff -Naur xfsprogs-4.7.0/Makefile xfsprogs-4.7.0-p/Makefile
--- xfsprogs-4.7.0/Makefile	2016-07-22 00:20:50.000000000 +0200
+++ xfsprogs-4.7.0-p/Makefile	2016-08-06 01:05:11.413205778 +0200
@@ -82,8 +82,11 @@
 quota: libxcmd
 repair: libxlog libxcmd
 copy: libxlog
+libxlog: libxfs
 mkfs: libxcmd
 
+libxlog-install-dev: libxfs-install-dev
+
 ifeq ($(HAVE_BUILDDEFS), yes)
 include $(BUILDRULES)
 else
