diff -Naur xfsprogs-2.5.6.orig/libxfs/linux.c xfsprogs-2.5.6/libxfs/linux.c
--- xfsprogs-2.5.6.orig/libxfs/linux.c	2003-10-10 16:33:37.000000000 +0200
+++ xfsprogs-2.5.6/libxfs/linux.c	2004-06-18 12:17:54.000000000 +0200
@@ -42,10 +42,10 @@
 extern char *progname;
 
 #ifndef BLKGETSIZE64
-# define BLKGETSIZE64	_IOR(0x12,114,sizeof(__uint64_t))
+# define BLKGETSIZE64	_IOR(0x12,114,size_t)
 #endif
 #ifndef BLKBSZSET
-# define BLKBSZSET	_IOW(0x12,113,sizeof(int))
+# define BLKBSZSET	_IOW(0x12,113,size_t)
 #endif
 #ifndef BLKSSZGET
 # define BLKSSZGET	_IO(0x12,104)
