--- speech-dispatcher-0.6.7/src/modules/module_main.c.orig	2009-05-20 11:40:37.000000000 +0100
+++ speech-dispatcher-0.6.7/src/modules/module_main.c	2009-05-20 11:40:46.000000000 +0100
@@ -22,7 +22,7 @@
  */
 
 /* So that gcc doesn't comply */
-int getline(char**, size_t*, FILE*);
+/*int getline(char**, size_t*, FILE*);*/
 
 #define PROCESS_CMD(command, function) \
 if (!strcmp(cmd_buf, #command"\n")){ \
--- speech-dispatcher-0.6.7/src/modules/module_utils.h.orig	2009-05-20 11:48:06.000000000 +0100
+++ speech-dispatcher-0.6.7/src/modules/module_utils.h	2009-05-20 11:48:14.000000000 +0100
@@ -421,7 +421,7 @@
 
 
 /* So that gcc doesn't complain */
-int getline(char**, size_t*, FILE*);
+/*int getline(char**, size_t*, FILE*);*/
 
 pthread_mutex_t module_stdout_mutex;
 
