--- g-wrap-1.9.6/configure.ac.moo	2006-03-27 16:31:41.000000000 -0500
+++ g-wrap-1.9.6/configure.ac	2006-03-27 16:31:15.000000000 -0500
@@ -95,7 +95,7 @@
 AM_CONDITIONAL(HAVE_SRFI_35, test "$HAVE_SRFI_35" = yes)
 
 dnl Check for optional libraries
-PKG_CHECK_MODULES(GLIB, glib, [have_glib=yes], [have_glib=no])
+PKG_CHECK_MODULES(GLIB, glib-2.0, [have_glib=yes], [have_glib=no])
 AM_CONDITIONAL(HAVE_GLIB, test "$have_glib" = yes)
 
 dnl Check for compiler flags and warnings. This should be last, since
--- g-wrap-1.9.6/configure.glib2	2006-02-18 15:10:25.000000000 -0500
+++ g-wrap-1.9.6/configure	2006-02-18 15:11:39.000000000 -0500
@@ -21699,8 +21699,8 @@
   if test "x$ac_cv_env_GLIB_CFLAGS_set" = "xset"; then
 	pkg_cv_GLIB_CFLAGS=$ac_cv_env_GLIB_CFLAGS_value
 elif test -n "$PKG_CONFIG"; then
-	if $PKG_CONFIG --exists "glib" >/dev/null 2>&1; then
-		pkg_cv_GLIB_CFLAGS=`$PKG_CONFIG --cflags "glib" 2>/dev/null`
+	if $PKG_CONFIG --exists "glib-2.0" >/dev/null 2>&1; then
+		pkg_cv_GLIB_CFLAGS=`$PKG_CONFIG --cflags "glib-2.0" 2>/dev/null`
 	else
 		pkg_failed=yes
 	fi
@@ -21718,8 +21718,8 @@
   if test "x$ac_cv_env_GLIB_LIBS_set" = "xset"; then
 	pkg_cv_GLIB_LIBS=$ac_cv_env_GLIB_LIBS_value
 elif test -n "$PKG_CONFIG"; then
-	if $PKG_CONFIG --exists "glib" >/dev/null 2>&1; then
-		pkg_cv_GLIB_LIBS=`$PKG_CONFIG --libs "glib" 2>/dev/null`
+	if $PKG_CONFIG --exists "glib-2.0" >/dev/null 2>&1; then
+		pkg_cv_GLIB_LIBS=`$PKG_CONFIG --libs "glib-2.0" 2>/dev/null`
 	else
 		pkg_failed=yes
 	fi
@@ -21731,7 +21731,7 @@
 echo "${ECHO_T}$pkg_cv_GLIB_LIBS" >&6
 
 if test $pkg_failed = yes; then
-	GLIB_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "glib"`
+	GLIB_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "glib-2.0"`
 	# Put the nasty error message in config.log where it belongs
 	echo "$GLIB_PKG_ERRORS" 1>&5
 
