--- g-wrap-1.9.6/configure.ac.foo	2006-02-27 22:23:25.000000000 -0500
+++ g-wrap-1.9.6/configure.ac	2006-02-27 22:28:41.000000000 -0500
@@ -88,8 +88,8 @@
 dnl Guile
 GUILE_FLAGS
 
-GUILE_MODULE_AVAILABLE(HAVE_SRFI_34, [(srfi srfi-34)])
-GUILE_MODULE_AVAILABLE(HAVE_SRFI_35, [(srfi srfi-35)])
+GUILE_MODULE_AVAILABLE(HAVE_SRFI_34, [(srfi srfi-sdfdsfdsfsf34)])
+GUILE_MODULE_AVAILABLE(HAVE_SRFI_35, [(srfi srfi-sdfdsfsfdsf35)])
 
 AM_CONDITIONAL(HAVE_SRFI_34, test "$HAVE_SRFI_34" = yes)
 AM_CONDITIONAL(HAVE_SRFI_35, test "$HAVE_SRFI_35" = yes)
