--- src/wl/sys/wl_linux.c	2015-09-19 00:47:30.000000000 +0200
+++ src.new/wl/sys/wl_linux.c	2022-08-02 10:16:25.059062466 +0200
@@ -724,7 +724,7 @@
 		WL_ALL_PASSIVE_ENAB(wl) ?  ", Passive Mode" : "", EPI_VERSION_STR);
 
 #ifdef BCMDBG
-	printf(" (Compiled in " SRCBASE " at " __TIME__ " on " __DATE__ ")");
+	printf(" (Compiled in " SRCBASE ")");
 #endif 
 	printf("\n");
 
@@ -2053,8 +2053,7 @@
 void
 wl_dump_ver(wl_info_t *wl, struct bcmstrbuf *b)
 {
-	bcm_bprintf(b, "wl%d: %s %s version %s\n", wl->pub->unit,
-		__DATE__, __TIME__, EPI_VERSION_STR);
+	bcm_bprintf(b, "wl%d: version %s\n", wl->pub->unit, EPI_VERSION_STR);
 }
 
 #if defined(BCMDBG)
