diff -Naur src/shared/linux_osl.c scr-crap-o-meter/shared/linux_osl.c
--- src/shared/linux_osl.c	2015-09-19 00:47:30.000000000 +0200
+++ scr-crap-o-meter/shared/linux_osl.c	2020-03-31 00:30:59.667169330 +0200
@@ -30,6 +30,7 @@
 #include <pcicfg.h>
 
 #include <linux/fs.h>
+#include <linux/io.h>
 
 #define PCI_CFG_RETRY 		10
 
diff -Naur src/wl/sys/wl_linux.c scr-crap-o-meter/wl/sys/wl_linux.c
--- src/wl/sys/wl_linux.c	2015-09-19 00:47:30.000000000 +0200
+++ scr-crap-o-meter/wl/sys/wl_linux.c	2020-03-31 00:39:52.180068119 +0200
@@ -93,6 +93,8 @@
 
 #include <wlc_wowl.h>
 
+#include <linux/io.h>
+
 static void wl_timer(ulong data);
 static void _wl_timer(wl_timer_t *t);
 static struct net_device *wl_alloc_linux_if(wl_if_t *wlif);
@@ -3335,10 +3337,17 @@
 }
 
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 10, 0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,6,0)
 static const struct file_operations wl_fops = {
 	.owner	= THIS_MODULE,
 	.read	= wl_proc_read,
 	.write	= wl_proc_write,
+#else
+static const struct proc_ops wl_fops = {
+	.proc_owner  = THIS_MODULE,
+	.proc_read   = wl_proc_read,
+	.proc_write  = wl_proc_write,
+#endif
 };
 #endif
 
