--- src/wl/sys/wl_linux.c	2015-09-19 00:47:30.000000000 +0200
+++ src.new/wl/sys/wl_linux.c	2022-03-22 10:39:20.763751076 +0100
@@ -3244,7 +3244,7 @@
 static ssize_t
 wl_proc_read(struct file *filp, char __user *buffer, size_t length, loff_t *offp)
 {
-	wl_info_t * wl = PDE_DATA(file_inode(filp));
+	wl_info_t * wl = pde_data(file_inode(filp));
 #endif
 	int bcmerror, len;
 	int to_user = 0;
@@ -3301,7 +3301,7 @@
 static ssize_t
 wl_proc_write(struct file *filp, const char __user *buff, size_t length, loff_t *offp)
 {
-	wl_info_t * wl = PDE_DATA(file_inode(filp));
+	wl_info_t * wl = pde_data(file_inode(filp));
 #endif
 	int from_user = 0;
 	int bcmerror;
