diff --git a/src/include/linuxver.h b/src/include/linuxver.h
index c30d374..3eed21d 100644
--- a/src/include/linuxver.h
+++ b/src/include/linuxver.h
@@ -19,8 +19,6 @@
 #include <linux/version.h>
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(2, 6, 0))
 #include <linux/config.h>
-#else
-#include <linux/autoconf.h>
 #endif
 #include <linux/module.h>
 
diff --git a/src/wl/sys/wl_linux.c b/src/wl/sys/wl_linux.c
index 3626b80..fbec8e8 100644
--- a/src/wl/sys/wl_linux.c
+++ b/src/wl/sys/wl_linux.c
@@ -40,6 +40,9 @@
 #include <linux/pci_ids.h>
 #define WLC_MAXBSSCFG		1	
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 32)
+#include <linux/sched.h>
+#endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 29)
 #include <net/lib80211.h>
 #endif
@@ -187,6 +190,7 @@ static struct pci_device_id wl_id_table[] = {
 	{ 0 }
 };
 MODULE_DEVICE_TABLE(pci, wl_id_table);
+MODULE_LICENSE("Mixed/Proprietary"); 
 
 #ifdef BCMDBG
 static int msglevel = 0xdeadbeef;
