--- x/src/wl/sys/wl_linux.c-	2010-06-08 10:26:17.320092425 +0100
+++ x/src/wl/sys/wl_linux.c	2010-06-08 10:42:16.824567775 +0100
@@ -1416,7 +1416,7 @@
 _wl_set_multicast_list(struct net_device *dev)
 {
 	wl_info_t *wl;
-	struct dev_mc_list *mclist;
+	struct netdev_hw_addr *ha;
 	int i;
 
 	if (!dev)
@@ -1430,14 +1430,15 @@
 	if (wl->pub->up) {
 		wl->pub->allmulti = (dev->flags & IFF_ALLMULTI)? TRUE: FALSE;
 
-		for (i = 0, mclist = dev->mc_list; mclist && (i < dev->mc_count);
-			i++, mclist = mclist->next) {
+		i = 0;
+		netdev_for_each_mc_addr(ha, dev) {
 			if (i >= MAXMULTILIST) {
 				wl->pub->allmulti = TRUE;
 				i = 0;
 				break;
 			}
-			wl->pub->multicast[i] = *((struct ether_addr*) mclist->dmi_addr);
+			wl->pub->multicast[i] = *((struct ether_addr*)ha->addr);
+			i++;
 		}
 		wl->pub->nmulticast = i;
 		wlc_set(wl->wlc, WLC_SET_PROMISC, (dev->flags & IFF_PROMISC));
