--- src/shared/linux_osl.c	2017-12-29 22:59:17.000000000 -0400
+++ /src/shared/linux_osl.c	2017-12-29 23:44:24.786247269 -0400
@@ -1076,11 +1076,21 @@
 {
 	struct file *fp = (struct file *)image;
 	int rdlen;
+	loff_t pos;
 
 	if (!image)
 		return 0;
 
-	rdlen = kernel_read(fp, fp->f_pos, buf, len);
+	pos = fp->f_pos;
+	rdlen = kernel_read(fp, 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4, 14, 0)
+			pos, 
+#endif
+			buf, len
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 14, 0)
+			,&pos 
+#endif
+	);
 	if (rdlen > 0)
 		fp->f_pos += rdlen;
 
