--- hslua-module-system-1.1.0.1/hslua-module-system.cabal	2001-09-09 03:46:40.000000000 +0200
+++ hslua-module-system-1.1.0.1.new/hslua-module-system.cabal	2023-12-10 22:36:53.725043318 +0100
@@ -37,7 +37,7 @@
   build-depends:       base                 >= 4.11   && < 5
                      , hslua-core           >= 2.1    && < 2.4
                      , hslua-packaging      >= 2.3    && < 2.4
-                     , text                 >= 1.2    && < 2.1
+                     , text                 >= 1.2    && < 2.2
   default-extensions:  LambdaCase
                      , OverloadedStrings
   ghc-options:         -Wall
