diff -Naur ultimatestunts-srcdata-0621.orig/configure ultimatestunts-srcdata-0621/configure
--- ultimatestunts-srcdata-0621.orig/configure	2006-08-30 19:08:36.000000000 +0200
+++ ultimatestunts-srcdata-0621/configure	2006-08-30 19:09:08.000000000 +0200
@@ -7404,7 +7404,7 @@
 echo "$as_me:$LINENO: result: $ac_cv_lib_openal_alGetError" >&5
 echo "${ECHO_T}$ac_cv_lib_openal_alGetError" >&6
 if test $ac_cv_lib_openal_alGetError = yes; then
-  soundlibs="-lopenal $soundlibs";
+  soundlibs="-lopenal -lalut $soundlibs";
 cat >>confdefs.h <<\_ACEOF
 #define HAVE_LIBOPENAL
 _ACEOF
diff -Naur ultimatestunts-srcdata-0621.orig/ultimatestunts/sndsample.cpp ultimatestunts-srcdata-0621/ultimatestunts/sndsample.cpp
--- ultimatestunts-srcdata-0621.orig/ultimatestunts/sndsample.cpp	2006-08-30 19:08:36.000000000 +0200
+++ ultimatestunts-srcdata-0621/ultimatestunts/sndsample.cpp	2006-08-30 19:09:33.000000000 +0200
@@ -96,10 +96,7 @@
 	//data (and default values)
 	void *wave = NULL;
 	ALsizei
-		format = AL_FORMAT_MONO16,
-		size = 0,
-		bits = 0,
-		freq = 44100;
+		size = 0;
 
 	CString extension = realfile.mid(realfile.length() - 4);
 	extension.toLower();
@@ -108,7 +105,7 @@
 	{
 		//the loading proc
 		ALboolean (*alutLoadVorbis)(ALuint, ALvoid *, ALint) =
-			(ALboolean (*)(ALuint, ALvoid *, ALint))alGetProcAddress((ALubyte *) VORBISFUNC);
+			(ALboolean (*)(ALuint, ALvoid *, ALint))alGetProcAddress(VORBISFUNC);
 
 		if(alutLoadVorbis == NULL)
 		{
@@ -134,7 +131,7 @@
 	{
 		//the loading proc
 		ALboolean (*alutLoadMP3)(ALuint, ALvoid *, ALint) =
-			(ALboolean (*)(ALuint, ALvoid *, ALint))alGetProcAddress((ALubyte *) MP3FUNC);
+			(ALboolean (*)(ALuint, ALvoid *, ALint))alGetProcAddress(MP3FUNC);
 
 		if(alutLoadMP3 == NULL)
 		{
@@ -158,9 +155,9 @@
 	}
 	else //default: load as wave file
 	{
-		alGenBuffers(1, &m_Buffer);
-		alutLoadWAV(realfile.c_str(), &wave, &format, &size, &bits, &freq);
-		alBufferData(m_Buffer, format, wave, size, freq);
+		m_Buffer = alutCreateBufferFromFile(realfile.c_str());
+		if(m_Buffer == AL_NONE)
+			printf("alutCreateBufferFromFile failed\n");
 	}
 
 	free(wave);
