diff -Naur ultimatestunts-srcdata-0621.orig/shared/lconfig.h ultimatestunts-srcdata-0621/shared/lconfig.h
--- ultimatestunts-srcdata-0621.orig/shared/lconfig.h	2006-08-30 18:54:03.000000000 +0200
+++ ultimatestunts-srcdata-0621/shared/lconfig.h	2006-08-30 18:55:17.000000000 +0200
@@ -38,7 +38,7 @@
 
 public:
 
-	CStringArray CLCData::findAllFields(CString) const;
+	CStringArray findAllFields(CString) const;
 	CLCData();
 	virtual ~CLCData();
 	bool push(CString, CString, CString, CString);
