diff -Naur haddock-library-1.11.0/haddock-library.cabal haddock-library-1.11.0.new/haddock-library.cabal
--- haddock-library-1.11.0/haddock-library.cabal	2001-09-09 03:46:40.000000000 +0200
+++ haddock-library-1.11.0.new/haddock-library.cabal	2022-10-14 10:32:46.252868439 +0200
@@ -39,7 +39,7 @@
   default-language: Haskell2010
 
   build-depends:
-    , base         >= 4.5     && < 4.17
+    , base         >= 4.5     && < 4.18
     , containers   ^>= 0.4.2.1 || ^>= 0.5.0.0 || ^>= 0.6.0.1
     , text         ^>= 1.2.3.0 || ^>= 2.0
     , parsec       ^>= 3.1.13.0
