diff -Naur mac-3.99-u4-b4.old/src/MACLib/APEDecompress.cpp mac-3.99-u4-b4/src/MACLib/APEDecompress.cpp
--- mac-3.99-u4-b4.old/src/MACLib/APEDecompress.cpp	2005-07-03 16:31:35.000000000 +0200
+++ mac-3.99-u4-b4/src/MACLib/APEDecompress.cpp	2006-05-01 22:55:36.000000000 +0200
@@ -23,7 +23,7 @@
     }
 
     // get format information
-    GetInfo(APE_INFO_WAVEFORMATEX, (int) &m_wfeInput);
+    GetInfo(APE_INFO_WAVEFORMATEX, (long) &m_wfeInput);
     m_nBlockAlign = GetInfo(APE_INFO_BLOCK_ALIGN);
 
     // initialize other stuff
@@ -452,7 +452,7 @@
             }
             else
             {
-                WAVEFORMATEX wfeFormat; GetInfo(APE_INFO_WAVEFORMATEX, (int) &wfeFormat, 0);
+                WAVEFORMATEX wfeFormat; GetInfo(APE_INFO_WAVEFORMATEX, (long) &wfeFormat, 0);
                 WAVE_HEADER WAVHeader; FillWaveHeader(&WAVHeader, 
                     (m_nFinishBlock - m_nStartBlock) * GetInfo(APE_INFO_BLOCK_ALIGN), 
                     &wfeFormat,    0);
diff -Naur mac-3.99-u4-b4.old/src/MACLib/APEInfo.cpp mac-3.99-u4-b4/src/MACLib/APEInfo.cpp
--- mac-3.99-u4-b4.old/src/MACLib/APEInfo.cpp	2005-05-30 14:53:50.000000000 +0200
+++ mac-3.99-u4-b4/src/MACLib/APEInfo.cpp	2006-05-01 23:03:41.000000000 +0200
@@ -254,7 +254,7 @@
             }
             else
             {
-                WAVEFORMATEX wfeFormat; GetInfo(APE_INFO_WAVEFORMATEX, (int) &wfeFormat, 0);
+                WAVEFORMATEX wfeFormat; GetInfo(APE_INFO_WAVEFORMATEX, (long) &wfeFormat, 0);
                 WAVE_HEADER WAVHeader; FillWaveHeader(&WAVHeader, m_APEFileInfo.nWAVDataBytes, &wfeFormat,
                     m_APEFileInfo.nWAVTerminatingBytes);
                 memcpy(pBuffer, &WAVHeader, sizeof(WAVE_HEADER));
@@ -311,7 +311,7 @@
         break;
     }
     case APE_INFO_IO_SOURCE:
-        nRetVal = (int) m_spIO.GetPtr();
+        nRetVal = (long) m_spIO.GetPtr();
         break;
     case APE_INFO_FRAME_BYTES:
     {
@@ -350,10 +350,10 @@
         break;
     }
     case APE_INFO_TAG:
-        nRetVal = (int) m_spAPETag.GetPtr();
+        nRetVal = (long) m_spAPETag.GetPtr();
         break;
     case APE_INTERNAL_INFO:
-        nRetVal = (int) &m_APEFileInfo;
+        nRetVal = (long) &m_APEFileInfo;
         break;
     default:
 	break;
diff -Naur mac-3.99-u4-b4.old/src/MACLib/APESimple.cpp mac-3.99-u4-b4/src/MACLib/APESimple.cpp
--- mac-3.99-u4-b4.old/src/MACLib/APESimple.cpp	2005-03-10 13:36:25.000000000 +0100
+++ mac-3.99-u4-b4/src/MACLib/APESimple.cpp	2006-05-01 23:05:47.000000000 +0200
@@ -291,14 +291,14 @@
         if (spAPEDecompress == NULL || nFunctionRetVal != ERROR_SUCCESS) throw(nFunctionRetVal);
 
         // get the input format
-        THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (int) &wfeInput))
+        THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (long) &wfeInput))
 
         // allocate space for the header
         spTempBuffer.Assign(new unsigned char [spAPEDecompress->GetInfo(APE_INFO_WAV_HEADER_BYTES)], TRUE);
         if (spTempBuffer == NULL) throw(ERROR_INSUFFICIENT_MEMORY);
 
         // get the header
-        THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAV_HEADER_DATA, (int) spTempBuffer.GetPtr(), spAPEDecompress->GetInfo(APE_INFO_WAV_HEADER_BYTES)));
+        THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAV_HEADER_DATA, (long) spTempBuffer.GetPtr(), spAPEDecompress->GetInfo(APE_INFO_WAV_HEADER_BYTES)));
 
         // initialize the output
         if (nOutputMode == UNMAC_DECODER_OUTPUT_WAV)
@@ -370,7 +370,7 @@
             {
                 spTempBuffer.Assign(new unsigned char[spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_BYTES)], TRUE);
                 if (spTempBuffer == NULL) throw(ERROR_INSUFFICIENT_MEMORY);
-                THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_DATA, (int) spTempBuffer.GetPtr(), spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_BYTES)))
+                THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_DATA, (long) spTempBuffer.GetPtr(), spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_BYTES)))
         
                 unsigned int nBytesToWrite = spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_BYTES);
                 unsigned int nBytesWritten = 0;
@@ -392,7 +392,7 @@
                 spTempBuffer.Assign(new unsigned char[nTerminatingBytes], TRUE);
                 if (spTempBuffer == NULL) throw(ERROR_INSUFFICIENT_MEMORY);
                 
-                THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_DATA, (int) spTempBuffer.GetPtr(), nTerminatingBytes))
+                THROW_ON_ERROR(spAPEDecompress->GetInfo(APE_INFO_WAV_TERMINATING_DATA, (long) spTempBuffer.GetPtr(), nTerminatingBytes))
 
                 if (bHasTag)
                 {
diff -Naur mac-3.99-u4-b4.old/src/MACLib/MD5.cpp mac-3.99-u4-b4/src/MACLib/MD5.cpp
--- mac-3.99-u4-b4.old/src/MACLib/MD5.cpp	2005-04-28 09:04:59.000000000 +0200
+++ mac-3.99-u4-b4/src/MACLib/MD5.cpp	2006-05-01 23:07:53.000000000 +0200
@@ -100,7 +100,7 @@
     CopyToLittleEndian (tempBuffer, in, 16);
     x = tempBuffer;
 #else
-    if ( (unsigned int)in & 3 ) {
+    if ( (unsigned long)in & 3 ) {
         memcpy ( tempBuffer, in, 64 );
         x = tempBuffer;
     } 
diff -Naur mac-3.99-u4-b4.old/src/MACLib/Old/APEDecompressOld.cpp mac-3.99-u4-b4/src/MACLib/Old/APEDecompressOld.cpp
--- mac-3.99-u4-b4.old/src/MACLib/Old/APEDecompressOld.cpp	2005-03-11 13:23:37.000000000 +0100
+++ mac-3.99-u4-b4/src/MACLib/Old/APEDecompressOld.cpp	2006-05-01 22:56:28.000000000 +0200
@@ -250,7 +250,7 @@
             }
             else
             {
-                WAVEFORMATEX wfeFormat; GetInfo(APE_INFO_WAVEFORMATEX, (int) &wfeFormat, 0);
+                WAVEFORMATEX wfeFormat; GetInfo(APE_INFO_WAVEFORMATEX, (long) &wfeFormat, 0);
                 WAVE_HEADER WAVHeader; FillWaveHeader(&WAVHeader, 
                     (m_nFinishBlock - m_nStartBlock) * GetInfo(APE_INFO_BLOCK_ALIGN), 
                     &wfeFormat,    0);
diff -Naur mac-3.99-u4-b4.old/src/MACLib/Old/UnMAC.cpp mac-3.99-u4-b4/src/MACLib/Old/UnMAC.cpp
--- mac-3.99-u4-b4.old/src/MACLib/Old/UnMAC.cpp	2005-03-10 13:36:26.000000000 +0100
+++ mac-3.99-u4-b4/src/MACLib/Old/UnMAC.cpp	2006-05-01 22:59:56.000000000 +0200
@@ -78,7 +78,7 @@
     // set the initialized flag to TRUE
     m_bInitialized = TRUE;
     
-    m_pAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (int) &m_wfeInput);
+    m_pAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (long) &m_wfeInput);
 
     // return a successful value
     return ERROR_SUCCESS;
@@ -197,7 +197,7 @@
     {
         m_pAPEDecompressCore->GenerateDecodedArrays(nBlocks, nSpecialCodes, FrameIndex, CPULoadBalancingFactor);
 
-        WAVEFORMATEX WaveFormatEx; m_pAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (int) &WaveFormatEx);
+        WAVEFORMATEX WaveFormatEx; m_pAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (long) &WaveFormatEx);
         m_pPrepare->UnprepareOld(m_pAPEDecompressCore->GetDataX(), m_pAPEDecompressCore->GetDataY(), nBlocks, &WaveFormatEx, 
             pOutputData, (unsigned int *) &CRC, (int *) &nSpecialCodes, m_pAPEDecompress->GetInfo(APE_INFO_FILE_VERSION));
     }
@@ -205,7 +205,7 @@
     {
         m_pAPEDecompressCore->GenerateDecodedArrays(nBlocks, nSpecialCodes, FrameIndex, CPULoadBalancingFactor);
         
-        WAVEFORMATEX WaveFormatEx; m_pAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (int) &WaveFormatEx);
+        WAVEFORMATEX WaveFormatEx; m_pAPEDecompress->GetInfo(APE_INFO_WAVEFORMATEX, (long) &WaveFormatEx);
         m_pPrepare->UnprepareOld(m_pAPEDecompressCore->GetDataX(), NULL, nBlocks, &WaveFormatEx, 
             pOutputData, (unsigned int *) &CRC, (int *) &nSpecialCodes, m_pAPEDecompress->GetInfo(APE_INFO_FILE_VERSION));
     }
