diff -Naur eboard-1.0.0.orig/configure eboard-1.0.0/configure
--- eboard-1.0.0.orig/configure	2007-01-02 14:04:42.000000000 +0100
+++ eboard-1.0.0/configure	2007-01-02 14:05:08.000000000 +0100
@@ -2,6 +2,7 @@
 
 use IO::Handle;
 
+my $DESTDIR	= "DSTDIR";
 my $prefix      = "/usr/local";
 my $package     = "eboard";
 my $version     = "1.0.0";
@@ -553,10 +554,10 @@
 print CONFIGMAKE "CXXFLAGS  = @cxxflags\n";
 print CONFIGMAKE "LDFLAGS   = @ldflags\n";
 
-print CONFIGMAKE "prefix    = $prefix\n";
-print CONFIGMAKE "bindir    = $prefix/bin\n";
-print CONFIGMAKE "mandir    = $prefix/man\n";
-print CONFIGMAKE "datadir   = $prefix/share/$package\n";
+print CONFIGMAKE "prefix    = $DESTDIR/$prefix\n";
+print CONFIGMAKE "bindir    = $DESTDIR/$prefix/bin\n";
+print CONFIGMAKE "mandir    = $DESTDIR/$prefix/man\n";
+print CONFIGMAKE "datadir   = $DESTDIR/$prefix/share/$package\n";
 print CONFIGMAKE "version   = $version\n";
 
 close CONFIGMAKE;
