From 4605e8d3a99ddcd75314051bd00b51d8ecee06a1 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Tue, 28 Jul 2009 03:32:41 +0000
Subject: vmware: update for resources/RAC API removal

---
diff --git a/src/vmware.c b/src/vmware.c
index 2d1eabd..b1a9070 100644
--- a/src/vmware.c
+++ b/src/vmware.c
@@ -18,7 +18,9 @@ char rcsId_vmware[] =
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
+#ifndef XSERVER_LIBPCIACCESS
 #include "xf86Resources.h"
+#endif
 
 #include "compiler.h"	/* inb/outb */
 
@@ -109,11 +111,15 @@ static SymTabRec VMWAREChipsets[] = {
     { -1,                  NULL }
 };
 
+#ifndef XSERVER_LIBPCIACCESS
 static resRange vmwareLegacyRes[] = {
     { ResExcIoBlock, SVGA_LEGACY_BASE_PORT,
       SVGA_LEGACY_BASE_PORT + SVGA_NUM_PORTS*sizeof(uint32)},
     _VGA_EXCLUSIVE, _END
 };
+#else
+static int vmwareLegacyRes = 0;
+#endif
 
 #if XSERVER_LIBPCIACCESS
 
diff --git a/src/vmware.h b/src/vmware.h
index a3502dd..31560b5 100644
--- a/src/vmware.h
+++ b/src/vmware.h
@@ -19,12 +19,14 @@
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Resources.h"
+
 
 #include <X11/extensions/panoramiXproto.h>
 
 #ifdef XSERVER_LIBPCIACCESS
 #include <pciaccess.h>
+#else
+#include "xf86Resources.h"
 #endif
 
 #include "compiler.h"	        /* inb/outb */
--
cgit v0.8.2

From 0fddce5511b761206bb5f59dcd17652f15bda5ef Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Tue, 28 Jul 2009 04:11:59 +0000
Subject: vmware: fix resource removal properly

---
diff --git a/src/vmware.c b/src/vmware.c
index b1a9070..9ea9fff 100644
--- a/src/vmware.c
+++ b/src/vmware.c
@@ -118,7 +118,7 @@ static resRange vmwareLegacyRes[] = {
     _VGA_EXCLUSIVE, _END
 };
 #else
-static int vmwareLegacyRes = 0;
+#define vmwareLegacyRes NULL
 #endif
 
 #if XSERVER_LIBPCIACCESS
--
cgit v0.8.2

