diff -Naur xf86-video-vmware-11.0.1/src/vmwaremodule.c xf86-video-vmware-11.0.1/src/vmwaremodule.c
--- xf86-video-vmware-11.0.1/src/vmwaremodule.c	2010-03-18 16:02:17.000000000 +0200
+++ xf86-video-vmware-11.0.1/src/vmwaremodule.c	2010-06-04 12:57:29.642813157 +0300
@@ -143,7 +143,7 @@
     return FALSE;
 }
 
-static Bool
+static void
 vmware_chain_module(pointer opts)
 {
     int vmwlegacy_devices;
@@ -156,6 +156,8 @@
     int i;
 
     vmware_devices = xf86MatchDevice(VMWARE_DRIVER_NAME, &gdevs);
+    if(vmware_devices <= 0 || gdevs == NULL)
+	return;
     vmwgfx_devices = xf86MatchDevice(VMWGFX_DRIVER_NAME, NULL);
     vmwlegacy_devices = xf86MatchDevice(VMWLEGACY_DRIVER_NAME, NULL);
 
