--- pykdeextensions-0.4.0/src/kdedistutils.py.orig	2006-11-04 22:56:06.000000000 +0100
+++ pykdeextensions-0.4.0/src/kdedistutils.py	2006-11-04 22:56:48.000000000 +0100
@@ -1983,7 +1983,7 @@
         self.announce("Checking for a working PyKDE...")
         
         # Try to import modules one by one. 
-        for k_module in ('dcop', 'kdecore', 'kdeui', 'kio', 'kfile', 'kparts', 'khtml', 'kspell'):
+        for k_module in ('dcop', 'kdecore', 'kdeui', 'kio', 'kfile', 'kspell'):
             self.announce(k_module)
             try:
                 exec('import ' + k_module)
