diff -ur a/setup/Linux/oss/build/osscore.c b/setup/Linux/oss/build/osscore.c
--- a/setup/Linux/oss/build/osscore.c	2014-09-04 00:15:31.000000000 +0200
+++ b/setup/Linux/oss/build/osscore.c	2015-05-23 18:48:49.510828645 +0200
@@ -36,6 +36,11 @@
 #define strlen oss_strlen
 #define strcpy oss_strcpy
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,0,0)
+#define write_cr4(A) __write_cr4(A)
+#define read_cr4(A) __read_cr4(A)
+#endif
+
 typedef struct _smap_t dmap_t;
 
 #include "soundcard.h"
