diff -Naur a/writerperfect/Library_wpftcalc.mk b/writerperfect/Library_wpftcalc.mk
--- a/writerperfect/Library_wpftcalc.mk	2024-01-11 19:27:11.000000000 +0100
+++ b/writerperfect/Library_wpftcalc.mk	2024-01-22 21:52:42.784058060 +0100
@@ -56,6 +56,7 @@
 	mwaw \
 	staroffice \
 	wps \
+	zlib \
 	lcms2 \
 	icu_headers \
 	icui18n \
diff -Naur a/writerperfect/Library_wpftdraw.mk b/writerperfect/Library_wpftdraw.mk
--- a/writerperfect/Library_wpftdraw.mk	2024-01-11 19:27:11.000000000 +0100
+++ b/writerperfect/Library_wpftdraw.mk	2024-01-22 21:53:09.728166851 +0100
@@ -60,6 +60,7 @@
 	visio \
 	wpg \
 	wpd \
+	zlib \
 	lcms2 \
 	libxml2 \
 	icu_headers \
diff -Naur a/writerperfect/Library_wpftimpress.mk b/writerperfect/Library_wpftimpress.mk
--- a/writerperfect/Library_wpftimpress.mk	2024-01-11 19:27:11.000000000 +0100
+++ b/writerperfect/Library_wpftimpress.mk	2024-01-22 21:53:27.940240377 +0100
@@ -50,6 +50,7 @@
 	odfgen \
 	staroffice \
 	revenge \
+	zlib \
 ))
 
 $(eval $(call gb_Library_add_exception_objects,wpftimpress,\
