--- a/helpcompiler/source/HelpLinker.cxx
+++ b/helpcompiler/source/HelpLinker.cxx
@@ -815,7 +815,7 @@
 
 extern "C" {
 
-static void StructuredXMLErrorFunction(SAL_UNUSED_PARAMETER void *, xmlErrorPtr error)
+static void StructuredXMLErrorFunction(SAL_UNUSED_PARAMETER void *, const xmlError* error)
 {
     std::string aErrorMsg = error->message;
     std::string aXMLParsingFile;
--- a/sax/source/fastparser/fastparser.cxx	2023-11-24 23:25:01.668121113 +0100
+++ b/sax/source/fastparser/fastparser.cxx	2023-11-24 23:25:42.273455930 +0100
@@ -579,7 +579,7 @@
 OUString lclGetErrorMessage( xmlParserCtxtPtr ctxt, std::u16string_view sSystemId, sal_Int32 nLine )
 {
     const char* pMessage;
-    xmlErrorPtr error = xmlCtxtGetLastError( ctxt );
+    const xmlError* error = xmlCtxtGetLastError( ctxt );
     if( error && error->message )
         pMessage = error->message;
     else
--- a/xmlsecurity/inc/xmlsec-wrapper.h	2023-11-25 09:33:26.948688438 +0100
+++ b/xmlsecurity/inc/xmlsec-wrapper.h	2023-11-25 09:34:06.057956862 +0100
@@ -28,6 +28,7 @@
 #define XMLSEC_NO_SIZE_T
 #endif
 
+#include <libxml/parser.h>
 #include <xmlsec/base64.h>
 #include <xmlsec/bn.h>
 #include <xmlsec/errors.h>
--- a/xmlsecurity/inc/xmlsec/saxhelper.hxx	2023-11-25 09:35:47.353242138 +0100
+++ b/xmlsecurity/inc/xmlsec/saxhelper.hxx	2023-11-25 09:36:13.442088272 +0100
@@ -23,6 +23,7 @@
 
 #include <string_view>
 
+#include <libxml/parser.h>
 #include <libxml/tree.h>
 
 #include <rtl/ustring.hxx>
--- a/filter/source/xsltfilter/LibXSLTTransformer.cxx	2023-11-25 10:16:37.272450399 +0100
+++ b/filter/source/xsltfilter/LibXSLTTransformer.cxx	2023-11-25 10:17:32.514236372 +0100
@@ -333,7 +333,7 @@
         }
         else
         {
-            xmlErrorPtr lastErr = xmlGetLastError();
+            const xmlError* lastErr = xmlGetLastError();
             OUString msg;
             if (lastErr)
                 msg = OStringToOUString(lastErr->message, RTL_TEXTENCODING_UTF8);
--- a/drawinglayer/source/tools/primitive2dxmldump.cxx	2023-11-25 12:32:39.416900868 +0100
+++ b/drawinglayer/source/tools/primitive2dxmldump.cxx	2023-11-25 12:33:20.582153304 +0100
@@ -15,6 +15,7 @@
 
 #include <math.h>
 #include <memory>
+#include <libxml/parser.h>
 #include <sal/log.hxx>
 
 #include <drawinglayer/primitive2d/bitmapprimitive2d.hxx>
--- a/unoxml/source/dom/documentbuilder.cxx	2023-11-25 13:44:01.196460141 +0100
+++ b/unoxml/source/dom/documentbuilder.cxx	2023-11-25 13:44:24.189252050 +0100
@@ -22,6 +22,7 @@
 #include <string.h>
 
 #include <libxml/xmlerror.h>
+#include <libxml/parser.h>
 
 #include <memory>
 
--- a/unoxml/source/dom/attr.cxx	2023-11-25 13:45:00.746511139 +0100
+++ b/unoxml/source/dom/attr.cxx	2023-11-25 13:45:30.999553119 +0100
@@ -22,6 +22,7 @@
 #include <string.h>
 
 #include <memory>
+#include <libxml/entities.h>
 
 #include <osl/diagnose.h>
 #include <sal/log.hxx>
--- a/unoxml/source/dom/entity.cxx	2023-11-25 13:46:09.528880144 +0100
+++ b/unoxml/source/dom/entity.cxx	2023-11-25 13:46:38.929892766 +0100
@@ -22,6 +22,7 @@
 #include <osl/diagnose.h>
 
 #include <string.h>
+#include <libxml/entities.h>
 
 using namespace css::uno;
 using namespace css::xml::dom;
--- a/unoxml/source/xpath/xpathapi.cxx	2023-11-25 13:47:12.659054468 +0100
+++ b/unoxml/source/xpath/xpathapi.cxx	2023-11-25 13:47:33.691778875 +0100
@@ -259,7 +259,7 @@
             SAL_WARN("unoxml", "libxml2 error: " << str);
         }
 
-        static void structured_error_func(void *, xmlErrorPtr error)
+        static void structured_error_func(void *, const xmlError* error)
         {
             SAL_WARN("unoxml", "libxml2 error: " << make_error_message(error));
         }
--- a/unoxml/source/xpath/xpathapi.cxx	2023-11-25 13:49:01.798813420 +0100
+++ b/unoxml/source/xpath/xpathapi.cxx	2023-11-25 13:50:35.577941162 +0100
@@ -26,6 +26,7 @@
 #include <libxml/xmlerror.h>
 #include <libxml/xpath.h>
 #include <libxml/xpathInternals.h>
+#include <libxml/xmlIO.h>
 
 #include <com/sun/star/xml/xpath/XPathException.hpp>
 
@@ -217,7 +218,7 @@
         return selectSingleNode(contextNode, expr);
     }
 
-    static OUString make_error_message(xmlErrorPtr pError)
+    static OUString make_error_message(const xmlError* pError)
     {
         OUStringBuffer buf;
         if (pError) {
--- a/unoxml/source/dom/document.cxx	2023-11-25 13:51:16.467269077 +0100
+++ b/unoxml/source/dom/document.cxx	2023-11-25 13:51:42.112101914 +0100
@@ -41,6 +41,7 @@
 #include <eventdispatcher.hxx>
 
 #include <string.h>
+#include <libxml/xmlIO.h>
 
 #include <osl/diagnose.h>
 
--- a/sc/source/core/tool/interpr7.cxx	2023-11-25 16:43:49.007012348 +0100
+++ b/sc/source/core/tool/interpr7.cxx	2023-11-25 16:44:29.272337368 +0100
@@ -31,6 +31,7 @@
 #include <cstring>
 #include <memory>
 #include <string_view>
+#include <libxml/parser.h>
 
 using namespace com::sun::star;
 
diff --git a/include/xmloff/xmltoken.hxx b/include/xmloff/xmltoken.hxx
index 4e6441841774..ba42fae4d035 100644
--- a/include/xmloff/xmltoken.hxx
+++ b/include/xmloff/xmltoken.hxx
@@ -744,7 +744,7 @@ namespace xmloff::token {
         XML_EMBEDDED_VISIBLE_AREA,
         XML_EMBOSSED,
         XML_EMISSIVE_COLOR,
-        XML_EMPTY,
+        XML_TOKEN_EMPTY,
         XML_EMPTY_LINE_REFRESH,
         XML_ENABLE_NUMBERING,
         XML_ENABLED,
diff --git a/sc/source/filter/xml/XMLExportDataPilot.cxx b/sc/source/filter/xml/XMLExportDataPilot.cxx
index da65bec0dab7..bd5f16d828ff 100644
--- a/sc/source/filter/xml/XMLExportDataPilot.cxx
+++ b/sc/source/filter/xml/XMLExportDataPilot.cxx
@@ -123,7 +123,7 @@ void ScXMLExportDataPilot::WriteDPCondition(const ScQueryEntry& aQueryEntry, boo
 
     if (aQueryEntry.IsQueryByEmpty())
     {
-        rExport.AddAttribute(XML_NAMESPACE_TABLE, XML_OPERATOR, GetXMLToken(XML_EMPTY));
+        rExport.AddAttribute(XML_NAMESPACE_TABLE, XML_OPERATOR, GetXMLToken(XML_TOKEN_EMPTY));
     }
     else if (aQueryEntry.IsQueryByNonEmpty())
     {
diff --git a/sc/source/filter/xml/XMLExportDatabaseRanges.cxx b/sc/source/filter/xml/XMLExportDatabaseRanges.cxx
index 6905e02d651b..e4307065bd92 100644
--- a/sc/source/filter/xml/XMLExportDatabaseRanges.cxx
+++ b/sc/source/filter/xml/XMLExportDatabaseRanges.cxx
@@ -376,7 +376,7 @@ private:
             case SC_EQUAL:
             {
                 if (rEntry.IsQueryByEmpty())
-                    return GetXMLToken(XML_EMPTY);
+                    return GetXMLToken(XML_TOKEN_EMPTY);
                 else if (rEntry.IsQueryByNonEmpty())
                     return GetXMLToken(XML_NOEMPTY);
 
diff --git a/sc/source/filter/xml/xmlfilti.cxx b/sc/source/filter/xml/xmlfilti.cxx
index 7585ce0c50fa..61f7cfe4a349 100644
--- a/sc/source/filter/xml/xmlfilti.cxx
+++ b/sc/source/filter/xml/xmlfilti.cxx
@@ -370,7 +370,7 @@ void ScXMLConditionContext::GetOperator(
         rEntry.eOp = SC_BOTPERC;
     else if (IsXMLToken(aOpStr, XML_BOTTOM_VALUES))
         rEntry.eOp = SC_BOTVAL;
-    else if (IsXMLToken(aOpStr, XML_EMPTY))
+    else if (IsXMLToken(aOpStr, XML_TOKEN_EMPTY))
         rEntry.SetQueryByEmpty();
     else if (aOpStr == u">")
         rEntry.eOp = SC_GREATER;
@@ -422,7 +422,7 @@ void SAL_CALL ScXMLConditionContext::endFastElement( sal_Int32 /*nElement*/ )
     if (maQueryItems.empty())
     {
         ScQueryEntry::Item& rItem = rEntry.GetQueryItem();
-        if (IsXMLToken(sOperator, XML_EMPTY))
+        if (IsXMLToken(sOperator, XML_TOKEN_EMPTY))
             return;
         if (IsXMLToken(sDataType, XML_NUMBER))
         {
@@ -753,7 +753,7 @@ void SAL_CALL ScXMLDPConditionContext::endFastElement( sal_Int32 /*nElement*/ )
     else
         aFilterField.eConnect = SC_AND;
     pFilterContext->SetIsCaseSensitive(bIsCaseSensitive);
-    if (IsXMLToken(sOperator, XML_EMPTY))
+    if (IsXMLToken(sOperator, XML_TOKEN_EMPTY))
         aFilterField.SetQueryByEmpty();
     else if (IsXMLToken(sOperator, XML_NOEMPTY))
         aFilterField.SetQueryByNonEmpty();
diff --git a/xmloff/source/core/xmltoken.cxx b/xmloff/source/core/xmltoken.cxx
index 6879f37db295..e11dc0d16111 100644
--- a/xmloff/source/core/xmltoken.cxx
+++ b/xmloff/source/core/xmltoken.cxx
@@ -756,7 +756,7 @@ namespace xmloff::token {
         TOKEN( "embedded-visible-area",           XML_EMBEDDED_VISIBLE_AREA ),
         TOKEN( "embossed",                        XML_EMBOSSED ),
         TOKEN( "emissive-color",                  XML_EMISSIVE_COLOR ),
-        TOKEN( "empty",                           XML_EMPTY ),
+        TOKEN( "empty",                           XML_TOKEN_EMPTY ),
         TOKEN( "empty-line-refresh",              XML_EMPTY_LINE_REFRESH ),
         TOKEN( "enable-numbering",                XML_ENABLE_NUMBERING ),
         TOKEN( "enabled",                         XML_ENABLED ),
diff --git a/xmloff/source/style/PageMasterStyleMap.cxx b/xmloff/source/style/PageMasterStyleMap.cxx
index 7b2cab4751aa..6f631289672b 100644
--- a/xmloff/source/style/PageMasterStyleMap.cxx
+++ b/xmloff/source/style/PageMasterStyleMap.cxx
@@ -157,7 +157,7 @@ const XMLPropertyMapEntry aXMLPageMasterStyleMap[] =
     PLMAP( PROP_FootnoteLineRelativeWidth, XML_NAMESPACE_STYLE,    XML__EMPTY,     XML_TYPE_PERCENT8|MID_FLAG_SPECIAL_ITEM,    CTF_PM_FTN_LINE_WIDTH ),
     PLMAP( PROP_FootnoteLineTextDistance, XML_NAMESPACE_STYLE,    XML__EMPTY,     XML_TYPE_MEASURE|MID_FLAG_SPECIAL_ITEM,    CTF_PM_FTN_LINE_DISTANCE ),
     PLMAP( PROP_FootnoteLineWeight,        XML_NAMESPACE_STYLE,    XML_FOOTNOTE_SEP,    XML_TYPE_MEASURE16|MID_FLAG_ELEMENT_ITEM,    CTF_PM_FTN_LINE_WEIGHT ),
-    PLMAP( PROP_FootnoteLineStyle,     XML_NAMESPACE_STYLE,    XML_EMPTY,  XML_TYPE_STRING|MID_FLAG_ELEMENT_ITEM,  CTF_PM_FTN_LINE_STYLE ),
+    PLMAP( PROP_FootnoteLineStyle,     XML_NAMESPACE_STYLE,    XML_TOKEN_EMPTY,  XML_TYPE_STRING|MID_FLAG_ELEMENT_ITEM,  CTF_PM_FTN_LINE_STYLE ),
     PLMAP_EXT(PROP_GutterMargin, XML_NAMESPACE_LO_EXT, XML_MARGIN_GUTTER, XML_TYPE_MEASURE, CTF_PM_MARGINGUTTER),
 
     //////////////////////////////////////////////////////////////////////////
