diff -Naur cdrtools-2.01.orig/RULES/mk-gmake.id cdrtools-2.01/RULES/mk-gmake.id
--- cdrtools-2.01.orig/RULES/mk-gmake.id	2004-08-03 23:33:18.000000000 +0200
+++ cdrtools-2.01/RULES/mk-gmake.id	2004-08-10 10:40:59.119647640 +0200
@@ -37,7 +37,7 @@
 
 ifndef		ARCH_DONE
 
-__gmake_warn:=	$(shell cat $(SRCROOT)/$(RULESDIR)/gmake.wrn 1>&2; sleep 15)
+__gmake_warn:=	$(shell cat $(SRCROOT)/$(RULESDIR)/gmake.wrn 1>&2)
 
 _MACHCMD=	(mach || uname -p || true)	2> /dev/null
 _ARCHCMD=	(arch || /usr/ucb/arch || true)	2> /dev/null
