--- cdrtools-3.01/rscsi/Makefile	2008-08-01 23:21:12.000000000 +0200
+++ cdrtools-3.01.new/rscsi/Makefile	2024-01-08 14:58:28.469710435 +0100
@@ -6,7 +6,7 @@
 include		$(SRCROOT)/$(RULESDIR)/rules.top
 ###########################################################################
 
-INSDIR=		sbin
+INSDIR=bin
 TARGET=		rscsi
 INSMODE=	4711
 INSUSR=		root
--- cdrtools-3.01/RULES/rules.prg	2015-03-26 22:34:35.000000000 +0100
+++ cdrtools-3.01.new/RULES/rules.prg	2024-01-08 15:03:27.104846934 +0100
@@ -43,10 +43,10 @@
 #
 #SHELL=		/bin/sh
 
-LN=		/bin/ln
-SYMLINK=	/bin/ln -s
-RM=		/bin/rm
-MV=		/bin/mv
+LN=		/usr/bin/ln
+SYMLINK=	/usr/bin/ln -s
+RM=		/usr/bin/rm
+MV=		/usr/bin/mv
 LORDER=		lorder
 TSORT=		tsort
 CTAGS=		vctags
