diff -Naur shadow-4.0.16.orig/etc/useradd shadow-4.0.16/etc/useradd
--- shadow-4.0.16.orig/etc/useradd	2003-05-07 16:04:53.000000000 +0200
+++ shadow-4.0.16/etc/useradd	2006-06-28 17:37:09.000000000 +0200
@@ -1,6 +1,6 @@
 # useradd defaults file
-GROUP=1000
-HOME=/home/users
+GROUP=100
+HOME=/home
 INACTIVE=-1
 EXPIRE=
 SHELL=/bin/bash
diff -Naur shadow-4.0.16.orig/src/useradd.c shadow-4.0.16/src/useradd.c
--- shadow-4.0.16.orig/src/useradd.c	2006-05-30 20:28:45.000000000 +0200
+++ shadow-4.0.16/src/useradd.c	2006-06-28 17:37:40.000000000 +0200
@@ -1757,13 +1757,6 @@
 		find_new_uid ();
 	}
 
-	/* do we have to add a group for that user? This is why we need to
-	 * open the group files in the open_files() function  --gafton */
-	if (!(nflg || gflg)) {
-		find_new_gid ();
-		grp_add ();
-	}
-
 	usr_update ();
 
 	if (mflg) {
