--- shadow-4.14.5/etc/login.defs	2023-09-25 17:52:15.000000000 +0200
+++ shadow-4.14.5.new/etc/login.defs	2024-02-16 16:11:48.467668156 +0100
@@ -158,8 +158,8 @@
 # *REQUIRED*  The default PATH settings, for superuser and normal users.
 #
 # (they are minimal, add the rest in the shell startup files)
-ENV_SUPATH	PATH=/sbin:/bin:/usr/sbin:/usr/bin
-ENV_PATH	PATH=/bin:/usr/bin
+ENV_SUPATH	PATH=/usr/bin
+ENV_PATH	PATH=/usr/bin
 
 #
 # Terminal permissions
@@ -214,10 +214,10 @@
 #	PASS_MIN_LEN	Minimum acceptable password length.
 #	PASS_WARN_AGE	Number of days warning given before a password expires.
 #
-PASS_MAX_DAYS	99999
-PASS_MIN_DAYS	0
+PASS_MAX_DAYS	356
+PASS_MIN_DAYS	7
 PASS_MIN_LEN	5
-PASS_WARN_AGE	7
+PASS_WARN_AGE	30
 
 #
 # If "yes", the user must be listed as a member of the first gid 0 group
@@ -296,7 +296,7 @@
 # phone, home phone).  If not defined, no changes are allowed.
 # For backward compatibility, "yes" = "rwh" and "no" = "frwh".
 #
-CHFN_RESTRICT		rwh
+CHFN_RESTRICT		frwh
 
 #
 # Password prompt (%s will be replaced by user name).
@@ -334,7 +334,7 @@
 # Note: If you use PAM, it is recommended to use a value consistent with
 # the PAM modules configuration.
 #
-#ENCRYPT_METHOD DES
+ENCRYPT_METHOD SHA512
 
 #
 # Only works if ENCRYPT_METHOD is set to SHA256 or SHA512.
@@ -426,7 +426,7 @@
 #
 # This also enables userdel(8) to remove user groups if no members exist.
 #
-USERGROUPS_ENAB yes
+USERGROUPS_ENAB no
 
 #
 # If set to a non-zero number, the shadow utilities will make sure that
