--- shadow-4.0.7/etc/login.defs.orig	Sun Mar  7 19:14:33 1999
+++ shadow-4.0.7/etc/login.defs	Wed Jun 23 20:17:06 1999
@@ -157,8 +157,8 @@
 # *REQUIRED*  The default PATH settings, for superuser and normal users.
 #
 # (they are minimal, add the rest in the shell startup files)
-ENV_SUPATH	PATH=/sbin:/bin:/usr/sbin:/usr/bin
-ENV_PATH	PATH=/bin:/usr/bin
+ENV_SUPATH	PATH=/usr/local/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin:/usr/X11R6/bin:/usr/lib/mono/bin:/usr/lib/java/jre/bin:/usr/lib/java/bin:/usr/lib/qt/bin
+ENV_PATH	PATH=/usr/local/bin:/usr/bin:/bin:/usr/X11R6/bin:/usr/lib/mono/bin:/usr/lib/java/jre/bin:/usr/lib/java/bin:/usr/lib/qt/bin
 
 #
 # Terminal permissions
@@ -172,7 +172,7 @@
 # TTYPERM to either 622 or 600.
 #
 TTYGROUP	tty
-TTYPERM		0600
+TTYPERM		0620
 
 #
 # Login configuration initializations:
@@ -217,7 +217,7 @@
 #
 # If compiled with cracklib support, where are the dictionaries
 #
-CRACKLIB_DICTPATH	/var/cache/cracklib/cracklib_dict
+#CRACKLIB_DICTPATH	/var/cache/cracklib/cracklib_dict
 
 #
 # Min/max values for automatic uid selection in useradd
@@ -269,7 +269,7 @@
 # phone, home phone).  If not defined, no changes are allowed.
 # For backward compatibility, "yes" = "rwh" and "no" = "frwh".
 # 
-CHFN_RESTRICT		rwh
+CHFN_RESTRICT		frwh
 
 #
 # Password prompt (%s will be replaced by user name).
@@ -286,7 +286,7 @@
 # Set to "no" if you need to copy encrypted passwords to other systems
 # which don't understand the new algorithm.  Default is "no".
 #
-#MD5_CRYPT_ENAB	no
+MD5_CRYPT_ENAB	yes
 
 #
 # List of groups to add to the user's supplementary group set
