diff -ur a/etc/login.defs b/etc/login.defs
--- a/etc/login.defs	2012-12-12 22:53:41.856156002 -0600
+++ b/etc/login.defs	2012-12-12 22:54:12.365162895 -0600
@@ -179,8 +179,8 @@
 #
 # Prefix these values with "0" to get octal, "0x" to get hexadecimal.
 #
-ERASECHAR	0177
-KILLCHAR	025
+#ERASECHAR	0177
+#KILLCHAR	025
 #ULIMIT		2097152
 
 # Default initial "umask" value used by login on non-PAM enabled systems.
@@ -274,7 +274,7 @@
 # phone, home phone).  If not defined, no changes are allowed.
 # For backward compatibility, "yes" = "rwh" and "no" = "frwh".
 # 
-CHFN_RESTRICT		rwh
+CHFN_RESTRICT		frwh
 
 #
 # Password prompt (%s will be replaced by user name).
@@ -296,7 +296,7 @@
 #
 # This variable is deprecated. You should use ENCRYPT_METHOD.
 #
-#MD5_CRYPT_ENAB	no
+MD5_CRYPT_ENAB	yes
 
 #
 # Only works if compiled with ENCRYPTMETHOD_SELECT defined:
Only in b/etc: login.defs.orig
