--- shadow/src/su.c	2009-10-31 20:12:51.000000000 +0100
+++ shadow/src/su.c	2009-10-31 20:33:32.000000000 +0100
@@ -869,7 +869,7 @@
 
 	cp = getdef_str ((pwent.pw_uid == 0) ? "ENV_SUPATH" : "ENV_PATH");
 	if (NULL == cp) {
-		addenv ("PATH=/bin:/usr/bin", NULL);
+		addenv ((pwent.pw_uid == 0) ? "PATH=/sbin:/bin:/usr/sbin:/usr/bin" : "PATH=/bin:/usr/bin", NULL);
 	} else if (strchr (cp, '=') != NULL) {
 		addenv (cp, NULL);
 	} else {
