--- pandoc-server-0.1.0.1/pandoc-server.cabal	2001-09-09 03:46:40.000000000 +0200
+++ pandoc-server-0.1.0.1.new/pandoc-server.cabal	2023-07-24 14:47:14.339774702 +0200
@@ -52,7 +52,7 @@
   build-depends: pandoc                >= 3.0,
                  pandoc-types          >= 1.22     && < 1.24,
                  containers            >= 0.6.0.1  && < 0.7,
-                 aeson                 >= 2.0      && < 2.2,
+                 aeson                 >= 2.0      && < 2.3,
                  bytestring            >= 0.9      && < 0.12,
                  base64                >= 0.4      && < 0.5,
                  doctemplates          >= 0.11     && < 0.12,
