--- kdenetwork-3.5.10/kopete/protocols/groupwise/libgroupwise/rtf.ll    2009-06-15 14:21:34.000000000 +0200
+++ new/kopete/protocols/groupwise/libgroupwise/rtf.ll  2009-06-15 14:22:19.000000000 +0200
@@ -570,7 +570,7 @@
 
         FontDef& def = p->fonts[m_nFont-1];
 
-        char *pp = strchr(str, ';');
+        const char *pp = strchr(str, ';');
         unsigned size;
         if (pp != NULL)
            size = (pp - str);

--- kdenetwork-3.5.10/kopete/protocols/groupwise/libgroupwise/rtf.cc    2009-06-15 14:21:34.000000000 +0200
+++ new/kopete/protocols/groupwise/libgroupwise/rtf.cc  2009-06-15 14:22:19.000000000 +0200
@@ -2235,7 +2235,7 @@
 
         FontDef& def = p->fonts[m_nFont-1];
 
-        char *pp = strchr(str, ';');
+        const char *pp = strchr(str, ';');
         unsigned size;
         if (pp != NULL)
            size = (pp - str);
--- kopete-3.5.10/kopete/protocols/oscar/liboscar/rtf.cc	2005-09-10 15:20:23.000000000 +0700
+++ kopete-3.5.10/kopete/protocols/oscar/liboscar/rtf.cc	2009-06-03 04:40:25.000000000 +0700
@@ -2131,7 +2131,7 @@
 
         FontDef& def = p->fonts[m_nFont-1];
 
-        char *pp = strchr(str, ';');
+        const char *pp = strchr(str, ';');
         unsigned size;
         if (pp != NULL)
            size = (pp - str);
--- kopete-3.5.10/kopete/protocols/oscar/liboscar/rtf.ll	2005-09-10 15:20:23.000000000 +0700
+++ kopete-3.5.10/kopete/protocols/oscar/liboscar/rtf.ll	2009-06-03 04:40:35.000000000 +0700
@@ -568,7 +568,7 @@
 
         FontDef& def = p->fonts[m_nFont-1];
 
-        char *pp = strchr(str, ';');
+        const char *pp = strchr(str, ';');
         unsigned size;
         if (pp != NULL)
            size = (pp - str);

