diff -Naur kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/base/asynctcpsocket.cc kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/base/asynctcpsocket.cc
--- kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/base/asynctcpsocket.cc	2006-10-01 19:26:45.000000000 +0200
+++ kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/base/asynctcpsocket.cc	2008-07-29 03:14:38.000000000 +0200
@@ -32,6 +32,7 @@
 #include "talk/base/byteorder.h"
 #include "talk/base/common.h"
 #include "talk/base/logging.h"
+#include <cstring>
 
 #if defined(_MSC_VER) && _MSC_VER < 1300
 namespace std {
diff -Naur kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/base/bytebuffer.cc kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/base/bytebuffer.cc
--- kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/base/bytebuffer.cc	2006-10-01 19:26:45.000000000 +0200
+++ kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/base/bytebuffer.cc	2008-07-29 03:14:38.000000000 +0200
@@ -30,6 +30,7 @@
 #include "talk/base/byteorder.h"
 #include <algorithm>
 #include <cassert>
+#include <cstring>
 
 #if defined(_MSC_VER) && _MSC_VER < 1300
 namespace std {
diff -Naur kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/base/host.cc kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/base/host.cc
--- kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/base/host.cc	2006-10-01 19:26:45.000000000 +0200
+++ kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/base/host.cc	2008-07-29 03:14:38.000000000 +0200
@@ -29,7 +29,8 @@
 #include "talk/base/logging.h"
 #include "talk/base/network.h"
 #include "talk/base/socket.h"
-#include <string>
+#include <cstring>
+#include <cstdlib>
 #include <iostream>
 #include <cassert>
 #include <errno.h>
diff -Naur kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/base/messagequeue.h kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/base/messagequeue.h
--- kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/base/messagequeue.h	2006-10-01 19:26:45.000000000 +0200
+++ kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/base/messagequeue.h	2008-07-29 03:14:38.000000000 +0200
@@ -32,6 +32,7 @@
 #include "talk/base/criticalsection.h"
 #include "talk/base/socketserver.h"
 #include "talk/base/jtime.h"
+#include <cstring>
 #include <vector>
 #include <queue>
 #include <algorithm>
diff -Naur kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/base/physicalsocketserver.cc kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/base/physicalsocketserver.cc
--- kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/base/physicalsocketserver.cc	2006-10-01 19:26:45.000000000 +0200
+++ kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/base/physicalsocketserver.cc	2008-07-29 03:14:38.000000000 +0200
@@ -30,11 +30,13 @@
 #endif
 
 #include <cassert>
+#include <algorithm>
 
 #ifdef POSIX
 extern "C" {
 #include <errno.h>
 #include <fcntl.h>
+#include <string.h>
 #include <sys/time.h>
 #include <unistd.h>
 }
diff -Naur kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/base/socketadapters.cc kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/base/socketadapters.cc
--- kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/base/socketadapters.cc	2006-10-01 19:26:45.000000000 +0200
+++ kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/base/socketadapters.cc	2008-07-29 03:14:38.000000000 +0200
@@ -42,6 +42,7 @@
 #endif
 
 #include <cassert>
+#include <cstring>
 
 #include "talk/base/base64.h"
 #include "talk/base/basicdefs.h"
diff -Naur kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/p2p/base/stun.cc kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/p2p/base/stun.cc
--- kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/p2p/base/stun.cc	2006-10-01 19:26:44.000000000 +0200
+++ kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/p2p/base/stun.cc	2008-07-29 03:14:38.000000000 +0200
@@ -29,6 +29,7 @@
 #include "talk/p2p/base/stun.h"
 #include <iostream>
 #include <cassert>
+#include <cstring>
 
 #if defined(_MSC_VER) && _MSC_VER < 1300
 namespace std {
diff -Naur kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/p2p/base/stunserver.cc kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/p2p/base/stunserver.cc
--- kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/p2p/base/stunserver.cc	2006-10-01 19:26:44.000000000 +0200
+++ kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/p2p/base/stunserver.cc	2008-07-29 03:14:38.000000000 +0200
@@ -28,6 +28,7 @@
 #include "talk/base/bytebuffer.h"
 #include "talk/p2p/base/stunserver.h"
 #include <iostream>
+#include <cstring>
 
 #ifdef POSIX
 extern "C" {
diff -Naur kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/xmpppassword.h kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/xmpppassword.h
--- kdenetwork-3.5.9/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/xmpppassword.h	2006-10-01 19:26:45.000000000 +0200
+++ kdenetwork-3.5.9-gcc43/kopete/protocols/jabber/jingle/libjingle/talk/xmpp/xmpppassword.h	2008-07-29 03:14:38.000000000 +0200
@@ -31,6 +31,8 @@
 #include "talk/base/linked_ptr.h"
 #include "talk/base/scoped_ptr.h"
 
+#include <cstring>
+
 namespace buzz {
 
 class XmppPasswordImpl {
