--- xxkb-1.11.orig/xxkb.man
+++ xxkb-1.11/xxkb.man
@@ -144,7 +144,7 @@
 .TP 4
 .B XXkb.controls.button_delete_and_forget 
 This mode in addition to previous one makes \fBxxkb\fR to forget the window
-which button is deleted.  It means the \fBxxkb\f will not remember the keyboard
+which button is deleted.  It means the \fBxxkb\fR will not remember the keyboard
 state changes in this window and restore this state when the window will be
 focused.
 .TP 4
--- xxkb-1.11.orig/ru-koi-3gr
+++ xxkb-1.11/ru-koi-3gr
@@ -0,0 +1,160 @@
+xkb_symbols {
+
+    // Describes the differences between a very simple en_US
+    // keyboard and a very simple Russian keybaord
+
+//    key	<CAPS> { [ ISO_Next_Group, Caps_Lock ]};
+    key	<CAPS> { [ ISO_Next_Group ]};
+
+    name[Group1]= "US/ASCII";
+    name[Group2]= "KOI8-R";
+    name[Group3]= "Cyrillic";
+
+    key	<TLDE> {	[	    grave,	asciitilde	],
+			[            0xa3,            0xb3	],
+			[     Cyrillic_io,     Cyrillic_IO	]	};
+    key	<LSGT> {	[	     less,	   greater	],
+			[	      bar,	 brokenbar	]	};
+    key	<AD01> {	[		q,		 Q	],
+			[            0xca,            0xea	],
+			[ Cyrillic_shorti, Cyrillic_SHORTI	]	};
+    key	<AE01> {	[		1,	    exclam	],
+			[		exclam,		 1	],
+			[		exclam,		 1	]	};
+    key	<AB01> {	[		z,		 Z	],
+			[            0xd1,            0xf1	],
+			[     Cyrillic_ya,     Cyrillic_YA	]	};
+    key	<AC02> {	[		s,		 S	],
+			[            0xd9,            0xf9	],
+			[   Cyrillic_yeru,   Cyrillic_YERU	]	};
+    key	<AC01> {	[		a,		 A	],
+			[            0xc6,            0xe6	],
+			[     Cyrillic_ef,     Cyrillic_EF	]	};
+    key	<AD02> {	[		w,		 W	],
+			[            0xc3,            0xe3	],
+			[    Cyrillic_tse,    Cyrillic_TSE	]	};
+    key	<AE02> {	[		2,		at	],
+			[		quotedbl,	 2	],
+			[		quotedbl,	 2	]	};
+    key	<AB03> {	[		c,		 C	],
+			[            0xd3,            0xf3	],
+			[     Cyrillic_es,     Cyrillic_ES	]	};
+    key	<AB02> {	[		x,		 X	],
+			[            0xde,            0xfe	],
+			[    Cyrillic_che,    Cyrillic_CHE	]	};
+    key	<AC03> {	[		d,		 D	],
+			[            0xd7,            0xf7	],
+			[     Cyrillic_ve,     Cyrillic_VE	]	};
+    key	<AD03> {	[		e,		 E	],
+			[            0xd5,            0xf5	],
+			[      Cyrillic_u,	Cyrillic_U	]	};
+    key	<AE04> {	[		4,	    dollar	],
+			[		asterisk,	 4	],
+			[		asterisk,	 4	]	};
+    key	<AE03> {	[		3,	numbersign	],
+			[		apostrophe,	 3	],
+			[		apostrophe,	 3	]	};
+    key	<AB04> {	[		v,		 V	],
+			[            0xcd,            0xed	],
+			[     Cyrillic_em,     Cyrillic_EM	]	};
+    key	<AC04> {	[		f,		 F	],
+			[            0xc1,            0xe1	],
+			[      Cyrillic_a,	Cyrillic_A	]	};
+    key	<AD05> {	[		t,		 T	],
+			[            0xc5,            0xe5	],
+			[     Cyrillic_ie,     Cyrillic_IE	]	};
+    key	<AD04> {	[		r,		 R	],
+			[            0xcb,            0xeb	],
+			[     Cyrillic_ka,     Cyrillic_KA	]	};
+    key	<AE05> {	[		5,	   percent	],
+			[		colon,		 5	],
+			[		colon,		 5	]	};
+    key	<AB06> {	[		n,		 N	],
+			[            0xd4,            0xf4	],
+			[     Cyrillic_te,     Cyrillic_TE	]	};
+    key	<AB05> {	[		b,		 B	],
+			[            0xc9,            0xe9	],
+			[      Cyrillic_i,	Cyrillic_I	]	};
+    key	<AC06> {	[		h,		 H	],
+			[            0xd2,            0xf2	],
+			[     Cyrillic_er,     Cyrillic_ER	]	};
+    key	<AC05> {	[		g,		 G	],
+			[            0xd0,            0xf0	],
+			[     Cyrillic_pe,     Cyrillic_PE	]	};
+    key	<AD06> {	[		y,		 Y	],
+			[            0xce,            0xee	],
+			[     Cyrillic_en,     Cyrillic_EN	]	};
+    key	<AE06> {	[		6,     asciicircum	],
+			[		comma,		 6	],
+			[		comma,		 6	]	};
+    key	<AB07> {	[		m,		 M	],
+			[            0xd8,            0xf8	],
+			[Cyrillic_softsign,Cyrillic_SOFTSIGN	]	};
+    key	<AC07> {	[		j,		 J	],
+			[            0xcf,            0xef	],
+			[      Cyrillic_o,	Cyrillic_O	]	};
+    key	<AD07> {	[		u,		 U	],
+			[            0xc7,            0xe7	],
+			[    Cyrillic_ghe,    Cyrillic_GHE	]	};
+    key	<AE07> {	[		7,	 ampersand	],
+			[		period,		 7	],
+			[		period,		 7	]	};
+    key	<AE08> {	[		8,	  asterisk	],
+			[		semicolon,	 8	],
+			[		semicolon,	 8	]	};
+    key	<AB08> {	[	    comma,	      less	],
+			[            0xc2,            0xe2	],
+			[     Cyrillic_be,     Cyrillic_BE	]	};
+    key	<AC08> {	[		k,		 K	],
+			[            0xcc,            0xec	],
+			[     Cyrillic_el,     Cyrillic_EL	]	};
+    key	<AD08> {	[		i,		 I	],
+			[            0xdb,            0xfb	],
+			[    Cyrillic_sha,    Cyrillic_SHA	]	};
+    key	<AD09> {	[		o,		 O	],
+			[            0xdd,            0xfd	],
+			[  Cyrillic_shcha,  Cyrillic_SHCHA	]	};
+    key	<AE10> {	[		0,	parenright	],
+			[		parenright,	 0	],
+			[		parenright,	 0	]	};
+    key	<AE09> {	[		9,	 parenleft	],
+			[		parenleft,	 9	],
+			[		parenleft,	 9	]	};
+    key	<AB09> {	[	   period,	   greater	],
+			[            0xc0,            0xe0	],
+			[     Cyrillic_yu,     Cyrillic_YU	]	};
+    key	<AB10> {	[	    slash,	  question	],
+			[	    slash,	  question	]	};
+    key	<AC09> {	[		l,		 L	],
+			[            0xc4,            0xe4	],
+			[     Cyrillic_de,     Cyrillic_DE	]	};
+    key	<AC10> {	[	semicolon,	     colon	],
+			[            0xd6,            0xf6	],
+			[    Cyrillic_zhe,    Cyrillic_ZHE	]	};
+    key	<AD10> {	[		p,		 P	],
+			[            0xda,            0xfa	],
+			[     Cyrillic_ze,     Cyrillic_ZE	]	};
+    key	<AE11> {	[	    minus,	underscore	],
+			[	    minus,	underscore	]	};
+    key	<AC11> {	[      apostrophe,	  quotedbl	],
+			[            0xdc,            0xfc	],
+			[      Cyrillic_e,	Cyrillic_E	]	};
+    key	<AD11> {	[     bracketleft,	 braceleft	],
+			[            0xc8,            0xe8	],
+			[     Cyrillic_ha,     Cyrillic_HA	]	};
+    key	<AE12> {	[	    equal,	      plus	],
+			[	    equal,	      plus	]	};
+    key	<AD12> {	[    bracketright,	braceright	],
+			[            0xdf,            0xff	],
+			[Cyrillic_hardsign,Cyrillic_HARDSIGN	]	};
+    key	<BKSL> {	[	backslash,	       bar	],
+			[	backslash,             bar	]	};
+
+    // Begin modifier mappings 
+
+    modifier_map Shift  { Shift_L };
+    modifier_map Lock   { Caps_Lock, ISO_Lock };
+    modifier_map Control{ Control_L };
+    modifier_map Mod3   { Mode_switch };
+};
+
--- xxkb-1.11.orig/Imakefile
+++ xxkb-1.11/Imakefile
@@ -3,7 +3,7 @@
 BASEDIR = $(PROJECTROOT)
 BINDIR = $(BASEDIR)/bin
 PIXMAPDIR = $(BASEDIR)/share/xxkb
-LOCAL_LIBRARIES = $(XONLYLIB) -lXpm -lXt
+LOCAL_LIBRARIES = $(XONLYLIB) -lXpm -lXt -lXext
 
 OBJS = xxkb.o resource.o wlist.o
 PIXMAPS = pixmaps/by15.xpm pixmaps/by48.xpm \
--- xxkb-1.11.orig/3gr.html
+++ xxkb-1.11/3gr.html
@@ -0,0 +1,56 @@
+<html>
+<title>Три раскладки в xxkb</title>
+
+<body bgcolor=#FFFFFF>
+
+  <H2>Как с помошью xxkb работать более чем с двумя
+раскладками клавиатуры</H2>
+
+<P> Одной из проблем при руссификации "иксов" является то, что некоторые
+приложения ("правильные") нормально вводят русские буквы, если в раскладке
+клавиатуры они закодированы двубайтными кодами - Cyrillic_A, Cyrillic_BE
+и т.д., а другие ("неправильные") понимают только однобайтные коды
+(которые используются в xmodmap).
+
+<P> Попытки заставить все приложения работать с одной и той же раскладкой,
+не всегда приводят к успеху.
+
+<P> При этом существует достаточно простое решение - держать в системе
+две "русских" раскладки - для "правильных" и "неправильных" программ.
+Неудобство в том, что вместе с латинской получается уже три раскладки и
+переключаться между ними с помощью одной клавиши становится очень неудобно.
+
+<P> При использовнии "штатного средства" -
+<a href=http://www.tsu.ru/~pascal/other/xkb/>клавиатурного модуля XKB</a>,
+можно с помощью файлов конфигурации этого XKB загрузить сразу три раскладки
+клавиатуры (вообще-то их может быть до 4-х). Но тогда клавиша-переключатель
+будет иметь не два состояния (лат/рус), а циклически перебирать все три
+раскладки.
+
+<P> В этом случае вам может облегчить жизнь программка xxkb.
+<ul>
+ <li> Предположим, что у вас уже есть подходящий файл конфигурации XKB
+с тремя раскладками клавиатуры. Если его нет, то можно взять
+<a href=http://www.tsu.ru/~pascal/other/xkb/ru-koi-3gr>здесь</a> и
+положить в <B>X11R6/lib/X11/xkb/symbols</B> (вам придется или переименовать
+его в <b>ru</b>, или подправить соответствующие файлы конфигурации XKB).
+ <li> Если вы используете вышеуказанный файл, то "правильная" раскладка
+(которая и должна использоваться "по умолчанию") в нем имеет номер группы 3,
+а "неправильная" - 2.
+ <li> Исправьте в файле настроек xxkb строчку
+<br> <b> XXkb.group.alt: 3 </b>
+<br> и, если вы хотите, назначение "флажков-иконок"
+<br> <b> XXkb.mainwindow.xpm.2: su48.xpm</b>
+<br> <b> XXkb.mainwindow.xpm.3: ru48.xpm</b>
+<br> <b> XXkb.button.xpm.2: su48.xpm</b>
+<br> <b> XXkb.button.xpm.3: ru48.xpm</b>
+ <li> Теперь, если у вас запущен xxkb, то каждом стартующем приложении
+"по умолчанию" будут переключаться две группы - "лат"/"правильный_рус".
+А для тех приложений, которым нужна "старая" раскладка можно правой кнопкой
+мыши выбрать "старую" русскую раскладку (она будет отображаться "советским"
+флажком). И в дальнейшем, в этом приложении тот же переключатель клавиатуры
+будут переключаться между "лат" и "старый_рус".
+</ul>
+
+</body>
+</html>
--- xxkb-1.11.orig/resource.c
+++ xxkb-1.11/resource.c
@@ -558,9 +558,7 @@
 	char *homedir, *filename;
 	char *str_list, *res_app_list, res_ctrls[256];
 	int i, j;
-#ifndef XT_RESOURCE_SEARCH
 	size_t len;
-#endif
 
 	/*
 	 * Init pixbuf engine
@@ -587,7 +585,10 @@
 	}
 	sprintf(filename, "%s/%s", APPDEFDIR, APPDEFFILE);
 #endif
-	db = XrmGetFileDatabase(filename);
+	if (filename)
+		db = XrmGetFileDatabase(filename);
+	else
+		db = NULL;
 	if (db == NULL) {
 		/*
 		 * this situation is not fatal if the user has all
@@ -618,12 +619,30 @@
 	/*
 	 * merge settings
 	 */
-	stat = XrmCombineFileDatabase(filename, &db, True);
+	if (filename)
+		stat = XrmCombineFileDatabase(filename, &db, True);
+	else
+		stat = NULL;
 	if (stat == 0 && db == NULL) {
 		/* failed */
 		warnx("Unable to find configuration data");
 		return 5;
 	}
+
+	if (!filename) {
+		/*
+		 * In case the user did not have a config file we have to come
+		 * up with one
+		 */
+		len = strlen(homedir) + 1 + strlen(USERDEFFILE);
+		filename = malloc(len + 1);
+		if (filename == NULL) {
+			warn(NULL);
+			XrmDestroyDatabase(db);
+			return 1;
+		}
+		sprintf(filename, "%s/%s", homedir, USERDEFFILE);
+	}
 	/*
 	 * start with the conf object
 	 */
--- xxkb-1.11.orig/debian/control
+++ xxkb-1.11/debian/control
@@ -0,0 +1,22 @@
+Source: xxkb
+Section: x11
+Priority: optional
+Build-Depends: debhelper (>= 7.0), xutils-dev, libx11-dev, libxpm-dev, libxt-dev, libxext-dev
+Maintainer: Artem Chuprina <ran@ran.pp.ru>
+Standards-Version: 3.8.3
+Homepage: http://sourceforge.net/projects/xxkb/
+
+Package: xxkb
+Architecture: any
+Depends: ${shlibs:Depends}, ${misc:Depends}
+Description: Keyboard state indicator and switcher for xkb
+ This program is a keyboard state indicator and switcher for xkb. Features:
+  - shows current xkb group (pixmap in its own window)
+  - allows switch group by mouse click
+  - allows individual state for every window
+  - can install its own button (indicator/mouse switcher) on every window's
+    title bar
+  - can restrict keyboard states for every window to only two ("main group" -
+    "alternative group") if xkb set up for more than two groups.
+ Bugs:
+  - documentation is partially in Russian (koi8-r charset) only
--- xxkb-1.11.orig/debian/README.Debian
+++ xxkb-1.11/debian/README.Debian
@@ -0,0 +1,9 @@
+xxkb for Debian
+----------------------
+
+Added 3gr.html (http://pascal.tsu.ru/other/xxkb/3gr.html) and ru-koi-3gr 
+(http://pascal.tsu.ru/other/xkb/ru-koi-3gr) - instructions how to use 3
+keyboard layouts with XKB and Russian XKB symbols with 3 groups to make work
+with Russian input both XKB-aware and not XKB-aware applications.
+
+ -- Artem Chuprina <ran@ran.pp.ru>, Fri Jan 28 20:12:56 MSK 2000
--- xxkb-1.11.orig/debian/docs
+++ xxkb-1.11/debian/docs
@@ -0,0 +1,5 @@
+README-Linux.koi8
+README.koi8
+README
+3gr.html
+ru-koi-3gr
--- xxkb-1.11.orig/debian/copyright
+++ xxkb-1.11/debian/copyright
@@ -0,0 +1,11 @@
+This package was debianized by Artem Chuprina <ran@ran.pp.ru>
+
+It was downloaded from http://xxkb.sourceforge.net
+
+Copyright (c) 1999-2009, by Ivan Pascal <pascal@tsu.ru>
+
+Artistic license. On Debian systems, full text of Artistic
+license is alailable in /usr/share/common-licenses/Artistic.
+
+Debian packaging is (C) 2000-2009 Artem Chuprina <ran@ran.pp.ru>
+and is licensed under the Artistic, see /usr/share/common-licenses/Artistic.
--- xxkb-1.11.orig/debian/watch
+++ xxkb-1.11/debian/watch
@@ -0,0 +1,2 @@
+version=3
+http://sf.net/xxkb/xxkb-(\d+(?:\.\d+)+)-src\.tar\.gz
--- xxkb-1.11.orig/debian/postrm.debhelper
+++ xxkb-1.11/debian/postrm.debhelper
@@ -0,0 +1,3 @@
+# Automatically added by dh_installmenu
+if [ -x "`which update-menus 2>/dev/null`" ]; then update-menus ; fi
+# End automatically added section
--- xxkb-1.11.orig/debian/menu
+++ xxkb-1.11/debian/menu
@@ -0,0 +1,2 @@
+?package(xxkb):needs="x11" section="Applications/System/Administration" \
+  title="xxkb" command="/usr/bin/xxkb"
--- xxkb-1.11.orig/debian/debhelper.log
+++ xxkb-1.11/debian/debhelper.log
@@ -0,0 +1,13 @@
+dh_installdirs
+dh_installdocs
+dh_installmenu
+dh_installmanpages
+dh_installchangelogs
+dh_link
+dh_strip
+dh_compress
+dh_installdeb
+dh_shlibdeps
+dh_gencontrol
+dh_md5sums
+dh_builddeb
--- xxkb-1.11.orig/debian/compat
+++ xxkb-1.11/debian/compat
@@ -0,0 +1 @@
+7
--- xxkb-1.11.orig/debian/postinst.debhelper
+++ xxkb-1.11/debian/postinst.debhelper
@@ -0,0 +1,5 @@
+# Automatically added by dh_installmenu
+if [ "$1" = "configure" ] && [ -x "`which update-menus 2>/dev/null`" ]; then
+	update-menus
+fi
+# End automatically added section
--- xxkb-1.11.orig/debian/substvars
+++ xxkb-1.11/debian/substvars
@@ -0,0 +1,2 @@
+shlibs:Depends=libc6 (>= 2.0), libx11-6, libxext6, libxpm4, libxt6
+misc:Depends=
--- xxkb-1.11.orig/debian/changelog
+++ xxkb-1.11/debian/changelog
@@ -0,0 +1,156 @@
+xxkb (1.11-2.1) unstable; urgency=low
+
+  * Non-maintainer upload.
+  * Fix "segfaults on startup": add patch from Sebastian Andrzej Siewior to
+    handle missing config file (closes: #562033).
+
+ -- gregor herrmann <gregoa@debian.org>  Sun, 17 Jan 2010 23:45:28 +0100
+
+xxkb (1.11-2) unstable; urgency=low
+
+  [ Artem Chuprina ]
+  * Build-Depends on xutils-dev
+  [ UNera ]
+  * Standards-Version bumped to 3.8.3:
+    - removed DH_COMPAT option from debian/rules.
+    - added call to dh_fixperms (closes a few lintian warnings).
+    - make clean errors aren't ignored (lintian warning).
+    - use dh_prep instead obsolete dh_clean -k (lintian warning).
+    - added copyright record like in source files, added link to
+      /usr/share/common-licenses/Artistic in copyright (lintian warning).
+    - menuitem was moved from obsolete Apps/Tools to Applications/System
+      (lintian warning).
+    - 'Homepage' was moved from description to control's tag (lintian warning).
+    - Fixed error in manpage (lintian warning).
+  * Merge Artem's changelog and unstable.
+  * Added debian/watch.
+
+ -- Artem Chuprina <ran@ran.pp.ru>  Sat, 19 Sep 2009 23:26:00 +0400
+
+xxkb (1.11-1) unstable; urgency=low
+
+  * New upstream version
+
+ -- Artem Chuprina <ran@ran.pp.ru>  Sat,  1 Nov 2008 19:17:20 +0300
+
+xxkb (1.10-2.2) unstable; urgency=medium
+
+  * Non-maintainer upload.
+  * Fix FTBFS by replacing obsolete xutils with xutils-dev in
+    Build-Depends (Closes: #485732).
+  * Set urgency to Б─°mediumБ─² as this bug affects testing too.
+  * Remove obsolete x-dev from Build-Depends too (Closes: #515483).
+  * debian/control: Move the Homepage where it belongs.
+  * debian/rules: Uncomment the dh_fixperms call (Closes: #391850).
+  * Menu transition: move from obsolete Б─°Apps/ToolsБ─² to
+    Б─°Applications/System/AdministrationБ─².
+
+ -- Cyril Brulebois <kibi@debian.org>  Sun, 21 Jul 2009 03:48:31 +0200
+
+xxkb (1.10-2.1) unstable; urgency=high
+
+  * Non-maintainer upload.
+  * Split xlibs-dev build-dep (Closes: #346869).
+
+ -- Luk Claes <luk@debian.org>  Fri, 20 Jan 2006 17:21:36 +0100
+
+xxkb (1.10-2) unstable; urgency=medium
+
+  * Fixed homepage
+  * List in the package description beautified (closes: #135082, #159175)
+  * Unreproducible bug #208012 reported as not present in the new version by
+    submitter (closes: #208012)
+  * Applied patch from Andreas Jochens <aj@andaco.de> (closes: #264160)
+
+ -- Artem Chuprina <ran@ran.pp.ru>  Tue, 28 Sep 2004 20:03:09 +0400
+
+xxkb (1.10-1) unstable; urgency=low
+
+  * New upstream version
+  * I hope this really (Closes: #114611)
+
+ -- Artem Chuprina <ran@ran.pp.ru>  Fri,  6 Jun 2003 18:23:43 +0400
+
+xxkb (1.8-1) unstable; urgency=low
+
+  * New upstream version
+
+ -- Artem Chuprina <ran@ran.pp.ru>  Sat, 10 Aug 2002 17:40:57 +0400
+
+xxkb (1.6-4) unstable; urgency=low
+
+  * Included dependency on debhelper (Closes: #124147)
+  * Fixed minor spelling errors (Closes: #125572)
+
+ -- Artem Chuprina <ran@ran.pp.ru>  Tue, 18 Dec 2001 14:24:20 +0300
+
+xxkb (1.6-3) unstable; urgency=low
+
+  * Minor corrections in documentation; fixed bug in packing of previous package
+  * Changed PIXMAPDIR and target in Imakefile
+  * Formally changed menu entry (really changed in previous versions).
+    (Closes: #114611)
+
+ -- Artem Chuprina <ran@ran.pp.ru>  Sat, 15 Dec 2001 15:13:22 +0300
+
+xxkb (1.6-2) unstable; urgency=low
+
+  * Technical rebuild
+
+ -- Artem Chuprina <ran@ran.pp.ru>  Fri,  9 Mar 2001 16:19:07 +0300
+
+xxkb (1.6-1) unstable; urgency=low
+
+  * New uptream verion. Really moved /usr/X11R6/lib/X11/app-defaults/XXkb to /etc/X11/app-defaults/XXkb. (Closes: #86340). I mean it. No kidding.
+
+ -- Daniel Ginsburg <dg@warpsolutions.com>  Tue,  6 Mar 2001 19:11:08 +0300
+
+xxkb (1.5.1-3) unstable; urgency=low
+
+  * Changed conffiles from /usr/X11R6/lib/X11/app-defaults/XXkb to
+    /etc/X11/app-defaults/XXkb. (Closes: #85385)
+
+ -- Peter Novodvorsky <nidd@debian.org>  Sun, 11 Feb 2001 07:30:51 +0300
+
+xxkb (1.5.1-2) unstable; urgency=low
+
+  * Pixmap dir moved to /usr/share/xxkb
+  * Added translation of README into English
+  * Sponsor upload by Peter Novodvorsky.  (closes: #81145)
+
+ -- Artem Chuprina <ran@ran.pp.ru>  Thu,  1 Feb 2001 05:05:49 +0300
+
+xxkb (1.5.1-1) unstable; urgency=low
+
+  * New upstream version: incorporated fix from 1.5-2 and fixed bug when xxkb
+    did not work with mainwindow turned off.
+
+ -- Artem Chuprina <ran@ran.pp.ru>  Fri, 24 Nov 2000 01:23:51 +0300
+
+xxkb (1.5-2) unstable; urgency=low
+
+  * fight with wm_class_class
+
+ -- Artem Chuprina <ran@ran.pp.ru>  Thu,  9 Nov 2000 15:40:41 +0300
+
+xxkb (1.5-1) unstable; urgency=low
+
+  * New upstream version
+
+ -- Artem Chuprina <ran@ran.pp.ru>  Mon, 23 Oct 2000 22:58:01 +0400
+
+xxkb (1.4-1) unstable; urgency=low
+
+  * New upstream version
+
+ -- Artem Chuprina <ran@ran.pp.ru>  Mon, 15 May 2000 12:36:08 +0400
+
+xxkb (1.3-1) unstable; urgency=low
+
+  * Moved to new upstream version and debhelper
+
+ -- Artem Chuprina <ran@ran.pp.ru>  Fri, 28 Jan 2000 20:35:02 +0300
+
+Local variables:
+mode: debian-changelog
+End:
--- xxkb-1.11.orig/debian/install
+++ xxkb-1.11/debian/install
@@ -0,0 +1,2 @@
+etc
+usr
--- xxkb-1.11.orig/debian/rules
+++ xxkb-1.11/debian/rules
@@ -0,0 +1,70 @@
+#!/usr/bin/make -f
+# Sample debian/rules that uses debhelper.
+# GNU copyright 1997 to 1999 by Joey Hess.
+
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+
+build: build-stamp
+build-stamp:
+	dh_testdir
+	xmkmf
+	$(MAKE)
+	touch build-stamp
+
+clean:
+	dh_testdir
+	dh_testroot
+	rm -f build-stamp
+	[ ! -f Makefile ] || $(MAKE) distclean
+	dh_clean
+
+install: build
+	dh_testdir
+	dh_testroot
+	dh_prep
+	dh_installdirs
+	# Add here commands to install the package into debian/tmp.
+	$(MAKE) install install.man DESTDIR=`pwd`/debian/tmp
+	mv debian/tmp/usr/share/man/man1/xxkb.1x \
+		debian/tmp/usr/share/man/man1/xxkb.1
+
+
+# Build architecture-independent files here.
+binary-indep: build install
+# We have nothing to do by default.
+
+# Build architecture-dependent files here.
+binary-arch: build install
+#	dh_testversion
+	dh_testdir
+	dh_testroot
+#	dh_installdebconf	
+	dh_installdocs
+#	dh_installexamples
+	dh_installmenu
+	dh_install
+#	dh_installemacsen
+#	dh_installpam
+#	dh_installinit
+#	dh_installcron
+	dh_installmanpages
+#	dh_installinfo
+#	dh_undocumented
+	dh_installchangelogs CHANGES.koi8
+	dh_link
+	dh_strip
+	dh_compress
+	dh_fixperms
+# You may want to make some executables suid here.
+#	dh_suidregister
+#	dh_makeshlibs
+	dh_installdeb
+#	dh_perl
+	dh_shlibdeps
+	dh_gencontrol
+	dh_md5sums
+	dh_builddeb
+
+binary: binary-indep binary-arch
+.PHONY: build clean binary-indep binary-arch binary install
