--- linuxwacom-0.8.2-1.orig/src/xdrv/xf86Wacom.h	2009-01-15 14:57:45.000000000 -0500
+++ linuxwacom-0.8.2-1/src/xdrv/xf86Wacom.h	2009-01-15 14:59:33.000000000 -0500
@@ -23,7 +23,7 @@
 /****************************************************************************/
 
 #include "../include/xdrv-config.h"
-#ifdef WCM_XORG_XSERVER_1_6
+#ifdef WCM_XORG
    #include <xorg-server.h>
    #include <xorgVersion.h>
 #else
--- linuxwacom-0.8.2-1.orig/src/xdrv/wcmConfig.c	2008-12-30 16:46:01.000000000 -0500
+++ linuxwacom-0.8.2-1/src/xdrv/wcmConfig.c	2009-01-15 15:11:13.000000000 -0500
@@ -994,7 +994,7 @@ static XF86ModuleVersionInfo xf86WcmVers
 	MODULEVENDORSTRING,
 	MODINFOSTRING1,
 	MODINFOSTRING2,
-#ifdef WCM_XORG_XSERVER_1_6
+#ifdef WCM_XORG
 	XORG_VERSION_CURRENT,
 #else
 	XF86_VERSION_CURRENT,
