--- text-short-0.1.5/text-short.cabal	2001-09-09 03:46:40.000000000 +0200
+++ text-short-0.1.5.new/text-short.cabal	2023-03-20 16:40:42.665078299 +0100
@@ -32,14 +32,14 @@
 
   other-modules:       Data.Text.Short.Internal
 
-  build-depends:       base        >= 4.7    && < 4.17
+  build-depends:       base        >= 4.7    && < 4.19
                      , bytestring  >= 0.10.4 && < 0.12
                      , hashable    >= 1.2.6  && < 1.5
                      , deepseq     >= 1.3    && < 1.5
                      , text        >= 1.0    && < 1.3 || >=2.0 && <2.1
                      , binary      >= 0.7.1  && < 0.9
-                     , ghc-prim    >= 0.3.1  && < 0.9
-                     , template-haskell >= 2.9.0.0 && <2.19
+                     , ghc-prim    >= 0.3.1  && < 0.11
+                     , template-haskell >= 2.9.0.0 && <2.21
 
   if impl(ghc >=8.0)
     build-depends: bytestring >= 0.10.8.0
