diff -Naur splashy-0.2.0.orig/Makefile.am splashy-0.2.0/Makefile.am
--- splashy-0.2.0.orig/Makefile.am	2006-10-15 21:22:31.000000000 +0200
+++ splashy-0.2.0/Makefile.am	2006-10-18 11:45:00.000000000 +0200
@@ -21,6 +21,7 @@
 		chmod 755 $(schemasdir)/splashy/schemas; \
 	fi
 	cp -r themes/default/ $(imagesdir)
+	cp -r themes/frugalware/ $(imagesdir)
 	cp -r schemas/ $(schemasdir)/splashy/schemas/
 	if test '!' -d $(DESTDIR)/$(sysconfdir)/default; then \
 		$(mkinstalldirs) $(DESTDIR)/$(sysconfdir)/default; \
@@ -33,6 +34,7 @@
 
 uninstall-hook:
 	rm -fr $(imagesdir)/default \
+	$(imagesdir)/frugalware \
 	$(datadir)/splashy/schemas \
 	$(DESTDIR)/$(sysconfdir)/default/splashy
 
@@ -44,6 +46,9 @@
 	     themes/default/error.jpg \
 	     themes/default/theme.xml \
 	     themes/default/FreeSans.ttf \
+	     themes/frugalware/background.jpg \
+	     themes/frugalware/theme.xml \
+	     themes/frugalware/FreeSans.ttf \
 	     schemas/theme.xsd \
 	     schemas/theme.xsl \
              scripts/default/splashy \
diff -Naur splashy-0.2.0.orig/doc/config.xml.in splashy-0.2.0/doc/config.xml.in
--- splashy-0.2.0.orig/doc/config.xml.in	2006-10-15 21:22:31.000000000 +0200
+++ splashy-0.2.0/doc/config.xml.in	2006-10-18 11:45:12.000000000 +0200
@@ -4,7 +4,7 @@
     <!-- themes directory: conventional path @EXPANDED_SYSCONFDIR@/splashy/themes -->
     <themes>@EXPANDED_SYSCONFDIR@/splashy/themes</themes>
     <!-- current theme could be relative the themes defined above or full path -->
-    <current_theme>default</current_theme>
+    <current_theme>frugalware</current_theme>
     <!-- full path to theme to fall back in case of problems. DO NOT CHANGE -->
     <default_theme>@EXPANDED_SYSCONFDIR@/splashy/themes/default</default_theme>
     <pid>/etc/splashy/splashy.pid</pid>
