diff -Naur splashy-0.3.0.orig/doc/config.xml.in splashy-0.3.0/doc/config.xml.in
--- splashy-0.3.0.orig/doc/config.xml.in	2007-01-23 21:20:57.000000000 +0100
+++ splashy-0.3.0/doc/config.xml.in	2007-01-26 15:00:44.000000000 +0100
@@ -4,7 +4,7 @@
     <!-- themes directory: conventional path @EXPANDED_SYSCONFDIR@/splashy/themes -->
     <themes>@EXPANDED_SYSCONFDIR@/splashy/themes</themes>
     <!-- current theme could be relative the themes defined above or full path -->
-    <current_theme>default</current_theme>
+    <current_theme>frugalware</current_theme>
     <!-- full path to theme to fall back in case of problems. DO NOT CHANGE -->
     <default_theme>@EXPANDED_SYSCONFDIR@/splashy/themes/default</default_theme>
     <pid>/etc/splashy/splashy.pid</pid>
diff -Naur splashy-0.3.0.orig/Makefile.am splashy-0.3.0/Makefile.am
--- splashy-0.3.0.orig/Makefile.am	2007-01-23 21:20:58.000000000 +0100
+++ splashy-0.3.0/Makefile.am	2007-01-26 15:00:26.000000000 +0100
@@ -10,7 +10,10 @@
 	themes/default/background.jpg \
 	themes/default/error.jpg \
 	themes/default/theme.xml \
-	themes/default/FreeSans.ttf 
+	themes/default/FreeSans.ttf \
+	themes/frugalware/background.jpg \
+	themes/frugalware/theme.xml \
+	themes/frugalware/FreeSans.ttf 
 
 
 # This will canonically be installed in /usr/share/splashy
