--- splashy-0.1.8/Makefile.am.orig	2006-07-13 02:46:09.000000000 +0200
+++ splashy-0.1.8/Makefile.am	2006-07-13 02:48:57.000000000 +0200
@@ -21,6 +21,7 @@
 		chmod 755 $(schemasdir)/splashy/schemas; \
 	fi
 	cp -r themes/default/ $(imagesdir)
+	cp -r themes/frugalware/ $(imagesdir)
 	cp -r themes/kubuntusplashy/ $(imagesdir)
 	cp -r themes/debian3/ $(imagesdir)
 	cp -r themes/debiansplashy/ $(imagesdir)
@@ -31,6 +32,7 @@
 
 uninstall-hook:
 	rm -fr $(imagesdir)/default \
+	$(imagesdir)/frugalware \
 	$(imagesdir)/kubuntusplashy \
 	$(imagesdir)/debian3 \
 	$(imagesdir)/debiansplashy \
@@ -47,6 +49,9 @@
 	     themes/default/error.jpg \
 	     themes/default/theme.xml \
 	     themes/default/FreeSans.ttf \
+	     themes/frugalware/background.jpg \
+	     themes/frugalware/theme.xml \
+	     themes/frugalware/FreeSans.ttf \
  	     themes/kubuntusplashy/theme.xml \
  	     themes/kubuntusplashy/README \
  	     themes/kubuntusplashy/shutdown.jpg \
--- splashy-0.1.8/doc/config.xml.in.orig	2006-07-13 02:54:57.000000000 +0200
+++ splashy-0.1.8/doc/config.xml.in	2006-07-13 02:55:06.000000000 +0200
@@ -4,7 +4,7 @@
     <!-- themes directory: conventional path @EXPANDED_SYSCONFDIR@/splashy/themes -->
     <themes>@EXPANDED_SYSCONFDIR@/splashy/themes</themes>
     <!-- current theme could be relative the themes defined above or full path -->
-    <current_theme>default</current_theme>
+    <current_theme>frugalware</current_theme>
     <!-- full path to theme to fall back in case of problems. DO NOT CHANGE -->
     <default_theme>@EXPANDED_SYSCONFDIR@/splashy/themes/default</default_theme>
     <fifo>/etc/splashy/splashy.fifo</fifo>
