diff -Naur splashy-0.3.0.orig/src/splashy_functions.c splashy-0.3.0/src/splashy_functions.c
--- splashy-0.3.0.orig/src/splashy_functions.c	2007-01-23 21:20:54.000000000 +0100
+++ splashy-0.3.0/src/splashy_functions.c	2007-01-26 15:08:24.000000000 +0100
@@ -1102,7 +1102,7 @@
 			{
 				splashy_child_exit ();
 			}
-			else if (key == DIKS_F2)
+			else if (0)
 			{
 				/*
 				 * when F2 is pressed we display a
