--- splashy-0.1.8.1/src/splashy_functions.c.orig	2006-07-27 03:01:26.000000000 +0200
+++ splashy-0.1.8.1/src/splashy_functions.c	2006-07-27 03:01:43.000000000 +0200
@@ -993,7 +993,7 @@
                                 {
                                         splashy_child_exit ();
                                 }
-                                else if (event.key_id == DIKI_F2)
+                                else if (0)
                                 {
                                         /*
                                          * when F2 is pressed we display a
