diff -Naur splashy-0.2.0.orig/src/splashy_main.c splashy-0.2.0/src/splashy_main.c
--- splashy-0.2.0.orig/src/splashy_main.c	2006-10-15 21:22:31.000000000 +0200
+++ splashy-0.2.0/src/splashy_main.c	2006-10-18 16:26:21.000000000 +0200
@@ -102,8 +102,8 @@
                 return 0;
         } else {
 		/* go to an unused tty */
-		int vt_fd = open ("/dev/tty8", O_RDWR);
-		switch_console(8, vt_fd);
+		int vt_fd = open ("/dev/tty1", O_RDWR);
+		switch_console(1, vt_fd);
 		close (vt_fd);
 	}
 
