--- semigroupoids-6.0.1/semigroupoids.cabal	2001-09-09 03:46:40.000000000 +0200
+++ semigroupoids-6.0.1.new/semigroupoids.cabal	2024-07-10 00:58:41.403691623 +0200
@@ -158,7 +158,7 @@
     build-depends: tagged >= 0.8.7 && < 1
 
   if flag(unordered-containers)
-    build-depends: hashable >= 1.2.7.0 && < 1.5,
+    build-depends: hashable >= 1.2.7.0 && < 1.6,
                    unordered-containers >= 0.2.8.0  && < 0.3
 
   hs-source-dirs: src
