--- amSynth-1.2.2/src/GUI/Knob.cc.old	2009-07-12 18:40:37.000000000 +0200
+++ amSynth-1.2.2/src/GUI/Knob.cc	2010-04-20 02:31:46.914164080 +0200
@@ -9,7 +9,7 @@
 using namespace std;
 
 Knob::Knob()
-:	adj(Glib::RefPtr<Gtk::Adjustment>::RefPtr(new Gtk::Adjustment(0.0, 0.0, 1.0, 0.01, 1.0, 0)))
+:	adj(new Gtk::Adjustment(0.0, 0.0, 1.0, 0.01, 1.0, 0))
 ,	frame (0)
 {
 	set_events (Gdk::BUTTON_PRESS_MASK | Gdk::BUTTON_RELEASE_MASK | Gdk::POINTER_MOTION_MASK);
@@ -33,9 +33,8 @@
 }
 
 void
-Knob::set_adjustment(Gtk::Adjustment* adjustment)
+Knob::set_adjustment(Gtk::Adjustment* adj)
 {
-	adj = Glib::RefPtr<Gtk::Adjustment>::RefPtr(adjustment);
 	adj->signal_value_changed().connect(mem_fun(*this, &Knob::on_adj_value_changed));
 	on_adj_value_changed();
 }
@@ -100,7 +99,7 @@
 }
 
 bool
-Knob::on_button_release_event	(GdkEventButton *ev)
+Knob::on_button_release_event	(GdkEventButton *)
 {
 	if(has_grab()) remove_modal_grab ();
 	return TRUE;
