diff -Naur gnupg-1.2.5.orig/po/Makefile.in.in gnupg-1.2.5/po/Makefile.in.in
--- gnupg-1.2.5.orig/po/Makefile.in.in	2002-06-29 15:43:03.000000000 +0200
+++ gnupg-1.2.5/po/Makefile.in.in	2004-08-17 21:55:26.129516384 +0200
@@ -29,7 +29,7 @@
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
 MKINSTALLDIRS = @MKINSTALLDIRS@
-mkinstalldirs = $(SHELL) `case "$(MKINSTALLDIRS)" in /*) echo "$(MKINSTALLDIRS)" ;; *) echo "$(top_builddir)/$(MKINSTALLDIRS)" ;; esac`
+mkinstalldirs = $(SHELL) $(MKINSTALLDIRS)
 
 CC = @CC@
 GMSGFMT = @GMSGFMT@
