diff -Naur git-1.5.2.3.orig/Documentation/Makefile git-1.5.2.3/Documentation/Makefile
--- git-1.5.2.3.orig/Documentation/Makefile	2007-07-02 10:11:39.000000000 +0200
+++ git-1.5.2.3/Documentation/Makefile	2007-07-07 14:49:41.000000000 +0200
@@ -118,13 +118,17 @@
 		$(ASCIIDOC_EXTRA) -agit_version=$(GIT_VERSION) -o $@+ $<
 	mv $@+ $@
 
+DTDDIR=/usr/share/sgml/docbook/dtd/xml-dtd-4.2
+
 %.1 %.5 %.7 : %.xml
-	xmlto -m callouts.xsl man $<
+	xsltproc --nonet --path $(DTDDIR) /etc/asciidoc/docbook-xsl/manpage.xsl $<
+	sed -i 's/\\(bu/*/' $@
 
 %.xml : %.txt
 	rm -f $@+ $@
 	$(ASCIIDOC) -b docbook -d manpage -f asciidoc.conf \
 		$(ASCIIDOC_EXTRA) -agit_version=$(GIT_VERSION) -o $@+ $<
+	sed -i '/<!DOCTYPE/s|\("http[^"].*"\)|"file://$(DTDDIR)/docbookx.dtd"|' $@+
 	mv $@+ $@
 
 user-manual.xml: user-manual.txt user-manual.conf
