diff -ur a/cmake/modules/FindPythonLibs.cmake b/cmake/modules/FindPythonLibs.cmake
--- a/cmake/modules/FindPythonLibs.cmake	2009-09-20 15:14:18.000000000 -0500
+++ b/cmake/modules/FindPythonLibs.cmake	2014-03-18 04:50:23.477664565 -0500
@@ -17,8 +17,10 @@
 
 IF(WIN32)
   FIND_LIBRARY(PYTHON_DEBUG_LIBRARY
-    NAMES python26_d python25_d python24_d python23_d python22_d python21_d python20_d python
+    NAMES python27_d python26_d python25_d python24_d python23_d python22_d python21_d python20_d python
     PATHS
+    [HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\2.7\\InstallPath]/libs/Debug
+    [HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\2.7\\InstallPath]/libs
     [HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\2.6\\InstallPath]/libs/Debug
     [HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\2.6\\InstallPath]/libs
     [HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\2.5\\InstallPath]/libs/Debug
@@ -41,7 +43,8 @@
 ENDIF(WIN32)
 
 FIND_LIBRARY(PYTHON_LIBRARY ${PREFERRED_PYTHON_VERSION}
-  NAMES python26 python2.6
+  NAMES python27 python2.7
+        python26 python2.6
         python25 python2.5
         python24 python2.4
         python23 python2.3
@@ -52,6 +55,7 @@
         python15 python1.5
 
   PATHS
+    [HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\2.7\\InstallPath]/libs
     [HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\2.6\\InstallPath]/libs
     [HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\2.5\\InstallPath]/libs
     [HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\2.4\\InstallPath]/libs
@@ -63,6 +67,7 @@
     [HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\1.5\\InstallPath]/libs
 
   PATH_SUFFIXES
+    python2.7/config
     python2.6/config
     python2.5/config
     python2.4/config
@@ -82,7 +87,7 @@
 SET(PYTHON_FRAMEWORK_INCLUDES)
 IF(Python_FRAMEWORKS)
   IF(NOT PYTHON_INCLUDE_PATH)
-    FOREACH(version 2.6 2.5 2.4 2.3 2.2 2.1 2.0 1.6 1.5)
+    FOREACH(version 2.7 2.6 2.5 2.4 2.3 2.2 2.1 2.0 1.6 1.5)
       FOREACH(dir ${Python_FRAMEWORKS})
         SET(PYTHON_FRAMEWORK_INCLUDES ${PYTHON_FRAMEWORK_INCLUDES}
           ${dir}/Versions/${version}/include/python${version})
@@ -96,6 +101,7 @@
 
   PATHS
     ${PYTHON_FRAMEWORK_INCLUDES}
+    [HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\2.7\\InstallPath]/include
     [HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\2.6\\InstallPath]/include
     [HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\2.5\\InstallPath]/include
     [HKEY_LOCAL_MACHINE\\SOFTWARE\\Python\\PythonCore\\2.4\\InstallPath]/include
@@ -108,6 +114,7 @@
 
   PATH_SUFFIXES
     ${PREFERRED_PYTHON_VERSION}
+    python2.7
     python2.6
     python2.5
     python2.4
diff -ur a/opensync/common/opensync_list.c b/opensync/common/opensync_list.c
--- a/opensync/common/opensync_list.c	2009-09-20 15:14:18.000000000 -0500
+++ b/opensync/common/opensync_list.c	2014-03-18 04:45:49.718674641 -0500
@@ -30,7 +30,7 @@
  * MT safe
  */
 
-#include <glib/gmem.h>
+#include <glib.h>
 #include "opensync_list.h"
 #include "opensync_internals.h"
 
diff -ur a/opensync/opensync_internals.h b/opensync/opensync_internals.h
--- a/opensync/opensync_internals.h	2009-09-20 15:14:18.000000000 -0500
+++ b/opensync/opensync_internals.h	2014-03-18 04:45:35.274675172 -0500
@@ -24,8 +24,6 @@
 #include <glib.h>
 #include <gmodule.h>
 #include <string.h>
-#include <glib/gprintf.h>
-#include <glib/gstdio.h>
 #include <sys/stat.h>
 
 #include "config.h"
diff -ur a/wrapper/opensync-error.i b/wrapper/opensync-error.i
--- a/wrapper/opensync-error.i	2009-09-20 15:14:18.000000000 -0500
+++ b/wrapper/opensync-error.i	2014-03-18 04:53:39.543657349 -0500
@@ -46,7 +46,7 @@
 		osync_error_set_from_error(&self, &source);
 	}
 
-	void set(ErrorType type, const char *msg) {
+	void set_(ErrorType type, const char *msg) {
 		osync_error_set(&self, type, "%s", msg);
 	}
 
