From: http://dev.rubyonrails.org/changeset/6894?format=diff&new=6894
Index: actionpack/lib/action_controller/assertions/selector_assertions.rb
===================================================================
--- actionpack/lib/action_controller/assertions/selector_assertions.rb (revision 6861)
+++ actionpack/lib/action_controller/assertions/selector_assertions.rb (revision 6894)
@@ -587,4 +587,6 @@
           unescaped= rjs_string.gsub('\"', '"')
           unescaped.gsub!('\n', "\n")
+          unescaped.gsub!('\076', '>')
+          unescaped.gsub!('\074', '<')
           # RJS encodes non-ascii characters.
           unescaped.gsub!(RJS_PATTERN_UNICODE_ESCAPED_CHAR) {|u| [$1.hex].pack('U*')}
