From: Sebastian Ramacher <sramacher@debian.org>
Date: Fri, 1 Dec 2023 23:22:16 +0100
Subject: Do not use contribs

---
 gtk/meson.build | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/gtk/meson.build b/gtk/meson.build
index 6f41201..c072f50 100644
--- a/gtk/meson.build
+++ b/gtk/meson.build
@@ -22,8 +22,8 @@ if hb_dir == ''
   hb_dir = meson.current_build_dir() / '..'
 endif
 
-hb_libdirs = [hb_dir / 'libhb', hb_dir / 'contrib/lib'] + get_option('extra-libdirs')
-hb_incdirs = include_directories(hb_dir / 'libhb', hb_dir / 'contrib/include')
+hb_libdirs = [hb_dir / 'libhb'] + get_option('extra-libdirs')
+hb_incdirs = include_directories(hb_dir / 'libhb')
 
 # External dependencies (required)
 ghb_deps = [
@@ -33,7 +33,6 @@ ghb_deps = [
   cc.find_library('mp3lame', dirs: hb_libdirs),
   dependency('dvdnav'),
   dependency('dvdread'),
-  dependency('dav1d'),
   dependency('SvtAv1Enc'),
   dependency('fribidi'),
   dependency('glib-2.0', version: glib_min),
@@ -63,7 +62,6 @@ ghb_deps = [
   dependency('vorbisenc'),
   dependency('vpx'),
   dependency('x264'),
-  dependency('zimg'),
   dependency('zlib'),
 ]
 
