diff --color -Naur fizz-2025.02.17.00/fizz/CMakeLists.txt fizz-2025.02.17.00.new/fizz/CMakeLists.txt
--- fizz-2025.02.17.00/fizz/CMakeLists.txt	2025-01-30 18:33:57.000000000 +0100
+++ fizz-2025.02.17.00.new/fizz/CMakeLists.txt	2025-02-20 17:46:20.969502277 +0100
@@ -50,7 +50,7 @@
 find_package(fmt CONFIG REQUIRED)
 
 find_package(OpenSSL REQUIRED)
-find_package(Glog REQUIRED)
+find_package(Glog CONFIG REQUIRED)
 find_package(Threads REQUIRED)
 find_package(Zstd REQUIRED)
 if (UNIX AND NOT APPLE)
