diff -Naur slocate-3.1.orig/Makefile slocate-3.1/Makefile
--- slocate-3.1.orig/Makefile	2006-10-06 13:54:04.000000000 +0200
+++ slocate-3.1/Makefile	2006-10-06 13:54:27.000000000 +0200
@@ -2,8 +2,8 @@
 	make -C src
 install:
 	make -C src install
-	install -d -m 0755 /usr/local/man/man1
-	install -m 0644 -o root -g root doc/slocate.1 /usr/local/man/man1
+	install -d -m 0755 $(DESTDIR)/usr/share/man/man1
+	install -m 0644 -o root -g root doc/slocate.1 $(DESTDIR)/usr/share/man/man1
 	
 uninstall:
 	make -C src uninstall
diff -Naur slocate-3.1.orig/src/Makefile slocate-3.1/src/Makefile
--- slocate-3.1.orig/src/Makefile	2006-10-06 13:54:04.000000000 +0200
+++ slocate-3.1/src/Makefile	2006-10-06 13:54:58.000000000 +0200
@@ -16,8 +16,7 @@
 	$(CC) -c $(CFLAGS) -o $@ $<
 
 install:
-	-groupadd slocate
-	install -m 2755 -o root -g slocate slocate /usr/local/bin/
+	install -m 755 -o root -g 21 slocate $(DESTDIR)/usr/bin/
 	
 uninstall:
 	rm -f /usr/local/bin/slocate
