diff -Naur monit-4.10.1-orig/file.c monit-4.10.1/file.c
--- monit-4.10.1-orig/file.c	2007-08-12 20:02:48.000000000 +0200
+++ monit-4.10.1/file.c	2007-11-26 12:08:21.000000000 +0100
@@ -158,6 +158,11 @@
     return (rcfile);
   }
   memset(rcfile, 0, STRLEN);
+  snprintf(rcfile, STRLEN, "/etc/monit/%s", MONITRC);
+  if(File_exist(rcfile)) {
+    return (rcfile);
+  }
+  memset(rcfile, 0, STRLEN);
   snprintf(rcfile, STRLEN, "/etc/%s", MONITRC);
   if(File_exist(rcfile)) {
     return (rcfile);
@@ -178,8 +183,8 @@
     return (rcfile);
   }
   LogError("%s: Cannot find the control file at "
-      "~/.%s, /etc/%s, %s/%s, /usr/local/etc/%s or at ./%s \n",
-      prog, MONITRC, MONITRC, SYSCONFDIR, MONITRC, MONITRC, MONITRC);
+      "~/.%s, /etc/monit/%s, /etc/%s, %s/%s, /usr/local/etc/%s or at ./%s \n",
+      prog, MONITRC, MONITRC, MONITRC, MONITRC, MONITRC, MONITRC);
   exit(1);
   
 }
