diff -Naur monit-5.4-orig/src/file.c monit-5.4/src/file.c
--- monit-5.4-orig/src/file.c	2012-05-06 11:40:45.000000000 +0200
+++ monit-5.4/src/file.c	2012-06-07 18:51:46.245492057 +0200
@@ -178,6 +178,10 @@
   if(file_exist(rcfile)) {
     return rcfile;
   }
+  snprintf(rcfile, STRLEN, "/etc/monit/%s", MONITRC);
+  if(file_exist(rcfile)) {
+    return rcfile;
+  }
   snprintf(rcfile, STRLEN, "%s/%s", SYSCONFDIR, MONITRC);
   if(file_exist(rcfile)) {
     return rcfile;
