--- file-orig.c	Tue Apr  5 16:56:26 2005
+++ file.c	Tue Apr  5 16:58:55 2005
@@ -155,6 +155,11 @@
     return (rcfile);
   }
   memset(rcfile, 0, STRLEN);
+  snprintf(rcfile, STRLEN, "/etc/monit/%s", MONITRC);
+  if(File_exist(rcfile)) {
+    return (rcfile);
+  }
+  memset(rcfile, 0, STRLEN);
   snprintf(rcfile, STRLEN, "/etc/%s", MONITRC);
   if(File_exist(rcfile)) {
     return (rcfile);
@@ -170,8 +175,8 @@
     return (rcfile);
   }
   LogError("%s: Cannot find the control file at "
-      "~/.%s, /etc/%s, /usr/local/etc/%s or at ./%s \n",
-      prog, MONITRC, MONITRC, MONITRC, MONITRC);
+      "~/.%s, /etc/monit/%s, /etc/%s, /usr/local/etc/%s or at ./%s \n",
+      prog, MONITRC, MONITRC, MONITRC, MONITRC, MONITRC);
   exit(1);
   
 }
