diff -u -r1.74 duplicity-bin
--- duplicity       21 Apr 2009 19:34:17 -0000      1.74
+++ duplicity       22 Apr 2009 02:34:02 -0000
@@ -765,7 +765,6 @@
         log.Notice(_("Last full backup date:") + " " + dup_time.timetopretty(last_full_time))
     else:
         log.Notice(_("Last full backup date: none"))
-        action = "full"
     if action == "inc" and last_full_time < globals.full_force_time:
         log.Notice(_("Last full backup is too old, forcing full backup"))
         action = "full"

