diff -Naur libetpan-0.50.orig/src/low-level/smtp/mailsmtp.c libetpan-0.50/src/low-level/smtp/mailsmtp.c
--- libetpan-0.50.orig/src/low-level/smtp/mailsmtp.c	2007-08-03 09:36:30.000000000 +0200
+++ libetpan-0.50/src/low-level/smtp/mailsmtp.c	2007-08-03 09:36:23.000000000 +0200
@@ -32,7 +32,7 @@
  */
 
 /*
- * $Id: mailsmtp.c,v 1.29 2007/07/21 18:44:40 hoa Exp $
+ * $Id: mailsmtp.c,v 1.30 2007/07/25 16:28:33 hoa Exp $
  */
 
 #ifdef HAVE_CONFIG_H
@@ -1257,11 +1257,11 @@
         unsigned int max_decoded;
         char * p;
         
-        p = strchr(f->pop3_response, '\r');
+        p = strchr(session->response, '\r');
         if (p != NULL) {
           * p = '\0';
         }
-        p = strchr(f->pop3_response, '\n');
+        p = strchr(session->response, '\n');
         if (p != NULL) {
           * p = '\0';
         }
