diff --git a/configure.in b/configure.in
index 58f248e..a2b8978 100644
--- a/configure.in
+++ b/configure.in
@@ -820,8 +820,8 @@ AC_ARG_ENABLE(menus-gnome2,
     
       if test "${PKG_CONFIG}" != ""; then
 	GNOME_VER=2
-	GNOME2_CFLAGS=`pkg-config --cflags gnome-desktop-2.0`
-	GNOME2_LIBS=`pkg-config --libs gnome-desktop-2.0`
+	GNOME2_CFLAGS=`pkg-config --cflags gnome-desktop-2.0 libgnomeui-2.0`
+	GNOME2_LIBS=`pkg-config --libs gnome-desktop-2.0 libgnomeui-2.0`
 
 	AC_DEFINE(CONFIG_GNOME_MENUS, 1, [Define to make IceWM more GNOME-friendly])
 
