diff -Naur i3-4.2/include/ipc.h i3-4.2-b/include/ipc.h 
--- i3-4.2/include/ipc.h	2012-04-25 23:21:25.000000000 +0200
+++ i3-4.2-b/include/ipc.h	2012-06-25 15:34:59.000000000 +0200
@@ -10,7 +10,7 @@
 #ifndef _IPC_H
 #define _IPC_H
 
-#include <ev.h>
+#include <libev/ev.h>
 #include <stdbool.h>
 #include <yajl/yajl_gen.h>
 #include <yajl/yajl_parse.h>

diff -Naur i3-4.2/src/main.c i3-4.2-b/src/main.c 
--- i3-4.2/src/main.c	2012-06-25 15:37:41.000000000 +0200
+++ i3-4.2-b/src/main.c	2012-06-25 15:37:50.000000000 +0200
@@ -7,7 +7,7 @@
  * main.c: Initialization, main loop
  *
  */
-#include <ev.h>
+#include <libev/ev.h>
 #include <fcntl.h>
 #include <sys/types.h>
 #include <sys/socket.h>

diff -Naur i3-4.2/src/sighandler.c i3-4.2-b/src/sighandler.c 
--- i3-4.2/src/sighandler.c	2012-04-25 23:21:25.000000000 +0200
+++ i3-4.2-b/src/sighandler.c	2012-06-25 15:41:31.000000000 +0200
@@ -11,7 +11,7 @@
  */
 #include "all.h"
 
-#include <ev.h>
+#include <libev/ev.h>
 #include <iconv.h>
 #include <signal.h>

diff -Naur i3-4.2/src/ipc.c i3-4.2-b/src/ipc.c
--- i3-4.2/src/ipc.c	2012-04-25 23:21:25.000000000 +0200
+++ i3-4.2-b/src/ipc.c	2012-06-25 15:43:33.000000000 +0200
@@ -13,7 +13,7 @@
 #include <sys/un.h>
 #include <fcntl.h>
 #include <libgen.h>
-#include <ev.h>
+#include <libev/ev.h>
 #include <yajl/yajl_gen.h>
 #include <yajl/yajl_parse.h>
 #include <yajl/yajl_version.h>

diff -Naur i3-4.2/i3-msg/main.c i3-4.2-b/i3-msg/main.c
--- i3-4.2/i3-msg/main.c	2012-04-25 23:21:25.000000000 +0200
+++ i3-4.2-b/i3-msg/main.c	2012-06-25 15:54:06.000000000 +0200
@@ -14,7 +14,7 @@
  * Additionally, it’s even useful sometimes :-).
  *
  */
-#include <ev.h>
+#include <libev/ev.h>
 #include <stdio.h>
 #include <stdbool.h>
 #include <sys/types.h>

diff -Naur i3-4.2/i3-input/main.c i3-4.2-b/i3-input/main.c
--- i3-4.2/i3-input/main.c	2012-04-25 23:21:32.000000000 +0200
+++ i3-4.2-b/i3-input/main.c	2012-06-25 15:54:20.000000000 +0200
@@ -8,7 +8,7 @@
  *                  to i3.
  *
  */
-#include <ev.h>
+#include <libev/ev.h>
 #include <stdio.h>
 #include <sys/types.h>
 #include <stdlib.h>

diff -Naur i3-4.2/i3-nagbar/main.c i3-4.2-b/i3-nagbar/main.c
--- i3-4.2/i3-nagbar/main.c	2012-04-25 23:21:25.000000000 +0200
+++ i3-4.2-b/i3-nagbar/main.c	2012-06-25 15:54:31.000000000 +0200
@@ -8,7 +8,7 @@
  * when the user has an error in his configuration file.
  *
  */
-#include <ev.h>
+#include <libev/ev.h>
 #include <stdio.h>
 #include <sys/types.h>
 #include <sys/wait.h>

diff -Naur i3-4.2/i3-config-wizard/main.c i3-4.2-b/i3-config-wizard/main.c
--- i3-4.2/i3-config-wizard/main.c	2012-04-25 23:21:25.000000000 +0200
+++ i3-4.2-b/i3-config-wizard/main.c	2012-06-25 15:54:45.000000000 +0200
@@ -21,7 +21,7 @@
 #define _WITH_GETLINE
 #endif
 
-#include <ev.h>
+#include <libev/ev.h>
 #include <stdio.h>
 #include <sys/types.h>
 #include <stdlib.h>

diff -Naur i3-4.2/i3bar/src/ipc.c i3-4.2-b/i3bar/src/ipc.c
--- i3-4.2/i3bar/src/ipc.c	2012-04-25 23:21:25.000000000 +0200
+++ i3-4.2-b/i3bar/src/ipc.c	2012-06-25 15:55:16.000000000 +0200
@@ -16,7 +16,7 @@
 #include <sys/socket.h>
 #include <sys/un.h>
 #include <i3/ipc.h>
-#include <ev.h>
+#include <libev/ev.h>
 
 #include "common.h"
 
diff -Naur i3-4.2/i3bar/src/main.c i3-4.2-b/i3bar/src/main.c 
--- i3-4.2/i3bar/src/main.c	2012-04-25 23:21:25.000000000 +0200
+++ i3-4.2-b/i3bar/src/main.c	2012-06-25 15:59:26.000000000 +0200
@@ -11,7 +11,7 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <errno.h>
-#include <ev.h>
+#include <libev/ev.h>
 #include <getopt.h>
 #include <glob.h>

diff -Naur i3-4.2/i3bar/src/xcb.c i3-4.2-b/i3bar/src/xcb.c 
--- i3-4.2/i3bar/src/xcb.c	2012-04-25 23:21:25.000000000 +0200
+++ i3-4.2-b/i3bar/src/xcb.c	2012-06-25 16:00:51.000000000 +0200
@@ -22,7 +22,7 @@
 #include <fcntl.h>
 #include <string.h>
 #include <i3/ipc.h>
-#include <ev.h>
+#include <libev/ev.h>
 #include <errno.h>
 #include <limits.h>
 #include <err.h>

diff -Naur i3-4.2/i3bar/src/child.c i3-4.2-b/i3bar/src/child.c 
--- i3-4.2/i3bar/src/child.c	2012-04-25 23:21:25.000000000 +0200
+++ i3-4.2-b/i3bar/src/child.c	2012-06-25 16:02:27.000000000 +0200
@@ -17,7 +17,7 @@
 #include <string.h>
 #include <errno.h>
 #include <err.h>
-#include <ev.h>
+#include <libev/ev.h>
 #include <yajl/yajl_common.h>
 #include <yajl/yajl_parse.h>
 #include <yajl/yajl_version.h>
