--- chromium-54.0.2832.2/chrome/test/BUILD.gn	2016-08-20 01:03:31.000000000 +0300
+++ chromium-54.0.2832.2.new/chrome/test/BUILD.gn	2016-08-20 15:34:36.390367940 +0300
@@ -471,7 +471,7 @@
             # suites, it seems like one or another starts timing out too.
             "../browser/ui/views/keyboard_access_browsertest.cc",
           ]
-          if (!use_ozone) {
+          if (!use_ozone && !use_gtk3) {
             sources += [ "../browser/ui/libgtk2ui/select_file_dialog_interactive_uitest.cc" ]
             configs += [ "//build/config/linux/gtk2:gtk2_internal_config" ]
           }
