=== modified file 'gdk-pixbuf/Makefile.am'
--- old/gdk-pixbuf/Makefile.am	2010-11-19 15:58:34 +0000
+++ new/gdk-pixbuf/Makefile.am	2010-11-19 16:00:47 +0000
@@ -684,6 +684,7 @@
 GdkPixbuf_2_0_gir_FILES = \
     $(libgdk_pixbufinclude_HEADERS) \
     $(libgdk_pixbuf_2_0_la_SOURCES)
+GdkPixbuf_2_0_gir_PACKAGES = gdk-pixbuf-2.0
 INTROSPECTION_GIRS += GdkPixbuf-2.0.gir
 
 girdir = $(datadir)/gir-1.0

