From 9671e0f93f49991954724889970aeb7939e380c5 Mon Sep 17 00:00:00 2001
From: Evan Nemerson <evan@coeus-group.com>
Date: Sat, 20 Nov 2010 17:39:07 +0000
Subject: introspection: Add C include and exported packages information to GIR

https://bugzilla.gnome.org/show_bug.cgi?id=635364
---
diff --git a/gdk-pixbuf/Makefile.am b/gdk-pixbuf/Makefile.am
index d4105d2..514ffe4 100644
--- a/gdk-pixbuf/Makefile.am
+++ b/gdk-pixbuf/Makefile.am
@@ -677,7 +677,8 @@ EXTRA_DIST +=					\
 if HAVE_INTROSPECTION
 
 GdkPixbuf-2.0.gir: libgdk_pixbuf-2.0.la Makefile
-GdkPixbuf_2_0_gir_SCANNERFLAGS = --warn-all --strip-prefix=Gdk
+GdkPixbuf_2_0_gir_SCANNERFLAGS = --warn-all --strip-prefix=Gdk --c-include="gdk-pixbuf/gdk-pixbuf.h"
+GdkPixbuf_2_0_gir_EXPORT_PACKAGES = gdk-pixbuf-2.0
 GdkPixbuf_2_0_gir_INCLUDES = GModule-2.0 Gio-2.0
 GdkPixbuf_2_0_gir_LIBS = libgdk_pixbuf-2.0.la
 GdkPixbuf_2_0_gir_CFLAGS = $(INCLUDES)
--
cgit v0.8.3.1
