diff -Naur xc.orig/programs/Xserver/dix/window.c xc/programs/Xserver/dix/window.c
--- xc.orig/programs/Xserver/dix/window.c	2004-07-31 10:24:13.000000000 +0200
+++ xc/programs/Xserver/dix/window.c	2004-11-17 18:14:08.000000000 +0100
@@ -299,6 +299,7 @@
 static void
 MakeRootTile(WindowPtr pWin)
 {
+#if 0
     ScreenPtr pScreen = pWin->drawable.pScreen;
     GCPtr pGC;
     unsigned char back[128];
@@ -340,6 +341,18 @@
 
    FreeScratchGC(pGC);
 
+#else
+   {
+     unsigned short 	red = 0 << 8, green = 0 << 8, blue = 0 << 8;
+     Pixel		pixel;
+     
+     AllocColor((ColormapPtr)SecurityLookupIDByType(wClient(pWin), pWin->optional->colormap,
+					      RT_COLORMAP, SecurityWriteAccess),
+		&red, &green, &blue, &pixel, wClient(pWin)->index);
+     ChangeWindowAttributes(pWin, CWBackPixel, &pixel, wClient(pWin));
+     (*pWin->drawable.pScreen->ClearToBackground)(pWin, 0, 0, 0, 0, 0);
+   }
+#endif
 }
 
 WindowPtr
