--- xc/lib/Xau/Imakefile.build-libs-with-pic	2001-01-17 20:42:24.000000000 +0100
+++ xc/lib/Xau/Imakefile	2003-08-11 18:52:39.000000000 +0200
@@ -37,6 +37,8 @@ SRCS =	AuDispose.c AuFileName.c AuGetAdd
 OBJS =	AuDispose.o AuFileName.o AuGetAddr.o AuGetBest.o AuLock.o \
 	AuRead.o AuUnlock.o AuWrite.o $(K5ENCOBJ)
 
+#define IncludeSharedObjectInNormalLib
+
 #include <Library.tmpl>
 
 DependTarget()
--- xc/lib/xkbfile/Imakefile.build-libs-with-pic	2001-08-27 19:40:59.000000000 +0200
+++ xc/lib/xkbfile/Imakefile	2003-08-11 18:52:39.000000000 +0200
@@ -34,6 +34,8 @@ XCOMM $XFree86: xc/lib/xkbfile/Imakefile
 
 ROOT_DEFINES = -DDFLT_XKB_CONFIG_ROOT=\"$(LIBDIR)/xkb\"
 
+#define IncludeSharedObjectInNormalLib
+
 #include <Library.tmpl>
 
 DependTarget()
--- xc/lib/xkbui/Imakefile.build-libs-with-pic	1999-03-14 04:21:21.000000000 +0100
+++ xc/lib/xkbui/Imakefile	2003-08-11 18:52:39.000000000 +0200
@@ -15,6 +15,8 @@ XCOMM $XFree86: xc/lib/xkbui/Imakefile,v
          HEADERS = XKBui.h
         INCLUDES = -I. -I$(EXTINCSRC) -I../xkbfile
 
+#define IncludeSharedObjectInNormalLib
+
 #include <Library.tmpl>
 
 DependTarget()
--- xc/programs/Xserver/hw/xfree86/parser/Imakefile.build-libs-with-pic	2002-05-31 20:46:02.000000000 +0200
+++ xc/programs/Xserver/hw/xfree86/parser/Imakefile	2003-08-11 18:52:40.000000000 +0200
@@ -36,9 +36,11 @@ CONFIG_DEFINES = -DXCONFIGDIR=\"$(XCONFI
 		 -DXCONFIGFILE=\"$(XCONFIGFILE)\" \
 		 -DXVERSION="$(XVERS)"
 
+#define IncludeSharedObjectInNormalLib
+
 #include <Library.tmpl>
 
-SpecialCObjectRule(scan,NullParameter,$(CONFIG_DEFINES) $(MODULEDEFINES) $(EXT_DEFINES))
+SpecialCObjectRule(scan,NullParameter,$(CONFIG_DEFINES) $(MODULEDEFINES) $(EXT_DEFINES) $(PICFLAGS))
 
 NormalProgramTarget(cpconfig,cpconfig.o $(OBJS),NullParameter,$(LOCAL_LIBRARIES),NullParameter)
 
