--- accountsservice-0.6.6/src/daemon.c.old	2011-03-17 20:56:31.000000000 +0100
+++ accountsservice-0.6.6/src/daemon.c	2011-03-18 21:52:55.000000000 +0100
@@ -1111,8 +1111,8 @@
                 argv[7] = NULL;
         }
         else if (cd->account_type == ACCOUNT_TYPE_STANDARD) {
-                argv[6] = cd->user_name;
-                argv[7] = NULL;
+                argv[4] = cd->user_name;
+                argv[5] = NULL;
         }
         else {
                 throw_error (context, ERROR_FAILED, "Don't know how to add user of type %d", cd->account_type);
