From 5295295ad3503a5879d06dec149c2c232d1825e5 Mon Sep 17 00:00:00 2001
From: Devin Lin <espidev@gmail.com>
Date: Sun, 3 Mar 2024 10:41:34 -0500
Subject: [PATCH] Remove unnecessary quickcompiler dependency

---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ef32ec9..fbc9ba7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,7 +48,6 @@ ecm_setup_version(${PROJECT_VERSION}
 
 find_package(PkgConfig REQUIRED)
 find_package(Qt6 ${QT_MIN_VERSION} REQUIRED NO_MODULE COMPONENTS Core Quick Test Gui QuickControls2 Widgets)
-find_package(Qt6 ${QT_MIN_VERSION} NO_MODULE COMPONENTS QuickCompiler)
 find_package(KF6 ${KF_MIN_VERSION} REQUIRED COMPONENTS
     Kirigami
     I18n
-- 
GitLab

