diff -Naur libX11-1.1.2/src/GetMoEv.c libX11-1.1.2-p/src/GetMoEv.c
--- libX11-1.1.2/src/GetMoEv.c	2007-06-02 19:46:55.000000000 +0200
+++ libX11-1.1.2-p/src/GetMoEv.c	2007-09-13 02:31:05.000000000 +0200
@@ -33,11 +33,11 @@
 
 XTimeCoord *XGetMotionEvents(
     register Display *dpy,
-    Time start, 
-    Time stop,
     Window w,
+    Time start,
+    Time stop,
     int *nEvents)  /* RETURN */
-{       
+{
     xGetMotionEventsReply rep;
     register xGetMotionEventsReq *req;
     XTimeCoord *tc = NULL;
@@ -53,7 +53,7 @@
         SyncHandle();
 	return (NULL);
 	}
-    
+
     if (rep.nEvents) {
 	if (! (tc = (XTimeCoord *)
 	       Xmalloc( (unsigned) 
