diff -Naur gfxboot-2.2.orig/Makefile gfxboot-2.2/Makefile
--- gfxboot-2.2.orig/Makefile	2003-08-18 15:27:45.000000000 +0200
+++ gfxboot-2.2/Makefile	2006-01-09 02:03:39.000000000 +0100
@@ -1,5 +1,5 @@
 CC	 = gcc
-CFLAGS	 = -g -Wall -Wno-sign-compare -O2 -fomit-frame-pointer
+CFLAGS	 += -Wall -Wno-sign-compare -fomit-frame-pointer
 X11LIBS	 = /usr/X11/lib
 THEMES	 = $(wildcard themes/*)
 LIBFILES = fixed_9x15 fixed_10x20 kroete.data happysuse.mod system.inc
@@ -18,7 +18,7 @@
 	$(CC) $(CFLAGS) $< -o $@
 
 bincode:  bincode.asm vocabulary.inc modplay_defines.inc modplay.inc kroete.inc
-	nasm -f bin -O3 -o $@ -l $(@).lst $<
+	nasm -f bin -O1 -o $@ -l $(@).lst $<
 
 bincode.h:  bincode bin2c
 	./bin2c bincode >bincode.h
