From d6d56af6e78c6010dc812febc7dd4fd101e47168 Mon Sep 17 00:00:00 2001
From: Adhemerval Zanella <adhemerval.zanella@linaro.org>
Date: Thu, 10 Apr 2025 12:08:22 +0000
Subject: [PATCH 43/47] elf: Fix arm-linux-gnueabihf build break from
 b861755a84

---
 elf/Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/elf/Makefile b/elf/Makefile
index 600bf3fe20..3a3ae56a24 100644
--- a/elf/Makefile
+++ b/elf/Makefile
@@ -62,7 +62,6 @@ dl-routines = \
   dl-exception \
   dl-execstack \
   dl-execstack-tunable \
-  dl-find_object \
   dl-fini \
   dl-init \
   dl-load \
-- 
2.49.0

