---
 sysdeps/i386/fpu/ftestexcept.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: glibc-2.7/sysdeps/i386/fpu/ftestexcept.c
===================================================================
--- glibc-2.7.orig/sysdeps/i386/fpu/ftestexcept.c	2004-03-05 11:14:48.000000000 +0100
+++ glibc-2.7/sysdeps/i386/fpu/ftestexcept.c	2008-01-18 14:02:19.000000000 +0100
@@ -26,7 +26,7 @@
 int
 fetestexcept (int excepts)
 {
-  int temp;
+  unsigned short temp;
   int xtemp = 0;
 
   /* Get current exceptions.  */
