diff -Naur glibc-2.3.5.orig/sysdeps/generic/wcstoul_l.c glibc-2.3.5/sysdeps/generic/wcstoul_l.c
--- glibc-2.3.5.orig/sysdeps/generic/wcstoul_l.c	2005-07-21 11:39:55.000000000 +0200
+++ glibc-2.3.5/sysdeps/generic/wcstoul_l.c	2005-07-21 11:40:01.000000000 +0200
@@ -1,5 +1,5 @@
 /* Convert string representing a number to integer value, using given locale.
-   Copyright (C) 1997, 2002, 2004 Free Software Foundation, Inc.
+   Copyright (C) 1997, 2002, 2004, 2005 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
    Contributed by Ulrich Drepper <drepper@cygnus.com>, 1997.
 
@@ -27,4 +27,4 @@
 extern unsigned long int ____wcstoul_l_internal (const wchar_t *, wchar_t **,
 						 int, int, __locale_t);
 
-#include <wcstol_l.c>
+#include "wcstol_l.c"
diff -Naur glibc-2.3.5.orig/sysdeps/wordsize-64/strtol_l.c glibc-2.3.5/sysdeps/wordsize-64/strtol_l.c
--- glibc-2.3.5.orig/sysdeps/wordsize-64/strtol_l.c	2005-07-21 11:39:55.000000000 +0200
+++ glibc-2.3.5/sysdeps/wordsize-64/strtol_l.c	2005-07-21 11:40:20.000000000 +0200
@@ -8,7 +8,9 @@
 #undef ____strtoll_l_internal
 #undef __strtoll_l
 #undef strtoll_l
+#if !UNSIGNED
 strong_alias (____strtol_l_internal, ____strtoll_l_internal)
 libc_hidden_ver (____strtol_l_internal, ____strtoll_l_internal)
 weak_alias (__strtol_l, __strtoll_l)
 weak_alias (__strtol_l, strtoll_l)
+#endif
diff -Naur glibc-2.3.5.orig/sysdeps/wordsize-64/wcstol_l.c glibc-2.3.5/sysdeps/wordsize-64/wcstol_l.c
--- glibc-2.3.5.orig/sysdeps/wordsize-64/wcstol_l.c	2005-07-21 11:39:55.000000000 +0200
+++ glibc-2.3.5/sysdeps/wordsize-64/wcstol_l.c	2005-07-21 11:40:26.000000000 +0200
@@ -8,6 +8,8 @@
 #undef ____wcstoll_l_internal
 #undef __wcstoll_l
 #undef wcstoll_l
+#if !UNSIGNED
 strong_alias (____wcstol_l_internal, ____wcstoll_l_internal)
 weak_alias (__wcstol_l, __wcstoll_l)
 weak_alias (__wcstol_l, wcstoll_l)
+#endif
