diff -Naur glibc-ports-2.16/sysdeps/arm/Makefile glibc-ports-2.16/sysdeps/arm/Makefile
--- glibc-ports-2.16/sysdeps/arm/Makefile	2011-12-23 20:34:10.000000000 +0100
+++ glibc-ports-2.16/sysdeps/arm/Makefile	2012-05-21 21:09:07.886428034 +0200
@@ -7,3 +7,8 @@
 ifeq ($(subdir),csu)
 gen-as-const-headers += tlsdesc.sym
 endif
+
+# to pull in __aeabi_read_tp, needed for tls
+ifeq ($(subdir),malloc)
+$(objpfx)libmemusage.so: $(common-objpfx)libc_nonshared.a
+endif
