--- glibc-2.3.3/sysdeps/generic/dl-tls.c	2004/12/18 20:55:18	1.1
+++ glibc-2.3.3/sysdeps/generic/dl-tls.c	2004/12/18 22:05:06
@@ -62,6 +62,13 @@ _dl_next_tls_modid (void)
       size_t disp = 0;
       struct dtv_slotinfo_list *runp = GL(dl_tls_dtv_slotinfo_list);
 
+      if (GL(dl_tls_max_dtv_idx) == GL(dl_tls_static_nelem))
+	{
+	  /* There is actually no gap.  */
+	  GL(dl_tls_dtv_gaps) = false;
+
+	  goto nogaps;
+	}
       /* Note that this branch will never be executed during program
 	 start since there are no gaps at that time.  Therefore it
 	 does not matter that the dl_tls_dtv_slotinfo is not allocated
