--- glibc-2.3.2/sysdeps/unix/sysv/linux/configure~	2004-06-05 14:03:22.118247000 +0200
+++ glibc-2.3.2/sysdeps/unix/sysv/linux/configure	2004-06-05 11:43:35.667961319 +0200
@@ -224,7 +224,7 @@
 /usr | /usr/)
   # 64-bit libraries on bi-arch platforms go in /lib64 instead of /lib
   case $machine in
-  sparc/sparc64 | x86_64 | powerpc/powerpc64 | s390/s390-64 | \
+  sparc/sparc64 | powerpc/powerpc64 | s390/s390-64 | \
   mips/mips64/n64/* )
     libc_cv_slibdir="/lib64"
     if test "$libdir" = '${exec_prefix}/lib'; then
--- glibc-2.3.2/sysdeps/unix/sysv/linux/configure.in~	2004-07-05 09:52:28.000000000 +0200
+++ glibc-2.3.2/sysdeps/unix/sysv/linux/configure.in	2004-07-05 12:18:07.995664571 +0200
@@ -157,7 +157,7 @@
 /usr | /usr/)
   # 64-bit libraries on bi-arch platforms go in /lib64 instead of /lib
   case $machine in
-  sparc/sparc64 | x86_64 | powerpc/powerpc64 | s390/s390-64 | \
+  sparc/sparc64 | powerpc/powerpc64 | s390/s390-64 | \
   mips/mips64/n64/* )
     libc_cv_slibdir="/lib64"
     if test "$libdir" = '${exec_prefix}/lib'; then
