diff -ur a/ports/sysdeps/arm/Makefile b/ports/sysdeps/arm/Makefile
--- a/ports/sysdeps/arm/Makefile	2012-10-07 03:50:53.989001976 -0500
+++ b/ports/sysdeps/arm/Makefile	2012-10-07 03:51:12.198001304 -0500
@@ -33,3 +33,8 @@
 ifeq ($(subdir),math)
 $(objpfx)libm.so: $(elfobjdir)/ld.so
 endif
+
+# to pull in __aeabi_read_tp, needed for tls
+ifeq ($(subdir),malloc)
+$(objpfx)libmemusage.so: $(common-objpfx)libc_nonshared.a
+endif
