diff -Naur glibc-2.6.1/nptl/sysdeps/pthread/pt-initfini.c glibc-2.6.1-if/nptl/sysdeps/pthread/pt-initfini.c
--- glibc-2.6.1/nptl/sysdeps/pthread/pt-initfini.c	2005-02-26 02:33:57.000000000 +0100
+++ glibc-2.6.1-if/nptl/sysdeps/pthread/pt-initfini.c	2007-08-09 02:33:03.000000000 +0200
@@ -72,7 +72,7 @@
 }
 
 SECTION (".init");
-extern void _init (void);
+extern void __attribute__ ((section (".init"))) _init (void);
 void
 _init (void)
 {
@@ -93,7 +93,7 @@
 asm ("\n/*@_fini_PROLOG_BEGINS*/");
 
 SECTION (".fini");
-extern void _fini (void);
+extern void __attribute__ ((section (".fini"))) _fini (void);
 void
 _fini (void)
 {
diff -Naur glibc-2.6.1/sysdeps/generic/initfini.c glibc-2.6.1-if/sysdeps/generic/initfini.c
--- glibc-2.6.1/sysdeps/generic/initfini.c	2004-08-16 06:50:53.000000000 +0200
+++ glibc-2.6.1-if/sysdeps/generic/initfini.c	2007-08-09 02:33:03.000000000 +0200
@@ -81,7 +81,7 @@
 }
 
 SECTION (".init");
-extern void _init (void);
+extern void __attribute__ ((section (".init"))) _init (void);
 void
 _init (void)
 {
@@ -107,7 +107,7 @@
 asm ("\n/*@_fini_PROLOG_BEGINS*/");
 
 SECTION (".fini");
-extern void _fini (void);
+extern void __attribute__ ((section (".fini"))) _fini (void);
 void
 _fini (void)
 {
