diff -Naur glibc-2.29/nscd/nscd.conf glibc-2.29-p/nscd/nscd.conf
--- glibc-2.29/nscd/nscd.conf	2019-01-31 17:45:36.000000000 +0100
+++ glibc-2.29-p/nscd/nscd.conf	2020-01-02 13:44:11.145378872 +0100
@@ -30,10 +30,10 @@
 #
 
 
-#	logfile			/var/log/nscd.log
+	logfile			/var/log/nscd/nscd.log
 #	threads			4
 #	max-threads		32
-#	server-user		nobody
+	server-user		nscd
 #	stat-user		somebody
 	debug-level		0
 #	reload-count		5
@@ -60,7 +60,8 @@
 	max-db-size		group		33554432
 	auto-propagate		group		yes
 
-	enable-cache		hosts		yes
+	# don't use that, it breaks forward/reverse lookup checks
+	enable-cache		hosts		no
 	positive-time-to-live	hosts		3600
 	negative-time-to-live	hosts		20
 	suggested-size		hosts		211
