diff -Naur glibc-2.29/nscd/nscd-client.h glibc-2.29-p/nscd/nscd-client.h
--- glibc-2.29/nscd/nscd-client.h	2019-01-31 17:45:36.000000000 +0100
+++ glibc-2.29-p/nscd/nscd-client.h	2020-01-02 13:09:00.142569150 +0100
@@ -36,10 +36,10 @@
 #define NSCD_VERSION 2
 
 /* Path of the file where the PID of the running system is stored.  */
-#define _PATH_NSCDPID	 "/var/run/nscd/nscd.pid"
+#define _PATH_NSCDPID	 "/run/nscd/nscd.pid"
 
 /* Path for the Unix domain socket.  */
-#define _PATH_NSCDSOCKET "/var/run/nscd/socket"
+#define _PATH_NSCDSOCKET "/run/nscd/socket"
 
 /* Path for the configuration file.  */
 #define _PATH_NSCDCONF	 "/etc/nscd.conf"
diff -Naur glibc-2.29/nscd/nscd.h glibc-2.29-p/nscd/nscd.h
--- glibc-2.29/nscd/nscd.h	2019-01-31 17:45:36.000000000 +0100
+++ glibc-2.29-p/nscd/nscd.h	2020-01-02 13:08:29.760353085 +0100
@@ -168,7 +168,7 @@
 #define _PATH_NSCD_NETGROUP_DB	"/var/db/nscd/netgroup"
 
 /* Path used when not using persistent storage.  */
-#define _PATH_NSCD_XYZ_DB_TMP	"/var/run/nscd/dbXXXXXX"
+#define _PATH_NSCD_XYZ_DB_TMP	"/run/nscd/dbXXXXXX"
 
 /* Maximum alignment requirement we will encounter.  */
 #define BLOCK_ALIGN_LOG 3
