--- merkaartor-0.17.2/3rdparty/qtsingleapplication-2.6_1-opensource/src/qtlocalpeer.cpp.old	2012-05-01 11:40:08.539279638 +0200
+++ merkaartor-0.17.2/3rdparty/qtsingleapplication-2.6_1-opensource/src/qtlocalpeer.cpp	2012-05-01 11:40:58.835282311 +0200
@@ -57,6 +57,8 @@
 #endif
 #if defined(Q_OS_UNIX)
 #include <time.h>
+#include <sys/types.h>
+#include <unistd.h>
 #endif
 
 namespace QtLP_Private {
