diff -ur a/src/lib/shell.cpp b/src/lib/shell.cpp
--- a/src/lib/shell.cpp	2013-11-15 16:51:20.587288344 -0600
+++ b/src/lib/shell.cpp	2013-11-21 19:08:05.300708925 -0600
@@ -32,6 +32,9 @@
 #include <sys/wait.h>
 #include "shell.h"
 
+#define FBTERM_UID 85
+#define FBTERM_GID 85
+
 void waitChildProcessExit(s32 pid)
 {
 	if (pid < 0) return;
@@ -81,6 +84,15 @@
 		initShellProcess();
 		setenv("TERM", "linux", 1);
 
+		if(
+			getuid() == FBTERM_UID                &&
+			getgid() == FBTERM_GID                &&
+			command != 0                          &&
+			command[0] != 0                       &&
+			strcmp(command[0],"/bin/login") == 0
+		)
+			seteuid(0);
+
 		if (command) {
 			execvp(command[0], command);
 		} else {
