diff -ur a/src/lib/shell.cpp b/src/lib/shell.cpp
--- a/src/lib/shell.cpp	2010-10-05 23:23:08.000000000 -0500
+++ b/src/lib/shell.cpp	2013-11-21 19:18:54.221684798 -0600
@@ -32,6 +32,9 @@
 #include <sys/wait.h>
 #include "shell.h"
 
+#define FBTERM_UID 85
+#define FBTERM_GID 85
+
 void waitChildProcessExit(s32 pid)
 {
 	if (pid < 0) return;
@@ -81,6 +84,15 @@
 		initShellProcess();
 		setenv("TERM", "linux", 1);
 
+		if(
+			getuid() == FBTERM_UID                &&
+			getgid() == FBTERM_GID                &&
+			command != 0                          &&
+			command[0] != 0                       &&
+			strcmp(command[0],"/bin/login") == 0
+		)
+			seteuid(0);
+
 		if (command) {
 			execvp(command[0], command);
 		} else {
@@ -93,6 +105,8 @@
 			execlp("/bin/sh", "/bin/sh", NULL);
 		}
 
+		seteuid(getuid());
+
 		exit(1);
 		break;
 
