--- network-uri-2.6.4.2/network-uri.cabal	2001-09-09 03:46:40.000000000 +0200
+++ network-uri-2.6.4.2.new/network-uri.cabal	2023-12-11 17:56:59.392000898 +0100
@@ -68,7 +68,7 @@
     Network.URI.Static
   build-depends:
     base >= 3 && < 5,
-    deepseq >= 1.1 && < 1.5,
+    deepseq >= 1.1 && < 1.6,
     parsec >= 3.1.12.0 && < 3.2,
     th-compat >= 0.1.1 && < 1.0
   build-depends: template-haskell
