diff -Naur gnumeric-1.12.36/doc/Makefile.am gnumeric-1.12.36-p/doc/Makefile.am
--- gnumeric-1.12.36/doc/Makefile.am	2017-03-20 23:41:47.000000000 +0100
+++ gnumeric-1.12.36-p/doc/Makefile.am	2017-11-30 16:05:27.175028193 +0100
@@ -8,7 +8,7 @@
 @YELP_HELP_RULES@
 
 HELP_ID=gnumeric
-HELP_LINGUAS=cs de es
+HELP_LINGUAS=de es
 HELP_FILES=gnumeric.xml
 
 HELP_EXTRA=					\
diff -Naur gnumeric-1.12.36/doc/Makefile.in gnumeric-1.12.36-p/doc/Makefile.in
--- gnumeric-1.12.36/doc/Makefile.in	2017-07-11 00:46:03.000000000 +0200
+++ gnumeric-1.12.36-p/doc/Makefile.in	2017-11-30 16:06:22.792563314 +0100
@@ -460,7 +460,7 @@
 noinst_DATA = $(am__append_1)
 DISTCLEANFILES = C/func.defs
 HELP_ID = gnumeric
-HELP_LINGUAS = cs de es
+HELP_LINGUAS = de es
 HELP_FILES = gnumeric.xml
 HELP_EXTRA = \
 	about-authors.xml			\
