diff -Naur apg-2.2.3/Makefile apg-2.2.3.new/Makefile
--- apg-2.2.3/Makefile	2003-08-07 21:10:30.000000000 +0530
+++ apg-2.2.3.new/Makefile	2007-09-13 21:01:30.000000000 +0530
@@ -6,7 +6,6 @@
 ##################################################################
 # Compilation flags
 # You should comment the line below for AIX+native cc
-FLAGS = -Wall
 
 ##################################################################
 # Libraries
@@ -39,11 +38,11 @@
 ##################################################################
 # Directories
 # Install dirs
-INSTALL_PREFIX = /usr/local
+INSTALL_PREFIX = ${DESTDIR}/usr
 APG_BIN_DIR = /bin
-APG_MAN_DIR = /man/man1
+APG_MAN_DIR = /share/man/man1
 APGD_BIN_DIR = /sbin
-APGD_MAN_DIR = /man/man8
+APGD_MAN_DIR = /share/man/man8
 
 ####################################################################
 # If you plan to install APG daemon you should look at lines below #
@@ -96,13 +95,13 @@
 cygwin: standalone
 
 apg:
-	${CC} ${FLAGS} -D${CRYPTED_PASS} -D${USE_SHA} -o ${PROGNAME} ${SOURCES} ${LIBS} ${LIBM}
+	${CC} ${CFLAGS} -D${CRYPTED_PASS} -D${USE_SHA} -o ${PROGNAME} ${SOURCES} ${LIBS} ${LIBM}
 
 apgd:
-	${CC} ${FLAGS} -DCLISERV -D${USE_SHA} -o ${CS_PROGNAME} ${SOURCES} ${CS_LIBS} ${LIBM}
+	${CC} ${CFLAGS} -DCLISERV -D${USE_SHA} -o ${CS_PROGNAME} ${SOURCES} ${CS_LIBS} ${LIBM}
 
 apgbfm:
-	${CC} ${FLAGS} -DAPGBFM -o ${BFM_PROGNAME} ${BFM_SOURCES} ${LIBM}
+	${CC} ${CFLAGS} -DAPGBFM -o ${BFM_PROGNAME} ${BFM_SOURCES} ${LIBM}
 
 strip:
 	strip ${PROGNAME}
