--- pywebkitgtk-1.0.2/Makefile.in	2008-11-01 03:27:14.000000000 +0100
+++ pywebkitgtk-1.0.2/Makefile.in	2009-01-25 16:05:47.000000000 +0100
@@ -742,7 +742,6 @@
 	    --register $(PYGTK_DEFSDIR)/gtk-types.defs \
 	    --override $*.override \
 	    --prefix py$* $*.defs) 2>&1 >gen-$*.c | tee $*.errors \
-	&& ! grep -q -v "^\*\*\*INFO\*\*\*" $*.errors \
 	&& cp gen-$*.c $*.c \
 	&& rm -f gen-$*.c
 
