diff -Naur qemu-launcher-1.7.0/Makefile qemu-launcher-1.7.0-p/Makefile
--- qemu-launcher-1.7.0/Makefile	2006-07-31 23:28:42.000000000 +0200
+++ qemu-launcher-1.7.0-p/Makefile	2006-08-22 13:48:39.660611976 +0200
@@ -2,8 +2,8 @@
 SHELL = /bin/sh
 
 # Intall things to the following directories
-PREFIX = /usr/local
-MANDIR = $(PREFIX)/share/man/man1
+PREFIX = /usr
+MANDIR = $(PREFIX)/man/man1
 BINDIR = $(PREFIX)/bin
 SHAREDIR = $(PREFIX)/share/qemu-launcher
 PIXMAPSDIR = $(PREFIX)/share/pixmaps
