--- Makefile	2003-07-26 21:15:00.000000000 +0200
+++ Makefile1	2008-10-10 00:36:19.000000000 +0200
@@ -15,10 +15,10 @@
 #    this must be a directory that's in your bin PATH.
 # MANDIR holds the man page.
 
-BINDIR = /opt/local/bin
+BINDIR = /usr/bin
 #BINDIR = /usr/local/bin
 #BINDIR = //C/bin
-MANDIR = /usr/local/man/man1
+MANDIR = /usr/man/man1
 # popular alternative
 #BINDIR = /opt/local/bin
 #MANDIR = /opt/local/man/man1
@@ -26,7 +26,7 @@
 
 ### if you have GNU gcc, use these definitions
 CC = gcc
-CFLAGS = -O2 -finline-functions
+CFLAGS ?= -O2 -finline-functions
 
 ### if you just have a standard UNIX, use these instead of GNU.
 ### CC must be an ANSI C compiler
