--- linux/phatch.desktop	2008-11-09 16:37:31.000000000 +0100
+++ linux/phatch.desktop	2009-08-05 01:12:26.000000000 +0200
@@ -11,5 +11,5 @@
 Categories=Graphics;Photography;GTK;
 GenericName[nl_NL]=Phatch Foto Batch Processor
 StartupNotify=true
-MimeType=application/x-phatch;image/bmp;image/gif;image/jpeg;image/jpg;image/png;image/tiff;image/x-bmp;image/x-pcx;image/x-png;image/x-portable-bitmap;image/x-portable-graymap;image/x-portable-pixmap;image/x-psd;image/x-sun-raster;image/x-tga;image/x-xbitmap;image/x-xpixmap;application/pcx;inode/directory;
+MimeType=application/x-phatch;image/bmp;image/gif;image/jpeg;image/jpg;image/png;image/tiff;image/x-bmp;image/x-pcx;image/x-png;image/x-portable-bitmap;image/x-portable-graymap;image/x-portable-pixmap;image/x-psd;image/x-sun-raster;image/x-tga;image/x-xbitmap;image/x-xpixmap;application/pcx;
 
