diff -ru minbif-1.0.5.old/CMakeLists.txt minbif-1.0.5/CMakeLists.txt
--- minbif-1.0.5.old/CMakeLists.txt	2011-12-04 15:24:51.000000000 +0100
+++ minbif-1.0.5/CMakeLists.txt	2013-01-23 18:12:01.386749825 +0100
@@ -67,7 +67,7 @@
 			MESSAGE(FATAL_ERROR "You need to enable CACA to enable VIDEO")
 		ENDIF(NOT ENABLE_CACA)
 		PKG_CHECK_MODULES(GSTREAMER gstreamer-0.10>=0.10.10 gstreamer-interfaces-0.10>=0.10)
-		PKG_CHECK_MODULES(FARSIGHT farsight2-0.10>=0.0.9)
+		PKG_CHECK_MODULES(FARSIGHT farstream-0.1>=0.0.9)
 		PKG_CHECK_MODULES(PURPLE_26 purple>=2.6.1)
 		IF(GSTREAMER_FOUND AND PURPLE_26_FOUND AND FARSIGHT_FOUND)
 			SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DHAVE_VIDEO")
