diff -ur a/src/irc/dcc.h b/src/irc/dcc.h
--- a/src/irc/dcc.h	2011-12-04 08:24:51.000000000 -0600
+++ b/src/irc/dcc.h	2012-06-08 05:06:25.512961231 -0500
@@ -22,6 +22,7 @@
 #include <string>
 #include <stdio.h>
 #include <stdint.h>
+#include <unistd.h>
 
 #include "im/ft.h"
 
diff -ur a/src/irc/server.h b/src/irc/server.h
--- a/src/irc/server.h	2011-12-04 08:24:51.000000000 -0600
+++ b/src/irc/server.h	2012-06-08 05:05:15.799612400 -0500
@@ -18,6 +18,7 @@
 #ifndef IRC_SERVER_H
 #define IRC_SERVER_H
 
+#include <unistd.h>
 #include <vector>
 
 #include "core/entity.h"
diff -ur a/src/server_poll/daemon_fork.cpp b/src/server_poll/daemon_fork.cpp
--- a/src/server_poll/daemon_fork.cpp	2011-12-04 08:24:51.000000000 -0600
+++ b/src/server_poll/daemon_fork.cpp	2012-06-08 05:02:17.064845776 -0500
@@ -20,7 +20,7 @@
 #include <cassert>
 #include <cstring>
 #include <cerrno>
-#include <glib/gmain.h>
+#include <glib.h>
 #include <sys/socket.h>
 #include <sys/stat.h>
 #include <arpa/inet.h>
diff -ur a/src/server_poll/inetd.cpp b/src/server_poll/inetd.cpp
--- a/src/server_poll/inetd.cpp	2011-12-04 08:24:51.000000000 -0600
+++ b/src/server_poll/inetd.cpp	2012-06-08 05:02:26.854613899 -0500
@@ -17,7 +17,7 @@
  */
 
 #include <cassert>
-#include <glib/gmain.h>
+#include <glib.h>
 
 #include "inetd.h"
 #include "irc/irc.h"
diff -ur a/src/sockwrap/sockwrap.h b/src/sockwrap/sockwrap.h
--- a/src/sockwrap/sockwrap.h	2011-12-04 08:24:51.000000000 -0600
+++ b/src/sockwrap/sockwrap.h	2012-06-08 05:00:46.918980892 -0500
@@ -20,6 +20,7 @@
 #include <netdb.h>
 #include <string>
 #include <vector>
+#include <unistd.h>
 #include "core/log.h"
 #include "core/config.h"
 #include "core/callback.h"
