diff -Naur udev-151/rules/rules.d/50-udev-default.rules udev-151-p/rules/rules.d/50-udev-default.rules
--- udev-151/rules/rules.d/50-udev-default.rules	2010-01-21 14:21:42.000000000 +0100
+++ udev-151-p/rules/rules.d/50-udev-default.rules	2010-02-19 21:17:23.000000000 +0100
@@ -77,9 +77,11 @@
 
 # floppy
 SUBSYSTEM=="block", KERNEL=="fd[0-9]", GROUP="floppy"
+KERNEL=="hd*", SUBSYSTEMS=="ide", ATTRS{media}=="floppy", OPTIONS+="all_partitions"
 
 # cdrom
 SUBSYSTEM=="block", KERNEL=="sr[0-9]*", SYMLINK+="scd%n", GROUP="cdrom"
+SUBSYSTEM=="block", KERNEL=="hd*", SUBSYSTEMS=="ide", ATTRS{media}=="cdrom", GROUP="cdrom"
 SUBSYSTEM=="scsi_generic", SUBSYSTEMS=="scsi", ATTRS{type}=="4|5", GROUP="cdrom"
 KERNEL=="pktcdvd[0-9]*", GROUP="cdrom"
 KERNEL=="pktcdvd", GROUP="cdrom"
