diff -Naur udev-070.orig/etc/udev/frugalware/udev.rules udev-070/etc/udev/frugalware/udev.rules
--- udev-070.orig/etc/udev/frugalware/udev.rules	2005-08-18 19:04:49.000000000 +0200
+++ udev-070/etc/udev/frugalware/udev.rules	2005-09-30 16:51:32.000000000 +0200
@@ -28,7 +28,7 @@
 # permissions
 KERNEL="pl2303",        NAME="%k", OWNER="500",  GROUP="500",  MODE="0666"
 KERNEL="visor",         NAME="%k", OWNER="root", GROUP="uucp", MODE="0666"
-KERNEL="camera*"        NAME="%k", OWNER="500",  GROUP="500",  MODE="0666"
+KERNEL="camera*"        NAME="%k", OWNER="root",  GROUP="camera",  MODE="0666"
 
 # console devices
 KERNEL="console"        NAME="%k", OWNER="root", GROUP="tty",  MODE="0600"
@@ -47,27 +47,27 @@
 KERNEL="cu[abiu]*"      NAME="%k", OWNER="root", GROUP="uucp",  MODE="0660"
 
 # sound devices
-KERNEL="audio*"         NAME="%k", OWNER="root", GROUP="sys",  MODE="0666"
-KERNEL="beep"           NAME="%k", OWNER="root", GROUP="root", MODE="0644"
-KERNEL="dsp*"           NAME="%k", OWNER="root", GROUP="sys",  MODE="0666"
-KERNEL="midi*"          NAME="%k", OWNER="root", GROUP="sys",  MODE="0666"
-KERNEL="mixer*"         NAME="%k", OWNER="root", GROUP="sys",  MODE="0666"
-KERNEL="music"          NAME="%k", OWNER="root", GROUP="sys",  MODE="0666"
-KERNEL="sequencer*"     NAME="%k", OWNER="root", GROUP="sys",  MODE="0666"
-KERNEL="controlC[0-9]*", NAME="snd/%k", OWNER="root", GROUP="sys",  MODE="0666"
-KERNEL="hw[CD0-9]*",     NAME="snd/%k", OWNER="root", GROUP="sys",  MODE="0666"
-KERNEL="pcm[CD0-9cp]*",  NAME="snd/%k", OWNER="root", GROUP="sys",  MODE="0666"
-KERNEL="midiC[D0-9]*",   NAME="snd/%k", OWNER="root", GROUP="sys",  MODE="0666"
-KERNEL="timer",          NAME="snd/%k", OWNER="root", GROUP="sys",  MODE="0666"
-KERNEL="seq",            NAME="snd/%k", OWNER="root", GROUP="sys",  MODE="0666"
-KERNEL="sndstat"        NAME="%k", OWNER="root", GROUP="sys",  MODE="0666"
-KERNEL="speaker"        NAME="%k", OWNER="root", GROUP="root", MODE="0644"
-KERNEL="sound/*"        NAME="%k", OWNER="root", GROUP="root", MODE="0666"
-KERNEL="admmidi*"       NAME="%k", OWNER="root", GROUP="root", MODE="0666"
-KERNEL="adsp*"          NAME="%k", OWNER="root", GROUP="root", MODE="0666"
-KERNEL="aload*"         NAME="%k", OWNER="root", GROUP="sys",  MODE="0666"
-KERNEL="amidi*"         NAME="%k", OWNER="root", GROUP="root", MODE="0666"
-KERNEL="amixer*"        NAME="%k", OWNER="root", GROUP="root", MODE="0666"
+KERNEL="audio*"         NAME="%k", OWNER="root", GROUP="audio",  MODE="0666"
+KERNEL="beep"           NAME="%k", OWNER="root", GROUP="audio", MODE="0644"
+KERNEL="dsp*"           NAME="%k", OWNER="root", GROUP="audio",  MODE="0666"
+KERNEL="midi*"          NAME="%k", OWNER="root", GROUP="audio",  MODE="0666"
+KERNEL="mixer*"         NAME="%k", OWNER="root", GROUP="audio",  MODE="0666"
+KERNEL="music"          NAME="%k", OWNER="root", GROUP="audio",  MODE="0666"
+KERNEL="sequencer*"     NAME="%k", OWNER="root", GROUP="audio",  MODE="0666"
+KERNEL="controlC[0-9]*", NAME="snd/%k", OWNER="root", GROUP="audio",  MODE="0666"
+KERNEL="hw[CD0-9]*",     NAME="snd/%k", OWNER="root", GROUP="audio",  MODE="0666"
+KERNEL="pcm[CD0-9cp]*",  NAME="snd/%k", OWNER="root", GROUP="audio",  MODE="0666"
+KERNEL="midiC[D0-9]*",   NAME="snd/%k", OWNER="root", GROUP="audio",  MODE="0666"
+KERNEL="timer",          NAME="snd/%k", OWNER="root", GROUP="audio",  MODE="0666"
+KERNEL="seq",            NAME="snd/%k", OWNER="root", GROUP="audio",  MODE="0666"
+KERNEL="sndstat"        NAME="%k", OWNER="root", GROUP="audio",  MODE="0666"
+KERNEL="speaker"        NAME="%k", OWNER="root", GROUP="audio", MODE="0644"
+KERNEL="sound/*"        NAME="%k", OWNER="root", GROUP="audio", MODE="0666"
+KERNEL="admmidi*"       NAME="%k", OWNER="root", GROUP="audio", MODE="0666"
+KERNEL="adsp*"          NAME="%k", OWNER="root", GROUP="audio", MODE="0666"
+KERNEL="aload*"         NAME="%k", OWNER="root", GROUP="audio",  MODE="0666"
+KERNEL="amidi*"         NAME="%k", OWNER="root", GROUP="audio", MODE="0666"
+KERNEL="amixer*"        NAME="%k", OWNER="root", GROUP="audio", MODE="0666"
 
 # input devices (mouse, joystick, etc)
 KERNEL="mice",		NAME="input/%k", OWNER="root", GROUP="root", MODE="0644"
@@ -94,11 +94,11 @@
 KERNEL="parport*"        NAME="%k", OWNER="root", GROUP="root", MODE="0660"
 
 # video devices
-KERNEL="card*"        NAME="dri/card%n", OWNER="root", GROUP="root", MODE="0666"
-KERNEL="fb[0-9]"         NAME="%k", OWNER="root", GROUP="tty",  MODE="0660"
-KERNEL="agpgart"         NAME="%k", OWNER="root", GROUP="root", MODE="0660"
-KERNEL="nvidia*"         NAME="%k", OWNER="root", GROUP="root", MODE="0666"
-KERNEL="video*"          NAME="%k", OWNER="root", GROUP="sys",  MODE="0660"
+KERNEL="card*"        NAME="dri/card%n", OWNER="root", GROUP="video", MODE="0666"
+KERNEL="fb[0-9]"         NAME="%k", OWNER="root", GROUP="video",  MODE="0660"
+KERNEL="agpgart"         NAME="%k", OWNER="root", GROUP="video", MODE="0660"
+KERNEL="nvidia*"         NAME="%k", OWNER="root", GROUP="video", MODE="0666"
+KERNEL="video*"          NAME="%k", OWNER="root", GROUP="video",  MODE="0660"
 
 # DISK devices
 KERNEL="ed[ab]*"         NAME="%k", OWNER="root", GROUP="disk", MODE="0640"
