diff -Naur udev-068.orig/etc/udev/frugalware/udev.rules udev-068/etc/udev/frugalware/udev.rules
--- udev-068.orig/etc/udev/frugalware/udev.rules	2005-08-18 22:14:14.000000000 +0200
+++ udev-068/etc/udev/frugalware/udev.rules	2005-08-18 22:37:00.000000000 +0200
@@ -8,6 +8,12 @@
 ENV{ID_CDROM_DVD}=="?*",        SYMLINK+="dvd%e"
 ENV{ID_CDROM_DVD_R}=="?*",      SYMLINK+="dvdrw%e"
 
+# permissions for IDE CD devices
+BUS="ide", KERNEL="*[!0-9]", PROGRAM="/bin/cat /proc/ide/%k/media", RESULT="cdrom*", NAME="%k", GROUP="cdrom", MODE="0660"
+
+# permissions for SCSI cd/sg/tape devices
+BUS="scsi", KERNEL="s[grt][0-9]*", SYSFS{type}="5", NAME="%k", GROUP="cdrom", MODE="0660"
+
 # create a symlink named after the device map name
 # note devmap_name comes with extras/multipath
 KERNEL="dm-[0-9]*", PROGRAM="/sbin/devmap_name %M %m", NAME="%k", SYMLINK="%c"
