diff -Naur udev-079.orig/etc/udev/frugalware/udev.rules udev-079/etc/udev/frugalware/udev.rules
--- udev-079.orig/etc/udev/frugalware/udev.rules	2006-01-05 11:37:20.000000000 +0100
+++ udev-079/etc/udev/frugalware/udev.rules	2006-01-05 11:44:31.000000000 +0100
@@ -4,6 +4,9 @@
 # See the udev.rules.examples file for more examples of how to create rules
 #

+# wait for sysfs
+ACTION=="add", DEVPATH=="/devices/*", ENV{PHYSDEVBUS}=="?*", WAIT_FOR_SYSFS="bus"
+
 BUS=="ide",     KERNEL=="hd[a-z]", ACTION=="add", IMPORT="/sbin/cdrom_id --export $tempnode"
 BUS=="scsi",    KERNEL="sr[0-9]*", ACTION=="add", IMPORT="/sbin/cdrom_id --export $tempnode"
 BUS=="scsi",    KERNEL="scd[a-z]", ACTION=="add", IMPORT="/sbin/cdrom_id --export $tempnode"
@@ -180,3 +180,23 @@
 KERNEL="ipstate*"     NAME="%k", OWNER="root", GROUP="root",   MODE="0660"
 KERNEL="spx"          NAME="%k", OWNER="root", GROUP="root",   MODE="0644"
 KERNEL="fwmonitor*"   NAME="%k", OWNER="root", GROUP="root",   MODE="0660"
+
+# debugging monitor
+RUN+="socket:/org/kernel/udev/monitor"
+
+# hotplug rules - module autoloading for different buses
+ACTION=="add", SUBSYSTEM=="pci", MODALIAS=="*", RUN+="/sbin/modprobe $modalias"
+ACTION=="add", SUBSYSTEM=="usb", MODALIAS=="*", RUN+="/sbin/modprobe $modalias"
+ACTION=="add", SUBSYSTEM=="pcmcia", MODALIAS=="*", RUN+="/sbin/modprobe $modalias"
+ACTION=="add", SUBSYSTEM=="pcmcia", MODALIAS=="*", RUN+="/bin/sh -c 'echo 1 > /sys/$DEVPATH/allow_func_id_match'"
+ACTION=="add", SUBSYSTEM=="ieee1394", MODALIAS=="*", RUN+="/sbin/modprobe $modalias"
+ACTION=="add", SUBSYSTEM=="scsi", SYSFS{type}=="[07]", RUN+="/sbin/modprobe sd_mod"
+ACTION=="add", SUBSYSTEM=="scsi", SYSFS{type}=="14", RUN+="/sbin/modprobe sd_mod"
+ACTION=="add", SUBSYSTEM=="scsi", SYSFS{type}=="[45]", RUN+="/sbin/modprobe sr_mod"
+ACTION=="add", SUBSYSTEM=="scsi", SYSFS{type}=="1", SYSFS{vendor}=="Onstream", SYSFS{model}!="ADR*", RUN+="/sbin/modprobe osst"
+ACTION=="add", SUBSYSTEM=="scsi", SYSFS{type}=="1", SYSFS{vendor}=="Onstream", SYSFS{model}=="ADR*", RUN+="/sbin/modprobe st"
+ACTION=="add", SUBSYSTEM=="scsi", SYSFS{type}=="1", SYSFS{vendor}!="Onstream", RUN+="/sbin/modprobe st"
+ACTION=="add", SUBSYSTEM=="scsi", SYSFS{type}=="[23689]", RUN+="/sbin/modprobe sg"
+
+# firmware
+ACTION=="add", SUBSYSTEM=="firmware", ENV{FIRMWARE}=="*", RUN+="/lib/udev/firmware_helper"
