diff -Naur udev-034.orig/etc/udev/udev.permissions.slackware udev-034/etc/udev/udev.permissions.slackware
--- udev-034.orig/etc/udev/udev.permissions.slackware	2004-10-08 01:37:37.000000000 +0200
+++ udev-034/etc/udev/udev.permissions.slackware	2004-10-09 20:25:43.606265352 +0200
@@ -5,7 +5,8 @@
 
 # console devices
 console:root:tty:0600
-tty[0-9][0-9]*:root:tty:0666
+tty:root:tty:0666
+tty[0-9]*:root:tty:0660
 vcs*:root:tty:0620
 tty[CIS]*:root:uucp:0660
 ttyUB[0-9]:root:uucp:0660
@@ -117,7 +118,7 @@
 rtc:root:root:0444
 urandom:root:root:0644
 full:root:sys:0666
-zero:root:root:0664
+zero:root:root:0666
 raw/*:root:disk:0660
 modreq:root:root:0644
 msr:root:root:0660
