diff -Naur udev-064.orig/etc/udev/frugalware/udev.rules udev-064/etc/udev/frugalware/udev.rules
--- udev-064.orig/etc/udev/frugalware/udev.rules	2005-07-23 18:27:02.000000000 +0200
+++ udev-064/etc/udev/frugalware/udev.rules	2005-07-30 12:16:18.000000000 +0200
@@ -4,8 +4,13 @@
 # See the udev.rules.examples file for more examples of how to create rules
 #
 
-# if this is a ide cdrom, name it the default name, and create a symlink to cdrom
-# BUS="ide", KERNEL="*[!0-9]", PROGRAM="/bin/cat /proc/ide/%k/media", RESULT="cdrom", NAME="%k", SYMLINK="cdrom"
+BUS=="ide",     KERNEL=="hd[a-z]", ACTION=="add", IMPORT="/sbin/cdrom_id --export $tempnode"
+BUS=="scsi",    KERNEL="sr[0-9]*", ACTION=="add", IMPORT="/sbin/cdrom_id --export $tempnode"
+BUS=="scsi",    KERNEL="scd[a-z]", ACTION=="add", IMPORT="/sbin/cdrom_id --export $tempnode"
+ENV{ID_CDROM}=="?*",            SYMLINK+="cdrom%e"
+ENV{ID_CDROM_CD_RW}=="?*",      SYMLINK+="cdrw%e"
+ENV{ID_CDROM_DVD}=="?*",        SYMLINK+="dvd%e"
+ENV{ID_CDROM_DVD_R}=="?*",      SYMLINK+="dvdrw%e"
 
 # create a symlink named after the device map name
 # note devmap_name comes with extras/multipath
@@ -112,6 +113,7 @@
 
 # floppy devices
 KERNEL="fd[0-3]*"        NAME="%k", OWNER="root", GROUP="floppy", MODE="0660"
+KERNEL="fd0"             SYMLINK="floppy"
 
 # IDE devices (hd, cdrom)
 KERNEL="hd*"             NAME="%k", OWNER="root", GROUP="disk",   MODE="0660"
