diff --git a/libmount/python/meson.build b/libmount/python/meson.build
index 72985eca6..7e1703063 100644
--- a/libmount/python/meson.build
+++ b/libmount/python/meson.build
@@ -1,3 +1,9 @@
+python_module = import('python')
+
+python = python_module.find_installation(
+    get_option('python'),
+    required : get_option('build-python'),
+    disabler : true)
 build_python = python.found()
 
 pylibmount_sources = '''
