--- util-linux-2.40/meson.build	2024-03-27 12:46:59.000000000 +0100
+++ util-linux-2.40.new/meson.build	2024-03-31 14:12:54.971710827 +0200
@@ -3015,7 +3015,7 @@
 syscalls_h = custom_target('syscalls.h',
   input : 'tools/all_syscalls',
   output : 'syscalls.h',
-  command : ['tools/all_syscalls', cc.cmd_array()] 
+  command : ['tools/all_syscalls', cc.cmd_array(), get_option('c_args')] 
 )
 
 if cc.compiles(fs.read('include/audit-arch.h'), name : 'has AUDIT_ARCH_NATIVE')
