diff -urN nautilus-2.16.0.orig/libnautilus-private/apps_nautilus_preferences.schemas.in nautilus-2.16.0/libnautilus-private/apps_nautilus_preferences.schemas.in
--- nautilus-2.16.0.orig/libnautilus-private/apps_nautilus_preferences.schemas.in	2006-09-17 15:13:36.000000000 +0100
+++ nautilus-2.16.0/libnautilus-private/apps_nautilus_preferences.schemas.in	2006-09-17 15:21:44.000000000 +0100
@@ -43,7 +43,7 @@
       <applyto>/apps/nautilus/preferences/always_use_browser</applyto>
       <owner>nautilus</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
          <short>Enables the classic Nautilus behavior, where all windows are browsers</short>
          <long>
@@ -309,11 +309,11 @@
       <applyto>/apps/nautilus/preferences/desktop_font</applyto>
       <owner>nautilus</owner>
       <type>string</type>
-      <default>Sans 10</default>
+      <default>DejaVu Sans 13</default>
       <locale name="C">
          <default><!-- Translators: please note this can choose the size. e.g.
 "Sans 15". Please do not change "Sans", only change the size if you need to. In
-most cases, this should be left alone. -->Sans 10</default>
+most cases, this should be left alone. -->DejaVu Sans 13</default>
          <short>Desktop font</short>
 	 <long>
 	  The font description used for the icons on the desktop.
@@ -815,7 +815,7 @@
       <applyto>/apps/nautilus/desktop/network_icon_visible</applyto>
       <owner>nautilus</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
       	<short>Network Servers icon visible on the desktop</short>
 	<long>
diff -urN nautilus-2.16.0.orig/libnautilus-private/nautilus-global-preferences.c nautilus-2.16.0/libnautilus-private/nautilus-global-preferences.c
--- nautilus-2.16.0.orig/libnautilus-private/nautilus-global-preferences.c	2006-09-17 15:13:36.000000000 +0100
+++ nautilus-2.16.0/libnautilus-private/nautilus-global-preferences.c	2006-09-17 15:24:25.000000000 +0100
@@ -645,7 +645,7 @@
 static gpointer
 default_font_callback (void)
 {
-	return g_strdup ("sans 12");
+	return g_strdup ("DejaVu Sans 13");
 }
 
 /*
