diff -urN nautilus-2.15.92.1/libnautilus-private/apps_nautilus_preferences.schemas.in nautilus-2.15.92.1.new/libnautilus-private/apps_nautilus_preferences.schemas.in
--- nautilus-2.15.92.1/libnautilus-private/apps_nautilus_preferences.schemas.in	2006-06-15 10:38:40.000000000 +0100
+++ nautilus-2.15.92.1.new/libnautilus-private/apps_nautilus_preferences.schemas.in	2006-08-24 10:25:15.000000000 +0100
@@ -43,7 +43,7 @@
       <applyto>/apps/nautilus/preferences/always_use_browser</applyto>
       <owner>nautilus</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
          <short>Enables the classic Nautilus behavior, where all windows are browsers</short>
          <long>
@@ -309,11 +309,11 @@
       <applyto>/apps/nautilus/preferences/desktop_font</applyto>
       <owner>nautilus</owner>
       <type>string</type>
-      <default>Sans 10</default>
+      <default>Luxi Sans 10</default>
       <locale name="C">
          <default><!-- Translators: please note this can choose the size. e.g.
 "Sans 15". Please do not change "Sans", only change the size if you need to. In
-most cases, this should be left alone. -->Sans 10</default>
+most cases, this should be left alone. -->Luxi Sans 10</default>
          <short>Desktop font</short>
 	 <long>
 	  The font description used for the icons on the desktop.
