diff -Naur libinput-1.12.0/meson.build libinput-1.12.0-p/meson.build
--- libinput-1.12.0/meson.build	2018-09-11 05:40:41.000000000 +0200
+++ libinput-1.12.0-p/meson.build	2018-09-14 16:51:06.936310213 +0200
@@ -8,7 +8,7 @@
 
 dir_data        = join_paths(get_option('prefix'), get_option('datadir'), 'libinput')
 dir_sysconf     = join_paths(get_option('prefix'), get_option('sysconfdir'), 'libinput')
-dir_libexec     = join_paths(get_option('prefix'), get_option('libexecdir'), 'libinput')
+dir_libexec     = join_paths(get_option('prefix'), get_option('libexecdir'))
 dir_lib         = join_paths(get_option('prefix'), get_option('libdir'))
 dir_man1        = join_paths(get_option('prefix'), get_option('mandir'), 'man1')
 dir_system_udev = join_paths(get_option('prefix'), 'lib', 'udev')
