diff -Naur libinput-1.11.0/meson.build libinput-1.11.0-p/meson.build
--- libinput-1.11.0/meson.build	2018-06-05 02:27:11.000000000 +0200
+++ libinput-1.11.0-p/meson.build	2018-06-08 19:26:53.083802479 +0200
@@ -404,7 +404,7 @@
 endif
 
 ############ tools ############
-libinput_tool_path = join_paths(get_option('prefix'), get_option('libexecdir'), 'libinput')
+libinput_tool_path = join_paths(get_option('prefix'), get_option('libexecdir'))
 config_h.set_quoted('LIBINPUT_TOOL_PATH', libinput_tool_path)
 tools_shared_sources = [ 'tools/shared.c',
 			 'tools/shared.h' ]
