diff -Naur okular/CMakeLists.txt okular-no-mobile/CMakeLists.txt
--- okular/CMakeLists.txt	2016-04-13 22:50:40.574414291 +0200
+++ okular-no-mobile/CMakeLists.txt	2016-04-13 22:55:15.312944749 +0200
@@ -69,7 +69,7 @@
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${PHONON_INCLUDES} core/synctex ${ZLIB_INCLUDE_DIR} ${CMAKE_BINARY_DIR}/core)
 
-add_subdirectory( mobile )
+
 option(BUILD_COVERAGE "Build the project with gcov support" OFF)
 
 if(BUILD_COVERAGE)
