--- libspnav-1.0/Makefile.in	2022-04-01 05:04:40.000000000 +0200
+++ libspnav-1.0.new/Makefile.in	2022-04-28 12:44:19.297733783 +0200
@@ -59,8 +59,8 @@
 		ln -s $(soname) $(devlink) || \
 		true
 	for h in $(hdr); do cp -p $(srcdir)/$$h $(DESTDIR)$(PREFIX)/include/; done
-	mkdir -p $(DESTDIR)$(PREFIX)/share/pkgconfig
-	cp spnav.pc $(DESTDIR)$(PREFIX)/share/pkgconfig/spnav.pc
+	mkdir -p $(DESTDIR)$(PREFIX)/lib/pkgconfig
+	cp spnav.pc $(DESTDIR)$(PREFIX)/lib/pkgconfig/spnav.pc
 
 .PHONY: uninstall
 uninstall:
@@ -70,5 +70,5 @@
 		rm -f $(DESTDIR)$(PREFIX)/$(libdir)/$(soname) $(DESTDIR)$(PREFIX)/$(libdir)/$(devlink) || \
 		true
 	for i in $(hdr); do rm -f $(DESTDIR)$(PREFIX)/include/$$i; done
-	rm -f $(DESTDIR)$(PREFIX)/share/pkgconfig/spnav.pc
+	rm -f $(DESTDIR)$(PREFIX)/lib/pkgconfig/spnav.pc
 
