diff -Naur fun-0.1.0/README fun-0.1.0.new/README
--- fun-0.1.0/README	2007-11-21 23:37:19.000000000 +0530
+++ fun-0.1.0.new/README	2007-11-22 19:21:49.000000000 +0530
@@ -26,7 +26,7 @@
 Using FUN
 =========
 You need to start the daemon by running the following command as root:
-# service fund start
+# service fun start
 
 Once the daemon is started, you may launch the client by running the
 following command:
diff -Naur fun-0.1.0/src/fun-ui.c fun-0.1.0.new/src/fun-ui.c
--- fun-0.1.0/src/fun-ui.c	2007-11-21 23:37:19.000000000 +0530
+++ fun-0.1.0.new/src/fun-ui.c	2007-11-22 19:21:29.000000000 +0530
@@ -483,7 +483,7 @@
 								"that the update notifier daemon is not running. FUN will attempt "
 								"to reconnect to the daemon every 45 seconds. \n\nYou can start the "
 								"update notifier daemon by running the following command as root: \n\n"
-								"'service fund start'");
+								"'service fun start'");
 	
 	fun_systray_create ();
 	fun_main_window_init ();
