diff -Naur fun-0.3.2/src/fun-ui.c fun-0.3.2.new/src/fun-ui.c
--- fun-0.3.2/src/fun-ui.c	2009-03-12 19:33:47.000000000 +0530
+++ fun-0.3.2.new/src/fun-ui.c	2009-03-14 00:27:34.000000000 +0530
@@ -397,6 +397,12 @@
 					G_CALLBACK(cb_fun_config_dlg_close_clicked),
 					(gpointer)glade_xml_get_widget(xml,"interval_spbtn"));
 	
+	/* don't destroy the dialog on delete-event */
+	g_signal_connect (G_OBJECT(fun_config_dlg),
+					  "delete_event",
+					  G_CALLBACK(gtk_widget_hide),
+					  (gpointer)fun_config_dlg);
+	
 	return;
 }
 
