--- openssl-0.9.8e/Makefile	2007-03-02 12:26:53.000000000 +0100
+++ openssl-0.9.8e/Makefile.new	2007-03-02 12:47:15.000000000 +0100
@@ -566,7 +566,7 @@
 			(grep -v "[	]"; true) | \
 			(cd $(INSTALL_PREFIX)$(MANDIR)/man$$sec/; \
 			 while read n; do \
-				$$here/util/point.sh $$fn.$${sec}$(MANSUFFIX) "$$n".$${sec}$(MANSUFFIX); \
+				$$here/util/point.sh openssl_$$fn.$${sec}$(MANSUFFIX) "$$n".$${sec}$(MANSUFFIX); \
 			 done); \
 	done; \
 	set -e; for i in doc/crypto/*.pod doc/ssl/*.pod; do \
--- openssl-0.9.8e/Makefile	2007-03-02 12:49:34.000000000 +0100
+++ openssl-0.9.8e/Makefile.new	2007-03-02 12:58:57.000000000 +0100
@@ -583,7 +583,7 @@
 			(grep -v "[	]"; true) | \
 			(cd $(INSTALL_PREFIX)$(MANDIR)/man$$sec/; \
 			 while read n; do \
-				$$here/util/point.sh $$fn.$${sec}$(MANSUFFIX) "$$n".$${sec}$(MANSUFFIX); \
+				$$here/util/point.sh openssl_$$fn.$${sec}$(MANSUFFIX) "$$n".$${sec}$(MANSUFFIX); \
 			 done); \
 	done
 
