# This patch fixes a FTBFS with gcc 4.5. Closes: #565050

diff -Naur tmw-0.0.29.1.orig/src/emoteshortcut.cpp tmw-0.0.29.1/src/emoteshortcut.cpp
--- tmw-0.0.29.1.orig/src/emoteshortcut.cpp	2009-04-15 20:55:28.000000000 +0200
+++ tmw-0.0.29.1/src/emoteshortcut.cpp	2010-01-13 18:10:24.000000000 +0100
@@ -26,7 +26,7 @@
 
 #include "utils/stringutils.h"
 
-EmoteShortcut::EmoteShortcut *emoteShortcut;
+EmoteShortcut *emoteShortcut;
 
 EmoteShortcut::EmoteShortcut():
     mEmoteSelected(0)
