diff -Naur xsys-2.0.8.orig/parse.c xsys-2.0.8/parse.c
--- xsys-2.0.8.orig/parse.c	2005-09-22 16:38:17.000000000 +0200
+++ xsys-2.0.8/parse.c	2005-09-22 16:38:47.000000000 +0200
@@ -451,6 +451,8 @@
 		fgets(buffer, bsize, fp);
 	else if((fp = fopen("/etc/turbolinux-release", "r")) != NULL)
 		fgets(buffer, bsize, fp);
+	else if((fp = fopen("/etc/frugalware-release", "r")) != NULL)
+		fgets(buffer, bsize, fp);
 	else
 		snprintf(buffer, bsize, "Unknown Distro");
 	if(fp != NULL) fclose(fp);
