diff -ur a/dracut.conf b/dracut.conf
--- a/dracut.conf	2013-03-26 10:55:03.000000000 -0500
+++ b/dracut.conf	2013-05-11 19:20:23.581018491 -0500
@@ -4,7 +4,7 @@
 
 # Sample dracut config file
 
-#logfile=/var/log/dracut.log
+logfile=/tmp/dracut.log
 #fileloglvl=6
 
 # Exact list of dracut modules to use.  Modules not listed here are not going
@@ -48,4 +48,4 @@
 
 # set the directory for temporary files
 # default: /var/tmp
-#tmpdir=/tmp
+tmpdir=/tmp
