diff -Naur dracut-047/modules.d/99fs-lib/module-setup.sh dracut-047-p/modules.d/99fs-lib/module-setup.sh
--- dracut-047/modules.d/99fs-lib/module-setup.sh	2018-02-19 10:00:33.000000000 +0100
+++ dracut-047-p/modules.d/99fs-lib/module-setup.sh	2018-04-26 17:00:51.493789476 +0200
@@ -41,7 +41,7 @@
 include_fs_helper_modules() {
     local dev=$1 fs=$2
     case "$fs" in
-        xfs|btrfs)
+        xfs|btrfs|ext4|ext3)
             instmods crc32c
             ;;
         f2fs)
@@ -57,7 +57,7 @@
         for_each_host_dev_fs include_fs_helper_modules
         :
     else
-        instmods crc32c
+        instmods crc32c crc32
     fi
 }
 
