--- dracut-050/dracut-functions.sh	2020-03-04 11:43:22.000000000 +0100
+++ dracut-050.new/dracut-functions.sh	2020-03-05 19:45:14.669137448 +0100
@@ -76,7 +76,8 @@
 
 ldconfig_paths()
 {
-    $DRACUT_LDCONFIG ${dracutsysrootdir:+-r ${dracutsysrootdir} -f /etc/ld.so.conf} -pN 2>/dev/null | grep -E -v '/(lib|lib64|usr/lib|usr/lib64)/[^/]*$' | sed -n 's,.* => \(.*\)/.*,\1,p' | sort | uniq
+	$DRACUT_LDCONFIG ${dracutsysrootdir:+-r ${dracutsysrootdir} -f /etc/ld.so.conf} -pN 2>/dev/null | grep -E -v '/(lib|lib64|usr/lib|usr/lib64|usr/lib32|lib32)/[^/]*$' | sed -n 's,.* => \(.*\)/.*,\1,p' | sort | uniq
+
 }
 
 # Version comparision function.  Assumes Linux style version scheme.
