diff -Naur dracut-045/dracut-functions.sh dracut-045-p/dracut-functions.sh
--- dracut-045/dracut-functions.sh	2017-04-03 10:26:25.000000000 +0200
+++ dracut-045-p/dracut-functions.sh	2017-10-20 18:25:00.066161598 +0200
@@ -52,7 +52,9 @@
 
 ldconfig_paths()
 {
-    ldconfig -pN 2>/dev/null | grep -E -v '/(lib|lib64|usr/lib|usr/lib64)/[^/]*$' | sed -n 's,.* => \(.*\)/.*,\1,p' | sort | uniq
+    ## well we don't want ghc , java  , firefox! and so on to be even considered to be added
+    ## also for sure we do NOT want lib32 stuff in initramfs .. So we just grep -v all ^/usr/lib/<something> too .. -- crazy --
+    ldconfig -pN 2>/dev/null | grep -E -v '/(lib|lib64|usr/lib|usr/lib64|usr/lib32|lib32)/[^/]*$' | sed -n 's,.* => \(.*\)/.*,\1,p' | grep -v '^/usr/lib' | sort | uniq
 }
 
 # Version comparision function.  Assumes Linux style version scheme.
