--- dracut-ng-101/modules.d/62bluetooth/module-setup.sh	2024-04-07 17:04:17.000000000 +0200
+++ dracut-ng-101.new/modules.d/62bluetooth/module-setup.sh	2024-04-26 20:28:16.650295881 +0200
@@ -5,7 +5,7 @@
 # Prerequisite check(s) for module.
 check() {
     # If the binary(s) requirements are not fulfilled the module can't be installed
-    require_any_binary /usr/lib/bluetooth/bluetoothd /usr/libexec/bluetooth/bluetoothd || return 1
+    require_any_binary /usr/lib/bluetooth/bluetoothd /usr/libexec/bluetooth/bluetoothd /usr/lib/bluez/bluetooth/bluetoothd || return 1
 
     if [[ $hostonly ]]; then
         # Warn user if bluetooth kernel module is loaded
