diff -Naur dracut-045/dracut-initramfs-restore.sh dracut-045-p/dracut-initramfs-restore.sh
--- dracut-045/dracut-initramfs-restore.sh	2017-04-03 10:26:25.000000000 +0200
+++ dracut-045-p/dracut-initramfs-restore.sh	2017-05-29 13:34:34.592759092 +0200
@@ -19,7 +19,7 @@
 if [[ $MACHINE_ID ]] && [[ -d /boot/${MACHINE_ID} || -L /boot/${MACHINE_ID} ]] ; then
     IMG="/boot/${MACHINE_ID}/${KERNEL_VERSION}/initrd"
 fi
-[[ -f $IMG ]] || IMG="/boot/initramfs-${KERNEL_VERSION}.img"
+[[ -f $IMG ]] || IMG="/boot/initrd-${KERNEL_VERSION}"
 
 cd /run/initramfs
 
diff -Naur dracut-045/dracut.sh dracut-045-p/dracut.sh
--- dracut-045/dracut.sh	2017-04-03 10:26:25.000000000 +0200
+++ dracut-045-p/dracut.sh	2017-05-29 13:34:34.593759119 +0200
@@ -770,7 +770,7 @@
         if [[ $MACHINE_ID ]] && ( [[ -d /boot/${MACHINE_ID} ]] || [[ -L /boot/${MACHINE_ID} ]] ); then
             outfile="/boot/${MACHINE_ID}/$kernel/initrd"
         else
-            outfile="/boot/initramfs-$kernel.img"
+            outfile="/boot/initrd-$kernel"
         fi
     fi
 fi
diff -Naur dracut-045/lsinitrd.sh dracut-045-p/lsinitrd.sh
--- dracut-045/lsinitrd.sh	2017-04-03 10:26:25.000000000 +0200
+++ dracut-045-p/lsinitrd.sh	2017-05-29 13:34:34.594759145 +0200
@@ -99,7 +99,7 @@
         && [[ -d /boot/${MACHINE_ID} || -L /boot/${MACHINE_ID} ]] ; then
         image="/boot/${MACHINE_ID}/${KERNEL_VERSION}/initrd"
     else
-        image="/boot/initramfs-${KERNEL_VERSION}.img"
+        image="/boot/initrd-${KERNEL_VERSION}"
     fi
 fi
 
diff -Naur dracut-045/mkinitrd-dracut.sh dracut-045-p/mkinitrd-dracut.sh
--- dracut-045/mkinitrd-dracut.sh	2017-04-03 10:26:25.000000000 +0200
+++ dracut-045-p/mkinitrd-dracut.sh	2017-05-29 13:34:34.594759145 +0200
@@ -16,7 +16,7 @@
     $cmd "       [--nocompress]"
     $cmd "       <initrd-image> <kernel-version>"
     $cmd ""
-    $cmd "       (ex: ${0##*/} /boot/initramfs-$kver.img $kver)"
+    $cmd "       (ex: ${0##*/} /boot/initrd-$kver $kver)"
 
     [[ $1 = '-n' ]] && exit 0
     exit 1
diff -Naur dracut-045/test/TEST-16-DMSQUASH/test.sh dracut-045-p/test/TEST-16-DMSQUASH/test.sh
--- dracut-045/test/TEST-16-DMSQUASH/test.sh	2017-04-03 10:26:25.000000000 +0200
+++ dracut-045-p/test/TEST-16-DMSQUASH/test.sh	2017-05-29 13:34:34.595759172 +0200
@@ -86,7 +86,7 @@
 	done
         inst_simple /etc/os-release
 	inst ./test-init.sh /sbin/init
-	inst "$TESTDIR"/initramfs.testing "/boot/initramfs-$KVERSION.img"
+	inst "$TESTDIR"/initramfs.testing "/boot/initrd-$KVERSION"
         [[ -f /etc/machine-id ]] && read MACHINE_ID < /etc/machine-id
 
 	VMLINUZ="/lib/modules/${KVERSION}/vmlinuz"
