diff -Naur dracut-050/modules.d/00systemd/module-setup.sh dracut-050-p/modules.d/00systemd/module-setup.sh
--- dracut-050/modules.d/00systemd/module-setup.sh	2020-03-04 11:43:22.000000000 +0100
+++ dracut-050-p/modules.d/00systemd/module-setup.sh	2020-03-28 11:53:59.072919983 +0100
@@ -29,7 +29,7 @@
 }
 
 installkernel() {
-    hostonly='' instmods autofs4 ipv6 algif_hash hmac sha256
+    hostonly='' instmods autofs4 algif_hash hmac sha256
     instmods -s efivarfs
 }
 
