diff -Naur dracut-050/modules.d/00systemd/module-setup.sh dracut-050-p/modules.d/00systemd/module-setup.sh
--- dracut-050/modules.d/00systemd/module-setup.sh	2020-03-04 11:43:22.000000000 +0100
+++ dracut-050-p/modules.d/00systemd/module-setup.sh	2020-03-29 16:01:40.487966347 +0200
@@ -101,7 +101,6 @@
         $systemdsystemunitdir/systemd-ask-password-console.path \
         $systemdsystemunitdir/systemd-udevd-control.socket \
         $systemdsystemunitdir/systemd-udevd-kernel.socket \
-        $systemdsystemunitdir/systemd-ask-password-plymouth.path \
         $systemdsystemunitdir/systemd-journald.socket \
         $systemdsystemunitdir/systemd-journald-audit.socket \
         $systemdsystemunitdir/systemd-ask-password-console.service \
@@ -114,7 +113,6 @@
         $systemdsystemunitdir/systemd-udevd.service \
         $systemdsystemunitdir/systemd-udev-trigger.service \
         $systemdsystemunitdir/systemd-udev-settle.service \
-        $systemdsystemunitdir/systemd-ask-password-plymouth.service \
         $systemdsystemunitdir/systemd-journald.service \
         $systemdsystemunitdir/systemd-vconsole-setup.service \
         $systemdsystemunitdir/systemd-volatile-root.service \
@@ -238,8 +236,7 @@
     for i in \
         emergency.target \
         rescue.target \
-        systemd-ask-password-console.service \
-        systemd-ask-password-plymouth.service \
+        systemd-ask-password-console.service
         ; do
         systemctl -q --root "$initdir" add-wants "$i" systemd-vconsole-setup.service
     done
