--- dracut-052/modules.d/00systemd/module-setup.sh	2021-02-15 17:30:54.000000000 +0100
+++ dracut-052.n/modules.d/00systemd/module-setup.sh	2021-02-23 11:11:37.136963937 +0100
@@ -103,7 +103,6 @@
         "$systemdsystemunitdir"/systemd-ask-password-console.path \
         "$systemdsystemunitdir"/systemd-udevd-control.socket \
         "$systemdsystemunitdir"/systemd-udevd-kernel.socket \
-        "$systemdsystemunitdir"/systemd-ask-password-plymouth.path \
         "$systemdsystemunitdir"/systemd-journald.socket \
         "$systemdsystemunitdir"/systemd-journald-audit.socket \
         "$systemdsystemunitdir"/systemd-ask-password-console.service \
@@ -116,7 +115,6 @@
         "$systemdsystemunitdir"/systemd-udevd.service \
         "$systemdsystemunitdir"/systemd-udev-trigger.service \
         "$systemdsystemunitdir"/systemd-udev-settle.service \
-        "$systemdsystemunitdir"/systemd-ask-password-plymouth.service \
         "$systemdsystemunitdir"/systemd-journald.service \
         "$systemdsystemunitdir"/systemd-vconsole-setup.service \
         "$systemdsystemunitdir"/systemd-volatile-root.service \
@@ -241,7 +239,6 @@
         emergency.target \
         rescue.target \
         systemd-ask-password-console.service \
-        systemd-ask-password-plymouth.service \
         ; do
         [[ -f "$systemdsystemunitdir"/$i ]] || continue
         $SYSTEMCTL -q --root "$initdir" add-wants "$i" systemd-vconsole-setup.service
