diff -Naur dracut-050/modules.d/00systemd/module-setup.sh dracut-050-p/modules.d/00systemd/module-setup.sh
--- dracut-050/modules.d/00systemd/module-setup.sh	2020-03-28 12:13:40.227817629 +0100
+++ dracut-050-p/modules.d/00systemd/module-setup.sh	2020-03-28 12:13:15.937719107 +0100
@@ -101,7 +101,6 @@
         $systemdsystemunitdir/systemd-ask-password-console.path \
         $systemdsystemunitdir/systemd-udevd-control.socket \
         $systemdsystemunitdir/systemd-udevd-kernel.socket \
-        $systemdsystemunitdir/systemd-ask-password-plymouth.path \
         $systemdsystemunitdir/systemd-journald.socket \
         $systemdsystemunitdir/systemd-journald-audit.socket \
         $systemdsystemunitdir/systemd-ask-password-console.service \
@@ -114,7 +113,6 @@
         $systemdsystemunitdir/systemd-udevd.service \
         $systemdsystemunitdir/systemd-udev-trigger.service \
         $systemdsystemunitdir/systemd-udev-settle.service \
-        $systemdsystemunitdir/systemd-ask-password-plymouth.service \
         $systemdsystemunitdir/systemd-journald.service \
         $systemdsystemunitdir/systemd-vconsole-setup.service \
         $systemdsystemunitdir/systemd-volatile-root.service \
