diff -Naur dracut-047/modules.d/90dmsquash-live/dmsquash-generator.sh dracut-047.new/modules.d/90dmsquash-live/dmsquash-generator.sh
--- dracut-047/modules.d/90dmsquash-live/dmsquash-generator.sh	2018-02-19 10:00:33.000000000 +0100
+++ dracut-047.new/modules.d/90dmsquash-live/dmsquash-generator.sh	2018-04-03 09:27:39.770986599 +0200
@@ -78,5 +78,4 @@
 {
     echo "[Unit]"
     echo "JobTimeoutSec=3000"
-    echo "JobRunningTimeoutSec=3000"
 } > "$GENERATOR_DIR/$_dev.device.d/timeout.conf"
diff -Naur dracut-047/modules.d/90livenet/livenet-generator.sh dracut-047.new/modules.d/90livenet/livenet-generator.sh
--- dracut-047/modules.d/90livenet/livenet-generator.sh	2018-02-19 10:00:33.000000000 +0100
+++ dracut-047.new/modules.d/90livenet/livenet-generator.sh	2018-04-03 09:27:47.559986347 +0200
@@ -57,5 +57,4 @@
 {
     echo "[Unit]"
     echo "JobTimeoutSec=3000"
-    echo "JobRunningTimeoutSec=3000"
 } > "$GENERATOR_DIR/dev-mapper-live\x2drw.device.d/timeout.conf"
diff -Naur dracut-047/modules.d/98dracut-systemd/rootfs-generator.sh dracut-047.new/modules.d/98dracut-systemd/rootfs-generator.sh
--- dracut-047/modules.d/98dracut-systemd/rootfs-generator.sh	2018-02-19 10:00:33.000000000 +0100
+++ dracut-047.new/modules.d/98dracut-systemd/rootfs-generator.sh	2018-04-03 09:27:32.335986840 +0200
@@ -32,7 +32,6 @@
         {
             echo "[Unit]"
             echo "JobTimeoutSec=$_timeout"
-            echo "JobRunningTimeoutSec=$_timeout"
         } > "$GENERATOR_DIR"/${_name}.device.d/timeout.conf
     fi
 }
diff -Naur dracut-047/modules.d/99base/dracut-lib.sh dracut-047.new/modules.d/99base/dracut-lib.sh
--- dracut-047/modules.d/99base/dracut-lib.sh	2018-02-19 10:00:33.000000000 +0100
+++ dracut-047.new/modules.d/99base/dracut-lib.sh	2018-04-03 09:27:21.155987203 +0200
@@ -938,7 +938,6 @@
             {
                 echo "[Unit]"
                 echo "JobTimeoutSec=$_timeout"
-                echo "JobRunningTimeoutSec=$_timeout"
             } > ${PREFIX}/etc/systemd/system/${_name}.device.d/timeout.conf
             type mark_hostonly >/dev/null 2>&1 && mark_hostonly /etc/systemd/system/${_name}.device.d/timeout.conf
             _needreload=1
