diff -urN Mesa-6.4.orig/bin/installmesa Mesa-6.4/bin/installmesa
--- Mesa-6.4.orig/bin/installmesa	2005-09-13 06:36:17.000000000 -0700
+++ Mesa-6.4/bin/installmesa	2005-10-13 14:10:39.000000000 -0700
@@ -60,7 +60,6 @@
 
 mkdir ${INCLUDE_DIR}
 mkdir ${INCLUDE_DIR}/GL
-mkdir ${INCLUDE_DIR}/GLES
 mkdir ${LIB_DIR}
 cp -f ${TOP}/include/GL/*.h ${INCLUDE_DIR}/GL
 # NOT YET: cp -f ${TOP}/include/GLES/*.h ${INCLUDE_DIR}/GLES
