--- gdb-6.3.orig/gdb/Makefile.in	2004-11-03 21:18:49.000000000 -0500
+++ gdb-6.3/gdb/Makefile.in	2004-11-09 23:04:57.204160812 -0500
@@ -121,10 +121,16 @@ BFD_SRC = $(srcdir)/$(BFD_DIR)
 BFD_CFLAGS = -I$(BFD_DIR) -I$(BFD_SRC)
 
 # Where is the READLINE library?  Typically in ../readline.
-READLINE_DIR = ../readline
-READLINE = $(READLINE_DIR)/libreadline.a
-READLINE_SRC = $(srcdir)/$(READLINE_DIR)
-READLINE_CFLAGS = -I$(READLINE_SRC)/..
+#READLINE_DIR = ../readline
+#READLINE = $(READLINE_DIR)/libreadline.a
+#READLINE_SRC = $(srcdir)/$(READLINE_DIR)
+#READLINE_CFLAGS = -I$(READLINE_SRC)/..
+#
+# Debian has a perfectly fine readline - use it.
+READLINE_SRC=/usr/include/readline
+READLINE=-lreadline
+READLINE_CFLAGS=
+READLINE_DIR=
 
 WARN_CFLAGS = @WARN_CFLAGS@
 WERROR_CFLAGS = @WERROR_CFLAGS@
