diff -Naur gdb-6.3.orig/gdb/doc/Makefile.in gdb-6.3/gdb/doc/Makefile.in
--- gdb-6.3.orig/gdb/doc/Makefile.in	2005-05-26 13:08:03.000000000 +0000
+++ gdb-6.3/gdb/doc/Makefile.in	2005-05-26 13:09:38.000000000 +0000
@@ -58,13 +58,13 @@
 
 # Where is the source dir for the READLINE library doc?  
 # Traditionally readline is in .. or .
-READLINE_DIR = ${gdbdir}/../readline/doc
+#READLINE_DIR = ${gdbdir}/../readline/doc
 
 # The GDB/MI docs come from a sibling directory ../mi
 GDBMI_DIR = ${gdbdir}/mi
 
 SET_TEXINPUTS = \
-   TEXINPUTS=${TEXIDIR}:.:$(srcdir):$(READLINE_DIR):$(GDBMI_DIR):$$TEXINPUTS
+   TEXINPUTS=${TEXIDIR}:.:$(srcdir):$(GDBMI_DIR):$$TEXINPUTS
 
 # Files which should be generated via 'info' and installed by 'install-info'
 INFO_DEPS = gdb.info gdbint.info stabs.info annotate.info
@@ -100,9 +100,7 @@
 GDB_DOC_SOURCE_INCLUDES = \
 	$(srcdir)/fdl.texi \
 	$(srcdir)/gpl.texi \
-	$(srcdir)/agentexpr.texi \
-	$(READLINE_DIR)/rluser.texi \
-	$(READLINE_DIR)/inc-hist.texinfo
+	$(srcdir)/agentexpr.texi
 GDB_DOC_BUILD_INCLUDES = \
 	gdb-cfg.texi \
 	GDBvn.texi
@@ -299,7 +297,7 @@
 
 # GDB MANUAL: info file
 gdb.info: ${GDB_DOC_FILES}
-	$(MAKEINFO) -I ${READLINE_DIR} -I ${GDBMI_DIR} -I $(srcdir) \
+	$(MAKEINFO) -I ${GDBMI_DIR} -I $(srcdir) \
 		-o gdb.info $(srcdir)/gdb.texinfo
 
 # GDB MANUAL: roff translations
@@ -375,7 +373,7 @@
 # GDB MANUAL: HTML file
 
 gdb_toc.html: ${GDB_DOC_FILES}
-	$(MAKEHTML) $(MAKEHTMLFLAGS) -I ${READLINE_DIR} -I ${GDBMI_DIR} -I $(srcdir) $(srcdir)/gdb.texinfo
+	$(MAKEHTML) $(MAKEHTMLFLAGS) -I ${GDBMI_DIR} -I $(srcdir) $(srcdir)/gdb.texinfo
 
 # Clean these up before each run.  Avoids a catch 22 with not being
 # able to re-generate these files (to fix a corruption) because these
