diff --git a/src/gens/emulator/g_main.c b/src/gens/emulator/g_main.c
index 1c9f4f5..3ac13eb 100644
--- a/src/gens/emulator/g_main.c
+++ b/src/gens/emulator/g_main.c
@@ -5,6 +5,7 @@
 #include <sys/stat.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <string.h>
 #include "g_main.h"
 #include "timer.h"
 #include "gens.h"
diff --git a/src/gens/emulator/g_main.h b/src/gens/emulator/g_main.h
index c7ea4b2..9090766 100644
--- a/src/gens/emulator/g_main.h
+++ b/src/gens/emulator/g_main.h
@@ -36,6 +36,8 @@ extern char **language_name;
 extern unsigned char Keys[];
 extern unsigned char joystate[];
 
+static int Build_Language_String (void);
+
 void SetWindowText(const char *text);
 
 int Set_Render(int Full, int Num, int Force);
diff --git a/src/gens/emulator/parse.c b/src/gens/emulator/parse.c
index 5bece4a..67f8b49 100644
--- a/src/gens/emulator/parse.c
+++ b/src/gens/emulator/parse.c
@@ -1,6 +1,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <getopt.h>
+#include <string.h>
 #include "port.h"
 #include "save.h"
 #include "cpu_68k.h"
