diff -Naur avidemux_2.5.3//avidemux/ADM_coreImage/include/ADM_mmxMacros.h avidemux_2.5.3-2//avidemux/ADM_coreImage/include/ADM_mmxMacros.h
--- avidemux_2.5.3//avidemux/ADM_coreImage/include/ADM_mmxMacros.h	2008-05-04 19:05:41.000000000 +0200
+++ avidemux_2.5.3-2//avidemux/ADM_coreImage/include/ADM_mmxMacros.h	2010-05-30 18:05:08.686065919 +0200
@@ -273,7 +273,7 @@
 			mmx_trace.d[1], mmx_trace.d[0]); \
 		__asm__ __volatile__ (#op " %0, %%" #reg \
 				      : /* nothing */ \
-				      : "X" (mem)); \
+				      : "ym" (mem)); \
 		__asm__ __volatile__ ("movq %%" #reg ", %0" \
 				      : "=X" (mmx_trace) \
 				      : /* nothing */ ); \
@@ -353,7 +353,7 @@
 #define	mmx_m2r(op, mem, reg) \
 	__asm__ __volatile__ (#op " %0, %%" #reg \
 			      : /* nothing */ \
-			      : "X" (mem))
+			      : "ym" (mem))
 
 #define	mmx_r2m(op, reg, mem) \
 	__asm__ __volatile__ (#op " %%" #reg ", %0" \
