diff -Naur KCDemu/kcdemu.desktop KCDemu-p/kcdemu.desktop
--- KCDemu/kcdemu.desktop	1970-01-01 01:00:00.000000000 +0100
+++ KCDemu-p/kcdemu.desktop	2006-07-19 12:28:21.000000000 +0200
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=KCDemu
+Comment=A GUI for the CDemu kernel module
+Exec=KCDemu
+Icon=cdimage
+Terminal=false
+Type=Application
+Categories=Qt;KDE;Utility;Application;
+Encoding=UTF-8
diff -Naur KCDemu/Makefile KCDemu-p/Makefile
--- KCDemu/Makefile	2005-11-19 20:20:13.000000000 +0100
+++ KCDemu-p/Makefile	2006-07-19 12:29:34.000000000 +0200
@@ -3,16 +3,19 @@
 OBJ		= lib
 SRC_ROOT	= `pwd`
 LFLAGS		= -L$(QTDIR)/lib/ -L$(KDEDIR)/lib/ -lqt-mt -lkdeui -lkio
-CFLAGS		= -I$(QTDIR)/include/ -I$(KDEDIR)/include/ -I$(SRC_ROOT)
+CFLAGS		+= -I$(QTDIR)/include/ -I$(KDEDIR)/include/ -I$(SRC_ROOT)
 ECHO		= @echo
 
 do-it-always:
 	for i in $(OBJ); do $(MAKE) -C $$i SRC_ROOT=$(SRC_ROOT); done
 	gcc -c KCDemu.cpp $(CFLAGS)
-	gcc -o $(TARGET) KCDemu.o $(addsuffix /*.o,$(OBJ)) $(LFLAGS)
+	gcc -o  $(TARGET) KCDemu.o $(addsuffix /*.o,$(OBJ)) $(LFLAGS)
 
 install:
-	cp KCDemu $(KDEDIR)/bin/
+	mkdir -p $(DESTDIR)$(KDEDIR)/bin
+	/usr/bin/install -c -g root -o root -m 755  KCDemu $(DESTDIR)$(KDEDIR)/bin/
+	mkdir -p $(DESTDIR)/usr/share/applications
+	/usr/bin/install -c -g root -o root -m 755 kcdemu.desktop $(DESTDIR)/usr/share/applications/
 
 clean:
 	for i in $(OBJ); do rm $$i/*.o; done
