diff -Naur blam-1.8.2.orig/configure.in blam-1.8.2/configure.in
--- blam-1.8.2.orig/configure.in	2005-09-24 02:40:16.000000000 +0200
+++ blam-1.8.2/configure.in	2005-09-24 02:41:03.000000000 +0200
@@ -51,14 +51,14 @@
 
 PKG_CHECK_MODULES(BASE_DEPENDENCIES, 
 		  mono >= $MONO_REQUIRED_VERSION
-		  gtk-sharp >= $GTKSHARP_REQUIRED_VERSION
+		  gtk-sharp-2.0 >= $GTKSHARP_REQUIRED_VERSION
                   gconf-2.0 >= $GCONF_REQUIRED_VERSION)
 
 PKG_CHECK_MODULES(BLAM,
-		  gtk-sharp >= $GTKSHARP_REQUIRED_VERSION
-                  gecko-sharp >= $GECKOSHARP_REQUIRED_VERSION
-		  gconf-sharp >= $GTKSHARP_REQUIRED_VERSION
-		  glade-sharp >= $GTKSHARP_REQUIRED_VERSION)
+		  gtk-sharp-2.0 >= $GTKSHARP_REQUIRED_VERSION
+                  gecko-sharp-2.0 >= $GECKOSHARP_REQUIRED_VERSION
+		  gconf-sharp-2.0 >= $GTKSHARP_REQUIRED_VERSION
+		  glade-sharp-2.0 >= $GTKSHARP_REQUIRED_VERSION)
 
 AC_SUBST(BLAM_LIBS)
 
