--- jdk16u-jdk-16-9/src/hotspot/share/gc/shenandoah/shenandoahPhaseTimings.hpp	2020-08-05 11:59:10.000000000 +0200
+++ jdk16u-jdk-16-9.new/src/hotspot/share/gc/shenandoah/shenandoahPhaseTimings.hpp	2021-04-21 13:52:11.915117976 +0200
@@ -25,6 +25,7 @@
 #ifndef SHARE_GC_SHENANDOAH_SHENANDOAHPHASETIMINGS_HPP
 #define SHARE_GC_SHENANDOAH_SHENANDOAHPHASETIMINGS_HPP
 
+#include "jfr/jfrEvents.hpp"
 #include "gc/shenandoah/shenandoahNumberSeq.hpp"
 #include "gc/shared/workerDataArray.hpp"
 #include "memory/allocation.hpp"
