diff -Naur fltk-2.0.x-r5274/glut/Makefile fltk-2.0.x-r5274-p/glut/Makefile
--- fltk-2.0.x-r5274/glut/Makefile	2006-07-28 10:58:12.000000000 +0200
+++ fltk-2.0.x-r5274-p/glut/Makefile	2006-08-08 22:53:28.622720472 +0200
@@ -1,121 +1,130 @@
 #
-# "$Id: Makefile 5273 2006-07-28 08:58:12Z fabien $"
+# Please report all bugs and problems to "fltk-bugs@fltk.org".
 #
-# Library makefile for the Fast Light Tool Kit (FLTK).
+
 #
-# Copyright 1998-2003 by Bill Spitzak and others.
+# Include common definitions...
 #
-# This library is free software; you can redistribute it and/or
-# modify it under the terms of the GNU Library General Public
-# License as published by the Free Software Foundation; either
-# version 2 of the License, or (at your option) any later version.
-#
-# This library is distributed in the hope that it will be useful,
-# but WITHOUT ANY WARRANTY; without even the implied warranty of
-# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-# Library General Public License for more details.
-#
-# You should have received a copy of the GNU Library General Public
-# License along with this library; if not, write to the Free Software
-# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307
-# USA.
+
+include ../makeinclude
+DSOCFLAGS += -DFL_GLUT_LIBRARY
+DSOCXXFLAGS += -DFL_GLUT_LIBRARY
+LDLIBS := $(GLUTLIBS) $(LDLIBS)
+
 #
-# Please report all bugs and problems to "fltk-bugs@fltk.org".
+# Files for this directory...
 #
 
 CPPFILES = \
 	glut_compatability.cxx \
 	glut_font.cxx
 
+CFILES	=
 
-DEMOS = \
-	glpuzzle fractals
+LIBNAME	=	$(LIBPREFIX)fltk2_glut$(LIBSUFFIX)
+DSONAME	=	$(DSOPREFIX)fltk2_glut$(DSOSUFFIX)
+DSOLINK = 	$(DSOPREFIX)fltk2_glut$(DSOSYMLINK)
+OBJECTS =	$(CPPFILES:.cxx=.o) $(CFILES:.c=.o)
 
-################################################################
+#
+# Make everything...
+#
 
+all:		$(LIBTARGET) $(DSOTARGET)
 
-include ../makeinclude
-DSOCFLAGS += -DFL_GLUT_LIBRARY
-DSOCXXFLAGS += -DFL_GLUT_LIBRARY
-LDLIBS := $(GLUTLIBS) $(LDLIBS)
+static:		../lib/$(LIBNAME)
 
-LIBNAME	= ../lib/$(LIBPREFIX)fltk2_glut$(LIBSUFFIX)
+shared:		../lib/$(DSONAME)
 
-OBJECTS = $(CPPFILES:.cxx=.o) $(CFILES:.c=.o)
 
-all:	static
+#
+# Clean old files...
+#
 
-static:	$(LIBNAME)
+clean:
+	$(RM) *.bck
+	$(RM) *.o
+	$(RM) core*
+	$(RM) ../lib/$(LIBNAME)
+	$(RM) ../lib/$(DSONAME)
+        ifeq ($(DSONAME), fltk2_glut.dll)
+	$(RM) lib$(DSONAME).a $(DSONAME)
+        endif
 
-shared:	$(DSO)$(DSOEXT)
+#
+# Make dependencies, excluding standard include directories...
+#
 
-demos: $(DEMOS)
+depend:
+	$(MAKEDEPEND) -I.. $(CPPFILES) $(CFILES) > makedepend
 
-programs: demos
 
-$(DEMOS): $(FLGLUTLIB) $(FLGLLIB) $(FLLIB)
+#
+# Include automatically generated dependencies...
+#
 
-$(LIBNAME): $(OBJECTS)
-	@rm -f $@
-	$(LIBCOMMAND) $@ $(OBJECTS)
-	$(RANLIB) $@
+include makedepend
 
-$(DSO).so.2 $(DSO).sl.2: $(OBJECTS)
-	@rm -f $@
-	$(DSOCOMMAND) $(DSONAME) $(OBJECTS) $(DSOLIBS)
-	mv $(DSONAME) $@
-
-DLLIMPORTLIBS = ../lib/$(LIBNAME)_gl.2.dll$(LIBEXT) ../lib/$(LIBNAME).2.dll$(LIBEXT)
-DLLIMPORTNAME = $(LIBNAME)_glut.2.dll$(LIBEXT)
-$(DSO).2.dll: $(OBJECTS)
-	@rm -f $@
-	$(DSOCOMMAND) $@ $(OBJECTS) $(DSOLIBS) $(DLLIMPORTLIBS)
-	mv $@.import ../lib/$(DLLIMPORTNAME)
-
-#.fl.cxx .fl.h:
-#	echo Generating $@...
-#	../fluid/fluid -c $<
-
-clean :
-	@rm -f *.o $(DEMOS) *.exe $(LIBNAME) $(DSO)$(DSOEXT) \
-	       ../lib/$(DLLIMPORTNAME) core *~ makedepend
-	@touch makedepend
 
-depend:	$(CPPFILES)
-	$(MAKEDEPEND) -I.. $(CPPFILES) > makedepend
+#
+# Make static libraries...
+#
 
-# Automatically generated dependencies...
-include makedepend
+../lib/$(LIBNAME): $(OBJECTS)
+	$(RM) $@
+	echo $(LIBCOMMAND) $@ ...
+	$(LIBCOMMAND) $@ $(OBJECTS)
+	$(RANLIB) $@
 
+../lib/$(DSONAME): $(OBJECTS)
+	echo $(DSOCOMMAND) $@ ...
+	$(DSOCOMMAND) $@ $(OBJECTS)
+	$(RM) ../lib/$(DSOLINK)
+	$(LN) $(DSONAME) ../lib/$(DSOLINK)
+
+../lib/fltk2_glut.dll: $(OBJECTS)
+	echo $(DSOCOMMAND) $@ ...
+	$(DSOCOMMAND) $@ $(OBJECTS) $(IMAGELIBS) $(LOCAL_IMAGELIBS) $(LINKFLTK) $(GLDLIBS)
+	cp ../lib/fltk2_gl.dll ./
+
+#
+# Install everything...
+#
+
+install:
+        ifneq (,$(wildcard ../lib/$(LIBNAME)))
+	echo "Installing static glut library in $(DESTDIR)$(libdir)"
+	$(MKDIR) $(DESTDIR)$(libdir)
+	$(RM) $(DESTDIR)$(libdir)/$(LIBNAME)
+	$(CP) ../lib/$(LIBNAME) $(DESTDIR)$(libdir)
+        endif
+        ifneq (,$(wildcard ../lib/$(DSONAME)))
+	echo "Installing shared glut library in $(DESTDIR)$(libdir)"
+	$(MKDIR) $(DESTDIR)$(libdir)
+	$(RM) $(DESTDIR)$(libdir)/$(DSONAME)
+	$(RM) $(DESTDIR)$(libdir)/$(DSOLINK)
+	$(CP) ../lib/$(DSONAME) $(DESTDIR)$(libdir)
+	$(LN) -s $(DSONAME) $(DESTDIR)$(libdir)/$(DSOLINK)
+        endif
+        ifeq ($(DSONAME), fltk2_glut.dll)
+	$(CP) lib$(DSONAME).a $(DESTDIR)$(libdir)
+        endif
 
-################################################################
+#
+# Uninstall the libraries...
+#
 
-install: $(INSTALLTARGETS)
 uninstall:
-	# echo nothing to do ?
-install_static: static
-	@echo "Installing static libraries..."
-	@mkdir -p $(libdir)
-	@rm -f $(libdir)/$(LIBNAME)_glut$(LIBEXT)
-	@cp $(LIBNAME) $(libdir)
-	@-chmod 644 $(libdir)/$(LIBNAME)_glut$(LIBEXT) 
-
-install_shared: shared
-	@echo "Installing shared libraries..."
-	@rm -f $(libdir)/$(DLLIMPORTNAME)
-	@if [ -f "$(libdir)/$(DSOLIBglutNAME)$(DSOEXT)" ]; then rm -f $(libdir)/$(DSOLIBglutNAME)$(DSOEXT)
-	@if [ "$(DSOSHORTEXT)" ]; then rm -f $(libdir)/$(DSOLIBglutNAME)$(DSOSHORTEXT); fi
-	@cp $(DSO)$(DSOEXT) $(libdir)
-	@chmod 755 $(libdir)/$(DSOLIBglutNAME)$(DSOEXT)
-	@if [ "$(DSOSHORTEXT)" ]; then \
-		ln -s $(DSOLIBglutNAME)$(DSOEXT) $(libdir)/$(DSOLIBglutNAME)$(DSOSHORTEXT); fi
-	@if [ -f "../lib/$(DLLIMPORTNAME)" ]; then cp ../lib/$(DLLIMPORTNAME) $(libdir); fi
+	echo "Removing static glut library from $(libdir)"
+	$(RM) $(libdir)/$(LIBNAME)
+	echo "Removing shared glut library from $(libdir)"
+	$(RM) $(libdir)/$(DSONAME)
+	$(RM) $(libdir)/$(DSOLINK)
+        ifeq ($(DSONAME), fltk2_glut.dll)
+	$(RM) $(libdir)/lib$(DSONAME).a
+        endif
 
-install_programs:
 
-#
-# End of "$Id: Makefile 5273 2006-07-28 08:58:12Z fabien $".
-#
 # DO NOT DELETE
 
 glut_compatability.o: ../config.h ../fltk/glut.h ../fltk/gl.h ../fltk/Color.h
@@ -130,3 +139,4 @@
 glut_font.o: ../fltk/GlWindow.h ../fltk/Window.h ../fltk/Group.h
 glut_font.o: ../fltk/Widget.h ../fltk/Style.h ../fltk/Rectangle.h
 glut_font.o: ../fltk/Cursor.h ../fltk/visual.h ../fltk/gl.h
+
diff -Naur fltk-2.0.x-r5274/README.windows fltk-2.0.x-r5274-p/README.windows
--- fltk-2.0.x-r5274/README.windows	2006-07-28 10:58:12.000000000 +0200
+++ fltk-2.0.x-r5274-p/README.windows	2006-08-08 22:53:07.346954880 +0200
@@ -1,8 +1,8 @@
 BUILDING FLTK UNDER MICROSOFT WINDOWS
 
     There are three ways to build FLTK under Microsoft Windows.  The
-    first is to use the Microsoft Visual C++ 6.0 project files under
-    the "ide/visualc" directory.  Just open (or double-click on) the
+    first is to use the Microsoft Visual C++ 5.0 project files under
+    the "visualc" directory.  Just open (or double-click on) the
     "fltk.dsw" file to get the whole shebang.
 
     The second method is to use Borland C++. Please read the
