diff -Naur fltk-2.0.x-r5508/fltk/MultiImage.h fltk-2.0.x-r5508-p/fltk/MultiImage.h
--- fltk-2.0.x-r5508/fltk/MultiImage.h	2006-10-02 00:11:29.000000000 +0200
+++ fltk-2.0.x-r5508-p/fltk/MultiImage.h	2006-10-07 20:27:55.000000000 +0200
@@ -49,7 +49,7 @@
   bool fills_rectangle() const;
   bool is_frame() const;
 
-  MultiImage::~MultiImage() { delete [] pairs; }
+  ~MultiImage() { delete [] pairs; }
 
   //! for MultiImage arrays construction using set() for post initialization
   MultiImage() { pairs=0; n_images = 0; } 
