--- seahorse-3.38.1/subprojects/libhandy/src/hdy-main.c	2021-02-15 16:09:45.517142300 +0100
+++ seahorse-3.38.1.new/subprojects/libhandy/src/hdy-main.c	2022-08-05 15:22:47.582838516 +0200
@@ -122,7 +122,7 @@
 static void
 hdy_style_init (void)
 {
-  static volatile gsize guard = 0;
+  static gsize guard = 0;
   g_autoptr (GtkCssProvider) css_provider = NULL;
   GtkSettings *settings;
 
@@ -162,7 +162,7 @@
 static void
 hdy_icons_init (void)
 {
-  static volatile gsize guard = 0;
+  static gsize guard = 0;
 
   if (!g_once_init_enter (&guard))
     return;
