--- agent/Makefile.in.orig	2007-04-11 01:54:22.000000000 -0500
+++ agent/Makefile.in	2007-04-11 01:55:00.000000000 -0500
@@ -742,7 +742,7 @@
 # Don't make the daemon setuid
 # Make the daemon setuid
 @HAVE_USER_MLOCK_FALSE@install-exec-hook:
-@HAVE_USER_MLOCK_FALSE@	chmod u+s $(DESTDIR)$(bindir)/seahorse-daemon
+@HAVE_USER_MLOCK_FALSE@	chmod u+s $(DESTDIR)$(bindir)/seahorse-agent
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
