diff -Naur seahorse-2.20.0/configure.in seahorse-2.20.0.new/configure.in
--- seahorse-2.20.0/configure.in	2007-09-19 09:16:40.000000000 +0530
+++ seahorse-2.20.0.new/configure.in	2007-09-25 12:19:15.000000000 +0530
@@ -607,13 +607,14 @@
         # - Bonobo plugins <= 2.12. 
         # - Glib type plugins > 2.13
     
-	PKG_CHECK_MODULES(GEDIT, gedit-2.16 >= 2.15.0, [with_gedit_plugin=yes],
+	PKG_CHECK_MODULES(GEDIT, gedit-2.20 >= 2.20.0, [with_gedit_plugin=yes],
+		[PKG_CHECK_MODULES(GEDIT, gedit-2.16 >= 2.15.0, [with_gedit_plugin=yes],
             [PKG_CHECK_MODULES(GEDIT, gedit-2.14 >= 2.13.0, [with_gedit_plugin=yes],
                 [PKG_CHECK_MODULES(GEDIT, gedit-2.12 >= 2.12.0, [with_gedit_bonobo=yes],
                     [PKG_CHECK_MODULES(GEDIT, gedit-2.10 >= 2.9.0, [with_gedit_bonobo=yes], 
                         [PKG_CHECK_MODULES(GEDIT, gedit-2.8 >= 2.8.0, [with_gedit_bonobo=yes], 
                             [PKG_CHECK_MODULES(GEDIT, gedit-2.6 >= 2.6.0, [with_gedit_bonobo=yes],
-                                [enable_gedit=no])])])])])])
+                                [enable_gedit=no])])])])])])])
     
         if test "$enable_gedit" = "yes"; then
             echo "enabling gedit plugin..."
