Index: kvirc/src/kvirc/kernel/kvi_ircurl.cpp
===================================================================
--- kvirc/src/kvirc/kernel/kvi_ircurl.cpp (revision 246)
+++ kvirc/src/kvirc/kernel/kvi_ircurl.cpp (revision 630)
@@ -44,4 +44,6 @@
 	// irc[6]://<server>[:<port>][/<channel>[?<pass>]]
 	KviStr szUrl = url;
+	szUrl.replaceAll("$","\\$");
+	szUrl.replaceAll(";","\\;");
 	bool bIpV6 = false;
 	bool bSSL = false;
