--- dblatex-0.3.4.old/setup.py	2012-08-26 18:13:19.000000000 +0000
+++ dblatex-0.3.4.new/setup.py	2012-08-26 18:11:37.000000000 +0000
@@ -365,10 +365,7 @@
             raise OSError("not found: %s" % ", ".join(mis_stys))
 
     def run(self):
-        if self.install_layout == "deb":
-            db = DebianInstaller(self)
-        else:
-            db = None
+        db = None
 
         if not(db) and not(self.nodeps):
             try:
