--- gvfs-1.38.3/meson.build	2019-07-17 11:55:08.000000000 +0200
+++ gvfs-1.38.3.new/meson.build	2020-06-25 16:31:22.077255306 +0200
@@ -353,7 +353,7 @@
 enable_afc = get_option('afc')
 if enable_afc
   libimobiledevice_dep = dependency('libimobiledevice-1.0', version: '>= 1.2')
-  libplist_dep = dependency('libplist', version: '>= 0.15')
+  libplist_dep = dependency('libplist-2.0', version: '>= 0.15')
 endif
 
 # *** Check if we should build with GOA volume monitor ***
