diff --git a/src/utils/bae.h b/src/utils/bae.h
index cffeec1..094a112 100644
--- a/src/utils/bae.h
+++ b/src/utils/bae.h
@@ -320,7 +320,7 @@ namespace BAE
         //title.remove(QRegExp(QString::fromUtf8("[·-`~!@#$%^&*()_—+=|:;<>«»,.?/{}\'\"\\\[\\\]\\\\]")));
         QString title=str;
         title=title.remove(QChar::Null);
-        title=title.contains('\u0000')?title.replace('\u0000',""):title;
+        //title=title.contains('\u0000')?title.replace('\u0000',""):title;
         title=title.contains("(")&&title.contains(")")?fixTitle(title,"(",")"):title;
         title=title.contains("[")&&title.contains("]")?fixTitle(title,"[","]"):title;
         title=title.contains("{")&&title.contains("}")?fixTitle(title,"{","}"):title;
