diff -ur qtoctave-0.10.1.old/qtoctave/src/configure.h qtoctave-0.10.1/qtoctave/src/configure.h
--- qtoctave-0.10.1.old/qtoctave/src/configure.h	2011-01-06 22:48:06.000000000 +0100
+++ qtoctave-0.10.1/qtoctave/src/configure.h	2012-11-01 12:49:53.331074814 +0100
@@ -1,11 +1,11 @@
 
 #ifndef __CONFIGURE_H__
 #define __CONFIGURE_H__
-#define CONFIG_PATH "/opt/qtoctave/share/qtoctave"
-#define HELP_PATH "/opt/qtoctave/share/doc/octave-html/octave_doc/index.html"
-#define ICON_PATH "/opt/qtoctave/share/qtoctave/images"
-#define LANG_PATH "/opt/qtoctave/share/qtoctave/lang"
-#define QTOCTAVE_HELP_PATH "/opt/qtoctave/share/doc/qtoctave/qtoctave_doc/index.html"
+#define CONFIG_PATH "/usr/share/qtoctave"
+#define HELP_PATH "/usr/share/doc/octave-html/octave_doc/index.html"
+#define ICON_PATH "/usr/share/qtoctave/images"
+#define LANG_PATH "/usr/share/qtoctave/lang"
+#define QTOCTAVE_HELP_PATH "/usr/share/doc/qtoctave/qtoctave_doc/index.html"
 
-#define PKG_CMD_PATH "/opt/qtoctave/share/qtoctave/pkg-commands.list"
+#define PKG_CMD_PATH "/usr/share/qtoctave/pkg-commands.list"
 #endif
