--- portmidi/pm_java/pmdefaults/PmDefaultsFrame.java	2010-09-29 02:41:47.000000000 +0300
+++ portmidi.new/pm_java/pmdefaults/PmDefaultsFrame.java	2014-02-24 02:29:31.000000000 +0200
@@ -103,13 +103,13 @@
     }
 
     private JLabel inputLabel;
-    private JComboBox inputSelection;
+    private JComboBox<String> inputSelection;
     // inputIds maps from the index of an item in inputSelection to the
     // device ID. Used to open the selected device.
     private ArrayList<Integer> inputIds;
     private ActivityLight inputActivity;
     private JLabel outputLabel;
-    private JComboBox outputSelection;
+    private JComboBox<String> outputSelection;
     // analogous to inputIds, outputIds maps selection index to device ID
     private ArrayList<Integer> outputIds;
     private JButton testButton;
@@ -326,7 +326,7 @@
         inputLabel.setText("Default Input");
         wholePanel.add(inputLabel);
 
-        inputSelection = new JComboBox();
+        inputSelection = new JComboBox<String>();
         inputSelection.addActionListener(this);
         inputSelection.setLocation(20, 30);
         inputSelection.setSize(inputSelection.getPreferredSize());
@@ -341,7 +341,7 @@
         outputLabel.setText("Default Output");
         wholePanel.add(outputLabel);
 
-        outputSelection = new JComboBox();
+        outputSelection = new JComboBox<String>();
         outputSelection.addActionListener(this);
         wholePanel.add(outputSelection);
         testButton = new JButton();
