--- fltk-2.0.x-r6671.orig/src/Image.cxx	2009-04-10 16:07:34.000000000 +0200
+++ fltk-2.0.x-r6671/src/Image.cxx	2009-04-10 16:03:22.000000000 +0200
@@ -20,6 +20,7 @@
 // Please report all bugs and problems to "fltk-bugs@fltk.org".
 
 #include <config.h>
+#include <cstring>
 #include <fltk/Image.h>
 #include <fltk/Widget.h>
 #include <fltk/events.h>
--- fltk-2.0.x-r6671.orig/src/setcolor.cxx	2009-04-10 16:07:19.000000000 +0200
+++ fltk-2.0.x-r6671/src/setcolor.cxx	2009-04-10 16:08:27.000000000 +0200
@@ -24,6 +24,7 @@
 #include <fltk/Color.h>
 #include <fltk/draw.h>
 #include <config.h>
+#include <cstring>
 #if defined(_WIN32) && USE_STOCK_BRUSH && _WIN32_WINNT<0x0500
 # undef _WIN32_WINNT
 # define _WIN32_WINNT 0x0500

