diff -Naur kdeaddons-3.2.3.orig/noatun-plugins/synaescope/polygon.h kdeaddons-3.2.3/noatun-plugins/synaescope/polygon.h
--- kdeaddons-3.2.3.orig/noatun-plugins/synaescope/polygon.h	2001-04-17 01:20:34.000000000 +0200
+++ kdeaddons-3.2.3/noatun-plugins/synaescope/polygon.h	2004-06-11 10:26:13.000000000 +0200
@@ -25,6 +25,10 @@
 template<class Pixel, class Combiner, int superSampleShift>
 struct PolygonEngine : public Bitmap<Pixel>
 {
+    using Bitmap<Pixel>::width;
+    using Bitmap<Pixel>::height;
+    using Bitmap<Pixel>::data;
+
 	PolygonEngine() : Bitmap<Pixel>(1) { }
 
 #define super (1<<superSampleShift)
diff -Naur kdeaddons-3.2.3.orig/noatun-plugins/tippercanoe/polygon.h kdeaddons-3.2.3/noatun-plugins/tippercanoe/polygon.h
--- kdeaddons-3.2.3.orig/noatun-plugins/tippercanoe/polygon.h	2001-04-17 00:04:03.000000000 +0200
+++ kdeaddons-3.2.3/noatun-plugins/tippercanoe/polygon.h	2004-06-11 10:26:13.000000000 +0200
@@ -23,6 +23,10 @@
 
 template<class Pixel, class Combiner, int superSampleShift>
 struct PolygonEngine : public Bitmap<Pixel> {
+    using Bitmap<Pixel>::width;
+    using Bitmap<Pixel>::height;
+    using Bitmap<Pixel>::data;
+
   PolygonEngine() : Bitmap<Pixel>(1) { }
   
 #define super (1<<superSampleShift)
