diff -ur blender.orig/extern/bFTGL/SConscript blender/extern/bFTGL/SConscript
--- blender.orig/extern/bFTGL/SConscript	2005-05-06 15:37:18.000000000 +0200
+++ blender/extern/bFTGL/SConscript	2005-06-02 19:32:47.000000000 +0200
@@ -21,7 +21,7 @@
 	ftgl_env.Append (CCFLAGS = cflags)
 
 ftgl_env.Append (CPPDEFINES = defines)
-ftgl_env.Append (CPPPATH = ['include',
+ftgl_env.Append (CPPPATH = ['/usr/X11R6/include','include',
                              'src'])
 
 ftgl_env.Append (CPPPATH = user_options_dict['FREETYPE_INCLUDE'])
diff -ur blender.orig/SConstruct blender/SConstruct
--- blender.orig/SConstruct	2005-05-29 21:14:21.000000000 +0200
+++ blender/SConstruct	2005-06-02 19:19:04.000000000 +0200
@@ -65,10 +65,10 @@
 	build_blender_dynamic = 'true'
 	build_blender_static = 'false'
 	build_blender_player = 'true'
-	build_blender_plugin = 'false'
+	build_blender_plugin = 'true'
 	release_flags = ['-O2']
 	debug_flags = ['-O2', '-g']
-	extra_flags = ['-pipe', '-funsigned-char']
+	extra_flags = ['-pipe', '-funsigned-char','-march=i686']
 	cxxflags = []
 	defines = []
 	warn_flags = ['-Wall', '-W']
@@ -76,7 +76,7 @@
 	platform_libs = ['m', 'util', 'stdc++']
 	platform_libpath = []
 	platform_linkflags = ['-pthread']
-	extra_includes = []
+	extra_includes = ['/usr/X11R6/include']
 	# z library information
 	z_lib = ['z']
 	z_libpath = ['/usr/lib']
@@ -117,16 +117,16 @@
 	python_include = [sysconfig.get_python_inc ()]
 	python_linkflags = Split (sysconfig.get_config_var('LINKFORSHARED'))
 	# International support information
-	ftgl_lib = []
-	ftgl_libpath = []
-	ftgl_include = ['#extern/bFTGL/include']
+	ftgl_lib = ['ftgl']
+	ftgl_libpath = ['/usr/lib']
+	ftgl_include = ['/usr/include/FTGL']
 	freetype_env.ParseConfig ('pkg-config --cflags --libs freetype2 2>/dev/null || freetype-config --cflags --libs 2>/dev/null')
 	freetype_lib = freetype_env.Dictionary()['LIBS']
 	freetype_libpath = freetype_env.Dictionary()['LIBPATH']
 	freetype_include = freetype_env.Dictionary()['CPPPATH']
-	gettext_lib = []
-	gettext_libpath = []
-	gettext_include = []
+	gettext_lib = ['gettextpo']
+	gettext_libpath = ['/usr/lib']
+	gettext_include = ['/usr/include']
 	# OpenAL library information
 	openal_lib = ['openal']
 	openal_libpath = ['/usr/lib']
diff -ur blender.orig/source/gameengine/GamePlayer/common/SConscript blender/source/gameengine/GamePlayer/common/SConscript
--- blender.orig/source/gameengine/GamePlayer/common/SConscript	2005-03-14 15:56:40.000000000 +0100
+++ blender/source/gameengine/GamePlayer/common/SConscript	2005-06-02 19:13:24.000000000 +0200
@@ -17,7 +17,7 @@
 				'GPC_RenderTools.cpp',
 				'GPC_System.cpp']
 
-gp_common_env.Append( CPPPATH = ['.',
+gp_common_env.Append( CPPPATH = ['/usr/X11R6/include','.',
 								'#intern/string',
 								'#intern/ghost',
 								'#intern/guardedalloc',
diff -ur blender.orig/source/gameengine/GamePlayer/ghost/SConscript blender/source/gameengine/GamePlayer/ghost/SConscript
--- blender.orig/source/gameengine/GamePlayer/ghost/SConscript	2004-06-04 14:58:45.000000000 +0200
+++ blender/source/gameengine/GamePlayer/ghost/SConscript	2005-06-02 19:04:57.000000000 +0200
@@ -11,7 +11,7 @@
 				'GPG_KeyboardDevice.cpp',
 				'GPG_System.cpp']
 
-gp_ghost_env.Append( CPPPATH = ['.',
+gp_ghost_env.Append( CPPPATH = ['/usr/X11R6/include','.',
 								'#intern/string',
 								'#intern/ghost',
 								'#intern/guardedalloc',
