diff -Naur blender.orig/SConstruct blender/SConstruct
--- blender.orig/SConstruct	2006-02-10 22:09:47.327667500 +0100
+++ blender/SConstruct	2006-02-10 22:12:29.877826250 +0100
@@ -74,11 +74,11 @@
 	use_fluidsim = 'true'
 	build_blender_dynamic = 'true'
 	build_blender_static = 'false'
-	build_blender_player = 'false'
-	build_blender_plugin = 'false'
+	build_blender_player = 'true'
+	build_blender_plugin = 'true'
 	release_flags = ['-O2']
 	debug_flags = ['-O2', '-g']
-	extra_flags = ['-pipe', '-funsigned-char']
+	extra_flags = ['-pipe', '-funsigned-char', '-march=i686']
 	cxxflags = []
 	defines = []
 	warn_flags = ['-Wall', '-W']
@@ -135,16 +135,16 @@
 	python_include = [sysconfig.get_python_inc ()]
 	python_linkflags = Split (sysconfig.get_config_var('LINKFORSHARED'))
 	# International support information
-	ftgl_lib = []
-	ftgl_libpath = []
-	ftgl_include = ['#extern/bFTGL/include']
+	ftgl_lib = ['ftgl']
+	ftgl_libpath = ['/usr/lib']
+	ftgl_include = ['/usr/include/FTGL']
 	freetype_env.ParseConfig ('pkg-config --cflags --libs freetype2 2>/dev/null || freetype-config --cflags --libs 2>/dev/null')
 	freetype_lib = freetype_env.Dictionary()['LIBS']
 	freetype_libpath = freetype_env.Dictionary()['LIBPATH']
 	freetype_include = freetype_env.Dictionary()['CPPPATH']
-	gettext_lib = []
-	gettext_libpath = []
-	gettext_include = []
+	gettext_lib = ['gettextpo']
+	gettext_libpath = ['/usr/lib']
+	gettext_include = ['/usr/include']
 	# OpenAL library information
 	openal_lib = ['openal']
 	openal_libpath = ['/usr/lib']
