--- blender-2.35/SConstruct	2004-11-14 21:37:32.000000000 +0100
+++ SConstruct	2004-11-15 20:55:04.398879656 +0100
@@ -52,7 +52,7 @@
 env = Environment ()
 
 if sys.platform == 'linux2' or sys.platform == 'linux-i386':
-	use_international = 'false'
+	use_international = 'true'
 	use_gameengine = 'true'
 	use_openal = 'false'
 	use_fmod = 'false'
@@ -63,7 +63,7 @@
 	build_blender_dynamic = 'true'
 	build_blender_static = 'false'
 	build_blender_player = 'true'
-	build_blender_plugin = 'false'
+	build_blender_plugin = 'true'
 	release_flags = ['-O2']
 	debug_flags = ['-O2', '-g']
 	extra_flags = ['-pipe', '-funsigned-char']
@@ -116,15 +116,15 @@
 	python_linkflags = Split (sysconfig.get_config_var('LINKFORSHARED'))
 	# International support information
 	ftgl_lib = ['ftgl']
-	ftgl_libpath = ['#../lib/linux-glibc2.2.5-i386/ftgl/lib']
-	ftgl_include = ['#../lib/linux-glibc2.2.5-i386/ftgl/include']
+	ftgl_libpath = ['/usr/lib']
+	ftgl_include = ['/usr/include/FTGL']
 	freetype_env.ParseConfig ('pkg-config --cflags --libs freetype2 2>/dev/null || freetype-config --cflags --libs 2>/dev/null')
 	freetype_lib = freetype_env.Dictionary()['LIBS']
 	freetype_libpath = freetype_env.Dictionary()['LIBPATH']
 	freetype_include = freetype_env.Dictionary()['CPPPATH']
-	gettext_lib = []
-	gettext_libpath = []
-	gettext_include = []
+	gettext_lib = ['gettextpo']
+	gettext_libpath = ['/usr/lib']
+	gettext_include = ['/usr/include']
 	# OpenAL library information
 	openal_lib = ['openal']
 	openal_libpath = ['/usr/lib']
