diff -Naur mp3info-0.8.4-orig/gmp3info.c mp3info-0.8.4/gmp3info.c
--- mp3info-0.8.4-orig/gmp3info.c	2001-07-16 11:17:16.000000000 +0200
+++ mp3info-0.8.4/gmp3info.c	2005-08-29 17:38:46.000000000 +0200
@@ -484,12 +484,7 @@
 }
 
 void about_mp3info(void) {
-	quick_popup(VERSION,"MP3Info
-			An ID3 Tag Editor\n\n
-			Original author: Ricardo Cerqueira <rmc@plug.pt>
-			Current maintainer: Cedric Tefft <cedric@earthling.net>
-			GTK Version: Ricardo Cerqueira <rmc@plug.pt>\n\n
-			Ricardo Cerqueira, Cedric Tefft   1999-2001\n\n");
+	quick_popup(VERSION,"MP3Info\nAn ID3 Tag Editor\n\nOriginal author: Ricardo Cerqueira <rmc@plug.pt>\nCurrent maintainer: Cedric Tefft <cedric@earthling.net>\nGTK Version: Ricardo Cerqueira <rmc@plug.pt>\n\nRicardo Cerqueira, Cedric Tefft   1999-2001\n\n");
 }
 
 /* rmcc has left the building */
