diff -Naur fuseiso-20060107.orig/src/Makefile.am fuseiso-20060107/src/Makefile.am
--- fuseiso-20060107.orig/src/Makefile.am	2005-10-09 19:31:15.000000000 +0200
+++ fuseiso-20060107/src/Makefile.am	2006-04-06 13:07:15.000000000 +0200
@@ -11,5 +11,5 @@
 testiso_LDFLAGS = $(all_libraries)
 noinst_HEADERS = isofs.h
 
-AM_CFLAGS = -D_FILE_OFFSET_BITS=64 `pkg-config --cflags fuse glib-2.0`
+AM_CFLAGS = -I/usr/src/linux/include -D_FILE_OFFSET_BITS=64 `pkg-config --cflags fuse glib-2.0`
 AM_LDFLAGS = `pkg-config --libs fuse glib-2.0` -lz
