diff -Naur sooperlooper/src/gui/main_panel.cpp sooperlooper.new/src/gui/main_panel.cpp
--- sooperlooper/src/gui/main_panel.cpp	2014-02-19 09:46:30.981214121 +0100
+++ sooperlooper.new/src/gui/main_panel.cpp	2014-02-19 09:50:01.470218000 +0100
@@ -1276,7 +1276,7 @@
 		}
 
 
-		wxString filename = do_file_selector (wxT("Choose file to save loop"), wxT("wav"), wxT("WAVE files (*.wav)|*.wav;*.WAV;*.Wav"),  wxSAVE|wxCHANGE_DIR|wxOVERWRITE_PROMPT);
+		wxString filename = do_file_selector (wxT("Choose file to save loop"), wxT("wav"), wxT("WAVE files (*.wav)|*.wav;*.WAV;*.Wav"),  wxFD_SAVE|wxFD_CHANGE_DIR|wxFD_OVERWRITE_PROMPT);
 		
 		if ( !filename.empty() )
 		{
@@ -1295,7 +1295,7 @@
 			index = 0;
 		}
 
-		wxString filename = do_file_selector (wxT("Choose file to open"), wxT(""), wxT("*.slsess"), wxOPEN|wxCHANGE_DIR);
+		wxString filename = do_file_selector (wxT("Choose file to open"), wxT(""), wxT("*.slsess"), wxFD_OPEN|wxFD_CHANGE_DIR);
 		
 		if ( !filename.empty() )
 		{
@@ -1407,7 +1407,7 @@
 
 void MainPanel::do_load_session ()
 {
-	wxString filename = do_file_selector (wxT("Choose session to load"), wxT("*.slsess"), wxT("*.slsess"), wxOPEN|wxCHANGE_DIR);
+	wxString filename = do_file_selector (wxT("Choose session to load"), wxT("*.slsess"), wxT("*.slsess"), wxFD_OPEN|wxFD_CHANGE_DIR);
 	
 	if ( !filename.empty() )
 	{
@@ -1418,7 +1418,7 @@
 
 void MainPanel::do_save_session (bool write_audio)
 {
-	wxString filename = do_file_selector (wxT("Choose file to save session"), wxT("slsess"), wxT("*.slsess"), wxSAVE|wxCHANGE_DIR|wxOVERWRITE_PROMPT);
+	wxString filename = do_file_selector (wxT("Choose file to save session"), wxT("slsess"), wxT("*.slsess"), wxFD_SAVE|wxFD_CHANGE_DIR|wxFD_OVERWRITE_PROMPT);
 	
 	if ( !filename.empty() )
 	{
