diff -Naur glibc-2.3.2-orig/stdio-common/sscanf.c glibc-2.3.2/stdio-common/sscanf.c
--- glibc-2.3.2-orig/stdio-common/sscanf.c	2002-08-10 11:09:08.000000000 -0700
+++ glibc-2.3.2/stdio-common/sscanf.c	2003-05-17 22:56:01.000000000 -0700
@@ -27,9 +27,7 @@
 /* Read formatted input from S, according to the format string FORMAT.  */
 /* VARARGS2 */
 int
-sscanf (s, format)
-     const char *s;
-     const char *format;
+sscanf (const char *s, const char *format, ...)
 {
   va_list arg;
   int done;
