--- containerd-2.0.0/containerd.service	2024-11-05 19:34:06.000000000 +0100
+++ containerd-2.0.0.new/containerd.service	2024-11-12 21:53:22.822066615 +0100
@@ -18,8 +18,8 @@
 After=network.target local-fs.target dbus.service
 
 [Service]
-ExecStartPre=-/sbin/modprobe overlay
-ExecStart=/usr/local/bin/containerd
+ExecStartPre=-/usr/bin/modprobe overlay
+ExecStart=/usr/bin/containerd
 
 Type=notify
 Delegate=yes
