--- containerd/containerd.service	2024-01-15 13:50:13.956501423 +0100
+++ containerd.new/containerd.service	2024-01-15 13:50:44.257725358 +0100
@@ -20,8 +20,8 @@
 [Service]
 #uncomment to enable the experimental sbservice (sandboxed) version of containerd/cri integration
 #Environment="ENABLE_CRI_SANDBOXES=sandboxed"
-ExecStartPre=-/sbin/modprobe overlay
-ExecStart=/usr/local/bin/containerd
+ExecStartPre=-/usr/bin/modprobe overlay
+ExecStart=/usr/bin/containerd
 
 Type=notify
 Delegate=yes
