diff --git a/Makefile b/Makefile
index 9d8cf8a1..5723beb4 100644
--- a/Makefile
+++ b/Makefile
@@ -5,8 +5,8 @@ ROOTDIR=$(dir $(abspath $(lastword $(MAKEFILE_LIST))))
 DESTDIR=/usr/local
 
 # Used to populate variables in version package.
-VERSION=$(shell git describe --match 'v[0-9]*' --dirty='.m' --always)
-REVISION=$(shell git rev-parse HEAD)$(shell if ! git diff --no-ext-diff --quiet --exit-code; then echo .m; fi)
+VERSION=v1.0.3
+REVISION=773c489c9c1b21a6d78b5c538cd395416ec50f88
 
 ifneq "$(strip $(shell command -v go 2>/dev/null))" ""
 	GOOS ?= $(shell go env GOOS)
