diff -Naur aqemu-0.9.1/CMakeLists.txt aqemu-0.9.1-p/CMakeLists.txt
--- aqemu-0.9.1/CMakeLists.txt	2016-05-13 17:48:02.000000000 +0200
+++ aqemu-0.9.1-p/CMakeLists.txt	2016-05-15 05:00:16.369139550 +0200
@@ -285,7 +285,7 @@
 		
 		SET( _out ${CMAKE_CURRENT_SOURCE_DIR}/${_basename}.rcc )
 		
-		ADD_CUSTOM_COMMAND( OUTPUT ${_out} COMMAND rcc ARGS ${_in} -binary -o ${_out} DEPENDS ${_in} )
+		ADD_CUSTOM_COMMAND( OUTPUT ${_out} COMMAND rcc-qt5 ARGS ${_in} -binary -o ${_out} DEPENDS ${_in} )
 		
 		SET( ${_sources} ${${_sources}} ${_out} )
 	ENDFOREACH( curFile )

 
