--- Embedded_Display/vncview.cpp	2010-02-26 13:40:40.000000000 +0000
+++ Embedded_Display_New/vncview.cpp	2011-12-25 19:05:07.000000000 +0000
@@ -27,8 +27,6 @@
     #include <QMessageBox>
     #include <QInputDialog>
     #define KMessageBox QMessageBox
-    #define error(parent, message, caption) \
-        critical(parent, caption, message)
 #else
     #include "settings.h"
     #include <KActionCollection>
