--- gmic-2.8.4/src/Makefile	2020-02-10 16:35:00.000000000 +0100
+++ gmic-2.8.4.new/src/Makefile	2020-03-23 13:04:16.530736553 +0100
@@ -289,8 +289,8 @@
 
 # Enable native support of webcams and video streaming, using the OpenCV library.
 # (https://opencv.org/)
-OPENCV_CFLAGS = -Dcimg_use_opencv $(shell pkg-config opencv --cflags || echo -I/usr/include/opencv) -I/usr/include/opencv
-OPENCV_LIBS = $(shell pkg-config opencv --libs || echo -lopencv_core -lopencv_highgui)
+OPENCV_CFLAGS = -Dcimg_use_opencv $(shell pkg-config opencv4 --cflags || echo -I/usr/include/opencv) -I/usr/include/opencv
+OPENCV_LIBS = $(shell  echo -lopencv_core -lopencv_videoio)
 
 # Enable support of most classical image file formats, using the GraphicsMagick++ library.
 # (http://www.graphicsmagick.org/Magick++/)
@@ -736,9 +736,9 @@
 	    mkdir -p $(DESTDIR)/usr/share/bash-completion/completions/; \
             cp -f ../resources/gmic_bashcompletion.sh $(DESTDIR)/usr/share/bash-completion/completions/gmic; \
 	  fi; \
-	  if [ -d /opt/local/etc/bash_completion.d/ ]; then \
-            mkdir -p $(DESTDIR)/opt/local/etc/bash_completion.d/; \
-            cp -f ../resources/gmic_bashcompletion.sh $(DESTDIR)/opt/local/etc/bash_completion.d/gmic; \
+	  if [ -d /opt/local/usr/share/bash-completion/completions/ ]; then \
+            mkdir -p $(DESTDIR)/opt/local/usr/share/bash-completion/completions/; \
+            cp -f ../resources/gmic_bashcompletion.sh $(DESTDIR)/opt/local/usr/share/bash-completion/completions/gmic; \
           fi; \
         fi
 
