diff -Naur gmic-2.8.4/zart/src/ImageConverter.cpp gmic-2.8.4.new/zart/src/ImageConverter.cpp
--- gmic-2.8.4/zart/src/ImageConverter.cpp	2020-02-10 16:34:58.000000000 +0100
+++ gmic-2.8.4.new/zart/src/ImageConverter.cpp	2020-03-18 15:37:27.654782860 +0100
@@ -70,7 +70,7 @@
 
   cv::Mat tmp(in->cols, in->rows, in->depth());
 
-  cvtColor(*in, tmp, (in->channels() == 1) ? CV_GRAY2RGB : CV_BGR2RGB);
+  cvtColor(*in, tmp, (in->channels() == 1) ? cv::COLOR_GRAY2RGB : cv::COLOR_BGR2RGB);
 
   const unsigned int w3 = 3 * tmp.cols;
   unsigned char * src = reinterpret_cast<unsigned char *>(tmp.ptr());
@@ -114,7 +114,7 @@
       dst += step;
     }
   }
-  cvtColor(**out, **out, CV_BGR2RGB);
+  cvtColor(**out, **out, cv::COLOR_BGR2RGB);
 }
 
 void ImageConverter::convert(const cimg_library::CImg<float> & in, QImage * out)
