tlb_flush_page is no longer exported on i686, either.

--- common/lib/modules/fglrx/build_mod/firegl_public.c.orig	2009-06-13 23:34:05.000000000 +0200
+++ common/lib/modules/fglrx/build_mod/firegl_public.c	2009-06-13 23:34:51.000000000 +0200
@@ -2308,7 +2308,7 @@
  * kernel <  2.6.27, on_each_cpu has 4 parameters.
  * kernel >= 2.6.27, on_each_cpu has 3 parameters (removed the "retry" parameter)
  */
-#if defined(__x86_64__) && (defined(__SMP__) || defined(CONFIG_SMP)) && (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,25)) 
+#if (defined(__SMP__) || defined(CONFIG_SMP)) && (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,25)) 
 #   if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,27))
         on_each_cpu(KCL_flush_tlb_one, &va, 1, 1);
 #   else
