diff -Naur fglrx/common/lib/modules/fglrx/build_mod/drm_compat.h fw_fglrx/common/lib/modules/fglrx/build_mod/drm_compat.h
--- common/lib/modules/fglrx/build_mod/drm_compat.h	2008-04-10 20:20:59.000000000 +0200
+++ common/lib/modules/fglrx/build_mod/drm_compat.h	2008-04-24 17:04:17.000000000 +0200
@@ -43,7 +43,7 @@
 #endif
 
 #ifndef MODULE_LICENSE
-#define MODULE_LICENSE(x) 
+#define MODULE_LICENSE("GPL") 
 #endif
 
 #ifndef preempt_disable
diff -Naur fglrx/common/lib/modules/fglrx/build_mod/firegl_public.c fw_fglrx/common/lib/modules/fglrx/build_mod/firegl_public.c
--- common/lib/modules/fglrx/build_mod/firegl_public.c	2008-04-10 20:20:59.000000000 +0200
+++ common/lib/modules/fglrx/build_mod/firegl_public.c	2008-04-24 17:04:56.000000000 +0200
@@ -233,7 +233,7 @@
 module_param(firegl, charp, 0);
 #endif
 #ifdef MODULE_LICENSE
-MODULE_LICENSE("Proprietary. (C) 2002 - ATI Technologies, Starnberg, GERMANY");
+MODULE_LICENSE("GPL");
 #endif
 #ifdef MODULE_DEVICE_TABLE
 MODULE_DEVICE_TABLE(pci, fglrx_pci_table);
