--- Makefile	2009-04-02 14:08:23.000000000 +0200
+++ Makefile	2010-04-14 22:31:16.283306486 +0200
@@ -89,11 +89,11 @@
 	-ccopt "-L." bigarray.cma gmp.cma 
 
 install:
-	mkdir -p $(PREFIX)/include $(PREFIX)/lib $(PREFIX)/bin
-	cp -f $(MLLIB_TOINSTALL) $(MLLIB_TOINSTALLx) $(PREFIX)/lib
-	cp -f $(CCINC_TOINSTALL) $(PREFIX)/include
-	for i in $(CCLIB_TOINSTALL); do if test -f $$i; then cp -f $$i $(PREFIX)/lib; fi; done
-	for i in $(CCBIN_TOINSTALL); do if test -f $$i; then cp -f $$i $(PREFIX)/bin; fi; done
+	mkdir -p $(DESTDIR)$(PREFIX)/include $(DESTDIR)$(PREFIX)/lib $(DESTDIR)$(PREFIX)/bin
+	install -m 644 $(MLLIB_TOINSTALL) $(MLLIB_TOINSTALLx) $(DESTDIR)$(PREFIX)/lib
+	install -m 644 $(CCINC_TOINSTALL) $(DESTDIR)$(PREFIX)/include
+	for i in $(CCLIB_TOINSTALL); do if test -f $$i; then install -m 644 $$i $(DESTDIR)$(PREFIX)/lib; fi; done
+	for i in $(CCBIN_TOINSTALL); do if test -f $$i; then install -m 644 $$i $(DESTDIR)$(PREFIX)/bin; fi; done
 
 #---------------------------------------
 # Misc rules
