diff -Naur mlgmpidl//gmp_caml.h mlgmpidl.new//gmp_caml.h
--- mlgmpidl//gmp_caml.h	2007-07-11 15:29:31.000000000 +0200
+++ mlgmpidl.new//gmp_caml.h	2011-02-10 17:00:47.302000032 +0100
@@ -46,10 +46,18 @@
 void camlidl_mpfr_ml2c(value val, __mpfr_struct* mpf);
 static inline
 value camlidl_mpfr_rnd_t_c2ml(mpfr_rnd_t* rnd)
-{ assert(*rnd>=0 && *rnd<GMP_RND_MAX); return Val_int(*rnd); }
+{ assert(*rnd>=0
+#ifdef GMP_RND_MAX
+&& *rnd<GMP_RND_MAX
+#endif
+); return Val_int(*rnd); }
 static inline
 void camlidl_mpfr_rnd_t_ml2c(value val, mpfr_rnd_t* rnd)
-{ *rnd = Int_val(val); assert(*rnd>=0 && *rnd<GMP_RND_MAX); }
+{ *rnd = Int_val(val); assert(*rnd>=0
+#ifdef GMP_RND_MAX
+&& *rnd<GMP_RND_MAX
+#endif
+); }
 #endif
 #endif
 
diff -Naur mlgmpidl//gmp_random_caml.c mlgmpidl.new//gmp_random_caml.c
--- mlgmpidl//gmp_random_caml.c	2009-04-02 16:25:47.000000000 +0200
+++ mlgmpidl.new//gmp_random_caml.c	2011-02-10 17:02:22.706000056 +0100
@@ -235,6 +235,7 @@
   return Val_unit;
 }
 
+#if 0
 value camlidl_gmp_random_mpfr_random(
 	value _v_ROP)
 {
@@ -246,4 +247,5 @@
   camlidl_free(_ctx);
   return Val_unit;
 }
+#endif
 
diff -Naur mlgmpidl//gmp_random.ml mlgmpidl.new//gmp_random.ml
--- mlgmpidl//gmp_random.ml	2009-04-02 16:25:47.000000000 +0200
+++ mlgmpidl.new//gmp_random.ml	2011-02-10 17:03:25.057000034 +0100
@@ -61,9 +61,9 @@
 module Mpfr = struct
 external urandomb : Mpfr.t -> state -> unit
 	= "camlidl_gmp_random_mpfr_urandomb"
-
+(*
 external random : Mpfr.t -> unit
 	= "camlidl_gmp_random_mpfr_random"
-
+*)
 end
 
diff -Naur mlgmpidl//gmp_random.mli mlgmpidl.new//gmp_random.mli
--- mlgmpidl//gmp_random.mli	2009-04-02 16:25:47.000000000 +0200
+++ mlgmpidl.new//gmp_random.mli	2011-02-10 17:03:54.119000033 +0100
@@ -61,9 +61,9 @@
 module Mpfr : sig
 external urandomb : Mpfr.t -> state -> unit
 	= "camlidl_gmp_random_mpfr_urandomb"
-
+(*
 external random : Mpfr.t -> unit
 	= "camlidl_gmp_random_mpfr_random"
-
+*)
 end
 
