diff -ur a/config.mk b/config.mk
--- a/config.mk	2013-03-30 09:03:46.000000000 -0500
+++ b/config.mk	2014-06-24 13:20:33.389881147 -0500
@@ -105,7 +105,7 @@
 $(error Cannot find $(GNUTLS))
 endif
 
-ifeq ($(shell pkg-config --exists json && echo 1), 1)
+ifeq ($(shell pkg-config --exists json-c && echo 1), 1)
 LIBS+=$(GNUTLS)
 else
 $(error Cannot find json-c)
diff -ur a/src/util/Makefile b/src/util/Makefile
--- a/src/util/Makefile	2013-03-30 09:03:46.000000000 -0500
+++ b/src/util/Makefile	2014-06-24 13:24:21.082872767 -0500
@@ -13,7 +13,7 @@
 TRANSPORT_SECURITY_JSON=transport_security_state_static.json
 TRANSPORT_SECURITY_CERTS=transport_security_state_static.certs
 
-LIBS=glib-2.0 gnutls json
+LIBS=glib-2.0 gnutls json-c
 CFLAGS+=$(shell pkg-config --cflags $(LIBS))
 LDFLAGS=$(shell pkg-config --libs $(LIBS))
 
