--- regex-tdfa-1.3.2.2/regex-tdfa.cabal	2001-09-09 03:46:40.000000000 +0200
+++ regex-tdfa-1.3.2.2.new/regex-tdfa.cabal	2024-07-10 22:03:39.239767275 +0200
@@ -1,6 +1,7 @@
 cabal-version:          1.12
 name:                   regex-tdfa
 version:                1.3.2.2
+x-revision:             3
 
 build-Type:             Simple
 license:                BSD3
@@ -25,8 +26,10 @@
   test/cases/*.txt
 
 tested-with:
-  GHC == 9.6.2
-  GHC == 9.4.5
+  GHC == 9.10.0
+  GHC == 9.8.2
+  GHC == 9.6.4
+  GHC == 9.4.8
   GHC == 9.2.8
   GHC == 9.0.2
   GHC == 8.10.7
@@ -35,10 +38,6 @@
   GHC == 8.4.4
   GHC == 8.2.2
   GHC == 8.0.2
-  GHC == 7.10.3
-  GHC == 7.8.4
-  GHC == 7.6.3
-  GHC == 7.4.2
 
 source-repository head
   type:                git
@@ -47,7 +46,7 @@
 source-repository this
   type:                git
   location:            https://github.com/haskell-hvr/regex-tdfa.git
-  tag:                 v1.3.2.2
+  tag:                 v1.3.2.2-r2
 
 flag force-O2
   default: False
@@ -107,11 +106,11 @@
   build-depends:        array              >= 0.4    && < 0.6
                       , base               >= 4.5    && < 5
                       , bytestring         >= 0.9.2  && < 0.13
-                      , containers         >= 0.4.2  && < 0.7
+                      , containers         >= 0.4.2  && < 0.8
                       , mtl                >= 2.1.3  && < 2.4
                       , parsec             == 3.1.*
                       , regex-base         == 0.94.*
-                      , text               >= 1.2.3  && < 2.1
+                      , text               >= 1.2.3  && < 2.2
 
   default-language:     Haskell2010
   default-extensions:   BangPatterns
@@ -164,7 +163,7 @@
 
   -- component-specific dependencies not inherited via 'regex-tdfa'
                       , directory          >= 1.1.0  && < 1.4
-                      , filepath           >= 1.3.0  && < 1.5
+                      , filepath           >= 1.3.0  && < 1.6
                       , utf8-string        >= 1.0.1  && < 1.1
 
   default-language:     Haskell2010
