diff -Naur ntfs-3g-0.20070207-RC1.orig/src/Makefile.am ntfs-3g-0.20070207-RC1/src/Makefile.am
--- ntfs-3g-0.20070207-RC1.orig/src/Makefile.am	2007-02-15 11:53:38.000000000 +0100
+++ ntfs-3g-0.20070207-RC1/src/Makefile.am	2007-02-15 11:53:43.000000000 +0100
@@ -38,7 +38,6 @@
 install-exec-hook:
 	$(INSTALL) -d $(DESTDIR)/sbin
 	$(LN_S) -f $(bindir)/ntfs-3g $(DESTDIR)/sbin/mount.ntfs-3g
-	$(LDCONFIG)
 
 install-data-hook:
 	$(INSTALL) -d $(DESTDIR)$(man8dir)
diff -Naur ntfs-3g-0.20070207-RC1.orig/src/Makefile.in ntfs-3g-0.20070207-RC1/src/Makefile.in
--- ntfs-3g-0.20070207-RC1.orig/src/Makefile.in	2007-02-15 11:53:38.000000000 +0100
+++ ntfs-3g-0.20070207-RC1/src/Makefile.in	2007-02-15 11:53:43.000000000 +0100
@@ -582,7 +582,6 @@
 install-exec-hook:
 	$(INSTALL) -d $(DESTDIR)/sbin
 	$(LN_S) -f $(bindir)/ntfs-3g $(DESTDIR)/sbin/mount.ntfs-3g
-	$(LDCONFIG)
 
 install-data-hook:
 	$(INSTALL) -d $(DESTDIR)$(man8dir)
