diff -Naur ntfs-3g-20070811-BETA/src/Makefile.am ntfs-3g-20070811-BETA-p/src/Makefile.am
--- ntfs-3g-20070811-BETA/src/Makefile.am	2006-08-02 00:45:07.000000000 +0200
+++ ntfs-3g-20070811-BETA-p/src/Makefile.am	2006-08-13 23:52:35.000000000 +0200
@@ -38,7 +38,6 @@
 install-exec-hook:
 	$(INSTALL) -d $(DESTDIR)/sbin
 	$(LN_S) -f $(bindir)/ntfs-3g $(DESTDIR)/sbin/mount.ntfs-3g
-	$(LDCONFIG)
 
 install-data-hook:
 	$(INSTALL) -d $(DESTDIR)$(man8dir)
diff -Naur ntfs-3g-20070811-BETA/src/Makefile.in ntfs-3g-20070811-BETA-p/src/Makefile.in
--- ntfs-3g-20070811-BETA/src/Makefile.in	2006-08-11 02:10:36.000000000 +0200
+++ ntfs-3g-20070811-BETA-p/src/Makefile.in	2006-08-13 23:53:55.000000000 +0200
@@ -579,7 +579,6 @@
 install-exec-hook:
 	$(INSTALL) -d $(DESTDIR)/sbin
 	$(LN_S) -f $(bindir)/ntfs-3g $(DESTDIR)/sbin/mount.ntfs-3g
-	$(LDCONFIG)
 
 install-data-hook:
 	$(INSTALL) -d $(DESTDIR)$(man8dir)
