diff -Naur libjpeg-turbo-2.0.0/CMakeLists.txt libjpeg-turbo-2.0.0-p/CMakeLists.txt
--- libjpeg-turbo-2.0.0/CMakeLists.txt	2018-07-27 18:47:48.000000000 +0200
+++ libjpeg-turbo-2.0.0-p/CMakeLists.txt	2018-11-04 13:51:17.488465423 +0100
@@ -107,7 +107,7 @@
   endif()
 endif()
 
-include(cmakescripts/GNUInstallDirs.cmake)
+include(GNUInstallDirs)
 
 set(CMAKE_INSTALL_RPATH ${CMAKE_INSTALL_FULL_LIBDIR})
 
