diff --git a/host/BorrowedImage.h b/host/BorrowedImage.h
index ed2890568..2bf2293d4 100644
--- a/host/BorrowedImage.h
+++ b/host/BorrowedImage.h
@@ -12,6 +12,8 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <cstdint>
+
 #pragma once
 
 namespace gfxstream {
diff --git a/host/compressedTextureFormats/AstcCpuDecompressor.h b/host/compressedTextureFormats/AstcCpuDecompressor.h
index 4823ee255..6d7843cd2 100644
--- a/host/compressedTextureFormats/AstcCpuDecompressor.h
+++ b/host/compressedTextureFormats/AstcCpuDecompressor.h
@@ -13,6 +13,7 @@
 // limitations under the License.
 #pragma once
 
+#include <cstdint>
 #include <memory>
 #include <string>
 
diff --git a/host/vulkan/BufferVk.h b/host/vulkan/BufferVk.h
index 9016fd745..96f5c2f55 100644
--- a/host/vulkan/BufferVk.h
+++ b/host/vulkan/BufferVk.h
@@ -12,6 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <cstdint>
 #include <memory>
 #include <vector>
 
