diff -Naur d4x-2.5.7.1/configure.ac d4x-2.5.7.1-gtk/configure.ac
--- d4x-2.5.7.1/configure.ac	2005-11-05 00:35:38.000000000 +0100
+++ d4x-2.5.7.1-gtk/configure.ac	2007-12-08 21:31:53.000000000 +0100
@@ -298,7 +298,6 @@
 
 # Other needed defines
 AC_DEFINE(_REENTRANT, , [Enable reenrant code of glibc])
-AC_DEFINE(GTK_DISABLE_DEPRECATED, , [Enable warnings about deprecated functions usage])
 AC_DEFINE(_FILE_OFFSET_BITS, 64, [To support files over 2GB])
 
 # Prepare all requiered flags
diff -Naur d4x-2.5.7.1/main/face/list.cc d4x-2.5.7.1-gtk/main/face/list.cc
--- d4x-2.5.7.1/main/face/list.cc	2006-03-20 01:29:41.000000000 +0100
+++ d4x-2.5.7.1-gtk/main/face/list.cc	2007-12-08 21:31:04.000000000 +0100
@@ -8,7 +8,6 @@
  *	but WITHOUT ANY WARRANTY; without even the implied warranty of
  *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
  */
-#define GTK_DISABLE_DEPRECATED
 
 #include <stdio.h>
 #include <gtk/gtk.h>
