diff -Naur liblastfm/admin/Makefile.rb liblastfm.new/admin/Makefile.rb
--- liblastfm/admin/Makefile.rb	2012-01-03 00:21:50.000000000 +0100
+++ liblastfm.new/admin/Makefile.rb	2012-01-03 00:23:01.000000000 +0100
@@ -6,7 +6,7 @@
 
 cwd = File.dirname( __FILE__ )
 require 'find'
-require "#{cwd}/platform.rb"
+require_relative "platform.rb"
 
 
 ######################################################################### defs
diff -Naur liblastfm/admin/qpp liblastfm.new/admin/qpp
--- liblastfm/admin/qpp	2012-01-03 00:21:50.000000000 +0100
+++ liblastfm.new/admin/qpp	2012-01-03 00:23:34.000000000 +0100
@@ -4,7 +4,7 @@
 
 cwd=File.dirname __FILE__
 require 'find'
-require "#{cwd}/findsrc"
+require_relative "findsrc"
 
 sources = Array.new
 headers = Array.new
