--- src/glade_support.c	2007-03-16 15:25:49.000000000 +0100
+++ src/glade_support.c	2009-01-24 16:57:15.000000000 +0100
@@ -37,10 +37,10 @@
  *                                                                       *
  *                                                                       *
  *************************************************************************/
-typedef struct signal {
+typedef struct signal_t {
 	gchar     *name;
 	GCallback callback;
-} signal;
+} signal_t;
 
 /*
 ** Signal handler callbascks.
@@ -206,7 +206,7 @@
 static gboolean
 find_and_connect_handler(
 		GtkWidget *widget,
-		signal *signals,
+		signal_t *signals,
 		const gchar *signal_name,
 		const gchar *handler_name)
 {
@@ -288,7 +288,7 @@
 		gpointer user_data)
 {
 	gint n;
-	signal entry_signals[] = {
+	signal_t entry_signals[] = {
 		{ "activate",           (GCallback)on_any_entry_almost_any },
 		{ "backspace",          (GCallback)on_any_entry_almost_any },
 		{ "copy-clipboard",     (GCallback)on_any_entry_almost_any },
@@ -319,7 +319,7 @@
 		gpointer user_data)
 {
 	variable *var;
-	signal widget_signals[] = {
+	signal_t widget_signals[] = {
 		{ "accel-closures-changed",    (GCallback)on_any_widget_almost_any },
 		{ "composited-changed",        (GCallback)on_any_widget_almost_any },
 		{ "grab-focus",                (GCallback)on_any_widget_almost_any },
