diff -Naur mc-4.6.0.orig/lib/mc.ext.in mc-4.6.0/lib/mc.ext.in
--- mc-4.6.0.orig/lib/mc.ext.in	2003-01-30 16:48:33.000000000 +0100
+++ mc-4.6.0/lib/mc.ext.in	2004-08-08 00:18:54.000000000 +0200
@@ -379,8 +379,8 @@
 
 ### Archives ###
 
-# .tgz, .tpz, .tar.gz, .tar.z, .tar.Z
-regex/\.t([gp]?z|ar\.g?[zZ])$
+# .tgz, .tpz, .tar.gz, .tar.z, .tar.Z, .fpm, .fdb
+regex/\.(t([gp]?z|ar\.g?[zZ])|f(pm|db))$
 	Open=%cd %p#utar
 	View=%view{ascii} gzip -dc %f 2>/dev/null | tar tvvf -
 
