diff -Naur mc-4.6.1-pre4a.old/src/util.c mc-4.6.1-pre4a/src/util.c
--- mc-4.6.1-pre4a.old/src/util.c	2005-03-21 12:32:16.000000000 +0100
+++ mc-4.6.1-pre4a/src/util.c	2005-03-21 12:32:26.000000000 +0100
@@ -1364,7 +1364,7 @@
 
 	*line = strtol(p, const_cast(char **, &p), 10);
 	if (*p == ';') {
-	    *column = strtol(p, const_cast(char **, &p), 10);
+	    *column = strtol(p+1, const_cast(char **, &p), 10);
 	    if (*p != '\n')
 	        *column = 0;
 	} else
