diff -Naur mc-4.6.1-pre4a.old/src/main.c mc-4.6.1-pre4a/src/main.c
--- mc-4.6.1-pre4a.old/src/main.c	2005-03-21 16:30:59.000000000 +0100
+++ mc-4.6.1-pre4a/src/main.c	2005-03-21 16:33:19.000000000 +0100
@@ -1606,7 +1606,7 @@
 void
 update_xterm_title_path (void)
 {
-    unsigned char *p, *s;
+    unsigned char *p, *s, *tmp;
 
     if (xterm_flag && xterm_title) {
 	p = s = g_strdup (strip_home_and_password (current_panel->cwd));
@@ -1614,8 +1614,15 @@
 	    if (!is_printable (*s))
 		*s = '?';
 	} while (*++s);
+#ifdef HAVE_SLANG
+	SLtt_set_alt_char_set (0);
+	tmp = g_strdup_printf("\33]0;mc - %s\7", p);
+	SLtt_write_string(tmp);
+	g_free(tmp);
+#else
 	fprintf (stdout, "\33]0;mc - %s\7", p);
 	fflush (stdout);
+#endif
 	g_free (p);
     }
 }
