diff -Naur mc-4.6.1.orig/syntax/sh.syntax mc-4.6.1/syntax/sh.syntax
--- mc-4.6.1.orig/syntax/sh.syntax	2004-12-02 11:47:14.000000000 +0100
+++ mc-4.6.1/syntax/sh.syntax	2005-11-26 23:29:33.000000000 +0100
@@ -13,7 +13,22 @@
     keyword ${*} brightgreen
     keyword { brightcyan
     keyword } brightcyan
-
+    keyword ( yellow
+    keyword ) yellow
+    keyword || yellow
+    keyword = magenta
+    keyword \\ yellow
+    keyword 0 brightred
+    keyword 1 brightred
+    keyword 2 brightred
+    keyword 3 brightred
+    keyword 4 brightred
+    keyword 5 brightred
+    keyword 6 brightred
+    keyword 7 brightred
+    keyword 8 brightred
+    keyword 9 brightred
+    
     keyword whole linestart #!\[\s\]*\n brightcyan black
 
     keyword $\* brightred
@@ -28,6 +43,7 @@
     keyword wholeright $\[0123456789\] brightred
 
     keyword wholeright $+ brightgreen
+    keyword wholeright --+ brightmagenta
 
     keyword $ brightgreen
 
@@ -68,6 +84,66 @@
     keyword whole until yellow
     keyword whole wait yellow
     keyword whole while yellow
+    keyword whole Fmessage yellow
+    keyword whole Fdie yellow
+    keyword whole Fcd yellow
+    keyword whole Fmkdir yellow
+    keyword whole Frm yellow
+    keyword whole Fcp yellow
+    keyword whole Fcpr yellow
+    keyword whole Fmv yellow
+    keyword whole Finstallrel yellow
+    keyword whole Finstall yellow
+    keyword whole Fdirschmod yellow
+    keyword whole Ffileschmod yellow
+    keyword whole Fdirschown yellow
+    keyword whole Ffileschown yellow
+    keyword whole Fexe yellow
+    keyword whole Fexerel yellow
+    keyword whole Ffile yellow
+    keyword whole Ffilerel yellow
+    keyword whole Fdoc yellow
+    keyword whole Fdocrel yellow
+    keyword whole Fln yellow
+    keyword whole Fsed yellow
+    keyword whole Fdeststrip yellow
+    keyword whole Fpatch yellow
+    keyword whole Fpatchall yellow
+    keyword whole Fconf yellow
+    keyword whole Fmake yellow
+    keyword whole Fmakeinstall yellow
+    keyword whole Fbuild yellow
+    keyword whole Frcd yellow
+    keyword whole Frcd2 yellow
+    keyword whole build  yellow
+    keyword whole Facu yellow
+    keyword whole Fsort yellow
+    keyword whole Funpack_makeself yellow
+    keyword whole Fautoconfize yellow
+    keyword whole Finclude yellow
+    keyword whole Flasttar yellow
+    keyword whole Fdesktop yellow
+    keyword whole pkgrel brightcyan
+    keyword whole pkgname brightcyan
+    keyword whole pkgver brightcyan
+    keyword whole pkgdesc brightcyan
+    keyword whole url brightcyan
+    keyword whole backup brightcyan
+    keyword whole groups brightcyan
+    keyword whole makedepends brightcyan
+    keyword whole depends brightcyan
+    keyword whole archs brightcyan
+    keyword whole up2date brightcyan
+    keyword whole source brightcyan
+    keyword whole sha1sums brightcyan
+    keyword whole force brightcyan
+    keyword whole nobuild brightcyan
+    keyword whole license brightcyan
+    keyword whole install brightcyan
+    keyword whole rodepends brightcyan
+    keyword whole conflicts brightcyan
+    keyword whole provides brightcyan
+    keyword whole replaces brightcyan
 
     keyword whole arch cyan
     keyword whole ash cyan
@@ -204,7 +280,7 @@
     keyword ${*} brightgreen
     keyword { brightcyan
     keyword } brightcyan
-
+    
     keyword $\* brightred
     keyword $@ brightred
     keyword $# brightred
diff -Naur mc-4.6.1.orig/syntax/Syntax mc-4.6.1/syntax/Syntax
--- mc-4.6.1.orig/syntax/Syntax	2005-05-26 09:21:34.000000000 +0200
+++ mc-4.6.1/syntax/Syntax	2005-11-26 23:49:20.000000000 +0100
@@ -144,5 +144,8 @@
 file Don_t_match_me Mail\sfolder ^From\s
 include mail.syntax
 
+file (FrugalBuild|..\*\\.install)$ FrugalBuild\sFPM
+include sh.syntax
+
 file .\* unknown
 include unknown.syntax
