--- xchm-1.23/src/chmframe.cpp~	2013-08-30 10:31:38.000000000 +0200
+++ xchm-1.23/src/chmframe.cpp	2014-01-18 16:36:29.792122290 +0100
@@ -265,7 +265,7 @@
 		enu.EnumerateFacenames();
 		_normalFonts = new wxArrayString;

-#if wxMAJOR_VERSION == 2 && wxMINOR_VERSION >= 7
+#if (wxMAJOR_VERSION == 2 && wxMINOR_VERSION >= 7) || (wxMAJOR_VERSION == 3)
 		*_normalFonts = enu.GetFacenames();
 #else
 		*_normalFonts = *enu.GetFacenames();
@@ -278,7 +278,7 @@
 		enu.EnumerateFacenames(wxFONTENCODING_SYSTEM, TRUE);
 		_fixedFonts = new wxArrayString;

-#if wxMAJOR_VERSION == 2 && wxMINOR_VERSION >= 7
+#if (wxMAJOR_VERSION == 2 && wxMINOR_VERSION >= 7) || (wxMAJOR_VERSION == 3)
 		*_fixedFonts = enu.GetFacenames();
 #else
 		*_fixedFonts = *enu.GetFacenames();
