diff -Naur monodoc-1.0.5.orig/monodoc.desktop monodoc-1.0.5/monodoc.desktop
--- monodoc-1.0.5.orig/monodoc.desktop	2004-11-13 20:56:52.000000000 +0100
+++ monodoc-1.0.5/monodoc.desktop	2005-02-11 11:36:23.100101760 +0100
@@ -1,7 +1,9 @@
 [Desktop Entry]
 Name=Monodoc
 GenericName=Documentation Browser
+GenericName[hu]=Dokumentációböngésző
 Comment=Monodoc Documentation Browser
+Comment[hu]=Monodoc dokumentációböngésző
 Exec=monodoc
 Icon=monodoc.png
 StartupNotify=false
