From 8896aff9728b7ac89de66f6bdc53cce256419d1b Mon Sep 17 00:00:00 2001
From: Jan Ciesko <jan.ciesko@gmail.com>
Date: Thu, 5 Jan 2023 12:31:39 -0800
Subject: [PATCH] UCT/BASE: Fix conflicting types for 'uct_md_mem_advise'.

---
 src/uct/base/uct_md.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/uct/base/uct_md.c b/src/uct/base/uct_md.c
index d7f4c326c0b..730f8a2300e 100644
--- a/src/uct/base/uct_md.c
+++ b/src/uct/base/uct_md.c
@@ -554,7 +554,7 @@ ucs_status_t uct_md_mem_free(uct_md_h md, uct_mem_h memh)
 
 ucs_status_t
 uct_md_mem_advise(uct_md_h md, uct_mem_h memh, void *addr, size_t length,
-                  unsigned advice)
+                  uct_mem_advice_t advice)
 {
     if ((length == 0) || (addr == NULL)) {
         return UCS_ERR_INVALID_PARAM;
