diff -Naur squeeze-0.2.2/libsqueeze/archive.c squeeze-0.2.2.new/libsqueeze/archive.c
--- squeeze-0.2.2/libsqueeze/archive.c	2008-01-04 19:57:34.000000000 +0530
+++ squeeze-0.2.2.new/libsqueeze/archive.c	2008-01-18 18:26:57.000000000 +0530
@@ -350,10 +350,14 @@
 const gchar *
 lsq_archive_get_status(const LSQArchive *archive)
 {
-	if(archive->command)
-		return lsq_archive_command_get_comment(archive->command);
-	else
-		return NULL;
+	if (archive)
+	{
+		if(archive->command)
+			return lsq_archive_command_get_comment(archive->command);
+		else
+			return NULL;
+	}
+	return NULL;
 }
 
 void
