diff -Naurp oxine-0.6/po/Makefile.in.in oxine-0.6-p/po/Makefile.in.in
--- oxine-0.6/po/Makefile.in.in	2006-08-30 08:46:38.000000000 +0200
+++ oxine-0.6-p/po/Makefile.in.in	2006-11-13 13:11:10.000000000 +0100
@@ -29,7 +29,7 @@ gettextsrcdir = $(datadir)/gettext/po
 
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
-MKINSTALLDIRS = @MKINSTALLDIRS@
+MKINSTALLDIRS = $(top_srcdir)/mkinstalldirs
 mkinstalldirs = $(SHELL) $(MKINSTALLDIRS)
 
 GMSGFMT = @GMSGFMT@
