diff -Naur kpogre-1.3.9/kpogre/kpogre.desktop kpogre-1.3.9-p/kpogre/kpogre.desktop
--- kpogre-1.3.9/kpogre/kpogre.desktop	2002-06-06 16:09:58.000000000 +0200
+++ kpogre-1.3.9-p/kpogre/kpogre.desktop	2006-07-21 14:36:53.824429152 +0200
@@ -2,10 +2,11 @@
 [Desktop Entry]
 Encoding=UTF-8
 Type=Application
+Categories=Qt;KDE;Utility;Application;
 Exec=kpogre -caption "%c" %i %m  
 Icon=kpogre.png
 DocPath=kpogre/index.html
-Comment=
+Comment=A grafical client for PostgreSQL database system for KDE
 Comment[de]=
 Terminal=0
 Name=KPoGre
