diff -Naur osynctool-0.39/tools/CMakeLists.txt osynctool-0.39-p/tools/CMakeLists.txt
--- osynctool-0.39/tools/CMakeLists.txt	2009-09-20 22:15:52.000000000 +0200
+++ osynctool-0.39-p/tools/CMakeLists.txt	2010-02-15 15:34:11.000000000 +0100
@@ -10,6 +10,15 @@
 
 ###### INSTALL ################### 
 
+
+if (SYSCONF_INSTALL_DIR)
+  set (SYSCONFDIR "${SYSCONF_INSTALL_DIR}")
+endif (SYSCONF_INSTALL_DIR)
+
+if (NOT SYSCONFDIR)
+   set (SYSCONFDIR "${CMAKE_INSTALL_PREFIX}/etc")
+endif (NOT SYSCONFDIR)
+
 INSTALL( TARGETS osynctool DESTINATION "${CMAKE_INSTALL_PREFIX}/bin" ) 
-INSTALL( PROGRAMS osynctool.sh DESTINATION "${CMAKE_INSTALL_PREFIX}/etc/bash_completion.d" ) 
+INSTALL( PROGRAMS osynctool.sh DESTINATION "${SYSCONFDIR}/bash_completion.d" ) 
 
