diff -NrU5 hydrogen-0.9.3.orig/src/gui/DrumkitManager.h hydrogen-0.9.3/src/gui/DrumkitManager.h
--- hydrogen-0.9.3.orig/src/gui/DrumkitManager.h	2008-04-19 16:48:03.000000000 +0200
+++ hydrogen-0.9.3/src/gui/DrumkitManager.h	2008-04-19 16:48:46.000000000 +0200
@@ -41,10 +41,11 @@
 #include <qtextedit.h>
 #include <qtimer.h>
 #include <qwidget.h>
 
 #include <vector>
+#include <cstdlib>
 
 #include "UI/DrumkitManager_UI.h"
 #include "lib/Object.h"
 #include "lib/Song.h"
 
diff -NrU5 hydrogen-0.9.3.orig/src/gui/main.cpp hydrogen-0.9.3/src/gui/main.cpp
--- hydrogen-0.9.3.orig/src/gui/main.cpp	2008-04-19 16:48:03.000000000 +0200
+++ hydrogen-0.9.3/src/gui/main.cpp	2008-04-19 16:48:46.000000000 +0200
@@ -41,10 +41,11 @@
 #include "lib/Preferences.h"
 #include "lib/DataPath.h"
 #include "lib/Exception.h"
 
 #include <iostream>
+#include <cstdlib>
 using namespace std;
 
 void showInfo();
 void showUsage();
 
diff -NrU5 hydrogen-0.9.3.orig/src/lib/drivers/OssDriver.cpp hydrogen-0.9.3/src/lib/drivers/OssDriver.cpp
--- hydrogen-0.9.3.orig/src/lib/drivers/OssDriver.cpp	2008-04-19 16:48:04.000000000 +0200
+++ hydrogen-0.9.3/src/lib/drivers/OssDriver.cpp	2008-04-19 16:50:09.000000000 +0200
@@ -28,10 +28,11 @@
 #ifdef OSS_SUPPORT
 
 #include "../Preferences.h"
 
 #include <pthread.h>
+#include <cstring>
 
 audioProcessCallback ossDriver_audioProcessCallback;
 bool ossDriver_running;
 pthread_t ossDriverThread;
 int oss_driver_bufferSize = -1;
diff -NrU5 hydrogen-0.9.3.orig/src/lib/FLACFile.cpp hydrogen-0.9.3/src/lib/FLACFile.cpp
--- hydrogen-0.9.3.orig/src/lib/FLACFile.cpp	2008-04-19 16:48:04.000000000 +0200
+++ hydrogen-0.9.3/src/lib/FLACFile.cpp	2008-04-19 16:48:46.000000000 +0200
@@ -25,10 +25,11 @@
 #include "FLACFile.h"
 #include "Sample.h"
 
 #include <vector>
 #include <fstream>
+#include <cstring>
 
 #ifdef FLAC_SUPPORT
 
 //#include "FLAC/file_decoder.h"
 #include <FLAC++/all.h>
diff -NrU5 hydrogen-0.9.3.orig/src/tools/HydrogenPlayer.cpp hydrogen-0.9.3/src/tools/HydrogenPlayer.cpp
--- hydrogen-0.9.3.orig/src/tools/HydrogenPlayer.cpp	2008-04-19 16:48:04.000000000 +0200
+++ hydrogen-0.9.3/src/tools/HydrogenPlayer.cpp	2008-04-19 16:48:46.000000000 +0200
@@ -26,10 +26,11 @@
 #include <qapplication.h>
 
 #include <iostream>
 #include <stdio.h>
 #include <string>
+#include <cstdlib>
 
 #include "lib/Hydrogen.h"
 #include "lib/LocalFileMng.h"
 #include "lib/Preferences.h"
 #include "lib/fx/LadspaFX.h"
