diff -Naurp hydrogen-0.9.3/src/lib/xml/tinyxml.h hydrogen-0.9.3-p/src/lib/xml/tinyxml.h
--- hydrogen-0.9.3/src/lib/xml/tinyxml.h	2005-10-18 00:32:04.000000000 +0200
+++ hydrogen-0.9.3-p/src/lib/xml/tinyxml.h	2006-11-15 19:30:03.000000000 +0100
@@ -823,9 +823,7 @@ public:
 #endif
 
 	/// Construct.
-	TiXmlDeclaration::TiXmlDeclaration( const char * _version,
-										const char * _encoding,
-										const char * _standalone );
+	TiXmlDeclaration( const char * _version, const char * _encoding, const char * _standalone );
 
 	virtual ~TiXmlDeclaration()	{}
 
