diff -Naur hydrogen-0.9.5.1/Sconstruct hydrogen-0.9.5.1.new/Sconstruct
--- hydrogen-0.9.5.1/Sconstruct	2012-02-19 22:54:05.000000000 +0100
+++ hydrogen-0.9.5.1.new/Sconstruct	2013-04-02 20:56:53.275931268 +0200
@@ -299,7 +299,6 @@
 
         for N in glob.glob('./data/i18n/hydrogen.*'):
             env.Alias(target="install", source=env.Install(dir= env['DESTDIR'] + env['prefix'] + '/share/hydrogen/data/i18n', source=N))
-	env.Alias(target="install", source=env.Install(dir= env['DESTDIR'] + env['prefix'] + '/share/hydrogen/data', source="./data/img"))
 
 	#add every img in ./data/img to the install list. 
 	os.path.walk("./data/img/",install_images,env) 
