diff -Naur fingerprint-gui-1.04/bin/fingerprint-gui/fingerprint-gui.pro fingerprint-gui-1.04.new/bin/fingerprint-gui/fingerprint-gui.pro
--- fingerprint-gui-1.04/bin/fingerprint-gui/fingerprint-gui.pro	2010-12-12 19:06:34.000000000 +0100
+++ fingerprint-gui-1.04.new/bin/fingerprint-gui/fingerprint-gui.pro	2013-03-22 21:45:08.842999454 +0100
@@ -6,7 +6,7 @@
 INCLUDEPATH += . ../../include ../../upek/include
 
 isEmpty(PREFIX) {
-	PREFIX = /usr/local
+	PREFIX = /usr
 }
 isEmpty(LIB) {
 	LIB = lib
@@ -21,8 +21,6 @@
 DOCDIR     = $${PREFIX}/share/doc/fingerprint-gui
 
 target.path     = $${BINDIR}
-desktop.path    = $${SHAREDIR}/applications
-desktop.files   = fingerprint-gui.desktop
 manpage.path    = $${MAN1DIR}
 manpage.files   = ../../man/$${TARGET}.1
 help.path       = $${DOCDIR}
diff -Naur fingerprint-gui-1.04/bin/fingerprint-helper/fingerprint-helper.pro fingerprint-gui-1.04.new/bin/fingerprint-helper/fingerprint-helper.pro
--- fingerprint-gui-1.04/bin/fingerprint-helper/fingerprint-helper.pro	2010-12-12 19:06:34.000000000 +0100
+++ fingerprint-gui-1.04.new/bin/fingerprint-helper/fingerprint-helper.pro	2013-03-22 19:27:11.668716084 +0100
@@ -6,7 +6,7 @@
 INCLUDEPATH += . ../../include ../../upek/include
 
 isEmpty(PREFIX) {
-	PREFIX = /usr/local
+	PREFIX = /usr
 }
 isEmpty(LIB) {
 	LIB = lib
diff -Naur fingerprint-gui-1.04/bin/fingerprint-identifier/fingerprint-identifier.pro fingerprint-gui-1.04.new/bin/fingerprint-identifier/fingerprint-identifier.pro
--- fingerprint-gui-1.04/bin/fingerprint-identifier/fingerprint-identifier.pro	2010-12-12 19:06:34.000000000 +0100
+++ fingerprint-gui-1.04.new/bin/fingerprint-identifier/fingerprint-identifier.pro	2013-03-22 19:27:21.713716428 +0100
@@ -5,7 +5,7 @@
 INCLUDEPATH += . ../../include ../../upek/include
 
 isEmpty(PREFIX) {
-	PREFIX = /usr/local
+	PREFIX = /usr
 }
 isEmpty(LIB) {
 	LIB = lib
diff -Naur fingerprint-gui-1.04/bin/fingerprint-pam/fingerprint-pam.pro fingerprint-gui-1.04.new/bin/fingerprint-pam/fingerprint-pam.pro
--- fingerprint-gui-1.04/bin/fingerprint-pam/fingerprint-pam.pro	2010-12-12 19:06:34.000000000 +0100
+++ fingerprint-gui-1.04.new/bin/fingerprint-pam/fingerprint-pam.pro	2013-03-22 19:27:31.753716772 +0100
@@ -5,7 +5,7 @@
 INCLUDEPATH += . ../../include ../../upek/include
 
 isEmpty(PREFIX) {
-	PREFIX = /usr/local
+	PREFIX = /usr
 }
 isEmpty(LIB) {
 	LIB = lib
diff -Naur fingerprint-gui-1.04/bin/fingerprint-plugin/fingerprint-plugin.pro fingerprint-gui-1.04.new/bin/fingerprint-plugin/fingerprint-plugin.pro
--- fingerprint-gui-1.04/bin/fingerprint-plugin/fingerprint-plugin.pro	2010-12-12 19:06:34.000000000 +0100
+++ fingerprint-gui-1.04.new/bin/fingerprint-plugin/fingerprint-plugin.pro	2013-03-22 19:27:41.615717110 +0100
@@ -5,7 +5,7 @@
 INCLUDEPATH += . ../../include ../../upek/include
 
 isEmpty(PREFIX) {
-	PREFIX = /usr/local
+	PREFIX = /usr
 }
 isEmpty(LIB) {
 	LIB = lib
diff -Naur fingerprint-gui-1.04/bin/fingerprint-polkit-agent/fingerprint-polkit-agent.pro fingerprint-gui-1.04.new/bin/fingerprint-polkit-agent/fingerprint-polkit-agent.pro
--- fingerprint-gui-1.04/bin/fingerprint-polkit-agent/fingerprint-polkit-agent.pro	2010-12-13 20:09:41.000000000 +0100
+++ fingerprint-gui-1.04.new/bin/fingerprint-polkit-agent/fingerprint-polkit-agent.pro	2013-03-22 19:27:51.962717464 +0100
@@ -5,7 +5,7 @@
 INCLUDEPATH += . ../../include ../../upek/include
 
 isEmpty(PREFIX) {
-    PREFIX = /usr/local
+    PREFIX = /usr
 }
 isEmpty(LIB) {
     LIB = lib
diff -Naur fingerprint-gui-1.04/bin/fingerprint-suid/fingerprint-suid.pro fingerprint-gui-1.04.new/bin/fingerprint-suid/fingerprint-suid.pro
--- fingerprint-gui-1.04/bin/fingerprint-suid/fingerprint-suid.pro	2010-12-28 10:43:15.000000000 +0100
+++ fingerprint-gui-1.04.new/bin/fingerprint-suid/fingerprint-suid.pro	2013-03-22 19:28:02.169717813 +0100
@@ -4,7 +4,7 @@
 INCLUDEPATH += . ../../include ../../upek/include
 
 isEmpty(PREFIX) {
-	PREFIX = /usr/local
+	PREFIX = /usr
 }
 isEmpty(LIB) {
 	LIB = lib
diff -Naur fingerprint-gui-1.04/src/FingerprintPAM.cpp fingerprint-gui-1.04.new/src/FingerprintPAM.cpp
--- fingerprint-gui-1.04/src/FingerprintPAM.cpp	2012-03-09 23:48:41.000000000 +0100
+++ fingerprint-gui-1.04.new/src/FingerprintPAM.cpp	2013-03-22 19:07:08.662674899 +0100
@@ -37,6 +37,7 @@
 #include <errno.h>
 #include <pwd.h>
 #include <string>
+#include <unistd.h>
 
 #include "../include/Globals.h"
 
diff -Naur fingerprint-gui-1.04/src/FingerprintPlugin.cpp fingerprint-gui-1.04.new/src/FingerprintPlugin.cpp
--- fingerprint-gui-1.04/src/FingerprintPlugin.cpp	2012-03-08 15:45:12.000000000 +0100
+++ fingerprint-gui-1.04.new/src/FingerprintPlugin.cpp	2013-03-22 19:04:56.561670377 +0100
@@ -25,7 +25,8 @@
 
 #include <syslog.h>
 #include <string>
-#include <qt4/QtGui/qwidget.h>
+#include <unistd.h>
+#include <QtGui/qwidget.h>
 
 #include "PluginDialog.h"
 
diff -Naur fingerprint-gui-1.04/src/IdentifierGUI.cpp fingerprint-gui-1.04.new/src/IdentifierGUI.cpp
--- fingerprint-gui-1.04/src/IdentifierGUI.cpp	2012-03-08 15:45:12.000000000 +0100
+++ fingerprint-gui-1.04.new/src/IdentifierGUI.cpp	2013-03-22 19:07:27.591675547 +0100
@@ -25,6 +25,7 @@
 
 #include <string>
 #include <iostream>
+#include <unistd.h>
 #include "IdentifierGUI.h"
 
 #define KEEP_ON_TOP \
diff -Naur fingerprint-gui-1.04/src/MainWindowImpl.cpp fingerprint-gui-1.04.new/src/MainWindowImpl.cpp
--- fingerprint-gui-1.04/src/MainWindowImpl.cpp	2012-03-08 15:45:12.000000000 +0100
+++ fingerprint-gui-1.04.new/src/MainWindowImpl.cpp	2013-03-22 19:05:28.715671478 +0100
@@ -7,7 +7,7 @@
  * Copyright (C) 2008-2012  Wolfgang Ullrich
  */
 
-#include <qt4/QtCore/qlocale.h>
+#include <QtCore/qlocale.h>
 
 
 /*
diff -Naur fingerprint-gui-1.04/src/PamGUI.cpp fingerprint-gui-1.04.new/src/PamGUI.cpp
--- fingerprint-gui-1.04/src/PamGUI.cpp	2012-03-08 15:45:12.000000000 +0100
+++ fingerprint-gui-1.04.new/src/PamGUI.cpp	2013-03-22 19:08:30.024677685 +0100
@@ -23,6 +23,7 @@
     Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#include <unistd.h>
 #include "PamGUI.h"
 #include "FingerprintHelper.h"
 
diff -Naur fingerprint-gui-1.04/src/PamNonGUI.cpp fingerprint-gui-1.04.new/src/PamNonGUI.cpp
--- fingerprint-gui-1.04/src/PamNonGUI.cpp	2012-03-08 15:45:12.000000000 +0100
+++ fingerprint-gui-1.04.new/src/PamNonGUI.cpp	2013-03-22 19:08:46.753678257 +0100
@@ -23,6 +23,7 @@
     Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
+#include <unistd.h>
 #include "PamNonGUI.h"
 #include "FingerprintHelper.h"
 
diff -Naur fingerprint-gui-1.04/src/PluginDialog.cpp fingerprint-gui-1.04.new/src/PluginDialog.cpp
--- fingerprint-gui-1.04/src/PluginDialog.cpp	2012-03-08 15:45:12.000000000 +0100
+++ fingerprint-gui-1.04.new/src/PluginDialog.cpp	2013-03-22 19:07:43.659676097 +0100
@@ -25,7 +25,8 @@
 
 #include <sys/poll.h>
 #include <string>
-#include <qt4/QtCore/qthread.h>
+#include <unistd.h>
+#include <QtCore/qthread.h>
 
 #include "../include/PluginDialog.h"
 
diff -Naur fingerprint-gui-1.04/src/PolkitListener.cpp fingerprint-gui-1.04.new/src/PolkitListener.cpp
--- fingerprint-gui-1.04/src/PolkitListener.cpp	2012-03-08 15:45:12.000000000 +0100
+++ fingerprint-gui-1.04.new/src/PolkitListener.cpp	2013-03-22 19:06:30.332673587 +0100
@@ -32,7 +32,7 @@
 #include <polkit-qt-1/PolkitQt1/Subject>
 #include <polkit-qt-1/PolkitQt1/Identity>
 #include <polkit-qt-1/PolkitQt1/Details>
-#include <qt4/QtGui/qdialogbuttonbox.h>
+#include <QtGui/qdialogbuttonbox.h>
 
 #include "../include/Globals.h"
 #include "../include/PolkitListener.h"
diff -Naur fingerprint-gui-1.04/src/UserSettings.cpp fingerprint-gui-1.04.new/src/UserSettings.cpp
--- fingerprint-gui-1.04/src/UserSettings.cpp	2012-03-08 15:45:12.000000000 +0100
+++ fingerprint-gui-1.04.new/src/UserSettings.cpp	2013-03-22 19:08:03.053676761 +0100
@@ -32,6 +32,7 @@
 #include <stdlib.h>
 #include <sys/mount.h>
 #include <errno.h>
+#include <unistd.h>
 
 #include <QFile>
 #include <QDir>
