diff -urN liblrdf-0.4.0/examples/showdefaults.c liblrdf-0.4.0_fixt/examples/showdefaults.c
--- liblrdf-0.4.0/examples/showdefaults.c	2004-11-08 19:12:32.000000000 -0800
+++ liblrdf-0.4.0_fixt/examples/showdefaults.c	2005-01-29 02:24:45.000000000 -0800
@@ -1,6 +1,6 @@
 #include <stdio.h>
 #include <stdlib.h>
-#include <ladspa.h>
+#include <src/ladspa.h>
 
 #include "lrdf.h"
 
diff -urN liblrdf-0.4.0/examples/showtaxonomy.c liblrdf-0.4.0_fixt/examples/showtaxonomy.c
--- liblrdf-0.4.0/examples/showtaxonomy.c	2004-11-08 19:12:32.000000000 -0800
+++ liblrdf-0.4.0_fixt/examples/showtaxonomy.c	2005-01-29 02:24:53.000000000 -0800
@@ -1,7 +1,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <ladspa.h>
+#include <src/ladspa.h>
 
 #include "lrdf.h"
 
