diff -Naur gnome-video-effects-0.4.3/Makefile.am gnome-video-effects-0.4.3-p/Makefile.am
--- gnome-video-effects-0.4.3/Makefile.am	2017-02-09 13:29:45.000000000 +0100
+++ gnome-video-effects-0.4.3-p/Makefile.am	2017-02-21 20:19:06.804666428 +0100
@@ -4,7 +4,7 @@
 # translations are included directly in the effect files.
 DIST_SUBDIRS = po
 
-pkgconfigdir = $(datadir)/pkgconfig
+pkgconfigdir = /usr/lib/pkgconfig
 pkgconfig_DATA = gnome-video-effects.pc
 
 %.effect: %.effect.in $(INTLTOOL_MERGE) $(wildcard $(top_srcdir)/po/*.po) ; LC_ALL=C $(INTLTOOL_MERGE) -d -u -c $(top_builddir)/po/.intltool-merge-cache $(top_srcdir)/po $< $@
diff -Naur gnome-video-effects-0.4.3/Makefile.in gnome-video-effects-0.4.3-p/Makefile.in
--- gnome-video-effects-0.4.3/Makefile.in	2017-02-09 13:31:54.000000000 +0100
+++ gnome-video-effects-0.4.3-p/Makefile.in	2017-02-21 20:19:06.808666527 +0100
@@ -331,7 +331,7 @@
 # This causes the po directory to be distributed, but not installed, as the
 # translations are included directly in the effect files.
 DIST_SUBDIRS = po
-pkgconfigdir = $(datadir)/pkgconfig
+pkgconfigdir = /usr/lib/pkgconfig
 pkgconfig_DATA = gnome-video-effects.pc
 effectsdir = $(pkgdatadir)
 effects_in_files = $(wildcard effects/*.effect.in)
