diff -ur a/plugins/gtkui/actions.c b/plugins/gtkui/actions.c
--- a/plugins/gtkui/actions.c	2012-05-19 06:50:51.554366843 -0500
+++ b/plugins/gtkui/actions.c	2012-05-19 06:51:27.098366334 -0500
@@ -133,7 +133,7 @@
                     g_signal_connect ((gpointer) actionitem, "activate",
                         G_CALLBACK (on_actionitem_activate),
                         action);
-                    g_object_set_data_full (G_OBJECT (actionitem), "plugaction", strdup (action->name), free);
+                    g_object_set_data_full (G_OBJECT (actionitem), "plugaction", g_strdup (action->name), free);
                     break;
                 }
                 *slash = 0;
