--- libxfcegui4-4.2.2/libxfcegui4/netk-tasklist.c	2005-05-15 09:53:29.000000000 +0200
+++ libxfcegui4-4.2.2/libxfcegui4/netk-tasklist.c	2005-07-31 16:19:44.000000000 +0200
@@ -2238,8 +2238,14 @@ 
     if (task->tasklist->priv->show_label) {
         task->label = gtk_label_new (text);
         gtk_widget_show (task->label);
-        gtk_table_attach (GTK_TABLE (table),
+        if(GTK_CHECK_VERSION(2, 7, 0)){
+            gtk_table_attach (GTK_TABLE (table),
+                      task->label, 1, 2, 0, 1, GTK_EXPAND|GTK_FILL|GTK_SHRINK, GTK_EXPAND, 0, 0);
+        }
+        else{
+            gtk_table_attach (GTK_TABLE (table),
                       task->label, 1, 2, 0, 1, GTK_EXPAND, GTK_EXPAND, 0, 0);
+        }
     }
 
     gtk_container_add (GTK_CONTAINER (task->button), table);
