From http://svn.mythtv.org/trac/raw-attachment/ticket/6414/mythmusic-visualize.diff

Index: mythplugins/mythmusic/mythmusic/visualize.h
===================================================================
--- mythplugins/mythmusic/mythmusic/visualize.h	(revision 20296)
+++ mythplugins/mythmusic/mythmusic/visualize.h	(working copy)
@@ -32,7 +32,7 @@
 #define myth_fftw_complex_cast fftw_complex
 #elif (myth_fftw_float == float)
 #define myth_fftw_complex_cast fftwf_complex
-#elif (myth_fftw_float == long double)
+#elif (myth_fftw_float == long)
 #define myth_fftw_complex_cast fftwl_complex
 #endif
 #elif	FFTW2_SUPPORT
