diff -Naur BashBurn-2.0.orig/Install.sh BashBurn-2.0/Install.sh
--- BashBurn-2.0.orig/Install.sh	2007-03-07 02:22:43.000000000 +0100
+++ BashBurn-2.0/Install.sh	2007-03-08 18:10:16.000000000 +0100
@@ -1,5 +1,10 @@
 #!/usr/bin/env bash
 
+if [ "$1" = "--destdir" ]
+then
+	BBDESTDIR=$2
+fi
+
 BBCONFFILE=/etc/bashburnrc
 BBROOTDIR=$(dirname $0)/
 BBINSTDIR=$(pwd)
@@ -30,8 +35,8 @@
 {
 	echo; echo -n "Installing files..."
 	# Does the path already exist
-	if [[ ! -e "${BBINSTALL_PATH}" ]]; then 
-		mkdir ${BBINSTALL_PATH}
+	if [[ ! -e "${BBDESTDIR}${BBINSTALL_PATH}" ]]; then 
+		mkdir -p ${BBDESTDIR}${BBINSTALL_PATH}
 	fi
 	cp -Rfp ${BBROOTDIR}BashBurn.sh \
 		${BBROOTDIR}burning \
@@ -42,12 +47,13 @@
 		${BBROOTDIR}misc \
 		${BBROOTDIR}func \
 		${BBROOTDIR}HOWTO \
-		${BBINSTALL_PATH}
+		${BBDESTDIR}${BBINSTALL_PATH}
 	echo -e "\033[1;32m[DONE]\033[1;0m"; echo
 
 	echo -n "Creating symlink..."
 	if [ "$SLACK" = "yes" ]; then
-        cd /usr/bin && ln -sf ${BBINSTALL_PATH}/BashBurn.sh bashburn && cd ${BBINSTDIR}
+	[ -d ${BBDESTDIR}/usr/bin ] || mkdir -p ${BBDESTDIR}/usr/bin
+        cd ${BBDESTDIR}/usr/bin && ln -sf ${BBINSTALL_PATH}/BashBurn.sh bashburn && cd ${BBINSTDIR}
 	else
 	# Nick - is this next line right if user selects an alternative install location?
         cd /usr/local/bin && ln -sf ${BBINSTALL_PATH}/BashBurn.sh bashburn && cd ${BBINSTDIR}
@@ -57,10 +63,10 @@
 	if [ "$SLACK" = "yes" ]; then
 	echo -e "Setting ownership correctly for"
 	echo -en "\033[1;36mSlackware\033[1;0m [root:root]..."
-	chown root:root /usr/bin/bashburn
-	chown -R root:root /usr/bin/BashBurn
-	chown -R root:root /usr/bin/BashBurn/lang
-	chown root:root /usr/bin/BashBurn/HOWTO
+	[ -e ${BBDESTDIR}/usr/bin/bashburn ] && chown root:root ${BBDESTDIR}/usr/bin/bashburn
+	chown -R root:root ${BBDESTDIR}/${BBINSTALL_PATH}
+	chown -R root:root ${BBDESTDIR}/${BBINSTALL_PATH}/lang
+	chown root:root ${BBDESTDIR}/${BBINSTALL_PATH}/HOWTO
 	echo -e "\033[1;32m[DONE]\033[1;0m"; echo
 	fi
 
@@ -80,8 +86,9 @@
 		echo -e "\033[1;32m[DONE]\033[1;0m"
 	else
 		echo; echo -n "Installing config file..."
-		cp -f ${BBROOTDIR}bashburnrc ${BBCONFFILE}
-		sed -e "s°^BBROOTDIR.*°BBROOTDIR: $BBINSTALL_PATH°" -i ${BBCONFFILE}
+		[ -d $(dirname ${BBDESTDIR}${BBCONFFILE}) ] || mkdir -p $(dirname ${BBDESTDIR}${BBCONFFILE})
+		cp -f ${BBROOTDIR}bashburnrc ${BBDESTDIR}${BBCONFFILE}
+		sed -e "s°^BBROOTDIR.*°BBROOTDIR: $BBINSTALL_PATH°" -i ${BBDESTDIR}${BBCONFFILE}
 		sleep 1s
 		echo -e "\033[1;32m[DONE]\033[1;0m"
 	fi
