diff -Naur psmisc-22.4.orig/configure.ac psmisc-22.4/configure.ac
--- psmisc-22.4.orig/configure.ac	2007-04-19 16:41:59.000000000 +0200
+++ psmisc-22.4/configure.ac	2007-04-18 01:23:52.000000000 +0200
@@ -58,6 +58,13 @@
 AC_TYPE_SIZE_T
 AC_CHECK_MEMBERS([struct stat.st_rdev])
 AC_TYPE_UID_T
+AC_CHECK_MEMBERS([struct user_regs_struct.orig_eax, struct user_regs_struct.eax, struct user_regs_struct.ebx, struct user_regs_struct.ecx, struct user_regs_struct.edx], [],[], [#include <linux/user.h>])
+AM_CONDITIONAL(WANT_PEEKFD,
+ test $ac_cv_member_struct_user_regs_struct_orig_eax = yes &&
+   test $ac_cv_member_struct_user_regs_struct_eax = yes &&
+   test $ac_cv_member_struct_user_regs_struct_ebx = yes &&
+   test $ac_cv_member_struct_user_regs_struct_ecx = yes &&
+   test $ac_cv_member_struct_user_regs_struct_edx = yes )
 
 dnl Check for language stuff
 AM_GNU_GETTEXT([external])
diff -Naur psmisc-22.4.orig/src/Makefile.am psmisc-22.4/src/Makefile.am
--- psmisc-22.4.orig/src/Makefile.am	2007-04-19 16:41:59.000000000 +0200
+++ psmisc-22.4/src/Makefile.am	2007-04-18 01:21:43.000000000 +0200
@@ -1,7 +1,10 @@
 
 AM_CFLAGS = -Wall -DLOCALEDIR=\"/usr/share/locale\"
 
-bin_PROGRAMS = fuser killall pstree oldfuser peekfd
+bin_PROGRAMS = fuser killall pstree oldfuser 
+if WANT_PEEKFD
+  bin_PROGRAMS += peekfd
+endif
 
 oldfuser_SOURCES = oldfuser.c comm.h signals.c signals.h loop.h i18n.h
 
