diff -ur src.orig/atom.h src/atom.h
--- src.orig/atom.h	2008-12-04 07:41:34.000000000 +0100
+++ src/atom.h	2009-09-08 16:32:03.000000000 +0200
@@ -21,6 +21,8 @@
 #ifndef ATOM_H
 #define ATOM_H
 
+#include <cstdlib>
+#include <cstring>
 #include "libghemicaldefine.h"
 #include "libghemical-features.h"
 
diff -ur src.orig/notice.h src/notice.h
--- src.orig/notice.h	2008-12-09 09:04:10.000000000 +0100
+++ src/notice.h	2009-09-08 16:35:26.000000000 +0200
@@ -21,6 +21,7 @@
 #ifndef NOTICE_H
 #define NOTICE_H
 
+#include <cstring>
 #include "libghemicaldefine.h"
 #include "libghemical-features.h"
 
diff -ur src.orig/sasaeval.cpp src/sasaeval.cpp
--- src.orig/sasaeval.cpp	2009-03-31 06:54:33.000000000 +0200
+++ src/sasaeval.cpp	2009-09-08 16:38:26.000000000 +0200
@@ -21,6 +21,9 @@
 #include "sasaeval.h"
 
 #include "engine.h"
+#include <vector>
+#include <algorithm>
+using namespace std;
 
 #include "local_i18n.h"
 #include "notice.h"
diff -ur src.orig/sasaeval.h src/sasaeval.h
--- src.orig/sasaeval.h	2008-12-16 12:21:58.000000000 +0100
+++ src/sasaeval.h	2009-09-08 16:39:50.000000000 +0200
@@ -25,6 +25,7 @@
 
 #include "typedef.h"
 #include "notice.h"
+#include <stdlib.h>
 
 #include <cstdlib>
 using namespace std;
