--- vault-0.3.1.5/vault.cabal	2001-09-09 03:46:40.000000000 +0200
+++ vault-0.3.1.5.new/vault.cabal	2022-11-17 22:25:20.405588161 +0100
@@ -28,8 +28,11 @@
                     ,GHC == 8.2.2
                     ,GHC == 8.4.4
                     ,GHC == 8.6.5
-                    ,GHC == 8.8.3
-                    ,GHC == 8.10.1
+                    ,GHC == 8.8.4
+                    ,GHC == 8.10.7
+                    ,GHC == 9.0.2
+                    ,GHC == 9.2.4
+                    ,GHC == 9.4.1
 
 extra-source-files:
     CHANGELOG.md
@@ -48,10 +51,10 @@
 
 Library
     hs-source-dirs:     src
-    build-depends:      base >= 4.5 && < 4.16,
+    build-depends:      base >= 4.5 && < 4.18,
                         containers >= 0.4 && < 0.7,
                         unordered-containers >= 0.2.3.0 && < 0.3,
-                        hashable >= 1.1.2.5 && < 1.4
+                        hashable >= 1.1.2.5 && < 1.5
 
     if impl(ghc < 8.0)
         build-depends:  semigroups >= 0.1 && < 1.0
