diff -ur src/install.rdf src2/install.rdf
--- src/install.rdf	2008-02-14 16:43:09.000000000 -0600
+++ src2/install.rdf	2008-07-18 15:57:39.000000000 -0500
@@ -13,8 +13,8 @@
     <em:targetApplication>
       <Description>
         <em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
-        <em:minVersion>3.0b3</em:minVersion>
-        <em:maxVersion>3.0b4pre</em:maxVersion>
+        <em:minVersion>3.0</em:minVersion>
+        <em:maxVersion>3.0.*</em:maxVersion>
       </Description>
     </em:targetApplication>
    
