--- src/Makefile	2020-01-21 16:11:10.000000000 +0100
+++ src.new/Makefile	2020-01-23 14:23:30.591265218 +0100
@@ -8,7 +8,6 @@
 DESTDIR ?=
 SRCDIR ?= $(PREFIX)/src
 DKMSDIR ?= $(SRCDIR)/wireguard
-DEPMOD ?= depmod
 
 PWD := $(shell pwd)
 
@@ -32,7 +31,6 @@
 
 module-install:
 	@$(MAKE) -C $(KERNELDIR) M=$(PWD) WIREGUARD_VERSION="$(WIREGUARD_VERSION)" modules_install
-	$(DEPMOD) -a $(KERNELRELEASE)
 
 install: module-install
 
