diff -Naur php-5.6.38/sapi/fpm/php-fpm.service.in php-5.6.38-p/sapi/fpm/php-fpm.service.in
--- php-5.6.38/sapi/fpm/php-fpm.service.in	2018-09-12 00:12:36.000000000 +0200
+++ php-5.6.38-p/sapi/fpm/php-fpm.service.in	2018-10-23 18:04:04.881513004 +0200
@@ -5,7 +5,7 @@
 [Service]
 Type=@php_fpm_systemd@
 PIDFile=@localstatedir@/run/php-fpm.pid
-ExecStart=@sbindir@/php-fpm --nodaemonize --fpm-config @sysconfdir@/php-fpm.conf
+ExecStart=/usr/sbin/php-fpm --nodaemonize --fpm-config @sysconfdir@/php/php-fpm.conf
 ExecReload=/bin/kill -USR2 $MAINPID
 
 [Install]
