--- php-5.2.2/ext/gd/libgd/gd_png.c.orig	2007-05-24 17:30:52.000000000 +0200
+++ php-5.2.2/ext/gd/libgd/gd_png.c	2007-05-24 17:33:32.000000000 +0200
@@ -71,7 +71,11 @@
 
 static void gdPngReadData (png_structp png_ptr, png_bytep data, png_size_t length)
 {
-	gdGetBuf(data, length, (gdIOCtx *) png_get_io_ptr(png_ptr));
+	int check;
+	check = gdGetBuf (data, length, (gdIOCtx *) png_get_io_ptr (png_ptr));
+	if (check != length) {
+		png_error(png_ptr, "Read Error: truncated data");
+	}
 }
 
 static void gdPngWriteData (png_structp png_ptr, png_bytep data, png_size_t length)
