diff -Naur php-5.6.40/ext/mysql/php_mysql_structs.h php-5.6.40-p/ext/mysql/php_mysql_structs.h
--- php-5.6.40/ext/mysql/php_mysql_structs.h	2019-01-09 10:54:13.000000000 +0100
+++ php-5.6.40-p/ext/mysql/php_mysql_structs.h	2019-04-05 16:27:06.022201877 +0200
@@ -40,6 +40,7 @@
 #include "mysql_mysqlnd.h"
 #else
 #include <mysql.h>
+#include <server/mysql_version.h>
 #endif
 
 #ifdef PHP_MYSQL_UNIX_SOCK_ADDR
