
Fix compile errors with GCC 4 due to bad C.  

--- php-5.0.4/ext/xmlrpc/libxmlrpc/xml_element.c.gcc4
+++ php-5.0.4/ext/xmlrpc/libxmlrpc/xml_element.c
@@ -113,7 +113,7 @@
 #include "expat.h"
 #include "encodings.h"
 
-#define my_free(thing)  if(thing) {free(thing); thing = 0;}
+#define my_free(thing)  if(thing) {free(thing); thing = NULL;}
 
 #define XML_DECL_START                 "<?xml"
 #define XML_DECL_START_LEN             sizeof(XML_DECL_START) - 1
@@ -189,7 +189,10 @@
 
       Q_Destroy(&root->children);
       Q_Destroy(&root->attrs);
-      my_free((char*)root->name);
+      if (root->name) {
+          free((char *)root->name);
+          root->name = NULL;
+      }
       simplestring_free(&root->text);
       my_free(root);
    }
