diff -up php-7.3.0RC1/ext/dba/config.m4.db60~ php-7.3.0RC1/ext/dba/config.m4
--- php-7.3.0RC1/ext/dba/config.m4.db60~	2018-09-13 22:48:02.122654318 +0200
+++ php-7.3.0RC1/ext/dba/config.m4	2018-09-13 22:48:47.222861935 +0200
@@ -277,7 +277,7 @@ AC_DEFUN([PHP_DBA_DB_CHECK],[
         ]])],[
           AC_EGREP_CPP(yes,[
 #include "$THIS_INCLUDE"
-#if DB_VERSION_MAJOR == $1 || ($1 == 4 && DB_VERSION_MAJOR == 5)
+#if DB_VERSION_MAJOR == $1 || ($1 == 4 && DB_VERSION_MAJOR >= 5)
             yes
 #endif
           ],[
