diff -Naur php-5.1.2.orig/pear/Makefile.frag php-5.1.2/pear/Makefile.frag
--- php-5.1.2.orig/pear/Makefile.frag	2005-12-12 13:39:50.000000000 +0100
+++ php-5.1.2/pear/Makefile.frag	2006-01-16 14:21:20.000000000 +0100
@@ -6,7 +6,7 @@
 PEAR_INSTALL_FLAGS = -n -dshort_open_tag=0 -dsafe_mode=0 -derror_reporting=E_ALL -ddetect_unicode=0
 
 install-pear-installer: $(SAPI_CLI_PATH)
-	@$(top_builddir)/sapi/cli/php $(PEAR_INSTALL_FLAGS) $(builddir)/install-pear-nozlib.phar -d "$(peardir)" -b "$(bindir)"
+	@$(top_builddir)/sapi/cli/php $(PEAR_INSTALL_FLAGS) $(builddir)/install-pear-nozlib.phar -d "$(INSTALL_ROOT)$(peardir)" -b "$(INSTALL_ROOT)$(bindir)"
 
 install-pear:
 	@echo "Installing PEAR environment:      $(INSTALL_ROOT)$(peardir)/"
