diff -Naur php-5.2.6.orig/scripts/phpize.in php-5.2.6/scripts/phpize.in
--- php-5.2.6.orig/scripts/phpize.in	2007-06-28 21:10:35.000000000 -0400
+++ php-5.2.6/scripts/phpize.in	2008-10-14 15:15:24.000000000 -0400
@@ -128,8 +128,13 @@
   (cd "$phpdir" && cp $FILES_BUILD "$builddir"/build)
   (cd "$phpdir" && cp $FILES "$builddir")
   (cd "$builddir" && cat acinclude.m4 ./build/libtool.m4 > aclocal.m4)
+  (cd "$builddir" && cat "/usr/share/aclocal/ltsugar.m4" >> aclocal.m4)
+  (cd "$builddir" && cat "/usr/share/aclocal/ltversion.m4" >> aclocal.m4)
+  (cd "$builddir" && cat "/usr/share/aclocal/lt~obsolete.m4" >> aclocal.m4)
+  (cd "$builddir" && cat "/usr/share/aclocal/ltoptions.m4" >> aclocal.m4)
 }
 
+
 phpize_replace_prefix()
 {
   $SED \
