diff -Naur cvs-1.11.23/lib/getline.c cvs-1.11.23-p/lib/getline.c
--- cvs-1.11.23/lib/getline.c	2005-04-04 22:46:05.000000000 +0200
+++ cvs-1.11.23-p/lib/getline.c	2008-05-13 22:17:57.000000000 +0200
@@ -155,7 +155,7 @@
 }
 
 int
-getline (lineptr, n, stream)
+my_getline (lineptr, n, stream)
      char **lineptr;
      size_t *n;
      FILE *stream;
diff -Naur cvs-1.11.23/lib/getline.h cvs-1.11.23-p/lib/getline.h
--- cvs-1.11.23/lib/getline.h	2005-04-04 22:46:05.000000000 +0200
+++ cvs-1.11.23-p/lib/getline.h	2008-05-13 22:18:06.000000000 +0200
@@ -12,7 +12,7 @@
 #define GETLINE_NO_LIMIT -1
 
 int
-  getline __PROTO ((char **_lineptr, size_t *_n, FILE *_stream));
+  my_getline __PROTO ((char **_lineptr, size_t *_n, FILE *_stream));
 int
   getline_safe __PROTO ((char **_lineptr, size_t *_n, FILE *_stream,
                          int limit));
