--- node-v14.17.5/src/cares_wrap.h	2021-08-11 03:55:38.000000000 +0200
+++ node-v14.17.5.new/src/cares_wrap.h	2021-08-12 11:15:16.199417955 +0200
@@ -22,7 +22,15 @@
 # include <netdb.h>
 #endif  // __POSIX__
 
-# include <ares_nameser.h>
+#if defined(__ANDROID__) || \
+    defined(__MINGW32__) || \
+    defined(__OpenBSD__) || \
+    defined(_MSC_VER)
+
+# include <nameser.h>
+#else
+# include <arpa/nameser.h>
+#endif
 
 namespace node {
 namespace cares_wrap {
