--- node-v23.1.0/configure.py	2024-10-24 14:44:04.000000000 +0200
+++ node-v23.1.0.new/configure.py	2024-10-29 16:24:45.409165156 +0100
@@ -1879,7 +1879,7 @@
   elif with_intl == 'system-icu':
     # ICU from pkg-config.
     o['variables']['v8_enable_i18n_support'] = 1
-    pkgicu = pkg_config('icu-i18n')
+    pkgicu = pkg_config('icu-i18n icu-uc')
     if not pkgicu[0]:
       error('''Could not load pkg-config data for "icu-i18n".
        See above errors or the README.md.''')
