diff -Naur node-v6.3.1/deps/npm/lib/config/core.js node-v6.3.1-p/deps/npm/lib/config/core.js
--- node-v6.3.1/deps/npm/lib/config/core.js	2016-07-21 22:44:28.000000000 +0200
+++ node-v6.3.1-p/deps/npm/lib/config/core.js	2016-08-12 00:46:20.116334114 +0200
@@ -151,16 +151,14 @@
     // Eg, `npm config get globalconfig --prefix ~/local` should
     // return `~/local/etc/npmrc`
     // annoying humans and their expectations!
-    if (conf.get('prefix')) {
-      var etc = path.resolve(conf.get('prefix'), 'etc')
-      mkdirp(etc, function () {
-        defaults.globalconfig = path.resolve(etc, 'npmrc')
-        defaults.globalignorefile = path.resolve(etc, 'npmignore')
-        afterUserContinuation()
-      })
-    } else {
+
+    // well this should be a var for 'sysconfigdir' from ./configure
+    var globalconfig = path.resolve('/etc', 'npm')
+    mkdirp(globalconfig, function () {
+      defaults.globalconfig = path.resolve(globalconfig, 'npmrc')
+      defaults.globalignorefile = path.resolve(globalconfig, 'npmignore')
       afterUserContinuation()
-    }
+    })
   }
 
   function afterUserContinuation () {
