diff -aur fme-1.1.2/src/ui.cc fme-1.1.2-2/src/ui.cc
--- fme-1.1.2/src/ui.cc	2008-07-27 09:33:08.000000000 -0500
+++ fme-1.1.2-2/src/ui.cc	2010-05-15 19:18:51.000000000 -0500
@@ -886,7 +886,7 @@
 	unsigned i=1;
 	for ( Glib::Container_Helpers::ArrayHandleIterator<Glib::Container_Helpers::TypeTraits<Glib::ustring> > iter=dirs.begin(); iter!=dirs.end(); iter++ ) {
 		try {
-			Glib::Dir::Dir directory(*iter);
+			Glib::Dir directory(*iter);
 #else
 	boost::regex re("\\:");
 	boost::sregex_token_iterator dir(path.begin(), path.end(), re, -1);
@@ -894,7 +894,7 @@
 
 	for ( unsigned i=1; dir!=end;) {
 		try {
-			Glib::Dir::Dir directory(*dir++);
+			Glib::Dir directory(*dir++);
 #endif /* GLIBMM_REGEX */
 			for ( Glib::DirIterator iter=directory.begin(); iter!=directory.end(); iter++ ) {
 				Gtk::TreeModel::Row rowCompletion = *(refCompletionModel->append());
@@ -1259,7 +1259,7 @@
 
 	for ( std::vector<Glib::ustring>::iterator iterPath = preferences.getPathsIcons().begin(); iterPath!=preferences.getPathsIcons().end(); iterPath++ ) {
 		try {
-			Glib::Dir::Dir directory(*iterPath);
+			Glib::Dir directory(*iterPath);
 			for ( Glib::DirIterator iter=directory.begin(); iter!=directory.end(); iter++ ) {
 				Glib::ustring icon_name = *iter;
 				if ( icon_name.find(command_name)!=Glib::ustring::npos ) {
