diff -aur ACE_wrappers/configure fw_ACE_wrappers/configure
--- ACE_wrappers/../configure	2010-03-03 15:24:00.000000000 +0100
+++ fw_ACE_wrappers/../configure	2010-06-11 15:11:16.000000000 +0200
@@ -9024,7 +9024,7 @@
 CRYPTO_set_locking_callback (0);
 SSLeay_add_ssl_algorithms ();
 SSL_load_error_strings ();
-SSL_METHOD * meth = TLSv1_method ();
+const SSL_METHOD * meth = TLSv1_method ();
 SSL_CTX * ctx = SSL_CTX_new (meth);
 SSL * ssl = SSL_new (ctx);
 int fd = 2000;  // Dummy file descriptor value.
@@ -9094,7 +9094,7 @@
 CRYPTO_set_locking_callback (0);
 SSLeay_add_ssl_algorithms ();
 SSL_load_error_strings ();
-SSL_METHOD * meth = TLSv1_method ();
+const SSL_METHOD * meth = TLSv1_method ();
 SSL_CTX * ctx = SSL_CTX_new (meth);
 SSL * ssl = SSL_new (ctx);
 int fd = 2000;  // Dummy file descriptor value.
@@ -9199,7 +9199,7 @@
 CRYPTO_set_locking_callback (0);
 SSLeay_add_ssl_algorithms ();
 SSL_load_error_strings ();
-SSL_METHOD * meth = TLSv1_method ();
+const SSL_METHOD * meth = TLSv1_method ();
 SSL_CTX * ctx = SSL_CTX_new (meth);
 SSL * ssl = SSL_new (ctx);
 int fd = 2000;  // Dummy file descriptor value.
