*** bzip2.c	2006-03-16 12:58:01.000000000 -0500
--- bzip2-new.c	2006-03-16 12:58:36.000000000 -0500
***************
*** 204,209 ****
--- 204,210 ----
  
  void DecompressBzipGzip ( GString *list , gchar *path , gboolean dummy , gboolean add )
  {
+     type = dummy;
      tmp = OpenTempFile ( dummy , NULL );
      if ( add ) g_child_watch_add ( compressor_pid , (GChildWatchFunc) AddToTar , list->str );
          else g_child_watch_add ( compressor_pid , (GChildWatchFunc) DeleteFromTar , list->str );
