diff -Naur qtwebkit-5.212.0-alpha2/Source/WebCore/dom/Document.cpp qtwebkit-5.212.0-alpha2.new/Source/WebCore/dom/Document.cpp
--- qtwebkit-5.212.0-alpha2/Source/WebCore/dom/Document.cpp	2017-06-15 13:56:11.000000000 +0200
+++ qtwebkit-5.212.0-alpha2.new/Source/WebCore/dom/Document.cpp	2020-02-04 14:12:29.397539211 +0100
@@ -4407,12 +4407,12 @@
     unsigned i = 0;
 
     UChar32 c;
-    U16_NEXT(characters, i, length, c)
+    U16_NEXT(characters, i, length, c);
     if (!isValidNameStart(c))
         return false;
 
     while (i < length) {
-        U16_NEXT(characters, i, length, c)
+        U16_NEXT(characters, i, length, c);
         if (!isValidNamePart(c))
             return false;
     }
@@ -4474,7 +4474,7 @@
 
     for (unsigned i = 0; i < length;) {
         UChar32 c;
-        U16_NEXT(qualifiedName, i, length, c)
+        U16_NEXT(qualifiedName, i, length, c);
         if (c == ':') {
             if (sawColon) {
                 ec = NAMESPACE_ERR;
