--- volti-0.1.6/src/main.py~	2010-01-16 13:23:19.000000000 +0000
+++ volti-0.1.6/src/main.py	2010-01-25 22:28:55.000000000 +0000
@@ -175,7 +175,10 @@
     def mixer_get_pid(self):
         pid = Popen(["pidof", "-x", preferences.prefs["mixer"]], stdout=PIPE).communicate()[0]
         if pid:
-            return int(pid)
+            try:
+                return int(pid)
+            except ValueError:
+                return None
         return None
 
     def mixer_get_active(self):
