diff -Naur xloadimage.4.1/Imakefile xloadimage.4.1/Imakefile
--- xloadimage.4.1//Imakefile	2011-02-11 13:05:53.000000000 +0100
+++ xloadimage.4.1/Imakefile	2011-07-27 14:05:57.464771119 +0200
@@ -22,10 +22,10 @@
 #define HAS_JPEG
 
 #ifdef HAS_TIFF
-TIFF_DEFS = -DHAS_TIFF
- TIFF_LIB = -ltiff
- TIFF_SRC = tiff.c
- TIFF_OBJ = tiff.o
+#TIFF_DEFS = -DHAS_TIFF
+# TIFF_LIB = -ltiff
+# TIFF_SRC = tiff.c
+# TIFF_OBJ = tiff.o
 #endif
 
 #ifdef HAS_JPEG
@@ -40,7 +40,7 @@
  PNG_SRC = png.c
  PNG_OBJ = png.o
 
-  EXTRA_DEFINES = $(TIFF_DEFS) $(JPEG_DEFS) $(PNG_DEFS)
+  EXTRA_DEFINES = $(TIFF_DEFS) $(JPEG_DEFS) $(PNG_DEFS) -DHAVE_LIBPNG -DHAVE_LIBJPEG
 LOCAL_LIBRARIES = $(TIFF_LIB)  $(JPEG_LIB) $(PNG_LIB) $(XLIB)
 
 SRCS = new.c niff.c value.c zio.c cmuwmraster.c faces.c fbm.c gif.c imagetypes.c img.c  mac.c mcidas.c mc_tables.c pbm.c pcx.c pdsuncomp.c  rle.c rlelib.c sunraster.c vff.c vicar.c xbitmap.c xpixmap.c  xwd.c bright.c clip.c compress.c dither.c fill.c halftone.c  merge.c reduce.c rotate.c smooth.c undither.c zoom.c config.c misc.c options.c root.c send.c window.c  xloadimage.c  $(TIFF_SRC) $(JPEG_SRC) $(PNG_SRC)
diff -Naur xloadimage.4.1//png.c xloadimage.4.1new//png.c
--- xloadimage.4.1//png.c	2011-02-11 12:47:37.000000000 +0100
+++ xloadimage.4.1new//png.c	2011-07-27 14:06:45.931931045 +0200
@@ -8,16 +8,17 @@
  * PNG_COLOR_TYPE_GRAY_ALPHA              O   O           16
  * PNG_COLOR_TYPE_RGB                     O   O  possible 16 * 3
  * PNG_COLOR_TYPE_RGB_ALPHA               O   O  possible 16 * 3
- * 
+ *
  * This code is based on jpeg.c and sample code from the libpng-1.0.5.
  * 2000/01/10: YOSHIDA Hiroshi
  *
  * TODO:
  * pngLoad(): Alpha channel, Transparency palette.
  * pngDump(): Dump.
- * 
+ *
  */
 
+#define _GETOPT_H
 
 #include "image.h"	/* xloadimage declarations */
 #ifdef HAVE_LIBPNG

