diff -Naur nomacs-3.8.0/ImageLounge/CMakeLists.txt nomacs-3.8.0-p/ImageLounge/CMakeLists.txt
--- nomacs-3.8.0/ImageLounge/CMakeLists.txt	2017-11-09 10:48:54.000000000 +0100
+++ nomacs-3.8.0-p/ImageLounge/CMakeLists.txt	2017-11-25 21:33:00.075920245 +0100
@@ -22,11 +22,6 @@
 set(NOMACS_VERSION ${NOMACS_VERSION_MAJOR}.${NOMACS_VERSION_MINOR})
 set(NOMACS_FULL_VERSION ${NOMACS_VERSION}.${NOMACS_VERSION_PATCH})
 
-# Codecov
-if(CMAKE_COMPILER_IS_GNUCXX)
-    include("cmake/CodeCoverage.cmake")
-    setup_target_for_coverage(${PROJECT_NAME}_coverage tests coverage)
-endif()
 
 if(CMAKE_CL_64)
 	SET(NMC_ARCHITECTURE "x64")
