diff -ur nomacs-1.6.2/CMakeLists.txt nomacs-1.6.2.n/CMakeLists.txt
--- nomacs-1.6.2/CMakeLists.txt	2013-12-20 15:11:00.000000000 +0100
+++ nomacs-1.6.2/CMakeLists.txt	2014-01-28 23:17:43.249575185 +0100
@@ -222,10 +222,8 @@
 	          
 	# OpenCV
 	SET(OpenCV_LIBS "")
-	if (PKG_CONFIG_FOUND) # not sure: pkgconfig is needed for old linux  with old old opencv systems
-		pkg_check_modules(OpenCV  opencv>=2.1.0)
-		SET(OpenCV_LIBS ${OpenCV_LIBRARIES})
-	endif(PKG_CONFIG_FOUND)
+	find_package(OpenCV REQUIRED)
+	SET(OpenCV_LIBS ${OpenCV_LIBRARIES})
 	IF (OpenCV_LIBS STREQUAL "") 
 		find_package(OpenCV 2.1.0 REQUIRED core imgproc)
 	ENDIF()
