diff -Naur nomacs-3.8.0/ImageLounge/cmake/Unix.cmake nomacs-3.8.0-p/ImageLounge/cmake/Unix.cmake
--- nomacs-3.8.0/ImageLounge/cmake/Unix.cmake	2017-11-09 10:48:54.000000000 +0100
+++ nomacs-3.8.0-p/ImageLounge/cmake/Unix.cmake	2018-02-16 17:39:49.921421542 +0100
@@ -104,7 +104,7 @@
   if(USE_SYSTEM_QUAZIP)
     SET(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
-    find_package(QuaZIP REQUIRED)
+    find_package(QuaZip5 REQUIRED)
     if(NOT QUAZIP_FOUND)
 	    message(FATAL_ERROR "QUAZIP not found. It's mandatory when used with ENABLE_QUAZIP enabled, you can also disable USE_SYSTEM_QUAZIP")
     else()
diff -Naur nomacs-3.8.0/ImageLounge/CMakeLists.txt nomacs-3.8.0-p/ImageLounge/CMakeLists.txt
--- nomacs-3.8.0/ImageLounge/CMakeLists.txt	2017-11-09 10:48:54.000000000 +0100
+++ nomacs-3.8.0-p/ImageLounge/CMakeLists.txt	2018-02-16 19:23:52.238508092 +0100
@@ -22,12 +22,6 @@
 set(NOMACS_VERSION ${NOMACS_VERSION_MAJOR}.${NOMACS_VERSION_MINOR})
 set(NOMACS_FULL_VERSION ${NOMACS_VERSION}.${NOMACS_VERSION_PATCH})
 
-# Codecov
-if(CMAKE_COMPILER_IS_GNUCXX)
-    include("cmake/CodeCoverage.cmake")
-    setup_target_for_coverage(${PROJECT_NAME}_coverage tests coverage)
-endif()
-
 if(CMAKE_CL_64)
 	SET(NMC_ARCHITECTURE "x64")
 else()
@@ -100,7 +94,7 @@
 
 if(CMAKE_COMPILER_IS_GNUCXX)
 	# Codecov
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -pedantic -pthread -g -O0 -fprofile-arcs -ftest-coverage")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -pedantic -pthread")
 endif()
 
 # find Qt
diff -Naur nomacs-3.8.0/ImageLounge/src/DkCore/DkBasicLoader.cpp nomacs-3.8.0-p/ImageLounge/src/DkCore/DkBasicLoader.cpp
--- nomacs-3.8.0/ImageLounge/src/DkCore/DkBasicLoader.cpp	2017-11-09 10:48:54.000000000 +0100
+++ nomacs-3.8.0-p/ImageLounge/src/DkCore/DkBasicLoader.cpp	2018-02-16 17:39:39.901386923 +0100
@@ -53,7 +53,7 @@
 
 // quazip
 #ifdef WITH_QUAZIP
-#include <quazip/JlCompress.h>
+#include <quazip5/JlCompress.h>
 #endif
 
 // opencv
diff -Naur nomacs-3.8.0/ImageLounge/src/DkCore/DkImageContainer.cpp nomacs-3.8.0-p/ImageLounge/src/DkCore/DkImageContainer.cpp
--- nomacs-3.8.0/ImageLounge/src/DkCore/DkImageContainer.cpp	2017-11-09 10:48:54.000000000 +0100
+++ nomacs-3.8.0-p/ImageLounge/src/DkCore/DkImageContainer.cpp	2018-02-16 17:39:39.898386912 +0100
@@ -41,7 +41,7 @@
 
 // quazip
 #ifdef WITH_QUAZIP
-#include <quazip/JlCompress.h>
+#include <quazip5/JlCompress.h>
 #endif
 #pragma warning(pop)		// no warnings from includes - end
 
diff -Naur nomacs-3.8.0/ImageLounge/src/DkCore/DkImageLoader.cpp nomacs-3.8.0-p/ImageLounge/src/DkCore/DkImageLoader.cpp
--- nomacs-3.8.0/ImageLounge/src/DkCore/DkImageLoader.cpp	2017-11-09 10:48:54.000000000 +0100
+++ nomacs-3.8.0-p/ImageLounge/src/DkCore/DkImageLoader.cpp	2018-02-16 17:39:39.896386905 +0100
@@ -76,7 +76,7 @@
 
 // quazip
 #ifdef WITH_QUAZIP
-#include <quazip/JlCompress.h>
+#include <quazip5/JlCompress.h>
 #endif
 
 // opencv
diff -Naur nomacs-3.8.0/ImageLounge/src/DkGui/DkDialog.cpp nomacs-3.8.0-p/ImageLounge/src/DkGui/DkDialog.cpp
--- nomacs-3.8.0/ImageLounge/src/DkGui/DkDialog.cpp	2017-11-09 10:48:54.000000000 +0100
+++ nomacs-3.8.0-p/ImageLounge/src/DkGui/DkDialog.cpp	2018-02-16 17:39:39.908386947 +0100
@@ -92,7 +92,7 @@
 #include <QPrinterInfo>
 // quazip
 #ifdef WITH_QUAZIP
-#include <quazip/JlCompress.h>
+#include <quazip5/JlCompress.h>
 #endif
 
 #pragma warning(pop)		// no warnings from includes - end
