diff -ruN a/ImageLounge/CMakeLists.txt b/ImageLounge/CMakeLists.txt
--- a/ImageLounge/CMakeLists.txt	2020-08-13 10:15:07.000000000 +0200
+++ b/ImageLounge/CMakeLists.txt	2020-10-06 00:47:58.624994682 +0200
@@ -188,26 +188,6 @@
 
 endif()
 
-if (Python_FOUND)
-    
-    add_custom_command(
-        TARGET ${DLL_CORE_NAME} PRE_BUILD 
-        COMMAND ${Python_EXECUTABLE}
-        ${CMAKE_CURRENT_SOURCE_DIR}/../scripts/versionupdate.py 
-        ${CMAKE_BINARY_DIR}/DkVersion.h)
-
-    # I am having issues with the PRE_BUILD above
-    # if version increment is off, we just need to run the update once (for version won't change)
-    # so let's do that:
-    execute_process(COMMAND ${Python_EXECUTABLE}
-            ${CMAKE_CURRENT_SOURCE_DIR}/../scripts/versionupdate.py 
-            ${CMAKE_BINARY_DIR}/DkVersion.h)
-
-else()
-    message(WARNING "Python not found: the version number will be incorrect!")
-endif()
-
-
 NMC_INSTALL()
 
 NMC_COPY_FILES()

