see https://trac.videolan.org/vlc/ticket/14908

diff -Naur vlc-2.2.2/modules/hw/vdpau/display.c vlc-2.2.2-p/modules/hw/vdpau/display.c
--- vlc-2.2.2/modules/hw/vdpau/display.c	2015-02-02 20:42:29.000000000 +0100
+++ vlc-2.2.2-p/modules/hw/vdpau/display.c	2016-04-08 11:39:25.762059918 +0200
@@ -156,10 +156,15 @@
     picture_pool_Delete(pool);
 }
 
+static bool force_fail;
+
 static picture_pool_t *Pool(vout_display_t *vd, unsigned requested_count)
 {
     vout_display_sys_t *sys = vd->sys;
 
+     if (force_fail)
+         return NULL;
+
     if (sys->pool == NULL)
         sys->pool = PoolAlloc(vd, requested_count);
     return sys->pool;
@@ -389,6 +394,7 @@
          || place.height != vd->fmt.i_visible_height)
         {
             vout_display_SendEventPicturesInvalid(vd);
+	    force_fail = true;
             return VLC_SUCCESS;
         }
 
