--- vlc-3.0.12/modules/access/live555.cpp	2019-03-29 20:01:15.000000000 +0100
+++ vlc-3.0.12.new/modules/access/live555.cpp	2021-01-21 09:34:09.898935050 +0100
@@ -852,7 +852,10 @@
             if( !p_sys->b_multicast )
             {
                 /* We need different rollover behaviour for multicast */
-                p_sys->b_multicast = IsMulticastAddress( sub->connectionEndpointAddress() );
+		struct sockaddr_storage connectionAddress;
+		sub->getConnectionEndpointAddress(connectionAddress);
+
+                p_sys->b_multicast = IsMulticastAddress(connectionAddress);
             }
 
             tk = (live_track_t*)malloc( sizeof( live_track_t ) );
