diff -Naur vlc-2.2.2/modules/access/rdp.c vlc-2.2.2-p/modules/access/rdp.c
--- vlc-2.2.2/modules/access/rdp.c	2015-04-30 15:45:33.000000000 +0200
+++ vlc-2.2.2-p/modules/access/rdp.c	2016-04-06 21:20:09.021938399 +0200
@@ -46,7 +46,7 @@
 #endif
 
 #if !defined(FREERDP_VERSION_MAJOR) || \
-    (defined(FREERDP_VERSION_MAJOR) && !(FREERDP_VERSION_MAJOR >= 1 && FREERDP_VERSION_MINOR >= 1 ))
+    (defined(FREERDP_VERSION_MAJOR) && !(FREERDP_VERSION_MAJOR > 1 || (FREERDP_VERSION_MAJOR == 1 && FREERDP_VERSION_MINOR >= 1)))
 # define SoftwareGdi sw_gdi
 # define Fullscreen fullscreen
 # define ServerHostname hostname
@@ -233,7 +233,7 @@
     vlcrdp_context_t * p_vlccontext = (vlcrdp_context_t *) p_instance->context;
 
     msg_Dbg( p_vlccontext->p_demux, "connected to desktop %dx%d (%d bpp)",
-#if (FREERDP_VERSION_MAJOR >= 1 && FREERDP_VERSION_MINOR >= 1 )
+#if defined(FREERDP_VERSION_MAJOR) && (FREERDP_VERSION_MAJOR > 1 || (FREERDP_VERSION_MAJOR == 1 && FREERDP_MINOR_VERSION >= 1))
              p_instance->settings->DesktopWidth,
              p_instance->settings->DesktopHeight,
              p_instance->settings->ColorDepth
