diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 7e815c7..d40acb3 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -70,19 +70,7 @@ else (NOT XSLTPROC_EXECUTABLE)
   message (STATUS "Looking for xsltproc... ${XSLTPROC_EXECUTABLE}")
 endif (NOT XSLTPROC_EXECUTABLE)
 
-message (STATUS "Looking for gpgme...")
-find_library (GPGME gpgme)
-if (NOT GPGME)
-  message (SEND_ERROR "The gpgme library is required.")
-else (NOT GPGME)
-  message (STATUS "Looking for gpgme... ${GPGME}")
-  execute_process (COMMAND gpgme-config --cflags
-    OUTPUT_VARIABLE GPGME_CFLAGS
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-  execute_process (COMMAND gpgme-config --libs
-    OUTPUT_VARIABLE GPGME_LDFLAGS
-    OUTPUT_STRIP_TRAILING_WHITESPACE)
-endif (NOT GPGME)
+pkg_check_modules (GPGME REQUIRED gpgme)
 
 if (WITH_LIBTHEIA)
   find_package(Theia 1.0.0 REQUIRED)

