diff -aur a/marave/editor/highlight/source-highlight-qt.sip b/marave/editor/highlight/source-highlight-qt.sip
--- a/marave/editor/highlight/source-highlight-qt.sip	2010-02-16 20:32:54.000000000 -0600
+++ b/marave/editor/highlight/source-highlight-qt.sip	2011-05-28 09:38:06.050998705 -0500
@@ -27,7 +27,7 @@
     void setDocument ( QTextDocument * doc );
     void rehighlightBlock ( const QTextBlock & block );
     const QString getLangDefFileFromFileName(const QString &filename);
-    void setDefaultToMonosapce(bool);
+    void setDefaultToMonospace(bool);
 private:
    Qt4SyntaxHighlighter(const srchiliteqt::Qt4SyntaxHighlighter&);
 
diff -aur a/marave/editor/spelltextedit.py b/marave/editor/spelltextedit.py
--- a/marave/editor/spelltextedit.py	2010-02-23 17:31:16.000000000 -0600
+++ b/marave/editor/spelltextedit.py	2011-05-28 09:38:11.926000460 -0500
@@ -208,7 +208,7 @@
                     if not self.dict and srchiliteqt:
                         self.highlighter.setDocument(None)
                         self.highlighter=srchiliteqt.Qt4SyntaxHighlighter(self.document())
-                        self.highlighter.setDefaultToMonosapce(False)
+                        self.highlighter.setDefaultToMonospace(False)
                         langName=self.highlighter.getLangDefFileFromFileName(fname)
                         if langName:
                             self.langChanged.emit(langName)
@@ -230,7 +230,7 @@
         if isinstance(self.highlighter, SpellHighlighter):
             self.killDict()
             self.highlighter=srchiliteqt.Qt4SyntaxHighlighter(self.document())
-            self.highlighter.setDefaultToMonosapce(False)
+            self.highlighter.setDefaultToMonospace(False)
         self.highlighter.setDocument(self.document())
         self.highlighter.init(lang)
         self.highlighter.setFormattingStyle(style)
