From 46f9e3e9a03f4531d5337d1e324dd4bc56e832d9 Mon Sep 17 00:00:00 2001
From: Oleg Grenrus <oleg.grenrus@iki.fi>
Date: Thu, 26 Mar 2020 00:52:06 +0200
Subject: [PATCH] Allow base-4.14

---
 .travis.yml | 111 ++++++++++++++++++++++++----------------------------
 assoc.cabal |   6 ++-
 2 files changed, 55 insertions(+), 62 deletions(-)

diff --git a/assoc.cabal b/assoc.cabal
index ec103d3..da1af6d 100644
--- a/assoc.cabal
+++ b/assoc.cabal
@@ -1,6 +1,7 @@
 cabal-version:      1.12
 name:               assoc
 version:            1.0.1
+x-revision:         1
 license:            BSD3
 license-file:       LICENSE
 synopsis:           swap and assoc: Symmetric and Semigroupy Bifunctors
@@ -27,7 +28,8 @@ tested-with:
      || ==8.2.2
      || ==8.4.4
      || ==8.6.5
-     || ==8.8.1
+     || ==8.8.3
+     || ==8.10.1
   , GHCJS ==8.4
 
 source-repository head
@@ -38,7 +40,7 @@ library
   default-language: Haskell2010
   hs-source-dirs:   src
   build-depends:
-      base        >=4.3   && <4.14
+      base        >=4.3   && <4.15
     , bifunctors  >=5.5.5 && <5.6
     , tagged      >=0.8.6 && <0.9
 
