diff -Naur audacious-0.1.orig/Plugins/Input/mikmod/plugin.c audacious-0.1/Plugins/Input/mikmod/plugin.c
--- audacious-0.1.orig/Plugins/Input/mikmod/plugin.c	2005-10-26 07:53:38.000000000 +0200
+++ audacious-0.1/Plugins/Input/mikmod/plugin.c	2005-11-16 14:25:46.000000000 +0100
@@ -1,4 +1,3 @@
-
 /*
    libmikmod for xmms
 
@@ -70,7 +69,6 @@
 static void *play_loop(void *arg);
 static void config_ok(GtkWidget * widget, gpointer data);
 
-
 static void aboutbox()
 {
 	GtkWidget *dialog_vbox1;
@@ -421,7 +419,7 @@
 
 	if (!mikmod_conf_window)
 	{
-		mikmod_conf_window = gtk_window_new(GTK_WINDOW_DIALOG);
+		mikmod_conf_window = gtk_window_new(GTK_WINDOW_POPUP);
 		gtk_object_set_data(GTK_OBJECT(mikmod_conf_window), "mikmod_conf_window", mikmod_conf_window);
 		gtk_window_set_title(GTK_WINDOW(mikmod_conf_window), _("MikMod Configuration"));
 		gtk_window_set_policy(GTK_WINDOW(mikmod_conf_window), FALSE, FALSE, FALSE);
