diff -Naur audacious-0.1.orig/audacious/audacious.desktop.in audacious-0.1/audacious/audacious.desktop.in
--- audacious-0.1.orig/audacious/audacious.desktop.in	2005-10-24 12:06:47.000000000 +0200
+++ audacious-0.1/audacious/audacious.desktop.in	2005-11-16 13:29:54.000000000 +0100
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Name=Audacious
 Comment=Play music
+Comment[hu]=Zene lejátszása
 Exec=audacious
 Icon=audacious.png
 MimeType=audio/x-scpls;audio/x-mpegurl;audio/mpegurl;audio/mp3;audio/x-mp3;audio/mpeg;audio/x-mpeg;audio/x-wav;application/x-ogg;
