diff -urN gdm-2.14.1/config/gdm.conf.in gdm-2.14.1.new/config/gdm.conf.in
--- gdm-2.14.1/config/gdm.conf.in	2006-04-07 20:47:20.000000000 +0100
+++ gdm-2.14.1.new/config/gdm.conf.in	2006-04-13 09:37:26.000000000 +0100
@@ -61,7 +61,7 @@
 
 # The greeter for local (non-xdmcp) logins.  Change gdmlogin to gdmgreeter to
 # get the new graphical greeter.
-#Greeter=@libexecdir@/gdmlogin
+Greeter=@libexecdir@/gdmgreeter
 
 # The greeter for xdmcp logins, usually you want a less graphically intensive
 # greeter here so it's better to leave this with gdmlogin
@@ -92,7 +92,7 @@
 # User and group used for running GDM GUI applicaitons.  By default this is set
 # to user "gdm" and group "gdm".  This user/group should have very limited
 # permissions and access to ony the gdm directories and files.
-User=gdm
+User=nobody
 Group=gdm
 
 # To try to kill all clients started at greeter time or in the Init script.
@@ -131,7 +131,7 @@
 # really a PATH style variable since 2.4.4.2 to allow actual interoperability
 # with KDM.  Note that <dmconfdir>/Sessions is there for backwards
 # compatibility reasons with 2.4.4.x.
-#SessionDesktopDir=/etc/X11/sessions/:@dmconfdir@/Sessions/:@datadir@/gdm/BuiltInSessions/:@datadir@/xsessions/
+SessionDesktopDir=/etc/X11/sessions/:@datadir@/gdm/BuiltInSessions
 # This is the default .desktop session.  One of the ones in SessionDesktopDir
 #DefaultSession=gnome.desktop
 # Better leave this blank and HOME will be used.  You can use syntax ~/ below
@@ -159,8 +159,8 @@
 # Automatic VT allocation.  Right now only works on Linux.  This way we force
 # X to use specific vts.  turn VTAllocation to false if this is causing
 # problems.
-#FirstVT=7
-#VTAllocation=true
+FirstVT=12
+VTAllocation=true
 # Should double login be treated with a warning (and possibility to change VT's
 # on Linux and FreeBSD systems for console logins)
 #DoubleLoginWarning=true
@@ -199,7 +199,7 @@
 [security]
 # Allow root to login.  It makes sense to turn this off for kiosk use, when
 # you want to minimize the possibility of break in.
-AllowRoot=true
+AllowRoot=false
 # Allow login as root via XDMCP.  This value will be overridden and set to
 # false if the /etc/default/login file exists and contains
 # "CONSOLE=/dev/login", and set to true if the /etc/default/login file exists
@@ -286,7 +286,7 @@
 #GtkRC=@datadir@/themes/Default/gtk-2.0/gtkrc
 
 # The GTK+ theme to use for the GUI.
-#GtkTheme=Default
+GtkTheme=Clearlooks
 # If to allow changing the GTK+ (widget) theme from the greeter.  Currently
 # this only affects the standard greeter as the graphical greeter does not yet
 # have this ability.
@@ -427,10 +427,10 @@
 # where the session saving stuff is disabled in GDM
 #ShowLastSession=true
 # Always use 24 hour clock no matter what the locale.
-#Use24Clock=auto
+Use24Clock=auto
 # Use circles in the password field.  Looks kind of cool actually, but only
 # works with certain fonts.
-#UseCirclesInEntry=false
+UseCirclesInEntry=true
 # Do not show any visible feedback in the password field. This is standard for
 # instance in console, xdm and ssh.
 #UseInvisibleInEntry=false
@@ -440,7 +440,7 @@
 # list then provide a list that is delimited by /: to the GraphicalThemes
 # key and set GraphicalThemeRand to true.  Otherwise use GraphicalTheme
 # and specify just one theme.
-#GraphicalTheme=circles
+GraphicalTheme=Frugalware
 #GraphicalThemes=circles/:happygnome
 GraphicalThemeDir=@datadir@/gdm/themes/
 GraphicalThemeRand=false
@@ -456,7 +456,7 @@
 # user input.  If SoundOnLogin is a file and the greeter finds the 'play'
 # executable (see daemon/SoundProgram) it will play that file instead of just
 # beeping.
-#SoundOnLogin=true
+SoundOnLogin=false
 #SoundOnLoginFile=
 # If SoundOnLoginSuccess, then the greeter will play a sound (as above) when a
 # user successfully logs in.
