diff -urN gdm-2.14.1/gui/gdmsetup.desktop.in gdm-2.14.1.new/gui/gdmsetup.desktop.in
--- gdm-2.14.1/gui/gdmsetup.desktop.in.in	2006-03-29 00:43:23.000000000 +0100
+++ gdm-2.14.1.new/gui/gdmsetup.desktop.in.in	2006-04-13 09:45:07.000000000 +0100
@@ -4,7 +4,7 @@
 _GenericName=Login Window Preferences
 _Comment=Configure the login window (GNOME Display Manager)
 TryExec=@sbindir@/gdmsetup
-Exec=gdmsetup
+Exec=gksu gdmsetup
 Icon=gdm-setup
 StartupNotify=true
 Terminal=false
