diff -Naur gdm-2.20.0.orig/configure.ac gdm-2.20.0/configure.ac
--- gdm-2.20.0.orig/configure.ac	2007-09-25 18:06:41.000000000 +0200
+++ gdm-2.20.0/configure.ac	2007-09-25 18:09:30.000000000 +0200
@@ -1100,8 +1100,8 @@
 	;;
 *)
 	HOST_GDM_USER_PATH="/bin:/usr/bin"
-	HOST_HALT_COMMAND="/usr/bin/poweroff;/sbin/poweroff;/sbin/shutdown -h now;/usr/sbin/shutdown -h now"
-	HOST_REBOOT_COMMAND="/usr/bin/reboot;/sbin/reboot;/sbin/shutdown -r now;/usr/sbin/shutdown -r now"
+	HOST_HALT_COMMAND="/sbin/poweroff;/sbin/shutdown -h now;/usr/sbin/shutdown -h now"
+	HOST_REBOOT_COMMAND="/sbin/reboot;/sbin/shutdown -r now;/usr/sbin/shutdown -r now"
 	HOST_SUSPEND_COMMAND=""
 	HOST_XSESSION_SHELL=/bin/sh
 	HOST_SOUND_PROGRAM=/usr/bin/play
