diff -Naur gdm-2.20.3/gui/gdmsetup.desktop.in.in gdm-2.20.3-p/gui/gdmsetup.desktop.in.in
--- gdm-2.20.3/gui/gdmsetup.desktop.in.in	2008-01-07 20:39:05.000000000 +0100
+++ gdm-2.20.3-p/gui/gdmsetup.desktop.in.in	2008-01-23 19:03:42.000000000 +0100
@@ -2,8 +2,8 @@
 Encoding=UTF-8
 _Name=Login Window
 _Comment=Configure GDM login window appearance and behavior
-TryExec=@sbindir@/gdmsetup
-Exec=@sbindir@/gdmsetup
+TryExec=gksu @sbindir@/gdmsetup
+Exec=gksu @sbindir@/gdmsetup
 Icon=gdmsetup
 StartupNotify=true
 Terminal=false
