diff -urN gdm-2.18.1.orig/gui/gdmphotosetup.desktop.in gdm-2.18.1/gui/gdmphotosetup.desktop.in
--- gdm-2.18.1.orig/gui/gdmphotosetup.desktop.in	2007-04-15 10:44:07.000000000 +0100
+++ gdm-2.18.1/gui/gdmphotosetup.desktop.in	2007-04-15 10:44:45.000000000 +0100
@@ -9,6 +9,7 @@
 StartupNotify=true
 Terminal=false
 Type=Application
+NotShowIn=GNOME;
 Categories=GNOME;GTK;Settings;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gdm
