diff -urN gdm-2.15.9/config/gdm.conf.in gdm-2.15.9.new/config/gdm.conf.in
--- gdm-2.15.9/config/gdm.conf.in	2006-08-03 18:36:26.000000000 +0100
+++ gdm-2.15.9.new/config/gdm.conf.in	2006-08-19 08:05:28.000000000 +0100
@@ -67,7 +67,7 @@
 
 # The greeter for local (non-xdmcp) logins.  Change gdmlogin to gdmgreeter to
 # get the new graphical greeter.
-#Greeter=@libexecdir@/gdmlogin
+Greeter=@libexecdir@/gdmgreeter
 
 # The greeter for xdmcp logins, usually you want a less graphically intensive
 # greeter here so it's better to leave this with gdmlogin
@@ -98,7 +98,7 @@
 # User and group used for running GDM GUI applicaitons.  By default this is set
 # to user "gdm" and group "gdm".  This user/group should have very limited
 # permissions and access to ony the gdm directories and files.
-User=gdm
+User=nobody
 Group=gdm
 
 # To try to kill all clients started at greeter time or in the Init script.
@@ -137,7 +137,7 @@
 # really a PATH style variable since 2.4.4.2 to allow actual interoperability
 # with KDM.  Note that <dmconfdir>/Sessions is there for backwards
 # compatibility reasons with 2.4.4.x.
-#SessionDesktopDir=/etc/X11/sessions/:@dmconfdir@/Sessions/:@datadir@/gdm/BuiltInSessions/:@datadir@/xsessions/
+SessionDesktopDir=/etc/X11/sessions/:@datadir@/gdm/BuiltInSessions/:@datadir@/xsessions/
 # This is the default .desktop session.  One of the ones in SessionDesktopDir
 #DefaultSession=gnome.desktop
 # Better leave this blank and HOME will be used.  You can use syntax ~/ below
@@ -165,8 +165,8 @@
 # Automatic VT allocation.  Right now only works on Linux.  This way we force
 # X to use specific vts.  turn VTAllocation to false if this is causing
 # problems.
-#FirstVT=7
-#VTAllocation=true
+FirstVT=12
+VTAllocation=true
 # Should double login be treated with a warning (and possibility to change VT's
 # on Linux and FreeBSD systems for console logins)
 #DoubleLoginWarning=true
@@ -205,7 +205,7 @@
 [security]
 # Allow root to login.  It makes sense to turn this off for kiosk use, when
 # you want to minimize the possibility of break in.
-AllowRoot=true
+AllowRoot=false
 # Allow login as root via XDMCP.  This value will be overridden and set to
 # false if the /etc/default/login file exists and contains
 # "CONSOLE=/dev/login", and set to true if the /etc/default/login file exists
@@ -294,7 +294,7 @@
 #GtkRC=@datadir@/themes/Default/gtk-2.0/gtkrc
 
 # The GTK+ theme to use for the GUI.
-#GtkTheme=Default
+GtkTheme=Clearlooks
 # If to allow changing the GTK+ (widget) theme from the greeter.  Currently
 # this only affects the standard greeter as the graphical greeter does not yet
 # have this ability.
@@ -441,10 +441,10 @@
 # where the session saving stuff is disabled in GDM
 #ShowLastSession=true
 # Always use 24 hour clock no matter what the locale.
-#Use24Clock=auto
+Use24Clock=auto
 # Use circles in the password field.  Looks kind of cool actually, but only
 # works with certain fonts.
-#UseCirclesInEntry=false
+UseCirclesInEntry=true
 # Do not show any visible feedback in the password field. This is standard for
 # instance in console, xdm and ssh.
 #UseInvisibleInEntry=false
@@ -454,7 +454,7 @@
 # list then provide a list that is delimited by /: to the GraphicalThemes
 # key and set GraphicalThemeRand to true.  Otherwise use GraphicalTheme
 # and specify just one theme.
-#GraphicalTheme=circles
+GraphicalTheme=Frugalware
 #GraphicalThemes=circles/:happygnome
 GraphicalThemeDir=@datadir@/gdm/themes/
 GraphicalThemeRand=false
@@ -470,7 +470,7 @@
 # user input.  If SoundOnLogin is a file and the greeter finds the 'play'
 # executable (see daemon/SoundProgram) it will play that file instead of just
 # beeping.
-#SoundOnLogin=true
+SoundOnLogin=false
 #SoundOnLoginFile=
 # If SoundOnLoginSuccess, then the greeter will play a sound (as above) when a
 # user successfully logs in.
@@ -558,7 +558,7 @@
 # Definition of the standard X server.
 [server-Standard]
 name=Standard server
-command=@X_SERVER@ @X_CONFIG_OPTIONS@ @XEVIE_OPTION@
+command=@X_SERVER@ @X_CONFIG_OPTIONS@ @XEVIE_OPTION@ -dpi 96
 flexible=true
 # Indicates that the X server should be started at a different process
 # priority.  Values can be any integer value accepted by the setpriority C
@@ -573,7 +573,7 @@
 [server-Terminal]
 name=Terminal server
 # Add -terminate to make things behave more nicely
-command=@X_SERVER@ @X_CONFIG_OPTIONS@ -terminate
+command=@X_SERVER@ @X_CONFIG_OPTIONS@ -terminate -dpi 96
 # Make this not appear in the flexible servers (we need extra params anyway,
 # and terminate would be bad for xdmcp choosing).  You can make a terminal
 # server flexible, but not with an indirect query.  If you need flexible
@@ -587,7 +587,7 @@
 # command line.
 [server-Chooser]
 name=Chooser server
-command=@X_SERVER@ @X_CONFIG_OPTIONS@
+command=@X_SERVER@ @X_CONFIG_OPTIONS@ -dpi 96
 # Make this not appear in the flexible servers for now, but if you wish to
 # allow a chooser server then make this true.  This is the only way to make a
 # flexible chooser server that behaves nicely.
