diff -urN gdm-2.18.0.orig/gui/gdmsetup.desktop.in.in gdm-2.18.0/gui/gdmsetup.desktop.in.in
--- gdm-2.18.0.orig/gui/gdmsetup.desktop.in.in	2007-03-14 16:33:17.000000000 +0000
+++ gdm-2.18.0/gui/gdmsetup.desktop.in.in	2007-03-14 16:34:53.000000000 +0000
@@ -2,8 +2,8 @@
 Encoding=UTF-8
 _Name=Login Window
 _Comment=Configure GDM login window appearance and behavior
-TryExec=@sbindir@/gdmsetup
-Exec=@sbindir@/gdmsetup
+TryExec=gksu @sbindir@/gdmsetup
+Exec=gksu @sbindir@/gdmsetup
 Icon=gdm-setup
 StartupNotify=true
 Terminal=false
