diff --git a/data/gdm-shell.session.in b/data/gdm-shell.session.in
index 496cc10..172f5e0 100644
--- a/data/gdm-shell.session.in
+++ b/data/gdm-shell.session.in
@@ -1,5 +1,5 @@
 [GNOME Session]
 Name=Display Manager
 RequiredComponents=gnome-shell;gnome-settings-daemon;
-IsRunnableHelper=bash -c 'gnome-shell --help | grep -q gdm-mode && @libexecdir@/gnome-session-check-accelerated'
+IsRunnableHelper=bash -c 'gnome-shell --help | grep -q gdm && @libexecdir@/gnome-session-check-accelerated'
 FallbackSession=gdm-fallback
