diff -ur a/data/gdm b/data/gdm
--- a/data/gdm	2012-03-27 16:01:25.988142525 -0500
+++ b/data/gdm	2012-03-27 16:01:50.430141854 -0500
@@ -8,5 +8,4 @@
 password   include     system-auth
 session    optional    pam_keyinit.so force revoke
 session    include     system-auth
-session    required    pam_loginuid.so
 session    optional    pam_console.so
diff -ur a/data/gdm-autologin b/data/gdm-autologin
--- a/data/gdm-autologin	2012-03-27 16:01:25.988142525 -0500
+++ b/data/gdm-autologin	2012-03-27 16:02:20.674143148 -0500
@@ -6,5 +6,4 @@
 password   include     system-auth
 session    optional    pam_keyinit.so force revoke
 session    include     system-auth
-session    required    pam_loginuid.so
 session    optional    pam_console.so
diff -ur a/data/gdm-welcome b/data/gdm-welcome
--- a/data/gdm-welcome	2012-03-27 16:01:25.988142525 -0500
+++ b/data/gdm-welcome	2012-03-27 16:02:38.334140889 -0500
@@ -4,6 +4,5 @@
 account    required    pam_nologin.so
 account    include     system-auth
 password   include     system-auth
-session    required    pam_loginuid.so
 session    optional    pam_keyinit.so force revoke
 session    include     system-auth
