From 419836c1a411bd541594363ffce7f508fa7c02f7 Mon Sep 17 00:00:00 2001
From: Rodrigo Moya <rodrigo@gnome-db.org>
Date: Wed, 27 Jul 2011 09:14:29 +0000
Subject: greeter: Use the new power DBus interface in g-s-d

https://bugzilla.gnome.org/show_bug.cgi?id=655399
---
diff --git a/gui/simple-greeter/gdm-greeter-panel.c b/gui/simple-greeter/gdm-greeter-panel.c
index f3f192a..598aadf 100644
--- a/gui/simple-greeter/gdm-greeter-panel.c
+++ b/gui/simple-greeter/gdm-greeter-panel.c
@@ -57,9 +57,9 @@
 #define CK_MANAGER_PATH      "/org/freedesktop/ConsoleKit/Manager"
 #define CK_MANAGER_INTERFACE "org.freedesktop.ConsoleKit.Manager"
 
-#define GPM_DBUS_NAME      "org.gnome.PowerManager"
-#define GPM_DBUS_PATH      "/org/gnome/PowerManager"
-#define GPM_DBUS_INTERFACE "org.gnome.PowerManager"
+#define GPM_DBUS_NAME      "org.gnome.SettingsDaemon"
+#define GPM_DBUS_PATH      "/org/gnome/SettingsDaemon/Power"
+#define GPM_DBUS_INTERFACE "org.gnome.SettingsDaemon.Power"
 
 #define KEY_DISABLE_RESTART_BUTTONS "/apps/gdm/simple-greeter/disable_restart_buttons"
 #define KEY_NOTIFICATION_AREA_PADDING "/apps/notification_area_applet/prefs/padding"
--
cgit v0.9.0.2

