diff -Naur lxsplit-0.1.1/Makefile lxsplit-0.1.1.new/Makefile
--- lxsplit-0.1.1/Makefile	2001-08-16 17:14:48.000000000 +0530
+++ lxsplit-0.1.1.new/Makefile	2008-03-15 11:27:53.000000000 +0530
@@ -5,11 +5,11 @@
 ####### Compiler, tools and options
 
 CC	=	gcc
-CFLAGS	=	-W -O2 -DNO_DEBUG
+#CFLAGS	=	-W -O2 -DNO_DEBUG
 LFLAGS	=	
 
 # This is where the lxsplit executable will be placed
-INSTALL_PATH = /usr/local/bin
+INSTALL_PATH = /usr/bin
 
 ####### Files
 
