diff -Naur lxsplit-0.1.1/func.c lxsplit-0.1.1.new/func.c
--- lxsplit-0.1.1/func.c	2001-08-16 17:01:15.000000000 +0530
+++ lxsplit-0.1.1.new/func.c	2008-03-15 11:30:07.000000000 +0530
@@ -1,4 +1,5 @@
 #include <stdlib.h>
+#include <string.h>
 #include <unistd.h>
 #include <stdio.h>
 
diff -Naur lxsplit-0.1.1/split.c lxsplit-0.1.1.new/split.c
--- lxsplit-0.1.1/split.c	2001-08-16 16:52:20.000000000 +0530
+++ lxsplit-0.1.1.new/split.c	2008-03-15 11:30:29.000000000 +0530
@@ -2,6 +2,8 @@
 #include <sys/stat.h>
 #include <unistd.h>
 #include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
 #include "func.h"
 
 int split_files( char *filename , unsigned long int _s)
