--- totem-2.30.0/configure.old	2010-03-29 16:13:34.000000000 +0000
+++ totem-2.30.0/configure	2010-04-23 21:46:06.000000000 +0000
@@ -18407,22 +18407,7 @@
 	fi
 done
 
-for good_element in gconfaudiosink gconfvideosink goom
-do
-	{ $as_echo "$as_me:${as_lineno-$LINENO}: checking GStreamer 0.10 $good_element plugin" >&5
-$as_echo_n "checking GStreamer 0.10 $good_element plugin... " >&6; }
-	if $gst010_inspect $good_element >/dev/null 2>/dev/null; then
-		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-$as_echo "yes" >&6; }
-	else
-		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: no" >&5
-$as_echo "no" >&6; }
-		as_fn_error "
-			Cannot find required GStreamer-0.10 plugin '$good_element'.
-			It should be part of gst-plugins-good. Please install it.
-		" "$LINENO" 5
-	fi
-done
+
 
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking Whether not to check for iso-codes" >&5
 $as_echo_n "checking Whether not to check for iso-codes... " >&6; }
