--- OpenRGB-release_0.8/OpenRGB.pro	2022-11-28 04:44:33.000000000 +0100
+++ OpenRGB-release_0.8.new/OpenRGB.pro	2022-12-08 15:06:17.109106204 +0100
@@ -1621,7 +1621,7 @@
     CONFIG(release, debug|release) {
         udev_rules.CONFIG       = no_check_exist
         udev_rules.target       = 60-openrgb.rules
-        udev_rules.path         = $$PREFIX/lib/udev/rules.d/
+        udev_rules.path         = /lib/udev/rules.d/
 
         exists($$udev_rules.target) {
             message($$udev_rules.target " - UDEV rules file exists. Removing from build")
@@ -1732,7 +1732,7 @@
     icon.files+=qt/OpenRGB.png
     metainfo.path=$$PREFIX/share/metainfo/
     metainfo.files+=qt/org.openrgb.OpenRGB.metainfo.xml
-    rules.path=$$PREFIX/lib/udev/rules.d/
+    rules.path=/lib/udev/rules.d/
     rules.files+=60-openrgb.rules
     INSTALLS += target desktop icon metainfo rules
 }
