diff -Naur unixODBC-2.2.14/exe/odbc-config.c unixODBC-2.2.14-64bit-p/exe/odbc-config.c
--- unixODBC-2.2.14/exe/odbc-config.c	2008-05-20 14:58:17.000000000 +0200
+++ unixODBC-2.2.14-64bit-p/exe/odbc-config.c	2009-03-01 14:12:11.000000000 +0100
@@ -66,12 +66,12 @@
     printf( "#ifndef HAVE_LONG_LONG\n #define HAVE_LONG_LONG\n#endif\n" );
 #endif
 
-#ifdef ODBCINT64
-    printf( "#ifndef ODBCINT64\n #define ODBCINT64\n#endif\n" );
+#ifdef ODBCINT64_TYPE
+    printf( "#ifndef ODBCINT64\n #define ODBCINT64 %s\n#endif\n", ODBCINT64_TYPE );
 #endif
 
-#ifdef UODBCINT64
-    printf( "#ifndef UODBCINT64\n #define UODBCINT64\n#endif\n" );
+#ifdef UODBCINT64_TYPE
+    printf( "#ifndef UODBCINT64\n #define UODBCINT64 %s\n#endif\n", UODBCINT64_TYPE );
 #endif
 
 #ifdef DISABLE_INI_CACHING
diff -Naur unixODBC-2.2.14/include/sqltypes.h unixODBC-2.2.14-64bit-p/include/sqltypes.h
--- unixODBC-2.2.14/include/sqltypes.h	2008-10-31 17:21:53.000000000 +0100
+++ unixODBC-2.2.14-64bit-p/include/sqltypes.h	2009-03-01 14:12:14.000000000 +0100
@@ -49,7 +49,7 @@
  */
 
 #ifndef SIZEOF_LONG_INT
-#include <unixodbc_conf.h>
+#include "unixodbc_conf.h"
 #endif
 
 #ifndef SIZEOF_LONG_INT
@@ -371,10 +371,14 @@
 # if (SIZEOF_LONG_INT == 8)
 #   define ODBCINT64	    long
 #   define UODBCINT64	unsigned long
+#   define ODBCINT64_TYPE	    "long"
+#   define UODBCINT64_TYPE	"unsigned long"
 # else
 #  ifdef HAVE_LONG_LONG
 #   define ODBCINT64	    long long
 #   define UODBCINT64	unsigned long long
+#   define ODBCINT64_TYPE	    "long long"
+#   define UODBCINT64_TYPE	"unsigned long long"
 #  else
 /*
  * may fail in some cases, but what else can we do ?
@@ -391,6 +395,8 @@
 };
 #   define ODBCINT64	    struct __bigint_struct
 #   define UODBCINT64	struct __bigint_struct_u
+#   define ODBCINT64_TYPE	    "struct __bigint_struct"
+#   define UODBCINT64_TYPE	"struct __bigint_struct_u"
 #  endif
 # endif
 #endif
