diff -Naur BitTorrent-5.0.7.orig/BitTorrent/launchmanycore.py BitTorrent-5.0.7/BitTorrent/launchmanycore.py
--- BitTorrent-5.0.7.orig/BitTorrent/launchmanycore.py	2007-03-02 02:12:50.000000000 +0100
+++ BitTorrent-5.0.7/BitTorrent/launchmanycore.py	2007-05-26 17:44:20.000000000 +0200
@@ -340,7 +340,7 @@
     def determine_filename(self, infohash):
         # THIS FUNCTION IS PARTICULARLY CONVOLUTED. BLECH! --Dave
         path, metainfo = self.torrent_cache[infohash]
-        path = efs2(path)
+        path = efs2(path.decode())
         name = metainfo.name_fs
         savein = efs2(self.config['save_in'])
         isdir = metainfo.is_batch
