diff -Naur BitTorrent-5.0.9/BitTorrent/GUI_wx/__init__.py BitTorrent-5.0.9-p/BitTorrent/GUI_wx/__init__.py
--- BitTorrent-5.0.9/BitTorrent/GUI_wx/__init__.py	2007-08-04 01:55:42.000000000 +0200
+++ BitTorrent-5.0.9-p/BitTorrent/GUI_wx/__init__.py	2007-12-05 17:29:36.000000000 +0100
@@ -22,7 +22,7 @@
 else:
     # doesn't work in py2exe
     try:
-        wxversion.select('2.6')
+        wxversion.select('2.8')
     except:
         pass
     
