--- exiv2-0.28.1/src/CMakeLists.txt	2023-11-06 13:17:09.000000000 +0100
+++ exiv2-0.28.1.new/src/CMakeLists.txt	2023-11-07 15:01:03.801330095 +0100
@@ -284,7 +284,7 @@
 configure_package_config_file(
   ../cmake/exiv2Config.cmake.in
   ${CMAKE_CURRENT_BINARY_DIR}/exiv2Config.cmake
-  INSTALL_DESTINATION "${CMAKE_INSTALL_DATADIR}/cmake/exiv2"
+  INSTALL_DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/exiv2"
 )
 
 install(FILES
@@ -294,7 +294,7 @@
     DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/exiv2)
 
 install(EXPORT exiv2Export
-    DESTINATION "${CMAKE_INSTALL_DATADIR}/cmake/exiv2"
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/exiv2"
     NAMESPACE Exiv2::
 )
 
@@ -302,5 +302,5 @@
     FILES
         ${CMAKE_CURRENT_BINARY_DIR}/exiv2ConfigVersion.cmake
         ${CMAKE_CURRENT_BINARY_DIR}/exiv2Config.cmake
-    DESTINATION "${CMAKE_INSTALL_DATADIR}/cmake/exiv2")
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/cmake/exiv2")
 
