diff -Naur bmpx-0.40.3/src/mb-tagger.hh bmpx-0.40.3-p/src/mb-tagger.hh
--- bmpx-0.40.3/src/mb-tagger.hh	2007-09-08 09:38:35.000000000 +0200
+++ bmpx-0.40.3-p/src/mb-tagger.hh	2007-09-10 11:10:27.000000000 +0200
@@ -48,7 +48,6 @@
 #include "library.hh"
 #include "musicbrainz/mbxml-v2.hh"
 #include "bmp/ui-types.hh"
-#include "lastfm-types.hh"
 
 using namespace Glib;
 
diff -Naur bmpx-0.40.3/src/shell.cc bmpx-0.40.3-p/src/shell.cc
--- bmpx-0.40.3/src/shell.cc	2007-09-08 09:38:35.000000000 +0200
+++ bmpx-0.40.3-p/src/shell.cc	2007-09-10 11:09:47.000000000 +0200
@@ -152,10 +152,6 @@
 #define PLAYER_SHELL_ACTION_BOOKMARKS         "player-shell-action-bookmarks"
 #define PLAYER_SHELL_ACTION_CREATE_BOOKMARK   "player-shell-action-create-bookmark"
 
-namespace
-{
-	const Bmp::DB::Variant dummy = double (2);
-}
 
 namespace Bmp
 {
diff -Naur bmpx-0.40.3/src/ui-part-jamendo.cc bmpx-0.40.3-p/src/ui-part-jamendo.cc
--- bmpx-0.40.3/src/ui-part-jamendo.cc	2007-09-08 09:38:35.000000000 +0200
+++ bmpx-0.40.3-p/src/ui-part-jamendo.cc	2007-09-10 11:10:49.000000000 +0200
@@ -48,10 +48,6 @@
 #define JAMENDO_ACTION_UI_PART_JAMENDO  "jamendo-action-ui-part-jamendo"
 #define JAMENDO_ACTION_UPDATE_DB        "jamendo-action-update-db"
 
-namespace
-{
-	const Bmp::DB::Variant dummy = double (2);
-}
 
 namespace Bmp
 {
