Index: configure.ac
===================================================================
--- configure.ac	(Revision 11299)
+++ configure.ac	(Arbeitskopie)
@@ -835,6 +835,7 @@
 echo 
 echo "    C compiler flags.............: $CFLAGS"
 echo "    C++ compiler flags...........: $CXXFLAGS"
+echo "    Linker flags.................: $LDFLAGS"
 echo
 echo "    Documentation generation.....: $generate_documentation"
 echo
