Index: jnetlib/Makefile.am
===================================================================
--- jnetlib/Makefile.am	(Revision 11299)
+++ jnetlib/Makefile.am	(Arbeitskopie)
@@ -1,7 +1,9 @@
+AUTOMAKE_OPTIONS = foreign
+
 noinst_LTLIBRARIES = libjnetlib.la
 
 libjnetlib_la_LDFLAGS = -pthread $(GLIB_LIBS) $(GLIBMM_LIBS) $(SIGC_LIBS) $(GTK_LIBS)
-libjnetlib_la_CXXFLAGS = -ggdb3 -s -O2 -Wall -DTHREAD_SAFE -D_THREAD_SAFE -D_REENTRANT $(GLIB_CFLAGS) $(SIGC_CFLAGS) $(GLIBMM_CFLAGS) $(GTK_CFLAGS)
+libjnetlib_la_CXXFLAGS = -DTHREAD_SAFE -D_THREAD_SAFE -D_REENTRANT $(GLIB_CFLAGS) $(SIGC_CFLAGS) $(GLIBMM_CFLAGS) $(GTK_CFLAGS)
 
 libjnetlib_la_SOURCES =  	\
 			asyncdns.cpp				\
Index: src/parser/Makefile.am
===================================================================
--- src/parser/Makefile.am	(Revision 11299)
+++ src/parser/Makefile.am	(Arbeitskopie)
@@ -1,3 +1,5 @@
+AUTOMAKE_OPTIONS = foreign
+
 noinst_LTLIBRARIES = libbmpparser.la
 
 libbmpparser_la_CXXFLAGS = \
@@ -3,5 +5,5 @@
 	$(BOOST_FILESYSTEM_CFLAGS) $(GLIBMM_CFLAGS) $(XML_CFLAGS) -I$(top_srcdir) -I$(top_srcdir)/include -I$(top_srcdir)/src
 
-libbmpparser_la_LDFLAGS = -ggdb3 $(BOOST_FILESYSTEM_LIBS)
+libbmpparser_la_LDFLAGS =  $(BOOST_FILESYSTEM_LIBS)
 
 libbmpparser_la_SOURCES =  \
Index: src/musicbrainz/Makefile.am
===================================================================
--- src/musicbrainz/Makefile.am	(Revision 11299)
+++ src/musicbrainz/Makefile.am	(Arbeitskopie)
@@ -1,3 +1,5 @@
+AUTOMAKE_OPTIONS = foreign
+
 noinst_LTLIBRARIES = libbmpmusicbrainz.la
 
 libbmpmusicbrainz_la_CXXFLAGS = \
@@ -3,5 +5,5 @@
 	$(SOUP_CFLAGS) $(BOOST_FILESYSTEM_CFLAGS) $(GLIBMM_CFLAGS) $(XML_CFLAGS) -I$(top_srcdir) -I$(top_srcdir)/include -I$(top_srcdir)/src
 
-libbmpmusicbrainz_la_LDFLAGS = -ggdb3 $(BOOST_FILESYSTEM_LIBS) $(SOUP_LIBS)
+libbmpmusicbrainz_la_LDFLAGS = $(BOOST_FILESYSTEM_LIBS) $(SOUP_LIBS)
 
 libbmpmusicbrainz_la_SOURCES =  		\
