diff -Naur bmpx-0.40.1/build.h bmpx-0.40.1-p/build.h
--- bmpx-0.40.1/build.h	2007-08-11 21:02:05.000000000 +0200
+++ bmpx-0.40.1-p/build.h	2007-08-30 01:27:00.000000000 +0200
@@ -1,8 +1,8 @@
 #ifndef _BUILD_H
 #define _BUILD_H
 
-#define BUILD_DATE "Sat Aug 11 21:02:05 CEST 2007"
-#define BUILD_BUILDUSER "mderezynski@milosz-laptop"
-#define BUILD_ARCH "Linux 2.6.20-16-generic i686"
+#define BUILD_DATE ""
+#define BUILD_BUILDUSER "Frugalware Linux"
+#define BUILD_ARCH ""
 
 #endif /* _BUILD_H */
diff -Naur bmpx-0.40.1/src/preferences.cc bmpx-0.40.1-p/src/preferences.cc
--- bmpx-0.40.1/src/preferences.cc	2007-08-11 20:41:51.000000000 +0200
+++ bmpx-0.40.1-p/src/preferences.cc	2007-08-30 01:26:15.000000000 +0200
@@ -604,8 +604,8 @@
       m_state_err = Gdk::Pixbuf::create_from_file( build_filename( BMP_IMAGE_DIR_STOCK, "silk-exclamation.png") );
 
       /* version string */
-      static boost::format version_f ("<span size='small' color='#606060'>BMPx %s:%s-R%s (%s / %s)</span>");
-      std::string version = (version_f % PACKAGE_VERSION % RV_LAST_CHANGED_DATE % RV_REVISION % BUILD_DATE % BUILD_BUILDUSER).str ();
+      static boost::format version_f ("<span size='small' color='#606060'>BMPx %s:%s-R%s ( %s )</span>");
+      std::string version = (version_f % PACKAGE_VERSION % RV_LAST_CHANGED_DATE % RV_REVISION  % BUILD_BUILDUSER).str ();
       dynamic_cast<Label*> (m_ref_xml->get_widget ("l_version"))->set_markup (version);
 
       m_ref_xml->get_widget ("cbox_audio_system", m_cbox_audio_system);
