diff --git a/docs/images/Makefile.am b/docs/images/Makefile.am
index 32a2a19..c28aa84 100644
--- a/docs/images/Makefile.am
+++ b/docs/images/Makefile.am
@@ -1,4 +1,4 @@
-image_DATA=main-window.png bmp.png active-source.png playing-source.png important.png tip.png favicon.ico bmp-logo-small.png gradient.png rect1869.png favicon.ico
+image_DATA=main-window.png bmp.png active-source.png playing-source.png important.png tip.png favicon.ico bmp-logo-small.png gradient.png rect1869.png
 imagedir=$(prefix)/share/doc/@PACKAGE@/images
 
 EXTRA_DIST=$(image_DATA)
diff --git a/src/ui-part-library.cc b/src/ui-part-library.cc
index cfe4b5c..a322d75 100644
--- a/src/ui-part-library.cc
+++ b/src/ui-part-library.cc
@@ -509,8 +509,8 @@ namespace Bmp
       if( node != NODE_BRANCH )
       {
         // Clone this node as one child row
-        UID uid (Bmp::AlbumArtist ((*i_toplevel)[mStoreArtistCR.artist]).bmpx_album_artist_id);
         Bmp::AlbumArtist aa ((*i_toplevel)[mStoreArtistCR.artist]);
+        UID uid (aa.bmpx_album_artist_id);
 
         UidIterMapIter uid_i = mUidIterMap.find (uid);
         if( uid_i != mUidIterMap.end() )
diff --git a/src/uri.cc b/src/uri.cc
index 4022882..5e76a54 100644
--- a/src/uri.cc
+++ b/src/uri.cc
@@ -35,6 +35,7 @@
 //  BMPx is covered by.
 
 #include <glib.h>
+#include <stdio.h>
 #include <string>
 #include <boost/algorithm/string.hpp>
 #include <glibmm.h>
diff --git a/widgets/popup.cc b/widgets/popup.cc
index 9ef4559..02767dc 100644
--- a/widgets/popup.cc
+++ b/widgets/popup.cc
@@ -308,7 +308,7 @@ namespace Bmp
 
 		if (m_location == ARROW_TOP)
 		{
-			Cairo::Matrix matrix = { 1, 0, 0, -1, 0, h };
+			Cairo::Matrix matrix(1, 0, 0, -1, 0, h);
 			cr->set_matrix (matrix);
 		}
 
