diff -Naur bmpx-0.40.13/libhal++/util.cc bmpx-0.40.13-p/libhal++/util.cc
--- bmpx-0.40.13/libhal++/util.cc	2007-10-18 19:17:21.000000000 +0200
+++ bmpx-0.40.13-p/libhal++/util.cc	2008-04-06 15:27:06.000000000 +0200
@@ -1,6 +1,6 @@
 #include <string>
 #include <cstring>
-
+#include <cstdlib>
 #include "types.hh"
 
 namespace Hal
diff -Naur bmpx-0.40.13/src/hal.hh bmpx-0.40.13-p/src/hal.hh
--- bmpx-0.40.13/src/hal.hh	2007-10-18 19:17:08.000000000 +0200
+++ bmpx-0.40.13-p/src/hal.hh	2008-04-06 16:09:35.000000000 +0200
@@ -30,6 +30,7 @@
 #endif //HAVE_CONFIG_H
 
 #include <string>
+#include <ctime>
 #include <map>
 #include <vector>
 
@@ -169,7 +170,7 @@
 
         void  device_condition            (std::string const& udi,
                                            std::string const& cond_name,
-                                           std::string const& cond_name );
+                                           std::string const& cond_name2 );
         void  device_added                (std::string const& udi );
         void  device_removed              (std::string const& udi );
         void  device_property             (std::string const& udi,
