diff -up bmpx-0.40.13/src/xds.cc.gcc43 bmpx-0.40.13/src/xds.cc
--- bmpx-0.40.13/src/xds.cc.gcc43	2007-10-18 19:17:08.000000000 +0200
+++ bmpx-0.40.13/src/xds.cc	2008-03-01 22:18:55.000000000 +0100
@@ -22,6 +22,7 @@
 //  permission is above and beyond the permissions granted by the GPL license
 //  BMPx is covered by.
 
+#include <cstring>
 #include <gtkmm.h>
 #include "xds.hh"
 #define MAX_URI_LEN 4096
diff -up bmpx-0.40.13/src/podcast.cc.gcc43 bmpx-0.40.13/src/podcast.cc
--- bmpx-0.40.13/src/podcast.cc.gcc43	2007-10-18 19:17:08.000000000 +0200
+++ bmpx-0.40.13/src/podcast.cc	2008-03-01 22:18:55.000000000 +0100
@@ -41,6 +41,7 @@
 #include <glibmm/markup.h>
 #include <iostream>
 #include <fstream>
+#include <cstring>
 
 #include "paths.hh"
 #include "util.hh"
diff -up bmpx-0.40.13/src/parser/libxml2-sax-base.cc.gcc43 bmpx-0.40.13/src/parser/libxml2-sax-base.cc
--- bmpx-0.40.13/src/parser/libxml2-sax-base.cc.gcc43	2007-10-18 19:17:06.000000000 +0200
+++ bmpx-0.40.13/src/parser/libxml2-sax-base.cc	2008-03-01 22:18:55.000000000 +0100
@@ -26,6 +26,7 @@
 #  include <config.h>
 #endif //HAVE_CONFIG_H
 
+#include <cstring>
 #include <libxml/parser.h>
 #include <glibmm.h>
 
diff -up bmpx-0.40.13/src/streams-shoutcast.cc.gcc43 bmpx-0.40.13/src/streams-shoutcast.cc
--- bmpx-0.40.13/src/streams-shoutcast.cc.gcc43	2007-10-18 19:17:08.000000000 +0200
+++ bmpx-0.40.13/src/streams-shoutcast.cc	2008-03-01 22:18:55.000000000 +0100
@@ -27,6 +27,7 @@
 
 #include <iostream>
 #include <sstream>
+#include <cstring>
 
 #include <glibmm.h>
 #include <glibmm/markup.h>
diff -up bmpx-0.40.13/src/ui-part-cdda.cc.gcc43 bmpx-0.40.13/src/ui-part-cdda.cc
--- bmpx-0.40.13/src/ui-part-cdda.cc.gcc43	2007-10-18 19:17:08.000000000 +0200
+++ bmpx-0.40.13/src/ui-part-cdda.cc	2008-03-01 22:18:55.000000000 +0100
@@ -35,6 +35,7 @@
 #include <iostream>
 #include <sstream>
 #include <string>
+#include <cstring>
 
 #include <boost/algorithm/string.hpp>
 #include <boost/format.hpp>
diff -up bmpx-0.40.13/src/minisoup.cc.gcc43 bmpx-0.40.13/src/minisoup.cc
--- bmpx-0.40.13/src/minisoup.cc.gcc43	2008-03-01 22:18:55.000000000 +0100
+++ bmpx-0.40.13/src/minisoup.cc	2008-03-01 22:18:55.000000000 +0100
@@ -26,6 +26,7 @@
 #  include <config.h>
 #endif //HAVE_CONFIG_H
 
+#include <cstring>
 #include <glib/gstdio.h>
 #include <glibmm/i18n.h>
 #include <boost/format.hpp>
diff -up bmpx-0.40.13/src/database.cc.gcc43 bmpx-0.40.13/src/database.cc
--- bmpx-0.40.13/src/database.cc.gcc43	2007-10-18 19:17:08.000000000 +0200
+++ bmpx-0.40.13/src/database.cc	2008-03-01 22:18:55.000000000 +0100
@@ -28,6 +28,7 @@
 
 #include <iostream>
 #include <string>
+#include <cstring>
 
 #include <glibmm.h>
 #include <glib/gstdio.h>
diff -up bmpx-0.40.13/src/video-widget.cc.gcc43 bmpx-0.40.13/src/video-widget.cc
--- bmpx-0.40.13/src/video-widget.cc.gcc43	2008-03-01 22:25:26.000000000 +0100
+++ bmpx-0.40.13/src/video-widget.cc	2008-03-01 22:25:31.000000000 +0100
@@ -22,6 +22,8 @@
 //  permission is above and beyond the permissions granted by the GPL license
 //  BMPx is covered by.
 
+#include <cstring>
+
 #include <gtkmm.h>
 #include "video-widget.hh"
 #include "audio/play.hh"
diff -up bmpx-0.40.13/src/uri.cc.gcc43 bmpx-0.40.13/src/uri.cc
--- bmpx-0.40.13/src/uri.cc.gcc43	2007-10-18 19:17:08.000000000 +0200
+++ bmpx-0.40.13/src/uri.cc	2008-03-01 22:18:55.000000000 +0100
@@ -39,6 +39,7 @@
 #include <boost/algorithm/string.hpp>
 #include <glibmm.h>
 #include <iostream>
+#include <cstring>
 
 #include "debug.hh"
 #include "uri.hh"
diff -up bmpx-0.40.13/src/vfs.hh.gcc43 bmpx-0.40.13/src/vfs.hh
--- bmpx-0.40.13/src/vfs.hh.gcc43	2007-10-18 19:17:08.000000000 +0200
+++ bmpx-0.40.13/src/vfs.hh	2008-03-01 22:18:55.000000000 +0100
@@ -26,6 +26,7 @@
 
 #include <iostream>
 #include <sstream>
+#include <cstring>
 #include <glibmm.h>
 #include <boost/shared_ptr.hpp>
 
diff -up bmpx-0.40.13/plugins/vfs/container/libcontainer_m3u.cc.gcc43 bmpx-0.40.13/plugins/vfs/container/libcontainer_m3u.cc
--- bmpx-0.40.13/plugins/vfs/container/libcontainer_m3u.cc.gcc43	2007-10-18 19:17:14.000000000 +0200
+++ bmpx-0.40.13/plugins/vfs/container/libcontainer_m3u.cc	2008-03-01 22:18:55.000000000 +0100
@@ -21,6 +21,8 @@
 //  permission is above and beyond the permissions granted by the GPL license
 //  BMPx is covered by.
 
+#include <cstring>
+
 #include <glibmm.h>
 #include <glibmm/i18n.h>
 
diff -up bmpx-0.40.13/json/json_writer.cpp.gcc43 bmpx-0.40.13/json/json_writer.cpp
--- bmpx-0.40.13/json/json_writer.cpp.gcc43	2007-10-18 19:17:09.000000000 +0200
+++ bmpx-0.40.13/json/json_writer.cpp	2008-03-01 22:18:55.000000000 +0100
@@ -3,6 +3,7 @@
 #include <assert.h>
 #include <stdio.h>
 #include <iostream>
+#include <cstring>
 
 #if _MSC_VER >= 1400 // VC++ 8.0
 #pragma warning( disable : 4996 )   // disable warning about strdup being deprecated.
diff -up bmpx-0.40.13/json/json_value.cpp.gcc43 bmpx-0.40.13/json/json_value.cpp
--- bmpx-0.40.13/json/json_value.cpp.gcc43	2007-10-18 19:17:09.000000000 +0200
+++ bmpx-0.40.13/json/json_value.cpp	2008-03-01 22:18:55.000000000 +0100
@@ -3,6 +3,7 @@
 #include <json/writer.h>
 #include <utility>
 #include <stdexcept>
+#include <cstring>
 #include "assert.h"
 #ifdef JSON_USE_CPPTL
 # include <cpptl/conststring.h>
diff -up bmpx-0.40.13/json/json_reader.cpp.gcc43 bmpx-0.40.13/json/json_reader.cpp
--- bmpx-0.40.13/json/json_reader.cpp.gcc43	2007-10-18 19:17:09.000000000 +0200
+++ bmpx-0.40.13/json/json_reader.cpp	2008-03-01 22:18:55.000000000 +0100
@@ -5,6 +5,7 @@
 #include <assert.h>
 #include <iostream>
 #include <stdexcept>
+#include <cstring>
 
 #if _MSC_VER >= 1400 // VC++ 8.0
 #pragma warning( disable : 4996 )   // disable warning about strdup being deprecated.
diff -up bmpx-0.40.13/jnetlib/httpget.cpp.gcc43 bmpx-0.40.13/jnetlib/httpget.cpp
--- bmpx-0.40.13/jnetlib/httpget.cpp.gcc43	2007-10-18 19:17:06.000000000 +0200
+++ bmpx-0.40.13/jnetlib/httpget.cpp	2008-03-01 22:18:55.000000000 +0100
@@ -20,6 +20,7 @@
 #include <iostream>
 #include <fstream>
 #include <vector>
+#include <cstring>
 
 const char* 
 JNL_HTTPGet::get_data ()
diff -up bmpx-0.40.13/jnetlib/httpserv.h.gcc43 bmpx-0.40.13/jnetlib/httpserv.h
--- bmpx-0.40.13/jnetlib/httpserv.h.gcc43	2007-10-18 19:17:06.000000000 +0200
+++ bmpx-0.40.13/jnetlib/httpserv.h	2008-03-01 22:18:55.000000000 +0100
@@ -12,6 +12,7 @@
 #ifndef _HTTPSERV_H_
 #define _HTTPSERV_H_
 
+#include <cstring>
 #include "connection.h"
 
 class JNL_HTTPServ
diff -up bmpx-0.40.13/jnetlib/listen.cpp.gcc43 bmpx-0.40.13/jnetlib/listen.cpp
--- bmpx-0.40.13/jnetlib/listen.cpp.gcc43	2008-03-01 22:23:45.000000000 +0100
+++ bmpx-0.40.13/jnetlib/listen.cpp	2008-03-01 22:24:13.000000000 +0100
@@ -6,6 +6,8 @@
 ** License: see jnetlib.h
 */
 
+#include <cstring>
+
 #include "netinc.h"
 #include "util.h"
 #include "listen.h"
diff -up bmpx-0.40.13/jnetlib/httpget.h.gcc43 bmpx-0.40.13/jnetlib/httpget.h
--- bmpx-0.40.13/jnetlib/httpget.h.gcc43	2007-10-18 19:17:06.000000000 +0200
+++ bmpx-0.40.13/jnetlib/httpget.h	2008-03-01 22:18:55.000000000 +0100
@@ -46,6 +46,7 @@
 
 #include <map>
 #include <string>
+#include <cstring>
 #include <glib.h>
 #include <glibmm.h>
 
diff -up bmpx-0.40.13/jnetlib/asyncdns.cpp.gcc43 bmpx-0.40.13/jnetlib/asyncdns.cpp
--- bmpx-0.40.13/jnetlib/asyncdns.cpp.gcc43	2008-03-01 22:23:22.000000000 +0100
+++ bmpx-0.40.13/jnetlib/asyncdns.cpp	2008-03-01 22:23:29.000000000 +0100
@@ -6,6 +6,8 @@
 ** License: see jnetlib.h
 */
 
+#include <cstring>
+
 #include "netinc.h"
 #include "util.h"
 #include "asyncdns.h"
diff -up bmpx-0.40.13/jnetlib/connection.cpp.gcc43 bmpx-0.40.13/jnetlib/connection.cpp
--- bmpx-0.40.13/jnetlib/connection.cpp.gcc43	2007-10-18 19:17:06.000000000 +0200
+++ bmpx-0.40.13/jnetlib/connection.cpp	2008-03-01 22:18:55.000000000 +0100
@@ -13,7 +13,7 @@
 #include "connection.h"
 
 #include <glib.h>
-
+#include <cstring>
 
 JNL_Connection::JNL_Connection(JNL_AsyncDNS *dns, int sendbufsize, int recvbufsize)
 {
