diff -Naur bmpx-0.32.0/m4/boost.m4 bmpx-0.32.0-p/m4/boost.m4
--- bmpx-0.32.0/m4/boost.m4	2006-10-02 00:34:15.000000000 +0200
+++ bmpx-0.32.0-p/m4/boost.m4	2006-10-03 12:41:41.000000000 +0200
@@ -11,7 +11,7 @@
 AC_DEFUN([BMP_WRAPPED_BOOST_CHECK],
     [AC_REQUIRE([BMP_BOOST_VARS])
      BMP_WRAPPED_CHECK(
-         [CXXFLAGS="-Wall -ansi -pedantic -Werror $BOOST_CFLAGS"
+         [CXXFLAGS="-Wall -ansi -pedantic  $BOOST_CFLAGS"
           LIBS="$BOOST_LIBS"
           $1
          ])
@@ -56,7 +56,7 @@
      AC_ARG_VAR([BOOST_FILESYSTEM_LIBS], [linker flags for Boost filesystem library])
 
      if test "$BOOST_FILESYSTEM_LIBS" = ""; then
-         BOOST_FILESYSTEM_LIBS="-lboost_filesystem"
+         BOOST_FILESYSTEM_LIBS="-lboost_filesystem-mt"
      fi
 
      BMP_WRAPPED_BOOST_CHECK(
@@ -99,7 +99,7 @@
      AC_ARG_VAR([BOOST_REGEX_LIBS], [linker flags for Boost regex library])
 
      if test "$BOOST_REGEX_LIBS" = ""; then
-         BOOST_REGEX_LIBS="-lboost_regex"
+         BOOST_REGEX_LIBS="-lboost_regex-mt"
      fi
 
      BMP_WRAPPED_BOOST_CHECK(
