diff -Naur bmpx-0.32.0/bindings/python/Makefile.am bmpx-0.32.0-p/bindings/python/Makefile.am
--- bmpx-0.32.0/bindings/python/Makefile.am	2006-10-02 00:34:25.000000000 +0200
+++ bmpx-0.32.0-p/bindings/python/Makefile.am	2006-10-03 21:49:38.000000000 +0200
@@ -6,7 +6,7 @@
 	$(PYTHON) setup.py build
 
 install:
-	$(PYTHON) setup.py install
+	$(PYTHON) setup.py install --prefix $(prefix) --root $(DESTDIR)
 
 clean:
 	$(PYTHON) setup.py clean
