diff -Naurp bmpx-0.34.3/configure.ac bmpx-0.34.3-p/configure.ac
--- bmpx-0.34.3/configure.ac	2006-11-12 11:32:15.000000000 +0100
+++ bmpx-0.34.3-p/configure.ac	2006-11-12 22:11:20.000000000 +0100
@@ -632,7 +632,7 @@ AC_SUBST(DEFAULT_DEVICE)
 
 # BUILD_*
 BUILD_DATE=$(LANG=en_US LC_ALL=en_US LOCALE=en_US date)
-BUILD_USER=$LOGNAME@$(hostname)
+BUILD_USER="Frugalware Linux"
 BUILD_ARCH="$(uname -sr) ($(uname -m))"
 AC_SUBST(BUILD_DATE)
 AC_SUBST(BUILD_USER)
diff -Naurp bmpx-0.34.3/src/preferences-ui.cc bmpx-0.34.3-p/src/preferences-ui.cc
--- bmpx-0.34.3/src/preferences-ui.cc	2006-11-12 08:21:31.000000000 +0100
+++ bmpx-0.34.3-p/src/preferences-ui.cc	2006-11-12 21:53:08.000000000 +0100
@@ -490,7 +490,7 @@ namespace Bmp
           signal_clicked().connect (sigc::mem_fun (this, &Preferences::hide));
 
       // Version String 
-      boost::format version_format ("<span size='small' color='#606060'>BMPx %s:%s-R%s (%s / %s)</span>");
+      boost::format version_format ("<span size='small' color='#606060'>BMPx %s:%s-R%s ( %s - %s )</span>");
       std::string version = (version_format % PACKAGE_VERSION % RV_LAST_CHANGED_DATE % RV_REVISION % BUILD_DATE % BUILD_BUILDUSER).str ();
       dynamic_cast<Gtk::Label *>(m_ref_xml->get_widget ("l_version"))->set_markup (version);
 
