diff -ur a/configure b/configure
--- a/configure	2012-11-26 04:47:00.701947663 -0600
+++ b/configure	2012-11-26 04:47:49.847968294 -0600
@@ -44135,9 +44135,9 @@
   dirname=""
   libname=""
       boost_headers="detail/lightweight_mutex.hpp thread/thread.hpp multi_index_container.hpp multi_index/key_extractors.hpp thread/mutex.hpp program_options/options_description.hpp iostreams/stream.hpp"
-      boost_libs="thread program_options iostreams"
+      boost_libs="thread program_options iostreams system"
 
-      cygnal_boost_libs="serialization date_time"
+      cygnal_boost_libs="serialization date_time system"
 
       newlist=$incllist
 
