--- gpgme-1.16.0/src/posix-io.c	2020-07-15 12:14:17.000000000 +0200
+++ gpgme-1.16.0.new/src/posix-io.c	2022-01-01 13:57:25.735658128 +0100
@@ -570,7 +570,7 @@
               if (fd_list[i].fd > fd)
                 fd = fd_list[i].fd;
             fd++;
-#if defined(__sun) || defined(__FreeBSD__)
+#if defined(__sun) || defined(__FreeBSD__) || defined(__GLIBC__)
             closefrom (fd);
             max_fds = fd;
 #else /*!__sun */
