--- gpgme-1.23.2/configure.ac	2023-11-28 16:42:31.000000000 +0100
+++ gpgme-1.23.2.new/configure.ac	2024-11-04 10:51:35.692619922 +0100
@@ -591,7 +591,7 @@
 	  # Reset everything, so that we can look for another Python.
           m4_foreach([mym4pythonver],
                      [[2.7],[3.4],[3.5],[3.6],[3.7],[3.8],[3.9],[3.10],
-                      [3.11],[3.12],[all]],
+                      [3.11],[3.12],[3.13],[all]],
            [unset PYTHON
 	    unset PYTHON_VERSION
 	    unset PYTHON_CPPFLAGS
