--- gpgme-1.24.0/configure.ac	2024-11-06 11:36:54.000000000 +0100
+++ gpgme-1.24.0.new/configure.ac	2024-11-10 16:12:49.419545347 +0100
@@ -614,7 +614,7 @@
 	if test "$found_py" = "1" -o "$found_py3" = "1"; then
 	  # Reset everything, so that we can look for another Python.
           m4_foreach([mym4pythonver],
-                     [[2.7],[3.6],[3.8],[3.9],[3.10],[3.11],[3.12],[all]],
+                     [[2.7],[3.6],[3.8],[3.9],[3.10],[3.11],[3.12],[3.13],[all]],
            [unset PYTHON
 	    unset PYTHON_VERSION
 	    unset PYTHON_CPPFLAGS
