diff -ur scorched.old/src/client/console/ConsoleImpl.cpp scorched/src/client/console/ConsoleImpl.cpp
--- scorched.old/src/client/console/ConsoleImpl.cpp	2009-02-18 20:06:23.000000000 +0200
+++ scorched/src/client/console/ConsoleImpl.cpp	2009-02-18 20:39:23.000000000 +0200
@@ -26,6 +26,7 @@
 #include <GLEXT/GLViewPort.h>
 #include <GLW/GLWFont.h>
 #include <GLW/GLWToolTip.h>
+#include <climits>
 
 ConsoleImpl::ConsoleImpl() : 
 	GameStateI("Console"),
diff -ur scorched.old/src/common/common/DefinesFile.cpp scorched/src/common/common/DefinesFile.cpp
--- scorched.old/src/common/common/DefinesFile.cpp	2009-02-18 20:06:22.000000000 +0200
+++ scorched/src/common/common/DefinesFile.cpp	2009-02-18 20:07:58.000000000 +0200
@@ -24,6 +24,7 @@
 #include <common/DefinesFile.h>
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <cstring>
 
 #pragma warning(disable : 4996)
 
diff -ur scorched.old/src/common/common/LoggerI.cpp scorched/src/common/common/LoggerI.cpp
--- scorched.old/src/common/common/LoggerI.cpp	2009-02-18 20:06:22.000000000 +0200
+++ scorched/src/common/common/LoggerI.cpp	2009-02-18 20:14:55.000000000 +0200
@@ -20,6 +20,7 @@
 
 #include <common/LoggerI.h>
 #include <time.h>
+#include <cstring>
 
 LoggerInfo::LoggerInfo(
 	const std::string &message, 
diff -ur scorched.old/src/common/common/main.h scorched/src/common/common/main.h
--- scorched.old/src/common/common/main.h	2009-02-18 20:06:22.000000000 +0200
+++ scorched/src/common/common/main.h	2009-02-18 20:07:18.000000000 +0200
@@ -40,7 +40,7 @@
 void _no_storage()
 {
 	printf("Failed to allocate memory!!");
-	std::exit(1);
+	exit(1);
 }
 
 void run_main(int argc, char *argv[], OptionsParameters &params)
diff -ur scorched.old/src/common/common/sha2.h scorched/src/common/common/sha2.h
--- scorched.old/src/common/common/sha2.h	2009-02-18 20:06:22.000000000 +0200
+++ scorched/src/common/common/sha2.h	2009-02-18 20:24:57.000000000 +0200
@@ -46,6 +46,7 @@
 
 #include <string>
 #include <stdexcept>
+#include <cstring>
 
 // NOTE: You may need to define things by hand for your system: 
 typedef unsigned char  sha_byte;           // Exactly 1 byte 
diff -ur scorched.old/src/common/landscapemap/HeightMapModifier.cpp scorched/src/common/landscapemap/HeightMapModifier.cpp
--- scorched.old/src/common/landscapemap/HeightMapModifier.cpp	2009-02-18 20:06:22.000000000 +0200
+++ scorched/src/common/landscapemap/HeightMapModifier.cpp	2009-02-18 20:28:50.000000000 +0200
@@ -26,6 +26,7 @@
 #include <image/ImageBitmap.h>
 #include <image/ImageFactory.h>
 #include <lang/LangResource.h>
+#include <climits>
 
 void HeightMapModifier::levelSurround(HeightMap &hmap)
 {
diff -ur scorched.old/src/common/lang/LangParam.cpp scorched/src/common/lang/LangParam.cpp
--- scorched.old/src/common/lang/LangParam.cpp	2009-02-18 20:06:22.000000000 +0200
+++ scorched/src/common/lang/LangParam.cpp	2009-02-18 20:22:06.000000000 +0200
@@ -20,6 +20,7 @@
 
 #include <lang/LangParam.h>
 #include <common/DefinesString.h>
+#include <cstdlib>
 
 LangParam::LangParam(const LangStringConverter &convValue)
 {
diff -ur scorched.old/src/common/lang/ResourceBundle.h scorched/src/common/lang/ResourceBundle.h
--- scorched.old/src/common/lang/ResourceBundle.h	2009-02-18 20:06:22.000000000 +0200
+++ scorched/src/common/lang/ResourceBundle.h	2009-02-18 20:20:12.000000000 +0200
@@ -23,6 +23,7 @@
 
 #include <set>
 #include <lang/ResourceBundleEntry.h>
+#include <cstring>
 
 class ResourceBundle
 {
diff -ur scorched.old/src/common/porting/windows.h scorched/src/common/porting/windows.h
--- scorched.old/src/common/porting/windows.h	2009-02-18 20:06:22.000000000 +0200
+++ scorched/src/common/porting/windows.h	2009-02-18 20:06:46.000000000 +0200
@@ -20,7 +20,7 @@
 typedef unsigned short  WORD;
 typedef unsigned char   BYTE;
 
-typedef int 	HWND;
+// typedef int 	HWND;
 typedef void * 	HINSTANCE;
 typedef int 	WPARAM;
 typedef int 	LPARAM;
