diff -Naur man-1.5n.orig/src/man.conf.in man-1.5n/src/man.conf.in
--- man-1.5n.orig/src/man.conf.in	2003-03-11 16:53:29.000000000 +0100
+++ man-1.5n/src/man.conf.in	2004-08-08 19:36:22.000000000 +0200
@@ -88,12 +88,9 @@
 # For use with utf-8, NROFF should be "nroff -mandoc" without -T option.
 # (Maybe - but today I need -Tlatin1 to prevent double conversion to utf8.)
 #
-# If you have a new troff (version 1.18.1?) and its colored output
-# causes problems, add the -c option to TROFF, NROFF, JNROFF.
-#
-TROFF		@troff@
-NROFF		@nroff@
-JNROFF		@jnroff@
+TROFF		@troff@ -c
+NROFF		@nroff@ -c
+JNROFF		@jnroff@ -c
 EQN		@eqn@
 NEQN		@neqn@
 JNEQN		@jneqn@
