diff -aur centericq-4.21.0/libicq2000-0.1/libicq2000/ContactTree.h fw_centericq-4.21.0/libicq2000-0.1/libicq2000/ContactTree.h
--- centericq-4.21.0/libicq2000-0.1/libicq2000/ContactTree.h	2004-03-06 20:02:12.000000000 +0100
+++ fw_centericq-4.21.0/libicq2000-0.1/libicq2000/ContactTree.h	2008-10-02 01:29:19.000000000 +0200
@@ -23,6 +23,7 @@
 #ifndef CONTACTTREE_H
 #define CONTACTTREE_H
 
+#include <algorithm>
 #include <list>
 #include <string>
 #include <map>
diff -aur centericq-4.21.0/libicq2000-0.1/src/Capabilities.cpp fw_centericq-4.21.0/libicq2000-0.1/src/Capabilities.cpp
--- centericq-4.21.0/libicq2000-0.1/src/Capabilities.cpp	2004-06-28 10:46:14.000000000 +0200
+++ fw_centericq-4.21.0/libicq2000-0.1/src/Capabilities.cpp	2008-10-02 01:57:13.000000000 +0200
@@ -21,6 +21,8 @@
 
 #include "Capabilities.h"
 
+#include <string.h>
+
 using std::set;
 
 namespace ICQ2000 {
diff -aur centericq-4.21.0/libicq2000-0.1/src/ICQ.cpp fw_centericq-4.21.0/libicq2000-0.1/src/ICQ.cpp
--- centericq-4.21.0/libicq2000-0.1/src/ICQ.cpp	2004-02-11 00:51:11.000000000 +0100
+++ fw_centericq-4.21.0/libicq2000-0.1/src/ICQ.cpp	2008-10-02 01:51:44.000000000 +0200
@@ -23,6 +23,7 @@
 
 #include "sstream_fix.h"
 #include <memory>
+#include <stdlib.h>
 
 using std::string;
 using std::list;
diff -aur centericq-4.21.0/libicq2000-0.1/src/socket.cpp fw_centericq-4.21.0/libicq2000-0.1/src/socket.cpp
--- centericq-4.21.0/libicq2000-0.1/src/socket.cpp	2005-01-07 22:28:28.000000000 +0100
+++ fw_centericq-4.21.0/libicq2000-0.1/src/socket.cpp	2008-10-02 01:44:24.000000000 +0200
@@ -21,6 +21,7 @@
 
 #include "sstream_fix.h"
 #include <algorithm>
+#include <string.h>
 
 #include "socket.h"
 
diff -aur centericq-4.21.0/libmsn-0.1/msn/connection.cpp fw_centericq-4.21.0/libmsn-0.1/msn/connection.cpp
--- centericq-4.21.0/libmsn-0.1/msn/connection.cpp	2004-06-29 01:17:07.000000000 +0200
+++ fw_centericq-4.21.0/libmsn-0.1/msn/connection.cpp	2008-10-02 02:07:58.000000000 +0200
@@ -38,6 +38,7 @@
 #include <time.h>
 #include <cassert>
 #include <cerrno>
+#include <cstring>
 
 namespace MSN
 {
diff -aur centericq-4.21.0/libmsn-0.1/msn/filetransfer.cpp fw_centericq-4.21.0/libmsn-0.1/msn/filetransfer.cpp
--- centericq-4.21.0/libmsn-0.1/msn/filetransfer.cpp	2004-06-29 12:59:04.000000000 +0200
+++ fw_centericq-4.21.0/libmsn-0.1/msn/filetransfer.cpp	2008-10-02 02:06:58.000000000 +0200
@@ -35,6 +35,8 @@
 
 #include <cassert>
 #include <cerrno>
+#include <cstdlib>
+#include <cstring>
 
 namespace MSN 
 {
diff -aur centericq-4.21.0/libmsn-0.1/msn/message.cpp fw_centericq-4.21.0/libmsn-0.1/msn/message.cpp
--- centericq-4.21.0/libmsn-0.1/msn/message.cpp	2004-06-21 23:57:38.000000000 +0200
+++ fw_centericq-4.21.0/libmsn-0.1/msn/message.cpp	2008-10-02 01:29:19.000000000 +0200
@@ -25,6 +25,7 @@
 #include <msn/util.h>
 #include <iomanip>
 #include <cassert>
+#include <cstdlib>
 
 namespace MSN 
 {
diff -aur centericq-4.21.0/libmsn-0.1/msn/notificationserver.cpp fw_centericq-4.21.0/libmsn-0.1/msn/notificationserver.cpp
--- centericq-4.21.0/libmsn-0.1/msn/notificationserver.cpp	2004-12-19 03:50:29.000000000 +0100
+++ fw_centericq-4.21.0/libmsn-0.1/msn/notificationserver.cpp	2008-10-02 02:20:35.000000000 +0200
@@ -29,6 +29,7 @@
 #include <algorithm>
 #include <cassert>
 #include <cctype>
+#include <cstring>
 
 #ifndef WIN32
 #include <unistd.h>
diff -aur centericq-4.21.0/libmsn-0.1/msn/util.cpp fw_centericq-4.21.0/libmsn-0.1/msn/util.cpp
--- centericq-4.21.0/libmsn-0.1/msn/util.cpp	2004-06-29 01:17:07.000000000 +0200
+++ fw_centericq-4.21.0/libmsn-0.1/msn/util.cpp	2008-10-02 08:57:52.000000000 +0200
@@ -23,6 +23,8 @@
 #include <msn/util.h>
 #include <cerrno>
 #include <cctype>
+#include <cstdlib>
+#include <cstring>
 
 namespace MSN 
 {    
