diff -Naur aria2-0.11.5/config.h.in aria2-0.11.5-p/config.h.in
--- aria2-0.11.5/config.h.in	2007-11-17 11:09:10.000000000 +0100
+++ aria2-0.11.5-p/config.h.in	2007-11-26 13:35:34.000000000 +0100
@@ -138,6 +138,9 @@
 /* Define to 1 if you have libcares. */
 #undef HAVE_LIBCARES
 
+/* Define 1 if ares_host_callback accepts timeouts(c-ares >= 1.5) */
+#undef HAVE_LIBCARES1_5
+
 /* Define to 1 if you have libgcrypt. */
 #undef HAVE_LIBGCRYPT
 
diff -Naur aria2-0.11.5/configure.ac aria2-0.11.5-p/configure.ac
--- aria2-0.11.5/configure.ac	2007-11-17 11:09:10.000000000 +0100
+++ aria2-0.11.5-p/configure.ac	2007-11-26 13:35:34.000000000 +0100
@@ -38,6 +38,9 @@
 AC_PROG_RANLIB
 AC_PROG_YACC
 
+# Setting language choice
+AC_LANG([C++])
+
 # Checks for libraries.
 if test "x$with_libxml2" = "xyes"; then
    AM_PATH_XML2([2.6.24], [have_libxml2=yes])
diff -Naur aria2-0.11.5/doc/Makefile.in aria2-0.11.5-p/doc/Makefile.in
--- aria2-0.11.5/doc/Makefile.in	2007-08-28 18:36:33.000000000 +0200
+++ aria2-0.11.5-p/doc/Makefile.in	2007-11-26 13:35:34.000000000 +0100
@@ -77,12 +77,12 @@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CFLAGS = @CFLAGS@
-CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CPPUNIT_CFLAGS = @CPPUNIT_CFLAGS@
 CPPUNIT_CONFIG = @CPPUNIT_CONFIG@
 CPPUNIT_LIBS = @CPPUNIT_LIBS@
 CXX = @CXX@
+CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
 CXXFLAGS = @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
diff -Naur aria2-0.11.5/m4/libcares.m4 aria2-0.11.5-p/m4/libcares.m4
--- aria2-0.11.5/m4/libcares.m4	2007-08-28 18:36:33.000000000 +0200
+++ aria2-0.11.5-p/m4/libcares.m4	2007-11-26 13:35:34.000000000 +0100
@@ -16,10 +16,30 @@
 CPPFLAGS_save=$CPPFLAGS
 
 LIBS="-L$libcares_prefix_lib $LIBS"
-CPPFLAGS="-I$libcares_prefix_include $CPPFLAGS"
+CPPFLAGS="-I$libcares_prefix_include -Wall $CPPFLAGS"
 
 AC_CHECK_LIB([cares], [ares_init], [have_libcares=yes])
+
 if test "x$have_libcares" = "xyes"; then
+
+    AC_MSG_CHECKING([whether ares_host_callback accepts timeouts(c-ares >= 1.5)])
+    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+    #include <ares.h>
+
+    void callback(void* arg, int status, int timeouts, struct hostent* host);
+    ]],
+    [[
+    ares_channel channel;
+    ares_gethostbyname(channel, "foo", 0, callback, 0);
+    ]])],
+    [have_libcares1_5=yes],
+    [have_libcares1_5=no])
+    AC_MSG_RESULT([$have_libcares1_5])
+
+    if test "x$have_libcares1_5" = "xyes"; then
+        AC_DEFINE([HAVE_LIBCARES1_5], [1], [Define 1 if ares_host_callback accepts timeouts(c-ares >= 1.5)])
+    fi
+
     AC_DEFINE([HAVE_LIBCARES], [1], [Define to 1 if you have libcares.])
     LIBCARES_LIBS="-L$libcares_prefix_lib -lcares"
     LIBCARES_CPPFLAGS="-I$libcares_prefix_include"
diff -Naur aria2-0.11.5/m4/Makefile.in aria2-0.11.5-p/m4/Makefile.in
--- aria2-0.11.5/m4/Makefile.in	2007-08-28 18:36:33.000000000 +0200
+++ aria2-0.11.5-p/m4/Makefile.in	2007-11-26 13:35:34.000000000 +0100
@@ -73,12 +73,12 @@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CFLAGS = @CFLAGS@
-CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CPPUNIT_CFLAGS = @CPPUNIT_CFLAGS@
 CPPUNIT_CONFIG = @CPPUNIT_CONFIG@
 CPPUNIT_LIBS = @CPPUNIT_LIBS@
 CXX = @CXX@
+CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
 CXXFLAGS = @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
diff -Naur aria2-0.11.5/Makefile.in aria2-0.11.5-p/Makefile.in
--- aria2-0.11.5/Makefile.in	2007-11-17 11:09:10.000000000 +0100
+++ aria2-0.11.5-p/Makefile.in	2007-11-26 13:35:34.000000000 +0100
@@ -99,12 +99,12 @@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CFLAGS = @CFLAGS@
-CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CPPUNIT_CFLAGS = @CPPUNIT_CFLAGS@
 CPPUNIT_CONFIG = @CPPUNIT_CONFIG@
 CPPUNIT_LIBS = @CPPUNIT_LIBS@
 CXX = @CXX@
+CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
 CXXFLAGS = @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
diff -Naur aria2-0.11.5/src/Makefile.in aria2-0.11.5-p/src/Makefile.in
--- aria2-0.11.5/src/Makefile.in	2007-09-03 12:32:19.000000000 +0200
+++ aria2-0.11.5-p/src/Makefile.in	2007-11-26 13:35:34.000000000 +0100
@@ -516,12 +516,12 @@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CFLAGS = @CFLAGS@
-CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CPPUNIT_CFLAGS = @CPPUNIT_CFLAGS@
 CPPUNIT_CONFIG = @CPPUNIT_CONFIG@
 CPPUNIT_LIBS = @CPPUNIT_LIBS@
 CXX = @CXX@
+CXXCPP = @CXXCPP@
 CXXDEPMODE = @CXXDEPMODE@
 CXXFLAGS = @CXXFLAGS@
 CYGPATH_W = @CYGPATH_W@
diff -Naur aria2-0.11.5/src/NameResolver.cc aria2-0.11.5-p/src/NameResolver.cc
--- aria2-0.11.5/src/NameResolver.cc	2007-08-28 18:36:33.000000000 +0200
+++ aria2-0.11.5-p/src/NameResolver.cc	2007-11-26 13:35:34.000000000 +0100
@@ -36,7 +36,11 @@
 
 #ifdef ENABLE_ASYNC_DNS
 
+#ifdef HAVE_LIBCARES1_5
+void callback(void* arg, int status, int timeouts, struct hostent* host) {
+#else
 void callback(void* arg, int status, struct hostent* host) {
+#endif // HAVE_LIBCARES1_5
   NameResolver* resolverPtr = (NameResolver*)arg;
 #ifdef HAVE_LIBARES
   // This block is required since the assertion in ares_strerror fails
diff -Naur aria2-0.11.5/src/NameResolver.h aria2-0.11.5-p/src/NameResolver.h
--- aria2-0.11.5/src/NameResolver.h	2007-08-28 18:36:33.000000000 +0200
+++ aria2-0.11.5-p/src/NameResolver.h	2007-11-26 13:35:34.000000000 +0100
@@ -48,10 +48,18 @@
 } /* end of extern "C" */
 #endif
 
+#ifdef HAVE_LIBCARES1_5
+void callback(void* arg, int status, int timeouts, struct hostent* host);
+#else
 void callback(void* arg, int status, struct hostent* host);
+#endif // HAVE_LIBCARES1_5
 
 class NameResolver {
+#ifdef HAVE_LIBCARES1_5
+  friend void callback(void* arg, int status, int timeouts, struct hostent* host);
+#else
   friend void callback(void* arg, int status, struct hostent* host);
+#endif // HAVE_LIBCARES1_5
 
 public:
   enum STATUS {
