Index: src/ActivePeerConnectionCommand.h
===================================================================
--- src/ActivePeerConnectionCommand.h	(revision 770)
+++ src/ActivePeerConnectionCommand.h	(revision 772)
@@ -55,7 +55,7 @@
   Time checkPoint;
   unsigned int _thresholdSpeed; // UNIT: byte/sec
   unsigned int _maxUploadSpeedLimit;
-  size_t _numNewConnection; // the number of the connection to establish.
+  unsigned int _numNewConnection; // the number of the connection to establish.
 public:
   ActivePeerConnectionCommand(int cuid,
 			      RequestGroup* requestGroup,
Index: src/ActivePeerConnectionCommand.cc
===================================================================
--- src/ActivePeerConnectionCommand.cc	(revision 770)
+++ src/ActivePeerConnectionCommand.cc	(revision 772)
@@ -89,11 +89,11 @@
        (!pieceStorage->downloadFinished() &&
 	(tstat.getDownloadSpeed() < _thresholdSpeed ||
 	 btRuntime->lessThanMinPeers()))) {
-      size_t numConnection = pieceStorage->downloadFinished() ?
+      unsigned int numConnection = pieceStorage->downloadFinished() ?
 	std::min(_numNewConnection,
 		 BtRuntime::MAX_PEERS-btRuntime->getConnections()) :
 	_numNewConnection;
-      for(size_t numAdd = numConnection;
+      for(unsigned int numAdd = numConnection;
 	  numAdd > 0 && peerStorage->isPeerAvailable(); --numAdd) {
 	PeerHandle peer = peerStorage->getUnusedPeer();
 	connectToPeer(peer);

