diff -Naur aria2c-0.13.1+2/src/IteratableChunkChecksumValidator.cc aria2c-0.13.1+2-p/src/IteratableChunkChecksumValidator.cc
--- aria2c-0.13.1+2/src/IteratableChunkChecksumValidator.cc	2008-02-24 10:43:31.000000000 +0100
+++ aria2c-0.13.1+2-p/src/IteratableChunkChecksumValidator.cc	2008-04-23 19:30:57.000000000 +0200
@@ -46,6 +46,7 @@
 #include "Logger.h"
 #include "messageDigest.h"
 #include <cerrno>
+#include <cstring>
 
 namespace aria2 {
 
diff -Naur aria2c-0.13.1+2/src/MessageDigestHelper.cc aria2c-0.13.1+2-p/src/MessageDigestHelper.cc
--- aria2c-0.13.1+2/src/MessageDigestHelper.cc	2008-02-28 18:40:47.000000000 +0100
+++ aria2c-0.13.1+2-p/src/MessageDigestHelper.cc	2008-04-23 19:34:42.000000000 +0200
@@ -39,6 +39,7 @@
 #include "DefaultDiskWriter.h"
 #include "Util.h"
 #include <cerrno>
+#include <cstring>
 
 namespace aria2 {
 
