diff -Naur sbcl-1.0.16/src/runtime/GNUmakefile sbcl-1.0.16-p/src/runtime/GNUmakefile
--- sbcl-1.0.16/src/runtime/GNUmakefile	2008-02-19 11:18:07.000000000 +0100
+++ sbcl-1.0.16-p/src/runtime/GNUmakefile	2008-05-19 15:40:58.000000000 +0200
@@ -23,7 +23,9 @@
 DEPEND_FLAGS = -MM
 GREP = grep
 
-CFLAGS = -g -Wall -Wsign-compare -O3
+CFLAGS ?= -g -Wall
+CFLAGS += -Wsign-compare
+CFLAGS += -DSBCL_HOME=\"/usr/lib/sbcl\" 
 ASFLAGS = $(CFLAGS)
 CPPFLAGS = -I.
 
