diff -Naur StepTalk.orig/Finders/ApplicationFinder/GNUmakefile StepTalk/Finders/ApplicationFinder/GNUmakefile
--- StepTalk.orig/Finders/ApplicationFinder/GNUmakefile	2007-07-29 23:09:44.000000000 +0200
+++ StepTalk/Finders/ApplicationFinder/GNUmakefile	2007-07-29 23:11:01.000000000 +0200
@@ -37,7 +37,7 @@
 ADDITIONAL_INCLUDE_DIRS += -I../../Frameworks/
 ADDITIONAL_LIB_DIRS += -L../../Frameworks/StepTalk/StepTalk.framework/Versions/Current/$(GNUSTEP_TARGET_LDIR)
 
-BUNDLE_INSTALL_DIR:=$(GNUSTEP_INSTALLATION_DIR)/Library/StepTalk/Finders
+BUNDLE_INSTALL_DIR:=$(DESTDIR)/$(GNUSTEP_SYSTEM_LIBRARY)/StepTalk/Finders
 
 -include GNUmakefile.preamble
 include $(GNUSTEP_MAKEFILES)/bundle.make
diff -Naur StepTalk.orig/Finders/DistributedFinder/GNUmakefile StepTalk/Finders/DistributedFinder/GNUmakefile
--- StepTalk.orig/Finders/DistributedFinder/GNUmakefile	2007-07-29 23:09:44.000000000 +0200
+++ StepTalk/Finders/DistributedFinder/GNUmakefile	2007-07-29 23:11:01.000000000 +0200
@@ -35,7 +35,7 @@
 ADDITIONAL_INCLUDE_DIRS += -I../../Frameworks/
 ADDITIONAL_LIB_DIRS += -L../../Frameworks/StepTalk/StepTalk.framework/Versions/Current/$(GNUSTEP_TARGET_LDIR)
 
-BUNDLE_INSTALL_DIR:=$(GNUSTEP_INSTALLATION_DIR)/Library/StepTalk/Finders
+BUNDLE_INSTALL_DIR:=$(DESTDIR)/$(GNUSTEP_SYSTEM_LIBRARY)/StepTalk/Finders
 
 -include GNUmakefile.preamble
 include $(GNUSTEP_MAKEFILES)/bundle.make
diff -Naur StepTalk.orig/Languages/Guile/GNUmakefile StepTalk/Languages/Guile/GNUmakefile
--- StepTalk.orig/Languages/Guile/GNUmakefile	2007-07-29 23:09:44.000000000 +0200
+++ StepTalk/Languages/Guile/GNUmakefile	2007-07-29 23:11:01.000000000 +0200
@@ -24,7 +24,7 @@
 
 BUNDLE_NAME        = Guile
 BUNDLE_EXTENSION  := .stlanguage
-BUNDLE_INSTALL_DIR:=$(GNUSTEP_INSTALLATION_DIR)/Library/StepTalk/Languages
+BUNDLE_INSTALL_DIR:=$(DESTDIR)/$(GNUSTEP_SYSTEM_LIBRARY)/StepTalk/Languages
     
 Guile_OBJC_FILES = \
         GuileEngine.m 
diff -Naur StepTalk.orig/Languages/MyLanguage/GNUmakefile StepTalk/Languages/MyLanguage/GNUmakefile
--- StepTalk.orig/Languages/MyLanguage/GNUmakefile	2007-07-29 23:09:44.000000000 +0200
+++ StepTalk/Languages/MyLanguage/GNUmakefile	2007-07-29 23:11:01.000000000 +0200
@@ -25,7 +25,7 @@
 ###########################################################################
 
 BUNDLE_EXTENSION  := .stlanguage
-BUNDLE_INSTALL_DIR:=$(GNUSTEP_INSTALLATION_DIR)/Library/StepTalk/Languages
+BUNDLE_INSTALL_DIR:=$(DESTDIR)/$(GNUSTEP_SYSTEM_LIBRARY)/StepTalk/Languages
 
 -include GNUmakefile.preamble
 include $(GNUSTEP_MAKEFILES)/bundle.make
diff -Naur StepTalk.orig/Languages/Smalltalk/GNUmakefile StepTalk/Languages/Smalltalk/GNUmakefile
--- StepTalk.orig/Languages/Smalltalk/GNUmakefile	2007-07-29 23:09:44.000000000 +0200
+++ StepTalk/Languages/Smalltalk/GNUmakefile	2007-07-29 23:11:01.000000000 +0200
@@ -24,7 +24,7 @@
 
 BUNDLE_NAME        = Smalltalk
 BUNDLE_EXTENSION  := .stlanguage
-BUNDLE_INSTALL_DIR:=$(GNUSTEP_INSTALLATION_DIR)/Library/StepTalk/Languages
+BUNDLE_INSTALL_DIR:=$(DESTDIR)/$(GNUSTEP_SYSTEM_LIBRARY)/StepTalk/Languages
     
 Smalltalk_OBJC_FILES = \
         SmalltalkEngine.m \
diff -Naur StepTalk.orig/Modules/AppKit/GNUmakefile StepTalk/Modules/AppKit/GNUmakefile
--- StepTalk.orig/Modules/AppKit/GNUmakefile	2007-07-29 23:09:44.000000000 +0200
+++ StepTalk/Modules/AppKit/GNUmakefile	2007-07-29 23:11:01.000000000 +0200
@@ -39,7 +39,7 @@
         
 AppKit_BUNDLE_LIBS += $(GUI_LIBS)
 
-BUNDLE_INSTALL_DIR:=$(GNUSTEP_INSTALLATION_DIR)/Library/StepTalk/Modules
+BUNDLE_INSTALL_DIR:=$(DESTDIR)/$(GNUSTEP_SYSTEM_LIBRARY)/StepTalk/Modules
 
 ADDITIONAL_OBJCFLAGS = -Wall -Wno-import
 
diff -Naur StepTalk.orig/Modules/Foundation/GNUmakefile StepTalk/Modules/Foundation/GNUmakefile
--- StepTalk.orig/Modules/Foundation/GNUmakefile	2007-07-29 23:09:44.000000000 +0200
+++ StepTalk/Modules/Foundation/GNUmakefile	2007-07-29 23:12:18.000000000 +0200
@@ -34,7 +34,7 @@
         
 Foundation_RESOURCE_FILES = ScriptingInfo.plist
 
-BUNDLE_INSTALL_DIR:=$(GNUSTEP_INSTALLATION_DIR)/Library/StepTalk/Modules
+BUNDLE_INSTALL_DIR:= $(DESTDIR)/$(GNUSTEP_SYSTEM_LIBRARY)/StepTalk/Modules
 ADDITIONAL_BUNDLE_LIBS = -lStepTalk
 ADDITIONAL_INCLUDE_DIRS += -I../../Frameworks/
 ADDITIONAL_LIB_DIRS += -L../../Frameworks/StepTalk/StepTalk.framework/Versions/Current/$(GNUSTEP_TARGET_LDIR)
diff -Naur StepTalk.orig/Modules/GDL2/GNUmakefile StepTalk/Modules/GDL2/GNUmakefile
--- StepTalk.orig/Modules/GDL2/GNUmakefile	2007-07-29 23:09:44.000000000 +0200
+++ StepTalk/Modules/GDL2/GNUmakefile	2007-07-29 23:11:01.000000000 +0200
@@ -41,7 +41,7 @@
 ADDITIONAL_INCLUDE_DIRS += -I../../Frameworks/
 ADDITIONAL_LIB_DIRS += -L../../Frameworks/StepTalk/StepTalk.framework/Versions/Current/$(GNUSTEP_TARGET_LDIR)
 
-BUNDLE_INSTALL_DIR:=$(GNUSTEP_INSTALLATION_DIR)/Library/StepTalk/Modules
+BUNDLE_INSTALL_DIR:=$(DESTDIR)/$(GNUSTEP_SYSTEM_LIBRARY)/StepTalk/Modules
 
 ADDITIONAL_OBJCFLAGS = -Wall -Wno-import
 
diff -Naur StepTalk.orig/Modules/GDL2/GNUmakefile.postamble StepTalk/Modules/GDL2/GNUmakefile.postamble
--- StepTalk.orig/Modules/GDL2/GNUmakefile.postamble	2007-07-29 23:09:44.000000000 +0200
+++ StepTalk/Modules/GDL2/GNUmakefile.postamble	2007-07-29 23:12:40.000000000 +0200
@@ -39,7 +39,7 @@
 # Things to do after installing
 after-install::
 	@(echo Copying scripting environments...; \
-	  cp GDL2.stenv $(GNUSTEP_INSTALLATION_DIR)/Library/StepTalk/Environments; \
+	  cp GDL2.stenv $(DESTIDR)/$(GNUSTEP_SYSTEM_LIBRARY)/StepTalk/Environments; \
 	)
 
 # Things to do before uninstalling
diff -Naur StepTalk.orig/Modules/ObjectiveC/GNUmakefile StepTalk/Modules/ObjectiveC/GNUmakefile
--- StepTalk.orig/Modules/ObjectiveC/GNUmakefile	2007-07-29 23:09:44.000000000 +0200
+++ StepTalk/Modules/ObjectiveC/GNUmakefile	2007-07-29 23:11:01.000000000 +0200
@@ -40,7 +40,7 @@
 ADDITIONAL_INCLUDE_DIRS += -I../../Frameworks/
 ADDITIONAL_LIB_DIRS += -L../../Frameworks/StepTalk/StepTalk.framework/Versions/Current/$(GNUSTEP_TARGET_LDIR)
 
-BUNDLE_INSTALL_DIR:=$(GNUSTEP_INSTALLATION_DIR)/Library/StepTalk/Modules
+BUNDLE_INSTALL_DIR:=$(DESTDIR)/$(GNUSTEP_SYSTEM_LIBRARY)/StepTalk/Modules
 
 ADDITIONAL_OBJCFLAGS = -Wall -Wno-import
 
diff -Naur StepTalk.orig/Modules/ReadlineTranscript/GNUmakefile StepTalk/Modules/ReadlineTranscript/GNUmakefile
--- StepTalk.orig/Modules/ReadlineTranscript/GNUmakefile	2007-07-29 23:09:44.000000000 +0200
+++ StepTalk/Modules/ReadlineTranscript/GNUmakefile	2007-07-29 23:11:01.000000000 +0200
@@ -32,7 +32,7 @@
         
 ReadlineTranscript_RESOURCE_FILES = ScriptingInfo.plist
 
-BUNDLE_INSTALL_DIR:=$(GNUSTEP_INSTALLATION_DIR)/Library/StepTalk/Modules
+BUNDLE_INSTALL_DIR:=$(DESTDIR)/$(GNUSTEP_SYSTEM_LIBRARY)/StepTalk/Modules
 
 ReadlineTranscript_BUNDLE_LIBS = -lStepTalk -lreadline
 ADDITIONAL_INCLUDE_DIRS += -I../../Frameworks/
diff -Naur StepTalk.orig/Modules/SimpleTranscript/GNUmakefile StepTalk/Modules/SimpleTranscript/GNUmakefile
--- StepTalk.orig/Modules/SimpleTranscript/GNUmakefile	2007-07-29 23:09:44.000000000 +0200
+++ StepTalk/Modules/SimpleTranscript/GNUmakefile	2007-07-29 23:11:01.000000000 +0200
@@ -32,7 +32,7 @@
         
 SimpleTranscript_RESOURCE_FILES = ScriptingInfo.plist
 
-BUNDLE_INSTALL_DIR:=$(GNUSTEP_INSTALLATION_DIR)/Library/StepTalk/Modules
+BUNDLE_INSTALL_DIR:=$(DESTDIR)/$(GNUSTEP_SYSTEM_LIBRARY)/StepTalk/Modules
 
 ADDITIONAL_BUNDLE_LIBS = -lStepTalk
 ADDITIONAL_INCLUDE_DIRS += -I../../Frameworks/
