--- gpsdrive-2.11/src/CMakeLists.txt.old	2011-02-17 09:49:21.000000000 +0000
+++ gpsdrive-2.11/src/CMakeLists.txt	2011-02-17 09:49:35.000000000 +0000
@@ -31,6 +31,7 @@ if (WITH_MAPNIK)
   find_package(Boost COMPONENTS filesystem system thread REQUIRED)
   find_package(Mapnik REQUIRED)
   add_definitions(-DMAPNIK)
+  add_definitions(-DBOOST_FILESYSTEM_VERSION=2)
 endif (WITH_MAPNIK)

 if (WITH_SPEECH)
