--- lxpanel-0.6.1/data/default/panels/panel.in	2013-03-08 05:28:01.000000000 +0100
+++ lxpanel-0.6.1b/data/default/panels/panel.in	2013-11-23 10:47:12.570514116 +0100
@@ -16,8 +16,8 @@
     usefontcolor=1
     fontcolor=#ffffff
     usefontsize=0
-    fontsize=10    
-    background=1
+    fontsize=10
+    background=0
     backgroundfile=@prefix@/share/lxpanel/images/background.png
 }
 
@@ -31,7 +31,7 @@
 Plugin {
     type = menu
     Config {
-        image=@prefix@/share/lxpanel/images/my-computer.png
+        image=@prefix@/share/lxpanel/images/frugalware-logo.png
         system {
         }
         separator {
@@ -55,10 +55,7 @@
             id=pcmanfm.desktop
         }
         Button {
-            id=gnome-terminal.desktop
-        }
-        Button {
-            id=firefox.desktop
+            id=lxterminal.desktop
         }
     }
 }
