diff -Naur lxpanel-0.5.8/configure.ac lxpanel-0.5.8.new/configure.ac 
--- lxpanel-0.5.8/configure.ac	2011-07-29 13:37:01.000000000 +0100
+++ lxpanel-0.5.8.new/configure.ac	2013-03-22 11:06:24.584834508 +0100
@@ -24,7 +24,8 @@
 # Checks for libraries.
 pkg_modules="gtk+-2.0 >= 2.12.0 \
              gio-unix-2.0 \
-             gthread-2.0"
+             gthread-2.0
+	     gmodule-2.0"
 #             libstartup-notification-1.0"
 PKG_CHECK_MODULES(PACKAGE, [$pkg_modules])
 AC_SUBST(PACKAGE_CFLAGS)
