--- kipi-plugins-0.1.3/kipi-plugins/common/exiv2iface/exiv2iface.cpp.ark	2007-03-21 14:44:45.000000000 +0100
+++ kipi-plugins-0.1.3/kipi-plugins/common/exiv2iface/exiv2iface.cpp	2007-03-21 14:44:54.000000000 +0100
@@ -97,7 +97,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot clear Exif data using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }    
 
@@ -114,7 +114,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot clear Iptc data using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }    
 
@@ -153,7 +153,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot get Exif data using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }       
     
@@ -192,7 +192,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot get Iptc data using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }       
     
@@ -214,7 +214,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set Exif data using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
 
@@ -236,7 +236,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set Iptc data using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }    
 
@@ -271,7 +271,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot load metadata using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
         return false;
     }
@@ -329,7 +329,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot save metadata using Exiv2 ("
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
         return false;
     }
@@ -362,7 +362,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set Program identity into image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
@@ -382,7 +382,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set image dimensions using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -404,7 +404,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set Exif Thumbnail using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -434,7 +434,7 @@
     {
         kdDebug() << "Cannot find Exif key '"
                   << exifTagName << "' into image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
@@ -451,7 +451,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set Exif tag string into image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
@@ -481,7 +481,7 @@
     {
         kdDebug() << "Cannot find Iptc key '"
                   << iptcTagName << "' into image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
@@ -498,7 +498,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set Iptc tag string into image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
@@ -522,7 +522,7 @@
     {
         kdDebug() << "Cannot find Exif key '"
                   << exifTagName << "' into image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -547,7 +547,7 @@
     {
         kdDebug() << "Cannot find Exif key '"
                   << exifTagName << "' into image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
@@ -572,7 +572,7 @@
     {
         kdDebug() << "Cannot find Iptc key '"
                   << iptcTagName << "' into image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
@@ -597,7 +597,7 @@
     {
         kdDebug() << "Cannot find Exif Rational value from key '"
                   << exifTagName << "' into image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
@@ -614,7 +614,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set Exif tag long value into image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
@@ -631,7 +631,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set Exif tag rational value into image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
@@ -653,7 +653,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot remove Exif tag using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -675,7 +675,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot remove Iptc tag using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -717,7 +717,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot get image preview using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
@@ -847,7 +847,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot parse Exif date & time tag using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -940,7 +940,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot parse Exif Orientation tag using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
@@ -1007,7 +1007,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set Exif Orientation tag using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -1107,7 +1107,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot get Exif GPS tag using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -1229,7 +1229,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set Exif GPS tag using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -1264,7 +1264,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot remove Exif GPS tag using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -1357,7 +1357,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot get IPTC Keywords from image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -1407,7 +1407,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set IPTC Keywords into image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -1440,7 +1440,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot get IPTC Subjects from image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -1490,7 +1490,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set IPTC Subjects into image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -1523,7 +1523,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot get IPTC Sub Categories from image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -1573,7 +1573,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set IPTC Sub Categories into image using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }        
     
@@ -1603,7 +1603,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot find Exif User Comment using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
@@ -1645,7 +1645,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot set Exif Comment using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
@@ -1718,7 +1718,7 @@
     catch( Exiv2::Error &e )
     {
         kdDebug() << "Cannot convert Comment using Exiv2 (" 
-                  << QString::fromLocal8Bit(e.what().c_str())
+                  << QString::fromLocal8Bit(e.what())
                   << ")" << endl;
     }
 
