--- scripts/postprocess-unix.sh.old	2015-12-19 17:04:54.770863096 +0100
+++ scripts/postprocess-unix.sh	2015-12-19 17:05:49.539823949 +0100
@@ -9,13 +9,13 @@
 TSDIRPI="${PIDIR}/ts"
 SPTDIR="${LCDIR}/support"
 DOCDIR="${SPTDIR}/doc"
-LRELEASE="lrelease"
+LRELEASE="lrelease-qt5"
 
 [ "$( which lrelease-qt4 )" ] && LRELEASE="lrelease-qt4"
 
 # Generate Help Files
 cd "${DOCDIR}"
-qcollectiongenerator LibreCADdoc.qhcp
+qcollectiongenerator-qt5 LibreCADdoc.qhcp
 
 cd "${THISDIR}"
 
