diff -Naur LibreCAD-2.1.3/librecad/src/ui/forms/qg_commandwidget.cpp LibreCAD-2.1.3-p/librecad/src/ui/forms/qg_commandwidget.cpp
--- LibreCAD-2.1.3/librecad/src/ui/forms/qg_commandwidget.cpp	2016-09-23 07:34:43.000000000 +0200
+++ LibreCAD-2.1.3-p/librecad/src/ui/forms/qg_commandwidget.cpp	2018-06-14 20:57:22.720317068 +0200
@@ -24,6 +24,7 @@
 **
 **********************************************************************/
 #include "qg_commandwidget.h"
+#include <QAction>
 #include <QKeyEvent>
 #include <algorithm>
 
diff -Naur LibreCAD-2.1.3/librecad/src/ui/generic/widgetcreator.cpp LibreCAD-2.1.3-p/librecad/src/ui/generic/widgetcreator.cpp
--- LibreCAD-2.1.3/librecad/src/ui/generic/widgetcreator.cpp	2016-09-23 07:34:43.000000000 +0200
+++ LibreCAD-2.1.3-p/librecad/src/ui/generic/widgetcreator.cpp	2018-06-14 20:57:58.504529906 +0200
@@ -27,6 +27,8 @@
 #include "widgetcreator.h"
 #include "ui_widgetcreator.h"
 
+#include <QAction>
+#include <QActionGroup>
 #include <QSettings>
 #include <QLineEdit>
 #include <QPushButton>
