diff -Naur libvirt-0.8.8.orig//configure.ac libvirt-0.8.8/configure.ac
--- libvirt-0.8.8.orig//configure.ac	2011-02-17 04:00:39.000000000 +0100
+++ libvirt-0.8.8/configure.ac	2011-02-23 13:15:14.000000000 +0100
@@ -332,13 +332,13 @@
 AC_PATH_PROG([IP_PATH], [ip], /sbin/ip, [/usr/sbin:$PATH])
 AC_DEFINE_UNQUOTED([IP_PATH], "$IP_PATH", [path to ip binary])
 
-AC_PATH_PROG([IPTABLES_PATH], [iptables], /sbin/iptables, [/usr/sbin:$PATH])
+AC_PATH_PROG([IPTABLES_PATH], [iptables], /usr/sbin/iptables, [/usr/sbin:$PATH])
 AC_DEFINE_UNQUOTED([IPTABLES_PATH], "$IPTABLES_PATH", [path to iptables binary])
 
-AC_PATH_PROG([IP6TABLES_PATH], [ip6tables], /sbin/ip6tables, [/usr/sbin:$PATH])
+AC_PATH_PROG([IP6TABLES_PATH], [ip6tables], /usr/sbin/ip6tables, [/usr/sbin:$PATH])
 AC_DEFINE_UNQUOTED([IP6TABLES_PATH], "$IP6TABLES_PATH", [path to ip6tables binary])
 
-AC_PATH_PROG([EBTABLES_PATH], [ebtables], /sbin/ebtables, [/usr/sbin:$PATH])
+AC_PATH_PROG([EBTABLES_PATH], [ebtables], /usr/sbin/ebtables, [/usr/sbin:$PATH])
 AC_DEFINE_UNQUOTED([EBTABLES_PATH], "$EBTABLES_PATH", [path to ebtables binary])
 
 
