diff --git a/src/qemu/qemu_cgroup.c b/src/qemu/qemu_cgroup.c
index e71d3fa..44211a6 100644
--- a/src/qemu/qemu_cgroup.c
+++ b/src/qemu/qemu_cgroup.c
@@ -46,7 +46,7 @@ static const char *const defaultDeviceACL[] = {
 int qemuCgroupControllerActive(struct qemud_driver *driver,
                                int controller)
 {
-    if (driver->cgroup == NULL)
+    /*if (driver->cgroup == NULL)*/
         return 0;
     if (!virCgroupMounted(driver->cgroup, controller))
         return 0;
@@ -202,7 +202,7 @@ int qemuSetupCgroup(struct qemud_driver *driver,
         (const char *const *)driver->cgroupDeviceACL :
         defaultDeviceACL;
 
-    if (driver->cgroup == NULL)
+    /*if (driver->cgroup == NULL)*/
         return 0; /* Not supported, so claim success */
 
     rc = virCgroupForDomain(driver->cgroup, vm->def->name, &cgroup, 1);
@@ -364,7 +364,7 @@ int qemuRemoveCgroup(struct qemud_driver *driver,
     virCgroupPtr cgroup;
     int rc;
 
-    if (driver->cgroup == NULL)
+    /*if (driver->cgroup == NULL)*/
         return 0; /* Not supported, so claim success */
 
     rc = virCgroupForDomain(driver->cgroup, vm->def->name, &cgroup, 0);
@@ -388,7 +388,7 @@ int qemuAddToCgroup(struct qemud_driver *driver,
     int ret = -1;
     int rc;
 
-    if (driver->cgroup == NULL)
+    /*if (driver->cgroup == NULL)*/
         return 0; /* Not supported, so claim success */
 
     rc = virCgroupForDomain(driver->cgroup, def->name, &cgroup, 0);
