diff -ur a/daemon/libvirtd.service.in b/daemon/libvirtd.service.in
--- a/daemon/libvirtd.service.in	2012-07-12 14:00:01.267787136 -0500
+++ b/daemon/libvirtd.service.in	2012-07-12 14:00:17.267416163 -0500
@@ -6,9 +6,6 @@
 [Unit]
 Description=Virtualization daemon
 After=syslog.target
-After=udev.target
-After=avahi.target
-After=dbus.target
 Before=libvirt-guests.service
 
 [Service]
diff -ur a/tools/libvirt-guests.service.in b/tools/libvirt-guests.service.in
--- a/tools/libvirt-guests.service.in	2012-07-12 14:00:01.317785977 -0500
+++ b/tools/libvirt-guests.service.in	2012-07-12 16:35:14.230173887 -0500
@@ -6,8 +6,10 @@
 EnvironmentFile=-/etc/sysconfig/libvirt-guests
 # Hack just call traditional service until we factor
 # out the code
-ExecStart=/etc/init.d/libvirt-guests start
-ExecStop=/etc/init.d/libvirt-guests stop
+ExecStart=/usr/libexec/libvirt-guests start
+ExecStop=/usr/libexec/libvirt-guests stop
+Type=oneshot
+RemainAfterExit=yes
 
 [Install]
 WantedBy=multi-user.target
