--- libvirt-2.1.0/tools/Makefile.am	2016-04-26 05:20:09.426241779 +0300
+++ libvirt-2.1.0.new/tools/Makefile.am	2016-08-11 22:51:41.546543058 +0300
@@ -367,7 +367,7 @@
 
 
 EXTRA_DIST += libvirt-guests.service.in
-SYSTEMD_UNIT_DIR = $(prefix)/lib/systemd/system
+SYSTEMD_UNIT_DIR = /lib/systemd/system
 
 if LIBVIRT_INIT_SCRIPT_SYSTEMD
 install-systemd: libvirt-guests.service install-sysconfig libvirt-guests.sh
--- libvirt-2.1.0/daemon/Makefile.am	2016-04-26 05:20:09.091245697 +0300
+++ libvirt-2.1.0.new/daemon/Makefile.am	2016-08-11 23:00:00.708231801 +0300
@@ -449,7 +449,7 @@
 
 if LIBVIRT_INIT_SCRIPT_SYSTEMD
 
-SYSTEMD_UNIT_DIR = $(prefix)/lib/systemd/system
+SYSTEMD_UNIT_DIR = /lib/systemd/system
 BUILT_SOURCES += libvirtd.service
 
 install-init-systemd: install-sysconfig libvirtd.service
--- libvirt-2.1.0/src/Makefile.am	2016-07-27 15:39:35.298706138 +0300
+++ libvirt-2.1.0.new/src/Makefile.am	2016-08-11 23:09:22.347367775 +0300
@@ -2650,7 +2650,7 @@
 if WITH_LIBVIRTD
 if LIBVIRT_INIT_SCRIPT_SYSTEMD
 
-SYSTEMD_UNIT_DIR = $(prefix)/lib/systemd/system
+SYSTEMD_UNIT_DIR = /lib/systemd/system
 
 BUILT_SOURCES += virtlockd.service virtlockd.socket \
 		 virtlogd.service virtlogd.socket
