--- libvirt-1.2.15/tools/Makefile.am	2015-01-23 13:46:24.000000000 +0200
+++ libvirt-1.2.15.new/tools/Makefile.am	2015-05-13 00:06:22.000000000 +0300
@@ -308,7 +308,7 @@
 
 
 EXTRA_DIST += libvirt-guests.service.in
-SYSTEMD_UNIT_DIR = $(prefix)/lib/systemd/system
+SYSTEMD_UNIT_DIR = /lib/systemd/system
 
 if LIBVIRT_INIT_SCRIPT_SYSTEMD
 install-systemd: libvirt-guests.service install-sysconfig libvirt-guests.sh
--- libvirt-1.2.15/daemon/Makefile.am	2015-04-28 04:15:40.000000000 +0300
+++ libvirt-1.2.15.new/daemon/Makefile.am	2015-05-13 00:07:20.000000000 +0300
@@ -387,7 +387,7 @@
 
 if LIBVIRT_INIT_SCRIPT_SYSTEMD
 
-SYSTEMD_UNIT_DIR = $(prefix)/lib/systemd/system
+SYSTEMD_UNIT_DIR = /lib/systemd/system
 BUILT_SOURCES += libvirtd.service libvirtd.socket
 
 install-init-systemd: install-sysconfig libvirtd.service libvirtd.socket
--- libvirt-1.2.15/src/Makefile.am	2015-04-30 12:04:46.000000000 +0300
+++ libvirt-1.2.15.new/src/Makefile.am	2015-05-13 00:08:11.000000000 +0300
@@ -2379,7 +2379,7 @@
 if WITH_LIBVIRTD
 if LIBVIRT_INIT_SCRIPT_SYSTEMD
 
-SYSTEMD_UNIT_DIR = $(prefix)/lib/systemd/system
+SYSTEMD_UNIT_DIR = /lib/systemd/system
 
 BUILT_SOURCES += virtlockd.service virtlockd.socket
 DISTCLEANFILES += virtlockd.service virtlockd.socket
