diff -Naur kdebase-workspace-4.3.4/kwin/compositingprefs.cpp kdebase-workspace-4.3.4-p/kwin/compositingprefs.cpp
--- kdebase-workspace-4.3.4/kwin/compositingprefs.cpp	2009-06-17 22:07:38.000000000 +0200
+++ kdebase-workspace-4.3.4-p/kwin/compositingprefs.cpp	2010-01-20 16:51:33.000000000 +0100
@@ -270,6 +270,9 @@
         {
         mDriver = "radeon";
         mVersion = Version( mGLRenderer.split(" ")[ 3 ] );
+        // Check that the version string is changed, and try the fifth element if it does
+        if (!mVersion.startsWith("20"))
+            mVersion = Version( mGLRenderer.split(' ')[ 5 ] );
         }
     else if( mGLRenderer == "Software Rasterizer" )
         {
@@ -286,6 +289,32 @@
 #endif
     }
 
+void CompositingPrefs::parseMesaVersion( const QString &version, int *major, int *minor )
+    {
+    *major = 0;
+    *minor = 0;
+
+    const QStringList tokens = version.split( ' ' );
+    int token = 0;
+    while( token < tokens.count() && !tokens.at( token ).endsWith( "Mesa" ) )
+        token++;
+
+    if( token < tokens.count() - 1 )
+        {
+        const QStringList version = tokens.at( token + 1 ).split( '.' );
+        if( version.count() >= 2 )
+            {
+            *major = version[ 0 ].toInt();
+
+            int end = 0;
+            while( end < version[ 1 ].length() && version[ 1 ][ end ].isDigit() )
+                end++;
+
+            *minor = version[ 1 ].left( end ).toInt();
+            }
+        }
+    }
+
 // See http://techbase.kde.org/Projects/KWin/HW for a list of some cards that are known to work.
 void CompositingPrefs::applyDriverSpecificOptions()
     {
@@ -324,12 +353,28 @@
             }
         }
     else if( mDriver == "radeon" )
-        { // radeon r200 only ?
+        {
         if( mGLRenderer.startsWith( "Mesa DRI R200" ) && mVersion >= Version( "20060602" ))
             {
             kDebug( 1212 ) << "radeon r200 >= 20060602, enabling compositing";
             mEnableCompositing = true;
             }
+        if( mGLRenderer.startsWith( "Mesa DRI R300" ) && mVersion >= Version( "20090101" )) // krazy:exclude=strings
+            {
+            kDebug( 1212 ) << "radeon r300 >= 20090101, enabling compositing";
+            mEnableCompositing = true;
+            }
+        if( mGLRenderer.startsWith( "Mesa DRI R600" ) )
+            {
+            // Enable compositing with Mesa 7.7 or later
+            int major, minor;
+            parseMesaVersion( mGLVersion, &major, &minor );
+            if( major > 7 || ( major == 7 && minor >= 7 ) )
+                {
+                kDebug( 1212 ) << "Radeon R600/R700, Mesa 7.7 or better. Enabling compositing.";
+                mEnableCompositing = true;
+                }
+            }
         }
     else if( mDriver == "fglrx" )
         { // radeon r200 only ?
diff -Naur kdebase-workspace-4.3.4/kwin/compositingprefs.h kdebase-workspace-4.3.4-p/kwin/compositingprefs.h
--- kdebase-workspace-4.3.4/kwin/compositingprefs.h	2008-10-17 13:00:26.000000000 +0200
+++ kdebase-workspace-4.3.4-p/kwin/compositingprefs.h	2010-01-20 16:51:33.000000000 +0100
@@ -69,6 +69,7 @@
 protected:
 
     void detectDriverAndVersion();
+    void parseMesaVersion( const QString &version, int *major, int *minor );
     void applyDriverSpecificOptions();
     static bool detectXgl();
 
