Index: kdebase-workspace-4.3.4/kcontrol/fonts/kxftconfig.h
===================================================================
--- kdebase-workspace-4.3.4.orig/kcontrol/fonts/kxftconfig.h
+++ kdebase-workspace-4.3.4/kcontrol/fonts/kxftconfig.h
@@ -80,7 +80,7 @@ class KXftConfig
         SubPixel(Type t, QDomNode &n) : Item(n), type(t) {}
         SubPixel(Type t=None)         : type(t)          {}
 
-        void reset() { Item::reset(); type=None; }
+        void reset() { Item::reset(); type=Rgb; }
 
         Type type;
     };
@@ -110,7 +110,7 @@ class KXftConfig
         Hint(Style s, QDomNode &n) : Item(n), style(s) {}
         Hint(Style s=NotSet)       : style(s)          {}
 
-        void reset() { Item::reset(); style=NotSet; }
+        void reset() { Item::reset(); style=Full; }
 
         Style style;
     };
