--- kde-workspace-4.8.2/libs/plasmagenericshell/panelshadows.cpp	2012-03-29 22:02:31.000000000 +0000
+++ kde-workspace-4.8.2.new/libs/plasmagenericshell/panelshadows.cpp	2012-05-03 22:10:07.000000000 +0000
@@ -122,6 +122,10 @@
 {
 #ifdef Q_WS_X11
     QPixmap pix = q->pixmap(element);
+    if(pix.width()<=0 || pix.height()<=0) {
+    	kError()<<"could not load "<<element<<" check your theme";
+    	return;
+    }
     if (pix.handle() == 0) {
         Pixmap xPix = XCreatePixmap(QX11Info::display(), QX11Info::appRootWindow(), pix.width(), pix.height(), 32);
         QPixmap tempPix = QPixmap::fromX11Pixmap(xPix, QPixmap::ExplicitlyShared);
