diff -Naur kdebase-workspace-4.3.98/kcontrol/krdb/themes/CMakeLists.txt kdebase-workspace-4.3.98-p/kcontrol/krdb/themes/CMakeLists.txt
--- kdebase-workspace-4.3.98/kcontrol/krdb/themes/CMakeLists.txt	2008-01-05 00:55:13.000000000 +0100
+++ kdebase-workspace-4.3.98-p/kcontrol/krdb/themes/CMakeLists.txt	2010-02-08 10:28:56.000000000 +0100
@@ -1,12 +1,12 @@
-add_subdirectory( HighContrastDark ) 
-add_subdirectory( HighContrastDark-big ) 
-add_subdirectory( HighContrastLight ) 
-add_subdirectory( HighContrastLight-big ) 
-add_subdirectory( YellowOnBlue ) 
-add_subdirectory( YellowOnBlue-big ) 
-add_subdirectory( KDE_Classic ) 
-add_subdirectory( Keramik ) 
+#add_subdirectory( HighContrastDark ) 
+#add_subdirectory( HighContrastDark-big ) 
+#add_subdirectory( HighContrastLight ) 
+#add_subdirectory( HighContrastLight-big ) 
+#add_subdirectory( YellowOnBlue ) 
+#add_subdirectory( YellowOnBlue-big ) 
+#add_subdirectory( KDE_Classic ) 
+#add_subdirectory( Keramik ) 
 add_subdirectory( Plastik ) 
-add_subdirectory( Platinum ) 
-add_subdirectory( Sunshine ) 
-add_subdirectory( Redmond ) 
+#add_subdirectory( Platinum ) 
+#add_subdirectory( Sunshine ) 
+#add_subdirectory( Redmond ) 
diff -Naur kdebase-workspace-4.3.98/kcontrol/style/CMakeLists.txt kdebase-workspace-4.3.98-p/kcontrol/style/CMakeLists.txt
--- kdebase-workspace-4.3.98/kcontrol/style/CMakeLists.txt	2009-11-02 13:01:14.000000000 +0100
+++ kdebase-workspace-4.3.98-p/kcontrol/style/CMakeLists.txt	2010-02-08 10:20:51.000000000 +0100
@@ -1,5 +1,5 @@
 KDE4_NO_ENABLE_FINAL(styles)
-add_subdirectory( keramik )
+#add_subdirectory( keramik )
 
 include_directories( ${QIMAGEBLITZ_INCLUDES} )
 
