diff -aur a/kcontrol/krdb/krdb.cpp b/kcontrol/krdb/krdb.cpp
--- a/kcontrol/krdb/krdb.cpp	2011-05-20 15:32:06.000000000 -0500
+++ b/kcontrol/krdb/krdb.cpp	2011-09-01 01:13:16.379657400 -0500
@@ -509,7 +509,10 @@
     {
       QString subPixel = generalCfgGroup.readEntry("XftSubPixel");
       if(!subPixel.isEmpty())
+      {
         contents += "Xft.rgba: " + subPixel + '\n';
+        contents += "Xft.lcdfilter: lcddefault\n";
+      }
     }
 
     KConfig _cfgfonts( "kcmfonts" );
