diff -Naur bcm43xx-fwcutter-004.orig/Makefile bcm43xx-fwcutter-004/Makefile
--- bcm43xx-fwcutter-004.orig/Makefile	2006-03-25 18:32:24.000000000 +0100
+++ bcm43xx-fwcutter-004/Makefile	2006-06-11 12:48:09.000000000 +0200
@@ -3,8 +3,9 @@
 FIRMWARE_INSTALL_DIR ?= /lib/firmware
 
 CC = cc
-PREFIX = /usr/local
-CFLAGS = -std=c99 -O2 -fomit-frame-pointer -Wall -pedantic -D_BSD_SOURCE
+PREFIX ?= /usr/local
+CFLAGS ?= -O2 -fomit-frame-pointer -Wall -pedantic
+CFLAGS += -D_BSD_SOURCE
 LDFLAGS =
 
 
@@ -18,13 +19,16 @@
 	$(CC) $(CFLAGS) -o bcm43xx-fwcutter $(OBJECTS) $(LDFLAGS)
 
 install: all
+	-install -d -o 0 -g 0 -m 755 $(PREFIX)/bin/
 	-install -o 0 -g 0 -m 755 bcm43xx-fwcutter $(PREFIX)/bin/
+	-install -d -o 0 -g 0 -m 755 $(PREFIX)/man/man1
+	-install -o 0 -g 0 -m 644 bcm43xx-fwcutter.1 $(PREFIX)/man/man1/
 
 clean:
 	-rm -f *~ *.o *.orig *.rej *.fw bcm43xx-fwcutter
 
 installfw:
-	-if ! [ -d $(FIRMWARE_INSTALL_DIR) ]; then mkdir $(FIRMWARE_INSTALL_DIR); fi
+	-if ! [ -d $(FIRMWARE_INSTALL_DIR) ]; then mkdir -p $(FIRMWARE_INSTALL_DIR); fi
 	-install -o 0 -g 0 -m 600 bcm43xx_*.fw $(FIRMWARE_INSTALL_DIR)
 
 # dependencies
