diff -Naur enchant-2.2.8/src/enchant.1.in enchant-2.2.8.new/src/enchant.1.in
--- enchant-2.2.8/src/enchant.1.in	2020-02-27 14:05:18.000000000 +0100
+++ enchant-2.2.8.new/src/enchant.1.in	2020-03-06 19:32:41.644633169 +0100
@@ -29,7 +29,7 @@
 .B "\-v"
 display version information and exit
 .SH ENCHANT ORDERING FILE
-Enchant uses global and per-user ordering files named \fIenchant.ordering\fR
+Enchant uses global and per-user ordering files named \fIenchant2.ordering\fR
 to decide which spelling provider to use for particular languages.
 The per-user file takes precedence.
 .PP
diff -Naur enchant-2.2.8/src/enchant2.ordering enchant-2.2.8.new/src/enchant2.ordering
--- enchant-2.2.8/src/enchant2.ordering	1970-01-01 01:00:00.000000000 +0100
+++ enchant-2.2.8.new/src/enchant2.ordering	2020-03-06 19:32:41.644633169 +0100
@@ -0,0 +1,7 @@
+*:hunspell,aspell
+fi:voikko,hunspell,aspell
+fi_FI:voikko,hunspell,aspell
+he:hspell,hunspell
+he_IL:hspell,hunspell
+tr:zemberek
+tr_TR:zemberek
diff -Naur enchant-2.2.8/src/lib.c enchant-2.2.8.new/src/lib.c
--- enchant-2.2.8/src/lib.c	2019-07-01 12:29:59.000000000 +0200
+++ enchant-2.2.8.new/src/lib.c	2020-03-06 19:32:50.529690947 +0100
@@ -921,7 +921,7 @@
 	GSList *conf_dirs = enchant_get_conf_dirs ();
 	for (GSList *iter = conf_dirs; iter; iter = iter->next)
 		{
-			char *ordering_file = g_build_filename (iter->data, "enchant.ordering", NULL);
+			char *ordering_file = g_build_filename (iter->data, "enchant2.ordering", NULL);
 			enchant_load_ordering_from_file (broker, ordering_file);
 			g_free (ordering_file);
 		}
diff -Naur enchant-2.2.8/src/Makefile.am enchant-2.2.8.new/src/Makefile.am
--- enchant-2.2.8/src/Makefile.am	2018-01-24 14:26:49.000000000 +0100
+++ enchant-2.2.8.new/src/Makefile.am	2020-03-06 19:32:50.530690954 +0100
@@ -20,7 +20,7 @@
 libenchant_includedir = $(pkgincludedir)-@ENCHANT_MAJOR_VERSION@
 libenchant_include_HEADERS = enchant.h enchant-provider.h enchant++.h
 
-pkgdata_DATA = enchant.ordering
+pkgdata_DATA = enchant2.ordering
 
 dist_man_MANS = enchant-@ENCHANT_MAJOR_VERSION@.1 enchant-lsmod-@ENCHANT_MAJOR_VERSION@.1
 
diff -Naur enchant-2.2.8/tests/broker/enchant_broker_set_ordering_tests.cpp enchant-2.2.8.new/tests/broker/enchant_broker_set_ordering_tests.cpp
--- enchant-2.2.8/tests/broker/enchant_broker_set_ordering_tests.cpp	2020-02-27 14:05:18.000000000 +0100
+++ enchant-2.2.8.new/tests/broker/enchant_broker_set_ordering_tests.cpp	2020-03-06 19:32:50.530690954 +0100
@@ -122,7 +122,7 @@
 	static void WriteStringToOrderingFile(const std::string& path, const std::string& contents)
 	{
 		CreateDirectory(path);
-		std::ofstream file(AddToPath(path, "enchant.ordering").c_str());
+		std::ofstream file(AddToPath(path, "enchant2.ordering").c_str());
 		file << contents;
 	}
 
@@ -344,7 +344,7 @@
 }
 
 /*
- * Ordering can also be set in enchant.ordering file:
+ * Ordering can also be set in enchant2.ordering file:
  * Language_Tag : Provider1, Provider2, ProviderN
  *
  * See enchant(1) for details of how the file is located.
