diff -Naur enchant-1.6.0/src/myspell/myspell_checker.cpp enchant-1.6.0-p/src/myspell/myspell_checker.cpp
--- enchant-1.6.0/src/myspell/myspell_checker.cpp	2010-04-01 22:53:37.000000000 +0200
+++ enchant-1.6.0-p/src/myspell/myspell_checker.cpp	2016-05-04 16:24:11.604863953 +0200
@@ -50,6 +50,12 @@
 #include <hunspell/hunspell.hxx>
 #endif
 
+/* hunspell >=1.4.0 */
+
+#ifndef MAXWORDLEN
+#   define MAXWORDLEN 100
+#endif
+
 ENCHANT_PLUGIN_DECLARE("Myspell")
 
 #define G_ICONV_INVALID (GIConv)-1
