diff -ur a/docs/kmscon.service b/docs/kmscon.service
--- a/docs/kmscon.service	2012-09-18 13:41:49.551382066 -0500
+++ b/docs/kmscon.service	2012-12-12 22:18:35.781259350 -0600
@@ -1,9 +1,12 @@
 [Unit]
 Description=KMS System Console
 Documentation=man:kmscon(1)
+After=systemd-user-sessions.service
+After=plymouth-quit-wait.service
+After=local.service
 
 [Service]
-ExecStart=/usr/bin/kmscon -l /bin/login
+ExecStart=/usr/bin/kmscon --vt=tty1 --seats=seat0 --no-switchvt --session-control
 
 [Install]
-WantedBy=multi-user.target
+WantedBy=graphical.target
diff -ur a/src/kmscon_conf.c b/src/kmscon_conf.c
--- a/src/kmscon_conf.c	2012-12-09 11:32:19.038735514 -0600
+++ b/src/kmscon_conf.c	2012-12-12 22:19:44.372264599 -0600
@@ -136,9 +136,9 @@
 		"\t    --render-timing         [off]   Print renderer timing information\n"
 		"\n"
 		"Font Options:\n"
-		"\t    --font-engine <engine>  [pango]\n"
+		"\t    --font-engine <engine>  [freetype2]\n"
 		"\t                              Font engine\n"
-		"\t    --font-size <points>    [15]\n"
+		"\t    --font-size <points>    [12]\n"
 		"\t                              Font size in points\n"
 		"\t    --font-name <name>      [monospace]\n"
 		"\t                              Font name\n"
@@ -589,7 +589,7 @@
 		CONF_OPTION_BOOL(0, "render-timing", &conf->render_timing, false),
 
 		/* Font Options */
-		CONF_OPTION_STRING(0, "font-engine", &conf->font_engine, "pango"),
+		CONF_OPTION_STRING(0, "font-engine", &conf->font_engine, "freetype2"),
 		CONF_OPTION_UINT(0, "font-size", &conf->font_size, 12),
 		CONF_OPTION_STRING(0, "font-name", &conf->font_name, "monospace"),
 		CONF_OPTION_UINT(0, "font-dpi", &conf->font_ppi, 96),
