--- stepmania-5.0.10/src/CMakeLists.txt	2015-10-30 07:55:52.000000000 +0100
+++ stepmania-5.0.10.new/src/CMakeLists.txt	2015-12-13 21:38:54.000000000 +0100
@@ -295,11 +295,9 @@
     if(CMAKE_SYSTEM_PROCESSOR MATCHES "x86_64")
       message("Host processor is 64bit")
       sm_add_compile_definition("${SM_EXE_NAME}" CPU_X86_64)
-    elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "x86")
+    else()
       message("Host processor is 32bit")
       sm_add_compile_definition("${SM_EXE_NAME}" CPU_X86)
-    else()
-      message("Unrecognized host processor type")
     endif()
   endif()
   if (${HAS_FFMPEG})
