--- stepmania-5.0.10/src/RageSurface_Save_JPEG.cpp	2015-10-30 07:55:52.000000000 +0100
+++ stepmania-5.0.10.new/src/RageSurface_Save_JPEG.cpp	2016-02-07 21:28:07.000000000 +0100
@@ -45,7 +45,7 @@
 	dest->pub.next_output_byte = dest->buffer;
 	dest->pub.free_in_buffer = OUTPUT_BUFFER_SIZE;
 
-	return TRUE;
+	return (jpeg::boolean)1;
 }
 
 
@@ -114,14 +114,14 @@
 	jpeg::jpeg_set_defaults(&cinfo);
 
 	if( bHighQual )
-		jpeg::jpeg_set_quality( &cinfo, 150, TRUE );
+		jpeg::jpeg_set_quality( &cinfo, 150, (jpeg::boolean)1 );
 	else
-		jpeg::jpeg_set_quality( &cinfo, 70, TRUE );
+		jpeg::jpeg_set_quality( &cinfo, 70, (jpeg::boolean)1 );
 
 	jpeg_RageFile_dest( &cinfo, f );
 
 	/* Start the compressor. */
-	jpeg::jpeg_start_compress( &cinfo, TRUE );
+	jpeg::jpeg_start_compress( &cinfo, (jpeg::boolean)1 );
 
 	/* Here we use the library's state variable cinfo.next_scanline as the
 	 * loop counter, so that we don't have to keep track ourselves.
