--- adjtimex-1.20/adjtimex.c.glibc	2004-09-19 17:37:29.000000000 +0200
+++ adjtimex-1.20/adjtimex.c	2005-03-16 13:16:02.514052584 +0100
@@ -35,15 +35,7 @@
 #include <sys/ioctl.h>
 #include <linux/rtc.h>
 
-#ifdef __alpha__
 extern int adjtimex(struct timex *);
-#else
-#ifdef __ia64__
-extern int adjtimex(struct timex *);
-#else
-_syscall1(int, adjtimex, struct timex *, txcp)
-#endif
-#endif
 int F_print = 0;
 
 #ifndef LOG_PATH
