diff -Naur MPlayer-1.0pre7.orig/etc/example.conf MPlayer-1.0pre7/etc/example.conf
--- MPlayer-1.0pre7.orig/etc/example.conf	2005-03-03 04:11:44.000000000 +0100
+++ MPlayer-1.0pre7/etc/example.conf	2005-06-15 20:11:23.216952224 +0200
@@ -157,6 +157,7 @@
 ## configuration file. In the command line you use:
 ## -aop list=resample:fout=44100 , but here it is:
 # aop=list=resample:fout=44100
+stop-xscreensaver=1
 
 ##
 ## You can also include other configfiles
