--- MPlayer-1.0pre8/etc/mplayer.desktop.orig	2006-06-12 01:57:45.000000000 +0200
+++ MPlayer-1.0pre8/etc/mplayer.desktop	2006-06-12 01:58:48.000000000 +0200
@@ -3,10 +3,16 @@
 Encoding=UTF-8
 Name=MPlayer
 GenericName=Multimedia player
+GenericName[de]=Multimedia Player
+GenericName[es]=Reproductor multimedia
+GenericName[fr]=Lecteur multimedia
+GenericName[hu]=Multimédia-lejátszó
+GenericName[it]=Lettore multimediale
 Comment=Multimedia player
 Comment[de]=Multimedia-Player
 Comment[es]=Reproductor multimedia
 Comment[fr]=Lecteur multimedia
+Comment[hu]=Multimédia-lejátszó
 Comment[it]=Lettore multimediale
 Icon=mplayer.xpm
 TryExec=gmplayer
