diff -Naur MPlayer-1.0pre8.orig/libavcodec/x264.c MPlayer-1.0pre8/libavcodec/x264.c
--- MPlayer-1.0pre8.orig/libavcodec/x264.c	2006-09-05 15:07:05.000000000 +0200
+++ MPlayer-1.0pre8/libavcodec/x264.c	2006-09-05 15:07:38.000000000 +0200
@@ -148,7 +148,7 @@
 
     // if neither crf nor cqp modes are selected we have to enable the RC
     // we do it this way because we cannot check if the bitrate has been set
-    if(!(avctx->crf || (avctx->cqp > -1))) x4->params.rc.b_cbr = 1;
+    /* Removed */
 
     x4->params.i_bframe = avctx->max_b_frames;
     x4->params.b_cabac = avctx->coder_type == FF_CODER_TYPE_AC;
diff -Naur MPlayer-1.0pre8.orig/libmpcodecs/ve_x264.c MPlayer-1.0pre8/libmpcodecs/ve_x264.c
--- MPlayer-1.0pre8.orig/libmpcodecs/ve_x264.c	2006-09-05 15:07:05.000000000 +0200
+++ MPlayer-1.0pre8/libmpcodecs/ve_x264.c	2006-09-05 15:07:17.000000000 +0200
@@ -275,7 +275,7 @@
                    "VBV requires both vbv_maxrate and vbv_bufsize.\n");
             return 0;
         }
-        mod->param.rc.b_cbr = 1;
+        /* Removed */
         mod->param.rc.i_bitrate = bitrate;
         mod->param.rc.f_rate_tolerance = ratetol;
         mod->param.rc.i_vbv_max_bitrate = vbv_maxrate;
