--- MPlayer-1.0rc3/libswscale/swscaleold.c	2010-08-03 11:45:53.000000000 +0000
+++ MPlayer-1.0rc3/libswscale/swscale.c	2010-08-03 12:39:52.000000000 +0000
@@ -960,8 +960,7 @@
 #endif
 
 #if ARCH_PPC
-#if (HAVE_ALTIVEC || defined (RUNTIME_CPUDETECT)) && CONFIG_GPL
-#undef COMPILE_C
+#if HAVE_ALTIVEC
 #define COMPILE_ALTIVEC
 #endif
 #endif //ARCH_PPC
@@ -1636,41 +1635,44 @@
 }
 
 static SwsFunc getSwsFunc(int flags){
+#if CONFIG_RUNTIME_CPUDETECT
+    int flags = c->flags;
 
-#if defined(RUNTIME_CPUDETECT) && CONFIG_GPL
 #if ARCH_X86
     // ordered per speed fastest first
-    if (flags & SWS_CPU_CAPS_MMX2)
+    if (flags & SWS_CPU_CAPS_MMX2) {
         return swScale_MMX2;
-    else if (flags & SWS_CPU_CAPS_3DNOW)
+    } else if (flags & SWS_CPU_CAPS_3DNOW) {
         return swScale_3DNow;
-    else if (flags & SWS_CPU_CAPS_MMX)
+    } else if (flags & SWS_CPU_CAPS_MMX) {
         return swScale_MMX;
-    else
+    } else {
         return swScale_C;
+    }
 
 #else
-#if ARCH_PPC
-    if (flags & SWS_CPU_CAPS_ALTIVEC)
+#ifdef COMPILE_ALTIVEC
+    if (flags & SWS_CPU_CAPS_ALTIVEC) {
         return swScale_altivec;
-    else
+    } else {
         return swScale_C;
+    }
 #endif
     return swScale_C;
 #endif /* ARCH_X86 */
-#else //RUNTIME_CPUDETECT
-#if   HAVE_MMX2
+#else //CONFIG_RUNTIME_CPUDETECT
+#if   COMPILE_TEMPLATE_MMX2
     return swScale_MMX2;
-#elif HAVE_AMD3DNOW
+#elif COMPILE_TEMPLATE_AMD3DNOW
     return swScale_3DNow;
-#elif HAVE_MMX
+#elif COMPILE_TEMPLATE_MMX
     return swScale_MMX;
-#elif HAVE_ALTIVEC
+#elif COMPILE_TEMPLATE_ALTIVEC
     return swScale_altivec;
 #else
     return swScale_C;
 #endif
-#endif //!RUNTIME_CPUDETECT
+#endif //!CONFIG_RUNTIME_CPUDETECT
 }
 
 static int PlanarToNV12Wrapper(SwsContext *c, uint8_t* src[], int srcStride[], int srcSliceY,
