diff -Naur XML-LibXML-1.59/Makefile.PL XML-LibXML-1.59-p/Makefile.PL
--- XML-LibXML-1.59/Makefile.PL	2006-08-02 12:46:56.000000000 +0200
+++ XML-LibXML-1.59-p/Makefile.PL	2006-08-21 00:03:39.000000000 +0200
@@ -310,8 +310,9 @@
      $script .= <<"INSTALL";
 
 install_sax_driver :
-\t\@\$(PERL) -I\$(INSTALLSITELIB) -MXML::SAX -e "XML::SAX->add_parser(q(XML::LibXML::SAX::Parser))->save_parsers()"
-\t\@\$(PERL) -I\$(INSTALLSITELIB) -MXML::SAX -e "XML::SAX->add_parser(q(XML::LibXML::SAX))->save_parsers()"
+# -- disable here and add to an *install file I don't know any other solution to solve this -- #
+#\t\@\$(PERL) -I\$(INSTALLSITELIB) -MXML::SAX -e "XML::SAX->add_parser(q(XML::LibXML::SAX::Parser))->save_parsers()"
+#\t\@\$(PERL) -I\$(INSTALLSITELIB) -MXML::SAX -e "XML::SAX->add_parser(q(XML::LibXML::SAX))->save_parsers()"
 
 INSTALL
    } else {
