--- cacti/graph_image.php	2007/03/04 20:17:57	3898
+++ cacti/graph_image.php	2007/06/04 06:41:13	3956
@@ -49,22 +49,22 @@
 $graph_data_array = array();
 
 /* override: graph start time (unix time) */
-if (!empty($_GET["graph_start"])) {
+if (!empty($_GET["graph_start"]) && $_GET["graph_start"] < 1600000000) {
 	$graph_data_array["graph_start"] = $_GET["graph_start"];
 }
 
 /* override: graph end time (unix time) */
-if (!empty($_GET["graph_end"])) {
+if (!empty($_GET["graph_end"]) && $_GET["graph_end"] < 1600000000) {
 	$graph_data_array["graph_end"] = $_GET["graph_end"];
 }
 
 /* override: graph height (in pixels) */
-if (!empty($_GET["graph_height"])) {
+if (!empty($_GET["graph_height"]) && $_GET["graph_height"] < 3000) {
 	$graph_data_array["graph_height"] = $_GET["graph_height"];
 }
 
 /* override: graph width (in pixels) */
-if (!empty($_GET["graph_width"])) {
+if (!empty($_GET["graph_width"]) && $_GET["graph_width"] < 3000) {
 	$graph_data_array["graph_width"] = $_GET["graph_width"];
 }
 
