--- egl-wayland-1.1.11/meson.build	2022-08-06 01:54:11.000000000 +0200
+++ egl-wayland-1.1.11.new/meson.build	2022-11-08 20:23:30.033601450 +0100
@@ -57,7 +57,7 @@
     output : '@BASENAME@',
     configuration : pkgconf,
     install : true,
-    install_dir : join_paths(get_option('datadir'), 'pkgconfig')
+    install_dir : join_paths(get_option('libdir'), 'pkgconfig')
 )
 
 subdir('wayland-eglstream')
