--- spyder-2.1.13/spyderlib/requirements.py.old	2013-03-03 13:58:51.243476607 +0100
+++ spyder-2.1.13/spyderlib/requirements.py	2013-03-03 13:59:31.580475870 +0100
@@ -14,8 +14,8 @@
     """Return True if actual_str version fit required_str requirement"""
     actual = actual_str.split('.')
     required = required_str.split('.')
-    return actual[0] < required[0] or \
-           (actual[0] == required[0] and actual[1] < required[1])
+    return int(actual[0]) < int(required[0]) or \
+           (actual[0] == required[0] and int(actual[1]) < int(required[1]))
 
 def show_warning(message):
     """Show warning using Tkinter if available"""
