diff -Naur audacious-plugins-1.5.1/src/amidi-plug/i_configure.c audacious-plugins-1.5.1-new/src/amidi-plug/i_configure.c
--- audacious-plugins-1.5.1/src/amidi-plug/i_configure.c	2008-05-23 19:44:19.000000000 -0300
+++ audacious-plugins-1.5.1-new/src/amidi-plug/i_configure.c	2008-05-28 08:39:45.000000000 -0300
@@ -29,6 +29,7 @@
 #include "i_configure-dummy.h"
 #include "i_utils.h"
 #include <audacious/auddrct.h>
+#include <audacious/plugin.h>
 
 
 amidiplug_cfg_backend_t * amidiplug_cfg_backend;
