diff -up synce-trayicon-0.15/src/synce-trayicon.c.orig synce-trayicon-0.15/src/synce-trayicon.c
--- synce-trayicon-0.15/src/synce-trayicon.c.orig	2010-12-08 21:47:11.000000000 +0100
+++ synce-trayicon-0.15/src/synce-trayicon.c	2010-12-08 21:47:47.000000000 +0100
@@ -204,7 +204,7 @@ event_notification(SynceTrayIcon *self, 
   }
 
   if (gtk_status_icon_is_embedded(GTK_STATUS_ICON(self))) {
-          priv->notification = notify_notification_new_with_status_icon (summary, body, NULL, GTK_STATUS_ICON(self));
+          priv->notification = notify_notification_new(summary, body, GTK_STATUS_ICON(self));
           notify_notification_show (priv->notification, NULL);
   } else {
           g_debug("%s: not embedded", G_STRFUNC);
