diff -Naur xmule-1.13.6.old/src/ED2KLink.h xmule-1.13.6/src/ED2KLink.h
--- xmule-1.13.6.old/src/ED2KLink.h	2006-02-28 23:38:45.000000000 +0100
+++ xmule-1.13.6/src/ED2KLink.h	2006-05-15 12:15:53.000000000 +0200
@@ -68,7 +68,7 @@
 class CED2KFileLink : public CED2KLink
 {
 public:
-    CED2KFileLink::CED2KFileLink(const TCHAR* name,const TCHAR* size, const TCHAR* hash,const TCHAR* sources);
+    CED2KFileLink(const TCHAR* name,const TCHAR* size, const TCHAR* hash,const TCHAR* sources);
     virtual ~CED2KFileLink();
 
     virtual LinkType GetKind() const;
diff -Naur xmule-1.13.6.old/src/PartFile.h xmule-1.13.6/src/PartFile.h
--- xmule-1.13.6.old/src/PartFile.h	2006-02-28 23:38:45.000000000 +0100
+++ xmule-1.13.6/src/PartFile.h	2006-05-15 12:14:50.000000000 +0200
@@ -113,8 +113,8 @@
     bool IsText();
 
 
-    wxString CPartFile::getPartfileStatus(); //<<--9/21/02
-    int32_t CPartFile::getTimeRemaining(); //<<--9/21/02
+    wxString getPartfileStatus(); //<<--9/21/02
+    int32_t getTimeRemaining(); //<<--9/21/02
     time_t lastseencomplete;
     int getPartfileStatusRang();
     wxString GetDownloadFileInfo();
diff -Naur xmule-1.13.6.old/src/SafeFile.h xmule-1.13.6/src/SafeFile.h
--- xmule-1.13.6.old/src/SafeFile.h	2006-02-28 23:38:45.000000000 +0100
+++ xmule-1.13.6/src/SafeFile.h	2006-05-15 12:14:50.000000000 +0200
@@ -25,14 +25,14 @@
 class CSafeFile : public CFile{
 public:
 	CSafeFile();
-	CSafeFile::CSafeFile(x::LPCSTR lpszFileName,x::UINT nOpenFlags);
+	CSafeFile(x::LPCSTR lpszFileName,x::UINT nOpenFlags);
 	x::UINT Read(void* lpBuf,x::UINT nCount);
 };
 
 class CSafeMemFile : public CMemFile{ // memFile
 public:
 	CSafeMemFile(x::UINT nGrowBytes = 0);
-	CSafeMemFile::CSafeMemFile(x::BYTE* lpBuffer,x::UINT nBufferSize,x::UINT nGrowBytes = 0);
+	CSafeMemFile(x::BYTE* lpBuffer,x::UINT nBufferSize,x::UINT nGrowBytes = 0);
 	x::UINT Read(void* lpBuf,x::UINT nCount);
 };
 #endif
