--- lxde-common-0.5.5/pcmanfm/pcmanfm.conf.in.old	2011-07-29 16:04:03.000000000 +0200
+++ lxde-common-0.5.5/pcmanfm/pcmanfm.conf.in	2011-07-30 17:41:01.000000000 +0200
@@ -4,7 +4,7 @@
 
 [desktop]
 wallpaper_mode=1
-wallpaper=@prefix@/share/lxde/wallpapers/lxde_blue.jpg
+wallpaper=/usr/share/lxde/wallpapers/frugalware.png
 desktop_bg=#000000
 desktop_fg=#ffffff
 desktop_shadow=#000000
--- lxde-common-0.5.5/lxpanel/panel.in.old	2011-07-29 16:04:03.000000000 +0200
+++ lxde-common-0.5.5/lxpanel/panel.in	2011-07-30 17:45:01.000000000 +0200
@@ -16,7 +16,7 @@
     usefontcolor=1
     fontcolor=#ffffff
     background=1
-    backgroundfile=@prefix@/share/lxpanel/images/background.png
+    backgroundfile=/usr/share/lxde/wallpapers/frugalware.png
 }
 
 Plugin {
@@ -29,7 +29,7 @@
 Plugin {
     type = menu
     Config {
-        image=@prefix@/share/lxde/images/lxde-icon.png
+        image=/usr/share/lxpanel/images/lxde-fw.png
         system {
         }
         separator {
@@ -130,12 +130,4 @@
     }
 }
 
-Plugin {
-    type = launchbar
-    Config {
-        Button {
-            id=lxde-logout.desktop
-        }
-    }
-}
 
