--- lxde-common-0.3.2.1old/pcmanfm/lxde.conf.in	2008-04-20 10:24:58.000000000 +0200
+++ lxde-common-0.3.2.1/pcmanfm/lxde.conf.in	2008-07-15 12:01:23.000000000 +0200
@@ -1,3 +1,3 @@
 [Desktop]
 show_wallpaper=1
-wallpaper=@prefix@/share/lxde/wallpapers/lxde_blue.jpg
+wallpaper=@prefix@/share/lxde/wallpapers/frugalware.png
--- lxde-common-0.5.0/lxpanel/panel.in.back	2008-06-16 08:10:24.000000000 +0000
+++ lxde-common-0.5.0/lxpanel/panel.in	2010-04-12 13:26:06.000000000 +0000
@@ -8,15 +8,18 @@
     widthtype=percent
     width=100
     height=26
-    transparent=0
+    transparent=1
     tintcolor=#000000
     alpha=0
+    autohide=0
+    heightwhenhidden=2
     setdocktype=1
     setpartialstrut=1
     usefontcolor=1
     fontcolor=#ffffff
-    background=1
-    backgroundfile=@prefix@/share/lxpanel/images/background.png
+    background=0
+    backgroundfile=/usr/share/lxde/wallpapers/frugalware.png
+    iconsize=24
 }
 
 Plugin {
@@ -29,7 +32,7 @@
 Plugin {
     type = menu
     Config {
-        image=@prefix@/share/lxde/images/lxde-icon.png
+        image=/usr/share/lxpanel/images/lxde-fw.png
         system {
         }
         separator {
@@ -49,12 +52,6 @@
 Plugin {
     type = launchbar
     Config {
-        Button {
-            id=pcmanfm.desktop
-        }
-        Button {
-            id=firefox.desktop
-        }
     }
 }
 
@@ -68,8 +65,10 @@
 Plugin {
     type = wincmd
     Config {
+        image=window-manager
         Button1=iconify
         Button2=shade
+        Toggle=0
     }
 }
 
@@ -97,23 +96,17 @@
     Config {
         tooltips=1
         IconsOnly=0
-        AcceptSkipPager=1
-        ShowIconified=1
-        ShowMapped=1
         ShowAllDesks=0
         UseMouseWheel=1
         UseUrgencyHint=1
         FlatButton=0
         MaxTaskWidth=150
         spacing=1
+        GroupedTasks=0
     }
 }
 
 Plugin {
-    type = netstat
-}
-
-Plugin {
     type = cpu
 }
 
@@ -127,15 +120,11 @@
         ClockFmt=%R
         TooltipFmt=%A %x
         BoldFont=0
+        IconOnly=0
     }
 }
 
 Plugin {
-    type = launchbar
-    Config {
-        Button {
-            id=lxde-logout.desktop
-        }
-    }
+    type = volumealsa
 }
 
