--- lxde-common-0.3.2.1old/lxpanel/panel.in	2008-04-20 12:09:34.000000000 +0200
+++ lxde-common-0.3.2.1/lxpanel/panel.in	2008-07-15 17:24:52.000000000 +0200
@@ -8,15 +8,15 @@
     widthtype=percent
     width=100
     height=26
-    transparent=0
+    transparent=1
     tintcolor=#000000
     alpha=0
     setdocktype=1
     setpartialstrut=1
     usefontcolor=1
     fontcolor=#ffffff
-    background=1
-    backgroundfile=@prefix@/share/lxpanel/images/background.png
+    background=0
+    backgroundfile=@prefix@/share/lxde/wallpapers/lxde-frugalware_1.png
 }
 
 Plugin {
@@ -29,7 +29,7 @@
 Plugin {
     type = menu
     Config {
-        image=@prefix@/share/lxpanel/images/my-computer.png
+        image=@prefix@/share/lxpanel/images/lxde-fw.png
         system {
         }
         separator {
@@ -53,10 +53,7 @@
             id=pcmanfm.desktop
         }
         Button {
-            id=gnome-terminal.desktop
-        }
-        Button {
-            id=firefox.desktop
+            id=lxterminal.desktop
         }
     }
 }
@@ -133,3 +130,7 @@
     }
 }
 
+Plugin {
+    type = volumealsa
+}
+

--- lxde-common-0.3.2.1old/pcmanfm/main.lxde.in	2008-04-20 10:24:58.000000000 +0200
+++ lxde-common-0.3.2.1/pcmanfm/main.lxde.in	2008-07-15 12:01:23.000000000 +0200
@@ -1,3 +1,3 @@
 [Desktop]
 show_wallpaper=1
-wallpaper=@prefix@/share/lxde/wallpapers/lxde_blue.jpg
+wallpaper=@prefix@/share/lxde/wallpapers/lxde-frugalware_1.png

--- lxde-common-0.3.2.1old/lxde-settings/config	2008-04-20 10:26:08.000000000 +0200
+++ lxde-common-0.3.2.1/lxde-settings/config	2008-07-15 12:39:04.000000000 +0200
@@ -4,7 +4,7 @@
 #iNet/DndDragThreshold
 #iNet/CursorBlink
 #iNet/CursorBlinkTime
-sNet/ThemeName=Clearlooks
+sNet/ThemeName=Frugalware
 sNet/IconThemeName=nuoveXT2
 #iGtk/CanChangeAccels=0
 #cGtk/ColorPalette


