From 34cd793a18138463133a72b713173d2d01d32d66 Mon Sep 17 00:00:00 2001
From: Christoph Wickert <christoph.wickert@googlemail.com>
Date: Sun, 3 Jan 2010 11:09:30 +0000
Subject: [PATCH 2/9] start xscreensaver after lxpanel and pcmanfm

---
 autostart |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/autostart b/autostart
index 17d7b58..bcee88e 100644
--- a/autostart
+++ b/autostart
@@ -1,3 +1,3 @@
-@xscreensaver -no-splash
 @lxpanel --profile LXDE
 @pcmanfm -d
+@xscreensaver -no-splash
-- 
1.6.6.1

