diff -Naur lxde-common-0.5.5/desktop.conf lxde-common-0.5.5new/desktop.conf
--- lxde-common-0.5.5/desktop.conf	2011-07-29 16:04:03.000000000 +0200
+++ lxde-common-0.5.5new/desktop.conf	2011-11-20 09:59:08.000000000 +0100
@@ -2,14 +2,17 @@
 window_manager=openbox-lxde
 
 [GTK]
-sNet/ThemeName=Clearlooks
-sNet/IconThemeName=nuoveXT2
-sGtk/FontName=Sans 10
+sNet/ThemeName=Frugalware
+sNet/IconThemeName=Frugalware
+sGtk/FontName=Droid Sans 10
 iGtk/ToolbarStyle=3
 iGtk/ButtonImages=1
 iGtk/MenuImages=1
 iGtk/CursorThemeSize=18
 iXft/Antialias=1
+iXft/Hinting=1
+sXft/HintStyle=hintslight
+sXft/RGBA=rgb
 
 [Mouse]
 AccFactor=20
diff -Naur lxde-common-0.5.5/lxpanel/panel.in lxde-common-0.5.5new/lxpanel/panel.in
--- lxde-common-0.5.5/lxpanel/panel.in	2011-07-29 16:04:03.000000000 +0200
+++ lxde-common-0.5.5new/lxpanel/panel.in	2011-11-20 10:03:53.000000000 +0100
@@ -7,7 +7,7 @@
     margin=0
     widthtype=percent
     width=100
-    height=26
+    height=24
     transparent=0
     tintcolor=#000000
     alpha=0
@@ -16,7 +16,8 @@
     usefontcolor=1
     fontcolor=#ffffff
     background=1
-    backgroundfile=@prefix@/share/lxpanel/images/background.png
+    backgroundfile=/usr/share/lxpanel/images/lxpanel-background.png
+    iconsize=22
 }
 
 Plugin {
@@ -29,7 +30,7 @@
 Plugin {
     type = menu
     Config {
-        image=@prefix@/share/lxde/images/lxde-icon.png
+        image=/usr/share/lxpanel/images/frugalware-logo.png
         system {
         }
         separator {
@@ -118,6 +119,10 @@
 }
 
 Plugin {
+    type = volumealsa
+}
+
+Plugin {
     type = tray
 }
 
diff -Naur lxde-common-0.5.5/pcmanfm/pcmanfm.conf.in lxde-common-0.5.5new/pcmanfm/pcmanfm.conf.in
--- lxde-common-0.5.5/pcmanfm/pcmanfm.conf.in	2011-07-29 16:04:03.000000000 +0200
+++ lxde-common-0.5.5new/pcmanfm/pcmanfm.conf.in	2011-11-20 09:20:56.000000000 +0100
@@ -4,10 +4,11 @@
 
 [desktop]
 wallpaper_mode=1
-wallpaper=@prefix@/share/lxde/wallpapers/lxde_blue.jpg
+wallpaper=/usr/share/lxde/wallpapers/frugalware.png
 desktop_bg=#000000
 desktop_fg=#ffffff
 desktop_shadow=#000000
+desktop_font=Droid Sans 11
 
 [ui]
 always_show_tabs=0

diff -Naur lxde-common-0.5.5/autostart lxde-common-0.5.5new/autostart 
--- lxde-common-0.5.5/autostart	2011-07-29 16:04:03.000000000 +0200
+++ lxde-common-0.5.5new/autostart	2013-04-10 19:00:52.428093770 +0200
@@ -1,3 +1,3 @@
-@lxpanel --profile LXDE
+@lxpanel --profile default
 @pcmanfm --desktop --profile LXDE
 @xscreensaver -no-splash
