diff -Naur d4x-2.5.0final.orig/main/face/lod.cc d4x-2.5.0final/main/face/lod.cc
--- d4x-2.5.0final.orig/main/face/lod.cc	2005-06-15 10:34:20.000000000 +0000
+++ d4x-2.5.0final/main/face/lod.cc	2005-06-15 10:34:38.000000000 +0000
@@ -126,21 +126,20 @@
 };
 
 /***************************************************************/
-
-struct GtkCellRendererProgress{
+struct GtkCellRendererProgressD4X{
 	GtkCellRenderer parent;
 	gfloat percent;
 	tDownload *dwn;
 };
 
-struct GtkCellRendererProgressClass{
+struct GtkCellRendererProgressD4XClass{
 	GtkCellRendererClass parent_class;
 };
 
-GtkType          gtk_cell_renderer_progress_get_type (void);
-GtkCellRenderer *gtk_cell_renderer_progress_new      (void);
+//GtkType          gtk_cell_renderer_progress_get_type (void);
+//GtkCellRenderer *gtk_cell_renderer_progress_new      (void);
 
-static void gtk_cell_renderer_progress_init (GtkCellRendererProgress *cellpixbuf){
+static void gtk_cell_renderer_progress_init (GtkCellRendererProgressD4X *cellpixbuf){
 };
 
 static void
@@ -172,7 +171,7 @@
 					 guint         param_id,
 					 const GValue *value,
 					 GParamSpec   *pspec){
-	GtkCellRendererProgress *renderer = (GtkCellRendererProgress *)object;
+	GtkCellRendererProgressD4X *renderer = (GtkCellRendererProgressD4X *)object;
 	switch (param_id){
 	case 1:
 		renderer->percent=g_value_get_float(value);
@@ -195,7 +194,7 @@
 				   GdkRectangle       *expose_area,
 				   GtkCellRendererState flags){
 	char tmpc[100];
-	float p=((GtkCellRendererProgress*)cell)->percent;
+	float p=((GtkCellRendererProgressD4X*)cell)->percent;
 
 	if (p>99.0 && p<100.0)
 		sprintf(tmpc,"%.1f",p);
@@ -223,7 +222,7 @@
 	switch (CFG.PROGRESS_MODE){
 	case 2:{
 		if (p<=0) break;
-		tDownload *temp=((GtkCellRendererProgress*)cell)->dwn;;
+		tDownload *temp=((GtkCellRendererProgressD4X*)cell)->dwn;;
 		if (temp && temp->segments && temp->finfo.size>0){
 			temp->segments->lock_public();
 			tSegment *tmp=temp->segments->get_first();
@@ -271,7 +270,7 @@
 };
 
 
-static void gtk_cell_renderer_progress_class_init (GtkCellRendererProgressClass *klass){
+static void gtk_cell_renderer_progress_class_init (GtkCellRendererProgressD4XClass *klass){
 	GObjectClass *object_class = G_OBJECT_CLASS (klass);
 	GtkCellRendererClass *cell_class = GTK_CELL_RENDERER_CLASS (klass);
 
@@ -300,19 +299,19 @@
 	{
 		static const GTypeInfo cell_progress_info =
 		{
-			sizeof (GtkCellRendererProgressClass),
+			sizeof (GtkCellRendererProgressD4XClass),
 			NULL,		/* base_init */
 			NULL,		/* base_finalize */
 			(GClassInitFunc) gtk_cell_renderer_progress_class_init,
 			NULL,		/* class_finalize */
 			NULL,		/* class_data */
-			sizeof (GtkCellRendererProgress),
+			sizeof (GtkCellRendererProgressD4X),
 			0,              /* n_preallocs */
 			(GInstanceInitFunc) gtk_cell_renderer_progress_init,
 		};
 
 		cell_progress_type = g_type_register_static (GTK_TYPE_CELL_RENDERER,
-							     "GtkCellRendererProgress",
+							     "GtkCellRendererProgressD4X",
 							     &cell_progress_info,
 							     GTypeFlags(0));
 	}
diff -Naur d4x-2.5.0final.orig/main/sndserv.cc d4x-2.5.0final/main/sndserv.cc
--- d4x-2.5.0final.orig/main/sndserv.cc	2005-06-15 10:34:20.000000000 +0000
+++ d4x-2.5.0final/main/sndserv.cc	2005-06-15 10:34:49.000000000 +0000
@@ -153,7 +153,7 @@
 
 #if G_BYTE_ORDER == G_LITTLE_ENDIAN
 	*len =(buf[3] << 24) | (buf[2] << 16) | (buf[1] << 8) | buf[0];
-#elif
+#else
 	*len =(buf[0] << 24) | (buf[1] << 16) | (buf[2] << 8) | buf[3];
 #endif
 
@@ -166,7 +166,7 @@
 		return 0;
 #if G_BYTE_ORDER == G_LITTLE_ENDIAN
 	*val = (buf[1] << 8) | buf[0];
-#elif
+#else
 	*val = (buf[0] << 8) | buf[1];
 #endif
 	return 1;
