diff -Naur kiconthemes-5.21.0/src/kicontheme.cpp kiconthemes-5.21.0-p/src/kicontheme.cpp
--- kiconthemes-5.21.0/src/kicontheme.cpp	2016-04-03 22:54:06.000000000 +0200
+++ kiconthemes-5.21.0-p/src/kicontheme.cpp	2016-05-13 13:00:09.481225178 +0200
@@ -557,9 +557,10 @@
 // static
 QString KIconTheme::defaultThemeName()
 {
-    return QStringLiteral("oxygen");
+    return QStringLiteral("breeze");
 }
 
+
 void KIconTheme::assignIconsToContextMenu(ContextMenus type,
         QList<QAction *> actions)
 {
