diff -Naur foo2zjs/Makefile foo2zjs-p/Makefile
--- foo2zjs/Makefile	2007-04-24 12:43:44.000000000 +0200
+++ foo2zjs-p/Makefile	2007-04-30 18:31:13.000000000 +0200
@@ -7,8 +7,7 @@
 VERSION=0.0
 
 # Installation prefix...
-PREFIX=/usr/local
-PREFIX=/usr
+PREFIX=$(DESTDIR)/usr
 
 # Pathnames for this package...
 BIN=$(PREFIX)/bin
@@ -22,7 +21,7 @@
 DOCDIR=$(PREFIX)/share/doc/foo2zjs/
 
 # Pathnames for referenced packages...
-FOODB=/usr/share/foomatic/db/source
+FOODB=$(DESTDIR)/usr/share/foomatic/db/source
 
 # User ID's
 LPuid=-olp
@@ -401,7 +400,7 @@
 	#
 	install -c $(PROGS) $(SHELLS) $(BIN)/
 	if [ "$(BINPROGS)" != "" ]; then \
-	    install -c $(BINPROGS) /bin/; \
+	    install -c $(BINPROGS) $(PREFIX)/bin/; \
 	fi
 	#
 	# Install gamma correction files.  These are just templates,
@@ -528,9 +527,9 @@
 	    fi; \
 	done
 
-MODEL=/usr/share/cups/model
-LOCALMODEL=/usr/local/share/cups/model
-PPD=/usr/share/ppd
+MODEL=$(DESTDIR)/usr/share/cups/model
+LOCALMODEL=$(DESTDIR)/usr/local/share/cups/model
+PPD=$(DESTDIR)/usr/share/ppd
 install-ppd:
 	#
 	# Install PPD files for CUPS
@@ -559,8 +558,8 @@
 	    done; \
 	fi
 
-USBDIR=/etc/hotplug/usb
-UDEVDIR=/etc/udev/rules.d
+USBDIR=$(DESTDIR)/etc/hotplug/usb
+UDEVDIR=$(DESTDIR)/etc/udev/rules.d
 RULES=hplj10xx.rules
 install-hotplug: install-hotplug-test install-hotplug-prog
 
@@ -916,8 +915,7 @@
 	install -c -m 644 README $(DOCDIR)
 	install -c -m 644 ChangeLog $(DOCDIR)
 
-GROFF=/usr/local/test/bin/groff
-GROFF=groff
+GROFF=/usr/bin/groff
 manual.pdf: $(MANPAGES)
 	-$(GROFF) -t -man $(MANPAGES) | ps2pdf - $@
 
