--- src/kdissert/canvasview/DCanvasLink.cpp~    2007-04-02 09:20:38.000000000 +0000
+++ src/kdissert/canvasview/DCanvasLink.cpp     2007-04-02 09:20:45.000000000 +0000
@@ -12,6 +12,7 @@
 *   GNU General Public License for more details.                          *
 ***************************************************************************/

+#include <cstdlib>
 #include <math.h>
 #include <qpainter.h>
 #include <qcolor.h> 
Index: src/kdissert/canvasview/DCanvasView.cpp
===================================================================
--- src/kdissert/canvasview/DCanvasView.cpp	(revision 2)
+++ src/kdissert/canvasview/DCanvasView.cpp	(working copy)
@@ -12,6 +12,7 @@
  *   GNU General Public License for more details.                          *
  ***************************************************************************/
 
+#include <limits>
 #include <qpoint.h>
 #include <qdragobject.h>
 #include <qevent.h>
 