diff -Naur dtach-0.7/Makefile.in dtach-0.7-makefile/Makefile.in
--- dtach-0.7/Makefile.in	2004-07-04 18:07:03.000000000 +0200
+++ dtach-0.7-makefile/Makefile.in	2006-07-23 19:10:11.054262248 +0200
@@ -5,6 +5,8 @@
 LIBS = @LIBS@
 VERSION = @PACKAGE_VERSION@
 VPATH = $(srcdir)
+PREFIX = /usr
+DESTDIR =
 
 OBJ = attach.o master.o main.o
 SRC = $(srcdir)/attach.c $(srcdir)/master.c $(srcdir)/main.c
@@ -17,6 +19,12 @@
 dtach: $(OBJ)
 	$(CC) -o $@ $(LDFLAGS) $(OBJ) $(LIBS)
 
+install:
+	mkdir -p $(DESTDIR)$(PREFIX)/bin
+	/usr/bin/install -c -g root -o root -m 755  dtach $(DESTDIR)$(PREFIX)/bin
+	mkdir -p $(DESTDIR)$(PREFIX)/man/man1
+	/usr/bin/install -c -g root -o root -m 644  dtach.1 $(DESTDIR)$(PREFIX)/man/man1
+	
 clean:
 	rm -f dtach $(OBJ) dtach-$(VERSION).tar.gz
 
