--- src/java/org/apache/ivy/ant/IvyAntSettings.java~	2014-05-23 19:56:50.000000000 +0200
+++ src/java/org/apache/ivy/ant/IvyAntSettings.java	2015-04-01 12:37:37.537520590 +0200
@@ -346,7 +346,7 @@
         File[] settingsLocations = new File[] {
                 new File(getProject().getBaseDir(), settingsFileName),
                 new File(getProject().getBaseDir(), "ivyconf.xml"), new File(settingsFileName),
-                new File("ivyconf.xml")};
+                new File("ivyconf.xml"), new File("/etc/ivy/ivysettings.xml")};
         for (int i = 0; i < settingsLocations.length; i++) {
             file = settingsLocations[i];
             task.log("searching settings file: trying " + file, Project.MSG_VERBOSE);
