diff -Naur fvwm-2.5.18.orig/bin/fvwm-menu-directory.in fvwm-2.5.18/bin/fvwm-menu-directory.in
--- fvwm-2.5.18.orig/bin/fvwm-menu-directory.in	2005-07-08 22:59:00.000000000 +0200
+++ fvwm-2.5.18/bin/fvwm-menu-directory.in	2006-11-25 14:02:45.000000000 +0100
@@ -279,7 +279,7 @@
 	my $itemStr = $dir && $itemF_eval? &evalItem($_name, $dir): $_name;
 	$itemStr = escapeItemName($itemStr);
 	my $act = !$dir || $checkSubdirs && !-x $dir? "Nop": !$reuse?
-		qq(Popup ") . escapeFvwmName($dir) . qq("$submenuPos):
+		qq(Popup ") . escapeFileName($dir) . qq("$submenuPos):
 		qq(PipeRead 'echo ") . escapeFileName($dir) . qq(" >$dirFile; )
 		. qq(echo Menu ") . escapeFvwmName($name) . qq(" WarpTitle');
 	return qq(+ "$iconDStr$itemStr" $act\n);
