diff -urN nvclock0.7/src/gtk/callbacks.c nvclock0.7_patched/src/gtk/callbacks.c
--- nvclock0.7/src/gtk/callbacks.c	2003-06-29 14:28:46.000000000 +0200
+++ nvclock0.7_patched/src/gtk/callbacks.c	2005-08-20 13:16:39.000000000 +0200
@@ -197,17 +197,17 @@
     nvclk = nv_card.get_gpu_speed();
 
     text = g_strdup_printf("%0.3f", memclk);
-    (gint)GTK_ADJUSTMENT(value_mem)->lower = nv_card.memclk_min;
-    (gint)GTK_ADJUSTMENT(value_mem)->upper = nv_card.memclk_max;
-    (gint)GTK_ADJUSTMENT(value_mem)->value = (memclk);
+    GTK_ADJUSTMENT(value_mem)->lower = nv_card.memclk_min;
+    GTK_ADJUSTMENT(value_mem)->upper = nv_card.memclk_max;
+    GTK_ADJUSTMENT(value_mem)->value = (memclk);
     gtk_adjustment_set_value(GTK_ADJUSTMENT(value_mem), memclk);
     gtk_entry_set_text(GTK_ENTRY(entry_mem), text);
     g_free(text);
 	
     text = g_strdup_printf("%0.3f", nvclk);
-    (gint)GTK_ADJUSTMENT(value_core)->lower = nv_card.nvclk_min;
-    (gint)GTK_ADJUSTMENT(value_core)->upper = nv_card.nvclk_max;
-    (gint)GTK_ADJUSTMENT(value_core)->value = (nvclk);
+    GTK_ADJUSTMENT(value_core)->lower = nv_card.nvclk_min;
+    GTK_ADJUSTMENT(value_core)->upper = nv_card.nvclk_max;
+    GTK_ADJUSTMENT(value_core)->value = (nvclk);
     gtk_adjustment_set_value(GTK_ADJUSTMENT(value_core), nvclk);
     gtk_entry_set_text(GTK_ENTRY(entry_core), text);
 
