diff -Naur sparkleshare-0.8.4/SparkleShare/Makefile.am sparkleshare-0.8.4new/SparkleShare/Makefile.am
--- sparkleshare-0.8.4/SparkleShare/Makefile.am	2012-03-18 19:58:25.000000000 +0100
+++ sparkleshare-0.8.4new/SparkleShare/Makefile.am	2012-03-26 18:41:04.000000000 +0200
@@ -25,5 +25,4 @@
 include $(top_srcdir)/build/build.mk
 	
 install-data-hook:
-	update-desktop-database $(datadir)/applications
 	echo "Installed protocol handler sparkleshare://"
diff -Naur sparkleshare-0.8.4/SparkleShare/Makefile.in sparkleshare-0.8.4new/SparkleShare/Makefile.in
--- sparkleshare-0.8.4/SparkleShare/Makefile.in	2012-03-18 21:26:25.000000000 +0100
+++ sparkleshare-0.8.4new/SparkleShare/Makefile.in	2012-03-26 18:41:20.000000000 +0200
@@ -786,7 +786,6 @@
 	fi;
 
 install-data-hook:
-	update-desktop-database $(datadir)/applications
 	echo "Installed protocol handler sparkleshare://"
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
