--- gsm-1.0-pl22/Makefile	2024-03-05 13:41:18.486735024 +0100
+++ gsm-1.0-pl22.new/Makefile	2024-03-05 13:42:10.766149396 +0100
@@ -356,11 +356,11 @@
 
 $(TOAST_INSTALL_BIN)/untoast:	$(TOAST_INSTALL_BIN)/toast
 		-rm $(RMFLAGS) $@
-		$(LN) $? $@
+		$(LN) toast $@
 
 $(TOAST_INSTALL_BIN)/tcat:	$(TOAST_INSTALL_BIN)/toast
 		-rm $(RMFLAGS) $@
-		$(LN) $? $@
+		$(LN) toast $@
 
 $(TOAST_INSTALL_MAN)/toast.1:	$(MAN)/toast.1
 		-rm $(RMFLAGS) $@
