diff -urN gparted-0.3.3.orig/gparted.desktop gparted-0.3.3/gparted.desktop
--- gparted-0.3.3.orig/gparted.desktop	2006-12-06 19:43:59.000000000 +0100
+++ gparted-0.3.3/gparted.desktop	2006-12-06 19:45:00.000000000 +0100
@@ -62,9 +62,10 @@
 Comment[vi]=Bộ hiệu chỉnh phân vùng Gnome
 Comment[zh_HK]=GNOME 分割區編輯器
 Comment[zh_TW]=GNOME 分割區編輯器
-Exec=gparted
+Exec=gksu gparted
 Icon=gparted.png
 Terminal=false
 Type=Application
 Categories=GNOME;Application;System;
 StartupNotify=true
+NotShowIn=KDE
diff -urN gparted-0.3.3.orig/gparted.desktop.in gparted-0.3.3/gparted.desktop.in
--- gparted-0.3.3.orig/gparted.desktop.in	2006-12-06 19:43:59.000000000 +0100
+++ gparted-0.3.3/gparted.desktop.in	2006-12-06 19:45:10.000000000 +0100
@@ -2,9 +2,10 @@
 Encoding=UTF-8
 _Name=GParted
 _Comment=GNOME Partition Editor
-Exec=gparted
+Exec=gksu gparted
 Icon=gparted.png
 Terminal=false
 Type=Application
 Categories=GNOME;Application;System;
 StartupNotify=true
+NotShowIn=KDE
