diff -Naur gparted-0.2.orig/gparted.desktop gparted-0.2/gparted.desktop
--- gparted-0.2.orig/gparted.desktop	2006-01-29 20:32:58.000000000 +0100
+++ gparted-0.2/gparted.desktop	2006-02-02 01:15:10.000000000 +0100
@@ -32,9 +32,10 @@
 Comment[hu]=GNOME partíciószerkesztő
 Comment[nl]=GNOME Partitiebewerker
 Comment[vi]=Bộ hiệu chỉnh phân vùng Gnome
-Exec=gparted
+Exec=gksu gparted
 Icon=gparted.png
 Terminal=false
 Type=Application
 Categories=GNOME;Application;System;
 StartupNotify=true
+NotShowIn=KDE
diff -Naur gparted-0.2.orig/gparted.desktop.in gparted-0.2/gparted.desktop.in
--- gparted-0.2.orig/gparted.desktop.in	2006-01-26 17:18:49.000000000 +0100
+++ gparted-0.2/gparted.desktop.in	2006-02-02 01:16:19.000000000 +0100
@@ -2,9 +2,10 @@
 Encoding=UTF-8
 _Name=GParted
 _Comment=GNOME Partition Editor
-Exec=gparted
+Exec=gksu gparted
 Icon=gparted.png
 Terminal=false
 Type=Application
 Categories=GNOME;Application;System;
 StartupNotify=true
+NotShowIn=KDE
