diff -Naurp gparted-0.3.1/configure.in gparted-0.3.1-p/configure.in
--- gparted-0.3.1/configure.in	2006-09-11 11:39:37.000000000 +0200
+++ gparted-0.3.1-p/configure.in	2006-11-20 23:13:02.000000000 +0100
@@ -48,7 +48,7 @@ int main ()
 	if ( sscanf( ped_get_version(), "%d.%d.%d", &major, &minor, &micro ) == 3 )
 		printf( "Found libparted %s\t", ped_get_version() ) ;
 
-        return ! ( major == 1 && minor == 7 && micro >= 1 ) ;
+        return ! ( major == 1 && minor == 8 && micro >= 0 ) ;
 }
 ,AC_MSG_RESULT(OK),AC_MSG_ERROR(*** Requires libparted >= 1.7.1) )
 
