diff -aur akode-2.0.2/akode/plugins/ffmpeg_decoder/ffmpeg_decoder.cpp fw_akode-2.0.2/akode/plugins/ffmpeg_decoder/ffmpeg_decoder.cpp
--- akode-2.0.2/akode/plugins/ffmpeg_decoder/ffmpeg_decoder.cpp	2006-08-10 20:37:20.000000000 +0200
+++ fw_akode-2.0.2/akode/plugins/ffmpeg_decoder/ffmpeg_decoder.cpp	2008-04-28 13:42:32.000000000 +0200
@@ -26,9 +26,9 @@
 #include "decoder.h"
 
 #include <assert.h>
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
-#include <ffmpeg/avio.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
+#include <libavformat/avio.h>
 
 #include "ffmpeg_decoder.h"
 #include <iostream>
