diff -Naur akode-2.0.2/akode/plugins/ffmpeg_decoder/Makefile.am akode-2.0.2-ffmpeg/akode/plugins/ffmpeg_decoder/Makefile.am
--- akode-2.0.2/akode/plugins/ffmpeg_decoder/Makefile.am	2005-11-19 13:02:20.000000000 +0100
+++ akode-2.0.2-ffmpeg/akode/plugins/ffmpeg_decoder/Makefile.am	2007-04-23 14:02:42.000000000 +0200
@@ -1,4 +1,6 @@
-INCLUDES = -I$(top_srcdir)/akode/lib -I$(top_builddir)/akode/lib $(AVCODEC_CFLAGS) $(AVFORMAT_CFLAGS) $(all_includes)
+
+NEW_FFMPEG_DEFINES = -D__STDC_CONSTANT_MACROS 
+INCLUDES = -I$(top_srcdir)/akode/lib -I$(top_builddir)/akode/lib $(AVCODEC_CFLAGS) $(NEW_FFMPEG_DEFINES) $(AVFORMAT_CFLAGS) $(all_includes)
 
 lib_LTLIBRARIES	= libakode_ffmpeg_decoder.la
 
