From 443e9c761c0e6f1f9b27c465c079354f00fdf13f Mon Sep 17 00:00:00 2001
From: Ryan Scott <ryan.gl.scott@gmail.com>
Date: Sun, 1 Jul 2018 06:43:58 -0400
Subject: [PATCH] Allow template-haskell-2.14, containers-0.6

---
 th-abstraction.cabal | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/th-abstraction.cabal b/th-abstraction.cabal
index 066331f..6d077aa 100644
--- a/th-abstraction.cabal
+++ b/th-abstraction.cabal
@@ -28,8 +28,8 @@ library
   other-modules:       Language.Haskell.TH.Datatype.Internal
   build-depends:       base             >=4.3   && <5,
                        ghc-prim,
-                       template-haskell >=2.5   && <2.14,
-                       containers       >=0.4   && <0.6
+                       template-haskell >=2.5   && <2.15,
+                       containers       >=0.4   && <0.7
   hs-source-dirs:      src
   default-language:    Haskell2010
 
