diff -Naur gftp-2.0.18.orig/docs/gftp.desktop gftp-2.0.18/docs/gftp.desktop
--- gftp-2.0.18.orig/docs/gftp.desktop	2004-11-03 23:11:57.000000000 +0100
+++ gftp-2.0.18/docs/gftp.desktop	2005-02-05 13:34:58.118164664 +0100
@@ -4,6 +4,7 @@
 Name=gFTP
 GenericName=gFTP
 Comment=Download and upload files using multiple file transfer protocols
+Comment[hu]=Fájlok le- és feltöltése többféle fájlátviteli protokoll használatával
 Exec=gftp %u
 TryExec=gftp
 Icon=gftp.png
