diff -Naur sddm-0.16.0/data/man/sddm.conf.rst.in sddm-0.16.0-p/data/man/sddm.conf.rst.in
--- sddm-0.16.0/data/man/sddm.conf.rst.in	2017-10-05 08:24:24.000000000 +0200
+++ sddm-0.16.0-p/data/man/sddm.conf.rst.in	2017-10-12 00:22:32.311809447 +0200
@@ -16,14 +16,14 @@
 
 Configuration loads all files in the configuration directories followed by the configuration file in the order listed below with the latter having the most precedent. Changes should be made to the local configurations.
 
-**@SYSTEM_CONFIG_DIR@*
-    System configuration directory
+**@SYSTEM_CONFIG_DIR@**
+        System configuration directory
 
 **@CONFIG_DIR@**
-	Local configuration directory
+        Local configuration directory
 
 **@CONFIG_FILE@**
-	Local configuration file for compatibility
+        Local configuration file for compatibility
 
 DESCRIPTION
 ===========
diff -Naur sddm-0.16.0/data/man/sddm-greeter.rst.in sddm-0.16.0-p/data/man/sddm-greeter.rst.in
--- sddm-0.16.0/data/man/sddm-greeter.rst.in	2017-10-05 08:24:24.000000000 +0200
+++ sddm-0.16.0-p/data/man/sddm-greeter.rst.in	2017-10-12 00:22:00.704939279 +0200
@@ -47,8 +47,14 @@
 FILES
 =====
 
+**@SYSTEM_CONFIG_DIR@**
+        System configuration directory
+
+**@CONFIG_DIR@**
+        Local configuration directory
+
 **@CONFIG_FILE@**
-	System configuration file
+	Local configuration file for compatibility
 
 **@DATA_INSTALL_DIR@/themes**
 	Where sddm looks for themes
diff -Naur sddm-0.16.0/data/man/sddm.rst.in sddm-0.16.0-p/data/man/sddm.rst.in
--- sddm-0.16.0/data/man/sddm.rst.in	2017-10-05 08:24:24.000000000 +0200
+++ sddm-0.16.0-p/data/man/sddm.rst.in	2017-10-12 00:22:08.325149069 +0200
@@ -46,11 +46,14 @@
 FILES
 =====
 
+**@SYSTEM_CONFIG_DIR@**
+        System configuration directory
+
 **@CONFIG_DIR@**
-	System configuration directory
+	Local configuration directory
 
 **@CONFIG_FILE@**
-	System configuration file
+	Local configuration file for compatibility
 
 **@DATA_INSTALL_DIR@/themes**
 	Where sddm looks for themes
