diff -Naur sddm-develop/services/sddm-autologin.pam sddm-develop-p/services/sddm-autologin.pam
--- sddm-develop/services/sddm-autologin.pam	2017-06-12 17:32:18.000000000 +0200
+++ sddm-develop-p/services/sddm-autologin.pam	2017-07-09 04:27:04.428378005 +0200
@@ -4,10 +4,6 @@
 auth        required    pam_shells.so
 auth        required    pam_nologin.so
 auth        required    pam_permit.so
--auth       optional    pam_gnome_keyring.so
--auth       optional    pam_kwallet5.so
 account     include     system-local-login
 password    include     system-local-login
 session     include     system-local-login
--session    optional    pam_gnome_keyring.so auto_start
--session    optional    pam_kwallet5.so auto_start
diff -Naur sddm-develop/services/sddm-greeter.pam sddm-develop-p/services/sddm-greeter.pam
--- sddm-develop/services/sddm-greeter.pam	2017-06-12 17:32:18.000000000 +0200
+++ sddm-develop-p/services/sddm-greeter.pam	2017-07-09 04:28:13.171239045 +0200
@@ -15,4 +15,5 @@
 # Setup session
 session		required pam_unix.so
 session		optional pam_systemd.so
-session		optional pam_elogind.so
+#enable once we remove ^
+#session		optional pam_elogind.so
diff -Naur sddm-develop/services/sddm.pam sddm-develop-p/services/sddm.pam
--- sddm-develop/services/sddm.pam	2017-06-12 17:32:18.000000000 +0200
+++ sddm-develop-p/services/sddm.pam	2017-07-09 04:29:12.448843952 +0200
@@ -1,15 +1,7 @@
 #%PAM-1.0
 
 auth		include		system-login
--auth		optional	pam_gnome_keyring.so
--auth   optional  pam_kwallet5.so
-
 account		include		system-login
-
 password	include		system-login
--password	optional	pam_gnome_keyring.so use_authtok
-
 session		optional	pam_keyinit.so force revoke
 session		include		system-login
--session		optional	pam_gnome_keyring.so auto_start
--session  optional  pam_kwallet5.so auto_start
