diff -Naur sddm-0.17.0/services/sddm-autologin.pam sddm-0.17.0-p/services/sddm-autologin.pam
--- sddm-0.17.0/services/sddm-autologin.pam	2017-12-05 17:00:16.000000000 +0100
+++ sddm-0.17.0-p/services/sddm-autologin.pam	2018-05-18 18:13:37.924727932 +0200
@@ -4,10 +4,16 @@
 auth        required    pam_shells.so
 auth        required    pam_nologin.so
 auth        required    pam_permit.so
--auth       optional    pam_gnome_keyring.so
 -auth       optional    pam_kwallet5.so
-account     include     system-local-login
-password    include     system-local-login
-session     include     system-local-login
--session    optional    pam_gnome_keyring.so auto_start
+-auth       optional    pam_gnome_keyring.so
+
+account     include     common-account
+
+password    include     common-password
+
+session     include     common-session
+session     required    pam_limits.so
 -session    optional    pam_kwallet5.so auto_start
+-session    optional    pam_gnome_keyring.so auto_start
+
+
diff -Naur sddm-0.17.0/services/sddm-greeter.pam sddm-0.17.0-p/services/sddm-greeter.pam
--- sddm-0.17.0/services/sddm-greeter.pam	2017-12-05 17:00:16.000000000 +0100
+++ sddm-0.17.0-p/services/sddm-greeter.pam	2018-05-18 18:11:28.663736017 +0200
@@ -15,4 +15,4 @@
 # Setup session
 session		required pam_unix.so
 -session		optional pam_systemd.so
--session		optional pam_elogind.so
+#-session		optional pam_elogind.so
diff -Naur sddm-0.17.0/services/sddm.pam sddm-0.17.0-p/services/sddm.pam
--- sddm-0.17.0/services/sddm.pam	2017-12-05 17:00:16.000000000 +0100
+++ sddm-0.17.0-p/services/sddm.pam	2018-05-18 18:11:28.663736017 +0200
@@ -1,15 +1,7 @@
 #%PAM-1.0
 
 auth		include		system-login
--auth		optional	pam_gnome_keyring.so
--auth   optional  pam_kwallet5.so
-
 account		include		system-login
-
 password	include		system-login
--password	optional	pam_gnome_keyring.so use_authtok
-
 session		optional	pam_keyinit.so force revoke
 session		include		system-login
--session		optional	pam_gnome_keyring.so auto_start
--session  optional  pam_kwallet5.so auto_start
