diff -Naur sddm-0.13.0/data/scripts/wayland-session sddm-0.13.0-p/data/scripts/wayland-session
--- sddm-0.13.0/data/scripts/wayland-session	2015-11-04 21:41:36.000000000 +0100
+++ sddm-0.13.0-p/data/scripts/wayland-session	2016-04-08 17:09:56.967351850 +0200
@@ -13,13 +13,6 @@
     [ -z "$BASH" ] && exec $SHELL $0 "$@"
     set +o posix
     [ -f /etc/profile ] && . /etc/profile
-    if [ -f $HOME/.bash_profile ]; then
-      . $HOME/.bash_profile
-    elif [ -f $HOME/.bash_login ]; then
-      . $HOME/.bash_login
-    elif [ -f $HOME/.profile ]; then
-      . $HOME/.profile
-    fi
     ;;
 */zsh)
     [ -z "$ZSH_NAME" ] && exec $SHELL $0 "$@"
@@ -27,22 +20,19 @@
     zhome=${ZDOTDIR:-$HOME}
     # zshenv is always sourced automatically.
     [ -f $zdir/zprofile ] && . $zdir/zprofile
-    [ -f $zhome/.zprofile ] && . $zhome/.zprofile
     [ -f $zdir/zlogin ] && . $zdir/zlogin
-    [ -f $zhome/.zlogin ] && . $zhome/.zlogin
     emulate -R sh
     ;;
   */csh|*/tcsh)
     # [t]cshrc is always sourced automatically.
     # Note that sourcing csh.login after .cshrc is non-standard.
     wlsess_tmp=`mktemp /tmp/wlsess-env-XXXXXX`
-    $SHELL -c "if (-f /etc/csh.login) source /etc/csh.login; if (-f ~/.login) source ~/.login; /bin/sh -c 'export -p' >! $wlsess_tmp"
+    $SHELL -c "if (-f /etc/csh.login) source /etc/csh.login; /bin/sh -c 'export -p' >! $wlsess_tmp"
     . $wlsess_tmp
     rm -f $wlsess_tmp
     ;;
   *) # Plain sh, ksh, and anything we do not know.
     [ -f /etc/profile ] && . /etc/profile
-    [ -f $HOME/.profile ] && . $HOME/.profile
     ;;
 esac
 
diff -Naur sddm-0.13.0/data/scripts/Xsession sddm-0.13.0-p/data/scripts/Xsession
--- sddm-0.13.0/data/scripts/Xsession	2015-11-04 21:41:36.000000000 +0100
+++ sddm-0.13.0-p/data/scripts/Xsession	2016-04-08 17:10:31.710265414 +0200
@@ -12,13 +12,6 @@
     [ -z "$BASH" ] && exec $SHELL $0 "$@"
     set +o posix
     [ -f /etc/profile ] && . /etc/profile
-    if [ -f $HOME/.bash_profile ]; then
-      . $HOME/.bash_profile
-    elif [ -f $HOME/.bash_login ]; then
-      . $HOME/.bash_login
-    elif [ -f $HOME/.profile ]; then
-      . $HOME/.profile
-    fi
     ;;
 */zsh)
     [ -z "$ZSH_NAME" ] && exec $SHELL $0 "$@"
@@ -26,27 +19,23 @@
     zhome=${ZDOTDIR:-$HOME}
     # zshenv is always sourced automatically.
     [ -f $zdir/zprofile ] && . $zdir/zprofile
-    [ -f $zhome/.zprofile ] && . $zhome/.zprofile
     [ -f $zdir/zlogin ] && . $zdir/zlogin
-    [ -f $zhome/.zlogin ] && . $zhome/.zlogin
     emulate -R sh
     ;;
   */csh|*/tcsh)
     # [t]cshrc is always sourced automatically.
     # Note that sourcing csh.login after .cshrc is non-standard.
     xsess_tmp=`mktemp /tmp/xsess-env-XXXXXX`
-    $SHELL -c "if (-f /etc/csh.login) source /etc/csh.login; if (-f ~/.login) source ~/.login; /bin/sh -c 'export -p' >! $xsess_tmp"
+    $SHELL -c "if (-f /etc/csh.login) source /etc/csh.login;  /bin/sh -c 'export -p' >! $xsess_tmp"
     . $xsess_tmp
     rm -f $xsess_tmp
     ;;
   *) # Plain sh, ksh, and anything we do not know.
     [ -f /etc/profile ] && . /etc/profile
-    [ -f $HOME/.profile ] && . $HOME/.profile
     ;;
 esac
 
 [ -f /etc/xprofile ] && . /etc/xprofile
-[ -f $HOME/.xprofile ] && . $HOME/.xprofile
 
 # run all system xinitrc shell scripts.
 if [ -d /etc/X11/xinit/xinitrc.d ]; then
