diff -Naur sddm-0.16.0/src/common/Configuration.h sddm-0.16.0-p/src/common/Configuration.h
--- sddm-0.16.0/src/common/Configuration.h	2017-10-05 08:24:24.000000000 +0200
+++ sddm-0.16.0-p/src/common/Configuration.h	2017-10-12 01:02:35.808843082 +0200
@@ -62,7 +62,7 @@
             Entry(ServerArguments,     QString,     _S("-nolisten tcp"),                        _S("Arguments passed to the X server invocation"));
             Entry(XephyrPath,          QString,     _S("/usr/bin/Xephyr"),                      _S("Path to Xephyr binary"));
             Entry(XauthPath,           QString,     _S("/usr/bin/xauth"),                       _S("Path to xauth binary"));
-            Entry(SessionDir,          QString,     _S("/usr/share/xsessions"),                 _S("Directory containing available X sessions"));
+            Entry(SessionDir,          QString,     _S("/etc/X11/sessions"),                    _S("Directory containing available X sessions"));
             Entry(SessionCommand,      QString,     _S(SESSION_COMMAND),                        _S("Path to a script to execute when starting the desktop session"));
 	    Entry(SessionLogFile,      QString,     _S(".local/share/sddm/xorg-session.log"),   _S("Path to the user session log file"));
 	    Entry(UserAuthFile,        QString,     _S(".Xauthority"),                          _S("Path to the Xauthority file"));
