--- garcon-0.1.9/data/xfce/xfce-applications.menu	2011-09-12 20:24:20.000000000 +0200
+++ garcon-0.1.9new/data/xfce/xfce-applications.menu	2011-12-30 18:54:37.000000000 +0100
@@ -162,4 +162,12 @@
         </Include>
     </Menu>
 
+    <Menu>
+    	<Name>Frugalware</Name>
+    	<Directory>Frugalware.directory</Directory>
+    	<Include>
+      		<Category>Frugalware</Category>
+    	</Include>
+    </Menu>
+
 </Menu>
