diff -Nurd Module.orig/rtmp_data.c Module/rtmp_data.c
--- Module.orig/rtmp_data.c	2006-06-17 22:12:58.000000000 +0200
+++ Module/rtmp_data.c	2007-07-15 21:26:25.000000000 +0200
@@ -1196,7 +1196,7 @@
 
  	                skb->dev = pAdapter->net_dev;
  	                memcpy(skb_put(skb, pRxD->DataByteCnt), pData, pRxD->DataByteCnt);
- 	                skb->mac.raw = skb->data;
+					skb_reset_mac_header(skb);
  	                skb->pkt_type = PACKET_OTHERHOST;
  	                skb->protocol = htons(ETH_P_802_2);
  	                skb->ip_summed = CHECKSUM_NONE;
diff -Nurd Module.orig/rtmp.h Module/rtmp.h
--- Module.orig/rtmp.h	2007-07-15 20:37:46.000000000 +0200
+++ Module/rtmp.h	2007-07-15 21:26:25.000000000 +0200
@@ -62,6 +62,13 @@
 #endif /*(LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) */
 #endif /* pci_name */
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,22))
+#define pci_module_init	pci_register_driver
+#endif
+
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,22))
+#define skb_reset_mac_header(skb) (skb->mac.raw = skb->data)
+#endif
 
 // Krellan: Limit range of user TxPower settings from -31 to +0 dBm.
 // We could accept -31 to +31 dBm, relative to 0 dBm which is defined
diff -Nurd Module.orig/rtmp_main.c Module/rtmp_main.c
--- Module.orig/rtmp_main.c	2007-07-15 20:37:46.000000000 +0200
+++ Module/rtmp_main.c	2007-07-15 21:26:52.000000000 +0200
@@ -340,7 +340,11 @@
         NICDisableInterrupt(pAd);
     }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,22)
+    status = request_irq(pAd->pPci_Dev->irq, &RTMPIsr, IRQF_SHARED, net_dev->name, net_dev);
+#else
     status = request_irq(pAd->pPci_Dev->irq, &RTMPIsr, SA_SHIRQ, net_dev->name, net_dev);
+#endif
     if (status)
     {
         goto out_module_put;
@@ -950,7 +954,8 @@
     PRTMP_ADAPTER pAdapter = (PRTMP_ADAPTER) dev->priv;
     int status;
 
-    pci_enable_device(pdev);
+    if (pci_enable_device(pdev))
+       return -EIO;
 
     printk(KERN_NOTICE "%s: got resume request\n", dev->name);
 
