--- Module/rtmp_data.c.orig	2007-07-10 15:16:31.000000000 +0200
+++ Module/rtmp_data.c	2007-07-10 15:16:51.000000000 +0200
@@ -1196,7 +1196,7 @@
 
  	                skb->dev = pAdapter->net_dev;
  	                memcpy(skb_put(skb, pRxD->DataByteCnt), pData, pRxD->DataByteCnt);
- 	                skb->mac.raw = skb->data;
+			skb_reset_mac_header(skb);
  	                skb->pkt_type = PACKET_OTHERHOST;
  	                skb->protocol = htons(ETH_P_802_2);
  	                skb->ip_summed = CHECKSUM_NONE;
