diff -Naur NetworkManager-1.10.0/src/nm-manager.c NetworkManager-1.10.0-p/src/nm-manager.c
--- NetworkManager-1.10.0/src/nm-manager.c	2017-11-10 16:46:40.000000000 +0100
+++ NetworkManager-1.10.0-p/src/nm-manager.c	2017-11-12 12:30:51.852368464 +0100
@@ -6250,7 +6250,6 @@
 	NMConfigData *config_data;
 	const NMGlobalDnsConfig *dns_config;
 	const char *type;
-	NMConnectivity *connectivity;
 
 	switch (prop_id) {
 	case PROP_VERSION:
@@ -6300,8 +6299,16 @@
 		g_value_set_boolean (value, nm_config_data_get_connectivity_uri (config_data) != NULL);
 		break;
 	case PROP_CONNECTIVITY_CHECK_ENABLED:
+#if WITH_CONCHECK
+        {
+                NMConnectivity *connectivity;
+
 		connectivity = nm_connectivity_get ();
 		g_value_set_boolean (value, nm_connectivity_check_enabled (connectivity));
+        }
+#else
+                g_value_set_boolean (value, FALSE);
+#endif
 		break;
 	case PROP_PRIMARY_CONNECTION:
 		nm_utils_g_value_set_object_path (value, priv->primary_connection);
diff -Naur NetworkManager-1.10.0/src/tests/config/test-config.c NetworkManager-1.10.0-p/src/tests/config/test-config.c
--- NetworkManager-1.10.0/src/tests/config/test-config.c	2017-11-10 16:46:25.000000000 +0100
+++ NetworkManager-1.10.0-p/src/tests/config/test-config.c	2017-11-12 13:33:06.590133823 +0100
@@ -318,6 +318,7 @@
 	g_object_unref (config);
 }
 
+#if WITH_CONCHECK
 static void
 test_config_connectivity_check (void)
 {
@@ -351,6 +352,7 @@
 
 	g_assert (remove (CONFIG_INTERN) == 0);
 }
+#endif
 
 static void
 test_config_no_auto_default (void)
@@ -1053,8 +1055,9 @@
 
 	g_test_add_func ("/config/set-values", test_config_set_values);
 	g_test_add_func ("/config/global-dns", test_config_global_dns);
+#if WITH_CONCHECK
 	g_test_add_func ("/config/connectivity-check", test_config_connectivity_check);
-
+#endif
 	g_test_add_func ("/config/signal", test_config_signal);
 
 	g_test_add_func ("/config/enable", test_config_enable);
