--- NetworkManager-1.8.2/src/devices/nm-device.c	2017-07-07 14:03:25.000000000 +0200
+++ NetworkManager-1.8.2.new/src/devices/nm-device.c	2017-09-09 17:52:40.000000000 +0200
@@ -10243,7 +10243,7 @@
 	NMDevicePrivate *priv = NM_DEVICE_GET_PRIVATE (self);
 	gs_free char *str_timeout = NULL;
 	gs_free char *tmp_str = NULL;
-	const char *args[] = { priv->gw_ping.binary, "-I", nm_device_get_ip_iface (self),
+	const char *args[] = { priv->gw_ping.binary, "-e", nm_device_get_ip_iface (self),
 	                       "-c", "1", "-w", NULL, priv->gw_ping.address, NULL };
 	gs_free_error GError *error = NULL;
 	gboolean ret;
