diff -aur rubygems-1.1.1/setup.rb fw_rubygems-1.1.1/setup.rb
--- rubygems-1.1.1/setup.rb	2008-04-02 01:22:20.000000000 +0200
+++ fw_rubygems-1.1.1/setup.rb	2008-04-27 03:29:37.000000000 +0200
@@ -94,7 +94,7 @@
     lib_dir = Gem::ConfigMap[:sitelibdir]
     bin_dir = Gem::ConfigMap[:bindir]
   else
-    lib_dir = File.join prefix, 'lib'
+    lib_dir = File.join prefix, 'lib/ruby/site_ruby/1.8'
     bin_dir = File.join prefix, 'bin'
 
     mkdir_p lib_dir
