diff -Naur rubygems-0.9.2/post-install.rb rubygems-0.9.2-p/post-install.rb
--- rubygems-0.9.2/post-install.rb	2007-02-02 13:37:35.000000000 +0100
+++ rubygems-0.9.2-p/post-install.rb	2007-02-22 04:25:18.000000000 +0100
@@ -77,7 +77,8 @@
 end
 
 def install_sources
-  $: << "lib"
+  #$: << "lib"
+  $:.unshift "lib"
   require 'rubygems'
   require 'rubygems/builder'
   require 'rubygems/installer'
@@ -85,7 +86,7 @@
     load "sources.gemspec"
     spec = Gem.sources_spec
     gem_file = Gem::Builder.new(spec).build
-    Gem::Installer.new(gem_file).install(true, Gem.dir, false)
+    #Gem::Installer.new(gem_file).install(false, Gem.dir, false)
   end
 end
 
@@ -114,8 +115,8 @@
 end
 
 install_sources
-remove_old_rdoc
-install_rdoc
+#remove_old_rdoc
+#install_rdoc
 install_windows_batch_files
-remove_stubs
+#remove_stubs
 
