diff -Naur sed-4.1.4.orig/doc/Makefile.in sed-4.1.4/doc/Makefile.in
--- sed-4.1.4.orig/doc/Makefile.in	2005-01-27 08:58:18.000000000 +0100
+++ sed-4.1.4/doc/Makefile.in	2005-02-03 10:30:19.527107736 +0100
@@ -622,9 +622,7 @@
 @MAKEINFO_HTML_TRUE@.PHONY: html
 
 @TEXI2HTML_HTML_TRUE@sed.html: sed.texi sed.info
-@TEXI2HTML_HTML_TRUE@	cp $(srcdir)/*.texi . && \
-@TEXI2HTML_HTML_TRUE@	  $(TEXI2HTML) sed.texi && \
-@TEXI2HTML_HTML_TRUE@	  rn *.texi
+@TEXI2HTML_HTML_TRUE@	  $(TEXI2HTML) sed.texi
 
 @TEXI2HTML_HTML_TRUE@html: sed.html
 
