--- src/org/objectweb/util/ant/MultipleCopy.java	2007-07-07 13:23:10.000000000 -0100
+++ src/org/objectweb/util/ant/MultipleCopy.java	2008-06-18 13:23:35.000000000 -0100
@@ -101,8 +101,8 @@
 		}
 
 		// deal with the filesets
-		for (int i = 0; i < filesets.size(); i++) {
-			FileSet fs = (FileSet) filesets.elementAt(i);
+		for (int i = 0; i < rcs.size(); i++) {
+			FileSet fs = (FileSet) rcs.elementAt(i);
 			DirectoryScanner ds = fs.getDirectoryScanner(getProject());
 			File fromDir = fs.getDir(getProject());
 
