diff -Naur cmatrix-1.2a.orig/Makefile.am cmatrix-1.2a/Makefile.am
--- cmatrix-1.2a.orig/Makefile.am	2006-05-06 15:32:59.087029000 +0200
+++ cmatrix-1.2a/Makefile.am	2006-05-06 15:33:05.539048144 +0200
@@ -23,7 +23,6 @@
 	    $(INSTALL_DATA) $(srcdir)/mtx.pcf /usr/lib/X11/fonts/misc; \
 	    $(INSTALL_DATA) $(srcdir)/mtx.pcf /usr/lib/X11/fonts/misc; \
 	    echo " Running mkfontdir /usr/lib/X11/fonts/misc..."; \
-	    $(MKFONTDIR) /usr/lib/X11/fonts/misc; \
 	    echo " Done.  If this is the first time you have installed CMatrix you will"; \
 	    echo " probably have to restart X window in order to use the mtx.pcf font."; \
 	else \
@@ -32,7 +31,6 @@
 	    $(INSTALL_DATA) $(srcdir)/mtx.pcf /var/tmp/fst/pkg/usr/X11R6/lib/X11/fonts/misc; \
 	    $(INSTALL_DATA) $(srcdir)/mtx.pcf /var/tmp/fst/pkg/usr/X11R6/lib/X11/fonts/misc; \
 	    echo " Running mkfontdir /usr/X11R6/lib/X11/fonts/misc..."; \
-	    $(MKFONTDIR) /usr/X11R6/lib/X11/fonts/misc; \
 	    echo " Done.  If this is the first time you have installed CMatrix you will"; \
 	    echo " probably have to restart X window in order to use the mtx.pcf font."; \
 	fi \
diff -Naur cmatrix-1.2a.orig/Makefile.in cmatrix-1.2a/Makefile.in
--- cmatrix-1.2a.orig/Makefile.in	2006-05-06 15:32:59.087029000 +0200
+++ cmatrix-1.2a/Makefile.in	2006-05-06 15:33:16.335406848 +0200
@@ -418,7 +418,6 @@
 	    $(INSTALL_DATA) $(srcdir)/mtx.pcf /var/tmp/fst/pkg/usr/lib/X11/fonts/misc; \
 	    $(INSTALL_DATA) $(srcdir)/mtx.pcf /var/tmp/fst/pkg/usr/lib/X11/fonts/misc; \
 	    echo " Running mkfontdir /usr/lib/X11/fonts/misc..."; \
-	    $(MKFONTDIR) /var/tmp/fst/pkg/usr/lib/X11/fonts/misc; \
 	    echo " Done.  If this is the first time you have installed CMatrix you will"; \
 	    echo " probably have to restart X window in order to use the mtx.pcf font."; \
 	else \
@@ -427,7 +426,6 @@
 	    $(INSTALL_DATA) $(srcdir)/mtx.pcf /var/tmp/fst/pkg/usr/X11R6/lib/X11/fonts/misc; \
 	    $(INSTALL_DATA) $(srcdir)/mtx.pcf /var/tmp/fst/pkg/usr/X11R6/lib/X11/fonts/misc; \
 	    echo " Running mkfontdir /usr/X11R6/lib/X11/fonts/misc..."; \
-	    $(MKFONTDIR) /var/tmp/fst/pkg/usr/X11R6/lib/X11/fonts/misc; \
 	    echo " Done.  If this is the first time you have installed CMatrix you will"; \
 	    echo " probably have to restart X window in order to use the mtx.pcf font."; \
 	fi \
