diff -Naur ghc-6.4.orig/mk/config.mk.in ghc-6.4/mk/config.mk.in
--- ghc-6.4.orig/mk/config.mk.in	2005-05-27 12:18:16.000000000 +0000
+++ ghc-6.4/mk/config.mk.in	2005-05-27 12:20:26.000000000 +0000
@@ -361,6 +361,13 @@
 ifeq "$(TARGETPLATFORM)" "ia64-unknown-linux"
 SplitObjs=NO
 endif
+ifeq "$(TARGETPLATFORM)" "x86_64-unknown-linux"
+SplitObjs=NO
+endif
+ifeq "$(TARGETPLATFORM)" "x86_64-unknown-linux-gnu"
+SplitObjs=NO
+endif
+
 
 # ----------------------------------------------------------------------------
 # Options for GHC's RTS
