diff -Naur kmess-1.5.1/kmess/crypt/sha1.cpp kmess-1.5.1-p/kmess/crypt/sha1.cpp
--- kmess-1.5.1/kmess/crypt/sha1.cpp	2008-02-23 13:02:43.000000000 +0100
+++ kmess-1.5.1-p/kmess/crypt/sha1.cpp	2008-06-11 10:55:14.000000000 +0200
@@ -36,6 +36,7 @@
 #include "bithelp.h"
 
 #include <stdio.h>
+#include <string.h>
 #include <sys/stat.h>
 
 Sha1::Sha1()
diff -Naur kmess-1.5.1/kmess/notification/passivepopupview.h kmess-1.5.1-p/kmess/notification/passivepopupview.h
--- kmess-1.5.1/kmess/notification/passivepopupview.h	2008-03-12 22:26:46.000000000 +0100
+++ kmess-1.5.1-p/kmess/notification/passivepopupview.h	2008-06-11 10:55:00.000000000 +0200
@@ -43,7 +43,7 @@
 
   public:  // Public methods
     // Class constructor
-    PassivePopupView(  QWidget *parent, QString icon, QString message, QString icon = QString::null );
+    PassivePopupView(  QWidget *parent, QString icon, QString message, QString icon2 = QString::null );
     // Changes the popup icon
     void changeIcon( QString icon );
     // Changes the popup status icon
