--- vinagre-2.91.91/data/org.gnome.Vinagre.gschema.xml.in.old	2011-03-17 22:05:09.000000000 +0100
+++ vinagre-2.91.91/data/org.gnome.Vinagre.gschema.xml.in	2011-03-17 22:05:53.000000000 +0100
@@ -27,7 +27,7 @@
       <_description>Set to "true" to always start the program listening for reverse connections.</_description>
     </key>
     <key name="active-plugins" type="as">
-      <default>[ 'vnc' ]</default>
+      <default>['rdp','vnc' ]</default>
       <_summary>Active plugins</_summary>
       <_description>List of active plugins. It contains the "Location" of the active plugins. See the .vinagre-plugin file for obtaining the "Location" of a given plugin.</_description>
     </key>
