diff -Naur vinagre-3.18.2/plugins/rdp/vinagre-rdp-tab.c vinagre-3.18.2-p/plugins/rdp/vinagre-rdp-tab.c
--- vinagre-3.18.2/plugins/rdp/vinagre-rdp-tab.c	2015-10-06 17:40:06.000000000 +0200
+++ vinagre-3.18.2-p/plugins/rdp/vinagre-rdp-tab.c	2016-01-21 21:37:58.811908788 +0100
@@ -1108,7 +1108,8 @@
   settings->encryption_level = ENCRYPTION_LEVEL_CLIENT_COMPATIBLE;
 #endif
 #include <freerdp/version.h>
-#if (FREERDP_VERSION_MAJOR == 1 && FREERDP_VERSION_MINOR >= 2 && FREERDP_VERSION_REVISION >= 1)
+#if (FREERDP_VERSION_MAJOR == 1 && FREERDP_VERSION_MINOR >= 2 && FREERDP_VERSION_REVISION >= 1) || \
+	 (FREERDP_VERSION_MAJOR == 2 && FREERDP_VERSION_MINOR >= 0 && FREERDP_VERSION_REVISION >= 0)
   settings->UseRdpSecurityLayer = FALSE;
 #else
   settings->DisableEncryption = FALSE;
