--- Makefile.old	2007-06-24 19:41:29.000000000 +0200
+++ Makefile	2007-06-24 19:41:02.000000000 +0200
@@ -32,11 +32,11 @@
 	rm -rf $(DIR_BIN)
 	
 install:
-	mkdir -p $(DIR_INSTALL_BIN)
-	cp -R $(DIR_BIN)/* $(DIR_INSTALL_BIN)
+	mkdir -p ${DESTDIR}/$(DIR_INSTALL_BIN)
+	cp -R $(DIR_BIN)/* ${DESTDIR}/$(DIR_INSTALL_BIN)
 ifndef RPM_DOC_DIR
-	mkdir -p $(DIR_INSTALL_DOC)
-	cp -R $(DIR_DOC)/* $(DIR_INSTALL_DOC)
+	mkdir -p ${DESTDIR}/$(DIR_INSTALL_DOC)
+	cp -R $(DIR_DOC)/* ${DESTDIR}/$(DIR_INSTALL_DOC)
 endif
 
 .PHONY: clean buildall

