diff -Naur skrooge-2.8.1/skgbasemodeler/CMakeLists.txt skrooge-2.8.1-p/skgbasemodeler/CMakeLists.txt
--- skrooge-2.8.1/skgbasemodeler/CMakeLists.txt	2017-04-15 21:06:47.000000000 +0200
+++ skrooge-2.8.1-p/skgbasemodeler/CMakeLists.txt	2017-06-15 14:35:12.364409572 +0200
@@ -55,15 +55,15 @@
 #need to link to some other libraries ? just add them here
 SET_TARGET_PROPERTIES(skgbasemodeler PROPERTIES VERSION ${SKG_VERSION} SOVERSION ${SOVERSION} )
 IF(Qt5_VERSION VERSION_LESS 5.7.0)
-    INCLUDE_DIRECTORIES(${SQLCIPHER_INCLUDE_DIRS} /usr/include/${Qca_LIBRARY})
+    INCLUDE_DIRECTORIES(${SQLCIPHER_INCLUDE_DIRS} /usr/include/Qca-q5/QtCrypto)
     TARGET_LINK_LIBRARIES(skgbasemodeler LINK_PUBLIC KF5::I18n KF5::IconThemes KF5::ConfigWidgets KF5::CoreAddons KF5::KDELibs4Support
-    Qt5::Core Qt5::Gui Qt5::Sql Qt5::Script Qt5::Xml Qt5::DBus Qt5::Concurrent qca-qt5
+    Qt5::Core Qt5::Gui Qt5::Sql Qt5::Script Qt5::Xml Qt5::DBus Qt5::Concurrent qca-qt5
     Grantlee5::Templates
     ${SQLCIPHER_LIBRARIES} )
 ELSE()
-    INCLUDE_DIRECTORIES(${SQLCIPHER_INCLUDE_DIRS} /usr/include/${Qca_LIBRARY})
+    INCLUDE_DIRECTORIES(${SQLCIPHER_INCLUDE_DIRS} /usr/include/Qca-qt5/QtCrypto)
     TARGET_LINK_LIBRARIES(skgbasemodeler LINK_PUBLIC KF5::I18n KF5::IconThemes KF5::ConfigWidgets KF5::CoreAddons KF5::KIOCore
-    Qt5::Core Qt5::Gui Qt5::Sql Qt5::Script Qt5::Xml Qt5::DBus Qt5::Concurrent qca-qt5
+    Qt5::Core Qt5::Gui Qt5::Sql Qt5::Script Qt5::Xml Qt5::DBus Qt5::Concurrent qca-qt5
     Grantlee5::Templates
     ${SQLCIPHER_LIBRARIES} )
 ENDIF()
