diff -Naur ruby-1.8.4/file.c ruby-1.8.4-p/file.c
--- ruby-1.8.4/file.c	2005-12-21 10:20:15.000000000 +0100
+++ ruby-1.8.4-p/file.c	2006-04-15 21:27:58.000000000 +0200
@@ -4328,8 +4328,8 @@
  *  read-only, which is reported as <code>0444</code>.
  */
 
-void
-Init_File()
+
+Init_File(void)
 {
     rb_mFileTest = rb_define_module("FileTest");
     rb_cFile = rb_define_class("File", rb_cIO);
diff -Naur ruby-1.8.4/gc.c ruby-1.8.4-p/gc.c
--- ruby-1.8.4/gc.c	2005-12-16 05:58:51.000000000 +0100
+++ ruby-1.8.4-p/gc.c	2006-04-15 20:39:52.000000000 +0200
@@ -422,7 +422,7 @@
 # define STACK_LEVEL_MAX (_stacksize - 4096)
 # undef HAVE_GETRLIMIT
 #elif defined(HAVE_GETRLIMIT)
-static unsigned int STACK_LEVEL_MAX = 655300;
+static unsigned long STACK_LEVEL_MAX = 655300;
 #else
 # define STACK_LEVEL_MAX 655300
 #endif
@@ -482,7 +482,7 @@
     (ret) = (STACK_LENGTH > STACK_LEVEL_MAX + GC_WATER_MARK);\
 } while (0)
 
-int
+long
 ruby_stack_length(p)
     VALUE **p;
 {
diff -Naur ruby-1.8.4/intern.h ruby-1.8.4-p/intern.h
--- ruby-1.8.4/intern.h	2005-11-23 03:06:00.000000000 +0100
+++ ruby-1.8.4-p/intern.h	2006-04-15 21:26:33.000000000 +0200
@@ -221,7 +221,7 @@
 void rb_thread_atfork _((void));
 VALUE rb_funcall_rescue __((VALUE, ID, int, ...));
 /* file.c */
-int eaccess _((const char*, int));
+//int eaccess _((const char*, int)); 
 VALUE rb_file_s_expand_path _((int, VALUE *));
 VALUE rb_file_expand_path _((VALUE, VALUE));
 void rb_file_const _((const char*, VALUE));
@@ -234,7 +234,7 @@
 /* gc.c */
 NORETURN(void rb_memerror __((void)));
 int ruby_stack_check _((void));
-int ruby_stack_length _((VALUE**));
+long ruby_stack_length _((VALUE**));
 char *rb_source_filename _((const char*));
 void rb_gc_mark_locations _((VALUE*, VALUE*));
 void rb_mark_tbl _((struct st_table*));
