--- a/src/app.c	2011-01-26 16:49:32.579000322 +0100
+++ a/src/app.c.new	2011-01-26 16:50:55.216000058 +0100
@@ -1269,8 +1269,7 @@
   self->priv->cutted      = NULL;
   self->priv->notify      = notify_notification_new("Rubrica Addressbook",
 						    NULL, 
-						    "info",
-						    NULL);
+						    "info");
   self->priv->show_card   = TRUE;
 
   g_object_set(self->priv->notebook, "homogeneous", FALSE, NULL);
