diff -up xf86-video-xgi-1.5.0/src/xgi_setup.c.xorg-server-1.5 xf86-video-xgi-1.5.0/src/xgi_setup.c
--- xf86-video-xgi-1.5.0/src/xgi_setup.c.xorg-server-1.5	2008-05-19 00:15:42.000000000 +0900
+++ xf86-video-xgi-1.5.0/src/xgi_setup.c	2008-05-19 00:15:49.000000000 +0900
@@ -39,8 +39,6 @@
 #include "xf86Pci.h"
 #include "xf86.h"
 #include "fb.h"
-#include "xf1bpp.h"
-#include "xf4bpp.h"
 #include "xf86_OSproc.h"
 #include "xf86Resources.h"
 #include "xf86Version.h"

diff -uNr xf86-video-xgi-1.5.0/src/vgatypes.h xf86-video-xgi-1.5.0.new/src/vgatypes.h
--- xf86-video-xgi-1.5.0/src/vgatypes.h	2007-08-28 06:55:51.000000000 +0900
+++ xf86-video-xgi-1.5.0.new/src/vgatypes.h	2009-03-12 19:55:10.000000000 +0900
@@ -56,7 +56,8 @@
 #include "osdef.h"
 
 #ifdef LINUX_XF86
-#include "xf86Version.h"
+#include <xorg-server.h>
+#include <xorgVersion.h>
 #include "xf86Pci.h"
 #endif
 
diff -uNr xf86-video-xgi-1.5.0/src/xgi.h xf86-video-xgi-1.5.0.new/src/xgi.h
--- xf86-video-xgi-1.5.0/src/xgi.h	2007-08-28 08:01:51.000000000 +0900
+++ xf86-video-xgi-1.5.0.new/src/xgi.h	2009-03-12 19:55:46.000000000 +0900
@@ -108,9 +108,7 @@
 #ifdef XF86DRI
 #define XGINEWDRI
 #undef XGINEWDRI2
-#if XF86_VERSION_CURRENT >= XF86_VERSION_NUMERIC(4,4,99,99,0)	/* Adapt this when the time has come */
 #define XGINEWDRI2
-#endif
 #include "xf86drm.h"
 #include "sarea.h"
 #define _XF86DRI_SERVER_
diff -uNr xf86-video-xgi-1.5.0/src/xgi_dac.c xf86-video-xgi-1.5.0.new/src/xgi_dac.c
--- xf86-video-xgi-1.5.0/src/xgi_dac.c	2007-06-26 07:39:57.000000000 +0900
+++ xf86-video-xgi-1.5.0.new/src/xgi_dac.c	2009-03-12 19:56:19.000000000 +0900
@@ -47,7 +47,8 @@
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Version.h"
+#include <xorg-server.h>
+#include <xorgVersion.h>
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
 #include "xf86DDC.h"
diff -uNr xf86-video-xgi-1.5.0/src/xgi_driver.c xf86-video-xgi-1.5.0.new/src/xgi_driver.c
--- xf86-video-xgi-1.5.0/src/xgi_driver.c	2007-08-28 07:56:34.000000000 +0900
+++ xf86-video-xgi-1.5.0.new/src/xgi_driver.c	2009-03-12 19:56:48.000000000 +0900
@@ -55,7 +55,8 @@
 #include "xf86_OSproc.h"
 #include "xf86Resources.h"
 #include "dixstruct.h"
-#include "xf86Version.h"
+#include <xorg-server.h>
+#include <xorgVersion.h>
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
 #include "xf86cmap.h"
diff -uNr xf86-video-xgi-1.5.0/src/xgi_setup.c xf86-video-xgi-1.5.0.new/src/xgi_setup.c
--- xf86-video-xgi-1.5.0/src/xgi_setup.c	2009-03-12 19:54:44.000000000 +0900
+++ xf86-video-xgi-1.5.0.new/src/xgi_setup.c	2009-03-12 19:57:09.000000000 +0900
@@ -41,7 +41,8 @@
 #include "fb.h"
 #include "xf86_OSproc.h"
 #include "xf86Resources.h"
-#include "xf86Version.h"
+#include <xorg-server.h>
+#include <xorgVersion.h>
 
 #include "xf86cmap.h"
 
diff -uNr xf86-video-xgi-1.5.0/src/xgi_vga.c xf86-video-xgi-1.5.0.new/src/xgi_vga.c
--- xf86-video-xgi-1.5.0/src/xgi_vga.c	2007-06-26 07:47:42.000000000 +0900
+++ xf86-video-xgi-1.5.0.new/src/xgi_vga.c	2009-03-12 19:57:40.000000000 +0900
@@ -38,7 +38,8 @@
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Version.h"
+#include <xorg-server.h>
+#include <xorgVersion.h>
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
 

