From 88c1dba4f9d28df97c996672f0267f6b2eb03f32 Mon Sep 17 00:00:00 2001
From: Robert Piasek <dagger@gentoo.org>
Date: Thu, 16 Jun 2011 12:42:25 +0100
Subject: [PATCH] Fix compilation against iptables 1.4.11

---
 src/iptables.c        |   16 ++++++++++++----
 tools/iptables-test.c |   16 ++++++++++++----
 2 files changed, 24 insertions(+), 8 deletions(-)

diff --git a/src/iptables.c b/src/iptables.c
index c799c25..f70be45 100644
--- a/src/iptables.c
+++ b/src/iptables.c
@@ -1088,9 +1088,13 @@ static int iptables_command(int argc, char *argv[])
 			if (xt_t->init != NULL)
 				xt_t->init(xt_t->t);
 			iptables_globals.opts =
-				xtables_merge_options(iptables_globals.opts,
-						     xt_t->extra_opts,
-						     &xt_t->option_offset);
+				xtables_merge_options(
+#if XTABLES_VERSION_CODE > 5
+						iptables_globals.orig_opts,
+#endif
+						iptables_globals.opts,
+						xt_t->extra_opts,
+						&xt_t->option_offset);
 			if (iptables_globals.opts == NULL)
 				goto out;
 
@@ -1112,7 +1116,11 @@ static int iptables_command(int argc, char *argv[])
 				xt_m->init(xt_m->m);
 			if (xt_m != xt_m->next) {
 				iptables_globals.opts =
-				xtables_merge_options(iptables_globals.opts,
+				xtables_merge_options(
+#if XTABLES_VERSION_CODE > 5
+						iptables_globals.orig_opts,
+#endif
+						iptables_globals.opts,
 						xt_m->extra_opts,
 						&xt_m->option_offset);
 				if (iptables_globals.opts == NULL)
diff --git a/tools/iptables-test.c b/tools/iptables-test.c
index e316cc9..7c5f05e 100644
--- a/tools/iptables-test.c
+++ b/tools/iptables-test.c
@@ -1075,9 +1075,13 @@ int main(int argc, char *argv[])
 			if (xt_t->init != NULL)
 				xt_t->init(xt_t->t);
 			connman_iptables_globals.opts =
-				xtables_merge_options(connman_iptables_globals.opts,
-						     xt_t->extra_opts,
-						     &xt_t->option_offset);
+				xtables_merge_options(
+#if XTABLES_VERSION_CODE > 5
+						connman_iptables_globals.orig_opts,
+#endif
+						connman_iptables_globals.opts,
+						xt_t->extra_opts,
+						&xt_t->option_offset);
 			if (connman_iptables_globals.opts == NULL)
 				goto out;
 
@@ -1101,7 +1105,11 @@ int main(int argc, char *argv[])
 				xt_m->init(xt_m->m);
 			if (xt_m != xt_m->next) {
 				connman_iptables_globals.opts =
-					xtables_merge_options(connman_iptables_globals.opts,
+					xtables_merge_options(
+#if XTABLES_VERSION_CODE > 5
+						connman_iptables_globals.orig_opts,
+#endif
+						connman_iptables_globals.opts,
 						xt_m->extra_opts,
 						&xt_m->option_offset);
 				if (connman_iptables_globals.opts == NULL)
-- 
1.7.5.3

