diff -Naur connman-1.31/configure.ac connman-1.31-p/configure.ac
--- connman-1.31/configure.ac	2015-12-28 04:05:49.000000000 +0100
+++ connman-1.31-p/configure.ac	2016-03-06 22:59:56.157434874 +0100
@@ -280,7 +280,7 @@
 				[enable PolicyKit support]),
 			[enable_polkit=${enableval}], [enable_polkit="no"])
 if (test "${enable_polkit}" != "no"); then
-	POLKIT_DATADIR="`$PKG_CONFIG --variable=actiondir polkit`"
+	POLKIT_DATADIR="`$PKG_CONFIG --variable=actiondir polkit-gobject-1`"
 	POLKIT_DATADIR=""
 	if (test -z "${POLKIT_DATADIR}"); then
 		POLKIT_DATADIR="${datadir}/polkit-1/actions"
