diff -Naur q4wine-1.3.7/src/q4wine-cli/CMakeLists.txt q4wine-1.3.7-p/src/q4wine-cli/CMakeLists.txt
--- q4wine-1.3.7/src/q4wine-cli/CMakeLists.txt	2018-04-26 23:33:08.000000000 +0200
+++ q4wine-1.3.7-p/src/q4wine-cli/CMakeLists.txt	2018-06-14 23:49:46.549209248 +0200
@@ -34,7 +34,7 @@
 ENDIF(${CMAKE_SYSTEM_NAME} MATCHES FreeBSD)
 
 if(QT5 MATCHES ON)
-    qt5_use_modules(q4wine-cli Core DBus Network Sql Widgets Xml)
+    target_link_libraries(q4wine-cli Qt5::Core Qt5::DBus Qt5::Network Qt5::Sql Qt5::Widgets Qt5::Xml)
     target_link_libraries(q4wine-cli q4wine-core ${BSD_LIBS})
 else(QT5 MATCHES ON)
 TARGET_LINK_LIBRARIES( q4wine-cli ${QT_LIBRARIES} q4wine-core ${BSD_LIBS})
diff -Naur q4wine-1.3.7/src/q4wine-gui/CMakeLists.txt q4wine-1.3.7-p/src/q4wine-gui/CMakeLists.txt
--- q4wine-1.3.7/src/q4wine-gui/CMakeLists.txt	2018-04-26 23:33:08.000000000 +0200
+++ q4wine-1.3.7-p/src/q4wine-gui/CMakeLists.txt	2018-06-14 23:51:22.298765328 +0200
@@ -209,7 +209,7 @@
 ENDIF(${CMAKE_SYSTEM_NAME} MATCHES FreeBSD)
 
 if (QT5 MATCHES ON)
-    qt5_use_modules(q4wine Core DBus Network Sql Widgets Xml)
+    target_link_libraries(q4wine Qt5::Core Qt5::DBus Qt5::Network Qt5::Sql Qt5::Widgets Qt5::Xml)
     target_link_libraries(q4wine q4wine-core ${BSD_LIBS} ${QTSINGLEAPPLIB})
 else (QT5 MATCHES ON)
 TARGET_LINK_LIBRARIES( q4wine ${QT_LIBRARIES} q4wine-core ${BSD_LIBS} ${QTSINGLEAPPLIB})
diff -Naur q4wine-1.3.7/src/q4wine-helper/CMakeLists.txt q4wine-1.3.7-p/src/q4wine-helper/CMakeLists.txt
--- q4wine-1.3.7/src/q4wine-helper/CMakeLists.txt	2018-04-26 23:33:08.000000000 +0200
+++ q4wine-1.3.7-p/src/q4wine-helper/CMakeLists.txt	2018-06-14 23:52:22.688116684 +0200
@@ -60,7 +60,7 @@
 ENDIF(${CMAKE_SYSTEM_NAME} MATCHES FreeBSD)
 
 if(QT5 MATCHES ON)
-    qt5_use_modules(q4wine-helper Core DBus Network Sql Widgets Xml)
+    target_link_libraries(q4wine-helper Qt5::Core Qt5::DBus Qt5::Network Qt5::Sql Qt5::Widgets Qt5::Xml)
     target_link_libraries(q4wine-helper q4wine-core ${BSD_LIBS})
 else(QT5 MATCHES ON)
 TARGET_LINK_LIBRARIES( q4wine-helper ${QT_LIBRARIES} q4wine-core ${BSD_LIBS})
diff -Naur q4wine-1.3.7/src/q4wine-lib/CMakeLists.txt q4wine-1.3.7-p/src/q4wine-lib/CMakeLists.txt
--- q4wine-1.3.7/src/q4wine-lib/CMakeLists.txt	2018-04-26 23:33:08.000000000 +0200
+++ q4wine-1.3.7-p/src/q4wine-lib/CMakeLists.txt	2018-06-14 23:58:37.434298424 +0200
@@ -85,7 +85,7 @@
 ENDIF( ${CMAKE_SYSTEM_NAME} MATCHES Darwin )
 
 if(QT5 MATCHES ON)
-    qt5_use_modules(q4wine-core Core DBus Network Sql Widgets Xml)
+    target_link_libraries(q4wine-core Qt5::Core Qt5::DBus Qt5::Network Qt5::Sql Qt5::Widgets Qt5::Xml)
 else(QT5 MATCHES ON)
 target_link_libraries( q4wine-core ${QT_LIBRARIES})
 endif(QT5 MATCHES ON)
diff -Naur q4wine-1.3.7/src/qtsingleapplication/CMakeLists.txt q4wine-1.3.7-p/src/qtsingleapplication/CMakeLists.txt
--- q4wine-1.3.7/src/qtsingleapplication/CMakeLists.txt	2018-04-26 23:33:08.000000000 +0200
+++ q4wine-1.3.7-p/src/qtsingleapplication/CMakeLists.txt	2018-06-14 23:54:46.091951293 +0200
@@ -23,5 +23,7 @@
 
 ADD_LIBRARY( qtsingleapp STATIC ${QSINGLEAPP_LIB_SOURCES} ${QSINGLEAPP_LIB_HEADERS} ${QTSINGLEAPPLICATION-LIB-MOC-SRCS}  )
 if(QT5 MATCHES ON)
-    qt5_use_modules(qtsingleapp Core Network Widgets)
-endif(QT5 MATCHES ON)
\ Kein Zeilenumbruch am Dateiende.
+    target_link_libraries(qtsingleapp Qt5::Core Qt5::Network Qt5::Widgets)
+endif(QT5 MATCHES ON)
+
+
