--- krusader-2.0.0.old/krusader/VFS/kiojobwrapper.cpp	2009-10-08 11:45:07.417003805 +0200
+++ krusader-2.0.0.new/krusader/VFS/kiojobwrapper.cpp	2009-10-08 11:45:55.356261793 +0200
@@ -120,7 +120,7 @@
 		job = PreservingCopyJob::createCopyJob( (PreserveMode)m_pmode, m_urlList, m_url, KIO::CopyJob::Move, false, m_showProgress );
 		break;
 	default:
-		fprintf( stderr, "Internal error: invalid job!\n" );
+		std::cerr << "Internal error: invalid job!\n";
 		break;
 	}
 	if( job ) {
