--- system-config-printer-1.0.8old/Makefile.in	2008-09-29 18:54:01.000000000 +0200
+++ system-config-printer-1.0.8/Makefile.in	2008-10-16 01:43:05.000000000 +0200
@@ -301,9 +301,6 @@
 	system-config-printer-applet		\
 	my-default-printer
 
-man_MANS = \
-	man/system-config-printer.1		\
-	man/system-config-printer-applet.1
 
 dbus_DATA = \
 	newprinternotification.conf
@@ -324,7 +321,6 @@
 	$(bin_SCRIPTS) \
 	setup.py \
 	$(EXPORT_MODULES) \
-	man/system-config-printer.xml \
 	newprinternotification.conf \
 	bootstrap \
 	mkinstalldirs \
@@ -432,51 +428,7 @@
 	  echo " rm -f '$(DESTDIR)$(pkgdatadir)/$$f'"; \
 	  rm -f "$(DESTDIR)$(pkgdatadir)/$$f"; \
 	done
-install-man1: $(man1_MANS) $(man_MANS)
-	@$(NORMAL_INSTALL)
-	test -z "$(man1dir)" || $(MKDIR_P) "$(DESTDIR)$(man1dir)"
-	@list='$(man1_MANS) $(dist_man1_MANS) $(nodist_man1_MANS)'; \
-	l2='$(man_MANS) $(dist_man_MANS) $(nodist_man_MANS)'; \
-	for i in $$l2; do \
-	  case "$$i" in \
-	    *.1*) list="$$list $$i" ;; \
-	  esac; \
-	done; \
-	for i in $$list; do \
-	  if test -f $(srcdir)/$$i; then file=$(srcdir)/$$i; \
-	  else file=$$i; fi; \
-	  ext=`echo $$i | sed -e 's/^.*\\.//'`; \
-	  case "$$ext" in \
-	    1*) ;; \
-	    *) ext='1' ;; \
-	  esac; \
-	  inst=`echo $$i | sed -e 's/\\.[0-9a-z]*$$//'`; \
-	  inst=`echo $$inst | sed -e 's/^.*\///'`; \
-	  inst=`echo $$inst | sed '$(transform)'`.$$ext; \
-	  echo " $(INSTALL_DATA) '$$file' '$(DESTDIR)$(man1dir)/$$inst'"; \
-	  $(INSTALL_DATA) "$$file" "$(DESTDIR)$(man1dir)/$$inst"; \
-	done
-uninstall-man1:
-	@$(NORMAL_UNINSTALL)
-	@list='$(man1_MANS) $(dist_man1_MANS) $(nodist_man1_MANS)'; \
-	l2='$(man_MANS) $(dist_man_MANS) $(nodist_man_MANS)'; \
-	for i in $$l2; do \
-	  case "$$i" in \
-	    *.1*) list="$$list $$i" ;; \
-	  esac; \
-	done; \
-	for i in $$list; do \
-	  ext=`echo $$i | sed -e 's/^.*\\.//'`; \
-	  case "$$ext" in \
-	    1*) ;; \
-	    *) ext='1' ;; \
-	  esac; \
-	  inst=`echo $$i | sed -e 's/\\.[0-9a-z]*$$//'`; \
-	  inst=`echo $$inst | sed -e 's/^.*\///'`; \
-	  inst=`echo $$inst | sed '$(transform)'`.$$ext; \
-	  echo " rm -f '$(DESTDIR)$(man1dir)/$$inst'"; \
-	  rm -f "$(DESTDIR)$(man1dir)/$$inst"; \
-	done
+
 install-dbusDATA: $(dbus_DATA)
 	@$(NORMAL_INSTALL)
 	test -z "$(dbusdir)" || $(MKDIR_P) "$(DESTDIR)$(dbusdir)"
@@ -1070,9 +1022,7 @@
 
 @INTLTOOL_DESKTOP_RULE@
 
-# The man pages are generated from DocBook XML.
-$(man_MANS): $(top_srcdir)/man/system-config-printer.xml
-	xmlto man -o man $<
+
 
 html:	$(EXPORT_MODULES)
 	rm -rf html







