diff -Naur snd-14.2/configure snd-14.2.new/configure
--- snd-14.2/configure	2013-11-21 22:10:26.000000000 +0100
+++ snd-14.2.new/configure	2013-11-25 09:34:46.490113440 +0100
@@ -5053,19 +5053,12 @@
 
 if test "$ac_snd_gui_choice" = none ; then
   if test x$PKG_CONFIG != xno ; then
-    if $PKG_CONFIG gtk+-3.0 --exists ; then
-      GTK_CFLAGS="`$PKG_CONFIG gtk+-3.0 --cflags`"
-      GTK_LIBS="`$PKG_CONFIG gtk+-3.0 --libs`"
-      GTK_LD_LIBS="`$PKG_CONFIG gtk+-3.0 --libs-only-L` `$PKG_CONFIG gtk+-3.0 --libs-only-l`"
-      ac_snd_gui_choice=gtk
-    else
       if $PKG_CONFIG gtk+-2.0 --exists ; then
         GTK_CFLAGS="`$PKG_CONFIG gtk+-2.0 --cflags`"
         GTK_LIBS="`$PKG_CONFIG gtk+-2.0 --libs`"
         GTK_LD_LIBS="`$PKG_CONFIG gtk+-2.0 --libs-only-L` `$PKG_CONFIG gtk+-2.0 --libs-only-l`"
         ac_snd_gui_choice=gtk
       fi
-    fi
   fi
   if test "$ac_snd_gui_choice" = gtk ; then
     GX_FILES="GTK_O_FILES"
