diff -Naur snd-14.1/snd-ladspa.c snd-14.1.new/snd-ladspa.c
--- snd-14.1/snd-ladspa.c	2013-09-27 01:16:13.000000000 +0200
+++ snd-14.1.new/snd-ladspa.c	2013-11-11 20:48:01.759229699 +0100
@@ -1191,7 +1191,7 @@
   if (descriptor->connect_port) 
     descriptor->connect_port(XEN_TO_C_Ladspa_Handle(ptr),
 			     XEN_TO_C_ULONG(port),
-			     double_to_float(samples->data, samples->length)
+			     double_to_float(mus_vct_data(samples), mus_vct_length(samples))
 			     );
   return(XEN_FALSE);
 }
