diff -Naur gnome-volume-manager-1.4.0.orig/gnome-volume-manager.schemas.in gnome-volume-manager-1.4.0/gnome-volume-manager.schemas.in
--- gnome-volume-manager-1.4.0.orig/gnome-volume-manager.schemas.in	2005-09-03 13:23:18.000000000 +0200
+++ gnome-volume-manager-1.4.0/gnome-volume-manager.schemas.in	2005-09-03 13:24:52.000000000 +0200
@@ -132,7 +132,7 @@
         <key>/schemas/desktop/gnome/volume_manager/autoplay_dvd</key>
 	<applyto>/desktop/gnome/volume_manager/autoplay_dvd</applyto>
 	<type>bool</type>
-	<default>true</default>
+	<default>false</default>
 	<locale name="C">
 	  <short>Play inserted video DVDs</short>
 	  <long>Whether, when a video DVD is inserted,
@@ -191,7 +191,7 @@
         <key>/schemas/desktop/gnome/volume_manager/autophoto_command</key>
 	<applyto>/desktop/gnome/volume_manager/autophoto_command</applyto>
 	<type>string</type>
-	<default></default>
+	<default>gnome-volume-manager-gthumb %h</default>
 	<locale name="C">
 	  <short>Photo management command</short>
 	  <long>Command to run when a digital camera is connected or media 
