diff -ruN gnome-volume-manager-old/src/manager.c gnome-volume-manager/src/manager.c
--- gnome-volume-manager-old/src/manager.c	2004-11-18 15:04:36.348949000 +0100
+++ gnome-volume-manager/src/manager.c	2004-11-18 15:04:36.562962730 +0100
@@ -969,9 +969,8 @@
 	char **volumes;
 	char *udi;
 	char *device_file;
-
-	if (!config.automount_media)
-		return;
+  char *drive_udi = NULL;
+  gboolean do_mount;
 
 	volumes = hal_find_device_by_capability (ctx, "volume", &num_volumes);
 	for (i = 0; i < num_volumes; i++) {
@@ -995,6 +994,20 @@
 		if (gvm_is_manual_mount(ctx, udi)) 
 			continue;
 
+    drive_udi = hal_device_get_property_string(ctx, udi, "info.parent");
+
+		if (hal_device_property_exists(ctx, drive_udi, "storage.hotpluggable") &&
+	      hal_device_get_property_bool(ctx, drive_udi, "storage.hotpluggable")) {
+			do_mount = config.automount_drives;
+		} else if (
+        hal_device_property_exists(ctx, drive_udi, "storage.removable") &&
+			  hal_device_get_property_bool(ctx, drive_udi, "storage.removable")) {
+			do_mount = config.automount_media;
+		} 
+    hal_free_string(drive_udi);
+    
+    if (!do_mount) continue;
+
 		device_file = hal_device_get_property_string (ctx, udi,
 							      "block.device");
 
