diff -ruN gnome-volume-manager-old/src/manager.c gnome-volume-manager/src/manager.c
--- gnome-volume-manager-old/src/manager.c	2004-11-18 15:05:44.994094000 +0100
+++ gnome-volume-manager/src/manager.c	2004-11-18 15:05:45.414120123 +0100
@@ -328,10 +328,6 @@
 
 	dbg ("Photos detected: %s\n", dcim_path);
 
-	/* add the "content.photos" capability to this device */
-	if (!hal_device_add_capability (hal_ctx, udi, "content.photos"))
-		warn ("failed to set content.photos on %s\n", device);
-
 	if (config.autophoto) {
 	  retval = TRUE;
 		askme = gtk_message_dialog_new (NULL, 0, GTK_MESSAGE_QUESTION,
