diff -ruN gnome-volume-manager-1.1.3-old/src/manager.c gnome-volume-manager-1.1.3/src/manager.c
--- gnome-volume-manager-1.1.3-old/src/manager.c	2005-02-23 15:33:51.000000000 +0100
+++ gnome-volume-manager-1.1.3/src/manager.c	2005-02-23 15:33:04.000000000 +0100
@@ -20,6 +20,7 @@
 #include <gdk/gdkx.h>
 #include <dbus/dbus.h>
 #include <dbus/dbus-glib.h>
+#include <dbus/dbus-glib-lowlevel.h>
 #include <libhal.h>
 #include <signal.h>
 
@@ -257,7 +258,6 @@
 gvm_parse_command(const char *device, const char *command, 
                   const char *path, const char *udi) {
 	gchar *new_command;
-	GError *error = NULL;
 	GString *exec = g_string_new (NULL);
 	char *p, *q;
   gchar *result, *tmp;
@@ -383,7 +383,7 @@
 	/*FIXME!! A better secondary label....*/
 	gtk_message_dialog_format_secondary_text (GTK_DIALOG (askme),
 			_("The file \"%s\" on the inserted media is an auto-run"
-			  "file."));
+			  "file."), path);
 
 	gtk_dialog_add_buttons (GTK_DIALOG (askme),
 				GTK_STOCK_CANCEL, GTK_RESPONSE_REJECT,
@@ -1225,7 +1225,7 @@
 	char *udi;
 	char *device_file;
   char *drive_udi = NULL;
-  gboolean do_mount;
+  gboolean do_mount = FALSE;
 
 	volumes = hal_find_device_by_capability (ctx, "volume", &num_volumes);
 	for (i = 0; i < num_volumes; i++) {
