diff -Naur gnome-volume-manager-1.4.0.orig/configure.in gnome-volume-manager-1.4.0/configure.in
--- gnome-volume-manager-1.4.0.orig/configure.in	2005-09-03 13:34:25.000000000 +0200
+++ gnome-volume-manager-1.4.0/configure.in	2005-09-03 13:34:51.000000000 +0200
@@ -8,14 +8,28 @@
 AC_PROG_CC
 
 AC_PATH_PROG(mount_path, mount, mount)
+AC_PATH_PROG(pmount_hal_path, pmount-hal, NOTFOUND)
 AC_PATH_PROG(umount_path, umount, umount)
+AC_PATH_PROG(pumount_path, pumount, NOTFOUND)
+
+if test ! x$pmount_hal_path = xNOTFOUND; then
+        mount_path="${pmount_hal_path} %h -e"
+else
+        mount_path="${mount_path} %d"
+fi
+
+if test ! x$pumount_path = xNOTFOUND; then
+        umount_path="${pumount_path} %d"
+else
+        umount_path="${umount_path} %d"
+fi
 
 AC_ARG_WITH(mount-command, AC_HELP_STRING(--with-mount-command], [Command to invoke in order to mount a device]), 
-            mount_command="$withval", mount_command="${mount_path} %d")
+            mount_command="$withval", mount_command="${mount_path}")
 AC_DEFINE_UNQUOTED([MOUNT_COMMAND], ["${mount_command}"], [mount command])
 
 AC_ARG_WITH(unmount-command, AC_HELP_STRING(--with-unmount-command], [Command to invoke in order to unmount a device]), 
-            unmount_command="$withval", unmount_command="${umount_path} %d")
+            unmount_command="$withval", unmount_command="${umount_path}")
 AC_DEFINE_UNQUOTED([UNMOUNT_COMMAND], ["${unmount_command}"], [unmount command])
 
 AC_PATH_PROG(nautilus_path, nautilus, nautilus)
