diff -urN gnome-volume-manager-2.17.0.orig/gnome-volume-properties.desktop.in gnome-volume-manager-2.17.0/gnome-volume-properties.desktop.in
--- gnome-volume-manager-2.17.0.orig/gnome-volume-properties.desktop.in	2007-04-15 09:43:33.000000000 +0100
+++ gnome-volume-manager-2.17.0/gnome-volume-properties.desktop.in	2007-04-15 09:45:00.000000000 +0100
@@ -6,7 +6,7 @@
 Exec=gnome-volume-properties
 Terminal=false
 Type=Application
-Categories=Application;Settings;GNOME;
+Categories=Settings;HardwareSettings;GNOME;GTK;
 OnlyShowIn=GNOME;
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-volume-manager
