diff -Naur libpng-1.2.12.orig/pngconf.h libpng-1.2.12/pngconf.h
--- libpng-1.2.12.orig/pngconf.h	2006-06-27 22:22:34.000000000 +0200
+++ libpng-1.2.12/pngconf.h	2006-09-05 02:33:48.000000000 +0200
@@ -313,9 +313,9 @@
 #    ifdef _SETJMP_H
      /* If you encounter a compiler error here, see the explanation
       * near the end of INSTALL.
+      * __png.h__ already includes setjmp.h;
+      * __dont__ include it again.;
       */
-         __png.h__ already includes setjmp.h;
-         __dont__ include it again.;
 #    endif
 #  endif /* __linux__ */
 
