--- tdesktop-5.3.1-full/Telegram/lib_webrtc/webrtc/platform/linux/webrtc_environment_linux.cpp	2024-08-01 10:18:49.000000000 +0200
+++ tdesktop-5.3.1-full.new/Telegram/lib_webrtc/webrtc/platform/linux/webrtc_environment_linux.cpp	2024-08-02 13:23:55.559006744 +0200
@@ -9,6 +9,7 @@
 #include "base/debug_log.h"
 
 #ifdef WEBRTC_USE_PIPEWIRE
+#include <modules/desktop_capture/linux/wayland/base_capturer_pipewire.h>
 #include <modules/portal/pipewire_utils.h>
 #endif // WEBRTC_USE_PIPEWIRE
 
