diff -Naur tdesktop-3.1.1-full/Telegram/ThirdParty/tgcalls/tgcalls/group/AudioStreamingPart.cpp tdesktop-3.1.1-full.new/Telegram/ThirdParty/tgcalls/tgcalls/group/AudioStreamingPart.cpp
--- tdesktop-3.1.1-full/Telegram/ThirdParty/tgcalls/tgcalls/group/AudioStreamingPart.cpp	2021-09-07 20:21:48.000000000 +0200
+++ tdesktop-3.1.1-full.new/Telegram/ThirdParty/tgcalls/tgcalls/group/AudioStreamingPart.cpp	2022-05-29 18:03:55.415318736 +0200
@@ -170,7 +170,7 @@
 
         _frame = av_frame_alloc();
 
-        AVInputFormat *inputFormat = av_find_input_format("ogg");
+        const AVInputFormat *inputFormat = av_find_input_format("ogg");
         if (!inputFormat) {
             _didReadToEnd = true;
             return;
@@ -209,7 +209,7 @@
             audioCodecParameters = inCodecpar;
             audioStream = inStream;
 
-            _durationInMilliseconds = (int)((inStream->duration + inStream->first_dts) * 1000 / 48000);
+	    _durationInMilliseconds = (int)(inStream->duration * av_q2d(inStream->time_base) * 1000);
 
             if (inStream->metadata) {
                 AVDictionaryEntry *entry = av_dict_get(inStream->metadata, "TG_META", nullptr, 0);
@@ -255,7 +255,7 @@
         }
 
         if (audioCodecParameters && audioStream) {
-            AVCodec *codec = avcodec_find_decoder(audioCodecParameters->codec_id);
+            const AVCodec *codec = avcodec_find_decoder(audioCodecParameters->codec_id);
             if (codec) {
                 _codecContext = avcodec_alloc_context3(codec);
                 ret = avcodec_parameters_to_context(_codecContext, audioCodecParameters);
diff -Naur tdesktop-3.1.1-full/Telegram/ThirdParty/tgcalls/tgcalls/group/VideoStreamingPart.cpp tdesktop-3.1.1-full.new/Telegram/ThirdParty/tgcalls/tgcalls/group/VideoStreamingPart.cpp
--- tdesktop-3.1.1-full/Telegram/ThirdParty/tgcalls/tgcalls/group/VideoStreamingPart.cpp	2021-09-07 20:21:48.000000000 +0200
+++ tdesktop-3.1.1-full.new/Telegram/ThirdParty/tgcalls/tgcalls/group/VideoStreamingPart.cpp	2022-05-29 18:00:57.531074889 +0200
@@ -337,7 +337,7 @@
 
         int ret = 0;
 
-        AVInputFormat *inputFormat = av_find_input_format(container.c_str());
+        const AVInputFormat *inputFormat = av_find_input_format(container.c_str());
         if (!inputFormat) {
             _didReadToEnd = true;
             return;
@@ -380,7 +380,7 @@
         }
 
         if (videoCodecParameters && videoStream) {
-            AVCodec *codec = avcodec_find_decoder(videoCodecParameters->codec_id);
+            const AVCodec *codec = avcodec_find_decoder(videoCodecParameters->codec_id);
             if (codec) {
                 _codecContext = avcodec_alloc_context3(codec);
                 ret = avcodec_parameters_to_context(_codecContext, videoCodecParameters);
diff -Naur tdesktop-3.1.1-full/Telegram/SourceFiles/media/audio/media_audio_capture.cpp tdesktop-3.1.1-full.new/Telegram/SourceFiles/media/audio/media_audio_capture.cpp
--- tdesktop-3.1.1-full/Telegram/SourceFiles/media/audio/media_audio_capture.cpp	2021-09-24 06:47:02.000000000 +0200
+++ tdesktop-3.1.1-full.new/Telegram/SourceFiles/media/audio/media_audio_capture.cpp	2022-05-29 18:42:39.745319857 +0200
@@ -147,7 +147,7 @@
 	AVIOContext *ioContext = nullptr;
 	AVFormatContext *fmtContext = nullptr;
 	AVStream *stream = nullptr;
-	AVCodec *codec = nullptr;
+	const AVCodec *codec = nullptr;
 	AVCodecContext *codecContext = nullptr;
 	bool opened = false;
 	bool processing = false;
diff -Naur tdesktop-3.1.1-full/Telegram/SourceFiles/media/audio/media_audio_ffmpeg_loader.h tdesktop-3.1.1-full.new/Telegram/SourceFiles/media/audio/media_audio_ffmpeg_loader.h
--- tdesktop-3.1.1-full/Telegram/SourceFiles/media/audio/media_audio_ffmpeg_loader.h	2021-09-24 06:47:02.000000000 +0200
+++ tdesktop-3.1.1-full.new/Telegram/SourceFiles/media/audio/media_audio_ffmpeg_loader.h	2022-05-29 18:45:00.354517058 +0200
@@ -60,7 +60,7 @@
 	uchar *ioBuffer = nullptr;
 	AVIOContext *ioContext = nullptr;
 	AVFormatContext *fmtContext = nullptr;
-	AVCodec *codec = nullptr;
+	const AVCodec *codec = nullptr;
 	int32 streamId = 0;
 
 	bool _opened = false;
