--- tdesktop-5.13.1-full/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp	2025-02-25 07:14:47.000000000 +0100
+++ tdesktop-5.13.1-full.new/Telegram/lib_base/base/platform/linux/base_linux_xdp_utilities.cpp	2025-04-07 17:31:10.746727461 +0200
@@ -16,7 +16,11 @@
 #if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
 #include <qpa/qplatformintegration.h>
 #include <private/qguiapplication_p.h>
+#if QT_VERSION >= QT_VERSION_CHECK(6, 9, 0)
+#include <private/qdesktopunixservices_p.h>
+#else
 #include <private/qgenericunixservices_p.h>
+#endif // Qt >= 6.9.0
 #endif // Qt >= 6.5.0
 
 #include <sstream>
@@ -39,7 +43,11 @@
 	}
 
 #if QT_VERSION >= QT_VERSION_CHECK(6, 5, 0)
+#if QT_VERSION >= QT_VERSION_CHECK(6, 9, 0)
+	if (const auto services = dynamic_cast<QDesktopUnixServices*>(
+#else
 	if (const auto services = dynamic_cast<QGenericUnixServices*>(
+#endif // Qt >= 6.9.0
 			QGuiApplicationPrivate::platformIntegration()->services())) {
 		return services->portalWindowIdentifier(window).toStdString();
 	}
