diff -Naur tdesktop-2.4.4-full/Telegram/CMakeLists.txt tdesktop-2.4.4-full.new/Telegram/CMakeLists.txt
--- tdesktop-2.4.4-full/Telegram/CMakeLists.txt	2020-10-23 18:40:48.000000000 +0200
+++ tdesktop-2.4.4-full.new/Telegram/CMakeLists.txt	2020-10-24 20:50:31.158771110 +0200
@@ -109,6 +109,7 @@
         desktop-app::external_xcb_screensaver
         desktop-app::external_xcb
         desktop-app::external_glib
+	-ldl -ljpeg -lvpx
     )
 
     if (NOT DESKTOP_APP_DISABLE_DBUS_INTEGRATION)
diff -Naur tdesktop-2.4.4-full/Telegram/lib_lottie/lottie/lottie_animation.cpp tdesktop-2.4.4-full.new/Telegram/lib_lottie/lottie/lottie_animation.cpp
--- tdesktop-2.4.4-full/Telegram/lib_lottie/lottie/lottie_animation.cpp	2020-08-31 11:03:37.000000000 +0200
+++ tdesktop-2.4.4-full.new/Telegram/lib_lottie/lottie/lottie_animation.cpp	2020-10-24 20:49:33.890303552 +0200
@@ -138,7 +138,7 @@
 	const auto string = UnpackGzip(content);
 	Assert(string.size() <= kMaxFileSize);
 
-#ifndef DESKTOP_APP_USE_PACKAGED_RLOTTIE
+#if 0 //ndef DESKTOP_APP_USE_PACKAGED_RLOTTIE
 	auto result = rlottie::Animation::loadFromData(
 		string,
 		std::string(),
