--- tdesktop-2.3.2-full/cmake/external/webrtc/CMakeLists.txt	2020-08-20 15:59:10.000000000 +0200
+++ tdesktop-2.3.2-full.new/cmake/external/webrtc/CMakeLists.txt	2020-08-31 15:13:55.918163369 +0200
@@ -13,7 +13,7 @@
     set(webrtc_loc ${libs_loc}/tg_owt/src)
 endif()
 
-set(webrtc_build_loc ${libs_loc}/tg_owt/out/$<CONFIG>)
+set(webrtc_build_loc /usr/lib)
 
 target_compile_definitions(external_webrtc
 INTERFACE
