--- tdesktop-2.4.0-full/cmake/external/webrtc/CMakeLists.txt	2020-09-29 19:19:33.000000000 +0200
+++ tdesktop-2.4.0-full.new/cmake/external/webrtc/CMakeLists.txt	2020-10-01 11:29:54.160526606 +0200
@@ -7,12 +7,8 @@
 add_library(external_webrtc INTERFACE IMPORTED GLOBAL)
 add_library(desktop-app::external_webrtc ALIAS external_webrtc)
 
-if (DESKTOP_APP_USE_PACKAGED)
-    find_package(tg_owt REQUIRED)
-    target_link_libraries(external_webrtc INTERFACE tg_owt::tg_owt)
-else()
-    set(webrtc_loc ${libs_loc}/tg_owt/src)
-    set(webrtc_build_loc ${libs_loc}/tg_owt/out/$<CONFIG>)
+    set(webrtc_loc /usr/include/tg_owt)
+    set(webrtc_build_loc /usr/lib)
 
     target_compile_definitions(external_webrtc
     INTERFACE
@@ -99,4 +95,3 @@
         ${webrtc_libs_list}
         desktop-app::external_openssl
     )
-endif()
