From: http://dev.rubyonrails.org/changeset/6893?format=diff&new=6893
--- activesupport/lib/active_support/json/encoders/core.rb (revision 6443)
+++ activesupport/lib/active_support/json/encoders/core.rb (revision 6893)
@@ -24,11 +24,13 @@
         "\r" =>    '\r',
         "\t" =>    '\t',
         '"' =>     '\"',
-        '\\' =>    '\\\\'
+        '\\' =>    '\\\\',
+	">" =>     '\076',
+	'<' =>     '\074'
       }
       
       define_encoder String do |string|
-        '"' + string.gsub(/[\010\f\n\r\t"\\]/) { |s|
+        '"' + string.gsub(/[\010\f\n\r\t"\\><]/) { |s|
           ESCAPED_CHARS[s]
         }.gsub(/([\xC0-\xDF][\x80-\xBF]|
                  [\xE0-\xEF][\x80-\xBF]{2}|
