diff -Naur bochs-2.1.1.orig/fpu/reg_ld_str.c bochs-2.1.1/fpu/reg_ld_str.c
--- bochs-2.1.1.orig/fpu/reg_ld_str.c	2004-02-11 23:28:52.000000000 +0100
+++ bochs-2.1.1/fpu/reg_ld_str.c	2005-05-16 01:18:20.000000000 +0200
@@ -315,7 +315,7 @@
 
 
 /* Get a short from user memory */
-int  BX_CPP_AttrRegparmN(1)
+int  BX_CPP_AttrRegparmN(2)
 FPU_load_int16(bx_address _s, FPU_REG *loaded_data)
 {
   s16 s, negative;
