diff -Naur maildir-0.24.4/src/maildir_gtk.c maildir-0.24.4-p/src/maildir_gtk.c
--- maildir-0.24.4/src/maildir_gtk.c	2007-01-09 07:59:57.000000000 +0100
+++ maildir-0.24.4-p/src/maildir_gtk.c	2008-06-20 23:45:59.000000000 +0200
@@ -337,7 +337,7 @@
 	if (!from_folder || from_folder->folder->klass != maildir_get_class())
 		return;
 
-	to_folder = foldersel_folder_sel(from_folder->folder, FOLDER_SEL_MOVE, NULL);
+	to_folder = foldersel_folder_sel(from_folder->folder, FOLDER_SEL_MOVE, NULL, FALSE);
 	if (!to_folder)
 		return;
 
