diff -Naur valgrind-3.15.0/coregrind/m_redir.c valgrind-3.15.0-p/coregrind/m_redir.c
--- valgrind-3.15.0/coregrind/m_redir.c	2019-04-13 12:56:21.000000000 +0200
+++ valgrind-3.15.0-p/coregrind/m_redir.c	2019-10-14 03:30:13.886097279 +0200
@@ -1342,9 +1342,11 @@
       add_hardwired_spec(
          "ld-linux.so.2", "index",
          (Addr)&VG_(x86_linux_REDIR_FOR_index), mandatory);
+#     ifdef FUCK_THIS_HACKS
       add_hardwired_spec(
          "ld-linux.so.2", "strlen",
          (Addr)&VG_(x86_linux_REDIR_FOR_strlen), mandatory);
+#    endif
    }
 
 #  elif defined(VGP_amd64_linux)
@@ -1369,6 +1371,7 @@
       add_hardwired_spec(
          "ld-linux-x86-64.so.2", "index",
          (Addr)&VG_(amd64_linux_REDIR_FOR_index), NULL);
+# 	  ifdef FUCK_THIS_HACKS
       add_hardwired_spec(
          "ld-linux-x86-64.so.2", "strlen",
          (Addr)&VG_(amd64_linux_REDIR_FOR_strlen),
@@ -1379,7 +1382,8 @@
             continue without it */
          complain_about_stripped_glibc_ldso
 #        endif
-      );   
+      );
+#      endif
    }
 
 #  elif defined(VGP_ppc32_linux)
