--- ebtables-v2.0.10-4/Makefile	2011-12-15 22:02:47.000000000 +0200
+++ ebtables-v2.0.10-4.new/Makefile	2016-08-12 02:47:16.664742460 +0300
@@ -90,7 +90,7 @@
 	$(CC) -shared $(LDFLAGS) -Wl,-soname,libebtc.so -o libebtc.so -lc $(OBJECTS2)
 
 ebtables: $(OBJECTS) ebtables-standalone.o libebtc.so
-	$(CC) $(CFLAGS) $(CFLAGS_SH_LIB) $(LDFLAGS) -o $@ ebtables-standalone.o -I$(KERNEL_INCLUDES) -L. -Lextensions -lebtc $(EXT_LIBSI) \
+	$(CC) $(CFLAGS) $(CFLAGS_SH_LIB) -o $@ ebtables-standalone.o -I$(KERNEL_INCLUDES) -L. -Lextensions -Wl,--no-as-needed $(EXT_LIBSI) -lebtc \
 	-Wl,-rpath,$(LIBDIR)
 
 ebtablesu: ebtablesu.c
