--- opera-8.54-20060330.6-shared-qt.i386-en/install.sh.bak	2006-04-06 10:37:25.000000000 +0200
+++ opera-8.54-20060330.6-shared-qt.i386-en/install.sh	2006-04-06 10:39:26.000000000 +0200
@@ -129,6 +129,7 @@
     man_dir="${OPERADESTDIR}${man_dir}"
     share_dir="${OPERADESTDIR}${share_dir}"
     plugin_dir="${OPERADESTDIR}${plugin_dir}"
+    desktop_dir="${prefix}/share/applications"
 }
 
 usage()
@@ -1856,10 +1857,7 @@
 	    done
 	fi
 	
-	for SHORTCUT_DIR in ${XDG_DATA_HOME}/applications /usr/local/share/applications /usr/share/applications; do
-	    test -d ${SHORTCUT_DIR} && break;
-	done
-	
+	SHORTCUT_DIR=${desktop_dir}	
 	if   test ! -d ${SHORTCUT_DIR}; then echo "Could not find shortcut installation directory, desktop entry not installed." >&2; return
 	elif test ! -w ${SHORTCUT_DIR}; then echo "Directory \"${SHORTCUT_DIR}\" not writable by user \"${USER}\", desktop entry not installed." >&2; return
 	fi
