--- opera-8.52-20060201.1-static-qt.i386-en/install.sh	2006-02-01 15:24:19.000000000 +0100
+++ install.sh	2006-03-05 00:07:00.000000000 +0100
@@ -118,6 +118,7 @@
 	doc_dir="${prefix}/share/doc/opera"
 	share_dir="${prefix}/share/opera"
 	plugin_dir="${prefix}/lib/opera/plugins"
+	desktop_dir="${prefix}/share/applications"
     fi
 }
 
@@ -1795,8 +1797,7 @@
 	    test -d ${ICON_DIR}/22x22/apps && test -w ${ICON_DIR}/22x22/apps && cp $cpv $share_dir/images/opera_22x22.png $DIR_HI/22x22/apps/opera.png
 	fi
 
-	for SHORTCUT_DIR in ${XDG_DATA_HOME}/applications /usr/local/share/applications /usr/share/applications; do test -d ${SHORTCUT_DIR} && break; done
-
+	SHORTCUT_DIR=${desktop_dir}
 	if   test ! -d ${SHORTCUT_DIR}; then echo "Could not find shortcut installation directory, desktop entry not installed." >&2; return
 	elif test ! -w ${SHORTCUT_DIR}; then echo "Directory \"${SHORTCUT_DIR}\" not writable by user \"${USER}\", desktop entry not installed." >&2; return
 	fi
