*** xorg-server-1.0.1/os/access.c	Thu Dec 22 07:13:56 2005
--- xorg-server-1.0.1/os/access.c	Thu Dec 22 07:13:33 2005
***************
*** 978,985 ****
--- 978,987 ----
  #ifdef DNETCONN
  	if (ifr->ifa_addr.sa_family == AF_DECnet) 
  	    continue;
  #endif /* DNETCONN */
+ 	if(!ifr->ifa_addr)
+             continue;
  	len = sizeof(*(ifr->ifa_addr));
  	family = ConvertAddr(ifr->ifa_addr, &len, (pointer *)&addr);
  	if (family == -1 || family == FamilyLocal) 
  	    continue;
