--- xgl-xorg/GL/symlink-mesa.sh.orig	2006-02-18 22:22:13.653757750 +0100
+++ xgl-xorg/GL/symlink-mesa.sh	2006-02-18 22:21:56.088660000 +0100
@@ -433,6 +433,8 @@
     action slang_compile_variable.h
     action slang_execute.c
     action slang_execute.h
+    action slang_library_noise.c
+    action slang_library_noise.h
     action slang_mesa.h
     action slang_preprocess.c
     action slang_preprocess.h
--- xgl-xorg/GL/mesa/shader/slang/Makefile.am.orig	2006-02-18 22:23:09.421243000 +0100
+++ xgl-xorg/GL/mesa/shader/slang/Makefile.am	2006-02-18 22:22:55.644382000 +0100
@@ -27,6 +27,7 @@
                       slang_compile_struct.c \
                       slang_compile_variable.c \
 		      slang_execute.c \
+		      slang_library_noise.c \
 		      slang_preprocess.c \
 		      slang_storage.c \
 		      slang_utility.c
