diff -Naur gnome-vfs-2.12.0.orig/modules/smb-method.c gnome-vfs-2.12.0/modules/smb-method.c
--- gnome-vfs-2.12.0.orig/modules/smb-method.c	2005-09-08 00:07:31.000000000 +0200
+++ gnome-vfs-2.12.0/modules/smb-method.c	2005-09-08 00:07:41.000000000 +0200
@@ -1486,7 +1486,7 @@
 
 		/* Important: perform_authentication leaves and re-enters the lock! */
 		while (perform_authentication (&actx) > 0) {
-			r = smb_context->close (smb_context, handle->file);
+			r = smb_context->close_fn (smb_context, handle->file);
 			actx.res = (r >= 0) ? GNOME_VFS_OK : gnome_vfs_result_from_errno ();
 		}
 
