diff -Naur gnome-vfs2-2.8.3-old/libgnomevfs/gnome-vfs-volume-ops.c gnome-vfs2-2.8.3/libgnomevfs/gnome-vfs-volume-ops.c
--- gnome-vfs2-2.8.3-old/libgnomevfs/gnome-vfs-volume-ops.c	2004-11-18 11:39:58.000000000 +0100
+++ gnome-vfs2-2.8.3/libgnomevfs/gnome-vfs-volume-ops.c	2004-11-18 11:40:38.000000000 +0100
@@ -57,6 +57,7 @@
 };
 
 static const char *umount_known_locations [] = {
+	"/usr/bin/pumount", "/bin/pumount",
 	"/sbin/umount", "/bin/umount",
 	"/usr/sbin/umount", "/usr/bin/umount",
 	NULL
@@ -76,7 +77,7 @@
 	int i;
 
 	for (i = 0; known_locations [i]; i++){
-		if (g_file_test (known_locations [i], G_FILE_TEST_EXISTS))
+		if (g_file_test (known_locations [i], G_FILE_TEST_IS_EXECUTABLE))
 			return known_locations [i];
 	}
 	return NULL;
@@ -339,7 +340,7 @@
 	       name = mount_point;
        }
 #else
-       name = mount_point;
+       name = device_path;
 #endif
        
        if (should_mount) {
