diff -Naur gnome-vfs-2.8.3.orig/libgnomevfs/gnome-vfs-filesystem-type.c gnome-vfs-2.8.3/libgnomevfs/gnome-vfs-filesystem-type.c
--- gnome-vfs-2.8.3.orig/libgnomevfs/gnome-vfs-filesystem-type.c	2004-07-24 16:39:43.000000000 +0200
+++ gnome-vfs-2.8.3/libgnomevfs/gnome-vfs-filesystem-type.c	2005-01-01 15:47:43.989711832 +0100
@@ -61,6 +61,7 @@
 	{ "nwfs"     , N_("Netware Volume"), 0 },
 	{ "proc"     , N_("System Volume"), 0 },
 	{ "procfs"   , N_("System Volume"), 0 },
+	{ "reiser4"  , N_("Reiser4 Linux Volume"), 1 },
 	{ "reiserfs" , N_("ReiserFS Linux Volume"), 1 },
 	{ "smbfs"    , N_("Windows Shared Volume"), 1 },
 	{ "supermount",N_("SuperMount Volume"), 0 },
