Index: pdftk-1.41/java_libs/com/lowagie/text/Makefile
===================================================================
--- pdftk-1.41.orig/java_libs/com/lowagie/text/Makefile	2007-11-08 14:03:26.800010177 +0100
+++ pdftk-1.41/java_libs/com/lowagie/text/Makefile	2007-11-08 14:03:29.072639728 +0100
@@ -25,14 +25,15 @@
 # the "$*" automatic variable, here
 #
 %.h : %.class
-	$(GCJH) --classpath="." $*;
-	$(RM) $<
+	$(GCJH) --classpath="/usr/share/java/libgcj-4.3.jar:$(PWD):." $*
 
 ##
 # targets
 
 all : $(library) $(headers)
 
+$(classes): $(sources)
+
 $(library) : $(objects)
 	$(AR) $(ARFLAGS) $(library) $(objects);
 
Index: pdftk-1.41/java_libs/com/lowagie/text/xml/xmp/Makefile
===================================================================
--- pdftk-1.41.orig/java_libs/com/lowagie/text/xml/xmp/Makefile	2007-11-08 14:03:26.820015719 +0100
+++ pdftk-1.41/java_libs/com/lowagie/text/xml/xmp/Makefile	2007-11-08 14:03:29.09264.370 +0100
@@ -25,8 +25,7 @@
 # the "$*" automatic variable, here
 #
 %.h : %.class
-	$(GCJH) --classpath="." $*;
-	$(RM) $<
+	$(GCJH) --classpath="/usr/share/java/libgcj-4.3.jar:$(PWD):." $*
 
 ##
 # targets
Index: pdftk-1.41/java_libs/com/lowagie/text/pdf/Makefile
===================================================================
--- pdftk-1.41.orig/java_libs/com/lowagie/text/pdf/Makefile	2007-11-08 14:03:26.832019044 +0100
+++ pdftk-1.41/java_libs/com/lowagie/text/pdf/Makefile	2007-11-08 14:03:29.09264.370 +0100
@@ -25,8 +25,7 @@
 # the "$*" automatic variable, here
 #
 %.h : %.class
-	$(GCJH) --classpath="." $*;
-	$(RM) $<
+	$(GCJH) --classpath="/usr/share/java/libgcj-4.3.jar:$(PWD):." $*
 
 ##
 # targets
@@ -37,4 +36,4 @@
 	$(AR) $(ARFLAGS) $(library) $(objects);
 
 clean :
-	$(RM) $(RMFLAGS) *.o *.h *.class *.a
+	$(RM) $(RMFLAGS) *.o *.h *.class *.a hyphenation/*.class
Index: pdftk-1.41/java_libs/com/lowagie/text/pdf/fonts/Makefile
===================================================================
--- pdftk-1.41.orig/java_libs/com/lowagie/text/pdf/fonts/Makefile	2007-11-08 14:03:26.852024586 +0100
+++ pdftk-1.41/java_libs/com/lowagie/text/pdf/fonts/Makefile	2007-11-08 14:03:29.09264.370 +0100
@@ -34,8 +34,7 @@
 # the "$*" automatic variable, here
 #
 %.h : %.class
-	$(GCJH) --classpath="." $*;
-	$(RM) $<
+	$(GCJH) --classpath="/usr/share/java/libgcj-4.3.jar:$(PWD):." $*
 
 ##
 # targets
Index: pdftk-1.41/java_libs/com/lowagie/text/pdf/codec/postscript/Makefile
===================================================================
--- pdftk-1.41.orig/java_libs/com/lowagie/text/pdf/codec/postscript/Makefile	2007-11-08 14:03:26.868029019 +0100
+++ pdftk-1.41/java_libs/com/lowagie/text/pdf/codec/postscript/Makefile	2007-11-08 14:03:29.09264.370 +0100
@@ -25,8 +25,7 @@
 # the "$*" automatic variable, here
 #
 %.h : %.class
-	$(GCJH) --classpath="." $*;
-	$(RM) $<
+	$(GCJH) --classpath="/usr/share/java/libgcj-4.3.jar:$(PWD):." $*
 
 ##
 # targets
Index: pdftk-1.41/java_libs/com/lowagie/text/pdf/codec/wmf/Makefile
===================================================================
--- pdftk-1.41.orig/java_libs/com/lowagie/text/pdf/codec/wmf/Makefile	2007-11-08 14:03:26.908040103 +0100
+++ pdftk-1.41/java_libs/com/lowagie/text/pdf/codec/wmf/Makefile	2007-11-08 14:03:29.09264.370 +0100
@@ -25,8 +25,7 @@
 # the "$*" automatic variable, here
 #
 %.h : %.class
-	$(GCJH) --classpath="." $*;
-	$(RM) $<
+	$(GCJH) --classpath="/usr/share/java/libgcj-4.3.jar:$(PWD):." $*
 
 ##
 # targets
Index: pdftk-1.41/java_libs/com/lowagie/text/pdf/codec/Makefile
===================================================================
--- pdftk-1.41.orig/java_libs/com/lowagie/text/pdf/codec/Makefile	2007-11-08 14:03:26.928045645 +0100
+++ pdftk-1.41/java_libs/com/lowagie/text/pdf/codec/Makefile	2007-11-08 14:03:29.100647487 +0100
@@ -25,8 +25,7 @@
 # the "$*" automatic variable, here
 #
 %.h : %.class
-	$(GCJH) --classpath="." $*;
-	$(RM) $<
+	$(GCJH) --classpath="/usr/share/java/libgcj-4.3.jar:$(PWD):." $*
 
 ##
 # targets
Index: pdftk-1.41/java_libs/com/lowagie/text/markup/Makefile
===================================================================
--- pdftk-1.41.orig/java_libs/com/lowagie/text/markup/Makefile	2007-11-08 14:03:26.948051187 +0100
+++ pdftk-1.41/java_libs/com/lowagie/text/markup/Makefile	2007-11-08 14:03:29.100647487 +0100
@@ -25,8 +25,7 @@
 # the "$*" automatic variable, here
 #
 %.h : %.class
-	$(GCJH) --classpath="." $*;
-	$(RM) $<
+	$(GCJH) --classpath="/usr/share/java/libgcj-4.3.jar:$(PWD):." $*
 
 ##
 # targets
Index: pdftk-1.41/java_libs/com/lowagie/bc/asn1/Makefile
===================================================================
--- pdftk-1.41.orig/java_libs/com/lowagie/bc/asn1/Makefile	2007-11-08 14:03:26.972057837 +0100
+++ pdftk-1.41/java_libs/com/lowagie/bc/asn1/Makefile	2007-11-08 14:03:29.100647487 +0100
@@ -25,8 +25,7 @@
 # the "$*" automatic variable, here
 #
 %.h : %.class
-	$(GCJH) --classpath="." $*;
-	$(RM) $<
+	$(GCJH) --classpath="/usr/share/java/libgcj-4.3.jar:$(PWD):." $*
 
 ##
 # targets
Index: pdftk-1.41/java_libs/Makefile
===================================================================
--- pdftk-1.41.orig/java_libs/Makefile	2007-11-08 14:03:26.992063379 +0100
+++ pdftk-1.41/java_libs/Makefile	2007-11-08 14:03:29.100647487 +0100
@@ -11,7 +11,7 @@
 java_libs_root= $(CURDIR)
 
 # append gcj flags
-export GCJFLAGS+= --encoding=UTF-8 --classpath="$(java_libs_root)"
+export GCJFLAGS+= -w --encoding=UTF-8 --classpath="$(java_libs_root)"
 
 all : libgcj_local itext
 
Index: pdftk-1.41/pdftk/Makefile.Base
===================================================================
--- pdftk-1.41.orig/pdftk/Makefile.Base	2007-11-08 14:03:42.008223078 +0100
+++ pdftk-1.41/pdftk/Makefile.Base	2007-11-08 14:03:50.490572824 +0100
@@ -73,4 +73,4 @@
 
 clean:
 	$(MAKE) -C $(java_libs_root) clean
-	rm -vf *.o
+	rm -vf *.o pdftk
