Index: pdftk/pdftk/Makefile.Base
===================================================================
--- pdftk.orig/pdftk/Makefile.Base	2007-11-14 13:26:12.343555702 +0100
+++ pdftk/pdftk/Makefile.Base	2007-11-14 13:26:12.363561247 +0100
@@ -31,18 +31,6 @@
 afms= $(wildcard $(java_libs_root)/com/lowagie/text/pdf/fonts/*.afm)
 afm_objects= $(patsubst %.afm, %.o, $(afms))
 
-# older versions of libgcj might not have the MD5 algorithm,
-# so I added it here; these *_local java files were grabbed from
-# libgcj CVS on March 7, 2004; diffed September 5, 2006 w/ gcc 4.1.1
-#
-# gnu/gcj/convert/Input_UnicodeBig.java was grabbed March 26, 2004; diffed September 5, 2006 w/ gcc 4.1.1
-#
-libgcj_local_libs = \
-$(java_libs_root)/java_local/security/security.a \
-$(java_libs_root)/gnu_local/java/security/provider/provider.a \
-$(java_libs_root)/gnu_local/java/security/security.a \
-$(java_libs_root)/gnu/gcj/convert/convert.a
-
 # this must already be set according to your platform Makefile;
 # we're just appending to it, here
 #
Index: pdftk/java_libs/com/lowagie/text/pdf/PdfEncryption.java
===================================================================
--- pdftk.orig/java_libs/com/lowagie/text/pdf/PdfEncryption.java	2007-11-14 13:24:35.736779184 +0100
+++ pdftk/java_libs/com/lowagie/text/pdf/PdfEncryption.java	2007-11-14 13:26:12.363561247 +0100
@@ -50,7 +50,7 @@
 
 package com.lowagie.text.pdf;
 
-import java_local.security.MessageDigest; // ssteward
+import java.security.MessageDigest;
 import com.lowagie.text.ExceptionConverter;
 
 /**
Index: pdftk/java_libs/Makefile
===================================================================
--- pdftk.orig/java_libs/Makefile	2007-11-14 13:26:12.343555702 +0100
+++ pdftk/java_libs/Makefile	2007-11-14 13:26:56.971925474 +0100
@@ -11,15 +11,9 @@
 java_libs_root= $(CURDIR)
 
 # append gcj flags
-export GCJFLAGS+= -w --encoding=UTF-8 --classpath="$(java_libs_root)"
+export GCJFLAGS+= -w --encoding=UTF-8 --classpath="/usr/share/java/libgcj-4.3.jar:$(java_libs_root)"
 
-all : libgcj_local itext
-
-libgcj_local :
-	$(MAKE) -C "$(java_libs_root)/gnu_local/java/security";
-	$(MAKE) -C "$(java_libs_root)/gnu_local/java/security/provider";
-	$(MAKE) -C "$(java_libs_root)/gnu/gcj/convert";
-	$(MAKE) -C "$(java_libs_root)/java_local/security";
+all : itext
 
 itext :
 	$(MAKE) -C "$(java_libs_root)/com/lowagie/text";
@@ -35,13 +29,7 @@
 	$(MAKE) -C "$(java_libs_root)/com/lowagie/bc/asn1";
 	$(MAKE) -C "$(java_libs_root)/com/lowagie/text/pdf/codec/postscript";
 
-clean : libgcj_local_clean itext_clean
-
-libgcj_local_clean :
-	$(MAKE) -iC "$(java_libs_root)/gnu_local/java/security" clean;
-	$(MAKE) -iC "$(java_libs_root)/gnu_local/java/security/provider" clean;
-	$(MAKE) -iC "$(java_libs_root)/gnu/gcj/convert" clean;
-	$(MAKE) -iC "$(java_libs_root)/java_local/security" clean;
+clean : itext_clean
 
 itext_clean :
 	$(MAKE) -iC "$(java_libs_root)/com/lowagie/text" clean;
