Description: Add method stubs to build against current APIs.
Forwarded: not-needed
Author: Onkar Shinde <onshinde@ubuntu.com>, James page <james.page@ubuntu.com>

Index: excalibur-logkit/src/java/org/apache/log/output/db/DefaultDataSource.java
===================================================================
--- excalibur-logkit.orig/src/java/org/apache/log/output/db/DefaultDataSource.java	2011-11-25 10:55:37.000000000 +0000
+++ excalibur-logkit/src/java/org/apache/log/output/db/DefaultDataSource.java	2011-11-25 10:59:19.640362541 +0000
@@ -20,6 +20,8 @@
 import java.sql.Connection;
 import java.sql.DriverManager;
 import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.util.logging.Logger;
 import javax.sql.DataSource;
 
 /**
@@ -111,4 +113,20 @@
     {
         m_logWriter = logWriter;
     }
+
+    public Object unwrap(Class iface) throws java.sql.SQLException
+    {
+       throw new UnsupportedOperationException();
+    }
+
+    public boolean isWrapperFor(Class iface) throws java.sql.SQLException
+    {
+       throw new UnsupportedOperationException();
+    }
+
+    public Logger getParentLogger() throws SQLFeatureNotSupportedException
+    {
+       throw new SQLFeatureNotSupportedException();
+    }
+
 }
