--- gyp/pylib/gyp/generator/ninja.py	2018-09-28 11:26:32.881809296 +0200
+++ gyp.new/pylib/gyp/generator/ninja.py	2018-09-28 11:27:44.298806980 +0200
@@ -1229,11 +1229,8 @@
       ldflags = env_ldflags + config.get('ldflags', [])
       if is_executable and len(solibs):
         rpath = 'lib/'
         if self.toolset != 'target':
           rpath += self.toolset
-          ldflags.append(r'-Wl,-rpath=\$$ORIGIN/%s' % rpath)
-        else:
-          ldflags.append('-Wl,-rpath=%s' % self.target_rpath)
         ldflags.append('-Wl,-rpath-link=%s' % rpath)
     self.WriteVariableList(ninja_file, 'ldflags',
                            map(self.ExpandSpecial, ldflags))
