--- base64-bytestring-1.2.1.0/base64-bytestring.cabal	2001-09-09 03:46:40.000000000 +0200
+++ base64-bytestring-1.2.1.0.new/base64-bytestring.cabal	2023-12-12 09:43:34.785808994 +0100
@@ -50,7 +50,7 @@
   other-modules:    Data.ByteString.Base64.Internal
   build-depends:
       base        >=4   && <5
-    , bytestring  >=0.9 && <0.12
+    , bytestring  >=0.9 && <0.13
 
   ghc-options:      -Wall -funbox-strict-fields
   default-language: Haskell2010
