Index: kdenlive-0.5/configure.in.in
===================================================================
--- kdenlive-0.5.orig/configure.in.in
+++ kdenlive-0.5/configure.in.in
@@ -49,6 +49,8 @@ AC_SUBST(LIBMLT_CFLAGS)
 
 dnl check libraries
 
+AC_CHECK_HEADERS([ffmpeg/avformat.h libavformat/avformat.h])
+
 # LIBRAW1394
 
 #PKG_CHECK_MODULES(LIBRAW1394, libraw1394 >= 1.1.0)
Index: kdenlive-0.5/kdenlive/krender.cpp
===================================================================
--- kdenlive-0.5.orig/kdenlive/krender.cpp
+++ kdenlive-0.5/kdenlive/krender.cpp
@@ -22,8 +22,14 @@
  *                                                                         *
  ***************************************************************************/
 
+#include "config.h"
+
 extern "C" {
+#if HAVE_LIBAVFORMAT_AVFORMAT_H
+#include <libavformat/avformat.h>
+#else
 #include <ffmpeg/avformat.h>
+#endif
 }
 
 #include <iostream>
