--- kdenlive/kdenlive.cpp
+++ kdenlive/kdenlive.cpp
@@ -17,6 +17,8 @@
 
 #define _ISOC99_SOURCE
 
+#include <cstdlib>
+
 // include files for QT
 #include <qdir.h>
 #include <qprinter.h>
--- kdenlive/krender.cpp
+++ kdenlive/krender.cpp
@@ -27,6 +27,7 @@
 }
 
 #include <iostream>
+#include <cstdlib>
 
 // ffmpeg Header files
 
--- kdenlive/kruler.cpp
+++ kdenlive/kruler.cpp
@@ -24,6 +24,7 @@
 
 #include <iostream>
 #include <cmath>
+#include <cstdlib>
 
 #include <kdebug.h>
 #include <kstyle.h>
--- kdenlive/ktrackview.cpp
+++ kdenlive/ktrackview.cpp
@@ -16,7 +16,8 @@
  ***************************************************************************/
 
 #include <iostream>
-#include <math.h>
+#include <cmath>
+#include <cstdlib>
 
 #include <qpainter.h>
 #include <qcursor.h>
