diff -Naur netkit-ntalk-0.17.orig/talkd/announce.c netkit-ntalk-0.17/talkd/announce.c
--- netkit-ntalk-0.17.orig/talkd/announce.c	2005-06-02 16:10:09.000000000 +0000
+++ netkit-ntalk-0.17/talkd/announce.c	2005-06-02 16:11:03.000000000 +0000
@@ -49,6 +49,7 @@
 #include <fcntl.h>
 #include <string.h>
 #include <paths.h>
+#include <time.h>
 #include "prot_talkd.h"
 #include "proto.h"
 
diff -Naur netkit-ntalk-0.17.orig/talkd/table.c netkit-ntalk-0.17/talkd/table.c
--- netkit-ntalk-0.17.orig/talkd/table.c	2005-06-02 16:10:09.000000000 +0000
+++ netkit-ntalk-0.17/talkd/table.c	2005-06-02 16:11:12.000000000 +0000
@@ -54,6 +54,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <netinet/in.h>
+#include <time.h>
 
 #include "prot_talkd.h"
 #include "proto.h"
