diff -Naur ipx-1.0.orig/Makefile ipx-1.0/Makefile
--- ipx-1.0.orig/Makefile	1996-02-07 20:44:12.000000000 +0100
+++ ipx-1.0/Makefile	2006-10-26 23:58:14.000000000 +0200
@@ -1,4 +1,4 @@
-CFLAGS = -O2 -Wall
+CFLAGS ?= -O2 -Wall
 UTILS = ipx_configure ipx_interface ipx_internal_net ipx_route
 all: $(UTILS)
 	
@@ -6,19 +6,14 @@
 	rm -f $(UTILS) *.o rip sap ipxrcv ipxsend
 
 install: $(UTILS)
+	mkdir -p $(DESTDIR)/sbin
+	mkdir -p $(DESTDIR)/usr/man/man8
 	for i in $(UTILS); \
 	do \
-		install --strip $$i /sbin; \
-		install $$i.8 /usr/man/man8; \
+		install --strip $$i $(DESTDIR)/sbin; \
+		install $$i.8 $(DESTDIR)/usr/man/man8; \
 	done
-	install init.ipx /etc/rc.d/init.d/ipx
-	install -m 0644 config.ipx /etc/sysconfig/ipx
-	rm -f /etc/rc.d/rc2.d/S15ipx
-	ln -sf /etc/rc.d/init.d/ipx /etc/rc.d/rc2.d/S15ipx
-	rm -f /etc/rc.d/rc3.d/S15ipx
-	ln -sf /etc/rc.d/init.d/ipx /etc/rc.d/rc3.d/S15ipx
-	rm -f /etc/rc.d/rc5.d/S15ipx
-	ln -sf /etc/rc.d/init.d/ipx /etc/rc.d/rc5.d/S15ipx
-	rm -f /etc/rc.d/rc6.d/K55ipx
-	ln -sf /etc/rc.d/init.d/ipx /etc/rc.d/rc6.d/K55ipx
-
+	mkdir -p $(DESTDIR)/etc/rc.d
+	install init.ipx $(DESTDIR)/etc/rc.d/rc.ipx
+	mkdir -p $(DESTDIR)/etc/sysconfig
+	install -m 0644 config.ipx $(DESTDIR)/etc/sysconfig/ipx
