diff -Naur ipx-1.0.orig/init.ipx ipx-1.0/init.ipx
--- ipx-1.0.orig/init.ipx	1996-01-03 01:27:45.000000000 +0100
+++ ipx-1.0/init.ipx	2006-10-26 23:57:31.000000000 +0200
@@ -1,17 +1,10 @@
 #!/bin/sh
 #
-# ipx       Bring up/down IPX networking
-#
-
-# Source function library.
-. /etc/rc.d/init.d/functions
+# chkconfig: 2345 08 97
+# description: Bring up/down IPX networking
 
-. /etc/sysconfig/network
 . /etc/sysconfig/ipx
 
-# Check that networking is up.
-[ ${NETWORKING} = "no" ] && exit 0
-
 # See how we were called.
 case "$1" in
   start)
@@ -34,7 +27,7 @@
         rm -f /var/lock/subsys/ipx
         ;;
   *)
-        echo "Usage: network {start|stop}"
+        echo "Usage: $0 {start|stop}"
         exit 1
 esac
 
