--- scribus-1.6.1/CMakeLists.txt	2024-01-07 18:06:51.000000000 +0100
+++ scribus-1.6.1.new/CMakeLists.txt	2024-05-21 22:24:02.970574560 +0200
@@ -407,16 +407,16 @@
 
 #C++11 Support
 #as of 1.5.5.svn, require C++11
-if(NOT WANT_CPP14 AND NOT WANT_CPP17)
+if(NOT WANT_CPP14 AND NOT WANT_CPP20)
 	message(STATUS "Enabling C++11 compiler features")
 	set(CMAKE_CXX_STANDARD 11)
 endif()
 if (WANT_CPP14)
 	message(STATUS "Enabling C++14 compiler features")
 	set(CMAKE_CXX_STANDARD 14)
-elseif(WANT_CPP17)
+elseif(WANT_CPP20)
 		message(STATUS "Enabling C++17 compiler features")
-		set(CMAKE_CXX_STANDARD 17)
+		set(CMAKE_CXX_STANDARD 20)
 endif()
 set (CMAKE_CXX_STANDARD_REQUIRED ON)
 set (CMAKE_CXX_EXTENSIONS OFF)
