diff -Naur scribus-1.5.6.1/scribus/pdf_analyzer.cpp scribus-1.5.6.1.new/scribus/pdf_analyzer.cpp
--- scribus-1.5.6.1/scribus/pdf_analyzer.cpp	2020-11-14 23:37:18.000000000 +0100
+++ scribus-1.5.6.1.new/scribus/pdf_analyzer.cpp	2021-01-18 14:42:44.047084848 +0100
@@ -139,7 +139,9 @@
 					{
 						pBase = cs->GetOwner()->GetObject(base.GetReference());
 					}
-					pBase->SetOwner(cs->GetOwner());
+#if (PODOFO_VERSION < PODOFO_MAKE_VERSION(0, 9, 7))
+                                        pBase->SetOwner(cs->GetOwner());
+#endif					
 					return getCSType(pBase);
 				}
 				if (csTypeName == "Separation")
