diff -Naur vim71.orig/runtime/autoload/tar.vim vim71/runtime/autoload/tar.vim
--- vim71.orig/runtime/autoload/tar.vim	2006-10-11 20:03:43.000000000 +0200
+++ vim71/runtime/autoload/tar.vim	2007-09-11 23:26:22.000000000 +0200
@@ -110,7 +110,7 @@
   if tarfile =~# '\.\(gz\|tgz\)$'
 "   call Decho("exe silent r! gzip -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -".g:tar_browseoptions." - ")
    exe "silent r! gzip -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -".g:tar_browseoptions." - "
-  elseif tarfile =~# '\.bz2$'
+  elseif tarfile =~# '\.\(bz2\|fpm\|fdb\)$'
 "   call Decho("exe silent r! bzip2 -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -".g:tar_browseoptions." - ")
    exe "silent r! bzip2 -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -".g:tar_browseoptions." - "
   else
@@ -197,7 +197,7 @@
   if tarfile =~# '\.\(gz\|tgz\)$'
 "   call Decho("exe silent r! gzip -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -OPxf - '".fname."'")
    exe "silent r! gzip -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -".g:tar_readoptions." - '".fname."'"
-  elseif tarfile =~# '\.bz2$'
+  elseif tarfile =~# '\.\(bz2\|fpm\|fdb\)$'
 "   call Decho("exe silent r! bzip2 -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -".g:tar_readoptions." - '".fname."'")
    exe "silent r! bzip2 -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -".g:tar_readoptions." - '".fname."'"
   else
@@ -283,9 +283,9 @@
    let tarfile = substitute(tarfile,'\.tgz','.tar','e')
    let compress= "gzip '".tarfile."'"
    let tgz     = 1
-  elseif tarfile =~# '\.bz2'
+  elseif tarfile =~# '\.\(bz2\|fpm\|fdb\)$'
    call system("bzip2 -d ".tarfile)
-   let tarfile = substitute(tarfile,'\.bz2','','e')
+   let tarfile = substitute(tarfile,'\.\(bz2\|fpm\|fdb\)$','','e')
    let compress= "bzip2 '".tarfile."'"
   endif
 
diff -Naur vim71.orig/runtime/filetype.vim vim71/runtime/filetype.vim
--- vim71.orig/runtime/filetype.vim	2007-05-10 17:14:37.000000000 +0200
+++ vim71/runtime/filetype.vim	2007-09-11 23:31:04.000000000 +0200
@@ -34,7 +34,7 @@
 " Pattern used to match file names which should not be inspected.
 " Currently finds compressed files.
 if !exists("g:ft_ignore_pat")
-  let g:ft_ignore_pat = '\.\(Z\|gz\|bz2\|zip\|tgz\)$'
+  let g:ft_ignore_pat = '\.\(Z\|gz\|bz2\|fpm\|fdb\|zip\|tgz\)$'
 endif
 
 " Function used for patterns that end in a star: don't set the filetype if the
@@ -1726,6 +1726,11 @@
   setf hog
 endfunc
 
+" FrugalBuild file
+au BufNewFile,BufRead FrugalBuild,*.install		setf sh
+
+" README.Frugalware file
+au BufNewFile,BufRead README.Frugalware		setf asciidoc 
 
 " Spec (Linux RPM)
 au BufNewFile,BufRead *.spec			setf spec
diff -Naur vim71.orig/runtime/plugin/tarPlugin.vim vim71/runtime/plugin/tarPlugin.vim
--- vim71.orig/runtime/plugin/tarPlugin.vim	2006-09-29 22:04:39.000000000 +0200
+++ vim71/runtime/plugin/tarPlugin.vim	2007-09-11 23:26:22.000000000 +0200
@@ -37,6 +37,8 @@
   au BufReadCmd   *.tar		call tar#Browse(expand("<amatch>"))
   au BufReadCmd   *.tar.gz	call tar#Browse(expand("<amatch>"))
   au BufReadCmd   *.tar.bz2	call tar#Browse(expand("<amatch>"))
+  au BufReadCmd   *.fpm		call tar#Browse(expand("<amatch>"))
+  au BufReadCmd   *.fdb		call tar#Browse(expand("<amatch>"))
   au BufReadCmd   *.tar.Z	call tar#Browse(expand("<amatch>"))
   au BufReadCmd   *.tgz		call tar#Browse(expand("<amatch>"))
 augroup END
diff -Naur vim71.orig/runtime/plugin/tarPlugin.vim.orig vim71/runtime/plugin/tarPlugin.vim.orig
--- vim71.orig/runtime/plugin/tarPlugin.vim.orig	1970-01-01 01:00:00.000000000 +0100
+++ vim71/runtime/plugin/tarPlugin.vim.orig	2006-09-29 22:04:39.000000000 +0200
@@ -0,0 +1,48 @@
+" tarPlugin.vim -- a Vim plugin for browsing tarfiles
+" Original was copyright (c) 2002, Michael C. Toren <mct@toren.net>
+" Modified by Charles E. Campbell, Jr.
+" Distributed under the GNU General Public License.
+"
+" Updates are available from <http://michael.toren.net/code/>.  If you
+" find this script useful, or have suggestions for improvements, please
+" let me know.
+" Also look there for further comments and documentation.
+"
+" This part only sets the autocommands.  The functions are in autoload/tar.vim.
+" ---------------------------------------------------------------------
+"  Load Once: {{{1
+if &cp || exists("g:loaded_tarPlugin")
+ finish
+endif
+let g:loaded_tarPlugin = 1
+let s:keepcpo          = &cpo
+set cpo&vim
+
+" ---------------------------------------------------------------------
+"  Public Interface: {{{1
+augroup tar
+  au!
+  au BufReadCmd   tarfile:*	call tar#Read(expand("<amatch>"), 1)
+  au FileReadCmd  tarfile:*	call tar#Read(expand("<amatch>"), 0)
+  au BufWriteCmd  tarfile:*	call tar#Write(expand("<amatch>"))
+  au FileWriteCmd tarfile:*	call tar#Write(expand("<amatch>"))
+
+  if has("unix")
+   au BufReadCmd   tarfile:*/*	call tar#Read(expand("<amatch>"), 1)
+   au FileReadCmd  tarfile:*/*	call tar#Read(expand("<amatch>"), 0)
+   au BufWriteCmd  tarfile:*/*	call tar#Write(expand("<amatch>"))
+   au FileWriteCmd tarfile:*/*	call tar#Write(expand("<amatch>"))
+  endif
+
+  au BufReadCmd   *.tar		call tar#Browse(expand("<amatch>"))
+  au BufReadCmd   *.tar.gz	call tar#Browse(expand("<amatch>"))
+  au BufReadCmd   *.tar.bz2	call tar#Browse(expand("<amatch>"))
+  au BufReadCmd   *.tar.Z	call tar#Browse(expand("<amatch>"))
+  au BufReadCmd   *.tgz		call tar#Browse(expand("<amatch>"))
+augroup END
+
+" ---------------------------------------------------------------------
+" Restoration And Modelines: {{{1
+" vim: fdm=marker
+let &cpo= s:keepcpo
+unlet s:keepcpo
diff -Naur vim71.orig/runtime/syntax/sh.vim vim71/runtime/syntax/sh.vim
--- vim71.orig/runtime/syntax/sh.vim	2007-05-07 22:12:00.000000000 +0200
+++ vim71/runtime/syntax/sh.vim	2007-09-11 23:46:48.000000000 +0200
@@ -436,6 +436,10 @@
  endif
 endif
 
+" Useful FrugalBuild Keywords: {{{1
+" ===================
+syn keyword shStatement Fmessage Fdie Fcd Fmkdir Frm Fcp Fcpr Fmv Finstallrel Finstall Fdirschmod Ffileschmod Fdirschown Ffileschown Fexe Fexerel Ffile Ffilerel Fdoc Fdocrel Fln Fsed Fdeststrip Fpatch Fpatchall Fconf Fmake Fmakeinstall Fbuild Frcd Frcd2 Facu Fsort Funpack_makeself Fautoconfize Finclude Flasttar Flasttgz Flasttarbz2 Fsplit Fextract Fcprrel Fman Fmanrel Ficon Ficonrel Fnant Fautoreconf Fdesktop Fdesktop2 Fwrapper Fmsgfmt
+
 " Synchronization: {{{1
 " ================
 if !exists("sh_minlines")
