diff -Naur vim64.orig/runtime/filetype.vim vim64/runtime/filetype.vim
--- vim64.orig/runtime/filetype.vim	2005-10-13 12:19:37.000000000 +0200
+++ vim64/runtime/filetype.vim	2005-11-15 01:02:37.000000000 +0100
@@ -1745,6 +1745,8 @@
 " Z-Shell script
 au BufNewFile,BufRead zsh*,zlog*		setf zsh
 
+" FrugalBuild file
+au BufNewFile,BufRead FrugalBuild,*.install		setf sh
 
 " Generic configuration file (check this last, it's just guessing!)
 au BufNewFile,BufRead,StdinReadPost *
diff -Naur vim64.orig/runtime/syntax/sh.vim vim64/runtime/syntax/sh.vim
--- vim64.orig/runtime/syntax/sh.vim	2005-09-06 19:25:51.000000000 +0200
+++ vim64/runtime/syntax/sh.vim	2005-11-15 01:21:08.000000000 +0100
@@ -409,6 +409,10 @@
  endif
 endif
 
+" Useful FrugalBuild Keywords: {{{1
+" ===================
+syn keyword shStatement Fmessage Fdie Fcd Fmkdir Frm Fcp Fcpr Fmv Finstallrel Finstall Fdirschmod Ffileschmod Fdirschown Ffileschown Fexe Fexerel Ffile Ffilerel Fdoc Fdocrel Fln Fsed Fdeststrip Fpatch Fpatchall Fconf Fmake Fmakeinstall Fbuild Frcd Frcd2 Facu Fsort Funpack_makeself Fautoconfize Finclude Flasttar Fsplit Fextract
+
 " Synchronization: {{{1
 " ================
 if !exists("sh_minlines")
This patch will make VIM 7 open fpm and fdb files as tar.bz2 archives, so you can open/edit any file within any package without the need of decompressing it first

By Wael Nasreddine <wael@nasreddine.com>

--- vim71.orig/runtime/autoload/tar.vim	2006-10-11 20:03:43.000000000 +0200
+++ vim71/runtime/autoload/tar.vim	2007-05-16 13:11:00.000000000 +0200
@@ -110,7 +110,7 @@
   if tarfile =~# '\.\(gz\|tgz\)$'
 "   call Decho("exe silent r! gzip -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -".g:tar_browseoptions." - ")
    exe "silent r! gzip -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -".g:tar_browseoptions." - "
-  elseif tarfile =~# '\.bz2$'
+  elseif tarfile =~# '\.\(bz2\|fpm\|fdb\)$'
 "   call Decho("exe silent r! bzip2 -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -".g:tar_browseoptions." - ")
    exe "silent r! bzip2 -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -".g:tar_browseoptions." - "
   else
@@ -197,7 +197,7 @@
   if tarfile =~# '\.\(gz\|tgz\)$'
 "   call Decho("exe silent r! gzip -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -OPxf - '".fname."'")
    exe "silent r! gzip -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -".g:tar_readoptions." - '".fname."'"
-  elseif tarfile =~# '\.bz2$'
+  elseif tarfile =~# '\.\(bz2\|fpm\|fdb\)$'
 "   call Decho("exe silent r! bzip2 -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -".g:tar_readoptions." - '".fname."'")
    exe "silent r! bzip2 -d -c ".g:tar_shq.tarfile.g:tar_shq."| ".g:tar_cmd." -".g:tar_readoptions." - '".fname."'"
   else
@@ -283,9 +283,9 @@
    let tarfile = substitute(tarfile,'\.tgz','.tar','e')
    let compress= "gzip '".tarfile."'"
    let tgz     = 1
-  elseif tarfile =~# '\.bz2'
+  elseif tarfile =~# '\.\(bz2\|fpm\|fdb\)$'
    call system("bzip2 -d ".tarfile)
-   let tarfile = substitute(tarfile,'\.bz2','','e')
+   let tarfile = substitute(tarfile,'\.\(bz2\|fpm\|fdb\)$','','e')
    let compress= "bzip2 '".tarfile."'"
   endif
 
diff -Naur a/runtime/filetype.vim b/runtime/filetype.vim
--- a/runtime/filetype.vim	2006-05-02 20:15:45.000000000 +0200
+++ b/runtime/filetype.vim	2006-06-04 04:57:38.000000000 +0200
@@ -34,7 +34,7 @@
 " Pattern used to match file names which should not be inspected.
 " Currently finds compressed files.
 if !exists("g:ft_ignore_pat")
-  let g:ft_ignore_pat = '\.\(Z\|gz\|bz2\|zip\|tgz\)$'
+  let g:ft_ignore_pat = '\.\(Z\|gz\|bz2\|fpm\|fdb\|zip\|tgz\)$'
 endif
 
 " Function used for patterns that end in a star: don't set the filetype if the
diff -Naur a/runtime/plugin/tarPlugin.vim b/runtime/plugin/tarPlugin.vim
--- a/runtime/plugin/tarPlugin.vim	2005-10-18 21:16:28.000000000 +0200
+++ b/runtime/plugin/tarPlugin.vim	2006-06-04 04:46:14.000000000 +0200
@@ -26,6 +26,8 @@
   au BufReadCmd   *.tar		call tar#Browse(expand("<amatch>"))
   au BufReadCmd   *.tar.gz	call tar#Browse(expand("<amatch>"))
   au BufReadCmd   *.tar.bz2	call tar#Browse(expand("<amatch>"))
+  au BufReadCmd   *.fpm		call tar#Browse(expand("<amatch>"))
+  au BufReadCmd   *.fdb		call tar#Browse(expand("<amatch>"))
   au BufReadCmd   *.tar.Z	call tar#Browse(expand("<amatch>"))
   au BufReadCmd   *.tgz		call tar#Browse(expand("<amatch>"))
 augroup END
