diff -Naur vim71.orig/runtime/filetype.vim vim71/runtime/filetype.vim
--- vim71.orig/runtime/filetype.vim	2007-05-10 17:14:37.000000000 +0200
+++ vim71/runtime/filetype.vim	2007-09-11 23:31:04.000000000 +0200
@@ -1726,6 +1726,11 @@
   setf hog
 endfunc
 
+" FrugalBuild file
+au BufNewFile,BufRead FrugalBuild,*.install		setf sh
+
+" README.Frugalware file
+au BufNewFile,BufRead README.Frugalware		setf asciidoc 
 
 " Spec (Linux RPM)
 au BufNewFile,BufRead *.spec			setf spec
diff -Naur vim71.orig/runtime/syntax/sh.vim vim71/runtime/syntax/sh.vim
--- vim71.orig/runtime/syntax/sh.vim	2007-05-07 22:12:00.000000000 +0200
+++ vim71/runtime/syntax/sh.vim	2007-09-11 23:46:48.000000000 +0200
@@ -436,6 +436,10 @@
  endif
 endif
 
+" Useful FrugalBuild Keywords: {{{1
+" ===================
+syn keyword shStatement Fmessage Fdie Fcd Fmkdir Frm Fcp Fcpr Fmv Finstallrel Finstall Fdirschmod Ffileschmod Fdirschown Ffileschown Fexe Fexerel Ffile Ffilerel Fdoc Fdocrel Fln Fsed Fdeststrip Fpatch Fpatchall Fconf Fmake Fmakeinstall Fbuild Frcd Frcd2 Facu Fsort Funpack_makeself Fautoconfize Finclude Flasttar Flasttgz Flasttarbz2 Fsplit Fextract Fcprrel Fman Fmanrel Ficon Ficonrel Fnant Fautoreconf Fdesktop Fdesktop2 Fwrapper Fmsgfmt
+
 " Synchronization: {{{1
 " ================
 if !exists("sh_minlines")
