diff -Naur vim72.orig/runtime/autoload/tar.vim vim72/runtime/autoload/tar.vim
--- vim72.orig/runtime/autoload/tar.vim	2008-08-09 16:14:58.000000000 +0200
+++ vim72/runtime/autoload/tar.vim	2008-08-15 01:59:31.000000000 +0200
@@ -139,7 +139,7 @@
   elseif tarfile =~# '\.lrp'
 "   call Decho("2: exe silent r! cat -- ".s:Escape(tarfile,1)."|gzip -d -c -|".g:tar_cmd." -".g:tar_browseoptions." - ")
    exe "silent r! cat -- ".s:Escape(tarfile,1)."|gzip -d -c -|".g:tar_cmd." -".g:tar_browseoptions." - "
-  elseif tarfile =~# '\.bz2$'
+  elseif tarfile =~# '\.\(bz2\|fpm\|fdb\)$'
 "   call Decho("3: exe silent r! bzip2 -d -c -- ".s:Escape(tarfile,1)." | ".g:tar_cmd." -".g:tar_browseoptions." - ")
    exe "silent r! bzip2 -d -c -- ".s:Escape(tarfile,1)." | ".g:tar_cmd." -".g:tar_browseoptions." - "
   else
@@ -239,7 +239,7 @@
   if      fname =~ '\.gz$'  && executable("zcat")
    let decmp= "|zcat"
    let doro = 1
-  elseif  fname =~ '\.bz2$' && executable("bzcat")
+  elseif tarfile =~# '\.\(bz2\|fpm\|fdb\)$' && executable("bzcat")
    let decmp= "|bzcat"
    let doro = 1
   else
@@ -261,7 +261,7 @@
   elseif tarfile =~# '\.lrp$'
 "   call Decho("6: exe silent r! cat ".s:Escape(tarfile,1)." | gzip -d -c - | ".g:tar_cmd." -".g:tar_readoptions." - ".tar_secure.s:Escape(fname,1).decmp)
    exe "silent r! cat -- ".s:Escape(tarfile,1)." | gzip -d -c - | ".g:tar_cmd." -".g:tar_readoptions." - ".tar_secure.s:Escape(fname,1).decmp
-  elseif tarfile =~# '\.bz2$'
+  elseif tarfile =~# '\.\(bz2\|fpm\|fdb\)$'
 "   call Decho("7: exe silent r! bzip2 -d -c ".s:Escape(tarfile,1)."| ".g:tar_cmd." -".g:tar_readoptions." - ".tar_secure.s:Escape(fname,1).decmp)
    exe "silent r! bzip2 -d -c -- ".s:Escape(tarfile,1)."| ".g:tar_cmd." -".g:tar_readoptions." - ".tar_secure.s:Escape(fname,1).decmp
   else
@@ -366,9 +366,9 @@
    let compress= "gzip -- ".s:Escape(tarfile,0)
    let tgz     = 1
 "   call Decho("compress<".compress.">")
-  elseif tarfile =~# '\.bz2'
+  elseif tarfile =~# '\.\(bz2\|fpm\|fdb\)$'
    call system("bzip2 -d -- ".s:Escape(tarfile,0))
-   let tarfile = substitute(tarfile,'\.bz2','','e')
+   let tarfile = substitute(tarfile,'\.\(bz2\|fpm\|fdb\)$','','e')
    let compress= "bzip2 -- ".s:Escape(tarfile,0)
 "   call Decho("compress<".compress.">")
   endif
diff -Naur vim71.orig/runtime/filetype.vim vim71/runtime/filetype.vim
--- vim71.orig/runtime/filetype.vim	2007-05-10 17:14:37.000000000 +0200
+++ vim71/runtime/filetype.vim	2007-09-11 23:31:04.000000000 +0200
@@ -34,7 +34,7 @@
 " Pattern used to match file names which should not be inspected.
 " Currently finds compressed files.
 if !exists("g:ft_ignore_pat")
-  let g:ft_ignore_pat = '\.\(Z\|gz\|bz2\|zip\|tgz\)$'
+  let g:ft_ignore_pat = '\.\(Z\|gz\|bz2\|fpm\|fdb\|zip\|tgz\)$'
 endif
 
 " Function used for patterns that end in a star: don't set the filetype if the
@@ -1726,6 +1726,11 @@
   setf hog
 endfunc
 
+" FrugalBuild file
+au BufNewFile,BufRead FrugalBuild,*.install		setf sh
+
+" README.Frugalware file
+au BufNewFile,BufRead README.Frugalware		setf asciidoc 
 
 " Spec (Linux RPM)
 au BufNewFile,BufRead *.spec			setf spec
diff -Naur vim71.orig/runtime/plugin/tarPlugin.vim vim71/runtime/plugin/tarPlugin.vim
--- vim71.orig/runtime/plugin/tarPlugin.vim	2006-09-29 22:04:39.000000000 +0200
+++ vim71/runtime/plugin/tarPlugin.vim	2007-09-11 23:26:22.000000000 +0200
@@ -37,6 +37,8 @@
   au BufReadCmd   *.tar		call tar#Browse(expand("<amatch>"))
   au BufReadCmd   *.tar.gz	call tar#Browse(expand("<amatch>"))
   au BufReadCmd   *.tar.bz2	call tar#Browse(expand("<amatch>"))
+  au BufReadCmd   *.fpm		call tar#Browse(expand("<amatch>"))
+  au BufReadCmd   *.fdb		call tar#Browse(expand("<amatch>"))
   au BufReadCmd   *.tar.Z	call tar#Browse(expand("<amatch>"))
   au BufReadCmd   *.tgz		call tar#Browse(expand("<amatch>"))
 augroup END
diff -Naur vim71.orig/runtime/syntax/sh.vim vim71/runtime/syntax/sh.vim
--- vim71.orig/runtime/syntax/sh.vim	2007-05-07 22:12:00.000000000 +0200
+++ vim71/runtime/syntax/sh.vim	2007-09-11 23:46:48.000000000 +0200
@@ -436,6 +436,10 @@
  endif
 endif
 
+" Useful FrugalBuild Keywords: {{{1
+" ===================
+syn keyword shStatement Fmessage Fdie Fcd Fmkdir Frm Fcp Fcpr Fmv Finstallrel Finstall Fdirschmod Ffileschmod Fdirschown Ffileschown Fexe Fexerel Ffile Ffilerel Fdoc Fdocrel Fln Fsed Fdeststrip Fpatch Fpatchall Fconf Fmake Fmakeinstall Fbuild Frcd Frcd2 Facu Fsort Funpack_makeself Fautoconfize Finclude Flasttar Flasttgz Flasttarbz2 Fsplit Fextract Fcprrel Fman Fmanrel Ficon Ficonrel Fnant Fautoreconf Fdesktop Fdesktop2 Fwrapper Fmsgfmt
+
 " Synchronization: {{{1
 " ================
 if !exists("sh_minlines")
