diff -Naur vim63.orig/runtime/vimrc_example.vim vim63/runtime/vimrc_example.vim
--- vim63.orig/runtime/vimrc_example.vim	2002-09-19 10:16:52.000000000 +0200
+++ vim63/runtime/vimrc_example.vim	2004-06-09 02:07:16.000000000 +0200
@@ -21,11 +21,7 @@
 " allow backspacing over everything in insert mode
 set backspace=indent,eol,start
 
-if has("vms")
-  set nobackup		" do not keep a backup file, use versions instead
-else
-  set backup		" keep a backup file
-endif
+set nobackup           " do not keep a backup file, use versions instead
 set history=50		" keep 50 lines of command line history
 set ruler		" show the cursor position all the time
 set showcmd		" display incomplete commands
@@ -38,7 +38,7 @@
 map Q gq
 
 " In many terminal emulators the mouse works just fine, thus enable it.
-set mouse=a
+"set mouse=a
 
 " Switch syntax highlighting on, when the terminal has colors
 " Also switch on highlighting the last used search pattern.
@@ -46,6 +42,7 @@
 if &t_Co > 2 || has("gui_running")
   syntax on
   set hlsearch
+  set bg=dark
 endif
 
 " Only do this part when compiled with support for autocommands.
