diff -urN dhcdbd-1.14/system.d/dhcdbd.conf dhcdbd-1.14.new/system.d/dhcdbd.conf
--- dhcdbd-1.14/system.d/dhcdbd.conf	2006-01-17 21:23:51.000000000 +0000
+++ dhcdbd-1.14.new/system.d/dhcdbd.conf	2006-07-20 16:18:26.000000000 +0100
@@ -7,10 +7,6 @@
 	    <allow send_interface="com.redhat.dhcp"/>
 	    <allow send_destination="com.redhat.dhcp"/>
     </policy>
-    <policy user="named">
-            <allow send_interface="com.redhat.dhcp"/>
-            <allow send_destination="com.redhat.dhcp"/>
-    </policy>
     <policy context="default">
             <deny own="com.redhat.dhcp"/>
             <deny send_destination="com.redhat.dhcp"/>
