--- subtitlecomposer-0.5.3/src/main/CMakeLists.txt	2009-04-17 19:09:12.000000000 +0000
+++ subtitlecomposer-0.5.3.new/src/main/CMakeLists.txt	2012-09-02 18:15:37.000000000 +0000
@@ -41,12 +41,15 @@
 
 KDE4_ADD_EXECUTABLE( subtitlecomposer ${subtitlecomposer_SRCS} )
 
+FIND_PACKAGE( X11 REQUIRED )
+
 TARGET_LINK_LIBRARIES(
 	subtitlecomposer
 	${KDE4_KDEUI_LIBS}
 	${KDE4_KHTML_LIBS}
 	${KDE4_KFILE_LIBS}
 	${KDE4_KROSSUI_LIBS}
+	${X11_LIBRARIES}
 	${common_LIBS}
 	${config_LIBS}
 	${core_LIBS}
