From d0bc81e8d7af7df512e5556a2d1da0c2053c75cd Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Tue, 28 Jul 2009 03:32:31 +0000
Subject: newport: update for resources/RAC API removal

---
diff --git a/src/newport.h b/src/newport.h
index f7354e9..d4435b7 100644
--- a/src/newport.h
+++ b/src/newport.h
@@ -23,7 +23,10 @@
 #include "xf86.h"
 #include "xf86_OSproc.h"
 #include "compiler.h"
+
+#ifndef XSERVER_LIBPCIACCESS
 #include "xf86Resources.h"
+#endif
 
 #include "xf86cmap.h"
 
diff --git a/src/newport_driver.c b/src/newport_driver.c
index 757fd12..64b643a 100644
--- a/src/newport_driver.c
+++ b/src/newport_driver.c
@@ -247,7 +247,9 @@ NewportProbe(DriverPtr drv, int flags)
 	Bool foundScreen = FALSE;
 	GDevPtr *devSections;
 	GDevPtr dev = NULL;
+#ifndef XSERVER_LIBPCIACCESS
 	resRange range[] = { {ResExcMemBlock ,0,0}, _END };
+#endif
 	unsigned probedIDs[NEWPORT_MAX_BOARDS];
 	memType base;
 
@@ -270,10 +272,12 @@ NewportProbe(DriverPtr drv, int flags)
 					ScrnInfoPtr pScrn = NULL;
 
 					entity = xf86ClaimNoSlot(drv, 0, dev, TRUE);
+#ifndef XSERVER_LIBPCIACCESS
 					base = (NEWPORT_BASE_ADDR0
 						+ busID * NEWPORT_BASE_OFFSET);
 					RANGE(range[0], base, base + sizeof(NewportRegs),\
 							ResExcMemBlock);
+#endif
 					pScrn = xf86AllocateScreen(drv, 0);
 					xf86AddEntityToScreen(pScrn, entity);
 					pScrn->driverVersion = NEWPORT_VERSION;
--
cgit v0.8.2

