diff -Naur user-manager-5.6.2/src/accountinfo.cpp user-manager-5.6.2-p/src/accountinfo.cpp
--- user-manager-5.6.2/src/accountinfo.cpp	2016-04-05 16:03:37.000000000 +0200
+++ user-manager-5.6.2-p/src/accountinfo.cpp	2016-04-08 15:41:07.497405808 +0200
@@ -56,7 +56,7 @@
     connect(m_info->realName, SIGNAL(textEdited(QString)), SLOT(hasChanged()));
     connect(m_info->email, SIGNAL(textEdited(QString)), SLOT(hasChanged()));
     connect(m_info->administrator, SIGNAL(clicked(bool)), SLOT(hasChanged()));
-    connect(m_info->automaticLogin, SIGNAL(clicked(bool)), SLOT(hasChanged()));
+    //connect(m_info->automaticLogin, SIGNAL(clicked(bool)), SLOT(hasChanged()));
     connect(m_passwordEdit, SIGNAL(focused()), SLOT(changePassword()));
     connect(m_passwordEdit, SIGNAL(textEdited(QString)), SLOT(changePassword()));
 
@@ -137,7 +137,7 @@
     m_info->realName->setText(m_model->data(m_index, AccountModel::RealName).toString());
     m_info->email->setText(m_model->data(m_index, AccountModel::Email).toString());
     m_info->administrator->setChecked(m_model->data(m_index, AccountModel::Administrator).toBool());
-    m_info->automaticLogin->setChecked(m_model->data(m_index, AccountModel::AutomaticLogin).toBool());
+    //m_info->automaticLogin->setChecked(m_model->data(m_index, AccountModel::AutomaticLogin).toBool());
     m_passwordEdit->clear();
 }
 
@@ -161,9 +161,9 @@
     if (!m_model->setData(m_index, m_info->administrator->isChecked(), AccountModel::Administrator)) {
         failed.append(AccountModel::Administrator);
     }
-    if (!m_model->setData(m_index, m_info->automaticLogin->isChecked(), AccountModel::AutomaticLogin)) {
-        failed.append(AccountModel::AutomaticLogin);
-    }
+    //if (!m_model->setData(m_index, m_info->automaticLogin->isChecked(), AccountModel::AutomaticLogin)) {
+    //    failed.append(AccountModel::AutomaticLogin);
+    //}
     if (m_infoToSave.contains(AccountModel::Password)) {
         if (!m_model->setData(m_index, m_infoToSave[AccountModel::Password], AccountModel::Password)) {
             failed.append(AccountModel::Password);
@@ -227,9 +227,9 @@
         infoToSave.insert(AccountModel::Administrator, m_info->administrator->isChecked());
     }
 
-    if (m_info->automaticLogin->isChecked() != m_model->data(m_index, AccountModel::AutomaticLogin).toBool()) {
-        infoToSave.insert(AccountModel::AutomaticLogin, m_info->automaticLogin->isChecked());
-    }
+    //if (m_info->automaticLogin->isChecked() != m_model->data(m_index, AccountModel::AutomaticLogin).toBool()) {
+    //    infoToSave.insert(AccountModel::AutomaticLogin, m_info->automaticLogin->isChecked());
+    //}
 
     if (m_infoToSave.contains(AccountModel::Face)) {
         infoToSave[AccountModel::Face] = m_infoToSave[AccountModel::Face];
diff -Naur user-manager-5.6.2/src/account.ui user-manager-5.6.2-p/src/account.ui
--- user-manager-5.6.2/src/account.ui	2016-04-05 16:03:37.000000000 +0200
+++ user-manager-5.6.2-p/src/account.ui	2016-04-08 15:29:28.308641646 +0200
@@ -182,20 +182,6 @@
         </property>
        </widget>
       </item>
-      <item row="5" column="0">
-       <widget class="QLabel" name="label_9">
-        <property name="text">
-         <string>Automatic Login:</string>
-        </property>
-       </widget>
-      </item>
-      <item row="5" column="1">
-       <widget class="QCheckBox" name="automaticLogin">
-        <property name="text">
-         <string/>
-        </property>
-       </widget>
-      </item>
       <item row="3" column="1">
        <widget class="QLineEdit" name="passwordEdit">
         <property name="sizePolicy">
