diff -Naur dvdshrink.orig/install.sh dvdshrink/install.sh
--- dvdshrink.orig/install.sh	2005-03-28 06:32:58.000000000 +0200
+++ dvdshrink/install.sh	2006-09-05 01:50:20.000000000 +0200
@@ -41,7 +41,8 @@
 if [ $BAD -gt 0 ]; then
    echo
    echo "   One or more of the utilities that DVDShrink requires are not installed or"
-   echo -n "   are not executable. Continue the installation? (y/n)   "
+   echo -n "   are not executable."
+   exit 1
    read YN
    if [ -z $YN ]; then exit; fi
    if [ `testistring $YN 'n'` -eq 1 ]; then exit; fi 
