diff -Naur dvdshrink.orig/install.sh dvdshrink/install.sh
--- dvdshrink.orig/install.sh	2005-03-28 06:32:58.000000000 +0200
+++ dvdshrink/install.sh	2006-09-05 03:14:11.000000000 +0200
@@ -13,6 +13,11 @@
    exit
 fi
 
+if [ $# -eq 1 ]
+then
+   DESTDIR=$1
+fi
+
 function testistring {
    INP=`echo $1 | tr 'a-z' 'A-Z'`
    TST=`echo $2 | tr 'a-z' 'A-Z'`
@@ -63,14 +68,15 @@
 echo "Creating application directories if needed"
 for DIR in doc applications; do
    if [ ! -d /usr/share/$DIR/dvdshrink ]; then
-      mkdir -p /usr/share/$DIR/dvdshrink
+      mkdir -p $DESTDIR/usr/share/$DIR/dvdshrink
    fi
 done
 
 echo
 echo "Installing files"
-cp usr/bin/* /usr/bin
-cp -R usr/share/* /usr/share
+mkdir -p $DESTDIR/usr/bin
+cp -a usr/bin/* $DESTDIR/usr/bin
+cp -R usr/share/* $DESTDIR/usr/share
 
 echo
 echo "Removing rc files if they exist"
