diff -Naur ois-v1-3/demos/FFConsoleDemo.cpp ois-v1-3-p/demos/FFConsoleDemo.cpp
--- ois-v1-3/demos/FFConsoleDemo.cpp	2010-04-01 05:38:04.000000000 +0200
+++ ois-v1-3-p/demos/FFConsoleDemo.cpp	2016-06-02 02:44:37.148477338 +0200
@@ -7,7 +7,7 @@
 #include <ios>
 #include <sstream>
 #include <vector>
-
+#include <unistd.h>
 using namespace std;
 
 ////////////////////////////////////Needed Windows Headers////////////
diff -Naur ois-v1-3/demos/OISConsole.cpp ois-v1-3-p/demos/OISConsole.cpp
--- ois-v1-3/demos/OISConsole.cpp	2010-08-16 00:51:34.000000000 +0200
+++ ois-v1-3-p/demos/OISConsole.cpp	2016-06-02 02:47:39.066187911 +0200
@@ -12,6 +12,7 @@
 #include <iostream>
 #include <vector>
 #include <sstream>
+#include <unistd.h>
 
 ////////////////////////////////////Needed Windows Headers////////////
 #if defined OIS_WIN32_PLATFORM
@@ -77,7 +78,7 @@
 	}
 	bool keyReleased( const KeyEvent &arg ) {
 		if( arg.key == KC_ESCAPE || arg.key == KC_Q )
-			appRunning = false;
+			appRunning = false;
 		std::cout << "KeyReleased {" << ((Keyboard*)(arg.device))->getAsString(arg.key) << "}\n";
 		return true;
 	}
diff -Naur ois-v1-3/src/linux/LinuxForceFeedback.cpp ois-v1-3-p/src/linux/LinuxForceFeedback.cpp
--- ois-v1-3/src/linux/LinuxForceFeedback.cpp	2010-04-01 05:06:55.000000000 +0200
+++ ois-v1-3-p/src/linux/LinuxForceFeedback.cpp	2016-06-02 02:39:39.975788107 +0200
@@ -26,6 +26,7 @@
 #include <cstdlib>
 #include <errno.h>
 #include <memory.h>
+#include <unistd.h>
 
 using namespace OIS;
 
diff -Naur ois-v1-3/src/linux/LinuxJoyStickEvents.cpp ois-v1-3-p/src/linux/LinuxJoyStickEvents.cpp
--- ois-v1-3/src/linux/LinuxJoyStickEvents.cpp	2010-07-26 03:28:41.000000000 +0200
+++ ois-v1-3-p/src/linux/LinuxJoyStickEvents.cpp	2016-06-02 02:40:12.303624219 +0200
@@ -31,7 +31,8 @@
 #include "OISException.h"
 
 #include <fcntl.h>        //Needed to Open a file descriptor
-#include <cassert>	
+#include <cassert>
+#include <unistd.h>
 #include <linux/input.h>
 
 
