diff -Naur audacious-1.1.2/audacious/audacious.desktop.in audacious-1.1.2-p/audacious/audacious.desktop.in
--- audacious-1.1.2/audacious/audacious.desktop.in	2006-08-25 13:57:52.000000000 +0200
+++ audacious-1.1.2-p/audacious/audacious.desktop.in	2006-08-29 22:44:19.588570056 +0200
@@ -5,7 +5,7 @@
 Exec=audacious
 Icon=audacious.png
 MimeType=audio/x-scpls;audio/x-mpegurl;audio/mpegurl;audio/mp3;audio/x-mp3;audio/mpeg;audio/x-mpeg;audio/x-wav;application/x-ogg;
-Categories=Application;GTK;AudioVideo;Player;
+Categories=KDE;GNOME;Application;AudioVideo;Player;
 Terminal=false
 Type=Application
 Encoding=UTF-8
