diff -Naur qbs-src-1.11.0/src/lib/corelib/tools/tools.pri qbs-src-1.11.0-p/src/lib/corelib/tools/tools.pri
--- qbs-src-1.11.0/src/lib/corelib/tools/tools.pri	2018-03-27 13:40:50.000000000 +0200
+++ qbs-src-1.11.0-p/src/lib/corelib/tools/tools.pri	2018-04-10 02:33:57.245014362 +0200
@@ -108,26 +108,7 @@
 }
 
 !qbs_no_dev_install {
-    tools_headers.files = \
-        $$PWD/architectures.h \
-        $$PWD/cleanoptions.h \
-        $$PWD/codelocation.h \
-        $$PWD/commandechomode.h \
-        $$PWD/error.h \
-        $$PWD/settings.h \
-        $$PWD/settingsmodel.h \
-        $$PWD/preferences.h \
-        $$PWD/profile.h \
-        $$PWD/processresult.h \
-        $$PWD/qbs_export.h \
-        $$PWD/buildoptions.h \
-        $$PWD/generateoptions.h \
-        $$PWD/generatorpluginmanager.h \
-        $$PWD/installoptions.h \
-        $$PWD/qbspluginmanager.h \
-        $$PWD/setupprojectparameters.h \
-        $$PWD/toolchains.h \
-        $$PWD/version.h
+    tools_headers.files = $$HEADERS
     tools_headers.path = $${QBS_INSTALL_PREFIX}/include/qbs/tools
     INSTALLS += tools_headers
 }
