diff -Naur qbs-src-1.11.1/src/lib/corelib/tools/tools.pri qbs-src-1.11.1-p/src/lib/corelib/tools/tools.pri
--- qbs-src-1.11.1/src/lib/corelib/tools/tools.pri	2018-04-23 15:07:22.000000000 +0200
+++ qbs-src-1.11.1-p/src/lib/corelib/tools/tools.pri	2018-05-06 16:04:52.675357669 +0200
@@ -108,25 +108,7 @@
 }
 
 !qbs_no_dev_install {
-    tools_headers.files = \
-        $$PWD/architectures.h \
-        $$PWD/buildoptions.h \
-        $$PWD/cleanoptions.h \
-        $$PWD/codelocation.h \
-        $$PWD/commandechomode.h \
-        $$PWD/error.h \
-        $$PWD/generateoptions.h \
-        $$PWD/installoptions.h \
-        $$PWD/preferences.h \
-        $$PWD/processresult.h \
-        $$PWD/profile.h \
-        $$PWD/projectgeneratormanager.h \
-        $$PWD/qbs_export.h \
-        $$PWD/settings.h \
-        $$PWD/settingsmodel.h \
-        $$PWD/setupprojectparameters.h \
-        $$PWD/toolchains.h \
-        $$PWD/version.h
+    tools_headers.files = $$HEADERS
     tools_headers.path = $${QBS_INSTALL_PREFIX}/include/qbs/tools
     INSTALLS += tools_headers
 }
