diff -Naur ntfs-3g_ntfsprogs-2022.10.3/ntfsprogs/Makefile.in ntfs-3g_ntfsprogs-2022.10.3.new/ntfsprogs/Makefile.in
--- ntfs-3g_ntfsprogs-2022.10.3/ntfsprogs/Makefile.in	2022-10-20 17:35:12.000000000 +0200
+++ ntfs-3g_ntfsprogs-2022.10.3.new/ntfsprogs/Makefile.in	2024-01-08 13:21:54.895868441 +0100
@@ -1532,7 +1532,7 @@
 @ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFSPROGS_TRUE@	$(LN_S) -f mkntfs.8 $(DESTDIR)$(man8dir)/mkfs.ntfs.8
 
 @ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFSPROGS_TRUE@uninstall-local:
-@ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFSPROGS_TRUE@	$(RM) -f $(DESTDIR)/sbin/mkfs.ntfs
+@ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFSPROGS_TRUE@	$(RM) -f $(DESTDIR)/usr/bin/mkfs.ntfs
 @ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFSPROGS_TRUE@	$(RM) -f $(DESTDIR)$(man8dir)/mkfs.ntfs.8
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
diff -Naur ntfs-3g_ntfsprogs-2022.10.3/src/Makefile.in ntfs-3g_ntfsprogs-2022.10.3.new/src/Makefile.in
--- ntfs-3g_ntfsprogs-2022.10.3/src/Makefile.in	2022-10-20 17:35:12.000000000 +0200
+++ ntfs-3g_ntfsprogs-2022.10.3.new/src/Makefile.in	2024-01-08 13:21:54.895868441 +0100
@@ -1003,9 +1003,9 @@
 @DISABLE_PLUGINS_FALSE@@ENABLE_NTFS_3G_TRUE@	$(MKDIR_P) $(DESTDIR)/$(plugindir)
 
 @ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFS_3G_TRUE@install-exec-local:	install-rootbinPROGRAMS
-@ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFS_3G_TRUE@	$(MKDIR_P) "$(DESTDIR)/sbin"
-@ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFS_3G_TRUE@	$(LN_S) -f "$(rootbindir)/ntfs-3g" "$(DESTDIR)/sbin/mount.ntfs-3g"
-@ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFS_3G_TRUE@	$(LN_S) -f "$(rootbindir)/lowntfs-3g" "$(DESTDIR)/sbin/mount.lowntfs-3g"
+@ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFS_3G_TRUE@	$(MKDIR_P) "$(DESTDIR)/usr/bin"
+@ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFS_3G_TRUE@	$(LN_S) -f "$(rootbindir)/ntfs-3g" "$(DESTDIR)/usr/bin/mount.ntfs-3g"
+@ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFS_3G_TRUE@	$(LN_S) -f "$(rootbindir)/lowntfs-3g" "$(DESTDIR)/usr/bin/mount.lowntfs-3g"
 
 @ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFS_3G_TRUE@install-data-local:	install-man8
 @ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFS_3G_TRUE@	$(LN_S) -f ntfs-3g.8 "$(DESTDIR)$(man8dir)/mount.ntfs-3g.8"
@@ -1013,7 +1013,7 @@
 
 @ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFS_3G_TRUE@uninstall-local:
 @ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFS_3G_TRUE@	$(RM) -f "$(DESTDIR)$(man8dir)/mount.ntfs-3g.8"
-@ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFS_3G_TRUE@	$(RM) -f "$(DESTDIR)/sbin/mount.ntfs-3g" "$(DESTDIR)/sbin/mount.lowntfs-3g"
+@ENABLE_MOUNT_HELPER_TRUE@@ENABLE_NTFS_3G_TRUE@	$(RM) -f "$(DESTDIR)/usr/bin/mount.ntfs-3g" "$(DESTDIR)/sbin/mount.lowntfs-3g"
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
