--- cookie-0.5.0/Web/Cookie.hs	2024-03-04 09:52:42.000000000 +0100
+++ cookie-0.5.0.new/Web/Cookie.hs	2024-11-28 10:36:59.748247840 +0100
@@ -56,7 +56,7 @@
 import Data.Text.Encoding (encodeUtf8Builder, decodeUtf8With)
 import Data.Text.Encoding.Error (lenientDecode)
 import Data.Maybe (isJust)
-import Data.Default.Class (Default (def))
+import Data.Default (Default (def))
 import Control.DeepSeq (NFData (rnf))
 
 -- | Textual cookies. Functions assume UTF8 encoding.
--- cookie-0.5.0/cookie.cabal	2024-03-04 09:52:42.000000000 +0100
+++ cookie-0.5.0.new/cookie.cabal	2024-11-28 10:38:38.801861537 +0100
@@ -19,7 +19,7 @@
                    , bytestring                >= 0.10.2
                    , time                      >= 1.5
                    , text                      >= 1.1
-                   , data-default-class
+                   , data-default
                    , deepseq
     exposed-modules: Web.Cookie
     ghc-options:     -Wall
