diff -Naur gimp-2.2.0.orig/data/misc/gimp.desktop.in.in gimp-2.2.0/data/misc/gimp.desktop.in.in
--- gimp-2.2.0.orig/data/misc/gimp.desktop.in.in	2004-08-09 13:04:42.000000000 +0200
+++ gimp-2.2.0/data/misc/gimp.desktop.in.in	2004-12-19 11:37:06.980732120 +0100
@@ -7,7 +7,7 @@
 _Comment=Create and edit images or photographs
 Exec=gimp-remote-@GIMP_APP_VERSION@ %U
 TryExec=gimp-@GIMP_APP_VERSION@
-Icon=@gimpdatadir@/images/@GIMP_DESKTOP_ICON@
+Icon=gimp.png
 Terminal=false
 Categories=Application;Graphics;2DGraphics;RasterGraphics;
 X-GNOME-Bugzilla-Bugzilla=GNOME
