diff -Naur QScintilla-1.73-gpl-2.0/Qt3/qscintilla.pro QScintilla-1.73-gpl-2.0-p/Qt3/qscintilla.pro
--- QScintilla-1.73-gpl-2.0/Qt3/qscintilla.pro	2007-05-26 11:59:42.000000000 +0200
+++ QScintilla-1.73-gpl-2.0-p/Qt3/qscintilla.pro	2007-05-27 19:19:13.000000000 +0200
@@ -29,9 +29,8 @@
 DEFINES = QSCINTILLA_MAKE_DLL QT SCI_LEXER
 
 # Handle both Qt v4 and v3.
-DESTDIR = $$[QT_INSTALL_LIBS]
 isEmpty(DESTDIR) {
-	DESTDIR = $(QTDIR)/lib
+	DESTDIR = $(INSTALL_ROOT)$(QTDIR)/lib
 }
 
 header.path = $$[QT_INSTALL_HEADERS]
diff -Naur QScintilla-1.73-gpl-2.0/Qt4/qscintilla.pro QScintilla-1.73-gpl-2.0-p/Qt4/qscintilla.pro
--- QScintilla-1.73-gpl-2.0/Qt4/qscintilla.pro	2007-05-26 11:59:42.000000000 +0200
+++ QScintilla-1.73-gpl-2.0-p/Qt4/qscintilla.pro	2007-05-27 19:18:34.000000000 +0200
@@ -29,9 +29,8 @@
 DEFINES = QSCINTILLA_MAKE_DLL QT SCI_LEXER
 
 # Handle both Qt v4 and v3.
-DESTDIR = $$[QT_INSTALL_LIBS]
 isEmpty(DESTDIR) {
-	DESTDIR = $(QTDIR)/lib
+	DESTDIR = $(INSTALL_ROOT)$(QTDIR)/lib
 }
 
 header.path = $$[QT_INSTALL_HEADERS]
