diff -Naur -X nope airtraf-1.1/src/libcommon/definition.h airtraf-1.1-p/src/libcommon/definition.h
--- airtraf-1.1/src/libcommon/definition.h	2003-01-22 04:35:00.000000000 +0100
+++ airtraf-1.1-p/src/libcommon/definition.h	2007-01-12 00:52:10.000000000 +0100
@@ -207,10 +207,10 @@
  **/
 typedef struct
 {
-  __u32 did __attribute__ ((packed));
-  __u16 status __attribute__ ((packed));
-  __u16 len __attribute__ ((packed));
-  __u32 data __attribute__ ((packed));
+  __u32 did;
+  __u16 status;
+  __u16 len;
+  __u32 data;
 } p80211item_t;
 
 /**
@@ -218,19 +218,19 @@
  **/
 typedef struct
 {
-  __u32 msgcode __attribute__ ((packed));
-  __u32 msglen __attribute__ ((packed));
-  __u8 devname[DEVNAME_LEN] __attribute__ ((packed));
-  p80211item_t hosttime __attribute__ ((packed));
-  p80211item_t mactime __attribute__ ((packed));
-  p80211item_t channel __attribute__ ((packed));
-  p80211item_t rssi __attribute__ ((packed));
-  p80211item_t sq __attribute__ ((packed));
-  p80211item_t signal __attribute__ ((packed));
-  p80211item_t noise __attribute__ ((packed));
-  p80211item_t rate __attribute__ ((packed));
-  p80211item_t istx __attribute__ ((packed));
-  p80211item_t frmlen __attribute__ ((packed));
+  __u32 msgcode;
+  __u32 msglen;
+  __u8 devname[DEVNAME_LEN];
+  p80211item_t hosttime;
+  p80211item_t mactime;
+  p80211item_t channel;
+  p80211item_t rssi;
+  p80211item_t sq;
+  p80211item_t signal;
+  p80211item_t noise;
+  p80211item_t rate;
+  p80211item_t istx;
+  p80211item_t frmlen;
 } wlan_ng_hdr_t;
 
 /**
@@ -242,13 +242,13 @@
 typedef struct
 {
   /* HFA384X RX frame descriptor */
-  __u16 status __attribute__ ((packed));
-  __u32 time __attribute__ ((packed));
-  __u8 silence __attribute__ ((packed));
-  __u8 signal __attribute__ ((packed));
-  __u8 rate __attribute__ ((packed));
-  __u8 rxflow __attribute__ ((packed));
-  __u32 reserved __attribute__ ((packed));
+  __u16 status;
+  __u32 time;
+  __u8 silence;
+  __u8 signal;
+  __u8 rate;
+  __u8 rxflow;
+  __u32 reserved;
 } hfa384x_descript_t;
 
 /**
@@ -260,23 +260,23 @@
  **/
 typedef struct
 {
-  hfa384x_descript_t frame_descriptor __attribute__ ((packed));
+  hfa384x_descript_t frame_descriptor;
   
   /* 802.11 */
-  __u16 frame_control __attribute__ ((packed));
-  __u16 duration_id __attribute__ ((packed));
-  __u8 mac1[6] __attribute__ ((packed));
-  __u8 mac2[6] __attribute__ ((packed));
-  __u8 mac3[6] __attribute__ ((packed));
-  __u16 sequence __attribute__ ((packed));
-  __u8 mac4[6] __attribute__ ((packed));
-  __u16 data_len __attribute__ ((packed));
+  __u16 frame_control;
+  __u16 duration_id;
+  __u8 mac1[6];
+  __u8 mac2[6];
+  __u8 mac3[6];
+  __u16 sequence;
+  __u8 mac4[6];
+  __u16 data_len;
 
   /* 802.3 */
-  __u8 dst_addr[6] __attribute__ ((packed));
-  __u8 src_addr[6] __attribute__ ((packed));
-  __u16 len __attribute__ ((packed));
-  __u8 crap[6] __attribute__ ((packed));
+  __u8 dst_addr[6];
+  __u8 src_addr[6];
+  __u16 len;
+  __u8 crap[6];
 
 } prism2_hdr_t;
    
@@ -289,13 +289,13 @@
  **/
 typedef struct
 {
-  __u16 frame_control __attribute__ ((packed));
-  __u16 duration_id __attribute__ ((packed));
-  __u8 mac1[6] __attribute__ ((packed));
-  __u8 mac2[6] __attribute__ ((packed));
-  __u8 mac3[6] __attribute__ ((packed));
-  __u16 sequence __attribute__ ((packed));
-  __u8 mac4[6] __attribute__ ((packed));
+  __u16 frame_control;
+  __u16 duration_id;
+  __u8 mac1[6];
+  __u8 mac2[6];
+  __u8 mac3[6];
+  __u16 sequence;
+  __u8 mac4[6];
 } wlan_hdr_t;
 
 /**
diff -Naur -X nope airtraf-1.1/src/libcommon/packet_card.c airtraf-1.1-p/src/libcommon/packet_card.c
--- airtraf-1.1/src/libcommon/packet_card.c	2003-01-22 04:35:00.000000000 +0100
+++ airtraf-1.1-p/src/libcommon/packet_card.c	2007-01-12 00:38:14.000000000 +0100
@@ -46,6 +46,7 @@
 #include <netinet/in.h>
 #include <linux/if_packet.h>
 #include <linux/if_ether.h>
+#include <linux/if.h>
 #include <linux/wireless.h> // wireless extensions
 #include <math.h> // math stuff...
 //#include <net/if.h>
diff -Naur -X nope airtraf-1.1/src/sniffd/autoconfig.c airtraf-1.1-p/src/sniffd/autoconfig.c
--- airtraf-1.1/src/sniffd/autoconfig.c	2007-01-12 01:12:15.000000000 +0100
+++ airtraf-1.1-p/src/sniffd/autoconfig.c	2007-01-12 00:39:52.000000000 +0100
@@ -50,6 +50,7 @@
 #include <sys/socket.h> /* For AF_INET & struct sockaddr */
 #include <netinet/in.h>   /* For struct sockaddr_in */
 #include <netinet/if_ether.h>
+#include <linux/if.h>
 #include <linux/wireless.h> /* For wireless extensions */
 
 #include "autoconfig.h"
diff -Naur -X nope airtraf-1.1/src/sniffd/server.c airtraf-1.1-p/src/sniffd/server.c
--- airtraf-1.1/src/sniffd/server.c	2007-01-12 01:12:15.000000000 +0100
+++ airtraf-1.1-p/src/sniffd/server.c	2007-01-12 00:58:13.000000000 +0100
@@ -143,7 +143,7 @@
       if (DEBUG) fprintf(stderr,"Error transmitting data\n");
     }
   } while (write_len == 0);
-  if (DEBUG) fprintf(stderr,"Data: %ld bytes sent\n",length);
+  if (DEBUG) fprintf(stderr,"Data: %ld bytes sent\n", (long)length);
 }
 
 /**
