diff -Naur cm/sndlib/premake4.lua cm.new/sndlib/premake4.lua
--- cm/sndlib/premake4.lua	2013-07-18 00:08:41.000000000 +0200
+++ cm.new/sndlib/premake4.lua	2014-03-09 18:20:15.702583809 +0100
@@ -92,11 +92,11 @@
    end
 end
 
-if (os.is64bit()) then
+-- if (os.is64bit()) then
    defines("SIZEOF_VOID_P=8")
-else
-   defines("SIZEOF_VOID_P=4")
-end
+-- else
+--   defines("SIZEOF_VOID_P=4")
+-- end
 
 
 -- TODO: WORDS_BIGENDIAN:
