diff -Naur libmirage-2.0.0/CMakeLists.txt libmirage-2.0.0.new/CMakeLists.txt
--- libmirage-2.0.0/CMakeLists.txt	2012-12-10 00:06:35.000000000 +0100
+++ libmirage-2.0.0.new/CMakeLists.txt	2013-03-04 20:02:54.670725920 +0100
@@ -27,8 +27,6 @@
 include (CheckCCompilerFlag)
 include (GNUInstallDirs)
 
-include (GObjectIntrospection)
-include (GtkDoc)
 include (Utilities)
 include (FileList)
 
@@ -50,10 +48,12 @@
 pkg_check_modules (SNDFILE REQUIRED sndfile>=1.0.0)
 
 if (INTROSPECTION_ENABLED)
+    include (GObjectIntrospection)
     pkg_check_modules (INTROSPECTION REQUIRED gobject-introspection-1.0>=1.30.0)
 endif ()
 
 if (GTKDOC_ENABLED)
+    include (GtkDoc)
     pkg_check_modules (GTKDOC REQUIRED gtk-doc>=1.4)
 endif ()
 
