diff -Naur pyzmq-18.0.0/setup.py pyzmq-18.0.0-p/setup.py
--- pyzmq-18.0.0/setup.py	2019-02-19 13:38:50.000000000 +0100
+++ pyzmq-18.0.0-p/setup.py	2019-03-03 20:36:43.873461919 +0100
@@ -1237,7 +1237,7 @@
 # Main setup
 #-----------------------------------------------------------------------------
 
-with open("README.md") as f:
+with open("README.md", encoding='utf-8') as f:
     long_desc = f.read()
 
 setup_args = dict(
