diff -Naur sim-0.9.3.orig/plugins/_core/cfgdlg.cpp sim-0.9.3/plugins/_core/cfgdlg.cpp
--- sim-0.9.3.orig/plugins/_core/cfgdlg.cpp	2005-08-05 14:19:33.000000000 +0200
+++ sim-0.9.3/plugins/_core/cfgdlg.cpp	2005-08-05 14:20:38.000000000 +0200
@@ -465,7 +465,7 @@
             continue;
         size_t size = 0;
         for (const DataDef *d = def; d->name; ++d)
-            size += sizeof(unsigned) * d->n_values;
+            size += sizeof(Data) * d->n_values;
         void *data = malloc(size);
         string cfg = client->getConfig();
         load_data(def, data, cfg.c_str());
diff -Naur sim-0.9.3.orig/sim/api/buffer.cpp sim-0.9.3/sim/api/buffer.cpp
--- sim-0.9.3.orig/sim/api/buffer.cpp	2005-08-05 14:19:34.000000000 +0200
+++ sim-0.9.3/sim/api/buffer.cpp	2005-08-05 14:21:23.000000000 +0200
@@ -285,6 +285,9 @@
 
 void Buffer::unpack(unsigned long &c)
 {
+    // FIXME: This needs to be rewritten for 64-bit machines.
+    // Kludge for now.
+    c = 0;
     if (unpack((char*)&c, 4) != 4) c = 0;
     SWAP_L(c);
 }
diff -Naur sim-0.9.3.orig/sim/api/contacts.cpp sim-0.9.3/sim/api/contacts.cpp
--- sim-0.9.3.orig/sim/api/contacts.cpp	2005-08-05 14:19:34.000000000 +0200
+++ sim-0.9.3/sim/api/contacts.cpp	2005-08-05 14:20:25.000000000 +0200
@@ -1331,7 +1331,7 @@
     const DataDef *def = client->protocol()->userDataDef();
     size_t size = 0;
     for (const DataDef *d = def; d->name; ++d)
-        size += sizeof(unsigned) * d->n_values;
+        size += sizeof(Data) * d->n_values;
     data.data = malloc(size);
     load_data(def, data.data, cfg);
     p->push_back(data);
@@ -1344,7 +1344,7 @@
     const DataDef *def = client->protocol()->userDataDef();
     size_t size = 0;
     for (const DataDef *d = def; d->name; ++d)
-        size += sizeof(unsigned) * d->n_values;
+        size += sizeof(Data) * d->n_values;
     data.data = malloc(size);
     load_data(def, data.data, NULL);
     p->push_back(data);
@@ -1560,7 +1560,7 @@
     }
     size_t size = 0;
     for (const DataDef *def = (*it).def; def->name; ++def)
-        size += sizeof(unsigned) * def->n_values;
+        size += sizeof(Data) * def->n_values;
     userData[id] = malloc(size);
     load_data((*it).def, userData[id], NULL);
     return userData[id];
diff -Naur sim-0.9.3.orig/sim/ui/listview.h sim-0.9.3/sim/ui/listview.h
--- sim-0.9.3.orig/sim/ui/listview.h	2005-08-05 14:19:34.000000000 +0200
+++ sim-0.9.3/sim/ui/listview.h	2005-08-05 14:21:01.000000000 +0200
@@ -84,7 +84,7 @@
     static bool canDecode(QMimeSource*);
     static Contact *decode(QMimeSource*);
 protected:
-    unsigned m_id;
+    unsigned long m_id;
 };
 
 #endif
