diff -Naur inetd-1.79s/inetd.c inetd-1.79s-p/inetd.c
--- inetd-1.79s/inetd.c	2004-07-21 21:13:46.000000000 +0200
+++ inetd-1.79s-p/inetd.c	2007-01-02 02:21:08.000000000 +0100
@@ -839,8 +839,8 @@
 		if (sep != 0) {
 			int i;
 
-#define SWAP(type, a, b) {type c=(type)a; (type)a=(type)b; (type)b=(type)c;}
-
+//#define SWAP(type, a, b) {type c=(type)a; (type)a=(type)b; (type)b=(type)c;}
+#define SWAP(type, a, b) {type c=(type)a; a=(type)b; b=(type)c;}
 			omask = sigblock(SIGBLOCK);
 			/*
 			 * sep->se_wait may be holding the pid of a daemon
