diff -Naur time-compat-1.9.2.2/Setup.hs time-compat-1.9.2.2.new/Setup.hs
--- time-compat-1.9.2.2/Setup.hs	1970-01-01 01:00:00.000000000 +0100
+++ time-compat-1.9.2.2.new/Setup.hs	2019-11-13 17:10:01.536473543 +0100
@@ -0,0 +1,2 @@
+import Distribution.Simple
+main = defaultMain
diff -Naur time-compat-1.9.2.2/time-compat.cabal time-compat-1.9.2.2.new/time-compat.cabal
--- time-compat-1.9.2.2/time-compat.cabal	2001-09-09 03:46:40.000000000 +0200
+++ time-compat-1.9.2.2.new/time-compat.cabal	2019-11-13 17:10:19.422593215 +0100
@@ -119,7 +119,7 @@
     , deepseq
     , QuickCheck        >=2.13   && <2.14
     , tagged            >=0.8.6  && <0.9
-    , tasty             >=1.2.1  && <1.3
+    , tasty             >=1  && <1.3
     , tasty-hunit       >=0.10   && <0.11
     , tasty-quickcheck  >=0.10   && <0.11
     , time-compat
