--- cabal-doctest-1.0.9/cabal-doctest.cabal	2001-09-09 03:46:40.000000000 +0200
+++ cabal-doctest-1.0.9.new/cabal-doctest.cabal	2022-11-23 14:53:44.729228980 +0100
@@ -1,5 +1,6 @@
 name:               cabal-doctest
 version:            1.0.9
+x-revision:         2
 synopsis:           A Setup.hs helper for running doctests
 description:
   As of now (end of 2021), there isn't @cabal doctest@
@@ -20,20 +21,21 @@
   README.md
 
 tested-with:
-  GHC == 7.0.4
-  GHC == 7.2.2
-  GHC == 7.4.2
-  GHC == 7.6.3
-  GHC == 7.8.4
-  GHC == 7.10.3
-  GHC == 8.0.2
-  GHC == 8.2.2
-  GHC == 8.4.4
-  GHC == 8.6.5
-  GHC == 8.8.4
+  GHC == 9.4.1
+  GHC == 9.2.4
+  GHC == 9.0.2
   GHC == 8.10.7
-  GHC == 9.0.1
-  GHC == 9.2.1
+  GHC == 8.8.4
+  GHC == 8.6.5
+  GHC == 8.4.4
+  GHC == 8.2.2
+  GHC == 8.0.2
+  GHC == 7.10.3
+  GHC == 7.8.4
+  GHC == 7.6.3
+  GHC == 7.4.2
+  GHC == 7.2.2
+  GHC == 7.0.4
 
 source-repository head
   type:     git
@@ -44,8 +46,8 @@
   other-modules:
   other-extensions:
   build-depends:
-      base       >=4.3  && <4.17
-    , Cabal      >=1.10 && <3.8
+      base       >=4.3  && <5
+    , Cabal      >=1.10 && <3.10
     , directory
     , filepath
 
