diff -Naur VTK-8.1.2/ThirdParty/mpi4py/vtkmpi4py/src/lib-mpi/config/openmpi.h VTK-8.1.2-p/ThirdParty/mpi4py/vtkmpi4py/src/lib-mpi/config/openmpi.h
--- VTK-8.1.2/ThirdParty/mpi4py/vtkmpi4py/src/lib-mpi/config/openmpi.h	2018-11-05 15:28:17.000000000 +0100
+++ VTK-8.1.2-p/ThirdParty/mpi4py/vtkmpi4py/src/lib-mpi/config/openmpi.h	2018-12-13 23:54:09.050250836 +0100
@@ -141,4 +141,13 @@
 
 #endif
 
+#if OMPI_NUMVERSION >= 40000
+#ifndef MPI_LB
+#undef  PyMPI_HAVE_MPI_LB
+#endif
+#ifndef MPI_UB
+#undef  PyMPI_HAVE_MPI_UB
+#endif
+#endif /* OMPI >= 4.0.0 */
+
 #endif /* !PyMPI_CONFIG_OPENMPI_H */
