diff -Naur VTK-9.2.6/ThirdParty/libproj/vtklibproj/src/proj_json_streaming_writer.hpp VTK-9.2.6.new/ThirdParty/libproj/vtklibproj/src/proj_json_streaming_writer.hpp
--- VTK-9.2.6/ThirdParty/libproj/vtklibproj/src/proj_json_streaming_writer.hpp	2023-02-15 05:03:53.000000000 +0100
+++ VTK-9.2.6.new/ThirdParty/libproj/vtklibproj/src/proj_json_streaming_writer.hpp	2023-09-05 21:15:31.418620868 +0200
@@ -33,6 +33,7 @@
 
 #include <vector>
 #include <string>
+#include <cstdint>
 
 #define CPL_DLL
 
