From d9bbc2c4916e7cb8815cd6b9e9e6f579ce59ecf6 Mon Sep 17 00:00:00 2001
From: Julien Schueller <schueller@phimeca.com>
Date: Thu, 29 Jun 2023 09:23:00 +0200
Subject: [PATCH] Filters/OpenTURNS: OPENTURNS_USE_FILE was removed

---
 Filters/OpenTURNS/CMakeLists.txt | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/Filters/OpenTURNS/CMakeLists.txt b/Filters/OpenTURNS/CMakeLists.txt
index 0dcde14e0b4..8bbc0cd41a4 100644
--- a/Filters/OpenTURNS/CMakeLists.txt
+++ b/Filters/OpenTURNS/CMakeLists.txt
@@ -1,6 +1,9 @@
 find_package(OpenTURNS 1.8 REQUIRED)
-# XXX(fixme)
-include(${OPENTURNS_USE_FILE})
+
+if (EXISTS "${OPENTURNS_USE_FILE}")
+  # removed in 1.21
+  include (${OPENTURNS_USE_FILE})
+endif ()
 
 set(classes
   vtkOTDensityMap
