diff -aur mlocate-0.18/Makefile.in fw_mlocate-0.18/Makefile.in
--- mlocate-0.18/Makefile.in	2007-08-25 00:51:49.000000000 +0200
+++ fw_mlocate-0.18/Makefile.in	2007-11-06 14:07:31.000000000 +0100
@@ -410,7 +410,7 @@
 top_srcdir = @top_srcdir@
 dbdir = $(localstatedir)/mlocate
 dbfile = $(dbdir)/mlocate.db
-groupname = mlocate
+groupname = locate
 updatedb_conf = $(sysconfdir)/updatedb.conf
 # autopoint puts its m4 files to the first directory
 ACLOCAL_AMFLAGS = -I m4 -I gnulib/m4
