--- gst-plugins-good/ext/speex/gstspeexdec.c	2007/05/26 15:25:18	1.40
+++ gst-plugins-good/ext/speex/gstspeexdec.c	2008/04/11 10:32:20	1.41
@@ -507,7 +507,7 @@
   if (!dec->header)
     goto no_header;
 
-  if (dec->header->mode >= SPEEX_NB_MODES)
+  if (dec->header->mode >= SPEEX_NB_MODES || dec->header->mode < 0)
     goto mode_too_old;
 
   dec->mode = (SpeexMode *) speex_mode_list[dec->header->mode];

