diff -Naur libdbusmenu-16.04.0/libdbusmenu-glib/Makefile.am libdbusmenu-16.04.0.new/libdbusmenu-glib/Makefile.am
--- libdbusmenu-16.04.0/libdbusmenu-glib/Makefile.am	2016-02-27 19:58:43.000000000 +0100
+++ libdbusmenu-16.04.0.new/libdbusmenu-glib/Makefile.am	2022-01-24 13:41:38.769350172 +0100
@@ -66,7 +66,7 @@
 libdbusmenu_glib_la_CFLAGS = \
 	$(DBUSMENUGLIB_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
-	-Wall -Werror -Wno-error=deprecated-declarations \
+	-Wall  -Wno-error=deprecated-declarations \
 	-DG_LOG_DOMAIN="\"LIBDBUSMENU-GLIB\""
 
 libdbusmenu_glib_la_LIBADD = \
diff -Naur libdbusmenu-16.04.0/libdbusmenu-glib/Makefile.in libdbusmenu-16.04.0.new/libdbusmenu-glib/Makefile.in
--- libdbusmenu-16.04.0/libdbusmenu-glib/Makefile.in	2016-02-27 20:00:13.000000000 +0100
+++ libdbusmenu-16.04.0.new/libdbusmenu-glib/Makefile.in	2022-01-24 13:41:38.769350172 +0100
@@ -498,7 +498,7 @@
 libdbusmenu_glib_la_CFLAGS = \
 	$(DBUSMENUGLIB_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
-	-Wall -Werror -Wno-error=deprecated-declarations \
+	-Wall  -Wno-error=deprecated-declarations \
 	-DG_LOG_DOMAIN="\"LIBDBUSMENU-GLIB\""
 
 libdbusmenu_glib_la_LIBADD = \
diff -Naur libdbusmenu-16.04.0/libdbusmenu-gtk/Makefile.am libdbusmenu-16.04.0.new/libdbusmenu-gtk/Makefile.am
--- libdbusmenu-16.04.0/libdbusmenu-gtk/Makefile.am	2016-02-27 19:58:43.000000000 +0100
+++ libdbusmenu-16.04.0.new/libdbusmenu-gtk/Makefile.am	2022-01-24 13:41:38.769350172 +0100
@@ -66,7 +66,7 @@
 	$(DBUSMENUGTK_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
 	-I$(top_srcdir) \
-	-Wall -Werror -Wno-error=deprecated-declarations \
+	-Wall  -Wno-error=deprecated-declarations \
 	-DG_LOG_DOMAIN="\"LIBDBUSMENU-GTK\""
 
 libdbusmenu_gtk_la_LIBADD = \
diff -Naur libdbusmenu-16.04.0/libdbusmenu-gtk/Makefile.in libdbusmenu-16.04.0.new/libdbusmenu-gtk/Makefile.in
--- libdbusmenu-16.04.0/libdbusmenu-gtk/Makefile.in	2016-02-27 20:00:14.000000000 +0100
+++ libdbusmenu-16.04.0.new/libdbusmenu-gtk/Makefile.in	2022-01-24 13:41:38.769350172 +0100
@@ -507,7 +507,7 @@
 	$(DBUSMENUGTK_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
 	-I$(top_srcdir) \
-	-Wall -Werror -Wno-error=deprecated-declarations \
+	-Wall  -Wno-error=deprecated-declarations \
 	-DG_LOG_DOMAIN="\"LIBDBUSMENU-GTK\""
 
 libdbusmenu_gtk_la_LIBADD = \
