--- libgnomecups-0.2.2.orig/libgnomecups/gnome-cups-request.c
+++ libgnomecups-0.2.2/libgnomecups/gnome-cups-request.c
@@ -349,7 +349,7 @@
 		if (request->response == NULL)
 			status = IPP_INTERNAL_ERROR;
 
-		if (status > IPP_OK_CONFLICT) {
+		if (status > IPP_OK_CONFLICT && status != IPP_NOT_FOUND) {
 			g_warning ("IPP request failed with status %d", status);
 			if (request->error != NULL)
 				*(request->error) = g_error_new (GNOME_CUPS_ERROR, 
--- libgnomecups-0.2.2.orig/libgnomecups/gnome-cups-printer.c
+++ libgnomecups-0.2.2/libgnomecups/gnome-cups-printer.c
@@ -510,6 +510,10 @@
 	num_dests = cups_get_dests (filename, num_dests, dests);
 	g_free (filename);
 
+	filename = g_build_filename (g_get_home_dir (), ".cups", "lpoptions", NULL);
+	num_dests = cups_get_dests (filename, num_dests, dests);
+	g_free (filename);
+
 	return num_dests;
 }
 
@@ -976,8 +980,11 @@
 {
 	g_return_val_if_fail (GNOME_CUPS_IS_PRINTER (printer), FALSE);
 
-	return (printer->details->device_uri != NULL) && 
-		(strcmp (printer->details->device_uri, "") != 0);
+	return printer->details->device_uri != NULL && 
+		strncmp(printer->details->device_uri, "smb:", 4) != 0 &&
+		strncmp(printer->details->device_uri, "http:", 5) != 0 &&
+		strncmp(printer->details->device_uri, "https:", 5) != 0 &&
+		strncmp(printer->details->device_uri, "ipp:", 4)  != 0;
 }
 
 void
@@ -1237,6 +1244,9 @@
 	
 }
 
+/* Define the CUPS-Add-Modify-Printer, see http://www.cups.org/documentation.php/spec-ipp.html#CUPS_ADD_MODIFY_PRINTER */
+#define CUPS_ADD_MODIFY_PRINTER 0x4003
+
 void
 gnome_cups_printer_set_description (GnomeCupsPrinter *printer,
 				    const char *description,
@@ -1252,7 +1262,9 @@
 		return;
 	}
 
-	request = gnome_cups_request_new_for_printer (IPP_SET_PRINTER_ATTRIBUTES,
+	/* As read in http://lists.samba.org/archive/samba-technical/2003-February/027044.html 
+	CUPS does not currently support IPP_SET_PRINTER_ATTRIBUTES, so a  is used */
+	request = gnome_cups_request_new_for_printer (CUPS_ADD_MODIFY_PRINTER,
 						      printer);
 	ippAddString (request, IPP_TAG_PRINTER, IPP_TAG_TEXT,
 		      "printer-info", NULL, description);
@@ -1283,9 +1295,9 @@
 	if (!strcmp (location, printer->details->location)) {
 		return;
 	}
-
+	/* Same as above (IPP_SET_PRINTER_ATTRIBUTES replaced by CUPS-Add-Modify-Printer ) */
 	request = gnome_cups_request_new_for_printer (
-		IPP_SET_PRINTER_ATTRIBUTES, printer);
+		CUPS_ADD_MODIFY_PRINTER, printer);
 	ippAddString (request, IPP_TAG_PRINTER, IPP_TAG_TEXT,
 		"printer-location", NULL, location);
 	response = gnome_cups_request_execute (request, NULL, "/admin/", error);
