diff -Naur udev-038.orig/Makefile udev-038/Makefile
--- udev-038.orig/Makefile	2004-10-14 01:50:21.000000000 +0200
+++ udev-038/Makefile	2004-10-15 22:54:58.000000000 +0200
@@ -362,7 +362,7 @@
 	@echo "Built $(RELEASE_NAME).tar.gz"
 
 
-install-initscript: etc/init.d/udev etc/init.d/udev.debian etc/init.d/udev.init.LSB
+install-initscript: etc/init.d/udev etc/init.d/udev.init.LSB
 	@if [ -f /etc/redhat-release ]; then \
 		$(INSTALL_DATA) -D etc/init.d/udev $(DESTDIR)$(initdir)/udev; \
 	elif [ -f /etc/SuSE-release ]; then \
