--- ocaml-3.08.3/otherlibs/num/bng_ia32.c.x86	2005-05-09 15:24:29.000000000 -0400
+++ ocaml-3.08.3/otherlibs/num/bng_ia32.c	2005-05-09 22:02:50.000000000 -0400
@@ -121,7 +121,7 @@
         "leal 4(%1), %1 \n\t"
         "decl %2 \n\t"
         "jnz 1b"
-        : "+&r" (a), "+&r" (b), "+&rm" (blen), "+&r" (out)
+        : "+&r" (a), "+&r" (b), "+m" (blen), "+&r" (out)
         : "rm" (d)
         : "eax", "edx");
   }
@@ -164,7 +164,7 @@
         "leal 4(%1), %1 \n\t"
         "decl %2 \n\t"
         "jnz 1b"
-        : "+&r" (a), "+&r" (b), "+&rm" (blen), "+&rm" (out), "=&r" (tmp)
+        : "+&r" (a), "+&r" (b), "+m" (blen), "+m" (out), "=&r" (tmp)
         : "rm" (d)
         : "eax", "edx");
   }
