diff --git a/libdha/sysdep/pci_linux.c b/libdha/sysdep/pci_linux.c
index 2848619..f81094e 100644
--- a/libdha/sysdep/pci_linux.c
+++ b/libdha/sysdep/pci_linux.c
@@ -6,10 +6,6 @@
 #include <errno.h>
 #ifdef __i386__
 #include <sys/perm.h>
-#else
-#ifndef __sparc__
-#include <sys/io.h>
-#endif
 #endif
 
 #include "config.h"
diff --git a/libdha/sysdep/pci_powerpc.c b/libdha/sysdep/pci_powerpc.c
index 667b4db..581b476 100644
--- a/libdha/sysdep/pci_powerpc.c
+++ b/libdha/sysdep/pci_powerpc.c
@@ -7,7 +7,6 @@
 static int pci_config_type( void ) { return 1; }
 #ifdef linux
 #include <fcntl.h>
-#include <sys/io.h>
 #include <linux/pci.h>
 #include "../../bswap.h"
 #endif
