diff -Naur hardinfo-0.4.2.3/configure hardinfo-0.4.2.3.new/configure
--- hardinfo-0.4.2.3/configure	2007-11-04 18:44:47.000000000 +0530
+++ hardinfo-0.4.2.3.new/configure	2008-03-01 23:21:40.000000000 +0530
@@ -56,7 +56,8 @@
 	x86_64)
 		ln -sf linux/x86_64 arch/this
 		ARCH="ARCH_x86_64"
-		LIBDIR="/usr/lib64" ;;
+		LIBDIR="/usr/lib"
+		PROC="k8" ;;
 	mips*)
 		ln -sf linux/mips arch/this
 		ARCH="ARCH_MIPS" ;;
@@ -215,7 +216,7 @@
 echo "SOUP_LIBS = ${SOUP_LIBS}" >> Makefile
 echo "SOUP_CFLAGS = ${SOUP_FLAGS}" >> Makefile
 echo "PACKAGE = `basename ${PWD}`" >> Makefile
-echo "ARCHOPTS = " >> Makefile
+echo "ARCHOPTS = -march=${PROC} -O2 -pipe " >> Makefile
 echo "LIBDIR = $LIBDIR" >> Makefile
 
 cat Makefile.in >> Makefile
diff -Naur hardinfo-0.4.2.3/Makefile.in hardinfo-0.4.2.3.new/Makefile.in
--- hardinfo-0.4.2.3/Makefile.in	2007-11-04 18:44:47.000000000 +0530
+++ hardinfo-0.4.2.3.new/Makefile.in	2008-03-01 23:20:05.000000000 +0530
@@ -1,7 +1,7 @@
 
 CC = gcc $(ARCHOPTS) -g
 CCSLOW = gcc -O0 -g
-CFLAGS = -fPIC -pipe -Wall -g $(GTK_CFLAGS) $(GLADE_CFLAGS) $(SOUP_CFLAGS) -I.
+CFLAGS = -fPIC -Wall -g $(GTK_CFLAGS) $(GLADE_CFLAGS) $(SOUP_CFLAGS) -I.
 
 # ----------------------------------------------------------------------------
 
