--- wbxml2-0.9.2old/configure.in	2006-07-11 13:47:44.000000000 +0200
+++ wbxml2-0.9.2new/configure.in	2008-08-02 09:37:16.000000000 +0200
@@ -1,11 +1,11 @@
 dnl  Process this file with autoconf to produce a configure script.
 AC_INIT(doxygen.h)
-AM_INIT_AUTOMAKE(wbxml2, 0.9.0)
+AM_INIT_AUTOMAKE(wbxml2, 0.9.2)
 AM_CONFIG_HEADER(config.h)
 
 
 dnl Define common variables
-VERSION="0.9.0"
+VERSION="0.9.2"
 AC_SUBST(VERSION)
 RELEASE="1"
 AC_SUBST(RELEASE)
