diff -Naur firefox-63.0/third_party/prio/include/mprio.h firefox-63.0-p/third_party/prio/include/mprio.h
--- firefox-63.0/third_party/prio/include/mprio.h	2018-10-18 22:06:15.000000000 +0200
+++ firefox-63.0-p/third_party/prio/include/mprio.h	2018-10-26 10:41:06.210080440 +0200
@@ -14,9 +14,9 @@
 #endif
 
 #include <msgpack.h>
-#include <nss/blapit.h>
-#include <nss/pk11pub.h>
-#include <nss/seccomon.h>
+#include <nss3/blapit.h>
+#include <nss3/pk11pub.h>
+#include <nss3/seccomon.h>
 #include <stdbool.h>
 #include <stddef.h>
 
diff -Naur firefox-63.0/third_party/prio/prio/encrypt.c firefox-63.0-p/third_party/prio/prio/encrypt.c
--- firefox-63.0/third_party/prio/prio/encrypt.c	2018-10-18 22:06:14.000000000 +0200
+++ firefox-63.0-p/third_party/prio/prio/encrypt.c	2018-10-26 10:41:06.238080613 +0200
@@ -6,9 +6,9 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. 
  */
 
-#include <nss/keyhi.h>
-#include <nss/keythi.h>
-#include <nss/pk11pub.h>
+#include <nss3/keyhi.h>
+#include <nss3/keythi.h>
+#include <nss3/pk11pub.h>
 #include <prerror.h>
 
 #include "encrypt.h"
diff -Naur firefox-63.0/third_party/prio/prio/prg.c firefox-63.0-p/third_party/prio/prio/prg.c
--- firefox-63.0/third_party/prio/prio/prg.c	2018-10-18 22:06:14.000000000 +0200
+++ firefox-63.0-p/third_party/prio/prio/prg.c	2018-10-26 10:41:06.222080514 +0200
@@ -7,8 +7,8 @@
  */
 
 #include <mprio.h>
-#include <nss/blapit.h>
-#include <nss/pk11pub.h>
+#include <nss3/blapit.h>
+#include <nss3/pk11pub.h>
 #include <string.h>
 
 #include "prg.h"
diff -Naur firefox-63.0/third_party/prio/prio/prg.h firefox-63.0-p/third_party/prio/prio/prg.h
--- firefox-63.0/third_party/prio/prio/prg.h	2018-10-18 22:06:14.000000000 +0200
+++ firefox-63.0-p/third_party/prio/prio/prg.h	2018-10-26 10:41:06.245080657 +0200
@@ -10,7 +10,7 @@
 #define __PRG_H__
 
 #include <mpi.h>
-#include <nss/blapit.h>
+#include <nss3/blapit.h>
 #include <stdlib.h>
 
 #include "config.h"
diff -Naur firefox-63.0/third_party/prio/prio/rand.c firefox-63.0-p/third_party/prio/prio/rand.c
--- firefox-63.0/third_party/prio/prio/rand.c	2018-10-18 22:06:14.000000000 +0200
+++ firefox-63.0-p/third_party/prio/prio/rand.c	2018-10-26 10:41:06.261080756 +0200
@@ -8,8 +8,8 @@
 
 #include <limits.h>
 #include <mprio.h>
-#include <nss/nss.h>
-#include <nss/pk11pub.h>
+#include <nss3/nss.h>
+#include <nss3/pk11pub.h>
 #include <nspr/prinit.h>
 
 #include "debug.h"
diff -Naur firefox-63.0/third_party/prio/prio/rand.h firefox-63.0-p/third_party/prio/prio/rand.h
--- firefox-63.0/third_party/prio/prio/rand.h	2018-10-18 22:06:14.000000000 +0200
+++ firefox-63.0-p/third_party/prio/prio/rand.h	2018-10-26 10:41:06.250080688 +0200
@@ -10,7 +10,7 @@
 #define __RAND_H__
 
 #include <mpi.h>
-#include <nss/seccomon.h>
+#include <nss3/seccomon.h>
 #include <stdlib.h>
 
 /*
diff -Naur firefox-63.0/third_party/prio/ptest/encrypt_test.c firefox-63.0-p/third_party/prio/ptest/encrypt_test.c
--- firefox-63.0/third_party/prio/ptest/encrypt_test.c	2018-10-18 22:06:15.000000000 +0200
+++ firefox-63.0-p/third_party/prio/ptest/encrypt_test.c	2018-10-26 10:41:06.287080917 +0200
@@ -7,11 +7,11 @@
  */
 
 #include <nspr.h>
-#include <nss/nss.h>
-#include <nss/secoidt.h>
-#include <nss/keyhi.h>
-#include <nss/pk11pub.h>
-#include <nss/cert.h>
+#include <nss3/nss.h>
+#include <nss3/secoidt.h>
+#include <nss3/keyhi.h>
+#include <nss3/pk11pub.h>
+#include <nss3/cert.h>
 
 #include "mutest.h"
 #include "prio/encrypt.h"
