diff --git a/testing/profiles/profileserver/user.js b/testing/profiles/profileserver/user.js
index b1a46af9821c8..f8f6e7a9cac84 100644
--- a/testing/profiles/profileserver/user.js
+++ b/testing/profiles/profileserver/user.js
@@ -9,3 +9,5 @@ user_pref("dom.timeout.enable_budget_timer_throttling", false);
 // TODO: Bug 1795750 - Re-enable this pref when we have a new version of the
 // Quitter XPI with a simpler version format.
 user_pref("extensions.webextensions.warnings-as-errors", false);
+// Bug XXX: Snap with PGO crashes on Mesa
+user_pref("gfx.webrender.software", true);
