diff -ur a/src/smi_driver.c b/src/smi_driver.c
--- a/src/smi_driver.c	2012-07-16 23:53:21.000000000 -0500
+++ b/src/smi_driver.c	2013-07-12 12:44:25.601285225 -0500
@@ -1750,7 +1750,6 @@
 		   "Done writing mode.  Register dump:\n");
     SMI_PrintRegs(pScrn);
 
-    miInitializeBackingStore(pScreen);
 
 #ifdef HAVE_XMODES
     xf86DiDGAInit(pScreen, (unsigned long)(pSmi->FBBase + pScrn->fbOffset));
