diff -Naur ffmpeg-0.4.9-pre1.orig/ffmpeg-config.1 ffmpeg-0.4.9-pre1/ffmpeg-config.1
--- ffmpeg-0.4.9-pre1.orig/ffmpeg-config.1	1970-01-01 01:00:00.000000000 +0100
+++ ffmpeg-0.4.9-pre1/ffmpeg-config.1	2005-08-07 20:19:12.000000000 +0200
@@ -0,0 +1,50 @@
+.TH ffmpeg-config 1 "2004-07-16" "ffmpeg"
+.SH NAME
+ffmpeg-config \- script to get information about the installed version of ffmpeg
+.SH SYNOPSIS
+.B ffmpeg-config
+[ --prefix=
+.IR DIR ]
+[ --version ] [ --libs [<extensions>]] [ --plugin-libs [<extensions>]] [ --cflags ]
+.SH DESCRIPTION
+.B ffmpeg-config
+is a tool that is used to configure and determine the compiler and linker
+flags that should be used to compile and link progams, libraries, and
+plugins that use ffmpeg.
+.SH OPTIONS
+.TP
+.B --cflags
+Print the C compiler flags that are necessary to compile a program or library
+that uses ffmpeg.
+.TP
+.B --libs { avcodec | avformat | postproc }
+Print the linker flags that are necessary to link a program that uses
+libavcodec, libavformat or libpostproc.
+.TP
+.B --plugin-libs { avcodec | avformat | postproc }
+Print the linker flags that are necessary to link a shared object (such as a
+plugin for a program) that uses libavcodec, libavformat or libpostproc.
+.TP
+.BI --prefix= DIR
+If specified, use PREFIX instead of the installation prefix that ffmpeg
+was built with when computing the output for the --cflags, --libs and
+--plugin-libs options. This option must be specified before any of the
+--cflags, --libs and --plugin-libs options.
+.TP
+.B --version
+Prints the currently installed version of ffmpeg on standard output.
+.SH EXAMPLES
+.TP
+gcc -o main.o $(ffmpeg-config --cflags) -c main.c
+is how you might use
+.B ffmpeg-config
+to compile a C source file for an executable program.
+.TP
+gcc -o my_app $(ffmpeg-config --libs avcodec) main.o util.o
+is how you might use
+.B ffmpeg-config
+to link compiled objects into an executable program that uses libavcodec.
+.SH AUTHOR
+This manual page was written for sdl-config by Branden Robinson, originally
+for Progeny Linux Systems, Inc., and the Debian Project. It was adapted to
+ffmpeg by Sam Hocevar.
diff -Naur ffmpeg-0.4.9-pre1.orig/ffmpeg-config.in ffmpeg-0.4.9-pre1/ffmpeg-config.in
--- ffmpeg-0.4.9-pre1.orig/ffmpeg-config.in	1970-01-01 01:00:00.000000000 +0100
+++ ffmpeg-0.4.9-pre1/ffmpeg-config.in	2005-08-07 20:22:16.000000000 +0200
@@ -0,0 +1,105 @@
+#!/bin/sh
+
+# ffmpeg-config
+
+prefix="/usr"
+exec_prefix="/usr/bin"
+version="@VERSION@"
+
+include_dir="/usr/include/ffmpeg"
+lib_dir="/usr/lib"
+
+link_libs=""
+
+usage()
+{
+    cat <<EOF
+Usage: ffmpeg-config [OPTIONS]
+Options:
+    [--prefix]
+    [--cflags]
+    [--libs [avcodec] [avformat] [postproc]]
+    [--plugin-libs [avcodec] [avformat] [postproc]]
+    [--version]
+EOF
+    exit $1
+}
+
+if test $# -eq 0; then
+    usage 1 1>&2
+fi
+
+while test $# -gt 0; do
+	case $1 in
+    --prefix)
+	    echo_prefix=yes
+	    ;;
+
+	--cflags)
+	    echo_cflags=yes
+	    ;;
+        
+	--libs)
+	    echo_libs=yes
+	    ;;
+
+	--plugin-libs)
+	    echo_plugin_libs=yes
+	    ;;
+
+	--version)
+	    echo_version=yes
+	    ;;
+
+	avcodec|avformat|postproc)
+	    if test "$echo_libs" = "yes"; then
+		link_libs="$link_libs -l$1"
+		if test "$1" != "avcodec"; then
+		    link_libs="$link_libs -lavcodec"
+		fi
+	    elif test "$echo_plugin_libs" = "yes"; then
+		link_plugin_libs="$link_plugin_libs -l${1}"
+		if test "$1" != "avcodec"; then
+		    link_plugin_libs="$link_plugin_libs -lavcodec"
+		fi
+	    else
+		usage 1 1>&2
+	    fi
+	    ;;
+
+	*)
+	    usage 1 1>&2
+	    ;;
+    esac
+  shift
+done
+
+if test "$echo_prefix" = "yes"; then
+    echo $prefix
+fi
+
+cflags="-I$include_dir"
+link_libs="$link_libs -lvorbis -lvorbisenc -ltheora -logg -la52 -lraw1394 -ldc1394_control -lz -lm"
+link_plugin_libs="$link_plugin_libs -lvorbis -lvorbisenc -ltheora -logg -la52 -lraw1394 -ldc1394_control -lz -lm"
+
+if test "$lib_dir" != "/usr/lib"; then
+    libs="-L$lib_dir"
+else
+    libs=""
+fi
+
+if test "$echo_cflags" = "yes"; then
+    echo $cflags
+fi
+
+if test "$echo_libs" = "yes"; then
+    echo $libs $link_libs
+fi
+
+if test "$echo_plugin_libs" = "yes"; then
+    echo $libs $link_plugin_libs
+fi
+
+if test "$echo_version" = "yes"; then
+    echo $version
+fi
