diff -Naur ffmpeg-0.4.9-pre1.orig/libavcodec/Makefile ffmpeg-0.4.9-pre1/libavcodec/Makefile
--- ffmpeg-0.4.9-pre1.orig/libavcodec/Makefile	2005-06-02 09:37:57.000000000 +0000
+++ ffmpeg-0.4.9-pre1/libavcodec/Makefile	2005-06-02 09:46:09.000000000 +0000
@@ -68,8 +68,9 @@
 
 # using builtin liba52 or runtime linked liba52.so.0
 ifneq ($(CONFIG_A52BIN),yes)
-OBJS+= liba52/bit_allocate.o liba52/bitstream.o liba52/downmix.o \
-	liba52/imdct.o  liba52/parse.o liba52/crc.o liba52/resample.o
+#OBJS+= liba52/bit_allocate.o liba52/bitstream.o liba52/downmix.o \
+#	liba52/imdct.o  liba52/parse.o liba52/crc.o liba52/resample.o
+EXTRALIBS += -la52
 endif
 endif
 
diff -Naur ffmpeg-0.4.9-pre1.orig/libavcodec/a52dec.c ffmpeg-0.4.9-pre1/libavcodec/a52dec.c
--- ffmpeg-0.4.9-pre1.orig/libavcodec/a52dec.c	2005-06-02 09:37:57.000000000 +0000
+++ ffmpeg-0.4.9-pre1/libavcodec/a52dec.c	2005-06-02 09:46:21.000000000 +0000
@@ -23,7 +23,7 @@
  */
 
 #include "avcodec.h"
-#include "liba52/a52.h"
+#include <a52dec/a52.h>
 
 #ifdef CONFIG_A52BIN
 #include <dlfcn.h>
--- ffmpeg-20060214/libavcodec/Makefile.orig	2006-02-14 14:29:13.000000000 +0100
+++ ffmpeg-20060214/libavcodec/Makefile	2006-02-14 14:30:39.000000000 +0100
@@ -302,6 +302,7 @@
 
 ifeq ($(CONFIG_X264),yes)
 OBJS+= x264.o
+EXTRALIBS+=-lpthread
 endif
 
 ifeq ($(CONFIG_MP3LAME),yes)
