--- freenx-client-0.9/nxcl/lib/notQt.cpp~	2008-03-11 00:13:08.000000000 +0100
+++ freenx-client-0.9/nxcl/lib/notQt.cpp	2008-04-05 20:36:05.000000000 +0200
@@ -19,6 +19,7 @@
 
 #include <iostream>
 #include <sstream>
+#include <cstdlib>
 extern "C" {
 #include <unistd.h>
 #include <errno.h>
--- freenx-client-0.9/nxcl/lib/nxclientlib.cpp~	2008-03-11 00:13:08.000000000 +0100
+++ freenx-client-0.9/nxcl/lib/nxclientlib.cpp	2008-04-05 20:54:18.000000000 +0200
@@ -27,6 +27,7 @@
 #include "../config.h"
 
 #include <fstream>
+#include <cstdlib>
 
 extern "C" {
     #include <errno.h>
--- freenx-client-0.9/nxcl/lib/nxclientlib.cpp~	2008-04-05 20:54:18.000000000 +0200
+++ freenx-client-0.9/nxcl/lib/nxclientlib.cpp	2008-04-06 06:37:37.000000000 +0200
@@ -28,6 +28,7 @@
 
 #include <fstream>
 #include <cstdlib>
+#include <cstring>
 
 extern "C" {
     #include <errno.h>
--- freenx-client-0.9/nxcl/nxcl/nxcl.cpp~	2008-03-11 00:13:08.000000000 +0100
+++ freenx-client-0.9/nxcl/nxcl/nxcl.cpp	2008-04-06 06:48:34.000000000 +0200
@@ -20,6 +20,7 @@
 #include "nxclientlib_i18n.h"
 #include "nxclientlib.h"
 #include <fstream>
+#include <cstdlib>
 
 #include "nxcl.h"
 
