Index: qtnx-0.9/qtnxwindow.cpp
===================================================================
--- qtnx-0.9.orig/qtnxwindow.cpp	2008-03-14 12:25:00.638618439 +0000
+++ qtnx-0.9/qtnxwindow.cpp	2008-03-14 12:25:34.639487284 +0000
@@ -290,7 +290,7 @@
 
     m_NXClient->setDepth(getDepth());
 
-    QString keyPath = "id.key";
+    QString keyPath = "/usr/share/qtnx/id.key";
 
 #ifdef Q_WS_MAC
     keyPath = binaryPath + "/id.key";
