--- dhcdbd-3.0/src/dbus_service.c.orig	2007-09-28 18:09:03.000000000 +0200
+++ dhcdbd-3.0/src/dbus_service.c	2007-09-28 18:10:58.000000000 +0200
@@ -33,6 +33,12 @@
 
 #include "dbus_service.h"
 
+#if (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MINOR == 1 && DBUS_VERSION_MICRO >= 1) || (DBUS_VERSION_MAJOR == 1 && DBUS_VERSION_MAJOR > 1) || (DBUS_VERSION_MAJOR > 1)
+#define dbus_watch_get_unix_fd dbus_watch_get_unix_fd
+#else
+#define dbus_watch_get_unix_fd dbus_watch_get_fd
+#endif
+
 typedef struct dbcs_s {
     DBusConnection *connection;
     DBusDispatchStatus dispatchStatus;
