diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-10.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-10.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-10.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-10.d	2019-05-19 02:41:25.495406466 +0200
@@ -32,7 +32,8 @@
 hook called: claim_file tmpdir/libtext.a \[@.* not claimed
 #...
 hook called: all symbols read.
-Sym: '_?func' Resolution: LDPR_PREVAILING_DEF
+Sym: '_?func' Resolution: LDPR_PREVAILING_DEF_IRONLY
 Sym: '_?func2' Resolution: LDPR_PREVAILING_DEF_IRONLY
+#...
 hook called: cleanup.
 #...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-11.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-11.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-11.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-11.d	2019-05-19 02:41:25.495406466 +0200
@@ -35,8 +35,9 @@
 hook called: claim_file tmpdir/libtext.a \[@.* CLAIMED
 #...
 hook called: all symbols read.
-Sym: '_?func' Resolution: LDPR_PREVAILING_DEF
+Sym: '_?func' Resolution: LDPR_PREVAILING_DEF_IRONLY
 Sym: '_?func2' Resolution: LDPR_PREVAILING_DEF_IRONLY
-Sym: '_?text' Resolution: LDPR_PREVAILING_DEF
+Sym: '_?text' Resolution: LDPR_PREVAILING_DEF_IRONLY
+#...
 hook called: cleanup.
 #...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-12.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-12.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-12.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-12.d	2019-05-19 02:41:25.495406466 +0200
@@ -1,6 +1,6 @@
 #...
-.*: symbol `func' definition: 0, visibility: 0, resolution: 2
-.*: symbol `func1' definition: 0, visibility: 1, resolution: 3
-.*: symbol `func2' definition: 0, visibility: 2, resolution: 3
-.*: symbol `func3' definition: 0, visibility: 3, resolution: 3
+.*: symbol `_?func' definition: 0, visibility: 0, resolution: .
+.*: symbol `_?func1' definition: 0, visibility: 1, resolution: 3
+.*: symbol `_?func2' definition: 0, visibility: 2, resolution: 3
+.*: symbol `_?func3' definition: 0, visibility: 3, resolution: 3
 #pass
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-13.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-13.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-13.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-13.d	2019-05-19 02:41:25.492406451 +0200
@@ -23,5 +23,3 @@
 hook called: claim_file .*/ld/testsuite/ld-plugin/func.c \[@0/.* CLAIMED
 hook called: claim_file tmpdir/text.o \[@0/.* not claimed
 #...
-.*main.c.*: undefined reference to `\.?func'
-#...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-14.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-14.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-14.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-14.d	2019-05-19 02:41:25.492406451 +0200
@@ -27,7 +27,6 @@
 hook called: claim_file tmpdir/text.o \[@0/.* not claimed
 #...
 hook called: all symbols read.
-.*: tmpdir/main.o: in function `main':
-.*main.c.*: undefined reference to `\.?func'
+#...
 hook called: cleanup.
 #...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-15.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-15.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-15.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-15.d	2019-05-19 02:41:25.492406451 +0200
@@ -28,7 +28,6 @@
 hook called: claim_file tmpdir/text.o \[@0/.* not claimed
 #...
 hook called: all symbols read.
-.*: tmpdir/main.o: in function `main':
-.*main.c.*: undefined reference to `\.?func'
+#...
 hook called: cleanup.
 #...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-16.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-16.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-16.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-16.d	2019-05-19 02:41:25.492406451 +0200
@@ -30,9 +30,8 @@
 hook called: claim_file tmpdir/text.o \[@0/.* not claimed
 #...
 hook called: all symbols read.
-Sym: '_?func' Resolution: LDPR_PREVAILING_DEF
+Sym: '_?func' Resolution: LDPR_PREVAILING_DEF_IRONLY
 Sym: '_?func2' Resolution: LDPR_PREVAILING_DEF_IRONLY
-.*: tmpdir/main.o: in function `main':
-.*main.c.*: undefined reference to `\.?func'
+#...
 hook called: cleanup.
 #...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-17.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-17.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-17.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-17.d	2019-05-19 02:41:25.493406456 +0200
@@ -31,7 +31,8 @@
 hook called: claim_file tmpdir/text.o \[@0/.* not claimed
 #...
 hook called: all symbols read.
-Sym: '_?func' Resolution: LDPR_PREVAILING_DEF
+Sym: '_?func' Resolution: LDPR_PREVAILING_DEF_IRONLY
 Sym: '_?func2' Resolution: LDPR_PREVAILING_DEF_IRONLY
+#...
 hook called: cleanup.
 #...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-18.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-18.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-18.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-18.d	2019-05-19 02:41:25.495406466 +0200
@@ -32,7 +32,8 @@
 hook called: claim_file tmpdir/libtext.a \[@.* not claimed
 #...
 hook called: all symbols read.
-Sym: '_?func' Resolution: LDPR_PREVAILING_DEF
+Sym: '_?func' Resolution: LDPR_PREVAILING_DEF_IRONLY
 Sym: '_?func2' Resolution: LDPR_PREVAILING_DEF_IRONLY
+#...
 hook called: cleanup.
 #...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-19.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-19.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-19.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-19.d	2019-05-19 02:41:25.495406466 +0200
@@ -35,8 +35,9 @@
 hook called: claim_file tmpdir/libtext.a \[@.* CLAIMED
 #...
 hook called: all symbols read.
-Sym: '_?func' Resolution: LDPR_PREVAILING_DEF
+Sym: '_?func' Resolution: LDPR_PREVAILING_DEF_IRONLY
 Sym: '_?func2' Resolution: LDPR_PREVAILING_DEF_IRONLY
-Sym: '_?text' Resolution: LDPR_PREVAILING_DEF
+Sym: '_?text' Resolution: LDPR_PREVAILING_DEF_IRONLY
+#...
 hook called: cleanup.
 #...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-20.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-20.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-20.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-20.d	2019-05-19 02:41:25.493406456 +0200
@@ -2,6 +2,5 @@
 Input: func.c \(tmpdir/libfunc.a\)
 Sym: '_?func' Resolution: LDPR_PREVAILING_DEF.*
 Sym: '_?func' Resolution: LDPR_PREVAILING_DEF.*
-.*: tmpdir/main.o: in function `main':
-.*main.c.*: undefined reference to `\.?func'
+#...
 hook called: cleanup.
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-21.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-21.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-21.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-21.d	2019-05-19 02:41:25.493406456 +0200
@@ -2,6 +2,5 @@
 Input: .*/ld/testsuite/ld-plugin/func.c \(.*/ld/testsuite/ld-plugin/func.c\)
 Sym: '_?func' Resolution: LDPR_PREVAILING_DEF.*
 Sym: '_?func' Resolution: LDPR_PREVAILING_DEF.*
-.*: tmpdir/main.o: in function `main':
-.*main.c.*: undefined reference to `\.?func'
+#...
 hook called: cleanup.
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-22.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-22.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-22.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-22.d	2019-05-19 02:41:25.493406456 +0200
@@ -2,6 +2,5 @@
 hook called: all symbols read.
 Sym: '_?func' Resolution: LDPR_PREVAILING_DEF.*
 Sym: '_?func' Resolution: LDPR_PREVAILING_DEF.*
-.*: tmpdir/main.o: in function `main':
-.*main.c.*: undefined reference to `\.?func'
+#...
 hook called: cleanup.
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-23.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-23.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-23.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-23.d	2019-05-19 02:41:25.493406456 +0200
@@ -2,6 +2,5 @@
 hook called: all symbols read.
 Sym: '_?func' Resolution: LDPR_PREVAILING_DEF.*
 Sym: '_?func' Resolution: LDPR_PREVAILING_DEF.*
-.*: tmpdir/main.o: in function `main':
-.*main.c.*: undefined reference to `\.?func'
+#...
 hook called: cleanup.
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-24.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-24.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-24.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-24.d	2019-05-19 02:41:25.493406456 +0200
@@ -2,4 +2,5 @@
 Input: .*/ld/testsuite/ld-plugin/func.c \(.*/ld/testsuite/ld-plugin/func.c\)
 Sym: '_?func' Resolution: LDPR_PREVAILING_DEF.*
 Sym: '_?func' Resolution: LDPR_PREVAILING_DEF.*
+#...
 hook called: cleanup.
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-25.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-25.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-25.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-25.d	2019-05-19 02:41:25.493406456 +0200
@@ -2,4 +2,5 @@
 hook called: all symbols read.
 Sym: '_?func' Resolution: LDPR_PREVAILING_DEF.*
 Sym: '_?func' Resolution: LDPR_PREVAILING_DEF.*
+#...
 hook called: cleanup.
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-28.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-28.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-28.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-28.d	2019-05-19 02:41:25.495406466 +0200
@@ -1 +1,2 @@
 .*: error: Error
+#...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-29.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-29.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-29.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-29.d	2019-05-19 02:41:25.493406456 +0200
@@ -1 +1,2 @@
 .*: warning: Warning
+#...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-30.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-30.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-30.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-30.d	2019-05-19 02:41:25.493406456 +0200
@@ -24,3 +24,4 @@
 hook called: claim_file tmpdir/func.o \[@0/.* not claimed
 hook called: claim_file tmpdir/text.o \[@0/.* not claimed
 hook called: claim_file tmpdir/libempty.a \[@.* not claimed
+#pass
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-6.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-6.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-6.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-6.d	2019-05-19 02:41:25.494406461 +0200
@@ -27,7 +27,6 @@
 hook called: claim_file tmpdir/text.o \[@0/.* not claimed
 #...
 hook called: all symbols read.
-.*: tmpdir/main.o: in function `main':
-.*main.c.*: undefined reference to `\.?func'
+#...
 hook called: cleanup.
 #...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-7.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-7.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-7.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-7.d	2019-05-19 02:41:25.494406461 +0200
@@ -28,7 +28,6 @@
 hook called: claim_file tmpdir/text.o \[@0/.* not claimed
 #...
 hook called: all symbols read.
-.*: tmpdir/main.o: in function `main':
-.*main.c.*: undefined reference to `\.?func'
+#...
 hook called: cleanup.
 #...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-8.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-8.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-8.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-8.d	2019-05-19 02:41:25.495406466 +0200
@@ -30,9 +30,8 @@
 hook called: claim_file tmpdir/text.o \[@0/.* not claimed
 #...
 hook called: all symbols read.
-Sym: '_?func' Resolution: LDPR_PREVAILING_DEF
+Sym: '_?func' Resolution: LDPR_PREVAILING_DEF_IRONLY
 Sym: '_?func2' Resolution: LDPR_PREVAILING_DEF_IRONLY
-.*: tmpdir/main.o: in function `main':
-.*main.c.*: undefined reference to `\.?func'
+#...
 hook called: cleanup.
 #...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/plugin-9.d binutils-2.32-p/ld/testsuite/ld-plugin/plugin-9.d
--- binutils-2.32/ld/testsuite/ld-plugin/plugin-9.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/plugin-9.d	2019-05-19 02:41:25.494406461 +0200
@@ -31,7 +31,8 @@
 hook called: claim_file tmpdir/text.o \[@0/.* not claimed
 #...
 hook called: all symbols read.
-Sym: '_?func' Resolution: LDPR_PREVAILING_DEF
+Sym: '_?func' Resolution: LDPR_PREVAILING_DEF_IRONLY
 Sym: '_?func2' Resolution: LDPR_PREVAILING_DEF_IRONLY
+#...
 hook called: cleanup.
 #...
diff -Naur binutils-2.32/ld/testsuite/ld-plugin/pr20070.d binutils-2.32-p/ld/testsuite/ld-plugin/pr20070.d
--- binutils-2.32/ld/testsuite/ld-plugin/pr20070.d	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-plugin/pr20070.d	2019-05-19 02:41:25.494406461 +0200
@@ -5,5 +5,6 @@
 Sym: 'undef' Resolution: LDPR_UNDEF
 Sym: 'weakundef' Resolution: LDPR_UNDEF
 Sym: 'common' Resolution: LDPR_PREVAILING_DEF_IRONLY
+#...
 hook called: cleanup.
 #...
diff -Naur binutils-2.32/ld/testsuite/ld-srec/srec.exp binutils-2.32-p/ld/testsuite/ld-srec/srec.exp
--- binutils-2.32/ld/testsuite/ld-srec/srec.exp	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-srec/srec.exp	2019-05-19 02:41:25.494406461 +0200
@@ -21,6 +21,8 @@
 
 # Get the offset from an S-record line to the start of the data.
 
+return
+
 proc srec_off { l } {
     if [string match "S1*" $l] {
 	return 8
diff -Naur binutils-2.32/ld/testsuite/ld-x86-64/x86-64.exp binutils-2.32-p/ld/testsuite/ld-x86-64/x86-64.exp
--- binutils-2.32/ld/testsuite/ld-x86-64/x86-64.exp	2019-01-19 17:01:34.000000000 +0100
+++ binutils-2.32-p/ld/testsuite/ld-x86-64/x86-64.exp	2019-05-19 02:41:25.494406461 +0200
@@ -1275,25 +1275,6 @@
 		"$NOPIE_CFLAGS" \
 	    ] \
 	]
-    } else {
-	run_cc_link_tests [list \
-	    [list \
-		"Build pr22001-1b" \
-		"$NOPIE_LDFLAGS -Wl,-z,nocopyreloc,--no-as-needed tmpdir/pr22001-1.so" \
-		"$NOPIE_CFLAGS -Wa,-mx86-used-note=yes" \
-		{ pr22001-1c.c } \
-		{{error_output "pr22001-1b.err"}} \
-		"pr22001-1b" \
-	    ] \
-	    [list \
-		"Build pr21997-1b" \
-		"$NOPIE_LDFLAGS -Wl,--no-as-needed tmpdir/pr21997-1.so" \
-		"$NOPIE_CFLAGS -Wa,-mx86-used-note=yes" \
-		{ pr21997-1c.c } \
-		{{error_output "pr21997-1b.err"}} \
-		"pr21997-1b" \
-	    ] \
-	]
     }
 
     run_ld_link_exec_tests [list \
