diff -urN autoconf-2.62.org/lib/autoconf/c.m4 autoconf-2.62/lib/autoconf/c.m4
--- autoconf-2.62.org/lib/autoconf/c.m4	2008-04-06 01:04:48.000000000 +0200
+++ autoconf-2.62/lib/autoconf/c.m4	2008-06-17 18:37:55.194174188 +0200
@@ -1416,7 +1416,7 @@
 	     #endif
 	     typedef int dummy;
 	    ]])],
-	 [ac_cv_c_bigendian=universal])],
+	 [ac_cv_c_bigendian=unknown])],
       [AC_DIAGNOSE([obsolete], [AC_C_BIGENDIAN suggests AC_CONFIG_HEADERS])])
     if test $ac_cv_c_bigendian = unknown; then
       # See if sys/param.h defines the BYTE_ORDER macro.

