diff -Naur kmymoney2-0.8.2.orig/kmymoney2/dialogs/kcurrencyeditdlg.cpp kmymoney2-0.8.2/kmymoney2/dialogs/kcurrencyeditdlg.cpp
--- kmymoney2-0.8.2.orig/kmymoney2/dialogs/kcurrencyeditdlg.cpp	2005-12-20 16:58:39.000000000 +0100
+++ kmymoney2-0.8.2/kmymoney2/dialogs/kcurrencyeditdlg.cpp	2006-01-17 20:07:05.000000000 +0100
@@ -41,6 +41,8 @@
 #include <kmessagebox.h>
 #include <kiconloader.h>
 
+#include <locale.h>
+
 // ----------------------------------------------------------------------------
 // Project Includes
 
