diff -aur k3b-1.0.4/plugins/decoder/ffmpeg/configure.in.in fw_k3b-1.0.4/plugins/decoder/ffmpeg/configure.in.in
--- k3b-1.0.4/plugins/decoder/ffmpeg/configure.in.in	2007-11-02 10:55:03.000000000 +0100
+++ fw_k3b-1.0.4/plugins/decoder/ffmpeg/configure.in.in	2008-04-28 22:11:06.000000000 +0200
@@ -21,8 +21,8 @@
 	AC_LANG_CPLUSPLUS
 	AC_COMPILE_IFELSE( 
 		extern "C" {
-		#include <ffmpeg/avformat.h>
-		#include <ffmpeg/avcodec.h>
+		#include <libavformat/avformat.h>
+		#include <libavcodec/avcodec.h>
 		}
 
 		int main() {
@@ -36,8 +36,8 @@
 	LIBS="-lavformat -lavcodec $LIBS"
 	AC_LINK_IFELSE( 
 		extern "C" {
-		#include <ffmpeg/avformat.h>
-		#include <ffmpeg/avcodec.h>
+		#include <libavformat/avformat.h>
+		#include <libavcodec/avcodec.h>
 		}
 
 		int main() {
diff -aur k3b-1.0.4/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp fw_k3b-1.0.4/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp
--- k3b-1.0.4/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp	2007-11-02 10:55:03.000000000 +0100
+++ fw_k3b-1.0.4/plugins/decoder/ffmpeg/k3bffmpegdecoder.cpp	2008-04-28 22:51:53.000000000 +0200
@@ -22,7 +22,7 @@
 #include <k3bpluginfactory.h>
 
 extern "C" {
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 }
 
 #include <math.h>
diff -aur k3b-1.0.4/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp fw_k3b-1.0.4/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp
--- k3b-1.0.4/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2007-11-02 10:55:03.000000000 +0100
+++ fw_k3b-1.0.4/plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2008-04-28 22:30:49.000000000 +0200
@@ -18,8 +18,8 @@
 #include "k3bffmpegwrapper.h"
 
 extern "C" {
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 }
 
 #include <string.h>
