--- k3b-1.0.5/libk3b/core/Makefile.in~	2008-05-27 10:34:44.000000000 +0200
+++ k3b-1.0.5/libk3b/core/Makefile.in	2009-10-28 01:41:22.000000000 +0100
@@ -387,7 +387,7 @@
 include_HEADERS = k3bcore.h k3bdefaultexternalprograms.h k3bexternalbinmanager.h \
 	k3bprocess.h k3bversion.h k3bglobals.h k3bjob.h k3bthread.h \
 	k3bthreadjob.h k3bglobalsettings.h k3bjobhandler.h \
-	k3b_export.h k3bjobhandler.h k3bsimplejobhandler.h
+	k3b_export.h k3bsimplejobhandler.h
 
 #>- METASOURCES = AUTO
 #>- all: all-am
--- k3b-1.0.5/libk3b/core/Makefile.am~	2008-05-27 10:24:09.000000000 +0200
+++ k3b-1.0.5/libk3b/core/Makefile.am	2009-10-28 01:41:43.000000000 +0100
@@ -13,7 +13,7 @@
 include_HEADERS = k3bcore.h k3bdefaultexternalprograms.h k3bexternalbinmanager.h \
 	k3bprocess.h k3bversion.h k3bglobals.h k3bjob.h k3bthread.h \
 	k3bthreadjob.h k3bglobalsettings.h k3bjobhandler.h \
-	k3b_export.h k3bjobhandler.h k3bsimplejobhandler.h
+	k3b_export.h k3bsimplejobhandler.h
 
 METASOURCES = AUTO
 
