diff -Naur k3b-1.91.0/k3bsetup/k3bsetup.desktop k3b-1.91.0-p/k3bsetup/k3bsetup.desktop
--- k3b-1.91.0/k3bsetup/k3bsetup.desktop	2010-03-09 07:47:24.000000000 +0100
+++ k3b-1.91.0-p/k3bsetup/k3bsetup.desktop	2010-03-31 10:42:48.585315300 +0200
@@ -108,6 +108,7 @@
 X-KDE-ServiceTypes=KCModule
 X-KDE-Library=kcm_k3bsetup
 X-KDE-ParentApp=kcontrol
+X-KDE-RootOnly=true
 X-KDE-System-Settings-Parent-Category=system
 X-KDE-Keywords=k3b,k3bsetup,cd,dvd
 Categories=Application;System;X-KDE-System;
