--- /plugins/decoder/ffmpeg/k3bffmpegwrapper.cpp	2009-12-12 00:07:36.000000000 +0100
+++ /plugins/decoder/ffmpeg/k3bffmpegwrapper.cppnew	2009-12-12 00:47:39.000000000 +0100
@@ -31,6 +31,9 @@
 #define FFMPEG_BUILD_PRE_4629
 #endif
 
+#ifndef CODEC_ID_MP3LAME
+#define CODEC_ID_MP3LAME CODEC_ID_MP3
+#endif
 
 K3bFFMpegWrapper* K3bFFMpegWrapper::s_instance = 0;
 
@@ -290,7 +293,7 @@
 #ifdef FFMPEG_BUILD_PRE_4629
     int len = avcodec_decode_audio( &d->formatContext->streams[0]->codec,
 #else
-    int len = avcodec_decode_audio( d->formatContext->streams[0]->codec,
+    int len = avcodec_decode_audio2( d->formatContext->streams[0]->codec,
 #endif
 				    (short*)d->outputBuffer, &d->outputBufferSize,
 				    d->packetData, d->packetSize );
