diff -Naur kinfocenter-5.6.2/Modules/CMakeLists.txt kinfocenter-5.6.2-p/Modules/CMakeLists.txt
--- kinfocenter-5.6.2/Modules/CMakeLists.txt	2016-04-05 15:42:11.000000000 +0200
+++ kinfocenter-5.6.2-p/Modules/CMakeLists.txt	2016-04-14 15:45:49.796664768 +0200
@@ -61,7 +61,8 @@
 endif()
 
 if (CMAKE_SYSTEM_NAME MATCHES Linux)
-    add_subdirectory(energy)
+    ##add_subdirectory(energy) - broken see https://bugs.kde.org/show_bug.cgi?id=347344
+    ## segfaults all the time
     add_subdirectory(fileindexermonitor)
 endif()
 
