diff -Naur kinfocenter-5.8.1/Modules/about-distro/src/Module.cpp kinfocenter-5.8.1-p/Modules/about-distro/src/Module.cpp
--- kinfocenter-5.8.1/Modules/about-distro/src/Module.cpp	2016-10-11 14:27:00.000000000 +0200
+++ kinfocenter-5.8.1-p/Modules/about-distro/src/Module.cpp	2016-10-13 18:31:53.766279410 +0200
@@ -243,8 +243,7 @@
 
 QString Module::plasmaVersion() const
 {
-    const QStringList &filePaths = QStandardPaths::locateAll(QStandardPaths::GenericDataLocation,
-                                                             QStringLiteral("xsessions/plasma.desktop"));
+    const QStringList &filePaths = QStringList("/etc/X11/sessions/plasma.desktop");
 
     if (filePaths.length() < 1) {
         return QString();
