diff -Nur gconf-editor-2.14.0/src/gconf-editor-window.c gconf-editor-2.14.0.new/src/gconf-editor-window.c
--- gconf-editor-2.14.0/src/gconf-editor-window.c	2006-01-30 07:19:32.000000000 +0100
+++ gconf-editor-2.14.0.new/src/gconf-editor-window.c	2006-03-26 12:33:58.000000000 +0200
@@ -950,7 +950,7 @@
 		gconf_add_recent_key (window, key);
 	}
 
-	g_free (value);
+	gconf_value_free (value);
 	g_free (key);
 	gtk_tree_path_free (path);
 }
