From: Lukas Schwaighofer <lukas@schwaighofer.name>
Date: Sun, 25 Nov 2018 16:41:41 +0100
Subject: Include changes since 6.03 in the changelog

Origin: http://www.syslinux.org/wiki/index.php?title=Syslinux_6_Changelog
Forwarded: not-needed
---
 NEWS | 55 +++++++++++++++++++++++++++++++++++++++++++++++++++++++
 1 file changed, 55 insertions(+)

diff --git a/NEWS b/NEWS
index c0846d8..ce80940 100644
--- a/NEWS
+++ b/NEWS
@@ -2,6 +2,61 @@ Starting with 1.47, changes marked with SYSLINUX, PXELINUX, ISOLINUX
 or EXTLINUX apply to that specific program only; other changes apply
 to all derivatives.
 
+Changes in 6.04~git20190221.05ac953c (unreleased):
+	* zlib: Update to version 1.2.11.
+	* libpng: Update to version 1.6.36.
+	* core/Makefile: Sort to be reproducible.
+	* efi/udp: Improve performance (Julien Viard de Galbert).
+	* keeppxe: Partial fix; the LINUX directive can pass
+	  the keeppxe parameter (Adam Goldman).
+	* chain.c32: Update "reactos=" option.
+	* keytab-lilo: Update to support kbd 2.0.3 format (Olivier Brunel).
+	* menu: Apply VSHIFT to scrollbar (Bruno Levert).
+	* ext4: Support 64bit feature (Paulo Alcantara).
+	* isohybrid: Use binary mode for compatibility with DOS/Windows
+	  (Colin Finck).
+	* isohybrid, MBR: Fix compatibility with CHS mode
+	  in certain BIOS (Martin Str|mberg, Thomas Schmitt).
+	* ISOLINUX: Fix the calculation of the boot info table
+	  default checksum (Thomas Schmitt); it was broken since 4.00.
+	* diag/geodsp improvements (Gene Cumm).
+	* HDT: Fixes, updates and improvements (Erwan Velu).
+	* load_linux: relocate protected-mode code as intended
+	  (Scot Doyle).
+	* SYSAPPEND: Fix space stripping (Dany St-Amant).
+	* SERIAL: Fix, allow octal and hex values on "port" and
+	  on "flow control" parameters (Andrew J. Schorr, Dany St-Amant).
+	* efi: Fix hang when no keyboard is present (Oliver Wagner).
+	* efi: Add support for firmware's architecture-specific
+	  optional configuration file names.
+	* core: Re-add gPXE/iPXE support for HTTP on pxelinux.0
+	  (Gene Cumm).
+	* core/http: Append port number to Host field if needed
+	  (Gene Cumm).
+	* chain.c32: Add options to ignore GPT crc checks
+	  (Michal Soltys).
+	* mboot.c32: Add ELF64 support for Multiboot1
+	  (Alex Barracuda72, Gene Cumm).
+	* XFS: Support XFS on-disk V5, node directories version 3,
+	  xfsprogs 4.x. (Paulo Alcantara).
+	* BTRFS: Fix logical_to_physical block address mapping
+	  (Paulo Alcantara).
+	* lua: Fixes and improvements;
+	  correct and improve its documentation (Ferenc Wágner).
+	* Old unmaintained gPXE code is no longer included.
+	* Code improvements.
+	* efi/udp: Improve performance (Julien Viard de Galbert).
+	* isohybrid: Use binary mode for compatibility with DOS/Windows
+	  (Colin Finck).
+	* isohybrid, MBR: Fix compatibility with CHS mode
+	  in certain BIOS (Martin Str|mberg, Thomas Schmitt).
+	* keeppxe: Partial fix; the LINUX directive can pass
+	  the keeppxe parameter (Adam Goldman).
+	* chain.c32: Update "reactos=" option.
+	* keytab-lilo: Update to support kbd 2.0.3 format (Olivier Brunel).
+	* menu: Apply VSHIFT to scrollbar (Bruno Levert).
+	* ext4: Support 64bit feature (Paulo Alcantara).
+
 Changes in 6.03:
 	* chain: Fix chainloading on 6.02 (Raphael S. Carvalho).
 	* core: Do not leak the __lowmem and __bss16 macros to modules;
