diff -ur a/mk/efi.mk b/mk/efi.mk
--- a/mk/efi.mk	2013-10-13 12:59:03.000000000 -0500
+++ b/mk/efi.mk	2013-11-09 06:40:33.578925260 -0600
@@ -23,11 +23,11 @@
 	EFI_SUBARCH = $(ARCH)
 endif
 
-EFIINC = $(shell $(topdir)/efi//find-gnu-efi.sh include $(EFI_SUBARCH))
+EFIINC = /usr/include/efi
 $(if $(EFIINC),, \
 	$(error Missing $(EFI_SUBARCH) gnu-efi header files))
 
-LIBDIR = $(shell $(topdir)/efi/find-gnu-efi.sh lib $(EFI_SUBARCH))
+LIBDIR = /usr/lib
 $(if $(LIBDIR),, \
 	$(error Missing $(EFI_SUBARCH) gnu-efi libraries))
 
@@ -45,8 +45,8 @@
 		-Wno-unused-parameter
 
 # gnuefi sometimes installs these under a gnuefi/ directory, and sometimes not
-CRT0 := $(shell find $(LIBDIR) -name crt0-efi-$(EFI_SUBARCH).o 2>/dev/null | tail -n1)
-LDSCRIPT := $(shell find $(LIBDIR) -name elf_$(EFI_SUBARCH)_efi.lds 2>/dev/null | tail -n1)
+CRT0 := $(LIBDIR)/crt0-efi-$(EFI_SUBARCH).o
+LDSCRIPT := $(LIBDIR)/elf_$(EFI_SUBARCH)_efi.lds
 
 LDFLAGS = -T $(SRC)/$(ARCH)/syslinux.ld -Bsymbolic -pie -nostdlib -znocombreloc \
 		-L$(LIBDIR) --hash-style=gnu -m elf_$(ARCH) $(CRT0) -E
