diff -ur a/Makefile b/Makefile
--- a/Makefile	2013-07-04 08:20:20.000000000 -0500
+++ b/Makefile	2013-11-09 07:02:42.763875841 -0600
@@ -123,7 +123,7 @@
 #
 
 ifndef EFI_BUILD
-MODULES = memdisk/memdisk memdump/memdump.com \
+MODULES = memdisk/memdisk \
 	com32/menu/*.c32 com32/modules/*.c32 com32/mboot/*.c32 \
 	com32/hdt/*.c32 com32/rosh/*.c32 com32/gfxboot/*.c32 \
 	com32/sysdump/*.c32 com32/lua/src/*.c32 com32/chain/*.c32 \
@@ -149,9 +149,7 @@
 	mbr/*.bin \
 	core/pxelinux.0 core/lpxelinux.0 \
 	core/isolinux.bin core/isolinux-debug.bin \
-	gpxe/gpxelinux.0 dos/syslinux.com \
-	win32/syslinux.exe win64/syslinux64.exe \
-	dosutil/*.com dosutil/*.sys \
+	gpxe/gpxelinux.0 \
 	$(MODULES)
 
 # BSUBDIRs build the on-target binary components.
@@ -170,8 +168,8 @@
 
 else
 
-BSUBDIRS = codepage com32 lzo core memdisk mbr memdump gpxe sample \
-	   diag libinstaller dos win32 win64 dosutil txt
+BSUBDIRS = codepage com32 lzo core memdisk mbr gpxe sample \
+	   libinstaller txt
 
 ITARGET  =
 IOBJECTS = $(ITARGET) \
@@ -186,14 +184,11 @@
 # Things to install in /usr/lib/syslinux
 INSTALL_AUX   =	core/pxelinux.0 gpxe/gpxelinux.0 gpxe/gpxelinuxk.0 \
 		core/isolinux.bin core/isolinux-debug.bin \
-		dos/syslinux.com core/lpxelinux.0 \
+		core/lpxelinux.0 \
 		mbr/*.bin $(INSTALLABLE_MODULES)
-INSTALL_AUX_OPT = win32/syslinux.exe win64/syslinux64.exe
-INSTALL_DIAG  =	diag/mbr/handoff.bin \
-		diag/geodsp/geodsp1s.img.xz diag/geodsp/geodspms.img.xz
 
 # These directories manage their own installables
-INSTALLSUBDIRS = com32 utils dosutil
+INSTALLSUBDIRS = com32 utils
 
 # Things to install in /boot/extlinux
 EXTBOOTINSTALL = $(INSTALLABLE_MODULES)
@@ -280,7 +275,7 @@
 
 #
 # List the dependencies to help out parallel builds.
-dos extlinux linux mtools win32 win64: libinstaller
+extlinux linux mtools: libinstaller
 libinstaller: core
 utils: mbr
 core: com32
@@ -315,9 +310,6 @@
 	install -m 755 -c $(INSTALL_SBIN) $(INSTALLROOT)$(SBINDIR)
 	mkdir -m 755 -p $(INSTALLROOT)$(AUXDIR)
 	install -m 644 -c $(INSTALL_AUX) $(INSTALLROOT)$(AUXDIR)
-	-install -m 644 -c $(INSTALL_AUX_OPT) $(INSTALLROOT)$(AUXDIR)
-	mkdir -m 755 -p $(INSTALLROOT)$(DIAGDIR)
-	install -m 644 -c $(INSTALL_DIAG) $(INSTALLROOT)$(DIAGDIR)
 	mkdir -m 755 -p $(INSTALLROOT)$(MANDIR)/man1
 	install -m 644 -c $(topdir)/man/*.1 $(INSTALLROOT)$(MANDIR)/man1
 	: mkdir -m 755 -p $(INSTALLROOT)$(MANDIR)/man8
