diff -Naur LVM2.2.03.02/scripts/dm_event_systemd_red_hat.service.in LVM2.2.03.02.new/scripts/dm_event_systemd_red_hat.service.in
--- LVM2.2.03.02/scripts/dm_event_systemd_red_hat.service.in	2018-12-18 15:22:34.000000000 +0100
+++ LVM2.2.03.02.new/scripts/dm_event_systemd_red_hat.service.in	2018-12-22 12:28:48.202771738 +0100
@@ -13,3 +13,6 @@
 Environment=SD_ACTIVATION=1
 PIDFile=@DMEVENTD_PIDFILE@
 OOMScoreAdjust=-1000
+
+[Install]
+WantedBy=sysinit.target
diff -Naur LVM2.2.03.02/scripts/lvm2_lvmpolld_systemd_red_hat.service.in LVM2.2.03.02.new/scripts/lvm2_lvmpolld_systemd_red_hat.service.in
--- LVM2.2.03.02/scripts/lvm2_lvmpolld_systemd_red_hat.service.in	2018-12-18 15:22:34.000000000 +0100
+++ LVM2.2.03.02.new/scripts/lvm2_lvmpolld_systemd_red_hat.service.in	2018-12-22 12:28:48.641771723 +0100
@@ -12,3 +12,6 @@
 ExecStart=@SBINDIR@/lvmpolld -t 60 -f
 Environment=SD_ACTIVATION=1
 PIDFile=@LVMPOLLD_PIDFILE@
+
+[Install]
+WantedBy=sysinit.target
