diff -Naur LVM2.2.03.01/scripts/dm_event_systemd_red_hat.service.in LVM2.2.03.01-p/scripts/dm_event_systemd_red_hat.service.in
--- LVM2.2.03.01/scripts/dm_event_systemd_red_hat.service.in	2018-10-31 15:54:24.000000000 +0100
+++ LVM2.2.03.01-p/scripts/dm_event_systemd_red_hat.service.in	2018-11-07 18:39:08.055103006 +0100
@@ -13,3 +13,6 @@
 Environment=SD_ACTIVATION=1
 PIDFile=@DMEVENTD_PIDFILE@
 OOMScoreAdjust=-1000
+
+[Install]
+WantedBy=sysinit.target
diff -Naur LVM2.2.03.01/scripts/lvm2_lvmpolld_systemd_red_hat.service.in LVM2.2.03.01-p/scripts/lvm2_lvmpolld_systemd_red_hat.service.in
--- LVM2.2.03.01/scripts/lvm2_lvmpolld_systemd_red_hat.service.in	2018-10-31 15:54:24.000000000 +0100
+++ LVM2.2.03.01-p/scripts/lvm2_lvmpolld_systemd_red_hat.service.in	2018-11-07 18:39:28.033226116 +0100
@@ -12,3 +12,6 @@
 ExecStart=@SBINDIR@/lvmpolld -t 60 -f
 Environment=SD_ACTIVATION=1
 PIDFile=@LVMPOLLD_PIDFILE@
+
+[Install]
+WantedBy=sysinit.target
diff -Naur LVM2.2.03.01/scripts/lvm2_monitoring_systemd_red_hat.service.in LVM2.2.03.01-p/scripts/lvm2_monitoring_systemd_red_hat.service.in
--- LVM2.2.03.01/scripts/lvm2_monitoring_systemd_red_hat.service.in	2018-10-31 15:54:24.000000000 +0100
+++ LVM2.2.03.01-p/scripts/lvm2_monitoring_systemd_red_hat.service.in	2018-11-07 18:40:58.574786340 +0100
@@ -1,8 +1,8 @@
 [Unit]
 Description=Monitoring of LVM2 mirrors, snapshots etc. using dmeventd or progress polling
 Documentation=man:dmeventd(8) man:lvcreate(8) man:lvchange(8) man:vgchange(8)
-Requires=dm-event.socket lvm2-lvmetad.socket
-After=dm-event.socket dm-event.service lvm2-lvmetad.socket lvm2-activation.service lvm2-lvmetad.service
+Requires=dm-event.socket
+After=dm-event.socket dm-event.service lvm2-activation.service
 Before=local-fs-pre.target
 DefaultDependencies=no
 Conflicts=shutdown.target
