diff -Naur cairo-dock-1.6.3.1/configure.ac cairo-dock-1.6.3.1.new/configure.ac
--- cairo-dock-1.6.3.1/configure.ac	2008-11-11 04:04:23.000000000 +0530
+++ cairo-dock-1.6.3.1.new/configure.ac	2009-01-31 14:04:03.000000000 +0530
@@ -32,7 +32,7 @@
 # Checks for library functions.
 # AC_CHECK_FUNCS([sscanf])
 
-PKG_CHECK_MODULES(PACKAGE, "gtk+-2.0 gthread-2.0 cairo librsvg-2.0 glitz dbus-1 dbus-glib-1 libxml-2.0")
+PKG_CHECK_MODULES(PACKAGE, "gtk+-2.0 gthread-2.0 cairo librsvg-2.0 glitz dbus-1 dbus-glib-1 libxml-2.0 xcomposite")
 
 AC_ARG_ENABLE(xextend,
 	[  --enable-xextend  :  Enable X Extensions],
