diff -urNad gnome-system-tools-2.15.0~/src/network/network.desktop.in.in gnome-system-tools-2.15.0/src/network/network.desktop.in.in
--- gnome-system-tools-2.15.0~/src/network/network.desktop.in.in	2006-05-27 17:28:17.000000000 +0200
+++ gnome-system-tools-2.15.0/src/network/network.desktop.in.in	2006-06-13 14:49:06.000000000 +0200
@@ -1,7 +1,8 @@
 [Desktop Entry]
 _Name=Networking
 _Comment=Configure network devices and connections
-Exec=network-admin
+Exec=gksu network-admin
+X-KDE-SubstituteUID=true
 Icon=gnome-fs-network
 Terminal=false
 Type=Application
diff -urNad gnome-system-tools-2.15.0~/src/services/services.desktop.in.in gnome-system-tools-2.15.0/src/services/services.desktop.in.in
--- gnome-system-tools-2.15.0~/src/services/services.desktop.in.in	2006-05-27 17:28:45.000000000 +0200
+++ gnome-system-tools-2.15.0/src/services/services.desktop.in.in	2006-06-13 14:49:13.000000000 +0200
@@ -1,7 +1,8 @@
 [Desktop Entry]
 _Name=Services
 _Comment=Configure which services will be run when the system starts
-Exec=services-admin
+Exec=gksu services-admin
+X-KDE-SubstituteUID=true
 Icon=gnome-fs-server
 Terminal=false
 Type=Application
diff -urNad gnome-system-tools-2.15.0~/src/shares/shares.desktop.in.in gnome-system-tools-2.15.0/src/shares/shares.desktop.in.in
--- gnome-system-tools-2.15.0~/src/shares/shares.desktop.in.in	2006-05-27 17:29:09.000000000 +0200
+++ gnome-system-tools-2.15.0/src/shares/shares.desktop.in.in	2006-06-13 14:49:21.000000000 +0200
@@ -1,7 +1,8 @@
 [Desktop Entry]
 _Name=Shared Folders
 _Comment=Configure which folders are available for your network neighborhood
-Exec=shares-admin
+Exec=gksu shares-admin
+X-KDE-SubstituteUID=true
 Icon=gnome-fs-share
 Terminal=false
 Type=Application
diff -urNad gnome-system-tools-2.15.0~/src/time/time.desktop.in.in gnome-system-tools-2.15.0/src/time/time.desktop.in.in
--- gnome-system-tools-2.15.0~/src/time/time.desktop.in.in	2006-05-27 17:29:51.000000000 +0200
+++ gnome-system-tools-2.15.0/src/time/time.desktop.in.in	2006-06-13 14:49:32.000000000 +0200
@@ -1,7 +1,8 @@
 [Desktop Entry]
 _Name=Time and Date
 _Comment=Change system time, date, and timezone
-Exec=time-admin
+Exec=gksu time-admin
+X-KDE-SubstituteUID=true
 Icon=config-date
 Terminal=false
 Type=Application
diff -urNad gnome-system-tools-2.15.0~/src/users/users.desktop.in.in gnome-system-tools-2.15.0/src/users/users.desktop.in.in
--- gnome-system-tools-2.15.0~/src/users/users.desktop.in.in	2006-05-27 17:30:28.000000000 +0200
+++ gnome-system-tools-2.15.0/src/users/users.desktop.in.in	2006-06-13 14:49:39.000000000 +0200
@@ -1,7 +1,8 @@
 [Desktop Entry]
 _Name=Users and Groups
 _Comment=Add or remove users and groups
-Exec=users-admin
+Exec=gksu users-admin
+X-KDE-SubstituteUID=true
 Icon=config-users
 Terminal=false
 Type=Application
