diff -urNad --exclude=CVS --exclude=.svn ./src/shares/nautilus/nautilus-shares.c /tmp/dpep-work.elM5c8/gnome-system-tools-1.4.0/src/shares/nautilus/nautilus-shares.c
--- ./src/shares/nautilus/nautilus-shares.c	2005-09-16 16:19:42.000000000 +0200
+++ /tmp/dpep-work.elM5c8/gnome-system-tools-1.4.0/src/shares/nautilus/nautilus-shares.c	2005-09-16 16:23:37.000000000 +0200
@@ -82,14 +82,16 @@
   GString          *cmd;
   gchar            *str, *dir;
 
-  cmd  = g_string_new ("shares-admin ");
+  cmd  = g_string_new ("gksudo -m \"You need administrator privileges to use "
+		       "this tool. Enter your password to modify "
+		       "your system configuration.\" \"shares-admin ");
   info = g_object_get_data (G_OBJECT (menu_item), "file");
 
   str = nautilus_file_info_get_uri (info);
   uri = gnome_vfs_uri_new (str);
   dir = get_path_from_url (gnome_vfs_uri_get_path (uri));
 
-  g_string_append_printf (cmd, "--add-share=\"%s\"", dir);
+  g_string_append_printf (cmd, "--add-share=\"%s\" \"", dir);
 
   g_spawn_command_line_async (cmd->str, NULL);
 
