diff -urNad --exclude=CVS --exclude=.svn ./src/boot/boot.desktop.in.in /tmp/dpep-work.Tq5STv/gnome-system-tools-1.4.0/src/boot/boot.desktop.in.in
--- ./src/boot/boot.desktop.in.in	2005-01-10 01:12:26.000000000 +0100
+++ /tmp/dpep-work.Tq5STv/gnome-system-tools-1.4.0/src/boot/boot.desktop.in.in	2005-09-16 22:16:05.000000000 +0200
@@ -1,7 +1,8 @@
 [Desktop Entry]
 _Name=Boot
 _Comment=Configure the system boot process
-Exec=boot-admin
+TryExec=gksu
+Exec=gksu boot-admin
 Icon=@pixmapsdir@/boot.png
 Terminal=false
 Type=Application
diff -urNad --exclude=CVS --exclude=.svn ./src/disks/disks.desktop.in.in /tmp/dpep-work.Tq5STv/gnome-system-tools-1.4.0/src/disks/disks.desktop.in.in
--- ./src/disks/disks.desktop.in.in	2005-01-10 01:12:26.000000000 +0100
+++ /tmp/dpep-work.Tq5STv/gnome-system-tools-1.4.0/src/disks/disks.desktop.in.in	2005-09-16 22:16:23.000000000 +0200
@@ -1,7 +1,8 @@
 [Desktop Entry]
 _Name=Disks
 _Comment=Configure the system disks
-Exec=disks-admin
+TryExec=gksu
+Exec=gksu disks-admin
 Icon=@pixmapsdir@/disks.png
 Terminal=false
 Type=Application
diff -urNad --exclude=CVS --exclude=.svn ./src/network/network.desktop.in.in /tmp/dpep-work.Tq5STv/gnome-system-tools-1.4.0/src/network/network.desktop.in.in
--- ./src/network/network.desktop.in.in	2005-01-10 01:12:26.000000000 +0100
+++ /tmp/dpep-work.Tq5STv/gnome-system-tools-1.4.0/src/network/network.desktop.in.in	2005-09-16 22:15:39.000000000 +0200
@@ -1,7 +1,8 @@
 [Desktop Entry]
 _Name=Networking
 _Comment=Configure network devices and connections
-Exec=network-admin
+TryExec=gksu
+Exec=gksu -u root network-admin
 Icon=@pixmapsdir@/network.png
 Terminal=false
 Type=Application
diff -urNad --exclude=CVS --exclude=.svn ./src/services/services.desktop.in.in /tmp/dpep-work.Tq5STv/gnome-system-tools-1.4.0/src/services/services.desktop.in.in
--- ./src/services/services.desktop.in.in	2005-01-10 01:12:27.000000000 +0100
+++ /tmp/dpep-work.Tq5STv/gnome-system-tools-1.4.0/src/services/services.desktop.in.in	2005-09-16 22:16:10.000000000 +0200
@@ -1,7 +1,8 @@
 [Desktop Entry]
 _Name=Services
 _Comment=Configure which services will be run when the system starts
-Exec=services-admin
+TryExec=gksu
+Exec=gksu services-admin
 Icon=@pixmapsdir@/services.png
 Terminal=false
 Type=Application
diff -urNad --exclude=CVS --exclude=.svn ./src/shares/shares.desktop.in.in /tmp/dpep-work.Tq5STv/gnome-system-tools-1.4.0/src/shares/shares.desktop.in.in
--- ./src/shares/shares.desktop.in.in	2005-06-13 00:03:17.000000000 +0200
+++ /tmp/dpep-work.Tq5STv/gnome-system-tools-1.4.0/src/shares/shares.desktop.in.in	2005-09-16 22:16:30.000000000 +0200
@@ -1,7 +1,8 @@
 [Desktop Entry]
 _Name=Shared Folders
 _Comment=Configure which folders are available for your network neighborhood
-Exec=shares-admin
+TryExec=gksu
+Exec=gksu shares-admin
 Icon=@pixmapsdir@/shares.png
 Terminal=false
 Type=Application
diff -urNad --exclude=CVS --exclude=.svn ./src/time/time.desktop.in.in /tmp/dpep-work.Tq5STv/gnome-system-tools-1.4.0/src/time/time.desktop.in.in
--- ./src/time/time.desktop.in.in	2005-01-10 01:12:27.000000000 +0100
+++ /tmp/dpep-work.Tq5STv/gnome-system-tools-1.4.0/src/time/time.desktop.in.in	2005-09-16 22:15:59.000000000 +0200
@@ -1,7 +1,8 @@
 [Desktop Entry]
 _Name=Time and Date
 _Comment=Change system time, date, and timezone
-Exec=time-admin
+TryExec=gksu
+Exec=gksu time-admin
 Icon=@pixmapsdir@/time.png
 Terminal=false
 Type=Application
diff -urNad --exclude=CVS --exclude=.svn ./src/users/users.desktop.in.in /tmp/dpep-work.Tq5STv/gnome-system-tools-1.4.0/src/users/users.desktop.in.in
--- ./src/users/users.desktop.in.in	2005-01-10 01:12:27.000000000 +0100
+++ /tmp/dpep-work.Tq5STv/gnome-system-tools-1.4.0/src/users/users.desktop.in.in	2005-09-16 22:16:18.000000000 +0200
@@ -1,7 +1,8 @@
 [Desktop Entry]
 _Name=Users and Groups
 _Comment=Add or remove users and groups
-Exec=users-admin
+TryExec=gksu
+Exec=gksu users-admin
 Icon=@pixmapsdir@/users.png
 Terminal=false
 Type=Application
