diff -Naur xawtv-3.94.orig/libng/plugins/read-qt.c xawtv-3.94/libng/plugins/read-qt.c
--- xawtv-3.94.orig/libng/plugins/read-qt.c	2005-08-05 16:43:09.000000000 +0200
+++ xawtv-3.94/libng/plugins/read-qt.c	2005-08-05 16:43:59.000000000 +0200
@@ -4,9 +4,9 @@
 #include <stdio.h>
 #include <string.h>
 #include <pthread.h>
-#include <quicktime/quicktime.h>
-#include <quicktime/colormodels.h>
-#include <quicktime/lqt.h>
+#include <lqt/quicktime.h>
+#include <lqt/colormodels.h>
+#include <lqt/lqt.h>
 
 #include "grab-ng.h"
 
diff -Naur xawtv-3.94.orig/libng/plugins/write-qt.c xawtv-3.94/libng/plugins/write-qt.c
--- xawtv-3.94.orig/libng/plugins/write-qt.c	2005-08-05 16:43:09.000000000 +0200
+++ xawtv-3.94/libng/plugins/write-qt.c	2005-08-05 16:43:45.000000000 +0200
@@ -4,9 +4,9 @@
 #include <stdio.h>
 #include <string.h>
 #include <pthread.h>
-#include <quicktime/quicktime.h>
-#include <quicktime/colormodels.h>
-#include <quicktime/lqt.h>
+#include <lqt/quicktime.h>
+#include <lqt/colormodels.h>
+#include <lqt/lqt.h>
 
 #include "grab-ng.h"
 
