diff -x '*~' -ur xawtv-3.94/console/fbtv.c xawtv-3.94-gcc4/console/fbtv.c
--- xawtv-3.94/console/fbtv.c	2004-06-30 14:26:52.000000000 +0200
+++ xawtv-3.94-gcc4/console/fbtv.c	2005-05-30 02:09:30.516434608 +0200
@@ -187,7 +187,7 @@
 
 #define NKEYTAB (sizeof(keytab)/sizeof(struct KEYTAB))
 
-static char              *snapbase;
+char              *snapbase;
 static char              default_title[128] = "???";
 static char              message[128] = "";
 
diff -x '*~' -ur xawtv-3.94/console/ftp.c xawtv-3.94-gcc4/console/ftp.c
--- xawtv-3.94/console/ftp.c	2003-02-14 15:14:04.000000000 +0100
+++ xawtv-3.94-gcc4/console/ftp.c	2005-05-30 02:07:26.185335808 +0200
@@ -186,7 +186,7 @@
 		s->connected = 0;
 	    }
 	    if (NULL != strstr(p,"Not connected")) {
-		if (ftp_connected)
+		if (ftp_connected(s))
 		    fprintf(stderr,"ftp: lost connection\n");
 		s->connected = 0;
 	    }
@@ -236,7 +236,7 @@
 	/* login */
 	ftp_send(s,3,"user",user,pass);
 	if (230 != ftp_recv(s)) {
-	    if (!ftp_connected)
+	    if (!ftp_connected(s))
 		continue;
 	    fprintf(stderr,"ftp: login incorrect\n");
 	    exit(1);
diff -x '*~' -ur xawtv-3.94/libng/grab-ng.h xawtv-3.94-gcc4/libng/grab-ng.h
--- xawtv-3.94/libng/grab-ng.h	2003-02-14 15:14:07.000000000 +0100
+++ xawtv-3.94-gcc4/libng/grab-ng.h	2005-05-30 02:08:39.157242392 +0200
@@ -25,7 +25,9 @@
 
 #if __STDC_VERSION__ < 199901
 # define restrict
-# define bool int
+# ifndef bool
+#  define bool int
+# endif
 #endif
 
 #define UNSET          (-1U)
diff -x '*~' -ur xawtv-3.94/libng/plugins/read-avi.c xawtv-3.94-gcc4/libng/plugins/read-avi.c
--- xawtv-3.94/libng/plugins/read-avi.c	2003-02-14 15:14:05.000000000 +0100
+++ xawtv-3.94-gcc4/libng/plugins/read-avi.c	2005-05-30 02:12:25.081896624 +0200
@@ -7,6 +7,7 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <sys/uio.h>
+#include <inttypes.h>
 
 #include "riff.h"
 #include "list.h"
@@ -64,7 +65,7 @@
     h->movi[h->movi_cnt].size  = size;
     h->movi_cnt++;
     if (ng_debug)
-	fprintf(stderr,"%*s[movie data list: 0x%llx+0x%llx]\n",
+	fprintf(stderr,"%*s[movie data list: 0x%" PRIx64 "+0x%" PRIx64 "]\n",
 		level, "", start, size);
 }
 
@@ -213,7 +214,7 @@
 	*pos += (chunk.size + 3) & ~0x03; /* 32-bit align */
 	if (FCCS(chunk.id) == id) {
 	    if (ng_debug)
-		fprintf(stderr,"avi: chunk %4.4s: 0x%llx+0x%x\n",
+		fprintf(stderr,"avi: chunk %4.4s: 0x%" PRIx64 "+0x%x\n",
 			chunk.id,*pos,chunk.size);
 	    return chunk.size;
 	}
