diff -Naur abiword-2.2.8.orig/abi/GNUmakefile.am abiword-2.2.8/abi/GNUmakefile.am
--- abiword-2.2.8.orig/abi/GNUmakefile.am	2005-06-25 14:39:27.000000000 +0000
+++ abiword-2.2.8/abi/GNUmakefile.am	2005-06-25 14:48:26.000000000 +0000
@@ -92,14 +92,17 @@
 	        $(INSTALL_DATA) $(dictionary_list) $(DESTDIR)$(pkgdatadir)/$$dir/ispell_dictionary_list.xml; \
 	      fi; \
 	    else \
-	      for file in `echo $(top_srcdir)/../abidistfiles/$$dir/* | grep -v MANIFEST`; do \
+	      for file in $(top_srcdir)/../abidistfiles/$$dir/* ; do \
 	        if test -f $$file ; then \
-		  sfile=`echo "$$file" | sed 's/\.[^.]*$$//'`; \
-		  if test "$$sfile.server" = $$file; then \
-		    $(INSTALL_DATA) $$file $(DESTDIR)$(bonobodir); \
-		  else \
-		    $(INSTALL_DATA) $$file $(DESTDIR)$(pkgdatadir)/$$dir; \
-		  fi; \
+		case "$$file" in \
+		    *.server) \
+		     $(INSTALL_DATA) $$file $(DESTDIR)$(bonobodir); \
+		    ;; \
+		    *MANIFEST*) ;; \
+		    *) \
+		     $(INSTALL_DATA) $$file $(DESTDIR)$(pkgdatadir)/$$dir; \
+		    ;; \
+		esac; \
 	        fi; \
 	      done; \
 	    fi; \
diff -Naur abiword-2.2.8.orig/abi/GNUmakefile.in abiword-2.2.8/abi/GNUmakefile.in
--- abiword-2.2.8.orig/abi/GNUmakefile.in	2005-06-25 14:39:28.000000000 +0000
+++ abiword-2.2.8/abi/GNUmakefile.in	2005-06-25 14:50:38.000000000 +0000
@@ -837,14 +837,17 @@
 	        $(INSTALL_DATA) $(dictionary_list) $(DESTDIR)$(pkgdatadir)/$$dir/ispell_dictionary_list.xml; \
 	      fi; \
 	    else \
-	      for file in `echo $(top_srcdir)/../abidistfiles/$$dir/* | grep -v MANIFEST`; do \
+	      for file in $(top_srcdir)/../abidistfiles/$$dir/* ; do \
 	        if test -f $$file ; then \
-		  sfile=`echo "$$file" | sed 's/\.[^.]*$$//'`; \
-		  if test "$$sfile.server" = $$file; then \
-		    $(INSTALL_DATA) $$file $(DESTDIR)$(bonobodir); \
-		  else \
-		    $(INSTALL_DATA) $$file $(DESTDIR)$(pkgdatadir)/$$dir; \
-		  fi; \
+		case "$$file" in \
+		    *.server) \
+			$(INSTALL_DATA) $$file $(DESTDIR)$(bonobodir); \
+		    ;; \
+		    *MANIFEST*) ;; \
+		    *) \
+			$(INSTALL_DATA) $$file $(DESTDIR)$(pkgdatadir)/$$dir; \
+		    ;; \
+		esac; \
 	        fi; \
 	      done; \
 	    fi; \
