=== modified file 'deja-dup/widgets/CMakeLists.txt'
--- deja-dup/widgets/CMakeLists.txt	2014-01-23 20:35:12 +0000
+++ deja-dup/widgets/CMakeLists.txt	2014-01-31 08:47:22 +0000
@@ -52,7 +52,7 @@
                 GENERATE_VAPI widgets
                 GENERATE_HEADER widgets)
 add_library(widgets SHARED ${VALA_C})
-target_link_libraries(widgets deja "${GTK_LDFLAGS} ${NOTIFY_LDFLAGS} ${PEAS_LDFLAGS} ${SECRET_LDFLAGS} ${UNITY_LDFLAGS}")
+target_link_libraries(widgets deja ${GTK_LDFLAGS} ${NOTIFY_LDFLAGS} ${PEAS_LDFLAGS} ${SECRET_LDFLAGS} ${UNITY_LDFLAGS})
 set_target_properties(widgets PROPERTIES
                       INCLUDE_DIRECTORIES "${CMAKE_BINARY_DIR}/libdeja;${CMAKE_SOURCE_DIR}/libdeja"
                       COMPILE_FLAGS "${GTK_CFLAGS} ${NOTIFY_CFLAGS} ${PEAS_CFLAGS} ${SECRET_CFLAGS} ${UNITY_CFLAGS} ${COMMON_CFLAGS}")
