--- deja-dup-28.0/deja-dup/widgets/WidgetUtils.vala	2013-07-16 15:38:27.000000000 +0200
+++ deja-dup-28.0.new/deja-dup/widgets/WidgetUtils.vala	2013-12-03 10:09:33.241325600 +0100
@@ -60,7 +60,7 @@
       // Use Legacy unless we detect a different shell.
       shell = ShellEnv.LEGACY;
       // Next check for Shell by notification capabilities
-      unowned List<string> caps = Notify.get_server_caps();
+      List<string> caps = Notify.get_server_caps();
       bool persistence = false, actions = false;
       foreach (string cap in caps) {
         if (cap == "persistence")
