--- blaze-markup-0.8.2.8/blaze-markup.cabal	2001-09-09 03:46:40.000000000 +0200
+++ blaze-markup-0.8.2.8.new/blaze-markup.cabal	2023-03-17 16:46:42.493644106 +0100
@@ -19,8 +19,8 @@
 Cabal-version: >= 1.10
 Tested-with:   GHC == 7.8.4, GHC == 7.10.3,
                GHC == 8.0.2, GHC == 8.2.2, GHC == 8.4.4,
-               GHC == 8.6.5, GHC == 8.8.4, GHC == 8.10.4,
-               GHC == 9.0.1
+               GHC == 8.6.5, GHC == 8.8.4, GHC == 8.10.7,
+               GHC == 9.0.1, GHC == 9.2.1
 
 Extra-source-files:
   CHANGELOG
@@ -39,9 +39,9 @@
     Text.Blaze.Renderer.Utf8
 
   Build-depends:
-    base          >= 4    && < 4.16,
+    base          >= 4    && < 4.19,
     blaze-builder >= 0.3  && < 0.5,
-    text          >= 0.10 && < 1.3,
+    text          >= 0.10 && < 2.1,
     bytestring    >= 0.9  && < 0.12
 
 Test-suite blaze-markup-tests
@@ -62,17 +62,18 @@
     Text.Blaze.Tests.Util
 
   Build-depends:
-    HUnit            >= 1.2  && < 1.7,
-    QuickCheck       >= 2.7  && < 2.15,
-    containers       >= 0.3  && < 0.7,
-    tasty            >= 1.0  && < 1.5,
-    tasty-hunit      >= 0.10 && < 0.11,
-    tasty-quickcheck >= 0.10 && < 0.11,
     -- Copied from regular dependencies...
-    base          >= 4    && < 4.16,
-    blaze-builder >= 0.3  && < 0.5,
-    text          >= 0.10 && < 1.3,
-    bytestring    >= 0.9  && < 0.12
+    base
+    , blaze-builder
+    , text
+    , bytestring
+    -- Extra dependencies
+    , HUnit            >= 1.2  && < 1.7
+    , QuickCheck       >= 2.7  && < 2.15
+    , containers       >= 0.3  && < 0.7
+    , tasty            >= 1.0  && < 1.5
+    , tasty-hunit      >= 0.10 && < 0.11
+    , tasty-quickcheck >= 0.10 && < 0.11
 
 Source-repository head
   Type:     git
