--- blaze-markup-0.8.3.0/blaze-markup.cabal	2001-09-09 03:46:40.000000000 +0200
+++ blaze-markup-0.8.3.0.new/blaze-markup.cabal	2023-12-12 13:43:27.592515431 +0100
@@ -51,7 +51,7 @@
   Build-depends:
     base          >= 4    && < 5,
     blaze-builder >= 0.3  && < 0.5,
-    text          >= 0.10 && < 2.1,
+    text          >= 0.10 && < 2.2,
     bytestring    >= 0.9  && < 0.13
 
 Test-suite blaze-markup-tests
