diff -aur a/configure.ac b/configure.ac
--- a/configure.ac	2010-03-17 17:13:25.000000000 -0500
+++ b/configure.ac	2010-12-29 15:22:08.051000224 -0600
@@ -85,7 +85,7 @@
         cairo >= $CAIRO_REQUIRED,
         pixman-1 >= $PIXMAN_REQUIRED,
         gtk+-2.0 >= $GTK_REQUIRED
-        vala-1.0 >= $VALA_REQUIRED])
+        vala-0.10 >= $VALA_REQUIRED])
 AC_SUBST(CAIRO_COMPMGR_CFLAGS)
 AC_SUBST(CAIRO_COMPMGR_LIBS)
 
diff -aur a/vapi/cairo-compmgr.deps b/vapi/cairo-compmgr.deps
--- a/vapi/cairo-compmgr.deps	2009-07-19 06:43:12.000000000 -0500
+++ b/vapi/cairo-compmgr.deps	2010-12-29 15:22:16.611000210 -0600
@@ -10,5 +10,5 @@
 atk
 gtk+-2.0
 gconf-2.0
-vala-1.0
+vala-0.10
 x11
