diff -aur a/configure.ac b/configure.ac
--- a/configure.ac	2010-03-17 17:13:25.000000000 -0500
+++ b/configure.ac	2010-12-29 15:20:49.949000217 -0600
@@ -92,13 +92,13 @@
 dnl ***************************************************************************
 dnl check for bfd
 dnl ***************************************************************************
-AC_CHECK_LIB(bfd, main, [BFD_LIBS="-lbfd"], [])
-AC_CHECK_LIB(iberty, main, [IBERTY_LIBS="-liberty"], [])
-EDEBUG_LIBS="$BFD_LIBS $IBERTY_LIBS"
-AC_SUBST(EDEBUG_LIBS)
-if test "$EDEBUG_LIBS" = "-lbfd -liberty"; then
-    AC_DEFINE(HAVE_EDEBUG, 1, [Enhanced debug enabled])
-fi
+#AC_CHECK_LIB(bfd, main, [BFD_LIBS="-lbfd"], [])
+#AC_CHECK_LIB(iberty, main, [IBERTY_LIBS="-liberty"], [])
+#EDEBUG_LIBS="$BFD_LIBS $IBERTY_LIBS"
+#AC_SUBST(EDEBUG_LIBS)
+#if test "$EDEBUG_LIBS" = "-lbfd -liberty"; then
+#    AC_DEFINE(HAVE_EDEBUG, 1, [Enhanced debug enabled])
+#fi
 
 dnl ****************************************************************************
 dnl Check for XInput2
