diff -Naur celestia-1.6.0//src/celengine/mesh.cpp celestia-1.6.0-2//src/celengine/mesh.cpp
--- celestia-1.6.0//src/celengine/mesh.cpp	2009-06-22 17:44:08.000000000 +0200
+++ celestia-1.6.0-2//src/celengine/mesh.cpp	2010-05-29 18:57:24.899450275 +0200
@@ -7,6 +7,7 @@
 // as published by the Free Software Foundation; either version 2
 // of the License, or (at your option) any later version.
 
+#include <cstdlib>
 #include "mesh.h"
 #include "rendcontext.h"
 #include "gl.h"
diff -Naur celestia-1.6.0//src/celengine/model.cpp celestia-1.6.0-2//src/celengine/model.cpp
--- celestia-1.6.0//src/celengine/model.cpp	2009-06-22 17:44:08.000000000 +0200
+++ celestia-1.6.0-2//src/celengine/model.cpp	2010-05-29 19:01:18.767450834 +0200
@@ -7,6 +7,7 @@
 // as published by the Free Software Foundation; either version 2
 // of the License, or (at your option) any later version.
 
+#include <cstdlib>
 #include "model.h"
 #include "rendcontext.h"
 #include <cassert>
diff -Naur celestia-1.6.0//src/celengine/overlay.h celestia-1.6.0-2//src/celengine/overlay.h
--- celestia-1.6.0//src/celengine/overlay.h	2009-06-22 17:44:08.000000000 +0200
+++ celestia-1.6.0-2//src/celengine/overlay.h	2010-05-29 18:55:54.478473739 +0200
@@ -10,6 +10,7 @@
 #ifndef _OVERLAY_H_
 #define _OVERLAY_H_
 
+#include <stdio.h>
 #include <string>
 #include <iostream>
 #include <celtxf/texturefont.h>
diff -Naur celestia-1.6.0//src/celengine/star.cpp celestia-1.6.0-2//src/celengine/star.cpp
--- celestia-1.6.0//src/celengine/star.cpp	2009-06-22 17:44:08.000000000 +0200
+++ celestia-1.6.0-2//src/celengine/star.cpp	2010-05-29 18:55:54.479450556 +0200
@@ -8,6 +8,7 @@
 // of the License, or (at your option) any later version.
 
 #include <celmath/mathlib.h>
+#include <cstdio>
 #include <cstring>
 #include <cassert>
 #include "celestia.h"
diff -Naur celestia-1.6.0//src/celestia/celx_object.cpp celestia-1.6.0-2//src/celestia/celx_object.cpp
--- celestia-1.6.0//src/celestia/celx_object.cpp	2010-05-29 19:08:30.294451113 +0200
+++ celestia-1.6.0-2//src/celestia/celx_object.cpp	2009-06-22 17:44:22.000000000 +0200
@@ -773,7 +773,7 @@
         markAlpha = 1.0f;
     
     Color markColorAlpha(0.0f, 1.0f, 0.0f, 0.9f);
-    markColorAlpha = Color::Color(markColor, markAlpha);
+    markColorAlpha = Color(markColor, markAlpha);
     
     const char* markLabel = celx.safeGetString(6, WrongType, "Fifth argument to object:mark must be a string");
     if (markLabel == NULL)
