diff -Naur notecase-1.1.9.orig/Makefile notecase-1.1.9/Makefile
--- notecase-1.1.9.orig/Makefile	2006-09-01 11:20:08.000000000 +0200
+++ notecase-1.1.9/Makefile	2006-09-01 11:23:24.000000000 +0200
@@ -245,24 +245,24 @@
 		name_len=`echo "$$name_len - 3" | bc`; \
 		loc=`echo $$loc | cut -c -$$name_len`; \
 		msgfmt $$i -o po/messages.mo; \
-		install -D -m 644 po/messages.mo "$(INSTALL)$(DATA_DIR)/locale/$$loc/LC_MESSAGES/notecase.mo"; \
+		install -D -m 644 po/messages.mo "$(INSTALL)$(DESTDIR)$(DATA_DIR)/locale/$$loc/LC_MESSAGES/notecase.mo"; \
 	done; \
 	rm -f po/messages.mo; 
 	@echo done installing translations
 
 pouninstall:
-	rm -vf $(DATA_DIR)/locale/*/LC_MESSAGES/notecase.mo
+	rm -vf $(DESTDIR)$(DATA_DIR)/locale/*/LC_MESSAGES/notecase.mo
 	@echo done uninstalling translations
 	
 #
 # install application
 #
 install: notecase poinstall
-	install -D -m 755 $(BIN)/notecase "$(INSTALL)$(BINARY_DIR)/notecase"
-	install -D -m 644 docs/help.ncd "$(INSTALL)$(DATA_DIR)/doc/notecase/help.ncd"
-	install -D -m 644 docs/notecase.desktop "$(INSTALL)$(DATA_DIR)/applications/notecase.desktop"
-	install -D -m 644 docs/notecase.xml "$(INSTALL)$(DATA_DIR)/mime/packages/notecase.xml"
-	install -D -m 644 res/notecase.xpm "$(INSTALL)$(DATA_DIR)/icons/notecase.xpm"
+	install -D -m 755 $(BIN)/notecase "$(INSTALL)$(DESTDIR)$(BINARY_DIR)/notecase"
+	install -D -m 644 docs/help.ncd "$(INSTALL)$(DESTDIR)$(DATA_DIR)/doc/notecase/help.ncd"
+	install -D -m 644 docs/notecase.desktop "$(INSTALL)$(DESTDIR)$(DATA_DIR)/applications/notecase.desktop"
+	install -D -m 644 docs/notecase.xml "$(INSTALL)$(DESTDIR)$(DATA_DIR)/mime/packages/notecase.xml"
+	install -D -m 644 res/notecase.xpm "$(INSTALL)$(DESTDIR)$(DATA_DIR)/icons/notecase.xpm"
 #TOFIX: how to prevent "fedora-" prefix?
 #	desktop-file-install --vendor=fedora \
 #		--dir=$(datadir)/applications \
@@ -278,9 +278,9 @@
 #
 uninstall:	pouninstall
 	rm -f  "$(BINARY_DIR)/notecase"
-	rm -f  "$(DATA_DIR)/applications/notecase.desktop"
-	rm -f  "$(DATA_DIR)/icons/notecase.xpm"
-	rm -rf "$(DATA_DIR)/doc/notecase"
+	rm -f  "$(DESTDIR)$(DATA_DIR)/applications/notecase.desktop"
+	rm -f  "$(DESTDIR)$(DATA_DIR)/icons/notecase.xpm"
+	rm -rf "$(DESTDIR)$(DATA_DIR)/doc/notecase"
 	@echo done uninstalling
 
 #
@@ -323,12 +323,6 @@
 convertsrc:
 	unix2dos -q -k $(SRCPATH)/*.cpp $(SRCPATH)/*.h $(GUIPATH)/*.cpp $(GUIPATH)/*.h $(GUIPATH)/gui/*.cpp $(GUIPATH)/gui/*.h
 
-#
-# - update Linux mime/desktop system
-#
-updatemime:
-	update-mime-database "$(DATA_DIR)/mime/"
-	update-desktop-database
 
 #
 # for debugging purposes
