diff -Naur inetlib-0.4/src/irc_abstract.c inetlib-0.4.fix/src/irc_abstract.c
--- inetlib-0.4/src/irc_abstract.c	2004-07-14 03:28:11.000000000 +0200
+++ inetlib-0.4.fix/src/irc_abstract.c	2006-06-26 02:55:24.000000000 +0200
@@ -228,15 +228,14 @@
 
     socks = _misc_build_select_list(&(server->sock));
 
-    if( t_out != -1 )
+    if( t_out != -1 ) {
     
-       return select( (server->sock) + 1, &socks, (fd_set *) 0, (fd_set *) 0, &timeout);
-    
-    else {
        timeout.tv_sec = t_out;
        timeout.tv_usec = 0;
+       return select( (server->sock) + 1, &socks, (fd_set *) 0, (fd_set *) 0, &timeout);
+    
+    } else
 
        return select( (server->sock) + 1, &socks, (fd_set *) 0, (fd_set *) 0, NULL);
-    }
 
 }
