diff -Naur linux-5.3.1/drivers/platform/x86/acer-wmi.c linux-5.3.1-acer/drivers/platform/x86/acer-wmi.c
--- linux-5.3.1/drivers/platform/x86/acer-wmi.c	2019-09-21 07:19:47.000000000 +0200
+++ linux-5.3.1-acer/drivers/platform/x86/acer-wmi.c	2019-09-27 16:10:57.372620915 +0200
@@ -237,6 +237,7 @@
 static int force_series;
 static bool ec_raw_mode;
 static bool has_type_aa;
+static bool be_silent;
 static u16 commun_func_bitmap;
 static u8 commun_fn_key_number;
 
@@ -245,11 +246,14 @@
 module_param(threeg, int, 0444);
 module_param(force_series, int, 0444);
 module_param(ec_raw_mode, bool, 0444);
+module_param(be_silent, bool, 0444);
+
 MODULE_PARM_DESC(mailled, "Set initial state of Mail LED");
 MODULE_PARM_DESC(brightness, "Set initial LCD backlight brightness");
 MODULE_PARM_DESC(threeg, "Set initial state of 3G hardware");
 MODULE_PARM_DESC(force_series, "Force a different laptop series");
 MODULE_PARM_DESC(ec_raw_mode, "Enable EC raw mode");
+MODULE_PARM_DESC(be_silent, "Disables Unknow functions messages");
 
 struct acer_data {
 	int mailled;
@@ -1773,8 +1777,10 @@
 		acer_gsensor_event();
 		break;
 	default:
-		pr_warn("Unknown function number - %d - %d\n",
-			return_value.function, return_value.key_num);
+		if (!be_silent) {
+			pr_warn("Unknown function number - %d - %d\n",
+				return_value.function, return_value.key_num);
+		}
 		break;
 	}
 }
