diff -Naur linux-5.9/drivers/cpufreq/acpi-cpufreq.c linux-5.9-p/drivers/cpufreq/acpi-cpufreq.c
--- linux-5.9/drivers/cpufreq/acpi-cpufreq.c	2020-10-11 23:15:50.000000000 +0200
+++ linux-5.9-p/drivers/cpufreq/acpi-cpufreq.c	2020-10-12 01:20:28.611218883 +0200
@@ -691,7 +691,8 @@
 		cpumask_copy(policy->cpus, topology_core_cpumask(cpu));
 	}
 
-	if (check_amd_hwpstate_cpu(cpu) && !acpi_pstate_strict) {
+	if (check_amd_hwpstate_cpu(cpu) && (c->x86 < 0x19) &&
+	    !acpi_pstate_strict) {
 		cpumask_clear(policy->cpus);
 		cpumask_set_cpu(cpu, policy->cpus);
 		cpumask_copy(data->freqdomain_cpus,
