diff --git a/drivers/iommu/amd/init.c b/drivers/iommu/amd/init.c
index c652f16eb702..b52ea079d555 100644
--- a/drivers/iommu/amd/init.c
+++ b/drivers/iommu/amd/init.c
@@ -1680,7 +1680,7 @@ static void init_iommu_perf_ctr(struct amd_iommu *iommu)
 	return;
 
 pc_false:
-	pci_err(pdev, "Unable to read/write to IOMMU perf counter.\n");
+	pci_info(pdev, "Unable to read/write to IOMMU perf counter.\n");
 	amd_iommu_pc_present = false;
 	return;
 }
