diff -Naur linux-5.11.3/drivers/gpu/drm/amd/display/dc/core/dc_link.c linux-5.11.3-p/drivers/gpu/drm/amd/display/dc/core/dc_link.c
--- linux-5.11.3/drivers/gpu/drm/amd/display/dc/core/dc_link.c	2021-03-04 12:15:45.000000000 +0100
+++ linux-5.11.3-p/drivers/gpu/drm/amd/display/dc/core/dc_link.c	2021-03-06 14:03:25.611661668 +0100
@@ -2566,7 +2566,6 @@
 			if (pipe_ctx->plane_state == NULL)
 				frame_ramp = 0;
 		} else {
-			ASSERT(false);
 			return false;
 		}
 
