diff -Naur linux-5.11.3/drivers/gpu/drm/amd/pm/swsmu/smu11/arcturus_ppt.c linux-5.11.3-p/drivers/gpu/drm/amd/pm/swsmu/smu11/arcturus_ppt.c
--- linux-5.11.3/drivers/gpu/drm/amd/pm/swsmu/smu11/arcturus_ppt.c	2021-03-04 12:15:45.000000000 +0100
+++ linux-5.11.3-p/drivers/gpu/drm/amd/pm/swsmu/smu11/arcturus_ppt.c	2021-03-05 20:07:51.061694191 +0100
@@ -1310,7 +1310,7 @@
 						       CMN2ASIC_MAPPING_WORKLOAD,
 						       profile_mode);
 	if (workload_type < 0) {
-		dev_err(smu->adev->dev, "Unsupported power profile mode %d on arcturus\n", profile_mode);
+		dev_dbg(smu->adev->dev, "Unsupported power profile mode %d on arcturus\n", profile_mode);
 		return -EINVAL;
 	}
 
diff -Naur linux-5.11.3/drivers/gpu/drm/amd/pm/swsmu/smu12/renoir_ppt.c linux-5.11.3-p/drivers/gpu/drm/amd/pm/swsmu/smu12/renoir_ppt.c
--- linux-5.11.3/drivers/gpu/drm/amd/pm/swsmu/smu12/renoir_ppt.c	2021-03-04 12:15:45.000000000 +0100
+++ linux-5.11.3-p/drivers/gpu/drm/amd/pm/swsmu/smu12/renoir_ppt.c	2021-03-05 20:07:51.062694197 +0100
@@ -676,7 +676,7 @@
 		 * TODO: If some case need switch to powersave/default power mode
 		 * then can consider enter WORKLOAD_COMPUTE/WORKLOAD_CUSTOM for power saving.
 		 */
-		dev_err_once(smu->adev->dev, "Unsupported power profile mode %d on RENOIR\n", profile_mode);
+		dev_dbg(smu->adev->dev, "Unsupported power profile mode %d on RENOIR\n", profile_mode);
 		return -EINVAL;
 	}
 
