diff -Naur linux-4.12/drivers/net/wireless/ath/ath10k/htt_rx.c linux-4.12-p/drivers/net/wireless/ath/ath10k/htt_rx.c
--- linux-4.12/drivers/net/wireless/ath/ath10k/htt_rx.c	2017-07-03 01:07:02.000000000 +0200
+++ linux-4.12-p/drivers/net/wireless/ath/ath10k/htt_rx.c	2017-07-09 16:55:51.190109393 +0200
@@ -1501,7 +1501,6 @@
 	 */
 
 	if (!rx_status->freq) {
-		ath10k_warn(ar, "no channel configured; ignoring frame(s)!\n");
 		return false;
 	}
 
