diff -Naur linux-4.20/drivers/net/wireless/ath/ath10k/mac.c linux-4.20-ath10k-no-warn/drivers/net/wireless/ath/ath10k/mac.c
--- linux-4.20/drivers/net/wireless/ath/ath10k/mac.c	2018-12-24 00:55:59.000000000 +0100
+++ linux-4.20-ath10k-no-warn/drivers/net/wireless/ath/ath10k/mac.c	2018-12-27 11:15:32.530350969 +0100
@@ -5646,8 +5646,7 @@
 				    arvif->vdev_id, ret);
 	}
 
-	if (changed & BSS_CHANGED_MCAST_RATE &&
-	    !WARN_ON(ath10k_mac_vif_chan(arvif->vif, &def))) {
+	if (changed & BSS_CHANGED_MCAST_RATE && !ath10k_mac_vif_chan(arvif->vif, &def)) {
 		band = def.chan->band;
 		rateidx = vif->bss_conf.mcast_rate[band] - 1;
 
@@ -5685,7 +5684,7 @@
 	}
 
 	if (changed & BSS_CHANGED_BASIC_RATES) {
-		if (WARN_ON(ath10k_mac_vif_chan(vif, &def))) {
+		if (ath10k_mac_vif_chan(vif, &def)) {
 			mutex_unlock(&ar->conf_mutex);
 			return;
 		}
