diff -Naur linux-5.5.6/fs/namespace.c linux-5.5.6-p/fs/namespace.c
--- linux-5.5.6/fs/namespace.c	2020-02-24 08:38:51.000000000 +0100
+++ linux-5.5.6-p/fs/namespace.c	2020-02-27 11:09:12.337903446 +0100
@@ -2478,7 +2478,7 @@
 
 		time64_to_tm(sb->s_time_max, 0, &tm);
 
-		pr_warn("%s filesystem being %s at %s supports timestamps until %04ld (0x%llx)\n",
+		pr_debug("%s filesystem being %s at %s supports timestamps until %04ld (0x%llx)\n",
 			sb->s_type->name,
 			is_mounted(mnt) ? "remounted" : "mounted",
 			mntpath,
