diff -Naur linux-4.15.1/drivers/edac/amd64_edac.c linux-4.15.1-p/drivers/edac/amd64_edac.c
--- linux-4.15.1/drivers/edac/amd64_edac.c	2018-02-03 17:58:44.000000000 +0100
+++ linux-4.15.1-p/drivers/edac/amd64_edac.c	2018-02-12 01:52:10.411149240 +0100
@@ -3020,17 +3020,6 @@
 		amd64_warn("Error restoring NB MCGCTL settings!\n");
 }
 
-/*
- * EDAC requires that the BIOS have ECC enabled before
- * taking over the processing of ECC errors. A command line
- * option allows to force-enable hardware ECC later in
- * enable_ecc_error_reporting().
- */
-static const char *ecc_msg =
-	"ECC disabled in the BIOS or no ECC capability, module will not load.\n"
-	" Either enable ECC checking or force module loading by setting "
-	"'ecc_enable_override'.\n"
-	" (Note that use of the override may cause unknown side effects.)\n";
 
 static bool ecc_enabled(struct pci_dev *F3, u16 nid)
 {
@@ -3083,7 +3072,6 @@
 		   nid, (ecc_en ? "enabled" : "disabled"));
 
 	if (!ecc_en || !nb_mce_en) {
-		amd64_info("%s", ecc_msg);
 		return false;
 	}
 	return true;
