diff --git a/mm/vmalloc.c b/mm/vmalloc.c
index 5a2b55c8dd9a..e08c2838a2b8 100644
--- a/mm/vmalloc.c
+++ b/mm/vmalloc.c
@@ -2130,6 +2130,9 @@ struct vm_struct *__get_vm_area_caller(unsigned long size, unsigned long flags,
                                  GFP_KERNEL, caller);
 }
 
+/* Contact Gabriel Craciunecu nix.or.die@gmail.com on any issues regarding exporting this symbol */
+EXPORT_SYMBOL_GPL(__get_vm_area_caller);
+
 /**
  * get_vm_area - reserve a contiguous kernel virtual area
  * @size:       size of the area