diff --git a/drivers/gpu/drm/i915/intel_lvds.c b/drivers/gpu/drm/i915/intel_lvds.c
index 75a9772..0f607cb 100644
--- a/drivers/gpu/drm/i915/intel_lvds.c
+++ b/drivers/gpu/drm/i915/intel_lvds.c
@@ -629,6 +629,13 @@ static const struct dmi_system_id bad_lid_status[] = {
 			DMI_MATCH(DMI_PRODUCT_NAME, "PC-81005"),
 		},
 	},
+	{
+		.ident = "Kapok M450C",
+		.matches = {
+			DMI_MATCH(DMI_BOARD_VENDOR, "KAPOK"),
+			DMI_MATCH(DMI_PRODUCT_NAME, "M450C"),
+		},
+	},
 	{ }
 };
 
