diff -Naur linux-5.6/tools/bootconfig/Makefile linux-5.6-p/tools/bootconfig/Makefile
--- linux-5.6/tools/bootconfig/Makefile	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6-p/tools/bootconfig/Makefile	2020-03-30 18:12:59.306528380 +0200
@@ -1,10 +1,10 @@
 # SPDX-License-Identifier: GPL-2.0
 # Makefile for bootconfig command
 
-bindir ?= /usr/bin
+sbindir ?= /usr/sbin
 
 HEADER = include/linux/bootconfig.h
-CFLAGS = -Wall -g -I./include
+CFLAGS += -Wall -g -I./include
 
 PROGS = bootconfig
 
@@ -14,7 +14,8 @@
 	$(CC) $(filter %.c,$^) $(CFLAGS) -o $@
 
 install: $(PROGS)
-	install bootconfig $(DESTDIR)$(bindir)
+	mkdir -p $(DESTDIR)$(sbindir)
+	install -m 0755 bootconfig $(DESTDIR)$(sbindir)
 
 test: bootconfig
 	./test-bootconfig.sh
