--- linux-5.14/tools/bootconfig/main.c	2021-08-30 00:04:50.000000000 +0200
+++ linux-5.14.new/tools/bootconfig/main.c	2021-08-31 11:00:06.890942307 +0200
@@ -114,7 +114,7 @@
 
 	xbc_for_each_key_value(leaf, val) {
 		if (xbc_node_compose_key(leaf, key, XBC_KEYLEN_MAX) < 0) {
-			fprintf(stderr, "Failed to compose key %d\n", ret);
+			fprintf(stderr, "Failed to compose key \n");
 			break;
 		}
 		printf("%s = ", key);
