diff -Naur linux-4.16.5/arch/x86/kernel/cpu/bugs.c linux-4.16.5-p/arch/x86/kernel/cpu/bugs.c
--- linux-4.16.5/arch/x86/kernel/cpu/bugs.c	2018-04-26 11:00:39.000000000 +0200
+++ linux-4.16.5-p/arch/x86/kernel/cpu/bugs.c	2018-04-28 02:34:28.424306602 +0200
@@ -292,13 +292,13 @@
 	if ((!boot_cpu_has(X86_FEATURE_PTI) &&
 	     !boot_cpu_has(X86_FEATURE_SMEP)) || is_skylake_era()) {
 		setup_force_cpu_cap(X86_FEATURE_RSB_CTXSW);
-		pr_info("Spectre v2 mitigation: Filling RSB on context switch\n");
+		pr_info("Mitigation: Filling RSB on context switch\n");
 	}
 
 	/* Initialize Indirect Branch Prediction Barrier if supported */
 	if (boot_cpu_has(X86_FEATURE_IBPB)) {
 		setup_force_cpu_cap(X86_FEATURE_USE_IBPB);
-		pr_info("Spectre v2 mitigation: Enabling Indirect Branch Prediction Barrier\n");
+		pr_info("Mitigation: Enabling Indirect Branch Prediction Barrier\n");
 	}
 
 	/*
@@ -307,7 +307,7 @@
 	 */
 	if (boot_cpu_has(X86_FEATURE_IBRS)) {
 		setup_force_cpu_cap(X86_FEATURE_USE_IBRS_FW);
-		pr_info("Enabling Restricted Speculation for firmware calls\n");
+		pr_info("Mitigation: Enabling Restricted Speculation for firmware calls\n");
 	}
 }
 
