diff -Naur linux-5.9/drivers/net/wireless/intel/iwlwifi/fw/dbg.c linux-5.9-p/drivers/net/wireless/intel/iwlwifi/fw/dbg.c
--- linux-5.9/drivers/net/wireless/intel/iwlwifi/fw/dbg.c	2020-10-11 23:15:50.000000000 +0200
+++ linux-5.9-p/drivers/net/wireless/intel/iwlwifi/fw/dbg.c	2020-10-12 00:56:12.171166232 +0200
@@ -2554,7 +2554,7 @@
 		return -EINVAL;
 
 	if (fwrt->dump.conf != FW_DBG_INVALID)
-		IWL_INFO(fwrt, "FW already configured (%d) - re-configuring\n",
+		IWL_DEBUG_FW_INFO(fwrt, "FW already configured (%d) - re-configuring\n",
 			 fwrt->dump.conf);
 
 	/* Send all HCMDs for configuring the FW debug */
diff -Naur linux-5.9/drivers/net/wireless/intel/iwlwifi/pcie/trans.c linux-5.9-p/drivers/net/wireless/intel/iwlwifi/pcie/trans.c
--- linux-5.9/drivers/net/wireless/intel/iwlwifi/pcie/trans.c	2020-10-11 23:15:50.000000000 +0200
+++ linux-5.9-p/drivers/net/wireless/intel/iwlwifi/pcie/trans.c	2020-10-12 00:57:02.204539087 +0200
@@ -929,7 +929,7 @@
 		return;
 	}
 
-	IWL_INFO(trans, "Applying debug destination %s\n",
+	IWL_DEBUG_INFO(trans, "Applying debug destination %s\n",
 		 get_fw_dbg_mode_string(dest->monitor_mode));
 
 	if (dest->monitor_mode == EXTERNAL_MODE)
