--- linux-2.6.16.noarch/arch/i386/kernel/cpu/cpufreq/powernow-k7.c~	2006-06-05 13:50:27.000000000 -0400
+++ linux-2.6.16.noarch/arch/i386/kernel/cpu/cpufreq/powernow-k7.c	2006-06-05 13:54:32.000000000 -0400
@@ -582,9 +582,7 @@ static int __init powernow_cpu_init (str
 	rdmsrl (MSR_K7_FID_VID_STATUS, fidvidstatus.val);
 
 	/* recalibrate cpu_khz */
-	result = recalibrate_cpu_khz();
-	if (result)
-		return result;
+	recalibrate_cpu_khz();
 
 	fsb = (10 * cpu_khz) / fid_codes[fidvidstatus.bits.CFID];
 	if (!fsb) {
