diff -Naur linux-2.6.15-rc1-a/drivers/net/wireless/Kconfig linux-2.6.15-rc1-b/drivers/net/wireless/Kconfig
--- linux-2.6.15-rc1-a/drivers/net/wireless/Kconfig	2005-11-13 05:11:39.451031050 +0100
+++ linux-2.6.15-rc1-b/drivers/net/wireless/Kconfig	2005-11-13 05:12:36.834130145 +0100
@@ -217,6 +217,15 @@
           say M here and read <file:Documentation/modules.txt>.  The module
           will be called ipw2200.ko.
 
+config IPW2200_MONITOR
+        bool "Enable promiscuous mode"
+        depends on IPW2200
+        ---help---
+          Enables promiscuous/monitor mode support for the ipw2200 driver.
+          With this feature compiled into the driver, you can switch to
+          promiscuous mode via the Wireless Tool's Monitor mode.  While in this
+          mode, no packets can be sent.
+
 config IPW2200_DEBUG
 	bool "Enable full debugging output in IPW2200 module."
 	depends on IPW2200

