diff -Naur linux-5.5.11/Makefile linux-5.5.11-p/Makefile
--- linux-5.5.11/Makefile	2020-03-21 08:15:56.000000000 +0100
+++ linux-5.5.11-p/Makefile	2020-03-23 17:14:43.417918491 +0100
@@ -981,7 +981,7 @@
     mod_compress_cmd = gzip -n -f
   endif # CONFIG_MODULE_COMPRESS_GZIP
   ifdef CONFIG_MODULE_COMPRESS_XZ
-    mod_compress_cmd = xz -f
+    mod_compress_cmd = xz -T0 -f
   endif # CONFIG_MODULE_COMPRESS_XZ
 endif # CONFIG_MODULE_COMPRESS
 export mod_compress_cmd
