diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index 158438bb0389..734718e45aaa 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -5245,7 +5245,10 @@ intel_pps_verify_state(struct drm_i915_private *dev_priv,
 
 	if (hw.t1_t3 != sw->t1_t3 || hw.t8 != sw->t8 || hw.t9 != sw->t9 ||
 	    hw.t10 != sw->t10 || hw.t11_t12 != sw->t11_t12) {
-		DRM_ERROR("PPS state mismatch\n");
+                /* seem buggy on 4.14.x .. mute that for now
+                 * even is not a real solution ..
+		  * DRM_ERROR("PPS state mismatch\n");
+                 */
 		intel_pps_dump_state("sw", sw);
 		intel_pps_dump_state("hw", &hw);
 	}
