diff -up linux-4.16/drivers/gpu/drm/nouveau/nouveau_backlight.c.omv~ linux-4.16/drivers/gpu/drm/nouveau/nouveau_backlight.c
--- linux-4.16/drivers/gpu/drm/nouveau/nouveau_backlight.c.omv~	2018-04-06 01:04:34.573357055 +0200
+++ linux-4.16/drivers/gpu/drm/nouveau/nouveau_backlight.c	2018-04-06 01:05:46.985579248 +0200
@@ -287,6 +287,7 @@ nouveau_backlight_init(struct drm_device
 		case NV_DEVICE_INFO_V0_FERMI:
 		case NV_DEVICE_INFO_V0_KEPLER:
 		case NV_DEVICE_INFO_V0_MAXWELL:
+		case NV_DEVICE_INFO_V0_PASCAL:
 			return nv50_backlight_init(connector);
 		default:
 			break;
