diff -Naur linux-5.3.1/drivers/nvme/host/pci.c linux-5.3.1-p/drivers/nvme/host/pci.c
--- linux-5.3.1/drivers/nvme/host/pci.c	2019-09-21 07:19:47.000000000 +0200
+++ linux-5.3.1-p/drivers/nvme/host/pci.c	2019-09-27 14:46:38.567818352 +0200
@@ -2524,8 +2524,11 @@
 	}
 
 	result = nvme_init_identify(&dev->ctrl);
-	if (result)
+	if (result) {
+		dev_warn(dev->ctrl.device,
+			"nvme_init_identify() failed\n");
 		goto out;
+	}
 
 	if (dev->ctrl.oacs & NVME_CTRL_OACS_SEC_SUPP) {
 		if (!dev->ctrl.opal_dev)
@@ -2547,13 +2550,19 @@
 
 	if (dev->ctrl.hmpre) {
 		result = nvme_setup_host_mem(dev);
-		if (result < 0)
+		if (result < 0) {
+			dev_warn(dev->ctrl.device,
+				"hmpre - nvme_setup_host_mem() failed\n");
 			goto out;
+		}
 	}
 
 	result = nvme_setup_io_queues(dev);
-	if (result)
+	if (result) {
+		dev_warn(dev->ctrl.device,
+			"nvme_setup_io_queues() failed\n");
 		goto out;
+	}
 
 	/*
 	 * Keep the controller around but remove all namespaces if we don't have
