diff -Naur linux-5.2.14-nvme-resume/drivers/nvme/host/pci.c linux-5.2.14-nvme/drivers/nvme/host/pci.c
--- linux-5.2.14-nvme-resume/drivers/nvme/host/pci.c	2019-09-11 21:23:38.746030937 +0200
+++ linux-5.2.14-nvme/drivers/nvme/host/pci.c	2019-09-11 23:29:45.638351764 +0200
@@ -1072,6 +1072,8 @@
 	u16 start, end;
 	int found;
 
+	if (!nvme_cqe_pending(nvmeq))
+                return 0;
 	/*
 	 * For a poll queue we need to protect against the polling thread
 	 * using the CQ lock.  For normal interrupt driven threads we have
