diff -Naur linux-5.10.4/drivers/nvme/host/pci.c linux-5.10.4-p/drivers/nvme/host/pci.c
--- linux-5.10.4/drivers/nvme/host/pci.c	2020-12-30 11:54:29.000000000 +0100
+++ linux-5.10.4-p/drivers/nvme/host/pci.c	2021-01-01 02:02:04.011358821 +0100
@@ -3224,7 +3224,8 @@
 		.driver_data = NVME_QUIRK_SINGLE_VECTOR |
 				NVME_QUIRK_128_BYTES_SQES |
 				NVME_QUIRK_SHARED_TAGS },
-
+	{ PCI_DEVICE(0x1987, 0x5012),	/* Patriot Viper VPN100  */
+		.driver_data = NVME_QUIRK_IGNORE_DEV_SUBNQN, },
 	{ PCI_DEVICE_CLASS(PCI_CLASS_STORAGE_EXPRESS, 0xffffff) },
 	{ 0, }
 };
