diff -Naur linux-4.9.6/arch/x86/kernel/apic/io_apic.c linux-4.9.6-p/arch/x86/kernel/apic/io_apic.c
--- linux-4.9.6/arch/x86/kernel/apic/io_apic.c	2017-01-26 08:25:24.000000000 +0100
+++ linux-4.9.6-p/arch/x86/kernel/apic/io_apic.c	2017-02-06 05:03:35.311334756 +0100
@@ -1876,7 +1876,6 @@
 	.irq_ack		= irq_chip_ack_parent,
 	.irq_eoi		= ioapic_ack_level,
 	.irq_set_affinity	= ioapic_set_affinity,
-	.irq_retrigger		= irq_chip_retrigger_hierarchy,
 	.flags			= IRQCHIP_SKIP_SET_WAKE,
 };
 
@@ -1888,7 +1887,6 @@
 	.irq_ack		= irq_chip_ack_parent,
 	.irq_eoi		= ioapic_ir_ack_level,
 	.irq_set_affinity	= ioapic_set_affinity,
-	.irq_retrigger		= irq_chip_retrigger_hierarchy,
 	.flags			= IRQCHIP_SKIP_SET_WAKE,
 };
 
