diff -Naur linux-5.6.5/drivers/acpi/ec.c linux-5.6.5-revert/drivers/acpi/ec.c
--- linux-5.6.5/drivers/acpi/ec.c	2020-04-17 16:13:49.000000000 +0200
+++ linux-5.6.5-revert/drivers/acpi/ec.c	2020-04-17 23:38:43.830573202 +0200
@@ -1646,6 +1646,7 @@
 
 		if (boot_ec && ec->command_addr == boot_ec->command_addr &&
 		    ec->data_addr == boot_ec->data_addr) {
+			boot_ec_is_ecdt = false;
 			/*
 			 * Trust PNP0C09 namespace location rather than
 			 * ECDT ID. But trust ECDT GPE rather than _GPE
@@ -1665,12 +1666,9 @@
 
 	if (ec == boot_ec)
 		acpi_handle_info(boot_ec->handle,
-				 "Boot %s EC initialization complete\n",
+				 "Boot %s EC used to handle transactions and events\n",
 				 boot_ec_is_ecdt ? "ECDT" : "DSDT");
 
-	acpi_handle_info(ec->handle,
-			 "EC: Used to handle transactions and events\n");
-
 	device->driver_data = ec;
 
 	ret = !!request_region(ec->data_addr, 1, "EC data");
