diff -Naur linux-2.6.24/include/linux/Kbuild linux-2.6.24-revert/include/linux/Kbuild
--- linux-2.6.24/include/linux/Kbuild	2008-02-27 13:51:11.000000000 +0100
+++ linux-2.6.24-revert/include/linux/Kbuild	2008-02-27 13:48:47.000000000 +0100
@@ -217,7 +217,6 @@
 unifdef-y += icmp.h
 unifdef-y += icmpv6.h
 unifdef-y += if_addr.h
-unifdef-y += if_addrlabel.h
 unifdef-y += if_arp.h
 unifdef-y += if_bridge.h
 unifdef-y += if_ec.h
