diff -up linux-5.3/drivers/nvme/host/core.c.34~ linux-5.3/drivers/nvme/host/core.c
--- linux-5.3/drivers/nvme/host/core.c.34~	2019-09-16 18:35:34.144511992 +0200
+++ linux-5.3/drivers/nvme/host/core.c	2019-09-16 18:36:50.351828001 +0200
@@ -2267,6 +2267,15 @@ static const struct nvme_core_quirk_entr
 		.vid = 0x14a4,
 		.fr = "22301111",
 		.quirks = NVME_QUIRK_SIMPLE_SUSPEND,
+	},
+	{
+		/* https://forum.openmandriva.org/t/nvme-ssd-m2-not-seen-by-omlx-4-0/2407
+		 * Unknown SSD ... Maybe ADATA/Hynix (a similar mn from ADTA
+		 * but vid seems to be Hynix)
+		 */
+		.vid = 0x1c5c,
+		.mn = "HFM128GDHTNG-8310B",
+		.quirks = NVME_QUIRK_NO_APST,
 	}
 };
 
