diff -Naur linux-4.20.6/drivers/nvme/host/core.c linux-4.20.6-p/drivers/nvme/host/core.c
--- linux-4.20.6/drivers/nvme/host/core.c	2019-01-31 08:15:47.000000000 +0100
+++ linux-4.20.6-p/drivers/nvme/host/core.c	2019-02-04 22:32:28.182827035 +0100
@@ -2046,6 +2046,15 @@
 		.vid = 0x1179,
 		.mn = "THNSF5256GPUK TOSHIBA",
 		.quirks = NVME_QUIRK_NO_APST,
+	},
+
+	{
+		/* https://forum.openmandriva.org/t/nvme-ssd-m2-not-seen-by-omlx-4-0/2407
+		 * Unknow SSD .. Maybe ADATA/Hynix ( a similar mn from ADTA but vid seems to be Hynix)
+		*/
+		.vid = 0x1c5c,
+		.mn = "HFM128GDHTNG-8310B",
+		.quirks = NVME_QUIRK_NO_APST,
 	}
 };
 
