diff -Naur linux-5.6/fs/fs_parser.c linux-5.6-p/fs/fs_parser.c
--- linux-5.6/fs/fs_parser.c	2020-03-30 00:25:41.000000000 +0200
+++ linux-5.6-p/fs/fs_parser.c	2020-03-30 16:28:31.520724795 +0200
@@ -324,24 +324,24 @@
 	bool good = true;
 
 	if (tbl_size == 0) {
-		pr_warn("VALIDATE C-TBL: Empty\n");
+		pr_warn("fs-parser: VALIDATE C-TBL: Empty\n");
 		return true;
 	}
 
 	for (i = 0; i < tbl_size; i++) {
 		if (!tbl[i].name) {
-			pr_err("VALIDATE C-TBL[%zu]: Null\n", i);
+			pr_err("fs-parser: VALIDATE C-TBL[%zu]: Null\n", i);
 			good = false;
 		} else if (i > 0 && tbl[i - 1].name) {
 			int c = strcmp(tbl[i-1].name, tbl[i].name);
 
 			if (c == 0) {
-				pr_err("VALIDATE C-TBL[%zu]: Duplicate %s\n",
+				pr_err("fs-parser: VALIDATE C-TBL[%zu]: Duplicate %s\n",
 				       i, tbl[i].name);
 				good = false;
 			}
 			if (c > 0) {
-				pr_err("VALIDATE C-TBL[%zu]: Missorted %s>=%s\n",
+				pr_err("fs-parser: VALIDATE C-TBL[%zu]: Missorted %s>=%s\n",
 				       i, tbl[i-1].name, tbl[i].name);
 				good = false;
 			}
@@ -349,7 +349,7 @@
 
 		if (tbl[i].value != special &&
 		    (tbl[i].value < low || tbl[i].value > high)) {
-			pr_err("VALIDATE C-TBL[%zu]: %s->%d const out of range (%d-%d)\n",
+			pr_err("fs-parser: VALIDATE C-TBL[%zu]: %s->%d const out of range (%d-%d)\n",
 			       i, tbl[i].name, tbl[i].value, low, high);
 			good = false;
 		}
@@ -368,7 +368,7 @@
 	const struct fs_parameter_spec *param, *p2;
 	bool good = true;
 
-	pr_notice("*** VALIDATE %s ***\n", name);
+	pr_notice("fs-parser: VALIDATE %s\n", name);
 
 	for (param = desc; param->name; param++) {
 		/* Check for duplicate parameter names */
@@ -376,7 +376,7 @@
 			if (strcmp(param->name, p2->name) == 0) {
 				if (is_flag(param) != is_flag(p2))
 					continue;
-				pr_err("VALIDATE %s: PARAM[%s]: Duplicate\n",
+				pr_err("fs-parser: VALIDATE %s: PARAM[%s]: Duplicate\n",
 				       name, param->name);
 				good = false;
 			}
