--- epiphany-1.0.1/data/epiphany.schemas.in.foo	Fri Oct 10 14:23:48 2003
+++ epiphany-1.0.1/data/epiphany.schemas.in	Fri Oct 10 14:24:42 2003
@@ -5,7 +5,7 @@
         <applyto>/apps/epiphany/general/homepage</applyto>
         <owner>epiphany</owner>
         <type>string</type>
-        <default>http://www.google.com</default>
+        <default>http://www.frugalware.org/</default>
         <locale name="C">
         <short>Home page</short>
         <long>Address of the user's home page.</long>
