--- channels/chan_iax2.c	2007/12/20 20:21:41	94255
+++ channels/chan_iax2.c	2007/12/20 20:22:22	94256
@@ -2704,7 +2704,7 @@
 		 * is because we only have the IP address and the host field might be
 		 * set as a name (and the reverse PTR might not match).
 		 */
-		if (var) {
+		if (var && sin) {
 			for (tmp = var; tmp; tmp = tmp->next) {
 				if (!strcasecmp(tmp->name, "host")) {
 					struct in_addr sin2 = { 0, };
