diff -Naur acme-0.0.67/solver/preference.go acme-0.0.67-p/solver/preference.go
--- acme-0.0.67/solver/preference.go	2018-02-13 00:18:36.000000000 +0100
+++ acme-0.0.67-p/solver/preference.go	2019-02-26 22:18:29.570907437 +0100
@@ -80,9 +80,10 @@
 
 // PreferFast prefers fast types.
 var PreferFast = TypePreferencer{
-	"tls-sni-01": 1,
-	"http-01":    0,
+	"http-01":    1,
 
+	// Disable tls-sni-01 is gone ..
+	"tls-sni-01": -10,
 	// Disable DNS challenges for now. They're practically unusable and the Let's
 	// Encrypt live server doesn't support them at this time anyway.
 	"dns-01": -10,
