--- pandoc-types-1.23.1/pandoc-types.cabal	2001-09-09 03:46:40.000000000 +0200
+++ pandoc-types-1.23.1.new/pandoc-types.cabal	2024-07-10 20:16:29.860589814 +0200
@@ -50,13 +50,13 @@
   Build-depends:     base >= 4.5 && < 5,
                      containers >= 0.3,
                      text,
-                     deepseq >= 1.4.1 && < 1.5,
+                     deepseq >= 1.4.1 && < 1.6,
                      syb >= 0.1 && < 0.8,
                      ghc-prim >= 0.2,
                      bytestring >= 0.9 && < 0.13,
                      aeson >= 0.6.2 && < 2.3,
                      transformers >= 0.2 && < 0.7,
-                     QuickCheck >= 2.10 && < 2.15
+                     QuickCheck >= 2.10 && < 2.16
   if !impl(ghc >= 8.0)
     Build-depends:   semigroups == 0.18.*
   ghc-options:       -Wall
@@ -77,7 +77,7 @@
                        test-framework >= 0.3 && < 0.9,
                        test-framework-hunit >= 0.2 && < 0.4,
                        test-framework-quickcheck2 >= 0.2.9 && < 0.4,
-                       QuickCheck >= 2.10 && < 2.15,
+                       QuickCheck >= 2.10 && < 2.16,
                        HUnit >= 1.2 && < 1.7,
                        template-haskell >= 2
   ghc-options:         -threaded -rtsopts -with-rtsopts=-N -Wall -O2
