diff --git a/parser/python_header.h b/parser/python_header.h
index 90dea6a5..6fef957a 100644
--- a/parser/python_header.h
+++ b/parser/python_header.h
@@ -13,11 +13,6 @@
 
 #include <language/duchain/duchainlock.h>
 
-#include "pyport.h"
-#ifndef _WIN32
-#include "pyconfig.h"
-#endif
-
 #include "Python.h"
 
 #include "ast.h"
--- kdev-python-24.08.3/CMakeLists.txt	2024-10-31 01:08:38.000000000 +0100
+++ kdev-python-24.08.3.new/CMakeLists.txt	2024-11-19 10:02:36.110600483 +0100
@@ -42,7 +42,7 @@
 
 set(Python3_USE_STATIC_LIBS FALSE)
 set(Python3_FIND_STRATEGY VERSION)
-find_package(Python3 3.4.3...<3.13 COMPONENTS Interpreter Development REQUIRED)
+find_package(Python3 3.4.3...<3.14 COMPONENTS Interpreter Development REQUIRED)
 
 configure_file(kdevpythonversion.h.cmake "${CMAKE_CURRENT_BINARY_DIR}/kdevpythonversion.h" @ONLY)
 
