--- kdev-python-24.05.0/CMakeLists.txt	2024-05-17 00:44:18.000000000 +0200
+++ kdev-python-24.05.0.new/CMakeLists.txt	2024-05-29 02:07:48.353614295 +0200
@@ -43,7 +43,7 @@
 
 set(Python3_USE_STATIC_LIBS FALSE)
 set(Python3_FIND_STRATEGY VERSION)
-find_package(Python3 3.4.3...<3.12 COMPONENTS Interpreter Development REQUIRED)
+find_package(Python3 3.4.3...<3.13 COMPONENTS Interpreter Development REQUIRED)
 
 configure_file(kdevpythonversion.h.cmake "${CMAKE_CURRENT_BINARY_DIR}/kdevpythonversion.h" @ONLY)
 
