diff --git a/rfs.h b/rfs.h
index c6ff02b..ed0dd13 100644
--- a/rfs.h
+++ b/rfs.h
@@ -28,6 +28,7 @@
 #include <linux/wait.h>
 #include <linux/sched.h>
 #include <linux/quotaops.h>
+#include <linux/slab.h>
 #include "redirfs.h"
 
 #define RFS_ADD_OP(ops_new, op) \
diff --git a/rfs_inode.c b/rfs_inode.c
index 34a3f37..43f4142 100644
--- a/rfs_inode.c
+++ b/rfs_inode.c
@@ -752,7 +752,7 @@ static int rfs_setattr_default(struct dentry *dentry, struct iattr *iattr)
 
 	if ((iattr->ia_valid & ATTR_UID && iattr->ia_uid != inode->i_uid) ||
 	    (iattr->ia_valid & ATTR_GID && iattr->ia_gid != inode->i_gid))
-		return rfs_dq_transfer(inode, iattr) ? -EDQUOT : 0;
+		return dquot_transfer(inode, iattr) ? -EDQUOT : 0;
 
 	return inode_setattr(inode, iattr);
 }
