diff -Naur doublecmd-0.5.3.1/plugins/build.sh doublecmd-0.5.3.1.new/plugins/build.sh
--- doublecmd-0.5.3.1/plugins/build.sh	2011-08-06 08:19:06.000000000 +0100
+++ doublecmd-0.5.3.1.new/plugins/build.sh	2012-03-01 16:31:28.000000000 +0000
@@ -15,7 +15,7 @@
 $lazbuild wcx/lzma/src/lzma.lpi $DC_ARCH
 $lazbuild wcx/rpm/src/rpm.lpi $DC_ARCH
 $lazbuild wcx/unrar/src/unrar.lpi $DC_ARCH
-$lazbuild wcx/zip/src/Zip.lpi $DC_ARCH
+#$lazbuild wcx/zip/src/Zip.lpi $DC_ARCH
 
 # WDX plugins
 $lazbuild wdx/rpm_wdx/src/rpm_wdx.lpi $DC_ARCH
@@ -52,9 +52,9 @@
 strip --strip-all unrar.wcx
 cd $pluginsdir
 
-cd wcx/zip/lib/
-strip --strip-all zip.wcx
-cd $pluginsdir
+#cd wcx/zip/lib/
+#strip --strip-all zip.wcx
+#cd $pluginsdir
 
 # Strip and rename WDX
 cd wdx/rpm_wdx/lib/
