--- witherable-0.4.2/witherable.cabal	2001-09-09 03:46:40.000000000 +0200
+++ witherable-0.4.2.new/witherable.cabal	2023-03-17 12:11:53.085857917 +0100
@@ -24,7 +24,7 @@
     Witherable
     Data.Witherable
   build-depends:       base                          >=4.9      && <5,
-                       base-orphans                  >=0.8.4    && <0.9,
+                       base-orphans                  >=0.8.4    && <0.10,
                        containers                    >=0.5.7.1  && <0.7,
                        hashable                      >=1.2.7.0  && <1.4,
                        transformers                  >=0.5.2.0  && <0.6,
