--- witherable-0.5/witherable.cabal	2001-09-09 03:46:40.000000000 +0200
+++ witherable-0.5.new/witherable.cabal	2024-07-10 16:26:38.536304010 +0200
@@ -26,7 +26,7 @@
   build-depends:       base                          >=4.9      && <5,
                        base-orphans                  >=0.8.4    && <0.10,
                        containers                    >=0.5.7.1  && <0.8,
-                       hashable                      >=1.2.7.0  && <1.5,
+                       hashable                      >=1.2.7.0  && <1.6,
                        transformers                  >=0.5.2.0  && <0.7,
                        unordered-containers          >=0.2.12.0 && <0.3,
                        vector                        >=0.12.2.0 && <0.14,
