diff -Naur mono-tools-1.1.9.orig/configure.in mono-tools-1.1.9/configure.in
--- mono-tools-1.1.9.orig/configure.in	2005-09-23 15:41:38.000000000 +0200
+++ mono-tools-1.1.9/configure.in	2005-09-23 15:42:24.000000000 +0200
@@ -33,16 +33,16 @@
 PKG_CHECK_MODULES(NUNIT, mono-nunit)
 AC_SUBST(NUNIT_LIBS)
 
-#PKG_CHECK_MODULES(GTK_SHARP, glade-sharp-2.0 gnome-sharp-2.0 gconf-sharp-2.0 gtkhtml-sharp-2.0, enable_gtks=yes, enable_gtks=no)
-#if test "x$enable_gtks" = "xno" ; then
+PKG_CHECK_MODULES(GTK_SHARP, glade-sharp-2.0 gnome-sharp-2.0 gconf-sharp-2.0 gtkhtml-sharp-2.0, enable_gtks=yes, enable_gtks=no)
+if test "x$enable_gtks" = "xno" ; then
 	PKG_CHECK_MODULES(GTK_SHARP, glade-sharp gnome-sharp gconf-sharp gtkhtml-sharp, enable_gtks=yes, enable_gtks=no)
 	if test "x$enable_gtks" = "xno" ; then
 		AC_MSG_ERROR([gtk-sharp not found])
 	fi
-#fi
+fi
 AC_SUBST(GTK_SHARP_LIBS)
 
-PKG_CHECK_MODULES(GECKO_SHARP, gecko-sharp = 0.6, enable_gecko=yes, enable_gecko=no)
+PKG_CHECK_MODULES(GECKO_SHARP, gecko-sharp-2.0 >= 0.6, enable_gecko=yes, enable_gecko=no)
 AC_SUBST(GECKO_SHARP_LIBS)
 AM_CONDITIONAL(ENABLE_GECKO, test "x$enable_gecko" = "xyes")
 
