diff -aur xmlrpc-c-1.11.00/lib/curl_transport/xmlrpc_curl_transport.c fw_xmlrpc-c-1.11.00/lib/curl_transport/xmlrpc_curl_transport.c
--- xmlrpc-c-1.11.00/lib/curl_transport/xmlrpc_curl_transport.c	2007-05-14 08:26:23.000000000 +0200
+++ fw_xmlrpc-c-1.11.00/lib/curl_transport/xmlrpc_curl_transport.c	2008-10-04 12:20:20.000000000 +0200
@@ -557,7 +557,7 @@
             curl_easy_setopt(curlSessionP, CURLOPT_SSLENGINE,
                              curlSetupP->sslEngine);
         if (curlSetupP->sslEngineDefault)
-            curl_easy_setopt(curlSessionP, CURLOPT_SSLENGINE_DEFAULT);
+            curl_easy_setopt(curlSessionP, CURLOPT_SSLENGINE_DEFAULT, 1);
         if (curlSetupP->sslVersion != XMLRPC_SSLVERSION_DEFAULT)
             curl_easy_setopt(curlSessionP, CURLOPT_SSLVERSION,
                              curlSetupP->sslVersion);
