diff -aur xmlrpc-c-1.11.00/include/xmlrpc-c/oldcppwrapper.hpp fw_xmlrpc-c-1.11.00/include/xmlrpc-c/oldcppwrapper.hpp
--- xmlrpc-c-1.11.00/include/xmlrpc-c/oldcppwrapper.hpp	2005-06-10 23:48:43.000000000 +0200
+++ fw_xmlrpc-c-1.11.00/include/xmlrpc-c/oldcppwrapper.hpp	2008-10-04 12:31:01.000000000 +0200
@@ -25,6 +25,7 @@
 // -Bryan 2005.03.12.
 
 
+#include <cstdlib>
 #include <string>
 #include <xmlrpc-c/base.h>
 #include <xmlrpc-c/client.h>
diff -aur xmlrpc-c-1.11.00/src/cpp/client_simple.cpp fw_xmlrpc-c-1.11.00/src/cpp/client_simple.cpp
--- xmlrpc-c-1.11.00/src/cpp/client_simple.cpp	2006-02-04 20:25:18.000000000 +0100
+++ fw_xmlrpc-c-1.11.00/src/cpp/client_simple.cpp	2008-10-04 12:42:50.000000000 +0200
@@ -1,3 +1,4 @@
+#include <cstring>
 #include <string>
 
 #include "xmlrpc-c/girerr.hpp"
diff -aur xmlrpc-c-1.11.00/src/cpp/packetsocket.cpp fw_xmlrpc-c-1.11.00/src/cpp/packetsocket.cpp
--- xmlrpc-c-1.11.00/src/cpp/packetsocket.cpp	2007-05-14 01:30:43.000000000 +0200
+++ fw_xmlrpc-c-1.11.00/src/cpp/packetsocket.cpp	2008-10-04 12:40:53.000000000 +0200
@@ -49,6 +49,7 @@
 #include <fcntl.h>
 #include <poll.h>
 #include <sys/types.h>
+#include <cstdlib>
 
 #include "c_util.h"
 #include "xmlrpc-c/string_int.h"
diff -aur xmlrpc-c-1.11.00/src/cpp/server_abyss.cpp fw_xmlrpc-c-1.11.00/src/cpp/server_abyss.cpp
--- xmlrpc-c-1.11.00/src/cpp/server_abyss.cpp	2007-05-14 08:28:50.000000000 +0200
+++ fw_xmlrpc-c-1.11.00/src/cpp/server_abyss.cpp	2008-10-04 12:36:54.000000000 +0200
@@ -1,4 +1,5 @@
 #include <cassert>
+#include <cstdlib>
 #include <string>
 #include <memory>
 #include <signal.h>
diff -aur xmlrpc-c-1.11.00/src/cpp/value.cpp fw_xmlrpc-c-1.11.00/src/cpp/value.cpp
--- xmlrpc-c-1.11.00/src/cpp/value.cpp	2007-06-05 05:49:49.000000000 +0200
+++ fw_xmlrpc-c-1.11.00/src/cpp/value.cpp	2008-10-04 12:33:54.000000000 +0200
@@ -28,6 +28,7 @@
 
 *****************************************************************************/
 
+#include <cstdlib>
 #include <string>
 #include <vector>
 #include <ctime>
diff -aur xmlrpc-c-1.11.00/src/cpp/XmlRpcCpp.cpp fw_xmlrpc-c-1.11.00/src/cpp/XmlRpcCpp.cpp
--- xmlrpc-c-1.11.00/src/cpp/XmlRpcCpp.cpp	2005-11-24 04:13:13.000000000 +0100
+++ fw_xmlrpc-c-1.11.00/src/cpp/XmlRpcCpp.cpp	2008-10-04 12:31:36.000000000 +0200
@@ -24,6 +24,7 @@
 // SUCH DAMAGE.
 
 
+#include <cstring>
 #include <string>
 #include "xmlrpc-c/oldcppwrapper.hpp"
 
