diff -Naurp xmlrpc-c-1.11.00/include/xmlrpc-c/packetsocket.hpp xmlrpc-c-1.11.00-gcc4/include/xmlrpc-c/packetsocket.hpp
--- xmlrpc-c-1.11.00/include/xmlrpc-c/packetsocket.hpp	2007-05-14 01:30:43.000000000 +0200
+++ xmlrpc-c-1.11.00-gcc4/include/xmlrpc-c/packetsocket.hpp	2007-08-29 02:09:59.000000000 +0200
@@ -24,22 +24,22 @@ class packet : public girmem::autoObject
 public:
     packet();
 
-    packet::packet(const unsigned char * const data,
+    packet(const unsigned char * const data,
                    size_t                const dataLength);
 
-    packet::packet(const char * const data,
+    packet(const char * const data,
                    size_t       const dataLength);
 
     ~packet();
 
     unsigned char *
-    packet::getBytes() const { return this->bytes; }
+    getBytes() const { return this->bytes; }
 
     size_t
-    packet::getLength() const { return this->length; }
+    getLength() const { return this->length; }
 
     void
-    packet::addData(const unsigned char * const data,
+    addData(const unsigned char * const data,
                     size_t                const dataLength);
 
 private:
@@ -59,7 +59,7 @@ class packetPtr: public girmem::autoObje
 public:
     packetPtr();
 
-    explicit packetPtr::packetPtr(packet * const packetP);
+    explicit packetPtr(packet * const packetP);
 
     packet *
     operator->() const;
@@ -117,23 +117,23 @@ private:
     } escAccum;
 
     void
-    packetSocket::bufferFinishedPacket();
+    bufferFinishedPacket();
 
     void
-    packetSocket::takeSomeEscapeSeq(const unsigned char * const buffer,
+    takeSomeEscapeSeq(const unsigned char * const buffer,
                                     size_t                const length,
                                     size_t *              const bytesTakenP);
 
     void
-    packetSocket::takeSomePacket(const unsigned char * const buffer,
+    takeSomePacket(const unsigned char * const buffer,
                                  size_t                const length,
                                  size_t *              const bytesTakenP);
 
     void
-    packetSocket::verifyNothingAccumulated();
+    verifyNothingAccumulated();
 
     void
-    packetSocket::readFromFile();
+    readFromFile();
 
 };
 
diff -Naurp xmlrpc-c-1.11.00/include/xmlrpc-c/server_pstream.hpp xmlrpc-c-1.11.00-gcc4/include/xmlrpc-c/server_pstream.hpp
--- xmlrpc-c-1.11.00/include/xmlrpc-c/server_pstream.hpp	2007-05-14 01:30:43.000000000 +0200
+++ xmlrpc-c-1.11.00-gcc4/include/xmlrpc-c/server_pstream.hpp	2007-08-29 02:07:10.000000000 +0200
@@ -30,12 +30,12 @@ public:
         } present;
     };
 
-    serverPstreamConn::serverPstreamConn(constrOpt const& opt);
+    serverPstreamConn(constrOpt const& opt);
 
-    serverPstreamConn::~serverPstreamConn();
+    ~serverPstreamConn();
 
     void
-    serverPstreamConn::runOnce(bool * const eofP);
+    runOnce(bool * const eofP);
 
 private:
 
@@ -54,10 +54,10 @@ private:
         // This is permanently connected to our fixed client.
 
     void
-    serverPstreamConn::establishRegistry(constrOpt const& opt);
+    establishRegistry(constrOpt const& opt);
 
     void
-    serverPstreamConn::establishPacketSocket(constrOpt const& opt);
+    establishPacketSocket(constrOpt const& opt);
 };
 
 
diff -Naurp xmlrpc-c-1.11.00/src/cpp/packetsocket.cpp xmlrpc-c-1.11.00-gcc4/src/cpp/packetsocket.cpp
--- xmlrpc-c-1.11.00/src/cpp/packetsocket.cpp	2007-05-14 01:30:43.000000000 +0200
+++ xmlrpc-c-1.11.00-gcc4/src/cpp/packetsocket.cpp	2007-08-29 02:03:28.000000000 +0200
@@ -39,6 +39,7 @@
 
 #define _BSD_SOURCE        // gets uint defined
 
+#include <cassert>
 #include <string>
 #include <queue>
 #include <iostream>
diff -Naurp xmlrpc-c-1.11.00/src/cpp/pstream.cpp xmlrpc-c-1.11.00-gcc4/src/cpp/pstream.cpp
--- xmlrpc-c-1.11.00/src/cpp/pstream.cpp	2007-05-14 01:30:43.000000000 +0200
+++ xmlrpc-c-1.11.00-gcc4/src/cpp/pstream.cpp	2007-08-29 02:03:35.000000000 +0200
@@ -32,6 +32,8 @@ using girerr::throwf;
 
 #include "xmlrpc-c/client_transport.hpp"
 
+using namespace std;
+
 namespace xmlrpc_c {
 
 
diff -Naurp xmlrpc-c-1.11.00/src/cpp/server_pstream.cpp xmlrpc-c-1.11.00-gcc4/src/cpp/server_pstream.cpp
--- xmlrpc-c-1.11.00/src/cpp/server_pstream.cpp	2007-05-14 01:30:43.000000000 +0200
+++ xmlrpc-c-1.11.00-gcc4/src/cpp/server_pstream.cpp	2007-08-29 02:03:38.000000000 +0200
@@ -30,6 +30,8 @@ using girerr::throwf;
 
 #include "xmlrpc-c/server_pstream.hpp"
 
+using namespace std;
+
 namespace xmlrpc_c {
 
 
