diff -Naur xmlrpc-c-1.06.04/Makefile.common xmlrpc-c-1.06.04-p/Makefile.common
--- xmlrpc-c-1.06.04/Makefile.common	2006-08-30 04:59:05.000000000 +0200
+++ xmlrpc-c-1.06.04-p/Makefile.common	2006-09-02 00:24:52.178643032 +0200
@@ -32,7 +32,7 @@
 CXXFLAGS_COMMON = -DNDEBUG
 
 ifeq ($(C_COMPILER_GNU),yes)
-  CFLAGS_COMMON += $(GCC_C_WARNINGS) -fno-common -g -O3
+  CFLAGS_COMMON += $(GCC_C_WARNINGS) -fno-common -g
 endif
 
 ifeq ($(CXX_COMPILER_GNU),yes)
