diff -u xmlrpc-c-1.06.04/src/cpp/Makefile xmlrpc-c-1.06.04-p/src/cpp/Makefile
--- xmlrpc-c-1.06.04/src/cpp/Makefile	2006-06-19 16:16:46.000000000 +0200
+++ xmlrpc-c-1.06.04-p/src/cpp/Makefile	2006-09-02 00:30:10.422262648 +0200
@@ -136,7 +136,7 @@
 #  RULES TO COMPILE OBJECT MODULES FOR LIBRARIES
 #-----------------------------------------------------------------------------
 
-CXXFLAGS = $(CXXFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD)
+CXXFLAGS = $(CXXFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD) -fPIC
 
 XmlRpcCpp.o:%.o:%.cpp
 	$(CXX) -c $(BASIC_INCLUDES) $(CXXFLAGS) $<
Gemeinsame Unterverzeichnisse: xmlrpc-c-1.06.04/src/cpp/test und xmlrpc-c-1.06.04-p/src/cpp/test.
