diff -Naur xmlrpc-c-1.07.00/Makefile.common xmlrpc-c-1.07.00-p/Makefile.common
--- xmlrpc-c-1.07.00/Makefile.common	2006-09-29 04:40:34.000000000 +0200
+++ xmlrpc-c-1.07.00-p/Makefile.common	2006-10-13 02:10:11.000000000 +0200
@@ -325,59 +325,7 @@
 ##############################################################################
 #                         CROSS-COMPONENT BUILD RULES                        #
 ##############################################################################
-
-$(BLDDIR)/lib/wininet_transport/xmlrpc_wininet_transport.lo: FORCE
-	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/wininet_transport/Makefile \
-	    $(notdir $@)
-
-$(BLDDIR)/lib/curl_transport/xmlrpc_curl_transport.lo: FORCE
-	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/curl_transport/Makefile \
-	    $(notdir $@)
-
-$(BLDDIR)/lib/libwww_transport/xmlrpc_libwww_transport.lo: FORCE
-	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/libwww_transport/Makefile \
-	    $(notdir $@)
-
-$(LIBXMLRPC_ABYSS): FORCE
-	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/abyss/src/Makefile \
-	    $(notdir $@)
-
-$(LIBXMLRPC_XMLPARSE): FORCE
-	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/expat/xmlparse/Makefile \
-	    $(notdir $@)
-
-$(LIBXMLRPC_XMLTOK): FORCE
-	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/expat/xmltok/Makefile \
-	    $(notdir $@)
-
-$(LIBXMLRPC_UTIL): FORCE
-	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/libutil/Makefile \
-	    $(notdir $@)
-
-$(LIBXMLRPC) $(LIBXMLRPC_CLIENT) $(LIBXMLRPC_SERVER): FORCE
-	$(MAKE) -C $(dir $@) -f $(SRCDIR)/src/Makefile \
-	    $(notdir $@)
-
-$(LIBXMLRPC_ABYSS_A): FORCE
-	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/abyss/src/Makefile \
-	    $(notdir $@)
-
-$(LIBXMLRPCPP) $(LIBXMLRPC_CLIENTPP) $(LIBXMLRPC_SERVERPP) \
-  $(LIBXMLRPC_SERVER_ABYSSPP) $(LIBXMLRPC_CPP): FORCE
-	$(MAKE) -C $(dir $@) -f $(SRCDIR)/src/cpp/Makefile \
-	    $(notdir $@)
-
-$(LIBXMLRPC_UTIL_A):         $(LIBXMLRPC_UTIL)
-$(LIBXMLRPC_A):              $(LIBXMLRPC)
-$(LIBXMLRPC_CLIENT_A):       $(LIBXMLRPC_CLIENT)
-$(LIBXMLRPC_SERVER_ABYSS_A): $(LIBXMLRPC_SERVER_ABYSS)
-$(LIBXMLRPC_CGI_A):          $(LIBXMLRPC_CGI)
-$(LIBXMLRPC_XMLPARSE_A):     $(LIBXMLRPC_XMLPARSE)
-$(LIBXMLRPC_XMLTOK_A):       $(LIBXMLRPC_XMLTOK)
-
-$(UTILS): FORCE
-	$(MAKE) -C $(dir $@) -f $(SRCDIR)/lib/util/Makefile \
-	    $(notdir $@)
+## *HOW* this should ever work ?
 
 # About version.h:  This is a built header file, which means it is a supreme
 # pain in the ass.  The biggest problem is that when we automatically make
@@ -399,10 +347,6 @@
 
 endif
 
-$(BLDDIR)/include/xmlrpc-c/config.h:
-	$(MAKE) -C $(BLDDIR)/include -f $(SRCDIR)/include/Makefile \
-          xmlrpc-c/config.h
-
 ifneq ($(OMIT_TRANSPORT_CONFIG_H),Y)
 $(BLDDIR)/transport_config.h:
 	$(MAKE) -C $(dir $@) -f $(SRCDIR)/GNUmakefile $(notdir $@)
