Index: b/pdns/dnsparser.cc
===================================================================
--- a/pdns/dnsparser.cc
+++ b/pdns/dnsparser.cc
@@ -20,6 +20,7 @@
 #include "dnswriter.hh"
 #include <boost/lexical_cast.hpp>
 #include <boost/algorithm/string.hpp>
+#include <cstdio>
 
 using namespace boost;
 
Index: b/pdns/rcpgenerator.cc
===================================================================
--- a/pdns/rcpgenerator.cc
+++ b/pdns/rcpgenerator.cc
@@ -20,6 +20,7 @@
 #include "dnsparser.hh"
 #include "misc.hh"
 #include <boost/lexical_cast.hpp>
+#include <cstdio>
 #include <iostream>
 #include "base64.hh"
 using namespace boost;
