diff -Naur amarok-1.2.4/amarok/src/collectionbrowser.h amarok-1.2.4-gcc4/amarok/src/collectionbrowser.h
--- amarok-1.2.4/amarok/src/collectionbrowser.h	2005-05-19 19:00:19.000000000 +0200
+++ amarok-1.2.4-gcc4/amarok/src/collectionbrowser.h	2005-06-22 19:23:09.634976456 +0200
@@ -28,6 +28,8 @@
 class KPopupMenu;
 class KProgress;
 
+class CollectionView;
+
 class CollectionBrowser: public QVBox
 {
     Q_OBJECT
diff -Naur amarok-1.2.4/amarok/src/mediabrowser.h amarok-1.2.4-gcc4/amarok/src/mediabrowser.h
--- amarok-1.2.4/amarok/src/mediabrowser.h	2005-05-19 19:00:19.000000000 +0200
+++ amarok-1.2.4-gcc4/amarok/src/mediabrowser.h	2005-06-22 19:28:19.371889288 +0200
@@ -22,6 +22,8 @@
 class QPalette;
 class QPushButton;
 
+class MediaDeviceView;
+class MediaDevice;
 
 class MediaItem : public KListViewItem
 {
diff -Naur amarok-1.2.4/amarok/src/playlistbrowser.h amarok-1.2.4-gcc4/amarok/src/playlistbrowser.h
--- amarok-1.2.4/amarok/src/playlistbrowser.h	2005-05-19 19:00:19.000000000 +0200
+++ amarok-1.2.4-gcc4/amarok/src/playlistbrowser.h	2005-06-22 19:25:39.398208984 +0200
@@ -22,6 +22,12 @@
 class QSplitter;
 class QTimer;
 
+class PlaylistBrowserItem;
+class PlaylistBrowserView;
+class SmartPlaylistView;
+class TrackItemInfo;
+class PlaylistTrackItem;
+
 class PlaylistBrowser : public QVBox
 {
 Q_OBJECT
diff -Naur amarok-1.2.4/amarok/src/smartplaylisteditor.h amarok-1.2.4-gcc4/amarok/src/smartplaylisteditor.h
--- amarok-1.2.4/amarok/src/smartplaylisteditor.h	2005-05-19 19:00:19.000000000 +0200
+++ amarok-1.2.4-gcc4/amarok/src/smartplaylisteditor.h	2005-06-22 19:30:09.097208504 +0200
@@ -18,6 +18,7 @@
 class QToolButton;
 class QVGroupBox;
 
+class CriteriaEditor;
 
 class SmartPlaylistEditor : public KDialogBase
 {
