From ec725bc7f0524c6155d4e975729da5dd105c2b13 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Franti=C5=A1ek=20Zatloukal?= <fzatlouk@redhat.com>
Date: Fri, 26 Apr 2024 13:24:54 +0200
Subject: [PATCH] logging: Allow to use spdlog from system if present

---
 source/utils/CMakeLists.txt | 21 ++++++++++++---------
 1 file changed, 12 insertions(+), 9 deletions(-)

diff --git a/source/utils/CMakeLists.txt b/source/utils/CMakeLists.txt
index cb6cfb1..175376b 100644
--- a/source/utils/CMakeLists.txt
+++ b/source/utils/CMakeLists.txt
@@ -1,15 +1,18 @@
 # Copyright (C) 2024 Intel Corporation
 # SPDX-License-Identifier: MIT
 
-include(FetchContent)
-set(SPDLOG_REPO https://github.com/gabime/spdlog)
-set(SPDLOG_TAG v1.13.0)
-FetchContent_Declare(
-    spdlog
-    GIT_REPOSITORY ${SPDLOG_REPO}
-    GIT_TAG ${SPDLOG_TAG}
-)
-FetchContent_makeAvailable(spdlog)
+find_package(spdlog)
+if (NOT spdlog_FOUND)
+  include(FetchContent)
+  set(SPDLOG_REPO https://github.com/gabime/spdlog)
+  set(SPDLOG_TAG v1.13.0)
+  FetchContent_Declare(
+      spdlog
+      GIT_REPOSITORY ${SPDLOG_REPO}
+      GIT_TAG ${SPDLOG_TAG}
+  )
+  FetchContent_makeAvailable(spdlog)
+endif (NOT spdlog_FOUND)
 
 add_library(utils
   STATIC
