diff -Naur ibus-table-1.9.16/engine/Makefile.am ibus-table-1.9.16-p/engine/Makefile.am
--- ibus-table-1.9.16/engine/Makefile.am	2017-01-16 14:47:30.000000000 +0100
+++ ibus-table-1.9.16-p/engine/Makefile.am	2017-02-27 23:10:57.233743561 +0100
@@ -22,7 +22,7 @@
 DOC2MAN = docbook2man
 SGML = ibus-table-createdb.sgml
 BUILT_MANS = ibus-table-createdb.1
-man_MANS = $(BUILT_MANS)
+man_MANS =
 
 engine_table_PYTHON = \
 	chinese_variants.py \
@@ -48,8 +48,6 @@
 	ibus-table-createdb.in \
 	ibus-engine-table.in \
 	table.xml.in \
-	$(SGML) \
-	$(BUILT_MANS) \
 	$(NULL)
 
 CLEANFILES = \
@@ -64,11 +62,6 @@
 	Makefile.in \
 	$(NULL)
 
-${man_MANS}: ${SGML}
-	$(AM_V_GEN) $(RM) $@; \
-	$(DOC2MAN) ${SGML}; \
-	$(RM) manpage.*
-
 table.xml: table.xml.in
 	( \
 		libexecdir=${libexecdir}; \
