diff -Naur testng-7.7.1/build-logic/code-quality/build.gradle.kts testng-7.7.1.new/build-logic/code-quality/build.gradle.kts
--- testng-7.7.1/build-logic/code-quality/build.gradle.kts	2022-12-26 03:42:52.000000000 +0100
+++ testng-7.7.1.new/build-logic/code-quality/build.gradle.kts	2023-03-15 09:06:04.568767101 +0100
@@ -15,6 +15,6 @@
 
 tasks.withType<KotlinCompile>().configureEach {
     kotlinOptions {
-        jvmTarget = "11"
+        jvmTarget = "19"
     }
 }
diff -Naur testng-7.7.1/build-logic/jvm/build.gradle.kts testng-7.7.1.new/build-logic/jvm/build.gradle.kts
--- testng-7.7.1/build-logic/jvm/build.gradle.kts	2022-12-26 03:42:52.000000000 +0100
+++ testng-7.7.1.new/build-logic/jvm/build.gradle.kts	2023-03-15 09:05:36.149561866 +0100
@@ -17,6 +17,6 @@
 
 tasks.withType<KotlinCompile>().configureEach {
     kotlinOptions {
-        jvmTarget = "11"
+        jvmTarget = "19"
     }
 }
diff -Naur testng-7.7.1/build-logic/jvm/src/main/kotlin/testng.kotlin-library.gradle.kts testng-7.7.1.new/build-logic/jvm/src/main/kotlin/testng.kotlin-library.gradle.kts
--- testng-7.7.1/build-logic/jvm/src/main/kotlin/testng.kotlin-library.gradle.kts	2022-12-26 03:42:52.000000000 +0100
+++ testng-7.7.1.new/build-logic/jvm/src/main/kotlin/testng.kotlin-library.gradle.kts	2023-03-15 09:02:26.011189582 +0100
@@ -12,6 +12,6 @@
 
 tasks.withType<KotlinCompile>().configureEach {
     kotlinOptions {
-        jvmTarget = "11"
+        jvmTarget = "19"
     }
 }
diff -Naur testng-7.7.1/build-logic/publishing/build.gradle.kts testng-7.7.1.new/build-logic/publishing/build.gradle.kts
--- testng-7.7.1/build-logic/publishing/build.gradle.kts	2022-12-26 03:42:52.000000000 +0100
+++ testng-7.7.1.new/build-logic/publishing/build.gradle.kts	2023-03-15 09:05:49.882661039 +0100
@@ -17,6 +17,6 @@
 
 tasks.withType<KotlinCompile>().configureEach {
     kotlinOptions {
-        jvmTarget = "11"
+        jvmTarget = "19"
     }
 }
