--- type-equality-1.0.1/type-equality.cabal	2001-09-09 03:46:40.000000000 +0200
+++ type-equality-1.0.1.new/type-equality.cabal	2024-12-25 21:16:42.756526005 +0100
@@ -44,7 +44,7 @@
 
 library
   default-language: Haskell2010
-  build-depends:    base >=4.12 && <4.21
+  build-depends:    base >=4.12 && <4.22
   hs-source-dirs:   src-hetero
   exposed-modules:  Data.Type.Equality.Hetero
   other-extensions: PolyKinds
