--- xml-conduit-1.9.1.1/xml-conduit.cabal	2001-09-09 03:46:40.000000000 +0200
+++ xml-conduit-1.9.1.1.new/xml-conduit.cabal	2022-11-23 15:26:50.200116990 +0100
@@ -17,19 +17,19 @@
 tested-with:     GHC >=8.0 && <8.12
 
 custom-setup
-    setup-depends:   base >= 4 && <5, Cabal, cabal-doctest >= 1 && <1.1
+    setup-depends:   base >= 4 && <5, Cabal, cabal-doctest >= 1.0.9 && <1.1
 
 library
-    build-depends:   base                      >= 4        && < 5
+    build-depends:   base                      >= 4.12     && < 5
                    , conduit                   >= 1.3      && < 1.4
                    , conduit-extra             >= 1.3      && < 1.4
-                   , resourcet                 >= 1.2      && < 1.3
+                   , resourcet                 >= 1.2      && < 1.4
                    , bytestring                >= 0.10.2
                    , text                      >= 0.7
                    , containers                >= 0.2
                    , xml-types                 >= 0.3.4    && < 0.4
                    , attoparsec                >= 0.10
-                   , transformers              >= 0.2      && < 0.6
+                   , transformers              >= 0.2      && < 0.7
                    , data-default-class
                    , blaze-markup              >= 0.5
                    , blaze-html                >= 0.5
