diff -Naur pbzip2-1.0.1.orig/Makefile pbzip2-1.0.1/Makefile
--- pbzip2-1.0.1.orig/Makefile	2007-04-07 18:08:27.000000000 +0200
+++ pbzip2-1.0.1/Makefile	2007-04-07 18:09:30.000000000 +0200
@@ -5,17 +5,17 @@
 CC=g++
 
 # Where you want pbzip2 installed when you do 'make install'
-PREFIX=/usr
+PREFIX=$(DESTDIR)/usr
 
 all: pbzip2
 
 # Standard pbzip2 compile
 pbzip2: pbzip2.cpp
-	$(CC) -O3 -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -o pbzip2 pbzip2.cpp -pthread -lpthread -lbz2
+	$(CC) $(CFLAGS) -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -o pbzip2 pbzip2.cpp -pthread -lpthread -lbz2
 
 # Choose this if you want to compile in a static version of the libbz2 library
 pbzip2-static: libbz2.a pbzip2.cpp
-	$(CC) -O3 -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -o pbzip2 pbzip2.cpp -pthread -lpthread -I. -L. -lbz2
+	$(CC) $(CFLAGS) -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -o pbzip2 pbzip2.cpp -pthread -lpthread -I. -L. -lbz2
 
 # Choose this if you want to compile pbzip2 without load average support for
 # systems that do not support it
