diff -Naur guile-1.8.1/configure guile-1.8.1-p/configure
--- guile-1.8.1/configure	2006-10-07 05:54:24.000000000 +0200
+++ guile-1.8.1-p/configure	2006-10-07 18:25:05.000000000 +0200
@@ -11020,7 +11020,7 @@
   # Append ld.so.conf contents to the search path
   if test -f /etc/ld.so.conf; then
     lt_ld_extra=`awk '/^include / { system(sprintf("cd /etc; cat %s", \$2)); skip = 1; } { if (!skip) print \$0; skip = 0; }' < /etc/ld.so.conf | $SED -e 's/#.*//;s/[:,	]/ /g;s/=[^=]*$//;s/=[^= ]* / /g;/^$/d' | tr '\n' ' '`
-    sys_lib_dlsearch_path_spec="/lib /usr/lib $lt_ld_extra"
+    sys_lib_dlsearch_path_spec="/lib64 /usr/lib64 /lib /usr/lib $lt_ld_extra"
   fi
 
   # We used to test for /lib/ld.so.1 and disable shared libraries on
diff -Naur guile-1.8.1/guile-config/guile-config.in guile-1.8.1-p/guile-config/guile-config.in
--- guile-1.8.1/guile-config/guile-config.in	2006-09-20 02:19:02.000000000 +0200
+++ guile-1.8.1-p/guile-config/guile-config.in	2006-10-07 18:28:30.000000000 +0200
@@ -151,9 +151,9 @@
     (display (string-join
 	      (list
 	       (get-build-info 'CFLAGS)
-	       "-lguile -lltdl" 
+	       "-lguile" 
                (if (or (string=? libdir "/usr/lib")
-                       (string=? libdir "/usr/lib/"))
+                       (string=? libdir "/usr/lib64"))
 		   ""
 		   (string-append "-L" (get-build-info 'libdir)))
 	       (string-join other-flags)
diff -Naur guile-1.8.1/guile-readline/configure guile-1.8.1-p/guile-readline/configure
--- guile-1.8.1/guile-readline/configure	2006-10-07 05:54:41.000000000 +0200
+++ guile-1.8.1-p/guile-readline/configure	2006-10-07 18:25:05.000000000 +0200
@@ -10413,7 +10413,7 @@
   # Append ld.so.conf contents to the search path
   if test -f /etc/ld.so.conf; then
     lt_ld_extra=`awk '/^include / { system(sprintf("cd /etc; cat %s", \$2)); skip = 1; } { if (!skip) print \$0; skip = 0; }' < /etc/ld.so.conf | $SED -e 's/#.*//;s/[:,	]/ /g;s/=[^=]*$//;s/=[^= ]* / /g;/^$/d' | tr '\n' ' '`
-    sys_lib_dlsearch_path_spec="/lib /usr/lib $lt_ld_extra"
+    sys_lib_dlsearch_path_spec="/lib64 /usr/lib64 /lib /usr/lib $lt_ld_extra"
   fi
 
   # We used to test for /lib/ld.so.1 and disable shared libraries on
diff -Naur guile-1.8.1/libguile/Makefile.in guile-1.8.1-p/libguile/Makefile.in
--- guile-1.8.1/libguile/Makefile.in	2006-10-07 05:54:33.000000000 +0200
+++ guile-1.8.1-p/libguile/Makefile.in	2006-10-07 18:25:05.000000000 +0200
@@ -1908,8 +1908,8 @@
 	@echo '#define SCM_LIBRARY_DIR "$(pkgdatadir)/$(GUILE_EFFECTIVE_VERSION)"'>>libpath.tmp
 	@echo '#define SCM_SITE_DIR "$(pkgdatadir)/site"' >> libpath.tmp
 	@echo '#define SCM_BUILD_INFO { \' >> libpath.tmp
-	@echo '	{ "srcdir", "'"`cd @srcdir@; pwd`"'" }, \' >> libpath.tmp
-	@echo ' { "top_srcdir",    "@top_srcdir_absolute@" }, \' >> libpath.tmp
+	@echo '	{ "srcdir", "" }, \' >> libpath.tmp
+	@echo ' { "top_srcdir",    "" }, \' >> libpath.tmp
 	@echo '	{ "prefix",	   "@prefix@" }, \' >> libpath.tmp
 	@echo '	{ "exec_prefix",   "@exec_prefix@" }, \' >> libpath.tmp
 	@echo '	{ "bindir",	   "@bindir@" }, \' >> libpath.tmp
@@ -1930,7 +1930,7 @@
 	@echo '	{ "guileversion", "@GUILE_VERSION@" }, \' >> libpath.tmp
 	@echo '	{ "libguileinterface", "@LIBGUILE_INTERFACE@" }, \' \
 		>> libpath.tmp
-	@echo '	{ "LIBS",	   "@GUILE_LIBS@" }, \' >> libpath.tmp
+	@echo '	{ "LIBS",	   "" }, \' >> libpath.tmp
 	@echo '	{ "CFLAGS",	   "@GUILE_CFLAGS@" }, \' >> libpath.tmp
 	@echo '	{ "buildstamp",    "'"`date`"'" }, \' >> libpath.tmp
 	@echo '}' >> libpath.tmp
