--- guile-1.8.0/libguile/Makefile.in.deplibs	2006-02-20 22:30:07.000000000 +0100
+++ guile-1.8.0/libguile/Makefile.in	2006-05-18 11:19:15.000000000 +0200
@@ -1903,8 +1903,8 @@
 	@echo '#define SCM_LIBRARY_DIR "$(pkgdatadir)/$(GUILE_EFFECTIVE_VERSION)"'>>libpath.tmp
 	@echo '#define SCM_SITE_DIR "$(pkgdatadir)/site"' >> libpath.tmp
 	@echo '#define SCM_BUILD_INFO { \' >> libpath.tmp
-	@echo '	{ "srcdir", "'"`cd @srcdir@; pwd`"'" }, \' >> libpath.tmp
-	@echo ' { "top_srcdir",    "@top_srcdir_absolute@" }, \' >> libpath.tmp
+	@echo '	{ "srcdir", "" }, \' >> libpath.tmp
+	@echo ' { "top_srcdir",    "" }, \' >> libpath.tmp
 	@echo '	{ "prefix",	   "@prefix@" }, \' >> libpath.tmp
 	@echo '	{ "exec_prefix",   "@exec_prefix@" }, \' >> libpath.tmp
 	@echo '	{ "bindir",	   "@bindir@" }, \' >> libpath.tmp
@@ -1925,7 +1925,7 @@
 	@echo '	{ "guileversion", "@GUILE_VERSION@" }, \' >> libpath.tmp
 	@echo '	{ "libguileinterface", "@LIBGUILE_INTERFACE@" }, \' \
 		>> libpath.tmp
-	@echo '	{ "LIBS",	   "@GUILE_LIBS@" }, \' >> libpath.tmp
+	@echo '	{ "LIBS",	   "" }, \' >> libpath.tmp
 	@echo '	{ "CFLAGS",	   "@GUILE_CFLAGS@" }, \' >> libpath.tmp
 	@echo '	{ "buildstamp",    "'"`date`"'" }, \' >> libpath.tmp
 	@echo '}' >> libpath.tmp
--- guile-1.8.0/guile-config/guile-config.in.deplibs	2006-02-12 14:29:09.000000000 +0100
+++ guile-1.8.0/guile-config/guile-config.in	2006-05-18 11:21:38.000000000 +0200
@@ -151,8 +151,8 @@
     (display (string-join
 	      (list
 	       (get-build-info 'CFLAGS)
-	       "-lguile -lltdl" 
-	       (if (string=? libdir "/usr/lib/")
+	       "-lguile" 
+	       (if (or (string=? libdir "/usr/lib") (string=? libdir "/usr/lib64"))
 		   ""
 		   (string-append "-L" (get-build-info 'libdir)))
 	       (string-join other-flags)
