--- guile-1.8.0/libguile/private-gc.h.stacksize	2006-02-12 14:29:12.000000000 +0100
+++ guile-1.8.0/libguile/private-gc.h	2006-05-08 09:12:44.000000000 +0200
@@ -134,7 +134,7 @@
 #define SCM_MAX(A, B) ((A) > (B) ? (A) : (B))
 #define SCM_MIN(A, B) ((A) < (B) ? (A) : (B))
 
-#define CELL_P(x)  (SCM_ITAG3 (x) == scm_tc3_cons)
+#define CELL_P(x)  ((SCM_UNPACK(x) & (sizeof(scm_t_cell)-1)) == scm_tc3_cons)
 
 /*
   gc-mark
