--- numlock.cpp.orig	2012-03-04 16:21:34.000000000 +0100
+++ numlock.cpp	2012-03-04 16:24:49.000000000 +0100
@@ -68,9 +68,11 @@
         return;
 
     if( flag == true )
-        XkbLockModifiers ( dpy, XkbUseCoreKbd, mask, 0);
-    else
         XkbLockModifiers ( dpy, XkbUseCoreKbd, mask, mask);
+    else
+        XkbLockModifiers ( dpy, XkbUseCoreKbd, mask, 0);
+
+    XSync( dpy, false );
 }
 
 void NumLock::setOn(Display *dpy) {
