diff -Naur notecase-1.2.3.orig/Makefile notecase-1.2.3/Makefile
--- notecase-1.2.3.orig/Makefile	2006-10-13 19:26:37.000000000 +0200
+++ notecase-1.2.3/Makefile	2006-10-13 19:28:11.000000000 +0200
@@ -243,7 +243,7 @@
 		rm -f po/messages.mo; \
 		msgfmt $$i -o po/messages.mo; \
 		echo "Installing: $(INSTALL)$(DATA_DIR)/locale/$$loc/LC_MESSAGES/notecase.mo";\
-		install -D -m 644 po/messages.mo "$(INSTALL)$(DATA_DIR)/locale/$$loc/LC_MESSAGES/notecase.mo"; \
+		install -D -m 644 po/messages.mo "$(INSTALL)$(DESTDIR)$(DATA_DIR)/locale/$$loc/LC_MESSAGES/notecase.mo"; \
 	done; \
 	rm -f po/messages.mo; 
 	@echo Done installing translations
@@ -256,11 +256,11 @@
 # install application
 #
 install: notecase poinstall
-	install -D -m 755 $(BIN)/notecase "$(INSTALL)$(BINARY_DIR)/notecase"
-	install -D -m 644 docs/help.ncd "$(INSTALL)$(DATA_DIR)/doc/notecase/help.ncd"
-	install -D -m 644 docs/notecase.desktop "$(INSTALL)$(DATA_DIR)/applications/notecase.desktop"
-	install -D -m 644 docs/notecase.xml "$(INSTALL)$(DATA_DIR)/mime/packages/notecase.xml"
-	install -D -m 644 res/notecase.xpm "$(INSTALL)$(DATA_DIR)/icons/notecase.xpm"
+	install -D -m 755 $(BIN)/notecase "$(INSTALL)$(DESTDIR)$(BINARY_DIR)/notecase"
+	install -D -m 644 docs/help.ncd "$(INSTALL)$(DESTDIR)$(DATA_DIR)/doc/notecase/help.ncd"
+	install -D -m 644 docs/notecase.desktop "$(INSTALL)$(DESTDIR)$(DATA_DIR)/applications/notecase.desktop"
+	install -D -m 644 docs/notecase.xml "$(INSTALL)$(DESTDIR)$(DATA_DIR)/mime/packages/notecase.xml"
+	install -D -m 644 res/notecase.xpm "$(INSTALL)$(DESTDIR)$(DATA_DIR)/icons/notecase.xpm"
 #TOFIX: how to prevent "fedora-" prefix?
 #	desktop-file-install --vendor=fedora \
 #		--dir=$(datadir)/applications \
@@ -321,12 +321,6 @@
 convertsrc:
 	unix2dos -q -k $(SRCPATH)/*.cpp $(SRCPATH)/*.h $(GUIPATH)/*.cpp $(GUIPATH)/*.h $(GUIPATH)/gui/*.cpp $(GUIPATH)/gui/*.h
 
-#
-# - update Linux mime/desktop system
-#
-updatemime:
-	update-mime-database "$(DATA_DIR)/mime/"
-	update-desktop-database
 
 #
 # for debugging purposes
