diff -Naur ppracer-0.3.1/src/ppgltk/FT/FTTextureGlyph.h ppracer-0.3.1-p/src/ppgltk/FT/FTTextureGlyph.h
--- ppracer-0.3.1/src/ppgltk/FT/FTTextureGlyph.h	2005-02-18 11:16:04.000000000 +0100
+++ ppracer-0.3.1-p/src/ppgltk/FT/FTTextureGlyph.h	2006-09-01 18:09:32.817106352 +0200
@@ -52,7 +52,7 @@
          * Reset the currently active texture to zero to get into a known state before
          * drawing a string. This is to get round possible threading issues.
          */
-        static void FTTextureGlyph::ResetActiveTexture(){ activeTextureID = 0;}
+        static void ResetActiveTexture(){ activeTextureID = 0;}
         
     private:
         /**
