diff -aur librep-0.90.2/configure librep-0.90.2-2/configure
--- librep-0.90.2/configure	2009-08-25 11:33:59.000000000 -0500
+++ librep-0.90.2-2/configure	2009-09-08 04:43:58.000000000 -0500
@@ -11999,7 +11999,7 @@
   pkg_cv_LIBFFI_CFLAGS=`$PKG_CONFIG --cflags "libffi >= $FFI_MIN_VER
   " 2>/dev/null`
 else
-  pkg_failed=yes
+  pkg_failed=no
 fi
     fi
 else
@@ -12020,7 +12020,7 @@
   pkg_cv_LIBFFI_LIBS=`$PKG_CONFIG --libs "libffi >= $FFI_MIN_VER
   " 2>/dev/null`
 else
-  pkg_failed=yes
+  pkg_failed=no
 fi
     fi
 else
