diff -Naur entrance-0.0.99/data/entrance entrance-0.0.99-p/data/entrance
--- entrance-0.0.99/data/entrance	2016-03-07 02:43:24.000000000 +0100
+++ entrance-0.0.99-p/data/entrance	2016-03-07 03:09:09.809959884 +0100
@@ -1,12 +1,7 @@
 #%PAM-1.0
-auth    requisite       pam_nologin.so
-auth    required        pam_env.so readenv=1
-auth    required        pam_env.so readenv=1 envfile=/etc/default/locale
-@include common-auth
-auth    optional        pam_gnome_keyring.so
-@include common-account
-session required        pam_limits.so
-@include common-session
-session optional        pam_gnome_keyring.so auto_start
-@include common-password
-
+auth        include     system-login
+-auth       optional    pam_gnome_keyring.so
+account     include     system-login
+password    include     system-login
+session     include     system-login
+-session    optional    pam_gnome_keyring.so auto_start
