diff -Naur entrance-0.0.99/data/entrance.service.in entrance-0.0.99-p/data/entrance.service.in
--- entrance-0.0.99/data/entrance.service.in	2016-03-07 02:43:24.000000000 +0100
+++ entrance-0.0.99-p/data/entrance.service.in	2016-03-07 03:28:22.012674146 +0100
@@ -1,14 +1,13 @@
 [Unit]
 Description=Enlightenment Display Manager
-Conflicts=getty@tty1.service plymouth-quit.service
-After=systemd-user-sessions.service getty@tty1.service plymouth-quit.service
+After=systemd-user-sessions.service getty@tty1.service
 
 [Service]
 
 Environment="LD_LIBRARY_PATH=@libdir@"
 IgnoreSIGPIPE=no
 ExecStart=@SBINDIR@/entrance --nodaemon
-Restart= always
+Restart=always
 StandardOutput=syslog
 StandardError=inherit
 
diff -Naur entrance-0.0.99/Makefile.in entrance-0.0.99-p/Makefile.in
--- entrance-0.0.99/Makefile.in	2016-03-07 03:00:16.000000000 +0100
+++ entrance-0.0.99-p/Makefile.in	2016-03-07 03:27:50.051813844 +0100
@@ -544,7 +544,7 @@
 pc_verbose = $(pc_verbose_@AM_V@)
 pc_verbose_ = $(pc_verbose_@AM_DEFAULT_V@)
 pc_verbose_0 = @echo "  SED     " $@;
-@HAVE_SYSTEMD_TRUE@systemddir = /usr/lib/systemd/system/
+@HAVE_SYSTEMD_TRUE@systemddir = /lib/systemd/system/
 @HAVE_SYSTEMD_TRUE@systemd_DATA = data/entrance.service
 edje_process = \
 $(edje) \
