diff -Naur opentrack-opentrack-2.3.10/cmake/opentrack-hier.cmake opentrack-opentrack-2.3.10.new/cmake/opentrack-hier.cmake
--- opentrack-opentrack-2.3.10/cmake/opentrack-hier.cmake	2018-02-21 11:13:33.000000000 +0100
+++ opentrack-opentrack-2.3.10.new/cmake/opentrack-hier.cmake	2019-01-16 15:47:17.243441359 +0100
@@ -28,15 +28,15 @@
     set(opentrack-i18n-path "./i18n")
     set(opentrack-hier-debug "./debug")
 else()
-    set(opentrack-hier-pfx "libexec/opentrack")
-    set(opentrack-hier-path "/../libexec/opentrack/")   # MUST HAVE A TRAILING BACKSLASH
+    set(opentrack-hier-pfx "lib/opentrack")
+    set(opentrack-hier-path "/../lib/opentrack/")   # MUST HAVE A TRAILING BACKSLASH
     set(opentrack-hier-doc "/share/doc/opentrack/")     # MUST HAVE A TRAILING BACKSLASH
     set(opentrack-doc-pfx "./share/doc/opentrack")
     set(opentrack-doc-src-pfx "./share/doc/opentrack/source-code")
     set(opentrack-install-rpath "${CMAKE_INSTALL_PREFIX}/${opentrack-hier-pfx}")
     set(opentrack-hier-str ARCHIVE DESTINATION lib/opentrack LIBRARY DESTINATION ${opentrack-hier-pfx} RUNTIME DESTINATION bin)
-    set(opentrack-i18n-pfx "./libexec/opentrack/i18n")
-    set(opentrack-i18n-path "../libexec/opentrack/i18n")
+    set(opentrack-i18n-pfx "./lib/opentrack/i18n")
+    set(opentrack-i18n-path "../lib/opentrack/i18n")
 endif()
 
 function(otr_escape_string var str)
