--- opentrack-opentrack-2.3.12/cmake/opentrack-hier.cmake	2020-07-01 08:35:57.000000000 +0200
+++ opentrack-opentrack-2.3.12.new/cmake/opentrack-hier.cmake	2020-08-21 11:37:18.477966062 +0200
@@ -31,7 +31,7 @@
     set(opentrack-debug "./debug")
     set(opentrack-install-rpath "")
 else()
-    set(opentrack-libexec "libexec/opentrack")
+    set(opentrack-libexec "lib/opentrack")
     set(opentrack-runtime-libexec "/../${opentrack-libexec}/")   # MUST HAVE A TRAILING BACKSLASH
     set(opentrack-runtime-doc "/share/doc/opentrack/")         # MUST HAVE A TRAILING BACKSLASH
     set(opentrack-bin "bin")
