--- opentrack-opentrack-2023.3.0/compat/process-list.hpp	2023-11-06 14:48:28.000000000 +0100
+++ opentrack-opentrack-2023.3.0.new/compat/process-list.hpp	2024-12-19 11:27:30.098898458 +0100
@@ -150,7 +150,7 @@
 
     while ((stack = procps_pids_get(info, PIDS_FETCH_TASKS_ONLY)))
     {
-        char  **p_cmdline = PIDS_VAL(rel_cmdline, strv,  stack, info);
+        char  **p_cmdline = PIDS_VAL(rel_cmdline, strv,  stack);
 
         // note, wine sets argv[0] so no parsing like in OSX case
         if (p_cmdline && p_cmdline[0] && p_cmdline[0][0] &&
