--- opentrack-opentrack-2022.3.2/cmake/opentrack-install.cmake	2022-10-19 19:24:13.000000000 +0200
+++ opentrack-opentrack-2022.3.2.new/cmake/opentrack-install.cmake	2022-11-05 19:16:47.852598605 +0100
@@ -24,7 +24,6 @@
 otr_install_dir("${opentrack-doc}" ${CMAKE_SOURCE_DIR}/3rdparty-notices)
 otr_install_dir("${opentrack-doc}" "${CMAKE_SOURCE_DIR}/settings" "${CMAKE_SOURCE_DIR}/contrib")
 
-INSTALL(CODE "FILE(MAKE_DIRECTORY \${CMAKE_INSTALL_PREFIX}/\${opentrack-hier-libexec}/presets)")
 
 if(WIN32)
     otr_install_misc(. FILES "${CMAKE_SOURCE_DIR}/bin/qt.conf")
