diff --git a/patches/src680/apply b/patches/src680/apply
index 2dc5da8..71e85ae 100644
--- a/patches/src680/apply
+++ b/patches/src680/apply
@@ -402,7 +402,7 @@ tools-urlobj-smb-scheme-patch.diff, n#261623, flr
 tools-qa-urlobj-unittest.diff, n#261623, flr
 
 # honour disable-*-link
-config_office-honour-disable-link.diff, i#83088, fridrich
+#config_office-honour-disable-link.diff, i#83088, fridrich
 
 # misc mozilla checks
 config_office-misc-mozilla-checks.diff, i#83090, fridrich
