Index: patches/src680/apply
===================================================================
--- patches/src680/apply	(revision 11042)
+++ patches/src680/apply	(working copy)
@@ -260,7 +260,7 @@
 cws-npower7.diff
 
 # npower8 (depends) e.g. must come after npower7
-cws-npower8.diff
+#cws-npower8.diff
 
 # major re-work / speed-up of configmgr
 # FIXME: when integrated upstream, remove the hack in Makefile.shared
@@ -1386,7 +1386,7 @@
 # core changes not yet upstream
 sc-source-ui-vba-makefile-mk.diff
 # fix --disable-vba build
-oovbaapi-disable-vba.diff
+#oovbaapi-disable-vba.diff
 
 # export of macro bindings for shapes
 sc-export-shape-macro-bindings.diff, n#304739, noelpwer
