diff -Naur ooo-build-2.0.2.9.orig/distro-configs/Common.conf.in ooo-build-2.0.2.9/distro-configs/Common.conf.in
--- ooo-build-2.0.2.9.orig/distro-configs/Common.conf.in	2006-03-08 15:06:12.000000000 +0100
+++ ooo-build-2.0.2.9/distro-configs/Common.conf.in	2006-05-14 12:30:03.000000000 +0200
@@ -1,7 +1,5 @@
 --disable-binfilter
---disable-crashdump
 --disable-epm
---enable-crashdump=no
 --enable-openldap
 --with-build-version=\"Build @OOO_BUILDVERSION@\"
 --with-lang=\"@OOO_LANGS@\"
diff -Naur ooo-build-2.0.2.9.orig/distro-configs/Frugalware.conf.in ooo-build-2.0.2.9/distro-configs/Frugalware.conf.in
--- ooo-build-2.0.2.9.orig/distro-configs/Frugalware.conf.in	2005-12-20 13:06:57.000000000 +0100
+++ ooo-build-2.0.2.9/distro-configs/Frugalware.conf.in	2006-05-16 16:18:13.000000000 +0200
@@ -4,17 +4,19 @@
 --with-system-curl
 --with-system-expat
 --with-system-neon
---with-system-gcc=yes
 --with-system-sndfile
 --with-system-cairo
 --with-system-boost
 --with-system-hunspell
 --with-system-odbc-headers
 --with-system-sane-headers
---with-java-home=$JAVA_HOME
+--with-system-gcc=yes
+--with-system-nas
 --with-javac
 --with-firefox
---without-java
+--with-java=gij
+--with-openclipart=/usr/share/openclipart
+--with-docdir=/usr/share/doc/openoffice.org
 --enable-libart
 --enable-libsn
 --enable-fontconfig
@@ -27,5 +29,5 @@
 --enable-cairo
 --enable-xsltproc
 --disable-rpath
---disable-mozilla 
+--disable-mozilla
 --disable-odk
diff -Naur ooo-build-2.0.2.9.orig/patches/src680/apply ooo-build-2.0.2.9/patches/src680/apply
--- ooo-build-2.0.2.9.orig/patches/src680/apply	2006-05-04 14:48:36.000000000 +0200
+++ ooo-build-2.0.2.9/patches/src680/apply	2006-05-16 16:19:13.000000000 +0200
@@ -61,7 +61,7 @@
 Mandriva64: MandrivaBase, 64bit
 # Frugalware Linux
 FrugalwareBase: LinuxCommon, FilePickers, GnomeVFS, GlobalDesktopFiles, NonArkCommon, NoRpath, NovellOnly, NovellOnlyUnix, NotDebian
-Frugalware: FrugalwareBase, FrugalwareOnly
+Frugalware: FrugalwareBase, FrugalwareOnly, GCJExperimental, SystemJavaBits
 Frugalware64: FrugalwareBase, 64bit, Frugalware64Only
 QiLinux: QiLinuxOnly, LinuxCommon, GlobalDesktopFiles, FilePickers
 #Gentoo
@@ -594,7 +594,7 @@
 [ FrugalwareOnly ]
 frugalware.diff
 xmlhelp-port-to-db-4.4.diff
-store-drop-gcc-libs-on-linking.diff
+gcc401-stlport45-include.diff
 
 [ Frugalware64Only ]
 frugalware64.diff
