diff -Naur ooo-build.orig/distro-configs/Frugalware64.conf.in ooo-build/distro-configs/Frugalware64.conf.in
--- ooo-build.orig/distro-configs/Frugalware64.conf.in	2006-05-02 18:38:32.000000000 +0200
+++ ooo-build/distro-configs/Frugalware64.conf.in	2006-05-20 02:15:33.000000000 +0200
@@ -19,6 +19,8 @@
 --with-system-xml-apis
 --with-system-nas
 --with-java=gij
+--with-openclipart=/usr/share/openclipart
+--with-docdir=/usr/share/doc/openoffice.org
 --enable-libart
 --enable-libsn
 --enable-kde
diff -Naur ooo-build.orig/distro-configs/Frugalware.conf.in ooo-build/distro-configs/Frugalware.conf.in
--- ooo-build.orig/distro-configs/Frugalware.conf.in	2006-05-02 18:38:32.000000000 +0200
+++ ooo-build/distro-configs/Frugalware.conf.in	2006-05-20 02:14:51.000000000 +0200
@@ -18,6 +18,8 @@
 --with-system-xerces
 --with-system-xml-apis
 --with-java=gij
+--with-openclipart=/usr/share/openclipart
+--with-docdir=/usr/share/doc/openoffice.org
 --enable-libart
 --enable-libsn
 --enable-fontconfig
