diff -Naur ooo-build.orig/distro-configs/Frugalware64.conf.in ooo-build/distro-configs/Frugalware64.conf.in
--- ooo-build.orig/distro-configs/Frugalware64.conf.in	2006-05-02 18:38:32.000000000 +0200
+++ ooo-build/distro-configs/Frugalware64.conf.in	2006-05-20 02:15:33.000000000 +0200
@@ -19,6 +19,8 @@
 --with-system-xml-apis
 --with-system-nas
 --with-java=gij
+--with-openclipart=/usr/share/openclipart
+--with-docdir=/usr/share/doc/openoffice.org
 --enable-libart
 --enable-libsn
 --enable-kde
diff -Naur ooo-build.orig/distro-configs/Frugalware.conf.in ooo-build/distro-configs/Frugalware.conf.in
--- ooo-build.orig/distro-configs/Frugalware.conf.in	2006-05-02 18:38:32.000000000 +0200
+++ ooo-build/distro-configs/Frugalware.conf.in	2006-05-20 02:14:51.000000000 +0200
@@ -18,6 +18,8 @@
 --with-system-xerces
 --with-system-xml-apis
 --with-java=gij
+--with-openclipart=/usr/share/openclipart
+--with-docdir=/usr/share/doc/openoffice.org
 --enable-libart
 --enable-libsn
 --enable-fontconfig
diff -Naur ooo-build.orig/patches/src680/apply ooo-build/patches/src680/apply
--- ooo-build.orig/patches/src680/apply	2006-05-19 13:03:18.000000000 +0200
+++ ooo-build/patches/src680/apply	2006-05-20 02:16:05.000000000 +0200
@@ -570,12 +570,14 @@
 frugalware.diff, pmladek
 xmlhelp-port-to-db-4.4.diff, i#58474, hmth
 gcc401-stlport45-include.diff
+disable-regcomp_java.diff
 
 
 [ Frugalware64Only ]
 frugalware64.diff, pmladek
 xmlhelp-port-to-db-4.4.diff, i#58474, hmth
 gcc401-stlport45-include.diff
+disable-regcomp_java.diff
 
 
 # ( mmp@oo.o deals with defaults apparently )
diff -Naur ooo-build.orig/patches/src680/disable-regcomp_java.diff ooo-build/patches/src680/disable-regcomp_java.diff
--- ooo-build.orig/patches/src680/disable-regcomp_java.diff	1970-01-01 01:00:00.000000000 +0100
+++ ooo-build/patches/src680/disable-regcomp_java.diff	2006-05-20 02:16:05.000000000 +0200
@@ -0,0 +1,11 @@
+--- solenv/bin/modules/installer/servicesfile.pm.orig	2006-04-29 11:25:25.000000000 +0200
++++ solenv/bin/modules/installer/servicesfile.pm	2006-04-29 12:24:45.000000000 +0200
+@@ -471,7 +473,7 @@
+ 	$python_error_occured = 0;
+ 
+ 	if ( $#unocomponents > -1 ) { $uno_error_occured = register_unocomponents(\@unocomponents, $regcompfileref, $servicesfile); }
+-	if ( $#javacomponents > -1 ) { $java_error_occured = register_javacomponents(\@javacomponents, $regcompfileref, $servicesfile, $regcomprdb); }
++	if ( 0 ) { $java_error_occured = register_javacomponents(\@javacomponents, $regcompfileref, $servicesfile, $regcomprdb); }
+ 	if ( $#pythoncomponents > -1 ) { $python_error_occured = register_pythoncomponents(\@pythoncomponents, $regcompfileref, $servicesfile, $includepatharrayref); }
+ 		
+ 	if ( $uno_error_occured || $java_error_occured || $python_error_occured ) { $registererrorflag = 1; }
