diff --git a/bin/package-ooo b/bin/package-ooo
index 50ceaf1..4f2644f 100755
--- a/bin/package-ooo
+++ b/bin/package-ooo
@@ -620,10 +620,17 @@ if test "z$VENDORNAME" != "zDebian" ; then
 	test -f $OODESTDIR/gid_Module_Optional_Kde && cp $OODESTDIR/gid_Module_Optional_Kde kde_list.txt || :
 	mv_file_between_flists kde_list.txt common_list.txt $OOINSTBASE/program/kdefilepicker
 	mv_file_between_flists kde_list.txt common_list.txt $OOINSTBASE/basis$VERSION/program/fps_kde.uno.so
-	mv_file_between_flists kde_list.txt common_list.txt $OOINSTBASE/basis$VERSION/program/libvclplug_kde[0-9]*l..so
+	mv_file_between_flists kde_list.txt common_list.txt $OOINSTBASE/basis$VERSION/program/libvclplug_kdel..so
 	mv_file_between_flists kde_list.txt common_list.txt $OOINSTBASE/basis$VERSION/program/libkabdrv1.so
 	add_used_directories kde_list.txt common_list.txt
-	
+
+	# create kde4 subpackage 
+	rm -f kde4_list.txt
+	mv_file_between_flists kde4_list.txt kde_list.txt    $OOINSTBASE/basis$VERSION/program/kde4be1.uno.so
+	mv_file_between_flists kde4_list.txt common_list.txt $OOINSTBASE/basis$VERSION/program/libvclplug_kde4l..so
+	mv_file_between_flists kde4_list.txt common_list.txt $OOINSTBASE/basis$VERSION/program/fps_kde4.uno.so
+	add_used_directories kde4_list.txt common_list.txt
+
 	# gnome subpackage
 	rm -f gnome_list.txt
 	test -f $OODESTDIR/gid_Module_Optional_Gnome && cp $OODESTDIR/gid_Module_Optional_Gnome gnome_list.txt || :
