Index: patches/src680/apply
===================================================================
--- patches/src680/apply	(revision 11042)
+++ patches/src680/apply	(working copy)
@@ -1875,7 +1875,7 @@
 SectionOwner => rodo
 SectionIssue => i#75399
 store-core.diff
-store-install.diff
+#store-install.diff
 store-registry.diff
 # don't burn 3% of startup doing CRC's we don't need
 speed-store.diff, i#78495, michael
