diff --git a/distro-configs/Frugalware.conf.in b/distro-configs/Frugalware.conf.in
index adfccbf..876127a 100644
--- a/distro-configs/Frugalware.conf.in
+++ b/distro-configs/Frugalware.conf.in
@@ -1,26 +1,10 @@
---with-intro-bitmaps=\"$TOOLSDIR/src/openintro_fw.bmp\"
---with-about-bitmaps=\"$TOOLSDIR/src/openabout_fw.bmp\"
 --with-vendor=\"Frugalware Linux\"
---with-system-curl
---with-system-expat
---with-system-neon
---with-system-sndfile
---with-system-boost
---with-system-hunspell
---with-system-odbc-headers
---with-system-nas
---with-system-freetype
 --with-jdk-home=$JAVA_HOME
 --with-java-target-version=1.5
---with-system-xalan
---with-system-xerces
---with-system-xml-apis
---with-system-lucene
 --with-serializer-jar=/usr/share/java/serializer.jar
 --with-xalan-jar=/usr/share/java/xalan.jar
 --with-xerces-jar=/usr/share/java/xercesImpl.jar
 --with-xml-apis-jar=/usr/share/java/xml-apis.jar
---with-lucene-core-jar=/usr/share/java/lucene-core.jar
 --with-lucene-analyzers-jar=/usr/share/java/lucene-analyzers.jar
 --disable-kde
 --enable-kde4
@@ -31,10 +15,15 @@
 --disable-rpath
 --disable-qadevooo
 --enable-gcjaot
---with-system-libxslt
---with-system-libwpd
 --without-fonts
 --with-system-mozilla=mozilla
---with-external-dict-dir=@libdir@/@OOOINSTALLDIRNAME@/basis@OOO_VERSION@/share/dictionaries
---with-external-hyph-dir=@libdir@/@OOOINSTALLDIRNAME@/basis@OOO_VERSION@/share/dictionaries
---with-external-thes-dir=@libdir@/@OOOINSTALLDIRNAME@/basis@OOO_VERSION@/share/dictionaries
+--with-system-libs
+--with-system-headers
+--without-system-saxon
+--with-external-tar=\"$TOOLSDIR/ext_source\"
+--without-afms
+--with-system-libtextcat
+--with-system-libtextcat-data
+--with-system-dicts
+--with-external-dict-dir=/usr/share/myspell
+--without-myspell-dicts
