Index: distro-configs/Frugalware.conf.in
===================================================================
--- distro-configs/Frugalware.conf.in	(revision 9063)
+++ distro-configs/Frugalware.conf.in	(working copy)
@@ -27,3 +27,4 @@
 --disable-qadevooo
 --enable-gcjaot
 --with-system-libxslt
+--with-system-libwpd
Index: distro-configs/Frugalware64.conf.in
===================================================================
--- distro-configs/Frugalware64.conf.in	(revision 9063)
+++ distro-configs/Frugalware64.conf.in	(working copy)
@@ -28,3 +28,4 @@
 --disable-qadevooo
 --enable-gcjaot
 --with-system-libxslt
+--with-system-libwpd
