From 7f0a1cc0e8afb1589fb6f37e17a80c824f17a52f Mon Sep 17 00:00:00 2001
From: Harald Sitter <sitter@kde.org>
Date: Tue, 15 Oct 2024 10:13:51 +0200
Subject: [PATCH] work around bugged qmlcachegen

BUG: 494281
---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 41297b6f..9290cf52 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -58,6 +58,12 @@ find_package(
 )
 qt_policy(SET QTP0001 NEW)
 
+if(Qt6_VERSION VERSION_EQUAL 6.8.0)
+    # https://bugreports.qt.io/browse/QTBUG-129797
+    message(NOTICE "Qt6_VERSION: ${Qt6_VERSION} detected. Enabling CAST_FROM_ASCII")
+    remove_definitions(-DQT_NO_CAST_FROM_ASCII)
+endif()
+
 find_package(
     KF6 ${KF_MIN_VERSION} REQUIRED
         CoreAddons
-- 
GitLab

