diff -Naur ladspa_sdk/src/makefile ladspa_sdk-patched/src/makefile
--- ladspa_sdk/src/makefile	2002-07-06 19:34:20.000000000 +0200
+++ ladspa_sdk-patched/src/makefile	2006-03-02 14:50:14.000000000 +0100
@@ -4,10 +4,10 @@
 #
 # Change these if you want to install somewhere else. In particularly
 # you may wish to remove the middle "local/" part of each entry.
-
-INSTALL_PLUGINS_DIR	=	/usr/local/lib/ladspa/
-INSTALL_INCLUDE_DIR	=	/usr/include/
-INSTALL_BINARY_DIR	=	/usr/local/bin/
+DESTDIR			=
+INSTALL_PLUGINS_DIR	=	${DESTDIR}/usr/lib/ladspa/
+INSTALL_INCLUDE_DIR	=	${DESTDIR}/usr/include/
+INSTALL_BINARY_DIR	=	${DESTDIR}/usr/bin/
 
 ###############################################################################
 #
@@ -51,15 +51,13 @@
 	@echo ---------------------------------------------
 	@echo First listen to the white noise input signal:
 	@echo ---------------------------------------------
-	play ../snd/noise.wav
 	@echo -------------------------
 	@echo Compare to plugin output.
 	@echo -------------------------
 	@echo Should be a noise band around 6000Hz, repeated quietly after 1s.
-	play /tmp/test.wav
 
 install:	targets
-	-mkdirhier	$(INSTALL_PLUGINS_DIR)				\
+	mkdir -p	$(INSTALL_PLUGINS_DIR)				\
 			$(INSTALL_INCLUDE_DIR)				\
 			$(INSTALL_BINARY_DIR)
 	cp ../plugins/* $(INSTALL_PLUGINS_DIR)
