diff -Naur libgphoto2-2.2.1.orig/libgphoto2_port/disk/disk.c libgphoto2-2.2.1/libgphoto2_port/disk/disk.c
--- libgphoto2-2.2.1.orig/libgphoto2_port/disk/disk.c	2006-04-19 03:54:26.000000000 +0200
+++ libgphoto2-2.2.1/libgphoto2_port/disk/disk.c	2006-09-28 17:01:30.000000000 +0200
@@ -175,7 +175,7 @@
 
 	libhal_free_string_array (volumes);
 	libhal_ctx_free (ctx);
-	dbus_connection_disconnect (dbus_connection);
+	/* dbus_connection_close (dbus_connection); */
 	dbus_connection_unref (dbus_connection);
 #else
 # ifdef HAVE_MNTENT_H
