diff -Naur octave-forge-2006.03.17/FIXES/Makefile octave-forge-2006.03.17-p/FIXES/Makefile
--- octave-forge-2006.03.17/FIXES/Makefile	2006-01-07 06:11:56.000000000 +0100
+++ octave-forge-2006.03.17-p/FIXES/Makefile	2006-10-08 03:58:22.000000000 +0200
@@ -47,14 +47,6 @@
 	$(MKOCTFILE) $(SORTDEFINES) $<
 
 # FIXES needs an install target because it isn't under main/ or extra/
-install:
-	@if test -f FIXES/NOINSTALL ; then \
-	    echo skipping FIXES ; \
-	else \
-	    echo installing FIXES to $(DESTDIR)$(MPATH)/FIXES ; \
-	    ../$(INSTALLOCT) . $(DESTDIR)$(MPATH)/FIXES \
-	    $(DESTDIR)$(OPATH) $(DESTDIR)$(XPATH) ; \
-	fi
 
 clean: 
 	-$(RM) *.o core octave-core core *.oct *~ randtest
