--- webkitgtk-2.46.3/Source/WTF/wtf/Platform.h	2024-08-19 08:28:38.539297800 +0200
+++ webkitgtk-2.46.3.new/Source/WTF/wtf/Platform.h	2024-10-31 17:18:58.444825731 +0100
@@ -115,6 +115,7 @@
 /* ICU configuration. Some of these match ICU defaults on some platforms, but we would like them consistently set everywhere we build WebKit. */
 #define U_HIDE_DEPRECATED_API 1
 #define U_SHOW_CPLUSPLUS_API 0
+#define U_SHOW_CPLUSPLUS_HEADER_API 0
 #ifdef __cplusplus
 #define UCHAR_TYPE char16_t
 #endif
