diff -Naur webkitgtk-2.32.0/Source/JavaScriptCore/runtime/IntlSegmenter.cpp webkitgtk-2.32.0.new/Source/JavaScriptCore/runtime/IntlSegmenter.cpp
--- webkitgtk-2.32.0/Source/JavaScriptCore/runtime/IntlSegmenter.cpp	2021-02-26 10:57:05.000000000 +0100
+++ webkitgtk-2.32.0.new/Source/JavaScriptCore/runtime/IntlSegmenter.cpp	2021-04-12 22:34:49.232246733 +0200
@@ -125,7 +125,7 @@
     auto upconvertedCharacters = Box<Vector<UChar>>::create(string.charactersWithoutNullTermination());
 
     UErrorCode status = U_ZERO_ERROR;
-    auto segmenter = std::unique_ptr<UBreakIterator, UBreakIteratorDeleter>(ubrk_safeClone(m_segmenter.get(), nullptr, nullptr, &status));
+    auto segmenter = std::unique_ptr<UBreakIterator, UBreakIteratorDeleter>(ubrk_clone(m_segmenter.get(), &status));
     if (U_FAILURE(status)) {
         throwTypeError(globalObject, scope, "failed to initialize Segments"_s);
         return { };
diff -Naur webkitgtk-2.32.0/Source/JavaScriptCore/runtime/IntlSegments.cpp webkitgtk-2.32.0.new/Source/JavaScriptCore/runtime/IntlSegments.cpp
--- webkitgtk-2.32.0/Source/JavaScriptCore/runtime/IntlSegments.cpp	2021-02-26 10:57:05.000000000 +0100
+++ webkitgtk-2.32.0.new/Source/JavaScriptCore/runtime/IntlSegments.cpp	2021-04-12 22:35:17.194428950 +0200
@@ -100,7 +100,7 @@
     auto scope = DECLARE_THROW_SCOPE(vm);
 
     UErrorCode status = U_ZERO_ERROR;
-    auto segmenter = std::unique_ptr<UBreakIterator, UBreakIteratorDeleter>(ubrk_safeClone(m_segmenter.get(), nullptr, nullptr, &status));
+    auto segmenter = std::unique_ptr<UBreakIterator, UBreakIteratorDeleter>(ubrk_clone(m_segmenter.get(), &status));
     if (U_FAILURE(status)) {
         throwTypeError(globalObject, scope, "failed to initialize SegmentIterator"_s);
         return nullptr;
