diff -Naur gnome-vfs-2.12.0.orig/schemas/desktop_gnome_url_handlers.schemas.in gnome-vfs-2.12.0/schemas/desktop_gnome_url_handlers.schemas.in
--- gnome-vfs-2.12.0.orig/schemas/desktop_gnome_url_handlers.schemas.in	2005-09-07 23:50:30.000000000 +0200
+++ gnome-vfs-2.12.0/schemas/desktop_gnome_url_handlers.schemas.in	2005-09-07 23:52:25.000000000 +0200
@@ -153,7 +153,7 @@
       <applyto>/desktop/gnome/url-handlers/http/command</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>epiphany %s</default>
+      <default>firefox %s</default>
       <locale name="C">
         <short>The handler for "http" URLs</short>
         <long>The command used to handle "http" URLs, if enabled.</long>
@@ -187,7 +187,7 @@
       <applyto>/desktop/gnome/url-handlers/https/command</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>epiphany %s</default>
+      <default>firefox %s</default>
       <locale name="C">
         <short>The handler for "https" URLs</short>
         <long>The command used to handle "https" URLs, if enabled.</long>
