diff -Naur gnome-vfs-2.12.0.orig/libgnomevfs/gnome-vfs-volume-monitor-daemon.c gnome-vfs-2.12.0/libgnomevfs/gnome-vfs-volume-monitor-daemon.c
--- gnome-vfs-2.12.0.orig/libgnomevfs/gnome-vfs-volume-monitor-daemon.c	2005-09-07 23:50:30.000000000 +0200
+++ gnome-vfs-2.12.0/libgnomevfs/gnome-vfs-volume-monitor-daemon.c	2005-09-07 23:56:37.000000000 +0200
@@ -968,6 +968,8 @@
 		vol->priv->is_user_visible = 1;
 		break;
 	default:
+		if (!strncmp (mount->mount_path, "/media/", 7))
+		    vol->priv->is_user_visible = 1;
 		break;
 	}
 	
