# upstream http://bugzilla.gnome.org/show_bug.cgi?id=300547

diff -Nur gnome-vfs2-2.10.0/libgnomevfs/gnome-vfs-volume-monitor-daemon.c gnome-vfs2-2.10.0.new/libgnomevfs/gnome-vfs-volume-monitor-daemon.c
--- gnome-vfs2-2.10.0/libgnomevfs/gnome-vfs-volume-monitor-daemon.c	2004-12-14 17:23:33.000000000 +0100
+++ gnome-vfs2-2.10.0.new/libgnomevfs/gnome-vfs-volume-monitor-daemon.c	2005-03-17 12:01:07.245801992 +0100
@@ -718,6 +718,9 @@
 			mount = l->data;
 			uri = gnome_vfs_get_uri_from_local_path (mount->mount_path);
 
+			if (uri == NULL)
+			    continue;
+
 			drive = _gnome_vfs_volume_monitor_find_fstab_drive_by_activation_uri (volume_monitor, uri);
 
 			if (drive != NULL) {
