diff --git a/src/ch7017/ch7017.c b/src/ch7017/ch7017.c
index 01e414f..6b27d05 100644
--- a/src/ch7017/ch7017.c
+++ b/src/ch7017/ch7017.c
@@ -34,7 +34,6 @@
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Resources.h"
 #include "compiler.h"
 #include "miscstruct.h"
 #include "xf86i2c.h"
diff --git a/src/ch7xxx/ch7xxx.c b/src/ch7xxx/ch7xxx.c
index 13ba102..cad507d 100644
--- a/src/ch7xxx/ch7xxx.c
+++ b/src/ch7xxx/ch7xxx.c
@@ -34,7 +34,6 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 #include <string.h>
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Resources.h"
 #include "compiler.h"
 #include "miscstruct.h"
 #include "xf86i2c.h"
diff --git a/src/i810_driver.c b/src/i810_driver.c
index 4b8c459..f158f5d 100644
--- a/src/i810_driver.c
+++ b/src/i810_driver.c
@@ -55,8 +55,6 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Resources.h"
-#include "xf86RAC.h"
 #include "xf86cmap.h"
 #include "compiler.h"
 #include "mibstore.h"
@@ -549,10 +547,6 @@ I810PreInit(ScrnInfoPtr pScrn, int flags)
 
    pI810->PciInfo = xf86GetPciInfoForEntity(pI810->pEnt->index);
 
-   if (xf86RegisterResources(pI810->pEnt->index, NULL, ResNone))
-      return FALSE;
-   pScrn->racMemFlags = RAC_FB | RAC_COLORMAP;
-
    /* Set pScrn->monitor */
    pScrn->monitor = pScrn->confScreen->monitor;
 
@@ -937,9 +931,6 @@ I810PreInit(ScrnInfoPtr pScrn, int flags)
 
    /*  We won't be using the VGA access after the probe */
    I810SetMMIOAccess(pI810);
-   xf86SetOperatingState(resVgaIo, pI810->pEnt->index, ResUnusedOpr);
-   xf86SetOperatingState(resVgaMem, pI810->pEnt->index, ResDisableOpr);
-
    return TRUE;
 }
 
diff --git a/src/i810_hwmc.c b/src/i810_hwmc.c
index 12ffdd9..1c3ffc9 100644
--- a/src/i810_hwmc.c
+++ b/src/i810_hwmc.c
@@ -41,7 +41,6 @@ THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Resources.h"
 #include "compiler.h"
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
diff --git a/src/i810_video.c b/src/i810_video.c
index 9528bc7..c82fcc9 100644
--- a/src/i810_video.c
+++ b/src/i810_video.c
@@ -41,7 +41,6 @@ THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Resources.h"
 #include "compiler.h"
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
diff --git a/src/i830_driver.c b/src/i830_driver.c
index 33079fe..f701231 100644
--- a/src/i830_driver.c
+++ b/src/i830_driver.c
@@ -51,8 +51,6 @@ USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Resources.h"
-#include "xf86RAC.h"
 #include "xf86Priv.h"
 #include "xf86cmap.h"
 #include "compiler.h"
@@ -1540,13 +1538,6 @@ I830PreInit(ScrnInfoPtr pScrn, int flags)
    if (!i830_open_drm_master(pScrn))
        xf86DrvMsg(pScrn->scrnIndex, X_ERROR, "Failed to become DRM master.\n");
 
-
-   if (xf86RegisterResources(pI830->pEnt->index, NULL, ResNone)) {
-      PreInitCleanup(pScrn);
-      return FALSE;
-   }
-
-   pScrn->racMemFlags = RAC_FB | RAC_COLORMAP;
    pScrn->monitor = pScrn->confScreen->monitor;
    pScrn->progClock = TRUE;
    pScrn->rgbBits = 8;
@@ -1632,8 +1623,6 @@ I830PreInit(ScrnInfoPtr pScrn, int flags)
 
        /*  We won't be using the VGA access after the probe. */
        I830SetMMIOAccess(pI830);
-       xf86SetOperatingState(resVgaIo, pI830->pEnt->index, ResUnusedOpr);
-       xf86SetOperatingState(resVgaMem, pI830->pEnt->index, ResDisableOpr);
    }
 
    /* Load the dri2 module if requested. */
diff --git a/src/i830_i2c.c b/src/i830_i2c.c
index 8392d0a..4ba073d 100644
--- a/src/i830_i2c.c
+++ b/src/i830_i2c.c
@@ -31,8 +31,6 @@ USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Resources.h"
-#include "xf86RAC.h"
 #include "xf86cmap.h"
 #include "compiler.h"
 #include "mibstore.h"
diff --git a/src/i830_video.c b/src/i830_video.c
index 92b6324..9fb0b56 100644
--- a/src/i830_video.c
+++ b/src/i830_video.c
@@ -56,7 +56,6 @@
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Resources.h"
 #include "compiler.h"
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
diff --git a/src/i915_hwmc.c b/src/i915_hwmc.c
index 6712caa..7e0f145 100644
--- a/src/i915_hwmc.c
+++ b/src/i915_hwmc.c
@@ -32,7 +32,6 @@
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Resources.h"
 #include "compiler.h"
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
diff --git a/src/ivch/ivch.c b/src/ivch/ivch.c
index dffe334..79da11a 100644
--- a/src/ivch/ivch.c
+++ b/src/ivch/ivch.c
@@ -31,7 +31,6 @@
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Resources.h"
 #include "compiler.h"
 #include "miscstruct.h"
 #include "xf86i2c.h"
diff --git a/src/sil164/sil164.c b/src/sil164/sil164.c
index c90bed3..e43107a 100644
--- a/src/sil164/sil164.c
+++ b/src/sil164/sil164.c
@@ -35,7 +35,6 @@ SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Resources.h"
 #include "compiler.h"
 #include "miscstruct.h"
 #include "xf86i2c.h"
diff --git a/src/tfp410/tfp410.c b/src/tfp410/tfp410.c
index 4d3bb8a..a8985f7 100644
--- a/src/tfp410/tfp410.c
+++ b/src/tfp410/tfp410.c
@@ -34,7 +34,6 @@
 
 #include "xf86.h"
 #include "xf86_OSproc.h"
-#include "xf86Resources.h"
 #include "compiler.h"
 #include "miscstruct.h"
 #include "xf86i2c.h"
