diff -Naur p7zip_4.43/install.sh p7zip_4.43-p/install.sh
--- p7zip_4.43/install.sh	2006-06-27 23:09:12.000000000 +0200
+++ p7zip_4.43-p/install.sh	2006-09-24 23:52:44.195103693 +0200
@@ -1,19 +1,15 @@
 #! /bin/sh
 
 # global install
-DEST_HOME=/usr/local
-# for testing 
-# DEST_HOME=${HOME}/INSTALL/usr/local
+DEST_HOME=/usr
 DEST_BIN=${DEST_HOME}/bin
 DEST_SHARE=${DEST_HOME}/lib/p7zip
-DEST_SHARE_DOC=${DEST_HOME}/share/doc/p7zip
 DEST_MAN=${DEST_HOME}/man
 DEST_DIR=
 [ "$1" ] && DEST_BIN=$1
 [ "$2" ] && DEST_SHARE=$2
 [ "$3" ] && DEST_MAN=$3
 [ "$4" ] && DEST_DIR=$4
-[ "$5" ] && DEST_SHARE_DOC=$5
 
 use_share="n"
 
@@ -39,13 +35,7 @@
 rm -f  ${DEST_DIR}${DEST_MAN}/man1/7z.1
 rm -f  ${DEST_DIR}${DEST_MAN}/man1/7za.1
 rm -f  ${DEST_DIR}${DEST_MAN}/man1/7zr.1
-rm -f  ${DEST_DIR}${DEST_SHARE_DOC}/README
-rm -f  ${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog
-if [ -d ${DEST_DIR}${DEST_SHARE_DOC}/DOCS ]
-then
-  find ${DEST_DIR}${DEST_SHARE_DOC}/DOCS -type d -exec chmod 777 {} \;
-  rm -fr ${DEST_DIR}${DEST_SHARE_DOC}/DOCS
-fi
+
 rmdir  ${DEST_DIR}${DEST_SHARE} 2> /dev/null
 
 if [ "${use_share}" = "o" ]
@@ -60,7 +50,7 @@
     strip     ${DEST_DIR}${DEST_SHARE}/7za
     chmod 555 ${DEST_DIR}${DEST_SHARE}/7za
     echo "#! /bin/sh" > ${DEST_DIR}${DEST_BIN}/7za
-    echo "${DEST_SHARE}/7za \"\$@\"" >> ${DEST_DIR}${DEST_BIN}/7za
+    echo "/usr/lib/p7zip/7za \"\$@\"" >> ${DEST_DIR}${DEST_BIN}/7za
     chmod 555 ${DEST_DIR}${DEST_BIN}/7za
   fi
 
@@ -72,7 +62,7 @@
     strip     ${DEST_DIR}${DEST_SHARE}/7zr
     chmod 555 ${DEST_DIR}${DEST_SHARE}/7zr
     echo "#! /bin/sh" > ${DEST_DIR}${DEST_BIN}/7zr
-    echo "${DEST_SHARE}/7zr \"\$@\"" >> ${DEST_DIR}${DEST_BIN}/7zr
+    echo "/usr/lib/p7zip/7zr \"\$@\"" >> ${DEST_DIR}${DEST_BIN}/7zr
     chmod 555 ${DEST_DIR}${DEST_BIN}/7zr
   fi
 
@@ -95,7 +85,7 @@
     cp -r bin/Codecs bin/Formats ${DEST_DIR}${DEST_SHARE}/
     chmod 555 ${DEST_DIR}${DEST_SHARE}/*/*
     echo "#! /bin/sh" > ${DEST_DIR}${DEST_BIN}/7z
-    echo "${DEST_SHARE}/7z \"\$@\"" >> ${DEST_DIR}${DEST_BIN}/7z
+    echo "/usr/lib/p7zip/7z \"\$@\"" >> ${DEST_DIR}${DEST_BIN}/7z
     chmod 555 ${DEST_DIR}${DEST_BIN}/7z
   fi
 
@@ -145,28 +135,4 @@
    chmod 444 ${DEST_DIR}${DEST_MAN}/man1/7zr.1
 fi
 
-if [ -f README ]
-then
-  echo "- installing ${DEST_DIR}${DEST_SHARE_DOC}/README"
-  mkdir -p ${DEST_DIR}${DEST_SHARE_DOC}
-  cp README ${DEST_DIR}${DEST_SHARE_DOC}/README
-  chmod 444 ${DEST_DIR}${DEST_SHARE_DOC}/README
-fi
-
-if [ -f ChangeLog ]
-then
-  echo "- installing ${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog"
-  mkdir -p ${DEST_DIR}${DEST_SHARE_DOC}
-  cp ChangeLog ${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog
-  chmod 444 ${DEST_DIR}${DEST_SHARE_DOC}/ChangeLog
-fi
-
-if [ -d DOCS ]
-then
-  echo "- installing HTML help in ${DEST_DIR}${DEST_SHARE_DOC}/DOCS"
-  mkdir -p ${DEST_DIR}${DEST_SHARE_DOC}
-  cp -r DOCS ${DEST_DIR}${DEST_SHARE_DOC}/DOCS
-  find ${DEST_DIR}${DEST_SHARE_DOC}/DOCS -type d -exec chmod 555 {} \;
-  find ${DEST_DIR}${DEST_SHARE_DOC}/DOCS -type f -exec chmod 444 {} \;
-fi
 
diff -Naur p7zip_4.43/makefile p7zip_4.43-p/makefile
--- p7zip_4.43/makefile	2006-09-23 09:05:48.000000000 +0200
+++ p7zip_4.43-p/makefile	2006-09-24 23:43:43.423313501 +0200
@@ -1,7 +1,7 @@
 
-DEST_BIN=/usr/local/bin
-DEST_SHARE=/usr/local/lib/p7zip
-DEST_MAN=/usr/local/man
+DEST_BIN=$(DESTDIR)/usr/bin
+DEST_SHARE=$(DESTDIR)/usr/lib/p7zip
+DEST_MAN=$(DESTDIR)/usr/man
 
 .PHONY: default all all2 7za sfx 7z 7zr common clean tar_bin depend test test_7z test_7zr
 
diff -Naur p7zip_4.43/makefile.linux_amd64 p7zip_4.43-p/makefile.linux_amd64
--- p7zip_4.43/makefile.linux_amd64	2006-05-19 20:52:42.000000000 +0200
+++ p7zip_4.43-p/makefile.linux_amd64	2006-09-24 23:43:43.423313501 +0200
@@ -1,8 +1,8 @@
 
-OPTFLAGS=-O
+OPTFLAGS = $(CXXFLAGS) 
 
 # use "-m32" to have a 32bits executable
-ALLFLAGS=-m64 ${OPTFLAGS} -s -fPIC \
+ALLFLAGS=-m64 ${OPTFLAGS} -fPIC \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	$(LOCAL_FLAGS)
diff -Naur p7zip_4.43/makefile.machine p7zip_4.43-p/makefile.machine
--- p7zip_4.43/makefile.machine	2006-09-23 09:06:30.000000000 +0200
+++ p7zip_4.43-p/makefile.machine	2006-09-24 23:43:43.424313349 +0200
@@ -3,9 +3,9 @@
 # LITTLE_ENDIAN and BIG_ENDIAN are already defined by the system headers
 #
 
-OPTFLAGS=-O
+OPTFLAGS = $(CXXFLAGS) 
 
-ALLFLAGS=${OPTFLAGS} -s -fPIC \
+ALLFLAGS=${OPTFLAGS}  -fPIC \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	$(LOCAL_FLAGS)
