--- llvm-3.2.src.old/tools/clang/lib/Driver/ToolChains.cpp	2012-12-16 17:59:27.000000000 +0200
+++ llvm-3.2.src.new/tools/clang/lib/Driver/ToolChains.cpp	2013-01-08 02:38:19.000000000 +0200
@@ -2228,8 +2228,12 @@
   if (DriverArgs.hasArg(options::OPT_nostdinc))
     return;
 
-  if (!DriverArgs.hasArg(options::OPT_nostdlibinc))
+  if (!DriverArgs.hasArg(options::OPT_nostdlibinc)) {
     addSystemInclude(DriverArgs, CC1Args, D.SysRoot + "/usr/local/include");
+    addSystemInclude(DriverArgs, CC1Args, D.SysRoot + "/usr/include/c++/4.7.2");
+    addSystemInclude(DriverArgs, CC1Args, D.SysRoot + "/usr/include/c++/4.7.2/x86_64-frugalware-linux");
+    addSystemInclude(DriverArgs, CC1Args, D.SysRoot + "/usr/include/c++/4.7.2/i686-frugalware-linux");
+  }
 
   if (!DriverArgs.hasArg(options::OPT_nobuiltininc)) {
     llvm::sys::Path P(D.ResourceDir);
