--- llvm-3.4.2.src/tools/clang/lib/Driver/ToolChains.cpp	2014-04-26 01:38:13.000000000 +0200
+++ llvm-3.4.2.src.new/tools/clang/lib/Driver/ToolChains.cpp	2014-10-11 00:33:04.000000000 +0200
@@ -1124,14 +1124,15 @@
   static const char *const X86_64Triples[] = {
     "x86_64-linux-gnu", "x86_64-unknown-linux-gnu", "x86_64-pc-linux-gnu",
     "x86_64-redhat-linux6E", "x86_64-redhat-linux", "x86_64-suse-linux",
-    "x86_64-manbo-linux-gnu", "x86_64-linux-gnu", "x86_64-slackware-linux"
+    "x86_64-manbo-linux-gnu", "x86_64-linux-gnu", "x86_64-slackware-linux",
+    "x86_64-frugalware-linux"
   };
   static const char *const X86LibDirs[] = { "/lib32", "/lib" };
   static const char *const X86Triples[] = {
     "i686-linux-gnu", "i686-pc-linux-gnu", "i486-linux-gnu", "i386-linux-gnu",
     "i386-redhat-linux6E", "i686-redhat-linux", "i586-redhat-linux",
     "i386-redhat-linux", "i586-suse-linux", "i486-slackware-linux",
-    "i686-montavista-linux"
+    "i686-montavista-linux", "i686-frugalware-linux"
   };
 
   static const char *const MIPSLibDirs[] = { "/lib" };
@@ -2588,8 +2589,14 @@
   if (DriverArgs.hasArg(options::OPT_nostdinc))
     return;
 
-  if (!DriverArgs.hasArg(options::OPT_nostdlibinc))
-    addSystemInclude(DriverArgs, CC1Args, SysRoot + "/usr/local/include");
+  if (!DriverArgs.hasArg(options::OPT_nostdlibinc)) {
+    addSystemInclude(DriverArgs, CC1Args, D.SysRoot + "/usr/local/include");
+    addSystemInclude(DriverArgs, CC1Args, D.SysRoot + "/usr/include/c++/@gcc_ver@");
+    addSystemInclude(DriverArgs, CC1Args, D.SysRoot + "/usr/include/c++/@gcc_ver@/backward");
+    addSystemInclude(DriverArgs, CC1Args, D.SysRoot + "/usr/include/c++/@gcc_ver@/i686-frugalware-linux");
+    addSystemInclude(DriverArgs, CC1Args, D.SysRoot + "/usr/include/c++/@gcc_ver@/x86_64-frugalware-linux");
+    addSystemInclude(DriverArgs, CC1Args, D.SysRoot + "/usr/include/c++/@gcc_ver@/arm-frugalware-linux-gnueabi");
+  }
 
   if (!DriverArgs.hasArg(options::OPT_nobuiltininc)) {
     SmallString<128> P(D.ResourceDir);
