--- llvm-3.0.src/Makefile.config.in	2011-10-13 17:27:34.000000000 +0000
+++ llvm-3.0.src.new/Makefile.config.in	2012-03-08 02:30:43.000000000 +0000
@@ -81,10 +81,10 @@
 LLVMMAKE := $(LLVM_SRC_ROOT)/make
 
 PROJ_bindir     := $(PROJ_prefix)/bin
-PROJ_libdir     := $(PROJ_prefix)/lib
+PROJ_libdir     := $(PROJ_prefix)/lib/llvm
 PROJ_datadir    := $(PROJ_prefix)/share
-PROJ_docsdir    := $(PROJ_prefix)/docs/llvm
-PROJ_etcdir     := $(PROJ_prefix)/etc/llvm
+PROJ_docsdir    := $(PROJ_prefix)/share/doc/llvm
+PROJ_etcdir     := /etc/llvm
 PROJ_includedir := $(PROJ_prefix)/include
 PROJ_infodir    := $(PROJ_prefix)/info
 PROJ_mandir     := $(PROJ_prefix)/share/man
