--- llvm-3.2.src.old/Makefile.config.in	2012-10-01 21:40:32.000000000 +0300
+++ llvm-3.2.src.new/Makefile.config.in	2012-12-26 05:32:56.000000000 +0200
@@ -97,8 +97,8 @@
 PROJ_bindir     := $(PROJ_prefix)/bin
 PROJ_libdir     := $(PROJ_prefix)/lib
 PROJ_datadir    := $(PROJ_prefix)/share
-PROJ_docsdir    := $(PROJ_prefix)/docs/llvm
-PROJ_etcdir     := $(PROJ_prefix)/etc/llvm
+PROJ_docsdir    := $(PROJ_prefix)/share/doc/llvm
+PROJ_etcdir     := /etc/llvm
 PROJ_includedir := $(PROJ_prefix)/include
 PROJ_infodir    := $(PROJ_prefix)/info
 PROJ_mandir     := $(PROJ_prefix)/share/man
