--- llvm-3.0.src/Makefile.rules	2011-10-11 12:51:44.000000000 +0000
+++ llvm-3.0.src.new/Makefile.rules	2012-03-08 02:30:43.000000000 +0000
@@ -546,9 +546,9 @@
 ifdef TOOLNAME
   LD.Flags += $(RPATH) -Wl,'$$ORIGIN/../lib'
   ifdef EXAMPLE_TOOL
-    LD.Flags += $(RPATH) -Wl,$(ExmplDir) $(DynamicFlag)
+    LD.Flags +=  $(DynamicFlag)
   else
-    LD.Flags += $(RPATH) -Wl,$(ToolDir) $(DynamicFlag)
+    LD.Flags +=  $(DynamicFlag)
   endif
 endif
 else
