diff -Naur ede/datas/desktop-links/Makefile ede-fw/datas/desktop-links/Makefile
--- ede/datas/desktop-links/Makefile	2006-09-02 22:52:51.000000000 +0200
+++ ede-fw/datas/desktop-links/Makefile	2006-09-03 20:58:12.865032240 +0200
@@ -13,7 +13,7 @@
 desktopdir = $(datadir)/ede/.ede/desktop
 
 DESKTOPLINKS     = \
-	Terminal.desktop
+	Terminal.desktop xmms.desktop Mozilla.desktop
 
 install:
 	$(MKINSTALLDIRS) $(desktopdir)
diff -Naur ede/datas/desktop-links/Mozilla.desktop ede-fw/datas/desktop-links/Mozilla.desktop
--- ede/datas/desktop-links/Mozilla.desktop	1970-01-01 01:00:00.000000000 +0100
+++ ede-fw/datas/desktop-links/Mozilla.desktop	2006-09-03 20:58:12.866032088 +0200
@@ -0,0 +1,8 @@
+# EFLTK Configuration - Version 2,003000
+
+[Desktop Entry]
+  Name=Mozilla Firefox
+  Name[sk]=Mozilla Firefox
+  Exec=firefox
+  Icon=mozilla.png
+
diff -Naur ede/datas/desktop-links/xmms.desktop ede-fw/datas/desktop-links/xmms.desktop
--- ede/datas/desktop-links/xmms.desktop	1970-01-01 01:00:00.000000000 +0100
+++ ede-fw/datas/desktop-links/xmms.desktop	2006-09-03 20:58:12.867031936 +0200
@@ -0,0 +1,8 @@
+# EFLTK Configuration - Version 2,003000
+
+[Desktop Entry]
+  Name=X Multimedia System
+  Exec=xmms
+  Icon=xmms_classic.png
+  Name[sk]=X Multimedia System
+
diff -Naur ede/datas/ede.conf ede-fw/datas/ede.conf
--- ede/datas/ede.conf	2006-09-02 22:52:51.000000000 +0200
+++ ede-fw/datas/ede.conf	2006-09-03 21:02:06.634493880 +0200
@@ -12,7 +12,8 @@
   Gridspacing=10
   Label Foreground=7
   Label Background=2037550592
-  OneClickExec=0
+  OneClickExec=1
+  AutoArrange=1
 
 [Panel]
   Volume Control=evolume
@@ -23,22 +24,22 @@
   CPUMonitor=true
   SoundMixer=true
   VariableWidthTaskbar=false
-  ShowUsernameOnMenu=true
+  ShowUsernameOnMenu=false
   ShowDesktop=true
   AllDesktops=false
 
 [Desktop]
-  Wallpaper=
+  Wallpaper=/usr/share/ede/wallpapers/frugalware.jpg
   Color=RGB(76,108,178)
   Opacity=205
-  Use=0
+  Use=1
 
 [Workspaces]
   Count=4
   (null)=9@9
 
 [Web]
-  Browser=mozilla
+  Browser=firefox
 
 [Terminal]
   Terminal=xterm
diff -Naur ede/datas/efltk.conf ede-fw/datas/efltk.conf
--- ede/datas/efltk.conf	2006-09-02 22:52:51.000000000 +0200
+++ ede-fw/datas/efltk.conf	2006-09-03 20:58:12.868031784 +0200
@@ -2,22 +2,22 @@
 
 
 [Images]
-  State Effects=1
+   State Effects=1
 
 [Menus]
-  Effects=0
+  Effect Type=1
   Delay=0
-  Subwindow Effect=0
-  Effect Type=0
+  Subwindow Effect=1
   Speed=1
+  Effects=1
 
 [Tooltips]
-  Effects=0
-  Effect Type=0
-  Enabled=1
+  Effect Type=1
   Delay=1
+  Effects=1
+  Enabled=1
 
 [MDI]
-  Animate=0
+  Animate=1
   Opaque=0
 
diff -Naur ede/datas/mime.conf ede-fw/datas/mime.conf
--- ede/datas/mime.conf	2006-09-02 22:52:45.000000000 +0200
+++ ede-fw/datas/mime.conf	2006-09-03 20:58:12.869031632 +0200
@@ -1,5 +1,9 @@
 # EFLTK Configuration - Version 2.000300
 
+[.fpm]
+  Name=FPM Package
+  Exec=
+
 [.rpm]
   Name=RPM Package
   Exec=einstaller
@@ -22,11 +26,11 @@
 
 [.html]
   Name=Internet WWW page
-  Exec=mozilla
+  Exec=firefox
 
 [.htm]
   Name=Internet WWW page
-  Exec=mozilla
+  Exec=firefox
 
 [.mp3]
   Name=Music file compressed with MP3 method
diff -Naur ede/datas/programs-links/Mozilla.desktop ede-fw/datas/programs-links/Mozilla.desktop
--- ede/datas/programs-links/Mozilla.desktop	2006-09-02 22:52:51.000000000 +0200
+++ ede-fw/datas/programs-links/Mozilla.desktop	2006-09-03 20:58:12.870031480 +0200
@@ -1,8 +1,8 @@
 # EFLTK Configuration - Version 2,003000
 
 [Desktop Entry]
-  Name=Mozilla
-  Name[sk]=Mozilla
-  Exec=mozilla
+  Name=Mozilla Firefox
+  Name[sk]=Mozilla Firefox
+  Exec=firefox
   Icon=mozilla.png
 
diff -Naur ede/datas/wmanager.conf ede-fw/datas/wmanager.conf
--- ede/datas/wmanager.conf	2006-09-02 22:52:51.000000000 +0200
+++ ede-fw/datas/wmanager.conf	2006-09-03 20:58:12.870031480 +0200
@@ -8,8 +8,8 @@
   Draw gradient=1
 
 [Resize]
-  Opaque resize=0
   Animate=1
+  Opaque resize=0
   Animate Speed=20
 
 
@@ -32,3 +32,9 @@
     MinimizeWindow=Alt+F7
     MaximizeWindow=Alt+F8
 
+[Misc]
+    Use frame=0
+    Frame color=RGB(0,0,128)
+    Use theme=1
+    Theme path=/usr/share/ede/themes/EdeAirlock/default.theme
+
