diff -Naur ede/datas/startede.in ede-p/datas/startede.in
--- ede/datas/startede.in	2007-11-16 14:18:28.000000000 +0100
+++ ede-p/datas/startede.in	2007-12-08 11:31:09.000000000 +0100
@@ -6,7 +6,7 @@
 XSETROOT=`which xsetroot 2> /dev/null`
 
 userresources=$HOME/.Xdefaults
-sysresources=/usr/X11R6/lib/X11/xinit/.Xresources
+sysresources=/usr/lib/X11/xinit/.Xresources
 
 if [ -f $sysresources ]; then
     xrdb -merge $sysresources
@@ -19,11 +19,8 @@
 PATH=$PATH:$PREFIX:$PREFIX/bin
 export PATH
 
-if [ -x $HOME/.ede ]; then
-    echo "EDE home dir exists, ok."
-else 
-    echo "Default EDE home dir for user $HOME does not exist. Creating one..."    
-    cp -R $PREFIX/share/ede/.ede $HOME
+if [ ! -d $HOME/.ede ]; then  
+    cp -R $PREFIX/share/ede/.ede  $HOME
 fi
 
 # If we have xsetroot than script shoult run it
@@ -36,7 +33,9 @@
 	xscreensaver -nosplash &
 fi
 
+## broken without that :/ - crazy -
+export LIBXCB_ALLOW_SLOPPY_LOCK=1
+
 exec edewm &
 exec eiconman &    
-etip &    
 exec eworkpanel
