diff -Naur ede/datas/startede.in ede-p/datas/startede.in
--- ede/datas/startede.in	2006-09-02 22:52:45.000000000 +0200
+++ ede-p/datas/startede.in	2006-09-03 16:39:20.988236416 +0200
@@ -6,7 +6,7 @@
 XSETROOT=`which xsetroot 2> /dev/null`
 
 userresources=$HOME/.Xdefaults
-sysresources=/usr/X11R6/lib/X11/xinit/.Xresources
+sysresources=/usr/lib/X11/xinit/.Xresources
 
 if [ -f $sysresources ]; then
     xrdb -merge $sysresources
@@ -19,11 +19,8 @@
 PATH=$PATH:$PREFIX:$PREFIX/bin
 export PATH
 
-if [ -x $HOME/.ede ]; then
-    echo "EDE home dir exists, ok."
-else 
-    echo "Default EDE home dir for user $HOME does not exist. Creating one..."    
-    cp -R $PREFIX/share/ede/.ede $HOME
+if [ ! -d $HOME/.ede ]; then  
+    cp -R $PREFIX/share/ede/.ede  $HOME
 fi
 
 # If we have xsetroot than script shoult run it
