diff -Naur libgtksourceviewmm-1.9.5/tests/test-search.cpp libgtksourceviewmm-1.9.5.new/tests/test-search.cpp
--- libgtksourceviewmm-1.9.5/tests/test-search.cpp	2007-09-02 04:05:56.000000000 +0530
+++ libgtksourceviewmm-1.9.5.new/tests/test-search.cpp	2008-04-29 12:25:57.000000000 +0530
@@ -3,6 +3,7 @@
 #include <fstream>
 #include <stdexcept>
 #include <string>
+#include <string.h>
 #include <gtkmm.h>
 #include <gtksourceviewmm.h>
 #include <glib/gmessages.h>
