diff -Naur libcap-2.25/Make.Rules libcap-2.25-p/Make.Rules
--- libcap-2.25/Make.Rules	2016-01-31 02:14:53.000000000 +0100
+++ libcap-2.25-p/Make.Rules	2016-10-26 02:30:49.524413853 +0200
@@ -48,9 +48,9 @@
 KERNEL_HEADERS := $(topdir)/libcap/include/uapi
 IPATH += -fPIC -I$(KERNEL_HEADERS) -I$(topdir)/libcap/include
 
-CC := gcc
-CFLAGS := -O2 -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
-BUILD_CC := $(CC)
+CC ?= gcc
+CFLAGS ?=  -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
+BUILD_CC ?= $(CC)
 BUILD_CFLAGS := $(CFLAGS) $(IPATH)
 AR := ar
 RANLIB := ranlib
@@ -60,7 +60,7 @@
         -Wstrict-prototypes -Wmissing-prototypes \
         -Wnested-externs -Winline -Wshadow
 LD=$(CC) -Wl,-x -shared
-LDFLAGS := #-g
+LDFLAGS ?= $(LDFLAGS
 BUILD_GPERF := $(shell which gperf >/dev/null 2>/dev/null && echo yes)
 
 SYSTEM_HEADERS = /usr/include
