diff -Naur -x '*.orig' -x Makefile DirectFB-0.9.25.1/gfxdrivers/i810/i810.c DirectFB-0.9.25.1-x86_64/gfxdrivers/i810/i810.c
--- DirectFB-0.9.25.1/gfxdrivers/i810/i810.c	2006-05-03 09:22:37.000000000 +0200
+++ DirectFB-0.9.25.1-x86_64/gfxdrivers/i810/i810.c	2006-07-13 23:58:02.000000000 +0200
@@ -21,6 +21,7 @@
    Free Software Foundation, Inc., 59 Temple Place - Suite 330,
    Boston, MA 02111-1307, USA.
 */
+#include <asm/types.h>
 #include <dfb_types.h>
 
 #include <stdio.h>
diff -Naur -x '*.orig' -x Makefile DirectFB-0.9.25.1/gfxdrivers/i810/i810_overlay.c DirectFB-0.9.25.1-x86_64/gfxdrivers/i810/i810_overlay.c
--- DirectFB-0.9.25.1/gfxdrivers/i810/i810_overlay.c	2005-10-19 01:14:09.000000000 +0200
+++ DirectFB-0.9.25.1-x86_64/gfxdrivers/i810/i810_overlay.c	2006-07-13 23:58:52.000000000 +0200
@@ -24,6 +24,7 @@
 */
 
 #include <sys/ioctl.h>
+#include <asm/types.h>
 
 #include <string.h>
 
diff -Naur -x '*.orig' -x Makefile DirectFB-0.9.25.1/gfxdrivers/i830/i830.c DirectFB-0.9.25.1-x86_64/gfxdrivers/i830/i830.c
--- DirectFB-0.9.25.1/gfxdrivers/i830/i830.c	2006-05-03 09:22:37.000000000 +0200
+++ DirectFB-0.9.25.1-x86_64/gfxdrivers/i830/i830.c	2006-07-14 00:00:10.000000000 +0200
@@ -23,6 +23,7 @@
    Free Software Foundation, Inc., 59 Temple Place - Suite 330,
    Boston, MA 02111-1307, USA.
 */
+#include <asm/types.h>
 #include "i830.h"
 
 #include <stdio.h>
diff -Naur -x '*.orig' -x Makefile DirectFB-0.9.25.1/gfxdrivers/i830/i830_overlay.c DirectFB-0.9.25.1-x86_64/gfxdrivers/i830/i830_overlay.c
--- DirectFB-0.9.25.1/gfxdrivers/i830/i830_overlay.c	2006-05-03 09:22:37.000000000 +0200
+++ DirectFB-0.9.25.1-x86_64/gfxdrivers/i830/i830_overlay.c	2006-07-14 00:00:41.000000000 +0200
@@ -43,6 +43,7 @@
  *        Matt Sottek <matthew.j.sottek@intel.com>
  */
 
+#include <asm/types.h>
 #include "i830.h"
 
 #include <math.h>
diff -Naur -x '*.orig' -x Makefile DirectFB-0.9.25.1/gfxdrivers/matrox/matrox_crtc2.c DirectFB-0.9.25.1-x86_64/gfxdrivers/matrox/matrox_crtc2.c
--- DirectFB-0.9.25.1/gfxdrivers/matrox/matrox_crtc2.c	2005-12-09 14:58:51.000000000 +0100
+++ DirectFB-0.9.25.1-x86_64/gfxdrivers/matrox/matrox_crtc2.c	2006-07-13 23:55:30.000000000 +0200
@@ -26,6 +26,7 @@
 */
 
 #include <stdio.h>
+#include <asm/types.h>
 #include <unistd.h>
 #include <stdlib.h>
 
diff -Naur -x '*.orig' -x Makefile DirectFB-0.9.25.1/gfxdrivers/matrox/matrox_screen_crtc2.c DirectFB-0.9.25.1-x86_64/gfxdrivers/matrox/matrox_screen_crtc2.c
--- DirectFB-0.9.25.1/gfxdrivers/matrox/matrox_screen_crtc2.c	2005-10-26 16:23:10.000000000 +0200
+++ DirectFB-0.9.25.1-x86_64/gfxdrivers/matrox/matrox_screen_crtc2.c	2006-07-13 23:55:53.000000000 +0200
@@ -26,6 +26,7 @@
 */
 
 #include <stdio.h>
+#include <asm/types.h>
 #include <unistd.h>
 #include <stdlib.h>
 
diff -Naur -x '*.orig' -x Makefile DirectFB-0.9.25.1/gfxdrivers/matrox/matrox_spic.c DirectFB-0.9.25.1-x86_64/gfxdrivers/matrox/matrox_spic.c
--- DirectFB-0.9.25.1/gfxdrivers/matrox/matrox_spic.c	2005-12-09 14:58:51.000000000 +0100
+++ DirectFB-0.9.25.1-x86_64/gfxdrivers/matrox/matrox_spic.c	2006-07-13 23:56:26.000000000 +0200
@@ -26,6 +26,7 @@
 */
 
 #include <stdio.h>
+#include <asm/types.h>
 #include <unistd.h>
 #include <stdlib.h>
 
diff -Naur -x '*.orig' -x Makefile DirectFB-0.9.25.1/gfxdrivers/nsc/nsc_galfns.c DirectFB-0.9.25.1-x86_64/gfxdrivers/nsc/nsc_galfns.c
--- DirectFB-0.9.25.1/gfxdrivers/nsc/nsc_galfns.c	2005-10-19 01:14:10.000000000 +0200
+++ DirectFB-0.9.25.1-x86_64/gfxdrivers/nsc/nsc_galfns.c	2006-07-14 00:02:18.000000000 +0200
@@ -144,6 +144,7 @@
 #include <stdlib.h>
 #include <unistd.h>
 #include <string.h>
+#include <asm/types.h>
 #include <sys/stat.h>
 #include <sys/sysmacros.h>
 #include <fcntl.h>
diff -Naur -x '*.orig' -x Makefile DirectFB-0.9.25.1/gfxdrivers/sis315/sis315.c DirectFB-0.9.25.1-x86_64/gfxdrivers/sis315/sis315.c
--- DirectFB-0.9.25.1/gfxdrivers/sis315/sis315.c	2006-05-03 09:22:37.000000000 +0200
+++ DirectFB-0.9.25.1-x86_64/gfxdrivers/sis315/sis315.c	2006-07-13 23:57:16.000000000 +0200
@@ -20,6 +20,7 @@
  */
 
 #include <stdio.h>
+#include <asm/types.h>
 #include <sys/ioctl.h>
 
 #include <directfb.h>
diff -Naur -x '*.orig' -x Makefile DirectFB-0.9.25.1/gfxdrivers/unichrome/uc_overlay.c DirectFB-0.9.25.1-x86_64/gfxdrivers/unichrome/uc_overlay.c
--- DirectFB-0.9.25.1/gfxdrivers/unichrome/uc_overlay.c	2005-12-01 19:51:06.000000000 +0100
+++ DirectFB-0.9.25.1-x86_64/gfxdrivers/unichrome/uc_overlay.c	2006-07-14 00:01:39.000000000 +0200
@@ -7,6 +7,7 @@
    version 2 of the License, or (at your option) any later version.
 */
 
+#include <asm/types.h>
 #include "unichrome.h"
 #include "uc_overlay.h"
 #include "vidregs.h"
diff -Naur -x '*.orig' -x Makefile DirectFB-0.9.25.1/inputdrivers/keyboard/keyboard.c DirectFB-0.9.25.1-x86_64/inputdrivers/keyboard/keyboard.c
--- DirectFB-0.9.25.1/inputdrivers/keyboard/keyboard.c	2005-10-19 01:14:13.000000000 +0200
+++ DirectFB-0.9.25.1-x86_64/inputdrivers/keyboard/keyboard.c	2006-07-13 23:54:12.000000000 +0200
@@ -28,6 +28,7 @@
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
+#include <asm/types.h>
 #include <unistd.h>
 
 #include <fcntl.h>
diff -Naur -x '*.orig' -x Makefile DirectFB-0.9.25.1/systems/fbdev/agp.c DirectFB-0.9.25.1-x86_64/systems/fbdev/agp.c
--- DirectFB-0.9.25.1/systems/fbdev/agp.c	2006-03-15 00:23:43.000000000 +0100
+++ DirectFB-0.9.25.1-x86_64/systems/fbdev/agp.c	2006-07-13 23:51:59.000000000 +0200
@@ -30,6 +30,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <asm/types.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
diff -Naur -x '*.orig' -x Makefile DirectFB-0.9.25.1/systems/fbdev/fbdev.c DirectFB-0.9.25.1-x86_64/systems/fbdev/fbdev.c
--- DirectFB-0.9.25.1/systems/fbdev/fbdev.c	2006-05-03 09:22:37.000000000 +0200
+++ DirectFB-0.9.25.1-x86_64/systems/fbdev/fbdev.c	2006-07-13 23:51:03.000000000 +0200
@@ -27,6 +27,7 @@
 
 #include <config.h>
 
+#include <asm/types.h>
 #include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
diff -Naur -x '*.orig' -x Makefile DirectFB-0.9.25.1/systems/fbdev/vt.c DirectFB-0.9.25.1-x86_64/systems/fbdev/vt.c
--- DirectFB-0.9.25.1/systems/fbdev/vt.c	2006-05-03 09:22:37.000000000 +0200
+++ DirectFB-0.9.25.1-x86_64/systems/fbdev/vt.c	2006-07-13 23:51:47.000000000 +0200
@@ -29,6 +29,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <asm/types.h>
 #include <unistd.h>
 #include <string.h>
 
