--- darktable-2.2.5/CMakeLists.txt	2017-05-28 16:22:21.000000000 +0200
+++ darktable-2.2.5.new/CMakeLists.txt	2017-08-29 12:04:42.274899623 +0200
@@ -321,7 +321,7 @@
   set(TESTBUILD_OPENCL_PROGRAMS OFF)
 
   # 3.9 is the first version with which this works.
-  find_package(LLVM 3.9 CONFIG)
+  find_package(LLVM 4.0 CONFIG)
   if (LLVM_FOUND)
     message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
 
