diff -Naur darktable-3.0.0/src/CMakeLists.txt darktable-3.0.0-p/src/CMakeLists.txt
--- darktable-3.0.0/src/CMakeLists.txt	2019-12-21 10:57:37.000000000 +0100
+++ darktable-3.0.0-p/src/CMakeLists.txt	2020-01-03 13:17:50.276765107 +0100
@@ -523,7 +523,7 @@
   # also, in IOPs, in-loops branches could be forced to be compiled as different loops variants
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${MARCH} ${DT_REQ_INSTRUCTIONS} -g")
   set(CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO} -O2")
-  set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O3")
+  set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} ")
   if(CMAKE_COMPILER_IS_GNUCC)
     if(BUILD_SSE2_CODEPATHS)
       set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -mfpmath=sse")
@@ -537,7 +537,7 @@
 
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${MARCH} ${DT_REQ_INSTRUCTIONS} -g")
   set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -O2")
-  set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3")
+  set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ")
   if(CMAKE_COMPILER_IS_GNUCXX)
     if(BUILD_SSE2_CODEPATHS)
       set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfpmath=sse")
diff -Naur darktable-3.0.0/src/external/LuaAutoC/CMakeLists.txt darktable-3.0.0-p/src/external/LuaAutoC/CMakeLists.txt
--- darktable-3.0.0/src/external/LuaAutoC/CMakeLists.txt	2019-12-21 10:57:37.000000000 +0100
+++ darktable-3.0.0-p/src/external/LuaAutoC/CMakeLists.txt	2020-01-03 13:18:33.906059021 +0100
@@ -5,7 +5,7 @@
 #
 # build liblautoc
 #
-set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -Wall -Wno-unused -O3 -g ")
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=gnu99 -Wall -Wno-unused  -g ")
 
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/include)
 
diff -Naur darktable-3.0.0/src/external/LuaAutoC/Makefile darktable-3.0.0-p/src/external/LuaAutoC/Makefile
--- darktable-3.0.0/src/external/LuaAutoC/Makefile	2019-12-21 10:57:37.000000000 +0100
+++ darktable-3.0.0-p/src/external/LuaAutoC/Makefile	2020-01-03 13:18:19.141959550 +0100
@@ -8,7 +8,7 @@
 LAC=lautoc.c
 LAC_OBJ=lautoc.o
 LAC_CPPFLAGS= -I./include $(LUA_INCLUDE_DIR)
-LAC_CFLAGS= -std=gnu99 -Wall -Werror -Wno-unused -O3 -g
+LAC_CFLAGS= -std=gnu99 -Wall -Werror -Wno-unused  -g
 LAC_LDFLAGS= $(LUA_LIBRARY_DIR)
 LAC_LIBS= $(LUA_LIBRARY)
 
diff -Naur darktable-3.0.0/src/external/rawspeed/cmake/compiler-flags.cmake darktable-3.0.0-p/src/external/rawspeed/cmake/compiler-flags.cmake
--- darktable-3.0.0/src/external/rawspeed/cmake/compiler-flags.cmake	2019-12-03 12:12:52.000000000 +0100
+++ darktable-3.0.0-p/src/external/rawspeed/cmake/compiler-flags.cmake	2020-01-03 13:18:53.787192985 +0100
@@ -143,7 +143,7 @@
     CMAKE_MODULE_LINKER_FLAGS_COVERAGE )
 
 # -fstack-protector-all
-set(SANITIZATION_DEFAULTS "-O3 -fno-optimize-sibling-calls")
+set(SANITIZATION_DEFAULTS " -fno-optimize-sibling-calls")
 
 set(asan "-fsanitize=address -fno-omit-frame-pointer -fno-common -U_FORTIFY_SOURCE")
 if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
@@ -167,7 +167,7 @@
     CMAKE_CXX_FLAGS_SANITIZE
     CMAKE_C_FLAGS_SANITIZE )
 
-set(fuzz "-O3 -ffast-math")
+set(fuzz " -ffast-math")
 
 if(NOT LIB_FUZZING_ENGINE)
   set(fuzz "${fuzz} ${asan} ${ubsan}")
@@ -238,5 +238,5 @@
 set(CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO} -O2")
 set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -O2")
 
-set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -O3")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3")
+set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} ")
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ")
