diff -Naur darktable-3.0.0/src/common/fast_guided_filter.h darktable-3.0.0-p/src/common/fast_guided_filter.h
--- darktable-3.0.0/src/common/fast_guided_filter.h	2019-12-21 10:57:37.000000000 +0100
+++ darktable-3.0.0-p/src/common/fast_guided_filter.h	2020-01-03 13:02:19.691530273 +0100
@@ -25,6 +25,7 @@
 
 #include "common/darktable.h"
 
+#if 0
 /** Note :
  * we use finite-math-only and fast-math because divisions by zero are manually avoided in the code
  * fp-contract=fast enables hardware-accelerated Fused Multiply-Add
@@ -41,6 +42,7 @@
                       "tree-loop-linear", "loop-block", "loop-strip-mine", \
                       "finite-math-only", "fp-contract=fast", "fast-math")
 #endif
+#endif
 
 #define MIN_FLOAT exp2f(-16.0f)
 
diff -Naur darktable-3.0.0/src/common/iop_order.c darktable-3.0.0-p/src/common/iop_order.c
--- darktable-3.0.0/src/common/iop_order.c	2019-12-21 10:57:37.000000000 +0100
+++ darktable-3.0.0-p/src/common/iop_order.c	2020-01-03 13:01:47.670319984 +0100
@@ -34,7 +34,7 @@
 
 #define DT_IOP_ORDER_INFO (darktable.unmuted & DT_DEBUG_IOPORDER)
 
-
+#if 0
 /** Note :
  * we do not use finite-math-only and fast-math because divisions by zero are not manually avoided in the code
  * fp-contract=fast enables hardware-accelerated Fused Multiply-Add
@@ -51,6 +51,7 @@
                       "fp-contract=fast", \
                       "tree-vectorize")
 #endif
+#endif
 
 static void _ioppr_insert_iop_after(GList **_iop_order_list, GList *history_list, const char *op_new, const char *op_previous, const int dont_move);
 static void _ioppr_insert_iop_before(GList **_iop_order_list, GList *history_list, const char *op_new, const char *op_next, const int dont_move);
diff -Naur darktable-3.0.0/src/common/luminance_mask.h darktable-3.0.0-p/src/common/luminance_mask.h
--- darktable-3.0.0/src/common/luminance_mask.h	2019-12-21 10:57:37.000000000 +0100
+++ darktable-3.0.0-p/src/common/luminance_mask.h	2020-01-03 13:01:17.807123955 +0100
@@ -28,6 +28,7 @@
 #include "develop/imageop_math.h"
 
 
+#if 0
 /** Note :
  * we use finite-math-only and fast-math because divisions by zero are manually avoided in the code
  * fp-contract=fast enables hardware-accelerated Fused Multiply-Add
@@ -44,6 +45,7 @@
                       "tree-loop-linear", "loop-block", "loop-strip-mine", \
                       "finite-math-only", "fp-contract=fast", "fast-math")
 #endif
+#endif
 
 #define MIN_FLOAT exp2f(-16.0f)
 
diff -Naur darktable-3.0.0/src/iop/choleski.h darktable-3.0.0-p/src/iop/choleski.h
--- darktable-3.0.0/src/iop/choleski.h	2019-12-21 10:57:37.000000000 +0100
+++ darktable-3.0.0-p/src/iop/choleski.h	2020-01-03 13:00:35.769848155 +0100
@@ -25,6 +25,7 @@
 #include "common/darktable.h"
 #include "develop/imageop_math.h"
 
+#if 0
 /** Note :
  * we use finite-math-only and fast-math because divisions by zero are manually avoided in the code
  * fp-contract=fast enables hardware-accelerated Fused Multiply-Add
@@ -41,6 +42,7 @@
                       "tree-loop-linear", "loop-block", "loop-strip-mine", \
                       "finite-math-only", "fp-contract=fast", "fast-math")
 #endif
+#endif
 
 /* DOCUMENTATION
  *
diff -Naur darktable-3.0.0/src/iop/filmicrgb.c darktable-3.0.0-p/src/iop/filmicrgb.c
--- darktable-3.0.0/src/iop/filmicrgb.c	2019-12-21 10:57:37.000000000 +0100
+++ darktable-3.0.0-p/src/iop/filmicrgb.c	2020-01-03 12:59:59.330609220 +0100
@@ -82,6 +82,7 @@
  * */
 
 
+#if 0
  /** Note :
  * we use finite-math-only and fast-math because divisions by zero are manually avoided in the code
  * fp-contract=fast enables hardware-accelerated Fused Multiply-Add
@@ -97,6 +98,7 @@
                       "tree-loop-linear", "loop-block", "loop-strip-mine", \
                       "finite-math-only", "fp-contract=fast", "fast-math")
 #endif
+#endif
 
 typedef enum dt_iop_filmicrgb_pickcolor_type_t
 {
diff -Naur darktable-3.0.0/src/iop/toneequal.c darktable-3.0.0-p/src/iop/toneequal.c
--- darktable-3.0.0/src/iop/toneequal.c	2019-12-21 10:57:37.000000000 +0100
+++ darktable-3.0.0-p/src/iop/toneequal.c	2020-01-03 12:59:11.143293448 +0100
@@ -109,6 +109,7 @@
 
 DT_MODULE_INTROSPECTION(2, dt_iop_toneequalizer_params_t)
 
+#if 0
 
 /** Note :
  * we use finite-math-only and fast-math because divisions by zero are manually avoided in the code
@@ -126,6 +127,7 @@
                       "finite-math-only", "fp-contract=fast", "fast-math", \
                       "tree-vectorize")
 #endif
+#endif
 
 #define UI_SAMPLES 256 // 128 is a bit small for 4K resolution
 #define CONTRAST_FULCRUM exp2f(-4.0f)
