diff -Naur darktable-3.4.0/src/common/fast_guided_filter.h darktable-3.4.0.new/src/common/fast_guided_filter.h
--- darktable-3.4.0/src/common/fast_guided_filter.h	2020-12-16 09:22:52.000000000 +0100
+++ darktable-3.4.0.new/src/common/fast_guided_filter.h	2021-01-14 14:09:30.340995797 +0100
@@ -33,17 +33,6 @@
  * the rest is loop reorganization and vectorization optimization
  **/
 
-#if defined(__GNUC__)
-#pragma GCC optimize ("unroll-loops", "tree-loop-if-convert", \
-                      "tree-loop-distribution", "no-strict-aliasing", \
-                      "loop-interchange", "loop-nest-optimize", "tree-loop-im", \
-                      "unswitch-loops", "tree-loop-ivcanon", "ira-loop-pressure", \
-                      "split-ivs-in-unroller", "variable-expansion-in-unroller", \
-                      "split-loops", "ivopts", "predictive-commoning",\
-                      "tree-loop-linear", "loop-block", "loop-strip-mine", \
-                      "finite-math-only", "fp-contract=fast", "fast-math")
-#endif
-
 #define MIN_FLOAT exp2f(-16.0f)
 
 
diff -Naur darktable-3.4.0/src/common/iop_order.c darktable-3.4.0.new/src/common/iop_order.c
--- darktable-3.4.0/src/common/iop_order.c	2020-12-16 09:22:52.000000000 +0100
+++ darktable-3.4.0.new/src/common/iop_order.c	2021-01-14 14:10:18.804273582 +0100
@@ -41,18 +41,6 @@
  * fp-contract=fast enables hardware-accelerated Fused Multiply-Add
  * the rest is loop reorganization and vectorization optimization
  **/
-#if defined(__GNUC__)
-#pragma GCC optimize ("unroll-loops", "tree-loop-if-convert", \
-                      "tree-loop-distribution", "no-strict-aliasing", \
-                      "loop-interchange", "loop-nest-optimize", "tree-loop-im", \
-                      "unswitch-loops", "tree-loop-ivcanon", "ira-loop-pressure", \
-                      "split-ivs-in-unroller", "variable-expansion-in-unroller", \
-                      "split-loops", "ivopts", "predictive-commoning",\
-                      "tree-loop-linear", "loop-block", "loop-strip-mine", \
-                      "fp-contract=fast", \
-                      "tree-vectorize")
-#endif
-
 const char *iop_order_string[] =
 {
   N_("custom"),
diff -Naur darktable-3.4.0/src/common/iop_profile.c darktable-3.4.0.new/src/common/iop_profile.c
--- darktable-3.4.0/src/common/iop_profile.c	2020-12-16 09:22:52.000000000 +0100
+++ darktable-3.4.0.new/src/common/iop_profile.c	2021-01-14 14:10:02.044177502 +0100
@@ -36,17 +36,6 @@
  * fp-contract=fast enables hardware-accelerated Fused Multiply-Add
  * the rest is loop reorganization and vectorization optimization
  **/
-#if defined(__GNUC__)
-#pragma GCC optimize ("unroll-loops", "tree-loop-if-convert", \
-                      "tree-loop-distribution", "no-strict-aliasing", \
-                      "loop-interchange", "loop-nest-optimize", "tree-loop-im", \
-                      "unswitch-loops", "tree-loop-ivcanon", "ira-loop-pressure", \
-                      "split-ivs-in-unroller", "variable-expansion-in-unroller", \
-                      "split-loops", "ivopts", "predictive-commoning",\
-                      "tree-loop-linear", "loop-block", "loop-strip-mine", \
-                      "fp-contract=fast", \
-                      "tree-vectorize")
-#endif
 
 static void _transform_from_to_rgb_lab_lcms2(const float *const image_in, float *const image_out, const int width,
                                              const int height, const dt_colorspaces_color_profile_type_t type,
diff -Naur darktable-3.4.0/src/common/luminance_mask.h darktable-3.4.0.new/src/common/luminance_mask.h
--- darktable-3.4.0/src/common/luminance_mask.h	2020-12-16 09:22:52.000000000 +0100
+++ darktable-3.4.0.new/src/common/luminance_mask.h	2021-01-14 14:09:47.730095454 +0100
@@ -34,17 +34,6 @@
  * the rest is loop reorganization and vectorization optimization
  **/
 
-#if defined(__GNUC__)
-#pragma GCC optimize ("unroll-loops", "tree-loop-if-convert", \
-                      "tree-loop-distribution", "no-strict-aliasing", \
-                      "loop-interchange", "loop-nest-optimize", "tree-loop-im", \
-                      "unswitch-loops", "tree-loop-ivcanon", "ira-loop-pressure", \
-                      "split-ivs-in-unroller", "variable-expansion-in-unroller", \
-                      "split-loops", "ivopts", "predictive-commoning",\
-                      "tree-loop-linear", "loop-block", "loop-strip-mine", \
-                      "finite-math-only", "fp-contract=fast", "fast-math")
-#endif
-
 #define MIN_FLOAT exp2f(-16.0f)
 
 
diff -Naur darktable-3.4.0/src/develop/blends/blendif_lab.c darktable-3.4.0.new/src/develop/blends/blendif_lab.c
--- darktable-3.4.0/src/develop/blends/blendif_lab.c	2020-12-16 09:22:52.000000000 +0100
+++ darktable-3.4.0.new/src/develop/blends/blendif_lab.c	2021-01-14 14:09:01.500830554 +0100
@@ -16,15 +16,6 @@
     along with darktable.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#if defined(__GNUC__)
-#pragma GCC optimize("unroll-loops", "tree-loop-if-convert", "tree-loop-distribution", "no-strict-aliasing",      \
-                     "loop-interchange", "loop-nest-optimize", "tree-loop-im", "unswitch-loops",                  \
-                     "tree-loop-ivcanon", "ira-loop-pressure", "split-ivs-in-unroller", "tree-loop-vectorize",    \
-                     "variable-expansion-in-unroller", "split-loops", "ivopts", "predictive-commoning",           \
-                     "tree-loop-linear", "loop-block", "loop-strip-mine", "finite-math-only", "fp-contract=fast", \
-                     "fast-math", "no-math-errno")
-#endif
-
 #include "common/colorspaces_inline_conversions.h"
 #include "common/math.h"
 #include "develop/blend.h"
diff -Naur darktable-3.4.0/src/develop/blends/blendif_raw.c darktable-3.4.0.new/src/develop/blends/blendif_raw.c
--- darktable-3.4.0/src/develop/blends/blendif_raw.c	2020-12-16 09:22:52.000000000 +0100
+++ darktable-3.4.0.new/src/develop/blends/blendif_raw.c	2021-01-14 14:08:47.434749970 +0100
@@ -16,15 +16,6 @@
     along with darktable.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#if defined(__GNUC__)
-#pragma GCC optimize("unroll-loops", "tree-loop-if-convert", "tree-loop-distribution", "no-strict-aliasing",      \
-                     "loop-interchange", "loop-nest-optimize", "tree-loop-im", "unswitch-loops",                  \
-                     "tree-loop-ivcanon", "ira-loop-pressure", "split-ivs-in-unroller", "tree-loop-vectorize",    \
-                     "variable-expansion-in-unroller", "split-loops", "ivopts", "predictive-commoning",           \
-                     "tree-loop-linear", "loop-block", "loop-strip-mine", "finite-math-only", "fp-contract=fast", \
-                     "fast-math", "no-math-errno")
-#endif
-
 #include "common/math.h"
 #include "develop/blend.h"
 #include "develop/imageop.h"
diff -Naur darktable-3.4.0/src/develop/blends/blendif_rgb_hsl.c darktable-3.4.0.new/src/develop/blends/blendif_rgb_hsl.c
--- darktable-3.4.0/src/develop/blends/blendif_rgb_hsl.c	2020-12-16 09:22:52.000000000 +0100
+++ darktable-3.4.0.new/src/develop/blends/blendif_rgb_hsl.c	2021-01-14 14:08:32.011661632 +0100
@@ -16,15 +16,6 @@
     along with darktable.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#if defined(__GNUC__)
-#pragma GCC optimize("unroll-loops", "tree-loop-if-convert", "tree-loop-distribution", "no-strict-aliasing",      \
-                     "loop-interchange", "loop-nest-optimize", "tree-loop-im", "unswitch-loops",                  \
-                     "tree-loop-ivcanon", "ira-loop-pressure", "split-ivs-in-unroller", "tree-loop-vectorize",    \
-                     "variable-expansion-in-unroller", "split-loops", "ivopts", "predictive-commoning",           \
-                     "tree-loop-linear", "loop-block", "loop-strip-mine", "finite-math-only", "fp-contract=fast", \
-                     "fast-math", "no-math-errno")
-#endif
-
 #include "common/colorspaces_inline_conversions.h"
 #include "common/math.h"
 #include "develop/blend.h"
diff -Naur darktable-3.4.0/src/develop/blends/blendif_rgb_jzczhz.c darktable-3.4.0.new/src/develop/blends/blendif_rgb_jzczhz.c
--- darktable-3.4.0/src/develop/blends/blendif_rgb_jzczhz.c	2020-12-16 09:22:52.000000000 +0100
+++ darktable-3.4.0.new/src/develop/blends/blendif_rgb_jzczhz.c	2021-01-14 14:08:11.811545946 +0100
@@ -16,15 +16,6 @@
     along with darktable.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#if defined(__GNUC__)
-#pragma GCC optimize("unroll-loops", "tree-loop-if-convert", "tree-loop-distribution", "no-strict-aliasing",      \
-                     "loop-interchange", "loop-nest-optimize", "tree-loop-im", "unswitch-loops",                  \
-                     "tree-loop-ivcanon", "ira-loop-pressure", "split-ivs-in-unroller", "tree-loop-vectorize",    \
-                     "variable-expansion-in-unroller", "split-loops", "ivopts", "predictive-commoning",           \
-                     "tree-loop-linear", "loop-block", "loop-strip-mine", "finite-math-only", "fp-contract=fast", \
-                     "fast-math", "no-math-errno")
-#endif
-
 #include "common/colorspaces_inline_conversions.h"
 #include "develop/blend.h"
 #include "develop/imageop.h"
diff -Naur darktable-3.4.0/src/iop/channelmixerrgb.c darktable-3.4.0.new/src/iop/channelmixerrgb.c
--- darktable-3.4.0/src/iop/channelmixerrgb.c	2020-12-16 09:22:52.000000000 +0100
+++ darktable-3.4.0.new/src/iop/channelmixerrgb.c	2021-01-14 14:10:49.052447019 +0100
@@ -49,17 +49,6 @@
  * fp-contract=fast enables hardware-accelerated Fused Multiply-Add
  * the rest is loop reorganization and vectorization optimization
  **/
-#if defined(__GNUC__)
-#pragma GCC optimize ("unroll-loops", "tree-loop-if-convert", \
-                      "tree-loop-distribution", "no-strict-aliasing", \
-                      "loop-interchange", "loop-nest-optimize", "tree-loop-im", \
-                      "unswitch-loops", "tree-loop-ivcanon", "ira-loop-pressure", \
-                      "split-ivs-in-unroller", "variable-expansion-in-unroller", \
-                      "split-loops", "ivopts", "predictive-commoning",\
-                      "tree-loop-linear", "loop-block", "loop-strip-mine", \
-                      "finite-math-only", "fp-contract=fast", "fast-math", \
-                      "tree-vectorize", "no-math-errno")
-#endif
 
 
 #define CHANNEL_SIZE 4
diff -Naur darktable-3.4.0/src/iop/choleski.h darktable-3.4.0.new/src/iop/choleski.h
--- darktable-3.4.0/src/iop/choleski.h	2020-12-16 09:22:52.000000000 +0100
+++ darktable-3.4.0.new/src/iop/choleski.h	2021-01-14 14:10:37.041378144 +0100
@@ -31,16 +31,6 @@
  * the rest is loop reorganization and vectorization optimization
  **/
 
-#if defined(__GNUC__)
-#pragma GCC optimize ("unroll-loops", "tree-loop-if-convert", \
-                      "tree-loop-distribution", "no-strict-aliasing", \
-                      "loop-interchange", "loop-nest-optimize", "tree-loop-im", \
-                      "unswitch-loops", "tree-loop-ivcanon", "ira-loop-pressure", \
-                      "split-ivs-in-unroller", "variable-expansion-in-unroller", \
-                      "split-loops", "ivopts", "predictive-commoning",\
-                      "tree-loop-linear", "loop-block", "loop-strip-mine", \
-                      "finite-math-only", "fp-contract=fast", "fast-math")
-#endif
 
 /* DOCUMENTATION
  *
diff -Naur darktable-3.4.0/src/iop/filmicrgb.c darktable-3.4.0.new/src/iop/filmicrgb.c
--- darktable-3.4.0/src/iop/filmicrgb.c	2020-12-16 09:22:52.000000000 +0100
+++ darktable-3.4.0.new/src/iop/filmicrgb.c	2021-01-14 14:11:02.676525153 +0100
@@ -94,15 +94,6 @@
  * fp-contract=fast enables hardware-accelerated Fused Multiply-Add
  * the rest is loop reorganization and vectorization optimization
  **/
-#if defined(__GNUC__)
-#pragma GCC optimize("unroll-loops", "tree-loop-if-convert", "tree-loop-distribution", "no-strict-aliasing",      \
-                     "loop-interchange", "loop-nest-optimize", "tree-loop-im", "unswitch-loops",                  \
-                     "tree-loop-ivcanon", "ira-loop-pressure", "split-ivs-in-unroller",                           \
-                     "variable-expansion-in-unroller", "split-loops", "ivopts", "predictive-commoning",           \
-                     "tree-loop-linear", "loop-block", "loop-strip-mine", "finite-math-only", "fp-contract=fast", \
-                     "fast-math", "no-math-errno")
-#endif
-
 
 typedef enum dt_iop_filmicrgb_methods_type_t
 {
diff -Naur darktable-3.4.0/src/iop/negadoctor.c darktable-3.4.0.new/src/iop/negadoctor.c
--- darktable-3.4.0/src/iop/negadoctor.c	2020-12-16 09:22:52.000000000 +0100
+++ darktable-3.4.0.new/src/iop/negadoctor.c	2021-01-14 14:11:17.915612558 +0100
@@ -40,17 +40,6 @@
 #include <math.h>
 #include <stdlib.h>
 
-#if defined(__GNUC__)
-#pragma GCC optimize ("unroll-loops", "tree-loop-if-convert", \
-                      "tree-loop-distribution", "no-strict-aliasing", \
-                      "loop-interchange", "loop-nest-optimize", "tree-loop-im", \
-                      "unswitch-loops", "tree-loop-ivcanon", "ira-loop-pressure", \
-                      "split-ivs-in-unroller", "variable-expansion-in-unroller", \
-                      "split-loops", "ivopts", "predictive-commoning",\
-                      "tree-loop-linear", "loop-block", "loop-strip-mine", \
-                      "finite-math-only", "fp-contract=fast", "fast-math")
-#endif
-
 /** DOCUMENTATION
  *
  * This module allows to invert scanned negatives and simulate their print on paper, based on Kodak Cineon
diff -Naur darktable-3.4.0/src/iop/toneequal.c darktable-3.4.0.new/src/iop/toneequal.c
--- darktable-3.4.0/src/iop/toneequal.c	2020-12-16 09:22:52.000000000 +0100
+++ darktable-3.4.0.new/src/iop/toneequal.c	2021-01-14 14:11:32.796697924 +0100
@@ -116,17 +116,6 @@
  * fp-contract=fast enables hardware-accelerated Fused Multiply-Add
  * the rest is loop reorganization and vectorization optimization
  **/
-#if defined(__GNUC__)
-#pragma GCC optimize ("unroll-loops", "tree-loop-if-convert", \
-                      "tree-loop-distribution", "no-strict-aliasing", \
-                      "loop-interchange", "loop-nest-optimize", "tree-loop-im", \
-                      "unswitch-loops", "tree-loop-ivcanon", "ira-loop-pressure", \
-                      "split-ivs-in-unroller", "variable-expansion-in-unroller", \
-                      "split-loops", "ivopts", "predictive-commoning",\
-                      "tree-loop-linear", "loop-block", "loop-strip-mine", \
-                      "finite-math-only", "fp-contract=fast", "fast-math", \
-                      "tree-vectorize")
-#endif
 
 #define UI_SAMPLES 256 // 128 is a bit small for 4K resolution
 #define CONTRAST_FULCRUM exp2f(-4.0f)
