diff -Naur signon-ui-master-075830f31096947fcea70802b8d3c142bccfbefa/signon-ui.pro signon-ui-master-075830f31096947fcea70802b8d3c142bccfbefa-p/signon-ui.pro
--- signon-ui-master-075830f31096947fcea70802b8d3c142bccfbefa/signon-ui.pro	2015-08-10 10:37:45.000000000 +0200
+++ signon-ui-master-075830f31096947fcea70802b8d3c142bccfbefa-p/signon-ui.pro	2016-04-01 15:40:22.675370840 +0200
@@ -5,8 +5,7 @@
 CONFIG += ordered
 SUBDIRS = \
     po \
-    src \
-    tests
+    src
 
 include(common-installs-config.pri)
 
diff -Naur signon-ui-master-075830f31096947fcea70802b8d3c142bccfbefa/src/request.cpp signon-ui-master-075830f31096947fcea70802b8d3c142bccfbefa-p/src/request.cpp
--- signon-ui-master-075830f31096947fcea70802b8d3c142bccfbefa/src/request.cpp	2015-08-10 10:37:45.000000000 +0200
+++ signon-ui-master-075830f31096947fcea70802b8d3c142bccfbefa-p/src/request.cpp	2016-04-01 15:40:51.906137213 +0200
@@ -18,7 +18,7 @@
  * with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#define HAS_XEMBED (QT_VERSION < QT_VERSION_CHECK(5, 0, 0))
+#undef HAS_XEMBED
 #define HAS_FOREIGN_QWINDOW (QT_VERSION >= QT_VERSION_CHECK(5, 1, 0) || \
                              defined(FORCE_FOREIGN_QWINDOW))
 #include "request.h"
