--- safe-exceptions-0.1.7.4/safe-exceptions.cabal	2023-06-26 19:37:00.000000000 +0200
+++ safe-exceptions-0.1.7.4.new/safe-exceptions.cabal	2023-12-13 11:25:32.714071659 +0100
@@ -17,7 +17,7 @@
   hs-source-dirs:      src
   exposed-modules:     Control.Exception.Safe
   build-depends:       base >= 4.11 && < 5
-                     , deepseq >= 1.2 && < 1.5
+                     , deepseq >= 1.2 && < 1.6
                      , exceptions >= 0.10 && < 0.11
                      , transformers >= 0.2 && < 0.7
   default-language:    Haskell2010
