diff -Naur hotplug-2004_03_29.orig/etc/hotplug/hotplug.functions hotplug-2004_03_29/etc/hotplug/hotplug.functions
--- hotplug-2004_03_29.orig/etc/hotplug/hotplug.functions	2004-03-29 21:25:59.000000000 +0200
+++ hotplug-2004_03_29/etc/hotplug/hotplug.functions	2004-07-20 13:55:10.000000000 +0200
@@ -140,7 +140,7 @@
 	    # it's not an error if a module doesn't exist or won't load.
 	    if $MODPROBE -n $MODULE >/dev/null 2>&1 &&
 		    ! $MODPROBE $MODULE >/dev/null 2>&1 ; then
-		mesg "... can't load module $MODULE"
+		debug_mesg "... can't load module $MODULE"
 	    else
 		# /etc/modules.conf may have set non-default module
 		# parameters ... handle per-device parameters in apps
@@ -164,7 +164,7 @@
 	fi
 
 	if [ $LOADED = false ]; then
-	    mesg "missing kernel or user mode driver $MODULE "
+	    debug_mesg "missing kernel or user mode driver $MODULE "
 	fi
 	if echo "$MODULE" | grep -q "usb-storage" > /dev/null 2>&1 ; then
 	    [ -x /usr/sbin/updfstab ] &&  /usr/sbin/updfstab
