diff -Naur hotplug-2004_03_29.orig/etc/hotplug/net.agent hotplug-2004_03_29/etc/hotplug/net.agent
--- hotplug-2004_03_29.orig/etc/hotplug/net.agent	2004-03-30 00:23:31.000000000 +0200
+++ hotplug-2004_03_29/etc/hotplug/net.agent	2004-07-20 13:56:50.000000000 +0200
@@ -62,9 +62,17 @@
 		fi;
 	    fi
 
+            # Frugalware Linux:
+            if [ -x /etc/rc.d/rc.interfaces ]; then
+                # Interface already up?  If so, skip.
+                if ! /sbin/ifconfig | grep "^${INTERFACE} " 1> /dev/null ; then
+                    debug_mesg run rc.interfaces
+                    exec /etc/rc.d/rc.interfaces
+                fi
+
 	    # RedHat and similar
-	    export IN_HOTPLUG=1
-	    if [ -x /sbin/ifup ]; then
+	    elif [ -x /sbin/ifup ]; then
+	        export IN_HOTPLUG=1
 		debug_mesg invoke ifup $INTERFACE
 		exec /sbin/ifup $INTERFACE
 	    else
