diff -Naur subtitleripper.orig/Makefile subtitleripper/Makefile
--- subtitleripper.orig/Makefile	2003-12-08 10:28:02.000000000 +0100
+++ subtitleripper/Makefile	2006-09-05 02:29:52.000000000 +0200
@@ -7,9 +7,9 @@
 LIBS    := -lm 
 INCLUDES :=
 
-### enable ppm support ###
+### enable netpbm support ###
 DEFINES  += -D_HAVE_LIB_PPM_
-LIBS     += -lppm
+LIBS     += -lnetpbm
 
 ### enable PNG support ###
 DEFINES += -D_HAVE_PNG_
