diff -Naur subtitleripper.orig/Makefile subtitleripper/Makefile
--- subtitleripper.orig/Makefile	2003-12-08 10:28:02.000000000 +0100
+++ subtitleripper/Makefile	2006-09-05 02:21:47.000000000 +0200
@@ -6,6 +6,8 @@
 DEFINES :=  
 LIBS    := -lm 
 INCLUDES :=
+PREFIX ?= /usr
+DESTDIR ?=
 
 ### enable ppm support ###
 DEFINES  += -D_HAVE_LIB_PPM_
@@ -96,6 +98,13 @@
 	@rm -rf $(PACKAGE)
 	@mv $(PACKAGE)-$(VERSION)-$(RELEASE).tgz $(HOME)/
 
+install:
+	mkdir -p $(DESTDIR)$(PREFIX)/bin
+	install -m 755 pgm2txt $(DESTDIR)$(PREFIX)/bin/
+	install -m 755 srttool $(DESTDIR)$(PREFIX)/bin/
+	install -m 755 subtitle2pgm $(DESTDIR)$(PREFIX)/bin/
+	install -m 755 subtitle2vobsub $(DESTDIR)$(PREFIX)/bin/
+
 # Build a rpm package (tested for SuSE 7.3)
 rpm:
 	@mkdir $(PACKAGE)
