diff -Naur vim62.orig/runtime/vimrc_example.vim vim62/runtime/vimrc_example.vim
--- vim62.orig/runtime/vimrc_example.vim	2002-09-19 10:16:52.000000000 +0200
+++ vim62/runtime/vimrc_example.vim	2004-03-21 14:36:48.000000000 +0100
@@ -21,11 +21,7 @@
 " allow backspacing over everything in insert mode
 set backspace=indent,eol,start
 
-if has("vms")
-  set nobackup		" do not keep a backup file, use versions instead
-else
-  set backup		" keep a backup file
-endif
+set nobackup		" do not keep a backup file, use versions instead
 set history=50		" keep 50 lines of command line history
 set ruler		" show the cursor position all the time
 set showcmd		" display incomplete commands
