diff -Naur aegisub/aegisub/src/menu.cpp aegisub.new/aegisub/src/menu.cpp
--- aegisub/src/menu.cpp	2013-07-04 20:05:39.000000000 +0200
+++ aegisub.new/src/menu.cpp	2013-11-06 17:02:30.254248453 +0100
@@ -462,11 +462,11 @@
 		window->Bind(wxEVT_COMMAND_MENU_SELECTED, &CommandManager::OnMenuClick, &menu->cm);
 		window->SetMenuBar(menu.get());
 
-#ifdef __WXGTK__
-		// GTK silently swallows keypresses for accelerators whose associated
-		// menu items are disabled. As we don't update the menu until it's
-		// opened, this means that conditional hotkeys don't work if the menu
-		// hasn't been opened since they became valid.
+#if defined(__WXGTK__) && !wxCHECK_VERSION(3, 0, 0)
+		// Older versions of wxGTK silently swallow keypresses for accelerators
+		// whose associated menu items are disabled. As we don't update the
+		// menu until it's opened, this means that conditional hotkeys don't
+		// work if the menu hasn't been opened since they became valid.
 		//
 		// To work around this, we completely disable accelerators from menu
 		// item. wxGTK doesn't expose any way to do this other that at wx
diff -Naur aegisub/aegisub/src/utils.cpp aegisub.new/aegisub/src/utils.cpp
--- aegisub/src/utils.cpp	2013-07-04 20:05:39.000000000 +0200
+++ aegisub.new/src/utils.cpp	2013-11-06 17:01:30.614247354 +0100
@@ -279,8 +279,7 @@
 	config::opt->Flush();
 
 #if defined(__WXMSW__)
-	wxStandardPaths stand;
-	wxExecute("\"" + stand.GetExecutablePath() + "\"");
+	wxExecute("\"" + wxStandardPaths::Get().GetExecutablePath() + "\"");
 #elif defined(__WXMAC__)
 	std::string bundle_path = agi::util::OSX_GetBundlePath();
 	std::string helper_path = agi::util::OSX_GetBundleAuxillaryExecutablePath("restart-helper");
@@ -290,8 +289,7 @@
 	LOG_I("util/restart/exec") << exec;
 	wxExecute(exec);
 #else
-	wxStandardPaths stand;
-	wxExecute(stand.GetExecutablePath());
+	wxExecute(wxStandardPaths::Get().GetExecutablePath());
 #endif
 }
 
