diff -Naur lives-0.9.8.5/Makefile.am lives-0.9.8.5-p/Makefile.am
--- lives-0.9.8.5/Makefile.am	2007-05-28 07:59:57.000000000 +0200
+++ lives-0.9.8.5-p/Makefile.am	2007-06-05 01:54:27.000000000 +0200
@@ -52,8 +52,8 @@
 	cp -f $(srcdir)/build-lives-rfx-plugin-multi $(DESTDIR)$(bindir) && chmod +rx $(DESTDIR)$(bindir)/build-lives-rfx-plugin-multi
 	cp -f $(srcdir)/default.keymap $(DESTDIR)$(datadir)/$(PACKAGE)/
 	$(DESTDIR)$(bindir)/build-lives-rfx-plugin-multi builtin $(DESTDIR)$(datadir) $(DESTDIR)$(bindir)
-	cp -f $(srcdir)/menu-entries/LiVES.desktop /usr/share/applications/
-	cp -f $(srcdir)/menu-entries/lives.xpm /usr/share/pixmaps/
+	cp -f $(srcdir)/menu-entries/LiVES.desktop $(DESTDIR)/usr/share/applications/LiVES.desktop
+	cp -f $(srcdir)/menu-entries/lives.xpm $(DESTDIR)/usr/share/pixmaps/lives.xpm
 
 	$(top_srcdir)/po/update_with_plugins.sh install $(PACKAGE) $(prefix) $(DESTDIR)
 
diff -Naur lives-0.9.8.5/Makefile.in lives-0.9.8.5-p/Makefile.in
--- lives-0.9.8.5/Makefile.in	2007-05-28 08:00:10.000000000 +0200
+++ lives-0.9.8.5-p/Makefile.in	2007-06-05 01:55:38.000000000 +0200
@@ -764,8 +764,8 @@
 	cp -f $(srcdir)/build-lives-rfx-plugin-multi $(DESTDIR)$(bindir) && chmod +rx $(DESTDIR)$(bindir)/build-lives-rfx-plugin-multi
 	cp -f $(srcdir)/default.keymap $(DESTDIR)$(datadir)/$(PACKAGE)/
 	$(DESTDIR)$(bindir)/build-lives-rfx-plugin-multi builtin $(DESTDIR)$(datadir) $(DESTDIR)$(bindir)
-	cp -f $(srcdir)/menu-entries/LiVES.desktop /usr/share/applications/
-	cp -f $(srcdir)/menu-entries/lives.xpm /usr/share/pixmaps/
+	cp -f $(srcdir)/menu-entries/LiVES.desktop $(DESTDIR)/usr/share/applications/LiVES.desktop
+	cp -f $(srcdir)/menu-entries/lives.xpm $(DESTDIR)/usr/share/pixmaps/lives.xpm
 
 	$(top_srcdir)/po/update_with_plugins.sh install $(PACKAGE) $(prefix) $(DESTDIR)
 
