diff -Naurp frozen-bubble-2.0.0/Makefile frozen-bubble-2.0.0-p/Makefile
--- frozen-bubble-2.0.0/Makefile	2006-10-05 21:39:10.000000000 +0200
+++ frozen-bubble-2.0.0-p/Makefile	2006-10-28 14:51:35.000000000 +0200
@@ -1,6 +1,6 @@
 DIRS = c_stuff po server
 
-PREFIX = /usr/local
+PREFIX = /usr
 DATADIR = $(PREFIX)/share
 LIBDIR = $(PREFIX)/lib
 BINDIR = $(PREFIX)/bin
diff -Naurp frozen-bubble-2.0.0/po/Makefile frozen-bubble-2.0.0-p/po/Makefile
--- frozen-bubble-2.0.0/po/Makefile	2006-10-26 21:26:24.000000000 +0200
+++ frozen-bubble-2.0.0-p/po/Makefile	2006-10-28 14:51:35.000000000 +0200
@@ -1,4 +1,4 @@
-PREFIX = /usr/local
+PREFIX = /usr
 DATADIR = $(PREFIX)/share
 LOCALEDIR = $(DATADIR)/locale
 
diff -Naurp frozen-bubble-2.0.0/server/Makefile frozen-bubble-2.0.0-p/server/Makefile
--- frozen-bubble-2.0.0/server/Makefile	2006-10-26 20:42:48.000000000 +0200
+++ frozen-bubble-2.0.0-p/server/Makefile	2006-10-28 14:52:19.000000000 +0200
@@ -1,8 +1,8 @@
-PREFIX = /usr/local
+PREFIX = /usr
 LIBDIR = $(PREFIX)/lib
 
-CFLAGS ?= -g -Wall -Werror -pipe `pkg-config glib-2.0 --cflags`
-LIBS ?= `pkg-config glib-2.0 --libs`
+CFLAGS += -g -Wall -pipe `pkg-config glib-2.0 --cflags`
+LIBS += `pkg-config glib-2.0 --libs`
 
 ALLSRC = fb-server.c log.c tools.c game.c net.c
 ALLOBJ = $(subst .c,.o,$(ALLSRC))
