--- lapack-3.3.1/SRC/Makefile	2011-03-07 16:11:26.000000000 +0100
+++ lapack-3.3.1/SRC/Makefile	2011-09-11 17:41:25.091427683 +0200
@@ -371,7 +371,7 @@
    zla_heamv.o zla_hercond_c.o zla_hercond_x.o zla_herpvgrw.o		\
    zla_lin_berr.o zlarscl2.o zlascl2.o zla_wwaddw.o
 
-all: ../$(LAPACKLIB)
+all: ../$(LAPACKLIB) shared
 
 ifdef USEXBLAS
 ALLXOBJ=$(SXLASRC) $(DXLASRC) $(CXLASRC) $(ZXLASRC) $(ALLXAUX)
@@ -384,6 +384,9 @@
 	$(ARCH) $(ARCHFLAGS) $@ $(ALLOBJ) $(ALLXOBJ)
 	$(RANLIB) $@
 
+shared: $(ALLOBJ) $(ALLXOBJ)
+	cc $(CFLAGS) -shared -Wl,-soname,liblapack.so.3 -o liblapack.so.3.3.1 $(ALLOBJ) -L.. -lm -lgfortran -lc
+
 single: $(SLASRC) $(ALLAUX) $(SCLAUX) 
 	$(ARCH) $(ARCHFLAGS) ../$(LAPACKLIB) $(SLASRC) $(ALLAUX) \
 	$(SCLAUX) $(DSLASRC)
@@ -434,4 +437,3 @@
 dla_wwaddw.o: dla_wwaddw.f ; $(FORTRAN) $(NOOPT) -c $< -o $@
 cla_wwaddw.o: cla_wwaddw.f ; $(FORTRAN) $(NOOPT) -c $< -o $@
 zla_wwaddw.o: zla_wwaddw.f ; $(FORTRAN) $(NOOPT) -c $< -o $@
-
