--- make.inc.old	2005-08-17 12:55:04.000000000 -0500
+++ make.inc	2005-08-17 12:34:07.000000000 -0500
@@ -34,8 +34,8 @@
 #  machine-specific, optimized BLAS library should be used whenever
 #  possible.)
 #
-BLASLIB      = ../../blas$(PLAT).a
-LAPACKLIB    = lapack$(PLAT).a
+BLASLIB      = /lib/libblas.a
+LAPACKLIB    = liblapack.a
 TMGLIB       = tmglib$(PLAT).a
 EIGSRCLIB    = eigsrc$(PLAT).a
 LINSRCLIB    = linsrc$(PLAT).a
