From 50f01c2d6b28110f8517a77960e456c1125bdb6e Mon Sep 17 00:00:00 2001
From: Arno Rehn <arno@arnorehn.de>
Date: Sat, 3 Dec 2011 17:57:43 +0100
Subject: [PATCH] fix a crash when generating the akonadi bindings

BUG: 283677
---
 type_compiler.cpp |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/type_compiler.cpp b/type_compiler.cpp
index 6ebf602..6e4ab70 100644
--- a/type_compiler.cpp
+++ b/type_compiler.cpp
@@ -191,6 +191,8 @@ void TypeCompiler::visitParameterDeclaration(ParameterDeclarationAST* node)
 
 void TypeCompiler::visitPtrOperator(PtrOperatorAST* node)
 {
+    if ( ! m_session->token_stream ) return;
+    if ( ! token_text(m_session->token_stream->kind(node->op)) ) return;
     if (token_text(m_session->token_stream->kind(node->op))[0] == '*') {
         QPair<bool, bool> cv = m_visitor->parseCv(node->cv);
         pointerDepth.append(cv.first);
-- 
1.7.8

