--- Makefile.am.old	2008-05-28 21:37:22.000000000 +0200
+++ Makefile.am	2008-07-18 15:19:06.000000000 +0200
@@ -89,8 +89,8 @@
 install-data-hook: 
 	$(UPDATE_DESKTOP)
 	if [ "x$(DIR_GIO)" != "x" ] ; then \
-		$(install_sh_DATA) $(top_builddir)/gio-sharp/gio/gio-sharp.dll $(libdir)/f-spot/; \
-		$(install_sh_DATA) $(top_builddir)/gio-sharp/gio/gio-sharp.dll.config $(libdir)/f-spot/; \
+		$(install_sh_DATA) $(top_builddir)/gio-sharp/gio/gio-sharp.dll $(DESTDIR)$(libdir)/f-spot/; \
+		$(install_sh_DATA) $(top_builddir)/gio-sharp/gio/gio-sharp.dll.config $(DESTDIR)$(libdir)/f-spot/; \
 	fi
 
 uninstall-hook: 
