diff -Naur unifdef-1.0/Makefile unifdef-1.0-p/Makefile
--- unifdef-1.0/Makefile	2000-02-12 04:54:36.000000000 +0100
+++ unifdef-1.0-p/Makefile	2006-06-14 01:10:30.000000000 +0200
@@ -9,5 +9,5 @@
 	rm -f unifdef *.o
 
 install: unifdef
-	install -m 0755 unifdef $(DEST)/bin
-	install -m 0644 unifdef.1 $(DEST)/man/man1 
+	install -m 0755 unifdef $(DESTDIR)$(DEST)/bin
+	install -m 0644 unifdef.1 $(DESTDIR)$(DEST)/man/man1 
