diff -Naur FreeImage/Source/LibRawLite/internal/dcraw_common.cpp FreeImage-p/Source/LibRawLite/internal/dcraw_common.cpp
--- FreeImage/Source/LibRawLite/internal/dcraw_common.cpp	2015-03-08 21:27:50.000000000 +0100
+++ FreeImage-p/Source/LibRawLite/internal/dcraw_common.cpp	2018-02-07 20:35:47.628666837 +0100
@@ -6937,7 +6937,7 @@
                 if (table_buf[iLensData+9] &&
                     (fabs(imgdata.lens.makernotes.CurFocal) < 0.1f))
                   imgdata.lens.makernotes.CurFocal =
-                    10*(table_buf[iLensData+9]>>2) * powf64(4, (table_buf[iLensData+9] & 0x03)-2);
+                    10*(table_buf[iLensData+9]>>2) * powf64((float)4, (table_buf[iLensData+9] & 0x03)-2);
                 if (table_buf[iLensData+10] & 0xf0)
                   imgdata.lens.makernotes.MaxAp4CurFocal =
                     powf64(2.0f, (float)((table_buf[iLensData+10] & 0xf0) >>4)/4.0f);
@@ -7809,7 +7809,7 @@
               {
                 if (table_buf[iLensData+9] && (fabs(imgdata.lens.makernotes.CurFocal) < 0.1f))
                   imgdata.lens.makernotes.CurFocal =
-                    10*(table_buf[iLensData+9]>>2) * powf64(4, (table_buf[iLensData+9] & 0x03)-2);
+                    10*(table_buf[iLensData+9]>>2) * powf64((float)4, (table_buf[iLensData+9] & 0x03)-2);
                 if (table_buf[iLensData+10] & 0xf0)
                   imgdata.lens.makernotes.MaxAp4CurFocal =
                     powf64(2.0f, (float)((table_buf[iLensData+10] & 0xf0) >>4)/4.0f);
