diff -Naur nilfs-utils-2.2.9/configure nilfs-utils-2.2.9.new/configure
--- nilfs-utils-2.2.9/configure	2022-06-20 12:08:01.000000000 +0200
+++ nilfs-utils-2.2.9.new/configure	2024-01-16 10:52:51.461216437 +0100
@@ -13973,7 +13973,7 @@
 
 test "x$prefix" = "xNONE" && prefix="/usr"
 test "x$exec_prefix" = "xNONE" && exec_prefix="${prefix}"
-root_sbindir=/sbin
+root_sbindir=/usr/bin
 
 sbindir=${exec_prefix}/sbin
 
