diff -Naur john-1.8.0-jumbo-1/src/DES_bs_b.c john-1.8.0-jumbo-1-p/src/DES_bs_b.c
--- john-1.8.0-jumbo-1/src/DES_bs_b.c	2014-05-16 15:40:40.000000000 +0200
+++ john-1.8.0-jumbo-1-p/src/DES_bs_b.c	2016-12-12 20:22:03.273436555 +0100
@@ -1247,6 +1247,9 @@
 #endif
 
 #if DES_bs_mt
+#if __GNUC__ >= 5
+extern
+#endif
 MAYBE_INLINE void DES_bs_set_salt_for_thread(int t, unsigned int salt)
 #else
 void DES_bs_set_salt(ARCH_WORD salt)
diff -Naur john-1.8.0-jumbo-1/src/MD5_std.c john-1.8.0-jumbo-1-p/src/MD5_std.c
--- john-1.8.0-jumbo-1/src/MD5_std.c	2014-05-16 15:40:40.000000000 +0200
+++ john-1.8.0-jumbo-1-p/src/MD5_std.c	2016-12-12 20:22:03.274436555 +0100
@@ -490,9 +490,15 @@
 #if MD5_std_mt
 #define MD5_body(x, out) \
 	MD5_body_for_thread(t, x, out)
+#if __GNUC__ >= 5
+extern
+#endif
 MAYBE_INLINE_BODY void MD5_body_for_thread(int t,
 	MD5_word x[15], MD5_word out[4])
 #else
+#if __GNUC__ >= 5
+extern
+#endif
 MAYBE_INLINE_BODY void MD5_body(MD5_word x[15], MD5_word out[4])
 #endif
 {
@@ -589,10 +595,16 @@
 #if MD5_std_mt
 #define MD5_body(x0, x1, out0, out1) \
 	MD5_body_for_thread(t, x0, x1, out0, out1)
+#if __GNUC__ >= 5
+extern
+#endif
 MAYBE_INLINE_BODY void MD5_body_for_thread(int t,
 	MD5_word x0[15], MD5_word x1[15],
 	MD5_word out0[4], MD5_word out1[4])
 #else
+#if __GNUC__ >= 5
+extern
+#endif
 MAYBE_INLINE_BODY void MD5_body(MD5_word x0[15], MD5_word x1[15],
 	MD5_word out0[4], MD5_word out1[4])
 #endif
