--- hamster-applet-2.24.2old/configure	2008-11-24 19:55:16.000000000 +0100
+++ hamster-applet-2.24.2/configure	2008-11-25 23:45:23.000000000 +0100
@@ -14678,40 +14678,8 @@
 echo "${ECHO_T}$PYGTK_DEFSDIR" >&6; }
 
 
-if test "x$PYTHON_VERSION" = "x2.5"; then
-    { echo "$as_me:$LINENO: checking for python sqlite module" >&5
-echo $ECHO_N "checking for python sqlite module... $ECHO_C" >&6; }
-    if { (echo "$as_me:$LINENO: DISPLAY= \$PYTHON -c '
-try:
-    import sqlite3
-except ImportError, e:
-    if str(e).find(\"sqlite3\") >= 0:
-        raise
-except:
-    pass
-    '") >&5
-  (DISPLAY= $PYTHON -c '
-try:
-    import sqlite3
-except ImportError, e:
-    if str(e).find("sqlite3") >= 0:
-        raise
-except:
-    pass
-    ') 2>&5
-  ac_status=$?
-  echo "$as_me:$LINENO: \$? = $ac_status" >&5
-  (exit $ac_status); }; then
-      { echo "$as_me:$LINENO: result: yes" >&5
-echo "${ECHO_T}yes" >&6; }
-    else
-      { echo "$as_me:$LINENO: result: no" >&5
-echo "${ECHO_T}no" >&6; }
-      { { echo "$as_me:$LINENO: error: Python 2.5: inbuilt sqlite is required to build hamster" >&5
-echo "$as_me: error: Python 2.5: inbuilt sqlite is required to build hamster" >&2;}
-   { (exit 1); exit 1; }; }
-    fi
-else
+
+
     { echo "$as_me:$LINENO: checking for pysqlite2 module" >&5
 echo $ECHO_N "checking for pysqlite2 module... $ECHO_C" >&6; }
     if { (echo "$as_me:$LINENO: DISPLAY= \$PYTHON -c '
@@ -14744,7 +14712,7 @@
 echo "$as_me: error: Python 2.4: pysqlite2 module required to build hamster" >&2;}
    { (exit 1); exit 1; }; }
     fi
-fi
+
 
 
 
