--- Kana no quiz-1.9.5/setup.py	2009-07-17 17:04:20.000000000 +0000
+++ Kana no quiz-1.9.5.new/setup.py	2012-03-18 23:40:33.000000000 +0000
@@ -1,4 +1,4 @@
-﻿#!/usr/bin/env python
+#!/usr/bin/env python
 """Kana no quiz!
 
    Copyleft 2003, 2004, 2005, 2006, 2007, 2008, 2009 Choplair-network.
@@ -26,11 +26,7 @@
 try: import py2exe
 except: pass
 
-if 'bdist_wininst' in sys.argv:
-   scriptfiles = glob.glob(os.path.join('data', 'script', '*.py*'))    
-elif "install" in sys.argv:
-   scriptfiles = [os.path.join("data", "script", "kana-no-quiz")]
-else: scriptfiles = []
+scriptfiles = [os.path.join("data", "script", "kana-no-quiz")]
 
 # Common data files.
 datafiles = [
@@ -115,20 +111,7 @@
    packages   =   ['kananoquiz'],
    package_dir   =   {'kananoquiz': 'data/py'},
    scripts      =   scriptfiles,
-   data_files   =   datafiles,
-   # The following is for py2exe...
-   windows    =   [{
-            'script': os.path.join("data", "script", "kana-no-quiz"),
-            'icon_resources': [(1, os.path.join("data", "img",
-               "icon.ico"))]
-            }],
-   options      =   {
-            'py2exe': {
-               'packages':'encodings',
-               'includes': 'cairo, pango, pangocairo, atk, '\
-                  'gobject, psyco, random, sets, gettext',
-               'compressed': 1, 'optimize': 1
-            }}
+   data_files   =   datafiles
    )
 
 if "install" in sys.argv:
