--- smb4k-2.0.82/doc/index.docbook	2018-02-25 11:11:55.000000000 +0100
+++ smb4k-2.0.82.new/doc/index.docbook	2018-03-16 13:22:37.260672921 +0100
@@ -27,11 +27,7 @@
     <holder>Alexander Reinholdt</holder>
   </copyright>
   <legalnotice>&underFDL;</legalnotice>
-<<<<<<< HEAD
-  <date>2018-01-07</date>
-=======
   <date>2018-02-22</date>
->>>>>>> 77700d2... Updated handbook.
   <releaseinfo>&smb4k; 2.1.0</releaseinfo><!--FIXME change releaseinfo number-->
   <abstract>
     <para>&smb4k; is an advanced network neighborhood browser and Samba share mounting utility.</para>
