--- smb4k-2.0.0/CMakeLists.txt	2017-03-11 10:03:54.000000000 +0100
+++ smb4k-2.0.0.new/CMakeLists.txt	2017-03-12 21:06:41.000000000 +0100
@@ -77,7 +77,6 @@
 if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/po")
   ki18n_install(po)
   if(KF5DocTools_FOUND)
-    kdoctools_install(po)
   endif(KF5DocTools_FOUND)
 endif(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/po")
 
@@ -108,4 +107,3 @@
 find_package(KF5I18n CONFIG REQUIRED)
 ki18n_install(po)
 find_package(KF5DocTools CONFIG REQUIRED)
-kdoctools_install(po)
