diff -Naur smb4k-2.0.0/CMakeLists.txt smb4k-2.0.0-p/CMakeLists.txt
--- smb4k-2.0.0/CMakeLists.txt	2017-03-11 10:03:54.000000000 +0100
+++ smb4k-2.0.0-p/CMakeLists.txt	2017-05-15 14:34:33.691399981 +0200
@@ -9,6 +9,7 @@
 cmake_minimum_required(VERSION 3.3 FATAL_ERROR)
 cmake_policy(SET CMP0037 NEW)
 cmake_policy(SET CMP0063 NEW)
+cmake_policy(SET CMP0002 OLD)
 
 find_package(ECM 1.1.0 REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
