diff -Naur gyachi-1.0.5/gyachi.desktop gyachi-1.0.5-p/gyachi.desktop
--- gyachi-1.0.5/gyachi.desktop	2006-03-20 11:02:18.000000000 +0100
+++ gyachi-1.0.5-p/gyachi.desktop	2006-08-24 21:17:18.723699376 +0200
@@ -6,7 +6,7 @@
 Comment[sr]=Програм за Yahoo!
 Comment=Yahoo! Client
 Exec=gyachi
-Icon=gyachi.png
+Icon=/usr/share/gyachi/pixmaps/gyach-icon.xpm
 Terminal=false
 Type=Application
-Categories=Application;Network;
+Categories=Gtk;Qt;Application;Network;
diff -Naur gyachi-1.0.5/Makefile.am gyachi-1.0.5-p/Makefile.am
--- gyachi-1.0.5/Makefile.am	2006-05-03 00:50:03.000000000 +0200
+++ gyachi-1.0.5-p/Makefile.am	2006-08-24 21:17:18.738697096 +0200
@@ -3,7 +3,6 @@
 
 util_DATA = gyachi.desktop
 utildir = $(datadir)/applications
-codecdir = $(libdir)/win32
 
 EXTRA_DIST = autogen.sh config.rpath mkinstalldirs gyachi.desktop gyachi.spec gyachi.spec.in
 
@@ -65,14 +64,6 @@
 	    fi \
 	  done \
 	fi
-	if test -d $(srcdir)/codecs; then \
-	  $(mkinstalldirs) $(DESTDIR)$(codecdir); \
-	  for codec in $(srcdir)/codecs/*; do \
-	    if test -f $$codec; then \
-	      $(INSTALL_DATA) $$codec $(DESTDIR)$(codecdir); \
-	    fi \
-	  done \
-	fi
 	
 uninstall-local:
 	rm -rf $(DESTDIR)$(pkgdatadir)
@@ -134,11 +125,3 @@
 	    fi \
 	  done \
 	fi
-	if test -d codecs; then \
-	  mkdir -p $(distdir)/codecs; \
-	  for codec in codecs/*; do \
-	    if test -f $$codec; then \
-	      cp -p $$codec $(distdir)/codecs; \
-	    fi \
-	  done \
-	fi
diff -Naur gyachi-1.0.5/Makefile.in gyachi-1.0.5-p/Makefile.in
--- gyachi-1.0.5/Makefile.in	2006-08-23 21:16:34.000000000 +0200
+++ gyachi-1.0.5-p/Makefile.in	2006-08-24 21:17:06.738521400 +0200
@@ -242,7 +242,6 @@
 SUBDIRS = intl po gyvoice client webcam plugins
 util_DATA = gyachi.desktop
 utildir = $(datadir)/applications
-codecdir = $(libdir)/win32
 EXTRA_DIST = autogen.sh config.rpath mkinstalldirs gyachi.desktop gyachi.spec gyachi.spec.in
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
@@ -768,14 +767,6 @@
 	    fi \
 	  done \
 	fi
-	if test -d $(srcdir)/codecs; then \
-	  $(mkinstalldirs) $(DESTDIR)$(codecdir); \
-	  for codec in $(srcdir)/codecs/*; do \
-	    if test -f $$codec; then \
-	      $(INSTALL_DATA) $$codec $(DESTDIR)$(codecdir); \
-	    fi \
-	  done \
-	fi
 
 uninstall-local:
 	rm -rf $(DESTDIR)$(pkgdatadir)
@@ -837,14 +828,7 @@
 	    fi \
 	  done \
 	fi
-	if test -d codecs; then \
-	  mkdir -p $(distdir)/codecs; \
-	  for codec in codecs/*; do \
-	    if test -f $$codec; then \
-	      cp -p $$codec $(distdir)/codecs; \
-	    fi \
-	  done \
-	fi
+
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
