--- gnome-do-0.5.0.1/configure.ac.old	2008-06-10 06:16:10.000000000 +0200
+++ gnome-do-0.5.0.1/configure.ac	2008-06-26 22:29:50.000000000 +0200
@@ -44,7 +44,7 @@
 PKG_CHECK_MODULES([NDESK_DBUS_GLIB_10], [ndesk-dbus-glib-1.0])
 PKG_CHECK_MODULES([GNOME_SHARP_20], [gnome-sharp-2.0])
 PKG_CHECK_MODULES([GNOME_DESKTOP_SHARP_20], [gnome-desktop-sharp-2.0])
-PKG_CHECK_MODULES([GNOME_KEYRING_SHARP], [gnome-keyring-sharp])
+PKG_CHECK_MODULES([GNOME_KEYRING_SHARP], [gnome-keyring-sharp-1.0])
 PKG_CHECK_MODULES([GLIB_SHARP_20], [glib-sharp-2.0])
 PKG_CHECK_MODULES([GCONF_SHARP_20], [gconf-sharp-2.0])
 PKG_CHECK_MODULES([GLADE_SHARP_20], [glade-sharp-2.0])
--- gnome-do-0.5.0.1/Do.Addins/Makefile.am.old	2008-06-10 02:48:12.000000000 +0200
+++ gnome-do-0.5.0.1/Do.Addins/Makefile.am	2008-06-26 22:43:05.000000000 +0200
@@ -101,7 +101,7 @@
 	-pkg:gconf-sharp-2.0 \
 	-pkg:glib-sharp-2.0 \
 	-pkg:gnome-desktop-sharp-2.0 \
-	-pkg:gnome-keyring-sharp \
+	-pkg:gnome-keyring-sharp-1.0 \
 	-pkg:gnome-sharp-2.0 \
 	-pkg:gnome-vfs-sharp-2.0 \
 	-pkg:gtk-sharp-2.0 \

