Index: gnome-do-0.8.3/libdo/Makefile.am
===================================================================
--- gnome-do-0.8.3.orig/libdo/Makefile.am
+++ gnome-do-0.8.3/libdo/Makefile.am
@@ -2,9 +2,6 @@ INCLUDES = -I$(top_srcdir)
 
 AM_CFLAGS = -DG_LOG_DOMAIN=\"libdo\"	\
 	    -DEGG_COMPILATION 		\
-	    -DGTK_DISABLE_DEPRECATED 	\
-	    -DGDK_DISABLE_DEPRECATED 	\
-	    -DG_DISABLE_DEPRECATED	\
 	    $(LIBDO_CFLAGS)
 
 
