diff -Naur gaim-2.0.0beta3.orig/src/Makefile.in gaim-2.0.0beta3/src/Makefile.in
--- gaim-2.0.0beta3.orig/src/Makefile.in	2006-07-26 18:33:47.306224344 +0200
+++ gaim-2.0.0beta3/src/Makefile.in	2006-07-26 18:36:01.190870784 +0200
@@ -136,13 +136,13 @@
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
+	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -fPIE
 LTCOMPILE = $(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) \
 	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
+	$(AM_CFLAGS) $(CFLAGS) -fPIE
 CCLD = $(CC)
 LINK = $(LIBTOOL) --tag=CC --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
-	$(AM_LDFLAGS) $(LDFLAGS) -o $@
+	$(AM_LDFLAGS) $(LDFLAGS) -o $@ -pie
 SOURCES = $(libgaim_client_la_SOURCES) $(gaim_SOURCES) \
 	$(gaim_client_example_SOURCES)
 DIST_SOURCES = $(am__libgaim_client_la_SOURCES_DIST) \
