diff -Naur gaim-2.0.0beta3.orig/src/protocols/silc/silc.c gaim-2.0.0beta3/src/protocols/silc/silc.c
--- gaim-2.0.0beta3.orig/src/protocols/silc/silc.c	2006-07-26 18:33:47.673168560 +0200
+++ gaim-2.0.0beta3/src/protocols/silc/silc.c	2006-07-26 18:34:49.144823448 +0200
@@ -268,6 +268,7 @@
 	GaimConnection *gc;
 	char pkd[256], prd[256];
 	const char *cipher, *hmac;
+	char *realname;
 	int i;
 
 	gc = account->gc;
@@ -298,11 +299,16 @@
 		client->username = silc_get_username();
 		gaim_account_set_username(account, client->username);
 	}
+	realname = silc_get_real_name();
 	if (gaim_account_get_user_info(account)) {
 		client->realname = strdup(gaim_account_get_user_info(account));
-	} else {
-		client->realname = silc_get_real_name();
+		free(realname);
+	} else if ((silc_get_real_name() != NULL) && (*realname != '\0')) {
+		client->realname = realname;
 		gaim_account_set_user_info(account, client->realname);
+	} else {
+		free(realname);
+		client->realname = strdup(_("Gaim User"));
 	}
 	client->hostname = silc_net_localhost();
 
