diff -ur gaim-2.0.0beta3/configure.ac gaim-2.0.0beta3-patched/configure.ac
--- gaim-2.0.0beta3/configure.ac	2006-03-26 07:45:35.000000000 +0000
+++ gaim-2.0.0beta3-patched/configure.ac	2006-05-07 23:13:46.000000000 +0000
@@ -183,15 +183,17 @@
 HOWL_CFLAGS=""
 HOWL_LIBS=""
 
-dnl Attempt to autodetect Howl
-PKG_CHECK_MODULES(HOWL, howl, [
-		howlincludes="yes"
-		howllibs="yes"
-	], [
-		AC_MSG_RESULT(no)
-		howlincludes="no"
-		howllibs="no"
-	])
+dnl First we attempt to use avahi-compat-howl
+PKG_CHECK_MODULES(HOWL, avahi-compat-howl,
+        [howlincludes="yes" howllibs="yes"],
+        [howlincludes="no" howllibs="no"])
+
+dnl If that fails, autodetect the "real" howl
+if test "$howlincludes" = "no" ; then
+        PKG_CHECK_MODULES(HOWL, howl,
+                [howlincludes="yes" howllibs="yes"],
+                [howlincludes="no" howllibs="no"])
+fi
 
 dnl Override HOWL_CFLAGS if the user specified an include dir
 if test "$ac_howl_includes" != "no"; then
