diff -Naur gaim-2.0.0beta3.orig/src/gtkblist.c gaim-2.0.0beta3/src/gtkblist.c
--- gaim-2.0.0beta3.orig/src/gtkblist.c	2006-07-26 18:33:53.980209744 +0200
+++ gaim-2.0.0beta3/src/gtkblist.c	2006-07-26 18:35:16.812617304 +0200
@@ -3422,7 +3422,7 @@
 	len = g_utf8_strlen(normalized, -1);
 	log_attrs = g_new(PangoLogAttr, len + 1);
 
-	pango_get_log_attrs(normalized, len, -1, NULL, log_attrs, len + 1);
+	pango_get_log_attrs(normalized, strlen(normalized), -1, NULL, log_attrs, len + 1);
 
 	word = normalized;
 	result = TRUE;
