diff -Naur gaim-2.0.0beta3/plugins/perl/Makefile.am gaim-2.0.0beta3-p/plugins/perl/Makefile.am
--- gaim-2.0.0beta3/plugins/perl/Makefile.am	2005-10-29 03:05:29.000000000 +0200
+++ gaim-2.0.0beta3-p/plugins/perl/Makefile.am	2006-07-26 23:54:18.152706064 +0200
@@ -92,7 +92,7 @@
 			${LN_S} -f ../${srcdir}/$$f $$f; \
 		done; \
 	fi
-	@cd common && $(perlpath) Makefile.PL $(PERL_MM_PARAMS)
+	@cd common && $(perlpath) Makefile.PL
 
 common/Makefile.PL: common/Makefile.PL.in $(top_builddir)/config.status
 	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)
@@ -101,7 +101,7 @@
 	@for dir in $(perl_dirs); do \
 		cd $$dir && \
 		if [ ! -f Makefile ]; then \
-			$(perlpath) Makefile.PL $(PERL_MM_PARAMS); \
+			$(perlpath) Makefile.PL; \
 		fi && \
 		($(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS) || \
 		 $(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS)) && \
diff -Naur gaim-2.0.0beta3/plugins/perl/Makefile.in gaim-2.0.0beta3-p/plugins/perl/Makefile.in
--- gaim-2.0.0beta3/plugins/perl/Makefile.in	2006-03-29 00:39:12.000000000 +0200
+++ gaim-2.0.0beta3-p/plugins/perl/Makefile.in	2006-07-26 23:56:25.307375616 +0200
@@ -749,7 +749,7 @@
 			${LN_S} -f ../${srcdir}/$$f $$f; \
 		done; \
 	fi
-	@cd common && $(perlpath) Makefile.PL $(PERL_MM_PARAMS)
+	@cd common && $(perlpath) Makefile.PL
 
 common/Makefile.PL: common/Makefile.PL.in $(top_builddir)/config.status
 	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)
@@ -758,7 +758,7 @@
 	@for dir in $(perl_dirs); do \
 		cd $$dir && \
 		if [ ! -f Makefile ]; then \
-			$(perlpath) Makefile.PL $(PERL_MM_PARAMS); \
+			$(perlpath) Makefile.PL; \
 		fi && \
 		($(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS) || \
 		 $(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS)) && \
