diff -Naur gaim-2.0.0beta3/plugins/musicmessaging/musicmessaging.c gaim-2.0.0beta3-p/plugins/musicmessaging/musicmessaging.c
--- gaim-2.0.0beta3/plugins/musicmessaging/musicmessaging.c	2006-02-11 20:16:37.000000000 +0100
+++ gaim-2.0.0beta3-p/plugins/musicmessaging/musicmessaging.c	2006-07-27 00:04:29.713734656 +0200
@@ -594,7 +594,6 @@
 	gtk_widget_show(sep);
 	gtk_widget_show(image);
 	gtk_widget_show(button);
-	
 	gtk_box_pack_start(GTK_BOX(GAIM_GTK_CONVERSATION(conv)->toolbar), sep, FALSE, FALSE, 0);
 	gtk_box_pack_start(GTK_BOX(GAIM_GTK_CONVERSATION(conv)->toolbar), button, FALSE, FALSE, 0);
 }
@@ -627,7 +626,7 @@
 	editor_path_label = gtk_label_new(_("Score Editor Path"));
 	editor_path_button = gtk_button_new_with_mnemonic(_("_Apply"));
 	
-	gtk_entry_set_text((GtkEntry*)editor_path, "/usr/local/bin/gscore");
+	gtk_entry_set_text((GtkEntry*)editor_path, "/usr/bin/gscore");
 	
 	g_signal_connect(G_OBJECT(editor_path_button), "clicked",
 					 G_CALLBACK(set_editor_path), editor_path);
@@ -678,7 +677,7 @@
 static void
 init_plugin(GaimPlugin *plugin) {
 	gaim_prefs_add_none("/plugins/gtk/musicmessaging");
-	gaim_prefs_add_string("/plugins/gtk/musicmessaging/editor_path", "/usr/local/bin/gscore");
+	gaim_prefs_add_string("/plugins/gtk/musicmessaging/editor_path", "/usr/bin/gscore");
 }
 
 GAIM_INIT_PLUGIN(musicmessaging, init_plugin, info);
