diff -Naur gaim-2.0.0beta4.orig/gtk/plugins/perl/Makefile.am gaim-2.0.0beta4/gtk/plugins/perl/Makefile.am
--- gaim-2.0.0beta4.orig/gtk/plugins/perl/Makefile.am	2006-10-26 12:28:46.000000000 +0200
+++ gaim-2.0.0beta4/gtk/plugins/perl/Makefile.am	2006-10-26 12:58:07.000000000 +0200
@@ -42,7 +42,7 @@
 			${LN_S} -f ../${srcdir}/$$f $$f; \
 		done; \
 	fi
-	@cd common && $(perlpath) Makefile.PL $(PERL_MM_PARAMS)
+	@cd common && $(perlpath) Makefile.PL
 
 common/Makefile.PL: common/Makefile.PL.in $(top_builddir)/config.status
 	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)
@@ -51,7 +51,7 @@
 	@for dir in $(perl_dirs); do \
 		cd $$dir && \
 		if [ ! -f Makefile ]; then \
-			$(perlpath) Makefile.PL $(PERL_MM_PARAMS); \
+			$(perlpath) Makefile.PL; \
 		fi && \
 		($(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS) || \
 		$(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS)) && \
diff -Naur gaim-2.0.0beta4.orig/gtk/plugins/perl/Makefile.in gaim-2.0.0beta4/gtk/plugins/perl/Makefile.in
--- gaim-2.0.0beta4.orig/gtk/plugins/perl/Makefile.in	2006-10-26 12:28:46.000000000 +0200
+++ gaim-2.0.0beta4/gtk/plugins/perl/Makefile.in	2006-10-26 12:58:07.000000000 +0200
@@ -579,7 +579,7 @@
 			${LN_S} -f ../${srcdir}/$$f $$f; \
 		done; \
 	fi
-	@cd common && $(perlpath) Makefile.PL $(PERL_MM_PARAMS)
+	@cd common && $(perlpath) Makefile.PL
 
 common/Makefile.PL: common/Makefile.PL.in $(top_builddir)/config.status
 	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)
@@ -588,7 +588,7 @@
 	@for dir in $(perl_dirs); do \
 		cd $$dir && \
 		if [ ! -f Makefile ]; then \
-			$(perlpath) Makefile.PL $(PERL_MM_PARAMS); \
+			$(perlpath) Makefile.PL; \
 		fi && \
 		($(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS) || \
 		$(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS)) && \
diff -Naur gaim-2.0.0beta4.orig/libgaim/plugins/perl/Makefile.am gaim-2.0.0beta4/libgaim/plugins/perl/Makefile.am
--- gaim-2.0.0beta4.orig/libgaim/plugins/perl/Makefile.am	2006-10-26 12:28:46.000000000 +0200
+++ gaim-2.0.0beta4/libgaim/plugins/perl/Makefile.am	2006-10-26 12:57:38.000000000 +0200
@@ -97,7 +97,7 @@
 			${LN_S} -f ../${srcdir}/$$f $$f; \
 		done; \
 	fi
-	@cd common && $(perlpath) Makefile.PL $(PERL_MM_PARAMS)
+	@cd common && $(perlpath) Makefile.PL
 
 common/Makefile.PL: common/Makefile.PL.in $(top_builddir)/config.status
 	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)
@@ -106,7 +106,7 @@
 	@for dir in $(perl_dirs); do \
 		cd $$dir && \
 		if [ ! -f Makefile ]; then \
-			$(perlpath) Makefile.PL $(PERL_MM_PARAMS); \
+			$(perlpath) Makefile.PL; \
 		fi && \
 		($(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS) || \
 		 $(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS)) && \
diff -Naur gaim-2.0.0beta4.orig/libgaim/plugins/perl/Makefile.in gaim-2.0.0beta4/libgaim/plugins/perl/Makefile.in
--- gaim-2.0.0beta4.orig/libgaim/plugins/perl/Makefile.in	2006-10-26 12:28:46.000000000 +0200
+++ gaim-2.0.0beta4/libgaim/plugins/perl/Makefile.in	2006-10-26 12:57:50.000000000 +0200
@@ -804,7 +804,7 @@
 			${LN_S} -f ../${srcdir}/$$f $$f; \
 		done; \
 	fi
-	@cd common && $(perlpath) Makefile.PL $(PERL_MM_PARAMS)
+	@cd common && $(perlpath) Makefile.PL
 
 common/Makefile.PL: common/Makefile.PL.in $(top_builddir)/config.status
 	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)
@@ -813,7 +813,7 @@
 	@for dir in $(perl_dirs); do \
 		cd $$dir && \
 		if [ ! -f Makefile ]; then \
-			$(perlpath) Makefile.PL $(PERL_MM_PARAMS); \
+			$(perlpath) Makefile.PL; \
 		fi && \
 		($(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS) || \
 		 $(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS)) && \
