--- cborg-0.2.10.0/cborg.cabal	2001-09-09 03:46:40.000000000 +0200
+++ cborg-0.2.10.0.new/cborg.cabal	2024-07-10 23:07:38.999673736 +0200
@@ -1,5 +1,5 @@
 name:                cborg
-version:             0.2.10.0
+version:             0.2.9.0
 synopsis:            Concise Binary Object Representation (CBOR)
 license:             BSD3
 license-file:        LICENSE.txt
@@ -90,7 +90,7 @@
 
   build-depends:
     array                   >= 0.4     && < 0.6,
-    base                    >= 4.11    && < 4.20,
+    base                    >= 4.11    && < 4.21,
     bytestring              >= 0.10.4  && < 0.13,
     containers              >= 0.5     && < 0.8,
     deepseq                 >= 1.0     && < 1.6,
@@ -148,7 +148,7 @@
 
   build-depends:
     array                   >= 0.4     && < 0.6,
-    base                    >= 4.11    && < 4.20,
+    base                    >= 4.11    && < 4.21,
     base-orphans,
     bytestring              >= 0.10.4  && < 0.13,
     text                    >= 1.1     && < 2.2,
@@ -159,7 +159,7 @@
     base16-bytestring       >= 1.0     && < 1.1,
     deepseq                 >= 1.0     && < 1.6,
     half                    >= 0.2.2.3 && < 0.4,
-    QuickCheck              >= 2.9     && < 2.15,
+    QuickCheck              >= 2.9     && < 2.16,
     random,
     scientific              >= 0.3     && < 0.4,
     tasty                   >= 0.11    && < 1.6,
