diff -Naur kwin-5.11.5/kcmkwin/CMakeLists.txt kwin-5.11.5-p/kcmkwin/CMakeLists.txt
--- kwin-5.11.5/kcmkwin/CMakeLists.txt	2018-01-02 13:34:16.849372782 +0100
+++ kwin-5.11.5-p/kcmkwin/CMakeLists.txt	2018-01-14 03:44:59.627934129 +0100
@@ -5,7 +5,7 @@
 add_subdirectory( kwindecoration )
 add_subdirectory( kwinrules )
 add_subdirectory( kwinscreenedges )
-add_subdirectory( kwinscripts )
+#add_subdirectory( kwinscripts )
 add_subdirectory( kwindesktop )
 
 if( KWIN_BUILD_TABBOX )
