diff -Naur kwin-5.20.0/kwin.kcfg kwin-5.20.0-p/kwin.kcfg
--- kwin-5.20.0/kwin.kcfg	2020-10-12 17:19:57.000000000 +0200
+++ kwin-5.20.0-p/kwin.kcfg	2020-10-15 02:54:25.170507944 +0200
@@ -127,7 +127,7 @@
                 <choice name="OnMainWindow"/>
                 <choice name="Maximizing"/>
             </choices>
-            <default>Placement::Smart</default>
+            <default>Placement::Centered</default>
         </entry>
         <entry name="AutoRaise" type="Bool">
             <default>false</default>
