diff -Naur kwin-5.20.0/kcmkwin/kwinoptions/kwinoptions_settings.kcfg kwin-5.20.0-p/kcmkwin/kwinoptions/kwinoptions_settings.kcfg
--- kwin-5.20.0/kcmkwin/kwinoptions/kwinoptions_settings.kcfg	2020-10-12 17:19:57.000000000 +0200
+++ kwin-5.20.0-p/kcmkwin/kwinoptions/kwinoptions_settings.kcfg	2020-10-15 04:58:11.526753969 +0200
@@ -51,7 +51,7 @@
             <choice name="ZeroCornered"></choice>
             <choice name="UnderMouse"></choice>
         </choices>
-        <default>0</default>
+        <default>4</default>
     </entry>
 
     <entry key="HideUtilityWindowsForInactive" type="Bool">
diff -Naur kwin-5.20.0/kwin.kcfg kwin-5.20.0-p/kwin.kcfg
--- kwin-5.20.0/kwin.kcfg	2020-10-12 17:19:57.000000000 +0200
+++ kwin-5.20.0-p/kwin.kcfg	2020-10-15 02:54:25.170507944 +0200
@@ -127,7 +127,7 @@
                 <choice name="OnMainWindow"/>
                 <choice name="Maximizing"/>
             </choices>
-            <default>Placement::Smart</default>
+            <default>Placement::Centered</default>
         </entry>
         <entry name="AutoRaise" type="Bool">
             <default>false</default>
diff -Naur kwin-5.20.0/rulesettings.kcfg kwin-5.20.0-p/rulesettings.kcfg
--- kwin-5.20.0/rulesettings.kcfg	2020-10-12 17:19:57.000000000 +0200
+++ kwin-5.20.0-p/rulesettings.kcfg	2020-10-15 04:00:30.806583415 +0200
@@ -68,7 +68,7 @@
     <entry name="placement" type="Enum">
       <label>Initial placement</label>
       <choices name="Placement::Policy"/>
-      <default code="true">Placement::Smart</default>
+      <default code="true">Placement::Centered</default>
     </entry>
     <entry name="placementrule" type="Int">
       <label>Initial placement rule type</label>
