diff --git a/src/plugins/screencast/screencaststream.cpp b/src/plugins/screencast/screencaststream.cpp
index 9ae96bc5d4..1577d09ab5 100644
--- a/src/plugins/screencast/screencaststream.cpp
+++ b/src/plugins/screencast/screencaststream.cpp
@@ -458,7 +458,6 @@ void ScreenCastStream::recordFrame(const QRegion &_damagedRegion)
 
     m_pendingDamages = {};
     if (m_pendingBuffer) {
-        qCWarning(KWIN_SCREENCAST) << "Dropping a screencast frame because the compositor is slow";
         return;
     }
 
