diff -up GraphicsMagick-1.1.10/Magick++/lib/Geometry.cpp.orig GraphicsMagick-1.1.10/Magick++/lib/Geometry.cpp
--- GraphicsMagick-1.1.10/Magick++/lib/Geometry.cpp.orig	2008-02-11 13:26:47.000000000 +0100
+++ GraphicsMagick-1.1.10/Magick++/lib/Geometry.cpp	2008-02-11 13:29:08.000000000 +0100
@@ -8,7 +8,7 @@
 #define MAGICK_IMPLEMENTATION
 
 #include "Magick++/Include.h"
-#include <string>
+#include <cstring>
 #include <ctype.h> // for isdigit
 
 using namespace std;
