--- weechat/src/common/weeconfig.c	2006-05-21 17:04:44.000000000 +0200
+++ src/common/weeconfig.c	2006-05-25 11:29:42.000000000 +0200
@@ -67,6 +67,7 @@
 int cfg_look_set_title;
 int cfg_look_startup_logo;
 int cfg_look_startup_version;
+int cfg_look_highlight_beep;
 char *cfg_look_weechat_slogan;
 char *cfg_look_charset_decode_iso;
 char *cfg_look_charset_decode_utf;
@@ -123,6 +124,10 @@
     N_("display WeeChat version at startup"),
     OPTION_TYPE_BOOLEAN, BOOL_FALSE, BOOL_TRUE, BOOL_TRUE,
     NULL, NULL, &cfg_look_startup_version, NULL, config_change_noop },
+  { "look_highlight_beep", N_("console beep on highlight"),
+    N_("console beep on highlight"),
+    OPTION_TYPE_BOOLEAN, BOOL_FALSE, BOOL_TRUE, BOOL_TRUE,
+    NULL, NULL, &cfg_look_highlight_beep, NULL, config_change_noop },
   { "look_weechat_slogan", N_("WeeChat slogan"),
     N_("WeeChat slogan (if empty, slogan is not used)"),
     OPTION_TYPE_STRING, 0, 0, 0,
--- weechat/src/common/weeconfig.h	2006-04-16 11:46:23.000000000 +0200
+++ src/common/weeconfig.h	2006-05-25 11:50:50.000000000 +0200
@@ -90,6 +90,7 @@
 extern int cfg_look_set_title;
 extern int cfg_look_startup_logo;
 extern int cfg_look_startup_version;
+extern int cfg_look_highlight_beep;
 extern char *cfg_look_weechat_slogan;
 extern char *cfg_look_charset_decode_iso;
 extern char *cfg_look_charset_decode_utf;
--- weechat/src/gui/gui-common.c	2006-05-20 11:45:03.000000000 +0200
+++ src/gui/gui-common.c	2006-05-25 11:42:31.000000000 +0200
@@ -159,8 +159,12 @@
     buffer->last_line->length += length;
     if (type & MSG_TYPE_MSG)
         buffer->last_line->line_with_message = 1;
-    if (type & MSG_TYPE_HIGHLIGHT)
+    if (type & MSG_TYPE_HIGHLIGHT) {
         buffer->last_line->line_with_highlight = 1;
+	if (cfg_look_highlight_beep) {
+		printf("\a");
+	}
+    }
     if ((type & MSG_TYPE_TIME) || (type & MSG_TYPE_NICK) || (type & MSG_TYPE_PREFIX))
         buffer->last_line->length_align += length;
     if (type & MSG_TYPE_NOLOG)
