diff -Naur mathgl-2.1.1/udav/prop_dlg.cpp mathgl-2.1.1new/udav/prop_dlg.cpp
--- mathgl-2.1.1/udav/prop_dlg.cpp	2012-11-26 07:58:42.000000000 +0100
+++ mathgl-2.1.1new/udav/prop_dlg.cpp	2012-12-31 14:31:15.939045225 +0100
@@ -115,7 +115,7 @@
 		<< QCoreApplication::applicationDirPath()+"/fonts";
 	filter << "*.vfm";
 #ifndef WIN32
-	paths << "/usr/local/share/mathgl/fonts/" << "/usr/local/mathgl/fonts/";
+	paths << "/usr/share/mathgl/fonts/" << "/usr/mathgl/fonts/";
 #endif
 	QFileInfoList info;
 	for(int i=0;i<paths.count();i++)
diff -Naur mathgl-2.1.1/udav/udav_wnd.cpp mathgl-2.1.1new/udav/udav_wnd.cpp
--- mathgl-2.1.1/udav/udav_wnd.cpp	2012-12-11 12:10:42.000000000 +0100
+++ mathgl-2.1.1new/udav/udav_wnd.cpp	2012-12-31 14:32:22.010074001 +0100
@@ -84,7 +84,7 @@
 #ifdef WIN32
 #define UDAV_DIR ""
 #else
-#define UDAV_DIR "/usr/local/share/udav/"
+#define UDAV_DIR "/usr/share/udav/"
 #endif
 #endif
 //-----------------------------------------------------------------------------
diff -Naur mathgl-2.1.1/udav/udav_wnd.h mathgl-2.1.1new/udav/udav_wnd.h
--- mathgl-2.1.1/udav/udav_wnd.h	2012-12-10 08:45:38.000000000 +0100
+++ mathgl-2.1.1new/udav/udav_wnd.h	2012-12-31 14:32:45.834087989 +0100
@@ -26,7 +26,7 @@
 #ifdef WIN32
 #define MGL_DOC_DIR ""
 #else
-#define MGL_DOC_DIR "/usr/local/share/doc/mathgl/"
+#define MGL_DOC_DIR "/usr/share/doc/mathgl/"
 #endif
 #endif
 //-----------------------------------------------------------------------------
