--- ./pkg/etc/acoc.conf	2006-03-31 18:25:10.000000000 +0300
+++ ./acoc.conf	2006-03-31 18:31:43.000000000 +0300
@@ -224,3 +224,23 @@
 
 [whereis]
 /^(.+): (.+)/					yellow,bold
+
+
+[pacman -Si]
+/^Name\s+:\s([\w.-]+)/                              green
+[pacman -Qi]
+/^Name\s+:\s([\w.-]+)/                              green
+[pacman -Qi$]
+/^([\w.-]+)\s([\w.-]+)/                 green,clear
+[pacman -Ss]
+/^([\w.-]+)\/([\w.-]+)\s+([\w.-]+)/     red,blue,magenta
+[pacman -Qs]
+/^([\w.-]+)\/([\w.-]+)\s+([\w.-]+)/     red,blue,magenta
+[pacman -Sl]
+/^([\w.-]+)\s([\w.-]+)\s([\w.-]+)/              clear,blue,clear
+[pacman -Qo]
+/^([\w.-\/]+)\sis\sowned\sby\s([\w.-]+)\s([\w.-]+)/     clear,blue,clear
+[pacman -Qe$]
+/^([\w.-]+)\s([\w.-]+)/                 blue,clear
+[pacman -Qg$]
+/^([\w.-]+)\s([\w.-]+)/                 blue,bold
