--- archivemail.orig	2007-12-03 12:36:15.000000000 +0100
+++ archivemail	2007-12-03 12:31:11.000000000 +0100
@@ -1122,7 +1122,10 @@
     old_temp_dir = tempfile.tempdir
     try:
         # create a temporary directory for us to work in securely
-        tempfile.tempdir = None
+        if 'TMPDIR' in os.environ.keys():
+            tempfile.tempdir = os.environ['TMPDIR']
+        else:
+            tempfile.tempdir = None
         new_temp_dir = tempfile.mkdtemp('archivemail')
         assert(new_temp_dir)
         _stale.temp_dir = new_temp_dir
