diff -Naur openinbrowser-1.1.orig/install.rdf openinbrowser-1.1/install.rdf
--- openinbrowser-1.1.orig/install.rdf	2007-01-15 18:42:17.000000000 +0100
+++ openinbrowser-1.1/install.rdf	2008-06-19 21:52:51.000000000 +0200
@@ -23,7 +23,7 @@
       <Description>
         <em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
         <em:minVersion>1.5</em:minVersion>
-        <em:maxVersion>2.0.0.*</em:maxVersion>
+        <em:maxVersion>3.0.0.*</em:maxVersion>
       </Description>
     </em:targetApplication>
     
