diff -Naur freetype-2.1.3-old/include/freetype/config/ftoption.h freetype-2.1.3/include/freetype/config/ftoption.h
--- freetype-2.1.3-old/include/freetype/config/ftoption.h	2002-11-06 14:32:53.000000000 -0800
+++ freetype-2.1.3/include/freetype/config/ftoption.h	2003-02-19 17:55:37.000000000 -0800
@@ -378,7 +378,7 @@
   /*   do not #undef this macro here, since the build system might         */
   /*   define for certain configurations                                   */
   /*                                                                       */
-/* #define  TT_CONFIG_OPTION_BYTECODE_INTERPRETER */
+#define  TT_CONFIG_OPTION_BYTECODE_INTERPRETER
 
 
   /*************************************************************************/
