diff -ur a/tools/Linux/xbmc.sh.in b/tools/Linux/xbmc.sh.in
--- a/tools/Linux/xbmc.sh.in	2012-11-01 00:48:53.380341117 -0500
+++ b/tools/Linux/xbmc.sh.in	2012-11-01 00:50:48.231391513 -0500
@@ -64,11 +64,11 @@
   echo -n " Kernel: " >> $FILE
   uname -rvs >> $FILE
   echo -n " Release: " >> $FILE
-  if which lsb_release > /dev/null; then
+  if [ -f /etc/frugalware-release ]; then
     echo >> $FILE
-    lsb_release -a 2> /dev/null | sed -e 's/^/    /' >> $FILE
+    cat /etc/frugalware-release | sed -e 's/^/    /' >> $FILE
   else
-    echo "lsb_release not available" >> $FILE
+    echo "frugalware-release not available" >> $FILE
   fi
   echo "############## END SYSTEM INFO ##############" >> $FILE
   echo >> $FILE
diff -ur a/xbmc/utils/SystemInfo.cpp b/xbmc/utils/SystemInfo.cpp
--- a/xbmc/utils/SystemInfo.cpp	2012-11-01 00:48:53.016340767 -0500
+++ b/xbmc/utils/SystemInfo.cpp	2012-11-01 00:53:15.657459157 -0500
@@ -506,7 +506,7 @@
                                         "/etc/arch-release",
                                         NULL };
 
-  FILE* pipe = popen("unset PYTHONHOME; unset PYTHONPATH; lsb_release -d | cut -f2", "r");
+  FILE* pipe = popen("cat /etc/frugalware-release | cut -f 1 -d ' '", "r");
   
   for (int i = 0; !pipe && release_file[i]; i++)
   {
