diff -ur a/tools/Linux/xbmc.sh.in b/tools/Linux/xbmc.sh.in
--- a/tools/Linux/xbmc.sh.in	2013-01-29 08:20:27.000000000 -0600
+++ b/tools/Linux/xbmc.sh.in	2013-02-12 18:27:06.949534539 -0600
@@ -64,11 +64,11 @@
   echo -n " Kernel: " >> $FILE
   uname -rvs >> $FILE
   echo -n " Release: " >> $FILE
-  if which lsb_release > /dev/null; then
+  if [ -f /etc/frugalware-release ]; then
     echo >> $FILE
-    lsb_release -a 2> /dev/null | sed -e 's/^/    /' >> $FILE
+    cat /etc/frugalware-release | sed -e 's/^/    /' >> $FILE
   else
-    echo "lsb_release not available" >> $FILE
+    echo "frugalware-release not available" >> $FILE
   fi
   echo "############## END SYSTEM INFO ##############" >> $FILE
   echo >> $FILE
diff -ur a/xbmc/utils/SystemInfo.cpp b/xbmc/utils/SystemInfo.cpp
--- a/xbmc/utils/SystemInfo.cpp	2013-01-29 08:20:27.000000000 -0600
+++ b/xbmc/utils/SystemInfo.cpp	2013-02-12 18:28:40.144531074 -0600
@@ -589,7 +589,7 @@
                                         NULL };
   CStdString result("");
 
-  FILE* pipe = popen("unset PYTHONHOME; unset PYTHONPATH; lsb_release -d  2>/dev/null | cut -f2", "r");
+  FILE* pipe = popen("cat /etc/frugalware-release | cut -f 1 -d ' '", "r");
   if (pipe)
   {
     char buffer[256] = {'\0'};
Only in b/xbmc/utils: SystemInfo.cpp.orig
Only in b/xbmc/utils: SystemInfo.cpp.rej
