diff -Naur cdemu-0.7/cdemu.c cdemu-0.7-p/cdemu.c
--- cdemu-0.7/cdemu.c	2004-11-20 07:34:42.000000000 +0100
+++ cdemu-0.7-p/cdemu.c	2006-06-21 01:57:34.000000000 +0200
@@ -509,7 +509,7 @@
 }
 #endif
 
-static int cdemu_ioctl(struct cdrom_device_info *cdi, unsigned int cmd, unsigned long arg)
+int cdemu_ioctl(struct cdrom_device_info *cdi, unsigned int cmd, unsigned long arg)
 {
 	struct cdemu_device *vc;
 	struct cdrom_msf msf;
@@ -639,11 +639,18 @@
 	.get_mcn				= NULL,
 	.reset					= cdemu_reset,
 	.audio_ioctl			= cdemu_audio_ioctl,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,17)
 	.dev_ioctl				= cdemu_ioctl,
 	.capability				= CDC_OPEN_TRAY | CDC_LOCK |
 							  CDC_MULTI_SESSION | CDC_MEDIA_CHANGED |
 							  CDC_PLAY_AUDIO |
-							  CDC_IOCTLS | CDC_DRIVE_STATUS,
+							CDC_IOCTLS | CDC_DRIVE_STATUS,
+#else
+	.capability                             = CDC_OPEN_TRAY | CDC_LOCK |
+                                                          CDC_MULTI_SESSION | CDC_MEDIA_CHANGED |
+                                                          CDC_PLAY_AUDIO |
+                                                    CDC_DRIVE_STATUS,
+#endif
 	.n_minors				= 1,
 /* RENE: 2.6.8 patch */
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,8)
