diff -Nur gnome-screensaver-0.0.21/data/gnome-screensaver.schemas.in gnome-screensaver-0.0.21.new/data/gnome-screensaver.schemas.in
--- gnome-screensaver-0.0.21/data/gnome-screensaver.schemas.in	2005-12-13 13:20:38.000000000 +0100
+++ gnome-screensaver-0.0.21.new/data/gnome-screensaver.schemas.in	2005-12-13 13:21:12.000000000 +0100
@@ -212,7 +212,7 @@
       <applyto>/apps/gnome-screensaver/user_switch_enabled</applyto>
       <owner>gnome-screensaver</owner>
       <type>bool</type>
-      <default>FALSE</default>
+      <default>TRUE</default>
       <locale name="C">
         <short>Allow user switching</short>
         <long>
