Index: data/gnome-screensaver-preferences.desktop.in
===================================================================
--- data/gnome-screensaver-preferences.desktop.in	(revision 1067)
+++ data/gnome-screensaver-preferences.desktop.in	(working copy)
@@ -6,7 +6,7 @@
 Exec=gnome-screensaver-preferences
 Terminal=false
 Type=Application
-Categories=Application;Settings;
+Categories=Application;Settings;DesktopSettings;
 OnlyShowIn=GNOME;
 StartupNotify=true
 X-GNOME-Bugzilla-Bugzilla=GNOME
