diff -urN gnome-screensaver-2.17.7.orig/data/gnome-screensaver.schemas.in gnome-screensaver-2.17.7/data/gnome-screensaver.schemas.in
--- gnome-screensaver-2.17.7.orig/data/gnome-screensaver.schemas.in	2007-02-22 07:27:32.000000000 +0000
+++ gnome-screensaver-2.17.7/data/gnome-screensaver.schemas.in	2007-02-22 07:32:27.000000000 +0000
@@ -36,7 +36,7 @@
       <applyto>/apps/gnome-screensaver/mode</applyto>
       <owner>gnome-screensaver</owner>
       <type>string</type>
-      <default>blank-only</default>
+      <default>single</default>
       <locale name="C">
         <short>Screensaver theme selection mode</short>
         <long>
@@ -55,7 +55,7 @@
       <owner>gnome-screensaver</owner>
       <type>list</type>
       <list_type>string</list_type>
-      <default>[]</default>
+      <default>[screensavers-frugal-floaters]</default>
       <locale name="C">
         <short>Screensaver themes</short>
         <long>
