--- gnome-screensaver-2.24.0old/data/gnome-screensaver.schemas.in	2008-04-30 04:30:11.000000000 +0200
+++ gnome-screensaver-2.24.0/data/gnome-screensaver.schemas.in	2008-10-27 15:29:02.000000000 +0100
@@ -230,7 +230,7 @@
       <applyto>/apps/gnome-screensaver/lock_dialog_theme</applyto>
       <owner>gnome-screensaver</owner>
       <type>string</type>
-      <default>default</default>
+      <default>newwave</default>
       <locale name="C">
         <short>Theme for lock dialog</short>
         <long>
