diff -aur swfdec-0.6.6/swfdec/swfdec_codec_ffmpeg.c fw_swfdec-0.6.6/swfdec/swfdec_codec_ffmpeg.c
--- swfdec-0.6.6/swfdec/swfdec_codec_ffmpeg.c	2008-04-10 16:20:41.000000000 +0200
+++ fw_swfdec-0.6.6/swfdec/swfdec_codec_ffmpeg.c	2008-04-29 00:02:51.000000000 +0200
@@ -21,8 +21,8 @@
 #include "config.h"
 #endif
 #include <string.h>
-#include <avcodec.h>
-#include <swscale.h>
+#include <libavcodec/avcodec.h>
+#include <libswscale/swscale.h>
 
 #include "swfdec_codec_audio.h"
 #include "swfdec_codec_video.h"
