diff -Naur gprename-2.3.orig/bin/gprename gprename-2.3/bin/gprename
--- gprename-2.3.orig/bin/gprename	2007-06-05 15:10:39.000000000 +0200
+++ gprename-2.3/bin/gprename	2007-06-16 13:25:36.000000000 +0200
@@ -30,7 +30,7 @@
 use utf8;                     # Because this file is in UTF8 and because we're using hardcoded accent in Help/About
 use Gtk2::Pango;              # To use text markup in TextView
 
-my $install_dir = '@INSTALLDIR@/share/gprename';
+my $install_dir = '/usr/share/gprename';
 
 # Change the font to courier for the Tree and the SimpleList
 Gtk2::Rc->parse_string(<<__);
@@ -146,7 +146,7 @@
 $window->show();
 
 # Set the window icon
-my $icon = '@INSTALLDIR@/share/pixmaps/gprename/gprename.png';
+my $icon = '/usr/share/pixmaps/gprename/gprename.png';
 my $pixbuf = Gtk2::Gdk::Pixbuf->new_from_file( $icon );
 $window->set_icon( $pixbuf );
 
@@ -1429,7 +1429,7 @@
    $buffer->create_tag( "big", size => 15 * PANGO_SCALE );
    $buffer->create_tag( "italic", style => 'italic' );
    my $iter = $buffer->get_start_iter;
-   my $icon = '@INSTALLDIR@/share/pixmaps/gprename/gprename.png';
+   my $icon = '/usr/share/pixmaps/gprename/gprename.png';
    my $pixbuf = Gtk2::Gdk::Pixbuf->new_from_file( $icon );
 
    $buffer->insert_pixbuf ($iter,  $pixbuf);
diff -Naur gprename-2.3.orig/Makefile gprename-2.3/Makefile
--- gprename-2.3.orig/Makefile	2007-06-05 14:36:38.000000000 +0200
+++ gprename-2.3/Makefile	2007-06-16 13:22:47.000000000 +0200
@@ -4,7 +4,7 @@
 # See the file README and COPYING for the Credits and the License
 
 # Globals:
-INSTALLDIR=/usr/local
+INSTALLDIR=/usr
 
 build:
 	# Create temporary build directory and modify file locations.
