diff -Naur ufoai-2.1.1-source.orig/build/ref_gl.mk ufoai-2.1.1-source/build/ref_gl.mk
--- ufoai-2.1.1-source.orig/build/ref_gl.mk	2007-12-18 09:41:10.000000000 +0100
+++ ufoai-2.1.1-source/build/ref_gl.mk	2007-12-18 09:41:45.000000000 +0100
@@ -66,7 +66,7 @@
 # Say about to build the target
 $(REF_GLX_TARGET) : $(REF_GLX_OBJS) $(REF_GL_OBJS) $(BUILDDIR)/.dirs
 	@echo " * [GLX] ... linking $(LNKFLAGS) ($(REF_GLX_LIBS))"; \
-		$(CC) $(LDFLAGS) $(SHARED_LDFLAGS) -o $@ $(REF_GLX_OBJS) $(REF_GL_OBJS) $(REF_GLX_LIBS) $(LNKFLAGS)
+		$(CC) $(LDFLAGS) $(SHARED_LDFLAGS) -o $@ $(REF_GLX_OBJS) $(REF_GL_OBJS) $(REF_GLX_LIBS) -lXxf86vm $(LNKFLAGS)
 
 #---------------------------------------------------------------------------------------------------------------------
 
