From ad1f3a914bf52b9398f427f923c8d278df453e45 Mon Sep 17 00:00:00 2001
From: Mathieu Duponchelle <mathieu@centricular.com>
Date: Fri, 13 Jul 2018 18:03:27 +0200
Subject: [PATCH 49/72] avcfg: actually remove call to g_object_getv

(Sorry about that)
---
 ext/libav/gstavcfg.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/ext/libav/gstavcfg.c b/ext/libav/gstavcfg.c
index ce9edd6..aee2c34 100644
--- a/ext/libav/gstavcfg.c
+++ b/ext/libav/gstavcfg.c
@@ -572,7 +572,6 @@ gst_ffmpeg_cfg_fill_context (GObject * object, AVCodecContext * context)
     if (!opt)
       continue;
 
-    g_object_getv (object, 1, &pspec->name, &value);
     g_object_get_property (object, pspec->name, &value);
     set_option_value (context, pspec, &value, opt);
   }
-- 
2.17.0

