From c881f831c369fd7e02ea91430948a7e30a1b6e03 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim@centricular.com>
Date: Wed, 18 Jul 2018 15:43:41 +0100
Subject: [PATCH 53/72] Remove obsolete #if 0-ed code

---
 ext/libav/gstav.c | 6 ------
 ext/libav/gstav.h | 4 ----
 2 files changed, 10 deletions(-)

diff --git a/ext/libav/gstav.c b/ext/libav/gstav.c
index 6125662..9afcdf1 100644
--- a/ext/libav/gstav.c
+++ b/ext/libav/gstav.c
@@ -164,12 +164,6 @@ plugin_init (GstPlugin * plugin)
   gst_ffmpegdemux_register (plugin);
   gst_ffmpegmux_register (plugin);
   gst_ffmpegdeinterlace_register (plugin);
-#if 0
-  gst_ffmpegscale_register (plugin);
-#endif
-#if 0
-  gst_ffmpegaudioresample_register (plugin);
-#endif
 
   /* Now we can return the pointer to the newly created Plugin object. */
   return TRUE;
diff --git a/ext/libav/gstav.h b/ext/libav/gstav.h
index 7f11a9e..df15ce8 100644
--- a/ext/libav/gstav.h
+++ b/ext/libav/gstav.h
@@ -44,10 +44,6 @@ extern gboolean gst_ffmpegviddec_register (GstPlugin * plugin);
 extern gboolean gst_ffmpegaudenc_register (GstPlugin * plugin);
 extern gboolean gst_ffmpegvidenc_register (GstPlugin * plugin);
 extern gboolean gst_ffmpegmux_register (GstPlugin * plugin);
-#if 0
-extern gboolean gst_ffmpegscale_register (GstPlugin * plugin);
-extern gboolean gst_ffmpegaudioresample_register (GstPlugin * plugin);
-#endif
 extern gboolean gst_ffmpegdeinterlace_register (GstPlugin * plugin);
 
 int gst_ffmpeg_avcodec_open (AVCodecContext *avctx, AVCodec *codec);
-- 
2.17.0

