--- gst-libav-1.22.0/ext/libav/gstavviddec.c	2023-01-23 20:29:34.000000000 +0100
+++ gst-libav-1.22.0.new/ext/libav/gstavviddec.c	2023-03-02 11:10:22.691324358 +0100
@@ -615,8 +615,12 @@
   if (ffmpegdec->max_threads == 0) {
     /* When thread type is FF_THREAD_FRAME, extra latency is introduced equal
      * to one frame per thread. We thus need to calculate the thread count ourselves */
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(59, 18, 100)	  
     if ((!(oclass->in_plugin->capabilities & AV_CODEC_CAP_AUTO_THREADS)) ||
         (ffmpegdec->context->thread_type & FF_THREAD_FRAME))
+#else
+    if (ffmpegdec->context->thread_type & FF_THREAD_FRAME)
+#endif
       ffmpegdec->context->thread_count =
           MIN (gst_ffmpeg_auto_max_threads (), 16);
     else
