--- gst-libav-1.8.3/ext/libav/gstavvidenc.c	2016-08-16 15:44:14.000000000 +0200
+++ gst-libav-1.8.3.new/ext/libav/gstavvidenc.c	2016-08-24 15:04:34.000000000 +0200
@@ -966,8 +966,7 @@
         || in_plugin->id == AV_CODEC_ID_Y41P
         || in_plugin->id == AV_CODEC_ID_012V
         || in_plugin->id == AV_CODEC_ID_YUV4
-        || in_plugin->id == AV_CODEC_ID_ZLIB
-        || in_plugin->id == AV_CODEC_ID_WRAPPED_AVFRAME) {
+        || in_plugin->id == AV_CODEC_ID_ZLIB) {
       goto next;
     }
 
--- gst-libav-1.8.3/ext/libav/gstavviddec.c	2016-08-16 15:44:27.000000000 +0200
+++ gst-libav-1.8.3.new/ext/libav/gstavviddec.c	2016-08-24 15:07:01.000000000 +0200
@@ -2044,8 +2044,7 @@
         || in_plugin->id == AV_CODEC_ID_Y41P
         || in_plugin->id == AV_CODEC_ID_012V
         || in_plugin->id == AV_CODEC_ID_YUV4
-        || in_plugin->id == AV_CODEC_ID_ZLIB
-        || in_plugin->id == AV_CODEC_ID_WRAPPED_AVFRAME) {
+        || in_plugin->id == AV_CODEC_ID_ZLIB) {
       goto next;
     }
 
