diff -Naur wireshark-2.0.2/wsutil/ws_version_info.c wireshark-2.0.2-p/wsutil/ws_version_info.c
--- wireshark-2.0.2/wsutil/ws_version_info.c	2016-02-26 20:58:30.000000000 +0100
+++ wireshark-2.0.2-p/wsutil/ws_version_info.c	2016-04-03 00:19:12.140445491 +0200
@@ -351,11 +351,7 @@
 const char *
 get_ws_vcs_version_info(void)
 {
-#ifdef VCSVERSION
-	return VERSION " (" VCSVERSION " from " VCSBRANCH ")";
-#else
-	return VERSION;
-#endif
+	return VERSION;
 }
 
 /*
