--- gremind-0.1.1/src/reminder.cc	2007-06-26 20:42:56.000000000 +0200
+++ gremind-0.1.1/src/reminder.cc	2008-06-21 15:02:04.000000000 +0200
@@ -25,6 +25,11 @@
 #include <cstring> // for std::memcpy()
 #include <fstream> // required by Reminder::save() and Reminder::load()
 
+#include <iostream>
+#include <sstream>
+#include <cstdlib> 
+
+
 using namespace Grem;
 
 Reminder::Reminder(std::tm* when, const Glib::ustring& msg, Mode mode)

