--- guvcview-src-0.9.2old/src/v4l2uvc.h	2008-08-13 18:25:12.000000000 +0200
+++ guvcview-src-0.9.2/src/v4l2uvc.h	2008-08-13 18:24:59.000000000 +0200
@@ -99,12 +99,8 @@
 
 #define V4L2_CID_PRIVATE_LAST			(V4L2_CID_WHITE_BALANCE_TEMPERATURE+1)
 
-enum  v4l2_exposure_auto_type {
-	V4L2_EXPOSURE_MANUAL = 1,
-	V4L2_EXPOSURE_AUTO = 2,
-	V4L2_EXPOSURE_SHUTTER_PRIORITY = 4,
-	V4L2_EXPOSURE_APERTURE_PRIORITY = 8
-};
+
+
 
 static const int exp_vals[]={
 						V4L2_EXPOSURE_MANUAL,
@@ -114,13 +110,6 @@
 				 };
 
 
-enum v4l2_power_line_frequency {
-	V4L2_CID_POWER_LINE_FREQUENCY_DISABLED	= 0,
-	V4L2_CID_POWER_LINE_FREQUENCY_50HZ	= 1,
-	V4L2_CID_POWER_LINE_FREQUENCY_60HZ	= 2,
-};
-
-
 #define UVC_DYN_CONTROLS
 /*
  * Dynamic controls
