diff -aur qosmic-1.4.8/qosmic.pro qosmic-1.4.8-2/qosmic.pro
--- qosmic-1.4.8/qosmic.pro	2009-12-15 16:10:36.000000000 -0600
+++ qosmic-1.4.8-2/qosmic.pro	2010-05-06 10:03:29.000000000 -0500
@@ -90,10 +90,10 @@
 }
 
 link_pkgconfig {
-	! system(pkg-config --atleast-version 2.7.18 flam3) {
+	system(pkg-config --atleast-version 2.7.18 flam3) {
 		error("Qosmic $$VERSION requires at least version 2.7.18 of flam3 to build.")
 	}
-	! system(pkg-config --max-version 2.7.9999 flam3) {
+	system(pkg-config --max-version 2.7.9999 flam3) {
 		error("Qosmic $$VERSION is not compatible with flam3 versions >=2.8")
 	}
 }
