diff -Naur ivman-0.6.12.orig/src/manager.c ivman-0.6.12/src/manager.c
--- ivman-0.6.12.orig/src/manager.c	2006-05-11 03:21:52.032824048 +0200
+++ ivman-0.6.12/src/manager.c	2006-05-11 03:23:23.049987336 +0200
@@ -1016,19 +1016,24 @@
                                               "volume.mount_point",
                                               &dbus_error);
 
-        while (( !mount_point || (strlen(mount_point) < 2) )
+        while (( !mount_point || (strlen(mount_point) < 1) )
                && (i < 5))
         {
             sleep(1);
             mount_point = (char *)
                 libhal_device_get_property_string(hal_ctx, udi,
-                                                  " volume.mount_point ",
+                                                  "volume.mount_point ",
                                                   &dbus_error);
+	    if( dbus_error_is_set(&dbus_error) )
+	     {
+	        ivm_check_dbus_error(&dbus_error);
+		 break;
+	     }
             i++;
         }
 
 
-        if ( !mount_point || strlen(mount_point) < 2)
+        if ( !mount_point || strlen(mount_point) < 1)
         {
             DEBUG(_("Couldn't get mount point of device %s after "
                     "5 seconds\n"), device);
