diff -Naur kdepim-runtime-4.4.0/kcm/CMakeLists.txt kdepim-runtime-4.4.0-p/kcm/CMakeLists.txt
--- kdepim-runtime-4.4.0/kcm/CMakeLists.txt	2009-12-14 11:54:42.000000000 +0100
+++ kdepim-runtime-4.4.0-p/kcm/CMakeLists.txt	2010-02-11 21:02:35.000000000 +0100
@@ -10,6 +10,7 @@
 kde4_add_ui_files(akonadi_server_kcm_srcs serverconfigmodule.ui
                                           servermysqlstorage.ui
                                           serverpsqlstorage.ui
+					  serversqlitestorage.ui
                                           serverstoragedriver.ui)
 kde4_add_plugin(kcm_akonadi_server ${akonadi_server_kcm_srcs})
 target_link_libraries(kcm_akonadi_server  ${KDE4_KIO_LIBS} ${KDEPIMLIBS_AKONADI_LIBS} ${AKONADI_COMMON_LIBRARIES})
diff -Naur kdepim-runtime-4.4.0/kcm/kcm_akonadi_resources.desktop kdepim-runtime-4.4.0-p/kcm/kcm_akonadi_resources.desktop
--- kdepim-runtime-4.4.0/kcm/kcm_akonadi_resources.desktop	2010-01-20 00:22:26.000000000 +0100
+++ kdepim-runtime-4.4.0-p/kcm/kcm_akonadi_resources.desktop	2010-02-11 21:03:43.000000000 +0100
@@ -1,6 +1,6 @@
 [Desktop Entry]
 Exec=kcmshell4 kcm_akonadi_resources
-Icon=network
+Icon=server-database
 Type=Service
 ServiceTypes=KCModule
 
diff -Naur kdepim-runtime-4.4.0/kcm/serverconfigmodule.cpp kdepim-runtime-4.4.0-p/kcm/serverconfigmodule.cpp
--- kdepim-runtime-4.4.0/kcm/serverconfigmodule.cpp	2009-12-14 11:54:42.000000000 +0100
+++ kdepim-runtime-4.4.0-p/kcm/serverconfigmodule.cpp	2010-02-11 21:16:43.000000000 +0100
@@ -57,7 +57,7 @@
   // supported drivers
   ui_driver.driverBox->addItem( "Mysql", QVariant( "QMYSQL" ) );
   ui_driver.driverBox->addItem( "PostgreSQL", QVariant( "QPSQL" ) );
-
+  ui_driver.driverBox->addItem( "SQlite", QVariant( "QSQLITE" ) );
   setButtons( KCModule::Default | KCModule::Apply );
 
   // MySQL
@@ -71,6 +71,11 @@
   ui_psql.setupUi( m_psqlWidget );
   m_stackWidget->setCurrentWidget( m_mysqlWidget );
 
+  m_sqliteWidget = new QWidget( this );
+  m_stackWidget->addWidget( m_sqliteWidget );
+  ui_sqlite.setupUi( m_sqliteWidget );
+  m_stackWidget->setCurrentWidget( m_mysqlWidget );
+
   QWidget *widget = new QWidget( this );
   layout->addWidget( widget );
   ui.setupUi( widget );
@@ -89,6 +94,8 @@
   connect( ui_psql.password, SIGNAL(textChanged(QString)), SLOT(changed()) );
   connect( ui_psql.port, SIGNAL(textChanged(QString)), SLOT(changed()) );
 
+  connect( ui_sqlite.name, SIGNAL(textChanged(QString)), SLOT(changed()) );
+
   connect( ui.startStopButton, SIGNAL(clicked()), SLOT(startStopClicked()) );
   connect( ui.restartButton, SIGNAL(clicked()), SLOT(restartClicked()) );
   connect( ui.selfTestButton, SIGNAL(clicked()), SLOT(selfTestClicked()) );
@@ -123,9 +130,14 @@
   ui_psql.port->setText( settings.value( "Port", "5432" ).toString() );
   settings.endGroup();
 
+  // sqlite group
+  settings.beginGroup( "QSQLITE" );
+  ui_sqlite.name->setText( settings.value( "Name", "" ).toString() );
+  settings.endGroup();
+
   // selected driver
   settings.beginGroup( "GENERAL" );
-  ui_driver.driverBox->setCurrentIndex( ui_driver.driverBox->findData( settings.value ( "Driver", "QMYSQL" ) ) );
+  ui_driver.driverBox->setCurrentIndex( ui_driver.driverBox->findData( settings.value ( "Driver", "QSQLITE" ) ) );
   driverChanged( ui_driver.driverBox->currentIndex() );
   settings.endGroup();
 
@@ -156,6 +168,11 @@
   settings.setValue( "Port", ui_psql.port->text() );
   settings.endGroup();
 
+  // sqlite group
+  settings.beginGroup( "QSQLITE" );
+  settings.setValue( "Name", ui_sqlite.name->text() );
+  settings.endGroup();
+
   // selected driver
   settings.beginGroup( "GENERAL" );
   settings.setValue( "Driver", ui_driver.driverBox->itemData( ui_driver.driverBox->currentIndex() ).toString() );
@@ -167,11 +184,11 @@
 
 void ServerConfigModule::defaults()
 {
-  ui_mysql.startServer->setChecked( true );
+  //ui_mysql.startServer->setChecked( true );
   // TODO: detect default server path
-  ui_mysql.name->setText( "akonadi" );
+  //ui_mysql.name->setText( "akonadi" );
 
-  ui_driver.driverBox->setCurrentIndex( ui_driver.driverBox->findData ("QMYSQL") );
+  ui_driver.driverBox->setCurrentIndex( ui_driver.driverBox->findData ("QSQLITE") );
 }
 
 void ServerConfigModule::updateStatus()
@@ -207,11 +224,12 @@
 
 void ServerConfigModule::driverChanged( int index )
 {
-  if( ui_driver.driverBox->itemData( index ).toString() == "QMYSQL" ) {
-    m_stackWidget->setCurrentWidget( m_mysqlWidget );
-  }
-  else {
-    m_stackWidget->setCurrentWidget( m_psqlWidget );
+	if( ui_driver.driverBox->itemData( index ).toString() == "QSQLITE" ) {
+		m_stackWidget->setCurrentWidget( m_sqliteWidget );
+	} else if ( ui_driver.driverBox->itemData( index ).toString() == "QMYSQL" ) {
+    		m_stackWidget->setCurrentWidget( m_mysqlWidget );
+  	} else {
+    		m_stackWidget->setCurrentWidget( m_psqlWidget );
   }
 }
 
diff -Naur kdepim-runtime-4.4.0/kcm/serverconfigmodule.h kdepim-runtime-4.4.0-p/kcm/serverconfigmodule.h
--- kdepim-runtime-4.4.0/kcm/serverconfigmodule.h	2009-12-14 11:54:42.000000000 +0100
+++ kdepim-runtime-4.4.0-p/kcm/serverconfigmodule.h	2010-02-11 21:18:38.000000000 +0100
@@ -23,6 +23,7 @@
 #include "ui_serverconfigmodule.h"
 #include "ui_servermysqlstorage.h"
 #include "ui_serverpsqlstorage.h"
+#include "ui_serversqlitestorage.h"
 #include "ui_serverstoragedriver.h"
 
 #include <KCModule>
@@ -52,11 +53,12 @@
     Ui::StorageDriver ui_driver;
     Ui::MySQLStoragePage ui_mysql;
     Ui::PSQLStoragePage  ui_psql;
+    Ui::SQLITEStoragePage  ui_sqlite;
 
     QStackedWidget *m_stackWidget;
     QWidget *m_mysqlWidget;
     QWidget *m_psqlWidget;
-
+    QWidget *m_sqliteWidget;
     QComboBox *m_driverBox;
 };
 
diff -Naur kdepim-runtime-4.4.0/kcm/serversqlitestorage.ui kdepim-runtime-4.4.0-p/kcm/serversqlitestorage.ui
--- kdepim-runtime-4.4.0/kcm/serversqlitestorage.ui	1970-01-01 01:00:00.000000000 +0100
+++ kdepim-runtime-4.4.0-p/kcm/serversqlitestorage.ui	2010-02-11 21:23:18.000000000 +0100
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ui version="4.0">
+ <class>SQLITEStoragePage</class>
+ <widget class="QWidget" name="SQLITEStoragePage">
+  <property name="geometry">
+   <rect>
+    <x>0</x>
+    <y>0</y>
+    <width>467</width>
+    <height>223</height>
+   </rect>
+  </property>
+  <layout class="QGridLayout" name="gridLayout_2">
+   <item row="0" column="0">
+    <widget class="QGroupBox" name="groupBox">
+     <property name="enabled">
+      <bool>true</bool>
+     </property>
+     <property name="title">
+      <string>External SQlite Server </string>
+     </property>
+     <layout class="QHBoxLayout" name="horizontalLayout">
+      <item>
+       <widget class="QLabel" name="label_6">
+        <property name="text">
+         <string>Database name:</string>
+        </property>
+        <property name="buddy">
+         <cstring>name</cstring>
+        </property>
+       </widget>
+      </item>
+      <item>
+       <widget class="KLineEdit" name="name">
+        <property name="enabled">
+         <bool>false</bool>
+        </property>
+        <property name="whatsThis">
+         <string>&lt;p&gt;&lt;b&gt;Database Name&lt;/b&gt;&lt;/p&gt;&lt;br&gt;Enter the name of your Akonadi database on your external SQlite server.</string>
+        </property>
+        <property name="autoFillBackground">
+         <bool>false</bool>
+        </property>
+       </widget>
+      </item>
+     </layout>
+    </widget>
+   </item>
+  </layout>
+ </widget>
+ <customwidgets>
+  <customwidget>
+   <class>KLineEdit</class>
+   <extends>QLineEdit</extends>
+   <header>klineedit.h</header>
+  </customwidget>
+ </customwidgets>
+ <resources/>
+ <connections/>
+</ui>
