diff -aur gst-plugins-bad-0.10.6/ext/mpeg2enc/gstmpeg2encpicturereader.cc fw_gst-plugins-bad-0.10.6/ext/mpeg2enc/gstmpeg2encpicturereader.cc
--- gst-plugins-bad-0.10.6/ext/mpeg2enc/gstmpeg2encpicturereader.cc	2008-02-04 17:22:11.000000000 +0100
+++ fw_gst-plugins-bad-0.10.6/ext/mpeg2enc/gstmpeg2encpicturereader.cc	2008-03-19 15:06:07.000000000 +0100
@@ -24,6 +24,7 @@
 #endif
 
 #include <encoderparams.hh>
+#include <string.h>
 
 #ifdef GST_MJPEGTOOLS_19x
 #include <imageplanes.hh>
