--- gst-plugins-bad-0.10.4/configure.ac	2006-12-21 16:10:47.000000000 +0100
+++ gst-plugins-bad-0.10.4-p/configure.ac	2007-06-12 14:03:37.000000000 +0200
@@ -339,9 +339,9 @@
 
 dnl *** DTS ***
 translit(dnm, m, l) AM_CONDITIONAL(USE_DTS, true)
-GST_CHECK_FEATURE(DTS, [dts library], dtsdec, [
-  GST_CHECK_LIBHEADER(DTS, dts_pic, dts_init, -lm, dts.h,
-    DTS_LIBS="-ldts_pic -lm")
+GST_CHECK_FEATURE(DTS, [dca library], dcadec, [
+  GST_CHECK_LIBHEADER(DTS, dca, dca_init, -lm, dca.h,
+    DTS_LIBS="-ldca -lm")
   AC_SUBST(DTS_LIBS)
 ])
 
