diff -Naur gst-plugins-bad-0.10.5/configure.ac gst-plugins-bad-0.10.5-p/configure.ac
--- gst-plugins-bad-0.10.5/configure.ac	2007-06-19 10:43:43.000000000 +0200
+++ gst-plugins-bad-0.10.5-p/configure.ac	2007-06-19 23:48:24.000000000 +0200
@@ -387,9 +387,9 @@
 
 dnl *** DTS ***
 translit(dnm, m, l) AM_CONDITIONAL(USE_DTS, true)
-AG_GST_CHECK_FEATURE(DTS, [dts library], dtsdec, [
-  AG_GST_CHECK_LIBHEADER(DTS, dts_pic, dts_init, -lm, dts.h,
-    DTS_LIBS="-ldts_pic -lm")
+AG_GST_CHECK_FEATURE(DTS, [dca library], dcadec, [
+  AG_GST_CHECK_LIBHEADER(DTS, dca, dca_init, -lm, dca.h,
+    DTS_LIBS="-ldca -lm")
   AC_SUBST(DTS_LIBS)
 ])
 
