From http://webcvs.freedesktop.org/gstreamer/gst-plugins-bad/ext/neon/gstneonhttpsrc.c?r1=1.13&r2=1.14&view=patch.
--- gst-plugins-bad-0.10.3.orig/ext/neon/gstneonhttpsrc.c	2006/04/29 15:36:15	1.13
+++ gst-plugins-bad-0.10.3/ext/neon/gstneonhttpsrc.c	2006/06/19 14:07:24	1.14
@@ -20,6 +20,10 @@
 #include <string.h>
 #include <unistd.h>
 
+#ifndef NE_FREE
+#define NEON_026_OR_LATER  1
+#endif
+
 #define HTTP_DEFAULT_HOST        "localhost"
 #define HTTP_DEFAULT_PORT        80
 #define HTTPS_DEFAULT_PORT       443
@@ -626,10 +630,13 @@
   if (parsed->host && !parsed->port) {
     goto clear;
   }
-
-  if (!parsed->path || parsed->authinfo) {
+#ifdef NEON_026_OR_LATER
+  if (!parsed->path || parsed->userinfo)
     goto clear;
-  }
+#else
+  if (!parsed->path || parsed->authinfo)
+    goto clear;
+#endif
 
   return TRUE;
 
