diff -Naur bootsplash-3.3.orig/Utilities/Makefile bootsplash-3.3/Utilities/Makefile
--- bootsplash-3.3.orig/Utilities/Makefile	2004-09-23 18:28:38.000000000 +0200
+++ bootsplash-3.3/Utilities/Makefile	2006-01-12 12:52:09.000000000 +0100
@@ -25,7 +25,7 @@
 LIBSS_MNG    = $(LIBDIR)/libmng-mini.a $(LIBDIR)/libz.a -lm
 LIBSD_MNG    = -lmng -lz -lm
 
-CFLAGS   = -Os -Wall -I/usr/include/freetype2
+CFLAGS   += -I/usr/include/freetype2
 
 LINKS    = $(CC) $(CFLAGS) $(LDFLAGS) -static -o $@
 LINKD    = $(CC) $(CFLAGS) $(LDFLAGS) -o $@
