http://bugzilla.gnome.org/show_bug.cgi?id=116814

diff -Naur gnome-session-2.8.1.orig/gnome-session/manager.c gnome-session-2.8.1/gnome-session/manager.c
--- gnome-session-2.8.1.orig/gnome-session/manager.c	2004-10-01 12:13:23.000000000 +0200
+++ gnome-session-2.8.1/gnome-session/manager.c	2005-01-20 20:03:24.000000000 +0100
@@ -1362,8 +1362,6 @@
       if (pending_list)
 	{
 	  Client* client = (Client*)pending_list->data;
-	  if (client->match_rule != MATCH_PROP)
-	    goto out;
 	}
 
       if (start_list)
