diff -PruN gnome-session-2.8.1.orig/gnome-session/manager.c gnome-session-2.8.1/gnome-session/manager.c
--- gnome-session-2.8.1.orig/gnome-session/manager.c	2004-10-01 12:13:23.000000000 +0200
+++ gnome-session-2.8.1/gnome-session/manager.c	2004-11-23 15:20:28.000000000 +0100
@@ -1381,6 +1381,7 @@
 		runlevel = client->priority;
 	    }
 	  if (pending_list)
+	    splash_hide ();
 	    goto out;
 	}
       save_state = MANAGER_IDLE;
diff -PruN gnome-session-2.8.1.orig/gnome-session/splash-widget.c gnome-session-2.8.1/gnome-session/splash-widget.c
--- gnome-session-2.8.1.orig/gnome-session/splash-widget.c	2004-09-24 18:35:38.000000000 +0200
+++ gnome-session-2.8.1/gnome-session/splash-widget.c	2004-11-23 15:19:57.000000000 +0100
@@ -605,3 +605,10 @@
 	}
 }
 
+void
+splash_hide (void)
+{
+	if (global_splash) {
+	gtk_widget_hide (GTK_WIDGET (global_splash));
+	}
+}
diff -PruN gnome-session-2.8.1.orig/gnome-session/splash-widget.h gnome-session-2.8.1/gnome-session/splash-widget.h
--- gnome-session-2.8.1.orig/gnome-session/splash-widget.h	2003-02-13 23:29:57.000000000 +0100
+++ gnome-session-2.8.1/gnome-session/splash-widget.h	2004-11-23 15:18:45.000000000 +0100
@@ -79,5 +79,6 @@
 void splash_start  (void);
 void splash_update (const gchar *text);
 void splash_stop   (void);
+void splash_hide   (void);
 
 #endif /* SPLASH_WIDGET_H */
