diff -urN gnome-session-2.17.91.orig/gnome-session/gnome-wm gnome-session-2.17.91/gnome-session/gnome-wm
--- gnome-session-2.17.91.orig/gnome-session/gnome-wm	2007-02-19 19:08:25.000000000 +0000
+++ gnome-session-2.17.91/gnome-session/gnome-wm	2007-02-19 19:09:25.000000000 +0000
@@ -1,7 +1,8 @@
 #!/bin/sh
 
 # The user can specify his prefered WM by setting the WINDOW_MANAGER
-# environment variable.
+# environment variable or setting the
+# /desktop/gnome/applications/window_manager/default gconf key.
 #
 # If this is not set, we search a list of known windowmanagers and use
 # the first one that is found in the users's PATH
@@ -37,6 +38,11 @@
   esac
 done
 
+# Get previously set window manager in gconf
+if [ ! "$DEFWM" ]; then
+  DEFWM=`gconftool-2 -g /desktop/gnome/applications/window_manager/default 2>/dev/null`
+fi
+
 # WINDOW_MANAGER overrides all
 
 if [ -z "$WINDOW_MANAGER" ] ; then
@@ -118,6 +124,9 @@
     ;;
 esac
 
+# Store the selected WM with gconf
+gconftool-2 -t string -s /desktop/gnome/applications/window_manager/current "$WINDOW_MANAGER"
+
 exec $WINDOW_MANAGER $OPT1 $OPT2 $OPT3
 
 echo "ERROR: No window manager could run!"
