diff -Naur gnome-session-2.20.0.orig/gnome-session/gnome-wm gnome-session-2.20.0/gnome-session/gnome-wm
--- gnome-session-2.20.0.orig/gnome-session/gnome-wm	2007-09-24 19:55:59.000000000 +0200
+++ gnome-session-2.20.0/gnome-session/gnome-wm	2007-09-24 19:59:19.000000000 +0200
@@ -37,6 +37,11 @@
   esac
 done
 
+# Get previously set window manager in gconf
+if [ -z "$DEFWM" -o "x$DEFWM" = "xgnome-wm" ]; then
+  DEFWM=`gconftool-2 -g /desktop/gnome/applications/window_manager/default 2>/dev/null`
+fi
+
 # WINDOW_MANAGER overrides all
 
 if [ -z "$WINDOW_MANAGER" ] ; then
@@ -128,6 +133,9 @@
     ;;
 esac
 
+# Store the selected WM with gconf
+gconftool-2 -t string -s /desktop/gnome/applications/window_manager/current "$WINDOW_MANAGER"
+
 exec $WINDOW_MANAGER $OPT1 $OPT2 $OPT3 $OPT4
 
 echo "ERROR: No window manager could run!"
