diff -ur gnome-session-2.16.0/gnome-session/logout.c gnome-session-2.16.0.patched/gnome-session/logout.c
--- gnome-session-2.16.0/gnome-session/logout.c	2006-09-29 18:30:17 +0100
+++ gnome-session-2.16.0.patched/gnome-session/logout.c	2006-09-29 18:32:09 +0100
@@ -796,8 +796,16 @@
    */
   if (!a11y_enabled && !getenv("LTSP_CLIENT"))
     {
-      XGrabServer (GDK_DISPLAY ());
-      gsm_foreach_screen (fadeout_screen);
+      Atom cmSnAtom;
+      Window currentCmSnOwner;
+      
+      cmSnAtom = XInternAtom (GDK_DISPLAY (), "_NET_WM_CM_S0", 0);
+      currentCmSnOwner = XGetSelectionOwner (GDK_DISPLAY (), cmSnAtom);
+
+      if (currentCmSnOwner == None) {
+	XGrabServer (GDK_DISPLAY ());
+	gsm_foreach_screen (fadeout_screen);
+      }
     }
 
   gtk_widget_show_all (dialog->dialog); 
@@ -823,8 +831,17 @@
 
   if (!a11y_enabled && !getenv("LTSP_CLIENT"))
     {
+      Atom cmSnAtom;
+      Window currentCmSnOwner;
+      
       hide_fadeout_windows ();
-      XUngrabServer (GDK_DISPLAY ());
+
+      cmSnAtom = XInternAtom (GDK_DISPLAY (), "_NET_WM_CM_S0", 0);
+      currentCmSnOwner = XGetSelectionOwner (GDK_DISPLAY (), cmSnAtom);
+
+      if (currentCmSnOwner == None) {
+        XUngrabServer (GDK_DISPLAY ());
+      }
 
       gdk_pointer_ungrab (GDK_CURRENT_TIME);
       gdk_keyboard_ungrab (GDK_CURRENT_TIME);
