--- gnome-session/tools/gnome-session-check-accelerated.c.old	2011-08-22 21:00:33.324631102 +0200
+++ gnome-session/tools/gnome-session-check-accelerated.c	2011-08-22 21:55:52.018141725 +0200
@@ -122,13 +122,14 @@
 
  read:
                 gdk_x11_display_error_trap_push (display);
-                XGetWindowProperty (GDK_DISPLAY_XDISPLAY (display), rootwin,
+                int i = XGetWindowProperty (GDK_DISPLAY_XDISPLAY (display), rootwin,
                                     is_accelerated_atom,
                                     0, G_MAXLONG, False, XA_CARDINAL, &type, &format, &nitems,
                                     &bytes_after, &data);
+		
                 gdk_x11_display_error_trap_pop_ignored (display);
 
-                if (type == XA_CARDINAL) {
+                if ((type == XA_CARDINAL) && (i>0)) {
                         glong *is_accelerated_ptr = (glong*) data;
 
                         if (*is_accelerated_ptr == ACCEL_CHECK_RUNNING) {
