--- gnome-session-2.24.0old/data/gnome-session.schemas.in	2008-09-22 23:21:08.000000000 +0200
+++ gnome-session-2.24.0/data/gnome-session.schemas.in	2008-10-01 16:39:35.000000000 +0200
@@ -7,7 +7,7 @@
          <applyto>/apps/gnome-session/options/show_splash_screen</applyto>
          <owner>gnome</owner>
          <type>bool</type>
-         <default>false</default>
+         <default>true</default>
          <locale name="C">
             <short>Show the splash screen</short>
             <long>Show the splash screen when the session starts up</long>
