diff -urN gnome-session-2.17.3.orig/gnome-session/gnome-wm gnome-session-2.17.3/gnome-session/gnome-wm
--- gnome-session-2.17.3.orig/gnome-session/gnome-wm	2006-12-03 18:27:22.000000000 +0000
+++ gnome-session-2.17.3/gnome-session/gnome-wm	2006-12-03 18:28:53.000000000 +0000
@@ -113,11 +113,12 @@
 
 case `basename $WINDOW_MANAGER` in
   compiz)
-    gnome-window-decorator &
-    OPT3=gconf
+    gtk-window-decorator &
+    OPT3=--use-cow
+    OPT4=gconf
     ;;
 esac
 
-exec $WINDOW_MANAGER $OPT1 $OPT2 $OPT3
+exec $WINDOW_MANAGER $OPT1 $OPT2 $OPT3 $OPT4
 
 echo "ERROR: No window manager could run!"
