diff -Nur gnome-session-2.13.90/gnome-session/main.c gnome-session-2.13.90.new/gnome-session/main.c
--- gnome-session-2.13.90/gnome-session/main.c	2006-01-10 01:48:20.000000000 +0100
+++ gnome-session-2.13.90.new/gnome-session/main.c	2006-01-28 19:12:13.000000000 +0100
@@ -344,6 +344,7 @@
     gsm_set_verbose (TRUE);
 
   /* Help eradicate the critical warnings in unstable releases of GNOME */
+  /*
   versions = g_strsplit (VERSION, ".", 3);
   if (versions && versions [0] && versions [1])
     {
@@ -356,7 +357,7 @@
 	}
     }
   g_strfreev (versions);
-      
+  */  
   set_lang();
   set_gtk1_theme_rcfile ();
 
