--- bluez-gnome-0.25/applet/obex.c	2008-03-14 04:35:47.000000000 +0530
+++ bluez-gnome-0.25.new/applet/obex.c	2008-05-08 20:35:23.000000000 +0530
@@ -139,11 +139,17 @@
 
 	dir = g_get_user_special_dir(G_USER_DIRECTORY_DOWNLOAD);
 
+	/* If the download directory is not set, use the user's Desktop dir 
+	 * so that the user can still receive files */
+	if (dir == NULL)
+		dir = g_strdup_printf ("%s/Desktop", g_get_home_dir());
+
 	dbus_g_proxy_begin_call(opp_server, "Start",
 				start_opp_notify, NULL, NULL,
 				G_TYPE_STRING, dir,
 				G_TYPE_BOOLEAN, TRUE,
 				G_TYPE_BOOLEAN, TRUE, G_TYPE_INVALID);
+	g_free (dir);
 }
 
 static void create_opp_server(DBusGProxy *manager)
@@ -206,11 +212,17 @@
 
 	dir = g_get_user_special_dir(G_USER_DIRECTORY_PUBLIC_SHARE);
 
+	/* If the share directory is not set, use the user's home directory
+	 * so that this service can work */
+	if (dir == NULL)
+		dir = g_strdup_printf (g_get_home_dir());
+
 	dbus_g_proxy_begin_call(ftp_server, "Start",
 				start_ftp_notify, NULL, NULL,
 				G_TYPE_STRING, dir,
 				G_TYPE_BOOLEAN, FALSE,
 				G_TYPE_BOOLEAN, TRUE, G_TYPE_INVALID);
+	g_free (dir);
 }
 
 static void create_ftp_server(DBusGProxy *manager)
