diff -urN bluez-gnome-0.7.orig/properties/bluetooth-properties.desktop.in bluez-gnome-0.7/properties/bluetooth-properties.desktop.in
--- bluez-gnome-0.7.orig/properties/bluetooth-properties.desktop.in	2007-05-26 16:05:34.000000000 +0100
+++ bluez-gnome-0.7/properties/bluetooth-properties.desktop.in	2007-05-26 16:05:47.000000000 +0100
@@ -6,5 +6,5 @@
 Exec=bluetooth-properties
 Terminal=false
 Type=Application
-Categories=Settings;
+Categories=GNOME;GTK;Settings;HardwareSettings;
 OnlyShowIn=GNOME;
