--- src/core/bell.c.orig	2013-02-02 14:49:34.022763810 +0100
+++ src/core/bell.c	2013-02-02 14:50:05.829763810 +0100
@@ -52,7 +52,7 @@
 #include "bell.h"
 #include "screen-private.h"
 #include "prefs.h"
-#include <canberra-gtk.h>
+#include <canberra-gtk2.h>
 
 /**
  * Flashes one entire screen.  This is done by making a window the size of the
--- src/core/workspace.c.orig	2013-02-02 14:49:43.208763810 +0100
+++ src/core/workspace.c	2013-02-02 14:50:23.184763810 +0100
@@ -29,7 +29,7 @@
 #include "prefs.h"
 #include <X11/Xatom.h>
 #include <string.h>
-#include <canberra-gtk.h>
+#include <canberra-gtk2.h>
 
 void meta_workspace_queue_calc_showing   (MetaWorkspace *workspace);
 static void set_active_space_hint        (MetaScreen *screen);
