diff -Naur linux-atm-2.5.2/src/extra/Makefile.in linux-atm-2.5.2.new/src/extra/Makefile.in
--- linux-atm-2.5.2/src/extra/Makefile.in	2010-12-28 17:06:10.000000000 +0100
+++ linux-atm-2.5.2.new/src/extra/Makefile.in	2024-01-16 10:25:34.697555214 +0100
@@ -379,10 +379,10 @@
 
 
 install-exec-hook:
-	$(MKDIR_P) $(DESTDIR)/lib/firmware
-	$(INSTALL_DATA) $(srcdir)/pca200e.bin $(DESTDIR)/lib/firmware
-	$(INSTALL_DATA) $(srcdir)/pca200e_ecd.bin2 $(DESTDIR)/lib/firmware
-	$(INSTALL_DATA) $(srcdir)/sba200e_ecd.bin2 $(DESTDIR)/lib/firmware
+	$(MKDIR_P) $(DESTDIR)/usr/lib/firmware
+	$(INSTALL_DATA) $(srcdir)/pca200e.bin $(DESTDIR)/usr/lib/firmware
+	$(INSTALL_DATA) $(srcdir)/pca200e_ecd.bin2 $(DESTDIR)/usr/lib/firmware
+	$(INSTALL_DATA) $(srcdir)/sba200e_ecd.bin2 $(DESTDIR)/usr/lib/firmware
 
 %.bin %.bin2: %.data
 	objcopy -Iihex $< -Obinary $@.gz
