--- ./src/config/Makefile.in.orig	2006-12-30 20:37:07.000000000 +0100
+++ ./src/config/Makefile.in	2006-12-30 20:37:30.000000000 +0100
@@ -294,10 +294,10 @@
 
 
 install-exec-local:
-	-cp hosts.atm /etc
+	-cp hosts.atm $(DESTDIR)/etc
 
 uninstall-local:
-	-rm /etc/hosts.atm
+	-rm $(DESTDIR)/etc/hosts.atm
 
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
