--- linux-atm-2.4.1/src/sigd/proto.c.orig	2004-04-21 20:33:39.327959120 -0400
+++ linux-atm-2.4.1/src/sigd/proto.c	2004-04-21 20:33:55.979427712 -0400
@@ -259,6 +259,7 @@
 		break;
 	    }
 	default:
+	    ;
     }
     va_end(ap);
     if ((size = q_close(&dsc)) >= 0) to_signaling(sock->sig,q_buffer,size);
@@ -288,6 +289,7 @@
 		}
 		break;
 	    default:
+	    ;
 	}
 	va_end(ap);
     }
--- linux-atm-2.4.1/src/lib/sapequal.c.orig	2004-04-21 20:18:48.773344000 -0400
+++ linux-atm-2.4.1/src/lib/sapequal.c	2004-04-21 20:19:15.103341232 -0400
@@ -65,6 +65,7 @@
 	    CHECK(l2.itu.window,a.l2.itu.window > b.l2.itu.window);
 	    break;
 	default:
+	    ;
     }
     switch (a.l3_proto) {
 	case ATM_L3_X25:
@@ -83,6 +84,7 @@
 	    if (a.l3.user != b.l3.user) return 0;
 	    break;
 	default:
+	    ;
     }
     return 1;
 }
--- linux-atm-2.4.1/src/maint/enitune.c.orig	2004-04-21 21:14:53.994752584 -0400
+++ linux-atm-2.4.1/src/maint/enitune.c	2004-04-21 21:15:03.371327128 -0400
@@ -44,6 +44,7 @@
 		if (*end || mult.rx <= 100) usage(name);
 		break;
 	    default:
+	    ;
 	}
     if (argc != optind+1) usage(name);
     sioc.number = strtol(argv[optind],&end,0);
