diff -Naur kbfx-0.4.9.3-20070117/cmakemodules/FindKdeLibs.cmake kbfx-0.4.9.3-20070117-p/cmakemodules/FindKdeLibs.cmake
--- kbfx-0.4.9.3-20070117/cmakemodules/FindKdeLibs.cmake	2006-12-14 09:00:32.000000000 +0100
+++ kbfx-0.4.9.3-20070117-p/cmakemodules/FindKdeLibs.cmake	2007-06-28 00:55:39.000000000 +0200
@@ -34,18 +34,7 @@
   /opt/bin
   )
 
-FIND_PROGRAM(CMAKE_UNAME uname ${BIN_SEARCH_PATHES})
-IF(CMAKE_UNAME)
-	EXECUTE_PROCESS(COMMAND ${CMAKE_UNAME} "-m"
-		OUTPUT_VARIABLE CMAKE_SYSTEM_PROCESSOR
-		ERROR_QUIET
-		)
-	IF("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "64")
-		SET(LIB_SUFFIX "64" CACHE STRING "Define suffix of directory name (32/64)" FORCE)
-	ELSE("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "64")
-		SET(LIB_SUFFIX "" CACHE STRING "Define suffix of directory name (32/64)" FORCE)
-	ENDIF("${CMAKE_SYSTEM_PROCESSOR}" MATCHES "64")
-ENDIF(CMAKE_UNAME)
+## rofl how drunk ..
 
 SET(INCLUDE_SEARCH_PATHES
   ${KDE3_INCLUDE_DIRS}

