diff -Naur kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgabout.cpp kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgabout.cpp
--- kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgabout.cpp	2006-09-22 13:19:33.000000000 +0200
+++ kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgabout.cpp	2006-09-25 15:26:30.000000000 +0200
@@ -38761,7 +38761,7 @@
     kActiveLabel1->setMaximumSize( QSize( 629, 100 ) );
     kActiveLabel1->setPaletteBackgroundPixmap( image2 );
     QFont kActiveLabel1_font(  kActiveLabel1->font() );
-    kActiveLabel1_font.setPointSize( 8 );
+    kActiveLabel1_font.setPointSize( 9 );
     kActiveLabel1->setFont( kActiveLabel1_font ); 
     kActiveLabel1->setFocusPolicy( KActiveLabel::NoFocus );
     kActiveLabel1->setTextFormat( KActiveLabel::AutoText );
@@ -38781,7 +38781,7 @@
     kActiveLabel19 = new KActiveLabel( this, "kActiveLabel19" );
     kActiveLabel19->setGeometry( QRect( -2, 175, 618, 48 ) );
     QFont kActiveLabel19_font(  kActiveLabel19->font() );
-    kActiveLabel19_font.setPointSize( 8 );
+    kActiveLabel19_font.setPointSize( 9 );
     kActiveLabel19->setFont( kActiveLabel19_font ); 
     kActiveLabel19->setFocusPolicy( KActiveLabel::NoFocus );
 
@@ -38794,13 +38794,13 @@
     textLabel3 = new QLabel( this, "textLabel3" );
     textLabel3->setGeometry( QRect( 1, 232, 485, 27 ) );
     QFont textLabel3_font(  textLabel3->font() );
-    textLabel3_font.setPointSize( 8 );
+    textLabel3_font.setPointSize( 9 );
     textLabel3->setFont( textLabel3_font ); 
 
     KbfxTeamInfo = new QTextEdit( this, "KbfxTeamInfo" );
     KbfxTeamInfo->setGeometry( QRect( 385, 264, 222, 180 ) );
     QFont KbfxTeamInfo_font(  KbfxTeamInfo->font() );
-    KbfxTeamInfo_font.setPointSize( 8 );
+    KbfxTeamInfo_font.setPointSize( 9 );
     KbfxTeamInfo->setFont( KbfxTeamInfo_font ); 
     KbfxTeamInfo->setFocusPolicy( QTextEdit::NoFocus );
     KbfxTeamInfo->setWrapPolicy( QTextEdit::AtWordOrDocumentBoundary );
@@ -38809,21 +38809,21 @@
     textLabel4 = new QLabel( this, "textLabel4" );
     textLabel4->setGeometry( QRect( 385, 445, 191, 16 ) );
     QFont textLabel4_font(  textLabel4->font() );
-    textLabel4_font.setPointSize( 8 );
+    textLabel4_font.setPointSize( 9 );
     textLabel4_font.setBold( TRUE );
     textLabel4->setFont( textLabel4_font ); 
 
     textLabel5 = new QLabel( this, "textLabel5" );
     textLabel5->setGeometry( QRect( 385, 460, 190, 57 ) );
     QFont textLabel5_font(  textLabel5->font() );
-    textLabel5_font.setPointSize( 8 );
+    textLabel5_font.setPointSize( 9 );
     textLabel5->setFont( textLabel5_font ); 
     textLabel5->setScaledContents( TRUE );
 
     textLabel6 = new KActiveLabel( this, "textLabel6" );
     textLabel6->setGeometry( QRect( 382, 520, 225, 21 ) );
     QFont textLabel6_font(  textLabel6->font() );
-    textLabel6_font.setPointSize( 8 );
+    textLabel6_font.setPointSize( 9 );
     textLabel6->setFont( textLabel6_font ); 
     textLabel6->setFocusPolicy( KActiveLabel::NoFocus );
 
diff -Naur kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgabout.ui kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgabout.ui
--- kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgabout.ui	2006-09-09 00:51:39.000000000 +0200
+++ kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgabout.ui	2006-09-25 15:32:16.000000000 +0200
@@ -181,7 +181,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="focusPolicy">
@@ -263,7 +263,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="focusPolicy">
@@ -310,7 +310,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="text">
@@ -331,7 +331,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="focusPolicy">
@@ -361,7 +361,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
                 <bold>1</bold>
             </font>
         </property>
@@ -383,7 +383,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="text">
@@ -407,7 +407,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="focusPolicy">
diff -Naur kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgbutton.cpp kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgbutton.cpp
--- kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgbutton.cpp	2006-09-22 13:19:29.000000000 +0200
+++ kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgbutton.cpp	2006-09-25 15:25:47.000000000 +0200
@@ -14242,7 +14242,7 @@
     groupBox9->setMinimumSize( QSize( 161, 73 ) );
     groupBox9->setMaximumSize( QSize( 161, 73 ) );
     QFont groupBox9_font(  groupBox9->font() );
-    groupBox9_font.setPointSize( 8 );
+    groupBox9_font.setPointSize( 9 );
     groupBox9->setFont( groupBox9_font ); 
 
     KbfxPixmapLabelHover = new KbfxPixmapLabel( groupBox9, "KbfxPixmapLabelHover" );
@@ -14259,7 +14259,7 @@
     groupBox10->setMinimumSize( QSize( 161, 73 ) );
     groupBox10->setMaximumSize( QSize( 161, 73 ) );
     QFont groupBox10_font(  groupBox10->font() );
-    groupBox10_font.setPointSize( 8 );
+    groupBox10_font.setPointSize( 9 );
     groupBox10->setFont( groupBox10_font ); 
 
     KbfxPixmapLabelPressed = new KbfxPixmapLabel( groupBox10, "KbfxPixmapLabelPressed" );
@@ -14277,7 +14277,7 @@
     kActiveLabel1->setMaximumSize( QSize( 629, 100 ) );
     kActiveLabel1->setPaletteBackgroundPixmap( image0 );
     QFont kActiveLabel1_font(  kActiveLabel1->font() );
-    kActiveLabel1_font.setPointSize( 8 );
+    kActiveLabel1_font.setPointSize( 9 );
     kActiveLabel1->setFont( kActiveLabel1_font ); 
     kActiveLabel1->setFocusPolicy( KActiveLabel::NoFocus );
     kActiveLabel1->setTextFormat( KActiveLabel::AutoText );
@@ -14304,7 +14304,7 @@
     textLabel6 = new QLabel( groupBoxNormal, "textLabel6" );
     textLabel6->setGeometry( QRect( 15, 20, 395, 16 ) );
     QFont textLabel6_font(  textLabel6->font() );
-    textLabel6_font.setPointSize( 8 );
+    textLabel6_font.setPointSize( 9 );
     textLabel6->setFont( textLabel6_font ); 
 
     KbfxURLRequesterNormal = new KURLRequester( groupBoxNormal, "KbfxURLRequesterNormal" );
@@ -14324,7 +14324,7 @@
     textLabel7 = new QLabel( groupBoxHover, "textLabel7" );
     textLabel7->setGeometry( QRect( 15, 20, 395, 16 ) );
     QFont textLabel7_font(  textLabel7->font() );
-    textLabel7_font.setPointSize( 8 );
+    textLabel7_font.setPointSize( 9 );
     textLabel7->setFont( textLabel7_font ); 
 
     KbfxURLRequesterHover = new KURLRequester( groupBoxHover, "KbfxURLRequesterHover" );
@@ -14341,7 +14341,7 @@
     groupBox7->setMinimumSize( QSize( 161, 73 ) );
     groupBox7->setMaximumSize( QSize( 161, 73 ) );
     QFont groupBox7_font(  groupBox7->font() );
-    groupBox7_font.setPointSize( 8 );
+    groupBox7_font.setPointSize( 9 );
     groupBox7->setFont( groupBox7_font ); 
     groupBox7->setAlignment( int( QGroupBox::AlignVCenter ) );
 
@@ -14390,7 +14390,7 @@
     textLabel8 = new QLabel( groupBoxPressed, "textLabel8" );
     textLabel8->setGeometry( QRect( 15, 20, 400, 16 ) );
     QFont textLabel8_font(  textLabel8->font() );
-    textLabel8_font.setPointSize( 8 );
+    textLabel8_font.setPointSize( 9 );
     textLabel8->setFont( textLabel8_font ); 
 
     groupBox11 = new QGroupBox( this, "groupBox11" );
@@ -14402,7 +14402,7 @@
     textLabel9 = new QLabel( groupBox11, "textLabel9" );
     textLabel9->setGeometry( QRect( 5, 25, 146, 45 ) );
     QFont textLabel9_font(  textLabel9->font() );
-    textLabel9_font.setPointSize( 8 );
+    textLabel9_font.setPointSize( 9 );
     textLabel9->setFont( textLabel9_font ); 
 
     groupBox8 = new QGroupBox( this, "groupBox8" );
diff -Naur kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgbutton.ui kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgbutton.ui
--- kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgbutton.ui	2006-09-20 01:35:31.000000000 +0200
+++ kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgbutton.ui	2006-09-25 15:32:36.000000000 +0200
@@ -137,7 +137,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="title">
@@ -223,7 +223,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="title">
@@ -312,7 +312,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="focusPolicy">
@@ -432,7 +432,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="text">
@@ -537,7 +537,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="text">
@@ -621,7 +621,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="title">
@@ -851,7 +851,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="text">
@@ -908,7 +908,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="text">
diff -Naur kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgfonts-old.cpp kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgfonts-old.cpp
--- kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgfonts-old.cpp	2006-09-22 13:19:36.000000000 +0200
+++ kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgfonts-old.cpp	2006-09-25 15:26:06.000000000 +0200
@@ -14230,7 +14230,7 @@
     kActiveLabel1->setGeometry( QRect( 1, 1, 629, 100 ) );
     kActiveLabel1->setPaletteBackgroundPixmap( image0 );
     QFont kActiveLabel1_font(  kActiveLabel1->font() );
-    kActiveLabel1_font.setPointSize( 8 );
+    kActiveLabel1_font.setPointSize( 9 );
     kActiveLabel1->setFont( kActiveLabel1_font ); 
     kActiveLabel1->setFocusPolicy( KActiveLabel::NoFocus );
     kActiveLabel1->setTextFormat( KActiveLabel::AutoText );
@@ -14276,14 +14276,14 @@
     kActiveLabel38 = new KActiveLabel( groupBox43, "kActiveLabel38" );
     kActiveLabel38->setGeometry( QRect( 98, 67, 65, 22 ) );
     QFont kActiveLabel38_font(  kActiveLabel38->font() );
-    kActiveLabel38_font.setPointSize( 8 );
+    kActiveLabel38_font.setPointSize( 9 );
     kActiveLabel38->setFont( kActiveLabel38_font ); 
     kActiveLabel38->setFocusPolicy( KActiveLabel::NoFocus );
 
     kActiveLabel39 = new KActiveLabel( groupBox43, "kActiveLabel39" );
     kActiveLabel39->setGeometry( QRect( 40, 97, 122, 18 ) );
     QFont kActiveLabel39_font(  kActiveLabel39->font() );
-    kActiveLabel39_font.setPointSize( 8 );
+    kActiveLabel39_font.setPointSize( 9 );
     kActiveLabel39->setFont( kActiveLabel39_font ); 
     kActiveLabel39->setFocusPolicy( KActiveLabel::NoFocus );
 
@@ -14308,7 +14308,7 @@
     kActiveLabel40 = new KActiveLabel( groupBox43, "kActiveLabel40" );
     kActiveLabel40->setGeometry( QRect( 45, 125, 115, 21 ) );
     QFont kActiveLabel40_font(  kActiveLabel40->font() );
-    kActiveLabel40_font.setPointSize( 8 );
+    kActiveLabel40_font.setPointSize( 9 );
     kActiveLabel40->setFont( kActiveLabel40_font ); 
     kActiveLabel40->setFocusPolicy( KActiveLabel::NoFocus );
 
@@ -14323,7 +14323,7 @@
     groupBox44 = new QGroupBox( groupBox43, "groupBox44" );
     groupBox44->setGeometry( QRect( 370, 192, 232, 81 ) );
     QFont groupBox44_font(  groupBox44->font() );
-    groupBox44_font.setPointSize( 8 );
+    groupBox44_font.setPointSize( 9 );
     groupBox44->setFont( groupBox44_font ); 
 
     kActiveLabel46 = new KActiveLabel( groupBox44, "kActiveLabel46" );
@@ -14333,28 +14333,28 @@
     kActiveLabel42 = new KActiveLabel( groupBox43, "kActiveLabel42" );
     kActiveLabel42->setGeometry( QRect( 45, 189, 115, 24 ) );
     QFont kActiveLabel42_font(  kActiveLabel42->font() );
-    kActiveLabel42_font.setPointSize( 8 );
+    kActiveLabel42_font.setPointSize( 9 );
     kActiveLabel42->setFont( kActiveLabel42_font ); 
     kActiveLabel42->setFocusPolicy( KActiveLabel::NoFocus );
 
     kActiveLabel41 = new KActiveLabel( groupBox43, "kActiveLabel41" );
     kActiveLabel41->setGeometry( QRect( 29, 159, 132, 21 ) );
     QFont kActiveLabel41_font(  kActiveLabel41->font() );
-    kActiveLabel41_font.setPointSize( 8 );
+    kActiveLabel41_font.setPointSize( 9 );
     kActiveLabel41->setFont( kActiveLabel41_font ); 
     kActiveLabel41->setFocusPolicy( KActiveLabel::NoFocus );
 
     kActiveLabel43_2 = new KActiveLabel( groupBox43, "kActiveLabel43_2" );
     kActiveLabel43_2->setGeometry( QRect( 67, 251, 95, 20 ) );
     QFont kActiveLabel43_2_font(  kActiveLabel43_2->font() );
-    kActiveLabel43_2_font.setPointSize( 8 );
+    kActiveLabel43_2_font.setPointSize( 9 );
     kActiveLabel43_2->setFont( kActiveLabel43_2_font ); 
     kActiveLabel43_2->setFocusPolicy( KActiveLabel::NoFocus );
 
     kActiveLabel43 = new KActiveLabel( groupBox43, "kActiveLabel43" );
     kActiveLabel43->setGeometry( QRect( 50, 220, 112, 25 ) );
     QFont kActiveLabel43_font(  kActiveLabel43->font() );
-    kActiveLabel43_font.setPointSize( 8 );
+    kActiveLabel43_font.setPointSize( 9 );
     kActiveLabel43->setFont( kActiveLabel43_font ); 
     kActiveLabel43->setFocusPolicy( KActiveLabel::NoFocus );
 
diff -Naur kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgfonts-old.ui kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgfonts-old.ui
--- kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgfonts-old.ui	2006-09-17 05:59:30.000000000 +0200
+++ kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgfonts-old.ui	2006-09-25 15:34:08.000000000 +0200
@@ -35,7 +35,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="focusPolicy">
@@ -207,7 +207,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="focusPolicy">
@@ -231,7 +231,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="focusPolicy">
@@ -385,7 +385,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="focusPolicy">
@@ -447,7 +447,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="title">
@@ -487,7 +487,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="focusPolicy">
@@ -511,7 +511,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="focusPolicy">
@@ -535,7 +535,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="focusPolicy">
@@ -559,7 +559,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="focusPolicy">
diff -Naur kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgstyle.cpp kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgstyle.cpp
--- kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgstyle.cpp	2006-09-22 13:19:34.000000000 +0200
+++ kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgstyle.cpp	2006-09-25 15:25:24.000000000 +0200
@@ -28452,7 +28452,7 @@
     kActiveLabel1->setMaximumSize( QSize( 629, 100 ) );
     kActiveLabel1->setPaletteBackgroundPixmap( image0 );
     QFont kActiveLabel1_font(  kActiveLabel1->font() );
-    kActiveLabel1_font.setPointSize( 8 );
+    kActiveLabel1_font.setPointSize( 9 );
     kActiveLabel1->setFont( kActiveLabel1_font ); 
     kActiveLabel1->setFocusPolicy( KActiveLabel::NoFocus );
     kActiveLabel1->setTextFormat( KActiveLabel::AutoText );
diff -Naur kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgstyle.ui kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgstyle.ui
--- kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgstyle.ui	2006-09-09 00:57:18.000000000 +0200
+++ kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgstyle.ui	2006-09-25 15:32:57.000000000 +0200
@@ -52,7 +52,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="focusPolicy">
diff -Naur kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgthemes.cpp kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgthemes.cpp
--- kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgthemes.cpp	2006-09-22 13:19:31.000000000 +0200
+++ kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgthemes.cpp	2006-09-25 15:24:16.000000000 +0200
@@ -18222,21 +18222,21 @@
     KbfxDeleteTheme = new QPushButton( this, "KbfxDeleteTheme" );
     KbfxDeleteTheme->setGeometry( QRect( 138, 507, 95, 29 ) );
     QFont KbfxDeleteTheme_font(  KbfxDeleteTheme->font() );
-    KbfxDeleteTheme_font.setPointSize( 8 );
+    KbfxDeleteTheme_font.setPointSize( 9 );
     KbfxDeleteTheme->setFont( KbfxDeleteTheme_font ); 
     KbfxDeleteTheme->setFocusPolicy( QPushButton::WheelFocus );
 
     KbfxInstallTheme = new QPushButton( this, "KbfxInstallTheme" );
     KbfxInstallTheme->setGeometry( QRect( 242, 507, 95, 29 ) );
     QFont KbfxInstallTheme_font(  KbfxInstallTheme->font() );
-    KbfxInstallTheme_font.setPointSize( 8 );
+    KbfxInstallTheme_font.setPointSize( 9 );
     KbfxInstallTheme->setFont( KbfxInstallTheme_font ); 
     KbfxInstallTheme->setFocusPolicy( QPushButton::WheelFocus );
 
     KbfxGetMoreThemes = new QPushButton( this, "KbfxGetMoreThemes" );
     KbfxGetMoreThemes->setGeometry( QRect( 345, 507, 95, 29 ) );
     QFont KbfxGetMoreThemes_font(  KbfxGetMoreThemes->font() );
-    KbfxGetMoreThemes_font.setPointSize( 8 );
+    KbfxGetMoreThemes_font.setPointSize( 9 );
     KbfxGetMoreThemes->setFont( KbfxGetMoreThemes_font ); 
     KbfxGetMoreThemes->setFocusPolicy( QPushButton::WheelFocus );
 
@@ -18282,7 +18282,7 @@
     textLabel2 = new QLabel( groupBox18, "textLabel2" );
     textLabel2->setGeometry( QRect( 10, 14, 150, 48 ) );
     QFont textLabel2_font(  textLabel2->font() );
-    textLabel2_font.setPointSize( 8 );
+    textLabel2_font.setPointSize( 9 );
     textLabel2->setFont( textLabel2_font ); 
     textLabel2->setAlignment( int( QLabel::WordBreak | QLabel::AlignCenter ) );
 
diff -Naur kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgthemes.ui kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgthemes.ui
--- kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgthemes.ui	2006-09-20 01:21:54.000000000 +0200
+++ kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgthemes.ui	2006-09-25 15:33:47.000000000 +0200
@@ -247,7 +247,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="focusPolicy">
@@ -277,7 +277,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="focusPolicy">
@@ -307,7 +307,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="focusPolicy">
@@ -508,7 +508,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="text">
diff -Naur kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgtooltip.cpp kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgtooltip.cpp
--- kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgtooltip.cpp	2006-09-22 13:19:35.000000000 +0200
+++ kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgtooltip.cpp	2006-09-25 15:24:39.000000000 +0200
@@ -14977,7 +14977,7 @@
     kActiveLabel1->setMaximumSize( QSize( 629, 100 ) );
     kActiveLabel1->setPaletteBackgroundPixmap( image0 );
     QFont kActiveLabel1_font(  kActiveLabel1->font() );
-    kActiveLabel1_font.setPointSize( 8 );
+    kActiveLabel1_font.setPointSize( 9 );
     kActiveLabel1->setFont( kActiveLabel1_font ); 
     kActiveLabel1->setFocusPolicy( KActiveLabel::NoFocus );
     kActiveLabel1->setTextFormat( KActiveLabel::AutoText );
@@ -15017,13 +15017,13 @@
     textLabel2->setGeometry( QRect( 517, 10, 106, 16 ) );
     textLabel2->setPaletteBackgroundColor( QColor( 255, 255, 255 ) );
     QFont textLabel2_font(  textLabel2->font() );
-    textLabel2_font.setPointSize( 8 );
+    textLabel2_font.setPointSize( 9 );
     textLabel2->setFont( textLabel2_font ); 
 
     textLabel1 = new QLabel( groupBox24, "textLabel1" );
     textLabel1->setGeometry( QRect( 11, 15, 337, 38 ) );
     QFont textLabel1_font(  textLabel1->font() );
-    textLabel1_font.setPointSize( 8 );
+    textLabel1_font.setPointSize( 9 );
     textLabel1->setFont( textLabel1_font ); 
 
     KbfxTooltipOnPixmapLabel = new QLabel( groupBox24, "KbfxTooltipOnPixmapLabel" );
@@ -15061,7 +15061,7 @@
     textLabel2_2->setGeometry( QRect( 510, 10, 117, 16 ) );
     textLabel2_2->setPaletteBackgroundColor( QColor( 255, 255, 255 ) );
     QFont textLabel2_2_font(  textLabel2_2->font() );
-    textLabel2_2_font.setPointSize( 8 );
+    textLabel2_2_font.setPointSize( 9 );
     textLabel2_2->setFont( textLabel2_2_font ); 
 
     KbfxTooltipAnimationPixmapLabel = new QLabel( groupBox26, "KbfxTooltipAnimationPixmapLabel" );
@@ -15073,7 +15073,7 @@
     textLabel1_2 = new QLabel( groupBox26, "textLabel1_2" );
     textLabel1_2->setGeometry( QRect( 11, 14, 337, 38 ) );
     QFont textLabel1_2_font(  textLabel1_2->font() );
-    textLabel1_2_font.setPointSize( 8 );
+    textLabel1_2_font.setPointSize( 9 );
     textLabel1_2->setFont( textLabel1_2_font ); 
 
     KbfxTooltipAnimationButton = new QPushButton( groupBox26, "KbfxTooltipAnimationButton" );
@@ -15103,7 +15103,7 @@
     textLabel1_3 = new QLabel( groupBox3, "textLabel1_3" );
     textLabel1_3->setGeometry( QRect( 9, 18, 288, 17 ) );
     QFont textLabel1_3_font(  textLabel1_3->font() );
-    textLabel1_3_font.setPointSize( 8 );
+    textLabel1_3_font.setPointSize( 9 );
     textLabel1_3->setFont( textLabel1_3_font ); 
 
     KbfxTooltipTextEdit = new KTextEdit( groupBox3, "KbfxTooltipTextEdit" );
diff -Naur kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgtooltip.ui kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgtooltip.ui
--- kbfx-0.4.9.2rc2/kbfxconfigapp/kbfxconfigdlgtooltip.ui	2006-09-17 10:39:03.000000000 +0200
+++ kbfx-0.4.9.2rc2-wtf/kbfxconfigapp/kbfxconfigdlgtooltip.ui	2006-09-25 15:33:15.000000000 +0200
@@ -75,7 +75,7 @@
         </property>
         <property name="font">
             <font>
-                <pointsize>8</pointsize>
+                <pointsize>9</pointsize>
             </font>
         </property>
         <property name="focusPolicy">
@@ -256,7 +256,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="text">
@@ -277,7 +277,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="text">
@@ -445,7 +445,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="text">
@@ -488,7 +488,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="text">
@@ -624,7 +624,7 @@
             </property>
             <property name="font">
                 <font>
-                    <pointsize>8</pointsize>
+                    <pointsize>9</pointsize>
                 </font>
             </property>
             <property name="text">
diff -Naur kbfx-0.4.9.2rc2/theme/default/kbfxfontrc kbfx-0.4.9.2rc2-wtf/theme/default/kbfxfontrc
--- kbfx-0.4.9.2rc2/theme/default/kbfxfontrc	2006-09-21 22:23:30.000000000 +0200
+++ kbfx-0.4.9.2rc2-wtf/theme/default/kbfxfontrc	2006-09-25 15:27:54.000000000 +0200
@@ -1,7 +1,7 @@
 [Fonts]
 ApplicationText=0,0,0
 CommentText=47,63,85
-FontSize=8
+FontSize=10
 GroupText=255,255,255
 LineColor=255,220,125
 SeparatorColor=0,0,0
