diff -Naur ktechlab-0.3.orig/src/itemdocument.h ktechlab-0.3/src/itemdocument.h
--- ktechlab-0.3.orig/src/itemdocument.h	2005-12-30 12:22:12.000000000 +0100
+++ ktechlab-0.3/src/itemdocument.h	2007-07-01 10:26:21.000000000 +0200
@@ -112,7 +112,7 @@
 		 * (such as PIC/START) have restrictions, and can only have one instance of
 		 * themselves on the canvas, and adds the operation to the undo list
 		 */
-		virtual Item* addItem( const QString &id, const QPoint &p, bool newItem ) = 0l;
+		virtual Item* addItem( const QString &id, const QPoint &p, bool newItem ) = 0;
 		/**
 		 * @returns A pointer to the canvas
 		 */
@@ -215,7 +215,7 @@
 		 * Set the given QCanvasItem (which will attempt to be casted to known
 		 * items to be deleted.
 		 */
-		virtual void appendDeleteList( QCanvasItem * ) = 0l;
+		virtual void appendDeleteList( QCanvasItem * ) = 0;
 		/**
 		 * Save the current state of the document to the undo/redo history.
 		 * @param actionTicket if this is non-negative, and the last state save
