From 148e0ccb50ce419e18e452eb7ccfe03cb4881634 Mon Sep 17 00:00:00 2001
From: =?latin2?q?Caol=E1n=20McNamara?= <caolanm@redhat.com>
Date: Tue, 5 Jul 2011 10:22:23 +0100
Subject: [PATCH] workaround for x86 gcc 4.6.1 link failure

---
 sc/Library_vbaobj.mk |   15 ++++++++++++++-
 1 files changed, 14 insertions(+), 1 deletions(-)

diff --git a/sc/Library_vbaobj.mk b/sc/Library_vbaobj.mk
index 51d0ed5..58dc936 100644
--- a/sc/Library_vbaobj.mk
+++ b/sc/Library_vbaobj.mk
@@ -123,7 +123,6 @@ $(eval $(call gb_Library_add_exception_objects,vbaobj,\
         sc/source/ui/vba/vbaquerytable \
         sc/source/ui/vba/vbarange \
         sc/source/ui/vba/vbasheetobject \
-        sc/source/ui/vba/vbasheetobjects \
         sc/source/ui/vba/vbastyle \
         sc/source/ui/vba/vbastyles \
         sc/source/ui/vba/vbatextboxshape \
@@ -138,6 +137,20 @@ $(eval $(call gb_Library_add_exception_objects,vbaobj,\
         sc/source/ui/vba/vbawsfunction \
 ))
 
+#32bit Fedora gcc 4.6.1 fails to link sc without this
+#TO-DO: make standalone reproducer for this and file
+#gcc bug
+ifeq ($(strip $(COM)),GCC)
+$(eval $(call gb_Library_add_cxxobjects,vbaobj,\
+        sc/source/ui/vba/vbasheetobjects \
+        , $(gb_COMPILERNOOPTFLAGS) $(gb_LinkTarget_EXCEPTIONFLAGS) \
+))
+else
+$(eval $(call gb_Library_add_exception_objects,vbaobj,\
+    sc/source/ui/vba/vbasheetobjects \
+))
+endif
+
 ifeq ($(OS),WNT)
 $(eval $(call gb_Library_add_linked_libs,vbaobj,\
 	advapi32 \
-- 
1.7.6

