diff --git a/configure.ac b/configure.ac
index a9c962f..543c46d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -8388,7 +8388,7 @@ AC_MSG_CHECKING([whether to enable HarfBuzz support])
 if test "$_os" != "WINNT" -a "$_os" != "Darwin"; then
     AC_MSG_RESULT([yes])
     ENABLE_HARFBUZZ="TRUE"
-    if test "$with_system_harfbuzz" = "yes"; then
+    if test "$with_system_harfbuzz" = "yes" -o "$with_system_libs" = "yes"; then
 	if $PKG_CONFIG --atleast-version 0.9.18 harfbuzz; then
 		libo_CHECK_SYSTEM_MODULE([harfbuzz],[HARFBUZZ],[harfbuzz-icu >= 0.9.18])
 	else
