--- solenv/bin/modules/installer/worker.pm.orig	2013-11-11 06:15:30.717721872 +0100
+++ solenv/bin/modules/installer/worker.pm	2013-11-11 08:57:10.882562943 +0100
@@ -23,6 +23,7 @@
 use File::stat;
 use File::Temp qw(tmpnam);
 use File::Path;
+use File::Basename;
 use installer::control;
 use installer::converter;
 use installer::exiter;
@@ -369,7 +370,9 @@
         my $onelink = ${$unixlinksarray}[$i];
         my $target = $onelink->{'Target'};
         my $destination = $onelink->{'destination'};
-        my $cmd = "ln -sf '$target' '$destdir$destination'";
+        my $cmd = "mkdir -p '" . dirname($destdir . $destination) . "'";
+        system($cmd) && die "Failed to execute \"$cmd\"";
+        $cmd = "ln -sf '$target' '$destdir$destination'";
 
         system($cmd) && die "Failed \"$cmd\"";
         push @lines, "$destination\n";
