--- microlens-th-0.4.3.10/microlens-th.cabal	2021-05-10 18:04:37.000000000 +0200
+++ microlens-th-0.4.3.10.new/microlens-th.cabal	2022-10-10 09:42:06.004041494 +0200
@@ -40,7 +40,7 @@
                      , microlens >=0.4.0 && <0.5
                      , containers >=0.5 && <0.7
                      , transformers
-                     , template-haskell >=2.8 && <2.19
+                     , template-haskell >=2.8 && <2.20
                      , th-abstraction >=0.4.1 && <0.5
 
   ghc-options:
