--- microlens-th-0.4.3.14/microlens-th.cabal	2001-09-09 03:46:40.000000000 +0200
+++ microlens-th-0.4.3.14.new/microlens-th.cabal	2024-04-16 14:51:02.917780274 +0200
@@ -45,7 +45,7 @@
                      , containers >=0.5 && <0.7
                      , transformers
                      , template-haskell >=2.8 && <2.22
-                     , th-abstraction >=0.4.1 && <0.7
+                     , th-abstraction >=0.4.1 && <0.8
 
   ghc-options:
     -Wall -fwarn-tabs
