diff -Naur smplayer-0.5.0/Makefile smplayer-0.5.0-p/Makefile
--- smplayer-0.5.0/Makefile	2007-05-25 19:20:44.000000000 +0200
+++ smplayer-0.5.0-p/Makefile	2007-06-13 18:05:13.000000000 +0200
@@ -1,13 +1,13 @@
 
-PREFIX=/usr/local
+PREFIX=/usr
 #PREFIX=/tmp/smplayer
 
 CONF_PREFIX=$(PREFIX)
 
 DATA_PATH=$(PREFIX)/share/smplayer
-DOC_PATH=$(PREFIX)/share/doc/packages/smplayer
+DOC_PATH=$(PREFIX)/share/doc/smplayer-DUMMY
 TRANSLATION_PATH=$(PREFIX)/share/smplayer/translations
-CONF_PATH=$(CONF_PREFIX)/etc/smplayer
+CONF_PATH=/etc/smplayer
 THEMES_PATH=$(PREFIX)/share/smplayer/themes
 SHORTCUTS_PATH=$(PREFIX)/share/smplayer/shortcuts
 
@@ -18,7 +18,7 @@
 KDE_ICONS=$(KDE_PREFIX)/share/icons/hicolor/
 KDE_APPLNK=$(KDE_PREFIX)/share/applications/
 
-QMAKE=qmake
+QMAKE=$(QTDIR)/bin/qmake
 
 DEFS=DATA_PATH=\\\"$(DATA_PATH)\\\" CONF_PATH=\\\"$(CONF_PATH)\\\" \
      TRANSLATION_PATH=\\\"$(TRANSLATION_PATH)\\\" \
diff -Naur smplayer-0.5.0/src/smplayer.pro smplayer-0.5.0-p/src/smplayer.pro
--- smplayer-0.5.0/src/smplayer.pro	2007-05-27 19:36:35.000000000 +0200
+++ smplayer-0.5.0-p/src/smplayer.pro	2007-06-13 18:00:26.000000000 +0200
@@ -198,7 +198,7 @@
   DEFINES += CONF_PATH=$(CONF_PATH)
   DEFINES += THEMES_PATH=$(THEMES_PATH)
   DEFINES += SHORTCUTS_PATH=$(SHORTCUTS_PATH)
-  #DEFINES += NO_DEBUG_ON_CONSOLE
+  DEFINES += NO_DEBUG_ON_CONSOLE
 
   #DEFINES += KDE_SUPPORT
   #INCLUDEPATH += /opt/kde3/include/
