diff -aur smplayer-0.6.4/Makefile fw_smplayer-0.6.4/Makefile
--- smplayer-0.6.4/Makefile	2008-10-13 23:16:45.000000000 +0200
+++ fw_smplayer-0.6.4/Makefile	2008-10-26 20:50:01.000000000 +0100
@@ -1,13 +1,13 @@
 
-PREFIX=/usr/local
+PREFIX=/usr
 #PREFIX=/tmp/smplayer
 
 CONF_PREFIX=$(PREFIX)
 
 DATA_PATH=$(PREFIX)/share/smplayer
-DOC_PATH=$(PREFIX)/share/doc/packages/smplayer
+DOC_PATH=$(PREFIX)/share/doc/smplayer-DUMMY
 TRANSLATION_PATH=$(PREFIX)/share/smplayer/translations
-CONF_PATH=$(CONF_PREFIX)/etc/smplayer
+CONF_PATH=/etc/smplayer
 THEMES_PATH=$(PREFIX)/share/smplayer/themes
 SHORTCUTS_PATH=$(PREFIX)/share/smplayer/shortcuts
 
diff -aur smplayer-0.6.4/src/smplayer.pro fw_smplayer-0.6.4/src/smplayer.pro
--- smplayer-0.6.4/src/smplayer.pro	2008-10-22 01:32:02.000000000 +0200
+++ fw_smplayer-0.6.4/src/smplayer.pro	2008-10-26 20:48:41.000000000 +0100
@@ -252,7 +252,7 @@
 	DEFINES += CONF_PATH=$(CONF_PATH)
 	DEFINES += THEMES_PATH=$(THEMES_PATH)
 	DEFINES += SHORTCUTS_PATH=$(SHORTCUTS_PATH)
-	#DEFINES += NO_DEBUG_ON_CONSOLE
+	DEFINES += NO_DEBUG_ON_CONSOLE
 
 	#DEFINES += KDE_SUPPORT
 	#INCLUDEPATH += /opt/kde3/include/
