diff -aur smplayer-0.5.62/Makefile fw_smplayer-0.5.62/Makefile
--- smplayer-0.5.62/Makefile	2007-09-15 11:03:23.000000000 +0200
+++ fw_smplayer-0.5.62/Makefile	2007-11-27 14:39:47.000000000 +0100
@@ -1,13 +1,13 @@
 
-PREFIX=/usr/local
+PREFIX=/usr
 #PREFIX=/tmp/smplayer
 
 CONF_PREFIX=$(PREFIX)
 
 DATA_PATH=$(PREFIX)/share/smplayer
-DOC_PATH=$(PREFIX)/share/doc/packages/smplayer
+DOC_PATH=$(PREFIX)/share/doc/smplayer-DUMMY
 TRANSLATION_PATH=$(PREFIX)/share/smplayer/translations
-CONF_PATH=$(CONF_PREFIX)/etc/smplayer
+CONF_PATH=/etc/smplayer
 THEMES_PATH=$(PREFIX)/share/smplayer/themes
 SHORTCUTS_PATH=$(PREFIX)/share/smplayer/shortcuts
 
diff -aur smplayer-0.5.62/src/smplayer.pro fw_smplayer-0.5.62/src/smplayer.pro
--- smplayer-0.5.62/src/smplayer.pro	2007-11-15 19:50:42.000000000 +0100
+++ fw_smplayer-0.5.62/src/smplayer.pro	2007-11-27 14:40:03.000000000 +0100
@@ -160,7 +160,7 @@
   DEFINES += CONF_PATH=$(CONF_PATH)
   DEFINES += THEMES_PATH=$(THEMES_PATH)
   DEFINES += SHORTCUTS_PATH=$(SHORTCUTS_PATH)
-  #DEFINES += NO_DEBUG_ON_CONSOLE
+  DEFINES += NO_DEBUG_ON_CONSOLE
 
   #DEFINES += KDE_SUPPORT
   #INCLUDEPATH += /opt/kde3/include/
