diff -u -N -r22534 -r22871
--- src/plugins/preauth/pkinit/pkinit_crypto_openssl.c	(.../pkinit_crypto_openssl.c)	(revision 22534)
+++ src/plugins/preauth/pkinit/pkinit_crypto_openssl.c	(.../pkinit_crypto_openssl.c)	(revision 22871)
@@ -3624,7 +3624,11 @@
     if (buf == NULL)
 	goto cleanup;
 
-    retval = EVP_PKEY_decrypt(buf, data, (int)data_len, pkey);
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+    retval = EVP_PKEY_decrypt_old(buf, data, (int)data_len, pkey);
+#else
+     retval = EVP_PKEY_decrypt(buf, data, (int)data_len, pkey);
+#endif
     if (retval <= 0) {
 	pkiDebug("unable to decrypt received data (len=%d)\n", data_len);
 	goto cleanup;
