diff -Naur xorg-server-1.10.2/hw/xfree86/Makefile.am xorg-server-1.10.2/hw/xfree86/Makefile.am
--- xorg-server-1.10.2/hw/xfree86/Makefile.am	2011-05-21 08:18:06.000000000 +0200
+++ xorg-server-1.10.2/hw/xfree86/Makefile.am	2011-05-29 19:14:27.089299777 +0200
@@ -101,10 +101,10 @@
 
 
 install-exec-local: install-binPROGRAMS
-	(cd $(DESTDIR)$(bindir) && rm -f X && ln -s Xorg X)
+	(cd $(DESTDIR)$(bindir) && rm -f X && ln -s Xorg-fwife X-fwife)
 if INSTALL_SETUID
-	chown root $(DESTDIR)$(bindir)/Xorg
-	chmod u+s $(DESTDIR)$(bindir)/Xorg
+	chown root $(DESTDIR)$(bindir)/Xorg-fwife
+	chmod u+s $(DESTDIR)$(bindir)/Xorg-fwife
 endif
 
 # Use variables from XORG_MANPAGE_SECTIONS and X Server configuration
@@ -117,4 +117,4 @@
 	$(AM_V_GEN)$(SED) $(CONF_SUBSTS) < $(srcdir)/xorgconf.cpp > $@
 
 relink:
-	$(AM_V_at)rm -f Xorg && $(MAKE) Xorg
+	$(AM_V_at)rm -f Xorg-fwife && $(MAKE) Xorg
