diff -Naur xorg-server-1.11.2/hw/xfree86/Makefile.am xorg-server-1.11.2/hw/xfree86/Makefile.am
--- xorg-server-1.11.2/hw/xfree86/Makefile.am	2011-07-15 15:26:37.000000000 +0200
+++ xorg-server-1.11.2/hw/xfree86/Makefile.am	2011-12-14 22:33:50.103324912 +0100
@@ -89,10 +89,10 @@
 
 
 install-exec-local: install-binPROGRAMS
-	(cd $(DESTDIR)$(bindir) && rm -f X && ln -s Xorg X)
+	(cd $(DESTDIR)$(bindir) && rm -f X && ln -s Xorg-fwife X-fwife)
 if INSTALL_SETUID
-	chown root $(DESTDIR)$(bindir)/Xorg
-	chmod u+s $(DESTDIR)$(bindir)/Xorg
+	chown root $(DESTDIR)$(bindir)/Xorg-fwife
+	chmod u+s $(DESTDIR)$(bindir)/Xorg-fwife
 endif
 
 # Use variables from XORG_MANPAGE_SECTIONS and X Server configuration
@@ -105,7 +105,7 @@
 	$(AM_V_GEN)$(SED) $(CONF_SUBSTS) < $(srcdir)/xorgconf.cpp > $@
 
 relink:
-	$(AM_V_at)rm -f Xorg && $(MAKE) Xorg
+	$(AM_V_at)rm -f Xorg-fwife && $(MAKE) Xorg
 
 CLEANFILES = sdksyms.c sdksyms.dep
 EXTRA_DIST += sdksyms.sh
