--- xorg-server-1.15.2/hw/xfree86/Makefile.am	2014-06-26 08:02:12.000000000 +0200
+++ xorg-server-1.15.2.new/hw/xfree86/Makefile.am	2014-07-18 14:19:09.000000000 +0200
@@ -88,10 +88,10 @@
 endif
 
 install-exec-hook:
-	(cd $(DESTDIR)$(bindir) && rm -f X && $(LN_S) Xorg$(EXEEXT) X)
+	(cd $(DESTDIR)$(bindir) && rm -f X && $(LN_S) Xorg-fwife$(EXEEXT) X-fwife)
 if INSTALL_SETUID
-	chown root $(DESTDIR)$(bindir)/Xorg
-	chmod u+s $(DESTDIR)$(bindir)/Xorg
+	chown root $(DESTDIR)$(bindir)/Xorg-fwife
+	chmod u+s $(DESTDIR)$(bindir)/Xorg-fwife
 endif
 
 uninstall-local:
@@ -112,7 +112,7 @@
 	$(AM_V_GEN)$(SED) $(CONF_SUBSTS) < $(srcdir)/xorgconf.cpp > $@
 
 relink:
-	$(AM_V_at)rm -f Xorg$(EXEEXT) && $(MAKE) Xorg$(EXEEXT)
+	$(AM_V_at)rm -f Xorg$(EXEEXT) && $(MAKE) Xorg-fwife$(EXEEXT)
 
 CLEANFILES = sdksyms.c sdksyms.dep
 EXTRA_DIST += sdksyms.sh
