--- xorg-server-1.7.4/hw/xfree86/Makefile.am	2009-11-30 06:42:31.000000000 +0100
+++ xorg-server-1.7.4-p/hw/xfree86/Makefile.am	2010-02-10 17:39:45.000000000 +0100
@@ -102,10 +102,10 @@
 
 
 install-exec-local: install-binPROGRAMS
-	(cd $(DESTDIR)$(bindir) && rm -f X && ln -s Xorg X)
+	(cd $(DESTDIR)$(bindir) && rm -f X && ln -s Xorg-fwife X-fwife)
 if INSTALL_SETUID
-	chown root $(DESTDIR)$(bindir)/Xorg
-	chmod u+s $(DESTDIR)$(bindir)/Xorg
+	chown root $(DESTDIR)$(bindir)/Xorg-fwife
+	chmod u+s $(DESTDIR)$(bindir)/Xorg-fwife
 endif
 
 CPP_FILES_FLAGS = \
@@ -114,7 +114,7 @@
 	-DMODULEPATH=\"$(DEFAULT_MODULE_PATH)\"
 
 relink:
-	rm -f Xorg && $(MAKE) Xorg
+	rm -f Xorg-fwife && $(MAKE) Xorg
 
 xorg.conf.example.pre: xorgconf.cpp
 	cp $(srcdir)/xorgconf.cpp $@
