diff -Naur xorg-server-1.8.1/hw/xfree86/Makefile.am xorg-server-1.8.1-2/hw/xfree86/Makefile.am
--- xorg-server-1.8.1/hw/xfree86/Makefile.am	2010-05-05 00:21:24.000000000 +0200
+++ xorg-server-1.8.1/hw/xfree86/Makefile.am	2010-06-03 17:16:00.039142074 +0200
@@ -102,10 +102,10 @@
 
 
 install-exec-local: install-binPROGRAMS
-	(cd $(DESTDIR)$(bindir) && rm -f X && ln -s Xorg X)
+	(cd $(DESTDIR)$(bindir) && rm -f X && ln -s Xorg-fwife X-fwife)
 if INSTALL_SETUID
-	chown root $(DESTDIR)$(bindir)/Xorg
-	chmod u+s $(DESTDIR)$(bindir)/Xorg
+	chown root $(DESTDIR)$(bindir)/Xorg-fwife
+	chmod u+s $(DESTDIR)$(bindir)/Xorg-fwife
 endif
 
 CPP_FILES_FLAGS = \
@@ -114,7 +114,7 @@
 	-DMODULEPATH=\"$(DEFAULT_MODULE_PATH)\"
 
 relink:
-	$(AM_V_at)rm -f Xorg && $(MAKE) Xorg
+	$(AM_V_at)rm -f Xorg-fwife && $(MAKE) Xorg
 
 xorg.conf.example.pre: xorgconf.cpp
 	cp $(srcdir)/xorgconf.cpp $@
