--- aeson-1.5.2.0/aeson.cabal	2001-09-09 03:46:40.000000000 +0200
+++ aeson-1.5.2.0.new/aeson.cabal	2020-07-22 16:44:15.074942537 +0200
@@ -136,7 +136,7 @@
   -- Other dependencies
   build-depends:
     attoparsec           >= 0.13.2.2 && < 0.14,
-    dlist                >= 0.8.0.4  && < 0.9,
+    dlist                >= 0.8.0.4  && < 1.1,
     hashable             >= 1.2.7.0  && < 1.4,
     primitive            >= 0.7.0.1  && < 0.8,
     scientific           >= 0.3.6.2  && < 0.4,
