--- aeson-2.1.2.1/aeson.cabal	2001-09-09 03:46:40.000000000 +0200
+++ aeson-2.1.2.1.new/aeson.cabal	2023-03-17 14:12:00.739139085 +0100
@@ -109,15 +109,14 @@
     , containers        >=0.5.7.1  && <0.7
     , deepseq           >=1.4.2.0  && <1.5
     , exceptions        >=0.10.4   && <0.11
-    , ghc-prim          >=0.5.0.0  && <0.10
-    , template-haskell  >=2.11.0.0 && <2.20
+    , ghc-prim          >=0.5.0.0  && <0.11
+    , template-haskell  >=2.11.0.0 && <2.21
     , text              >=1.2.3.0  && <1.3  || >=2.0 && <2.1
     , time              >=1.6.0.1  && <1.13
 
   -- Compat
   build-depends:
-      base-compat-batteries  >=0.10.0 && <0.13
-    , generically            >=0.1    && <0.2
+      base-compat-batteries  >=0.10.0 && <0.14
     , time-compat            >=1.9.6  && <1.10
 
   if !impl(ghc >=8.6)
@@ -130,16 +129,16 @@
     , dlist                 >=0.8.0.4  && <1.1
     , hashable              >=1.3.5.0  && <1.5
     , indexed-traversable   >=0.1.2    && <0.2
-    , OneTuple              >=0.3.1    && <0.4
+    , OneTuple              >=0.3.1    && <0.4.2
     , primitive             >=0.7.3.0  && <0.9
     , QuickCheck            >=2.14.2   && <2.15
     , scientific            >=0.3.7.0  && <0.4
-    , semialign             >=1.2      && <1.3
-    , strict                >=0.4      && <0.5
+    , semialign             >=1.2      && <1.4
+    , strict                >=0.4      && <0.6
     , tagged                >=0.8.6    && <0.9
     , text-short            >=0.1.5    && <0.2
     , th-abstraction        >=0.3.0.0  && <0.6
-    , these                 >=1.1.1.1  && <1.2
+    , these                 >=1.1.1.1  && <1.3
     , unordered-containers  >=0.2.10.0 && <0.3
     , uuid-types            >=1.0.5    && <1.1
     , vector                >=0.12.0.1 && <0.14
@@ -207,7 +206,6 @@
     , dlist
     , filepath
     , generic-deriving      >=1.10     && <1.15
-    , generically
     , ghc-prim              >=0.2
     , hashable
     , indexed-traversable
