diff -Naur aeson-2.1.1.0/aeson.cabal aeson-2.1.1.0.new/aeson.cabal
--- aeson-2.1.1.0/aeson.cabal	2001-09-09 03:46:40.000000000 +0200
+++ aeson-2.1.1.0.new/aeson.cabal	2022-10-13 18:38:10.802794954 +0200
@@ -107,7 +107,6 @@
   -- Compat
   build-depends:
       base-compat-batteries  >=0.10.0 && <0.13
-    , generically            >=0.1    && <0.2
     , time-compat            >=1.9.6  && <1.10
 
   if !impl(ghc >=8.6)
@@ -197,7 +196,6 @@
     , dlist
     , filepath
     , generic-deriving      >=1.10     && <1.15
-    , generically
     , ghc-prim              >=0.2
     , hashable
     , indexed-traversable
