--- aeson-2.2.3.0/aeson.cabal	2001-09-09 03:46:40.000000000 +0200
+++ aeson-2.2.3.0.new/aeson.cabal	2024-07-10 19:08:12.452430186 +0200
@@ -115,7 +115,7 @@
     , character-ps          ^>=0.1
     , data-fix              ^>=0.3.2
     , dlist                 ^>=1.0
-    , hashable              ^>=1.4.6.0
+    , hashable              ^>=1.5
     , indexed-traversable   ^>=0.1.2
     , integer-conversion    ^>=0.1
     , integer-logarithms    ^>=1.0.3.1
