--- aeson-1.4.3.0/aeson.cabal	2001-09-09 03:46:40.000000000 +0200
+++ aeson-1.4.3.0.new/aeson.cabal	2019-05-21 16:17:39.425211039 +0200
@@ -144,13 +144,13 @@
 
     -- not in LTS-12.10
     tagged               >= 0.8.5    && < 0.9,
-    primitive            >= 0.6.3.0  && < 0.7
+    primitive            >= 0.6.3.0  && < 0.8
 
   -- Other dependencies
   build-depends:
     attoparsec           >= 0.13.2.2 && < 0.14,
     dlist                >= 0.8.0.4  && < 0.9,
-    hashable             >= 1.2.7.0  && < 1.3,
+    hashable             >= 1.2.7.0  && < 1.4,
     scientific           >= 0.3.6.2  && < 0.4,
     th-abstraction       >= 0.2.8.0  && < 0.4,
     time-locale-compat   >= 0.1.1.5  && < 0.2,
From c1907c9013a5a54dbc50838cb93be97ded5c8c01 Mon Sep 17 00:00:00 2001
From: Adam Bergmark <adam@bergmark.nl>
Date: Sat, 18 May 2019 16:08:54 +0200
Subject: [PATCH] Support primitive 0.7.* (fixes #702)

---
 Data/Aeson/Types/FromJSON.hs | 4 ++++
 Data/Aeson/Types/ToJSON.hs   | 4 ++++
 aeson.cabal                  | 2 +-
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/Data/Aeson/Types/FromJSON.hs b/Data/Aeson/Types/FromJSON.hs
index c915235e..7277a8ba 100644
--- a/Data/Aeson/Types/FromJSON.hs
+++ b/Data/Aeson/Types/FromJSON.hs
@@ -149,7 +149,9 @@ import qualified Data.Primitive.SmallArray as PM
 import qualified Data.Primitive.Types as PM
 
 #if MIN_VERSION_primitive(0,6,4)
+#if !MIN_VERSION_primitive(0,7,0)
 import qualified Data.Primitive.UnliftedArray as PM
+#endif
 import qualified Data.Primitive.PrimArray as PM
 #endif
 
@@ -1938,10 +1940,12 @@ instance FromJSON a => FromJSON (PM.SmallArray a) where
 instance (PM.Prim a,FromJSON a) => FromJSON (PM.PrimArray a) where
   parseJSON = fmap Exts.fromList . parseJSON
 
+#if !MIN_VERSION_primitive(0,7,0)
 instance (PM.PrimUnlifted a,FromJSON a) => FromJSON (PM.UnliftedArray a) where
   parseJSON = fmap Exts.fromList . parseJSON
 #endif
 #endif
+#endif
 
 -------------------------------------------------------------------------------
 -- time
diff --git a/Data/Aeson/Types/ToJSON.hs b/Data/Aeson/Types/ToJSON.hs
index 52c64170..d6b22cd3 100644
--- a/Data/Aeson/Types/ToJSON.hs
+++ b/Data/Aeson/Types/ToJSON.hs
@@ -135,7 +135,9 @@ import qualified Data.Primitive.SmallArray as PM
 import qualified Data.Primitive.Types as PM
 
 #if MIN_VERSION_primitive(0,6,4)
+#if !MIN_VERSION_primitive(0,7,0)
 import qualified Data.Primitive.UnliftedArray as PM
+#endif
 import qualified Data.Primitive.PrimArray as PM
 #endif
 
@@ -1957,11 +1959,13 @@ instance (PM.Prim a,ToJSON a) => ToJSON (PM.PrimArray a) where
   toJSON = toJSON . Exts.toList
   toEncoding = toEncoding . Exts.toList
 
+#if !MIN_VERSION_primitive(0,7,0)
 instance (PM.PrimUnlifted a,ToJSON a) => ToJSON (PM.UnliftedArray a) where
   toJSON = toJSON . Exts.toList
   toEncoding = toEncoding . Exts.toList
 #endif
 #endif
+#endif
 
 -------------------------------------------------------------------------------
 -- time
