--- aeson-2.2.1.0/aeson.cabal	2001-09-09 03:46:40.000000000 +0200
+++ aeson-2.2.1.0.new/aeson.cabal	2023-12-11 23:54:37.536076209 +0100
@@ -126,7 +126,7 @@
     , integer-logarithms    >=1.0.3.1  && <1.1
     , network-uri           >=2.6.4.1  && <2.7
     , OneTuple              >=0.4.1.1  && <0.5
-    , primitive             >=0.8.0.0  && <0.9
+    , primitive             >=0.8.0.0  && <0.10
     , QuickCheck            >=2.14.3   && <2.15
     , scientific            >=0.3.7.0  && <0.4
     , semialign             >=1.3      && <1.4
