--- mozilla/mail/app/profile/all-thunderbird.js.~1.1.2.11.~	2004-12-06 05:39:38.000000000 +0100
+++ mozilla/mail/app/profile/all-thunderbird.js	2004-12-14 10:49:05.435231693 +0100
@@ -152,7 +152,7 @@
 
 #ifdef MOZ_WIDGET_GTK2
 // whether to check if we're the default mail client on startup (GNOME)
-pref("mail.checkDefaultMail", true);
+pref("mail.checkDefaultMail", false);
 // whether to check if we're the default news client on startup (GNOME)
 pref("mail.checkDefaultNews", false);
 #endif
