--- mozilla/toolkit/mozapps/extensions/src/nsExtensionManager.js.in.quietextensions	2004-10-12 08:19:47.000000000 +0200
+++ mozilla/toolkit/mozapps/extensions/src/nsExtensionManager.js.in	2005-01-13 17:38:40.571100014 +0100
@@ -1823,7 +1823,7 @@
         cr.selectSkin(KEY_DEFAULT_THEME, true);
       }
       catch (e) {
-        dump("*** nsExtensionManager::_disableObsoleteExtensions - failure, catching exception so finalize window can close\n");
+//        dump("*** nsExtensionManager::_disableObsoleteExtensions - failure, catching exception so finalize window can close\n");
       }
       win.close();
       
@@ -2087,7 +2087,7 @@
       this._writeDefaults(false);
     }
     catch (e) { 
-      dump("*** ExtensionManager:_updateManifests: no access privileges to application directory, skipping.\n"); 
+//      dump("*** ExtensionManager:_updateManifests: no access privileges to application directory, skipping.\n"); 
     };
     this._writeComponentManifest(true);
     this._writeDefaults(true);
@@ -2807,7 +2807,7 @@
   _ensureDS: function nsExtensionManager__ensureDS ()
   {
     if (!this._ds) {
-      dump("*** loading the extensions datasource\n");
+      // dump("*** loading the extensions datasource\n");
       this._ds = new nsExtensionsDataSource();
       if (this._ds) {
         this._ds.loadExtensions(false);
