--- mozilla/mail/components/gnome/nsMailGNOMEIntegration.cpp.foo	2004-10-18 20:15:51.000000000 -0400
+++ mozilla/mail/components/gnome/nsMailGNOMEIntegration.cpp	2004-10-18 20:15:57.000000000 -0400
@@ -229,7 +229,7 @@
 NS_IMETHODIMP
 nsMailGNOMEIntegration::GetShowDialog(PRBool *aShow)
 {
-  *aShow = (mShowMailDialog || mShowNewsDialog);
+  *aShow = PR_FALSE;
   return NS_OK;
 }
 
--- mozilla/mail/components/prefwindow/content/pref-mailnews.xul.foo	2004-10-18 20:05:27.000000000 -0400
+++ mozilla/mail/components/prefwindow/content/pref-mailnews.xul	2004-10-18 20:09:33.000000000 -0400
@@ -52,10 +52,6 @@
 #ifdef XP_MACOSX
     _elementIDs.push("newMailNotificationBounce");
 #endif
-#ifdef MOZ_WIDGET_GTK2
-    _elementIDs.push("checkDefaultMail");
-    _elementIDs.push("checkDefaultNews");
-#endif
   ]]>
   </script>
   
@@ -72,33 +68,6 @@
               accesskey="&setDefaultNewsClient.accesskey;" 
               oncommand="onEnableNews();"/>
 #endif
-#ifdef MOZ_WIDGET_GTK2
-      <vbox align="left" id="defaultClientBox">
-        <label value="&whenStartingCheck.label;"/>
-        <grid flex="1" style="margin-left: 20px;">
-          <columns>
-            <column/>
-            <column flex="1" align="left"/>
-          </columns>
-          <rows>
-          <row>
-            <checkbox id="checkDefaultMail" prefstring="mail.checkDefaultMail"
-                      label="&mailApplication.label;"
-                      accesskey="&checkDefaultMail.accesskey;"/>
-            <button label="&checkNow.label;" accesskey="&checkNow.accesskey;"
-                    oncommand="checkDefaultMailNow(false)"/>
-          </row>
-          <row>
-            <checkbox id="checkDefaultNews" prefstring="mail.checkDefaultNews"
-                      label="&newsApplication.label;"
-                      accesskey="&checkDefaultNews.accesskey;"/>
-            <button label="&checkNow.label;" accesskey="&checkNow.accesskey;"
-                    oncommand="checkDefaultMailNow(true)"/>
-          </row>
-          </rows>
-        </grid>
-      </vbox>
-#endif
     <separator class="thin"/>
     <checkbox id="mailRememberLastMsg" label="&rememberLastMsg.label;"
               prefstring="mailnews.remember_selected_message"
