--- mozilla/uriloader/exthandler/unix/nsGNOMERegistry.cpp.foo	2004-03-06 09:41:00.000000000 -0500
+++ mozilla/uriloader/exthandler/unix/nsGNOMERegistry.cpp	2004-10-18 18:55:55.000000000 -0400
@@ -283,6 +283,11 @@
 
   gchar *nativeCommand = g_filename_from_utf8(handlerApp->command,
                                               -1, NULL, NULL, NULL);
+
+  // Some versions of the mime stuff will add whitespace to the end or
+  // beginning of the command
+  nativeCommand = g_strstrip(nativeCommand);
+
   if (!nativeCommand) {
     NS_ERROR("Could not convert helper app command to filesystem encoding");
     _gnome_vfs_mime_application_free(handlerApp);
