Index: netwerk/protocol/http/public/nsIHttpProtocolHandler.idl
===================================================================
RCS file: /cvsroot/mozilla/netwerk/protocol/http/public/nsIHttpProtocolHandler.idl,v
retrieving revision 1.7
diff -d -u -p -6 -r1.7 nsIHttpProtocolHandler.idl
--- netwerk/protocol/http/public/nsIHttpProtocolHandler.idl	14 Nov 2003 20:51:55 -0000	1.7
+++ netwerk/protocol/http/public/nsIHttpProtocolHandler.idl	16 Dec 2004 19:50:51 -0000
@@ -99,12 +99,22 @@ interface nsIHttpProtocolHandler : nsIPr
     attribute ACString language;
 
     /**
      * Get/Set the application comment misc portion.
      */
     attribute ACString misc;
+
+    /**
+     * The application string (e.g. "Firefox")
+     */
+    attribute ACString app;
+
+    /**
+     * The application sub string (e.g. "1.0")
+     */
+    attribute ACString appSub;
 };
 
 %{C++
 /**
  * At initialization time, the HTTP handler will initialize each service
  * registered under this category:
Index: netwerk/protocol/http/src/nsHttpHandler.cpp
===================================================================
RCS file: /cvsroot/mozilla/netwerk/protocol/http/src/nsHttpHandler.cpp,v
retrieving revision 1.94.2.1.2.3
diff -d -u -p -6 -r1.94.2.1.2.3 nsHttpHandler.cpp
--- netwerk/protocol/http/src/nsHttpHandler.cpp	15 Sep 2004 19:43:43 -0000	1.94.2.1.2.3
+++ netwerk/protocol/http/src/nsHttpHandler.cpp	16 Dec 2004 19:50:51 -0000
@@ -214,12 +214,14 @@ nsHttpHandler::Init()
     LOG(("> app-version = %s\n", mAppVersion.get()));
     LOG(("> platform = %s\n", mPlatform.get()));
     LOG(("> oscpu = %s\n", mOscpu.get()));
     LOG(("> security = %s\n", mSecurity.get()));
     LOG(("> language = %s\n", mLanguage.get()));
     LOG(("> misc = %s\n", mMisc.get()));
+    LOG(("> app = %s\n", mApp.get()));
+    LOG(("> app-sub = %s\n", mAppSub.get()));
     LOG(("> vendor = %s\n", mVendor.get()));
     LOG(("> vendor-sub = %s\n", mVendorSub.get()));
     LOG(("> vendor-comment = %s\n", mVendorComment.get()));
     LOG(("> product = %s\n", mProduct.get()));
     LOG(("> product-sub = %s\n", mProductSub.get()));
     LOG(("> product-comment = %s\n", mProductComment.get()));
@@ -520,16 +522,18 @@ nsHttpHandler::BuildUserAgent()
                            mOscpu.Length() +
                            mLanguage.Length() +
                            mMisc.Length() +
                            mProduct.Length() +
                            mProductSub.Length() +
                            mProductComment.Length() +
+                           mApp.Length() +
+                           mAppSub.Length() +
                            mVendor.Length() +
                            mVendorSub.Length() +
                            mVendorComment.Length() +
-                           22);
+                           24);
 
     // Application portion
     mUserAgent.Assign(mAppName);
     mUserAgent += '/';
     mUserAgent += mAppVersion;
     mUserAgent += ' ';
@@ -563,12 +567,22 @@ nsHttpHandler::BuildUserAgent()
             mUserAgent += " (";
             mUserAgent += mProductComment;
             mUserAgent += ')';
         }
     }
 
+    // App portion
+    if (!mApp.IsEmpty()) {
+        mUserAgent += ' ';
+        mUserAgent += mApp;
+        if (!mAppSub.IsEmpty()) {
+            mUserAgent += '/';
+            mUserAgent += mAppSub;
+        }
+    }
+
     // Vendor portion
     if (!mVendor.IsEmpty()) {
         mUserAgent += ' ';
         mUserAgent += mVendor;
         if (!mVendorSub.IsEmpty()) {
             mUserAgent += '/';
@@ -704,12 +718,23 @@ nsHttpHandler::PrefsChanged(nsIPrefBranc
         prefs->GetCharPref(UA_PREF("appVersion"),
             getter_Copies(mAppVersion));
         if (mAppVersion.IsEmpty())
             mAppVersion.Adopt(nsCRT::strdup(UA_APPVERSION));
         mUserAgentIsDirty = PR_TRUE;
     }
+    if (PREF_CHANGED(UA_PREF("app"))) {
+        prefs->GetCharPref(UA_PREF("app"),
+            getter_Copies(mApp));
+        mUserAgentIsDirty = PR_TRUE;
+    }
+    if (PREF_CHANGED(UA_PREF("appSub"))) {
+        prefs->GetCharPref(UA_PREF("appSub"),
+            getter_Copies(mAppSub));
+        mUserAgentIsDirty = PR_TRUE;
+    }
+
 
     // Gather vendor values.
     if (PREF_CHANGED(UA_PREF("vendor"))) {
         prefs->GetCharPref(UA_PREF("vendor"),
             getter_Copies(mVendor));
         mUserAgentIsDirty = PR_TRUE;
@@ -1448,18 +1473,50 @@ NS_IMETHODIMP
 nsHttpHandler::GetAppVersion(nsACString &value)
 {
     value = mAppVersion;
     return NS_OK;
 }
 
+
+NS_IMETHODIMP
+nsHttpHandler::GetApp(nsACString &value)
+{
+    value = mApp;
+    return NS_OK;
+}
+
+NS_IMETHODIMP
+nsHttpHandler::GetAppSub(nsACString &value)
+{
+    value = mAppSub;
+    return NS_OK;
+}
+
+NS_IMETHODIMP
+nsHttpHandler::SetApp(const nsACString &value)
+{
+    mApp = value;
+    mUserAgentIsDirty = PR_TRUE;
+    return NS_OK;
+}
+
+NS_IMETHODIMP
+nsHttpHandler::SetAppSub(const nsACString &value)
+{
+    mAppSub = value;
+    mUserAgentIsDirty = PR_TRUE;
+    return NS_OK;
+}
+
 NS_IMETHODIMP
 nsHttpHandler::GetVendor(nsACString &value)
 {
     value = mVendor;
     return NS_OK;
 }
+
 NS_IMETHODIMP
 nsHttpHandler::SetVendor(const nsACString &value)
 {
     mVendor = value;
     mUserAgentIsDirty = PR_TRUE;
     return NS_OK;
@@ -1468,12 +1525,13 @@ nsHttpHandler::SetVendor(const nsACStrin
 NS_IMETHODIMP
 nsHttpHandler::GetVendorSub(nsACString &value)
 {
     value = mVendorSub;
     return NS_OK;
 }
+
 NS_IMETHODIMP
 nsHttpHandler::SetVendorSub(const nsACString &value)
 {
     mVendorSub = value;
     mUserAgentIsDirty = PR_TRUE;
     return NS_OK;
Index: netwerk/protocol/http/src/nsHttpHandler.h
===================================================================
RCS file: /cvsroot/mozilla/netwerk/protocol/http/src/nsHttpHandler.h,v
retrieving revision 1.37.2.1
diff -d -u -p -6 -r1.37.2.1 nsHttpHandler.h
--- netwerk/protocol/http/src/nsHttpHandler.h	8 May 2004 00:37:08 -0000	1.37.2.1
+++ netwerk/protocol/http/src/nsHttpHandler.h	16 Dec 2004 19:50:51 -0000
@@ -264,12 +264,14 @@ private:
     nsXPIDLCString mVendor;
     nsXPIDLCString mVendorSub;
     nsXPIDLCString mVendorComment;
     nsXPIDLCString mProduct;
     nsXPIDLCString mProductSub;
     nsXPIDLCString mProductComment;
+    nsXPIDLCString mApp;
+    nsXPIDLCString mAppSub;
 
     nsCString      mUserAgent;
     nsXPIDLCString mUserAgentOverride;
     PRPackedBool   mUserAgentIsDirty; // true if mUserAgent should be rebuilt
 
     PRPackedBool   mUseCache;
Index: mail/app/profile/all-thunderbird.js
===================================================================
RCS file: /cvsroot/mozilla/mail/app/profile/all-thunderbird.js,v
retrieving revision 1.1.2.11
diff -d -u -p -6 -r1.1.2.11 all-thunderbird.js
--- mail/app/profile/all-thunderbird.js	6 Dec 2004 04:39:38 -0000	1.1.2.11
+++ mail/app/profile/all-thunderbird.js	16 Dec 2004 19:50:51 -0000
@@ -32,14 +32,14 @@
  * and other provisions required by the GPL or the LGPL. If you do not delete
  * the provisions above, a recipient may use your version of this file under
  * the terms of any one of the NPL, the GPL or the LGPL.
  *
  * ***** END LICENSE BLOCK ***** */
 
-pref("general.useragent.vendor", "Thunderbird");
-pref("general.useragent.vendorSub", 
+pref("general.useragent.app", "Thunderbird");
+pref("general.useragent.appSub", 
 #expand __APP_VERSION__
 );
 
 // This is this application's unique identifier used by the Extension System to identify
 // this application as an extension target, and by the SmartUpdate system to identify
 // this application to the Update server.
