diff -Naur thunderbird-45.3.0/mozilla/mozglue/build/arm.cpp thunderbird-45.3.0-p/mozilla/mozglue/build/arm.cpp
--- thunderbird-45.3.0/mozilla/mozglue/build/arm.cpp	2016-08-25 19:38:13.000000000 +0200
+++ thunderbird-45.3.0-p/mozilla/mozglue/build/arm.cpp	2016-09-28 15:44:43.485396867 +0200
@@ -104,7 +104,9 @@
 
 #  elif defined(__linux__) || defined(ANDROID)
 #    include <stdio.h>
+#define _GLIBCXX_INCLUDE_NEXT_C_HEADERS
 #    include <stdlib.h>
+#undef _GLIBCXX_INCLUDE_NEXT_C_HEADERS
 #    include <string.h>
 
 enum{
diff -Naur thunderbird-45.3.0/mozilla/nsprpub/config/make-system-wrappers.pl thunderbird-45.3.0-p/mozilla/nsprpub/config/make-system-wrappers.pl
--- thunderbird-45.3.0/mozilla/nsprpub/config/make-system-wrappers.pl	2016-08-25 19:38:14.000000000 +0200
+++ thunderbird-45.3.0-p/mozilla/nsprpub/config/make-system-wrappers.pl	2016-09-28 15:44:43.484396867 +0200
@@ -19,7 +19,9 @@
     open OUT, ">$output_dir/$_";
     print OUT "#pragma GCC system_header\n";  # suppress include_next warning
     print OUT "#pragma GCC visibility push(default)\n";
+    print OUT "#define _GLIBCXX_INCLUDE_NEXT_C_HEADERS\n";
     print OUT "#include_next \<$_\>\n";
+    print OUT "#undef _GLIBCXX_INCLUDE_NEXT_C_HEADERS\n";
     print OUT "#pragma GCC visibility pop\n";
     close OUT;
 }
