diff -Naur tovid-0.30/src/tovidgui tovid-0.30-p/src/tovidgui
--- tovid-0.30/src/tovidgui	2006-11-20 06:54:41.000000000 +0100
+++ tovid-0.30-p/src/tovidgui	2007-12-06 03:06:10.000000000 +0100
@@ -21,20 +21,20 @@
 import sys
 import traceback
 
-# Try to import wx 2.6; print error if it's unavailable
+# Try to import wx 2.8; print error if it's unavailable
 try:
     import wxversion
-    if wxversion.checkInstalled('2.6'):
-        wxversion.select('2.6-unicode')
+    if wxversion.checkInstalled('2.8'):
+        wxversion.select('2.8-unicode')
     else:
-        print "tovidgui requires wxPython 2.6; please install or upgrade wxPython."
+        print "tovidgui requires wxPython 2.8; please install or upgrade wxPython."
         sys.exit()
     import wx
 except ImportError:
     traceback.print_exc()
     print "There was an error importing the 'wx' libraries. The above"
     print "output should help you find what went wrong. Re-installing"
-    print "wxPython 2.6 (or upgrading from wxPython 2.4 to 2.6) may"
+    print "wxPython 2.8 (or upgrading from wxPython 2.6 to 2.8) may"
     print "help. Consult the tovid homepage (tovid.org) for further"
     print "assistance."
     print "Sorry, 'tovidgui' will not work."
