--- th-compat-0.1.4/th-compat.cabal	2001-09-09 03:46:40.000000000 +0200
+++ th-compat-0.1.4.new/th-compat.cabal	2023-12-10 22:44:54.174775645 +0100
@@ -33,8 +33,9 @@
                    , GHC == 8.8.4
                    , GHC == 8.10.7
                    , GHC == 9.0.2
-                   , GHC == 9.2.3
-                   , GHC == 9.4.1
+                   , GHC == 9.2.8
+                   , GHC == 9.4.5
+                   , GHC == 9.6.2
 extra-source-files:  CHANGELOG.md, README.md
 
 source-repository head
@@ -44,7 +45,7 @@
 library
   exposed-modules:     Language.Haskell.TH.Syntax.Compat
   build-depends:       base             >= 4.3 && < 5
-                     , template-haskell >= 2.5 && < 2.20
+                     , template-haskell >= 2.5 && < 2.22
   if !impl(ghc >= 8.0)
     build-depends:     fail             == 4.9.*
                      , transformers     >= 0.2 && < 0.7
@@ -63,10 +64,10 @@
   other-modules:       Language.Haskell.TH.Syntax.CompatSpec
                        Types
   build-depends:       base             >= 4.3 && < 5
-                     , base-compat      >= 0.6 && < 0.13
+                     , base-compat      >= 0.6 && < 0.14
                      , hspec            >= 2   && < 3
                      , mtl              >= 2.1 && < 2.4
-                     , template-haskell >= 2.5 && < 2.20
+                     , template-haskell >= 2.5 && < 2.22
                      , th-compat
   build-tool-depends:  hspec-discover:hspec-discover >= 2
   hs-source-dirs:      tests
