--- th-compat-0.1.5/th-compat.cabal	2001-09-09 03:46:40.000000000 +0200
+++ th-compat-0.1.5.new/th-compat.cabal	2024-07-09 00:37:41.100765031 +0200
@@ -47,7 +47,7 @@
 library
   exposed-modules:     Language.Haskell.TH.Syntax.Compat
   build-depends:       base             >= 4.3 && < 5
-                     , template-haskell >= 2.5 && < 2.22
+                     , template-haskell >= 2.5 && < 2.23
   if !impl(ghc >= 8.0)
     build-depends:     fail             == 4.9.*
                      , transformers     >= 0.2 && < 0.7
@@ -66,10 +66,10 @@
   other-modules:       Language.Haskell.TH.Syntax.CompatSpec
                        Types
   build-depends:       base             >= 4.3 && < 5
-                     , base-compat      >= 0.6 && < 0.14
+                     , base-compat      >= 0.6 && < 0.15
                      , hspec            >= 2   && < 3
                      , mtl              >= 2.1 && < 2.4
-                     , template-haskell >= 2.5 && < 2.22
+                     , template-haskell >= 2.5 && < 2.23
                      , th-compat
   build-tool-depends:  hspec-discover:hspec-discover >= 2
   hs-source-dirs:      tests
