diff -Naur monodevelop-0.5.1.orig/monodevelop.desktop monodevelop-0.5.1/monodevelop.desktop
--- monodevelop-0.5.1.orig/monodevelop.desktop	2004-06-27 06:53:36.000000000 +0200
+++ monodevelop-0.5.1/monodevelop.desktop	2005-02-12 13:43:41.727698992 +0100
@@ -3,7 +3,9 @@
 Encoding=UTF-8
 Name=MonoDevelop
 GenericName=Integrated Development Environment
+GenericName[hu]=Integrált fejlesztői környezet
 Comment=Integrated Development Environment
+Comment[hu]=Integrált fejlesztői környezet
 Exec=monodevelop
 TryExec=monodevelop
 Icon=monodevelop.png
