diff -Naur monodevelop-0.7.orig/Core/src/AddIns/DebuggerAddIn/DebuggerPaths.cs.in monodevelop-0.7/Core/src/AddIns/DebuggerAddIn/DebuggerPaths.cs.in
--- monodevelop-0.7.orig/Core/src/AddIns/DebuggerAddIn/DebuggerPaths.cs.in	1970-01-01 01:00:00.000000000 +0100
+++ monodevelop-0.7/Core/src/AddIns/DebuggerAddIn/DebuggerPaths.cs.in	2005-09-23 16:19:46.000000000 +0200
@@ -0,0 +1,8 @@
+
+namespace MonoDevelop.Debugger
+{
+	public class DebuggerPaths
+	{
+		public static readonly string VisualizerPath = "@pkglibdir@/Debugger";
+	}
+}
diff -Naur monodevelop-0.7.orig/Core/src/AddIns/DebuggerAddIn/DebuggingService.cs monodevelop-0.7/Core/src/AddIns/DebuggerAddIn/DebuggingService.cs
--- monodevelop-0.7.orig/Core/src/AddIns/DebuggerAddIn/DebuggingService.cs	2005-09-23 16:19:20.000000000 +0200
+++ monodevelop-0.7/Core/src/AddIns/DebuggerAddIn/DebuggingService.cs	2005-09-23 16:19:46.000000000 +0200
@@ -38,7 +38,6 @@
 #endif
 		public DebuggingService()
 		{
-			DebuggerBackend.Initialize ();
 #if NET_2_0
 			attr_handler = new DebugAttributeHandler();
 #endif
diff -Naur monodevelop-0.7.orig/monodevelop.in monodevelop-0.7/monodevelop.in
--- monodevelop-0.7.orig/monodevelop.in	2005-09-23 16:19:20.000000000 +0200
+++ monodevelop-0.7/monodevelop.in	2005-09-23 16:21:15.000000000 +0200
@@ -1,5 +1,7 @@
 #!/bin/sh
 
+export MOZILLA_FIVE_HOME=/usr/lib/firefox
+
 function reWrite {
     ret=""
     WD=`pwd`
