diff -Naur Sphinx-2.0.1/doc/conf.py Sphinx-2.0.1-p/doc/conf.py
--- Sphinx-2.0.1/doc/conf.py	2019-04-08 04:25:09.000000000 +0200
+++ Sphinx-2.0.1-p/doc/conf.py	2019-04-09 22:16:07.192308171 +0200
@@ -92,13 +92,13 @@
 man_pages = [
     ('contents', 'sphinx-all', 'Sphinx documentation generator system manual',
      'Georg Brandl', 1),
-    ('man/sphinx-build', 'sphinx-build', 'Sphinx documentation generator tool',
+    ('man/sphinx-build-3', 'sphinx-build', 'Sphinx documentation generator tool',
      '', 1),
-    ('man/sphinx-quickstart', 'sphinx-quickstart', 'Sphinx documentation '
+    ('man/sphinx-quickstart-3', 'sphinx-quickstart', 'Sphinx documentation '
      'template generator', '', 1),
-    ('man/sphinx-apidoc', 'sphinx-apidoc', 'Sphinx API doc generator tool',
+    ('man/sphinx-apidoc-3', 'sphinx-apidoc', 'Sphinx API doc generator tool',
      '', 1),
-    ('man/sphinx-autogen', 'sphinx-autogen', 'Generate autodoc stub pages',
+    ('man/sphinx-autogen-3', 'sphinx-autogen', 'Generate autodoc stub pages',
      '', 1),
 ]
 
diff -Naur Sphinx-2.0.1/setup.py Sphinx-2.0.1-p/setup.py
--- Sphinx-2.0.1/setup.py	2019-04-08 04:25:09.000000000 +0200
+++ Sphinx-2.0.1-p/setup.py	2019-04-09 22:16:07.193308177 +0200
@@ -216,10 +216,10 @@
     include_package_data=True,
     entry_points={
         'console_scripts': [
-            'sphinx-build = sphinx.cmd.build:main',
-            'sphinx-quickstart = sphinx.cmd.quickstart:main',
-            'sphinx-apidoc = sphinx.ext.apidoc:main',
-            'sphinx-autogen = sphinx.ext.autosummary.generate:main',
+            'sphinx-build-3 = sphinx.cmd.build:main',
+            'sphinx-quickstart-3 = sphinx.cmd.quickstart:main',
+            'sphinx-apidoc-3 = sphinx.ext.apidoc:main',
+            'sphinx-autogen-3 = sphinx.ext.autosummary.generate:main',
         ],
         'distutils.commands': [
             'build_sphinx = sphinx.setup_command:BuildDoc',
diff -Naur Sphinx-2.0.1/sphinx/cmd/build.py Sphinx-2.0.1-p/sphinx/cmd/build.py
--- Sphinx-2.0.1/sphinx/cmd/build.py	2019-04-08 17:10:05.000000000 +0200
+++ Sphinx-2.0.1-p/sphinx/cmd/build.py	2019-04-09 22:16:07.193308177 +0200
@@ -85,7 +85,7 @@
 def jobs_argument(value):
     # type: (str) -> int
     """
-    Special type to handle 'auto' flags passed to 'sphinx-build' via -j flag. Can
+    Special type to handle 'auto' flags passed to 'sphinx-build-3' via -j flag. Can
     be expanded to handle other special scaling requests, such as setting job count
     to cpu_count.
     """
@@ -107,12 +107,12 @@
         description=__("""
 Generate documentation from source files.
 
-sphinx-build generates documentation from the files in SOURCEDIR and places it
+sphinx-build-3 generates documentation from the files in SOURCEDIR and places it
 in OUTPUTDIR. It looks for 'conf.py' in SOURCEDIR for the configuration
-settings.  The 'sphinx-quickstart' tool may be used to generate template files,
+settings.  The 'sphinx-quickstart-3' tool may be used to generate template files,
 including 'conf.py'
 
-sphinx-build can create documentation in different formats. A format is
+sphinx-build-3 can create documentation in different formats. A format is
 selected by specifying the builder name on the command line; it defaults to
 HTML. Builders can also perform other tasks related to documentation
 processing.
diff -Naur Sphinx-2.0.1/sphinx/cmd/make_mode.py Sphinx-2.0.1-p/sphinx/cmd/make_mode.py
--- Sphinx-2.0.1/sphinx/cmd/make_mode.py	2019-04-08 04:25:09.000000000 +0200
+++ Sphinx-2.0.1-p/sphinx/cmd/make_mode.py	2019-04-09 22:16:07.194308184 +0200
@@ -2,7 +2,7 @@
     sphinx.cmd.make_mode
     ~~~~~~~~~~~~~~~~~~~~
 
-    sphinx-build -M command-line handling.
+    sphinx-build-3 -M command-line handling.
 
     This replaces the old, platform-dependent and once-generated content
     of Makefile / make.bat.
diff -Naur Sphinx-2.0.1/sphinx/cmd/quickstart.py Sphinx-2.0.1-p/sphinx/cmd/quickstart.py
--- Sphinx-2.0.1/sphinx/cmd/quickstart.py	2019-04-08 17:10:05.000000000 +0200
+++ Sphinx-2.0.1-p/sphinx/cmd/quickstart.py	2019-04-09 22:16:07.194308184 +0200
@@ -263,7 +263,7 @@
         print()
         print(bold(__('Error: an existing conf.py has been found in the '
                       'selected root path.')))
-        print(__('sphinx-quickstart will not overwrite existing Sphinx projects.'))
+        print(__('sphinx-quickstart-3 will not overwrite existing Sphinx projects.'))
         print()
         d['path'] = do_prompt(__('Please enter a new root path (or just Enter '
                                  'to exit)'), '', is_path)
@@ -335,7 +335,7 @@
         print()
         print(bold(__('Error: the master file %s has already been found in the '
                       'selected root path.') % (d['master'] + d['suffix'])))
-        print(__('sphinx-quickstart will not overwrite the existing file.'))
+        print(__('sphinx-quickstart-3 will not overwrite the existing file.'))
         print()
         d['master'] = do_prompt(__('Please enter a new file name, or rename the '
                                    'existing file and press Enter'), d['master'])
@@ -357,7 +357,7 @@
     if 'makefile' not in d:
         print(__('''
 A Makefile and a Windows command file can be generated for you so that you
-only have to run e.g. `make html' instead of invoking sphinx-build
+only have to run e.g. `make html' instead of invoking sphinx-build-3
 directly.'''))
         d['makefile'] = do_prompt(__('Create Makefile? (y/n)'), 'y', boolean)
 
@@ -453,8 +453,8 @@
 Use the Makefile to build the docs, like so:
    make builder
 ''') or __('''\
-Use the sphinx-build command to build the docs, like so:
-   sphinx-build -b builder %s %s
+Use the sphinx-build-3 command to build the docs, like so:
+   sphinx-build-3 -b builder %s %s
 ''') % (srcdir, builddir)) + __('''\
 where "builder" is one of the supported builders, e.g. html, latex or linkcheck.
 '''))
@@ -498,9 +498,9 @@
         description=__("""
 Generate required files for a Sphinx project.
 
-sphinx-quickstart is an interactive tool that asks some questions about your
+sphinx-quickstart-3 is an interactive tool that asks some questions about your
 project and then generates a complete documentation directory and sample
-Makefile to be used with sphinx-build.
+Makefile to be used with sphinx-build-3.
 """))
 
     parser.add_argument('-q', '--quiet', action='store_true', dest='quiet',
@@ -617,7 +617,7 @@
                 print()
                 print(bold(__('Error: specified path is not a directory, or sphinx'
                               ' files already exist.')))
-                print(__('sphinx-quickstart only generate into a empty directory.'
+                print(__('sphinx-quickstart-3 only generate into a empty directory.'
                          ' Please specify a new root path.'))
                 return 1
         else:
diff -Naur Sphinx-2.0.1/sphinx/cmdline.py Sphinx-2.0.1-p/sphinx/cmdline.py
--- Sphinx-2.0.1/sphinx/cmdline.py	2019-04-08 17:10:05.000000000 +0200
+++ Sphinx-2.0.1-p/sphinx/cmdline.py	2019-04-09 22:16:07.194308184 +0200
@@ -2,7 +2,7 @@
     sphinx.cmdline
     ~~~~~~~~~~~~~~
 
-    sphinx-build command-line handling.
+    sphinx-build-3 command-line handling.
 
     :copyright: Copyright 2007-2019 by the Sphinx team, see AUTHORS.
     :license: BSD, see LICENSE for details.
diff -Naur Sphinx-2.0.1/sphinx/ext/apidoc.py Sphinx-2.0.1-p/sphinx/ext/apidoc.py
--- Sphinx-2.0.1/sphinx/ext/apidoc.py	2019-04-08 04:25:09.000000000 +0200
+++ Sphinx-2.0.1-p/sphinx/ext/apidoc.py	2019-04-09 22:16:07.195308190 +0200
@@ -356,7 +356,7 @@
                         default='rst',
                         help=__('file suffix (default: rst)'))
     parser.add_argument('-F', '--full', action='store_true', dest='full',
-                        help=__('generate a full project with sphinx-quickstart'))
+                        help=__('generate a full project with sphinx-quickstart-3'))
     parser.add_argument('-a', '--append-syspath', action='store_true',
                         dest='append_syspath',
                         help=__('append module_path to sys.path, used when --full is given'))
diff -Naur Sphinx-2.0.1/sphinx/ext/autosummary/generate.py Sphinx-2.0.1-p/sphinx/ext/autosummary/generate.py
--- Sphinx-2.0.1/sphinx/ext/autosummary/generate.py	2019-04-08 04:25:09.000000000 +0200
+++ Sphinx-2.0.1-p/sphinx/ext/autosummary/generate.py	2019-04-09 22:17:23.385784428 +0200
@@ -11,7 +11,7 @@
     Example Makefile rule::
 
        generate:
-               sphinx-autogen -o source/generated source/*.rst
+               sphinx-autogen-3 -o source/generated source/*.rst
 
     :copyright: Copyright 2007-2019 by the Sphinx team, see AUTHORS.
     :license: BSD, see LICENSE for details.
@@ -46,7 +46,7 @@
 
 
 class DummyApplication:
-    """Dummy Application class for sphinx-autogen command."""
+    """Dummy Application class for sphinx-autogen-3 command."""
 
     def __init__(self):
         # type: () -> None
@@ -364,7 +364,7 @@
         description=__("""
 Generate ReStructuredText using autosummary directives.
 
-sphinx-autogen is a frontend to sphinx.ext.autosummary.generate. It generates
+sphinx-autogen-3 is a frontend to sphinx.ext.autosummary.generate. It generates
 the reStructuredText files from the autosummary directives contained in the
 given input files.
 
diff -Naur Sphinx-2.0.1/sphinx/ext/doctest.py Sphinx-2.0.1-p/sphinx/ext/doctest.py
--- Sphinx-2.0.1/sphinx/ext/doctest.py	2019-04-08 17:10:05.000000000 +0200
+++ Sphinx-2.0.1-p/sphinx/ext/doctest.py	2019-04-09 22:16:07.196308196 +0200
@@ -273,7 +273,7 @@
         return self.save_linecache_getlines(filename, module_globals)  # type: ignore
 
 
-# the new builder -- use sphinx-build.py -b doctest to run
+# the new builder -- use sphinx-build-3.py -b doctest to run
 
 class DocTestBuilder(Builder):
     """
diff -Naur Sphinx-2.0.1/sphinx/make_mode.py Sphinx-2.0.1-p/sphinx/make_mode.py
--- Sphinx-2.0.1/sphinx/make_mode.py	2019-04-08 17:10:05.000000000 +0200
+++ Sphinx-2.0.1-p/sphinx/make_mode.py	2019-04-09 22:16:07.196308196 +0200
@@ -2,7 +2,7 @@
     sphinx.make_mode
     ~~~~~~~~~~~~~~~~
 
-    sphinx-build -M command-line handling.
+    sphinx-build-3 -M command-line handling.
 
     This replaces the old, platform-dependent and once-generated content
     of Makefile / make.bat.
