diff -Naur snd123-1.0.2/libvlc/libcdio-0.75/lib/driver/gnu_linux.c snd123-1.0.2-p/libvlc/libcdio-0.75/lib/driver/gnu_linux.c
--- snd123-1.0.2/libvlc/libcdio-0.75/lib/driver/gnu_linux.c	2005-07-07 08:53:04.000000000 +0200
+++ snd123-1.0.2-p/libvlc/libcdio-0.75/lib/driver/gnu_linux.c	2008-04-13 22:25:16.000000000 +0200
@@ -60,6 +60,7 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <mntent.h>
+#include <limits.h>
 
 #include <linux/cdrom.h>
 #include <scsi/scsi.h>
