--- beast-0.8.2/beast-gtk/bstsnetrouter.cc	2013-07-06 01:22:58.000000000 +0300
+++ beast-0.8.2.new/beast-gtk/bstsnetrouter.cc	2014-02-23 22:12:25.000000000 +0200
@@ -316,14 +316,14 @@
 }
 
 static void
-bst_router_popup_select (gpointer user_data, size_t action_id)
+bst_router_popup_select (gpointer user_data, long unsigned int action_id)
 {
   BstSNetRouter *self = BST_SNET_ROUTER (user_data);
   gxk_action_group_select (self->canvas_tool, action_id);
 }
 
 static void
-bst_router_run_method (gpointer user_data, size_t action_id)
+bst_router_run_method (gpointer user_data, long unsigned int action_id)
 {
   BstSNetRouter *self = BST_SNET_ROUTER (user_data);
   BseCategory *cat = bse_category_from_id (action_id);
