--- beast-0.8.2/configure.in	2013-07-06 01:22:58.000000000 +0300
+++ beast-0.8.2.new/configure.in	2014-02-22 01:34:40.000000000 +0200
@@ -10,7 +10,7 @@
 #
 AC_INIT(beast, m4_esyscmd(./mkrelease.sh shellvar configure.in:BST_VERSION))
 AC_CONFIG_SRCDIR([bse/bseconfig.h.in])
-AM_CONFIG_HEADER(configure.h)
+AC_CONFIG_HEADERS(configure.h)
 AC_CONFIG_AUX_DIR([autotools])
 AC_CONFIG_MACRO_DIR([autotools])
 AC_PREREQ(2.57)
@@ -431,7 +431,7 @@
     guile_tests=0
     GUILE_VERSION=none
     REQUIRED_VERSION=1.6
-    AC_PATH_PROG(GUILE_CONFIG, guile-config)
+    AC_PATH_PROG(GUILE_CONFIG, guile1-config)
     if test -n "$GUILE_CONFIG"; then
 	guile_tests=`expr $guile_tests + 1`
 	GUILE_VERSION=`$GUILE_CONFIG info guileversion`
