--- gstreamer-1.18.0/pkgconfig/meson.build	2020-09-08 01:01:36.071851000 +0200
+++ gstreamer-1.18.0.new/pkgconfig/meson.build	2020-09-10 09:56:01.307675710 +0200
@@ -4,7 +4,7 @@
 pkgconf.set('exec_prefix', '${prefix}')
 pkgconf.set('libdir', '${prefix}/@0@'.format(get_option('libdir')))
 pkgconf.set('libexecdir', '${prefix}/@0@'.format(get_option('libexecdir')))
-pkgconf.set('includedir', '${prefix}/@0@'.format(get_option('includedir')))
+pkgconf.set('includedir', get_option('includedir'))
 pkgconf.set('GST_API_VERSION', apiversion)
 pkgconf.set('VERSION', gst_version)
 pkgconf.set('LIBM', mathlib.found() ? '-lm' : '')
