--- protontricks-1.12.0/src/protontricks/steam.py	2024-09-16 08:55:57.000000000 +0200
+++ protontricks-1.12.0.new/src/protontricks/steam.py	2024-11-10 02:08:21.495425976 +0100
@@ -8,8 +8,7 @@
 
 import vdf
 
-from ._vdf import binary_loads as vendored_binary_loads
-from .util import is_steam_deck, lower_dict
+from .util import lower_dict, is_steam_deck
 
 __all__ = (
     "COMMON_STEAM_DIRS", "SteamApp", "find_steam_installations",
@@ -562,19 +561,9 @@
 
             i += section_size
 
-            try:
-                vdf_d = vdf.binary_loads(
-                    data[i:i+vdf_section_size], key_table=key_table
-                )
-            except TypeError:
-                # System 'vdf' is too old and does not support 'key_table',
-                # use the bundled one instead. This is cursed, but it's
-                # so far the only reasonable option without a proper maintained
-                # release on PyPI.
-                vdf_d = vendored_binary_loads(
-                    data[i:i+vdf_section_size], key_table=key_table
-                )
-
+            vdf_d = vdf.binary_loads(
+                data[i:i+vdf_section_size], key_table=key_table
+            )
             vdf_d = lower_dict(vdf_d)
             yield vdf_d
 
