--- radeon-profile-daemon/radeon-profile-daemon.pro	2019-02-20 21:39:54.000000000 +0100
+++ radeon-profile-daemon.new/radeon-profile-daemon.pro	2019-02-21 11:17:11.447812618 +0100
@@ -33,10 +33,10 @@
 
 DESTDIR= target
 
-bin.path = /usr/bin
+bin.path = /var/tmp/fst/pkg/usr/bin
 bin.files = target/radeon-profile-daemon
 
-service.path = /usr/lib/systemd/system
+service.path = /var/tmp/fst/pkg/lib/systemd/system
 service.files = extra/radeon-profile-daemon.service
 
 INSTALLS += \
