diff -Naur a2ps-4.13.orig/Makefile.in a2ps-4.13/Makefile.in
--- a2ps-4.13.orig/Makefile.in	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/Makefile.in	2005-06-08 15:52:29.000000000 +0000
@@ -100,7 +100,7 @@
 INTLDEPS = @INTLDEPS@
 INTLLIBS = @INTLLIBS@
 INTLOBJS = @INTLOBJS@
-LEX = @LEX@
+LEX = @LEX@ -CFe
 LEX_OUTPUT_ROOT = @LEX_OUTPUT_ROOT@
 LIBOBJS = @LIBOBJS@
 LIBTOOL = @LIBTOOL@
diff -Naur a2ps-4.13.orig/config.h.in a2ps-4.13/config.h.in
--- a2ps-4.13.orig/config.h.in	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/config.h.in	2005-06-08 15:52:35.000000000 +0000
@@ -380,7 +380,7 @@
 #undef SYNTAX_TABLE
 
 /* Define if `sys_siglist' is declared by <signal.h> or <unistd.h>. */
-#undef SYS_SIGLIST_DECLARED
+#undef HAVE_DECL_SYS_SIGLIST
 
 /* Define if you can safely include both <sys/time.h> and <time.h>. */
 #undef TIME_WITH_SYS_TIME
diff -Naur a2ps-4.13.orig/configure.in a2ps-4.13/configure.in
--- a2ps-4.13.orig/configure.in	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/configure.in	2005-06-08 15:52:35.000000000 +0000
@@ -24,6 +24,8 @@
 ## Checks for programs.  ##
 ## --------------------- ##
 
+m4_pattern_allow([.*M4])
+m4_pattern_allow([AC_MSG_WARN])
 AC_PROG_AWK
 AC_PROG_GNU_M4
 AC_PROG_INSTALL
@@ -82,7 +84,7 @@
 AC_TYPE_PID_T
 AC_TYPE_SIGNAL
 AC_STRUCT_TM
-AC_CHECK_MEMBERS((struct passwd.pw_gecos),,,
+AC_CHECK_MEMBERS([struct passwd.pw_gecos],,,
 [#if HAVE_PWD_H
 # include <pwd.h>
 #endif])
@@ -141,7 +143,7 @@
 # Check and replace broken or missing "fnmatch"
 AC_FUNC_FNMATCH
 if test $ac_cv_func_fnmatch_works = no; then
-  LIBOBJS="$LIBOBJS fnmatch.o"
+  AC_LIBOBJ([fnmatch])
   AC_DEFINE_UNQUOTED(fnmatch, rpl_fnmatch,
 	  [Define to rpl_fnmatch if the replacement function should be used.])
 fi
@@ -149,7 +151,7 @@
 # Needed checks for tinyterm.[ch]
 ad_TINYTERM_CHECKS
 
-AC_CHECK_DECLS((fputs, strlen, strtoul, strtoull))
+AC_CHECK_DECLS([fputs, strlen, strtoul, strtoull])
 
 # Internationalization.
 AM_GNU_GETTEXT
@@ -273,7 +275,7 @@
 # Nice special printers
 ########################
 # Look for GV or Ghostview
-ad_CHECK_PROGS((gv, ghostview))
+ad_CHECK_PROGS([gv, ghostview])
 if test "$COM_gv$COM_ghostview" = "##"; then
   COM_DISPLAY="#"
 fi
@@ -286,7 +288,7 @@
 ad_CHECK_PROG(gzip,
 [a2ps works better with gzip.
 Consider fetching gzip at any ftp site proposing GNU programs])
-ad_CHECK_PROGS((bzip, bzip2))
+ad_CHECK_PROGS([bzip, bzip2])
 
 # DVI delegation
 ad_CHECK_PROG(dvips)
@@ -295,7 +297,7 @@
 ad_CHECK_PROG(convert)
 
 # PDF delegation
-ad_CHECK_PROGS((acroread, pdf2ps))
+ad_CHECK_PROGS([acroread, pdf2ps])
 COM_acroread4='#'
 AC_SUBST(COM_acroread4)
 # If we have acroread, is it a version which supports -shrink etc.?
@@ -313,7 +315,7 @@
 fi
 
 # Outputting PDF
-ad_CHECK_PROGS((distill, ps2pdf))
+ad_CHECK_PROGS([distill, ps2pdf])
 test "$COM_distill$COM_ps2pdf" = "##" && COM_PS2PDF="#"
 AC_SUBST(COM_PS2PDF)
 
@@ -326,7 +328,7 @@
 Consider fetching groff at any ftp site proposing GNU programs])
 
 # Texinfo delegation
-ad_CHECK_PROGS((makeinfo, tex, latex))
+ad_CHECK_PROGS([makeinfo, tex, latex])
 COM_TEXI="$COM_tex$COM_makeinfo$COM_dvips"
 COM_LATEX="$COM_latex$COM_dvips"
 AC_SUBST(COM_TEXI)
@@ -336,12 +338,6 @@
 ## Epilogue.  ##
 ## ---------- ##
 
-# This is necessary so that .o files in LIBOBJS are also built via
-# the ANSI2KNR-filtering rules.
-LIBOBJS=`echo "$LIBOBJS" | sed 's/\.o /\$U.o /g;s/\.o$/\$U.o/'`
-# Autoconf 2.14a is not ready for libtool
-LTLIBOBJS=`echo "$LIBOBJS" | sed 's/\.o/\.lo/g'`
-AC_SUBST(LTLIBOBJS)
 LTALLOCA=`echo "$ALLOCA" | sed 's/\.o/\.lo/g'`
 AC_SUBST(LTALLOCA)
 
diff -Naur a2ps-4.13.orig/etc/Makefile.in a2ps-4.13/etc/Makefile.in
--- a2ps-4.13.orig/etc/Makefile.in	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/etc/Makefile.in	2005-06-08 15:52:23.000000000 +0000
@@ -297,7 +297,9 @@
 
 # Building the correct a2ps.cfg
 a2ps.cfg: a2ps_cfg Makefile
-	sed "s!@libpath@!$(libpath)!" a2ps_cfg > a2ps.cfg
+	sed -e "s!@libpath@!$(libpath)!" \
+	    -e "s!@apppath@!$(PSFONT_PATH)!" \
+	    a2ps_cfg > a2ps.cfg
 
 # Building a time stamp to know the version.
 README: README.in Makefile
diff -Naur a2ps-4.13.orig/etc/a2ps_cfg.in a2ps-4.13/etc/a2ps_cfg.in
--- a2ps-4.13.orig/etc/a2ps_cfg.in	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/etc/a2ps_cfg.in	2005-06-08 15:52:23.000000000 +0000
@@ -73,6 +73,7 @@
 # It may be useful to extend it so that a2ps can see some
 # TeX or X11 resources: it likes AFM files and PF[AB] files.
 #AppendLibraryPath: foo
+AppendLibraryPath: @apppath@
 
 
 #################################################################
diff -Naur a2ps-4.13.orig/lib/Makefile.in a2ps-4.13/lib/Makefile.in
--- a2ps-4.13.orig/lib/Makefile.in	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/lib/Makefile.in	2005-06-08 15:52:29.000000000 +0000
@@ -100,7 +100,7 @@
 INTLDEPS = @INTLDEPS@
 INTLLIBS = @INTLLIBS@
 INTLOBJS = @INTLOBJS@
-LEX = @LEX@
+LEX = @LEX@ -CFe
 LEX_OUTPUT_ROOT = @LEX_OUTPUT_ROOT@
 LIBOBJS = @LIBOBJS@
 LIBTOOL = @LIBTOOL@
diff -Naur a2ps-4.13.orig/lib/fonts.l a2ps-4.13/lib/fonts.l
--- a2ps-4.13.orig/lib/fonts.l	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/lib/fonts.l	2005-06-08 15:52:29.000000000 +0000
@@ -178,7 +178,7 @@
 
 %}
 
-%option yylineno
+/* %option yylineno */
 %option prefix="afm"
 %option outfile="lex.yy.c"
 
@@ -524,7 +524,7 @@
   const char * filename;
   FILE * stream;
 
-  yylineno = 0;
+  /* yylineno = 0; */
 
   /* Find its AFM file: first resolve its font file name */
   filename = fonts_map_resolve_alias (job->fonts_map, font->key);
diff -Naur a2ps-4.13.orig/lib/signame.c a2ps-4.13/lib/signame.c
--- a2ps-4.13.orig/lib/signame.c	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/lib/signame.c	2005-06-08 15:52:35.000000000 +0000
@@ -62,7 +62,7 @@
 
 #else	/* HAVE_SYS_SIGLIST.  */
 
-#ifndef SYS_SIGLIST_DECLARED
+#if HAVE_DECL_SYS_SIGLIST == 0
 extern char *sys_siglist[];
 #endif	/* Not SYS_SIGLIST_DECLARED.  */
 
diff -Naur a2ps-4.13.orig/m4/atexit.m4 a2ps-4.13/m4/atexit.m4
--- a2ps-4.13.orig/m4/atexit.m4	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/m4/atexit.m4	2005-06-08 15:52:35.000000000 +0000
@@ -14,7 +14,7 @@
   [AC_CHECK_FUNCS(atexit)
    if test $ac_cv_func_atexit = no; then
      AC_CHECK_FUNCS(on_exit)
-     LIBOBJS="$LIBOBJS atexit.o"
+     AC_LIBOBJ([atexit])
      if test $ac_cv_func_on_exit = no; then
        AC_DEFINE_UNQUOTED(exit, rpl_exit,
 			  [Define to rpl_exit if the replacement function should be used])
diff -Naur a2ps-4.13.orig/m4/c-bs-a.m4 a2ps-4.13/m4/c-bs-a.m4
--- a2ps-4.13.orig/m4/c-bs-a.m4	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/m4/c-bs-a.m4	2005-06-08 15:52:47.000000000 +0000
@@ -2,7 +2,7 @@
 
 dnl From Paul Eggert.
 
-AC_DEFUN(AC_C_BACKSLASH_A,
+AC_DEFUN([AC_C_BACKSLASH_A],
 [
   AC_CACHE_CHECK([whether backslash-a works in strings], ac_cv_c_backslash_a,
    [AC_TRY_COMPILE([],
@@ -22,7 +22,7 @@
   fi
 ])
 
-AC_DEFUN(jm_PREREQ_QUOTEARG,
+AC_DEFUN([jm_PREREQ_QUOTEARG],
 [
   AC_CHECK_FUNCS(isascii mbrtowc)
   AC_CHECK_HEADERS(limits.h stdlib.h string.h wchar.h)
diff -Naur a2ps-4.13.orig/m4/error.m4 a2ps-4.13/m4/error.m4
--- a2ps-4.13.orig/m4/error.m4	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/m4/error.m4	2005-06-08 15:52:47.000000000 +0000
@@ -1,13 +1,12 @@
 dnl From Jim Meyering.  Use this if you use the GNU error.[ch].
 dnl FIXME: Migrate into libit
 
-AC_DEFUN(AM_FUNC_ERROR_AT_LINE,
+AC_DEFUN([AM_FUNC_ERROR_AT_LINE],
 [AC_CACHE_CHECK([for error_at_line], am_cv_lib_error_at_line,
  [AC_TRY_LINK([],[error_at_line(0, 0, "", 0, "");],
               am_cv_lib_error_at_line=yes,
 	      am_cv_lib_error_at_line=no)])
  if test $am_cv_lib_error_at_line = no; then
-   LIBOBJS="$LIBOBJS error.$ac_objext"
+   AC_LIBOBJ([error])
  fi
- AC_SUBST(LIBOBJS)dnl
 ])
diff -Naur a2ps-4.13.orig/m4/fullpath.m4 a2ps-4.13/m4/fullpath.m4
--- a2ps-4.13.orig/m4/fullpath.m4	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/m4/fullpath.m4	2005-06-08 15:52:35.000000000 +0000
@@ -37,12 +37,6 @@
     COM_$1=""
   fi
 fi
-ifval([$2],
-[if test "[$]COM_$1" = "#"; then
-  AC_MSG_WARN([============================================================
-$2])
-  AC_MSG_WARN([============================================================])
-fi])
 AC_SUBST($1)
 AC_SUBST(COM_$1)])
 
diff -Naur a2ps-4.13.orig/m4/lex.m4 a2ps-4.13/m4/lex.m4
--- a2ps-4.13.orig/m4/lex.m4	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/m4/lex.m4	2005-06-08 15:52:47.000000000 +0000
@@ -3,7 +3,7 @@
 
 dnl AM_PROG_LEX
 dnl Look for flex, lex or missing, then run AC_PROG_LEX and AC_DECL_YYTEXT
-AC_DEFUN(AM_PROG_LEX,
+AC_DEFUN([AM_PROG_LEX],
 [AC_REQUIRE([AM_MISSING_HAS_RUN])
 AC_CHECK_PROGS(LEX, flex lex, [${am_missing_run}flex])
 AC_PROG_LEX
diff -Naur a2ps-4.13.orig/m4/libtool.m4 a2ps-4.13/m4/libtool.m4
--- a2ps-4.13.orig/m4/libtool.m4	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/m4/libtool.m4	2005-06-08 15:52:47.000000000 +0000
@@ -22,7 +22,7 @@
 ## the same distribution terms that you use for the rest of that program.
 
 # serial 40 AC_PROG_LIBTOOL
-AC_DEFUN(AC_PROG_LIBTOOL,
+AC_DEFUN([AC_PROG_LIBTOOL],
 [AC_REQUIRE([AC_LIBTOOL_SETUP])dnl
 
 # Save cache, so that ltconfig can load it
@@ -52,7 +52,7 @@
 exec 5>>./config.log
 ])
 
-AC_DEFUN(AC_LIBTOOL_SETUP,
+AC_DEFUN([AC_LIBTOOL_SETUP],
 [AC_PREREQ(2.13)dnl
 AC_REQUIRE([AC_ENABLE_SHARED])dnl
 AC_REQUIRE([AC_ENABLE_STATIC])dnl
@@ -132,16 +132,16 @@
 ])
 
 # AC_LIBTOOL_DLOPEN - enable checks for dlopen support
-AC_DEFUN(AC_LIBTOOL_DLOPEN, [AC_BEFORE([$0],[AC_LIBTOOL_SETUP])])
+AC_DEFUN([AC_LIBTOOL_DLOPEN], [AC_BEFORE([$0],[AC_LIBTOOL_SETUP])])
 
 # AC_LIBTOOL_WIN32_DLL - declare package support for building win32 dll's
-AC_DEFUN(AC_LIBTOOL_WIN32_DLL, [AC_BEFORE([$0], [AC_LIBTOOL_SETUP])])
+AC_DEFUN([AC_LIBTOOL_WIN32_DLL], [AC_BEFORE([$0], [AC_LIBTOOL_SETUP])])
 
 # AC_ENABLE_SHARED - implement the --enable-shared flag
 # Usage: AC_ENABLE_SHARED[(DEFAULT)]
 #   Where DEFAULT is either `yes' or `no'.  If omitted, it defaults to
 #   `yes'.
-AC_DEFUN(AC_ENABLE_SHARED, [dnl
+AC_DEFUN([AC_ENABLE_SHARED], [dnl
 define([AC_ENABLE_SHARED_DEFAULT], ifelse($1, no, no, yes))dnl
 AC_ARG_ENABLE(shared,
 changequote(<<, >>)dnl
@@ -167,14 +167,14 @@
 ])
 
 # AC_DISABLE_SHARED - set the default shared flag to --disable-shared
-AC_DEFUN(AC_DISABLE_SHARED, [AC_BEFORE([$0],[AC_LIBTOOL_SETUP])dnl
+AC_DEFUN([AC_DISABLE_SHARED], [AC_BEFORE([$0],[AC_LIBTOOL_SETUP])dnl
 AC_ENABLE_SHARED(no)])
 
 # AC_ENABLE_STATIC - implement the --enable-static flag
 # Usage: AC_ENABLE_STATIC[(DEFAULT)]
 #   Where DEFAULT is either `yes' or `no'.  If omitted, it defaults to
 #   `yes'.
-AC_DEFUN(AC_ENABLE_STATIC, [dnl
+AC_DEFUN([AC_ENABLE_STATIC], [dnl
 define([AC_ENABLE_STATIC_DEFAULT], ifelse($1, no, no, yes))dnl
 AC_ARG_ENABLE(static,
 changequote(<<, >>)dnl
@@ -200,7 +200,7 @@
 ])
 
 # AC_DISABLE_STATIC - set the default static flag to --disable-static
-AC_DEFUN(AC_DISABLE_STATIC, [AC_BEFORE([$0],[AC_LIBTOOL_SETUP])dnl
+AC_DEFUN([AC_DISABLE_STATIC], [AC_BEFORE([$0],[AC_LIBTOOL_SETUP])dnl
 AC_ENABLE_STATIC(no)])
 
 
@@ -208,7 +208,7 @@
 # Usage: AC_ENABLE_FAST_INSTALL[(DEFAULT)]
 #   Where DEFAULT is either `yes' or `no'.  If omitted, it defaults to
 #   `yes'.
-AC_DEFUN(AC_ENABLE_FAST_INSTALL, [dnl
+AC_DEFUN([AC_ENABLE_FAST_INSTALL], [dnl
 define([AC_ENABLE_FAST_INSTALL_DEFAULT], ifelse($1, no, no, yes))dnl
 AC_ARG_ENABLE(fast-install,
 changequote(<<, >>)dnl
@@ -234,11 +234,11 @@
 ])
 
 # AC_ENABLE_FAST_INSTALL - set the default to --disable-fast-install
-AC_DEFUN(AC_DISABLE_FAST_INSTALL, [AC_BEFORE([$0],[AC_LIBTOOL_SETUP])dnl
+AC_DEFUN([AC_DISABLE_FAST_INSTALL], [AC_BEFORE([$0],[AC_LIBTOOL_SETUP])dnl
 AC_ENABLE_FAST_INSTALL(no)])
 
 # AC_PROG_LD - find the path to the GNU or non-GNU linker
-AC_DEFUN(AC_PROG_LD,
+AC_DEFUN([AC_PROG_LD],
 [AC_ARG_WITH(gnu-ld,
 [  --with-gnu-ld           assume the C compiler uses GNU ld [default=no]],
 test "$withval" = no || with_gnu_ld=yes, with_gnu_ld=no)
@@ -308,7 +308,7 @@
 AC_PROG_LD_GNU
 ])
 
-AC_DEFUN(AC_PROG_LD_GNU,
+AC_DEFUN([AC_PROG_LD_GNU],
 [AC_CACHE_CHECK([if the linker ($LD) is GNU ld], ac_cv_prog_gnu_ld,
 [# I'd rather use --version here, but apparently some GNU ld's only accept -v.
 if $LD -v 2>&1 </dev/null | egrep '(GNU|with BFD)' 1>&5; then
@@ -319,7 +319,7 @@
 ])
 
 # AC_PROG_NM - find the path to a BSD-compatible name lister
-AC_DEFUN(AC_PROG_NM,
+AC_DEFUN([AC_PROG_NM],
 [AC_MSG_CHECKING([for BSD-compatible nm])
 AC_CACHE_VAL(ac_cv_path_NM,
 [if test -n "$NM"; then
@@ -353,7 +353,7 @@
 ])
 
 # AC_CHECK_LIBM - check for math library
-AC_DEFUN(AC_CHECK_LIBM,
+AC_DEFUN([AC_CHECK_LIBM],
 [AC_REQUIRE([AC_CANONICAL_HOST])dnl
 LIBM=
 case "$lt_target" in
@@ -378,7 +378,7 @@
 # '${top_builddir}/' (note the single quotes!) if your package is not
 # flat, and, if you're not using automake, define top_builddir as
 # appropriate in the Makefiles.
-AC_DEFUN(AC_LIBLTDL_CONVENIENCE, [AC_BEFORE([$0],[AC_LIBTOOL_SETUP])dnl
+AC_DEFUN([AC_LIBLTDL_CONVENIENCE], [AC_BEFORE([$0],[AC_LIBTOOL_SETUP])dnl
   case "$enable_ltdl_convenience" in
   no) AC_MSG_ERROR([this package needs a convenience libltdl]) ;;
   "") enable_ltdl_convenience=yes
@@ -397,7 +397,7 @@
 # flat, and, if you're not using automake, define top_builddir as
 # appropriate in the Makefiles.
 # In the future, this macro may have to be called after AC_PROG_LIBTOOL.
-AC_DEFUN(AC_LIBLTDL_INSTALLABLE, [AC_BEFORE([$0],[AC_LIBTOOL_SETUP])dnl
+AC_DEFUN([AC_LIBLTDL_INSTALLABLE], [AC_BEFORE([$0],[AC_LIBTOOL_SETUP])dnl
   AC_CHECK_LIB(ltdl, main,
   [test x"$enable_ltdl_install" != xyes && enable_ltdl_install=no],
   [if test x"$enable_ltdl_install" = xno; then
@@ -418,13 +418,13 @@
 ])
 
 dnl old names
-AC_DEFUN(AM_PROG_LIBTOOL, [indir([AC_PROG_LIBTOOL])])dnl
-AC_DEFUN(AM_ENABLE_SHARED, [indir([AC_ENABLE_SHARED], $@)])dnl
-AC_DEFUN(AM_ENABLE_STATIC, [indir([AC_ENABLE_STATIC], $@)])dnl
-AC_DEFUN(AM_DISABLE_SHARED, [indir([AC_DISABLE_SHARED], $@)])dnl
-AC_DEFUN(AM_DISABLE_STATIC, [indir([AC_DISABLE_STATIC], $@)])dnl
-AC_DEFUN(AM_PROG_LD, [indir([AC_PROG_LD])])dnl
-AC_DEFUN(AM_PROG_NM, [indir([AC_PROG_NM])])dnl
+AC_DEFUN([AM_PROG_LIBTOOL], [indir([AC_PROG_LIBTOOL])])dnl
+AC_DEFUN([AM_ENABLE_SHARED], [indir([AC_ENABLE_SHARED], $@)])dnl
+AC_DEFUN([AM_ENABLE_STATIC], [indir([AC_ENABLE_STATIC], $@)])dnl
+AC_DEFUN([AM_DISABLE_SHARED], [indir([AC_DISABLE_SHARED], $@)])dnl
+AC_DEFUN([AM_DISABLE_STATIC], [indir([AC_DISABLE_STATIC], $@)])dnl
+AC_DEFUN([AM_PROG_LD], [indir([AC_PROG_LD])])dnl
+AC_DEFUN([AM_PROG_NM], [indir([AC_PROG_NM])])dnl
 
 dnl This is just to silence aclocal about the macro not being used
 ifelse([AC_DISABLE_FAST_INSTALL])dnl
diff -Naur a2ps-4.13.orig/m4/malloc.m4 a2ps-4.13/m4/malloc.m4
--- a2ps-4.13.orig/m4/malloc.m4	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/m4/malloc.m4	2005-06-08 15:52:35.000000000 +0000
@@ -27,8 +27,7 @@
 	 jm_cv_func_working_malloc=no)
   ])
   if test $jm_cv_func_working_malloc = no; then
-    AC_SUBST(LIBOBJS)
-    LIBOBJS="$LIBOBJS malloc.$ac_objext"
+    AC_LIBOBJ([malloc])
     AC_DEFINE_UNQUOTED(malloc, rpl_malloc,
       [Define to rpl_malloc if the replacement function should be used.])
   fi
diff -Naur a2ps-4.13.orig/m4/obstack.m4 a2ps-4.13/m4/obstack.m4
--- a2ps-4.13.orig/m4/obstack.m4	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/m4/obstack.m4	2005-06-08 15:52:47.000000000 +0000
@@ -1,7 +1,7 @@
 dnl From Jim Meyering.
 dnl FIXME: migrate into libit.
 
-AC_DEFUN(AM_FUNC_OBSTACK,
+AC_DEFUN([AM_FUNC_OBSTACK],
 [AC_CACHE_CHECK([for obstacks], am_cv_func_obstack,
  [AC_TRY_LINK([#include "obstack.h"],
 	      [struct obstack *mem;obstack_free(mem,(char *) 0)],
@@ -10,6 +10,6 @@
  if test $am_cv_func_obstack = yes; then
    AC_DEFINE(HAVE_OBSTACK,1,[Define if libc includes obstacks])
  else
-   LIBOBJS="$LIBOBJS obstack.$ac_objext"
+   AC_LIBOBJ([obstack])
  fi
 ])
diff -Naur a2ps-4.13.orig/m4/realloc.m4 a2ps-4.13/m4/realloc.m4
--- a2ps-4.13.orig/m4/realloc.m4	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/m4/realloc.m4	2005-06-08 15:52:35.000000000 +0000
@@ -27,8 +27,7 @@
 	 jm_cv_func_working_realloc=no)
   ])
   if test $jm_cv_func_working_realloc = no; then
-    AC_SUBST(LIBOBJS)
-    LIBOBJS="$LIBOBJS realloc.$ac_objext"
+    AC_LIBOBJ([realloc])
     AC_DEFINE_UNQUOTED(realloc, rpl_realloc,
       [Define to rpl_realloc if the replacement function should be used.])
   fi
diff -Naur a2ps-4.13.orig/m4/rename.m4 a2ps-4.13/m4/rename.m4
--- a2ps-4.13.orig/m4/rename.m4	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/m4/rename.m4	2005-06-08 15:52:35.000000000 +0000
@@ -27,7 +27,7 @@
    test "$ac_cv_func_rename_works" = no; then
   AC_CHECK_FUNCS(link)
   dnl FIXME: Define MVDIR
-  LIBOBJS="$LIBOBJS rename.o"
+  AC_LIBOBJ([rename.o])
   AC_DEFINE_UNQUOTED(rename, rpl_rename,
    [Define to the replacement function if should be used.])
 fi])
diff -Naur a2ps-4.13.orig/m4/strcasecmp.m4 a2ps-4.13/m4/strcasecmp.m4
--- a2ps-4.13.orig/m4/strcasecmp.m4	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/m4/strcasecmp.m4	2005-06-08 15:52:35.000000000 +0000
@@ -13,7 +13,7 @@
  if test $ac_cv_func_strcasecmp = no; then
    AC_CHECK_FUNCS(stricmp)
    if test $ac_cv_func_stricmp = no; then
-     LIBOBJS="$LIBOBJS strcasecmp.o"
+     AC_LIBOBJ([strcasecmp])
    else
      AC_DEFINE_UNQUOTED(strcasecmp, stricmp,
 	  [Define to the replacement function if should be used.])
@@ -25,7 +25,7 @@
  if test $ac_cv_func_strncasecmp = no; then
    AC_CHECK_FUNCS(strnicmp)
    if test $ac_cv_func_strnicmp = no; then
-     LIBOBJS="$LIBOBJS strncasecmp.o"
+     AC_LIBOBJ([strncasecmp])
    else
      AC_DEFINE_UNQUOTED(strncasecmp, strnicmp,
 	  [Define to the replacement function if should be used.])
diff -Naur a2ps-4.13.orig/m4/strftim.m4 a2ps-4.13/m4/strftim.m4
--- a2ps-4.13.orig/m4/strftim.m4	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/m4/strftim.m4	2005-06-08 15:52:35.000000000 +0000
@@ -12,7 +12,7 @@
   AC_CHECK_HEADERS(limits.h)
   AC_FUNC_MKTIME
   if test $ac_cv_func_strftime = no; then
-    LIBOBJS="$LIBOBJS strftime.o"
+    AC_LIBOBJ([strftime])
     if test $ac_cv_func_working_mktime = no; then
        AC_DEFINE_UNQUOTED(mktime, rpl_mktime,
 	[Define to rpl_mktime if the replacement function should be used.])
diff -Naur a2ps-4.13.orig/ogonkify/configure.in a2ps-4.13/ogonkify/configure.in
--- a2ps-4.13.orig/ogonkify/configure.in	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/ogonkify/configure.in	2005-06-08 15:52:35.000000000 +0000
@@ -45,7 +45,7 @@
  x/*) # User specified a version of Perl to use by absolute path.
      AC_SUBST(PERL);;
  x)   # She didn't specify at all.
-      AC_PATH_PROG(PERL, 'perl perl5', no) ;;
+      AC_PATH_PROG(PERL, [perl perl5], no) ;;
  x*)  # She specified the name, but we want the path.
      AC_PATH_PROG(PERL, $PERL, no) ;;
 esac
diff -Naur a2ps-4.13.orig/src/Makefile.in a2ps-4.13/src/Makefile.in
--- a2ps-4.13.orig/src/Makefile.in	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/src/Makefile.in	2005-06-08 15:52:29.000000000 +0000
@@ -100,7 +100,7 @@
 INTLDEPS = @INTLDEPS@
 INTLLIBS = @INTLLIBS@
 INTLOBJS = @INTLOBJS@
-LEX = @LEX@
+LEX = @LEX@ -CFe
 LEX_OUTPUT_ROOT = @LEX_OUTPUT_ROOT@
 LIBOBJS = @LIBOBJS@
 LIBTOOL = @LIBTOOL@
@@ -154,7 +154,7 @@
 BUILT_SOURCES = parsessh.h
 
 YFLAGS = -dtv
-LEX = @LEX@
+LEX = @LEX@ -CFe
 # I don't really understand why I have to put srcdir here, but
 # it is needed for yacc and lex files (seems related to #line, but
 # I really don't understand why)...
diff -Naur a2ps-4.13.orig/src/lexps.l a2ps-4.13/src/lexps.l
--- a2ps-4.13.orig/src/lexps.l	2005-06-08 15:52:16.000000000 +0000
+++ a2ps-4.13/src/lexps.l	2005-06-08 15:52:29.000000000 +0000
@@ -35,7 +35,7 @@
 void pslex_initialize PARAMS ((void));
 %}
 
-%option yylineno
+/* %option yylineno */
 %option prefix="ps"
 %option outfile="lex.yy.c"
 
