diff -Naur a2ps-4.13/configure a2ps-4.13-p/configure
--- a2ps-4.13/configure	2000-03-08 18:43:57.000000000 +0100
+++ a2ps-4.13-p/configure	2007-03-18 04:38:47.000000000 +0100
@@ -2672,7 +2672,8 @@
 LIBTOOL_DEPS="$ac_aux_dir/ltconfig $ac_aux_dir/ltmain.sh"
 
 # Always use our own libtool.
-LIBTOOL='$(SHELL) $(top_builddir)/libtool'
+#LIBTOOL='$(SHELL) $(top_builddir)/libtool'
+LIBTOOL='/usr/bin/libtool'
 
 # Redirect the config.log output again, so that the ltconfig log is not
 # clobbered by the next message.
