diff -Naurp dmidecode-2.8/Makefile dmidecode-2.8-p/Makefile
--- dmidecode-2.8/Makefile	2005-10-03 20:38:20.000000000 +0200
+++ dmidecode-2.8-p/Makefile	2006-11-21 02:02:25.000000000 +0100
@@ -9,7 +9,7 @@
 #
 
 CC      = gcc
-CFLAGS  = -W -Wall -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-qual \
+CFLAGS  += -W -Wall -Wshadow -Wstrict-prototypes -Wpointer-arith -Wcast-qual \
           -Wcast-align -Wwrite-strings -Wmissing-prototypes -Winline -Wundef
 #CFLAGS += -DBIGENDIAN
 #CFLAGS += -DALIGNMENT_WORKAROUND
@@ -19,10 +19,10 @@ CFLAGS += -O2
 #CFLAGS += -g
 
 # Pass linker flags here
-LDFLAGS =
+LDFLAGS += 
 
 DESTDIR =
-prefix  = /usr/local
+prefix  = /usr
 sbindir = $(prefix)/sbin
 mandir  = $(prefix)/share/man
 man8dir = $(mandir)/man8
