diff -Naur digikam-8.1.0/core/app/DigikamCoreTarget.cmake digikam-8.1.0.new/core/app/DigikamCoreTarget.cmake
--- digikam-8.1.0/core/app/DigikamCoreTarget.cmake	2023-07-05 17:20:49.000000000 +0200
+++ digikam-8.1.0.new/core/app/DigikamCoreTarget.cmake	2023-11-13 13:43:20.935850157 +0100
@@ -151,7 +151,6 @@
                       ${TIFF_LIBRARIES}
                       PNG::PNG
                       ${JPEG_LIBRARIES}
-                      exiv2lib
 
                       ${OPENMP_LDFLAGS}
 
@@ -164,6 +163,22 @@
                       opencv_flann
 )
 
+if(LibExiv2_FOUND)
+
+    target_link_libraries(digikamcore
+                          PRIVATE
+                          ${LibExiv2_LIBRARIES}
+    )
+
+else()
+
+    target_link_libraries(digikamcore
+                          PRIVATE
+                          exiv2lib
+    )
+
+endif()
+
 if(Qt6_FOUND)
     target_link_libraries(digikamcore
                           PRIVATE
diff -Naur digikam-8.1.0/core/cmake/rules/RulesLibExiv2.cmake digikam-8.1.0.new/core/cmake/rules/RulesLibExiv2.cmake
--- digikam-8.1.0/core/cmake/rules/RulesLibExiv2.cmake	2023-07-05 17:20:49.000000000 +0200
+++ digikam-8.1.0.new/core/cmake/rules/RulesLibExiv2.cmake	2023-11-13 13:43:20.935850157 +0100
@@ -4,7 +4,17 @@
 # SPDX-License-Identifier: BSD-3-Clause
 #
 
-find_package(exiv2 REQUIRED)
+find_package(LibExiv2 OPTIONAL_COMPONENTS)
+
+if(LibExiv2_FOUND)
+
+    find_package(LibExiv2 REQUIRED)
+
+else()
+
+    find_package(exiv2 REQUIRED)
+
+endif()
 
 set_package_properties("exiv2"     PROPERTIES
                        DESCRIPTION "Required to build digiKam"
@@ -13,18 +23,40 @@
                        PURPOSE     "Library to manage image metadata"
 )
 
-if("${exiv2_VERSION}" VERSION_LESS ${EXIV2_MIN_VERSION})
+if(LibExiv2_FOUND)
 
-    message(FATAL_ERROR "Exiv2 version is too old (${exiv2_VERSION})! Minimal version required:${EXIV2_MIN_VERSION}.")
+    if("${LibExiv2_VERSION}" VERSION_LESS ${EXIV2_MIN_VERSION})
 
-endif()
+        message(FATAL_ERROR "Exiv2 version is too old (${LibExiv2_VERSION})! Minimal version required:${EXIV2_MIN_VERSION}.")
+
+    endif()
+
+    if("${LibExiv2_VERSION}" VERSION_LESS "0.27.99")
 
-if("${exiv2_VERSION}" VERSION_LESS "0.27.99")
+        set(EXIV2_CXX_STANDARD 11)
 
-    set(EXIV2_CXX_STANDARD 11)
+    else()
+
+        set(EXIV2_CXX_STANDARD 17)
+
+    endif()
 
 else()
 
-    set(EXIV2_CXX_STANDARD 17)
+    if("${exiv2_VERSION}" VERSION_LESS ${EXIV2_MIN_VERSION})
+
+        message(FATAL_ERROR "Exiv2 version is too old (${exiv2_VERSION})! Minimal version required:${EXIV2_MIN_VERSION}.")
+
+    endif()
+
+    if("${exiv2_VERSION}" VERSION_LESS "0.27.99")
+
+        set(EXIV2_CXX_STANDARD 11)
+
+    else()
+
+        set(EXIV2_CXX_STANDARD 17)
+
+    endif()
 
 endif()
diff -Naur digikam-8.1.0/core/CMakeLists.txt digikam-8.1.0.new/core/CMakeLists.txt
--- digikam-8.1.0/core/CMakeLists.txt	2023-07-05 17:20:49.000000000 +0200
+++ digikam-8.1.0.new/core/CMakeLists.txt	2023-11-13 13:43:20.933850141 +0100
@@ -255,7 +255,17 @@
 # ==============================================================================
 
 PRINT_LIBRARY_STATUS("libboostgraph" "https://github.com/boostorg/boost"                 "(version >= 1.43.0)"                Boost_FOUND)
-PRINT_LIBRARY_STATUS("libexiv2"      "https://github.com/Exiv2/exiv2"                    "(version >= ${EXIV2_MIN_VERSION}"   exiv2_FOUND)
+
+if(LibExiv2_FOUND)
+
+    PRINT_LIBRARY_STATUS("libexiv2"  "https://github.com/Exiv2/exiv2"                    "(version >= ${EXIV2_MIN_VERSION}"   LibExiv2_FOUND)
+
+else()
+
+    PRINT_LIBRARY_STATUS("libexiv2"  "https://github.com/Exiv2/exiv2"                    "(version >= ${EXIV2_MIN_VERSION}"   exiv2_FOUND)
+
+endif()
+
 PRINT_LIBRARY_STATUS("libexpat"      "https://libexpat.github.io"                        "(version >= 2.0.0)"                 EXPAT_FOUND)
 PRINT_LIBRARY_STATUS("libjpeg"       "https://github.com/libjpeg-turbo/libjpeg-turbo"    "(version >= 6b)"                    JPEG_FOUND)
 PRINT_LIBRARY_STATUS("libkde"        "https://invent.kde.org/frameworks/"                "(version >= ${KF5_MIN_VERSION})"    KF5_FOUND)
@@ -306,7 +316,7 @@
 # ==============================================================================
 
 if(Boost_FOUND                          AND
-   exiv2_FOUND                          AND
+   (LibExiv2_FOUND OR exiv2_FOUND)      AND
    EXPAT_FOUND                          AND
    JPEG_FOUND                           AND
    KF5_FOUND                            AND
diff -Naur digikam-8.1.0/core/libs/metadataengine/CMakeLists.txt digikam-8.1.0.new/core/libs/metadataengine/CMakeLists.txt
--- digikam-8.1.0/core/libs/metadataengine/CMakeLists.txt	2023-07-05 17:20:49.000000000 +0200
+++ digikam-8.1.0.new/core/libs/metadataengine/CMakeLists.txt	2023-11-13 13:45:21.200848722 +0100
@@ -18,7 +18,7 @@
 
 # Exiv2 version < 0.28 use auto_ptr instead unique_ptr which is deprecated.
 
-if("${exiv2_VERSION}" VERSION_LESS "0.27.99")
+if("${LibExiv2_VERSION}" VERSION_LESS "0.27.99")
 
     DISABLE_GCC_COMPILER_WARNINGS("5.99.99" "-Wno-deprecated")
 
@@ -112,9 +112,19 @@
     $<TARGET_PROPERTY:KF5::ConfigCore,INTERFACE_INCLUDE_DIRECTORIES>
     $<TARGET_PROPERTY:KF5::I18n,INTERFACE_INCLUDE_DIRECTORIES>
 
-    $<TARGET_PROPERTY:exiv2lib,INTERFACE_INCLUDE_DIRECTORIES>
 )
 
+if(LibExiv2_FOUND)
+
+    include_directories($<TARGET_PROPERTY:LibExiv2_INCLUDE_DIRS>)
+
+else()
+
+    include_directories($<TARGET_PROPERTY:exiv2lib,INTERFACE_INCLUDE_DIRECTORIES>)
+
+endif()
+
+
 # Used by digikamcore
 add_library(core_dmetadata_obj OBJECT ${libdmetadata_SRCS})
 
