--- digikam-5.9.0/core/libs/dmetadata/metaengine.cpp	2018-03-19 10:41:58.000000000 +0100
+++ digikam-5.9.0.new/core/libs/dmetadata/metaengine.cpp	2018-12-28 12:37:56.380849325 +0100
@@ -152,10 +152,13 @@
 
 QString MetaEngine::Exiv2Version()
 {
-    // Since 0.14.0 release, we can extract run-time version of Exiv2.
-    // else we return make version.
 
-    return QString::fromLatin1(Exiv2::version());
+#if EXIV2_TEST_VERSION(0,27,0)
+    return QLatin1String(Exiv2::versionString().c_str());
+#else
+     return QLatin1String(Exiv2::version());
+#endif
+
 }
 
 QString MetaEngine::sidecarFilePathForFile(const QString& path)
