diff -Naur knb.orig/scripts/knbchk.sh knb/scripts/knbchk.sh
--- knb.orig/scripts/knbchk.sh	2006-10-01 14:31:02.000000000 +0200
+++ knb/scripts/knbchk.sh	2007-11-24 18:51:37.000000000 +0100
@@ -47,11 +47,8 @@
     exit 0
 fi
 
-# searching for pid readed from pid file in process list  
-TMPPID=`ps ux -U $USER | grep $MYPID | head -n 1 | awk '{ print $2 }'` 
-
 # is bot allready running?
-if [ ! "$TMPPID" = "$MYPID" ]; then
+if [ ! -d "/proc/$MYPID" ]; then
     # no, so we start it now (and removing pid file) 
     echo "-+- Starting Knb bot.."
     rm -fr $PIDFILE
