diff -Naur sysdig-0.30.2/CMakeLists.txt sysdig-0.30.2.new/CMakeLists.txt
--- sysdig-0.30.2/CMakeLists.txt	2022-11-14 11:37:40.000000000 +0100
+++ sysdig-0.30.2.new/CMakeLists.txt	2023-02-15 15:04:24.013151223 +0100
@@ -86,7 +86,7 @@
 	endif()
 
 	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${CMAKE_COMMON_FLAGS}")
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_COMMON_FLAGS} -std=c++0x")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_COMMON_FLAGS} -std=c++17")
 
 	set(CMAKE_C_FLAGS_DEBUG "${SYSDIG_DEBUG_FLAGS}")
 	set(CMAKE_CXX_FLAGS_DEBUG "${SYSDIG_DEBUG_FLAGS}")
