diff -Naur sysdig-0.26.0/userspace/libsinsp/CMakeLists.txt sysdig-0.26.0-link/userspace/libsinsp/CMakeLists.txt
--- sysdig-0.26.0/userspace/libsinsp/CMakeLists.txt	2019-05-22 18:31:53.000000000 +0200
+++ sysdig-0.26.0-link/userspace/libsinsp/CMakeLists.txt	2019-05-24 05:53:10.862027626 +0200
@@ -155,16 +155,14 @@
 	if(USE_BUNDLED_OPENSSL)
 		add_dependencies(sinsp openssl)
 	endif()
-	if(USE_BUNDLED_CURL)
-		add_dependencies(sinsp curl)
-		target_link_libraries(sinsp
-			"${CURL_LIBRARIES}")
-	endif()
-	if(USE_BUNDLED_TBB)
-		add_dependencies(sinsp tbb)
-		target_link_libraries(sinsp
-			"${TBB_LIB}")
-	endif()
+
+	add_dependencies(sinsp curl)
+	target_link_libraries(sinsp
+		"${CURL_LIBRARIES}")
+
+	add_dependencies(sinsp tbb)
+	target_link_libraries(sinsp
+		"${TBB_LIB}")
 
 	if(NOT APPLE)
 		add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/cri.grpc.pb.cc
