diff -Naur sysdig-0.26.0/userspace/libsinsp/cursesui.cpp sysdig-0.26.0-pt/userspace/libsinsp/cursesui.cpp
--- sysdig-0.26.0/userspace/libsinsp/cursesui.cpp	2019-05-22 18:31:53.000000000 +0200
+++ sysdig-0.26.0-pt/userspace/libsinsp/cursesui.cpp	2019-05-24 06:28:58.780393239 +0200
@@ -2292,7 +2292,7 @@
 					{
 						f = compiler.compile();
 					}
-					catch(sinsp_exception e)
+					catch(sinsp_exception const& e)
 					{
 						//
 						// Backup the cursor position
diff -Naur sysdig-0.26.0/userspace/sysdig/csysdig.cpp sysdig-0.26.0-pt/userspace/sysdig/csysdig.cpp
--- sysdig-0.26.0/userspace/sysdig/csysdig.cpp	2019-05-22 18:31:53.000000000 +0200
+++ sysdig-0.26.0-pt/userspace/sysdig/csysdig.cpp	2019-05-24 06:29:45.896694107 +0200
@@ -830,7 +830,7 @@
 				{
 					inspector->open("");
 				}
-				catch(sinsp_exception e)
+				catch(sinsp_exception const& e)
 				{
 					open_success = false;
 				}
diff -Naur sysdig-0.26.0/userspace/sysdig/sysdig.cpp sysdig-0.26.0-pt/userspace/sysdig/sysdig.cpp
--- sysdig-0.26.0/userspace/sysdig/sysdig.cpp	2019-05-22 18:31:53.000000000 +0200
+++ sysdig-0.26.0-pt/userspace/sysdig/sysdig.cpp	2019-05-24 06:30:25.583947349 +0200
@@ -1453,7 +1453,7 @@
 				{
 					inspector->open("");
 				}
-				catch(sinsp_exception e)
+				catch(sinsp_exception const& e)
 				{
 					open_success = false;
 				}
