diff -urN system-tools-backends-1.9.1/Utils/Platform.pm system-tools-backends-1.9.1.new/Utils/Platform.pm
--- system-tools-backends-1.9.1/Utils/Platform.pm	2006-01-02 15:32:34.000000000 +0000
+++ system-tools-backends-1.9.1.new/Utils/Platform.pm	2006-06-18 12:18:09.000000000 +0100
@@ -381,18 +381,25 @@
 
 sub check_slackware
 {
-  open RELEASE, "$gst_prefix/etc/slackware-version" or return -1;
-  while (<RELEASE>)
+  if (stat("/etc/frugalware-release"))
   {
-    chomp;
-    if (/^Slackware ([0-9.]+)/)
+    return "slackware-10.0.0";
+  }
+  else
+  {
+    open RELEASE, "$gst_prefix/etc/slackware-version" or return -1;
+    while (<RELEASE>)
     {
-        close RELEASE;
-        return "slackware-$1";
+      chomp;
+      if (/^Slackware ([0-9.]+)/)
+      {
+          close RELEASE;
+          return "slackware-$1";
+      }
     }
+    close RELEASE;
+    return -1;
   }
-  close RELEASE;
-  return -1;
 }
 
 sub check_gentoo
