--- regex-compat-0.95.2.1/regex-compat.cabal	2001-09-09 03:46:40.000000000 +0200
+++ regex-compat-0.95.2.1.new/regex-compat.cabal	2023-07-24 10:45:57.461332589 +0200
@@ -52,7 +52,7 @@
 library
   exposed-modules: Text.Regex
 
-  build-depends: base        >= 4.3 && < 4.16
+  build-depends: base        >= 4.3 && < 5
                , regex-base  == 0.94.*
                , regex-posix == 0.96.*
                , array       >= 0.3 && < 0.6
