--- hslua-aeson-2.3.0.1/hslua-aeson.cabal	2001-09-09 03:46:40.000000000 +0200
+++ hslua-aeson-2.3.0.1.new/hslua-aeson.cabal	2023-07-21 17:04:04.014767815 +0200
@@ -29,7 +29,7 @@
 common common-options
   default-language:    Haskell2010
   build-depends:       base                 >= 4.11   && < 5
-                     , aeson                >= 1.5    && < 2.2
+                     , aeson                >= 1.5    && < 2.3
                      , bytestring           >= 0.10.2 && < 0.12
                      , containers           >= 0.5.9  && < 0.7
                      , hashable             >= 1.2    && < 1.5
