diff -Naur docbook2X-0.8.3.orig/perl/db2x_texixml.pl docbook2X-0.8.3/perl/db2x_texixml.pl
--- docbook2X-0.8.3.orig/perl/db2x_texixml.pl	2004-07-16 00:03:26.000000000 +0200
+++ docbook2X-0.8.3/perl/db2x_texixml.pl	2007-12-18 10:35:37.000000000 +0100
@@ -60,7 +60,7 @@
 use Getopt::Long;
 Getopt::Long::Configure('bundling');
 my $cmdoptions = {
-    'encoding' => 'us-ascii',
+    'encoding' => 'utf-8',
     'list-files' => 0,
     'to-stdout' => 0,
     'info' => 0,
