diff -Naur extra-cmake-modules-5.27.0/kde-modules/KDEInstallDirs.cmake extra-cmake-modules-5.27.0-p/kde-modules/KDEInstallDirs.cmake
--- extra-cmake-modules-5.27.0/kde-modules/KDEInstallDirs.cmake	2016-09-15 14:44:43.000000000 +0200
+++ extra-cmake-modules-5.27.0-p/kde-modules/KDEInstallDirs.cmake	2016-10-22 21:40:01.635914907 +0200
@@ -428,7 +428,7 @@
     _define_relative(LIBEXECDIR LIBDIR "libexec"
         "executables for internal use by programs and libraries"
         LIBEXEC_INSTALL_DIR)
-    _define_non_cache(LIBEXECDIR_KF5 "${CMAKE_INSTALL_LIBEXECDIR}/kf5")
+    _define_non_cache(LIBEXECDIR_KF5 "${CMAKE_INSTALL_LIBEXECDIR}")
 endif()
 if(NOT KDE_INSTALL_DIRS_NO_DEPRECATED)
     set(KF5_LIBEXEC_INSTALL_DIR "${CMAKE_INSTALL_LIBEXECDIR_KF5}")
