--- extra-cmake-modules-5.77.0/find-modules/FindPythonModuleGeneration.cmake	2020-12-04 21:30:12.000000000 +0100
+++ extra-cmake-modules-5.77.0.new/find-modules/FindPythonModuleGeneration.cmake	2020-12-14 17:21:45.076509170 +0100
@@ -258,7 +258,7 @@
   message(STATUS "Found ${libclang_LIBRARY}")
 endif()
 
-foreach (pyversion "2" "3")
+foreach (pyversion "3")
   set(GPB_PYTHON_COMMAND ${GPB_PYTHON${pyversion}_COMMAND})
   if (NOT GPB_PYTHON_COMMAND)
     continue()
