diff -Naur uwsgi-2.0.13.1/uwsgiconfig.py uwsgi-2.0.13.1-p/uwsgiconfig.py
--- uwsgi-2.0.13.1/uwsgiconfig.py	2016-05-12 19:37:49.000000000 +0200
+++ uwsgi-2.0.13.1-p/uwsgiconfig.py	2016-09-29 04:52:06.282097133 +0200
@@ -672,7 +672,7 @@
             self.include_path += os.environ['UWSGI_INCLUDES'].split(',')
 
 
-        self.cflags = ['-O2', '-I.', '-Wall', '-Werror', '-D_LARGEFILE_SOURCE', '-D_FILE_OFFSET_BITS=64'] + os.environ.get("CFLAGS", "").split() + self.get('cflags','').split()
+        self.cflags = ['-I.', '-Wall', '-D_LARGEFILE_SOURCE', '-D_FILE_OFFSET_BITS=64'] + os.environ.get("CFLAGS", "").split() + self.get('cflags','').split()
 
         report['kernel'] = uwsgi_os
 
