--- uwsgi-2.0.14/plugins/mono/uwsgiplugin.py	2016-10-03 10:16:59.000000000 +0200
+++ uwsgi-2.0.14.new/plugins/mono/uwsgiplugin.py	2017-03-01 15:42:22.000000000 +0100
@@ -2,6 +2,7 @@
 NAME='mono'
 
 CFLAGS = os.popen('pkg-config --cflags mono-2').read().rstrip().split()
+CFLAGS.extend(["-I/usr/include/glib-2.0", "-I/usr/include", "-I/usr/lib/glib-2.0/include"])
 LDFLAGS = []
 LIBS = os.popen('pkg-config --libs mono-2').read().rstrip().split() 
 GCC_LIST = ['mono_plugin']
