diff -Naur uwsgi-2.0.16/plugins/lua/uwsgiplugin.py uwsgi-2.0.16-p/plugins/lua/uwsgiplugin.py
--- uwsgi-2.0.16/plugins/lua/uwsgiplugin.py	2018-02-10 11:00:57.000000000 +0100
+++ uwsgi-2.0.16-p/plugins/lua/uwsgiplugin.py	2018-02-10 22:43:53.367514109 +0100
@@ -3,7 +3,7 @@
 LUAINC = os.environ.get('UWSGICONFIG_LUAINC')
 LUALIB = os.environ.get('UWSGICONFIG_LUALIB')
 LUALIBPATH = os.environ.get('UWSGICONFIG_LUALIBPATH')
-LUAPC = os.environ.get('UWSGICONFIG_LUAPC', 'lua5.1')
+LUAPC = os.environ.get('UWSGICONFIG_LUAPC', 'lua')
 
 # we LUAINC/LUALIB/LUALIBPATH override the LUAPC for backwards compat
 if LUAINC:
@@ -12,7 +12,7 @@
 	try:
 		CFLAGS = os.popen('pkg-config --cflags %s' % LUAPC).read().rstrip().split()
 	except:
-		CFLAGS = ['-I/usr/include/lua5.1']
+		CFLAGS = ['-I/usr/include']
 
 if LUALIB:
 	LIBS = ['-l%s' % LUALIB]
@@ -20,7 +20,7 @@
 	try:
 		LIBS = os.popen('pkg-config --libs %s' % LUAPC).read().rstrip().split() 
 	except:
-		LIBS = ['-llua5.1']
+		LIBS = ['-llua']
 
 if LUALIBPATH:
 	LDFLAGS = ['-L%s' % LUALIBPATH]
