diff -Naur Vc-1.3.0/cmake/OptimizeForArchitecture.cmake Vc-1.3.0-p/cmake/OptimizeForArchitecture.cmake
--- Vc-1.3.0/cmake/OptimizeForArchitecture.cmake	2016-10-27 09:05:02.000000000 +0200
+++ Vc-1.3.0-p/cmake/OptimizeForArchitecture.cmake	2016-11-16 17:10:17.668601666 +0100
@@ -178,7 +178,7 @@
 endmacro()
 
 macro(OptimizeForArchitecture)
-   set(TARGET_ARCHITECTURE "auto" CACHE STRING "CPU architecture to optimize for. \
+   set(TARGET_ARCHITECTURE "none" CACHE STRING "CPU architecture to optimize for. \
 Using an incorrect setting here can result in crashes of the resulting binary because of invalid instructions used. \
 Setting the value to \"auto\" will try to optimize for the architecture where cmake is called. \
 Other supported values are: \"none\", \"generic\", \"core\", \"merom\" (65nm Core2), \
