diff -Naur pacman-2.9.5.orig/scripts/gensync pacman-2.9.5/scripts/gensync
--- pacman-2.9.5.orig/scripts/gensync	2005-05-20 21:16:19.000000000 +0200
+++ pacman-2.9.5/scripts/gensync	2005-05-20 21:18:03.000000000 +0200
@@ -75,6 +75,7 @@
 	cat $1 |grep -v "up2date=\`" >/tmp/.FrugalBuild
 	source /tmp/.FrugalBuild || return 1
 	rm /tmp/.FrugalBuild
+	[ "$nobuild" ] && return 0
 	cd $gstmpdir
 	mkdir $pkgname-$pkgver-$pkgrel || return 1
 	cd $pkgname-$pkgver-$pkgrel
diff -Naur pacman-2.9.5.orig/scripts/updatesync pacman-2.9.5/scripts/updatesync
--- pacman-2.9.5.orig/scripts/updatesync	2005-05-20 22:09:44.000000000 +0200
+++ pacman-2.9.5/scripts/updatesync	2005-05-20 22:10:36.000000000 +0200
@@ -74,6 +74,7 @@
 	unset pkgname pkgver pkgrel pkgdesc force
 	unset groups replaces provides depends conflicts
 	source $1 || return 1
+	[ "$nobuild" ] && return 0
 	cd $ustmpdir
 	mkdir $pkgname-$pkgver-$pkgrel || return 1
 	cd $pkgname-$pkgver-$pkgrel
diff -Naur pacman-2.9.5.orig/doc/makepkg.8.in pacman-2.9.5/doc/makepkg.8.in
--- pacman-2.9.5.orig/doc/makepkg.8.in	2004-08-04 07:57:17.000000000 +0200
+++ pacman-2.9.5/doc/makepkg.8.in	2005-05-20 21:28:51.000000000 +0200
@@ -241,6 +241,12 @@
 if its an older version.
 
 .TP
+.B nobuild
+If this directive set, gensync will ignore this package, so users must build
+these packages on their machines, they will not be able to install them with
+\fBpacman -S\fP. Useful for closed-source, but freeware programs.
+
+.TP
 .B url
 This field contains an optional URL that is associated with the piece of software
 being packaged.  This is typically the project's website.
