diff -Naur pacman-2.9.7.orig/src/pacman.c pacman-2.9.7/src/pacman.c
--- pacman-2.9.7.orig/src/pacman.c	2005-11-11 01:45:16.569573000 +0100
+++ pacman-2.9.7/src/pacman.c	2005-11-11 01:48:40.498317750 +0100
@@ -2003,20 +2003,24 @@
 
 			if(nb) {
 				char *temp;
+				int fd;
 				char *md5_local, *md5_pkg;
 				char *sha1_local, *sha1_pkg;
 
-				md5_local = MDFile(expath);
-				sha1_local = SHAFile(expath);
 				/* extract the package's version to a temporary file and md5 or sha1 it */
 				temp = strdup("/tmp/pacman_XXXXXX");
-				mkstemp(temp);
+				fd = mkstemp(temp);
 				archive_entry_set_pathname (entry, temp);
 				if (archive_read_extract (archive, entry, ARCHIVE_EXTRACT_FLAGS) != ARCHIVE_OK) {
 					logaction(stderr, "could not extract %s: %s", pathname, strerror(errno));
 					errors++;
+					unlink(temp);
+					FREE(temp);
+					close(fd);
 					continue;
 				}
+				md5_local = MDFile(expath);
+				sha1_local = SHAFile(expath);
 				md5_pkg = MDFile(temp);
 				sha1_pkg = SHAFile(temp);
 				/* append the new md5 / sha1 hash to it's respective entry in info->backup
@@ -2122,6 +2126,7 @@
 				FREE(sha1_orig);
 				unlink(temp);
 				FREE(temp);
+				close(fd);
 			} else {
 				if(!notouch) {
 					/*vprint("  %s\n", expath);*/
