diff -Naur pacman-2.9.7.orig/src/pacman.c pacman-2.9.7/src/pacman.c
--- pacman-2.9.7.orig/src/pacman.c	2005-11-30 00:34:04.348095750 +0100
+++ pacman-2.9.7/src/pacman.c	2005-11-30 00:34:26.649489500 +0100
@@ -127,7 +127,22 @@
 	if (percent > 100) { return (1); }
 	hash = percent/6.25;
 
-	printf("(%d/%d) %s", remain, howmany, pkgname);
+	if (howmany < 10) {
+	    printf("(%d/%d) %s", remain, howmany, pkgname);
+	} else if ((howmany > 10) && (remain < 10)) {
+	    printf("( %d/%d) %s", remain, howmany, pkgname);
+	} else if ((howmany > 10) && (remain > 10)) {
+	    printf("(%d/%d) %s", remain, howmany, pkgname);
+	} else if ((howmany > 100) && (remain < 10)) {
+	    printf("(  %d/%d) %s", remain, howmany, pkgname);
+	} else if ((howmany > 100) && (remain < 100)) {
+	    printf("( %d/%d) %s", remain, howmany, pkgname);
+	} else if ((howmany > 100) && (remain < 100)) {
+	    printf("(%d/%d) %s", remain, howmany, pkgname);
+	} else {
+	    printf("(%d/%d) %s", remain, howmany, pkgname);
+	}
+
 	if (strlen(pkgname)<35)
 	    for (i=35-strlen(pkgname)-1; i>0; i--)
 		printf(" ");
