diff -BurN pacman-2.9.7/scripts/makepkg pacman-2.9.7-patched/scripts/makepkg
--- pacman-2.9.7/scripts/makepkg	2005-11-03 01:27:22.000000000 +0100
+++ pacman-2.9.7-patched/scripts/makepkg	2005-11-03 01:26:41.000000000 +0100
@@ -28,6 +28,11 @@
 # Frugalware extensions to makepkg
 [ -e /usr/lib/frugalware/fwmakepkg ] && . /usr/lib/frugalware/fwmakepkg
 
+## repoman.conf for $fst_root
+[ -f /etc/repoman.conf ] && source /etc/repoman.conf
+
+## why is abs.conf still here ? we dont need it and with this patch I broke the $ABSROOT support -- crazy -- 
+
 # source Arch's abs.conf if it's present
 [ -f /etc/abs/abs.conf ] && source /etc/abs/abs.conf
 
@@ -152,17 +157,17 @@
 			# TODO: check deps again to make sure they were resolved
 		elif [ "$DEP_SRC" = "1" ]; then
 			# install missing deps by building them from source.
-			# we look for each package name in $ABSROOT and build it.
-			if [ "$ABSROOT" = "" ]; then
-				error "The ABSROOT environment variable is not defined."
+			# we look for each package name in $FST_ROOT and build it.
+			if [ "$fst_root" = "" ]; then
+				error "The FSTROOT environment variable is not defined."
 				exit 1
 			fi
 			# TODO: handle version comparators (eg, glibc>=2.2.5)
 			msg "Building missing dependencies..."
 			for dep in $deplist; do
-				candidates=`find $ABSROOT -type d -name "$dep"`
+				candidates=`find $fst_root -type d -name "$dep"`
 				if [ "$candidates" = "" ]; then
-					error "Could not find \"$dep\" under $ABSROOT"
+					error "Could not find \"$dep\" under $fst_root"
 					exit 1
 				fi
 				success=0
