diff -Naur pacman-2.9.5.orig/scripts/makepkg pacman-2.9.5/scripts/makepkg
--- pacman-2.9.5.orig/scripts/makepkg	2004-12-19 03:09:59.000000000 +0100
+++ pacman-2.9.5/scripts/makepkg	2005-05-23 14:12:10.000000000 +0200
@@ -656,6 +656,9 @@
 done
 
 # check for an install script
+if [ "$install" == "" ]; then
+	[ -e $startdir/$pkgname.install ] && install=$pkgname.install
+fi
 if [ "$install" != "" ]; then
 	msg "Copying install script..."
 	cp $startdir/$install $startdir/pkg/.INSTALL
