diff -Naur pacman-2.9.99.23.orig/lib/libalpm/add.c pacman-2.9.99.23/lib/libalpm/add.c
--- pacman-2.9.99.23.orig/lib/libalpm/add.c	2006-03-17 19:13:26.000000000 +0100
+++ pacman-2.9.99.23/lib/libalpm/add.c	2006-03-17 21:17:07.000000000 +0100
@@ -441,12 +441,12 @@
 					!strcmp(pathname, ".CHANGELOG")) {
 					if(!strcmp(pathname, ".CHANGELOG")) {
 						/* the changelog goes inside the db */
-						snprintf(expath, PATH_MAX, "%s/%s/%s-%s/changelog", db->path,
-							db->treename, info->name, info->version);
+						snprintf(expath, PATH_MAX, "%s/%s-%s/changelog", db->path,
+							info->name, info->version);
 					} else {
 						/* the install script goes inside the db */
-						snprintf(expath, PATH_MAX, "%s/%s/%s-%s/install", db->path,
-							db->treename, info->name, info->version);
+						snprintf(expath, PATH_MAX, "%s/%s-%s/install", db->path,
+							info->name, info->version);
 					}
 				} else {
 					/* build the new pathname relative to handle->root */
