--- pacman-2.9.7.orig/scripts/makepkg.orig	2005-12-09 15:25:52.000000000 +0100
+++ pacman-2.9.7.orig/scripts/makepkg	2005-12-09 15:26:10.000000000 +0100
@@ -146,14 +146,14 @@
 				unset FAKEROOTKEY
 			fi
 			sudo pacman -D $deplist
-			if [ "$INFAKEROOT" = "1" ]; then
-				export FAKEROOTKEY=$FAKEROOTKEY2
-				unset FAKEROOTKEY2
-			fi
 			if [ "$?" = "127" ]; then
 				error "Failed to install missing dependencies."
 				exit 1
 			fi
+			if [ "$INFAKEROOT" = "1" ]; then
+				export FAKEROOTKEY=$FAKEROOTKEY2
+				unset FAKEROOTKEY2
+			fi
 			# TODO: check deps again to make sure they were resolved
 		elif [ "$DEP_SRC" = "1" ]; then
 			# install missing deps by building them from source.
