diff -Naur pacman-2.8.1.orig/doc/makepkg.8.in pacman-2.8.1/doc/makepkg.8.in
--- pacman-2.8.1.orig/scripts/makepkg	2005-05-26 10:05:02.000000000 +0200
+++ pacman-2.8.1/scripts/makepkg	2005-05-26 10:27:09.000000000 +0200
@@ -195,6 +195,7 @@
 	echo "  -p <buildscript> Use an alternate build script (instead of FrugalBuild)"
 	echo "  -r, --rmdeps     Remove installed dependencies after a successful build"
 	echo "  -s, --syncdeps   Install missing dependencies with pacman"
+	echo "  -u, --noup2date  Do not check for newer version"
 	echo "  -w <destdir>     Write package to <destdir> instead of the working dir"
 	echo
 	echo "  if -p is not specified, makepkg will look for a FrugalBuild"
@@ -216,6 +217,7 @@
 NOSTRIP=0
 NOBUILD=0
 RMDEPS=0
+NOUP2DATE=0
 BUILDSCRIPT="./FrugalBuild"
 
 ARGLIST=$@
@@ -236,6 +238,7 @@
 		--nocolor)    USE_COLOR="n" ;;
 		--genmd5)     GENMD5=1 ;;
 		--rmdeps)     RMDEPS=1 ;;
+		--noup2date)     NOUP2DATE=1 ;;
 		--help)
 			usage
 			exit 0
@@ -245,7 +248,7 @@
 			exit 1
 			;;
 		-*)
-			while getopts "acCsbdehifgj:Lmnorp:w:-" opt; do
+			while getopts "acCsbdehifgj:Lmnorp:uw:-" opt; do
 				case $opt in
 					a) SEARCHDEPS=1 ;;
 					c) CLEANUP=1 ;;
@@ -267,6 +270,7 @@
 					p) BUILDSCRIPT=$OPTARG ;;
 					r) RMDEPS=1 ;;
 					s) DEP_BIN=1 ;;
+					u) NOUP2DATE=1 ;;
 					w) PKGDEST=$OPTARG ;;
 					-)
 						OPTIND=0
@@ -388,6 +392,31 @@
 
 cd $startdir
 
+# version checking
+if [ "$NOUP2DATE" = "0" ]; then
+	msg "Checking for newer version..."
+	if echo "$up2date"|grep -q " "; then
+		if type -p `echo "$up2date"|sed 's/^\([^ ]*\) .*/\1/'` &>/dev/null; then
+			up2date=`eval "$up2date"`
+		else
+			warning "The $cmd program is missing.  Cannot check for newer version!"
+			up2date=""
+			sleep 1
+		fi
+	fi
+	if [ -z "$up2date" ]; then
+		warning "Version checking informations are missing. Cannot check for newer version."
+	else
+		if [ "$pkgver" != "$up2date" ]; then
+			echo "   FAILED" >&2
+			error "Newer version available! ($pkgver < $up2date)"
+			exit 1
+		else
+		echo "   Passed" >&2
+		fi
+	fi
+fi
+
 # retrieve sources
 msg "Retrieving Sources..."
 mkdir -p src
diff -Naur pacman-2.9.5.orig/doc/makepkg.8.in pacman-2.9.5/doc/makepkg.8.in
--- pacman-2.9.5.orig/doc/makepkg.8.in	2004-08-04 07:57:17.000000000 +0200
+++ pacman-2.9.5/doc/makepkg.8.in	2005-05-26 11:07:49.000000000 +0200
@@ -38,6 +38,7 @@
 makedepends=('bash' 'mawk')
 depends=('glibc' 'zlib')
 archs=('i686')
+up2date="lynx -dump http://www.kernel.org/pub/linux/utils/kernel/modutils/v2.4/|grep LATEST |sed -n 's/.*S-\([0-9\.]*\) .*/\1/; 1 p'"
 source=(ftp://ftp.kernel.org/pub/linux/utils/kernel/$pkgname/v2.4/$pkgname-$pkgver.tar.bz2 \\
         modules.conf)
 md5sums=('2c0cca3ef6330a187c6ef4fe41ecaa4d' \\
@@ -258,6 +259,12 @@
 \fIsource\fP array.  (eg, install=modutils.install)
 
 .TP
+.B up2date
+This directive should contain a command that prints the current upstream stable
+version of the project. This way we can check for newer version without visiting
+manually the project's website (see above).
+
+.TP
 .B source \fI(array)\fP 
 The \fIsource\fP line is an array of source files required to build the
 package.  Source files must reside in the same directory as the PKGBUILD
@@ -384,6 +391,11 @@
 pacman will download the missing packages from a package repository and
 install them for you.
 .TP
+.B "\-u, \-\-noup2date"
+Do not check for newer version before starting build. Normally makepkg will
+prevent you from building obsolete source by mistake. If you know what you are
+doing, you can disable this feature with this switch.
+.TP
 .B "\-w <destdir>"
 Write the resulting package file to the directory \fI<destdir>\fP instead of the
 current working directory.
