diff -Naur pacman-2.9.6.orig/etc/makepkg.conf pacman-2.9.6/etc/makepkg.conf
--- pacman-2.9.6.orig/etc/makepkg.conf	2005-06-10 23:31:25.000000000 +0200
+++ pacman-2.9.6/etc/makepkg.conf	2005-06-13 10:27:59.000000000 +0200
@@ -22,6 +22,9 @@
 # SMP Systems
 #export MAKEFLAGS="-j 2"
 
+# Build time of static binutils in seconds
+SBU="1"
+
 # Enable fakeroot for building packages as a non-root user
 export USE_FAKEROOT="y"
 
diff -Naur pacman-2.9.6.orig/scripts/makepkg pacman-2.9.6/scripts/makepkg
--- pacman-2.9.6.orig/scripts/makepkg	2005-06-10 22:54:29.000000000 +0200
+++ pacman-2.9.6/scripts/makepkg	2005-06-13 10:26:55.000000000 +0200
@@ -557,12 +557,21 @@
 
 # build
 msg "Starting build()..."
+stime=`date +%s`
 build 2>&1
 if [ $? -gt 0 ]; then
 	error "Build Failed.  Aborting..."
 	exit 2
 fi
 
+# count sbu
+if [ "$SBU" == "1" ]; then
+	# $SBU not yet set
+	msg "Elapsed Time: $(($(date +%s)-$stime)) seconds"
+elif [ ! -z "$SBU" ]; then
+	 msg "Elapsed Time: `echo -e "scale=2\n$(($(date +%s)-$stime))/$SBU"|bc /dev/stdin|sed 's/^\./0./'` SBU"
+fi
+
 # documentation
 msg "Preparing package documentation..."
 cd $startdir
