diff -Naur pacman-2.9.5.orig/doc/makepkg.8.in pacman-2.9.5/doc/makepkg.8.in
--- pacman-2.9.5.orig/doc/makepkg.8.in	2004-08-04 07:57:17.000000000 +0200
+++ pacman-2.9.5/doc/makepkg.8.in	2005-05-20 21:28:51.000000000 +0200
@@ -241,6 +241,12 @@
 if its an older version.
 
 .TP
+.B nobuild
+If this directive set, gensync will ignore this package, so users must build
+these packages on their machines, they will not be able to install them with
+\fBpacman -S\fP. Useful for closed-source, but freeware programs.
+
+.TP
 .B url
 This field contains an optional URL that is associated with the piece of software
 being packaged.  This is typically the project's website.
diff -Naur pacman-2.9.5.orig/scripts/gensync pacman-2.9.5/scripts/gensync
--- pacman-2.9.5.orig/scripts/gensync	2004-10-23 19:27:07.000000000 +0200
+++ pacman-2.9.5/scripts/gensync	2005-05-23 14:03:25.000000000 +0200
@@ -163,7 +163,9 @@
 echo "gensync: building database entries, generating md5sums..." >&2
 cd `dirname $2`
 for file in `find $rootdir/* -name FrugalBuild`; do
+	unset nobuild
 	source $file || die "errors parsing $file"
+	[ "$nobuild" ] && continue
 	if [ "$pkgdir" != "" ]; then
 		pkgfile="$pkgdir/$pkgname-$pkgver-$pkgrel.pkg.tar.gz"
 	else
diff -Naur pacman-2.9.5.orig/scripts/updatesync pacman-2.9.5/scripts/updatesync
--- pacman-2.9.5.orig/scripts/updatesync	2004-09-25 19:59:49.000000000 +0200
+++ pacman-2.9.5/scripts/updatesync	2005-05-23 14:04:32.000000000 +0200
@@ -205,6 +205,7 @@
 	delete_entry $option
 
 	source $option || die "errors parsing $option"
+	if [ ! "$nobuild" ]; then
 	if [ "$pkgdir" != "" ]; then
 		pkgfile="$pkgdir/$pkgname-$pkgver-$pkgrel.pkg.tar.gz"
 	else
@@ -216,6 +217,7 @@
 	[ -z $pkgmd5sum ] && die "error generating checksum for $pkgfile"
 	echo "updatesync: creating entry for $option" >&2
 	db_write_entry $option || die "error writing entry for $option"
+	fi
 else
 	# DELETE
 	delete_entry $option
