diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN gksu-2.0.0.orig/configure.ac gksu-2.0.0/configure.ac
--- gksu-2.0.0.orig/configure.ac	2006-09-23 16:08:21.000000000 -0400
+++ gksu-2.0.0/configure.ac	2008-03-31 15:21:46.000000000 -0400
@@ -66,7 +66,7 @@ AC_ARG_ENABLE(nautilus-extension, [  --e
 AM_CONDITIONAL(ENABLE_NAUTILUS_EXTENSION, test x$enable_nautilus_extension = xyes)
 
 if test x$enable_nautilus_extension = xyes; then
-   PKG_CHECK_MODULES(NAUTILUS, [libgksu2 >= 2.0.0, libnautilus-extension])
+   PKG_CHECK_MODULES(NAUTILUS, [libgksu2 >= 2.0.0, libnautilus-extension, gnome-vfs-2.0])
 fi
 
 AC_CONFIG_FILES([
