diff -Naur varnish-1.1/Makefile.am varnish-1.1.new/Makefile.am
--- varnish-1.1/Makefile.am	2007-07-20 17:00:04.000000000 +0530
+++ varnish-1.1.new/Makefile.am	2007-07-31 21:19:08.000000000 +0530
@@ -10,4 +10,4 @@
 EXTRA_DIST = LICENSE autogen.sh varnishapi.pc.in
 
 install-data-local:
-	$(install_sh) -d -m 0755 $(localstatedir)/varnish
+	$(install_sh) -d -m 0755 $(DESTDIR)$(localstatedir)/varnish
