diff -Naur irssi-0.8.10.orig/src/irc/dcc/dcc.c irssi-0.8.10/src/irc/dcc/dcc.c
--- irssi-0.8.10.orig/src/irc/dcc/dcc.c	2006-01-18 14:25:19.000000000 +0200
+++ irssi-0.8.10/src/irc/dcc/dcc.c	2006-01-18 14:41:20.000000000 +0200
@@ -58,8 +58,8 @@
 
 	pos = gslist_find_string(dcc_types, type);
 	if (pos != NULL) {
-                dcc_types = g_slist_remove(dcc_types, pos->data);
 		g_free(pos->data);
+                dcc_types = g_slist_remove(dcc_types, pos->data);
 	}
 }
 
