diff -Naur frugalwareutils-0.8.2/configure.ac frugalwareutils-0.8.2-p/configure.ac
--- frugalwareutils-0.8.2/configure.ac	2008-07-27 19:54:31.000000000 +0200
+++ frugalwareutils-0.8.2-p/configure.ac	2008-08-01 23:41:56.000000000 +0200
@@ -77,7 +77,7 @@
 		LIB_CFLAGS="$CFLAGS"
 		;;
 	x86_64-*)
-		CFLAGS="$CFLAGS -march=k8"
+		CFLAGS="$CFLAGS -march=x86-64"
 		LIB_CFLAGS="$CFLAGS -fPIC"
 		;;
 esac
