--- libindicator-12.10.1/libindicator/Makefile.am	2012-09-19 17:38:05.000000000 +0200
+++ libindicator-12.10.1.new/libindicator/Makefile.am	2016-05-19 14:09:21.000000000 +0200
@@ -57,7 +57,7 @@
 	$(LIBINDICATOR_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
 	-DG_LOG_DOMAIN=\"libindicator\" \
-	-Wall -Werror
+	-Wall -Wno-deprecated-declarations
 
 libindicator_la_LIBADD = \
 	$(LIBINDICATOR_LIBS)
--- libindicator-12.10.1/libindicator/Makefile.in	2012-09-19 17:40:09.000000000 +0200
+++ libindicator-12.10.1.new/libindicator/Makefile.in	2016-05-19 14:09:31.000000000 +0200
@@ -380,7 +380,7 @@
 	$(LIBINDICATOR_CFLAGS) \
 	$(COVERAGE_CFLAGS) \
 	-DG_LOG_DOMAIN=\"libindicator\" \
-	-Wall -Werror
+	-Wall -Wno-deprecated-declarations
 
 libindicator_la_LIBADD = \
 	$(LIBINDICATOR_LIBS)
