diff -Naur zstd-1.5.2/contrib/linux-kernel/test/Makefile zstd-1.5.2.new/contrib/linux-kernel/test/Makefile
--- zstd-1.5.2/contrib/linux-kernel/test/Makefile	2022-01-20 22:17:18.000000000 +0100
+++ zstd-1.5.2.new/contrib/linux-kernel/test/Makefile	2022-08-02 16:34:05.732107164 +0200
@@ -24,7 +24,7 @@
 LINUX_ZSTD_OBJECTS    := $(LINUX_ZSTD_OBJECTS0:.S=.o)
 
 %.o: %.S
-	$(COMPILE.S) $(OUTPUT_OPTION) $<
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) $^ -o $@
 
 liblinuxzstd.a: $(LINUX_ZSTD_OBJECTS)
 	$(AR) $(ARFLAGS) $@ $^
diff -Naur zstd-1.5.2/lib/libzstd.mk zstd-1.5.2.new/lib/libzstd.mk
--- zstd-1.5.2/lib/libzstd.mk	2022-01-20 22:17:18.000000000 +0100
+++ zstd-1.5.2.new/lib/libzstd.mk	2022-08-02 16:34:05.732107164 +0200
@@ -95,9 +95,8 @@
             -Wvla -Wformat=2 -Winit-self -Wfloat-equal -Wwrite-strings \
             -Wredundant-decls -Wmissing-prototypes -Wc++-compat
 CFLAGS   += $(DEBUGFLAGS) $(MOREFLAGS)
-ASFLAGS  += $(DEBUGFLAGS) $(MOREFLAGS) $(CFLAGS)
 LDFLAGS  += $(MOREFLAGS)
-FLAGS     = $(CPPFLAGS) $(CFLAGS) $(ASFLAGS) $(LDFLAGS)
+FLAGS     = $(CPPFLAGS) $(CFLAGS) $(LDFLAGS)
 
 ifndef ALREADY_APPENDED_NOEXECSTACK
 export ALREADY_APPENDED_NOEXECSTACK := 1
@@ -105,12 +104,10 @@
 LDFLAGS += -z noexecstack
 endif
 ifeq ($(shell echo | $(CC) $(FLAGS) -Wa,--noexecstack -x assembler -Werror -c - -o $(VOID) 2>$(VOID) && echo 1 || echo 0),1)
-CFLAGS  += -Wa,--noexecstack
-# CFLAGS are also added to ASFLAGS
+CFLAGS += -Wa,--noexecstack
 else ifeq ($(shell echo | $(CC) $(FLAGS) -Qunused-arguments -Wa,--noexecstack -x assembler -Werror -c - -o $(VOID) 2>$(VOID) && echo 1 || echo 0),1)
 # See e.g.: https://github.com/android/ndk/issues/171
-CFLAGS  += -Qunused-arguments -Wa,--noexecstack
-# CFLAGS are also added to ASFLAGS
+CFLAGS += -Qunused-arguments -Wa,--noexecstack
 endif
 endif
 
diff -Naur zstd-1.5.2/lib/Makefile zstd-1.5.2.new/lib/Makefile
--- zstd-1.5.2/lib/Makefile	2022-01-20 22:17:18.000000000 +0100
+++ zstd-1.5.2.new/lib/Makefile	2022-08-02 16:34:05.732107164 +0200
@@ -212,11 +212,11 @@
 
 $(ZSTD_DYNLIB_DIR)/%.o : %.S | $(ZSTD_DYNLIB_DIR)
 	@echo AS $@
-	$(COMPILE.S) $(OUTPUT_OPTION) $<
+	$(COMPILE.c) $(OUTPUT_OPTION) $<
 
 $(ZSTD_STATLIB_DIR)/%.o : %.S | $(ZSTD_STATLIB_DIR)
 	@echo AS $@
-	$(COMPILE.S) $(OUTPUT_OPTION) $<
+	$(COMPILE.c) $(OUTPUT_OPTION) $<
 
 MKDIR ?= mkdir
 $(BUILD_DIR) $(ZSTD_DYNLIB_DIR) $(ZSTD_STATLIB_DIR):
diff -Naur zstd-1.5.2/programs/Makefile zstd-1.5.2.new/programs/Makefile
--- zstd-1.5.2/programs/Makefile	2022-01-20 22:17:18.000000000 +0100
+++ zstd-1.5.2.new/programs/Makefile	2022-08-02 16:34:05.733107107 +0200
@@ -320,7 +320,7 @@
 
 $(BUILD_DIR)/%.o : %.S | $(BUILD_DIR)
 	@echo AS $@
-	$(COMPILE.S) $(OUTPUT_OPTION) $<
+	$(COMPILE.c) $(OUTPUT_OPTION) $<
 
 MKDIR ?= mkdir
 $(BUILD_DIR): ; $(MKDIR) -p $@
diff -Naur zstd-1.5.2/tests/fuzz/Makefile zstd-1.5.2.new/tests/fuzz/Makefile
--- zstd-1.5.2/tests/fuzz/Makefile	2022-01-20 22:17:18.000000000 +0100
+++ zstd-1.5.2.new/tests/fuzz/Makefile	2022-08-02 16:34:05.733107107 +0200
@@ -12,7 +12,6 @@
 CFLAGS ?= -O3
 CXXFLAGS ?= -O3
 CPPFLAGS ?=
-ASFLAGS ?=
 LDFLAGS ?=
 ARFLAGS ?=
 LIB_FUZZING_ENGINE ?= libregression.a
@@ -44,7 +43,6 @@
 	-Wredundant-decls -Wno-deprecated-declarations \
 	-g -fno-omit-frame-pointer
 FUZZ_CFLAGS := $(FUZZ_EXTRA_FLAGS) $(CFLAGS)
-FUZZ_ASFLAGS := $(FUZZ_EXTRA_FLAGS) $(ASFLAGS)
 FUZZ_CXXFLAGS := $(FUZZ_EXTRA_FLAGS) -std=c++11 $(CXXFLAGS)
 FUZZ_LDFLAGS := -pthread $(LDFLAGS)
 FUZZ_ARFLAGS := $(ARFLAGS)
@@ -129,7 +127,7 @@
 	$(CC) $(FUZZ_CPPFLAGS) $(FUZZ_CFLAGS) $(FUZZ_ROUND_TRIP_FLAGS) $< -c -o $@
 
 rt_lib_decompress_%.o: $(ZSTDDIR)/decompress/%.S
-	$(CC) $(FUZZ_CPPFLAGS) $(FUZZ_ASFLAGS) $(FUZZ_ROUND_TRIP_FLAGS) $< -c -o $@
+	$(CC) $(FUZZ_CPPFLAGS) $(FUZZ_CFLAGS) $(FUZZ_ROUND_TRIP_FLAGS) $< -c -o $@
 
 rt_lib_dictBuilder_%.o: $(ZSTDDIR)/dictBuilder/%.c
 	$(CC) $(FUZZ_CPPFLAGS) $(FUZZ_CFLAGS) $(FUZZ_ROUND_TRIP_FLAGS) $< -c -o $@
@@ -153,7 +151,7 @@
 	$(CC) $(FUZZ_CPPFLAGS) $(FUZZ_CFLAGS) $< -c -o $@
 
 d_lib_decompress_%.o: $(ZSTDDIR)/decompress/%.S
-	$(CC) $(FUZZ_CPPFLAGS) $(FUZZ_ASFLAGS) $< -c -o $@
+	$(CC) $(FUZZ_CPPFLAGS) $(FUZZ_CFLAGS) $< -c -o $@
 
 d_lib_dictBuilder_%.o: $(ZSTDDIR)/dictBuilder/%.c
 	$(CC) $(FUZZ_CPPFLAGS) $(FUZZ_CFLAGS) $< -c -o $@
diff -Naur zstd-1.5.2/tests/Makefile zstd-1.5.2.new/tests/Makefile
--- zstd-1.5.2/tests/Makefile	2022-01-20 22:17:18.000000000 +0100
+++ zstd-1.5.2.new/tests/Makefile	2022-08-02 16:34:05.733107107 +0200
@@ -118,7 +118,7 @@
 	$(CC) -c $(CPPFLAGS) $(CFLAGS) $< -o $@
 
 zstdd_%.o : $(ZSTDDIR)/decompress/%.S
-	$(CC) -c $(CPPFLAGS) $(ASFLAGS) $< -o $@
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) $< -o $@
 
 zstdmt%.o : CPPFLAGS += $(MULTITHREAD_CPP)
 
@@ -132,7 +132,7 @@
 	$(CC) -c $(CPPFLAGS) $(CFLAGS) $< -o $@
 
 zstdmt_d_%.o : $(ZSTDDIR)/decompress/%.S
-	$(CC) -c $(CPPFLAGS) $(ASFLAGS) $< -o $@
+	$(CC) -c $(CPPFLAGS) $(CFLAGS) $< -o $@
 
 fullbench32: CPPFLAGS += -m32
 fullbench fullbench32 : CPPFLAGS += $(MULTITHREAD_CPP) -Wno-deprecated-declarations
