diff -Naur MPlayer-1.0pre4.orig/loader/ldt_keeper.c MPlayer-1.0pre4/loader/ldt_keeper.c
--- MPlayer-1.0pre4.orig/loader/ldt_keeper.c	2003-09-06 02:14:08.000000000 +0200
+++ MPlayer-1.0pre4/loader/ldt_keeper.c	2004-06-27 03:19:15.000000000 +0200
@@ -26,9 +26,7 @@
 #include <asm/ldt.h>
 // 2.5.xx+ calls this user_desc:
 #include <linux/version.h>
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,47)
 #define modify_ldt_ldt_s user_desc
-#endif
 /* prototype it here, so we won't depend on kernel headers */
 #ifdef  __cplusplus
 extern "C" {
