diff -Naur txt2regex-0.8.orig/txt2regex-0.8.sh txt2regex-0.8/txt2regex-0.8.sh
--- txt2regex-0.8.orig/txt2regex-0.8.sh	2004-09-28 21:51:01.000000000 +0200
+++ txt2regex-0.8/txt2regex-0.8.sh	2005-12-30 00:20:43.000000000 +0100
@@ -517,7 +517,7 @@
 
 doMenu(){
   local -a Menui
-  eval Menui=(\"\${$1[@]}\"); menu_n=$((${#Menui[*]}-1))    # ini
+  eval "Menui=(\"\${$1[@]}\")"; menu_n=$((${#Menui[*]}-1))    # ini
 
   if [ "$f_i" == 1 ]; then
     gotoxy $x_hist $y_hist
