diff -Naur soprano-2.4.0.1/backends/sesame2/jniwrapper.cpp soprano-2.4.0.1-p/backends/sesame2/jniwrapper.cpp
--- soprano-2.4.0.1/backends/sesame2/jniwrapper.cpp	2009-08-21 09:05:13.000000000 +0200
+++ soprano-2.4.0.1-p/backends/sesame2/jniwrapper.cpp	2010-02-17 20:57:48.000000000 +0100
@@ -67,12 +67,11 @@
         JavaVMInitArgs vmArgs;
         JavaVMOption vmOptions[4];
         vmOptions[0].optionString = ( char* )"-Djava.class.path="SESAME2_CLASSPATH;
-        vmOptions[1].optionString = ( char* )"-verbose:jni,gc,class";
-        vmOptions[2].optionString = ( char* )"-Xms256m";
-        vmOptions[3].optionString = ( char* )"-Xmx256m";
+        vmOptions[1].optionString = ( char* )"-Xms256m";
+        vmOptions[2].optionString = ( char* )"-Xmx256m";
         vmArgs.version = JNI_VERSION_1_4;
         vmArgs.options = vmOptions;
-        vmArgs.nOptions = 4;
+        vmArgs.nOptions = 3;
 
         // create the VM
         JavaVM* jvm = 0;

