--- plasma-workspace-5.27.9.1/CMakeLists.txt	2023-10-25 12:05:42.000000000 +0200
+++ plasma-workspace-5.27.9.1.new/CMakeLists.txt	2023-11-24 00:10:54.436107377 +0100
@@ -183,7 +183,7 @@
   find_package(Qt${QT_MAJOR_VERSION} ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS PrintSupport)
 endif()
 
-find_package(AppStreamQt 0.10.6)
+find_package(AppStreamQt5 1.0.0)
 set_package_properties(AppStreamQt PROPERTIES DESCRIPTION "Access metadata for listing available software"
                        URL "https://www.freedesktop.org/wiki/Distributions/AppStream/"
                        TYPE OPTIONAL)
