diff -Naur plasma-workspace-5.6.2/ksmserver/shutdown.cpp plasma-workspace-5.6.2-p/ksmserver/shutdown.cpp
--- plasma-workspace-5.6.2/ksmserver/shutdown.cpp	2016-04-05 15:58:32.000000000 +0200
+++ plasma-workspace-5.6.2-p/ksmserver/shutdown.cpp	2016-04-11 15:30:05.965002077 +0200
@@ -276,7 +276,7 @@
 // callbacks
 void KSMServer::saveYourselfDone( KSMClient* client, bool success )
 {
-    if ( state == Idle ) {
+    if ( state != Shutdown && state != Checkpoint ) {
         // State saving when it's not shutdown or checkpoint. Probably
         // a shutdown was canceled and the client is finished saving
         // only now. Discard the saved state in order to avoid
