diff -Naur plasma-workspace-5.9.5/ksplash/ksplashqml/splashapp.cpp plasma-workspace-5.9.5-p/ksplash/ksplashqml/splashapp.cpp
--- plasma-workspace-5.9.5/ksplash/ksplashqml/splashapp.cpp	2017-04-25 15:34:07.000000000 +0200
+++ plasma-workspace-5.9.5-p/ksplash/ksplashqml/splashapp.cpp	2017-04-28 10:49:58.872488986 +0200
@@ -35,7 +35,8 @@
 #include <KWayland/Client/registry.h>
 #include <KWayland/Client/plasmashell.h>
 
-#define TEST_STEP_INTERVAL 2000
+#define TEST_STEP_INTERVAL 2000 // 2 sec
+#define STEP_INTERVAL 700  // 0,7 sec
 
 /**
  * There are 6 used stages in ksplash
@@ -104,7 +105,7 @@
 
         setStage(m_stage + 1);
 
-        m_timer.start(TEST_STEP_INTERVAL, this);
+        m_timer.start(STEP_INTERVAL, this);
     }
 }
 
diff -Naur plasma-workspace-5.9.5/ksplash/ksplashqml/splashwindow.cpp plasma-workspace-5.9.5-p/ksplash/ksplashqml/splashwindow.cpp
--- plasma-workspace-5.9.5/ksplash/ksplashqml/splashwindow.cpp	2017-04-25 15:34:07.000000000 +0200
+++ plasma-workspace-5.9.5-p/ksplash/ksplashqml/splashwindow.cpp	2017-04-28 10:50:27.420242985 +0200
@@ -70,7 +70,7 @@
 
     //be sure it will be eventually closed
     //FIXME: should never be stuck
-    QTimer::singleShot(30000, this, &QWindow::close);
+    QTimer::singleShot(8000, this, &QWindow::close);
 }
 
 void SplashWindow::setStage(int stage)
