diff -Naur plasma-workspace-5.10.3/ksplash/ksplashqml/main.cpp plasma-workspace-5.10.3-p/ksplash/ksplashqml/main.cpp
--- plasma-workspace-5.10.3/ksplash/ksplashqml/main.cpp	2017-06-27 12:00:38.000000000 +0200
+++ plasma-workspace-5.10.3-p/ksplash/ksplashqml/main.cpp	2017-06-27 20:48:03.948002742 +0200
@@ -24,6 +24,9 @@
 #include <QTextStream>
 
 #include <iostream>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
 #include <unistd.h>
 
 void logMessageHandler(QtMsgType type, const char *msg)
@@ -83,6 +86,16 @@
 
             return 0;
         }
+
+        int devNull = open("/dev/null", O_RDWR);
+        if (devNull == -1) {
+            return -1;
+        }
+        // replace stdin,stdout,stderr, otherwise startkde will block
+        dup2(devNull, 0);
+        dup2(devNull, 1);
+        dup2(devNull, 2);
+        close(devNull);
     }
 
     //enable to send log output to /tmp/ksplash
@@ -91,13 +104,6 @@
     QQuickWindow::setDefaultAlphaBuffer(true);
     SplashApp app(argc, argv);
 
-    if (!test && !nofork) {
-        // close stdin,stdout,stderr, otherwise startkde will block
-        close(0);
-        close(1);
-        close(2);
-    }
-
     return app.exec();
 }
 
