diff -Naur plasma-workspace-5.6.2/startkde/startkde.cmake plasma-workspace-5.6.2-p/startkde/startkde.cmake
--- plasma-workspace-5.6.2/startkde/startkde.cmake	2016-04-05 15:58:32.000000000 +0200
+++ plasma-workspace-5.6.2-p/startkde/startkde.cmake	2016-04-16 20:03:46.837934043 +0200
@@ -95,7 +95,14 @@
 kdeglobalsfile=$configDir/kdeglobals
 test -f $kdeglobalsfile || {
 cat >$kdeglobalsfile <<EOF
+[KDE]
+widgetStyle=breeze
+
+[Icons]
+Theme=breeze
+
 [General]
+ColorScheme=Breeze
 XftAntialias=true
 XftHintStyle=hintmedium
 XftSubPixel=none
diff -Naur plasma-workspace-5.6.2/startkde/startplasmacompositor.cmake plasma-workspace-5.6.2-p/startkde/startplasmacompositor.cmake
--- plasma-workspace-5.6.2/startkde/startplasmacompositor.cmake	2016-04-05 15:58:32.000000000 +0200
+++ plasma-workspace-5.6.2-p/startkde/startplasmacompositor.cmake	2016-04-16 20:05:07.453187620 +0200
@@ -55,7 +55,14 @@
 kdeglobalsfile=$configDir/kdeglobals
 test -f $kdeglobalsfile || {
 cat >$kdeglobalsfile <<EOF
+[KDE]
+widgetStyle=breeze
+
+[Icons]
+Theme=breeze
+
 [General]
+ColorScheme=Breeze
 XftAntialias=true
 XftHintStyle=hintmedium
 XftSubPixel=none
