diff -Naur plasma-workspace-5.6.0/startkde/startkde.cmake plasma-workspace-5.6.0-p/startkde/startkde.cmake
--- plasma-workspace-5.6.0/startkde/startkde.cmake	2016-03-17 12:03:23.000000000 +0100
+++ plasma-workspace-5.6.0-p/startkde/startkde.cmake	2016-03-23 20:54:04.921363928 +0100
@@ -15,19 +15,21 @@
 trap 'echo GOT SIGHUP' HUP
 
 # we have to unset this for Darwin since it will screw up KDE's dynamic-loading
-unset DYLD_FORCE_FLAT_NAMESPACE
+# unset DYLD_FORCE_FLAT_NAMESPACE
 
 # in case we have been started with full pathname spec without being in PATH
 bindir=`echo "$0" | sed -n 's,^\(/.*\)/[^/][^/]*$,\1,p'`
 if [ -n "$bindir" ]; then
-  qbindir=`qtpaths --binaries-dir`
+  qbindir=`qtpaths-qt5 --binaries-dir`
   qdbus=$qbindir/qdbus
   case $PATH in
     $bindir|$bindir:*|*:$bindir|*:$bindir:*) ;;
     *) PATH=$bindir:$PATH; export PATH;;
   esac
 else
-  qdbus=qdbus
+## welll no-no-no -- we want qt5 qdbus wth
+ qbindir=`qtpaths-qt5 --binaries-dir`
+ qdbus=$qbindir/qdbus
 fi
 
 # Check if a KDE session already is running and whether it's possible to connect to X
@@ -189,7 +191,7 @@
 
 # Source scripts found in <config locations>/plasma-workspace/env/*.sh
 # (where <config locations> correspond to the system and user's configuration
-# directories, as identified by Qt's qtpaths,  e.g.  $HOME/.config
+# directories, as identified by Qt's qtpaths-qt5,  e.g.  $HOME/.config
 # and /etc/xdg/ on Linux)
 #
 # This is where you can define environment variables that will be available to
@@ -201,8 +203,7 @@
 # For anything else (that doesn't set env vars, or that needs a window manager),
 # better use the Autostart folder.
 
-# TODO: Use GenericConfigLocation once we depend on Qt 5.4
-scriptpath=`qtpaths --paths ConfigLocation | tr ':' '\n' | sed 's,$,/plasma-workspace,g'`
+scriptpath=`qtpaths-qt5 --paths GenericConfigLocation | tr ':' '\n' | sed 's,$,/plasma-workspace,g'`
 
 # Add /env/ to the directory to locate the scripts to be sourced
 for prefix in `echo $scriptpath`; do
diff -Naur plasma-workspace-5.6.0/startkde/startplasmacompositor.cmake plasma-workspace-5.6.0-p/startkde/startplasmacompositor.cmake
--- plasma-workspace-5.6.0/startkde/startplasmacompositor.cmake	2016-03-17 12:03:23.000000000 +0100
+++ plasma-workspace-5.6.0-p/startkde/startplasmacompositor.cmake	2016-03-23 20:54:04.922363955 +0100
@@ -6,14 +6,16 @@
 # in case we have been started with full pathname spec without being in PATH
 bindir=`echo "$0" | sed -n 's,^\(/.*\)/[^/][^/]*$,\1,p'`
 if [ -n "$bindir" ]; then
-  qbindir=`qtpaths --binaries-dir`
+  qbindir=`qtpaths-qt5 --binaries-dir`
   qdbus=$qbindir/qdbus
   case $PATH in
     $bindir|$bindir:*|*:$bindir|*:$bindir:*) ;;
     *) PATH=$bindir:$PATH; export PATH;;
   esac
 else
-  qdbus=qdbus
+## well no-no-no - *again* :D
+ qbindir=`qtpaths-qt5 --binaries-dir`
+ qdbus=$qbindir/qdbus
 fi
 
 # We need to create config folder so we can write startupconfigkeys
@@ -126,7 +128,7 @@
 
 # Source scripts found in <config locations>/plasma-workspace/env/*.sh
 # (where <config locations> correspond to the system and user's configuration
-# directories, as identified by Qt's qtpaths,  e.g.  $HOME/.config
+# directories, as identified by Qt's qtpaths-qt5,  e.g.  $HOME/.config
 # and /etc/xdg/ on Linux)
 #
 # This is where you can define environment variables that will be available to
@@ -138,8 +140,7 @@
 # For anything else (that doesn't set env vars, or that needs a window manager),
 # better use the Autostart folder.
 
-# TODO: Use GenericConfigLocation once we depend on Qt 5.4
-scriptpath=`qtpaths --paths ConfigLocation | tr ':' '\n' | sed 's,$,/plasma-workspace,g'`
+scriptpath=`qtpaths-qt5 --paths GenericConfigLocation | tr ':' '\n' | sed 's,$,/plasma-workspace,g'`
 
 # Add /env/ to the directory to locate the scripts to be sourced
 for prefix in `echo $scriptpath`; do
