diff -Naur plasma-workspace-5.8.0/startkde/startkde.cmake plasma-workspace-5.8.0-p/startkde/startkde.cmake
--- plasma-workspace-5.8.0/startkde/startkde.cmake	2016-09-29 13:35:01.000000000 +0200
+++ plasma-workspace-5.8.0-p/startkde/startkde.cmake	2016-10-04 16:55:05.808419064 +0200
@@ -20,14 +20,14 @@
 # in case we have been started with full pathname spec without being in PATH
 bindir=`echo "$0" | sed -n 's,^\(/.*\)/[^/][^/]*$,\1,p'`
 if [ -n "$bindir" ]; then
-  qbindir=`qtpaths --binaries-dir`
-  qdbus=$qbindir/qdbus
+  qbindir=`qtpaths-qt5 --binaries-dir`
+  qdbus="$qbindir/qdbus"
   case $PATH in
     $bindir|$bindir:*|*:$bindir|*:$bindir:*) ;;
     *) PATH=$bindir:$PATH; export PATH;;
   esac
 else
-  qdbus=qdbus
+  qdbus="qdbus-qt5"
 fi
 
 # Check if a KDE session already is running and whether it's possible to connect to X
@@ -95,7 +95,14 @@
 kdeglobalsfile=$configDir/kdeglobals
 test -f $kdeglobalsfile || {
 cat >$kdeglobalsfile <<EOF
+[KDE]
+widgetStyle=breeze
+
+[Icons]
+Theme=breeze
+
 [General]
+ColorScheme=Breeze
 XftAntialias=true
 XftHintStyle=hintmedium
 XftSubPixel=none
@@ -206,7 +213,7 @@
 # better use the Autostart folder.
 
 # TODO: Use GenericConfigLocation once we depend on Qt 5.4
-scriptpath=`qtpaths --paths ConfigLocation | tr ':' '\n' | sed 's,$,/plasma-workspace,g'`
+scriptpath=`qtpaths-qt5 --paths ConfigLocation | tr ':' '\n' | sed 's,$,/plasma-workspace,g'`
 
 # Add /env/ to the directory to locate the scripts to be sourced
 for prefix in `echo $scriptpath`; do
@@ -359,7 +366,7 @@
   exit 1
 fi
 
-qdbus org.kde.KSplash /KSplash org.kde.KSplash.setStage kinit
+$qdbus org.kde.KSplash /KSplash org.kde.KSplash.setStage kinit &
 
 # finally, give the session control to the session manager
 # see kdebase/ksmserver for the description of the rest of the startup sequence
@@ -388,7 +395,7 @@
 
 if test x"$wait_drkonqi"x = x"true"x ; then
     # wait for remaining drkonqi instances with timeout (in seconds)
-    wait_drkonqi_timeout=`kreadconfig5 --file startkderc --group WaitForDrKonqi --key Timeout --default 900`
+    wait_drkonqi_timeout=`kreadconfig5 --file startkderc --group WaitForDrKonqi --key Timeout --default 120`
     wait_drkonqi_counter=0
     while $qdbus | grep "^[^w]*org.kde.drkonqi" > /dev/null ; do
         sleep 5
diff -Naur plasma-workspace-5.8.0/startkde/startplasma.cmake plasma-workspace-5.8.0-p/startkde/startplasma.cmake
--- plasma-workspace-5.8.0/startkde/startplasma.cmake	2016-09-29 13:35:01.000000000 +0200
+++ plasma-workspace-5.8.0-p/startkde/startplasma.cmake	2016-10-04 16:54:01.768418115 +0200
@@ -18,6 +18,20 @@
 # * Then ksmserver is started which takes control of the rest of the startup sequence
 
 # We need to create config folder so we can write startupconfigkeys
+# in case we have been started with full pathname spec without being in PATH
+bindir=`echo "$0" | sed -n 's,^\(/.*\)/[^/][^/]*$,\1,p'`
+if [ -n "$bindir" ]; then
+  qbindir=`qtpaths-qt5 --binaries-dir`
+  qdbus="$qbindir/qdbus"
+  case $PATH in
+    $bindir|$bindir:*|*:$bindir|*:$bindir:*) ;;
+    *) PATH=$bindir:$PATH; export PATH;;
+  esac
+else
+  qdbus="qdbus-qt5"
+fi
+
+
 if [  ${XDG_CONFIG_HOME} ]; then
   configDir=$XDG_CONFIG_HOME;
 else
@@ -128,7 +142,7 @@
   exit 1
 fi
 
-qdbus org.kde.KSplash /KSplash org.kde.KSplash.setStage kinit
+$qdbus org.kde.KSplash /KSplash org.kde.KSplash.setStage kinit &
 
 # finally, give the session control to the session manager
 # see kdebase/ksmserver for the description of the rest of the startup sequence
diff -Naur plasma-workspace-5.8.0/startkde/startplasmacompositor.cmake plasma-workspace-5.8.0-p/startkde/startplasmacompositor.cmake
--- plasma-workspace-5.8.0/startkde/startplasmacompositor.cmake	2016-09-29 13:35:01.000000000 +0200
+++ plasma-workspace-5.8.0-p/startkde/startplasmacompositor.cmake	2016-10-04 16:54:54.270418893 +0200
@@ -6,14 +6,14 @@
 # in case we have been started with full pathname spec without being in PATH
 bindir=`echo "$0" | sed -n 's,^\(/.*\)/[^/][^/]*$,\1,p'`
 if [ -n "$bindir" ]; then
-  qbindir=`qtpaths --binaries-dir`
-  qdbus=$qbindir/qdbus
+  qbindir=`qtpaths-qt5 --binaries-dir`
+  qdbus="$qbindir/qdbus"
   case $PATH in
     $bindir|$bindir:*|*:$bindir|*:$bindir:*) ;;
     *) PATH=$bindir:$PATH; export PATH;;
   esac
 else
-  qdbus=qdbus
+  qdbus="qdbus-qt5"
 fi
 
 # We need to create config folder so we can write startupconfigkeys
@@ -55,7 +55,14 @@
 kdeglobalsfile=$configDir/kdeglobals
 test -f $kdeglobalsfile || {
 cat >$kdeglobalsfile <<EOF
+[KDE]
+widgetStyle=breeze
+
+[Icons]
+Theme=breeze
+
 [General]
+ColorScheme=Breeze
 XftAntialias=true
 XftHintStyle=hintmedium
 XftSubPixel=none
@@ -148,7 +155,7 @@
 # better use the Autostart folder.
 
 # TODO: Use GenericConfigLocation once we depend on Qt 5.4
-scriptpath=`qtpaths --paths ConfigLocation | tr ':' '\n' | sed 's,$,/plasma-workspace,g'`
+scriptpath=`qtpaths-qt5 --paths ConfigLocation | tr ':' '\n' | sed 's,$,/plasma-workspace,g'`
 
 # Add /env/ to the directory to locate the scripts to be sourced
 for prefix in `echo $scriptpath`; do
