--- a/src/AudioFileFactory.cxx
+++ b/src/AudioFileFactory.cxx
@@ -25,6 +25,8 @@
 #include "Resampler.H"
 #include <FL/filename.H>
 
+#include <cstring>
+
 namespace nle
 {
 
--- a/src/AudioFileQT.cxx
+++ b/src/AudioFileQT.cxx
@@ -22,6 +22,8 @@
 #include "AudioFileQT.H"
 #include "ErrorDialog/IErrorHandler.H"
 
+#include <cstdlib>
+
 
 namespace nle
 {
--- a/src/FilmStripFactory.cxx
+++ b/src/FilmStripFactory.cxx
@@ -25,6 +25,8 @@
 #include "IVideoFile.H"
 #include "FilmStrip.H"
 
+#include <cstring>
+
 
 namespace nle
 {
--- a/src/FolderBrowser.cxx
+++ b/src/FolderBrowser.cxx
@@ -21,6 +21,7 @@
 
 #include <cstdlib>
 #include <iostream>
+#include <cstring>
 
 namespace nle
 {
--- a/src/JackPlaybackCore.cxx
+++ b/src/JackPlaybackCore.cxx
@@ -39,6 +39,9 @@
 #include "ErrorDialog/IErrorHandler.H"
 #include <iostream>
 
+#include <cstdlib>
+#include <cstring>
+
 #define VIDEO_DRIFT_LIMIT 2 //Calculate this based on frame size
 #define FRAMES 4096
 
--- a/src/NodeFilterFrei0rFactoryPlugin.cxx
+++ b/src/NodeFilterFrei0rFactoryPlugin.cxx
@@ -3,6 +3,7 @@
 #include "NodeFilterFrei0rFactoryPlugin.H"
 #include "Frei0rNode.H"
 #include <iostream>
+#include <cstring>
 
 
 NodeFilterFrei0rFactoryPlugin::NodeFilterFrei0rFactoryPlugin( std::string filename )
--- a/src/PortAudioPlaybackCore.cxx
+++ b/src/PortAudioPlaybackCore.cxx
@@ -38,6 +38,8 @@
 #include "ErrorDialog/IErrorHandler.H"
 #include <iostream>
 
+#include <cstdlib>
+
 #define VIDEO_DRIFT_LIMIT (2 * 1200) //Calculate this based on frame size
 #define FRAMES 4096
 
--- a/src/VideoFileFactory.cxx
+++ b/src/VideoFileFactory.cxx
@@ -22,6 +22,7 @@
 #include "VideoFileFactory.H"
 #include "VideoFileMpeg3.H"
 #include <FL/filename.H>
+#include <cstring>
 namespace nle
 {
 
--- a/src/VideoThumbnails.cxx
+++ b/src/VideoThumbnails.cxx
@@ -25,6 +25,8 @@
 #include "VideoFileFactory.H"
 #include "ImageClip.H"
 
+#include <cstring>
+
 namespace nle
 {
 
--- a/src/fl_font_browser.cpp
+++ b/src/fl_font_browser.cpp
@@ -28,6 +28,8 @@
 
 
 #include "fl_font_browser.h"
+#include <cstdlib>
+#include <cstring>
 using namespace std;
 
 void Fl_Font_Browser::callback(void (*cb)(Fl_Widget *, void *), void *d ) 
