--- warzone2100/lib/ivis_opengl/gfx_api_vk.cpp	2023-11-06 20:11:19.000000000 +0100
+++ warzone2100.new/lib/ivis_opengl/gfx_api_vk.cpp	2023-11-20 13:08:05.490316727 +0100
@@ -62,7 +62,7 @@
 #if VK_HEADER_VERSION >= 260
 # define WZ_THROW_VK_RESULT_EXCEPTION(result, message) vk::detail::throwResultException(result, message)
 #else
-# define WZ_THROW_VK_RESULT_EXCEPTION(result, message) vk::throwResultException(result, message)
+# define WZ_THROW_VK_RESULT_EXCEPTION(result, message) vk::detail::throwResultException(result, message)
 #endif
 
 const uint32_t minSupportedVulkanVersion = VK_API_VERSION_1_0;
