diff -ur a/clientgui/gtk/taskbarex.cpp b/clientgui/gtk/taskbarex.cpp
--- a/clientgui/gtk/taskbarex.cpp	2012-07-22 23:50:13.738921066 -0500
+++ b/clientgui/gtk/taskbarex.cpp	2012-07-22 23:51:17.424346623 -0500
@@ -25,8 +25,7 @@
 #include <libnotify/notify.h>
 #endif
 
-#include <glib/gtypes.h>
-#include <glib-object.h>
+#include <glib.h>
 
 #include "BOINCGUIApp.h"
 #include "gtk/taskbarex.h"
