--- mc-4.8.29/misc/mc.ext.ini.in	2023-01-01 19:39:31.000000000 +0100
+++ mc-4.8.29.new/misc/mc.ext.ini.in	2023-01-22 21:26:31.345493088 +0100
@@ -404,6 +404,12 @@
 Open=%cd %p/deba://
 Ciew=%view{ascii} @EXTHELPERSDIR@/package.sh view deba
 
+[fw]
+Regex/\.(fpm|fdb)$
+Open=%cd %p/utar://
+View=%view{ascii} /usr/libexec/mc/ext.d/archive.sh view tar.xz
+
+
 [ISO9660]
 Shell=.iso
 ShellIgnoreCase=true
