diff -Naurp libburn-0.2.6.2/configure.ac libburn-0.2.6.2-p/configure.ac
--- libburn-0.2.6.2/configure.ac	2006-12-19 06:04:10.000000000 +0100
+++ libburn-0.2.6.2-p/configure.ac	2006-12-20 01:34:43.000000000 +0100
@@ -25,7 +25,7 @@ dnl if MAJOR or MINOR version changes, b
 dnl
 BURN_MAJOR_VERSION=0
 BURN_MINOR_VERSION=2
-BURN_MICRO_VERSION=6.2
+BURN_MICRO_VERSION=6
 BURN_INTERFACE_AGE=0
 BURN_BINARY_AGE=0
 BURN_VERSION=$BURN_MAJOR_VERSION.$BURN_MINOR_VERSION.$BURN_MICRO_VERSION
@@ -63,9 +63,6 @@ AC_C_BIGENDIAN
 dnl Large file support
 AC_SYS_LARGEFILE
 AC_FUNC_FSEEKO
-if test ! $ac_cv_func_fseeko; then
-   AC_ERROR([Libburn requires largefile support.])
-fi
 
 AC_PROG_LIBTOOL
 AC_SUBST(LIBTOOL_DEPS)
