diff -Naur apm-2.0.1/package.json apm-2.0.1.new/package.json
--- apm-2.0.1/package.json	2018-08-06 16:25:17.000000000 +0200
+++ apm-2.0.1.new/package.json	2018-08-07 09:14:01.265930683 +0200
@@ -29,11 +29,9 @@
     "fs-plus": "2.x",
     "git-utils": "^4.0",
     "hosted-git-info": "^2.1.4",
-    "keytar": "^4.0",
+    "keytar": "^4.2.1",
     "mv": "2.0.0",
     "ncp": "~0.5.1",
-    "node-gyp": "3.4.0",
-    "npm": "6.2.0",
     "open": "0.0.4",
     "plist": "git+https://github.com/nathansobo/node-plist.git",
     "q": "~0.9.7",
diff -Naur apm-2.0.1/src/dedupe.coffee apm-2.0.1.new/src/dedupe.coffee
--- apm-2.0.1/src/dedupe.coffee	2018-08-06 16:25:17.000000000 +0200
+++ apm-2.0.1.new/src/dedupe.coffee	2018-08-07 09:13:22.939931925 +0200
@@ -17,7 +17,7 @@
     @atomPackagesDirectory = path.join(@atomDirectory, 'packages')
     @atomNodeDirectory = path.join(@atomDirectory, '.node-gyp')
     @atomNpmPath = require.resolve('npm/bin/npm-cli')
-    @atomNodeGypPath = require.resolve('node-gyp/bin/node-gyp')
+    @atomNodeGypPath = require.resolve('npm/node_modules/node-gyp/bin/node-gyp')
 
   parseOptions: (argv) ->
     options = yargs(argv).wrap(100)
diff -Naur apm-2.0.1/src/install.coffee apm-2.0.1.new/src/install.coffee
--- apm-2.0.1/src/install.coffee	2018-08-06 16:25:17.000000000 +0200
+++ apm-2.0.1.new/src/install.coffee	2018-08-07 09:13:23.003931923 +0200
@@ -84,7 +84,7 @@
     opts = {env, cwd: @atomDirectory}
     opts.streaming = true if @verbose
 
-    atomNodeGypPath = process.env.ATOM_NODE_GYP_PATH or require.resolve('node-gyp/bin/node-gyp')
+    atomNodeGypPath = process.env.ATOM_NODE_GYP_PATH or require.resolve('npm/node_modules/node-gyp/bin/node-gyp')
     @fork atomNodeGypPath, installNodeArgs, opts, (code, stderr='', stdout='') ->
       if code is 0
         callback()
