diff -Naur recordmydesktop-0.2.4/configure recordmydesktop-0.2.4-p/configure
--- recordmydesktop-0.2.4/configure	2006-08-11 03:57:11.000000000 +0200
+++ recordmydesktop-0.2.4-p/configure	2006-08-25 17:25:09.000000000 +0200
@@ -5391,7 +5391,7 @@
 
 LDFLAGS="$LD_FLAGS -L$x_libraries $X_LIBS $X_EXTRA_LIBS $X_PRE_LIBS"
 
-CFLAGS="${CFLAGS} -O3 -Wall -D_THREAD_SAFE -pthread"
+CFLAGS="${CFLAGS}  -Wall -D_THREAD_SAFE -pthread"
 
 
 
diff -Naur recordmydesktop-0.2.4/configure.ac recordmydesktop-0.2.4-p/configure.ac
--- recordmydesktop-0.2.4/configure.ac	2006-08-11 03:33:02.000000000 +0200
+++ recordmydesktop-0.2.4-p/configure.ac	2006-08-25 17:25:09.000000000 +0200
@@ -28,7 +28,7 @@
 
 LDFLAGS="$LD_FLAGS -L$x_libraries $X_LIBS $X_EXTRA_LIBS $X_PRE_LIBS"
 
-CFLAGS="${CFLAGS} -O3 -Wall -D_THREAD_SAFE -pthread"
+CFLAGS="${CFLAGS}  -Wall -D_THREAD_SAFE -pthread"
 
 
 
diff -Naur recordmydesktop-0.2.4/src/Makefile.am recordmydesktop-0.2.4-p/src/Makefile.am
--- recordmydesktop-0.2.4/src/Makefile.am	2006-07-29 02:46:08.000000000 +0200
+++ recordmydesktop-0.2.4-p/src/Makefile.am	2006-08-25 17:25:45.000000000 +0200
@@ -24,5 +24,5 @@
 
 INCLUDES= $(all_includes) -I../include -I$x_includes
 
-recordmydesktop_LDFLAGS = -D_THREAD_SAFE -pthread -Wall -O3
+recordmydesktop_LDFLAGS = -D_THREAD_SAFE -pthread -Wall 
 
diff -Naur recordmydesktop-0.2.4/src/Makefile.in recordmydesktop-0.2.4-p/src/Makefile.in
--- recordmydesktop-0.2.4/src/Makefile.in	2006-08-11 04:04:44.000000000 +0200
+++ recordmydesktop-0.2.4-p/src/Makefile.in	2006-08-25 17:25:43.000000000 +0200
@@ -74,7 +74,7 @@
 
 INCLUDES = $(all_includes) -I../include -I$x_includes
 
-recordmydesktop_LDFLAGS = -D_THREAD_SAFE -pthread -Wall -O3
+recordmydesktop_LDFLAGS = -D_THREAD_SAFE -pthread -Wall 
 mkinstalldirs = $(SHELL) $(top_srcdir)/mkinstalldirs
 CONFIG_HEADER = ../config.h
 CONFIG_CLEAN_FILES = 
