diff --git a/m4/ax_python.m4 b/m4/ax_python.m4
index 17975e0..ba13a68 100644
--- a/m4/ax_python.m4
+++ b/m4/ax_python.m4
@@ -59,7 +59,7 @@ AC_DEFUN([AX_PYTHON],
 [
 AC_MSG_CHECKING(for python build information)
 AC_MSG_RESULT([])
-for python in python2.6 python2.5 python2.4 python2.3 python2.2 python2.1 python; do
+for python in python2.7 python2.6 python2.5 python2.4 python2.3 python2.2 python2.1 python; do
     AC_CHECK_PROGS(PYTHON_BIN, [$python])
     ax_python_bin=$PYTHON_BIN
     if test x$ax_python_bin != x; then
