diff -Naur FreeNX-0_5_0-LWE.orig/freenx-server/node.conf.sample FreeNX-0_5_0-LWE/freenx-server/node.conf.sample
--- FreeNX-0_5_0-LWE.orig/freenx-server/node.conf.sample	2005-08-07 11:53:57.000000000 +0200
+++ FreeNX-0_5_0-LWE/freenx-server/node.conf.sample	2006-06-24 01:11:09.000000000 +0200
@@ -334,7 +334,7 @@
 #########################################################################
 
 # When set to 1 this will automatically resume started sessions
-#ENABLE_AUTORECONNECT="0"
+ENABLE_AUTORECONNECT="1"
 
 # When set to 1 this will automatically resume started sessions
 # but only if an older client version is used
@@ -371,7 +371,7 @@
 
 # The number of seconds we wait for the nxagent to start before
 # deciding startup has failed
-#AGENT_STARTUP_TIMEOUT="60"
+AGENT_STARTUP_TIMEOUT="60"
 
 # The font server the agent will use. If set to "" no font server is used.
 # For this to do any good, the client has to have the same font server set
