diff -Naur FreeNX-0_5_0-LWE.orig/freenx-server/nxserver FreeNX-0_5_0-LWE/freenx-server/nxserver
--- FreeNX-0_5_0-LWE.orig/freenx-server/nxserver	2005-08-07 11:53:57.000000000 +0200
+++ FreeNX-0_5_0-LWE/freenx-server/nxserver	2006-03-15 01:43:31.000000000 +0100
@@ -785,7 +785,7 @@
 	# Pick one based on "round-robin random"
 	SERVER_HOSTS=( $SERVER_LOADBALANCING )
 	SERVER_NR_OF_HOSTS=${#SERVER_HOSTS[@]}
-	let SERVER_NR=(RANDOM % SERVER_NR_OF_HOSTS)
+	let SERVER_NR=$(($RANDOM % $SERVER_NR_OF_HOSTS))
 	SERVER_HOST_RR=${SERVER_HOSTS[$SERVER_NR]}
 	echo $SERVER_HOST_RR
 }
