--- llvm-project-13.0.0.src/libclc/CMakeLists.txt	2021-09-24 18:18:10.000000000 +0200
+++ llvm-project-13.0.0.src.new/libclc/CMakeLists.txt	2021-12-23 11:46:27.068080684 +0100
@@ -183,7 +183,7 @@
 
 # pkg-config file
 configure_file( libclc.pc.in libclc.pc @ONLY )
-install( FILES ${CMAKE_CURRENT_BINARY_DIR}/libclc.pc DESTINATION ${CMAKE_INSTALL_DATADIR}/pkgconfig )
+install( FILES ${CMAKE_CURRENT_BINARY_DIR}/libclc.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig )
 install( DIRECTORY generic/include/clc DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} )
 
 if( ENABLE_RUNTIME_SUBNORMAL )
