--- libclc/configure.py	2015-03-19 16:55:24.000000000 +0100
+++ libclc.new/configure.py	2015-03-19 18:12:54.000000000 +0100
@@ -71,7 +71,7 @@
 llvm_system_libs = llvm_config(['--system-libs'])
 llvm_bindir = llvm_config(['--bindir'])
 llvm_core_libs = llvm_config(['--libs', 'core', 'bitreader', 'bitwriter']) + ' ' + \
-                 llvm_system_libs + ' ' + \
+                 llvm_system_libs + ' -latomic ' + \
                  llvm_config(['--ldflags'])
 llvm_cxxflags = llvm_config(['--cxxflags']) + ' -fno-exceptions -fno-rtti'
 llvm_libdir = llvm_config(['--libdir'])
