diff -aur a/src/bin/url_utils.c b/src/bin/url_utils.c
--- a/src/bin/url_utils.c	2011-09-06 16:11:10.416405202 -0500
+++ b/src/bin/url_utils.c	2011-09-06 16:11:52.184404115 -0500
@@ -28,7 +28,6 @@
 #include <unistd.h>
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 #include <Ecore_File.h>
diff -aur a/src/modules/activity/photo/photo_slideshow_view.c b/src/modules/activity/photo/photo_slideshow_view.c
--- a/src/modules/activity/photo/photo_slideshow_view.c	2011-09-06 16:11:10.417405202 -0500
+++ b/src/modules/activity/photo/photo_slideshow_view.c	2011-09-07 01:48:08.324931631 -0500
@@ -129,7 +129,6 @@
 {
     Smart_Data *sd = data;
     evas_object_show(sd->controls);
-    elm_notify_timer_init(sd->controls);
 }
 
 static void
diff -aur a/src/modules/browser/valhalla/valhalla.c b/src/modules/browser/valhalla/valhalla.c
--- a/src/modules/browser/valhalla/valhalla.c	2011-09-06 16:11:10.419405202 -0500
+++ b/src/modules/browser/valhalla/valhalla.c	2011-09-07 01:26:21.145980232 -0500
@@ -282,9 +282,9 @@
     valhalla_db_stmt_t *stmt;
     const valhalla_db_metares_t *metares;
     valhalla_db_restrict_t r1 =
-        VALHALLA_DB_RESTRICT_STR(EQUAL, "title", NULL, priority);
+        VALHALLA_DB_RESTRICT_STR(EQUAL, "title", NULL, NULL, priority);
     valhalla_db_restrict_t r2 =
-        VALHALLA_DB_RESTRICT_STR(EQUAL, "track", NULL, priority);
+        VALHALLA_DB_RESTRICT_STR(EQUAL, "track", NULL, NULL, priority);
     unsigned int i;
 
     struct {
@@ -326,9 +326,9 @@
     valhalla_db_stmt_t *stmt;
     const valhalla_db_metares_t *metares;
     valhalla_db_item_t search =
-        VALHALLA_DB_SEARCH_TEXT(item->meta, NIL, item->priority);
+        VALHALLA_DB_SEARCH_TEXT(item->meta, NIL, NULL, item->priority);
     valhalla_db_restrict_t r1 =
-        VALHALLA_DB_RESTRICT_STR(IN, id->meta, id->data, item->priority);
+        VALHALLA_DB_RESTRICT_STR(IN, id->meta, id->data, NULL, item->priority);
     valhalla_db_restrict_t *r = NULL;
 
     if (tree_meta[id->it].level > LEVEL_ONE)
