diff -Nur gadmin-samba-0.2.5.orig/Makefile.in gadmin-samba-0.2.5/Makefile.in
--- gadmin-samba-0.2.5.orig/Makefile.in	2008-04-28 20:59:27.000000000 +0200
+++ gadmin-samba-0.2.5/Makefile.in	2008-05-02 21:59:24.000000000 +0200
@@ -670,12 +670,12 @@
 	if test -d $(DESTDIR)$(bindir) && test ! -f $(DESTDIR)$(bindir)/gadmin-samba-pdf; then \
 	   cp scripts/gadmin-samba-pdf $(bindir)/; \
 	fi;
-	@echo Installing example logon script to static /home/netlogon.
-	if test ! -d /home/netlogon; then \
-	   mkdir -p /home/netlogon; \
+	@echo Installing example logon script to static $(DESTDIR)$(sysconfdir)/gadmin-samba/netlogon.
+	if test ! -d $(DESTDIR)$(sysconfdir)/gadmin-samba/netlogon; then \
+	   mkdir -p $(DESTDIR)$(sysconfdir)/gadmin-samba/netlogon; \
 	fi;
-	if test -d /home/netlogon && test ! -f /home/netlogon/example.bat; then \
-	   cp scripts/example.bat /home/netlogon/; \
+	if test -d $(DESTDIR)$(sysconfdir)/gadmin-samba/netlogon && test ! -f $(DESTDIR)$(sysconfdir)/gadmin-samba/netlogon/example.bat; then \
+	   cp scripts/example.bat $(DESTDIR)$(sysconfdir)/gadmin-samba/netlogon/; \
 	fi;
 	@echo ""
 	@echo "Start GADMIN-SAMBA by typing gadmin-samba in a terminal"
