--- th-lift-0.8.2/th-lift.cabal	2001-09-09 03:46:40.000000000 +0200
+++ th-lift-0.8.2.new/th-lift.cabal	2022-11-15 16:47:48.242747486 +0100
@@ -26,7 +26,7 @@
     old versions of their respective libraries, as the same @Lift@ instances
     are also present upstream on newer versions.
 Category:           Language
-Tested-With:        GHC==7.0.4, GHC==7.2.2, GHC==7.4.2, GHC==7.6.3, GHC==7.8.4, GHC==7.10.3, GHC==8.0.2, GHC==8.2.2, GHC==8.4.4, GHC==8.6.5, GHC==8.8.3, GHC==8.10.1
+Tested-With:        GHC==7.0.4, GHC==7.2.2, GHC==7.4.2, GHC==7.6.3, GHC==7.8.4, GHC==7.10.3, GHC==8.0.2, GHC==8.2.2, GHC==8.4.4, GHC==8.6.5, GHC==8.8.4, GHC==8.10.4, GHC==9.0.1
 build-type:         Simple
 Extra-source-files: CHANGELOG.md
 
@@ -47,7 +47,7 @@
   Build-Depends:    base              >= 4.3  && < 5,
                     ghc-prim,
                     th-abstraction   >= 0.2.3 && < 0.5,
-                    template-haskell >= 2.5   && < 2.18
+                    template-haskell >= 2.5   && < 2.20
   ghc-options:      -Wall
 
 Test-Suite test
@@ -57,6 +57,8 @@
   Hs-Source-Dirs:   t
   other-modules:    Foo
   ghc-options:      -Wall
+  if impl(ghc >= 9.0)
+    ghc-options:    -fenable-th-splice-warnings
   Build-Depends:    base,
                     ghc-prim,
                     th-lift,
