--- th-lift-0.8.4/th-lift.cabal	2001-09-09 03:46:40.000000000 +0200
+++ th-lift-0.8.4.new/th-lift.cabal	2024-07-11 19:01:09.716693012 +0200
@@ -46,8 +46,8 @@
   Hs-Source-Dirs:   src
   Build-Depends:    base             >= 4.3 && < 5,
                     ghc-prim,
-                    th-abstraction   >= 0.3 && < 0.7,
-                    template-haskell >= 2.5 && < 2.22
+                    th-abstraction   >= 0.3 && < 0.8,
+                    template-haskell >= 2.5 && < 2.23
   ghc-options:      -Wall
 
 Test-Suite test
