diff -Naur dvb-apps/util/dst-utils/dst_test.c dvb-apps-p/util/dst-utils/dst_test.c
--- dvb-apps/util/dst-utils/dst_test.c	2018-02-11 15:57:05.135990355 +0100
+++ dvb-apps-p/util/dst-utils/dst_test.c	2018-02-11 15:56:29.330965527 +0100
@@ -111,15 +111,6 @@
 	return 0;
 }
 
-static int dst_set_pid(int cafd)
-{
-	if ((ioctl(cafd, CA_SET_PID)) < 0) {
-		printf("%s: ioctl failed ..\n", __FUNCTION__);
-		return -1;
-	}
-
-	return 0;
-}
 
 static int dst_get_descr(int cafd)
 {
@@ -229,10 +220,6 @@
 				printf("%s: Reset\n", __FUNCTION__);
 				dst_reset(cafd);
 				break;
-			case 'p':
-				printf("%s: PID\n", __FUNCTION__);
-				dst_set_pid(cafd);
-				break;
 			case 'g':
 				printf("%s: Get Desc\n", __FUNCTION__);
 				dst_get_descr(cafd);
