diff -Naur jq-jq-1.5/jv_parse.c jq-jq-1.5-p/jv_parse.c
--- jq-jq-1.5/jv_parse.c	2015-08-18 06:25:04.000000000 +0200
+++ jq-jq-1.5-p/jv_parse.c	2018-03-17 00:54:33.254744098 +0100
@@ -383,7 +383,7 @@
 
 static void tokenadd(struct jv_parser* p, char c) {
   assert(p->tokenpos <= p->tokenlen);
-  if (p->tokenpos == p->tokenlen) {
+  if (p->tokenpos >= (p->tokenlen - 1)) {
     p->tokenlen = p->tokenlen*2 + 256;
     p->tokenbuf = jv_mem_realloc(p->tokenbuf, p->tokenlen);
   }
@@ -485,7 +485,7 @@
     TRY(value(p, v));
   } else {
     // FIXME: better parser
-    p->tokenbuf[p->tokenpos] = 0; // FIXME: invalid
+    p->tokenbuf[p->tokenpos] = 0;
     char* end = 0;
     double d = jvp_strtod(&p->dtoa, p->tokenbuf, &end);
     if (end == 0 || *end != 0)
