diff -Naur Singular-3-1-1//Singular/Makefile.in Singular-3-1-1nzw//Singular/Makefile.in
--- Singular-3-1-1//Singular/Makefile.in	2010-02-08 19:32:36.000000000 +0100
+++ Singular-3-1-1nzw//Singular/Makefile.in	2010-09-19 18:30:36.365000067 +0200
@@ -22,12 +22,12 @@
 exec_prefix 	= @exec_prefix@
 libdir 		= @libdir@
 # program executable goes here
-bindir		= @bindir@
+bindir		= ${prefix}/bin
 # includes are taken from here
 includedir	= @includedir@
-slibdir		= ${prefix}/LIB
-install_bindir  = ${install_prefix}/${SINGUNAME}
-install_slibdir = ${install_prefix}/LIB
+slibdir		= ${prefix}/lib/singular/LIB
+install_bindir  = ${install_prefix}/bin
+install_slibdir = ${install_prefix}/lib/singular/LIB
 
 ##
 ## various programs
@@ -524,7 +524,6 @@
 	${INSTALL_PROGRAM} ${SINGULAR} ${bindir}/Singular${EXEC_EXT};
 
 installslib: LIB
-	test -r ${slibdir} || ${LN_S} `pwd`/LIB ${slibdir}
 
 installslib-nolns: LIB
 	${MKINSTALLDIRS} ${slibdir}
