--- these-1.2.1/these.cabal	2001-09-09 03:46:40.000000000 +0200
+++ these-1.2.1.new/these.cabal	2024-07-10 00:32:20.477787685 +0200
@@ -66,7 +66,7 @@
   -- Bifunctor type class comes from bifunctor-classes-compat
   build-depends:
       assoc     >=1.1.1   && <1.2
-    , hashable  >=1.4.4.0 && <1.5
+    , hashable  >=1.4.4.0 && <1.6
 
   if !impl(ghc >=9.6)
     build-depends: foldable1-classes-compat >=0.1 && <0.2
