diff -Naur vim64.orig/runtime/filetype.vim vim64/runtime/filetype.vim
--- vim64.orig/runtime/filetype.vim	2005-10-13 12:19:37.000000000 +0200
+++ vim64/runtime/filetype.vim	2005-11-15 01:02:37.000000000 +0100
@@ -1745,6 +1745,8 @@
 " Z-Shell script
 au BufNewFile,BufRead zsh*,zlog*		setf zsh
 
+" FrugalBuild file
+au BufNewFile,BufRead FrugalBuild,*.install		setf sh
 
 " Generic configuration file (check this last, it's just guessing!)
 au BufNewFile,BufRead,StdinReadPost *
diff -Naur vim64.orig/runtime/syntax/sh.vim vim64/runtime/syntax/sh.vim
--- vim64.orig/runtime/syntax/sh.vim	2005-09-06 19:25:51.000000000 +0200
+++ vim64/runtime/syntax/sh.vim	2005-11-15 01:21:08.000000000 +0100
@@ -409,6 +409,10 @@
  endif
 endif
 
+" Useful FrugalBuild Keywords: {{{1
+" ===================
+syn keyword shStatement Fmessage Fdie Fcd Fmkdir Frm Fcp Fcpr Fmv Finstallrel Finstall Fdirschmod Ffileschmod Fdirschown Ffileschown Fexe Fexerel Ffile Ffilerel Fdoc Fdocrel Fln Fsed Fdeststrip Fpatch Fpatchall Fconf Fmake Fmakeinstall Fbuild Frcd Frcd2 Facu Fsort Funpack_makeself Fautoconfize Finclude Flasttar Fsplit
+
 " Synchronization: {{{1
 " ================
 if !exists("sh_minlines")
