diff -Naur vim72.orig/runtime/vimrc_example.vim vim72/runtime/vimrc_example.vim
--- vim72.orig/runtime/vimrc_example.vim	2008-07-02 22:11:53.000000000 +0200
+++ vim72/runtime/vimrc_example.vim	2008-08-15 01:54:10.000000000 +0200
@@ -21,11 +21,7 @@
 " allow backspacing over everything in insert mode
 set backspace=indent,eol,start
 
-if has("vms")
-  set nobackup		" do not keep a backup file, use versions instead
-else
-  set backup		" keep a backup file
-endif
+set nobackup            " do not keep a backup file, use versions instead
 set history=50		" keep 50 lines of command line history
 set ruler		" show the cursor position all the time
 set showcmd		" display incomplete commands
@@ -42,15 +38,16 @@
 inoremap <C-U> <C-G>u<C-U>
 
 " In many terminal emulators the mouse works just fine, thus enable it.
-if has('mouse')
-  set mouse=a
-endif
+"if has('mouse')
+"  set mouse=a
+"endif
 
 " Switch syntax highlighting on, when the terminal has colors
 " Also switch on highlighting the last used search pattern.
 if &t_Co > 2 || has("gui_running")
   syntax on
   set hlsearch
+  set bg=dark
 endif
 
 " Only do this part when compiled with support for autocommands.
