diff -Naur gcompris-8.4.7/src/boards/python/clickanddraw.py gcompris-8.4.7-p/src/boards/python/clickanddraw.py
--- gcompris-8.4.7/src/boards/python/clickanddraw.py	2008-02-14 01:45:14.000000000 +0100
+++ gcompris-8.4.7-p/src/boards/python/clickanddraw.py	2008-11-16 18:41:48.000000000 +0100
@@ -107,6 +107,6 @@
           self.gamewon = 1
           self.timeout = gobject.timeout_add(1500, self.lauch_bonus) # The level is complete -> Bonus display
 
-        else : #Action to execute if the selected point isn't the last one of this level
+        else : #Action to execute if the selected point is not the last one of this level
           self.POINT[(idpt+1)].set(fill_color='blue') #Set color in blue to next point. Too easy ???
           self.actu = self.actu+1 #self.actu update to set it at actual value of selected point
diff -Naur gcompris-8.4.7/src/boards/python/drawnumber.py gcompris-8.4.7-p/src/boards/python/drawnumber.py
--- gcompris-8.4.7/src/boards/python/drawnumber.py	2008-02-14 01:45:14.000000000 +0100
+++ gcompris-8.4.7-p/src/boards/python/drawnumber.py	2008-11-16 18:41:26.000000000 +0100
@@ -185,7 +185,7 @@
           self.gamewon = 1
           self.timeout = gobject.timeout_add(1500, self.lauch_bonus) # The level is complete -> Bonus display
 
-        else : #Action to execute if the selected point isn't the last one of this level
+        else : #Action to execute if the selected point is not the last one of this level
           #self.POINT[(idpt+1)].set(fill_color='blue') #Set color in blue to next point. Too easy ???
           self.actu=self.actu+1 #self.actu update to set it at actual value of selected point
 
