--- ./src/java/org/apache/fop/render/ps/AbstractPSTranscoder.java.orig	2008-05-03 02:41:12.000000000 +0200
+++ ./src/java/org/apache/fop/render/ps/AbstractPSTranscoder.java	2008-05-03 02:41:45.000000000 +0200
@@ -36,7 +36,7 @@
 import org.apache.fop.fonts.FontSetup;
 import org.apache.fop.svg.AbstractFOPTranscoder;
 import org.apache.xmlgraphics.java2d.ps.AbstractPSDocumentGraphics2D;
-import org.apache.xmlgraphics.java2d.ps.TextHandler;
+import org.apache.xmlgraphics.java2d.TextHandler;
 
 import org.w3c.dom.Document;
 import org.w3c.dom.svg.SVGLength;
--- ./src/java/org/apache/fop/render/ps/NativeTextHandler.java.orig	2008-05-03 02:42:46.000000000 +0200
+++ ./src/java/org/apache/fop/render/ps/NativeTextHandler.java	2008-05-03 02:43:02.000000000 +0200
@@ -29,7 +29,7 @@
 import org.apache.fop.fonts.FontTriplet;
 
 import org.apache.xmlgraphics.java2d.ps.PSGraphics2D;
-import org.apache.xmlgraphics.java2d.ps.TextHandler;
+import org.apache.xmlgraphics.java2d.TextHandler;
 import org.apache.xmlgraphics.ps.PSGenerator;
 
 /**
