--- rss-glx_0.8.1-orig/reallyslick/cpp_src/skyrocket_sound.cpp
+++ rss-glx_0.8.1/reallyslick/cpp_src/skyrocket_sound.cpp
@@ -24,6 +24,7 @@
 #include <AL/al.h>
 #include <AL/alut.h>
 
+#include <cstdlib>
 #include <list>
 
 #include "loadTexture.h"
--- rss-glx_0.8.1-orig/reallyslick/Implicit/impSurface.cpp
+++ rss-glx_0.8.1/reallyslick/Implicit/impSurface.cpp
@@ -21,6 +21,7 @@
 #include <Implicit/impSurface.h>
 #include <math.h>
 #include <iostream>
+#include <cstring>
 
 
 
