--- tg_owt/src/modules/video_coding/codecs/h264/h264_decoder_impl.cc	2022-05-29 18:25:09.853627663 +0200
+++ tg_owt.new/src/modules/video_coding/codecs/h264/h264_decoder_impl.cc	2022-05-29 18:26:08.001051089 +0200
@@ -202,7 +202,7 @@
   // a pointer |this|.
   av_context_->opaque = this;
 
-  AVCodec* codec = avcodec_find_decoder(av_context_->codec_id);
+  const AVCodec* codec = avcodec_find_decoder(av_context_->codec_id);
   if (!codec) {
     // This is an indication that FFmpeg has not been initialized or it has not
     // been compiled/initialized with the correct set of codecs.
