--- a/os-prober.orig	2008-01-07 16:59:25.000000000 +0100
+++ b/os-prober	2008-01-07 17:00:54.000000000 +0100
@@ -36,8 +36,7 @@
 
 	# Also detect OSes on LVM volumes (assumes LVM is active)
 	if type lvs >/dev/null 2>&1; then
-		echo "$(log-output -t os-prober --pass-stdout \
-			lvs --noheadings --separator : -o vg_name,lv_name |
+		echo "$(lvs --noheadings --separator : -o vg_name,lv_name 2>/dev/null|
 			sed "s|-|--|g;s|^[[:space:]]*\(.*\):\(.*\)$|/dev/mapper/\1-\2|")"
 	fi
 }
