diff -Naur os-prober.orig/os-probes/mounted/common/90linux-distro os-prober/os-probes/mounted/common/90linux-distro
--- os-prober.orig/os-probes/mounted/common/90linux-distro	2006-06-16 00:24:48.000000000 +0200
+++ os-prober/os-probes/mounted/common/90linux-distro	2006-06-16 20:39:12.000000000 +0200
@@ -14,7 +14,10 @@
 # can come first.
 # TODO: look for ld-linux.so on arches that have it
 if ls $dir/lib/ld*.so* >/dev/null 2>/dev/null; then
-	if [ -e "$dir/etc/debian_version" ]; then
+	if [ -e "$dir/etc/frugalware-release" ]; then
+		short="Frugalware"
+		long="$(cat $dir/etc/frugalware-release) - $(ls $dir/boot/vmlinuz-*|sed -n 's|.*vmlinuz-||;$ p')"
+	elif [ -e "$dir/etc/debian_version" ]; then
 		short="Debian"
 		long=$(printf "Debian GNU/Linux (%s)\n" "$(cat $dir/etc/debian_version)")
 	# RPM derived distributions may also have a redhat-release or
