diff --git a/waflib/Context.py b/waflib/Context.py
index ab6b154..ac858d6 100644
--- a/waflib/Context.py
+++ b/waflib/Context.py
@@ -2,7 +2,7 @@
 # encoding: utf-8
 # WARNING! Do not edit! https://waf.io/book/index.html#_obtaining_the_waf_file
 
-import os,re,imp,sys
+import os,re,types,sys
 from waflib import Utils,Errors,Logs
 import waflib.Node
 HEXVERSION=0x2000e00
@@ -344,7 +344,7 @@ def load_module(path,encoding=None):
 		return cache_modules[path]
 	except KeyError:
 		pass
-	module=imp.new_module(WSCRIPT_FILE)
+	module=types.ModuleType(WSCRIPT_FILE)
 	try:
 		code=Utils.readf(path,m='r',encoding=encoding)
 	except EnvironmentError:
