diff --git a/waflib/ConfigSet.py b/waflib/ConfigSet.py
index 8212586..94a08d7 100644
--- a/waflib/ConfigSet.py
+++ b/waflib/ConfigSet.py
@@ -4,7 +4,7 @@
 
 import copy,re,os
 from waflib import Logs,Utils
-re_imp=re.compile('^(#)*?([^#=]*?)\ =\ (.*?)$',re.M)
+re_imp=re.compile(r'^(#)*?([^#=]*?)\ =\ (.*?)$',re.M)
 class ConfigSet(object):
 	__slots__=('table','parent')
 	def __init__(self,filename=None):
diff --git a/waflib/Utils.py b/waflib/Utils.py
index 273ebb0..d7f9436 100644
--- a/waflib/Utils.py
+++ b/waflib/Utils.py
@@ -440,7 +440,7 @@ def unversioned_sys_platform():
 		return s
 	if s=='cli'and os.name=='nt':
 		return'win32'
-	return re.split('\d+$',s)[0]
+	return re.split(r'\d+$',s)[0]
 def nada(*k,**kw):
 	pass
 class Timer(object):
diff --git a/waflib/ansiterm.py b/waflib/ansiterm.py
index 1d8bc78..1ab9c5c 100644
--- a/waflib/ansiterm.py
+++ b/waflib/ansiterm.py
@@ -175,7 +175,7 @@ else:
 			self._csinfo.bVisible=0
 			windll.kernel32.SetConsoleCursorInfo(self.hconsole,byref(self._csinfo))
 		ansi_command_table={'A':move_up,'B':move_down,'C':move_right,'D':move_left,'E':next_line,'F':prev_line,'G':set_column,'H':set_cursor,'f':set_cursor,'J':clear_screen,'K':clear_line,'h':show_cursor,'l':hide_cursor,'m':set_color,'s':push_cursor,'u':pop_cursor,}
-		ansi_tokens=re.compile('(?:\x1b\[([0-9?;]*)([a-zA-Z])|([^\x1b]+))')
+		ansi_tokens=re.compile(r'(?:\x1b\[([0-9?;]*)([a-zA-Z])|([^\x1b]+))')
 		def write(self,text):
 			try:
 				wlock.acquire()
