diff -Naur mesa-18.1.0/src/gallium/drivers/llvmpipe/lp_screen.c mesa-18.1.0.new/src/gallium/drivers/llvmpipe/lp_screen.c
--- mesa-18.1.0/src/gallium/drivers/llvmpipe/lp_screen.c	2018-05-19 01:28:40.000000000 +0200
+++ mesa-18.1.0.new/src/gallium/drivers/llvmpipe/lp_screen.c	2018-05-22 11:16:16.731745521 +0200
@@ -449,6 +449,13 @@
    if (!format_desc)
       return FALSE;
 
+   if ((bind & PIPE_BIND_RENDER_TARGET) &&
+       format != PIPE_FORMAT_R9G9B9E5_FLOAT &&
+       format != PIPE_FORMAT_R11G11B10_FLOAT &&
+       util_format_is_float(format)) {
+      return FALSE;
+   }
+
    assert(target == PIPE_BUFFER ||
           target == PIPE_TEXTURE_1D ||
           target == PIPE_TEXTURE_1D_ARRAY ||
diff -Naur mesa-18.1.0/src/gallium/drivers/softpipe/sp_screen.c mesa-18.1.0.new/src/gallium/drivers/softpipe/sp_screen.c
--- mesa-18.1.0/src/gallium/drivers/softpipe/sp_screen.c	2018-05-19 01:28:40.000000000 +0200
+++ mesa-18.1.0.new/src/gallium/drivers/softpipe/sp_screen.c	2018-05-22 11:16:16.731745521 +0200
@@ -397,6 +397,13 @@
    if (!format_desc)
       return FALSE;
 
+   if ((bind & PIPE_BIND_RENDER_TARGET) &&
+       format != PIPE_FORMAT_R9G9B9E5_FLOAT &&
+       format != PIPE_FORMAT_R11G11B10_FLOAT &&
+       util_format_is_float(format)) {
+      return FALSE;
+   }
+
    if (sample_count > 1)
       return FALSE;
 
