From 1d24e6ec1d13c2b88b858997900aa8b69de92b13 Mon Sep 17 00:00:00 2001
From: Ben Skeggs <bskeggs@redhat.com>
Date: Mon, 06 Jun 2011 00:17:59 +0000
Subject: dri/nouveau: fix gnome-shell segfault

Signed-off-by: Ben Skeggs <bskeggs@redhat.com>
---
diff --git a/src/mesa/drivers/dri/nouveau/nouveau_context.c b/src/mesa/drivers/dri/nouveau/nouveau_context.c
index 71dde41..22b9957 100644
--- a/src/mesa/drivers/dri/nouveau/nouveau_context.c
+++ b/src/mesa/drivers/dri/nouveau/nouveau_context.c
@@ -261,7 +261,7 @@ nouveau_update_renderbuffers(__DRIcontext *dri_ctx, __DRIdrawable *draw)
 		assert(!ret);
 	}
 
-	_mesa_resize_framebuffer(NULL, fb, draw->w, draw->h);
+	_mesa_resize_framebuffer(ctx, fb, draw->w, draw->h);
 }
 
 static void
--
cgit v0.8.3-6-g21f6
