diff -Naur Mesa-6.5.1.orig/configs/linux-dri-x86-64 Mesa-6.5.1/configs/linux-dri-x86-64
--- Mesa-6.5.1.orig/configs/linux-dri-x86-64	2006-09-16 11:47:41.000000000 +0200
+++ Mesa-6.5.1/configs/linux-dri-x86-64	2006-09-16 11:48:11.000000000 +0200
@@ -20,5 +20,5 @@
 # the new interface.  i810 are missing because there is no x86-64
 # system where they could *ever* be used.
 #
-DRI_DIRS = i915 mach64 mga r128 r200 radeon tdfx unichrome savage r300
+DRI_DIRS = i915 i965 mach64 mga r128 r200 radeon tdfx unichrome savage r300
 
