diff -Naur Mesa-6.5.1.orig/configs/default Mesa-6.5.1/configs/default
--- Mesa-6.5.1.orig/configs/default	2006-10-07 12:12:00.000000000 +0200
+++ Mesa-6.5.1/configs/default	2006-10-07 12:13:12.000000000 +0200
@@ -80,7 +80,7 @@
 
 # Installation directories (for make install)
 INSTALL_DIR = /usr/local
-DRI_DRIVER_INSTALL_DIR = /usr/X11R6/lib/modules/dri
+DRI_DRIVER_INSTALL_DIR = /usr/lib/dri
 
 # Where libGL will look for DRI hardware drivers
 DRI_DRIVER_SEARCH_DIR = $(DRI_DRIVER_INSTALL_DIR)
diff -Naur Mesa-6.5.1.orig/configs/linux-dri Mesa-6.5.1/configs/linux-dri
--- Mesa-6.5.1.orig/configs/linux-dri	2006-10-07 12:12:00.000000000 +0200
+++ Mesa-6.5.1/configs/linux-dri	2006-10-07 12:18:37.000000000 +0200
@@ -9,7 +9,7 @@
 CC = gcc
 CXX = g++
 
-#MKDEP = /usr/X11R6/bin/makedepend
+MKDEP = /usr/bin/makedepend
 #MKDEP = gcc -M
 #MKDEP_OPTIONS = -MF depend
 
@@ -24,7 +24,7 @@
 	-DPTHREADS -DUSE_EXTERNAL_DXTN_LIB=1 -DIN_DRI_DRIVER \
 	-DGLX_DIRECT_RENDERING -DHAVE_ALIAS -DHAVE_POSIX_MEMALIGN
 
-X11_INCLUDES = -I/usr/X11R6/include
+X11_INCLUDES = 
 
 CFLAGS = -Wall -Wmissing-prototypes -std=c99 -ffast-math \
 	$(OPT_FLAGS) $(PIC_FLAGS) $(ARCH_FLAGS) $(DEFINES) $(ASM_FLAGS)
@@ -35,7 +35,7 @@
 ASM_SOURCES = 
 
 # Library/program dependencies
-EXTRA_LIB_PATH=-L/usr/X11R6/lib
+EXTRA_LIB_PATH=-L/usr/lib
 
 LIBDRM_CFLAGS = `pkg-config --cflags libdrm`
 LIBDRM_LIB = `pkg-config --libs libdrm`
diff -Naur Mesa-6.5.1.orig/configs/linux-dri-x86-64 Mesa-6.5.1/configs/linux-dri-x86-64
--- Mesa-6.5.1.orig/configs/linux-dri-x86-64	2006-10-07 12:12:00.000000000 +0200
+++ Mesa-6.5.1/configs/linux-dri-x86-64	2006-10-07 12:19:07.000000000 +0200
@@ -11,14 +11,14 @@
 ASM_SOURCES = $(X86-64_SOURCES)
 ASM_API = $(X86-64_API)
 
-LIB_DIR = lib64
+LIB_DIR = lib
 
 # Library/program dependencies
-EXTRA_LIB_PATH=-L/usr/X11R6/lib64
+EXTRA_LIB_PATH=-L/usr/lib
 
 # ffb, gamma, and sis are missing because they have not be converted to use
 # the new interface.  i810 are missing because there is no x86-64
 # system where they could *ever* be used.
 #
-DRI_DIRS = i915tex i915 i965 mach64 mga r128 r200 radeon tdfx unichrome savage r300
+DRI_DIRS = i915tex i915 i965 mach64 mga r128 r200 radeon tdfx unichrome savage r300 nouveau
 
