--- mesa-19.0.5/src/mesa/main/context.c	2019-05-21 23:11:01.000000000 +0200
+++ mesa-19.0.5.new/src/mesa/main/context.c	2019-06-02 10:46:06.413981001 +0200
@@ -1760,10 +1760,6 @@
 
          check_init_viewport(newCtx, drawBuffer->Width, drawBuffer->Height);
       }
-      else {
-         _mesa_reference_framebuffer(&newCtx->WinSysDrawBuffer, NULL);
-         _mesa_reference_framebuffer(&newCtx->WinSysReadBuffer, NULL);
-      }
 
       if (newCtx->FirstTimeCurrent) {
          handle_first_current(newCtx);
