diff -Naur mesa-18.2.0/src/gallium/drivers/swr/Makefile.am mesa-18.2.0-p/src/gallium/drivers/swr/Makefile.am
--- mesa-18.2.0/src/gallium/drivers/swr/Makefile.am	2018-09-07 23:18:07.000000000 +0200
+++ mesa-18.2.0-p/src/gallium/drivers/swr/Makefile.am	2018-09-10 15:36:53.456889287 +0200
@@ -49,7 +49,8 @@
 # SWR_AVX_CXXFLAGS needed for intrinsic usage in swr api headers
 libmesaswr_la_CXXFLAGS = \
 	$(SWR_AVX_CXXFLAGS) \
-	$(COMMON_CXXFLAGS)
+	$(COMMON_CXXFLAGS) \
+	-fpermissive
 
 if HAVE_SWR_AVX
 libmesaswr_la_CXXFLAGS += -DHAVE_SWR_AVX
