diff -aur wmii-3.6/config.mk fw_wmii-3.6/config.mk
--- wmii-3.6/config.mk	2007-11-16 14:59:15.000000000 +0100
+++ fw_wmii-3.6/config.mk	2007-11-27 17:08:54.000000000 +0100
@@ -1,12 +1,12 @@
 # Customize below to fit your system
 
 # paths
-PREFIX = /usr/local
-BIN = ${PREFIX}/bin
-MAN = ${PREFIX}/share/man
-ETC = ${PREFIX}/etc
-LIBDIR = ${PREFIX}/lib
-INCLUDE = ${PREFIX}/include
+PREFIX = /usr
+BIN = ${DESTDIR}/${PREFIX}/bin
+MAN = ${DESTDIR}/${PREFIX}/share/man
+ETC = ${DESTDIR}/${PREFIX}/etc
+LIBDIR = ${DESTDIR}/${PREFIX}/lib
+INCLUDE = ${DESTDIR}/${PREFIX}/include
 
 # Includes and libs
 INCPATH = .:${ROOT}/include:${INCLUDE}:/usr/include
