diff -Naur genpower-1.0.2.orig/genpowerfail genpower-1.0.2/genpowerfail
--- genpower-1.0.2.orig/genpowerfail	2001-02-05 22:28:37.000000000 +0100
+++ genpower-1.0.2/genpowerfail	2004-04-03 18:23:34.000000000 +0200
@@ -6,19 +6,19 @@
 #		This script is also being run when the power comes
 #		up again (if it does in time!)
 #
-# Version:	/etc/genpowerfail 1.0.1
+# Version:	/sbin/genpowerfail 1.0.1
 #
 # Author:	Tom Webster <webster@kaiwan.com>
 #
 
 # Set the path.
-PATH=/sbin:/etc:/bin:/usr/bin
+PATH=/sbin:/bin:/usr/bin:/usr/sbin
 
 # Set location of upsstatus file
 statpath="/etc/upsstatus"
 
 # Set location of file containing PID of running shutdowns
-spidpath="/etc/shutdownpid"
+spidpath="/var/run/shutdownpid"
 
 # See what happened.
 case "$1" in
@@ -63,7 +63,7 @@
 		shutdown -c "THE POWER IS BACK"
 		;;
   	*)
-		echo "Usage: /etc/genpowerfail {start|stop}"
+		echo "Usage: /sbin/genpowerfail {start|stop}"
 		exit 1
 		;;
   esac
diff -Naur genpower-1.0.2.orig/Makefile genpower-1.0.2/Makefile
--- genpower-1.0.2.orig/Makefile	2001-02-05 22:28:37.000000000 +0100
+++ genpower-1.0.2/Makefile	2004-04-03 18:24:16.000000000 +0200
@@ -8,7 +8,7 @@
 
 NAME  		= genpower-1.0.1
 BINDIR 		= /sbin
-SCRIPTDIR 	= /etc
+SCRIPTDIR 	= /sbin
 MANDIR		= /usr/man/man8
 OWNER 		= root
 GROUP 		= root
