diff -Naur FreeNX-0_5_0-LWE.orig/freenx-server/node.conf.sample FreeNX-0_5_0-LWE/freenx-server/node.conf.sample
--- FreeNX-0_5_0-LWE.orig/freenx-server/node.conf.sample	2005-08-07 11:53:57.000000000 +0200
+++ FreeNX-0_5_0-LWE/freenx-server/node.conf.sample	2006-07-15 18:24:20.000000000 +0200
@@ -305,7 +305,7 @@
 
 # The key that contains the name of the complete path of command name
 # 'xauth'.
-#COMMAND_XAUTH=/usr/X11R6/bin/xauth
+#COMMAND_XAUTH=/usr/bin/xauth
 
 # The key that contains the name of the complete path of command name
 # 'smbmount'.
diff -Naur FreeNX-0_5_0-LWE.orig/freenx-server/nxclient FreeNX-0_5_0-LWE/freenx-server/nxclient
--- FreeNX-0_5_0-LWE.orig/freenx-server/nxclient	2005-08-07 11:53:57.000000000 +0200
+++ FreeNX-0_5_0-LWE/freenx-server/nxclient	2006-07-15 18:24:36.000000000 +0200
@@ -60,7 +60,7 @@
 else
 	dialog_interface="xmessage"
 	xmessage=$(which xmessage 2>/dev/null)
-	[ -z "$xmessage" ] && xmessage="/usr/X11R6/bin/xmessage"
+	[ -z "$xmessage" ] && xmessage="/usr/bin/xmessage"
 fi
 
 #
diff -Naur FreeNX-0_5_0-LWE.orig/freenx-server/nxloadconfig FreeNX-0_5_0-LWE/freenx-server/nxloadconfig
--- FreeNX-0_5_0-LWE.orig/freenx-server/nxloadconfig	2005-08-07 11:53:57.000000000 +0200
+++ FreeNX-0_5_0-LWE/freenx-server/nxloadconfig	2006-07-15 18:24:43.000000000 +0200
@@ -152,7 +152,7 @@
 COMMAND_START_GNOME=gnome-session
 COMMAND_START_CDE=cdwm
 COMMAND_XTERM=xterm
-COMMAND_XAUTH=/usr/X11R6/bin/xauth
+COMMAND_XAUTH=/usr/bin/xauth
 COMMAND_SMBMOUNT=smbmount
 COMMAND_SMBUMOUNT=smbumount
 COMMAND_NETCAT=netcat
