diff -ur a/libvo/videodev_mjpeg.h b/libvo/videodev_mjpeg.h
--- a/libvo/videodev_mjpeg.h	2012-10-30 00:35:05.660290723 -0500
+++ b/libvo/videodev_mjpeg.h	2012-10-30 00:36:39.881309452 -0500
@@ -130,6 +130,7 @@
 /*
 Private IOCTL to set up for displaying MJPEG
 */
+#define BASE_VIDIOCPRIVATE 192
 #define MJPIOC_G_PARAMS       _IOR ('v', BASE_VIDIOCPRIVATE+0,  struct mjpeg_params)
 #define MJPIOC_S_PARAMS       _IOWR('v', BASE_VIDIOCPRIVATE+1,  struct mjpeg_params)
 #define MJPIOC_REQBUFS        _IOWR('v', BASE_VIDIOCPRIVATE+2,  struct mjpeg_requestbuffers)
