--- pipewire-1.0.0/spa/plugins/aec/meson.build	2023-11-26 09:02:52.000000000 +0100
+++ pipewire-1.0.0.new/spa/plugins/aec/meson.build	2023-11-27 14:27:00.287003419 +0100
@@ -5,12 +5,14 @@
   install : true,
   install_dir : spa_plugindir / 'aec')
 
-if webrtc_dep.found()
-  aec_webrtc = shared_library('spa-aec-webrtc',
-    [ 'aec-webrtc.cpp' ],
-    include_directories : [ configinc ],
-    dependencies : [ spa_dep, webrtc_dep ],
-    install : true,
-    install_dir : spa_plugindir / 'aec')
+if get_option('echo-cancel-webrtc').allowed()
+  if webrtc_dep.found()
+    aec_webrtc = shared_library('spa-aec-webrtc',
+      [ 'aec-webrtc.cpp' ],
+      include_directories : [ configinc ],
+      dependencies : [ spa_dep, webrtc_dep ],
+      install : true,
+      install_dir : spa_plugindir / 'aec')
+  endif
 endif
 
