--- sonic-visualiser-1.5/data/fft/FFTMemoryCache.cpp.gcc44	2009-03-16 13:04:09.000000000 -0400
+++ sonic-visualiser-1.5/data/fft/FFTMemoryCache.cpp	2009-03-26 13:30:33.124732276 -0400
@@ -17,6 +17,7 @@
 #include "system/System.h"
 
 #include <iostream>
+#include <cstdio> // perror
 #include <cstdlib>
 
 //#define DEBUG_FFT_MEMORY_CACHE 1
--- sonic-visualiser-1.5/data/fileio/FileReadThread.cpp.gcc44	2009-03-16 13:04:26.000000000 -0400
+++ sonic-visualiser-1.5/data/fileio/FileReadThread.cpp	2009-03-26 13:31:51.140732496 -0400
@@ -20,6 +20,7 @@
 
 #include <iostream>
 #include <unistd.h>
+#include <cstdio> // perror
 
 //#define DEBUG_FILE_READ_THREAD 1
 
--- sonic-visualiser-1.5/base/Thread.cpp.gcc44	2009-03-16 13:04:33.000000000 -0400
+++ sonic-visualiser-1.5/base/Thread.cpp	2009-03-26 13:28:35.802982120 -0400
@@ -22,6 +22,7 @@
 //#define DEBUG_MUTEX_LOCKER 1
 
 #include <iostream>
+#include <cstdio> // perror
 
 Thread::Thread(Type type, QObject *parent) :
     QThread(parent),
--- sonic-visualiser-1.5/base/RingBuffer.h.gcc44	2009-03-16 13:04:33.000000000 -0400
+++ sonic-visualiser-1.5/base/RingBuffer.h	2009-03-26 13:26:04.569982992 -0400
@@ -26,6 +26,7 @@
 #include "system/System.h"
 #include "Scavenger.h"
 
+#include <cstdio>  // perror
 #include <cstring> // memcpy, memset &c
 
 //#define DEBUG_RINGBUFFER 1
--- sonic-visualiser-1.5/rdf/RDFFeatureWriter.h.gcc44	2009-03-16 13:04:39.000000000 -0400
+++ sonic-visualiser-1.5/rdf/RDFFeatureWriter.h	2009-03-26 13:38:33.891742875 -0400
@@ -20,6 +20,12 @@
 #ifndef _RDF_FEATURE_WRITER_H_
 #define _RDF_FEATURE_WRITER_H_
 
+// GCC 4.4.x: cstdint provides uint64_t, but it is still experimental
+//#include <cstdint>
+// while <stdint.h> defines it as long long long, which the C++ compiler
+// does not like. ugh for C <-> C++ impedance mismatch!
+#define uint64_t unsigned long long
+// GCC 4.4x END
 #include <string>
 #include <map>
 #include <set>
--- sonic-visualiser-1.5/system/System.cpp.gcc44	2009-03-16 13:03:42.000000000 -0400
+++ sonic-visualiser-1.5/system/System.cpp	2009-03-26 13:42:34.443747793 -0400
@@ -19,6 +19,7 @@
 #include <QString>
 
 #include <stdint.h>
+#include <cstdio> // file functions, perror
 
 #ifndef _WIN32
 #include <signal.h>
