diff -Naur apmd.orig/apmd.c apmd/apmd.c
--- apmd.orig/apmd.c	2001-11-25 23:43:32.000000000 +0100
+++ apmd/apmd.c	2004-07-27 10:40:26.000000000 +0200
@@ -71,8 +71,8 @@
 
 
 #ifdef APM_TRACE
-#define ADEBUG(lev,args...) \
-    if (APM_TRACE>=lev) syslog(LOG_DEBUG, __FUNCTION__ ": " args)
+#define ADEBUG(lev, text, args...) \
+    if (APM_TRACE>=lev) syslog(LOG_DEBUG, "%s : " text, __FUNCTION__, args)
 #else
 #define ADEBUG(args...)
 #endif
diff -Naur apmd.orig/apmsleep.c apmd/apmsleep.c
--- apmd.orig/apmsleep.c	2001-05-11 22:30:08.000000000 +0200
+++ apmd/apmsleep.c	2004-07-27 10:40:32.000000000 +0200
@@ -42,11 +42,15 @@
  *
  */
 
+typedef unsigned char u8;
+typedef unsigned short u16;
+typedef unsigned int u32;
+
 #include <stdio.h>
 #include <linux/version.h>
 
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,2,0)
-#include <asm/spinlock.h>
+// #include <asm/spinlock.h>
 #endif
 
 #include <linux/mc146818rtc.h>
