diff -Naur apmd/apmsleep.c apmd-p/apmsleep.c
--- apmd/apmsleep.c	2001-05-11 22:30:08.000000000 +0200
+++ apmd-p/apmsleep.c	2008-07-21 23:58:20.000000000 +0200
@@ -43,6 +43,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <linux/version.h>
 
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,2,0)
diff -Naur apmd/xapm.c apmd-p/xapm.c
--- apmd/xapm.c	1999-06-05 23:00:42.000000000 +0200
+++ apmd-p/xapm.c	2008-07-21 23:58:03.000000000 +0200
@@ -24,6 +24,7 @@
  */
 #include <stdio.h>
 #include <getopt.h>
+#include <stdlib.h>
 #include <X11/Intrinsic.h>
 #include <X11/StringDefs.h>
 #include <X11/cursorfont.h>
