=== modified file 'src/bamf-legacy-window-test.c'
--- src/bamf-legacy-window-test.c	2013-08-08 12:32:39 +0000
+++ src/bamf-legacy-window-test.c	2014-01-07 19:15:14 +0000
@@ -427,7 +427,7 @@
   g_return_if_fail (BAMF_IS_LEGACY_WINDOW_TEST (window));
   BamfLegacyWindowTest *self = BAMF_LEGACY_WINDOW_TEST (window);
 
-  return g_hash_table_insert (self->hints, g_strdup (name), g_strdup (value));
+  g_hash_table_insert (self->hints, g_strdup (name), g_strdup (value));
 }
 
 static char *

