=== modified file 'src/bamf-application.c'
--- src/bamf-application.c	2013-08-08 12:48:26 +0000
+++ src/bamf-application.c	2014-02-14 06:44:55 +0000
@@ -470,7 +470,7 @@
                                GFile **out_desktop_file, GFile **out_icon_file,
                                GCancellable *cancellable)
 {
-  g_return_val_if_fail (out_desktop_file, NULL);
+  g_return_val_if_fail (out_desktop_file, FALSE);
 
   if (!apps_dir)
     {

