diff -Naur lablgtk-2.10.0/src/Makefile lablgtk-2.10.0-opt/src/Makefile
--- lablgtk-2.10.0/src/Makefile	2007-09-25 12:56:49.000000000 +0200
+++ lablgtk-2.10.0-opt/src/Makefile	2007-10-28 18:07:15.000000000 +0100
@@ -46,7 +46,8 @@
 #MLLINK += -cclib -lcamlrund
 MLBYTEFLAGS = -g -dtypes
 else
-CFLAGS = -DG_DISABLE_ASSERT -DG_DISABLE_CAST_CHECKS $(GTKCFLAGS)
+OPTFLAGS ?=
+CFLAGS = $(OPTFLAGS) -DG_DISABLE_ASSERT -DG_DISABLE_CAST_CHECKS $(GTKCFLAGS)
 ifneq ($(TOOLCHAIN),msvc)
 CFLAGS += -O
 endif
