diff -up aqbanking-3.8.2/src/plugins/backends/aqhbci/ui/qt3/libs/logmanager/error.cpp.foo aqbanking-3.8.2/src/plugins/backends/aqhbci/ui/qt3/libs/logmanager/error.cpp
--- aqbanking-3.8.2/src/plugins/backends/aqhbci/ui/qt3/libs/logmanager/error.cpp.foo	2009-03-05 15:45:22.000000000 -0500
+++ aqbanking-3.8.2/src/plugins/backends/aqhbci/ui/qt3/libs/logmanager/error.cpp	2009-03-05 15:46:09.000000000 -0500
@@ -32,6 +32,7 @@
 
 #include "assert.h"
 #include "error.h"
+#include <cstdio>
 
 namespace HBCI {
 
