diff -Naur taskjuggler-2.2.0/TaskJugglerUI/taskjuggler.desktop taskjuggler-2.2.0-p/TaskJugglerUI/taskjuggler.desktop
--- taskjuggler-2.2.0/TaskJugglerUI/taskjuggler.desktop	2005-12-04 16:07:42.000000000 +0100
+++ taskjuggler-2.2.0-p/TaskJugglerUI/taskjuggler.desktop	2006-08-28 02:02:53.900114984 +0200
@@ -9,5 +9,5 @@
 Comment=Project Management Software
 MimeType=application/x-tjp;application/x-tjx
 InitialPreference=4
-Categories=Qt;KDE;ProjectManagement
+Categories=Qt;KDE;ProjectManagement;Utility;Application;
 Terminal=0
