--- tracker-0.6.6old/configure.ac	2008-03-03 05:13:34.000000000 +0100
+++ tracker-0.6.6/configure.ac	2008-12-05 11:43:37.000000000 +0100
@@ -38,7 +38,7 @@
 DBUS_REQUIRED=0.60
 GLIB_REQUIRED=2.14.0
 PANGO_REQUIRED=1.0.0
-GMIME_REQUIRED=2.1.0
+
 
 PKG_CHECK_MODULES(GLIB2, [ glib-2.0 >= $GLIB_REQUIRED ])
 AC_SUBST(GLIB2_CFLAGS)
@@ -64,10 +64,6 @@
 AC_SUBST(PANGO_CFLAGS)
 AC_SUBST(PANGO_LIBS)
 
-# Check for GMime
-PKG_CHECK_MODULES(GMIME, [ gmime-2.0 >= $GMIME_REQUIRED ])
-AC_SUBST(GMIME_CFLAGS)
-AC_SUBST(GMIME_LIBS)
 
 # Check for Dbus 0.50 or higher
 PKG_CHECK_MODULES(DBUS, [ dbus-1 >= $DBUS_REQUIRED dbus-glib-1 >= $DBUS_REQUIRED ])
@@ -79,6 +75,51 @@
 AC_SUBST(LIBPNG_CFLAGS)
 AC_SUBST(LIBPNG_LIBS)
 
+dnl Timezone checks
+AC_CACHE_CHECK(for tm_gmtoff in struct tm, ac_cv_struct_tm_gmtoff,
+	AC_TRY_COMPILE([
+		#include <time.h>
+		], [
+		struct tm tm;
+		tm.tm_gmtoff = 1;
+		], ac_cv_struct_tm_gmtoff="yes", ac_cv_struct_tm_gmtoff="no"))
+if test "$ac_cv_struct_tm_gmtoff" = "yes"; then
+	AC_DEFINE(HAVE_TM_GMTOFF, 1, [Define if struct tm has a tm_gmtoff member])
+else
+	AC_CACHE_CHECK(for timezone variable, ac_cv_var_timezone,
+		AC_TRY_COMPILE([
+			#include <time.h>
+		], [
+			timezone = 1;
+		], ac_cv_var_timezone="yes", ac_cv_var_timezone="no"))
+	if test "$ac_cv_var_timezone" = "yes"; then
+		AC_DEFINE(HAVE_TIMEZONE, 1, [Define if libc defines a timezone variable])
+		AC_CACHE_CHECK(for altzone variable, ac_cv_var_altzone,
+			AC_TRY_COMPILE([
+				#include <time.h>
+			], [
+				altzone = 1;
+			], ac_cv_var_altzone="yes", ac_cv_var_altzone="no"))
+		if test "$ac_cv_var_altzone" = "yes"; then
+			AC_DEFINE(HAVE_ALTZONE, 1, [Define if libc defines an altzone variable])
+		fi
+	else
+		AC_CACHE_CHECK(for _timezone variable, ac_cv_var__timezone,
+			AC_TRY_COMPILE([
+				#include <time.h>
+			], [
+				_timezone = 1;
+			], ac_cv_var__timezone="yes", ac_cv_var__timezone="no"))
+		if test "$ac_cv_var__timezone" = "yes"; then
+			AC_DEFINE(HAVE__TIMEZONE, 1, [Define if libc defines a _timezone variable])
+		else
+			AC_ERROR(unable to find a way to determine timezone)
+		fi
+	fi
+fi
+
+
+
 ####################################################################
 # DBus Service
 ####################################################################
