Index: trunk/src/airodump-ng.c
===================================================================
--- trunk/src/airodump-ng.c (revision 282)
+++ trunk/src/airodump-ng.c (revision 288)
@@ -306,4 +306,6 @@
     unsigned int long crc;
 
+    if((unsigned)caplen > sizeof(h80211)) return 1;
+
     m_bmac = 16;
     m_smac = 10;
@@ -371,4 +373,6 @@
 
     textlen = G.sk_len;
+
+    if((unsigned)textlen > sizeof(text)) return 1;
 
     memcpy(text, G.sharedkey[0]+24, textlen);
@@ -1034,4 +1038,5 @@
 //                n = ( p[1] > 32 ) ? 32 : p[1];
                 n = p[1];
+                n &= 0xFF; //cut down to a max length of 256
 
                 for( i = 0; i < n; i++ )
@@ -1093,4 +1098,5 @@
 //                n = ( p[1] > 32 ) ? 32 : p[1];
                 n = p[1];
+                n &= 0xFF; //cut down to a max length of 256
 
                 memset( ap_cur->essid, 0, 256 );
