http://bugs.gentoo.org/show_bug.cgi?id=88421

Index: svc_auth_gss.c
===================================================================
RCS file: /cvsroot/nfs/nfs-utils/support/rpc/svc_auth_gss.c,v
diff -u -r1.1 -r1.2
--- nfs-utils/support/rpc/svc_auth_gss.c
+++ nfs-utils/support/rpc/svc_auth_gss.c
@@ -382,7 +382,7 @@
 			return (AUTH_FAILED);
 		}
 		auth->svc_ah_ops = &svc_auth_gss_ops;
-		SVCAUTH_PRIVATE(auth) = gd;
+		auth->svc_ah_private = (caddr_t)gd;
 		rqst->rq_xprt->xp_auth = auth;
 	}
 	else gd = SVCAUTH_PRIVATE(rqst->rq_xprt->xp_auth);
