diff -Naur nfs-utils-2.6.4/tools/nfsrahead/99-nfs.rules nfs-utils-2.6.4.new/tools/nfsrahead/99-nfs.rules
--- nfs-utils-2.6.4/tools/nfsrahead/99-nfs.rules	2023-11-17 18:32:39.000000000 +0100
+++ nfs-utils-2.6.4.new/tools/nfsrahead/99-nfs.rules	2024-01-06 22:18:38.028616465 +0100
@@ -1 +1 @@
-SUBSYSTEM=="bdi", ACTION=="add", PROGRAM="/usr/libexec/nfsrahead %k", ATTR{read_ahead_kb}="%c"
+SUBSYSTEM=="bdi", ACTION=="add", PROGRAM="/usr/bin/nfsrahead %k", ATTR{read_ahead_kb}="%c"
diff -Naur nfs-utils-2.6.4/utils/mount/Makefile.am nfs-utils-2.6.4.new/utils/mount/Makefile.am
--- nfs-utils-2.6.4/utils/mount/Makefile.am	2023-11-17 18:32:39.000000000 +0100
+++ nfs-utils-2.6.4.new/utils/mount/Makefile.am	2024-01-06 22:18:20.312485876 +0100
@@ -4,7 +4,7 @@
 # is disabled as may be appropriate when /sbin is a symlink.
 # Note that we don't use "if CONFIG_SBIN_OVERRIDE" as that
 # causes autotools to notice the override and disable it.
-@CONFIG_SBIN_OVERRIDE_TRUE@sbindir = /sbin
+@CONFIG_SBIN_OVERRIDE_TRUE@sbindir = /usr/bin
 
 man8_MANS	= mount.nfs.man umount.nfs.man
 man5_MANS	= nfs.man
diff -Naur nfs-utils-2.6.4/utils/nfsdcltrack/Makefile.am nfs-utils-2.6.4.new/utils/nfsdcltrack/Makefile.am
--- nfs-utils-2.6.4/utils/nfsdcltrack/Makefile.am	2023-11-17 18:32:39.000000000 +0100
+++ nfs-utils-2.6.4.new/utils/nfsdcltrack/Makefile.am	2024-01-06 22:18:20.312485876 +0100
@@ -5,7 +5,7 @@
 # disabled to install in /usr/sbin anyway.
 # Note that we don't use "if CONFIG_SBIN_OVERRIDE" as that
 # causes autotools to notice the override and disable it.
-@CONFIG_SBIN_OVERRIDE_TRUE@sbindir = /sbin
+@CONFIG_SBIN_OVERRIDE_TRUE@sbindir = /usr/bin
 
 man8_MANS	= nfsdcltrack.man
 EXTRA_DIST	= $(man8_MANS)
diff -Naur nfs-utils-2.6.4/utils/nfsidmap/id_resolver.conf nfs-utils-2.6.4.new/utils/nfsidmap/id_resolver.conf
--- nfs-utils-2.6.4/utils/nfsidmap/id_resolver.conf	2023-11-17 18:32:39.000000000 +0100
+++ nfs-utils-2.6.4.new/utils/nfsidmap/id_resolver.conf	2024-01-06 22:18:30.544561297 +0100
@@ -1 +1 @@
-create	id_resolver	*	*	/usr/sbin/nfsidmap -t 600 %k %d
+create	id_resolver	*	*	/usr/bin/nfsidmap -t 600 %k %d
