diff --git a/utils/exportfs/exportfs.c b/utils/exportfs/exportfs.c
index 593a8eb..331e57e 100644
--- a/utils/exportfs/exportfs.c
+++ b/utils/exportfs/exportfs.c
@@ -13,6 +13,7 @@
 #endif
 
 #include <sys/vfs.h>
+#include <sys/stat.h>
 #include <unistd.h>
 #include <stdlib.h>
 #include <string.h>
diff --git a/utils/mount/mount.c b/utils/mount/mount.c
index 06e2804..cd04db9 100644
--- a/utils/mount/mount.c
+++ b/utils/mount/mount.c
@@ -24,6 +24,7 @@
 
 #include <unistd.h>
 #include <sys/types.h>
+#include <sys/stat.h>
 #include <stdio.h>
 #include <string.h>
 #include <errno.h>
diff --git a/utils/mount/network.c b/utils/mount/network.c
index 04a62ab..03a96d0 100644
--- a/utils/mount/network.c
+++ b/utils/mount/network.c
@@ -37,6 +37,7 @@
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/wait.h>
+#include <sys/stat.h>
 #include <netinet/in.h>
 #include <rpc/rpc.h>
 #include <rpc/pmap_prot.h>
