--- openvr-1.14.15/src/CMakeLists.txt	2020-09-18 20:20:27.000000000 +0200
+++ openvr-1.14.15.new/src/CMakeLists.txt	2020-11-06 15:57:22.663902777 +0100
@@ -106,7 +106,7 @@
 
 # Generate a .pc file for linux environments
 if(PLATFORM_NAME MATCHES "linux")
-    set(INSTALL_PKGCONFIG_DIR "${CMAKE_INSTALL_PREFIX}/share/pkgconfig" CACHE PATH "Installation directory for pkgconfig (.pc) files")
+    set(INSTALL_PKGCONFIG_DIR "${CMAKE_INSTALL_PREFIX}/lib/pkgconfig" CACHE PATH "Installation directory for pkgconfig (.pc) files")
     CONFIGURE_FILE("openvr.pc.in" "openvr.pc" @ONLY)
 
     set(OPENVR_PC ${CMAKE_CURRENT_BINARY_DIR}/openvr.pc)
