diff -Naur openvr-1.16.8/src/openvr_api_public.cpp openvr-1.16.8.new/src/openvr_api_public.cpp
--- openvr-1.16.8/src/openvr_api_public.cpp	2021-02-25 01:57:03.000000000 +0100
+++ openvr-1.16.8.new/src/openvr_api_public.cpp	2021-03-08 12:35:03.608566421 +0100
@@ -2,12 +2,12 @@
 #define VR_API_EXPORT 1
 #include "openvr.h"
 #include "ivrclientcore.h"
-#include <vrcore/pathtools_public.h>
-#include <vrcore/sharedlibtools_public.h>
-#include <vrcore/envvartools_public.h>
+#include "vrcommon/pathtools_public.h"
+#include "vrcommon/sharedlibtools_public.h"
+#include "vrcommon/envvartools_public.h"
 #include "hmderrors_public.h"
-#include <vrcore/strtools_public.h>
-#include <vrcore/vrpathregistry_public.h>
+#include "vrcommon/strtools_public.h"
+#include "vrcommon/vrpathregistry_public.h"
 #include <mutex>
 
 using vr::EVRInitError;
diff -Naur openvr-1.16.8/src/vrcommon/dirtools_public.cpp openvr-1.16.8.new/src/vrcommon/dirtools_public.cpp
--- openvr-1.16.8/src/vrcommon/dirtools_public.cpp	2021-02-25 01:57:03.000000000 +0100
+++ openvr-1.16.8.new/src/vrcommon/dirtools_public.cpp	2021-03-08 12:35:03.608566421 +0100
@@ -1,7 +1,7 @@
 //========= Copyright Valve Corporation ============//
-#include <vrcore/dirtools_public.h>
-#include <vrcore/strtools_public.h>
-#include <vrcore/pathtools_public.h>
+#include "vrcommon/dirtools_public.h"
+#include "vrcommon/strtools_public.h"
+#include "vrcommon/pathtools_public.h"
 
 #include <errno.h>
 #include <string.h>
diff -Naur openvr-1.16.8/src/vrcommon/envvartools_public.cpp openvr-1.16.8.new/src/vrcommon/envvartools_public.cpp
--- openvr-1.16.8/src/vrcommon/envvartools_public.cpp	2021-02-25 01:57:03.000000000 +0100
+++ openvr-1.16.8.new/src/vrcommon/envvartools_public.cpp	2021-03-08 12:35:03.609566426 +0100
@@ -1,6 +1,6 @@
 //========= Copyright Valve Corporation ============//
-#include <vrcore/envvartools_public.h>
-#include <vrcore/strtools_public.h>
+#include "vrcommon/envvartools_public.h"
+#include "vrcommon/strtools_public.h"
 #include <stdlib.h>
 #include <string>
 #include <cctype>
diff -Naur openvr-1.16.8/src/vrcommon/pathtools_public.cpp openvr-1.16.8.new/src/vrcommon/pathtools_public.cpp
--- openvr-1.16.8/src/vrcommon/pathtools_public.cpp	2021-02-25 01:57:03.000000000 +0100
+++ openvr-1.16.8.new/src/vrcommon/pathtools_public.cpp	2021-03-08 12:35:03.609566426 +0100
@@ -1,6 +1,6 @@
 //========= Copyright Valve Corporation ============//
-#include <vrcore/strtools_public.h>
-#include <vrcore/pathtools_public.h>
+#include "vrcommon/strtools_public.h"
+#include "vrcommon/pathtools_public.h"
 
 #if defined( _WIN32)
 #include <windows.h>
@@ -66,7 +66,7 @@
 		return "";
 	}
 #else
-	AssertMsg( false, "Implement Plat_GetExecutablePath" );
+	//AssertMsg( false, "Implement Plat_GetExecutablePath" );
 	return "";
 #endif
 
diff -Naur openvr-1.16.8/src/vrcommon/sharedlibtools_public.cpp openvr-1.16.8.new/src/vrcommon/sharedlibtools_public.cpp
--- openvr-1.16.8/src/vrcommon/sharedlibtools_public.cpp	2021-02-25 01:57:03.000000000 +0100
+++ openvr-1.16.8.new/src/vrcommon/sharedlibtools_public.cpp	2021-03-08 12:35:03.609566426 +0100
@@ -1,5 +1,5 @@
 //========= Copyright Valve Corporation ============//
-#include <vrcore/sharedlibtools_public.h>
+#include "vrcommon/sharedlibtools_public.h"
 #include <string.h>
 
 #if defined(_WIN32)
diff -Naur openvr-1.16.8/src/vrcommon/strtools_public.cpp openvr-1.16.8.new/src/vrcommon/strtools_public.cpp
--- openvr-1.16.8/src/vrcommon/strtools_public.cpp	2021-02-25 01:57:03.000000000 +0100
+++ openvr-1.16.8.new/src/vrcommon/strtools_public.cpp	2021-03-08 12:35:52.096803758 +0100
@@ -1,5 +1,6 @@
 //========= Copyright Valve Corporation ============//
-#include <vrcore/strtools_public.h>
+#include "vrcommon/strtools_public.h"
+//#include "vrcommon/assert.h"
 #include <string.h>
 #include <stdio.h>
 #include <stdlib.h>
@@ -9,7 +10,7 @@
 #include <functional>
 #include <locale>
 #include <codecvt>
-#include <vrcore/assert.h>
+#include <cstdarg>
 
 #if defined( _WIN32 )
 #include <windows.h>
@@ -128,7 +129,7 @@
 	// Something went fairly wrong
 	if ( unSize < 0 )
 	{
-		AssertMsg( false, "Format string parse failure" );
+		//AssertMsg( false, "Format string parse failure" );
 		return "";
 	}
 
@@ -149,7 +150,7 @@
 	// Double check, just in case
 	if ( unSize < 0 )
 	{
-		AssertMsg( false, "Format string parse failure" );
+		//AssertMsg( false, "Format string parse failure" );
 		return "";
 	}
 
diff -Naur openvr-1.16.8/src/vrcommon/vrpathregistry_public.cpp openvr-1.16.8.new/src/vrcommon/vrpathregistry_public.cpp
--- openvr-1.16.8/src/vrcommon/vrpathregistry_public.cpp	2021-02-25 01:57:03.000000000 +0100
+++ openvr-1.16.8.new/src/vrcommon/vrpathregistry_public.cpp	2021-03-08 12:35:03.695566846 +0100
@@ -1,11 +1,11 @@
 //========= Copyright Valve Corporation ============//
 
-#include <vrcore/vrpathregistry_public.h>
-#include <json/json.h>
-#include <vrcore/pathtools_public.h>
-#include <vrcore/envvartools_public.h>
-#include <vrcore/strtools_public.h>
-#include <vrcore/dirtools_public.h>
+#include "vrcommon/vrpathregistry_public.h"
+#include "json/json.h"
+#include "vrcommon/pathtools_public.h"
+#include "vrcommon/envvartools_public.h"
+#include "vrcommon/strtools_public.h"
+#include "vrcommon/dirtools_public.h"
 
 #if defined( WIN32 )
 #include <windows.h>
