--- stellarium-25.1/CMakeLists.txt	2025-03-23 12:29:18.000000000 +0100
+++ stellarium-25.1.new/CMakeLists.txt	2025-04-07 17:15:19.164256220 +0200
@@ -847,11 +847,18 @@
      ENDIF()
 ENDIF()
 
-CPMAddPackage(NAME md4c
+CPMFindPackage(NAME md4c
               URL https://github.com/mity/md4c/archive/refs/tags/release-0.4.8.tar.gz
               URL_HASH SHA256=4a457df853425b6bb6e3457aa1d1a13bccec587a04c38c622b1013a0da41439f
               EXCLUDE_FROM_ALL yes
               OPTIONS "BUILD_MD2HTML_EXECUTABLE OFF")
+IF(md4c_ADDED)
+	SET(MD4C_STEL_LIBRARIES md4c-html)
+ELSE()
+	# *_LIBRARIES variable is not set by CPM (unlike find_package; see
+	# https://github.com/cpm-cmake/CPM.cmake/issues/132).
+        SET(MD4C_STEL_LIBRARIES md4c::md4c-html)
+ENDIF()
 
 ########### Set some global variables ###########
 IF(UNIX AND NOT WIN32)
