diff -Naur modsecurity-v3.0.4/src/engine/lua.cc modsecurity-v3.0.4.new/src/engine/lua.cc
--- modsecurity-v3.0.4/src/engine/lua.cc	2020-01-13 14:09:28.000000000 +0100
+++ modsecurity-v3.0.4.new/src/engine/lua.cc	2020-08-07 10:36:54.000404356 +0200
@@ -155,11 +155,6 @@
             case LUA_ERRMEM:
                 e.assign("Memory error. ");
                 break;
-#ifndef WITH_LUA_5_1
-            case LUA_ERRGCMM:
-                e.assign("Garbage Collector error. ");
-                break;
-#endif
         }
         e.append(lua_tostring(L, -1));
         ms_dbg_a(t, 2, e);
