diff -Naur a/install.rdf b/install.rdf
--- a/install.rdf	2009-06-30 19:43:00.000000000 +0200
+++ b/install.rdf	2010-01-27 02:41:22.000000000 +0100
@@ -46,7 +46,7 @@
     <em:targetApplication>
       <Description>
         <em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
-        <em:maxVersion>3.5.*</em:maxVersion>
+        <em:maxVersion>3.6.*</em:maxVersion>
         <em:minVersion>1.0</em:minVersion>
       </Description>
     </em:targetApplication>
