--- Makefile.unix	2008-09-04 17:12:23.000000000 +0200
+++ Makefile.unix	2008-09-15 16:00:40.000000000 +0200
@@ -1,5 +1,5 @@
 # note if you change the prefix also update egoboo.sh
-PREFIX   := ${HOME}/.local
+PREFIX   := ${Fdestdir}/usr
 EGO_SRC  := camera.c char.c Client.c Clock.c common-file.c configfile.c \
                  egoboostrutil.c enchant.c Font.c game.c gltexture.c graphicfan.c \
                  graphicmad.c graphic.c graphicprt.c id_normals.c input.c \
@@ -28,13 +28,13 @@
    
 
 ${EGO_BIN}: ${EGO_OBJ} ${ENET_OBJ}
-   ${CC} -o $@ $^ ${LDFLAGS}
+	${CC} -o $@ $^ ${LDFLAGS}
 
 install: ${EGO_BIN}
-   mkdir -p ${PREFIX}/bin
-   mkdir -p ${PREFIX}/libexec
-   install -m 755 ${EGO_BIN} ${PREFIX}/libexec
-   install -p -m 755 ${EGO_BIN}.sh ${PREFIX}/bin/${EGO_BIN}
+	mkdir -p ${PREFIX}/bin
+	mkdir -p ${PREFIX}/libexec
+	install -m 755 ${EGO_BIN} ${PREFIX}/libexec
+	install -p -m 755 ${EGO_BIN}.sh ${PREFIX}/bin/${EGO_BIN}
 
 clean:
-   rm -f ${ENET_OBJ} ${EGO_OBJ} ${EGO_BIN} test_sdl.o test_sdl
\ Pas de fin de ligne à la fin du fichier.
+	rm -f ${ENET_OBJ} ${EGO_OBJ} ${EGO_BIN} test_sdl.o test_sdl
\ Pas de fin de ligne à la fin du fichier.
