--- src/game/Makefile.orig	2011-12-14 15:00:21.907724126 +0100
+++ src/game/Makefile	2011-12-14 15:00:47.833724109 +0100
@@ -34,7 +34,7 @@
 
 CC      := gcc
 INC     := -I. -I.. -I../enet/include ${SDLCONF_I} -I./extensions -I./file_formats -I./platform
-LDFLAGS := ${SDLCONF_L} -L../enet/lib -lSDL_ttf -lSDL_mixer -lGL -lGLU -lSDL_image -lphysfs -lenet
+LDFLAGS := ${SDLCONF_L} -L../enet/lib -lSDL_ttf -lSDL_mixer -lGL -lGLU -lSDL_image -lphysfs -lenet -lm
 
 # use different options if the environmental variable PREFIX is defined
 ifdef ($(PREFIX),"")
