#patch for optimization
--- source/game/Makefile.unix	2008-11-23 15:01:02.000000000 +0100
+++ Makefile.unix	2008-12-13 13:21:33.000000000 +0100
@@ -1,5 +1,5 @@
 # note if you change the prefix also update egoboo.sh
-PREFIX	:= ${HOME}/.local
+PREFIX	:= /usr
 EGO_SRC  := camera.c char.c Client.c Clock.c common-file.c configfile.c \
                  egobooendian.c egoboostrutil.c enchant.c Font.c game.c gltexture.c \
                  graphicfan.c graphicmad.c graphic.c graphicprt.c input.c \
@@ -19,7 +19,7 @@
 CC      := gcc
 OPT     := -Os
 INC     := -I. -I../enet/include -I.. ${SDLCONF_I}
-CFLAGS  := ${OPT} ${INC}
+CFLAGS  := ${CFLAGS} ${INC}
 LDFLAGS := ${SDLCONF_L} -lSDL_ttf -lSDL_mixer -lGL -lGLU
 
 EGO_BIN := egoboo

#patch for installing
--- source/Makefile	2008-11-10 10:24:28.000000000 +0100
+++ Makefile	2008-12-13 13:34:01.000000000 +0100
@@ -1,7 +1,6 @@
 # Top level Makefile for Egoboo 2.x
 
-PREFIX	:= ${HOME}/.local
-PROJ_NAME	:= egoboo
+PREFIX	:= /usr
 
 all:
 	make -C enet all
@@ -13,7 +12,7 @@
 
 install: 
 	mkdir -p ${PREFIX}/bin
-	install -p -m 755 game/{PROJ_NAME}.sh ${PREFIX}/bin/${PROJ_NAME}
+	install -p -m 755 game/egoboo.sh ${PREFIX}/bin/egoboo
 
 	mkdir -p ${PREFIX}/libexec
-	install -m 755 game/${PROJ_NAME} ${PREFIX}/libexec
\ Pas de fin de ligne à la fin du fichier.
+	install -m 755 game/egoboo ${PREFIX}/libexec

#patch for executable
--- source/game/egoboo.sh	2008-11-09 19:49:21.000000000 +0100
+++ egoboo.sh	2008-12-13 13:36:39.000000000 +0100
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-EGOBOO_PREFIX=${HOME}/.local
+EGOBOO_PREFIX="/usr"
 EGOBOO_SHARED="${EGOBOO_PREFIX}/share/egoboo"
 
 # exit on any error
