--- gnome-do-plugins-0.8.0/configure.ac	2009-01-29 21:03:08.000000000 +0100
+++ gnome-do-plugins-0.8.0/configure.ac	2009-02-01 16:50:47.000000000 +0100
@@ -129,12 +129,6 @@
 GNOME-Screenshot/Makefile
 GNOME-Session/Makefile
 GNOME-Terminal/Makefile
-GoogleCalculator/Makefile
-GoogleCalendar/Makefile
-GoogleContacts/Makefile
-GoogleDocs/Makefile
-GoogleMaps/Makefile
-GoogleSearch/Makefile
 ImageShack/Makefile
 JIRA/Makefile
 Launchpad/Makefile
--- gnome-do-plugins-0.8.0/Makefile.am	2009-01-22 20:33:48.000000000 +0100
+++ gnome-do-plugins-0.8.0/Makefile.am	2009-02-01 16:54:16.000000000 +0100
@@ -24,12 +24,6 @@
 	GNOME-Screenshot \
 	GNOME-Session \
 	GNOME-Terminal \
-	GoogleCalculator \
-	GoogleCalendar \
-	GoogleContacts \
-	GoogleDocs \
-	GoogleMaps \
-	GoogleSearch \
 	ImageShack \
 	JIRA \
 	Launchpad \
