diff -Naur faad2/configure.in faad2-p/configure.in
--- faad2/configure.in	2006-08-13 16:13:31.000000000 +0200
+++ faad2-p/configure.in	2006-11-18 03:00:56.000000000 +0100
@@ -29,8 +29,6 @@
 AC_ARG_WITH(xmms,[  --with-xmms             compile XMMS-1 plugin],
 	     WITHXMMS=$withval, WITHXMMS=no)
 
-AC_ARG_WITH(bmp,[  --with-bmp              compile beep media player plugin],
-	     [WITHBMP=$withval;WITHXMMS=$withval], [WITHBMP=no])
 
 AC_ARG_WITH(drm,[  --with-drm              compile libfaad with DRM support],
 	     WITHDRM=$withval, WITHDRM=no)
@@ -114,11 +112,6 @@
  AM_CONDITIONAL(HAVE_XMMS, false)
 fi
 
-if test x$WITHBMP = xyes; then
-  AC_DEFINE([HAVE_BMP], 1, [User wants beep media player plugin built])
-  AM_CONDITIONAL([HAVE_XMMS], true)
-  AM_CONDITIONAL([HAVE_BMP], true)
-fi
 
 if test x$WITHDRM = xyes; then
   AC_DEFINE(DRM, 1, [Define if you want to use libfaad together with Digital Radio Mondiale (DRM)])
diff -Naur faad2/plugins/Makefile.am faad2-p/plugins/Makefile.am
--- faad2/plugins/Makefile.am	2006-08-07 20:47:45.000000000 +0200
+++ faad2-p/plugins/Makefile.am	2006-11-18 03:00:56.000000000 +0100
@@ -8,10 +8,6 @@
 if HAVE_XMMS
 SUBDIRS = xmms
 else
-if HAVE_BMP
-SUBDIRS = bmp
-else
 SUBDIRS =
-endif #HAVE_BMP
 endif #HAVE_XMMS
 endif #HAVE_MPEG4IP_PLUG
diff -Naur faad2/plugins/xmms/src/Makefile.am faad2-p/plugins/xmms/src/Makefile.am
--- faad2/plugins/xmms/src/Makefile.am	2006-08-07 20:47:45.000000000 +0200
+++ faad2-p/plugins/xmms/src/Makefile.am	2006-11-18 03:02:11.000000000 +0100
@@ -1,12 +1,7 @@
-if HAVE_BMP
-libdir=$(shell pkg-config --variable=input-plugin-dir bmp)
-local_CFLAGS=$(shell pkg-config --cflags bmp)
-local_LDFLAGS=$(shell pkg-config --libs bmp)
-else
+
 local_CFLAGS=`$(XMMS_CONFIG) --cflags` -Wall
 local_LDFLAGS=`$(XMMS_CONFIG) --libs`
 libdir = `$(XMMS_CONFIG) --input-plugin-dir`
-endif #HAVE_BMP
 lib_LTLIBRARIES = libmp4.la
 
 libmp4_la_CFLAGS = $(local_CFLAGS) -Wall \
