From 5eb7fec958bc6ba8a1a2b0be4916cac818866e1c Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@gmail.com>
Date: Wed, 18 Jul 2012 08:46:49 +0000
Subject: mach64: fix build on 32-bit.

tvout code only builds on 32-bit, fix build.

Signed-off-by: Dave Airlie <airlied@redhat.com>
---
diff --git a/src/atiscreen.c b/src/atiscreen.c
index 2957877..b1fbd3c 100644
--- a/src/atiscreen.c
+++ b/src/atiscreen.c
@@ -580,7 +580,7 @@ ATIScreenInit(SCREEN_INIT_ARGS_DECL)
 #ifdef TV_OUT
     /* Fix-up TV out after ImpacTV probe */
     if (pATI->OptionTvOut && pATI->Chip < ATI_CHIP_264GTPRO)
-        ATISwitchMode(0, pScreenInfo->currentMode, 0);
+        ATISwitchMode(SWITCH_MODE_ARGS(pScreenInfo, pScreenInfo->currentMode));
 #endif /* TV_OUT */
 
 #ifdef XF86DRI_DEVEL
--
cgit v0.9.0.2-2-gbebe
