diff -up netatalk-2.0.4/etc/afpd/desktop.c.extern_ucreator netatalk-2.0.4/etc/afpd/desktop.c
--- netatalk-2.0.4/etc/afpd/desktop.c.extern_ucreator	2009-07-14 16:26:36.984610055 +0200
+++ netatalk-2.0.4/etc/afpd/desktop.c	2009-07-14 16:27:01.607580009 +0200
@@ -312,7 +312,7 @@ addicon_err:
 }
 
 static const u_char	utag[] = { 0, 0, 0, 0 };
-static const u_char	ucreator[] = { 0, 0, 0, 0 };/* { 'U', 'N', 'I', 'X' };*/
+u_char			ucreator[] = { 0, 0, 0, 0 };/* { 'U', 'N', 'I', 'X' };*/
 static const u_char	utype[] = { 0, 0, 0, 0 };/* { 'T', 'E', 'X', 'T' };*/
 static const short	usize = 256;
 
