diff -up netatalk-2.0.3/etc/uams/Makefile.am.uams_no_pie netatalk-2.0.3/etc/uams/Makefile.am
--- netatalk-2.0.3/etc/uams/Makefile.am.uams_no_pie	2008-10-10 10:32:59.000000000 +0200
+++ netatalk-2.0.3/etc/uams/Makefile.am	2008-10-10 10:32:59.000000000 +0200
@@ -1,5 +1,5 @@
 # Makefile.am for etc/uams/
-
+LDFLAGS =
 SUBDIRS = uams_krb4
 
 #
diff -up netatalk-2.0.3/macros/gssapi-check.m4.uams_no_pie netatalk-2.0.3/macros/gssapi-check.m4
--- netatalk-2.0.3/macros/gssapi-check.m4.uams_no_pie	2008-10-10 10:32:59.000000000 +0200
+++ netatalk-2.0.3/macros/gssapi-check.m4	2008-10-10 10:32:59.000000000 +0200
@@ -126,7 +126,7 @@ AC_DEFUN([NETATALK_GSSAPI_CHECK], 
 	if test x"$ac_cv_func_gss_acquire_cred" = x"yes"; then
    		AC_DEFINE(HAVE_GSSAPI,1,[Whether to enable GSSAPI support])
 		AC_MSG_RESULT([yes])
-		GSSAPI_LIBS="$LDFLAGS $LIBS"
+		GSSAPI_LIBS="$LIBS"
 	else
 		AC_MSG_RESULT([no])
 		if test x"$compilegssapi" = x"yes"; then
