diff -Naur vigra-1.11.1/include/vigra/separableconvolution.hxx vigra-1.11.1.new/include/vigra/separableconvolution.hxx
--- vigra-1.11.1/include/vigra/separableconvolution.hxx	2017-05-19 17:01:08.000000000 +0200
+++ vigra-1.11.1.new/include/vigra/separableconvolution.hxx	2022-08-05 12:06:53.239433240 +0200
@@ -1409,11 +1409,7 @@
         {}
 
         ~InitProxy()
-#ifndef _MSC_VER
-            throw(PreconditionViolation)
-#elif _MSC_VER >= 1900
             noexcept(false)
-#endif
         {
             vigra_precondition(count_ == 1 || count_ == sum_,
                   "Kernel1D::initExplicitly(): "
diff -Naur vigra-1.11.1/include/vigra/stdconvolution.hxx vigra-1.11.1.new/include/vigra/stdconvolution.hxx
--- vigra-1.11.1/include/vigra/stdconvolution.hxx	2017-05-19 17:01:08.000000000 +0200
+++ vigra-1.11.1.new/include/vigra/stdconvolution.hxx	2022-08-05 12:06:53.240433182 +0200
@@ -792,11 +792,7 @@
         {}
 
         ~InitProxy()
-#ifndef _MSC_VER
-            throw(PreconditionViolation)
-#elif _MSC_VER >= 1900
             noexcept(false)
-#endif
         {
             vigra_precondition(count_ == 1 || count_ == sum_,
                                "Kernel2D::initExplicitly(): "
