diff -dupr a/src/chrome/browser/first_run.cc b/src/chrome/browser/first_run.cc
--- a/src/chrome/browser/first_run/first_run.cc	2010-03-10 10:21:33.000000000 -0700
+++ b/src/chrome/browser/first_run/first_run.cc	2010-03-10 13:08:12.000000000 -0700
@@ -92,8 +92,7 @@
 
   // The standard location of the master prefs is next to the chrome binary.
   FilePath master_prefs;
-  if (!PathService::Get(base::DIR_EXE, &master_prefs))
-    return true;
+  master_prefs = FilePath("/etc/chromium-browser");
   master_prefs = master_prefs.AppendASCII(installer::kDefaultMasterPrefs);
 
   installer::MasterPreferences prefs(master_prefs);
