--- chromium-13.0.782.112/chrome/browser/first_run/first_run.cc	2011-08-06 10:01:51.000000000 +0200
+++ chromium-13.0.782.112new/chrome/browser/first_run/first_run.cc	2011-08-15 11:49:16.000000000 +0200
@@ -101,8 +101,7 @@ bool FirstRun::ProcessMasterPreferences(
 
   // The standard location of the master prefs is next to the chrome binary.
   FilePath master_prefs;
-  if (!PathService::Get(base::DIR_EXE, &master_prefs))
-    return true;
+  master_prefs = FilePath("/etc/chromium-browser");
   master_prefs = master_prefs.AppendASCII(installer::kDefaultMasterPrefs);
 
   installer::MasterPreferences prefs(master_prefs);
