--- media/filters/ffmpeg_demuxer.cc.orig	2010-06-10 12:28:10.000000000 +0200
+++ media/filters/ffmpeg_demuxer.cc	2010-06-10 12:29:13.000000000 +0200
@@ -413,9 +413,8 @@
     CodecType codec_type = codec_context->codec_type;
     if (codec_type == CODEC_TYPE_AUDIO || codec_type == CODEC_TYPE_VIDEO) {
       AVStream* stream = format_context_->streams[i];
-      // WebM is currently strictly VP8 and Vorbis.
-      if (kDemuxerIsWebm && (stream->codec->codec_id != CODEC_ID_VP8 &&
-        stream->codec->codec_id != CODEC_ID_VORBIS)) {
+      // WebM is currently strictly Vorbis.
+      if (kDemuxerIsWebm && (stream->codec->codec_id != CODEC_ID_VORBIS)) {
         packet_streams_.push_back(NULL);
         continue;
       }
--- net/base/mime_util.cc.orig	2010-06-10 12:30:13.000000000 +0200
+++ net/base/mime_util.cc	2010-06-10 12:30:38.000000000 +0200
@@ -227,7 +227,6 @@
 #endif
   "theora",
   "vorbis",
-  "vp8"
 };
 
 // Note: does not include javascript types list (see supported_javascript_types)
@@ -297,7 +296,7 @@
 };
 
 static const MediaFormatStrict format_codec_mappings[] = {
-  { "video/webm", "vorbis,vp8,vp8.0" },
+  { "video/webm", "vorbis" },
   { "audio/webm", "vorbis" }
 };
 
