--- chromium-browser-5.0.342.8/src/sandbox/linux/seccomp/sandbox.cc
+++ chromium-browser-5.0.342.8/src/sandbox/linux/seccomp/sandbox.cc
@@ -444,8 +444,8 @@
   }
   processFdPub_                  = pair[0];
   cloneFdPub_                    = pair[2];
-  SecureMemArgs::Args* secureMem = createTrustedProcess(pair[0], pair[1],
-                                                        pair[2], pair[3]);
+  SecureMemArgs* secureMem = createTrustedProcess(pair[0], pair[1],
+                                                  pair[2], pair[3]);
 
   // We find all libraries that have system calls and redirect the system
   // calls to the sandbox. If we miss any system calls, the application will be

