diff -ur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2012-04-23 04:46:49.000000000 -0500
+++ b/CMakeLists.txt	2013-01-01 08:55:04.007295933 -0600
@@ -140,7 +140,7 @@
 else (WL_UNIT_TESTS)
   message(STATUS "Disabled Unit Tests")
   set (Boost_FIND_COMPONENTS signals)
-  set (Boost_USE_STATIC_LIBS   ON)
+  set (Boost_USE_STATIC_LIBS   OFF)
   set (Boost_USE_MULTITHREADED ON)
   set (Boost_DETAILED_FAILURE_MSG ON)
   find_package(Boost 1.35.0 COMPONENTS signals REQUIRED)
diff -ur a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2012-04-23 04:46:49.000000000 -0500
+++ b/src/CMakeLists.txt	2013-01-01 08:56:09.704293515 -0600
@@ -57,7 +57,7 @@
   endif (DEFINED MINGW)
 endif (DEFINED MSVC)
 
-add_library (widelands_all STATIC ${WL_SRCS_CC} ${WL_SRCS_H})
+add_library (widelands_all SHARED ${WL_SRCS_CC} ${WL_SRCS_H})
 
 set (WL_SOURCES_ROOT ${CMAKE_CURRENT_SOURCE_DIR})
 set (WL_SOURCES_BINARY_ROOT ${CMAKE_CURRENT_BINARY_DIR})
