diff -ur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2014-02-22 10:06:51.000000000 +0100
+++ b/CMakeLists.txt	2014-02-23 16:53:27.054141998 +0100
@@ -108,9 +108,9 @@
 if (WL_UNIT_TESTS)
   message(STATUS "Enabled unit tests")
   if (DEFINED MSVC)
-    set (BUILD_SHARED_LIBS OFF)
-	set (Boost_USE_STATIC_RUNTIME ON)
-	set (Boost_USE_STATIC_LIBS ON)
+    set (BUILD_SHARED_LIBS ON)
+	set (Boost_USE_STATIC_RUNTIME OFF)
+	set (Boost_USE_STATIC_LIBS OFF)
   else (DEFINED MSVC)
     set (Boost_USE_STATIC_LIBS  OFF)
   endif (DEFINED MSVC)
diff -ur a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	2014-02-22 10:06:51.000000000 +0100
+++ b/src/CMakeLists.txt	2014-02-23 16:53:48.993141075 +0100
@@ -57,7 +57,7 @@
   endif (DEFINED MINGW)
 endif (DEFINED MSVC)
 
-add_library (widelands_all STATIC ${WL_SRCS_CC} ${WL_SRCS_H})
+add_library (widelands_all SHARED ${WL_SRCS_CC} ${WL_SRCS_H})
 
 set (WL_SOURCES_ROOT ${CMAKE_CURRENT_SOURCE_DIR})
 set (WL_SOURCES_BINARY_ROOT ${CMAKE_CURRENT_BINARY_DIR})
