--- ogre-14.3.1/Components/Overlay/include/ImGui.i	2024-10-23 00:56:30.000000000 +0200
+++ ogre-14.3.1.new/Components/Overlay/include/ImGui.i	2024-11-03 17:21:29.393853633 +0100
@@ -49,7 +49,7 @@
 #ifdef SWIGPYTHON
 // match the signature of the by value variants
 %typemap(argout) float[4], float[3], float[2] {
-    $result = SWIG_Python_AppendOutput($result, SWIG_NewPointerObj($1, $descriptor(ImVec4*), 0));
+    $result = SWIG_AppendOutput($result, SWIG_NewPointerObj($1, $descriptor(ImVec4*), 0));
 }
 
 // for PlotHistogram, PlotLines
