diff -Naur iptraf-2.7.0.orig/src/othptab.c iptraf-2.7.0/src/othptab.c
--- iptraf-2.7.0.orig/src/othptab.c	2005-06-09 16:48:17.000000000 +0000
+++ iptraf-2.7.0/src/othptab.c	2005-06-09 16:49:12.000000000 +0000
@@ -16,6 +16,7 @@
 
 ***/
 
+#include <net/if.h>
 #include <asm/types.h>
 #include <linux/if_ether.h>
 #include <linux/if_tr.h>
diff -Naur iptraf-2.7.0.orig/src/tcptable.h iptraf-2.7.0/src/tcptable.h
--- iptraf-2.7.0.orig/src/tcptable.h	2005-06-09 16:48:17.000000000 +0000
+++ iptraf-2.7.0/src/tcptable.h	2005-06-09 16:48:45.000000000 +0000
@@ -19,12 +19,12 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 #include <net/if_arp.h>
+#include <net/if.h>
 #include <asm/types.h>
 #include <linux/if_packet.h>
 #include <linux/if_ether.h>
 #include <linux/if_fddi.h>
 #include <linux/if_tr.h>
-#include <net/if.h>
 #include <netinet/ip.h>
 #include <netinet/udp.h>
 #include "servname.h"
