diff -Naur k3d-0.6.0.0/m4/k3d_with_external_boost.m4 k3d-0.6.0.0-p/m4/k3d_with_external_boost.m4
--- k3d-0.6.0.0/m4/k3d_with_external_boost.m4	2006-07-04 05:08:57.000000000 +0200
+++ k3d-0.6.0.0-p/m4/k3d_with_external_boost.m4	2006-08-19 17:30:39.272429448 +0200
@@ -9,32 +9,27 @@
 AC_ARG_WITH(external-boost, [  --with-external-boost   use an external boost library instead of the version included with the K-3D source [[yes]]],[],[withval=yes])
 if test x$withval != xno; then
 	AC_CHECK_HEADER([boost/version.hpp],
-	[
-		K3D_BOOST_SUBDIR=""
-		K3D_BOOST_CPPFLAGS=""
-		K3D_BOOST_LIBS="-lboost_regex -lboost_date_time -lboost_filesystem"
-		AC_MSG_RESULT([Using external boost library])
-	],
-	[
-		AC_MSG_ERROR([couldn't find boost library ... either install boost or re-run configure using the --without-external-boost option])
-	])
-	AC_CHECK_LIB(boost_regex, main, [],
-	[
-		AC_MSG_ERROR([couldn't find boost_regex library])
-	])
-	AC_CHECK_LIB(boost_date_time, main, [],
-	[
-		AC_MSG_ERROR([couldn't find boost_date_time library])
-	])
-	AC_CHECK_LIB(boost_filesystem, main, [],
-	[
-		AC_MSG_ERROR([couldn't find boost_filesystem library])
-	])
-else
-	K3D_BOOST_SUBDIR="boost"
-	K3D_BOOST_CPPFLAGS="-I\$(top_srcdir)/boost"
-	K3D_BOOST_LIBS='$(top_builddir)/boost/libs/regex/src/libregexpp.la $(top_builddir)/boost/libs/date_time/src/gregorian/libgregorian.la $(top_builddir)/boost/libs/date_time/src/posix_time/libposixtime.la $(top_builddir)/boost/libs/filesystem/src/libfilesystempp.la'
-	AC_MSG_RESULT([Using internal boost library])
+        [
+                K3D_BOOST_SUBDIR=""
+                K3D_BOOST_CPPFLAGS=""
+                K3D_BOOST_LIBS="-lboost_regex-mt -lboost_date_time-mt -lboost_filesystem-mt"
+                AC_MSG_RESULT([Using external boost library])
+        ],
+        [
+          AC_MSG_ERROR([couldn't find boost-mt library])
+        ])
+        AC_CHECK_LIB(boost_regex-mt, main, [],
+        [
+                AC_MSG_ERROR([couldn't find boost_regex-mt library])
+        ])
+        AC_CHECK_LIB(boost_date_time-mt, main, [],
+        [
+                AC_MSG_ERROR([couldn't find boost_date_time-mt library])
+        ])
+        AC_CHECK_LIB(boost_filesystem-mt, main, [],
+        [
+                AC_MSG_ERROR([couldn't find boost_filesystem-mt library])
+        ])
 fi
 ])
 
