diff --git a/k3dsdk/gzstream.cpp b/k3dsdk/gzstream.cpp
index b72f43c..4d3b1f4 100644
--- a/k3dsdk/gzstream.cpp
+++ b/k3dsdk/gzstream.cpp
@@ -31,6 +31,7 @@
 
 #include <zlib.h>
 #include <string.h>
+#include <stdio.h>
 
 namespace k3d
 {
diff --git a/k3dsdk/path.cpp b/k3dsdk/path.cpp
index 182f090..82bf386 100644
--- a/k3dsdk/path.cpp
+++ b/k3dsdk/path.cpp
@@ -35,6 +35,7 @@
 	#include <dirent.h>
 	#include <fcntl.h>
 	#include <unistd.h>
+	#include <stdio.h>
 
 #endif // !K3D_PLATFORM_WIN32
 
diff --git a/modules/jpeg/jpegreader.cpp b/modules/jpeg/jpegreader.cpp
index ef1dc35..cef5084 100644
--- a/modules/jpeg/jpegreader.cpp
+++ b/modules/jpeg/jpegreader.cpp
@@ -34,6 +34,7 @@ extern "C"
 {
 
 #include <jpeglib.h>
+#include <stdio.h>
 
 } // extern "C"
 
