--- k3d-0.6.7.0/k3dsdk/node.cpp~	2006-05-31 22:27:27.000000000 +0200
+++ k3d-0.6.7.0/k3dsdk/node.cpp	2008-04-01 11:32:56.000000000 +0200
@@ -26,6 +26,7 @@
 #include "iplugin_factory.h"
 #include "node.h"
 
+#include <sigc++/sigc++.h>
 #include <algorithm>
 #include <iostream>
 
--- k3d-0.6.7.0/k3dsdk/mesh_modifier.h~	2006-02-19 06:43:39.000000000 +0100
+++ k3d-0.6.7.0/k3dsdk/mesh_modifier.h	2008-04-01 11:32:52.000000000 +0200
@@ -27,6 +27,8 @@
 #include "imesh_source.h"
 #include "mesh.h"
 
+#include <sigc++/sigc++.h>
+
 namespace k3d
 {
 
