diff -Naurp qbittorrent-20061110/configure qbittorrent-20061110-p/configure
--- qbittorrent-20061110/configure	2006-11-10 15:56:24.000000000 +0100
+++ qbittorrent-20061110-p/configure	2006-11-12 00:48:01.000000000 +0100
@@ -256,8 +256,8 @@ class qc_libtorrent : public ConfObj
 {
 public:
 	qc_libtorrent(Conf *c) : ConfObj(c) {}
-	QString name() const { return "libtorrent >= 0.11"; }
-	QString shortname() const { return "libtorrent"; }
+	QString name() const { return "rblibtorrent >= 0.11"; }
+	QString shortname() const { return "rblibtorrent"; }
 	bool exec(){
           QString s;
           QStringList sl;
diff -Naurp qbittorrent-20061110/qbittorrent.qc qbittorrent-20061110-p/qbittorrent.qc
--- qbittorrent-20061110/qbittorrent.qc	2006-11-10 15:56:24.000000000 +0100
+++ qbittorrent-20061110-p/qbittorrent.qc	2006-11-12 00:48:01.000000000 +0100
@@ -5,7 +5,7 @@
   <dep type='qt41'>
     <required/>
   </dep>
-  <dep type='libtorrent'>
+  <dep type='rblibtorrent'>
     <required/>
   </dep>
   <dep type='libboost'>
diff -Naurp qbittorrent-20061110/qcm/libtorrent.qcm qbittorrent-20061110-p/qcm/libtorrent.qcm
--- qbittorrent-20061110/qcm/libtorrent.qcm	2006-11-10 15:56:16.000000000 +0100
+++ qbittorrent-20061110-p/qcm/libtorrent.qcm	2006-11-12 00:48:01.000000000 +0100
@@ -7,8 +7,8 @@ class qc_libtorrent : public ConfObj
 {
 public:
 	qc_libtorrent(Conf *c) : ConfObj(c) {}
-	QString name() const { return "libtorrent >= 0.11"; }
-	QString shortname() const { return "libtorrent"; }
+	QString name() const { return "rblibtorrent >= 0.11"; }
+	QString shortname() const { return "rblibtorrent"; }
 	bool exec(){
           QString s;
           QStringList sl;
diff -Naurp qbittorrent-20061110/src/src.pro qbittorrent-20061110-p/src/src.pro
--- qbittorrent-20061110/src/src.pro	2006-11-10 15:56:24.000000000 +0100
+++ qbittorrent-20061110-p/src/src.pro	2006-11-12 00:48:29.000000000 +0100
@@ -26,7 +26,7 @@ QMAKE_CXXFLAGS_RELEASE += -fwrapv
 QMAKE_CXXFLAGS_DEBUG += -fwrapv
 LIBS += -lcurl
 CONFIG += link_pkgconfig
-PKGCONFIG += libtorrent
+PKGCONFIG += rblibtorrent
 QT += xml network
 
 DEFINES += QT_NO_CAST_TO_ASCII
