diff -Naur qbittorrent-0.9.1/configure qbittorrent-0.9.1-p/configure
--- qbittorrent-0.9.1/configure	2007-04-03 23:51:26.000000000 +0200
+++ qbittorrent-0.9.1-p/configure	2007-04-05 18:21:12.000000000 +0200
@@ -295,8 +295,8 @@
 {
 public:
 	qc_libtorrent(Conf *c) : ConfObj(c) {}
-	QString name() const { return "libtorrent >= 0.12"; }
-	QString shortname() const { return "libtorrent"; }
+	QString name() const { return "rblibtorrent >= 0.12"; }
+	QString shortname() const { return "rblibtorrent"; }
 	bool exec(){
         	QString s;
 		s = conf->getenv("QC_WITH_LIBTORRENT_INC");
diff -Naur qbittorrent-0.9.1/qbittorrent.qc qbittorrent-0.9.1-p/qbittorrent.qc
--- qbittorrent-0.9.1/qbittorrent.qc	2007-03-15 16:51:11.000000000 +0100
+++ qbittorrent-0.9.1-p/qbittorrent.qc	2007-04-05 18:21:29.000000000 +0200
@@ -5,7 +5,7 @@
   <dep type='qt42'>
     <required/>
   </dep>
-  <dep type='libtorrent'>
+  <dep type='rblibtorrent'>
     <required/>
   </dep>
   <dep type='libboost'>
diff -Naur qbittorrent-0.9.1/qcm/libtorrent.qcm qbittorrent-0.9.1-p/qcm/libtorrent.qcm
--- qbittorrent-0.9.1/qcm/libtorrent.qcm	2007-04-03 23:51:26.000000000 +0200
+++ qbittorrent-0.9.1-p/qcm/libtorrent.qcm	2007-04-05 18:21:58.000000000 +0200
@@ -9,8 +9,8 @@
 {
 public:
 	qc_libtorrent(Conf *c) : ConfObj(c) {}
-	QString name() const { return "libtorrent >= 0.12"; }
-	QString shortname() const { return "libtorrent"; }
+	QString name() const { return "rblibtorrent >= 0.12"; }
+	QString shortname() const { return "rblibtorrent"; }
 	bool exec(){
         	QString s;
 		s = conf->getenv("QC_WITH_LIBTORRENT_INC");
