diff -ur kdevelop-3.2.1/parts/snippet/snippetitem.h kdevelop-3.2.1-gcc4/parts/snippet/snippetitem.h
--- kdevelop-3.2.1/parts/snippet/snippetitem.h	2005-05-23 14:09:41.000000000 +0200
+++ kdevelop-3.2.1-gcc4/parts/snippet/snippetitem.h	2005-06-10 06:22:12.266888112 +0200
@@ -14,6 +14,8 @@
 
 class QString;
 
+class SnippetGroup;
+
 /**
 This class represents one CodeSnippet-Item in the listview.
 It also holds the needed data for one snippet.
