diff -Naur lm-sensors-3-6-0/prog/init/lm_sensors.service lm-sensors-3-6-0-p/prog/init/lm_sensors.service
--- lm-sensors-3-6-0/prog/init/lm_sensors.service	2019-10-17 17:43:18.000000000 +0200
+++ lm-sensors-3-6-0-p/prog/init/lm_sensors.service	2020-01-12 13:49:11.831018069 +0100
@@ -2,12 +2,9 @@
 Description=Initialize hardware monitoring sensors
 
 [Service]
-EnvironmentFile=/etc/sysconfig/lm_sensors
 Type=oneshot
 RemainAfterExit=yes
-ExecStart=-/sbin/modprobe -qab $BUS_MODULES $HWMON_MODULES
 ExecStart=/usr/bin/sensors -s
-ExecStop=-/sbin/modprobe -qabr $BUS_MODULES $HWMON_MODULES
 
 [Install]
 WantedBy=multi-user.target
