--- boson-all-0.12/code/boson/bosonufoglwidget.cpp	2005-09-06 15:45:31.000000000 +0200
+++ boson-all-0.12/code/boson/bosonufoglwidget.cpp.gcc41	2006-06-04 22:33:01.000000000 +0200
@@ -35,11 +35,11 @@
 	BoUfoRealProfiling()
 	{
 	}
-	void BoUfoRealProfiling::push(const QString& name)
+	void push(const QString& name)
 	{
 		boProfiling->push(name);
 	}
-	void BoUfoRealProfiling::pop()
+	void pop()
 	{
 		boProfiling->pop();
 	}
--- boson-all-0.12/code/boson/qlistviewitemnumber.cpp.orig	2006-06-04 22:56:40.000000000 +0200
+++ boson-all-0.12/code/boson/qlistviewitemnumber.cpp	2006-06-04 22:57:11.000000000 +0200
@@ -51,7 +51,7 @@
 QString QListViewItemNumberPrefix::key(int column, bool ascending) const
 {
  QString k = QListViewItemNumber::key(column, ascending);
- QRegExp r("^[0-9]+(\.[0-9]+)?");
+ QRegExp r("^[0-9]+(\\.[0-9]+)?");
  if (r.search(k) >= 0) {
 	return r.cap(0);
  }
