diff -aur r1000_v1.07/src/Makefile fw_r1000_v1.07/src/Makefile
--- r1000_v1.07/src/Makefile	2007-06-22 03:38:18.000000000 +0200
+++ fw_r1000_v1.07/src/Makefile	2007-09-08 12:33:21.000000000 +0200
@@ -21,19 +21,6 @@
 # 
 ################################################################################
 
-KVER           := $(shell uname -r)
-KDIR           := /lib/modules/$(KVER)/build
-KMISC          := /lib/modules/$(KVER)/kernel/drivers/net/
-KEXT           := $(shell echo $(KVER) | sed -ne 's/^2\.[567]\..*/k/p')o
-KFLAG          := 2$(shell echo $(KVER) | sed -ne 's/^2\.[4]\..*/4/p')x
+r1000-objs := r1000_ioctl.o r1000_n.o
 
-modules:
-	$(MAKE) -f Makefile_linux24x
-	strip --strip-debug r1000.$(KEXT)
-
-clean:
-	rm -rf *.o *.ko *~ core* .dep* .*.d .*.cmd *.mod.c *.a *.s .*.flags
-
-
-install:
-	install -m 644 -c r1000.$(KEXT) $(KMISC)
+obj-m += r1000.o
diff -aur r1000_v1.07/src/r1000_n.c fw_r1000_v1.07/src/r1000_n.c
--- r1000_v1.07/src/r1000_n.c	2007-08-17 19:36:53.000000000 +0200
+++ fw_r1000_v1.07/src/r1000_n.c	2007-09-08 12:34:27.000000000 +0200
@@ -72,9 +72,9 @@
 MODULE_DEVICE_TABLE (pci, r1000_pci_tbl);
 MODULE_AUTHOR ("Realtek");
 MODULE_DESCRIPTION ("Linux device driver for Realtek Ethernet Controllers");
-MODULE_PARM (speed, "1-" __MODULE_STRING(MAX_UNITS) "i");
-MODULE_PARM (duplex, "1-" __MODULE_STRING(MAX_UNITS) "i");
-MODULE_PARM (autoneg, "1-" __MODULE_STRING(MAX_UNITS) "i");
+MODULE_PARM_DESC (speed, "1-" __MODULE_STRING(MAX_UNITS) "i");
+MODULE_PARM_DESC (duplex, "1-" __MODULE_STRING(MAX_UNITS) "i");
+MODULE_PARM_DESC (autoneg, "1-" __MODULE_STRING(MAX_UNITS) "i");
 MODULE_LICENSE("GPL");
 
 
