diff -aur r1000_v1.06/src/Makefile fw-r1000/src/Makefile
--- r1000_v1.06/src/Makefile	2007-04-16 03:01:47.000000000 +0200
+++ fw-r1000/src/Makefile	2007-05-29 18:47:17.000000000 +0200
@@ -1,20 +1,3 @@
-# ------------------------------------------------------ #
-# Makefile for RealTek Ethernet Linux Driver, 2004/04/16 #
-# ------------------------------------------------------ #
+r1000-objs := r1000_ioctl.o r1000_n.o
 
-KVER           := $(shell uname -r)
-KDIR           := /lib/modules/$(KVER)/build
-KMISC          := /lib/modules/$(KVER)/kernel/drivers/net/
-KEXT           := $(shell echo $(KVER) | sed -ne 's/^2\.[567]\..*/k/p')o
-KFLAG          := 2$(shell echo $(KVER) | sed -ne 's/^2\.[4]\..*/4/p')x
-
-modules:
-	$(MAKE) -f Makefile_linux24x
-	strip --strip-debug r1000.$(KEXT)
-
-clean:
-	rm -rf *.o *.ko *~ core* .dep* .*.d .*.cmd *.mod.c *.a *.s .*.flags
-
-
-install:
-	install -m 644 -c r1000.$(KEXT) $(KMISC)
+obj-m += r1000.o
diff -aur r1000_v1.06/src/r1000_n.c fw-r1000/src/r1000_n.c
--- r1000_v1.06/src/r1000_n.c	2007-04-17 16:24:19.000000000 +0200
+++ fw-r1000/src/r1000_n.c	2007-05-29 18:56:17.000000000 +0200
@@ -49,9 +49,9 @@
 MODULE_DEVICE_TABLE (pci, r1000_pci_tbl);
 MODULE_AUTHOR ("Realtek");
 MODULE_DESCRIPTION ("Linux device driver for Realtek Ethernet Controllers");
-MODULE_PARM (speed, "1-" __MODULE_STRING(MAX_UNITS) "i");
-MODULE_PARM (duplex, "1-" __MODULE_STRING(MAX_UNITS) "i");
-MODULE_PARM (autoneg, "1-" __MODULE_STRING(MAX_UNITS) "i");
+MODULE_PARM_DESC (speed, "1-" __MODULE_STRING(MAX_UNITS) "i");
+MODULE_PARM_DESC (duplex, "1-" __MODULE_STRING(MAX_UNITS) "i");
+MODULE_PARM_DESC (autoneg, "1-" __MODULE_STRING(MAX_UNITS) "i");
 MODULE_LICENSE("GPL");
 
 
