--- panel-plugin/weather.h.orig	2011-12-22 09:33:38.886730045 +0100
+++ panel-plugin/weather.h	2011-12-22 09:37:04.665726242 +0100
@@ -21,8 +21,8 @@
 #include <libxfce4panel/libxfce4panel.h>
 #include <libxfce4util/libxfce4util.h>
 #define PLUGIN_WEBSITE   "http://goodies.xfce.org/projects/panel-plugins/xfce4-weather-plugin"
-#define PARTNER_ID       "1121946239"
-#define LICENSE_KEY      "3c4cd39ee5dec84f"
+#define PARTNER_ID       "1003666583"
+#define LICENSE_KEY      "4128909340a9b2fc"
 
 G_BEGIN_DECLS
 
