diff -Naur monotheka-0.0.5.orig/System/monotheka.desktop.in monotheka-0.0.5/System/monotheka.desktop.in
--- monotheka-0.0.5.orig/System/monotheka.desktop.in	2005-03-13 10:45:59.000000000 +0100
+++ monotheka-0.0.5/System/monotheka.desktop.in	2005-08-12 18:56:15.008273192 +0200
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Name=Monotheka
 Comment=DivX/DVD movie database
+Comment[hu]=DivX/DVD filmadatbázis
 Exec=@BOIL_BINDIR@/monotheka
 Icon=@BOIL_DATADIR@/pixmaps/monotheka.png
 Terminal=false
