diff -Naur kdevelop-3.5.1/languages/ada/app_templates/adahello/ada-Makefile.cvs kdevelop-3.5.1-p/languages/ada/app_templates/adahello/ada-Makefile.cvs
--- kdevelop-3.5.1/languages/ada/app_templates/adahello/ada-Makefile.cvs	2005-09-10 10:22:38.000000000 +0200
+++ kdevelop-3.5.1-p/languages/ada/app_templates/adahello/ada-Makefile.cvs	2008-05-02 18:25:00.000000000 +0200
@@ -1,6 +1,7 @@
 default: all
 
 all:
+	libtoolize -v -i -c -f
 	aclocal
 	autoheader
 	automake
diff -Naur kdevelop-3.5.1/languages/cpp/app_templates/automakeempty/automakeempty-Makefile.cvs kdevelop-3.5.1-p/languages/cpp/app_templates/automakeempty/automakeempty-Makefile.cvs
--- kdevelop-3.5.1/languages/cpp/app_templates/automakeempty/automakeempty-Makefile.cvs	2007-10-08 11:50:06.000000000 +0200
+++ kdevelop-3.5.1-p/languages/cpp/app_templates/automakeempty/automakeempty-Makefile.cvs	2008-05-02 18:25:42.000000000 +0200
@@ -1,6 +1,7 @@
 default: all
 
 all:
+	libtoolize -v -i -c -f
 	aclocal
 	autoheader
 	automake
diff -Naur kdevelop-3.5.1/languages/cpp/app_templates/chello/c-Makefile.cvs kdevelop-3.5.1-p/languages/cpp/app_templates/chello/c-Makefile.cvs
--- kdevelop-3.5.1/languages/cpp/app_templates/chello/c-Makefile.cvs	2005-09-10 10:22:36.000000000 +0200
+++ kdevelop-3.5.1-p/languages/cpp/app_templates/chello/c-Makefile.cvs	2008-05-02 18:25:34.000000000 +0200
@@ -1,6 +1,7 @@
 default: all
 
 all:
+	libtoolize -v -i -c -f
 	aclocal
 	autoheader
 	automake
diff -Naur kdevelop-3.5.1/languages/cpp/app_templates/chello_gba/c-Makefile.cvs kdevelop-3.5.1-p/languages/cpp/app_templates/chello_gba/c-Makefile.cvs
--- kdevelop-3.5.1/languages/cpp/app_templates/chello_gba/c-Makefile.cvs	2005-09-10 10:22:34.000000000 +0200
+++ kdevelop-3.5.1-p/languages/cpp/app_templates/chello_gba/c-Makefile.cvs	2008-05-02 18:26:06.000000000 +0200
@@ -1,6 +1,7 @@
 default: all
 
 all:
+	libtoolize -v -i -c -f
 	aclocal
 	autoheader
 	automake
diff -Naur kdevelop-3.5.1/languages/cpp/app_templates/cppcurseshello/cpp-Makefile.cvs kdevelop-3.5.1-p/languages/cpp/app_templates/cppcurseshello/cpp-Makefile.cvs
--- kdevelop-3.5.1/languages/cpp/app_templates/cppcurseshello/cpp-Makefile.cvs	2007-05-14 09:41:17.000000000 +0200
+++ kdevelop-3.5.1-p/languages/cpp/app_templates/cppcurseshello/cpp-Makefile.cvs	2008-05-02 18:25:52.000000000 +0200
@@ -1,6 +1,7 @@
 default: all
 
 all:
+	libtoolize -v -i -c -f
 	aclocal
 	automake
 	autoconf
diff -Naur kdevelop-3.5.1/languages/cpp/app_templates/cpphello/cpp-Makefile.cvs kdevelop-3.5.1-p/languages/cpp/app_templates/cpphello/cpp-Makefile.cvs
--- kdevelop-3.5.1/languages/cpp/app_templates/cpphello/cpp-Makefile.cvs	2005-09-10 10:22:33.000000000 +0200
+++ kdevelop-3.5.1-p/languages/cpp/app_templates/cpphello/cpp-Makefile.cvs	2008-05-02 18:25:07.000000000 +0200
@@ -1,6 +1,7 @@
 default: all
 
 all:
+	libtoolize -v -i -c -f
 	aclocal
 	autoheader
 	automake
diff -Naur kdevelop-3.5.1/languages/cpp/app_templates/cppsdlhello/cpp-Makefile.cvs kdevelop-3.5.1-p/languages/cpp/app_templates/cppsdlhello/cpp-Makefile.cvs
--- kdevelop-3.5.1/languages/cpp/app_templates/cppsdlhello/cpp-Makefile.cvs	2005-09-10 10:22:34.000000000 +0200
+++ kdevelop-3.5.1-p/languages/cpp/app_templates/cppsdlhello/cpp-Makefile.cvs	2008-05-02 18:25:25.000000000 +0200
@@ -1,6 +1,7 @@
 default: all
 
 all:
+	libtoolize -v -i -c -f
 	aclocal
 	autoheader
 	automake
diff -Naur kdevelop-3.5.1/languages/cpp/app_templates/win32gui/win32gui-Makefile.cvs kdevelop-3.5.1-p/languages/cpp/app_templates/win32gui/win32gui-Makefile.cvs
--- kdevelop-3.5.1/languages/cpp/app_templates/win32gui/win32gui-Makefile.cvs	2005-09-10 10:22:36.000000000 +0200
+++ kdevelop-3.5.1-p/languages/cpp/app_templates/win32gui/win32gui-Makefile.cvs	2008-05-02 18:25:59.000000000 +0200
@@ -1,6 +1,7 @@
 default: all
 
 all:
+	libtoolize -v -i -c -f
 	aclocal
 	autoheader
 	automake
diff -Naur kdevelop-3.5.1/languages/cpp/app_templates/win32hello/win32-Makefile.cvs kdevelop-3.5.1-p/languages/cpp/app_templates/win32hello/win32-Makefile.cvs
--- kdevelop-3.5.1/languages/cpp/app_templates/win32hello/win32-Makefile.cvs	2005-09-10 10:22:31.000000000 +0200
+++ kdevelop-3.5.1-p/languages/cpp/app_templates/win32hello/win32-Makefile.cvs	2008-05-02 18:25:15.000000000 +0200
@@ -1,6 +1,7 @@
 default: all
 
 all:
+	libtoolize -v -i -c -f
 	aclocal
 	autoheader
 	automake
diff -Naur kdevelop-3.5.1/languages/fortran/app_templates/fortranhello/fortran-Makefile.cvs kdevelop-3.5.1-p/languages/fortran/app_templates/fortranhello/fortran-Makefile.cvs
--- kdevelop-3.5.1/languages/fortran/app_templates/fortranhello/fortran-Makefile.cvs	2005-09-10 10:22:41.000000000 +0200
+++ kdevelop-3.5.1-p/languages/fortran/app_templates/fortranhello/fortran-Makefile.cvs	2008-05-02 18:24:34.000000000 +0200
@@ -1,6 +1,7 @@
 default: all
 
 all:
+	libtoolize -v -i -c -f
 	aclocal
 	autoheader
 	automake
