diff -aurN pygtk-2.9.2.orig/gtk/gtk-2.10.defs pygtk-2.9.2/gtk/gtk-2.10.defs
--- pygtk-2.9.2.orig/gtk/gtk-2.10.defs	2006-06-15 18:47:17.000000000 +0200
+++ pygtk-2.9.2/gtk/gtk-2.10.defs	2006-06-22 21:01:39.000000000 +0200
@@ -1925,21 +1941,6 @@
   )
 )
 
-(define-method get_print_to_file
-  (of-object "GtkPrintSettings")
-  (c-name "gtk_print_settings_get_print_to_file")
-  (return-type "gboolean")
-)
-
-(define-method set_print_to_file
-  (of-object "GtkPrintSettings")
-  (c-name "gtk_print_settings_set_print_to_file")
-  (return-type "none")
-  (parameters
-    '("gboolean" "print_to_file")
-  )
-)
-
 (define-method get_print_pages
   (of-object "GtkPrintSettings")
   (c-name "gtk_print_settings_get_print_pages")
diff -aurN pygtk-2.9.2.orig/gtk/gtk-2.10-types.defs pygtk-2.9.2/gtk/gtk-2.10-types.defs
--- pygtk-2.9.2.orig/gtk/gtk-2.10-types.defs	2006-06-13 23:02:08.000000000 +0200
+++ pygtk-2.9.2/gtk/gtk-2.10-types.defs	2006-06-23 00:32:51.000000000 +0200
@@ -35,7 +35,7 @@
   (gtype-id "GTK_TYPE_PRINT_OPERATION")
 )
 
-(define-object PrintOperationPreview
+(define-interface PrintOperationPreview
   (in-module "Gtk")
   (c-name "GtkPrintOperationPreview")
   (gtype-id "GTK_TYPE_PRINT_OPERATION_PREVIEW")
diff -aurN pygtk-2.9.2.orig/gtk/gtk-base.defs pygtk-2.9.2/gtk/gtk-base.defs
--- pygtk-2.9.2.orig/gtk/gtk-base.defs	2006-06-08 01:12:45.000000000 +0200
+++ pygtk-2.9.2/gtk/gtk-base.defs	2006-06-16 02:52:48.000000000 +0200
@@ -383,14 +383,6 @@
   )
 )
 
-(define-function accel_group_from_accel_closure
-  (c-name "gtk_accel_group_from_accel_closure")
-  (return-type "GtkAccelGroup*")
-  (parameters
-    '("GClosure*" "closure")
-  )
-)
-
 (define-function accelerator_valid
   (c-name "gtk_accelerator_valid")
   (return-type "gboolean")
diff -aurN pygtk-2.9.2.orig/gtk/gtkunixprint.override pygtk-2.9.2/gtk/gtkunixprint.override
--- pygtk-2.9.2.orig/gtk/gtkunixprint.override	2006-06-15 14:11:35.000000000 +0200
+++ pygtk-2.9.2/gtk/gtkunixprint.override	2006-06-22 17:51:01.000000000 +0200
@@ -151,8 +151,6 @@
     static char *kwlist[] = { "callback", "user_data", NULL };
     PyObject *pyfunc, *pyarg = NULL;
     PyGtkCustomNotify *cunote;
-    GError *error = NULL;
-    gboolean ret;
   
     if (!PyArg_ParseTupleAndKeywords(args, kwargs,
                                      "O|O:gtk.unixprint.PrintJob.send",
@@ -169,18 +167,10 @@
     Py_INCREF(cunote->func);
     Py_XINCREF(cunote->data);
  
-    ret = gtk_print_job_send(GTK_PRINT_JOB(self->obj),
+    gtk_print_job_send(GTK_PRINT_JOB(self->obj),
                              pygtk_print_job_complete_func_cb,
-                             cunote, pygtk_custom_destroy_notify, &error);
+                             cunote, pygtk_custom_destroy_notify);
  
-    if (!ret) {
-        Py_XDECREF(cunote->func);
-        Py_XDECREF(cunote->data);
-        g_free(cunote);
-    }
-
-    if (pyg_error_check(&error))
-        return NULL;
- 
-    return PyBool_FromLong(ret);
+    Py_INCREF(Py_None);
+    return Py_None;
 }
