--- gsoap-2.7/soapcpp2/stdsoap2.cpp.orig	2006-06-06 13:39:10.000000000 +0200
+++ gsoap-2.7/soapcpp2/stdsoap2.cpp	2006-06-06 13:40:53.000000000 +0200
@@ -3737,7 +3737,7 @@
           const char *ext_str = OBJ_nid2sn(OBJ_obj2nid(X509_EXTENSION_get_object(ext)));
           if (!strcmp(ext_str, "subjectAltName"))
           { int j;
-            unsigned char *data;
+            unsigned const char *data;
             STACK_OF(CONF_VALUE) *val;
             X509V3_EXT_METHOD *meth = X509V3_EXT_get(ext);
             if (!meth)
