diff -ur a/Makefile.defaults b/Makefile.defaults
--- a/Makefile.defaults	2012-01-01 15:42:38.000000000 -0600
+++ b/Makefile.defaults	2012-12-06 17:03:13.047418193 -0600
@@ -55,7 +55,7 @@
 
 # Enable NetSurf's use of libwebp for displaying WebPs
 # Valid options: YES, NO
-NETSURF_USE_WEBP := NO
+NETSURF_USE_WEBP := YES
 
 # Enable NetSurf's use of gstreamer for displaying videos
 # Valid options: YES, NO
