diff -urNad mlview-0.9.0.orig/src/mlview-validator.cc mlview-0.9.0/src/mlview-validator.cc
--- mlview-0.9.0.orig/src/mlview-validator.cc	2008-04-09 00:27:53.000000000 +0100
+++ mlview-0.9.0/src/mlview-validator.cc	2008-04-09 00:28:31.000000000 +0100
@@ -26,6 +26,8 @@
 #include "mlview-validator.h"
 #include "mlview-safe-ptr-utils.h"
 
+#include <memory>
+
 #define MESSAGE_LINE_LENGTH 55
 namespace mlview
 {
diff -urNad mlview-0.9.0.orig/tests/testcompl2.cc mlview-0.9.0/tests/testcompl2.cc
--- mlview-0.9.0.orig/tests/testcompl2.cc	2008-04-09 00:27:53.000000000 +0100
+++ mlview-0.9.0/tests/testcompl2.cc	2008-04-09 00:30:47.000000000 +0100
@@ -1,4 +1,5 @@
 #include <iostream>
+#include <cstring>
 #include <libxml/xmlregexp.h>
 #include <libxml/xmlautomata.h>
 #include <libxml/valid.h>
