diff -up sip-4.19.10/siplib/siplib.c.omv~ python-sip-4.19.10/python2/siplib/siplib.c
--- sip-4.19.10/siplib/siplib.c.omv~	2018-06-26 23:12:43.738463426 +0200
+++ sip-4.19.10/siplib/siplib.c	2018-06-26 23:14:06.340466507 +0200
@@ -13073,7 +13073,11 @@ static PyObject *sipEnumType_getattro(Py
     name_str = _PyUnicode_AsString(name);
 #else
     /* We don't handle Unicode names. */
+#if PY_VERSION_HEX >= 0x03000000
     if (!PyStringCheck(name))
+#else
+    if (!PyString_Check(name))
+#endif
     {
         PyErr_Format(PyExc_TypeError,
                 "attribute name must be string, not '%.200s'",
