diff -ru linux-wlan-ng-0.2.3.orig/src/prism2/driver/prism2_cs.c linux-wlan-ng-0.2.3/src/prism2/driver/prism2_cs.c
--- linux-wlan-ng-0.2.3.orig/src/prism2/driver/prism2_cs.c	2005-09-30 18:19:26.000000000 +0200
+++ linux-wlan-ng-0.2.3/src/prism2/driver/prism2_cs.c	2006-03-30 10:46:51.000000000 +0200
@@ -136,7 +136,9 @@
 	},
 	.attach = prism2sta_attach,
 	.detach = prism2sta_detach,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,16)
 	.owner = THIS_MODULE,
+#endif
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,12)
 	.id_table = prism2_cs_ids,
 	.event = prism2sta_event,
diff -ru linux-wlan-ng-0.2.3.orig/src/prism2/driver/prism2_usb.c linux-wlan-ng-0.2.3/src/prism2/driver/prism2_usb.c
--- linux-wlan-ng-0.2.3.orig/src/prism2/driver/prism2_usb.c	2005-09-19 19:57:08.000000000 +0200
+++ linux-wlan-ng-0.2.3/src/prism2/driver/prism2_usb.c	2006-03-30 10:47:16.000000000 +0200
@@ -313,7 +313,7 @@
 
 
 struct usb_driver prism2_usb_driver = {
-#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,4,19))
+#if (LINUX_VERSION_CODE > KERNEL_VERSION(2,4,19) && LINUX_VERSION_CODE < KERNEL_VERSION(2,6,16))
 	.owner = THIS_MODULE,
 #endif
 	.name = "prism2_usb",
