--- src/p80211/p80211netdev.c
+++ src/p80211/p80211netdev.c
@@ -82,6 +82,8 @@
 #if WIRELESS_EXT > 12
 #include <net/iw_handler.h>
 #endif
+#include <net/net_namespace.h>
+
 
 /*================================================================*/
 /* Project Includes */
@@ -167,11 +169,11 @@
 	DBFENTER;
 
 #ifdef CONFIG_PROC_FS
-	if (proc_net != NULL) {
+	if (init_net.proc_net != NULL) {
 		proc_p80211 = create_proc_entry(
 				"p80211", 
 				(S_IFDIR|S_IRUGO|S_IXUGO),
-				proc_net);
+				init_net.proc_net);
 	}
 #endif
 	DBFEXIT;
@@ -196,7 +198,7 @@
 	DBFENTER;
 #ifdef CONFIG_PROC_FS
 	if (proc_p80211 != NULL) {
-		remove_proc_entry("p80211", proc_net);
+		remove_proc_entry("p80211", init_net.proc_net);
 	}
 #endif
 	DBFEXIT;
--- src/prism2/driver/prism2_cs.c
+++ src/prism2/driver/prism2_cs.c
@@ -517,7 +517,6 @@
 #endif
 
 	/* Register the network device and get assigned a name */
-	SET_MODULE_OWNER(wlandev->netdev);
 	SET_NETDEV_DEV(wlandev->netdev,  &handle_to_dev(pdev));
 	if (register_wlandev(wlandev) != 0) {
 		WLAN_LOG_NOTICE("prism2sta_cs: register_wlandev() failed.\n");
@@ -1080,7 +1079,6 @@
 	link->dev = &hw->node;
 
 	/* Register the network device and get assigned a name */
-	SET_MODULE_OWNER(wlandev->netdev);
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,11) )
 	SET_NETDEV_DEV(wlandev->netdev,  &handle_to_dev(link->handle));
 #endif
--- src/prism2/driver/prism2_pci.c
+++ src/prism2/driver/prism2_pci.c
@@ -119,7 +119,6 @@
 	/* Register the wlandev, this gets us a name and registers the
 	 * linux netdevice.
 	 */
-	SET_MODULE_OWNER(wlandev->netdev);
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0))
        SET_NETDEV_DEV(wlandev->netdev, &(pdev->dev));
 #endif
--- src/prism2/driver/prism2_plx.c
+++ src/prism2/driver/prism2_plx.c
@@ -228,7 +228,6 @@
 	/* Register the wlandev, this gets us a name and registers the
 	 * linux netdevice.
 	 */
-	SET_MODULE_OWNER(wlandev->netdev);
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0))
        SET_NETDEV_DEV(wlandev->netdev, &(pdev->dev));
 #endif
--- src/prism2/driver/prism2_usb.c
+++ src/prism2/driver/prism2_usb.c
@@ -131,7 +131,6 @@
 	/* Register the wlandev, this gets us a name and registers the
 	 * linux netdevice.
 	 */
-	SET_MODULE_OWNER(wlandev->netdev);
 #if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0))
 	SET_NETDEV_DEV(wlandev->netdev, &(interface->dev));
 #endif
