diff -Naur linux-wlan-ng-0.2.8.orig/src/p80211/p80211conv.c linux-wlan-ng-0.2.8/src/p80211/p80211conv.c
--- linux-wlan-ng-0.2.8.orig/src/p80211/p80211conv.c	2007-03-19 16:37:00.000000000 +0100
+++ linux-wlan-ng-0.2.8/src/p80211/p80211conv.c	2007-07-10 14:53:04.000000000 +0200
@@ -498,7 +498,7 @@
 	}
 
 	skb->protocol = eth_type_trans(skb, netdev);
-	skb->mac.raw = (unsigned char *) e_hdr; /* new MAC header */
+	skb->mac_header = (unsigned char *) e_hdr; /* new MAC header */
 
         /* jkriegl: process signal and noise as set in hfa384x_int_rx() */
 	/* jkriegl: only process signal/noise if requested by iwspy */
diff -Naur linux-wlan-ng-0.2.8.orig/src/p80211/p80211netdev.c linux-wlan-ng-0.2.8/src/p80211/p80211netdev.c
--- linux-wlan-ng-0.2.8.orig/src/p80211/p80211netdev.c	2007-03-19 16:37:00.000000000 +0100
+++ linux-wlan-ng-0.2.8/src/p80211/p80211netdev.c	2007-07-10 14:53:04.000000000 +0200
@@ -389,7 +389,7 @@
 
 				/* set up various data fields */
 				skb->dev = dev;
-				skb->mac.raw = skb->data ;
+				skb_reset_mac_header(skb);
 				skb->ip_summed = CHECKSUM_NONE;
 				skb->pkt_type = PACKET_OTHERHOST;
 				skb->protocol = htons(ETH_P_80211_RAW); 
diff -Naur linux-wlan-ng-0.2.8.orig/src/prism2/driver/hfa384x.c linux-wlan-ng-0.2.8/src/prism2/driver/hfa384x.c
--- linux-wlan-ng-0.2.8.orig/src/prism2/driver/hfa384x.c	2007-03-19 16:37:00.000000000 +0100
+++ linux-wlan-ng-0.2.8/src/prism2/driver/hfa384x.c	2007-07-10 14:53:04.000000000 +0200
@@ -3645,7 +3645,7 @@
 		/* the prism2 cards don't return the FCS */
 		datap = skb_put(skb, WLAN_CRC_LEN);
 		memset (datap, 0xff, WLAN_CRC_LEN);
-		skb->mac.raw = skb->data;
+		skb_reset_mac_header(skb);
 
 		/* Attach the rxmeta, set some stuff */
 		p80211skb_rxmeta_attach(wlandev, skb);
diff -Naur linux-wlan-ng-0.2.8.orig/src/prism2/driver/hfa384x_usb.c linux-wlan-ng-0.2.8/src/prism2/driver/hfa384x_usb.c
--- linux-wlan-ng-0.2.8.orig/src/prism2/driver/hfa384x_usb.c	2007-03-19 16:37:00.000000000 +0100
+++ linux-wlan-ng-0.2.8/src/prism2/driver/hfa384x_usb.c	2007-07-10 14:53:04.000000000 +0200
@@ -4255,7 +4255,7 @@
 		/* The prism2 series does not return the CRC */
 		memset(skb_put(skb, WLAN_CRC_LEN), 0xff, WLAN_CRC_LEN);
 
-		skb->mac.raw = skb->data;
+		skb_reset_mac_header(skb);
 
 		/* Attach the rxmeta, set some stuff */
 		p80211skb_rxmeta_attach(wlandev, skb);
