diff --git a/gui/start.sh b/gui/start.sh
index 389203a..d62ff1e 100644
--- a/gui/start.sh
+++ b/gui/start.sh
@@ -18,7 +18,7 @@ mount -t tmpfs none /tmp
 echo " * Initialising mtab.."
 ln -sf /proc/self/mounts /etc/mtab
 echo " * Load kernel modules.."
-modprobe isofs
+modprobe isofs ext2 ext3 ext4
 modprobe ntfs
 modprobe BusLogic
 modprobe -q ehci-hcd
diff --git a/src/plugins/hotplug.c b/src/plugins/hotplug.c
index a14bf1c..b28eee5 100644
--- a/src/plugins/hotplug.c
+++ b/src/plugins/hotplug.c
@@ -64,7 +64,7 @@ int run(GList **config)
 	fw_system("mount -t tmpfs none /tmp");
 	fw_system("mount -t tmpfs none /run");
 	fw_system("ln -sf /proc/self/mounts /etc/mtab");
-	fw_system("modprobe isofs");
+	fw_system("modprobe isofs ext2 etx3 ext4");
 	fw_system("modprobe ntfs");
 	fw_system("modprobe BusLogic");
 
