diff --git a/gui/start.sh b/gui/start.sh
index 389203a..74ae571 100644
--- a/gui/start.sh
+++ b/gui/start.sh
@@ -19,6 +19,9 @@ echo " * Initialising mtab.."
 ln -sf /proc/self/mounts /etc/mtab
 echo " * Load kernel modules.."
 modprobe isofs
+modprobe ext2
+modprobe ext3
+modprobe ext4
 modprobe ntfs
 modprobe BusLogic
 modprobe -q ehci-hcd
diff --git a/src/plugins/hotplug.c b/src/plugins/hotplug.c
index a14bf1c..3aac189 100644
--- a/src/plugins/hotplug.c
+++ b/src/plugins/hotplug.c
@@ -65,6 +65,9 @@ int run(GList **config)
 	fw_system("mount -t tmpfs none /run");
 	fw_system("ln -sf /proc/self/mounts /etc/mtab");
 	fw_system("modprobe isofs");
+	fw_system("modprobe ext2");
+	fw_system("modprobe ext3");
+	fw_system("modprobe ext4");
 	fw_system("modprobe ntfs");
 	fw_system("modprobe BusLogic");
 
