diff --git a/Makefile b/Makefile
index fd28a02..532e4a4 100644
--- a/Makefile
+++ b/Makefile
@@ -228,9 +228,9 @@ initrd_gz: clean config.mak devices initrd
 	gzip -9 -c initrd-$(CARCH).img > initrd-$(CARCH).img.gz
 
 create_usb_img:
-	dd if=/dev/zero of=frugalware-$(FWVER)-$(CARCH)-usb.img bs=516096c count=$(CYL_COUNT)
+	dd if=/dev/zero of=frugalware-$(FWVER)-$(CARCH)-usb.img bs=516096c count=$(shell echo $(CYL_COUNT) + 2 | bc)
 ifneq ($(CARCH),ppc)
-	echo -e 'n\np\n1\n\n\na\n1\nw'|/sbin/fdisk -u -C$(CYL_COUNT) -S63 -H16 frugalware-$(FWVER)-$(CARCH)-usb.img || true
+	echo -e 'n\np\n1\n\n\na\n1\nw'|/sbin/fdisk -u -C$(shell echo $(CYL_COUNT) + 2 | bc) -S63 -H16 frugalware-$(FWVER)-$(CARCH)-usb.img || true
 endif
 
 usb_img: check_root create_usb_img
