diff --git a/src/plugins/postconf.c b/src/plugins/postconf.c
index f0b3d12..da6648b 100644
--- a/src/plugins/postconf.c
+++ b/src/plugins/postconf.c
@@ -245,14 +245,11 @@ int run(GList **config)
 
 	dialog_vars.backtitle=gen_backtitle(_("Post-install configuration"));
 
-	// TODO: somehow /proc gets mounted sometimes
-	// this is just a workaround, we should find and fix the affected pkgs
-	ptr = g_strdup_printf("umount %s/proc", TARGETDIR);
-	fw_system(ptr);
-	free(ptr);
-
 	// this prevents mounting and umounting dev/proc/sys for each config
 	// tool
+	ptr = g_strdup_printf("mount none -t devtmpfs %s/dev", TARGETDIR);
+	fw_system(ptr);
+	free(ptr);
 	ptr = g_strdup_printf("mount /proc -o bind %s/proc", TARGETDIR);
 	fw_system(ptr);
 	free(ptr);
