diff -Nur hylafax-4.4.4.orig/configure hylafax-4.4.4/configure
--- hylafax-4.4.4.orig/configure	2007-11-21 15:45:23.000000000 +0100
+++ hylafax-4.4.4/configure	2008-04-20 17:36:02.000000000 +0200
@@ -1306,7 +1306,7 @@
 fi
 
 cat>dummy.C<<EOF
-#include "new.h"
+#include <new>
 struct foo {
     int x;
     foo();
@@ -1418,8 +1418,8 @@
 CheckForCXXRuntime()
 {
 cat>t.c++<<EOF
-#include "iostream.h"
-int main(){ cout << "Hello World!" << endl; return 0;}
+#include <iostream>
+int main(){ std::cout << "Hello World!" << std::endl; return 0;}
 EOF
     runMake t "t:; \${C++F} \${C++FILE} t.c++${MAKECXXOVERRIDE}"
 }
