--- jpobst-Pinta-b70b79d/autogen.sh.old	2010-07-01 15:03:14.000000000 +0000
+++ jpobst-Pinta-b70b79d/autogen.sh	2010-07-07 16:57:45.000000000 +0000
@@ -83,15 +83,7 @@
         DIE=1
 }
 
-($XBUILD /version) < /dev/null > /dev/null 2>&1 || {
-        echo
-        echo "You must have mono-xbuild installed to compile $PROJECT."
-        DIE=1
-}
 
-if test "$DIE" -eq 1; then
-        exit 1
-fi
 
 xbuild_version=`xbuild /version | grep '^XBuild' | egrep -o '([0-9]+\.?){2,}'`
 check_version "$xbuild_version" "2.4" 2> /dev/null
