diff -Nurp opencl-clang-0a5a9f67b56431ef7b9436d1af812df6dfb44975-orig/CMakeLists.txt opencl-clang-0a5a9f67b56431ef7b9436d1af812df6dfb44975/CMakeLists.txt
--- opencl-clang-0a5a9f67b56431ef7b9436d1af812df6dfb44975-orig/CMakeLists.txt	2020-02-11 09:45:20.000000000 +0000
+++ opencl-clang-0a5a9f67b56431ef7b9436d1af812df6dfb44975/CMakeLists.txt	2020-03-02 10:52:07.857275590 +0000
@@ -230,34 +230,7 @@ add_llvm_library(${TARGET_NAME} SHARED
 # The list of clang libraries is taken from clang makefile
 # (build/tools/clang/tools/driver/CMakeFiles/clang.dir/link.txt)
 # All duplicate libraries are there on purpose
-    clangBasic
-    clangCodeGen
-    clangDriver
-    clangFrontend
-    clangFrontendTool
-    clangCodeGen
-    clangRewriteFrontend
-    clangARCMigrate
-    clangStaticAnalyzerFrontend
-    clangStaticAnalyzerCheckers
-    clangStaticAnalyzerCore
-    clangCrossTU
-    clangIndex
-    clangFrontend
-    clangDriver
-    clangParse
-    clangSerialization
-    clangSema
-    clangAnalysis
-    clangEdit
-    clangFormat
-    clangToolingInclusions
-    clangToolingCore
-    clangRewrite
-    clangASTMatchers
-    clangAST
-    clangLex
-    clangBasic
+    clang-cpp
     ${ADDITIONAL_LIBS}
   )
 
