--- leatherman-1.6.0/cmake/cflags.cmake	2019-02-22 15:18:43.000000000 +0100
+++ leatherman-1.6.0.new/cmake/cflags.cmake	2019-06-16 17:45:05.713841911 +0200
@@ -3,7 +3,7 @@
 # not set CMAKE_CXX_FLAGS globally because gtest is not warning-clean.
 if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "\\w*Clang")
     if (ENABLE_CXX_WERROR)
-        set(CMAKE_CXX_FLAGS "-Werror ${CMAKE_CXX_FLAGS}")
+        set(CMAKE_CXX_FLAGS " ${CMAKE_CXX_FLAGS}")
     endif()
     set(LEATHERMAN_CXX_FLAGS "-std=c++11 -Wall -Wextra -Wno-unused-parameter -Wno-tautological-constant-out-of-range-compare ${CMAKE_CXX_FLAGS}")
 
@@ -44,7 +44,7 @@
     set(CMAKE_CXX_FLAGS "-Wno-maybe-uninitialized ${CMAKE_CXX_FLAGS}")
 
     if (ENABLE_CXX_WERROR)
-        set(CMAKE_CXX_FLAGS "-Werror ${CMAKE_CXX_FLAGS}")
+        set(CMAKE_CXX_FLAGS " ${CMAKE_CXX_FLAGS}")
     endif()
 
     # missing-field-initializers is disabled because GCC can't make up their mind how to treat C++11 initializers
