diff -Naur lablgtk-2.4.0.orig/src/ml_panel.c lablgtk-2.4.0/src/ml_panel.c
--- lablgtk-2.4.0.orig/src/ml_panel.c	2005-07-20 21:15:23.000000000 +0200
+++ lablgtk-2.4.0/src/ml_panel.c	2005-07-20 21:15:39.000000000 +0200
@@ -128,7 +128,7 @@
 static void weak_notify(gpointer data, GObject *applet)
 {
   value *glob_root = data;
-  Pointer_val(*glob_root) = NULL;
+  Field (glob_root, 1) = 0;
   ml_global_root_destroy(glob_root);
 }
 
