diff -Naur octave-2.1.73/liboctave/ArrayN.h octave-2.1.73-p/liboctave/ArrayN.h
--- octave-2.1.73/liboctave/ArrayN.h	2005-05-02 13:16:12.000000000 +0200
+++ octave-2.1.73-p/liboctave/ArrayN.h	2006-10-08 00:52:10.000000000 +0200
@@ -63,7 +63,7 @@
   ArrayN (const dim_vector& dv) : Array<T> (dv) { }
 
   ArrayN (const dim_vector& dv, const T& val)
-    : Array<T> (dv) { fill (val); }
+    : Array<T> (dv) { Array<T>::fill (val); }
 
   template <class U>
   explicit ArrayN (const Array2<U>& a) : Array<T> (a, a.dims ()) { }
diff -Naur octave-2.1.73/liboctave/DiagArray2.h octave-2.1.73-p/liboctave/DiagArray2.h
--- octave-2.1.73/liboctave/DiagArray2.h	2005-05-02 13:16:16.000000000 +0200
+++ octave-2.1.73-p/liboctave/DiagArray2.h	2006-10-08 00:52:10.000000000 +0200
@@ -125,7 +125,7 @@
     {
       this->dimensions = dim_vector (r, c);
 
-      fill (val);
+      Array<T>::fill (val);
     }
 
   DiagArray2 (const Array<T>& a) : Array<T> (a)
