diff -ur a/src/apm_driver.c b/src/apm_driver.c
--- a/src/apm_driver.c	2012-07-16 23:52:01.000000000 -0500
+++ b/src/apm_driver.c	2013-07-12 12:38:40.827298044 -0500
@@ -1784,7 +1784,6 @@
     }
 #endif
 
-    miInitializeBackingStore(pScreen);
     xf86SetBackingStore(pScreen);
     xf86SetSilkenMouse(pScreen);
 
diff -ur a/src/apm.h b/src/apm.h
--- a/src/apm.h	2012-07-16 23:52:01.000000000 -0500
+++ b/src/apm.h	2013-07-12 12:38:28.598298498 -0500
@@ -33,9 +33,6 @@
 /* All drivers initialising the SW cursor need this */
 #include "mipointer.h"
 
-/* All drivers implementing backing store need this */
-#include "mibstore.h"
-
 /* I2C support */
 #include "xf86i2c.h"
 
