--- kdebase-3.4.0/startkde.orig	2005-03-04 12:20:04.000000000 +0100
+++ kdebase-3.4.0/startkde	2005-03-20 12:15:46.000000000 +0100
@@ -3,6 +3,9 @@
 #  DEFAULT KDE STARTUP SCRIPT ( KDE-3.4 )
 #
 
+export XDG_DATA_DIRS=$KDEDIR/share
+export XDG_CONFIG_DIRS=$KDEDIR/etc/xdg
+
 # When the X server dies we get a HUP signal from xinit. We must ignore it
 # because we still need to do some cleanup.
 trap 'echo GOT SIGHUP' HUP
