--- kdebase-3.4.2.orig/startkde	2005-07-20 12:04:12.000000000 +0200
+++ kdebase-3.4.2/startkde	2005-07-31 13:03:26.000000000 +0200
@@ -217,6 +217,11 @@
   xmessage -geometry 500x100 "Could not start kdeinit. Check your installation."
 fi
 
+# Frugalware uses ivman for volume manager.
+if [ -x /usr/bin/ivman ]; then
+    /usr/bin/ivman -s --nofork &
+fi
+
 # finally, give the session control to the session manager
 # if the KDEWM environment variable has been set, then it will be used as KDE's
 # window manager instead of kwin.
