diff -Naur kdebase-3.4.2.orig/kcontrol/fonts/fonts.cpp kdebase-3.4.2/kcontrol/fonts/fonts.cpp
--- kdebase-3.4.2.orig/kcontrol/fonts/fonts.cpp	2005-05-23 14:13:33.000000000 +0200
+++ kdebase-3.4.2/kcontrol/fonts/fonts.cpp	2005-08-01 11:01:10.000000000 +0200
@@ -403,7 +403,7 @@
 
 void FontAASettings::defaults()
 {
-  excludeRange->setChecked(true);
+  excludeRange->setChecked(false);
   excludeFrom->setValue(8.0);
   excludeTo->setValue(15.0);
   useSubPixel->setChecked(false);
@@ -526,16 +526,16 @@
 
   // Keep in sync with kdelibs/kdecore/kglobalsettings.cpp
 
-  QFont f0("helvetica", 12);
-  QFont f1("courier", 12);
-  QFont f2("helvetica", 10);
-  QFont f3("helvetica", 12, QFont::Bold);
-  QFont f4("helvetica", 11);
-
-  f0.setPointSize(12);
-  f1.setPointSize(12);
-  f2.setPointSize(10);
-  f3.setPointSize(12);
+  QFont f0("Sans Serif", 13);
+  QFont f1("Monospace", 13);
+  QFont f2("Sans Serif", 13);
+  QFont f3("Luxi Sans", 13, QFont::Bold);
+  QFont f4("Sans Serif", 11);
+
+  f0.setPointSize(13);
+  f1.setPointSize(13);
+  f2.setPointSize(13);
+  f3.setPointSize(13);
   f4.setPointSize(11);
 
   defaultFontList << f0 << f1 << f2 << f0 << f3 << f4 << f0;
@@ -658,7 +658,7 @@
   for ( int i = 0; i < (int) fontUseList.count(); i++ )
     fontUseList.at( i )->setDefault();
 
-  useAA = false;
+  useAA = true;
   cbAA->setChecked(useAA);
   aaSettings->defaults();
   emit changed(true);
diff -Naur kdebase-3.4.2.orig/kcontrol/kdm/kdm-font.cpp kdebase-3.4.2/kcontrol/kdm/kdm-font.cpp
--- kdebase-3.4.2.orig/kcontrol/kdm/kdm-font.cpp	2005-05-23 14:13:53.000000000 +0200
+++ kdebase-3.4.2/kcontrol/kdm/kdm-font.cpp	2005-08-01 11:00:38.000000000 +0200
@@ -90,9 +90,9 @@
 
 void KDMFontWidget::set_def()
 {
-  stdFontChooser->setFont(QFont("helvetica", 12));
-  failFontChooser->setFont(QFont("helvetica", 12, QFont::Bold));
-  greetingFontChooser->setFont(QFont("charter", 24));
+  stdFontChooser->setFont(QFont("Sans Serif", 13));
+  failFontChooser->setFont(QFont("Sans Serif", 13, QFont::Bold));
+  greetingFontChooser->setFont(QFont("Sans Serif", 24));
 }
 
 void KDMFontWidget::save()
@@ -128,7 +128,7 @@
 void KDMFontWidget::defaults()
 {
   set_def();
-  aacb->setChecked(false);
+  aacb->setChecked(true);
 }
 
 #include "kdm-font.moc"
diff -Naur kdebase-3.4.2.orig/kcontrol/krdb/themes/Plastik/Plastik.xml kdebase-3.4.2/kcontrol/krdb/themes/Plastik/Plastik.xml
--- kdebase-3.4.2.orig/kcontrol/krdb/themes/Plastik/Plastik.xml	2005-05-23 14:13:28.000000000 +0200
+++ kdebase-3.4.2/kcontrol/krdb/themes/Plastik/Plastik.xml	2005-08-01 11:02:01.000000000 +0200
@@ -113,12 +113,12 @@
   </panel>
   <widgets name="Plastik" />
   <fonts>
-    <font value="Arial,12,-1,5,50,0,0,0,0,0" object="General" />
-    <fixed value="Courier New,10,-1,5,50,0,0,0,0,0" object="General" />
-    <toolBarFont value="Arial,12,-1,5,50,0,0,0,0,0" object="General" />
-    <menuFont value="Arial,12,-1,5,50,0,0,0,0,0" object="General" />
-    <activeFont value="Arial,12,-1,5,75,0,0,0,0,0" object="WM" />
-    <taskbarFont value="Arial,11,-1,5,50,0,0,0,0,0" object="General" />
+    <font value="Sans Serif,13,-1,5,50,0,0,0,0,0" object="General" />
+    <fixed value="Monospace,13,-1,5,50,0,0,0,0,0" object="General" />
+    <toolBarFont value="Sans Serif,13,-1,5,50,0,0,0,0,0" object="General" />
+    <menuFont value="Sans Serif,13,-1,5,50,0,0,0,0,0" object="General" />
+    <activeFont value="Luxi Sans,13,-1,5,75,0,0,0,0,0" object="WM" />
+    <taskbarFont value="Sans Serif,11,-1,5,50,0,0,0,0,0" object="General" />
     <StandardFont value="" object="FMSettings" />
   </fonts>
 </ktheme>
diff -Naur kdebase-3.4.2.orig/kdm/config.def kdebase-3.4.2/kdm/config.def
--- kdebase-3.4.2.orig/kdm/config.def	2005-05-23 14:15:07.000000000 +0200
+++ kdebase-3.4.2/kdm/config.def	2005-08-01 11:00:38.000000000 +0200
@@ -2048,37 +2048,37 @@
 
 Key: GreetFont
 Type: string
-Default: "charter,24,5,0,50,0"
-CDefault: "charter,24,bold"
+Default: "Sans Serif,24,5,0,50,0"
+CDefault: "Sans Serif,24,bold"
 User: greeter:font
-Instance: #*/"charter,20,5,0,50,0"
+Instance: #*/"Sans Serif,20,5,0,50,0"
 Comment: &
 Description:
  The font for the greeter headline.
 
 Key: StdFont
 Type: string
-Default: "helvetica,12,5,0,50,0"
-CDefault: "helvetica,12"
+Default: "Sans Serif,13,5,0,50,0"
+CDefault: "Sans Serif,13"
 User: greeter(normalFont):font
-Instance: #*/"helvetica,10,5,0,50,0"
+Instance: #*/"Sans Serif,11,5,0,50,0"
 Comment: &
 Description:
  The normal font used in the greeter.
 
 Key: FailFont
 Type: string
-Default: "helvetica,12,5,0,75,0"
-CDefault: "helvetica,12,bold"
+Default: "Sans Serif,13,5,0,75,0"
+CDefault: "Sans Serif,13,bold"
 User: greeter:font
-Instance: #*/"helvetica,10,5,0,75,0"
+Instance: #*/"Sans Serif,11,5,0,75,0"
 Comment: &
 Description:
  The font used for the <quote>Login Failed</quote> message.
 
 Key: AntiAliasing
 Type: bool
-Default: false
+Default: true
 User: greeter-c
 Instance: #*/!
 Comment: &
diff -Naur kdebase-3.4.2.orig/kpersonalizer/keyecandypage.cpp kdebase-3.4.2/kpersonalizer/keyecandypage.cpp
--- kdebase-3.4.2.orig/kpersonalizer/keyecandypage.cpp	2005-05-23 14:13:21.000000000 +0200
+++ kdebase-3.4.2/kpersonalizer/keyecandypage.cpp	2005-08-01 11:02:22.000000000 +0200
@@ -594,20 +594,9 @@
 void KEyeCandyPage::enableAntialiasingFonts(bool enable, bool reset){
 	if (!reset){
 		if ( ( (sys->isXfromXFreeInc() && (sys->getXRelease() > 40000000)) || sys->isXfromXOrg() )
-		     && sys->getRenderSupport()
-		     && (*userFonts.at(1)).contains("elvetica")) {
+		     && sys->getRenderSupport() ) {
 			QSettings().writeEntry("/qt/useXft", enable);
 			QSettings().writeEntry("/qt/enableXft", enable);
-			if (enable) {
-				KGlobal::config()->setGroup("General");
-				KGlobal::config()->writeEntry("fixed", sys->getFixedWidthFont().toString(), true, true);
-				KGlobal::config()->writeEntry("font", sys->getNormalFont().toString(), true, true);
-				KGlobal::config()->writeEntry("menuFont", sys->getNormalFont().toString(), true, true);
-				KGlobal::config()->writeEntry("taskbarFont", sys->getSmallFont().toString(), true, true);
-				KGlobal::config()->writeEntry("toolBarFont", sys->getNormalFont().toString(), true, true);
-				KGlobal::config()->setGroup("WM");
-				KGlobal::config()->writeEntry("activeFont", sys->getBoldFont().toString(), true, true);
-			}
 		}
 	} else {
 		QSettings().writeEntry("/qt/useXft", b_useXft);
@@ -717,13 +706,13 @@
 	b_useXft=QSettings().readBoolEntry("/qt/useXft");
 	b_enableXft=QSettings().readBoolEntry("/qt/enableXft");
 	KGlobal::config()->setGroup("General");
-	userFonts.append(KGlobal::config()->readEntry("fixed", "courier,10,-1,5,50,0,0,0,0,0"));
-	userFonts.append(KGlobal::config()->readEntry("font", "helvetica,12,-1,5,50,0,0,0,0,0"));
-	userFonts.append(KGlobal::config()->readEntry("menuFont", "helvetica,12,-1,5,50,0,0,0,0,0"));
-	userFonts.append(KGlobal::config()->readEntry("taskbarFont", "helvetica,11,-1,5,50,0,0,0,0,0"));
-	userFonts.append(KGlobal::config()->readEntry("toolBarFont", "helvetica,12,-1,5,50,0,0,0,0,0"));
+	userFonts.append(KGlobal::config()->readEntry("fixed", "Monospace,13,-1,5,50,0,0,0,0,0"));
+	userFonts.append(KGlobal::config()->readEntry("font", "Sans Serif,13,-1,5,50,0,0,0,0,0"));
+	userFonts.append(KGlobal::config()->readEntry("menuFont", "Sans Serif,13,-1,5,50,0,0,0,0,0"));
+	userFonts.append(KGlobal::config()->readEntry("taskbarFont", "Sans Serif,11,-1,5,50,0,0,0,0,0"));
+	userFonts.append(KGlobal::config()->readEntry("toolBarFont", "Sans Serif,13,-1,5,50,0,0,0,0,0"));
 	KGlobal::config()->setGroup("WM");
-	userFonts.append(KGlobal::config()->readEntry("activeFont", "helvetica,12,-1,5,75,0,0,0,0,0"));
+	userFonts.append(KGlobal::config()->readEntry("activeFont", "Luxi Sans,13,-1,5,75,0,0,0,0,0"));
 
 	KGlobal::config()->setGroup("PanelIcons");
 	st_UserGamma.EffectPanel=KGlobal::config()->readEntry("ActiveEffect", "none");
diff -Naur kdebase-3.4.2.orig/ksplashml/themeengine/standard/themestandard.cpp kdebase-3.4.2/ksplashml/themeengine/standard/themestandard.cpp
--- kdebase-3.4.2.orig/ksplashml/themeengine/standard/themestandard.cpp	2005-05-23 14:14:47.000000000 +0200
+++ kdebase-3.4.2/ksplashml/themeengine/standard/themestandard.cpp	2005-08-01 11:00:38.000000000 +0200
@@ -32,7 +32,7 @@
 
 ThemeStandard::ThemeStandard( QWidget *parent, const char *name, const QStringList &args )
   :ThemeEngine( parent, name, args ), mIcon(0L), mPrevIcon(0L), mIconCount(0), mStdIconWidth(-1),
-  mIconPos(WndIcon::HBottomLeft), mSbAtTop(false), mSbVisible(true), mSbPbVisible(true), mSbFontName("helvetica"),
+  mIconPos(WndIcon::HBottomLeft), mSbAtTop(false), mSbVisible(true), mSbPbVisible(true), mSbFontName("Sans Serif"),
   mSbFontSz(16), mSbFontBold(true), mSbFontItalic(false), mSbFont(QFont()), mSbFg(QColor()), mSbBg(QColor()),
   mSbIcon("run"), mIconsVisible(true), mIconsJumping(true), mSplashScreen("(Default)")
 {
@@ -148,7 +148,7 @@
   mSbVisible = cfg->readBoolEntry( "Statusbar Visible", true);
   mSbPbVisible = cfg->readBoolEntry( "Progress Visible", true);
 
-  mSbFontName = cfg->readEntry( "Statusbar Font", "helvetica" );
+  mSbFontName = cfg->readEntry( "Statusbar Font", "Sans Serif" );
   mSbFontSz = cfg->readNumEntry( "Statusbar Font Size", 16 );
   mSbFontBold = cfg->readBoolEntry( "Statusbar Font Bold", true );
   mSbFontItalic = cfg->readBoolEntry( "Statusbar Font Italic", false );
