diff -Naur kdebase-3.5.6/kicker/applets/clock/clockapplet.kcfg kdebase-3.5.6-p/kicker/applets/clock/clockapplet.kcfg
--- kdebase-3.5.6/kicker/applets/clock/clockapplet.kcfg	2005-09-10 10:25:35.000000000 +0200
+++ kdebase-3.5.6-p/kicker/applets/clock/clockapplet.kcfg	2007-02-24 03:31:42.000000000 +0100
@@ -77,7 +77,7 @@
     </entry>
     <entry name="DigitalShowDate" type="Bool" key="Show_Date">
       <label>Show date.</label>
-      <default>false</default>
+      <default>true</default>
     </entry>
     <entry name="DigitalShowDayOfWeek" type="Bool" key="Show_DayOfWeek">
       <label>Show day of week.</label>
@@ -85,7 +85,7 @@
     </entry>
     <entry name="DigitalShowFrame" type="Bool" key="Show_Frame">
       <label>Show frame.</label>
-      <default>false</default>
+      <default>true</default>
     </entry>
     <entry name="DigitalForegroundColor" type="Color" key="Foreground_Color">
       <label>Foreground color.</label>
