diff -Naur kdebase-3.5.9/kioslave/media/mediamanager/halbackend.cpp kdebase-3.5.9-p/kioslave/media/mediamanager/halbackend.cpp
--- kdebase-3.5.9/kioslave/media/mediamanager/halbackend.cpp	2008-02-13 10:40:36.000000000 +0100
+++ kdebase-3.5.9-p/kioslave/media/mediamanager/halbackend.cpp	2008-02-23 02:42:34.000000000 +0100
@@ -445,7 +445,7 @@
 
     medium->mountableState(
         libhal_volume_get_device_file(halVolume),		/* Device node */
-        libhal_volume_get_mount_point(halVolume),		/* Mount point */
+        QString::fromUtf8(libhal_volume_get_mount_point(halVolume)),	/* Mount point */
         libhal_volume_get_fstype(halVolume),			/* Filesystem type */
         libhal_volume_is_mounted(halVolume) );			/* Mounted ? */
 
