diff -Naur kdebase-3.5.4/kcontrol/fonts/fonts.cpp kdebase-3.5.4-p/kcontrol/fonts/fonts.cpp
--- kdebase-3.5.4/kcontrol/fonts/fonts.cpp	2006-07-22 10:15:24.000000000 +0200
+++ kdebase-3.5.4-p/kcontrol/fonts/fonts.cpp	2006-09-29 12:28:53.854449713 +0200
@@ -309,7 +309,7 @@
     KConfig kglobals("kdeglobals", false, false);
 
     kglobals.setGroup("General");
-    hStyle=KXftConfig::Hint::Medium;
+    hStyle=KXftConfig::Hint::Full;
     xft.setHintStyle(hStyle);
     xft.apply();  // Save this setting
     kglobals.writeEntry("XftHintStyle", KXftConfig::toStr(hStyle));
@@ -434,7 +434,7 @@
     if(hintingStyle->currentText()==i18n(KXftConfig::description((KXftConfig::Hint::Style)s).utf8()))
       return (KXftConfig::Hint::Style)s;
 
-  return KXftConfig::Hint::Medium;
+  return KXftConfig::Hint::Full;
 }
 #endif
 
diff -Naur kdebase-3.5.4/kcontrol/krdb/krdb.cpp kdebase-3.5.4-p/kcontrol/krdb/krdb.cpp
--- kdebase-3.5.4/kcontrol/krdb/krdb.cpp	2006-07-22 10:15:22.000000000 +0200
+++ kdebase-3.5.4-p/kcontrol/krdb/krdb.cpp	2006-09-29 12:28:53.875446520 +0200
@@ -511,7 +511,7 @@
   {
     kglobals.setGroup("General");
 
-    QString hintStyle(kglobals.readEntry("XftHintStyle", "hintmedium")),
+    QString hintStyle(kglobals.readEntry("XftHintStyle", "hintfull")),
             subPixel(kglobals.readEntry("XftSubPixel"));
 
     contents += "Xft.antialias: ";
