diff -Naur kdebase-3.5.1/ksplashml/kcmksplash/installer.cpp kdebase-3.5.1-p1/ksplashml/kcmksplash/installer.cpp
--- kdebase-3.5.1/ksplashml/kcmksplash/installer.cpp	2005-09-10 10:25:50.000000000 +0200
+++ kdebase-3.5.1-p1/ksplashml/kcmksplash/installer.cpp	2006-03-17 20:21:50.000000000 +0100
@@ -248,7 +248,7 @@
 //-----------------------------------------------------------------------------
 void SplashInstaller::defaults()
 {
-  mThemesList->setCurrentItem(findTheme("Default"));
+  mThemesList->setCurrentItem(findTheme("Frugalware"));
   emit changed( true );
 }
 
@@ -256,7 +256,7 @@
 {
   KConfig cnf("ksplashrc");
   cnf.setGroup("KSplash");
-  QString curTheme = cnf.readEntry("Theme","Default");
+  QString curTheme = cnf.readEntry("Theme","Frugalware");
   mThemesList->setCurrentItem(findTheme(curTheme));
   emit changed( false );
 }
diff -Naur kdebase-3.5.1/ksplashml/wndmain.cpp kdebase-3.5.1-p1/ksplashml/wndmain.cpp
--- kdebase-3.5.1/ksplashml/wndmain.cpp	2005-09-10 10:25:51.000000000 +0200
+++ kdebase-3.5.1-p1/ksplashml/wndmain.cpp	2006-03-17 20:22:07.000000000 +0100
@@ -95,7 +95,7 @@
   if (mTheme.isEmpty())
   {
     config->setGroup( "KSplash" );
-    mTheme = config->readEntry( "Theme", "Default" );
+    mTheme = config->readEntry( "Theme", "Frugalware" );
   }
   loadTheme( mTheme ); // Guaranteed to return a valid theme.
 }
