diff -Naur kdebase-3.5.6/startkde kdebase-3.5.6-p/startkde
--- kdebase-3.5.6/startkde	2007-01-15 12:32:26.000000000 +0100
+++ kdebase-3.5.6-p/startkde	2007-02-24 02:40:27.000000000 +0100
@@ -58,6 +58,7 @@
 mkdir -m 700 -p $kdehome
 mkdir -m 700 -p $kdehome/share
 mkdir -m 700 -p $kdehome/share/config
+mkdir -m 700  -p $kdehome/env
 cat >$kdehome/share/config/startupconfigkeys <<EOF
 kcminputrc Mouse cursorTheme ''
 kcminputrc Mouse cursorSize ''
@@ -70,6 +71,34 @@
 kcmrandrrc [Screen3]
 kcmfonts General forceFontDPI 0
 EOF
+## LOL gtk-qt-engine hack to set our themes etc by default 
+if [ ! -f $HOME/.gtkrc-2.0-kde ]; then
+	cat >$HOME/.gtkrc-2.0-kde <<EOF
+# This file was written by KDE
+# You can edit it in the KDE control center, under "GTK Styles and Fonts"
+
+include "/usr/share/themes/QtCurve/gtk-2.0/gtkrc"
+include "/etc/gtk-2.0/gtkrc"
+
+style "user-font"
+{
+        font_name="DejaVu Sans 12"
+}
+widget_class "*" style "user-font"
+
+gtk-theme-name="QtCurve"
+gtk-font-name="DejaVu Sans 12"
+EOF
+fi
+if [ ! -f $kdehome/env/gtk-qt-engine.rc.sh ]; then 
+	cat >$kdehome/env/gtk-qt-engine.rc.sh <<EOF
+#!/bin/bash
+
+#Make sure our customised gtkrc file is loaded.
+export GTK2_RC_FILES=$HOME/.gtkrc-2.0-kde
+EOF
+chmod 0755 $kdehome/env/gtk-qt-engine.rc.sh
+fi
 kstartupconfig
 if test $? -ne 0; then
     xmessage -geometry 500x100 "Could not start kstartupconfig. Check your installation."
@@ -283,19 +312,20 @@
 echo 'startkde: Starting up...'  1>&2
 
 # run KPersonalizer before the session, if this is the first login
-if test "$kpersonalizerrc_general_firstlogin" = "true"; then
-    # start only dcopserver, don't start whole kdeinit (takes too long)
-    echo 'startkde: Running kpersonalizer...'  1>&2
-    dcopserver
-    kwin --lock &
-    kpersonalizer --before-session
-    # handle kpersonalizer restarts (language change)
-    while test $? -eq 1; do
-        kpersonalizer --r --before-session
-    done
-    dcopquit kwin
-    dcopserver_shutdown --wait
-fi
+## comment out so the defaults are there 
+#if test "$kpersonalizerrc_general_firstlogin" = "true"; then
+#    # start only dcopserver, don't start whole kdeinit (takes too long)
+#    echo 'startkde: Running kpersonalizer...'  1>&2
+#    dcopserver
+#    kwin --lock &
+#    kpersonalizer --before-session
+#    # handle kpersonalizer restarts (language change)
+#    while test $? -eq 1; do
+#        kpersonalizer --r --before-session
+#    done
+#    dcopquit kwin
+#    dcopserver_shutdown --wait
+#fi
 
 if test -z "$dl"; then
   # the splashscreen and progress indicator
