diff -Naur kdebase-3.5.1/kcontrol/fonts/fonts.cpp kdebase-3.5.1-dejavu/kcontrol/fonts/fonts.cpp
--- kdebase-3.5.1/kcontrol/fonts/fonts.cpp	2005-10-10 17:03:43.000000000 +0200
+++ kdebase-3.5.1-dejavu/kcontrol/fonts/fonts.cpp	2006-03-10 16:13:05.000000000 +0100
@@ -508,11 +508,11 @@
 
   // Keep in sync with kdelibs/kdecore/kglobalsettings.cpp
 
-  QFont f0("Sans Serif", 10);
-  QFont f1("Monospace", 10);
-  QFont f2("Sans Serif", 10);
-  QFont f3("Sans Serif", 9, QFont::Bold);
-  QFont f4("Sans Serif", 10);
+  QFont f0("DejaVu Sans", 13);
+  QFont f1("DejaVu Sans Mono", 13);
+  QFont f2("DejaVu Sans", 13);
+  QFont f3("DejaVu Serif Condensed", 12, QFont::Bold);
+  QFont f4("DejaVu Sans", 13);
 
   f0.setPointSize(10);
   f1.setPointSize(10);
