diff -Naur kdebase-3.5.1/kcontrol/fonts/fonts.cpp kdebase-3.5.1-p/kcontrol/fonts/fonts.cpp
--- kdebase-3.5.1/kcontrol/fonts/fonts.cpp	2005-10-10 17:03:43.000000000 +0200
+++ kdebase-3.5.1-p/kcontrol/fonts/fonts.cpp	2006-03-17 20:15:26.000000000 +0100
@@ -508,17 +508,17 @@
 
   // Keep in sync with kdelibs/kdecore/kglobalsettings.cpp
 
-  QFont f0("Sans Serif", 10);
-  QFont f1("Monospace", 10);
-  QFont f2("Sans Serif", 10);
-  QFont f3("Sans Serif", 9, QFont::Bold);
-  QFont f4("Sans Serif", 10);
-
-  f0.setPointSize(10);
-  f1.setPointSize(10);
-  f2.setPointSize(10);
-  f3.setPointSize(9);
-  f4.setPointSize(10);
+  QFont f0("DejaVu Sans", 13);
+  QFont f1("DejaVu Sans Mono", 13);
+  QFont f2("DejaVu Sans", 13);
+  QFont f3("DejaVu Serif Condensed", 12, QFont::Bold);
+  QFont f4("DejaVu Sans", 13);
+
+  f0.setPointSize(13);
+  f1.setPointSize(13);
+  f2.setPointSize(13);
+  f3.setPointSize(12);
+  f4.setPointSize(13);
 
   defaultFontList << f0 << f1 << f2 << f0 << f3 << f4 << f0;
 
diff -Naur kdebase-3.5.1/kcontrol/fonts/kxftconfig.cpp kdebase-3.5.1-p/kcontrol/fonts/kxftconfig.cpp
--- kdebase-3.5.1/kcontrol/fonts/kxftconfig.cpp	2005-11-19 12:23:23.000000000 +0100
+++ kdebase-3.5.1-p/kcontrol/fonts/kxftconfig.cpp	2006-03-17 20:15:26.000000000 +0100
@@ -195,7 +195,7 @@
                 files.append(f);
         }
 #if (FC_VERSION>=20300)
-        else if(system && dExists(f) && -1!=f.find(QRegExp("/conf\\.d/?$")))
+        else if(system && dExists(f) && -1!=f.find(QRegExp("/frugalware\\.d/?$")))
             return dirSyntax(f)+constKdeRootFcFile;   // This ones good enough for me!
 #endif
     }
diff -Naur kdebase-3.5.1/kcontrol/krdb/themes/Plastik/Plastik.xml kdebase-3.5.1-p/kcontrol/krdb/themes/Plastik/Plastik.xml
--- kdebase-3.5.1/kcontrol/krdb/themes/Plastik/Plastik.xml	2005-09-10 10:25:01.000000000 +0200
+++ kdebase-3.5.1-p/kcontrol/krdb/themes/Plastik/Plastik.xml	2006-03-17 20:17:30.000000000 +0100
@@ -113,12 +113,12 @@
   </panel>
   <widgets name="Plastik" />
   <fonts>
-    <font value="Arial,12,-1,5,50,0,0,0,0,0" object="General" />
-    <fixed value="Courier New,10,-1,5,50,0,0,0,0,0" object="General" />
-    <toolBarFont value="Arial,12,-1,5,50,0,0,0,0,0" object="General" />
-    <menuFont value="Arial,12,-1,5,50,0,0,0,0,0" object="General" />
-    <activeFont value="Arial,12,-1,5,75,0,0,0,0,0" object="WM" />
-    <taskbarFont value="Arial,11,-1,5,50,0,0,0,0,0" object="General" />
+    <font value="DejaVu Sans,12,-1,5,50,0,0,0,0,0" object="General" />
+    <fixed value="DejaVu Sans,10,-1,5,50,0,0,0,0,0" object="General" />
+    <toolBarFont value="DejaVu Sans,12,-1,5,50,0,0,0,0,0" object="General" />
+    <menuFont value="DejaVu Sans,12,-1,5,50,0,0,0,0,0" object="General" />
+    <activeFont value="DejaVu Sans,12,-1,5,75,0,0,0,0,0" object="WM" />
+    <taskbarFont value="DejaVu Sans,11,-1,5,50,0,0,0,0,0" object="General" />
     <StandardFont value="" object="FMSettings" />
   </fonts>
 </ktheme>
