diff -Naur kdebase-3.5.1/startkde kdebase-3.5.1-p/startkde
--- kdebase-3.5.1/startkde	2006-01-19 18:03:19.000000000 +0100
+++ kdebase-3.5.1-p/startkde	2006-03-17 20:26:13.000000000 +0100
@@ -258,19 +258,20 @@
 echo 'startkde: Starting up...'  1>&2
 
 # run KPersonalizer before the session, if this is the first login
-if test "$kpersonalizerrc_general_firstlogin" = "true"; then
-    # start only dcopserver, don't start whole kdeinit (takes too long)
-    echo 'startkde: Running kpersonalizer...'  1>&2
-    dcopserver
-    kwin --lock &
-    kpersonalizer --before-session
-    # handle kpersonalizer restarts (language change)
-    while test $? -eq 1; do
-        kpersonalizer --r --before-session
-    done
-    dcopquit kwin
-    dcopserver_shutdown --wait
-fi
+## comment out so the defaults are there 
+#if test "$kpersonalizerrc_general_firstlogin" = "true"; then
+#    # start only dcopserver, don't start whole kdeinit (takes too long)
+#    echo 'startkde: Running kpersonalizer...'  1>&2
+#    dcopserver
+#    kwin --lock &
+#    kpersonalizer --before-session
+#    # handle kpersonalizer restarts (language change)
+#    while test $? -eq 1; do
+#        kpersonalizer --r --before-session
+#    done
+#    dcopquit kwin
+#    dcopserver_shutdown --wait
+#fi
 
 # the splashscreen and progress indicator
 case "$ksplashrc_ksplash_theme" in 
