diff -Naur kdebase-3.5.1/kcontrol/konq/desktopbehavior_impl.cpp kdebase-3.5.1-misc/kcontrol/konq/desktopbehavior_impl.cpp
--- kdebase-3.5.1/kcontrol/konq/desktopbehavior_impl.cpp	2006-01-19 18:00:48.000000000 +0100
+++ kdebase-3.5.1-misc/kcontrol/konq/desktopbehavior_impl.cpp	2006-03-10 17:07:11.000000000 +0100
@@ -304,7 +304,7 @@
     g_pConfig->setGroup( "General" );
     vrootBox->setChecked( g_pConfig->readBoolEntry( "SetVRoot", false ) );
     iconsEnabledBox->setChecked( g_pConfig->readBoolEntry( "Enabled", true ) );
-    autoLineupIconsBox->setChecked( g_pConfig->readBoolEntry( "AutoLineUpIcons", false ) );
+    autoLineupIconsBox->setChecked( g_pConfig->readBoolEntry( "AutoLineUpIcons", true ) );
 
     //
     g_pConfig->setGroup( "Mouse Buttons" );
diff -Naur kdebase-3.5.1/kdesktop/kdesktop.kcfg kdebase-3.5.1-misc/kdesktop/kdesktop.kcfg
--- kdebase-3.5.1/kdesktop/kdesktop.kcfg	2006-01-19 18:02:00.000000000 +0100
+++ kdebase-3.5.1-misc/kdesktop/kdesktop.kcfg	2006-03-10 17:07:11.000000000 +0100
@@ -51,7 +51,7 @@
       <!--     if (!config->readBoolEntry("CopyDesktopLinks", true)) -->
     </entry>
     <entry key="AutoLineUpIcons" type="Bool">
-      <default>false</default>
+      <default>true</default>
       <label>Automatically line up icons</label>
       <whatsthis>Check this option if you want to see your icons automatically aligned to the grid when you move them.</whatsthis>
     </entry>
diff -Naur kdebase-3.5.1/konsole/konsole/konsole.cpp kdebase-3.5.1-misc/konsole/konsole/konsole.cpp
--- kdebase-3.5.1/konsole/konsole/konsole.cpp	2006-01-19 18:02:25.000000000 +0100
+++ kdebase-3.5.1-misc/konsole/konsole/konsole.cpp	2006-03-10 17:18:56.000000000 +0100
@@ -1584,7 +1584,10 @@
 
       // (1) set menu items and Konsole members
 
-      QFont tmpFont = KGlobalSettings::fixedFont();
+      // KGlobalSettings::fixedFont() is nicer for a unified look, but for
+      // konsole, speed matters more than anything else - and nothing beats
+      // the speed of bitmap fonts
+      QFont tmpFont = QFont("Fixed [misc]", 14); //KGlobalSettings::fixedFont();
       defaultFont = config->readFontEntry("defaultfont", &tmpFont);
 
       //set the schema
diff -Naur kdebase-3.5.1/konsole/konsole/konsole_part.cpp kdebase-3.5.1-misc/konsole/konsole/konsole_part.cpp
--- kdebase-3.5.1/konsole/konsole/konsole_part.cpp	2005-10-10 17:04:22.000000000 +0200
+++ kdebase-3.5.1-misc/konsole/konsole/konsole_part.cpp	2006-03-10 17:18:56.000000000 +0100
@@ -561,7 +561,10 @@
 
   n_encoding = config->readNumEntry("encoding",0);
 
-  QFont tmpFont = KGlobalSettings::fixedFont();
+  // KGlobalSettings::fixedFont() is nicer for a unified look, but for
+  // konsole, speed matters more than anything else - and nothing beats
+  // the speed of bitmap fonts
+  QFont tmpFont = QFont("Fixed [misc]", 14); //KGlobalSettings::fixedFont();
   defaultFont = config->readFontEntry("defaultfont", &tmpFont);
 
   QString schema = config->readEntry("Schema");
diff -Naur kdebase-3.5.1/libkonq/konq_settings.cc kdebase-3.5.1-misc/libkonq/konq_settings.cc
--- kdebase-3.5.1/libkonq/konq_settings.cc	2005-10-10 17:04:21.000000000 +0200
+++ kdebase-3.5.1-misc/libkonq/konq_settings.cc	2006-03-10 17:08:51.000000000 +0100
@@ -90,7 +90,7 @@
   
   d->m_iconTextWidth = config->readNumEntry( "TextWidth", DEFAULT_TEXTWIDTH );
   if ( d->m_iconTextWidth == DEFAULT_TEXTWIDTH )
-    d->m_iconTextWidth = QFontMetrics(m_standardFont).width("0000000000");
+    d->m_iconTextWidth = QFontMetrics(m_standardFont).width("0000000000000");
 
   m_iconTextHeight = config->readNumEntry( "TextHeight", 0 );
   if ( m_iconTextHeight == 0 ) {
