diff -Naur kdebase-3.5.4/kioslave/media/services/media_unmount.desktop kdebase-3.5.4-p/kioslave/media/services/media_unmount.desktop
--- kdebase-3.5.4/kioslave/media/services/media_unmount.desktop	2006-05-23 13:33:34.000000000 +0200
+++ kdebase-3.5.4-p/kioslave/media/services/media_unmount.desktop	2006-08-04 15:51:48.006923537 +0200
@@ -1,5 +1,5 @@
 [Desktop Entry]
-ServiceTypes=media/cdrom_mounted,media/cdwriter_mounted,media/dvd_mounted,media/floppy5_mounted,media/floppy_mounted,media/hdd_mounted,media/nfs_mounted,media/smb_mounted,media/zip_mounted,media/vcd,media/svcd,media/dvdvideo
+ServiceTypes=media/cdrom_mounted,media/cdwriter_mounted,media/dvd_mounted,media/floppy5_mounted,media/floppy_mounted,media/hdd_mounted,media/nfs_mounted,media/smb_mounted,media/zip_mounted,media/vcd,media/svcd,media/dvdvideo,media/camera_mounted,media/removable_mounted
 Actions=MediaUnmount;
 X-KDE-Priority=TopLevel
 X-KDE-MediaNotifierHide=true
