diff -Naur kdebase-3.5.4/kioslave/media/mediamanager/halbackend.cpp kdebase-3.5.4-p/kioslave/media/mediamanager/halbackend.cpp
--- kdebase-3.5.4/kioslave/media/mediamanager/halbackend.cpp	2006-07-22 10:15:45.000000000 +0200
+++ kdebase-3.5.4-p/kioslave/media/mediamanager/halbackend.cpp	2006-08-04 01:40:38.321731768 +0200
@@ -112,6 +112,14 @@
     DBusError error;
     dbus_error_init(&error);
     dbus_connection = dbus_connection_open_private(DBUS_SYSTEM_BUS, &error);
+
+    if (!dbus_connection || dbus_error_is_set(&error)) {
+        dbus_error_free(&error);
+        libhal_ctx_free(m_halContext);
+        m_halContext = NULL;
+        return false;
+    }
+
     dbus_bus_register(dbus_connection, &error);
 
     if (dbus_error_is_set(&error)) {
