diff -ur kdebase-3.5.5.orig/kioslave/media/mediamanager/halbackend.cpp kdebase-3.5.5/kioslave/media/mediamanager/halbackend.cpp
--- kdebase-3.5.5.orig/kioslave/media/mediamanager/halbackend.cpp	2006-11-16 23:23:08.000000000 +0200
+++ kdebase-3.5.5/kioslave/media/mediamanager/halbackend.cpp	2006-11-16 23:41:15.000000000 +0200
@@ -20,6 +20,7 @@
 #include "linuxcdpolling.h"
 
 #include <stdlib.h>
+#include <locale.h>
 
 #include <kapplication.h>
 #include <qeventloop.h>
@@ -873,6 +874,15 @@
             result << tmp;
     }
 
+    if (valids.contains("locale"))
+    {
+        if (fstype == "ntfs-3g")
+        {
+            tmp = QString("locale=%1" ).arg(setlocale(LC_ALL, ""));
+            result << tmp;
+        }
+    }
+
     QString mount_point = libhal_device_get_property_QString(m_halContext, name.latin1(), "volume.mount_point");
     if (mount_point.isEmpty())
         mount_point = libhal_device_get_property_QString(m_halContext, name.latin1(), "volume.policy.desired_mount_point");
@@ -1152,6 +1162,11 @@
         soptions << QString("shortname=%1").arg(valids["shortname"]);
     }
 
+    if (valids.contains("locale"))
+    {
+        soptions << QString("locale=%1").arg(valids["locale"]);
+    }
+
     if (valids.contains("journaling"))
     {
         QString option = valids["journaling"];
