diff -Naurp kdebase-3.5.5/konqueror/konq_mainwindow.cc kdebase-3.5.5-p/konqueror/konq_mainwindow.cc
--- kdebase-3.5.5/konqueror/konq_mainwindow.cc	2006-10-01 19:32:03.000000000 +0200
+++ kdebase-3.5.5-p/konqueror/konq_mainwindow.cc	2006-10-27 19:47:43.000000000 +0200
@@ -5843,9 +5843,9 @@ static int current_memory_usage( int* li
     int usage_sum = 0;
 #if defined(KDE_MALLINFO_STDLIB) || defined(KDE_MALLINFO_MALLOC)
     // ugly hack for kdecore/malloc
-    extern int kde_malloc_is_used;
-    free( calloc( 4, 4 )); // trigger setting kde_malloc_is_used
-    if( kde_malloc_is_used )
+//    extern int kde_malloc_is_used;
+//    free( calloc( 4, 4 )); // trigger setting kde_malloc_is_used
+    if( 0 )
     {
 	struct mallinfo m = mallinfo();
 	usage_sum = m.hblkhd + m.uordblks;
