diff -Naurp amsn-0.96/amsn.desktop amsn-0.96-p/amsn.desktop
--- amsn-0.96/amsn.desktop	2006-06-24 19:48:53.000000000 +0200
+++ amsn-0.96-p/amsn.desktop	2006-11-16 17:50:33.000000000 +0100
@@ -2,9 +2,9 @@
 Encoding=UTF-8
 Name=aMSN
 Exec=amsn
-Icon=msn.png
+Icon=/usr/share/icons/hicolor/48x48/msn.png
 Info=AMSN
-Categories=Application;Network;
+Categories=KDE;GNOME;Application;Network;
 Comment=MSN Messenger for Linux
 Comment[fr]=MSN Messenger pour Linux
 Comment[de]=MSN Messenger fr Linux
diff -Naurp amsn-0.96/Makefile.in amsn-0.96-p/Makefile.in
--- amsn-0.96/Makefile.in	2006-11-01 18:05:49.000000000 +0100
+++ amsn-0.96-p/Makefile.in	2006-11-16 17:58:21.000000000 +0100
@@ -7,7 +7,7 @@ VPATH		:= $(srcdir)
 
 STATIC		:= @STATIC@
 
-doc_files	:= docs INSTALL CREDITS FAQ GNUGPL HELP README TODO AGREEMENT remote.help
+doc_files	:= docs
 
 FILES_TO_INSTALL := *.tcl hotmlog.htm amsn amsn-remote amsn-remote-CLI amsn.desktop langlist
 FILES_TO_INSTALL += $(doc_files)
