--- amsn/utils/linux/capture/libng/plugins/Rules.mk	(revision 12409)
+++ amsn/utils/linux/capture/libng/plugins/Rules.mk	(working copy)
@@ -11,7 +11,7 @@
 	rm -f $(TARGETS-plugins) $(capture_dir)/libng/plugins/*.o
 
 $(TARGETS-plugins): CFLAGS+=$(V4L_CFLAGS) -I$(capture_dir) -I$(capture_dir)/libng -I$(capture_dir)/structs
-$(TARGETS-plugins): LDFLAGS+=$(foreach rp,$(RPATH),"-Wl,-rpath=$(rp)/$(capture_dir)/libng")
+$(TARGETS-plugins): LDFLAGS+=-Wl,-rpath=$(libdir)/amsn/utils/capture/libng
 $(TARGETS-plugins): MORE_LIBS=-L$(capture_dir)/libng -lng
 
 $(TARGETS-plugins): | $(capture_dir)/libng/libng.so
--- amsn/utils/linux/capture/Rules.mk	(revision 12409)
+++ amsn/utils/linux/capture/Rules.mk	(working copy)
@@ -4,7 +4,7 @@
 $(OBJS-capture): CFLAGS+=-I$(capture_dir) -I$(capture_dir)/libng
 
 $(TARGETS-capture): MORE_LIBS=-L$(capture_dir)/libng -lng
-$(TARGETS-capture): LDFLAGS+=$(foreach rp,$(RPATH),"-Wl,-rpath=$(rp)/$(capture_dir)/libng")
+$(TARGETS-capture): LDFLAGS+=-Wl,-rpath=$(libdir)/amsn/utils/capture/libng
 
 $(TARGETS-capture): $(OBJS-capture) | $(capture_dir)/libng/libng.so
 	@$(echo_link_so)

