--- amsn/autoupdate.tcl	2008/03/11 20:50:28	9648
+++ amsn/autoupdate.tcl	2008/11/14 16:47:29	10726
@@ -540,7 +540,7 @@
 	}
 
 	#///////////////////////////////////////////////////////////////////////
-	package require http
+	package require -exact http 2.4.4
 
 	proc check_web_version { token } {
 		global version rcversion weburl
--- amsn/proxy.tcl	2008/10/27 18:05:35	10649
+++ amsn/proxy.tcl	2008/11/14 16:47:29	10726
@@ -9,7 +9,7 @@
 ::Version::setSubversionId {$Id$}
 
 package provide Proxy 0.1
-package require http
+package require -exact http 2.4.4
 
 # This should be converted to a proper package, to use with package require
 source socks.tcl	;# SOCKS5 proxy support
