Index: amsn/Makefile.in
===================================================================
--- amsn/Makefile.in	(revision 11991)
+++ amsn/Makefile.in	(revision 11992)
@@ -16,7 +16,7 @@
 FILES_TO_INSTALL += utils/pixmapscroll utils/pixmapmenu utils/contentmanager
 FILES_TO_INSTALL += utils/scalable-bg utils/sha1 utils/snit utils/BWidget-1.9.0 utils/dpbrowser
 FILES_TO_INSTALL += plugins utils/sexytile
-FILES_TO_INSTALL += utils/log utils/uri utils/combobox utils/voipcontrols
+FILES_TO_INSTALL += utils/log utils/uri utils/combobox utils/voipcontrols utils/ca-certificates.crt
 
 FILES_TO_EXCLUDE := lang/genpage.c lang/addkey.tcl lang/lang1.tmpl lang/lang2.tmpl lang/lang3.tmpl lang/langchk.sh lang/complete.pl lang/genlangfiles.c
 
Index: amsn/utils/windows/NSIS/aMSN.nsi
===================================================================
--- amsn/utils/windows/NSIS/aMSN.nsi	(revision 11991)
+++ amsn/utils/windows/NSIS/aMSN.nsi	(revision 11992)
@@ -200,6 +200,8 @@
   File /r /x "genpage.c" /x "convert.tcl" /x "missing.py" /x "sortlang" /x "addkey.tcl" /x "*.tmpl" /x "langchk.sh" /x "complete.pl" /x "genlangfiles.c" /x ".svn" "..\..\..\lang\*.*"
   SetOutPath "$INSTDIR\scripts\skins"
   File /r /x ".svn" "..\..\..\skins\default"
+  SetOutPath "$INSTDIR\scripts\utils"
+  File /r /x ".svn" "..\..\..\utils\ca-certificates.crt"
   SetOutPath "$INSTDIR\scripts\utils\base64"
   File /r /x ".svn" "..\..\..\utils\base64\*.*"
   SetOutPath "$INSTDIR\scripts\utils\sasl"
