diff -Naur hexchat-2.12.4/src/fe-gtk/joind.c hexchat-2.12.4-p/src/fe-gtk/joind.c
--- hexchat-2.12.4/src/fe-gtk/joind.c	2016-03-20 02:05:19.000000000 +0100
+++ hexchat-2.12.4-p/src/fe-gtk/joind.c	2017-10-26 22:06:49.095968066 +0200
@@ -248,7 +248,7 @@
 	if (serv->network)
 		if (g_ascii_strcasecmp(((ircnet*)serv->network)->name, "freenode") == 0)
 		{
-			gtk_entry_set_text (GTK_ENTRY (entry1), "#hexchat");
+			gtk_entry_set_text (GTK_ENTRY (entry1), "#frugalware");
 		}
 
 	gtk_widget_grab_focus (okbutton1);
