--- calligra-2.4.0/filters/words/hancomword/pole.cpp.old	2012-04-12 19:51:45.465338721 +0200
+++ calligra-2.4.0/filters/words/hancomword/pole.cpp	2012-04-12 19:55:09.036353276 +0200
@@ -867,8 +867,8 @@
         unsigned k = 109;
         unsigned mblock = header->mbat_start;
         for (unsigned r = 0; r < header->num_mbat; r++) {
-            unsigned long r = loadBigBlock(mblock, buffer2, bbat->blockSize);
-            if (r != bbat->blockSize) {
+            unsigned long r2 = loadBigBlock(mblock, buffer2, bbat->blockSize);
+            if (r2 != bbat->blockSize) {
                 delete[] buffer2;
                 return;
             }
