diff -udr yafray-0.0.8.orig/SConstruct yafray-0.0.8/SConstruct
--- yafray-0.0.8.orig/SConstruct	2005-11-27 22:25:55.000000000 +0000
+++ yafray-0.0.8/SConstruct	2005-11-27 23:20:30.000000000 +0000
@@ -4,6 +4,7 @@
 import globalinfo
 
 prefix=ARGUMENTS.get('prefix','/usr/local')
+installprefix=ARGUMENTS.get('installprefix','')
 
 ficheros = {
 'darwin' : 'darwin-settings',
diff -udr yafray-0.0.8.orig/linux-settings.py yafray-0.0.8/linux-settings.py
--- yafray-0.0.8.orig/linux-settings.py	2005-11-27 22:25:55.000000000 +0000
+++ yafray-0.0.8/linux-settings.py	2005-11-27 23:20:02.000000000 +0000
@@ -5,15 +5,18 @@
 
 srcroot = os.getcwd()
 prefix = ''
+installprefix=''
 
 def init(args): 
 	global prefix
+	global installprefix
 	prefix = args.get('prefix','/usr/local')
+	installprefix = args.get('installprefix','')
 
-def get_libpath(args): return prefix+"/lib"
-def get_pluginpath(args): return prefix+"/lib/yafray"
-def get_binpath(args): return prefix+"/bin"
-def get_confpath(args): return prefix+"/etc"
+def get_libpath(args): return installprefix+'/'+prefix+"/lib"
+def get_pluginpath(args): return installprefix+'/'+prefix+"/lib/yafray"
+def get_binpath(args): return installprefix+'/'+prefix+"/bin"
+def get_confpath(args): return installprefix+"/etc"
 def get_cxxflags(args):
 	debug=args.get('debug',0)
 	flags='-Wall -DHAVE_CONFIG_H -D_PTHREADS'
