--- zig-0.14.1/build.zig	2025-05-23 06:31:41.000000000 +0200
+++ zig-0.14.1.new/build.zig	2025-06-09 19:16:32.497349836 +0200
@@ -443,7 +443,6 @@
         .skip_non_native = skip_non_native,
         .skip_libc = skip_libc,
         .use_llvm = use_llvm,
-        .max_rss = 2 * 1024 * 1024 * 1024,
     }));
 
     test_modules_step.dependOn(tests.addModuleTests(b, .{
@@ -507,7 +506,6 @@
         .skip_libc = skip_libc,
         .use_llvm = use_llvm,
         // I observed a value of 5605064704 on the M2 CI.
-        .max_rss = 6165571174,
     }));
 
     const unit_tests_step = b.step("test-unit", "Run the compiler source unit tests");
@@ -703,7 +701,6 @@
 
     const exe = b.addExecutable(.{
         .name = "zig",
-        .max_rss = 7_800_000_000,
         .root_module = compiler_mod,
     });
     exe.stack_size = stack_size;
