diff -Naur p7zip_4.42/install.sh p7zip_4.42-p/install.sh
--- p7zip_4.42/install.sh	2006-05-19 20:52:40.000000000 +0200
+++ p7zip_4.42-p/install.sh	2006-05-29 14:49:47.000000000 +0200
@@ -1,9 +1,7 @@
 #! /bin/sh
 
 # global install
-DEST_HOME=/usr/local
-# for testing 
-# DEST_HOME=${HOME}/INSTALL/usr/local
+DEST_HOME=/usr
 DEST_BIN=${DEST_HOME}/bin
 DEST_SHARE=${DEST_HOME}/lib/p7zip
 DEST_MAN=${DEST_HOME}/man
@@ -96,7 +94,7 @@
     chmod 555 ${DEST_DIR}${DEST_BIN}/7za
   fi
 
-  if [-x bin/7zr ]
+  if [ -x bin/7zr ]
   then
     echo "- installing ${DEST_DIR}${DEST_BIN}/7zr"
     mkdir -p ${DEST_DIR}${DEST_BIN}
diff -Naur p7zip_4.42/makefile p7zip_4.42-p/makefile
--- p7zip_4.42/makefile	2006-05-28 15:00:01.000000000 +0200
+++ p7zip_4.42-p/makefile	2006-05-29 14:47:50.000000000 +0200
@@ -1,7 +1,7 @@
 
-DEST_BIN=/usr/local/bin
-DEST_SHARE=/usr/local/lib/p7zip
-DEST_MAN=/usr/local/man
+DEST_BIN=$(DESTDIR)/usr/bin
+DEST_SHARE=$(DESTDIR)/usr/lib/p7zip
+DEST_MAN=$(DESTDIR)/usr/man
 
 .PHONY: all all2 7za sfx 7z 7zr common clean tar_src tar_bin depend test test_7z test_7zr
 
diff -Naur p7zip_4.42/makefile.linux_amd64 p7zip_4.42-p/makefile.linux_amd64
--- p7zip_4.42/makefile.linux_amd64	2006-05-19 20:52:41.000000000 +0200
+++ p7zip_4.42-p/makefile.linux_amd64	2006-05-29 14:50:18.000000000 +0200
@@ -1,8 +1,8 @@
 
-OPTFLAGS=-O
+OPTFLAGS = $(CXXFLAGS) 
 
 # use "-m32" to have a 32bits executable
-ALLFLAGS=-m64 ${OPTFLAGS} -s -fPIC \
+ALLFLAGS=-m64 ${OPTFLAGS} -fPIC \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	$(LOCAL_FLAGS)
diff -Naur p7zip_4.42/makefile.machine p7zip_4.42-p/makefile.machine
--- p7zip_4.42/makefile.machine	2006-05-28 15:00:07.000000000 +0200
+++ p7zip_4.42-p/makefile.machine	2006-05-29 14:48:01.000000000 +0200
@@ -3,9 +3,9 @@
 # LITTLE_ENDIAN and BIG_ENDIAN are already defined by the system headers
 #
 
-OPTFLAGS=-O
+OPTFLAGS = $(CXXFLAGS) 
 
-ALLFLAGS=${OPTFLAGS} -s -fPIC \
+ALLFLAGS=${OPTFLAGS}  -fPIC \
     	-D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE \
 	-DNDEBUG -D_REENTRANT -DENV_UNIX \
 	$(LOCAL_FLAGS)
