diff -Naur gtk-recordMyDesktop-0.1/gtk-recordmydesktop.desktop gtk-recordMyDesktop-0.1-p/gtk-recordmydesktop.desktop
--- gtk-recordMyDesktop-0.1/gtk-recordmydesktop.desktop	2006-09-05 08:58:49.000000000 +0200
+++ gtk-recordMyDesktop-0.1-p/gtk-recordmydesktop.desktop	2006-09-05 22:14:29.336643192 +0200
@@ -1,10 +1,10 @@
 [Desktop Entry]
 Encoding=UTF-8
-_Name=gtk-recordMyDesktop
-_Comment=Frontend for recordMyDesktop
+Name=gtk-recordMyDesktop
+Comment=Frontend for recordMyDesktop
 Exec=gtk-recordMyDesktop
 Terminal=false
 Type=Application
 Icon=gtk-recordmydesktop.png
-Categories=GNOME;Application;AudioVideo;
+Categories=GNOME;KDE;Application;AudioVideo;
 StartupNotify=false
