diff -Naurp tastymenu-0.6.5/src/tastylistview.h tastymenu-0.6.5-p/src/tastylistview.h
--- tastymenu-0.6.5/src/tastylistview.h	2007-02-16 21:06:03.000000000 +0100
+++ tastymenu-0.6.5-p/src/tastylistview.h	2007-02-20 22:46:16.000000000 +0100
@@ -28,6 +28,7 @@
 #include <kservicegroup.h>
 #include <kdesktopfile.h>
 #include <qpixmap.h>
+#include <qheader.h>
 #include <qtooltip.h>
 #include <qtimer.h>
 
