diff -Naur cdparanoia-III-alpha9.8/paranoia/Makefile.in cdparanoia-III-alpha9.8-p/paranoia/Makefile.in
--- cdparanoia-III-alpha9.8/paranoia/Makefile.in	1999-12-14 05:28:04.000000000 +0100
+++ cdparanoia-III-alpha9.8-p/paranoia/Makefile.in	2007-01-26 00:21:18.000000000 +0100
@@ -44,7 +44,7 @@
 	$(RANLIB) libcdda_paranoia.a
 
 libcdda_paranoia.so: 	$(OFILES)	
-	$(CC) -fpic -shared -o libcdda_paranoia.so.0.$(VERSION) -Wl,-soname -Wl,libcdda_paranoia.so.0 $(OFILES)
+	$(CC) $(OFILES) -fpic -shared -o libcdda_paranoia.so.0.$(VERSION) -Wl,-soname -Wl,libcdda_paranoia.so.0 $(LDFLAGS) $(LIBFLAGS) $(LIBS)
 	[ -e libcdda_paranoia.so.0 ] || ln -s libcdda_paranoia.so.0.$(VERSION) libcdda_paranoia.so.0
 	[ -e libcdda_paranoia.so ] || ln -s libcdda_paranoia.so.0.$(VERSION) libcdda_paranoia.so
 
