diff --git a/po/ccinput.mo b/po/ccinput.mo
index 86e8caa..a81d321 100755
Binary files a/po/ccinput.mo and b/po/ccinput.mo differ
diff --git a/scim/IMEngine/scim_ccin_imengine.cpp b/scim/IMEngine/scim_ccin_imengine.cpp
index 0c4ad8a..0e10094 100755
--- a/scim/IMEngine/scim_ccin_imengine.cpp
+++ b/scim/IMEngine/scim_ccin_imengine.cpp
@@ -48,6 +48,7 @@
 #include <unistd.h>
 #include <iostream>
 #include <stdio.h>
+#include <libintl.h>
 
 #ifdef HAVE_CONFIG_Hu
   #include <config.h>
diff --git a/scim/SetupUI/scim_ccin_imengine_setup.cpp b/scim/SetupUI/scim_ccin_imengine_setup.cpp
index 300c128..9430e85 100755
--- a/scim/SetupUI/scim_ccin_imengine_setup.cpp
+++ b/scim/SetupUI/scim_ccin_imengine_setup.cpp
@@ -32,6 +32,8 @@
 
 #include <stdio.h>
 #include <iostream>
+#include <libintl.h>
+#include <string.h>
 
 #ifndef INPUT_METHOD_NAME 
   #define INPUT_METHOD_NAME "ccinput"
