diff -Naur deluge-0.5.4/setup.py deluge-0.5.4.new/setup.py
--- deluge-0.5.4/setup.py	2007-08-05 17:40:25.000000000 +0530
+++ deluge-0.5.4.new/setup.py	2007-08-09 00:22:20.000000000 +0530
@@ -153,8 +153,9 @@
             'boost_thread', 'z', 'pthread', 'ssl']
     print 'Libraries nomt' 
 elif boosttype == "mt":
-    librariestype = ['boost_filesystem-mt', 'boost_date_time-mt',
-            'boost_thread-mt', 'z', 'pthread', 'ssl']
+    librariestype = ['boost_filesystem-gcc42-mt',
+			'boost_date_time-gcc42-mt',
+            'boost_thread-gcc42-mt', 'z', 'pthread', 'ssl']
     print 'Libraries mt'
 
 def fetchCpp():
