diff -Naur deluge-0.5.6.2/setup.py deluge-0.5.6.2.new/setup.py
--- deluge-0.5.6.2/setup.py	2007-10-31 15:12:11.000000000 +0530
+++ deluge-0.5.6.2.new/setup.py	2007-10-31 21:21:41.000000000 +0530
@@ -167,8 +167,8 @@
             'boost_thread', 'z', 'pthread', 'ssl']
         print 'Libraries nomt' 
     elif boosttype == "mt":
-        librariestype = ['boost_filesystem-mt', 'boost_date_time-mt',
-            'boost_thread-mt', 'z', 'pthread', 'ssl']
+        librariestype = ['boost_filesystem-gcc42-mt', 'boost_date_time-gcc42-mt',
+            'boost_thread-gcc42-mt', 'z', 'pthread', 'ssl']
         print 'Libraries mt'
 else:
         librariestype = ['boost_filesystem-mt', 'boost_date_time-mt',
