diff -Naur deluge-0.5.0/setup.py deluge-0.5.0.new/setup.py
--- deluge-0.5.0/setup.py	2007-03-18 22:09:56.000000000 +0530
+++ deluge-0.5.0.new/setup.py	2007-04-29 14:08:16.000000000 +0530
@@ -85,9 +85,9 @@
                     include_dirs = ['./libtorrent', './libtorrent/include', 
                     			'./libtorrent/include/libtorrent', 
                     			'/usr/include/python' + pythonVersion],
-                    libraries = ['boost_filesystem', 'boost_date_time',
-					'boost_program_options', 'boost_regex',
-					'boost_serialization', 'boost_thread', 
+                    libraries = ['boost_filesystem-gcc42-mt', 'boost_date_time-gcc42-mt',
+					'boost_program_options-gcc42-mt', 'boost_regex-gcc42-mt',
+					'boost_serialization-gcc42-mt', 'boost_thread-gcc42-mt', 
 					'z', 'pthread'],
                     extra_compile_args = ["-Wno-missing-braces"],
                     sources = ['src/deluge_core.cpp',
