--- freetuxtv-0.1.1old/src/freetuxtv-player.h	2008-07-17 19:32:56.000000000 +0200
+++ freetuxtv-0.1.1/src/freetuxtv-player.h	2008-08-30 14:19:36.000000000 +0200
@@ -13,7 +13,7 @@
 
 #include <glib-object.h>
 #include <gtk/gtk.h>
-#include <vlc/libvlc.h>
+#include <vlc/vlc.h>
 
 G_BEGIN_DECLS

--- freetuxtv-0.1.1old/src/freetuxtv-player.c	2008-07-17 19:32:56.000000000 +0200
+++ freetuxtv-0.1.1/src/freetuxtv-player.c	2008-08-30 14:26:57.000000000 +0200
@@ -155,10 +155,13 @@
 void
 freetuxtv_player_fullscreen (FreetuxTVPlayer *self)
 {
-	libvlc_input_t *input_t;
+/*
+	libvlc_input_t * libvlc_playlist_get_input( vlcinstance *, _vlcexcep ) 
+	
 	libvlc_exception_t _vlcexcep;
         libvlc_exception_init (&_vlcexcep);
-        if(self->vlcinstance != NULL){
+	
+       if(self->vlcinstance != NULL){
 		// Mode plein écran si la chaine est en cours de lecture
 		if (libvlc_playlist_isplaying (self->vlcinstance, &_vlcexcep)) {
 			input_t = libvlc_playlist_get_input(self->vlcinstance,
@@ -168,6 +171,7 @@
 			on_vlc_exception (self, &_vlcexcep);
 		}
         }
+*/
 }
 
 static void 
