diff -Naur vnc_unixsrc/Xvnc/config/cf/vnc.def vnc_unixsrc-p/Xvnc/config/cf/vnc.def
--- vnc_unixsrc/Xvnc/config/cf/vnc.def	2006-08-10 13:01:07.000000000 +0200
+++ vnc_unixsrc-p/Xvnc/config/cf/vnc.def	2007-04-27 14:43:17.000000000 +0200
@@ -121,7 +121,7 @@
 #define ProjectRoot /usr/openwin
 #else
 #ifdef LinuxArchitecture
-#define ProjectRoot /usr/X11R6
+#define ProjectRoot /usr
 #endif
 #endif
 
diff -Naur vnc_unixsrc/Xvnc/programs/Xserver/os/wrapper.c vnc_unixsrc-p/Xvnc/programs/Xserver/os/wrapper.c
--- vnc_unixsrc/Xvnc/programs/Xserver/os/wrapper.c	2000-06-11 14:00:52.000000000 +0200
+++ vnc_unixsrc-p/Xvnc/programs/Xserver/os/wrapper.c	2007-04-27 14:44:09.000000000 +0200
@@ -39,7 +39,7 @@
 
 /* This is normally set in the Imakefile */
 #ifndef XSERVER_PATH
-#define XSERVER_PATH	"/usr/X11R6/bin/X"
+#define XSERVER_PATH	"/usr/bin/X"
 #endif
 
 #include <stdio.h>
