diff -Naur libsndfile-1.0.17.orig/configure.ac libsndfile-1.0.17/configure.ac
--- libsndfile-1.0.17.orig/configure.ac	2006-09-01 09:14:59.000000000 +0200
+++ libsndfile-1.0.17/configure.ac	2006-09-01 09:15:31.000000000 +0200
@@ -386,12 +386,12 @@
 #====================================================================================
 # Settings for the HTML documentation.
 
-htmldocdir=$prefix/share/doc/libsndfile1-dev/html
+htmldocdir=$prefix/share/doc/$PACKAGE_NAME-$PACKAGE_VERSION/html
 
 if test $prefix = "NONE" ; then
-	htmldocdir=/usr/local/share/doc/libsndfile1-dev/html
+	htmldocdir=/usr/local/share/doc/$PACKAGE_NAME-$PACKAGE_VERSION/html
 else
-	htmldocdir=$prefix/share/doc/libsndfile1-dev/html
+	htmldocdir=$prefix/share/doc/$PACKAGE_NAME-$PACKAGE_VERSION/html
 	fi
 
 if test x$enable_bow_docs = "xyes" ; then
