diff -Naurp mod_perl-2.0.3-old/Changes mod_perl-2.0.3/Changes
--- mod_perl-2.0.3-old/Changes	2006-11-29 03:02:17.000000000 -0500
+++ mod_perl-2.0.3/Changes	2007-03-30 19:05:50.000000000 -0400
@@ -10,6 +10,11 @@ Also refer to the Apache::Test changes l
 
 =over 3
 
+=item 2.0.4-dev
+
+fix unescaped variable interprolation in regular expression
+[Randal L. Schwartz <merlyn[at]stonehenge.com>, Fred Moyer <fred[at]redhotpenguin.com>] 
+
 =item 2.0.3 November 28, 2006
 
 Prevent things in %INC that are not stat() able
diff -Naurp mod_perl-2.0.3-old/ModPerl-Registry/lib/ModPerl/RegistryCooker.pm mod_perl-2.0.3/ModPerl-Registry/lib/ModPerl/RegistryCooker.pm
--- mod_perl-2.0.3-old/ModPerl-Registry/lib/ModPerl/RegistryCooker.pm	2006-11-19 18:31:41.000000000 -0500
+++ mod_perl-2.0.3/ModPerl-Registry/lib/ModPerl/RegistryCooker.pm	2007-03-30 19:08:05.000000000 -0400
@@ -337,7 +337,7 @@ sub namespace_from_uri {
     my $self = shift;
 
     my $path_info = $self->{REQ}->path_info;
-    my $script_name = $path_info && $self->{URI} =~ /$path_info$/
+    my $script_name = $path_info && $self->{URI} =~ /\Q$path_info\E$/
         ? substr($self->{URI}, 0, length($self->{URI}) - length($path_info))
         : $self->{URI};
 
