diff -Naur squashfs4.3/squashfs-tools/mksquashfs.c squashfs4.3.new/squashfs-tools/mksquashfs.c
--- squashfs-tools/mksquashfs.c	2014-05-13 00:18:20.000000000 +0200
+++ squashfs-tools/mksquashfs.c	2017-05-22 19:57:45.000000000 +0200
@@ -5131,7 +5131,16 @@
 					 argv[0]);
 				exit(1);
 			}
-			/* convert from bytes to Mbytes */
+
+			/*
+			 * convert from bytes to Mbytes, ensuring the value
+			 * does not overflow a signed int
+			 */
+			if(number >= (1LL << 51)) {
+				ERROR("%s: -mem invalid mem size\n", argv[0]);
+				exit(1);
+			}
+
 			total_mem = number / 1048576;
 			if(total_mem < (SQUASHFS_LOWMEM / SQUASHFS_TAKE)) {
 				ERROR("%s: -mem should be %d Mbytes or "
