diff -Naur squashfs-tools.orig/Makefile squashfs-tools/Makefile
--- squashfs-tools.orig/Makefile	2007-01-12 12:15:45.000000000 +0100
+++ squashfs-tools/Makefile	2007-01-12 12:30:59.000000000 +0100
@@ -1,6 +1,7 @@
 INCLUDEDIR = .
 
-CFLAGS := -I$(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -O2
+CFLAGS ?= -O2
+CFLAGS += -I$(INCLUDEDIR) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 
 all: mksquashfs unsquashfs
 
