diff -aur Makefile Makefile
--- Makefile	2011-02-28 14:04:15.000000000 -0600
+++ Makefile	2011-04-18 16:32:55.339823672 -0500
@@ -26,7 +26,7 @@
 # To build using XZ Utils liblzma - install the library and uncomment
 # the XZ_SUPPORT line below.
 #
-#XZ_SUPPORT = 1
+XZ_SUPPORT = 1
 
 
 ############ Building LZO support ##############
@@ -37,8 +37,8 @@
 # LZO_SUPPORT line below. If needed, uncomment and set LZO_DIR to the
 # installation prefix.
 #
-#LZO_SUPPORT = 1
-#LZO_DIR = /usr/local
+LZO_SUPPORT = 1
+LZO_DIR = /usr
 
 ########### Building LZMA support #############
 #
@@ -57,7 +57,7 @@
 # work) - download and unpack it, uncomment and set LZMA_DIR to unpacked source,
 # and uncomment the LZMA_SUPPORT line below.
 #
-#LZMA_XZ_SUPPORT = 1
+LZMA_XZ_SUPPORT = 1
 #LZMA_SUPPORT = 1
 #LZMA_DIR = ../../../../LZMA/lzma465
 
