diff -Naur recoll-1.5.4/mk/localdefs.in recoll-1.5.4-p/mk/localdefs.in
--- recoll-1.5.4/mk/localdefs.in	2006-01-30 10:32:28.000000000 +0100
+++ recoll-1.5.4-p/mk/localdefs.in	2006-10-06 22:39:43.000000000 +0200
@@ -12,4 +12,4 @@
 LOCALCXXFLAGS = $(INCICONV) $(XAPIANCXXFLAGS) \
 	      -DRECOLL_DATADIR=\"$(RECOLL_DATADIR)\"
 
-CXXFLAGS = -g -O2 -Wall -Wno-unused 
+CXXFLAGS += -g -Wall -Wno-unused 
