diff -Naur recoll-1.8.1/rcldb/rcldb.cpp recoll-1.8.1-p/rcldb/rcldb.cpp
--- recoll-1.8.1/rcldb/rcldb.cpp	2007-02-14 11:09:31.000000000 +0100
+++ recoll-1.8.1-p/rcldb/rcldb.cpp	2007-05-18 14:05:56.000000000 +0200
@@ -1531,7 +1531,7 @@
 		     const string& base)
 {
     Xapian::Stem stemmer(lang);
-    if (!stemmer.stem_word(word).compare(stemmer.stem_word(base))) {
+    if (!stemmer(word).compare(stemmer(base))) {
 	LOGDEB2(("Rcl::Db::stemDiffers: same for %s and %s\n", 
 		word.c_str(), base.c_str()));
 	return false;
diff -Naur recoll-1.8.1/rcldb/stemdb.cpp recoll-1.8.1-p/rcldb/stemdb.cpp
--- recoll-1.8.1/rcldb/stemdb.cpp	2007-01-19 16:19:51.000000000 +0100
+++ recoll-1.8.1-p/rcldb/stemdb.cpp	2007-05-18 14:10:27.000000000 +0200
@@ -115,7 +115,7 @@
 		// (*it).c_str(), *sit));
 		continue;
 	    }
-	    string stem = stemmer.stem_word(*it);
+	    string stem = stemmer(*it);
 	    //cerr << "word " << *it << " stem " << stem << endl;
 	    if (stem == *it) {
 		++stemconst;
@@ -226,7 +226,7 @@
 {
     try {
 	Xapian::Stem stemmer(lang);
-	string stem = stemmer.stem_word(term);
+	string stem = stemmer(term);
 	LOGDEB(("stemExpand: [%s] stem-> [%s]\n", term.c_str(), stem.c_str()));
 
 	// Open stem database
