diff -Naur libexif-gtk-0.3.5.orig/libexif-gtk/gtk-exif-content-list.c libexif-gtk-0.3.5/libexif-gtk/gtk-exif-content-list.c
--- libexif-gtk-0.3.5.orig/libexif-gtk/gtk-exif-content-list.c	2007-07-16 14:35:31.000000000 +0200
+++ libexif-gtk-0.3.5/libexif-gtk/gtk-exif-content-list.c	2004-10-17 15:54:23.000000000 +0200
@@ -405,12 +405,7 @@
 	if (!gtk_exif_content_list_get_iter (list, e, &iter)) return;
 	gtk_list_store_set (list->priv->store, &iter,
 			    VALUE_COLUMN,
-#ifdef HAVE_EXIF_0_6_12
-			    exif_entry_get_value (e, s, sizeof (s)), 
-#else
-			    exif_entry_get_value (e), 
-#endif
-			    -1);
+			    exif_entry_get_value (e, s, sizeof (s)), -1);
 }
 
 void
@@ -436,13 +431,8 @@
 	gtk_list_store_append (list->priv->store, &iter);
 	gtk_list_store_set (GTK_LIST_STORE (list->priv->store), &iter,
 			NAME_COLUMN, exif_tag_get_name (e->tag),
-			VALUE_COLUMN,
-#ifdef HAVE_EXIF_0_6_12
-				exif_entry_get_value (e, s, sizeof (s)),
-#else
-				exif_entry_get_value (e),
-#endif
-				ENTRY_COLUMN, e, -1);
+			VALUE_COLUMN, exif_entry_get_value (e, s, sizeof (s)),
+			ENTRY_COLUMN, e, -1);
 	g_signal_emit (list, signals[ENTRY_ADDED], 0, e);
 }
 
diff -Naur libexif-gtk-0.3.5.orig/libexif-gtk/gtk-exif-entry-generic.c libexif-gtk-0.3.5/libexif-gtk/gtk-exif-entry-generic.c
--- libexif-gtk-0.3.5.orig/libexif-gtk/gtk-exif-entry-generic.c	2007-07-16 14:35:31.000000000 +0200
+++ libexif-gtk-0.3.5/libexif-gtk/gtk-exif-entry-generic.c	2004-10-17 15:54:23.000000000 +0200
@@ -136,11 +136,7 @@
 	label = gtk_label_new ("Value:");
 	gtk_widget_show (label);
 	gtk_table_attach (GTK_TABLE (table), label, 0, 1, 3, 4, 0, 0, 0, 0);
-#ifdef HAVE_EXIF_0_6_12
 	label = gtk_label_new (exif_entry_get_value (e, s, sizeof (s)));
-#else
-	label = gtk_label_new (exif_entry_get_value (e));
-#endif
 	gtk_widget_show (label);
 	gtk_table_attach (GTK_TABLE (table), label, 1, 2, 3, 4, 0, 0, 0, 0);
 
