--- libadwaita-1.4.0/src/adw-about-window.c	2023-09-14 22:12:41.000000000 +0200
+++ libadwaita-1.4.0.new/src/adw-about-window.c	2023-11-24 10:49:35.792934833 +0100
@@ -2004,8 +2004,12 @@
     g_free (appid_desktop);
   }
 
-  releases = as_component_get_releases (component);
-
+#if AS_CHECK_VERSION (1, 0, 0)
+   releases = as_release_list_get_entries (as_component_get_releases_plain (component));
+#else
+   releases = as_component_get_releases (component);
+#endif
+	
   if (release_notes_version) {
     guint release_index = 0;
 
@@ -2038,12 +2042,17 @@
   }
 
   name = as_component_get_name (component);
-  developer_name = as_component_get_developer_name (component);
   project_license = as_component_get_project_license (component);
   issue_url = as_component_get_url (component, AS_URL_KIND_BUGTRACKER);
   support_url = as_component_get_url (component, AS_URL_KIND_HELP);
   website_url = as_component_get_url (component, AS_URL_KIND_HOMEPAGE);
 
+#if AS_CHECK_VERSION (1, 0, 0)
+  developer_name = as_developer_get_name (as_component_get_developer (component));
+#else
+  developer_name = as_component_get_developer_name (component);
+#endif
+
   adw_about_window_set_application_icon (self, application_id);
 
   if (name)
