--- acx100-0.3.35/pci.c.orig	2006-09-21 00:02:04.000000000 +0200
+++ acx100-0.3.35/pci.c	2006-09-21 00:03:11.000000000 +0200
@@ -33,6 +33,9 @@
 
 #include <linux/config.h>
 #include <linux/version.h>
+#ifndef UTS_RELEASE /* required for Lx 2.6.18 and above */
+#include <linux/utsrelease.h>
+#endif
 #include <linux/compiler.h> /* required for Lx 2.6.8 ?? */
 #include <linux/kernel.h>
 #include <linux/module.h>
