diff -aur acx-mac80211-20071003/acx_config.h fw_acx-mac80211-20071003/acx_config.h
--- acx-mac80211-20071003/acx_config.h	2007-10-03 17:42:39.000000000 +0200
+++ fw_acx-mac80211-20071003/acx_config.h	2007-11-01 20:06:21.000000000 +0100
@@ -1,6 +1,6 @@
 /* temporary hack until proper Kconfig integration */
-//#define CONFIG_ACX_MAC80211_PCI 1
-//#define CONFIG_ACX_MAC80211_USB 1
+#define CONFIG_ACX_MAC80211_PCI 1
+#define CONFIG_ACX_MAC80211_USB 1
 
 #define ACX_RELEASE "v0.3.36-mac80211"
 
