Only in b: kernel-3.8.patch
diff -ur a/mem.c b/mem.c
--- a/mem.c	2013-01-27 13:09:48.000000000 -0600
+++ b/mem.c	2013-05-13 06:22:33.169511069 -0500
@@ -2216,7 +2216,7 @@
  * ==================================================
  */
 
-static int __devinit acxmem_probe(struct platform_device *pdev)
+static int  acxmem_probe(struct platform_device *pdev)
 {
 	acx_device_t *adev = NULL;
 	const char *chip_name;
@@ -2392,7 +2392,7 @@
  * pdev - ptr to PCI device structure containing info about pci
  * configuration
  */
-static int __devexit acxmem_remove(struct platform_device *pdev)
+static int  acxmem_remove(struct platform_device *pdev)
 {
 	struct ieee80211_hw *hw = (struct ieee80211_hw *)
 		platform_get_drvdata(pdev);
@@ -2594,7 +2594,7 @@
 		.name = "acx-mem",
 	},
 	.probe = acxmem_probe,
-	.remove = __devexit_p(acxmem_remove),
+	.remove = acxmem_remove,
 
 #ifdef CONFIG_PM
 	.suspend = acxmem_e_suspend,
diff -ur a/pci.c b/pci.c
--- a/pci.c	2013-01-27 13:09:48.000000000 -0600
+++ b/pci.c	2013-05-13 06:22:37.275510916 -0500
@@ -1039,7 +1039,7 @@
  * id	- ptr to the device id entry that matched this device
  */
 #ifdef CONFIG_PCI
-static int __devinit acxpci_probe(struct pci_dev *pdev,
+static int  acxpci_probe(struct pci_dev *pdev,
 				const struct pci_device_id *id)
 {
 	unsigned long mem_region1 = 0;
@@ -1292,7 +1292,7 @@
  *
  * pdev - ptr to PCI device structure containing info about pci configuration
  */
-static void __devexit acxpci_remove(struct pci_dev *pdev)
+static void  acxpci_remove(struct pci_dev *pdev)
 {
 	struct ieee80211_hw *hw
 		= (struct ieee80211_hw *) pci_get_drvdata(pdev);
@@ -1478,7 +1478,7 @@
  */
 
 #if 0 // use later ?
-static struct acxpci_device_info acxpci_info_tbl[] __devinitdata = {
+static struct acxpci_device_info acxpci_info_tbl[] data = {
         [0] = {
                 .part_name      = "acx111",
                 .helper_image   = "tiacx1111r16", // probly wrong !!
@@ -1505,7 +1505,7 @@
 	.name		= "acx_pci",
 	.id_table	= acxpci_id_tbl,
 	.probe		= acxpci_probe,
-	.remove		= __devexit_p(acxpci_remove),
+	.remove		= acxpci_remove,
 #ifdef CONFIG_PM
 	.suspend	= acxpci_e_suspend,
 	.resume		= acxpci_e_resume
@@ -1603,7 +1603,7 @@
 };
 
 
-static __devinit int vlynq_probe(struct vlynq_device *vdev,
+static  int vlynq_probe(struct vlynq_device *vdev,
 				 struct vlynq_device_id *id)
 {
 	int result = -EIO, i;
@@ -1785,7 +1785,7 @@
 	return result;
 }
 
-static __devexit void vlynq_remove(struct vlynq_device *vdev)
+static  void vlynq_remove(struct vlynq_device *vdev)
 {
 	struct ieee80211_hw *hw = vlynq_get_drvdata(vdev);
 	acx_device_t *adev = hw2adev(hw);
@@ -1851,7 +1851,7 @@
 	.name = "acx_vlynq",
 	.id_table = acx_vlynq_id,
 	.probe = vlynq_probe,
-	.remove = __devexit_p(vlynq_remove),
+	.remove = vlynq_remove,
 };
 #endif /* CONFIG_VLYNQ */
 
