diff --git a/pci.c b/pci.c
index af76968..0bb678e 100644
--- a/pci.c
+++ b/pci.c
@@ -31,7 +31,7 @@
 */
 #define ACX_PCI 1
 
-#include <linux/utsrelease.h>
+#include <generated/utsrelease.h>
 #include <linux/version.h>
 #include <linux/kernel.h>
 #include <linux/module.h>
diff --git a/usb.c b/usb.c
index bde3476..e7edd1a 100644
--- a/usb.c
+++ b/usb.c
@@ -45,7 +45,7 @@
 #define ACX_USB 1
 
 #include <linux/version.h>
-#include <linux/utsrelease.h>
+#include <generated/utsrelease.h>
 #include <linux/types.h>
 #include <linux/module.h>
 #include <linux/moduleparam.h>
