--- common.c
+++ common.c
@@ -4237,8 +4237,8 @@
 	       "(type: 0x%08X, ID: %d, MAC: "
 	       MAC_FMT ")\n", conf->type, conf->if_id, MAC_ARG(conf->mac_addr));
 #else
-	       "(type: 0x%08X, ID: %d, MAC: %s)\n",
-	       conf->type, conf->if_id, print_mac(mac, conf->mac_addr));
+	       "(type: 0x%08X, MAC: %s)\n",
+	       conf->type, print_mac(mac, conf->mac_addr));
 #endif
 
       out_unlock:
@@ -4280,8 +4280,8 @@
 	       MAC_FMT ")\n",
 	       conf->type, conf->if_id, MAC_ARG(conf->mac_addr));
 #else
-	       "(type: 0x%08X, ID: %d, MAC: %s)\n",
-	       conf->type, conf->if_id, print_mac(mac, conf->mac_addr));
+	       "(type: 0x%08X, MAC: %s)\n",
+	       conf->type, print_mac(mac, conf->mac_addr));
 #endif
 	FN_EXIT0;
 }
