From 4248454395bfd8aaeecde627d31138424b5b3eb7 Mon Sep 17 00:00:00 2001
From: Frank Neuhaus <fneuhaus@uni-koblenz.de>
Date: Thu, 19 Aug 2021 13:46:38 +0200
Subject: [PATCH] Fix syntax error that occurs with gcc11

---
 include/liblas/bounds.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/liblas/bounds.hpp b/include/liblas/bounds.hpp
index e8c2e737..e35340bd 100644
--- a/include/liblas/bounds.hpp
+++ b/include/liblas/bounds.hpp
@@ -186,7 +186,7 @@ class Bounds
     
 public:
 
-Bounds<T>()
+Bounds()
 {
     ranges.resize(0);
 }
