diff -Naur acerhk-0.5.34.orig/acerhk.c acerhk-0.5.34/acerhk.c
--- acerhk-0.5.34.orig/acerhk.c	2006-08-18 18:50:35.000000000 +0200
+++ acerhk-0.5.34/acerhk.c	2006-11-30 23:25:29.000000000 +0100
@@ -35,8 +35,6 @@
  *
  */
 
-#include <linux/config.h>
-
 /* This driver is heavily dependent on the architecture, don't let anyone
  * without an X86 machine use it. I doubt that there are laptops out there
  * which would need this driver and are not X86, so it doesn't matter anyway.
@@ -45,6 +43,9 @@
 
 #include <linux/version.h>
 
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,18)
+#include <linux/config.h>
+#endif
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
 #define KERNEL26
 #include <linux/moduleparam.h>
