--- acerhk-0.5.35/acerhk.c.orig	2007-11-12 21:01:25.000000000 +0100
+++ acerhk-0.5.35/acerhk.c	2009-06-12 19:46:45.000000000 +0200
@@ -2668,7 +2668,6 @@
     printk(KERN_INFO"acerhk: could not create /proc/driver/acerhk\n");
   }
   else {
-    proc_acer_dir->owner = THIS_MODULE;
     /* now create several files, first general info ... */
     entry = create_proc_read_entry("info",
                                    0444, proc_acer_dir, acerhk_proc_info, NULL);
@@ -2677,7 +2676,6 @@
       remove_proc_entry("driver/acerhk", NULL);
       retval = 0;
     } else {
-      entry->owner = THIS_MODULE;
       /* ... last pressed key ... */
       entry = create_proc_read_entry("key",
                                      0444, proc_acer_dir, acerhk_proc_key, NULL);
@@ -2687,7 +2685,6 @@
         remove_proc_entry("driver/acerhk", NULL);
         retval = 0;
       } else {
-        entry->owner = THIS_MODULE;
         /* ... and led control file */
         entry = create_proc_entry("led", 0222, proc_acer_dir);
         if (entry == NULL) {
@@ -2699,7 +2696,6 @@
         }
         else {
           entry->write_proc = acerhk_proc_led;
-          entry->owner = THIS_MODULE;
           /* ... and wireless led controll file */
           entry = create_proc_entry("wirelessled", 0222, proc_acer_dir);
           if (entry == NULL) {
@@ -2712,7 +2708,6 @@
           }
           else {
             entry->write_proc = acerhk_proc_wirelessled;
-            entry->owner = THIS_MODULE;
             /* ... and bluetooth led controll file */
             entry = create_proc_entry("blueled", 0222, proc_acer_dir);
             if (entry == NULL) {
@@ -2725,7 +2720,6 @@
               retval = 0;
             } else {
               entry->write_proc = acerhk_proc_blueled;
-              entry->owner = THIS_MODULE;
               retval = 1;
 #ifdef ACERDEBUG
               /* add extra file for debugging purposes */
@@ -2742,7 +2736,6 @@
               }
               else {
                 entry->write_proc = acerhk_proc_debug;
-                entry->owner = THIS_MODULE;
                 retval = 1;
               }
 #endif
