--- a/vhba.c.ori	2024-02-02 11:43:46.000000000 +0100
+++ b/vhba.c	2024-09-07 12:01:23.790828243 +0200
@@ -1049,7 +1049,7 @@
     return 0;
 }
 
-static int vhba_remove (struct platform_device *pdev)
+static void vhba_remove (struct platform_device *pdev)
 {
     struct vhba_host *vhost;
     struct Scsi_Host *shost;
@@ -1061,8 +1061,6 @@
     scsi_host_put(shost);
 
     kfree(vhost->commands);
-
-    return 0;
 }
 
 static void vhba_release (struct device * dev)
