--- SuiteSparse-5.8.0/Makefile	2020-07-03 18:11:39.000000000 +0200
+++ SuiteSparse-5.8.0.new/Makefile	2020-07-14 17:07:54.622432009 +0200
@@ -16,7 +16,7 @@
 # installs all libraries SuiteSparse/lib.
 go: metis
 	( cd SuiteSparse_config && $(MAKE) )
-	( cd Mongoose && $(MAKE) CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
+	( cd Mongoose && $(MAKE) library CMAKE_OPTIONS='$(CMAKE_OPTIONS)' )
 	( cd AMD && $(MAKE) )
 	( cd BTF && $(MAKE) )
 	( cd CAMD && $(MAKE) )
