diff -Naur madwifi-ng-r1491-20060404.orig/tools/Makefile madwifi-ng-r1491-20060404/tools/Makefile
--- madwifi-ng-r1491-20060404.orig/tools/Makefile	2006-01-04 09:16:11.000000000 +0100
+++ madwifi-ng-r1491-20060404/tools/Makefile	2006-04-07 11:28:07.000000000 +0200
@@ -53,7 +53,7 @@
 all:	${ALL}
 
 INCS+=	-I. -I${HAL} -I${DEPTH}
-CFLAGS=	-g -O2 -Wall
+CFLAGS ?=	-g -O2 -Wall
 ALL_CFLAGS= ${CFLAGS} ${INCS}
 LDFLAGS=
 
