diff --exclude-from=/home/dang/.diffrc -u -ruN xerces-c-src.orig/obj/Makefile.in xerces-c-src/obj/Makefile.in
--- xerces-c-src.orig/obj/Makefile.in	2005-08-26 12:19:57.000000000 -0400
+++ xerces-c-src/obj/Makefile.in	2005-10-04 12:01:33.000000000 -0400
@@ -46,6 +46,7 @@
 include ../src/xercesc/Makefile.incl
 
 LIB=${XERCESCROOT}/lib
+MLIBDIR=lib
 ## OS390BATCH
 ifeq (${OS390BATCH},1)
 BATCH_TARGET = "//'${LOADMOD}(IXM4C56)'"
@@ -223,26 +224,26 @@
 ifeq (${OS390BATCH},1)
 	@echo "Nothing to do for OS390BATCH ..."
 else
-	-mkdir -p ${DESTDIR}${PREFIX}/lib
+	-mkdir -p ${DESTDIR}${PREFIX}/${MLIBDIR}
 ifeq (${PLATFORM}, CYGWIN)
 	-mkdir -p ${DESTDIR}${PREFIX}/bin
 	$(CP) ${FQ_SO_NAME} ${DESTDIR}${PREFIX}/bin
-	$(CP) $(XML_LIB_DIR)/${LIBNAME}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a ${DESTDIR}${PREFIX}/lib
-	(cd ${DESTDIR}$(PREFIX)/lib; rm -f ${LINK_NAME}.a libxerces.dll.a; ln -s ${LIBNAME}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a ${LINK_NAME}.a; rm -f libxerces.dll.a; ln -s ${LIBNAME}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a libxerces.dll.a)
+	$(CP) $(XML_LIB_DIR)/${LIBNAME}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a ${DESTDIR}${PREFIX}/${MLIBDIR}
+	(cd ${DESTDIR}$(PREFIX)/${MLIBDIR}; rm -f ${LINK_NAME}.a libxerces.dll.a; ln -s ${LIBNAME}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a ${LINK_NAME}.a; rm -f libxerces.dll.a; ln -s ${LIBNAME}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a libxerces.dll.a)
 
 	$(CP) ${FQ_SO_DEPDOM} ${DESTDIR}${PREFIX}/bin
-	$(CP) $(XML_LIB_DIR)/${LIBDEPDOM}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a ${DESTDIR}${PREFIX}/lib
-	(cd ${DESTDIR}$(PREFIX)/lib; rm -f ${LINK_DEPDOM}.a libxercesdepdom.dll.a; ln -s ${LIBDEPDOM}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a ${LINK_DEPDOM}.a; rm -f libxercesdepdom.dll.a; ln -s ${LIBDEPDOM}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a libxercesdepdom.dll.a)
+	$(CP) $(XML_LIB_DIR)/${LIBDEPDOM}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a ${DESTDIR}${PREFIX}/${MLIBDIR}
+	(cd ${DESTDIR}$(PREFIX)/${MLIBDIR}; rm -f ${LINK_DEPDOM}.a libxercesdepdom.dll.a; ln -s ${LIBDEPDOM}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a ${LINK_DEPDOM}.a; rm -f libxercesdepdom.dll.a; ln -s ${LIBDEPDOM}${SO_TARGET_VERSION}${SHLIBSUFFIX}.a libxercesdepdom.dll.a)
 else
-	$(CP) ${FQ_REAL_NAME} ${DESTDIR}$(PREFIX)/lib
-	(cd ${DESTDIR}$(PREFIX)/lib; rm -f ${SO_NAME}; ln -s ${REAL_NAME} ${SO_NAME}; rm -f ${LINK_NAME}; ln -s ${REAL_NAME} ${LINK_NAME})
+	$(CP) ${FQ_REAL_NAME} ${DESTDIR}$(PREFIX)/${MLIBDIR}
+	(cd ${DESTDIR}$(PREFIX)/${MLIBDIR}; rm -f ${SO_NAME}; ln -s ${REAL_NAME} ${SO_NAME}; rm -f ${LINK_NAME}; ln -s ${REAL_NAME} ${LINK_NAME})
 	
-	$(CP) ${FQ_REAL_DEPDOM} ${DESTDIR}$(PREFIX)/lib
-	(cd ${DESTDIR}$(PREFIX)/lib; rm -f ${SO_DEPDOM}; ln -s ${REAL_DEPDOM} ${SO_DEPDOM}; rm -f ${LINK_DEPDOM}; ln -s ${REAL_DEPDOM} ${LINK_DEPDOM})	
+	$(CP) ${FQ_REAL_DEPDOM} ${DESTDIR}$(PREFIX)/${MLIBDIR}
+	(cd ${DESTDIR}$(PREFIX)/${MLIBDIR}; rm -f ${SO_DEPDOM}; ln -s ${REAL_DEPDOM} ${SO_DEPDOM}; rm -f ${LINK_DEPDOM}; ln -s ${REAL_DEPDOM} ${LINK_DEPDOM})	
 endif
 ifeq (${PLATFORM},OS390)
-	$(CP) ${LIBNAME}${VER}${OS390SIDEDECK} ${DESTDIR}$(PREFIX)/lib
-	$(CP) ${LIBDEPDOM}${VER}${OS390SIDEDECK} ${DESTDIR}$(PREFIX)/lib	
+	$(CP) ${LIBNAME}${VER}${OS390SIDEDECK} ${DESTDIR}$(PREFIX)/${MLIBDIR}
+	$(CP) ${LIBDEPDOM}${VER}${OS390SIDEDECK} ${DESTDIR}$(PREFIX)/${MLIBDIR}	
 endif
 endif
 
