diff -Naur Etoile-0.4.1/Bundles/PowerMenulet/GNUmakefile Etoile-0.4.1.new/Bundles/PowerMenulet/GNUmakefile
--- Etoile-0.4.1/Bundles/PowerMenulet/GNUmakefile	2009-03-17 18:49:23.000000000 +0530
+++ Etoile-0.4.1.new/Bundles/PowerMenulet/GNUmakefile	2009-07-16 13:21:10.000000000 +0530
@@ -3,6 +3,8 @@
 BUNDLE_NAME = PowerMenulet
 BUNDLE_EXTENSION = .menulet
 
+ADDITIONAL_LDFLAGS += -L.
+
 PowerMenulet_OBJC_FILES = \
 	PowerMenulet.m 
 
diff -Naur Etoile-0.4.1/Bundles/VolumeControlMenulet/GNUmakefile Etoile-0.4.1.new/Bundles/VolumeControlMenulet/GNUmakefile
--- Etoile-0.4.1/Bundles/VolumeControlMenulet/GNUmakefile	2009-03-17 18:49:22.000000000 +0530
+++ Etoile-0.4.1.new/Bundles/VolumeControlMenulet/GNUmakefile	2009-07-16 13:21:35.000000000 +0530
@@ -3,6 +3,8 @@
 BUNDLE_NAME = VolumeControlMenulet
 BUNDLE_EXTENSION = .menulet
 
+ADDITIONAL_LDFLAGS += -L.
+
 VolumeControlMenulet_OBJC_FILES = \
 	VolumeControlMenulet.m 
 
