diff -urN sabayon-2.12.4.orig/admin-tool/lockdownappliersabayon.py sabayon-2.12.4/admin-tool/lockdownappliersabayon.py
--- sabayon-2.12.4.orig/admin-tool/lockdownappliersabayon.py	2006-12-27 14:14:38.000000000 +0000
+++ sabayon-2.12.4/admin-tool/lockdownappliersabayon.py	2006-12-27 14:23:21.000000000 +0000
@@ -20,7 +20,7 @@
 import util
 import sessionwindow
 
-from sabayon.lockdown import lockdownapplier
+from Pessulus import lockdownapplier
 
 def dprint (fmt, *args):
     util.debug_print (util.DEBUG_ADMINTOOL, fmt % args)
diff -urN sabayon-2.12.4.orig/admin-tool/sessionwindow.py sabayon-2.12.4/admin-tool/sessionwindow.py
--- sabayon-2.12.4.orig/admin-tool/sessionwindow.py	2006-12-27 14:14:38.000000000 +0000
+++ sabayon-2.12.4/admin-tool/sessionwindow.py	2006-12-27 14:23:21.000000000 +0000
@@ -29,7 +29,7 @@
 import saveconfirm
 import aboutdialog
 from lockdownappliersabayon import LockdownApplierSabayon
-from sabayon.lockdown import maindialog as lockdowndialog
+from Pessulus import maindialog as lockdowndialog
 from config import *
 
 _ui_string = '''
