diff -Naur sabayon-2.20.1.orig/admin-tool/lockdownappliersabayon.py sabayon-2.20.1/admin-tool/lockdownappliersabayon.py
--- sabayon-2.20.1.orig/admin-tool/lockdownappliersabayon.py	2007-11-02 11:08:21.000000000 +0100
+++ sabayon-2.20.1/admin-tool/lockdownappliersabayon.py	2007-11-02 11:09:16.000000000 +0100
@@ -22,7 +22,7 @@
 
 import debuglog
 import errors
-from sabayon.lockdown import lockdownapplier
+from Pessulus import lockdownapplier
 
 def dprint (fmt, *args):
     debuglog.debug_log (False, debuglog.DEBUG_LOG_DOMAIN_ADMIN_TOOL, fmt % args)
diff -Naur sabayon-2.20.1.orig/admin-tool/sessionwindow.py sabayon-2.20.1/admin-tool/sessionwindow.py
--- sabayon-2.20.1.orig/admin-tool/sessionwindow.py	2007-11-02 11:08:21.000000000 +0100
+++ sabayon-2.20.1/admin-tool/sessionwindow.py	2007-11-02 11:09:59.000000000 +0100
@@ -31,7 +31,7 @@
 import debuglog
 import errors
 from lockdownappliersabayon import LockdownApplierSabayon
-from sabayon.lockdown import maindialog as lockdowndialog
+from Pessulus import maindialog as lockdowndialog
 from config import *
 
 _ui_string = '''
