diff -urN sabayon-2.17.90.orig/admin-tool/sabayon.desktop.in sabayon-2.17.90/admin-tool/sabayon.desktop.in
--- sabayon-2.17.90.orig/admin-tool/sabayon.desktop.in	2007-02-10 16:12:17.000000000 +0000
+++ sabayon-2.17.90/admin-tool/sabayon.desktop.in	2007-02-10 16:12:29.000000000 +0000
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 _Name=User Profile Editor
 _Comment=Establish and Edit Profiles for Users
-Exec=sabayon
+Exec=gksu sabayon
 Terminal=false
 Type=Application
 Icon=sabayon.png
