diff -Naur sabayon-2.22.0.orig/admin-tool/sabayon.desktop sabayon-2.22.0/admin-tool/sabayon.desktop
--- sabayon-2.22.0.orig/admin-tool/sabayon.desktop	2008-04-15 15:42:19.000000000 +0200
+++ sabayon-2.22.0/admin-tool/sabayon.desktop	2008-04-15 15:42:38.000000000 +0200
@@ -123,7 +123,7 @@
 Comment[zh_CN]=为用户建立并编辑配置文件
 Comment[zh_HK]=建立與編輯使用者設定組合
 Comment[zh_TW]=建立與編輯使用者設定組合
-Exec=sabayon
+Exec=gksu sabayon
 Terminal=false
 Type=Application
 Icon=sabayon
diff -Naur sabayon-2.22.0.orig/admin-tool/sabayon.desktop.in sabayon-2.22.0/admin-tool/sabayon.desktop.in
--- sabayon-2.22.0.orig/admin-tool/sabayon.desktop.in	2008-04-15 15:42:19.000000000 +0200
+++ sabayon-2.22.0/admin-tool/sabayon.desktop.in	2008-04-15 15:42:49.000000000 +0200
@@ -1,7 +1,7 @@
 [Desktop Entry]
 _Name=User Profile Editor
 _Comment=Establish and Edit Profiles for Users
-Exec=sabayon
+Exec=gksu sabayon
 Terminal=false
 Type=Application
 Icon=sabayon
diff -Naur sabayon-2.22.0.orig/admin-tool/sabayon.desktop.in.in sabayon-2.22.0/admin-tool/sabayon.desktop.in.in
--- sabayon-2.22.0.orig/admin-tool/sabayon.desktop.in.in	2008-04-15 15:42:19.000000000 +0200
+++ sabayon-2.22.0/admin-tool/sabayon.desktop.in.in	2008-04-15 15:42:46.000000000 +0200
@@ -1,7 +1,7 @@
 [Desktop Entry]
 _Name=User Profile Editor
 _Comment=Establish and Edit Profiles for Users
-Exec=sabayon
+Exec=gksu sabayon
 Terminal=false
 Type=Application
 Icon=sabayon
