diff -Naur sabayon-2.20.1.orig/admin-tool/sabayon.desktop sabayon-2.20.1/admin-tool/sabayon.desktop
--- sabayon-2.20.1.orig/admin-tool/sabayon.desktop	2007-11-02 11:08:21.000000000 +0100
+++ sabayon-2.20.1/admin-tool/sabayon.desktop	2007-11-02 11:11:02.000000000 +0100
@@ -119,7 +119,7 @@
 Comment[zh_CN]=为用户建立并编辑配置文件
 Comment[zh_HK]=建立與編輯使用者設定組合
 Comment[zh_TW]=建立與編輯使用者設定組合
-Exec=gnomesu sabayon
+Exec=gksu sabayon
 Terminal=false
 Type=Application
 Icon=sabayon
diff -Naur sabayon-2.20.1.orig/admin-tool/sabayon.desktop.in sabayon-2.20.1/admin-tool/sabayon.desktop.in
--- sabayon-2.20.1.orig/admin-tool/sabayon.desktop.in	2007-11-02 11:08:21.000000000 +0100
+++ sabayon-2.20.1/admin-tool/sabayon.desktop.in	2007-11-02 11:10:56.000000000 +0100
@@ -2,7 +2,7 @@
 Encoding=UTF-8
 _Name=User Profile Editor
 _Comment=Establish and Edit Profiles for Users
-Exec=gnomesu sabayon
+Exec=gksu sabayon
 Terminal=false
 Type=Application
 Icon=sabayon
