--- html5lib-python-1.0b10/setup.py	2016-07-15 04:36:07.000000000 +0300
+++ html5lib-python-1.0b10.new/setup.py	2016-08-13 20:24:26.469571694 +0300
@@ -8,7 +8,7 @@
 from setuptools import setup, find_packages, __version__ as setuptools_version
 from pkg_resources import parse_version
 
-if parse_version(setuptools_version) < parse_version("18.5"):
+if parse_version(setuptools_version) < parse_version("0.6"):
     print("html5lib requires setuptools version 18.5 or above; "
           "please upgrade before installing (you have %s)" % setuptools_version)
     sys.exit(1)
@@ -60,7 +60,7 @@
       install_requires=[
           'six',
           'webencodings',
-          'setuptools>=18.5'
+          'setuptools>=0.6'
       ],
       extras_require={
           # A empty extra that only has a conditional marker will be
--- html5lib-python-1.0b10/requirements-install.sh	2016-07-15 04:36:07.000000000 +0300
+++ html5lib-python-1.0b10.new/requirements-install.sh	2016-08-13 20:23:39.538558942 +0300
@@ -6,7 +6,7 @@
 fi
 
 # Make sure we're running setuptools >= 18.5
-pip install -U pip setuptools>=18.5
+pip install -U pip setuptools>=0.6
 
 pip install -U -r requirements-test.txt
 
