diff -urNad opal~/Makefile.in opal/Makefile.in
--- opal~/Makefile.in	2006-04-09 12:24:00.000000000 +0200
+++ opal/Makefile.in	2006-04-15 23:55:53.000000000 +0200
@@ -579,65 +579,6 @@
 endif # OPAL_SYSTEM_SPEEX
 
 
-# iLBC
-
-ILBC_DIR 	= $(OPAL_SRCDIR)/codec/iLBC
-
-VPATH_C		+= $(ILBC_DIR)
-VPATH_CXX	+= $(ILBC_DIR)
-
-HEADER_FILES	+= $(OPAL_INCDIR)/codec/ilbccodec.h
-SOURCES	+= $(OPAL_SRCDIR)/codec/ilbccodec.cxx
-
-HEADER_FILES	+= $(ILBC_DIR)/iLBC_define.h
-HEADER_FILES	+= $(ILBC_DIR)/iLBC_decode.h
-SOURCES		+= $(ILBC_DIR)/iLBC_decode.c
-HEADER_FILES	+= $(ILBC_DIR)/iLBC_encode.h
-SOURCES		+= $(ILBC_DIR)/iLBC_encode.c
-HEADER_FILES	+= $(ILBC_DIR)/FrameClassify.h
-SOURCES		+= $(ILBC_DIR)/FrameClassify.c
-HEADER_FILES	+= $(ILBC_DIR)/LPCdecode.h
-SOURCES		+= $(ILBC_DIR)/LPCdecode.c
-HEADER_FILES	+= $(ILBC_DIR)/LPCencode.h
-SOURCES		+= $(ILBC_DIR)/LPCencode.c
-HEADER_FILES	+= $(ILBC_DIR)/StateConstructW.h
-SOURCES		+= $(ILBC_DIR)/StateConstructW.c
-HEADER_FILES	+= $(ILBC_DIR)/StateSearchW.h
-SOURCES		+= $(ILBC_DIR)/StateSearchW.c
-HEADER_FILES	+= $(ILBC_DIR)/anaFilter.h
-SOURCES		+= $(ILBC_DIR)/anaFilter.c
-HEADER_FILES	+= $(ILBC_DIR)/constants.h
-SOURCES		+= $(ILBC_DIR)/constants.c
-HEADER_FILES	+= $(ILBC_DIR)/createCB.h
-SOURCES		+= $(ILBC_DIR)/createCB.c
-HEADER_FILES	+= $(ILBC_DIR)/doCPLC.h
-SOURCES		+= $(ILBC_DIR)/doCPLC.c
-HEADER_FILES	+= $(ILBC_DIR)/enhancer.h
-SOURCES		+= $(ILBC_DIR)/enhancer.c
-HEADER_FILES	+= $(ILBC_DIR)/filter.h
-SOURCES		+= $(ILBC_DIR)/filter.c
-HEADER_FILES	+= $(ILBC_DIR)/gainquant.h
-SOURCES		+= $(ILBC_DIR)/gainquant.c
-HEADER_FILES	+= $(ILBC_DIR)/getCBvec.h
-SOURCES		+= $(ILBC_DIR)/getCBvec.c
-HEADER_FILES	+= $(ILBC_DIR)/helpfun.h
-SOURCES		+= $(ILBC_DIR)/helpfun.c
-HEADER_FILES	+= $(ILBC_DIR)/hpInput.h
-SOURCES		+= $(ILBC_DIR)/hpInput.c
-HEADER_FILES	+= $(ILBC_DIR)/hpOutput.h
-SOURCES		+= $(ILBC_DIR)/hpOutput.c
-HEADER_FILES	+= $(ILBC_DIR)/iCBConstruct.h
-SOURCES		+= $(ILBC_DIR)/iCBConstruct.c
-HEADER_FILES	+= $(ILBC_DIR)/iCBSearch.h
-SOURCES		+= $(ILBC_DIR)/iCBSearch.c
-HEADER_FILES	+= $(ILBC_DIR)/lsf.h
-SOURCES		+= $(ILBC_DIR)/lsf.c
-HEADER_FILES	+= $(ILBC_DIR)/packing.h
-SOURCES		+= $(ILBC_DIR)/packing.c
-HEADER_FILES	+= $(ILBC_DIR)/syntFilter.h
-SOURCES		+= $(ILBC_DIR)/syntFilter.c
-
-
 VIC_DIR = $(OPAL_SRCDIR)/codec/vic
 
 SOURCES += $(OPAL_SRCDIR)/codec/h261codec.c \
diff -ru orig/opal_v2_2_1/include/codec/allcodecs.h opal_v2_2_1/include/codec/allcodecs.h
--- orig/opal_v2_2_1/include/codec/allcodecs.h  2006-01-01 20:19:32.000000000 +0100
+++ opal_v2_2_1/include/codec/allcodecs.h       2006-04-16 12:55:03.501545261 +0200
@@ -81,9 +81,6 @@
 #include <codec/speexcodec.h>
 OPAL_REGISTER_SPEEX();
 
-#include <codec/ilbccodec.h>
-OPAL_REGISTER_iLBC();
-
 #include <codec/vidcodec.h>
 OPAL_REGISTER_UNCOMPRESSED_VIDEO();
 
Only in orig/opal_v2_2_1/include/codec: ilbccodec.h
diff -ru orig/opal_v2_2_1/include/h323/h323.h opal_v2_2_1/include/h323/h323.h
--- orig/opal_v2_2_1/include/h323/h323.h        2004-03-11 07:54:26.000000000 +0100
+++ opal_v2_2_1/include/h323/h323.h     2006-04-16 13:16:28.106553131 +0200
@@ -485,7 +485,6 @@
 #ifndef NO_SPEEX
 #include <codec/speexcodec.h>
 #endif
-#include <codec/ilbccodec.h>
 #endif // NO_H323_AUDIO_CODECS
 
 #ifndef NO_OPAL_VIDEO
