--- libtrash-2.5/src/rename.c	2004-11-07 23:01:09.000000000 +0100
+++ libtrash-2.5/src/rename.c	2006-11-29 12:20:59.000000000 +0100
@@ -217,7 +217,7 @@
    
    /* We don't use GNU libc's canonicalize_file_name() directly for reasons explained in unlink.c: */
    
-   absolute_newpath = build_absolute_path(newpath);
+   absolute_newpath = build_absolute_path(newpath,1);
    
    if (!absolute_newpath)
      {
--- libtrash-2.5/src/unlink.c	2004-11-07 22:41:06.000000000 +0100
+++ libtrash-2.5/src/unlink.c	2006-11-29 12:20:59.000000000 +0100
@@ -153,7 +153,7 @@
     build_absolute_path(), which takes care of these complications for us:
     */
 
-   absolute_path = build_absolute_path(pathname);
+   absolute_path = build_absolute_path(pathname,1);
 
    if (!absolute_path)
      {
