diff --color -Naur edencommon-2025.02.17.00/CMakeLists.txt edencommon-2025.02.17.00.new/CMakeLists.txt
--- edencommon-2025.02.17.00/CMakeLists.txt	2025-01-30 18:35:12.000000000 +0100
+++ edencommon-2025.02.17.00.new/CMakeLists.txt	2025-02-21 17:02:21.996495520 +0100
@@ -50,7 +50,7 @@
   "${CMAKE_CURRENT_SOURCE_DIR}/build/fbcode_builder/CMake"
   ${CMAKE_MODULE_PATH})
 
-find_package(Glog MODULE REQUIRED)
+find_package(Glog CONFIG REQUIRED)
 include_directories(${GLOG_INCLUDE_DIR})
 
 find_package(Gflags REQUIRED)
