--- pccts.orig/sorcerer/lib/sorlist.c	1999-06-29 21:15:56.000000000 -0500
+++ pccts/sorcerer/lib/sorlist.c	2003-10-11 13:59:17.000000000 -0500
@@ -28,13 +28,7 @@
  */
 #include <stdio.h>
 #include <setjmp.h>
-
-#ifdef PCCTS_USE_STDARG
 #include <stdarg.h>
-#else
-#include <varargs.h>
-#endif
-
 #include "CASTBase.h"
 #include "sorlist.h"
 
--- pccts.orig/sorcerer/lib/sintstack.c	1999-06-30 09:08:06.000000000 -0500
+++ pccts/sorcerer/lib/sintstack.c	2003-10-11 14:09:43.000000000 -0500
@@ -28,13 +28,7 @@
  */
 #include <stdio.h>
 #include <setjmp.h>
-
-#ifdef PCCTS_USE_STDARG
 #include <stdarg.h>
-#else
-#include <varargs.h>
-#endif
-
 #include "CASTBase.h"
 #include "sintstack.h"
 
