--- pesign-113/Make.defaults	2019-05-10 20:53:51.000000000 +0200
+++ pesign-113.new/Make.defaults	2019-05-11 12:43:06.188959122 +0200
@@ -35,7 +35,7 @@
 
 SOFLAGS	= -shared
 clang_cflags =
-gcc_cflags = -Wmaybe-uninitialized -grecord-gcc-switches -flto
+gcc_cflags = -Wmaybe-uninitialized -grecord-gcc-switches
 cflags	= $(CFLAGS) $(ARCH3264) \
 	-Wall -Wextra -Wsign-compare -Wno-unused-result \
 	-Wno-unused-function -Wno-missing-field-initializers \
