diff -Naur kgpg-16.11.80/kgpginterface.cpp kgpg-16.11.80-p/kgpginterface.cpp
--- kgpg-16.11.80/kgpginterface.cpp	2016-11-16 09:14:50.000000000 +0100
+++ kgpg-16.11.80-p/kgpginterface.cpp	2016-11-25 12:04:25.290296114 +0100
@@ -277,7 +277,10 @@
 	}
 
 	if (p.exitCode() != 0) {
-		KMessageBox::detailedError(Q_NULLPTR, i18n("An error occurred while scanning your keyring"), log);
+		// FIXME FIXME
+		// well since og gnupg 2.1 a lot options moved to dirmngr ..
+		// we always have debug and verbose on it seems .. so mute that for now
+		//KMessageBox::detailedError(Q_NULLPTR, i18n("An error occurred while scanning your keyring"), log);
 		log.clear();
 	}
 
