diff -Naur linuxdoc-tools-0.9.71/rtf-fix/rtf2rtf.l linuxdoc-tools-0.9.71-rtf-fix/rtf-fix/rtf2rtf.l
--- linuxdoc-tools-0.9.71/rtf-fix/rtf2rtf.l	2015-10-20 13:22:17.000000000 +0200
+++ linuxdoc-tools-0.9.71-rtf-fix/rtf-fix/rtf2rtf.l	2015-12-15 14:38:03.975885045 +0100
@@ -71,7 +71,8 @@
 int current;
 int skipnewline = 0;
 
-extern char *yytext;
+/* no need for that with out flex version */
+/* extern char *yytext; */
 
 #if   YY_FLEX_MAJOR_VERSION > 2 \
  ||  (YY_FLEX_MAJOR_VERSION == 2 && YY_FLEX_MINOR_VERSION > 5) \
