diff -Naur aircrack-ng-0.6.orig/Makefile aircrack-ng-0.6/Makefile
--- aircrack-ng-0.6.orig/Makefile	2006-06-21 00:11:12.000000000 +0200
+++ aircrack-ng-0.6/Makefile	2006-07-10 00:09:29.000000000 +0200
@@ -1,5 +1,5 @@
 CC              = gcc
-CFLAGS          = -g -W -Wall -O3
+CFLAGS          ?= -g -W -Wall -O3
 OPTFLAGS        = -D_FILE_OFFSET_BITS=64
 
 iCC             = /opt/intel/cc/9.0/bin/icc
