diff -urN awstats-6.4/tools/awstats_buildstaticpages.pl awstats-6.4_fw/tools/awstats_buildstaticpages.pl
--- awstats-6.4/tools/awstats_buildstaticpages.pl	2005-02-20 17:11:06.000000000 +0100
+++ awstats-6.4_fw/tools/awstats_buildstaticpages.pl	2005-08-25 14:21:58.045625424 +0200
@@ -293,8 +293,8 @@
 # Check if AWSTATS prog is found
 my $AwstatsFound=0;
 if (-s "$Awstats") { $AwstatsFound=1; }
-elsif (-s "/usr/local/awstats/wwwroot/cgi-bin/awstats.pl") {
-	$Awstats="/usr/local/awstats/wwwroot/cgi-bin/awstats.pl";
+elsif (-s "/usr/lib/awstats/wwwroot/cgi-bin/awstats.pl") {
+	$Awstats="/usr/lib/awstats/wwwroot/cgi-bin/awstats.pl";
 	$AwstatsFound=1;
 }
 if (! $AwstatsFound) {
diff -urN awstats-6.4/tools/awstats_configure.pl awstats-6.4_fw/tools/awstats_configure.pl
--- awstats-6.4/tools/awstats_configure.pl	2005-01-15 23:42:58.000000000 +0100
+++ awstats-6.4_fw/tools/awstats_configure.pl	2005-08-25 14:21:58.045625424 +0200
@@ -29,10 +29,10 @@
 $AWSTATS_DIRDATA_PATH
 /;
 $AWSTATS_PATH='';
-$AWSTATS_ICON_PATH='/usr/local/awstats/wwwroot/icon';
-$AWSTATS_CSS_PATH='/usr/local/awstats/wwwroot/css';
-$AWSTATS_CLASSES_PATH='/usr/local/awstats/wwwroot/classes';
-$AWSTATS_CGI_PATH='/usr/local/awstats/wwwroot/cgi-bin';
+$AWSTATS_ICON_PATH='/usr/lib/awstats/wwwroot/icon';
+$AWSTATS_CSS_PATH='/usr/lib/awstats/wwwroot/css';
+$AWSTATS_CLASSES_PATH='/usr/lib/awstats/wwwroot/classes';
+$AWSTATS_CGI_PATH='/usr/lib/awstats/wwwroot/cgi-bin';
 $AWSTATS_MODEL_CONFIG='/etc/awstats/awstats.model.conf';		# Used only when configure ran on linux
 $AWSTATS_DIRDATA_PATH='/var/lib/awstats';						# Used only when configure ran on linux
 
@@ -60,7 +60,7 @@
 @WEBCONF=(
 'C:/Program Files/Apache Group/Apache2/conf/httpd.conf',
 'C:/Program Files/Apache Group/Apache/conf/httpd.conf',
-'/etc/httpd/httpd.conf',
+'/etc/httpd/conf/httpd.conf',
 '/usr/local/apache/conf/httpd.conf',
 '/usr/local/apache2/conf/httpd.conf'
 );
@@ -308,13 +308,13 @@
 	$AWSTATS_PATH=`pwd`; $AWSTATS_PATH =~ s/[\r\n]//;
 	$AWSTATS_PATH=~s/tools[\\\/]?$//;
 	$AWSTATS_PATH=~s/[\\\/]$//;
-	if ($AWSTATS_PATH ne '/usr/local/awstats') {
-		print "Warning: AWStats standard directory on Linux OS is '/usr/local/awstats'.\n";
+	if ($AWSTATS_PATH ne '/usr/lib/awstats') {
+		print "Warning: AWStats standard directory on Linux OS is '/usr/lib/awstats'.\n";
 		print "If you want to use standard directory, you should first move all content\n";
 		print "of AWStats distribution from current directory:\n";
 		print "$AWSTATS_PATH\n";
 		print "to standard directory:\n";
-		print "/usr/local/awstats\n";
+		print "/usr/lib/awstats\n";
 		print "And then, run configure.pl from this location.\n";
 		print "Do you want to continue setup from this NON standard directory [yN] ? ";
 		my $bidon='';
@@ -418,7 +418,7 @@
 		if ($bidon) { $ApachePath{"$bidon"}=1; }
 
 		print "\n".($bidon?"Now, enter":"Enter")." full config file path of your Web server.\n";
-		print "Example: /etc/httpd/httpd.conf\n";
+		print "Example: /etc/httpd/conf/httpd.conf\n";
 		print "Example: /usr/local/apache2/conf/httpd.conf\n";
 		print "Example: c:\\Program files\\apache group\\apache\\conf\\httpd.conf\n";
 		$bidon='';
diff -urN awstats-6.4/tools/awstats_updateall.pl awstats-6.4_fw/tools/awstats_updateall.pl
--- awstats-6.4/tools/awstats_updateall.pl	2004-06-07 14:27:26.000000000 +0200
+++ awstats-6.4_fw/tools/awstats_updateall.pl	2005-08-25 14:21:58.046625272 +0200
@@ -121,8 +121,8 @@
 	# Check if AWSTATS prog is found
 	my $AwstatsFound=0;
 	if (-s "$Awstats") { $AwstatsFound=1; }
-	elsif (-s "/usr/local/awstats/wwwroot/cgi-bin/awstats.pl") {
-		$Awstats="/usr/local/awstats/wwwroot/cgi-bin/awstats.pl";
+	elsif (-s "/usr/lib/awstats/wwwroot/cgi-bin/awstats.pl") {
+		$Awstats="/usr/lib/awstats/wwwroot/cgi-bin/awstats.pl";
 		$AwstatsFound=1;
 	}
 	if (! $AwstatsFound) {
diff -urN awstats-6.4/tools/httpd_conf awstats-6.4_fw/tools/httpd_conf
--- awstats-6.4/tools/httpd_conf	2004-01-13 14:25:56.000000000 +0100
+++ awstats-6.4_fw/tools/httpd_conf	1970-01-01 01:00:00.000000000 +0100
@@ -1,23 +0,0 @@
-#
-# Content of this file, with correct values, can be automatically added to
-# your Apache server by using the AWStats configure.pl tool.
-#
-
-#
-# Directives to add to your Apache conf file to allow use of AWStats as a CGI.
-# Note that path "/usr/local/awstats/" must reflect your AWStats install path.
-#
-Alias /awstatsclasses "/usr/local/awstats/wwwroot/classes/"
-Alias /awstatscss "/usr/local/awstats/wwwroot/css/"
-Alias /awstatsicons "/usr/local/awstats/wwwroot/icon/"
-ScriptAlias /awstats/ "/usr/local/awstats/wwwroot/cgi-bin/"
-
-#
-# This is to permit URL access to scripts/files in AWStats directory.
-#
-<Directory "/usr/local/awstats/wwwroot">
-    Options None
-    AllowOverride None
-    Order allow,deny
-    Allow from all
-</Directory>
diff -urN awstats-6.4/wwwroot/cgi-bin/awstats.model.conf awstats-6.4_fw/wwwroot/cgi-bin/awstats.model.conf
--- awstats-6.4/wwwroot/cgi-bin/awstats.model.conf	2005-02-24 22:08:40.000000000 +0100
+++ awstats-6.4_fw/wwwroot/cgi-bin/awstats.model.conf	2005-08-25 14:21:58.047625120 +0200
@@ -48,7 +48,7 @@
 # If there is several log files from load balancing servers :
 # Example: "/pathtotools/logresolvemerge.pl *.log |"
 #
-LogFile="/var/log/httpd/mylog.log"
+LogFile="/var/log/httpd/access_log"
 
 
 # Enter the log file type you want to analyze.
@@ -145,7 +145,7 @@
 # Example: "ftp.domain.com"
 # Example: "domain.com"
 #
-SiteDomain=""
+SiteDomain="localhost"
 
 
 # Enter here all other possible domain names, addresses or virtual host
@@ -195,7 +195,7 @@
 # Example: "C:/awstats_data_dir"
 # Default: "."          (means same directory as awstats.pl)
 #
-DirData="."
+DirData="/var/lib/awstats"
 
 
 # Relative or absolute web URL of your awstats cgi-bin directory.
diff -urN awstats-6.4/wwwroot/cgi-bin/awstats.pl awstats-6.4_fw/wwwroot/cgi-bin/awstats.pl
--- awstats-6.4/wwwroot/cgi-bin/awstats.pl	2005-02-24 22:02:23.000000000 +0100
+++ awstats-6.4_fw/wwwroot/cgi-bin/awstats.pl	2005-08-25 14:21:58.054624056 +0200
@@ -1835,7 +1835,7 @@
 	# Windows and standard package:        		"$DIR/plugins" (plugins in same dir than awstats.pl)
 	# Redhat :                                  "/usr/local/awstats/wwwroot/cgi-bin/plugins"
 	# Debian package :                    		"/usr/share/awstats/plugins"
-	my @PossiblePluginsDir=("$DIR/plugins","/usr/local/awstats/wwwroot/cgi-bin/plugins","/usr/share/awstats/plugins");
+	my @PossiblePluginsDir=("$DIR/plugins","/usr/lib/awstats/wwwroot/cgi-bin/plugins","/usr/share/awstats/plugins");
  	my %DirAddedInINC=();
 
 	#Removed for security reason
