diff -Naur awstats-6.7-orig/tools/awstats_buildstaticpages.pl awstats-6.7/tools/awstats_buildstaticpages.pl
--- awstats-6.7-orig/tools/awstats_buildstaticpages.pl	2006-02-08 22:26:14.000000000 +0100
+++ awstats-6.7/tools/awstats_buildstaticpages.pl	2007-07-07 11:53:09.000000000 +0200
@@ -127,13 +127,13 @@
 	# Check config file in common possible directories :
 	# Windows :                   				"$DIR" (same dir than awstats.pl)
 	# Standard, Mandrake and Debian package :	"/etc/awstats"
-	# Other possible directories :				"/usr/local/etc/awstats", "/etc"
+	# Other possible directories :				"/usr/lib/etc/awstats", "/etc"
 	# FHS standard, Suse package : 				"/etc/opt/awstats"
 	my $configdir=shift;
 	my @PossibleConfigDir=();
 
 	if ($configdir) { @PossibleConfigDir=("$configdir"); }
-	else { @PossibleConfigDir=("$AwstatsDir","$DIR","/etc/awstats","/usr/local/etc/awstats","/etc","/etc/opt/awstats"); }
+	else { @PossibleConfigDir=("$AwstatsDir","$DIR","/etc/awstats","/usr/lib/etc/awstats","/etc","/etc/opt/awstats"); }
 
 	# Open config file
 	$FileConfig=$FileSuffix='';
@@ -293,8 +293,8 @@
 # Check if AWSTATS prog is found
 my $AwstatsFound=0;
 if (-s "$Awstats") { $AwstatsFound=1; }
-elsif (-s "/usr/local/awstats/wwwroot/cgi-bin/awstats.pl") {
-	$Awstats="/usr/local/awstats/wwwroot/cgi-bin/awstats.pl";
+elsif (-s "/usr/lib/awstats/wwwroot/cgi-bin/awstats.pl") {
+	$Awstats="/usr/lib/awstats/wwwroot/cgi-bin/awstats.pl";
 	$AwstatsFound=1;
 }
 if (! $AwstatsFound) {
diff -Naur awstats-6.7-orig/tools/awstats_configure.pl awstats-6.7/tools/awstats_configure.pl
--- awstats-6.7-orig/tools/awstats_configure.pl	2007-04-02 20:30:52.000000000 +0200
+++ awstats-6.7/tools/awstats_configure.pl	2007-07-07 11:45:16.000000000 +0200
@@ -29,10 +29,10 @@
 $AWSTATS_DIRDATA_PATH
 /;
 $AWSTATS_PATH='';
-$AWSTATS_ICON_PATH='/usr/local/awstats/wwwroot/icon';
-$AWSTATS_CSS_PATH='/usr/local/awstats/wwwroot/css';
-$AWSTATS_CLASSES_PATH='/usr/local/awstats/wwwroot/classes';
-$AWSTATS_CGI_PATH='/usr/local/awstats/wwwroot/cgi-bin';
+$AWSTATS_ICON_PATH='/usr/lib/awstats/wwwroot/icon';
+$AWSTATS_CSS_PATH='/usr/lib/awstats/wwwroot/css';
+$AWSTATS_CLASSES_PATH='/usr/lib/awstats/wwwroot/classes';
+$AWSTATS_CGI_PATH='/usr/lib/awstats/wwwroot/cgi-bin';
 $AWSTATS_MODEL_CONFIG='/etc/awstats/awstats.model.conf';		# Used only when configure ran on linux
 $AWSTATS_DIRDATA_PATH='/var/lib/awstats';						# Used only when configure ran on linux
 
@@ -61,7 +61,7 @@
 'C:/Program Files/Apache Group/Apache2/conf/httpd.conf',
 'C:/Program Files/Apache Group/Apache/conf/httpd.conf',
 '/Applications/MAMP/conf/apache/httpd.conf',
-'/etc/httpd/httpd.conf',
+'/etc/httpd/conf/httpd.conf',
 '/usr/local/apache/conf/httpd.conf',
 '/usr/local/apache2/conf/httpd.conf',
 );
@@ -309,13 +309,13 @@
 	$AWSTATS_PATH=`pwd`; $AWSTATS_PATH =~ s/[\r\n]//;
 	$AWSTATS_PATH=~s/tools[\\\/]?$//;
 	$AWSTATS_PATH=~s/[\\\/]$//;
-	if ($AWSTATS_PATH ne '/usr/local/awstats') {
-		print "Warning: AWStats standard directory on Linux OS is '/usr/local/awstats'.\n";
+	if ($AWSTATS_PATH ne '/usr/lib/awstats') {
+		print "Warning: AWStats standard directory on Linux OS is '/usr/lib/awstats'.\n";
 		print "If you want to use standard directory, you should first move all content\n";
 		print "of AWStats distribution from current directory:\n";
 		print "$AWSTATS_PATH\n";
 		print "to standard directory:\n";
-		print "/usr/local/awstats\n";
+		print "/usr/lib/awstats\n";
 		print "And then, run configure.pl from this location.\n";
 		print "Do you want to continue setup from this NON standard directory [yN] ? ";
 		my $bidon='';
@@ -419,7 +419,7 @@
 		if ($bidon) { $ApachePath{"$bidon"}=1; }
 
 		print "\n".($bidon?"Now, enter":"Enter")." full config file path of your Web server.\n";
-		print "Example: /etc/httpd/httpd.conf\n";
+		print "Example: /etc/httpd/conf/httpd.conf\n";
 		print "Example: /usr/local/apache2/conf/httpd.conf\n";
 		print "Example: c:\\Program files\\apache group\\apache\\conf\\httpd.conf\n";
 		$bidon='';
diff -Naur awstats-6.7-orig/tools/awstats_updateall.pl awstats-6.7/tools/awstats_updateall.pl
--- awstats-6.7-orig/tools/awstats_updateall.pl	2006-07-24 00:57:48.000000000 +0200
+++ awstats-6.7/tools/awstats_updateall.pl	2007-07-07 11:45:42.000000000 +0200
@@ -130,8 +130,8 @@
 	# Check if AWSTATS prog is found
 	my $AwstatsFound=0;
 	if (-s "$Awstats") { $AwstatsFound=1; }
-	elsif (-s "/usr/local/awstats/wwwroot/cgi-bin/awstats.pl") {
-		$Awstats="/usr/local/awstats/wwwroot/cgi-bin/awstats.pl";
+	elsif (-s "/usr/lib/awstats/wwwroot/cgi-bin/awstats.pl") {
+		$Awstats="/usr/lib/awstats/wwwroot/cgi-bin/awstats.pl";
 		$AwstatsFound=1;
 	}
 	if (! $AwstatsFound) {
diff -Naur awstats-6.7-orig/tools/httpd_conf awstats-6.7/tools/httpd_conf
--- awstats-6.7-orig/tools/httpd_conf	2006-07-23 00:08:08.000000000 +0200
+++ awstats-6.7/tools/httpd_conf	1970-01-01 01:00:00.000000000 +0100
@@ -1,23 +0,0 @@
-#
-# Content of this file, with correct values, can be automatically added to
-# your Apache server by using the AWStats configure.pl tool.
-#
-
-#
-# Directives to add to your Apache conf file to allow use of AWStats as a CGI.
-# Note that path "/usr/local/awstats/" must reflect your AWStats install path.
-#
-Alias /awstatsclasses "/usr/local/awstats/wwwroot/classes/"
-Alias /awstatscss "/usr/local/awstats/wwwroot/css/"
-Alias /awstatsicons "/usr/local/awstats/wwwroot/icon/"
-ScriptAlias /awstats/ "/usr/local/awstats/wwwroot/cgi-bin/"
-
-#
-# This is to permit URL access to scripts/files in AWStats directory.
-#
-<Directory "/usr/local/awstats/wwwroot">
-    Options None
-    AllowOverride None
-    Order allow,deny
-    Allow from all
-</Directory>
diff -Naur awstats-6.7-orig/wwwroot/cgi-bin/awstats.model.conf awstats-6.7/wwwroot/cgi-bin/awstats.model.conf
--- awstats-6.7-orig/wwwroot/cgi-bin/awstats.model.conf	2007-04-02 20:34:36.000000000 +0200
+++ awstats-6.7/wwwroot/cgi-bin/awstats.model.conf	2007-07-07 11:50:05.000000000 +0200
@@ -48,7 +48,7 @@
 # If there are several log files from load balancing servers :
 # Example: "/pathtotools/logresolvemerge.pl *.log |"
 #
-LogFile="/var/log/httpd/mylog.log"
+LogFile="/var/log/httpd/access_log"
 
 
 # Enter the log file type you want to analyze.
@@ -150,7 +150,7 @@
 # Example: "ftp.domain.com"
 # Example: "domain.com"
 #
-SiteDomain=""
+SiteDomain="localhost"
 
 
 # Enter here all other possible domain names, addresses or virtual host
@@ -200,7 +200,7 @@
 # Example: "C:/awstats_data_dir"
 # Default: "."          (means same directory as awstats.pl)
 #
-DirData="."
+DirData="/var/lib/awstats"
 
 
 # Relative or absolute web URL of your awstats cgi-bin directory.
diff -Naur awstats-6.7-orig/wwwroot/cgi-bin/awstats.pl awstats-6.7/wwwroot/cgi-bin/awstats.pl
--- awstats-6.7-orig/wwwroot/cgi-bin/awstats.pl	2007-03-06 18:35:20.000000000 +0100
+++ awstats-6.7/wwwroot/cgi-bin/awstats.pl	2007-07-07 11:52:14.000000000 +0200
@@ -1136,7 +1136,7 @@
 	# Check config file in common possible directories :
 	# Windows :                   				"$DIR" (same dir than awstats.pl)
 	# Standard, Mandrake and Debian package :	"/etc/awstats"
-	# Other possible directories :				"/usr/local/etc/awstats", "/etc"
+	# Other possible directories :				"/usr/lib/etc/awstats", "/etc"
 	# FHS standard, Suse package : 				"/etc/opt/awstats"
 	my $configdir=shift;
 	my @PossibleConfigDir=();
@@ -1153,7 +1153,7 @@
 			@PossibleConfigDir=("$configdir");
 		#}
 	}
-	else { @PossibleConfigDir=("$DIR","/etc/awstats","/usr/local/etc/awstats","/etc","/etc/opt/awstats"); }
+	else { @PossibleConfigDir=("$DIR","/etc/awstats","/usr/lib/etc/awstats","/etc","/etc/opt/awstats"); }
 
 	# Open config file
 	$FileConfig=$FileSuffix='';
@@ -1938,7 +1938,7 @@
 	# Windows and standard package:        		"$DIR/plugins" (plugins in same dir than awstats.pl)
 	# Redhat :                                  "/usr/local/awstats/wwwroot/cgi-bin/plugins"
 	# Debian package :                    		"/usr/share/awstats/plugins"
-	my @PossiblePluginsDir=("$DIR/plugins","/usr/local/awstats/wwwroot/cgi-bin/plugins","/usr/share/awstats/plugins");
+	my @PossiblePluginsDir=("$DIR/plugins","/usr/lib/awstats/wwwroot/cgi-bin/plugins","/usr/share/awstats/plugins");
  	my %DirAddedInINC=();
 
 	#Removed for security reason
