diff -Naur shadow-4.0.4.1.orig/libmisc/xmalloc.c shadow-4.0.4.1/libmisc/xmalloc.c
--- shadow-4.0.4.1.orig/libmisc/xmalloc.c	2003-04-22 12:59:22.000000000 +0200
+++ shadow-4.0.4.1/libmisc/xmalloc.c	2004-04-28 23:09:08.000000000 +0200
@@ -13,7 +13,6 @@
 RCSID ("$Id: xmalloc.c,v 1.4 2003/04/22 10:59:22 kloczek Exp $")
 #include <stdio.h>
 #include "defines.h"
-extern char *malloc ();
 
 char *xmalloc (size_t size)
 {
