diff -Naur unison-2.13.16.orig/Makefile unison-2.13.16/Makefile
--- unison-2.13.16.orig/Makefile	2007-05-24 17:25:49.000000000 +0200
+++ unison-2.13.16/Makefile	2007-05-24 17:26:23.000000000 +0200
@@ -43,7 +43,7 @@
 
 ########################################################################
 ## Miscellaneous developer-only switches
-DEBUGGING=true
+DEBUGGING=false
 PROFILING=false
 STATIC=false
 
@@ -68,7 +68,7 @@
 ######################################################################
 # Installation
 
-INSTALLDIR = $(HOME)/bin/
+INSTALLDIR = $(DESTDIR)$(HOME)/bin/
 
 # This has two names because on OSX the file INSTALL shadows the target 'install'!
 install: doinstall
