diff --git a/ps/output.c b/ps/output.c
index 87bf9de..b3cf257 100644
--- a/ps/output.c
+++ b/ps/output.c
@@ -204,6 +204,39 @@ int pr_nop(char *restrict const outbuf, const proc_t *restrict const pp){
   return snprintf(outbuf, COLWID, "%c", '-');
 }
 
+static int pr_cgroup(char *restrict const outbuf, const proc_t *restrict const pp){
+  char filename[48];
+  FILE *fd;
+  int counter = 0;
+  int c;
+  int is_cgroup = 0;
+
+  outbuf[0]='\0';
+  snprintf(filename, sizeof filename, "/proc/%d/cgroup", pp->tgid);
+  fd = fopen(filename, "r");
+  if (likely(fd == NULL)) goto fail;
+  while (( (c = fgetc(fd)) != EOF) && (counter<665)) {
+    if (is_cgroup == 0) {
+      if (c == ':') {
+        is_cgroup = 1;
+        if (counter>0)
+          outbuf[counter++]=';';
+      }
+    }else
+      if ((c == '\n') || (c == '\0'))
+        is_cgroup = 0;
+      else
+        outbuf[counter++]=c;
+  }
+  outbuf[counter]='\0';
+  close(fd);
+  if (counter>0) 
+    return counter;
+fail:
+  outbuf[0] = '-';
+  outbuf[1] = '\0';
+  return 1;
+}
 
 /********* Unix 98 ************/
 
@@ -1293,6 +1326,7 @@ static const format_struct format_array[] = {
 {"bsdtime",   "TIME",    pr_bsdtime,  sr_nop,     6,   0,    LNX, ET|RIGHT},
 {"c",         "C",       pr_c,        sr_pcpu,    2,   0,    SUN, ET|RIGHT},
 {"caught",    "CAUGHT",  pr_sigcatch, sr_nop,     9,   0,    BSD, TO|SIGNAL}, /*sigcatch*/
+{"cgroup",    "CGROUP",  pr_cgroup,   sr_nop,    35,   0,    LNX, PO|LEFT},  /* cgroups*/
 {"class",     "CLS",     pr_class,    sr_sched,   3,   0,    XXX, TO|LEFT},
 {"cls",       "CLS",     pr_class,    sr_sched,   3,   0,    HPU, TO|RIGHT}, /*says HPUX or RT*/
 {"cmaj_flt",  "-",       pr_nop,      sr_cmaj_flt, 1,  0,    LNX, AN|RIGHT},
diff --git a/ps/ps.1 b/ps/ps.1
index 64953d5..0a5f6f3 100644
--- a/ps/ps.1
+++ b/ps/ps.1
@@ -904,6 +904,10 @@ width of the field, a 32 or 64 bits mask in hexadecimal format is
 displayed.  (alias\ \fBsig_catch\fR,\ \fBsigcatch\fR).
 T}
 
+cgroup	CGROUP	T{
+display control groups to which the process belonges.
+t}
+
 class	CLS	T{
 scheduling class of the process.  (alias\ \fBpolicy\fR,\ \fBcls\fR).
 Field's possible values are:
