--- mate-panel/nothing.cP.orig	2014-02-16 17:28:37.529414744 +0100
+++ mate-panel/nothing.cP	2014-02-16 17:29:00.590218967 +0100
@@ -9,7 +9,7 @@
 #if GTK_CHECK_VERSION (3, 0, 0)
 #include <gdk/gdkkeysyms-compat.h>
 #endif
-#include <canberra-gtk.h>
+#include <canberra-gtk2.h>
 
 #ifdef CA_CHECK_VERSION
 #if CA_CHECK_VERSION(0, 13)
@@ -1571,4 +1571,4 @@
 
 	return FALSE;
 }
-#endif
\ Pas de fin de ligne à la fin du fichier
+#endif
