diff -Naur frobby_v0.8.2/src/test/TestSuite.cpp frobby_v0.8.2/src/test/TestSuite.cpp
--- frobby_v0.8.2/src/test/TestSuite.cpp	2009-07-07 05:40:18.000000000 +0200
+++ frobby_v0.8.2/src/test/TestSuite.cpp	2009-11-09 23:09:15.093908794 +0100
@@ -14,6 +14,7 @@
    You should have received a copy of the GNU General Public License
    along with this program.  If not, see http://www.gnu.org/licenses/.
 */
+#include <algorithm>
 #include "stdinc.h"
 #include "TestSuite.h"
 
