diff -Nru libbtctl-0.6.0.orig/src/Makefile.in libbtctl-0.6.0/src/Makefile.in
--- libbtctl-0.6.0.orig/src/Makefile.in	2005-11-22 00:02:52.000000000 +0200
+++ libbtctl-0.6.0/src/Makefile.in	2006-04-22 12:11:12.586318250 +0300
@@ -239,7 +239,7 @@
 btctl.defs btctl.override
 
 lib_LTLIBRARIES = libbtctl.la
-@HAVE_PYTHON_TRUE@pydir = $(PYTHON_PREFIX)/lib/python$(PYTHON_VERSION)/site-packages/
+@HAVE_PYTHON_TRUE@pydir = $(libdir)/python$(PYTHON_VERSION)/site-packages/
 @HAVE_PYTHON_FALSE@pydir = 
 @HAVE_PYTHON_TRUE@py_LTLIBRARIES = btctl.la
 @HAVE_PYTHON_FALSE@py_LTLIBRARIES = 
