diff -Nur desktop-effects-0.7.0/desktop-effects.c desktop-effects-0.7.0.new/desktop-effects.c
--- desktop-effects-0.7.0/desktop-effects.c	2007-01-16 22:17:26.000000000 +0100
+++ desktop-effects-0.7.0.new/desktop-effects.c	2007-01-21 00:50:59.000000000 +0100
@@ -32,7 +32,7 @@
 #include <glib/gi18n.h>
 
 #define PLUGIN_LIST_KEY		"/apps/compiz/general/allscreens/options/active_plugins"
-#define WINDOW_MANAGER_KEY	"/apps/gnome-session/rh/window_manager"
+#define WINDOW_MANAGER_KEY	"/desktop/gnome/applications/window_manager/default"
 
 typedef struct App App;
 
@@ -91,7 +91,7 @@
 	return METACITY;
     }
     
-    if (str && strcmp (str, "compiz") == 0)
+    if (str && strcmp (str, "/usr/bin/compiz") == 0)
     {
 	return COMPIZ;
     }
@@ -283,7 +283,7 @@
 static void
 apply_settings (App *app, Settings *settings)
 {
-    const char *str = settings->enabled? "compiz" : "metacity";
+    const char *str = settings->enabled? "/usr/bin/compiz" : "/usr/bin/metacity";
     char *session_file;
     
     gconf_client_set_string (app->gconf,
