diff -urN desktop-effects-0.6.163.orig/desktop-effects.c desktop-effects-0.6.163/desktop-effects.c
--- desktop-effects-0.6.163.orig/desktop-effects.c	2006-12-03 20:25:04.000000000 +0000
+++ desktop-effects-0.6.163/desktop-effects.c	2006-12-03 20:25:16.000000000 +0000
@@ -19,7 +19,7 @@
 #include <glib/gi18n.h>
 
 #define PLUGIN_LIST_KEY		"/apps/compiz/general/allscreens/options/active_plugins"
-#define WINDOW_MANAGER_KEY	"/apps/gnome-session/ubuntu/window_manager"
+#define WINDOW_MANAGER_KEY	"/apps/gnome-session/frugalware/window_manager"
 
 typedef struct App App;
 
