diff -Naur vbetool-0.7.orig/x86-common.c vbetool-0.7/x86-common.c
--- vbetool-0.7.orig/x86-common.c	2007-05-13 10:55:52.000000000 +0200
+++ vbetool-0.7/x86-common.c	2007-05-13 10:56:02.000000000 +0200
@@ -190,14 +190,14 @@
 static inline unsigned int
 get_int_seg(int i)
 {
-	return *(unsigned short *)(i * 4 + 2);
+	return (unsigned short)(i * 4 + 2);
 }
 
 
 static inline unsigned int
 get_int_off(int i)
 {
-	return *(unsigned short *)(i * 4);
+	return (unsigned short)(i * 4);
 }
 
 int LRMI_common_init(void)
