diff --color -Naur gsl-2.8/Makefile.am gsl-2.8.new/Makefile.am
--- gsl-2.8/Makefile.am	2018-11-27 22:51:13.000000000 +0100
+++ gsl-2.8.new/Makefile.am	2025-02-20 18:08:44.203062348 +0100
@@ -29,10 +29,10 @@
 bin_PROGRAMS = gsl-randist gsl-histogram
 
 gsl_randist_SOURCES = gsl-randist.c
-gsl_randist_LDADD = libgsl.la cblas/libgslcblas.la
+gsl_randist_LDADD = libgsl.la 
 
 gsl_histogram_SOURCES = gsl-histogram.c
-gsl_histogram_LDADD = libgsl.la cblas/libgslcblas.la
+gsl_histogram_LDADD = libgsl.la 
 
 check_SCRIPTS = test_gsl_histogram.sh pkgconfig.test
 TESTS = test_gsl_histogram.sh pkgconfig.test
diff --color -Naur gsl-2.8/Makefile.in gsl-2.8.new/Makefile.in
--- gsl-2.8/Makefile.in	2024-05-25 15:26:32.000000000 +0200
+++ gsl-2.8.new/Makefile.in	2025-02-20 18:08:44.203405563 +0100
@@ -154,10 +154,10 @@
 	$(libgsl_la_LDFLAGS) $(LDFLAGS) -o $@
 am_gsl_histogram_OBJECTS = gsl-histogram.$(OBJEXT)
 gsl_histogram_OBJECTS = $(am_gsl_histogram_OBJECTS)
-gsl_histogram_DEPENDENCIES = libgsl.la cblas/libgslcblas.la
+gsl_histogram_DEPENDENCIES = libgsl.la 
 am_gsl_randist_OBJECTS = gsl-randist.$(OBJEXT)
 gsl_randist_OBJECTS = $(am_gsl_randist_OBJECTS)
-gsl_randist_DEPENDENCIES = libgsl.la cblas/libgslcblas.la
+gsl_randist_DEPENDENCIES = libgsl.la 
 SCRIPTS = $(bin_SCRIPTS)
 AM_V_P = $(am__v_P_@AM_V@)
 am__v_P_ = $(am__v_P_@AM_DEFAULT_V@)
@@ -659,9 +659,9 @@
 m4datadir = $(datadir)/aclocal
 m4data_DATA = gsl.m4
 gsl_randist_SOURCES = gsl-randist.c
-gsl_randist_LDADD = libgsl.la cblas/libgslcblas.la
+gsl_randist_LDADD = libgsl.la 
 gsl_histogram_SOURCES = gsl-histogram.c
-gsl_histogram_LDADD = libgsl.la cblas/libgslcblas.la
+gsl_histogram_LDADD = libgsl.la 
 check_SCRIPTS = test_gsl_histogram.sh pkgconfig.test
 TESTS = test_gsl_histogram.sh pkgconfig.test
 
