diff -Naurp kdbus-0.8.6/src/dbusobject.cpp kdbus-0.8.6-p/src/dbusobject.cpp
--- kdbus-0.8.6/src/dbusobject.cpp	2006-02-18 06:23:48.000000000 +0100
+++ kdbus-0.8.6-p/src/dbusobject.cpp	2006-11-17 23:25:27.000000000 +0100
@@ -42,6 +42,7 @@ public:
 
     QString discoverPath( QDomElement const & elem );
 
+    QDomDocument doc;
     QDBusProxy * proxy;
     DBusObject * p;
 };
@@ -88,19 +89,19 @@ DBusObject::DBusObject(
 
     // Now iterate through self, enumerating interfaces, properties, and
     // other objects.
-    myElem = myElem.firstChild().toElement();
+    QDomElement childElement = myElem.firstChild().toElement();
 
     try {
         do {
-            if ( myElem.tagName() == "node" )
-                new DBusObject( this, myElem, d->proxy );
-            else if ( myElem.tagName() == "interface" )
-                new DBusInterface( this, myElem );
+            if ( childElement.tagName() == "node" )
+                new DBusObject( this, childElement, d->proxy );
+            else if ( childElement.tagName() == "interface" )
+                new DBusInterface( this, childElement );
             else
                 kdWarning() << "Unhandled node in introspect data: "
-                            << myElem.tagName() << endl;
-            myElem = myElem.nextSibling().toElement();
-        } while ( !myElem.isNull() );
+                            << childElement.tagName() << endl;
+            childElement = childElement.nextSibling().toElement();
+        } while ( !childElement.isNull() );
     }
     catch ( std::runtime_error const & e ) {
         m_ok = false;
@@ -128,7 +129,6 @@ throw ( QDBusSendError, QDBusXmlError ) 
             .arg( proxy->lastError().message() )
         );
 
-    QDomDocument doc;
     if ( !doc.setContent( reply[0].toString() ) )
         throw QDBusXmlError(
             i18n(
