diff -Nur jokosher-0.1/jokosher.desktop jokosher-0.1.new/jokosher.desktop
--- jokosher-0.1/jokosher.desktop	2006-07-19 01:28:39.000000000 +0200
+++ jokosher-0.1.new/jokosher.desktop	2006-08-01 23:27:20.000000000 +0200
@@ -3,9 +3,10 @@
 Name=Jokosher
 GenericName=Jokosher audio editing
 Comment=Multi-track non-linear audio editing
-Exec=Jokosher.py
-Icon=jokosher.png
+Exec=jokosher
+Icon=/usr/share/jokosher/jokosher.png
 Terminal=false
 Type=Application
 Categories=Application;AudioVideo;
-Version=0.1
\ Kein Zeilenumbruch am Dateiende.
+Version=0.1
+MimeType=application/x-jokosher
