diff -Naurp QtCurve-KDE3-0.46/admin/cvs.sh QtCurve-KDE3-0.46-p/admin/cvs.sh
--- QtCurve-KDE3-0.46/admin/cvs.sh	2006-11-05 18:44:45.000000000 +0100
+++ QtCurve-KDE3-0.46-p/admin/cvs.sh	2006-11-29 16:43:15.000000000 +0100
@@ -70,7 +70,7 @@ case $AUTOMAKE_STRING in
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9*) : ;;
+  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* |  automake*1.10* ) : ;;
   "" )
     echo "*** AUTOMAKE NOT FOUND!."
     echo "*** KDE requires automake $required_automake_version"
