diff -urN compiz-0.3.2.orig/plugins/compiz.schemas.in compiz-0.3.2/plugins/compiz.schemas.in
--- compiz-0.3.2.orig/plugins/compiz.schemas.in	2006-11-18 20:04:20.000000000 +0000
+++ compiz-0.3.2/plugins/compiz.schemas.in	2006-11-18 20:04:54.000000000 +0000
@@ -8,7 +8,7 @@
             <owner>compiz</owner>
             <type>list</type>
             <list_type>string</list_type>
-            <default>[gconf,@DEFAULT_PLUGINS@]</default>
+            <default>[gconf,png,svg,decoration,move,place,resize,screenshot,water,wobbly,fade,minimize,cube,switcher,scale,rotate]</default>
             <locale name="C">
                 <short>Active Plugins</short>
                 <long>List of currently active plugins</long>
