diff -Naur KTrafficAnalyzer-0.5.4.1/lib/TrafficWindow.cpp KTrafficAnalyzer-0.5.4.1-p/lib/TrafficWindow.cpp
--- KTrafficAnalyzer-0.5.4.1/lib/TrafficWindow.cpp	2010-08-02 09:20:20.000000000 +0200
+++ KTrafficAnalyzer-0.5.4.1-p/lib/TrafficWindow.cpp	2016-01-28 19:45:06.720268216 +0100
@@ -77,7 +77,7 @@
     dialog->addPage(statisticsWidget, i18n("Statistics"), "kchart");
 
     //Create correct stats skeleton
-    /*delete KTAStats::self();
+    delete KTAStats::self();
     KTAStats::self(this->trafficInterface->getInterfaceName().prepend("ktastats"));
     //Create stats dialog
     stats = new KConfigDialog(this, "KTAStats", KTAStats::self());
@@ -92,7 +92,7 @@
     stats->addPage(monthlyStatsWidget, i18n("Monthly"), "accessories-calculator");
     //Update traffic stats in real-time
     QObject::connect(this, SIGNAL(updateStatsDialog(int)), dailyStatsWidget, SLOT(updateStatistics(int)));
-    QObject::connect(this, SIGNAL(updateStatsDialog(int)), monthlyStatsWidget, SLOT(updateStatistics(int)));*/
+    QObject::connect(this, SIGNAL(updateStatsDialog(int)), monthlyStatsWidget, SLOT(updateStatistics(int)));
 
     //Create display for realtime traffic graph
     this->graphDisplay = new GraphDisplay(this);
@@ -132,7 +132,7 @@
 
     //Add context menu entries at the SysTray
     this->sysTray->contextMenu()->addAction(i18n("Settings"), dialog, SLOT(show()));
-    this->sysTray->contextMenu()->addAction(i18n("Stats"), stats, SLOT(show()));
+    //this->sysTray->contextMenu()->addAction(i18n("Stats"), stats, SLOT(show()));
 
     //Restore size?
     if (KTASettings::rememberSize()) {
@@ -550,10 +550,11 @@
     QObject::connect(this, SIGNAL(updateStatsDialog(int)), dailyStatsWidget, SLOT(updateStatistics(int)));
     QObject::connect(this, SIGNAL(updateStatsDialog(int)), monthlyStatsWidget, SLOT(updateStatistics(int)));
 
-    //Update menu-bar
-    this->menuBar()->removeAction(this->statsEntryAction);
+    //Update menu-bar & systray
+    this->sysTray->contextMenu()->removeAction(this->statsEntryAction);
     this->statsEntryAction = this->menuBar()->addAction(i18n("Stats"), stats, SLOT(show()));
     this->menuBar()->insertAction(this->exitEntryAction, this->statsEntryAction);
+    this->sysTray->contextMenu()->addAction(this->statsEntryAction);
 
     //Reset memory
     this->memoryUpload->reset();
