--- servant-server-0.20.2/servant-server.cabal	2001-09-09 03:46:40.000000000 +0200
+++ servant-server-0.20.2.new/servant-server.cabal	2024-12-28 15:00:52.222559799 +0100
@@ -116,7 +116,7 @@
   -- Bundled with GHC: Lower bound to not force re-installs
   -- text and mtl are bundled starting with GHC-8.4
   build-depends:
-    , base          >=4.14     && <4.21
+    , base          >=4.14     && <4.22
     , bytestring    >=0.10.8.1 && <0.13
     , constraints   >=0.2      && <0.15
     , containers    >=0.5.7.1  && <0.8
