--- servant-server-0.20/servant-server.cabal	2001-09-09 03:46:40.000000000 +0200
+++ servant-server-0.20.new/servant-server.cabal	2023-07-24 09:33:20.887060897 +0200
@@ -73,7 +73,7 @@
   -- strict dependency as we re-export 'servant' things.
   build-depends:
       servant             >= 0.20     && < 0.21
-    , http-api-data       >= 0.4.1    && < 0.6
+    , http-api-data       >= 0.4.1    && < 0.7
 
   -- Other dependencies: Lower bound around what is in the latest Stackage LTS.
   -- Here can be exceptions if we really need features from the newer versions.
