diff -Naur old-locale-1.0.0.7/old-locale.cabal old-locale-1.0.0.7-p/old-locale.cabal
--- old-locale-1.0.0.7/old-locale.cabal	2014-11-21 11:45:10.000000000 +0100
+++ old-locale-1.0.0.7-p/old-locale.cabal	2017-03-17 17:54:28.565646559 +0100
@@ -31,5 +31,5 @@
     exposed-modules:
         System.Locale
 
-    build-depends: base >= 4.2 && < 4.9
+    build-depends: base >= 4.2
     ghc-options: -Wall
