diff -Naur GConf-2.16.0.orig/gconf/gconftool.c GConf-2.16.0/gconf/gconftool.c
--- GConf-2.16.0.orig/gconf/gconftool.c	2006-10-05 14:37:28.000000000 +0200
+++ GConf-2.16.0/gconf/gconftool.c	2006-10-05 20:34:56.000000000 +0200
@@ -960,8 +960,9 @@
 
       umask (022);
       retval = do_makefile_install (conf, args, FALSE);
-      
+
       gconf_engine_unref (conf);
+	 g_spawn_command_line_sync ("/usr/bin/killall -q -TERM " GCONF_SERVERDIR "/" GCONFD, NULL, NULL, NULL, NULL);
 
       return retval;
     }
@@ -974,6 +975,7 @@
       retval = do_makefile_install (conf, args, TRUE);
       
       gconf_engine_unref (conf);
+	 g_spawn_command_line_sync ("/usr/bin/killall -q -TERM " GCONF_SERVERDIR "/" GCONFD, NULL, NULL, NULL, NULL);
 
       return retval;
     }
