--- GConf2-2.13.5/gconf/gconftool.c	2006-03-03 14:30:01.000000000 +0100
+++ GConf2-2.13.5/gconf/gconftool.c	2006-03-03 14:38:46.000000000 +0100
@@ -476,6 +476,7 @@
 static int do_recursive_unset (GConfEngine* conf, const gchar** args);
 static int do_all_subdirs(GConfEngine* conf, const gchar** args);
 static int do_load_file(GConfEngine* conf, LoadType load_type, gboolean unload, const gchar* file, const gchar** base_dirs);
+static int do_sync(GConfEngine* conf);
 static int do_short_docs (GConfEngine *conf, const gchar **args);
 static int do_long_docs (GConfEngine *conf, const gchar **args);
 static int do_get_schema_name (GConfEngine *conf, const gchar **args);
@@ -841,6 +842,8 @@
       gint retval;
 
       retval = do_load_file(conf, LOAD_SCHEMA_FILE, FALSE, schema_file, NULL);
+      if (!retval)
+	retval = do_sync(conf);
 
       gconf_engine_unref(conf);
 
@@ -853,6 +856,8 @@
       gint retval;
 
       retval = do_load_file(conf, LOAD_ENTRY_FILE, FALSE, entry_file, args);
+      if (!retval)
+	retval = do_sync(conf);
 
       gconf_engine_unref(conf);
 
@@ -865,6 +870,8 @@
       gint retval;
 
       retval = do_load_file(conf, LOAD_ENTRY_FILE, TRUE, unload_entry_file, args);
+      if (!retval)
+	retval = do_sync(conf);
 
       gconf_engine_unref(conf);
 
@@ -3629,7 +3636,6 @@
 
   xmlDocPtr doc;
   xmlNodePtr iter;
-  GError * err = NULL;
   /* file comes from the command line, is thus in locale charset */
   gchar *utf8_file = g_locale_to_utf8 (file, -1, NULL, NULL, NULL);;
 
@@ -3691,7 +3697,17 @@
           
       iter = iter->next;
     }
+  
+  return 0;
+#undef LOAD_TYPE_TO_LIST
+#undef LOAD_TYPE_TO_ROOT
+}
 
+static int
+do_sync(GConfEngine* conf)
+{
+  GError * err = NULL;
+  
   gconf_engine_suggest_sync(conf, &err);
 
   if (err != NULL)
@@ -3703,9 +3719,6 @@
     }
   
   return 0;
-
-#undef LOAD_TYPE_TO_LIST
-#undef LOAD_TYPE_TO_ROOT
 }
 
 static int

