diff -Naur pngnq.orig/Makefile pngnq/Makefile
--- pngnq.orig/Makefile	2006-02-23 15:34:15.281221624 +0100
+++ pngnq/Makefile	2006-02-23 15:36:01.686045632 +0100
@@ -2,16 +2,16 @@
 
 CC=gcc
 
-PNGINC = ../libpng
+PNGINC = /usr/include
 PNGLIB = /usr/lib
 
-ZINC = ../zlib
-ZLIB = ../zlib
+ZINC = /usr/include
+ZLIB = /usr/lib
 
-CFLAGS = -O3 -g -Wall -I. -I$(PNGINC) -I$(ZINC) -funroll-loops -fomit-frame-pointer
+CFLAGS += -Wall -I. -I$(PNGINC) -I$(ZINC) -funroll-loops -fomit-frame-pointer
 
-#LDFLAGS = -L$(PNGLIB) -lpng -L$(ZLIB) -lz -lm
-LDFLAGS = $(PNGLIB)/libpng.a -L$(ZLIB) -lz -lm
+LDFLAGS = -L$(PNGLIB) -lpng -L$(ZLIB) -lz -lm
+#LDFLAGS = $(PNGLIB)/libpng.a -L$(ZLIB) -lz -lm
 #LDFLAGS = $(PNGLIB)/libpng.a $(ZLIB)/libz.a -lm
 
 OBJS = pngnq.o rwpng.o neuquant32.o
@@ -23,3 +23,6 @@
 
 clean:
 	rm -f pngnq $(OBJS)
+
+install: pngnq
+	/usr/bin/install -c -g root -o root -m 755 pngnq $(DESTDIR)/usr/bin/
