--- Linux-PAM-1.4.0/configure.ac	2020-06-08 12:17:27.000000000 +0200
+++ Linux-PAM-1.4.0.new/configure.ac	2020-07-17 13:54:25.078102620 +0200
@@ -25,15 +25,6 @@
         then
                 sysconfdir="/etc"
         fi
-	if test ${libdir} = '${exec_prefix}/lib'
-	then
-		case "$host_cpu" in
-		    x86_64|ppc64|s390x|sparc64)
-			libdir="/lib64" ;;
-		    *)
-			libdir="/lib" ;;
-		esac
-	fi
 	if test ${sbindir} = '${exec_prefix}/sbin'
         then
                 sbindir="/sbin"
