--- Linux-PAM-1.4.0/modules/pam_namespace/Makefile.am	2020-06-08 12:17:27.000000000 +0200
+++ Linux-PAM-1.4.0.new/modules/pam_namespace/Makefile.am	2020-07-17 14:32:50.562524236 +0200
@@ -18,7 +18,7 @@
 securelibdir = $(SECUREDIR)
 secureconfdir = $(SCONFIGDIR)
 namespaceddir = $(SCONFIGDIR)/namespace.d
-servicedir = $(prefix)/lib/systemd/system
+servicedir = /lib/systemd/system
 
 AM_CFLAGS = -I$(top_srcdir)/libpam/include -I$(top_srcdir)/libpamc/include \
         -DSECURECONF_DIR=\"$(SCONFIGDIR)/\" $(WARN_CFLAGS)
