diff --git a/backend/sm3600.h b/backend/sm3600.h
index 2ecbeb2..5d8a8d8 100644
--- a/backend/sm3600.h
+++ b/backend/sm3600.h
@@ -77,7 +77,11 @@ Start: 2.4.2001
 
 /* ====================================================================== */
 
+#if !defined(__STDC_VERSION__) || __STDC_VERSION__ < 202311L
 typedef enum { false, true } TBool;
+#else
+typedef bool TBool;
+#endif /* GCC < 15 */
 
 typedef SANE_Status TState;
 
