=== modified file 'protocols/oscar/ssi.c'
--- protocols/oscar/ssi.c
+++ protocols/oscar/ssi.c
@@ -151,7 +151,7 @@
 
 	cur = sess->ssi.items;
 	while (cur) {
-		if (cur->name)  g_free(cur->name);
+		g_free(cur->name);
 		if (cur->data)  aim_freetlvchain((aim_tlvlist_t **)&cur->data);
 		delitem = cur;
 		cur = cur->next;
@@ -1224,7 +1224,7 @@
 		guint16 namelen, tbslen;
 
 		if (!sess->ssi.items) {
-			if (!(sess->ssi.items = g_new(struct aim_ssi_item, 1)))
+			if (!(sess->ssi.items = g_new0(struct aim_ssi_item, 1)))
 				return -ENOMEM;
 			cur = sess->ssi.items;
 		} else {

