--- zlib-1.2.3/configure	2005-07-11 22:11:57.000000000 +0200
+++ zlib-1.2.3-p/configure	2007-03-10 01:38:43.000000000 +0100
@@ -73,7 +73,8 @@
 
 if test "$gcc" -eq 1 && ($cc -c $cflags $test.c) 2>/dev/null; then
   CC="$cc"
-  SFLAGS=${CFLAGS-"-fPIC -O3"}
+  SFLAGS=${CFLAGS-"-O3"}
+  SFLAGS="${SFLAGS} -fPIC"
   CFLAGS="$cflags"
   case `(uname -s || echo unknown) 2>/dev/null` in
   Linux | linux | GNU | GNU/*) LDSHARED=${LDSHARED-"$cc -shared -Wl,-soname,libz.so.1"};;
