diff -Naur nano-2.7.1/doc/nanorc.sample.in nano-2.7.1-p/doc/nanorc.sample.in
--- nano-2.7.1/doc/nanorc.sample.in	2016-10-26 09:41:14.000000000 +0200
+++ nano-2.7.1-p/doc/nanorc.sample.in	2016-10-29 16:27:58.982780986 +0200
@@ -24,7 +24,7 @@
 # set autoindent
 
 ## Back up files to the current filename plus a tilde.
-# set backup
+set backup
 
 ## The directory to put unique backup files in.
 # set backupdir ""
@@ -39,18 +39,18 @@
 ## paragraphs.  They cannot contain blank characters.  Only closing
 ## punctuation, optionally followed by closing brackets, can end
 ## sentences.
-# set brackets ""')>]}"
+set brackets ""')>]}"
 
 ## Do case-sensitive searches by default.
 # set casesensitive
 
 ## Constantly display the cursor position in the statusbar.  Note that
 ## this overrides "quickblank".
-# set constantshow
+set constantshow
 ## (The old form, 'const', is deprecated.)
 
 ## Use cut-to-end-of-line by default.
-# set cut
+set cut
 
 ## Set the line length for wrapping text and justifying paragraphs.
 ## If the value is 0 or less, the wrapping point will be the screen
@@ -58,10 +58,10 @@
 # set fill -8
 
 ## Remember the used search/replace strings for the next session.
-# set historylog
+set historylog
 
 ## Make the justify command kill whitespace at the end of lines.
-# set justifytrim
+set justifytrim
 
 ## Display line numbers to the left of the text.
 # set linenumbers
@@ -74,7 +74,7 @@
 ## The opening and closing brackets that can be found by bracket
 ## searches.  They cannot contain blank characters.  The former set must
 ## come before the latter set, and both must be in the same order.
-# set matchbrackets "(<[{)>]}"
+set matchbrackets "(<[{)>]}"
 
 ## Use the blank line below the titlebar as extra editing space.
 # set morespace
@@ -87,7 +87,7 @@
 
 ## Switch on multiple file buffers (inserting a file will put it into
 ## a separate buffer).
-# set multibuffer
+set multibuffer
 
 ## Don't convert files from DOS/Mac format.
 # set noconvert
@@ -96,10 +96,10 @@
 # set nohelp
 
 ## Don't add newlines to the ends of files.
-# set nonewlines
+set nonewlines
 
 ## Don't wrap text at all.
-# set nowrap
+set nowrap
 
 ## Set operating directory.  nano will not read or write files outside
 ## this directory and its subdirectories.  Also, the current directory
@@ -108,7 +108,7 @@
 # set operatingdir ""
 
 ## Remember the cursor position in each file for the next editing session.
-# set positionlog
+set positionlog
 ## (The old form, 'poslog', is deprecated.)
 
 ## Preserve the XON and XOFF keys (^Q and ^S).
@@ -118,7 +118,7 @@
 ## paragraphs.  They cannot contain blank characters.  Only closing
 ## punctuation, optionally followed by closing brackets, can end
 ## sentences.
-# set punct "!.?"
+set punct "!.?"
 
 ## Do quick statusbar blanking.  Statusbar messages will disappear after
 ## 1 keystroke instead of 26.  Note that "const" overrides this.
@@ -139,7 +139,7 @@
 # set rebindkeypad
 
 ## Do extended regular expression searches by default.
-# set regexp
+set regexp
 
 ## Put the cursor on the highlighted item in the file browser;
 ## useful for people who use a braille display.
@@ -160,7 +160,7 @@
 
 ## Use this spelling checker instead of the internal one.  This option
 ## does not properly have a default value.
-# set speller "aspell -x -c"
+set speller "aspell -x -c"
 
 ## Allow nano to be suspended.
 # set suspend
@@ -197,11 +197,11 @@
 
 ## Paint the interface elements of nano.
 ## These are examples; by default there are no colors.
-# set titlecolor brightwhite,blue
-# set numbercolor cyan
-# set statuscolor brightwhite,green
-# set keycolor green
-# set functioncolor yellow
+set titlecolor brightwhite,blue
+set numbercolor cyan
+set statuscolor brightwhite,green
+set keycolor green
+set functioncolor yellow
 
 
 ## Setup of syntax coloring.
@@ -246,7 +246,7 @@
 ## them easier to remember and faster to type using nano's -Y option.
 ##
 ## To include all existing syntax definitions, you can do:
-# include "@PKGDATADIR@/*.nanorc"
+include "@PKGDATADIR@/*.nanorc"
 
 
 ## Key bindings.
