diff -Naur nano-2.8.5/doc/sample.nanorc.in nano-2.8.5-p/doc/sample.nanorc.in
--- nano-2.8.5/doc/sample.nanorc.in	2017-06-25 09:31:27.000000000 +0200
+++ nano-2.8.5-p/doc/sample.nanorc.in	2017-06-25 22:59:45.358904517 +0200
@@ -24,7 +24,7 @@
 # set autoindent
 
 ## Back up files to the current filename plus a tilde.
-# set backup
+set backup
 
 ## The directory to put unique backup files in.
 # set backupdir ""
@@ -39,18 +39,18 @@
 ## paragraphs.  They cannot contain blank characters.  Only closing
 ## punctuation, optionally followed by closing brackets, can end
 ## sentences.
-# set brackets ""')>]}"
+set brackets ""')>]}"
 
 ## Do case-sensitive searches by default.
 # set casesensitive
 
 ## Constantly display the cursor position in the status bar.  Note that
 ## this overrides "quickblank".
-# set constantshow
+set constantshow
 ## (The old form, 'const', is deprecated.)
 
 ## Use cut-to-end-of-line by default.
-# set cut
+set cut
 
 ## Set the line length for wrapping text and justifying paragraphs.
 ## If the value is 0 or less, the wrapping point will be the screen
@@ -58,13 +58,13 @@
 # set fill -8
 
 ## Remember the used search/replace strings for the next session.
-# set historylog
+set historylog
 
 ## Make the justify command kill whitespace at the end of lines.
-# set justifytrim
+set justifytrim
 
 ## Display line numbers to the left of the text.
-# set linenumbers
+set linenumbers
 
 ## Enable vim-style lock-files.  This is just to let a vim user know you
 ## are editing a file [s]he is trying to edit and vice versa. There are
@@ -74,7 +74,7 @@
 ## The opening and closing brackets that can be found by bracket
 ## searches.  They cannot contain blank characters.  The former set must
 ## come before the latter set, and both must be in the same order.
-# set matchbrackets "(<[{)>]}"
+set matchbrackets "(<[{)>]}"
 
 ## Use the blank line below the title bar as extra editing space.
 # set morespace
@@ -87,7 +87,7 @@
 
 ## Switch on multiple file buffers (inserting a file will put it into
 ## a separate buffer).
-# set multibuffer
+set multibuffer
 
 ## Don't convert files from DOS/Mac format.
 # set noconvert
@@ -100,10 +100,10 @@
 # set nopauses
 
 ## Don't add newlines to the ends of files.
-# set nonewlines
+set nonewlines
 
 ## Don't wrap text at all.
-# set nowrap
+set nowrap
 
 ## Set operating directory.  nano will not read or write files outside
 ## this directory and its subdirectories.  Also, the current directory
@@ -112,7 +112,7 @@
 # set operatingdir ""
 
 ## Remember the cursor position in each file for the next editing session.
-# set positionlog
+set positionlog
 ## (The old form, 'poslog', is deprecated.)
 
 ## Preserve the XON and XOFF keys (^Q and ^S).
@@ -122,7 +122,7 @@
 ## paragraphs.  They cannot contain blank characters.  Only closing
 ## punctuation, optionally followed by closing brackets, can end
 ## sentences.
-# set punct "!.?"
+set punct "!.?"
 
 ## Do quick status-bar blanking.  Status-bar messages will disappear after
 ## 1 keystroke instead of 26.  Note that "const" overrides this.
@@ -143,7 +143,7 @@
 # set rebindkeypad
 
 ## Do extended regular expression searches by default.
-# set regexp
+set regexp
 
 ## Put the cursor on the highlighted item in the file browser;
 ## useful for people who use a braille display.
@@ -164,7 +164,7 @@
 
 ## Use this spelling checker instead of the internal one.  This option
 ## does not properly have a default value.
-# set speller "aspell -x -c"
+set speller "aspell -x -c"
 
 ## Allow nano to be suspended.
 # set suspend
@@ -201,13 +201,13 @@
 
 ## Paint the interface elements of nano.
 ## These are examples; by default there are no colors.
-# set titlecolor brightwhite,blue
-# set statuscolor brightwhite,green
-# set numbercolor cyan
-# set keycolor cyan
-# set functioncolor green
+set titlecolor brightwhite,blue
+set statuscolor brightwhite,green
+set numbercolor cyan
+set keycolor cyan
+set functioncolor green
 ## In root's .nanorc you might want to use:
-# set titlecolor brightwhite,red
+#set titlecolor brightwhite,red
 # set statuscolor brightwhite,red
 # set numbercolor magenta
 # set keycolor brightmagenta
@@ -256,7 +256,7 @@
 ## them easier to remember and faster to type using nano's -Y option.
 ##
 ## To include all existing syntax definitions, you can do:
-# include "@PKGDATADIR@/*.nanorc"
+include "@PKGDATADIR@/*.nanorc"
 
 
 ## Key bindings.
