diff -Naur nano-3.0/doc/sample.nanorc.in nano-3.0-p/doc/sample.nanorc.in
--- nano-3.0/doc/sample.nanorc.in	2018-09-09 11:39:27.000000000 +0200
+++ nano-3.0-p/doc/sample.nanorc.in	2018-09-10 12:33:26.751551047 +0200
@@ -27,7 +27,7 @@
 # set autoindent
 
 ## Back up files to the current filename plus a tilde.
-# set backup
+set backup
 
 ## The directory to put unique backup files in.
 # set backupdir ""
@@ -38,7 +38,7 @@
 ## The characters treated as closing brackets when justifying paragraphs.
 ## This may not include any blank characters.  Only closing punctuation,
 ## optionally followed by these closing brackets, can end sentences.
-# set brackets ""')>]}"
+set brackets ""')>]}"
 
 ## Do case-sensitive searches by default.
 # set casesensitive
@@ -57,7 +57,7 @@
 # set fill -8
 
 ## Remember the used search/replace strings for the next session.
-# set historylog
+set historylog
 
 ## Display line numbers to the left of the text.
 # set linenumbers
@@ -70,7 +70,7 @@
 ## The opening and closing brackets that can be found by bracket
 ## searches.  They cannot contain blank characters.  The former set must
 ## come before the latter set, and both must be in the same order.
-# set matchbrackets "(<[{)>]}"
+set matchbrackets "(<[{)>]}"
 
 ## Use the blank line below the title bar as extra editing space.
 # set morespace
@@ -83,7 +83,7 @@
 
 ## Switch on multiple file buffers (inserting a file will put it into
 ## a separate buffer).
-# set multibuffer
+set multibuffer
 
 ## Don't convert files from DOS/Mac format.
 # set noconvert
@@ -99,7 +99,7 @@
 # set nopauses
 
 ## Do not automatically hard-wrap text.
-# set nowrap
+set nowrap
 
 ## Set operating directory.  nano will not read or write files outside
 ## this directory and its subdirectories.  Also, the current directory
@@ -117,7 +117,7 @@
 ## paragraphs.  They cannot contain blank characters.  Only closing
 ## punctuation, optionally followed by closing brackets, can end
 ## sentences.
-# set punct "!.?"
+set punct "!.?"
 
 ## Do quick status-bar blanking.  Status-bar messages will disappear after
 ## 1 keystroke instead of 26.  Note that "constantshow" overrides this.
@@ -134,7 +134,7 @@
 # set rebindkeypad
 
 ## Do extended regular expression searches by default.
-# set regexp
+set regexp
 
 ## Put the cursor on the highlighted item in the file browser;
 ## useful for people who use a braille display.
@@ -155,7 +155,7 @@
 
 ## Use this spelling checker instead of the internal one.  This option
 ## does not have a default value.
-# set speller "aspell -x -c"
+set speller "aspell -x -c"
 
 ## Allow nano to be suspended.
 # set suspend
@@ -196,13 +196,13 @@
 
 ## Paint the interface elements of nano.  These are examples;
 ## by default there are no colors, except for errorcolor.
-# set titlecolor brightwhite,blue
-# set statuscolor brightwhite,green
-# set errorcolor brightwhite,red
-# set selectedcolor brightwhite,magenta
-# set numbercolor cyan
-# set keycolor cyan
-# set functioncolor green
+set titlecolor brightwhite,blue
+set statuscolor brightwhite,green
+set errorcolor brightwhite,red
+set selectedcolor brightwhite,magenta
+set numbercolor cyan
+set keycolor cyan
+set functioncolor green
 ## In root's .nanorc you might want to use:
 # set titlecolor brightwhite,magenta
 # set statuscolor brightwhite,magenta
@@ -255,7 +255,7 @@
 ## them easier to remember and faster to type using nano's -Y option.
 
 ## To include all existing syntax definitions, you can do:
-# include "@PKGDATADIR@/*.nanorc"
+include "@PKGDATADIR@/*.nanorc"
 
 
 ## Key bindings.
