--- nano-5.9/doc/sample.nanorc.in	2021-10-06 10:44:45.000000000 +0200
+++ nano-5.9.new/doc/sample.nanorc.in	2021-12-13 16:49:17.509705128 +0100
@@ -22,7 +22,7 @@
 # set autoindent
 
 ## Back up files to the current filename plus a tilde.
-# set backup
+set backup
 
 ## The directory to put unique backup files in.
 # set backupdir ""
@@ -36,7 +36,7 @@
 ## The characters treated as closing brackets when justifying paragraphs.
 ## This may not include any blank characters.  Only closing punctuation,
 ## optionally followed by these closing brackets, can end sentences.
-# set brackets ""')>]}"
+set brackets ""')>]}"
 
 ## Automatically hard-wrap the current line when it becomes overlong.
 # set breaklonglines
@@ -59,7 +59,7 @@
 # set fill -8
 
 ## Remember the used search/replace strings for the next session.
-# set historylog
+set historylog
 
 ## Display a "scrollbar" on the righthand side of the edit window.
 # set indicator
@@ -81,7 +81,7 @@
 ## The opening and closing brackets that are found by a matching-bracket
 ## search.  This may not contain blank characters.  The opening brackets
 ## must come before the closing ones, and they must be in the same order.
-# set matchbrackets "(<[{)>]}"
+set matchbrackets "(<[{)>]}"
 
 ## Suppress the title bar and show the filename plus a cursor-position
 ## percentage in the space of the status bar.
@@ -95,7 +95,7 @@
 
 ## Switch on multiple file buffers (inserting a file will put it into
 ## a separate buffer).
-# set multibuffer
+set multibuffer
 
 ## Don't convert files from DOS/Mac format.
 # set noconvert
@@ -121,7 +121,7 @@
 ## The characters treated as closing punctuation when justifying paragraphs.
 ## This may not contain blank characters.  Only these closing punctuations,
 ## optionally followed by closing brackets, can end sentences.
-# set punct "!.?"
+set punct "!.?"
 
 ## Make status-bar messages disappear after 1 keystroke instead of after 20.
 # set quickblank
@@ -139,6 +139,7 @@
 ## Do regular-expression searches by default.
 ## Regular expressions are of the extended type (ERE).
 # set regexp
+set regexp
 
 ## Save a changed buffer automatically on exit; don't prompt.
 # set saveonexit
@@ -161,7 +162,7 @@
 
 ## Use this spelling checker instead of the internal one.  This option
 ## does not have a default value.
-# set speller "aspell -x -c"
+set speller "aspell -x -c"
 
 ## Use the end of the title bar for some state flags: I = auto-indenting,
 ## M = mark, L = hard-wrapping long lines, R = recording, S = soft-wrapping.
@@ -215,6 +216,14 @@
 # set numbercolor cyan
 # set keycolor cyan
 # set functioncolor green
+set titlecolor brightwhite,blue
+set statuscolor brightwhite,green
+set errorcolor brightwhite,red
+set selectedcolor brightwhite,magenta
+set stripecolor ,yellow
+set numbercolor cyan
+set keycolor cyan
+set functioncolor green
 
 ## In root's .nanorc you might want to use:
 # set titlecolor bold,white,magenta
@@ -234,7 +243,7 @@
 ## For all details, see 'man nanorc', section SYNTAX HIGHLIGHTING.
 
 ## To include most of the existing syntax definitions, you can do:
-# include "@PKGDATADIR@/*.nanorc"
+include "@PKGDATADIR@/*.nanorc"
 
 ## Or you can select just the ones you need.  For example:
 # include "@PKGDATADIR@/html.nanorc"
