--- nano-5.3/doc/sample.nanorc.in	2020-09-28 14:57:36.000000000 +0200
+++ nano-5.3.new/doc/sample.nanorc.in	2020-10-09 13:49:51.304869102 +0200
@@ -22,7 +22,7 @@
 # set autoindent
 
 ## Back up files to the current filename plus a tilde.
-# set backup
+set backup
 
 ## The directory to put unique backup files in.
 # set backupdir ""
@@ -36,7 +36,7 @@
 ## The characters treated as closing brackets when justifying paragraphs.
 ## This may not include any blank characters.  Only closing punctuation,
 ## optionally followed by these closing brackets, can end sentences.
-# set brackets ""')>]}"
+set brackets ""')>]}"
 
 ## Automatically hard-wrap the current line when it becomes overlong.
 # set breaklonglines
@@ -60,7 +60,7 @@
 # set fill -8
 
 ## Remember the used search/replace strings for the next session.
-# set historylog
+set historylog
 
 ## Display a "scrollbar" on the righthand side of the edit window.
 # set indicator
@@ -82,7 +82,7 @@
 ## The opening and closing brackets that can be found by bracket
 ## searches.  They cannot contain blank characters.  The former set must
 ## come before the latter set, and both must be in the same order.
-# set matchbrackets "(<[{)>]}"
+set matchbrackets "(<[{)>]}"
 
 ## Enable mouse support, if available for your system.  When enabled,
 ## mouse clicks can be used to place the cursor, set the mark (with a
@@ -92,7 +92,7 @@
 
 ## Switch on multiple file buffers (inserting a file will put it into
 ## a separate buffer).
-# set multibuffer
+set multibuffer
 
 ## Don't convert files from DOS/Mac format.
 # set noconvert
@@ -119,7 +119,7 @@
 ## paragraphs.  They cannot contain blank characters.  Only closing
 ## punctuation, optionally followed by closing brackets, can end
 ## sentences.
-# set punct "!.?"
+set punct "!.?"
 
 ## Do quick status-bar blanking.  Status-bar messages will disappear after
 ## 1 keystroke instead of 26.  Note that "constantshow" overrides this.
@@ -138,6 +138,7 @@
 ## Do regular-expression searches by default.
 ## Regular expressions are of the extended type (ERE).
 # set regexp
+set regexp
 
 ## Save a changed buffer automatically on exit; don't prompt.
 # set saveonexit
@@ -160,7 +161,7 @@
 
 ## Use this spelling checker instead of the internal one.  This option
 ## does not have a default value.
-# set speller "aspell -x -c"
+set speller "aspell -x -c"
 
 ## Use the end of the title bar for some state flags: I = auto-indenting,
 ## M = mark, L = hard-wrapping long lines, R = recording, S = soft-wrapping.
@@ -214,6 +215,14 @@
 # set numbercolor cyan
 # set keycolor cyan
 # set functioncolor green
+set titlecolor brightwhite,blue
+set statuscolor brightwhite,green
+set errorcolor brightwhite,red
+set selectedcolor brightwhite,magenta
+set stripecolor ,yellow
+set numbercolor cyan
+set keycolor cyan
+set functioncolor green
 
 ## In root's .nanorc you might want to use:
 # set titlecolor bold,lightwhite,magenta
@@ -231,7 +240,7 @@
 ## For all details, see 'man nanorc', section SYNTAX HIGHLIGHTING.
 
 ## To include most of the existing syntax definitions, you can do:
-# include "@PKGDATADIR@/*.nanorc"
+include "@PKGDATADIR@/*.nanorc"
 
 ## Or you can select just the ones you need.  For example:
 # include "@PKGDATADIR@/html.nanorc"
