diff -Naur ttmkfdir-3.0.9.orig/Makefile ttmkfdir-3.0.9/Makefile
--- ttmkfdir-3.0.9.orig/Makefile	2006-10-06 15:08:27.000000000 +0200
+++ ttmkfdir-3.0.9/Makefile	2006-10-06 15:08:48.000000000 +0200
@@ -24,8 +24,8 @@
 FREETYPE_INCL=$(shell freetype-config --cflags)
 FREETYPE_LIB=$(shell freetype-config --libs)
 
-OPTFLAGS=
-DEBUG=-ggdb
+OPTFLAGS=$(CFLAGS)
+DEBUG=
 CXX=g++
 CXXFLAGS=-Wall -pedantic $(FREETYPE_INCL) $(DEBUG) $(OPTFLAGS)
 LDFLAGS=$(FREETYPE_LIB) $(DEBUG)
