diff -aur libXfont-X11R7.2-1.2.7/src/FreeType/fttools.c fw_libXfont-X11R7.2-1.2.7/src/FreeType/fttools.c
--- libXfont-X11R7.2-1.2.7/src/FreeType/fttools.c	2007-01-09 01:08:44.000000000 +0100
+++ fw_libXfont-X11R7.2-1.2.7/src/FreeType/fttools.c	2008-01-25 18:59:40.000000000 +0100
@@ -43,13 +43,6 @@
 #include FT_TRUETYPE_IDS_H
 #include "ft.h"
 
-/* backward compatibility hack */
-#if (FREETYPE_VERSION < 2001008)
-# ifndef ft_isdigit
-#  define ft_isdigit isdigit
-# endif
-#endif
-
 #ifndef LSBFirst
 #define LSBFirst 0
 #define MSBFirst 1
