--- dazuko_redirfs.c	2008/03/12 11:42:38	1.1
+++ dazuko_redirfs.c	2008/03/12 12:05:19
@@ -284,7 +284,7 @@
 	if (xfs->nd == NULL || xfs->free_full_filename)
 		return 0;
 	
-	if (xfs->nd->mnt == NULL || xfs->nd->dentry == NULL)
+	if (xfs->nd->path.mnt == NULL || xfs->nd->path.dentry == NULL)
 		return 0;
 
 	/* check if we need to allocate a buffer */
@@ -304,7 +304,7 @@
 	/* make sure we don't already have a vfsmount */
 	if (!xfs->mntput_vfsmount)
 	{
-		xfs->vfsmount = mntget(xfs->nd->mnt);
+		xfs->vfsmount = mntget(xfs->nd->path.mnt);
 
 		/* the vfsmount will need to be put back */
 		xfs->mntput_vfsmount = 1;
@@ -313,13 +313,13 @@
 	/* make sure we don't already have a dentry */
 	if (!xfs->dput_dentry)
 	{
-		xfs->dentry = dget(xfs->nd->dentry);
+		xfs->dentry = dget(xfs->nd->path.dentry);
 
 		/* the dentry will need to be put back */
 		xfs->dput_dentry = 1;
 	}
 
-	temp = d_path(xfs->dentry, xfs->vfsmount, xfs->buffer, PAGE_SIZE);
+	temp = d_path(&xfs->nd->path, xfs->buffer, PAGE_SIZE);
 
 	/* make sure we really got a new filename */
 	if (temp == NULL)
