diff -Naur dazuko-2.3.1-pre3.orig/configure dazuko-2.3.1-pre3/configure
--- dazuko-2.3.1-pre3.orig/configure	2006-08-30 17:32:27.000000000 +0200
+++ dazuko-2.3.1-pre3/configure	2006-08-30 17:32:39.000000000 +0200
@@ -782,7 +782,7 @@
 
 	echo "" >> Makefile
 	echo "dummy_rule:" >> Makefile
-        echo "	\$(MAKE) -C ${LINUX_OBJ} include/linux/version.h include/asm scripts" >> Makefile
+#        echo "	\$(MAKE) -C ${LINUX_OBJ} include/linux/version.h include/asm scripts" >> Makefile
         echo "	\$(MAKE) -C ${LINUX_OBJ} SUBDIRS=\"`pwd`\" modules" >> Makefile
 	echo "	touch dummy_rule" >> Makefile
 
