diff --git a/configure b/configure
index 1caa547..bd57abc 100755
--- a/configure
+++ b/configure
@@ -117,7 +117,7 @@ linux_check_objdir()
 	then
 		echo -n "kernel build source in ${LINUX_CHECK_OBJ}... "
 
-		if [ -f "${LINUX_CHECK_OBJ}/include/linux/version.h" -a -f "${LINUX_CHECK_OBJ}/include/linux/autoconf.h" ]
+		if [ -f "${LINUX_CHECK_OBJ}/include/linux/version.h" -a -f "${LINUX_CHECK_OBJ}/include/generated/autoconf.h" ]
 		then
 			LINUX_OBJ="$LINUX_CHECK_OBJ"
 			echo "yes"
diff --git a/linux_conf.c b/linux_conf.c
index 70e9981..0e29e67 100644
--- a/linux_conf.c
+++ b/linux_conf.c
@@ -1,11 +1,9 @@
 #ifdef USE_CONFIG_H
 #else
-#include <linux/autoconf.h>
+#include <generated/autoconf.h>
 #endif
 #include <linux/version.h>
-#ifdef USE_UTSRELEASE_H
-#include <linux/utsrelease.h>
-#endif
+#include <generated/utsrelease.h>
 #include <stdio.h>
 
 #ifndef KERNEL_VERSION
