diff --git a/dazuko_redirfs.c b/dazuko_redirfs.c
index 35e4638..348d223 100644
--- a/dazuko_redirfs.c
+++ b/dazuko_redirfs.c
@@ -52,7 +52,7 @@
 ssize_t linux_dazuko_device_read(struct file *, char __user *, size_t, loff_t *);
 ssize_t linux_dazuko_device_write(struct file *, const char __user *, size_t, loff_t *);
 int linux_dazuko_device_open(struct inode *, struct file *);
-int linux_dazuko_device_ioctl(struct inode *inode, struct file *file, unsigned int cmd, unsigned long param);
+long linux_dazuko_device_ioctl(struct file *file, unsigned int cmd, unsigned long param);
 int linux_dazuko_device_release(struct inode *, struct file *);
 
 #ifndef WITHOUT_UDEV
@@ -86,7 +86,7 @@ static struct file_operations	fops = {
 		.owner		= THIS_MODULE,
 		.read		= linux_dazuko_device_read,
 		.write		= linux_dazuko_device_write,
-		.ioctl		= linux_dazuko_device_ioctl,
+		.unlocked_ioctl		= linux_dazuko_device_ioctl,
 		.open		= linux_dazuko_device_open,
 		.release	= linux_dazuko_device_release,
 	};
@@ -906,7 +906,7 @@ ssize_t linux_dazuko_device_write(struct file *file, const char *buffer, size_t
 	return XP_ERROR_INTERRUPT;
 }
 
-int linux_dazuko_device_ioctl(struct inode *inode, struct file *file, unsigned int cmd, unsigned long param)
+long linux_dazuko_device_ioctl(struct file *file, unsigned int cmd, unsigned long param)
 {
 	/* A daemon uses this function to interact with
 	 * the kernel. A daemon can set scanning parameters,
