---
 dazuko_redirfs.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/dazuko_redirfs.c
+++ b/dazuko_redirfs.c
@@ -659,7 +659,7 @@ int dazuko_sys_generic(struct inode *ino
 		event_p.set_mode = 1;
 		event_p.pid = current->pid;
 		event_p.set_pid = 1;
-		event_p.uid = current->uid;
+		event_p.uid = current_uid();
 		event_p.set_uid = 1;
 
 		dfs = (struct dazuko_file_struct *)xp_malloc(sizeof(struct dazuko_file_struct));
