diff -Naur sddm-kcm-5.17.1/src/advancedconfig.cpp sddm-kcm-5.17.1-p/src/advancedconfig.cpp
--- sddm-kcm-5.17.1/src/advancedconfig.cpp	2019-10-22 17:46:56.000000000 +0200
+++ sddm-kcm-5.17.1-p/src/advancedconfig.cpp	2019-10-26 20:12:15.950686424 +0200
@@ -190,9 +190,12 @@
         qDebug() << "Cannot find cursor theme value.";
     }
 
+	// wth -dpi 0
     if (!dpiValue.isEmpty()) {
-        args[QStringLiteral("kde_settings.conf/X11/ServerArguments")] = dpiArgument;
-    }
+        if (dpiValue >= QStringLiteral("24")) {
+           args[QStringLiteral("kde_settings.conf/X11/ServerArguments")] = dpiArgument;
+        }
+    }
     else {
         qDebug() << "Cannot find scaling DPI value.";
     }
