diff --git a/src/advancedconfig.cpp b/src/advancedconfig.cpp
--- a/src/advancedconfig.cpp
+++ b/src/advancedconfig.cpp
@@ -190,7 +190,8 @@
         qDebug() << "Cannot find cursor theme value.";
     }
 
-    if (!dpiValue.isEmpty()) {
+    // minimal possible valid step from Force font DPI setings is 24
+    if (!dpiValue.isEmpty() && dpiValue.toInt() >= 24) {
         args[QStringLiteral("kde_settings.conf/X11/ServerArguments")] = dpiArgument;
     }
     else {

