diff -Naur sddm-kcm-5.17.1/src/CMakeLists.txt sddm-kcm-5.17.1-p/src/CMakeLists.txt
--- sddm-kcm-5.17.1/src/CMakeLists.txt	2019-10-22 17:46:56.000000000 +0200
+++ sddm-kcm-5.17.1-p/src/CMakeLists.txt	2019-10-26 19:13:55.097262783 +0200
@@ -1,7 +1,7 @@
 set(SDDM_CONFIG_FILE                 "${CMAKE_INSTALL_FULL_SYSCONFDIR}/sddm.conf"        CACHE PATH      "Path of the sddm config file")
 set(SDDM_CONFIG_DIR                  "${CMAKE_INSTALL_FULL_SYSCONFDIR}/sddm.conf.d"      CACHE PATH      "Path of the sddm config directory")
 set(SDDM_SYSTEM_CONFIG_DIR           "${CMAKE_INSTALL_PREFIX}/lib/sddm/sddm.conf.d"      CACHE PATH      "Path of the system sddm config directory")
-set(XSESSIONS_DIR                    "${CMAKE_INSTALL_PREFIX}/share/xsessions"           CACHE PATH      "Path of the xsessions")
+set(XSESSIONS_DIR                    "${CMAKE_INSTALL_FULL_SYSCONFDIR}/X11/sessions"     CACHE PATH      "Path of the xsessions")
 set(WAYLAND_SESSIONS_DIR             "${CMAKE_INSTALL_PREFIX}/share/wayland-sessions"    CACHE PATH      "Path of the wayland sessions")
 
 configure_file(config.h.in config.h IMMEDIATE @ONLY)
