diff --git a/configure.ac b/configure.ac
index 5e4cff126798d7ffa7e3ec05b8292c34aee2387e..8c20e50813e4cd274011b3ecda6415ff7f4b18c2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1054,6 +1054,10 @@ AG_GST_CHECK_FEATURE(VPX, [VPX decoder], vpx, [
       AC_DEFINE(HAVE_VPX_1_4, 1, [Defined if the VPX library version is 1.4 or bigger])
     ], [true])
 
+    PKG_CHECK_MODULES(VPX_180, vpx >= 1.8.0, [
+      AC_DEFINE(HAVE_VPX_1_8, 1, [Defined if the VPX library version is 1.8 or bigger])
+    ], [true])
+
     LIBS="$OLD_LIBS"
     CFLAGS="$OLD_CFLAGS"
   fi
diff --git a/ext/vpx/gstvpxdec.c b/ext/vpx/gstvpxdec.c
index c3f0f625f116525d3fbbfc20ba0130756c86561f..da4322513e26fcf270e9054d1d5ac2728d61fadf 100644
--- a/ext/vpx/gstvpxdec.c
+++ b/ext/vpx/gstvpxdec.c
@@ -62,6 +62,7 @@ gst_vpx_dec_post_processing_flags_get_type (void)
     {C_FLAGS (VP8_DEBLOCK), "Deblock", "deblock"},
     {C_FLAGS (VP8_DEMACROBLOCK), "Demacroblock", "demacroblock"},
     {C_FLAGS (VP8_ADDNOISE), "Add noise", "addnoise"},
+#ifndef HAVE_VPX_1_8
     {C_FLAGS (VP8_DEBUG_TXT_FRAME_INFO),
           "Print frame information",
         "visualize-frame-info"},
@@ -74,6 +75,7 @@ gst_vpx_dec_post_processing_flags_get_type (void)
     {C_FLAGS (VP8_DEBUG_TXT_RATE_INFO),
           "Print video rate info",
         "visualize-rate-info"},
+#endif
     {C_FLAGS (VP8_MFQE), "Multi-frame quality enhancement", "mfqe"},
     {0, NULL, NULL}
   };
diff --git a/ext/vpx/meson.build b/ext/vpx/meson.build
index 6ed44072833c5d697ad91d8f46a2a789666ca1c8..0d2340b322ceb33988fb67b37b1994ee57de2eb3 100644
--- a/ext/vpx/meson.build
+++ b/ext/vpx/meson.build
@@ -48,6 +48,10 @@ if vpx_dep.found()
     vpx_args += '-DHAVE_VPX_1_4'
   endif
 
+  if dependency('vpx', version : '>=1.8.0', required : false).found()
+    vpx_args += '-DHAVE_VPX_1_8'
+  endif
+
   gstvpx = library('gstvpx',
     vpx_sources,
     c_args : gst_plugins_good_args + vpx_args,
