diff -aur windowlab-1.39/Makefile windowlab-1.39-2/Makefile
--- windowlab-1.39/Makefile	2009-11-05 15:04:45.000000000 -0600
+++ windowlab-1.39-2/Makefile	2010-03-13 06:59:25.000000000 -0600
@@ -5,7 +5,7 @@
 EXTRA_LIBS += -lXext
 
 # Set this to the hardcoded location of all files if it's not /
-PREFIX = /usr/local
+PREFIX = /usr
 
 # Set this to the directory, below PREFIX, where man pages 
 # are expected. Below this directory, the target "install"
@@ -20,7 +20,7 @@
 CONFDIR = /etc/X11/windowlab
 
 # Set this to the location of the global configuration files
-SYSCONFDIR = $(CONFPREFIX)$(CONFDIR)
+SYSCONFDIR = /etc/X11/windowlab
 
 # Information about the location of the menurc file
 ifndef MENURC
@@ -34,9 +34,9 @@
 
 # Uncomment to add freetype support (requires XFree86 4.0.2 or later)
 # This needs -lXext above, even if you have disabled shape support
-#DEFINES += -DXFT
-#EXTRA_INC += `pkg-config --cflags xft`
-#EXTRA_LIBS += `pkg-config --libs xft`
+DEFINES += -DXFT
+EXTRA_INC += `pkg-config --cflags xft`
+EXTRA_LIBS += `pkg-config --libs xft`
 
 # Uncomment for debugging info (abandon all hope, ye who enter here)
 #DEFINES += -DDEBUG
