diff -Naur xf86-video-ati-19.0.0/src/radeon_glamor.c xf86-video-ati-19.0.0-p/src/radeon_glamor.c
--- xf86-video-ati-19.0.0/src/radeon_glamor.c	2019-03-06 17:56:24.000000000 +0100
+++ xf86-video-ati-19.0.0-p/src/radeon_glamor.c	2019-03-10 12:04:56.573499972 +0100
@@ -238,7 +238,7 @@
 		if (info->shadow_primary) {
 			if (usage != CREATE_PIXMAP_USAGE_BACKING_PIXMAP)
 				return fbCreatePixmap(screen, w, h, depth, usage);
-		} else if (usage != CREATE_PIXMAP_USAGE_BACKING_PIXMAP) {
+		} else {
 			pixmap = glamor_create_pixmap(screen, w, h, depth, usage);
 			if (pixmap)
 			    return pixmap;
