diff -up opencascade-7.5.0/src/IVtkDraw/IVtkDraw.cxx.vtk opencascade-7.5.0/src/IVtkDraw/IVtkDraw.cxx
--- opencascade-7.5.0/src/IVtkDraw/IVtkDraw.cxx.vtk	2020-11-03 07:50:01.000000000 -0700
+++ opencascade-7.5.0/src/IVtkDraw/IVtkDraw.cxx	2023-01-12 21:27:51.669725256 -0700
@@ -31,10 +31,8 @@
 #include <NCollection_List.hxx>
 #include <NCollection_DataMap.hxx>
 #include <TopTools_DataMapOfIntegerShape.hxx>
-#include <OpenGl_GraphicDriver.hxx>
 #include <V3d.hxx>
 #include <V3d_TypeOfOrientation.hxx>
-#include <Aspect_DisplayConnection.hxx>
 
 #include <IVtk_Types.hxx>
 #include <IVtkVTK_ShapeData.hxx>
@@ -74,6 +72,9 @@
 #include <vtkSmartPointer.h>
 #include <vtkTIFFWriter.h>
 #include <vtkWindowToImageFilter.h>
+// Move X11 including files after vtk
+#include <OpenGl_GraphicDriver.hxx>
+#include <Aspect_DisplayConnection.hxx>
 #ifndef _WIN32
   #include <X11/X.h>
   #include <X11/Shell.h>
diff -up opencascade-7.5.0/src/IVtkDraw/IVtkDraw_Interactor.hxx.vtk opencascade-7.5.0/src/IVtkDraw/IVtkDraw_Interactor.hxx
--- opencascade-7.5.0/src/IVtkDraw/IVtkDraw_Interactor.hxx.vtk	2020-11-03 07:50:01.000000000 -0700
+++ opencascade-7.5.0/src/IVtkDraw/IVtkDraw_Interactor.hxx	2023-01-14 19:16:25.310673547 -0700
@@ -20,16 +20,6 @@
 #include <Standard_Macro.hxx>
 #include <Standard_Transient.hxx>
 
-#ifdef _WIN32
-#include <windows.h>
-#else
-#include <X11/X.h>
-#include <X11/Xlib.h>
-#include <X11/Shell.h>
-#include <X11/Xutil.h>
-#include <tk.h>
-#endif
-
 // prevent disabling some MSVC warning messages by VTK headers 
 #include <Standard_WarningsDisable.hxx>
 #include <vtkRenderWindowInteractor.h>
@@ -40,6 +30,17 @@
 #include <IVtkDraw_HighlightAndSelectionPipeline.hxx>
 #include <Aspect_Window.hxx>
 
+// Include X11 headers after VTK for 9.2
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <X11/X.h>
+#include <X11/Xlib.h>
+#include <X11/Shell.h>
+#include <X11/Xutil.h>
+#include <tk.h>
+#endif
+
 class vtkWin32RenderWindowInteractor;
 typedef vtkSmartPointer<IVtkTools_ShapePicker> PSelector;
 
diff -up occt-V7_6_3/src/IVtkDraw/IVtkDraw_Interactor.cxx.vtk occt-V7_6_3/src/IVtkDraw/IVtkDraw_Interactor.cxx
--- occt-V7_6_3/src/IVtkDraw/IVtkDraw_Interactor.cxx.vtk	2022-07-22 02:53:38.000000000 -0600
+++ occt-V7_6_3/src/IVtkDraw/IVtkDraw_Interactor.cxx	2023-01-14 21:07:30.367040080 -0700
@@ -15,21 +15,6 @@
 
 // prevent disabling some MSVC warning messages by VTK headers 
 #include <Standard_WarningsDisable.hxx>
-#ifdef _WIN32
-#include <vtkWin32RenderWindowInteractor.h>
-#include <vtkWin32OpenGLRenderWindow.h>
-#else
-#include <GL/glx.h>
-
-// Preventing naming collisions between
-// GLX and VTK versions 9.0 and above
-#ifdef AllValues
-#undef AllValues
-#endif
-
-#include <vtkXRenderWindowInteractor.h>
-#include <vtkXOpenGLRenderWindow.h>
-#endif
 #include <vtkActor.h>
 #include <vtkActorCollection.h>
 #include <vtkCommand.h>
@@ -45,6 +30,16 @@
 #include <IVtkTools_ShapeObject.hxx>
 #include <IVtkTools_ShapeDataSource.hxx>
 
+// Include X11 headers after VTK for VTK 9.2.5
+#ifdef _WIN32
+#include <vtkWin32RenderWindowInteractor.h>
+#include <vtkWin32OpenGLRenderWindow.h>
+#else
+#include <GL/glx.h>
+#include <vtkXRenderWindowInteractor.h>
+#include <vtkXOpenGLRenderWindow.h>
+#endif
+
 #include <Message.hxx>
 #include <Message_Messenger.hxx>
 
