diff -Naur xine-lib-1-rc5.orig/src/libw32dll/wine/ldt_keeper.c xine-lib-1-rc5/src/libw32dll/wine/ldt_keeper.c
--- xine-lib-1-rc5.orig/src/libw32dll/wine/ldt_keeper.c	2004-06-13 01:19:56.000000000 +0200
+++ xine-lib-1-rc5/src/libw32dll/wine/ldt_keeper.c	2004-07-01 01:22:30.000000000 +0200
@@ -67,10 +67,7 @@
 #include <asm/unistd.h>
 #include <asm/ldt.h>
 /* 2.5.xx+ calls this user_desc: */
-#include <linux/version.h>
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,47)
 #define modify_ldt_ldt_s user_desc
-#endif
 /* prototype it here, so we won't depend on kernel headers */
 #ifdef  __cplusplus
 extern "C" {
