diff -Naur usbutils-0.71.orig/update-usbids.sh usbutils-0.71/update-usbids.sh
--- usbutils-0.71.orig/update-usbids.sh	2005-07-01 12:24:55.000000000 +0200
+++ usbutils-0.71/update-usbids.sh	2005-07-01 12:29:57.000000000 +0200
@@ -4,7 +4,7 @@
 
 set -e
 SRC="http://www.linux-usb.org/usb.ids"
-DEST=usb.ids
+DEST=/usr/share/misc/usb.ids
 
 if which wget >/dev/null ; then
 	DL="wget -O $DEST.new $SRC"
