Index: xserver-xorg-input-evtouch/evtouch.c
===================================================================
--- xserver-xorg-input-evtouch.orig/evtouch.c	2007-09-17 17:20:21.736761738 +0900
+++ xserver-xorg-input-evtouch/evtouch.c	2007-09-17 17:31:22.274403633 +0900
@@ -492,9 +492,6 @@ DeviceOn (DeviceIntPtr dev)
 
         priv->buffer = XisbNew(local->fd, 64);
 
-        DBG (9, XisbTrace (priv->buffer, 1));
-
-
         if (!priv->buffer) 
         {
                 xf86CloseSerial(local->fd);
@@ -1207,7 +1204,6 @@ EVTouchPreInit(InputDriverPtr drv, IDevP
                     priv->diff[i][0], priv->diff[i][1]);
         }
         
-        xf86AlwaysCore(local, TRUE);
         priv->touch_flags = 0;
 
         local->history_size = xf86SetIntOption( local->options, "HistorySize", 0 );
