diff src/plugins/select.c src/plugins/select.c
--- src/plugins/select.c
+++ src/plugins/select.c
@@ -248,6 +248,7 @@ int prepare_pkgdb(char *srcdev)
 			}
 			fprintf(fp, "[options]\n");
 			fprintf(fp, "LogFile     = /var/log/pacman-g2.log\n");
+			fprintf(fp, "MaxTries = 5\n");
 			fprintf(fp, "[%s]\n", PACCONF);
 			fprintf(fp, "Server = file://%s/frugalware-%s\n\n", SOURCEDIR, ARCH);
 			fclose(fp);


