--- bonnie++-2.00a/Makefile.in	2024-01-16 13:16:43.955161465 +0100
+++ bonnie++-2.00a.new/Makefile.in	2024-01-16 13:17:52.270637196 +0100
@@ -52,8 +52,8 @@
 	$(LINK) generate_randfile.o -o generate_randfile
 
 install-bin: $(EXE) $(EXES)
-	mkdir -p $(eprefix)/bin $(eprefix)/sbin
-	@INSTALL_PROGRAM@ @stripping@ $(EXES) $(eprefix)/sbin
+	mkdir -p $(eprefix)/bin $(eprefix)/bin
+	@INSTALL_PROGRAM@ @stripping@ $(EXES) $(eprefix)/bin
 	@INSTALL_PROGRAM@ @stripping@ $(EXE) $(eprefix)/bin
 	@INSTALL_SCRIPT@ $(SCRIPTS) $(eprefix)/bin
 
