diff -Naur at-3.1.8.orig/Makefile.in at-3.1.8/Makefile.in
--- at-3.1.8.orig/Makefile.in	2002-01-18 07:55:33.000000000 +0100
+++ at-3.1.8/Makefile.in	2005-07-25 22:26:21.000000000 +0200
@@ -92,14 +92,6 @@
 	$(INSTALL) -g root -o root -m 755 -d $(IROOT)$(sbindir)
 	$(INSTALL) -g root -o root -m 755 -d $(IROOT)$(docdir)
 	$(INSTALL) -g root -o root -m 755 -d $(IROOT)$(atdocdir)
-	$(INSTALL) -m 755 -d $(IROOT)$(ATJOB_DIR)
-	$(INSTALL) -g $(DAEMON_GROUPNAME) -o $(DAEMON_USERNAME) -m 755 -d $(IROOT)$(ATSPOOL_DIR)
-	chmod 700 $(IROOT)$(ATJOB_DIR) $(IROOT)$(ATSPOOL_DIR)
-	chown $(DAEMON_USERNAME):$(DAEMON_GROUPNAME) $(IROOT)$(ATJOB_DIR) $(IROOT)$(ATSPOOL_DIR)
-	touch $(IROOT)$(LFILE)
-	chmod 600 $(IROOT)$(LFILE)
-	chown $(DAEMON_USERNAME):$(DAEMON_GROUPNAME) $(IROOT)$(LFILE)
-	test -f $(IROOT)$(etcdir)/at.allow || test -f $(IROOT)$(etcdir)/at.deny || $(INSTALL) -o root -m 600 at.deny $(IROOT)$(etcdir)/
 	$(INSTALL) -g root -o root -m 4755 -s at $(IROOT)$(bindir)
 	$(LN_S) -f at $(IROOT)$(bindir)/atq
 	$(LN_S) -f at $(IROOT)$(bindir)/atrm
