diff -Naur at-3.1.9.orig/Makefile.in at-3.1.9/Makefile.in
--- at-3.1.9.orig/Makefile.in	2005-08-03 10:51:42.000000000 +0200
+++ at-3.1.9/Makefile.in	2005-08-03 10:53:24.000000000 +0200
@@ -93,12 +93,6 @@
 	$(INSTALL) -g root -o root -m 755 -d $(IROOT)$(sbindir)
 	$(INSTALL) -g root -o root -m 755 -d $(IROOT)$(docdir)
 	$(INSTALL) -g root -o root -m 755 -d $(IROOT)$(atdocdir)
-	$(INSTALL) -g $(DAEMON_GROUPNAME) -o $(DAEMON_USERNAME) -m 755 -d $(IROOT)$(ATSPOOL_DIR) $(IROOT)$(ATJOB_DIR)
-	chmod 1770 $(IROOT)$(ATSPOOL_DIR) $(IROOT)$(ATJOB_DIR)
-	touch $(IROOT)$(LFILE)
-	chmod 600 $(IROOT)$(LFILE)
-	chown $(DAEMON_USERNAME):$(DAEMON_GROUPNAME) $(IROOT)$(LFILE)
-	test -f $(IROOT)$(etcdir)/at.allow || test -f $(IROOT)$(etcdir)/at.deny || $(INSTALL) -o root -g $(DAEMON_GROUPNAME) -m 640 at.deny $(IROOT)$(etcdir)/
 	$(INSTALL) -g $(DAEMON_GROUPNAME) -o $(DAEMON_USERNAME) -m 6755 -s at $(IROOT)$(bindir)
 	$(LN_S) -f at $(IROOT)$(bindir)/atq
 	$(LN_S) -f at $(IROOT)$(bindir)/atrm
