--- gtkimageview-1.6.2/tests/testlib/testlib.h	1970-01-01 01:00:00.000000000 +0100
+++ gtkimageview-1.6.2/tests/testlib/testlib.h	2008-10-28 23:32:34.000000000 +0100
@@ -0,0 +1,10 @@
+#ifndef __TESTLIB_H__
+#define __TESTLIB_H__
+
+#include <gtk/gtk.h>
+
+void          fake_realize                   (GtkWidget *widget);
+gboolean      g_main_context_wait_for_event  (GMainContext *context,
+                                              int           timeout);
+
+#endif

