diff -Naurp wxGTK-2.6.4/src/gtk/button.cpp wxGTK-2.6.4-p/src/gtk/button.cpp
--- wxGTK-2.6.4/src/gtk/button.cpp	2007-03-20 16:50:07.000000000 +0100
+++ wxGTK-2.6.4-p/src/gtk/button.cpp	2007-09-23 01:53:30.000000000 +0200
@@ -89,7 +89,7 @@ gtk_button_style_set_callback( GtkWidget
             right_border += default_border->right;
             top_border += default_border->top;
             bottom_border += default_border->bottom;
-            g_free( default_border );
+            gtk_border_free( default_border );
         }
 #else
         left_border = 6;
diff -Naurp wxGTK-2.6.4/src/gtk/window.cpp wxGTK-2.6.4-p/src/gtk/window.cpp
--- wxGTK-2.6.4/src/gtk/window.cpp	2007-03-20 16:50:07.000000000 +0100
+++ wxGTK-2.6.4-p/src/gtk/window.cpp	2007-09-23 01:53:30.000000000 +0200
@@ -3273,7 +3273,7 @@ void wxWindowGTK::DoSetSize( int x, int 
                 right_border += default_border->right;
                 top_border += default_border->top;
                 bottom_border += default_border->bottom;
-                g_free( default_border );
+                gtk_border_free( default_border );
             }
 #else
             left_border = 6;
