diff -Naur gcc-4.3.0/libjava/configure.ac gcc-4.3.0-p/libjava/configure.ac
--- gcc-4.3.0/libjava/configure.ac	2007-12-22 16:48:46.000000000 +0100
+++ gcc-4.3.0-p/libjava/configure.ac	2008-03-16 13:54:45.000000000 +0100
@@ -114,7 +114,7 @@
 AC_CHECK_TOOL(AR, ar)
 AC_CHECK_TOOL(RANLIB, ranlib, :)
 AC_PROG_AWK
-AC_CHECK_PROGS([JAR], [jar fastjar gjar], no)
+AC_CHECK_PROGS([JAR], [fastjar], yes)
 AC_PATH_PROG([ZIP], [zip], no)
 AC_PATH_PROG([UNZIP], [unzip], unzip)
 
