diff -ur a/gcc/ada/cal.c b/gcc/ada/cal.c
--- a/gcc/ada/cal.c	2009-04-09 17:00:19.000000000 +0200
+++ b/gcc/ada/cal.c	2015-03-01 01:34:53.744615329 +0100
@@ -74,6 +74,7 @@
 #endif
 #endif
 
+#include <time.h>
 void
 __gnat_timeval_to_duration (struct timeval *t, long *sec, long *usec)
 {
diff -ur a/gcc/ada/socket.c b/gcc/ada/socket.c
--- a/gcc/ada/socket.c	2014-08-12 21:49:19.000000000 +0200
+++ b/gcc/ada/socket.c	2015-03-01 01:34:12.438365517 +0100
@@ -88,6 +88,7 @@
 #include <string.h>
 /* Required for memcpy() */
 
+#include <unistd.h>
 extern void __gnat_disable_sigpipe (int fd);
 extern void __gnat_disable_all_sigpipes (void);
 extern int  __gnat_create_signalling_fds (int *fds);
