diff -Naur gcc-5.3.0/contrib/texi2pod.pl gcc-5.3.0-p/contrib/texi2pod.pl
--- gcc-5.3.0/contrib/texi2pod.pl	2014-08-11 09:29:41.000000000 +0200
+++ gcc-5.3.0-p/contrib/texi2pod.pl	2016-05-19 17:24:08.660026282 +0200
@@ -316,7 +316,7 @@
 	@columns = ();
 	for $column (split (/\s*\@tab\s*/, $1)) {
 	    # @strong{...} is used a @headitem work-alike
-	    $column =~ s/^\@strong{(.*)}$/$1/;
+	    $column =~ s/^\@strong\{(.*)\}$/$1/;
 	    push @columns, $column;
 	}
 	$_ = "\n=item ".join (" : ", @columns)."\n";
