--- gexiv2-0.10.5/gexiv2/gexiv2-stream-io.h	2017-03-20 12:04:52.000000000 +0100
+++ gexiv2-0.10.5.new/gexiv2/gexiv2-stream-io.h	2017-04-29 22:37:31.000000000 +0200
@@ -39,7 +39,7 @@
 	virtual Exiv2::byte* mmap (bool isWriteable = false);
 	virtual int munmap ();
 	virtual long tell () const;
-	virtual long size () const;
+	virtual size_t size () const;
 	virtual bool isopen () const;
 	virtual int error () const;
 	virtual bool eof () const;
--- gexiv2-0.10.5/gexiv2/gexiv2-stream-io.cpp	2017-03-11 21:41:13.000000000 +0100
+++ gexiv2-0.10.5.new/gexiv2/gexiv2-stream-io.cpp	2017-04-29 22:39:45.000000000 +0200
@@ -142,7 +142,7 @@
     return cb->Position (cb->handle);
 }
 
-long StreamIo::size () const {
+size_t StreamIo::size () const {
     return cb->Length (cb->handle);
 }
 
