diff -Naur fweelin-0.6.orig/configure.ac fweelin-0.6/configure.ac
--- fweelin-0.6.orig/configure.ac	2008-10-04 00:50:04.000000000 +0200
+++ fweelin-0.6/configure.ac	2012-10-05 12:00:05.399539305 +0200
@@ -98,7 +98,7 @@
 		http://www.http://www.mega-nerd.com/libsndfile/)])
 
 CFLAGS="$CFLAGS $SDL_CFLAGS"
-LIBS="$X_LIBS $X_PRE_LIBS -L/usr/X11R6/lib -lX11 $X_EXTRA_LIBS $LIBS"
+LIBS="$X_LIBS $X_PRE_LIBS -logg -lSDL -L/usr/X11R6/lib -lX11 $X_EXTRA_LIBS $LIBS"
 LIBS="$LIBS $SDL_LIBS"
 
 # Checks for header files.
