diff -Naur cups-2.2.8/conf/cups-files.conf.in cups-2.2.8-p/conf/cups-files.conf.in
--- cups-2.2.8/conf/cups-files.conf.in	2018-06-05 18:06:54.000000000 +0200
+++ cups-2.2.8-p/conf/cups-files.conf.in	2018-09-06 23:57:24.600440435 +0200
@@ -11,8 +11,8 @@
 
 # Default user and group for filters/backends/helper programs; this cannot be
 # any user or group that resolves to ID 0 for security reasons...
-#User @CUPS_USER@
-#Group @CUPS_GROUP@
+User @CUPS_USER@
+Group @CUPS_GROUP@
 
 # Administrator user group, used to match @SYSTEM in cupsd.conf policy rules...
 # This cannot contain the Group value for security reasons...
