diff -Naur cups-2.2.10/config.h.in cups-2.2.10-p/config.h.in
--- cups-2.2.10/config.h.in	2018-12-07 20:41:56.000000000 +0100
+++ cups-2.2.10-p/config.h.in	2019-01-17 01:16:02.432718196 +0100
@@ -126,7 +126,7 @@
 #define CUPS_SBINDIR	"/usr/sbin"
 #define CUPS_SERVERBIN	"/usr/lib/cups"
 #define CUPS_SERVERROOT	"/etc/cups"
-#define CUPS_STATEDIR	"/var/run/cups"
+#define CUPS_STATEDIR	"/run/cups"
 
 
 /*
diff -Naur cups-2.2.10/man/cups-files.conf.man.in cups-2.2.10-p/man/cups-files.conf.man.in
--- cups-2.2.10/man/cups-files.conf.man.in	2018-12-07 20:41:56.000000000 +0100
+++ cups-2.2.10-p/man/cups-files.conf.man.in	2019-01-17 01:16:02.417718117 +0100
@@ -205,7 +205,7 @@
 .TP 5
 \fBStateDir \fIdirectory\fR
 Specifies the directory to use for PID and local certificate files.
-The default is "/var/run/cups" or "/etc/cups" depending on the platform.
+The default is "/run/cups" or "/etc/cups" depending on the platform.
 .\"#SyncOnClose
 .TP 5
 \fBSyncOnClose Yes\fR
diff -Naur cups-2.2.10/scheduler/org.cups.cupsd.plist cups-2.2.10-p/scheduler/org.cups.cupsd.plist
--- cups-2.2.10/scheduler/org.cups.cupsd.plist	2018-12-07 20:41:56.000000000 +0100
+++ cups-2.2.10-p/scheduler/org.cups.cupsd.plist	2019-01-17 01:16:02.420718132 +0100
@@ -40,7 +40,7 @@
 				<key>SockPathMode</key>
 				<integer>49663</integer>
 				<key>SockPathName</key>
-				<string>/private/var/run/cupsd</string>
+				<string>/private/run/cupsd</string>
 			</dict>
 		</array>
 	</dict>
diff -Naur cups-2.2.10/scheduler/process.c cups-2.2.10-p/scheduler/process.c
--- cups-2.2.10/scheduler/process.c	2018-12-07 20:41:56.000000000 +0100
+++ cups-2.2.10-p/scheduler/process.c	2019-01-17 01:16:02.429718180 +0100
@@ -181,7 +181,7 @@
 	       " #\"^/private/var/lib/\""
 	       " #\"^/private/var/log/\""
 	       " #\"^/private/var/mysql/\""
-	       " #\"^/private/var/run/\""
+	       " #\"^/private/run/\""
 	       " #\"^/private/var/spool/\""
 	       " #\"^/Library/Application Support/\""
 	       " #\"^/Library/Caches/\""
@@ -226,7 +226,7 @@
                  " #\"^/private/var/lib$\""
                  " #\"^/private/var/log$\""
                  " #\"^/private/var/mysql$\""
-                 " #\"^/private/var/run$\""
+                 " #\"^/private/run$\""
                  " #\"^/private/var/spool$\""
                  " #\"^/private/var/tmp$\""
                  " #\"^/private/var/tmp/\""
@@ -315,7 +315,7 @@
   cupsFilePuts(fp, "(allow distributed-notification-post)\n");
   /* Allow outbound networking to local services */
   cupsFilePuts(fp, "(allow network-outbound"
-		   "\n       (regex #\"^/private/var/run/\" #\"^/private/tmp/\" #\"^/private/var/tmp/\")");
+		   "\n       (regex #\"^/private/run/\" #\"^/private/tmp/\" #\"^/private/var/tmp/\")");
   for (lis = (cupsd_listener_t *)cupsArrayFirst(Listeners);
        lis;
        lis = (cupsd_listener_t *)cupsArrayNext(Listeners))
diff -Naur cups-2.2.10/xcode/config.h cups-2.2.10-p/xcode/config.h
--- cups-2.2.10/xcode/config.h	2018-12-07 20:41:56.000000000 +0100
+++ cups-2.2.10-p/xcode/config.h	2019-01-17 01:16:02.414718101 +0100
@@ -102,7 +102,7 @@
  * Do we have domain socket support, and if so what is the default one?
  */
 
-#define CUPS_DEFAULT_DOMAINSOCKET "/private/var/run/cupsd"
+#define CUPS_DEFAULT_DOMAINSOCKET "/private/run/cupsd"
 
 
 /*
