diff -urN virt-manager-0.2.6.orig/src/virtManager/remote.py virt-manager-0.2.6/src/virtManager/remote.py
--- virt-manager-0.2.6.orig/src/virtManager/remote.py	2006-11-16 20:59:26.000000000 +0000
+++ virt-manager-0.2.6/src/virtManager/remote.py	2006-11-16 20:59:56.000000000 +0000
@@ -25,23 +25,23 @@
 
         self.engine = engine
 
-    @dbus.service.method("com.redhat.virt.manager", in_signature="s")
+    @dbus.service.method("com.redhat.virt.manager")
     def show_domain_creator(self, uri):
         self.engine.show_create(uri)
 
-    @dbus.service.method("com.redhat.virt.manager", in_signature="ss")
+    @dbus.service.method("com.redhat.virt.manager")
     def show_domain_editor(self, uri, uuid):
         self.engine.show_details_config(uri, uuid)
 
-    @dbus.service.method("com.redhat.virt.manager", in_signature="ss")
+    @dbus.service.method("com.redhat.virt.manager")
     def show_domain_performance(self, uri, uuid):
         self.engine.show_details_performance(uri, uuid)
 
-    @dbus.service.method("com.redhat.virt.manager", in_signature="ss")
+    @dbus.service.method("com.redhat.virt.manager")
     def show_domain_console(self, uri, uuid):
         self.engine.show_console(uri, uuid)
 
-    @dbus.service.method("com.redhat.virt.manager", in_signature="s")
+    @dbus.service.method("com.redhat.virt.manager")
     def show_host_summary(self, uri):
         self.engine.show_manager(uri)
 
