diff --git a/src/virt-manager.py.in b/src/virt-manager.py.in
index fefa1c3..c285dac 100755
--- a/src/virt-manager.py.in
+++ b/src/virt-manager.py.in
@@ -273,7 +273,7 @@ def main():
     # on as if all were fine. Ultimately bad stuff happens,
     # so lets catch it here & get the hell out...
     import warnings
-    warnings.filterwarnings('error', module='gtk')
+    warnings.filterwarnings('error', module='gtk', append=True)
     try:
         import gobject
 
