diff -Naur pth-2.0.4.orig/configure pth-2.0.4/configure
--- pth-2.0.4.orig/configure	2005-05-31 10:31:42.000000000 +0000
+++ pth-2.0.4/configure	2005-05-31 10:32:43.000000000 +0000
@@ -22117,7 +22117,7 @@
     -e '/,[ 	]*$/N' \
     -e 's/,[ 	]*\n[ 	]*/, /' \
     -e 'tjoin' |\
-egrep "[^a-zA-Z0-9_]accept[ 	]*\\($gpat\\)" | head -1 |\
+egrep "[^a-zA-Z0-9_]accept[ 	]*\\($gpat\\)" | head -n 1 |\
 sed -e "s/.*[^a-zA-Z0-9_]accept[ 	]*($spat).*/\\1/" \
     -e 's/(\*[a-zA-Z_][a-zA-Z_0-9]*)/(*)/' \
     -e 's/^[ 	]*//' -e 's/[ 	]*$//' \
@@ -22244,7 +22244,7 @@
     -e '/,[ 	]*$/N' \
     -e 's/,[ 	]*\n[ 	]*/, /' \
     -e 'tjoin' |\
-egrep "[^a-zA-Z0-9_]poll[ 	]*\\($gpat\\)" | head -1 |\
+egrep "[^a-zA-Z0-9_]poll[ 	]*\\($gpat\\)" | head -n 1 |\
 sed -e "s/.*[^a-zA-Z0-9_]poll[ 	]*($spat).*/\\1/" \
     -e 's/(\*[a-zA-Z_][a-zA-Z_0-9]*)/(*)/' \
     -e 's/^[ 	]*//' -e 's/[ 	]*$//' \
