--- a/layout/applications.menu
+++ b/layout/applications.menu
@@ -69,6 +69,17 @@
       </And>
     </Include>
   </Menu> <!-- End Education -->
+  
+  <!-- Frugalware -->
+  <Menu>
+    <Name>Frugalware</Name>
+    <Directory>Frugalware.directory</Directory>
+    <Include>
+      <And>
+        <Category>Frugalware</Category>
+      </And>
+    </Include>
+  </Menu> <!-- End Frugalware -->
 
   <!-- Games -->
   <Menu>
