diff -Naur gnome-menus-2.20.0.orig/desktop-directories/Hardware.directory.in gnome-menus-2.20.0/desktop-directories/Hardware.directory.in
--- gnome-menus-2.20.0.orig/desktop-directories/Hardware.directory.in	2007-09-22 00:28:45.000000000 +0200
+++ gnome-menus-2.20.0/desktop-directories/Hardware.directory.in	2007-09-22 00:30:05.000000000 +0200
@@ -1,6 +1,6 @@
 [Desktop Entry]
 _Name=Hardware
 _Comment=Settings for several hardware devices
-Icon=preferences-desktop-peripherals
+Icon=system
 Type=Directory
 Encoding=UTF-8
--- gnome-menus-2.24.1/layout/settings.menu	2008-10-22 11:15:08.000000000 +0200
+++ gnome-menus-2.24.1new/layout/settings.menu	2008-11-18 12:30:50.000000000 +0100
@@ -24,15 +24,96 @@
     <Directory>Settings.directory</Directory>
     <Include>
       <And>
-        <Category>Settings</Category>
+<Category>Settings</Category>
         <Not>
           <Or>
             <Category>System</Category>
+	   <Category>X-GNOME-PersonalSettings</Category><!-- delete category move to submenu Frugalware-->
+           <Category>DesktopSettings</Category><!-- delete category move to submenu Frugalware-->
+           <Category>X-GNOME-NetworkSettings</Category><!-- delete category move to submenu Frugalware-->
+           <Category>HardwareSettings</Category><!-- delete category move to submenu Frugalware-->
+           <Category>X-GNOME-SystemSettings</Category>	<!-- delete category move to submenu Frugalware-->
             <Filename>gnomecc.desktop</Filename>
           </Or>
         </Not>
       </And>
     </Include>
+
+<!-- Begin personal Submenu for Frugalware-->
+  <Menu>
+    <Name>Personal</Name>
+    <Directory>Personal.directory</Directory>
+    <Include>
+      <And>
+        <Category>Settings</Category>
+        <Category>X-GNOME-PersonalSettings</Category>
+      </And>
+    </Include>
+  </Menu>  <!-- End Personal -->
+
+  <!-- Look and Feel submenu -->
+  <Menu>
+    <Name>Look and Feel</Name>
+    <Directory>LookAndFeel.directory</Directory>
+    <Include>
+      <And>
+        <Category>Settings</Category>
+        <Category>DesktopSettings</Category>
+      </And>
+    </Include>
+    </Menu> <!-- End Look and Feel -->
+
+  <!-- Internet and Network submenu -->
+  <Menu>
+    <Name>Internet and Network</Name>
+    <Directory>InternetAndNetwork.directory</Directory>
+    <Include>
+      <And>
+        <Category>Settings</Category>
+        <Category>X-GNOME-NetworkSettings</Category>
+      </And>
+    </Include>
+  </Menu> <!-- End Internet and Network -->
+  
+  <!-- Hardware submenu -->
+  <Menu>
+    <Name>Hardware</Name>
+    <Directory>Hardware.directory</Directory>
+    <Include>
+      <And>
+        <Category>Settings</Category>
+        <Category>HardwareSettings</Category>
+      </And>
+    </Include>
+  </Menu> <!-- End Hardware -->
+
+  <!-- System submenu -->
+  <Menu>
+    <Name>System</Name>
+    <Directory>System.directory</Directory>
+    <Include>
+      <And>
+        <Category>Settings</Category>
+        <Or>
+          <Category>X-GNOME-SystemSettings</Category>
+          <And>
+            <Category>System</Category>
+            <Not>
+              <Or>
+                <Category>X-GNOME-PersonalSettings</Category>
+                <Category>DesktopSettings</Category>
+                <Category>HardwareSettings</Category>
+                <Category>X-GNOME-NetworkSettings</Category> 
+              </Or>
+            </Not>
+          </And>
+        </Or>
+      </And>
+    </Include>
+  </Menu> 
+<!-- End System for Frugalware-->
+
+
   </Menu>
 
   <!-- System Settings -->
@@ -45,6 +126,7 @@
         <Category>System</Category>
       </And>
     </Include>
+	
   </Menu>     <!-- End System Settings -->
 
   <Layout>
@@ -54,4 +136,6 @@
     <Merge type="files"/>
   </Layout>
 
+
+
 </Menu> <!-- End Settings -->

