diff -Naur gnome-menus-2.20.0.orig/desktop-directories/Hardware.directory.in gnome-menus-2.20.0/desktop-directories/Hardware.directory.in
--- gnome-menus-2.20.0.orig/desktop-directories/Hardware.directory.in	2007-09-22 00:28:45.000000000 +0200
+++ gnome-menus-2.20.0/desktop-directories/Hardware.directory.in	2007-09-22 00:30:05.000000000 +0200
@@ -1,6 +1,6 @@
 [Desktop Entry]
 _Name=Hardware
 _Comment=Settings for several hardware devices
-Icon=preferences-desktop-peripherals
+Icon=system
 Type=Directory
 Encoding=UTF-8
--- gnome-menus-2.24.0old/layout/settings.menu	2008-09-22 23:10:56.000000000 +0200
+++ gnome-menus-2.24.0/layout/settings.menu	2008-10-01 13:06:11.000000000 +0200
@@ -28,6 +28,11 @@
         <Not>
           <Or>
             <Category>System</Category>
+	    <Category>X-GNOME-PersonalSettings</Category>
+           <Category>DesktopSettings</Category>
+           <Category>X-GNOME-NetworkSettings</Category>
+           <Category>HardwareSettings</Category>
+           <Category>X-GNOME-SystemSettings</Category>	
             <Filename>gnomecc.desktop</Filename>
           </Or>
         </Not>
@@ -47,6 +52,81 @@
     </Include>
   </Menu>     <!-- End System Settings -->
 
+  <!-- Personal Submenu -->
+  <Menu>
+    <Name>Personal</Name>
+    <Directory>Personal.directory</Directory>
+    <Include>
+      <And>
+        <Category>Settings</Category>
+        <Category>X-GNOME-PersonalSettings</Category>
+      </And>
+    </Include>
+  </Menu>  <!-- End Personal -->
+
+   <Menu>
+    <Name>Look and Feel</Name>
+    <Directory>LookAndFeel.directory</Directory>
+     <Include>
+       <And>
+         <Category>Settings</Category>
+         <Category>DesktopSettings</Category>
+      </And>
+     </Include>
+  </Menu>
+
+  <!-- Internet and Network submenu -->
+  <Menu>
+    <Name>Internet and Network</Name>
+    <Directory>InternetAndNetwork.directory</Directory>
+    <Include>
+      <And>
+        <Category>Settings</Category>
+        <Category>X-GNOME-NetworkSettings</Category>
+      </And>
+    </Include>
+  </Menu> <!-- End Internet and Network -->
+  
+  <!-- Hardware submenu -->
+  <Menu>
+    <Name>Hardware</Name>
+    <Directory>Hardware.directory</Directory>
+    <Include>
+      <And>
+        <Category>Settings</Category>
+        <Category>HardwareSettings</Category>
+      </And>
+    </Include>
+  </Menu> <!-- End Hardware -->
+
+  <!-- System submenu -->
+  <Menu>
+    <Name>System</Name>
+    <Directory>System.directory</Directory>
+    <Include>
+      <And>
+        <Category>Settings</Category>
+        <Or>
+          <Category>X-GNOME-SystemSettings</Category>
+          <And>
+            <Category>System</Category>
+            <Not>
+              <Or>
+                <Category>X-GNOME-PersonalSettings</Category>
+                <Category>DesktopSettings</Category>
+                <Category>HardwareSettings</Category>
+                <Category>X-GNOME-NetworkSettings</Category> 
+              </Or>
+            </Not>
+          </And>
+        </Or>
+      </And>
+    </Include>
+  </Menu> <!-- End System -->
+
+
+
+
   <Layout>
     <Menuname>Preferences</Menuname>
     <Menuname>Administration</Menuname>
