diff -Naur e1000e-3.6.0/src/netdev.c e1000e-3.6.0-p/src/netdev.c
--- e1000e-3.6.0/src/netdev.c	2019-09-09 20:28:11.384088231 +0200
+++ e1000e-3.6.0-p/src/netdev.c	2020-03-31 00:20:58.683967144 +0200
@@ -6675,7 +6675,13 @@
  * e1000_tx_timeout - Respond to a Tx Hang
  * @netdev: network interface device structure
  **/
+
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5,6,0)
 static void e1000_tx_timeout(struct net_device *netdev)
+#else
+static void e1000_tx_timeout(struct net_device *netdev, unsigned int txqueue)
+#endif
+
 {
 	struct e1000_adapter *adapter = netdev_priv(netdev);
 
