--- e1000e-3.4.2.4/src/netdev.c	2019-03-17 13:08:40.000000000 +0100
+++ e1000e-3.4.2.4.new/src/netdev.c	2019-07-16 11:21:42.410947150 +0200
@@ -6623,7 +6623,7 @@
 						  adapter->tx_fifo_limit) + 2));
 
 #ifdef HAVE_SKB_XMIT_MORE
-		if (!skb->xmit_more ||
+		if (!netdev_xmit_more() ||
 		    netif_xmit_stopped(netdev_get_tx_queue(netdev, 0))) {
 			if (adapter->flags2 & FLAG2_PCIM2PCI_ARBITER_WA)
 				e1000e_update_tdt_wa(tx_ring,
