diff -Naur e1000e-3.3.5.3/src/netdev.c e1000e-3.3.5.3-p/src/netdev.c
--- e1000e-3.3.5.3/src/netdev.c	2016-11-09 17:39:01.000000000 +0100
+++ e1000e-3.3.5.3-p/src/netdev.c	2017-03-03 22:27:21.345067617 +0100
@@ -7665,19 +7665,19 @@
 
 		vector = 0;
 		msix_irq = adapter->msix_entries[vector].vector;
-		disable_irq(msix_irq);
+		disable_hardirq(msix_irq);
 		e1000_intr_msix_rx(msix_irq, netdev);
 		enable_irq(msix_irq);
 
 		vector++;
 		msix_irq = adapter->msix_entries[vector].vector;
-		disable_irq(msix_irq);
+		disable_hardirq(msix_irq);
 		e1000_intr_msix_tx(msix_irq, netdev);
 		enable_irq(msix_irq);
 
 		vector++;
 		msix_irq = adapter->msix_entries[vector].vector;
-		disable_irq(msix_irq);
+		disable_hardirq(msix_irq);
 		e1000_msix_other(msix_irq, netdev);
 		enable_irq(msix_irq);
 	}
@@ -7702,13 +7702,13 @@
 		e1000_intr_msix(adapter->pdev->irq, netdev);
 		break;
 	case E1000E_INT_MODE_MSI:
-		disable_irq(adapter->pdev->irq);
-		e1000_intr_msi(adapter->pdev->irq, netdev);
+		if (disable_hardirq(adapter->pdev->irq))
+			e1000_intr_msi(adapter->pdev->irq, netdev);
 		enable_irq(adapter->pdev->irq);
 		break;
 	default:		/* E1000E_INT_MODE_LEGACY */
-		disable_irq(adapter->pdev->irq);
-		e1000_intr(adapter->pdev->irq, netdev);
+		if (disable_hardirq(adapter->pdev->irq))
+			 e1000_intr(adapter->pdev->irq, netdev);
 		enable_irq(adapter->pdev->irq);
 		break;
 	}
