--- bluemindo-0.2.1/Makefile.old	2008-06-22 15:06:14.000000000 +0200
+++ bluemindo-0.2.1/Makefile	2008-06-22 21:39:02.000000000 +0200
@@ -11,8 +11,8 @@
 	install -m644 data/glade/*.glade $(DATADIR)/bluemindo/glade
 	install -m666 COPYING $(DATADIR)/bluemindo
 
-	install -m644 data/misc/Bluemindo.desktop $(DATADIR)/applications
-	install -m644 data/misc/bluemindo.png $(DATADIR)/pixmaps
+	install -m644 data/misc/Bluemindo.desktop $(DATADIR)/applications/Bluemindo.desktop
+	install -m644 data/misc/bluemindo.png $(DATADIR)/pixmaps/bluemindo.png
 	install -m755 data/misc/bluemindo $(BIN)
 
 	for sourcedir in `find src/ -type d | grep -v '.svn' | grep -v '.pyc' | sed 's:src/::g'` ; do \
