--- build.xml	2007-11-13 20:59:00.000000000 +0200
+++ build.xml.fixed	2008-06-12 22:53:57.000000000 +0200
@@ -9,7 +9,7 @@
 	<property name="compiler.generate.no.warnings" value="off"/>
 	<property name="compiler.args" value=""/>
 	<property name="compiler.max.memory" value="128m"/>
-	<property name="compiler.verbose" value="true"/>
+	<property name="compiler.verbose" value="false"/>
 	
 	<!-- Ignore these always -->
 	<patternset id="ignored.files">
@@ -165,7 +165,7 @@
 		</jar>
 	</target>
 	
-	<target name="compile" depends="compile.production,compile.tests,run.tests" description="Compile all jaudiotagger code"/>
+	<target name="compile" depends="compile.production" description="Compile all jaudiotagger code"/>
 
 	<target name="compile.production" description="Compile production classes">
 	  
@@ -197,13 +197,6 @@
 		<!--Delete old file -->
 		<delete file="cobertura.ser" />
 		
-		<!--Instrument the classes for code coverage -->
-		<cobertura-instrument todir="${instrumented.dir}">
-	    <ignore regex="org.apache.log4j.*" />
-	    <fileset dir="${output.dir}">
-	        <include name="**/*.class" />	        
-	    </fileset>	    
-    </cobertura-instrument>
 		
 		
 	</target>
@@ -264,8 +257,6 @@
 			</fileset>
   			<report format="frames" todir="${junitreport.dir}"/>
 		</junitreport>
-		
-		<cobertura-report format="html" destdir="${coveragereport.dir}" srcdir="${src.dir}" />
 	</target>
 	
 
@@ -305,4 +296,4 @@
 	<!-- Build all Javadoc -->
 	<target name="build.javadoc" depends="build.user.javadoc,build.dev.javadoc" description="Build user and Developer javaDoc">
 	</target>
-</project>
\ Dosya sonunda yenisatır yok.
+</project>

