diff -Naur nethogs-0.8.7/Makefile nethogs-0.8.7.new/Makefile
--- nethogs-0.8.7/Makefile	2022-04-04 16:27:44.000000000 +0200
+++ nethogs-0.8.7.new/Makefile	2024-01-16 11:20:30.368118839 +0100
@@ -1,7 +1,7 @@
 export VERSION := $(shell ./determineVersion.sh)
 
 #export PREFIX := /usr
-export PREFIX ?= /usr/local
+export PREFIX ?= /usr
 
 all: decpcap_test test nethogs
 
diff -Naur nethogs-0.8.7/src/MakeApp.mk nethogs-0.8.7.new/src/MakeApp.mk
--- nethogs-0.8.7/src/MakeApp.mk	2022-04-04 16:27:44.000000000 +0200
+++ nethogs-0.8.7.new/src/MakeApp.mk	2024-01-16 11:20:30.361118789 +0100
@@ -1,4 +1,4 @@
-sbin := $(PREFIX)/sbin
+sbin := $(PREFIX)/bin
 
 all: nethogs decpcap_test
 
