diff -ur a/autogen.sh b/autogen.sh
--- a/autogen.sh	2010-03-31 10:13:52.000000000 -0500
+++ b/autogen.sh	2011-11-04 04:54:12.206340278 -0500
@@ -202,15 +202,15 @@
   GNU_OR_DIE=0
 }
 
-WGET=`find_command 'wget*'`
-($WGET --version) < /dev/null > /dev/null 2>&1 ||
-{
-  echo
-  echo "You must have wget installed to compile $progname."
-  echo "Download the appropriate package for your distribution, or get the"
-  echo "source tarball from ftp://ftp.gnu.org/pub/gnu/wget"
-  GNU_OR_DIE=0
-}
+#WGET=`find_command 'wget*'`
+#($WGET --version) < /dev/null > /dev/null 2>&1 ||
+#{
+#  echo
+#  echo "You must have wget installed to compile $progname."
+#  echo "Download the appropriate package for your distribution, or get the"
+#  echo "source tarball from ftp://ftp.gnu.org/pub/gnu/wget"
+#  GNU_OR_DIE=0
+#}
 
 if test "$GNU_OR_DIE" -eq 0; then
   exit 1
