--- hdparm-9.60/Makefile	2020-06-26 12:55:43.000000000 +0200
+++ hdparm-9.60.new/Makefile	2020-12-01 14:10:21.240080856 +0100
@@ -13,7 +13,7 @@
 CC ?= gcc
 STRIP ?= strip
 
-CFLAGS := -O2 -W -Wall -Wbad-function-cast -Wcast-align -Wpointer-arith -Wcast-qual -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -fkeep-inline-functions -Wwrite-strings -Waggregate-return -Wnested-externs -Wtrigraphs $(CFLAGS)
+CFLAGS += -W -Wall -Wbad-function-cast -Wcast-align -Wpointer-arith -Wcast-qual -Wshadow -Wstrict-prototypes -Wmissing-prototypes -Wmissing-declarations -fkeep-inline-functions -Wwrite-strings -Waggregate-return -Wnested-externs -Wtrigraphs
 
 INSTALL = install
 INSTALL_DATA = $(INSTALL) -m 644
@@ -23,13 +23,13 @@
 OBJS = hdparm.o identify.o sgio.o sysfs.o geom.o fallocate.o fibmap.o fwdownload.o dvdspeed.o wdidle3.o apt.o
 
 all:
-	$(MAKE) -j4 hdparm
+	$(MAKE) hdparm
 
 static:
 	LDFLAGS=-static $(MAKE)
 
 hdparm: Makefile hdparm.h sgio.h $(OBJS)
-	$(CC) $(LDFLAGS) -o hdparm $(OBJS)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o hdparm $(OBJS)
 	$(STRIP) hdparm
 
 hdparm.o:	hdparm.h sgio.h
