diff -up ufraw-0.15/dcraw.cc.gcc-4.4 ufraw-0.15/dcraw.cc
--- ufraw-0.15/dcraw.cc.gcc-4.4	2008-12-23 08:19:41.000000000 +0100
+++ ufraw-0.15/dcraw.cc	2009-03-02 15:56:58.724512142 +0100
@@ -8337,8 +8337,8 @@ int CLASS main (int argc, const char **a
   static int timestamp_only=0, thumbnail_only=0, identify_only=0;
   static int user_qual=-1, user_black=-1, user_sat=-1, user_flip=-1;
   static int use_fuji_rotate=1, write_to_stdout=0, quality, i, c;
-  static char opm, opt, *ofname, *cp, *bpfile=0;
-  static const char *sp, *dark_frame=0, *write_ext;
+  static char opm, opt, *ofname, *sp, *cp, *bpfile=0;
+  static const char *dark_frame=0, *write_ext;
   static struct utimbuf ut;
   static FILE *ofp;
 #ifndef NO_LCMS

