diff -aur a/plugins/zzogl-pg/opengl/ZZoglDrawing.cpp b/plugins/zzogl-pg/opengl/ZZoglDrawing.cpp
--- a/plugins/zzogl-pg/opengl/ZZoglDrawing.cpp	2011-04-06 02:13:40.000000000 +0000
+++ b/plugins/zzogl-pg/opengl/ZZoglDrawing.cpp	2011-10-11 15:04:55.362520104 +0000
@@ -41,14 +41,6 @@
 	curvb.SetCurrentPrim();
 }
 
-// return true if triangle SHOULD be painted.
-// Hackish and should be replaced.
-bool __forceinline NoHighlights(int i)
-{
-	//Old code
-	return (!(conf.settings().xenosaga_spec) || !vb[i].zbuf.zmsk || prim->iip) ;
-}
-
 // Not inlining for the moment to avoid getting 'unresolved external symbol' errors in Windows. 
 // This could also be resolved by moving the function into the header...
 void Kick::KickVertex(bool adc)
diff -aur a/plugins/zzogl-pg/opengl/ZZoglDrawing.h b/plugins/zzogl-pg/opengl/ZZoglDrawing.h
--- a/plugins/zzogl-pg/opengl/ZZoglDrawing.h	2011-04-06 02:13:40.000000000 +0000
+++ b/plugins/zzogl-pg/opengl/ZZoglDrawing.h	2011-10-11 15:05:22.642523792 +0000
@@ -23,7 +23,7 @@
 #include "Util.h"
 #include "GS.h"
 
-extern bool __forceinline NoHighlights(int i);
+#define NoHighlights(I) (!(conf.settings().xenosaga_spec) || !vb[(I)].zbuf.zmsk || prim->iip)
 
 enum PRIM_TYPE {
     PRIM_POINT = 0,
