diff -Naur xmule-1.13.6.old/src/ClientCredits.cpp xmule-1.13.6/src/ClientCredits.cpp
--- xmule-1.13.6.old/src/ClientCredits.cpp	2006-02-28 23:38:45.000000000 +0100
+++ xmule-1.13.6/src/ClientCredits.cpp	2006-05-15 12:25:41.000000000 +0200
@@ -116,7 +116,7 @@
 void CClientCredits::ShowCHandPK(uint32_t challenge, unsigned char * pk, int length)
 {
     std::ostringstream internalbuffer;
-    internalbuffer << "    credits=" << (uint32_t)this << "  challenge=" << challenge << "  length=" << length << "  publickey=";
+    internalbuffer << "    credits=" << (ulong)this << "  challenge=" << challenge << "  length=" << length << "  publickey=";
     for (int i = 0 ; i < length ; i++)
     {
         if (i < 5 || i > length - 6)
diff -Naur xmule-1.13.6.old/src/StatisticsDlg.cpp xmule-1.13.6/src/StatisticsDlg.cpp
--- xmule-1.13.6.old/src/StatisticsDlg.cpp	2006-02-28 23:38:45.000000000 +0100
+++ xmule-1.13.6/src/StatisticsDlg.cpp	2006-05-15 12:26:40.000000000 +0200
@@ -957,7 +957,7 @@
         stattree->Expand(item);
         temp = wxT("");
         
-        for (ix = 0 ; ix < 3 * (int) stattree->GetItemData(item); ++ix)
+        for (ix = 0 ; ix < 3 * (long) stattree->GetItemData(item); ++ix)
         {
             temp += wxT("&nbsp;");
         }
