diff -Naur jed-0.99-16.orig/src/Makefile.in jed-0.99-16/src/Makefile.in
--- jed-0.99-16.orig/src/Makefile.in	2002-10-20 08:55:30.000000000 +0200
+++ jed-0.99-16/src/Makefile.in	2004-08-08 23:17:50.615663288 +0200
@@ -51,10 +51,10 @@
 #---------------------------------------------------------------------------
 # 1. Linux GPM Mouse support
 #    Uncomment next FOUR lines for GPM mouse support
-#MOUSEFLAGS = -DHAS_GPM_MOUSE
-#MOUSELIB = -lgpm
-#GPMMOUSEO = gpmmouse.o
-#OBJGPMMOUSEO = $(OBJDIR)/$(GPMMOUSEO)
+MOUSEFLAGS = -DHAS_GPM_MOUSE
+MOUSELIB = -lgpm
+GPMMOUSEO = gpmmouse.o
+OBJGPMMOUSEO = $(OBJDIR)/$(GPMMOUSEO)
 
  
 # 2. XFree86 XRENDERFONT (Anti-aliased font) support for XJED
