diff -Naur jed-0.99-18/src/Makefile.in jed-0.99-18.gpm/src/Makefile.in
--- jed-0.99-18/src/Makefile.in	2006-02-16 04:09:50.000000000 +0100
+++ jed-0.99-18.gpm/src/Makefile.in	2006-04-03 18:35:27.000000000 +0200
@@ -52,10 +52,10 @@
 #---------------------------------------------------------------------------
 # 1. Linux GPM Mouse support
 #    Uncomment next FOUR lines for GPM mouse support
-#MOUSEFLAGS = -DUSE_GPM_MOUSE
-#MOUSELIB = -lgpm
-#GPMMOUSEO = gpmmouse.o
-#OBJGPMMOUSEO = $(OBJDIR)/$(GPMMOUSEO)
+MOUSEFLAGS = -DUSE_GPM_MOUSE
+MOUSELIB = -lgpm
+GPMMOUSEO = gpmmouse.o
+OBJGPMMOUSEO = $(OBJDIR)/$(GPMMOUSEO)
 
  
 # 2. XFree86 XRENDERFONT (Anti-aliased font) support for XJED
