--- mpd-0.20.21/src/decoder/plugins/FluidsynthDecoderPlugin.cxx	2017-01-27 08:46:51.000000000 +0100
+++ mpd-0.20.21.new/src/decoder/plugins/FluidsynthDecoderPlugin.cxx	2018-09-17 13:39:37.252539444 +0200
@@ -64,7 +64,7 @@
  * logging library.
  */
 static void
-fluidsynth_mpd_log_function(int level, char *message, gcc_unused void *data)
+fluidsynth_mpd_log_function(int level, const char *message, gcc_unused void *data)
 {
 	Log(fluidsynth_domain,
 	    fluidsynth_level_to_mpd(fluid_log_level(level)),
