diff -Naur pngnq-0.4.orig/Makefile pngnq-0.4/Makefile
--- pngnq-0.4.orig/Makefile	2006-06-14 21:37:53.659892960 +0200
+++ pngnq-0.4/Makefile	2006-06-14 21:39:18.708963536 +0200
@@ -2,13 +2,13 @@
 
 CC=gcc
 
-PNGINC = ../libpng
+PNGINC = /usr/include
 PNGLIB = /usr/lib
 
-ZINC = ../zlib
-ZLIB = ../zlib
+ZINC = /usr/include
+ZLIB = /usr/lib
 
-CFLAGS = -O3 -g -Wall -pedantic -I. -I$(PNGINC) -I$(ZINC) -funroll-loops -fomit-frame-pointer
+CFLAGS += -Wall -pedantic -I. -I$(PNGINC) -I$(ZINC) -funroll-loops -fomit-frame-pointer
 
 LDFLAGS = -L$(PNGLIB) -lpng -L$(ZLIB) -lz -lm
 #LDFLAGS = $(PNGLIB)/libpng.a -L$(ZLIB) -lz -lm
@@ -26,3 +26,6 @@
 
 clean:
 	rm -f pngnq pngcomp *.o
+
+install:
+	/usr/bin/install -c -g root -o root -m 755 pngnq $(DESTDIR)/usr/bin
