diff -Naur 0.5.orig/Makefile 0.5/Makefile
--- 0.5.orig/Makefile	2007-11-11 17:33:55.000000000 +0100
+++ 0.5/Makefile	2007-11-11 17:35:04.000000000 +0100
@@ -5,13 +5,13 @@
 
 DISTFILES = LICENSE README pngnq.1 pngnq pngcomp 
 
-PNGINC = ../libpng
+PNGINC = /usr/include
 PNGLIB = /usr/lib
 
-ZINC = ../zlib
-ZLIB = ../zlib
+ZINC = /usr/include
+ZLIB = /usr/lib
 
-CFLAGS = -O3 -g -Wall -pedantic -I. -I$(PNGINC) -I$(ZINC) -funroll-loops -fomit-frame-pointer -std=c99
+CFLAGS += -Wall -pedantic -I. -I$(PNGINC) -I$(ZINC) -funroll-loops -fomit-frame-pointer -std=c99
 
 LDFLAGS = -L$(PNGLIB) -lpng -L$(ZLIB) -lz -lm
 #LDFLAGS = $(PNGLIB)/libpng.a -L$(ZLIB) -lz -lm
@@ -41,3 +41,5 @@
 bindist: all
 	tar -cz -f ../pngnq-$(VERSION)-$(ARCH).tar.gz $(DISTFILES)
 
+install:
+	/usr/bin/install -c -g root -o root -m 755 pngnq $(DESTDIR)/usr/bin
