diff -Naur 0.5/Makefile 0.5-p/Makefile
--- 0.5/Makefile	2007-11-14 22:03:03.000000000 +0100
+++ 0.5-p/Makefile	2008-05-17 19:02:34.000000000 +0200
@@ -27,15 +27,15 @@
 # Change these to suit your system's requirements
 BINDIR=/usr/bin
 MANDIR=/usr/share/man/man1
-DOCDIR=/usr/share/doc/pngnq
+DOCDIR=/usr/share/doc/pngnq-$(VERSION)
 
 # Build options
 PNGINC = ${shell libpng-config --cflags}
 PNGLIB = ${shell libpng-config --ldflags}
 
-CFLAGS = -O3 -g -Wall -pedantic -I. -D VERSION=\"$(VERSION)\" $(PNGINC) -funroll-loops -fomit-frame-pointer 
+CFLAGS +=  -g -Wall -pedantic -I. -D VERSION=\"$(VERSION)\" $(PNGINC) -funroll-loops -fomit-frame-pointer 
 
-LDFLAGS = $(PNGLIB)
+LDFLAGS = $(PNGLIB) 
 
 OBJS = pngnq.o rwpng.o neuquant32.o freegetopt/getopt.o
 #############################################
@@ -72,7 +72,9 @@
 
 # Install
 install: all
-	install -t ${BINDIR} pngnq pngcomp
-	install -t ${MANDIR} pngnq.1 
-	install -d ${DOCDIR} 
-	install -t ${DOCDIR} README LICENSE README.pngcomp 
+	install -d $(DESTDIR)${MANDIR}
+	install -d $(DESTDIR)${DOCDIR}
+	install -d $(DESTDIR)${BINDIR}
+	install -t $(DESTDIR)${BINDIR} pngnq pngcomp
+	install -t $(DESTDIR)${MANDIR} pngnq.1 
+	install -t $(DESTDIR)${DOCDIR} README LICENSE README.pngcomp 
