--- rdma-core-49.0/CMakeLists.txt	2023-11-12 15:16:56.000000000 +0100
+++ rdma-core-49.0.new/CMakeLists.txt	2024-01-09 21:56:36.813709783 +0100
@@ -544,7 +544,7 @@
 # drm headers
 
 # Check if the headers have been installed by kernel-headers
-find_path(DRM_INCLUDE_DIRS "drm.h" PATH_SUFFIXES "drm")
+find_path(DRM_INCLUDE_DIRS "drm.h" PATH_SUFFIXES "libdrm")
 
 # Alternatively the headers could have been installed by libdrm
 if (NOT DRM_INCLUDE_DIRS)
