diff -Naur gtkwave-3.0.28/configure gtkwave-3.0.28.new/configure
--- gtkwave-3.0.28/configure	2007-04-20 06:44:14.000000000 +0530
+++ gtkwave-3.0.28.new/configure	2007-06-06 23:32:12.000000000 +0530
@@ -13,8 +13,8 @@
 	rm -f contrib/rtlbrowse/Makefile.in
 
 	X=""
-	echo "Build GTKWave for GTK+-1.x or 2.x? Please enter '1' or '2'"
-	read X
+	# Build for GTK2 Please enter '1' or '2'"
+	X="2"
 
 	if [ "$X" = "" ]
 		then X="2"
@@ -48,8 +48,8 @@
 # in case of automatic build, default to gtk+-1.x...
 
 X=""
-echo "Build GTKWave for GTK+-1.x or 2.x? Please enter '1' or '2'"
-read X
+# Build for GTK2 Please enter '1' or '2'"
+X="2"
 
 if [ "$X" = "" ]
 	then X="1"
diff -Naur gtkwave-3.0.28/contrib/pccts/antlr/Makefile.in gtkwave-3.0.28.new/contrib/pccts/antlr/Makefile.in
--- gtkwave-3.0.28/contrib/pccts/antlr/Makefile.in	2007-04-20 06:45:07.000000000 +0530
+++ gtkwave-3.0.28.new/contrib/pccts/antlr/Makefile.in	2007-06-06 23:33:20.000000000 +0530
@@ -170,7 +170,7 @@
 DLG=../bin/dlg
 OBJ_EXT=o
 OUT_OBJ = -o
-CFLAGS= $(COPT) -I. -I$(SET) -I$(PCCTS_H) -DUSER_ZZSYN $(COTHER) -DZZLEXBUFSIZE=32000
+CFLAGS+= -I. -I$(SET) -I$(PCCTS_H) -DUSER_ZZSYN $(COTHER) -DZZLEXBUFSIZE=32000
 #
 # SGI Users, use this CFLAGS
 #
@@ -179,7 +179,7 @@
         globals.o hash.o lex.o main.o misc.o set.o pred.o egman.o mrhoist.o fcache.o
 
 antlr : $(OBJ) $(SRC)
-		$(CC) $(CFLAGS) -o antlr $(OBJ)
+		$(CC) $(CFLAGS) $(LDFLAGS) -o antlr $(OBJ)
 		mv antlr ../bin/.
 
 # what files does PCCTS generate (both ANTLR and DLG)
diff -Naur gtkwave-3.0.28/contrib/pccts/dlg/Makefile.in gtkwave-3.0.28.new/contrib/pccts/dlg/Makefile.in
--- gtkwave-3.0.28/contrib/pccts/dlg/Makefile.in	2007-04-20 06:44:47.000000000 +0530
+++ gtkwave-3.0.28.new/contrib/pccts/dlg/Makefile.in	2007-06-06 23:33:20.000000000 +0530
@@ -118,14 +118,14 @@
 COPT=-O
 ANTLR=../bin/antlr
 DLG=../bin/dlg
-CFLAGS= $(COPT) -I. -I$(SET) -I$(PCCTS_H) -DUSER_ZZSYN -DZZLEXBUFSIZE=32000
+CFLAGS+= -I. -I$(SET) -I$(PCCTS_H) -DUSER_ZZSYN -DZZLEXBUFSIZE=32000
 OBJ_EXT=o
 OUT_OBJ = -o
 OBJ = dlg_p.o dlg_a.o main.o err.o set.o support.o output.o \
         relabel.o automata.o
 
 dlg : $(OBJ) $(SRC)
-		$(CC) $(CFLAGS) -o dlg $(OBJ)
+		$(CC) $(CFLAGS) $(LDFLAGS) -o dlg $(OBJ)
 		mv dlg ../bin/.
 
 SRC = dlg_p.c dlg_a.c main.c err.c $(SET)/set.c support.c output.c \
diff -Naur gtkwave-3.0.28/contrib/pccts/sorcerer/Makefile.in gtkwave-3.0.28.new/contrib/pccts/sorcerer/Makefile.in
--- gtkwave-3.0.28/contrib/pccts/sorcerer/Makefile.in	2007-04-20 06:44:37.000000000 +0530
+++ gtkwave-3.0.28.new/contrib/pccts/sorcerer/Makefile.in	2007-06-06 23:33:27.000000000 +0530
@@ -11,7 +11,7 @@
 
 CC=@CC@
 COPT=-g
-CFLAGS = $(COPT) -Ih -I. -I$(PCCTS_H) -I$(SET) -DZZLEXBUFSIZE=32000
+CFLAGS += -Ih -I. -I$(PCCTS_H) -I$(SET) -DZZLEXBUFSIZE=32000
 AFLAGS = -gh -k 2 -gt
 DFLAGS = -C2
 GRM = sor.g
@@ -25,7 +25,7 @@
 	(cd ./lib; $(MAKE) CC="$(CC)" COPT="$(COPT)")
 
 sor: $(OBJ) $(SRC)
-	$(CC) -o sor $(CFLAGS) $(OBJ)
+	$(CC) $(LDFLAGS) -o sor $(CFLAGS) $(OBJ)
 	mv sor $(BIN)/.
 
 sor.o : sor.c mode.h tokens.h
diff -Naur gtkwave-3.0.28/contrib/pccts/support/genmk/Makefile.in gtkwave-3.0.28.new/contrib/pccts/support/genmk/Makefile.in
--- gtkwave-3.0.28/contrib/pccts/support/genmk/Makefile.in	2007-04-20 06:44:43.000000000 +0530
+++ gtkwave-3.0.28.new/contrib/pccts/support/genmk/Makefile.in	2007-06-06 23:33:27.000000000 +0530
@@ -10,11 +10,11 @@
 CC=@CC@
 COPT=-O
 #CFLAGS=-I../../h -DPC
-CFLAGS=$(COPT) -I../../h
+CFLAGS += -I../../h
 BAG=../../bin/bag
 
 genmk: $(OBJ) $(SRC) ../../h/pcctscfg.h
-	$(CC) -o genmk $(OBJ)
+	$(CC) $(LDFLAGS) -o genmk $(OBJ)
 
 clean:
 	rm -rf core *.o
diff -Naur gtkwave-3.0.28/contrib/rtlbrowse/Makefile_GTK1.in gtkwave-3.0.28.new/contrib/rtlbrowse/Makefile_GTK1.in
--- gtkwave-3.0.28/contrib/rtlbrowse/Makefile_GTK1.in	2007-04-20 06:44:16.000000000 +0530
+++ gtkwave-3.0.28.new/contrib/rtlbrowse/Makefile_GTK1.in	2007-06-06 23:32:39.000000000 +0530
@@ -3,7 +3,7 @@
 
 AIXFLAGS = -bmaxdata:0xd0000000/dsa -D_WAVE_BE32
 BIGFILES = -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
-CFLAGS = -O2 `gtk-config --cflags` -I../../src/libz -I../../src/libbz2 -I../../src/helpers -I../../ -I../../src/
+CFLAGS += `gtk-config --cflags` -I../../src/libz -I../../src/libbz2 -I../../src/helpers -I../../ -I../../src/
 LDFLAGS= `gtk-config --libs`
 DEBUGS =
 AET2LIB =
diff -Naur gtkwave-3.0.28/contrib/rtlbrowse/Makefile_GTK2.in gtkwave-3.0.28.new/contrib/rtlbrowse/Makefile_GTK2.in
--- gtkwave-3.0.28/contrib/rtlbrowse/Makefile_GTK2.in	2007-04-20 06:44:16.000000000 +0530
+++ gtkwave-3.0.28.new/contrib/rtlbrowse/Makefile_GTK2.in	2007-06-06 23:33:13.000000000 +0530
@@ -3,9 +3,9 @@
 
 AIXFLAGS = -bmaxdata:0xd0000000/dsa -D_WAVE_BE32
 BIGFILES = -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
-CFLAGS = -O2 -DWAVE_USE_GTK2 `pkg-config gtk+-2.0 --cflags` -I../../src/libz -I../../src/libbz2 -I../../src/helpers -I../../ -I../../src/
+CFLAGS += -DWAVE_USE_GTK2 $(shell pkg-config gtk+-2.0 --cflags) -I../../src/libz -I../../src/libbz2 -I../../src/helpers -I../../ -I../../src/
 OBSOLETE_CFLAGS = -DGTK_ENABLE_BROKEN
-LDFLAGS= `pkg-config gtk+-2.0 --libs`
+LDFLAGS+= $(shell pkg-config gtk+-2.0 --libs)
 DEBUGS =
 AET2LIB =
 PTHLIB = 
diff -Naur gtkwave-3.0.28/contrib/vermin/Makefile.in gtkwave-3.0.28.new/contrib/vermin/Makefile.in
--- gtkwave-3.0.28/contrib/vermin/Makefile.in	2007-04-22 02:38:51.000000000 +0530
+++ gtkwave-3.0.28.new/contrib/vermin/Makefile.in	2007-06-06 23:33:27.000000000 +0530
@@ -22,7 +22,7 @@
 BIN = $(PCCTS)/bin
 ANTLR = $(BIN)/antlr
 DLG = $(BIN)/dlg
-CFLAGS = -O2 -I. -I$(ANTLR_H) $(COTHER)
+CFLAGS += -I. -I$(ANTLR_H) $(COTHER)
 AFLAGS = -ga -k 2 -gl
 DFLAGS = -ga -C2 -i
 GRM = verilog.g
@@ -38,7 +38,7 @@
 	cp vermin ../../bin/vermin
 
 vermin : $(SRC) $(OBJ)
-	$(CC) -o vermin $(CFLAGS) $(OBJ) -lm
+	$(CC) $(LDFLAGS) -o vermin $(CFLAGS) $(OBJ) -lm
 
 verilog.o : $(MOD_FILE) $(TOKENS) verilog.c
 	$(CC) -c $(CFLAGS) -o verilog.o verilog.c
diff -Naur gtkwave-3.0.28/Makefile.in gtkwave-3.0.28.new/Makefile.in
--- gtkwave-3.0.28/Makefile.in	2007-04-28 08:08:41.000000000 +0530
+++ gtkwave-3.0.28.new/Makefile.in	2007-06-06 23:32:31.000000000 +0530
@@ -17,45 +17,45 @@
 
 install:
 	cd bin;\
-	install -d $(bindir);\
-	install -m 0755 -c gtkwave  $(bindir)/gtkwave;\
-	install -m 0755 -c twinwave  $(bindir)/twinwave;\
-	install -m 0755 -c mvl2vcd  $(bindir)/mvl2vcd;\
-	install -m 0755 -c mvl2lxt  $(bindir)/mvl2lxt;\
-	install -m 0755 -c tex2vcd  $(bindir)/tex2vcd;\
-	install -m 0755 -c vcd2lxt  $(bindir)/vcd2lxt;\
-	install -m 0755 -c vcd2lxt2 $(bindir)/vcd2lxt2;\
-	install -m 0755 -c lxt2vcd  $(bindir)/lxt2vcd;\
-	install -m 0755 -c vcd2vzt  $(bindir)/vcd2vzt;\
-	install -m 0755 -c vzt2vcd  $(bindir)/vzt2vcd;\
-	install -m 0755 -c vztminer $(bindir)/vztminer;\
-	install -m 0755 -c lxt2miner $(bindir)/lxt2miner;\
-	install -m 0755 -c tla2vcd  $(bindir)/tla2vcd;\
-	install -m 0755 -c ghwdump  $(bindir)/ghwdump;\
-	install -m 0755 -c rtlbrowse  $(bindir)/rtlbrowse;\
-	install -m 0755 -c vermin  $(bindir)/vermin;\
-	install -m 0755 -c shmidcat  $(bindir)/shmidcat;\
+	install -d $(DESTDIR)/$(bindir);\
+	install -m 0755 -c gtkwave  $(DESTDIR)/$(bindir)/gtkwave;\
+	install -m 0755 -c twinwave  $(DESTDIR)/$(bindir)/twinwave;\
+	install -m 0755 -c mvl2vcd  $(DESTDIR)/$(bindir)/mvl2vcd;\
+	install -m 0755 -c mvl2lxt  $(DESTDIR)/$(bindir)/mvl2lxt;\
+	install -m 0755 -c tex2vcd  $(DESTDIR)/$(bindir)/tex2vcd;\
+	install -m 0755 -c vcd2lxt  $(DESTDIR)/$(bindir)/vcd2lxt;\
+	install -m 0755 -c vcd2lxt2 $(DESTDIR)/$(bindir)/vcd2lxt2;\
+	install -m 0755 -c lxt2vcd  $(DESTDIR)/$(bindir)/lxt2vcd;\
+	install -m 0755 -c vcd2vzt  $(DESTDIR)/$(bindir)/vcd2vzt;\
+	install -m 0755 -c vzt2vcd  $(DESTDIR)/$(bindir)/vzt2vcd;\
+	install -m 0755 -c vztminer $(DESTDIR)/$(bindir)/vztminer;\
+	install -m 0755 -c lxt2miner $(DESTDIR)/$(bindir)/lxt2miner;\
+	install -m 0755 -c tla2vcd  $(DESTDIR)/$(bindir)/tla2vcd;\
+	install -m 0755 -c ghwdump  $(DESTDIR)/$(bindir)/ghwdump;\
+	install -m 0755 -c rtlbrowse  $(DESTDIR)/$(bindir)/rtlbrowse;\
+	install -m 0755 -c vermin  $(DESTDIR)/$(bindir)/vermin;\
+	install -m 0755 -c shmidcat  $(DESTDIR)/$(bindir)/shmidcat;\
 	cd ../man;\
-	install -d $(mandir)/man1;\
-	install -m 0644 -c gtkwave.1 $(mandir)/man1/gtkwave.1;\
-	install -m 0644 -c twinwave.1 $(mandir)/man1/twinwave.1;\
-	install -m 0644 -c lxt2vcd.1 $(mandir)/man1/lxt2vcd.1;\
-	install -m 0644 -c mvl2lxt.1 $(mandir)/man1/mvl2lxt.1;\
-	install -m 0644 -c mvl2vcd.1 $(mandir)/man1/mvl2vcd.1;\
-	install -m 0644 -c tex2vcd.1 $(mandir)/man1/tex2vcd.1;\
-	install -m 0644 -c vcd2lxt.1 $(mandir)/man1/vcd2lxt.1;\
-	install -m 0644 -c vcd2lxt2.1 $(mandir)/man1/vcd2lxt2.1;\
-	install -m 0644 -c vcd2vzt.1 $(mandir)/man1/vcd2vzt.1;\
-	install -m 0644 -c vzt2vcd.1 $(mandir)/man1/vzt2vcd.1;\
-	install -m 0644 -c vztminer.1 $(mandir)/man1/vztminer.1;\
-	install -m 0644 -c lxt2miner.1 $(mandir)/man1/lxt2miner.1;\
-	install -m 0644 -c tla2vcd.1 $(mandir)/man1/tla2vcd.1;\
-	install -m 0644 -c ghwdump.1 $(mandir)/man1/ghwdump.1;\
-	install -m 0644 -c vermin.1 $(mandir)/man1/vermin.1;\
-	install -m 0644 -c shmidcat.1 $(mandir)/man1/shmidcat.1;\
-	install -m 0644 -c rtlbrowse.1 $(mandir)/man1/rtlbrowse.1;\
-	install -d $(mandir)/man5;\
-	install -m 0644 -c gtkwaverc.5 $(mandir)/man5/gtkwaverc.5
+	install -d $(DESTDIR)/$(mandir)/man1;\
+	install -m 0644 -c gtkwave.1 $(DESTDIR)/$(mandir)/man1/gtkwave.1;\
+	install -m 0644 -c twinwave.1 $(DESTDIR)/$(mandir)/man1/twinwave.1;\
+	install -m 0644 -c lxt2vcd.1 $(DESTDIR)/$(mandir)/man1/lxt2vcd.1;\
+	install -m 0644 -c mvl2lxt.1 $(DESTDIR)/$(mandir)/man1/mvl2lxt.1;\
+	install -m 0644 -c mvl2vcd.1 $(DESTDIR)/$(mandir)/man1/mvl2vcd.1;\
+	install -m 0644 -c tex2vcd.1 $(DESTDIR)/$(mandir)/man1/tex2vcd.1;\
+	install -m 0644 -c vcd2lxt.1 $(DESTDIR)/$(mandir)/man1/vcd2lxt.1;\
+	install -m 0644 -c vcd2lxt2.1 $(DESTDIR)/$(mandir)/man1/vcd2lxt2.1;\
+	install -m 0644 -c vcd2vzt.1 $(DESTDIR)/$(mandir)/man1/vcd2vzt.1;\
+	install -m 0644 -c vzt2vcd.1 $(DESTDIR)/$(mandir)/man1/vzt2vcd.1;\
+	install -m 0644 -c vztminer.1 $(DESTDIR)/$(mandir)/man1/vztminer.1;\
+	install -m 0644 -c lxt2miner.1 $(DESTDIR)/$(mandir)/man1/lxt2miner.1;\
+	install -m 0644 -c tla2vcd.1 $(DESTDIR)/$(mandir)/man1/tla2vcd.1;\
+	install -m 0644 -c ghwdump.1 $(DESTDIR)/$(mandir)/man1/ghwdump.1;\
+	install -m 0644 -c vermin.1 $(DESTDIR)/$(mandir)/man1/vermin.1;\
+	install -m 0644 -c shmidcat.1 $(DESTDIR)/$(mandir)/man1/shmidcat.1;\
+	install -m 0644 -c rtlbrowse.1 $(DESTDIR)/$(mandir)/man1/rtlbrowse.1;\
+	install -d $(DESTDIR)/$(mandir)/man5;\
+	install -m 0644 -c gtkwaverc.5 $(DESTDIR)/$(mandir)/man5/gtkwaverc.5
 
 
 dist:
diff -Naur gtkwave-3.0.28/src/helpers/Makefile.in gtkwave-3.0.28.new/src/helpers/Makefile.in
--- gtkwave-3.0.28/src/helpers/Makefile.in	2007-04-20 06:46:44.000000000 +0530
+++ gtkwave-3.0.28.new/src/helpers/Makefile.in	2007-06-06 23:33:43.000000000 +0530
@@ -1,7 +1,7 @@
 CC = @CC@
 BIGFILES = -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
 AIXFLAGS = -bmaxdata:0xd0000000/dsa
-CFLAGS = -O2 -I../libz -I../libbz2 -I../ -I../../
+CFLAGS += -I../libz -I../libbz2 -I../ -I../../
 DEBUGS =
 PTHLIB = 
 NSLLIB = 
@@ -22,10 +22,10 @@
 	cp vztminer ../../bin; cp lxt2miner ../../bin; cp tla2vcd ../../bin; cp ghwdump ../../bin; cp shmidcat ../../bin
 
 mvl2vcd: mvl2vcd.c
-	$(CC) $(CFLAGS) -o mvl2vcd mvl2vcd.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o mvl2vcd mvl2vcd.c
 
 shmidcat: shmidcat.c
-	$(CC) $(CFLAGS) -o shmidcat shmidcat.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o shmidcat shmidcat.c
 
 lxt_write.o: lxt_write.c lxt_write.h
 	$(CC) $(CFLAGS) -c lxt_write.c
@@ -47,37 +47,37 @@
 mvl2lxt.o: mvl2lxt.c 
 	$(CC) $(CFLAGS) -c mvl2lxt.c
 mvl2lxt: mvl2lxt.o lxt_write.o ../libz/libz.a ../libbz2/libbz2.a
-	$(CC) $(CFLAGS) -o mvl2lxt mvl2lxt.o lxt_write.o ../libz/libz.a ../libbz2/libbz2.a
+	$(CC) $(CFLAGS) $(LDFLAGS) -o mvl2lxt mvl2lxt.o lxt_write.o ../libz/libz.a ../libbz2/libbz2.a
 
 tex2vcd: tex2vcd.c
-	$(CC) $(CFLAGS) -o tex2vcd tex2vcd.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -o tex2vcd tex2vcd.c
 
 vcd2lxt: vcd2lxt.c lxt_write.o v2l_debug.o ../libz/libz.a ../libbz2/libbz2.a
-	$(CC) $(CFLAGS) -o vcd2lxt vcd2lxt.c lxt_write.o v2l_debug.o ../libz/libz.a ../libbz2/libbz2.a
+	$(CC) $(CFLAGS) $(LDFLAGS) -o vcd2lxt vcd2lxt.c lxt_write.o v2l_debug.o ../libz/libz.a ../libbz2/libbz2.a
 
 lxt2vcd: lxt2_read.o lxt2vcd.c scopenav.o ../libz/libz.a
-	$(CC) $(CFLAGS) -o lxt2vcd lxt2vcd.c lxt2_read.o scopenav.o ../libz/libz.a
+	$(CC) $(CFLAGS) $(LDFLAGS) -o lxt2vcd lxt2vcd.c lxt2_read.o scopenav.o ../libz/libz.a
 
 vcd2lxt2: vcd2lxt2.c lxt2_write.o v2l_debug_lxt2.o ../libz/libz.a
-	$(CC) $(CFLAGS) -o vcd2lxt2 vcd2lxt2.c lxt2_write.o v2l_debug_lxt2.o ../libz/libz.a
+	$(CC) $(CFLAGS) $(LDFLAGS) -o vcd2lxt2 vcd2lxt2.c lxt2_write.o v2l_debug_lxt2.o ../libz/libz.a
 
 vzt2vcd: vzt_read.o vzt2vcd.c scopenav.o ../libz/libz.a ../libbz2/libbz2.a
-	$(CC) $(CFLAGS) -o vzt2vcd vzt2vcd.c vzt_read.o scopenav.o ../libz/libz.a ../libbz2/libbz2.a $(PTHLIB) $(NSLLIB)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o vzt2vcd vzt2vcd.c vzt_read.o scopenav.o ../libz/libz.a ../libbz2/libbz2.a $(PTHLIB) $(NSLLIB)
 
 vcd2vzt: vcd2vzt.c vzt_write.o v2l_debug_lxt2.o ../libz/libz.a
-	$(CC) $(CFLAGS) -o vcd2vzt vcd2vzt.c vzt_write.o v2l_debug_lxt2.o ../libz/libz.a ../libbz2/libbz2.a $(NSLLIB)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o vcd2vzt vcd2vzt.c vzt_write.o v2l_debug_lxt2.o ../libz/libz.a ../libbz2/libbz2.a $(NSLLIB)
 
 vztminer: vztminer.c vzt_read.o ../libz/libz.a ../libbz2/libbz2.a
-	$(CC) $(CFLAGS) -o vztminer vztminer.c vzt_read.o ../libz/libz.a ../libbz2/libbz2.a $(PTHLIB) $(NSLLIB)
+	$(CC) $(CFLAGS) $(LDFLAGS) -o vztminer vztminer.c vzt_read.o ../libz/libz.a ../libbz2/libbz2.a $(PTHLIB) $(NSLLIB)
 
 lxt2miner: lxt2miner.c lxt2_read.o ../libz/libz.a
-	$(CC) $(CFLAGS) -o lxt2miner lxt2miner.c lxt2_read.o ../libz/libz.a
+	$(CC) $(CFLAGS) $(LDFLAGS) -o lxt2miner lxt2miner.c lxt2_read.o ../libz/libz.a
 
 tla2vcd: tla2vcd.c ../gnu_regex.c ../gnu_regex.h ../libz/libz.a
-	$(CC) $(CFLAGS) -o tla2vcd tla2vcd.c ../gnu_regex.c ../libz/libz.a
+	$(CC) $(CFLAGS) $(LDFLAGS) -o tla2vcd tla2vcd.c ../gnu_regex.c ../libz/libz.a
 
 ghwdump: ghwdump.c ../ghwlib.c
-	$(CC) $(CFLAGS) -I../ -o ghwdump ghwdump.c ../ghwlib.c 
+	$(CC) $(CFLAGS) -I../ $(LDFLAGS) $(LDFLAGS) -o ghwdump ghwdump.c ../ghwlib.c 
 
 ../libz/libz.a:
 	cd ../libz ; make
diff -Naur gtkwave-3.0.28/src/libbz2/Makefile.in gtkwave-3.0.28.new/src/libbz2/Makefile.in
--- gtkwave-3.0.28/src/libbz2/Makefile.in	2007-04-20 06:46:49.000000000 +0530
+++ gtkwave-3.0.28.new/src/libbz2/Makefile.in	2007-06-06 23:32:39.000000000 +0530
@@ -8,7 +8,7 @@
 
 # Suitably paranoid flags to avoid bugs in gcc-2.7
 BIGFILES = -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
-CFLAGS = -O2
+CFLAGS +=
 
 OBJS= blocksort.o  \
       huffman.o    \
diff -Naur gtkwave-3.0.28/src/libz/Makefile.in gtkwave-3.0.28.new/src/libz/Makefile.in
--- gtkwave-3.0.28/src/libz/Makefile.in	2007-04-20 06:46:30.000000000 +0530
+++ gtkwave-3.0.28.new/src/libz/Makefile.in	2007-06-06 23:32:39.000000000 +0530
@@ -8,7 +8,7 @@
 
 # Suitably paranoid flags to avoid bugs in gcc-2.7
 BIGFILES = -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
-CFLAGS = -O2 -I../../
+CFLAGS += -I../../
 
 OBJS= 	adler32.o \
 	compress.o \
diff -Naur gtkwave-3.0.28/src/Makefile_GTK2.in gtkwave-3.0.28.new/src/Makefile_GTK2.in
--- gtkwave-3.0.28/src/Makefile_GTK2.in	2007-04-20 06:45:49.000000000 +0530
+++ gtkwave-3.0.28.new/src/Makefile_GTK2.in	2007-06-06 23:33:05.000000000 +0530
@@ -1,7 +1,7 @@
 CC = @CC@
 AIXFLAGS = -bmaxdata:0xd0000000/dsa -D_WAVE_BE32
 BIGFILES = -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
-CFLAGS = -O2 -DWAVE_USE_GTK2 -I./libz -I./libbz2 -I./helpers -I../
+CFLAGS += -DWAVE_USE_GTK2 -I./libz -I./libbz2 -I./helpers -I../
 OBSOLETE_CFLAGS = -DGTK_ENABLE_BROKEN
 DEBUGS = -DDEBUG_FACILITIES -DDEBUG_PRINTF -DDEBUG_MALLOC -DSTRICT_VCD_ONLY -DDEBUG_MALLOC_LINES
 AET2LIB =
@@ -43,10 +43,10 @@
 	cp twinwave ../bin
 
 gtkwave:$(OBJS) libz/libz.a libbz2/libbz2.a
-	$(CC) $(CFLAGS) $(OBJS) `pkg-config gtk+-2.0 --libs` $(AET2LIB) libz/libz.a libbz2/libbz2.a -o gtkwave $(PTHLIB) $(NSLLIB)
+	$(CC) $(CFLAGS) $(LDFLAGS) $(OBJS) $(shell pkg-config gtk+-2.0 --libs) $(AET2LIB) libz/libz.a libbz2/libbz2.a -o gtkwave $(PTHLIB) $(NSLLIB)
 
 twinwave: twinwave.c
-	$(CC) $(CFLAGS) `pkg-config gtk+-2.0 --cflags` -o twinwave twinwave.c `pkg-config gtk+-2.0 --libs`
+	$(CC) $(CFLAGS) $(shell pkg-config gtk+-2.0 --cflags) -o twinwave twinwave.c $(shell pkg-config gtk+-2.0 --libs)
 
 libz/libz.a:
 	cd libz ; make
@@ -55,7 +55,7 @@
 	cd libbz2 ; make
 
 .c.o:	$(H_FILES)
-	$(CC) $(CFLAGS) `pkg-config gtk+-2.0 --cflags` -c -o $@ $<
+	$(CC) $(CFLAGS) $(shell pkg-config gtk+-2.0 --cflags) -c -o $@ $<
 
 clean: 
 	rm -f core *.core *.o gtkwave ../bin/gtkwave twinwave ../bin/twinwave
