--- compiz/plugins/decoration.c	2006-08-15 01:04:34.000000000 -0400
+++ compiz/plugins/decoration.c	2006-08-13 03:19:28.000000000 -0400
@@ -650,10 +651,12 @@
 	wd->quad[i].matrix.x0 += x0 * wd->quad[i].matrix.xx;
 	wd->quad[i].matrix.y0 += y0 * wd->quad[i].matrix.yy;
 
+	wd->quad[i].matrix.xy =
+        wd->quad[i].matrix.xx*wd->decor->quad[i].m.xy;
+	wd->quad[i].matrix.yx = 
+        wd->quad[i].matrix.yy*wd->decor->quad[i].m.yx;
 	wd->quad[i].matrix.xx *= wd->decor->quad[i].m.xx;
 	wd->quad[i].matrix.yy *= wd->decor->quad[i].m.yy;
-	wd->quad[i].matrix.xy *= wd->decor->quad[i].m.xy;
-	wd->quad[i].matrix.yx *= wd->decor->quad[i].m.yx;
 
 	wd->quad[i].matrix.x0 -=
 	    wd->quad[i].box.x1 * wd->quad[i].matrix.xx +
