diff -urN compiz-20061001.orig/gtk/gnome/Makefile.am compiz-20061001/gtk/gnome/Makefile.am
--- compiz-20061001.orig/gtk/gnome/Makefile.am	2006-10-01 14:29:52.000000000 +0100
+++ compiz-20061001/gtk/gnome/Makefile.am	2006-10-01 14:33:54.000000000 +0100
@@ -21,12 +21,14 @@
 	$(libcompiz_module)
 
 install-exec-local:
-	if test -w $(moduledir); then ( \
+	mkdir -p -- "$(DESTDIR)$(moduledir)"
+	if test -w $(DESTDIR)$(moduledir); then ( \
 		$(LIBTOOL) --mode=install $(INSTALL) $(libcompiz_module) \
-		$(moduledir) \
+		$(DESTDIR)$(moduledir) \
 		); \
 	fi
-	if test -w $(desktopfilesdir); then ( \
-		$(INSTALL) -m 644 $(desktopfiles_files) $(desktopfilesdir) \
+	mkdir -p -- "$(DESTDIR)$(desktopfilesdir)"
+	if test -w $(DESTDIR)$(desktopfilesdir); then ( \
+		$(INSTALL) -m 644 $(desktopfiles_files) $(DESTDIR)$(desktopfilesdir) \
 		); \
 	fi
