diff -Naur arpwatch-3.5/Makefile.in arpwatch-3.5.new/Makefile.in
--- arpwatch-3.5/Makefile.in	2023-12-03 01:48:53.000000000 +0100
+++ arpwatch-3.5.new/Makefile.in	2024-01-14 23:42:21.838447555 +0100
@@ -131,7 +131,6 @@
 	    $(INSTALL) -v -m 444 $(srcdir)/$(PROG).8 $(DESTDIR)$(MANDEST)/man8
 	@diff $(srcdir)/arpsnmp.8 $(DESTDIR)$(MANDEST)/man8 >/dev/null 2>&1 || \
 	    $(INSTALL) -v -m 444 $(srcdir)/arpsnmp.8 $(DESTDIR)$(MANDEST)/man8
-@HAVE_FREEBSD_TRUE@	@$(INSTALL) -v -m 555 arpwatch.sh $(DESTDIR)$(prefix)/etc/rc.d/arpwatch
 
 clean:
 	rm -f $(CLEANFILES)
diff -Naur arpwatch-3.5/report.c arpwatch-3.5.new/report.c
--- arpwatch-3.5/report.c	2023-09-05 19:50:56.000000000 +0200
+++ arpwatch-3.5.new/report.c	2024-01-14 23:42:21.846447615 +0100
@@ -1,3 +1,4 @@
+#include <time.h>
 /*
  * Copyright (c) 1990, 1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 2000, 2009, 2010, 2012, 2013, 2016, 2019, 2021, 2023
  * The Regents of the University of California. All rights reserved.
