--- qemu-0.7.1/kqemu/kqemu-linux.c       25 Apr 2005 22:14:39 -0000      1.4
+++ qemu-0.7.1/kqemu/kqemu-linux.c       25 Jul 2005 22:17:34 -0000
@@ -28,7 +28,13 @@
 #define pfn_to_page(pfn) (mem_map + (pfn))
 #endif

-#ifndef PAGE_KERNEL_EXEC
+#ifdef PAGE_KERNEL_EXEC
+#if defined(__i386__)
+/* problem : i386 kernels usually don't export __PAGE_KERNEL_EXEC */
+#undef PAGE_KERNEL_EXEC
+#define PAGE_KERNEL_EXEC __pgprot(__PAGE_KERNEL & ~_PAGE_NX)
+#endif
+#else
 #define PAGE_KERNEL_EXEC PAGE_KERNEL
 #endif
