From 8e785ff6cf5f1d2e48f713f7d4f0ee8872bb9d79 Mon Sep 17 00:00:00 2001
From: Stef Walter <stefw@collabora.co.uk>
Date: Mon, 21 Nov 2011 13:10:53 +0000
Subject: Don't include internal glib headers

 * Fixes the build with glib 2.31
---
diff --git a/egg/egg-bytes.h b/egg/egg-bytes.h
index 57cba03..c30a43a 100644
--- a/egg/egg-bytes.h
+++ b/egg/egg-bytes.h
@@ -24,8 +24,7 @@
 #ifndef __EGG_BYTES_H__
 #define __EGG_BYTES_H__
 
-#include <glib/gtypes.h>
-#include <glib/garray.h>
+#include <glib.h>
 
 /**
  * EggBytes:
--
cgit v0.9.0.2

