--- weston-8.0.0/meson.build	2020-01-24 21:05:18.395629600 +0100
+++ weston-8.0.0.new/meson.build	2020-01-30 13:49:02.748123627 +0100
@@ -36,7 +36,7 @@
 dir_libexec = join_paths(dir_prefix, get_option('libexecdir'))
 dir_module_weston = join_paths(dir_lib, 'weston')
 dir_module_libweston = join_paths(dir_lib, 'libweston-@0@'.format(libweston_major))
-dir_data_pc = join_paths(dir_data, 'pkgconfig')
+dir_data_pc = join_paths(dir_lib, 'pkgconfig')
 dir_lib_pc = join_paths(dir_lib, 'pkgconfig')
 dir_man = join_paths(dir_prefix, get_option('mandir'))
 dir_protocol_libweston = join_paths('libweston-@0@'.format(libweston_major), 'protocols')
