diff -Naur zapping-0.8.orig/Makefile.in zapping-0.8/Makefile.in
--- zapping-0.8.orig/Makefile.in	2004-11-11 15:46:49.000000000 +0100
+++ zapping-0.8/Makefile.in	2004-11-11 18:42:55.552655700 +0100
@@ -269,1 +269,1 @@
-desktopdir = $(datadir)/applications
+desktopdir = $(libdir)/applications
diff -Naur zapping-0.8.orig/zapping.desktop.in zapping-0.8/zapping.desktop.in
--- zapping-0.8.orig/zapping.desktop.in	2004-11-02 17:51:18.000000000 +0100
+++ zapping-0.8/zapping.desktop.in	2004-11-11 18:43:56.089368251 +0100
@@ -1,10 +1,11 @@
 [Desktop Entry]
 Version=0.9
 Encoding=UTF-8
-_Name=Zapping TV Viewer
+_Name=Zapping
+_Comment=TV viewer
 Type=Application
 Exec=zapping
-Icon=zapping/gnome-television.png
+Icon=zapping.png
 Terminal=false
 StartupNotify=true
 Categories=GNOME;Application;Multimedia;AudioVideo;Video;TV;
