diff -Naur netkit-timed-0.17.orig/timed/timed/timed.c netkit-timed-0.17/timed/timed/timed.c
--- netkit-timed-0.17.orig/timed/timed/timed.c	1999-12-12 19:05:07.000000000 +0100
+++ netkit-timed-0.17/timed/timed/timed.c	2004-07-20 10:17:48.000000000 +0200
@@ -898,7 +898,7 @@
 void
 get_goodgroup(int force)
 {
-# define NG_DELAY (30*60*CLK_TCK)	/* 30 minutes */
+# define NG_DELAY (30*60*HZ)	/* 30 minutes */
 	static unsigned long last_update = -NG_DELAY;
 	unsigned long new_update;
 /*	struct hosttbl *htp; */
