diff -Naur ghc-8.0.2/aclocal.m4 ghc-8.0.2-p/aclocal.m4
--- ghc-8.0.2/aclocal.m4	2017-01-03 16:59:18.000000000 +0100
+++ ghc-8.0.2-p/aclocal.m4	2017-03-17 15:57:18.389731566 +0100
@@ -227,7 +227,7 @@
 
     checkVendor() {
         case [$]1 in
-        dec|unknown|hp|apple|next|sun|sgi|ibm|montavista|portbld)
+        dec|unknown|hp|apple|next|sun|sgi|ibm|montavista|portbld|frugalware)
             ;;
         *)
             echo "Unknown vendor [$]1"
diff -Naur ghc-8.0.2/configure.ac ghc-8.0.2-p/configure.ac
--- ghc-8.0.2/configure.ac	2017-01-04 15:17:17.000000000 +0100
+++ ghc-8.0.2-p/configure.ac	2017-03-17 15:57:40.580307264 +0100
@@ -220,7 +220,6 @@
     echo "This GHC (${WithGhc}) does not generate code for the build platform"
     echo "   GHC target platform    : $bootstrap_target"
     echo "   Desired build platform : $BuildPlatform"
-    exit 1
 fi
 
 # Testing if we shall enable shared libs support on Solaris.
