diff -urNad gnome-system-tools-2.15.4~/src/shares/share-settings.c gnome-system-tools-2.15.4/src/shares/share-settings.c
--- gnome-system-tools-2.15.4~/src/shares/share-settings.c	2006-09-04 05:13:18.000000000 +0200
+++ gnome-system-tools-2.15.4/src/shares/share-settings.c	2006-09-08 21:27:42.000000000 +0200
@@ -55,6 +55,8 @@
 {
 	gchar *name;
 
+	g_return_if_fail (path != NULL);
+
 	if (strcmp (path, "/") == 0)
 		name = g_strdup (_("File System"));
 	else
@@ -435,7 +437,7 @@
 	GtkTreeIter   iter;
 	gboolean      path_exists;
 	gchar        *title;
-	GtkWidget    *name_entry;
+	GtkWidget    *name_entry, *file_chooser;
 
 	share  = NULL;
 	dialog = share_settings_prepare_dialog (path, standalone);
@@ -456,6 +458,12 @@
 		g_object_set_data (name_entry, "modified", GINT_TO_POINTER (FALSE));
 		gtk_window_set_title (GTK_WINDOW (dialog), "Share Folder");
 
+		if (!path) {
+			/* make sure the path gets filled */
+			file_chooser = gst_dialog_get_widget (tool->main_dialog, "share_path");
+			path = gtk_file_chooser_get_current_folder (GTK_FILE_CHOOSER (file_chooser));
+		}
+
 		share_settings_set_name_from_folder (path);
 	}
 
