--- shared-mime-info-2.2/meson.build	2022-03-27 12:19:00.000000000 +0200
+++ shared-mime-info-2.2.new/meson.build	2022-12-09 10:46:05.029965488 +0100
@@ -74,7 +74,7 @@
         'prefix': get_option('prefix'),
         'VERSION': meson.project_version()
     },
-    install_dir: get_option('datadir') / 'pkgconfig'
+    install_dir: get_option('libdir') / 'pkgconfig'
 )
 
 if get_option('update-mimedb')
