diff --git a/src/scim_hangul_imengine.cpp b/src/scim_hangul_imengine.cpp
index 0420cd8..8473fb7 100644
--- a/src/scim_hangul_imengine.cpp
+++ b/src/scim_hangul_imengine.cpp
@@ -34,6 +34,7 @@
 #endif
 
 #include <scim.h>
+#include <string.h>
 #include "scim_hangul_imengine.h"
 
 #ifdef HAVE_GETTEXT
diff --git a/src/scim_hangul_imengine_setup.cpp b/src/scim_hangul_imengine_setup.cpp
index e2d3bbb..72496c5 100644
--- a/src/scim_hangul_imengine_setup.cpp
+++ b/src/scim_hangul_imengine_setup.cpp
@@ -35,6 +35,7 @@
 #endif
 
 #include <scim.h>
+#include <string.h>
 #include <gtk/scimkeyselection.h>
 
 #ifdef HAVE_GETTEXT
