diff --git a/half.cabal b/half.cabal
index 8696192..7fb01da 100644
--- a/half.cabal
+++ b/half.cabal
@@ -29,6 +29,7 @@ tested-with:   GHC == 7.4.2
              , GHC == 8.6.5
              , GHC == 8.8.4
              , GHC == 8.10.3
+             , GHC == 9.8.1
 
 source-repository head
     type: git
@@ -46,7 +47,7 @@ library
     build-depends:
         base >=4.5 && <5,
         binary >=0.5.1.0 && <0.9,
-        deepseq >=1.3.0.0 && <1.5,
+        deepseq >=1.3.0.0 && <1.6,
         template-haskell
 
     if !impl(ghcjs)
