diff -Naur pngcrush-1.6.4/pngcrush.h pngcrush-1.6.4-p/pngcrush.h
--- pngcrush-1.6.4/pngcrush.h	2006-03-09 22:32:55.000000000 +0100
+++ pngcrush-1.6.4-p/pngcrush.h	2006-06-08 22:05:34.000000000 +0200
@@ -18,10 +18,7 @@
 #define PNGCRUSH_LIBPNG_VER 10007
 #endif
 
-#if !defined(PNG_USE_PNGGCCRD) && !defined(PNG_USE_PNGVCRD) && \
-    !defined(PNG_NO_ASSEMBLER_CODE)
 #define PNG_NO_ASSEMBLER_CODE
-#endif
 
 #ifndef PNG_NO_ZALLOC_ZERO
 #  define PNG_NO_ZALLOC_ZERO  /* speeds it up a little */
