diff -aur libzrtpcpp-1.0.0/src/Makefile.in fw_libzrtpcpp-1.0.0/src/Makefile.in
--- libzrtpcpp-1.0.0/src/Makefile.in	2007-11-30 00:13:52.000000000 +0100
+++ fw_libzrtpcpp-1.0.0/src/Makefile.in	2007-12-04 21:25:09.000000000 +0100
@@ -248,7 +248,6 @@
 RELEASE = $(LT_VERSION) -release $(LT_RELEASE)
 lib_LTLIBRARIES = libzrtpcpp.la
 # libzrtpcpp_la_LIBADD = @LIBS@
-libzrtpcpp_la_LDFLAGS = $(RELEASE)
 
 # http://www.gnu.org/software/libtool/manual.html
 @ZRTP_OPENSSL_TRUE@zrtp_src_o = libzrtpcpp/crypto/openssl/ZrtpDH.cxx \
