diff -Naur qwt-5.0.2/qwtconfig.pri qwt-5.0.2-p/qwtconfig.pri
--- qwt-5.0.2/qwtconfig.pri	2007-06-11 07:41:29.000000000 +0200
+++ qwt-5.0.2-p/qwtconfig.pri	2007-11-26 20:25:03.000000000 +0100
@@ -3,7 +3,7 @@
 ######################################################################
 
 unix {
-    INSTALLBASE    = /usr/local/qwt-5.0.2
+    INSTALLBASE    = /usr
 }
 
 win32 {
@@ -12,7 +12,7 @@
 
 target.path    = $$INSTALLBASE/lib
 headers.path   = $$INSTALLBASE/include
-doc.path       = $$INSTALLBASE/doc
+doc.path       = $$INSTALLBASE/share
 
 ######################################################################
 # qmake internal options
@@ -58,7 +58,7 @@
 # QwtSVGItem.
 ######################################################################
 
-#CONFIG     += QwtSVGItem
+CONFIG     += QwtSVGItem
 
 ######################################################################
 # If you have a commercial license you can use the MathML renderer
@@ -75,7 +75,7 @@
 # Otherwise you have to build it from the designer directory.
 ######################################################################
 
-CONFIG     += QwtDesigner
+CONFIG     += QwtDesigner
 
 ######################################################################
 # If you want to auto build the examples, enable the line below
