diff -Naur dazuko-2.2.1.orig/Makefile dazuko-2.2.1/Makefile
--- dazuko-2.2.1.orig/Makefile	2006-06-10 17:33:36.000000000 +0200
+++ dazuko-2.2.1/Makefile	2006-06-10 20:15:25.000000000 +0200
@@ -9,7 +9,6 @@
 dazuko-objs := dazuko_core.o dazuko_transport.o dazuko_linux26_lsm.o dazuko_linux26.o
 
 dummy_rule:
-	$(MAKE) -C /usr/src/linux include/linux/version.h include/asm scripts
 	$(MAKE) -C /usr/src/linux SUBDIRS="/var/tmp/fst/src/dazuko-2.2.1" modules
 	touch dummy_rule
 
