diff -Naur jack-audio-connection-kit-0.103.0/configure.ac jack-audio-connection-kit-0.103.0-no-multilib/configure.ac
--- jack-audio-connection-kit-0.103.0/configure.ac	2007-03-06 06:24:05.000000000 +0100
+++ jack-audio-connection-kit-0.103.0-no-multilib/configure.ac	2007-04-21 00:29:04.000000000 +0200
@@ -100,26 +100,6 @@
 
 AC_SUBST(OS_LDFLAGS)
 
-#
-# We need to establish suitable defaults for a 64-bit OS
-libnn=lib
-case "${host_os}" in
-    linux*)
-    case "${host_cpu}" in
-	x86_64|mips64|ppc64|sparc64|s390x)
-        libnn=lib64
-	;;
-    esac
-    ;;
-    solaris*)
-    ## libnn=lib/sparcv9 ## on 64-bit only, but that's compiler-specific
-    ;;  
-esac
-
-## take care not to  override the command-line setting
-if test "${libdir}" = '${exec_prefix}/lib'; then
-  libdir='${exec_prefix}/'${libnn}
-fi
 
 # system-dependent config.h values
 test "x$JACK_THREAD_STACK_TOUCH" = "x" && JACK_THREAD_STACK_TOUCH=500000
