diff -Nur xmms2-0.4DrKosmos/src/include/xmmsclient/xmmsclient++/helpers.h xmms2-0.4DrKosmos.new/src/include/xmmsclient/xmmsclient++/helpers.h
--- xmms2-0.4DrKosmos/src/include/xmmsclient/xmmsclient++/helpers.h	2007-11-15 19:25:07.000000000 +0100
+++ xmms2-0.4DrKosmos.new/src/include/xmmsclient/xmmsclient++/helpers.h	2008-06-05 00:32:11.000000000 +0200
@@ -32,6 +32,7 @@
 #include <string>
 #include <list>
 #include <vector>
+#include <limits.h>
 
 namespace Xmms
 {
