diff -Naur zfs-zfs-2.0.0/config/kernel.m4 zfs-zfs-2.0.0-p/config/kernel.m4
--- zfs-zfs-2.0.0/config/kernel.m4	2020-11-30 19:13:14.000000000 +0100
+++ zfs-zfs-2.0.0-p/config/kernel.m4	2020-12-15 13:20:11.728985213 +0100
@@ -330,7 +330,7 @@
 	AS_IF([test "$utsrelease"], [
 		kernsrcver=`(echo "#include <$utsrelease>";
 		             echo "kernsrcver=UTS_RELEASE") |
-		             ${CPP} -I $kernelbuild/include - |
+		             ${CXXCPP} -I $kernelbuild/include - |
 		             grep "^kernsrcver=" | cut -d \" -f 2`
 
 		AS_IF([test -z "$kernsrcver"], [
diff -Naur zfs-zfs-2.0.0/configure.ac zfs-zfs-2.0.0-p/configure.ac
--- zfs-zfs-2.0.0/configure.ac	2020-11-30 19:13:14.000000000 +0100
+++ zfs-zfs-2.0.0-p/configure.ac	2020-12-15 13:19:27.100717382 +0100
@@ -48,6 +48,7 @@
 AC_PROG_INSTALL
 AC_PROG_CC
 AC_PROG_LIBTOOL
+AC_PROG_CXXCPP
 PKG_PROG_PKG_CONFIG
 AM_PROG_AS
 AM_PROG_CC_C_O
