diff -ur a/module/Makefile.in b/module/Makefile.in
--- a/module/Makefile.in	2014-06-12 15:58:09.000000000 -0500
+++ b/module/Makefile.in	2014-07-18 06:05:10.375381535 -0500
@@ -5,7 +5,7 @@
 subdir-m += zfs
 subdir-m += zpios
 
-INSTALL_MOD_DIR ?= extra
+INSTALL_MOD_DIR ?= kernel/extra
 
 ZFS_MODULE_CFLAGS += -include @SPL_OBJ@/spl_config.h
 ZFS_MODULE_CFLAGS += -include @abs_top_builddir@/zfs_config.h
