diff -Naur zfs-2.1.9/config/kernel-generic_io_acct.m4 zfs-2.1.9.new/config/kernel-generic_io_acct.m4
--- zfs-2.1.9/config/kernel-generic_io_acct.m4	2023-01-25 20:11:41.091744538 +0100
+++ zfs-2.1.9.new/config/kernel-generic_io_acct.m4	2023-03-24 13:22:20.506893348 +0100
@@ -2,7 +2,20 @@
 dnl # Check for generic io accounting interface.
 dnl #
 AC_DEFUN([ZFS_AC_KERNEL_SRC_GENERIC_IO_ACCT], [
-	ZFS_LINUX_TEST_SRC([bdev_io_acct], [
+	ZFS_LINUX_TEST_SRC([bdev_io_acct_62], [
+		#include <linux/blkdev.h>
+	], [
+		struct block_device *bdev = NULL;
+		struct bio *bio = NULL;
+		unsigned long passed_time = 0;
+		unsigned long start_time;
+
+		start_time = bdev_start_io_acct(bdev, bio_op(bio),
+		    passed_time);
+		bdev_end_io_acct(bdev, bio_op(bio), bio_sectors(bio), start_time);
+	])
+
+	ZFS_LINUX_TEST_SRC([bdev_io_acct_old], [
 		#include <linux/blkdev.h>
 	], [
 		struct block_device *bdev = NULL;
@@ -68,13 +81,18 @@
 	dnl # disk_start_io_acct() and disk_end_io_acct() have been replaced by
 	dnl # bdev_start_io_acct() and bdev_end_io_acct().
 	dnl #
-	AC_MSG_CHECKING([whether generic bdev_*_io_acct() are available])
-	ZFS_LINUX_TEST_RESULT([bdev_io_acct], [
+	AC_MSG_CHECKING([whether 6.2+ bdev_*_io_acct() are available])
+	ZFS_LINUX_TEST_RESULT([bdev_io_acct_62], [
 		AC_MSG_RESULT(yes)
-		AC_DEFINE(HAVE_BDEV_IO_ACCT, 1, [bdev_*_io_acct() available])
+		AC_DEFINE(HAVE_BDEV_IO_ACCT_62, 1, [bdev_*_io_acct() available])
 	], [
 		AC_MSG_RESULT(no)
-
+		AC_MSG_CHECKING([whether pre-6.3 bdev_*_io_acct() are available])
+		ZFS_LINUX_TEST_RESULT([bdev_io_acct_old], [
+			AC_MSG_RESULT(yes)
+			AC_DEFINE(HAVE_BDEV_IO_ACCT_OLD, 1, [bdev_*_io_acct() available])
+		], [
+		AC_MSG_RESULT(no)
 		dnl #
 		dnl # 5.12 API,
 		dnl #
@@ -137,3 +155,4 @@
 		])
 	])
 ])
+])
\ No newline at end of file
diff -Naur zfs-2.1.9/include/os/linux/kernel/linux/blkdev_compat.h zfs-2.1.9.new/include/os/linux/kernel/linux/blkdev_compat.h
--- zfs-2.1.9/include/os/linux/kernel/linux/blkdev_compat.h	2023-01-25 20:11:42.214766890 +0100
+++ zfs-2.1.9.new/include/os/linux/kernel/linux/blkdev_compat.h	2023-03-24 13:22:20.507893356 +0100
@@ -555,7 +555,10 @@
     struct gendisk *disk __attribute__((unused)),
     int rw __attribute__((unused)), struct bio *bio)
 {
-#if defined(HAVE_BDEV_IO_ACCT)
+#if defined(HAVE_BDEV_IO_ACCT_62)
+	return (bdev_start_io_acct(bio->bi_bdev, bio_op(bio),
+	    jiffies));
+#elif defined(HAVE_BDEV_IO_ACCT_OLD)
 	return (bdev_start_io_acct(bio->bi_bdev, bio_sectors(bio),
 	    bio_op(bio), jiffies));
 #elif defined(HAVE_DISK_IO_ACCT)
@@ -581,7 +584,9 @@
     struct gendisk *disk __attribute__((unused)),
     int rw __attribute__((unused)), struct bio *bio, unsigned long start_time)
 {
-#if defined(HAVE_BDEV_IO_ACCT)
+#if defined(HAVE_BDEV_IO_ACCT_62)
+	bdev_end_io_acct(bio->bi_bdev, bio_op(bio), bio_sectors(bio), start_time);
+#elif defined(HAVE_BDEV_IO_ACCT_OLD)
 	bdev_end_io_acct(bio->bi_bdev, bio_op(bio), start_time);
 #elif defined(HAVE_DISK_IO_ACCT)
 	disk_end_io_acct(disk, bio_op(bio), start_time);
