--- zfs-zfs-0.7.12/include/zpios-ctl.h	2018-11-08 23:38:37.000000000 +0100
+++ zfs-zfs-0.7.12.new/include/zpios-ctl.h	2018-12-24 17:14:24.629599093 +0100
@@ -181,9 +181,11 @@
 zpios_timespec_now(void)
 {
 	zpios_timespec_t zts_now;
-	struct timespec ts_now;
-
-	ts_now = current_kernel_time();
+#if defined(HAVE_INODE_TIMESPEC64_TIMES)
+	inode_timespec_t ts_now = current_kernel_time64();
+#else
+	inode_timespec_t ts_now  = current_kernel_time();
+#endif
 	zts_now.ts_sec  = ts_now.tv_sec;
 	zts_now.ts_nsec = ts_now.tv_nsec;
 
