--- db-4.4.16/dist/Makefile.in.ark	2005-12-14 14:31:25.000000000 +0800
+++ db-4.4.16/dist/Makefile.in	2005-12-14 14:31:48.000000000 +0800
@@ -92,7 +92,7 @@
 #
 # Java support is optional and requires shared librarires.
 ##################################################
-CLASSPATH=	$(JAVA_CLASSTOP)
+CLASSPATH=     $(JAVA_CLASSTOP):/usr/share/java/libgcj-$(shell gcc -dumpversion).jar
 LIBJSO_LIBS=	@LIBJSO_LIBS@ @LIBSO_LIBS@
 
 JAR=		@JAR@
--- db-4.4.20/dist/Makefile.in.orig	2006-09-27 03:05:09.000000000 +0200
+++ db-4.4.20/dist/Makefile.in	2006-09-27 03:05:58.000000000 +0200
@@ -755,7 +755,7 @@
 	@test -d $(JAVA_EXCLASSTOP) || \
 	   ($(mkdir) -p $(JAVA_EXCLASSTOP) && \
 		$(chmod) $(dmode) $(JAVA_EXCLASSTOP))
-	$(JAVAC) -classpath $(libj_jarfile) -d $(JAVA_EXCLASSTOP) \
+	$(JAVAC) -classpath $(libj_jarfile):/usr/share/java/libgcj-$(shell gcc -dumpversion).jar -d $(JAVA_EXCLASSTOP) \
 		$(JAVACFLAGS) $(JAVA_EXSRCS)
 	cd $(JAVA_EXCLASSTOP) && $(JAR) cf ../$(libj_exjarfile) .
 
