diff -Naur db-4.5.20.orig/dist/Makefile.in db-4.5.20/dist/Makefile.in
--- db-4.5.20.orig/dist/Makefile.in	2006-09-13 18:22:38.000000000 +0200
+++ db-4.5.20/dist/Makefile.in	2006-10-05 23:30:29.000000000 +0200
@@ -92,7 +92,7 @@
 #
 # Java support is optional and requires shared librarires.
 ##################################################
-CLASSPATH=	$(JAVA_CLASSTOP)
+CLASSPATH=	$(JAVA_CLASSTOP):/usr/share/java/libgcj-$(shell gcc -dumpversion).jar
 LIBJSO_LIBS=	@LIBJSO_LIBS@ @LIBSO_LIBS@
 SWIGCFLAGS=	@SWIGCFLAGS@
 
@@ -627,7 +627,7 @@
 	@test -d $(JAVA_EXCLASSTOP) || \
 	   ($(MKDIR) -p $(JAVA_EXCLASSTOP) && \
 		$(CHMOD) $(dmode) $(JAVA_EXCLASSTOP))
-	$(JAVAC) -classpath $(libj_jarfile) -d $(JAVA_EXCLASSTOP) \
+	$(JAVAC) -classpath $(libj_jarfile):/usr/share/java/libgcj-$(shell gcc -dumpversion).jar -d $(JAVA_EXCLASSTOP) \
 		$(JAVACFLAGS) $(JAVA_EXSRCS)
 	cd $(JAVA_EXCLASSTOP) && $(JAR) cf ../$(libj_exjarfile) .
 
