diff -ur a/Make.defaults b/Make.defaults
--- a/Make.defaults	2013-03-29 14:03:14.000000000 -0500
+++ b/Make.defaults	2013-11-09 05:43:49.370051831 -0600
@@ -63,8 +63,8 @@
 TOPDIR  =
 ALLSUBDIRS = ia32 ia64 x86_64 fs choosers devschemes tools
 
-HOSTARCH   = $(shell dpkg-architecture -qDEB_BUILD_ARCH | sed s,i[3456789]86,ia32, | sed s,amd64,x86_64, )
-ARCH	   := $(shell dpkg-architecture -qDEB_BUILD_ARCH | sed s,i[3456789]86,ia32, | sed s,amd64,x86_64, )
+HOSTARCH   = $(shell uname -m | sed 's|i[0-9]86|ia32|')
+ARCH	   := $(shell uname -m | sed 's|i[0-9]86|ia32|')
 INCDIR	   = -I. -I$(TOPDIR) -I$(EFIINC) -I$(EFIINC)/$(ARCH) -I$(EFIINC)/protocol -I$(TOPDIR)/efi110
 CPPFLAGS   = -DCONFIG_$(ARCH) 
 
@@ -115,9 +115,9 @@
 	ARCH3264 = -m64
 	LD3264 = -melf_x86_64
 
-	GNUEFILIB := $(GNUEFILIB)64
-	EFILIB    := $(EFILIB)64
-	EFICRT0   := $(EFICRT0)64
+	GNUEFILIB := $(GNUEFILIB)
+	EFILIB    := $(EFILIB)
+	EFICRT0   := $(EFICRT0)
   endif
 
         CFLAGS   += -DEFI_FUNCTION_WRAPPER
@@ -129,9 +129,9 @@
 	ARCH3264 = -m32
 	LD3264 = -melf_i386
 
-	GNUEFILIB := /usr/lib32
-	EFILIB    := /usr/lib32
-	EFICRT0   := /usr/lib32
+	GNUEFILIB := /usr/lib
+	EFILIB    := /usr/lib
+	EFICRT0   := /usr/lib
   endif
 endif
 
