--- python-libtorrent/setup.py.orig	2007-01-18 16:16:31.000000000 +0100
+++ python-libtorrent/setup.py	2007-01-18 16:18:13.000000000 +0100
@@ -21,9 +21,9 @@
 module1 = Extension('torrent',
                     include_dirs = ['./include', './include/libtorrent',
 												'/usr/include/python' + pythonVersion],
-                    libraries = ['boost_filesystem', 'boost_date_time',
-											'boost_program_options', 'boost_regex',
-											'boost_serialization', 'boost_thread', 'z', 'pthread'],
+                    libraries = ['boost_filesystem-gcc42-mt', 'boost_date_time-gcc42-mt',
+											'boost_program_options-gcc42-mt', 'boost_regex-gcc42-mt',
+											'boost_serialization-gcc42-mt', 'boost_thread-gcc42-mt', 'z', 'pthread'],
                     sources = ['alert.cpp',
 										 'allocate_resources.cpp',
 										 'bt_peer_connection.cpp',
