diff -Naur seafile-client-6.1.1/CMakeLists.txt seafile-client-6.1.1-p/CMakeLists.txt
--- seafile-client-6.1.1/CMakeLists.txt	2017-09-19 12:18:55.000000000 +0200
+++ seafile-client-6.1.1-p/CMakeLists.txt	2018-06-15 22:31:46.239401100 +0200
@@ -763,9 +763,9 @@
   ${EXTRA_LIBS}
 )
 
-QT5_USE_MODULES(seafile-applet Core Gui Widgets Network)
+TARGET_LINK_LIBRARIES(seafile-applet Qt5::Core Qt5::Gui Qt5::Widgets Qt5::Network)
 IF (BUILD_SHIBBOLETH_SUPPORT)
-QT5_USE_MODULES(seafile-applet ${WEBKIT_NAME} ${WEBKIT_WIDGETS_NAME})
+TARGET_LINK_LIBRARIES(seafile-applet Qt5::WebEngine Qt5::WebEngineWidgets)
 ENDIF()
 
 ## QtBus
@@ -773,7 +773,7 @@
   TARGET_LINK_LIBRARIES(seafile-applet
     ${QT_QTDBUS_LIBRARIES}
   )
-  QT5_USE_MODULES(seafile-applet ${USE_QT_LIBRARIES})
+  TARGET_LINK_LIBRARIES(seafile-applet Qt5::DBus)
 ENDIF()
 
 ### Xcode-related, build as a osx bundle
@@ -822,7 +822,7 @@
 
         INCLUDE_DIRECTORIES(${Qt5Test_INCLUDE_DIRS})
 
-        QT5_USE_MODULES(${testname} ${USE_QT_LIBRARIES} Test)
+        TARGET_LINK_LIBRARIES(${testname} ${USE_QT_LIBRARIES} Qt5::Test)
 
         ADD_TEST(${testname} ${CMAKE_CURRENT_BINARY_DIR}/tests/${testname})
 
