diff -Naur seafile-client-6.1.1/src/ui/private-share-dialog.h seafile-client-6.1.1-pp/src/ui/private-share-dialog.h
--- seafile-client-6.1.1/src/ui/private-share-dialog.h	2017-09-19 12:18:55.000000000 +0200
+++ seafile-client-6.1.1-pp/src/ui/private-share-dialog.h	2018-06-15 22:40:06.143083347 +0200
@@ -12,6 +12,7 @@
 #include <QStringList>
 #include <QStyledItemDelegate>
 #include <QTableView>
+#include <QHeaderView>
 #include "ui_private-share-dialog.h"
 
 #include "account.h"
