diff -u -r1.7 -r1.8
--- ltp/testcases/kernel/syscalls/getsid/getsid02.c	2003/08/27 14:53:47	1.7
+++ ltp/testcases/kernel/syscalls/getsid/getsid02.c	2005/12/22 20:00:43	1.8
@@ -54,6 +54,14 @@
  */
 #define _GNU_SOURCE 1
 
+/* 
+ * When attempting to build on SUSE 10, the make fails trying to compile
+ * because CONFIG_BASE_SMALL is undefined.
+ */
+#ifndef CONFIG_BASE_SMALL
+#define CONFIG_BASE_SMALL 0
+#endif
+
 #include "test.h"
 #include "usctest.h"
 
diff -u -r1.7 -r1.8
--- ltp/testcases/kernel/syscalls/setpriority/setpriority04.c	2003/06/05 19:45:52	1.7
+++ ltp/testcases/kernel/syscalls/setpriority/setpriority04.c	2005/12/22 20:00:43	1.8
@@ -59,6 +59,14 @@
 #include <sys/resource.h>
 
 /*
+ *  * When attempting to build on SUSE 10, the make fails trying to compile
+ *   * because CONFIG_BASE_SMALL is undefined.
+ *    */
+#ifndef CONFIG_BASE_SMALL
+#define CONFIG_BASE_SMALL 0
+#endif
+
+/*
  * See the Makefile for comments about the following preprocessor code.
  */
 #ifndef _LTP_TASKS_H
diff -u -r1.10 -r1.11
--- ltp/testcases/kernel/syscalls/wait4/wait402.c	2005/07/11 22:29:12	1.10
+++ ltp/testcases/kernel/syscalls/wait4/wait402.c	2005/12/22 20:00:44	1.11
@@ -63,6 +63,14 @@
 #include <string.h>
 
 /*
+ *  * When attempting to build on SUSE 10, the make fails trying to compile
+ *   * because CONFIG_BASE_SMALL is undefined.
+ *    */
+#ifndef CONFIG_BASE_SMALL
+#define CONFIG_BASE_SMALL 0
+#endif
+
+/*
  * See the Makefile for comments about the following preprocessor code.
  */
 #ifndef _LTP_TASKS_H
