diff --color -Naur mvfst-2025.02.17.00/CMakeLists.txt mvfst-2025.02.17.00.new/CMakeLists.txt
--- mvfst-2025.02.17.00/CMakeLists.txt	2025-01-30 23:51:03.000000000 +0100
+++ mvfst-2025.02.17.00.new/CMakeLists.txt	2025-02-21 08:50:49.361949328 +0100
@@ -52,7 +52,7 @@
 find_package(fmt REQUIRED)
 find_package(folly REQUIRED)
 find_package(Fizz REQUIRED)
-find_package(Glog REQUIRED)
+find_package(Glog CONFIG REQUIRED)
 find_package(Threads)
 
 SET(GFLAG_DEPENDENCIES "")
