--- Module/rt_config.h	2007/05/10 08:45:32	1.8
+++ Module/rt_config.h	2007/06/18 19:39:09	1.9
@@ -88,6 +88,22 @@
 #include <asm/irq.h>
 #include <asm/uaccess.h>
 
+#ifndef pci_name
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0))
+#define pci_name(__pPci_Dev)	(__pPci_Dev)->dev.bus_id
+#else /* (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) */
+#define pci_name(__pPci_Dev)	(__pPci_Dev)->slot_name
+#endif /*(LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) */
+#endif /* pci_name */
+
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,22))
+#define pci_module_init	pci_register_driver
+#endif
+
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,22))
+#define skb_reset_mac_header(skb) (skb->mac.raw = skb->data)
+#endif
+
 #if WIRELESS_EXT > 12
 #include <net/iw_handler.h> /* New driver API */
 #endif /* WIRELESS_EXT > 12 */
--- Module/rtmp.h	2007/04/25 05:28:43	1.10
+++ Module/rtmp.h	2007/06/18 19:39:10	1.11
@@ -55,14 +55,6 @@
 #define netdev_priv(_dev)	( _dev->priv )
 #endif
 
-#ifndef pci_name
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0))
-#define pci_name(__pPci_Dev)	(__pPci_Dev)->dev.bus_id
-#else /* (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) */
-#define pci_name(__pPci_Dev)	(__pPci_Dev)->slot_name
-#endif /*(LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)) */
-#endif /* pci_name */
-
 //
 //  MACRO for debugging information
 //
--- Module/rtmp_data.c	2007/03/21 04:12:20	1.6
+++ Module/rtmp_data.c	2007/06/18 19:39:10	1.7
@@ -110,7 +110,7 @@
                     {
                        skb->dev = pAd->net_dev;
                        memcpy(skb_put(skb, pRxD->DataByteCnt), pData, pRxD->DataByteCnt);
-                       skb->mac.raw = skb->data;
+                       skb_reset_mac_header(skb);
                        skb->pkt_type = PACKET_OTHERHOST;
                        skb->protocol = htons(ETH_P_802_2);
                        skb->ip_summed = CHECKSUM_NONE;
