diff -NruBb zinf-2.2.5ori/base/unix/src/bootstrap.cpp zinf-2.2.5/base/unix/src/bootstrap.cpp
--- zinf-2.2.5ori/base/unix/src/bootstrap.cpp	2003-11-21 01:21:58.000000000 +0100
+++ zinf-2.2.5/base/unix/src/bootstrap.cpp	2005-05-19 16:57:50.000000000 +0200
@@ -149,7 +149,9 @@
                         ResolvePath(path);
 
                     strncpy(pPtr, path.c_str(),path.length());
-                    pPtr += strlen(pPtr) + 1;
+		    
+		    pPtr[path.length()]='\0';
+		    pPtr += path.length() + 1;
                 }
                 *pPtr = 0;
 
