--- zinf-2.2.5.orig/base/include/player.h	2004-02-02 23:09:35.000000000 +0530
+++ zinf-2.2.5/base/include/player.h	2006-09-09 14:48:13.000000000 +0530
@@ -155,13 +155,13 @@
     
     #define _EQUALIZER_ENABLE_
     #ifdef  _EQUALIZER_ENABLE_
-    void Player::SetEQData(Event *pEvent);
+    void SetEQData(Event *pEvent);
     #endif // _EQUALIZER_ENABLE_
     #undef  _EQUALIZER_ENABLE_    
 
     #define _VISUAL_ENABLE_
     #ifdef  _VISUAL_ENABLE_
-    void Player::SendVisBuf(Event *pEvent); 
+    void SendVisBuf(Event *pEvent); 
     #endif // _VISUAL_ENABLE_
     #undef  _VISUAL_ENABLE_  
 
--- zinf-2.2.5.orig/base/include/queue.h
+++ zinf-2.2.5/base/include/queue.h
@@ -24,6 +24,8 @@
 #ifndef INCLUDED_QUEUE_H_
 #define INCLUDED_QUEUE_H_
 
+#include <cassert>
+
 #include "config.h"
 #include "mutex.h"