=== modified file 'src/pidgin-libnotify.c'
--- old/src/pidgin-libnotify.c	2009-02-18 22:26:12 +0000
+++ new/src/pidgin-libnotify.c	2009-02-19 03:38:20 +0000
@@ -706,6 +706,20 @@
 	pidgin_blist_toggle_visibility();
 }
 
+static gboolean
+force_load_once (gpointer data)
+{
+	PurplePlugin * plugin = (PurplePlugin *)data;
+
+	if (!purple_prefs_get_bool("/plugins/gtk/libnotify/auto_loaded")) {
+		purple_plugin_load(plugin);
+		purple_plugins_save_loaded(PIDGIN_PREFS_ROOT "/plugins/loaded");
+		purple_prefs_set_bool("/plugins/gtk/libnotify/auto_loaded", TRUE);
+	}
+
+	return FALSE;
+}
+
 static void
 notify_check_caps_helper (gpointer data, gpointer user_data)
 {
@@ -877,6 +891,9 @@
 	purple_prefs_add_bool ("/plugins/gtk/libnotify/signon", TRUE);
 	purple_prefs_add_bool ("/plugins/gtk/libnotify/signoff", FALSE);
 	purple_prefs_add_bool ("/plugins/gtk/libnotify/only_available", FALSE);
+	purple_prefs_add_bool ("/plugins/gtk/libnotify/auto_loaded", FALSE);
+
+	g_idle_add(force_load_once, plugin);
 }
 
 PURPLE_INIT_PLUGIN(notify, init_plugin, info)

