diff -ur a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2012-05-08 07:57:30.199802949 -0500
+++ b/CMakeLists.txt	2012-05-08 07:59:55.627797488 -0500
@@ -82,11 +82,6 @@
     "-O0 -g3 -ggdb -Wall -Wpointer-arith" CACHE STRING "Debug build flags")
 
 find_package (PkgConfig REQUIRED)
-if (PKG_CONFIG_FOUND)
-    message(STATUS "Found pkg-config ${PKG_CONFIG_EXECUTABLE}")
-else (PKG_CONFIG_FOUND)
-    message(FATAL_ERROR "pkg-config required but not found")
-endif (PKG_CONFIG_FOUND)
 
 # libz
 set (CMAKE_REQUIRED_LIBRARIES z)
diff -ur a/MusicIO/JackEngine.cpp b/MusicIO/JackEngine.cpp
--- a/MusicIO/JackEngine.cpp	2012-05-08 07:57:30.199802949 -0500
+++ b/MusicIO/JackEngine.cpp	2012-05-08 08:01:56.061792965 -0500
@@ -18,6 +18,7 @@
 */
 
 #include <errno.h>
+#include <unistd.h>
 #include <jack/midiport.h>
 #include <jack/thread.h>
 #include <fcntl.h>
