diff -Naur ardour-2.7.1.orig/libs/pbd/convert.cc ardour-2.7.1/libs/pbd/convert.cc
--- ardour-2.7.1.orig/libs/pbd/convert.cc	2008-12-09 15:34:12.544005281 -0500
+++ ardour-2.7.1/libs/pbd/convert.cc	2008-12-09 15:34:47.962006119 -0500
@@ -17,6 +17,7 @@
 
 */
 
+#include <cstdio>
 #include <cmath>
 #include <locale>
 #include <algorithm>
diff -Naur ardour-2.7.1.orig/libs/pbd/pbd/convert.h ardour-2.7.1/libs/pbd/pbd/convert.h
--- ardour-2.7.1.orig/libs/pbd/pbd/convert.h	2008-12-09 15:34:12.546005001 -0500
+++ ardour-2.7.1/libs/pbd/pbd/convert.h	2008-12-09 16:01:25.177005281 -0500
@@ -20,6 +20,7 @@
 #ifndef __pbd_convert_h__
 #define __pbd_convert_h__
 
+#include <stdint.h>
 #include <string>
 #include <vector>
 #include <sstream>
diff -Naur ardour-2.7.1.orig/libs/pbd/shortpath.cc ardour-2.7.1/libs/pbd/shortpath.cc
--- ardour-2.7.1.orig/libs/pbd/shortpath.cc	2008-12-09 15:34:12.550005001 -0500
+++ ardour-2.7.1/libs/pbd/shortpath.cc	2008-12-09 15:44:23.709005001 -0500
@@ -17,6 +17,7 @@
 
 */
 
+#include <stdint.h>
 #include <pbd/shortpath.h>
 
 using namespace Glib;
diff -Naur ardour-2.7.1.orig/libs/pbd/stacktrace.cc ardour-2.7.1/libs/pbd/stacktrace.cc
--- ardour-2.7.1.orig/libs/pbd/stacktrace.cc	2008-12-09 15:34:12.550005001 -0500
+++ ardour-2.7.1/libs/pbd/stacktrace.cc	2008-12-09 15:46:27.671005281 -0500
@@ -17,6 +17,7 @@
 
 */
 
+#include <cstdio>
 #include <pbd/stacktrace.h>
 #include <iostream>
 

