diff -Naur xfce-mcs-plugins-4.4.0/plugins/ui_plugin/ui_plugin.c xfce-mcs-plugins-4.4.0.new/plugins/ui_plugin/ui_plugin.c
--- xfce-mcs-plugins-4.4.0/plugins/ui_plugin/ui_plugin.c	2007-01-20 21:15:31.000000000 +0530
+++ xfce-mcs-plugins-4.4.0.new/plugins/ui_plugin/ui_plugin.c	2007-01-24 18:58:16.000000000 +0530
@@ -117,16 +117,16 @@
 #define RCFILE  "gtk.xml"
 #define CHANNEL "settings"
 #define PLUGIN_NAME "ui"
-#define DEFAULT_THEME "Default"
-#define INITIAL_THEME "Xfce"
-#define DEFAULT_ICON_THEME "hicolor"
-#define INITIAL_ICON_THEME "Rodent"
+#define DEFAULT_THEME "Frugalware"
+#define INITIAL_THEME "Frugalware"
+#define DEFAULT_ICON_THEME "Tango"
+#define INITIAL_ICON_THEME "Tango"
 #define DEFAULT_TOOLBAR_STYLE "icons"
-#define DEFAULT_FONT "Sans 9"
+#define DEFAULT_FONT "Sans 12"
 #define DEFAULT_XFT_ANTIALIAS 1
 #define DEFAULT_XFT_HINTING 1
 #define DEFAULT_XFT_HINTSTYLE "hintfull"
-#define DEFAULT_XFT_RGBA "none"
+#define DEFAULT_XFT_RGBA "vrgb"
 
 #define MAX_ELEMENTS_BEFORE_SCROLLING 6
 #define BORDER 5
