diff -urN xfce-mcs-plugins-4.3.99.1.orig/plugins/ui_plugin/ui_plugin.c xfce-mcs-plugins-4.3.99.1/plugins/ui_plugin/ui_plugin.c
--- xfce-mcs-plugins-4.3.99.1.orig/plugins/ui_plugin/ui_plugin.c	2006-09-23 11:50:38.000000000 +0100
+++ xfce-mcs-plugins-4.3.99.1/plugins/ui_plugin/ui_plugin.c	2006-09-23 11:52:22.000000000 +0100
@@ -117,12 +117,12 @@
 #define RCFILE  "gtk.xml"
 #define CHANNEL "settings"
 #define PLUGIN_NAME "ui"
-#define DEFAULT_THEME "Default"
-#define INITIAL_THEME "Xfce"
-#define DEFAULT_ICON_THEME "hicolor"
-#define INITIAL_ICON_THEME "Rodent"
+#define DEFAULT_THEME "Frugalware"
+#define INITIAL_THEME "Frugalware"
+#define DEFAULT_ICON_THEME "Tango"
+#define INITIAL_ICON_THEME "Tango"
 #define DEFAULT_TOOLBAR_STYLE "icons"
-#define DEFAULT_FONT "Sans 9"
+#define DEFAULT_FONT "Sans 12"
 #define DEFAULT_XFT_ANTIALIAS 1
 #define DEFAULT_XFT_HINTING 1
 #define DEFAULT_XFT_HINTSTYLE "hintfull"
