diff -Naur xfce-mcs-plugins-4.3.99.1.orig/plugins/ui_plugin/ui_plugin.c xfce-mcs-plugins-4.3.99.1/plugins/ui_plugin/ui_plugin.c
--- xfce-mcs-plugins-4.3.99.1.orig/plugins/ui_plugin/ui_plugin.c	2006-09-02 20:48:47.000000000 +0530
+++ xfce-mcs-plugins-4.3.99.1/plugins/ui_plugin/ui_plugin.c	2006-09-05 12:43:07.000000000 +0530
@@ -117,10 +117,10 @@
 #define RCFILE  "gtk.xml"
 #define CHANNEL "settings"
 #define PLUGIN_NAME "ui"
-#define DEFAULT_THEME "Default"
-#define INITIAL_THEME "Xfce"
-#define DEFAULT_ICON_THEME "hicolor"
-#define INITIAL_ICON_THEME "Rodent"
+#define DEFAULT_THEME "Frugalware"
+#define INITIAL_THEME "Frugalware"
+#define DEFAULT_ICON_THEME "Tango"
+#define INITIAL_ICON_THEME "Tango"
 #define DEFAULT_TOOLBAR_STYLE "icons"
 #define DEFAULT_FONT "Sans 9"
 #define DEFAULT_XFT_ANTIALIAS 1
