diff --git a/src/ast_2dtool.c b/src/ast_2dtool.c
index bb99f9d..684e400 100644
--- a/src/ast_2dtool.c
+++ b/src/ast_2dtool.c
@@ -27,7 +27,6 @@
 #include "xf86_OSproc.h"
 #include "xf86cmap.h"
 #include "compiler.h"
-#include "mibstore.h"
 #include "vgaHW.h"
 #include "mipointer.h"
 #include "micmap.h"
@@ -36,7 +35,6 @@
 #include "regionstr.h"
 #include "xf86xv.h"
 #include <X11/extensions/Xv.h>
-#include "vbe.h"
 
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
diff --git a/src/ast_accel.c b/src/ast_accel.c
index 34f9c65..0e7e5d3 100644
--- a/src/ast_accel.c
+++ b/src/ast_accel.c
@@ -27,7 +27,6 @@
 #include "xf86_OSproc.h"
 #include "xf86cmap.h"
 #include "compiler.h"
-#include "mibstore.h"
 #include "vgaHW.h"
 #include "mipointer.h"
 #include "micmap.h"
@@ -36,7 +35,6 @@
 #include "regionstr.h"
 #include "xf86xv.h"
 #include <X11/extensions/Xv.h>
-#include "vbe.h"
 
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
diff --git a/src/ast_cursor.c b/src/ast_cursor.c
index f967ce9..c672564 100644
--- a/src/ast_cursor.c
+++ b/src/ast_cursor.c
@@ -27,7 +27,6 @@
 #include "xf86_OSproc.h"
 #include "xf86cmap.h"
 #include "compiler.h"
-#include "mibstore.h"
 #include "vgaHW.h"
 #include "mipointer.h"
 #include "micmap.h"
@@ -36,7 +35,6 @@
 #include "regionstr.h"
 #include "xf86xv.h"
 #include <X11/extensions/Xv.h>
-#include "vbe.h"
 
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
diff --git a/src/ast_driver.c b/src/ast_driver.c
index 080b051..43fcdbc 100644
--- a/src/ast_driver.c
+++ b/src/ast_driver.c
@@ -31,7 +31,6 @@
 #endif
 #include "xf86cmap.h"
 #include "compiler.h"
-#include "mibstore.h"
 #include "vgaHW.h"
 #include "mipointer.h"
 #include "micmap.h"
@@ -40,7 +39,6 @@
 #include "regionstr.h"
 #include "xf86xv.h"
 #include <X11/extensions/Xv.h>
-#include "vbe.h"
 
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
@@ -889,7 +887,6 @@ ASTScreenInit(SCREEN_INIT_ARGS_DECL)
    }
 #endif /* end of Accel_2D */
      
-   miInitializeBackingStore(pScreen);
    xf86SetBackingStore(pScreen);
    xf86SetSilkenMouse(pScreen);
 
diff --git a/src/ast_mode.c b/src/ast_mode.c
index 3238f39..2e06062 100644
--- a/src/ast_mode.c
+++ b/src/ast_mode.c
@@ -27,7 +27,6 @@
 #include "xf86_OSproc.h"
 #include "xf86cmap.h"
 #include "compiler.h"
-#include "mibstore.h"
 #include "vgaHW.h"
 #include "mipointer.h"
 #include "micmap.h"
@@ -36,7 +35,6 @@
 #include "regionstr.h"
 #include "xf86xv.h"
 #include <X11/extensions/Xv.h>
-#include "vbe.h"
 
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
diff --git a/src/ast_tool.c b/src/ast_tool.c
index d11f20d..c2bfac6 100644
--- a/src/ast_tool.c
+++ b/src/ast_tool.c
@@ -27,7 +27,6 @@
 #include "xf86_OSproc.h"
 #include "xf86cmap.h"
 #include "compiler.h"
-#include "mibstore.h"
 #include "vgaHW.h"
 #include "mipointer.h"
 #include "micmap.h"
@@ -36,7 +35,6 @@
 #include "regionstr.h"
 #include "xf86xv.h"
 #include <X11/extensions/Xv.h>
-#include "vbe.h"
 
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
diff --git a/src/ast_vgatool.c b/src/ast_vgatool.c
index ed8c906..378822a 100644
--- a/src/ast_vgatool.c
+++ b/src/ast_vgatool.c
@@ -27,7 +27,6 @@
 #include "xf86_OSproc.h"
 #include "xf86cmap.h"
 #include "compiler.h"
-#include "mibstore.h"
 #include "vgaHW.h"
 #include "mipointer.h"
 #include "micmap.h"
@@ -36,7 +35,6 @@
 #include "regionstr.h"
 #include "xf86xv.h"
 #include <X11/extensions/Xv.h>
-#include "vbe.h"
 
 #include "xf86PciInfo.h"
 #include "xf86Pci.h"
@@ -2711,7 +2709,7 @@ void vGetDefaultSettings(ScrnInfoPtr pScrn)
 Bool InitVGA(ScrnInfoPtr pScrn, ULONG Flags)
 {
    ASTRecPtr pAST;
-   ULONG ulData;
+   uint32_t ulData;
 
    pAST = ASTPTR(pScrn);
 
@@ -3007,7 +3005,7 @@ GetVGAEDID(ScrnInfoPtr pScrn, unsigned char *pEDIDBuffer)
 Bool bInitAST1180(ScrnInfoPtr pScrn)
 {
     ASTRecPtr pAST;	
-    ULONG ulData;
+    uint32_t ulData;
 
     pAST = ASTPTR(pScrn);    
 
