diff -Naur lxqt-build-tools-0.5.0/cmake/modules/LXQtCreatePkgConfigFile.cmake lxqt-build-tools-0.5.0-p/cmake/modules/LXQtCreatePkgConfigFile.cmake
--- lxqt-build-tools-0.5.0/cmake/modules/LXQtCreatePkgConfigFile.cmake	2018-05-21 19:06:51.000000000 +0200
+++ lxqt-build-tools-0.5.0-p/cmake/modules/LXQtCreatePkgConfigFile.cmake	2018-06-15 01:39:28.496985584 +0200
@@ -104,7 +104,7 @@
     endif()
 
     if (NOT DEFINED USER_INCLUDEDIR_PREFIX)
-        set(_PKGCONFIG_INCLUDEDIR_PREFIX "\${prefix}/${CMAKE_INSTALL_INCLUDEDIR}")
+        set(_PKGCONFIG_INCLUDEDIR_PREFIX "\${prefix}/include")
     endif()
 
     if (NOT DEFINED USER_LIBDIR_PREFIX)
