--- frameworkintegration-5.112.0/CMakeLists.txt	2023-11-04 10:59:53.000000000 +0100
+++ frameworkintegration-5.112.0-new/CMakeLists.txt	2023-11-23 23:39:06.070000811 +0100
@@ -47,7 +47,7 @@
    find_package(KF5I18n ${KF_DEP_VERSION} REQUIRED)
 
    find_package(packagekitqt${QT_MAJOR_VERSION})
-   find_package(AppStreamQt 0.10.4)
+   find_package(AppStreamQt5 1.0.0)
 endif()
 
 ecm_set_disabled_deprecation_versions(
