diff -urNad --exclude=CVS --exclude=.svn ./src/AIModel/AIBase.cxx /tmp/dpep-work.zJXVy8/flightgear-0.9.8/src/AIModel/AIBase.cxx
--- ./src/AIModel/AIBase.cxx	2005-09-11 16:01:29.000000000 +0200
+++ /tmp/dpep-work.zJXVy8/flightgear-0.9.8/src/AIModel/AIBase.cxx	2005-09-11 16:51:34.109727904 +0200
@@ -386,5 +386,5 @@
 }
 
 int FGAIBase::_getID() const {
-    return (int)(this);
+    return (long)(this);
 }
diff -urNad --exclude=CVS --exclude=.svn ./src/Scripting/NasalSys.cxx /tmp/dpep-work.zJXVy8/flightgear-0.9.8/src/Scripting/NasalSys.cxx
--- ./src/Scripting/NasalSys.cxx	2004-11-15 21:12:03.000000000 +0100
+++ /tmp/dpep-work.zJXVy8/flightgear-0.9.8/src/Scripting/NasalSys.cxx	2005-09-11 16:52:21.066589376 +0200
@@ -86,7 +86,7 @@
 
     char buf[256];
     if(!name) {
-        sprintf(buf, "FGNasalScript@%8.8x", (int)script);
+        sprintf(buf, "FGNasalScript@%8.8x", (long)script);
         name = buf;
     }
 
