diff -Naur bip-0.8.9/src/log.c bip-0.8.9-p/src/log.c
--- bip-0.8.9/src/log.c	2012-01-25 11:23:16.000000000 +0100
+++ bip-0.8.9-p/src/log.c	2018-02-09 21:39:46.528973091 +0100
@@ -439,8 +439,8 @@
 	log_write(logdata, channel, logdata->buffer);
 }
 
-void log_part(log_t *logdata, const const char *ircmask, const char *channel,
-		const const char *message)
+void log_part(log_t *logdata, const char *ircmask, const char *channel,
+		const char *message)
 {
 	if (message)
 		snprintf(logdata->buffer, LOGLINE_MAXLEN,
