--- spirv-headers/CMakeLists.txt	2021-12-30 18:27:42.907104682 +0100
+++ spirv-headers.new/CMakeLists.txt	2021-12-30 18:28:09.294666509 +0100
@@ -128,6 +128,6 @@
     configure_file(${CMAKE_CURRENT_SOURCE_DIR}/SPIRV-Headers.pc.in ${CMAKE_BINARY_DIR}/SPIRV-Headers.pc @ONLY)
     install(
         FILES "${CMAKE_BINARY_DIR}/SPIRV-Headers.pc"
-        DESTINATION ${CMAKE_INSTALL_DATADIR}/pkgconfig
+	DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig
     )
 endif()
