--- extensions/Makefile.am	2009-04-17 13:05:15.000000000 -0400
+++ extensions/Makefile.am.new	2009-06-02 16:39:38.659858086 -0400
@@ -1,7 +1,7 @@
 if OS_WIN32
 AM_LDFLAGS = -module -no-undefined
 else
-AM_LDFLAGS = -module
+AM_LDFLAGS = -module -avoid-version
 endif
 
 noinst_HEADERS = util.h
@@ -12,7 +12,7 @@
 	-I$(top_srcdir)/babl		\
 	-I$(top_srcdir)/extensions
 
-extdir = $(DESTDIR)$(libdir)/babl-@BABL_API_VERSION@
+extdir = $(libdir)/babl-@BABL_API_VERSION@
 ext_LTLIBRARIES = \
 	CIE-Lab.la      \
 	gegl-fixups.la  \
--- extensions/Makefile.in	2009-05-20 16:42:51.000000000 -0400
+++ extensions/Makefile.in.new	2009-06-02 16:47:39.090987339 -0400
@@ -50,7 +50,7 @@
     *) f=$$p;; \
   esac;
 am__strip_dir = `echo $$p | sed -e 's|^.*/||'`;
-am__installdirs = "$(DESTDIR)$(extdir)"
+am__installdirs = "$(extdir)"
 extLTLIBRARIES_INSTALL = $(INSTALL)
 LTLIBRARIES = $(ext_LTLIBRARIES)
 CIE_Lab_la_LIBADD =
@@ -233,7 +233,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-@OS_WIN32_FALSE@AM_LDFLAGS = -module
+@OS_WIN32_FALSE@AM_LDFLAGS = -module -avoid-version
 @OS_WIN32_TRUE@AM_LDFLAGS = -module -no-undefined
 noinst_HEADERS = util.h
 AM_CPPFLAGS = \
@@ -242,7 +242,7 @@
 	-I$(top_srcdir)/babl		\
 	-I$(top_srcdir)/extensions
 
-extdir = $(DESTDIR)$(libdir)/babl-@BABL_API_VERSION@
+extdir = $(libdir)/babl-@BABL_API_VERSION@
 ext_LTLIBRARIES = \
 	CIE-Lab.la      \
 	gegl-fixups.la  \
