From d9e34c59faf63162f18dc87f599e36de024f557c Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Tue, 28 Jul 2009 03:32:35 +0000
Subject: rendition: update for resources/RAC API removal

---
diff --git a/src/rendition.c b/src/rendition.c
index db62f8e..7a9e18b 100644
--- a/src/rendition.c
+++ b/src/rendition.c
@@ -538,7 +538,6 @@ renditionPreInit(ScrnInfoPtr pScreenInfo, int flags)
     pRendition->PciInfo = xf86GetPciInfoForEntity(pRendition->pEnt->index);
     pRendition->pcitag= pciTag(pRendition->PciInfo->bus,
                pRendition->PciInfo->device, pRendition->PciInfo->func);
-#endif
 
     /*
      * XXX This could be refined if some VGA memory resources are not
@@ -549,11 +548,12 @@ renditionPreInit(ScrnInfoPtr pScreenInfo, int flags)
     if (xf86RegisterResources(pRendition->pEnt->index, NULL, ResExclusive))
          return FALSE;
 
+
     /* Operations for which memory access is required. */
     pScreenInfo->racMemFlags = RAC_FB | RAC_CURSOR;
     /* Operations for which I/O access is required. (XXX Check this) */
     pScreenInfo->racIoFlags = RAC_FB | RAC_COLORMAP | RAC_CURSOR | RAC_VIEWPORT;
-    
+#endif
     /* determine depth, bpp, etc. */
     if (!xf86SetDepthBpp(pScreenInfo, 0, 0, 0, Support32bppFb))
         return FALSE;
diff --git a/src/rendition.h b/src/rendition.h
index 2b43cba..84809fa 100644
--- a/src/rendition.h
+++ b/src/rendition.h
@@ -37,11 +37,13 @@
 /* All drivers using the mi colormap manipulation need this */
 #include "micmap.h"
 
+#ifndef XSERVER_LIBPCIACCESS
 /* Needed by Resources Access Control (RAC) */
 #include "xf86RAC.h"
 
 /* Several predefined resource lists */
 #include "xf86Resources.h"
+#endif
 
 /* Needed by the Shadow Framebuffer */
 #include "shadowfb.h"
--
cgit v0.8.2

