diff -Naur VirtualBox-5.2.0/src/VBox/Installer/linux/Makefile.include.footer VirtualBox-5.2.0-p/src/VBox/Installer/linux/Makefile.include.footer
--- VirtualBox-5.2.0/src/VBox/Installer/linux/Makefile.include.footer	2017-10-18 09:12:51.000000000 +0200
+++ VirtualBox-5.2.0-p/src/VBox/Installer/linux/Makefile.include.footer	2017-10-19 21:59:33.607843277 +0200
@@ -63,8 +63,7 @@
 
 install: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
-	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR); \
-	PATH="$(PATH):/bin:/sbin" depmod -a;
+	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR);
 
 clean:
 	for f in $(sort $(dir $(OBJS))); do rm -f $$f/*.o $$f/.*.cmd $$f/.*.flags; done
