diff -Naur VirtualBox-6.0.0/src/VBox/Installer/linux/Makefile.include.footer VirtualBox-6.0.0-p/src/VBox/Installer/linux/Makefile.include.footer
--- VirtualBox-6.0.0/src/VBox/Installer/linux/Makefile.include.footer	2018-12-18 14:00:02.000000000 +0100
+++ VirtualBox-6.0.0-p/src/VBox/Installer/linux/Makefile.include.footer	2018-12-19 01:00:18.376934280 +0100
@@ -72,8 +72,7 @@
 
 install: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
-	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR); \
-	PATH="$(PATH):/bin:/sbin" depmod -a; sync
+	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR); sync
 
 clean:
 	for f in $(sort $(dir $(OBJS))); do rm -f $$f/*.o $$f/.*.cmd $$f/.*.flags; done
