diff -Naur VirtualBox-6.0.6/src/VBox/Installer/linux/Makefile-footer.gmk VirtualBox-6.0.6-p/src/VBox/Installer/linux/Makefile-footer.gmk
--- VirtualBox-6.0.6/src/VBox/Installer/linux/Makefile-footer.gmk	2019-04-16 12:16:46.000000000 +0200
+++ VirtualBox-6.0.6-p/src/VBox/Installer/linux/Makefile-footer.gmk	2019-04-23 11:42:58.552143970 +0200
@@ -70,8 +70,7 @@
 
 install: $(VBOXMOD_0_TARGET)
 	@mkdir -p $(MODULE_DIR); \
-	install -m 0644 -o root -g root $(VBOXMOD_0_TARGET).$(VBOXMOD_EXT) $(MODULE_DIR); \
-	PATH="$(PATH):/bin:/sbin" depmod -a; sync
+	install -m 0644 -o root -g root $(VBOXMOD_0_TARGET).$(VBOXMOD_EXT) $(MODULE_DIR); sync
 
 clean:
 	for f in $(sort $(dir $(VBOXMOD_OBJS))); do rm -f $$f/*.o $$f/.*.cmd $$f/.*.flags; done
