diff -Naur VirtualBox-5.2.20/src/VBox/Additions/linux/drm/vbox_mode.c VirtualBox-5.2.20-p/src/VBox/Additions/linux/drm/vbox_mode.c
--- VirtualBox-5.2.20/src/VBox/Additions/linux/drm/vbox_mode.c	2018-10-15 16:25:07.000000000 +0200
+++ VirtualBox-5.2.20-p/src/VBox/Additions/linux/drm/vbox_mode.c	2018-10-23 01:01:37.023700374 +0200
@@ -495,9 +495,11 @@
 	drm_connector_update_edid_property(connector, (struct edid *)edid);
 #else
 	drm_mode_connector_update_edid_property(connector, (struct edid *)edid);
-}
+
 #endif
 
+}
+
 static int vbox_get_modes(struct drm_connector *connector)
 {
 	struct vbox_connector *vbox_connector = NULL;
