Index: configure
===================================================================
--- configure	(revision 184)
+++ configure	(working copy)
@@ -36,9 +36,9 @@
 GENISOIMAGE="genisoimage"
 MKISOFS="mkisofs"
 INCXALAN=""
-LIBXALAN="-lxalan-c"
+LIBXALAN="-lxalan-c -pthread"
 INCXERCES=""
-LIBXERCES="-lxerces-c"
+LIBXERCES="-lxerces-c -pthread"
 LIBUUID="-luuid"
 INCSDL="/usr/include/SDL"
 LIBSDL="-lSDL"
@@ -51,7 +51,7 @@
 LIBZ="-lz"
 INCPNG=""
 LIBPNG="-lpng"
-QTDIR="/usr/qt/3 /usr/share/qt3"
+QTDIR="/usr/lib/qt"
 KBUILDDIR="`cd $(dirname $0); pwd`/kBuild"
 DEVDIR="`cd $(dirname $0); pwd`/tools"
 if [ -d /lib/modules/`uname -r`/build ]; then
