diff -Naur VirtualBox-6.1.12/src/VBox/Additions/common/VBoxGuest/linux/Makefile VirtualBox-6.1.12-p/src/VBox/Additions/common/VBoxGuest/linux/Makefile
--- VirtualBox-6.1.12/src/VBox/Additions/common/VBoxGuest/linux/Makefile	2020-07-11 10:03:29.000000000 +0200
+++ VirtualBox-6.1.12-p/src/VBox/Additions/common/VBoxGuest/linux/Makefile	2020-08-22 15:13:03.307887161 +0200
@@ -29,7 +29,7 @@
 include $(obj)/Makefile-header.gmk
 VBOXGUEST_DIR = $(VBOX_MODULE_SRC_DIR)
 
-#VBOX_WITHOUT_COMBINED_SOURCES=1
+VBOX_WITHOUT_COMBINED_SOURCES=1
 
 VBOXMOD_NAME = vboxguest
 VBOXMOD_OBJS = \
diff -Naur VirtualBox-6.1.12/src/VBox/Additions/linux/Makefile VirtualBox-6.1.12-p/src/VBox/Additions/linux/Makefile
--- VirtualBox-6.1.12/src/VBox/Additions/linux/Makefile	2020-07-11 10:03:31.000000000 +0200
+++ VirtualBox-6.1.12-p/src/VBox/Additions/linux/Makefile	2020-08-22 15:13:34.187098087 +0200
@@ -14,7 +14,7 @@
 # hope that it will be useful, but WITHOUT ANY WARRANTY of any kind.
 #
 
-ifneq ($(KERNELRELEASE),)
+ifneq ($(KBUILD_EXTMOD),)
 
 # Building from kBuild (make -C <kernel_directory> M=`pwd`)
 # or inside a kernel source tree.
@@ -31,7 +31,7 @@
 VBOX_QUIET    :=
 VBOX_QUIET_SH := set -x;
  endif
-
+PWD ?= $(shell pwd)
 all: vboxguest vboxsf vboxvideo
 
 vboxguest:
@@ -49,8 +49,9 @@
 	    if [ -f vboxguest/Module.symvers ]; then \
 	        cp vboxguest/Module.symvers vboxsf; \
 	    fi; \
+        export KBUILD_EXTRA_SYMBOLS=${PWD}/vboxsf/Module.symvers; \
 	    echo "=== Building 'vboxsf' module ==="; \
-	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) KBUILD_EXTRA_SYMBOLS=$(abspath vboxsf/Module.symvers) -C vboxsf || exit 1; \
+	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxsf || exit 1; \
 	    if [ -f vboxsf/vboxsf.ko ]; then \
 	        cp vboxsf/vboxsf.ko .; \
 	    else \
@@ -64,8 +65,9 @@
 	    if [ -f vboxguest/Module.symvers ]; then \
 	        cp vboxguest/Module.symvers vboxvideo; \
 	    fi; \
+        export KBUILD_EXTRA_SYMBOLS=${PWD}/vboxvideo/Module.symvers; \
 	    echo "=== Building 'vboxvideo' module ==="; \
-	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) KBUILD_EXTRA_SYMBOLS=$(abspath vboxvideo/Module.symvers) -C vboxvideo || exit 1; \
+	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxvideo || exit 1; \
 	    if [ -f vboxvideo/vboxvideo.ko ]; then \
 	        cp vboxvideo/vboxvideo.ko .; \
 	    elif [ -f vboxvideo/vboxvideo.o ]; then \
