--- VirtualBox-6.0.4/src/VBox/Additions/linux/sharedfolders/vfsmod.c	2019-01-25 19:12:35.000000000 +0100
+++ VirtualBox-6.0.4.new/src/VBox/Additions/linux/sharedfolders/vfsmod.c	2019-03-04 13:41:34.990552909 +0100
@@ -40,6 +40,9 @@
 #include "revision-generated.h"
 #include "product-generated.h"
 #include "VBoxGuestR0LibInternal.h"
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 0, 0)
+#include <uapi/linux/mount.h>
+#endif
 #if LINUX_VERSION_CODE < KERNEL_VERSION(3, 3, 0)
 # include <linux/mount.h>
 #endif
