Index: src/VBox/Additions/linux/sharedfolders/utils.c
===================================================================
--- a/src/VBox/Additions/linux/sharedfolders/utils.c
+++ b/src/VBox/Additions/linux/sharedfolders/utils.c
@@ -1163,5 +1163,9 @@
  * @note Caller holds no relevant locks, just a dentry reference.
  */
-#if RTLNX_VER_MIN(3,6,0)
+
+#if RTLNX_VER_MIN(6,14,0)
+static int vbsf_dentry_revalidate(struct inode *parentDir, const struct qstr *parentName,
+                                  struct dentry *dentry, unsigned flags)
+#elif RTLNX_VER_MIN(3,6,0)
 static int vbsf_dentry_revalidate(struct dentry *dentry, unsigned flags)
 #elif RTLNX_VER_MIN(2,6,0)
@@ -1171,4 +1175,8 @@
 #endif
 {
+#if RTLNX_VER_MIN(6,14,0)
+    /* Optional, not in use currently. */
+    RT_NOREF(parentDir, parentName);
+#endif
 #if RTLNX_VER_RANGE(2,6,0,  3,6,0)
     int const flags = nd ? nd->flags : 0;

