--- vboxguest/Makefile.include.footer	2017-04-28 17:04:21.000000000 +0200
+++ vboxguest.new/Makefile.include.footer	2017-05-14 22:11:48.000000000 +0200
@@ -92,7 +92,6 @@
 install: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
 	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR); \
-	PATH="$(PATH):/bin:/sbin" depmod -a;
 
 endif # eq($(MAKECMDGOALS),clean)
 
diff -Naur vboxsf/Makefile.include.footer vboxsf.new/Makefile.include.footer
--- vboxsf/Makefile.include.footer	2017-04-28 17:04:21.000000000 +0200
+++ vboxsf.new/Makefile.include.footer	2017-05-14 22:11:48.000000000 +0200
@@ -92,7 +92,6 @@
 install: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
 	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR); \
-	PATH="$(PATH):/bin:/sbin" depmod -a;
 
 endif # eq($(MAKECMDGOALS),clean)
 
diff -Naur vboxvideo/Makefile.include.footer vboxvideo.new/Makefile.include.footer
--- vboxvideo/Makefile.include.footer	2017-04-28 17:04:21.000000000 +0200
+++ vboxvideo.new/Makefile.include.footer	2017-05-14 22:11:48.000000000 +0200
@@ -92,7 +92,6 @@
 install: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
 	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR); \
-	PATH="$(PATH):/bin:/sbin" depmod -a;
 
 endif # eq($(MAKECMDGOALS),clean)
 
