diff -Naur VirtualBox-6.1.12/src/VBox/Additions/linux/sharedfolders/vfsmod.c VirtualBox-6.1.12-p/src/VBox/Additions/linux/sharedfolders/vfsmod.c
--- VirtualBox-6.1.12/src/VBox/Additions/linux/sharedfolders/vfsmod.c	2020-07-11 10:03:32.000000000 +0200
+++ VirtualBox-6.1.12-p/src/VBox/Additions/linux/sharedfolders/vfsmod.c	2020-08-14 14:29:19.676072075 +0200
@@ -53,8 +53,10 @@
 #include <linux/seq_file.h>
 #include <linux/vfs.h>
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2, 5, 62)
+# if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
 # include <linux/vermagic.h>
 #endif
+#endif
 #include <VBox/err.h>
 #include <iprt/path.h>
 
diff -Naur VirtualBox-6.1.12/src/VBox/HostDrivers/Support/linux/SUPDrv-linux.c VirtualBox-6.1.12-p/src/VBox/HostDrivers/Support/linux/SUPDrv-linux.c
--- VirtualBox-6.1.12/src/VBox/HostDrivers/Support/linux/SUPDrv-linux.c	2020-07-11 10:10:43.000000000 +0200
+++ VirtualBox-6.1.12-p/src/VBox/HostDrivers/Support/linux/SUPDrv-linux.c	2020-08-15 00:09:47.484697930 +0200
@@ -756,21 +756,17 @@
 
 RTCCUINTREG VBOXCALL supdrvOSChangeCR4(RTCCUINTREG fOrMask, RTCCUINTREG fAndMask)
 {
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 20, 0)
-    RTCCUINTREG uOld = this_cpu_read(cpu_tlbstate.cr4);
-    RTCCUINTREG uNew = (uOld & fAndMask) | fOrMask;
-    if (uNew != uOld)
-    {
-        this_cpu_write(cpu_tlbstate.cr4, uNew);
-        __write_cr4(uNew);
-    }
-#else
-    RTCCUINTREG uOld = ASMGetCR4();
-    RTCCUINTREG uNew = (uOld & fAndMask) | fOrMask;
-    if (uNew != uOld)
-        ASMSetCR4(uNew);
-#endif
-    return uOld;
+	 /* only works on 5.8+ */
+	unsigned long fSavedFlags;
+
+	local_irq_save(fSavedFlags);
+	RTCCUINTREG const uOld = cr4_read_shadow();
+	cr4_update_irqsoff(fOrMask, ~fAndMask); /* Same as this function, only it is not returning the old value. */
+	AssertMsg(cr4_read_shadow() == ((uOld & fAndMask) | fOrMask),
+                        ("fOrMask=%#RTreg fAndMask=%#RTreg uOld=%#RTreg; new cr4=%#llx\n", fOrMask, fAndMask, uOld, cr4_read_shadow()));
+	local_irq_restore(fSavedFlags);
+
+	return uOld;
 }
 
 
diff -Naur VirtualBox-6.1.12/src/VBox/Runtime/r0drv/linux/alloc-r0drv-linux.c VirtualBox-6.1.12-p/src/VBox/Runtime/r0drv/linux/alloc-r0drv-linux.c
--- VirtualBox-6.1.12/src/VBox/Runtime/r0drv/linux/alloc-r0drv-linux.c	2020-07-11 10:11:26.000000000 +0200
+++ VirtualBox-6.1.12-p/src/VBox/Runtime/r0drv/linux/alloc-r0drv-linux.c	2020-08-14 14:22:37.796941099 +0200
@@ -153,6 +153,8 @@
 
 
 #ifdef RTMEMALLOC_EXEC_VM_AREA
+
+
 /**
  * Allocate executable kernel memory in the module range.
  *
@@ -168,7 +170,12 @@
     struct vm_struct   *pVmArea;
     size_t              iPage;
 
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 8, 0)
+    pVmArea = __get_vm_area_caller(cbAlloc, VM_ALLOC, MODULES_VADDR, MODULES_END,
+                                   __builtin_return_address(0));
+#else
     pVmArea = __get_vm_area(cbAlloc, VM_ALLOC, MODULES_VADDR, MODULES_END);
+#endif
     if (!pVmArea)
         return NULL;
     pVmArea->nr_pages = 0;    /* paranoia? */
@@ -201,6 +208,12 @@
 # endif
         pVmArea->nr_pages = cPages;
         pVmArea->pages    = papPages;
+# if LINUX_VERSION_CODE >= KERNEL_VERSION(5, 8, 0)
+        unsigned long start = (unsigned long)pVmArea->addr;
+        unsigned long size = get_vm_area_size(pVmArea);
+
+        if (!map_kernel_range(start, size, PAGE_KERNEL_EXEC, papPages))
+#else
         if (!map_vm_area(pVmArea, PAGE_KERNEL_EXEC,
 # if LINUX_VERSION_CODE < KERNEL_VERSION(3, 17, 0)
                          &papPagesIterator
@@ -208,6 +221,7 @@
                          papPages
 # endif
                          ))
+#endif
         {
             PRTMEMLNXHDREX pHdrEx = (PRTMEMLNXHDREX)pVmArea->addr;
             pHdrEx->pVmArea     = pVmArea;
diff -Naur VirtualBox-6.1.12/src/VBox/Runtime/r0drv/linux/memobj-r0drv-linux.c VirtualBox-6.1.12-p/src/VBox/Runtime/r0drv/linux/memobj-r0drv-linux.c
--- VirtualBox-6.1.12/src/VBox/Runtime/r0drv/linux/memobj-r0drv-linux.c	2020-07-11 10:11:26.000000000 +0200
+++ VirtualBox-6.1.12-p/src/VBox/Runtime/r0drv/linux/memobj-r0drv-linux.c	2020-08-14 14:22:31.100239263 +0200
@@ -222,9 +222,17 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 5, 0)
         ulAddr = vm_mmap(NULL, R3PtrFixed, cb, fLnxProt, MAP_SHARED | MAP_ANONYMOUS | MAP_FIXED, 0);
 #else
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
         down_write(&pTask->mm->mmap_sem);
+#else
+        down_write(&pTask->mm->mmap_lock);
+#endif
         ulAddr = do_mmap(NULL, R3PtrFixed, cb, fLnxProt, MAP_SHARED | MAP_ANONYMOUS | MAP_FIXED, 0);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
         up_write(&pTask->mm->mmap_sem);
+#else
+        up_write(&pTask->mm->mmap_lock);
+#endif
 #endif
     }
     else
@@ -232,9 +240,17 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 5, 0)
         ulAddr = vm_mmap(NULL, 0, cb, fLnxProt, MAP_SHARED | MAP_ANONYMOUS, 0);
 #else
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
         down_write(&pTask->mm->mmap_sem);
+#else
+        down_write(&pTask->mm->mmap_lock);
+#endif
         ulAddr = do_mmap(NULL, 0, cb, fLnxProt, MAP_SHARED | MAP_ANONYMOUS, 0);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
         up_write(&pTask->mm->mmap_sem);
+#else
+        up_write(&pTask->mm->mmap_lock);
+#endif
 #endif
         if (    !(ulAddr & ~PAGE_MASK)
             &&  (ulAddr & (uAlignment - 1)))
@@ -269,13 +285,29 @@
     Assert(pTask == current); RT_NOREF_PV(pTask);
     vm_munmap((unsigned long)pv, cb);
 #elif defined(USE_RHEL4_MUNMAP)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
     down_write(&pTask->mm->mmap_sem);
+#else
+    down_write(&pTask->mm->mmap_lock);
+#endif
     do_munmap(pTask->mm, (unsigned long)pv, cb, 0); /* should it be 1 or 0? */
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
     up_write(&pTask->mm->mmap_sem);
 #else
+    up_write(&pTask->mm->mmap_lock);
+#endif
+#else
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
     down_write(&pTask->mm->mmap_sem);
+#else
+    down_write(&pTask->mm->mmap_lock);
+#endif
     do_munmap(pTask->mm, (unsigned long)pv, cb);
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
     up_write(&pTask->mm->mmap_sem);
+#else
+    up_write(&pTask->mm->mmap_lock);
+#endif
 #endif
 }
 
@@ -593,7 +625,11 @@
                 size_t              iPage;
                 Assert(pTask);
                 if (pTask && pTask->mm)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
                     down_read(&pTask->mm->mmap_sem);
+#else
+                    down_read(&pTask->mm->mmap_lock);
+#endif
 
                 iPage = pMemLnx->cPages;
                 while (iPage-- > 0)
@@ -608,7 +644,11 @@
                 }
 
                 if (pTask && pTask->mm)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
                     up_read(&pTask->mm->mmap_sem);
+#else
+                    up_read(&pTask->mm->mmap_lock);
+#endif
             }
             /* else: kernel memory - nothing to do here. */
             break;
@@ -1076,7 +1116,11 @@
     papVMAs = (struct vm_area_struct **)RTMemAlloc(sizeof(*papVMAs) * cPages);
     if (papVMAs)
     {
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
         down_read(&pTask->mm->mmap_sem);
+#else
+        down_read(&pTask->mm->mmap_lock);
+#endif
 
         /*
          * Get user pages.
@@ -1162,7 +1206,11 @@
                 papVMAs[rc]->vm_flags |= VM_DONTCOPY | VM_LOCKED;
             }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
             up_read(&pTask->mm->mmap_sem);
+#else
+            up_read(&pTask->mm->mmap_lock);
+#endif
 
             RTMemFree(papVMAs);
 
@@ -1189,7 +1237,11 @@
 #endif
         }
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
         up_read(&pTask->mm->mmap_sem);
+#else
+        up_read(&pTask->mm->mmap_lock);
+#endif
 
         RTMemFree(papVMAs);
         rc = VERR_LOCK_FAILED;
@@ -1604,7 +1656,11 @@
             const size_t    cPages    = (offSub + cbSub) >> PAGE_SHIFT;
             size_t          iPage;
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
             down_write(&pTask->mm->mmap_sem);
+#else
+            down_write(&pTask->mm->mmap_lock);
+#endif
 
             rc = VINF_SUCCESS;
             if (pMemLnxToMap->cPages)
@@ -1721,7 +1777,11 @@
             }
 #endif /* CONFIG_NUMA_BALANCING */
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 8, 0)
             up_write(&pTask->mm->mmap_sem);
+#else
+            up_write(&pTask->mm->mmap_lock);
+#endif
 
             if (RT_SUCCESS(rc))
             {
