diff -Naur vboxdrv/Makefile vboxdrv.new/Makefile
--- vboxdrv/Makefile	2017-05-14 21:39:53.000000000 +0200
+++ vboxdrv.new/Makefile	2017-05-14 21:57:19.000000000 +0200
@@ -306,7 +306,6 @@
 install: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
 	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR); \
-	PATH="$(PATH):/bin:/sbin" depmod -a; \
 	rm -f /etc/vbox/module_not_compiled
 
 install_rpm: $(MODULE)
diff -Naur vboxnetadp/Makefile vboxnetadp.new/Makefile
--- vboxnetadp/Makefile	2017-05-14 21:39:53.000000000 +0200
+++ vboxnetadp.new/Makefile	2017-05-14 21:57:19.000000000 +0200
@@ -208,7 +208,6 @@
 install: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
 	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR); \
-	PATH="$(PATH):/bin:/sbin" depmod -a; \
 	rm -f /etc/vbox/module_not_compiled
 
 install_rpm: $(MODULE)
diff -Naur vboxnetflt/Makefile vboxnetflt.new/Makefile
--- vboxnetflt/Makefile	2017-05-14 21:39:53.000000000 +0200
+++ vboxnetflt.new/Makefile	2017-05-14 21:57:19.000000000 +0200
@@ -212,7 +212,6 @@
 install: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
 	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR); \
-	PATH="$(PATH):/bin:/sbin" depmod -a; \
 	rm -f /etc/vbox/module_not_compiled
 
 install_rpm: $(MODULE)
diff -Naur vboxpci/Makefile vboxpci.new/Makefile
--- vboxpci/Makefile	2017-05-14 21:39:53.000000000 +0200
+++ vboxpci.new/Makefile	2017-05-14 21:57:19.000000000 +0200
@@ -211,7 +211,6 @@
 install: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
 	install -m 0644 -o root -g root $(MODULE).$(MODULE_EXT) $(MODULE_DIR); \
-	PATH="$(PATH):/bin:/sbin" depmod -a;
 
 install_rpm: $(MODULE)
 	@mkdir -p $(MODULE_DIR); \
