--- VirtualBox-4.3.0/src/VBox/VMM/VMMR3/DBGFDisas.cpp	2013-10-15 14:13:44.000000000 +0300
+++ VirtualBox-4.3.0.new/src/VBox/VMM/VMMR3/DBGFDisas.cpp	2013-10-17 16:07:44.620363840 +0300
@@ -353,8 +353,10 @@
     else if (   DIS_FMT_SEL_IS_REG(u32Sel)
              && DIS_FMT_SEL_GET_REG(u32Sel) == DISSELREG_SS
              && pSelInfo->GCPtrBase == 0
-             && pSelInfo->cbLimit   >= UINT32_MAX
-             && PATMIsPatchGCAddr(pState->pVM, pState->Cpu.uInstrAddr))
+#ifdef VBOX_WITH_RAW_MODE
+             && PATMIsPatchGCAddr(pState->pVM, pState->Cpu.uInstrAddr)
+#endif
+             && pSelInfo->cbLimit   >= UINT32_MAX)
     {
         DBGFR3AddrFromFlat(pState->pVM->pUVM, &Addr, uAddress);
         rc = VINF_SUCCESS;
