diff -Naur VirtualBox-6.0.4/src/VBox/Additions/linux/Makefile VirtualBox-6.0.4-p/src/VBox/Additions/linux/Makefile
--- VirtualBox-6.0.4/src/VBox/Additions/linux/Makefile	2019-01-25 19:12:34.000000000 +0100
+++ VirtualBox-6.0.4-p/src/VBox/Additions/linux/Makefile	2019-03-04 17:56:49.014608679 +0100
@@ -19,7 +19,7 @@
 # Building from kBuild (make -C <kernel_directory> M=`pwd`).
 # KBUILD_EXTMOD is set to $(M) in this case.
 
-obj-m = vboxguest/ vboxsf/ vboxvideo/
+obj-m = vboxguest/ vboxsf/
 
 else # ! KBUILD_EXTMOD
 
@@ -50,22 +50,7 @@
 	    echo; \
 	fi
 
-vboxvideo:
-	@if [ -d vboxvideo ]; then \
-	    if [ -f vboxguest/Module.symvers ]; then \
-	        cp vboxguest/Module.symvers vboxvideo; \
-	    fi; \
-	    echo "=== Building 'vboxvideo' module ==="; \
-	    $(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxvideo || exit 1; \
-	    if [ -f vboxvideo/vboxvideo.ko ]; then \
-	        cp vboxvideo/vboxvideo.ko .; \
-	    elif [ -f vboxvideo/vboxvideo.o ]; then \
-	        cp vboxvideo/vboxvideo.o .; \
-	    fi; \
-	    echo; \
-	fi
-
-all: vboxguest vboxsf vboxvideo
+all: vboxguest vboxsf
 
 install:
 	@$(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxguest install
@@ -81,22 +66,17 @@
 	@if [ -d vboxsf ]; then \
 	    $(MAKE) -C vboxsf clean; \
 	fi
-	@if [ -d vboxvideo ]; then \
-	    $(MAKE) -C vboxvideo clean; \
-	fi
-	rm -f vboxguest.*o vboxsf.*o vboxvideo.*o
+	rm -f vboxguest.*o vboxsf.*o
 
 check:
 	@$(MAKE) KBUILD_VERBOSE=$(KBUILD_VERBOSE) -C vboxguest check
 
 load:
-	@/sbin/rmmod vboxvideo || true
 	@/sbin/rmmod vboxvfs || true
 	@/sbin/rmmod vboxsf || true
 	@/sbin/rmmod vboxguest || true
 	@/sbin/insmod vboxguest.ko
 	@if [ -f vboxsf.ko ]; then /sbin/insmod vboxsf.ko; fi
-	@if [ -f vboxvideo.ko ]; then /sbin/insmod vboxvideo.ko; fi
 
 .PHONY: vboxguest vboxsf vboxvideo all install clean check load
 
