--- VirtualBox-4.1.2_OSE/configure	2011-09-10 23:42:17.000000000 +0000
+++ VirtualBox-4.1.2_OSE.new/configure	2011-09-10 23:25:46.000000000 +0000
@@ -1852,26 +1852,6 @@
   echo "compiling the following source file:" >> $LOG
   cat $ODIR.tmp_src.c >> $LOG
   echo "using the following command line:" >> $LOG
-  echo "$CC -m32 -O -Wall -o $ODIR.tmp_out $ODIR.tmp_src.c" >> $LOG
-  $CC -m32 -O -Wall -o $ODIR.tmp_out $ODIR.tmp_src.c >> $LOG 2>&1
-  if [ $? -ne 0 ]; then
-    echo
-    echo "  Cannot compile 32-bit applications (missing headers and/or libraries)!"
-    echo "  Check the file $LOG for detailed error information."
-    fail
-  else
-    echo "executing the binary" >> $LOG
-    $ODIR.tmp_out 2> $ODIR.test_execute.log
-    rc=$?
-    cat $ODIR.test_execute.log >> $LOG
-    if [ $rc -ne 0 ]; then
-      echo
-      echo "  Cannot execute 32-bit applications! Either enable 32-bit support in the"
-      echo "  kernel configuration or use --disable-vmmraw to disable 32-bit guests."
-      fail
-      return 1
-    fi
-  fi
   log_success ""
 }
 
--- VirtualBox-4.1.2_OSE/src/VBox/Makefile.kmk	2011-01-14 20:15:50.000000000 +0000
+++ VirtualBox-4.1.2_OSE.new/src/VBox/Makefile.kmk	2011-09-10 22:34:53.000000000 +0000
@@ -17,7 +17,7 @@
 
 SUB_DEPTH = ../..
 include $(KBUILD_PATH)/subheader.kmk
-
+VBOX_WITH_RAW_MODE=
 
 ifdef VBOX_ONLY_ADDITIONS
  include $(PATH_SUB_CURRENT)/Runtime/Makefile.kmk
