--- src/VBox/Main/VMMDevInterface.cpp.old	2008-01-27 16:50:02.000000000 +0100
+++ src/VBox/Main/VMMDevInterface.cpp	2008-01-27 16:50:28.000000000 +0100
@@ -27,6 +27,7 @@
 #include <VBox/VBoxGuest.h>
 #include <VBox/shflsvc.h>
 #include <iprt/asm.h>
+#include <stdlib.h>
 
 #ifdef VBOX_HGCM
 #include "hgcm/HGCM.h"
--- src/VBox/Frontends/VirtualBox/src/VBoxGlobal.cpp.old	2008-01-27 16:49:32.000000000 +0100
+++ src/VBox/Frontends/VirtualBox/src/VBoxGlobal.cpp	2008-01-27 16:49:48.000000000 +0100
@@ -28,6 +28,8 @@
 #include "VBoxRegistrationDlg.h"
 #endif
 
+#include <stdlib.h>
+
 #include <qapplication.h>
 #include <qmessagebox.h>
 #include <qpixmap.h>
