Index: src/recompiler/InnoTek/op-validate.sed
===================================================================
--- src/recompiler/InnoTek/op-validate.sed.orig
+++ src/recompiler/InnoTek/op-validate.sed
@@ -63,6 +63,8 @@ s/^[[:blank:]]*ret[[:blank:]]*\n*[[:blan
 /\.Lfe[0-9][0-9]*:/d
 /\.LFE[0-9][0-9]*:/d
 /size[[:space:]]/d
+/p2align[[:space:]]/d
+/^[/#][[:space:]]0[[:space:]]\"\"[[:space:]]2[[:space:]]*$/d
 /^[/#]NO_APP[[:space:]]*$/d
 /^$/!b bad
 b end
