diff -urN VirtualBox-6.1.2/src/VBox/Devices/Makefile.kmk VirtualBox-6.1.0/src/VBox/Devices/Makefile.kmk
--- VirtualBox-6.1.2/src/VBox/Devices/Makefile.kmk	2020-01-13 11:51:54.000000000 +0100
+++ VirtualBox-6.1.0/src/VBox/Devices/Makefile.kmk	2019-12-10 18:50:52.000000000 +0100
@@ -1267,7 +1248,7 @@
 	$(if $(VBOX_WITH_NAT_SERVICE),VBOX_WITH_NAT_SERVICE,)
  ServicesR0_SOURCES   = \
  	Network/SrvIntNetR0.cpp \
-#        $(if $(VBOX_WITH_PCI_PASSTHROUGH),Bus/SrvPciRawR0.cpp,)
+        $(if $(VBOX_WITH_PCI_PASSTHROUGH),Bus/SrvPciRawR0.cpp,)
  Network/SrvIntNetR0.cpp_CXXFLAGS := $(if-expr $(KBUILD_TARGET) == "win",,$(VBOX_GCC_Wno-array_bounds))
 
 
