diff -Naur galeon-2.0.4.old/mozilla/ProgressListener.cpp galeon-2.0.4/mozilla/ProgressListener.cpp
--- galeon-2.0.4.old/mozilla/ProgressListener.cpp	2009-06-18 20:21:18.000000000 +0530
+++ galeon-2.0.4/mozilla/ProgressListener.cpp	2009-06-18 20:23:45.000000000 +0530
@@ -745,7 +745,7 @@
 	GulCString helperId(helperDesc);
 
 	// Format <usertime>:<helperapp id>
-	char *colon = strchr (helperId.get(), ':');
+	char *colon = (char*) strchr (helperId.get(), ':');
 	g_return_val_if_fail (colon, NS_ERROR_FAILURE);
 
 	const char *helperappid = colon+1;
