--- galeon-2.0.7/configure.orig	2009-02-15 09:43:45.000000000 +0100
+++ galeon-2.0.7/configure	2009-02-15 09:44:22.000000000 +0100
@@ -24514,7 +24514,7 @@
 for i in $TEST_MOZILLA_INCLUDE_DIRS ; do
    CXXFLAGS="$CXXFLAGS -I$MOZILLA_INCLUDE_ROOT/$i"
 done
-CXXFLAGS="$CXXFLAGS -DMOZILLA_STRICT_API=1"
+CXXFLAGS="$CXXFLAGS -DMOZILLA_STRICT_API=1 `pkg-config --cflags nspr`"
 
 
 { $as_echo "$as_me:$LINENO: checking whether we have a mozilla debug build" >&5
