diff -Naur bochs-2.3.5/gui/wx.cc bochs-2.3.5-p/gui/wx.cc
--- bochs-2.3.5/gui/wx.cc	2006-10-25 19:40:56.000000000 +0200
+++ bochs-2.3.5-p/gui/wx.cc	2007-12-05 18:57:29.000000000 +0100
@@ -832,9 +832,7 @@
     case WXK_NUMPAD_UP:            bx_key = BX_KEY_KP_UP;        break;
     case WXK_NUMPAD_RIGHT:         bx_key = BX_KEY_KP_RIGHT;     break;
     case WXK_NUMPAD_DOWN:          bx_key = BX_KEY_KP_DOWN;      break;
-    case WXK_NUMPAD_PRIOR:         bx_key = BX_KEY_KP_PAGE_UP;   break;
     case WXK_NUMPAD_PAGEUP:        bx_key = BX_KEY_KP_PAGE_UP;   break;
-    case WXK_NUMPAD_NEXT:          bx_key = BX_KEY_KP_PAGE_DOWN; break;
     case WXK_NUMPAD_PAGEDOWN:      bx_key = BX_KEY_KP_PAGE_DOWN; break;
     case WXK_NUMPAD_END:           bx_key = BX_KEY_KP_END;       break;
     case WXK_NUMPAD_BEGIN:         bx_key = BX_KEY_KP_HOME;      break;
@@ -860,7 +858,6 @@
     case 220: bx_key = BX_KEY_BACKSLASH;     break; // \|
     case 222: bx_key = BX_KEY_SINGLE_QUOTE;  break; // '"
     case 305: bx_key = BX_KEY_KP_5;          break; // keypad 5
-    case 392: bx_key = BX_KEY_KP_ADD;        break; // keypad plus
 
     default:
       wxLogMessage(wxT ("Unhandled key event: %i (0x%x)"), key, key);
