diff --color -Naur simgear-v2024.1.1/simgear/nasal/cppbind/NasalHash.hxx simgear-v2024.1.1.new/simgear/nasal/cppbind/NasalHash.hxx
--- simgear-v2024.1.1/simgear/nasal/cppbind/NasalHash.hxx	2025-02-20 09:42:31.000000000 +0100
+++ simgear-v2024.1.1.new/simgear/nasal/cppbind/NasalHash.hxx	2025-05-26 15:56:55.538779440 +0200
@@ -19,6 +19,8 @@
 #ifndef SG_NASAL_HASH_HXX_
 #define SG_NASAL_HASH_HXX_
 
+#include <boost/mpl/if.hpp>
+
 #include "from_nasal.hxx"
 #include "to_nasal.hxx"
 
