diff -Naur simgear-2019.1.1/simgear/nasal/cppbind/NasalHash.hxx simgear-2019.1.1.new/simgear/nasal/cppbind/NasalHash.hxx
--- simgear-2019.1.1/simgear/nasal/cppbind/NasalHash.hxx	2019-05-13 17:34:25.000000000 +0200
+++ simgear-2019.1.1.new/simgear/nasal/cppbind/NasalHash.hxx	2019-10-21 16:41:09.269495345 +0200
@@ -23,6 +23,7 @@
 #include "to_nasal.hxx"
 
 #include <simgear/structure/map.hxx>
+#include <boost/utility/enable_if.hpp>
 #include <boost/iterator/iterator_facade.hpp>
 #if BOOST_VERSION >= 105600
 #include <boost/core/enable_if.hpp>
