--- ansi-wl-pprint-0.6.9/ansi-wl-pprint.cabal	2001-09-09 03:46:40.000000000 +0200
+++ ansi-wl-pprint-0.6.9.new/ansi-wl-pprint.cabal	2022-11-18 12:38:24.203675068 +0100
@@ -1,6 +1,8 @@
+cabal-version:       >= 1.10
 name:                ansi-wl-pprint
 version:             0.6.9
-cabal-version:       >= 1.10
+x-revision:          3
+
 category:            User Interfaces, Text
 synopsis:            The Wadler/Leijen Pretty Printer for colored ANSI terminal output
 description:         {
@@ -17,7 +19,7 @@
 bug-reports:         http://github.com/ekmett/ansi-wl-pprint/issues
 homepage:            http://github.com/ekmett/ansi-wl-pprint
 build-type:          Simple
-tested-with:         GHC==7.0.2, GHC==7.2.2, GHC==7.4.2, GHC==7.6.3, GHC==7.8.4, GHC==7.10.3, GHC==8.0.2, GHC==8.2.2, GHC==8.4.3, GHC==8.6.5, GHC==8.8.1
+tested-with:         GHC==7.0.4, GHC==7.2.2, GHC==7.4.2, GHC==7.6.3, GHC==7.8.4, GHC==7.10.3, GHC==8.0.2, GHC==8.2.2, GHC==8.4.3, GHC==8.6.5, GHC==8.8.4, GHC==8.10.7, GHC==9.0.1, GHC==9.2.1
 
 source-repository head
   type: git
@@ -42,9 +44,9 @@
       ghc-options: -Wnoncanonical-monadfail-instances
   else
     -- see also notes in Text.PrettyPrint.ANSI.Leijen
-    build-depends: semigroups >= 0.18.5 && < 0.20
+    build-depends: semigroups >= 0.18.5 && < 0.21
 
-  build-depends: ansi-terminal >= 0.9.1 && < 0.10
+  build-depends: ansi-terminal >= 0.9.1 && < 0.12
   build-depends: base >= 4.3 && < 5
 
   if impl(ghc >= 7.4)
