diff --git a/src/scim_phrase.cpp b/src/scim_phrase.cpp
index b8618ca..2534c5f 100644
--- a/src/scim_phrase.cpp
+++ b/src/scim_phrase.cpp
@@ -28,6 +28,7 @@
 #define SCIM_PHRASE_MAX_RELATION 1000
 
 #include <scim.h>
+#include <string.h>
 #include "scim_pinyin_private.h"
 #include "scim_phrase.h"
 
diff --git a/src/scim_pinyin.cpp b/src/scim_pinyin.cpp
index 0c281ec..6ee795e 100644
--- a/src/scim_pinyin.cpp
+++ b/src/scim_pinyin.cpp
@@ -29,6 +29,7 @@
 #define Uses_SCIM_LOOKUP_TABLE
 
 #include <scim.h>
+#include <string.h>
 #include "scim_pinyin.h"
 
 // Internal functions
diff --git a/src/scim_pinyin_phrase.cpp b/src/scim_pinyin_phrase.cpp
index e44e9df..fce993c 100644
--- a/src/scim_pinyin_phrase.cpp
+++ b/src/scim_pinyin_phrase.cpp
@@ -27,6 +27,7 @@
 #define Uses_SCIM_LOOKUP_TABLE
 
 #include <scim.h>
+#include <string.h>
 #include "scim_pinyin_private.h"
 #include "scim_phrase.h"
 #include "scim_pinyin.h"
diff --git a/src/scim_special_table.cpp b/src/scim_special_table.cpp
index 6a46e8f..e1c7cc7 100644
--- a/src/scim_special_table.cpp
+++ b/src/scim_special_table.cpp
@@ -22,6 +22,7 @@
 
 #include <time.h>
 #include <scim.h>
+#include <string.h>
 #include "scim_pinyin_private.h"
 #include "scim_special_table.h"
 
