--- xfce4-session-4.5.92/xfce4-session/xfsm-manager.c	2008-11-14 15:11:08.000000000 +0530
+++ xfce4-session.new/xfce4-session/xfsm-manager.c	2008-11-30 23:53:37.000000000 +0530
@@ -1,4 +1,4 @@
-/* $Id: xfsm-manager.c 28625 2008-11-06 09:40:43Z kelnos $ */
+/* $Id: xfsm-manager.c 28938 2008-11-28 06:14:37Z kelnos $ */
 /*-
  * Copyright (c) 2003-2006 Benedikt Meurer <benny@xfce.org>
  * Copyright (c) 2008 Brian Tarricone <bjt23@cornell.edu>
@@ -910,14 +910,13 @@
       xfsm_manager_start_client_save_timeout (manager, client);
     }
 
-  if ((manager->failsafe_mode || previous_id != NULL)
-      && manager->state == XFSM_MANAGER_STARTUP)
+  if (previous_id != NULL && manager->state == XFSM_MANAGER_STARTUP)
     {
-      /* Only continue the startup if we are either in Failsafe mode (which
-       * means that we don't have any previous_id at all) or the previous_id
-       * matched one of the starting_properties. If there was no match
-       * above, previous_id will be NULL here.
-       * See http://bugs.xfce.org/view_bug_page.php?f_id=212 for details.
+      /* Only continue the startup if the previous_id matched one of
+       * the starting_properties. If there was no match above,
+       * previous_id will be NULL here.  We don't need to continue when
+       * in failsafe mode because in that case the failsafe session is
+       * started all at once.
        */
       if (g_queue_peek_head (manager->starting_properties) == NULL)
         xfsm_startup_session_continue (manager);
@@ -1105,7 +1104,7 @@
                                                     manager->shutdown_type,
                                                     &error)
                   && (error->domain != DBUS_GERROR
-                      || error->code != DBUS_GERROR_TIMED_OUT)))
+                      || error->code != DBUS_GERROR_NO_REPLY)))
             {
               xfce_message_dialog (NULL, _("Shutdown Failed"),
                                    GTK_STOCK_DIALOG_ERROR,
