--- openttd-14.1/CMakeLists.txt	2024-05-03 22:51:29.000000000 +0200
+++ openttd-14.1.new/CMakeLists.txt	2024-10-29 18:44:48.971875759 +0100
@@ -152,7 +152,7 @@
                 find_package(Fontconfig)
             endif()
             find_package(Harfbuzz)
-            find_package(ICU OPTIONAL_COMPONENTS i18n)
+            find_package(ICU OPTIONAL_COMPONENTS uc)
         endif()
     endif()
 endif()
