diff -Naur firestarter-1.0.3/src/policyview.c firestarter-1.0.3-p/src/policyview.c
--- firestarter-1.0.3/src/policyview.c	2005-01-29 13:32:08.000000000 +0100
+++ firestarter-1.0.3-p/src/policyview.c	2009-03-02 00:50:16.000000000 +0100
@@ -146,7 +146,7 @@
 
 	g_strfreev (tokens);
 	if (view != NULL)
-		gtk_tree_iter_free (iter);
+		g_free (iter);
 	return TRUE;
 }
 
@@ -428,7 +428,7 @@
 				gtk_list_store_remove (GTK_LIST_STORE (model), iter); /* Remove from the view */
 				remove_line_from_file (rule_file, (int)position);
 
-				gtk_tree_iter_free (iter);
+				g_free (iter);
 			}
 
 			append_to_file (rule_file, data, TRUE);
diff -Naur firestarter-1.0.3/src/preferences.c firestarter-1.0.3-p/src/preferences.c
--- firestarter-1.0.3/src/preferences.c	2005-01-29 13:32:08.000000000 +0100
+++ firestarter-1.0.3-p/src/preferences.c	2009-03-02 00:50:36.000000000 +0100
@@ -244,7 +244,7 @@
 	                    0, SECTION_INTERFACE_POLICY,
 			    1, _("Policy"),
 			    -1);
-	gtk_tree_iter_free (top_section);
+	g_free (top_section);
 
 	gtk_tree_store_append (store, iter, NULL);
 	top_section = gtk_tree_iter_copy (iter);
@@ -272,8 +272,8 @@
 	                    0, SECTION_FIREWALL_ADVANCED,
 			    1, _("Advanced Options"),
 			    -1);
-	gtk_tree_iter_free (top_section);
-	gtk_tree_iter_free (iter);
+	g_free (top_section);
+	g_free (iter);
 
 	return GTK_TREE_MODEL (store);
 }
diff -Naur firestarter-1.0.3/src/statusview.c firestarter-1.0.3-p/src/statusview.c
--- firestarter-1.0.3/src/statusview.c	2005-01-29 13:32:08.000000000 +0100
+++ firestarter-1.0.3-p/src/statusview.c	2009-03-02 00:50:27.000000000 +0100
@@ -773,7 +773,7 @@
 {
 	Connection_entry *entry = (Connection_entry *)data;
 	
-	gtk_tree_iter_free (entry->ref);
+	g_free (entry->ref);
 	g_free (data);
 }
 
