diff -ur a/libs/imwrapper/src/purple/PurpleIMFactory.h b/libs/imwrapper/src/purple/PurpleIMFactory.h
--- a/libs/imwrapper/src/purple/PurpleIMFactory.h	2012-05-08 06:27:33.492005629 -0500
+++ b/libs/imwrapper/src/purple/PurpleIMFactory.h	2012-05-08 06:28:02.751004529 -0500
@@ -24,7 +24,6 @@
 #include <imwrapper/IMAccount.h>
 
 #include <glib.h>
-#include <glib/gtypes.h>
 
 class PurpleAccountMngr;
 class PurpleChatMngr;
