Subject: hack to get orcus/autotools build against boost 1.53
From: Bjoern Michaelsen <bjoern.michaelsen@canonical.com>
Bug: https://bugs.freedesktop.org/show_bug.cgi?id=55435
Forwarded: https://bugs.freedesktop.org/attachment.cgi?id=80565&action=edit
--
--- liborcus-0.5.1.orig/src/Makefile.am
+++ liborcus-0.5.1/src/Makefile.am
@@ -39,6 +39,8 @@ orcus_xml_dump_LDADD = \
 	liborcus/liborcus-@ORCUS_API_VERSION@.la
 
 orcus_xml_dump_CPPFLAGS = -I$(top_builddir)/lib/liborcus/liborcus.la $(COMMON_CPPFLAGS)
+orcus_xml_dump_LDFLAGS = -lboost_system
+
 
 # orcus-test-common
 
--- liborcus-0.5.1.orig/src/parser/Makefile.am
+++ liborcus-0.5.1/src/parser/Makefile.am
@@ -18,7 +18,7 @@ liborcus_parser_@ORCUS_API_VERSION@_la_S
 	zip_archive.cpp \
 	zip_archive_stream.cpp
 
-liborcus_parser_@ORCUS_API_VERSION@_la_LDFLAGS = -no-undefined
+liborcus_parser_@ORCUS_API_VERSION@_la_LDFLAGS = -no-undefined -lboost_system
 liborcus_parser_@ORCUS_API_VERSION@_la_LIBADD = \
 	 $(ZLIB_LIBS)
 
