diff -Naur bash_completion.orig/bash_completion bash_completion/bash_completion
--- bash_completion.orig/bash_completion	2004-07-11 20:36:39.000000000 +0200
+++ bash_completion/bash_completion	2004-10-15 14:24:47.995790306 +0200
@@ -5283,7 +5283,7 @@
 			;;
 		-*-charset)
 			COMPREPLY=( $( mkisofs -input-charset help 2>&1 | \
-					tail +3 | grep "^$cur") )
+					tail -n +3 | grep "^$cur") )
 			return 0
 			;;
 		-uid)
