diff -Naur bash_completion.orig/bash_completion bash_completion/bash_completion
--- bash_completion.orig/bash_completion	2005-07-21 21:21:22.000000000 +0200
+++ bash_completion/bash_completion	2006-01-12 17:33:20.000000000 +0100
@@ -2474,7 +2474,7 @@
 	    fi
 	    # append any available aliases from config files
 	    if [ ${#config[@]} -gt 0 ] && [ -n "$aliases" ]; then
-		hosts=$( compgen -W "$( sed -ne "s/^[Hh][Oo][Ss][Tt]["$'\t '"]*\([^*?]*\)$/\1/p" ${config[@]} )" -- $ocur )
+		hosts=$( compgen -W "$( sed -ne 's/^[Hh][Oo][Ss][Tt][\t ]*\([^*?]*\)$/\1/p' ${config[@]} )" -- $ocur )
 		COMPREPLY=( ${COMPREPLY[@]} $hosts )
 	    fi
 
