--- bash_completion.orig	2006-03-01 17:20:18.000000000 +0100
+++ bash_completion	2007-05-08 11:57:27.000000000 +0200
@@ -78,7 +78,7 @@
 # Do NOT break these over multiple lines.
 #
 # START exclude -- do NOT remove this line
-complete -f -X '!*.?(t)bz?(2)' bunzip2 bzcat bzcmp bzdiff bzegrep bzfgrep bzgrep
+complete -f -X '!*.@(?(t)bz?(2)|fpm)' bunzip2 bzcat bzcmp bzdiff bzegrep bzfgrep bzgrep
 complete -f -X '!*.@(zip|ZIP|jar|JAR|exe|EXE|pk3|war|wsz|ear|zargo|xpi|sxw|ott)' unzip zipinfo
 complete -f -X '*.Z' compress znew
 complete -f -X '!*.@(Z|gz|tgz|Gz|dz)' gunzip zcmp zdiff zcat zegrep zfgrep zgrep zless zmore
