diff -aur bash-completion-1.1/bash_completion fw_bash-completion-1.1//bash_completion
--- bash-completion-1.1/bash_completion	2009-10-10 16:26:09.000000000 +0200
+++ fw_bash-completion-1.1//bash_completion	2010-05-22 19:35:54.000000000 +0200
@@ -105,7 +105,7 @@
 complete -f -X '!*.Z' uncompress
 # lzcmp, lzdiff intentionally not here, see Debian: #455510
 complete -f -X '!*.lzma' lzcat lzegrep lzfgrep lzgrep lzless lzmore unlzma
-complete -f -X '!*.@(xz|lzma)' unxz xzcat
+complete -f -X '!*.@(xz|lzma|fpm)' unxz xzcat
 complete -f -X '!*.@(gif|jp?(e)g|miff|tif?(f)|pn[gm]|p[bgp]m|bmp|xpm|ico|xwd|tga|pcx|GIF|JP?(E)G|MIFF|TIF?(F)|PN[GM]|P[BGP]M|BMP|XPM|ICO|XWD|TGA|PCX)' ee
 complete -f -X '!*.@(gif|jp?(e)g|tif?(f)|png|p[bgp]m|bmp|x[bp]m|rle|rgb|pcx|fits|pm|GIF|JPG|JP?(E)G|TIF?(F)|PNG|P[BGP]M|BMP|X[BP]M|RLE|RGB|PCX|FITS|PM)' xv qiv
 complete -f -X '!*.@(@(?(e)ps|?(E)PS|pdf|PDF)?(.gz|.GZ|.bz2|.BZ2|.Z))' gv ggv kghostview
