diff -Naur sim-0.9.3.orig/plugins/_core/commands.cpp sim-0.9.3/plugins/_core/commands.cpp
--- sim-0.9.3.orig/plugins/_core/commands.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/_core/commands.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -188,27 +188,27 @@
         clear();
         break;
     case EventToolbarCreate:
-        return (void*)createBar((unsigned)(e->param()));
+        return (void*)createBar((unsigned long)(e->param()));
     case EventToolbarRemove:
-        removeBar((unsigned)(e->param()));
+        removeBar((unsigned long)(e->param()));
         break;
     case EventShowBar:
         b = (BarShow*)(e->param());
         return show(b->bar_id, b->parent);
     case EventMenuCreate:
-        return (void*)createMenu((unsigned)(e->param()));
+        return (void*)createMenu((unsigned long)(e->param()));
     case EventMenuRemove:
-        removeMenu((unsigned)(e->param()));
+        removeMenu((unsigned long)(e->param()));
         break;
     case EventGetMenu:
         return (void*)get((CommandDef*)(e->param()));
     case EventGetMenuDef:
-        return (void*)getDef((unsigned)(e->param()));
+        return (void*)getDef((unsigned long)(e->param()));
     case EventProcessMenu:
         mp = (ProcessMenuParam*)(e->param());
         return (void*)processMenu(mp->id, mp->param, mp->key);
     case EventMenuCustomize:
-        customizeMenu((unsigned)(e->param()));
+        customizeMenu((unsigned long)(e->param()));
         break;
     default:
         break;
diff -Naur sim-0.9.3.orig/plugins/_core/container.cpp sim-0.9.3/plugins/_core/container.cpp
--- sim-0.9.3.orig/plugins/_core/container.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/_core/container.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -680,7 +680,7 @@
     case EventCommandExec:
         cmd = (CommandDef*)(e->param());
         userWnd = m_tabBar->currentWnd();
-        if (userWnd && ((unsigned)(cmd->param) == userWnd->id())){
+        if (userWnd && ((unsigned long)(cmd->param) == userWnd->id())){
             if (cmd->menu_id == MenuContainerContact){
                 m_tabBar->raiseTab(cmd->id);
                 return e->param();
@@ -702,7 +702,7 @@
     case EventCheckState:
         cmd = (CommandDef*)(e->param());
         userWnd = m_tabBar->currentWnd();
-        if (userWnd && ((unsigned)(cmd->param) == userWnd->id()) &&
+        if (userWnd && ((unsigned long)(cmd->param) == userWnd->id()) &&
                 (cmd->menu_id == MenuContainerContact) &&
                 (cmd->id == CmdContainerContacts)){
             list<UserWnd*> userWnds = m_tabBar->windows();
diff -Naur sim-0.9.3.orig/plugins/_core/core.cpp sim-0.9.3/plugins/_core/core.cpp
--- sim-0.9.3.orig/plugins/_core/core.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/_core/core.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -1782,7 +1782,7 @@
             return e->param();
         }
     case EventRemovePreferences:{
-            unsigned id = (unsigned)(e->param());
+            unsigned id = (unsigned long)(e->param());
             Event eCmd(EventCommandRemove, (void*)id);
             eCmd.process();
             preferences.erase(id);
@@ -1858,7 +1858,7 @@
             return e->param();
         }
     case EventRemoveMessageType:{
-            unsigned id = (unsigned)(e->param());
+            unsigned id = (unsigned long)(e->param());
             CommandDef *def;
             def = CorePlugin::m_plugin->messageTypes.find(id);
             if (def){
@@ -2015,7 +2015,7 @@
             return NULL;
         }
     case EventDefaultAction:{
-            unsigned contact_id = (unsigned)(e->param());
+            unsigned contact_id = (unsigned long)(e->param());
             unsigned index = 0;
             for (list<msg_id>::iterator it = CorePlugin::m_plugin->unread.begin(); it != CorePlugin::m_plugin->unread.end(); ++it, index++){
                 if ((*it).contact != contact_id)
@@ -2167,7 +2167,7 @@
             return e->param();
         }
     case EventContactOnline:{
-            Contact *contact = getContacts()->contact((unsigned)(e->param()));
+            Contact *contact = getContacts()->contact((unsigned long)(e->param()));
             if (contact){
                 CoreUserData *data = (CoreUserData*)(contact->getUserData(user_data_id));
                 if (data->OpenOnOnline.bValue){
@@ -2200,7 +2200,7 @@
                 return NULL;
             }
             if (cmd->id == CmdContactClients){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact == NULL)
                     return NULL;
                 vector<clientContact> ways;
@@ -2474,7 +2474,7 @@
                 return e->param();
             }
             if (cmd->menu_id == MenuContainer){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact){
                     unsigned nContainers = 1;
                     QWidgetList  *list = QApplication::topLevelWidgets();
@@ -2515,7 +2515,7 @@
             }
             if (cmd->menu_id == MenuMessage){
                 cmd->flags &= ~COMMAND_CHECKED;
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact){
                     vector<clientContact> ways;
                     getWays(ways, contact);
@@ -2774,9 +2774,9 @@
                 if (mdef->create == NULL)
                     return NULL;
                 msg = mdef->create(NULL);
-                msg->setContact((unsigned)(cmd->param));
+                msg->setContact((unsigned long)(cmd->param));
                 if (mdef->flags & MESSAGE_SILENT){
-                    Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                    Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                     if (contact){
                         ClientDataIterator it(contact->clientData);
                         void *data;
@@ -2872,7 +2872,7 @@
             }
 
             if (cmd->id == CmdSeparate){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact == NULL)
                     return NULL;
                 unsigned n = cmd->menu_id - CmdContactClients - 1;
@@ -2921,7 +2921,7 @@
                 return e->param();
             }
             if (cmd->id == CmdHistory){
-                unsigned id = (unsigned)(cmd->param);
+                unsigned id = (unsigned long)(cmd->param);
                 HistoryWindow *wnd = NULL;
                 QWidgetList  *list = QApplication::topLevelWidgets();
                 QWidgetListIt it(*list);
@@ -3176,7 +3176,7 @@
                 return e->param();
             }
             if ((cmd->menu_id > CmdContactResource) && (cmd->menu_id <= CmdContactResource + 0x100)){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 CommandDef *def = messageTypes.find(cmd->id);
                 if (def && contact){
                     unsigned nRes = cmd->menu_id - CmdContactResource - 1;
@@ -3203,7 +3203,7 @@
                                 getToken(res, ',');
                                 MessageDef *mdef = (MessageDef*)(def->param);
                                 Message *msg = mdef->create(NULL);
-                                msg->setContact((unsigned)(cmd->param));
+                                msg->setContact((unsigned long)(cmd->param));
                                 msg->setClient(cc.client->dataName(data).c_str());
                                 msg->setResource(QString::fromUtf8(res.c_str()));
                                 Event eOpen(EventOpenMessage, &msg);
@@ -3217,7 +3217,7 @@
                 return NULL;
             }
             if ((cmd->menu_id > CmdContactClients) && (cmd->menu_id <= CmdContactClients + 0x100)){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 CommandDef *def = messageTypes.find(cmd->id);
                 if (def && contact){
                     unsigned n = cmd->menu_id - CmdContactClients - 1;
@@ -3241,7 +3241,7 @@
 
                         MessageDef *mdef = (MessageDef*)(def->param);
                         Message *msg = mdef->create(NULL);
-                        msg->setContact((unsigned)(cmd->param));
+                        msg->setContact((unsigned long)(cmd->param));
                         msg->setClient(cc.client->dataName(data).c_str());
                         Event eOpen(EventOpenMessage, &msg);
                         eOpen.process();
@@ -3305,7 +3305,7 @@
     QWidget *w;
     Contact *contact = NULL;
     Group   *group   = NULL;
-    unsigned id = (unsigned)(cmd->param);
+    unsigned id = (unsigned long)(cmd->param);
     if (cmd->menu_id == MenuContact){
         contact = getContacts()->contact(id);
         if (contact == NULL)
diff -Naur sim-0.9.3.orig/plugins/_core/historywnd.cpp sim-0.9.3/plugins/_core/historywnd.cpp
--- sim-0.9.3.orig/plugins/_core/historywnd.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/_core/historywnd.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -131,7 +131,7 @@
     }
     if (e->type() == EventCheckState){
         CommandDef *cmd = (CommandDef*)(e->param());
-        if ((cmd->id == CmdHistoryDirection) && ((unsigned)(cmd->param) == m_id)){
+        if ((cmd->id == CmdHistoryDirection) && ((unsigned long)(cmd->param) == m_id)){
             cmd->flags &= ~COMMAND_CHECKED;
             if (m_bDirection)
                 cmd->flags |= COMMAND_CHECKED;
@@ -146,7 +146,7 @@
     }
     if (e->type() == EventCommandExec){
         CommandDef *cmd = (CommandDef*)(e->param());
-        if ((unsigned)(cmd->param) != m_id)
+        if ((unsigned long)(cmd->param) != m_id)
             return NULL;
         if (cmd->id == CmdHistoryDirection){
             bool bDirection = ((cmd->flags & COMMAND_CHECKED) != 0);
diff -Naur sim-0.9.3.orig/plugins/_core/msgview.cpp sim-0.9.3/plugins/_core/msgview.cpp
--- sim-0.9.3.orig/plugins/_core/msgview.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/_core/msgview.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -802,7 +802,7 @@
         return NULL;
     }
     if (e->type() == EventHistoryConfig){
-        unsigned id = (unsigned)(e->param());
+        unsigned id = (unsigned long)(e->param());
         if (id && (id != m_id))
             return NULL;
         reload();
diff -Naur sim-0.9.3.orig/plugins/_core/tmpl.cpp sim-0.9.3/plugins/_core/tmpl.cpp
--- sim-0.9.3.orig/plugins/_core/tmpl.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/_core/tmpl.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -146,7 +146,7 @@
         if (tagName == "IP"){
             Event e(EventGetContactIP, contact);
             struct in_addr addr;
-            addr.s_addr = (unsigned)(e.process());
+            addr.s_addr = (unsigned long)(e.process());
             res += inet_ntoa(addr);
             continue;
         }
diff -Naur sim-0.9.3.orig/plugins/_core/usercfg.cpp sim-0.9.3/plugins/_core/usercfg.cpp
--- sim-0.9.3.orig/plugins/_core/usercfg.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/_core/usercfg.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -463,7 +463,7 @@
             return NULL;
         }
     case EventCommandRemove:
-        removeCommand((unsigned)(e->param()));
+        removeCommand((unsigned long)(e->param()));
         return NULL;
     case EventLanguageChanged:
     case EventPluginChanged:
diff -Naur sim-0.9.3.orig/plugins/_core/userview.cpp sim-0.9.3/plugins/_core/userview.cpp
--- sim-0.9.3.orig/plugins/_core/userview.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/_core/userview.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -359,7 +359,7 @@
     case EventCommandExec:{
             CommandDef *cmd = (CommandDef*)(e->param());
             if (cmd->menu_id == MenuContact){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact){
                     if (cmd->id == CmdContactDelete){
                         QListViewItem *item = findContactItem(contact->id());
@@ -404,7 +404,7 @@
                         while ((w = it.current()) != NULL){
                             if (w->inherits("Container")){
                                 Container *c =  static_cast<Container*>(w);
-                                wnd = c->wnd((unsigned)(cmd->param));
+                                wnd = c->wnd((unsigned long)(cmd->param));
                                 if (wnd)
                                     break;
                             }
@@ -428,7 +428,7 @@
                 }
             }
             if (cmd->menu_id == MenuContactGroup){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact){
                     Group *grp = getContacts()->group(cmd->id - CmdContactGroup);
                     if (grp && (grp->id() != contact->getGroup())){
@@ -440,7 +440,7 @@
                 }
             }
             if (cmd->menu_id == MenuContainer){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact){
                     Container *from = NULL;
                     Container *to = NULL;
@@ -516,7 +516,7 @@
                 return e->param();
             }
             if (cmd->id == CmdGrpRename){
-                QListViewItem *item = findGroupItem((unsigned)(cmd->param));
+                QListViewItem *item = findGroupItem((unsigned long)(cmd->param));
                 if (item){
                     setCurrentItem(item);
                     renameGroup();
@@ -524,7 +524,7 @@
                 return e->param();
             }
             if (cmd->id == CmdGrpUp){
-                unsigned grp_id = (unsigned)(cmd->param);
+                unsigned grp_id = (unsigned long)(cmd->param);
                 getContacts()->moveGroup(grp_id, true);
                 QListViewItem *item = findGroupItem(grp_id);
                 if (item){
@@ -534,7 +534,7 @@
                 return e->param();
             }
             if (cmd->id == CmdGrpDown){
-                unsigned grp_id = (unsigned)(cmd->param);
+                unsigned grp_id = (unsigned long)(cmd->param);
                 getContacts()->moveGroup(grp_id, false);
                 QListViewItem *item = findGroupItem(grp_id);
                 if (item){
@@ -544,7 +544,7 @@
                 return e->param();
             }
             if (cmd->id == CmdGrpDelete){
-                unsigned grp_id = (unsigned)(cmd->param);
+                unsigned grp_id = (unsigned long)(cmd->param);
                 QListViewItem *item = findGroupItem(grp_id);
                 Group *g = getContacts()->group(grp_id);
                 if (item && g){
@@ -578,14 +578,14 @@
             }
             if (cmd->menu_id == MenuContact){
                 if (cmd->id == CmdContactTitle){
-                    Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                    Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                     if (contact){
                         cmd->text_wrk = strdup(contact->getName().utf8());
                         return e->param();
                     }
                 }
                 if (cmd->id == CmdShowAlways){
-                    Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                    Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                     if (contact){
                         ListUserData *data = (ListUserData*)(contact->getUserData(CorePlugin::m_plugin->list_data_id, true));
                         cmd->flags &= ~COMMAND_CHECKED;
@@ -601,7 +601,7 @@
                     QWidget * w;
                     while ((w = it.current()) != NULL){
                         if (w->inherits("Container")){
-                            wnd = static_cast<Container*>(w)->wnd((unsigned)(cmd->param));
+                            wnd = static_cast<Container*>(w)->wnd((unsigned long)(cmd->param));
                             if (wnd)
                                 break;
                         }
@@ -654,7 +654,7 @@
             if (cmd->menu_id == MenuContactGroup){
                 if (cmd->id == CmdContactGroup){
                     unsigned grpId = 0;
-                    Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                    Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                     if (contact)
                         grpId = contact->getGroup();
                     unsigned nGroups = 0;
@@ -689,7 +689,7 @@
                 }
             }
             if (cmd->menu_id == MenuGroup){
-                unsigned grp_id = (unsigned)(cmd->param);
+                unsigned grp_id = (unsigned long)(cmd->param);
                 if (grp_id){
                     if (cmd->id == CmdGrpTitle){
                         Group *g = getContacts()->group(grp_id);
@@ -739,14 +739,14 @@
 
 void UserView::deleteGroup(void *p)
 {
-    Group *grp = getContacts()->group((unsigned)p);
+    Group *grp = getContacts()->group((unsigned long)p);
     if (grp)
         delete grp;
 }
 
 void UserView::deleteContact(void *p)
 {
-    Contact *contact = getContacts()->contact((unsigned)p);
+    Contact *contact = getContacts()->contact((unsigned long)p);
     if (contact == NULL)
         return;
     ContactItem *item = findContactItem(contact->id());
diff -Naur sim-0.9.3.orig/plugins/action/action.cpp sim-0.9.3/plugins/action/action.cpp
--- sim-0.9.3.orig/plugins/action/action.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/action/action.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -133,7 +133,7 @@
     if (e->type() == EventCheckState){
         CommandDef *cmd = (CommandDef*)(e->param());
         if ((cmd->id == CmdAction) && (cmd->menu_id == MenuContact)){
-            Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+            Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
             if (contact == NULL)
                 return NULL;
             ActionUserData *data = (ActionUserData*)(contact->getUserData(action_data_id));
@@ -180,7 +180,7 @@
         CommandDef *cmd = (CommandDef*)(e->param());
         if ((cmd->menu_id == MenuContact) && (cmd->id >= CmdAction)){
             unsigned n = cmd->id - CmdAction;
-            Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+            Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
             if (contact == NULL)
                 return NULL;
             ActionUserData *data = (ActionUserData*)(contact->getUserData(action_data_id));
diff -Naur sim-0.9.3.orig/plugins/dock/dockwnd.cpp sim-0.9.3/plugins/dock/dockwnd.cpp
--- sim-0.9.3.orig/plugins/dock/dockwnd.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/dock/dockwnd.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -772,7 +772,7 @@
     XFree( hints );
 
     Event eArgc(EventArgc);
-    int argc = (int)eArgc.process();
+    int argc = (long)eArgc.process();
     Event eArgv(EventArgv);
     char **argv = (char**)eArgv.process();
     XSetCommand(dsp, win, argv, argc);
diff -Naur sim-0.9.3.orig/plugins/filter/filter.cpp sim-0.9.3/plugins/filter/filter.cpp
--- sim-0.9.3.orig/plugins/filter/filter.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/filter/filter.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -188,7 +188,7 @@
         CommandDef *cmd = (CommandDef*)(e->param());
         if (cmd->id == CmdIgnore){
             cmd->flags &= ~BTN_HIDE;
-            Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+            Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
             if (contact && contact->getGroup())
                 cmd->flags |= BTN_HIDE;
             return e->param();
@@ -208,7 +208,7 @@
         }
         if (cmd->menu_id == MenuContactGroup){
             if (cmd->id == CmdIgnoreList){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact == NULL)
                     return NULL;
                 cmd->flags &= COMMAND_CHECKED;
@@ -221,7 +221,7 @@
     if (e->type() == EventCommandExec){
         CommandDef *cmd = (CommandDef*)(e->param());
         if (cmd->id == CmdIgnore){
-            Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+            Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
             if (contact){
                 QString text = i18n("Add %1 to ignore list?") .arg(contact->getName());
                 Command cmd;
@@ -288,7 +288,7 @@
         }
         if (cmd->menu_id == MenuContactGroup){
             if (cmd->id == CmdIgnoreList){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact == NULL)
                     return NULL;
                 contact->setIgnore((cmd->flags & COMMAND_CHECKED) == 0);
@@ -397,7 +397,7 @@
 
 void FilterPlugin::addToIgnore(void *p)
 {
-    Contact *contact = getContacts()->contact((unsigned)p);
+    Contact *contact = getContacts()->contact((unsigned long)p);
     if (contact && !contact->getIgnore()){
         contact->setIgnore(true);
         Event e(EventContactChanged, contact);
diff -Naur sim-0.9.3.orig/plugins/floaty/floaty.cpp sim-0.9.3/plugins/floaty/floaty.cpp
--- sim-0.9.3.orig/plugins/floaty/floaty.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/floaty/floaty.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -155,7 +155,7 @@
     case EventCheckState:{
             CommandDef *cmd = (CommandDef*)(e->param());
             if (cmd->id == CmdFloaty){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact){
                     FloatyUserData *data = (FloatyUserData*)(contact->userData.getUserData(user_data_id, false));
                     if (data){
@@ -173,7 +173,7 @@
     case EventCommandExec:{
             CommandDef *cmd = (CommandDef*)(e->param());
             if (cmd->id == CmdFloaty){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact){
                     FloatyUserData *data = (FloatyUserData*)(contact->userData.getUserData(user_data_id, false));
                     if (data){
@@ -183,7 +183,7 @@
                         contact->userData.freeUserData(user_data_id);
                     }else{
                         data = (FloatyUserData*)(contact->userData.getUserData(user_data_id, true));
-                        FloatyWnd *wnd = new FloatyWnd(this, (unsigned)(cmd->param));
+                        FloatyWnd *wnd = new FloatyWnd(this, (unsigned long)(cmd->param));
                         wnd->move(0, 0);
                         wnd->show();
                     }
diff -Naur sim-0.9.3.orig/plugins/gpg/gpg.cpp sim-0.9.3/plugins/gpg/gpg.cpp
--- sim-0.9.3.orig/plugins/gpg/gpg.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/gpg/gpg.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -292,7 +292,7 @@
                 }
                 if (cmd->id == MessageGPGUse){
                     cmd->flags &= ~COMMAND_CHECKED;
-                    Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                    Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                     if (contact == NULL)
                         return NULL;
                     GpgUserData *data = (GpgUserData*)(contact->userData.getUserData(user_data_id, false));
@@ -308,7 +308,7 @@
     case EventCommandExec:{
             CommandDef *cmd = (CommandDef*)(e->param());
             if ((cmd->menu_id == MenuMessage) && (cmd->id == MessageGPGUse)){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact == NULL)
                     return NULL;
                 GpgUserData *data = (GpgUserData*)(contact->userData.getUserData(user_data_id, false));
@@ -369,7 +369,7 @@
                     GpgUserData *data = (GpgUserData*)(contact->userData.getUserData(user_data_id, false));
                     if (data && data->Key.ptr && data->Use.bValue){
                         QString output = QFile::decodeName(user_file("m.").c_str());
-                        output += QString::number((unsigned)ms->msg);
+                        output += QString::number((unsigned long)ms->msg);
                         QString input = output + ".in";
                         QFile in(input);
                         if (!in.open(IO_WriteOnly | IO_Truncate)){
@@ -423,7 +423,7 @@
                 char SIGN_KEY[] = "-----BEGIN PGP PUBLIC KEY BLOCK-----";
                 if (text.left(strlen(SIGN_MSG)) == SIGN_MSG){
                     QString output = QFile::decodeName(user_file("m.").c_str());
-                    output += QString::number((unsigned)msg);
+                    output += QString::number((unsigned long)msg);
                     QString input = output + ".in";
                     QFile in(input);
                     if (!in.open(IO_WriteOnly | IO_Truncate)){
@@ -460,7 +460,7 @@
                 }
                 if (text.left(strlen(SIGN_KEY)) == SIGN_KEY){
                     QString input = QFile::decodeName(user_file("m.").c_str());
-                    input  += QString::number((unsigned)msg);
+                    input  += QString::number((unsigned long)msg);
                     input += ".in";
                     QFile in(input);
                     if (!in.open(IO_WriteOnly | IO_Truncate)){
diff -Naur sim-0.9.3.orig/plugins/icq/icqclient.cpp sim-0.9.3/plugins/icq/icqclient.cpp
--- sim-0.9.3.orig/plugins/icq/icqclient.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/icq/icqclient.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -2623,7 +2623,7 @@
         CommandDef *cmd = (CommandDef*)(e->param());
         if (cmd->menu_id == MenuEncoding){
             if (cmd->id == CmdChangeEncoding){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact == NULL)
                     return NULL;
                 ClientDataIterator itClient(contact->clientData, this);
@@ -2725,7 +2725,7 @@
         }
         if ((cmd->bar_id == ToolBarContainer) || (cmd->bar_id == BarHistory)){
             if (cmd->id == CmdChangeEncoding){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact == NULL){
                     cmd->flags |= BTN_HIDE;
                     return e->param();
@@ -2749,7 +2749,7 @@
         }
         if (cmd->menu_id == MenuContactGroup){
             if (cmd->id == CmdVisibleList){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact == NULL)
                     return NULL;
                 for (unsigned i = 0; i < getContacts()->nClients(); i++){
@@ -2772,7 +2772,7 @@
                 return bOK ? e->param() : NULL;
             }
             if (cmd->id == CmdInvisibleList){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact == NULL)
                     return NULL;
                 for (unsigned i = 0; i < getContacts()->nClients(); i++){
@@ -2803,7 +2803,7 @@
                 ICQPlugin::core->setShowAllEncodings(!ICQPlugin::core->getShowAllEncodings());
                 return e->param();
             }
-            Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+            Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
             if (contact == NULL)
                 return NULL;
             QCString codecStr;
@@ -2896,7 +2896,7 @@
         }
         if (cmd->menu_id == MenuContactGroup){
             if (cmd->id == CmdVisibleList){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact == NULL)
                     return NULL;
                 ICQUserData *data;
@@ -2909,7 +2909,7 @@
                 return e->param();
             }
             if (cmd->id == CmdInvisibleList){
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact == NULL)
                     return NULL;
                 ICQUserData *data;
diff -Naur sim-0.9.3.orig/plugins/icq/icqdirect.cpp sim-0.9.3/plugins/icq/icqdirect.cpp
--- sim-0.9.3.orig/plugins/icq/icqdirect.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/icq/icqdirect.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -277,7 +277,7 @@
     }
     if (m_state != Logged){
         ICQPlugin *plugin = static_cast<ICQPlugin*>(m_client->protocol()->plugin());
-        log_packet(m_socket->readBuffer, false, plugin->ICQDirectPacket, number((unsigned)this).c_str());
+        log_packet(m_socket->readBuffer, false, plugin->ICQDirectPacket, number((unsigned long)this).c_str());
     }
     switch (m_state){
     case Logged:{
@@ -402,7 +402,7 @@
     if (m_version >= 7)
         m_socket->writeBuffer.pack(0x00000000L);
     ICQPlugin *plugin = static_cast<ICQPlugin*>(m_client->protocol()->plugin());
-    log_packet(m_socket->writeBuffer, true, plugin->ICQDirectPacket, number((unsigned)this).c_str());
+    log_packet(m_socket->writeBuffer, true, plugin->ICQDirectPacket, number((unsigned long)this).c_str());
     m_socket->write();
 }
 
@@ -413,7 +413,7 @@
     m_socket->writeBuffer.pack((unsigned short)0x0001);
     m_socket->writeBuffer.pack((unsigned short)0x0000);
     ICQPlugin *plugin = static_cast<ICQPlugin*>(m_client->protocol()->plugin());
-    log_packet(m_socket->writeBuffer, true, plugin->ICQDirectPacket, number((unsigned)this).c_str());
+    log_packet(m_socket->writeBuffer, true, plugin->ICQDirectPacket, number((unsigned long)this).c_str());
     m_socket->write();
 }
 
@@ -506,7 +506,7 @@
     case WaitInit2:
         if (m_bIncoming){
             ICQPlugin *plugin = static_cast<ICQPlugin*>(m_client->protocol()->plugin());
-            log_packet(m_socket->readBuffer, false, plugin->ICQDirectPacket, number((unsigned)this).c_str());
+            log_packet(m_socket->readBuffer, false, plugin->ICQDirectPacket, number((unsigned long)this).c_str());
             if (m_version < 8){
                 if (m_data->Direct.ptr){
                     m_socket->error_state("Direct connection already established");
@@ -517,7 +517,7 @@
                 break;
             }
             plugin = static_cast<ICQPlugin*>(m_client->protocol()->plugin());
-            log_packet(m_socket->readBuffer, false, plugin->ICQDirectPacket, number((unsigned)this).c_str());
+            log_packet(m_socket->readBuffer, false, plugin->ICQDirectPacket, number((unsigned long)this).c_str());
             m_socket->readBuffer.incReadPos(13);
             char p[16];
             m_socket->readBuffer.unpack(p, 16);
@@ -1559,7 +1559,7 @@
     }
     m_name += number(m_data->Uin.value);
     m_name += ".";
-    m_name += number((unsigned)this);
+    m_name += number((unsigned long)this);
     return m_name.c_str();
 }
 
diff -Naur sim-0.9.3.orig/plugins/icq/icqsearch.cpp sim-0.9.3/plugins/icq/icqsearch.cpp
--- sim-0.9.3.orig/plugins/icq/icqsearch.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/icq/icqsearch.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -350,7 +350,7 @@
 void *ICQSearch::processEvent(Event *e)
 {
     if (e->type() == EventRandomChat){
-        m_randomUin = (unsigned)(e->param());
+        m_randomUin = (unsigned long)(e->param());
         if (m_randomUin == 0){
             edtStatus->setText(i18n("Search fail"));
         }else{
diff -Naur sim-0.9.3.orig/plugins/jabber/jabberclient.cpp sim-0.9.3/plugins/jabber/jabberclient.cpp
--- sim-0.9.3.orig/plugins/jabber/jabberclient.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/jabber/jabberclient.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -448,7 +448,7 @@
                 cmd->flags |= COMMAND_RECURSIVE;
                 return e->param();
             }
-            Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+            Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
             if (contact == NULL)
                 return NULL;
             clientData *data;
@@ -504,7 +504,7 @@
                 if (url.isEmpty())
                     url = QString::fromUtf8(getServer());
             }else{
-                Contact *contact = getContacts()->contact((unsigned)(cmd->param));
+                Contact *contact = getContacts()->contact((unsigned long)(cmd->param));
                 if (contact){
                     clientData *data;
                     ClientDataIterator it(contact->clientData, this);
@@ -560,7 +560,7 @@
     }
     if (e->type() == EventTemplateExpanded){
         TemplateExpand *t = (TemplateExpand*)(e->param());
-        setStatus((unsigned)(t->param), quoteString(t->tmpl, quoteNOBR).utf8());
+        setStatus((unsigned long)(t->param), quoteString(t->tmpl, quoteNOBR).utf8());
     }
     if (e->type() == EventContactChanged){
         Contact *contact = (Contact*)(e->param());
@@ -1881,7 +1881,7 @@
             if (!resource.empty())
                 resource += ";";
             const char *dicon = get_icon(data, atol(get_str(data->ResourceStatus, i)), false);
-            resource += number((unsigned)dicon);
+            resource += number((unsigned long)dicon);
             resource += ",";
             resource += quoteChars(get_str(data->Resources, i), ";");
         }
diff -Naur sim-0.9.3.orig/plugins/shortcuts/shortcuts.cpp sim-0.9.3/plugins/shortcuts/shortcuts.cpp
--- sim-0.9.3.orig/plugins/shortcuts/shortcuts.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/shortcuts/shortcuts.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -503,7 +503,7 @@
         }
     }
     if (e->type() == EventCommandRemove){
-        unsigned id = (unsigned)(e->param());
+        unsigned id = (unsigned long)(e->param());
         MAP_STR::iterator it_key = oldKeys.find(id);
         if (it_key != oldKeys.end())
             oldKeys.erase(it_key);
diff -Naur sim-0.9.3.orig/plugins/sound/sound.cpp sim-0.9.3/plugins/sound/sound.cpp
--- sim-0.9.3.orig/plugins/sound/sound.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/plugins/sound/sound.cpp	2005-08-05 15:11:35.000000000 +0200
@@ -256,7 +256,7 @@
         bool bEnable = !data->Disable.bValue;
         if (bEnable && data->NoSoundIfActive.bValue){
             Event e(EventActiveContact);
-            if ((unsigned)(e.process()) == contact->id())
+            if ((unsigned long)(e.process()) == contact->id())
                 bEnable = false;
         }
         if (bEnable){
diff -Naur sim-0.9.3.orig/sim/api/cfg.cpp sim-0.9.3/sim/api/cfg.cpp
--- sim-0.9.3.orig/sim/api/cfg.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/sim/api/cfg.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -567,7 +567,7 @@
                 break;
             case DATA_ULONG:
             case DATA_LONG:
-                data->value = (unsigned)(def->def_value);
+                data->value = (unsigned long)(def->def_value);
                 break;
             case DATA_BOOL:
                 data->bValue = (def->def_value != NULL);
diff -Naur sim-0.9.3.orig/sim/api/cmddef.cpp sim-0.9.3/sim/api/cmddef.cpp
--- sim-0.9.3.orig/sim/api/cmddef.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/sim/api/cmddef.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -121,7 +121,7 @@
         }
         break;
     case EventCommandRemove:
-        if (delCommand((unsigned)(e->param())))
+        if (delCommand((unsigned long)(e->param())))
             cfg.clear();
         break;
     }
diff -Naur sim-0.9.3.orig/sim/api/plugins.cpp sim-0.9.3/sim/api/plugins.cpp
--- sim-0.9.3.orig/sim/api/plugins.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/sim/api/plugins.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -214,7 +214,7 @@
             return;
     }
     Event eStart(EventInit);
-    if ((int)eStart.process() == -1) {
+    if ((long)eStart.process() == -1) {
         log(L_ERROR,"EventInit failed - aborting!");
         m_bAbort = true;
         return;
@@ -248,7 +248,7 @@
         p = (CmdParam*)(e->param());
         return (void*)findParam(p->arg, p->descr, p->value);
     case EventPluginGetInfo:
-        return getInfo((unsigned)(e->param()));
+        return getInfo((unsigned long)(e->param()));
     case EventApplyPlugin:
         return (void*)setInfo((const char*)(e->param()));
     case EventPluginsUnload:
@@ -429,7 +429,7 @@
         info.base = m_base;
     }
     info.plugin = info.info->create(info.base, m_bInInit, info.config);
-    if ((unsigned)(info.plugin) == ABORT_LOADING){
+    if ((unsigned long)(info.plugin) == ABORT_LOADING){
         m_bAbort = true;
         info.plugin = NULL;
     }
diff -Naur sim-0.9.3.orig/sim/api/simapi.cpp sim-0.9.3/sim/api/simapi.cpp
--- sim-0.9.3.orig/sim/api/simapi.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/sim/api/simapi.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -491,7 +491,7 @@
 {
     Event e(EventGetIcon, (void*)name);
     const QIconSet *res = (const QIconSet*)e.process();
-    if ((unsigned)res == (unsigned)(-1))
+    if ((unsigned long)res == (unsigned)(-1))
         res = NULL;
     return res;
 }
diff -Naur sim-0.9.3.orig/sim/ui/toolbtn.cpp sim-0.9.3/sim/ui/toolbtn.cpp
--- sim-0.9.3.orig/sim/ui/toolbtn.cpp	2005-08-05 15:10:57.000000000 +0200
+++ sim-0.9.3/sim/ui/toolbtn.cpp	2005-08-05 15:11:22.000000000 +0200
@@ -616,7 +616,7 @@
             toolBarChanged();
         break;
     case EventCommandRemove:
-        button = buttons->remove((unsigned)(e->param()));
+        button = buttons->remove((unsigned long)(e->param()));
         if (button)
             delete button;
         break;
