--- ./Makefile.org.orig	2003-04-08 04:54:32.000000000 -0700
+++ ./Makefile.org	2003-04-20 13:44:59.000000000 -0700
@@ -10,6 +10,7 @@
 SHLIB_MAJOR=
 SHLIB_MINOR=
 SHLIB_EXT=
+SHLIB_SOVER=0
 PLATFORM=dist
 OPTIONS=
 CONFIGURE_ARGS=
@@ -283,7 +284,7 @@
 	fi; \
 	( set -x; ${CC} ${SHARED_LDFLAGS} \
 		-shared -o lib$$i.so.${SHLIB_MAJOR}.${SHLIB_MINOR} \
-		-Wl,-soname=lib$$i.so.${SHLIB_MAJOR}.${SHLIB_MINOR} \
+		-Wl,-soname=lib$$i.so.${SHLIB_SOVER} \
 		-Wl,-Bsymbolic \
 		-Wl,--whole-archive lib$$i.a \
 		-Wl,--no-whole-archive $$libs ${EX_LIBS} -lc ) || exit 1; \
