diff -ur openssl-0.9.7g.orig/Makefile.org openssl-0.9.7g/Makefile.org
--- openssl-0.9.7g.orig/Makefile.org	2005-03-15 10:46:13.000000000 +0100
+++ openssl-0.9.7g/Makefile.org	2005-04-14 14:58:46.000000000 +0200
@@ -10,6 +10,7 @@
 SHLIB_MAJOR=
 SHLIB_MINOR=
 SHLIB_EXT=
+SHLIB_SOVER=0
 PLATFORM=dist
 OPTIONS=
 CONFIGURE_ARGS=
@@ -306,7 +307,7 @@
 	fi; \
 	( set -x; ${CC} ${SHARED_LDFLAGS} \
 		-shared -o lib$$i.so.${SHLIB_MAJOR}.${SHLIB_MINOR} \
-		-Wl,-soname=lib$$i.so.${SHLIB_MAJOR}.${SHLIB_MINOR} \
+		-Wl,-soname=lib$$i.so.${SHLIB_SOVER} \
 		-Wl,-Bsymbolic \
 		-Wl,--whole-archive lib$$i.a \
 		-Wl,--no-whole-archive $$libs ${EX_LIBS} -lc ) || exit 1; \
