--- qtwebengine-everywhere-src-6.9.0/src/3rdparty/chromium/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc	2025-03-29 01:50:57.000000000 +0100
+++ qtwebengine-everywhere-src-6.9.0.new/src/3rdparty/chromium/third_party/webrtc/modules/video_capture/linux/pipewire_session.cc	2025-04-04 13:58:43.272633129 +0200
@@ -85,7 +85,7 @@
       .param = OnNodeParam,
   };
 
-  pw_node_add_listener(proxy_, &node_listener_, &node_events, this);
+  pw_node_add_listener(reinterpret_cast<pw_node*>(proxy_), &node_listener_, &node_events, this);
 }
 
 // static
@@ -117,7 +117,7 @@
       uint32_t id = info->params[i].id;
       if (id == SPA_PARAM_EnumFormat &&
           info->params[i].flags & SPA_PARAM_INFO_READ) {
-        pw_node_enum_params(that->proxy_, 0, id, 0, UINT32_MAX, nullptr);
+        pw_node_enum_params(reinterpret_cast<pw_node*>(that->proxy_), 0, id, 0, UINT32_MAX, nullptr);
         break;
       }
     }
