diff -Naur Thunar-0.4.0rc1/configure Thunar-0.4.0rc1-p/configure
--- Thunar-0.4.0rc1/configure	2006-09-03 13:59:12.000000000 +0200
+++ Thunar-0.4.0rc1-p/configure	2006-09-23 16:20:52.000000000 +0200
@@ -30022,7 +30022,7 @@
     echo "$as_me:$LINENO: result: $enable_debug" >&5
 echo "${ECHO_T}$enable_debug" >&6
 
-    PLATFORM_CPPFLAGS="$PLATFORM_CPPFLAGS -NDEBUG"
+    PLATFORM_CPPFLAGS="$PLATFORM_CPPFLAGS -DNDEBUG"
 
     PLATFORM_CPPFLAGS="$PLATFORM_CPPFLAGS -DG_DISABLE_CAST_CHECKS"
 
diff -Naur Thunar-0.4.0rc1/configure.in Thunar-0.4.0rc1-p/configure.in
--- Thunar-0.4.0rc1/configure.in	2006-09-03 13:58:09.000000000 +0200
+++ Thunar-0.4.0rc1-p/configure.in	2006-09-23 16:20:52.000000000 +0200
@@ -311,7 +311,7 @@
   AC_MSG_RESULT([$enable_debug])
 
   dnl Disable debugging (release build)
-  PLATFORM_CPPFLAGS="$PLATFORM_CPPFLAGS -NDEBUG"
+  PLATFORM_CPPFLAGS="$PLATFORM_CPPFLAGS -DNDEBUG"
 
   dnl Disable object cast checks
   PLATFORM_CPPFLAGS="$PLATFORM_CPPFLAGS -DG_DISABLE_CAST_CHECKS"
