diff -Naur ktp-call-ui-19.08.3/libqtf/CMakeLists.txt ktp-call-ui-19.08.3-p/libqtf/CMakeLists.txt
--- ktp-call-ui-19.08.3/libqtf/CMakeLists.txt	2017-03-23 23:34:40.000000000 +0100
+++ ktp-call-ui-19.08.3-p/libqtf/CMakeLists.txt	2019-11-14 12:20:45.333434177 +0100
@@ -21,6 +21,7 @@
 add_library(qtf STATIC qtf.cpp)
 
 target_link_libraries(qtf
+    Qt5::DBus
     ${QTGSTREAMER_LIBRARIES}
     ${TP_FARSTREAM_LDFLAGS}
     ${FARSTREAM_LDFLAGS}
