diff -aur ne-2.1/src/ext.c ne-2.1-2/src/ext.c
--- ne-2.1/src/ext.c	2010-05-10 11:42:34.000000000 -0500
+++ ne-2.1-2/src/ext.c	2010-09-07 01:08:08.300002507 -0500
@@ -49,6 +49,7 @@
 	{ "eps", "ps" },
 	{ "f", "fortran" },
 	{ "for", "fortran" },
+	{ "frugalbuild", "frugalbuild" },
 	{ "h", "c" },
 	{ "h++", "c" },
 	{ "hpp", "c" },
diff -aur ne-2.1/src/prefs.c ne-2.1-2/src/prefs.c
--- ne-2.1/src/prefs.c	2010-04-23 04:05:06.000000000 -0500
+++ ne-2.1-2/src/prefs.c	2010-09-07 01:34:24.314998176 -0500
@@ -60,7 +60,7 @@
 	for(i = strlen(filename); i-- != 0;)
 		if (filename[i] == '.') return &filename[i + 1];
 
-	return NULL;
+	return filename;
 }
 
 
