diff -Naur subversion-1.4.0/build/ac-macros/neon.m4 subversion-1.4.0-p/build/ac-macros/neon.m4
--- subversion-1.4.0/build/ac-macros/neon.m4	2006-01-07 01:56:50.000000000 +0100
+++ subversion-1.4.0-p/build/ac-macros/neon.m4	2006-09-26 17:21:35.886018022 +0200
@@ -50,9 +50,13 @@
       NEON_VERSION=`cat $abs_srcdir/neon/.version`
       AC_MSG_RESULT([$NEON_VERSION])
 
-      if test -n "`echo \"$NEON_VERSION\" | grep '^0\.25\.'`" ; then
+      if test -n "`echo \"$NEON_VERSION\" | grep '^0\.2[[56]]\.'`" ; then
         AC_DEFINE_UNQUOTED([SVN_NEON_0_25], [1],
-                           [Defined if have Neon 0.25.x instead of 0.24.7.])
+                           [Define to 1 if you have Neon 0.25 or later.])
+      fi
+      if test -n "`echo \"$NEON_VERSION\" | grep '^0\.26\.'`" ; then
+        AC_DEFINE_UNQUOTED([SVN_NEON_0_26], [1],
+                           [Define to 1 if you have Neon 0.26 or later.])
       fi
 
       for svn_allowed_neon in $NEON_ALLOWED_LIST; do
@@ -123,9 +127,13 @@
       NEON_VERSION=`$neon_config --version | sed -e 's/^neon //'`
       AC_MSG_RESULT([$NEON_VERSION])
 
-      if test -n "`echo \"$NEON_VERSION\" | grep '^0\.25\.'`" ; then
+      if test -n "`echo \"$NEON_VERSION\" | grep '^0\.2[[56]]\.'`" ; then
         AC_DEFINE_UNQUOTED([SVN_NEON_0_25], [1],
-                           [Defined if have Neon 0.25.x instead of 0.24.7.])
+                           [Define to 1 if you have Neon 0.25 or later.])
+      fi
+      if test -n "`echo \"$NEON_VERSION\" | grep '^0\.26\.'`" ; then
+        AC_DEFINE_UNQUOTED([SVN_NEON_0_26], [1],
+                           [Define to 1 if you have Neon 0.26 or later.])
       fi
 
       for svn_allowed_neon in $NEON_ALLOWED_LIST; do
diff -Naur subversion-1.4.0/configure.in subversion-1.4.0-p/configure.in
--- subversion-1.4.0/configure.in	2006-07-13 15:48:48.000000000 +0200
+++ subversion-1.4.0-p/configure.in	2006-09-26 17:21:35.887017870 +0200
@@ -69,8 +69,8 @@
 
 # Either a space-separated list of allowable Neon versions, or "any" to
 # mean allow anything.
-NEON_ALLOWED_LIST="0.24.7 0.25.0 0.25.1 0.25.2 0.25.3 0.25.4 0.25.5"
-NEON_LATEST_WORKING_VER=0.25.5
+NEON_ALLOWED_LIST="0.24.7 0.25.0 0.25.1 0.25.2 0.25.3 0.25.4 0.25.5 0.26.1"
+NEON_LATEST_WORKING_VER=0.26.1
 NEON_URL="http://www.webdav.org/neon/neon-${NEON_LATEST_WORKING_VER}.tar.gz"
 dnl You can skip the neon version check only if you know what you are doing 
 AC_ARG_ENABLE(neon-version-check,
diff -Naur subversion-1.4.0/subversion/libsvn_ra_dav/session.c subversion-1.4.0-p/subversion/libsvn_ra_dav/session.c
--- subversion-1.4.0/subversion/libsvn_ra_dav/session.c	2006-03-25 19:01:16.000000000 +0100
+++ subversion-1.4.0-p/subversion/libsvn_ra_dav/session.c	2006-09-26 17:21:35.888017718 +0200
@@ -1448,7 +1448,11 @@
 static void
 lock_receiver(void *userdata,
               const struct ne_lock *lock,
+#ifdef SVN_NEON_0_26
+              const ne_uri *uri,
+#else
               const char *uri,
+#endif
               const ne_status *status)
 {
   struct receiver_baton *rb = userdata;
