diff -rN -u xqed-1.0.3.orig/Makefile xqed-1.0.3/Makefile
--- xqed-1.0.3.orig/Makefile	2004-03-05 19:28:08.000000000 +0100
+++ xqed-1.0.3/Makefile	2005-06-15 11:19:58.000000000 +0200
@@ -1,13 +1,13 @@
 BIN=/usr/bin
 
 install: Qed.py Xqed.py xqed setup.py
-	python setup.py install
-	cp qed xqed $(BIN)
-	-@ln -s $(BIN)/xqed $(BIN)/angolul
-	-@ln -s $(BIN)/xqed $(BIN)/magyarul
-	-@mkdir /usr/share/qed
-	cp eng-hu.dic eng-hu.index pro.help use.help /usr/share/qed
-	cp xqed.xbm xqed.gif xqed-bg.gif xqed-local.gif xqed-idiom.gif xqed-window.gif pronouncing /usr/share/qed
+	python setup.py install --root="$(DESTDIR)"
+	cp qed xqed $(DESTDIR)$(BIN)
+	-@ln -s xqed $(DESTDIR)$(BIN)/angolul
+	-@ln -s xqed $(DESTDIR)$(BIN)/magyarul
+	-@mkdir -p $(DESTDIR)/usr/share/qed
+	cp eng-hu.dic eng-hu.index pro.help use.help $(DESTDIR)/usr/share/qed
+	cp xqed.xbm xqed.gif xqed-bg.gif xqed-local.gif xqed-idiom.gif xqed-window.gif pronouncing $(DESTDIR)/usr/share/qed
 
 clean:
 	-@rm -fr build
