diff -Naur virtuoso-opensource-6.1.0/binsrc/Makefile.am virtuoso-opensource-6.1.0-build-only-what-kde-need/binsrc/Makefile.am
--- virtuoso-opensource-6.1.0/binsrc/Makefile.am	2009-04-21 23:21:06.000000000 +0200
+++ virtuoso-opensource-6.1.0-build-only-what-kde-need/binsrc/Makefile.am	2010-02-08 18:14:39.000000000 +0100
@@ -19,8 +19,7 @@
 #  
 #  
 
-SUBDIRS = dav mono virtuoso tests rdf_mappers driver maildrop sqldoc hosting bpel b3s tutorial yacutia samples vsp ws sync vspx vad cached_resources virtodbc virtoledb virtuoso_sink xddl VirtuosoClient.Net oat isparql jena sesame sesame2 sesame3 redland
-
+SUBDIRS = virtuoso driver 
 
 # ----------------------------------------------------------------------
 #
diff -Naur virtuoso-opensource-6.1.0/binsrc/virtuoso/Makefile.am virtuoso-opensource-6.1.0-build-only-what-kde-need/binsrc/virtuoso/Makefile.am
--- virtuoso-opensource-6.1.0/binsrc/virtuoso/Makefile.am	2009-10-16 18:49:43.000000000 +0200
+++ virtuoso-opensource-6.1.0-build-only-what-kde-need/binsrc/virtuoso/Makefile.am	2010-02-08 18:13:28.000000000 +0100
@@ -41,7 +41,7 @@
 
 LIBS		+= @ZLIB_LIB@
 
-lib_LTLIBRARIES = libvirtuoso-t.la $(IODBC_LIBS) $(MONO_LIBS) $(IODBC_MONO_LIBS)
+noinst_LTLIBRARIES = libvirtuoso-t.la $(IODBC_LIBS) $(MONO_LIBS) $(IODBC_MONO_LIBS)
 bin_PROGRAMS = virtuoso-t $(IODBC_PROGS)
 
 noinst_SCRIPTS = virtuoso
diff -Naur virtuoso-opensource-6.1.0/libsrc/Makefile.am virtuoso-opensource-6.1.0-build-only-what-kde-need/libsrc/Makefile.am
--- virtuoso-opensource-6.1.0/libsrc/Makefile.am	2007-12-11 14:47:44.000000000 +0100
+++ virtuoso-opensource-6.1.0-build-only-what-kde-need/libsrc/Makefile.am	2010-02-08 18:13:28.000000000 +0100
@@ -19,7 +19,7 @@
 #  
 #  
 
-SUBDIRS = util zlib odbcsdk Dk Thread langfunc Wi plugin Tidy Xml.new JDBCDriverType4
+SUBDIRS = util zlib odbcsdk Dk Thread langfunc Wi plugin Tidy Xml.new
 
 noinst_HEADERS = Dk.h libutil.h
 
diff -Naur virtuoso-opensource-6.1.0/libsrc/Wi/http_client.c virtuoso-opensource-6.1.0-build-only-what-kde-need/libsrc/Wi/http_client.c
--- virtuoso-opensource-6.1.0/libsrc/Wi/http_client.c	2010-02-02 21:27:43.000000000 +0100
+++ virtuoso-opensource-6.1.0-build-only-what-kde-need/libsrc/Wi/http_client.c	2010-02-08 18:15:01.000000000 +0100
@@ -412,7 +412,7 @@
   socksreq[8] = 0; /* no name */
   if (name)
     {
-      strncat ((char*)socksreq + 8, name, sizeof(socksreq) - 8);
+      strncat ((char*)socksreq + 8, name, sizeof(socksreq) - 8 - strlen(name) - 1);
       socksreq[sizeof (socksreq) - 1] = 0;
       packetsize = 9 + strlen ((char *) socksreq + 8);
     }
diff -Naur virtuoso-opensource-6.1.0/libsrc/Wi/sqlovdb.c virtuoso-opensource-6.1.0-build-only-what-kde-need/libsrc/Wi/sqlovdb.c
--- virtuoso-opensource-6.1.0/libsrc/Wi/sqlovdb.c	2009-08-19 09:54:34.000000000 +0200
+++ virtuoso-opensource-6.1.0-build-only-what-kde-need/libsrc/Wi/sqlovdb.c	2010-02-08 18:15:01.000000000 +0100
@@ -104,7 +104,7 @@
   return loc;
 }
 
-#define NO_VDB GPF_T1 ("This build does not include virtual database support.")
+#define NO_VDB do { GPF_T1 ("This build does not include virtual database support."); abort(); } while(0)
 
 void
 sqlo_table_new_locus (sqlo_t * so, df_elt_t * tb_dfe, remote_ds_t * rds, dk_set_t col_preds, dk_set_t * after_test, dk_set_t after_join_test, dk_set_t * vdb_join_test)
diff -Naur virtuoso-opensource-6.1.0/Makefile.am virtuoso-opensource-6.1.0-build-only-what-kde-need/Makefile.am
--- virtuoso-opensource-6.1.0/Makefile.am	2010-02-03 17:09:01.000000000 +0100
+++ virtuoso-opensource-6.1.0-build-only-what-kde-need/Makefile.am	2010-02-08 18:13:28.000000000 +0100
@@ -23,7 +23,7 @@
 ACLOCAL_AMFLAGS 	= -I binsrc/config
 
 
-SUBDIRS = . docsrc libsrc binsrc appsrc
+SUBDIRS = . libsrc binsrc
 
 dist_doc_DATA = \
 	AUTHORS \
