--- SystemTray/src/main.c.orig	2005-12-21 12:22:13.000000000 +0100
+++ SystemTray/src/main.c	2005-12-21 12:36:06.000000000 +0100
@@ -160,12 +160,6 @@
     return gdk_handler(display, error);
 }
 
-typedef struct applet_info {
-  PanelLocation loc;
-  int margin;
-} AppletInfo;
-AppletInfo *get_position(GtkWidget *plug);
-
 #define SHORT_OPS "hv"
 
 #ifdef HAVE_GETOPT_LONG
@@ -231,7 +225,7 @@
       window_id = atol(argv[1]);
       plug = gtk_plug_new(window_id);
 
-      info = get_position(plug);
+      info = rox_applet_get_position(plug);
       if (info) 
 	{
 	  location = info->loc;
