diff -Naur libechonest-2.3.1/CMakeLists.txt libechonest-2.3.1-p/CMakeLists.txt
--- libechonest-2.3.1/CMakeLists.txt	2015-02-07 20:56:50.000000000 +0100
+++ libechonest-2.3.1-p/CMakeLists.txt	2018-06-16 01:58:42.056799006 +0200
@@ -39,9 +39,6 @@
     endif()
     include( ${QT_USE_FILE} )
 
-    macro(qt5_use_modules)
-    endmacro()
-
     macro(qt_wrap_cpp)
         qt4_wrap_cpp(${ARGN})
     endmacro()
diff -Naur libechonest-2.3.1/src/CMakeLists.txt libechonest-2.3.1-p/src/CMakeLists.txt
--- libechonest-2.3.1/src/CMakeLists.txt	2015-02-07 20:56:50.000000000 +0100
+++ libechonest-2.3.1-p/src/CMakeLists.txt	2018-06-16 02:04:20.118732390 +0200
@@ -1,6 +1,9 @@
 
 include_directories( ${QJSON_INCLUDE_DIR}/.. ${CMAKE_CURRENT_SOURCE_DIR} )
 
+find_package(Qt5Network REQUIRED)
+find_package(Qt5Xml REQUIRED)
+
 set( LIBECHONEST_SRC
         Track.cpp
         Song.cpp
@@ -47,7 +50,7 @@
 add_library( ${ECHONEST_LIB_TARGET_NAME} SHARED ${LIBECHONEST_SRC} )
 target_link_libraries( ${ECHONEST_LIB_TARGET_NAME} ${QT_QTCORE_LIBRARY} ${QT_QTNETWORK_LIBRARY} ${QJSON_LIBRARIES} )
 set_target_properties( ${ECHONEST_LIB_TARGET_NAME} PROPERTIES VERSION ${ECHONEST_LIB_VERSION} SOVERSION ${ECHONEST_LIB_VERSION_SONAME} )
-qt5_use_modules( ${ECHONEST_LIB_TARGET_NAME} Core Network Xml )
+target_link_libraries( ${ECHONEST_LIB_TARGET_NAME} Qt5::Core Qt5::Network Qt5::Xml )
 install( TARGETS ${ECHONEST_LIB_TARGET_NAME}   RUNTIME DESTINATION bin
                             LIBRARY DESTINATION lib${LIB_SUFFIX}
                             ARCHIVE DESTINATION lib${LIB_SUFFIX}
diff -Naur libechonest-2.3.1/tests/CMakeLists.txt libechonest-2.3.1-p/tests/CMakeLists.txt
--- libechonest-2.3.1/tests/CMakeLists.txt	2015-02-07 20:56:50.000000000 +0100
+++ libechonest-2.3.1-p/tests/CMakeLists.txt	2018-06-16 02:05:15.285047706 +0200
@@ -1,5 +1,8 @@
 ########### next target ###############
 # QTEST_MAIN is using QApplication when QT_GUI_LIB is defined
+find_package(Qt5Network REQUIRED)
+find_package(Qt5Test REQUIRED)
+
 remove_definitions(-DQT_GUI_LIB)
 
 add_definitions( -DDATA_DIR=\"${CMAKE_CURRENT_SOURCE_DIR}/test_data\" )
@@ -17,7 +20,7 @@
     add_executable(${_name} ${_source})
     target_link_libraries(${_name} ${ECHONEST_LIB_TARGET_NAME} ${QT_QTCORE_LIBRARY} ${QT_QTTEST_LIBRARY} ${QTCORE_QTNETWORK_LIBRARY} ${QT_QTXML_LIBRARY} ${QJSON_LIBRARIES} )
     add_test(${_name}-test ${EXECUTABLE_OUTPUT_PATH}/${_name})
-    qt5_use_modules(${_name} Network Test)
+    target_link_libraries(${_name} Qt5::Network Qt5::Test)
 endmacro(add_echonest_test)
 
 add_echonest_test( ArtistTest.cpp )
