diff -Naur xorg-server-0.99.3.orig/hw/xfree86/os-support/linux/lnx_agp.c xorg-server-0.99.3/hw/xfree86/os-support/linux/lnx_agp.c
--- xorg-server-0.99.3.orig/hw/xfree86/os-support/linux/lnx_agp.c	2005-11-19 13:58:12.336920250 +0100
+++ xorg-server-0.99.3/hw/xfree86/os-support/linux/lnx_agp.c	2005-11-19 13:58:33.118219000 +0100
@@ -20,6 +20,8 @@
 #include "xf86OSpriv.h"
 
 #if defined(linux)
+#undef _LINUX_TYPES_H
+#include <linux/types.h>
 #include <asm/ioctl.h>
 #include <linux/agpgart.h>
 #elif defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(__NetBSD__) || defined(__OpenBSD__) || defined(__DragonFly__)
