diff -rc darcsweb/darcsweb.cgi darcsweb.patch/darcsweb.cgi
*** darcsweb/darcsweb.cgi	2005-09-07 14:30:11.000000000 +0200
--- darcsweb.patch/darcsweb.cgi	2005-09-07 14:45:00.000000000 +0200
***************
*** 19,24 ****
--- 19,26 ----
  import xml.sax
  from xml.sax.saxutils import escape
  
+ # first search config in /etc/darcsweb/
+ sys.path.insert(0, '/etc/darcsweb/')
  
  # empty configuration class, we will fill it in later depending on the repo
  class config:
Seulement dans darcsweb.patch/: darcsweb.cgi~
