diff -aur vice-2.2/src/arch/unix/x11/xaw/x11video.c vice-2.2-2/src/arch/unix/x11/xaw/x11video.c
--- vice-2.2/src/arch/unix/x11/xaw/x11video.c	2009-10-21 11:46:45.000000000 -0500
+++ vice-2.2-2/src/arch/unix/x11/xaw/x11video.c	2010-05-15 20:16:38.000000000 -0500
@@ -59,6 +59,8 @@
 #include <X11/Xutil.h>
 #include <X11/Intrinsic.h>
 #include <X11/cursorfont.h>
+#include <X11/Xmd.h>
+#include <X11/extensions/shmproto.h>
 
 #include <stdio.h>
 #include <stdlib.h>
@@ -86,7 +88,6 @@
 #endif
 
 static log_t x11video_log = LOG_ERR;
-
 #ifdef HAVE_XVIDEO
 #include "video/video-resources.h"
 #endif
