diff -Naur libgksu-1.9.2.orig/libgksu/libgksu.c libgksu-1.9.2/libgksu/libgksu.c
--- libgksu-1.9.2.orig/libgksu/libgksu.c	2006-04-30 22:12:47.000000000 +0000
+++ libgksu-1.9.2/libgksu/libgksu.c	2006-05-15 01:00:55.000000000 +0000
@@ -1079,7 +1079,7 @@
 
   /* get the authorization token */
   tmp = g_strdup_printf ("%s list %s | "
-			 "head -1 | awk '{ print $3 }'",
+			 "head -n 1 | awk '{ print $3 }'",
 			 xauth_bin,
 			 context->display);
   if ((xauth_output = popen (tmp, "r")) == NULL)
