diff -Naur libgksu-2.0.4.orig/libgksu/libgksu.c libgksu-2.0.4/libgksu/libgksu.c
--- libgksu-2.0.4.orig/libgksu/libgksu.c	2007-02-15 11:41:12.000000000 +0100
+++ libgksu-2.0.4/libgksu/libgksu.c	2007-02-15 11:41:17.000000000 +0100
@@ -1084,7 +1084,7 @@
 
   /* get the authorization token */
   tmp = g_strdup_printf ("%s list %s | "
-			 "head -1 | awk '{ print $3 }'",
+			 "head -n 1 | awk '{ print $3 }'",
 			 xauth_bin,
 			 display);
   if ((xauth_output = popen (tmp, "r")) == NULL)
