diff -Naur pysqlite-2.4.0/src/connection.c pysqlite-2.4.0-fix-utf8/src/connection.c
--- pysqlite-2.4.0/src/connection.c	2007-11-25 17:56:11.000000000 +0100
+++ pysqlite-2.4.0-fix-utf8/src/connection.c	2007-12-21 20:44:44.000000000 +0100
@@ -83,7 +83,7 @@
     Py_INCREF(&PyUnicode_Type);
     self->text_factory = (PyObject*)&PyUnicode_Type;
 
-    if (PyString_Check(database)) {
+    if (PyString_Check(database) || PyUnicode_Check(database)) {
         Py_BEGIN_ALLOW_THREADS
         rc = sqlite3_open(PyString_AsString(database), &self->db);
         Py_END_ALLOW_THREADS
