--- yaml-cpp-yaml-cpp-0.7.0/CMakeLists.txt	2021-07-10 17:53:22.000000000 +0200
+++ yaml-cpp-yaml-cpp-0.7.0.new/CMakeLists.txt	2022-03-05 17:21:32.312247682 +0100
@@ -154,7 +154,7 @@
 		"${PROJECT_BINARY_DIR}/yaml-cpp-config-version.cmake"
     DESTINATION "${CMAKE_INSTALL_DATADIR}/cmake/yaml-cpp")
   install(FILES "${PROJECT_BINARY_DIR}/yaml-cpp.pc"
-    DESTINATION ${CMAKE_INSTALL_DATADIR}/pkgconfig)
+	  DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
 endif()
 
 if(YAML_CPP_BUILD_TESTS)
