--- generically-0.1.1/generically.cabal	2001-09-09 03:46:40.000000000 +0200
+++ generically-0.1.1.new/generically.cabal	2023-07-14 13:22:54.248432305 +0200
@@ -33,7 +33,7 @@
 
 library
   default-language: Haskell2010
-  build-depends:    base >=4.9 && <4.18
+  build-depends:    base >=4.9 && <4.19
   if impl(ghc >= 9.4) && !impl(ghc >= 9.6)
     build-depends:  base-orphans >=0.8.8 && <0.10
   hs-source-dirs:   src
