--- viewer/svutil.cpp	2008-04-22 02:07:25.000000000 +0200
+++ viewer/svutil.cpp	2009-06-09 22:21:00.000000000 +0200
@@ -22,7 +22,7 @@
 
 #ifndef GRAPHICS_DISABLED
 
-#include "svutil.h"
+#include <svutil.h>
 
 #ifdef WIN32
 #include <windows.h>
@@ -43,6 +43,10 @@
 #endif
 
 #include <iostream>
+// Patch for tesseract-2.04 (compile with gcc 4.4)
+#include <string.h>
+#include <cstdio>
+#include <stdlib.h>
 
 const int kBufferSize = 65536;
 const int kMaxMsgSize = 4096;
