--- viewer/svutil.cpp	2008-04-22 02:07:25.000000000 +0200
+++ viewer/svutil.cpp	2009-06-09 22:21:00.000000000 +0200
@@ -20,7 +20,7 @@
 // SVUtil contains the SVSync and SVNetwork classes, which are used for
 // thread/process creation & synchronization and network connection.
 
-#include "svutil.h"
+#include <svutil.h>
 
 #ifdef WIN32
 #include <windows.h>
@@ -38,6 +38,10 @@
 
 #include <iostream>
 #include <string>
+// Patch for tesseract-2.03 (compile with gcc 4.4)
+#include <string.h>
+#include <cstdio>
+#include <stdlib.h>
 
 const int kBufferSize = 65536;
 const int kMaxMsgSize = 4096;
