diff -Naur qtiplot-0.9.8.5/qtiplot/qtiplot.pro qtiplot-0.9.8.5/qtiplot/qtiplot.pro
--- qtiplot-0.9.8.5/qtiplot/qtiplot.pro	2011-02-14 12:21:47.000000000 +0100
+++ qtiplot-0.9.8.5/qtiplot/qtiplot.pro	2011-04-13 13:19:28.830160242 +0200
@@ -14,12 +14,14 @@
 INCLUDEPATH       += $$QWT_INCLUDEPATH
 INCLUDEPATH       += $$QWT3D_INCLUDEPATH
 INCLUDEPATH       += $$GSL_INCLUDEPATH
+INCLUDEPATH       += $$QTASSISTANT_INCLUDEPATH
 
 # configurable libs
 LIBS         += $$MUPARSER_LIBS
 LIBS         += $$QWT_LIBS
 LIBS         += $$QWT3D_LIBS
 LIBS         += $$GSL_LIBS
+LIBS         += $$QTASSISTANT_LIBS
 
 #############################################################################
 ###################### BASIC PROJECT PROPERTIES #############################

