diff -Naur qtiplot-0.9.8.9/3rdparty/qwtplot3d/include/qwt3d_global.h qtiplot-0.9.8.9new/3rdparty/qwtplot3d/include/qwt3d_global.h
--- qtiplot-0.9.8.9/3rdparty/qwtplot3d/include/qwt3d_global.h	2011-08-24 12:25:09.000000000 +0200
+++ qtiplot-0.9.8.9new/3rdparty/qwtplot3d/include/qwt3d_global.h	2012-05-03 21:42:00.319533324 +0200
@@ -2,6 +2,7 @@
 #define QWT3D_GLOBAL_H
 
 #include <qglobal.h>
+#include <GL/glu.h>
 #if QT_VERSION < 0x040000
 #include <qmodules.h>
 #endif
diff -Naur qtiplot-0.9.8.9/3rdparty/qwtplot3d/include/qwt3d_openglhelper.h qtiplot-0.9.8.9new/3rdparty/qwtplot3d/include/qwt3d_openglhelper.h
--- qtiplot-0.9.8.9/3rdparty/qwtplot3d/include/qwt3d_openglhelper.h	2011-08-24 12:25:09.000000000 +0200
+++ qtiplot-0.9.8.9new/3rdparty/qwtplot3d/include/qwt3d_openglhelper.h	2012-05-03 21:42:27.350535011 +0200
@@ -2,6 +2,7 @@
 #define __openglhelper_2003_06_06_15_49__
 
 #include "qglobal.h"
+#include <GL/glu.h>
 #if QT_VERSION < 0x040000
 #include <qgl.h>
 #else
