diff -Naur inkscape-0.45.1/configure.ac inkscape-0.45.1-p/configure.ac
--- inkscape-0.45.1/configure.ac	2007-03-20 14:44:13.000000000 +0100
+++ inkscape-0.45.1-p/configure.ac	2007-06-07 12:23:41.000000000 +0200
@@ -41,7 +41,6 @@
 AC_LANG(C++)
 AC_ISC_POSIX
 AC_PROG_CXX
-AM_PROG_CC_STDC
 AM_PROG_AS
 AC_HEADER_STDC
 
diff -Naur inkscape-0.45.1/src/dom/cssparser.h inkscape-0.45.1-p/src/dom/cssparser.h
--- inkscape-0.45.1/src/dom/cssparser.h	2007-03-13 19:06:21.000000000 +0100
+++ inkscape-0.45.1-p/src/dom/cssparser.h	2007-06-07 12:23:16.000000000 +0200
@@ -71,6 +71,7 @@
      */
     virtual bool parseFile(const DOMString &str);
 
+#include <glib.h>
 
 protected:
 
diff -Naur inkscape-0.45.1/src/dom/lsimpl.h inkscape-0.45.1-p/src/dom/lsimpl.h
--- inkscape-0.45.1/src/dom/lsimpl.h	2007-03-13 19:06:21.000000000 +0100
+++ inkscape-0.45.1-p/src/dom/lsimpl.h	2007-06-07 12:23:16.000000000 +0200
@@ -35,6 +35,7 @@
 #include "events.h"
 #include "traversal.h"
 #include "ls.h"
+#include <glib.h>
 
 
 #include "xmlreader.h"
diff -Naur inkscape-0.45.1/src/dom/svg/svgparser.h inkscape-0.45.1-p/src/dom/svg/svgparser.h
--- inkscape-0.45.1/src/dom/svg/svgparser.h	2007-03-13 19:06:21.000000000 +0100
+++ inkscape-0.45.1-p/src/dom/svg/svgparser.h	2007-06-07 12:23:16.000000000 +0200
@@ -43,6 +43,7 @@
 namespace svg
 {
 
+#include <glib.h>
 
 class SvgParser
 {
diff -Naur inkscape-0.45.1/src/dom/uri.h inkscape-0.45.1-p/src/dom/uri.h
--- inkscape-0.45.1/src/dom/uri.h	2007-03-13 19:06:21.000000000 +0100
+++ inkscape-0.45.1-p/src/dom/uri.h	2007-06-07 12:23:16.000000000 +0200
@@ -34,6 +34,7 @@
 
 #include "dom.h"
 
+#include <glib.h>
 
 namespace org
 {
diff -Naur inkscape-0.45.1/src/dom/util/ziptool.h inkscape-0.45.1-p/src/dom/util/ziptool.h
--- inkscape-0.45.1/src/dom/util/ziptool.h	2007-03-13 19:06:21.000000000 +0100
+++ inkscape-0.45.1-p/src/dom/util/ziptool.h	2007-06-07 12:23:16.000000000 +0200
@@ -37,6 +37,7 @@
 #include <vector>
 #include <string>
 
+#include <glib.h>
 
 //########################################################################
 //#  A D L E R  3 2
diff -Naur inkscape-0.45.1/src/dom/xmlreader.h inkscape-0.45.1-p/src/dom/xmlreader.h
--- inkscape-0.45.1/src/dom/xmlreader.h	2007-03-13 19:06:21.000000000 +0100
+++ inkscape-0.45.1-p/src/dom/xmlreader.h	2007-06-07 12:23:16.000000000 +0200
@@ -41,6 +41,7 @@
 namespace dom
 {
 
+#include <glib.h>
 
 
 class XmlReader
diff -Naur inkscape-0.45.1/src/io/inkscapestream.h inkscape-0.45.1-p/src/io/inkscapestream.h
--- inkscape-0.45.1/src/io/inkscapestream.h	2007-03-13 19:06:22.000000000 +0100
+++ inkscape-0.45.1-p/src/io/inkscapestream.h	2007-06-07 12:23:16.000000000 +0200
@@ -13,6 +13,7 @@
 
 
 #include <glibmm.h>
+#include <glib.h>
 
 namespace Inkscape
 {
