diff -Naur inkscape-0.46-orig/src/2geom/sbasis-math.cpp inkscape-0.46/src/2geom/sbasis-math.cpp
--- inkscape-0.46-orig/src/2geom/sbasis-math.cpp	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/2geom/sbasis-math.cpp	2009-04-05 18:21:30.000000000 -0400
@@ -38,11 +38,11 @@
 //#define ZERO 1e-3
 
 
-namespace Geom {
+#include <math.h>
+#include <cstdio>
 
+namespace Geom {
 
-#include <stdio.h>
-#include <math.h>
 
 //-|x|-----------------------------------------------------------------------
 Piecewise<SBasis> abs(SBasis const &f){
diff -Naur inkscape-0.46-orig/src/2geom/svg-path-parser.h inkscape-0.46/src/2geom/svg-path-parser.h
--- inkscape-0.46-orig/src/2geom/svg-path-parser.h	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/2geom/svg-path-parser.h	2009-04-05 18:21:30.000000000 -0400
@@ -32,6 +32,7 @@
 #ifndef SEEN_SVG_PATH_PARSER_H
 #define SEEN_SVG_PATH_PARSER_H
 
+#include <cstdio>
 #include <vector>
 #include <iterator>
 #include <stdexcept>
diff -Naur inkscape-0.46-orig/src/dom/domimpl.cpp inkscape-0.46/src/dom/domimpl.cpp
--- inkscape-0.46-orig/src/dom/domimpl.cpp	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/dom/domimpl.cpp	2009-04-05 18:21:30.000000000 -0400
@@ -30,6 +30,8 @@
 
 #include "domimpl.h"
 
+#include <cstdio>
+
 namespace org
 {
 namespace w3c
diff -Naur inkscape-0.46-orig/src/dom/io/domstream.h inkscape-0.46/src/dom/io/domstream.h
--- inkscape-0.46-orig/src/dom/io/domstream.h	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/dom/io/domstream.h	2009-04-05 18:21:30.000000000 -0400
@@ -33,6 +33,8 @@
 
 #include <dom/dom.h>
 
+#include <cstdio>
+
 namespace org
 {
 namespace w3c
diff -Naur inkscape-0.46-orig/src/dom/io/socket.h inkscape-0.46/src/dom/io/socket.h
--- inkscape-0.46-orig/src/dom/io/socket.h	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/dom/io/socket.h	2009-04-05 18:34:14.000000000 -0400
@@ -29,6 +29,7 @@
  *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
+#include <cstdio>
 #include "dom/dom.h"
 
 #ifdef HAVE_SSL
diff -Naur inkscape-0.46-orig/src/dom/io/stringstream.h inkscape-0.46/src/dom/io/stringstream.h
--- inkscape-0.46-orig/src/dom/io/stringstream.h	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/dom/io/stringstream.h	2009-04-05 18:22:50.000000000 -0400
@@ -29,7 +29,7 @@
  *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
-
+#include <cstdio>
 #include "domstream.h"
 
 
diff -Naur inkscape-0.46-orig/src/dom/svg/svgtypes.h inkscape-0.46/src/dom/svg/svgtypes.h
--- inkscape-0.46-orig/src/dom/svg/svgtypes.h	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/dom/svg/svgtypes.h	2009-04-05 18:21:30.000000000 -0400
@@ -47,6 +47,7 @@
 #include "dom/smil.h"
 
 
+#include <cstdio>
 #include <math.h>
 
 
diff -Naur inkscape-0.46-orig/src/dom/xpathtoken.h inkscape-0.46/src/dom/xpathtoken.h
--- inkscape-0.46-orig/src/dom/xpathtoken.h	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/dom/xpathtoken.h	2009-04-05 18:21:30.000000000 -0400
@@ -35,6 +35,7 @@
 
 #include <math.h>
 
+#include <cstdio>
 #include <vector>
 
 namespace org
diff -Naur inkscape-0.46-orig/src/io/inkscapestream.h inkscape-0.46/src/io/inkscapestream.h
--- inkscape-0.46-orig/src/io/inkscapestream.h	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/io/inkscapestream.h	2009-04-05 18:21:30.000000000 -0400
@@ -12,6 +12,7 @@
  */
 
 
+#include <cstdio>
 #include <glibmm.h>
 
 namespace Inkscape
diff -Naur inkscape-0.46-orig/src/io/sys.cpp inkscape-0.46/src/io/sys.cpp
--- inkscape-0.46-orig/src/io/sys.cpp	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/io/sys.cpp	2009-04-05 18:21:30.000000000 -0400
@@ -22,6 +22,8 @@
 #include <glibmm/ustring.h>
 #include <gtk/gtkmessagedialog.h>
 
+#include <cstdio>
+
 #include "prefs-utils.h"
 #include "sys.h"
 
diff -Naur inkscape-0.46-orig/src/libavoid/vertices.h inkscape-0.46/src/libavoid/vertices.h
--- inkscape-0.46-orig/src/libavoid/vertices.h	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/libavoid/vertices.h	2009-04-05 18:21:30.000000000 -0400
@@ -23,6 +23,7 @@
 #ifndef AVOID_VERTICES_H
 #define AVOID_VERTICES_H
 
+#include <cstdio>
 #include <list>
 #include <set>
 #include <map>
diff -Naur inkscape-0.46-orig/src/libnr/nr-matrix-fns.cpp inkscape-0.46/src/libnr/nr-matrix-fns.cpp
--- inkscape-0.46-orig/src/libnr/nr-matrix-fns.cpp	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/libnr/nr-matrix-fns.cpp	2009-04-05 18:21:30.000000000 -0400
@@ -1,5 +1,7 @@
 #include <libnr/nr-matrix-fns.h>
 
+#include <cstdio>
+
 namespace NR {
 
 Matrix elliptic_quadratic_form(Matrix const &m) {
diff -Naur inkscape-0.46-orig/src/libnr/nr-matrix.cpp inkscape-0.46/src/libnr/nr-matrix.cpp
--- inkscape-0.46-orig/src/libnr/nr-matrix.cpp	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/libnr/nr-matrix.cpp	2009-04-05 18:21:30.000000000 -0400
@@ -11,6 +11,7 @@
  * This code is in public domain
  */
 
+#include <cstdio>
 #include <cstdlib>
 #include "nr-matrix.h"
 
diff -Naur inkscape-0.46-orig/src/streams-handles.cpp inkscape-0.46/src/streams-handles.cpp
--- inkscape-0.46-orig/src/streams-handles.cpp	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/streams-handles.cpp	2009-04-05 18:21:30.000000000 -0400
@@ -12,6 +12,7 @@
 #include "streams-handles.h"
 #include "uri.h"
 
+#include <cstdio>
 #include <iostream>
 
 namespace Inkscape {
diff -Naur inkscape-0.46-orig/src/streams-zlib.cpp inkscape-0.46/src/streams-zlib.cpp
--- inkscape-0.46-orig/src/streams-zlib.cpp	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/streams-zlib.cpp	2009-04-05 18:21:30.000000000 -0400
@@ -9,6 +9,7 @@
  * Released under GNU LGPL, read the file 'COPYING.LIB' for more information
  */
 
+#include <cstdio>
 #include <cstring>
 #include <string>
 #include <string.h>
diff -Naur inkscape-0.46-orig/src/xml/event.cpp inkscape-0.46/src/xml/event.cpp
--- inkscape-0.46-orig/src/xml/event.cpp	2009-04-05 16:47:42.000000000 -0400
+++ inkscape-0.46/src/xml/event.cpp	2009-04-05 18:21:30.000000000 -0400
@@ -14,6 +14,7 @@
  */
 
 #include <glib.h> // g_assert()
+#include <cstdio>
 
 #include "event.h"
 #include "event-fns.h"

