diff -ru src/dialogs/export.cpp src/dialogs/export.cpp
--- src/dialogs/export.cpp	2009-05-25 02:21:18.067302917 +0300
+++ src/dialogs/export.cpp	2009-05-25 02:21:17.804059091 +0300
@@ -20,6 +20,9 @@
 # include "config.h"
 #endif

+// This has to be included prior to anything that includes setjmp.h, it croaks otherwise
+#include <png.h>
+
 #include <gtk/gtk.h>
 #include <gtkmm/box.h>
 #include <gtkmm/buttonbox.h>
@@ -60,7 +63,6 @@
 #include "io/sys.h"

 #include "helper/png-write.h"
-#include <png.h>


 #define SP_EXPORT_MIN_SIZE 1.0
@@ -1041,9 +1043,7 @@
 static gchar *
 filename_add_extension (const gchar *filename, const gchar *extension)
 {
-  gchar *dot;
-
-  dot = strrchr (filename, '.');
+  const gchar *dot = strrchr (filename, '.');
   if ( !dot )
     return g_strconcat (filename, ".", extension, NULL);
   {
diff -ru src/document.cpp src/document.cpp
--- src/document.cpp	2009-05-25 02:21:18.052299751 +0300
+++ src/document.cpp	2009-05-25 02:21:17.797058619 +0300
@@ -746,7 +746,7 @@
         }

         if ( NULL != document_language ) {
-            gchar *pos = strchr(document_language, '_');
+	    const char *pos = strchr(document_language, '_');
             if ( NULL != pos ) {
                 return Glib::ustring(document_language, pos - document_language);
             }
diff -ru src/dom/io/domstream.cpp src/dom/io/domstream.cpp
--- src/dom/io/domstream.cpp	2009-05-25 02:21:18.011048623 +0300
+++ src/dom/io/domstream.cpp	2009-05-25 02:21:17.755301212 +0300
@@ -34,6 +34,7 @@
  *
  */

+#include <cstdio>
 #include <math.h>
 #include <stdarg.h>

diff -ru src/dom/io/gzipstream.cpp src/dom/io/gzipstream.cpp
--- src/dom/io/gzipstream.cpp	2009-05-25 02:21:18.012048750 +0300
+++ src/dom/io/gzipstream.cpp	2009-05-25 02:21:17.755301212 +0300
@@ -29,6 +29,8 @@

 #include "gzipstream.h"

+#include <cstdio>
+
 #include "dom/util/ziptool.h"


diff -ru src/dom/io/socket.cpp src/dom/io/socket.cpp
--- src/dom/io/socket.cpp	2009-05-25 02:21:18.011048623 +0300
+++ src/dom/io/socket.cpp	2009-05-25 02:21:17.754336705 +0300
@@ -35,6 +35,7 @@
 #include <sys/filio.h>   // needed on Solaris 8
 #endif

+#include <cstdio>
 #include "socket.h"
 #include "dom/util/thread.h"

diff -ru src/dom/io/uristream.cpp src/dom/io/uristream.cpp
--- src/dom/io/uristream.cpp	2009-05-25 02:21:18.012048750 +0300
+++ src/dom/io/uristream.cpp	2009-05-25 02:21:17.755301212 +0300
@@ -41,6 +41,7 @@


 #include "uristream.h"
+#include <cstdio>



diff -ru src/dom/lsimpl.cpp src/dom/lsimpl.cpp
--- src/dom/lsimpl.cpp	2009-05-25 02:21:18.014048586 +0300
+++ src/dom/lsimpl.cpp	2009-05-25 02:21:17.760050211 +0300
@@ -33,6 +33,7 @@
 #include "lsimpl.h"

 #include <stdarg.h>
+#include <cstdio>

 namespace org
 {
diff -ru src/dom/svg/svgparser.cpp src/dom/svg/svgparser.cpp
--- src/dom/svg/svgparser.cpp	2009-05-25 02:21:18.014048586 +0300
+++ src/dom/svg/svgparser.cpp	2009-05-25 02:21:17.759333709 +0300
@@ -32,6 +32,7 @@
 #include "dom/cssparser.h"
 #include "dom/charclass.h"

+#include <cstdio>
 #include <stdarg.h>

 #define SVG_NAMESPACE "http://www.w3.org/2000/svg"
diff -ru src/main.cpp src/main.cpp
--- src/main.cpp	2009-05-25 02:21:18.069308689 +0300
+++ src/main.cpp	2009-05-25 02:21:17.810107556 +0300
@@ -28,6 +28,9 @@
 #endif
 #include "path-prefix.h"

+// This has to be included prior to anything that includes setjmp.h, it croaks otherwise
+#include <png.h>
+
 #include <gtk/gtkmessagedialog.h>

 #ifdef HAVE_IEEEFP_H
@@ -111,7 +114,6 @@

 #include "main-cmdlineact.h"

-#include <png.h>
 #include <errno.h>

 enum {
diff -ru src/prefix.cpp src/prefix.cpp
--- src/prefix.cpp	2009-05-25 02:21:18.037083918 +0300
+++ src/prefix.cpp	2009-05-25 02:21:17.787058605 +0300
@@ -355,7 +355,8 @@
 char *
 br_extract_dir (const char *path)
 {
-	char *end, *result;
+	const char *end;
+	char *result;

 	br_return_val_if_fail (path != (char*)NULL, (char*)NULL);

@@ -390,7 +391,8 @@
 char *
 br_extract_prefix (const char *path)
 {
-	char *end, *tmp, *result;
+	const char *end;
+	char *tmp, *result;

 	br_return_val_if_fail (path != (char*)NULL, (char*)NULL);

diff -ru src/sp-image.cpp src/sp-image.cpp
--- src/sp-image.cpp	2009-05-25 02:21:18.023053851 +0300
+++ src/sp-image.cpp	2009-05-25 02:21:17.777050418 +0300
@@ -17,6 +17,9 @@
 # include "config.h"
 #endif

+// This has to be included prior to anything that includes setjmp.h, it croaks otherwise
+#include <png.h>
+
 #include <cstring>
 #include <string>
 #include <libnr/nr-matrix-fns.h>
@@ -41,7 +44,6 @@
 #include "libnr/nr-matrix-fns.h"

 #include "io/sys.h"
-#include <png.h>
 #if ENABLE_LCMS
 #include "color-profile-fns.h"
 #include "color-profile.h"

