diff -up network-manager-applet-0.9.2.0/m4/compiler_warnings.m4.depre NetworkManager-0.9.2.0/network-manager-applet-0.9.2.0/m4/compiler_warnings.m4
--- network-manager-applet-0.9.2.0/m4/compiler_warnings.m4.depre	2011-11-10 15:03:07.528830185 -0600
+++ network-manager-applet-0.9.2.0/m4/compiler_warnings.m4	2011-11-10 15:05:28.273070638 -0600
@@ -7,7 +7,7 @@ if test "$GCC" = "yes" -a "$set_more_war
 	AC_MSG_RESULT(yes)
 	CFLAGS="-Wall -Werror -std=gnu89 $CFLAGS"
 
-	for option in -Wshadow -Wmissing-declarations -Wmissing-prototypes \
+	for option in -Wshadow -Wmissing-declarations -Wmissing-prototypes -Wno-deprecated-declarations -Wno-error=deprecated-declarations \
 		      -Wdeclaration-after-statement -Wstrict-prototypes \
 		      -Wfloat-equal -Wno-unused-parameter -Wno-sign-compare \
 		      -fno-strict-aliasing; do
diff -up network-manager-applet-0.9.2.0/configure.ac.foo NetworkManager-0.9.2.0/network-manager-applet-0.9.2.0/configure.ac
--- network-manager-applet-0.9.2.0/configure.ac.foo	2011-11-10 15:10:45.268107655 -0600
+++ network-manager-applet-0.9.2.0/configure.ac	2011-11-10 15:10:51.902024720 -0600
@@ -179,8 +179,6 @@ dnl
 dnl Compiler flags
 dnl
 NM_COMPILER_WARNINGS
-# Use --enable-maintainer-mode to disabled deprecated symbols
-GNOME_MAINTAINER_MODE_DEFINES
 
 
 AC_CONFIG_FILES([

