--- HTTP-4000.4.1/HTTP.cabal	2001-09-09 03:46:40.000000000 +0200
+++ HTTP-4000.4.1.new/HTTP.cabal	2023-12-13 13:27:20.396911417 +0100
@@ -109,9 +109,9 @@
   -- note the test harness constraints should be kept in sync with these
   -- where dependencies are shared
   build-depends:
-      base          >= 4.6.0.0   && < 4.17
+      base          >= 4.6.0.0   && < 4.20
     , array         >= 0.3.0.2   && < 0.6
-    , bytestring    >= 0.9.1.5   && < 0.12
+    , bytestring    >= 0.9.1.5   && < 0.13
     , parsec        >= 2.0       && < 3.2
     , time          >= 1.1.2.3   && < 1.13
     , transformers  >= 0.2.0.0   && < 0.7
