diff -Naur HTTP-4000.4.1/HTTP.cabal HTTP-4000.4.1.new/HTTP.cabal
--- HTTP-4000.4.1/HTTP.cabal	2001-09-09 03:46:40.000000000 +0200
+++ HTTP-4000.4.1.new/HTTP.cabal	2022-10-14 10:05:56.747217614 +0200
@@ -109,7 +109,7 @@
   -- note the test harness constraints should be kept in sync with these
   -- where dependencies are shared
   build-depends:
-      base          >= 4.6.0.0   && < 4.17
+      base          >= 4.6.0.0   && < 4.18
     , array         >= 0.3.0.2   && < 0.6
     , bytestring    >= 0.9.1.5   && < 0.12
     , parsec        >= 2.0       && < 3.2
