diff -Naur hplip-3.23.12/prnt/cupsext/cupsext.c hplip-3.23.12.new/prnt/cupsext/cupsext.c
--- hplip-3.23.12/prnt/cupsext/cupsext.c	2023-11-27 16:06:40.000000000 +0100
+++ hplip-3.23.12.new/prnt/cupsext/cupsext.c	2024-05-17 12:43:32.104217474 +0200
@@ -101,6 +101,12 @@
 
 // static http_t * http = NULL;     /* HTTP object */
 
+void _releaseCupsInstance(void);
+int addCupsPrinter(char *name, char *device_uri, char *location, char *ppd_file, char *model, char *info);
+int setDefaultCupsPrinter(char *pr_name);
+int delCupsPrinter(char *pr_name);
+int controlCupsPrinter(char *pr_name, int op);
+
 PyObject *releaseCupsInstance(PyObject *self, PyObject *args)
 {
     _releaseCupsInstance();
diff -Naur hplip-3.23.12/protocol/hp_ipp.c hplip-3.23.12.new/protocol/hp_ipp.c
--- hplip-3.23.12/protocol/hp_ipp.c	2023-11-27 16:06:40.000000000 +0100
+++ hplip-3.23.12.new/protocol/hp_ipp.c	2024-05-17 12:40:47.791928853 +0200
@@ -21,6 +21,7 @@
 #include <cups/cups.h>
 #include <cups/language.h>
 #include <cups/ppd.h>
+#include <stdio.h>
 #include <syslog.h>
 #include <stdarg.h>
 #include <sys/types.h>
@@ -44,7 +45,7 @@
 		dst[size - 1] = '\0'; \
     } while (0)
 
-http_t *acquireCupsInstance()
+http_t *acquireCupsInstance(void)
 {
     if (http == NULL)
     {
@@ -54,7 +55,7 @@
     return http;
 }
 
-void _releaseCupsInstance()
+void _releaseCupsInstance(void)
 {
     if (http)
     {
