diff -Naur hplip-3.23.12/Makefile.am hplip-3.23.12.new/Makefile.am
--- hplip-3.23.12/Makefile.am	2023-11-27 16:06:40.000000000 +0100
+++ hplip-3.23.12.new/Makefile.am	2024-05-17 12:17:03.164743549 +0200
@@ -306,13 +306,6 @@
 pcarddir = $(hplipdir)/pcard
 dist_pcard_DATA = pcard/__init__.py pcard/photocard.py
 
-# pcardext
-pcardextdir = $(pyexecdir)
-pcardext_LTLIBRARIES = pcardext.la
-pcardext_la_LDFLAGS = -module -avoid-version
-pcardext_la_SOURCES = pcard/pcardext/pcardext.c pcard/fat.c
-pcardext_la_CFLAGS = -I$(PYTHONINCLUDEDIR)
-
 # prnt
 prntdir = $(hplipdir)/prnt
 dist_prnt_DATA = prnt/cups.py prnt/__init__.py prnt/ldl.py prnt/pcl.py prnt/colorcal.py
diff -Naur hplip-3.23.12/protocol/hp_ipp.c hplip-3.23.12.new/protocol/hp_ipp.c
--- hplip-3.23.12/protocol/hp_ipp.c	2023-11-27 16:06:40.000000000 +0100
+++ hplip-3.23.12.new/protocol/hp_ipp.c	2024-05-17 12:17:03.164743549 +0200
@@ -18,6 +18,7 @@
 
 \******************************************************************************/
 
+#define _GNU_SOURCE
 #include <cups/cups.h>
 #include <cups/language.h>
 #include <cups/ppd.h>
diff -Naur hplip-3.23.12/scan/sane/hpaio.c hplip-3.23.12.new/scan/sane/hpaio.c
--- hplip-3.23.12/scan/sane/hpaio.c	2023-11-27 16:06:41.000000000 +0100
+++ hplip-3.23.12.new/scan/sane/hpaio.c	2024-05-17 12:17:03.164743549 +0200
@@ -50,6 +50,7 @@
 #include "sclpml.h"
 #include "escl.h"
 #include "io.h"
+#include "orblite.h"
 #include "orblitei.h"
 
 
diff -Naur hplip-3.23.12/scan/sane/orblite.c hplip-3.23.12.new/scan/sane/orblite.c
--- hplip-3.23.12/scan/sane/orblite.c	2023-11-27 16:06:41.000000000 +0100
+++ hplip-3.23.12.new/scan/sane/orblite.c	2024-05-17 12:17:03.164743549 +0200
@@ -20,6 +20,7 @@
 #include "orblitei.h"
 #include "orblite.h"//Added New
 #include <math.h>
+#include <dlfcn.h>
 #include "utils.h"
 #include "io.h"
 
@@ -193,27 +194,6 @@
 
 }
 
-static int bb_unload(SANE_THandle ps)
-{
-   _DBG("Calling orblite bb_unload: \n");
-   if (ps->bb_handle)
-   {
-      dlclose(ps->bb_handle);
-      ps->bb_handle = NULL;
-   }
-   if (ps->hpmud_handle)
-   {
-      dlclose(ps->hpmud_handle);
-      ps->hpmud_handle = NULL;
-   }
-//   if (ps->math_handle)
-//   { 
-//      dlclose(ps->math_handle);
-//      ps->math_handle = NULL;
-//   }
-   return 0;
-}
-
 SANE_Status 
 orblite_init (SANE_Int * version_code, SANE_Auth_Callback authorize)
 {
diff -Naur hplip-3.23.12/scan/sane/sclpml.c hplip-3.23.12.new/scan/sane/sclpml.c
--- hplip-3.23.12/scan/sane/sclpml.c	2023-11-27 16:06:41.000000000 +0100
+++ hplip-3.23.12.new/scan/sane/sclpml.c	2024-05-17 12:17:03.163743541 +0200
@@ -46,6 +46,9 @@
 #define DEBUG_DECLARE_ONLY
 #include "sanei_debug.h"
 
+/* This file is not built with _GNU_SOURCE. */
+char *strcasestr(const char *, const char *);
+
 //# define  SCLPML_DEBUG
 # ifdef SCLPML_DEBUG
    # define _DBG(args...) syslog(LOG_INFO, __FILE__ " " STRINGIZE(__LINE__) ": " args)
