diff -Naur hplip-3.23.12/scan/sane/bb_ledm.c hplip-3.23.12.new/scan/sane/bb_ledm.c
--- hplip-3.23.12/scan/sane/bb_ledm.c	2023-11-27 16:06:40.000000000 +0100
+++ hplip-3.23.12.new/scan/sane/bb_ledm.c	2024-05-17 12:58:07.577143137 +0200
@@ -16,6 +16,7 @@
 # include <string.h>
 # include <fcntl.h>
 # include <math.h>
+# include <unistd.h>
 # include "sane.h"
 # include "saneopts.h"
 # include "hpmud.h"
diff -Naur hplip-3.23.12/scan/sane/http.c hplip-3.23.12.new/scan/sane/http.c
--- hplip-3.23.12/scan/sane/http.c	2023-11-27 16:06:41.000000000 +0100
+++ hplip-3.23.12.new/scan/sane/http.c	2024-05-17 12:57:38.044908314 +0200
@@ -36,6 +36,7 @@
 #include <string.h>
 #include <syslog.h>
 #include <ctype.h>
+#include <unistd.h>
 #include "hpmud.h"
 #include "http.h"
 
diff -Naur hplip-3.23.12/scan/sane/sclpml.c hplip-3.23.12.new/scan/sane/sclpml.c
--- hplip-3.23.12/scan/sane/sclpml.c	2023-11-27 16:06:41.000000000 +0100
+++ hplip-3.23.12.new/scan/sane/sclpml.c	2024-05-17 12:58:46.690454100 +0200
@@ -30,6 +30,7 @@
 #include <unistd.h>
 #include <fcntl.h>
 #include <string.h>
+#include <sys/time.h>
 #include "sane.h"
 #include "saneopts.h"
 #include "common.h"
