diff -Naur hplip-3.17.11/prnt/hpcups/genPCLm.cpp hplip-3.17.11.new/prnt/hpcups/genPCLm.cpp
--- hplip-3.17.11/prnt/hpcups/genPCLm.cpp	2017-11-23 13:10:22.000000000 +0100
+++ hplip-3.17.11.new/prnt/hpcups/genPCLm.cpp	2017-11-25 22:39:17.397873474 +0100
@@ -350,7 +350,7 @@
 	static int fileCntr=0;
 
 	if(1)
-		return true;
+		return (boolean)true;
 	snprintf(outFileName,sizeof(outFileName),"%s/hp_%s_cups_outfile_%04d",CUPS_TMP_DIR, user_name, fileCntr);
  
 	fileCntr++;
@@ -364,7 +364,7 @@
 	}
 	fwrite(ptr,numBytes,1,outputFile);
 	fclose(outputFile); 
-	return(true);
+	return((boolean)true);
 }
 
 void PCLmGenerator::initOutBuff(char *buff, sint32 size)
@@ -1110,7 +1110,7 @@
 		}
 	}
 	//origTail++;
-	return(true);
+	return((boolean)true);
 }
 
 char* PCLmGenerator::getInputBinString(jobInputBin bin)
