From bea6d0e4915c091d6ffa00ca4a374c6b279f3acf Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Mon, 5 Sep 2016 19:33:50 +0200
Subject: [PATCH 1/4] makepkg.conf * kill ARM

---
 etc/makepkg.conf.in | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/etc/makepkg.conf.in b/etc/makepkg.conf.in
index cbf4051..226621c 100644
--- a/etc/makepkg.conf.in
+++ b/etc/makepkg.conf.in
@@ -23,12 +23,6 @@ case $(uname -m) in
 		CFLAGS="-march=x86-64 -mtune=generic -O2 -pipe"
 		CXXFLAGS="-march=x86-64 -mtune=generic -O2 -pipe"
 		;;
-	arm*)
-		CARCH="arm"
-		CHOST="arm-frugalware-linux"
-		CFLAGS="-march=armv5te -O2 -pipe"
-		CXXFLAGS="-march=armv5te -O2 -pipe"
-		;;
 esac
 export CARCH CHOST CFLAGS CXXFLAGS
 
-- 
2.9.3

