From 41f703c97a0dac99f5798b7e4fd726efb3e49c72 Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Tue, 27 Oct 2015 21:09:50 +0100
Subject: [PATCH] makepkg: fix optimization check

---
 scripts/makepkg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index 73bea6e..cb3a4de 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -1199,7 +1199,7 @@ if [ "$CHROOT" = "1" -a "$INCHROOT" != "1" ]; then
 		chroot_clean
 		# check optimization and write it to $BUILDSCRIPT
 		msg "Checking for optimization..."
-		if grep -q -- "\($CFLAGS\|$CXXFLAGS\)" "${BUILDLOG}" ; then
+		if cat "${BUILDLOG}" | grep -v HOST_ | grep -q -- "\($CFLAGS\|$CXXFLAGS\)" ; then
 			if ! grep -q "# optimization" "${BUILDSCRIPT}" ; then
 				$ECHO -e "\n# optimization OK" >> "${BUILDSCRIPT}"
 			fi
-- 
2.6.2

