From bbf3d7ca852a1b3685b993efc16cca236f724694 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Mon, 28 Sep 2015 12:22:44 +0200
Subject: [PATCH] * fix problems with umounting chroots when dirmngr was 
 running

---
 scripts/makepkg | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/scripts/makepkg b/scripts/makepkg
index b0591b8..011b78e 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -832,6 +832,8 @@ makepkg_init() {
 }
 
 chroot_umount() {
+    msg "Kill dirmngr if it is running"
+    gpgconf --kill dirmngr
     msg "Attempting to umount chroot directories..."
     umount $CHROOTDIR/proc >/dev/null
     umount $CHROOTDIR/sys >/dev/null
-- 
2.5.0

