From 792c9b9193a4c96980e05a655f7fbe01bec9d7f1 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Thu, 25 Jan 2024 11:23:19 +0100
Subject: [PATCH] * check for TERM before trying to use tput

---
 scripts/makepkg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index 212e70fa..47f86d11 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -1066,7 +1066,7 @@ if [[ -n ${_MAKEFLAGS} ]]; then
        export MAKEFLAGS="${_MAKEFLAGS}"
 fi
 
-if [ "$USE_COLOR" = "Y" -o "$USE_COLOR" = "y" ]; then
+if [ -n "$TERM" ] - a [ "$USE_COLOR" = "Y" -o "$USE_COLOR" = "y" ]; then
     if [ "$(/usr/bin/tput colors)" -lt 3 ]; then
         USE_COLOR="n"
         warning "Disabling color as the terminal does not support it."
-- 
2.43.0

