From 719695510d2174e69815c0a586a83b55b81da9aa Mon Sep 17 00:00:00 2001
From: James Buren <ryuo@frugalware.org>
Date: Fri, 22 Jun 2012 05:14:50 -0500
Subject: [PATCH] makepkg.conf.in: march is x86-64 not x86_64

---
 etc/makepkg.conf.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/etc/makepkg.conf.in b/etc/makepkg.conf.in
index 109ede7..4ea8e79 100644
--- a/etc/makepkg.conf.in
+++ b/etc/makepkg.conf.in
@@ -20,8 +20,8 @@ case $(uname -m) in
 	x86_64)
 		CARCH="x86_64"
 		CHOST="x86_64-frugalware-linux"
-		CFLAGS="-march=x86_64 -mtune=generic -O2 -pipe"
-		CXXFLAGS="-march=x86_64 -mtune=generic -O2 -pipe"
+		CFLAGS="-march=x86-64 -mtune=generic -O2 -pipe"
+		CXXFLAGS="-march=x86-64 -mtune=generic -O2 -pipe"
 		;;
 	arm*)
 		CARCH="arm"
-- 
1.7.11

