From 9c55eba19b8722f4a10ebb7c14ba3375f1d9e44f Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Mon, 28 Sep 2015 15:26:12 +0200
Subject: [PATCH] * killall gpg components _in_ chroot

---
 scripts/makepkg | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index 011b78e..dc4e618 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -832,8 +832,6 @@ makepkg_init() {
 }
 
 chroot_umount() {
-    msg "Kill dirmngr if it is running"
-    gpgconf --kill dirmngr
     msg "Attempting to umount chroot directories..."
     umount $CHROOTDIR/proc >/dev/null
     umount $CHROOTDIR/sys >/dev/null
@@ -1906,6 +1904,13 @@ if [ "$GENSHA1" = "0" ]; then
 					msg "Importing $pkgname.key"
 					gpg --import $startdir/$pkgname.key
 				fi
+				msg "Killing all gpg components"
+				gpgconf --kill gpg
+				gpgconf --kill gpg-agent
+				gpgconf --kill scdaemon
+				gpgconf --kill gpgsm
+				gpgconf --kill dirmngr
+				gpgconf --kill pinentry
 				msg "Validating source files with gpg"
 				errors=0
 				idx=0
@@ -1936,6 +1941,13 @@ if [ "$GENSHA1" = "0" ]; then
 					fi
 					idx=$(($idx+1))
 				done
+				msg "Killing all gpg components"
+				gpgconf --kill gpg
+				gpgconf --kill gpg-agent
+				gpgconf --kill scdaemon
+				gpgconf --kill gpgsm
+				gpgconf --kill dirmngr
+				gpgconf --kill pinentry
 				if [ $errors -gt 0 ]; then
 					error "One or more files did not pass the validity check!"
 					return 1
-- 
2.5.0

