From 7c90cc591e60dea0b9957771119c636a9d480157 Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Wed, 6 Jan 2016 20:51:50 +0100
Subject: [PATCH] makepkg - frugalware package is allowed to add all kind
 folders

---
 scripts/makepkg | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index ded62a7..f11b328 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -1728,11 +1728,13 @@ do
 	## /lib64 and /usr/lib/64 .. this is a symlink in our layout
 	## so don't install there
 	if [ -d ${dir}/lib64 ] || [ -d ${dir}/usr/lib64 ]; then
-		error "Your package has broken /lib or /usr/lib PATH"
-		error "Files are installed in 'lib64'"
-		error "but should be /lib or /usr/lib!"
-		error "Bailing out, please fix your package!"
-		exit 1
+		if [[ ! $pkgname =~ "frugalware" ]]; then
+			error "Your package has broken /lib or /usr/lib PATH"
+			error "Files are installed in 'lib64'"
+			error "but should be /lib or /usr/lib!"
+			error "Bailing out, please fix your package!"
+			exit 1
+		fi
 	fi
 	## don't insstall files in /usr/<file>
 	files=$(find ${dir}/usr -maxdepth 1 -type f)
-- 
2.6.4

