From 68a20e4351cc76452d3e31e12fee8b4a6a0e7ee6 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Sun, 6 Dec 2015 20:01:23 +0100
Subject: [PATCH] * try to fix gpg verify issues

---
 scripts/makepkg | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index cb3a4de..90e797b 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -1452,6 +1452,7 @@ if [ "$GENSHA1" = "0" ]; then
 				if [ ! -d ~/.gnupg ]; then
 					msg "Configuring gpg..."
 					gpg --list-keys
+					$ECHO "keyserver hkp://pool.sks-keyservers.net" >>~/.gnupg/gpg.conf
 					$ECHO "keyserver-options auto-key-retrieve" >>~/.gnupg/gpg.conf
 				fi
 				if [ -e $startdir/$pkgname.key ]; then
@@ -1459,7 +1460,7 @@ if [ "$GENSHA1" = "0" ]; then
 					gpg --import $startdir/$pkgname.key
 				fi
 				msg "Killing dirmngr"
-				killall dirmngr
+				killall -u fst dirmngr
 				msg "Validating source files with gpg"
 				errors=0
 				idx=0
@@ -1491,7 +1492,7 @@ if [ "$GENSHA1" = "0" ]; then
 					idx=$(($idx+1))
 				done
 				msg "Killing dirmngr"
-				killall dirmngr
+				killall -u fst dirmngr
 				if [ $errors -gt 0 ]; then
 					error "One or more files did not pass the validity check!"
 					exit 1
-- 
2.6.3

