From f4e2b76f5136fcfc7a23d5b1571a3661107e696c Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Mon, 28 Sep 2015 15:47:35 +0200
Subject: [PATCH] * only kill dirmngr

---
 scripts/makepkg | 18 ++++--------------
 1 file changed, 4 insertions(+), 14 deletions(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index dc4e618..fb8bb84 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -1904,13 +1904,8 @@ if [ "$GENSHA1" = "0" ]; then
 					msg "Importing $pkgname.key"
 					gpg --import $startdir/$pkgname.key
 				fi
-				msg "Killing all gpg components"
-				gpgconf --kill gpg
-				gpgconf --kill gpg-agent
-				gpgconf --kill scdaemon
-				gpgconf --kill gpgsm
-				gpgconf --kill dirmngr
-				gpgconf --kill pinentry
+				msg "Killing dirmngr"
+				killall dirmngr
 				msg "Validating source files with gpg"
 				errors=0
 				idx=0
@@ -1941,13 +1936,8 @@ if [ "$GENSHA1" = "0" ]; then
 					fi
 					idx=$(($idx+1))
 				done
-				msg "Killing all gpg components"
-				gpgconf --kill gpg
-				gpgconf --kill gpg-agent
-				gpgconf --kill scdaemon
-				gpgconf --kill gpgsm
-				gpgconf --kill dirmngr
-				gpgconf --kill pinentry
+				msg "Killing dirmngr"
+				killall dirmngr
 				if [ $errors -gt 0 ]; then
 					error "One or more files did not pass the validity check!"
 					return 1
-- 
2.5.0

