From 4980f34c8554594c8935c3a3370797d34f31ea4b Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Wed, 1 Feb 2017 12:52:03 +0100
Subject: [PATCH] fix mercurial cmd

---
 scripts/makepkg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index e2487658..41468d7e 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -1284,7 +1284,7 @@ if [ "$NOUP2DATE" = "0" ]; then
 					cvs)	cmd="cvs";;
 					subversion)	cmd="subversion";;
 					git) 	cmd="git";;
-					mercurial)	cmd="mercurial";;
+					mercurial)	cmd="hg";;
 					bzr)	cmd="bzr";;
 				esac
 			else
-- 
2.11.0

