From 8f776f4e178d219b620a90f5bbf7eccfef71f247 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Fri, 29 Dec 2023 16:12:43 +0100
Subject: [PATCH] * systemd now installs everything into /usr

---
 scripts/makepkg | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index 7ec35ed4..212e70fa 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -2118,14 +2118,6 @@ do
 			exit 1
 		fi
 	done
-	## wrong unit path for systemd
-	if [ -d ${dir}/usr/lib/systemd/system ]; then
-		error "Your package has broken PATH for systemd unit files"
-		error "Files are installed in /usr/lib/systemd/system"
-		error "but should be /lib/systemd/system!"
-		error "Bailing out, please fix your package!"
-		exit 1
-	fi
 	## /lib64 and /usr/lib/64 .. this is a symlink in our layout
 	## so don't install there
 	if [ -d ${dir}/lib64 ] || [ -d ${dir}/usr/lib64 ]; then
-- 
2.43.0

