From 15239ddaabded16686facc39dd57e675e1da1888 Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Sat, 26 Jun 2010 00:08:40 +0200
Subject: [PATCH] fix percent calculation , should close #4252

---
 lib/libpacman/add.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/lib/libpacman/add.c b/lib/libpacman/add.c
index 34186f2..29167c1 100644
--- a/lib/libpacman/add.c
+++ b/lib/libpacman/add.c
@@ -444,7 +444,7 @@ int _pacman_add_commit(pmtrans_t *trans, pmdb_t *db)
 				STRNCPY(pathname, archive_entry_pathname (entry), PATH_MAX);
 
 				if (info->size != 0)
-		    			percent = archive_position_uncompressed(archive) / info->size;
+		    			percent = (double)archive_position_uncompressed(archive) / info->size;
 				if (needdisp == 0) {
 					PROGRESS(trans, cb_state, what, (int)(percent * 100), howmany, (howmany - remain +1));
 				}
-- 
1.7.1

