From fe97b73d1bb8c30d9f5d636c9ac6aa301d2509b0 Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Wed, 14 Sep 2016 16:50:33 +0200
Subject: [PATCH] workaround LDFLAGS mess here

---
 scripts/makepkg | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/scripts/makepkg b/scripts/makepkg
index 5bf8aa1..70edbb9 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -1357,6 +1357,25 @@ if grep -i -q "^# Compiling Time: [~0-9\.]\+ SBU$" $BUILDSCRIPT && \
 	fi
 fi
 
+### we use at least -Wl,--hash-style=both and conditional
+## -Wl,--as-needed. Gcc6 WIP will add -Wl,-O1.. HOWEVER some build server(s)
+## seems to use veryyyyyy old makepkg.conf even witout --hash-style=both..
+## check here and workaround this when needed..
+
+if ! $ECHO $LDFLAGS | grep -q "\-Wl,--hash-style=both" ; then
+        warning "Broken LDFLAGS found, -Wl,--hash-style=both is missing from makepkg.conf!!"
+        warning "Enabling workaround, please fix your setup!!"
+        LDFLAGS+=" -Wl,--hash-style=both"
+        export LDFLAGS
+fi
+
+if ! $ECHO $LDFLAGS | grep -q "\-Wl,-O1" ; then
+        warning "Broken LDFLAGS found, -Wl,-O1 is missing from makepkg.conf!!"
+        warning "Enabling workaround, please fix your setup!!"
+        LDFLAGS+=" -Wl,-O1"
+        export LDFLAGS
+fi
+
 
 if [ "`check_option ASNEEDED`" ]; then
         error "You are using deprecated option 'asneeded'."
-- 
2.9.3

