From 48a8aa6c57f98046d7000e6289f35735f3cbbc9f Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Mon, 28 Sep 2015 12:30:53 +0200
Subject: [PATCH] * use multithreaded compression and higher compression level
 for xz

---
 etc/makepkg.conf.in | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/etc/makepkg.conf.in b/etc/makepkg.conf.in
index 2650e4c..cbf4051 100644
--- a/etc/makepkg.conf.in
+++ b/etc/makepkg.conf.in
@@ -96,3 +96,6 @@ SBU="1"
 
 # Uncomment the following line if you want build time to be more "human readable."
 #HRBT="1"
+
+# Options for xz used in compression
+export XZ_OPT="--threads=0 -9"
-- 
2.5.0

