From 83bdc1464844bdac50b5d02acc822e477eb3ff5a Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Mon, 5 Sep 2016 19:34:50 +0200
Subject: [PATCH 2/4] makepkg.conf LDFLAGS * added -Wl,-O1 to LDFLAGS

---
 etc/makepkg.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/etc/makepkg.conf.in b/etc/makepkg.conf.in
index 226621c..da3c5e9 100644
--- a/etc/makepkg.conf.in
+++ b/etc/makepkg.conf.in
@@ -26,7 +26,7 @@ case $(uname -m) in
 esac
 export CARCH CHOST CFLAGS CXXFLAGS
 
-export LDFLAGS="-Wl,--hash-style=both"
+export LDFLAGS="-Wl,--hash-style=both -Wl,-O1"
 
 # SMP Systems
 #export MAKEFLAGS="-j2"
-- 
2.9.3

