From 3b0826c7cb39b9fb1ddf665693aa3fdce0574760 Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Fri, 9 Sep 2016 11:14:38 +0200
Subject: [PATCH 2/4] remove duplicated check

---
 scripts/makepkg | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index 1dd241b..5bf8aa1 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -904,10 +904,7 @@ else
 	error "Please make sure that you edit /etc/makepkg.conf and set the PACKAGER= variable"
 	exit 1
 fi
-if [ -z "$PKG_EXT" ]; then
-	error "Please make sure that you edit /etc/makepkg.conf and set the PKG_EXT= variable"
-	exit 1
-fi
+
 if [ ! -f $BUILDSCRIPT ]; then
 	error "$BUILDSCRIPT does not exist."
 	exit 1
-- 
2.10.0

