From b543652625e2648e9957bd059fd50c9eefed75c3 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Thu, 25 Jan 2024 13:33:02 +0100
Subject: [PATCH 2/2] * fix test

---
 scripts/makepkg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index 47f86d11..dc7f05d1 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -1066,7 +1066,7 @@ if [[ -n ${_MAKEFLAGS} ]]; then
        export MAKEFLAGS="${_MAKEFLAGS}"
 fi
 
-if [ -n "$TERM" ] - a [ "$USE_COLOR" = "Y" -o "$USE_COLOR" = "y" ]; then
+if [ -n "$TERM" ] && { [ "$USE_COLOR" = "Y" ] || [ "$USE_COLOR" = "y" ]; }; then
     if [ "$(/usr/bin/tput colors)" -lt 3 ]; then
         USE_COLOR="n"
         warning "Disabling color as the terminal does not support it."
-- 
2.43.0

