From c36872b0562b01d1e18e569c0c4e10b5cc1b1cab Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Thu, 22 Feb 2024 13:08:28 +0100
Subject: [PATCH 3/3] * allow checking files with spaces in symbolic links
 check

---
 scripts/makepkg | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index 366c430e..e998ed44 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -2279,7 +2279,8 @@ done
 # check symbolic links
 msg "Checking symbolic links..."
 for d in $startdir/{pkg,pkg.*} ; do
-	for l in `find $d -type l 2> /dev/null` ; do
+	find $d -type l -print0 2> /dev/null | while read -d $'\0' l; do
+		IFS=''
 		if [ ! -e "$l" -a ! -e "$d/`ls -l $l | awk '{print $NF}'`" ]; then
 			if [ "$d" = "$startdir/pkg" ]; then
 				msg2 "Maybe broken link ${l#$d} in pkg $pkgname found."
-- 
2.43.0

