From 029213b14ad2ab4533657d23eaeea4f25b232a4d Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Mon, 26 Aug 2024 16:35:23 +0200
Subject: [PATCH] * check for sqv

---
 scripts/makepkg | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index da958717..2f965f86 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -1854,7 +1854,7 @@ if [ "$GENSHA1" = "0" ]; then
 		fi
 		# gpg validation
 		if [ ${#signatures[@]} -eq ${#source[@]} ]; then
-			if [ `type -p gpg` ]; then
+			if [ `type -p sqv` ]; then
 				errors=0
 				idx=0
 				for netfile in "${source[@]}"; do
@@ -1862,7 +1862,7 @@ if [ "$GENSHA1" = "0" ]; then
 					sig=`strip_url "${signatures[$idx]}"`
 					$ECHO -n "    $file ... " >&2
 					if [ ! -z "$sig" ]; then
-						sqv --keyring $startdir/$pkgname.key $sig $file >/dev/null 2>&1
+						sqv -v --keyring $startdir/$pkgname.key $sig $file
 						if [ $? -ne 0 ]; then
 							$ECHO "FAILED" >&2
 							errors=1
@@ -1879,7 +1879,7 @@ if [ "$GENSHA1" = "0" ]; then
 					exit 1
 				fi
 			else
-				warning "The gpg program is missing.  Cannot verify source files!"
+				warning "The sqv program is missing.  Cannot verify source files!"
 				sleep 1
 			fi
 		fi
-- 
2.46.0

