From eba116454f7bd647ff3b863e7ba71a59ca22498d Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Wed, 5 Jun 2024 13:05:40 +0200
Subject: [PATCH 2/2] * default link with gold

---
 scripts/makepkg | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index b7d5e3d1..ff48abb2 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -257,12 +257,21 @@ setflags() {
 	fi
 
 	## LD's
-	# Since we try with gold default lets have
+	_LD_PLUGIN=" -fuse-ld=mold"
+	# Since we try with mold default lets have
 	# options to switch them with options=()
-	if [ "`check_option LDGOLD`" -a "`check_option LDLLD`" ]; then
+	if [ "`check_option LDBFD`" -a "`check_option LDLLD`" ]; then
 		error "Having ldbfd and ldlld enabled is not possible."
 		Fdie
 	fi
+        if [ "`check_option LDBFD`" -a "`check_option LDGOLD`" ]; then
+		error "Having ldbfd and ldgold enabled is not possible."
+		Fdie
+	fi
+        if [ "`check_option LDLLD`" -a "`check_option LDGOLD`" ]; then
+                error "Having ldlld and ldgold enabled is not possible."
+                Fdie
+        fi
 
 	if [ "`check_option LDBFD`" ]; then
 		_LD_PLUGIN=" -fuse-ld=bfd"
-- 
2.45.2

