From ea886ea1bd32e917ad46c4aadc8570a16c35bbd8 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Tue, 13 Aug 2024 10:28:31 +0200
Subject: [PATCH] * enable libeatmydata

---
 scripts/makepkg | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/scripts/makepkg b/scripts/makepkg
index ff48abb2..6bec0222 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -59,6 +59,7 @@ RMDEPS=0
 CHROOT=1
 NOUP2DATE=0
 NODOCS=0
+SYNCFS=0
 TREE=""
 CACHEURL=""
 
@@ -365,6 +366,17 @@ setflags() {
 		_LDFLAGS+=" -Wl,-z,defs"
 	fi
 
+	if [ "`check_option SYNCFS`" ]; then
+		warning "Syncing enabled"
+	else
+		if [ -z "$LD_PRELOAD" ]; then
+			  LD_PRELOAD=/usr/lib/libeatmydata/libeatmydata.so
+		  else
+			    LD_PRELOAD="$LD_PRELOAD":/usr/lib/libeatmydata/libeatmydata.so
+		fi
+		export LD_PRELOAD
+	fi
+
 	## put all together.
 	## FB w/out build()
 	[[ -n ${_FB_CFLAGS} ]] && _FLAGS+=" ${_FB_CFLAGS}"
-- 
2.46.0

