From 36dd0f395033b18153f8d1249b9a10e6ac7c9c67 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Thu, 15 Aug 2024 11:46:19 +0200
Subject: [PATCH] * fix fuse-ld with clang

---
 scripts/makepkg | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index ae4175ad..37c4c38f 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -387,8 +387,14 @@ setflags() {
 	# just sanitize all FLAGS here
 
 	_LDFLAGS="$(/usr/bin/echo ${_LDFLAGS[@]} | tr [:space:] '\n' | awk '!a[$0]++' | tr '\n' ' ')"
-	_CFLAGS="${_MFLAGS} ${_OFLAG} ${_FLAGS} ${_LD_PLUGIN} -pipe"
-	_CXXFLAGS="${_MFLAGS} ${_OFLAG} ${_FLAGS} ${_LD_PLUGIN} -pipe"
+	# clang complains when using fuse-ld when compiling
+	if [ "`check_option CLANG`" ]; then
+		_CFLAGS="${_MFLAGS} ${_OFLAG} ${_FLAGS} -pipe"
+		_CXXFLAGS="${_MFLAGS} ${_OFLAG} ${_FLAGS} -pipe"
+	else
+                _CFLAGS="${_MFLAGS} ${_OFLAG} ${_FLAGS} ${_LD_PLUGIN} -pipe"
+                _CXXFLAGS="${_MFLAGS} ${_OFLAG} ${_FLAGS} ${_LD_PLUGIN} -pipe"
+	fi
 	LDFLAGS_=" ${_LDFLAGS} ${_CFLAGS} ${_CXXFLAGS}"
 
 	CFLAGS="$(/usr/bin/echo ${_CFLAGS[@]} | tr [:space:] '\n' | awk '!a[$0]++' | tr '\n' ' ')"
-- 
2.46.0

