From c997da65bc0a35a33180dbf8a513e686927655d0 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Tue, 13 Aug 2024 10:43:39 +0200
Subject: [PATCH 2/2] * fix path

---
 scripts/makepkg | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index 6bec0222..ae4175ad 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -370,9 +370,9 @@ setflags() {
 		warning "Syncing enabled"
 	else
 		if [ -z "$LD_PRELOAD" ]; then
-			  LD_PRELOAD=/usr/lib/libeatmydata/libeatmydata.so
+			  LD_PRELOAD=/usr/lib/libeatmydata.so
 		  else
-			    LD_PRELOAD="$LD_PRELOAD":/usr/lib/libeatmydata/libeatmydata.so
+			    LD_PRELOAD="$LD_PRELOAD":/usr/lib/libeatmydata.so
 		fi
 		export LD_PRELOAD
 	fi
-- 
2.46.0

