diff -Naur pacman-g2-3.8.1//lib/libpacman/package.c pacman-g2-3.8.1new//lib/libpacman/package.c
--- pacman-g2-3.8.1//lib/libpacman/package.c	2011-01-10 22:44:52.000000000 +0100
+++ pacman-g2-3.8.1new//lib/libpacman/package.c	2011-02-17 23:00:08.355426001 +0100
@@ -352,7 +352,7 @@
 				}
 
 				uname (&name);
-				if(strcmp(name.machine, info->arch)) {
+				if(strncmp(name.machine, info->arch, strlen(info->arch))) {
 					_pacman_log(PM_LOG_ERROR, _("wrong package architecture in %s"), pkgfile);
 					pm_errno = PM_ERR_WRONG_ARCH;
 					unlink(descfile);
