diff -Naur pacman-g2-3.7.0/lib/libftp/Makefile.am pacman-g2-3.7.0-glibc28/lib/libftp/Makefile.am
--- pacman-g2-3.7.0/lib/libftp/Makefile.am	2008-06-01 01:26:49.000000000 +0200
+++ pacman-g2-3.7.0-glibc28/lib/libftp/Makefile.am	2008-07-17 18:05:16.000000000 +0200
@@ -1,5 +1,5 @@
 AUTOMAKE_OPTIONS = gnu
-DEFINES = -DFTPLIB_DEFMODE=FTPLIB_PORT -D_REENTRANT
+DEFINES = -DFTPLIB_DEFMODE=FTPLIB_PORT -D_REENTRANT -D_GNU_SOURCE
 AM_CFLAGS = $(DEFINES)
 
 #SOMAJOR = 3
