diff -ur a/lib/libftp/ftplib.c b/lib/libftp/ftplib.c
--- a/lib/libftp/ftplib.c	2013-03-03 05:55:16.275090684 -0600
+++ b/lib/libftp/ftplib.c	2013-03-03 05:57:11.760086433 -0600
@@ -956,6 +956,7 @@
 GLOBALDEF int FtpClose(netbuf *nData)
 {
 	netbuf *ctrl;
+	int errcode;
 	switch (nData->dir)
 	{
 		case FTPLIB_WRITE:
@@ -968,8 +969,9 @@
 			shutdown(nData->handle,2);
 			net_close(nData->handle);
 			ctrl = nData->ctrl;
+			errcode = nData->errcode;
 			free(nData);
-			if (ctrl && (nData->errcode != ETIMEDOUT))
+			if (ctrl && (errcode != ETIMEDOUT))
 			{
 				ctrl->data = NULL;
 				return(readresp('2', ctrl));
