diff -Naur pacman-g2-3.9.4/scripts/makepkg pacman-g2-3.9.4-p/scripts/makepkg
--- pacman-g2-3.9.4/scripts/makepkg	2020-02-16 02:43:58.378276474 +0100
+++ pacman-g2-3.9.4-p/scripts/makepkg	2020-02-16 02:43:28.422059579 +0100
@@ -318,7 +318,6 @@
 
 	if [ "`check_option NOLTO`" ]; then
 		warning "Disabling C/CXX FLAGS: -flto"
-		_FLAGS+=" -fno-lto"
 	else
 		if [ ! "`check_option CLANG`" ]; then
 			_lto_cpus="$(getconf _NPROCESSORS_ONLN)"
