--- pacman-g2-3.9.0/scripts/makepkg	2013-02-18 13:07:02.000000000 +0200
+++ pacman-g2-3.9.0.new/scripts/makepkg	2015-04-05 17:17:16.000000000 +0300
@@ -366,6 +366,7 @@
 
 chroot_umount() {
     msg "Attempting to umount chroot directories..."
+    umount $CHROOTDIR/dev/shm >/dev/null
     umount $CHROOTDIR/proc >/dev/null
     umount $CHROOTDIR/sys >/dev/null
     umount $CHROOTDIR/dev >/dev/null
@@ -383,6 +384,7 @@
     mount -t proc none $CHROOTDIR/proc >/dev/null
     mount -t sysfs none $CHROOTDIR/sys >/dev/null
     mount -t devtmpfs none $CHROOTDIR/dev >/dev/null
+    mount -o bind /dev/shm $CHROOTDIR/dev/shm
     mount -o bind /var/cache/pacman-g2 $CHROOTDIR/var/cache/pacman-g2 >/dev/null
     mount -o bind $CCACHE_BASEDIR/$pkgname \
     $CHROOTDIR/var/cache/ccache/$pkgname >/dev/null
