diff --git a/scripts/makepkg b/scripts/makepkg
index a84c3ab..23d4924 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -114,6 +114,13 @@ Fextract() {
 # makepkg configuration
 [ -f /etc/makepkg.conf ] && source /etc/makepkg.conf
 
+## both are hacks , remove after 0.9 and move to makepkg.conf
+
+if `$ECHO $CHOST|grep -q linux-gnu`; then
+        unset CHOST
+        export CHOST="`arch`-frugalware-linux"
+fi
+
 # we need -fno-strict-aliasing for gcc 4.2*++
 _gccver=`type -p gcc >/dev/null && gcc -dumpversion`
 if [ "${_gccver%%.?}" == "4.3" ]; then
