--- pacman-g2-3.9.4/bindings/perl/Makefile.in	2018-01-30 22:06:55.000000000 +0100
+++ pacman-g2-3.9.4.new/bindings/perl/Makefile.in	2023-08-28 14:54:47.807834893 +0200
@@ -8,7 +8,7 @@
 CFLAGS += -fPIC
 endif
 LDFLAGS += -L$(top_srcdir)/lib/libpacman/.libs -lpacman
-LIBDIR += $(shell perl -V|grep site|sed 's/^ *//;s|/usr|$(prefix)|;q')
+LIBDIR += $(shell perl "-V::installsitelib:" | cut -d \' -f 2)
 
 all: Core.so
 
