--- pacman-g2-3.9.4/bindings/python/Makefile.in	2018-01-30 22:06:55.000000000 +0100
+++ pacman-g2-3.9.4.new/bindings/python/Makefile.in	2022-05-05 15:15:02.611815830 +0200
@@ -2,13 +2,13 @@
 prefix = @prefix@
 
 CFLAGS = $(subst -Werror,,@CFLAGS@)
-CFLAGS += $(shell python -c 'from distutils import sysconfig; print "-I" + sysconfig.get_python_inc()')
+CFLAGS += $(shell python3 -c 'from distutils import sysconfig; print("-I" + sysconfig.get_python_inc())')
 CFLAGS += -I$(top_srcdir)/lib/libpacman
 ifeq ($(shell arch),x86_64)
 CFLAGS += -fPIC
 endif
 LDFLAGS += -L$(top_srcdir)/lib/libpacman/.libs -lpacman
-LIBDIR += $(shell python -c 'from distutils import sysconfig; print sysconfig.get_python_lib()'|sed 's|/usr|$(prefix)|')
+LIBDIR += $(shell python3 -c 'from distutils import sysconfig; print(sysconfig.get_python_lib())'|sed 's|/usr|$(prefix)|')
 
 all: _pacman.so pacman.pyc
 
@@ -21,12 +21,10 @@
 pacman_wrap.c:
 	cp $(top_srcdir)/lib/libpacman/pacman.h ./
 	cp $(top_srcdir)/bindings/pacman.i ./
-	swig -python pacman.i
-	# strip the unnecessary prefixes
-	sed -i 's/^pacman_//;s/^PM_//' pacman.py
+	swig -python -py3 pacman.i
 
 pacman.pyc: pacman.py
-	python -c "import compileall; compileall.compile_dir('.',1,'.')"
+	python3 -c "import compileall; compileall.compile_dir('.',1,'.',legacy=True)"
 
 pacman.py: pacman_wrap.c
 
