diff --git a/lib/libpacman/add.c b/lib/libpacman/add.c
index 0927f18d..5f8c19ec 100644
--- a/lib/libpacman/add.c
+++ b/lib/libpacman/add.c
@@ -261,6 +261,7 @@ int _pacman_add_prepare(pmtrans_t *trans, pmlist_t **data)
 		info=(pmpkg_t *)lp->data;
 		for (rmlist=info->removes; rmlist!=NULL; rmlist=rmlist->next) {
 			snprintf(rm_fname, PATH_MAX, "%s%s", handle->root, (char *)rmlist->data);
+			_pacman_log(PM_LOG_FLOW1, _("removes(): about to delete %s%s"), handle->root, (char *)rmlist->data);
 			remove(rm_fname);
 		}
 	}
