diff --git a/etc/pacman-g2.conf.in b/etc/pacman-g2.conf.in
index 23c24b3..d581d05 100644
--- a/etc/pacman-g2.conf.in
+++ b/etc/pacman-g2.conf.in
@@ -19,10 +19,10 @@ OldDelay    = 3
 # tree to avoid conflicts
 
 # -current
-Include = /etc/pacman-g2/repos/frugalware-current
+#Include = /etc/pacman-g2/repos/frugalware-current
 
 # -stable
-#Include = /etc/pacman-g2/repos/frugalware
+Include = /etc/pacman-g2/repos/frugalware
 
 # An example of a custom package repository.  See the pacman-g2 manpage for
 # tips on creating your own repositories.
