diff -ur a/etc/pacman-g2.conf.in b/etc/pacman-g2.conf.in
--- a/etc/pacman-g2.conf.in	2012-06-14 09:31:03.000000000 -0500
+++ b/etc/pacman-g2.conf.in	2013-02-01 16:09:26.957177982 -0600
@@ -19,10 +19,10 @@
 # tree to avoid conflicts
 
 # -current
-Include = /etc/pacman-g2/repos/frugalware-current
+#Include = /etc/pacman-g2/repos/frugalware-current
 
 # -stable
-#Include = /etc/pacman-g2/repos/frugalware
+Include = /etc/pacman-g2/repos/frugalware
 
 # An example of a custom package repository.  See the pacman-g2 manpage for
 # tips on creating your own repositories.
