diff --git a/etc/pacman.conf.in b/etc/pacman.conf.in
index d8e9374..9656565 100644
--- a/etc/pacman.conf.in
+++ b/etc/pacman.conf.in
@@ -19,10 +19,10 @@ OldDelay    = 3
 # tree to avoid conflicts
 
 # -current
-Include = /etc/pacman.d/frugalware-current
+#Include = /etc/pacman.d/frugalware-current
 
 # -stable
-#Include = /etc/pacman.d/frugalware
+Include = /etc/pacman.d/frugalware
 
 # An example of a custom package repository.  See the pacman-g2 manpage for
 # tips on creating your own repositories.
