diff -Naur pacman-g2-3.9.4/configure.ac pacman-g2-3.9.4.new/configure.ac
--- pacman-g2-3.9.4/configure.ac	2018-01-30 22:06:55.000000000 +0100
+++ pacman-g2-3.9.4.new/configure.ac	2022-04-01 20:00:15.015686368 +0200
@@ -54,6 +54,11 @@
 AM_GNU_GETTEXT_VERSION(0.13.1)
 AM_CONDITIONAL(LINKSTATIC, test "$enable_static" = "yes")
 
+WL_WHOLE_ARCHIVE_HACK="-Wl,--whole-archive"
+WL_NO_WHOLE_ARCHIVE_HACK="-Wl,--no-whole-archive"
+AC_SUBST(WL_WHOLE_ARCHIVE_HACK)
+AC_SUBST(WL_NO_WHOLE_ARCHIVE_HACK)
+
 dnl Humor lowers blood pressure
 AC_MSG_CHECKING(your blood pressure)
 AC_MSG_RESULT([a bit high, but we can proceed])
diff -Naur pacman-g2-3.9.4/src/pacman-g2/Makefile.am pacman-g2-3.9.4.new/src/pacman-g2/Makefile.am
--- pacman-g2-3.9.4/src/pacman-g2/Makefile.am	2018-01-30 22:06:55.000000000 +0100
+++ pacman-g2-3.9.4.new/src/pacman-g2/Makefile.am	2022-04-01 20:07:04.638835856 +0200
@@ -20,9 +20,9 @@
 pacman_g2_LDADD = \
 	$(top_builddir)/lib/libpacman/libpacman.la
 
-pacman_g2_static_LDADD = $(pacman_g2_LDADD)
+pacman_g2_static_LDADD = $(pacman_g2_LDADD) @WL_WHOLE_ARCHIVE_HACK@ -lpthread -lrt -lpthread @WL_NO_WHOLE_ARCHIVE_HACK@
 
-pacman_g2_static_LDFLAGS = -all-static $(LDFLAGS)
+pacman_g2_static_LDFLAGS = -static -pthread $(LDFLAGS)
 
 install-exec-hook:
 	ln -sf pacman-g2 $(DESTDIR)$(bindir)/pacman
