diff -Naur pacman-g2-3.9.2/scripts/makepkg pacman-g2-3.9.2-strip2/scripts/makepkg
--- pacman-g2-3.9.2/scripts/makepkg	2016-10-22 08:30:15.175212510 +0200
+++ pacman-g2-3.9.2-strip2/scripts/makepkg	2016-10-22 08:32:33.083214555 +0200
@@ -587,11 +587,11 @@
 	fi
 	if [ ! "`check_option NOSTRIP`" -a "$NOSTRIP" = "0" ]; then
 		msg "Stripping symbols from libraries from $subpkg$1..."
-		find $dir/{,usr,usr/local,opt/*}/{lib,lib32} -type f -not -name "*.dll" -not -name "*.exe" \
+		find $dir/{,usr,usr/local,opt/*}/{lib,lib32} -type f -not -name "*.dll" -not -name "*.exe" -not -name "*.a"  -not -name "*.o" \
 			-exec /usr/bin/strip -s '{}' \; 2>&1 \
 			| grep -v "No such file" | grep -v "format not recognized"
 		msg "Stripping symbols from binaries from $subpkg$p..."
-		find $dir/{,usr,usr/local,usr/i686-frugalware-linux,opt/*}/{bin,sbin} -type f -not -name "*.dll" -not -name "*.exe" \
+		find $dir/{,usr,usr/local,usr/i686-frugalware-linux,opt/*}/{bin,sbin} -type f -not -name "*.dll" -not -name "*.exe" -not -name "*.a"  -not -name "*.o" \
 			-exec /usr/bin/strip '{}' \; 2>&1 \
 			| grep -v "No such file" | grep -v "format not recognized"
 	fi
