diff -Naur pacman-g2-3.9.2/scripts/makepkg pacman-g2-3.9.2-strip/scripts/makepkg
--- pacman-g2-3.9.2/scripts/makepkg	2016-10-16 20:07:36.199206578 +0200
+++ pacman-g2-3.9.2-strip/scripts/makepkg	2016-10-16 20:28:42.268225344 +0200
@@ -586,12 +586,12 @@
 		subpkg="subpackage "
 	fi
 	if [ ! "`check_option NOSTRIP`" -a "$NOSTRIP" = "0" ]; then
-		msg "Stripping debugging symbols from libraries from $subpkg$1..."
-		find $dir/{,usr,usr/local,opt/*}/lib -type f -not -name "*.dll" -not -name "*.exe" \
-			-exec /usr/bin/strip --strip-debug '{}' \; 2>&1 \
+		msg "Stripping symbols from libraries from $subpkg$1..."
+		find $dir/{,usr,usr/local,opt/*}/{lib,lib32} -type f -not -name "*.dll" -not -name "*.exe" \
+			-exec /usr/bin/strip -s '{}' \; 2>&1 \
 			| grep -v "No such file" | grep -v "format not recognized"
 		msg "Stripping symbols from binaries from $subpkg$p..."
-		find $dir/{,usr,usr/local,opt/*}/{bin,sbin} -type f -not -name "*.dll" -not -name "*.exe" \
+		find $dir/{,usr,usr/local,usr/i686-frugalware-linux,opt/*}/{bin,sbin} -type f -not -name "*.dll" -not -name "*.exe" \
 			-exec /usr/bin/strip '{}' \; 2>&1 \
 			| grep -v "No such file" | grep -v "format not recognized"
 	fi
