From 857f450dada3b95fceddfa0004caf8bed7b3bd50 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Mon, 19 Aug 2024 14:08:50 +0200
Subject: [PATCH] * use sqv directly

---
 scripts/makepkg | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index fd99b083..da958717 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -1862,17 +1862,7 @@ if [ "$GENSHA1" = "0" ]; then
 					sig=`strip_url "${signatures[$idx]}"`
 					$ECHO -n "    $file ... " >&2
 					if [ ! -z "$sig" ]; then
-						echo "$sig" | grep -q "^$file" >/dev/null 2>&1
-						if [ $? -ne 0 ]; then
-							case $file in
-								*.gz)  cmd='zcat'  ;;
-								*.bz2) cmd='bzcat' ;;
-								*.xz)  cmd='xzcat' ;;
-							esac
-						else
-							cmd='cat'
-						fi
-						$cmd $file | sqv --keyring $startdir/$pkgname.key $sig - >/dev/null 2>&1
+						sqv --keyring $startdir/$pkgname.key $sig $file >/dev/null 2>&1
 						if [ $? -ne 0 ]; then
 							$ECHO "FAILED" >&2
 							errors=1
-- 
2.46.0

