From 71e980d7dea53d8f17c932161bd03f114d690261 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Mon, 19 Aug 2024 11:15:35 +0200
Subject: [PATCH] * use gpgv to verify signatures

---
 scripts/makepkg | 14 +-------------
 1 file changed, 1 insertion(+), 13 deletions(-)

diff --git a/scripts/makepkg b/scripts/makepkg
index 37c4c38f..cd51f2d7 100755
--- a/scripts/makepkg
+++ b/scripts/makepkg
@@ -1855,18 +1855,6 @@ if [ "$GENSHA1" = "0" ]; then
 		# gpg validation
 		if [ ${#signatures[@]} -eq ${#source[@]} ]; then
 			if [ `type -p gpg` ]; then
-				if [ ! -d ~/.gnupg ]; then
-					msg "Configuring gpg..."
-					gpg --list-keys
-					$ECHO "keyserver hkp://pool.sks-keyservers.net" >>~/.gnupg/gpg.conf
-					$ECHO "keyserver-options auto-key-retrieve" >>~/.gnupg/gpg.conf
-				fi
-				if [ -e $startdir/$pkgname.key ]; then
-					msg "Importing $pkgname.key"
-					gpg --import $startdir/$pkgname.key
-				fi
-				msg "Killing dirmngr"
-				killall -u fst dirmngr
 				msg "Validating source files with gpg"
 				errors=0
 				idx=0
@@ -1885,7 +1873,7 @@ if [ "$GENSHA1" = "0" ]; then
 						else
 							cmd='cat'
 						fi
-						$cmd $file | gpg --batch --verify $sig - >/dev/null 2>&1
+						$cmd $file |  gpgv --keyring <(gpg -o - --dearmor $startdir/$pkgname.key ) $sig - >/dev/null 2>&1
 						if [ $? -ne 0 ]; then
 							$ECHO "FAILED" >&2
 							errors=1
-- 
2.46.0

