--- pacman-g2-3.9.4/scripts/makepkg	2023-12-31 21:42:36.510015989 +0100
+++ pacman-g2-3.9.4.new/scripts/makepkg	2023-12-31 21:42:16.052868407 +0100
@@ -26,7 +26,7 @@
 #
 
 ## bash's 'echo' is buggy and does not check on write() for errors
-ECHO="/bin/echo"
+ECHO="/usr/bin/echo"
 startdir=`pwd`
 PKGDEST="$startdir"
 USE_COLOR="n"
@@ -366,14 +366,14 @@
 	# need that bc we may pull in dplicated flags from FB and/or Finclde
 	# just sanitize all FLAGS here
 
-	_LDFLAGS="$(/bin/echo ${_LDFLAGS[@]} | tr [:space:] '\n' | awk '!a[$0]++' | tr '\n' ' ')"
+	_LDFLAGS="$(/usr/bin/echo ${_LDFLAGS[@]} | tr [:space:] '\n' | awk '!a[$0]++' | tr '\n' ' ')"
 	_CFLAGS="${_MFLAGS} ${_OFLAG} ${_FLAGS} ${_LD_PLUGIN} -pipe"
 	_CXXFLAGS="${_MFLAGS} ${_OFLAG} ${_FLAGS} ${_LD_PLUGIN} -pipe"
 	LDFLAGS_=" ${_LDFLAGS} ${_CFLAGS} ${_CXXFLAGS}"
 
-	CFLAGS="$(/bin/echo ${_CFLAGS[@]} | tr [:space:] '\n' | awk '!a[$0]++' | tr '\n' ' ')"
-	CXXFLAGS="$(/bin/echo ${_CXXFLAGS[@]} | tr [:space:] '\n' | awk '!a[$0]++' | tr '\n' ' ')"
-	LDFLAGS="$(/bin/echo ${LDFLAGS_[@]} | tr [:space:] '\n' | awk '!a[$0]++' | tr '\n' ' ')"
+	CFLAGS="$(/usr/bin/echo ${_CFLAGS[@]} | tr [:space:] '\n' | awk '!a[$0]++' | tr '\n' ' ')"
+	CXXFLAGS="$(/usr/bin/echo ${_CXXFLAGS[@]} | tr [:space:] '\n' | awk '!a[$0]++' | tr '\n' ' ')"
+	LDFLAGS="$(/usr/bin/echo ${LDFLAGS_[@]} | tr [:space:] '\n' | awk '!a[$0]++' | tr '\n' ' ')"
 
 	export LDFLAGS CFLAGS CXXFLAGS
 
@@ -1049,7 +1049,7 @@
 
 
 ## -t foo,bar -> foo
-MAKEPKG_TREE=$(/bin/echo $TREE | cut -d, -f1)
+MAKEPKG_TREE=$(/usr/bin/echo $TREE | cut -d, -f1)
 
 [[ -z ${MAKEFLAGS} ]] && export MAKEFLAGS="-j`getconf _NPROCESSORS_ONLN`"
 
@@ -2309,7 +2309,7 @@
 			if [ "$d" = "$startdir/pkg" ]; then
 				msg2 "Maybe broken link ${l#$d} in pkg $pkgname found."
 			else
-				if [[ "$(/bin/echo ${d##*/pkg.} | sed 's/.*-\(.*\)/\1/')" =~ "devel" ]]; then
+				if [[ "$(/usr/bin/echo ${d##*/pkg.} | sed 's/.*-\(.*\)/\1/')" =~ "devel" ]]; then
 					continue
 				else
 					msg2 "Maybe broken link ${l#$d} in pkg ${d##*/pkg.} found."
@@ -2319,7 +2319,7 @@
 			if [ "$d" = "$startdir/pkg" ]; then
 				msg2 "Broken link ${l#$d} in pkg $pkgname found."
 			else
-				if [[ "$(/bin/echo ${d##*/pkg.} | sed 's/.*-\(.*\)/\1/')" =~ "devel" ]]; then
+				if [[ "$(/usr/bin/echo ${d##*/pkg.} | sed 's/.*-\(.*\)/\1/')" =~ "devel" ]]; then
 					continue
 				else
 					msg2 "Broken link ${l#$d} in pkg ${d##*/pkg.} found."
