Index: sslcon.c
===================================================================
RCS file: /cvsroot/mozilla/security/nss/lib/ssl/sslcon.c,v
retrieving revision 1.34
diff -p -u -9 -r1.34 mozilla/security/nss/lib/ssl/sslcon.c
--- mozilla/security/nss/lib/ssl/sslcon.c	14 Nov 2006 01:09:54 -0000	1.34
+++ mozilla/security/nss/lib/ssl/sslcon.c	19 Dec 2006 08:08:39 -0000
@@ -1975,19 +1975,22 @@ ssl2_ClientHandleServerCert(sslSocket *s
 unsigned char *
 ssl_FormatSSL2Block(unsigned modulusLen, SECItem *data)
 {
     unsigned char *block;
     unsigned char *bp;
     int padLen;
     SECStatus rv;
     int i;
 
-    PORT_Assert (data->len <= (modulusLen - (3 + RSA_BLOCK_MIN_PAD_LEN)));
+    if (modulusLen < data->len + (3 + RSA_BLOCK_MIN_PAD_LEN)) {
+	PORT_SetError(SEC_ERROR_BAD_KEY);
+    	return NULL;
+    }
     block = (unsigned char *) PORT_Alloc(modulusLen);
     if (block == NULL)
 	return NULL;
 
     bp = block;
 
     /*
      * All RSA blocks start with two octets:
      *	0x00 || BlockType
