diff -Naur GammaRay-2.8.1/ui/resources/CMakeLists.txt GammaRay-2.8.1-p/ui/resources/CMakeLists.txt
--- GammaRay-2.8.1/ui/resources/CMakeLists.txt	2017-09-05 23:29:29.000000000 +0200
+++ GammaRay-2.8.1-p/ui/resources/CMakeLists.txt	2017-10-14 20:47:47.112192388 +0200
@@ -4,13 +4,13 @@
     foreach(size 16 24 32 48 64 128 256 512)
         foreach(scale 1 2 3)
             set(scaled_size ${size}x${size})
+            set(source gammaray/GammaRay-${scaled_size}.png)
 
             if(${scale} GREATER 1)
                 set(scaled_size ${size}x${size}@${scale})
+                set(source gammaray/GammaRay-${scaled_size}x.png)
             endif()
 
-            set(source gammaray/GammaRay-${scaled_size}x.png)
-
             if(EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${source})
                 install(
                     FILES ${CMAKE_CURRENT_SOURCE_DIR}/${source}
