From d0c9f9c8a0c62c53812c25e5bd77119f214cd1f8 Mon Sep 17 00:00:00 2001
From: Volker Krause <volker.krause@kdab.com>
Date: Mon, 20 Aug 2018 13:47:35 +0200
Subject: [PATCH 1/2] Adapt to Qt 5.12 V4 identifier changes

---
 plugins/qmlsupport/qmlcontextpropertyadaptor.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/plugins/qmlsupport/qmlcontextpropertyadaptor.cpp b/plugins/qmlsupport/qmlcontextpropertyadaptor.cpp
index 2c183f083..1c24eaa44 100644
--- a/plugins/qmlsupport/qmlcontextpropertyadaptor.cpp
+++ b/plugins/qmlsupport/qmlcontextpropertyadaptor.cpp
@@ -100,8 +100,13 @@ void QmlContextPropertyAdaptor::doSetObject(const ObjectInstance &oi)
     QV4::IdentifierHashEntry *e = propNames.d->entries;
     QV4::IdentifierHashEntry *end = e + propNames.d->alloc;
     while (e < end) {
+#if QT_VERSION < QT_VERSION_CHECK(5, 12, 0)
         if (e->identifier)
             m_contextPropertyNames.push_back(e->identifier->string);
+#else
+        if (e->identifier.isValid())
+            m_contextPropertyNames.push_back(e->identifier.toQString());
+#endif
         ++e;
     }
 #else

From 2cd4585e287a2bcf80572505537e5e75d020c5f5 Mon Sep 17 00:00:00 2001
From: Volker Krause <volker.krause@kdab.com>
Date: Wed, 22 Aug 2018 09:59:42 +0200
Subject: [PATCH 2/2] Compile with Qt 5.12

---
 plugins/qmlsupport/qmlbindingprovider.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/plugins/qmlsupport/qmlbindingprovider.cpp b/plugins/qmlsupport/qmlbindingprovider.cpp
index efc6f2b4d..0d210aa4c 100644
--- a/plugins/qmlsupport/qmlbindingprovider.cpp
+++ b/plugins/qmlsupport/qmlbindingprovider.cpp
@@ -35,6 +35,7 @@
 // Qt
 #include <private/qqmlabstractbinding_p.h>
 #include <private/qqmlbinding_p.h>
+#include <private/qqmlglobal_p.h>
 #include <private/qqmlvaluetypeproxybinding_p.h>
 #include <private/qqmldata_p.h>
 #include <private/qqmlproperty_p.h>
