diff -Naur claws-mail-3.15.1/src/plugins/Makefile.in claws-mail-3.15.1-p/src/plugins/Makefile.in
--- claws-mail-3.15.1/src/plugins/Makefile.in	2017-08-29 21:37:07.000000000 +0200
+++ claws-mail-3.15.1-p/src/plugins/Makefile.in	2017-09-24 01:05:38.718099345 +0200
@@ -95,13 +95,6 @@
 target_triplet = @target@
 subdir = src/plugins
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
-am__aclocal_m4_deps = $(top_srcdir)/m4/check-type.m4 \
-	$(top_srcdir)/m4/gnupg-check-typedef.m4 \
-	$(top_srcdir)/m4/gnupg.m4 $(top_srcdir)/m4/gpgme.m4 \
-	$(top_srcdir)/m4/isc-posix.m4 $(top_srcdir)/m4/libtool.m4 \
-	$(top_srcdir)/m4/ltoptions.m4 $(top_srcdir)/m4/ltsugar.m4 \
-	$(top_srcdir)/m4/ltversion.m4 $(top_srcdir)/m4/lt~obsolete.m4 \
-	$(top_srcdir)/m4/spamassassin.m4 $(top_srcdir)/configure.ac
 am__configure_deps = $(am__aclocal_m4_deps) $(CONFIGURE_DEPENDENCIES) \
 	$(ACLOCAL_M4)
 DIST_COMMON = $(srcdir)/Makefile.am $(am__DIST_COMMON)
