diff -up gvfs-1.5.4/configure.ac.servicedir gvfs-1.5.4/configure.ac
--- gvfs-1.5.4/configure.ac.servicedir	2010-02-22 10:39:26.000000000 -0500
+++ gvfs-1.5.4/configure.ac	2010-02-25 19:46:43.964765020 -0500
@@ -49,7 +49,7 @@ PKG_CHECK_MODULES(DBUS, dbus-1)
 
 AC_ARG_WITH(dbus_service_dir,
 		AS_HELP_STRING([--with-dbus-service-dir=PATH],[choose directory for dbus service files]),
-		[default=PREFIX/share/dbus-1/services]], with_dbus_service_dir="$withval", with_dbus_service_dir=$datadir/dbus-1/services)
+                with_dbus_service_dir="$withval", with_dbus_service_dir=$datadir/dbus-1/services)
 DBUS_SERVICE_DIR=$with_dbus_service_dir
 AC_SUBST(DBUS_SERVICE_DIR)
 
