diff -Naur xfce4-settings-4.5.92/xfsettingsd/registry.c xfce4-settings-4.5.92.new/xfsettingsd/registry.c
--- xfce4-settings-4.5.92/xfsettingsd/registry.c	2008-11-10 12:01:06.000000000 +0530
+++ xfce4-settings-4.5.92.new/xfsettingsd/registry.c	2008-11-20 18:57:01.000000000 +0530
@@ -243,18 +243,18 @@
     g_value_set_boolean (&properties[XSETTING_ENTRY_NET_ENABLEINPUTFEEDBACKSOUNDS].value, FALSE);
 
     /* Xft settings */
-    g_value_set_int (&properties[XSETTING_ENTRY_XFT_ANTIALIAS].value, -1);
+    g_value_set_int (&properties[XSETTING_ENTRY_XFT_ANTIALIAS].value, 1);
     g_value_set_int (&properties[XSETTING_ENTRY_XFT_HINTING].value, -1);
-    g_value_set_string (&properties[XSETTING_ENTRY_XFT_HINTSTYLE].value, "hintnone");
-    g_value_set_string (&properties[XSETTING_ENTRY_XFT_RGBA].value, "none");
-    g_value_set_int (&properties[XSETTING_ENTRY_XFT_DPI].value, -1);
+    g_value_set_string (&properties[XSETTING_ENTRY_XFT_HINTSTYLE].value, "hintfull");
+    g_value_set_string (&properties[XSETTING_ENTRY_XFT_RGBA].value, "vrgb");
+    g_value_set_int (&properties[XSETTING_ENTRY_XFT_DPI].value, 72);
     /* GTK settings */
     g_value_set_boolean (&properties[XSETTING_ENTRY_GTK_CANCHANGEACCELS].value, FALSE);
     g_value_set_string (&properties[XSETTING_ENTRY_GTK_COLORPALETTE].value,
                     "black:white:gray50:red:purple:blue:light "
                     "blue:green:yellow:orange:lavender:brown:goldenrod4:dodger "
                     "blue:pink:light green:gray10:gray30:gray75:gray90");
-    g_value_set_string (&properties[XSETTING_ENTRY_GTK_FONTNAME].value, "Sans 10");
+    g_value_set_string (&properties[XSETTING_ENTRY_GTK_FONTNAME].value, "Sans 12");
     g_value_set_string (&properties[XSETTING_ENTRY_GTK_ICONSIZES].value, NULL);
     g_value_set_string (&properties[XSETTING_ENTRY_GTK_KEYTHEMENAME].value, NULL);
     g_value_set_string (&properties[XSETTING_ENTRY_GTK_TOOLBARSTYLE].value, "Icons");
diff -Naur xfce4-settings-4.5.92/xfsettingsd/xsettings.xml xfce4-settings-4.5.92.new/xfsettingsd/xsettings.xml
--- xfce4-settings-4.5.92/xfsettingsd/xsettings.xml	2008-11-10 11:30:40.000000000 +0530
+++ xfce4-settings-4.5.92.new/xfsettingsd/xsettings.xml	2008-11-20 18:56:48.000000000 +0530
@@ -2,7 +2,7 @@
 
 <channel name="xsettings" version="1.0">
   <property name="Net" type="empty">
-    <property name="ThemeName" type="string" value="Xfce"/>
-    <property name="IconThemeName" type="string" value="Rodent"/>
+    <property name="ThemeName" type="string" value="Frugalware"/>
+    <property name="IconThemeName" type="string" value="Tango"/>
   </property>
 </channel>
