diff -aur a/xfsettingsd/registry.c b/xfsettingsd/registry.c
--- a/xfsettingsd/registry.c	2011-01-30 10:34:15.000000000 -0600
+++ b/xfsettingsd/registry.c	2011-03-15 16:32:40.134000088 -0500
@@ -384,7 +384,7 @@
     str = g_value_get_string (&properties[XSETTING_ENTRY_XFT_RGBA].value);
     update_property (props, "Xft.rgba", str);
 
-    /*update_property (props, "Xft.lcdfilter", g_str_equal (str, "rgb") ? "lcddefault" : "none");*/
+    update_property (props, "Xft.lcdfilter", (g_str_equal(str,"none") || g_str_equal(str,"unknown")) ? "lcdnone" : "lcddefault");
 
     if (g_value_get_int (&properties[XSETTING_ENTRY_XFT_HINTING].value))
         str = g_value_get_string (&properties[XSETTING_ENTRY_XFT_HINTSTYLE].value);
