diff -aur xfce4-settings-4.6.1/dialogs/keyboard-settings/xfce-keyboard-settings.c xfce4-settings-4.6.1-2/dialogs/keyboard-settings/xfce-keyboard-settings.c
--- xfce4-settings-4.6.1/dialogs/keyboard-settings/xfce-keyboard-settings.c	2009-03-02 11:48:00.000000000 -0600
+++ xfce4-settings-4.6.1-2/dialogs/keyboard-settings/xfce-keyboard-settings.c	2009-11-17 15:37:02.000000000 -0600
@@ -378,7 +378,7 @@
   xkl_config_rec_get_from_server (settings->priv->xkl_rec_config, settings->priv->xkl_engine);
 
   settings->priv->xkl_registry = xkl_config_registry_get_instance (settings->priv->xkl_engine);
-  xkl_config_registry_load (settings->priv->xkl_registry);
+  xkl_config_registry_load (settings->priv->xkl_registry, FALSE);
 
   /* Tab */
   xkb_tab_layout_vbox = glade_xml_get_widget (settings->priv->glade_xml, "xkb_tab_layout_vbox");
