diff -ur a/daemon/gvfsafpvolume.c b/daemon/gvfsafpvolume.c
--- a/daemon/gvfsafpvolume.c	2012-05-09 12:01:24.352008506 -0500
+++ b/daemon/gvfsafpvolume.c	2012-05-09 12:01:50.554007521 -0500
@@ -1082,7 +1082,7 @@
   return;
 
 error:
-  g_clear_object (info);
+  g_clear_object (&info);
   g_simple_async_result_take_error (simple, err);
   g_simple_async_result_complete (simple);
   g_object_unref (simple);
