diff -Naur OpenColorIO-1.1.0/src/core/CMakeLists.txt OpenColorIO-1.1.0-p/src/core/CMakeLists.txt
--- OpenColorIO-1.1.0/src/core/CMakeLists.txt	2018-01-05 02:38:27.000000000 +0100
+++ OpenColorIO-1.1.0-p/src/core/CMakeLists.txt	2018-05-19 01:00:42.474247284 +0200
@@ -24,7 +24,7 @@
         set(EXTERNAL_COMPILE_FLAGS "${EXTERNAL_COMPILE_FLAGS} /WX")
     endif()
 else()
-    set(EXTERNAL_COMPILE_FLAGS "${EXTERNAL_COMPILE_FLAGS} -Werror")
+    set(EXTERNAL_COMPILE_FLAGS "${EXTERNAL_COMPILE_FLAGS} ")
 endif()
 
 # SHARED
diff -Naur OpenColorIO-1.1.0/src/pyglue/CMakeLists.txt OpenColorIO-1.1.0-p/src/pyglue/CMakeLists.txt
--- OpenColorIO-1.1.0/src/pyglue/CMakeLists.txt	2018-01-05 02:38:27.000000000 +0100
+++ OpenColorIO-1.1.0-p/src/pyglue/CMakeLists.txt	2018-05-19 01:00:58.635318692 +0200
@@ -23,7 +23,7 @@
 # Process all warnings as errors
 # Unfortunately Windows still has a warning
 if(UNIX)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ")
 endif()
 
 find_package(PythonLibs)
