diff -Naur gmerlin-encoders-0.2.2/configure.ac gmerlin-encoders-0.2.2-p/configure.ac
--- gmerlin-encoders-0.2.2/configure.ac	2006-07-02 13:35:39.000000000 +0200
+++ gmerlin-encoders-0.2.2-p/configure.ac	2006-08-26 16:55:16.880418192 +0200
@@ -115,7 +115,7 @@
 
 SPEEX_REQUIRED="1.0.4"
 
-PKG_CHECK_MODULES(SPEEX, speex >= $SPEEX_REQUIRED, have_speex="true", have_speex="false")
+PKG_CHECK_MODULES(SPEEX, speex = $SPEEX_REQUIRED, have_speex="true", have_speex="false")
 AC_SUBST(SPEEX_REQUIRED)
 AC_SUBST(SPEEX_LIBS)
 AC_SUBST(SPEEX_CFLAGS)
