From 48897c4e5467c62cb36ab191eab6ce779f8714da Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Sat, 20 Feb 2010 05:31:02 +0100
Subject: [PATCH] gfpm-logviewer.c: fix segfault and actually make that work since it didn't

---
 src/gfpm-logviewer.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/src/gfpm-logviewer.c b/src/gfpm-logviewer.c
index 591aafb..7672623 100644
--- a/src/gfpm-logviewer.c
+++ b/src/gfpm-logviewer.c
@@ -18,6 +18,8 @@
  *  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
  */
 
+
+
 #include "gfpm-logviewer.h"
 #include "gfpm-messages.h"
 #include "gfpm-interface.h"
@@ -43,6 +45,7 @@ typedef struct _LogViewItem
 #define DMK_FILE "/share/gfpm/datemsk"
 
 int getdate_err;
+struct tm *getdate(const char *string);
 
 /* Log viewer widgets */
 static GtkWidget *gfpm_logviewer_dlg;
-- 
1.6.5.3

