--- gfpm-1.3.2/data/gfpm.ui	2009-09-05 17:42:44.000000000 +0200
+++ gfpm-1.3.2/data/gfpm.ui	2009-09-11 17:35:52.000000000 +0200
@@ -8,7 +8,7 @@
     <property name="default_height">600</property>
     <signal name="delete_event" handler="gtk_main_quit"/>
     <child>
-      <object class="GtkVBox" id="vbox1">
+      <object class="GtkVBox" id="vbox111">
         <property name="visible">True</property>
         <child>
           <object class="GtkMenuBar" id="menubar1">
@@ -48,7 +48,7 @@
                         <property name="label" translatable="yes">Repositories</property>
                         <property name="visible">True</property>
                         <property name="tooltip_text" translatable="yes">Manage Repositories</property>
-                        <property name="image">image1</property>
+                        <property name="image">image1111</property>
                         <property name="use_stock">False</property>
                       </object>
                     </child>
@@ -91,7 +91,7 @@
                         <property name="label" translatable="yes">_Optimize package database</property>
                         <property name="visible">True</property>
                         <property name="use_underline">True</property>
-                        <property name="image">image3</property>
+                        <property name="image">image3333</property>
                         <property name="use_stock">False</property>
                         <signal name="activate" handler="gtk_widget_show" object="optimize_db_dlg"/>
                       </object>
@@ -102,7 +102,7 @@
                         <property name="visible">True</property>
                         <property name="events">GDK_POINTER_MOTION_MASK | GDK_POINTER_MOTION_HINT_MASK | GDK_BUTTON_PRESS_MASK | GDK_BUTTON_RELEASE_MASK</property>
                         <property name="use_underline">True</property>
-                        <property name="image">image4</property>
+                        <property name="image">image4444</property>
                         <property name="use_stock">False</property>
                       </object>
                     </child>
@@ -278,11 +278,11 @@
                               </packing>
                             </child>
                             <child>
-                              <object class="GtkAlignment" id="alignment1">
+                              <object class="GtkAlignment" id="alignment1111">
                                 <property name="visible">True</property>
                                 <property name="top_padding">5</property>
                                 <child>
-                                  <object class="GtkVBox" id="vbox3">
+                                  <object class="GtkVBox" id="vbox333">
                                     <property name="visible">True</property>
                                     <child>
                                       <object class="GtkComboBox" id="combobox_repos">
@@ -489,7 +489,7 @@
                         <property name="left_padding">5</property>
                         <property name="right_padding">5</property>
                         <child>
-                          <object class="GtkScrolledWindow" id="scrolledwindow1">
+                          <object class="GtkScrolledWindow" id="scrolledwindow1111">
                             <property name="visible">True</property>
                             <property name="can_focus">False</property>
                             <property name="hscrollbar_policy">automatic</property>
@@ -807,7 +807,7 @@
                       </object>
                     </child>
                     <child type="label">
-                      <object class="GtkLabel" id="label1">
+                      <object class="GtkLabel" id="label1111">
                         <property name="visible">True</property>
                         <property name="label" translatable="yes">&lt;b&gt;Information&lt;/b&gt;</property>
                         <property name="use_markup">True</property>
@@ -1307,14 +1307,14 @@
             <property name="label_xalign">0</property>
             <property name="shadow_type">none</property>
             <child>
-              <object class="GtkAlignment" id="alignment1">
+              <object class="GtkAlignment" id="alignment11111">
                 <property name="visible">True</property>
                 <property name="top_padding">6</property>
                 <property name="bottom_padding">6</property>
                 <property name="left_padding">6</property>
                 <property name="right_padding">6</property>
                 <child>
-                  <object class="GtkVBox" id="vbox1">
+                  <object class="GtkVBox" id="vbox1111">
                     <property name="visible">True</property>
                     <property name="orientation">vertical</property>
                     <child>
@@ -3915,7 +3915,7 @@
       </object>
     </child>
   </object>
-  <object class="GtkImage" id="image1">
+  <object class="GtkImage" id="image1111">
     <property name="visible">True</property>
     <property name="icon_name">gfpm</property>
   </object>
@@ -3924,17 +3924,17 @@
     <property name="stock">gtk-clear</property>
     <property name="icon-size">1</property>
   </object>
-  <object class="GtkImage" id="image3">
+  <object class="GtkImage" id="image3333">
     <property name="visible">True</property>
     <property name="stock">gtk-refresh</property>
     <property name="icon-size">1</property>
   </object>
-  <object class="GtkImage" id="image4">
+  <object class="GtkImage" id="image4444">
     <property name="visible">True</property>
     <property name="stock">gtk-dialog-info</property>
     <property name="icon-size">1</property>
   </object>
-  <object class="GtkImage" id="image5">
+  <object class="GtkImage" id="image5555">
     <property name="visible">True</property>
     <property name="icon_name">gtk-clear</property>
   </object>
