diff -Naur gfpm-1.0.3/src/gfpm-interface.c gfpm-1.0.3.new/src/gfpm-interface.c
--- gfpm-1.0.3/src/gfpm-interface.c	2007-10-02 22:29:50.000000000 +0530
+++ gfpm-1.0.3.new/src/gfpm-interface.c	2007-10-08 19:03:29.000000000 +0530
@@ -1401,8 +1401,6 @@
 					-1);
 			pacman_pkg_free (pm_lpkg);
 			pacman_pkg_free (pm_spkg);
-			g_free (v1);
-			g_free (v2);
 		}
 		while (gtk_events_pending())
 			gtk_main_iteration ();
