diff -Naur gfpm-1.1.0/data/gfpm.glade gfpm-1.1.0.new/data/gfpm.glade
--- gfpm-1.1.0/data/gfpm.glade	2008-03-02 01:22:31.000000000 +0530
+++ gfpm-1.1.0.new/data/gfpm.glade	2008-03-04 21:41:03.000000000 +0530
@@ -201,7 +201,6 @@
             </child>
             <child>
               <widget class="GtkToolButton" id="button_help1">
-                <property name="visible">True</property>
                 <property name="label">Help</property>
                 <property name="use_underline">True</property>
                 <property name="stock_id">gtk-help</property>
