diff -Naur gfpm-1.0.1/src/gfpm-progress.c gfpm-1.0.1.new/src/gfpm-progress.c
--- gfpm-1.0.1/src/gfpm-progress.c	2007-09-05 22:01:50.000000000 +0530
+++ gfpm-1.0.1.new/src/gfpm-progress.c	2007-09-05 22:05:20.000000000 +0530
@@ -194,8 +194,8 @@
 		return;
 	gtk_widget_hide (rate_box);
 	gtk_widget_hide (rec_label);
-	while (gtk_events_pending ())
-		gtk_main_iteration ();
+	//while (gtk_events_pending ())
+	//	gtk_main_iteration ();
 	switch (event)
 	{
 		case PM_TRANS_EVT_CHECKDEPS_START:
