diff -Naur gfpm-1.1.1/src/gfpm-progress.c gfpm-1.1.1.new/src/gfpm-progress.c
--- gfpm-1.1.1/src/gfpm-progress.c	2008-03-06 18:27:15.000000000 +0530
+++ gfpm-1.1.1.new/src/gfpm-progress.c	2008-03-15 14:21:57.000000000 +0530
@@ -98,7 +98,7 @@
 cb_gfpm_close_button_clicked (GtkWidget *button, gpointer data)
 {
 	pacman_trans_release ();
-//	gfpm_progress_show (FALSE);
+	gfpm_progress_show (FALSE);
 
 	return;
 }
