diff -Naur gfpm-1.2.0/src/gfpm-progress.c gfpm-1.2.0.new/src/gfpm-progress.c
--- gfpm-1.2.0/src/gfpm-progress.c	2009-03-11 22:27:38.000000000 +0530
+++ gfpm-1.2.0.new/src/gfpm-progress.c	2009-03-14 19:22:41.000000000 +0530
@@ -286,6 +286,10 @@
 		gfpm_progress_set_sub_text (sub_text);
 		g_free (sub_text);
 	}
+	/* set percentage */
+	char text[6];
+	snprintf (text, sizeof(text), "%d %%", percent);
+	gtk_progress_bar_set_text (progressbar, text);
 	gtk_progress_bar_set_fraction (progressbar, (float)percent/100);
 	g_free (main_text);
 	while (gtk_events_pending())
