diff -uraN boo.orig/default.build boo/default.build
--- boo.orig/default.build	2008-02-23 16:09:19.000000000 +0100
+++ boo/default.build	2008-02-23 16:13:14.000000000 +0100
@@ -53,7 +53,7 @@
 
 	<property name="fixture" value="" dynamic="true" />
 
-	<target name="all" depends="booc, booi, booish, boo-pkgconfig, Boo.Lang.CodeDom, Boo.Lang.Useful, boo, Boo.Microsoft.Build.Tasks, update-vs2005-env">	
+	<target name="all" depends="booc, booi, booish, boo-pkgconfig, Boo.Lang.CodeDom, Boo.Lang.Useful, boo">	
 	</target>
 	
 	<target name="set-release-configuration">
