--- src/Frames/MusikFrame.cpp.old	2005-11-22 03:43:39.000000000 +0800
+++ src/Frames/MusikFrame.cpp	2007-03-05 16:00:40.000000000 +0800
@@ -380,9 +380,7 @@
 	g_SourcesCtrl->SetAlignment(wxLAYOUT_LEFT);
 	g_SourcesCtrl->SetOrientation(wxLAYOUT_VERTICAL);
 	g_SourcesCtrl->SetSashBorder(wxSASH_RIGHT, true);
-	g_SourcesCtrl->SetSashBorder(wxSASH_RIGHT, true);
-	g_SourcesCtrl->SetDefaultBorderSize(3);
-	g_SourcesCtrl->SetExtraBorderSize(1);
+	g_SourcesCtrl->SetDefaultBorderSize(5);
 	//---------------------//
 	//--- activity area ---//
 	//---------------------//
@@ -392,8 +390,7 @@
 	g_ActivityAreaCtrl->SetAlignment(wxLAYOUT_TOP);
 	g_ActivityAreaCtrl->SetOrientation(wxLAYOUT_HORIZONTAL);
 	g_ActivityAreaCtrl->SetSashBorder(wxSASH_BOTTOM, true);
-	g_ActivityAreaCtrl->SetDefaultBorderSize(3);
-	g_ActivityAreaCtrl->SetExtraBorderSize(1);
+	g_ActivityAreaCtrl->SetDefaultBorderSize(5);
 	
 
 

