--- src/Classes/PlaylistCtrl.cpp.old	2005-11-22 03:43:38.000000000 +0800
+++ src/Classes/PlaylistCtrl.cpp	2005-12-17 23:47:08.000000000 +0800
@@ -195,7 +195,7 @@
 	EVT_UPDATE_UI				( MUSIK_PLAYLIST_DISPLAY_SMART,											CPlaylistCtrl::OnUpdateUIDisplaySmart)
 	EVT_MENU					( MUSIK_PLAYLIST_DISPLAY_FIT,											CPlaylistCtrl::OnDisplayFit			)
 	EVT_MENU					( MUSIK_PLAYLIST_CLEARPLAYERLIST,										CPlaylistCtrl::OnClearPlayerlist	)
-	EVT_KEY_DOWN				(																		CPlaylistCtrl::OnKeyDown			)
+	EVT_KEY_UP				(																		CPlaylistCtrl::OnKeyDown			)
 	EVT_LIST_COL_CLICK			( -1,																	CPlaylistCtrl::OnColumnClick		)
 
 	//---------------------------------------------------------//
