--- src/Classes/SourcesBox.cpp.old	2005-11-22 03:43:38.000000000 +0800
+++ src/Classes/SourcesBox.cpp	2006-09-10 02:04:44.000000000 +0800
@@ -953,7 +953,7 @@
 		#if defined(__WXMSW__) && !defined(USE_GENERICLISTCTRL)
 			SetColumnWidth	( main_col, nWidth );
 		#else
-			SetColumnWidth( main_col, nWidth - wxSystemSettings::GetMetric(wxSYS_HSCROLL_Y) /*- GetColumnWidth( 0 )*/ - 1 );			
+			SetColumnWidth( main_col, nWidth /*- GetColumnWidth( 0 )*/ - 1 );
 		#endif 
 	}
 }
@@ -1592,10 +1592,10 @@
 END_EVENT_TABLE()
 
 CSourcesBox::CSourcesBox( wxWindow *parent )
-	: wxSashLayoutWindow( parent, MUSIK_SOURCES, wxDefaultPosition , wxDefaultSize, wxNO_BORDER | wxCLIP_CHILDREN |wxSW_3D|wxTAB_TRAVERSAL )
+	: wxSashLayoutWindow( parent, MUSIK_SOURCES, wxDefaultPosition , wxDefaultSize, wxNO_BORDER | wxCLIP_CHILDREN |wxSW_3DSASH|wxTAB_TRAVERSAL )
 {
 
-	m_pPanel = new wxPanel( this, -1, wxDefaultPosition , wxDefaultSize , wxNO_BORDER|wxCLIP_CHILDREN|wxTAB_TRAVERSAL );
+	m_pPanel = new wxPanel( this, -1, wxDefaultPosition , wxDefaultSize , wxSUNKEN_BORDER|wxCLIP_CHILDREN|wxTAB_TRAVERSAL );
 	//--- CSourcesListBox ---//
 	m_pListBox	= new CSourcesListBox( m_pPanel );
 	m_pPictureBox = new CPictureBox(m_pPanel);
