diff -Naur libmikmod-3.1.11.orig/configure.in libmikmod-3.1.11/configure.in
--- libmikmod-3.1.11.orig/configure.in	2005-06-01 16:00:37.000000000 +0000
+++ libmikmod-3.1.11/configure.in	2005-06-01 16:00:52.000000000 +0000
@@ -257,10 +257,10 @@
 if test "`uname`" = "Linux" && test $libmikmod_driver_alsa = yes
 then
 	libmikmod_driver_alsa=no
-	AC_CHECK_HEADERS(sys/asoundlib.h)
-	if test $ac_cv_header_sys_asoundlib_h = yes
+	AC_CHECK_HEADERS(alsa/asoundlib.h)
+	if test $ac_cv_header_alsa_asoundlib_h = yes
 	then
-		AC_CHECK_LIB(asound,snd_cards,libmikmod_driver_alsa=yes)
+		AC_CHECK_LIB(asound,snd_card_load,libmikmod_driver_alsa=yes)
 	fi
 else
 	libmikmod_driver_alsa=no
@@ -377,7 +377,7 @@
 AC_CHECK_FUNCS(setenv snprintf srandom)
 AC_REPLACE_FUNCS(strcasecmp strdup strstr)
 # Change extension, as we use libtool
-LIBOBJS="`echo $LIBOBJS|sed -e 's/\.o/\.lo/g'`"
+#LIBOBJS="`echo $LIBOBJS|sed -e 's/\.o/\.lo/g'`"
 
 AC_HEADER_EGREP(srandom,math.h,AC_DEFINE(SRANDOM_IN_MATH_H))
 
diff -Naur libmikmod-3.1.11.orig/drivers/drv_alsa.c libmikmod-3.1.11/drivers/drv_alsa.c
--- libmikmod-3.1.11.orig/drivers/drv_alsa.c	2005-06-01 16:00:37.000000000 +0000
+++ libmikmod-3.1.11/drivers/drv_alsa.c	2005-06-01 16:00:47.000000000 +0000
@@ -47,7 +47,7 @@
 #include <stdlib.h>
 #include <string.h>
 
-#include <sys/asoundlib.h>
+#include <alsa/asoundlib.h>
 #if defined(SND_LIB_VERSION) && (SND_LIB_VERSION >= 0x500)
 #undef DRV_ALSA
 #endif
diff -Naur libmikmod-3.1.11.orig/include/mikmod_internals.h libmikmod-3.1.11/include/mikmod_internals.h
--- libmikmod-3.1.11.orig/include/mikmod_internals.h	2005-06-01 16:00:37.000000000 +0000
+++ libmikmod-3.1.11/include/mikmod_internals.h	2005-06-01 16:00:47.000000000 +0000
@@ -50,7 +50,7 @@
 /*========== More type definitions */
 
 /* SLONGLONG: 64bit, signed */
-#if defined (__arch64__) || defined(__alpha)
+#if defined(__arch64__) || defined(__alpha) || defined(__x86_64) || defined(_LP64)
 typedef long		SLONGLONG;
 #define NATIVE_64BIT_INT
 #elif defined(__WATCOMC__)
diff -Naur libmikmod-3.1.11.orig/macintosh/mikmod.h libmikmod-3.1.11/macintosh/mikmod.h
--- libmikmod-3.1.11.orig/macintosh/mikmod.h	2005-06-01 16:00:37.000000000 +0000
+++ libmikmod-3.1.11/macintosh/mikmod.h	2005-06-01 16:00:47.000000000 +0000
@@ -85,7 +85,7 @@
 
 
 
-#if defined(__arch64__) || defined(__alpha)
+#if defined(__arch64__) || defined(__alpha) || defined(__x86_64) || defined(_LP64)
 /* 64 bit architectures */
 
 typedef signed char     SBYTE;      /* 1 byte, signed */
diff -Naur libmikmod-3.1.11.orig/win32/mikmod_build.h libmikmod-3.1.11/win32/mikmod_build.h
--- libmikmod-3.1.11.orig/win32/mikmod_build.h	2005-06-01 16:00:37.000000000 +0000
+++ libmikmod-3.1.11/win32/mikmod_build.h	2005-06-01 16:00:47.000000000 +0000
@@ -85,7 +85,7 @@
 
 
 
-#if defined(__arch64__) || defined(__alpha)
+#if defined(__arch64__) || defined(__alpha) || defined(__x86_64) || defined(_LP64)
 /* 64 bit architectures */
 
 typedef signed char     SBYTE;      /* 1 byte, signed */
