--- installpkg.orig	2004-06-18 20:51:06.000000000 +0200
+++ installpkg	2004-06-18 20:51:14.000000000 +0200
@@ -4,15 +4,18 @@
 # installpkg for pacman
 # distributed under GPL License
 
-version=0.5
+version=0.6
 
 if ! [ -e *fpm ]; then
 	echo "No package in this directory!"
 elif [ `ls *.fpm|wc|tr -s ' '|cut -d ' ' -f 3` == 1 ]; then
 	pkgname=`cat FrugalBuild |grep pkgname|sed -n '1 p'|cut -d = -f 2`
-	cmda="rm -f ../../../frugalware/`ls |grep "^$pkgname-[[:digit:]]"|grep fpm`"
-	echo "$cmda"
-	$cmda
+	oldpkg=`cd ../../../frugalware/;ls |grep "^$pkgname-[[:digit:]]"|grep fpm`
+	if ! [ -z "$oldpkg" ]; then
+		cmda="rm -f ../../../frugalware/$oldpkg"
+		echo "$cmda"
+		$cmda
+	fi
 	cmdb="mv `ls *fpm` ../../../frugalware/"
 	echo "$cmdb"
 	$cmdb
