diff -Naur XGngeo-16/setup.py XGngeo-16.new/setup.py
--- XGngeo-16/setup.py	2006-09-25 01:33:40.000000000 +0530
+++ XGngeo-16.new/setup.py	2007-06-03 12:41:23.000000000 +0530
@@ -71,26 +71,3 @@
 			[os.path.join("data", "locale", "pt_BR", "LC_MESSAGES", "xgngeo.mo")])
 		]
 	)
-	
-if "install" in sys.argv:
-	#Post-install stuffs (Unix).
-	if os.name == "posix":
-		#Determining the best start-up script path prefix.
-		prefix = None
-		str = "--prefix="
-
-		for val in sys.argv: #Check if mentioned as command parameter.
-			if val[:len(str)] == str:
-				prefix = val[len(str):]
-				break
-
-		if not prefix:
-			prefix = sys.prefix #Use Python prefix.
-
-		startup_script = os.path.join(prefix, "bin", "xgngeo") #Startup script path.
-		startup_script_dir = os.path.dirname(startup_script) 
-		
-		import shutil
-		if not os.path.exists(startup_script_dir): os.makedirs(startup_script_dir)
-		shutil.copy("./data/script/xgngeo_startup.py", startup_script)
-		print "XGngeo start-up script put into `%s'." % startup_script_dir
