diff --git a/configure.ac b/configure.ac
index 409e5fd..dec388c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -85,8 +85,8 @@ AC_SUBST(DFB_LIBDIR)
 # to modify src/Makefile.am where it says "STATIC"
 # sample options to include to directfb-config below (before ' | sed ...'):
 #--input=keyboard --imageprovider=png --font=ft2,default 
-DFB_LIBS="$(eval directfb-config --libs | sed 's/$/ \\/')"
-#AC_DEFINE([DFB_LIBS], [$DFB_LIBS], [directfb libs])
+DFB_LIBS="$(eval directfb-config --libs --input=keyboard --imageprovider=png --font=ft2,default| sed 's/$/ \\/')"
+AC_DEFINE([DFB_LIBS], [$DFB_LIBS], [directfb libs])
 AC_SUBST(DFB_LIBS)
 
 # TODO slurp the doc/*.xml.in or .xml files and create xml_format.h
diff --git a/src/Makefile.am b/src/Makefile.am
index a93a1f1..5e1e23f 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -38,21 +38,21 @@ libsplashycnf_la_LDFLAGS = -export-symbols-regex splashy_.*\
 			   -version-info $(SPL_SONAME)
 
 splashy_SOURCES = splashy_main.c splashy_functions.c
-splashy_LDADD = libsplashycnf.la libsplashy.la -lpthread -lgcc_s -lm  -lglib-2.0 @DFB_LIBS@
+splashy_LDADD = libsplashycnf.la libsplashy.la -lpthread -lm  -lglib-2.0 @DFB_LIBS@
 
 # '-lgcc_s' is provided with -static-libgcc and this is needed when compiling statically only
 # uncomment the following for static builds:
-#splashy_LDFLAGS = -all-static -static-libgcc $(DFBHACK)
+splashy_LDFLAGS = -all-static -static-libgcc $(DFBHACK)
 # all other builds:
 # FIXME doesn't DFB_LIBS provide this from configure.ac? $(shell directfb-config --libs) 
-splashy_LDFLAGS = -rpath $(libdir)/directfb-1.0-0/systems:$(libdir)/directfb-1.0-0/wm:/usr/lib64/directfb-1.0-0/interfaces/IDirectFBFont/:$(libdir)/directfb-1.0-0/interfaces/IDirectFBImageProvider:$(libdir)/directfb-1.0-0/inputdrivers
+#splashy_LDFLAGS = -rpath $(libdir)/directfb-1.0-0/systems:$(libdir)/directfb-1.0-0/wm:/usr/lib64/directfb-1.0-0/interfaces/IDirectFBFont/:$(libdir)/directfb-1.0-0/interfaces/IDirectFBImageProvider:$(libdir)/directfb-1.0-0/inputdrivers
 
 splashy_config_SOURCES = splashy_config-main.c splashy_config-functions.c \
 	xml_functions.c
 splashy_config_LDFLAGS = -lsplashycnf -lglib-2.0 -lmagic -lz
 
 # uncomment the following if you need static builds for splashy_update as well:
-#splashy_update_LDFLAGS = -all-static -static-libgcc
+splashy_update_LDFLAGS = -all-static -static-libgcc
 splashy_update_SOURCES = splashy_update.c
 
 install-exec-hook:
