diff -up epdfview-0.1.6/src/IDocumentLink.h.orig epdfview-0.1.6/src/IDocumentLink.h
--- epdfview-0.1.6/src/IDocumentLink.h.orig	2008-02-14 11:21:01.000000000 +0100
+++ epdfview-0.1.6/src/IDocumentLink.h	2008-02-14 11:21:06.000000000 +0100
@@ -33,7 +33,7 @@ namespace ePDFView
     class IDocumentLink
     {
         public:
-            IDocumentLink (gdouble x1, gdouble y1, gdouble x2, gdouble y1);
+            IDocumentLink (gdouble x1, gdouble y1, gdouble x2, gdouble y2);
             virtual ~IDocumentLink (void);
 
             virtual void activate (IDocument *document) = 0;
