diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4e78dda..af745c5 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,9 +18,6 @@ include(KDECMakeSettings)
 include(KDECompilerSettings NO_POLICY_SCOPE)
 include(ECMOptionalAddSubdirectory)
 
-include(FindPkgConfig)
-pkg_check_modules(GCONF REQUIRED gconf-2.0)
-pkg_check_modules(GOBJECT REQUIRED gobject-2.0)
 
 find_package(Qt5 ${QT_MIN_VERSION} REQUIRED COMPONENTS
     Core
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 21a6066..6d38932 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -1,5 +1,4 @@
-include_directories(${PULSEAUDIO_INCLUDE_DIR} ${GLIB2_INCLUDE_DIR}     ${GCONF_INCLUDE_DIRS}
-    ${GOBJECT_INCLUDE_DIRS})
+include_directories(${PULSEAUDIO_INCLUDE_DIR} ${GLIB2_INCLUDE_DIR})
 
 set(SRC_LIST
     card.cpp
@@ -15,8 +14,6 @@ set(SRC_LIST
     ref.cpp
     sink.cpp
     sinkinput.cpp
-    modulemanager.cpp
-    gconfitem.cpp
     source.cpp
     sourceoutput.cpp
     stream.cpp
@@ -32,9 +29,8 @@ target_link_libraries(QPulseAudioPrivate
     Qt5::Gui
     ${PULSEAUDIO_LIBRARY}
     ${PULSEAUDIO_MAINLOOP_LIBRARY}
-    ${GCONF_LDFLAGS}
-    ${GOBJECT_LDFLAGS}
 )
+
 install(TARGETS QPulseAudioPrivate ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 add_subdirectory(kcm)
diff --git a/src/kcm/package/contents/ui/CardView.qml b/src/kcm/package/contents/ui/CardView.qml
index fb7eb2f..a1ce20d 100644
--- a/src/kcm/package/contents/ui/CardView.qml
+++ b/src/kcm/package/contents/ui/CardView.qml
@@ -19,32 +19,8 @@
 */
 
 import QtQuick 2.0
-import QtQuick.Layouts 1.1
-import QtQuick.Controls 1.3
-
-import org.kde.plasma.private.volume 0.1
-
-ColumnLayout {
-    property alias model: view.model
-    property alias emptyText: view.emptyText
-    PulseView {
-        id: view
-        Layout.fillHeight: true
-        Layout.fillWidth: true
-        delegate: CardListItem {}
-    }
-    ModuleManager {
-        id: moduleManager
-    }
-    CheckBox {
-        text: i18n("Add virtual output device for simultaneous output on all local sound cards")
-        checked: moduleManager.combineSinks
-        onCheckedChanged: moduleManager.combineSinks = checked;
-    }
-    CheckBox {
-        text: i18n("Automatically switch all running streams when a new output becomes available")
-        checked: moduleManager.switchOnConnect
-        onCheckedChanged: moduleManager.switchOnConnect = checked;
-    }
 
+PulseView {
+    delegate: CardListItem {}
 }
+
diff --git a/src/qml/plugin.cpp b/src/qml/plugin.cpp
index e4604df..6efdc6c 100644
--- a/src/qml/plugin.cpp
+++ b/src/qml/plugin.cpp
@@ -27,7 +27,6 @@
 #include "sink.h"
 #include "source.h"
 #include "context.h"
-#include "modulemanager.h"
 
 #include "globalactioncollection.h"
 #include "volumeosd.h"
@@ -50,7 +49,6 @@ void Plugin::registerTypes(const char* uri)
     qmlRegisterType<QPulseAudio::SinkModel>(uri, 0, 1, "SinkModel");
     qmlRegisterType<QPulseAudio::SinkInputModel>(uri, 0, 1, "SinkInputModel");
     qmlRegisterType<QPulseAudio::SourceModel>(uri, 0, 1, "SourceModel");
-    qmlRegisterType<QPulseAudio::ModuleManager>(uri, 0, 1, "ModuleManager");
     qmlRegisterType<QPulseAudio::SourceOutputModel>(uri, 0, 1, "SourceOutputModel");
     qmlRegisterType<QPulseAudio::StreamRestoreModel>(uri, 0, 1, "StreamRestoreModel");
     qmlRegisterType<GlobalAction>(uri, 0, 1, "GlobalAction");
