diff -Naur module-init-tools-3.4/Makefile.am module-init-tools-3.4-p/Makefile.am
--- module-init-tools-3.4/Makefile.am	2007-10-07 23:51:46.000000000 +0200
+++ module-init-tools-3.4-p/Makefile.am	2007-12-19 13:31:01.000000000 +0100
@@ -31,7 +31,7 @@
 
 sbin_PROGRAMS = insmod modprobe rmmod depmod modinfo insmod.static
 bin_PROGRAMS = lsmod
-sbin_SCRIPTS = generate-modprobe.conf
+#sbin_SCRIPTS = generate-modprobe.conf
 INSTALL = $(SHELL) $(top_srcdir)/install-with-care
 
 TESTS = tests/runtests
diff -Naur module-init-tools-3.4/modprobe.c module-init-tools-3.4-p/modprobe.c
--- module-init-tools-3.4/modprobe.c	2007-10-07 23:57:23.000000000 +0200
+++ module-init-tools-3.4-p/modprobe.c	2007-12-19 13:29:38.000000000 +0100
@@ -1226,7 +1226,6 @@
 
 static const char *default_configs[] = 
 {
-	"/etc/modprobe.conf",
 	"/etc/modprobe.d",
 };
 
