diff --git a/Makefile.am b/Makefile.am
index defe6cc..02c447b 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -28,11 +28,10 @@ mandir =$(shell if [ @mandir@ = $(prefix)/man ]; then if [ $(prefix) = / ]; then
 
 TESTSUITE := $(shell find tests -type f ! -name '*~') tests/vg-suppressions
 
-EXTRA_DIST = generate-modprobe.conf modprobe.devfs FAQ CODING stress_modules.sh install-with-care $(SGML) $(man_MANS) $(TESTSUITE)
+EXTRA_DIST = modprobe.devfs FAQ CODING stress_modules.sh install-with-care $(SGML) $(man_MANS) $(TESTSUITE)
 
 sbin_PROGRAMS = insmod modprobe rmmod depmod modinfo insmod.static
 bin_PROGRAMS = lsmod
-sbin_SCRIPTS = generate-modprobe.conf
 noinst_PROGRAMS=modindex
 INSTALL = $(SHELL) $(top_srcdir)/install-with-care
 
diff --git a/modprobe.c b/modprobe.c
index 474232d..0c4f789 100644
--- a/modprobe.c
+++ b/modprobe.c
@@ -1340,7 +1340,6 @@ static int read_config_file_bin(const char *filename,
 
 static const char *default_configs[] = 
 {
-	"/etc/modprobe.conf",
 	"/etc/modprobe.d",
 };
 
