diff -Naur ksudoku-0.3/src/glwindow.cpp ksudoku-0.3.time/src/glwindow.cpp
--- ksudoku-0.3/src/glwindow.cpp	2005-09-29 17:16:04.000000000 +0200
+++ ksudoku-0.3.time/src/glwindow.cpp	2006-01-08 21:59:36.000000000 +0100
@@ -3,6 +3,7 @@
 #include <qcursor.h>
 #include "ksudoku.h"
 #include <kmessagebox.h>
+#include <time.h>
 GLUquadricObj *quadratic;											// Used For Our Quadric
 
 const float PI2 = 2.0*3.1415926535f;								// PI Squared
diff -Naur ksudoku-0.3/src/ksudokuview.cpp ksudoku-0.3.time/src/ksudokuview.cpp
--- ksudoku-0.3/src/ksudokuview.cpp	2005-09-29 17:15:47.000000000 +0200
+++ ksudoku-0.3.time/src/ksudokuview.cpp	2006-01-08 21:59:40.000000000 +0100
@@ -17,6 +17,7 @@
 #include <qfile.h>
 #include "ksudoku.h"
 #include "knewdlg.h"
+#include <time.h>
 
 ksudokuView::ksudokuView(QWidget *parent)
     : QWidget(parent),
