diff -Naur ed-0.2.orig/configure.in ed-0.2/configure.in
--- ed-0.2.orig/configure.in	2005-07-21 12:13:53.000000000 +0200
+++ ed-0.2/configure.in	2005-07-21 12:14:20.000000000 +0200
@@ -11,11 +11,11 @@
 AC_C_CONST
 AC_HEADER_STDC
 AC_CHECK_HEADERS(limits.h memory.h string.h unistd.h locale.h)
-AC_CHECK_FUNCS(setbuffer sigsetjmp sigaction strerror)
+AC_CHECK_FUNCS(setbuffer siglongjmp sigaction strerror)
 AC_FUNC_VPRINTF
 AC_FUNC_ALLOCA
 if test "$ALLOCA" = alloca.o; then
-  AC_SUBST(LIBOBJS)LIBOBJS="$LIBOBJS xmalloc.o error.o"
+  AC_LIBOBJ(xmalloc) AC_LIBOBJ(error)
 fi
 
 AC_OUTPUT(Makefile)
diff -Naur ed-0.2.orig/main.c ed-0.2/main.c
--- ed-0.2.orig/main.c	2005-07-21 12:13:53.000000000 +0200
+++ ed-0.2/main.c	2005-07-21 12:14:20.000000000 +0200
@@ -68,7 +68,7 @@
 #include "getopt.h"
 
 
-#ifdef HAVE_SIGSETJMP
+#ifdef HAVE_SIGLONGJMP
 sigjmp_buf env;
 #else
 jmp_buf env;
@@ -219,7 +219,7 @@
   reliable_signal (SIGHUP, signal_hup);
   reliable_signal (SIGQUIT, SIG_IGN);
   reliable_signal (SIGINT, signal_int);
-#ifdef HAVE_SIGSETJMP
+#ifdef HAVE_SIGLONGJMP
   if (status = sigsetjmp (env, 1))
 #else
   if (status = setjmp (env))
diff -Naur ed-0.2.orig/signal.c ed-0.2/signal.c
--- ed-0.2.orig/signal.c	2005-07-21 12:13:53.000000000 +0200
+++ ed-0.2/signal.c	2005-07-21 12:14:20.000000000 +0200
@@ -39,7 +39,7 @@
      void (*hndlr) ();
 {
 #ifndef HAVE_SIGACTION
-  signal (sno, hndlr);
+  return signal (sno, hndlr);
 #else
   struct sigaction sa, osa;
 
@@ -76,7 +76,7 @@
 }
 
 
-#ifdef HAVE_SIGSETJMP
+#ifdef HAVE_SIGLONGJMP
 extern sigjmp_buf env;
 #else
 extern jmp_buf env;
@@ -116,7 +116,7 @@
   if (!sigactive)
     quit (1);
   sigflags &= ~(1 << (signo - 1));
-#ifdef HAVE_SIGSETJMP
+#ifdef HAVE_SIGLONGJMP
   siglongjmp (env, -1);
 #else
   longjmp (env, -1);
