diff -ur a/module/Makefile.in b/module/Makefile.in
--- a/module/Makefile.in	2014-06-12 15:58:19.000000000 -0500
+++ b/module/Makefile.in	2014-07-18 04:35:42.041579117 -0500
@@ -1,7 +1,7 @@
 subdir-m += spl
 subdir-m += splat
 
-INSTALL_MOD_DIR ?= extra
+INSTALL_MOD_DIR ?= kernel/extra
 
 SPL_MODULE_CFLAGS  = -I@abs_top_srcdir@/include
 SPL_MODULE_CFLAGS += -include @abs_top_builddir@/spl_config.h
