diff -ur a/module/Makefile.in b/module/Makefile.in
--- a/module/Makefile.in	2014-06-12 15:58:19.000000000 -0500
+++ b/module/Makefile.in	2014-07-18 04:35:42.041579117 -0500
@@ -1,7 +1,7 @@
 subdir-m += spl
 subdir-m += splat
 
-INSTALL_MOD_DIR ?= extra
+INSTALL_MOD_DIR ?= kernel/extra
 
 SPL_MODULE_CFLAGS  = -I@abs_top_srcdir@/include
 SPL_MODULE_CFLAGS += -include @abs_top_builddir@/spl_config.h
--- spl-0.6.5.7/include/linux/file_compat.h	2015-12-24 01:18:07.000000000 +0100
+++ spl-0.6.5.7.new/include/linux/file_compat.h	2016-07-26 11:42:53.335661282 +0200
@@ -76,8 +76,8 @@
 #define	spl_filp_fsync(fp, sync)	vfs_fsync(fp, (fp)->f_dentry, sync)
 #endif /* HAVE_2ARGS_VFS_FSYNC */
 
-#define	spl_inode_lock(ip)		mutex_lock(&(ip)->i_mutex)
-#define	spl_inode_unlock(ip)		mutex_unlock(&(ip)->i_mutex)
+#define	spl_inode_lock(ip)		inode_lock(ip)
+#define	spl_inode_unlock(ip)		inode_unlock(ip)
 
 #endif /* SPL_FILE_COMPAT_H */
 
