diff -ur a/configure b/configure
--- a/configure	2011-12-11 16:01:12.000000000 -0600
+++ b/configure	2012-09-19 21:24:08.889699997 -0500
@@ -2511,7 +2511,7 @@
 VERSION_MINOR=17
 VERSION_REVISION=0
 
-versioninfo=3:0:0
+versioninfo=99:0:0
 
 am__api_version='1.11'
 
diff -ur a/configure.ac b/configure.ac
--- a/configure.ac	2011-12-11 15:58:50.000000000 -0600
+++ b/configure.ac	2012-09-19 21:24:01.824700252 -0500
@@ -54,7 +54,7 @@
 VERSION_MINOR=version_minor
 VERSION_REVISION=version_revision
 AC_SUBST(versioninfo)
-versioninfo=3:0:0
+versioninfo=99:0:0
 
 AM_INIT_AUTOMAKE([foreign])
 m4_ifdef([AM_SILENT_RULES],[AM_SILENT_RULES([yes])])
