diff -aur Python-3.1.2/setup.py fw_Python-3.1.2//setup.py
--- Python-3.1.2/setup.py	2010-01-01 17:29:51.000000000 +0100
+++ fw_Python-3.1.2//setup.py	2010-07-05 20:16:45.081001449 +0200
@@ -912,7 +912,7 @@
                         if self.compiler.find_library_file(lib_dirs, 'ndbm'):
                             ndbm_libs = ['ndbm']
                         else:
-                            ndbm_libs = []
+                            ndbm_libs = ['gdbm']
                         print("building dbm using ndbm")
                         dbmext = Extension('_dbm', ['_dbmmodule.c'],
                                            define_macros=[
