diff -Naur unix.orig/tclUnixNotfy.c unix/tclUnixNotfy.c
--- unix.orig/tclUnixNotfy.c	2005-06-01 20:12:21.000000000 +0200
+++ unix/tclUnixNotfy.c	2005-06-08 23:20:35.000000000 +0200
@@ -275,7 +275,7 @@
      */
 
     if (notifierCount == 0) {
-	int result;
+	int result, ignored;
 	if (triggerPipe < 0) {
 	    panic("Tcl_FinalizeNotifier: notifier pipe not initialized");
 	}
@@ -294,7 +294,7 @@
 	close(triggerPipe);
 
 	Tcl_ConditionWait(&notifierCV, &notifierMutex, NULL);
-	result = Tcl_JoinThread(notifierThread);
+	result = Tcl_JoinThread(notifierThread, &ignored);
 	if (result) {
 	    Tcl_Panic("Tcl_FinalizeNotifier: unable to join notifier thread");
 	}
