diff -ur a/src/MurmurHash/MurmurHash3.cpp b/src/MurmurHash/MurmurHash3.cpp
--- a/src/MurmurHash/MurmurHash3.cpp	2012-05-08 08:14:00.128765772 -0500
+++ b/src/MurmurHash/MurmurHash3.cpp	2012-05-08 08:14:16.623765152 -0500
@@ -29,7 +29,7 @@
 
 #else	// defined(_MSC_VER)
 
-#define	FORCE_INLINE __attribute__((always_inline))
+#define	FORCE_INLINE __attribute__((always_inline)) inline
 
 inline uint32_t rotl32 ( uint32_t x, int8_t r )
 {
