diff --git a/src/sisusb_driver.c b/src/sisusb_driver.c
index 539b9f1..0fdc8e9 100644
--- a/src/sisusb_driver.c
+++ b/src/sisusb_driver.c
@@ -39,7 +39,6 @@
 #include "shadowfb.h"
 #include "micmap.h"
 #include "mipointer.h"
-#include "mibstore.h"
 
 #include "sisusb_regs.h"
 #include "sisusb_dac.h"
@@ -1907,7 +1906,6 @@ SISUSBScreenInit(SCREEN_INIT_ARGS_DECL)
     /* Initialize the accelerators */
     SiSUSBAccelInit(pScreen);
 
-    miInitializeBackingStore(pScreen);
     xf86SetBackingStore(pScreen);
 
     /* NO SilkenMouse. Never. Ever. */
