Index: kimono-4.7.3/soprano/CMakeLists.txt
===================================================================
--- kimono-4.7.3.orig/soprano/CMakeLists.txt
+++ kimono-4.7.3/soprano/CMakeLists.txt
@@ -22,7 +22,7 @@ ADD_DEPENDENCIES(soprano-sharp smokesopr
 TARGET_LINK_LIBRARIES(soprano-sharp ${SMOKE_SOPRANO_LIBRARY}
     ${SMOKE_BASE_LIBRARY}
     ${QYOTO_LIBRARY}
-    ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${SOPRANO_LIBRARIES} ${SOPRANO_INDEX_LIBRARIES}
+    ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} ${SOPRANO_LIBRARIES}
 	${SOPRANO_CLIENT_LIBRARIES} ${SOPRANO_SERVER_LIBRARIES})
 
 CSHARP_ADD_LIBRARY(soprano "${SRC_CS}" REFERENCES ${QYOTO_ASSEMBLY} COMPILE_FLAGS /warn:0 /keyfile:${QYOTO_KEYFILE})