--- gnomeartng-0.6.0old/src/CConfiguration.cs	2008-07-15 15:48:44.000000000 +0200
+++ gnomeartng-0.6.0/src/CConfiguration.cs	2008-07-16 21:26:22.000000000 +0200
@@ -281,24 +281,10 @@
 		
 		private void setDistributionDependendSettings(){
 			getDistribution();
-			gdmFile="gdm.conf";
-			gdmCustomFile="gdm.conf-custom";
+			gdmFile="custom.conf";
+			gdmCustomFile="custom.conf";
+			sudoCommand="gksudo"; 
 			attribPrep="";
-			switch (distribution) {
-			 case DistriType.dtKubuntu: 
-				sudoCommand="kdesu";
-				break; 
-			 case DistriType.dtUbuntu: 
-				sudoCommand="gksudo"; 
-				break;
-			 case DistriType.dtSuse:
-				sudoCommand="gnomesu"; 
-				gdmFile="custom.conf";
-				gdmCustomFile="custom.conf";
-				attribPrep="--command=";
-				break;
-			 default: throw new Exception("Unknown distribution...aborting!!");
-			 }
 			gdmPath="/etc/gdm/";
 		}
 		
