--- gnomeartng-0.7.0/src/CConfiguration.cs	2009-07-16 17:56:09.000000000 +0200
+++ gnomeartng-0.7.0new/src/CConfiguration.cs	2009-07-17 10:48:03.000000000 +0200
@@ -76,6 +76,7 @@
 			Suse,
 			Fedora,
 			PcLinuxOs,
+			Frugalware,
 			Unknown
 		}
 		
@@ -319,6 +320,11 @@
 			 case DistriType.PcLinuxOs: 
 				sudoCommand="gksu"; 
 				break;
+			 case DistriType.Frugalware:
+				gdmFile="custom.conf";
+				gdmCustomFile="custom.conf";
+				sudoCommand="gksudo"; 
+				break;
 			 default: throw new Exception("Unknown distribution...aborting!!");
 			 }
 			gdmPath="/etc/gdm/";
@@ -330,6 +336,10 @@
 			distribution = DistriType.Unknown;
 			distributionVersion = DistriVersion.Unknown;
 
+			//Frugalware 
+			if (sContent.IndexOf("frugalware",StringComparison.CurrentCulture)>-1)
+				distribution=DistriType.Frugalware;
+
 			//(K)Ubuntu, Suse
 			if (sContent.IndexOf("kubuntu",StringComparison.CurrentCulture)>-1)
 				distribution=DistriType.Kubuntu;
