--- gnomeartng-0.5.1/CFileDownloader.cs.old	2008-05-23 19:10:57.000000000 +0200
+++ gnomeartng-0.5.1/CFileDownloader.cs	2008-06-25 15:48:47.000000000 +0200
@@ -62,7 +62,9 @@
 			try	{
 				// Create a request to the file we are downloading
 				webRequest = (HttpWebRequest)WebRequest.Create(From);
-				webRequest.UserAgent="Gnome-Art NexGen";
+				//webRequest.UserAgent="Gnome-Art NexGen";
+				//Spoof the user-agent
+				webRequest.UserAgent="Mozilla/5.0 (X11; U; Linux i686; fr; rv:1.9b5) Gecko/2008041514 Firefox/3.0b5"; 
 				webRequest.Timeout = 4000; 
 				webRequest.KeepAlive=holdConnection;
 				// Set default authentication for retrieving the file
@@ -81,7 +83,8 @@
 					
 				// Open the URL for download
 				WebClient client = new WebClient ();
-				client.Headers.Add ("user-agent", "Gnome-Art NexGen");
+				//client.Headers.Add ("user-agent", "Gnome-Art NexGen");
+				client.Headers.Add ("user-agent", "Mozilla/5.0 (X11; U; Linux i686; fr; rv:1.9b5) Gecko/2008041514 Firefox/3.0b5");				
 				strResponse = client.OpenRead (From);
 
 				// Create a new file stream where we will be saving the data (local drive)
