Only in b: a
diff -ur a/imap/imapd.dist.in b/imap/imapd.dist.in
--- a/imap/imapd.dist.in	2012-07-29 08:13:45.812969180 -0500
+++ b/imap/imapd.dist.in	2012-07-29 08:16:44.096059622 -0500
@@ -178,13 +178,13 @@
 # with or without TLS then just comment this out, and add AUTH=PLAIN to
 # IMAP_CAPABILITY
 
-IMAP_CAPABILITY_TLS="$IMAP_CAPABILITY AUTH=PLAIN"
+IMAP_CAPABILITY_TLS="IMAP4rev1 UIDPLUS CHILDREN NAMESPACE THREAD=ORDEREDSUBJECT THREAD=REFERENCES SORT QUOTA IDLE AUTH=PLAIN"
 
 ##NAME: IMAP_TLS_ORIG:0
 #
 # For use by webadmin
 
-IMAP_CAPABILITY_TLS_ORIG="$IMAP_CAPABILITY_ORIG AUTH=PLAIN"
+IMAP_CAPABILITY_TLS_ORIG="IMAP4rev1 UIDPLUS CHILDREN NAMESPACE THREAD=ORDEREDSUBJECT THREAD=REFERENCES SORT QUOTA AUTH=CRAM-MD5 AUTH=CRAM-SHA1 AUTH=CRAM-SHA256 IDLE AUTH=PLAIN"
 
 ##NAME: IMAP_DISABLETHREADSORT:0
 #
diff -ur a/imap/imapd-ssl.dist.in b/imap/imapd-ssl.dist.in
--- a/imap/imapd-ssl.dist.in	2012-07-29 08:13:45.810968180 -0500
+++ b/imap/imapd-ssl.dist.in	2012-07-29 09:01:37.328904047 -0500
@@ -74,11 +74,11 @@
 
 IMAPDSSLSTART=NO
 
-##NAME: IMAPDSTARTTLS:0
+##NAME: IMAP_STARTTLS:0
 #
 #  Whether or not to implement IMAP STARTTLS extension instead:
 
-IMAPDSTARTTLS=YES
+IMAP_STARTTLS=YES
 
 ##NAME: IMAP_TLS_REQUIRED:1
 #
@@ -130,13 +130,7 @@
 # DEFAULT VALUES:
 #
 # SSL23 (OpenSSL), or "TLS_1:TLS1:SSL3" (GnuTLS)
-
-##NAME: TLS_STARTTLS_PROTOCOL:0
-# 
-# TLS_STARTTLS_PROTOCOL is used instead of TLS_PROTOCOL for the IMAP STARTTLS
-# extension, as opposed to IMAP over SSL on port 993.
-#
-# It takes the same values for OpenSSL/GnuTLS as TLS_PROTOCOL
+TLS_PROTOCOL=""
 
 ##NAME: TLS_CIPHER_LIST:0
 #
diff -ur a/imap/pop3d-ssl.dist.in b/imap/pop3d-ssl.dist.in
--- a/imap/pop3d-ssl.dist.in	2012-07-29 08:13:45.810968180 -0500
+++ b/imap/pop3d-ssl.dist.in	2012-07-29 09:00:28.900709565 -0500
@@ -115,15 +115,7 @@
 # DEFAULT VALUES:
 #
 # SSL23 (OpenSSL), or "TLS_1:TLS1:SSL3" (GnuTLS)
-
-##NAME: TLS_STARTTLS_PROTOCOL:0
-# 
-# TLS_STARTTLS_PROTOCOL is used instead of TLS_PROTOCOL for the POP3 STARTTLS
-# extension, as opposed to POP3 over SSL on port 995.
-#
-# It takes the same values for OpenSSL/GnuTLS as TLS_PROTOCOL
-
-TLS_STARTTLS_PROTOCOL=TLS1
+TLS_PROTOCOL=TLS1
 
 ##NAME: TLS_CIPHER_LIST:0
 #
