diff -ur a/entry.c b/entry.c
--- a/entry.c	2012-03-08 04:22:29.000000000 -0600
+++ b/entry.c	2013-11-09 06:04:56.466004719 -0600
@@ -406,9 +406,9 @@
 	WCHAR *error_buf;
 	EFI_STATUS err;
 	EFI_LOADED_IMAGE *info;
-	CHAR16 *name, *options;
+	CHAR16 *name = NULL, *options;
 	UINT32 options_size;
-	char *cmdline;
+	char *cmdline = NULL;
 
 	InitializeLib(image, _table);
 	sys_table = _table;
diff -ur a/Makefile b/Makefile
--- a/Makefile	2012-03-08 04:22:29.000000000 -0600
+++ b/Makefile	2013-11-09 06:13:55.458984679 -0600
@@ -51,8 +51,8 @@
 INCDIR := /usr/include
 
 # gnuefi sometimes installs these under a gnuefi/ directory, and sometimes not
-CRT0 := $(shell find $(LIBDIR) -name crt0-efi-$(ARCH).o 2>/dev/null | tail -n1)
-LDSCRIPT := $(shell find $(LIBDIR) -name elf_$(ARCH)_efi.lds 2>/dev/null | tail -n1)
+CRT0 := $(LIBDIR)/crt0-efi-$(ARCH).o
+LDSCRIPT := $(LIBDIR)/elf_$(ARCH)_efi.lds
 
 CFLAGS=-I. -I$(INCDIR)/efi -I$(INCDIR)/efi/$(ARCH) \
 		-DEFI_FUNCTION_WRAPPER -fPIC -fshort-wchar -ffreestanding \
