Description: Don't run the engine for nspluginviewer.
Author: Fathi Boudra <fabo@debian.org>
Last-Update: 2008-05-01
--- a/src/engine.cpp
+++ b/src/engine.cpp
@@ -110,7 +110,8 @@ Engine::Engine()
 	    commandLine.contains("metacity") ||
 	    commandLine.contains("xfwm4") ||
 	    commandLine.contains("gnome-settings-daemon") ||
-	    commandLine.contains("gnome-panel"))
+	    commandLine.contains("gnome-panel") ||
+	    commandLine.contains("nspluginviewer"))
 	{
 		m_enabled = false;
 		qDebug() << "Disabling the GTK-Qt Theme Engine for" << commandLine;
