Description: Disables the Bonobo features
Origin: vendor
Author: Didier Raboud <didier@raboud.com>
Bug: http://bugs.debian.org/541451
Last-Update: 2009-08-21

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,15 +13,8 @@ ENDIF (NOT GTK_INCLUDE_DIR)
 MESSAGE(STATUS "Found GTK include dir: ${GTK_INCLUDE_DIR}")
 MESSAGE(STATUS "Found GTK library dir: ${GTK_LIB_DIR}")
 
-# Find Bonobo
-PKGCONFIG(libbonoboui-2.0 BONOBO_INCLUDE_DIR BONOBO_LIB_DIR BONOBO_LINK_FLAGS BONOBO_CFLAGS)
-IF (NOT BONOBO_INCLUDE_DIR)
-	MESSAGE(STATUS "bonoboui not found.  Some features of the theme engine will not work as intended.")
-ELSE (NOT BONOBO_INCLUDE_DIR)
-	MESSAGE(STATUS "Found Bonobo include dir: ${BONOBO_INCLUDE_DIR}")
-	MESSAGE(STATUS "Found Bonobo library dir: ${BONOBO_LIB_DIR}")
-	SET(BONOBO_DEFINITIONS -D HAVE_BONOBO)
-ENDIF (NOT BONOBO_INCLUDE_DIR)
+# Don't find Bonobo
+MESSAGE(STATUS "Bonobo has been disabled on because it is deprecated and pulls too much of Gnome.")
 
 # Find msgfmt
 SET(CMAKE_MODULE_PATH ${CMAKE_SOURCE_DIR}/CMake/Modules)
