Index: gtk-qt-engine-1.1+svn20080816/src/rcproperties.cpp
===================================================================
--- gtk-qt-engine-1.1+svn20080816.orig/src/rcproperties.cpp	2008-12-29 08:38:22.000000000 -0500
+++ gtk-qt-engine-1.1+svn20080816/src/rcproperties.cpp	2008-12-29 08:38:26.000000000 -0500
@@ -388,8 +388,12 @@
 	iconMappings += doIconMapping("gtk-goto-first", "actions/go-first.png");
 	iconMappings += doIconMapping("gtk-goto-last", "actions/go-last.png");
 	iconMappings += doIconMapping("gtk-goto-top", "actions/go-top.png");
+	iconMappings += doIconMapping("gtk-go-back-ltr", "actions/go-previous.png");
+	iconMappings += doIconMapping("gtk-go-back-rtl", "actions/go-next.png");
 	iconMappings += doIconMapping("gtk-go-back", "actions/go-previous.png");
 	iconMappings += doIconMapping("gtk-go-down", "actions/go-down.png");
+	iconMappings += doIconMapping("gtk-go-forward-ltr", "actions/go-next.png");
+	iconMappings += doIconMapping("gtk-go-forward-rtl", "actions/go-previous.png");
 	iconMappings += doIconMapping("gtk-go-forward", "actions/go-next.png");
 	iconMappings += doIconMapping("gtk-go-up", "actions/go-up.png");
 	iconMappings += doIconMapping("gtk-harddisk", "devices/drive-harddisk.png");
@@ -437,8 +441,8 @@
 	iconMappings += doIconMapping("gtk-select-all", "actions/edit-select-all.png");
 	iconMappings += doIconMapping("gtk-select-color", "actions/color-picker.png");
 	iconMappings += doIconMapping("gtk-select-font", "actions/format-font-size-more.png");
-	//iconMappings += doIconMapping("gtk-sort-ascending", "??");
-	//iconMappings += doIconMapping("gtk-sort-descending", "??");
+	iconMappings += doIconMapping("gtk-sort-ascending", "actions/view-sort-ascending");
+	iconMappings += doIconMapping("gtk-sort-descending", "actions/view-sort-descending");
 	iconMappings += doIconMapping("gtk-spell-check", "actions/tools-check-spelling.png");
 	iconMappings += doIconMapping("gtk-stop", "actions/process-stop.png");
 	iconMappings += doIconMapping("gtk-strikethrough", "actions/format-text-strikethrough.png");
