diff -ur a/util/MultiplayerCommon.cpp b/util/MultiplayerCommon.cpp
--- a/util/MultiplayerCommon.cpp	2012-07-18 00:17:10.000000000 -0500
+++ b/util/MultiplayerCommon.cpp	2013-01-01 08:01:00.154415323 -0600
@@ -411,7 +411,7 @@
 void Sleep(int ms)
 {
     boost::xtime t;
-    boost::xtime_get(&t, boost::TIME_UTC);
+    boost::xtime_get(&t, boost::TIME_UTC_);
     int ns_sum = t.nsec + ms * 1000000;
     const int NANOSECONDS_PER_SECOND = 1000000000;
     int delta_secs = ns_sum / NANOSECONDS_PER_SECOND;
