--- FreeOrion/CMakeLists.txt	2013-12-25 19:10:54.000000000 +0200
+++ FreeOrion.new/CMakeLists.txt	2013-12-25 15:39:59.000000000 +0200
@@ -93,6 +93,8 @@
 
 find_package(OGRE)
 
+find_package(Threads REQUIRED)
+
 configure_file(
     ${CMAKE_SOURCE_DIR}/ogre_plugins.cfg.in
     ${CMAKE_BINARY_DIR}/ogre_plugins.cfg
@@ -254,6 +256,7 @@
     )
 endif ()
 
+install(TARGETS freeorioncommon DESTINATION lib)
 ########################################
 # Packaging                            #
 ########################################
--- FreeOrion/parse/CMakeLists.txt	2013-06-30 21:19:06.000000000 +0300
+++ FreeOrion.new/parse/CMakeLists.txt	2013-12-25 15:04:15.000000000 +0200
@@ -55,3 +55,5 @@
 if (BUILD_TESTS)
     add_subdirectory(test)
 endif ()
+
+install(TARGETS freeorionparse DESTINATION lib)
