diff -Naur src-tarball/network/ClientNetworking.cpp src-tarball-p/network/ClientNetworking.cpp
--- src-tarball/network/ClientNetworking.cpp	2017-04-24 20:15:38.000000000 +0200
+++ src-tarball-p/network/ClientNetworking.cpp	2017-05-02 13:36:12.303711721 +0200
@@ -1,5 +1,6 @@
 #define WIN32_LEAN_AND_MEAN
 
+#include <boost/optional/optional.hpp>
 #include "ClientNetworking.h"
 
 #include "Message.h"
@@ -27,7 +28,6 @@
 #include <boost/bind.hpp>
 #include <boost/thread/condition.hpp>
 #include <boost/thread/thread.hpp>
-#include <boost/optional/optional.hpp>
 
 #include <thread>
 
