diff --git a/data/mono.pc.in b/data/mono.pc.in
index 01e0a9e..6da0960 100644
--- a/data/mono.pc.in
+++ b/data/mono.pc.in
@@ -7,8 +7,6 @@ sysconfdir=@sysconfdir@
 Name: Mono
 Description: Mono Runtime
 Version: @VERSION@
-## Commented out because SLE hides devel files in the SLE SDK,
-## which not all customers will have.
-#Requires: glib-2.0 gthread-2.0
+Requires: glib-2.0 gthread-2.0
 Libs: -L${libdir} @export_ldflags@ -lmono @libmono_ldflags@
 Cflags: -I${includedir} @libmono_cflags@
