diff -ur a/src/examples/ClientApp.java b/src/examples/ClientApp.java
--- a/src/examples/ClientApp.java	2014-02-04 21:00:58.838527541 -0600
+++ b/src/examples/ClientApp.java	2014-02-04 21:12:23.478502085 -0600
@@ -61,7 +61,7 @@
  * content as text and as rendered HTML.
  *
  * @author Sean C. Sullivan
- * @author Ortwin Glück
+ * @author Ortwin Glueck
  * @author Michael Becke
  */
 public class ClientApp {
diff -ur a/src/examples/TrivialApp.java b/src/examples/TrivialApp.java
--- a/src/examples/TrivialApp.java	2014-02-04 21:00:58.838527541 -0600
+++ b/src/examples/TrivialApp.java	2014-02-04 21:12:44.482501304 -0600
@@ -47,7 +47,7 @@
  * how to use the Jakarta HttpClient API.
  *
  * @author <a href="mailto:jsdever@apache.org">Jeff Dever</a>
- * @author Ortwin Glück
+ * @author Ortwin Glueck
  */
 public class TrivialApp
 {
diff -ur a/src/java/org/apache/commons/httpclient/auth/BasicScheme.java b/src/java/org/apache/commons/httpclient/auth/BasicScheme.java
--- a/src/java/org/apache/commons/httpclient/auth/BasicScheme.java	2014-02-04 21:00:58.840527541 -0600
+++ b/src/java/org/apache/commons/httpclient/auth/BasicScheme.java	2014-02-04 21:15:45.123494588 -0600
@@ -46,7 +46,7 @@
  * @author <a href="mailto:remm@apache.org">Remy Maucherat</a>
  * @author Rodney Waldhoff
  * @author <a href="mailto:jsdever@apache.org">Jeff Dever</a>
- * @author Ortwin Gl?ck
+ * @author Ortwin Glueck
  * @author Sean C. Sullivan
  * @author <a href="mailto:adrian@ephox.com">Adrian Sutton</a>
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
diff -ur a/src/java/org/apache/commons/httpclient/auth/DigestScheme.java b/src/java/org/apache/commons/httpclient/auth/DigestScheme.java
--- a/src/java/org/apache/commons/httpclient/auth/DigestScheme.java	2014-02-04 21:00:58.840527541 -0600
+++ b/src/java/org/apache/commons/httpclient/auth/DigestScheme.java	2014-02-04 21:15:45.125494588 -0600
@@ -67,7 +67,7 @@
  * @author <a href="mailto:remm@apache.org">Remy Maucherat</a>
  * @author Rodney Waldhoff
  * @author <a href="mailto:jsdever@apache.org">Jeff Dever</a>
- * @author Ortwin Gl?ck
+ * @author Ortwin Glueck
  * @author Sean C. Sullivan
  * @author <a href="mailto:adrian@ephox.com">Adrian Sutton</a>
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
diff -ur a/src/java/org/apache/commons/httpclient/auth/HttpAuthenticator.java b/src/java/org/apache/commons/httpclient/auth/HttpAuthenticator.java
--- a/src/java/org/apache/commons/httpclient/auth/HttpAuthenticator.java	2014-02-04 21:00:58.839527541 -0600
+++ b/src/java/org/apache/commons/httpclient/auth/HttpAuthenticator.java	2014-02-04 21:15:45.126494588 -0600
@@ -62,7 +62,7 @@
  * @author <a href="mailto:remm@apache.org">Remy Maucherat</a>
  * @author Rodney Waldhoff
  * @author <a href="mailto:jsdever@apache.org">Jeff Dever</a>
- * @author Ortwin Glï¿½ck
+ * @author Ortwin Glueck
  * @author Sean C. Sullivan
  * @author <a href="mailto:adrian@ephox.com">Adrian Sutton</a>
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
diff -ur a/src/java/org/apache/commons/httpclient/auth/NTLMScheme.java b/src/java/org/apache/commons/httpclient/auth/NTLMScheme.java
--- a/src/java/org/apache/commons/httpclient/auth/NTLMScheme.java	2014-02-04 21:00:58.840527541 -0600
+++ b/src/java/org/apache/commons/httpclient/auth/NTLMScheme.java	2014-02-04 21:15:45.128494588 -0600
@@ -43,7 +43,7 @@
  * @author <a href="mailto:remm@apache.org">Remy Maucherat</a>
  * @author Rodney Waldhoff
  * @author <a href="mailto:jsdever@apache.org">Jeff Dever</a>
- * @author Ortwin Gl???ck
+ * @author Ortwin Glueck
  * @author Sean C. Sullivan
  * @author <a href="mailto:adrian@ephox.com">Adrian Sutton</a>
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
diff -ur a/src/java/org/apache/commons/httpclient/ConnectMethod.java b/src/java/org/apache/commons/httpclient/ConnectMethod.java
--- a/src/java/org/apache/commons/httpclient/ConnectMethod.java	2014-02-04 21:00:58.843527540 -0600
+++ b/src/java/org/apache/commons/httpclient/ConnectMethod.java	2014-02-04 21:15:36.292494916 -0600
@@ -38,7 +38,7 @@
 /**
  * Establishes a tunneled HTTP connection via the CONNECT method.
  *
- * @author Ortwin Gl???ck
+ * @author Ortwin Glueck
  * @author dIon Gillard
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
  * @author <a href="mailto:oleg@ural.ru">Oleg Kalnichevski</a>
diff -ur a/src/java/org/apache/commons/httpclient/HttpClientError.java b/src/java/org/apache/commons/httpclient/HttpClientError.java
--- a/src/java/org/apache/commons/httpclient/HttpClientError.java	2014-02-04 21:00:58.842527540 -0600
+++ b/src/java/org/apache/commons/httpclient/HttpClientError.java	2014-02-04 21:15:36.297494916 -0600
@@ -33,7 +33,7 @@
 /**
  * Signals that an error has occurred.
  * 
- * @author Ortwin Gl?ck
+ * @author Ortwin Glueck
  * @version $Revision: 480424 $ $Date: 2006-11-29 06:56:49 +0100 (Wed, 29 Nov 2006) $
  * @since 3.0
  */
diff -ur a/src/java/org/apache/commons/httpclient/HttpClient.java b/src/java/org/apache/commons/httpclient/HttpClient.java
--- a/src/java/org/apache/commons/httpclient/HttpClient.java	2014-02-04 21:00:58.843527540 -0600
+++ b/src/java/org/apache/commons/httpclient/HttpClient.java	2014-02-04 21:15:36.298494916 -0600
@@ -48,7 +48,7 @@
  * @author <a href="mailto:rwaldhoff@apache.org">Rodney Waldhoff</a>
  * @author Sean C. Sullivan
  * @author <a href="mailto:dion@apache.org">dIon Gillard</a>
- * @author Ortwin Gl?ck
+ * @author Ortwin Glueck
  * @author <a href="mailto:becke@u.washington.edu">Michael Becke</a>
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
  * @author Sam Maloney
diff -ur a/src/java/org/apache/commons/httpclient/HttpContentTooLargeException.java b/src/java/org/apache/commons/httpclient/HttpContentTooLargeException.java
--- a/src/java/org/apache/commons/httpclient/HttpContentTooLargeException.java	2014-02-04 21:00:58.840527541 -0600
+++ b/src/java/org/apache/commons/httpclient/HttpContentTooLargeException.java	2014-02-04 21:15:36.302494916 -0600
@@ -3,7 +3,7 @@
 /**
  * Signals that the response content was larger than anticipated. 
  * 
- * @author Ortwin Glück
+ * @author Ortwin Glueck
  */
 public class HttpContentTooLargeException extends HttpException {
     private int maxlen;
diff -ur a/src/java/org/apache/commons/httpclient/methods/PostMethod.java b/src/java/org/apache/commons/httpclient/methods/PostMethod.java
--- a/src/java/org/apache/commons/httpclient/methods/PostMethod.java	2014-02-04 21:00:58.842527540 -0600
+++ b/src/java/org/apache/commons/httpclient/methods/PostMethod.java	2014-02-04 21:15:49.574494423 -0600
@@ -62,7 +62,7 @@
  * @author <a href="mailto:remm@apache.org">Remy Maucherat</a>
  * @author <a href="mailto:dsale@us.britannica.com">Doug Sale</a>
  * @author <a href="mailto:jsdever@apache.org">Jeff Dever</a>
- * @author Ortwin Gl???ck
+ * @author Ortwin Glueck
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
  * @author <a href="mailto:oleg@ural.ru">Oleg Kalnichevski</a>
  *
diff -ur a/src/java/org/apache/commons/httpclient/WireLogInputStream.java b/src/java/org/apache/commons/httpclient/WireLogInputStream.java
--- a/src/java/org/apache/commons/httpclient/WireLogInputStream.java	2014-02-04 21:00:58.840527541 -0600
+++ b/src/java/org/apache/commons/httpclient/WireLogInputStream.java	2014-02-04 21:15:36.336494915 -0600
@@ -37,7 +37,7 @@
 /**
  * Logs all data read to the wire LOG.
  *
- * @author Ortwin Glï¿½ck
+ * @author Ortwin Glueck
  * @author <a href="mailto:mbowler@GargoyleSoftware.com">Mike Bowler</a>
  * @author <a href="mailto:oleg@ural.ru">Oleg Kalnichevski</a>
  * 
diff -ur a/src/test/org/apache/commons/httpclient/TestHttps.java b/src/test/org/apache/commons/httpclient/TestHttps.java
--- a/src/test/org/apache/commons/httpclient/TestHttps.java	2014-02-04 21:00:58.846527540 -0600
+++ b/src/test/org/apache/commons/httpclient/TestHttps.java	2014-02-04 21:07:20.532513349 -0600
@@ -49,7 +49,7 @@
  *    (see build.xml)
  *
  * @author Rodney Waldhoff
- * @author Ortwin Glück
+ * @author Ortwin Glueck
  * @version $Id: TestHttps.java 480424 2006-11-29 05:56:49Z bayard $
  */
 public class TestHttps extends TestCase {
diff -ur a/src/test/org/apache/commons/httpclient/TestURIUtil2.java b/src/test/org/apache/commons/httpclient/TestURIUtil2.java
--- a/src/test/org/apache/commons/httpclient/TestURIUtil2.java	2014-02-04 21:00:58.845527540 -0600
+++ b/src/test/org/apache/commons/httpclient/TestURIUtil2.java	2014-02-04 21:07:02.685514013 -0600
@@ -37,7 +37,7 @@
 /**
  * Tests the util.URIUtil class.
  *
- * @author Ortwin Glück
+ * @author Ortwin Glueck
  *
  * @since 2.0
  */
