diff --git a/qc-driver.c b/qc-driver.c
index d303148..2f3a91b 100644
--- a/qc-driver.c
+++ b/qc-driver.c
@@ -49,8 +49,8 @@
 #include <linux/quickcam.h>
 #endif
 #include <linux/version.h>
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,18)) || (HAVE_UTSRELEASE_H == 1)
-#include <linux/utsrelease.h>
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,18))
+#include <generated/utsrelease.h>
 #endif
 #include <linux/module.h>
 
diff --git a/qc-memory.c b/qc-memory.c
index 39a95f5..aaf74d7 100644
--- a/qc-memory.c
+++ b/qc-memory.c
@@ -27,7 +27,7 @@
  */
 /* }}} */
 /* {{{ [fold] Includes */
-#include <linux/autoconf.h>
+#include <generated/autoconf.h>
 #include <linux/version.h>
 
 #ifdef CONFIG_SMP
diff --git a/quickcam.h b/quickcam.h
index 2b54bf2..2c8014d 100644
--- a/quickcam.h
+++ b/quickcam.h
@@ -76,7 +76,7 @@ struct qc_userlut {
 
 #ifdef __KERNEL__
 
-#include <linux/autoconf.h>
+#include <generated/autoconf.h>
 #include <linux/version.h>
 
 #ifdef CONFIG_SMP
