diff --git a/qc-driver.c b/qc-driver.c
index 2f3a91b..7bedad6 100644
--- a/qc-driver.c
+++ b/qc-driver.c
@@ -2518,7 +2518,7 @@ fail:	if (qcdebug&QC_DEBUGMUTEX) PDEBUG("up(%p) in qc_v4l_mmap() : %i", qc, sem_
 /* }}} */
 /* {{{ [fold] qc_v4l_ioctl(struct video_device *dev, unsigned int cmd, void *arg) */
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,0)
-static int qc_v4l_ioctl(struct inode *inode, struct file *file, unsigned int cmd, unsigned long arg)
+static long qc_v4l_ioctl(struct file *file, unsigned int cmd, unsigned long arg)
 #else
 static int qc_v4l_ioctl(struct video_device *dev, unsigned int cmd, void *argp)
 #endif
@@ -3021,7 +3021,7 @@ static struct file_operations qc_v4l_fops = {
 	release:	qc_v4l_close,
 	read:		qc_v4l_read,
 //	write:		qc_v4l_write,
-	ioctl:		qc_v4l_ioctl,
+	unlocked_ioctl:	qc_v4l_ioctl,
 	mmap:		qc_v4l_mmap,
 	poll:		qc_v4l_poll,
 };
