diff -Naur qc-usb-0.6.4.orig/qc-driver.c qc-usb-0.6.4/qc-driver.c
--- qc-usb-0.6.4.orig/qc-driver.c	2006-05-06 04:51:50.000000000 +0200
+++ qc-usb-0.6.4/qc-driver.c	2006-09-21 13:29:59.000000000 +0200
@@ -58,6 +58,9 @@
 #include <linux/malloc.h>
 #endif
 #include <linux/kernel.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,18)
+#include <linux/utsrelease.h>
+#endif
 #include <linux/init.h>
 #include <linux/sched.h>
 #include <linux/list.h>
diff -Naur qc-usb-0.6.4.orig/quickcam.h qc-usb-0.6.4/quickcam.h
--- qc-usb-0.6.4.orig/quickcam.h	2006-05-06 05:00:26.000000000 +0200
+++ qc-usb-0.6.4/quickcam.h	2006-09-21 13:30:39.000000000 +0200
@@ -93,6 +93,9 @@
 #endif
 
 #include <linux/videodev.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,18)
+#include <media/v4l2-dev.h>
+#endif
 #include <linux/usb.h>
 #include <asm/pgtable.h>		/* This is required for testing pte_offset_map */
 #include <linux/spinlock.h>
