diff -Naur file-4.10.orig/magic/Magdir/zisofs file-4.10/magic/Magdir/zisofs
--- file-4.10.orig/magic/Magdir/zisofs	1970-01-01 01:00:00.000000000 +0100
+++ file-4.10/magic/Magdir/zisofs	2004-07-26 10:50:09.000000000 +0200
@@ -0,0 +1,8 @@
+
+#------------------------------------------------------------------------------
+# zisofs: file(1) magic for zisofs/RockRidge compressed files
+#
+# from H. Peter Anvin <hpa@zytor.com> May 4, 2001
+#
+0	string	\x37\xE4\x53\x96\xC9\xDB\xD6\x07	zisofs/Rockridge compressed file
+>8	lelong		x		- %d bytes
diff -Naur file-4.10.orig/magic/Makefile.in file-4.10/magic/Makefile.in
--- file-4.10.orig/magic/Makefile.in	2004-07-24 22:38:55.000000000 +0200
+++ file-4.10/magic/Makefile.in	2004-07-26 10:50:17.000000000 +0200
@@ -306,6 +306,7 @@
 Magdir/xenix \
 Magdir/xwindows \
 Magdir/zilog \
+Magdir/zisofs \
 Magdir/zyxel
 
 subdir = magic
