diff -Naur file-4.10.orig/src/apprentice.c file-4.10/src/apprentice.c
--- file-4.10.orig/src/apprentice.c	2004-07-24 22:38:56.000000000 +0200
+++ file-4.10/src/apprentice.c	2004-07-26 10:56:50.000000000 +0200
@@ -165,8 +165,8 @@
 	}
 #ifndef COMPILE_ONLY
 	if ((rv = apprentice_map(ms, &magic, &nmagic, fn)) == -1) {
-		if (ms->flags & MAGIC_CHECK)
-			file_magwarn("using regular magic file `%s'", fn);
+		/* if (ms->flags & MAGIC_CHECK)
+			file_magwarn("using regular magic file `%s'", fn); */
 		rv = apprentice_file(ms, &magic, &nmagic, fn, action);
 		if (rv != 0)
 			return -1;
