--- kiax-0.8.51-src/lib/libiax2/src/iax.c.orig	2008-08-15 19:12:59.000000000 +0200
+++ kiax-0.8.51-src/lib/libiax2/src/iax.c	2008-08-15 19:13:27.000000000 +0200
@@ -1553,7 +1553,7 @@
 		snprintf(iax_errstr, sizeof(iax_errstr), "Invalid hostname: %s", tmp);
 		return -1;
 	}
-	memcpy(&session->peeraddr.sin_addr, hp->h_addr, sizeof(session->peeraddr.sin_addr));
+	memcpy(&session->peeraddr.sin_addr, hp->h_addr_list[0], sizeof(session->peeraddr.sin_addr));
 	session->peeraddr.sin_port = htons(portno);
 	session->peeraddr.sin_family = AF_INET;
 	strncpy(session->username, peer, sizeof(session->username) - 1);
@@ -1946,7 +1946,7 @@
 		snprintf(iax_errstr, sizeof(iax_errstr), "Invalid hostname: %s", hostname);
 		return -1;
 	}
-	memcpy(&session->peeraddr.sin_addr, hp->h_addr, sizeof(session->peeraddr.sin_addr));
+	memcpy(&session->peeraddr.sin_addr, hp->h_addr_list[0], sizeof(session->peeraddr.sin_addr));
 	session->peeraddr.sin_port = htons(portno);
 	session->peeraddr.sin_family = AF_INET;
 	res = send_command(session, AST_FRAME_IAX, IAX_COMMAND_NEW, 0, ied.buf, ied.pos, -1);
