--- irssi-python-svn4430/src/objects/connect-object.c.orig	2007-10-19 13:33:12.000000000 +0200
+++ irssi-python-svn4430/src/objects/connect-object.c	2007-10-19 13:39:16.000000000 +0200
@@ -112,6 +112,15 @@
     RET_AS_STRING_OR_NONE(self->data->realname);
 }
 
+PyDoc_STRVAR(PyConnect_use_ssl_doc,
+    "If we use ssl"
+);
+static PyObject *PyConnect_use_ssl_get(PyConnect *self, void *closure)
+{
+    RET_NULL_IF_INVALID(self->data);
+    return PyBool_FromLong(self->data->use_ssl);
+}
+
 /* Get/Set */
 static PyGetSetDef PyConnect_getseters[] = {
     {"address", (getter)PyConnect_address_get, NULL,
@@ -128,6 +137,8 @@
         PyConnect_username_doc, NULL},
     {"realname", (getter)PyConnect_realname_get, NULL,
         PyConnect_realname_doc, NULL},
+    {"use_ssl", (getter)PyConnect_use_ssl_get, NULL,
+        PyConnect_use_ssl_doc, NULL},
     {NULL}
 };
 
