--- wormux-0.7/configure.orig	2006-04-04 23:41:28.000000000 +0200
+++ wormux-0.7/configure	2006-04-04 23:43:34.000000000 +0200
@@ -6752,9 +6752,7 @@
         GCC_FLAGS="$GCC_FLAGS -Wno-unused-parameter"
 
     if test "${debug}" = "yes"; then
-        GCC_FLAGS="$GCC_FLAGS -O0 -g -DDEBUG"
-    else
-        GCC_FLAGS="$GCC_FLAGS -O2"
+        GCC_FLAGS="$GCC_FLAGS -DDEBUG"
     fi
 
     CFLAGS="$CFLAGS $GCC_FLAGS"
