--- wormux-0.7.orig/src/graphic/color.h	2006-04-04 23:20:52.000000000 +0200
+++ wormux-0.7/src/graphic/color.h	2006-04-04 23:21:35.000000000 +0200
@@ -37,7 +37,7 @@
 		Color();
 		Color(Uint8 r, Uint8 g, Uint8 b, Uint8 a);
 
-		bool Color::operator==(const Color &color) const;
+		bool operator==(const Color &color) const;
 
 		void SetColor(Uint8 r, Uint8 g, Uint8 b, Uint8 a);
 
--- wormux-0.7.orig/src/graphic/surface.h	2006-04-04 23:20:52.000000000 +0200
+++ wormux-0.7/src/graphic/surface.h	2006-04-04 23:23:06.000000000 +0200
@@ -52,7 +52,7 @@
 			void SetAutoFree(bool newAutoFree);
 
 			void SetSurface(SDL_Surface *newSurface, bool freePrevious = true);
-			void Surface::NewSurface(const Point2i &size, Uint32 flags, bool useAlpha = true);
+			void NewSurface(const Point2i &size, Uint32 flags, bool useAlpha = true);
 
 			SDL_Surface *GetSurface();
 
--- wormux-0.7.orig/src/game/game_loop.h	2006-04-04 23:20:52.000000000 +0200
+++ wormux-0.7/src/game/game_loop.h	2006-04-04 23:24:28.000000000 +0200
@@ -72,10 +72,10 @@
 private:
   GameLoop();
 
-  void GameLoop::InitGameData_NetServer();
-  void GameLoop::InitGameData_NetClient();
-  void GameLoop::InitData_Local();
-  void GameLoop::InitData();
+  void InitGameData_NetServer();
+  void InitGameData_NetClient();
+  void InitData_Local();
+  void InitData();
     
   void RefreshClock();
   void CallDraw();
--- wormux-0.7.orig/src/weapon/air_attack.h	2006-04-04 23:25:53.000000000 +0200
+++ wormux-0.7/src/weapon/air_attack.h	2006-04-04 23:26:45.000000000 +0200
@@ -69,7 +69,7 @@
   bool PeutLacherObus() const;
   int LitCibleX() const;
   int GetDirection() const;
-  void Avion::SignalGhostState (bool was_dead);
+  void SignalGhostState (bool was_dead);
 };
 
 class AirAttack : public Weapon
--- wormux-0.7.orig/src/weapon/cluster_bomb.h	2006-04-04 23:25:53.000000000 +0200
+++ wormux-0.7/src/weapon/cluster_bomb.h	2006-04-04 23:28:04.000000000 +0200
@@ -72,7 +72,7 @@
   ClusterLauncher();
 
  private:
-  ClusterBombConfig& ClusterLauncher::cfg() ;
+  ClusterBombConfig& cfg() ;
 };
 
 #endif
--- wormux-0.7.orig/src/gui/box.cpp	2006-04-05 00:39:01.000000000 +0200
+++ wormux-0.7/src/gui/box.cpp	2006-04-05 00:40:03.000000000 +0200
@@ -56,7 +56,6 @@
   for (it = widgets.begin(); 
        it != widgets.end(); 
        ++it){
-    assert(it != NULL);
     if( (*it)->Clic(mousePosition, button) )
       return true;
   }
@@ -106,8 +105,6 @@
        it != widgets.end(); 
        ++it ){
 
-    assert(it != NULL);
-
     if( it == widgets.begin() )
       _y += border.y - margin;
 
@@ -154,7 +151,6 @@
   for( it = widgets.begin(); 
        it != widgets.end(); 
        ++it ){
-    assert(it != NULL);   
 
     if( it == widgets.begin() )
       _x += border.x - margin;
