diff -Naur plasma-desktop-5.17.1/kcms/fonts/package/contents/ui/main.qml plasma-desktop-5.17.1-p/kcms/fonts/package/contents/ui/main.qml
--- plasma-desktop-5.17.1/kcms/fonts/package/contents/ui/main.qml	2019-10-22 17:34:22.000000000 +0200
+++ plasma-desktop-5.17.1-p/kcms/fonts/package/contents/ui/main.qml	2019-10-26 21:08:07.239109349 +0200
@@ -236,8 +236,10 @@
                     enabled: dpiCheckBox.checked
                     value: kcm.fontAASettings.dpi !== 0 ? kcm.fontAASettings.dpi : 96
                     onValueModified: kcm.fontAASettings.dpi = value
-                    to: 1000
-                    from: 1
+                    // to: need to divide to stepSize
+                    to: 1008
+					// lowest vaue here can be == stepSize, that is because 0 means off
+                    from: 24
                 }
             }
 
