--- plasma-desktop-5.22.2.1/kcms/CMakeLists.txt	2021-06-22 21:05:40.000000000 +0200
+++ plasma-desktop-5.22.2.1.new/kcms/CMakeLists.txt	2021-06-25 17:06:37.426368584 +0200
@@ -16,7 +16,6 @@
 add_subdirectory( landingpage )
 add_subdirectory( keys )
 add_subdirectory( ksmserver )
-add_subdirectory( users )
 
 if (CMAKE_SYSTEM_NAME MATCHES Linux)
    # this one doesn't seem to be very portable, Alex
