diff -Naur plasma-desktop-5.20.0/kcms/CMakeLists.txt plasma-desktop-5.20.0-p/kcms/CMakeLists.txt
--- plasma-desktop-5.20.0/kcms/CMakeLists.txt	2020-10-08 18:20:24.000000000 +0200
+++ plasma-desktop-5.20.0-p/kcms/CMakeLists.txt	2020-10-15 00:33:01.664441906 +0200
@@ -17,7 +17,8 @@
 add_subdirectory( keys )
 add_subdirectory( ksmserver )
 add_subdirectory( nightcolor )
-add_subdirectory( users )
+# broken s*it
+#add_subdirectory( users )
 
 add_subdirectory( hardware )
 add_subdirectory( desktoppaths )
