diff -Naur plasma-desktop-5.6.2/knetattach/org.kde.knetattach.desktop plasma-desktop-5.6.2-p/knetattach/org.kde.knetattach.desktop
--- plasma-desktop-5.6.2/knetattach/org.kde.knetattach.desktop	2016-04-05 15:52:13.000000000 +0200
+++ plasma-desktop-5.6.2-p/knetattach/org.kde.knetattach.desktop	2016-04-08 23:02:50.148984860 +0200
@@ -177,7 +177,7 @@
 GenericName[zh_TW]=網路資料夾精靈
 Exec=knetattach
 InitialPreference=6
-Icon=knetattach
+Icon=preferences-system-network-sharing
 Type=Application
 Terminal=false
 X-KDE-StartupNotify=true
