diff -Naur plasma-desktop-5.20.0/kcms/ksmserver/smserversettings.kcfg plasma-desktop-5.20.0-p/kcms/ksmserver/smserversettings.kcfg
--- plasma-desktop-5.20.0/kcms/ksmserver/smserversettings.kcfg	2020-10-08 18:20:24.000000000 +0200
+++ plasma-desktop-5.20.0-p/kcms/ksmserver/smserversettings.kcfg	2020-10-15 03:38:04.689515824 +0200
@@ -24,7 +24,7 @@
         <choice name="emptySession"/>
       </choices>
       <label>On login</label>
-      <default>restorePreviousLogout</default>
+      <default>emptySession</default>
     </entry>
     <entry name="excludeApps" type="String">
       <label>Applications to be excluded from session</label>
