diff -Naur plasma-desktop-5.10.0/kcms/icons/icons.cpp plasma-desktop-5.10.0-p/kcms/icons/icons.cpp
--- plasma-desktop-5.10.0/kcms/icons/icons.cpp	2017-05-26 11:04:20.000000000 +0200
+++ plasma-desktop-5.10.0-p/kcms/icons/icons.cpp	2017-06-03 17:54:29.493662051 +0200
@@ -179,7 +179,7 @@
     mDefaultEffect[1].color2 = QColor(0,0,0);
     mDefaultEffect[2].color2 = QColor(0,0,0);
 
-    const int defDefSizes[] = { 32, 22, 22, 16, 48, 32 };
+    const int defDefSizes[] = { 32, 22, 22, 16, 256, 32 };
 
     KIconLoader::Group i;
     QStringList::ConstIterator it;
