diff -Naur plasma-desktop-5.9.2/applets/kicker/package/contents/config/main.xml plasma-desktop-5.9.2-p/applets/kicker/package/contents/config/main.xml
--- plasma-desktop-5.9.2/applets/kicker/package/contents/config/main.xml	2017-02-14 14:25:00.000000000 +0100
+++ plasma-desktop-5.9.2-p/applets/kicker/package/contents/config/main.xml	2017-02-20 00:25:08.602177511 +0100
@@ -8,7 +8,7 @@
   <group name="General">
     <entry name="icon" type="String">
       <label>The name of the icon to use for the compact representation (e.g. on a small panel).</label>
-      <default>start-here-kde</default>
+      <default>kde</default>
     </entry>
     <entry name="useCustomButtonImage" type="Bool">
       <label>Whether to use a custom image instead of an icon in the compact representation (e.g. on a small panel).</label>
diff -Naur plasma-desktop-5.9.2/applets/kicker/package/metadata.desktop plasma-desktop-5.9.2-p/applets/kicker/package/metadata.desktop
--- plasma-desktop-5.9.2/applets/kicker/package/metadata.desktop	2017-02-14 14:25:00.000000000 +0100
+++ plasma-desktop-5.9.2-p/applets/kicker/package/metadata.desktop	2017-02-20 00:25:23.949568044 +0100
@@ -90,7 +90,7 @@
 Comment[zh_TW]=以啟動器為基底的彈出式串連選單
 
 Type=Service
-Icon=start-here-kde
+Icon=kde
 X-KDE-ServiceTypes=Plasma/Applet
 
 X-Plasma-API=declarativeappletscript
diff -Naur plasma-desktop-5.9.2/applets/kickoff/package/contents/config/main.xml plasma-desktop-5.9.2-p/applets/kickoff/package/contents/config/main.xml
--- plasma-desktop-5.9.2/applets/kickoff/package/contents/config/main.xml	2017-02-14 14:25:00.000000000 +0100
+++ plasma-desktop-5.9.2-p/applets/kickoff/package/contents/config/main.xml	2017-02-20 00:25:40.360985703 +0100
@@ -16,7 +16,7 @@
     </entry>
     <entry name="icon" type="String">
       <label>The name of the icon used in the compact representation (e.g. on a small panel).</label>
-      <default>start-here-kde</default>
+      <default>kde</default>
     </entry>
     <entry name="favorites" type="StringList">
       <label>List of general favorites. Supported values are menu id's (usually .desktop file names), special URLs that expand into default applications (e.g. preferred://browser), document URLs and KPeople contact URIs.</label>
diff -Naur plasma-desktop-5.9.2/applets/kickoff/package/contents/ui/ConfigGeneral.qml plasma-desktop-5.9.2-p/applets/kickoff/package/contents/ui/ConfigGeneral.qml
--- plasma-desktop-5.9.2/applets/kickoff/package/contents/ui/ConfigGeneral.qml	2017-02-14 14:25:00.000000000 +0100
+++ plasma-desktop-5.9.2-p/applets/kickoff/package/contents/ui/ConfigGeneral.qml	2017-02-20 00:26:07.654680424 +0100
@@ -53,7 +53,7 @@
 
             KQuickAddons.IconDialog {
                 id: iconDialog
-                onIconNameChanged: cfg_icon = iconName || "start-here-kde" // TODO use actual default
+                onIconNameChanged: cfg_icon = iconName || "kde" // TODO use actual default
             }
 
             // just to provide some visual feedback, cannot have checked without checkable enabled
@@ -96,7 +96,7 @@
             PlasmaComponents.MenuItem {
                 text: i18nc("@item:inmenu Reset icon to default", "Clear Icon")
                 icon: "edit-clear"
-                onClicked: cfg_icon = "start-here-kde" // TODO reset to actual default
+                onClicked: cfg_icon = "kde" // TODO reset to actual default
             }
         }
     }
diff -Naur plasma-desktop-5.9.2/applets/kickoff/package/metadata.desktop plasma-desktop-5.9.2-p/applets/kickoff/package/metadata.desktop
--- plasma-desktop-5.9.2/applets/kickoff/package/metadata.desktop	2017-02-14 14:25:00.000000000 +0100
+++ plasma-desktop-5.9.2-p/applets/kickoff/package/metadata.desktop	2017-02-20 00:26:21.898043025 +0100
@@ -162,7 +162,7 @@
 Comment[x-test]=xxLauncher to start applicationsxx
 Comment[zh_CN]=应用程序的启动器
 Comment[zh_TW]=應用程式啟動器
-Icon=start-here-kde
+Icon=kde
 Type=Service
 X-KDE-ServiceTypes=Plasma/Applet
 
