diff -ru boost/function/function_template.hpp boost/function/function_template.hpp
--- boost/function/function_template.hpp	2009-05-10 10:31:29.000000000 +0100
+++ boost/function/function_template.hpp	2009-05-10 10:49:46.000000000 +0100
@@ -950,10 +950,10 @@
           f.vtable->manager(f.functor, this->functor,
                             boost::detail::function::move_functor_tag);
 		  f.vtable = 0;
-#if !defined(BOOST_NO_EXCEPTIONS)      
         } else {
           clear();
         }
+#if !defined(BOOST_NO_EXCEPTIONS)      
       } catch (...) {
         vtable = 0;
         throw;
