--- boost/function/function_template.hpp.orig	2008-12-19 10:32:27.000000000 +0000
+++ boost/function/function_template.hpp	2008-12-19 10:33:27.000000000 +0000
@@ -950,10 +950,10 @@
           f.vtable->manager(f.functor, this->functor,
                             boost::detail::function::move_functor_tag);
 		  f.vtable = 0;
-#if !defined(BOOST_NO_EXCEPTIONS)      
         } else {
           clear();
         }
+#if !defined(BOOST_NO_EXCEPTIONS)      
       } catch (...) {
         vtable = 0;
         throw;
