diff -Naur boost_1_34_0.orig/tools/jam/src/jam.h boost_1_34_0/tools/jam/src/jam.h
--- boost_1_34_0.orig/tools/jam/src/jam.h	2007-05-25 13:34:42.000000000 +0200
+++ boost_1_34_0/tools/jam/src/jam.h	2007-05-25 13:35:11.000000000 +0200
@@ -480,6 +480,10 @@
 # define OSPLAT "OSPLAT=IA64"
 # endif
 
+# if defined( __x86_64__ )
+# define OSPLAT "OSPLAT=x86_64"
+# endif
+
 # ifdef __s390__
 # define OSPLAT "OSPLAT=390"
 # endif
