diff -Naur boost_1_39_0.orig/Jamroot boost_1_39_0/Jamroot
--- boost_1_39_0.orig/Jamroot	2009-04-14 11:00:59.000000000 +0200
+++ boost_1_39_0/Jamroot	2009-06-28 22:55:02.000000000 +0200
@@ -344,7 +344,7 @@
         else
         {
             local result = [ common.format-name
-                <base> 
+                <base> <threading> 
                 -$(BUILD_ID)
                 : $(name) : $(type) : $(property-set) ] ;
 
@@ -359,7 +359,7 @@
               ( ! ( [ $(property-set).get <target-os> ] in windows cygwin darwin aix ) &&
                 ! ( [ $(property-set).get <toolset> ] in pgi ) )
             {
-                result = $(result).$(BOOST_VERSION)  ;
+#                result = $(result).$(BOOST_VERSION)  ;
             }
 
             return $(result) ;
