--- boost_1_58_0/boost/serialization/deque.hpp	2015-04-06 12:48:24.000000000 +0300
+++ boost_1_58_0.new/boost/serialization/deque.hpp	2015-05-21 04:01:12.000000000 +0300
@@ -24,6 +24,8 @@
 #include <boost/serialization/detail/stack_constructor.hpp>
 #include <boost/serialization/detail/is_default_constructible.hpp>
 #include <boost/serialization/split_free.hpp>
+#include <boost/serialization/type_info_implementation.hpp>
+#include <boost/archive/basic_archive.hpp>
 
 namespace boost { 
 namespace serialization {
