diff -Naur boost_1_34_0/Jamfile.v2 boost_1_34_0-p/Jamfile.v2
--- boost_1_34_0/Jamfile.v2	2007-05-11 05:40:34.000000000 +0200
+++ boost_1_34_0-p/Jamfile.v2	2007-05-13 19:13:52.000000000 +0200
@@ -312,7 +312,7 @@
     ;
 if $(layout-versioned)
 {
-    install-requirements += <install-header-subdir>boost-$(BOOST_VERSION_TAG)/boost ;
+    install-requirements += <install-header-subdir>boost ;
 }
 else
 {
