diff -Naur boost_1_88_0/boost/context/fiber_fcontext.hpp boost_1_88_0.new/boost/context/fiber_fcontext.hpp
--- boost_1_88_0/boost/context/fiber_fcontext.hpp	2025-04-03 13:37:24.000000000 +0200
+++ boost_1_88_0.new/boost/context/fiber_fcontext.hpp	2025-05-26 15:19:35.608734477 +0200
@@ -344,6 +344,8 @@
 
     ~fiber() {
         if ( BOOST_UNLIKELY( nullptr != fctx_) ) {
+            detail::manage_exception_state exstate;
+            boost::ignore_unused(exstate);
             detail::ontop_fcontext(
 #if defined(BOOST_NO_CXX14_STD_EXCHANGE)
                     detail::exchange( fctx_, nullptr),
