--- llvm-3.0.src/bindings/ocaml/llvm/Makefile	2011-10-14 20:38:02.000000000 +0000
+++ llvm-3.0.src.new/bindings/ocaml/llvm/Makefile	2012-03-08 02:30:43.000000000 +0000
@@ -30,7 +30,7 @@
 $(OcamlDir)/META.llvm: META.llvm
 	$(Verb) $(CP) -f $< $@
 
-install-meta:: $(ObjDir)/META.llvm
+install-meta:: $(ObjDir)/../META.llvm
 	$(Echo) "Install $(BuildMode) $(DestMETA)"
 	$(Verb) $(MKDIR) $(PROJ_libocamldir)
 	$(Verb) $(DataInstall) META.llvm "$(DestMETA)"
