diff -Naur llvm-2.6/CMakeLists.txt llvm-2.6-p/CMakeLists.txt
--- llvm-2.6/CMakeLists.txt	2009-08-18 17:29:35.000000000 +0200
+++ llvm-2.6-p/CMakeLists.txt	2010-01-28 14:24:30.000000000 +0100
@@ -4,7 +4,7 @@
 cmake_minimum_required(VERSION 2.6.1)
 
 set(PACKAGE_NAME llvm)
-set(PACKAGE_VERSION 2.6svn)
+set(PACKAGE_VERSION 2.6)
 set(PACKAGE_STRING "${PACKAGE_NAME} ${PACKAGE_VERSION}")
 set(PACKAGE_BUGREPORT "llvmbugs@cs.uiuc.edu")
 
