diff -Naur acetoneiso2/src/acetoneiso2.pro acetoneiso2-dos60-v2-fix/src/acetoneiso2.pro
--- acetoneiso2/src/acetoneiso2.pro	2007-11-05 15:07:36.000000000 +0100
+++ acetoneiso2-dos60-v2-fix/src/acetoneiso2.pro	2007-11-07 20:05:06.000000000 +0100
@@ -21,7 +21,7 @@
     menu.files = menu/*
     menu.path = /usr/share/applications
     dos.files = dos6.0/*
-    dos.path = /opt/acetoneiso/dos6.0
+    dos.path = /usr/share/acetoneiso/dos6.0
     dmg.files = dmg2img/dmg2img
     dmg.path = /usr/bin
     INSTALLS += target menuico menu dos dmg
diff -Naur acetoneiso2/src/acetoneiso.cpp acetoneiso2-dos60-v2-fix/src/acetoneiso.cpp
--- acetoneiso2/src/acetoneiso.cpp	2007-11-05 15:08:13.000000000 +0100
+++ acetoneiso2-dos60-v2-fix/src/acetoneiso.cpp	2007-11-07 20:06:09.000000000 +0100
@@ -885,7 +885,7 @@
 
 void acetoneiso::eldos()
 {
-QDir dosDir( "/opt/acetoneiso/dos6.0" );
+QDir dosDir( "/usr/share/acetoneiso/dos6.0" );
 QString fileName = QFileDialog::getSaveFileName(this, tr("AcetoneISO2::Save ISO file"),QDir::home().path() + "/acetone_bootable" ,tr("Images (*.iso)"));
 
 if(!fileName.isNull() )
