diff -Naur AcetoneISO2/src/acetoneiso.cpp AcetoneISO2-no-dos-kthx/src/acetoneiso.cpp
--- AcetoneISO2/src/acetoneiso.cpp	2007-10-24 19:42:13.000000000 +0200
+++ AcetoneISO2-no-dos-kthx/src/acetoneiso.cpp	2007-10-30 01:04:52.000000000 +0100
@@ -70,7 +70,7 @@
     connect( actionAbout, SIGNAL( triggered() ), this, SLOT( about() ) );
     connect( actionGeneric_Mount, SIGNAL( triggered() ), this, SLOT( gmount() ) );
     connect( actionGeneric_Umount, SIGNAL( triggered() ), this, SLOT( gumount() ) );
-    connect( actionDOS_Boot, SIGNAL( triggered() ), this, SLOT( eldos() ) );
+    /* connect( actionDOS_Boot, SIGNAL( triggered() ), this, SLOT( eldos() ) ); */
     connect( actionGeneric_Floppy_Emulation, SIGNAL( triggered() ), this, SLOT( elfloppy() ) );
     connect( actionGeneric_No_Emulation, SIGNAL( triggered() ), this, SLOT( eltorito() ) );
     connect( actionExtract_Boot_Image, SIGNAL( triggered() ), this, SLOT( getboot() ) );
@@ -882,9 +882,10 @@
 } 
 }
 /////////////////////////////////////////////
-
+/*
 void acetoneiso::eldos()
 {
+
 QDir dosDir( QDir::home().path() + "/.acetoneiso/dos6.0" );
 QString fileName = QFileDialog::getSaveFileName(this, tr("AcetoneISO2::Save ISO file"),QDir::home().path() + "/acetone_bootable" ,tr("Images (*.iso)"));
 
@@ -895,6 +896,9 @@
 	 progBarra();
 	}
 }
+*/
+
+/* guys that's broken */
 /////////////////////////////////////////////
 void acetoneiso::elfloppy()
 {
diff -Naur AcetoneISO2/src/acetoneiso.h AcetoneISO2-no-dos-kthx/src/acetoneiso.h
--- AcetoneISO2/src/acetoneiso.h	2007-10-24 19:05:24.000000000 +0200
+++ AcetoneISO2-no-dos-kthx/src/acetoneiso.h	2007-10-30 01:05:47.000000000 +0100
@@ -41,7 +41,7 @@
     void about();
     void gmount();
     void gumount();
-    void eldos();
+    /* void eldos(); */
     void elfloppy();
     void eltorito();
     void backupaudio();
diff -Naur AcetoneISO2/src/acetoneiso.ui AcetoneISO2-no-dos-kthx/src/acetoneiso.ui
--- AcetoneISO2/src/acetoneiso.ui	2007-10-24 19:06:26.000000000 +0200
+++ AcetoneISO2-no-dos-kthx/src/acetoneiso.ui	2007-10-30 01:33:25.000000000 +0100
@@ -1509,7 +1509,6 @@
      <property name="title" >
       <string>El Torito</string>
      </property>
-     <addaction name="actionDOS_Boot" />
      <addaction name="actionGeneric_Floppy_Emulation" />
      <addaction name="actionGeneric_No_Emulation" />
     </widget>
@@ -1598,14 +1597,6 @@
     <string>Check</string>
    </property>
   </action>
-  <action name="actionDOS_Boot" >
-   <property name="icon" >
-    <iconset resource="acetoneiso.qrc" >:/images/configure.png</iconset>
-   </property>
-   <property name="text" >
-    <string>DOS Boot</string>
-   </property>
-  </action>
   <action name="actionGeneric_Floppy_Emulation" >
    <property name="icon" >
     <iconset resource="acetoneiso.qrc" >:/images/configure.png</iconset>
diff -Naur AcetoneISO2/src/acetoneiso2.pro AcetoneISO2-no-dos-kthx/src/acetoneiso2.pro
--- AcetoneISO2/src/acetoneiso2.pro	2007-10-24 18:04:35.000000000 +0200
+++ AcetoneISO2-no-dos-kthx/src/acetoneiso2.pro	2007-10-30 01:04:52.000000000 +0100
@@ -20,11 +20,11 @@
     menuico.path = /usr/share/pixmaps
     menu.files = menu/*
     menu.path = /usr/share/applications
-    dos.files = dos6.0/*
-    dos.path = ~/.acetoneiso/dos6.0
+    #dos.files = dos6.0/*
+    #dos.path = ~/.acetoneiso/dos6.0
     dmg.files = dmg2img/dmg2img
     dmg.path = /usr/bin
-    INSTALLS += target menuico menu dos dmg
+    INSTALLS += target menuico menu dmg
 
  TRANSLATIONS = locale/acetoneiso_it.ts \
 		locale/acetoneiso_fr.ts \
