--- solid-5.21.0.old/src/solid/devices/frontend/devicemanager_p.h	2016-04-04 00:08:10.000000000 +0300
+++ solid-5.21.0.new/src/solid/devices/frontend/devicemanager_p.h	2016-04-15 04:02:42.751456230 +0300
@@ -29,6 +29,7 @@
 #include <QtCore/QPointer>
 #include <QtCore/QSharedData>
 #include <QtCore/QThreadStorage>
+#include <QtCore/QRegularExpression>
 
 namespace Solid
 {
--- solid-5.21.0.old/src/solid/devices/frontend/devicemanager.cpp	2016-04-04 00:08:10.000000000 +0300
+++ solid-5.21.0.new/src/solid/devices/frontend/devicemanager.cpp	2016-04-15 03:51:35.213357947 +0300
@@ -114,8 +114,24 @@
 
         QStringList udis = backend->devicesFromQuery(parentUdi, type);
 
-        Q_FOREACH (const QString &udi, udis) {
-            list.append(Device(udi));
+        if(type==DeviceInterface::Type::Processor) {
+            QMap<int, QString> coresMap;
+            QRegularExpression re("cpu([0-9]+)");
+            Q_FOREACH (const QString &udi, udis) {
+                QRegularExpressionMatch match = re.match(udi);
+                if(match.hasMatch()) {
+                     QString matched = match.captured(1);
+                     coresMap[matched.toUInt()] = udi;
+                }
+            }
+            for(QMap<int, QString>::const_iterator ii=coresMap.constBegin(); ii != coresMap.constEnd(); ii++) {
+                list.append(Device(ii.value()));
+            }
+        }
+        else  {
+            Q_FOREACH (const QString &udi, udis) {
+                list.append(Device(udi));
+            }
         }
     }
 
