diff -Naur SugarSuite-Full-3.5.1a.orig/acceptDecline.php SugarSuite-Full-3.5.1a/acceptDecline.php
--- SugarSuite-Full-3.5.1a.orig/acceptDecline.php	2005-10-09 11:31:00.000000000 +0200
+++ SugarSuite-Full-3.5.1a/acceptDecline.php	2005-10-28 23:07:20.000000000 +0200
@@ -21,10 +21,6 @@
  * Contributor(s): ______________________________________.
  ********************************************************************************/
 
-if (substr(phpversion(), 0, 1) == "5") {
-        ini_set("zend.ze1_compatibility_mode", "1");
-}
-
 include_once('config.php');
 //require_once('log4php/LoggerManager.php');
 require_once('include/database/PearDatabase.php');
diff -Naur SugarSuite-Full-3.5.1a.orig/export.php SugarSuite-Full-3.5.1a/export.php
--- SugarSuite-Full-3.5.1a.orig/export.php	2005-10-09 11:31:00.000000000 +0200
+++ SugarSuite-Full-3.5.1a/export.php	2005-10-28 23:07:38.000000000 +0200
@@ -21,10 +21,6 @@
  * Contributor(s): ______________________________________.
  ********************************************************************************/
 
-if (substr(phpversion(), 0, 1) == "5") {
-        ini_set("zend.ze1_compatibility_mode", "1");
-}
-
 include_once('config.php');
 require_once('log4php/LoggerManager.php');
 require_once('include/database/PearDatabase.php');
diff -Naur SugarSuite-Full-3.5.1a.orig/install.php SugarSuite-Full-3.5.1a/install.php
--- SugarSuite-Full-3.5.1a.orig/install.php	2005-10-09 11:31:00.000000000 +0200
+++ SugarSuite-Full-3.5.1a/install.php	2005-10-28 23:06:43.000000000 +0200
@@ -32,10 +32,6 @@
 
 session_start();
 
-if (substr(phpversion(), 0, 1) == "5"){
-   ini_set("zend.ze1_compatibility_mode", "1");
-}
-
 require_once('include/utils.php');
 
 $current_language = 'en_us';
diff -Naur SugarSuite-Full-3.5.1a.orig/json.php SugarSuite-Full-3.5.1a/json.php
--- SugarSuite-Full-3.5.1a.orig/json.php	2005-10-09 11:31:00.000000000 +0200
+++ SugarSuite-Full-3.5.1a/json.php	2005-10-28 23:08:40.000000000 +0200
@@ -21,12 +21,6 @@
  * Contributor(s): ______________________________________.
  ********************************************************************************/
 
-if (substr(phpversion(), 0, 1) == "5") {
-        ini_set("zend.ze1_compatibility_mode", "1");
-}
-
-
-
 include_once('config.php');
 require_once('log4php/LoggerManager.php');
 require_once('include/database/PearDatabase.php');
diff -Naur SugarSuite-Full-3.5.1a.orig/json_server.php SugarSuite-Full-3.5.1a/json_server.php
--- SugarSuite-Full-3.5.1a.orig/json_server.php	2005-10-09 11:31:00.000000000 +0200
+++ SugarSuite-Full-3.5.1a/json_server.php	2005-10-28 23:09:47.000000000 +0200
@@ -46,11 +46,6 @@
  */ 
 $SUPPORTED_METHODS = array('retrieve','query','set_accept_status');
 
-if (substr(phpversion(), 0, 1) == "5") {
-$GLOBALS['log']->debug("JSON_SERVER: setting zend.ze1_compatibility_mode");
-        ini_set("zend.ze1_compatibility_mode", "1");
-}
-
 // check for old config format.
 if(empty($sugar_config) && isset($dbconfig['db_host_name']))
 {
diff -Naur SugarSuite-Full-3.5.1a.orig/pdf.php SugarSuite-Full-3.5.1a/pdf.php
--- SugarSuite-Full-3.5.1a.orig/pdf.php	2005-10-09 11:31:00.000000000 +0200
+++ SugarSuite-Full-3.5.1a/pdf.php	2005-10-28 23:10:08.000000000 +0200
@@ -21,10 +21,6 @@
  * Contributor(s): ______________________________________.
  ********************************************************************************/
 
-if (substr(phpversion(), 0, 1) == "5") {
-        ini_set("zend.ze1_compatibility_mode", "1");
-}
-
 include_once('config.php');
 require_once('log4php/LoggerManager.php');
 require_once('include/database/PearDatabase.php');
diff -Naur SugarSuite-Full-3.5.1a.orig/soap.php SugarSuite-Full-3.5.1a/soap.php
--- SugarSuite-Full-3.5.1a.orig/soap.php	2005-10-09 11:31:00.000000000 +0200
+++ SugarSuite-Full-3.5.1a/soap.php	2005-10-28 23:10:24.000000000 +0200
@@ -21,10 +21,6 @@
  * Contributor(s): ______________________________________.
  ********************************************************************************/
 
-if (substr(phpversion(), 0, 1) == "5") {
-	ini_set("zend.ze1_compatibility_mode", "1");
-}
-
 ob_start();
 
 require_once('log4php/LoggerManager.php');
diff -Naur SugarSuite-Full-3.5.1a.orig/upgrade/index.php SugarSuite-Full-3.5.1a/upgrade/index.php
--- SugarSuite-Full-3.5.1a.orig/upgrade/index.php	2005-10-09 11:31:00.000000000 +0200
+++ SugarSuite-Full-3.5.1a/upgrade/index.php	2005-10-28 23:10:55.000000000 +0200
@@ -33,10 +33,6 @@
 
 session_start();
 
-if (substr(phpversion(), 0, 1) == "5"){
-	ini_set("zend.ze1_compatibility_mode", "1");
-}
-
 require_once( "upgrade/dir_inc.php" );
 if(!file_exists('config.php')){
 	// add something here
