--- bmpx-0.12.9/src/system_control.c~	2005-11-22 23:40:59.000000000 +0100
+++ bmpx-0.12.9/src/system_control.c	2006-01-03 19:47:50.000000000 +0100
@@ -1756,7 +1756,7 @@
 
     if (!dbus_g_proxy_call (bus_proxy, "RequestName", &error,
 			    G_TYPE_STRING, BMP_DBUS_INTERFACE,
-			    G_TYPE_UINT, DBUS_NAME_FLAG_PROHIBIT_REPLACEMENT,
+			    G_TYPE_UINT, DBUS_NAME_FLAG_DO_NOT_QUEUE,
 			    G_TYPE_INVALID,
 			    G_TYPE_UINT, &request_name_result,
 			    G_TYPE_INVALID))
