diff -urN bmpx-0.12.9/configure.ac bmpx-0.12.9.az/configure.ac
--- bmpx-0.12.9/configure.ac	2005-11-23 00:41:07.000000000 +0200
+++ bmpx-0.12.9.az/configure.ac	2005-12-02 13:13:26.000000000 +0200
@@ -404,13 +404,13 @@
 
 dnl XCS
 XCS_CFLAGS='-I$(top_builddir)/xcs'
-XCS_LIBS='$(top_builddir)/xcs/libxcs.a'
+XCS_LIBS='$(top_builddir)/xcs/libxcs.la'
 AC_SUBST([XCS_CFLAGS])
 AC_SUBST([XCS_LIBS])
 
 dnl GOA
 GOA_CFLAGS='-I$(top_builddir)/goa'
-GOA_LIBS='$(top_builddir)/goa/libgoa.a'
+GOA_LIBS='$(top_builddir)/goa/libgoa.la'
 AC_SUBST([GOA_CFLAGS])
 AC_SUBST([GOA_LIBS])
 
diff -urN bmpx-0.12.9/goa/Makefile.am bmpx-0.12.9.az/goa/Makefile.am
--- bmpx-0.12.9/goa/Makefile.am	2005-11-23 00:41:20.000000000 +0200
+++ bmpx-0.12.9.az/goa/Makefile.am	2005-12-02 13:12:01.000000000 +0200
@@ -1,16 +1,15 @@
-lib_LIBRARIES = libgoa.a
+lib_LTLIBRARIES = libgoa.la
 
 INCLUDES = \
 	$(GLIB_CFLAGS)
 
-AM_LDFLAGS = -all-static
-LIBADD = $(GLIB_LIBS)
+libgoa_la_LIBADD = $(GLIB_LIBS)
 
 BUILT_SOURCES = \
         libgoa.c \
         libgoa.h
 
-nodist_libgoa_a_SOURCES = \
+nodist_libgoa_la_SOURCES = \
 	libgoa.c \
 	libgoa.h
 
diff -urN bmpx-0.12.9/xcs/Makefile.am bmpx-0.12.9.az/xcs/Makefile.am
--- bmpx-0.12.9/xcs/Makefile.am	2005-11-23 00:41:22.000000000 +0200
+++ bmpx-0.12.9.az/xcs/Makefile.am	2005-12-02 13:12:01.000000000 +0200
@@ -1,16 +1,15 @@
-lib_LIBRARIES = libxcs.a
+noinst_LTLIBRARIES = libxcs.la
 
 INCLUDES = \
 	$(GLIB_CFLAGS) $(XML_CFLAGS) $(GTK_CFLAGS) $(XCS_DEFINES)
 
-AM_LDFLAGS = -all-static
-LIBADD = $(GLIB_LIBS) $(XML_LIBS)
+libxcs_la_LIBADD = $(GLIB_LIBS) $(XML_LIBS)
 
 if HAVE_GUI
-LIBADD += $(GTK_LIBS)
+libxcs_la_LIBADD += $(GTK_LIBS)
 endif
 
-libxcs_a_SOURCES = \
+libxcs_la_SOURCES = \
         xcs.c \
         xcs.h \
 	xml.c \
