diff -Naur mpeg2dec-0.4.0.orig/libmpeg2/cpu_accel.c mpeg2dec-0.4.0/libmpeg2/cpu_accel.c
--- mpeg2dec-0.4.0.orig/libmpeg2/cpu_accel.c	2005-07-29 17:41:59.000000000 +0200
+++ mpeg2dec-0.4.0/libmpeg2/cpu_accel.c	2005-07-29 17:44:42.000000000 +0200
@@ -30,14 +30,62 @@
 #include "mpeg2_internal.h"
 
 #ifdef ACCEL_DETECT
+
+static int has_cpuid() {
+    long a, c;
+    
+    /* Code from libavcodec */
+    __asm__ __volatile__ (
+	    /* See if CPUID instructions supported */
+	    /* Get copies of EFLAGS into eax and exc */
+	    "pushf\n\t"
+	    "pop %0\n\t"
+	    "mov %0, %1\n\t"
+	    /* Toggle the ID bit in one copy and store to the EFLAGS reg */
+	    "xor $0x200000, %0\n\t"
+	    "push %0\n\t"
+	    "popf\n\t"
+	    /* Get the (hopefully modified) EFLAGS */
+	    "pushf\n\t"
+	    "pop %0\n\t"
+	    : "=a" (a), "=c" (c)
+	    :
+	    : "cc"
+	);
+    return (a!=c);
+}
+
+#ifdef ARCH_X86_64 && __X86_64__
+#	define REG_b "rbx"
+#	define REG_S "rsi"
+#else
+#	define REG_b "ebx"
+#	define REG_S "esi"
+#endif
+
+static void cpuid(unsigned int ax, unsigned int *p) {
+    __asm__ __volatile__ 
+	    ("mov %%"REG_b", %%"REG_S"\n\t"
+		"cpuid\n\t"
+		"xchg %%"REG_b", %%"REG_S
+		: "=a" (p[0]), "=S" (p[1]),
+		  "=c" (p[2]), "=d" (p[3])
+		: "0" (ax));
+}
+
 #ifdef ARCH_X86
 static inline uint32_t arch_accel (void)
 {
-    uint32_t eax, ebx, ecx, edx;
+    unsigned int regs[4];
     int AMD;
     uint32_t caps;
 
-#if !defined(PIC) && !defined(__PIC__)
+#define eax regs[0]
+#define ebx regs[1]
+#define ecx regs[2]
+#define edx regs[3]
+
+/*#if !defined(PIC) && !defined(__PIC__)
 #define cpuid(op,eax,ebx,ecx,edx)	\
     __asm__ ("cpuid"			\
 	     : "=a" (eax),		\
@@ -46,7 +94,7 @@
 	       "=d" (edx)		\
 	     : "a" (op)			\
 	     : "cc")
-#else	/* PIC version : save ebx */
+#else	/* PIC version : save ebx 
 #define cpuid(op,eax,ebx,ecx,edx)	\
     __asm__ ("push %%ebx\n\t"		\
 	     "cpuid\n\t"		\
@@ -73,18 +121,18 @@
 	     : "=r" (eax),
 	       "=r" (ebx)
 	     :
-	     : "cc");
+	     : "cc"); */
 
-    if (eax == ebx)		/* no cpuid */
+    if (!has_cpuid)		/* no cpuid */
 	return 0;
 
-    cpuid (0x00000000, eax, ebx, ecx, edx);
+    cpuid (0x00000000, regs);
     if (!eax)			/* vendor string only */
 	return 0;
 
     AMD = (ebx == 0x68747541) && (ecx == 0x444d4163) && (edx == 0x69746e65);
 
-    cpuid (0x00000001, eax, ebx, ecx, edx);
+    cpuid (0x00000001, regs);
     if (! (edx & 0x00800000))	/* no MMX */
 	return 0;
 
@@ -92,11 +140,11 @@
     if (edx & 0x02000000)	/* SSE - identical to AMD MMX extensions */
 	caps = MPEG2_ACCEL_X86_MMX | MPEG2_ACCEL_X86_MMXEXT;
 
-    cpuid (0x80000000, eax, ebx, ecx, edx);
+    cpuid (0x80000000, regs);
     if (eax < 0x80000001)	/* no extended capabilities */
 	return caps;
 
-    cpuid (0x80000001, eax, ebx, ecx, edx);
+    cpuid (0x80000001, regs);
 
     if (edx & 0x80000000)
 	caps |= MPEG2_ACCEL_X86_3DNOW;
@@ -108,6 +156,54 @@
 }
 #endif /* ARCH_X86 */
 
+#ifdef ARCH_X86_64
+static inline uint32_t arch_accel (void)
+{
+    unsigned int regs[4];
+    int AMD;
+    uint32_t caps;
+    
+#define eax regs[0]
+#define ebx regs[1]
+#define ecx regs[2]
+#define edx regs[3]
+
+
+
+    if (!has_cpuid())		/* no cpuid */
+	return 0;
+
+    cpuid (0x00000000, regs);
+    if (!eax)			/* vendor string only */
+	return 0;
+
+    AMD = (ebx == 0x68747541) && (ecx == 0x444d4163) && (edx == 0x69746e65);
+
+    cpuid (0x00000001, regs);
+    if (! (edx & 0x00800000))	/* no MMX */
+	return 0;
+
+    caps = MPEG2_ACCEL_X86_MMX;
+    if (edx & 0x02000000)	/* SSE - identical to AMD MMX extensions */
+	caps = MPEG2_ACCEL_X86_MMX | MPEG2_ACCEL_X86_MMXEXT;
+
+    cpuid (0x80000000, regs);
+    if (eax < 0x80000001)	/* no extended capabilities */
+	return caps;
+
+    cpuid (0x80000001, regs);
+
+    if (edx & 0x80000000)
+	caps |= MPEG2_ACCEL_X86_3DNOW;
+
+    if (AMD && (edx & 0x00400000))	/* AMD MMX extensions */
+	caps |= MPEG2_ACCEL_X86_MMXEXT;
+
+    return caps;
+}
+#endif /* ARCH_X86_64 */
+
+
 #if defined(ARCH_PPC) || defined(ARCH_SPARC)
 #include <signal.h>
 #include <setjmp.h>
@@ -212,7 +308,7 @@
 
     accel = 0;
 #ifdef ACCEL_DETECT
-#if defined (ARCH_X86) || defined (ARCH_PPC) || defined (ARCH_ALPHA) || defined (ARCH_SPARC)
+#if defined (ARCH_X86) || defined (ARCH_X86_64) || defined (ARCH_PPC) || defined (ARCH_ALPHA) || defined (ARCH_SPARC)
     accel = arch_accel ();
 #endif
 #endif
diff -Naur mpeg2dec-0.4.0.orig/libmpeg2/cpu_state.c mpeg2dec-0.4.0/libmpeg2/cpu_state.c
--- mpeg2dec-0.4.0.orig/libmpeg2/cpu_state.c	2005-07-29 17:41:59.000000000 +0200
+++ mpeg2dec-0.4.0/libmpeg2/cpu_state.c	2005-07-29 17:47:46.000000000 +0200
@@ -29,14 +29,14 @@
 #include "mpeg2.h"
 #include "attributes.h"
 #include "mpeg2_internal.h"
-#ifdef ARCH_X86
+#if defined(ARCH_X86) || defined(ARCH_X86_64)
 #include "mmx.h"
 #endif
 
 void (* mpeg2_cpu_state_save) (cpu_state_t * state) = NULL;
 void (* mpeg2_cpu_state_restore) (cpu_state_t * state) = NULL;
 
-#ifdef ARCH_X86
+#if defined(ARCH_X86) || defined(ARCH_X86_64)
 static void state_restore_mmx (cpu_state_t * state)
 {
     emms ();
@@ -115,7 +115,7 @@
 
 void mpeg2_cpu_state_init (uint32_t accel)
 {
-#ifdef ARCH_X86
+#if defined(ARCH_X86) || defined(ARCH_X86_64)
     if (accel & MPEG2_ACCEL_X86_MMX) {
 	mpeg2_cpu_state_restore = state_restore_mmx;
     }
diff -Naur mpeg2dec-0.4.0.orig/libmpeg2/idct.c mpeg2dec-0.4.0/libmpeg2/idct.c
--- mpeg2dec-0.4.0.orig/libmpeg2/idct.c	2005-07-29 17:41:59.000000000 +0200
+++ mpeg2dec-0.4.0/libmpeg2/idct.c	2005-07-29 17:48:35.000000000 +0200
@@ -235,7 +235,7 @@
 
 void mpeg2_idct_init (uint32_t accel)
 {
-#ifdef ARCH_X86
+#if defined(ARCH_X86) || defined(ARCH_X86_64)
     if (accel & MPEG2_ACCEL_X86_MMXEXT) {
 	mpeg2_idct_copy = mpeg2_idct_copy_mmxext;
 	mpeg2_idct_add = mpeg2_idct_add_mmxext;
diff -Naur mpeg2dec-0.4.0.orig/libmpeg2/idct_mmx.c mpeg2dec-0.4.0/libmpeg2/idct_mmx.c
--- mpeg2dec-0.4.0.orig/libmpeg2/idct_mmx.c	2005-07-29 17:41:59.000000000 +0200
+++ mpeg2dec-0.4.0/libmpeg2/idct_mmx.c	2005-07-29 17:49:05.000000000 +0200
@@ -23,7 +23,7 @@
 
 #include "config.h"
 
-#ifdef ARCH_X86
+#if defined(ARCH_X86) || defined(ARCH_X86_64)
 
 #include <inttypes.h>
 
diff -Naur mpeg2dec-0.4.0.orig/libmpeg2/motion_comp.c mpeg2dec-0.4.0/libmpeg2/motion_comp.c
--- mpeg2dec-0.4.0.orig/libmpeg2/motion_comp.c	2005-07-29 17:41:59.000000000 +0200
+++ mpeg2dec-0.4.0/libmpeg2/motion_comp.c	2005-07-29 17:49:33.000000000 +0200
@@ -33,7 +33,7 @@
 
 void mpeg2_mc_init (uint32_t accel)
 {
-#ifdef ARCH_X86
+#if defined(ARCH_X86) || defined(ARCH_X86_64)
     if (accel & MPEG2_ACCEL_X86_MMXEXT)
 	mpeg2_mc = mpeg2_mc_mmxext;
     else if (accel & MPEG2_ACCEL_X86_3DNOW)
diff -Naur mpeg2dec-0.4.0.orig/libmpeg2/motion_comp_mmx.c mpeg2dec-0.4.0/libmpeg2/motion_comp_mmx.c
--- mpeg2dec-0.4.0.orig/libmpeg2/motion_comp_mmx.c	2005-07-29 17:41:59.000000000 +0200
+++ mpeg2dec-0.4.0/libmpeg2/motion_comp_mmx.c	2005-07-29 17:50:07.000000000 +0200
@@ -23,7 +23,7 @@
 
 #include "config.h"
 
-#ifdef ARCH_X86
+#if defined (ARCH_X86) || defined (ARCH_X86_64)
 
 #include <inttypes.h>
 
