diff -Naur rstart-0.99.1.orig/Makefile.in rstart-0.99.1/Makefile.in
--- rstart-0.99.1.orig/Makefile.in	2005-11-13 03:07:26.987519000 +0100
+++ rstart-0.99.1/Makefile.in	2005-11-13 03:07:43.540553500 +0100
@@ -806,7 +806,7 @@
 	done;								\
 									\
 	cd $(DESTDIR)$(DATA_DIR)/contexts;				\
-	sed -e '/^$$/d' -e '/^#/d' ${srcdir}/contexts/@Aliases |	\
+	sed -e '/^$$/d' -e '/^#/d' @Aliases |	\
 	    while read real aliases; do					\
 	        for i in $$aliases; do					\
 	            $(RM) $$i;						\
