diff -Naurp libkarma-0.0.5/karma-sharp/Makefile libkarma-0.0.5-p/karma-sharp/Makefile
--- libkarma-0.0.5/karma-sharp/Makefile	2006-08-03 01:04:25.000000000 +0200
+++ libkarma-0.0.5-p/karma-sharp/Makefile	2006-11-07 20:26:59.000000000 +0100
@@ -26,10 +26,10 @@ karma-sharp.pc: FORCE
 
 ifdef HAS_MONO
 install: $(ASSEMBLY) karma-sharp.pc
-	../install-sh -d $(PREFIX)/lib/karma-sharp
-	../install-sh -d $(PREFIX)/lib/pkgconfig
-	../install-sh -t $(PREFIX)/lib/karma-sharp $(ASSEMBLY)
-	../install-sh -t $(PREFIX)/lib/pkgconfig -m 0644 karma-sharp.pc
+	../install-sh -d $(DESTDIR)$(PREFIX)/lib/karma-sharp
+	../install-sh -d $(DESTDIR)$(PREFIX)/lib/pkgconfig
+	../install-sh -t $(DESTDIR)$(PREFIX)/lib/karma-sharp $(ASSEMBLY)
+	../install-sh -t $(DESTDIR)$(PREFIX)/lib/pkgconfig -m 0644 karma-sharp.pc
 else
 install:
 
diff -Naurp libkarma-0.0.5/Makefile libkarma-0.0.5-p/Makefile
--- libkarma-0.0.5/Makefile	2006-08-03 01:04:25.000000000 +0200
+++ libkarma-0.0.5-p/Makefile	2006-11-07 20:25:38.000000000 +0100
@@ -1,6 +1,6 @@
 LK_SRC=src
 TOOLSDIR=tools
-PREFIX=/usr/local
+PREFIX=/usr
 VERSION=0.0.5
 
 export PREFIX
diff -Naurp libkarma-0.0.5/src/Makefile libkarma-0.0.5-p/src/Makefile
--- libkarma-0.0.5/src/Makefile	2006-08-03 01:04:24.000000000 +0200
+++ libkarma-0.0.5-p/src/Makefile	2006-11-07 20:24:04.000000000 +0100
@@ -19,12 +19,12 @@ CFLAGS+=-Wmissing-declarations -Wno-swit
 default: all
 
 install:
-	../install-sh -m 0755 -d $(PREFIX)/lib
-	../install-sh -m 0755 -d $(PREFIX)/include
-	../install-sh -m 0755 -d $(PREFIX)/include/libkarma
-	../install-sh -m 0644 -t $(PREFIX)/lib/ $(LIBDIR)/$(TARGET).a
-	../install-sh -m 0755 -t $(PREFIX)/lib/ $(LIBDIR)/$(TARGET).so
-	../install-sh -m 0644 -t $(PREFIX)/include/libkarma/ $(INST_HEADERS)
+	../install-sh -m 0755 -d $(DESTDIR)$(PREFIX)/lib
+	../install-sh -m 0755 -d $(DESTDIR)$(PREFIX)/include
+	../install-sh -m 0755 -d $(DESTDIR)$(PREFIX)/include/libkarma
+	../install-sh -m 0644 -t $(DESTDIR)$(PREFIX)/lib/ $(LIBDIR)/$(TARGET).a
+	../install-sh -m 0755 -t $(DESTDIR)$(PREFIX)/lib/ $(LIBDIR)/$(TARGET).so
+	../install-sh -m 0644 -t $(DESTDIR)$(PREFIX)/include/libkarma/ $(INST_HEADERS)
 
 uninstall:
 	$(RM) $(PREFIX)/lib/$(TARGET).a $(PREFIX)/lib/$(TARGET).so*
diff -Naurp libkarma-0.0.5/tools/Makefile libkarma-0.0.5-p/tools/Makefile
--- libkarma-0.0.5/tools/Makefile	2006-08-03 01:04:25.000000000 +0200
+++ libkarma-0.0.5-p/tools/Makefile	2006-11-07 20:24:38.000000000 +0100
@@ -27,8 +27,8 @@ karma_helper: karma_helper.c
 	$(CC) $(INCLUDES) $(CFLAGS) $@.c -o $@ -lusb
 
 install:
-	../install-sh -m 0755 -d $(PREFIX)/bin
-	../install-sh -m 0755 -t $(PREFIX)/bin/ $(TOOLS)
+	../install-sh -m 0755 -d $(DESTDIR)$(PREFIX)/bin
+	../install-sh -m 0755 -t $(DESTDIR)$(PREFIX)/bin/ $(TOOLS)
 
 uninstall:
 	cd $(PREFIX)/bin && $(RM) $(TOOLS)
