--- pekwm-0.1.6.orig/src/CfgParser.cc	2007-05-08 10:10:13.000000000 +0200
+++ pekwm-0.1.6/src/CfgParser.cc	2008-04-22 22:41:21.000000000 +0200
@@ -11,10 +11,11 @@
 #include "Util.hh"
 
 #include <iostream>
 #include <memory>
 #include <cassert>
+#include <cstring>
 
 #ifdef HAVE_GETTEXT
 #include <libintl.h>
 #define _(S) gettext(S)
 #else // !HAVE_GETTEXT
--- pekwm-0.1.6.orig/src/CfgParser.hh	2007-05-08 10:10:13.000000000 +0200
+++ pekwm-0.1.6/src/CfgParser.hh	2008-04-22 22:41:21.000000000 +0200
@@ -30,10 +30,11 @@
 
 #include <list>
 #include <map>
 #include <set>
 #include <string>
+#include <cstring>
 #include <iostream>
 #include <cstdlib>
 
 //! @brief Configuration file parser.
 class CfgParser {
--- pekwm-0.1.6.orig/src/CfgParserSource.cc	2007-05-08 10:10:13.000000000 +0200
+++ pekwm-0.1.6/src/CfgParserSource.cc	2008-04-22 22:41:21.000000000 +0200
@@ -11,10 +11,11 @@
 //
 
 #include "CfgParserSource.hh"
 #include "Util.hh"
 
+#include <cstdlib>
 #include <iostream>
 
 extern "C" {
 #include <unistd.h>
 }
--- pekwm-0.1.6.orig/src/ColorHandler.cc	2007-05-08 10:10:13.000000000 +0200
+++ pekwm-0.1.6/src/ColorHandler.cc	2008-04-22 22:41:21.000000000 +0200
@@ -5,10 +5,11 @@
 // See the LICENSE file for more information.
 //
 
 #include "ColorHandler.hh"
 #include "PScreen.hh"
+#include <cstring>
 
 #ifdef DEBUG
 #include <iostream>
 using std::cerr;
 using std::endl;
--- pekwm-0.1.6.orig/src/ColorHandler.hh	2007-05-08 10:10:13.000000000 +0200
+++ pekwm-0.1.6/src/ColorHandler.hh	2008-04-22 22:41:21.000000000 +0200
@@ -13,10 +13,11 @@
 
 #include "pekwm.hh"
 
 #include <list>
 #include <string>
+#include <cstring>
 
 class ColorHandler {
 public:
     class Entry {
     public:
--- pekwm-0.1.6.orig/src/PImageNativeLoaderXpm.cc	2007-05-08 10:10:13.000000000 +0200
+++ pekwm-0.1.6/src/PImageNativeLoaderXpm.cc	2008-04-22 22:41:21.000000000 +0200
@@ -11,10 +11,11 @@
 #ifdef HAVE_IMAGE_XPM
 
 #include "PImageNativeLoaderXpm.hh"
 
 #include <iostream>
+#include <cstring>
 
 using std::cerr;
 using std::endl;
 using std::string;
 
--- pekwm-0.1.6.orig/src/RegexString.cc	2007-05-08 10:10:13.000000000 +0200
+++ pekwm-0.1.6/src/RegexString.cc	2008-04-22 22:41:21.000000000 +0200
@@ -9,11 +9,11 @@
 //
 
 #include "../config.h"
 
 #include "RegexString.hh"
-
+#include <cstdlib>
 #include <iostream>
 
 #ifdef HAVE_GETTEXT
 #include <libintl.h>
 #define _(S) gettext(S)
--- pekwm-0.1.6.orig/src/Util.cc	2007-05-08 10:10:13.000000000 +0200
+++ pekwm-0.1.6/src/Util.cc	2008-04-22 22:42:12.000000000 +0200
@@ -11,10 +11,11 @@
 //
 
 #include "../config.h"
 #include "Util.hh"
 
+#include <cstdlib>
 #include <algorithm>
 #include <iostream>
 #include <sstream>
 
 #include <unistd.h>
--- pekwm-0.1.6.orig/src/Util.hh	2007-05-08 10:10:13.000000000 +0200
+++ pekwm-0.1.6/src/Util.hh	2008-04-22 22:41:21.000000000 +0200
@@ -12,10 +12,11 @@
 #define _UTIL_HH_
 
 #include "Types.hh"
 
 #include <string>
+#include <cstring>
 #include <vector>
 #include <functional>
 #include <sstream>
 
 //! @brief Namespace Util used for various small file/string tasks.
