diff -aur emerald-0.2.1/configure.ac fw_emerald-0.2.1/configure.ac
--- emerald-0.2.1/configure.ac	2007-01-30 05:23:43.000000000 +0100
+++ fw_emerald-0.2.1/configure.ac	2007-10-02 08:48:27.000000000 +0200
@@ -42,6 +42,12 @@
 AC_MSG_NOTICE([Using PKG_CONFIG_PATH=$PKG_CONFIG_PATH])
 
 PKG_CHECK_MODULES(EMERALD, $EMERALD_REQUIRES)
+PKG_CHECK_EXISTS(libwnck-1.0 >= 2.19.3,
+		 [have_libwnck_2_19_3=yes], [have_libwnck_2_19_3=no])
+if test "$have_libwnck_2_19_3" = yes; then
+	AC_DEFINE(HAVE_LIBWNCK_2_19_3, 1,
+		  [Define to 1 if libwnck version >= 2_19_3])
+fi
 
 AC_SUBST(EMERALD_REQUIRES)
 CFLAGS="$CFLAGS $EMERALD_CFLAGS"
diff -aur emerald-0.2.1/src/main.c fw_emerald-0.2.1/src/main.c
--- emerald-0.2.1/src/main.c	2007-03-17 20:29:38.000000000 +0100
+++ fw_emerald-0.2.1/src/main.c	2007-10-02 08:49:14.000000000 +0200
@@ -3781,7 +3781,9 @@
 		return;
 	case WNCK_WINDOW_NORMAL:
 	case WNCK_WINDOW_DIALOG:
+#ifndef HAVE_LIBWNCK_2_19_3
 	case WNCK_WINDOW_MODAL_DIALOG:
+#endif
 	case WNCK_WINDOW_TOOLBAR:
 	case WNCK_WINDOW_MENU:
 	case WNCK_WINDOW_UTILITY:
