diff -Naur camlp5-6.02.1/etc/pr_o.ml camlp5-6.02.1/etc/pr_o.ml
--- camlp5-6.02.1/etc/pr_o.ml	2010-11-21 18:17:45.000000000 +0100
+++ camlp5-6.02.1/etc/pr_o.ml	2011-02-13 20:09:38.817000010 +0100
@@ -2080,7 +2080,7 @@
 value label_ipatt expr pc (p, oe) =
   match Pcaml.unvala oe with
   [ Some e -> pprintf pc "~%p:%p" patt p expr e
-  | None -> patt pc p ]
+  | None -> pprintf pc "~%p" patt p ]
 ;
 
 EXTEND_PRINTER
