diff --git a/configure.in b/configure.in
index 3840ff2..0513b5f 100644
--- a/configure.in
+++ b/configure.in
@@ -142,8 +142,8 @@ AC_CHECK_LIB(ncurses, initscr)
 if test "${ac_cv_lib_ncurses_initscr}" != "yes"; then
   AC_MSG_ERROR([can not find the ncurses library.])
 fi
-AC_CHECK_LIB(cdk, initCDKScreen)
-if test "${ac_cv_lib_cdk_initCDKScreen}" != "yes"; then
+AC_CHECK_LIB(cdkw, initCDKScreen)
+if test "${ac_cv_lib_cdkw_initCDKScreen}" != "yes"; then
   AC_MSG_ERROR([can not find the cdk library.])
 fi
 if test ${USE_CRACKLIB} == 1; then
@@ -187,7 +187,7 @@ if test ${INCOMPATIBLE} -gt 0; then
   CDK_UNUSABLE=1
 
   # this file only exists in v5
-  CDK_VERSION_H=`dirname "${CDK_H}"`"/cdk_version.h"
+  CDK_VERSION_H="/usr/include/cdk/cdk_version.h"
   if test -f "${CDK_VERSION_H}"; then
     CDK_VERSION=`grep "CDK_VERSION_PATCH" ${CDK_VERSION_H} | cut -d " " -f 3 | sed -e 's/"//g'`
     AC_MSG_NOTICE([CDK version patch: ${CDK_VERSION}])
diff --git a/interface_gui.h b/interface_gui.h
index bb659e6..abae3e5 100644
--- a/interface_gui.h
+++ b/interface_gui.h
@@ -27,7 +27,7 @@
 /* #############################################################################
  * includes
  */
-#include <cdk/cdk.h>
+#include <cdk.h>
 
 
 /* #############################################################################
diff --git a/options.h b/options.h
index e4fb2a4..2e0f0be 100644
--- a/options.h
+++ b/options.h
@@ -28,7 +28,7 @@
  * includes
  */
 /* we need this for the version info only */
-#include <cdk/cdk.h>
+#include <cdk.h>
 
 
 /* #############################################################################
