--- kmod-34/meson.build	2025-02-21 21:28:48.000000000 +0100
+++ kmod-34.new/meson.build	2025-02-26 12:15:20.914986270 +0100
@@ -219,8 +219,6 @@
       error('User provided @0@, \'@1@\' is not an absolute path.'
           .format(dir_option, customdir))
     endif
-    # Strip all leading/trailing and re-add only the leading one.
-    customdir = '' / customdir.strip('/')
   endif
   cdata.set_quoted(quoted, customdir)
   set_variable(dir_option, customdir)
