--- nitroshare-desktop-0.3.4/src/dist/nitroshare.py.in	2017-10-12 23:50:31.000000000 +0200
+++ nitroshare-desktop-0.3.4.new/src/dist/nitroshare.py.in	2023-08-31 15:34:32.103494946 +0200
@@ -27,8 +27,8 @@
 
 from json import loads
 from os import path
-from urllib import url2pathname
-from urlparse import urlparse
+from urllib.request import url2pathname
+from urllib.parse import urlparse
 
 import gi
 gi.require_version('Gtk', '3.0')
