diff -Naur libabw-0.1.1/src/lib/Makefile.am libabw-0.1.1-boost/src/lib/Makefile.am
--- libabw-0.1.1/src/lib/Makefile.am	2014-12-20 08:44:20.000000000 +0100
+++ libabw-0.1.1-boost/src/lib/Makefile.am	2016-01-28 09:35:30.586836257 +0100
@@ -26,7 +26,7 @@
 	$(top_builddir)/src/lib/tokens.h \
 	$(top_builddir)/src/lib/tokenhash.h
 
-libabw_@ABW_MAJOR_VERSION@_@ABW_MINOR_VERSION@_la_LIBADD  = $(REVENGE_LIBS) $(LIBXML_LIBS) $(ZLIB_LIBS) @LIBABW_WIN32_RESOURCE@
+libabw_@ABW_MAJOR_VERSION@_@ABW_MINOR_VERSION@_la_LIBADD  = $(REVENGE_LIBS) $(LIBXML_LIBS) $(ZLIB_LIBS) -lboost_system @LIBABW_WIN32_RESOURCE@
 libabw_@ABW_MAJOR_VERSION@_@ABW_MINOR_VERSION@_la_DEPENDENCIES = @LIBABW_WIN32_RESOURCE@
 libabw_@ABW_MAJOR_VERSION@_@ABW_MINOR_VERSION@_la_LDFLAGS = $(version_info) -export-dynamic $(no_undefined)
 libabw_@ABW_MAJOR_VERSION@_@ABW_MINOR_VERSION@_la_SOURCES = \
diff -Naur libabw-0.1.1/src/lib/Makefile.in libabw-0.1.1-boost/src/lib/Makefile.in
--- libabw-0.1.1/src/lib/Makefile.in	2014-12-20 14:23:18.000000000 +0100
+++ libabw-0.1.1-boost/src/lib/Makefile.in	2016-01-28 09:36:54.901045501 +0100
@@ -369,7 +369,7 @@
 	$(top_builddir)/src/lib/tokens.h \
 	$(top_builddir)/src/lib/tokenhash.h
 
-libabw_@ABW_MAJOR_VERSION@_@ABW_MINOR_VERSION@_la_LIBADD = $(REVENGE_LIBS) $(LIBXML_LIBS) $(ZLIB_LIBS) @LIBABW_WIN32_RESOURCE@
+libabw_@ABW_MAJOR_VERSION@_@ABW_MINOR_VERSION@_la_LIBADD = $(REVENGE_LIBS) $(LIBXML_LIBS) $(ZLIB_LIBS) -lboost_system @LIBABW_WIN32_RESOURCE@
 libabw_@ABW_MAJOR_VERSION@_@ABW_MINOR_VERSION@_la_DEPENDENCIES = @LIBABW_WIN32_RESOURCE@
 libabw_@ABW_MAJOR_VERSION@_@ABW_MINOR_VERSION@_la_LDFLAGS = $(version_info) -export-dynamic $(no_undefined)
 libabw_@ABW_MAJOR_VERSION@_@ABW_MINOR_VERSION@_la_SOURCES = \
