diff -ruN mjpegtools-1.6.2.orig/utils/altivec/fdct_idct.c mjpegtools-1.6.2/utils/altivec/fdct_idct.c
--- mjpegtools-1.6.2.orig/utils/altivec/fdct_idct.c	2003-10-09 09:05:09.000000000 +0200
+++ mjpegtools-1.6.2/utils/altivec/fdct_idct.c	2004-08-05 09:38:31.808037456 +0200
@@ -177,14 +177,14 @@
 
 /* Post-scaling matrix -- scaled by 1 */
 static const vector signed short PostScale[8] = {
-    (vector signed short)VCONST(4095, 5681, 5351, 4816, 4095, 4816, 5351, 5681),
-    (vector signed short)VCONST(5681, 7880, 7422, 6680, 5681, 6680, 7422, 7880),
-    (vector signed short)VCONST(5351, 7422, 6992, 6292, 5351, 6292, 6992, 7422),
-    (vector signed short)VCONST(4816, 6680, 6292, 5663, 4816, 5663, 6292, 6680),
-    (vector signed short)VCONST(4095, 5681, 5351, 4816, 4095, 4816, 5351, 5681),
-    (vector signed short)VCONST(4816, 6680, 6292, 5663, 4816, 5663, 6292, 6680),
-    (vector signed short)VCONST(5351, 7422, 6992, 6292, 5351, 6292, 6992, 7422),
-    (vector signed short)VCONST(5681, 7880, 7422, 6680, 5681, 6680, 7422, 7880)
+    (const vector signed short)VCONST(4095, 5681, 5351, 4816, 4095, 4816, 5351, 5681),
+    (const vector signed short)VCONST(5681, 7880, 7422, 6680, 5681, 6680, 7422, 7880),
+    (const vector signed short)VCONST(5351, 7422, 6992, 6292, 5351, 6292, 6992, 7422),
+    (const vector signed short)VCONST(4816, 6680, 6292, 5663, 4816, 5663, 6292, 6680),
+    (const vector signed short)VCONST(4095, 5681, 5351, 4816, 4095, 4816, 5351, 5681),
+    (const vector signed short)VCONST(4816, 6680, 6292, 5663, 4816, 5663, 6292, 6680),
+    (const vector signed short)VCONST(5351, 7422, 6992, 6292, 5351, 6292, 6992, 7422),
+    (const vector signed short)VCONST(5681, 7880, 7422, 6680, 5681, 6680, 7422, 7880)
 };
 
 /***************************************************************
@@ -219,7 +219,7 @@
  * of lvx which is more efficient given the remainder of the instruction mix.
  */
 static const vector signed short SpecialConstants =
-  (vector signed short)VCONST(23170, 13573, 6518, 21895, -23170, -21895, 0, 0);
+  (const vector signed short)VCONST(23170, 13573, 6518, 21895, -23170, -21895, 0, 0);
 
 #ifdef ORIGINAL_SOURCE
 void DCT(short *input, short *output) {
@@ -364,14 +364,14 @@
 
 /* Pre-Scaling matrix -- scaled by 1 */
 static const vector signed short PreScale[8] = {
-    (vector signed short)VCONST(4095, 5681, 5351, 4816, 4095, 4816, 5351, 5681),
-    (vector signed short)VCONST(5681, 7880, 7422, 6680, 5681, 6680, 7422, 7880),
-    (vector signed short)VCONST(5351, 7422, 6992, 6292, 5351, 6292, 6992, 7422),
-    (vector signed short)VCONST(4816, 6680, 6292, 5663, 4816, 5663, 6292, 6680),
-    (vector signed short)VCONST(4095, 5681, 5351, 4816, 4095, 4816, 5351, 5681),
-    (vector signed short)VCONST(4816, 6680, 6292, 5663, 4816, 5663, 6292, 6680),
-    (vector signed short)VCONST(5351, 7422, 6992, 6292, 5351, 6292, 6992, 7422),
-    (vector signed short)VCONST(5681, 7880, 7422, 6680, 5681, 6680, 7422, 7880)
+    (const vector signed short)VCONST(4095, 5681, 5351, 4816, 4095, 4816, 5351, 5681),
+    (const vector signed short)VCONST(5681, 7880, 7422, 6680, 5681, 6680, 7422, 7880),
+    (const vector signed short)VCONST(5351, 7422, 6992, 6292, 5351, 6292, 6992, 7422),
+    (const vector signed short)VCONST(4816, 6680, 6292, 5663, 4816, 5663, 6292, 6680),
+    (const vector signed short)VCONST(4095, 5681, 5351, 4816, 4095, 4816, 5351, 5681),
+    (const vector signed short)VCONST(4816, 6680, 6292, 5663, 4816, 5663, 6292, 6680),
+    (const vector signed short)VCONST(5351, 7422, 6992, 6292, 5351, 6292, 6992, 7422),
+    (const vector signed short)VCONST(5681, 7880, 7422, 6680, 5681, 6680, 7422, 7880)
 };
 
 

