diff -Naur mjpegtools-1.8.0/configure.ac mjpegtools-1.8.0-p/configure.ac
--- mjpegtools-1.8.0/configure.ac	2006-08-26 14:00:51.000000000 +0200
+++ mjpegtools-1.8.0-p/configure.ac	2006-08-26 14:02:24.000000000 +0200
@@ -538,8 +538,8 @@
 
 dnl maddog:  check out AC-COMPILE-WARNINGS ???????
 
-CFLAGS="$ARCHFLAGS $CFLAGS"
-CXXFLAGS="$ARCHFLAGS $CXXFLAGS"
+CFLAGS="$CFLAGS"
+CXXFLAGS="$CXXFLAGS"
 
 if test "x${GCC}" != "xyes" ; then
   enable_compile_warnings=no
