--- tomahawk/CMakeModules/FindLibLastFm.cmake	2016-05-11 18:56:08.000000000 +0200
+++ tomahawk.new/CMakeModules/FindLibLastFm.cmake	2016-05-11 23:20:09.000000000 +0200
@@ -14,7 +14,7 @@
 # Include dir
 find_path(LIBLASTFM_INCLUDE_DIR
   # Track.h doesn't exist in liblastfm-0.3.1, was called Track back then
-  NAMES lastfm${LASTFM_LIB_SUFFIX}/Track.h
+  NAMES lastfm/Track.h
   PATHS ${KDE4_INCLUDE_DIR}
 )
 
--- tomahawk/src/libtomahawk/utils/TomahawkUtils.cpp	2016-05-11 18:56:10.000000000 +0200
+++ tomahawk.new/src/libtomahawk/utils/TomahawkUtils.cpp	2016-05-11 23:49:26.000000000 +0200
@@ -33,7 +33,7 @@
 
 #ifdef LIBLASTFM_FOUND
     #if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-    #include <lastfm5/ws.h>
+    #include <lastfm/ws.h>
     #else
     #include <lastfm/ws.h>
     #endif
--- tomahawk/src/libtomahawk/accounts/lastfm/LastFmConfig.cpp	2016-05-11 18:56:09.000000000 +0200
+++ tomahawk.new/src/libtomahawk/accounts/lastfm/LastFmConfig.cpp	2016-05-12 00:08:02.000000000 +0200
@@ -29,17 +29,10 @@
 #include "utils/Closure.h"
 #include "utils/NetworkAccessManager.h"
 
-#if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-#include <lastfm5/ws.h>
-#include <lastfm5/User.h>
-#include <lastfm5/XmlQuery.h>
-#include <lastfm5/Track.h>
-#else
 #include <lastfm/ws.h>
 #include <lastfm/User.h>
 #include <lastfm/XmlQuery.h>
 #include <lastfm/Track.h>
-#endif
 
 using namespace Tomahawk::Accounts;
 
--- tomahawk/src/tomahawk/TomahawkApp.h	2016-05-11 18:56:10.000000000 +0200
+++ tomahawk.new/src/tomahawk/TomahawkApp.h	2016-05-12 00:09:14.000000000 +0200
@@ -64,11 +64,7 @@
 }
 
 #ifdef LIBLASTFM_FOUND
-    #if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-    #include <lastfm5/NetworkAccessManager.h>
-    #else
     #include <lastfm/NetworkAccessManager.h>
-    #endif
     #include "Scrobbler.h"
 #endif
 
--- tomahawk/src/tomahawk/Scrobbler.h	2016-05-11 18:56:10.000000000 +0200
+++ tomahawk.new/src/tomahawk/Scrobbler.h	2016-05-12 00:10:15.000000000 +0200
@@ -24,11 +24,7 @@
 #include "infosystem/InfoSystem.h"
 
 
-#if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-#include <lastfm5/ScrobblePoint.h>
-#else
 #include <lastfm/ScrobblePoint.h>
-#endif
 #include <QObject>
 
 /**
--- tomahawk/src/libtomahawk/accounts/lastfm/LastFmInfoPlugin.h	2016-05-11 18:56:09.000000000 +0200
+++ tomahawk.new/src/libtomahawk/accounts/lastfm/LastFmInfoPlugin.h	2016-05-12 00:11:28.000000000 +0200
@@ -24,15 +24,9 @@
 #include "infosystem/InfoSystemWorker.h"
 #include "DllMacro.h"
 
-#if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-#include <lastfm5/Track.h>
-#include <lastfm5/Audioscrobbler.h>
-#include <lastfm5/ScrobblePoint.h>
-#else
 #include <lastfm/Track.h>
 #include <lastfm/Audioscrobbler.h>
 #include <lastfm/ScrobblePoint.h>
-#endif
 
 #include <QObject>
 
--- tomahawk/src/libtomahawk/accounts/lastfm/LastFmInfoPlugin.cpp	2016-05-11 18:56:09.000000000 +0200
+++ tomahawk.new/src/libtomahawk/accounts/lastfm/LastFmInfoPlugin.cpp	2016-05-12 00:12:12.000000000 +0200
@@ -34,13 +34,8 @@
 #include "TomahawkSettings.h"
 #include "utils/NetworkAccessManager.h"
 
-#if QT_VERSION >= QT_VERSION_CHECK(5,0,0)
-#include <lastfm5/ws.h>
-#include <lastfm5/XmlQuery.h>
-#else
 #include <lastfm/ws.h>
 #include <lastfm/XmlQuery.h>
-#endif
 
 using namespace Tomahawk::Accounts;
 using namespace Tomahawk::InfoSystem;
