Index: src/core/curl_get.cc
===================================================================
--- rtorrent/src/core/curl_get.cc	(revision 868)
+++ rtorrent/src/core/curl_get.cc	(working copy)
@@ -66,7 +66,7 @@
   if (m_stream == NULL)
     throw torrent::internal_error("Tried to call CurlGet::start without a valid output stream.");
 
-  m_handle = curl_easy_init();
+  if ((m_handle = curl_easy_init()) == NULL) return;
 
   curl_easy_setopt(m_handle, CURLOPT_URL,            m_url.c_str());
   curl_easy_setopt(m_handle, CURLOPT_WRITEFUNCTION,  &curl_get_receive_write);
