diff -Naur rtorrent-0.7.9.orig/rak/functional.h rtorrent-0.7.9/rak/functional.h
--- rtorrent-0.7.9.orig/rak/functional.h	2007-07-19 22:23:09.000000000 +0100
+++ rtorrent-0.7.9/rak/functional.h	2008-04-11 20:32:13.000000000 +0100
@@ -37,6 +37,8 @@
 #ifndef RAK_FUNCTIONAL_H
 #define RAK_FUNCTIONAL_H
 
+#include <cstdlib>
+
 #include <functional>
 
 namespace rak {
diff -Naur rtorrent-0.7.9.orig/src/command_ui.cc rtorrent-0.7.9/src/command_ui.cc
--- rtorrent-0.7.9.orig/src/command_ui.cc	2007-10-19 18:20:56.000000000 +0100
+++ rtorrent-0.7.9/src/command_ui.cc	2008-04-11 20:33:26.000000000 +0100
@@ -36,6 +36,8 @@
 
 #include "config.h"
 
+#include <ctime>
+
 #include <rak/functional.h>
 #include <rak/functional_fun.h>
 #include <sigc++/adaptors/bind.h>
diff -Naur rtorrent-0.7.9.orig/src/display/text_element_value.cc rtorrent-0.7.9/src/display/text_element_value.cc
--- rtorrent-0.7.9.orig/src/display/text_element_value.cc	2007-08-28 14:09:34.000000000 +0100
+++ rtorrent-0.7.9/src/display/text_element_value.cc	2008-04-11 20:25:53.000000000 +0100
@@ -34,6 +34,8 @@
 //           Skomakerveien 33
 //           3185 Skoppum, NORWAY
 
+#include <ctime>
+
 #include "config.h"
 
 #include "globals.h"
diff -Naur rtorrent-0.7.9.orig/src/display/window_file_list.cc rtorrent-0.7.9/src/display/window_file_list.cc
--- rtorrent-0.7.9.orig/src/display/window_file_list.cc	2007-07-19 22:23:11.000000000 +0100
+++ rtorrent-0.7.9/src/display/window_file_list.cc	2008-04-11 20:28:45.000000000 +0100
@@ -36,6 +36,8 @@
 
 #include "config.h"
 
+#include <cstdlib>
+
 #include <torrent/path.h>
 #include <torrent/data/file.h>
 #include <torrent/data/file_list.h>
diff -Naur rtorrent-0.7.9.orig/src/rpc/command_map.cc rtorrent-0.7.9/src/rpc/command_map.cc
--- rtorrent-0.7.9.orig/src/rpc/command_map.cc	2007-10-20 19:58:46.000000000 +0100
+++ rtorrent-0.7.9/src/rpc/command_map.cc	2008-04-11 20:30:02.000000000 +0100
@@ -36,6 +36,8 @@
 
 #include "config.h"
 
+#include <cstdlib>
+
 #include <torrent/exceptions.h>
 #include <torrent/object.h>
 #include <torrent/data/file_list_iterator.h>
diff -Naur rtorrent-0.7.9.orig/src/rpc/exec_file.cc rtorrent-0.7.9/src/rpc/exec_file.cc
--- rtorrent-0.7.9.orig/src/rpc/exec_file.cc	2007-08-11 19:59:23.000000000 +0100
+++ rtorrent-0.7.9/src/rpc/exec_file.cc	2008-04-11 20:30:26.000000000 +0100
@@ -36,7 +36,7 @@
 
 #include "config.h"
 
-#include <string>
+#include <cstring>
 #include <unistd.h>
 #include <rak/path.h>
 #include <sys/types.h>
diff -Naur rtorrent-0.7.9.orig/src/rpc/parse.cc rtorrent-0.7.9/src/rpc/parse.cc
--- rtorrent-0.7.9.orig/src/rpc/parse.cc	2007-08-20 02:21:03.000000000 +0100
+++ rtorrent-0.7.9/src/rpc/parse.cc	2008-04-11 20:30:42.000000000 +0100
@@ -37,6 +37,7 @@
 #include "config.h"
 
 #include <locale>
+#include <cstring>
 #include <rak/path.h>
 #include <torrent/exceptions.h>
 
