diff -Naur libtorrent-rasterbar-0.16.10/include/libtorrent/disk_buffer_pool.hpp libtorrent-rasterbar-0.16.10.new/include/libtorrent/disk_buffer_pool.hpp
--- libtorrent-rasterbar-0.16.10/include/libtorrent/disk_buffer_pool.hpp	2013-03-17 09:15:20.000000000 +0100
+++ libtorrent-rasterbar-0.16.10.new/include/libtorrent/disk_buffer_pool.hpp	2013-05-13 12:49:58.146926833 +0200
@@ -42,9 +42,7 @@
 #include <fstream>
 #endif
 
-#if defined TORRENT_DEBUG || TORRENT_RELEASE_ASSERTS
 #include <map>
-#endif
 
 namespace libtorrent
 {
