diff -ur aqualung-0.9beta9.1.orig/src/decoder/dec_lavc.h aqualung-0.9beta9.1/src/decoder/dec_lavc.h
--- aqualung-0.9beta9.1.orig/src/decoder/dec_lavc.h	2007-04-21 02:57:14.000000000 -0500
+++ aqualung-0.9beta9.1/src/decoder/dec_lavc.h	2008-08-01 10:32:36.000000000 -0500
@@ -23,8 +23,8 @@
 #define _DEC_LAVC_H
 
 #ifdef HAVE_LAVC
-#include <ffmpeg/avcodec.h>
-#include <ffmpeg/avformat.h>
+#include <libavcodec/avcodec.h>
+#include <libavformat/avformat.h>
 #endif /* HAVE_LAVC */
 
 #include "file_decoder.h"
