diff -Naur libclaw-1.7.0/claw/graph.hpp libclaw-1.7.0new/claw/graph.hpp
--- libclaw-1.7.0/claw/graph.hpp	2011-06-12 12:01:53.000000000 +0200
+++ libclaw-1.7.0new/claw/graph.hpp	2012-05-06 11:18:27.701908599 +0200
@@ -39,6 +39,8 @@
 #include <iterator>
 #include <utility>
 
+#include <cstddef>
+
 namespace claw
 {
 
diff -Naur libclaw-1.7.0/claw/impl/basic_socketbuf.tpp libclaw-1.7.0new/claw/impl/basic_socketbuf.tpp
--- libclaw-1.7.0/claw/impl/basic_socketbuf.tpp	2011-06-12 12:01:53.000000000 +0200
+++ libclaw-1.7.0new/claw/impl/basic_socketbuf.tpp	2012-05-06 11:19:47.730079768 +0200
@@ -178,7 +178,7 @@
 		       length, 0 );
 
   if ( write_count >= 0 )
-    setp( m_output_buffer, m_output_buffer + m_output_buffer_size );
+    this->setp( m_output_buffer, m_output_buffer + m_output_buffer_size );
   else
     result = -1;
 
@@ -214,11 +214,11 @@
 
   if ( read_count > 0 )
     {
-      setg( m_input_buffer, m_input_buffer, m_input_buffer + read_count);
+      this->setg( m_input_buffer, m_input_buffer, m_input_buffer + read_count);
       result = this->sgetc();
     }
   else
-    setg( m_input_buffer, m_input_buffer + m_input_buffer_size,
+    this->setg( m_input_buffer, m_input_buffer + m_input_buffer_size,
           m_input_buffer + m_input_buffer_size );
 
   return result;
@@ -243,7 +243,7 @@
       result = traits_type::not_eof(c);
 
       if ( !traits_type::eq_int_type(c, traits_type::eof()) )
-        sputc(c);
+        this->sputc(c);
     }
 
   return result;
diff -Naur libclaw-1.7.0/claw/impl/binary_node.tpp libclaw-1.7.0new/claw/impl/binary_node.tpp
--- libclaw-1.7.0/claw/impl/binary_node.tpp	2011-06-12 12:01:53.000000000 +0200
+++ libclaw-1.7.0new/claw/impl/binary_node.tpp	2012-05-06 11:20:06.311441363 +0200
@@ -22,6 +22,9 @@
 
   contact: julien.jorge@gamned.org
 */
+
+#include <cstddef>
+
 /**
  * \file binary_node.tpp
  * \brief The binary_node class implementation.
diff -Naur libclaw-1.7.0/claw/impl/socket_stream.tpp libclaw-1.7.0new/claw/impl/socket_stream.tpp
--- libclaw-1.7.0/claw/impl/socket_stream.tpp	2011-06-12 12:01:53.000000000 +0200
+++ libclaw-1.7.0new/claw/impl/socket_stream.tpp	2012-05-06 11:20:33.015835356 +0200
@@ -39,7 +39,7 @@
 ( int read_delay )
   : m_buffer(read_delay)
 {
-  init(&m_buffer);
+  this->init(&m_buffer);
 } // basic_socket_stream::basic_socket_stream()
 
 /*----------------------------------------------------------------------------*/
@@ -55,7 +55,7 @@
 ( const char* address, int port, int read_delay )
   : m_buffer(read_delay)
 {
-  init(&m_buffer);
+  this->init(&m_buffer);
   open(address, port);
 } // basic_socket_stream::basic_socket_stream()
 