diff -Naur ndiswrapper-1.3rc1.orig/driver/Makefile ndiswrapper-1.3rc1/driver/Makefile
--- ndiswrapper-1.3rc1.orig/driver/Makefile	2005-09-23 12:24:05.000000000 +0200
+++ ndiswrapper-1.3rc1/driver/Makefile	2005-09-23 12:24:40.000000000 +0200
@@ -99,6 +99,10 @@
 ntoskernel_io_exports.h: ntoskernel_io.c
 usb_exports.h: usb.c
 
+ifeq ($(CONFIG_MK8),y)
+CONFIG_X86_64=y
+endif
+
 ifeq ($(CONFIG_X86_64),y)
 OBJS += x86_64_stubs.o
 x86_64_stubs.o: x86_64_stubs.h x86_64_stubs.S
