diff -ur a/usr/src/nv/conftest.sh b/usr/src/nv/conftest.sh
--- a/usr/src/nv/conftest.sh	2011-07-12 12:05:40.000000000 -0500
+++ b/usr/src/nv/conftest.sh	2012-03-23 11:29:38.098924735 -0500
@@ -123,7 +123,7 @@
     CFLAGS="$BASE_CFLAGS $MACH_CFLAGS $OUTPUT_CFLAGS -I$HEADERS $AUTOCONF_CFLAGS"
 
     if [ "$ARCH" = "i386" -o "$ARCH" = "x86_64" ]; then
-        CFLAGS="$CFLAGS -I$SOURCES/arch/x86/include"
+        CFLAGS="$CFLAGS -I$SOURCES/arch/x86/include -I$SOURCES/arch/x86/include/generated"
     fi
     if [ -n "$BUILD_PARAMS" ]; then
         CFLAGS="$CFLAGS -D$BUILD_PARAMS"
