diff -ur a/usr/src/nv/nv-linux.h b/usr/src/nv/nv-linux.h
--- a/usr/src/nv/nv-linux.h	2013-11-27 16:47:09.000000000 -0600
+++ b/usr/src/nv/nv-linux.h	2014-05-03 22:53:03.403744368 -0500
@@ -207,6 +207,7 @@
 
 #if defined(CONFIG_ACPI) 
 #include <acpi/acpi.h>
+#include <acpi/acpi_bus.h>
 #include <acpi/acpi_drivers.h>
 #if defined(NV_ACPI_DEVICE_OPS_HAS_MATCH) || defined(ACPI_VIDEO_HID)
 #define NV_LINUX_ACPI_EVENTS_SUPPORTED 1
