diff -Naur mixxx-1.5.0.orig/src/mixxx.pro mixxx-1.5.0/src/mixxx.pro
--- mixxx-1.5.0.orig/src/mixxx.pro	2007-02-23 18:51:58.000000000 +0100
+++ mixxx-1.5.0/src/mixxx.pro	2007-04-21 14:06:04.000000000 +0200
@@ -490,8 +490,10 @@
 	SOURCES += script/python/*.cpp
 	DEFINES += __PYTHON__
 
-	LIBS += $$system(python-config)
-	exists(/usr/include/python2.4/Python.h) {
+	LIBS += $$system(python-config --libs)
+	exists(/usr/include/python2.7/Python.h) {
+		INCLUDEPATH += /usr/include/python2.7/
+	} else:exists(/usr/include/python2.4/Python.h) {
 		INCLUDEPATH += /usr/include/python2.4/
 	} else:exists(/usr/include/python2.3/Python.h) {
 		INCLUDEPATH += /usr/include/python2.3/
