--- src/nessus-plugins/Makefile.old	2006-05-22 22:24:26.000000000 +0200
+++ nessus-plugins/Makefile	2006-05-22 22:24:46.000000000 +0200
@@ -23,7 +23,7 @@
 		$(DESTDIR)${libdir}/nessus/plugins
 	
 install-nasl:	install-dirs
-	for scripts in scripts/*.nasl scripts/*.nbin; do \
+	for scripts in scripts/*.nasl; do \
 	$(INSTALL) -o $(installuser) -m 444 $$scripts \
 		$(DESTDIR)${libdir}/nessus/plugins; \
 	done
