diff -Naur pilot-link-0.11.8.orig/pilot-link.m4 pilot-link-0.11.8/pilot-link.m4
--- pilot-link-0.11.8.orig/pilot-link.m4	2006-08-25 00:45:28.000000000 +0200
+++ pilot-link-0.11.8/pilot-link.m4	2006-08-25 00:45:47.000000000 +0200
@@ -1,4 +1,4 @@
-AC_DEFUN(AC_PILOT_LINK_HOOK,
+AC_DEFUN([AC_PILOT_LINK_HOOK],
 [
 PILOT_LINK_VERS=0
 PILOT_LINK_MAJOR=11
diff -Naur pilot-link-0.11.8.orig/pilot-link.m4.in pilot-link-0.11.8/pilot-link.m4.in
--- pilot-link-0.11.8.orig/pilot-link.m4.in	2006-08-25 00:45:28.000000000 +0200
+++ pilot-link-0.11.8/pilot-link.m4.in	2006-08-25 00:45:42.000000000 +0200
@@ -1,4 +1,4 @@
-AC_DEFUN(AC_PILOT_LINK_HOOK,
+AC_DEFUN([AC_PILOT_LINK_HOOK],
 [
 PILOT_LINK_VERS=@PILOT_LINK_VERS@
 PILOT_LINK_MAJOR=@PILOT_LINK_MAJOR@
