diff -Naur grep-2.5.orig/src/Makefile.in grep-2.5/src/Makefile.in
--- grep-2.5.orig/src/Makefile.in	2002-03-13 07:23:51.000000000 -0800
+++ grep-2.5/src/Makefile.in	2002-06-06 11:39:31.000000000 -0700
@@ -354,8 +354,8 @@
 
 
 install-exec-local:
-	@SYMLINK@ @SYMLINKFROM@ $(bindir)/egrep
-	@SYMLINK@ @SYMLINKFROM@ $(bindir)/fgrep
+	@SYMLINK@ @SYMLINKFROM@ $(DESTDIR)$(bindir)/egrep
+	@SYMLINK@ @SYMLINKFROM@ $(DESTDIR)$(bindir)/fgrep
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
