diff -Naur hk_classes-0.8.3/hk_dbaseclasses/xbase/xbdbf.cpp hk_classes-0.8.3.new/hk_dbaseclasses/xbase/xbdbf.cpp
--- hk_classes-0.8.3/hk_dbaseclasses/xbase/xbdbf.cpp	2006-04-24 20:04:30.000000000 +0200
+++ hk_classes-0.8.3.new/hk_dbaseclasses/xbase/xbdbf.cpp	2012-06-04 17:43:06.000000000 +0200
@@ -50,6 +50,7 @@
 #include <io.h>
 #endif
 #include <errno.h>
+#include <unistd.h>
 
 /*! \file xbdbf.cpp
 */
diff -Naur hk_classes-0.8.3/hk_dbaseclasses/xbase/xblock.cpp hk_classes-0.8.3.new/hk_dbaseclasses/xbase/xblock.cpp
--- hk_classes-0.8.3/hk_dbaseclasses/xbase/xblock.cpp	2006-04-24 20:04:30.000000000 +0200
+++ hk_classes-0.8.3.new/hk_dbaseclasses/xbase/xblock.cpp	2012-06-04 17:54:48.000000000 +0200
@@ -59,6 +59,7 @@
 
 //#include <stdio.h>
 //#include <stdlib.h>
+#include <unistd.h>
 
 /*! \file xblock.cpp
 */
