diff -Naur wordpress-orig/xmlrpc.php wordpress/xmlrpc.php
--- wordpress-orig/xmlrpc.php	2007-05-11 23:40:15.000000000 +0200
+++ wordpress/xmlrpc.php	2007-06-13 09:18:01.000000000 +0200
@@ -538,7 +538,7 @@
 		$username				= $args[1];
 		$password				= $args[2];
 		$category				= $args[3];
-		$max_results			= $args[4];
+		$max_results			= (int)$args[4];
 
 		if(!$this->login_pass_ok($username, $password)) {
 			return($this->error);
