diff -Naur wordpress-orig/wp-includes/query.php wordpress/wp-includes/query.php
--- wordpress-orig/wp-includes/query.php	2007-12-28 02:04:30.000000000 +0100
+++ wordpress/wp-includes/query.php	2008-04-29 07:31:40.000000000 +0200
@@ -453,6 +453,7 @@
 		$qv['day'] = (int) $qv['day'];
 		$qv['w'] = (int) $qv['w'];
 		$qv['m'] =  (int) $qv['m'];
+		$qv['cat'] = preg_replace( '|[^0-9,-]|', '', $qv['cat'] ); // comma separated list of positive or negative integers
 		if ( '' !== $qv['hour'] ) $qv['hour'] = (int) $qv['hour'];
 		if ( '' !== $qv['minute'] ) $qv['minute'] = (int) $qv['minute'];
 		if ( '' !== $qv['second'] ) $qv['second'] = (int) $qv['second'];
diff -Naur wordpress-orig/wp-includes/theme.php wordpress/wp-includes/theme.php
--- wordpress-orig/wp-includes/theme.php	2007-09-13 00:19:26.000000000 +0200
+++ wordpress/wp-includes/theme.php	2008-04-29 07:32:05.000000000 +0200
@@ -343,8 +343,8 @@
 
 function get_category_template() {
 	$template = '';
-	if ( file_exists(TEMPLATEPATH . "/category-" . get_query_var('cat') . '.php') )
-		$template = TEMPLATEPATH . "/category-" . get_query_var('cat') . '.php';
+	if ( file_exists(TEMPLATEPATH . "/category-" . absint( get_query_var('cat') ) . '.php') )
+		$template = TEMPLATEPATH . "/category-" . absint( get_query_var('cat') ) . '.php';
 	elseif ( file_exists(TEMPLATEPATH . "/category.php") )
 		$template = TEMPLATEPATH . "/category.php";
 
