diff -p -up metisse-0.4.1/fvwm-insitu/fvwm/builtins.c.format-security metisse-0.4.1/fvwm-insitu/fvwm/builtins.c
--- metisse-0.4.1/fvwm-insitu/fvwm/builtins.c.format-security	2008-12-17 13:33:54.000000000 +0100
+++ metisse-0.4.1/fvwm-insitu/fvwm/builtins.c	2008-12-17 13:34:50.000000000 +0100
@@ -2850,7 +2850,7 @@ void CMD_Echo(F_CMD_ARGS)
 			action[len-1]='\0';
 		}
 	}
-	fvwm_msg(ECHO,"Echo",action);
+	fvwm_msg(ECHO,"Echo","%s",action);
 
 	return;
 }
diff -p -up metisse-0.4.1/fvwm-insitu/libs/FRenderInit.c.format-security metisse-0.4.1/fvwm-insitu/libs/FRenderInit.c
--- metisse-0.4.1/fvwm-insitu/libs/FRenderInit.c.format-security	2008-12-17 13:32:00.000000000 +0100
+++ metisse-0.4.1/fvwm-insitu/libs/FRenderInit.c	2008-12-17 13:32:24.000000000 +0100
@@ -129,7 +129,7 @@ Bool FRenderGetErrorText(int code, char 
 		    code <= FRenderErrorBase +
 		    (sizeof(error_names) / sizeof(char *)) -1)
 		{
-			sprintf(msg, error_names[code - FRenderErrorBase]);
+			sprintf(msg, "%s", error_names[code - FRenderErrorBase]);
 			return 1;
 		}
 	}
diff -p -up metisse-0.4.1/fvwm-insitu/modules/FvwmIconMan/functions.c.format-security metisse-0.4.1/fvwm-insitu/modules/FvwmIconMan/functions.c
--- metisse-0.4.1/fvwm-insitu/modules/FvwmIconMan/functions.c.format-security	2008-12-17 13:39:49.000000000 +0100
+++ metisse-0.4.1/fvwm-insitu/modules/FvwmIconMan/functions.c	2008-12-17 13:43:59.000000000 +0100
@@ -522,7 +522,7 @@ int builtin_ret(int numargs, BuiltinArg 
 
 int builtin_print(int numargs, BuiltinArg *args)
 {
-	char *s, buf[256];
+	char *s;
 
 	ConsoleDebug(FUNCTIONS, "print: %s\n", args[0].value.string_value);
 
@@ -533,8 +533,7 @@ int builtin_print(int numargs, BuiltinAr
 	}
 	else
 	{
-		sprintf(buf, "%s\n", s);
-		ConsoleMessage(buf);
+		ConsoleMessage("%s\n",s);
 	}
 
 	return 0;
diff -p -up metisse-0.4.1/xserver/metisse/init.c.format-security metisse-0.4.1/xserver/metisse/init.c
--- metisse-0.4.1/xserver/metisse/init.c.format-security	2008-12-17 14:16:14.000000000 +0100
+++ metisse-0.4.1/xserver/metisse/init.c	2008-12-17 14:17:05.000000000 +0100
@@ -1124,7 +1124,7 @@ void rfbLog(char *format, ...)
 
     time(&clock);
     strftime(buf, 255, "%F %T ", localtime(&clock));
-    fprintf(stderr, buf);
+    fprintf(stderr, "%s", buf);
 
     vfprintf(stderr, format, args);
     fflush(stderr);
