diff -Naur metisse-0.4.1/FvwmCompositor/main/AWidgets.cxx metisse-0.4.1/FvwmCompositor/main/AWidgets.cxx
--- metisse-0.4.1/FvwmCompositor/main/AWidgets.cxx	2008-06-30 10:11:59.000000000 +0200
+++ metisse-0.4.1/FvwmCompositor/main/AWidgets.cxx	2009-06-15 21:28:36.000000000 +0200
@@ -18,6 +18,8 @@
 #include "main/AScreen.H"
 #include "desktop/MetisseWindow.H"
 
+#include <cstdio>
+
 
 // --------------------------------------------------------------------
 
diff -Naur metisse-0.4.1/FvwmCompositor/main/ExtendedInputs.cxx metisse-0.4.1/FvwmCompositor/main/ExtendedInputs.cxx
--- metisse-0.4.1/FvwmCompositor/main/ExtendedInputs.cxx	2008-07-02 14:23:32.000000000 +0200
+++ metisse-0.4.1/FvwmCompositor/main/ExtendedInputs.cxx	2009-06-15 21:33:30.000000000 +0200
@@ -18,6 +18,8 @@
 #include <nucleo/gl/window/glWindow.H>
 #include <nucleo/gl/glUtils.H>
 
+#include <cstdio>
+
 using namespace nucleo ;
 
 int AExtDevice::handleEvents(glWindow::event e)
diff -Naur metisse-0.4.1/FvwmCompositor/main/LayerManager.cxx metisse-0.4.1/FvwmCompositor/main/LayerManager.cxx
--- metisse-0.4.1/FvwmCompositor/main/LayerManager.cxx	2008-08-25 15:54:44.000000000 +0200
+++ metisse-0.4.1/FvwmCompositor/main/LayerManager.cxx	2009-06-15 21:37:32.000000000 +0200
@@ -17,6 +17,7 @@
 #include "LayerManager.H"
 
 #include <X11/X.h>
+#include <cstdio>
 
 #include "main/AScreen.H"
 #include "desktop/MetisseWindow.H"
diff -Naur metisse-0.4.1/FvwmCompositor/main/WidgetManager.cxx metisse-0.4.1/FvwmCompositor/main/WidgetManager.cxx
--- metisse-0.4.1/FvwmCompositor/main/WidgetManager.cxx	2008-06-26 14:32:30.000000000 +0200
+++ metisse-0.4.1/FvwmCompositor/main/WidgetManager.cxx	2009-06-15 21:28:18.000000000 +0200
@@ -22,6 +22,7 @@
 
 #include <iostream>
 #include <vector>
+#include <cstdio>
 //#include <sstream>
 #include <stdexcept>
 
diff -Naur metisse-0.4.1/FvwmCompositor/renderer/goodies.cxx metisse-0.4.1/FvwmCompositor/renderer/goodies.cxx
--- metisse-0.4.1/FvwmCompositor/renderer/goodies.cxx	2008-06-26 14:32:30.000000000 +0200
+++ metisse-0.4.1/FvwmCompositor/renderer/goodies.cxx	2009-06-15 21:27:56.000000000 +0200
@@ -17,6 +17,7 @@
 #include "goodies.H"
 
 #include <math.h>
+#include <cstdio>
 
 #include "fvwmmodule/AFvwm.H"
 #include "FoldablePolygon.H"
