--- vector-0.13.1.0/vector.cabal	2001-09-09 03:46:40.000000000 +0200
+++ vector-0.13.1.0.new/vector.cabal	2024-07-10 01:27:41.889913609 +0200
@@ -145,7 +145,7 @@
   Install-Includes:
         vector.h
 
-  Build-Depends: base >= 4.9 && < 4.20
+  Build-Depends: base >= 4.9 && < 4.21
                , primitive >= 0.6.4.0 && < 0.10
                , deepseq >= 1.1 && < 1.6
                , vector-stream >= 0.1 && < 0.2
