--- vector-0.12.0.3/vector.cabal	2001-09-09 03:46:40.000000000 +0200
+++ vector-0.12.0.3.new/vector.cabal	2019-11-26 14:08:12.478629070 +0100
@@ -1,5 +1,5 @@
 Name:           vector
-Version:        0.12.0.3
+Version:        0.12.0.1
 -- don't forget to update the changelog file!
 License:        BSD3
 License-File:   LICENSE
@@ -39,7 +39,10 @@
   GHC == 7.6.3,
   GHC == 7.8.4,
   GHC == 7.10.3,
-  GHC == 8.0.1
+  GHC == 8.0.2,
+  GHC == 8.2.2,
+  GHC == 8.4.4,
+  GHC == 8.6.4
 
 Cabal-Version:  >=1.10
 Build-Type:     Simple
@@ -144,13 +147,13 @@
   Install-Includes:
         vector.h
 
-  Build-Depends: base >= 4.5 && < 4.13
+  Build-Depends: base >= 4.5 && < 4.14
                , primitive >= 0.5.0.1 && < 0.8
                , ghc-prim >= 0.2 && < 0.6
                , deepseq >= 1.1 && < 1.5
   if !impl(ghc > 8.0)
     Build-Depends: fail == 4.9.*
-                 , semigroups >= 0.18 && < 0.19
+                 , semigroups >= 0.18 && < 0.20
 
   Ghc-Options: -O2 -Wall
 
@@ -189,8 +192,8 @@
 
   hs-source-dirs: tests
   Build-Depends: base >= 4.5 && < 5, template-haskell, base-orphans >= 0.6, vector,
-                 random, primitive,
-                 QuickCheck >= 2.9 && < 2.10 , HUnit, test-framework,
+                 primitive, random,
+                 QuickCheck >= 2.9 && < 2.12 , HUnit, test-framework,
                  test-framework-hunit, test-framework-quickcheck2,
                  transformers >= 0.2.0.0
 
@@ -227,8 +230,8 @@
 
   hs-source-dirs: tests
   Build-Depends: base >= 4.5 && < 5, template-haskell, base-orphans >= 0.6, vector,
-                 random, primitive,
-                 QuickCheck >= 2.9 && < 2.10 , HUnit, test-framework,
+                 primitive, random,
+                 QuickCheck >= 2.9 && < 2.12 , HUnit, test-framework,
                  test-framework-hunit, test-framework-quickcheck2,
                  transformers >= 0.2.0.0
 
