From d041deff7db7028a061aff8ad9a59e9f1c640d6b Mon Sep 17 00:00:00 2001
From: Daniel Exner <dex@dragonslave.de>
Date: Mon, 21 Oct 2013 20:59:58 +0200
Subject: [PATCH] * initial FrugalWare support

---
 appliance/packagelist.in | 60 ++++++++++++++++++++++++++++++++++++++++++++++++
 configure.ac             |  3 +++
 2 files changed, 63 insertions(+)

diff --git a/appliance/packagelist.in b/appliance/packagelist.in
index aecf55a..3158a09 100644
--- a/appliance/packagelist.in
+++ b/appliance/packagelist.in
@@ -113,6 +113,7 @@
   zfs-fuse
 #endif /* ARCHLINUX */
 
+#ifndef FRUGALWARE
 acl
 attr
 bash
@@ -167,6 +168,65 @@ zerofree
 #ifdef VALGRIND_DAEMON
 valgrind
 #endif
+#endif /* FRUGALWARE */
 
 /* Define this by doing: ./configure --with-extra-packages="..." */
 EXTRA_PACKAGES
+
+#ifdef FRUGALWARE
+augeas
+btrfs-progs
+cryptsetup-luks
+e2fsprogs
+cdrkit
+grub2
+hfsplus
+iproute2
+iputils
+kernel
+libcap
+ntfsprogs
+ntfs-3g
+openssh
+pcre
+reiserfsprogs
+syslinux
+systemd
+vim
+xz
+yajl
+xfsprogs-acl
+xfsprogs-attr
+bash
+binutils
+bzip2
+coreutils
+cpio
+diffutils
+dosfstools
+file
+findutils
+gawk
+gptfdisk
+grep
+gzip
+jfsutils
+kmod
+less
+libxml2
+lsof
+lsscsi
+lvm2
+mdadm
+module-init-tools
+parted
+procps
+psmisc
+rsync
+sed
+strace
+syslinux
+tar
+util-linux
+xfsprogs
+#endif /* FRUGALWARE */
diff --git a/configure.ac b/configure.ac
index 603c4af..39269ef 100644
--- a/configure.ac
+++ b/configure.ac
@@ -557,6 +557,9 @@ fi
 if test -f /etc/SuSE-release; then
     DISTRO=SUSE
 fi
+if test -f /etc/frugalware-release; then
+    DISTRO=FRUGALWARE
+fi
 AC_MSG_RESULT([$DISTRO])
 AC_SUBST([DISTRO])
 
-- 
1.8.4

