diff -Naur gnome-mag-0.14.5.orig/magnifier/x11/gmag-graphical-server.c gnome-mag-0.14.5/magnifier/x11/gmag-graphical-server.c
--- gnome-mag-0.14.5.orig/magnifier/x11/gmag-graphical-server.c	2007-06-18 17:36:13.000000000 +0200
+++ gnome-mag-0.14.5/magnifier/x11/gmag-graphical-server.c	2007-06-18 17:36:23.000000000 +0200
@@ -68,7 +68,7 @@
 static Display           *dpy = NULL;
 static guint              dpy_gsource = 0;
 static Window             root_window, mag_window;
-static gboolean           use_damage, use_compositor;
+static gboolean           use_damage, use_compositor, use_windows;
 static gint               fixes_event_base = 0;
 static DamageHandlerFunc *damage_events_handler = NULL;
 static gint               damage_event_base = 0;
