--- valencia-0.3.1/Makefile.old	2010-12-27 17:29:42.000000000 +0100
+++ valencia-0.3.1/Makefile	2010-12-27 17:49:10.000000000 +0100
@@ -28,7 +28,7 @@
 DIST_TAR_BZ2 = $(DIST_TAR).bz2
 DIST_TAR_GZ = $(DIST_TAR).gz
 
-ICON_DIR = ~/.local/share/icons/hicolor/128x128/apps
+ICON_DIR = $(prefix)/share/icons/hicolor/128x128/apps
 
 all: valencia.gedit-plugin libvalencia.so
 	
@@ -42,9 +42,8 @@
 	$(VALAC) $(VFLAGS) -X --shared -X -fPIC --vapidir=. $(PACKAGES) $(SOURCES) -o $@
 
 install: libvalencia.so valencia.gedit-plugin
-	@ [ `whoami` != "root" ] || ( echo 'Run make install as yourself, not as root.' ; exit 1 )
-	mkdir -p ~/.gnome2/gedit/plugins
-	cp $(OUTPUTS) ~/.gnome2/gedit/plugins
+	mkdir -p $(prefix)/lib/gedit-2/plugins
+	cp $(OUTPUTS) $(prefix)/lib/gedit-2/plugins
 	mkdir -p $(ICON_DIR)
 	cp -p valencia.png $(ICON_DIR)
 
