--- lxdm.orig/data/lxdm.conf.in	2010-02-23 23:32:31.000000000 +0100
+++ lxdm/data/lxdm.conf.in	2010-02-24 00:52:27.000000000 +0100
@@ -5,11 +5,11 @@
 greeter=@FULL_LIBEXECDIR@/lxdm-greeter-gtk
 
 [server]
-# arg=/usr/bin/X -nr vt1
+#arg=/usr/bin/X -nr vt12
 
 [display]
 gtk_theme=Clearlooks
-bg=/usr/share/backgrounds/default.png
+bg=/usr/share/lxde/wallpapers/frugalware.png
 bottom_pane=1
 lang=1
 theme=Industrial

