--- lxdm-0.3.0/data/lxdm..conf.in.old	2010-09-24 17:43:49.000000000 +0200
+++ lxdm-0.3.0/data/lxdm.conf.in	2011-04-16 16:31:22.000000000 +0200
@@ -8,11 +8,11 @@
 # arg=/usr/bin/X -nr vt1
 
 [display]
-gtk_theme=Clearlooks
-bg=/usr/share/backgrounds/default.png
+#gtk_theme=Adwaita
+bg=/usr/share/lxde/wallpapers/frugalware.png
 bottom_pane=1
 lang=1
-theme=Industrial
+#theme=Industrial
 
 [input]
 
