--- lxdm-0.4.1/data/lxdm.conf	2011-07-29 15:44:51.000000000 +0200
+++ lxdm-0.4.1new/data/lxdm.conf	2011-11-26 11:32:51.000000000 +0100
@@ -16,7 +16,7 @@
 # xauth_path=/tmp
 
 ## greeter used to welcome the user
-greeter=/usr/local/libexec/lxdm-greeter-gtk
+greeter=/usr/libexec/lxdm-greeter-gtk
 
 [server]
 ## arg used to start xserver, not fully function
@@ -27,7 +27,7 @@
 gtk_theme=Clearlooks
 
 ## background of the greeter
-bg=/usr/share/backgrounds/default.png
+bg=/usr/share/lxdm/themes/Frugalware/wall.png
 
 ## if show bottom pane
 bottom_pane=1
