From 3eb70bffb730e147ba743dc4ffc7c053f2c9db42 Mon Sep 17 00:00:00 2001
From: dgod <dgod.osa@gmail.com>
Date: Fri, 22 Jul 2011 20:12:23 +0800
Subject: [PATCH] fix lost data entry when install

---
 data/Makefile.am |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/data/Makefile.am b/data/Makefile.am
index aabe9e6..6d3fad6 100644
--- a/data/Makefile.am
+++ b/data/Makefile.am
@@ -46,6 +46,9 @@ theme_DATA= \
 	themes/Industrial/lang.png \
 	themes/Industrial/session.png \
 	themes/Industrial/greeter.ui \
+	themes/Industrial/greeter-gtk3.ui \
+	themes/Industrial/gtk.css \
+	themes/Industrial/nobody.png \
 	$(NULL)
 
 EXTRA_DIST= \
-- 
1.7.0.1


