diff -Naur lxdm-0.5.0/data/lxdm.conf lxdm-0.5.0b/data/lxdm.conf
--- lxdm-0.5.0/data/lxdm.conf	2012-05-24 15:13:44.000000000 +0200
+++ lxdm-0.5.0b/data/lxdm.conf	2014-12-10 09:58:36.215280541 +0100
@@ -31,10 +31,10 @@
 
 [display]
 ## gtk theme used by greeter
-gtk_theme=Clearlooks
+gtk_theme=Frugalware
 
 ## background of the greeter
-bg=/usr/share/backgrounds/default.png
+bg=/usr/share/lxdm/themes/Frugalware/background.png
 
 ## if show bottom pane
 bottom_pane=1
@@ -43,10 +43,10 @@
 transparent_pane=0
 
 ## if show sessions
-show_sessions=0
+show_sessions=1
 
 ## if show language select control
-lang=1
+lang=0
 
 ## if show keyboard layout select control
 keyboard=0
@@ -58,7 +58,7 @@
 show_time=1
 
 ## the theme of greeter
-theme=Industrial
+theme=Frugalware
 
 [input]
 
@@ -70,5 +70,5 @@
 white=
 
 ## blacklist user
-black=
+black=ftp syncpkgd rsyncd partimag
 
diff -Naur lxdm-0.5.0/data/lxdm.in lxdm-0.5.0b/data/lxdm.in
--- lxdm-0.5.0/data/lxdm.in	2011-11-03 13:52:31.000000000 +0100
+++ lxdm-0.5.0b/data/lxdm.in	2014-12-10 09:58:15.061584245 +0100
@@ -1,31 +1,23 @@
 #!/bin/sh
 
-[ -f /etc/sysconfig/i18n ] && . /etc/sysconfig/i18n
-
-if [ -z "$LANG" -a -e /etc/sysconfig/language ]; then
-	. /etc/sysconfig/language
-	if [ -n "$RC_LANG"]; then
-		LANG=$RC_LANG
-	fi
+if [ -f /etc/locale.conf ]; then
+    . /etc/locale.conf
 fi
 
 if [ -n "$LANG" ]; then
-	export LANG
+    export LANG
 fi
 
-[ -f /etc/sysconfig/desktop ] && . /etc/sysconfig/desktop
-[ -f /etc/sysconfig/windowmanager ] && . /etc/sysconfig/windowmanager
-
 if [ -n "$DEFAULT_WM" ]; then
-	PREFERRED=$DEFAULT_WM
+    PREFERRED=$DEFAULT_WM
 fi
 
 if [ -n "$DESKTOP" ]; then
-	export DESKTOP
+    export DESKTOP
 fi
 
 if [ -n "$PREFERRED" ]; then
-	export PREFERRED
+    export PREFERRED
 fi
 
-exec @prefix@/sbin/lxdm-binary $*
+exec @prefix@/sbin/lxdm-binary $* >/dev/null 2>&1
