diff -urN sabayon-2.12.3.orig/admin-tool/sabayon.desktop sabayon-2.12.3/admin-tool/sabayon.desktop
--- sabayon-2.12.3.orig/admin-tool/sabayon.desktop	2006-01-30 22:10:59.000000000 +0000
+++ sabayon-2.12.3/admin-tool/sabayon.desktop	2006-03-21 18:24:13.000000000 +0000
@@ -12,9 +12,7 @@
 Name[hu]=Profilszerkesző használata
 Name[ja]=プロファイル・エディタ
 Name[lt]=Naudotojų profilių redaktorius
-Name[mk]=Уредувач на профили за корисници
 Name[nb]=Redigering av brukerprofiler
-Name[nl]=Profieleditor
 Name[no]=Redigering av brukerprofiler
 Name[pa]=ਉਪਭੋਗੀ ਪਰੋਫਾਇਲ ਸੰਪਾਦਕ
 Name[sr]=Уређивач корисничких поставки
@@ -40,10 +38,8 @@
 Comment[hu]=Felhasználók profiljainak létesítése és szerkesztése
 Comment[ja]=ユーザ向けのプロファイルを確立したり編集します
 Comment[lt]=Kurti ir redaguoti naudotojų profilius
-Comment[mk]=Формирање и уредување на Профили за Корисниците
 Comment[nb]=Opprett og rediger brukerprofiler
 Comment[ne]=प्रयोगकर्ताहरु को लागि प्रोफाइलहरु स्थापना र सम्पादन गर्नुहोस
-Comment[nl]=Profielen maken en bewerken
 Comment[no]=Opprett og rediger brukerprofiler
 Comment[sr]=Образује и уређује поставке за кориснике
 Comment[sr@Latn]=Obrazuje i uređuje postavke za korisnike
@@ -54,9 +50,9 @@
 Comment[zh_CN]=为用户建立并编辑配置文件
 Comment[zh_HK]=建立與編輯使用者設定組合
 Comment[zh_TW]=建立與編輯使用者設定組合
-Exec=sabayon
+Exec=gksu sabayon
 Terminal=false
 Type=Application
 Icon=sabayon.png
 StartupNotify=true
-Categories=GNOME;Application;SystemSetup;
+Categories=Categories=GNOME;GTK;Application;System;Settings;
diff -urN sabayon-2.12.3.orig/admin-tool/sabayon.desktop.in sabayon-2.12.3/admin-tool/sabayon.desktop.in
--- sabayon-2.12.3.orig/admin-tool/sabayon.desktop.in	2005-10-27 14:54:17.000000000 +0100
+++ sabayon-2.12.3/admin-tool/sabayon.desktop.in	2006-03-21 18:24:07.000000000 +0000
@@ -2,9 +2,9 @@
 Encoding=UTF-8
 _Name=User Profile Editor
 _Comment=Establish and Edit Profiles for Users
-Exec=sabayon
+Exec=gksu sabayon
 Terminal=false
 Type=Application
 Icon=sabayon.png
 StartupNotify=true
-Categories=GNOME;Application;SystemSetup;
+Categories=Categories=GNOME;GTK;Application;System;Settings;
diff -urN sabayon-2.12.3.orig/configure sabayon-2.12.3/configure
--- sabayon-2.12.3.orig/configure	2006-01-31 17:08:48.000000000 +0000
+++ sabayon-2.12.3/configure	2006-03-21 18:22:04.000000000 +0000
@@ -24889,6 +24889,27 @@
   with_distro="mandriva"
 fi
 
+  echo "$as_me:$LINENO: checking for /etc/frugalware-release" >&5
+echo $ECHO_N "checking for /etc/frugalware-release... $ECHO_C" >&6
+if test "${ac_cv_file__etc_frugalware_release+set}" = set; then
+  echo $ECHO_N "(cached) $ECHO_C" >&6
+else
+  test "$cross_compiling" = yes &&
+  { { echo "$as_me:$LINENO: error: cannot check for file existence when cross compiling" >&5
+echo "$as_me: error: cannot check for file existence when cross compiling" >&2;}
+   { (exit 1); exit 1; }; }
+if test -r "/etc/frugalware-release"; then
+  ac_cv_file__etc_frugalware_release=yes
+else
+  ac_cv_file__etc_frugalware_release=no
+fi
+fi
+echo "$as_me:$LINENO: result: $ac_cv_file__etc_frugalware_release" >&5
+echo "${ECHO_T}$ac_cv_file__etc_frugalware_release" >&6
+if test $ac_cv_file__etc_frugalware_release = yes; then
+  with_distro="frugalware"
+fi
+
 fi
 with_distro=`echo ${with_distro} | tr '[:upper:]' '[:lower:]' `
 
@@ -24932,6 +24953,12 @@
       install_xinitrc_script="yes"
       XINITRC_SYSCONFDIR="X11/xinit.d"
       ;;
+    frugalware)
+      XSESSION="/etc/gdm/Xsession"
+      SESSION_NAME="gnome-session"
+      install_xinitrc_script="yes"
+      XINITRC_SYSCONFDIR="X11/xinit/xinitrc.d"
+      ;;
     *)
       echo "Your distribution (${with_distro}) is not yet supported!  (patches welcome)"
       exit 1
diff -urN sabayon-2.12.3.orig/configure.ac sabayon-2.12.3/configure.ac
--- sabayon-2.12.3.orig/configure.ac	2006-01-27 00:23:12.000000000 +0000
+++ sabayon-2.12.3/configure.ac	2006-03-21 18:21:40.000000000 +0000
@@ -99,6 +99,7 @@
   AC_CHECK_FILE(/etc/SuSE-release, with_distro="suse")
   AC_CHECK_FILE(/etc/mandriva-release, with_distro="mandriva")
   AC_CHECK_FILE(/etc/mandrakelinux-release, with_distro="mandriva")
+  AC_CHECK_FILE(/etc/frugalware-release, with_distro="frugalware")
 fi
 with_distro=`echo ${with_distro} | tr '[[:upper:]]' '[[:lower:]]' `
 
@@ -142,6 +143,12 @@
       install_xinitrc_script="yes"
       XINITRC_SYSCONFDIR="X11/xinit.d"
       ;;
+    frugalware)
+      XSESSION="/etc/gdm/Xsession"
+      SESSION_NAME="gnome-session"
+      install_xinitrc_script="yes"
+      XINITRC_SYSCONFDIR="X11/xinit/xinitrc.d"
+      ;;
     *)
       echo "Your distribution (${with_distro}) is not yet supported!  (patches welcome)"
       exit 1
diff -urN sabayon-2.12.3.orig/lib/config.py sabayon-2.12.3/lib/config.py
--- sabayon-2.12.3.orig/lib/config.py	2006-01-31 17:09:10.000000000 +0000
+++ sabayon-2.12.3/lib/config.py	2006-03-21 18:52:57.000000000 +0000
@@ -22,11 +22,11 @@
 
 PACKAGE    = "sabayon"
 VERSION    = "2.12.3"
-LOCALEDIR = "/opt/gnome2/share/locale"
-DATADIR    = "/opt/gnome2/share"
-SYSCONFDIR = "/opt/gnome2/etc"
-LIBEXECDIR = "/opt/gnome2/libexec"
-SBINDIR    = "/opt/gnome2/sbin"
+LOCALEDIR = "/usr/share/locale"
+DATADIR    = "/usr/share"
+SYSCONFDIR = "/usr/etc"
+LIBEXECDIR = "/usr/libexec"
+SBINDIR    = "/usr/sbin"
 
 GLADEDIR    = os.path.join (DATADIR, "sabayon", "glade")
 PROFILESDIR = os.path.join (SYSCONFDIR, "desktop-profiles")
@@ -53,8 +53,8 @@
 NOLOGIN_SHELL = "/sbin/nologin"
 SKEL_HOMEDIR  = "/etc/skel"
 
-XNEST_ARGV   = [ "/usr/X11R6/bin/Xnest", "-terminate", "-audit", "0", "-nolisten", "tcp" ]
-SESSION_ARGV = [ "/etc/X11/xdm/Xsession", "gnome" ]
+XNEST_ARGV   = [ "/usr/bin/Xnest", "-terminate", "-audit", "0", "-nolisten", "tcp" ]
+SESSION_ARGV = [ "/etc/gdm/Xsession", "gnome-session" ]
 USERMOD_ARGV = [ "/usr/sbin/usermod" ]
 
 PANEL_KEY_BASE = "/apps/panel"
