diff -Naur qscintilla-1.65-gpl-1.6.orig/qt/qscintilla.pro qscintilla-1.65-gpl-1.6/qt/qscintilla.pro
--- qscintilla-1.65-gpl-1.6.orig/qt/qscintilla.pro	2006-02-08 19:02:13.091234750 +0100
+++ qscintilla-1.65-gpl-1.6/qt/qscintilla.pro	2006-02-08 19:02:21.815780000 +0100
@@ -25,7 +25,7 @@
 
 TEMPLATE = lib
 TARGET = qscintilla
-DESTDIR = $(QTDIR)/lib
+DESTDIR = $(INSTALL_ROOT)/$(QTDIR)/lib
 CONFIG += qt warn_off release dll thread
 INCLUDEPATH = . ../include ../src
 DEFINES = QEXTSCINTILLA_MAKE_DLL QT SCI_LEXER
