diff -Naur libmypaint-1.3.0/autogen.sh libmypaint-1.3.0-p/autogen.sh
--- libmypaint-1.3.0/autogen.sh	2016-12-01 23:47:14.000000000 +0100
+++ libmypaint-1.3.0-p/autogen.sh	2018-07-27 16:37:13.201186700 +0200
@@ -9,14 +9,14 @@
 # tools and you shouldn't use this script.  Just call ./configure
 # directly.
 
-ACLOCAL=${ACLOCAL-aclocal-1.13}
+ACLOCAL=${ACLOCAL-aclocal-1.16}
 AUTOCONF=${AUTOCONF-autoconf}
 AUTOHEADER=${AUTOHEADER-autoheader}
-AUTOMAKE=${AUTOMAKE-automake-1.13}
+AUTOMAKE=${AUTOMAKE-automake-1.16}
 LIBTOOLIZE=${LIBTOOLIZE-libtoolize}
 
 AUTOCONF_REQUIRED_VERSION=2.62
-AUTOMAKE_REQUIRED_VERSION=1.13
+AUTOMAKE_REQUIRED_VERSION=1.16
 INTLTOOL_REQUIRED_VERSION=0.40.1
 LIBTOOL_REQUIRED_VERSION=1.5
 LIBTOOL_WIN32_REQUIRED_VERSION=2.2
@@ -127,9 +127,9 @@
 if ($AUTOMAKE --version) < /dev/null > /dev/null 2>&1; then
    AUTOMAKE=$AUTOMAKE
    ACLOCAL=$ACLOCAL
-elif (automake-1.15 --version) < /dev/null > /dev/null 2>&1; then
-   AUTOMAKE=automake-1.15
-   ACLOCAL=aclocal-1.15
+elif (automake-1.16 --version) < /dev/null > /dev/null 2>&1; then
+   AUTOMAKE=automake-1.16
+   ACLOCAL=aclocal-1.16
 elif (automake-1.14 --version) < /dev/null > /dev/null 2>&1; then
    AUTOMAKE=automake-1.14
    ACLOCAL=aclocal-1.14
