diff -Naur vlc-0.8.4a/modules/gui/skins2/utils/var_tree.hpp vlc-0.8.4a-gcc41/modules/gui/skins2/utils/var_tree.hpp
--- vlc-0.8.4a/modules/gui/skins2/utils/var_tree.hpp	2005-12-11 22:36:20.000000000 +0100
+++ vlc-0.8.4a-gcc41/modules/gui/skins2/utils/var_tree.hpp	2006-05-04 13:12:04.000000000 +0200
@@ -88,7 +88,7 @@
 
         /// Parent node
         VarTree *parent() { return m_pParent; }
-        void VarTree::checkParents( VarTree *pParent );
+        void checkParents( VarTree *pParent );
 
         Iterator uncle();
 
