--- vlc-0.8.2/modules/access/smb.c~	2005-06-25 15:43:11.000000000 +0200
+++ vlc-0.8.2/modules/access/smb.c	2005-09-06 16:53:32.000000000 +0200
@@ -301,7 +301,7 @@
     access_sys_t *p_sys = p_access->p_sys;
 
 #ifdef USE_CTX
-    p_sys->p_smb->close( p_sys->p_smb, p_sys->p_file );
+    p_sys->p_smb->close_fn( p_sys->p_smb, p_sys->p_file );
     smbc_free_context( p_sys->p_smb, 1 );
 #else
     smbc_close( p_sys->i_smb );
