diff -Naur sylpheed-3.7.0/configure sylpheed-3.7.0.new/configure
--- sylpheed-3.7.0/configure	2018-01-30 10:06:04.000000000 +0100
+++ sylpheed-3.7.0.new/configure	2023-02-24 09:57:44.035700824 +0100
@@ -16307,9 +16307,9 @@
 
 $as_echo "#define USE_ENCHANT 1" >>confdefs.h
 
-			LIBS="$LIBS `$PKG_CONFIG --libs enchant`"
+			LIBS="$LIBS `$PKG_CONFIG --libs enchant-2`"
 
-						if $PKG_CONFIG --atleast-version 1.4.2 enchant ; then
+						if $PKG_CONFIG --atleast-version 1.4.2 enchant-2 ; then
 				CFLAGS="$CFLAGS `$PKG_CONFIG --cflags dbus-glib-1`"
 				LIBS="$LIBS `$PKG_CONFIG --libs dbus-glib-1`"
 			fi
diff -Naur sylpheed-3.7.0/src/compose.c sylpheed-3.7.0.new/src/compose.c
--- sylpheed-3.7.0/src/compose.c	2017-11-17 01:58:43.000000000 +0100
+++ sylpheed-3.7.0.new/src/compose.c	2023-02-24 09:57:42.994692821 +0100
@@ -71,7 +71,7 @@
 #  include <gtk/gtkradiomenuitem.h>
 #  include <gtkspell/gtkspell.h>
 #if USE_ENCHANT
-#  include <enchant/enchant.h>
+#  include <enchant-2/enchant.h>
 #else
 #  include <aspell.h>
 #endif
