--- hslua-typing-0.1.0/hslua-typing.cabal	2001-09-09 03:46:40.000000000 +0200
+++ hslua-typing-0.1.0.new/hslua-typing.cabal	2023-12-10 22:16:55.579692556 +0100
@@ -57,7 +57,7 @@
   build-depends:       containers           >= 0.5.9  && < 0.7
                      , hslua-core           >= 2.3    && < 2.4
                      , hslua-marshalling    >= 2.3    && < 2.4
-                     , text                 >= 1.2    && < 2.1
+                     , text                 >= 1.2    && < 2.2
   default-language:    Haskell2010
 
 test-suite test-hslua-typing
