diff -Naur FLU_2.14/FLU/Flu_Tree_Browser.h FLU_2.14.new/FLU/Flu_Tree_Browser.h
--- FLU_2.14/FLU/Flu_Tree_Browser.h	2004-11-05 22:33:20.000000000 +0530
+++ FLU_2.14.new/FLU/Flu_Tree_Browser.h	2007-03-12 16:32:29.000000000 +0530
@@ -346,7 +346,7 @@
     { return rdata.insertionMode; }
 
   //! \return whether the point \c (x,y) is inside the entry area (not on the scrollbars)
-  bool Flu_Tree_Browser :: inside_entry_area( int x, int y );
+  bool inside_entry_area( int x, int y );
 
   //! Set the title of the Tree (also the label for the root entry)
   inline void label( const char *l )
