diff -Naur FLU_2.14/FLU/Flu_Tree_Browser.h FLU_2.14-p/FLU/Flu_Tree_Browser.h
--- FLU_2.14/FLU/Flu_Tree_Browser.h	2004-11-05 18:03:20.000000000 +0100
+++ FLU_2.14-p/FLU/Flu_Tree_Browser.h	2008-05-18 00:55:42.000000000 +0200
@@ -1078,7 +1078,7 @@
       //! Remove the entry identified by path \b fullpath from this node
       /*! \return the unique id of the removed entry, or \c 0 if no matching entry was found */
       inline unsigned int remove( const char *fullpath )
-	{ return( (unsigned int)modify( fullpath, REMOVE, tree->rdata ) ); }
+	{ return( (unsigned long)modify( fullpath, REMOVE, tree->rdata ) ); }
 
       //! Remove the entry identified by unique id \b id from this node
       /*! \return the unique id of the removed entry, or \c 0 if no matching entry was found */
diff -Naur FLU_2.14/src/Flu_File_Chooser.cpp FLU_2.14-p/src/Flu_File_Chooser.cpp
--- FLU_2.14/src/Flu_File_Chooser.cpp	2004-11-02 01:33:31.000000000 +0100
+++ FLU_2.14-p/src/Flu_File_Chooser.cpp	2008-05-18 00:55:42.000000000 +0200
@@ -2773,7 +2773,7 @@
   const Fl_Menu_Item *selection = entryPopup.popup();
   if( selection )
     {
-      int handler = (int)selection->user_data();
+      int handler = (long)selection->user_data();
       switch( handler )
 	{
 	case ACTION_NEW_FOLDER:
diff -Naur FLU_2.14/src/Flu_Tree_Browser.cpp FLU_2.14-p/src/Flu_Tree_Browser.cpp
--- FLU_2.14/src/Flu_Tree_Browser.cpp	2004-11-05 18:03:20.000000000 +0100
+++ FLU_2.14-p/src/Flu_Tree_Browser.cpp	2008-05-18 00:55:28.000000000 +0200
@@ -2983,7 +2983,7 @@
 
 unsigned int Flu_Tree_Browser :: remove( const char *fullpath )
 {
-  return( (unsigned int)root.modify( fullpath, Node::REMOVE, rdata ) );
+  return( (unsigned long)root.modify( fullpath, Node::REMOVE, rdata ) );
 }
 
 unsigned int Flu_Tree_Browser :: remove( const char *path, const char *text )
