diff -up qtwebengine-everywhere-src-5.11.2/src/3rdparty/chromium/media/base/decoder_buffer.h.0010~ qtwebengine-everywhere-src-5.11.2/src/3rdparty/chromium/media/base/decoder_buffer.h
--- qtwebengine-everywhere-src-5.11.2/src/3rdparty/chromium/media/base/decoder_buffer.h.0010~	2018-08-27 10:21:31.000000000 +0200
+++ qtwebengine-everywhere-src-5.11.2/src/3rdparty/chromium/media/base/decoder_buffer.h	2018-09-20 21:23:04.184848022 +0200
@@ -8,6 +8,8 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#include <libavcodec/avcodec.h>
+
 #include <memory>
 #include <string>
 #include <utility>
@@ -37,7 +39,7 @@ class MEDIA_EXPORT DecoderBuffer
     : public base::RefCountedThreadSafe<DecoderBuffer> {
  public:
   enum {
-    kPaddingSize = 64,
+    kPaddingSize = AV_INPUT_BUFFER_PADDING_SIZE > 64 ? AV_INPUT_BUFFER_PADDING_SIZE : 64,
 #if defined(ARCH_CPU_ARM_FAMILY)
     kAlignmentSize = 16
 #else
