diff -up qtwebengine-everywhere-src-5.12.0/src/3rdparty/chromium/media/base/decoder_buffer.h.6~ qtwebengine-everywhere-src-5.12.0/src/3rdparty/chromium/media/base/decoder_buffer.h
--- qtwebengine-everywhere-src-5.12.0/src/3rdparty/chromium/media/base/decoder_buffer.h.6~	2018-12-17 03:19:05.294678112 +0100
+++ qtwebengine-everywhere-src-5.12.0/src/3rdparty/chromium/media/base/decoder_buffer.h	2018-12-17 03:19:05.311678346 +0100
@@ -8,7 +8,9 @@
 #include <stddef.h>
 #include <stdint.h>
 
+extern "C" {
 #include <libavcodec/avcodec.h>
+}
 
 #include <memory>
 #include <string>
diff -up qtwebengine-everywhere-src-5.12.0/src/3rdparty/chromium/media/filters/BUILD.gn.6~ qtwebengine-everywhere-src-5.12.0/src/3rdparty/chromium/media/filters/BUILD.gn
--- qtwebengine-everywhere-src-5.12.0/src/3rdparty/chromium/media/filters/BUILD.gn.6~	2018-11-13 19:25:11.000000000 +0100
+++ qtwebengine-everywhere-src-5.12.0/src/3rdparty/chromium/media/filters/BUILD.gn	2018-12-17 03:20:03.375479407 +0100
@@ -142,6 +142,11 @@ source_set("filters") {
       "media_file_checker.cc",
       "media_file_checker.h",
     ]
+    libs += [
+      "avcodec",
+      "avformat",
+      "avutil",
+    ]
   }
 
   if (media_use_libvpx) {
@@ -169,11 +174,23 @@ source_set("filters") {
         "ffmpeg_h264_to_annex_b_bitstream_converter.h",
       ]
 
+      libs += [
+        "avcodec",
+        "avformat",
+        "avutil",
+      ]
+
       if (enable_hevc_demuxing) {
         sources += [
           "ffmpeg_h265_to_annex_b_bitstream_converter.cc",
           "ffmpeg_h265_to_annex_b_bitstream_converter.h",
         ]
+
+        libs += [
+          "avcodec",
+          "avformat",
+          "avutil",
+        ]
       }
     }
 
@@ -182,6 +199,12 @@ source_set("filters") {
         "ffmpeg_video_decoder.cc",
         "ffmpeg_video_decoder.h",
       ]
+
+      libs += [
+        "avcodec",
+        "avformat",
+        "avutil",
+      ]
     }
   }
 
