diff -up qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/chromium/build/config/chromecast/BUILD.gn.9~ qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/chromium/build/config/chromecast/BUILD.gn
--- qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/chromium/build/config/chromecast/BUILD.gn.9~	2019-05-17 22:38:03.000000000 +0200
+++ qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/chromium/build/config/chromecast/BUILD.gn	2019-05-29 14:55:39.545614952 +0200
@@ -14,15 +14,6 @@ config("static_config") {
       # http://b/26390825
       "-Wl,--exclude-libs=libffmpeg.a",
     ]
-
-    if (!is_android) {
-      ldflags += [
-        # We want to statically link libstdc++/libgcc on Linux.
-        # (On Android, libstdc++ and libgcc aren't used.)
-        "-static-libstdc++",
-        "-static-libgcc",
-      ]
-    }
   }
 }
 
diff -up qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/gn/build/gen.py.9~ qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/gn/build/gen.py
--- qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/gn/build/gen.py.9~	2019-05-29 14:55:39.546614952 +0200
+++ qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/gn/build/gen.py	2019-05-29 17:11:36.492697850 +0200
@@ -350,7 +350,6 @@ def WriteGNNinja(path, platform, host, o
 
     if platform.is_linux():
       ldflags.extend([
-          '-static-libstdc++',
           '-Wl,--as-needed',
       ])
       # This is needed by libc++.
diff -up qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/gn/infra/recipes/gn.expected/cipd_exists.json.9~ qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/gn/infra/recipes/gn.expected/cipd_exists.json
--- qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/gn/infra/recipes/gn.expected/cipd_exists.json.9~	2019-05-17 22:38:03.000000000 +0200
+++ qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/gn/infra/recipes/gn.expected/cipd_exists.json	2019-05-29 14:55:39.546614952 +0200
@@ -263,4 +263,4 @@
     "recipe_result": null,
     "status_code": 0
   }
-]
\ No newline at end of file
+]
diff -up qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/gn/infra/recipes/gn.expected/cipd_register.json.9~ qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/gn/infra/recipes/gn.expected/cipd_register.json
--- qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/gn/infra/recipes/gn.expected/cipd_register.json.9~	2019-05-17 22:38:03.000000000 +0200
+++ qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/gn/infra/recipes/gn.expected/cipd_register.json	2019-05-29 14:55:39.546614952 +0200
@@ -279,4 +279,4 @@
     "recipe_result": null,
     "status_code": 0
   }
-]
\ No newline at end of file
+]
diff -up qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/gn/infra/recipes/gn.py.9~ qtwebengine-everywhere-src-5.13.0-beta4/src/3rdparty/gn/infra/recipes/gn.py
