diff -up qtwebengine-everywhere-src-5.12.1/mkspecs/features/functions.prf.fix-extractcflag qtwebengine-everywhere-src-5.12.1/mkspecs/features/functions.prf
--- qtwebengine-everywhere-src-5.12.1/mkspecs/features/functions.prf.fix-extractcflag	2019-02-01 09:25:44.950965875 -0600
+++ qtwebengine-everywhere-src-5.12.1/mkspecs/features/functions.prf	2019-02-01 09:28:39.290041131 -0600
@@ -11,7 +11,7 @@ defineReplace(getChromiumSrcDir) {
 }
 
 defineReplace(extractCFlag) {
-    CFLAGS = $$QMAKE_CC $$QMAKE_CFLAGS
+    CFLAGS = $$QMAKE_CC $$QMAKE_CFLAGS $$QMAKE_CFLAGS_RELEASE
     OPTION = $$find(CFLAGS, $$1)
     OPTION = $$split(OPTION, =)
     PARAM = $$member(OPTION, 1)
