diff -up qtwebengine-everywhere-src-5.12.1/src/3rdparty/chromium/third_party/tcmalloc/chromium/src/base/linux_syscall_support.h.gcc9 qtwebengine-everywhere-src-5.12.1/src/3rdparty/chromium/third_party/tcmalloc/chromium/src/base/linux_syscall_support.h
--- qtwebengine-everywhere-src-5.12.1/src/3rdparty/chromium/third_party/tcmalloc/chromium/src/base/linux_syscall_support.h.gcc9	2019-03-03 08:43:11.558507679 -0600
+++ qtwebengine-everywhere-src-5.12.1/src/3rdparty/chromium/third_party/tcmalloc/chromium/src/base/linux_syscall_support.h	2019-03-03 08:43:44.941779231 -0600
@@ -1485,7 +1485,7 @@ struct kernel_stat {
                                "d"(LSS_SYSCALL_ARG(parent_tidptr)),
                                "r"(LSS_SYSCALL_ARG(newtls)),
                                "r"(LSS_SYSCALL_ARG(child_tidptr))
-                             : "rsp", "memory", "r8", "r10", "r11", "rcx");
+                             : "memory", "r8", "r10", "r11", "rcx");
       }
       LSS_RETURN(int, __res);
     }
