--- klamav-0.42.orig/src/ctunit.cpp	2008-01-07 00:00:31.000000000 +0800
+++ klamav-0.42/src/ctunit.cpp	2008-04-26 13:00:11.000000000 +0800
@@ -15,8 +15,9 @@
 #include "cti18n.h"
 #include <vector>
 #include <string>
-#include <stdio.h>    // sprintf
-#include <ctype.h>    // tolower
+#include <cstdlib>    // atoi
+#include <cstdio>     // sprintf
+#include <cctype>     // tolower
 
 using namespace std;
