diff -Naurp ndiswrapper-1.33/driver/Makefile ndiswrapper-1.33-no-depmod-thx/driver/Makefile
--- ndiswrapper-1.33/driver/Makefile	2006-12-29 05:13:07.000000000 +0100
+++ ndiswrapper-1.33-no-depmod-thx/driver/Makefile	2007-01-01 15:04:18.000000000 +0100
@@ -289,9 +289,6 @@ install: prereq_check win2lin_stubs.h ge
 	mkdir -p $(INST_DIR)
 	install -m 0644 $(MODULE) $(INST_DIR)
 	-@/bin/rm -rf $(SRC_DIR)/.tmp_versions
-ifndef DIST_DESTDIR
-	-/sbin/depmod -a $(KVERS) -b $(DESTDIR)
-endif
 
 dist:
 	@for file in $(DISTFILES); do \
