diff -Naur librsvg-2.9.5/configure librsvg-2.9.5.orig/configure
+++ librsvg-2.9.5/configure	2005-06-11 11:13:56.000000000 +0000
--- librsvg-2.9.5.orig/configure	2005-01-19 01:14:43.000000000 +0000
@@ -23487,7 +23487,7 @@
 MOZILLA_CFLAGS=
 if test -z "$MOZILLA_CONFIG"; then
   # Extract the first word of "mozilla-config", so it can be a program name with args.
+set dummy firefox-config; ac_word=$2
-set dummy mozilla-config; ac_word=$2
 echo "$as_me:$LINENO: checking for $ac_word" >&5
 echo $ECHO_N "checking for $ac_word... $ECHO_C" >&6
 if test "${ac_cv_path_MOZILLA_CONFIG+set}" = set; then
@@ -23528,7 +23528,7 @@
 
 fi
 if test "x$MOZILLA_CONFIG" != "xno"; then
+  _mozilla_include_dir=`firefox-config --cflags|sed 's/-I\(.*\) .*/\1/'`
-  _mozilla_include_dir=`mozilla-config --cflags|sed 's/-I\(.*\) .*/\1/'`
   MOZILLA_CFLAGS="-I$_mozilla_include_dir/plugin -I$_mozilla_include_dir/java -I$_mozilla_include_dir/nspr"
 else
 
@@ -23624,8 +23624,8 @@
   if test $succeeded = yes; then
      build_mozilla_plugin=yes
   else
+     { echo "$as_me:$LINENO: WARNING: firefox-config not found. Mozilla/Netscape plugin will not be built" >&5
+echo "$as_me: WARNING: firefox-config not found. Mozilla/Netscape plugin will not be built" >&2;}
-     { echo "$as_me:$LINENO: WARNING: mozilla-config not found. Mozilla/Netscape plugin will not be built" >&5
-echo "$as_me: WARNING: mozilla-config not found. Mozilla/Netscape plugin will not be built" >&2;}
   fi
 
 fi
