diff -Naur kshowmail-3.1.2/admin/cvs.sh kshowmail-3.1.2-p/admin/cvs.sh
--- kshowmail-3.1.2/admin/cvs.sh	2007-03-12 16:35:23.000000000 +0100
+++ kshowmail-3.1.2-p/admin/cvs.sh	2007-06-08 23:03:27.000000000 +0200
@@ -68,7 +68,7 @@
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9*)
+  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.10* )
     echo "*** $AUTOMAKE_STRING found."
     UNSERMAKE=no
     ;;
