Index: fs/namei.c
===================================================================
RCS file: linux-2.6.19/fs/namei.c,v
retrieving revision 1.1
retrieving revision 1.3
diff -u -p -r1.1 -r1.3
--- fs/namei.c	1 Dec 2006 11:38:24 -0000	1.1
+++ fs/namei.c	8 Dec 2006 07:47:33 -0000	1.3
@@ -1251,7 +1251,7 @@ int __user_path_lookup_open(const char _
  * needs parent already locked. Doesn't follow mounts.
  * SMP-safe.
  */
-static struct dentry * __lookup_hash(struct qstr *name, struct dentry * base, struct nameidata *nd)
+struct dentry * __lookup_hash(struct qstr *name, struct dentry * base, struct nameidata *nd)
 {
 	struct dentry * dentry;
 	struct inode *inode;
@@ -2761,6 +2761,7 @@ EXPORT_SYMBOL(follow_up);
 EXPORT_SYMBOL(get_write_access); /* binfmt_aout */
 EXPORT_SYMBOL(getname);
 EXPORT_SYMBOL(lock_rename);
+EXPORT_SYMBOL(__lookup_hash);
 EXPORT_SYMBOL(lookup_one_len);
 EXPORT_SYMBOL(page_follow_link_light);
 EXPORT_SYMBOL(page_put_link);
Index: include/linux/namei.h
===================================================================
RCS file: linux-2.6.19/include/linux/namei.h,v
retrieving revision 1.1
retrieving revision 1.3
diff -u -p -r1.1 -r1.3
--- include/linux/namei.h	1 Dec 2006 11:38:24 -0000	1.1
+++ include/linux/namei.h	8 Dec 2006 07:47:33 -0000	1.3
@@ -76,6 +76,7 @@ extern struct file *lookup_instantiate_f
 extern struct file *nameidata_to_filp(struct nameidata *nd, int flags);
 extern void release_open_intent(struct nameidata *);
 
+struct dentry * __lookup_hash(struct qstr *name, struct dentry * base, struct nameidata *nd);
 extern struct dentry * lookup_one_len(const char *, struct dentry *, int);
 
 extern int follow_down(struct vfsmount **, struct dentry **);
