Index: beyond/fs/squashfs/inode.c
===================================================================
--- beyond.orig/fs/squashfs/inode.c
+++ beyond/fs/squashfs/inode.c
@@ -2143,9 +2143,7 @@ static int __init init_inodecache(void)
 
 static void destroy_inodecache(void)
 {
-	if (kmem_cache_destroy(squashfs_inode_cachep))
-		printk(KERN_INFO "squashfs_inode_cache: not all structures "
-			"were freed\n");
+	kmem_cache_destroy(squashfs_inode_cachep);
 }
 
 
Index: beyond/fs/unionfs/lookup.c
===================================================================
--- beyond.orig/fs/unionfs/lookup.c
+++ beyond/fs/unionfs/lookup.c
@@ -399,9 +399,7 @@ void destroy_dentry_cache(void)
 {
 	if (!unionfs_dentry_cachep)
 		return;
-	if (kmem_cache_destroy(unionfs_dentry_cachep))
-		printk(KERN_ERR
-		       "unionfs_dentry_cache: not all structures were freed\n");
+	kmem_cache_destroy(unionfs_dentry_cachep);
 	return;
 }
 
Index: beyond/fs/unionfs/rdstate.c
===================================================================
--- beyond.orig/fs/unionfs/rdstate.c
+++ beyond/fs/unionfs/rdstate.c
@@ -47,11 +47,8 @@ void destroy_filldir_cache(void)
 {
 	if (!unionfs_filldir_cachep)
 		return;
-	if (kmem_cache_destroy(unionfs_filldir_cachep)) {
-		printk(KERN_ERR
-		       "unionfs_filldir_cache: not all structures were freed\n");
-	}
-	return;
+	kmem_cache_destroy(unionfs_filldir_cachep); 
+   	return;
 }
 
 /* This is a tuning parameter that tells us roughly how big to make the
Index: beyond/fs/unionfs/super.c
===================================================================
--- beyond.orig/fs/unionfs/super.c
+++ beyond/fs/unionfs/super.c
@@ -380,9 +380,7 @@ void destroy_inode_cache(void)
 	print_entry_location();
 	if (!unionfs_inode_cachep)
 		goto out;
-	if (kmem_cache_destroy(unionfs_inode_cachep))
-		printk(KERN_ERR
-		       "unionfs_inode_cache: not all structures were freed\n");
+	kmem_cache_destroy(unionfs_inode_cachep);
       out:
 	print_exit_location();
 	return;
