Index: beyond/fs/squashfs/inode.c
===================================================================
--- beyond.orig/fs/squashfs/inode.c
+++ beyond/fs/squashfs/inode.c
@@ -604,7 +604,7 @@ static struct inode *squashfs_iget(struc
 			i->i_fop = &generic_ro_fops;
 			i->i_mode |= S_IFREG;
 			i->i_blocks = ((i->i_size - 1) >> 9) + 1;
-			i->i_blksize = PAGE_CACHE_SIZE;
+
 			SQUASHFS_I(i)->u.s1.fragment_start_block = frag_blk;
 			SQUASHFS_I(i)->u.s1.fragment_size = frag_size;
 			SQUASHFS_I(i)->u.s1.fragment_offset = inodep->offset;
@@ -657,7 +657,7 @@ static struct inode *squashfs_iget(struc
 			i->i_fop = &generic_ro_fops;
 			i->i_mode |= S_IFREG;
 			i->i_blocks = ((i->i_size - 1) >> 9) + 1;
-			i->i_blksize = PAGE_CACHE_SIZE;
+
 			SQUASHFS_I(i)->u.s1.fragment_start_block = frag_blk;
 			SQUASHFS_I(i)->u.s1.fragment_size = frag_size;
 			SQUASHFS_I(i)->u.s1.fragment_offset = inodep->offset;
Index: beyond/fs/squashfs/squashfs2_0.c
===================================================================
--- beyond.orig/fs/squashfs/squashfs2_0.c
+++ beyond/fs/squashfs/squashfs2_0.c
@@ -229,7 +229,7 @@ static struct inode *squashfs_iget_2(str
 			i->i_atime.tv_sec = inodep->mtime;
 			i->i_ctime.tv_sec = inodep->mtime;
 			i->i_blocks = ((i->i_size - 1) >> 9) + 1;
-			i->i_blksize = PAGE_CACHE_SIZE;
+
 			SQUASHFS_I(i)->u.s1.fragment_start_block = frag_blk;
 			SQUASHFS_I(i)->u.s1.fragment_size = frag_size;
 			SQUASHFS_I(i)->u.s1.fragment_offset = inodep->offset;
