diff -Naur kplayer-0.6.0/admin/cvs.sh kplayer-0.6.0-p/admin/cvs.sh
--- kplayer-0.6.0/admin/cvs.sh	2006-08-25 18:28:08.000000000 +0200
+++ kplayer-0.6.0-p/admin/cvs.sh	2007-01-23 23:35:13.000000000 +0100
@@ -68,7 +68,7 @@
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9*)
+  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.10* )
     echo "*** $AUTOMAKE_STRING found."
     UNSERMAKE=no
     ;;
