--- radeon-profile/radeon-profile.pro	2019-02-11 21:37:15.000000000 +0100
+++ radeon-profile.new/radeon-profile.pro	2019-02-12 10:16:41.890969053 +0100
@@ -98,13 +98,13 @@
     translations/strings.pl.ts \
     translations/strings.hr.ts
 
-bin.path = /usr/bin
+bin.path = /var/tmp/fst/pkg/usr/bin
 bin.files = target/radeon-profile
 
-desktop.path = /usr/share/applications
+desktop.path = /var/tmp/fst/pkg/usr/share/applications
 desktop.files = extra/radeon-profile.desktop
 
-icon.path = /usr/share/icons/hicolor/512x512/apps
+icon.path = /var/tmp/fst/pkg/usr/share/icons/hicolor/512x512/apps
 icon.files = extra/radeon-profile.png
 
 INSTALLS += \
