--- a/source/mupen64plus-core/src/device/r4300/idec.h	2019-02-10 19:25:29.000000000 +0100
+++ b/source/mupen64plus-core/src/device/r4300/idec.h	2022-08-04 11:13:43.999609500 +0200
@@ -82,6 +82,6 @@
 
 #define IDEC_U53(r4300, iw, u53, u5) (void*)(((char*)(r4300)) + idec_u53((iw), (u53), (u5)))
 
-const char* g_r4300_opcodes[R4300_OPCODES_COUNT];
+extern const char* g_r4300_opcodes[R4300_OPCODES_COUNT];
 
 #endif
--- a/source/mupen64plus-core/src/main/workqueue.h	2019-02-10 19:25:29.000000000 +0100
+++ b/source/mupen64plus-core/src/main/workqueue.h	2022-08-04 11:13:43.999609500 +0200
@@ -27,7 +27,6 @@
 
 struct work_struct;
 
-struct work_struct *work;
 typedef void (*work_func_t)(struct work_struct *work);
 struct work_struct {
     work_func_t func;
