diff -Nur gnome-mount-0.5/src/gnome-mount-properties.c gnome-mount-0.5.new/src/gnome-mount-properties.c
--- gnome-mount-0.5/src/gnome-mount-properties.c	2007-03-30 18:26:58.000000000 +0200
+++ gnome-mount-0.5.new/src/gnome-mount-properties.c	2007-03-30 18:27:49.000000000 +0200
@@ -94,10 +94,10 @@
 
 	if (drive == NULL && volume != NULL)
 		drive = gnome_vfs_volume_get_drive (volume);
-
+#if 0
 	g_message ("drive = %x", drive);
 	g_message ("volume = %x", volume);
-
+#endif
 	if (drive != NULL)
 		drive_hal_udi = gnome_vfs_drive_get_hal_udi (drive);
 	if (volume != NULL)
