diff -urN gnome-mount-0.5.orig/src/gnome-mount-properties-view.c gnome-mount-0.5/src/gnome-mount-properties-view.c
--- gnome-mount-0.5.orig/src/gnome-mount-properties-view.c	2006-08-13 22:17:22.000000000 -0600
+++ gnome-mount-0.5/src/gnome-mount-properties-view.c	2006-10-16 14:15:31.000000000 -0600
@@ -34,6 +34,7 @@
 #include <sys/time.h>
 #include <string.h>
 #include <gconf/gconf-client.h>
+#include <libgnomevfs/gnome-vfs-utils.h>
 
 #ifdef __FreeBSD__
 #include <fstab.h>
diff -urN gnome-mount-0.5.orig/src/gnome-mount-properties.c gnome-mount-0.5/src/gnome-mount-properties.c
--- gnome-mount-0.5.orig/src/gnome-mount-properties.c	2006-07-16 14:33:58.000000000 -0600
+++ gnome-mount-0.5/src/gnome-mount-properties.c	2006-10-16 14:11:13.000000000 -0600
@@ -31,6 +31,8 @@
 #include <libnautilus-extension/nautilus-info-provider.h>
 #include <libnautilus-extension/nautilus-property-page-provider.h>
 
+#include <gtk/gtklabel.h>
+
 #include "gnome-mount-properties-view.h"
 
 static DBusConnection *dbus_connection;

