diff -urN gnome-mount-0.5.orig/gnome-mount.schemas.in gnome-mount-0.5/gnome-mount.schemas.in
--- gnome-mount-0.5.orig/gnome-mount.schemas.in	2007-04-04 08:06:41.000000000 +0100
+++ gnome-mount-0.5/gnome-mount.schemas.in	2007-04-04 08:06:54.000000000 +0100
@@ -48,6 +48,30 @@
           <long>A list of default mount options for volumes formatted with the ntfs file system.</long>
         </locale>
       </schema>
+      
+      <schema>
+        <key>/schemas/system/storage/default_options/ntfs-3g/mount_options</key>
+        <applyto>/system/storage/default_options/ntfs-3g/mount_options</applyto>
+        <type>list</type>
+        <list_type>string</list_type>
+        <default>[locale=]</default>
+        <locale name="C">
+          <short>Default mount options for ntfs-3g fs</short>
+          <long>A list of default mount options for volumes formatted with the ntfs file system using ntfs-3g.</long>
+        </locale>
+      </schema> 
+      
+      <schema>
+        <key>/schemas/system/storage/default_options/ntfs-fuse/mount_options</key>
+        <applyto>/system/storage/default_options/ntfs-fuse/mount_options</applyto>
+        <type>list</type>
+        <list_type>string</list_type>
+        <default>[locale=,uid=,umask=077]</default>
+        <locale name="C">
+          <short>Default mount options for ntfs-fuse fs</short>
+          <long>A list of default mount options for volumes formatted with the ntfs file system using ntfs-fuse.</long>
+        </locale>
+      </schema> 
 
 
   </schemalist>
diff -urN gnome-mount-0.5.orig/src/gnome-mount.c gnome-mount-0.5/src/gnome-mount.c
--- gnome-mount-0.5.orig/src/gnome-mount.c	2007-04-04 08:06:41.000000000 +0100
+++ gnome-mount-0.5/src/gnome-mount.c	2007-04-04 08:06:54.000000000 +0100
@@ -1353,6 +1353,11 @@
 					g_free (option);
 					option = g_strdup_printf ("uid=%u", getuid ());
 				}
+				/* use the same kind of workaround for the "locale=" option (use by ntfs-3g and ntfs-fuse) */
+				if (strcmp (option, "locale=") == 0) {
+					g_free (option);
+					option = g_strdup_printf ("locale=%s", setlocale( LC_ALL, "" ));
+				}
 				g_ptr_array_add (mount_options, option);
 
 				g_slist_free_1 (l);
