diff -urN gnome-mount-0.6.orig/gnome-mount.schemas.in gnome-mount-0.6/gnome-mount.schemas.in
--- gnome-mount-0.6.orig/gnome-mount.schemas.in	2007-04-11 08:28:59.000000000 +0100
+++ gnome-mount-0.6/gnome-mount.schemas.in	2007-04-11 08:39:38.000000000 +0100
@@ -6,7 +6,7 @@
         <applyto>/system/storage/default_options/vfat/mount_options</applyto>
         <type>list</type>
         <list_type>string</list_type>
-        <default>[shortname=lower,uid=]</default>
+        <default>[shortname=mixed,uid=,umask=077,exec]</default>
         <locale name="C">
           <short>Default mount options for vfat fs</short>
           <long>A list of default mount options for volumes formatted with the vfat file system.</long>
@@ -18,7 +18,7 @@
         <applyto>/system/storage/default_options/iso9660/mount_options</applyto>
         <type>list</type>
         <list_type>string</list_type>
-        <default>[uid=]</default>
+        <default>[uid=,exec]</default>
         <locale name="C">
           <short>Default mount options for iso9660 fs</short>
           <long>A list of default mount options for volumes formatted with the iso9660 file system.</long>
@@ -30,7 +30,7 @@
         <applyto>/system/storage/default_options/udf/mount_options</applyto>
         <type>list</type>
         <list_type>string</list_type>
-        <default>[uid=]</default>
+        <default>[uid=,exec]</default>
         <locale name="C">
           <short>Default mount options for udf fs</short>
           <long>A list of default mount options for volumes formatted with the udf file system.</long>
@@ -42,7 +42,7 @@
         <applyto>/system/storage/default_options/ntfs/mount_options</applyto>
         <type>list</type>
         <list_type>string</list_type>
-        <default>[umask=222]</default>
+        <default>[umask=222,exec]</default>
         <locale name="C">
           <short>Default mount options for ntfs fs</short>
           <long>A list of default mount options for volumes formatted with the ntfs file system.</long>
@@ -64,7 +64,7 @@
         <applyto>/system/storage/default_options/ntfs-3g/mount_options</applyto>
         <type>list</type>
         <list_type>string</list_type>
-        <default>[locale=]</default>
+        <default>[locale=,exec]</default>
         <locale name="C">
           <short>Default mount options for ntfs-3g fs</short>
           <long>A list of default mount options for volumes formatted with the ntfs file system using ntfs-3g.</long>
