diff -urN gnome-power-manager-2.15.90.orig/data/gnome-power-manager.schemas.in gnome-power-manager-2.15.90/data/gnome-power-manager.schemas.in
--- gnome-power-manager-2.15.90.orig/data/gnome-power-manager.schemas.in	2006-07-08 10:25:14.000000000 +0100
+++ gnome-power-manager-2.15.90/data/gnome-power-manager.schemas.in	2006-07-28 11:05:36.000000000 +0100
@@ -185,7 +185,7 @@
       <applyto>/apps/gnome-power-manager/can_hibernate</applyto>
       <owner>gnome-power-manager</owner>
       <type>bool</type>
-      <default>true</default>
+      <default>false</default>
       <locale name="C">
         <short>Hibernate enabled</short>
         <long>If the user is authorized to hibernate the computer.</long>
