diff -urN gnome-power-manager-2.15.92.orig/data/gnome-power-manager.conf gnome-power-manager-2.15.92/data/gnome-power-manager.conf
--- gnome-power-manager-2.15.92.orig/data/gnome-power-manager.conf	2006-01-10 17:46:13.000000000 +0000
+++ gnome-power-manager-2.15.92/data/gnome-power-manager.conf	2006-08-21 10:08:15.000000000 +0100
@@ -2,7 +2,7 @@
  "-//freedesktop//DTD D-BUS Bus Configuration 1.0//EN"
  "http://www.freedesktop.org/standards/dbus/1.0/busconfig.dtd">
 <busconfig>
-	<policy at_console="true">
+	<policy group="users">
 		<allow own="org.gnome.PowerManager"/>
 		<allow send_destination="org.gnome.PowerManager"/>
 		<allow send_interface="org.gnome.PowerManager"/>
