--- gnome-power-manager-2.24.0old/data/gnome-power-manager.schemas.in	2008-09-04 12:05:18.000000000 +0200
+++ gnome-power-manager-2.24.0/data/gnome-power-manager.schemas.in	2008-10-02 15:05:09.000000000 +0200
@@ -411,7 +411,7 @@
       <applyto>/apps/gnome-power-manager/general/can_hibernate</applyto>
       <owner>gnome-power-manager</owner>
       <type>bool</type>
-      <default>true</default>
+      <default>false</default>
       <locale name="C">
         <short>Hibernate enabled</short>
         <long>If the user is authorized to hibernate the computer.</long>
