--- src/kfilebox-0.4.5/kfilebox.pro	2010-09-24 14:11:46.000000000 +0200
+++ kfilebox.pro	2010-09-27 17:46:02.000000000 +0200
@@ -37,10 +37,44 @@
     core/dolphin.h \
     core/konqueror.h
 RESOURCES += DropboxIcons.qrc
-
+TRANSLIST = ar \
+    br \
+    de \
+#    en \
+    es \
+    fr \
+    gl \
+    tr \
+    ru \
+    pl \
+    cs \
+    it \
+    lt \
+    nl \
+    pt \
+    si \
+    zh
+for(language, TRANSLIST):TRANSLATIONS += locale/$${language}/kfilebox.po
 FORMS += mainwindow.ui \
     installer/installerform.ui
-
+!isEmpty(TRANSLATIONS) { 
+    isEmpty(QMAKE_LRELEASE) { 
+        win32:QMAKE_LRELEASE = msgfmt.exe
+        else:QMAKE_LRELEASE = msgfmt
+    }
+    TSQM.name = msgfmt \
+        ${QMAKE_FILE_IN}
+    TSQM.input = TRANSLATIONS
+    TSQM.output = ${QMAKE_FILE_PATH}/${QMAKE_FILE_BASE}.mo
+    TSQM.commands = $$QMAKE_LRELEASE \
+        -o \
+        $$TSQM.output \
+        ${QMAKE_FILE_IN}
+    TSQM.CONFIG = no_link
+    QMAKE_EXTRA_COMPILERS += TSQM
+    PRE_TARGETDEPS += compiler_TSQM_make_all
+}
+else:message(No translation files in project)
 
 DATADIR=/usr/share
 
@@ -108,128 +142,60 @@
 iconox128.files += img/hicolor/128x128/apps/kfilebox.png
 INSTALLS += iconox128
 
-transinstallar.extra = find . -name \"*.cpp\" -o -name \"*.h\"  | xargs xgettext -C --kde --keyword=tr --keyword=translate:2 -o $${OUT_PWD}/messages.po; \
-    msgmerge locale/ar/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/ar/kfilebox.mo; \
-    msgfmt -o $$DATADIR/locale/ar/LC_MESSAGES/kfilebox.mo locale/ar/kfilebox.po
 transinstallar.path = $$DATADIR/locale/ar/LC_MESSAGES
-transinstallar.files = locale/ar/kfilebox.mo
+transinstallar.files = locale/ar/kdropbox.mo
 INSTALLS += transinstallar
-transinstallbr.extra = msgmerge locale/br/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/br/kfilebox.po; \
-    msgfmt -o locale/br/kfilebox.mo locale/br/kfilebox.po; \
-    cp -f locale/br/kfilebox.mo $$DATADIR/locale/br/LC_MESSAGES
 transinstallbr.path = $$DATADIR/locale/br/LC_MESSAGES
-transinstallbr.files = locale/br/kfilebox.mo
+transinstallbr.files = locale/br/kdropbox.mo
 INSTALLS += transinstallbr
-transinstallde.extra = msgmerge locale/de/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/de/kfilebox.po; \
-    msgfmt -o locale/de/kfilebox.mo locale/de/kfilebox.po; \
-    cp -f locale/de/kfilebox.mo $$DATADIR/locale/de/LC_MESSAGES
 transinstallde.path = $$DATADIR/locale/de_DE/LC_MESSAGES
-transinstallde.files = locale/de/kfilebox.mo
+transinstallde.files = locale/de/kdropbox.mo
 INSTALLS += transinstallde
-
-transinstallen.extra = msgmerge locale/en/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/en/kfilebox.po; \
-    msgfmt -o locale/en/kfilebox.mo locale/en/kfilebox.po; \
-    cp -f locale/en/kfilebox.mo $$DATADIR/locale/en/LC_MESSAGES
-transinstallen.path = $$DATADIR/locale/en/LC_MESSAGES
-transinstallen.files = locale/en/kfilebox.mo
-INSTALLS += transinstallen
-
-transinstallen_gb.extra = cp -f locale/en/kfilebox.mo $$DATADIR/locale/en_GB/LC_MESSAGES
+#transinstallen.path = $$DATADIR/locale/en/LC_MESSAGES
+#transinstallen.files = locale/en/kdropbox.mo
+#INSTALLS += transinstallen
 transinstallen_gb.path = $$DATADIR/locale/en_GB/LC_MESSAGES
-transinstallen_gb.files = locale/en/kfilebox.mo
+transinstallen_gb.files = locale/en/kdropbox.mo
 INSTALLS += transinstallen_gb
-
-transinstalles.extra = msgmerge locale/es/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/es/kfilebox.po; \
-    msgfmt -o locale/es/kfilebox.mo locale/es/kfilebox.po; \
-    cp -f locale/es/kfilebox.mo $$DATADIR/locale/es/LC_MESSAGES
 transinstalles.path = $$DATADIR/locale/es/LC_MESSAGES
-transinstalles.files = locale/es/kfilebox.mo
+transinstalles.files = locale/es/kdropbox.mo
 INSTALLS += transinstalles
-
-transinstallfr.extra = msgmerge locale/fr/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/fr/kfilebox.po; \
-    msgfmt -o locale/fr/kfilebox.mo locale/fr/kfilebox.po; \
-    cp -f locale/fr/kfilebox.mo $$DATADIR/locale/fr/LC_MESSAGES
 transinstallfr.path = $$DATADIR/locale/fr/LC_MESSAGES
-transinstallfr.files = locale/fr/kfilebox.mo
+transinstallfr.files = locale/fr/kdropbox.mo
 INSTALLS += transinstallfr
-
-transinstallgl.extra = msgmerge locale/gl/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/gl/kfilebox.po; \
-    msgfmt -o locale/gl/kfilebox.mo locale/gl/kfilebox.po; \
-    cp -f locale/gl/kfilebox.mo $$DATADIR/locale/gl/LC_MESSAGES
 transinstallgl.path = $$DATADIR/locale/gl/LC_MESSAGES
-transinstallgl.files = locale/gl/kfilebox.mo
+transinstallgl.files = locale/gl/kdropbox.mo
 INSTALLS += transinstallgl
-
-transinstalltr.extra = msgmerge locale/tr/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/tr/kfilebox.po; \
-    msgfmt -o locale/tr/kfilebox.mo locale/tr/kfilebox.po; \
-    cp -f locale/tr/kfilebox.mo $$DATADIR/locale/tr/LC_MESSAGES
 transinstalltr.path = $$DATADIR/locale/tr/LC_MESSAGES
-transinstalltr.files = locale/tr/kfilebox.mo
+transinstalltr.files = locale/tr/kdropbox.mo
 INSTALLS += transinstalltr
-
-transinstallru.extra = msgmerge locale/ru/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/ru/kfilebox.po; \
-    msgfmt -o locale/ru/kfilebox.mo locale/ru/kfilebox.po; \
-    cp -f locale/ru/kfilebox.mo $$DATADIR/locale/ru/LC_MESSAGES
 transinstallru.path = $$DATADIR/locale/ru/LC_MESSAGES
-transinstallru.files = locale/ru/kfilebox.mo
+transinstallru.files = locale/ru/kdropbox.mo
 INSTALLS += transinstallru
-
-transinstallcs.extra = msgmerge locale/cs/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/cs/kfilebox.po; \
-    msgfmt -o locale/cs/kfilebox.mo locale/cs/kfilebox.po; \
-    cp -f locale/cs/kfilebox.mo $$DATADIR/locale/cs/LC_MESSAGES
 transinstallcs.path = $$DATADIR/locale/cs/LC_MESSAGES
-transinstallcs.files = locale/cs/kfilebox.mo
+transinstallcs.files = locale/cs/kdropbox.mo
 INSTALLS += transinstallcs
-
-transinstallpl.extra = msgmerge locale/pl/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/pl/kfilebox.po; \
-    msgfmt -o locale/pl/kfilebox.mo locale/pl/kfilebox.po; \
-    cp -f locale/pl/kfilebox.mo $$DATADIR/locale/pl/LC_MESSAGES
 transinstallpl.path = $$DATADIR/locale/pl/LC_MESSAGES
-transinstallpl.files = locale/pl/kfilebox.mo
+transinstallpl.files = locale/pl/kdropbox.mo
 INSTALLS += transinstallpl
-
-transinstallit.extra = msgmerge locale/it/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/it/kfilebox.po; \
-    msgfmt -o locale/it/kfilebox.mo locale/it/kfilebox.po; \
-    cp -f locale/it/kfilebox.mo $$DATADIR/locale/it/LC_MESSAGES
 transinstallit.path = $$DATADIR/locale/it/LC_MESSAGES
-transinstallit.files = locale/it/kfilebox.mo
+transinstallit.files = locale/it/kdropbox.mo
 INSTALLS += transinstallit
-
-transinstalllt.extra = msgmerge locale/lt/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/lt/kfilebox.po; \
-    msgfmt -o locale/lt/kfilebox.mo locale/lt/kfilebox.po; \
-    cp -f locale/lt/kfilebox.mo $$DATADIR/locale/lt/LC_MESSAGES
 transinstalllt.path = $$DATADIR/locale/lt/LC_MESSAGES
-transinstalllt.files = locale/lt/kfilebox.mo
+transinstalllt.files = locale/lt/kdropbox.mo
 INSTALLS += transinstalllt
-
-transinstallnl.extra = msgmerge locale/nl/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/nl/kfilebox.po; \
-    msgfmt -o locale/nl/kfilebox.mo locale/nl/kfilebox.po; \
-    cp -f locale/nl/kfilebox.mo $$DATADIR/locale/nl/LC_MESSAGES
 transinstallnl.path = $$DATADIR/locale/nl/LC_MESSAGES
-transinstallnl.files = locale/nl/kfilebox.mo
+transinstallnl.files = locale/nl/kdropbox.mo
 INSTALLS += transinstallnl
-
-transinstallpt.extra = msgmerge locale/pt/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/pt/kfilebox.po; \
-    msgfmt -o locale/pt/kfilebox.mo locale/pt/kfilebox.po; \
-    cp -f locale/pt/kfilebox.mo $$DATADIR/locale/pt/LC_MESSAGES
 transinstallpt.path = $$DATADIR/locale/pt/LC_MESSAGES
-transinstallpt.files = locale/pt/kfilebox.mo
+transinstallpt.files = locale/pt/kdropbox.mo
 INSTALLS += transinstallpt
-
-transinstallsi.extra = msgmerge locale/si/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/si/kfilebox.po; \
-    msgfmt -o locale/si/kfilebox.mo locale/si/kfilebox.po; \
-    cp -f locale/si/kfilebox.mo $$DATADIR/locale/si/LC_MESSAGES
 transinstallsi.path = $$DATADIR/locale/si/LC_MESSAGES
-transinstallsi.files = locale/si/kfilebox.mo
+transinstallsi.files = locale/si/kdropbox.mo
 INSTALLS += transinstallsi
-
-transinstallzh.extra = msgmerge locale/zh/kfilebox.po messages.po > tmp.po; mv -f tmp.po locale/zh/kfilebox.po; \
-    msgfmt -o locale/zh/kfilebox.mo locale/zh/kfilebox.po; \
-    cp -f locale/zh/kfilebox.mo $$DATADIR/locale/zh/LC_MESSAGES
 transinstallzh.path = $$DATADIR/locale/zh/LC_MESSAGES
-transinstallzh.files = locale/zh/kfilebox.mo
+transinstallzh.files = locale/zh/kdropbox.mo
 INSTALLS += transinstallzh
-transinstallzh_cn.extra =  cp -f locale/zh/kfilebox.mo $$DATADIR/locale/zh_CN/LC_MESSAGES
-transinstallzh_cn.path = $$DATADIR/locale/zh_CN/LC_MESSAGES
-transinstallzh_cn.files = locale/zh/kfilebox.mo
+transinstallzh_cn.path = $$DATADIR/locale/zh_cn/LC_MESSAGES
+transinstallzh_cn.files = locale/zh/kdropbox.mo
 INSTALLS += transinstallzh_cn
