--- FrugalBuild-20031029-1	2004-05-28 11:22:03.000000000 +0200
+++ FrugalBuild-20040626-1	2004-06-29 12:23:12.000000000 +0200
@@ -1,38 +1,25 @@
-# Last Modified: Wed Apr 28 16:24:44 CEST 2004
+# Last Modified: Tue Jun 29 12:21:38 CEST 2004
 # Compiling Time: ~1 minute
 # Maintainer: VMiklos <mamajom@axelero.hu>
 
 pkgname=codecs
-pkgver=20031029
+pkgver=20040626
 pkgrel=1
 pkgdesc="Binary codecs for MPlayer"
 url="http://www1.mplayerhq.hu/MPlayer/releases/codecs/"
 depends=()
-source=(http://www1.mplayerhq.hu/MPlayer/releases/codecs/win32codecs.tar.bz2 \
-        http://www1.mplayerhq.hu/MPlayer/releases/codecs/essential.tar.bz2 \
-	http://www1.mplayerhq.hu/MPlayer/releases/codecs/xanimdlls.tar.bz2)
-md5sums=('10e934a4a5609fa2e094acb0f6c1b976' '4748ecae87f71e8bda9cb2e2a9bd30b4' \
-         'c2787acd99d905fa995ad40a8f52abb8')
+up2date=`lynx -dump http://www.mplayerhq.hu/MPlayer/releases/codecs/MD5SUMS|sed -n -e 's/.tar.bz2//' -e '1 p'|cut -d ' ' -f 3|cut -d - -f 2`
+source=(http://www1.mplayerhq.hu/MPlayer/releases/codecs/win32codecs-$pkgver.tar.bz2 \
+        http://www1.mplayerhq.hu/MPlayer/releases/codecs/essential-$pkgver.tar.bz2 \
+	http://www1.mplayerhq.hu/MPlayer/releases/codecs/xanimdlls-$pkgver.tar.bz2)
+md5sums=('a29b1f03b425546bdd321cfc3e6ffdea' 'a23c6a22bc79f1b0ddc2285f8e2240c2' \
+         'a7c980a1a94fbac6ebd34c7485a1891d')
 
-# up2date
-up2date_win32codecs=`lynx -dump http://www1.mplayerhq.hu/MPlayer/releases/codecs/|grep -1  codecpack|sed -n '2 p'|tr -s ' '|cut -d ' ' -f 6`
-up2date_essential=`lynx -dump http://www1.mplayerhq.hu/MPlayer/releases/codecs/|grep -1 All-in-one|tail -1|tr -s ' ' |cut -d ' ' -f 2`
-up2date_xanimdlls=`lynx -dump http://www1.mplayerhq.hu/MPlayer/releases/codecs/|grep -1 XAnim|sed -n '3 p'|tr -s ' '|cut -d ' ' -f 2`
-
-
-if [ "${md5sums[0]}" == "$up2date_win32codecs" ] && \
-   [ "${md5sums[1]}" == "$up2date_essential" ] && \
-   [ "${md5sums[2]}" == "$up2date_xanimdlls" ]; then
-	up2date=$pkgver
-else
-	up2date=$RANDOM
-fi
-
-# we have to use makepkg --nostrip!
+# we have to use makepkg --nostrip
 
 build() {
-   mkdir -p $startdir/pkg/usr/lib/win32
-   mv $startdir/src/*/* $startdir/pkg/usr/lib/win32
-   chown -R root.root $startdir/pkg/usr/lib/win32
+	mkdir -p $startdir/pkg/usr/lib/win32
+	mv $startdir/src/*/* $startdir/pkg/usr/lib/win32
+	chown -R root.root $startdir/pkg/usr/lib/win32
 	chmod 755 $startdir/pkg/usr/lib/win32/*
 }
