diff -Naur opensc-0.22.0/configure.ac opensc-0.22.0.new/configure.ac
--- opensc-0.22.0/configure.ac	2021-08-10 11:09:20.000000000 +0200
+++ opensc-0.22.0.new/configure.ac	2022-11-12 15:45:12.847429838 +0100
@@ -1089,7 +1089,7 @@
 	CFLAGS="-pedantic ${CFLAGS}"
 fi
 if test "${enable_strict}" = "yes"; then
-	CFLAGS="-Wall -Wextra -Wno-unused-parameter -Werror -Wstrict-aliasing=2 ${CFLAGS}"
+	CFLAGS="-Wall -Wextra -Wno-unused-parameter  -Wstrict-aliasing=2 ${CFLAGS}"
 fi
 
 AC_CONFIG_FILES([
diff -Naur opensc-0.22.0/src/tools/pkcs11-tool.c opensc-0.22.0.new/src/tools/pkcs11-tool.c
--- opensc-0.22.0/src/tools/pkcs11-tool.c	2021-08-10 11:09:20.000000000 +0200
+++ opensc-0.22.0.new/src/tools/pkcs11-tool.c	2022-11-12 15:45:10.338570984 +0100
@@ -48,6 +48,7 @@
 #include <openssl/asn1t.h>
 #include <openssl/rsa.h>
 #include <openssl/pem.h>
+#include <openssl/provider.h>
 #if !defined(OPENSSL_NO_EC) && !defined(OPENSSL_NO_ECDSA)
 #include <openssl/ec.h>
 #include <openssl/ecdsa.h>
@@ -66,6 +67,10 @@
 #include "util.h"
 #include "libopensc/sc-ossl-compat.h"
 
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+OSSL_PROVIDER *legacy_provider = NULL;
+#endif
+
 #ifdef _WIN32
 #ifndef STDOUT_FILENO
 #define STDOUT_FILENO 1
@@ -4979,7 +4984,6 @@
 	CK_ULONG        hashLen1, hashLen2;
 	CK_MECHANISM_TYPE firstMechType;
 	CK_SESSION_INFO sessionInfo;
-
 	CK_MECHANISM_TYPE mechTypes[] = {
 		CKM_MD5,
 		CKM_SHA_1,
@@ -5074,6 +5078,11 @@
 
 	for (i = 0; mechTypes[i] != 0xffffff; i++) {
 		ck_mech.mechanism = mechTypes[i];
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+		if (!legacy_provider) {
+			legacy_provider = OSSL_PROVIDER_load(NULL, "legacy");
+		}
+#endif
 
 		rv = p11->C_DigestInit(session, &ck_mech);
 		if (rv == CKR_MECHANISM_INVALID)
@@ -5239,6 +5248,11 @@
 		EVP_sha256(),
 	};
 #endif
+#if OPENSSL_VERSION_NUMBER >= 0x30000000L
+	if (!legacy_provider) {
+		legacy_provider = OSSL_PROVIDER_load(NULL, "legacy");
+	}
+#endif
 
 	rv = p11->C_SignInit(session, ck_mech, privKeyObject);
 	/* mechanism not implemented, don't test */
