--- unicode-data-0.6.0/unicode-data.cabal	2001-09-09 03:46:40.000000000 +0200
+++ unicode-data-0.6.0.new/unicode-data.cabal	2024-12-25 21:20:43.202865235 +0100
@@ -118,7 +118,7 @@
   hs-source-dirs: lib
   ghc-options: -O2
   build-depends:
-      base >=4.7 && < 4.21
+      base >=4.7 && < 4.22
   if impl(ghc < 9.0)
     -- Required for unpackCString#
     build-depends: ghc-prim
@@ -132,7 +132,7 @@
   other-modules:
       Unicode.CharSpec
   build-depends:
-      base             >= 4.7   && < 4.21
+      base             >= 4.7   && < 4.22
     , hspec            >= 2.0   && < 2.12
     , unicode-data
   if flag(dev-has-icu)
@@ -159,7 +159,7 @@
     Unicode.Char.NumericBench
     Unicode.Char.Numeric.CompatBench
   build-depends:
-    base        >= 4.7   && < 4.21,
+    base        >= 4.7   && < 4.22,
     deepseq     >= 1.1   && < 1.6,
     tasty-bench >= 0.2.5 && < 0.4,
     tasty       >= 1.4.1 && < 1.6,
