diff -Nurp sim-0.9.4.1/plugins/remote/Makefile.am sim-0.9.4.1-p/plugins/remote/Makefile.am
--- sim-0.9.4.1/plugins/remote/Makefile.am	2006-02-03 10:54:16.000000000 +0100
+++ sim-0.9.4.1-p/plugins/remote/Makefile.am	2006-10-24 19:31:17.000000000 +0200
@@ -11,7 +11,7 @@ simctrl_SOURCES = \
 
 INCLUDES=-I../../sim -I../_core $(all_includes)
 
-mime_DATA = x-icq.desktop
+mime_DATA = x-sim-icq.desktop
 mimedir = $(kde_mimedir)/application
 service_DATA = simctrl.desktop
 servicedir = $(kde_servicesdir)
diff -Nurp sim-0.9.4.1/plugins/remote/Makefile.in sim-0.9.4.1-p/plugins/remote/Makefile.in
--- sim-0.9.4.1/plugins/remote/Makefile.in	2006-10-21 19:10:31.000000000 +0200
+++ sim-0.9.4.1-p/plugins/remote/Makefile.in	2006-10-24 19:30:13.000000000 +0200
@@ -381,7 +381,7 @@ simctrl_SOURCES = \
  simctrl.cpp
 
 INCLUDES = -I../../sim -I../_core $(all_includes)
-mime_DATA = x-icq.desktop
+mime_DATA = x-sim-icq.desktop
 mimedir = $(kde_mimedir)/application
 service_DATA = simctrl.desktop
 servicedir = $(kde_servicesdir)
@@ -791,7 +791,7 @@ clean-metasources:
 	-rm -f  remotecfg.moc remote.moc
 
 #>+ 2
-KDE_DIST=simremote.vcproj remote.h remotecfg.h simext.def simremote.rc simipc.vcproj simipc.cpp SimControl.cpp simext.vcproj simctrl.vcproj SimControl.rgs simext.cpp Makefile.in sim_remote.cpp SIM_ext.h IContextMenuImpl.h simctrl.dsp x-icq.desktop SIM_ext.rgs SimControl.h simext.idl simext.dsp SIM_ext.cpp IShellExtInitImpl.h remote.rc simremote.idl simext.rc remotecfgbase.ui remote.vcproj configure.in.in simctrl.desktop simremote.def simremote.dsp Makefile.am remote.dsp simipc.dsp 
+KDE_DIST=simremote.vcproj remote.h remotecfg.h simext.def simremote.rc simipc.vcproj simipc.cpp SimControl.cpp simext.vcproj simctrl.vcproj SimControl.rgs simext.cpp Makefile.in sim_remote.cpp SIM_ext.h IContextMenuImpl.h simctrl.dsp x-sim-icq.desktop SIM_ext.rgs SimControl.h simext.idl simext.dsp SIM_ext.cpp IShellExtInitImpl.h remote.rc simremote.idl simext.rc remotecfgbase.ui remote.vcproj configure.in.in simctrl.desktop simremote.def simremote.dsp Makefile.am remote.dsp simipc.dsp 
 
 #>+ 4
 clean-ui:
diff -Nurp sim-0.9.4.1/plugins/remote/simctrl.desktop sim-0.9.4.1-p/plugins/remote/simctrl.desktop
--- sim-0.9.4.1/plugins/remote/simctrl.desktop	2006-01-15 13:53:27.000000000 +0100
+++ sim-0.9.4.1-p/plugins/remote/simctrl.desktop	2006-10-24 19:31:57.000000000 +0200
@@ -3,7 +3,7 @@ Encoding=UTF-8
 Name=simctrl
 Icon=licq
 Type=Service
-MimeType=application/x-icq;
+MimeType=application/x-sim-icq;
 GenericName=SIM remote control
 Terminal=false
 Exec=simctrl
diff -Nurp sim-0.9.4.1/plugins/remote/x-icq.desktop sim-0.9.4.1-p/plugins/remote/x-icq.desktop
--- sim-0.9.4.1/plugins/remote/x-icq.desktop	2006-01-15 13:53:27.000000000 +0100
+++ sim-0.9.4.1-p/plugins/remote/x-icq.desktop	1970-01-01 01:00:00.000000000 +0100
@@ -1,51 +0,0 @@
-[Desktop Entry]
-Encoding=UTF-8
-Comment=ICQ Contact
-Comment[ar]=جهة اتصال على ICQ
-Comment[be]=Кантакт ICQ
-Comment[bg]=Връзка с ICQ Contact
-Comment[br]=Darempred ICQ
-Comment[bs]=ICQ kontakt
-Comment[ca]=Contacte ICQ
-Comment[cs]=ICQ kontakt
-Comment[cy]=Cysylltiad ICQ
-Comment[da]=ICQ-Kontakt
-Comment[de]=ICQ-Kontakt
-Comment[el]=Επαφή ICQ
-Comment[es]=Contacto de ICQ
-Comment[et]=ICQ kontakt
-Comment[eu]=ICQ kontaktua
-Comment[fi]=ICQ-kontakti
-Comment[fr]=Contact ICQ
-Comment[gl]=Contacto ICQ
-Comment[he]=איש-קשר ICQ
-Comment[hi]=आईसीक्यू सम्पर्क
-Comment[hr]=ICQ kontakt
-Comment[hu]=ICQ-kapcsolat
-Comment[is]=ICQ tengiliður
-Comment[it]=Contatto ICQ
-Comment[ja]=ICQコンタクト
-Comment[mk]=Контакт на ICQ 
-Comment[nb]=ICQ kontakt
-Comment[nl]=ICQ contact
-Comment[nn]=ICQ-kontakt
-Comment[pl]=Kontakt ICQ
-Comment[pt]=Contacto de ICQ
-Comment[pt_BR]=Contato ICQ
-Comment[ru]=Контакт ICQ
-Comment[se]=ICQ-oktavuohta
-Comment[sk]=Kontakt ICQ
-Comment[sl]=Stik ICQ
-Comment[sr]=ICQ контакт
-Comment[sr@Latn]=ICQ kontakt
-Comment[sv]=ICQ-kontakt
-Comment[ta]=ICQ தொடர்பு
-Comment[tg]=Пайвастшавии ICQ 
-Comment[tr]=ICQ Bağlantısı
-Comment[uk]=Контакт ICQ
-Comment[xx]=xxICQ Contactxx
-Comment[zh_CN]=ICQ 联系人
-Type=MimeType
-MimeType=application/x-icq
-Patterns=*.uin;*.icq
-Icon=licq
diff -Nurp sim-0.9.4.1/plugins/remote/x-sim-icq.desktop sim-0.9.4.1-p/plugins/remote/x-sim-icq.desktop
--- sim-0.9.4.1/plugins/remote/x-sim-icq.desktop	1970-01-01 01:00:00.000000000 +0100
+++ sim-0.9.4.1-p/plugins/remote/x-sim-icq.desktop	2006-10-24 19:32:48.000000000 +0200
@@ -0,0 +1,51 @@
+[Desktop Entry]
+Encoding=UTF-8
+Comment=ICQ Contact
+Comment[ar]=جهة اتصال على ICQ
+Comment[be]=Кантакт ICQ
+Comment[bg]=Връзка с ICQ Contact
+Comment[br]=Darempred ICQ
+Comment[bs]=ICQ kontakt
+Comment[ca]=Contacte ICQ
+Comment[cs]=ICQ kontakt
+Comment[cy]=Cysylltiad ICQ
+Comment[da]=ICQ-Kontakt
+Comment[de]=ICQ-Kontakt
+Comment[el]=Επαφή ICQ
+Comment[es]=Contacto de ICQ
+Comment[et]=ICQ kontakt
+Comment[eu]=ICQ kontaktua
+Comment[fi]=ICQ-kontakti
+Comment[fr]=Contact ICQ
+Comment[gl]=Contacto ICQ
+Comment[he]=איש-קשר ICQ
+Comment[hi]=आईसीक्यू सम्पर्क
+Comment[hr]=ICQ kontakt
+Comment[hu]=ICQ-kapcsolat
+Comment[is]=ICQ tengiliður
+Comment[it]=Contatto ICQ
+Comment[ja]=ICQコンタクト
+Comment[mk]=Контакт на ICQ 
+Comment[nb]=ICQ kontakt
+Comment[nl]=ICQ contact
+Comment[nn]=ICQ-kontakt
+Comment[pl]=Kontakt ICQ
+Comment[pt]=Contacto de ICQ
+Comment[pt_BR]=Contato ICQ
+Comment[ru]=Контакт ICQ
+Comment[se]=ICQ-oktavuohta
+Comment[sk]=Kontakt ICQ
+Comment[sl]=Stik ICQ
+Comment[sr]=ICQ контакт
+Comment[sr@Latn]=ICQ kontakt
+Comment[sv]=ICQ-kontakt
+Comment[ta]=ICQ தொடர்பு
+Comment[tg]=Пайвастшавии ICQ 
+Comment[tr]=ICQ Bağlantısı
+Comment[uk]=Контакт ICQ
+Comment[xx]=xxICQ Contactxx
+Comment[zh_CN]=ICQ 联系人
+Type=MimeType
+MimeType=application/x-sim-icq
+Patterns=*.uin;*.icq
+Icon=licq
