=== modified file 'gtk/gtk.symbols'
--- old/gtk/gtk.symbols	2010-07-29 17:26:15 +0000
+++ new/gtk/gtk.symbols	2010-07-29 17:26:21 +0000
@@ -5277,6 +5277,7 @@
 gtk_widget_get_realized
 gtk_widget_set_mapped
 gtk_widget_get_mapped
+ubuntu_gtk_widget_set_has_grab
 #endif
 #endif
 

=== modified file 'gtk/gtkwidget.c'
--- old/gtk/gtkwidget.c	2010-07-29 17:24:37 +0000
+++ new/gtk/gtkwidget.c	2010-07-29 17:29:04 +0000
@@ -11399,5 +11399,15 @@
   return res;
 }
 
+void
+ubuntu_gtk_widget_set_has_grab (GtkWidget *widget,
+                                gboolean   has_grab)
+{
+  if (has_grab)
+    GTK_OBJECT_FLAGS (widget) |= GTK_HAS_GRAB;
+  else
+    GTK_OBJECT_FLAGS (widget) &= ~(GTK_HAS_GRAB);
+}
+
 #define __GTK_WIDGET_C__
 #include "gtkaliasdef.c"

=== modified file 'gtk/gtkwidget.h'
--- old/gtk/gtkwidget.h	2010-07-29 17:24:37 +0000
+++ new/gtk/gtkwidget.h	2010-07-29 17:27:51 +0000
@@ -1342,6 +1342,9 @@
 						       GtkWidget *toplevel,
 						       gpointer   user_data);
 
+void   ubuntu_gtk_widget_set_has_grab  (GtkWidget *widget,
+                                        gboolean   has_grab);
+
 G_END_DECLS
 
 #endif /* __GTK_WIDGET_H__ */

