diff -Naur gtk+-2.10.14/modules/printbackends/cups/gtkcupsutils.c gtk+-2.10.14-p/modules/printbackends/cups/gtkcupsutils.c
--- gtk+-2.10.14/modules/printbackends/cups/gtkcupsutils.c	2007-07-16 21:45:18.000000000 +0200
+++ gtk+-2.10.14-p/modules/printbackends/cups/gtkcupsutils.c	2007-09-07 20:20:19.000000000 +0200
@@ -30,8 +30,10 @@
 #include <stdlib.h>
 #include <time.h>
 
-#if CUPS_VERSION_MAJOR > 1 || (CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR > 1) || (CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR == 1 && CUPS_VERSION_PATCH >= 20)
-#define HAVE_HTTP_AUTHSTRING 1
+#if (CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR == 1 && CUPS_VERSION_PATCH >= 20) || (CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR == 2 && CUPS_VERSION_PATCH >= 0)
+#define OLD_API 1
+#elif (CUPS_VERSION_MAJOR == 1 && CUPS_VERSION_MINOR == 3 && CUPS_VERSION_PATCH >= 0)
+#define NEW_API 1
 #endif
 
 typedef void (*GtkCupsRequestStateFunc) (GtkCupsRequest *request);
@@ -627,8 +629,10 @@
   httpClearFields(request->http);
   httpSetField(request->http, HTTP_FIELD_CONTENT_LENGTH, length);
   httpSetField(request->http, HTTP_FIELD_CONTENT_TYPE, "application/ipp");
-#ifdef HAVE_HTTP_AUTHSTRING
+#if defined(OLD_API)
   httpSetField(request->http, HTTP_FIELD_AUTHORIZATION, request->http->authstring);
+#elif defined(NEW_API)
+  httpSetField (request->http, HTTP_FIELD_AUTHORIZATION, httpGetAuthString (request->http));
 #endif
 
   if (httpPost(request->http, request->resource))
@@ -966,8 +970,10 @@
     }
 
   httpClearFields(request->http);
-#ifdef HAVE_HTTP_AUTHSTRING
+#if defined(OLD_API)
   httpSetField(request->http, HTTP_FIELD_AUTHORIZATION, request->http->authstring);
+#elif defined(NEW_API)
+  httpSetField (request->http, HTTP_FIELD_AUTHORIZATION, httpGetAuthString (request->http));
 #endif
 
   if (httpGet(request->http, request->resource))
