diff -Naur gtk+-2.24.0//gdk/Makefile.am gtk+-2.24.0.patched//gdk/Makefile.am
--- gtk+-2.24.0//gdk/Makefile.am	2011-01-05 16:51:20.000000000 +0100
+++ gtk+-2.24.0.patched//gdk/Makefile.am	2011-02-03 23:18:41.000000000 +0100
@@ -201,6 +201,7 @@
 Gdk_2_0_gir_INCLUDES = Gio-2.0 GdkPixbuf-2.0 Pango-1.0 cairo-1.0
 Gdk_2_0_gir_LIBS = $(gdktargetlib)
 Gdk_2_0_gir_FILES = $(introspection_files)
+Gdk_2_0_gir_PACKAGES = gdk-2.0
 Gdk_2_0_gir_CFLAGS = $(INCLUDES)
 INTROSPECTION_GIRS += Gdk-2.0.gir
 
diff -Naur gtk+-2.24.0//gtk/Makefile.am gtk+-2.24.0.patched//gtk/Makefile.am
--- gtk+-2.24.0//gtk/Makefile.am	2011-01-07 05:41:26.000000000 +0100
+++ gtk+-2.24.0.patched//gtk/Makefile.am	2011-02-03 23:19:11.000000000 +0100
@@ -1004,6 +1004,7 @@
 		-DGTK_TEXT_USE_INTERNAL_UNSUPPORTED_API
 Gtk_2_0_gir_LIBS = $(gtktargetlib)
 Gtk_2_0_gir_FILES = $(addprefix $(srcdir)/, $(introspection_files))
+Gtk_2_0_gir_PACKAGES = gtk+-2.0
 INTROSPECTION_GIRS += Gtk-2.0.gir
 
 girdir = $(datadir)/gir-1.0
