--- connection-0.3.1/connection.cabal	2019-09-02 06:33:24.000000000 +0200
+++ connection-0.3.1.new/connection.cabal	2023-07-24 09:13:37.921408040 +0200
@@ -16,7 +16,7 @@
 Build-Type:          Simple
 Category:            Network
 stability:           experimental
-Cabal-Version:       >=1.6
+Cabal-Version:       >=1.8
 Homepage:            https://github.com/vincenthz/hs-connection
 extra-source-files:  README.md
                      CHANGELOG.md
@@ -28,12 +28,12 @@
                    , containers
                    , data-default-class
                    , network >= 2.6.3
-                   , tls >= 1.4
+                   , tls >= 1.7
                    , socks >= 0.6
-                   , x509 >= 1.5
-                   , x509-store >= 1.5
-                   , x509-system >= 1.5
-                   , x509-validation >= 1.5
+                   , crypton-x509 >= 0.33
+                   , crypton-x509-store >= 1.6.9
+                   , crypton-x509-system >= 1.6.7
+                   , crypton-x509-validation >= 1.6.12
   Exposed-modules:   Network.Connection
   Other-modules:     Network.Connection.Types
   ghc-options:       -Wall
