diff --git a/Makefile b/Makefile
index 6e4538c..8d74b2e 100644
--- a/Makefile
+++ b/Makefile
@@ -2,7 +2,7 @@ XSESSIONDIR=/usr/share/xsessions
 BINDIR=/usr/local/bin
 CONFDIR=/etc/xdg
 DATADIR=/usr/local/share
-MODULEDIR=/usr/lib/python2.5/site-packages
+MODULEDIR=/usr/lib/python2.7/site-packages
 
 install:
 	install -d $(XSESSIONDIR) $(BINDIR) $(CONFDIR)/PycaWM $(DATADIR)/PycaWM $(MODULEDIR)/pycawm $(MODULEDIR)/pycawm/plugins
@@ -12,7 +12,7 @@ install:
 	install -m644 conf/*.png $(DATADIR)/PycaWM
 	install -m644 pycawm/*.py $(MODULEDIR)/pycawm
 	install -m644 pycawm/plugins/*.py  $(MODULEDIR)/pycawm/plugins
-	python2.5 -O -c "from compileall import compile_dir; compile_dir('$(MODULEDIR)/pycawm')"
+	python2.7 -O -c "from compileall import compile_dir; compile_dir('$(MODULEDIR)/pycawm')"
 
 
 uninstall:
diff --git a/conf/config-example.py b/conf/config-example.py
index 6cfb61b..0634d00 100755
--- a/conf/config-example.py
+++ b/conf/config-example.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.5
+#!/usr/bin/python2.7
 # -*- coding: utf-8 -*-
 
 #  PycaWM
@@ -63,7 +63,7 @@ pika_file = os.path.normpath(os.path.join(os.getcwd(), sys.argv[0]))
 # The function to call when a key is pressed takes the wm as its only argument.
 
 pika.add_key('C-M-Escape', lambda wm: wm.run_command('xkill'))
-pika.add_key('C-M-r', lambda wm: wm.switch_wm('/usr/bin/python2.5', pika_file))
+pika.add_key('C-M-r', lambda wm: wm.switch_wm('/usr/bin/python2.7', pika_file))
 pika.add_key('C-M-q', lambda wm: wm.quit())
 pika.add_key('M-Tab', lambda wm: ClientsSwitchMenu(wm, 'M-Tab'))
 
diff --git a/pycawm_launcher b/pycawm_launcher
index 53497bb..06f24b3 100755
--- a/pycawm_launcher
+++ b/pycawm_launcher
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.5
+#!/usr/bin/python2.7
 # -*- coding: utf-8 -*-
 
 #  PycaWM
@@ -22,4 +22,4 @@ from os import execl
 
 from pycawm.basedirspec import load_first_config
 
-execl('/usr/bin/python2.5', 'python2.5', load_first_config('config.py'))
+execl('/usr/bin/python2.7', 'python2.7', load_first_config('config.py'))
diff --git a/test.py b/test.py
index f9f6ac3..7607c73 100755
--- a/test.py
+++ b/test.py
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.5
+#!/usr/bin/python2.7
 # -*- coding: utf-8 -*-
 
 #  PycaWM
diff --git a/utils/pycarepl b/utils/pycarepl
index 7668035..bb0dfb7 100755
--- a/utils/pycarepl
+++ b/utils/pycarepl
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.5
+#!/usr/bin/python2.7
 # -*- coding: utf-8 -*-
 
 #  PycaWM
