--- configure.ac
+++ configure_fw.ac
@@ -40,6 +40,9 @@
 AC_CHECK_LIB(jpeg, jpeg_finish_decompress, JPEG_LIBS="-ljpeg", , -lm)
 AC_SUBST([JPEG_LIBS])
 
+AC_CHECK_FUNC([dlopen], [],
+	      AC_CHECK_LIB([dl], [dlopen], DLOPEN_LIBS="-ldl"))
+
 AC_CHECK_LIB(X11, XOpenDisplay, X_LIBS="$X_LIBS -lX11")
 AC_CHECK_LIB(Xext, XextFindDisplay, X_LIBS="$X_LIBS -lXext",,-lX11)
 AC_CHECK_LIB(Xss, XScreenSaverSuspend, X_LIBS="$X_LIBS -lXss",,-lXext -lX11)

--- lib/Makefile.am
+++ lib/Makefile_fw.am
@@ -74,6 +74,8 @@
 	../libusb/libargyllusb.la				\
 	../icc/libargyllicc.la					\
 	$(TIFF_LIBS)						\
+	$(JPEG_LIBS) 						\
+	$(DLOPEN_LIBS) 						\
 	$(X_LIBS)						\
 	$(NULL)
 

