--- raidtools-1.00.3/raid_io.c.orig	2006-11-03 22:52:32.000000000 +0100
+++ raidtools-1.00.3/raid_io.c	2006-11-03 23:51:02.000000000 +0100
@@ -81,23 +81,6 @@
 }
 #undef F
 
-#if !(defined(__alpha__) || defined(__sparc_v9__) || defined(__ia64__) || defined(__x86_64__) || defined(__PIC__))
-# ifndef __NR__llseek
-#  ifdef __sparc__
-#   define __NR__llseek		236
-#  else
-#   define __NR__llseek		140
-#  endif
-# endif
-
-static int _llseek (unsigned int, unsigned long,
-		unsigned long, long long *, unsigned int);
-
-static _syscall5( int, _llseek, unsigned int, fd, unsigned long, offset_high,
-		unsigned long, offset_low, long long *, result,
-		unsigned int, origin)
-#endif
-
 long long raidseek (unsigned int fd, unsigned long blk)
 {
 	unsigned long long offset = (unsigned long long)blk *
@@ -105,14 +88,7 @@
 	long long result;
 	int retval;
 
-#if defined(__alpha__) || defined(__sparc_v9__) || defined(__ia64__) || defined(__x86_64__) || defined(__PIC__)
 	return lseek(fd, offset, SEEK_SET);
-#else
-	retval = _llseek (fd, ((unsigned long long) offset) >> 32,
-			((unsigned long long) offset) & 0xffffffff,
-			&result, SEEK_SET);
-	return (retval == -1 ? (long long) retval : result);
-#endif
 }
 
 int upgrade_sb (int fd, mdp_super_t *sb, md_cfg_entry_t * cfg, int verbose)
