diff -Naur qiv-2.1-pre11/Makefile qiv-2.1-pre11.new/Makefile
--- qiv-2.1-pre11/Makefile	2007-04-25 06:22:39.000000000 +0530
+++ qiv-2.1-pre11.new/Makefile	2007-04-28 12:42:40.000000000 +0530
@@ -58,9 +58,8 @@
 ######################################################################
 
 CC        = gcc
-CFLAGS    = -O2 -Wall \
-	    -fcaller-saves -ffast-math -fno-strength-reduce \
-	    -fthread-jumps #-march=pentium #-DSTAT_MACROS_BROKEN
+CFLAGS    += -fcaller-saves -ffast-math -fno-strength-reduce \
+	    	-fthread-jumps #-march=pentium #-DSTAT_MACROS_BROKEN
 #CFLAGS    = -O2 -Wall -fomit-frame-pointer -finline-functions \
 #	    -fcaller-saves -ffast-math -fno-strength-reduce \
 #	    -fthread-jumps #-march=pentium #-DSTAT_MACROS_BROKEN
