Index: gnome-bluetooth-0.7.0/src/Makefile.am
===================================================================
--- gnome-bluetooth-0.7.0.orig/src/Makefile.am
+++ gnome-bluetooth-0.7.0/src/Makefile.am
@@ -33,41 +33,42 @@ gnomebt-chooser.lo: gnomebt-controller.h
 
 libgnomebt_la_LDFLAGS = \
 	-version-info $(GNOMEBT_LT_VERSION)
+libgnomebt_la_LIBADD = $(GNOME_BLUETOOTH_LIBS)
 
 gnomebt_chooser_test_SOURCES = \
 	gnomebt-chooser-test.c
 
 gnomebt_chooser_test_LDADD = \
-	$(GNOME_BLUETOOTH_LIBS) libgnomebt.la
+	libgnomebt.la $(GNOME_BLUETOOTH_LIBS)
 
 gnomebt_controller_test_SOURCES = \
 	gnomebt-controller-test.c
 
 gnomebt_controller_test_LDADD = \
-	$(GNOME_BLUETOOTH_LIBS) libgnomebt.la
+	libgnomebt.la $(GNOME_BLUETOOTH_LIBS)
 
 gnomebt_fileactiondialog_test_SOURCES = \
 	gnomebt-fileactiondialog-test.c
 
 gnomebt_fileactiondialog_test_LDADD = \
-	$(GNOME_BLUETOOTH_LIBS) libgnomebt.la
+	libgnomebt.la $(GNOME_BLUETOOTH_LIBS)
 
 gnomebt_permissiondialog_test_SOURCES = \
 	gnomebt-permissiondialog-test.c
 
 gnomebt_permissiondialog_test_LDADD = \
-	$(GNOME_BLUETOOTH_LIBS) libgnomebt.la
+	libgnomebt.la $(GNOME_BLUETOOTH_LIBS)
 
 spinner_test_SOURCES = \
 	spinner-test.c
 
 spinner_test_LDADD = \
-	$(GNOME_BLUETOOTH_LIBS) libgnomebt.la
+	libgnomebt.la $(GNOME_BLUETOOTH_LIBS)
 
 gconftest_SOURCES = \
 	gconftest.c
 
-gconftest_LDFLAGS = \
+gconftest_LDADD = \
 	$(GNOME_BLUETOOTH_LIBS) $(BLUETOOTH_LIBS)
 
 GOB_DIST = \
@@ -121,13 +122,13 @@ gnomebt-chooser-py.c: gnomebt-chooser.de
 pydir = @pythondir@/gnomebt/
 
 controller_la_SOURCES = gnomebt-controller-py.c gnomebt-controller-pymodule.c
-controller_la_LIBADD =  libgnomebt.la
-controller_la_LDFLAGS = $(GNOME_BLUETOOTH_PYTHON_LIBS) \
-			$(GNOME_BLUETOOTH_LIBS) \
-			-module -avoid-version
+controller_la_LIBADD =  libgnomebt.la \
+			$(GNOME_BLUETOOTH_PYTHON_LIBS) \
+			$(GNOME_BLUETOOTH_LIBS)
+controller_la_LDFLAGS =	-module -avoid-version
 
 chooser_la_SOURCES = gnomebt-chooser-py.c gnomebt-chooser-pymodule.c
-chooser_la_LIBADD =  libgnomebt.la
-chooser_la_LDFLAGS = 	$(GNOME_BLUETOOTH_PYTHON_LIBS) \
-			$(GNOME_BLUETOOTH_LIBS) \
-			-module -avoid-version
+chooser_la_LIBADD =  libgnomebt.la \
+			$(GNOME_BLUETOOTH_PYTHON_LIBS) \
+			$(GNOME_BLUETOOTH_LIBS)
+chooser_la_LDFLAGS = -module -avoid-version
