Description: Fix 64-bit issues with the ODBC support.
Author: Steve Langasek <vorlon@debian.org>
Origin: http://bugs.debian.org/cgi-bin/bugreport.cgi?msg=10;filename=mdbtools-64bit.patch;att=1;bug=360436
Bug-Debian: http://bugs.debian.org/360436

Index: mdbtools/src/odbc/odbc.c
===================================================================
--- mdbtools.orig/src/odbc/odbc.c	2010-04-18 02:42:09.000000000 +0100
+++ mdbtools/src/odbc/odbc.c	2010-04-18 03:03:06.000000000 +0100
@@ -208,10 +208,10 @@
 SQLRETURN SQL_API SQLDescribeParam(
     SQLHSTMT           hstmt,
     SQLUSMALLINT       ipar,
-    SQLSMALLINT FAR   *pfSqlType,
-    SQLUINTEGER FAR   *pcbParamDef,
-    SQLSMALLINT FAR   *pibScale,
-    SQLSMALLINT FAR   *pfNullable)
+    SQLSMALLINT       *pfSqlType,
+    SQLULEN           *pcbParamDef,
+    SQLSMALLINT       *pibScale,
+    SQLSMALLINT       *pfNullable)
 {
 	TRACE("SQLDescribeParam");
 	return SQL_SUCCESS;
@@ -220,9 +220,9 @@
 SQLRETURN SQL_API SQLExtendedFetch(
     SQLHSTMT           hstmt,
     SQLUSMALLINT       fFetchType,
-    SQLINTEGER         irow,
-    SQLUINTEGER FAR   *pcrow,
-    SQLUSMALLINT FAR  *rgfRowStatus)
+    SQLLEN             irow,
+    SQLULEN           *pcrow,
+    SQLUSMALLINT      *rgfRowStatus)
 {
 struct _hstmt *stmt = (struct _hstmt *) hstmt;
 struct _hdbc *dbc = (struct _hdbc *) stmt->hdbc;
@@ -296,8 +296,8 @@
 
 SQLRETURN SQL_API SQLParamOptions(
     SQLHSTMT           hstmt,
-    SQLUINTEGER        crow,
-    SQLUINTEGER FAR   *pirow)
+    SQLULEN            crow,
+    SQLULEN FAR       *pirow)
 {
 	TRACE("SQLParamOptions");
 	return SQL_SUCCESS;
@@ -346,7 +346,7 @@
 
 SQLRETURN SQL_API SQLSetPos(
     SQLHSTMT           hstmt,
-    SQLUSMALLINT       irow,
+    SQLSETPOSIROW      irow,
     SQLUSMALLINT       fOption,
     SQLUSMALLINT       fLock)
 {
@@ -397,11 +397,11 @@
     SQLSMALLINT        fParamType,
     SQLSMALLINT        fCType,
     SQLSMALLINT        fSqlType,
-    SQLUINTEGER        cbColDef,
+    SQLULEN            cbColDef,
     SQLSMALLINT        ibScale,
     SQLPOINTER         rgbValue,
-    SQLINTEGER         cbValueMax,
-    SQLINTEGER FAR    *pcbValue)
+    SQLLEN             cbValueMax,
+    SQLLEN FAR        *pcbValue)
 {
 struct _hstmt *stmt;
 
@@ -501,8 +501,8 @@
     SQLUSMALLINT       icol,
     SQLSMALLINT        fCType,
     SQLPOINTER         rgbValue,
-    SQLINTEGER         cbValueMax,
-    SQLINTEGER FAR    *pcbValue)
+    SQLLEN             cbValueMax,
+    SQLLEN FAR        *pcbValue)
 {
 	struct _hstmt *stmt = (struct _hstmt *) hstmt;
 	struct _sql_bind_info *cur, *newitem;
@@ -593,7 +593,7 @@
     SQLSMALLINT        cbColNameMax,
     SQLSMALLINT FAR   *pcbColName,
     SQLSMALLINT FAR   *pfSqlType,
-    SQLUINTEGER FAR   *pcbColDef, /* precision */
+    SQLULEN FAR       *pcbColDef, /* precision */
     SQLSMALLINT FAR   *pibScale,
     SQLSMALLINT FAR   *pfNullable)
 {
@@ -648,7 +648,7 @@
     SQLPOINTER         rgbDesc,
     SQLSMALLINT        cbDescMax,
     SQLSMALLINT FAR   *pcbDesc,
-    SQLINTEGER FAR    *pfDesc)
+    SQLLEN FAR        *pfDesc)
 {
 	int namelen, i;
 	struct _hstmt *stmt;
@@ -968,7 +968,7 @@
 
 SQLRETURN SQL_API SQLRowCount(
     SQLHSTMT           hstmt,
-    SQLINTEGER FAR    *pcrow)
+    SQLLEN FAR        *pcrow)
 {
 struct _hstmt *stmt=(struct _hstmt *)hstmt;
 
@@ -1001,10 +1001,10 @@
     SQLUSMALLINT       ipar,
     SQLSMALLINT        fCType,
     SQLSMALLINT        fSqlType,
-    SQLUINTEGER        cbParamDef,
+    SQLULEN            cbParamDef,
     SQLSMALLINT        ibScale,
     SQLPOINTER         rgbValue,
-    SQLINTEGER FAR     *pcbValue)
+    SQLLEN FAR        *pcbValue)
 {
 	TRACE("SQLSetParam");
 	return SQL_SUCCESS;
@@ -1124,8 +1124,8 @@
     SQLUSMALLINT       icol,
     SQLSMALLINT        fCType,
     SQLPOINTER         rgbValue,
-    SQLINTEGER         cbValueMax,
-    SQLINTEGER FAR    *pcbValue)
+    SQLLEN             cbValueMax,
+    SQLLEN     FAR    *pcbValue)
 {
 	struct _hstmt *stmt;
 	struct _hdbc *dbc;
@@ -1429,7 +1429,7 @@
 SQLRETURN SQL_API SQLPutData(
     SQLHSTMT           hstmt,
     SQLPOINTER         rgbValue,
-    SQLINTEGER         cbValue)
+    SQLLEN             cbValue)
 {
 	TRACE("SQLPutData");
 	return SQL_SUCCESS;
@@ -1438,7 +1438,7 @@
 SQLRETURN SQL_API SQLSetConnectOption(
     SQLHDBC            hdbc,
     SQLUSMALLINT       fOption,
-    SQLUINTEGER        vParam)
+    SQLULEN            vParam)
 {
 	TRACE("SQLSetConnectOption");
 	return SQL_SUCCESS;
@@ -1447,7 +1447,7 @@
 SQLRETURN SQL_API SQLSetStmtOption(
     SQLHSTMT           hstmt,
     SQLUSMALLINT       fOption,
-    SQLUINTEGER        vParam)
+    SQLULEN            vParam)
 {
 	TRACE("SQLSetStmtOption");
 	return SQL_SUCCESS;
