diff -Naur dbus-sharp-0.8.1/configure.ac dbus-sharp-0.8.1.new/configure.ac
--- dbus-sharp-0.8.1/configure.ac	2014-06-24 19:10:54.000000000 +0200
+++ dbus-sharp-0.8.1.new/configure.ac	2024-04-11 16:26:34.935403331 +0200
@@ -23,9 +23,9 @@
 MONO_REQ_VERSION=1.1.13
 PKG_CHECK_MODULES(MONO, mono >= $MONO_REQ_VERSION)
 
-AC_PATH_PROG(GMCS, gmcs, no)
+AC_PATH_PROG(GMCS, mcs, no)
 if test "x$GMCS" = "xno"; then
-	AC_MSG_ERROR([You need to install gmcs])
+	AC_MSG_ERROR([You need to install mcs])
 fi
 AC_SUBST(GMCS)
 
