diff -Naur gimmix-0.4.0.old/src/gimmix-systray.c gimmix-0.4.0/src/gimmix-systray.c
--- gimmix-0.4.0.old/src/gimmix-systray.c	2007-02-23 23:41:43.000000000 +0530
+++ gimmix-0.4.0/src/gimmix-systray.c	2007-02-25 01:18:50.000000000 +0530
@@ -105,16 +105,27 @@
 static gboolean
 cb_systray_enter_notify (GtkWidget *widget, GdkEventCrossing *event, gpointer data)
 {
-	GdkScreen *screen; 
+	GdkScreen *screen = NULL;
+	GdkScreen *def_screen = NULL;
 	GdkRectangle rectangle;
 	gint x, y;
 	gint w, h;
+	gint top;
 	
 	if (strncasecmp(cfg_get_key_value(conf, "enable_notification"), "true", 4) != 0)
 	return FALSE;
-
+	
+	/* Check where to place our tooltip */
+	def_screen = gdk_screen_get_default ();
+	w = gdk_screen_get_width (def_screen);
+	h = gdk_screen_get_height (def_screen);
 	/* Get the location of the system tray icon */
 	gdk_window_get_origin ((GTK_WIDGET(icon)->window), &x, &y);
+	if (h-y >= 100)
+		top = 1; /* tooltip should be placed on top */
+	else
+		top = 0; /* tooltip should be placed on bottom */
+	w = h = 0;
 	
 	/* Move the tooltip off-screen to calculate the exact co-ordinates */
 	rectangle.x = 2500;
@@ -127,8 +138,17 @@
 	gtk_window_get_size (GTK_WINDOW(stooltip), &w, &h);
 	
 	/* Good, now lets move it back to where it should be */
-	rectangle.x = x-(w/4);
-	rectangle.y = y-120;
+	if (top == 1)
+	{	
+		rectangle.x = x-(w/4);
+		rectangle.y = y-25;
+	}
+	else
+	{
+		rectangle.x = x-(w/4);
+		rectangle.y = y-120;
+	}
+
 	sexy_tooltip_position_to_rect (SEXY_TOOLTIP(stooltip), &rectangle, screen);
 
 	return TRUE;
