diff -urN gimmix-0.1b.orig/Makefile gimmix-0.1b/Makefile
--- gimmix-0.1b.orig/Makefile	2006-10-27 14:10:19.000000000 +0100
+++ gimmix-0.1b/Makefile	2006-10-27 14:23:12.000000000 +0100
@@ -2,19 +2,19 @@
 # gimmix-0.1beta
 
 # Installation directory
-PREFIX=/usr
+PREFIX=$(DESTDIR)/usr
 
 ALLFILES=*.c
 
-CONFUSE_LIB=`pkg-config libconfuse --libs`
-NOTIFY_CFLAGS=`pkg-config libnotify --cflags`
-NOTIFY_LIB=`pkg-config libnotify --libs`
-GLADE_CFLAGS=`pkg-config libglade-2.0 --cflags`
-GLADE_LIB=`pkg-config libglade-2.0 --libs`
-MPD_CFLAGS=`pkg-config libmpd --cflags`
-MPD_LIB=`pkg-config libmpd --libs`
+CONFUSE_LIB=$(shell pkg-config libconfuse --libs)
+NOTIFY_CFLAGS=$(shell pkg-config libnotify --cflags)
+NOTIFY_LIB=$(shell pkg-config libnotify --libs)
+GLADE_CFLAGS=$(shell pkg-config libglade-2.0 --cflags)
+GLADE_LIB=$(shell pkg-config libglade-2.0 --libs)
+MPD_CFLAGS=$(shell pkg-config libmpd --cflags)
+MPD_LIB=$(shell pkg-config libmpd --libs)
 
-CFLAGS=-march=i686 -pipe -O2 -Wall -fPIC -DPREFIX=\"$(PREFIX)\" $(NOTIFY_CFLAGS) $(GLADE_CFLAGS) $(MPD_CFLAGS) -I.
+CFLAGS += -fPIC -DPREFIX=\"$(PREFIX)\" $(NOTIFY_CFLAGS) $(GLADE_CFLAGS) $(MPD_CFLAGS) -I.
 LIBS=$(CONFUSE_LIB) $(NOTIFY_LIB) $(GLADE_LIB) $(MPD_LIB)
 
 OBJS = config.o gimmix-core.o playlist.o interface.o gimmix.o
@@ -47,4 +47,3 @@
 	install -D gimmix $(PREFIX)/bin/gimmix
 	install -D -m644 pixmaps/gimmix.png $(PREFIX)/share/pixmaps/gimmix.png
 	install -D -m644 glade/gimmix.glade $(PREFIX)/share/gimmix/gimmix.glade
-
