--- configure.in.old	2010-11-30 22:46:20.190215458 -0500
+++ configure.in	2010-11-30 22:47:07.138144807 -0500
@@ -11,9 +11,10 @@
 dnl Checks for programs.
 AC_PROG_CC
 AC_PROG_RANLIB
+AM_PROG_CC_C_O
 
 dnl various CFLAGS
-CFLAGS="$CFLAGS -g -Wall"
+CFLAGS="$CFLAGS -Wall"
 #CFLAGS="-O0 -g -Wall"
 
 dnl Checks for libraries.
@@ -36,6 +37,7 @@
 dnl L10n support
 ALL_LINGUAS="en de"
 
+AC_GNU_SOURCE
 AM_GNU_GETTEXT
 
 XGETTEXTFLAGS=
