--- hal-0.5.14/hal.conf.in.old	2009-02-04 22:07:23.000000000 +0000
+++ hal-0.5.14/hal.conf.in	2010-02-12 16:38:00.000000000 +0000
@@ -55,5 +55,22 @@
 
   </policy>
 
+  <!-- Default policy for the exported interfaces; if PolicyKit is not used
+       for access control you will need to modify this -->
+  <policy group="0">
+    <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"
+	   send_destination="org.freedesktop.Hal"/>
+    <allow send_interface="org.freedesktop.Hal.Device.VideoAdapterPM"
+	   send_destination="org.freedesktop.Hal"/>
+    <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"
+	   send_destination="org.freedesktop.Hal"/>
+   </policy>
+
+    <policy group="storage">
+    <allow send_interface="org.freedesktop.Hal.Device.Volume"
+	   send_destination="org.freedesktop.Hal"/>
+    <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"
+	   send_destination="org.freedesktop.Hal"/>
+  </policy>
 </busconfig>
 
