diff -Naur hal-0.5.10/tools/linux/hal-system-power-hibernate-linux hal-0.5.10.new/tools/linux/hal-system-power-hibernate-linux
--- hal-0.5.10/tools/linux/hal-system-power-hibernate-linux	2007-07-26 23:30:28.000000000 +0530
+++ hal-0.5.10.new/tools/linux/hal-system-power-hibernate-linux	2008-03-01 11:56:55.000000000 +0530
@@ -26,6 +26,10 @@
 if [ -x /usr/sbin/pm-hibernate ] ; then
 	/usr/sbin/pm-hibernate $QUIRKS
 	RET=$?
+elif [ -w "/sys/power/state" ] ; then
+	# use the raw kernel interface
+	echo "disk" > /sys/power/state
+	RET=$?
 else
 	unsupported
 fi
diff -Naur hal-0.5.10/tools/linux/hal-system-power-suspend-linux hal-0.5.10.new/tools/linux/hal-system-power-suspend-linux
--- hal-0.5.10/tools/linux/hal-system-power-suspend-linux	2007-07-26 23:30:28.000000000 +0530
+++ hal-0.5.10.new/tools/linux/hal-system-power-suspend-linux	2008-03-01 11:57:03.000000000 +0530
@@ -39,6 +39,10 @@
 if [ -x "/usr/sbin/pm-suspend" ] ; then
 	/usr/sbin/pm-suspend $QUIRKS
 	RET=$?
+elif [ -w "/sys/power/state" ] ; then
+	# use the raw kernel interface
+	echo "mem" > /sys/power/state
+	RET=$?
 else
 	# TODO: add support
 	unsupported
