From b10f077898d916ce12558675cfa96d90d1c070e8 Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Mon, 28 Aug 2017 15:33:08 +0200
Subject: [PATCH 2/4] workaround vbox with efi

---
 src/modules/bootloader/main.py | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/modules/bootloader/main.py b/src/modules/bootloader/main.py
index da9641fbe..13f83f733 100644
--- a/src/modules/bootloader/main.py
+++ b/src/modules/bootloader/main.py
@@ -142,6 +142,15 @@ def install_grub(efi_directory, fw_type):
                                               efi_bootloader_id,
                                               "grubx64.efi")}
         shutil.copy2(efi_file_source[efi_bitness], install_efi_boot_directory)
+        # detect Vbox and workaround efi boot issue with grub
+        vbox_efi_file_name = "startup.nsh"
+        vbox_efi_file_path = install_efi_directory + '/' + vbox_efi_file_name
+        vbox_sysfile = open("/sys/devices/virtual/dmi/id/bios_version", "r")
+        vbox_out = vbox_sysfile.read(10)
+        if 'VirtualBox' in vbox_out:
+            with open(vbox_efi_file_path, "w") as vb:
+                vb.write("\\EFI\\boot\\grubx64.efi")
+            # end
     else:
         print("Bootloader: grub (bios)")
         if libcalamares.globalstorage.value("bootLoader") is None:
-- 
2.14.1

