From 8ffab7ec8e4a2fcb3e9650f5676ca6e0098f4d86 Mon Sep 17 00:00:00 2001
From: Gabriel Craciunescu <crazy@frugalware.org>
Date: Sun, 22 Oct 2017 02:20:29 +0200
Subject: [PATCH 2/6] chrootcfg: our pacman doesn't have cachedir args

---
 src/modules/chrootcfg/main.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/modules/chrootcfg/main.py b/src/modules/chrootcfg/main.py
index 459cb7158..1bb315de8 100644
--- a/src/modules/chrootcfg/main.py
+++ b/src/modules/chrootcfg/main.py
@@ -138,9 +138,9 @@ class PacmanController:
         cachedir = join(self.root, "var/cache/pacman-g2/pkg")
         dbdir = join(self.root, "var/lib/pacman-g2")
 
-        args = 'pacman --noconfirm {}  --root {} --dbpath {} {}'
+        args = 'pacman --noconfirm {} --root {} --dbpath {} {}'
         cmd = args.format(
-            '-U' if local else '-Sy', cachedir, self.root, dbdir,
+            '-U' if local else '-Sy', self.root, dbdir,
             ' '.join(pkglist)
         )
 
-- 
2.14.2

