From faea845162cfc8b7e2c9c6b7b1e03c595d48a608 Mon Sep 17 00:00:00 2001
From: Gabriel Craciunescu <crazy@frugalware.org>
Date: Sun, 22 Oct 2017 03:29:09 +0200
Subject: [PATCH 3/6] chrootcfg: remove dbpatch too

---
 src/modules/chrootcfg/main.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/modules/chrootcfg/main.py b/src/modules/chrootcfg/main.py
index 1bb315de8..b0bc09d98 100644
--- a/src/modules/chrootcfg/main.py
+++ b/src/modules/chrootcfg/main.py
@@ -134,13 +134,13 @@ class PacmanController:
     def install(self, pkglist, local=False):
         """Install the given packagelist
         """
-
+        # not used
         cachedir = join(self.root, "var/cache/pacman-g2/pkg")
         dbdir = join(self.root, "var/lib/pacman-g2")
 
-        args = 'pacman --noconfirm {} --root {} --dbpath {} {}'
+        args = 'pacman --noconfirm {} --root {} {}'
         cmd = args.format(
-            '-U' if local else '-Sy', self.root, dbdir,
+            '-U' if local else '-Sy', self.root,
             ' '.join(pkglist)
         )
 
-- 
2.14.2

