From e87497dc764470f72a7b7b8bedf0dbecba768da4 Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Mon, 28 Aug 2017 15:41:58 +0200
Subject: [PATCH 4/4] install virtualbox-guest-additions when vbox is detected

---
 src/modules/packages/main.py | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/modules/packages/main.py b/src/modules/packages/main.py
index 227712fa5..0a09ebcec 100644
--- a/src/modules/packages/main.py
+++ b/src/modules/packages/main.py
@@ -69,6 +69,11 @@ class PackageManager:
 
             check_target_env_call(["pacman", pacman_flags, "--noconfirm",
                                    "--noprogressbar", "--nointegrity"] + pkgs)
+            vbox_sysfile = open("/sys/devices/virtual/dmi/id/bios_version", "r")
+            vbox_out = vbox_sysfile.read(10)
+            vpkg = "virtualbox-guest-additions"
+            if 'VirtualBox' in vbox_out:
+                check_target_env_call(["pacman", pacman_flags, "--noconfirm", vpkg])
             check_target_env_call(["pacman", "-Scc", "--noconfirm"])
         elif self.backend == "portage":
             check_target_env_call(["emerge", "-v"] + pkgs)
-- 
2.14.1

