From 00e5013768ae999e3dd4ba569d300bbabf593043 Mon Sep 17 00:00:00 2001
From: Gabriel Craciunescu <crazy@frugalware.org>
Date: Sun, 22 Oct 2017 03:41:10 +0200
Subject: [PATCH 4/6] chrootcfg: added --noarch --noprogressbar to install

---
 src/modules/chrootcfg/main.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/modules/chrootcfg/main.py b/src/modules/chrootcfg/main.py
index b0bc09d98..3d5f06e82 100644
--- a/src/modules/chrootcfg/main.py
+++ b/src/modules/chrootcfg/main.py
@@ -138,7 +138,7 @@ class PacmanController:
         cachedir = join(self.root, "var/cache/pacman-g2/pkg")
         dbdir = join(self.root, "var/lib/pacman-g2")
 
-        args = 'pacman --noconfirm {} --root {} {}'
+        args = 'pacman --noconfirm --noarch --noprogressbar {} --root {} {}'
         cmd = args.format(
             '-U' if local else '-Sy', self.root,
             ' '.join(pkglist)
-- 
2.14.2

