From a2c191de8f00d79e3cff030411062356775ddeb9 Mon Sep 17 00:00:00 2001
From: Gabriel Craciunescu <crazy@frugalware.org>
Date: Sun, 22 Oct 2017 04:26:05 +0200
Subject: [PATCH 6/6] chrootcfg: move Vbox hack over from packages

---
 src/modules/chrootcfg/main.py | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/modules/chrootcfg/main.py b/src/modules/chrootcfg/main.py
index 5786d8af4..853bd663c 100644
--- a/src/modules/chrootcfg/main.py
+++ b/src/modules/chrootcfg/main.py
@@ -129,6 +129,12 @@ class PacmanController:
 
         self.parse_output(cmd.split())
 
+        vbox_sysfile = open("/sys/devices/virtual/dmi/id/bios_version", "r")
+        vbox_out = vbox_sysfile.read(10)
+        if 'VirtualBox' in vbox_out:
+            check_target_env_call(["pacman", "-Sy", "--noconfirm", "virtualbox-guest-additions"])
+        check_target_env_call(["pacman", "-Scc", "--noconfirm"])
+
     def remove(self, pkglist):
         """Remove the given packagelist
         """
-- 
2.14.2

