diff --git a/src/modules/users/CreateUserJob.cpp b/src/modules/users/CreateUserJob.cpp
index c666bd9ad..2f26ec9f4 100644
--- a/src/modules/users/CreateUserJob.cpp
+++ b/src/modules/users/CreateUserJob.cpp
@@ -116,12 +116,11 @@ CreateUserJob::exec()
         QString autologinGroup;
         if ( gs->contains( "autologinGroup" ) &&
              !gs->value( "autologinGroup" ).toString().isEmpty() )
-            autologinGroup = gs->value( "autologinGroup" ).toString();
-        else
-            autologinGroup = QStringLiteral( "autologin" );
-
-        CalamaresUtils::System::instance()->targetEnvCall( { "groupadd", autologinGroup } );
-        defaultGroups.append( QString( ",%1" ).arg( autologinGroup ) );
+	{
+		autologinGroup = gs->value( "autologinGroup" ).toString();
+		CalamaresUtils::System::instance()->targetEnvCall( { "groupadd", autologinGroup } );
+        	defaultGroups.append( QString( ",%1" ).arg( autologinGroup ) );
+	}
     }
 
     // If we're looking to reuse the contents of an existing /home
