diff -Naur lirc-0.10.1/daemons/lircd.cpp lirc-0.10.1.new/daemons/lircd.cpp
--- lirc-0.10.1/daemons/lircd.cpp	2017-09-10 10:52:19.000000000 +0200
+++ lirc-0.10.1.new/daemons/lircd.cpp	2022-06-02 10:40:21.350309394 +0200
@@ -485,8 +485,7 @@
 
 	if (curr_driver->fd != -1 && curr_driver->drvctl_func) {
 		if ((curr_driver->features & LIRC_CAN_SET_REC_CARRIER)
-		    || (curr_driver->features & LIRC_CAN_SET_REC_TIMEOUT)
-		    || (curr_driver->features & LIRC_CAN_SET_REC_FILTER)) {
+		    || (curr_driver->features & LIRC_CAN_SET_REC_TIMEOUT)) {
 				ret = setup_frequency() && setup_timeout();
 		}
 	}
@@ -2066,8 +2065,6 @@
 			const char* button_name;
 			int reps;
 
-			if (curr_driver->drvctl_func && (curr_driver->features & LIRC_CAN_NOTIFY_DECODE))
-				curr_driver->drvctl_func(DRVCTL_NOTIFY_DECODE, NULL);
 
 			get_release_data(&remote_name, &button_name, &reps);
 
diff -Naur lirc-0.10.1/tools/lirc-lsplugins.cpp lirc-0.10.1.new/tools/lirc-lsplugins.cpp
--- lirc-0.10.1/tools/lirc-lsplugins.cpp	2017-09-10 10:52:19.000000000 +0200
+++ lirc-0.10.1.new/tools/lirc-lsplugins.cpp	2022-06-02 10:39:39.857026044 +0200
@@ -57,8 +57,7 @@
 	"#    c: LIRC_CAN_SET_SEND_CARRIER\n" \
 	"#    d: LIRC_CAN_SET_SEND_DUTY_CYCLE\n" \
 	"#    t: LIRC_CAN_SET_TRANSMITTER_MASK\n" \
-	"#    C: LIRC_CAN_MEASURE_CARRIER\n" \
-	"#    D: LIRC_CAN_NOTIFY_DECODE\n"
+	"#    C: LIRC_CAN_MEASURE_CARRIER\n"
 
 const struct option options[] = {
 	{ "plugindir",	  required_argument, NULL, 'U' },
@@ -291,7 +290,7 @@
 	char buff[256];
 
 	snprintf(buff, sizeof(buff),
-		 "%c%c%c%c%c%c%c%c%c%c%c%c%c ",
+		 "%c%c%c%c%c%c%c%c%c%c%c%c ",
 		 get(LIRC_CAN_SEND_RAW, 'R', hw),
 		 get(LIRC_CAN_SEND_PULSE, 'P', hw),
 		 get(LIRC_CAN_SEND_MODE2, 'M', hw),
@@ -303,8 +302,7 @@
 		 get(LIRC_CAN_SET_SEND_CARRIER, 'c', hw),
 		 get(LIRC_CAN_SET_SEND_DUTY_CYCLE, 'd', hw),
 		 get(LIRC_CAN_SET_TRANSMITTER_MASK, 't', hw),
-		 get(LIRC_CAN_MEASURE_CARRIER, 'C', hw),
-		 get(LIRC_CAN_NOTIFY_DECODE, 'D', hw)
+		 get(LIRC_CAN_MEASURE_CARRIER, 'C', hw)
 		 );
 	line->features = strdup(buff);
 }
