--- lirc-0.9.4d/contrib/60-lirc.rules	2017-01-17 08:42:58.000000000 +0100
+++ lirc-0.9.4d.new/contrib/60-lirc.rules	2017-03-17 22:06:28.000000000 +0100
@@ -6,4 +6,4 @@
 # Grant rw permission to the lirc group for USB devices using acl(5).
 
 ACTION=="add", SUBSYSTEM=="usb", ENV{DEVNAME}=="?*", \
-    RUN+="/usr/bin/sh -c '/usr/bin/setfacl -m g:lirc:rw %E{DEVNAME} 2>/dev/null || :'"
+    RUN+="/bin/sh -c '/bin/setfacl -m g:lirc:rw %E{DEVNAME} 2>/dev/null || :'"
