diff -Naur lirc-0.10.1/Makefile.am lirc-0.10.1-p/Makefile.am
--- lirc-0.10.1/Makefile.am	2017-09-10 10:52:19.000000000 +0200
+++ lirc-0.10.1-p/Makefile.am	2019-06-02 21:29:03.160448826 +0200
@@ -16,9 +16,6 @@
 VERBOSE                 = $(subst 0,,$(V))
 AM_LIBTOOLFLAGS         = $(if $(VERBOSE),,--silent)
 
-if HAVE_PYTHON35
-PYTHON_TARBALL          = python-pkg/dist/lirc-$(BASE_VERSION).tar.gz
-endif
 
 EXTRA_DIST              = README autogen.sh CONTRIBUTE.md ChangeLog \
                           COPYING.curl COPYING.ciniparser
