diff -Naur lirc-0.10.1/daemons/lircd.cpp lirc-0.10.1-p/daemons/lircd.cpp
--- lirc-0.10.1/daemons/lircd.cpp	2017-09-10 10:52:19.000000000 +0200
+++ lirc-0.10.1-p/daemons/lircd.cpp	2019-01-17 00:18:43.648518187 +0100
@@ -948,7 +948,7 @@
 
 	lirc_log_open("lircd", nodaemon, loglevel);
 
-	/* create pid lockfile in /var/run */
+	/* create pid lockfile in /run */
 	fd = open(pidfile, O_RDWR | O_CREAT, 0644);
 	if (fd > 0)
 		pidf = fdopen(fd, "r+");
diff -Naur lirc-0.10.1/lib/lirc/lirc_config.h lirc-0.10.1-p/lib/lirc/lirc_config.h
--- lirc-0.10.1/lib/lirc/lirc_config.h	2017-09-10 10:52:58.000000000 +0200
+++ lirc-0.10.1-p/lib/lirc/lirc_config.h	2019-01-17 00:18:43.668518294 +0100
@@ -5,10 +5,10 @@
  * @ingroup driver_api
  */
 
-/** lircd socket file name - beneath $varrundir (default /var/run/lirc) */
+/** lircd socket file name - beneath $varrundir (default /run/lirc) */
 #define DEV_LIRCD       "lircd"
 
-/** lircmd socket file name - beneath $varrundir (default /var/run/lirc) */
+/** lircmd socket file name - beneath $varrundir (default /run/lirc) */
 #define DEV_LIRCM       "lircm"
 
 /** lircd,conf  file name - beneath SYSCONFDIR (default /etc) */
@@ -71,7 +71,7 @@
 /** Environment variable overriding options file path. */
 #define LIRC_OPTIONS_VAR        "LIRC_OPTIONS_PATH"
 
-/** Default permissions for /var/run/lircd. */
+/** Default permissions for /run/lircd. */
 #define DEFAULT_PERMISSIONS     "666"
 
 /** Default timeout (ms) while waiting for socket. */
diff -Naur lirc-0.10.1/lib/lirc/paths.h lirc-0.10.1-p/lib/lirc/paths.h
--- lirc-0.10.1/lib/lirc/paths.h	2017-09-10 10:52:58.000000000 +0200
+++ lirc-0.10.1-p/lib/lirc/paths.h	2019-01-17 00:18:43.671518310 +0100
@@ -2,7 +2,7 @@
 #define PATHS_H
 #define DATADIR          "/usr/local/share"
 #define SYSCONFDIR       "/usr/local/etc"
-#define VARRUNDIR        "/usr/local/var/run"
+#define VARRUNDIR        "/usr/local/run"
 #define LOCALSTATEDIR    "/usr/local/var"
 #define LIBDIR           "/usr/local/lib"
 #endif
diff -Naur lirc-0.10.1/lib/lirc_client.dox lirc-0.10.1-p/lib/lirc_client.dox
--- lirc-0.10.1/lib/lirc_client.dox	2017-09-10 10:52:57.000000000 +0200
+++ lirc-0.10.1-p/lib/lirc_client.dox	2019-01-17 00:18:43.660518251 +0100
@@ -117,7 +117,7 @@
 </P>
 <p>
     By default the client connects to the hard-coded default path,
-    usually /var/run/lirc/lircd. The environment variable LIRC_SOCKET_PATH
+    usually /run/lirc/lircd. The environment variable LIRC_SOCKET_PATH
     can be used to connect to another socket.
 </p>
 <P>
diff -Naur lirc-0.10.1/lib/lirc_config.h lirc-0.10.1-p/lib/lirc_config.h
--- lirc-0.10.1/lib/lirc_config.h	2017-09-10 10:52:19.000000000 +0200
+++ lirc-0.10.1-p/lib/lirc_config.h	2019-01-17 00:18:43.657518235 +0100
@@ -5,10 +5,10 @@
  * @ingroup driver_api
  */
 
-/** lircd socket file name - beneath $varrundir (default /var/run/lirc) */
+/** lircd socket file name - beneath $varrundir (default /run/lirc) */
 #define DEV_LIRCD       "lircd"
 
-/** lircmd socket file name - beneath $varrundir (default /var/run/lirc) */
+/** lircmd socket file name - beneath $varrundir (default /run/lirc) */
 #define DEV_LIRCM       "lircm"
 
 /** lircd,conf  file name - beneath SYSCONFDIR (default /etc) */
@@ -71,7 +71,7 @@
 /** Environment variable overriding options file path. */
 #define LIRC_OPTIONS_VAR        "LIRC_OPTIONS_PATH"
 
-/** Default permissions for /var/run/lircd. */
+/** Default permissions for /run/lircd. */
 #define DEFAULT_PERMISSIONS     "666"
 
 /** Default timeout (ms) while waiting for socket. */
diff -Naur lirc-0.10.1/lirc_options.conf lirc-0.10.1-p/lirc_options.conf
--- lirc-0.10.1/lirc_options.conf	2017-09-10 10:52:19.000000000 +0200
+++ lirc-0.10.1-p/lirc_options.conf	2019-01-17 00:18:43.645518171 +0100
@@ -10,8 +10,8 @@
 nodaemon        = False
 driver          = devinput
 device          = auto
-output          = /var/run/lirc/lircd
-pidfile         = /var/run/lirc/lircd.pid
+output          = /run/lirc/lircd
+pidfile         = /run/lirc/lircd.pid
 plugindir       = /usr/lib/lirc/plugins
 permission      = 666
 allow-simulate  = No
diff -Naur lirc-0.10.1/python-pkg/doc/mainpage.dox lirc-0.10.1-p/python-pkg/doc/mainpage.dox
--- lirc-0.10.1/python-pkg/doc/mainpage.dox	2017-09-10 10:52:19.000000000 +0200
+++ lirc-0.10.1-p/python-pkg/doc/mainpage.dox	2019-01-17 00:18:43.636518122 +0100
@@ -14,6 +14,6 @@
 An otherwise undocumented module is config.py. This, which can be
 imported using *import lirc.config*, provides access to the build
 configuration  e. g., SYSCONFDIR (usually /etc) and VARRUNDIR
-(typically /var/run).
+(typically /run).
 
 */
diff -Naur lirc-0.10.1/python-pkg/lirc/client.py lirc-0.10.1-p/python-pkg/lirc/client.py
--- lirc-0.10.1/python-pkg/lirc/client.py	2017-09-10 10:52:19.000000000 +0200
+++ lirc-0.10.1-p/python-pkg/lirc/client.py	2019-01-17 00:18:43.639518138 +0100
@@ -17,7 +17,7 @@
 #
 #   The otherwise undocumented file config.py, which can be imported using
 #   <i>import lirc.config</i>,  provides access to the paths defined when
-#   running configure such as VARRUNDIR (often /var/run) and SYSCONFDIR
+#   running configure such as VARRUNDIR (often /run) and SYSCONFDIR
 #   (typically /etc).
 
 #   pylint: disable=W0613
diff -Naur lirc-0.10.1/tools/irtestcase.cpp lirc-0.10.1-p/tools/irtestcase.cpp
--- lirc-0.10.1/tools/irtestcase.cpp	2017-09-10 10:52:19.000000000 +0200
+++ lirc-0.10.1-p/tools/irtestcase.cpp	2019-01-17 00:18:43.633518106 +0100
@@ -30,7 +30,7 @@
 	"irtestcase [-p prog -l lircrc] [-t testdata] <socket>\n"
 	"irtestcase [ħ | -v]\n\n"
 	"<socket> is the socket connecting to lircd. Defaults to a hardcoded\n"
-	"default value, usually /var/run/lirc/lircd. Respects LIRC_SOCKET_PATH in\n"
+	"default value, usually /run/lirc/lircd. Respects LIRC_SOCKET_PATH in\n"
 	"environment.\n\n"
 	"Options:\n"
 	"   -l  lircrc  Log also translated symbols using lircrc type config file.\n"
