diff -Naur krusader-1.60.0.orig/krusader/MountMan/kmountman.h krusader-1.60.0/krusader/MountMan/kmountman.h
--- krusader-1.60.0.orig/krusader/MountMan/kmountman.h	2004-10-27 16:08:27.000000000 +0200
+++ krusader-1.60.0/krusader/MountMan/kmountman.h	2005-10-30 23:29:17.000000000 +0100
@@ -45,6 +45,8 @@
 #include <stdlib.h>
 #include <math.h>
 
+class KMountManGUI;
+
 class KMountMan : public QObject {
    Q_OBJECT
    friend class KMountManGUI;
diff -Naur krusader-1.60.0.orig/krusader/Panel/krdetailedview.h krusader-1.60.0/krusader/Panel/krdetailedview.h
--- krusader-1.60.0.orig/krusader/Panel/krdetailedview.h	2005-02-08 23:01:10.000000000 +0100
+++ krusader-1.60.0/krusader/Panel/krdetailedview.h	2005-10-30 23:29:40.000000000 +0100
@@ -40,6 +40,8 @@
 #include "krview.h"
 #include "krviewitem.h"
 
+class KrDetailedViewItem;
+
 // extends KrViewProperties to add detailedview-only properties
 class KrDetailedViewProperties: public KrViewProperties {
 public:
diff -Naur krusader-1.60.0.orig/krusader/Panel/listpanel.h krusader-1.60.0/krusader/Panel/listpanel.h
--- krusader-1.60.0.orig/krusader/Panel/listpanel.h	2005-02-04 17:15:10.000000000 +0100
+++ krusader-1.60.0/krusader/Panel/listpanel.h	2005-10-30 23:28:45.000000000 +0100
@@ -71,6 +71,7 @@
 class SyncBrowseButton;
 class KrBookmarkButton;
 class KPushButton;
+class ListPanelFunc;
 
 class ListPanel : public QWidget {
    friend class ListPanelFunc;
