diff -Naur seq24-0.8.7/src/event.cpp seq24-0.8.7.new/src/event.cpp
--- seq24-0.8.7/src/event.cpp	2006-06-02 08:50:20.000000000 +0530
+++ seq24-0.8.7.new/src/event.cpp	2008-11-24 00:46:16.000000000 +0530
@@ -17,6 +17,7 @@
 //  Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 //
 //-----------------------------------------------------------------------------
+#include <string.h>
 #include "event.h"
 
 event::event()
