diff -ur xmms-scrobbler-0.3.8.1/tags/id3v2.c xmms-scrobbler-0.3.8.1-fixed/tags/id3v2.c
--- xmms-scrobbler-0.3.8.1/tags/id3v2.c	2005-02-20 11:11:22.000000000 -0800
+++ xmms-scrobbler-0.3.8.1-fixed/tags/id3v2.c	2006-07-15 16:38:29.000000000 -0700
@@ -409,6 +409,14 @@
 			resync_t *unsync = checkunsync(data, framedata->len);
 			framedata->len -= unsync->count;
 			free(unsync);
+
+			if (framedata->len == 0)
+			{
+				free(data);
+				framedata->data = realloc(framedata->data, 1);
+				strcpy(framedata->data, "\0");
+				return framedata;
+			}
 		}
 		if(encoding == 0x00)
 		{
@@ -477,6 +485,9 @@
 		
 		framedata = parseFrame(&bp, end, id3_data);
 		
+		if (framedata == NULL)
+			continue;
+
 		id3v2->items = realloc(id3v2->items, (id3v2->numitems + 1) * 
 			sizeof(framedata_t *));
 		id3v2->items[id3v2->numitems++] = framedata;
