diff -ru xfce4-power-manager-0.8.3.1.orig/src/xfpm-dpms.c xfce4-power-manager-0.8.3.1/src/xfpm-dpms.c
--- xfce4-power-manager-0.8.3.1.orig/src/xfpm-dpms.c	2009-07-15 11:47:37.000000000 +0200
+++ xfce4-power-manager-0.8.3.1/src/xfpm-dpms.c	2009-08-11 01:07:59.715110853 +0200
@@ -32,7 +32,6 @@
 
 #include <X11/Xproto.h>
 #include <X11/extensions/dpms.h>
-#include <X11/extensions/dpmsstr.h>
 
 #include <libxfce4util/libxfce4util.h>
 

