diff -Naur hal-0.5.10/configure.in hal-0.5.10-p/configure.in
--- hal-0.5.10/configure.in	2007-09-25 17:47:27.000000000 +0200
+++ hal-0.5.10-p/configure.in	2007-11-02 15:03:49.000000000 +0100
@@ -244,7 +244,7 @@
    AC_DEFINE(USE_PARTED,1,[Whether libparted is to be used])
    AC_CHECK_LIB(uuid, uuid_generate, [], AC_MSG_ERROR([*** uuid library (libuuid) not found]))
    AC_CHECK_LIB(dl, dlopen,          [], AC_MSG_ERROR([*** dl library (libdl) not found]))
-   AC_MSG_CHECKING([for libparted == 1.7.1, 1.8.0, 1.8.1, 1.8.2, 1.8.6, 1.8.7])
+   AC_MSG_CHECKING([for libparted >= 1.7.1 or >= 1.8.0])
    LDFLAGS=-lparted
    AC_RUN_IFELSE([AC_LANG_SOURCE([[
    #include <stdio.h>
@@ -258,18 +258,14 @@
 	if ( sscanf( ped_get_version(), "%d.%d.%d", &major, &minor, &micro ) == 3 )
 		printf( "Found libparted %s", ped_get_version() ) ;
 
-	if ((major == 1 && minor == 7 && micro == 1) ||
-	    (major == 1 && minor == 8 && micro == 0) ||
-	    (major == 1 && minor == 8 && micro == 1) ||
-	    (major == 1 && minor == 8 && micro == 2) ||
-	    (major == 1 && minor == 8 && micro == 6) ||
-	    (major == 1 && minor == 8 && micro == 7))
+	if ((major == 1 && minor == 7 && micro >= 1) ||
+	    (major == 1 && minor == 8 && micro >= 0))
 	   return 0;
 
 	return 1;
     }
     ]])], AC_MSG_RESULT([]),
-       AC_MSG_ERROR([*** Requires libparted == 1.7.1/1.8.0/1.8.1/1.8.2/1.8.6 or 1.8.7]))
+       AC_MSG_ERROR([*** Requires libparted >= 1.7.1 or >= 1.8.0]))
     PARTED_LIBS=-lparted
     AC_SUBST(PARTED_LIBS)
     LDFLAGS=
