diff --git a/doc/api/.gitignore b/doc/api/.gitignore
diff --git a/doc/api/Makefile.am b/doc/api/Makefile.am
diff --git a/doc/api/hal-docs.xml b/doc/api/hal-docs.xml
diff --git a/hald/hald_dbus.c b/hald/hald_dbus.c
index 7a51bfb..741a52e 100644
--- a/hald/hald_dbus.c
+++ b/hald/hald_dbus.c
@@ -2902,7 +2902,8 @@ device_is_executing_method (HalDevice *d
 
 	ret = FALSE;
 
-	if (g_hash_table_lookup_extended (udi_to_method_queue, d->udi, &origkey, (gpointer) &queue)) {
+	if (udi_to_method_queue != NULL &&
+	    g_hash_table_lookup_extended (udi_to_method_queue, d->udi, &origkey, (gpointer) &queue)) {
 
 		if (queue != NULL) {
 			MethodInvocation *mi;
diff --git a/tools/hal-storage-shared.c b/tools/hal-storage-shared.c
