diff -Naur hal-0.5.9.orig/configure.in hal-0.5.9/configure.in
--- hal-0.5.9.orig/configure.in	2007-04-03 06:36:44.000000000 +0200
+++ hal-0.5.9/configure.in	2007-05-14 23:48:12.000000000 +0200
@@ -193,7 +193,7 @@
    AC_DEFINE(USE_PARTED,1,[Whether libparted is to be used])
    AC_CHECK_LIB(uuid, uuid_generate, [], AC_MSG_ERROR([*** uuid library (libuuid) not found]))
    AC_CHECK_LIB(dl, dlopen,          [], AC_MSG_ERROR([*** dl library (libdl) not found]))
-   AC_MSG_CHECKING([for libparted == 1.7.1, 1.8.0, 1.8.1, 1.8.2, 1.8.6])
+   AC_MSG_CHECKING([for libparted == 1.7.1, 1.8.0, 1.8.1, 1.8.2, 1.8.6, 1.8.7])
    LDFLAGS=-lparted
    AC_TRY_RUN(
    #include <stdio.h>
@@ -211,12 +211,13 @@
 	    (major == 1 && minor == 8 && micro == 0) ||
 	    (major == 1 && minor == 8 && micro == 1) ||
 	    (major == 1 && minor == 8 && micro == 2) ||
-	    (major == 1 && minor == 8 && micro == 6))
+	    (major == 1 && minor == 8 && micro == 6) ||
+	    (major == 1 && minor == 8 && micro == 7))
 	   return 0;
 
 	return 1;
     }
-	,AC_MSG_RESULT( ),AC_MSG_ERROR([*** Requires libparted == 1.7.1/1.8.0/1.8.1/1.8.2 or 1.8.6]) )
+	,AC_MSG_RESULT( ),AC_MSG_ERROR([*** Requires libparted == 1.7.1/1.8.0/1.8.1/1.8.2/1.8.6 or 1.8.7]) )
     PARTED_LIBS=-lparted
     AC_SUBST(PARTED_LIBS)
     LDFLAGS=
