diff --git a/tools/hal-storage-mount.c b/tools/hal-storage-mount.c
index ef8d1a9..47412a8 100644
--- a/tools/hal-storage-mount.c
+++ b/tools/hal-storage-mount.c
@@ -728,6 +728,7 @@ #endif
 		 */
 		if (strcmp (libhal_volume_get_fstype (volume), "vfat") != 0 &&
 		    strcmp (libhal_volume_get_fstype (volume), "iso9660") != 0 &&
+		    strcmp (libhal_volume_get_fstype (volume), "ntfs") != 0 &&
 		    strcmp (libhal_volume_get_fstype (volume), "udf") != 0) {
 			pol_change_uid = wants_to_change_uid;
 		}
