diff --git a/tools/hal-storage-mount.c b/tools/hal-storage-mount.c
index 597c50e..5b98876 100644
--- a/tools/hal-storage-mount.c
+++ b/tools/hal-storage-mount.c
@@ -67,7 +67,7 @@ #define MOUNT_OPTIONS	"noexec,nosuid"
 #define MOUNT_TYPE_OPT	"-F"
 #else
 #define MOUNT		"/bin/mount"
-#define MOUNT_OPTIONS	"noexec,nosuid,nodev"
+#define MOUNT_OPTIONS	"nosuid,nodev"
 #define MOUNT_TYPE_OPT	"-t"
 #endif
 
