diff -Naur hal-0.5.11/hal.conf.in hal-0.5.11-p-conf/hal.conf.in
--- hal-0.5.11/hal.conf.in	2008-05-08 01:24:13.000000000 +0200
+++ hal-0.5.11-p-conf/hal.conf.in	2009-01-26 13:32:20.000000000 +0100
@@ -16,17 +16,25 @@
 
   <!-- Allow anyone to invoke methods on the Manager and Device interfaces -->
   <policy context="default">
-    <allow send_interface="org.freedesktop.Hal.Manager"/>
-    <allow send_interface="org.freedesktop.Hal.Device"/>
+    <allow send_interface="org.freedesktop.DBus.Introspectable"
+           send_destination="org.freedesktop.Hal"/>
+    <allow send_interface="org.freedesktop.Hal.Manager"
+	   send_destination="org.freedesktop.Hal"/>
+    <allow send_interface="org.freedesktop.Hal.Device"
+	   send_destination="org.freedesktop.Hal"/>
     <allow receive_interface="org.freedesktop.Hal.Manager"
            receive_sender="org.freedesktop.Hal"/>
     <allow receive_interface="org.freedesktop.Hal.Device"
            receive_sender="org.freedesktop.Hal"/>
 
-    <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
-    <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
-    <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
-    <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
+    <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"
+	   send_destination="org.freedesktop.Hal"/>
+    <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"
+           send_destination="org.freedesktop.Hal"/>
+    <allow send_interface="org.freedesktop.Hal.Device.Volume"
+	   send_destination="org.freedesktop.Hal"/>
+    <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"
+	   send_destination="org.freedesktop.Hal"/>
     <allow receive_interface="org.freedesktop.Hal.Device.SystemPowerManagement"
 	   receive_sender="org.freedesktop.Hal"/>
     <allow receive_interface="org.freedesktop.Hal.Device.LaptopPanel"
@@ -39,12 +47,20 @@
 
   <!-- Default policy for the exported interfaces; if PolicyKit is not used
        for access control you will need to modify this -->
-  <policy context="default">
-    <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
-    <allow send_interface="org.freedesktop.Hal.Device.VideoAdapterPM"/>
-    <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
-    <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
-    <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
+  <policy group="0">
+    <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"
+	   send_destination="org.freedesktop.Hal"/>
+    <allow send_interface="org.freedesktop.Hal.Device.VideoAdapterPM"
+	   send_destination="org.freedesktop.Hal"/>
+    <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"
+	   send_destination="org.freedesktop.Hal"/>
+   </policy>
+
+    <policy group="storage">
+    <allow send_interface="org.freedesktop.Hal.Device.Volume"
+	   send_destination="org.freedesktop.Hal"/>
+    <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"
+	   send_destination="org.freedesktop.Hal"/>
   </policy>
 
 </busconfig>
