diff -Naur hal-0.5.7.1/hal.conf.in hal-0.5.7.1-p/hal.conf.in
--- hal-0.5.7.1/hal.conf.in	2006-01-10 20:03:45.000000000 +0100
+++ hal-0.5.7.1-p/hal.conf.in	2006-08-04 13:56:09.584432200 +0200
@@ -33,8 +33,9 @@
     <deny send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
   </policy>
 
-  <!-- This will not work if pam_console support is not enabled -->
-  <policy at_console="true">
+  <!-- This will not work if pam_console support is not enabled 
+   well we don't have PAM so disable this  -->
+  <policy at_console="false">
     <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
     <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
     <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
@@ -44,6 +45,8 @@
   <!-- You can change this to a more suitable user, or make per-group -->
   <policy user="0">
     <allow send_interface="org.freedesktop.Hal.Device.SystemPowerManagement"/>
+  </policy>
+  <policy group="storage">
     <allow send_interface="org.freedesktop.Hal.Device.LaptopPanel"/>
     <allow send_interface="org.freedesktop.Hal.Device.Volume"/>
     <allow send_interface="org.freedesktop.Hal.Device.Volume.Crypto"/>
