--- hal-0.5.8.1.vanilla/configure	2006-09-19 22:24:41.000000000 +0200
+++ hal-0.5.8.1/configure	2006-09-26 21:09:20.000000000 +0200
@@ -22383,9 +22383,9 @@
 
 fi;
 BUILD_MACBOOKPRO=no
-if test "x$use_macbookpro" = "xyes" ; then
+if test "x$with_macbookpro" = "xyes" ; then
    BUILD_MACBOOKPRO=yes
-elif test "x$use_macbookpro" = "x" ; then
+elif test "x$with_macbookpro" = "x" ; then
   if test "$HAVE_LIBPCI" != "false" ; then
     case "${host}" in
       i[3456]86-*-linux*)
