diff -Naurp hal-0.5.8.1/configure.in hal-0.5.8.1-p/configure.in
--- hal-0.5.8.1/configure.in	2006-11-21 02:36:05.000000000 +0100
+++ hal-0.5.8.1-p/configure.in	2006-11-21 02:37:34.000000000 +0100
@@ -250,9 +250,9 @@ AC_CHECK_HEADERS(pci/pci.h, [
 dnl see if we should build macbookpro utils
 AC_ARG_WITH(macbookpro,     [  --with-macbookpro       Whether to build Macbook Pro utils (auto)])
 BUILD_MACBOOKPRO=no
-if test "x$use_macbookpro" = "xyes" ; then
+if test "x$with_macbookpro" = "xyes" ; then
    BUILD_MACBOOKPRO=yes
-elif test "x$use_macbookpro" = "x" ; then
+elif test "x$with_macbookpro" = "x" ; then
   if test "$HAVE_LIBPCI" != "false" ; then
     case "${host}" in
       i[[3456]]86-*-linux*)
