diff -Naur hal-0.5.8.1/configure.in hal-0.5.8.1-p/configure.in
--- hal-0.5.8.1/configure.in	2006-09-19 22:24:05.000000000 +0200
+++ hal-0.5.8.1-p/configure.in	2007-02-06 03:37:09.000000000 +0100
@@ -250,10 +250,10 @@
 dnl see if we should build macbookpro utils
 AC_ARG_WITH(macbookpro,     [  --with-macbookpro       Whether to build Macbook Pro utils (auto)])
 BUILD_MACBOOKPRO=no
-if test "x$use_macbookpro" = "xyes" ; then
+if test "x$with_macbookpro" = "xyes" ; then
    BUILD_MACBOOKPRO=yes
-elif test "x$use_macbookpro" = "x" ; then
-  if test "$HAVE_LIBPCI" != "false" ; then
+elif test "x$with_macbookpro" = "x" ; then
+  if test "x$USE_LIBPCI" != "xno" ; then
     case "${host}" in
       i[[3456]]86-*-linux*)
         BUILD_MACBOOKPRO=yes
