diff -Naurp hal-0.5.8.1/configure.in hal-0.5.8.1-p/configure.in
--- hal-0.5.8.1/configure.in	2006-09-19 22:24:05.000000000 +0200
+++ hal-0.5.8.1-p/configure.in	2006-11-21 02:34:51.000000000 +0100
@@ -108,7 +108,7 @@ if test "x$use_parted" = "xyes" ; then
    AC_DEFINE(USE_PARTED,1,[Whether libparted is to be used])
    AC_CHECK_LIB(uuid, uuid_generate, [], AC_MSG_ERROR([*** uuid library (libuuid) not found]))
    AC_CHECK_LIB(dl, dlopen,          [], AC_MSG_ERROR([*** dl library (libdl) not found]))
-   AC_MSG_CHECKING(for libparted == 1.7.1)
+   AC_MSG_CHECKING(for libparted >= 1.7.1)
    LDFLAGS=-lparted
    AC_TRY_RUN(
    #include <stdio.h>
@@ -122,9 +122,9 @@ if test "x$use_parted" = "xyes" ; then
 	if ( sscanf( ped_get_version(), "%d.%d.%d", &major, &minor, &micro ) == 3 )
 		printf( "Found libparted %s", ped_get_version() ) ;
 
-        return ! ( major == 1 && minor == 7 && micro == 1 ) ;
+        return ! ( major == 1 && minor >= 7 && micro >= 0 ) ;
     }
-	,AC_MSG_RESULT( ),AC_MSG_ERROR(*** Requires libparted == 1.7.1) )
+	,AC_MSG_RESULT( ),AC_MSG_ERROR(*** Requires libparted >= 1.7.1) )
     PARTED_LIBS=-lparted
     AC_SUBST(PARTED_LIBS)
     LDFLAGS=
