diff -Naur hpoj-0.91.orig/apps/cmdline/Makefile.in hpoj-0.91/apps/cmdline/Makefile.in
--- hpoj-0.91.orig/apps/cmdline/Makefile.in	2002-06-14 03:49:49.000000000 +0200
+++ hpoj-0.91/apps/cmdline/Makefile.in	2005-08-20 19:30:40.000000000 +0200
@@ -5,7 +5,7 @@
 DISTCLEANS=Makefile
 INCLUDES=Makefile ../../include/ptal.h ../../include/hpojip.h
 
-CFLAGS=-O -Wall -g @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@ @VAR_RUN_PREFIX_CFLAGS@
+CFLAGS += @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@ @VAR_RUN_PREFIX_CFLAGS@
 
 all:		$(GOALS)
 
diff -Naur hpoj-0.91.orig/apps/xojpanel/Makefile.in hpoj-0.91/apps/xojpanel/Makefile.in
--- hpoj-0.91.orig/apps/xojpanel/Makefile.in	2002-06-13 02:40:44.000000000 +0200
+++ hpoj-0.91/apps/xojpanel/Makefile.in	2005-08-20 19:30:40.000000000 +0200
@@ -3,7 +3,7 @@
 MOC=@QT_MOC@
 
 LFLAGS = @LIBRARY_CMDLINE@ @LIBQT_CMDLINE@ -lptal
-CFLAGS = -O -Wall -g -I/usr/X11R6/include @INCLUDE_CMDLINE@
+CFLAGS += -I/usr/X11R6/include @INCLUDE_CMDLINE@
 
 PIXMAPS = hpojlcd.xpm hpoj_lcdmon.xpm hpoj_mini.xpm
 COMMON_HEADERS = Makefile xojpanel.h
diff -Naur hpoj-0.91.orig/lib/hpojip/Makefile.in hpoj-0.91/lib/hpojip/Makefile.in
--- hpoj-0.91.orig/lib/hpojip/Makefile.in	2002-01-08 02:24:19.000000000 +0100
+++ hpoj-0.91/lib/hpojip/Makefile.in	2005-08-20 19:30:40.000000000 +0200
@@ -26,7 +26,7 @@
 # TODO: Detect system endianness in configure script!
 # For now, little-endian (Intel LSB-first) is assumed.
 
-CFLAGS=-O -Wall -g @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@
+CFLAGS += @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@
 
 %.shared.o:	%.c $(INCLUDES)
 		$(CC) $(CFLAGS) -DHPOJIP_INTERNAL -fPIC -c -o $@ $<
diff -Naur hpoj-0.91.orig/lib/ptal/Makefile.in hpoj-0.91/lib/ptal/Makefile.in
--- hpoj-0.91.orig/lib/ptal/Makefile.in	2002-10-16 01:34:28.000000000 +0200
+++ hpoj-0.91/lib/ptal/Makefile.in	2005-08-20 19:30:40.000000000 +0200
@@ -17,7 +17,7 @@
 MLCDDIR=../../mlcd
 INCLUDES=Makefile $(INCLUDEDIR)/ptal.h ptal-internal.h $(MLCDDIR)/mlcd.h
 
-CFLAGS=-O -Wall -g @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@ @VAR_RUN_PREFIX_CFLAGS@ -DUCD_COMPATIBLE
+CFLAGS += @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@ @VAR_RUN_PREFIX_CFLAGS@ -DUCD_COMPATIBLE
 
 %.shared.o:	%.c $(INCLUDES)
 		$(CC) $(CFLAGS) -fPIC -c -o $@ $<
diff -Naur hpoj-0.91.orig/lib/sane/Makefile.in hpoj-0.91/lib/sane/Makefile.in
--- hpoj-0.91.orig/lib/sane/Makefile.in	2002-07-10 03:13:39.000000000 +0200
+++ hpoj-0.91/lib/sane/Makefile.in	2005-08-20 19:30:40.000000000 +0200
@@ -16,7 +16,7 @@
 INCLUDEDIR=../../include
 INCLUDES=Makefile $(INCLUDEDIR)/ptal.h hpoj.h hpoj-tables.h sane.h saneopts.h
 
-CFLAGS=-O -Wall -g @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@
+CFLAGS += @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@
 
 %.shared.o:	%.c $(INCLUDES)
 		$(CC) $(CFLAGS) -fPIC -c -o $@ $<
diff -Naur hpoj-0.91.orig/Makefile.in hpoj-0.91/Makefile.in
--- hpoj-0.91.orig/Makefile.in	2003-09-23 03:16:41.000000000 +0200
+++ hpoj-0.91/Makefile.in	2005-08-20 20:01:56.000000000 +0200
@@ -118,14 +118,6 @@
 	@ echo
 	@ export PATH=/sbin:/usr/sbin:$$PATH:/usr/local/bin ; \
 	if test `id -u` -eq 0 ; then \
-		if test -d /etc/profile.d ; then \
-			echo "export PATH=$(bindir):$(sbindir):\$$PATH" >/etc/profile.d/hpoj.sh ; \
-			echo "setenv PATH $(bindir):$(sbindir):\$$PATH" >/etc/profile.d/hpoj.csh ; \
-			chmod +x /etc/profile.d/hpoj.sh /etc/profile.d/hpoj.csh ; \
-		else \
-			echo "You may need to add $(bindir) and $(sbindir) to your PATH." ; \
-		fi ; \
-		echo ; \
 		for dir in /etc/init.d /etc/rc.d/init.d /sbin/init.d XXXX ; do \
 			if test $$dir = XXXX ; then \
 				if ! test -d /usr/local/etc/rc.d ; then \
@@ -158,17 +150,7 @@
 				echo "Setting libsane-hpoj.so symlink." ; \
 				$(LN_S) -f $(libdir)/libsane-hpoj.so $(SANE_BACKEND_DIR)/libsane-hpoj.so.1 ; \
 			fi ; \
-			if ! grep ^hpoj $(SANE_ETC_DIR)/dll.conf >/dev/null 2>/dev/null ; then \
-				echo "Adding hpoj entry to $(SANE_ETC_DIR)/dll.conf." ; \
-				echo hpoj >>$(SANE_ETC_DIR)/dll.conf ; \
-			fi ; \
-		fi ; \
-		echo ; \
-		if ! grep $(libdir) /etc/ld.so.conf >/dev/null 2>/dev/null ; then \
-			echo "Adding $(libdir) to /etc/ld.so.conf." ; \
-			echo $(libdir) >>/etc/ld.so.conf ; \
 		fi ; \
-		ldconfig ; \
 	else \
 		if ! test -d /etc/profile.d || ! test -f /etc/profile.d/hpoj.sh || ! test -f /etc/profile.d/hpoj.csh ; then \
 			echo "You may need to add $(bindir) and $(sbindir) to your PATH." ; \
diff -Naur hpoj-0.91.orig/mlcd/Makefile.in hpoj-0.91/mlcd/Makefile.in
--- hpoj-0.91.orig/mlcd/Makefile.in	2003-04-30 04:39:31.000000000 +0200
+++ hpoj-0.91/mlcd/Makefile.in	2005-08-20 19:30:40.000000000 +0200
@@ -19,7 +19,7 @@
 		transport/ExMlcTransport.h \
 
 PWD=$(shell pwd)
-CFLAGS=-I$(PWD) -I$(PWD)/transport -O -g -Wall -DEX_TRANSPORT_UNIX_PORT -DPAR_PLATFORM_@PAR_PLATFORM@ -DUSB_PLATFORM_@USB_PLATFORM@ -DJD_DEBUGLITE @MLCD_CFLAGS@ @VAR_RUN_PREFIX_CFLAGS@
+CFLAGS += -I$(PWD) -I$(PWD)/transport -DEX_TRANSPORT_UNIX_PORT -DPAR_PLATFORM_@PAR_PLATFORM@ -DUSB_PLATFORM_@USB_PLATFORM@ -DJD_DEBUGLITE @MLCD_CFLAGS@ @VAR_RUN_PREFIX_CFLAGS@
 LFLAGS=@MLCD_LFLAGS@
 
 %.o :		%.cpp $(INCLUDES)
