--- hpoj-0.91/apps/cmdline/Makefile.in.optflags	2005-02-16 11:32:32.643046812 +0000
+++ hpoj-0.91/apps/cmdline/Makefile.in	2005-02-16 11:32:49.105898371 +0000
@@ -5,7 +5,7 @@
 DISTCLEANS=Makefile
 INCLUDES=Makefile ../../include/ptal.h ../../include/hpojip.h
 
-CFLAGS=-O -Wall -g @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@ @VAR_RUN_PREFIX_CFLAGS@
+CFLAGS += @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@ @VAR_RUN_PREFIX_CFLAGS@
 
 all:		$(GOALS)
 
--- hpoj-0.91/apps/xojpanel/Makefile.in.optflags	2005-02-16 11:32:57.774240773 +0000
+++ hpoj-0.91/apps/xojpanel/Makefile.in	2005-02-16 11:33:15.887777423 +0000
@@ -3,7 +3,7 @@
 MOC=@QT_MOC@
 
 LFLAGS = @LIBRARY_CMDLINE@ @LIBQT_CMDLINE@ -lptal
-CFLAGS = -O -Wall -g -I/usr/X11R6/include @INCLUDE_CMDLINE@
+CFLAGS += -I/usr/X11R6/include @INCLUDE_CMDLINE@
 
 PIXMAPS = hpojlcd.xpm hpoj_lcdmon.xpm hpoj_mini.xpm
 COMMON_HEADERS = Makefile xojpanel.h
--- hpoj-0.91/lib/hpojip/Makefile.in.optflags	2005-02-16 11:33:27.385579295 +0000
+++ hpoj-0.91/lib/hpojip/Makefile.in	2005-02-16 11:33:38.664423235 +0000
@@ -26,7 +26,7 @@
 # TODO: Detect system endianness in configure script!
 # For now, little-endian (Intel LSB-first) is assumed.
 
-CFLAGS=-O -Wall -g @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@
+CFLAGS += @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@
 
 %.shared.o:	%.c $(INCLUDES)
 		$(CC) $(CFLAGS) -DHPOJIP_INTERNAL -fPIC -c -o $@ $<
--- hpoj-0.91/lib/ptal/Makefile.in.optflags	2005-02-16 11:33:49.688316111 +0000
+++ hpoj-0.91/lib/ptal/Makefile.in	2005-02-16 11:34:00.776196953 +0000
@@ -17,7 +17,7 @@
 MLCDDIR=../../mlcd
 INCLUDES=Makefile $(INCLUDEDIR)/ptal.h ptal-internal.h $(MLCDDIR)/mlcd.h
 
-CFLAGS=-O -Wall -g @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@ @VAR_RUN_PREFIX_CFLAGS@ -DUCD_COMPATIBLE
+CFLAGS += @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@ @VAR_RUN_PREFIX_CFLAGS@ -DUCD_COMPATIBLE
 
 %.shared.o:	%.c $(INCLUDES)
 		$(CC) $(CFLAGS) -fPIC -c -o $@ $<
--- hpoj-0.91/lib/sane/Makefile.in.optflags	2005-02-16 11:34:12.602936798 +0000
+++ hpoj-0.91/lib/sane/Makefile.in	2005-02-16 11:34:24.799606180 +0000
@@ -16,7 +16,7 @@
 INCLUDEDIR=../../include
 INCLUDES=Makefile $(INCLUDEDIR)/ptal.h hpoj.h hpoj-tables.h sane.h saneopts.h
 
-CFLAGS=-O -Wall -g @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@
+CFLAGS += @DEFINES_CMDLINE@ @INCLUDE_CMDLINE@ @LIBRARY_CMDLINE@
 
 %.shared.o:	%.c $(INCLUDES)
 		$(CC) $(CFLAGS) -fPIC -c -o $@ $<
--- hpoj-0.91/mlcd/Makefile.in.optflags	2005-02-16 11:34:32.749087269 +0000
+++ hpoj-0.91/mlcd/Makefile.in	2005-02-16 11:34:48.706038666 +0000
@@ -19,7 +19,7 @@
 		transport/ExMlcTransport.h \
 
 PWD=$(shell pwd)
-CFLAGS=-I$(PWD) -I$(PWD)/transport -O -g -Wall -DEX_TRANSPORT_UNIX_PORT -DPAR_PLATFORM_@PAR_PLATFORM@ -DUSB_PLATFORM_@USB_PLATFORM@ -DJD_DEBUGLITE @MLCD_CFLAGS@ @VAR_RUN_PREFIX_CFLAGS@
+CFLAGS += -I$(PWD) -I$(PWD)/transport -DEX_TRANSPORT_UNIX_PORT -DPAR_PLATFORM_@PAR_PLATFORM@ -DUSB_PLATFORM_@USB_PLATFORM@ -DJD_DEBUGLITE @MLCD_CFLAGS@ @VAR_RUN_PREFIX_CFLAGS@
 LFLAGS=@MLCD_LFLAGS@
 
 %.o :		%.cpp $(INCLUDES)
