diff -Naur qtdeclarative-opensource-src-5.6.1/src/qml/qml/qqmltypeloader.cpp qtdeclarative-opensource-src-5.6.1-p/src/qml/qml/qqmltypeloader.cpp
--- qtdeclarative-opensource-src-5.6.1/src/qml/qml/qqmltypeloader.cpp	2016-05-20 22:56:32.000000000 +0200
+++ qtdeclarative-opensource-src-5.6.1-p/src/qml/qml/qqmltypeloader.cpp	2016-06-28 18:38:59.960333683 +0200
@@ -1961,7 +1961,8 @@
         QList<TypeCache::Iterator> unneededTypes;
         for (TypeCache::Iterator iter = m_typeCache.begin(), end = m_typeCache.end(); iter != end; ++iter)  {
             QQmlTypeData *typeData = iter.value();
-            if (typeData->m_compiledData && typeData->m_compiledData->count() == 1) {
+            if (typeData->m_compiledData && typeData->count() == 1
+		    && typeData->m_compiledData->count() == 1) {
                 // There are no live objects of this type
                 unneededTypes.append(iter);
             }
