diff -urN dhcpcd-3.0.17.orig/pathnames.h dhcpcd-3.0.17/pathnames.h
--- dhcpcd-3.0.17.orig/pathnames.h	2007-05-30 12:14:01.000000000 +0100
+++ dhcpcd-3.0.17/pathnames.h	2007-05-30 12:14:57.000000000 +0100
@@ -36,11 +36,11 @@
 #define NTPFILE             ETCDIR "/ntp.conf"
 #define NTPDRIFTFILE        ETCDIR "/ntp.drift"
 #define NTPLOGFILE          "/var/log/ntp.log"
-#define NTPSERVICE          ETCDIR "/init.d/ntpd"
+#define NTPSERVICE          ETCDIR "/rc.d/rc.ntpd"
 #define NTPRESTARTARGS      "--quiet", "conditionalrestart"
 
 #define OPENNTPFILE         ETCDIR "/ntpd.conf"
-#define OPENNTPSERVICE      ETCDIR "/init.d/ntpd"
+#define OPENNTPSERVICE      ETCDIR "/rc.d/rc.ntpd"
 #define OPENNTPRESTARTARGS  "--quiet", "conditionalrestart"
 
 #define DEFAULT_SCRIPT      ETCDIR "/" PACKAGE ".sh"
