diff -urN dhcpcd-3.0.2.orig/pathnames.h dhcpcd-3.0.2/pathnames.h
--- dhcpcd-3.0.2.orig/pathnames.h	2006-12-03 20:32:08.000000000 +0000
+++ dhcpcd-3.0.2/pathnames.h	2006-12-03 20:35:51.000000000 +0000
@@ -32,10 +32,10 @@
 #define DEFAULT_SCRIPT		ETCDIR "/" PACKAGE ".sh"
 
 #define STATEDIR		"/var"
-#define PIDFILE			STATEDIR "/run/" PACKAGE "-%s.pid"
+#define PIDFILE			ETCDIR "/dhcpc/" PACKAGE "-%s.pid"
 
-#define CONFIGDIR		STATEDIR "/lib/" PACKAGE
-#define INFOFILE		CONFIGDIR "/" PACKAGE "-%s.info"
+#define CONFIGDIR		ETCDIR "/dhcpc"
+#define INFOFILE		ETCDIR "/dhcpc/" PACKAGE "-%s.info"
 
 #define NTPLOGFILE		"/var/log/ntp.log"
 
