diff -Naur dhcpcd-3.1.5.orig/config.h dhcpcd-3.1.5/config.h
--- dhcpcd-3.1.5.orig/config.h	2007-08-26 22:04:38.000000000 +0200
+++ dhcpcd-3.1.5/config.h	2007-08-26 22:05:56.000000000 +0200
@@ -61,11 +61,11 @@
 #define NTPFILE             ETCDIR "/ntp.conf"
 #define NTPDRIFTFILE        ETCDIR "/ntp.drift"
 #define NTPLOGFILE          "/var/log/ntp.log"
-#define NTPSERVICE          ETCDIR "/init.d/ntpd"
+#define NTPSERVICE          ETCDIR "/rc.d/rc.ntpd"
 #define NTPRESTARTARGS      "--nodeps", "--quiet", "conditionalrestart"
 
 #define OPENNTPFILE         ETCDIR "/ntpd.conf"
-#define OPENNTPSERVICE      ETCDIR "/init.d/ntpd"
+#define OPENNTPSERVICE      ETCDIR "/rc.d/rc.ntpd"
 #define OPENNTPRESTARTARGS  "--nodeps", "--quiet", "conditionalrestart"
 
 #define DEFAULT_SCRIPT      ETCDIR "/" PACKAGE ".sh"
