diff -urN dhcpcd-3.0.17/pathnames.h dhcpcd-3.0.17.new/pathnames.h
--- dhcpcd-3.0.17/pathnames.h	2007-04-14 08:54:09.000000000 +0100
+++ dhcpcd-3.0.17.new/pathnames.h	2007-04-30 18:40:51.000000000 +0100
@@ -40,15 +40,15 @@
 #define NTPRESTARTARGS      "--quiet", "conditionalrestart"
 
 #define OPENNTPFILE         ETCDIR "/ntpd.conf"
-#define OPENNTPSERVICE      ETCDIR "/init.d/ntpd"
+#define OPENNTPSERVICE      ETCDIR "/rc.d/rc.ntpd"
 #define OPENNTPRESTARTARGS  "--quiet", "conditionalrestart"
 
 #define DEFAULT_SCRIPT      ETCDIR "/" PACKAGE ".sh"
 
 #define STATEDIR            "/var"
-#define PIDFILE             STATEDIR "/run/" PACKAGE "-%s.pid"
+#define PIDFILE             ETCDIR "/dhcpc/" PACKAGE "-%s.pid"
 
-#define CONFIGDIR           STATEDIR "/lib/" PACKAGE
-#define INFOFILE            CONFIGDIR "/" PACKAGE "-%s.info"
+#define CONFIGDIR           ETCDIR "/dhcpc"
+#define INFOFILE            ETCDIR "/dhcpc/" PACKAGE "-%s.info"
 
 #endif
