diff -urN dhcpcd-3.0.2.orig/dhcpcd.sh dhcpcd-3.0.2/dhcpcd.sh
--- dhcpcd-3.0.2.orig/dhcpcd.sh	2006-12-03 20:52:22.000000000 +0000
+++ dhcpcd-3.0.2/dhcpcd.sh	2006-12-03 20:52:40.000000000 +0000
@@ -25,20 +25,20 @@
 
 case "${state}" in
     up)
-    logger -s -p local0.info -t dhcpcd.sh \
+    logger -p local0.info -t dhcpcd.sh \
     "interface ${INTERFACE} has been configured with old IP=${IPADDR}"
     # Put your code here for when the interface has been brought up with an
     # old IP address here
     ;;
 
     new)
-    logger -s -p local0.info -t dhcpcd.sh \
+    logger -p local0.info -t dhcpcd.sh \
     "interface ${INTERFACE} has been configured with new IP=${IPADDR}"
     # Put your code here for when the interface has been brought up with a
     # new IP address
     ;;
 
-    down) logger -s -p local0.info -t dhcpcd.sh \
+    down) logger -p local0.info -t dhcpcd.sh \
     "interface ${INTERFACE} has been brought down"
     # Put your code here for the when the interface has been shut down
     ;;
