diff -Naur dhcpcd-3.1.8.orig/config.h dhcpcd-3.1.8/config.h
--- dhcpcd-3.1.8.orig/config.h	2007-12-19 12:02:56.000000000 +0100
+++ dhcpcd-3.1.8/config.h	2007-12-19 12:03:36.000000000 +0100
@@ -72,7 +72,7 @@
 # define SERVICE             "OPENRC"
 # define NISSERVICE          ETCDIR "/init.d/ypbind"
 # define NISRESTARTARGS      "--nodeps", "--quiet", "conditionalrestart"
-# define NTPSERVICE          ETCDIR "/init.d/ntpd"
+# define NTPSERVICE          ETCDIR "/rc.d/ntpd"
 # define NTPRESTARTARGS      "--nodeps", "--quiet", "conditionalrestart"
 #elif ENABLE_BSDRC
 # define SERVICE             "BSDRC"
@@ -96,7 +96,7 @@
 # define SERVICE             "SYSV"
 # define NISSERVICE          ETCDIR "/init.d/ypbind"
 # define NISRESTARTARGS      "restart"
-# define NTPSERVICE          ETCDIR "/init.d/ntpd"
+# define NTPSERVICE          ETCDIR "/rc.d/ntpd"
 # define NTPRESTARTARGS      "restart"
 #else
 # undef ENABLE_NIS
