--- http2-5.0.0/http2.cabal	2001-09-09 03:46:40.000000000 +0200
+++ http2-5.0.0.new/http2.cabal	2023-12-13 11:46:59.205344073 +0100
@@ -117,7 +117,7 @@
         base >=4.9 && <5,
         array >= 0.5 && < 0.6,
         async >= 2.2 && < 2.3,
-        bytestring >= 0.10 && < 0.12,
+        bytestring >= 0.10 && < 0.13,
         containers >= 0.6 && < 0.7,
         stm >= 2.5 && < 2.6,
         case-insensitive >= 1.2 && < 1.3,
