diff -Naur knetworkmanager-0.9/NetworkManager-kde4.conf knetworkmanager-0.9-p/NetworkManager-kde4.conf
--- knetworkmanager-0.9/NetworkManager-kde4.conf	2009-11-02 16:24:49.000000000 +0100
+++ knetworkmanager-0.9-p/NetworkManager-kde4.conf	2010-04-06 11:44:57.826746155 +0200
@@ -23,7 +23,7 @@
                 <allow send_destination="org.freedesktop.NetworkManagerUserSettings"
                        send_interface="org.freedesktop.NetworkManagerSettings.Connection.Secrets"/>
         </policy>
-        <policy at_console="true">
+        <policy group="netdev">
                 <allow own="org.freedesktop.NetworkManagerUserSettings"/>
 
                 <allow send_destination="org.freedesktop.NetworkManagerUserSettings"
