diff -Naur metacity-2.8.8.orig/src/metacity.schemas.in metacity-2.8.8/src/metacity.schemas.in
--- metacity-2.8.8.orig/src/metacity.schemas.in	2004-12-06 19:31:16.000000000 +0100
+++ metacity-2.8.8/src/metacity.schemas.in	2005-01-27 15:57:15.339924408 +0100
@@ -145,7 +145,7 @@
       <applyto>/apps/metacity/general/titlebar_font</applyto>   
       <owner>metacity</owner>
       <type>string</type>
-      <default>Sans Bold 10</default>
+      <default>Luxi Sans Bold 10</default>
       <locale name="C">
          <short>Window title font</short>
          <long>
