diff -Naur metacity-2.10.0.orig/src/metacity.schemas.in metacity-2.10.0/src/metacity.schemas.in
--- metacity-2.10.0.orig/src/metacity.schemas.in	2005-02-24 23:02:44.000000000 +0100
+++ metacity-2.10.0/src/metacity.schemas.in	2005-03-17 14:26:11.379907280 +0100
@@ -114,7 +114,7 @@
       <applyto>/apps/metacity/general/theme</applyto>
       <owner>metacity</owner>
       <type>string</type>
-      <default>Simple</default>
+      <default>Clearlooks</default>
       <locale name="C">
          <short>Current theme</short>
          <long>
