diff -urN metacity-2.20.0.orig/src/metacity.schemas metacity-2.20.0/src/metacity.schemas
--- metacity-2.20.0.orig/src/metacity.schemas	2007-10-04 22:28:50.000000000 +0100
+++ metacity-2.20.0/src/metacity.schemas	2007-10-04 22:29:35.000000000 +0100
@@ -212,7 +212,7 @@
       <applyto>/apps/metacity/general/theme</applyto>
       <owner>metacity</owner>
       <type>string</type>
-      <default>Clearlooks</default>
+      <default>Inverted</default>
       <locale name="C">
          <short>Current theme</short>
          <long>
@@ -243,7 +243,7 @@
       <applyto>/apps/metacity/general/titlebar_font</applyto>   
       <owner>metacity</owner>
       <type>string</type>
-      <default>Sans Bold 10</default>
+      <default>DejaVu Sans Bold 13</default>
       <locale name="C">
         <short>Window title font</short>
         <long>
diff -urN metacity-2.20.0.orig/src/metacity.schemas.in metacity-2.20.0/src/metacity.schemas.in
--- metacity-2.20.0.orig/src/metacity.schemas.in	2007-10-04 22:28:50.000000000 +0100
+++ metacity-2.20.0/src/metacity.schemas.in	2007-10-04 22:29:28.000000000 +0100
@@ -212,7 +212,7 @@
       <applyto>/apps/metacity/general/theme</applyto>
       <owner>metacity</owner>
       <type>string</type>
-      <default>Clearlooks</default>
+      <default>Inverted</default>
       <locale name="C">
          <short>Current theme</short>
          <long>
@@ -243,7 +243,7 @@
       <applyto>/apps/metacity/general/titlebar_font</applyto>   
       <owner>metacity</owner>
       <type>string</type>
-      <default>Sans Bold 10</default>
+      <default>DejaVu Sans Bold 13</default>
       <locale name="C">
         <short>Window title font</short>
         <long>
--- metacity-2.24.0old/src/core/prefs.c	2008-08-18 14:29:06.000000000 +0200
+++ metacity-2.24.0/src/core/prefs.c	2008-09-29 21:54:35.000000000 +0200
@@ -1038,8 +1038,8 @@
    * elsewhere that will do everything possible to fallback to an existing theme
    * if the one here does not exist.
    */
-  titlebar_font = pango_font_description_from_string ("Sans Bold 10");
-  current_theme = g_strdup ("Atlanta");
+  titlebar_font = pango_font_description_from_string ("DejaVu Sans Bold 13");
+  current_theme = g_strdup ("Inverted");
   
   init_button_layout();
 #endif /* HAVE_GCONF */
@@ -1365,7 +1365,7 @@
 {
   /* Fallback crackrock */
   if (string_value == NULL)
-    current_theme = g_strdup ("Atlanta");
+    current_theme = g_strdup ("Inverted");
   else
     current_theme = g_strdup (string_value);
 }

