diff -urN metacity-2.18.2.orig/src/metacity.schemas metacity-2.18.2/src/metacity.schemas
--- metacity-2.18.2.orig/src/metacity.schemas	2007-05-06 20:17:54.000000000 +0100
+++ metacity-2.18.2/src/metacity.schemas	2007-05-06 20:24:31.000000000 +0100
@@ -2678,7 +2678,7 @@
       <applyto>/apps/metacity/general/theme</applyto>
       <owner>metacity</owner>
       <type>string</type>
-      <default>Clearlooks</default>
+      <default>Frugalware</default>
       <locale name="C">
          <short>Current theme</short>
          <long>
@@ -3409,7 +3409,7 @@
       <applyto>/apps/metacity/general/titlebar_font</applyto>   
       <owner>metacity</owner>
       <type>string</type>
-      <default>Sans Bold 10</default>
+      <default>DejaVu Sans Bold 13</default>
       <locale name="C">
         <short>Window title font</short>
         <long>
diff -urN metacity-2.18.2.orig/src/metacity.schemas.in metacity-2.18.2/src/metacity.schemas.in
--- metacity-2.18.2.orig/src/metacity.schemas.in	2007-05-06 20:17:54.000000000 +0100
+++ metacity-2.18.2/src/metacity.schemas.in	2007-05-06 20:24:22.000000000 +0100
@@ -161,7 +161,7 @@
       <applyto>/apps/metacity/general/theme</applyto>
       <owner>metacity</owner>
       <type>string</type>
-      <default>Clearlooks</default>
+      <default>Frugalware</default>
       <locale name="C">
          <short>Current theme</short>
          <long>
@@ -192,7 +192,7 @@
       <applyto>/apps/metacity/general/titlebar_font</applyto>   
       <owner>metacity</owner>
       <type>string</type>
-      <default>Sans Bold 10</default>
+      <default>DejaVu Sans Bold 13</default>
       <locale name="C">
         <short>Window title font</short>
         <long>
diff -urN metacity-2.18.2.orig/src/prefs.c metacity-2.18.2/src/prefs.c
--- metacity-2.18.2.orig/src/prefs.c	2007-05-06 20:17:54.000000000 +0100
+++ metacity-2.18.2/src/prefs.c	2007-05-06 20:18:33.000000000 +0100
@@ -514,8 +514,8 @@
    * elsewhere that will do everything possible to fallback to an existing theme
    * if the one here does not exist.
    */
-  titlebar_font = pango_font_description_from_string ("Sans Bold 10");
-  current_theme = g_strdup ("Atlanta");
+  titlebar_font = pango_font_description_from_string ("DejaVu Sans Bold 13");
+  current_theme = g_strdup ("Frugalware");
 #endif /* HAVE_GCONF */
   
   /* Load keybindings prefs */
@@ -1165,7 +1165,7 @@
   if (current_theme == NULL)
     {
       /* Fallback crackrock */
-      current_theme = g_strdup ("Atlanta");
+      current_theme = g_strdup ("Frugalware");
       changed = TRUE;
     }
   
