diff -urN metacity-2.16.1.orig/src/metacity.schemas metacity-2.16.1/src/metacity.schemas
--- metacity-2.16.1.orig/src/metacity.schemas	2006-09-17 15:51:31.000000000 +0100
+++ metacity-2.16.1/src/metacity.schemas	2006-09-17 15:57:02.000000000 +0100
@@ -2563,7 +2563,7 @@
       <applyto>/apps/metacity/general/theme</applyto>
       <owner>metacity</owner>
       <type>string</type>
-      <default>Clearlooks</default>
+      <default>Simplebox-custom-1.5</default>
       <locale name="C">
          <short>Current theme</short>
          <long>
@@ -3274,7 +3274,7 @@
       <applyto>/apps/metacity/general/titlebar_font</applyto>   
       <owner>metacity</owner>
       <type>string</type>
-      <default>Sans Bold 10</default>
+      <default>DejaVu Sans Bold 13</default>
       <locale name="C">
         <short>Window title font</short>
         <long>
diff -urN metacity-2.16.1.orig/src/metacity.schemas.in metacity-2.16.1/src/metacity.schemas.in
--- metacity-2.16.1.orig/src/metacity.schemas.in	2006-09-17 15:51:31.000000000 +0100
+++ metacity-2.16.1/src/metacity.schemas.in	2006-09-17 15:54:10.000000000 +0100
@@ -161,7 +161,7 @@
       <applyto>/apps/metacity/general/theme</applyto>
       <owner>metacity</owner>
       <type>string</type>
-      <default>Clearlooks</default>
+      <default>Simplebox-custom-1.5</default>
       <locale name="C">
          <short>Current theme</short>
          <long>
@@ -192,7 +192,7 @@
       <applyto>/apps/metacity/general/titlebar_font</applyto>   
       <owner>metacity</owner>
       <type>string</type>
-      <default>Sans Bold 10</default>
+      <default>DejaVu Sans Bold 13</default>
       <locale name="C">
         <short>Window title font</short>
         <long>
diff -urN metacity-2.16.1.orig/src/prefs.c metacity-2.16.1/src/prefs.c
--- metacity-2.16.1.orig/src/prefs.c	2006-09-17 15:51:30.000000000 +0100
+++ metacity-2.16.1/src/prefs.c	2006-09-17 15:56:35.000000000 +0100
@@ -512,8 +512,8 @@
    * elsewhere that will do everything possible to fallback to an existing theme
    * if the one here does not exist.
    */
-  titlebar_font = pango_font_description_from_string ("Sans Bold 10");
-  current_theme = g_strdup ("Atlanta");
+  titlebar_font = pango_font_description_from_string ("DejaVu Sans Bold 13");
+  current_theme = g_strdup ("Simplebox-custom-1.5");
 #endif /* HAVE_GCONF */
   
   /* Load keybindings prefs */
@@ -1163,7 +1163,7 @@
   if (current_theme == NULL)
     {
       /* Fallback crackrock */
-      current_theme = g_strdup ("Atlanta");
+      current_theme = g_strdup ("Simplebox-custom-1.5");
       changed = TRUE;
     }
   
