diff -urN metacity-2.15.21.orig/src/metacity.schemas metacity-2.15.21/src/metacity.schemas
--- metacity-2.15.21.orig/src/metacity.schemas	2006-08-07 18:31:08.000000000 +0100
+++ metacity-2.15.21/src/metacity.schemas	2006-08-18 13:40:17.000000000 +0100
@@ -2483,7 +2483,7 @@
       <applyto>/apps/metacity/general/theme</applyto>
       <owner>metacity</owner>
       <type>string</type>
-      <default>Clearlooks</default>
+      <default>Simplebox-custom-1.5</default>
       <locale name="C">
          <short>Current theme</short>
          <long>
@@ -3184,7 +3184,7 @@
       <applyto>/apps/metacity/general/titlebar_font</applyto>   
       <owner>metacity</owner>
       <type>string</type>
-      <default>Sans Bold 10</default>
+      <default>Luxi Sans Bold 10</default>
       <locale name="C">
         <short>Window title font</short>
         <long>
diff -urN metacity-2.15.21.orig/src/metacity.schemas.in metacity-2.15.21/src/metacity.schemas.in
--- metacity-2.15.21.orig/src/metacity.schemas.in	2006-04-15 03:34:43.000000000 +0100
+++ metacity-2.15.21/src/metacity.schemas.in	2006-08-18 13:40:01.000000000 +0100
@@ -161,7 +161,7 @@
       <applyto>/apps/metacity/general/theme</applyto>
       <owner>metacity</owner>
       <type>string</type>
-      <default>Clearlooks</default>
+      <default>Simplebox-custom-1.5</default>
       <locale name="C">
          <short>Current theme</short>
          <long>
@@ -192,7 +192,7 @@
       <applyto>/apps/metacity/general/titlebar_font</applyto>   
       <owner>metacity</owner>
       <type>string</type>
-      <default>Sans Bold 10</default>
+      <default>Luxi Sans Bold 10</default>
       <locale name="C">
         <short>Window title font</short>
         <long>
