diff --color -Naur gimp-2.10.38/plug-ins/file-tiff/file-tiff-load.c gimp-2.10.38.new/plug-ins/file-tiff/file-tiff-load.c
--- gimp-2.10.38/plug-ins/file-tiff/file-tiff-load.c	2024-05-03 02:33:35.000000000 +0200
+++ gimp-2.10.38.new/plug-ins/file-tiff/file-tiff-load.c	2024-07-29 15:31:54.755800962 +0200
@@ -1301,8 +1301,8 @@
 
       /* any resolution info in the file? */
       {
-        gfloat   xres = 72.0;
-        gfloat   yres = 72.0;
+        gdouble   xres = 72.0;
+        gdouble   yres = 72.0;
         gushort  read_unit;
         GimpUnit unit = GIMP_UNIT_PIXEL; /* invalid unit */

diff --git a/plug-ins/metadata/metadata-editor.c b/plug-ins/metadata/metadata-editor.c
index a3992e40ddc..5bf6ef709f1 100644
--- a/plug-ins/metadata/metadata-editor.c
+++ b/plug-ins/metadata/metadata-editor.c
@@ -2846,7 +2846,7 @@ metadata_dialog_editor_set_metadata (GExiv2Metadata  *metadata,
                             }
                           else
                             {
-                              if (! g_strv_contains (values, equiv_values[evi]))
+                              if (! g_strv_contains ((const gchar * const *) values, equiv_values[evi]))
                                 {
                                   gchar *tmpvalue;
 
 
