diff -Naur compiz-0.8.8/kde/window-decorator-kde4/window.cpp compiz-0.8.8/kde/window-decorator-kde4/window.cpp
--- compiz-0.8.8/kde/window-decorator-kde4/window.cpp	2010-05-21 13:18:14.000000000 +0200
+++ compiz-0.8.8/kde/window-decorator-kde4/window.cpp	2012-05-06 11:44:01.819887105 +0200
@@ -21,6 +21,8 @@
  * Author: David Reveman <davidr@novell.com>
  */
 
+#include <unistd.h>
+
 #include "window.h"
 #include "decorator.h"
 #include "options.h"
@@ -986,10 +988,10 @@
 {
     Atom    atom = Atoms::compizWindowBlurDecor;
     QRegion topQRegion, bottomQRegion, leftQRegion, rightQRegion;
-    Region  topRegion = NULL;
-    Region  bottomRegion = NULL;
-    Region  leftRegion = NULL;
-    Region  rightRegion = NULL;
+    ::Region  topRegion = NULL;
+    ::Region  bottomRegion = NULL;
+    ::Region  leftRegion = NULL;
+    ::Region  rightRegion = NULL;
     int     size = 0;
     int     w, h;
 