--- http-media-0.8.0.0/http-media.cabal	2019-04-16 00:57:56.000000000 +0200
+++ http-media-0.8.0.0.new/http-media.cabal	2023-03-20 11:38:32.917644415 +0100
@@ -71,8 +71,8 @@
     Network.HTTP.Media.Utils
 
   build-depends:
-    base             >= 4.7  && < 4.13,
-    bytestring       >= 0.10 && < 0.11,
+    base             >= 4.7  && < 4.19,
+    bytestring       >= 0.10 && < 0.12,
     case-insensitive >= 1.0  && < 1.3,
     containers       >= 0.5  && < 0.7,
     utf8-string      >= 0.3  && < 1.1
@@ -122,8 +122,8 @@
     Network.HTTP.Media.Utils
 
   build-depends:
-    base                       >= 4.7  && < 4.13,
-    bytestring                 >= 0.10 && < 0.11,
+    base                       >= 4.7  && < 4.18,
+    bytestring                 >= 0.10 && < 0.12,
     case-insensitive           >= 1.0  && < 1.3,
     containers                 >= 0.5  && < 0.7,
     utf8-string                >= 0.3  && < 1.1,
