From a6b075926b3d22aa504a0aef842c275105853b2d Mon Sep 17 00:00:00 2001
From: 5kft <5kft@users.noreply.github.com>
Date: Thu, 30 Nov 2023 16:10:33 -0800
Subject: [PATCH 1/2] changes for kernel v6.7

---
 os_dep/linux/ioctl_cfg80211.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/os_dep/linux/ioctl_cfg80211.c b/os_dep/linux/ioctl_cfg80211.c
index e2b8d67..373a77e 100644
--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -4965,14 +4965,22 @@ static int cfg80211_rtw_start_ap(struct wiphy *wiphy, struct net_device *ndev,
 }
 
 static int cfg80211_rtw_change_beacon(struct wiphy *wiphy, struct net_device *ndev,
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0))
+		struct cfg80211_ap_update *info)
+#else
 		struct cfg80211_beacon_data *info)
+#endif
 {
 	int ret = 0;
 	_adapter *adapter = (_adapter *)rtw_netdev_priv(ndev);
 
 	RTW_INFO(FUNC_NDEV_FMT"\n", FUNC_NDEV_ARG(ndev));
 
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 7, 0))
+	ret = rtw_add_beacon(adapter, info->beacon.head, info->beacon.head_len, info->beacon.tail, info->beacon.tail_len);
+#else
 	ret = rtw_add_beacon(adapter, info->head, info->head_len, info->tail, info->tail_len);
+#endif
 
 	return ret;
 }

From e7363dcf435b574043cd918a4abcbcfafc48499a Mon Sep 17 00:00:00 2001
From: 5kft <5kft@users.noreply.github.com>
Date: Thu, 30 Nov 2023 16:21:45 -0800
Subject: [PATCH 2/2] fix build warning for kernel v6.4+

---
 include/osdep_service.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/osdep_service.h b/include/osdep_service.h
index 35988cc..40cf213 100644
--- a/include/osdep_service.h
+++ b/include/osdep_service.h
@@ -408,7 +408,11 @@ static __inline void thread_enter(char *name)
 	printf("%s", "RTKTHREAD_enter");
 #endif
 }
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 4, 0))
+void __noreturn thread_exit(_completion *comp);
+#else
 void thread_exit(_completion *comp);
+#endif
 void _rtw_init_completion(_completion *comp);
 void _rtw_wait_for_comp_timeout(_completion *comp);
 void _rtw_wait_for_comp(_completion *comp);
