--- spring/rts/System/Sound/OpenAL/Sound.h	2019-12-02 14:02:34.303825125 +0100
+++ spring.new/rts/System/Sound/OpenAL/Sound.h	2019-12-02 14:03:37.641252911 +0100
@@ -18,10 +18,8 @@
 class CSoundSource;
 class SoundBuffer;
 class SoundItem;
-struct ALCdevice_struct;
-typedef struct ALCdevice_struct ALCdevice;
-struct ALCcontext_struct;
-typedef struct ALCcontext_struct ALCcontext;
+class ALCcontext;
+class ALCdevice;
 
 /// Default sound system implementation (OpenAL)
 class CSound : public ISound
