diff -aur a/AI/Skirmish/E323AI/AAStar.h b/AI/Skirmish/E323AI/AAStar.h
--- a/AI/Skirmish/E323AI/AAStar.h	2011-01-07 18:45:27.000000000 -0600
+++ b/AI/Skirmish/E323AI/AAStar.h	2011-09-05 20:12:27.684082517 -0500
@@ -1,6 +1,7 @@
 #ifndef E323ASTAR_H
 #define E323ASTAR_H
 
+#include <cstddef>
 #include <queue>
 #include <vector>
 #include <list>
diff -aur a/rts/Rendering/GLContext.h b/rts/Rendering/GLContext.h
--- a/rts/Rendering/GLContext.h	2011-01-07 18:45:23.000000000 -0600
+++ b/rts/Rendering/GLContext.h	2011-09-05 20:24:56.633049898 -0500
@@ -23,6 +23,7 @@
 //            (use "data" for the pointer to the instance).
 
 // TODO: add event driven system (like alttab, resolution changed, dualscreen, ...)
+#include <cstddef>
 
 namespace GLContext
 {
diff -aur a/rts/System/MemPool.h b/rts/System/MemPool.h
--- a/rts/System/MemPool.h	2011-01-07 18:45:23.000000000 -0600
+++ b/rts/System/MemPool.h	2011-09-05 20:29:02.515044723 -0500
@@ -3,6 +3,7 @@
 #ifndef _MEM_POOL_H_
 #define _MEM_POOL_H_
 
+#include <cstddef>
 #include <new>
 
 const size_t MAX_MEM_SIZE=200;
