--- spring_98.0/rts/System/creg/TypeDeduction.h	2014-10-07 22:09:51.000000000 +0200
+++ spring_98.0.new/rts/System/creg/TypeDeduction.h	2015-05-13 15:43:17.036817586 +0200
@@ -42,6 +42,7 @@
 CREG_SUPPORT_BASIC_TYPE(unsigned char, crUChar)
 CREG_SUPPORT_BASIC_TYPE(boost::int64_t, crInt64)
 CREG_SUPPORT_BASIC_TYPE(boost::uint64_t, crUInt64)
+CREG_SUPPORT_BASIC_TYPE(long unsigned int, crUInt64)
 CREG_SUPPORT_BASIC_TYPE(float, crFloat)
 CREG_SUPPORT_BASIC_TYPE(double, crDouble)
 CREG_SUPPORT_BASIC_TYPE(bool, crBool)
