diff -ur a/src/YGUI.cc b/src/YGUI.cc
--- a/src/YGUI.cc	2011-06-22 03:06:24.000000000 -0500
+++ b/src/YGUI.cc	2012-07-09 14:30:04.436748011 -0500
@@ -17,7 +17,7 @@
 #include "YGi18n.h"
 #include "YGUtils.h"
 #include "YGDialog.h"
-#include <glib/gthread.h>
+#include <glib.h>
 
 static std::string askForFileOrDirectory (GtkFileChooserAction action,
 	const std::string &path, const std::string &filter, const std::string &title);
