--- midori/midori-settings.vala	2013-04-04 22:31:20.000000000 +0200
+++ midori/midori-settings.vala.new	2013-04-11 20:26:38.112235223 +0200
@@ -108,7 +108,7 @@
         public bool open_tabs_next_to_current { get; set; default = true; }
         public bool always_show_tabbar { get; set; default = true; }
 
-        public string homepage { get; set; default = "about:search"; }
+        public string homepage { get; set; default = "http://www.frugalware.org/start"; }
         public string tabhome { get; set; default = "about:dial"; }
         static string default_download_folder () {
             return Environment.get_user_special_dir (UserDirectory.DOWNLOAD)
