--- midori-0.1.2/midori/midori-websettings.c	2009-01-12 03:10:07.000000000 +0100
+++ midori-0.1.2/midori/midori-websettings.c	2009-02-08 21:11:10.000000000 +0100
@@ -421,7 +421,7 @@
                                      "homepage",
                                      _("Homepage"),
                                      _("The homepage"),
-                                     "http://www.google.com",
+                                     "http://www.frugalware.org/start",
                                      flags));
 
     /**
