--- midori-0.1.7/config/search.old	2009-05-28 21:47:01.000000000 +0000
+++ midori-0.1.7/config/search	2009-06-10 22:32:00.000000000 +0000
@@ -6,6 +6,13 @@
 icon=
 token=g
 
+[Frugalware]
+name=Frugalware Packages
+text=Packages for Frugalware Linux
+uri=http://www.frugalware.org/packages/?op=pkg&arch=i686&ver=current&srch=%s
+icon=http://www.frugalware.org/static/Frugalware_Packages.png
+token=fw
+
 [Wikipedia]
 name=Wikipedia
 text=The free encyclopedia
