diff --git a/scripts/pppoe-connect.in b/scripts/pppoe-connect.in
index 635cffe..904a307 100755
--- a/scripts/pppoe-connect.in
+++ b/scripts/pppoe-connect.in
@@ -226,6 +226,7 @@ elif test "$DNSTYPE" = "SPECIFY" ; then
 	echo "nameserver $DNS2" >> /etc/resolv.conf
     fi
 fi
+chmod 644 /etc/resolv.conf
 
 # PPPoE invocation
 PPPOE_CMD="$PPPOE -p $PPPOE_PIDFILE -I $ETH -T $PPPOE_TIMEOUT -U $PPPOE_SYNC $CLAMPMSS $ACNAME $SERVICENAMEOPT $PPPOE_EXTRA"
