diff -Naur wepdecrypt-0.7.orig/Makefile wepdecrypt-0.7/Makefile
--- wepdecrypt-0.7.orig/Makefile	2005-08-20 02:24:09.000000000 +0200
+++ wepdecrypt-0.7/Makefile	2006-01-19 15:03:18.000000000 +0100
@@ -1,5 +1,6 @@
 include config.mak
 
+DESTDIR=
 INSTDIR=/usr/bin
 
 all:
@@ -15,25 +16,28 @@
 	rm config.log config.mak config.status
 
 install:
-	./install-sh -m 755 src/wepdecrypt $(INSTDIR)
+	mkdir -p $(DESTDIR)$(INSTDIR)
+	./install-sh -m 755 src/wepdecrypt $(DESTDIR)$(INSTDIR)
 
 	@if [ $(USEGUI) = YES ]; then \
-	        ./install-sh -m 755 src/gwepdecrypt $(INSTDIR); echo "./install-sh -m 755 src/gwepdecrypt $(INSTDIR)"; fi
+	        ./install-sh -m 755 src/gwepdecrypt $(DESTDIR)$(INSTDIR); echo "./install-sh -m 755 src/gwepdecrypt $(DESTDIR)$(INSTDIR)"; fi
 
-	./install-sh -m 755 run/wepdecrypt_word $(INSTDIR)
-	./install-sh -m 755 run/wepdecrypt_inc $(INSTDIR)
+	./install-sh -m 755 run/wepdecrypt_word $(DESTDIR)$(INSTDIR)
+	./install-sh -m 755 run/wepdecrypt_inc $(DESTDIR)$(INSTDIR)
 
+	mkdir -p $(DESTDIR)/etc
 	@if test -f /etc/wepdecrypt.conf ; then \
 		echo "/etc/wepdecrypt.conf already exists..."; \
 	else echo "creating /etc/wepdecrypt.conf..."; \
-		./install-sh -m 644 conf/wepdecrypt.conf /etc ; fi
+		./install-sh -m 644 conf/wepdecrypt.conf $(DESTDIR)/etc ; fi
 
-	./install-sh -m 644 doc/wepdecrypt.1.gz /usr/share/man/man1/
-	mkdir /usr/share/doc/wepdecrypt-0.7
-	./install-sh -m 644 doc/manual.html /usr/share/doc/wepdecrypt-0.7/
-	./install-sh -m 644 doc/manual.txt /usr/share/doc/wepdecrypt-0.7/
-	./install-sh -m 644 README /usr/share/doc/wepdecrypt-0.7/
-	./install-sh -m 644 CHANGELOG /usr/share/doc/wepdecrypt-0.7/
+	mkdir -p $(DESTDIR)/usr/share/man/man1/
+	./install-sh -m 644 doc/wepdecrypt.1.gz $(DESTDIR)/usr/share/man/man1/
+	mkdir -p $(DESTDIR)/usr/share/doc/wepdecrypt-0.7
+	./install-sh -m 644 doc/manual.html $(DESTDIR)/usr/share/doc/wepdecrypt-0.7/
+	./install-sh -m 644 doc/manual.txt $(DESTDIR)/usr/share/doc/wepdecrypt-0.7/
+	./install-sh -m 644 README $(DESTDIR)/usr/share/doc/wepdecrypt-0.7/
+	./install-sh -m 644 CHANGELOG $(DESTDIR)/usr/share/doc/wepdecrypt-0.7/
 
 uninstall:
 	@if test -f /usr/bin/gwepdecrypt ; then                 rm /usr/bin/gwepdecrypt; echo "rm usr/bin/gwepdecrypt"; fi
