--- julia/src/julia_internal.h.0	2018-09-04 16:31:16.074841286 +0200
+++ julia/src/julia_internal.h	2018-09-04 16:31:45.377840722 +0200
@@ -623,12 +623,9 @@
 #  include <libunwind.h>
 typedef unw_context_t bt_context_t;
 typedef unw_cursor_t bt_cursor_t;
-#  if (!defined(SYSTEM_LIBUNWIND) || UNW_VERSION_MAJOR > 1 ||   \
-       (UNW_VERSION_MAJOR == 1 && UNW_VERSION_MINOR > 1))
 // Enable our memory manager only for libunwind with our patch or
 // on a newer release
 #    define JL_UNW_HAS_FORMAT_IP 1
-#  endif
 #else
 // Unwinding is disabled
 typedef int bt_context_t;
