diff -Naur libsynaptics-0.14.4c.orig/Makefile libsynaptics-0.14.4c/Makefile
--- libsynaptics-0.14.4c.orig/Makefile	2006-01-02 09:57:06.000000000 +0100
+++ libsynaptics-0.14.4c/Makefile	2006-02-04 13:57:39.000000000 +0100
@@ -1,5 +1,6 @@
 #LIBVER=$( cat version.h | grep LIBSYNAPTICS_VERSIONSTR | cut -d ' ' -f 3 | sed 's/\"//g' )
 LIBVER=0.14.4c
+DESTDIR=
 
 all: libsynaptics.so Makefile
 	g++ -I. syntest.cpp -fPIC -L. -lsynaptics -o syntest
@@ -18,20 +19,21 @@
 	rm .systeminfo; rm .params_1; rm .params_2
 
 install: Makefile libsynaptics.so
-	cp libsynaptics.*so /usr/lib 
-	mkdir -p /usr/include/synaptics
+	mkdir -p $(DESTDIR)/usr/lib
+	cp libsynaptics.*so $(DESTDIR)/usr/lib 
+	mkdir -p $(DESTDIR)/usr/include/synaptics
 	
-	cp synaptics.h /usr/include/synaptics
-	cp pad.h /usr/include/synaptics
-	cp Shm_*_*_* /usr/include/synaptics
-	cp syndebug.h /usr/include/synaptics
-	cp synparams.h /usr/include/synaptics
-	cp synshm.h /usr/include/synaptics
-	cp version.h /usr/include/synaptics
+	cp synaptics.h $(DESTDIR)/usr/include/synaptics
+	cp pad.h $(DESTDIR)/usr/include/synaptics
+	cp Shm_*_*_* $(DESTDIR)/usr/include/synaptics
+	cp syndebug.h $(DESTDIR)/usr/include/synaptics
+	cp synparams.h $(DESTDIR)/usr/include/synaptics
+	cp synshm.h $(DESTDIR)/usr/include/synaptics
+	cp version.h $(DESTDIR)/usr/include/synaptics
 
 uninstall:
-	rm /usr/lib/libsynaptics.*so
-	rm -rf /usr/include/synaptics
+	rm $(DESTDIR)/usr/lib/libsynaptics.*so
+	rm -rf $(DESTDIR)/usr/include/synaptics
 
 test: all
 	sh syntest.sh
