--- scientific-0.3.8.0/scientific.cabal	2001-09-09 03:46:40.000000000 +0200
+++ scientific-0.3.8.0.new/scientific.cabal	2024-12-27 13:20:45.876069960 +0100
@@ -1,5 +1,6 @@
 name:               scientific
 version:            0.3.8.0
+x-revision:         1
 synopsis:           Numbers represented using scientific notation
 description:
   "Data.Scientific" provides the number type 'Scientific'. Scientific numbers are
@@ -78,15 +79,15 @@
 
   ghc-options:      -Wall
   build-depends:
-      base                >=4.5      && <4.21
+      base                >=4.5      && <4.22
     , binary              >=0.8.6.0  && <0.9
     , bytestring          >=0.10.8.2 && <0.13
     , containers          >=0.6.0.1  && <0.8
     , deepseq             >=1.4.4.0  && <1.6
-    , hashable            >=1.4.4.0  && <1.5
+    , hashable            >=1.4.4.0  && <1.6
     , integer-logarithms  >=1.0.3.1  && <1.1
     , primitive           >=0.9.0.0  && <0.10
-    , template-haskell    >=2.14.0.0 && <2.23
+    , template-haskell    >=2.14.0.0 && <2.24
     , text                >=1.2.3.0  && <1.3  || >=2.0 && <2.2
 
   if impl(ghc >=9.0)
