diff -Naur gettext-0.19.8.1/gettext-tools/gnulib-lib/libcroco/cr-statement.c gettext-0.19.8.1-p/gettext-tools/gnulib-lib/libcroco/cr-statement.c
--- gettext-0.19.8.1/gettext-tools/gnulib-lib/libcroco/cr-statement.c	2019-03-11 20:40:30.000000000 +0100
+++ gettext-0.19.8.1-p/gettext-tools/gnulib-lib/libcroco/cr-statement.c	2019-05-20 21:39:53.879323076 +0200
@@ -2607,7 +2607,7 @@
         g_return_if_fail (a_fp && a_this);
         str = cr_statement_ruleset_to_string (a_this, a_indent);
         if (str) {
-                fprintf (a_fp, str);
+                fprintf (a_fp, "%s", str);
                 g_free (str);
                 str = NULL;
         }
@@ -2658,7 +2658,7 @@
         str = cr_statement_charset_to_string (a_this,
                                               a_indent) ;
         if (str) {
-                fprintf (a_fp, str) ;
+                fprintf (a_fp, "%s", str) ;
                 g_free (str) ;
                 str = NULL ;
         }
@@ -2685,7 +2685,7 @@
 
         str = cr_statement_at_page_rule_to_string (a_this, a_indent) ;
         if (str) {
-                fprintf (a_fp, str);
+                fprintf (a_fp, "%s", str);
                 g_free (str) ;
                 str = NULL ; 
         }
@@ -2711,7 +2711,7 @@
 
         str = cr_statement_media_rule_to_string (a_this, a_indent) ;
         if (str) {
-                fprintf (a_fp, str) ;
+                fprintf (a_fp, "%s", str) ;
                 g_free (str) ;
                 str = NULL ;
         }
@@ -2737,7 +2737,7 @@
 
         str = cr_statement_import_rule_to_string (a_this, a_indent) ;
         if (str) {
-                fprintf (a_fp, str) ;
+                fprintf (a_fp, "%s", str) ;
                 g_free (str) ;
                 str = NULL ;
         }
