--- hslua-2.2.1/hslua.cabal	2001-09-09 03:46:40.000000000 +0200
+++ hslua-2.2.1.new/hslua.cabal	2023-02-22 14:30:13.972074661 +0100
@@ -44,7 +44,7 @@
                      , bytestring              >= 0.10.2 && < 0.12
                      , containers              >= 0.5.9  && < 0.7
                      , exceptions              >= 0.8    && < 0.11
-                     , hslua-aeson             >= 2.2    && < 2.3
+                     , hslua-aeson             >= 2.2    && < 2.4
                      , hslua-core              >= 2.2.1  && < 2.3
                      , hslua-classes           >= 2.2    && < 2.3
                      , hslua-marshalling       >= 2.2.1  && < 2.3
