diff -aur tetex-src-3.0/texk/dvipdfm/pngimage.c tetex-src-3.0-2/texk/dvipdfm/pngimage.c
--- tetex-src-3.0/texk/dvipdfm/pngimage.c	2002-12-25 11:51:16.000000000 -0600
+++ tetex-src-3.0-2/texk/dvipdfm/pngimage.c	2010-05-06 10:40:24.000000000 -0500
@@ -42,7 +42,7 @@
   rewind (png_file);
   if (fread (sigbytes, 1, sizeof(sigbytes), png_file) !=
       sizeof(sigbytes) ||
-      (!png_check_sig (sigbytes, sizeof(sigbytes))))
+      (png_sig_cmp (sigbytes, 0, sizeof(sigbytes))))
     return 0;
   else
     return 1;
diff -aur tetex-src-3.0/texk/dvipdfm/thumbnail.c tetex-src-3.0-2/texk/dvipdfm/thumbnail.c
--- tetex-src-3.0/texk/dvipdfm/thumbnail.c	2002-12-25 11:51:16.000000000 -0600
+++ tetex-src-3.0-2/texk/dvipdfm/thumbnail.c	2010-05-06 10:41:42.000000000 -0500
@@ -79,7 +79,7 @@
   }
   if (fread (sigbytes, 1, sizeof(sigbytes), thumb_file) !=
       sizeof(sigbytes) ||
-      (!png_check_sig (sigbytes, sizeof(sigbytes)))) {
+      (png_sig_cmp (sigbytes, 0, sizeof(sigbytes)))) {
     fprintf (stderr, "\nThumbnail not a png file! Skipping\n");
     return NULL;
   }
