diff -Naur shaderc-2020.5/CMakeLists.txt shaderc-2020.5.new/CMakeLists.txt
--- shaderc-2020.5/CMakeLists.txt	2021-02-19 22:08:08.000000000 +0100
+++ shaderc-2020.5.new/CMakeLists.txt	2021-04-16 16:44:18.242701251 +0200
@@ -112,7 +112,6 @@
 
 # Configure subdirectories.
 # We depend on these for later projects, so they should come first.
-add_subdirectory(third_party)
 
 add_subdirectory(libshaderc_util)
 add_subdirectory(libshaderc)
diff -Naur shaderc-2020.5/glslc/CMakeLists.txt shaderc-2020.5.new/glslc/CMakeLists.txt
--- shaderc-2020.5/glslc/CMakeLists.txt	2021-02-19 22:08:08.000000000 +0100
+++ shaderc-2020.5.new/glslc/CMakeLists.txt	2021-04-16 16:44:10.676657190 +0200
@@ -53,7 +53,6 @@
 target_include_directories(glslc_exe PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/.. ${spirv-tools_SOURCE_DIR}/include)
 set_target_properties(glslc_exe PROPERTIES OUTPUT_NAME glslc)
 target_link_libraries(glslc_exe PRIVATE glslc shaderc_util shaderc)
-add_dependencies(glslc_exe build-version)
 
 shaderc_add_tests(
   TEST_PREFIX glslc
