diff -ur a/configure.ac b/configure.ac
--- a/configure.ac	2009-10-18 06:50:39.000000000 -0500
+++ b/configure.ac	2013-01-01 10:07:41.988135537 -0600
@@ -22,7 +22,7 @@
 
 CFLAGS="$CFLAGS $DEP_CFLAGS $GNOME_KEYRING_CFLAGS"
 CXXFLAGS="$CFLAGS"
-LIBS="$DEP_LIBS $GNOME_KEYRING_LIBS"
+LIBS="$DEP_LIBS $GNOME_KEYRING_LIBS -lX11"
 
 #AC_PROG_LIBTOOL
 
diff -ur a/src/lsof.cpp b/src/lsof.cpp
--- a/src/lsof.cpp	2009-10-18 06:47:13.000000000 -0500
+++ b/src/lsof.cpp	2013-01-01 10:03:46.532144203 -0600
@@ -19,6 +19,7 @@
 #include <stdio.h>
 #include <string>
 #include <vector>
+#include <unistd.h>
 #include <assert.h>
 #include <cstring>
 #include "lsof.h"
