--- http-api-data-0.6/http-api-data.cabal	2001-09-09 03:46:40.000000000 +0200
+++ http-api-data-0.6.new/http-api-data.cabal	2024-04-15 19:21:14.353570886 +0200
@@ -50,7 +50,7 @@
 
     -- other-dependencies
     build-depends:
-                     cookie                >= 0.4.3    && < 0.5
+                     cookie                >= 0.4.3    && < 0.6
                    , hashable              >= 1.2.7.0  && < 1.5
                    , http-types            >= 0.12.3   && < 0.13
                    , text-iso8601          >= 0.1      && < 0.2
