--- http-api-data-0.6.1/http-api-data.cabal	2001-09-09 03:46:40.000000000 +0200
+++ http-api-data-0.6.1.new/http-api-data.cabal	2024-07-11 15:10:59.146667049 +0200
@@ -51,7 +51,7 @@
     -- other-dependencies
     build-depends:
                      cookie                >= 0.4.3    && < 0.6
-                   , hashable              >= 1.4.4.0  && < 1.5
+                   , hashable              >= 1.4.4.0  && < 1.6
                    , http-types            >= 0.12.4   && < 0.13
                    , text-iso8601          >= 0.1.1    && < 0.2
                    , tagged                >= 0.8.8    && < 0.9
