--- KVideoEncoder-0.07/Makefile	2006-06-06 09:37:28.000000000 +0200
+++ KVideoEncoder-0.07-p/Makefile	2006-06-06 09:36:16.000000000 +0200
@@ -4,11 +4,13 @@
 	@cd src && make
 
 install: kvideoencoder
-	cp src/kvideoencoder /usr/bin/
-	@echo /usr > /etc/kvideoencoder
-	@mkdir -p /usr/share/apps/kvideoencoder
-	cp src/*qm /usr/share/apps/kvideoencoder/
-	cp src/kvideoencoder.desktop /usr/share/applications/
+	@mkdir -p $(DESTDIR)/usr/bin
+	cp src/kvideoencoder $(DESTDIR)/usr/bin/
+	@echo /usr > $(DESTDIR)/etc/kvideoencoder
+	@mkdir -p $(DESTDIR)/usr/share/apps/kvideoencoder
+	cp src/*qm $(DESTDIR)/usr/share/apps/kvideoencoder/
+	@mkdir -p $(DESTDIR)/usr/share/applications
+	cp src/kvideoencoder.desktop $(DESTDIR)/usr/share/applications/
    
 uninstall:
 	rm /usr/bin/kvideoencoder
