diff -Naur htdig-3.2.0b6.orig/htsearch/Collection.h htdig-3.2.0b6/htsearch/Collection.h
--- htdig-3.2.0b6.orig/htsearch/Collection.h	2004-05-28 15:15:24.000000000 +0200
+++ htdig-3.2.0b6/htsearch/Collection.h	2006-05-04 03:02:08.000000000 +0200
@@ -36,9 +36,9 @@
                const char *docExcerpt);
     ~Collection();
 
-    void Collection::Open();
+    void Open();
 
-    void Collection::Close(); 
+    void Close(); 
 
     char *getWordFile() { return wordFile.get(); }
     DocumentRef         *getDocumentRef(int id);
