--- build.xml	2007-03-09 21:16:08.000000000 +0200
+++ build.xml.fixed	2007-04-08 14:32:56.000000000 +0300
@@ -144,9 +144,7 @@
          <src path="${java.src.dockbar}"/>
          <src path="${java.src.drag}"/>
          <src path="${java.src.perspective}"/>
-      </javac>
-      <javac classpathref="javac.classpath.demo" srcdir="${java.src.demo}" destdir="${bin.dir.demo}" debug="true" source="1.4"/>
-      <javac classpathref="javac.classpath.test" srcdir="${java.src.test}" destdir="${bin.dir.test}" debug="true" source="1.4"/>
+      </javac>
    </target>
 
 
@@ -279,22 +277,22 @@
    <target name="compile.unix.lib" depends="native.init,compile,resources" if="os.unixWithXlib" 
            description="Compile the shared library if you're on a UNIX of some sort">
 
-      <getprefixlibrarykey property="library.key"/>
 
       <property name="c.dir" value="${basedir}/src/c/core/org/flexdock/docking/drag/outline/xlib"/>
       <property name="so.dir" value="${build.dir}/bin/org/flexdock/docking/drag/outline/xlib"/>
 
 
       <!-- Cleanup existing .so or ensure the dir exists -->
-      <delete file="${so.dir}/libRubberBand${library.key}.so" failonerror="false"/>
+      <delete file="${so.dir}/libRubberBand.so" failonerror="false"/>
       <mkdir dir="${so.dir}"/>
 
-      <exec executable="gcc" failonerror="true">
+      <exec executable="${CC}" failonerror="true">
+         <arg value="-fPIC"/>
          <arg value="-lX11"/>
          <arg value="-L/usr/X11R6/lib"/>
          <arg value="-I/usr/X11R6/include"/>
          <arg value="-o"/>
-         <arg value="${so.dir}/libRubberBand${library.key}.so"/>
+         <arg value="${so.dir}/libRubberBand.so"/>
          <arg value="-shared"/>
          <arg value="${c.dir}/RubberBand.c"/>
          <arg line='${sdk.include.gcc.args}'/> <!-- potentially multiple include dirs -->
@@ -328,16 +326,6 @@
 
       <!-- Don't try to make a generic demo jar, 
            here shouldn't be native code in the demos. -->
-      <jar jarfile="${build.dir}/${dist.demo.jar.name}" basedir="${bin.dir.demo}" excludes="${jar.excludes}">
-         <!-- NOTE: don't use index="true", the classpath lookup of the other jar will not work.
-              See: http://forum.java.sun.com/thread.jspa?forumID=22&threadID=568620,
-              I believe this BUG is the cause: http://bugs.sun.com/bugdatabase/view_bug.do?bug_id=4490798 -->
-         <manifest>
-            <!-- matches core jar name -->
-            <attribute name="Class-Path" value="${dist.jar.name} lib/commons-logging-1.1.jar lib/jmf/lib/jmf.jar lib/looks-2.1.1.jar"/> 
-            <attribute name="Main-Class" value="org.flexdock.demos.AllDemos"/>
-         </manifest>
-      </jar>
 
       <mkdir dir="${build.dir}/lib"/>
       <copy todir="${build.dir}/lib">
