diff -Naur nano-2.0.1/doc/nanorc.sample.in nano-2.0.1-p/doc/nanorc.sample.in
--- nano-2.0.1/doc/nanorc.sample.in	2006-11-02 07:32:02.000000000 +0100
+++ nano-2.0.1-p/doc/nanorc.sample.in	2006-12-02 12:39:31.000000000 +0100
@@ -16,10 +16,10 @@
 ## ", ', ), >, ], and }.
 
 ## Use auto-indentation.
-# set autoindent
+#set autoindent
 
 ## Backup files to filename~.
-# set backup
+set backup
 
 ## The directory to put unique backup files in.
 # set backupdir ""
@@ -35,7 +35,7 @@
 ## punctuation, optionally followed by closing brackets, can end
 ## sentences.
 ##
-# set brackets ""')>]}"
+set brackets ""')>]}"
 
 ## Do case sensitive searches by default.
 # set casesensitive
@@ -45,7 +45,7 @@
 # set const
 
 ## Use cut to end of line by default.
-# set cut
+set cut
 
 ## Set the line length for wrapping text and justifying paragraphs.
 ## If fill is 0 or less, the line length will be the screen width less
@@ -60,7 +60,7 @@
 ## searches.  They cannot contain blank characters.  The former set must
 ## come before the latter set, and both must be in the same order.
 ##
-# set matchbrackets "(<[{)>]}"
+set matchbrackets "(<[{)>]}"
 
 ## Use the blank line below the titlebar as extra editing space.
 # set morespace
@@ -76,7 +76,7 @@
 ## separate buffer).  You must have configured with --enable-multibuffer
 ## for this to work.
 ##
-# set multibuffer
+set multibuffer
 
 ## Don't convert files from DOS/Mac format.
 # set noconvert
@@ -91,7 +91,7 @@
 # set nonewlines
 
 ## Don't wrap text at all.
-# set nowrap
+set nowrap
 
 ## Set operating directory.  nano will not read or write files outside
 ## this directory and its subdirectories.  Also, the current directory
@@ -108,7 +108,7 @@
 ## punctuation, optionally followed by closing brackets, can end
 ## sentences.
 ##
-# set punct "!.?"
+set punct "!.?"
 
 ## Do quick statusbar blanking.  Statusbar messages will disappear after
 ## 1 keystroke instead of 26.  Note that "const" overrides this.
@@ -129,7 +129,7 @@
 # set rebindkeypad
 
 ## Do extended regular expression searches by default.
-# set regexp
+set regexp
 
 ## Make the Home key smarter.  When Home is pressed anywhere but at the
 ## very beginning of non-whitespace characters on a line, the cursor
@@ -215,46 +215,50 @@
 
 
 ## Nanorc files
-# include "@PKGDATADIR@/nanorc.nanorc"
+include "@PKGDATADIR@/nanorc.nanorc"
 
 ## C/C++
-# include "@PKGDATADIR@/c.nanorc"
+include "@PKGDATADIR@/c.nanorc"
 
 ## HTML
-# include "@PKGDATADIR@/html.nanorc"
+include "@PKGDATADIR@/html.nanorc"
 
 ## TeX
-# include "@PKGDATADIR@/tex.nanorc"
+include "@PKGDATADIR@/tex.nanorc"
 
 ## Quoted emails (under e.g. mutt)
-# include "@PKGDATADIR@/mutt.nanorc"
+include "@PKGDATADIR@/mutt.nanorc"
 
 ## Patch files
-# include "@PKGDATADIR@/patch.nanorc"
+include "@PKGDATADIR@/patch.nanorc"
 
 ## Manpages
-# include "@PKGDATADIR@/man.nanorc"
+include "@PKGDATADIR@/man.nanorc"
 
 ## Groff
-# include "@PKGDATADIR@/groff.nanorc"
+include "@PKGDATADIR@/groff.nanorc"
 
 ## Perl
-# include "@PKGDATADIR@/perl.nanorc"
+include "@PKGDATADIR@/perl.nanorc"
 
 ## Python
-# include "@PKGDATADIR@/python.nanorc"
+include "@PKGDATADIR@/python.nanorc"
 
 ## Ruby
-# include "@PKGDATADIR@/ruby.nanorc"
+include "@PKGDATADIR@/ruby.nanorc"
 
 ## Java
-# include "@PKGDATADIR@/java.nanorc"
+include "@PKGDATADIR@/java.nanorc"
 
 ## Assembler
-# include "@PKGDATADIR@/asm.nanorc"
+include "@PKGDATADIR@/asm.nanorc"
 
 ## Bourne shell scripts
-# include "@PKGDATADIR@/sh.nanorc"
+include "@PKGDATADIR@/sh.nanorc"
 
 ## POV-Ray
-# include "@PKGDATADIR@/pov.nanorc"
+include "@PKGDATADIR@/pov.nanorc"
+
+## FrugalBuild
+include "@PKGDATADIR@/frugalbuild.nanorc"
+
diff -Naur nano-2.0.1/doc/syntax/frugalbuild.nanorc nano-2.0.1-p/doc/syntax/frugalbuild.nanorc
--- nano-2.0.1/doc/syntax/frugalbuild.nanorc	1970-01-01 01:00:00.000000000 +0100
+++ nano-2.0.1-p/doc/syntax/frugalbuild.nanorc	2006-12-02 12:47:50.000000000 +0100
@@ -0,0 +1,42 @@
+# (c) 2006 Gabriel Craciunescu <crazy@frugalware.org>
+# frugalbuild.nanorc for Frugalware
+# distributed under GPL License
+
+syntax "frugalbuild" "^.*FrugalBuild$"
+color brightwhite start="^" end="$"
+color brightcyan "^(pkgname|pkgver|pkgrel|pkgdesc|url|license|source|groups|backup|depends)"
+color brightcyan "^(|makedepends|conflicts|provides|replaces|up2date|archs|stable|rodepends)"
+color magenta "^(subpkgs|subinstall|subdescs|subarchs|suboptions|subdepends|subgroups|subrodepends)"
+color magenta "^(sublicense|subreplaces|subremoves|subconflicts|subprovides|subbackup|subbackup|suboptions)"
+color brightred "\<(configure|make)\>"
+color brightcyan "\<(DESTDIR|PREFIX|prefix|sysconfdir|datadir|libdir|includedir|mandir)\>"
+color red "F(build|cd|mkdir|doc|docrel|sed|die|message|make|makeinstall|patch|patchall|rcd|rcd2|file|ln|filerel|chmod|split|rm|cp|cpr|mv)"
+color red "F(desktop|autoreconf|cprrel|acu|dirschmod|fileschmod|dirschown|fileschown|exerel|exe|iconrel|icon|deststrip|conf|nant|sort)"
+color red "F(unpack_makeself|autoconfize|lasttar|lasttgz|lasttarbz2|msgfmt|desktop2)"
+color brightblue "\<F(srcdir|destdir|prefix|menudir|include|confopts|sysconfdir|localstatedir)\>"
+color brightblue "\<(md5sums|sha1sums)\>"
+color brightblue "^(options)"
+color white """ "'" "(<|>|@|)"
+## sourceforge.sh
+color brightcyan "_F_sourceforge_(prefix|ext|mirror|dirname|name|broken_up2date)"
+## util.sh
+color brightcyan "_F_(cd_path|conf_configure|conf_perl_pipefrom|rcd_name)"
+## kernel.sh
+color brightcyan "_F_kernel_(stable|rc|git|mm|vmlinuz|verbose|name|ver|dontsedarch|dontfakeversion|rel)"
+## perl.sh
+color brightcyan "_F_perl_(sourcename|ext|url)"
+## scm.sh
+color brightcyan "_F_scm_(type|url|password|tag|module)"
+## kdeapps.sh
+color brightcyan "_F_kdeapps_(id|id2)"
+## gnome-scriptlet.sh
+color brightcyan "_F_gnome_(schemas|entries|desktop|scrollkeeper|mime|iconcache|name)"
+color red "Fbuild_(slice_scrollkeeper|gnome_scriptlet)"
+## mono.sh
+color red "F(monoexport|monocleanup|build_mono)"
+## some misc stuff
+icolor white "^# .*modified+:"
+icolor white "^# .*(Time|time)+:"
+icolor white "^# .*M.*r+:"
+icolor white "^# .*C.*r+:"
+
diff -Naur nano-2.0.1/doc/syntax/Makefile.am nano-2.0.1-p/doc/syntax/Makefile.am
--- nano-2.0.1/doc/syntax/Makefile.am	2006-11-02 07:32:02.000000000 +0100
+++ nano-2.0.1-p/doc/syntax/Makefile.am	2006-12-02 12:35:26.000000000 +0100
@@ -12,6 +12,7 @@
 		python.nanorc \
 		ruby.nanorc \
 		sh.nanorc \
-		tex.nanorc
+		tex.nanorc \
+		frugalbuild.nanorc
 
 EXTRA_DIST = $(pkgdata_DATA)
