diff -Naur coq-8.3/Makefile coq-8.3/Makefile
--- coq-8.3/Makefile	2010-10-13 21:53:28.000000000 +0200
+++ coq-8.3/Makefile	2010-10-18 13:29:41.384000036 +0200
@@ -160,9 +160,15 @@
 stage1 $(STAGE1_TARGETS) : always
 	$(call stage-template,1)
 
+$(STAGE1_TARGETS_IMPLICIT) : always
+	$(call stage-template,1)
+
 stage2 $(STAGE2_TARGETS) : stage1
 	$(call stage-template,2)
 
+$(STAGE2_TARGETS_IMPLICIT) : stage1
+	$(call stage-template,2)
+
 # Nota:
 # - world is one of the targets in $(STAGE2_TARGETS), hence launching
 # "make" or "make world" leads to recursion into stage1 then stage2
diff -Naur coq-8.3/Makefile.common coq-8.3/Makefile.common
--- coq-8.3/Makefile.common	2010-06-23 15:17:17.000000000 +0200
+++ coq-8.3/Makefile.common	2010-10-18 13:31:29.767000038 +0200
@@ -374,10 +374,12 @@
 STAGE1_TARGETS:= $(STAGE1) $(COQDEPBOOT) \
   $(GENFILES) \
   source-doc revision toplevel/mltop.byteml toplevel/mltop.optml \
-  $(STAGE1_ML4:.ml4=.ml4-preprocessed) %.o
+  $(STAGE1_ML4:.ml4=.ml4-preprocessed)
+
+STAGE1_TARGETS_IMPLICIT:=%.o
 
 ifdef CM_STAGE1
- STAGE1_TARGETS+=$(CAML_OBJECT_PATTERNS)
+ STAGE1_TARGETS_IMPLICIT+=$(CAML_OBJECT_PATTERNS)
 endif
 
 ## Enumeration of targets that require being done at stage2
@@ -402,12 +404,13 @@
   printers debug initplugins plugins \
   world install coqide coqide-files coq coqlib \
   coqlight states check init theories theories-light \
-  $(DOC_TARGETS) $(VO_TARGETS) validate \
-  %.vo %.glob states/% install-% %.ml4-preprocessed \
+  $(DOC_TARGETS) $(VO_TARGETS) validate
+
+STAGE2_TARGETS_IMPLICIT:=%.vo %.glob states/% install-% %.ml4-preprocessed \
   $(DOC_TARGET_PATTERNS)
 
 ifndef CM_STAGE1
- STAGE2_TARGETS+=$(CAML_OBJECT_PATTERNS)
+ STAGE2_TARGETS_IMPLICIT+=$(CAML_OBJECT_PATTERNS)
 endif
 
 
