--- integer-logarithms-1.0.3.1/integer-logarithms.cabal	2001-09-09 03:46:40.000000000 +0200
+++ integer-logarithms-1.0.3.1.new/integer-logarithms.cabal	2024-07-09 18:09:05.359948247 +0200
@@ -1,5 +1,6 @@
 name:               integer-logarithms
 version:            1.0.3.1
+x-revision:         6
 cabal-version:      >=1.10
 author:             Daniel Fischer
 copyright:
@@ -34,7 +35,13 @@
      || ==8.4.4
      || ==8.6.4
      || ==8.8.4
-     || ==8.10.2
+     || ==8.10.4
+     || ==9.0.2
+     || ==9.2.8
+     || ==9.4.8
+     || ==9.6.4
+     || ==9.8.2
+     || ==9.10.1
   , GHCJS ==8.4
 
 extra-source-files:
@@ -56,8 +63,8 @@
   hs-source-dirs:   src
   build-depends:
       array     >=0.3 && <0.6
-    , base      >=4.3 && <4.16
-    , ghc-prim  >=0   && <0.8
+    , base      >=4.3 && <4.21
+    , ghc-prim  <0.12
 
   if !impl(ghc >=7.10)
     build-depends: nats >=1.1.2 && <1.2
@@ -65,7 +72,7 @@
   if impl(ghc >=9.0)
     build-depends:
         base        >=4.15
-      , ghc-bignum  >=1.0  && <1.1
+      , ghc-bignum  >=1.0  && <1.4
 
     if !flag(integer-gmp)
       build-depends: invalid-cabal-flag-settings <0
@@ -118,7 +125,7 @@
     , integer-logarithms
     , QuickCheck          >=2.14.1 && <2.15
     , smallcheck          >=1.2    && <1.3
-    , tasty               >=0.10   && <1.4
+    , tasty               >=0.10   && <1.6
     , tasty-hunit         >=0.9    && <0.11
     , tasty-quickcheck    >=0.8    && <0.11
     , tasty-smallcheck    >=0.8    && <0.9
