diff -Nru lineakd-0.8.3.orig/lineakd/lineakkb.def lineakd-0.8.3/lineakd/lineakkb.def
--- lineakd-0.8.3.orig/lineakd/lineakkb.def	2005-10-25 14:19:13.000000000 +0200
+++ lineakd-0.8.3/lineakd/lineakkb.def	2005-10-25 14:20:07.000000000 +0200
@@ -2244,6 +2244,7 @@
     AudioRaiseVolume = 158
     AudioLowerVolume = 165
     WWW              = 130
+  [END KEYS]
 [END GENK-WTTOV]
 
 #### END Genius Keyboards ####
@@ -4537,7 +4538,7 @@
     VolumeDown  = 174
     VolumeUp    = 176
     Web|Home    = 178
-  END KEYS]
+  [END KEYS]
 [END MMNKP]
 
 [MNKpro]
