diff -Naur lineakd-0.8.4/utils/Makefile.am lineakd-0.8.4.new/utils/Makefile.am
--- lineakd-0.8.4/utils/Makefile.am	2005-08-01 10:12:32.000000000 +0530
+++ lineakd-0.8.4.new/utils/Makefile.am	2007-03-16 11:41:49.000000000 +0530
@@ -37,42 +37,3 @@
 DEFS = @DEFS@ -DCONFDIR=\"@sysconfdir@\" -DPLUGINDIR=\"@libdir@/lineakd/plugins\"
 helper_suid = $(HELPER_SUID)
 
-if EVTEST
-evtest.1.bz2:
-	bzip2 -c evtest.1 > evtest.1.bz2
-endif
-
-send_to_keyboard.8.bz2:
-	bzip2 -c send_to_keyboard.8 > send_to_keyboard.8.bz2
-xsendkeycode.8.bz2:
-	bzip2 -c xsendkeycode.8 > xsendkeycode.8.bz2
-xsendkeys.8.bz2:
-	bzip2 -c xsendkeys.8 > xsendkeys.8.bz2
-
-#install-data-local: send_to_keyboard.8.bz2 evtest.1.bz2 xsendkeycode.8.bz2 xsendkeys.8.bz2
-if EVTEST
-install-data-local: send_to_keyboard.8.bz2 evtest.1.bz2 xsendkeycode.8.bz2 xsendkeys.8.bz2
-	$(mkinstalldirs) $(mandir)/man1/
-	$(INSTALL_DATA) $(srcdir)/evtest.1.bz2 $(mandir)/man1/evtest.1.bz2
-else
-install-data-local: send_to_keyboard.8.bz2 xsendkeycode.8.bz2 xsendkeys.8.bz2
-endif
-if SEND_TO_KEYBOARD
-	$(mkinstalldirs) $(mandir)/man8/
-	chmod 755 $(DESTDIR)$(prefix)/sbin/send_to_keyboard
-	if test "$(HELPER_SUID)" != "no" ; then \
-	   chown root $(DESTDIR)$(prefix)/sbin/send_to_keyboard ; \
-	   chmod +s $(DESTDIR)$(prefix)/sbin/send_to_keyboard ; \
-	fi
-	$(INSTALL_DATA) $(srcdir)/send_to_keyboard.8.bz2 $(mandir)/man8/send_to_keyboard.8.bz2
-endif
-	$(INSTALL_DATA) $(srcdir)/xsendkeycode.8.bz2 $(mandir)/man8/xsendkeycode.8.bz2
-	$(INSTALL_DATA) $(srcdir)/xsendkeys.8.bz2 $(mandir)/man8/xsendkeys.8.bz2
-
-uninstall-local:
-	-rm -f $(mandir)/man8/send_to_keyboard.8.bz2
-	-rm -f $(mandir)/man8/xsendkeycode.8.bz2
-	-rm -f $(mandir)/man8/xsendkeys.8.bz2
-if EVTEST
-	-rm -f $(mandir)/man1/evtest.1.bz2
-endif
diff -Naur lineakd-0.8.4/utils/Makefile.in lineakd-0.8.4.new/utils/Makefile.in
--- lineakd-0.8.4/utils/Makefile.in	2005-11-09 20:49:28.000000000 +0530
+++ lineakd-0.8.4.new/utils/Makefile.in	2007-03-16 11:44:20.000000000 +0530
@@ -649,34 +649,6 @@
 xsendkeys.8.bz2:
 	bzip2 -c xsendkeys.8 > xsendkeys.8.bz2
 
-#install-data-local: send_to_keyboard.8.bz2 evtest.1.bz2 xsendkeycode.8.bz2 xsendkeys.8.bz2
-@EVTEST_TRUE@install-data-local: send_to_keyboard.8.bz2 evtest.1.bz2 xsendkeycode.8.bz2 xsendkeys.8.bz2
-@EVTEST_TRUE@	$(mkinstalldirs) $(mandir)/man1/
-@EVTEST_TRUE@	$(INSTALL_DATA) $(srcdir)/evtest.1.bz2 $(mandir)/man1/evtest.1.bz2
-@EVTEST_FALSE@install-data-local: send_to_keyboard.8.bz2 xsendkeycode.8.bz2 xsendkeys.8.bz2
-@SEND_TO_KEYBOARD_TRUE@	$(mkinstalldirs) $(mandir)/man8/
-@SEND_TO_KEYBOARD_TRUE@	chmod 755 $(DESTDIR)$(prefix)/sbin/send_to_keyboard
-@SEND_TO_KEYBOARD_TRUE@	if test "$(HELPER_SUID)" != "no" ; then \
-@SEND_TO_KEYBOARD_TRUE@	   chown root $(DESTDIR)$(prefix)/sbin/send_to_keyboard ; \
-@SEND_TO_KEYBOARD_TRUE@	   chmod +s $(DESTDIR)$(prefix)/sbin/send_to_keyboard ; \
-@SEND_TO_KEYBOARD_TRUE@	fi
-@SEND_TO_KEYBOARD_TRUE@	$(INSTALL_DATA) $(srcdir)/send_to_keyboard.8.bz2 $(mandir)/man8/send_to_keyboard.8.bz2
-	$(INSTALL_DATA) $(srcdir)/xsendkeycode.8.bz2 $(mandir)/man8/xsendkeycode.8.bz2
-	$(INSTALL_DATA) $(srcdir)/xsendkeys.8.bz2 $(mandir)/man8/xsendkeys.8.bz2
-
-uninstall-local:
-#>- 	-rm -f $(mandir)/man8/send_to_keyboard.8.bz2
-#>+ 1
-	-rm -f $(DESTDIR)$(mandir)/man8/send_to_keyboard.8.bz2
-#>- 	-rm -f $(mandir)/man8/xsendkeycode.8.bz2
-#>+ 1
-	-rm -f $(DESTDIR)$(mandir)/man8/xsendkeycode.8.bz2
-#>- 	-rm -f $(mandir)/man8/xsendkeys.8.bz2
-#>+ 1
-	-rm -f $(DESTDIR)$(mandir)/man8/xsendkeys.8.bz2
-#>- @EVTEST_TRUE@	-rm -f $(mandir)/man1/evtest.1.bz2
-#>+ 1
-@EVTEST_TRUE@	-rm -f $(DESTDIR)$(mandir)/man1/evtest.1.bz2
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
