--- semialign-1.2.0.1/semialign.cabal	2001-09-09 03:46:40.000000000 +0200
+++ semialign-1.2.0.1.new/semialign.cabal	2022-10-13 16:53:27.663950673 +0200
@@ -1,6 +1,7 @@
 cabal-version:      >=1.10
 name:               semialign
 version:            1.2.0.1
+x-revision:         3
 synopsis:
   Align and Zip type-classes from the common Semialign ancestor.
 
@@ -30,9 +31,10 @@
      || ==8.4.4
      || ==8.6.5
      || ==8.8.4
-     || ==8.10.4
-     || ==9.0.1
-     || ==9.2.1
+     || ==8.10.7
+     || ==9.0.2
+     || ==9.2.4
+     || ==9.4.1
   , GHCJS ==8.4
 
 source-repository head
@@ -66,7 +68,7 @@
 
   -- ghc boot libs
   build-depends:
-      base          >=4.5.1.0 && <4.17
+      base          >=4.5.1.0 && <4.18
     , containers    >=0.4.2.1 && <0.7
     , transformers  >=0.3.0.0 && <0.7
 
@@ -80,7 +82,7 @@
     , indexed-traversable-instances  >=0.1      && <0.2
     , tagged                         >=0.8.6    && <0.9
     , unordered-containers           >=0.2.8.0  && <0.3
-    , vector                         >=0.12.0.2 && <0.13
+    , vector                         >=0.12.0.2 && <0.14
 
   -- base shims
   if !impl(ghc >=8.2)
@@ -88,7 +90,7 @@
 
   if !impl(ghc >=8.0)
     build-depends:
-        semigroups           >=0.18.5  && <0.20
+        semigroups           >=0.18.5  && <0.21
       , transformers         >=0.3.0.0 && <0.7
       , transformers-compat  >=0.6.5   && <0.8
 
