diff -Naur kpogre-1.5.8/kpogre/DbObjects/kpgtreeitem.h kpogre-1.5.8-p/kpogre/DbObjects/kpgtreeitem.h
--- kpogre-1.5.8/kpogre/DbObjects/kpgtreeitem.h	2007-10-25 14:25:43.000000000 +0200
+++ kpogre-1.5.8-p/kpogre/DbObjects/kpgtreeitem.h	2007-11-11 23:37:00.000000000 +0100
@@ -22,6 +22,7 @@
 #define QT_NO_ASCII_CAST
 
 #include <qlistview.h>
+#include <cassert>
 
 // include files for libpgxx 
 #include <pqxx/transaction> 
