--- src/main.c.orig	2011-10-25 14:09:57.161042945 +0200
+++ src/main.c	2011-10-25 14:14:21.961042330 +0200
@@ -233,11 +233,10 @@
     }
 
     xmlIndentTreeOutput = 1;
-    if (!obt_xml_load_config_file(parse_i,
-                                  "openbox",
-                                  (obc_config_file ?
-                                   obc_config_file : "rc.xml"),
-                                  "openbox_config"))
+    if (!((obc_config_file &&
+	    obt_xml_load_file(parse_i, obc_config_file, "openbox_config")) ||
+            obt_xml_load_config_file(parse_i, "openbox", "rc.xml",
+		    		     "openbox_config")))
     {
         obconf_error(_("Failed to load an rc.xml. You have probably failed to install Openbox properly."), TRUE);
         exit_with_error = TRUE;
