diff -ur a/modules.d/50plymouth/module-setup.sh b/modules.d/50plymouth/module-setup.sh
--- a/modules.d/50plymouth/module-setup.sh	2012-02-13 20:12:29.825785878 -0600
+++ b/modules.d/50plymouth/module-setup.sh	2012-02-13 20:15:13.444779796 -0600
@@ -18,7 +18,7 @@
         case $_modname in
             *.ko)      grep -q drm_crtc_init $_modname ;;
             *.ko.gz)  zgrep -q drm_crtc_init $_modname ;;
-            *.ko.xz) xzgrep -q drm_crtc_init $_modname ;;
+            *.ko.xz)  xz -dc < $_modname | grep -q 'drm_crtc_init' ;;
         esac
         if test $? -eq 0; then
             # if the hardware is present, include module even if it is not currently loaded,
