diff -ur a/dracut-functions b/dracut-functions
--- a/dracut-functions	2011-12-15 02:19:05.000000000 -0600
+++ b/dracut-functions	2012-01-22 11:34:30.247234753 -0600
@@ -320,7 +320,7 @@
         inst "${_src%/*}/.${_src##*/}.hmac" "${target%/*}/.${target##*/}.hmac"
     fi
     ddebug "Installing $_src"
-    cp --sparse=always -pfL "$_src" "${initdir}/$target" 
+    cp --sparse=always -pfL "$_src" "${initdir}/$target"
 }
 
 # find symlinks linked to given library file
@@ -464,7 +464,7 @@
     if [[ -d $_realsrc ]]; then
         inst_dir "$_realsrc"
     else
-        inst "$_realsrc" && mkdir -m 0755 -p "${_target%/*}" 
+        inst "$_realsrc" && mkdir -m 0755 -p "${_target%/*}"
     fi
     if [[ -e "${_src}" ]]; then
         ln -sfn $(convert_abs_rel "${_src}" "${_realsrc}") "$_target"
@@ -855,6 +855,7 @@
     if ! [[ $hostonly ]]; then
         _filtercmd='find "$srcmods/kernel/$1" "$srcmods/extra"'
         _filtercmd+=' "$srcmods/weak-updates" -name "*.ko" -o -name "*.ko.gz"'
+        _filtercmd+=' -o -name "*.ko.xz"'
         _filtercmd+=' 2>/dev/null'
     else
         _filtercmd='cut -d " " -f 1 </proc/modules|xargs modinfo -F filename '
@@ -867,13 +868,17 @@
                 $2 $initdir/$$.ko && echo "$_modname"
                 rm -f $initdir/$$.ko
                 ;;
+            *.ko.xz) xz -dc "$_modname" > $initdir/$$.ko
+                $2 $initdir/$$.ko && echo "$_modname"
+                rm -f $initdir/$$.ko
+                ;;
         esac
     done
 )
 find_kernel_modules_by_path () (
     if ! [[ $hostonly ]]; then
         find "$srcmods/kernel/$1" "$srcmods/extra" "$srcmods/weak-updates" \
-          -name "*.ko" -o -name "*.ko.gz" 2>/dev/null
+          -name "*.ko" -o -name "*.ko.gz" -o -name "*.ko.xz" 2>/dev/null
     else
         cut -d " " -f 1 </proc/modules \
         | xargs modinfo -F filename -k $kernel 2>/dev/null
diff -ur a/modules.d/40network/module-setup.sh b/modules.d/40network/module-setup.sh
--- a/modules.d/40network/module-setup.sh	2011-12-15 02:19:05.000000000 -0600
+++ b/modules.d/40network/module-setup.sh	2012-01-22 11:44:20.329212595 -0600
@@ -36,6 +36,7 @@
                 case "$_fname" in
                     *.ko)    _fcont="$(<        $_fname)" ;;
                     *.ko.gz) _fcont="$(gzip -dc $_fname)" ;;
+                    *.ko.xz) _fcont="$(xz -dc   $_fname)" ;;
                 esac
                 [[   $_fcont =~ $_net_drivers
                 && ! $_fcont =~ iw_handler_get_spy ]] \
diff -ur a/modules.d/50plymouth/module-setup.sh b/modules.d/50plymouth/module-setup.sh
--- a/modules.d/50plymouth/module-setup.sh	2011-12-15 02:19:05.000000000 -0600
+++ b/modules.d/50plymouth/module-setup.sh	2012-01-22 16:32:21.323563620 -0600
@@ -13,8 +13,12 @@
 installkernel() {
     local _modname
     # Include KMS capable drm drivers
-    for _modname in $(find "$srcmods/kernel/drivers/gpu/drm" "$srcmods/extra" -name '*.ko' 2>/dev/null); do
-        grep -q drm_crtc_init  $_modname && instmods $_modname
+    for _modname in $(find "$srcmods/kernel/drivers/gpu/drm" "$srcmods/extra" \( -name '*.ko' -o -name "*.ko.gz" -o -name "*.ko.xz" \) 2>/dev/null); do
+        case $_modname in
+            *.ko)    grep   -q drm_crtc_init  $_modname && instmods $_modname ;;
+            *.ko.gz) zgrep  -q drm_crtc_init  $_modname && instmods $_modname ;;
+            *.ko.xz) xz -dc < $_modname | grep -q 'drm_crtc_init' && instmods $_modname ;;
+        esac
     done
 }
 
diff -ur a/modules.d/90kernel-modules/module-setup.sh b/modules.d/90kernel-modules/module-setup.sh
--- a/modules.d/90kernel-modules/module-setup.sh	2011-12-15 02:19:05.000000000 -0600
+++ b/modules.d/90kernel-modules/module-setup.sh	2012-01-22 11:45:55.161209054 -0600
@@ -13,6 +13,7 @@
                 while read _f; do case "$_f" in
                     *.ko)    [[ $(<         $_f) =~ $_blockfuncs ]] && echo "$_f" ;;
                     *.ko.gz) [[ $(gzip -dc <$_f) =~ $_blockfuncs ]] && echo "$_f" ;;
+                    *.ko.xz) [[ $(xz -dc   <$_f) =~ $_blockfuncs ]] && echo "$_f" ;;
                     esac
                 done
             }
diff -ur a/modules.d/90multipath/module-setup.sh b/modules.d/90multipath/module-setup.sh
--- a/modules.d/90multipath/module-setup.sh	2011-12-15 02:19:05.000000000 -0600
+++ b/modules.d/90multipath/module-setup.sh	2012-01-22 11:46:47.937207050 -0600
@@ -40,6 +40,7 @@
         while read _f; do case "$_f" in
             *.ko)    [[ $(<         $_f) =~ $_mpfuncs ]] && echo "$_f" ;;
             *.ko.gz) [[ $(gzip -dc <$_f) =~ $_mpfuncs ]] && echo "$_f" ;;
+            *.ko.xz) [[ $(xz -dc   <$_f) =~ $_mpfuncs ]] && echo "$_f" ;;
             esac
         done
     }
diff -ur a/modules.d/95iscsi/module-setup.sh b/modules.d/95iscsi/module-setup.sh
--- a/modules.d/95iscsi/module-setup.sh	2011-12-15 02:19:05.000000000 -0600
+++ b/modules.d/95iscsi/module-setup.sh	2012-01-22 13:57:26.247912660 -0600
@@ -48,6 +48,7 @@
         while read _f; do case "$_f" in
             *.ko)    [[ $(<         $_f) =~ $_iscsifuncs ]] && echo "$_f" ;;
             *.ko.gz) [[ $(gzip -dc <$_f) =~ $_iscsifuncs ]] && echo "$_f" ;;
+            *.ko.xz) [[ $(xz -dc   <$_f) =~ $_iscsifuncs ]] && echo "$_f" ;;
             esac
         done
     }
