diff -ur a/modules.d/03modsign/module-setup.sh b/modules.d/03modsign/module-setup.sh
--- a/modules.d/03modsign/module-setup.sh	2014-03-19 11:16:08.000000000 -0500
+++ b/modules.d/03modsign/module-setup.sh	2014-05-04 01:00:06.547460936 -0500
@@ -29,7 +29,7 @@
 # called by dracut
 install() {
     inst_dir /lib/modules/keys
-    inst_binary /usr/bin/keyctl
+    inst_binary /bin/keyctl
 
     inst_hook pre-trigger 01 "$moddir/load-modsign-keys.sh"
 
