diff --color -Naur watchman-2025.02.17.00/CMakeLists.txt watchman-2025.02.17.00.new/CMakeLists.txt
--- watchman-2025.02.17.00/CMakeLists.txt	2025-02-11 06:04:00.000000000 +0100
+++ watchman-2025.02.17.00.new/CMakeLists.txt	2025-02-24 10:44:49.499870023 +0100
@@ -333,7 +333,7 @@
 find_package(Gflags REQUIRED)
 include_directories(SYSTEM ${GFLAGS_INCLUDE_DIR})
 
-find_package(Glog REQUIRED)
+find_package(Glog CONFIG REQUIRED)
 add_compile_definitions(GLOG_NO_ABBREVIATED_SEVERITIES)
 
 # We indirectly depend on boost.  This logic needs to match
