--- lighttpd-1.4.56/doc/config/lighttpd.conf	2020-11-29 23:32:01.000000000 +0100
+++ lighttpd-1.4.56.new/doc/config/lighttpd.conf	2020-12-01 11:13:10.109392939 +0100
@@ -14,7 +14,7 @@
 ## chroot example aswell.
 ##
 var.log_root    = "/var/log/lighttpd"
-var.server_root = "/srv/www"
+var.server_root = "/var/www/htdocs"
 var.state_dir   = "/run"
 var.home_dir    = "/var/lib/lighttpd"
 var.conf_dir    = "/etc/lighttpd"
@@ -36,7 +36,7 @@
 #var.vhosts_dir  = "/vhosts"
 #var.conf_dir    = "/etc"
 #
-#server.chroot   = "/srv/www"
+#server.chroot   = "/var/www/htdocs"
 
 ##
 ## Some additional variables to make the configuration easier
@@ -101,8 +101,8 @@
 ## Run as a different username/groupname.
 ## This requires root permissions during startup. 
 ##
-server.username  = "lighttpd"
-server.groupname = "lighttpd"
+server.username  = "nobody"
+server.groupname = "nobody"
 
 ##
 ## Enable lighttpd to serve requests on sockets received from systemd
