--- lighttpd-1.4.79/doc/config/lighttpd.annotated.conf	2025-04-04 07:28:20.000000000 +0200
+++ lighttpd-1.4.79.new/doc/config/lighttpd.annotated.conf	2025-05-21 12:43:26.939418207 +0200
@@ -14,7 +14,7 @@
 ## chroot example as well.
 ##
 var.log_root    = "/var/log/lighttpd"
-var.server_root = "/srv/www"
+var.server_root = "/var/www/htdocs"
 var.state_dir   = "/run"
 var.home_dir    = "/var/lib/lighttpd"
 var.conf_dir    = "/etc/lighttpd"
@@ -36,7 +36,7 @@
 #var.vhosts_dir  = "/vhosts"
 #var.conf_dir    = "/etc"
 #
-#server.chroot   = "/srv/www"
+#server.chroot   = "/var/www/htdocs"
 
 ##
 ## Some additional variables to make the configuration easier
@@ -97,8 +97,8 @@
 ## Run as a different username/groupname.
 ## This requires root permissions during startup.
 ##
-server.username  = "lighttpd"
-server.groupname = "lighttpd"
+server.username  = "nobody"
+server.groupname = "nobody"
 
 ##
 ## Enable lighttpd to serve requests on sockets received from systemd
