diff -Naur lighttpd-1.4.51/doc/config/lighttpd.conf lighttpd-1.4.51-p/doc/config/lighttpd.conf
--- lighttpd-1.4.51/doc/config/lighttpd.conf	2018-10-15 00:35:58.000000000 +0200
+++ lighttpd-1.4.51-p/doc/config/lighttpd.conf	2018-10-27 23:06:46.362492159 +0200
@@ -14,7 +14,7 @@
 ## chroot example aswell.
 ##
 var.log_root    = "/var/log/lighttpd"
-var.server_root = "/srv/www"
+var.server_root = "/var/www/htdocs"
 var.state_dir   = "/var/run"
 var.home_dir    = "/var/lib/lighttpd"
 var.conf_dir    = "/etc/lighttpd"
@@ -36,7 +36,7 @@
 #var.vhosts_dir  = "/vhosts"
 #var.conf_dir    = "/etc"
 #
-#server.chroot   = "/srv/www"
+#server.chroot   = "/var/www/htdocs"
 
 ##
 ## Some additional variables to make the configuration easier
@@ -101,8 +101,8 @@
 ## Run as a different username/groupname.
 ## This requires root permissions during startup. 
 ##
-server.username  = "lighttpd"
-server.groupname = "lighttpd"
+server.username  = "nobody"
+server.groupname = "nobody"
 
 ## 
 ## enable core files.
