diff -ur a/src/GUI.cpp b/src/GUI.cpp
--- a/src/GUI.cpp	2011-11-05 21:42:35.000000000 -0500
+++ b/src/GUI.cpp	2012-11-30 00:20:30.516968696 -0600
@@ -779,7 +779,7 @@
 void GUI::Wait(unsigned int ms)
 {
     boost::xtime t;
-    boost::xtime_get(&t, boost::TIME_UTC);
+    boost::xtime_get(&t, boost::TIME_UTC_);
     unsigned int ns_sum = t.nsec + ms * 1000000;
     const unsigned int NANOSECONDS_PER_SECOND = 1000000000;
     unsigned int delta_secs = ns_sum / NANOSECONDS_PER_SECOND;
