diff -Naur pagerankstatus-0.9.7.orig/install.rdf pagerankstatus-0.9.7/install.rdf
--- pagerankstatus-0.9.7.orig/install.rdf	2006-10-25 16:55:01.000000000 +0200
+++ pagerankstatus-0.9.7/install.rdf	2008-06-19 21:45:55.000000000 +0200
@@ -11,7 +11,7 @@
 		<Description>
 			<em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
 			<em:minVersion>0.9</em:minVersion>
-			<em:maxVersion>2.0+</em:maxVersion>
+			<em:maxVersion>3.0+</em:maxVersion>
 		</Description>
 	</em:targetApplication>
 
