--- pagerankstatus-0.9.6/install.rdf.orig	2006-10-24 21:57:05.000000000 +0200
+++ pagerankstatus-0.9.6/install.rdf	2006-10-24 21:57:35.000000000 +0200
@@ -11,7 +11,7 @@
 		<Description>
 			<em:id>{ec8030f7-c20a-464f-9b0e-13a3a9e97384}</em:id>
 			<em:minVersion>0.9</em:minVersion>
-			<em:maxVersion>1.5+</em:maxVersion>
+			<em:maxVersion>2.0+</em:maxVersion>
 		</Description>
 	</em:targetApplication>
 
