diff -urN monodevelop-20060904.orig/configure.in monodevelop-20060904/configure.in
--- monodevelop-20060904.orig/configure.in	2006-09-04 14:58:21.000000000 +0100
+++ monodevelop-20060904/configure.in	2006-09-04 14:59:20.000000000 +0100
@@ -254,7 +254,7 @@
 	# get zip to create jar/xpi for chrome
 	AC_PATH_PROG(ZIP, zip)
 else
-	MOZILLA_HOME="`$PKG_CONFIG --variable=libdir mozilla-gtkmozembed`"
+	MOZILLA_HOME="`$PKG_CONFIG --variable=libdir firefox-gtkmozembed`"
 	AC_SUBST(MOZILLA_HOME)
 fi
 
diff -urN monodevelop-20060904.orig/monodevelop.in monodevelop-20060904/monodevelop.in
--- monodevelop-20060904.orig/monodevelop.in	2006-09-04 14:58:21.000000000 +0100
+++ monodevelop-20060904/monodevelop.in	2006-09-04 14:59:35.000000000 +0100
@@ -18,7 +18,7 @@
 
 if test -n "$MOZILLA_FIVE_HOME"; then
 	MOZILLA_HOME=$MOZILLA_FIVE_HOME
-elif [ -f @MOZILLA_HOME@/chrome/comm.jar ]; then
+elif [ -f @MOZILLA_HOME@/libgtkembedmoz.so ]; then
         MOZILLA_HOME=@MOZILLA_HOME@
 elif grep GRE_PATH /etc/gre.d/*.conf > /dev/null ; then
 	MOZILLA_HOME=$(grep -h GRE_PATH= /etc/gre.d/*.conf | cut -d '"' -f 2 -d = | head -n 1)
