diff -urN monodevelop-0.13.1.orig/configure.in monodevelop-0.13.1/configure.in
--- monodevelop-0.13.1.orig/configure.in	2007-03-15 20:08:54.000000000 +0000
+++ monodevelop-0.13.1/configure.in	2007-03-15 20:11:45.000000000 +0000
@@ -252,6 +252,8 @@
 	if !(test -n "$MOZILLA_HOME"); then
 		if test -n "$MOZILLA_FIVE_HOME"; then
 			MOZILLA_HOME=$MOZILLA_FIVE_HOME
+		elif [ -e /usr/lib/firefox ]; then
+			MOZILLA_HOME=/usr/lib/firefox
 		elif grep GRE_PATH /etc/gre.d/*.conf > /dev/null ; then
 			MOZILLA_HOME=$(grep -h GRE_PATH= /etc/gre.d/*.conf | cut -d '"' -f 2 -d = | head -n 1)
 		elif [ $(which mozilla 2>/dev/null) ] && grep MOZILLA_FIVE_HOME= "$(which mozilla)" > /dev/null ; then
diff -urN monodevelop-0.13.1.orig/monodevelop monodevelop-0.13.1/monodevelop
--- monodevelop-0.13.1.orig/monodevelop	2007-03-15 20:08:54.000000000 +0000
+++ monodevelop-0.13.1/monodevelop	2007-03-15 20:11:26.000000000 +0000
@@ -19,6 +19,8 @@
 #(NOTE: any changes to this block should be kept in sync with the one in configure.in)
 if test -n "$MOZILLA_FIVE_HOME"; then
 	MOZILLA_HOME=$MOZILLA_FIVE_HOME
+elif [ -e /usr/lib/firefox ]; then
+	MOZILLA_HOME=/usr/lib/firefox
 elif [ -f /opt/mozilla/lib/chrome/comm.jar ]; then
         MOZILLA_HOME=/opt/mozilla/lib
 elif grep GRE_PATH /etc/gre.d/*.conf > /dev/null ; then
