--- _monodevelop-1.9.1/Makefile.in	2008-11-19 17:43:06.000000000 +0100
+++ _monodevelop-1.9.1/Makefile.in	2009-02-11 20:44:30.000000000 +0100
@@ -222,7 +222,7 @@
 	build/bin/MonoDevelop.Ide.addin.xml \
 	build/bin/MonoDevelop.Projects.addin.xml \
 	build/bin/MonoDevelop.Projects.Gui.addin.xml \
-	build/AddIns/MonoDevelop.SourceEditor.addin.xml \
+	build/AddIns/MonoDevelop.SourceEditor2.addin.xml \
 	build/AddIns/BackendBindings/BooBinding.addin.xml \
 	build/AddIns/BackendBindings/CSharpBinding.addin.xml \
 	build/AddIns/BackendBindings/ILAsmBinding.addin.xml \
--- _monodevelop-1.9.1/src/addins/Makefile.in	2008-11-19 17:43:04.000000000 +0100
+++ _monodevelop-1.9.1/src/addins/Makefile.in	2009-02-11 21:21:49.000000000 +0100
@@ -29,7 +29,7 @@
 NORMAL_UNINSTALL = :
 PRE_UNINSTALL = :
 POST_UNINSTALL = :
-@ENABLE_GTKSOURCEVIEW2_TRUE@am__append_1 = MonoDevelop.SourceEditor
+@ENABLE_GTKSOURCEVIEW2_TRUE@am__append_1 = MonoDevelop.SourceEditor2
 subdir = src/addins
 DIST_COMMON = $(srcdir)/Makefile.am $(srcdir)/Makefile.in
 ACLOCAL_M4 = $(top_srcdir)/aclocal.m4
@@ -59,8 +59,7 @@
 	MonoDevelop.GtkCore MonoDevelop.SourceEditor2 \
 	MonoDevelop.XmlEditor AspNetAddIn MonoDevelop.WebReferences \
 	MonoDevelop.RegexToolkit CBinding MonoDevelop.AssemblyBrowser \
-	MonoDevelop.Moonlight MonoDevelop.CodeMetrics WindowsPlatform \
-	MonoDevelop.SourceEditor
+	MonoDevelop.Moonlight MonoDevelop.CodeMetrics WindowsPlatform
 DISTFILES = $(DIST_COMMON) $(DIST_SOURCES) $(TEXINFOS) $(EXTRA_DIST)
 ACLOCAL = @ACLOCAL@
 ALL_LINGUAS = @ALL_LINGUAS@
