--- screenfetch-dev.orig	2012-06-29 16:10:02.342221167 +0200
+++ screenfetch-dev	2012-06-29 16:10:36.198215440 +0200
@@ -436,7 +436,7 @@
 detectpkgs () {
 	case $distro in
 		'Arch Linux'|'ParabolaGNU/Linux-libre'|'Chakra') pkgs=$(pacman -Qq | wc -l) ;;
-		'Frugalware') pkgs=$(pacman-g2 -Qq | wc -l) ;;
+		'Frugalware') pkgs=$(pacman-g2 -Q | wc -l) ;;
 		'Ubuntu'|'Mint'|'SolusOS'|'Debian'|'LMDE'|'CrunchBang'|'Peppermint'|'LinuxDeepin') pkgs=$(dpkg --get-selections | wc -l) ;;
 		'Slackware') pkgs=$(ls -1 /var/log/packages | wc -l) ;;
 		'Gentoo') pkgs=$(ls -d /var/db/pkg/*/* | wc -l) ;;
