diff -Naur spca5xx-20051001.orig/drivers/usb/spca5xx.h spca5xx-20051001/drivers/usb/spca5xx.h
--- spca5xx-20051001.orig/drivers/usb/spca5xx.h	2005-10-30 00:17:46.000000000 +0200
+++ spca5xx-20051001/drivers/usb/spca5xx.h	2005-10-30 00:20:02.000000000 +0200
@@ -12,6 +12,7 @@
 #include <linux/videodev.h>
 #include <linux/smp_lock.h>
 #include <linux/usb.h>
+#include <linux/version.h>
 /*
  *      tq_scheduler disappeared @ lk-2.4.0-test12
  *      (right when <linux/sched.h> newly defined TQ_ACTIVE)
