diff --exclude='*.orig' -NrU5 boson-all-0.13.orig/code/boson/programs/mainnoguimain.cpp boson-all-0.13/code/boson/programs/mainnoguimain.cpp
--- boson-all-0.13.orig/code/boson/programs/mainnoguimain.cpp	2006-03-09 18:30:29.000000000 +0100
+++ boson-all-0.13/code/boson/programs/mainnoguimain.cpp	2008-05-05 16:52:40.000000000 +0200
@@ -55,11 +55,11 @@
     { 0, 0, 0 }
 };
 
 static bool parseArgs(MainNoGUIStartOptions* options, KCmdLineArgs* args);
 static bool parseAddComputerArgs(MainNoGUIStartOptions* options, KCmdLineArgs* args);
-static bool parsePlayFieldArgs(MainNoGUIStartOptions* args, KCmdLineArgs* args);
+static bool parsePlayFieldArgs(MainNoGUIStartOptions* options, KCmdLineArgs* args);
 
 static void postBosonConfigInit();
 
 int main(int argc, char **argv)
 {
diff --exclude='*.orig' -NrU5 boson-all-0.13.orig/code/ufo/include/ufo/util/urectangle.hpp boson-all-0.13/code/ufo/include/ufo/util/urectangle.hpp
--- boson-all-0.13.orig/code/ufo/include/ufo/util/urectangle.hpp	2005-12-01 00:13:04.000000000 +0100
+++ boson-all-0.13/code/ufo/include/ufo/util/urectangle.hpp	2008-05-05 16:52:40.000000000 +0200
@@ -32,10 +32,13 @@
 
 #include "upoint.hpp"
 #include "udimension.hpp"
 #include "uinsets.hpp"
 
+#include <cstdlib>
+#include <memory>
+
 namespace ufo {
 
 /** @short An abstraction of a rectangle (x, y, width and height).
   * @ingroup appearance
   *
diff --exclude='*.orig' -NrU5 boson-all-0.13.orig/code/ufo/src/gl/ugl_driver.cpp boson-all-0.13/code/ufo/src/gl/ugl_driver.cpp
--- boson-all-0.13.orig/code/ufo/src/gl/ugl_driver.cpp	2005-12-01 00:13:04.000000000 +0100
+++ boson-all-0.13/code/ufo/src/gl/ugl_driver.cpp	2008-05-05 16:52:40.000000000 +0200
@@ -27,10 +27,12 @@
 
 #include "ufo/gl/ugl_driver.hpp"
 
 #include "ufo/usharedlib.hpp"
 
+#include <cstdlib>
+
 #ifdef UFO_OS_WIN32
 #define UFO_GL_LIB "OPENGL32.DLL"
 #else
 #define UFO_GL_LIB "libGL.so.1"
 #endif
diff --exclude='*.orig' -NrU5 boson-all-0.13.orig/code/ufo/src/uabstractcontext.cpp boson-all-0.13/code/ufo/src/uabstractcontext.cpp
--- boson-all-0.13.orig/code/ufo/src/uabstractcontext.cpp	2005-08-22 02:58:03.000000000 +0200
+++ boson-all-0.13/code/ufo/src/uabstractcontext.cpp	2008-05-05 16:52:40.000000000 +0200
@@ -48,10 +48,11 @@
 // for key bindings
 #include "ufo/uinputmap.hpp"
 #include "ufo/ukeystroke.hpp"
 #include "ufo/events/uactionevent.hpp"
 
+#include <cstdlib>
 
 using namespace ufo;
 
 //#define DEBUG_SEND 1
 //#define DEBUG_FIRE_MOUSE_EVENT 1
diff --exclude='*.orig' -NrU5 boson-all-0.13.orig/code/ufo/src/uabstractdisplay.cpp boson-all-0.13/code/ufo/src/uabstractdisplay.cpp
--- boson-all-0.13.orig/code/ufo/src/uabstractdisplay.cpp	2005-12-01 00:13:04.000000000 +0100
+++ boson-all-0.13/code/ufo/src/uabstractdisplay.cpp	2008-05-05 16:52:40.000000000 +0200
@@ -43,10 +43,11 @@
 #include "ufo/gl/ugl_image.hpp"
 #include "ufo/image/uimageio.hpp"
 #include "ufo/uvolatiledata.hpp"
 
 #include <bitset>
+#include <algorithm>
 
 namespace ufo {
 
 // FIXME: this should be in a file udisplay.cpp
 UDisplay * UDisplay::m_default = NULL;
diff --exclude='*.orig' -NrU5 boson-all-0.13.orig/code/ufo/src/uabstracttoolkit.cpp boson-all-0.13/code/ufo/src/uabstracttoolkit.cpp
--- boson-all-0.13.orig/code/ufo/src/uabstracttoolkit.cpp	2005-12-01 00:13:04.000000000 +0100
+++ boson-all-0.13/code/ufo/src/uabstracttoolkit.cpp	2008-05-05 16:52:40.000000000 +0200
@@ -60,10 +60,13 @@
 #include "ufo/ux/uxwgldriver.hpp"
 #endif
 
 #include "ufo/uvideodriver.hpp"
 
+#include <climits>
+#include <cstring>
+
 namespace ufo {
 bool operator<(const UPluginInfo & p1, const UPluginInfo & p2) {
 	return ((p1.category < p2.category) || (p1.feature < p2.feature));
 }
 }
diff --exclude='*.orig' -NrU5 boson-all-0.13.orig/code/ufo/src/ubuttongroup.cpp boson-all-0.13/code/ufo/src/ubuttongroup.cpp
--- boson-all-0.13.orig/code/ufo/src/ubuttongroup.cpp	2005-12-01 00:13:04.000000000 +0100
+++ boson-all-0.13/code/ufo/src/ubuttongroup.cpp	2008-05-05 16:52:40.000000000 +0200
@@ -29,10 +29,12 @@
 
 #include "ufo/widgets/ubutton.hpp"
 // special case for radio buttons
 #include "ufo/widgets/uradiobutton.hpp"
 
+#include <algorithm>
+
 using namespace ufo;
 
 UFO_IMPLEMENT_DYNAMIC_CLASS(UButtonGroup, UObject)
 
 UButtonGroup::UButtonGroup()
diff --exclude='*.orig' -NrU5 boson-all-0.13.orig/code/ufo/src/ucontextgroup.cpp boson-all-0.13/code/ufo/src/ucontextgroup.cpp
--- boson-all-0.13.orig/code/ufo/src/ucontextgroup.cpp	2005-12-01 00:13:04.000000000 +0100
+++ boson-all-0.13/code/ufo/src/ucontextgroup.cpp	2008-05-05 16:52:40.000000000 +0200
@@ -29,10 +29,12 @@
 
 #include "ufo/ucontext.hpp"
 #include "ufo/utoolkit.hpp"
 #include "ufo/uvolatiledata.hpp"
 
+#include <algorithm>
+
 using namespace ufo;
 
 UFO_IMPLEMENT_DYNAMIC_CLASS(UContextGroup, UObject)
 
 
diff --exclude='*.orig' -NrU5 boson-all-0.13.orig/code/ufo/src/ufo_debug.cpp boson-all-0.13/code/ufo/src/ufo_debug.cpp
--- boson-all-0.13.orig/code/ufo/src/ufo_debug.cpp	2005-12-01 00:13:04.000000000 +0100
+++ boson-all-0.13/code/ufo/src/ufo_debug.cpp	2008-05-05 16:52:40.000000000 +0200
@@ -25,10 +25,11 @@
  * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA *
  ***************************************************************************/
 
 #include "ufo/ufo_global.hpp"
 
+#include <cstdlib>
 #include <cstring>
 #include <cstdarg>
 #include <cstdio> // for vsnprintf
 
 ufo::UDebugStream &
diff --exclude='*.orig' -NrU5 boson-all-0.13.orig/code/ufo/src/ukeystroke.cpp boson-all-0.13/code/ufo/src/ukeystroke.cpp
--- boson-all-0.13.orig/code/ufo/src/ukeystroke.cpp	2005-12-01 00:13:04.000000000 +0100
+++ boson-all-0.13/code/ufo/src/ukeystroke.cpp	2008-05-05 16:52:40.000000000 +0200
@@ -28,10 +28,11 @@
 #include "ufo/ukeystroke.hpp"
 
 #include "ufo/events/ukeyevent.hpp"
 #include "ufo/util/ustring.hpp"
 
+#include <algorithm>
 
 using namespace ufo;
 
 UFO_IMPLEMENT_ABSTRACT_CLASS(UKeyStroke, UObject)
 
diff --exclude='*.orig' -NrU5 boson-all-0.13.orig/code/ufo/src/umenumanager.cpp boson-all-0.13/code/ufo/src/umenumanager.cpp
--- boson-all-0.13.orig/code/ufo/src/umenumanager.cpp	2005-12-01 00:13:04.000000000 +0100
+++ boson-all-0.13/code/ufo/src/umenumanager.cpp	2008-05-05 16:52:40.000000000 +0200
@@ -36,10 +36,12 @@
 #include "ufo/events/umouseevent.hpp"
 #include "ufo/events/ukeyevent.hpp"
 #include "ufo/widgets/upopupmenu.hpp"
 #include "ufo/widgets/urootpane.hpp"
 
+#include <algorithm>
+
 using namespace ufo;
 
 UFO_IMPLEMENT_DYNAMIC_CLASS(UMenuManager, UObject)
 
 UMenuManager * UMenuManager::sm_menuManager = new UMenuManager();
diff --exclude='*.orig' -NrU5 boson-all-0.13.orig/code/ufo/src/uobject.cpp boson-all-0.13/code/ufo/src/uobject.cpp
--- boson-all-0.13.orig/code/ufo/src/uobject.cpp	2005-12-01 00:13:04.000000000 +0100
+++ boson-all-0.13/code/ufo/src/uobject.cpp	2008-05-05 16:52:40.000000000 +0200
@@ -28,10 +28,12 @@
 #include "ufo/uobject.hpp"
 
 #include "ufo/signals/ufo_signals.hpp"
 #include "ufo/ufo_types.hpp"
 
+#include <algorithm>
+
 using namespace ufo;
 
 #ifdef UFO_RTTI
 
 //
diff --exclude='*.orig' -NrU5 boson-all-0.13.orig/code/ufo/src/ux/uxdisplay.cpp boson-all-0.13/code/ufo/src/ux/uxdisplay.cpp
--- boson-all-0.13.orig/code/ufo/src/ux/uxdisplay.cpp	2005-12-01 00:13:04.000000000 +0100
+++ boson-all-0.13/code/ufo/src/ux/uxdisplay.cpp	2008-05-05 16:52:40.000000000 +0200
@@ -41,10 +41,12 @@
 
 #include "ufo/widgets/urootpane.hpp"
 
 #include "ufo/gl/ugl_driver.hpp"
 
+#include <algorithm>
+
 using namespace ufo;
 
 UFO_IMPLEMENT_DYNAMIC_CLASS(UXDisplay, UAbstractDisplay)
 
 class UXDummyDriver : public UVideoDriver {
diff --exclude='*.orig' -NrU5 boson-all-0.13.orig/code/ufo/src/widgets/uwidget.cpp boson-all-0.13/code/ufo/src/widgets/uwidget.cpp
--- boson-all-0.13.orig/code/ufo/src/widgets/uwidget.cpp	2005-12-01 00:13:04.000000000 +0100
+++ boson-all-0.13/code/ufo/src/widgets/uwidget.cpp	2008-05-05 16:52:40.000000000 +0200
@@ -67,10 +67,12 @@
 
 #include "ufo/ugraphics.hpp"
 
 #include "ufo/umodel.hpp"
 
+#include <algorithm>
+
 using namespace ufo;
 
 class PaintNotifier
 {
 public:
