--- code/boson/info/boinfo_linux.cpp
+++ code/boson/info/boinfo_linux.cpp
@@ -27,7 +27,6 @@
 #include <qregexp.h>
 #include <qstringlist.h>
 
-#include <linux/version.h>
 
 QString readFile(const char* fileName)
 {
@@ -72,7 +71,6 @@
 	insert(OSVersionString, kernelVersion);
  }
 
- kernelVersion = UTS_RELEASE;
  insert(BoInfo::OSVersionString + BoInfo::CompileOffset, kernelVersion);
 
  QString kernelModules = readFile("/proc/modules");

