--- speechd-0.9.1/configure.ac	2019-05-10 10:40:53.000000000 +0200
+++ speechd-0.9.1.new/configure.ac	2019-05-14 20:16:26.786785467 +0200
@@ -150,7 +150,7 @@
 			[AC_MSG_FAILURE([python 3 or greater is not available])])])])
 AM_CONDITIONAL([HAVE_PYTHON], [test $enable_python = "yes"])
 
-output_modules="cicero dummy festival generic"
+output_modules="dummy festival generic"
 # checks for output modules
 # check for espeak support
 AC_ARG_WITH([espeak],
--- speechd-0.9.1/src/modules/Makefile.am	2019-05-10 10:40:53.000000000 +0200
+++ speechd-0.9.1.new/src/modules/Makefile.am	2019-05-14 20:16:41.303784996 +0200
@@ -34,7 +34,7 @@
 	$(DOTCONF_CFLAGS) $(GLIB_CFLAGS) $(GTHREAD_CFLAGS) \
 	$(ibmtts_include) $(SNDFILE_CFLAGS)
 
-modulebin_PROGRAMS = sd_dummy sd_generic sd_festival sd_cicero
+modulebin_PROGRAMS = sd_dummy sd_generic sd_festival 
 
 sd_dummy_SOURCES = dummy.c $(audio_SOURCES) $(common_SOURCES) \
 	module_utils_addvoice.c
@@ -55,11 +55,6 @@
 	$(audio_dlopen_modules) \
 	$(common_LDADD)
 
-sd_cicero_SOURCES = cicero.c $(audio_SOURCES) $(common_SOURCES)
-sd_cicero_LDADD = $(top_builddir)/src/common/libcommon.la \
-	$(audio_dlopen_modules) \
-	$(common_LDADD)
-
 if flite_support
 modulebin_PROGRAMS += sd_flite
 sd_flite_SOURCES = flite.c $(audio_SOURCES) $(common_SOURCES)
