--- ConsoleKit-0.3.0/data/ConsoleKitold.conf	2008-04-19 07:39:49.000000000 +0200
+++ ConsoleKit-0.3.0/data/ConsoleKit.conf	2009-05-27 11:49:34.000000000 +0200
@@ -31,6 +31,9 @@
            send_member="CloseSession"/>
     <allow send_interface="org.freedesktop.ConsoleKit.Manager"
            send_member="GetSeats"/>
+    <allow send_destination="org.freedesktop.ConsoleKit"
+           send_interface="org.freedesktop.ConsoleKit.Manager"
+           send_member="GetSessions"/>
     <allow send_interface="org.freedesktop.ConsoleKit.Manager"
            send_member="GetSessionForCookie"/>
     <allow send_interface="org.freedesktop.ConsoleKit.Manager"
@@ -45,6 +48,7 @@
            send_member="GetSystemIdleHint"/>
     <allow send_interface="org.freedesktop.ConsoleKit.Manager"
            send_member="GetSystemIdleSinceHint"/>
+   
 
     <allow send_interface="org.freedesktop.ConsoleKit.Seat"
            send_member="GetId"/>
@@ -91,6 +95,8 @@
            send_member="GetIdleHint"/>
     <allow send_interface="org.freedesktop.ConsoleKit.Session"
            send_member="GetIdleSinceHint"/>
+    <allow send_interface="org.freedesktop.ConsoleKit.Session"
+           send_member="SetIdleHint"/>
   </policy>
 
 </busconfig>
