--- mpeg2dec-0.4.0b.orig/libmpeg2/configure.incl
+++ mpeg2dec-0.4.0b/libmpeg2/configure.incl
@@ -1,7 +1,12 @@
 AC_SUBST([LIBMPEG2_CFLAGS])
 
 dnl avoid -fPIC when possible
-AC_LIBTOOL_NON_PIC([LIBMPEG2_CFLAGS="$LIBMPEG2_CFLAGS -prefer-non-pic"])
+case "$host" in
+    i?86-* | k?-* )
+        AC_LIBTOOL_NON_PIC([LIBMPEG2_CFLAGS="$LIBMPEG2_CFLAGS -prefer-non-pic"])
+    ;;
+esac
+ 
 
 dnl check for cpudetect
 AC_ARG_ENABLE([accel-detect],

