diff -Naur builds/cuda_cupti/extras/CUPTI/samples/activity_trace_async/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/activity_trace_async/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/activity_trace_async/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/activity_trace_async/Makefile	2022-11-07 17:18:56.643852476 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/autorange_profiling/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/autorange_profiling/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/autorange_profiling/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/autorange_profiling/Makefile	2022-11-07 17:18:56.622853654 +0100
@@ -15,7 +15,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/callback_event/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/callback_event/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/callback_event/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/callback_event/Makefile	2022-11-07 17:18:56.675850680 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/callback_metric/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/callback_metric/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/callback_metric/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/callback_metric/Makefile	2022-11-07 17:18:56.678850512 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/callback_profiling/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/callback_profiling/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/callback_profiling/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/callback_profiling/Makefile	2022-11-07 17:18:56.728847706 +0100
@@ -15,7 +15,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/callback_timestamp/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/callback_timestamp/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/callback_timestamp/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/callback_timestamp/Makefile	2022-11-07 17:18:56.635852925 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS),Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/checkpoint_kernels/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/checkpoint_kernels/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/checkpoint_kernels/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/checkpoint_kernels/Makefile	2022-11-07 17:18:56.664851297 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/concurrent_profiling/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/concurrent_profiling/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/concurrent_profiling/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/concurrent_profiling/Makefile	2022-11-07 17:18:56.693849670 +0100
@@ -15,7 +15,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/cuda_graphs_trace/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/cuda_graphs_trace/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/cuda_graphs_trace/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/cuda_graphs_trace/Makefile	2022-11-07 17:18:56.671850905 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/cuda_memory_trace/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/cuda_memory_trace/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/cuda_memory_trace/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/cuda_memory_trace/Makefile	2022-11-07 17:18:56.710848716 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/cupti_correlation/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/cupti_correlation/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/cupti_correlation/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/cupti_correlation/Makefile	2022-11-07 17:18:56.632853093 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/cupti_external_correlation/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/cupti_external_correlation/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/cupti_external_correlation/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/cupti_external_correlation/Makefile	2022-11-07 17:18:56.653851915 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/cupti_metric_properties/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/cupti_metric_properties/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/cupti_metric_properties/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/cupti_metric_properties/Makefile	2022-11-07 17:18:56.721848099 +0100
@@ -15,7 +15,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/cupti_nvtx/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/cupti_nvtx/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/cupti_nvtx/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/cupti_nvtx/Makefile	2022-11-07 17:18:56.697849446 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/cupti_query/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/cupti_query/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/cupti_query/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/cupti_query/Makefile	2022-11-07 17:18:56.656851747 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/event_multi_gpu/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/event_multi_gpu/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/event_multi_gpu/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/event_multi_gpu/Makefile	2022-11-07 17:18:56.717848323 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS),Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/event_sampling/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/event_sampling/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/event_sampling/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/event_sampling/Makefile	2022-11-07 17:18:56.705848996 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS),Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/extensions/src/profilerhost_util/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/extensions/src/profilerhost_util/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/extensions/src/profilerhost_util/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/extensions/src/profilerhost_util/Makefile	2022-11-07 17:18:56.667851129 +0100
@@ -12,7 +12,7 @@
 TARGET_OS ?= linux
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/nested_range_profiling/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/nested_range_profiling/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/nested_range_profiling/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/nested_range_profiling/Makefile	2022-11-07 17:18:56.660851522 +0100
@@ -15,7 +15,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/nvlink_bandwidth/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/nvlink_bandwidth/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/nvlink_bandwidth/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/nvlink_bandwidth/Makefile	2022-11-07 17:18:56.701849221 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS),Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/profiling_injection/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/profiling_injection/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/profiling_injection/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/profiling_injection/Makefile	2022-11-07 17:18:56.725847874 +0100
@@ -15,7 +15,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 NVCCFLAGS :=
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/sass_source_map/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/sass_source_map/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/sass_source_map/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/sass_source_map/Makefile	2022-11-07 17:18:56.628853317 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS),Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/unified_memory/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/unified_memory/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/unified_memory/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/unified_memory/Makefile	2022-11-07 17:18:56.649852139 +0100
@@ -14,7 +14,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS),Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_cupti/extras/CUPTI/samples/userrange_profiling/Makefile builds.new/cuda_cupti/extras/CUPTI/samples/userrange_profiling/Makefile
--- builds/cuda_cupti/extras/CUPTI/samples/userrange_profiling/Makefile	2022-09-29 20:18:34.000000000 +0200
+++ builds.new/cuda_cupti/extras/CUPTI/samples/userrange_profiling/Makefile	2022-11-07 17:18:56.639852700 +0100
@@ -15,7 +15,7 @@
 TARGET_OS ?= $(shell uname | tr A-Z a-z)
 
 # Set required library paths.
-# In the case of cross-compilation, set the libs to the correct ones under /usr/local/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
+# In the case of cross-compilation, set the libs to the correct ones under /opt/cuda/targets/<TARGET_ARCH>-<TARGET_OS>/lib
 
 ifeq ($(OS), Windows_NT)
     LIB_PATH ?= ..\..\lib64
diff -Naur builds/cuda_nvml_dev/nvml/example/Makefile builds.new/cuda_nvml_dev/nvml/example/Makefile
--- builds/cuda_nvml_dev/nvml/example/Makefile	2022-09-29 20:18:35.000000000 +0200
+++ builds.new/cuda_nvml_dev/nvml/example/Makefile	2022-11-07 17:18:56.608854439 +0100
@@ -9,7 +9,7 @@
 DRIVER_BRANCH := $(shell nvidia-smi | grep Driver | cut -f 3 -d' ' | cut -f 1 -d '.')
 
 # Location of the CUDA Toolkit
-CUDA_PATH ?= "/usr/local/cuda-8.0"
+CUDA_PATH ?= "/opt/cuda-8.0"
 
 ifeq (${ARCH},$(filter ${ARCH},32 64))
     # If correct architecture and libnvidia-ml library is not found 
diff -Naur builds/libcufile/gds/samples/Makefile builds.new/libcufile/gds/samples/Makefile
--- builds/libcufile/gds/samples/Makefile	2022-09-29 20:18:45.000000000 +0200
+++ builds.new/libcufile/gds/samples/Makefile	2022-11-07 17:18:56.732847481 +0100
@@ -34,9 +34,9 @@
 ################################################################################
 
 # Common includes and paths for CUDA
-CUDA_PATH   := /usr/local/cuda
-CUFILE_PATH ?= /usr/local/cuda/targets/x86_64-linux/lib/
-CUFILE_INCLUDE_PATH ?= /usr/local/cuda/targets/x86_64-linux/include/
+CUDA_PATH   := /opt/cuda
+CUFILE_PATH ?= /opt/cuda/targets/x86_64-linux/lib/
+CUFILE_INCLUDE_PATH ?= /opt/cuda/targets/x86_64-linux/include/
 CXXFLAGS    := -Wall
 CXXFLAGS    += -I $(CUDA_PATH)/include/ 
 CXXFLAGS    += -I $(CUFILE_INCLUDE_PATH)
