diff -urN gpm-1.20.1.orig/src/lib/liblow.c gpm-1.20.1/src/lib/liblow.c
--- gpm-1.20.1.orig/src/lib/liblow.c	2002-12-30 23:13:37.000000000 +0100
+++ gpm-1.20.1/src/lib/liblow.c	2002-12-30 23:56:54.000000000 +0100
@@ -199,7 +199,7 @@
    Gpm_Stst *new = NULL;
    char* sock_name = 0;
 
-   option.consolename = NULL;
+   option.consolename;
 
    /*....................................... First of all, check xterm */
 
