Avoid useless debug info under certain circumstances.

--- gpm-1.20.1/src/lib/report-lib.c.orig	Sat Feb 23 16:42:23 2002
+++ gpm-1.20.1/src/lib/report-lib.c	Fri May  3 20:16:02 2002
@@ -30,10 +30,10 @@
    va_start(ap,text);
 
    switch(stat) {
-      case GPM_STAT_INFO : string = GPM_TEXT_INFO ; break;
+      case GPM_STAT_INFO : string = GPM_TEXT_INFO ; return;
       case GPM_STAT_WARN : string = GPM_TEXT_WARN ; break;
       case GPM_STAT_ERR  : string = GPM_TEXT_ERR  ; break;
-      case GPM_STAT_DEBUG: string = GPM_TEXT_DEBUG; break;
+      case GPM_STAT_DEBUG: string = GPM_TEXT_DEBUG; return;
       case GPM_STAT_OOPS : string = GPM_TEXT_OOPS; break;
    }
    fprintf(stderr,"%s[%s(%d)]:\n",string,file,line);
--- gpm-1.20.0/src/report.c.orig	Sat Feb 23 16:42:23 2002
+++ gpm-1.20.0/src/report.c	Fri May  3 20:12:51 2002
@@ -130,6 +130,7 @@
       case GPM_RUN_DAEMON:
          switch(stat) {
             case GPM_STAT_INFO:
+	       break;
 #ifdef HAVE_VSYSLOG
                syslog(LOG_INFO | LOG_USER, GPM_STRING_INFO);
                vsyslog(LOG_INFO | LOG_USER, text, ap);
--- gpm-1.20.1/src/lib/liblow.c.orig	2004-03-31 15:06:45.000000000 -0500
+++ gpm-1.20.1/src/lib/liblow.c	2004-03-31 15:09:32.000000000 -0500
@@ -202,7 +202,9 @@
 
    consolename = NULL;
 
+#if 0
    gpm_report(GPM_PR_DEBUG,"VC: %d",flag);
+#endif
 
    /*....................................... First of all, check xterm */
 
diff -Naur gpm-1.20.1.orig/src/lib/liblow.c gpm-1.20.1/src/lib/liblow.c
--- gpm-1.20.1.orig/src/lib/liblow.c	2002-12-24 23:57:16.000000000 +0100
+++ gpm-1.20.1/src/lib/liblow.c	2005-11-29 13:36:42.000000000 +0100
@@ -325,8 +325,6 @@
            */
           close(gpm_fd); /* the socket */
           if ((gpm_fd=open(GPM_NODE_DEV,O_RDWR))==-1) {
-            gpm_report(GPM_PR_ERR,GPM_MESS_DOUBLE_S,GPM_NODE_DEV
-                                                   ,strerror(errno));
             goto err;
           }
          if (fstat(gpm_fd,&stbuf)==-1 || (stbuf.st_mode&S_IFMT)!=S_IFCHR) {
@@ -369,7 +367,6 @@
 
   /*....................................... Error: free all memory */
    err:
-   gpm_report(GPM_PR_ERR,"Oh, oh, it's an error! possibly I die! ");
    do {
       new=gpm_stack->next;
       free(gpm_stack);
