diff -Naur glest-source-3.1.2/shared_lib/sources/platform/posix/socket.cpp glest-source-3.1.2.new/shared_lib/sources/platform/posix/socket.cpp
--- glest-source-3.1.2/shared_lib/sources/platform/posix/socket.cpp	2008-02-16 19:04:09.000000000 +0530
+++ glest-source-3.1.2.new/shared_lib/sources/platform/posix/socket.cpp	2008-04-08 19:35:15.000000000 +0530
@@ -10,6 +10,8 @@
 
 #include <stdexcept>
 #include <sstream>
+#include <cstdlib>
+#include <cstring>
 #if defined(HAVE_SYS_IOCTL_H)
 #define BSD_COMP /* needed for FIONREAD on Solaris2 */
 #include <sys/ioctl.h>
diff -Naur glest-source-3.1.2/shared_lib/sources/util/conversion.cpp glest-source-3.1.2.new/shared_lib/sources/util/conversion.cpp
--- glest-source-3.1.2/shared_lib/sources/util/conversion.cpp	2008-01-23 01:40:59.000000000 +0530
+++ glest-source-3.1.2.new/shared_lib/sources/util/conversion.cpp	2008-04-08 19:34:48.000000000 +0530
@@ -13,6 +13,8 @@
 
 #include <stdexcept>
 #include <cstdio>
+#include <cstdlib>
+#include <cstring>
 
 #include "leak_dumper.h"
 
diff -Naur glest-source-3.1.2/shared_lib/sources/util/properties.cpp glest-source-3.1.2.new/shared_lib/sources/util/properties.cpp
--- glest-source-3.1.2/shared_lib/sources/util/properties.cpp	2008-02-08 16:13:47.000000000 +0530
+++ glest-source-3.1.2.new/shared_lib/sources/util/properties.cpp	2008-04-08 19:33:54.000000000 +0530
@@ -13,7 +13,7 @@
 
 #include <fstream>
 #include <stdexcept>
-
+#include <cstring>
 #include "conversion.h"
 #include "leak_dumper.h"
 
diff -Naur glest-source-3.1.2/shared_lib/sources/util/util.cpp glest-source-3.1.2.new/shared_lib/sources/util/util.cpp
--- glest-source-3.1.2/shared_lib/sources/util/util.cpp	2008-01-23 01:40:59.000000000 +0530
+++ glest-source-3.1.2.new/shared_lib/sources/util/util.cpp	2008-04-08 19:34:37.000000000 +0530
@@ -13,6 +13,7 @@
 
 #include <ctime>
 #include <cassert>
+#include <cstring>
 #include <stdexcept>
 
 #include "leak_dumper.h"
