diff -Naur ode-0.5.old/ode/test/test_space.cpp ode-0.5/ode/test/test_space.cpp
--- ode-0.5.old/ode/test/test_space.cpp	2004-04-22 22:38:51.000000000 +0200
+++ ode-0.5/ode/test/test_space.cpp	2006-05-01 22:23:50.000000000 +0200
@@ -124,8 +124,8 @@
 static void nearCallback (void *data, dGeomID o1, dGeomID o2)
 {
   int i,j;
-  i = (int) dGeomGetData (o1);
-  j = (int) dGeomGetData (o2);
+  i = (long) dGeomGetData (o1);
+  j = (long) dGeomGetData (o2);
   if (i==j)
     printf ("collision (%d,%d) is between the same object\n",i,j);
   if (!good_matrix[i][j] || !good_matrix[j][i])
