--- xscreensaver-5.42/hacks/fontglide.c	2018-09-03 00:56:45.000000000 +0200
+++ xscreensaver-5.42.new/hacks/fontglide.c	2018-12-29 23:10:58.132955297 +0100
@@ -2421,8 +2421,6 @@
 
   textclient_close (s->tc);
 
-//  if (s->b && s->b != s->window) XFreePixmap (dpy, s->b);
-//  if (s->ba && s->ba != s->b) XFreePixmap (dpy, s->ba);
   XFreeGC (dpy, s->bg_gc);
   if (s->charset) free (s->charset);
   if (s->font_override) free (s->font_override);
