--- src/src/main.cpp	2006-05-30 16:33:56.000000000 +0530
+++ src/src/main.cpp	2006-05-30 16:34:24.000000000 +0530
@@ -65,7 +65,7 @@
 class wxMyFileDropTarget : public wxFileDropTarget
 {
 public:
-    wxMyFileDropTarget::wxMyFileDropTarget(MainFrame *frame):m_frame(frame){}
+    wxMyFileDropTarget(MainFrame *frame):m_frame(frame){}
     virtual bool OnDropFiles(wxCoord x, wxCoord y, const wxArrayString& filenames)
     {
         if(!m_frame) return false;
