--- ispc-1.16.0/CMakeLists.txt	2021-06-12 02:41:24.000000000 +0200
+++ ispc-1.16.0.new/CMakeLists.txt	2021-07-14 13:39:40.938697512 +0200
@@ -281,7 +281,7 @@
     list(APPEND ISPC_TARGETS wasm-i32x4)
 endif()
 
-set(CLANG_LIBRARY_LIST clangFrontend clangDriver clangSerialization clangParse clangSema clangAnalysis clangAST clangBasic clangEdit clangLex)
+set(CLANG_LIBRARY_LIST clang-cpp)
 set(LLVM_COMPONENTS engine ipo bitreader bitwriter instrumentation linker option frontendopenmp)
 
 if (X86_ENABLED)
