diff -Naur DVDStyler-1.4.orig/src/ButtonPropDlg.cpp DVDStyler-1.4/src/ButtonPropDlg.cpp
--- DVDStyler-1.4.orig/src/ButtonPropDlg.cpp	2005-08-05 12:27:38.000000000 +0200
+++ DVDStyler-1.4/src/ButtonPropDlg.cpp	2005-08-05 12:27:13.000000000 +0200
@@ -115,14 +115,14 @@
 int ButtonPropDlg::GetTsi()
 {
   if (m_titlesetChoice->GetSelection() >= 0)
-	return ((int) m_titlesetChoice->GetClientData(m_titlesetChoice->GetSelection()))-1;
+	return ((long) m_titlesetChoice->GetClientData(m_titlesetChoice->GetSelection()))-1;
   return 0;
 }
 
 int ButtonPropDlg::GetPgci()
 {
   if (m_targetChoice->GetSelection() >= 0)
-	return (int) m_targetChoice->GetClientData(m_targetChoice->GetSelection());
+	return (long) m_targetChoice->GetClientData(m_targetChoice->GetSelection());
   return 0;
 }
 
@@ -296,9 +296,9 @@
 	return false;
   }
   int n = 1;
-  int tsi = ((int)GetClientData(n++))-1;
+  int tsi = ((long)GetClientData(n++))-1;
   m_object->SetActionTsi(tsi != m_tsi ? tsi : -2);
-  m_object->SetActionPgci((int)GetClientData(n++));
+  m_object->SetActionPgci((long)GetClientData(n++));
   m_object->SetActionChapter(GetInt(n++));
   if (GetBool(n++))
 	m_object->SetCustomAction(GetString(n));
