--- vkeybd/Makefile.old	2009-05-14 18:20:38.000000000 +0000
+++ vkeybd/Makefile	2009-05-14 22:58:39.000000000 +0000
@@ -8,7 +8,7 @@
 #
 # installation directory
 #
-PREFIX = /usr/local
+PREFIX = /usr
 # binary and Tcl script are put there
 BIN_DIR = $(PREFIX)/bin
 # man page
@@ -27,7 +27,7 @@
 # device selections -- multiple avaialble
 # to disable the device, set value 0 (do not comment out!)
 #
-USE_AWE = 1
+USE_AWE = 0
 USE_MIDI = 1
 USE_ALSA = 1
 USE_LADCCA = 0
@@ -35,7 +35,7 @@
 #
 # Tcl/Tk library -- depends on your distribution
 #
-TCL_VERSION = 8.4
+TCL_VERSION = 8.5
 TCLLIB = -ltcl$(TCL_VERSION)
 TCLINC =
 TKLIB = -ltk$(TCL_VERSION)
