diff -ur a/sbin/update-ca-certificates b/sbin/update-ca-certificates
--- sbin/update-ca-certificates	2013-06-10 12:29:57.000000000 -0500
+++ sbin/update-ca-certificates	2013-06-25 19:36:50.420975896 -0500
@@ -148,16 +148,5 @@
 
 echo "$ADDED_CNT added, $REMOVED_CNT removed; done."
 
-HOOKSDIR=/etc/ca-certificates/update.d
-echo -n "Running hooks in $HOOKSDIR...."
-VERBOSE_ARG=
-[ "$verbose" = 0 ] || VERBOSE_ARG=--verbose
-eval run-parts $VERBOSE_ARG --test -- $HOOKSDIR | while read hook
-do
-  ( cat $ADDED
-    cat $REMOVED ) | $hook || echo E: $hook exited with code $?.
-done
-echo "done."
-
 # vim:set et sw=2:
 
