diff -ur a/sbin/update-ca-certificates b/sbin/update-ca-certificates
--- a/sbin/update-ca-certificates	2013-09-07 02:40:28.000000000 -0500
+++ b/sbin/update-ca-certificates	2013-10-23 15:58:51.792268327 -0500
@@ -107,6 +107,12 @@
   remove "$CERTSDIR/$crt"
 done
 
+# Remove certificates which have symbolic links leading to nowhere.
+for i in "$ETCCERTSDIR/"*
+do
+  test -L "$i" -a ! -f "$i" && remove "${i%.pem}"
+done
+
 sed -e '/^$/d' -e '/^#/d' -e '/^!/d' $CERTSCONF | while read crt
 do
   if ! test -f "$CERTSDIR/$crt"
