--- ca-certificates/sbin/update-ca-certificates	2018-04-10 01:43:49.000000000 +0200
+++ ca-certificates.new/sbin/update-ca-certificates	2018-10-23 11:42:11.028798591 +0200
@@ -174,9 +174,9 @@
   # only run if set of files has changed
   if [ "$verbose" = 0 ]
   then
-    openssl rehash . > /dev/null
+    openssl-c_rehash.sh . > /dev/null
   else
-    openssl rehash .
+    openssl-c_rehash.sh .
   fi
 fi
 
