diff -uNr fwbuilder-2.1.14-orig/src/antlr/CharScanner.hpp fwbuilder-2.1.14/src/antlr/CharScanner.hpp
--- fwbuilder-2.1.14-orig/src/antlr/CharScanner.hpp	2007-06-07 04:33:56.000000000 +0200
+++ fwbuilder-2.1.14/src/antlr/CharScanner.hpp	2008-01-03 16:00:52.000000000 +0100
@@ -11,6 +11,7 @@
 #include <antlr/config.hpp>
 
 #include <map>
+#include <cstring>
 
 #ifdef HAS_NOT_CCTYPE_H
 #include <ctype.h>
diff -uNr fwbuilder-2.1.14-orig/src/ipt/NATCompiler_ipt.cpp fwbuilder-2.1.14/src/ipt/NATCompiler_ipt.cpp
--- fwbuilder-2.1.14-orig/src/ipt/NATCompiler_ipt.cpp	2007-01-15 02:21:58.000000000 +0100
+++ fwbuilder-2.1.14/src/ipt/NATCompiler_ipt.cpp	2008-01-03 16:11:36.000000000 +0100
@@ -53,6 +53,7 @@
 #include <functional>
 #include <iomanip>
 #include <iostream>
+#include <cstring>
 
 #include <assert.h>
 #include <ctype.h>
diff -uNr fwbuilder-2.1.14-orig/src/ipt/PolicyCompiler_PrintRule.cpp fwbuilder-2.1.14/src/ipt/PolicyCompiler_PrintRule.cpp
--- fwbuilder-2.1.14-orig/src/ipt/PolicyCompiler_PrintRule.cpp	2007-07-19 03:46:30.000000000 +0200
+++ fwbuilder-2.1.14/src/ipt/PolicyCompiler_PrintRule.cpp	2008-01-03 16:11:18.000000000 +0100
@@ -60,6 +60,7 @@
 #include <iomanip>
 #include <fstream>
 #include <sstream>
+#include <cstring>
 
 #include <assert.h>

diff -uNr fwbuilder-2.1.18.orig/src/cisco_lib/PolicyCompiler_cisco.cpp fwbuilder-2.1.18/src/cisco_lib/PolicyCompiler_cisco.cpp
--- fwbuilder-2.1.18.orig/src/cisco_lib/PolicyCompiler_cisco.cpp        2008-03-06 07:48:56.000000000 +0100
+++ fwbuilder-2.1.18/src/cisco_lib/PolicyCompiler_cisco.cpp     2008-04-19 14:15:41.000000000 +0200
@@ -41,6 +41,7 @@
 #include "fwbuilder/AddressTable.h"

 #include <iostream>
+#include <cstring>
 #if __GNUC__ > 3 || \
     (__GNUC__ == 3 && (__GNUC_MINOR__ > 2 || (__GNUC_MINOR__ == 2 ) ) ) || \
      _MSC_VER
diff -uNr fwbuilder-2.1.18.orig/src/pix/NATCompiler_pix.cpp fwbuilder-2.1.18/src/pix/NATCompiler_pix.cpp
--- fwbuilder-2.1.18.orig/src/pix/NATCompiler_pix.cpp   2008-03-06 07:48:59.000000000 +0100
+++ fwbuilder-2.1.18/src/pix/NATCompiler_pix.cpp        2008-04-19 14:23:22.000000000 +0200
@@ -44,6 +44,7 @@
 #include <algorithm>
 #include <functional>
 #include <iostream>
+#include <cstring>
 #include <assert.h>

 using namespace libfwbuilder;
