diff -aur gmpc-serverstats-0.17.0/src/plugin.c gmpc-serverstats-0.17.0-2/src/plugin.c
--- gmpc-serverstats-0.17.0/src/plugin.c	2008-12-25 15:58:59.000000000 -0600
+++ gmpc-serverstats-0.17.0-2/src/plugin.c	2010-05-16 02:12:21.000000000 -0500
@@ -120,8 +120,8 @@
 			PL3_CAT_TITLE,"Server Statistics", 
 			PL3_CAT_INT_ID, "/",
 			PL3_CAT_ICON_ID, "mpd",
-			PL3_CAT_PROC, TRUE,
-			PL3_CAT_ICON_SIZE,GTK_ICON_SIZE_DND,-1);
+			PL3_CAT_PROC_DEPRECATED, TRUE,
+			PL3_CAT_ICON_SIZE_DEPRECATED,GTK_ICON_SIZE_DND,-1);
 	/** 
 	 * remove odl reference if exists 
 	 */
@@ -146,7 +146,7 @@
 		if(path)
 		{
 			gint *indices = gtk_tree_path_get_indices(path);
-			debug_printf(DEBUG_INFO,"Saving myself to position: %i\n", indices[0]);
+			//debug_printf(DEBUG_INFO,"Saving myself to position: %i\n", indices[0]);
 			cfg_set_single_value_as_int(config, "serverstats","position",indices[0]);
 			gtk_tree_path_free(path);
 		}
@@ -627,7 +627,7 @@
 {
     gchar *path = gmpc_plugin_get_data_path(&plugin);
     gchar *url = g_build_path(G_DIR_SEPARATOR_S,path, "serverstats", NULL);
-    debug_printf(DEBUG_WARNING,"Found url: %s\n", url);
+    //debug_printf(DEBUG_WARNING,"Found url: %s\n", url);
     /*
     gchar *path = NULL;
 
