diff -Naur libtool-2.4.6/m4/gnulib-common.m4 libtool-2.4.6-p/m4/gnulib-common.m4
--- libtool-2.4.6/m4/gnulib-common.m4	2015-01-16 19:32:34.000000000 +0100
+++ libtool-2.4.6-p/m4/gnulib-common.m4	2018-01-27 20:45:04.027479126 +0100
@@ -278,12 +278,12 @@
       dnl but prefer ${host}-ar over ar (useful for cross-compiling).
       AC_CHECK_TOOL([AR], [ar], [ar])
       if test -z "$ARFLAGS"; then
-        ARFLAGS='cru'
+        ARFLAGS='crD'
       fi
     fi
   else
     if test -z "$ARFLAGS"; then
-      ARFLAGS='cru'
+      ARFLAGS='crD'
     fi
   fi
   AC_SUBST([AR])
diff -Naur libtool-2.4.6/m4/libtool.m4 libtool-2.4.6-p/m4/libtool.m4
--- libtool-2.4.6/m4/libtool.m4	2015-01-20 17:15:19.000000000 +0100
+++ libtool-2.4.6-p/m4/libtool.m4	2018-01-27 20:45:18.566489313 +0100
@@ -1042,8 +1042,8 @@
 _LT_EOF
       echo "$LTCC $LTCFLAGS -c -o conftest.o conftest.c" >&AS_MESSAGE_LOG_FD
       $LTCC $LTCFLAGS -c -o conftest.o conftest.c 2>&AS_MESSAGE_LOG_FD
-      echo "$AR cru libconftest.a conftest.o" >&AS_MESSAGE_LOG_FD
-      $AR cru libconftest.a conftest.o 2>&AS_MESSAGE_LOG_FD
+      echo "$AR crD libconftest.a conftest.o" >&AS_MESSAGE_LOG_FD
+      $AR crD libconftest.a conftest.o 2>&AS_MESSAGE_LOG_FD
       echo "$RANLIB libconftest.a" >&AS_MESSAGE_LOG_FD
       $RANLIB libconftest.a 2>&AS_MESSAGE_LOG_FD
       cat > conftest.c << _LT_EOF
@@ -1493,7 +1493,7 @@
 m4_defun([_LT_PROG_AR],
 [AC_CHECK_TOOLS(AR, [ar], false)
 : ${AR=ar}
-: ${AR_FLAGS=cru}
+: ${AR_FLAGS=crD}
 _LT_DECL([], [AR], [1], [The archiver])
 _LT_DECL([], [AR_FLAGS], [1], [Flags to create an archive])
 
