diff -Naur xbsql-hk_classes-0.13/xbsql/xbsql.l xbsql-hk_classes-0.13.new/xbsql/xbsql.l
--- xbsql-hk_classes-0.13/xbsql/xbsql.l	2005-04-19 19:17:50.000000000 +0200
+++ xbsql-hk_classes-0.13.new/xbsql/xbsql.l	2012-06-05 21:54:14.000000000 +0200
@@ -12,12 +12,12 @@
 %%
 
 \'			{
-			string	('\'')		;
+			String	('\'')		;
 			return	T_STRING	;
 			}
 
 \"			{
-			  string('\"');
+			  String('\"');
                           return T_NAME;
 			}
 
@@ -131,7 +131,7 @@
 #define	HASHMSK	(HASHSIZ-1)
 static	KEYWORD	*hashtab[HASHSIZ] ;
 
-static	void	string
+static	void	String
 	(	int	qch
 	)
 {
diff -Naur xbsql-hk_classes-0.13/xbsql/xbsql.y xbsql-hk_classes-0.13.new/xbsql/xbsql.y
--- xbsql-hk_classes-0.13/xbsql/xbsql.y	2005-04-19 19:17:50.000000000 +0200
+++ xbsql-hk_classes-0.13.new/xbsql/xbsql.y	2012-06-05 21:55:06.000000000 +0200
@@ -13,7 +13,7 @@
 
 extern	void		yyerror	(const char *, ...) ;
 extern	int		yylex	()	;
-static	void		string	(int)	;
+static	void		String	(int)	;
 static	int		nextChar()	;
 static	int		nextData(char *,  int, int) ;
 static	char		*textp		;
