diff -urN archive-1.9.5/Archive/formats.py archive-1.9.5_fw/Archive/formats.py
--- archive-1.9.5/Archive/formats.py	2005-04-17 17:29:31.000000000 +0200
+++ archive-1.9.5_fw/Archive/formats.py	2005-08-25 17:29:10.000000000 +0200
@@ -137,7 +137,7 @@
 tgz = Extract('tgz',	"gunzip -c - | tar xf -")
 tbz = Extract('tar.bz2', "bunzip2 -c - | tar xf -")
 tarz = Extract('tar.Z', "uncompress -c - | tar xf -")
-rar = Extract('rar',	"rar x '%s'")
+rar = Extract('rar',	"unrar x '%s'")
 ace = Extract('ace',   "unace x '%s'")
 tar = Extract('tar',	"tar xf -")
 rpm = Extract('rpm',	"rpm2cpio - | cpio -id --quiet")
@@ -267,12 +267,12 @@
 		
 		# Compressed streams
 		if string(0, '\037\213'):
-			if self.path.endswith('.tar.gz') or self.path.endswith('.tgz'):
+			if self.path.endswith('.tar.gz') or self.path.endswith('.tgz') or self.path.endswith('.fpm'):
 				return tgz
 			return gz
 		if string(0, 'BZh') or string(0, 'BZ'):
 			if self.path.endswith('.tar.bz') or self.path.endswith('.tar.bz2') or \
-			   self.path.endswith('.tbz') or self.path.endswith('.tbz2'):
+			   self.path.endswith('.tbz') or self.path.endswith('.tbz2') or self.path.endswith('.fpm'):
 				return tbz
 			return bz2
 		if string(0, 'begin '):
