See http://bugs.frugalware.org/2945.

diff -Naur java-gcj-compat-1.0.78.orig/Makefile.am java-gcj-compat-1.0.78/Makefile.am
--- java-gcj-compat-1.0.78.orig/Makefile.am	2007-06-26 21:51:19.000000000 +0200
+++ java-gcj-compat-1.0.78/Makefile.am	2008-04-09 21:41:44.000000000 +0200
@@ -19,6 +19,7 @@
 	ln -sf ../../../../../bin/gij$(gcc_suffix) $(DESTDIR)$(JRE_BIN_DIR)/java
 	ln -sf ../../../../bin/fastjar$(gcc_suffix) $(DESTDIR)$(SDK_BIN_DIR)/jar
 	ln -sf ../../../../bin/sinjdoc $(DESTDIR)$(SDK_BIN_DIR)/javadoc
+	ln -sf ../../../../bin/gnative2ascii $(DESTDIR)$(SDK_BIN_DIR)/native2ascii
 	ln -sf ../../../../bin/grmic$(gcc_suffix) $(DESTDIR)$(SDK_BIN_DIR)/rmic
 	ln -sf ../../../../bin/gjavah$(gcc_suffix) $(DESTDIR)$(SDK_BIN_DIR)/javah
 	ln -sf ../../../../bin/ecj $(DESTDIR)$(SDK_BIN_DIR)/javac
@@ -35,6 +36,7 @@
 	ln -sf $(SDK_BIN_DIR)/appletviewer $(DESTDIR)$(prefix)/bin/appletviewer
 	ln -sf $(SDK_BIN_DIR)/jarsigner $(DESTDIR)$(prefix)/bin/jarsigner
 	ln -sf $(SDK_BIN_DIR)/javadoc $(DESTDIR)$(prefix)/bin/javadoc
+	ln -sf $(SDK_BIN_DIR)/native2ascii $(DESTDIR)$(prefix)/bin/native2ascii
 	ln -sf $(SDK_BIN_DIR)/javah $(DESTDIR)$(prefix)/bin/javah
 	ln -sf $(SDK_BIN_DIR)/rmic $(DESTDIR)$(prefix)/bin/rmic
 	ln -sf $(JRE_BIN_DIR)/java $(DESTDIR)$(prefix)/bin/java
