diff -Naur java-gcj-compat-1.0.77/Makefile.am java-gcj-compat-1.0.77-p/Makefile.am
--- java-gcj-compat-1.0.77/Makefile.am	2007-06-26 21:51:19.000000000 +0200
+++ java-gcj-compat-1.0.77-p/Makefile.am	2008-03-08 20:43:21.000000000 +0100
@@ -17,7 +17,7 @@
 	$(mkinstalldirs) $(DESTDIR)$(SDK_INCLUDE_DIR)/$(OS)
 	ln -sf ../../../../bin/gij$(gcc_suffix) $(DESTDIR)$(SDK_BIN_DIR)/java
 	ln -sf ../../../../../bin/gij$(gcc_suffix) $(DESTDIR)$(JRE_BIN_DIR)/java
-	ln -sf ../../../../bin/fastjar$(gcc_suffix) $(DESTDIR)$(SDK_BIN_DIR)/jar
+	ln -sf ../../../../bin/gjar$(gcc_suffix) $(DESTDIR)$(SDK_BIN_DIR)/jar
 	ln -sf ../../../../bin/sinjdoc $(DESTDIR)$(SDK_BIN_DIR)/javadoc
 	ln -sf ../../../../bin/grmic$(gcc_suffix) $(DESTDIR)$(SDK_BIN_DIR)/rmic
 	ln -sf ../../../../bin/gjavah$(gcc_suffix) $(DESTDIR)$(SDK_BIN_DIR)/javah
