diff -Naur bluez-hcidump-1.42/parser/hci.c bluez-hcidump-1.42-p/parser/hci.c
--- bluez-hcidump-1.42/parser/hci.c	2008-06-15 14:46:13.000000000 +0200
+++ bluez-hcidump-1.42-p/parser/hci.c	2010-02-13 20:06:28.000000000 +0100
@@ -1973,7 +1973,7 @@
 
 static inline void read_inquiry_transmit_power_level_dump(int level, struct frame *frm)
 {
-	read_inquiry_transmit_power_level_rp *rp = frm->ptr;
+	read_inq_response_tx_power_level_rp *rp = frm->ptr;
 
 	p_indent(level, frm);
 	printf("status 0x%2.2x level %d\n", rp->status, rp->level);
@@ -2325,7 +2325,7 @@
 		case OCF_READ_EXT_INQUIRY_RESPONSE:
 			read_ext_inquiry_response_dump(level, frm);
 			return;
-		case OCF_READ_INQUIRY_TRANSMIT_POWER_LEVEL:
+		case OCF_READ_INQ_RESPONSE_TX_POWER_LEVEL:
 			read_inquiry_transmit_power_level_dump(level, frm);
 			return;
 		case OCF_READ_DEFAULT_ERROR_DATA_REPORTING:
