Index: xen-unstable/tools/vncfb/Makefile
===================================================================
--- xen-unstable.orig/tools/vncfb/Makefile
+++ xen-unstable/tools/vncfb/Makefile
@@ -1,14 +1,24 @@
-CFLAGS += -g -Wall
+XEN_ROOT=../..
+include $(XEN_ROOT)/tools/Rules.mk
 
-ifneq ($(XENDIR),)
-CFLAGS += -I$(XENDIR)/tools/libxc -I$(XENDIR)/linux-2.6.12-xenU/include
-LDFLAGS += -L$(XENDIR)/tools/libxc
-endif
+CFLAGS += -I$(XEN_LIBXC) -I$(XEN_XENSTORE) -I$(XEN_ROOT)/linux-2.6-xen-sparse/include
+LDFLAGS += -L$(XEN_LIBXC) -L$(XEN_XENSTORE)
 
-all: vncfb sdlfb
+INSTALL         = install
+INSTALL_PROG    = $(INSTALL) -m0755
+INSTALL_DIR     = $(INSTALL) -d -m0755
+
+.PHONY: all
+all: build
+
+.PHONY: build
+build: mk-symlinks
+	$(MAKE) vncfb sdlfb
 
 install: all
-	cp vncfb /usr/lib/xen/bin/vncfb
+	$(INSTALL_DIR) -p $(DESTDIR)/usr/$(LIBDIR)/xen/bin
+	$(INSTALL_PROG) vncfb $(DESTDIR)/usr/$(LIBDIR)/xen/bin
+	$(INSTALL_PROG) sdlfb $(DESTDIR)/usr/$(LIBDIR)/xen/bin
 
 sdlfb: sdlfb.o xenfb.o
 
Index: xen-unstable/tools/Makefile
===================================================================
--- xen-unstable.orig/tools/Makefile
+++ xen-unstable/tools/Makefile
@@ -18,6 +18,7 @@ SUBDIRS-$(VTPM_TOOLS) += vtpm
 SUBDIRS-y += xenstat
 SUBDIRS-y += libaio
 SUBDIRS-y += blktap
+SUBDIRS-y += vncfb
 
 # These don't cross-compile
 ifeq ($(XEN_COMPILE_ARCH),$(XEN_TARGET_ARCH))
