Index: xen-unstable/tools/python/xen/xm/create.py
===================================================================
--- xen-unstable.orig/tools/python/xen/xm/create.py
+++ xen-unstable/tools/python/xen/xm/create.py
@@ -808,9 +808,8 @@ def preprocess_access_control(vals):
 
 def preprocess_ip(vals):
     if vals.ip or vals.dhcp != 'off':
-        dummy_nfs_server = '1.2.3.4'
         ip = (vals.ip
-          + ':' + (vals.nfs_server or dummy_nfs_server)
+          + ':' + (vals.nfs_server or '')
           + ':' + vals.gateway
           + ':' + vals.netmask
           + ':' + vals.hostname
