Index: xen-unstable/tools/python/xen/xm/create.py
===================================================================
--- xen-unstable.orig/tools/python/xen/xm/create.py
+++ xen-unstable/tools/python/xen/xm/create.py
@@ -902,8 +902,7 @@ def spawn_vnc(display):
     returns the port that the vncviewer is listening on and sets the global
     vncpid.  On failure, returns 0.  Note that vncviewer is daemonized.
     """
-    vncargs = (["vncviewer", "-log", "*:stdout:0",
-            "-listen", "%d" % (VNC_BASE_PORT + display) ])
+    vncargs = (["vncviewer", "-listen", "%d" % (display) ])
     global vncpid
     vncpid = daemonize("vncviewer", vncargs)
     if vncpid == 0:
