Index: xen-unstable/tools/python/xen/xm/shutdown.py
===================================================================
--- xen-unstable.orig/tools/python/xen/xm/shutdown.py
+++ xen-unstable/tools/python/xen/xm/shutdown.py
@@ -67,6 +67,8 @@ def wait_reboot(opts, doms, rcs):
     opts.info("All domains rebooted")
 
 def wait_shutdown(opts, doms):
+    from xen.xend.xenstore.xstransact import xstransact
+    doms_to_cleanup = doms[:]
     while doms:
         alive = server.xend.domains(0)
         dead = []
@@ -77,6 +79,17 @@ def wait_shutdown(opts, doms):
             opts.info("Domain %s terminated" % d)
             doms.remove(d)
         time.sleep(1)
+        # Now all the domains are terminated, but wait until the devices are
+        # cleaned up.
+        for d in doms_to_cleanup:
+            info = server.xend.domain(d)
+            domid = int(sxp.child_value(info, 'domid', '-1'))
+            device_class_path = '/local/domain/0/backend/vbd/%d/' % domid
+            while True:
+                devices = xstransact.List(device_class_path)
+                if len(devices) == 0:
+                    break
+                time.sleep(1)
     opts.info("All domains terminated")
 
 def shutdown(opts, doms, mode, wait):
