Index: xen-unstable/tools/misc/xend
===================================================================
--- xen-unstable.orig/tools/misc/xend
+++ xen-unstable/tools/misc/xend
@@ -64,14 +64,7 @@ def check_logging():
     try:
         import logging
     except ImportError:
-        hline()
-        msg("Python logging is not installed.")
-        msg("Use 'make install-logging' at the xen root to install.")
-        msg("")
-        msg("Alternatively download and install from")
-        msg("http://www.red-dove.com/python_logging.html")
-        hline()
-        raise CheckError("logging is not installed")
+        raise CheckError("logging is not installed; try Python 2.3 or later")
 
 def check_user():
     """Check that the effective user id is 0 (root).
Index: xen-unstable/tools/python/xen/xend/XendDomain.py
===================================================================
--- xen-unstable.orig/tools/python/xen/xend/XendDomain.py
+++ xen-unstable/tools/python/xen/xend/XendDomain.py
@@ -467,6 +467,8 @@ class XendDomain:
 
             if dominfo.getDomid() == PRIV_DOMAIN:
                 raise XendError("Cannot save privileged domain %s" % domid)
+            if dominfo.readVm('image/ostype') == "hvm":
+                raise XendError("Cannot save fully virtualized domains")
 
             fd = os.open(dst, os.O_WRONLY | os.O_CREAT | os.O_TRUNC)
             try:
