--- xen-unstable.hg/tools/examples/xend-config.sxp.http	2006-04-25 21:57:10.000000000 -0400
+++ xen-unstable.hg/tools/examples/xend-config.sxp	2006-04-25 21:57:35.000000000 -0400
@@ -14,7 +14,7 @@
 #(logfile /var/log/xend.log)
 #(loglevel DEBUG)
 
-#(xend-http-server no)
+(xend-http-server yes)
 #(xend-unix-server no)
 #(xend-tcp-xmlrpc-server no)
 #(xend-unix-xmlrpc-server yes)
@@ -35,7 +35,7 @@
 # Specifying 'localhost' prevents remote connections.
 # Specifying the empty string '' (the default) allows all connections.
 #(xend-address '')
-#(xend-address localhost)
+(xend-address localhost)
 
 # Address xend should listen on for relocation-socket connections, if
 # xend-relocation-server is set.
