Index: xen-3.0.3-testing/docs/Makefile
===================================================================
--- xen-3.0.3-testing.orig/docs/Makefile
+++ xen-3.0.3-testing/docs/Makefile
@@ -12,8 +12,8 @@ LATEX2HTML	:= latex2html
 DOXYGEN		:= doxygen
 POD2MAN		:= pod2man
 
-pkgdocdir	:= /usr/share/doc/xen
-mandir		:= /usr/share/man
+pkgdocdir	:= $(DOCDIR)
+mandir		:= $(MANDIR)
 
 DOC_MAN5SRC	:= $(wildcard man/*.pod.5)
 DOC_MAN1SRC	:= $(wildcard man/*.pod.1)
Index: xen-3.0.3-testing/tools/xentrace/Makefile
===================================================================
--- xen-3.0.3-testing.orig/tools/xentrace/Makefile
+++ xen-3.0.3-testing/tools/xentrace/Makefile
@@ -43,14 +43,14 @@ install: build
 	[ -d $(DESTDIR)/usr/bin ] || $(INSTALL_DIR) $(DESTDIR)/usr/bin
 	[ -z "$(LIBBIN)" ] || [ -d $(DESTDIR)/usr/$(LIBDIR)/xen/bin ] || \
 		$(INSTALL_DIR) $(DESTDIR)/usr/$(LIBDIR)/xen/bin
-	[ -d $(DESTDIR)/usr/share/man/man1 ] || \
-		$(INSTALL_DIR) $(DESTDIR)/usr/share/man/man1
-	[ -d $(DESTDIR)/usr/share/man/man8 ] || \
-		$(INSTALL_DIR) $(DESTDIR)/usr/share/man/man8
+	[ -d $(DESTDIR)$(MANDIR)/man1 ] || \
+		$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man1
+	[ -d $(DESTDIR)$(MANDIR)/man8 ] || \
+		$(INSTALL_DIR) $(DESTDIR)$(MANDIR)/man8
 	$(INSTALL_PROG) $(BIN) $(SCRIPTS) $(DESTDIR)/usr/bin
 	[ -z "$(LIBBIN)" ] || $(INSTALL_PROG) $(LIBBIN) $(DESTDIR)/usr/$(LIBDIR)/xen/bin
-	$(INSTALL_DATA) $(MAN1) $(DESTDIR)/usr/share/man/man1
-	$(INSTALL_DATA) $(MAN8) $(DESTDIR)/usr/share/man/man8
+	$(INSTALL_DATA) $(MAN1) $(DESTDIR)$(MANDIR)/man1
+	$(INSTALL_DATA) $(MAN8) $(DESTDIR)$(MANDIR)/man8
 
 .PHONY: clean
 clean:
Index: xen-3.0.3-testing/tools/examples/Makefile
===================================================================
--- xen-3.0.3-testing.orig/tools/examples/Makefile
+++ xen-3.0.3-testing/tools/examples/Makefile
@@ -43,18 +43,6 @@ XEN_HOTPLUG_SCRIPTS = xen-backend.agent
 UDEV_RULES_DIR = /etc/udev
 UDEV_RULES = xen-backend.rules
 
-DI = $(if $(DISTDIR),$(shell readlink -f $(DISTDIR)),)
-DE = $(if $(DESTDIR),$(shell readlink -f $(DESTDIR)),)
-ifeq ($(findstring $(DI),$(DE)),$(DI))
-HOTPLUGS=install-hotplug install-udev
-else
-ifeq ($(shell [ -x /usr/bin/udevinfo ] && [ `/usr/bin/udevinfo -V | sed -e 's/^[^0-9]* \([0-9]\{1,\}\)[^0-9]\{0,\}/\1/'` -ge 059 ] && echo 1),1)
-HOTPLUGS=install-udev
-else
-HOTPLUGS=install-hotplug
-endif
-endif
-
 .PHONY: all
 all:
 
@@ -62,15 +50,15 @@ all:
 build:
 
 .PHONY: install
-install: all install-initd install-configs install-scripts $(HOTPLUGS)
+install: all install-initd install-configs install-scripts install-udev
 
 .PHONY: install-initd
 install-initd:
 	[ -d $(DESTDIR)/etc/init.d ] || $(INSTALL_DIR) $(DESTDIR)/etc/init.d
-	[ -d $(DESTDIR)/etc/sysconfig ] || $(INSTALL_DIR) $(DESTDIR)/etc/sysconfig
+	[ -d $(DESTDIR)/var/adm/fillup-templates ] || $(INSTALL_DIR) $(DESTDIR)/var/adm/fillup-templates/
 	$(INSTALL_PROG) $(XEND_INITD) $(DESTDIR)/etc/init.d
 	$(INSTALL_PROG) $(XENDOMAINS_INITD) $(DESTDIR)/etc/init.d
-	$(INSTALL_PROG) $(XENDOMAINS_SYSCONFIG) $(DESTDIR)/etc/sysconfig/xendomains
+	$(INSTALL_PROG) $(XENDOMAINS_SYSCONFIG) $(DESTDIR)/var/adm/fillup-templates/
 
 .PHONY: install-configs
 install-configs: $(XEN_CONFIGS)
@@ -111,9 +99,7 @@ install-udev:
 		$(INSTALL_DIR) $(DESTDIR)$(UDEV_RULES_DIR)/rules.d
 	for i in $(UDEV_RULES); \
 	    do \
-	    $(INSTALL_DATA) $$i $(DESTDIR)$(UDEV_RULES_DIR); \
-	    ( cd $(DESTDIR)$(UDEV_RULES_DIR)/rules.d ; \
-	        ln -sf ../$$i . ) \
+	    $(INSTALL_DATA) $$i $(DESTDIR)$(UDEV_RULES_DIR)/rules.d; \
 	done
 
 .PHONY: clean
Index: xen-3.0.3-testing/tools/security/Makefile
===================================================================
--- xen-3.0.3-testing.orig/tools/security/Makefile
+++ xen-3.0.3-testing/tools/security/Makefile
@@ -73,9 +73,9 @@ install: all $(ACM_CONFIG_FILE)
 	$(INSTALL_DIR) -p $(DESTDIR)$(ACM_SECGEN_CGIDIR)
 	$(INSTALL_PROG) -p $(ACM_INST_CGI) $(DESTDIR)$(ACM_SECGEN_CGIDIR)
 ifndef XEN_PYTHON_NATIVE_INSTALL
-	python python/setup.py install --home="$(DESTDIR)/usr"
+	python python/setup.py install --home="$(DESTDIR)/usr" --prefix=""
 else
-	python python/setup.py install --root="$(DESTDIR)"
+	python python/setup.py install --root="$(DESTDIR)" --prefix="/usr"
 endif
 else
 .PHONY: all
Index: xen-3.0.3-testing/tools/pygrub/Makefile
===================================================================
--- xen-3.0.3-testing.orig/tools/pygrub/Makefile
+++ xen-3.0.3-testing/tools/pygrub/Makefile
@@ -15,7 +15,7 @@ install: all
 	$(INSTALL_DIR) -p $(DESTDIR)/var/lib/xen
 else
 install: all
-	CFLAGS="$(CFLAGS)" python setup.py install --root="$(DESTDIR)"
+	CFLAGS="$(CFLAGS)" python setup.py install --root="$(DESTDIR)" --prefix="/usr"
 	$(INSTALL_DIR) -p $(DESTDIR)/var/lib/xen
 endif
 
Index: xen-3.0.3-testing/tools/python/Makefile
===================================================================
--- xen-3.0.3-testing.orig/tools/python/Makefile
+++ xen-3.0.3-testing/tools/python/Makefile
@@ -14,7 +14,7 @@ install: all
 	CFLAGS="$(CFLAGS)" python setup.py install --home="$(DESTDIR)/usr" --prefix="" --force
 else
 install: all
-	CFLAGS="$(CFLAGS)" python setup.py install --root="$(DESTDIR)" --force
+	CFLAGS="$(CFLAGS)" python setup.py install --root="$(DESTDIR)" --prefix="/usr" --force
 endif
 
 .PHONY: test
Index: xen-3.0.3-testing/xen/Makefile
===================================================================
--- xen-3.0.3-testing.orig/xen/Makefile
+++ xen-3.0.3-testing/xen/Makefile
@@ -31,8 +31,11 @@ _install: $(TARGET).gz
 	$(INSTALL_DATA) $(TARGET)-syms $(DESTDIR)/boot/$(notdir $(TARGET))-syms-$(XEN_FULLVERSION)
 	[ -d $(DESTDIR)/usr/include/xen/io ] || \
 		$(INSTALL_DIR) $(DESTDIR)/usr/include/xen/io
+	[ -d $(DESTDIR)/usr/include/xen/hvm ] || \
+		$(INSTALL_DIR) $(DESTDIR)/usr/include/xen/hvm
 	$(INSTALL_DATA) include/public/*.h $(DESTDIR)/usr/include/xen
 	$(INSTALL_DATA) include/public/io/*.h $(DESTDIR)/usr/include/xen/io
+	$(INSTALL_DATA) include/public/hvm/*.h $(DESTDIR)/usr/include/xen/hvm
 	$(INSTALL_DATA) include/public/COPYING $(DESTDIR)/usr/include/xen
 
 .PHONY: _debug
