Index: xen-unstable/tools/pygrub/src/pygrub
===================================================================
--- xen-unstable.orig/tools/pygrub/src/pygrub
+++ xen-unstable/tools/pygrub/src/pygrub
@@ -432,9 +432,9 @@ if __name__ == "__main__":
         print >> sys.stderr, "Usage: %s [-q|--quiet] [--output=] [--entry=] <image>" %(sys.argv[0],)
 
     try:
-        opts, args = getopt.gnu_getopt(sys.argv[1:], 'qh::',
+        opts, args = getopt.gnu_getopt(sys.argv[1:], 'qvh::',
                                    ["quiet", "help", "output=", "entry=",
-                                    "isconfig"])
+                                    "isconfig", "root=", "disks=", "verbose="])
     except getopt.GetoptError:
         usage()
         sys.exit(1)
