diff -Naur vtun.orig/configure.in vtun/configure.in
--- vtun.orig/configure.in	2005-07-20 12:12:17.000000000 +0200
+++ vtun/configure.in	2005-07-20 12:12:33.000000000 +0200
@@ -98,10 +98,10 @@
 AC_SEARCH_LIBS(nanosleep, rt posix4)
 
 dnl Check for setproctitle in libutil
-AC_SEARCH_LIBS(setproctitle, util bsd, AC_DEFINE(HAVE_SETPROC_TITLE) )
+AC_SEARCH_LIBS(setproctitle, util bsd, AC_DEFINE([HAVE_SETPROC_TITLE], 1, [Have setproc title]) )
 
 if test "$SHAPER" = "yes"; then
-   AC_DEFINE(HAVE_SHAPER)
+   AC_DEFINE([HAVE_SHAPER], 1, [Have shaper])
 fi
 
 if test "$ZLIB" = "yes"; then
@@ -109,7 +109,7 @@
    AC_CHECKING( for ZLIB Library and Header files ... )
    AC_CHECK_LIB(z, deflate,
 	LIBS="$LIBS -lz"
-	AC_DEFINE(HAVE_ZLIB),
+	AC_DEFINE([HAVE_ZLIB], 1, [Use zlib]),
 	AC_MSG_ERROR( Zlib library not found.)
    )
 fi
@@ -117,12 +117,13 @@
 if test "$LZO" = "yes"; then
    AC_MSG_RESULT()
    AC_CHECKING( for LZO Library and Header files ... )
-   AC_SEARCH_HEADERS(lzo1x.h, 
+   AC_SEARCH_HEADERS(lzo/lzo1x.h, 
 	$LZO_HDR_DIR /usr/include/lzo "" /usr/local/include,
-        AC_CHECK_LIB(lzo, lzo1x_decompress,
+        AC_CHECK_LIB(lzo2, lzo1x_1_15_compress,
 	    [
-	       LIBS="$LIBS -llzo"
-	       AC_DEFINE(HAVE_LZO)
+	       LIBS="$LIBS -llzo2"
+	       AC_DEFINE([HAVE_LZO], 1, [LZO Support enabled])
+	       AC_DEFINE([LZO_HEADER_DIR], 1, [LZO 2 Support enabled])
 	    ],
 	    AC_MSG_ERROR( LZO library not found. )
 	),
@@ -134,7 +135,7 @@
    AC_MSG_RESULT()
    AC_CHECKING( for md5 Library and Header files ... )
 AC_SEARCH_HEADERS(md5.h,
-    $SSL_HDR_DIR /usr/include/openssl "" /usr/include /usr/include/ssl /usr/local/include /usr/local/ssl/include, /usr/include/sys, 
+    $SSL_HDR_DIR /usr/include/openssl "" /usr/include /usr/include/ssl /usr/local/include /usr/local/ssl/include, 
     ,
     AC_MSG_ERROR( SSL headers not found. )
    )
@@ -148,7 +149,7 @@
 	AC_CHECK_LIB(crypto, BF_set_key,
 	    [
 	       LIBS="$LIBS -lcrypto"
-	       AC_DEFINE(HAVE_SSL)
+	       AC_DEFINE([HAVE_SSL], 1, [SSL Support])
 	    ],
 	    AC_MSG_ERROR( SSL library not found. )
 	),
@@ -207,6 +208,6 @@
 REL=`echo 'substr($Name,0)$' | tr -d '$: \-' | sed 's/^[A-Za-z]*//' | sed 's/\_/\./'`
 changequote([,])
 
-AC_DEFINE_UNQUOTED(VTUN_VER, "$REL `date '+%m/%d/%Y'`")
+AC_DEFINE_UNQUOTED(VTUN_VER, "$REL `date '+%m/%d/%Y'`", [Vtun Version])
 
 AC_OUTPUT(Makefile)
diff -Naur vtun.orig/lfd_lzo.c vtun/lfd_lzo.c
--- vtun.orig/lfd_lzo.c	2005-07-20 12:12:17.000000000 +0200
+++ vtun/lfd_lzo.c	2005-07-20 12:12:46.000000000 +0200
@@ -34,12 +34,16 @@
 #include "lib.h"
 
 #ifdef HAVE_LZO
-
+#ifdef LZO_HEADER_DIR
+#include "lzo/lzoutil.h"
+#include "lzo/lzo1x.h"
+#else
 #include "lzo1x.h"
+#endif
 
 static lzo_byte *zbuf;
 static lzo_voidp wmem;
-static int zbuf_size = VTUN_FRAME_SIZE * VTUN_FRAME_SIZE / 64 + 16 + 3;
+static int zbuf_size = VTUN_FRAME_SIZE * VTUN_FRAME_SIZE / 8 + 128 + 3;
 
 /* Pointer to compress function */
 int (*lzo1x_compress)(const lzo_byte *src, lzo_uint  src_len,
@@ -102,7 +106,7 @@
  */  
 int comp_lzo(int len, char *in, char **out)
 { 
-     unsigned int zlen = 0;    
+     lzo_uint zlen = 0;    
      int err;
      
      if( (err=lzo1x_compress((void *)in,len,zbuf,&zlen,wmem)) != LZO_E_OK ){
@@ -116,7 +120,7 @@
 
 int decomp_lzo(int len, char *in, char **out)
 {
-     unsigned int zlen = 0;
+     lzo_uint zlen = 0;
      int err;
 
      if( (err=lzo1x_decompress((void *)in,len,zbuf,&zlen,wmem)) != LZO_E_OK ){
