--- freetype-2.2.1.orig/include/freetype/config/ftoption.h	2006-09-27 00:55:44.000000000 +0300
+++ freetype-2.2.1/include/freetype/config/ftoption.h	2006-10-31 03:08:14.000000000 +0200
@@ -92,7 +92,7 @@
   /* This is done to allow FreeType clients to run unmodified, forcing     */
   /* them to display normal gray-level anti-aliased glyphs.                */
   /*                                                                       */
-/* #define FT_CONFIG_OPTION_SUBPIXEL_RENDERING */
+#define FT_CONFIG_OPTION_SUBPIXEL_RENDERING
 
 
   /*************************************************************************/
