
Index: freetype-2.3.5/include/freetype/config/ftoption.h
===================================================================
--- freetype-2.3.5.orig/include/freetype/config/ftoption.h	2007-07-07 01:14:27.000000000 -0700
+++ freetype-2.3.5/include/freetype/config/ftoption.h	2007-07-07 01:14:42.000000000 -0700
@@ -505,7 +505,7 @@
   /*     ...                                                               */
   /*   }                                                                   */
   /*                                                                       */
-#define TT_CONFIG_OPTION_UNPATENTED_HINTING
+/* #define TT_CONFIG_OPTION_UNPATENTED_HINTING */
 
 
   /*************************************************************************/

