
Index: include/freetype/config/ftoption.h
===================================================================
--- include/freetype/config/ftoption.h.orig	2007-07-07 01:14:19.000000000 -0700
+++ include/freetype/config/ftoption.h	2007-07-07 01:14:21.000000000 -0700
@@ -458,7 +458,7 @@
   /*   Do not #undef this macro here, since the build system might         */
   /*   define it for certain configurations only.                          */
   /*                                                                       */
-/* #define TT_CONFIG_OPTION_BYTECODE_INTERPRETER */
+#define TT_CONFIG_OPTION_BYTECODE_INTERPRETER
 
 
   /*************************************************************************/

