diff -Naur tea-22.3.0/src.pro tea-22.3.0.new/src.pro
--- tea-22.3.0/src.pro	2009-01-21 19:10:22.000000000 +0530
+++ tea-22.3.0.new/src.pro	2009-02-01 12:33:04.000000000 +0530
@@ -26,8 +26,11 @@
     xml \
     script \
     svg
+;target.path = /usr/local/bin
+DESTDIR = /var/tmp/fst/pkg
+PREFIX = $$DESTDIR/usr
 TARGET = bin/tea
-target.path = /usr/local/bin
+target.path = $$PREFIX/bin
 INSTALLS += target
 RESOURCES += application.qrc \
     rlvn.qrc
