--- tea-18.0.0/src.pro	2008-07-04 19:51:40.000000000 +0530
+++ tea-18.0.0.new/src.pro	2008-07-11 13:16:52.000000000 +0530
@@ -22,13 +22,13 @@
 QT += network xml \
       script svg
       
-
-;PREFIX = /usr/local
-;TARGET = tea
-;target.path = $$PREFIX/bin
-
+DESTDIR = /var/tmp/fst/pkg/
+PREFIX = $$DESTDIR/usr
 TARGET = bin/tea
-target.path = /usr/local/bin
+target.path = $$PREFIX/bin
+
+;TARGET = bin/tea
+;target.path = /usr/local/bin
 
 INSTALLS += target
 
