diff -Naurp QScintilla2-1.71-gpl-2-snapshot-20061104/Qt4/qscintilla.pro QScintilla2-1.71-gpl-2-snapshot-20061104-p/Qt4/qscintilla.pro
--- QScintilla2-1.71-gpl-2-snapshot-20061104/Qt4/qscintilla.pro	2006-11-16 04:55:53.000000000 +0100
+++ QScintilla2-1.71-gpl-2-snapshot-20061104-p/Qt4/qscintilla.pro	2006-11-17 02:31:29.000000000 +0100
@@ -29,9 +29,8 @@ INCLUDEPATH = . ../include ../src
 DEFINES = QSCINTILLA_MAKE_DLL QT SCI_LEXER
 
 # Handle both Qt v4 and v3.
-DESTDIR = $$[QT_INSTALL_LIBS]
 isEmpty(DESTDIR) {
-	DESTDIR = $(QTDIR)/lib
+	DESTDIR = $(INSTALL_ROOT)$(QTDIR)/lib
 }
 
 header.path = $$[QT_INSTALL_HEADERS]
