diff -rupN OpenCOLLADA-69b844d/COLLADAFramework/include/COLLADAFWInstanceBindingBase.h OpenCOLLADA-69b844d-new/COLLADAFramework/include/COLLADAFWInstanceBindingBase.h
--- OpenCOLLADA-69b844d/COLLADAFramework/include/COLLADAFWInstanceBindingBase.h	2014-08-28 05:30:44.000000000 +0200
+++ OpenCOLLADA-69b844d-new/COLLADAFramework/include/COLLADAFWInstanceBindingBase.h	2014-10-08 23:53:23.152124797 +0200
@@ -15,6 +15,11 @@
 #include "COLLADAFWInstanceBase.h"
 #include "COLLADAFWMaterialBinding.h"
 
+#include "COLLADABUURI.h"
+
+#include <vector>
+
+
 namespace COLLADAFW
 {
 
