diff -ur a/Makefile b/Makefile
--- a/Makefile	2012-08-19 20:58:07.262612478 -0500
+++ b/Makefile	2012-08-19 21:03:46.712239866 -0500
@@ -13,7 +13,7 @@
 CROSS_COMPILE=
 
 # Support for compressed pci.ids (yes/no, default: detect)
-ZLIB=
+ZLIB=no
 
 # Support for resolving ID's by DNS (yes/no, default: detect)
 DNS=
@@ -25,11 +25,11 @@
 ABI_VERSION=.3
 
 # Installation directories
-PREFIX=/usr/local
+PREFIX=/usr
 SBINDIR=$(PREFIX)/sbin
-SHAREDIR=$(PREFIX)/share
+SHAREDIR=$(PREFIX)/share/hwdata
 IDSDIR=$(SHAREDIR)
-MANDIR:=$(shell if [ -d $(PREFIX)/share/man ] ; then echo $(PREFIX)/share/man ; else echo $(PREFIX)/man ; fi)
+MANDIR:=$(PREFIX)/share/man
 INCDIR=$(PREFIX)/include
 LIBDIR=$(PREFIX)/lib
 PKGCFDIR=$(LIBDIR)/pkgconfig
