diff -Naur pciutils-2.2.1.orig/Makefile pciutils-2.2.1/Makefile
--- pciutils-2.2.1.orig/Makefile	2005-11-26 16:59:24.963713000 +0100
+++ pciutils-2.2.1/Makefile	2005-11-26 17:00:09.758512500 +0100
@@ -10,7 +10,7 @@
 PREFIX=/usr/local
 SBINDIR=$(PREFIX)/sbin
 SHAREDIR=$(PREFIX)/share
-IDSDIR=$(SHAREDIR)
+IDSDIR=$(SHAREDIR)/misc
 MANDIR:=$(shell if [ -d $(PREFIX)/share/man ] ; then echo $(PREFIX)/share/man ; else echo $(PREFIX)/man ; fi)
 INSTALL=install
 DIRINSTALL=install -d
