diff --git a/src/trident_dga.c b/src/trident_dga.c
index 0263aa7..c4b5b9a 100644
--- a/src/trident_dga.c
+++ b/src/trident_dga.c
@@ -37,10 +37,10 @@
 static Bool TRIDENT_OpenFramebuffer(ScrnInfoPtr, char **, unsigned char **, 
 					int *, int *, int *);
 static Bool TRIDENT_SetMode(ScrnInfoPtr, DGAModePtr);
-static void TRIDENT_Sync(ScrnInfoPtr);
 static int  TRIDENT_GetViewport(ScrnInfoPtr);
 static void TRIDENT_SetViewport(ScrnInfoPtr, int, int, int);
 #ifdef HAVE_XAA_H
+static void TRIDENT_Sync(ScrnInfoPtr);
 static void TRIDENT_FillRect(ScrnInfoPtr, int, int, int, int, unsigned long);
 static void TRIDENT_BlitRect(ScrnInfoPtr, int, int, int, int, int, int);
 #if 0
@@ -56,8 +56,8 @@ DGAFunctionRec TRIDENTDGAFuncs = {
    TRIDENT_SetMode,
    TRIDENT_SetViewport,
    TRIDENT_GetViewport,
-   TRIDENT_Sync,
 #ifdef HAVE_XAA_H
+   TRIDENT_Sync,
    TRIDENT_FillRect,
    TRIDENT_BlitRect,
 #if 0
diff --git a/src/trident_driver.c b/src/trident_driver.c
index dc12b76..408c870 100644
--- a/src/trident_driver.c
+++ b/src/trident_driver.c
@@ -56,7 +56,6 @@
 
 #include "mipointer.h"
 
-#include "mibstore.h"
 #include "shadow.h"
 #include "trident.h"
 #include "trident_regs.h"
@@ -3037,7 +3036,6 @@ TRIDENTScreenInit(SCREEN_INIT_ARGS_DECL)
     	TridentAccelInit(pScreen);
     }
 
-    miInitializeBackingStore(pScreen);
     xf86SetBackingStore(pScreen);
 
     /* Initialise cursor functions */
