--- smc-1.9/src/overworld/world_manager.cpp	2009-08-16 18:44:52.000000000 +0200
+++ smc-1.9.new/src/overworld/world_manager.cpp	2011-02-10 13:04:19.029260001 +0100
@@ -111,14 +111,14 @@
 void cOverworld_Manager :: Load_Dir( const std::string &dir, bool user_dir /* = 0 */ ) 
 {
 	// set world directory
-	fs::path full_path( dir, fs::native );
+	fs::path full_path( dir );
 	fs::directory_iterator end_iter;
 
 	for( fs::directory_iterator dir_itr( full_path ); dir_itr != end_iter; ++dir_itr )
 	{
 		try
 		{
-			std::string current_dir = dir_itr->path().leaf();
+			std::string current_dir = dir_itr->path().filename().native();
 
 			// only directories with an existing description
 			if( fs::is_directory( *dir_itr ) && File_Exists( dir + "/" + current_dir + "/description.xml" ) )
@@ -145,7 +145,7 @@
 		}
 		catch( const std::exception &ex )
 		{
-			printf( "%s %s\n", dir_itr->path().leaf().c_str(), ex.what() );
+			printf( "%s %s\n", dir_itr->path().filename().c_str(), ex.what() );
 		}
 	}
 }
Only in smc-1.9.new/src: utilities.o
Only in smc-1.9.new/src: vector.o
diff -ur smc-1.9/src/video/video.cpp smc-1.9.new/src/video/video.cpp
--- smc-1.9/src/video/video.cpp	2011-02-10 13:06:45.880260001 +0100
+++ smc-1.9.new/src/video/video.cpp	2011-02-10 13:06:25.319259961 +0100
@@ -735,7 +735,7 @@
 		{
 			try
 			{
-				fs::remove_all( fs::path( m_imgcache_dir, fs::native ) );
+				fs::remove_all( fs::path( m_imgcache_dir ) );
 			}
 			// could happen if a file is locked or we have no write rights
 			catch( const std::exception &ex )
@@ -757,7 +757,7 @@
 	// no cache available
 	if( !Dir_Exists( imgcache_dir_active ) )
 	{
-		fs::create_directories( fs::path( imgcache_dir_active + "/" GAME_PIXMAPS_DIR, fs::native ) );
+		fs::create_directories( fs::path( imgcache_dir_active + "/" GAME_PIXMAPS_DIR ) );
 	}
 	// cache available
 	else
