--- smc-1.4/configure.ac.orig	2008-01-01 16:19:58.000000000 +0100
+++ smc-1.4/configure.ac	2008-01-01 16:29:24.000000000 +0100
@@ -26,6 +26,10 @@
 CPPFLAGS="$CPPFLAGS $SDL_CFLAGS"
 LIBS="$LIBS $SDL_LIBS"
 
+# Check for png library
+AC_CHECK_LIB(png, png_create_write_struct, ,
+	AC_MSG_ERROR([Unable to find the PNG library]))
+
 # Check for SDL_image library
 AC_CHECK_LIB(SDL_image, IMG_LoadPNG_RW, ,
 	AC_MSG_ERROR([Unable to find the SDL_image library with PNG support]))
