--- src/GParted_Core.cc	2008-02-04 09:35:47.000000000 -0500
+++ src/GParted_Core.cc.new	2008-02-07 15:26:32.000000000 -0500
@@ -38,8 +38,6 @@
 #include "../include/ufs.h"
 
 #include <cerrno>
-#include <cstdio>
-#include <csignal>
 #include <sys/statvfs.h>	
 
 std::vector<Glib::ustring> libparted_messages ; //see ped_exception_handler()
@@ -47,13 +45,6 @@
 namespace GParted
 {
 
-// Clean up the fdi when terminating
-void sig_fdi_cleanup( int signum )
-{
-	remove( "/usr/share/hal/fdi/policy/gparted-disable-automount.fdi" );
-	raise( signum );
-}
-
 GParted_Core::GParted_Core() 
 {
 	lp_device = NULL ;
@@ -63,44 +54,6 @@
 
 	ped_exception_set_handler( ped_exception_handler ) ; 
 
-	//make sure that the fdi is cleaned up on all but the most forceful
-	//program exits
-	struct sigaction sa;
-	sa.sa_handler = sig_fdi_cleanup;
-	sigemptyset( &sa.sa_mask );
-	sa.sa_flags = SA_RESETHAND;
-	if( sigaction( SIGHUP, &sa, NULL ) == -1 ||
-	    sigaction( SIGINT, &sa, NULL ) == -1 ||
-	    sigaction( SIGQUIT, &sa, NULL ) == -1 ||
-	    sigaction( SIGILL, &sa, NULL ) == -1 ||
-	    sigaction( SIGABRT, &sa, NULL ) == -1 ||
-	    sigaction( SIGFPE, &sa, NULL ) == -1 ||
-	    sigaction( SIGSEGV, &sa, NULL ) == -1 ||
-	    sigaction( SIGPIPE, &sa, NULL ) == -1 ||
-	    sigaction( SIGALRM, &sa, NULL ) == -1 ||
-	    sigaction( SIGTERM, &sa, NULL ) == -1 ||
-	    sigaction( SIGUSR1, &sa, NULL ) == -1 ||
-	    sigaction( SIGUSR2, &sa, NULL ) == -1 ||
-	    sigaction( SIGBUS, &sa, NULL ) == -1 ||
-	    sigaction( SIGXCPU, &sa, NULL ) == -1 ||
-	    sigaction( SIGXFSZ, &sa, NULL ) == -1 )
-		perror( "Could not set signal handler" );
-	
-	//disable automount //FIXME: temporary hack, till i find a better solution...
-	std::ofstream fdi_file( "/usr/share/hal/fdi/policy/gparted-disable-automount.fdi" ) ;
-	if ( fdi_file )
-	{
-		fdi_file << "<deviceinfo version='0.2'>" ;
-		fdi_file << "<device>" ;
-		fdi_file << "<match key='storage.hotpluggable' bool='true'>" ;
-		fdi_file << "<merge key='storage.automount_enabled_hint' type='bool'>false</merge>" ;
-		fdi_file << "</match>" ;
-		fdi_file << "</device>" ;
-		fdi_file << "</deviceinfo>" ;
-
-		fdi_file .close() ;
-	}	
-
 	//get valid flags ...
 	for ( PedPartitionFlag flag = ped_partition_flag_next( static_cast<PedPartitionFlag>( NULL ) ) ;
 	      flag ;
@@ -2332,8 +2285,6 @@
 	if ( p_filesystem )
 		delete p_filesystem ;
 
-	//remove .fdi file..
-	remove( "/usr/share/hal/fdi/policy/gparted-disable-automount.fdi" ) ;
 }
 	
 } //GParted

