Description: fix FTBFS with gcc 4.7
Author: Paul Tagliamonte <tag@pault.ag>
Bug-Debian: http://bugs.debian.org/667194

--- a/src/gsc_text_window.h	2012-04-18 22:03:41.340661672 -0400
+++ b/src/gsc_text_window.h	2012-04-18 22:03:46.264661939 -0400
@@ -126,7 +126,7 @@ class GscTextWindow : public AppUIResWid
 		// by default, delete_event calls hide().
 		bool on_delete_event_before(GdkEventAny* e)
 		{
-			destroy(this);  // deletes this object and nullifies instance
+			this->destroy(this);  // deletes this object and nullifies instance
 			return true;  // event handled, don't call default virtual handler
 		}
 
@@ -184,7 +184,7 @@ class GscTextWindow : public AppUIResWid
 
 		void on_close_window_button_clicked()
 		{
-			destroy(this);
+			this->destroy(this);
 		}
 
 
