diff -Naur hotplug-2004_09_23.orig/etc/hotplug/net.agent hotplug-2004_09_23/etc/hotplug/net.agent
--- hotplug-2004_09_23.orig/etc/hotplug/net.agent	2004-12-28 11:53:03.700631192 +0100
+++ hotplug-2004_09_23/etc/hotplug/net.agent	2004-12-28 12:53:21.883583776 +0100
@@ -62,9 +62,17 @@
 		fi;
 	    fi
 
+	    # Frugalware Linux
+	    if [ -x /etc/rc.d/rc.interfaces ]; then
+		# Interface already up
+		if ! /sbin/ifconfig | grep "^${INTERFACE} " 1> /dev/null ; then
+		    debug_msg run rc.interfaces
+		    exec /etc/rc.d/rc.interfaces
+		fi
+		
 	    # RedHat and similar
-	    export IN_HOTPLUG=1
-	    if [ -x /sbin/ifup ]; then
+	    elif [ -x /sbin/ifup ]; then
+		export IN_HOTPLUG=1
 		debug_mesg invoke ifup $INTERFACE
 		exec /sbin/ifup $INTERFACE
 
@@ -91,14 +99,10 @@
 	    exit 0
 	    ;;
 	*)
-	    # right now it looks like only Gentoo wants to care about
-	    # unregistering network devices...
-	    if [ -f /etc/gentoo-release ]; then
-	    	script=/etc/init.d/net.$INTERFACE
-		if [ -x "$script" ]; then
-		    debug_mesg invoke "$script" --quiet stop
-		    exec "$script" --quiet stop
-		fi
+	    # Does the interface use dhcpcd? If it does, we must kill the daemon.
+	    if [ -r /etc/dhcpc/dhcpcd-$INTERFACE.pid ]; then
+		debug_mesg killing /etc/dhcpc/dhcpcd-$INTERFACE.pid
+		kill -15 `cat /etc/dhcpc/dhcpcd-$INTERFACE.pid`
 	    fi
 	    ;;
     esac
