diff -Naur live/live555.pc.in live-pkgconfig/live555.pc.in
--- live/live555.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ live-pkgconfig/live555.pc.in	2016-04-06 20:47:32.266673969 +0200
@@ -0,0 +1,9 @@
+prefix=@PREFIX@
+libdir=@LIBDIR@
+includedir=${prefix}/include
+
+Name: live555
+Description: multimedia RTSP streaming library
+Version: @VERSION@
+Cflags: -I${includedir}/liveMedia -I${includedir}/groupsock -I${includedir}/BasicUsageEnvironment -I${includedir}/UsageEnvironment
+Libs: -L${libdir} -lliveMedia -lgroupsock -lBasicUsageEnvironment -lUsageEnvironment
diff -Naur live/Makefile.head live-pkgconfig/Makefile.head
--- live/Makefile.head	2016-04-01 12:19:55.000000000 +0200
+++ live-pkgconfig/Makefile.head	2016-04-06 20:53:31.300155982 +0200
@@ -1 +1,4 @@
 ##### Change the following for your environment:
+PREFIX = /usr
+LIBDIR = /usr/lib
+VERSION = FWVER
diff -Naur live/Makefile.tail live-pkgconfig/Makefile.tail
--- live/Makefile.tail	2016-04-01 12:19:55.000000000 +0200
+++ live-pkgconfig/Makefile.tail	2016-04-06 20:47:32.266673969 +0200
@@ -20,7 +20,12 @@
 	cd $(MEDIA_SERVER_DIR) ; $(MAKE)
 	cd $(PROXY_SERVER_DIR) ; $(MAKE)
 
-install:
+install_shared_libraries:
+	install -d $(DESTDIR)$(LIBDIR)/pkgconfig
+	sed "s#@PREFIX@#$(PREFIX)#;s#@LIBDIR@#$(LIBDIR)#;s#@VERSION@#$(VERSION)#" live555.pc.in > $(DESTDIR)$(LIBDIR)/pkgconfig/live555.pc
+	chmod 644 $(DESTDIR)$(LIBDIR)/pkgconfig/live555.pc
+
+install: $(INSTALL2)
 	cd $(LIVEMEDIA_DIR) ; $(MAKE) install
 	cd $(GROUPSOCK_DIR) ; $(MAKE) install
 	cd $(USAGE_ENVIRONMENT_DIR) ; $(MAKE) install
