diff -Naur xawtv-3.95.orig/configure xawtv-3.95/configure
--- xawtv-3.95.orig/configure	2007-07-17 19:00:44.000000000 +0200
+++ xawtv-3.95/configure	2007-07-17 19:02:21.000000000 +0200
@@ -6812,10 +6812,7 @@
 
 echo "$as_me:$LINENO: checking for X11 config directory" >&5
 echo $ECHO_N "checking for X11 config directory... $ECHO_C" >&6
-x11conf=/usr/X11R6/lib/X11
-if test -d /etc/X11; then
-	x11conf=/etc/X11
-fi
+x11conf=/etc/X11
 echo "$as_me:$LINENO: result: $x11conf" >&5
 echo "${ECHO_T}$x11conf" >&6
 
