diff -urN monotheka-0.0.7.orig/MainWindow/boil monotheka-0.0.7/MainWindow/boil
--- monotheka-0.0.7.orig/MainWindow/boil	2005-04-15 22:45:09.000000000 +0100
+++ monotheka-0.0.7/MainWindow/boil	2006-08-31 08:58:56.000000000 +0100
@@ -1,11 +1,11 @@
 NAME="Monotheka"
 TYPE="exe"
 
-REF_PKG="gtk-sharp \
-gnome-sharp \
-gtkhtml-sharp \
-glade-sharp \
-gconf-sharp"
+REF_PKG="gtk-sharp-2.0 \
+gnome-sharp-2.0 \
+gtkhtml-sharp-2.0 \
+glade-sharp-2.0 \
+gconf-sharp-2.0"
 
 REF_INTERNAL="Util.dll PluginLib.dll MovieDatabase.dll"
 DEPENDS="Util MovieDatabase PluginLib"
diff -urN monotheka-0.0.7.orig/monoboil.config monotheka-0.0.7/monoboil.config
--- monotheka-0.0.7.orig/monoboil.config	2006-04-01 16:43:40.000000000 +0100
+++ monotheka-0.0.7/monoboil.config	2006-08-31 08:58:26.000000000 +0100
@@ -1,4 +1,4 @@
 BOIL_SOFTWARE="Monotheka"
 BOIL_VERSION="0.0.7"
 BOIL_CODENAME="Quickfix"
-BOIL_DEPENDENCIES_PKG="mono-1.0.6 gtk-sharp-1.0.4 glade-sharp-1.0.4 gnome-sharp-1.0.4 gconf-sharp-1.0.4 sqlite-2.8"
+BOIL_DEPENDENCIES_PKG="mono-1.0.6 gtk-sharp-2.0-2.10.0 glade-sharp-2.0-2.10.0 gnome-sharp-2.0-2.16.0 gconf-sharp-2.0-2.16.0 sqlite-2.8"
diff -urN monotheka-0.0.7.orig/PluginLib/boil monotheka-0.0.7/PluginLib/boil
--- monotheka-0.0.7.orig/PluginLib/boil	2005-03-25 14:54:20.000000000 +0000
+++ monotheka-0.0.7/PluginLib/boil	2006-08-31 08:59:10.000000000 +0100
@@ -1,6 +1,6 @@
 NAME=PluginLib
 TYPE=library
 REF_INTERNAL="MovieDatabase.dll Util.dll"
-REF_PKG="gtk-sharp"
+REF_PKG="gtk-sharp-2.0"
 DEPENDS=MovieDatabase
 SOURCES="PluginLib.cs FilmPortal.cs"
diff -urN monotheka-0.0.7.orig/Plugins/AMCImport/boil monotheka-0.0.7/Plugins/AMCImport/boil
--- monotheka-0.0.7.orig/Plugins/AMCImport/boil	2005-07-30 18:52:36.000000000 +0100
+++ monotheka-0.0.7/Plugins/AMCImport/boil	2006-08-31 09:00:06.000000000 +0100
@@ -1,7 +1,7 @@
 NAME=AMCImport
 TYPE=library
 DEPENDS="PluginLib MovieDatabase Util"
-REF_PKG="gtk-sharp gnome-sharp glade-sharp"
+REF_PKG="gtk-sharp-2.0 gnome-sharp-2.0 glade-sharp-2.0"
 REF_INTERNAL="MovieDatabase.dll PluginLib.dll Util.dll"
 SOURCES="AMCImport.cs Importer.cs ToggleList.cs"
 RESOURCES="AMCImport.glade"
diff -urN monotheka-0.0.7.orig/Plugins/AVIGet/boil monotheka-0.0.7/Plugins/AVIGet/boil
--- monotheka-0.0.7.orig/Plugins/AVIGet/boil	2005-03-18 19:03:55.000000000 +0000
+++ monotheka-0.0.7/Plugins/AVIGet/boil	2006-08-31 09:00:54.000000000 +0100
@@ -2,5 +2,5 @@
 TYPE=library
 DEPENDS="PluginLib MovieDatabase Util"
 REF_INTERNAL="MovieDatabase.dll PluginLib.dll Util.dll"
-REF_PKG="gtk-sharp"
+REF_PKG="gtk-sharp-2.0"
 SOURCES="AVIGet.cs RIFFReader.cs"
diff -urN monotheka-0.0.7.orig/Plugins/CSVImport/boil monotheka-0.0.7/Plugins/CSVImport/boil
--- monotheka-0.0.7.orig/Plugins/CSVImport/boil	2005-03-13 09:46:26.000000000 +0000
+++ monotheka-0.0.7/Plugins/CSVImport/boil	2006-08-31 08:59:41.000000000 +0100
@@ -1,7 +1,7 @@
 NAME=CSVImport
 TYPE=library
 DEPENDS="PluginLib MovieDatabase Util"
-REF_PKG="gtk-sharp gnome-sharp glade-sharp"
+REF_PKG="gtk-sharp-2.0 gnome-sharp-2.0 glade-sharp-2.0"
 REF_INTERNAL="MovieDatabase.dll PluginLib.dll Util.dll"
 SOURCES="CSVImport.cs ToggleList.cs Importer.cs"
 RESOURCES="CVSImport.glade"
diff -urN monotheka-0.0.7.orig/Plugins/FilmwebGet/boil monotheka-0.0.7/Plugins/FilmwebGet/boil
--- monotheka-0.0.7.orig/Plugins/FilmwebGet/boil	2005-03-25 14:55:42.000000000 +0000
+++ monotheka-0.0.7/Plugins/FilmwebGet/boil	2006-08-31 09:00:17.000000000 +0100
@@ -2,6 +2,6 @@
 TYPE=library
 DEPENDS="PluginLib MovieDatabase Util"
 REF_INTERNAL="MovieDatabase.dll PluginLib.dll Util.dll"
-REF_PKG="gtk-sharp"
+REF_PKG="gtk-sharp-2.0"
 SOURCES="FilmwebGet.cs SearchResultsPage.cs MoviePage.cs"
 RESOURCES="filmweb_logo.png"
diff -urN monotheka-0.0.7.orig/Plugins/HTMLExport/boil monotheka-0.0.7/Plugins/HTMLExport/boil
--- monotheka-0.0.7.orig/Plugins/HTMLExport/boil	2005-04-22 21:29:52.000000000 +0100
+++ monotheka-0.0.7/Plugins/HTMLExport/boil	2006-08-31 09:00:43.000000000 +0100
@@ -1,7 +1,7 @@
 NAME=HTMLExport
 TYPE=library
 DEPENDS="MovieDatabase PluginLib Util"
-REF_PKG="gtk-sharp gnome-sharp glade-sharp"
+REF_PKG="gtk-sharp-2.0 gnome-sharp-2.0 glade-sharp-2.0"
 REF_INTERNAL="MovieDatabase.dll PluginLib.dll Util.dll"
 SOURCES="HTMLExport.cs Template.cs HTMLSnippet.cs HTMLWriter.cs HTMLExport_Window.cs"
 RESOURCES="HTMLExport.glade nopreview.png"
diff -urN monotheka-0.0.7.orig/Plugins/IMDBGet/boil monotheka-0.0.7/Plugins/IMDBGet/boil
--- monotheka-0.0.7.orig/Plugins/IMDBGet/boil	2005-04-04 23:50:20.000000000 +0100
+++ monotheka-0.0.7/Plugins/IMDBGet/boil	2006-08-31 09:00:25.000000000 +0100
@@ -2,6 +2,6 @@
 TYPE=library
 DEPENDS="PluginLib MovieDatabase Util"
 REF_INTERNAL="MovieDatabase.dll PluginLib.dll Util.dll"
-REF_PKG="gtk-sharp"
+REF_PKG="gtk-sharp-2.0"
 SOURCES="IMDBGet.cs SearchResultsPage.cs MoviePage.cs"
 RESOURCES="imdb_logo.png"
diff -urN monotheka-0.0.7.orig/Plugins/TXTExport/boil monotheka-0.0.7/Plugins/TXTExport/boil
--- monotheka-0.0.7.orig/Plugins/TXTExport/boil	2005-05-07 12:20:00.000000000 +0100
+++ monotheka-0.0.7/Plugins/TXTExport/boil	2006-08-31 08:59:54.000000000 +0100
@@ -1,7 +1,7 @@
 NAME=TXTExport
 TYPE=library
 DEPENDS="PluginLib MovieDatabase Util"
-REF_PKG="gtk-sharp gnome-sharp glade-sharp"
+REF_PKG="gtk-sharp-2.0 gnome-sharp-2.0 glade-sharp-2.0"
 REF_INTERNAL="MovieDatabase.dll PluginLib.dll Util.dll"
 SOURCES="TXTExport.cs TextWriter.cs MailToEvolution.cs"
 RESOURCES="TXTExport.glade MailToEvolution.glade"
diff -urN monotheka-0.0.7.orig/Util/boil monotheka-0.0.7/Util/boil
--- monotheka-0.0.7.orig/Util/boil	2005-04-22 21:32:14.000000000 +0100
+++ monotheka-0.0.7/Util/boil	2006-08-31 09:01:10.000000000 +0100
@@ -1,6 +1,6 @@
 NAME="Util"
 TYPE="library"
-REF_PKG="gtk-sharp glade-sharp gconf-sharp"
+REF_PKG="gtk-sharp-2.0 glade-sharp-2.0 gconf-sharp-2.0"
 
 SOURCES="Strings.cs \
 ListWindow.cs \
