diff -urN gwget-0.98.1.orig/src/Makefile.in gwget-0.98.1/src/Makefile.in
--- gwget-0.98.1.orig/src/Makefile.in	2006-10-23 15:21:08.000000000 +0100
+++ gwget-0.98.1/src/Makefile.in	2006-10-23 15:21:58.000000000 +0100
@@ -648,7 +648,7 @@
 
 
 $(IDL_SOURCES): $(IDL_FILE) $(ORBIT_IDL) 	 
-	$(ORBIT_IDL) -I $(LIBBONOBO_IDL) -I $(BONOBO_ACTIVATION_IDL) $(srcdir)/$(IDL_FILE) 	 
+	$(ORBIT_IDL) -I /usr/share/idl/bonobo-2.0 -I /usr/share/idl/bonobo-activation-2.0 $(srcdir)/$(IDL_FILE) 	 
 
 gwget-marshal.h: gwget-marshal.list $(GLIB_GENMARSHAL)
 	        $(GLIB_GENMARSHAL) $< --header --prefix=gwget_marshal > $@
