diff -urN gwget-0.98.1.orig/src/main_window_cb.c gwget-0.98.1/src/main_window_cb.c
--- gwget-0.98.1.orig/src/main_window_cb.c	2006-10-23 15:26:13.000000000 +0100
+++ gwget-0.98.1/src/main_window_cb.c	2006-10-23 15:26:36.000000000 +0100
@@ -24,7 +24,6 @@
 #include <libgnome/gnome-init.h>
 #include <libgnomevfs/gnome-vfs-utils.h>
 #include <libgnomevfs/gnome-vfs-mime-utils.h>
-#include <gnome-vfs-module-2.0/libgnomevfs/gnome-vfs-mime.h>
 
 #include "main_window.h"
 #include "main_window_cb.h"
