diff --git a/kphone/kcallwidget.cpp b/kphone/kcallwidget.cpp
index cb57851..9927da8 100644
--- a/kphone/kcallwidget.cpp
+++ b/kphone/kcallwidget.cpp
@@ -92,9 +92,9 @@ KCallWidget::KCallWidget( KSipAuthentication *auth, CallAudio *callaudio, SipCal
 	: call( initcall ), phoneView( parent )
 {
 	if( call->getSubject() == tr("Incoming call") )
-		QDialog::QDialog( NULL, name, FALSE, WStyle_DialogBorder );
+		QDialog( NULL, name, FALSE, WStyle_DialogBorder );
 	else
-		QDialog::QDialog( parent, name, FALSE, WStyle_DialogBorder );
+		QDialog( parent, name, FALSE, WStyle_DialogBorder );
 		
 	sipauthentication = auth;
 	missedCalls.setAutoDelete( true );
