diff -Naur kphone.old/kphone/kcallwidget.h kphone/kphone/kcallwidget.h
--- kphone.old/kphone/kcallwidget.h	2006-03-31 20:00:14.000000000 -0500
+++ kphone/kphone/kcallwidget.h	2006-03-31 20:00:54.000000000 -0500
@@ -67,7 +67,7 @@
 	void clickHangup( void );
 	void setHide( void );
 	bool isHided( void ) const { return hided; }
-	void KCallWidget::setDTMFSender( QString &s );
+	void setDTMFSender( QString &s );
 
 public slots:
 	void pleaseDial( const SipUri &dialuri );
diff -Naur kphone.old/kphone/kphoneview.h kphone/kphone/kphoneview.h
--- kphone.old/kphone/kphoneview.h	2006-03-31 20:00:14.000000000 -0500
+++ kphone/kphone/kphoneview.h	2006-03-31 20:00:43.000000000 -0500
@@ -54,7 +54,7 @@
 	QString getStunSrv( void );
 	void kphoneQuit( void );
 	bool getState( void );
-	KCallWidget *KPhoneView::DoCall( QString num, SipCall::CallType ctype );
+	KCallWidget *DoCall( QString num, SipCall::CallType ctype );
 
 signals:
 	void stateChanged( void );
