diff -Naur popa3d-1.0.orig/Makefile popa3d-1.0/Makefile
--- popa3d-1.0.orig/Makefile	2003-11-17 01:43:35.000000000 +0100
+++ popa3d-1.0/Makefile	2005-06-11 20:38:32.000000000 +0200
@@ -5,11 +5,11 @@
 INSTALL = install -c
 CFLAGS = -Wall -O2 -fomit-frame-pointer
 # You may use OpenSSL's MD5 routines instead of the ones supplied here
-#CFLAGS += -DHAVE_OPENSSL
+CFLAGS += -DHAVE_OPENSSL
 LDFLAGS = -s
 LIBS =
 # Linux with glibc, FreeBSD, NetBSD
-#LIBS += -lcrypt
+LIBS += -lcrypt
 # HP-UX trusted system
 #LIBS += -lsec
 # Solaris (POP_STANDALONE, POP_VIRTUAL)
@@ -21,10 +21,10 @@
 # libwrap may also want this
 #LIBS += -lnsl
 # OpenSSL (-DHAVE_OPENSSL)
-#LIBS += -lcrypto
+LIBS += -lcrypto
 
 DESTDIR =
-PREFIX = /usr/local
+PREFIX = /usr
 SBINDIR = $(PREFIX)/sbin
 MANDIR = $(PREFIX)/man
 
diff -Naur popa3d-1.0.orig/params.h popa3d-1.0/params.h
--- popa3d-1.0.orig/params.h	2005-05-26 17:35:15.000000000 +0200
+++ popa3d-1.0/params.h	2005-06-11 20:46:05.000000000 +0200
@@ -13,7 +13,7 @@
 /*
  * Are we going to be a standalone server or start via an inetd clone?
  */
-#define POP_STANDALONE			0
+#define POP_STANDALONE			1
 
 #if POP_STANDALONE
 
@@ -103,13 +103,13 @@
  * A pseudo-user to run as before authentication.  The user and its UID
  * must not be used for any other purpose.
  */
-#define POP_USER			POP_SERVER
+#define POP_USER			"pop"
 
 /*
  * An empty directory to chroot to before authentication.  The directory
  * and its parent directories must not be writable by anyone but root.
  */
-#define POP_CHROOT			"/var/empty"
+#define POP_CHROOT			"/var/spool/pop"
 
 /*
  * Sessions will be closed if idle for longer than POP_TIMEOUT seconds.
@@ -191,7 +191,7 @@
  *
  * #undef this for qmail-style $HOME/Mailbox mailboxes.
  */
-#define MAIL_SPOOL_PATH			"/var/mail"
+#define MAIL_SPOOL_PATH			"/var/spool/mail"
 
 #ifndef MAIL_SPOOL_PATH
 /*
