diff -Naur chkrootkit-0.46.orig/chkrootkit chkrootkit-0.46/chkrootkit
--- chkrootkit-0.46.orig/chkrootkit	2005-10-28 11:16:18.611039500 +0200
+++ chkrootkit-0.46/chkrootkit	2005-10-28 11:29:10.083253500 +0200
@@ -10,6 +10,15 @@
 # (C)1997-2005 Nelson Murilo, Pangeia Informatica, AMS Foundation and others.
 # All rights reserved
 
+# Gentoo specific : Could use `type <command> | cut -f 3 -d " "`
+IFPROMISC="/usr/sbin/ifpromisc"
+CHKLASTLOG="/usr/sbin/chklastlog"
+CHKPROC="/usr/sbin/chkproc"
+CHKWTMP="/usr/sbin/chkwtmp"
+CHKUTMP="/usr/sbin/chkutmp"
+CHECK_WTMPX="/usr/sbin/check_wtmpx"
+STRINGS="/usr/sbin/strings-static"
+
 ### workaround for some Bourne shell implementations
 unalias login > /dev/null 2>&1
 unalias ls > /dev/null 2>&1
@@ -125,7 +134,7 @@
 
     if [ "${EXPERT}" = "t" ]; then
         expertmode_output "${egrep} ^asp ${ROOTDIR}etc/inetd.conf"
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
@@ -137,7 +146,7 @@
         if [ "${QUIET}" != "t" ]; then echo "not infected"; fi
         return ${NOT_INFECTED}
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${ASP_LABEL}" >/dev/null 2>&1; then
+    if ${STRINGS} -a ${CMD} | ${egrep} "${ASP_LABEL}" >/dev/null 2>&1; then
        echo "INFECTED"
        STATUS=${INFECTED}
     else
@@ -158,23 +167,23 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "./ifpromisc" -v
+        expertmode_output "${IFPROMISC}" -v
         return 5
     fi
-    if [ ! -x ./ifpromisc ]; then
-      echo "not tested: can't exec ./ifpromisc"
+    if [ ! -x ${IFPROMISC} ]; then
+      echo "not tested: can't exec ${IFPROMISC}"
       return ${NOT_TESTED}
     else
-      [ "${QUIET}" != "t" ] && ./ifpromisc -v || ./ifpromisc -q
+      [ "${QUIET}" != "t" ] && ${IFPROMISC} -v || ${IFPROMISC} -q
     fi
 }
 
 chkutmp() {
-    if [ ! -x ./chkutmp ]; then
-      echo "not tested: can't exec ./chkutmp"
+    if [ ! -x ${CHKUTMP} ]; then
+      echo "not tested: can't exec ${CHKUTMP}"
       return ${NOT_TESTED}
     fi
-    if ./chkutmp
+    if ${CHKUTMP}
     then
       if [ "${QUIET}" != "t" ]; then echo "chkutmp: nothing deleted"; fi
     fi
@@ -182,8 +191,8 @@
 }
 
 z2 () {
-    if [ ! -x ./chklastlog ]; then
-      echo "not tested: can't exec ./chklastlog"
+    if [ ! -x ${CHKLASTLOG} ]; then
+      echo "not tested: can't exec ${CHKLASTLOG}"
       return ${NOT_TESTED}
     fi
 
@@ -196,7 +205,7 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "./chklastlog -f ${WTMP} -l ${LASTLOG}"
+        expertmode_output "${CHKLASTLOG} -f ${WTMP} -l ${LASTLOG}"
         return 5
     fi
 
@@ -207,21 +216,21 @@
 }
 
 wted () {
-    if [ ! -x ./chkwtmp ]; then
-      echo "not tested: can't exec ./chkwtmp"
+    if [ ! -x ${CHKWTMP} ]; then
+      echo "not tested: can't exec ${CHKWTMP}"
       return ${NOT_TESTED}
     fi
 
    if [ "$SYSTEM" = "SunOS" ]; then
-       if [ ! -x ./check_wtmpx ]; then
-          echo "not tested: can't exec ./check_wtmpx"
+       if [ ! -x ${CHECK_WTMPX} ]; then
+          echo "not tested: can't exec ${CHECK_WTMPX}"
        else
           if [ "${EXPERT}" = "t" ]; then
-             expertmode_output "./check_wtmpx"
+             expertmode_output "${CHECK_WTMPX}"
               return 5
           fi
 	  if [ -f ${ROOTDIR}var/adm/wtmp ]; then
-             if ./check_wtmpx
+             if ${CHECK_WTMPX}
                 then
                 if [ "${QUIET}" != "t" ]; then \
                    echo "check_wtmpx: nothing deleted in /var/adm/wtmpx"; fi
@@ -232,12 +241,12 @@
        WTMP=`loc wtmp wtmp "${ROOTDIR}var/log ${ROOTDIR}var/adm"`
 
        if [ "${EXPERT}" = "t" ]; then
-          expertmode_output "./chkwtmp -f ${WTMP}"
+          expertmode_output "${CHKWTMP} -f ${WTMP}"
           return 5
        fi
     fi
 
-    if ./chkwtmp -f ${WTMP}
+    if ${CHKWTMP} -f ${WTMP}
     then
       if [ "${QUIET}" != "t" ]; then echo "chkwtmp: nothing deleted"; fi
     fi
@@ -276,8 +285,8 @@
     prog=""
     if [  \( "${SYSTEM}" = "Linux"  -o \( "${SYSTEM}" = "FreeBSD" -a \
        `echo ${V} | ${awk} '{ if ($1 > 4.3) print 1; else print 0 }'` -eq 1 \) \) -a "${ROOTDIR}" = "/" ]; then
-      [ ! -x ./chkproc ] && prog="./chkproc"
-      [ ! -x ./chkdirs ] && prog="$prog ./chkdirs"
+      [ ! -x ${CHKPROC} ] && prog="${CHKPROC}"
+      [ ! -x ${CHKDIRS} ] && prog="$prog ${CHKDIRS}"
       if [ "$prog" != "" ]; then
 #        echo "not tested: can't exec $prog"
         return ${NOT_TESTED}
@@ -289,7 +298,7 @@
          PV=`$ps -V 2>/dev/null| $cut -d " " -f 3 |$awk -F . '{ print $1 "." $2 $3 }' | ${awk} '{ if ($0 > 3.19) print 3; else if ($0 < 2.11) print 1; else print 2 }'`
          [ "$PV" = "" ] &&  PV=2
          [ "${SYSTEM}" = "SunOS" ] && PV=0
-         expertmode_output "./chkproc -v -v -p $PV"
+         expertmode_output "${CHKPROC} -v -v -p $PV"
          return 5
       fi
 
@@ -315,7 +324,7 @@
       if [ "${DEBUG}" = "t" ]; then
            ${echo} "*** PV=$PV ***"
       fi
-      if ./chkproc -p ${PV}
+      if ${CHKPROC} -p ${PV}
       then
            if [ "${QUIET}" != "t" ]; then echo "chkproc: nothing detected"; fi
       else
@@ -1173,14 +1182,14 @@
 
     case "${SYSTEM}" in
        Linux)
-          if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" \
+          if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" \
              >/dev/null 2>&1
           then
              STATUS=${INFECTED}
           fi;;
        FreeBSD)
           [ `echo $V | awk '{ if ( $1 >= 5.0) print 1; else print 0 }'` -eq 1 ] && n=1 || n=2
-          if [ `${strings} -a ${CMD} | \
+          if [ `${STRINGS} -a ${CMD} | \
                 ${egrep} -c "${GENERIC_ROOTKIT_LABEL}"` -ne $n ]
           then
              STATUS=${INFECTED}
@@ -1197,16 +1206,16 @@
     REDHAT_PAM_LABEL="*NOT*"
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
     case "${SYSTEM}" in
        Linux)
-          if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" \
+          if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" \
           >/dev/null 2>&1
              then
-             if ${strings} -a ${CMD} | ${egrep} "${REDHAT_PAM_LABEL}" \
+             if ${STRINGS} -a ${CMD} | ${egrep} "${REDHAT_PAM_LABEL}" \
              >/dev/null 2>&1
                 then
                 :
@@ -1216,7 +1225,7 @@
           fi;;
        FreeBSD)
           [ `echo $V | awk '{ if ($1 >= 5.0) print 1; else print 0}'` -eq 1 ] && n=1 || n=2
-          if [ `${strings} -a ${CMD} | ${egrep} -c "${GENERIC_ROOTKIT_LABEL}"` -ne $n ]
+          if [ `${STRINGS} -a ${CMD} | ${egrep} -c "${GENERIC_ROOTKIT_LABEL}"` -ne $n ]
              then
              STATUS=${INFECTED}
           fi;;
@@ -1229,13 +1238,13 @@
     CMD=`loc login login $pth`
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
     if [ "$SYSTEM" = "SunOS" ]; then
       TROJED_L_L="porcao|/bin/xstat"
-      if ${strings} -a ${CMD} | ${egrep} "${TROJED_L_L}" >/dev/null 2>&1 ]; then
+      if ${STRINGS} -a ${CMD} | ${egrep} "${TROJED_L_L}" >/dev/null 2>&1 ]; then
           return ${INFECTED}
        else
           return ${NOT_TESTED}
@@ -1243,7 +1252,7 @@
     fi
     GENERAL="^root$"
     TROJED_L_L="vejeta|^xlogin|^@\(#\)klogin\.c|lets_log|sukasuka|/usr/lib/.ark?|SucKIT"
-    ret=`${strings} -a ${CMD} | ${egrep} -c "${GENERAL}"`
+    ret=`${STRINGS} -a ${CMD} | ${egrep} -c "${GENERAL}"`
     if [ ${ret} -gt 0 ]; then
         case ${ret} in
         1) [ "${SYSTEM}" = "OpenBSD" -a `echo $V | ${awk} '{ if ($1 < 2.7 ||
@@ -1255,7 +1264,7 @@
         *) STATUS=${INFECTED};;
         esac
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${TROJED_L_L}" 2>&1 >/dev/null
+    if ${STRINGS} -a ${CMD} | ${egrep} "${TROJED_L_L}" 2>&1 >/dev/null
        then
         STATUS=${INFECTED}
     fi
@@ -1271,7 +1280,7 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-       expertmode_output "${strings} -a ${CMD}"
+       expertmode_output "${STRINGS} -a ${CMD}"
     fi
 
     if [ "${SYSTEM}" = "OpenBSD" -o "${SYSTEM}" = "SunOS" -o "${SYSTEM}" \
@@ -1279,7 +1288,7 @@
     then
        return ${NOT_TESTED}
     fi
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}|/lib/security" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}|/lib/security" \
     >/dev/null 2>&1
     then
        STATUS=${INFECTED}
@@ -1297,11 +1306,11 @@
     fi
 
     if [ "${EXPERT}" = "t" ]; then
-        expertmode_output "${strings} -a ${CMD}"
+        expertmode_output "${STRINGS} -a ${CMD}"
         return 5
     fi
 
-    if ${strings} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" \
+    if ${STRINGS} -a ${CMD} | ${egrep} "${GENERIC_ROOTKIT_LABEL}" \
     >/dev/null 2>&1
     then
        STATUS=${INFECTED}
@@ -1609,18 +1618,18 @@
 
    if [ "${SYSTEM}" = "Linux" ]
    then
-      if [ ! -x ./strings-static ]; then
-        printn "can't exec ./strings-static, "
+      if [ ! -x ${STRINGS} ]; then
+        printn "can't exec ${STRINGS}, "
         return ${NOT_TESTED}
       fi
 
       if [ "${EXPERT}" = "t" ]; then
-          expertmode_output "./strings-static -a ${CMD}"
+          expertmode_output "${STRINGS} -a ${CMD}"
           return 5
       fi
 
       ### strings must be a statically linked binary.
-      if ./strings-static -a ${CMD} > /dev/null 2>&1
+      if ${STRINGS} -a ${CMD} > /dev/null 2>&1
       then
          STATUS=${INFECTED}
       fi
