diff -Naur KTrafficAnalyzer-0.3.8-1/Makefile KTrafficAnalyzer-0.3.8-1-p/Makefile
--- KTrafficAnalyzer-0.3.8-1/Makefile	2007-06-24 18:32:49.000000000 +0200
+++ KTrafficAnalyzer-0.3.8-1-p/Makefile	2007-07-03 10:38:23.000000000 +0200
@@ -1,34 +1,24 @@
 MAKE		= make
 TARGET		= KTrafficAnalyzer
-OBJ			= settings stats lib
+OBJ		= settings stats lib
 SRC_ROOT	:= $(shell pwd)
-LIB_PATH    = lib`if [ "$(HOSTTYPE)" = "x86_64" ]; then echo "64";  fi`
-LFLAGS		= $(ADD_LFLAGS) -L$(QTDIR)/$(LIB_PATH)/ -L$(KDEDIR)/$(LIB_PATH)/ -lqt-mt -lkdeui -lkio
-CFLAGS		= $(ADD_CFLAGS) -I$(QTDIR)/include/ -I$(KDEDIR)/include/ -I$(SRC_ROOT)
+LFLAGS		+= $(ADD_LFLAGS) -L$(QTDIR)/lib -L$(KDEDIR)/lib -lqt-mt -lkdeui -lkio
+CFLAGS		+= $(ADD_CFLAGS) -I$(QTDIR)/include/ -I$(KDEDIR)/include/ -I$(SRC_ROOT)
 ECHO		= echo
 KDEDIR		:= $(shell kde-config --prefix)
-MOC         = $(QTDIR)/bin/moc
-UIC         = $(QTDIR)/bin/uic
-UBUNTU		:= $(shell lsb_release -a | grep Ubuntu)
-ifdef UBUNTU
-	ADD_CFLAGS += -I/usr/include/kde -I/usr/include/qt3
-	ADD_LFLAGS += -L/usr/lib/qt3/ -L/usr/lib/kde3
-endif
+MOC         	= $(QTDIR)/bin/moc
+UIC         	= $(QTDIR)/bin/uic
 
 all:
-	if test "$(KDEDIR)" = ""; then $(ECHO) KDEDIR not set!; exit 1; fi;
-	if test "$(QTDIR)" = ""; then $(ECHO) QTDIR not set!; exit 1; fi;
+	
 	for i in $(OBJ); do $(MAKE) -C $$i SRC_ROOT=$(SRC_ROOT) "CFLAGS=$(CFLAGS)" "MOC=$(MOC)" "UIC=$(UIC)"; done
 	gcc -c KTrafficAnalyzer.cpp $(CFLAGS)
 	gcc -o $(TARGET) *.o $(addsuffix /*.o,$(OBJ)) $(LFLAGS)
 
 install:
-	if test "$(KDEDIR)" = ""; then $(ECHO) KDEDIR not set!; exit 1; fi;
-	cp KTrafficAnalyzer $(KDEDIR)/bin/
-	ln -sf $(KDEDIR)/bin/KTrafficAnalyzer $(KDEDIR)/bin/ktrafficanalyzer
+	cp KTrafficAnalyzer $(DESTDIR)$(KDEDIR)/bin/
 
 uninstall:
-	if test "$(KDEDIR)" = ""; then $(ECHO) KDEDIR not set!; exit 1; fi;
 	rm $(KDEDIR)/bin/ktrafficanalyzer
 	rm $(KDEDIR)/bin/KTrafficAnalyzer
 
