--- hwdata-0.363/Makefile	2022-10-03 10:54:32.000000000 +0200
+++ hwdata-0.363.new/Makefile	2022-10-06 10:02:55.081946527 +0200
@@ -52,8 +52,8 @@
 		mkdir -p -m 755 $(DESTDIR)$(libdir)/modprobe.d ;\
 		install -m 644 -T blacklist.conf $(DESTDIR)$(libdir)/modprobe.d/dist-blacklist.conf ;\
 	fi;
-	mkdir -p -m 755 $(DESTDIR)$(datadir)/pkgconfig
-	install -m 644 hwdata.pc $(DESTDIR)$(datadir)/pkgconfig/
+	mkdir -p -m 755 $(DESTDIR)$(libdir)/pkgconfig
+	install -m 644 hwdata.pc $(DESTDIR)$(libdir)/pkgconfig/
 
 commit:
 	git commit -vas ||:
