--- iotop-1.25/Makefile	2023-09-03 23:09:01.000000000 +0200
+++ iotop-1.25.new/Makefile	2024-01-15 16:45:09.856741018 +0100
@@ -102,12 +102,12 @@
 	$(E) STRIP $(TARGET)
 	$(Q)$(STRIP) $(TARGET)
 	$(E) INSTALL $(TARGET)
-	$(Q)$(INSTALL) -D -m 0755 $(TARGET) $(PREFIX)/sbin/$(TARGET)
+	$(Q)$(INSTALL) -D -m 0755 $(TARGET) $(PREFIX)/bin/$(TARGET)
 	$(Q)$(INSTALL) -D -m 0644 iotop.8 $(PREFIX)/share/man/man8/iotop.8
 
 uninstall:
 	$(E) UNINSTALL $(TARGET)
-	$(Q)rm -f $(PREFIX)/sbin/$(TARGET)
+	$(Q)rm -f $(PREFIX)/bin/$(TARGET)
 	$(Q)rm -f $(PREFIX)/share/man/man8/iotop.8
 
 bld/.mkdir:
