Index: libopenobex-1.2/ircp/ircp_io.c
===================================================================
--- libopenobex-1.2.orig/ircp/ircp_io.c	2006-05-10 00:31:25.000000000 +0200
+++ libopenobex-1.2/ircp/ircp_io.c	2006-05-10 00:48:06.000000000 +0200
@@ -126,13 +126,20 @@
 	if(ircp_nameok(name) == FALSE)
 		return -1;
 
-	//TODO! Rename file if already exist.
+	if (path == NULL || strnlen(path,sizeof(diskname)) == 0)
+	        path = ".";
+	if (snprintf(diskname, sizeof(diskname), "%s/%s", path, name) >= sizeof(diskname))
+	        return -1;
+
+	/* never overwrite an existing file */
+	fd = open(diskname, O_RDWR | O_CREAT | O_EXCL, DEFFILEMODE);
+	if (fd < 0 &&
+	    snprintf(diskname, sizeof(diskname), "%s/%s_XXXXXX", path, name) < sizeof(diskname))
+	        fd = mkstemp(diskname);
 
-	snprintf(diskname, MAXPATHLEN, "%s/%s", path, name);
+	if (fd >= 0)
+	        DEBUG(4, "Creating file %s\n", diskname);
 
-	DEBUG(4, "Creating file %s\n", diskname);
-
-	fd = open(diskname, O_RDWR | O_CREAT | O_TRUNC, DEFFILEMODE);
 	return fd;
 }
 
@@ -150,7 +157,10 @@
 			return -1;
 	}
 
-	snprintf(newpath, MAXPATHLEN, "%s/%s", path, dir);
+	if (strnlen(path,sizeof(newpath)) != 0)
+		snprintf(newpath, sizeof(newpath), "%s/%s", path, dir);
+	else
+		strncpy(newpath, dir, sizeof(newpath));
 
 	DEBUG(4, "path = %s dir = %s, flags = %d\n", path, dir, flags);
 	if(stat(newpath, &statbuf) == 0) {
