From 9b92428cb7b423751a61a51d5934a58051d38318 Mon Sep 17 00:00:00 2001
From: Jan Sundermeyer <sunderme@gmx.de>
Date: Tue, 1 Apr 2025 19:45:26 +0200
Subject: [PATCH] set higher minimum cmake version

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1972b0a0c..5e39225fb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.1.0)
+cmake_minimum_required(VERSION 3.10.0)
 
 project(TeXstudio VERSION 4.8.7 LANGUAGES C CXX)
 SET(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake ${CMAKE_MODULE_PATH})
