diff -ur a/src/utils_glx.h b/src/utils_glx.h
--- a/src/utils_glx.h	2012-10-05 10:02:58.000000000 -0500
+++ b/src/utils_glx.h	2012-10-26 22:00:13.039341049 -0500
@@ -27,6 +27,13 @@
 #include <GL/glxext.h>
 #include <vdpau/vdpau.h>
 
+#if GL_GLEXT_VERSION >= 85
+/* XXX: PFNGLMULTITEXCOORD2FPROC got out of the GL_VERSION_1_3_DEPRECATED
+   block and is not defined if GL_VERSION_1_3 is defined in <GL/gl.h>
+   Redefine the type here as an interim solution */
+typedef void (*PFNGLMULTITEXCOORD2FPROC) (GLenum target, GLfloat s, GLfloat t);
+#endif
+
 /* GL_NV_vdpau_interop */
 #if GL_GLEXT_VERSION < 64
 typedef GLintptr GLvdpauSurfaceNV;
