diff -ur a/src/vdpau_dump.c b/src/vdpau_dump.c
--- a/src/vdpau_dump.c	2012-10-05 10:02:58.000000000 -0500
+++ b/src/vdpau_dump.c	2013-07-12 14:21:19.990069043 -0500
@@ -59,8 +59,6 @@
         _(VAEncSequenceParameterBufferType);
         _(VAEncPictureParameterBufferType);
         _(VAEncSliceParameterBufferType);
-        _(VAEncH264VUIBufferType);
-        _(VAEncH264SEIBufferType);
 #endif
 #if VA_CHECK_VERSION(0,31,1)
         _(VAQMatrixBufferType);
