diff -Npaur cups-1.1.23/pdftops/GfxFont.cxx cups-1.1.23-patched/pdftops/GfxFont.cxx
--- cups-1.1.23/pdftops/GfxFont.cxx	2004-02-02 23:41:09.000000000 +0100
+++ cups-1.1.23-patched/pdftops/GfxFont.cxx	2005-01-28 02:04:15.613249488 +0100
@@ -969,7 +969,7 @@ Gushort *Gfx8BitFont::getCodeToGIDMap(Tr
   if ((nameToGID = ff->getNameToGID())) {
     for (i = 0; i < 256; ++i) {
       if (!map[i] && (charName = enc[i])) {
-	map[i] = (Gushort)(int)nameToGID->lookup(charName);
+	map[i] = (Gushort)(long)nameToGID->lookup(charName);
       }
     }
     delete nameToGID;
