Index: mozilla/security/manager/ssl/src/nsKeygenHandler.cpp
===================================================================
--- mozilla.orig/security/manager/ssl/src/nsKeygenHandler.cpp	2005-07-20 15:31:22.000000000 -0400
+++ mozilla/security/manager/ssl/src/nsKeygenHandler.cpp	2007-11-12 18:10:29.000000000 -0500
@@ -75,6 +75,15 @@
     { nsnull, 0 }, 
 };
 
+DERTemplate SECAlgorithmIDTemplate[] = {
+    { DER_SEQUENCE,
+	  0, NULL, sizeof(SECAlgorithmID) },
+    { DER_OBJECT_ID,
+	  offsetof(SECAlgorithmID,algorithm), },
+    { DER_OPTIONAL | DER_ANY,
+	  offsetof(SECAlgorithmID,parameters), },
+    { 0, }
+};
 
 DERTemplate CERTSubjectPublicKeyInfoTemplate[] = {
     { DER_SEQUENCE,
@@ -95,16 +104,6 @@
     { 0, }
 };
 
-DERTemplate SECAlgorithmIDTemplate[] = {
-    { DER_SEQUENCE,
-	  0, NULL, sizeof(SECAlgorithmID) },
-    { DER_OBJECT_ID,
-	  offsetof(SECAlgorithmID,algorithm), },
-    { DER_OPTIONAL | DER_ANY,
-	  offsetof(SECAlgorithmID,parameters), },
-    { 0, }
-};
-
 const SEC_ASN1Template SECKEY_PQGParamsTemplate[] = {
     { SEC_ASN1_SEQUENCE, 0, NULL, sizeof(PQGParams) },
     { SEC_ASN1_INTEGER, offsetof(PQGParams,prime) },
