--- yaml-0.11.8.0/yaml.cabal	2022-02-24 13:18:58.000000000 +0100
+++ yaml-0.11.8.0.new/yaml.cabal	2022-11-23 19:02:08.523273567 +0100
@@ -1,6 +1,6 @@
 cabal-version: 1.12
 
--- This file has been generated from package.yaml by hpack version 0.34.4.
+-- This file has been generated from package.yaml by hpack version 0.35.0.
 --
 -- see: https://github.com/sol/hpack
 
@@ -74,7 +74,7 @@
     , filepath
     , libyaml ==0.1.*
     , mtl
-    , resourcet >=0.3 && <1.3
+    , resourcet >=0.3 && <1.4
     , scientific >=0.3
     , template-haskell
     , text
@@ -103,20 +103,20 @@
     , filepath
     , libyaml ==0.1.*
     , mtl
-    , resourcet >=0.3 && <1.3
+    , resourcet >=0.3 && <1.4
     , scientific >=0.3
     , template-haskell
     , text
     , transformers >=0.1
     , unordered-containers
     , vector
+  default-language: Haskell2010
   if flag(no-examples)
     buildable: False
   else
     build-depends:
         raw-strings-qq
       , yaml
-  default-language: Haskell2010
 
 executable json2yaml
   main-is: json2yaml.hs
@@ -138,7 +138,7 @@
     , libyaml ==0.1.*
     , mtl
     , optparse-applicative
-    , resourcet >=0.3 && <1.3
+    , resourcet >=0.3 && <1.4
     , scientific >=0.3
     , template-haskell
     , text
@@ -146,9 +146,9 @@
     , unordered-containers
     , vector
     , yaml
+  default-language: Haskell2010
   if flag(no-exe)
     buildable: False
-  default-language: Haskell2010
 
 executable yaml2json
   main-is: yaml2json.hs
@@ -174,7 +174,7 @@
     , libyaml ==0.1.*
     , mtl
     , optparse-applicative
-    , resourcet >=0.3 && <1.3
+    , resourcet >=0.3 && <1.4
     , scientific >=0.3
     , template-haskell
     , text
@@ -182,9 +182,9 @@
     , unordered-containers
     , vector
     , yaml
+  default-language: Haskell2010
   if flag(no-exe)
     buildable: False
-  default-language: Haskell2010
 
 test-suite spec
   type: exitcode-stdio-1.0
@@ -214,7 +214,7 @@
     , mockery
     , mtl
     , raw-strings-qq
-    , resourcet >=0.3 && <1.3
+    , resourcet >=0.3 && <1.4
     , scientific >=0.3
     , template-haskell
     , temporary
