--- evolution-data-server-1.7.92/camel/camel-filter-driver.c.fix-fd-leak	2006-08-29 17:17:18.000000000 -0400
+++ evolution-data-server-1.7.92/camel/camel-filter-driver.c	2006-08-29 17:17:34.000000000 -0400
@@ -715,6 +715,7 @@
 	GPtrArray *args;
 	child_watch_data_t child_watch_data;
 	GSource *source;
+	GMainContext *context;
 	
 	if (argc < 1 || argv[0]->value.string[0] == '\0')
 		return 0;
@@ -802,7 +803,9 @@
 	camel_object_unref (parser);
 	
  wait:
-	child_watch_data.loop = g_main_loop_new (g_main_context_new (), FALSE);
+	context = g_main_context_new ();
+	child_watch_data.loop = g_main_loop_new (context, FALSE);
+	g_main_context_unref (context);
 
 	source = g_child_watch_source_new (child_pid);
 	g_source_set_callback (source, (GSourceFunc) child_watch, &child_watch_data, NULL);
--- evolution-data-server-1.7.92/camel/camel-filter-search.c.fix-fd-leak	2006-08-29 17:17:04.000000000 -0400
+++ evolution-data-server-1.7.92/camel/camel-filter-search.c	2006-08-29 17:17:34.000000000 -0400
@@ -533,7 +533,6 @@
 	g_spawn_close_pid (pid);
 
 	child_watch_data->child_status = status;
-
 	g_main_loop_quit (child_watch_data->loop);
 }
 
@@ -549,6 +548,7 @@
 	GPtrArray *args;
 	child_watch_data_t child_watch_data;
 	GSource *source;
+	GMainContext *context;
 	
 	if (argc < 1 || argv[0]->value.string[0] == '\0')
 		return 0;
@@ -590,7 +590,9 @@
 	camel_stream_flush (stream);
 	camel_object_unref (stream);
 	
-	child_watch_data.loop = g_main_loop_new (g_main_context_new (), FALSE);
+	context = g_main_context_new ();
+	child_watch_data.loop = g_main_loop_new (context, FALSE);
+	g_main_context_unref (context);
 
 	source = g_child_watch_source_new (child_pid);
 	g_source_set_callback (source, (GSourceFunc) child_watch, &child_watch_data, NULL);
