--- evolution-data-server-1.7.92/src/server-logging.c.chain-finalize	2006-08-29 16:30:38.000000000 -0400
+++ evolution-data-server-1.7.92/src/server-logging.c	2006-08-29 16:31:31.000000000 -0400
@@ -58,6 +58,8 @@
 	}
 	g_slist_free (priv->handlers);
 	priv->handlers = NULL;
+
+	G_OBJECT_CLASS (parent_class)->dispose (object);
 }
 
 static void
@@ -66,6 +68,8 @@
 	ServerLogging *logging = SERVER_LOGGING (object);
 
 	g_free (logging->priv);
+
+	G_OBJECT_CLASS (parent_class)->finalize (object);
 }
 
 static void
