diff -Naur gnome-settings-daemon-2.22.1.orig/data/desktop_gnome_font_rendering.schemas.in gnome-settings-daemon-2.22.1/data/desktop_gnome_font_rendering.schemas.in
--- gnome-settings-daemon-2.22.1.orig/data/desktop_gnome_font_rendering.schemas.in	2008-04-10 17:09:07.000000000 +0200
+++ gnome-settings-daemon-2.22.1/data/desktop_gnome_font_rendering.schemas.in	2008-04-10 17:09:40.000000000 +0200
@@ -6,6 +6,7 @@
 	    <applyto>/desktop/gnome/font_rendering/dpi</applyto>
             <owner>gnome</owner>
 	    <type>float</type>
+	    <default>96</default>
             <locale name="C">
 	        <short>DPI</short>
                 <long>
