diff -Naur gnome-settings-daemon-3.26.0/plugins/rfkill/Makefile.am gnome-settings-daemon-3.26.0-p/plugins/rfkill/Makefile.am
--- gnome-settings-daemon-3.26.0/plugins/rfkill/Makefile.am	2017-09-01 14:12:11.000000000 +0200
+++ gnome-settings-daemon-3.26.0-p/plugins/rfkill/Makefile.am	2017-09-29 12:51:14.749699260 +0200
@@ -30,7 +30,7 @@
 org.gnome.SettingsDaemon.Rfkill.desktop: $(desktop_in_files) Makefile
 	$(AM_V_GEN) sed -e "s|\@libexecdir\@|$(libexecdir)|" $< > $@
 
-udevrulesdir = $(prefix)/lib/udev/rules.d
+udevrulesdir = /lib/udev/rules.d
 udevrules_DATA = 61-gnome-settings-daemon-rfkill.rules
 
 EXTRA_DIST = $(udevrules_DATA) $(desktop_in_files)
diff -Naur gnome-settings-daemon-3.26.0/plugins/rfkill/Makefile.in gnome-settings-daemon-3.26.0-p/plugins/rfkill/Makefile.in
--- gnome-settings-daemon-3.26.0/plugins/rfkill/Makefile.in	2017-09-11 20:15:46.000000000 +0200
+++ gnome-settings-daemon-3.26.0-p/plugins/rfkill/Makefile.in	2017-09-29 12:51:14.754699395 +0200
@@ -472,7 +472,7 @@
 desktopdir = $(sysconfdir)/xdg/autostart
 desktop_in_files = org.gnome.SettingsDaemon.Rfkill.desktop.in
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
-udevrulesdir = $(prefix)/lib/udev/rules.d
+udevrulesdir = /lib/udev/rules.d
 udevrules_DATA = 61-gnome-settings-daemon-rfkill.rules
 EXTRA_DIST = $(udevrules_DATA) $(desktop_in_files)
 CLEANFILES = $(desktop_DATA)
