From 79acde1a8f402ffc1c0e1afec196928f8e1164f5 Mon Sep 17 00:00:00 2001
From: Richard Hughes <richard@hughsie.com>
Date: Tue, 30 Aug 2011 07:46:40 +0000
Subject: power: Do not exit gnome-settings-daemon if upower fails to load

---
diff --git a/plugins/power/gsd-power-manager.c b/plugins/power/gsd-power-manager.c
index ed88853..f983890 100644
--- a/plugins/power/gsd-power-manager.c
+++ b/plugins/power/gsd-power-manager.c
@@ -1079,7 +1079,7 @@ engine_coldplug (GsdPowerManager *manager)
         /* get devices from UPower */
         ret = up_client_enumerate_devices_sync (manager->priv->up_client, NULL, &error);
         if (!ret) {
-                g_error ("failed to get device list: %s", error->message);
+                g_warning ("failed to get device list: %s", error->message);
                 g_error_free (error);
                 goto out;
         }
--
cgit v0.9.0.2

