--- hslua-module-path-1.1.0/hslua-module-path.cabal	2001-09-09 03:46:40.000000000 +0200
+++ hslua-module-path-1.1.0.new/hslua-module-path.cabal	2023-12-13 13:22:48.432753812 +0100
@@ -35,7 +35,7 @@
                      , hslua-core        >= 2.1    && < 2.4
                      , hslua-marshalling >= 2.1    && < 2.4
                      , hslua-packaging   >= 2.3    && < 2.4
-                     , text              >= 1.2    && < 2.1
+                     , text              >= 1.2    && < 2.2
 
   ghc-options:         -Wall
                        -Wcompat
