--- configure.orig	2008-08-28 23:15:30.000000000 +0300
+++ configure	2008-08-29 00:00:45.000000000 +0300
@@ -39456,8 +39456,8 @@
 
 # Check for fribidi
 
-   { echo "$as_me:$LINENO: checking for fribidi-config" >&5
-echo $ECHO_N "checking for fribidi-config... $ECHO_C" >&6; }
+   { echo "$as_me:$LINENO: checking for fribidi.pc" >&5
+echo $ECHO_N "checking for fribidi.pc... $ECHO_C" >&6; }
    if test -n "$FRIBIDI_CONFIG"; then
         kde_cv_path="$FRIBIDI_CONFIG";
    else
@@ -39474,22 +39474,20 @@
 	for dir in $PATH; do
 	  dirs="$dirs $dir"
 	done
-	if test -z ""; then  	  dirs="${prefix}/bin ${exec_prefix}/bin /usr/local/bin /opt/local/bin $dirs"
-        else  	  dirs="$dirs ${prefix}/bin ${exec_prefix}/bin /usr/local/bin /opt/local/bin"
-	fi
+    dirs="$dirs ${prefix}/lib/pkgconfig /usr/lib/pkgconfig"
 	IFS=$kde_save_IFS
 
         for dir in $dirs; do
-	  if test -x "$dir/fribidi-config"; then
+	  if test -e "$dir/fribidi.pc"; then
 	    if test -n ""
 	    then
-              evalstr="$dir/fribidi-config  2>&1 "
+              evalstr="/usr/bin/pkg-config fribidi  2>&1 "
 	      if eval $evalstr; then
-                kde_cv_path="$dir/fribidi-config"
+                kde_cv_path="/usr/bin/pkg-config fribidi"
                 break
 	      fi
-            else
-		kde_cv_path="$dir/fribidi-config"
+        else
+        kde_cv_path="/usr/bin/pkg-config fribidi"
                 break
 	    fi
           fi
@@ -39500,9 +39498,6 @@
 
 fi
 
-
-      eval "kde_cv_path=\"`echo '$kde_cv_path_'$kde_cache`\""
-
    fi
 
    if test -z "$kde_cv_path" || test "$kde_cv_path" = NONE; then
diff -ru kdegraphics-3.5.10/ksvg/impl/libs/libtext2path/src/Converter.cpp kdegraphics-3.5.10-orig/ksvg/impl/libs/libtext2path/src/Converter.cpp
--- ksvg/impl/libs/libtext2path/src/Converter.cpp	2008-08-29 00:12:09.000000000 +0300
+++ ksvg/impl/libs/libtext2path/src/Converter.cpp	2008-08-29 00:11:20.000000000 +0300
@@ -23,7 +23,7 @@
 #include "myboost/shared_ptr.hpp"
 #include <fontconfig/fontconfig.h>
 #include <fribidi/fribidi.h>
-#include <fribidi/fribidi_types.h>
+#include <fribidi/fribidi-types.h>
 
 #include "Font.h"
 #include "Glyph.h"
