diff -Naur libquvi-0.9.4/configure.ac libquvi-0.9.4.new/configure.ac
--- libquvi-0.9.4/configure.ac	2013-11-04 13:55:25.000000000 +0100
+++ libquvi-0.9.4.new/configure.ac	2020-08-07 09:55:47.863272100 +0200
@@ -170,8 +170,6 @@
   doc/dox/Makefile
   doc/dox/Doxyfile
   doc/Makefile
-  examples/Makefile
-  examples/lib/Makefile
   src/misc/Makefile
   src/gcrypt/Makefile
   src/curl/Makefile
@@ -182,8 +180,6 @@
   src/quvi-0.9/quvi/Makefile
   src/quvi-0.9/Makefile
   src/Makefile
-  tests/Makefile
-  tests/lib/Makefile
   libquvi-0.9.pc
   po/Makefile.in])
 AC_OUTPUT
diff -Naur libquvi-0.9.4/Makefile.am libquvi-0.9.4.new/Makefile.am
--- libquvi-0.9.4/Makefile.am	2013-11-04 13:55:25.000000000 +0100
+++ libquvi-0.9.4.new/Makefile.am	2020-08-07 09:56:08.816425669 +0200
@@ -4,7 +4,7 @@
 pkgconfigdir=	$(libdir)/pkgconfig
 pkgconfig_DATA=	libquvi-0.9.pc
 
-SUBDIRS=po src examples doc tests
+SUBDIRS=po src doc
 
 .PHONY: doc distdoc distdoc-html distdoc-latex distdoc-pdf ChangeLog VERSION test test-mem
 
diff -Naur libquvi-0.9.4/Makefile.in libquvi-0.9.4.new/Makefile.in
--- libquvi-0.9.4/Makefile.in	2013-11-10 08:45:46.000000000 +0100
+++ libquvi-0.9.4.new/Makefile.in	2020-08-07 09:55:21.855081502 +0200
@@ -342,7 +342,7 @@
 EXTRA_DIST = libquvi-0.9.pc.in gen-ver.sh VERSION
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libquvi-0.9.pc
-SUBDIRS = po src examples doc tests
+SUBDIRS = po src doc
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
