diff -Naur lua-5.1.orig/etc/lua.pc lua-5.1/etc/lua.pc
--- lua-5.1.orig/etc/lua.pc	2006-02-07 17:54:19.000000000 +0100
+++ lua-5.1/etc/lua.pc	2006-03-01 14:56:06.000000000 +0100
@@ -6,7 +6,7 @@
 V= 5.1
 
 # grep '^INSTALL_.*=' ../Makefile | sed 's/INSTALL_TOP/prefix/'
-prefix= /usr/local
+prefix= /usr
 INSTALL_BIN= ${prefix}/bin
 INSTALL_INC= ${prefix}/include
 INSTALL_LIB= ${prefix}/lib
diff -Naur lua-5.1.orig/Makefile lua-5.1/Makefile
--- lua-5.1.orig/Makefile	2006-02-16 16:45:17.000000000 +0100
+++ lua-5.1/Makefile	2006-03-01 14:56:52.000000000 +0100
@@ -11,7 +11,7 @@
 # if INSTALL_TOP is not an absolute path. (Man pages are installed from the
 # doc directory.)
 #
-INSTALL_TOP= /usr/local
+INSTALL_TOP= /usr
 INSTALL_BIN= $(INSTALL_TOP)/bin
 INSTALL_INC= $(INSTALL_TOP)/include
 INSTALL_LIB= $(INSTALL_TOP)/lib
diff -Naur lua-5.1.orig/src/luaconf.h lua-5.1/src/luaconf.h
--- lua-5.1.orig/src/luaconf.h	2006-02-10 18:44:06.000000000 +0100
+++ lua-5.1/src/luaconf.h	2006-03-01 14:55:53.000000000 +0100
@@ -82,7 +82,7 @@
 	".\\?.dll;"  LUA_CDIR"?.dll;" LUA_CDIR"loadall.dll"
 
 #else
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"/usr/"
 #define LUA_LDIR	LUA_ROOT "share/lua/5.1/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/5.1/"
 #define LUA_PATH_DEFAULT  \
