--- FrugalBuild.orig	2004-05-12 12:53:47.000000000 +0200
+++ FrugalBuild	2004-05-12 13:25:15.000000000 +0200
@@ -3,20 +3,19 @@
 # Maintainer: VMiklos <mamajom@axelero.hu>
 
 pkgname=kernel-headers
-pkgver=2.4.26
+pkgver=2.6.6
 pkgrel=1
 pkgdesc="Linux kernel include files"
 url="http://www.kernel.org"
 depends=()
-up2date=`lynx -dump http://www.kernel.org/kdist/finger_banner |tr -s ' '|grep 2.4|sed -n '1 p'|cut -d ' ' -f 10`
-source=(http://www.kernel.org/pub/linux/kernel/v2.4/linux-$pkgver.tar.bz2)
-md5sums=('88d7aefa03c92739cb70298a0b486e2c')
+up2date=`lynx -dump http://www.kernel.org/kdist/finger_banner |tr -s ' ' |cut -d ' ' -f 10 |sed -n '1 p'`
+source=(ftp://ftp.kernel.org/pub/linux/kernel/v2.6/linux-$pkgver.tar.bz2)
+md5sums=('5218790bc3db41e77a7422969639a9ad')
 
 build() {
   cd $startdir/src/linux-$pkgver/include
   mkdir -p $startdir/pkg/usr/include/
   mv linux $startdir/pkg/usr/include/
-  mv asm-i386 $startdir/pkg/usr/include/
-  cd $startdir/pkg/usr/include/
-  ln -s asm-i386 asm
+  mv asm-i386 $startdir/pkg/usr/include/asm
+  mv asm-generic $startdir/pkg/usr/include/
 }
