diff -ur a/agent/pseudotcp.c b/agent/pseudotcp.c
--- a/agent/pseudotcp.c	2014-03-06 19:21:05.000000000 -0600
+++ b/agent/pseudotcp.c	2014-09-29 05:06:19.834525655 -0500
@@ -946,24 +946,24 @@
     return FALSE;
   }
 
-  if (*timeout == 0 || *timeout < now)
+  if (*timeout == 0 || *timeout < ((long) now))
     *timeout = now + CLOSED_TIMEOUT;
 
   if (priv->state == TCP_CLOSED) {
-    *timeout = min (*timeout, now + CLOSED_TIMEOUT);
+    *timeout = min (*timeout, ((long) now) + CLOSED_TIMEOUT);
     return TRUE;
   }
 
-  *timeout = min (*timeout, now + DEFAULT_TIMEOUT);
+  *timeout = min (*timeout, ((long) now) + DEFAULT_TIMEOUT);
 
   if (priv->t_ack) {
-    *timeout = min(*timeout, priv->t_ack + priv->ack_delay);
+    *timeout = min(*timeout, (long) (priv->t_ack + priv->ack_delay));
   }
   if (priv->rto_base) {
-    *timeout = min(*timeout, priv->rto_base + priv->rx_rto);
+    *timeout = min(*timeout, (long) (priv->rto_base + priv->rx_rto));
   }
   if (priv->snd_wnd == 0) {
-    *timeout = min(*timeout, priv->lastsend + priv->rx_rto);
+    *timeout = min(*timeout, (long) (priv->lastsend + priv->rx_rto));
   }
 
   return TRUE;
