--- rocprofiler-register-rocm-6.4.0/CMakeLists.txt	2025-02-18 01:53:49.000000000 +0100
+++ rocprofiler-register-rocm-6.4.0.new/CMakeLists.txt	2025-04-21 12:34:51.936528048 +0200
@@ -100,7 +100,8 @@
 enable_testing()
 include(CTest)
 
-add_subdirectory(external)
+find_package(fmt REQUIRED)
+find_package(glog REQUIRED)
 add_subdirectory(source)
 
 include(rocprofiler_register_config_install)
@@ -113,6 +114,5 @@
     add_subdirectory(samples)
 endif()
 
-include(rocprofiler_register_config_packaging)
 
 rocprofiler_register_print_features()
