--- GNUmakefile.orig	2011-07-01 15:16:33.000000000 +0200
+++ GNUmakefile	2011-07-01 15:17:17.000000000 +0200
@@ -49,7 +49,7 @@
 
 BUNDLE_NAME=
 
-BUNDLE_INSTALL_DIR=$(GNUSTEP_BUNDLES)/SQLClient
+BUNDLE_INSTALL_DIR=$(DESTDIR)/$(GNUSTEP_SYSTEM_ROOT)/Library/Bundles/SQLClient
 
 # In some systems and situations the dynamic linker needs to haved the
 # SQLClient, gnustep-base, and objc libraries explicityly linked into
