diff -ur a/src/vesa.c b/src/vesa.c
--- a/src/vesa.c	2012-07-18 05:02:49.000000000 -0500
+++ b/src/vesa.c	2013-07-12 13:23:52.075197238 -0500
@@ -50,7 +50,6 @@
 #include "mipointer.h"
 
 /* All drivers implementing backing store need this */
-#include "mibstore.h"
 
 /* Colormap handling */
 #include "micmap.h"
@@ -1081,7 +1080,6 @@
     VESADGAInit(pScrn, pScreen);
 
     xf86SetBlackWhitePixels(pScreen);
-    miInitializeBackingStore(pScreen);
     xf86SetBackingStore(pScreen);
 
     /* software cursor */
