From f8412d7c5d62bde3c03b7a40734d74fbcebc8beb Mon Sep 17 00:00:00 2001
From: Badlop <badlop@process-one.net>
Date: Fri, 9 Jul 2010 20:18:59 +0200
Subject: [PATCH] When using OTP R14, use public_key library instead of old ssl (EJAB-953)

---

diff -u b/src/ejabberd_s2s_in.erl b/src/ejabberd_s2s_in.erl
--- b/src/ejabberd_s2s_in.erl
+++ b/src/ejabberd_s2s_in.erl
@@ -48,9 +48,9 @@
 
 -include("ejabberd.hrl").
 -include("jlib.hrl").
--include_lib("ssl/include/ssl_pkix.hrl").
--define(PKIXEXPLICIT, 'OTP-PKIX').
--define(PKIXIMPLICIT, 'OTP-PKIX').
+-include_lib("public_key/include/public_key.hrl"). 
+-define(PKIXEXPLICIT, 'OTP-PUB-KEY').
+-define(PKIXIMPLICIT, 'OTP-PUB-KEY').
 -include("XmppAddr.hrl").
 
 -define(DICT, dict).
--- a/src/tls/tls.erl
+++ b/src/tls/tls.erl
@@ -232,7 +232,7 @@ close(#tlssock{tcpsock = TCPSocket, tlsport = Port}) ->
 get_peer_certificate(#tlssock{tlsport = Port}) ->
     case port_control(Port, ?GET_PEER_CERTIFICATE, []) of
 	<<0, BCert/binary>> ->
-	    case catch ssl_pkix:decode_cert(BCert, [pkix]) of
+		case catch public_key:pkix_decode_cert(BCert, [pkix]) of
 		{ok, Cert} ->
 		    {ok, Cert};
 		_ ->
