diff -Naur marlin_build_los-linspireclear-style-1.orig/admin/cvs.sh marlin_build_los-linspireclear-style-1/admin/cvs.sh
--- marlin_build_los-linspireclear-style-1.orig/admin/cvs.sh	2005-02-18 03:45:07.000000000 +0100
+++ marlin_build_los-linspireclear-style-1/admin/cvs.sh	2005-07-31 16:30:58.000000000 +0200
@@ -77,7 +77,6 @@
   * )
     echo "*** YOU'RE USING $AUTOMAKE_STRING."
     echo "*** KDE requires automake 1.6"
-    exit 1
     ;;
 esac
 }
