diff -Naur ROX-Menu.orig/src/menu_applet.c ROX-Menu/src/menu_applet.c
--- ROX-Menu.orig/src/menu_applet.c	2005-06-04 13:31:53.000000000 +0200
+++ ROX-Menu/src/menu_applet.c	2005-06-04 13:31:26.000000000 +0200
@@ -324,7 +324,7 @@
 		exit(1);
 	}
 	dest_dir = g_build_path(G_DIR_SEPARATOR_S, menu_dir_path, ("SystemApps"), NULL);
-	ret = symlink("/usr/apps", dest_dir);
+	ret = symlink("/usr/share/Apps", dest_dir);
 	if (ret == 0) status = 0;
 	g_free(dest_dir);
 	dest_dir = g_build_path(G_DIR_SEPARATOR_S, menu_dir_path, ("LocalApps"), NULL);
