diff -Naur lumina-1.3.0-p1/src-qt5/core/lumina-desktop/fluxboxconf/fluxbox-init-rc lumina-1.3.0-p1-fw/src-qt5/core/lumina-desktop/fluxboxconf/fluxbox-init-rc
--- lumina-1.3.0-p1/src-qt5/core/lumina-desktop/fluxboxconf/fluxbox-init-rc	2017-06-30 14:14:48.000000000 +0200
+++ lumina-1.3.0-p1-fw/src-qt5/core/lumina-desktop/fluxboxconf/fluxbox-init-rc	2017-09-16 19:09:47.723535840 +0200
@@ -66,7 +66,7 @@
 session.titlebar.right:	Minimize Maximize Close
 session.colorsPerChannel:	4
 session.opaqueMove:	False
-session.styleFile:	/usr/local/share/fluxbox/styles/bias_allblack
+session.styleFile:	/usr/share/fluxbox/styles/BlueNight
 session.doubleClickInterval:	250
 session.iconbar:	false
 session.cacheLife:	5
@@ -78,4 +78,4 @@
 session.autoRaiseDelay:	250
 session.cacheMax:	200
 session.imageDither:	false
-session.forcePseudoTransparency:	false
+session.forcePseudoTransparency:	false
