diff -Naur knemo-0.4.6/src/knemod/knemod.desktop knemo-0.4.6-p/src/knemod/knemod.desktop
--- knemo-0.4.6/src/knemod/knemod.desktop	2006-10-03 21:54:25.000000000 +0200
+++ knemo-0.4.6-p/src/knemod/knemod.desktop	2007-02-25 05:56:09.000000000 +0100
@@ -5,7 +5,8 @@
 X-KDE-ModuleType=Library
 X-KDE-Library=knemod
 X-KDE-FactoryName=knemod
-X-KDE-Kded-load-on-demand=true
+X-KDE-Kded-autoload=true
+X-KDE-Kded-load-on-demand=false
 Name=KNemo
 Name[sv]=Knemo
 Name[xx]=xxKNemoxx
