--- a/src/utils/confparser.h.orig	2008-02-22 01:42:09.000000000 -0700
+++ b/src/utils/confparser.h	2008-09-30 18:24:10.000000000 -0600
@@ -25,6 +25,8 @@
 #ifndef _CONFPARSER_H_
 #define _CONFPARSER_H_
 
+#include <string.h>
+
 #include<iostream>
 #include<cstring>
 #include<string>
--- a/src/kernel/cpdf.cc
+++ b/src/kernel/cpdf.cc
@@ -2347,7 +2347,7 @@ using namespace std;
 		instance->file = file;
 		kernelPrintDbg(debug::DBG_INFO, "Instance created successfully openMode=" << openMode);
 		return instance;
-	}catch(exception &e)
+	}catch(std::exception &e)
 	{
 		kernelPrintDbg(DBG_CRIT, "Pdf instance creation failed. cause="<<e.what());
 		string what=string("CPdf open failed. reason=")+e.what();
@@ -2631,7 +2631,7 @@ using namespace utils;
 		try
 		{
 			minPos = getNodePosition(*this, newValue, &nodeCountCache);
-		}catch(exception &e)
+		}catch(std::exception &e)
 		{
 			// position can't be determined
 			// no special handling is needed, minPos keeps its value
@@ -2690,7 +2690,7 @@ using namespace utils;
 				kernelPrintDbg(DBG_WARN, "page with original position="<<i->first<<" is ambiguous. Invalidating.");
 				// page position is ambiguous and so it has to be invalidate
 				i->second->invalidate();
-			}catch(exception & e)
+			}catch(std::exception & e)
 			{
 				kernelPrintDbg(DBG_CRIT, "Unexpected error. cause="<<e.what());
 				assert(!"Possibly bug.");
