diff -Naur kchmviewer-2.5/src/iconstorage.h kchmviewer-2.5-gcc41/src/iconstorage.h
--- kchmviewer-2.5/src/iconstorage.h	2006-04-02 04:09:14.000000000 +0200
+++ kchmviewer-2.5-gcc41/src/iconstorage.h	2006-04-05 19:05:16.000000000 +0200
@@ -57,7 +57,7 @@
 	const QPixmap * getBookIconPixmap (unsigned int id);
 	const QPixmap * getToolbarPixmap (pixmap_index_t pix);
 	const QPixmap * getApplicationIcon();
-	const QPixmap * KCHMIconStorage::getCloseWindowIcon();
+	const QPixmap * getCloseWindowIcon();
 
 private:
 	const QPixmap * returnOrLoadImage (unsigned int id, const png_memory_image_t * image);
