diff -Naur kchmviewer-1.1.orig/kio-msits/msits.h kchmviewer-1.1/kio-msits/msits.h
--- kchmviewer-1.1.orig/kio-msits/msits.h	2005-11-11 18:32:07.368732000 +0100
+++ kchmviewer-1.1/kio-msits/msits.h	2005-11-11 18:32:20.069525750 +0100
@@ -56,7 +56,7 @@
 	}
 
 	// Retrieve an object from the CHM file
-	inline size_t RetrieveObject (const chmUnitInfo *ui, unsigned char *buffer, LONGUINT64 fileOffset, LONGINT64 bufferSize)
+	inline size_t RetrieveObject (chmUnitInfo *ui, unsigned char *buffer, LONGUINT64 fileOffset, LONGINT64 bufferSize)
 	{
 		return ::chm_retrieve_object(m_chmFile, ui, buffer, fileOffset, bufferSize);
 	}
diff -Naur kchmviewer-1.1.orig/src/kchmsearchtoolbar.cpp kchmviewer-1.1/src/kchmsearchtoolbar.cpp
--- kchmviewer-1.1.orig/src/kchmsearchtoolbar.cpp	2005-11-11 18:32:07.360731500 +0100
+++ kchmviewer-1.1/src/kchmsearchtoolbar.cpp	2005-11-11 18:32:12.549055750 +0100
@@ -152,7 +152,7 @@
 					connect (menu_sublang, SIGNAL( activated(int) ), this, SLOT ( onMenuActivated(int) ));
 				}
 					
-				menuid = menu_sublang->insertItem( item->country, (int) item );
+				menuid = menu_sublang->insertItem( item->country, (long) item );
 				continue;
 			}
 		}
@@ -163,11 +163,11 @@
 		if ( menu_sublang )
 		{
 			menu_sublang->insertItem( item->country );
-			menuid = menu_enclist->insertItem( item->charset, menu_sublang, (int) item );
+			menuid = menu_enclist->insertItem( item->charset, menu_sublang, (long) item );
 			menu_sublang = 0;
 		}
 		else
-			menuid = menu_enclist->insertItem( item->charset, (int) item );
+			menuid = menu_enclist->insertItem( item->charset, (long) item );
 	}
 
 	menu_view->insertItem( tr("&Set encoding"), menu_enclist );
@@ -252,8 +252,8 @@
 void KCHMSearchAndViewToolbar::setChosenEncodingInMenu( const KCHMTextEncoding::text_encoding_t * enc)
 {
 	if ( m_checkedEncodingInMenu )
-		menu_enclist->setItemChecked ((int)m_checkedEncodingInMenu, false);
+		menu_enclist->setItemChecked ((long)m_checkedEncodingInMenu, false);
 	
-	menu_enclist->setItemChecked ((int)enc, true);
+	menu_enclist->setItemChecked ((long)enc, true);
 	m_checkedEncodingInMenu = enc;
 }
diff -Naur kchmviewer-1.1.orig/src/xchmfile.cpp kchmviewer-1.1/src/xchmfile.cpp
--- kchmviewer-1.1.orig/src/xchmfile.cpp	2005-11-11 18:32:07.356731250 +0100
+++ kchmviewer-1.1/src/xchmfile.cpp	2005-11-11 18:32:12.553056000 +0100
@@ -646,7 +646,7 @@
 }
 
 
-size_t CHMFile::RetrieveObject(const chmUnitInfo *ui, unsigned char *buffer,
+size_t CHMFile::RetrieveObject(chmUnitInfo *ui, unsigned char *buffer,
 							   LONGUINT64 fileOffset, LONGINT64 bufferSize)
 {
 	return ::chm_retrieve_object(m_chmFile, ui, buffer, fileOffset,
@@ -1043,7 +1043,7 @@
 }
 
 
-bool CHMFile::GetFileContentAsString(QString& str, const chmUnitInfo *ui)
+bool CHMFile::GetFileContentAsString(QString& str, chmUnitInfo *ui)
 {
 	QByteArray buf (ui->length + 1);
 			
diff -Naur kchmviewer-1.1.orig/src/xchmfile.h kchmviewer-1.1/src/xchmfile.h
--- kchmviewer-1.1.orig/src/xchmfile.h	2005-11-11 18:32:07.356731250 +0100
+++ kchmviewer-1.1/src/xchmfile.h	2005-11-11 18:32:12.553056000 +0100
@@ -231,10 +231,10 @@
 	  \param bufferSize The size of the buffer.
 	  \return 0 on error, length of chunk retrieved otherwise.
 	 */
-	size_t RetrieveObject(const chmUnitInfo *ui, unsigned char *buffer, LONGUINT64 fileOffset, LONGINT64 bufferSize);
+	size_t RetrieveObject(chmUnitInfo *ui, unsigned char *buffer, LONGUINT64 fileOffset, LONGINT64 bufferSize);
 
 	//! Puts in the str parameter the contents of the file referred by ui.
-	bool GetFileContentAsString (QString& str, const chmUnitInfo *ui);
+	bool GetFileContentAsString (QString& str, chmUnitInfo *ui);
 	
 	//! Puts in the str parameter the contents of the file referred by location.
 	bool GetFileContentAsString (QString& str, QString location);
