diff -Naur kchmviewer-1.1.orig/src/kchmsearchtoolbar.cpp kchmviewer-1.1/src/kchmsearchtoolbar.cpp
--- kchmviewer-1.1.orig/src/kchmsearchtoolbar.cpp	2005-11-11 18:32:07.360731500 +0100
+++ kchmviewer-1.1/src/kchmsearchtoolbar.cpp	2005-11-11 18:32:12.549055750 +0100
@@ -152,7 +152,7 @@
 					connect (menu_sublang, SIGNAL( activated(int) ), this, SLOT ( onMenuActivated(int) ));
 				}
 					
-				menuid = menu_sublang->insertItem( item->country, (int) item );
+				menuid = menu_sublang->insertItem( item->country, (long) item );
 				continue;
 			}
 		}
@@ -163,11 +163,11 @@
 		if ( menu_sublang )
 		{
 			menu_sublang->insertItem( item->country );
-			menuid = menu_enclist->insertItem( item->charset, menu_sublang, (int) item );
+			menuid = menu_enclist->insertItem( item->charset, menu_sublang, (long) item );
 			menu_sublang = 0;
 		}
 		else
-			menuid = menu_enclist->insertItem( item->charset, (int) item );
+			menuid = menu_enclist->insertItem( item->charset, (long) item );
 	}
 
 	menu_view->insertItem( tr("&Set encoding"), menu_enclist );
@@ -252,8 +252,8 @@
 void KCHMSearchAndViewToolbar::setChosenEncodingInMenu( const KCHMTextEncoding::text_encoding_t * enc)
 {
 	if ( m_checkedEncodingInMenu )
-		menu_enclist->setItemChecked ((int)m_checkedEncodingInMenu, false);
+		menu_enclist->setItemChecked ((long)m_checkedEncodingInMenu, false);
 	
-	menu_enclist->setItemChecked ((int)enc, true);
+	menu_enclist->setItemChecked ((long)enc, true);
 	m_checkedEncodingInMenu = enc;
 }
