diff -Naur kchmviewer-1.1.orig/configure kchmviewer-1.1/configure
--- kchmviewer-1.1.orig/configure	2005-11-11 18:21:13.951896000 +0100
+++ kchmviewer-1.1/configure	2005-11-11 18:21:40.209537000 +0100
@@ -31372,8 +31372,8 @@
 
 EXTRA_SUBDIRS=""
 CHMLIBDIR="chmlib"
-CHM_INCLUDES="\$(top_srcdir)/$CHMLIBDIR/"
-CHM_LIBS="\$(top_builddir)/$CHMLIBDIR/libchm.la"
+CHM_INCLUDES="/usr/include/"
+CHM_LIBS="/usr/lib/libchm.la"
 
 
 # Check whether --with-kde or --without-kde was given.
diff -Naur kchmviewer-1.1.orig/kio-msits/msits.h kchmviewer-1.1/kio-msits/msits.h
--- kchmviewer-1.1.orig/kio-msits/msits.h	2005-11-11 18:21:13.955896250 +0100
+++ kchmviewer-1.1/kio-msits/msits.h	2005-11-11 18:22:09.115343500 +0100
@@ -29,7 +29,7 @@
 #include <qstring.h>
 #include <qcstring.h>
 
-#include "chm_lib.h"
+#include <chm_lib.h>
 
 
 class ProtocolMSITS : public KIO::SlaveBase
diff -Naur kchmviewer-1.1.orig/Makefile.am kchmviewer-1.1/Makefile.am
--- kchmviewer-1.1.orig/Makefile.am	2005-11-11 18:21:13.955896250 +0100
+++ kchmviewer-1.1/Makefile.am	2005-11-11 18:22:09.119343750 +0100
@@ -1,4 +1,4 @@
 # don't warn about missing AUTHORS, NEWS, ChangeLog .. files
 AUTOMAKE_OPTIONS = foreign 1.4
 
-SUBDIRS = chmlib src @EXTRA_SUBDIRS@
+SUBDIRS = src @EXTRA_SUBDIRS@
diff -Naur kchmviewer-1.1.orig/Makefile.in kchmviewer-1.1/Makefile.in
--- kchmviewer-1.1.orig/Makefile.in	2005-11-11 18:21:13.955896250 +0100
+++ kchmviewer-1.1/Makefile.in	2005-11-11 18:22:42.629438000 +0100
@@ -324,7 +324,7 @@
 
 # don't warn about missing AUTHORS, NEWS, ChangeLog .. files
 AUTOMAKE_OPTIONS = foreign 1.4
-SUBDIRS = chmlib src @EXTRA_SUBDIRS@
+SUBDIRS = src @EXTRA_SUBDIRS@
 #>- all: all-recursive
 #>+ 1
 all: docs-am  all-recursive
diff -Naur kchmviewer-1.1.orig/src/xchmfile.h kchmviewer-1.1/src/xchmfile.h
--- kchmviewer-1.1.orig/src/xchmfile.h	2005-11-11 18:21:13.943895500 +0100
+++ kchmviewer-1.1/src/xchmfile.h	2005-11-11 18:22:09.119343750 +0100
@@ -39,7 +39,7 @@
 #include "kchmtreeviewitem.h"
 #include "kchmtextencoding.h"
 
-#include "chm_lib.h"
+#include <chm_lib.h>
 
 class KCHMSearchResult
 {
