--- sakura-1.0.6/Makefile.in.old	2006-07-04 22:39:41.000000000 +0530
+++ sakura-1.0.6/Makefile.in		2006-07-04 22:40:00.000000000 +0530
@@ -40,7 +40,7 @@
 install: all
 	@$(call makedir,$(BINDIR),$(DIRMODE))
 	@$(call install_files,src/sakura,$(BINDIR),a=rx)
-	@$(call makedir,/usr/share/pixmaps,$(DIRMODE))
+	@$(call makedir,$(DATADIR)/pixmaps,$(DIRMODE))
 	@$(call install_files,terminal-tango.png,$(DATADIR)/pixmaps,a=rx)
 	@$(call makedir,$(DATADIR)/share/locale,$(DIRMODE))
 	@for i in $(MOFILES); do \
