diff -Naur coriander-2.0.0-rc1.orig/configure.in coriander-2.0.0-rc1/configure.in
--- coriander-2.0.0-rc1.orig/configure.in	2006-08-18 17:48:04.000000000 +0200
+++ coriander-2.0.0-rc1/configure.in	2006-08-18 17:46:41.000000000 +0200
@@ -317,7 +317,7 @@
 AC_DEFUN([AC_CHECK_FFMPEG],[
 	AC_SUBST(FFMPEG_CFLAGS)
 	AC_SUBST(FFMPEG_LIBS)
-	AC_CHECK_PROG(have_ffmpeg_config, ffmpeg-config, "found", "no")
+	AC_CHECK_PROG(have_ffmpeg_config, pkg-config, "found", "no")
 	if test x$have_ffmpeg_config = xfound; then 
 	     AC_MSG_CHECKING(ffmpeg version)    
 	     AC_TRY_COMPILE([
@@ -334,8 +334,8 @@
 	     ],ffmpeg=ok,ffmpeg=outdated);
 	     AC_MSG_RESULT($ffmpeg)
  	     if test x$ffmpeg = xok; then
-	          FFMPEG_LIBS=`ffmpeg-config --libs avformat avcodec`
-	          FFMPEG_CFLAGS=`ffmpeg-config --cflags`
+	          FFMPEG_LIBS=`pkg-config --libs libavformat libavcodec`
+	          FFMPEG_CFLAGS=`pkg-config --cflags libavformat libavcodec`
 	          AC_DEFINE(HAVE_FFMPEG,1,[defined if ffmpeg is available])
 	     else
                   AC_ERROR([ffmpeg is too old. 
