--- vector-stream-0.1.0.0/vector-stream.cabal	2022-06-19 15:11:39.000000000 +0200
+++ vector-stream-0.1.0.0.new/vector-stream.cabal	2023-03-17 09:58:03.863408218 +0100
@@ -43,8 +43,8 @@
   Hs-Source-Dirs:
         src
 
-  Build-Depends: base >= 4.9 && < 4.17
-               , ghc-prim >= 0.2 && < 0.9
+  Build-Depends: base >= 4.9 && < 4.19
+               , ghc-prim >= 0.2 && < 0.11
 
 source-repository head
   type:     git
