--- vector-stream-0.1.0.1/vector-stream.cabal	2001-09-09 03:46:40.000000000 +0200
+++ vector-stream-0.1.0.1.new/vector-stream.cabal	2024-12-25 21:30:10.782073494 +0100
@@ -21,17 +21,17 @@
         as a backbone for vector package fusion functionality.
 
 Tested-With:
-  GHC == 8.0.2,
-  GHC == 8.2.2,
-  GHC == 8.4.4,
-  GHC == 8.6.5,
-  GHC == 8.8.4,
-  GHC == 8.10.4,
-  GHC == 9.0.2,
-  GHC == 9.2.8,
-  GHC == 9.4.6,
-  GHC == 9.6.2
-  GHC == 9.8.1
+  GHC == 8.0.2
+  GHC == 8.2.2
+  GHC == 8.4.4
+  GHC == 8.6.5
+  GHC == 8.8.4
+  GHC == 8.10.7
+  GHC == 9.0.2
+  GHC == 9.2.8
+  GHC == 9.4.8
+  GHC == 9.6.4
+  GHC == 9.8.2
 
 Cabal-Version:  >=1.10
 Build-Type:     Simple
@@ -49,8 +49,8 @@
   Hs-Source-Dirs:
         src
 
-  Build-Depends: base >= 4.9 && < 4.20
-               , ghc-prim >= 0.2 && < 0.12
+  Build-Depends: base >= 4.9 && < 4.22
+               , ghc-prim >= 0.2 && < 0.14
 
 source-repository head
   type:     git
