diff -Naur vector-stream-0.1.0.0/vector-stream.cabal vector-stream-0.1.0.0.new/vector-stream.cabal
--- vector-stream-0.1.0.0/vector-stream.cabal	2022-06-19 15:11:39.000000000 +0200
+++ vector-stream-0.1.0.0.new/vector-stream.cabal	2022-10-12 15:52:55.513880197 +0200
@@ -43,8 +43,8 @@
   Hs-Source-Dirs:
         src
 
-  Build-Depends: base >= 4.9 && < 4.17
-               , ghc-prim >= 0.2 && < 0.9
+  Build-Depends: base >= 4.9 && < 4.18
+               , ghc-prim >= 0.2 && < 0.10
 
 source-repository head
   type:     git
