diff --exclude-from=/home/dang/.scripts/diffrc -up -ruN bug-buddy-2.20.0.orig/gnome-breakpad/gnome-breakpad.cc bug-buddy-2.20.0/gnome-breakpad/gnome-breakpad.cc
--- bug-buddy-2.20.0.orig/gnome-breakpad/gnome-breakpad.cc	2007-09-04 15:53:55.000000000 -0400
+++ bug-buddy-2.20.0/gnome-breakpad/gnome-breakpad.cc	2007-10-02 11:23:54.000000000 -0400
@@ -291,22 +291,59 @@ gtk_module_init (int *argc, char** argv[
 	
 	if (bugbuddy && !g_getenv ("GNOME_DISABLE_CRASH_DIALOG")) {
 #ifdef ENABLE_GOOGLE_BREAKPAD
+        	static struct sigaction old_action;
+
+		sigaction(SIGSEGV, NULL, &old_action);
+		if (old_action.sa_handler != SIG_DFL)
+			return 0;
+
+                sigaction(SIGABRT, NULL, &old_action);
+		if (old_action.sa_handler != SIG_DFL)
+			return 0;
+
+                sigaction(SIGTRAP, NULL, &old_action);
+		if (old_action.sa_handler != SIG_DFL)
+			return 0;
+
+                sigaction(SIGFPE, NULL, &old_action);
+		if (old_action.sa_handler != SIG_DFL)
+			return 0;
+
+                sigaction(SIGBUS, NULL, &old_action);
+		if (old_action.sa_handler != SIG_DFL)
+			return 0;
+
   		static ExceptionHandler handler("/tmp", check_if_gdb,
 						MinidumpCallback, NULL, true);
 #else
         	static struct sigaction *setptr;
+        	static struct sigaction old_action;
         	struct sigaction sa;
                 memset(&sa, 0, sizeof(sa));
                 setptr = &sa;
 
                 sa.sa_handler = bugbuddy_segv_handle;
 
-                sigaction(SIGSEGV, setptr, NULL);
-                sigaction(SIGABRT, setptr, NULL);
-                sigaction(SIGTRAP, setptr, NULL);
-                sigaction(SIGFPE, setptr, NULL);
-                sigaction(SIGBUS, setptr, NULL);
+                sigaction(SIGSEGV, NULL, &old_action);
+		if (old_action.sa_handler == SIG_DFL)
+                	sigaction(SIGSEGV, setptr, NULL);
+
+                sigaction(SIGABRT, NULL, &old_action);
+		if (old_action.sa_handler == SIG_DFL)
+                	sigaction(SIGABRT, setptr, NULL);
+
+                sigaction(SIGTRAP, NULL, &old_action);
+		if (old_action.sa_handler == SIG_DFL)
+                	sigaction(SIGTRAP, setptr, NULL);
+
+                sigaction(SIGFPE, NULL, &old_action);
+		if (old_action.sa_handler == SIG_DFL)
+                	sigaction(SIGFPE, setptr, NULL);
+
+                sigaction(SIGBUS, NULL, &old_action);
+		if (old_action.sa_handler == SIG_DFL)
+                	sigaction(SIGBUS, setptr, NULL);
 #endif
 	}
-  
+	return 0;
 }
