diff -Naur bug-buddy-2.16.0.orig/src/distribution.h bug-buddy-2.16.0/src/distribution.h
--- bug-buddy-2.16.0.orig/src/distribution.h	2006-11-30 00:58:06.000000000 +0100
+++ bug-buddy-2.16.0/src/distribution.h	2006-11-30 00:58:25.000000000 +0100
@@ -57,6 +57,7 @@
 	{ "Red Hat",       "/etc/redhat-release",     &redhat_phy },
 	{ "Fedora",        "/etc/fedora-release",     &redhat_phy },
 	{ "Gentoo",        "/etc/gentoo-release",     &redhat_phy },
+	{ "Frugalware",    "/etc/frugalware-release", &redhat_phy },
 	{ "Solaris",       "/etc/release",  	      &redhat_phy },
 	{ "LSB Linux",     "/etc/lsb-release",        &lsb_phy },
 	{ "Debian",        "/etc/debian_version",     &debian_phy },
