diff -urN bug-buddy-2.17.4/src/distribution.c bug-buddy-2.17.4.new/src/distribution.c
--- bug-buddy-2.17.4/src/distribution.c	2007-01-11 20:13:00.000000000 +0000
+++ bug-buddy-2.17.4.new/src/distribution.c	2007-03-04 17:57:49.000000000 +0000
@@ -82,6 +82,7 @@
 	{ "TurboLinux",    "/etc/turbolinux-release", &turbolinux_phy },
 	{ "SuSE",          "/etc/SuSE-release",       &redhat_phy },
 	{ "Red Hat",       "/etc/redhat-release",     &redhat_phy },
+	{ "Frugalware",    "/etc/frugalware-release", &redhat_phy },
 	{ "Fedora",        "/etc/fedora-release",     &redhat_phy },
 	{ "Gentoo",        "/etc/gentoo-release",     &redhat_phy },
 	{ "Solaris",       "/etc/release",  	      &redhat_phy },
