diff -Naur libdlna-0.2.4/src/audio_aac.c libdlna-0.2.4.new/src/audio_aac.c
--- libdlna-0.2.4/src/audio_aac.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/audio_aac.c	2014-12-19 10:02:53.883958803 +0100
@@ -311,7 +311,7 @@
     return AUDIO_PROFILE_INVALID;
 
   /* check for AAC variants codec */
-  if (ac->codec_id != CODEC_ID_AAC)
+  if (ac->codec_id != AV_CODEC_ID_AAC)
     return AUDIO_PROFILE_INVALID;
   
   switch (type)
diff -Naur libdlna-0.2.4/src/audio_ac3.c libdlna-0.2.4.new/src/audio_ac3.c
--- libdlna-0.2.4/src/audio_ac3.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/audio_ac3.c	2014-12-19 10:02:53.883958803 +0100
@@ -39,7 +39,7 @@
     return AUDIO_PROFILE_INVALID;
 
   /* check for AC3 codec */
-  if (ac->codec_id != CODEC_ID_AC3)
+  if (ac->codec_id != AV_CODEC_ID_AC3)
     return AUDIO_PROFILE_INVALID;
   
   /* supported channels: 1/0, 2/0, 3/0, 2/1, 3/1, 2/2, 3/2 */
diff -Naur libdlna-0.2.4/src/audio_amr.c libdlna-0.2.4.new/src/audio_amr.c
--- libdlna-0.2.4/src/audio_amr.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/audio_amr.c	2014-12-19 10:02:53.884958835 +0100
@@ -53,7 +53,7 @@
   if (!ac)
     return 0;
 
-  if (ac->codec_id != CODEC_ID_AMR_NB)
+  if (ac->codec_id != AV_CODEC_ID_AMR_NB)
     return 0;
   
   /* only mono is supported */
@@ -89,7 +89,7 @@
   if (!ac)
     return 0;
 
-  if (ac->codec_id != CODEC_ID_AMR_WB)
+  if (ac->codec_id != AV_CODEC_ID_AMR_WB)
     return 0;
   
   /* valid sampling rates: 8, 16, 24, 32 and 48 kHz */
diff -Naur libdlna-0.2.4/src/audio_atrac3.c libdlna-0.2.4.new/src/audio_atrac3.c
--- libdlna-0.2.4/src/audio_atrac3.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/audio_atrac3.c	2014-12-19 10:02:53.884958835 +0100
@@ -39,7 +39,7 @@
     return AUDIO_PROFILE_INVALID;
 
 #if LIBAVCODEC_VERSION_INT >= ((51<<16)+(40<<8)+4)
-  if (ac->codec_id == CODEC_ID_ATRAC3)
+  if (ac->codec_id == AV_CODEC_ID_ATRAC3)
     return AUDIO_PROFILE_ATRAC;
 #endif
 
diff -Naur libdlna-0.2.4/src/audio_g726.c libdlna-0.2.4.new/src/audio_g726.c
--- libdlna-0.2.4/src/audio_g726.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/audio_g726.c	2014-12-19 10:02:53.884958835 +0100
@@ -31,7 +31,7 @@
   if (!ac)
     return AUDIO_PROFILE_INVALID;
 
-  if (ac->codec_id != CODEC_ID_ADPCM_G726)
+  if (ac->codec_id != AV_CODEC_ID_ADPCM_G726)
     return AUDIO_PROFILE_INVALID;
   
   if (ac->channels != 1)
diff -Naur libdlna-0.2.4/src/audio_lpcm.c libdlna-0.2.4.new/src/audio_lpcm.c
--- libdlna-0.2.4/src/audio_lpcm.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/audio_lpcm.c	2014-12-19 10:02:53.885958866 +0100
@@ -45,8 +45,8 @@
     return AUDIO_PROFILE_INVALID;
 
   /* check for 16-bit signed network-endian PCM codec  */
-  if (ac->codec_id != CODEC_ID_PCM_S16BE &&
-      ac->codec_id != CODEC_ID_PCM_S16LE)
+  if (ac->codec_id != AV_CODEC_ID_PCM_S16BE &&
+      ac->codec_id != AV_CODEC_ID_PCM_S16LE)
     return AUDIO_PROFILE_INVALID;
 
   /* supported channels: mono or stereo */
diff -Naur libdlna-0.2.4/src/audio_mp2.c libdlna-0.2.4.new/src/audio_mp2.c
--- libdlna-0.2.4/src/audio_mp2.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/audio_mp2.c	2014-12-19 10:02:53.885958866 +0100
@@ -31,7 +31,7 @@
   if (!ac)
     return AUDIO_PROFILE_INVALID;
 
-  if (ac->codec_id != CODEC_ID_MP2)
+  if (ac->codec_id != AV_CODEC_ID_MP2)
     return AUDIO_PROFILE_INVALID;
   
   if (ac->channels > 5)
diff -Naur libdlna-0.2.4/src/audio_mp3.c libdlna-0.2.4.new/src/audio_mp3.c
--- libdlna-0.2.4/src/audio_mp3.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/audio_mp3.c	2014-12-19 10:02:53.885958866 +0100
@@ -47,7 +47,7 @@
     return 0;
 
   /* check for MP3 codec */
-  if (ac->codec_id != CODEC_ID_MP3)
+  if (ac->codec_id != AV_CODEC_ID_MP3)
     return 0;
   
   /* only mono and stereo are supported */
diff -Naur libdlna-0.2.4/src/audio_wma.c libdlna-0.2.4.new/src/audio_wma.c
--- libdlna-0.2.4/src/audio_wma.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/audio_wma.c	2014-12-19 10:02:53.886958898 +0100
@@ -55,7 +55,7 @@
     return ap;
 
   /* check for WMA codec */
-  if (ac->codec_id != CODEC_ID_WMAV1 && ac->codec_id != CODEC_ID_WMAV2)
+  if (ac->codec_id != AV_CODEC_ID_WMAV1 && ac->codec_id != AV_CODEC_ID_WMAV2)
     return ap;
 
   if (ac->sample_rate <= 48000)
diff -Naur libdlna-0.2.4/src/av_mpeg1.c libdlna-0.2.4.new/src/av_mpeg1.c
--- libdlna-0.2.4/src/av_mpeg1.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/av_mpeg1.c	2014-12-19 10:02:53.886958898 +0100
@@ -42,7 +42,7 @@
     return NULL;
   
   /* check for MPEG-1 video codec */
-  if (codecs->vc->codec_id != CODEC_ID_MPEG1VIDEO)
+  if (codecs->vc->codec_id != AV_CODEC_ID_MPEG1VIDEO)
     return NULL;
 
   /* video bitrate must be CBR at 1,151,929.1 bps */
@@ -56,23 +56,23 @@
   */
   if (codecs->vc->width == 352 && codecs->vc->height == 288)
   {
-    if (codecs->vs->r_frame_rate.num != 25 &&
-        codecs->vs->r_frame_rate.den != 1)
+    if (codecs->vs->avg_frame_rate.num != 25 &&
+        codecs->vs->avg_frame_rate.den != 1)
           return NULL;
   }
   else if (codecs->vc->width == 352 && codecs->vc->height == 240)
   {
-    if ((codecs->vs->r_frame_rate.num != 30000 &&
-         codecs->vs->r_frame_rate.den != 1001) ||
-        (codecs->vs->r_frame_rate.num != 24000 &&
-         codecs->vs->r_frame_rate.den != 1001))
+    if ((codecs->vs->avg_frame_rate.num != 30000 &&
+         codecs->vs->avg_frame_rate.den != 1001) ||
+        (codecs->vs->avg_frame_rate.num != 24000 &&
+         codecs->vs->avg_frame_rate.den != 1001))
           return NULL;
   }
   else
     return NULL;
 
   /* check for MPEG-1 Layer-2 audio codec */
-  if (codecs->ac->codec_id != CODEC_ID_MP2)
+  if (codecs->ac->codec_id != AV_CODEC_ID_MP2)
     return NULL;
   
   /* supported channels: stereo only */
diff -Naur libdlna-0.2.4/src/av_mpeg2.c libdlna-0.2.4.new/src/av_mpeg2.c
--- libdlna-0.2.4/src/av_mpeg2.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/av_mpeg2.c	2014-12-19 10:02:53.887958929 +0100
@@ -379,8 +379,8 @@
   int i;
 
   /* determine region through frame rate */
-  if ((codecs->vs->r_frame_rate.num == 30000 &&
-       codecs->vs->r_frame_rate.den == 1001)) /* NTSC */
+  if ((codecs->vs->avg_frame_rate.num == 30000 &&
+       codecs->vs->avg_frame_rate.den == 1001)) /* NTSC */
   {
     for (i = 0; i < (int) (sizeof (mpeg_ps_es_valid_streams_ntsc)
                            / sizeof (mpeg_ps_es_stream_t)); i++)
@@ -402,8 +402,8 @@
     /* invalid resolution */
     return NULL;
   }
-  else if (codecs->vs->r_frame_rate.num == 25 &&
-           codecs->vs->r_frame_rate.den == 1) /* PAL */
+  else if (codecs->vs->avg_frame_rate.num == 25 &&
+           codecs->vs->avg_frame_rate.den == 1) /* PAL */
   {
     for (i = 0; i < (int) (sizeof (mpeg_ps_es_valid_streams_pal)
                            / sizeof (mpeg_ps_es_stream_t)); i++)
@@ -453,15 +453,15 @@
   int i;
   
   /* check for MPEG-2 MP@LL profile */
-  if (codecs->ac->codec_id == CODEC_ID_AAC)
+  if (codecs->ac->codec_id == AV_CODEC_ID_AAC)
   {
     /* 352x288 only */
     if (codecs->vc->width != 352 && codecs->vc->height != 288)
       return NULL;
 
     /* 30 fps */
-    if (codecs->vs->r_frame_rate.num != 30 &&
-        codecs->vs->r_frame_rate.den != 1)
+    if (codecs->vs->avg_frame_rate.num != 30 &&
+        codecs->vs->avg_frame_rate.den != 1)
       return NULL;
 
     /* video bitrate is less or equal to 4 Mbps */
@@ -486,8 +486,8 @@
   }
 
   /* check for Region: only Europe supports 25 fps (50i) */
-  if (codecs->vs->r_frame_rate.num == 25 &&
-      codecs->vs->r_frame_rate.den == 1)
+  if (codecs->vs->avg_frame_rate.num == 25 &&
+      codecs->vs->avg_frame_rate.den == 1)
   {
     for (i = 0; i < (int) (sizeof (mpeg_ts_valid_streams_eu_sd)
                            / sizeof (mpeg_ts_stream_t)); i++)
@@ -535,7 +535,7 @@
   if (ctx->bit_rate > 19392700)
     return NULL;
 
-  if (codecs->ac->codec_id != CODEC_ID_AC3)
+  if (codecs->ac->codec_id != AV_CODEC_ID_AC3)
     return NULL;
 
   /* 48 KHz only */
@@ -559,9 +559,9 @@
     if (mpeg_ts_valid_streams_na_sd[i].width == codecs->vc->width &&
         mpeg_ts_valid_streams_na_sd[i].height == codecs->vc->height &&
         mpeg_ts_valid_streams_na_sd[i].fps_num
-        == codecs->vs->r_frame_rate.num &&
+        == codecs->vs->avg_frame_rate.num &&
         mpeg_ts_valid_streams_na_sd[i].fps_den
-        == codecs->vs->r_frame_rate.num)
+        == codecs->vs->avg_frame_rate.num)
     {
       switch (st)
       {
@@ -584,9 +584,9 @@
     if (mpeg_ts_valid_streams_na_hd[i].width == codecs->vc->width &&
         mpeg_ts_valid_streams_na_hd[i].height == codecs->vc->height &&
         mpeg_ts_valid_streams_na_hd[i].fps_num
-        == codecs->vs->r_frame_rate.num &&
+        == codecs->vs->avg_frame_rate.num &&
         mpeg_ts_valid_streams_na_hd[i].fps_den
-        == codecs->vs->r_frame_rate.num)
+        == codecs->vs->avg_frame_rate.num)
     {
       switch (st)
       {
@@ -615,7 +615,7 @@
     return NULL;
   
   /* check for MPEG-2 video codec */
-  if (codecs->vc->codec_id != CODEC_ID_MPEG2VIDEO)
+  if (codecs->vc->codec_id != AV_CODEC_ID_MPEG2VIDEO)
     return NULL;
 
   switch (st)
diff -Naur libdlna-0.2.4/src/av_mpeg4_part10.c libdlna-0.2.4.new/src/av_mpeg4_part10.c
--- libdlna-0.2.4/src/av_mpeg4_part10.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/av_mpeg4_part10.c	2014-12-19 10:02:53.887958929 +0100
@@ -1151,7 +1151,7 @@
     /* QCIF */
     if (vc->bit_rate <= 128000 && ctx->bit_rate <= 256000)
     {
-      if (vs->r_frame_rate.num == 15 && vs->r_frame_rate.num == 1)
+      if (vs->avg_frame_rate.num == 15 && vs->avg_frame_rate.num == 1)
         return AVC_VIDEO_PROFILE_BL_QCIF15;
       else
         return AVC_VIDEO_PROFILE_BL_L1B_QCIF;
@@ -1221,7 +1221,7 @@
     return NULL;
 
   /* check for H.264/AVC codec */
-  if (codecs->vc->codec_id != CODEC_ID_H264)
+  if (codecs->vc->codec_id != AV_CODEC_ID_H264)
     return NULL;
 
   /* check for a supported container */
diff -Naur libdlna-0.2.4/src/av_mpeg4_part2.c libdlna-0.2.4.new/src/av_mpeg4_part2.c
--- libdlna-0.2.4/src/av_mpeg4_part2.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/av_mpeg4_part2.c	2014-12-19 10:02:53.888958960 +0100
@@ -42,13 +42,13 @@
   int codec_id;
   mpeg4_vcodec_type_t type;
 } avf_vcodec_mapping[] = {
-  { CODEC_ID_H263,                      MPEG4_VCODEC_H263 },
-  { CODEC_ID_H263I,                     MPEG4_VCODEC_H263 },
-  { CODEC_ID_H263P,                     MPEG4_VCODEC_H263 },
-  { CODEC_ID_MPEG4,                     MPEG4_VCODEC_P2 },
-  { CODEC_ID_MSMPEG4V1,                 MPEG4_VCODEC_P2 },
-  { CODEC_ID_MSMPEG4V2,                 MPEG4_VCODEC_P2 },
-  { CODEC_ID_MSMPEG4V3,                 MPEG4_VCODEC_P2 },
+  { AV_CODEC_ID_H263,                      MPEG4_VCODEC_H263 },
+  { AV_CODEC_ID_H263I,                     MPEG4_VCODEC_H263 },
+  { AV_CODEC_ID_H263P,                     MPEG4_VCODEC_H263 },
+  { AV_CODEC_ID_MPEG4,                     MPEG4_VCODEC_P2 },
+  { AV_CODEC_ID_MSMPEG4V1,                 MPEG4_VCODEC_P2 },
+  { AV_CODEC_ID_MSMPEG4V2,                 MPEG4_VCODEC_P2 },
+  { AV_CODEC_ID_MSMPEG4V3,                 MPEG4_VCODEC_P2 },
   { -1, 0 }
 };
    
@@ -780,7 +780,7 @@
   {
     if (res[i].width == vc->width &&
         res[i].height == vc->height &&
-        (vs->r_frame_rate.num / vs->r_frame_rate.num)
+        (vs->avg_frame_rate.num / vs->avg_frame_rate.num)
         <= (res[i].fps_num / res[i].fps_den))
       return 1;
   }
diff -Naur libdlna-0.2.4/src/av_wmv9.c libdlna-0.2.4.new/src/av_wmv9.c
--- libdlna-0.2.4/src/av_wmv9.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/av_wmv9.c	2014-12-19 10:02:53.889958991 +0100
@@ -193,8 +193,8 @@
   for (i = 0; i < size / (int) sizeof (wmv9_profile_t); i++)
     if (vc->width <= profile[i].max_width &&
         vc->height <= profile[i].max_height &&
-        vs->r_frame_rate.num == profile[i].fps_num &&
-        vs->r_frame_rate.den == profile[i].fps_den &&
+        vs->avg_frame_rate.num == profile[i].fps_num &&
+        vs->avg_frame_rate.den == profile[i].fps_den &&
         vc->bit_rate <= profile[i].max_bitrate)
       return 1;
 
@@ -245,7 +245,7 @@
     return NULL;
 
   /* check for WMV3 (Simple and Main profiles) video codec */
-  if (codecs->vc->codec_id != CODEC_ID_WMV3)
+  if (codecs->vc->codec_id != AV_CODEC_ID_WMV3)
     return NULL;
 
   /* get video profile */
diff -Naur libdlna-0.2.4/src/image_jpeg.c libdlna-0.2.4.new/src/image_jpeg.c
--- libdlna-0.2.4/src/image_jpeg.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/image_jpeg.c	2014-12-19 10:02:53.889958991 +0100
@@ -92,10 +92,10 @@
     return NULL;
 
   /* check for JPEG compliant codec */
-  if (codecs->vc->codec_id != CODEC_ID_MJPEG &&
-      codecs->vc->codec_id != CODEC_ID_MJPEGB &&
-      codecs->vc->codec_id != CODEC_ID_LJPEG &&
-      codecs->vc->codec_id != CODEC_ID_JPEGLS)
+  if (codecs->vc->codec_id != AV_CODEC_ID_MJPEG &&
+      codecs->vc->codec_id != AV_CODEC_ID_MJPEGB &&
+      codecs->vc->codec_id != AV_CODEC_ID_LJPEG &&
+      codecs->vc->codec_id != AV_CODEC_ID_JPEGLS)
     return NULL;
 
   for (i = 0; jpeg_profiles_mapping[i].profile; i++)
diff -Naur libdlna-0.2.4/src/image_png.c libdlna-0.2.4.new/src/image_png.c
--- libdlna-0.2.4/src/image_png.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/image_png.c	2014-12-19 10:02:53.890959022 +0100
@@ -76,7 +76,7 @@
     return NULL;
   
   /* check for PNG compliant codec */
-  if (codecs->vc->codec_id != CODEC_ID_PNG)
+  if (codecs->vc->codec_id != AV_CODEC_ID_PNG)
     return NULL;
 
   for (i = 0; png_profiles_mapping[i].profile; i++)
diff -Naur libdlna-0.2.4/src/profiles.c libdlna-0.2.4.new/src/profiles.c
--- libdlna-0.2.4/src/profiles.c	2012-02-26 20:34:00.000000000 +0100
+++ libdlna-0.2.4.new/src/profiles.c	2014-12-19 10:02:53.890959022 +0100
@@ -287,7 +287,7 @@
     return NULL;
   }
 
-  if (av_find_stream_info (ctx) < 0)
+  if (avformat_find_stream_info (ctx, NULL) < 0)
   {
     if (dlna->verbosity)
       fprintf (stderr, "can't find stream info\n");
@@ -334,7 +334,7 @@
     p = p->next;
   }
 
-  av_close_input_file (ctx);
+  avformat_close_input (ctx);
   free (codecs);
   return profile;
 }
