diff -Naur rpm-4.4.9.orig/lib/Makefile.am rpm-4.4.9/lib/Makefile.am
--- rpm-4.4.9.orig/lib/Makefile.am	2007-09-07 13:59:07.000000000 +0200
+++ rpm-4.4.9/lib/Makefile.am	2007-09-07 14:23:43.000000000 +0200
@@ -48,7 +48,8 @@
 	$(top_builddir)/rpmio/librpmio.la \
 	@WITH_POPT_LIB@ \
 	@WITH_SELINUX_LIB@ \
-	@LTLIBINTL@
+	@LTLIBINTL@ \
+	@WITH_BEECRYPT_LIB@
 
 if HAVE_LD_VERSION_SCRIPT
 librpm_la_LDFLAGS += -Wl,--version-script=$(srcdir)/librpm.vers
