diff -Naur rpm-4.4.9.orig/configure rpm-4.4.9/configure
--- rpm-4.4.9.orig/configure	2007-09-07 13:59:02.000000000 +0200
+++ rpm-4.4.9/configure	2007-09-07 14:08:17.000000000 +0200
@@ -36454,11 +36454,11 @@
 fi
 
 MARK64=
-if ! echo "${libdir}" | grep -q '64$' ; then
-    case "${target_cpu}" in
-    x86_64*|ppc64*|powerpc64*|sparc64*|s390x*)	MARK64=64 ;;
-    esac
-fi
+#if ! echo "${libdir}" | grep -q '64$' ; then
+#    case "${target_cpu}" in
+#    x86_64*|ppc64*|powerpc64*|sparc64*|s390x*)	MARK64=64 ;;
+#    esac
+#fi
 
 
 # For some systems we know that we have ld_version scripts.
diff -Naur rpm-4.4.9.orig/popt/configure rpm-4.4.9/popt/configure
--- rpm-4.4.9.orig/popt/configure	2007-09-07 13:59:02.000000000 +0200
+++ rpm-4.4.9/popt/configure	2007-09-07 14:09:14.000000000 +0200
@@ -20873,11 +20873,11 @@
 
 
 MARK64=
-if ! echo "${libdir}" | grep -q '64$' ; then
-    case "${target_cpu}" in
-    x86_64*|powerpc64*|ppc64*|sparc64*|s390x*) MARK64=64 ;;
-    esac
-fi
+#if ! echo "${libdir}" | grep -q '64$' ; then
+#    case "${target_cpu}" in
+#    x86_64*|powerpc64*|ppc64*|sparc64*|s390x*) MARK64=64 ;;
+#    esac
+#fi
 
 
 
diff -Naur rpm-4.4.9.orig/scripts/Makefile.in rpm-4.4.9/scripts/Makefile.in
--- rpm-4.4.9.orig/scripts/Makefile.in	2007-09-07 13:59:02.000000000 +0200
+++ rpm-4.4.9/scripts/Makefile.in	2007-09-07 14:08:50.000000000 +0200
@@ -347,7 +347,7 @@
 
 installprefix = $(DESTDIR)
 bin_SCRIPTS = rpm2cpio
-configdir = ${prefix}/lib/rpm
+configdir = ${libdir}/rpm
 config_SCRIPTS = \
 	brp-compress brp-python-bytecompile brp-java-gcjcompile brp-redhat \
 	brp-strip brp-strip-comment-note brp-nobuildrootpath \
