diff -Naur digikam-1.2.0/libs/jpegutils/libjpeg-62/transupp.cpp digikam-1.2.0-p/libs/jpegutils/libjpeg-62/transupp.cpp
--- digikam-1.2.0/libs/jpegutils/libjpeg-62/transupp.cpp	2010-03-29 00:18:42.000000000 +0200
+++ digikam-1.2.0-p/libs/jpegutils/libjpeg-62/transupp.cpp	2010-03-29 12:23:37.292190855 +0200
@@ -16,9 +16,10 @@
  * it's helpful to let it call jround_up() and jcopy_block_row().
  */
 #define JPEG_INTERNALS
-
+extern "C" {
 #include "jinclude.h"
 #include "jpeglib.h"
+}
 #include "transupp.h"		/* My own external interface */
 
 namespace Digikam
