--- digikam-4.10.0_old/extra/libkface/libkface/libopencv.h.cmake.in	2015-05-10 11:09:07.000000000 +0200
+++ digikam-4.10.0/extra/libkface/libkface/libopencv.h.cmake.in	2015-07-18 14:13:55.276829677 +0200
@@ -53,5 +53,9 @@
 
 #include <opencv2/core/core.hpp>
+#if CV_MAJOR_VERSION == 2
 #include <opencv2/core/internal.hpp>
+#elif CV_MAJOR_VERSION == 3
+#include <opencv2/core/private.hpp>
+#endif
 #include <opencv2/contrib/contrib.hpp>
