--- setup.py	2009-05-08 18:11:20.000000000 +0200
+++ setup.py	2009-07-10 11:35:08.000000000 +0200
@@ -168,16 +168,6 @@
 shutil.copytree('src', 'staging')
 os.unlink(os.path.join('staging', 'ufw-init'))
 os.unlink(os.path.join('staging', 'ufw-init-functions'))
-
-exe = 'iptables'
-(rc, out) = cmd([exe, '-V'])
-if rc != 0:
-    raise OSError(errno.ENOENT, "Could not find version for '%s'" % (exe))
-version = re.sub('^v', '', re.split('\s', out)[1])
-print "Found '%s' version '%s'" % (exe, version)
-if version < "1.4":
-    print >> sys.stderr, "WARN: version '%s' has limited IPv6 support. See README for details." % (version)
-
 setup (name='ufw',
       version=ufw_version,
       description='front-end for Linux firewalling',
