diff -Naur pidgin-2.0.0/libpurple/plugins/perl/Makefile.am pidgin-2.0.0-perl-fix/libpurple/plugins/perl/Makefile.am
--- pidgin-2.0.0/libpurple/plugins/perl/Makefile.am	2007-05-03 20:14:10.000000000 +0200
+++ pidgin-2.0.0-perl-fix/libpurple/plugins/perl/Makefile.am	2007-05-04 21:28:57.000000000 +0200
@@ -92,7 +92,7 @@
 			${LN_S} -f ../${srcdir}/$$f $$f; \
 		done; \
 	fi
-	@cd common && $(perlpath) Makefile.PL $(PERL_MM_PARAMS)
+	@cd common && $(perlpath) Makefile.PL
 
 common/Makefile.PL: common/Makefile.PL.in $(top_builddir)/config.status
 	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)
@@ -101,7 +101,7 @@
 	@for dir in $(perl_dirs); do \
 		cd $$dir && \
 		if [ ! -f Makefile ]; then \
-			$(perlpath) Makefile.PL $(PERL_MM_PARAMS); \
+			$(perlpath) Makefile.PL; \
 		fi && \
 		($(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS) || \
 		 $(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS)) && \
diff -Naur pidgin-2.0.0/libpurple/plugins/perl/Makefile.in pidgin-2.0.0-perl-fix/libpurple/plugins/perl/Makefile.in
--- pidgin-2.0.0/libpurple/plugins/perl/Makefile.in	2007-05-04 07:24:28.000000000 +0200
+++ pidgin-2.0.0-perl-fix/libpurple/plugins/perl/Makefile.in	2007-05-04 21:28:57.000000000 +0200
@@ -786,7 +786,7 @@
 			${LN_S} -f ../${srcdir}/$$f $$f; \
 		done; \
 	fi
-	@cd common && $(perlpath) Makefile.PL $(PERL_MM_PARAMS)
+	@cd common && $(perlpath) Makefile.PL
 
 common/Makefile.PL: common/Makefile.PL.in $(top_builddir)/config.status
 	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)
@@ -795,7 +795,7 @@
 	@for dir in $(perl_dirs); do \
 		cd $$dir && \
 		if [ ! -f Makefile ]; then \
-			$(perlpath) Makefile.PL $(PERL_MM_PARAMS); \
+			$(perlpath) Makefile.PL; \
 		fi && \
 		($(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS) || \
 		 $(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS)) && \
diff -Naur pidgin-2.0.0/pidgin/plugins/perl/Makefile.am pidgin-2.0.0-perl-fix/pidgin/plugins/perl/Makefile.am
--- pidgin-2.0.0/pidgin/plugins/perl/Makefile.am	2007-05-03 20:14:10.000000000 +0200
+++ pidgin-2.0.0-perl-fix/pidgin/plugins/perl/Makefile.am	2007-05-04 21:28:57.000000000 +0200
@@ -42,7 +42,7 @@
 			${LN_S} -f ../${srcdir}/$$f $$f; \
 		done; \
 	fi
-	@cd common && $(perlpath) Makefile.PL $(PERL_MM_PARAMS)
+	@cd common && $(perlpath) Makefile.PL
 
 common/Makefile.PL: common/Makefile.PL.in $(top_builddir)/config.status
 	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)
@@ -51,7 +51,7 @@
 	@for dir in $(perl_dirs); do \
 		cd $$dir && \
 		if [ ! -f Makefile ]; then \
-			$(perlpath) Makefile.PL $(PERL_MM_PARAMS); \
+			$(perlpath) Makefile.PL; \
 		fi && \
 		($(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS) || \
 		$(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS)) && \
diff -Naur pidgin-2.0.0/pidgin/plugins/perl/Makefile.in pidgin-2.0.0-perl-fix/pidgin/plugins/perl/Makefile.in
--- pidgin-2.0.0/pidgin/plugins/perl/Makefile.in	2007-05-04 07:24:43.000000000 +0200
+++ pidgin-2.0.0-perl-fix/pidgin/plugins/perl/Makefile.in	2007-05-04 21:28:57.000000000 +0200
@@ -598,7 +598,7 @@
 			${LN_S} -f ../${srcdir}/$$f $$f; \
 		done; \
 	fi
-	@cd common && $(perlpath) Makefile.PL $(PERL_MM_PARAMS)
+	@cd common && $(perlpath) Makefile.PL
 
 common/Makefile.PL: common/Makefile.PL.in $(top_builddir)/config.status
 	cd $(top_builddir) && $(SHELL) ./config.status $(subdir)/$@ $(am__depfiles_maybe)
@@ -607,7 +607,7 @@
 	@for dir in $(perl_dirs); do \
 		cd $$dir && \
 		if [ ! -f Makefile ]; then \
-			$(perlpath) Makefile.PL $(PERL_MM_PARAMS); \
+			$(perlpath) Makefile.PL; \
 		fi && \
 		($(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS) || \
 		$(MAKE) CC="$(CC)" CCFLAGS="$(PERL_CFLAGS) $(CFLAGS)" $(PERL_EXTRA_OPTS)) && \
