diff -Naur accountsservice-0.6.43/src/daemon.c accountsservice-0.6.43-groups/src/daemon.c
--- accountsservice-0.6.43/src/daemon.c	2016-09-06 21:48:50.000000000 +0200
+++ accountsservice-0.6.43-groups/src/daemon.c	2016-10-09 17:33:11.384532429 +0200
@@ -52,6 +52,8 @@
 #define PATH_SHADOW "/etc/shadow"
 #define PATH_GROUP "/etc/group"
 #define PATH_GDM_CUSTOM "/etc/gdm/custom.conf"
+// can be easy done with an patch for ./configure
+#define DISTRO_DEFAULT_GROUPS "audio,camera,cdrom,lp,scanner,video,uucp,storage,netdev,locate"
 
 enum {
         PROP_0,
@@ -984,9 +986,13 @@
                 argv[8] = NULL;
         }
         else if (cd->account_type == ACCOUNT_TYPE_STANDARD) {
-                argv[4] = "--";
-                argv[5] = cd->user_name;
-                argv[6] = NULL;
+		// without this we get default from useradd eg: just users group
+                argv[4] = "-G";
+                argv[5] = DISTRO_DEFAULT_GROUPS;
+                argv[6] = "--";
+                argv[7] = cd->user_name;
+                argv[8] = NULL;
+
         }
         else {
                 throw_error (context, ERROR_FAILED, "Don't know how to add user of type %d", cd->account_type);
