--- src/io/prom/forth.cc	2005-05-24 16:27:59.000000000 -0500
+++ src/io/prom/forth.cc.orig	2011-09-07 16:09:47.000000000 -0500
@@ -205,10 +205,12 @@
 	return true;
 }
 
+#if 0
 String &ForthVM::getToken(const String &delimiters)
 {
 	
 }
+#endif
 
 bool ForthVM::consumeSpace(bool except)
 {
@@ -539,6 +541,7 @@
 
 uint32 ForthVar::getExecToken(ForthVM &vm)
 {
+   return 0;
 }
 
 void ForthVar::interprete(ForthVM &vm)
--- src/io/prom/promboot.cc	2011-07-12 09:37:52.000000000 -0500
+++ src/io/prom/promboot.cc.orig	2011-09-07 16:15:45.000000000 -0500
@@ -476,6 +476,7 @@
 			loader_size -= size;
 		}
 	}
+	return false;
 }
 
 extern uint32 gMemorySize;
--- src/io/serial/serial.cc	2006-10-11 13:39:02.000000000 -0500
+++ src/io/serial/serial.cc.orig	2011-09-07 16:19:44.000000000 -0500
@@ -148,6 +148,7 @@
 		static char bl[10];
 		sprintf(bl, "%08x", a);
 	}
+	return NULL;
 }
 
 virtual bool readDeviceIO(uint r, uint32 address, uint32 &data, uint size)
--- src/debug/debugger.cc	2008-02-07 14:38:29.000000000 -0600
+++ src/debug/debugger.cc.orig	2011-09-07 16:21:08.000000000 -0500
@@ -562,6 +562,7 @@
 		return true;
 	}
 #endif
+return false;
 }
 
 bool Debugger::parse(const String &str)
