diff -Naur traverso-0.42.0/CMakeLists.txt traverso-0.42.0-hack/CMakeLists.txt
--- traverso-0.42.0/CMakeLists.txt	2007-12-10 13:51:49.000000000 +0100
+++ traverso-0.42.0-hack/CMakeLists.txt	2008-07-08 11:40:21.000000000 +0200
@@ -252,9 +252,6 @@
 		LIST(APPEND TRAVERSO_DEFINES -DMP3_ENCODE_SUPPORT)
 		MESSAGE("-- Building with mp3 encoding support enabled")
 		SET(HAVE_MP3_ENCODING TRUE)
-	ENDIF(LAME_FOUND)
-ELSE(WANT_MP3_ENCODE)
-	SET(HAVE_MP3_ENCODING FALSE)
 ENDIF(WANT_MP3_ENCODE)
 
 
