--- traverso-0.42.0.orig/src/traverso/Main.cpp
+++ traverso-0.42.0/src/traverso/Main.cpp
@@ -31,6 +31,7 @@
 // Always put me below _all_ includes, this is needed
 // in case we run with memory leak detection enabled!
 #include "Debugger.h"
+#include <cstdlib>
 
 
 Traverso* traverso;
--- traverso-0.42.0.orig/src/traverso/songcanvas/TimeLineView.cpp
+++ traverso-0.42.0/src/traverso/songcanvas/TimeLineView.cpp
@@ -48,6 +48,7 @@
 // Always put me below _all_ includes, this is needed
 // in case we run with memory leak detection enabled!
 #include "Debugger.h"
+#include <cstdlib>
 
 
 #define MARKER_SOFT_SELECTION_DISTANCE 50
--- traverso-0.42.0.orig/src/core/ReadSource.cpp
+++ traverso-0.42.0/src/core/ReadSource.cpp
@@ -35,7 +35,9 @@
 // Always put me below _all_ includes, this is needed
 // in case we run with memory leak detection enabled!
 #include "Debugger.h"
-
+#ifndef LONG_LONG_MAX
+#define LONG_LONG_MAX __LONG_LONG_MAX__
+#endif
 
 /**
  *	\class ReadSource

