Index: traverso-0.42.0/src/common/fpu.cc
===================================================================
--- traverso-0.42.0.orig/src/common/fpu.cc
+++ traverso-0.42.0/src/common/fpu.cc
@@ -75,11 +75,12 @@ FPU::FPU ()
 	if (cpuflags & (1 << 24)) {
 		
 		char* fxbuf = 0;
+		char** temp = &fxbuf;
 		
 #ifdef NO_POSIX_MEMALIGN
 		if ((fxbuf = (char *) malloc(512)) == 0)
 #else
-		if (posix_memalign ((void**)&fxbuf, 16, 512)) 
+		if (posix_memalign ((void**)temp, 16, 512)) 
 #endif			
 		{
 			printf("FPU() ERROR: cannot allocate 16 byte aligned buffer for h/w feature detection");
