diff -Naur libglvnd-1.2.0/configure.ac libglvnd-1.2.0.new/configure.ac
--- libglvnd-1.2.0/configure.ac	2019-09-25 17:48:46.000000000 +0200
+++ libglvnd-1.2.0.new/configure.ac	2019-10-25 14:12:54.088331371 +0200
@@ -169,8 +169,9 @@
 
 if test "x$enable_x11" = "xyes" ; then
     PKG_CHECK_MODULES([X11], [x11])
-    AC_DEFINE([USE_X11], 1,
-        [Define to 1 if X11 support is enabled.])
+else
+    AC_DEFINE([EGL_NO_X11], 1,
+        [Define to 1 if X11 support is disabled.])
 fi
 if test "x$enable_glx" = "xyes" ; then
     PKG_CHECK_MODULES([XEXT], [xext])
diff -Naur libglvnd-1.2.0/src/EGL/libegl.c libglvnd-1.2.0.new/src/EGL/libegl.c
--- libglvnd-1.2.0/src/EGL/libegl.c	2019-09-25 17:48:46.000000000 +0200
+++ libglvnd-1.2.0.new/src/EGL/libegl.c	2019-10-25 14:12:54.088331371 +0200
@@ -36,7 +36,7 @@
 #include <unistd.h>
 #include <sys/mman.h>
 
-#if defined(USE_X11)
+#if !defined(EGL_NO_X11)
 #include <X11/Xlib.h>
 #endif
 
@@ -180,7 +180,7 @@
 
 static EGLBoolean IsX11Display(void *dpy)
 {
-#if defined(USE_X11)
+#if !defined(EGL_NO_X11)
     void *alloc;
     void *handle;
     void *XAllocID = NULL;
@@ -197,9 +197,9 @@
     }
 
     return (XAllocID != NULL && XAllocID == alloc);
-#else // defined(USE_X11)
+#else // !defined(EGL_NO_X11)
     return EGL_FALSE;
-#endif // defined(USE_X11)
+#endif // !defined(EGL_NO_X11)
 }
 
 static EGLBoolean IsWaylandDisplay(void *native_display)
