diff -Naur qbittorrent-0.9.2/configure qbittorrent-0.9.2-p/configure
--- qbittorrent-0.9.2/configure	2007-04-06 02:39:58.000000000 +0200
+++ qbittorrent-0.9.2-p/configure	2007-04-15 01:16:48.000000000 +0200
@@ -295,8 +295,8 @@
 {
 public:
 	qc_libtorrent(Conf *c) : ConfObj(c) {}
-	QString name() const { return "libtorrent >= 0.11 (>= 0.12 ADVISED)"; }
-	QString shortname() const { return "libtorrent"; }
+	QString name() const { return "rblibtorrent >= 0.11 (>= 0.12 ADVISED)"; }
+	QString shortname() const { return "rblibtorrent"; }
 	bool exec(){
         	QString s;
 		s = conf->getenv("QC_WITH_LIBTORRENT_INC");
diff -Naur qbittorrent-0.9.2/qbittorrent.qc qbittorrent-0.9.2-p/qbittorrent.qc
--- qbittorrent-0.9.2/qbittorrent.qc	2007-03-15 16:51:11.000000000 +0100
+++ qbittorrent-0.9.2-p/qbittorrent.qc	2007-04-15 01:15:00.000000000 +0200
@@ -5,7 +5,7 @@
   <dep type='qt42'>
     <required/>
   </dep>
-  <dep type='libtorrent'>
+  <dep type='rblibtorrent'>
     <required/>
   </dep>
   <dep type='libboost'>
diff -Naur qbittorrent-0.9.2/qcm/libtorrent.qcm qbittorrent-0.9.2-p/qcm/libtorrent.qcm
--- qbittorrent-0.9.2/qcm/libtorrent.qcm	2007-04-06 02:39:58.000000000 +0200
+++ qbittorrent-0.9.2-p/qcm/libtorrent.qcm	2007-04-15 01:20:31.000000000 +0200
@@ -9,8 +9,8 @@
 {
 public:
 	qc_libtorrent(Conf *c) : ConfObj(c) {}
-	QString name() const { return "libtorrent >= 0.11 (>= 0.12 ADVISED)"; }
-	QString shortname() const { return "libtorrent"; }
+	QString name() const { return "rblibtorrent >= 0.11 (>= 0.12 ADVISED)"; }
+	QString shortname() const { return "rblibtorrent"; }
 	bool exec(){
         	QString s;
 		s = conf->getenv("QC_WITH_LIBTORRENT_INC");
diff -Naur qbittorrent-0.9.2/src/src.pro qbittorrent-0.9.2-p/src/src.pro
--- qbittorrent-0.9.2/src/src.pro	2007-04-10 23:02:47.000000000 +0200
+++ qbittorrent-0.9.2-p/src/src.pro	2007-04-15 01:21:13.000000000 +0200
@@ -29,7 +29,7 @@
 QMAKE_CXXFLAGS_DEBUG += -fwrapv
 #LIBS += -lcurl
 CONFIG += link_pkgconfig
-PKGCONFIG += libtorrent
+PKGCONFIG += rblibtorrent
 QT += network
 
 DEFINES += QT_NO_CAST_TO_ASCII
