diff -Naur qbittorrent-1.0.0/configure qbittorrent-1.0.0-frugalware/configure
--- qbittorrent-1.0.0/configure	2008-04-11 19:01:38.000000000 +0200
+++ qbittorrent-1.0.0-frugalware/configure	2008-07-14 10:44:43.000000000 +0200
@@ -370,8 +370,8 @@
 {
 public:
 	qc_libtorrent(Conf *c) : ConfObj(c) {}
-	QString name() const { return "libtorrent >= 0.13"; }
-	QString shortname() const { return "libtorrent"; }
+	QString name() const { return "libtorrent-rasterbar >= 0.13"; }
+	QString shortname() const { return "libtorrent-rasterbar"; }
 	bool exec(){
         	QString s;
 		s = conf->getenv("QC_WITH_LIBTORRENT_INC");
@@ -405,7 +405,7 @@
 
 		s = conf->getenv("QC_WITH_LIBTORRENT_LIB");
 		if(!s.isEmpty()) {
-      			if(!conf->checkLibrary(s, "torrent")) {
+      			if(!conf->checkLibrary(s, "torrent-rasterbar")) {
 			        return false;
 			}
       			conf->addLib(QString("-L") + s);
@@ -417,7 +417,7 @@
 			sl << "/usr/local/lib64/";
 			bool found = false;
 			foreach(s, sl){
-				if(conf->checkLibrary(s, "torrent")){
+				if(conf->checkLibrary(s, "torrent-rasterbar")){
 					found = true;
 					break;
 				}
diff -Naur qbittorrent-1.0.0/qbittorrent.qc qbittorrent-1.0.0-frugalware/qbittorrent.qc
--- qbittorrent-1.0.0/qbittorrent.qc	2008-04-11 19:01:38.000000000 +0200
+++ qbittorrent-1.0.0-frugalware/qbittorrent.qc	2008-07-14 10:44:10.000000000 +0200
@@ -6,7 +6,7 @@
   <dep type='qt4'>
     <required/>
   </dep>
-  <dep type='libtorrent'>
+  <dep type='libtorrent-rasterbar'>
     <required/>
   </dep>
   <dep type='libboost'>
diff -Naur qbittorrent-1.0.0/qcm/libtorrent.qcm qbittorrent-1.0.0-frugalware/qcm/libtorrent.qcm
--- qbittorrent-1.0.0/qcm/libtorrent.qcm	2008-04-11 19:01:38.000000000 +0200
+++ qbittorrent-1.0.0-frugalware/qcm/libtorrent.qcm	2008-07-14 10:44:10.000000000 +0200
@@ -10,8 +10,8 @@
 {
 public:
 	qc_libtorrent(Conf *c) : ConfObj(c) {}
-	QString name() const { return "libtorrent >= 0.13"; }
-	QString shortname() const { return "libtorrent"; }
+	QString name() const { return "libtorrent-rasterbar >= 0.13"; }
+	QString shortname() const { return "libtorrent-rasterbar"; }
 	bool exec(){
         	QString s;
 		s = conf->getenv("QC_WITH_LIBTORRENT_INC");
diff -Naur qbittorrent-1.0.0/src/src.pro qbittorrent-1.0.0-frugalware/src/src.pro
--- qbittorrent-1.0.0/src/src.pro	2008-04-11 19:01:38.000000000 +0200
+++ qbittorrent-1.0.0-frugalware/src/src.pro	2008-07-14 10:44:10.000000000 +0200
@@ -81,7 +81,7 @@
 QMAKE_CXXFLAGS_DEBUG += -fwrapv -O1
 
 CONFIG += link_pkgconfig
-PKGCONFIG += "libtorrent libcurl"
+PKGCONFIG += "libtorrent-rasterbar libcurl"
 
 !contains(DEFINES, HAVE_MAGICK){
   message(ImageMagick disabled)
