diff -Naur qbittorrent-1.0.0/configure qbittorrent-1.0.0-p/configure
--- qbittorrent-1.0.0/configure	2008-04-11 19:01:38.000000000 +0200
+++ qbittorrent-1.0.0-p/configure	2008-05-07 02:51:32.000000000 +0200
@@ -370,8 +370,8 @@
 {
 public:
 	qc_libtorrent(Conf *c) : ConfObj(c) {}
-	QString name() const { return "libtorrent >= 0.13"; }
-	QString shortname() const { return "libtorrent"; }
+	QString name() const { return "rblibtorrent >= 0.13"; }
+	QString shortname() const { return "rblibtorrent"; }
 	bool exec(){
         	QString s;
 		s = conf->getenv("QC_WITH_LIBTORRENT_INC");
diff -Naur qbittorrent-1.0.0/qbittorrent.qc qbittorrent-1.0.0-p/qbittorrent.qc
--- qbittorrent-1.0.0/qbittorrent.qc	2008-04-11 19:01:38.000000000 +0200
+++ qbittorrent-1.0.0-p/qbittorrent.qc	2008-05-07 02:50:31.000000000 +0200
@@ -6,7 +6,7 @@
   <dep type='qt4'>
     <required/>
   </dep>
-  <dep type='libtorrent'>
+  <dep type='rblibtorrent'>
     <required/>
   </dep>
   <dep type='libboost'>
diff -Naur qbittorrent-1.0.0/qcm/libtorrent.qcm qbittorrent-1.0.0-p/qcm/libtorrent.qcm
--- qbittorrent-1.0.0/qcm/libtorrent.qcm	2008-04-11 19:01:38.000000000 +0200
+++ qbittorrent-1.0.0-p/qcm/libtorrent.qcm	2008-05-07 02:51:18.000000000 +0200
@@ -10,8 +10,8 @@
 {
 public:
 	qc_libtorrent(Conf *c) : ConfObj(c) {}
-	QString name() const { return "libtorrent >= 0.13"; }
-	QString shortname() const { return "libtorrent"; }
+	QString name() const { return "rblibtorrent >= 0.13"; }
+	QString shortname() const { return "rblibtorrent"; }
 	bool exec(){
         	QString s;
 		s = conf->getenv("QC_WITH_LIBTORRENT_INC");
diff -Naur qbittorrent-1.0.0/src/src.pro qbittorrent-1.0.0-p/src/src.pro
--- qbittorrent-1.0.0/src/src.pro	2008-04-11 19:01:38.000000000 +0200
+++ qbittorrent-1.0.0-p/src/src.pro	2008-05-07 02:52:12.000000000 +0200
@@ -81,7 +81,7 @@
 QMAKE_CXXFLAGS_DEBUG += -fwrapv -O1
 
 CONFIG += link_pkgconfig
-PKGCONFIG += "libtorrent libcurl"
+PKGCONFIG += "rblibtorrent libcurl"
 
 !contains(DEFINES, HAVE_MAGICK){
   message(ImageMagick disabled)
