--- common/xfdesktop-common.h.orig	2015-03-08 13:50:20.583961232 +0100
+++ common/xfdesktop-common.h	2015-03-08 13:50:54.327287913 +0100
@@ -35,7 +35,7 @@
 #include <stdarg.h>
 
 #define XFDESKTOP_CHANNEL        "xfce4-desktop"
-#define DEFAULT_BACKDROP         DATADIR "/backgrounds/xfce/xfce-teal.jpg"
+#define DEFAULT_BACKDROP         DATADIR "/xfce4/backgrounds/frugalware.png"
 #define DEFAULT_ICON_FONT_SIZE   12
 #define DEFAULT_ICON_SIZE        48
 #define ITHEME_FLAGS             (GTK_ICON_LOOKUP_USE_BUILTIN \
