--- xfdesktop-4.9.3/common/xfdesktop-common.h	2012-04-14 10:00:51.000000000 +0200
+++ xfdesktop-4.9.3new/common/xfdesktop-common.h	2012-04-14 10:19:30.000000000 +0200
@@ -32,7 +32,8 @@
 #include <X11/Xlib.h>
 
 #define XFDESKTOP_CHANNEL        "xfce4-desktop"
-#define DEFAULT_BACKDROP         DATADIR "/backgrounds/xfce/xfce-blue.jpg"
+/* Use Frugalware default background image */
+#define DEFAULT_BACKDROP         DATADIR "/xfce4/backdrops/frugalware.png"
 #define DEFAULT_BACKDROP_LIST    "xfce4/desktop/backdrop.list"
 #define DEFAULT_ICON_FONT_SIZE   12
 #define DEFAULT_ICON_SIZE        32
