diff -Naur xfdesktop-4.3.99.1.orig/panel-plugin/desktop-menu-plugin.c xfdesktop-4.3.99.1/panel-plugin/desktop-menu-plugin.c
--- xfdesktop-4.3.99.1.orig/panel-plugin/desktop-menu-plugin.c	2006-09-02 22:44:12.000000000 +0530
+++ xfdesktop-4.3.99.1/panel-plugin/desktop-menu-plugin.c	2006-09-05 11:18:47.000000000 +0530
@@ -58,8 +58,8 @@
 #include "xfce4-popup-menu.h"
 
 #define BORDER 8
-#define DEFAULT_BUTTON_ICON  DATADIR "/pixmaps/xfce4_xicon1.png"
-#define DEFAULT_BUTTON_TITLE "Xfce Menu"
+#define DEFAULT_BUTTON_ICON  DATADIR "/pixmaps/xfce4-fw.png"
+#define DEFAULT_BUTTON_TITLE "Menu"
 
 #ifndef CHECK_RUNNING_PLUGIN
 #define CHECK_RUNNING_PLUGIN 0

diff -Naur xfdesktop-4.3.99.1.orig/common/xfdesktop-common.h xfdesktop-4.3.99.1/common/xfdesktop-common.h
--- xfdesktop-4.3.99.1.orig/common/xfdesktop-common.h	2006-09-02 22:44:18.000000000 +0530
+++ xfdesktop-4.3.99.1/common/xfdesktop-common.h	2006-09-06 18:30:55.000000000 +0530
@@ -31,7 +31,7 @@
 #include <X11/Xlib.h>
 
 #define BACKDROP_CHANNEL         "BACKDROP"
-#define DEFAULT_BACKDROP         DATADIR "/xfce4/backdrops/xfce-smoke.png"
+#define DEFAULT_BACKDROP         DATADIR "/xfce4/backdrops/xfce-frugalware_1.png"
 #define LIST_TEXT                "# xfce backdrop list"
 #define XFDESKTOP_SELECTION_FMT  "XFDESKTOP_SELECTION_%d"
 #define XFDESKTOP_IMAGE_FILE_FMT "XFDESKTOP_IMAGE_FILE_%d"
