diff -Naur xfdesktop-4.4.0/settings/appearance-settings.c xfdesktop-4.4.0.new/settings/appearance-settings.c
--- xfdesktop-4.4.0/settings/appearance-settings.c	2007-01-21 02:48:37.000000000 +0530
+++ xfdesktop-4.4.0.new/settings/appearance-settings.c	2007-03-19 11:33:09.000000000 +0530
@@ -279,9 +279,9 @@
             if(setting)
                 bp->show_image = setting->data.v_int == 0 ? FALSE : TRUE;
             else {
-                bp->show_image = FALSE;
+                bp->show_image = TRUE;
                 mcs_manager_set_int(mcs_plugin->manager, setting_name,
-                        BACKDROP_CHANNEL, 0);
+                        BACKDROP_CHANNEL, 1);
             }
             
             /* the color style */
