diff -Naur kchmviewer-7.7/src/src.pro kchmviewer-7.7-p/src/src.pro
--- kchmviewer-7.7/src/src.pro	2016-10-05 09:35:34.149265000 +0200
+++ kchmviewer-7.7-p/src/src.pro	2017-03-13 23:13:23.028257398 +0100
@@ -108,26 +108,19 @@
     HEADERS += dbus_interface.h
     SOURCES += dbus_interface.cpp
     CONFIG += dbus
-    LIBS += ../lib/libebook/libebook.a
-    POST_TARGETDEPS += ../lib/libebook/libebook.a
+    ## compile error without !
+    LIBS += ../lib/libebook/libebook.a -lchm -lzip
+    POST_TARGETDEPS += ../lib/libebook/libebook.a 
 }
 
 greaterThan(QT_MAJOR_VERSION, 4) {
-    # Qt 5
-    greaterThan(QT_MINOR_VERSION, 5) {
-        # Qt 5.6+
-        error("You use Qt5.6+ - QWebEngine is not yet suitable for kchmviewer and is not supported")
-        QT += webengine webenginewidgets
-        DEFINES += USE_WEBENGINE
-        SOURCES += viewwindow_webengine.cpp dataprovider_qwebengine.cpp
-        HEADERS += dataprovider_qwebengine.h viewwindow_webengine.h
-    } else {
-        # Qt 5.0-5.5
-        QT += webkit webkitwidgets
-        DEFINES += USE_WEBKIT
-        SOURCES += viewwindow_webkit.cpp dataprovider_qwebkit.cpp
-        HEADERS += dataprovider_qwebkit.h viewwindow_webkit.h
-    }
+	# Qt5 
+	## well why f**** fail on broken 
+	## WebEngine code while WebKit backend works!
+	QT += webkit webkitwidgets
+	DEFINES += USE_WEBKIT
+	SOURCES += viewwindow_webkit.cpp dataprovider_qwebkit.cpp
+	HEADERS += dataprovider_qwebkit.h viewwindow_webkit.h
 } else {
     message("Qt4 is not supported anymore, please do not report any errors")
     QT += webkit webkitwidgets
