diff -Naur eflete-1.19.1/configure.ac eflete-1.19.1-p/configure.ac
--- eflete-1.19.1/configure.ac	2017-07-21 15:02:41.000000000 +0200
+++ eflete-1.19.1-p/configure.ac	2017-08-08 18:19:07.158859644 +0200
@@ -14,7 +14,7 @@
 
 dnl m4_define([package], "EFL\ Edje\ Theme\ Editor")
 m4_define([efl_version], [1.19.0])
-m4_define([efl_version_max], [1.19.9999])
+m4_define([efl_version_max], [1.21.999])
 
 m4_define([enventor_version], [0.5.0])
 
