diff -Naur drush-orig/commands/pm/package_handler/wget.inc drush/commands/pm/package_handler/wget.inc
--- drush-orig/commands/pm/package_handler/wget.inc	2012-03-23 21:19:27.000000000 +0100
+++ drush/commands/pm/package_handler/wget.inc	2012-03-31 16:32:40.553052565 +0200
@@ -70,6 +70,9 @@
 
   // Extract the tarball.
   $file_list = drush_tarball_extract($path, $request['base_project_path'], TRUE);
+  $drupal_root = getcwd();
+  $buf = stat($drupal_root);
+  drush_shell_exec("chown -R %s:%s %s", $buf["uid"], $buf["gid"], $request['base_project_path']);
 
   // Move untarred directory to project_dir, if distinct.
   if (($request['project_type'] == 'core') || (($request['project_type'] == 'profile') && (drush_get_option('variant', 'core') == 'core'))) {
