diff -Naur drush-orig/commands/pm/package_handler/wget.inc drush/commands/pm/package_handler/wget.inc
--- drush-orig/commands/pm/package_handler/wget.inc	2010-12-24 22:42:05.000000000 +0100
+++ drush/commands/pm/package_handler/wget.inc	2011-01-14 10:36:44.815000636 +0100
@@ -83,6 +83,8 @@
   $tarpath = basename($tarpath, '.tgz');
   $tarpath .= '.tar';
   drush_shell_exec("tar -xf %s", $tarpath);
+  $buf = stat($olddir);
+  drush_shell_exec("chown -R %s:%s .", $buf["uid"], $buf["gid"]);
 
   // Move untarred directory to project_dir, if distinct.
   if (($request['project_type'] == 'core') || (($request['project_type'] == 'profile') && (drush_get_option('variant', 'core') == 'core'))) {
