diff -Naur boost-jam-3.1.16/build.jam boost-jam-3.1.16-flags/build.jam
--- boost-jam-3.1.16/build.jam	2007-12-03 04:27:54.000000000 +0100
+++ boost-jam-3.1.16-flags/build.jam	2007-12-08 17:38:26.000000000 +0100
@@ -176,7 +176,7 @@
 ## GCC 2.x, 3.x, 4.x
 toolset gcc gcc : "-o " : -D
     : -pedantic -fno-strict-aliasing
-    [ opt --release : [ opt --symbols : -g : -s ] -O3 ]
+    [ opt --release : [ opt --symbols : -g : -s ] $(CFLAGS) ]
     [ opt --debug : -g -O0 -fno-inline ]
     -I$(--python-include) -I$(--extra-include) -Wno-long-long
     : -L$(--python-lib[1]) -l$(--python-lib[2]) ;
diff -Naur boost-jam-3.1.16/build.sh boost-jam-3.1.16-flags/build.sh
--- boost-jam-3.1.16/build.sh	2007-12-03 04:27:54.000000000 +0100
+++ boost-jam-3.1.16-flags/build.sh	2007-12-08 17:36:50.000000000 +0100
@@ -12,7 +12,7 @@
 echo_run ()
 {
     echo "$@"
-    $@
+    eval "$@"
     r=$?
     if test $r -ne 0 ; then
         exit $r
@@ -265,7 +265,7 @@
     echo_run rm -rf bootstrap
     echo_run mkdir bootstrap
     if test ! -r jamgram.y -o ! -r jamgramtab.h ; then
-        echo_run ${BOOST_JAM_CC} ${BOOST_JAM_OPT_YYACC} ${YYACC_SOURCES}
+        echo_run ${BOOST_JAM_CC} ${CFLAGS} ${BOOST_JAM_OPT_YYACC} ${YYACC_SOURCES}
         if test -x "./bootstrap/yyacc0" ; then
             echo_run ./bootstrap/yyacc0 jamgram.y jamgramtab.h jamgram.yy
         fi
@@ -279,16 +279,20 @@
         mv -f y.tab.h jamgram.h
     fi
     if test ! -r jambase.c ; then
-        echo_run ${BOOST_JAM_CC} ${BOOST_JAM_OPT_MKJAMBASE} ${MKJAMBASE_SOURCES}
+        echo_run ${BOOST_JAM_CC} ${CFLAGS} ${BOOST_JAM_OPT_MKJAMBASE} ${MKJAMBASE_SOURCES}
         if test -x "./bootstrap/mkjambase0" ; then
             echo_run ./bootstrap/mkjambase0 jambase.c Jambase
         fi
     fi
-    echo_run ${BOOST_JAM_CC} ${BOOST_JAM_OPT_JAM} ${BJAM_SOURCES}
+    echo_run ${BOOST_JAM_CC} ${CFLAGS} ${BOOST_JAM_OPT_JAM} ${BJAM_SOURCES}
 fi
 if test -x "./bootstrap/jam0" ; then
     if test "${BJAM_UPDATE}" != "update" ; then
         echo_run ./bootstrap/jam0 -f build.jam --toolset=$BOOST_JAM_TOOLSET "--toolset-root=$BOOST_JAM_TOOLSET_ROOT" clean
     fi
-    echo_run ./bootstrap/jam0 -f build.jam --toolset=$BOOST_JAM_TOOLSET "--toolset-root=$BOOST_JAM_TOOLSET_ROOT" "$@"
+    if test -z "$BOOST_JAM_TOOLSET_ROOT"; then
+        echo_run ./bootstrap/jam0 -f build.jam --toolset=$BOOST_JAM_TOOLSET "$@"
+    else
+        echo_run ./bootstrap/jam0 -f build.jam --toolset=$BOOST_JAM_TOOLSET "--toolset-root=$BOOST_JAM_TOOLSET_ROOT" "$@"
+    fi
 fi
