--- xfig.3.2.4.orig/Imakefile   2003-02-23 14:04:25.000000000 +0100
+++ xfig.3.2.4/Imakefile        2003-02-23 14:12:31.000000000 +0100
@@ -105,10 +105,10 @@
 XCOMM Then be sure to change the XAW3DINC to point to the directory where your
 XCOMM 3D Athena widget headers are located
 
-XCOMM #define XAW3D
+#define XAW3D
 
 #ifdef XAW3D
-XAW3DINC = -I/usr/include/X11/Xaw3d
+XAW3DINC = -I/usr/X11R6/include/X11/Xaw3d
 DUSEXAW3D = -DXAW3D
 XAWLIB = -lXaw3d
 #endif


