diff -Naur secure_delete-3.1/Makefile secure_delete-3.1.new/Makefile
--- secure_delete-3.1/Makefile	2003-10-29 22:07:06.000000000 +0100
+++ secure_delete-3.1.new/Makefile	2013-04-24 10:31:27.319782864 +0200
@@ -1,13 +1,13 @@
 CC=gcc
-OPT=-O2 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
+OPT=$(CFLAGS) -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
 #OPT=-Wall -D_DEBUG_ -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE
-INSTALL_DIR=/usr/local/bin
-MAN_DIR=/usr/local/man
+INSTALL_DIR=$(prefix)/bin
+MAN_DIR=$(prefix)/share/man
 DOC_DIR=/usr/share/doc/secure_delete
 OPT_MOD=-D__KERNEL__ -DMODULE -fomit-frame-pointer -fno-strict-aliasing -pipe -mpreferred-stack-boundary=2
 #LD_MOD=-r
 
-all: sdel-lib.o srm sfill sswap smem sdel-mod.o
+all: sdel-lib.o srm sfill sswap smem
 	@echo
 	@echo "A Puritan is someone who is deathly afraid that someone, somewhere, is"
 	@echo "having fun."
@@ -21,16 +21,16 @@
 sdel-lib.o: sdel-lib.c
 	$(CC) ${OPT} -c sdel-lib.c
 
-srm: srm.c
+srm: sdel-lib.o srm.c
 	$(CC) ${OPT} -o srm srm.c sdel-lib.o
 	-strip srm
-sfill: sfill.c
+sfill: sdel-lib.o sfill.c
 	$(CC) ${OPT} -o sfill sfill.c sdel-lib.o
 	-strip sfill
-sswap: sswap.c
+sswap: sdel-lib.o sswap.c
 	$(CC) ${OPT} -o sswap sswap.c sdel-lib.o
 	-strip sswap
-smem: smem.c
+smem: sdel-lib.o smem.c
 	$(CC) ${OPT} -o smem smem.c sdel-lib.o
 	-strip smem
 
@@ -45,9 +45,9 @@
 	mkdir -p -m 755 ${MAN_DIR}/man1 2> /dev/null
 	cp -f srm.1 sfill.1 sswap.1 smem.1 ${MAN_DIR}/man1
 	chmod 644 ${MAN_DIR}/man1/srm.1 ${MAN_DIR}/man1/sfill.1 ${MAN_DIR}/man1/sswap.1 ${MAN_DIR}/man1/smem.1
-	mkdir -p -m 755 ${DOC_DIR} 2> /dev/null
-	cp -f CHANGES FILES README secure_delete.doc usenix6-gutmann.doc ${DOC_DIR}
-	-test -e sdel-mod.o && cp -f sdel-mod.o /lib/modules/`uname -r`/kernel/drivers/char
+#	mkdir -p -m 755 ${DOC_DIR} 2> /dev/null
+#	cp -f CHANGES FILES README secure_delete.doc usenix6-gutmann.doc ${DOC_DIR}
+#	-test -e sdel-mod.o && cp -f sdel-mod.o /lib/modules/`uname -r`/kernel/drivers/char
 #	@-test '!' -e sdel-mod.o -a `uname -s` = 'Linux' && echo "type \"make sdel-mod install\" to compile and install the Linux loadable kernel module for secure delete"
 	@echo
 	@echo "If men could get pregnant, abortion would be a sacrament."
