diff -Naur gplflash-0.4.13.orig/plugin/Makefile.am gplflash-0.4.13/plugin/Makefile.am
--- gplflash-0.4.13.orig/plugin/Makefile.am	2005-01-19 19:30:52.000000000 +0100
+++ gplflash-0.4.13/plugin/Makefile.am	2006-03-11 22:25:18.000000000 +0100
@@ -17,7 +17,7 @@
 
 libnpflash_la_SOURCES = plugin.c npunix.c
 libnpflash_la_CFLAGS = -fpic -fPIC -DPIC $(OPTIMIZER) $(PLUGIN_DEFINES) -I../lib $(X_CFLAGS)
-libnpflash_la_LDFLAGS = -L../lib/.libs  -lflash
+libnpflash_la_LDFLAGS = -L../lib/.libs  -lflash -lXt
 
 install-pluginLTLIBRARIES: $(plugin_LIBRARIES)
 	mkdir -p "$(DESTDIR)$(PLUGIN_DIR)"
