diff -Naur setserial-2.17/setserial.c setserial-2.17-p/setserial.c
--- setserial-2.17/setserial.c	2000-01-27 16:40:52.000000000 +0100
+++ setserial-2.17-p/setserial.c	2017-03-06 15:34:44.691261010 +0100
@@ -15,6 +15,9 @@
 #include <termios.h>
 #include <string.h>
 #include <errno.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
 
 #ifdef HAVE_ASM_IOCTLS_H
 #include <asm/ioctls.h>
