--- ./rc.serial.orig	Thu Jan 27 07:47:30 2000
+++ ./rc.serial	Sun May 13 13:27:57 2001
@@ -14,8 +14,6 @@
 # XXXX For now, the autosave feature doesn't work if you are
 # using the multiport feature; it doesn't save the multiport configuration
 # (for now).  Autosave also doesn't work for the hayes devices.  
-# Will fix later...
-#
 #
 
 RCLOCKFILE=/var/lock/subsys/serial
@@ -87,7 +85,7 @@
 		mv /etc/.serial.conf.new /etc/serial.conf
 		echo "done."
 	fi
-	if test -n $MODULE ; then
+	if test -n "$MODULE" ; then
 		module=`grep $MODULE_REGEXP /proc/modules | awk '{print $1}'`
 		if test -z "$module" ; then 
 			echo "The $DRIVER_NAME driver is not loaded."
@@ -108,22 +106,22 @@
 # If not stop, it must be a start....
 #
 
-if test -n $MODULE -a "$LOADED" != "yes" ; then 
-	if insmod -fm $MODULE $DRIVER_ARG \
-		> /tmp/$DRIVER.map 2> /tmp/$DRIVER.$$; then :; 
+if test -n "$MODULE" -a "$LOADED" != "yes" ; then 
+	if insmod -f $MODULE $DRIVER_ARG ; then
+          true
 	else
 		echo "Couldn't load $DRIVER_NAME driver."
-		echo "See error logs in /tmp/$DRIVER.$$"
 		exit 1
 	fi
-	/bin/rm -f /tmp/$DRIVER.$$
 fi
 
 if test -f /etc/serial.conf ; then
         if test -n ${SETSERIAL} ; then
 		grep -v ^# < /etc/serial.conf | while read device args
 		do
-		    ${SETSERIAL} -z $device $args
+                    if [ ! "$device" = "" -a ! "$args" = "" ]; then
+                        ${SETSERIAL} -z $device $args
+                    fi
 		done 
 	fi
 else
