diff --color -Naur dmraid/1.0.0.rc16-3/dmraid/tools/dmevent_tool.c dmraid.new/1.0.0.rc16-3/dmraid/tools/dmevent_tool.c
--- a/tools/dmevent_tool.c	2010-10-27 13:31:48.000000000 +0200
+++ b/tools/dmevent_tool.c	2025-05-21 15:24:03.143285849 +0200
@@ -123,7 +123,7 @@
 }
 
 /* Increment option counters. */
-static void _process_opt(int opt, const char *cmd)
+static int _process_opt(int opt, const char *cmd)
 {
 	struct opt_def_struct {
 		const char opt;		/* Option character. */
@@ -148,7 +148,7 @@
 
 			optc[o->type]++;
 			optc[OPT_SUM]++;
-			return;
+			return 0;
 		}
 	}
 
