diff -aur qlbar-0.1.4/src/main.cc qlbar-0.1.4-2/src/main.cc
--- qlbar-0.1.4/src/main.cc	2008-08-13 13:08:24.000000000 -0500
+++ qlbar-0.1.4-2/src/main.cc	2009-01-08 04:43:21.000000000 -0600
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <string.h>
 #include <unistd.h>
 #include <getopt.h>
 
diff -aur qlbar-0.1.4/src/qlbar.cc qlbar-0.1.4-2/src/qlbar.cc
--- qlbar-0.1.4/src/qlbar.cc	2008-08-13 13:08:24.000000000 -0500
+++ qlbar-0.1.4-2/src/qlbar.cc	2009-01-08 04:43:21.000000000 -0600
@@ -1,7 +1,7 @@
 #include <stdio.h>
 #include <X11/extensions/shape.h>
 #include <pthread.h>
-
+#include <unistd.h>
 #include "constants.h"
 #include "qlbar.h"
 
diff -aur qlbar-0.1.4/src/qlconf.cc qlbar-0.1.4-2/src/qlconf.cc
--- qlbar-0.1.4/src/qlconf.cc	2008-08-14 12:42:59.000000000 -0500
+++ qlbar-0.1.4-2/src/qlconf.cc	2009-01-08 04:43:21.000000000 -0600
@@ -1,6 +1,7 @@
 #include <regex.h>
 #include <fstream>
 #include <vector>
+#include <string.h>
 #include <stdlib.h>
 
 #include "qlconf.h"
diff -aur qlbar-0.1.4/src/qlitem.cc qlbar-0.1.4-2/src/qlitem.cc
--- qlbar-0.1.4/src/qlitem.cc	2008-08-13 13:08:24.000000000 -0500
+++ qlbar-0.1.4-2/src/qlitem.cc	2009-01-08 04:43:21.000000000 -0600
@@ -1,4 +1,6 @@
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include <unistd.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
