diff -ru springlobby-0.101.old/src/utils/curlhelper.h springlobby-0.101/src/utils/curlhelper.h
--- springlobby-0.101.old/src/utils/curlhelper.h	2011-10-21 21:25:26.727389966 +0200
+++ springlobby-0.101/src/utils/curlhelper.h	2011-10-21 21:25:50.740389420 +0200
@@ -12,7 +12,6 @@
 #endif
 
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 
 extern "C"
diff -ru springlobby-0.101.old/src/utils/downloader.cpp springlobby-0.101/src/utils/downloader.cpp
--- springlobby-0.101.old/src/utils/downloader.cpp	2011-10-21 21:25:26.727389966 +0200
+++ springlobby-0.101/src/utils/downloader.cpp	2011-10-21 21:26:05.284389062 +0200
@@ -29,7 +29,6 @@
 #include "../uiutils.h"
 #include "conversion.h"
 #include <curl/curl.h>
-#include <curl/types.h>
 #include <curl/easy.h>
 #include "curlhelper.h"
 
