diff -urN httpd-2.0.54/modules/ssl/ssl_toolkit_compat.h httpd-2.0.54_fw/modules/ssl/ssl_toolkit_compat.h
--- httpd-2.0.54/modules/ssl/ssl_toolkit_compat.h	2005-02-04 21:21:18.000000000 +0100
+++ httpd-2.0.54_fw/modules/ssl/ssl_toolkit_compat.h	2005-08-25 16:51:42.000000000 +0200
@@ -211,4 +211,11 @@
 #define SSL_SESS_CACHE_NO_INTERNAL  SSL_SESS_CACHE_NO_INTERNAL_LOOKUP
 #endif
 
+#ifndef PEM_F_DEF_CALLBACK
+#ifdef PEM_F_PEM_DEF_CALLBACK
+/* In OpenSSL 0.9.8 PEM_F_DEF_CALLBACK was renamed */
+#define PEM_F_DEF_CALLBACK PEM_F_PEM_DEF_CALLBACK
+#endif
+#endif
+
 #endif /* SSL_TOOLKIT_COMPAT_H */
