From bc589a353e60095445695f8b6977de480643c745 Mon Sep 17 00:00:00 2001
From: Laurent Rene de Cotret <l.rdc@socivolta.com>
Date: Mon, 6 Jan 2025 11:37:07 -0500
Subject: [PATCH] Support for random-1.3

---
 QuickCheck.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/QuickCheck.cabal b/QuickCheck.cabal
index ad64ec72..9cdfd002 100644
--- a/QuickCheck.cabal
+++ b/QuickCheck.cabal
@@ -81,7 +81,7 @@ library
     Build-depends: random >= 1.0.0.3 && < 1.2.0
     cpp-options: -DOLD_RANDOM
   else
-    Build-depends: random >= 1.2.0 && < 1.3
+    Build-depends: random >= 1.2.0 && < 1.4
 
   -- We always use splitmix directly rather than going through StdGen
   -- (it's somewhat more efficient).
