diff -up gmime-2.5.8/tests/Makefile.am.bak gmime-2.5.8/tests/Makefile.am
--- gmime-2.5.8/tests/Makefile.am.bak	2011-06-17 15:31:05.000000000 +0200
+++ gmime-2.5.8/tests/Makefile.am	2011-06-17 15:30:40.000000000 +0200
@@ -39,7 +39,7 @@ endif
 noinst_PROGRAMS = $(AUTOMATED_TESTS) $(MANUAL_TESTS)
 
 DEPS = $(top_builddir)/gmime/libgmime-$(GMIME_API_VERSION).la
-LDADDS = $(top_builddir)/gmime/libgmime-$(GMIME_API_VERSION).la $(GLIB_LIBS)
+LDADDS = $(top_builddir)/gmime/libgmime-$(GMIME_API_VERSION).la -lgpg-error $(GLIB_LIBS)
 
 test_best_SOURCES = test-best.c
 test_best_LDFLAGS = 
diff -up gmime-2.5.8/tests/test-pkcs7.c.bak gmime-2.5.8/tests/test-pkcs7.c
--- gmime-2.5.8/tests/test-pkcs7.c.bak	2011-03-18 04:44:35.000000000 +0100
+++ gmime-2.5.8/tests/test-pkcs7.c	2011-06-17 15:30:15.000000000 +0200
@@ -32,6 +32,7 @@
 
 #include <gmime/gmime.h>
 #include <gmime/gmime-pkcs7-context.h>
+#include <gpg-error.h>
 
 #include "testsuite.h"
 

