diff -urN gimmie-0.2.6.orig/gimmie/gimmie_computer.py gimmie-0.2.6/gimmie/gimmie_computer.py
--- gimmie-0.2.6.orig/gimmie/gimmie_computer.py	2007-04-05 17:16:39.000000000 +0100
+++ gimmie-0.2.6/gimmie/gimmie_computer.py	2007-04-05 17:17:00.000000000 +0100
@@ -483,7 +483,7 @@
 
         self.system_settings_menu_tree = None
         try:
-            self.system_settings_menu_tree = MenuTree("system-settings.menu")
+            self.system_settings_menu_tree = MenuTree("settings.menu")
             self.system_settings_menu_tree.connect("reload", lambda x: self.emit("reload"))
         except ValueError:
             pass
