--- trunk/gimmie/gimmie_computer.py	2007/03/22 00:44:46	385
+++ trunk/gimmie/gimmie_computer.py	2007/03/22 20:22:15	389
@@ -103,7 +103,8 @@
                     uri = volume.get_activation_uri()
                     break
             else:
-                raise ValueError, "Cannot find URI to open for drive '%s'" % drive
+                raise ValueError, "Cannot find URI to open for drive '%s'" % \
+                      drive.get_display_name()
 
         Item.__init__(self,
                       uri=uri,
@@ -214,7 +215,10 @@
         yield self.cd_burner
 
         for drive in self.vol_monitor.get_connected_drives():
-            yield DriveItem(drive)
+            try:
+                yield DriveItem(drive)
+            except ValueError:
+                pass
 
 
 class PrinterItem(Item):
