diff -urN gimmie-0.2.3.orig/gimmie/gimmie_util.py gimmie-0.2.3/gimmie/gimmie_util.py
--- gimmie-0.2.3.orig/gimmie/gimmie_util.py	2007-02-09 17:25:46.000000000 +0000
+++ gimmie-0.2.3/gimmie/gimmie_util.py	2007-02-09 17:26:01.000000000 +0000
@@ -625,7 +625,6 @@
     DEFAULT_BOOKMARKS = [
         ("firefox.desktop", "application/x-desktop", "gimmie.gimmie_applications.DesktopFileItem"),
         ("nautilus.desktop", "application/x-desktop", "gimmie.gimmie_applications.DesktopFileItem"),
-        (os.path.abspath("README"), "text/plain", "gimmie.gimmie_file.FileItem"),
         ("aim:goim?screenname=\"orphennui\"", "gaim/buddy", "gimmie.gimmie_gaim.GaimBuddy"),
         ]
     
