diff -urN gimmie-0.2.1.orig/gimmie_trash.py gimmie-0.2.1/gimmie_trash.py
--- gimmie-0.2.1.orig/gimmie_trash.py	2007-01-19 15:04:19.000000000 +0000
+++ gimmie-0.2.1/gimmie_trash.py	2007-01-19 15:04:49.000000000 +0000
@@ -73,16 +73,16 @@
             if result[1] == gnomevfs.NotSupportedError:
                 continue
 
-            trash_uri = str(result[0])
-            if trash_uri in self.trash_monitors.keys():
-                continue
-
-            self.trash_dirs[volume_uri] = trash_uri
-            self.trash_monitors[trash_uri] = gnomevfs.monitor_add(trash_uri,
-                                                                  gnomevfs.MONITOR_DIRECTORY,
-                                                                  self._trash_dir_changed)
+            #trash_uri = str(result[0])
+            #if trash_uri in self.trash_monitors.keys():
+            #    continue
+
+            #self.trash_dirs[volume_uri] = trash_uri
+            #self.trash_monitors[trash_uri] = gnomevfs.monitor_add(trash_uri,
+            #                                                      gnomevfs.MONITOR_DIRECTORY,
+            #                                                      self._trash_dir_changed)
             # simulate a change to update item count
-            self._trash_dir_changed(trash_uri, None, gnomevfs.MONITOR_EVENT_CHANGED)
+            #self._trash_dir_changed(trash_uri, None, gnomevfs.MONITOR_EVENT_CHANGED)
             break
 
     def _trash_dir_changed(self, monitor_uri, info_uri, event_type):
diff -urN gimmie-0.2.1.orig/gimmie_util.py gimmie-0.2.1/gimmie_util.py
--- gimmie-0.2.1.orig/gimmie_util.py	2007-01-19 15:04:19.000000000 +0000
+++ gimmie-0.2.1/gimmie_util.py	2007-01-19 15:04:58.000000000 +0000
@@ -625,7 +625,6 @@
     DEFAULT_BOOKMARKS = [
         ("firefox.desktop", "application/x-desktop"),
         ("nautilus.desktop", "application/x-desktop"),
-        (os.path.abspath("README"), "text/plain"),
         ("aim:goim?screenname=\"orphennui\"", "gaim/buddy"),
         ]
     
