diff -aur pipslite-1.0.0/ekpd/ecbteg.c fw-pipslite-1.0.0/ekpd/ecbteg.c
--- pipslite-1.0.0/ekpd/ecbteg.c	2006-01-24 01:38:22.000000000 +0100
+++ fw-pipslite-1.0.0/ekpd/ecbteg.c	2007-05-21 20:11:11.000000000 +0200
@@ -3224,7 +3224,7 @@
 				}
 
 				for (cnt=0;(cnt<length)&&(cnt<BD_size-6);cnt++) {
-					(BYTE)pTemp[cnt] = HD_BUFFER[cnt+6];
+					pTemp[cnt] = HD_BUFFER[cnt+6];
 				}
 				length -= cnt;
 			}
diff -aur pipslite-1.0.0/ekpstm/inkbox.c fw-pipslite-1.0.0/ekpstm/inkbox.c
--- pipslite-1.0.0/ekpstm/inkbox.c	2005-11-14 08:17:36.000000000 +0100
+++ fw-pipslite-1.0.0/ekpstm/inkbox.c	2007-05-21 20:45:03.000000000 +0200
@@ -186,7 +186,7 @@
 	return;
 }
 
-enum Inkbox_Id 
+enum Inkset_Id 
 inkbox_get_inkid( unsigned long id )
 {
 	enum Inkbox_Id color;
diff -aur pipslite-1.0.0/ekpstm/main.c fw-pipslite-1.0.0/ekpstm/main.c
--- pipslite-1.0.0/ekpstm/main.c	2005-11-14 08:17:36.000000000 +0100
+++ fw-pipslite-1.0.0/ekpstm/main.c	2007-05-21 20:14:04.000000000 +0200
@@ -70,6 +70,7 @@
           break;
 	  
         default:
+	  return;
         }
     }
   
