diff -Naur bmp-0.9.7.orig/beep/bmp.desktop.in bmp-0.9.7/beep/bmp.desktop.in
--- bmp-0.9.7.orig/beep/bmp.desktop.in	2004-07-07 09:56:34.000000000 +0200
+++ bmp-0.9.7/beep/bmp.desktop.in	2004-12-12 10:58:27.000000000 +0100
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Name=Beep Media Player
 Comment=Play music
+Comment[hu]=Zene lejátszása
 Exec=beep-media-player
 Icon=bmp.svg
 MimeType=audio/x-scpls;audio/x-mpegurl;audio/mpegurl;audio/mp3;audio/x-mp3;audio/mpeg;audio/x-mpeg;audio/x-wav;application/x-ogg
