--- evdi-1.14.6/Makefile	2024-08-14 10:12:08.000000000 +0200
+++ evdi-1.14.6.new/Makefile	2024-08-15 11:53:56.143384002 +0200
@@ -3,7 +3,7 @@
 #
 
 FLAGS=-Werror -Wextra -Wall -Wno-error=missing-field-initializers -Werror=sign-compare
-FLAGS_C=$(FLAGS) -Wmissing-prototypes -Wstrict-prototypes -Werror=discarded-qualifiers
+FLAGS_C=$(FLAGS) -Wmissing-prototypes -Wstrict-prototypes
 FLAGS_CXX=$(FLAGS)
 
 .PHONY: module library pyevdi
