--- evdi-1.14.7/Makefile	2024-09-20 11:14:26.000000000 +0200
+++ evdi-1.14.7.new/Makefile	2024-10-01 15:38:49.954381012 +0200
@@ -2,8 +2,8 @@
 # Copyright (c) 2015 - 2020 DisplayLink (UK) Ltd.
 #
 
-FLAGS=-Werror -Wextra -Wall -Wno-error=missing-field-initializers -Werror=sign-compare
-FLAGS_C=$(FLAGS) -Wmissing-prototypes -Wstrict-prototypes -Werror=discarded-qualifiers
+FLAGS=-Wextra -Wall -Wno-error=missing-field-initializers -Werror=sign-compare
+FLAGS_C=$(FLAGS) -Wmissing-prototypes -Wstrict-prototypes
 FLAGS_CXX=$(FLAGS)
 
 .PHONY: module library pyevdi
