diff -Naur evdi/module/evdi_drm_drv.c evdi.new/module/evdi_drm_drv.c
--- evdi/module/evdi_drm_drv.c	2023-11-13 13:31:34.193919378 +0100
+++ evdi.new/module/evdi_drm_drv.c	2023-11-13 13:32:11.653235481 +0100
@@ -124,9 +124,13 @@
 
 	.fops = &evdi_driver_fops,
 
+#if KERNEL_VERSION(6, 6, 0) > LINUX_VERSION_CODE
 	.prime_fd_to_handle = drm_gem_prime_fd_to_handle,
+#endif
 	.gem_prime_import = drm_gem_prime_import,
+#if KERNEL_VERSION(6, 6, 0) > LINUX_VERSION_CODE
 	.prime_handle_to_fd = drm_gem_prime_handle_to_fd,
+#endif
 #if KERNEL_VERSION(5, 11, 0) <= LINUX_VERSION_CODE || defined(EL8)
 #else
 	.preclose = evdi_driver_preclose,
diff -Naur evdi/module/evdi_fb.c evdi.new/module/evdi_fb.c
--- evdi/module/evdi_fb.c	2023-11-13 13:31:34.227919665 +0100
+++ evdi.new/module/evdi_fb.c	2023-11-13 13:32:11.653235481 +0100
@@ -418,7 +418,10 @@
 	info->fix.smem_start = (unsigned long)efbdev->efb.obj->vmapping;
 
 #if KERNEL_VERSION(4, 20, 0) <= LINUX_VERSION_CODE || defined(EL8)
+// KERNEL_VERSION(6, 6, 0) has removed FBINFO_DEFAULT as it is equal to 0, which is the default value
+#ifdef FBINFO_DEFAULT
 	info->flags = FBINFO_DEFAULT;
+#endif
 #else
 	info->flags = FBINFO_DEFAULT | FBINFO_CAN_FORCE_OUTPUT;
 #endif
