--- evdi-1.5.1/module/evdi_fb.c	2018-10-30 08:41:20.000000000 +0100
+++ evdi-1.5.1.new/module/evdi_fb.c	2018-12-24 17:33:00.795562902 +0100
@@ -367,7 +367,7 @@
 	info->fix.smem_len = size;
 	info->fix.smem_start = (unsigned long)ufbdev->ufb.obj->vmapping;
 
-	info->flags = FBINFO_DEFAULT | FBINFO_CAN_FORCE_OUTPUT;
+	info->flags = FBINFO_DEFAULT;
 	info->fbops = &evdifb_ops;
 #if KERNEL_VERSION(4, 11, 0) > LINUX_VERSION_CODE
 	drm_fb_helper_fill_fix(info, fb->pitches[0], fb->depth);
