diff -Naur evdi-1.6.4/module/evdi_fb.c evdi-1.6.4-p/module/evdi_fb.c
--- evdi-1.6.4/module/evdi_fb.c	2019-11-21 08:42:24.000000000 +0100
+++ evdi-1.6.4-p/module/evdi_fb.c	2020-03-31 01:45:58.118987557 +0200
@@ -561,7 +561,12 @@
 		struct fb_info *info;
 
 		info = efbdev->helper.fbdev;
+#if KERNEL_VERSION(5, 6, 0) > LINUX_VERSION_CODE
 		unlink_framebuffer(info);
+#else
+		unregister_framebuffer(info);
+#endif
+
 	}
 }
 #endif /* CONFIG_FB */
diff -Naur evdi-1.6.4/module/evdi_gem.c evdi-1.6.4-p/module/evdi_gem.c
--- evdi-1.6.4/module/evdi_gem.c	2019-11-21 08:42:24.000000000 +0100
+++ evdi-1.6.4-p/module/evdi_gem.c	2020-03-31 01:37:25.326497172 +0200
@@ -454,6 +454,7 @@
 {
 }
 
+#if KERNEL_VERSION(5, 6, 0) > LINUX_VERSION_CODE
 static void *evdi_dmabuf_kmap(__always_unused struct dma_buf *dma_buf,
 			__always_unused unsigned long page_num)
 {
@@ -466,6 +467,7 @@
 			__always_unused void *addr)
 {
 }
+#endif
 
 #if KERNEL_VERSION(4, 19, 0) > LINUX_VERSION_CODE
 static void *evdi_dmabuf_kmap_atomic(__always_unused struct dma_buf *dma_buf,
@@ -504,7 +506,7 @@
 	.map_atomic = evdi_dmabuf_kmap_atomic,
 	.unmap = evdi_dmabuf_kunmap,
 	.unmap_atomic = evdi_dmabuf_kunmap_atomic,
-#else
+#elif KERNEL_VERSION(5, 6, 0) > LINUX_VERSION_CODE
 	.map = evdi_dmabuf_kmap,
 	.unmap = evdi_dmabuf_kunmap,
 #endif
