--- artsc/Makefile.am
+++ artsc/Makefile.am
@@ -1,4 +1,4 @@
-AM_CFLAGS = -DARTSC_BACKEND='"$(libdir)/libartscbackend.la"' 
+AM_CFLAGS = -DARTSC_BACKEND='"$(libdir)/libartscbackend.so"' 
 AM_CPPFLAGS = -DCOMPILING_ARTSC
 lib_LTLIBRARIES = libartsc.la libartscbackend.la libartsdsp.la libartsdsp_st.la
 FLOWLIBS = $(top_builddir)/flow/libartsflow.la
--- mcop/extensionloader.cc
+++ mcop/extensionloader.cc
@@ -56,7 +56,14 @@
 	/* this will catch all startup classes here */
 	StartupManager::setExtensionLoader(this);
 
-    lt_dlinit();
+    	lt_dlinit();
+
+	string::size_type len = dlfilename.length();
+	if (len > 0 && dlfilename[len-1] == 'a' && dlfilename[len-2] == 'l') {
+          dlfilename[len-1] = 'o';
+          dlfilename[len-2] = 's';
+        }
+
 	handle = lt_dlopen(dlfilename.c_str());
 
 	StartupManager::setExtensionLoader(0);
