--- SDL-1.2.7+1.2.8cvs20041007.orig/include/SDL_endian.h	2004-02-26 14:26:07.000000000 +0000
+++ SDL-1.2.7+1.2.8cvs20041007/include/SDL_endian.h	2005-01-13 00:23:35.304661645 +0000
@@ -68,7 +68,7 @@
 #elif defined(__GNUC__) && defined(__x86_64__)
 static __inline__ Uint16 SDL_Swap16(Uint16 x)
 {
-	__asm__("xchgb %b0,%h0" : "=q" (x) :  "0" (x));
+	__asm__("xchgb %b0,%h0" : "=Q" (x) :  "0" (x));
 	return x;
 }
 #elif defined(__GNUC__) && (defined(__powerpc__) || defined(__ppc__))
