--- hslua-module-text-1.1.0.1/hslua-module-text.cabal	2001-09-09 03:46:40.000000000 +0200
+++ hslua-module-text-1.1.0.1.new/hslua-module-text.cabal	2023-12-12 11:22:25.039958875 +0100
@@ -35,7 +35,7 @@
   build-depends:       base                 >= 4.11   && < 5
                      , hslua-core           >= 2.3    && < 2.4
                      , hslua-packaging      >= 2.3    && < 2.4
-                     , text                 >= 1.2    && < 2.1
+                     , text                 >= 1.2    && < 2.2
   ghc-options:         -Wall
                        -Wincomplete-record-updates
                        -Wnoncanonical-monad-instances
