diff -urN glade3-3.0.1.orig/data/glade-3.desktop.in.in glade3-3.0.1/data/glade-3.desktop.in.in
--- glade3-3.0.1.orig/data/glade-3.desktop.in.in	2006-08-19 19:50:03.000000000 +0100
+++ glade3-3.0.1/data/glade-3.desktop.in.in	2006-08-28 10:41:22.000000000 +0100
@@ -1,7 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
-_Name=Glade Interface Designer
-_GenericName=User Interface Designer
+_Name=Glade 3 Interface Designer
+_GenericName=User Interface Designer (version 3)
 _Comment=Create or open user interface designs for GTK+ applications
 Exec=glade-3 %F
 Terminal=false
