diff -Naur glade3-3.0.0/glade-3.desktop.in glade3-3.0.0-icon/glade-3.desktop.in
--- glade3-3.0.0/glade-3.desktop.in	2006-08-12 03:18:59.000000000 +0200
+++ glade3-3.0.0-icon/glade-3.desktop.in	2006-08-18 23:57:16.305740848 +0200
@@ -7,7 +7,7 @@
 Terminal=false
 StartupNotify=true
 Type=Application
-Icon=glade-3
+Icon=/usr/share/icons/hicolor/48x48/apps/glade-3.png
 Categories=Application;Development;
 MimeType=application/x-glade;
 X-GNOME-Bugzilla-Bugzilla=GNOME
