--- postfix-3.3.3/makedefs	2018-02-03 18:20:46.000000000 +0100
+++ postfix-3.3.3.new/makedefs	2019-02-28 13:45:37.787511865 +0100
@@ -546,7 +546,7 @@
 		: ${SHLIB_ENV="LD_LIBRARY_PATH=`pwd`/lib"}
 		: ${PLUGIN_LD="${CC-gcc} -shared"}
 		;;
-  Linux.[34].*)	SYSTYPE=LINUX$RELEASE_MAJOR
+  Linux.[345].*)	SYSTYPE=LINUX$RELEASE_MAJOR
 		case "$CCARGS" in
 		 *-DNO_DB*) ;;
 		 *-DHAS_DB*) ;;
--- postfix-3.3.3/src/util/sys_defs.h	2019-02-28 14:12:34.763459435 +0100
+++ postfix-3.3.3.new/src/util/sys_defs.h	2019-02-28 13:44:40.428513725 +0100
@@ -748,7 +748,7 @@
  /*
   * LINUX.
   */
-#if defined(LINUX2) || defined(LINUX3) || defined(LINUX4)
+#if defined(LINUX2) || defined(LINUX3) || defined(LINUX4) || defined(LINUX5)
 #define SUPPORTED
 #define UINT32_TYPE	unsigned int
 #define UINT16_TYPE	unsigned short
