diff -Naur postfix-2.11.7/src/util/dict_db.c postfix-2.11.7-p/src/util/dict_db.c
--- postfix-2.11.7/src/util/dict_db.c	2012-01-25 01:41:08.000000000 +0100
+++ postfix-2.11.7-p/src/util/dict_db.c	2016-01-22 01:09:49.633395717 +0100
@@ -693,7 +693,8 @@
 	msg_fatal("set DB cache size %d: %m", dict_db_cache_size);
     if (type == DB_HASH && db->set_h_nelem(db, DICT_DB_NELM) != 0)
 	msg_fatal("set DB hash element count %d: %m", DICT_DB_NELM);
-#if DB_VERSION_MAJOR == 5 || (DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR > 0)
+#if DB_VERSION_MAJOR == 6 || DB_VERSION_MAJOR == 5 || \
+	(DB_VERSION_MAJOR == 4 && DB_VERSION_MINOR > 0)
     if ((errno = db->open(db, 0, db_path, 0, type, db_flags, 0644)) != 0)
 	FREE_RETURN(dict_surrogate(class, path, open_flags, dict_flags,
 				   "open database %s: %m", db_path));
