diff -Naur inetlib-0.4.orig/include/structs.h inetlib-0.4/include/structs.h
--- inetlib-0.4.orig/include/structs.h	2004-07-14 14:11:52.000000000 +0200
+++ inetlib-0.4/include/structs.h	2006-02-14 00:09:07.000000000 +0100
@@ -49,9 +49,6 @@
 
     //! Where the parsed data will be stored.
     struct _irc_message msg;
-
-    //! Where the data will be stored and classified.
-    struct _irc_ordered_msg o_msg;
 };
 
 /*******************************************************************/
