diff -Naur inetlib-0.4.orig/Makefile inetlib-0.4/Makefile
--- inetlib-0.4.orig/Makefile	2006-02-15 01:47:01.474048872 +0100
+++ inetlib-0.4/Makefile	2006-02-15 01:47:08.623961920 +0100
@@ -14,7 +14,7 @@
 CFLAGS += -I$(INCLUDE)
      
 $(LIB)/libinetlib.a: $(OBJS)
-	ld -m elf_i386 -r -o $(LIB)/libinetlib.a $(OBJS)
+	ld -r -o $(LIB)/libinetlib.a $(OBJS)
        
 $(OBJ)/libinetlib.o: $(SRC)/libinetlib.c
 	$(CC) $(CFLAGS) -c $(SRC)/libinetlib.c -o $(OBJ)/libinetlib.o
