--- wayland-protocols-1.24/meson.build	2021-11-23 10:13:33.066573400 +0100
+++ wayland-protocols-1.24.new/meson.build	2021-12-02 14:13:29.765789692 +0100
@@ -91,7 +91,7 @@
 pkgconfig_configuration.set('PACKAGE', 'wayland-protocols')
 pkgconfig_configuration.set('WAYLAND_PROTOCOLS_VERSION', wayland_protocols_version)
 
-pkg_install_dir = join_paths(get_option('datadir'), 'pkgconfig')
+pkg_install_dir = join_paths(get_option('libdir'), 'pkgconfig')
 configure_file(
 	input: 'wayland-protocols.pc.in',
 	output: 'wayland-protocols.pc',
