diff -Naur noip-2.1.3/Makefile noip-2.1.3-p/Makefile
--- noip-2.1.3/Makefile	2006-02-24 17:45:51.000000000 +0100
+++ noip-2.1.3-p/Makefile	2006-08-01 16:19:07.273136270 +0200
@@ -1,8 +1,8 @@
 TGT=noip2
 CC=gcc
 
-PREFIX=/usr/local
-CONFDIR=${PREFIX}/etc
+PREFIX=/usr
+CONFDIR=/etc
 BINDIR=${PREFIX}/bin
 
 # these defines are for Linux
@@ -21,14 +21,12 @@
 # ARCH=sun
 
 ${TGT}: Makefile ${TGT}.c 
-	${CC} -Wall -g -O2 -D${ARCH} -DPREFIX=\"${PREFIX}\" ${TGT}.c -o ${TGT} ${LIBS}
+	${CC} -Wall -g $(CFLAGS) -D${ARCH} -DPREFIX=\"${PREFIX}\" ${TGT}.c -o ${TGT} ${LIBS}
 
 install: ${TGT} 
-	if [ ! -d ${BINDIR} ]; then mkdir -p ${BINDIR};fi
-	if [ ! -d ${CONFDIR} ]; then mkdir -p ${CONFDIR};fi
-	cp ${TGT} ${BINDIR}/${TGT}
-	${BINDIR}/${TGT} -C -c /tmp/no-ip2.conf
-	mv /tmp/no-ip2.conf ${CONFDIR}/no-ip2.conf
+	mkdir -p $(DESTDIR)$(BINDIR)
+	cp $(TGT) $(DESTDIR)$(BINDIR)/$(TGT)
+
 
 package: ${TGT}
 	rm  -f *.bak
