diff -Naur scribus-1.2.4.orig/scribus/scribus.cpp scribus-1.2.4/scribus/scribus.cpp
--- scribus-1.2.4.orig/scribus/scribus.cpp	2005-12-22 08:41:39.000000000 +0100
+++ scribus-1.2.4/scribus/scribus.cpp	2006-01-05 15:22:45.000000000 +0100
@@ -8934,7 +8934,7 @@
 	Prefs.Language = "English";
 	pfad = LIBDIR;
 	pfad += "dicts/";
-	QDir d(pfad, "*.dic", QDir::Name, QDir::Files | QDir::NoSymLinks);
+	QDir d(pfad, "*.dic", QDir::Name, QDir::Files);
 	if ((d.exists()) && (d.count() != 0))
 	{
 		for (uint dc = 0; dc < d.count(); ++dc)
