diff -Naur scribus-1.3.0.orig/scribus/scribus.cpp scribus-1.3.0/scribus/scribus.cpp
--- scribus-1.3.0.orig/scribus/scribus.cpp	2005-07-20 12:44:58.000000000 +0200
+++ scribus-1.3.0/scribus/scribus.cpp	2005-07-20 12:46:14.000000000 +0200
@@ -10293,7 +10293,7 @@
 	LangTransl.clear();
 	Prefs.Language = "English";
 	pfad = QDir::convertSeparators(ScPaths::instance().libDir() + "dicts/");
-	QDir d(pfad, "*.dic", QDir::Name, QDir::Files | QDir::NoSymLinks);
+	QDir d(pfad, "*.dic", QDir::Name, QDir::Files);
 	if ((d.exists()) && (d.count() != 0))
 	{
 		LanguageManager langmgr;
