diff -Naur scribus-1.2.1.orig/scribus.desktop scribus-1.2.1/scribus.desktop
--- scribus-1.2.1.orig/scribus.desktop	2005-01-07 23:48:42.000000000 +0100
+++ scribus-1.2.1/scribus.desktop	2005-01-27 15:11:42.432884744 +0100
@@ -50,7 +50,7 @@
 Exec=scribus %f
 MimeType=application/x-scribus
 Terminal=false
-Icon=scribusicon.png
+Icon=/usr/share/scribus/icons/scribusicon.png
 Type=Application
 X-StandardInstall=false
 Categories=Qt;Office;X-Misc;
