diff -Naur gfpm-mcs-plugin-0.1.1/src/gfpm-plugin.c gfpm-mcs-plugin-0.1.1.new/src/gfpm-plugin.c
--- gfpm-mcs-plugin-0.1.1/src/gfpm-plugin.c	2007-10-07 19:17:24.000000000 +0530
+++ gfpm-mcs-plugin-0.1.1.new/src/gfpm-plugin.c	2008-02-24 12:05:59.000000000 +0530
@@ -40,7 +40,7 @@
 
 static gchar *xterm_argv[] = {
     "xterm",
-    "-T", "GFpm starter -- DONT'T CLOSE THAT WINDOW",
+    "-T", "GFpm starter -- DON'T CLOSE THAT WINDOW",
     "-geometry", "50x1",
     "-e", "su -c " GFPM_BIN,
     NULL 
