--- input-remapper-1.4.2/setup.py	2022-03-19 09:34:57.000000000 +0100
+++ input-remapper-1.4.2.new/setup.py	2022-05-31 19:33:08.656710584 +0200
@@ -116,10 +116,10 @@
         ("/usr/share/input-remapper/", glob.glob("data/*")),
         ("/usr/share/applications/", ["data/input-remapper.desktop"]),
         ("/usr/share/polkit-1/actions/", ["data/input-remapper.policy"]),
-        ("/usr/lib/systemd/system", ["data/input-remapper.service"]),
+        ("/lib/systemd/system", ["data/input-remapper.service"]),
         ("/etc/dbus-1/system.d/", ["data/inputremapper.Control.conf"]),
         ("/etc/xdg/autostart/", ["data/input-remapper-autoload.desktop"]),
-        ("/usr/lib/udev/rules.d", ["data/99-input-remapper.rules"]),
+        ("/lib/udev/rules.d", ["data/99-input-remapper.rules"]),
         ("/usr/bin/", ["bin/input-remapper-gtk"]),
         ("/usr/bin/", ["bin/input-remapper-service"]),
         ("/usr/bin/", ["bin/input-remapper-control"]),
--- input-remapper-1.4.2/data/99-input-remapper.rules	2022-03-19 09:34:57.000000000 +0100
+++ input-remapper-1.4.2.new/data/99-input-remapper.rules	2022-05-31 19:32:14.755293150 +0200
@@ -7,4 +7,5 @@
 #   journalctl -f
 # to get available variables:
 #   udevadm monitor --environment --udev --subsystem input
-ACTION=="add", SUBSYSTEM=="input", RUN+="/bin/input-remapper-control --command autoload --device $env{DEVNAME}"
+ACTION=="add", SUBSYSTEM=="input", RUN+="/usr/bin/key-mapper-control --command autoload --device $env{DEVNAME}"
+
