diff -Naur input-remapper-1.5.0/data/99-input-remapper.rules input-remapper-1.5.0.new/data/99-input-remapper.rules
--- input-remapper-1.5.0/data/99-input-remapper.rules	2022-06-22 20:55:43.000000000 +0200
+++ input-remapper-1.5.0.new/data/99-input-remapper.rules	2022-09-16 15:38:10.817609990 +0200
@@ -7,4 +7,4 @@
 #   journalctl -f
 # to get available variables:
 #   udevadm monitor --environment --udev --subsystem input
-ACTION=="add", SUBSYSTEM=="input", ENV{ID_PATH}!="platform-sound", RUN+="/bin/input-remapper-control --command autoload --device $env{DEVNAME}"
+ACTION=="add", SUBSYSTEM=="input", ENV{ID_PATH}!="platform-sound", RUN+="/usr/bin/input-remapper-control --command autoload --device $env{DEVNAME}"
diff -Naur input-remapper-1.5.0/setup.py input-remapper-1.5.0.new/setup.py
--- input-remapper-1.5.0/setup.py	2022-06-22 20:55:43.000000000 +0200
+++ input-remapper-1.5.0.new/setup.py	2022-09-16 15:37:46.178005082 +0200
@@ -116,10 +116,10 @@
         ("/usr/share/input-remapper/", glob.glob("data/*")),
         ("/usr/share/applications/", ["data/input-remapper.desktop"]),
         ("/usr/share/polkit-1/actions/", ["data/input-remapper.policy"]),
-        ("/usr/lib/systemd/system", ["data/input-remapper.service"]),
+        ("/lib/systemd/system", ["data/input-remapper.service"]),
         ("/etc/dbus-1/system.d/", ["data/inputremapper.Control.conf"]),
         ("/etc/xdg/autostart/", ["data/input-remapper-autoload.desktop"]),
-        ("/usr/lib/udev/rules.d", ["data/99-input-remapper.rules"]),
+        ("/lib/udev/rules.d", ["data/99-input-remapper.rules"]),
         ("/usr/bin/", ["bin/input-remapper-gtk"]),
         ("/usr/bin/", ["bin/input-remapper-service"]),
         ("/usr/bin/", ["bin/input-remapper-control"]),
