diff --git a/src/platformtheme/khintssettings.cpp b/src/platformtheme/khintssettings.cpp
index f225017..90c9690 100644
--- a/src/platformtheme/khintssettings.cpp
+++ b/src/platformtheme/khintssettings.cpp
@@ -146,9 +146,10 @@ QVariant KHintsSettings::readConfigValue(const QString &group, const QString &ke
         }
     }
 
-    KConfigGroup lnfCg(mDefaultLnfConfig, group);
+    KConfigGroup lnfCg(mDefaultLnfConfig, "kdeglobals");
+    lnfCg = KConfigGroup(&lnfCg, group);
     if (lnfCg.isValid()) {
-        return lnfCg.readEntry(key, defaultValue);
+        return  lnfCg.readEntry(key, defaultValue);
     }
 
     return defaultValue;
