diff -urN screenlets-0.0.7.orig/src/share/add-screenlet.py screenlets-0.0.7/src/share/add-screenlet.py
--- screenlets-0.0.7.orig/src/share/add-screenlet.py	2007-04-03 20:56:16.000000000 +0100
+++ screenlets-0.0.7/src/share/add-screenlet.py	2007-04-03 20:56:34.000000000 +0100
@@ -38,7 +38,7 @@
 	try:
 		for cname in names:
 			#print "Adding: "+cname+" to "+str(backend)
-			if backend.add(cname):
+			if backend.add(cname, False):
 				print cname + " successfully created."
 			else:
 				print "Class " + cname + " not found."
