--- eric4-4.3.0/install.py	2009-02-08 14:10:24.000000000 +0100
+++ eric4-4.3.0-p/install.py	2009-02-09 23:34:04.000000000 +0100
@@ -190,7 +190,7 @@
     fname = os.path.join(pdir, "__init__.py")
     if not os.path.exists(fname):
         if not os.path.exists(pdir):
-            os.mkdir(pdir,  0755)
+            os.makedirs(pdir,  0755)
         f = open(fname, "wb")
         f.write(\
 '''# -*- coding: utf-8 -*-
@@ -215,7 +215,10 @@
     if not specialInstall and qtDataDir is None:
         try:
             from PyQt4 import pyqtconfig
-            qtDataDir = pyqtconfig._pkg_config["qt_data_dir"]
+	    if distDir:
+            	qtDataDir = distDir + pyqtconfig._pkg_config["qt_data_dir"]
+	    else:
+		qtDataDir = pyqtconfig._pkg_config["qt_data_dir"]
         except (AttributeError, ImportError):
             print
             print "Please enter the name of the Qt data directory."
@@ -494,7 +497,7 @@
        cfg['ericTemplatesDir'],
        cfg['ericCodeTemplatesDir'], cfg['ericOthersDir'],
        cfg['bindir'], cfg['mdir'], 
-       getAPIsPath(), apis)
+       getAPIsPath()[len(distDir):], apis)
     copyToFile(fn, config)
 
 def doDependancyChecks():    
