diff -up eric4-4.1.1/install.py.patch eric4-4.1.1/install.py
--- eric4-4.1.1/install.py.patch	2008-03-01 23:54:03.000000000 +0100
+++ eric4-4.1.1/install.py	2008-03-01 23:54:24.000000000 +0100
@@ -184,8 +184,6 @@ def createGlobalPluginsDir():
     global cfg, distDir
     
     pdir = os.path.join(cfg['mdir'], "eric4plugins")
-    if distDir:
-        pdir = distDir + pdir
     fname = os.path.join(pdir, "__init__.py")
     if not os.path.exists(fname):
         if not os.path.exists(pdir):
