--- ./install.py.orig	2008-02-03 07:30:54.000000000 -0700
+++ ./install.py	2008-02-04 03:01:52.959774413 -0700
@@ -182,8 +182,11 @@ def createGlobalPluginsDir():
     Create the global plugins directory, if it doesn't exist.
     """
     global modDir
+    global distDir
     
     pdir = os.path.join(modDir, "eric4plugins")
+    if distDir:
+        pdir = distDir + pdir
     fname = os.path.join(pdir, "__init__.py")
     if not os.path.exists(fname):
         if not os.path.exists(pdir):
