diff -Naur bzip2-1.0.4.orig/Makefile bzip2-1.0.4/Makefile
--- bzip2-1.0.4.orig/Makefile	2007-01-03 04:49:21.000000000 +0100
+++ bzip2-1.0.4/Makefile	2007-01-06 14:36:23.000000000 +0100
@@ -21,7 +21,7 @@
 LDFLAGS=
 
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-Wall -Winline -O2 -g $(BIGFILES)
+CFLAGS+=-Wall -Winline $(BIGFILES)
 
 # Where you want it installed when you do 'make install'
 PREFIX=/usr/local
diff -Naur bzip2-1.0.4.orig/Makefile-libbz2_so bzip2-1.0.4/Makefile-libbz2_so
--- bzip2-1.0.4.orig/Makefile-libbz2_so	2007-01-03 03:00:55.000000000 +0100
+++ bzip2-1.0.4/Makefile-libbz2_so	2007-01-06 14:36:43.000000000 +0100
@@ -24,7 +24,7 @@
 SHELL=/bin/sh
 CC=gcc
 BIGFILES=-D_FILE_OFFSET_BITS=64
-CFLAGS=-fpic -fPIC -Wall -Winline -O2 -g $(BIGFILES)
+CFLAGS+=-fpic -fPIC -Wall -Winline $(BIGFILES)
 
 OBJS= blocksort.o  \
       huffman.o    \
