diff -Naur lgeneral-1.2beta-3.orig/src/ai.c lgeneral-1.2beta-3/src/ai.c
--- lgeneral-1.2beta-3.orig/src/ai.c	2005-09-22 15:48:36.000000000 +0200
+++ lgeneral-1.2beta-3/src/ai.c	2005-09-22 15:49:45.000000000 +0200
@@ -508,7 +508,7 @@
 */
 void ai_finalize( void )
 {
-printf(__FUNCTION__ "\n");
+printf("%s\n", __func__);
     if ( finalized )
         return;
 printf("Really finalized\n");
diff -Naur lgeneral-1.2beta-3.orig/src/ai_group.c lgeneral-1.2beta-3/src/ai_group.c
--- lgeneral-1.2beta-3.orig/src/ai_group.c	2005-09-22 15:48:36.000000000 +0200
+++ lgeneral-1.2beta-3/src/ai_group.c	2005-09-22 15:49:53.000000000 +0200
@@ -560,7 +560,7 @@
         }
     }
     if ( unit == 0 ) {
-        printf("ERROR: " __FUNCTION__ ": null unit detected\n");
+        printf("ERROR: %s: null unit detected\n", __func__ );
 	return 0;
     }
     if ( group->state == GS_ART_FIRE )
