--- blop-0.2.8/configure	2004-06-15 11:11:25.000000000 +0200
+++ blop-0.2.8-patched/configure	2006-03-02 04:20:08.000000000 +0100
@@ -4533,8 +4533,7 @@
   enableval="$enable_debug"
   CFLAGS="-pipe -Wall -O0 -g"
 else
-  CFLAGS="-pipe -Wall -O3 -Wno-unused -DNO_DEBUG -DPIC -fPIC \
-            -ffast-math -fomit-frame-pointer -funroll-loops"
+  CFLAGS="-Wall DUMMY  -Wno-unused -DNO_DEBUG -DPIC -fPIC" 
 
 fi;
 
