diff -Naur tap-plugins-0.7.0/Makefile tap-plugins-0.7.0-patched/Makefile
--- tap-plugins-0.7.0/Makefile	2004-08-17 14:34:43.000000000 +0200
+++ tap-plugins-0.7.0-patched/Makefile	2006-03-02 01:06:32.000000000 +0100
@@ -23,8 +23,9 @@
 # Change this if you want to install somewhere else. In particular
 # you may wish to remove the middle "local/" part of the path.
 
-INSTALL_PLUGINS_DIR	=	/usr/local/lib/ladspa/
-INSTALL_LRDF_DIR	=	/usr/local/share/ladspa/rdf/
+DESTDIR			=
+INSTALL_PLUGINS_DIR	=	${DESTDIR}/usr/lib/ladspa/
+INSTALL_LRDF_DIR	=	${DESTDIR}/usr/share/ladspa/rdf/
 
 # NO EDITING below this line is required
 # if all you want to do is install and use the plugins.
