diff -Naur noteedit-2.8.0-orig/configure noteedit-2.8.0/configure
--- noteedit-2.8.0-orig/configure	2005-07-11 19:17:01.000000000 +0200
+++ noteedit-2.8.0/configure	2005-09-02 14:24:01.000000000 +0200
@@ -31041,6 +31041,14 @@
 _ACEOF
 
 		rm -f compvers.cc compvers.o compvers
+	elif test "X$ccversion" = "X4"
+	then
+
+cat >>confdefs.h <<\_ACEOF
+#define GCC_MAJ_VERS 4
+_ACEOF
+
+		rm -f compvers.cc compvers.o compvers
 	else
 		echo ""
 		echo "Cannot determine g++ version"
