diff -Naur noteedit-2.8.0-orig/noteedit/staffProps.ui noteedit-2.8.0/noteedit/staffProps.ui
--- noteedit-2.8.0-orig/noteedit/staffProps.ui	2005-07-05 20:46:39.000000000 +0200
+++ noteedit-2.8.0/noteedit/staffProps.ui	2006-02-17 20:27:09.000000000 +0100
@@ -543,13 +543,5 @@
 </slots>
 <layoutdefaults spacing="6" margin="11"/>
 <includehints>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
 </includehints>
 </UI>
diff -Naur noteedit-2.8.0-orig/noteedit/timesigDia.ui noteedit-2.8.0/noteedit/timesigDia.ui
--- noteedit-2.8.0-orig/noteedit/timesigDia.ui	2005-07-05 20:46:39.000000000 +0200
+++ noteedit-2.8.0/noteedit/timesigDia.ui	2006-02-17 20:26:45.000000000 +0100
@@ -357,7 +357,5 @@
 </slots>
 <layoutdefaults spacing="6" margin="11"/>
 <includehints>
-    <includehint>nscaleedit.h</includehint>
-    <includehint>nscaleedit.h</includehint>
 </includehints>
 </UI>
