diff -ur a/Makefile b/Makefile
--- a/Makefile	2014-02-16 23:44:57.000000000 +0100
+++ b/Makefile	2015-03-18 11:24:52.638426986 +0100
@@ -86,7 +86,7 @@
 $(LLVM_PROGS): LD := g++
 LLVM_LDFLAGS := $(shell llvm-config --ldflags)
 LLVM_CFLAGS := $(shell llvm-config --cflags | sed -e "s/-DNDEBUG//g")
-LLVM_LIBS := $(shell llvm-config --libs)
+LLVM_LIBS := $(shell llvm-config --libs) -pthread -ldl -lncurses 
 PROGRAMS += $(LLVM_PROGS)
 INST_PROGRAMS += sparse-llvm sparsec
 sparse-llvm.o: BASIC_CFLAGS += $(LLVM_CFLAGS)
