--- ./src/openmedialib/plugins/avformat/avformat_plugin.cpp.orig	2009-07-22 11:39:31.000000000 +0200
+++ ./src/openmedialib/plugins/avformat/avformat_plugin.cpp	2009-07-22 11:42:26.000000000 +0200
@@ -28,7 +28,7 @@
 
 
 extern "C" {
-#include <avformat.h>
+#include <libavformat/avformat.h>
 }
 
 namespace oml = olib::openmedialib;
@@ -308,7 +308,7 @@
 
 				// Close the output file
 				if ( !( fmt_->flags & AVFMT_NOFILE ) )
-					url_fclose( &oc_->pb );
+					url_fclose( oc_->pb );
 
 				// Free the context
 				av_free( oc_ );
@@ -1156,7 +1156,7 @@
 			int error = av_open_input_file( &context_, opl::to_string( resource ).c_str( ), format_, 0, params_ ) < 0;
 
 			// Check for streaming
-			if ( error == 0 && url_is_streamed( &context_->pb ) )
+			if ( error == 0 && url_is_streamed( context_->pb ) )
 			{
 				is_seekable_ = false;
 				key_search_ = true;
@@ -1917,7 +1917,7 @@
 {
 	public:
 		// filter_type overloads
-		explicit avformat_resampler_filter::avformat_resampler_filter(const opl::wstring &)
+		explicit avformat_resampler_filter(const opl::wstring &)
 			: filter_type()
 			, prop_output_channels_(pcos::key::from_string("channels"))
 			, prop_output_sample_freq_(pcos::key::from_string("frequency"))
@@ -2012,7 +2012,7 @@
 			}
 		}
 
-		frame_type_ptr avformat_resampler_filter::fetch()
+		frame_type_ptr fetch()
 		{
 			// Get the filter input
 			input_type_ptr input = fetch_slot( );
--- ./src/openmedialib/ml/filter.hpp.orig	2009-07-22 11:34:45.000000000 +0200
+++ ./src/openmedialib/ml/filter.hpp	2009-07-22 11:34:55.000000000 +0200
@@ -8,6 +8,7 @@
 #define OPENMEDIALIB_FILTER_INC_
 
 #include <vector>
+#include <stdio.h>
 #include <boost/shared_ptr.hpp>
 #include <openmedialib/ml/frame.hpp>
 #include <openmedialib/ml/input.hpp>
--- ./src/openimagelib/plugins/jpg/jpg_plugin.cpp.orig	2009-07-22 11:33:49.000000000 +0200
+++ ./src/openimagelib/plugins/jpg/jpg_plugin.cpp	2009-07-22 11:34:01.000000000 +0200
@@ -8,6 +8,7 @@
 #include <cstdlib>
 #include <vector>
 #include <string>
+#include <stdio.h>
 
 #include <openimagelib/plugins/jpg/jpg_plugin.hpp>
 
--- ./src/openimagelib/il/utility.cpp.orig	2009-07-22 11:32:37.000000000 +0200
+++ ./src/openimagelib/il/utility.cpp	2009-07-22 11:32:59.000000000 +0200
@@ -8,6 +8,7 @@
 #include <algorithm>
 #include <cmath>
 #include <limits>
+#include <stdio.h>
 
 using namespace std;
 
