diff -Naur openlibraries-0.4.0/src/openassetlib/al/base64_codec.cpp openlibraries-0.4.0-p/src/openassetlib/al/base64_codec.cpp
--- openlibraries-0.4.0/src/openassetlib/al/base64_codec.cpp	2006-10-18 12:21:03.000000000 +0200
+++ openlibraries-0.4.0-p/src/openassetlib/al/base64_codec.cpp	2008-04-06 00:25:57.000000000 +0200
@@ -5,6 +5,7 @@
 // For more information, see http://www.openlibraries.org.
 
 #include <openassetlib/al/base64_codec.hpp>
+#include <cstring>
 
 namespace olib { namespace openassetlib {
 
diff -Naur openlibraries-0.4.0/src/openeffectslib/fx/content_handler_libxml.cpp openlibraries-0.4.0-p/src/openeffectslib/fx/content_handler_libxml.cpp
--- openlibraries-0.4.0/src/openeffectslib/fx/content_handler_libxml.cpp	2006-05-19 17:25:15.000000000 +0200
+++ openlibraries-0.4.0-p/src/openeffectslib/fx/content_handler_libxml.cpp	2008-04-06 00:50:08.000000000 +0200
@@ -7,6 +7,7 @@
 
 #include <openpluginlib/pl/utf8_utils.hpp>
 #include <openeffectslib/fx/content_handler_libxml.hpp>
+#include <cstring>
 
 namespace olib { namespace openeffectslib { namespace fx {
 
diff -Naur openlibraries-0.4.0/src/openpluginlib/pl/content_handler_libxml.cpp openlibraries-0.4.0-p/src/openpluginlib/pl/content_handler_libxml.cpp
--- openlibraries-0.4.0/src/openpluginlib/pl/content_handler_libxml.cpp	2006-05-19 17:25:16.000000000 +0200
+++ openlibraries-0.4.0-p/src/openpluginlib/pl/content_handler_libxml.cpp	2008-04-05 23:09:10.000000000 +0200
@@ -7,6 +7,7 @@
 
 #include <openpluginlib/pl/utf8_utils.hpp>
 #include <openpluginlib/pl/content_handler_libxml.hpp>
+#include <string.h>
 
 namespace
 {
diff -Naur openlibraries-0.4.0/src/openpluginlib/pl/pcos/key.hpp openlibraries-0.4.0-p/src/openpluginlib/pl/pcos/key.hpp
--- openlibraries-0.4.0/src/openpluginlib/pl/pcos/key.hpp	2006-11-01 12:18:02.000000000 +0100
+++ openlibraries-0.4.0-p/src/openpluginlib/pl/pcos/key.hpp	2008-04-05 23:28:12.000000000 +0200
@@ -12,6 +12,7 @@
 
 // std
 #include <vector>
+#include <iostream>
 
 namespace olib { namespace openpluginlib { namespace pcos {
 
