diff -Naur openlibraries-0.4.0/src/openobjectlib/sg/appearance.cpp openlibraries-0.4.0-p/src/openobjectlib/sg/appearance.cpp
--- openlibraries-0.4.0/src/openobjectlib/sg/appearance.cpp	2006-09-24 10:08:09.000000000 +0200
+++ openlibraries-0.4.0-p/src/openobjectlib/sg/appearance.cpp	2007-05-27 15:50:03.000000000 +0200
@@ -54,7 +54,7 @@
 	{
 		typedef opl::m_url::const_iterator const_iterator;
 		
-		const_iterator I = std::find_if( m.begin( ), m.end( ), fs::exists );
+		const_iterator I = std::find_if( m.begin( ), m.end( ), fs::exists<fs::path> );
 		if( I != m.end( ) )
 		{
 			p = *I;
