diff -ur a/driver/linux/Makefile.in b/driver/linux/Makefile.in
--- a/driver/linux/Makefile.in	2012-05-23 00:35:50.387916633 -0500
+++ b/driver/linux/Makefile.in	2012-05-23 00:53:38.517876921 -0500
@@ -30,15 +30,10 @@
 	$(MAKE) -C $(KDIR) SUBDIRS=$(PWD) modules
 
 install:
-	if [ -e "/lib/udev/rules.d" ] ; then cp -f 20-crystalhd.rules /lib/udev/rules.d/ ; fi
-	if [ -e "/etc/udev/rules.d" ] ; then cp -f 20-crystalhd.rules /etc/udev/rules.d/ ; fi
-	if [ -d "/lib/modules/$(shell uname -r)/kernel/drivers/staging/crystalhd" ] ; \
-		then install -m 0644 crystalhd.ko /lib/modules/$(shell uname -r)/kernel/drivers/staging/crystalhd/ ; \
-	else \
-		install -d /lib/modules/$(shell uname -r)/kernel/drivers/video/broadcom ; \
-		install -m 0644 crystalhd.ko /lib/modules/$(shell uname -r)/kernel/drivers/video/broadcom ; \
-	fi
-	/sbin/depmod -a
+	install -d $(DESTDIR)/lib/udev/rules.d
+	install -m 0644 20-crystalhd.rules $(DESTDIR)/lib/udev/rules.d
+	install -d $(DESTDIR)/lib/modules/$(shell uname -r)/kernel/drivers/video/broadcom
+	install -m 0644 crystalhd.ko $(DESTDIR)/lib/modules/$(shell uname -r)/kernel/drivers/video/broadcom
 
 clean:
 	rm -f *.map *.list *.o *.ko crystalhd.mod.c $(OBJ)
