diff -ur a/linux_lib/libcrystalhd/Makefile b/linux_lib/libcrystalhd/Makefile
--- a/linux_lib/libcrystalhd/Makefile	2012-05-23 01:41:04.676771098 -0500
+++ b/linux_lib/libcrystalhd/Makefile	2012-05-23 01:42:36.461767686 -0500
@@ -24,7 +24,7 @@
 # -D_USE_SHMEM_
 
 CPPFLAGS += ${INCLUDES}
-CPPFLAGS += -O2 -Wall -fPIC -shared -fstrict-aliasing -msse2
+CPPFLAGS += -O2 -Wall -fPIC -shared -fstrict-aliasing
 LDFLAGS = -Wl,-soname,${BCLIB_SL} -pthread
 
 SRCFILES = 	libcrystalhd_if.cpp \
