diff -Naur hal-info-20071011/configure.in hal-info-20071011-wth/configure.in
--- hal-info-20071011/configure.in	2007-09-25 22:32:01.000000000 +0200
+++ hal-info-20071011-wth/configure.in	2007-10-15 15:32:14.000000000 +0200
@@ -9,8 +9,16 @@
 
 PKG_PROG_PKG_CONFIG
 
-if ! $PKG_CONFIG --atleast-version 0.5.10 hal; then
-   AC_MSG_ERROR([hal 0.5.10 or later is required for this version of hal-info])
+
+PKG_CHECK_MODULES([hal], [hal], [have_hal_installed="yes"], [have_hal_installed="no"])
+
+if test "x$have_hal_installed" = "xyes"; then
+   if ! $PKG_CONFIG --atleast-version 0.5.10 hal; then
+	hal_now="`$PKG_CONFIG --modversion hal`"
+   	AC_MSG_WARN([found $hal_now but hal 0.5.10 or later is required for this version of hal-info])
+   fi
+else
+   	AC_MSG_WARN([Hal is not installed but hal 0.5.10 or later is required for this version of hal-info])
 fi
 
 dnl ---------------------------------------------------------------------------
diff -Naur hal-info-20071011/VERSION hal-info-20071011-wth/VERSION
--- hal-info-20071011/VERSION	1970-01-01 01:00:00.000000000 +0100
+++ hal-info-20071011-wth/VERSION	2007-10-15 15:32:40.000000000 +0200
@@ -0,0 +1 @@
+20071011
