diff -Naur gnome-panel-2.12.0.orig/gnome-panel/panel-default-setup.entries.in gnome-panel-2.12.0/gnome-panel/panel-default-setup.entries.in
--- gnome-panel-2.12.0.orig/gnome-panel/panel-default-setup.entries.in	2005-09-08 03:32:33.000000000 +0200
+++ gnome-panel-2.12.0/gnome-panel/panel-default-setup.entries.in	2005-09-08 03:35:52.000000000 +0200
@@ -252,7 +252,7 @@
       <key>objects/browser_launcher/launcher_location</key>
       <schema_key>/schemas/apps/panel/objects/launcher_location</schema_key>
       <value>
-        <string>@DATADIR@/applications/epiphany.desktop</string>
+        <string>@DATADIR@/applications/firefox.desktop</string>
       </value>
     </entry>
 
