diff -urN gnome-panel-2.18.0.orig/gnome-panel/gnome-panel.desktop gnome-panel-2.18.0/gnome-panel/gnome-panel.desktop
--- gnome-panel-2.18.0.orig/gnome-panel/gnome-panel.desktop	2007-03-15 19:52:25.000000000 +0000
+++ gnome-panel-2.18.0/gnome-panel/gnome-panel.desktop	2007-03-15 19:53:11.000000000 +0000
@@ -13,4 +13,4 @@
 X-GNOME-Bugzilla-Version=2.18.0
 Categories=GNOME;GTK;Utility;Core;
 OnlyShowIn=GNOME;
-NoDisplay=True
+NoDisplay=true
diff -urN gnome-panel-2.18.0.orig/gnome-panel/gnome-panel.desktop.in gnome-panel-2.18.0/gnome-panel/gnome-panel.desktop.in
--- gnome-panel-2.18.0.orig/gnome-panel/gnome-panel.desktop.in	2007-03-15 19:52:25.000000000 +0000
+++ gnome-panel-2.18.0/gnome-panel/gnome-panel.desktop.in	2007-03-15 19:53:08.000000000 +0000
@@ -13,4 +13,4 @@
 X-GNOME-Bugzilla-Version=@VERSION@
 Categories=GNOME;GTK;Utility;Core;
 OnlyShowIn=GNOME;
-NoDisplay=True
+NoDisplay=true
