diff -Nrup gnome-panel-2.26.2.orig/libpanel-applet/panel-applet.c gnome-panel-2.26.2/libpanel-applet/panel-applet.c
--- gnome-panel-2.26.2.orig/libpanel-applet/panel-applet.c	2009-05-20 13:44:11.000000000 +0100
+++ gnome-panel-2.26.2/libpanel-applet/panel-applet.c	2009-06-01 23:10:24.318415195 +0100
@@ -161,7 +161,8 @@ panel_applet_associate_schemas_in_dir (G
 			gconf_engine_associate_schema (client->engine,
 						       key, schema_key, error);
 
-			if (gconf_entry_get_value (applet_entry) == NULL ||
+			if (applet_entry == NULL ||
+			    gconf_entry_get_value (applet_entry) == NULL ||
 			    gconf_entry_get_is_default (applet_entry)) {
 				/* unset the key: gconf_client_get_entry()
 				 * brought an invalid entry in the client

