--- trunk/gnome-panel/menu.c	2007/02/27 07:40:38	10268
+++ trunk/gnome-panel/menu.c	2007/02/27 10:27:11	10269
@@ -1237,7 +1237,7 @@
 		if (panel_menu_have_icons ())
 			gtk_image_menu_item_set_image (
 				GTK_IMAGE_MENU_ITEM (menuitem), image);
-	} else
+	} else if (icon_size != GTK_ICON_SIZE_INVALID)
 		g_signal_connect (menuitem, "size_request",
 				  G_CALLBACK (image_menuitem_size_request),
 				  GINT_TO_POINTER (icon_size));
--- trunk/gnome-panel/panel-menu-items.c	2007/02/27 07:40:38	10268
+++ trunk/gnome-panel/panel-menu-items.c	2007/02/27 10:27:11	10269
@@ -904,7 +904,7 @@
 		image = NULL;
 
 	setup_menuitem (GTK_WIDGET (menuitem),
-			panel_menu_icon_get_size (),
+			image ? panel_menu_icon_get_size () : GTK_ICON_SIZE_INVALID,
 			image,
 			_("Places"),
 			TRUE);
@@ -934,7 +934,7 @@
 		image = NULL;
 
 	setup_menuitem (GTK_WIDGET (menuitem),
-			panel_menu_icon_get_size (),
+			image ? panel_menu_icon_get_size () : GTK_ICON_SIZE_INVALID,
 			image,
 			_("System"),
 			TRUE);
