diff -Naur faad2/common/mp4ff/Makefile.am faad2-libmp4ff/common/mp4ff/Makefile.am
--- faad2/common/mp4ff/Makefile.am	2006-05-07 20:09:00.000000000 +0200
+++ faad2-libmp4ff/common/mp4ff/Makefile.am	2007-11-02 14:20:01.000000000 +0100
@@ -1,4 +1,6 @@
-noinst_LTLIBRARIES = libmp4ff.la
+lib_LTLIBRARIES = libmp4ff.la
+
+include_HEADERS = mp4ff.h
 
 libmp4ff_la_CFLAGS = -DUSE_TAGGING=1
 
diff -Naur faad2/common/mp4ff/mp4ff.h faad2-libmp4ff/common/mp4ff/mp4ff.h
--- faad2/common/mp4ff/mp4ff.h	2007-11-01 13:33:29.000000000 +0100
+++ faad2-libmp4ff/common/mp4ff/mp4ff.h	2007-11-02 14:19:30.000000000 +0100
@@ -35,7 +35,7 @@
 extern "C" {
 #endif /* __cplusplus */
 
-#include "mp4ff_int_types.h"
+#include <stdint.h>
 
 /* file callback structure */
 typedef struct
