diff -Naur muttng-20051113.orig/doc/Makefile.in muttng-20051113/doc/Makefile.in
--- muttng-20051113.orig/doc/Makefile.in	2005-10-07 14:11:02.000000000 +0200
+++ muttng-20051113/doc/Makefile.in	2005-11-14 17:57:58.000000000 +0100
@@ -35,7 +35,8 @@
 
 MAKEDOC_CPP = $(CPP) $(INCLUDES) $(DEFS) $(CPPFLAGS) -D_MAKEDOC -C 
 
-DISTFILES = Makefile.in dotlock.man  	\
+DISTFILES = Makefile.in 		\
+	dotlock.man  			\
 	muttbug.man			\
 	mutt.man			\
 	PGP-Notes.txt			\
@@ -53,7 +54,8 @@
 	muttng-manual.xsl		\
 	muttng-tex.xsl			\
 	muttng.sty			\
-	html.xsl chunk.xsl
+	html.xsl			\
+	chunk.xsl
 
 BUILT_DISTFILES = stamp-doc-xml stamp-doc-man stamp-doc-chunked \
 		  manual.txt manual.html
