Author: Andreas Tille <tille@debian.org>
Last-Update: Mon, 26 Feb 2024 07:36:57 +0100
Origin: https://github.com/glennrp/pmt/pull/2/commits/720533055469af660e8e79a127999d1c626e95f7
Forwarded: https://sourceforge.net/p/pmt/bugs/82/
Description: Do not build unused Adler32 code
Bug-Debian: https://bugs.debian.org/1064160

--- a/pngcrush.c
+++ b/pngcrush.c
@@ -5512,8 +5512,10 @@ int main(int argc, char *argv[])
                      * they were already checked in the pngcrush_measure_idat
                      * function
                      */
+# ifdef PNG_IGNORE_ADLER32
                     png_set_option(read_ptr, PNG_IGNORE_ADLER32,
                         PNG_OPTION_ON);
+# endif
                     png_set_crc_action(read_ptr, PNG_CRC_QUIET_USE,
                                        PNG_CRC_QUIET_USE);
                 }
