diff -Naur hal-0.5.6.orig/fdi/policy/10osvendor/10-storage-policy.fdi hal-0.5.6/fdi/policy/10osvendor/10-storage-policy.fdi
--- hal-0.5.6.orig/fdi/policy/10osvendor/10-storage-policy.fdi	2006-01-18 17:46:53.285993500 +0100
+++ hal-0.5.6/fdi/policy/10osvendor/10-storage-policy.fdi	2006-01-18 17:49:51.169110500 +0100
@@ -7,10 +7,11 @@
     <match key="info.udi" string="/org/freedesktop/Hal/devices/computer">
       <merge key="storage.policy.default.mount_root" type="string">/media</merge>
       <merge key="storage.policy.default.use_managed_keyword" type="bool">true</merge>
-      <merge key="storage.policy.default.managed_keyword.primary" type="string">managed</merge>
+      <merge key="storage.policy.default.managed_keyword.primary" type="string">comment=managed</merge>
       <merge key="storage.policy.default.managed_keyword.secondary" type="string">kudzu</merge>
       <merge key="storage.policy.default.mount_option.noauto" type="bool">true</merge>
-      <merge key="storage.policy.default.mount_option.pamconsole" type="bool">true</merge>
+      <merge key="storage.policy.default.mount_option.pamconsole" type="bool">false</merge>
+      <merge key="storage.policy.default.mount_option.user" type="bool">true</merge>
       <merge key="storage.policy.default.mount_option.exec" type="bool">true</merge>
     </match>
   </device>
@@ -96,6 +97,9 @@
     <match key="block.is_volume" bool="true">
       <match key="volume.fsusage" string="filesystem">
 	<!-- skip for drives with the no partitions hint (they are handled above) -->
+	<match key="@block.storage_device:storage.no_partitions_hint" bool="true">
+	    <merge key="volume.policy.desired_mount_point" type="copy_property">@block.storage_device:storage.policy.desired_mount_point</merge>
+	</match>
 	<match key="@block.storage_device:storage.no_partitions_hint" bool="false">
 
 	  <merge key="volume.policy.should_mount" type="bool">true</merge>
