diff -Naur hal-0.5.3.orig/configure.in hal-0.5.3/configure.in
--- hal-0.5.3.orig/configure.in	2005-07-19 14:06:29.000000000 +0200
+++ hal-0.5.3/configure.in	2005-07-19 14:10:01.000000000 +0200
@@ -35,16 +35,16 @@
 
 AC_CHECK_HEADERS([sys/ioccom.h])
 
-AC_CHECK_TYPE(__u8, ,
-              [AC_DEFINE(__s8,  int8_t,   [Signed 8 bits type])
-               AC_DEFINE(__u8,  uint8_t,  [Unsigned 8 bits type])
-               AC_DEFINE(__s16, int16_t,  [Signed 16 bits type])
-               AC_DEFINE(__u16, uint16_t, [Unsigned 16 bits type])
-               AC_DEFINE(__s32, int32_t,  [Signed 32 bits type])
-               AC_DEFINE(__u32, uint32_t, [Unsigned 32 bits type])
-               AC_DEFINE(__s64, int64_t,  [Signed 64 bits type])
-               AC_DEFINE(__u64, uint64_t, [Unsigned 64 bits type])],
-               /usr/include/sys/types.h)
+#AC_CHECK_TYPE(__u8, ,
+#              [AC_DEFINE(__s8,  int8_t,   [Signed 8 bits type])
+#               AC_DEFINE(__u8,  uint8_t,  [Unsigned 8 bits type])
+#               AC_DEFINE(__s16, int16_t,  [Signed 16 bits type])
+#               AC_DEFINE(__u16, uint16_t, [Unsigned 16 bits type])
+#               AC_DEFINE(__s32, int32_t,  [Signed 32 bits type])
+#               AC_DEFINE(__u32, uint32_t, [Unsigned 32 bits type])
+#               AC_DEFINE(__s64, int64_t,  [Signed 64 bits type])
+#               AC_DEFINE(__u64, uint64_t, [Unsigned 64 bits type])],
+#               /usr/include/sys/types.h)
 
 AC_ARG_WITH(os-type,     [  --with-os-type=<os>     Distribution or OS (redhat)])
 AC_ARG_WITH(pid-file,    [  --with-pid-file=<file>  PID file for HAL daemon])
@@ -291,9 +291,9 @@
 AC_MSG_RESULT(yes)
 
 # Check for BLKGETSIZE64
-AC_CHECK_TYPE(pgoff_t, ,
-              [AC_DEFINE(pgoff_t, unsigned long, [Index into the pagecache])],
-               /usr/include/sys/types.h)
+#AC_CHECK_TYPE(pgoff_t, ,
+#              [AC_DEFINE(pgoff_t, unsigned long, [Index into the pagecache])],
+#               /usr/include/sys/types.h)
 AC_MSG_CHECKING(for BLKGETSIZE64)
 AC_TRY_COMPILE(
 [#include <sys/types.h>
diff -Naur hal-0.5.3.orig/volume_id/util.h hal-0.5.3/volume_id/util.h
--- hal-0.5.3.orig/volume_id/util.h	2005-07-19 14:06:29.000000000 +0200
+++ hal-0.5.3/volume_id/util.h	2005-07-19 14:10:28.000000000 +0200
@@ -30,6 +30,7 @@
 #endif
 
 #include <stdint.h>
+#include <asm/types.h>
 
 /* size of superblock buffer, reiserfs block is at 64k */
 #define SB_BUFFER_SIZE				0x11000
