--- redis-8.0.2/src/zmalloc.h	2025-05-27 14:39:18.000000000 +0200
+++ redis-8.0.2.new/src/zmalloc.h	2025-06-16 10:35:43.914438745 +0200
@@ -17,7 +17,7 @@
 
 #if defined(USE_TCMALLOC)
 #define ZMALLOC_LIB ("tcmalloc-" __xstr(TC_VERSION_MAJOR) "." __xstr(TC_VERSION_MINOR))
-#include <google/tcmalloc.h>
+#include <gperftools/tcmalloc.h>
 #if (TC_VERSION_MAJOR == 1 && TC_VERSION_MINOR >= 6) || (TC_VERSION_MAJOR > 1)
 #define HAVE_MALLOC_SIZE 1
 #define zmalloc_size(p) tc_malloc_size(p)
