diff -Naur redis-5.0.1/redis.conf redis-5.0.1-p/redis.conf
--- redis-5.0.1/redis.conf	2018-11-07 13:23:06.000000000 +0100
+++ redis-5.0.1-p/redis.conf	2018-11-09 21:14:13.024160439 +0100
@@ -144,7 +144,7 @@
 #                        UPSTART_JOB or NOTIFY_SOCKET environment variables
 # Note: these supervision methods only signal "process is ready."
 #       They do not enable continuous liveness pings back to your supervisor.
-supervised no
+supervised systemd
 
 # If a pid file is specified, Redis writes it where specified at startup
 # and removes it at exit.
@@ -155,7 +155,7 @@
 #
 # Creating a pid file is best effort: if Redis is not able to create it
 # nothing bad happens, the server will start and run normally.
-pidfile /var/run/redis_6379.pid
+pidfile /var/run/redis.pid
 
 # Specify the server verbosity level.
 # This can be one of:
@@ -168,7 +168,7 @@
 # Specify the log file name. Also the empty string can be used to force
 # Redis to log on the standard output. Note that if you use standard
 # output for logging but daemonize, logs will be sent to /dev/null
-logfile ""
+logfile "/var/log/redis/redis.log"
 
 # To enable logging to the system logger, just set 'syslog-enabled' to yes,
 # and optionally update the other syslog parameters to suit your needs.
@@ -260,7 +260,7 @@
 # The Append Only File will also be created inside this directory.
 #
 # Note that you must specify a directory here, not a file name.
-dir ./
+dir /var/lib/redis/
 
 ################################# REPLICATION #################################
 
diff -Naur redis-5.0.1/sentinel.conf redis-5.0.1-p/sentinel.conf
--- redis-5.0.1/sentinel.conf	2018-11-07 13:23:06.000000000 +0100
+++ redis-5.0.1-p/sentinel.conf	2018-11-09 21:13:51.517032076 +0100
@@ -33,7 +33,7 @@
 # Specify the log file name. Also the empty string can be used to force
 # Sentinel to log on the standard output. Note that if you use standard
 # output for logging but daemonize, logs will be sent to /dev/null
-logfile ""
+logfile "/var/log/redis/sentinel.log"
 
 # sentinel announce-ip <ip>
 # sentinel announce-port <port>
