diff -Naur qt-virt-manager-0.45.75/CMakeLists.txt qt-virt-manager-0.45.75-p/CMakeLists.txt
--- qt-virt-manager-0.45.75/CMakeLists.txt	2017-10-06 10:07:47.000000000 +0200
+++ qt-virt-manager-0.45.75-p/CMakeLists.txt	2017-12-26 04:16:15.533132143 +0100
@@ -1030,7 +1030,7 @@
 endif()
 message(STATUS "Build with Libcacard: ${WITH_LIBCACARD}")
 find_package(LibVirt              REQUIRED)
-find_package("${TERM_WIDGET_LIB}" REQUIRED)
+pkg_check_modules(QTERMWIDGET ${TERM_WIDGET_LIB} REQUIRED)
 find_package(LibVNCServer         REQUIRED)
 include_directories(
     ${GLib2_INCLUDE_DIRS}
