--- SQLiteStudio3/coreSQLiteStudio/common/utils.cpp	2015-11-24 21:09:30.000000000 +0200
+++ SQLiteStudio3.new/coreSQLiteStudio/common/utils.cpp	2015-12-27 00:03:20.562950865 +0200
@@ -11,6 +11,7 @@
 #include <QDebug>
 #include <QRegularExpression>
 #include <QDir>
+#include <QDataStream>
 
 #ifdef Q_OS_LINUX
 #include <sys/utsname.h>
--- SQLiteStudio3/coreSQLiteStudio/db/db.cpp	2015-11-24 21:09:30.000000000 +0200
+++ SQLiteStudio3.new/coreSQLiteStudio/db/db.cpp	2015-12-27 00:19:11.862643145 +0200
@@ -1,6 +1,7 @@
 #include "db.h"
 #include <QMetaEnum>
 #include <QDebug>
+#include <QDataStream>
 
 Db::Db()
 {
--- SQLiteStudio3/sqlitestudiocli/cli_config.cpp	2015-11-24 21:09:31.000000000 +0200
+++ SQLiteStudio3.new/sqlitestudiocli/cli_config.cpp	2015-12-27 00:55:48.910089419 +0200
@@ -1,4 +1,5 @@
 #include "cli_config.h"
+#include <QDataStream>
 
 CFG_DEFINE(Cli)
 
