diff -Naur SQLiteStudio3/guiSQLiteStudio/mainwindow.cpp SQLiteStudio3-p/guiSQLiteStudio/mainwindow.cpp
--- SQLiteStudio3/guiSQLiteStudio/mainwindow.cpp	2016-11-04 16:27:35.000000000 +0100
+++ SQLiteStudio3-p/guiSQLiteStudio/mainwindow.cpp	2018-06-15 23:42:14.359294348 +0200
@@ -42,6 +42,7 @@
 #include "services/codeformatter.h"
 #include <QMdiSubWindow>
 #include <QDebug>
+#include <QStyle>
 #include <QStyleFactory>
 #include <QUiLoader>
 #include <QInputDialog>
