--- gridtables-0.1.0.0/gridtables.cabal	2001-09-09 03:46:40.000000000 +0200
+++ gridtables-0.1.0.0.new/gridtables.cabal	2023-12-12 17:33:51.050947383 +0100
@@ -30,7 +30,7 @@
   build-depends:       base             >= 4.12     && < 5
                      , array
                      , parsec           >= 3.1      && < 3.2
-                     , text             >= 1.1.1.0  && < 1.3 || >= 2.0 && < 2.1
+                     , text             >= 1.1.1.0  && < 1.3 || >= 2.0 && < 2.2
   default-language:    Haskell2010
   default-extensions:  OverloadedStrings
   other-extensions:    FlexibleContexts
