--- qt-x11-free-3.3.5/src/kernel/qsizepolicy.h	2005-09-02 14:43:02.000000000 +0200
+++ qt-x11-free-3.3.5-patched/src/kernel/qsizepolicy.h	2005-09-29 01:09:28.000000000 +0200
@@ -88,7 +88,7 @@
     }
 
     void setHorData( SizeType d ) { data = (Q_UINT32)(data & ~HMask) | d; }
-    void setVerData( SizeType d ) { data = (Q_UINT32)(data & ~(HMask << HSize)) |
+    void setVerData( SizeType d ) { data = (Q_UINT32)(data & ~VMask) |
 					   (d << HSize); }
 
     void setHeightForWidth( bool b ) { data = b ? (Q_UINT32)( data | ( 1 << 2*HSize ) )
