diff -Naur qtractor-0.9.1/ChangeLog qtractor-0.9.1-p/ChangeLog
--- qtractor-0.9.1/ChangeLog	2018-05-29 16:01:11.000000000 +0200
+++ qtractor-0.9.1-p/ChangeLog	2018-06-14 16:08:26.318335231 +0200
@@ -6,6 +6,11 @@
 
 GIT HEAD
 
+- Fix build for Qt >= 5.11.0 (by David Geiger, thanks).
+
+
+0.9.1  2018-05-29  Pre-LAC2018 Release Frenzy.
+
 - Displaying MIDI note(on) events as diamonds instead
   of simple rectangles (aka. Drum Mode) is now being
   introduced as an optional MIDI track property (Drums)
diff -Naur qtractor-0.9.1/qtractor.spec.in qtractor-0.9.1-p/qtractor.spec.in
--- qtractor-0.9.1/qtractor.spec.in	2018-05-29 16:01:11.000000000 +0200
+++ qtractor-0.9.1-p/qtractor.spec.in	2018-06-14 16:08:26.318335231 +0200
@@ -116,7 +116,7 @@
 
 %changelog
 * Tue May 29 2018 Rui Nuno Capela <rncbc@rncbc.org> 0.9.1
-- A pre-LAC2018 release frenzy finale.
+- Pre-LAC2018 release frenzy.
 * Thu Mar 22 2018 Rui Nuno Capela <rncbc@rncbc.org> 0.9.0
 - Early Spring'18 beta release.
 * Tue Jan 30 2018 Rui Nuno Capela <rncbc@rncbc.org> 0.8.6
diff -Naur qtractor-0.9.1/src/qtractorMeter.cpp qtractor-0.9.1-p/src/qtractorMeter.cpp
--- qtractor-0.9.1/src/qtractorMeter.cpp	2018-05-29 16:01:11.000000000 +0200
+++ qtractor-0.9.1-p/src/qtractorMeter.cpp	2018-06-14 16:14:42.908585254 +0200
@@ -1,7 +1,7 @@
 // qtractorMeter.cpp
 //
 /****************************************************************************
-   Copyright (C) 2005-2017, rncbc aka Rui Nuno Capela. All rights reserved.
+   Copyright (C) 2005-2018, rncbc aka Rui Nuno Capela. All rights reserved.
 
    This program is free software; you can redistribute it and/or
    modify it under the terms of the GNU General Public License
@@ -38,7 +38,7 @@
 #include <QResizeEvent>
 
 #include <math.h>
-
+#include <QAction>
 
 //----------------------------------------------------------------------------
 // qtractorMeterScale -- Meter bridge scale widget.
diff -Naur qtractor-0.9.1/src/qtractorMeter.h qtractor-0.9.1-p/src/qtractorMeter.h
--- qtractor-0.9.1/src/qtractorMeter.h	2018-05-29 16:01:11.000000000 +0200
+++ qtractor-0.9.1-p/src/qtractorMeter.h	2018-06-14 16:08:26.318335231 +0200
@@ -1,7 +1,7 @@
 // qtractorMeter.h
 //
 /****************************************************************************
-   Copyright (C) 2005-2017, rncbc aka Rui Nuno Capela. All rights reserved.
+   Copyright (C) 2005-2018, rncbc aka Rui Nuno Capela. All rights reserved.
 
    This program is free software; you can redistribute it and/or
    modify it under the terms of the GNU General Public License
diff -Naur qtractor-0.9.1/src/qtractorTrackButton.cpp qtractor-0.9.1-p/src/qtractorTrackButton.cpp
--- qtractor-0.9.1/src/qtractorTrackButton.cpp	2018-05-29 16:01:11.000000000 +0200
+++ qtractor-0.9.1-p/src/qtractorTrackButton.cpp	2018-06-14 16:09:54.401863293 +0200
@@ -1,7 +1,7 @@
 // qtractorTrackButton.cpp
 //
 /****************************************************************************
-   Copyright (C) 2005-2014, rncbc aka Rui Nuno Capela. All rights reserved.
+   Copyright (C) 2005-2018, rncbc aka Rui Nuno Capela. All rights reserved.
 
    This program is free software; you can redistribute it and/or
    modify it under the terms of the GNU General Public License
@@ -22,11 +22,10 @@
 #include "qtractorAbout.h"
 #include "qtractorTrackButton.h"
 
-#include "qtractorMidiControlObserverForm.h"
-
 #include "qtractorMidiControlObserver.h"
+#include "qtractorMidiControlObserverForm.h"
 
-
+#include <QAction>
 
 //----------------------------------------------------------------------------
 // qtractorMidiControlButton -- MIDI controller observer tool button.
diff -Naur qtractor-0.9.1/src/qtractorTrackButton.h qtractor-0.9.1-p/src/qtractorTrackButton.h
--- qtractor-0.9.1/src/qtractorTrackButton.h	2018-05-29 16:01:11.000000000 +0200
+++ qtractor-0.9.1-p/src/qtractorTrackButton.h	2018-06-14 16:08:26.318335231 +0200
@@ -1,7 +1,7 @@
 // qtractorTrackButton.h
 //
 /****************************************************************************
-   Copyright (C) 2005-2014, rncbc aka Rui Nuno Capela. All rights reserved.
+   Copyright (C) 2005-2018, rncbc aka Rui Nuno Capela. All rights reserved.
 
    This program is free software; you can redistribute it and/or
    modify it under the terms of the GNU General Public License
@@ -24,10 +24,10 @@
 
 #include "qtractorTrack.h"
 
-#include <QPushButton>
-
 #include "qtractorObserverWidget.h"
 
+#include <QPushButton>
+
 
 //----------------------------------------------------------------------------
 // qtractorMidiControlButton -- MIDI controller observer tool button.
diff -Naur qtractor-0.9.1/src/translations/qtractor_cs.ts qtractor-0.9.1-p/src/translations/qtractor_cs.ts
--- qtractor-0.9.1/src/translations/qtractor_cs.ts	2018-05-29 16:01:11.000000000 +0200
+++ qtractor-0.9.1-p/src/translations/qtractor_cs.ts	2018-06-14 16:08:26.320335243 +0200
@@ -9563,12 +9563,12 @@
 <context>
     <name>qtractorMixerMeter</name>
     <message>
-        <location filename="../qtractorMeter.cpp" line="312"/>
+        <location filename="../qtractorMeter.cpp" line="313"/>
         <source>Pan</source>
         <translation>Vyvážení</translation>
     </message>
     <message>
-        <location filename="../qtractorMeter.cpp" line="324"/>
+        <location filename="../qtractorMeter.cpp" line="325"/>
         <source>Gain</source>
         <translation>Zesílení</translation>
     </message>
@@ -12261,17 +12261,17 @@
 <context>
     <name>qtractorTrackButton</name>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="89"/>
+        <location filename="../qtractorTrackButton.cpp" line="91"/>
         <source>Record</source>
         <translation>Nahrát</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="94"/>
+        <location filename="../qtractorTrackButton.cpp" line="96"/>
         <source>Mute</source>
         <translation>Ztlumit</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="99"/>
+        <location filename="../qtractorTrackButton.cpp" line="101"/>
         <source>Solo</source>
         <translation>Sólo</translation>
     </message>
diff -Naur qtractor-0.9.1/src/translations/qtractor_de.ts qtractor-0.9.1-p/src/translations/qtractor_de.ts
--- qtractor-0.9.1/src/translations/qtractor_de.ts	2018-05-29 16:01:11.000000000 +0200
+++ qtractor-0.9.1-p/src/translations/qtractor_de.ts	2018-06-14 16:08:26.321335249 +0200
@@ -9445,12 +9445,12 @@
 <context>
     <name>qtractorMixerMeter</name>
     <message>
-        <location filename="../qtractorMeter.cpp" line="312"/>
+        <location filename="../qtractorMeter.cpp" line="313"/>
         <source>Pan</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <location filename="../qtractorMeter.cpp" line="324"/>
+        <location filename="../qtractorMeter.cpp" line="325"/>
         <source>Gain</source>
         <translation type="unfinished">Verstärkung</translation>
     </message>
@@ -12117,17 +12117,17 @@
 <context>
     <name>qtractorTrackButton</name>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="89"/>
+        <location filename="../qtractorTrackButton.cpp" line="91"/>
         <source>Record</source>
         <translation>Aufnahme</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="94"/>
+        <location filename="../qtractorTrackButton.cpp" line="96"/>
         <source>Mute</source>
         <translation type="unfinished"></translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="99"/>
+        <location filename="../qtractorTrackButton.cpp" line="101"/>
         <source>Solo</source>
         <translation type="unfinished"></translation>
     </message>
diff -Naur qtractor-0.9.1/src/translations/qtractor_es.ts qtractor-0.9.1-p/src/translations/qtractor_es.ts
--- qtractor-0.9.1/src/translations/qtractor_es.ts	2018-05-29 16:01:11.000000000 +0200
+++ qtractor-0.9.1-p/src/translations/qtractor_es.ts	2018-06-14 16:08:26.322335255 +0200
@@ -9559,12 +9559,12 @@
 <context>
     <name>qtractorMixerMeter</name>
     <message>
-        <location filename="../qtractorMeter.cpp" line="312"/>
+        <location filename="../qtractorMeter.cpp" line="313"/>
         <source>Pan</source>
         <translation>Panorámica</translation>
     </message>
     <message>
-        <location filename="../qtractorMeter.cpp" line="324"/>
+        <location filename="../qtractorMeter.cpp" line="325"/>
         <source>Gain</source>
         <translation>Ganancia</translation>
     </message>
@@ -12256,17 +12256,17 @@
 <context>
     <name>qtractorTrackButton</name>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="89"/>
+        <location filename="../qtractorTrackButton.cpp" line="91"/>
         <source>Record</source>
         <translation>Grabar</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="94"/>
+        <location filename="../qtractorTrackButton.cpp" line="96"/>
         <source>Mute</source>
         <translation>Silenciar</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="99"/>
+        <location filename="../qtractorTrackButton.cpp" line="101"/>
         <source>Solo</source>
         <translation>Solo</translation>
     </message>
diff -Naur qtractor-0.9.1/src/translations/qtractor_fr.ts qtractor-0.9.1-p/src/translations/qtractor_fr.ts
--- qtractor-0.9.1/src/translations/qtractor_fr.ts	2018-05-29 16:01:11.000000000 +0200
+++ qtractor-0.9.1-p/src/translations/qtractor_fr.ts	2018-06-14 16:08:26.323335261 +0200
@@ -9562,12 +9562,12 @@
 <context>
     <name>qtractorMixerMeter</name>
     <message>
-        <location filename="../qtractorMeter.cpp" line="312"/>
+        <location filename="../qtractorMeter.cpp" line="313"/>
         <source>Pan</source>
         <translation>Pan</translation>
     </message>
     <message>
-        <location filename="../qtractorMeter.cpp" line="324"/>
+        <location filename="../qtractorMeter.cpp" line="325"/>
         <source>Gain</source>
         <translation>Gain</translation>
     </message>
@@ -12260,17 +12260,17 @@
 <context>
     <name>qtractorTrackButton</name>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="89"/>
+        <location filename="../qtractorTrackButton.cpp" line="91"/>
         <source>Record</source>
         <translation>Enregistrer</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="94"/>
+        <location filename="../qtractorTrackButton.cpp" line="96"/>
         <source>Mute</source>
         <translation>Muet</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="99"/>
+        <location filename="../qtractorTrackButton.cpp" line="101"/>
         <source>Solo</source>
         <translation>Solo</translation>
     </message>
diff -Naur qtractor-0.9.1/src/translations/qtractor_it.ts qtractor-0.9.1-p/src/translations/qtractor_it.ts
--- qtractor-0.9.1/src/translations/qtractor_it.ts	2018-05-29 16:01:11.000000000 +0200
+++ qtractor-0.9.1-p/src/translations/qtractor_it.ts	2018-06-14 16:08:26.325335273 +0200
@@ -9563,12 +9563,12 @@
 <context>
     <name>qtractorMixerMeter</name>
     <message>
-        <location filename="../qtractorMeter.cpp" line="312"/>
+        <location filename="../qtractorMeter.cpp" line="313"/>
         <source>Pan</source>
         <translation type="unfinished">Bilanciamento</translation>
     </message>
     <message>
-        <location filename="../qtractorMeter.cpp" line="324"/>
+        <location filename="../qtractorMeter.cpp" line="325"/>
         <source>Gain</source>
         <translation type="unfinished">Guadagno</translation>
     </message>
@@ -12260,17 +12260,17 @@
 <context>
     <name>qtractorTrackButton</name>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="89"/>
+        <location filename="../qtractorTrackButton.cpp" line="91"/>
         <source>Record</source>
         <translation>Registra</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="94"/>
+        <location filename="../qtractorTrackButton.cpp" line="96"/>
         <source>Mute</source>
         <translation>Mute</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="99"/>
+        <location filename="../qtractorTrackButton.cpp" line="101"/>
         <source>Solo</source>
         <translation>Solo</translation>
     </message>
diff -Naur qtractor-0.9.1/src/translations/qtractor_ja.ts qtractor-0.9.1-p/src/translations/qtractor_ja.ts
--- qtractor-0.9.1/src/translations/qtractor_ja.ts	2018-05-29 16:01:11.000000000 +0200
+++ qtractor-0.9.1-p/src/translations/qtractor_ja.ts	2018-06-14 16:08:26.326335279 +0200
@@ -9561,12 +9561,12 @@
 <context>
     <name>qtractorMixerMeter</name>
     <message>
-        <location filename="../qtractorMeter.cpp" line="312"/>
+        <location filename="../qtractorMeter.cpp" line="313"/>
         <source>Pan</source>
         <translation>パン</translation>
     </message>
     <message>
-        <location filename="../qtractorMeter.cpp" line="324"/>
+        <location filename="../qtractorMeter.cpp" line="325"/>
         <source>Gain</source>
         <translation>ゲイン</translation>
     </message>
@@ -12259,17 +12259,17 @@
 <context>
     <name>qtractorTrackButton</name>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="89"/>
+        <location filename="../qtractorTrackButton.cpp" line="91"/>
         <source>Record</source>
         <translation>録音</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="94"/>
+        <location filename="../qtractorTrackButton.cpp" line="96"/>
         <source>Mute</source>
         <translation>ミュート</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="99"/>
+        <location filename="../qtractorTrackButton.cpp" line="101"/>
         <source>Solo</source>
         <translation>ソロ</translation>
     </message>
diff -Naur qtractor-0.9.1/src/translations/qtractor_pt.ts qtractor-0.9.1-p/src/translations/qtractor_pt.ts
--- qtractor-0.9.1/src/translations/qtractor_pt.ts	2018-05-29 16:01:11.000000000 +0200
+++ qtractor-0.9.1-p/src/translations/qtractor_pt.ts	2018-06-14 16:08:26.327335285 +0200
@@ -9562,12 +9562,12 @@
 <context>
     <name>qtractorMixerMeter</name>
     <message>
-        <location filename="../qtractorMeter.cpp" line="312"/>
+        <location filename="../qtractorMeter.cpp" line="313"/>
         <source>Pan</source>
         <translation>Panorâmico</translation>
     </message>
     <message>
-        <location filename="../qtractorMeter.cpp" line="324"/>
+        <location filename="../qtractorMeter.cpp" line="325"/>
         <source>Gain</source>
         <translation>Ganho</translation>
     </message>
@@ -12261,17 +12261,17 @@
 <context>
     <name>qtractorTrackButton</name>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="94"/>
+        <location filename="../qtractorTrackButton.cpp" line="96"/>
         <source>Mute</source>
         <translation>Mudo</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="99"/>
+        <location filename="../qtractorTrackButton.cpp" line="101"/>
         <source>Solo</source>
         <translation>Solo</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="89"/>
+        <location filename="../qtractorTrackButton.cpp" line="91"/>
         <source>Record</source>
         <translation>Gravar</translation>
     </message>
diff -Naur qtractor-0.9.1/src/translations/qtractor_ru.ts qtractor-0.9.1-p/src/translations/qtractor_ru.ts
--- qtractor-0.9.1/src/translations/qtractor_ru.ts	2018-05-29 16:01:11.000000000 +0200
+++ qtractor-0.9.1-p/src/translations/qtractor_ru.ts	2018-06-14 16:08:26.328335291 +0200
@@ -9476,12 +9476,12 @@
 <context>
     <name>qtractorMixerMeter</name>
     <message>
-        <location filename="../qtractorMeter.cpp" line="312"/>
+        <location filename="../qtractorMeter.cpp" line="313"/>
         <source>Pan</source>
         <translation type="unfinished">Панорама</translation>
     </message>
     <message>
-        <location filename="../qtractorMeter.cpp" line="324"/>
+        <location filename="../qtractorMeter.cpp" line="325"/>
         <source>Gain</source>
         <translation type="unfinished">Усиление</translation>
     </message>
@@ -12153,17 +12153,17 @@
 <context>
     <name>qtractorTrackButton</name>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="89"/>
+        <location filename="../qtractorTrackButton.cpp" line="91"/>
         <source>Record</source>
         <translation>Включить режим записи</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="94"/>
+        <location filename="../qtractorTrackButton.cpp" line="96"/>
         <source>Mute</source>
         <translation>Приглушить</translation>
     </message>
     <message>
-        <location filename="../qtractorTrackButton.cpp" line="99"/>
+        <location filename="../qtractorTrackButton.cpp" line="101"/>
         <source>Solo</source>
         <translation>Сделать солирующей</translation>
     </message>
