diff -Naur giflib-5.1.9/Makefile giflib-5.1.9-p/Makefile
--- giflib-5.1.9/Makefile	2019-03-28 19:57:23.000000000 +0100
+++ giflib-5.1.9-p/Makefile	2019-03-30 18:29:09.215698103 +0100
@@ -6,19 +6,18 @@
 # of code space in the shared library.
 
 #
-OFLAGS = -O0 -g
-OFLAGS  = -O2
-CFLAGS  = -std=gnu99 -fPIC -Wall -Wno-format-truncation $(OFLAGS)
+CC    = gcc
+CFLAGS  += -std=gnu99 -fPIC -Wall -Wno-format-truncation
 
 SHELL = /bin/sh
 TAR = tar
 INSTALL = install
 
-PREFIX = /usr/local
-BINDIR = $(PREFIX)/bin
-INCDIR = $(PREFIX)/include
-LIBDIR = $(PREFIX)/lib
-MANDIR = $(PREFIX)/share/man
+PREFIX ?= /usr
+BINDIR ?= $(PREFIX)/bin
+INCDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
+MANDIR ?= $(PREFIX)/share/man
 
 # No user-serviceable parts below this line
 
@@ -28,15 +27,11 @@
 LIBPOINT=0
 LIBVER=$(LIBMAJOR).$(LIBMINOR).$(LIBPOINT)
 
-SOURCES = dgif_lib.c egif_lib.c gifalloc.c gif_err.c gif_font.c \
-	gif_hash.c openbsd-reallocarray.c
-HEADERS = gif_hash.h  gif_lib.h  gif_lib_private.h
+SOURCES = dgif_lib.c egif_lib.c getarg.c gifalloc.c gif_err.c gif_font.c \
+	gif_hash.c openbsd-reallocarray.c qprintf.c quantize.c
+HEADERS = getarg.h  gif_hash.h  gif_lib.h  gif_lib_private.h
 OBJECTS = $(SOURCES:.c=.o)
 
-USOURCES = qprintf.c quantize.c getarg.c 
-UHEADERS = getarg.h
-UOBJECTS = $(USOURCES:.c=.o)
-
 # Some utilities are installed
 INSTALLABLE = \
 	gif2rgb \
@@ -62,10 +57,9 @@
 
 LDLIBS=libgif.a -lm
 
-all: libgif.so libgif.a libutil.so libutil.a $(UTILS)
-	$(MAKE) -C doc
+all: libgif.so libgif.a $(UTILS)
 
-$(UTILS):: libgif.a libutil.a
+$(UTILS):: libgif.a
 
 libgif.so: $(OBJECTS) $(HEADERS)
 	$(CC) $(CFLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,libgif.so.$(LIBMAJOR) -o libgif.so $(OBJECTS)
@@ -73,14 +67,8 @@
 libgif.a: $(OBJECTS) $(HEADERS)
 	$(AR) rcs libgif.a $(OBJECTS)
 
-libutil.so: $(UOBJECTS) $(UHEADERS)
-	$(CC) $(CFLAGS) -shared $(LDFLAGS) -Wl,-soname -Wl,libutil.so.$(LIBMAJOR) -o libutil.so $(UOBJECTS)
-
-libutil.a: $(UOBJECTS) $(UHEADERS)
-	$(AR) rcs libutil.a $(UOBJECTS)
-
 clean:
-	rm -f $(UTILS) $(TARGET) libgetarg.a libgif.a libgif.so libutil.a libutil.so *.o
+	rm -f $(UTILS) $(TARGET) libgetarg.a libgif.a libgif.so
 	rm -f libgif.so.$(LIBMAJOR).$(LIBMINOR).$(LIBPOINT)
 	rm -f libgif.so.$(LIBMAJOR)
 	rm -fr doc/*.1 *.html doc/staging
@@ -134,7 +122,7 @@
 	     doc/whatsinagif \
 
 DSOURCES = Makefile *.[ch]
-DOCS = doc/*.xml doc/*.1 doc/*.html doc/index.html.in doc/00README doc/Makefile
+DOCS = doc/*.xml doc/*.1 doc/*.html doc/*.txt doc/index.html.in doc/00README doc/Makefile
 ALL =  $(DSOURCES) $(DOCS) tests pic $(EXTRAS)
 giflib-$(VERSION).tar.gz: $(ALL)
 	$(TAR) --transform='s:^:giflib-$(VERSION)/:' -czf giflib-$(VERSION).tar.gz $(ALL)
