diff -Naur giflib-5.1.6/Makefile giflib-5.1.6-p/Makefile
--- giflib-5.1.6/Makefile	2019-02-12 15:26:13.000000000 +0100
+++ giflib-5.1.6-p/Makefile	2019-03-02 23:11:47.148313484 +0100
@@ -16,7 +16,7 @@
 TAR = tar
 INSTALL = install
 
-PREFIX = /usr/local
+PREFIX = /usr
 BINDIR = $(PREFIX)/bin
 INCDIR = $(PREFIX)/include
 LIBDIR = $(PREFIX)/lib
@@ -61,7 +61,6 @@
 LDLIBS=libgif.a -lm
 
 all: libgif.so libgif.a $(UTILS)
-	$(MAKE) -C doc
 
 $(UTILS):: libgif.a
 
@@ -96,8 +95,8 @@
 	ln -sf libgif.so.$(LIBVER) "$(DESTDIR)$(LIBDIR)/libgif.so.$(LIBMAJOR)"
 	ln -sf libgif.so.$(LIBMAJOR) "$(DESTDIR)$(LIBDIR)/libgif.so"
 install-man:
-	$(INSTALL) -d "$(DESTDIR)$(MANDIR)"
-	$(INSTALL) -m 644 doc/*.1 "$(DESTDIR)$(MANDIR)"
+	$(INSTALL) -d "$(DESTDIR)$(MANDIR)/man1"
+	$(INSTALL) -m 644 doc/*.1 "$(DESTDIR)$(MANDIR)/man1"
 uninstall: uninstall-man uninstall-include uninstall-lib uninstall-bin
 uninstall-bin:
 	cd "$(DESTDIR)$(BINDIR)" && rm -f $(INSTALLABLE)
