diff -Naur giflib-5.2.2/Makefile giflib-5.2.2.new/Makefile
--- giflib-5.2.2/Makefile	2024-02-19 02:01:50.000000000 +0100
+++ giflib-5.2.2.new/Makefile	2024-02-20 13:24:51.688096773 +0100
@@ -6,19 +6,18 @@
 # of code space in the shared library.
 
 #
-OFLAGS = -O0 -g
-OFLAGS  = -O2
-CFLAGS  = -std=gnu99 -fPIC -Wall -Wno-format-truncation $(OFLAGS)
+CC    = gcc
+CFLAGS  += -std=gnu99 -fPIC -Wall -Wno-format-truncation
 
 SHELL = /bin/sh
 TAR = tar
 INSTALL = install
 
-PREFIX = /usr/local
-BINDIR = $(PREFIX)/bin
-INCDIR = $(PREFIX)/include
-LIBDIR = $(PREFIX)/lib
-MANDIR = $(PREFIX)/share/man
+PREFIX ?= /usr
+BINDIR ?= $(PREFIX)/bin
+INCDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
+MANDIR ?= $(PREFIX)/share/man
 
 # No user-serviceable parts below this line
 
