diff -Naur giflib-5.1.9/Makefile giflib-5.1.9-p/Makefile
--- giflib-5.1.9/Makefile	2019-03-28 19:57:23.000000000 +0100
+++ giflib-5.1.9-p/Makefile	2019-03-29 00:22:15.739599282 +0100
@@ -6,19 +6,17 @@
 # of code space in the shared library.
 
 #
-OFLAGS = -O0 -g
-OFLAGS  = -O2
-CFLAGS  = -std=gnu99 -fPIC -Wall -Wno-format-truncation $(OFLAGS)
+CFLAGS  += -std=gnu99 -fPIC -Wall -Wno-format-truncation
 
 SHELL = /bin/sh
 TAR = tar
 INSTALL = install
 
-PREFIX = /usr/local
-BINDIR = $(PREFIX)/bin
-INCDIR = $(PREFIX)/include
-LIBDIR = $(PREFIX)/lib
-MANDIR = $(PREFIX)/share/man
+PREFIX ?= /usr
+BINDIR ?= $(PREFIX)/bin
+INCDIR ?= $(PREFIX)/include
+LIBDIR ?= $(PREFIX)/lib
+MANDIR ?= $(PREFIX)/share/man
 
 # No user-serviceable parts below this line
 
@@ -63,7 +61,6 @@
 LDLIBS=libgif.a -lm
 
 all: libgif.so libgif.a libutil.so libutil.a $(UTILS)
-	$(MAKE) -C doc
 
 $(UTILS):: libgif.a libutil.a
 
