--- scipy-1.11.3/pyproject.toml	2023-09-27 21:51:29.000000000 +0200
+++ scipy-1.11.3.new/pyproject.toml	2023-11-02 03:01:14.661029919 +0100
@@ -16,7 +16,7 @@
     # (too many changes in 3.0)
     "Cython>=0.29.35,<3.0",   # when updating version, also update check in meson.build
     # Reason for `<`: future-proofing (2.11.1 is the most recent version)
-    "pybind11>=2.10.4,<2.11.1",
+    "pybind11>=2.10.4",
     # Reason for `<`: future-proofing (0.14.0 released and known to work)
     "pythran>=0.12.0,<0.15.0",
 
