diff -Naur libgnome-2.10.0.orig/schemas/desktop_gnome_interface.schemas.in libgnome-2.10.0/schemas/desktop_gnome_interface.schemas.in
--- libgnome-2.10.0.orig/schemas/desktop_gnome_interface.schemas.in	2004-08-18 09:21:38.000000000 +0200
+++ libgnome-2.10.0/schemas/desktop_gnome_interface.schemas.in	2005-03-17 14:15:47.854697520 +0100
@@ -141,7 +141,7 @@
       <applyto>/desktop/gnome/interface/gtk_theme</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Simple</default>
+      <default>Clearlooks</default>
       <locale name="C">
 	<short>Gtk+ Theme</short>
 	<long>Basename of the default theme used by gtk+.</long>
