diff -Naur libgnome-2.20.0.orig/schemas/desktop_gnome_applications_browser.schemas libgnome-2.20.0/schemas/desktop_gnome_applications_browser.schemas
--- libgnome-2.20.0.orig/schemas/desktop_gnome_applications_browser.schemas	2007-09-21 23:28:33.000000000 +0200
+++ libgnome-2.20.0/schemas/desktop_gnome_applications_browser.schemas	2007-09-21 23:29:04.000000000 +0200
@@ -6,7 +6,7 @@
       <applyto>/desktop/gnome/applications/browser/exec</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>mozilla</default>
+      <default>firefox</default>
       <locale name="C">
         <short>Default browser</short>
         <long>Default browser for all URLs</long>
diff -Naur libgnome-2.20.0.orig/schemas/desktop_gnome_applications_browser.schemas.in libgnome-2.20.0/schemas/desktop_gnome_applications_browser.schemas.in
--- libgnome-2.20.0.orig/schemas/desktop_gnome_applications_browser.schemas.in	2007-09-21 23:28:33.000000000 +0200
+++ libgnome-2.20.0/schemas/desktop_gnome_applications_browser.schemas.in	2007-09-21 23:29:40.000000000 +0200
@@ -6,7 +6,7 @@
       <applyto>/desktop/gnome/applications/browser/exec</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>mozilla</default>
+      <default>firefox</default>
       <locale name="C">
         <short>Default browser</short>
         <long>Default browser for all URLs</long>
diff -Naur libgnome-2.20.0.orig/schemas/desktop_gnome_background.schemas libgnome-2.20.0/schemas/desktop_gnome_background.schemas
--- libgnome-2.20.0.orig/schemas/desktop_gnome_background.schemas	2007-09-21 23:28:33.000000000 +0200
+++ libgnome-2.20.0/schemas/desktop_gnome_background.schemas	2007-09-21 23:30:50.000000000 +0200
@@ -381,7 +381,7 @@
         <applyto>/desktop/gnome/background/picture_options</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>zoom</default>
+        <default>stretched</default>
         <locale name="C">
         <short>Picture Options</short>
         <long>Determines how the image set by wallpaper_filename is
@@ -780,7 +780,7 @@
         <applyto>/desktop/gnome/background/picture_filename</applyto>
         <owner>gnome</owner>
         <type>string</type>
-	<default>/usr/share/pixmaps/backgrounds/gnome/background-default.jpg</default>
+	<default>/usr/share/pixmaps/backgrounds/gnome/frugalware/frugalware.jpg</default>
         <locale name="C">
         <short>Picture Filename</short>
         <long>File to use for the background image</long>
diff -Naur libgnome-2.20.0.orig/schemas/desktop_gnome_background.schemas.in libgnome-2.20.0/schemas/desktop_gnome_background.schemas.in
--- libgnome-2.20.0.orig/schemas/desktop_gnome_background.schemas.in	2007-09-21 23:28:33.000000000 +0200
+++ libgnome-2.20.0/schemas/desktop_gnome_background.schemas.in	2007-09-21 23:31:19.000000000 +0200
@@ -16,7 +16,7 @@
         <applyto>/desktop/gnome/background/picture_options</applyto>
         <owner>gnome</owner>
         <type>string</type>
-        <default>zoom</default>
+        <default>stretched</default>
         <locale name="C">
         <short>Picture Options</short>
         <long>Determines how the image set by wallpaper_filename is
@@ -29,7 +29,7 @@
         <applyto>/desktop/gnome/background/picture_filename</applyto>
         <owner>gnome</owner>
         <type>string</type>
-	<default>/usr/share/pixmaps/backgrounds/gnome/background-default.jpg</default>
+	<default>/usr/share/pixmaps/backgrounds/gnome/frugalware/frugalware.jpg</default>
         <locale name="C">
         <short>Picture Filename</short>
         <long>File to use for the background image</long>
diff -Naur libgnome-2.20.0.orig/schemas/desktop_gnome_interface.schemas libgnome-2.20.0/schemas/desktop_gnome_interface.schemas
--- libgnome-2.20.0.orig/schemas/desktop_gnome_interface.schemas	2007-09-21 23:28:33.000000000 +0200
+++ libgnome-2.20.0/schemas/desktop_gnome_interface.schemas	2007-09-21 23:32:34.000000000 +0200
@@ -4134,7 +4134,7 @@
       <applyto>/desktop/gnome/interface/icon_theme</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>gnome</default>
+      <default>Mist</default>
       <locale name="C">
         <short>Icon Theme</short>
         <long>Icon theme to use for the panel, nautilus etc.</long>
@@ -5307,7 +5307,7 @@
       <applyto>/desktop/gnome/interface/font_name</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Sans 10</default>
+      <default>DejaVu Sans 13</default>
       <locale name="C">
 	<short>Default font</short>
 	<long>Name of the default font used by gtk+.</long>
@@ -6430,7 +6430,7 @@
       <key>/schemas/desktop/gnome/interface/document_font_name</key>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Sans 10</default>
+      <default>DejaVu Sans 13</default>
       <locale name="C">
 	<short>Document font</short>
 	<long>Name of the default font used for reading documents</long>
@@ -6761,7 +6761,7 @@
       <key>/schemas/desktop/gnome/interface/monospace_font_name</key>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Monospace 10</default>
+      <default>DejaVu Mono 13</default>
       <locale name="C">
 	<short>Monospace font</short>
 	<long>Name of a monospaced (fixed-width) font for use
diff -Naur libgnome-2.20.0.orig/schemas/desktop_gnome_interface.schemas.in libgnome-2.20.0/schemas/desktop_gnome_interface.schemas.in
--- libgnome-2.20.0.orig/schemas/desktop_gnome_interface.schemas.in	2007-09-21 23:28:33.000000000 +0200
+++ libgnome-2.20.0/schemas/desktop_gnome_interface.schemas.in	2007-09-21 23:33:21.000000000 +0200
@@ -130,7 +130,7 @@
       <applyto>/desktop/gnome/interface/icon_theme</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>gnome</default>
+      <default>Mist</default>
       <locale name="C">
         <short>Icon Theme</short>
         <long>Icon theme to use for the panel, nautilus etc.</long>
@@ -163,7 +163,7 @@
       <applyto>/desktop/gnome/interface/font_name</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Sans 10</default>
+      <default>DejaVu Sans 13</default>
       <locale name="C">
 	<short>Default font</short>
 	<long>Name of the default font used by gtk+.</long>
@@ -196,7 +196,7 @@
       <key>/schemas/desktop/gnome/interface/document_font_name</key>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Sans 10</default>
+      <default>DejaVu Sans 13</default>
       <locale name="C">
 	<short>Document font</short>
 	<long>Name of the default font used for reading documents</long>
@@ -207,7 +207,7 @@
       <key>/schemas/desktop/gnome/interface/monospace_font_name</key>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Monospace 10</default>
+      <default>DejaVu Mono 13</default>
       <locale name="C">
 	<short>Monospace font</short>
 	<long>Name of a monospaced (fixed-width) font for use
diff -Naur libgnome-2.20.0.orig/schemas/desktop_gnome_interface.schemas.in.in libgnome-2.20.0/schemas/desktop_gnome_interface.schemas.in.in
--- libgnome-2.20.0.orig/schemas/desktop_gnome_interface.schemas.in.in	2007-09-21 23:28:33.000000000 +0200
+++ libgnome-2.20.0/schemas/desktop_gnome_interface.schemas.in.in	2007-09-21 23:34:49.000000000 +0200
@@ -6,7 +6,7 @@
       <applyto>/desktop/gnome/interface/accessibility</applyto>
       <owner>gnome</owner>
       <type>bool</type>
-      <default>@A11Y_ENABLED@</default>
+      <default>false</default>
       <locale name="C">
         <short>Enable Accessibility</short>
         <long>Whether Applications should have accessibility support</long>
@@ -130,7 +130,7 @@
       <applyto>/desktop/gnome/interface/icon_theme</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>gnome</default>
+      <default>Mist</default>
       <locale name="C">
         <short>Icon Theme</short>
         <long>Icon theme to use for the panel, nautilus etc.</long>
@@ -141,7 +141,7 @@
       <applyto>/desktop/gnome/interface/gtk_theme</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Clearlooks</default>
+      <default>Frugalware</default>
       <locale name="C">
 	<short>Gtk+ Theme</short>
 	<long>Basename of the default theme used by gtk+.</long>
@@ -163,7 +163,7 @@
       <applyto>/desktop/gnome/interface/font_name</applyto>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Sans 10</default>
+      <default>DejaVu Sans 13</default>
       <locale name="C">
 	<short>Default font</short>
 	<long>Name of the default font used by gtk+.</long>
@@ -196,7 +196,7 @@
       <key>/schemas/desktop/gnome/interface/document_font_name</key>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Sans 10</default>
+      <default>DejaVu Sans 13</default>
       <locale name="C">
 	<short>Document font</short>
 	<long>Name of the default font used for reading documents</long>
@@ -207,7 +207,7 @@
       <key>/schemas/desktop/gnome/interface/monospace_font_name</key>
       <owner>gnome</owner>
       <type>string</type>
-      <default>Monospace 10</default>
+      <default>DejaVu Mono 13</default>
       <locale name="C">
 	<short>Monospace font</short>
 	<long>Name of a monospaced (fixed-width) font for use
diff -Naur libgnome-2.20.0.orig/schemas/desktop_gnome_sound.schemas libgnome-2.20.0/schemas/desktop_gnome_sound.schemas
--- libgnome-2.20.0.orig/schemas/desktop_gnome_sound.schemas	2007-09-21 23:28:33.000000000 +0200
+++ libgnome-2.20.0/schemas/desktop_gnome_sound.schemas	2007-09-21 23:35:08.000000000 +0200
@@ -504,7 +504,7 @@
       <applyto>/desktop/gnome/sound/enable_esd</applyto>
       <owner>gnome</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
         <short>Enable ESD</short>
         <long>Enable sound server startup.</long>
diff -Naur libgnome-2.20.0.orig/schemas/desktop_gnome_sound.schemas.in libgnome-2.20.0/schemas/desktop_gnome_sound.schemas.in
--- libgnome-2.20.0.orig/schemas/desktop_gnome_sound.schemas.in	2007-09-21 23:28:33.000000000 +0200
+++ libgnome-2.20.0/schemas/desktop_gnome_sound.schemas.in	2007-09-21 23:35:15.000000000 +0200
@@ -29,7 +29,7 @@
       <applyto>/desktop/gnome/sound/enable_esd</applyto>
       <owner>gnome</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
         <short>Enable ESD</short>
         <long>Enable sound server startup.</long>
