From 3a7f5fb4d0c5e632c439c7ae3e8189a64f77c853 Mon Sep 17 00:00:00 2001
From: Dagobert Michelsen <dam@opencsw.org>
Date: Tue, 29 Nov 2011 13:55:57 +0100
Subject: [PATCH] Add missing ppi.h from upstream GIT

---
 ppi.h |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)
 create mode 100644 ppi.h

diff --git a/ppi.h b/ppi.h
new file mode 100644
index 0000000..fa3359e
--- /dev/null
+++ b/ppi.h
@@ -0,0 +1,8 @@
+typedef struct ppi_header {
+	uint8_t		ppi_ver;
+	uint8_t		ppi_flags;
+	uint16_t	ppi_len;
+	uint32_t	ppi_dlt;
+} ppi_header_t;
+
+#define	PPI_HDRLEN	8
-- 
1.7.6.1
