diff -Nur wvstreams-4.2.2.orig/linuxstreams/wvinterface.cc wvstreams-4.2.2/linuxstreams/wvinterface.cc
--- wvstreams-4.2.2.orig/linuxstreams/wvinterface.cc	2005-11-24 21:46:47.000000000 +0200
+++ wvstreams-4.2.2/linuxstreams/wvinterface.cc	2006-02-03 21:19:32.635112750 +0200
@@ -21,6 +21,7 @@
 #include <errno.h>
 #include <linux/sockios.h>
 
+#include <linux/compiler.h>
 #define _LINUX_IF_H /* Hack to prevent loading linux/if.h */
 #include <linux/wireless.h>
 
