diff -Nru wvstreams-4.4.orig/configure.ac wvstreams-4.4/configure.ac
--- wvstreams-4.4.orig/configure.ac	2007-08-17 22:43:47.000000000 +0300
+++ wvstreams-4.4/configure.ac	2007-08-17 22:52:07.000000000 +0300
@@ -488,7 +488,6 @@
 
 # xplc
 if test "$with_xplc" != "no"; then
-    if test "$with_xplc" = ""; then
 	if pkg-config --modversion xplc-${xplc_version}; then
             WV_APPEND(CPPFLAGS, [`pkg-config --cflags xplc-${xplc_version}`])
 	    WV_APPEND(LDFLAGS, [`pkg-config --libs xplc-${xplc_version}`])
@@ -500,15 +499,6 @@
 	else
 	    with_xplc='no'
 	fi
-    else
-        WV_APPEND(CPPFLAGS, [-I$with_xplc/include])
-	WV_APPEND(LDFLAGS, [-L$with_xplc])
-	AC_CHECK_HEADERS(xplc/core.h,, [with_xplc=no], [#define UNSTABLE])
-	LIBS_save="$LIBS"
-	AC_CHECK_LIB(xplc, XPLC_getServiceManager,, [with_xplc=no])
-	LIBS="$LIBS_save"
-	xplc_pc="xplc-${xplc_version}"
-    fi
     # What if we can't find it?
     if test "$with_xplc" = "no" && test -d xplc; then
         WV_APPEND(CPPFLAGS, [-I\$(WVSTREAMS)/xplc/include])
@@ -516,7 +506,7 @@
         AC_CONFIG_SUBDIRS(xplc)
         build_xplc=yes
         with_xplc='$(WVSTREAMS)/xplc'
-	xplc_pc='wvxplc'
+        xplc_pc='wvxplc'
     fi
 fi
 
