diff -Nru wvstreams-4.4.orig/urlget/wvhttppool.cc wvstreams-4.4/urlget/wvhttppool.cc
--- wvstreams-4.4.orig/urlget/wvhttppool.cc	2007-07-20 00:22:36.000000000 +0300
+++ wvstreams-4.4/urlget/wvhttppool.cc	2007-08-17 23:07:41.000000000 +0300
@@ -43,7 +43,12 @@
     {
         WvBufUrlStream *x = new WvBufUrlStream;
         outstream = x;
-        x->death_notify = (WvStream **)&outstream;
+        union {
+            WvBufUrlStream** ppbufurlstream;
+            WvStream**       ppstream;
+        } s;
+        s.ppbufurlstream = &outstream;
+        x->death_notify = s.ppstream;
         x->url = url;
 
         putstream = content_source;
