diff -Nur wvstreams-4.2.2.orig/wvrules.mk wvstreams-4.2.2/wvrules.mk
--- wvstreams-4.2.2.orig/wvrules.mk	2005-11-24 21:46:51.000000000 +0200
+++ wvstreams-4.2.2/wvrules.mk	2006-02-03 21:26:59.439036250 +0200
@@ -296,7 +296,7 @@
 %.E: %.cc;	$(call wvcxx,$@,$<,$*,,-E)
 %.E: %.cpp;	$(call wvcxx,$@,$<,$*,,-E)
 
-%.moc: %.h;	moc -o $@ $<
+%.moc: %.h;	$(MOC) -o $@ $<
 
 %: %.o;		$(call wvlink,$@,$^) 
 %.t: %.t.o;	$(call wvlink,$@,$(call reverse,$(filter %.o,$^)) $(filter-out %.o,$^) $(LIBWVTEST))
