diff -Nru wvstreams-4.4.orig/linuxstreams/wvinterface.cc wvstreams-4.4/linuxstreams/wvinterface.cc
--- wvstreams-4.4.orig/linuxstreams/wvinterface.cc	2007-07-20 00:22:51.000000000 +0300
+++ wvstreams-4.4/linuxstreams/wvinterface.cc	2007-08-17 22:38:13.000000000 +0300
@@ -21,6 +21,7 @@
 #include <errno.h>
 #include <linux/sockios.h>
 
+#include <sys/sysctl.h> /* Needed for __user definition */
 #define _LINUX_IF_H /* Hack to prevent loading linux/if.h */
 #include <linux/wireless.h>
 
