diff -Nru wvstreams-4.4.orig/wvrules.mk wvstreams-4.4/wvrules.mk
--- wvstreams-4.4.orig/wvrules.mk	2007-08-17 22:45:07.000000000 +0300
+++ wvstreams-4.4/wvrules.mk	2007-08-17 22:53:05.000000000 +0300
@@ -294,7 +294,7 @@
 %.E: %.cc;	$(call wvcxx,$@,$<,$*,,-E)
 %.E: %.cpp;	$(call wvcxx,$@,$<,$*,,-E)
 
-%.moc: %.h;	moc -o $@ $<
+%.moc: %.h;	$(MOC) -o $@ $<
 
 %: %.o;		$(call wvlink,$@,$^) 
 %.t: %.t.o;	$(call wvlink,$@,$(call reverse,$(filter %.o,$^)) $(filter-out %.o,$^) $(LIBWVTEST))
