diff -Nur wvstreams-4.2.2.orig/configure.ac wvstreams-4.2.2/configure.ac
--- wvstreams-4.2.2.orig/configure.ac	2006-01-12 20:31:50.000000000 +0200
+++ wvstreams-4.2.2/configure.ac	2006-02-03 21:22:25.153894500 +0200
@@ -483,9 +483,9 @@
 # tcl
 if test "$with_tcl" != "no"; then
     CPPFLAGS_save="$CPPFLAGS"
-    WV_APPEND(CPPFLAGS, -I/usr/include/tcl8.3)
+    WV_APPEND(CPPFLAGS, -I/usr/include)
     AC_CHECK_HEADERS(tcl.h,, [with_tcl=no])
-    AC_CHECK_LIB(tcl8.3, TclInterpInit,, [with_tcl=no])
+    AC_CHECK_LIB(tcl8.4, TclInterpInit,, [with_tcl=no])
     if test "$with_tcl" != "no"; then
         CPPFLAGS="$CPPFLAGS_save"
     fi
diff -Nur wvstreams-4.2.2.orig/vars.mk wvstreams-4.2.2/vars.mk
--- wvstreams-4.2.2.orig/vars.mk	2005-11-24 21:46:51.000000000 +0200
+++ wvstreams-4.2.2/vars.mk	2006-02-03 21:22:25.153894500 +0200
@@ -234,3 +234,5 @@
 
 libwvgtk.a libwvgtk.so: $(call objects,gtk)
 libwvgtk.so: -lgtk -lgdk libwvstreams.so libwvutils.so
+
+libuniconf_tcl.so: bindings/uniconf_tcl.o -ltcl8.4 -luniconf
