diff -Nur wvstreams-4.2.2.orig/configure.ac wvstreams-4.2.2/configure.ac
--- wvstreams-4.2.2.orig/configure.ac	2006-01-12 20:31:50.000000000 +0200
+++ wvstreams-4.2.2/configure.ac	2006-02-03 21:24:41.546418500 +0200
@@ -542,7 +542,6 @@
 
 # xplc
 if test "$with_xplc" != "no"; then
-    if test "$with_xplc" = ""; then
 	if pkg-config --modversion xplc-${xplc_version}; then
             WV_APPEND(CPPFLAGS, [`pkg-config --cflags xplc-${xplc_version}`])
 	    WV_APPEND(LDFLAGS, [`pkg-config --libs xplc-${xplc_version}`])
@@ -554,15 +553,6 @@
 	else
 	    with_xplc='no'
 	fi
-    else
-        WV_APPEND(CPPFLAGS, [-I$with_xplc/include])
-	WV_APPEND(LDFLAGS, [-L$with_xplc])
-	AC_CHECK_HEADERS(xplc/core.h,, [with_xplc=no], [#define UNSTABLE])
-	LIBS_save="$LIBS"
-	AC_CHECK_LIB(xplc, XPLC_getServiceManager,, [with_xplc=no])
-	LIBS="$LIBS_save"
-	xplc_pc="xplc-${xplc_version}"
-    fi
     # What if we can't find it?
     if test "$with_xplc" = "no" && test -d xplc; then
         WV_APPEND(CPPFLAGS, [-I\$(WVSTREAMS)/xplc/include])
@@ -570,7 +560,7 @@
         AC_CONFIG_SUBDIRS(xplc)
         build_xplc=yes
         with_xplc='$(WVSTREAMS)/xplc'
-	xplc_pc='wvxplc'
+        xplc_pc='wvxplc'
     fi
 fi
 
