diff -Naur jahshaka/source/Jahshaka/JahLibraries/jahdatabase/jahdatabase.cpp jahshaka-p/source/Jahshaka/JahLibraries/jahdatabase/jahdatabase.cpp
--- jahshaka/source/Jahshaka/JahLibraries/jahdatabase/jahdatabase.cpp	2006-09-15 19:14:53.000000000 +0200
+++ jahshaka-p/source/Jahshaka/JahLibraries/jahdatabase/jahdatabase.cpp	2008-04-06 03:28:04.000000000 +0200
@@ -10,6 +10,7 @@
 
 #include "jahdatabase.h"
 #include <sqlite3.h>
+#include <cstdlib>
 
 static const unsigned char databaseVersionMajor		= 2;
 static const unsigned char databaseVersionMinor		= 1;
diff -Naur jahshaka/source/Jahshaka/JahLibraries/jahformatter/valueFormatters.cpp jahshaka-p/source/Jahshaka/JahLibraries/jahformatter/valueFormatters.cpp
--- jahshaka/source/Jahshaka/JahLibraries/jahformatter/valueFormatters.cpp	2006-05-01 12:52:19.000000000 +0200
+++ jahshaka-p/source/Jahshaka/JahLibraries/jahformatter/valueFormatters.cpp	2008-04-06 03:36:05.000000000 +0200
@@ -6,6 +6,8 @@
 
 // std
 #include <cmath>
+#include <cstdlib>
+#include <climits>
 
 TimecodeValueFormatter::TimecodeValueFormatter()
     : m_fps( 30 ),
diff -Naur jahshaka/source/Jahshaka/JahModules/editing_mlt/preview.h jahshaka-p/source/Jahshaka/JahModules/editing_mlt/preview.h
--- jahshaka/source/Jahshaka/JahModules/editing_mlt/preview.h	2006-09-15 19:14:56.000000000 +0200
+++ jahshaka-p/source/Jahshaka/JahModules/editing_mlt/preview.h	2008-04-06 04:19:41.000000000 +0200
@@ -12,6 +12,7 @@
 
 #include <qcombobox.h>
 #include <mlt++/Mlt.h>
+#include <stdlib.h>
 
 class QComboBox;
 
@@ -71,7 +72,7 @@
 				Mlt::Factory::init( );
 				Mlt::Factory::setenv( "MLT_NORMALISATION", pal ? "PAL" : "NTSC" );
 #else
-				setenv( "MLT_NORMALISATION", pal ? "PAL" : "NTSC", 1 );
+					setenv( "MLT_NORMALISATION", pal ? "PAL" : "NTSC", 1 );
 				Mlt::Factory::init( );
 #endif
 				init( );
diff -Naur jahshaka/source/Jahshaka/JahModules/editing_mlt/render.cpp jahshaka-p/source/Jahshaka/JahModules/editing_mlt/render.cpp
--- jahshaka/source/Jahshaka/JahModules/editing_mlt/render.cpp	2006-09-15 19:14:56.000000000 +0200
+++ jahshaka-p/source/Jahshaka/JahModules/editing_mlt/render.cpp	2008-04-06 03:51:47.000000000 +0200
@@ -22,6 +22,8 @@
 
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <cstring>
+#include <stdlib.h>
 
 // Local non-class functions
 
diff -Naur jahshaka/source/Jahshaka/JahWidgets/nodes/jahnodes.cpp jahshaka-p/source/Jahshaka/JahWidgets/nodes/jahnodes.cpp
--- jahshaka/source/Jahshaka/JahWidgets/nodes/jahnodes.cpp	2006-09-15 19:14:58.000000000 +0200
+++ jahshaka-p/source/Jahshaka/JahWidgets/nodes/jahnodes.cpp	2008-04-06 05:03:06.000000000 +0200
@@ -12,6 +12,7 @@
 //for preferences global
 #include "jahpreferences.h"
 #include "jahformatter.h"
+#include <cstdlib>
 
 static uint mainCount = 0;
 
diff -Naur jahshaka/source/OpenLibraries/openassetlib/v2_openassetlib/src/Database.cpp jahshaka-p/source/OpenLibraries/openassetlib/v2_openassetlib/src/Database.cpp
--- jahshaka/source/OpenLibraries/openassetlib/v2_openassetlib/src/Database.cpp	2006-09-15 19:14:59.000000000 +0200
+++ jahshaka-p/source/OpenLibraries/openassetlib/v2_openassetlib/src/Database.cpp	2008-04-06 03:01:06.000000000 +0200
@@ -7,6 +7,7 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <cstring>
 #include "sqlite3.h"
 #include "Database.h"
 #include "Asset.h"
diff -Naur jahshaka/source/OpenLibraries/opengpulib/glsl_objects.cpp jahshaka-p/source/OpenLibraries/opengpulib/glsl_objects.cpp
--- jahshaka/source/OpenLibraries/opengpulib/glsl_objects.cpp	2006-09-15 19:15:00.000000000 +0200
+++ jahshaka-p/source/OpenLibraries/opengpulib/glsl_objects.cpp	2008-04-06 03:17:59.000000000 +0200
@@ -10,6 +10,7 @@
 #include "glsl_objects.h"
 
 #include <fstream>
+#include <cstring>
 using namespace std;
 
 GlslProgram::GlslProgram(vector<GlslShader*>& shader_vector)
