diff -Naur exaile_0.2b.orig/mmkeys/Makefile exaile_0.2b/mmkeys/Makefile
--- exaile_0.2b.orig/mmkeys/Makefile	2006-06-09 18:40:47.015009880 +0200
+++ exaile_0.2b/mmkeys/Makefile	2006-06-09 18:41:16.136582728 +0200
@@ -2,7 +2,7 @@
 
 PYTHON_DIR = /usr/include/python2.4/
 
-CFLAGS = -O2 `pkg-config --cflags gtk+-2.0 pygtk-2.0` -I$(PYTHON_DIR)
+CFLAGS = -fPIC -O2 `pkg-config --cflags gtk+-2.0 pygtk-2.0` -I$(PYTHON_DIR)
 LDFLAGS = `pkg-config --libs gtk+-2.0 pygtk-2.0`
 
 mmkeys.so: mmkeyspy.o mmkeys.o mmkeysmodule.o
