diff -Naur Amaya/amaya/HTMLedit.c Amaya/amaya/HTMLedit.c.new
--- Amaya/amaya/HTMLedit.c    2009-12-09 11:11:22.000000000 +0100
+++ Amaya/amaya/HTMLedit.cnew    2010-05-20 19:15:22.000000000 +0200
@@ -1689,7 +1689,7 @@
                 {
 #ifdef _WX
                   wxString tmpFile = TtaConvMessageToWX(tempURL);
-                  wxFile::wxFile (tmpFile, wxFile::write);
+                  wxFile (tmpFile, wxFile::write);
 #endif /* _WX */
                 }
             }
diff -Naur Amaya/thotlib/base/platform.c Amaya/thotlib/base/platform.c.new
--- Amaya/thotlib/base/platform.c    2009-12-02 12:23:07.000000000 +0100
+++ Amaya/thotlib/base/platform.c.new    2010-05-20 19:00:49.000000000 +0200
@@ -329,9 +329,9 @@
     {
       tmp = (char *)TtaGetMemory (strlen(targetFileName)+10);
       sprintf (tmp, "%s.tmp", targetFileName);
-      wxFile::wxFile (targetFile, wxFile::write);
+      wxFile (targetFile, wxFile::write);
       tmpFile = TtaConvMessageToWX(tmp);
-      wxFile::wxFile (tmpFile, wxFile::write);
+      wxFile (tmpFile, wxFile::write);
       result = wxFile::Exists(targetFile);
       if (result)
         result = wxConcatFiles (tmpFile, sourceFile, targetFile);
