diff -Naur gzip-1.3.12.orig/gzip.c gzip-1.3.12/gzip.c
--- gzip-1.3.12.orig/gzip.c	2007-03-20 06:09:51.000000000 +0100
+++ gzip-1.3.12/gzip.c	2007-09-07 19:54:29.000000000 +0200
@@ -1637,7 +1637,7 @@
 	}
       }
 
-    if (futimens (ofd, ofname, timespec) != 0)
+    if (gzip_futimens (ofd, ofname, timespec) != 0)
       {
 	int e = errno;
 	WARN ((stderr, "%s: ", program_name));
diff -Naur gzip-1.3.12.orig/lib/utimens.c gzip-1.3.12/lib/utimens.c
--- gzip-1.3.12.orig/lib/utimens.c	2007-01-18 09:33:34.000000000 +0100
+++ gzip-1.3.12/lib/utimens.c	2007-09-07 19:54:15.000000000 +0200
@@ -75,7 +75,7 @@
    Return 0 on success, -1 (setting errno) on failure.  */
 
 int
-futimens (int fd ATTRIBUTE_UNUSED,
+gzip_futimens (int fd ATTRIBUTE_UNUSED,
 	  char const *file, struct timespec const timespec[2])
 {
   /* Some Linux-based NFS clients are buggy, and mishandle time stamps
@@ -185,5 +185,5 @@
 int
 utimens (char const *file, struct timespec const timespec[2])
 {
-  return futimens (-1, file, timespec);
+  return gzip_futimens (-1, file, timespec);
 }
diff -Naur gzip-1.3.12.orig/lib/utimens.h gzip-1.3.12/lib/utimens.h
--- gzip-1.3.12.orig/lib/utimens.h	2007-02-23 19:25:21.000000000 +0100
+++ gzip-1.3.12/lib/utimens.h	2007-09-07 19:54:06.000000000 +0200
@@ -1,3 +1,3 @@
 #include <time.h>
-int futimens (int, char const *, struct timespec const [2]);
+int gzip_futimens (int, char const *, struct timespec const [2]);
 int utimens (char const *, struct timespec const [2]);
