--- cups-filters-1.25.0/utils/cups-browsed.service	2019-06-06 15:17:11.000000000 +0200
+++ cups-filters-1.25.0.new/utils/cups-browsed.service	2019-06-24 09:48:59.940539899 +0200
@@ -1,11 +1,11 @@
 [Unit]
 Description=Make remote CUPS printers available locally
-Requires=cups.service
-After=cups.service avahi-daemon.service
+Requires=org.cups.cupsd.service
+After=org.cups.cupsd.service avahi-daemon.service
 Wants=avahi-daemon.service
 
 [Service]
-ExecStart=/usr/sbin/cups-browsed
+ExecStart=/usr/bin/cups-browsed
 
 [Install]
 WantedBy=multi-user.target
