diff -Naur avifile-0.7-0.7.45/drivers/libdha/kernelhelper/Makefile.am avifile-0.7-0.7.45-p/drivers/libdha/kernelhelper/Makefile.am
--- avifile-0.7-0.7.45/drivers/libdha/kernelhelper/Makefile.am	2002-08-25 20:29:33.000000000 +0200
+++ avifile-0.7-0.7.45-p/drivers/libdha/kernelhelper/Makefile.am	2006-08-25 23:14:21.000000000 +0200
@@ -1,7 +1,7 @@
 
 EXTRA_DIST = README dhahelper.c
 
-CFLAGS = -O2 -Wall $(LINUX_CFLAGS)
+AM_CFLAGS = -O2 -Wall $(LINUX_CFLAGS)
 
 KVERSION = $(shell $(SHELL) -c 'uname -r')
 moddir = /lib/modules/$(KVERSION)/misc
diff -Naur avifile-0.7-0.7.45/ffmpeg/libavcodec/i386/Makefile.am avifile-0.7-0.7.45-p/ffmpeg/libavcodec/i386/Makefile.am
--- avifile-0.7-0.7.45/ffmpeg/libavcodec/i386/Makefile.am	2005-08-28 22:57:53.000000000 +0200
+++ avifile-0.7-0.7.45-p/ffmpeg/libavcodec/i386/Makefile.am	2006-08-25 23:16:15.000000000 +0200
@@ -30,7 +30,7 @@
 
 # without -O2 compiler may complain about 'fixed or forbidden register spilled'
 #INSTRUMENT = -finstrument-functions
-CFLAGS = $(FFMPEG_CFLAGS) -O2 $(INSTRUMENT)
+AM_CFLAGS = $(FFMPEG_CFLAGS) -O2 $(INSTRUMENT)
 
 MAINTAINERCLEANFILES = Makefile.in
 #SUFFIXES = .s
diff -Naur avifile-0.7-0.7.45/ffmpeg/libavcodec/Makefile.am avifile-0.7-0.7.45-p/ffmpeg/libavcodec/Makefile.am
--- avifile-0.7-0.7.45/ffmpeg/libavcodec/Makefile.am	2006-03-06 10:10:15.000000000 +0100
+++ avifile-0.7-0.7.45-p/ffmpeg/libavcodec/Makefile.am	2006-08-25 23:15:40.000000000 +0200
@@ -236,7 +236,7 @@
 AM_CPPFLAGS = $(LTNOPIC) -DHAVE_AV_CONFIG_H -DHAVE_THREADS -I$(srcdir)/../libavutil -I$(srcdir)/..
 
 #INSTRUMENT = -finstrument-functions
-CFLAGS = $(FFMPEG_CFLAGS) $(INSTRUMENT)
+AM_CFLAGS = $(FFMPEG_CFLAGS) $(INSTRUMENT)
 #-fno-omit-frame-pointer -O0
 
 check_PROGRAMS = apiexample
diff -Naur avifile-0.7-0.7.45/ffmpeg/libavutil/Makefile.am avifile-0.7-0.7.45-p/ffmpeg/libavutil/Makefile.am
--- avifile-0.7-0.7.45/ffmpeg/libavutil/Makefile.am	2005-08-28 23:13:29.000000000 +0200
+++ avifile-0.7-0.7.45-p/ffmpeg/libavutil/Makefile.am	2006-08-25 23:17:12.000000000 +0200
@@ -32,7 +32,7 @@
 AM_CPPFLAGS = $(LTNOPIC) -DHAVE_AV_CONFIG_H -DHAVE_THREADS -I$(srcdir)/../libavcodec -I$(srcdir)/..
 
 #INSTRUMENT = -finstrument-functions
-CFLAGS = $(FFMPEG_CFLAGS) $(INSTRUMENT)
+AM_CFLAGS = $(FFMPEG_CFLAGS) $(INSTRUMENT)
 #-fno-omit-frame-pointer -O0
 
 CLEANFILES = .depend *.a *.la
diff -Naur avifile-0.7-0.7.45/plugins/libwin32/loader/dmo/Makefile.am avifile-0.7-0.7.45-p/plugins/libwin32/loader/dmo/Makefile.am
--- avifile-0.7-0.7.45/plugins/libwin32/loader/dmo/Makefile.am	2004-04-20 15:33:53.000000000 +0200
+++ avifile-0.7-0.7.45-p/plugins/libwin32/loader/dmo/Makefile.am	2006-08-25 23:18:30.000000000 +0200
@@ -14,6 +14,6 @@
  dmo_guids.c
 
 AM_CPPFLAGS = $(LTNOPIC) -I$(srcdir)/..
-CFLAGS = $(WIN32_CFLAGS)
+AM_CFLAGS = $(WIN32_CFLAGS)
 
 MAINTAINERCLEANFILES = Makefile.in
diff -Naur avifile-0.7-0.7.45/plugins/libwin32/loader/dshow/Makefile.am avifile-0.7-0.7.45-p/plugins/libwin32/loader/dshow/Makefile.am
--- avifile-0.7-0.7.45/plugins/libwin32/loader/dshow/Makefile.am	2004-04-20 15:33:58.000000000 +0200
+++ avifile-0.7-0.7.45-p/plugins/libwin32/loader/dshow/Makefile.am	2006-08-25 23:19:44.000000000 +0200
@@ -21,6 +21,6 @@
  outputpin.c
 
 AM_CPPFLAGS = $(LTNOPIC) -DAVIFILE -I$(srcdir)/..
-CFLAGS = $(WIN32_CFLAGS)
+AM_CFLAGS = $(WIN32_CFLAGS)
 
 MAINTAINERCLEANFILES = Makefile.in
diff -Naur avifile-0.7-0.7.45/plugins/libwin32/loader/Makefile.am avifile-0.7-0.7.45-p/plugins/libwin32/loader/Makefile.am
--- avifile-0.7-0.7.45/plugins/libwin32/loader/Makefile.am	2005-02-15 13:20:32.000000000 +0100
+++ avifile-0.7-0.7.45-p/plugins/libwin32/loader/Makefile.am	2006-08-25 23:18:01.000000000 +0200
@@ -52,7 +52,7 @@
 # -mno-omit-leaf-frame-pointer added to workaround bug with WMV8 and gcc-3.2
 AM_CFLAGS = -fno-omit-frame-pointer 
 #-mno-omit-leaf-frame-pointer
-CFLAGS = $(WIN32_CFLAGS)
+AM_CFLAGS = $(WIN32_CFLAGS)
 #-O2 -ffast-math
 #-fno-omit-frame-pointer
 #-O3 -fomit-frame-pointer
