--- avifile-0.7-0.7.45/plugins/libmp3lame_audioenc/mp3encoder.cpp	2006-03-05 21:45:26.000000000 +0100
+++ avifile-0.7-0.7.45-p/plugins/libmp3lame_audioenc/mp3encoder.cpp	2006-05-15 14:02:58.000000000 +0200
@@ -30,7 +30,7 @@
 	short  nCodecDelay;
     };
 public:
-    MP3Encoder::MP3Encoder(const CodecInfo& info, const WAVEFORMATEX* format)
+    MP3Encoder(const CodecInfo& info, const WAVEFORMATEX* format)
 	:IAudioEncoder(info)
     {
 	in_fmt=*format;
