diff -Naur avifile-0.7-0.7.45/configure.in avifile-0.7-0.7.45-p/configure.in
--- avifile-0.7-0.7.45/configure.in	2006-08-25 23:38:50.000000000 +0200
+++ avifile-0.7-0.7.45-p/configure.in	2006-08-26 00:11:01.000000000 +0200
@@ -92,7 +92,7 @@
 	    2*) GCCOPT=cpu ;;
 	    *) GCCOPT=tune ;;
 	esac
-        DEFAULT_FLAGS="-m$GCCOPT=$DEFAULT_FLAGS -march=$DEFAULT_FLAGS -ffast-math -fomit-frame-pointer"
+        DEFAULT_FLAGS="-ffast-math -fomit-frame-pointer"
     fi
     AC_DEFINE(ARCH_X86, 1, [Define if you want to build on x86 architecture.])
     AC_DEFINE(HAVE_MMX, 1, [Define if you want to build with mmx code.])
