diff -Naur avifile-0.7-0.7.45/acinclude.m4 avifile-0.7-0.7.45-p/acinclude.m4
--- avifile-0.7-0.7.45/acinclude.m4	2006-08-25 23:38:50.000000000 +0200
+++ avifile-0.7-0.7.45-p/acinclude.m4	2006-08-25 23:35:55.000000000 +0200
@@ -523,7 +523,7 @@
 	AC_DEFINE(HAVE_GPROF, 1, [Define if you want to build ffmpeg with gprof support.])
     fi
 
-    test -z "$FFMPEG_CFLAGS" && FFMPEG_CFLAGS="-O4 $DEFAULT_FLAGS"
+    test -z "$FFMPEG_CFLAGS" && FFMPEG_CFLAGS=" $DEFAULT_FLAGS"
     FFMPEG_CFLAGS="$FFMPEG_CFLAGS $DEFAULT_DEFINES"
 
     AC_CHECK_GNU_VECTOR(FFMPEG_SSE="-msse";
diff -Naur avifile-0.7-0.7.45/lib/common/Makefile.am avifile-0.7-0.7.45-p/lib/common/Makefile.am
--- avifile-0.7-0.7.45/lib/common/Makefile.am	2006-08-25 23:38:50.000000000 +0200
+++ avifile-0.7-0.7.45-p/lib/common/Makefile.am	2006-08-25 23:36:17.000000000 +0200
@@ -33,7 +33,7 @@
 check_PROGRAMS = testimage
 testimage_SOURCES = testimage.cpp
 testimage_LDFLAGS = -g ../libaviplay.la
-#testimage_CXXFLAGS = -g -O4
+#testimage_CXXFLAGS = -g 
 
 imtest.o: image.o mmx.o
 	$(CXXCOMPILE) -DIMTEST -o $@ -c $<
diff -Naur avifile-0.7-0.7.45/lib/common/Makefile.in avifile-0.7-0.7.45-p/lib/common/Makefile.in
--- avifile-0.7-0.7.45/lib/common/Makefile.in	2006-08-25 23:38:50.000000000 +0200
+++ avifile-0.7-0.7.45-p/lib/common/Makefile.in	2006-08-25 23:36:14.000000000 +0200
@@ -660,7 +660,7 @@
 	mostlyclean-generic mostlyclean-libtool pdf pdf-am ps ps-am \
 	tags uninstall uninstall-am uninstall-info-am
 
-#testimage_CXXFLAGS = -g -O4
+#testimage_CXXFLAGS = -g 
 
 imtest.o: image.o mmx.o
 	$(CXXCOMPILE) -DIMTEST -o $@ -c $<
diff -Naur avifile-0.7-0.7.45/m4/ffmpeg.m4 avifile-0.7-0.7.45-p/m4/ffmpeg.m4
--- avifile-0.7-0.7.45/m4/ffmpeg.m4	2006-08-25 23:38:50.000000000 +0200
+++ avifile-0.7-0.7.45-p/m4/ffmpeg.m4	2006-08-25 23:36:33.000000000 +0200
@@ -231,7 +231,7 @@
 	AC_DEFINE(HAVE_GPROF, 1, [Define if you want to build ffmpeg with gprof support.])
     fi
 
-    test -z "$FFMPEG_CFLAGS" && FFMPEG_CFLAGS="-O4 $DEFAULT_FLAGS"
+    test -z "$FFMPEG_CFLAGS" && FFMPEG_CFLAGS=" $DEFAULT_FLAGS"
     FFMPEG_CFLAGS="$FFMPEG_CFLAGS $DEFAULT_DEFINES"
 
     AC_CHECK_GNU_VECTOR(FFMPEG_SSE="-msse";
