--- colord-0.1.30/rules/69-cd-sensors.rules.in	2013-02-06 15:43:44.000000000 +0100
+++ colord-0.1.30.new/rules/69-cd-sensors.rules.in	2013-02-17 22:16:17.000000000 +0100
@@ -83,7 +83,6 @@
 # color calibration device
 ENV{COLORD_SENSOR_KIND}=="*?", ENV{COLOR_MEASUREMENT_DEVICE}="1"
 ENV{COLORD_SENSOR_KIND}=="*?", ENV{ID_MODEL}=="", IMPORT{program}="usb_id --export %p"
-ENV{COLORD_SENSOR_KIND}=="*?", ENV{ID_MODEL_FROM_DATABASE}=="", IMPORT{program}="usb-db %p"
 
 # Allow the daemon to access the color devices
 ENV{COLORD_SENSOR_KIND}=="*?", GROUP="@daemon_user@"
