From fb039f5553cdc009d2855ba698ea9d0672334727 Mon Sep 17 00:00:00 2001
From: Gabriel Craciunescu <nix.or.die@gmail.com>
Date: Mon, 5 Nov 2018 03:11:06 +0100
Subject: [PATCH] pacman fixes

---
 src/modules/packages/main.py | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/src/modules/packages/main.py b/src/modules/packages/main.py
index bffd6a945..bd126629f 100644
--- a/src/modules/packages/main.py
+++ b/src/modules/packages/main.py
@@ -264,16 +264,17 @@ class PMPacman(PackageManager):
             pacman_flags = "-S"
 
         check_target_env_call(["pacman", pacman_flags,
-                               "--noconfirm"] + pkgs)
+                               "--noconfirm", "--noprogressbar", "--noarch"] + pkgs)
 
     def remove(self, pkgs):
-        check_target_env_call(["pacman", "-Rs", "--noconfirm"] + pkgs)
+        check_target_env_call(["pacman", "-Rs", "--noconfirm", "--noprogressbar", "--noarch"] + pkgs)
 
     def update_db(self):
         check_target_env_call(["pacman", "-Sy"])
 
     def update_system(self):
-        check_target_env_call(["pacman", "-Su"])
+        # don't Su but Syu , really .. Anyway that should not be here no need
+        check_target_env_call(["pacman", "-Syu", "--noconfirm", "--noprogressbar"])
 
 
 class PMPortage(PackageManager):
-- 
2.19.1

