diff -Naur calamares-3.2.18/src/modules/partition/jobs/FillGlobalStorageJob.cpp calamares-3.2.18-p/src/modules/partition/jobs/FillGlobalStorageJob.cpp
--- calamares-3.2.18/src/modules/partition/jobs/FillGlobalStorageJob.cpp	2020-01-28 17:42:21.000000000 +0100
+++ calamares-3.2.18-p/src/modules/partition/jobs/FillGlobalStorageJob.cpp	2020-02-13 03:31:31.121174177 +0100
@@ -89,7 +89,7 @@
     map[ "fs" ] = partition->fileSystem().name( { QStringLiteral("C") } );  // Untranslated
     if ( partition->fileSystem().type() == FileSystem::Luks &&
          dynamic_cast< FS::luks& >( partition->fileSystem() ).innerFS() )
-        map[ "fs" ] = dynamic_cast< FS::luks& >( partition->fileSystem() ).innerFS()->name();
+        map[ "fs" ] = dynamic_cast< FS::luks& >( partition->fileSystem() ).innerFS()->name( { QStringLiteral("C") } );
     map[ "uuid" ] = uuid;
 
     // Debugging for inside the loop in createPartitionList(),
