diff -Naur calamares-3.2.7/src/modules/welcome/checker/partman_devices.c calamares-3.2.7-p/src/modules/welcome/checker/partman_devices.c
--- calamares-3.2.7/src/modules/welcome/checker/partman_devices.c	2019-04-27 20:12:03.000000000 +0200
+++ calamares-3.2.7-p/src/modules/welcome/checker/partman_devices.c	2019-05-03 18:39:38.789723224 +0200
@@ -120,7 +120,7 @@
     for (dev = NULL; NULL != (dev = ped_device_get_next(dev));)
     {
         long long dev_size = process_device(dev);
-        if (dev_size > required_space)
+        if (dev_size >= required_space)
         {
             big_enough = true;
             break;
