--- rsyslog-8.2502.0/configure.ac	2025-02-16 14:10:33.000000000 +0100
+++ rsyslog-8.2502.0.new/configure.ac	2025-04-24 15:21:54.136999397 +0200
@@ -1217,19 +1217,15 @@
         [enable_libgcrypt=yes]
 )
 if test "x$enable_libgcrypt" = "xyes"; then
-	AC_PATH_PROG([LIBGCRYPT_CONFIG],[libgcrypt-config],[no])
-        if test "x${LIBGCRYPT_CONFIG}" = "xno"; then
-           AC_MSG_FAILURE([libgcrypt-config not found in PATH])
-        fi
         AC_CHECK_LIB(
-		[gcrypt],
-        	[gcry_cipher_open],
-        	[LIBGCRYPT_CFLAGS="`${LIBGCRYPT_CONFIG} --cflags`"
-        	LIBGCRYPT_LIBS="`${LIBGCRYPT_CONFIG} --libs`"
-        	],
-        	[AC_MSG_FAILURE([libgcrypt is missing])],
-        	[`${LIBGCRYPT_CONFIG} --libs --cflags`]
-        	)
+               [gcrypt],
+               [gcry_cipher_open],
+               [LIBGCRYPT_CFLAGS="`pkg-config --cflags libgcrypt gpg-error`"
+               LIBGCRYPT_LIBS="`pkg-config --libs libgcrypt gpg-error`"
+               ],
+               [AC_MSG_FAILURE([libgcrypt is missing])],
+               [`pkg-config --libs --cflags libgcrypt gpg-error`]
+               )
 	AC_DEFINE([ENABLE_LIBGCRYPT], [1], [Indicator that LIBGCRYPT is present])
 fi
 AM_CONDITIONAL(ENABLE_LIBGCRYPT, test x$enable_libgcrypt = xyes)
