diff -Naur nessus-core.orig/nessusd/nessus-check-signature.c nessus-core/nessusd/nessus-check-signature.c
--- nessus-core.orig/nessusd/nessus-check-signature.c	2005-09-06 00:43:06.000000000 +0200
+++ nessus-core/nessusd/nessus-check-signature.c	2005-09-06 00:46:29.000000000 +0200
@@ -36,7 +36,7 @@
 int generate_signature(char * filename)
 {
  RSA * rsa = NULL;
- FILE * fp = fopen(NESSUSD_STATEDIR "/nessus_org.priv.pem", "r");
+ FILE * fp = fopen(NESSUSD_CONFDIR "/nessus_org.priv.pem", "r");
  unsigned char  * result;
  unsigned int len;
  int i;
@@ -116,7 +116,7 @@
 {
  unsigned char md[SHA_DIGEST_LENGTH+1];
  RSA * rsa = NULL;
- FILE * fp = fopen(NESSUSD_STATEDIR "/nessus_org.pem", "r");
+ FILE * fp = fopen(NESSUSD_CONFDIR "/nessus_org.pem", "r");
 
  char sig[16384];
  unsigned char bin_sig[8192];
@@ -134,7 +134,7 @@
 
  if ( fp == NULL )
  {
-  fprintf(stderr, "Open %s/nessus_org.pem : %s\n", NESSUSD_STATEDIR, strerror(errno));
+  fprintf(stderr, "Open %s/nessus_org.pem : %s\n", NESSUSD_CONFDIR, strerror(errno));
   return -1;
  }
 
diff -Naur nessus-core.orig/nessus-mkcert-client.in nessus-core/nessus-mkcert-client.in
--- nessus-core.orig/nessus-mkcert-client.in	2005-09-06 00:43:06.000000000 +0200
+++ nessus-core/nessus-mkcert-client.in	2005-09-06 00:45:26.000000000 +0200
@@ -84,8 +84,8 @@
 
 
 
-NESSUSPRIV="$localstatedir/nessus/CA"
-NESSUSPUB="$sharedstatedir/nessus/CA"
+NESSUSPRIV="$sysconfdir/nessus/CA"
+NESSUSPUB="$sysconfdir/nessus/ssl"
 
 while [ ! -d "$NESSUSPRIV" ]; do
     echo_no_nl "Nessus server 'private' directory: "
diff -Naur nessus-core.orig/nessus-mkcert.in nessus-core/nessus-mkcert.in
--- nessus-core.orig/nessus-mkcert.in	2005-09-06 00:43:06.000000000 +0200
+++ nessus-core/nessus-mkcert.in	2005-09-06 00:44:25.000000000 +0200
@@ -92,8 +92,8 @@
    exit 1
 esac
      
-NESSUSPRIV="$localstatedir/nessus/CA"
-NESSUSPUB="$sharedstatedir/nessus/CA"
+NESSUSPRIV="$sysconfdir/nessus/CA"
+NESSUSPUB="$sysconfdir/nessus/ssl"
 if [ ! -d "$NESSUSPRIV" ]; then
     mkdir -p "$NESSUSPRIV"
     chmod 0700 "$NESSUSPRIV"
