diff -Naur gdhcpd-0.2.9.orig/desktop/net-gdhcpd.desktop gdhcpd-0.2.9/desktop/net-gdhcpd.desktop
--- gdhcpd-0.2.9.orig/desktop/net-gdhcpd.desktop	2006-02-07 01:05:15.000000000 +0100
+++ gdhcpd-0.2.9/desktop/net-gdhcpd.desktop	2006-03-05 23:09:09.000000000 +0100
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=GDHCPD
+Name[hu]=GDHCPD
 Name[de]=GDHCPD
 Name[es]=GDHCPD
 Name[fi]=GDHCPD
@@ -15,8 +16,9 @@
 Name[zh_TW]=GDHCPD
 GenericName=GDHCPD
 Comment=DHCPD DHCP server administration
-Exec=gdhcpd
+Comment=Grafikus DHCPD adminisztrátor
+Exec=gksu /usr/sbin/gdhcpd
 Icon=gdhcpd.png
 Terminal=false
 Type=Application
-Categories=Application;Network;X-Red-Hat-ServerConfig;X-Red-Hat-Base;
+Categories=Application;Network;
