diff -Naurd gdhcpd-0.2.8.orig/desktop/net-gdhcpd.desktop gdhcpd-0.2.8/desktop/net-gdhcpd.desktop
--- gdhcpd-0.2.8.orig/desktop/net-gdhcpd.desktop	2004-12-12 20:48:19.000000000 +0100
+++ gdhcpd-0.2.8/desktop/net-gdhcpd.desktop	2005-08-14 17:17:41.000000000 +0200
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=GDHCPD
+Name[hu]=GDHCPD
 Name[de]=GDHCPD
 Name[es]=GDHCPD
 Name[fi]=GDHCPD
@@ -15,8 +16,9 @@
 Name[zh_TW]=GDHCPD
 GenericName=GDHCPD
 Comment=Graphical DHCPD administration
-Exec=gdhcpd
+Comment=Grafikus DHCPD adminisztrátor
+Exec=gksu /usr/sbin/gdhcpd
 Icon=gdhcpd.png
 Terminal=false
 Type=Application
-Categories=Application;Network;X-Red-Hat-ServerConfig;X-Red-Hat-Base;
+Categories=Application;Network;
