diff -Naur XML-Twig-3.26/Makefile.PL XML-Twig-3.26-do_not_ask_build_it/Makefile.PL
--- XML-Twig-3.26/Makefile.PL	2006-06-21 16:02:49.000000000 +0200
+++ XML-Twig-3.26-do_not_ask_build_it/Makefile.PL	2006-07-20 20:58:01.000000000 +0200
@@ -13,10 +13,7 @@
 xml_spellcheck|y|spellcheck XML files skipping tags
 QUERIES
     my ($program, $default, $description) = split /\|/, $_;
-    if (prompt("Do you want to install '$program' ($description)?", 
-               $default) =~ /^y/) {
 	push(@programs, $program);
-    }
 }
 
 WriteMakefile(
