--- trunk/gobject/gtypemodule.h	2006/12/29 06:12:11	5179
+++ trunk/gobject/gtypemodule.h	2007/09/20 10:56:34	5777
@@ -86,6 +86,7 @@
 static void \
 type_name##_register_type (GTypeModule *type_module) \
 { \
+  GType g_define_type_id; \
   const GTypeInfo g_define_type_info = { \
     sizeof (TypeName##Class), \
     (GBaseInitFunc) NULL, \
@@ -103,6 +104,7 @@
 						     #TypeName, \
 						     &g_define_type_info, \
 						     (GTypeFlags) flags); \
+  g_define_type_id = type_name##_type_id; \
   { CODE ; } \
 }
