diff -Naur glib-2.14.1/glib/gmain.c glib-2.14.1-p/glib/gmain.c
--- glib-2.14.1/glib/gmain.c	2007-09-16 19:32:03.000000000 +0200
+++ glib-2.14.1-p/glib/gmain.c	2007-09-27 19:29:16.000000000 +0200
@@ -665,6 +665,11 @@
   else
     main_contexts_without_pipe = g_slist_remove (main_contexts_without_pipe, 
 						 context);
+
+  if(context->cond != NULL)
+  {
+    g_cond_free(context->cond);
+  }
 #endif
   
   g_free (context);
