diff -ur a/modules/json/jsonparser.c b/modules/json/jsonparser.c
--- a/modules/json/jsonparser.c	2014-03-12 08:12:11.000000000 -0500
+++ b/modules/json/jsonparser.c	2014-06-24 13:47:46.971821023 -0500
@@ -187,7 +187,7 @@
   if (tok->err != json_tokener_success)
     {
       msg_error ("Unparsable JSON stream encountered",
-                 evt_tag_str ("error", json_tokener_errors[tok->err]), NULL);
+                 evt_tag_str ("error", json_tokener_error_desc(tok->err)), NULL);
       json_tokener_free (tok);
       return FALSE;
     }
