diff -Naur lynx2-8-8/lynx.cfg lynx2-8-8-p/lynx.cfg
--- lynx2-8-8/lynx.cfg	2014-03-09 22:43:10.000000000 +0100
+++ lynx2-8-8-p/lynx.cfg	2017-08-13 15:23:03.210732563 +0200
@@ -3334,11 +3334,11 @@
 
 .h2 CONNECT_TIMEOUT
 # Specifies (in seconds) connect timeout. Default value is rather huge.
-#CONNECT_TIMEOUT:18000
+CONNECT_TIMEOUT:280
 
 .h2 READ_TIMEOUT
 # Specifies (in seconds) read-timeout. Default value is rather huge.
-#READ_TIMEOUT:18000
+READ_TIMEOUT:280
 
 .h1 Internal Behavior
 # These settings control internal lynx behavior - the way it interacts with the
