--- hslua-packaging-2.3.0/hslua-packaging.cabal	2001-09-09 03:46:40.000000000 +0200
+++ hslua-packaging-2.3.0.new/hslua-packaging.cabal	2023-12-10 22:29:38.811651090 +0100
@@ -39,7 +39,7 @@
                      , hslua-objectorientation >= 2.3    && < 2.4
                      , hslua-typing            >= 0.1    && < 0.2
                      , mtl                     >= 2.2    && < 2.4
-                     , text                    >= 1.2    && < 2.1
+                     , text                    >= 1.2    && < 2.2
   ghc-options:         -Wall
                        -Wincomplete-record-updates
                        -Wnoncanonical-monad-instances
