diff -up lincity-ng-1.91.beta/src/lincity/modules/market.cpp.BAD lincity-ng-1.91.beta/src/lincity/modules/market.cpp
--- lincity-ng-1.91.beta/src/lincity/modules/market.cpp.BAD	2008-02-19 15:54:52.000000000 -0500
+++ lincity-ng-1.91.beta/src/lincity/modules/market.cpp	2008-02-19 15:55:04.000000000 -0500
@@ -6,6 +6,7 @@
 #include "modules.h"
 #include "market.h"
 #include "lincity-ng/ErrorInterface.hpp"
+#include <cstdlib>
 
 int get_jobs(int x, int y, int jobs)
 {
