diff -Naur telepathy-qt-master/CMakeLists.txt telepathy-qt-master-glib-fix/CMakeLists.txt
--- telepathy-qt-master/CMakeLists.txt	2016-03-02 13:51:39.000000000 +0100
+++ telepathy-qt-master-glib-fix/CMakeLists.txt	2016-03-31 18:52:08.503745226 +0200
@@ -117,6 +117,7 @@
 # set the flags (-fPIE) here so that CHECK_CXX_SOURCE_COMPILES works properly if qt5 was built with
 # -reduce-locations
 set(CMAKE_REQUIRED_FLAGS "${TP_QT_EXECUTABLE_LINKER_FLAGS}")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -DQT_NO_VERSION_TAGGING=1")
 
 CHECK_CXX_SOURCE_COMPILES("
 #include <QtCore/QtGlobal>
