--- dayplanner-0.9.2old/devel-tools/GenDesktop	2008-08-04 17:27:27.000000000 +0200
+++ dayplanner-0.9.2/devel-tools/GenDesktop	2008-08-14 23:03:59.000000000 +0200
@@ -111,13 +111,9 @@
 print $Desktop "Version=1.0\n";
 print $Desktop "Encoding=UTF-8\n";
 print $Desktop "MimeType=text/calendar\n";
-print $Desktop "Categories=X-MandrivaLinux-Office-TimeManagement;Office;Calendar;GTK;GNOME;\n";
-if($BinDir eq '.') {
-	print $Desktop "Exec=dayplanner\n";
-} else {
-	print $Desktop "Exec=$BinDir/dayplanner\n";
-}
-print $Desktop "Icon=$Icon\n";
+print $Desktop "Categories=Office;Calendar;GTK;GNOME;\n";
+print $Desktop "Exec=dayplanner\n";
+print $Desktop "Icon=/usr/share/dayplanner/art/dayplanner-48x48.png\n";
 print $Desktop "Name=Day Planner\n";
 print $Desktop "StartupNotify=false\n";
 print $Desktop "Terminal=false\n";
