--- dayplanner-0.10old/devel-tools/GenDesktop	2009-03-25 19:09:21.000000000 +0100
+++ dayplanner-0.10/devel-tools/GenDesktop	2009-04-03 22:00:51.000000000 +0200
@@ -111,12 +111,8 @@
 print $Desktop "Encoding=UTF-8\n";
 print $Desktop "MimeType=text/calendar\n";
 print $Desktop "Categories=Office;Calendar;GTK;GNOME;\n";
-if($BinDir eq '.') {
-	print $Desktop "Exec=dayplanner\n";
-} else {
-	print $Desktop "Exec=$BinDir/dayplanner\n";
-}
-print $Desktop "Icon=$Icon\n";
+print $Desktop "Exec=dayplanner\n";
+print $Desktop "Icon=/usr/share/dayplanner/art/dayplanner-48x48.png\n";
 print $Desktop "Name=Day Planner\n";
 print $Desktop "StartupNotify=false\n";
 print $Desktop "Terminal=false\n";
