diff -urN libgksu-2.0.4.orig/gksu-properties/gksu-properties.desktop.in libgksu-2.0.4/gksu-properties/gksu-properties.desktop.in
--- libgksu-2.0.4.orig/gksu-properties/gksu-properties.desktop.in	2007-04-15 10:22:55.000000000 +0100
+++ libgksu-2.0.4/gksu-properties/gksu-properties.desktop.in	2007-04-15 10:27:59.000000000 +0100
@@ -7,5 +7,5 @@
 StartupNotify=true
 Terminal=false
 Type=Application
-Categories=GNOME;GTK;Application;Settings;AdvancedSettings;
+Categories=GNOME;GTK;Application;Settings;AdvancedSettings;X-GNOME-PersonalSettings;
 OnlyShowIn=GNOME;XFCE;
