--- ./files.h.orig	2003-02-22 16:19:28.000000000 -0800
+++ ./files.h	2003-02-22 16:20:27.000000000 -0800
@@ -18,9 +18,9 @@
 #include <utmp.h>
 
 #define WTMP_FILE_LOC "/var/log/wtmp"
-#define ACCT_FILE_LOC "/var/account/pacct"
-#define SAVACCT_FILE_LOC "/var/account/savacct"
-#define USRACCT_FILE_LOC "/var/account/usracct"
+#define ACCT_FILE_LOC "/var/log/pacct"
+#define SAVACCT_FILE_LOC "/var/log/savacct"
+#define USRACCT_FILE_LOC "/var/log/usracct"
 
 #ifndef AHZ
 #define AHZ 64
