diff -ur a/interfaces/C/Makefile.am b/interfaces/C/Makefile.am
--- a/interfaces/C/Makefile.am	2012-06-28 05:50:33.000000000 -0500
+++ b/interfaces/C/Makefile.am	2012-09-13 03:21:49.203212613 -0500
@@ -73,7 +73,7 @@
 #	    0.10.2		    3:0:1
 #	    0.11		    4:0:0
 
-LIBPPL_C_LT_CURRENT =  4
+LIBPPL_C_LT_CURRENT =  99
 LIBPPL_C_LT_REVISION = 0
 LIBPPL_C_LT_AGE =      0
 
diff -ur a/interfaces/C/Makefile.in b/interfaces/C/Makefile.in
--- a/interfaces/C/Makefile.in	2012-06-28 05:50:40.000000000 -0500
+++ b/interfaces/C/Makefile.in	2012-09-13 04:14:21.918094213 -0500
@@ -462,7 +462,7 @@
 #	    0.10.1		    2:0:0 (was not updated by mistake)
 #	    0.10.2		    3:0:1
 #	    0.11		    4:0:0
-LIBPPL_C_LT_CURRENT = 4
+LIBPPL_C_LT_CURRENT = 99
 LIBPPL_C_LT_REVISION = 0
 LIBPPL_C_LT_AGE = 0
 BUILT_SOURCES = \
