diff -urN system-tools-backends-2.1.1.orig/Utils/Platform.pm system-tools-backends-2.1.1/Utils/Platform.pm
--- system-tools-backends-2.1.1.orig/Utils/Platform.pm	2006-12-07 17:25:10.000000000 +0000
+++ system-tools-backends-2.1.1/Utils/Platform.pm	2006-12-07 17:26:39.000000000 +0000
@@ -70,6 +70,7 @@
   "slackware-10.0.0" => [ "Slackware", "10.0.0", "" ],
   "slackware-10.1.0" => [ "Slackware", "10.1.0", "" ],
   "slackware-10.2.0" => [ "Slackware", "10.2.0", "" ],
+  "frugalware"      => [ "Frugalware Linux", "", "" ],
   "freebsd-4"       => [ "FreeBSD", "4", "" ],
   "freebsd-5"       => [ "FreeBSD", "5", "" ],
   "freebsd-6"       => [ "FreeBSD", "6", "" ],
@@ -136,6 +137,7 @@
      "slackware-10.0.0" => "slackware-9.1.0",
      "slackware-10.1.0" => "slackware-9.1.0",
      "slackware-10.2.0" => "slackware-9.1.0",
+     "frugalware"       => "slackware-9.1.0",
      "suse-9.1"         => "suse-9.0",
      "ubuntu-5.04"      => "debian-3.0",
      "ubuntu-5.10"      => "debian-3.0",
@@ -369,6 +371,7 @@
                   [ \&check_distro_file, "/etc/slackware-version", "slackware", "^Slackware (\\S+)" ],
                   [ \&check_distro_file, "/etc/vlos-release", "vlos", "^VLOS.*\\s+(\\S+)" ],
                   [ \&check_file_exists, "/usr/portage", "gentoo" ],
+                  [ \&check_file_exists, "/etc/frugalware-release", "frugalware" ],
                   [ \&check_distro_file, "/etc/pld-release", "pld", "^(\\S+) PLD Linux" ],
                   [ \&check_rpath ],
                   [ \&check_file_exists, "/etc/arch-release", "archlinux" ],
