diff -Naur system-tools-backends-2.4.0/Utils/Platform.pm system-tools-backends-2.4.0-p/Utils/Platform.pm
--- system-tools-backends-2.4.0/Utils/Platform.pm	2007-09-12 00:43:37.000000000 +0200
+++ system-tools-backends-2.4.0-p/Utils/Platform.pm	2007-09-26 03:02:42.000000000 +0200
@@ -74,6 +74,7 @@
   "slackware-10.2.0" => [ "Slackware", "10.2.0", "" ],
   "slackware-11.0.0" => [ "Slackware", "11.0.0", "" ],
   "slackware-12.0.0" => [ "Slackware", "12.0.0", "" ],
+  "frugalware"      => [ "Frugalware Linux", "", "" ],
   "bluewhite64-12.0.0" => [ "Bluewhite64", "12.0.0", "" ],
   "freebsd-4"       => [ "FreeBSD", "4", "" ],
   "freebsd-5"       => [ "FreeBSD", "5", "" ],
@@ -144,6 +145,7 @@
      "slackware-10.2.0" => "slackware-9.1.0",
      "slackware-11.0.0" => "slackware-9.1.0",
      "slackware-12.0.0" => "slackware-9.1.0",
+     "frugalware"       => "slackware-9.1.0",
      "bluewhite64-12.0.0" => "slackware-9.1.0",
      "suse-9.1"         => "suse-9.0",
      "ubuntu-5.04"      => "debian-3.0",
@@ -381,6 +383,7 @@
                   [ \&check_distro_file, "/etc/bluewhite64-version", "bluewhite64", "^Bluewhite64 (\\S+)" ],
                   [ \&check_distro_file, "/etc/vlos-release", "vlos", "^VLOS.*\\s+(\\S+)" ],
                   [ \&check_file_exists, "/usr/portage", "gentoo" ],
+		  [ \&check_file_exists, "/etc/frugalware-release", "frugalware" ],
                   [ \&check_distro_file, "/etc/pld-release", "pld", "^(\\S+) PLD Linux" ],
                   [ \&check_rpath ],
                   [ \&check_file_exists, "/etc/arch-release", "archlinux" ],
