diff -urN system-tools-backends-1.9.3/Utils/Platform.pm system-tools-backends-1.9.3.new/Utils/Platform.pm
--- system-tools-backends-1.9.3/Utils/Platform.pm	2006-08-12 17:01:31.000000000 +0100
+++ system-tools-backends-1.9.3.new/Utils/Platform.pm	2006-08-15 08:19:01.000000000 +0100
@@ -137,6 +137,7 @@
      "slackware-10.0.0" => "slackware-9.1.0",
      "slackware-10.1.0" => "slackware-9.1.0",
      "slackware-10.2.0" => "slackware-9.1.0",
+     "frugalware"	=> "slackware-9.1.0",
      "suse-9.1"         => "suse-9.0",
      "ubuntu-5.04"      => "debian-3.0",
      "ubuntu-5.10"      => "debian-3.0",
