diff -Nurp koffice-1.6.0/filters/krita/jpeg/krita_jpeg.desktop koffice-1.6.0-p/filters/krita/jpeg/krita_jpeg.desktop
--- koffice-1.6.0/filters/krita/jpeg/krita_jpeg.desktop	2006-10-07 15:33:33.000000000 +0200
+++ koffice-1.6.0-p/filters/krita/jpeg/krita_jpeg.desktop	2006-10-23 02:34:13.000000000 +0200
@@ -1,5 +1,5 @@
 [Desktop Entry]
-Categories=
+NoDisplay=true
 Encoding=UTF-8
 Exec=krita %u
 GenericName=Application for Drawing and Handling of Images
diff -Nurp koffice-1.6.0/karbon/data/karbon.desktop koffice-1.6.0-p/karbon/data/karbon.desktop
--- koffice-1.6.0/karbon/data/karbon.desktop	2006-10-07 15:38:10.000000000 +0200
+++ koffice-1.6.0-p/karbon/data/karbon.desktop	2006-10-23 02:36:28.000000000 +0200
@@ -63,4 +63,4 @@ InitialPreference=5
 Type=Application
 Icon=karbon
 X-KDE-NativeMimeType=application/x-karbon
-Categories=Qt;KDE;Office;X-KDE-More;
+Categories=Qt;KDE;Office;
diff -Nurp koffice-1.6.0/kchart/kchart.desktop koffice-1.6.0-p/kchart/kchart.desktop
--- koffice-1.6.0/kchart/kchart.desktop	2006-10-07 15:35:49.000000000 +0200
+++ koffice-1.6.0-p/kchart/kchart.desktop	2006-10-23 02:37:42.000000000 +0200
@@ -95,4 +95,4 @@ X-KDE-NativeMimeType=application/vnd.oas
 X-KDE-NativeOasisMimeType=application/vnd.oasis.opendocument.chart
 X-KDE-StartupNotify=true
 X-DCOP-ServiceType=Multi
-Categories=Qt;KDE;Office;X-KDE-More;
+Categories=Qt;KDE;Office;
diff -Nurp koffice-1.6.0/kformula/kformula.desktop koffice-1.6.0-p/kformula/kformula.desktop
--- koffice-1.6.0/kformula/kformula.desktop	2006-10-07 15:27:55.000000000 +0200
+++ koffice-1.6.0-p/kformula/kformula.desktop	2006-10-23 02:39:29.000000000 +0200
@@ -75,4 +75,4 @@ Icon=kformula
 X-KDE-NativeMimeType=application/vnd.oasis.opendocument.formula
 X-DCOP-ServiceType=Multi
 DocPath=kformula/index.html
-Categories=Qt;KDE;Office;X-KDE-More;
+Categories=Qt;KDE;Office;
diff -Nurp koffice-1.6.0/krita/krita.desktop koffice-1.6.0-p/krita/krita.desktop
--- koffice-1.6.0/krita/krita.desktop	2006-10-07 15:35:38.000000000 +0200
+++ koffice-1.6.0-p/krita/krita.desktop	2006-10-23 02:34:54.000000000 +0200
@@ -69,7 +69,7 @@ GenericName[zh_TW]=繪圖與影像編輯
 MimeType=application/x-krita
 Type=Application
 Icon=krita
-Categories=Qt;KDE;Office;Graphics;
+Categories=Qt;KDE;Graphics;
 X-KDE-NativeMimeType=application/x-krita
 X-KDE-StartupNotify=true
 X-DCOP-ServiceType=Multi
diff -Nurp koffice-1.6.0/kugar/kudesigner/kudesigner.desktop koffice-1.6.0-p/kugar/kudesigner/kudesigner.desktop
--- koffice-1.6.0/kugar/kudesigner/kudesigner.desktop	2006-10-07 15:33:42.000000000 +0200
+++ koffice-1.6.0-p/kugar/kudesigner/kudesigner.desktop	2006-10-23 02:47:05.000000000 +0200
@@ -3,7 +3,7 @@
 Encoding=UTF-8
 Type=Application
 Exec=kudesigner %u
-Icon=kudesigner
+Icon=kugar
 DocPath=kugar/index.html
 Terminal=false
 Name=Kugar Designer
@@ -137,5 +137,4 @@ MimeType=application/x-kudesigner
 X-KDE-NativeMimeType=application/x-kudesigner
 X-KDE-Library=libkudesignerpart
 ServiceTypes=KOfficePart,KParts/ReadOnlyPart,KParts/ReadWritePart
-
-Categories=Qt;KDE;Office;X-KDE-More;
+Categories=Qt;KDE;Office;
diff -Nurp koffice-1.6.0/tools/kthesaurus/KThesaurus.desktop koffice-1.6.0-p/tools/kthesaurus/KThesaurus.desktop
--- koffice-1.6.0/tools/kthesaurus/KThesaurus.desktop	2006-10-07 15:29:15.000000000 +0200
+++ koffice-1.6.0-p/tools/kthesaurus/KThesaurus.desktop	2006-10-23 03:15:21.000000000 +0200
@@ -69,4 +69,5 @@ GenericName[xh]=Amagama Ahlobeneyo
 GenericName[zh_CN]=相关单词
 GenericName[zh_TW]=相關單字
 X-KDE-StartupNotify=true
-Categories=Qt;KDE;Office;X-KDE-More;
+Categories=Qt;KDE;Office;
+NoDisplay=true
