diff -Naur koffice-2.1.1/kchart/CMakeLists.txt koffice-2.1.1-p/kchart/CMakeLists.txt
--- koffice-2.1.1/kchart/CMakeLists.txt	2010-01-08 11:43:13.000000000 +0100
+++ koffice-2.1.1-p/kchart/CMakeLists.txt	2010-01-31 16:51:34.000000000 +0100
@@ -57,36 +57,36 @@
 
 ########### target kchartpart ###############
 
-#set(kchartpart_PART_SRCS KChartFactoryInit.cpp )
+set(kchartpart_PART_SRCS KChartFactoryInit.cpp )
 
-#kde4_add_plugin(kchartpart WITH_PREFIX ${kchartpart_PART_SRCS})
+kde4_add_plugin(kchartpart WITH_PREFIX ${kchartpart_PART_SRCS})
 
 
-#target_link_libraries(kchartpart
-#	${KDE4_KDECORE_LIBS}
-#	kchartcommon
-	#chartshapelib
-#)
+target_link_libraries(kchartpart
+	${KDE4_KDECORE_LIBS}
+	kchartcommon
+	chartshapelib
+)
 
-#install(TARGETS kchartpart  DESTINATION ${PLUGIN_INSTALL_DIR})
+install(TARGETS kchartpart  DESTINATION ${PLUGIN_INSTALL_DIR})
 
 
 ########### target kchart ###############
 
-#set(kchart_KDEINIT_SRCS main.cpp )
+set(kchart_KDEINIT_SRCS main.cpp )
 
-#kde4_add_kdeinit_executable( kchart ${kchart_KDEINIT_SRCS})
+kde4_add_kdeinit_executable( kchart ${kchart_KDEINIT_SRCS})
 
-#target_link_libraries(kdeinit_kchart  ${KDE4_KDECORE_LIBS} ${QT_QTGUI_LIBRARY} komain )
+target_link_libraries(kdeinit_kchart  ${KDE4_KDECORE_LIBS} ${QT_QTGUI_LIBRARY} komain )
 
-#install(TARGETS kdeinit_kchart  ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kdeinit_kchart  ${INSTALL_TARGETS_DEFAULT_ARGS})
 
-#target_link_libraries( kchart kdeinit_kchart )
-#install(TARGETS kchart  ${INSTALL_TARGETS_DEFAULT_ARGS})
+target_link_libraries( kchart kdeinit_kchart )
+install(TARGETS kchart  ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### install files ###############
 
 install( FILES  kchartpart.desktop  DESTINATION ${SERVICES_INSTALL_DIR})
-#install( FILES  kchart.rc kchart_readonly.rc  DESTINATION ${DATA_INSTALL_DIR}/kchart)
-#install( FILES  kchart.desktop  DESTINATION ${XDG_APPS_INSTALL_DIR})
-#install( FILES  kchart.kcfg  DESTINATION  ${KCFG_INSTALL_DIR} )
+install( FILES  kchart.rc kchart_readonly.rc  DESTINATION ${DATA_INSTALL_DIR}/kchart)
+install( FILES  kchart.desktop  DESTINATION ${XDG_APPS_INSTALL_DIR})
+install( FILES  kchart.kcfg  DESTINATION  ${KCFG_INSTALL_DIR} )
