diff -Naur koffice-2.1.1/kformula/CMakeLists.txt koffice-2.1.1-p/kformula/CMakeLists.txt
--- koffice-2.1.1/kformula/CMakeLists.txt	2010-01-08 11:36:56.000000000 +0100
+++ koffice-2.1.1-p/kformula/CMakeLists.txt	2010-01-31 16:40:57.000000000 +0100
@@ -51,14 +51,14 @@
 
 target_link_libraries(kdeinit_kformula_bin komain )
 
-#install(TARGETS kdeinit_kformula_bin  ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kdeinit_kformula_bin  ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 set_target_properties(kdeinit_kformula_bin PROPERTIES OUTPUT_NAME kdeinit_kformula)
 
 target_link_libraries( kformula_bin kdeinit_kformula_bin )
 set_target_properties( kformula_bin PROPERTIES OUTPUT_NAME kformula)
 
-#install(TARGETS kformula_bin  ${INSTALL_TARGETS_DEFAULT_ARGS})
+install(TARGETS kformula_bin  ${INSTALL_TARGETS_DEFAULT_ARGS})
 
 ########### install files ###############
 
