diff -Naur sbcl-0.9.4-x86-64-linux.orig/install.sh sbcl-0.9.4-x86-64-linux/install.sh
--- sbcl-0.9.4-x86-64-linux.orig/install.sh	2005-09-04 20:54:40.000000000 +0200
+++ sbcl-0.9.4-x86-64-linux/install.sh	2005-09-04 20:54:59.000000000 +0200
@@ -95,28 +95,28 @@
 cp doc/sbcl.1 $BUILD_ROOT$MAN_DIR/man1/ && echo " man $BUILD_ROOT$MAN_DIR/man1/sbcl.1"
 
 # info
-for info in doc/manual/*.info
-do
-  cp $info $BUILD_ROOT$INFO_DIR/ \
-      && echo -n " info $BUILD_ROOT$INFO_DIR/`basename $info` $BUILD_ROOT$INFO_DIR/dir" \
-      && ( install-info $BUILD_ROOT$INFO_DIR/`basename $info` $BUILD_ROOT$INFO_DIR/dir > /dev/null 2>&1 \
-           || echo -n " (could not add to system catalog)" ) \
-      && echo
-done
+#for info in doc/manual/*.info
+#do
+#  cp $info $BUILD_ROOT$INFO_DIR/ \
+#      && echo -n " info $BUILD_ROOT$INFO_DIR/`basename $info` $BUILD_ROOT$INFO_DIR/dir" \
+#      && ( install-info $BUILD_ROOT$INFO_DIR/`basename $info` $BUILD_ROOT$INFO_DIR/dir > /dev/null 2>&1 \
+#           || echo -n " (could not add to system catalog)" ) \
+#      && echo
+#done
 
 # pdf
-for pdf in doc/manual/*.pdf
-do
-  cp $pdf $BUILD_ROOT$DOC_DIR/ \
-      && echo " pdf $BUILD_ROOT$DOC_DIR/`basename $pdf`"
-done
+#for pdf in doc/manual/*.pdf
+#do
+#  cp $pdf $BUILD_ROOT$DOC_DIR/ \
+#      && echo " pdf $BUILD_ROOT$DOC_DIR/`basename $pdf`"
+#done
 
 # html
-for html in doc/manual/sbcl doc/manual/asdf
-do
-  test -d $html && cp -R -L $html $BUILD_ROOT$DOC_DIR/html \
-      && echo " html $BUILD_ROOT$DOC_DIR/html/`basename $html`/index.html"
-done
+#for html in doc/manual/sbcl doc/manual/asdf
+#do
+#  test -d $html && cp -R -L $html $BUILD_ROOT$DOC_DIR/html \
+#      && echo " html $BUILD_ROOT$DOC_DIR/html/`basename $html`/index.html"
+#done
 
 for f in BUGS SUPPORT CREDITS COPYING NEWS
 do
