diff -Naur gcc-3.4.4-20050522-1.orig/gcc/config/i386/cygming.h gcc-3.4.4-20050522-1/gcc/config/i386/cygming.h
--- gcc-3.4.4-20050522-1.orig/gcc/config/i386/cygming.h	2004-11-03 09:57:03.000000000 +0100
+++ gcc-3.4.4-20050522-1/gcc/config/i386/cygming.h	2005-10-15 12:44:03.000000000 +0200
@@ -329,7 +329,7 @@
 /* DWARF2 Unwinding doesn't work with exception handling yet.  To make
    it work, we need to build a libgcc_s.dll, and dcrt0.o should be
    changed to call __register_frame_info/__deregister_frame_info.  */
-#define DWARF2_UNWIND_INFO 1
+#define DWARF2_UNWIND_INFO 0
 
 /* Don't assume anything about the header files.  */
 #define NO_IMPLICIT_EXTERN_C
