diff -Naur chromium-65.0.3325.146/chrome/browser/vr/sample_queue.h chromium-65.0.3325.146-p/chrome/browser/vr/sample_queue.h
--- chromium-65.0.3325.146/chrome/browser/vr/sample_queue.h	2018-03-07 00:04:20.000000000 +0100
+++ chromium-65.0.3325.146-p/chrome/browser/vr/sample_queue.h	2018-03-08 17:57:08.600144962 +0100
@@ -6,6 +6,7 @@
 #define CHROME_BROWSER_VR_SAMPLE_QUEUE_H_
 
 #include <vector>
+#include <stdint.h>
 
 #include "base/macros.h"
 
diff -Naur chromium-65.0.3325.146/components/assist_ranker/ranker_example_util.cc chromium-65.0.3325.146-p/components/assist_ranker/ranker_example_util.cc
--- chromium-65.0.3325.146/components/assist_ranker/ranker_example_util.cc	2018-03-07 00:04:24.000000000 +0100
+++ chromium-65.0.3325.146-p/components/assist_ranker/ranker_example_util.cc	2018-03-08 17:55:46.490678020 +0100
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include <math.h>
 #include "components/assist_ranker/ranker_example_util.h"
 #include "base/bit_cast.h"
 #include "base/format_macros.h"
