--- chromium-58.0.3029.81/ui/gfx/render_text_harfbuzz.h	2017-04-19 21:06:54.000000000 +0200
+++ chromium-58.0.3029.81.new/ui/gfx/render_text_harfbuzz.h	2017-04-20 10:51:36.000000000 +0200
@@ -12,7 +12,7 @@
 #include <vector>
 
 #include "base/macros.h"
-#include "third_party/harfbuzz-ng/src/hb.h"
+#include <harfbuzz/hb.h>
 #include "third_party/icu/source/common/unicode/ubidi.h"
 #include "third_party/icu/source/common/unicode/uscript.h"
 #include "ui/gfx/render_text.h"
