diff -Naur chromium-65.0.3325.146/media/video/video_encode_accelerator.h chromium-65.0.3325.146-p/media/video/video_encode_accelerator.h
--- chromium-65.0.3325.146/media/video/video_encode_accelerator.h	2018-03-07 00:04:35.000000000 +0100
+++ chromium-65.0.3325.146-p/media/video/video_encode_accelerator.h	2018-03-08 05:37:47.876172483 +0100
@@ -161,7 +161,7 @@
   // handed back to the client, and afterwards the |flush_callback| will be
   // called. The FlushCallback takes a boolean argument: |true| indicates the
   // flush is complete; |false| indicates the flush is cancelled due to errors
-  // or destruction. The client should not invoke Flush() or Encode() while the\
+  // or destruction. The client should not invoke Flush() or Encode() while the
   // previous Flush() is not finished yet.
   virtual void Flush(FlushCallback flush_callback);
 
