diff -up chromium-79.0.3945.88/tools/gn/build/gen.py.41~ chromium-79.0.3945.88/tools/gn/build/gen.py
--- chromium-79.0.3945.88/tools/gn/build/gen.py.41~	2019-12-23 23:08:34.915379587 +0100
+++ chromium-79.0.3945.88/tools/gn/build/gen.py	2019-12-23 23:09:39.021376477 +0100
@@ -96,7 +96,7 @@ def main(argv):
   parser.add_option('--no-strip', action='store_true',
                     help='Don\'t strip release build. Useful for profiling.')
   parser.add_option('--no-static-libstdc++', action='store_true',
-                    default=False, dest='no_static_libstdcpp',
+                    default=True, dest='no_static_libstdcpp',
                     help='Don\'t link libstdc++ statically')
   options, args = parser.parse_args(argv)
 
