diff -up chromium-46.0.2490.86/media/cast/receiver/audio_decoder_unittest.cc.use_system_opus chromium-46.0.2490.86/media/cast/receiver/audio_decoder_unittest.cc
--- chromium-46.0.2490.86/media/cast/receiver/audio_decoder_unittest.cc.use_system_opus	2015-10-07 04:42:12.000000000 +0200
+++ chromium-46.0.2490.86/media/cast/receiver/audio_decoder_unittest.cc	2015-11-11 16:11:13.535250609 +0100
@@ -13,7 +13,7 @@
 #include "media/cast/test/utility/audio_utility.h"
 #include "media/cast/test/utility/standalone_cast_environment.h"
 #include "testing/gtest/include/gtest/gtest.h"
-#include "third_party/opus/src/include/opus.h"
+#include <opus/opus.h>
 
 namespace media {
 namespace cast {
