diff -up chromium-55.0.2883.75/sandbox/linux/BUILD.gn.sandboxpie chromium-55.0.2883.75/sandbox/linux/BUILD.gn
--- chromium-55.0.2883.75/sandbox/linux/BUILD.gn.sandboxpie	2016-12-01 18:02:17.000000000 -0500
+++ chromium-55.0.2883.75/sandbox/linux/BUILD.gn	2016-12-12 16:26:06.863426221 -0500
@@ -279,11 +279,17 @@ if (is_linux) {
       # For ULLONG_MAX
       "-std=gnu99",
 
+      "-fPIE",
+
       # These files have a suspicious comparison.
       # TODO fix this and re-enable this warning.
       "-Wno-sign-compare",
     ]
 
+    ldflags = [
+      "-pie",
+    ]
+
     import("//build/config/compiler/compiler.gni")
     import("//build/config/sanitizers/sanitizers.gni")
     if (is_component_build || using_sanitizer) {
@@ -293,7 +299,7 @@ if (is_linux) {
       # other flags that executable_config might have.
       configs -= [ "//build/config:executable_config" ]
       if (!use_gold) {
-        ldflags = [ "-Wl,--disable-new-dtags" ]
+        ldflags += [ "-Wl,--disable-new-dtags" ]
       }
     }
 
