--- chromium-67.0.3396.62/build/config/compiler/BUILD.gn	2018-05-30 10:43:03.000000000 +0200
+++ chromium-67.0.3396.62.new/build/config/compiler/BUILD.gn	2018-05-31 16:09:56.489182567 +0200
@@ -251,7 +251,7 @@
       if (is_debug) {
         cflags += [ "-fstack-protector-strong" ]
       } else {
-        cflags += [ "-fstack-protector" ]
+        cflags += [ "-fstack-protector-strong" ]
       }
     } else if ((is_posix && !is_chromeos && !is_nacl) || is_fuchsia) {
       # TODO(phajdan.jr): Use -fstack-protector-strong when our gcc supports it.
@@ -263,7 +263,7 @@
         cflags += [ "-fno-stack-protector" ]
       } else if (current_os != "aix") {
         # Not available on aix.
-        cflags += [ "-fstack-protector" ]
+        cflags += [ "-fstack-protector-strong" ]
       }
     }
   } else {
@@ -680,7 +680,11 @@
     if (current_cpu == "x64") {
       cflags += [
         "-m64",
+        "-fPIE",
         "-march=x86-64",
+        "-mtune=generic",
+        "-fPIC",
+        "-Wp,-D_FORTIFY_SOURCE=2",
       ]
       ldflags += [ "-m64" ]
     } else if (current_cpu == "x86") {
