diff -Naur chromium-65.0.3325.146/build/config/compiler/BUILD.gn chromium-65.0.3325.146-p/build/config/compiler/BUILD.gn
--- chromium-65.0.3325.146/build/config/compiler/BUILD.gn	2018-03-07 00:04:15.000000000 +0100
+++ chromium-65.0.3325.146-p/build/config/compiler/BUILD.gn	2018-03-08 19:23:33.125044606 +0100
@@ -248,12 +248,12 @@
       # The strong variant of the stack protector significantly increases
       # binary size, so only enable it in debug mode.
       if (is_debug) {
-        cflags += [ "-fstack-protector-strong" ]
+        cflags += [ "-fstack-protector-strong-strong" ]
       } else {
-        cflags += [ "-fstack-protector" ]
+        cflags += [ "-fstack-protector-strong" ]
       }
     } else if (is_posix && !is_chromeos && !is_nacl) {
-      # TODO(phajdan.jr): Use -fstack-protector-strong when our gcc supports it.
+      # TODO(phajdan.jr): Use -fstack-protector-strong-strong when our gcc supports it.
       # See also https://crbug.com/533294
       cflags += [ "--param=ssp-buffer-size=4" ]
 
@@ -262,7 +262,7 @@
         cflags += [ "-fno-stack-protector" ]
       } else if (current_os != "aix") {
         # Not available on aix.
-        cflags += [ "-fstack-protector" ]
+        cflags += [ "-fstack-protector-strong" ]
       }
     }
   }
@@ -639,7 +639,10 @@
     if (current_cpu == "x64") {
       cflags += [
         "-m64",
+        "-fPIE",
         "-march=x86-64",
+        "-mtune=generic",
+        "-fPIC",
       ]
       ldflags += [ "-m64" ]
     } else if (current_cpu == "x86") {
