--- chromium-129.0.6668.58.new/build/config/clang/BUILD.gn	2024-09-20 11:39:02.391029638 +0200
+++ chromium-129.0.6668.58/build/config/clang/BUILD.gn	2024-09-20 12:59:06.198557111 +0200
@@ -191,7 +191,7 @@
         _dir = "darwin"
       } else if (is_linux || is_chromeos) {
         if (current_cpu == "x64") {
-          _dir = "x86_64-unknown-linux-gnu"
+          _dir = "x86_64-frugalware-linux"
         } else if (current_cpu == "x86") {
           _dir = "i386-unknown-linux-gnu"
         } else if (current_cpu == "arm") {
