--- src/chrome/test/chromedriver/embed_version_in_cpp.py	2013/10/10 00:32:56	227841
+++ src/chrome/test/chromedriver/embed_version_in_cpp.py	2013/10/10 00:33:17	227842
@@ -25,9 +25,12 @@
   options, args = parser.parse_args()
 
   version = open(options.version_file, 'r').read().strip()
-  revision = lastchange.FetchVersionInfo(None).revision.strip()
+  revision = lastchange.FetchVersionInfo(None).revision
+  if revision:
+    version += '.' + revision.strip()
+
   global_string_map = {
-      'kChromeDriverVersion': version + '.' + revision
+      'kChromeDriverVersion': version
   }
   cpp_source.WriteSource('version',
                          'chrome/test/chromedriver',
