diff -Naur chromium-53.0.2785.116/v8/gypfiles/toolchain.gypi chromium-53.0.2785.116-gcc6/v8/gypfiles/toolchain.gypi
--- chromium-53.0.2785.116/v8/gypfiles/toolchain.gypi	2016-09-15 03:06:53.000000000 +0200
+++ chromium-53.0.2785.116-gcc6/v8/gypfiles/toolchain.gypi	2016-09-22 13:24:44.246588268 +0200
@@ -74,6 +74,10 @@
     # Some versions of GCC 4.5 seem to need -fno-strict-aliasing.
     'v8_no_strict_aliasing%': 0,
 
+     # V8 needs support for illegal null this pointers, and must disable
+     # optimizations in GCC 6 that relies on correct behavior.
+    'v8_no_delete_null_pointer_checks%': 0,
+
     # Chrome needs this definition unconditionally. For standalone V8 builds,
     # it's handled in gypfiles/standalone.gypi.
     'want_separate_host_toolset%': 1,
@@ -1074,6 +1078,10 @@
           [ 'v8_no_strict_aliasing==1', {
             'cflags': [ '-fno-strict-aliasing' ],
           }],
+           [ 'v8_no_delete_null_pointer_checks==1', {
+            'cflags_cc': [ '-fno-delete-null-pointer-checks' ],
+          }],
+
         ],  # conditions
       }],
       ['OS=="solaris"', {
