--- hotdoc/setup.py	2020-09-08 15:54:19.115442683 +0200
+++ hotdoc.new/setup.py	2020-09-09 10:39:31.104443178 +0200
@@ -416,7 +416,7 @@
         INSTALL_REQUIRES += [
             'pkgconfig==1.1.0',
             'cchardet',
-            'networkx==1.11'
+            'networkx'
         ]
         PACKAGE_DATA['hotdoc.extensions.gi'] = ['html_templates/*']
         PACKAGE_DATA['hotdoc.extensions.gi.transition_scripts'] = ['translate_sections.sh']
--- hotdoc/setup.py	2020-09-09 14:41:41.499827124 +0200
+++ hotdoc.new/setup.py	2020-09-09 15:32:47.076221325 +0200
@@ -315,7 +315,7 @@
     'lxml',
     'schema',
     'appdirs',
-    'wheezy.template==0.1.167',
+    'wheezy.template',
     'toposort>=1.4',
     'xdg>=4.0.0',
 ]
