diff -Naur hyperscan-5.0.0/CMakeLists.txt hyperscan-5.0.0-pcre/CMakeLists.txt
--- hyperscan-5.0.0/CMakeLists.txt	2018-07-09 18:45:44.000000000 +0200
+++ hyperscan-5.0.0-pcre/CMakeLists.txt	2018-11-09 15:59:28.775918377 +0100
@@ -452,7 +452,7 @@
 # PCRE check, we have a fixed requirement for PCRE to use Chimera
 # and hscollider
 set(PCRE_REQUIRED_MAJOR_VERSION 8)
-set(PCRE_REQUIRED_MINOR_VERSION 41)
+set(PCRE_REQUIRED_MINOR_VERSION 43)
 set(PCRE_REQUIRED_VERSION ${PCRE_REQUIRED_MAJOR_VERSION}.${PCRE_REQUIRED_MINOR_VERSION})
 include (${CMAKE_MODULE_PATH}/pcre.cmake)
 if (NOT CORRECT_PCRE_VERSION)
