diff -uraN nant-0.85.orig/src/NAnt.DotNet/Tasks/ScriptTask.cs nant-0.85/src/NAnt.DotNet/Tasks/ScriptTask.cs
--- nant-0.85.orig/src/NAnt.DotNet/Tasks/ScriptTask.cs	2008-03-27 11:34:21.257506822 +0100
+++ nant-0.85/src/NAnt.DotNet/Tasks/ScriptTask.cs	2008-03-27 11:38:37.163506860 +0100
@@ -516,7 +516,7 @@
         #region Private Static Methods
 
         private static CodeDomProvider CreateCodeDomProvider(string typeName, string assemblyName) {
-            Assembly providerAssembly = Assembly.LoadWithPartialName(assemblyName);
+            Assembly providerAssembly = Assembly.Load(assemblyName);
             if (providerAssembly == null) {
                 throw new ArgumentException(string.Format(CultureInfo.InvariantCulture,
                     ResourceUtils.GetString("NA2037"), assemblyName));
