--- nant-0.85/src/NAnt.Core/Task.cs	2005-10-21 18:13:06.000000000 +0100
+++ nant-0.85/src/NAnt.Core/Task.cs.new	2007-11-12 13:13:05.000000000 +0000
@@ -400,7 +400,7 @@
         #region Protected Instance Methods
 
         /// <summary><note>Deprecated (to be deleted).</note></summary>
-        [Obsolete("Deprecated- Use InitializeTask instead")]
+        //[Obsolete("Deprecated- Use InitializeTask instead")]
         protected override void InitializeElement(XmlNode elementNode) {
             // Just defer for now so that everything just works
             InitializeTask(elementNode);
@@ -415,4 +415,4 @@
 
         #endregion Protected Instance Methods
     }
-}
\ No newline at end of file
+}
