diff -Naur xchat-systray-integration-2.4.5/src/menu.c xchat-systray-integration-2.4.5.new/src/menu.c
--- xchat-systray-integration-2.4.5/src/menu.c	2004-04-30 18:08:10.000000000 +0530
+++ xchat-systray-integration-2.4.5.new/src/menu.c	2008-03-02 19:12:25.000000000 +0530
@@ -10,7 +10,6 @@
 	char command[156]="exec ";
 	strcat(command, bpath);
 	xchat_command( ph, command );
-	g_free(command);
 
 }
 
@@ -20,7 +19,6 @@
 	char command[156]="exec ";
 	strcat(command, epath);
 	xchat_command( ph, command );
-	g_free(command);
 
 }
 
@@ -30,7 +28,6 @@
 	char command[156]="exec ";
 	strcat(command, ppath);
 	xchat_command( ph, command );
-	g_free(command);
 
 }
 
