--- libextractor-1.3/src/plugins/exiv2_extractor.cc	2013-06-25 13:02:05.000000000 +0200
+++ libextractor-1.3.new/src/plugins/exiv2_extractor.cc	2017-04-29 22:47:09.000000000 +0200
@@ -180,7 +180,7 @@
    *
    * @return -1 on error
    */
-  virtual long int size (void) const;
+  virtual size_t size (void) const;
 
   /**
    * Check if file is open.
@@ -445,7 +445,7 @@
  *
  * @return -1 on error
  */
-long int 
+size_t
 ExtractorIO::size (void) const
 {
   return (long) ec->get_size (ec->cls);
