--- shiboken2/libshiboken/embed/signature_bootstrap.py	2022-06-30 09:54:16.000000000 +0200
+++ shiboken2.new/libshiboken/embed/signature_bootstrap.py	2022-11-03 12:54:35.082930005 +0100
@@ -211,7 +211,7 @@
         return self if self._mod2path.get(fullname) else None
 
     def load_module(self, fullname):
-        import importlib
+        import importlib.machinery
         import sys
 
         filename = self._mod2path.get(fullname)
