diff -Naur aMule-2.1.3/src/ServerWnd.cpp aMule-2.1.3-p/src/ServerWnd.cpp
--- aMule-2.1.3/src/ServerWnd.cpp	2006-03-17 03:06:48.000000000 +0100
+++ aMule-2.1.3-p/src/ServerWnd.cpp	2007-12-07 10:09:40.000000000 +0100
@@ -70,7 +70,7 @@
 
 	CastChild( ID_SRV_SPLITTER, wxSplitterWindow )->SetSashPosition(splitter_pos, true);
 	CastChild( ID_SRV_SPLITTER, wxSplitterWindow )->SetSashGravity(0.5f);
-	CastChild( IDC_SERVERLISTURL, wxTextCtrl )->SetValue(wxT("http://ocbmaurice.dyns.net/pl/slist.pl?download/server-best.met"));
+	CastChild( IDC_SERVERLISTURL, wxTextCtrl )->SetValue(wxT("http://www.server-met.de/dl.php?load=gz&trace=33250645.6944"));
 
 	// Insert two columns, currently without a header
 	wxListCtrl* ED2KInfoList = CastChild( ID_ED2KINFO, wxListCtrl );
