diff -Naurd foo2zjs.orig/Makefile foo2zjs/Makefile
--- foo2zjs.orig/Makefile	2004-01-12 16:17:08.000000000 +0100
+++ foo2zjs/Makefile	2004-01-12 16:20:41.000000000 +0100
@@ -5,8 +5,7 @@
 VERSION=0.0
 
 # Installation prefix...
-PREFIX=/usr/local
-PREFIX=/usr
+PREFIX=$(DESTDIR)/usr
 
 # Pathnames for this package...
 BIN=$(PREFIX)/bin
@@ -15,7 +14,7 @@
 DOCDIR=$(PREFIX)/share/doc/foo2zjs/
 
 # Pathnames for referenced packages...
-FOODB=/usr/share/foomatic/db/source
+FOODB=$(DESTDIR)/usr/share/foomatic/db/source
 
 # User ID's
 LPuid=-olp
@@ -388,7 +387,7 @@
 ppd:
 	./getweb ppd
 
-MODEL=/usr/share/cups/model
+MODEL=$(DESTDIR)/usr/share/cups/model
 install-ppd:
 	#
 	# Install PPD files for CUPS
