--- foo2zjs.orig/Makefile	2005-12-30 06:40:43.000000000 +0100
+++ foo2zjs/Makefile	2006-01-03 20:30:31.000000000 +0100
@@ -6,8 +6,7 @@
 VERSION=0.0
 
 # Installation prefix...
-PREFIX=/usr/local
-PREFIX=/usr
+PREFIX=$(DESTDIR)/usr
 
 # Pathnames for this package...
 BIN=$(PREFIX)/bin
@@ -18,7 +17,7 @@
 DOCDIR=$(PREFIX)/share/doc/foo2zjs/
 
 # Pathnames for referenced packages...
-FOODB=/usr/share/foomatic/db/source
+FOODB=$(DESTDIR)/usr/share/foomatic/db/source
 
 # User ID's
 LPuid=-olp
@@ -249,7 +248,7 @@
 	# Install driver, wrapper, and development tools
 	#
 	install -c $(PROGS) $(SHELLS) $(BIN)/
-	install -c $(BINPROGS) /bin/
+	install -c $(BINPROGS) $(PREFIX)/bin/
 	#
 	# Install gamma correction files.  These are just templates,
 	# and don't actually do anything right now.  If anybody wants
@@ -357,7 +356,7 @@
 	    fi; \
 	done
 
-MODEL=/usr/share/cups/model
+MODEL=$(DESTDIR)/usr/share/cups/model
 install-ppd:
 	#
 	# Install PPD files for CUPS
@@ -369,8 +368,8 @@
 	    done; \
 	fi
 
-USBDIR=/etc/hotplug/usb
-UDEVDIR=/etc/udev/rules.d
+USBDIR=$(DESTDIR)/etc/hotplug/usb
+UDEVDIR=$(DESTDIR)/etc/udev/rules.d
 RULES=hplj10xx.rules
 install-hotplug:
 	[ -d $(USBDIR) ] || install -d -m 755 $(USBDIR)/
@@ -609,8 +608,7 @@
 	install -c -m 644 README $(DOCDIR)
 	install -c -m 644 ChangeLog $(DOCDIR)
 
-GROFF=/usr/local/test/bin/groff
-GROFF=groff
+GROFF=/usr/bin/groff
 manual.pdf: $(MANPAGES)
 	-$(GROFF) -t -man $(MANPAGES) | ps2pdf - $@
 
