diff -Naur foo2zjs.orig/Makefile foo2zjs/Makefile
--- foo2zjs.orig/Makefile	2005-11-25 16:33:11.000000000 +0100
+++ foo2zjs/Makefile	2005-11-30 18:13:48.000000000 +0100
@@ -6,8 +6,7 @@
 VERSION=0.0
 
 # Installation prefix...
-PREFIX=/usr/local
-PREFIX=/usr
+PREFIX=$(DESTDIR)/usr
 
 # Pathnames for this package...
 BIN=$(PREFIX)/bin
@@ -18,7 +17,7 @@
 DOCDIR=$(PREFIX)/share/doc/foo2zjs/
 
 # Pathnames for referenced packages...
-FOODB=/usr/share/foomatic/db/source
+FOODB=$(DESTDIR)/usr/share/foomatic/db/source
 
 # User ID's
 LPuid=-olp
@@ -340,8 +339,8 @@
 	    fi; \
 	done
 
-USBDIR=/etc/hotplug/usb
-UDEVDIR=/etc/udev/rules.d
+USBDIR=$(DESTDIR)/etc/hotplug/usb
+UDEVDIR=$(DESTDIR)/etc/udev/rules.d
 RULES=hplj10xx.rules
 install-hotplug:
 	[ -d $(USBDIR) ] || install -d -m 755 $(USBDIR)/
@@ -503,7 +502,7 @@
 ppd:
 	./getweb ppd
 
-MODEL=/usr/share/cups/model
+MODEL=$(DESTDIR)/usr/share/cups/model
 install-ppd:
 	#
 	# Install PPD files for CUPS
@@ -560,8 +559,7 @@
 	install -c -m 644 README $(DOCDIR)
 	install -c -m 644 ChangeLog $(DOCDIR)
 
-GROFF=/usr/local/test/bin/groff
-GROFF=groff
+GROFF=/usr/bin/groff
 manual.pdf: $(MANPAGES)
 	-$(GROFF) -t -man $(MANPAGES) | ps2pdf - $@
 
