diff -urN evolution-sharp-0.11.1.orig/configure.in evolution-sharp-0.11.1/configure.in
--- evolution-sharp-0.11.1.orig/configure.in	2006-04-29 17:44:50.000000000 +0100
+++ evolution-sharp-0.11.1/configure.in	2006-08-30 17:56:26.000000000 +0100
@@ -122,6 +122,7 @@
 PKG_CHECK_MODULES(ESHELL22, "evolution-shell-2.2", eshellver="2.2", foo=bar)
 PKG_CHECK_MODULES(ESHELL24, "evolution-shell-2.4", eshellver="2.4", foo=bar)
 PKG_CHECK_MODULES(ESHELL26, "evolution-shell-2.6", eshellver="2.6", foo=bar)
+PKG_CHECK_MODULES(ESHELL28, "evolution-shell-2.8", eshellver="2.8", foo=bar)
 
 if test "x$eshellver" = "x"; then
 	AC_MSG_ERROR([You need to install evolution-shell])
