--- library/base/base/xml_util_functions.h.~1~	2011-03-11 00:03:28.000000000 +0100
+++ library/base/base/xml_util_functions.h	2011-03-14 18:57:34.571514279 +0100
@@ -20,10 +20,6 @@
 #ifndef _XML_UTIL_FUNCTIONS_H_
 #define _XML_UTIL_FUNCTIONS_H_
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-
 #include "base/common.h"
 //#include "util.h"
 
@@ -31,6 +27,10 @@
 #include <libxml/tree.h>
 //#endif
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 typedef char base_bool;
 
 BASELIBRARY_PUBLIC_FUNC xmlDocPtr base_xmlParseFile(const char *filename);

