=== modified file 'configure.in'
--- configure.in	2011-06-29 17:49:13 +0000
+++ configure.in	2011-10-17 10:05:52 +0000
@@ -246,7 +246,7 @@
 cfg=sysconfig.get_config_vars()
 lib = os.path.join(cfg['LIBDIR'], cfg['LDLIBRARY'])
 if os.path.exists(lib):
-    print lib")]
+    print \"-L\" + cfg['LIBDIR'] + \" -l\" + cfg['LDLIBRARY'].lstrip(\"lib\").rstrip(\".so\")")]
 PYTHON_CFLAGS=[$(python -c "from distutils import sysconfig
 print sysconfig.get_python_inc()")]
 PYTHON_CFLAGS="-I$PYTHON_CFLAGS"

=== modified file 'frontend/linux/workbench/main.cpp'
--- frontend/linux/workbench/main.cpp	2011-04-25 17:08:20 +0000
+++ frontend/linux/workbench/main.cpp	2011-10-17 13:25:23 +0000
@@ -247,7 +247,8 @@
   if (!wboptions.quit_when_done)
   {
     Gtk::Window *splash = show_splash();
-  
+    splash->get_window()->freeze_updates(); // Added to stop idle window update requests storm in gnome-3
+
     while (app.events_pending())
       app.iteration(false);
 

=== modified file 'library/forms/gtk/src/lf_utilities.cpp'
--- library/forms/gtk/src/lf_utilities.cpp	2011-08-05 20:43:23 +0000
+++ library/forms/gtk/src/lf_utilities.cpp	2011-10-17 13:26:09 +0000
@@ -759,6 +759,7 @@
 #endif
   window->process_updates(true);
   show_now();
+  window->freeze_updates(); // Added to stop idle window update requests storm in gnome-3
 }
 
 

