diff -Naur print-manager-16.08.2/add-printer/DevicesModel.cpp print-manager-16.08.2-p/add-printer/DevicesModel.cpp
--- print-manager-16.08.2/add-printer/DevicesModel.cpp	2016-09-12 18:59:25.000000000 +0200
+++ print-manager-16.08.2-p/add-printer/DevicesModel.cpp	2016-10-28 01:20:11.436747098 +0200
@@ -340,7 +340,6 @@
 {
     qWarning() << error <<  message;
     groupedDevicesFallback();
-    emit errorMessage(i18n("Failed to group devices: '%1'",error.message()));
     emit loaded();
 }
 
diff -Naur print-manager-16.08.2/libkcups/SelectMakeModel.cpp print-manager-16.08.2-p/libkcups/SelectMakeModel.cpp
--- print-manager-16.08.2/libkcups/SelectMakeModel.cpp	2016-09-12 18:59:25.000000000 +0200
+++ print-manager-16.08.2-p/libkcups/SelectMakeModel.cpp	2016-10-28 01:19:24.628746404 +0200
@@ -243,8 +243,6 @@
 
     // Show the PPDs anyway
     m_gotBestDrivers = true;
-    ui->messageWidget->setText(i18n("Failed to search for a recommended driver: '%1'", error.message()));
-    ui->messageWidget->animatedShow();
     setModelData();
 }
 
