--- xpn-1.0.0.orig/xpn.py
+++ xpn-1.0.0/xpn.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 import os,shutil
 #os.environ['PATH'] += ';'+os.path.join('gtk/lib')+';'+os.path.join('gtk/bin') #OLD
 #os.environ['PATH'] = os.path.join('gtk/lib')+';'+os.path.join('gtk/bin')+';'+os.path.join('gtk\\lib')+';'+os.path.join('gtk\\bin')+';'+os.environ['PATH']  #NEW
--- xpn-1.0.0.orig/xpn_src/Show_Logs.py
+++ xpn-1.0.0/xpn_src/Show_Logs.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 import gtk
 import os
 from xpn_src.UserDir import get_wdir
--- xpn-1.0.0.orig/xpn_src/add_tag.py
+++ xpn-1.0.0/xpn_src/add_tag.py
@@ -1,4 +1,3 @@
-#!/usr/bin/env python
 import gtk
 import os
 from locale import getdefaultlocale
--- xpn.orig/xpn.py
+++ xpn/xpn.py
@@ -2631,21 +2631,21 @@
     def load_languages(self):
         #loading translation
         if self.configs["lang"]=="it":
-            it=gettext.translation("xpn","lang",["it"])
+            it=gettext.translation("xpn","/usr/share/locale",["it"])
             it.install()
             try:
                 #trying to force GTK translation
                 locale.setlocale(locale.LC_MESSAGES,"it_IT")
             except: pass
         elif self.configs["lang"]=="fr":
-            fr=gettext.translation("xpn","lang",["fr"])
+            fr=gettext.translation("xpn","/usr/share/locale",["fr"])
             fr.install()
             try:
                 #trying to force GTK translation
                 locale.setlocale(locale.LC_MESSAGES,"fr_FR")
             except: pass
         elif self.configs["lang"]=="de":
-            de=gettext.translation("xpn","lang",["de"])
+            de=gettext.translation("xpn","/usr/share/locale",["de"])
             de.install()
             try:
                 #trying to force GTK translation

