--- SConstruct	2008-04-20 19:05:35.000000000 -0500
+++ SConstruct.orig	2010-04-05 22:58:15.007357388 -0500
@@ -27,14 +27,14 @@
 opts.Add('LIBPATH', 'library path', [])
 
 opts.Add('tolua_bin', 'the resulting binary', 'tolua++')
-opts.Add('tolua_lib', 'the resulting library', 'tolua++')
+opts.Add('tolua_lib', 'the resulting library', 'tolua++-5.1')
 opts.Add('TOLUAPP', 'the name of the tolua++ binary (to use with built_dev=1)', 'tolua++')
 
 opts.Add('prefix', 'The installation prefix')
 opts.Add('build_dev', 'Build for development (uses tolua to rebuild toluabind.c with the embeded scripts', 0)
 opts.Add('build_failsafe', "Build using 'factory default' toluabind file (in case build_dev fails)", 0)
 opts.Add('ENV', 'The environment variables')
-opts.Add('shared', 'Build a shared object', False)
+opts.Add('shared', 'Build a shared object', True)
 opts.Update(env)
 Help(opts.GenerateHelpText(env))
 
--- config_linux.py	2005-11-30 16:31:27.000000000 -0600
+++ config_linux.py.orig	2010-04-05 21:20:38.122745603 -0500
@@ -4,8 +4,8 @@
 
 # Compiler flags (based on Debian's installation of lua)
 #LINKFLAGS = ['-g']
-CCFLAGS = ['-I/usr/include/lua50', '-O2', '-ansi', '-Wall']
-#CCFLAGS = ['-I/usr/include/lua50', '-g']
+CCFLAGS = ['-I/usr/include/lua', '-O2', '-ansi', '-Wall']
+#CCFLAGS = ['-I/usr/include/lua', '-g']
 
 # this is the default directory for installation. Files will be installed on
 # <prefix>/bin, <prefix>/lib and <prefix>/include when you run 'scons install'
@@ -15,8 +15,8 @@
 #
 # You can see more 'generic' options for POSIX systems on config_posix.py
 
-prefix = '/usr/local'
+prefix = '/usr'
 
 # libraries (based on Debian's installation of lua)
-LIBS = ['lua50', 'lualib50', 'dl', 'm']
+LIBS = ['lua', 'liblua', 'dl', 'm']
 
--- custom-5.1.py	2008-04-20 19:05:35.000000000 -0500
+++ custom-5.1.py.orig	2010-04-05 21:31:16.832389012 -0500
@@ -1,9 +1,9 @@
-CCFLAGS = ['-I/usr/include/lua5.1', '-O2', '-ansi']
+CCFLAGS = ['-I/usr/include/lua', '-O2', '-ansi']
 #LIBPATH = ['/usr/local/lib']
-LIBS = ['lua5.1', 'dl', 'm']
+LIBS = ['lua', 'dl', 'm']
 prefix = '/mingw'
 #build_dev=1
-tolua_bin = 'tolua++5.1'
-tolua_lib = 'tolua++5.1'
+tolua_bin = 'tolua++'
+tolua_lib = 'tolua++-5.1'
 TOLUAPP = 'tolua++5.1'
 



