diff -Naur libdvdnav-0.1.10/configure libdvdnav-0.1.10-p/configure
--- libdvdnav-0.1.10/configure	2004-06-10 23:13:17.000000000 +0200
+++ libdvdnav-0.1.10-p/configure	2006-08-25 20:34:13.268761248 +0200
@@ -21459,7 +21459,7 @@
     ;;
 esac
 
-CFLAGS="-O3 -Wall -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE $CFLAGS"
+CFLAGS=" -Wall -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE $CFLAGS"
 DEBUG_CFLAGS="-g -DDEBUG $CFLAGS"
 
 
diff -Naur libdvdnav-0.1.10/configure.ac libdvdnav-0.1.10-p/configure.ac
--- libdvdnav-0.1.10/configure.ac	2004-06-10 23:13:02.000000000 +0200
+++ libdvdnav-0.1.10-p/configure.ac	2006-08-25 20:34:19.039883904 +0200
@@ -175,7 +175,7 @@
 dnl cflags
 dnl ---------------------------------------------
 dnl Common cflags for all platforms
-CFLAGS="-O3 -Wall -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE $CFLAGS"
+CFLAGS=" -Wall -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE $CFLAGS"
 DEBUG_CFLAGS="-g -DDEBUG $CFLAGS"
 
 AC_SUBST(DEBUG_CFLAGS)
