diff -Naur bcpg-jdk15on-148/src/org/bouncycastle/openpgp/test/PGPUnicodeTest.java bcpg-jdk15on-148.new/src/org/bouncycastle/openpgp/test/PGPUnicodeTest.java
--- bcpg-jdk15on-148/src/org/bouncycastle/openpgp/test/PGPUnicodeTest.java	2013-02-08 09:54:46.000000000 +0100
+++ bcpg-jdk15on-148.new/src/org/bouncycastle/openpgp/test/PGPUnicodeTest.java	2013-02-16 10:30:53.303969432 +0100
@@ -75,7 +75,7 @@
         {
             BigInteger keyId = new BigInteger("362961283C48132B9F14C5C3EC87272EFCB986D2", 16);
 
-            String passphrase = new String("Händle".getBytes("UTF-16"), "UTF-16");
+            String passphrase = new String("Handle".getBytes("UTF-16"), "UTF-16");
 //			FileInputStream passwordFile = new FileInputStream("testdata/passphrase_for_test.txt");
 //			byte[] password = new byte[passwordFile.available()];
 //			passwordFile.read(password);
