diff -Naur bcpg-jdk15on-149/src/org/bouncycastle/openpgp/test/PGPUnicodeTest.java bcpg-jdk15on-149.new/src/org/bouncycastle/openpgp/test/PGPUnicodeTest.java
--- bcpg-jdk15on-149/src/org/bouncycastle/openpgp/test/PGPUnicodeTest.java	2013-05-31 14:17:50.000000000 +0200
+++ bcpg-jdk15on-149.new/src/org/bouncycastle/openpgp/test/PGPUnicodeTest.java	2013-06-02 09:48:51.178849935 +0200
@@ -75,7 +75,7 @@
         {
             BigInteger keyId = new BigInteger("362961283C48132B9F14C5C3EC87272EFCB986D2", 16);
 
-            String passphrase = new String("Händle".getBytes("UTF-16"), "UTF-16");
+            String passphrase = new String("Handle".getBytes("UTF-16"), "UTF-16");
 //            FileInputStream passwordFile = new FileInputStream("testdata/passphrase_for_test.txt");
 //            byte[] password = new byte[passwordFile.available()];
 //            passwordFile.read(password);
