--- makefile	2007-10-02 17:55:40.000000000 +0530
+++ makefile.new	2007-10-02 17:59:07.000000000 +0530
@@ -2,15 +2,18 @@
 prefix = /usr
 
 install:
+	mkdir -p $(prefix)/bin
 	mkdir -p $(prefix)/share/blueman
 	mkdir -p $(prefix)/share/blueman/icons
 	mkdir -p $(prefix)/share/blueman/icons/devices
 	mkdir -p $(prefix)/share/blueman/icons/signal
+	mkdir -p ${prefix}/share/applications
+	mkdir -p ${prefix}/share/pixmaps
 	install icons/*.* $(prefix)/share/blueman/icons
 	install icons/devices/*.* $(prefix)/share/blueman/icons/devices
 	install icons/signal/*.* $(prefix)/share/blueman/icons/signal
-	install *.desktop /usr/share/applications
-	
+	install *.desktop ${prefix}/share/applications
+
 	mkdir -p $(prefix)/share/blueman/lib
 	
 	install src/*.py $(prefix)/share/blueman/lib/
@@ -18,7 +21,7 @@
 	
 	install src/*.glade $(prefix)/share/blueman/
 	
-	install src/blueman $(prefix)/bin
+	install src/blueman $(prefix)/bin/
 	
 uninstall:
 	-rm -r $(prefix)/share/blueman
