diff -Naur blueman-0.3/src/blueman/Gui/toolbar.py blueman-0.3.new/src/blueman/Gui/toolbar.py
--- blueman-0.3/src/blueman/Gui/toolbar.py	2007-12-04 18:18:00.000000000 +0530
+++ blueman-0.3.new/src/blueman/Gui/toolbar.py	2008-02-24 22:48:32.000000000 +0530
@@ -284,7 +284,7 @@
 			dest = Globals.main_list.get_selected_row()[0]
 
 			#self.set_status(_("Sending file..."))
-			os.system("gnome-obex-send \"" + filename + "\" --dest=" + dest + " &")
+			os.system("gnome-obex-send \"" + filename + "\" --device=" + dest + " &")
 			#os.system("gnome-obex-send \"" + filename + "\" --dest=" + dest + " 2>/dev/null")
 		
 				
