diff -Naur ipw2200-1.0.4.orig/net/ieee80211.h ipw2200-1.0.4/net/ieee80211.h
--- ipw2200-1.0.4.orig/net/ieee80211.h	2005-05-17 22:11:08.000000000 +0200
+++ ipw2200-1.0.4/net/ieee80211.h	2005-06-18 14:12:23.000000000 +0200
@@ -718,11 +718,6 @@
 #define MAC_ARG(x) ((u8*)(x))[0],((u8*)(x))[1],((u8*)(x))[2],((u8*)(x))[3],((u8*)(x))[4],((u8*)(x))[5]
 
 
-extern inline int is_multicast_ether_addr(const u8 *addr)
-{
-	return ((addr[0] != 0xff) && (0x01 & addr[0]));
-}
-
 extern inline int is_broadcast_ether_addr(const u8 *addr)
 {
 	return ((addr[0] == 0xff) && (addr[1] == 0xff) && (addr[2] == 0xff) &&   \
