--- libetonyek-0.1.9/configure.ac	2018-12-29 09:15:33.000000000 +0100
+++ libetonyek-0.1.9.new/configure.ac	2020-02-10 10:30:51.329029742 +0100
@@ -24,7 +24,7 @@
 # Configure options
 # =================
 AC_ARG_WITH([mdds],
-            AS_HELP_STRING([--with-mdds=1.4|1.2|1.0|0.x], [Specify which version of mdds to use (1.2 is the default)]),
+            AS_HELP_STRING([--with-mdds=1.5|1.4|1.2|1.0|0.x], [Specify which version of mdds to use (1.2 is the default)]),
             [], [with_mdds="1.2"])
 
 # ===========================
