diff -Naur bootchart-0.8.orig/script/bootchartd bootchart-0.8/script/bootchartd
--- bootchart-0.8.orig/script/bootchartd	2005-02-08 23:31:01.000000000 +0100
+++ bootchart-0.8/script/bootchartd	2005-07-20 20:28:06.869742256 +0200
@@ -183,6 +183,8 @@
 		echo "system.release = $( /bin/sed q /etc/gentoo-release )" >> "$HEADER"
 	elif [ -f /etc/SuSE-release ]; then
 		echo "system.release = $( /bin/sed q /etc/SuSE-release )" >> "$HEADER"
+	elif [ -f /etc/frugalware-release ]; then
+		echo "system.release = $( /bin/sed q /etc/frugalware-release )" >> "$HEADER"
 	else
 		echo "system.release = $( /bin/sed 's/\\.//g;q' /etc/issue )" >> "$HEADER"
 	fi
