diff -Naur aalib-1.4.0/src/aafire.c aalib-1.4.0-fix/src/aafire.c
--- aalib-1.4.0/src/aafire.c	2001-04-26 17:30:03.000000000 +0200
+++ aalib-1.4.0-fix/src/aafire.c	2015-10-07 17:48:29.426443759 +0200
@@ -1,4 +1,5 @@
 #include <stdio.h>
+#include <stdlib.h>
 #include "aalib.h"
 
 #define XSIZ aa_imgwidth(context)
diff -Naur aalib-1.4.0/src/aainfo.c aalib-1.4.0-fix/src/aainfo.c
--- aalib-1.4.0/src/aainfo.c	2001-04-26 16:37:31.000000000 +0200
+++ aalib-1.4.0-fix/src/aainfo.c	2015-10-07 17:48:29.426443759 +0200
@@ -1,4 +1,4 @@
-
+#include <stdlib.h>
 #include "aalib.h"
 #include "aaint.h"
 int main(int argc, char **argv)
diff -Naur aalib-1.4.0/src/aakbdreg.c aalib-1.4.0-fix/src/aakbdreg.c
--- aalib-1.4.0/src/aakbdreg.c	2001-04-26 16:37:31.000000000 +0200
+++ aalib-1.4.0-fix/src/aakbdreg.c	2015-10-07 17:50:48.165013618 +0200
@@ -1,4 +1,5 @@
 #include <malloc.h>
+#include <string.h>
 #include "config.h"
 #include "aalib.h"
 #include "aaint.h"
diff -Naur aalib-1.4.0/src/aalib.c aalib-1.4.0-fix/src/aalib.c
--- aalib-1.4.0/src/aalib.c	2001-04-26 16:37:31.000000000 +0200
+++ aalib-1.4.0-fix/src/aalib.c	2015-10-07 17:48:29.426443759 +0200
@@ -1,5 +1,6 @@
 #include <stdio.h>
 #include <string.h>
+#include <stdlib.h>
 #include <malloc.h>
 #include "aalib.h"
 #include "aaint.h"
diff -Naur aalib-1.4.0/src/aalinuxkbd.c aalib-1.4.0-fix/src/aalinuxkbd.c
--- aalib-1.4.0/src/aalinuxkbd.c	2001-04-26 16:37:31.000000000 +0200
+++ aalib-1.4.0-fix/src/aalinuxkbd.c	2015-10-07 17:58:20.258644996 +0200
@@ -13,6 +13,7 @@
 #include <sys/types.h>
 #include <unistd.h>
 #include <stdio.h>
+#include <stdlib.h>
 #include <fcntl.h>
 #include <signal.h>
 #include <termios.h>
diff -Naur aalib-1.4.0/src/aamoureg.c aalib-1.4.0-fix/src/aamoureg.c
--- aalib-1.4.0/src/aamoureg.c	2001-04-26 16:37:31.000000000 +0200
+++ aalib-1.4.0-fix/src/aamoureg.c	2015-10-07 17:51:42.346407748 +0200
@@ -1,4 +1,5 @@
 #include <malloc.h>
+#include <string.h>
 #include "config.h"
 #include "aalib.h"
 #include "aaint.h"
diff -Naur aalib-1.4.0/src/aaregist.c aalib-1.4.0-fix/src/aaregist.c
--- aalib-1.4.0/src/aaregist.c	2001-04-26 16:37:31.000000000 +0200
+++ aalib-1.4.0-fix/src/aaregist.c	2015-10-07 17:50:09.546019920 +0200
@@ -1,4 +1,5 @@
 #include <malloc.h>
+#include <string.h>
 #include "config.h"
 #include "aalib.h"
 #include "aaint.h"
diff -Naur aalib-1.4.0/src/aasavefont.c aalib-1.4.0-fix/src/aasavefont.c
--- aalib-1.4.0/src/aasavefont.c	2001-04-26 16:37:31.000000000 +0200
+++ aalib-1.4.0-fix/src/aasavefont.c	2015-10-07 17:48:29.426443759 +0200
@@ -1,3 +1,4 @@
+#include <stdlib.h>
 #include "aalib.h"
 int main(int argc, char **argv)
 {
diff -Naur aalib-1.4.0/src/aatest.c aalib-1.4.0-fix/src/aatest.c
--- aalib-1.4.0/src/aatest.c	2001-04-26 16:37:31.000000000 +0200
+++ aalib-1.4.0-fix/src/aatest.c	2015-10-07 17:54:00.349958683 +0200
@@ -1,3 +1,5 @@
+#include <stdlib.h>
+#include <string.h>
 #include "aalib.h"
 int main(int argc, char **argv)
 {
