From 8c6ed05317169b45574ab47a5ff9ac772d1ac4e4 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Mon, 28 Aug 2023 16:40:32 +0200
Subject: [PATCH] * don't refuse usr/share/pkgconfig

---
 repoman | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/repoman b/repoman
index 2c6a603..1b398d5 100755
--- a/repoman
+++ b/repoman
@@ -411,12 +411,6 @@ upload()
                         echo "Please fix your package."
                         exit 1
 		fi
-		if echo $src | grep -q fpm$ && pacman-g2 -Ql -p $src | grep -q 'usr/share/pkgconfig'; then
-                        echo "$src has invalid pkgconfig path , files are installed"
-                        echo "in /usr/share/pkgconfig , should be /usr/lib/pkgconfig, refusing to upload!"
-                        echo "Please fix your package."
-                        exit 1
-		fi
 		if echo $src | grep -q fpm$ && pacman-g2 -Ql -p $src | grep -q 'usr/share/perl5'; then
                         echo "$src has invalid perl path , files are installed"
                         echo "in /usr/share/perl5 , refusing to upload!"
-- 
2.41.0

