From e89d1ed4fef6a10aa2e311f1bfa19982fd59c513 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Tue, 13 Aug 2024 20:20:14 +0200
Subject: [PATCH] * allow syncfs

---
 fblint | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fblint b/fblint
index 8c661ed..3a32afb 100755
--- a/fblint
+++ b/fblint
@@ -134,7 +134,7 @@ check "groups" "[ ! -z \"$groups\" ]"
 # with it
 check "empty option" "[ -z \"$option\" ]"
 check "valid first group" "! echo $groups |grep -q -- - || echo $groups|grep -q -- -extra$"
-check "valid options()" "[ -z \"$(echo ${options[@]}|tr ' ' '\n' |grep -v '\(clang\|ldlld\|ldbfd\|ldgold\|ldefs\|nowfsecurity\|noclash\|ldlld\|ldbfd\|ofaster\|ofast\|omedium\|osmall\|odebug\|nodevel\|nolto\|nopie\|plt\|nonow\|noprotector\|norelro\|nofortify\|libtool\|static\|nodocs\|nostrip\|force\|nobuild\|nofakeroot\|scriptlet\|stick\|nomirror\|noversrc\|noccache\|asneeded\|noasneeded\)')\" ]"
+check "valid options()" "[ -z \"$(echo ${options[@]}|tr ' ' '\n' |grep -v '\(clang\|ldlld\|ldbfd\|ldgold\|ldefs\|nowfsecurity\|noclash\|ldlld\|ldbfd\|ofaster\|ofast\|omedium\|osmall\|odebug\|nodevel\|nolto\|nopie\|plt\|nonow\|noprotector\|norelro\|nofortify\|libtool\|static\|nodocs\|nostrip\|force\|nobuild\|nofakeroot\|scriptlet\|stick\|nomirror\|noversrc\|noccache\|asneeded\|noasneeded\|syncfs\)')\" ]"
 check "archs" "[ ! -z \"$archs\" ]"
 check "up2date" "test -n \"`echo $up2date|sed 's/\$(/\\\\\$(/g'`\""
 _fblint_have_remote_source=n
-- 
2.46.0

