From 3e0aa4da801e504532a28b15a662c7fdfda2dc04 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Wed, 14 Aug 2024 11:08:37 +0200
Subject: [PATCH] * fix command

---
 repoman | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/repoman b/repoman
index 63f37b7..52afe48 100755
--- a/repoman
+++ b/repoman
@@ -172,7 +172,7 @@ server()
 
 		upload)
 		shift
-		if [ $# -lt 4 ]; then
+		if [ $# -lt 3 ]; then
 			echo "Too few parameters!"
 			echo "usage: repoman server upload path yourtmpfile targetfile"
 			return 1
@@ -419,7 +419,10 @@ upload()
 		[ -z "$tmp" ] && exit 1
 		msg "Uploading file: $src (to $tmp)"
 		[[ -z "$repoman_port" ]] && repoman_port="22"
-        $repoman_scp -P$repoman_port $src $host:$tmp
+		$repoman_scp -P$repoman_port $src $host:$tmp
+		if ! _ssh "upload '$path' '$tmp' '$dest/$src'"; then
+			return 1
+		fi
 	done
 }
 
-- 
2.46.0

