diff -Naur pacman-tools-1.2.6/repoman pacman-tools-1.2.6/repoman
--- pacman-tools-1.2.6/repoman	2011-02-27 10:28:26.000000000 +0100
+++ pacman-tools-1.2.6/repoman	2011-03-07 23:09:03.065152768 +0100
@@ -936,7 +936,13 @@
 
 # Initializing variables.
 version='0.3.1'
-[ -z "$arch" ] && arch=`uname -m`
+if [ -z "$arch" ]; then
+	if [[ "`uname -m`" == arm* ]]; then
+		arch="arm"
+	else
+		arch="`uname -m`"
+    fi
+fi
 mktemp_local="mktemp -p $HOME"
 sudo_env="sudo -E"
 unset LANG LC_ALL
