diff -Naur pacman-tools-1.2.6/repoman pacman-tools-1.2.6new/repoman
--- pacman-tools-1.2.6/repoman	2011-02-27 10:28:26.000000000 +0100
+++ pacman-tools-1.2.6new/repoman	2011-03-07 23:09:03.065152768 +0100
@@ -380,7 +380,7 @@
 			## using 127.0.0.1 .. scp -S really sucks and I get a lot problems
 			## with rsync -e with all this tunneling =)
 			## the other solution may be to force host when tunnel is used to be user@127.0.0.1
-			## since -p <port> user@local gives out user@genesis .. 
+			## since -p <port> user@local gives out user@genesis ..
 			msg "Uploading file: $src (to $tmp) over local tunnel."
                         $repoman_scp $src 127.0.0.1:$tmp
                 else
@@ -936,7 +936,13 @@
 
 # Initializing variables.
 version='0.3.1'
-[ -z "$arch" ] && arch=`uname -m`
+if [ -z "$arch" ]; then
+	if [[ "`uname -m`" == arm* ]]; then
+		arch="arm"
+	else
+		arch="`uname -m`"
+    fi
+fi
 mktemp_local="mktemp -p $HOME"
 sudo_env="sudo -E"
 unset LANG LC_ALL
