diff --git a/repoman.conf b/repoman.conf
index 35d41f6..37ca7b2 100644
--- a/repoman.conf
+++ b/repoman.conf
@@ -7,8 +7,8 @@
 
 # the order is important, the first repo will be used as a default for
 # developer actions (like up, del and so on)
-source /etc/repoman.d/current
-#source /etc/repoman.d/stable
+#source /etc/repoman.d/current
+source /etc/repoman.d/stable
 
 # default repo to search the include dir in
 fst_root=/var/fst/
