diff -ur a/repoman.conf b/repoman.conf
--- a/repoman.conf	2012-10-10 01:09:41.000000000 -0500
+++ b/repoman.conf	2013-02-02 03:57:31.884598372 -0600
@@ -7,8 +7,8 @@
 
 # the order is important, the first repo will be used as a default for
 # developer actions (like up, del and so on)
-source /etc/repoman.d/current
-#source /etc/repoman.d/stable
+#source /etc/repoman.d/current
+source /etc/repoman.d/stable
 
 # default repo to search the include dir in
 fst_root=/var/fst/
