From 6e66828c8f604d1285955e237ccba39ac8db9f25 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Wed, 14 Aug 2024 10:41:52 +0200
Subject: [PATCH] * stop using sha1sum after transfer: not needed

---
 repoman | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/repoman b/repoman
index 1b398d5..63f37b7 100755
--- a/repoman
+++ b/repoman
@@ -174,19 +174,14 @@ server()
 		shift
 		if [ $# -lt 4 ]; then
 			echo "Too few parameters!"
-			echo "usage: repoman server upload path yourtmpfile sha1 targetfile"
+			echo "usage: repoman server upload path yourtmpfile targetfile"
 			return 1
 		fi
-		local path="$1" tmp="$2" sha="$3" dest="$4"
-		shift 4
+		local path="$1" tmp="$2" dest="$3"
+		shift 3
 		check_absolute_path
 		cd $path
 		get_lock || return 1
-		if ! echo "$sha  $tmp" |sha1sum -c - >/dev/null 2>&1; then
-			echo "The uploaded file is corrupted :-/"
-			rm $tmp
-			return 1
-		fi
 		# strip leading / and any ..
 		if ! mv -f $tmp "`echo $dest|sed 's|^/||;s|\.\./||g'`"; then
 			echo "Failed to move the uploaded file"
@@ -425,10 +420,6 @@ upload()
 		msg "Uploading file: $src (to $tmp)"
 		[[ -z "$repoman_port" ]] && repoman_port="22"
         $repoman_scp -P$repoman_port $src $host:$tmp
-		sha=`sha1sum $src|sed 's/\(.*\)  .*/\1/'`
-		if ! _ssh "upload '$path' '$tmp' '$sha' '$dest/$src'"; then
-			return 1
-		fi
 	done
 }
 
-- 
2.46.0

