--- pacman-tools-1.3.5/syncpkgd/Makefile	2023-01-27 14:06:20.000000000 +0100
+++ pacman-tools-1.3.5.new/syncpkgd/Makefile	2024-01-03 16:57:07.563445673 +0100
@@ -4,9 +4,9 @@
 	mkdir -p  $(DESTDIR)/etc/syncpkgd
 	install -g 2 -o 12 -m600 dconfig.py $(DESTDIR)/etc/syncpkgd/dconfig.py
 	install -m644 ctlconfig.py $(DESTDIR)/etc/syncpkgd/ctlconfig.py
-	mkdir -p  $(DESTDIR)/lib/systemd/system
-	install -m644 syncpkgcd.service $(DESTDIR)/lib/systemd/system/syncpkgcd.service
-	install -m644 syncpkgd.service $(DESTDIR)/lib/systemd/system/syncpkgd.service
+	mkdir -p  $(DESTDIR)/usr/lib/systemd/system
+	install -m644 syncpkgcd.service $(DESTDIR)/usr/lib/systemd/system/syncpkgcd.service
+	install -m644 syncpkgd.service $(DESTDIR)/usr/lib/systemd/system/syncpkgd.service
 	mkdir -p  $(DESTDIR)/usr/bin
 	install syncpkgcd.py $(DESTDIR)/usr/bin/syncpkgcd
 	install syncpkgdctl.py $(DESTDIR)/usr/bin/syncpkgdctl
