--- base64-0.4.2.4/base64.cabal	2001-09-09 03:46:40.000000000 +0200
+++ base64-0.4.2.4.new/base64.cabal	2023-12-12 15:14:23.891572397 +0100
@@ -51,10 +51,10 @@
     Data.ByteString.Base64.Internal.W64.Loop
 
   build-depends:
-      base           >=4.14     && <4.17
-    , bytestring     >=0.10     && <0.12
-    , deepseq        >=1.4.3.0  && <1.5
-    , text           ^>=1.2
+      base           >=4.14     && <4.20
+    , bytestring     >=0.10     && <0.13
+    , deepseq        >=1.4.3.0  && <1.6
+    , text           >=1.5
     , text-short     ^>=0.1
 
   hs-source-dirs:   src
