--- base64-0.4.2.4/base64.cabal	2001-09-09 03:46:40.000000000 +0200
+++ base64-0.4.2.4.new/base64.cabal	2022-11-15 15:51:41.742059138 +0100
@@ -51,10 +51,10 @@
     Data.ByteString.Base64.Internal.W64.Loop
 
   build-depends:
-      base           >=4.14     && <4.17
+      base           >=4.14     && <4.18
     , bytestring     >=0.10     && <0.12
     , deepseq        >=1.4.3.0  && <1.5
-    , text           ^>=1.2
+    , text            >=1.2 && <1.3 || >=2.0 && <2.1
     , text-short     ^>=0.1
 
   hs-source-dirs:   src
@@ -68,7 +68,7 @@
   other-modules:    Internal
   main-is:          Main.hs
   build-depends:
-      base               >=4.14 && <4.17
+      base               >=4.14 && <4.18
     , base64
     , base64-bytestring
     , bytestring
@@ -88,7 +88,7 @@
   hs-source-dirs:   benchmarks
   main-is:          Base64Bench.hs
   build-depends:
-      base               >=4.14 && <4.17
+      base               >=4.14 && <4.18
     , base64
     , base64-bytestring
     , bytestring
