--- base64-1.0/base64.cabal	2001-09-09 03:46:40.000000000 +0200
+++ base64-1.0.new/base64.cabal	2024-07-10 22:15:57.287897111 +0200
@@ -20,7 +20,7 @@
   README.md
   MIGRATION-1.0.md
 
-tested-with:     GHC ==8.10.7 || ==9.0.2 || ==9.2.5 || ==9.4.7 || ==9.6.3 || ==9.8.1
+tested-with:     GHC ==8.10.7 || ==9.0.2 || ==9.2.8 || ==9.4.8 || ==9.6.5 || ==9.8.2 || ==9.10.1
 
 source-repository head
   type:     git
@@ -54,7 +54,7 @@
     Data.ByteString.Base64.Internal.W64.Loop
 
   build-depends:
-      base           >=4.14     && <4.20
+      base           >=4.14     && <4.22
     , bytestring     >=0.11     && <0.13
     , deepseq        >=1.4.4.0  && <1.6
     , text           >=2.0      && <2.3
@@ -71,7 +71,7 @@
   other-modules:    Internal
   main-is:          Main.hs
   build-depends:
-      base               >=4.14 && <4.20
+      base               >=4.14 && <4.22
     , base64
     , base64-bytestring
     , bytestring         >=0.11
@@ -91,7 +91,7 @@
   hs-source-dirs:   benchmarks
   main-is:          Base64Bench.hs
   build-depends:
-      base               >=4.14 && <4.20
+      base               >=4.14 && <4.22
     , base64
     , base64-bytestring
     , bytestring         >=0.11
