diff -aur commoncpp2-1.5.6/include/config.h.in fw-commoncpp2-1.5.6/include/config.h.in
--- commoncpp2-1.5.6/include/config.h.in	2007-05-26 14:17:39.000000000 +0200
+++ fw-commoncpp2-1.5.6/include/config.h.in	2007-05-26 15:17:04.000000000 +0200
@@ -1082,9 +1082,17 @@
 #endif
 #endif
 
+#define GCC_VERSION (__GNUC__ * 10000 \
+		+ __GNUC_MINOR__ * 100 \
+		+ __GNUC_PATCHLEVEL__)
+
 #if defined(__cplusplus)
 #if defined(HAVE_GCC_BITS_ATOMIC) || defined(HAVE_GCC_CXX_BITS_ATOMIC)
-#include <bits/atomicity.h>
+#if GCC_VERSION >= 40200
+ #include <ext/atomicity.h>
+#else
+ #include <bits/atomicity.h>
+#endif
 #define HAVE_ATOMIC
 #endif
 #endif
