diff -Naur i3lock-2.5/i3lock.c i3lock-2.5b/i3lock.c
--- i3lock-2.5/i3lock.c	2013-10-21 18:26:28.000000000 +0200
+++ i3lock-2.5b/i3lock.c	2013-06-09 11:49:00.000000000 +0200
@@ -20,7 +20,7 @@
 #include <X11/Xlib-xcb.h>
 #include <getopt.h>
 #include <string.h>
-#include <ev.h>
+#include <libev/ev.h>
 #include <sys/mman.h>
 #include <X11/XKBlib.h>
 #include <X11/extensions/XKBfile.h>
diff -Naur i3lock-2.5/unlock_indicator.c i3lock-2.5b/unlock_indicator.c
--- i3lock-2.5/unlock_indicator.c	2013-10-21 18:26:06.000000000 +0200
+++ i3lock-2.5b/unlock_indicator.c	2013-06-09 11:49:00.000000000 +0200
@@ -10,7 +10,7 @@
 #include <stdlib.h>
 #include <math.h>
 #include <xcb/xcb.h>
-#include <ev.h>
+#include <libev/ev.h>
 #include <cairo.h>
 #include <cairo/cairo-xcb.h>


