--- gsettings-desktop-schemas-0.1.7/schemas/org.gnome.desktop.background.gschema.xml.in.in.old	2011-02-25 22:42:12.000000000 +0100
+++ gsettings-desktop-schemas-0.1.7/schemas/org.gnome.desktop.background.gschema.xml.in.in	2011-02-25 22:42:58.000000000 +0100
@@ -15,7 +15,7 @@
       </_description>
     </key>
     <key name="picture-uri" type="s">
-      <default>'file://@datadir@/themes/Adwaita/backgrounds/adwaita-timed.xml'</default>
+      <default>'file://@datadir@/pixmaps/backgrounds/gnome/frugalware/frugalware.png'</default>
       <_summary>Picture URI</_summary>
       <_description>
         URI to use for the background image. Not that the backend only supports
