--- schemas/org.gnome.desktop.wm.preferences.gschema.xml.in
+++ schemas/org.gnome.desktop.wm.preferences.gschema.xml.in.fw
@@ -25,7 +25,7 @@
        </description>
     </key>
     <key name="button-layout" type="s">
-      <default>'appmenu:close'</default>
+      <default>'appmenu:minimize,maximize,close'</default>
       <summary>Arrangement of buttons on the titlebar</summary>
       <description>
         Arrangement of buttons on the titlebar. The value should be a string,

