--- gsettings-desktop-schemas-46.0/schemas/org.gnome.desktop.background.gschema.xml.in	2024-03-16 22:59:12.000000000 +0100
+++ gsettings-desktop-schemas-46.0.new/schemas/org.gnome.desktop.background.gschema.xml.in	2024-06-12 17:25:01.878911774 +0200
@@ -57,7 +57,7 @@
       </description>
     </key>
     <key name="show-desktop-icons" type="b">
-      <default>false</default>
+      <default>true</default>
       <summary>Have file manager handle the desktop</summary>
       <description>
         If set to true, then file manager will draw the icons on the desktop.
