--- schemas/org.gnome.desktop.background.gschema.xml.in
+++ schemas/org.gnome.desktop.background.gschema.xml.in.good
@@ -48,7 +48,7 @@
       </description>
     </key>
     <key name="show-desktop-icons" type="b">
-      <default>false</default>
+      <default>true</default>
       <summary>Have file manager handle the desktop</summary>
       <description>
         If set to true, then file manager will draw the icons on the desktop.
