--- gsettings-desktop-schemas-0.1.7/schemas/org.gnome.desktop.screensaver.gschema.xml.in.in.old	2011-02-25 22:51:26.000000000 +0100
+++ gsettings-desktop-schemas-0.1.7/schemas/org.gnome.desktop.screensaver.gschema.xml.in.in	2011-02-25 22:53:05.000000000 +0100
@@ -11,12 +11,12 @@
       <_description>Set this to TRUE to lock the screen when the screensaver goes active.</_description>
     </key>
     <key name="mode" enum="org.gnome.desktop.GDesktopScreensaverMode">
-      <default>'blank-only'</default>
+      <default>'single'</default>
       <_summary>Screensaver theme selection mode</_summary>
       <_description>The selection mode used by screensaver.  May be "blank-only" to enable the screensaver without using any theme on activation, "single" to enable screensaver using only one theme on activation (specified in "themes" key), and "random" to enable the screensaver using a random theme on activation.</_description>
     </key>
     <key name="themes" type="as">
-      <default>[]</default>
+      <default>['screensavers-frugal-floaters']</default>
       <_summary>Screensaver themes</_summary>
       <_description>This key specifies the list of themes to be used by the screensaver. It's ignored when "mode" key is  "blank-only", should provide the theme name when "mode" is "single", and should provide a list of themes when "mode" is "random".</_description>
     </key>
