--- gsettings-desktop-schemas-2.91.92/schemas/org.gnome.desktop.background.gschema.xml.in.in.old	2011-03-21 15:42:04.000000000 +0100
+++ gsettings-desktop-schemas-2.91.92/schemas/org.gnome.desktop.background.gschema.xml.in.in	2011-03-22 12:34:08.000000000 +0100
@@ -22,6 +22,11 @@
         local (file://) URIs.
       </_description>
     </key>
+    <key type="s" name="picture-filename">
+      <default>'/usr/share/pixmaps/backgrounds/gnome/frugalware/frugalware.png'</default>
+      <summary>Picture Filename</summary>
+      <description>File to use for the background image.</description>
+    </key>
     <key name="picture-opacity" type="i">
       <range min="0" max="100"/>
       <default>100</default>
