diff -Naur fish-1.20.1.orig/init/fish_function.fish fish-1.20.1/init/fish_function.fish
--- fish-1.20.1.orig/init/fish_function.fish	2006-01-26 17:30:03.000000000 +0100
+++ fish-1.20.1/init/fish_function.fish	2006-02-02 14:05:31.000000000 +0100
@@ -116,7 +116,7 @@
 
 		# If we are in a graphical environment, check if there is a graphical
 		# browser to use instead.
-		if test (echo $DISPLAY) -a \( "$XAUTHORITY" = "$HOME/.Xauthority" -o "$XAUTHORITY" = "" \)
+		if test (echo $WINDOWID) -a \( "$LOGNAME" = "$USER" \)
 			for i in $graphical_browsers
 				if which $i 2>/dev/null >/dev/null
 					set fish_browser $i
