diff -aur 2006.1.5.3/src/MDIText.cpp 2006.1.5.3-2/src/MDIText.cpp
--- 2006.1.5.3/src/MDIText.cpp	2006-10-20 16:41:00.000000000 -0500
+++ 2006.1.5.3-2/src/MDIText.cpp	2010-05-15 02:57:57.000000000 -0500
@@ -19,7 +19,7 @@
 {
 	setObjectName( "MDIText" );
 	setType( ctText );
-	setWindowIcon( QIcon::QIcon( ":/Icons/Icons/projectshowfile.png" ) );
+	setWindowIcon( QIcon( ":/Icons/Icons/projectshowfile.png" ) );
 	//
 	QHBoxLayout* hboxLayout = new QHBoxLayout;
 	hboxLayout->setObjectName( "hboxLayout" );
