diff -Naur 2006.1.5.3/src/3rdparty/TextEditor/src/CppSyntaxHighlighter.h 2006.1.5.3-p/src/3rdparty/TextEditor/src/CppSyntaxHighlighter.h
--- 2006.1.5.3/src/3rdparty/TextEditor/src/CppSyntaxHighlighter.h	2006-10-20 23:40:58.000000000 +0200
+++ 2006.1.5.3-p/src/3rdparty/TextEditor/src/CppSyntaxHighlighter.h	2007-11-07 18:24:43.000000000 +0100
@@ -39,6 +39,7 @@
 	};
 	static const Keyword keywords[];
 	//
+	Q_OBJECT
 private:
 	void highlightKeyword( int, const QString& );
 	//
