diff -Naur netkit-telnet-0.17.orig/telnet/Makefile netkit-telnet-0.17/telnet/Makefile
--- netkit-telnet-0.17.orig/telnet/Makefile	2005-09-04 12:36:58.000000000 +0200
+++ netkit-telnet-0.17/telnet/Makefile	2005-09-04 12:37:35.000000000 +0200
@@ -6,7 +6,7 @@
 #CXXFLAGS:=$(patsubst -O2, -g, $(CXXFLAGS))
 
 # -DAUTHENTICATE
-CXXFLAGS += -DUSE_TERMIO -DKLUDGELINEMODE
+CXXFLAGS += -DUSE_TERMIO -DKLUDGELINEMODE -D_GNU_SOURCE
 LIBS = $(LIBTERMCAP)
 
 SRCS = commands.cc main.cc network.cc ring.cc sys_bsd.cc telnet.cc \
diff -Naur netkit-telnet-0.17.orig/telnetd/Makefile netkit-telnet-0.17/telnetd/Makefile
--- netkit-telnet-0.17.orig/telnetd/Makefile	2005-09-04 12:36:58.000000000 +0200
+++ netkit-telnet-0.17/telnetd/Makefile	2005-09-04 12:38:37.000000000 +0200
@@ -10,7 +10,7 @@
 
 CFLAGS += '-DISSUE_FILE="/etc/issue.net"' -DPARANOID_TTYS \
 	   -DNO_REVOKE -DKLUDGELINEMODE -DDIAGNOSTICS \
-	   -DLOGIN_WRAPPER=\"/usr/lib/telnetlogin\"
+	   -DLOGIN_WRAPPER=\"/usr/lib/telnetlogin\" -D_GNU_SOURCE
 # LIBS += $(LIBTERMCAP)
 
 OBJS = telnetd.o state.o termstat.o slc.o sys_term.o utility.o \
