diff -Naur netkit-telnet-0.17.orig/configure netkit-telnet-0.17/configure
--- netkit-telnet-0.17.orig/configure	2005-06-02 16:22:59.000000000 +0000
+++ netkit-telnet-0.17/configure	2005-06-02 16:25:06.000000000 +0000
@@ -554,14 +554,14 @@
     echo "CC=$CC"
     echo "CXX=$CXX"
     if [ x$CC_WARNINGS != x ]; then
-        CFLAGS="$CFLAGS $WARNINGS"
+        CFLAGS="$CFLAGS -trigraphs $WARNINGS"
     fi
 
     if [ x$CXX_WARNINGS != x ]; then
         CXXFLAGS="$CXXFLAGS $WARNINGS"
     fi
 
-    echo "CFLAGS=$CFLAGS" | sed 's/= */=/'
+    echo "CFLAGS=$CFLAGS -trigraphs" | sed 's/= */=/'
     echo "CXXFLAGS=$CXXFLAGS" | sed 's/= */=/'
     echo "LDFLAGS=$LDFLAGS" | sed 's/= */=/'
     echo "LIBS=$LIBS" | sed 's/= */=/'
diff -Naur netkit-telnet-0.17.orig/telnet/main.cc netkit-telnet-0.17/telnet/main.cc
--- netkit-telnet-0.17.orig/telnet/main.cc	2005-06-02 16:22:59.000000000 +0000
+++ netkit-telnet-0.17/telnet/main.cc	2005-06-02 16:23:07.000000000 +0000
@@ -46,6 +46,7 @@
 #include <sys/types.h>
 #include <getopt.h>
 #include <string.h>
+#include <stdlib.h>
 
 #include "ring.h"
 #include "externs.h"
diff -Naur netkit-telnet-0.17.orig/telnet/netlink.cc netkit-telnet-0.17/telnet/netlink.cc
--- netkit-telnet-0.17.orig/telnet/netlink.cc	2005-06-02 16:22:59.000000000 +0000
+++ netkit-telnet-0.17/telnet/netlink.cc	2005-06-02 16:23:07.000000000 +0000
@@ -8,6 +8,7 @@
 #include <sys/ioctl.h>
 #include <sys/time.h>
 #include <netdb.h>
+#include <string.h>
 #include "netlink.h"
 #include "proto.h"
 #include "ring.h"
diff -Naur netkit-telnet-0.17.orig/telnet/network.cc netkit-telnet-0.17/telnet/network.cc
--- netkit-telnet-0.17.orig/telnet/network.cc	2005-06-02 16:22:59.000000000 +0000
+++ netkit-telnet-0.17/telnet/network.cc	2005-06-02 16:23:07.000000000 +0000
@@ -42,6 +42,7 @@
 #include <sys/time.h>
 #include <errno.h>
 #include <arpa/telnet.h>
+#include <stdlib.h>
 
 #include "ring.h"
 #include "defines.h"
diff -Naur netkit-telnet-0.17.orig/telnet/terminal.cc netkit-telnet-0.17/telnet/terminal.cc
--- netkit-telnet-0.17.orig/telnet/terminal.cc	2005-06-02 16:22:59.000000000 +0000
+++ netkit-telnet-0.17/telnet/terminal.cc	2005-06-02 16:23:07.000000000 +0000
@@ -45,6 +45,8 @@
 #include <signal.h>
 #include <errno.h>
 #include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
 
 #include "ring.h"
 #include "defines.h"
diff -Naur netkit-telnet-0.17.orig/telnet/utilities.cc netkit-telnet-0.17/telnet/utilities.cc
--- netkit-telnet-0.17.orig/telnet/utilities.cc	2005-06-02 16:22:59.000000000 +0000
+++ netkit-telnet-0.17/telnet/utilities.cc	2005-06-02 16:23:07.000000000 +0000
@@ -47,6 +47,9 @@
 #include <sys/socket.h>
 #include <unistd.h>
 #include <ctype.h>
+#include <time.h>
+#include <stdlib.h>
+#include <string.h>
 
 #include "ring.h"
 #include "defines.h"
diff -Naur netkit-telnet-0.17.orig/telnetd/utility.c netkit-telnet-0.17/telnetd/utility.c
--- netkit-telnet-0.17.orig/telnetd/utility.c	2005-06-02 16:22:59.000000000 +0000
+++ netkit-telnet-0.17/telnetd/utility.c	2005-06-02 16:23:53.000000000 +0000
@@ -41,6 +41,7 @@
 
 #include <stdarg.h>
 #include <sys/utsname.h>
+#include <time.h>
 
 #ifdef AUTHENTICATE
 #include <libtelnet/auth.h>
