diff -Naur krita-3.1.4/libs/ui/KisResourceBundleManifest.cpp krita-3.1.4-p/libs/ui/KisResourceBundleManifest.cpp
--- krita-3.1.4/libs/ui/KisResourceBundleManifest.cpp	2017-05-26 08:47:55.000000000 +0200
+++ krita-3.1.4-p/libs/ui/KisResourceBundleManifest.cpp	2017-06-17 18:21:47.225792422 +0200
@@ -22,7 +22,9 @@
 #include <QDomElement>
 #include <QDomNode>
 #include <QDomNodeList>
-#include <QForeachContainer>
+#if QT_VERSION < 0x050900
+ #include <QForeachContainer>
+#endif
 
 #include <KoXmlNS.h>
 #include <KoXmlReader.h>
