--- clr-rocm-6.4.0/hipamd/src/CMakeLists.txt	2025-04-02 17:04:37.000000000 +0200
+++ clr-rocm-6.4.0.new/hipamd/src/CMakeLists.txt	2025-04-22 10:48:26.485370887 +0200
@@ -49,9 +49,9 @@
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCC)
-  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
   set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-error=deprecated-declarations")
-  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-error=deprecated-declarations")
 endif()
 
