diff -Naur suxpanel-0.4a.orig/gtkmisc.c suxpanel-0.4a/gtkmisc.c
--- suxpanel-0.4a.orig/gtkmisc.c	2006-03-05 05:26:50.000000000 +0100
+++ suxpanel-0.4a/gtkmisc.c	2006-03-05 05:27:53.000000000 +0100
@@ -49,11 +49,11 @@
 gtkm_widget_set_cursor(GtkWidget *widget, GdkCursorType cursor_type)
 {
         GdkCursor *cursor;
-        
+
         cursor = gdk_cursor_new(cursor_type);
         gdk_window_set_cursor(GDK_WINDOW(widget->window), cursor);
-        g_free(cursor);
-        
+        gdk_cursor_unref(cursor);
+
         while(gtk_events_pending())
-                gtk_main_iteration();        
+                gtk_main_iteration();
 }
