--- jxrlib-0.2.4/libjxr.pc.in	2020-05-11 11:07:07.000000000 +0200
+++ jxrlib-0.2.4.new/libjxr.pc.in	2022-05-12 16:36:28.171142071 +0200
@@ -1,4 +1,4 @@
-prefix=%(DIR_INSTALL)s
+prefix=%(PREFIX)s
 exec_prefix=${prefix}
 libdir=${exec_prefix}/lib
 includedir=${prefix}/include
--- jxrlib-0.2.4/Makefile	2020-05-11 11:07:07.000000000 +0200
+++ jxrlib-0.2.4.new/Makefile	2022-05-12 16:40:39.755076830 +0200
@@ -305,7 +305,7 @@
 	rm -rf $(DIR_BUILD)/*App $(DIR_BUILD)/**/*.o $(DIR_BUILD)/**/*.class $(DIR_BUILD)/libj*.a $(DIR_BUILD)/libj*.$(LIBSUFFIX) $(DIR_BUILD)/libjxr.pc $(DIR_BUILD)/$(CXXDECAPP) $(DIR_BUILD)/$(JAR)
 
 $(DIR_BUILD)/libjxr.pc: $(DIR_SRC)/libjxr.pc.in
-	@python -c 'import os; d = { "DIR_INSTALL": "$(DIR_INSTALL)", "JXR_VERSION": "$(JXR_VERSION)", "JXR_ENDIAN": "$(ENDIANFLAG)" }; fin = open("$<", "r"); fout = open("$@", "w+"); fout.writelines( [ l % d for l in fin.readlines()])'
+	@python -c 'import os; d = { "PREFIX": "$(PREFIX)", "JXR_VERSION": "$(JXR_VERSION)", "JXR_ENDIAN": "$(ENDIANFLAG)" }; fin = open("$<", "r"); fout = open("$@", "w+"); fout.writelines( [ l % d for l in fin.readlines()])'
 
 install: all $(DIR_BUILD)/libjxr.pc
 	install -d $(DIR_INSTALL)/lib/pkgconfig $(DIR_INSTALL)/bin $(DIR_INSTALL)/include/libjxr/common  $(DIR_INSTALL)/include/libjxr/image/x86 $(DIR_INSTALL)/include/libjxr/glue $(DIR_INSTALL)/include/libjxr/test $(DIR_INSTALL)/share/doc/jxr-$(JXR_VERSION)
