From 43071845a28e5b2e7aa105956bf26762b979fa1e Mon Sep 17 00:00:00 2001
From: l0drex <l0drex@pm.me>
Date: Wed, 9 Nov 2022 16:35:29 +0100
Subject: [PATCH] Fix import

---
 src/yin_yang.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/yin_yang.py b/src/yin_yang.py
index 4a93fea..7106077 100755
--- a/src/yin_yang.py
+++ b/src/yin_yang.py
@@ -13,7 +13,7 @@
 import time
 from threading import Thread
 
-from daemon_handler import update_times
+from src.daemon_handler import update_times
 from src.meta import PluginKey
 from src.config import config, plugins
 
