--- Yin-Yang-3.2.4/src/plugins/vscode.py	2023-05-12 17:02:49.000000000 +0200
+++ Yin-Yang-3.2.4.new/src/plugins/vscode.py	2023-05-18 19:13:40.485448884 +0200
@@ -16,6 +16,7 @@
     '/usr/lib/code-insiders/extensions',
     '/usr/share/code/resources/app/extensions',
     '/usr/share/code-insiders/resources/app/extensions',
+    '/usr/share/vscode/resources/app/extensions',
     '/opt/visual-studio-code/resources/app/extensions/',
     '/opt/visual-studio-code-insiders/resources/app/extensions/',
     '/var/lib/snapd/snap/code/current/usr/share/code/resources/app/extensions/',
