--- Yin-Yang-3.2.1/src/plugins/vscode.py	2022-11-09 16:32:25.000000000 +0100
+++ Yin-Yang-3.2.1.new/src/plugins/vscode.py	2022-12-29 09:15:33.899058568 +0100
@@ -13,6 +13,7 @@
     str(Path.home()) + '/.vscode-oss/extensions',
     '/usr/lib/code/extensions',
     '/usr/share/code/resources/app/extensions',
+    '/usr/share/vscode/resources/app/extensions',
     '/opt/visual-studio-code/resources/app/extensions/',
     '/var/lib/snapd/snap/code/current/usr/share/code/resources/app/extensions/'
 ]
