diff -ur a/pycparser/c_lexer.py b/pycparser/c_lexer.py
--- a/pycparser/c_lexer.py	2013-08-03 08:15:32.000000000 -0500
+++ b/pycparser/c_lexer.py	2014-03-10 15:30:29.114690762 -0500
@@ -9,8 +9,8 @@
 import re
 import sys
 
-from .ply import lex
-from .ply.lex import TOKEN
+from ply import lex
+from ply.lex import TOKEN
 
 
 class CLexer(object):
diff -ur a/pycparser/c_parser.py b/pycparser/c_parser.py
--- a/pycparser/c_parser.py	2013-08-03 08:15:32.000000000 -0500
+++ b/pycparser/c_parser.py	2014-03-10 15:30:44.805690185 -0500
@@ -8,7 +8,7 @@
 #------------------------------------------------------------------------------
 import re
 
-from .ply import yacc
+from ply import yacc
 
 from . import c_ast
 from .c_lexer import CLexer
