diff -Naur pycparser-2.21/pycparser/c_lexer.py pycparser-2.21.new/pycparser/c_lexer.py
--- pycparser-2.21/pycparser/c_lexer.py	2021-10-20 20:50:33.000000000 +0200
+++ pycparser-2.21.new/pycparser/c_lexer.py	2022-02-15 13:30:37.352657763 +0100
@@ -8,8 +8,8 @@
 #------------------------------------------------------------------------------
 import re
 
-from .ply import lex
-from .ply.lex import TOKEN
+from ply import lex
+from ply.lex import TOKEN
 
 
 class CLexer(object):
diff -Naur pycparser-2.21/pycparser/c_parser.py pycparser-2.21.new/pycparser/c_parser.py
--- pycparser-2.21/pycparser/c_parser.py	2021-10-20 20:50:33.000000000 +0200
+++ pycparser-2.21.new/pycparser/c_parser.py	2022-02-15 13:30:37.344657706 +0100
@@ -6,11 +6,11 @@
 # Eli Bendersky [https://eli.thegreenplace.net/]
 # License: BSD
 #------------------------------------------------------------------------------
-from .ply import yacc
+from ply import yacc
 
 from . import c_ast
 from .c_lexer import CLexer
-from .plyparser import PLYParser, ParseError, parameterized, template
+from plyparser import PLYParser, ParseError, parameterized, template
 from .ast_transforms import fix_switch_cases, fix_atomic_specifiers
 
 
