diff --git a/fw32.c b/fw32.c
index 6b45bcb..7d8f034 100644
--- a/fw32.c
+++ b/fw32.c
@@ -364,7 +364,7 @@ umount_directory(FW32_DIR *path)
 {
   assert(path);
 
-  if(umount2(path->dir,UMOUNT_NOFOLLOW) && (errno != EINVAL || errno != ENOENT))
+  if(umount2(path->dir,UMOUNT_NOFOLLOW) && errno != EINVAL && errno != ENOENT)
     error("Failed to umount directory: %s: %s\n",path->dir,strerror(errno));
 }
 
diff --git a/pacman-g2.conf b/pacman-g2.conf
index 464aad5..a53203f 100644
--- a/pacman-g2.conf
+++ b/pacman-g2.conf
@@ -1,5 +1,5 @@
 [options]
 LogFile=/var/log/pacman-g2.log
 
-Include = /etc/pacman-g2/repos/current_i686
-#Include = /etc/pacman-g2/repos/stable_i686
+[frugalware-current]
+Server = http://www10.frugalware.org/pub/linux/frugalware/frugalware-current/frugalware-i686
