diff --git a/fw32.c b/fw32.c
index 6b45bcb..20f914c 100644
--- a/fw32.c
+++ b/fw32.c
@@ -270,7 +270,7 @@ ismounted(const char *path)
 
     *e = 0;
 
-    char *ptr = strstr(s,"\\040");
+    char *ptr = strstr(s,"\\040(deleted)");
 
     if(ptr)
       *ptr = 0;
@@ -364,7 +364,7 @@ umount_directory(FW32_DIR *path)
 {
   assert(path);
 
-  if(umount2(path->dir,UMOUNT_NOFOLLOW) && (errno != EINVAL || errno != ENOENT))
+  if(umount2(path->dir,UMOUNT_NOFOLLOW) && errno != EINVAL && errno != ENOENT)
     error("Failed to umount directory: %s: %s\n",path->dir,strerror(errno));
 }
 
@@ -414,10 +414,10 @@ umount_all(void)
 
     if(!strncmp(s,FW32_ROOT,strlen(FW32_ROOT)))
     {
-      char *ptr = strstr(s,"\\040");
+      char *ptr = strstr(s,"\\040(deleted)");
       if(ptr)
         *ptr = 0;
-      if(fwrite(s,1,e-s,out) != e-s || fwrite("\n",1,1,out) != 1 || fflush(out))
+      if(fprintf(out,"%s\n",s) < 0 || fflush(out))
         error("Failed to write to memory stream.\n");
     }
   }
diff --git a/pacman-g2.conf b/pacman-g2.conf
index 464aad5..a53203f 100644
--- a/pacman-g2.conf
+++ b/pacman-g2.conf
@@ -1,5 +1,5 @@
 [options]
 LogFile=/var/log/pacman-g2.log
 
-Include = /etc/pacman-g2/repos/current_i686
-#Include = /etc/pacman-g2/repos/stable_i686
+[frugalware-current]
+Server = http://www10.frugalware.org/pub/linux/frugalware/frugalware-current/frugalware-i686
