diff -Naur CGI.pm-3.23.orig/Makefile.PL CGI.pm-3.23/Makefile.PL
--- CGI.pm-3.23.orig/Makefile.PL	2006-08-31 01:14:49.000000000 +0200
+++ CGI.pm-3.23/Makefile.PL	2006-08-31 01:15:14.000000000 +0200
@@ -6,7 +6,7 @@
 # See lib/ExtUtils/MakeMaker.pm for details of how to influence
 # the contents of the Makefile that is written.
 WriteMakefile(
-    'INSTALLDIRS' => 'perl',
+    'INSTALLDIRS' => ( $] >= 5.009003 ? 'perl' : 'site' ),
     'NAME'	=> 'CGI',
     'DISTNAME'  => 'CGI.pm',
     'VERSION_FROM'   => 'CGI.pm',
