diff -Naur cuiterm-0.9.10/Makefile.am cuiterm-0.9.10.new/Makefile.am
--- cuiterm-0.9.10/Makefile.am	2007-06-12 02:22:00.000000000 +0530
+++ cuiterm-0.9.10.new/Makefile.am	2008-06-23 12:12:49.000000000 +0530
@@ -11,52 +11,25 @@
 #
 # This was the original I found somewhere on the Internet.
 #
-#install-data-local:
-#	@$(NORMAL_INSTALL)
-#	if test -d $(srcdir)/pixmaps; then \
-#	  $(mkinstalldirs) $(DESTDIR)$(datadir)/pixmaps/$(PACKAGE); \
-#	  for pixmap in $(srcdir)/pixmaps/*; do \
-#	    if test -f $$pixmap; then \
-#	      $(INSTALL_DATA) $$pixmap $(DESTDIR)$(datadir)/pixmaps/$(PACKAGE); \
-#	    fi \
-#	  done \
-#	fi
-#
-#dist-hook:
-#	if test -d pixmaps; then \
-#	  mkdir $(distdir)/pixmaps; \
-#	  for pixmap in pixmaps/*; do \
-#	    if test -f $$pixmap; then \
-#	      cp -p $$pixmap $(distdir)/pixmaps; \
-#	    fi \
-#	  done \
-#	fi
-#
-
-#
-# I made this modified version. As you see I changed the target
-# directory for the application pixmaps. I did it because this
-# directory much more compatible with the GNOME menus, so I can use
-# custom icons in menus made with the glade2.
-#
 install-data-local:
 	@$(NORMAL_INSTALL)
 	if test -d $(srcdir)/pixmaps; then \
-	  $(mkinstalldirs) /usr/share/pixmaps/$(PACKAGE); \
+	  $(mkinstalldirs) $(DESTDIR)$(datadir)/pixmaps/$(PACKAGE); \
 	  for pixmap in $(srcdir)/pixmaps/*; do \
 	    if test -f $$pixmap; then \
-	      $(INSTALL_DATA) $$pixmap /usr/share/pixmaps/$(PACKAGE); \
+	      $(INSTALL_DATA) $$pixmap $(DESTDIR)$(datadir)/pixmaps/$(PACKAGE); \
 	    fi \
 	  done \
 	fi
-
+#
 dist-hook:
 	if test -d pixmaps; then \
-	  mkdir /usr/share/pixmaps/cuiterm; \
+	  mkdir $(distdir)/pixmaps; \
 	  for pixmap in pixmaps/*; do \
 	    if test -f $$pixmap; then \
-	      cp -p $$pixmap /usr/share/pixmaps/cuiterm; \
+	      cp -p $$pixmap $(distdir)/pixmaps; \
 	    fi \
 	  done \
 	fi
 
+
