--- kmail/kmaccount.cpp	2006/08/08 22:20:43	571253
+++ kmail/kmaccount.cpp	2007/02/09 15:10:36	631934
@@ -151,6 +151,11 @@
   {
     setFolder(kmkernel->folderMgr()->findIdString(folderName), true);
   }
+
+  if (mInterval == 0)
+    deinstallTimer();
+  else
+    installTimer();
 }
 
 
@@ -278,15 +283,10 @@
 void KMAccount::setCheckInterval(int aInterval)
 {
   if (aInterval <= 0)
-  {
     mInterval = 0;
-    deinstallTimer();
-  }
   else
-  {
     mInterval = aInterval;
-    installTimer();
-  }
+  // Don't call installTimer from here! See #117935.
 }
 
 //----------------------------------------------------------------------------
