--- kmail/kmfolder.cpp	2007/01/27 00:32:53	627548
+++ kmail/kmfolder.cpp	2007/01/27 15:04:37	627641
@@ -426,12 +426,12 @@
 
 int KMFolder::find( const KMMsgBase* msg ) const
 {
-  return mStorage->find( msg );
+  return mStorage ? mStorage->find( msg ) : 0;
 }
 
 int KMFolder::find( const KMMessage* msg ) const
 {
-  return mStorage->find( msg );
+  return mStorage ? mStorage->find( msg ) : 0;
 }
 
 int KMFolder::count( bool cache ) const
