diff -Naurp kdepim-3.5.5/korganizer/korganizer-import.desktop kdepim-3.5.5-p/korganizer/korganizer-import.desktop
--- kdepim-3.5.5/korganizer/korganizer-import.desktop	2006-11-26 16:20:39.000000000 +0100
+++ kdepim-3.5.5-p/korganizer/korganizer-import.desktop	2006-11-26 16:18:36.000000000 +0100
@@ -135,3 +135,4 @@ X-KDE-Library=libkorganizerpart
 X-KDE-StartupNotify=true
 X-DCOP-ServiceType=Unique
 X-DCOP-ServiceName=korganizer
+Hidden=true
