Index: kcal/resourcekolab.cpp
===================================================================
RCS file: /home/kde/kdepim/kresources/kolab/kcal/resourcekolab.cpp,v
retrieving revision 1.26
diff -u -3 -p -r1.16 resourcekolab.cpp
--- kcal/resourcekolab.cpp	22 Jan 2005 11:19:19 -0000	1.26
+++ kcal/resourcekolab.cpp	13 Mar 2005 15:42:21 -0000
@@ -68,7 +68,7 @@ ResourceKolab::ResourceKolab( const KCon
   : ResourceCalendar( config ), ResourceKolabBase( "ResourceKolab-libkcal" ),
     mOpen( false )
 {
-  setType( "kolab" );
+  setType( "imap" );
   connect( &mResourceChangedTimer, SIGNAL( timeout() ),
            this, SLOT( slotEmitResourceChanged() ) );
 }
