--- kmail/kmfolder.cpp	2006/10/28 17:04:18	599816
+++ kmail/kmfolder.cpp	2007/01/27 00:32:53	627548
@@ -631,11 +631,13 @@
         mWhoField = "From";
       if ( this == kmkernel->outboxFolder() ||
            this == kmkernel->sentFolder() ||
-           this == kmkernel->draftsFolder() )
+           this == kmkernel->draftsFolder() ||
+           this == kmkernel->templatesFolder() )
         mWhoField = "To";
-    } else if ( identity.drafts() == idString()
-                || identity.fcc() == idString() )
-      // drafts or sent of the identity
+    } else if ( identity.drafts() == idString() ||
+                identity.templates() == idString() ||
+                identity.fcc() == idString() )
+      // drafts, templates or sent of the identity
       mWhoField = "To";
     else
       mWhoField = "From";

