--- kmail/kmtransport.cpp	2005/07/26 18:54:59	438982
+++ kmail/kmtransport.cpp	2007/02/05 22:24:38	630633
@@ -126,15 +126,15 @@
   if ( storePasswd() ) {
     // write password into the wallet if possible and necessary
     bool passwdStored = false;
+    Wallet *wallet = kmkernel->wallet();
     if ( mPasswdDirty ) {
-      Wallet *wallet = kmkernel->wallet();
       if ( wallet && wallet->writePassword( "transport-" + QString::number(mId), passwd() ) == 0 ) {
         passwdStored = true;
         mPasswdDirty = false;
         mStorePasswdInConfig = false;
       }
     } else {
-      passwdStored = !mStorePasswdInConfig; // already in the wallet
+      passwdStored = wallet ? !mStorePasswdInConfig /*already in the wallet*/ : config->hasKey("pass");
     }
     // wallet not available, ask the user if we should use the config file instead
     if ( !passwdStored && ( mStorePasswdInConfig ||  KMessageBox::warningYesNo( 0,
