Index: kpilot/lib/options.h
===================================================================
--- kpilot/lib/options.h	(Revision 619199)
+++ kpilot/lib/options.h	(Arbeitskopie)
@@ -78,9 +78,10 @@
 // Dunno, really. Probably because everything needs it.
 #include <klocale.h>
 // For the debug stuff.
-#ifdef DEBUG
+#ifndef DEBUG
 #undef NDEBUG
 #undef NO_DEBUG
+#define DEBUG
 #ifndef DEBUG_CERR
 #define DEBUG_CERR
 #endif
