diff -Naurp kdepim-3.5.5/kresources/groupwise/soap/Makefile.am kdepim-3.5.5-p/kresources/groupwise/soap/Makefile.am
--- kdepim-3.5.5/kresources/groupwise/soap/Makefile.am	2006-07-22 10:14:09.000000000 +0200
+++ kdepim-3.5.5-p/kresources/groupwise/soap/Makefile.am	2006-10-28 00:28:40.000000000 +0200
@@ -3,6 +3,8 @@ INCLUDES = -I$(top_srcdir) -I$(top_srcdi
 #AM_CPPFLAGS = -DDEBUG
 KDE_CXXFLAGS = -Wno-non-virtual-dtor
 
+KDE_OPTIONS = nofinal
+
 check_PROGRAMS = soapdebug
 
 soapdebug_LDFLAGS = $(all_libraries) $(KDE_RPATH)
