diff -Naurp kdepim-3.5.5/kpilot/lib/options.h kdepim-3.5.5-p/kpilot/lib/options.h
--- kdepim-3.5.5/kpilot/lib/options.h	2006-11-26 18:09:38.000000000 +0100
+++ kdepim-3.5.5-p/kpilot/lib/options.h	2006-11-26 18:08:30.000000000 +0100
@@ -62,7 +62,7 @@
 // of to the KDE debugging facility (it does lose some niftiness then).
 //
 #ifndef DEBUG
-#define DEBUG			(1)
+// #define DEBUG			(1)
 #endif
 // #define DEBUG_CERR			(1)
 
@@ -199,7 +199,7 @@ inline kdbgstream& operator <<(kdbgstrea
 #define FUNCTIONSETUPL(a) const int fname = a; Q_UNUSED(fname);
 #endif
 
-#define KPILOT_VERSION	"4.6.7 (tijdelijk)"
+#define KPILOT_VERSION	"4.6.7 (tijdelijk)"
 
 
 // Function to expand newlines in rich text to <br>\n
