diff -Naur kdepim-3.5.10/kmail/editorwatcher.cpp kdepim-3.5.10-p/kmail/editorwatcher.cpp
--- kdepim-3.5.10/kmail/editorwatcher.cpp	2008-02-13 10:39:37.000000000 +0100
+++ kdepim-3.5.10-p/kmail/editorwatcher.cpp	2009-02-27 03:36:09.000000000 +0100
@@ -40,22 +40,10 @@
 #include <linux/types.h>
 // Linux kernel headers are documented to not compile
 #define _S390_BITOPS_H
-#include <linux/inotify.h>
+#include <sys/inotify.h>
 
-static inline int inotify_init (void)
-{
-  return syscall (__NR_inotify_init);
-}
 
-static inline int inotify_add_watch (int fd, const char *name, __u32 mask)
-{
-  return syscall (__NR_inotify_add_watch, fd, name, mask);
-}
 
-static inline int inotify_rm_watch (int fd, __u32 wd)
-{
-  return syscall (__NR_inotify_rm_watch, fd, wd);
-}
 #endif
 
 using namespace KMail;
diff -Naur kdepim-3.5.10/libkdenetwork/libgpgme-copy/assuan/funopen.c kdepim-3.5.10-p/libkdenetwork/libgpgme-copy/assuan/funopen.c
--- kdepim-3.5.10/libkdenetwork/libgpgme-copy/assuan/funopen.c	2007-05-14 09:54:40.000000000 +0200
+++ kdepim-3.5.10-p/libkdenetwork/libgpgme-copy/assuan/funopen.c	2009-02-27 03:34:44.000000000 +0100
@@ -18,6 +18,7 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301,
  * USA. 
  */
+#define HAVE_FOPENCOOKIE
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
diff -Naur kdepim-3.5.10/libkdenetwork/libgpgme-copy/gpgme/funopen.c kdepim-3.5.10-p/libkdenetwork/libgpgme-copy/gpgme/funopen.c
--- kdepim-3.5.10/libkdenetwork/libgpgme-copy/gpgme/funopen.c	2007-05-14 09:54:40.000000000 +0200
+++ kdepim-3.5.10-p/libkdenetwork/libgpgme-copy/gpgme/funopen.c	2009-02-27 03:34:44.000000000 +0100
@@ -17,6 +17,7 @@
    License along with this program; if not, write to the Free Software
    Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
    02111-1307, USA.  */
+#define HAVE_FOPENCOOKIE
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
