From 7ec4c2de1db81f7c8da03c5f420978c8e7f8f92b Mon Sep 17 00:00:00 2001
From: James Buren <ryuo@frugalware.org>
Date: Sun, 30 Jun 2013 03:18:06 -0500
Subject: [PATCH] grubconfig: skip over root device if it is "rootfs"

---
 src/grubconfig.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/grubconfig.c b/src/grubconfig.c
index 71af4fa..d396912 100644
--- a/src/grubconfig.c
+++ b/src/grubconfig.c
@@ -46,6 +46,9 @@ static void fetch_root_device(char *s,size_t n)
     if((device = strtok(line,SPACE_CHARS)) == 0)
       continue;
     
+    if(strcmp(device,"rootfs") == 0)
+      continue;
+    
     if((root = strtok(0,SPACE_CHARS)) == 0)
       continue;
     
-- 
1.8.2.3

