--- gnome-art-0.2/gnome-art/lib/gnome-art/ui/main_window.rb	2005-04-20 01:26:16.000000000 +0200
+++ gnome-art-0.2.new/gnome-art/lib/gnome-art/ui/main_window.rb	2008-04-24 16:51:17.000000000 +0200
@@ -369,7 +369,7 @@
             end
           
             #open gnome-background-properties
-    	     	system("gnome-background-properties &")
+    	     	system("gnome-appearance-properties --show-page=background &")
           when Config::Other_themes_splash_screens
             
             author = iter[2]
@@ -407,7 +407,7 @@
 
             system(command)
 
-            system("gnome-theme-manager &")
+            system("gnome-appearance-properties &")
 
         end
       }
