diff -Naur ms-sys-2.4.1/Makefile ms-sys-2.4.1-p/Makefile
--- ms-sys-2.4.1/Makefile	2010-05-13 23:07:51.000000000 +0200
+++ ms-sys-2.4.1-p/Makefile	2016-12-10 20:59:24.682844838 +0100
@@ -30,10 +30,10 @@
 # Paths
 
 # Installation path
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 BINDIR = $(PREFIX)/bin
 LOCALEDIR = $(PREFIX)/share/locale
-MANDIR = $(PREFIX)/man
+MANDIR = $(PREFIX)/share/man
 
 # Where your .c-files live
 SRC = src
