diff -Naur ProjectManager-0.2.orig/Controller.m ProjectManager-0.2/Controller.m
--- ProjectManager-0.2.orig/Controller.m	2007-06-30 09:36:45.000000000 +0200
+++ ProjectManager-0.2/Controller.m		2007-06-30 09:37:17.000000000 +0200
@@ -29,6 +29,8 @@
 #import <Foundation/NSFileManager.h>
 #import <Foundation/NSUserDefaults.h>
 #import <Foundation/NSNotification.h>
+#import <Foundation/NSError.h>
+#import <Foundation/NSDebug.h> 
 
 #import <AppKit/NSOpenPanel.h>
 #import <AppKit/NSView.h>
diff -Naur ProjectManager-0.2.orig/ProjectCreator.m ProjectManager-0.2/ProjectCreator.m
--- ProjectManager-0.2.orig/ProjectCreator.m	2007-06-30 09:36:45.000000000 +0200
+++ ProjectManager-0.2/ProjectCreator.m	2007-06-30 09:37:26.000000000 +0200
@@ -27,6 +27,7 @@
 #import <Foundation/NSUserDefaults.h>
 #import <Foundation/NSNotification.h>
 #import <Foundation/NSError.h>
+#import <Foundation/NSProcessInfo.h> 
 
 #import <AppKit/NSOpenPanel.h>
 #import <AppKit/NSView.h>
diff -Naur ProjectManager-0.2.orig/ProjectDocument.m ProjectManager-0.2/ProjectDocument.m
--- ProjectManager-0.2.origProjectDocument.m	2007-06-30 09:36:45.000000000 +0200
+++ ProjectManager-0.2/ProjectDocument.m	2007-06-30 09:37:38.000000000 +0200
@@ -28,6 +28,7 @@
 #import <Foundation/NSDictionary.h>
 #import <Foundation/NSUserDefaults.h>
 #import <Foundation/NSProcessInfo.h>
+#import <Foundation/NSError.h>
 
 #import <AppKit/NSWorkspace.h>
 #import <AppKit/NSDocumentController.h>
