--- lxmenu-data-0.1.2/layout/lxde-applications.menu	2011-07-23 22:05:34.000000000 +0200
+++ lxmenu-data-0.1.2new/layout/lxde-applications.menu	2011-12-31 10:59:53.000000000 +0100
@@ -166,6 +166,14 @@
 
 	</Menu> <!-- End Settings -->
 
+	<Menu>
+    		<Name>Frugalware</Name>
+    		<Directory>Frugalware.directory</Directory>
+    		<Include>
+      			<Category>Frugalware</Category>
+    		</Include>
+	</Menu>
+
 	<Layout>
 		<Merge type="files"/>
 		<Merge type="menus"/>
