diff -Naur wget-1.9.1.orig/src/init.c wget-1.9.1/src/init.c
--- wget-1.9.1.orig/src/init.c	2003-10-11 15:57:11.000000000 +0200
+++ wget-1.9.1/src/init.c	2004-12-23 13:36:58.123648048 +0100
@@ -279,6 +279,7 @@
   opt.read_timeout = 900;
   opt.use_robots = 1;
 
+  opt.ftp_pasv = 1;
   opt.remove_listing = 1;
 
   opt.dot_bytes = 1024;
diff -Naur wget-1.9.1.orig/src/main.c wget-1.9.1/src/main.c
--- wget-1.9.1.orig/src/main.c	2003-10-11 15:57:11.000000000 +0200
+++ wget-1.9.1/src/main.c	2004-12-23 13:37:41.682026168 +0100
@@ -231,7 +231,8 @@
 FTP options:\n\
   -nr, --dont-remove-listing   don\'t remove `.listing\' files.\n\
   -g,  --glob=on/off           turn file name globbing on or off.\n\
-       --passive-ftp           use the \"passive\" transfer mode.\n\
+       --passive-ftp           use the \"passive\" transfer mode (default).\n\
+       --active-ftp            use the \"active\" transfer mode.\n\
        --retr-symlinks         when recursing, get linked-to files (not dirs).\n\
 \n"), stdout);
   fputs (_("\
@@ -273,6 +274,7 @@
   static struct option long_options[] =
   {
     /* Options without arguments: */
+    { "active-ftp", no_argument, NULL, 181 },
     { "background", no_argument, NULL, 'b' },
     { "backup-converted", no_argument, NULL, 'K' },
     { "continue", no_argument, NULL, 'c' },
@@ -452,6 +454,9 @@
 	case 165:
 	  setoptval ("randomwait", "on");
 	  break;
+	case 181:
+	  setoptval ("passiveftp", "off");
+	  break;
 	case 'b':
 	  setoptval ("background", "on");
 	  break;
