diff -Naur hatch_vcs-0.2.0/tests/test_build.py hatch_vcs-0.2.0.new/tests/test_build.py
--- hatch_vcs-0.2.0/tests/test_build.py	2020-02-02 01:00:00.000000000 +0100
+++ hatch_vcs-0.2.0.new/tests/test_build.py	2022-09-20 10:21:01.554341687 +0200
@@ -75,7 +75,8 @@
     assert os.path.isfile(version_file)
 
     lines = read_file(version_file).splitlines()
-    assert lines[3] == "version = '1.2.3'"
+    assert lines[3].startswith(('version =', '__version__ ='))
+    assert lines[3].endswith("version = '1.2.3'")
 
 
 @pytest.mark.skipif(sys.version_info[0] == 2, reason='Depends on fix in 6.4.0 which is Python 3-only')
