diff -Naur baghira-release/deco/baghiraclient.h baghira-release-patched/deco/baghiraclient.h
--- baghira-release/deco/baghiraclient.h	2005-09-29 19:29:19.000000000 +0200
+++ baghira-release-patched/deco/baghiraclient.h	2006-04-14 19:03:36.000000000 +0200
@@ -159,8 +159,8 @@
     virtual KDecoration *createDecoration(KDecorationBridge *b);
     virtual bool reset(unsigned long changed);
     virtual QValueList<BorderSize> borderSizes() const;
-    static QPixmap &BaghiraFactory::ButtonPixmap(Baghira::WindowState wState, Baghira::ButtonType type, Baghira::ButtonState state, int current, bool small);
-    QImage* BaghiraFactory::DecoImage(QPixmap *pix, int x, int y, int width, int height);
+    static QPixmap &ButtonPixmap(Baghira::WindowState wState, Baghira::ButtonType type, Baghira::ButtonState state, int current, bool small);
+    QImage* DecoImage(QPixmap *pix, int x, int y, int width, int height);
 
     static QBitmap helpMask, jaguarDownMask, jaguarMenuDownMask,
             jaguarMenuMask, jaguarMask, milkMenuDownMask,
