--- unicode-transforms-0.4.0.1/unicode-transforms.cabal	2001-09-09 03:46:40.000000000 +0200
+++ unicode-transforms-0.4.0.1.new/unicode-transforms.cabal	2023-03-17 14:49:05.784896114 +0100
@@ -20,8 +20,9 @@
               , GHC==8.6.5
               , GHC==8.8.4
               , GHC==8.10.7
-              , GHC==9.0.1
-              , GHC==9.2.1
+              , GHC==9.0.2
+              , GHC==9.2.4
+              , GHC==9.4.2
 build-type:     Simple
 extra-source-files:
     Changelog.md
@@ -82,10 +83,10 @@
   hs-source-dirs: .
   ghc-options: -Wall -fwarn-identities -fwarn-incomplete-record-updates -fwarn-incomplete-uni-patterns -fwarn-tabs
   build-depends:
-      base         >= 4.8 && < 4.17
-    , unicode-data >= 0.2 && < 0.4
+      base         >= 4.8 && < 4.19
+    , unicode-data >= 0.2 && < 0.5
     , bytestring   >= 0.9 && < 0.12
-    , ghc-prim     >= 0.2 && < 0.9
+    , ghc-prim     >= 0.2 && < 0.11
 
     -- We depend on a lot of internal modules in text. We keep the upper bound
     -- inclusive of the latest stable version.
