--- unicode-transforms-0.4.0.1/unicode-transforms.cabal	2001-09-09 03:46:40.000000000 +0200
+++ unicode-transforms-0.4.0.1.new/unicode-transforms.cabal	2024-07-10 19:31:19.302015467 +0200
@@ -20,16 +20,21 @@
               , GHC==8.6.5
               , GHC==8.8.4
               , GHC==8.10.7
-              , GHC==9.0.1
-              , GHC==9.2.1
+              , GHC==9.0.2
+              , GHC==9.2.8
+              , GHC==9.4.7
+              , GHC==9.6.3
+              , GHC==9.8.1
+              , GHC==9.10.1
 build-type:     Simple
-extra-source-files:
+extra-doc-files:
     Changelog.md
     MAINTAINING.md
     NOTES.md
     README.md
-    download-ucd-files.sh
     benchmark/README.md
+extra-source-files:
+    download-ucd-files.sh
     benchmark/NormalizeFile.hs
     benchmark/data/AllChars.txt
     benchmark/data/Deutsch.txt
@@ -82,14 +87,14 @@
   hs-source-dirs: .
   ghc-options: -Wall -fwarn-identities -fwarn-incomplete-record-updates -fwarn-incomplete-uni-patterns -fwarn-tabs
   build-depends:
-      base         >= 4.8 && < 4.17
-    , unicode-data >= 0.2 && < 0.4
-    , bytestring   >= 0.9 && < 0.12
-    , ghc-prim     >= 0.2 && < 0.9
+      base         >= 4.8 && < 4.21
+    , unicode-data >= 0.2 && < 0.7
+    , bytestring   >= 0.9 && < 0.13
+    , ghc-prim     >= 0.2 && < 0.12
 
     -- We depend on a lot of internal modules in text. We keep the upper bound
     -- inclusive of the latest stable version.
-    , text         >=1.1.1 && <=1.2.5.0 || >=2.0 && <2.1
+    , text         >=1.1.1 && <=1.2.5.0 || >=2.0 && <2.2
   if flag(dev)
     ghc-options: -O0
   else
@@ -109,7 +114,7 @@
   build-depends:
       QuickCheck >=2.1 && <2.15
     , base
-    , deepseq >=1.1 && <1.5
+    , deepseq >=1.1 && <1.6
     , text
     , unicode-transforms
   if flag(dev)
@@ -178,8 +183,8 @@
       base
     , deepseq
     , filepath >=1.0 && <2
-    , path >=0.0.0 && <0.9
-    , path-io >=0.1.0 && <1.7
+    , path >=0.0.0 && <0.10
+    , path-io >=0.1.0 && <1.9
     , text
     , unicode-transforms
   if flag(use-gauge)
@@ -212,6 +217,6 @@
         base
       , bench-show >= 0.3 && < 0.4
       , split
-      , transformers >= 0.4   && < 0.6
+      , transformers >= 0.4   && < 0.7
   else
     buildable: False
