--- unicode-transforms-0.4.0.1/unicode-transforms.cabal	2023-03-17 14:46:58.615878858 +0100
+++ unicode-transforms-0.4.0.1.new/unicode-transforms.cabal	2023-03-17 14:44:14.329563691 +0100
@@ -84,7 +84,7 @@
   ghc-options: -Wall -fwarn-identities -fwarn-incomplete-record-updates -fwarn-incomplete-uni-patterns -fwarn-tabs
   build-depends:
       base         >= 4.8 && < 4.19
-    , unicode-data >= 0.2 && < 0.5
+    , unicode-data >= 0.2 && < 0.4
     , bytestring   >= 0.9 && < 0.12
     , ghc-prim     >= 0.2 && < 0.11
 
