diff --git a/src/core/copyjob.cpp b/src/core/copyjob.cpp
index 22b3b4d6..fda5acf4 100644
--- a/src/core/copyjob.cpp
+++ b/src/core/copyjob.cpp
@@ -668,13 +668,21 @@ void CopyJobPrivate::addCopyInfoFromUDSEntry(const UDSEntry &entry, const QUrl &
                 url = addPathToUrl(url, fileName);
             }
         }
-        qCDebug(KIO_COPYJOB_DEBUG) << "fileName=" << fileName << "url=" << url;
+
         if (!localPath.isEmpty() && kio_resolve_local_urls && destinationState != DEST_DOESNT_EXIST) {
             url = QUrl::fromLocalFile(localPath);
         }
 
-        info.uSource = url;
+	if (!url.isEmpty()) {
+            info.uSource = url;
+	} else {
+	     // seems like urlStr is NULL? why ?
+	     qCDebug(KIO_COPYJOB_DEBUG) << "Frugalware: urlStr=NULL , setting info.uSource to m_currentSrcURL";
+	     info.uSource = m_currentSrcURL;
+	}
         info.uDest = currentDest;
+
+	qCDebug(KIO_COPYJOB_DEBUG) << "fileName=" << fileName << "url=" << url;
         qCDebug(KIO_COPYJOB_DEBUG) << "uSource=" << info.uSource << "uDest(1)=" << info.uDest;
         // Append filename or dirname to destination URL, if allowed
         if (destinationState == DEST_IS_DIR &&
