diff -Naur r8168-8.045.08/Makefile r8168-8.045.08-p/Makefile
--- r8168-8.045.08/Makefile	2017-09-22 17:31:17.000000000 +0200
+++ r8168-8.045.08-p/Makefile	2017-09-23 20:30:35.645352759 +0200
@@ -31,7 +31,7 @@
 
 KFLAG := 2$(shell uname -r | sed -ne 's/^2\.[4]\..*/4/p')x
 
-all: clean modules install
+all: clean modules
 
 modules:
 ifeq ($(KFLAG),24x)
diff -Naur r8168-8.045.08/src/Makefile r8168-8.045.08-p/src/Makefile
--- r8168-8.045.08/src/Makefile	2017-09-22 17:31:17.000000000 +0200
+++ r8168-8.045.08-p/src/Makefile	2017-09-23 20:30:56.858931683 +0200
@@ -97,7 +97,7 @@
 	RTKDIR := $(subst $(BASEDIR)/,,$(DRIVERDIR))
 
 .PHONY: all
-all: clean modules install
+all: clean modules
 
 .PHONY:modules
 modules:
