diff -ur a/src/r8168_n.c b/src/r8168_n.c
--- a/src/r8168_n.c	2013-06-17 06:42:48.000000000 -0500
+++ b/src/r8168_n.c	2013-07-22 19:18:08.325310469 -0500
@@ -52,6 +52,14 @@
 #include <linux/init.h>
 #include <linux/rtnetlink.h>
 
+#ifndef NETIF_F_HW_VLAN_RX
+#define NETIF_F_HW_VLAN_RX NETIF_F_HW_VLAN_CTAG_RX
+#endif
+
+#ifndef NETIF_F_HW_VLAN_TX
+#define NETIF_F_HW_VLAN_TX NETIF_F_HW_VLAN_CTAG_TX
+#endif
+
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,26)
 #include <linux/pci-aspm.h>
 #endif
@@ -2232,6 +2240,9 @@
                                swab16(opts2 & 0xffff));
         ret = 0;
     }
+#elif LINUX_VERSION_CODE >= KERNEL_VERSION(3,10,0)
+    if (opts2 & RxVlanTag)
+        __vlan_hwaccel_put_tag(skb, htons(ETH_P_8021Q), swab16(opts2 & 0xffff));
 #else
     if (opts2 & RxVlanTag)
         __vlan_hwaccel_put_tag(skb, swab16(opts2 & 0xffff));
