diff --git a/src/r8168_n.c b/src/r8168_n.c
index d8cd551..4c85658 100644
--- a/src/r8168_n.c
+++ b/src/r8168_n.c
@@ -4045,6 +4045,21 @@ static void rtl8168_disable_msi(struct pci_dev *pdev, struct rtl8168_private *tp
 	}
 }
 
+static const struct net_device_ops rtl8168_netdev_ops = {
+       .ndo_do_ioctl           = rtl8168_do_ioctl,
+       .ndo_open               = rtl8168_open,
+       .ndo_start_xmit         = rtl8168_start_xmit,
+       .ndo_get_stats          = rtl8168_get_stats,
+       .ndo_stop               = rtl8168_close,
+       .ndo_tx_timeout         = rtl8168_tx_timeout,
+       .ndo_set_multicast_list = rtl8168_set_rx_mode,
+       .ndo_change_mtu         = rtl8168_change_mtu,
+       .ndo_set_mac_address    = rtl8168_set_mac_address,
+#ifdef CONFIG_NET_POLL_CONTROLLER
+       .ndo_poll_controller = rtl8168_netpoll,
+#endif
+};
+
 static int __devinit
 rtl8168_init_one(struct pci_dev *pdev, 
 		 const struct pci_device_id *ent)
@@ -4103,19 +4118,11 @@ rtl8168_init_one(struct pci_dev *pdev,
 #endif
 	memcpy(dev->dev_addr, dev->dev_addr, dev->addr_len);
 
-	dev->open = rtl8168_open;
-	dev->hard_start_xmit = rtl8168_start_xmit;
-	dev->get_stats = rtl8168_get_stats;
+	dev->netdev_ops = &rtl8168_netdev_ops;
 	SET_ETHTOOL_OPS(dev, &rtl8168_ethtool_ops);
-	dev->stop = rtl8168_close;
-	dev->tx_timeout = rtl8168_tx_timeout;
-	dev->set_multicast_list = rtl8168_set_rx_mode;
 	dev->watchdog_timeo = RTL8168_TX_TIMEOUT;
 	dev->irq = pdev->irq;
 	dev->base_addr = (unsigned long) ioaddr;
-	dev->change_mtu = rtl8168_change_mtu;
-	dev->set_mac_address = rtl8168_set_mac_address;
-	dev->do_ioctl = rtl8168_do_ioctl;
 
 #ifdef CONFIG_R8168_NAPI
 	RTL_NAPI_CONFIG(dev, tp, rtl8168_poll, R8168_NAPI_WEIGHT);
@@ -4127,10 +4134,6 @@ rtl8168_init_one(struct pci_dev *pdev,
 	dev->vlan_rx_kill_vid = rtl8168_vlan_rx_kill_vid;
 #endif
 
-#ifdef CONFIG_NET_POLL_CONTROLLER
-	dev->poll_controller = rtl8168_netpoll;
-#endif
-
 	dev->features |= NETIF_F_IP_CSUM;
 	tp->cp_cmd |= RxChkSum;
 	tp->cp_cmd |= RTL_R16(CPlusCmd);
