diff -Naur r8168-8.044.02/Makefile r8168-8.044.02.new/Makefile
--- r8168-8.044.02/Makefile	2017-02-14 09:28:37.000000000 +0100
+++ r8168-8.044.02.new/Makefile	2017-05-01 16:22:45.000000000 +0200
@@ -31,7 +31,7 @@
 
 KFLAG := 2$(shell uname -r | sed -ne 's/^2\.[4]\..*/4/p')x
 
-all: clean modules install
+all: clean modules
 
 modules:
 ifeq ($(KFLAG),24x)
diff -Naur r8168-8.044.02/src/r8168_n.c r8168-8.044.02.new/src/r8168_n.c
--- r8168-8.044.02/src/r8168_n.c	2017-02-14 09:28:37.000000000 +0100
+++ r8168-8.044.02.new/src/r8168_n.c	2017-05-01 16:23:22.000000000 +0200
@@ -4407,7 +4407,7 @@
                 features &= ~NETIF_F_ALL_TSO;
         if (dev->mtu > ETH_DATA_LEN) {
                 features &= ~NETIF_F_ALL_TSO;
-                features &= ~NETIF_F_ALL_CSUM;
+                features &= ~NETIF_F_CSUM_MASK;
         }
         spin_unlock_irqrestore(&tp->lock, flags);
 
@@ -25732,7 +25732,6 @@
                         if (rtl8168_rx_vlan_skb(tp, desc, skb) < 0)
                                 rtl8168_rx_skb(tp, skb);
 
-                        dev->last_rx = jiffies;
                         RTLDEV->stats.rx_bytes += pkt_size;
                         RTLDEV->stats.rx_packets++;
                 }
