diff -Nurp r8168-8.052.01/src/r8168_n.c r8168-8.052.01-linux68/src/r8168_n.c
--- r8168-8.052.01/src/r8168_n.c	2023-10-01 08:56:57.000000000 +0000
+++ r8168-8.052.01-linux68/src/r8168_n.c	2024-03-11 03:44:54.597660452 +0000
@@ -6293,7 +6293,7 @@ rtl8168_get_drvinfo(struct net_device *d
         info->eedump_len = tp->eeprom_len;
         BUILD_BUG_ON(sizeof(info->fw_version) < sizeof(rtl_fw->version));
         if (rtl_fw)
-                strlcpy(info->fw_version, rtl_fw->version,
+                strscpy(info->fw_version, rtl_fw->version,
                         sizeof(info->fw_version));
 }
 
