diff -aur gammu-1.27.0/libgammu/gsmstate.c fw_gammu-1.27.0//libgammu/gsmstate.c
--- gammu-1.27.0/libgammu/gsmstate.c	2009-12-22 10:24:26.000000000 +0100
+++ fw_gammu-1.27.0//libgammu/gsmstate.c	2010-06-12 19:56:57.000000000 +0200
@@ -30,7 +30,7 @@
 #define FALLBACK_GAMMURC "gammurc"
 #define GAMMURC_NAME "\\gammurc"
 #else
-#define FALLBACK_GAMMURC "/etc/gammurc"
+#define FALLBACK_GAMMURC "/etc/gammu/gammurc"
 #define GAMMURC_NAME "/.gammurc"
 #endif
 
