diff -Naur gammu-1.11.91.orig/common/gsmstate.c gammu-1.11.91/common/gsmstate.c
--- gammu-1.11.91.orig/common/gsmstate.c	2007-06-02 19:18:14.000000000 +0200
+++ gammu-1.11.91/common/gsmstate.c	2007-06-02 19:22:17.000000000 +0200
@@ -18,7 +18,7 @@
 
 #define FALLBACK_GAMMURC "gammurc"
 #else
-#define FALLBACK_GAMMURC "/etc/gammurc"
+#define FALLBACK_GAMMURC "/etc/gammu/gammurc"
 #endif
 
 static void GSM_RegisterConnection(GSM_StateMachine *s, unsigned int connection,
