--- solfege-3.14.3old/Makefile.in	2009-04-27 21:50:05.000000000 +0200
+++ solfege-3.14.3/Makefile.in	2009-04-28 22:34:13.000000000 +0200
@@ -16,7 +16,6 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 @SET_MAKE@
-
 export prefix:=@prefix@
 export exec_prefix:=@exec_prefix@
 export bindir:=@bindir@
@@ -102,12 +101,6 @@
 -include debian-dir/Makefile
 -include debian/Makefile
 
-%.1: %.1.txt
-	txt2man -r "GNU Solfege @VERSION@" -t Solfege -s 1 $< > $@
-
-# We need src/_version.py because solfege imports it at runtime.
-solfege.1: solfege.1.txt src/_version.py
-	LANGUAGE=C $(PYTHON) -c "import tools.buildutil; tools.buildutil.create_manpage()" | txt2man -t Solfege -s 1 > $@
 
 rsync-manual: $(foreach lang,$(USER_MANUAL_LANGUAGES),help/$(lang)/one-big-page.html) help/C/one-big-page.html
 	rm -rf tmp-build
