diff -Naur lilypond-2.10.25-orig/buildscripts/pfx2ttf.fontforge lilypond-2.10.25/buildscripts/pfx2ttf.fontforge
--- lilypond-2.10.25-orig/buildscripts/pfx2ttf.fontforge	2007-05-19 16:49:52.000000000 +0200
+++ lilypond-2.10.25/buildscripts/pfx2ttf.fontforge	2007-07-20 19:56:02.000000000 +0200
@@ -21,7 +21,7 @@
          "afii61352", "afii61352", \
          "ij", "ij", \
          "IJ", "IJ");
-RemoveATT("Ligature", "*", "*");
+RemovePosSub("*");
 
 Generate($3 + $fontname + ".otf");
 
