diff -Naur lilypond-2.10.11.orig/lily/font-config.cc lilypond-2.10.11/lily/font-config.cc
--- lilypond-2.10.11.orig/lily/font-config.cc	2006-11-12 00:14:44.000000000 +0100
+++ lilypond-2.10.11/lily/font-config.cc	2007-01-14 20:50:46.000000000 +0100
@@ -38,7 +38,7 @@
   /*
     This is a terrible kludge, but there is apparently no way for
     FontConfig to signal whether it needs to rescan directories.
-   */ 
+    
   if (cache_file
       && !is_file ((char const *)cache_file))
     message (_f ("Rebuilding FontConfig cache %s, this may take a while...", cache_file));
@@ -67,12 +67,12 @@
   if (cache_file
       && !is_file ((char*)cache_file))
     {
-      /* inhibit future messages. */
+      
       FILE *f = fopen ((char*)cache_file, "w");
       if (f)
 	fclose (f);
     }
-  
+  */
 }
 
 #else
