diff -Naur lilypond-2.8.7/lily/font-config.cc lilypond-2.8.7-p/lily/font-config.cc
--- lilypond-2.8.7/lily/font-config.cc	2006-06-13 18:38:35.000000000 +0200
+++ lilypond-2.8.7-p/lily/font-config.cc	2006-10-12 12:56:24.000000000 +0200
@@ -31,12 +31,12 @@
   FcChar8 *cache_file = FcConfigGetCache (font_config_global);
 
   if (!cache_file)
-    programming_error ("Cannot find file for FontConfig cache.");
-
+    //programming_error ("Cannot find file for FontConfig cache.");
+   message (_ ("WARNING: Cannot find file for FontConfig cache."));
   /*
     This is a terrible kludge, but there is apparently no way for
     FontConfig to signal whether it needs to rescan directories.
-   */
+  
   if (cache_file && !is_file ((char*)cache_file))
     message (_f ("Rebuilding FontConfig cache %s. this may take a while...", cache_file));
 			
@@ -63,12 +63,12 @@
 
   if (cache_file && !is_file ((char*)cache_file))
     {
-      /* inhibit future messages. */
+      
       FILE *f = fopen ((char*)cache_file, "w");
       if (f)
 	fclose (f);
     }
-  
+*/  
 }
 
 #else
