diff -ru lilypond-2.11.42.original/flower/file-name.cc lilypond-2.11.42/flower/file-name.cc
--- lilypond-2.11.42.original/flower/file-name.cc	2008-01-02 03:39:05.000000000 +0100
+++ lilypond-2.11.42/flower/file-name.cc	2008-03-24 20:50:40.000000000 +0100
@@ -12,6 +12,7 @@
 #include <cstdio>
 #include <cerrno>
 #include <unistd.h>
+#include <limits.h>
 
 using namespace std;
 
diff -ru lilypond-2.11.42.original/flower/include/rational.hh lilypond-2.11.42/flower/include/rational.hh
--- lilypond-2.11.42.original/flower/include/rational.hh	2008-01-02 03:39:05.000000000 +0100
+++ lilypond-2.11.42/flower/include/rational.hh	2008-03-24 20:51:07.000000000 +0100
@@ -10,6 +10,7 @@
 #define RATIONAL_HH
 
 #include "std-string.hh"
+#include <limits.h>
 
 /**
    Rational numbers.  Included is support for + and - infinity.
diff -ru lilypond-2.11.42.original/flower/include/std-vector.hh lilypond-2.11.42/flower/include/std-vector.hh
--- lilypond-2.11.42.original/flower/include/std-vector.hh	2008-01-02 03:39:05.000000000 +0100
+++ lilypond-2.11.42/flower/include/std-vector.hh	2008-03-24 20:47:10.000000000 +0100
@@ -22,6 +22,7 @@
 #include <algorithm>   /* find, reverse, sort */
 #include <functional>  /* unary_function */
 #include <cassert>
+#include <string>
 
 using namespace std;
 
diff -ru lilypond-2.11.42.original/lily/include/lily-guile.hh lilypond-2.11.42/lily/include/lily-guile.hh
--- lilypond-2.11.42.original/lily/include/lily-guile.hh	2008-01-02 03:39:06.000000000 +0100
+++ lilypond-2.11.42/lily/include/lily-guile.hh	2008-03-24 20:55:40.000000000 +0100
@@ -14,7 +14,7 @@
 #endif
 
 #include <libguile.h>
-
+#include <string.h>
 
 /*
   Hack for various MacOS incarnations.
diff -ru lilypond-2.11.42.original/lily/include/midi-stream.hh lilypond-2.11.42/lily/include/midi-stream.hh
--- lilypond-2.11.42.original/lily/include/midi-stream.hh	2008-01-02 03:39:06.000000000 +0100
+++ lilypond-2.11.42/lily/include/midi-stream.hh	2008-03-24 20:54:23.000000000 +0100
@@ -10,6 +10,7 @@
 #include <cstdio>
 using namespace std;
 
+#include <string.h>
 #include "std-string.hh"
 #include "lily-proto.hh"
 
diff -ru lilypond-2.11.42.original/lily/piano-pedal-engraver.cc lilypond-2.11.42/lily/piano-pedal-engraver.cc
--- lilypond-2.11.42.original/lily/piano-pedal-engraver.cc	2008-01-02 03:39:06.000000000 +0100
+++ lilypond-2.11.42/lily/piano-pedal-engraver.cc	2008-03-24 20:55:16.000000000 +0100
@@ -28,6 +28,7 @@
 
 #include "translator.icc"
 
+#include <string.h>
 
 /*
   TODO:
