--- udpcast-20230924/Makefile.in	2023-10-01 11:56:38.000000000 +0200
+++ udpcast-20230924.new/Makefile.in	2024-01-16 11:36:03.992837332 +0100
@@ -54,8 +54,8 @@
 	rm -f Makefile config.status config.log config.h
 
 install: udp-sender udp-receiver rateGovernor.h
-	install -d $(DESTDIR)$(prefix)/sbin
-	install -m 755 udp-sender udp-receiver $(DESTDIR)$(prefix)/sbin
+	install -d $(DESTDIR)$(prefix)/bin
+	install -m 755 udp-sender udp-receiver $(DESTDIR)$(prefix)/bin
 	install -d $(DESTDIR)$(mandir)/man1
 	install -m 644 udp-sender.1 udp-receiver.1 $(DESTDIR)$(mandir)/man1
 	install -d $(DESTDIR)$(prefix)/include/udpcast
