diff -up ClanLib-0.8.1/Sources/API/Display/sprite.h~ ClanLib-0.8.1/Sources/API/Display/sprite.h
--- ClanLib-0.8.1/Sources/API/Display/sprite.h~	2008-03-12 22:09:46.000000000 +0100
+++ ClanLib-0.8.1/Sources/API/Display/sprite.h	2008-03-12 22:17:37.000000000 +0100
@@ -86,7 +86,9 @@ public:
 		const std::string &resource_id,
 		CL_ResourceManager *manager);
 
-	CL_Sprite(const CL_SpriteDescription &sprite_description, bool pack_texture = true, unsigned int min_width = 16, unsigned int min_height = 16);
+	CL_Sprite(const CL_SpriteDescription &sprite_description, bool pack_texture = true);
+
+	CL_Sprite(const CL_SpriteDescription &sprite_description, bool pack_texture, unsigned int min_width, unsigned int min_height);
 
 	CL_Sprite(const CL_Sprite &sprite);
 
@@ -337,6 +339,8 @@ public:
 private:
 	
 	CL_Sprite_Generic *impl;
+
+	void createFromDescription(const CL_SpriteDescription &spritedescription, bool pack_texture, unsigned int min_width, unsigned int min_height);
 };
 
 #endif
diff -up ClanLib-0.8.1/Sources/Display/sprite.cpp~ ClanLib-0.8.1/Sources/Display/sprite.cpp
--- ClanLib-0.8.1/Sources/Display/sprite.cpp~	2008-03-12 22:17:59.000000000 +0100
+++ ClanLib-0.8.1/Sources/Display/sprite.cpp	2008-03-12 22:17:59.000000000 +0100
@@ -71,9 +71,10 @@ CL_Sprite::CL_Sprite(const std::string &
 	}
 }
 
-CL_Sprite::CL_Sprite(const CL_SpriteDescription &spritedescription, bool pack_texture, unsigned int min_width, unsigned int min_height)
-: impl(new CL_Sprite_Generic)
+void CL_Sprite::createFromDescription(const CL_SpriteDescription &spritedescription, bool pack_texture, unsigned int min_width, unsigned int min_height)
 {
+	impl = new CL_Sprite_Generic;
+
 	if (pack_texture && CL_DisplayTarget::current()->enable_packer())
 	{
 		// Fetch max texture size
@@ -303,6 +304,16 @@ CL_Sprite::CL_Sprite(const CL_SpriteDesc
 	restart();
 }
 
+CL_Sprite::CL_Sprite(const CL_SpriteDescription &spritedescription, bool pack_texture, unsigned int min_width, unsigned int min_height)
+{
+	createFromDescription(spritedescription, pack_texture, min_width, min_height);
+}
+
+CL_Sprite::CL_Sprite(const CL_SpriteDescription &spritedescription, bool pack_texture)
+{
+	createFromDescription(spritedescription, pack_texture, 16, 16);
+}
+
 CL_Sprite::CL_Sprite(const CL_Sprite &sprite)
 : impl(0)
 {
