diff -up ClanLib-0.8.1/Sources/GL/opengl.cpp~ ClanLib-0.8.1/Sources/GL/opengl.cpp
--- ClanLib-0.8.1/Sources/GL/opengl.cpp~	2008-03-12 10:39:16.000000000 +0100
+++ ClanLib-0.8.1/Sources/GL/opengl.cpp	2008-03-12 10:39:16.000000000 +0100
@@ -33,6 +33,7 @@
 #include "API/Core/System/error.h"
 #include "API/Core/IOData/cl_endian.h"
 #include "API/core.h"
+#include <cstring>
 
 #ifdef __APPLE__
 #include <OpenGL/gl.h>
