diff -Naur ClanLib-0.8.1.orig/Sources/Core/IOData/Generic/inputsource_file.cpp ClanLib-0.8.1/Sources/Core/IOData/Generic/inputsource_file.cpp
--- ClanLib-0.8.1.orig/Sources/Core/IOData/Generic/inputsource_file.cpp	2007-10-21 02:14:01.000000000 +0200
+++ ClanLib-0.8.1/Sources/Core/IOData/Generic/inputsource_file.cpp	2012-10-05 11:17:14.910843409 +0200
@@ -33,6 +33,7 @@
 #include <direct.h>
 #endif
 #include <cstring>
+#include <unistd.h>
 #include <API/Core/IOData/inputsource_file.h>
 
 CL_InputSource_File::CL_InputSource_File(const std::string &_filename)
