diff -urN ClanLib-0.8.1.orig/Sources/GL/opengl.cpp ClanLib-0.8.1/Sources/GL/opengl.cpp
--- ClanLib-0.8.1.orig/Sources/GL/opengl.cpp	2006-10-26 21:55:01.000000000 -0400
+++ ClanLib-0.8.1/Sources/GL/opengl.cpp	2008-03-19 14:25:18.000000000 -0400
@@ -51,6 +51,8 @@
 #endif
 #endif
 
+#include <cstring>
+
 std::vector<std::string> CL_OpenGL::m_ignored_gl_extension;
 
 CL_ProcAddress *CL_OpenGL::get_proc_address(const std::string& function_name)
@@ -414,4 +416,4 @@
 #endif
 	
 	m_ignored_gl_extension.push_back(extension_name);
-}
\ No newline at end of file
+}
