--- pinot-0.76/pinot.desktop	2007-03-28 11:52:34.000000000 +0200
+++ pinot-0.76-p/pinot.desktop	2007-09-06 23:05:06.000000000 +0200
@@ -16,7 +16,7 @@
 Comment[sv]=Sök på webben och i dina dokument
 Exec=pinot
 StartupNotify=true
-Icon=pinot.png
+Icon=pinot
 Terminal=false
 Type=Application
 Categories=GTK;Core;Network;Filesystem;
--- pinot-0.76/pinot-dbus-daemon.desktop	2006-11-12 03:50:07.000000000 +0100
+++ pinot-0.76-p/pinot-dbus-daemon.desktop	2007-09-06 23:04:55.000000000 +0200
@@ -2,7 +2,7 @@
 Name=Pinot
 Comment=Search the Web and your documents
 Exec=pinot-dbus-daemon
-Icon=pinot.png
+Icon=pinot
 Terminal=false
 Type=Application
 Categories=
