Index: Search/Makefile.am
===================================================================
--- Search/Makefile.am	(Revision 954)
+++ Search/Makefile.am	(Arbeitskopie)
@@ -28,7 +28,6 @@
 noinst_LTLIBRARIES = libSearch.la
 endif
 
-bin_PROGRAMS = pinot-search
 
 libSearch_la_SOURCES = \
 	AbstractGenerator.cpp \
@@ -55,16 +54,25 @@
 	SOAPEnvNS.cpp
 endif
 
-pinot_search_SOURCES = \
-	pinot-search.cpp
+bin_PROGRAMS = pinot-search
 
 pinot_search_LDADD = -L../Utils -L../Tokenize -L../Collect -L../Index -LGoogle -Lxesam \
 	@SEARCH_LIBS@ -lIndex -lCollect -lTokenize -lUtils -lBasicUtils \
 	@GLIBMM_LIBS@ @INDEX_LIBS@ @SOAP_LIBS@ @XML_LIBS@ @GMIME_LIBS@ @HTTP_LIBS@ @MISC_LIBS@
 
+pinot_search_SOURCES = \
+        pinot-search.cpp
+
+if HAVE_SOAP
+pinot_search_DEPENDENCIES = libSearch.la libSearchSOAP.la
+else
+pinot_search_DEPENDENCIES = libSearch.la
+endif
+ 
 SOAPEnvC.cpp : SOAPEnv.h
 	soapcpp2 -pSOAPEnv SOAPEnv.h
 
+
 AM_CXXFLAGS = -I$(srcdir)/../Utils -I$(srcdir)/../Tokenize -I$(srcdir)/../Tokenize/filters \
 	-I$(srcdir)/../Collect -I$(srcdir)/../Index -IGoogle -Ixesam \
 	@HTTP_CFLAGS@ @XML_CFLAGS@ @INDEX_CFLAGS@ @SOAP_CFLAGS@ @GMIME_CFLAGS@ @GLIBMM_CFLAGS@
Index: Index/Makefile.am
===================================================================
--- Index/Makefile.am	(Revision 954)
+++ Index/Makefile.am	(Arbeitskopie)
@@ -12,8 +12,6 @@
 
 noinst_LTLIBRARIES = libIndex.la
 
-bin_PROGRAMS = pinot-index
-
 libIndex_la_SOURCES = \
 	DBusXapianIndex.cpp \
 	FilterWrapper.cpp \
@@ -23,13 +21,17 @@
 	XapianDatabaseFactory.cpp \
 	XapianIndex.cpp
 
-pinot_index_SOURCES = \
-	pinot-index.cpp
+bin_PROGRAMS = pinot-index
 
 pinot_index_LDADD = -L../Utils -L../Tokenize -L../Collect \
 	-lIndex -lCollect -lTokenize -lBasicUtils -lUtils \
 	@GLIBMM_LIBS@ @INDEX_LIBS@ @DBUS_LIBS@ @XML_LIBS@ @GMIME_LIBS@ @HTTP_LIBS@ @MISC_LIBS@
 
+pinot_index_SOURCES = \
+        pinot-index.cpp
+
+pinot_index_DEPENDENCIES = libIndex.la
+
 AM_CXXFLAGS = -I$(srcdir)/../Utils -I$(srcdir)/../Tokenize -I$(srcdir)/../Tokenize/filters \
 	-I$(srcdir)/../Collect @HTTP_CFLAGS@ @GMIME_CFLAGS@ @XML_CFLAGS@ @DBUS_CFLAGS@ @INDEX_CFLAGS@
 
