diff -upNr pinot-0.83.orign/IndexSearch/cjkv/CJKVTokenizer.cc pinot-0.83/IndexSearch/cjkv/CJKVTokenizer.cc
--- pinot-0.83.orign/IndexSearch/cjkv/CJKVTokenizer.cc	2008-02-26 14:16:53.000000000 +0100
+++ pinot-0.83/IndexSearch/cjkv/CJKVTokenizer.cc	2008-03-08 09:24:41.000000000 +0100
@@ -19,7 +19,7 @@
 
 #include <cctype>
 #include <cassert>
-
+#include <cstring>
 #include "CJKVTokenizer.h"
 
 #ifndef HAVE_UNICODE_H
diff -upNr pinot-0.83.orign/IndexSearch/Xapian/XapianEngine.cpp pinot-0.83/IndexSearch/Xapian/XapianEngine.cpp
--- pinot-0.83.orign/IndexSearch/Xapian/XapianEngine.cpp	2008-02-23 08:53:32.000000000 +0100
+++ pinot-0.83/IndexSearch/Xapian/XapianEngine.cpp	2008-03-08 09:23:17.000000000 +0100
@@ -21,6 +21,7 @@
 #include <unistd.h>
 #include <time.h>
 #include <string>
+#include <cstring>
 #include <vector>
 #include <iostream>
 #include <algorithm>
diff -upNr pinot-0.83.orign/UI/GTK2/src/pinot-dbus-daemon.cc pinot-0.83/UI/GTK2/src/pinot-dbus-daemon.cc
--- pinot-0.83.orign/UI/GTK2/src/pinot-dbus-daemon.cc	2008-02-21 07:46:44.000000000 +0100
+++ pinot-0.83/UI/GTK2/src/pinot-dbus-daemon.cc	2008-03-08 09:38:37.000000000 +0100
@@ -26,6 +26,7 @@
 #include <unistd.h>
 #include <iostream>
 #include <fstream>
+#include <cstring>
 #include <sigc++/sigc++.h>
 #include <glibmm.h>
 #include <glibmm/thread.h>

