diff -Naur pinot-0.98-old/IndexSearch/cjkv/CJKVTokenizer.h pinot-0.98/IndexSearch/cjkv/CJKVTokenizer.h
--- pinot-0.98-old/IndexSearch/cjkv/CJKVTokenizer.h	2011-11-19 05:53:22.857315171 +0000
+++ pinot-0.98/IndexSearch/cjkv/CJKVTokenizer.h	2011-11-19 05:54:09.354733872 +0000
@@ -25,7 +25,7 @@
 #ifdef HAVE_UNICODE_H
 #include <unicode.h>
 #else
-#include <glib/gunicode.h>
+#include <glib.h>
 #define unicode_char_t gunichar
 #endif