diff -ur a/connectpairs/connectpairs.cc b/connectpairs/connectpairs.cc
--- a/connectpairs/connectpairs.cc	2013-12-10 14:11:49.000000000 -0600
+++ b/connectpairs/connectpairs.cc	2014-09-28 04:09:15.624831649 -0500
@@ -85,7 +85,7 @@
 	unsigned k;
 
 	/** An estimate of the size of the genome. */
-	size_t genomeSize = 3000000000;
+	size_t genomeSize = 3000000000UL;
 
 	/** Write the de Bruijn graph to this file. */
 	static string graphPath;
