diff -Naur tse3-0.3.0-orig/src/tse3/app/Application.h tse3-0.3.0/src/tse3/app/Application.h
--- tse3-0.3.0-orig/src/tse3/app/Application.h	2005-05-25 15:33:05.000000000 +0200
+++ tse3-0.3.0/src/tse3/app/Application.h	2005-09-05 01:10:59.000000000 +0200
@@ -35,12 +35,12 @@
     namespace Cmd
     {
         class CommandHistory;
-    };
+    }
 
     namespace Ins
     {
         class Destination;
-    };
+    }
 
     /**
      * The App namespace contains classes that use the @ref TSE3
diff -Naur tse3-0.3.0-orig/src/tse3/app/Record.h tse3-0.3.0/src/tse3/app/Record.h
--- tse3-0.3.0-orig/src/tse3/app/Record.h	2005-05-25 15:33:05.000000000 +0200
+++ tse3-0.3.0/src/tse3/app/Record.h	2005-09-05 01:11:11.000000000 +0200
@@ -36,7 +36,7 @@
     namespace Cmd
     {
         class CommandHistory;
-    };
+    }
 
     namespace App
     {
diff -Naur tse3-0.3.0-orig/src/tse3/listen/Phrase.h tse3-0.3.0/src/tse3/listen/Phrase.h
--- tse3-0.3.0-orig/src/tse3/listen/Phrase.h	2005-05-25 15:33:10.000000000 +0200
+++ tse3-0.3.0/src/tse3/listen/Phrase.h	2005-09-05 01:08:04.000000000 +0200
@@ -17,6 +17,8 @@
 #ifndef TSE3_LISTEN_PHRASE_H
 #define TSE3_LISTEN_PHRASE_H
 
+#include "tse3/listen/PhraseEdit.h"
+
 namespace TSE3
 {
     class Phrase;
diff -Naur tse3-0.3.0-orig/src/tse3/Midi.h tse3-0.3.0/src/tse3/Midi.h
--- tse3-0.3.0-orig/src/tse3/Midi.h	2005-05-25 15:33:11.000000000 +0200
+++ tse3-0.3.0/src/tse3/Midi.h	2005-09-05 00:18:22.000000000 +0200
@@ -819,6 +819,6 @@
                 const MidiEvent &e1;
         };
     };
-};
+}
 
 #endif
diff -Naur tse3-0.3.0-orig/src/tse3/Mutex.h tse3-0.3.0/src/tse3/Mutex.h
--- tse3-0.3.0-orig/src/tse3/Mutex.h	2005-05-25 15:33:11.000000000 +0200
+++ tse3-0.3.0/src/tse3/Mutex.h	2005-09-05 01:13:50.000000000 +0200
@@ -315,7 +315,7 @@
 #endif
                 }
         };
-    };
+    }
 }
 
 #endif
diff -Naur tse3-0.3.0-orig/src/tse3/Notifier.h tse3-0.3.0/src/tse3/Notifier.h
--- tse3-0.3.0-orig/src/tse3/Notifier.h	2005-05-25 15:33:11.000000000 +0200
+++ tse3-0.3.0/src/tse3/Notifier.h	2005-09-05 00:19:02.000000000 +0200
@@ -290,7 +290,7 @@
                 const p4_type       &p4;                                 // (*)
         };
 
-    };
+    }
 
     /**************************************************************************
      * Public Notifier framework API
@@ -367,7 +367,7 @@
              */
             typedef Listener<interface_type> listener_type;
 
-            friend class listener_type;
+            friend class Listener<interface_type>;
 
         protected:
 
@@ -602,7 +602,7 @@
                 if (notifiers.erase(notifier)) notifier->detach(this);
             }
 
-            friend class notifier_type;
+            friend class Notifier<interface_type>;
 
         protected:
 
diff -Naur tse3-0.3.0-orig/src/tse3/util/MulDiv.h tse3-0.3.0/src/tse3/util/MulDiv.h
--- tse3-0.3.0-orig/src/tse3/util/MulDiv.h	2005-05-25 15:33:08.000000000 +0200
+++ tse3-0.3.0/src/tse3/util/MulDiv.h	2005-09-05 00:18:46.000000000 +0200
@@ -34,7 +34,7 @@
          * @return val * num / div (which coped with integer overflows)
          */
         extern int muldiv(int val, int num, int div);
-    };
+    }
 }
 
 #endif
