diff -Naur tse3-0.3.0-orig/configure tse3-0.3.0/configure
--- tse3-0.3.0-orig/configure	2005-05-25 15:37:20.000000000 +0200
+++ tse3-0.3.0/configure	2005-10-23 11:22:15.000000000 +0200
@@ -19857,7 +19857,7 @@
 /* end confdefs.h.  */
 
 #include <$alsa_header_prefix/asoundlib.h>
-int main() { return SND_LIB_MINOR != 5 && SND_LIB_MINOR != 9; }
+int main() { return SND_LIB_MINOR != 5 && SND_LIB_MINOR != 9 && SND_LIB_MINOR != 0; }
 
 _ACEOF
 rm -f conftest$ac_exeext
@@ -19965,6 +19965,13 @@
 _ACEOF
 
     fi
+    if test "$alsa_version" = "1.0"; then
+
+cat >>confdefs.h <<\_ACEOF
+#define TSE3_WITH_ALSA_1_0_X 1
+_ACEOF
+
+    fi
   fi
 
   LDFLAGS="$ac_ldflags_save"
