diff -Naur djvulibre-3.5.17/configure djvulibre-3.5.17-p/configure
--- djvulibre-3.5.17/configure	2006-03-23 17:37:23.000000000 +0100
+++ djvulibre-3.5.17-p/configure	2006-10-06 17:50:18.000000000 +0200
@@ -26970,7 +26970,7 @@
 else
   echo "$as_me:$LINENO: checking g++ version" >&5
 echo $ECHO_N "checking g++ version... $ECHO_C" >&6
-  cxxver="`$CXX --version | head -1 |
+  cxxver="`$CXX --version | head -n 1 |
              sed -e 's/^.*[^0-9.]\([2-4]\.[0-9.]*\).*$/\1/'`"
   echo "$as_me:$LINENO: result: $cxxver" >&5
 echo "${ECHO_T}$cxxver" >&6
diff -Naur djvulibre-3.5.17/configure.ac djvulibre-3.5.17-p/configure.ac
--- djvulibre-3.5.17/configure.ac	2006-03-23 17:37:24.000000000 +0100
+++ djvulibre-3.5.17-p/configure.ac	2006-10-06 17:50:18.000000000 +0200
@@ -326,7 +326,7 @@
 else
   AC_MSG_CHECKING([g++ version]) 
 changequote(<<, >>)dnl
-  cxxver="`$CXX --version | head -1 | 
+  cxxver="`$CXX --version | head -n 1 | 
              sed -e 's/^.*[^0-9.]\([2-4]\.[0-9.]*\).*$/\1/'`"
 changequote([, ])dnl 
   AC_MSG_RESULT($cxxver)
