diff -Naur djvulibre-3.5.17/gui/bin2cpp/Makefile.in djvulibre-3.5.17-p/gui/bin2cpp/Makefile.in
--- djvulibre-3.5.17/gui/bin2cpp/Makefile.in	2004-06-11 18:42:25.000000000 +0200
+++ djvulibre-3.5.17-p/gui/bin2cpp/Makefile.in	2006-10-06 18:40:03.000000000 +0200
@@ -45,10 +45,10 @@
 LIBDJVU = ${top_builddir}/libdjvu/libdjvulibre.la
 
 INCS = -I${top_builddir} -I${top_srcdir}/libdjvu -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS}  ${JPEG_CFLAGS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${JPEG_LIBS} ${THREAD_LIBS} @LIBS@
-CFLAGS = @CPPFLAGS@ @CFLAGS@ ${FLAGS}
-CXXFLAGS = @CPPFLAGS@ @CXXFLAGS@ ${FLAGS}
+CFLAGS += @CPPFLAGS@  ${FLAGS}
+CXXFLAGS += @CPPFLAGS@  ${FLAGS}
 
 PROGRAMS = bin2cpp
 
diff -Naur djvulibre-3.5.17/gui/djview/Makefile.in djvulibre-3.5.17-p/gui/djview/Makefile.in
--- djvulibre-3.5.17/gui/djview/Makefile.in	2004-08-24 19:58:55.000000000 +0200
+++ djvulibre-3.5.17-p/gui/djview/Makefile.in	2006-10-06 18:39:18.000000000 +0200
@@ -157,13 +157,13 @@
 
 INCS = -I${top_builddir} -I${top_srcdir}/libdjvu ${SRCINCS} -I.
 MISCFLAGS = ${JPEG_CFLAGS} ${THREAD_CFLAGS} ${QT_CFLAGS} ${X_CFLAGS}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${MISCFLAGS}
+FLAGS = ${DEFS} ${INCS} ${MISCFLAGS}
 
 LIBS= @LDFLAGS@ ${QT_LIBS} ${X_LIBS} ${JPEG_LIBS} ${THREAD_LIBS} @LIBS@
 
-CFLAGS = @CPPFLAGS@ @CFLAGS@ ${FLAGS}
+CFLAGS += @CPPFLAGS@  ${FLAGS}
 
-CXXFLAGS = @CPPFLAGS@ @CXXFLAGS@ ${CXXRPOFLAGS} ${FLAGS}
+CXXFLAGS += @CPPFLAGS@  ${CXXRPOFLAGS} ${FLAGS}
 
 djview: ${OBJS}
 	${LIBTOOL} --mode=link \
diff -Naur djvulibre-3.5.17/gui/nsdejavu/Makefile.in djvulibre-3.5.17-p/gui/nsdejavu/Makefile.in
--- djvulibre-3.5.17/gui/nsdejavu/Makefile.in	2004-08-24 19:58:55.000000000 +0200
+++ djvulibre-3.5.17-p/gui/nsdejavu/Makefile.in	2006-10-06 18:38:12.000000000 +0200
@@ -49,7 +49,7 @@
  -I${top_srcdir}/gui/shared/res \
  -I${top_srcdir}/gui/npsdk 
 
-CFLAGS = @CPPFLAGS@ ${DEFS} ${OPTS} ${INCS} ${NSDEJAVU_CFLAGS} 
+CFLAGS += @CPPFLAGS@ ${DEFS} ${INCS} ${NSDEJAVU_CFLAGS} 
 
 LIBS = @LDFLAGS@ ${NSDEJAVU_LIBS} 
 
diff -Naur djvulibre-3.5.17/libdjvu/Makefile.in djvulibre-3.5.17-p/libdjvu/Makefile.in
--- djvulibre-3.5.17/libdjvu/Makefile.in	2005-12-04 02:36:26.000000000 +0100
+++ djvulibre-3.5.17-p/libdjvu/Makefile.in	2006-10-06 18:37:15.000000000 +0200
@@ -51,10 +51,10 @@
 LIBDJVU_VER = @LIBDJVU_VER@
 
 INCS = -I${top_builddir} -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${JPEG_LIBS} ${THREAD_LIBS} @LIBS@
-CFLAGS = @CPPFLAGS@ @CFLAGS@ ${FLAGS}
-CXXFLAGS = @CPPFLAGS@ @CXXFLAGS@ ${CXXPICFLAGS} ${CXXRPOFLAGS} ${FLAGS}
+CFLAGS += @CPPFLAGS@  ${FLAGS}
+CXXFLAGS += @CPPFLAGS@  ${CXXPICFLAGS} ${CXXRPOFLAGS} ${FLAGS}
 
 all: ${LIBDJVU_LA} 
 
diff -Naur djvulibre-3.5.17/tools/jb2cmp/Makefile.in djvulibre-3.5.17-p/tools/jb2cmp/Makefile.in
--- djvulibre-3.5.17/tools/jb2cmp/Makefile.in	2005-03-13 20:17:32.000000000 +0100
+++ djvulibre-3.5.17-p/tools/jb2cmp/Makefile.in	2006-10-06 18:36:41.000000000 +0200
@@ -44,10 +44,10 @@
 THREAD_CFLAGS = @THREAD_CFLAGS@
 
 INCS = -I${top_builddir} -I${top_srcdir} -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS}  ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${THREAD_LIBS} @LIBS@
-CFLAGS = @CPPFLAGS@ @CFLAGS@ ${FLAGS}
-CXXFLAGS = @CPPFLAGS@ @CXXFLAGS@ ${CXXRPOFLAGS} ${FLAGS}
+CFLAGS += @CPPFLAGS@  ${FLAGS}
+CXXFLAGS += @CPPFLAGS@  ${CXXRPOFLAGS} ${FLAGS}
 
 
 OBJS = classify.o cuts.o frames.o patterns.o
diff -Naur djvulibre-3.5.17/tools/Makefile.in djvulibre-3.5.17-p/tools/Makefile.in
--- djvulibre-3.5.17/tools/Makefile.in	2005-08-23 18:49:23.000000000 +0200
+++ djvulibre-3.5.17-p/tools/Makefile.in	2006-10-06 18:32:54.000000000 +0200
@@ -50,11 +50,11 @@
 LIBDJVU = ${top_builddir}/libdjvu/libdjvulibre.la
 
 INCS = -I${top_builddir} -I${top_srcdir} -I${top_srcdir}/libdjvu -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${JPEG_CFLAGS} ${TIFF_CFLAGS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS}  ${JPEG_CFLAGS} ${TIFF_CFLAGS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${JPEG_LIBS} ${THREAD_LIBS} @LIBS@
 MORELIBS= @LDFLAGS@ ${JPEG_LIBS} ${TIFF_LIBS} ${THREAD_LIBS} @LIBS@
-CFLAGS = @CPPFLAGS@ @CFLAGS@ ${FLAGS}
-CXXFLAGS = @CPPFLAGS@ @CXXFLAGS@ ${CXXRPOFLAGS} ${FLAGS}
+CFLAGS += @CPPFLAGS@  ${FLAGS}
+CXXFLAGS += @CPPFLAGS@  ${CXXRPOFLAGS} ${FLAGS}
 
 SUBDIRS = jb2cmp
 
diff -Naur djvulibre-3.5.17/xmltools/Makefile.in djvulibre-3.5.17-p/xmltools/Makefile.in
--- djvulibre-3.5.17/xmltools/Makefile.in	2004-08-24 19:58:56.000000000 +0200
+++ djvulibre-3.5.17-p/xmltools/Makefile.in	2006-10-06 18:32:23.000000000 +0200
@@ -48,10 +48,10 @@
 LIBDJVU = ${top_builddir}/libdjvu/libdjvulibre.la
 
 INCS = -I${top_builddir} -I${top_srcdir}/libdjvu -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS}  ${JPEG_CFLAGS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${JPEG_LIBS} ${THREAD_LIBS} @LIBS@
-CFLAGS = @CPPFLAGS@ @CFLAGS@ ${FLAGS}
-CXXFLAGS = @CPPFLAGS@ @CXXFLAGS@ ${CXXRPOFLAGS} ${FLAGS}
+CFLAGS += @CPPFLAGS@ ${FLAGS}
+CXXFLAGS += @CPPFLAGS@ ${CXXRPOFLAGS} ${FLAGS}
 
 PROGRAMS = djvutoxml djvuxmlparser 
 
