--- key-mapper-1.0.0/setup.py	2021-05-08 20:28:13.000000000 +0200
+++ key-mapper-1.0.0.new/setup.py	2021-08-13 09:49:02.397577413 +0200
@@ -69,10 +69,10 @@
         ('/usr/share/key-mapper/', glob.glob('data/*')),
         ('/usr/share/applications/', ['data/key-mapper.desktop']),
         ('/usr/share/polkit-1/actions/', ['data/key-mapper.policy']),
-        ('/usr/lib/systemd/system', ['data/key-mapper.service']),
+        ('/lib/systemd/system', ['data/key-mapper.service']),
         ('/etc/dbus-1/system.d/', ['data/keymapper.Control.conf']),
         ('/etc/xdg/autostart/', ['data/key-mapper-autoload.desktop']),
-        ('/usr/lib/udev/rules.d', ['data/key-mapper.rules']),
+        ('/lib/udev/rules.d', ['data/key-mapper.rules']),
         ('/usr/bin/', ['bin/key-mapper-gtk']),
         ('/usr/bin/', ['bin/key-mapper-service']),
         ('/usr/bin/', ['bin/key-mapper-control']),
--- key-mapper-1.0.0/data/key-mapper.rules	2021-05-08 20:28:13.000000000 +0200
+++ key-mapper-1.0.0.new/data/key-mapper.rules	2021-08-13 09:52:40.409260311 +0200
@@ -1,3 +1,3 @@
 # udevadm monitor --property
 # udevadm info --query=all --name=/dev/input/event3
-ACTION=="add", SUBSYSTEM=="input", RUN+="/bin/key-mapper-control --command autoload --device $env{DEVNAME}"
+ACTION=="add", SUBSYSTEM=="input", RUN+="/usr/bin/key-mapper-control --command autoload --device $env{DEVNAME}"
