diff -Naur openvpn-2.0/config.h.in openvpn-2.0.1_rc4/config.h.in
--- openvpn-2.0/config.h.in	2005-04-18 02:01:40.000000000 +0000
+++ openvpn-2.0.1_rc4/config.h.in	2005-06-15 16:30:45.000000000 +0000
@@ -378,6 +378,9 @@
 /* Path to iproute tool */
 #undef IPROUTE_PATH
 
+/* Use lzo/ directory prefix for LZO header files (for LZO 2.0) */
+#undef LZO_HEADER_DIR
+
 /* Name of package */
 #undef PACKAGE
 
diff -Naur openvpn-2.0/configure openvpn-2.0.1_rc4/configure
--- openvpn-2.0/configure	2005-04-18 02:01:39.000000000 +0000
+++ openvpn-2.0.1_rc4/configure	2005-06-15 16:30:44.000000000 +0000
@@ -9287,8 +9287,129 @@
 
 
 if test "$LZO" = "yes"; then
+   LZO_H=""
    { echo "$as_me:$LINENO: checking for LZO Library and Header files..." >&5
 echo "$as_me: checking for LZO Library and Header files..." >&6;}
+   if test "${ac_cv_header_lzo_lzo1x_h+set}" = set; then
+  echo "$as_me:$LINENO: checking for lzo/lzo1x.h" >&5
+echo $ECHO_N "checking for lzo/lzo1x.h... $ECHO_C" >&6
+if test "${ac_cv_header_lzo_lzo1x_h+set}" = set; then
+  echo $ECHO_N "(cached) $ECHO_C" >&6
+fi
+echo "$as_me:$LINENO: result: $ac_cv_header_lzo_lzo1x_h" >&5
+echo "${ECHO_T}$ac_cv_header_lzo_lzo1x_h" >&6
+else
+  # Is the header compilable?
+echo "$as_me:$LINENO: checking lzo/lzo1x.h usability" >&5
+echo $ECHO_N "checking lzo/lzo1x.h usability... $ECHO_C" >&6
+cat >conftest.$ac_ext <<_ACEOF
+/* confdefs.h.  */
+_ACEOF
+cat confdefs.h >>conftest.$ac_ext
+cat >>conftest.$ac_ext <<_ACEOF
+/* end confdefs.h.  */
+$ac_includes_default
+#include <lzo/lzo1x.h>
+_ACEOF
+rm -f conftest.$ac_objext
+if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
+  (eval $ac_compile) 2>conftest.er1
+  ac_status=$?
+  grep -v '^ *+' conftest.er1 >conftest.err
+  rm -f conftest.er1
+  cat conftest.err >&5
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); } &&
+	 { ac_try='test -z "$ac_c_werror_flag"
+			 || test ! -s conftest.err'
+  { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
+  (eval $ac_try) 2>&5
+  ac_status=$?
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); }; } &&
+	 { ac_try='test -s conftest.$ac_objext'
+  { (eval echo "$as_me:$LINENO: \"$ac_try\"") >&5
+  (eval $ac_try) 2>&5
+  ac_status=$?
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); }; }; then
+  ac_header_compiler=yes
+else
+  echo "$as_me: failed program was:" >&5
+sed 's/^/| /' conftest.$ac_ext >&5
+
+ac_header_compiler=no
+fi
+rm -f conftest.err conftest.$ac_objext conftest.$ac_ext
+echo "$as_me:$LINENO: result: $ac_header_compiler" >&5
+echo "${ECHO_T}$ac_header_compiler" >&6
+
+# Is the header present?
+echo "$as_me:$LINENO: checking lzo/lzo1x.h presence" >&5
+echo $ECHO_N "checking lzo/lzo1x.h presence... $ECHO_C" >&6
+cat >conftest.$ac_ext <<_ACEOF
+/* confdefs.h.  */
+_ACEOF
+cat confdefs.h >>conftest.$ac_ext
+cat >>conftest.$ac_ext <<_ACEOF
+/* end confdefs.h.  */
+#include <lzo/lzo1x.h>
+_ACEOF
+if { (eval echo "$as_me:$LINENO: \"$ac_cpp conftest.$ac_ext\"") >&5
+  (eval $ac_cpp conftest.$ac_ext) 2>conftest.er1
+  ac_status=$?
+  grep -v '^ *+' conftest.er1 >conftest.err
+  rm -f conftest.er1
+  cat conftest.err >&5
+  echo "$as_me:$LINENO: \$? = $ac_status" >&5
+  (exit $ac_status); } >/dev/null; then
+  if test -s conftest.err; then
+    ac_cpp_err=$ac_c_preproc_warn_flag
+    ac_cpp_err=$ac_cpp_err$ac_c_werror_flag
+  else
+    ac_cpp_err=
+  fi
+else
+  ac_cpp_err=yes
+fi
+if test -z "$ac_cpp_err"; then
+  ac_header_preproc=yes
+else
+  echo "$as_me: failed program was:" >&5
+sed 's/^/| /' conftest.$ac_ext >&5
+
+  ac_header_preproc=no
+fi
+rm -f conftest.err conftest.$ac_ext
+echo "$as_me:$LINENO: result: $ac_header_preproc" >&5
+echo "${ECHO_T}$ac_header_preproc" >&6
+
+# So?  What about this header?
+case $ac_header_compiler:$ac_header_preproc:$ac_c_preproc_warn_flag in
+  yes:no: )
+    { echo "$as_me:$LINENO: WARNING: lzo/lzo1x.h: accepted by the compiler, rejected by the preprocessor!" >&5
+echo "$as_me: WARNING: lzo/lzo1x.h: accepted by the compiler, rejected by the preprocessor!" >&2;}
+    { echo "$as_me:$LINENO: WARNING: lzo/lzo1x.h: proceeding with the compiler's result" >&5
+echo "$as_me: WARNING: lzo/lzo1x.h: proceeding with the compiler's result" >&2;}
+    ac_header_preproc=yes
+    ;;
+  no:yes:* )
+    ;;
+esac
+echo "$as_me:$LINENO: checking for lzo/lzo1x.h" >&5
+echo $ECHO_N "checking for lzo/lzo1x.h... $ECHO_C" >&6
+if test "${ac_cv_header_lzo_lzo1x_h+set}" = set; then
+  echo $ECHO_N "(cached) $ECHO_C" >&6
+else
+  ac_cv_header_lzo_lzo1x_h=$ac_header_preproc
+fi
+echo "$as_me:$LINENO: result: $ac_cv_header_lzo_lzo1x_h" >&5
+echo "${ECHO_T}$ac_cv_header_lzo_lzo1x_h" >&6
+
+fi
+if test $ac_cv_header_lzo_lzo1x_h = yes; then
+   LZO_H="2"
+else
    if test "${ac_cv_header_lzo1x_h+set}" = set; then
   echo "$as_me:$LINENO: checking for lzo1x.h" >&5
 echo $ECHO_N "checking for lzo1x.h... $ECHO_C" >&6
@@ -9407,7 +9528,17 @@
 
 fi
 if test $ac_cv_header_lzo1x_h = yes; then
-  echo "$as_me:$LINENO: checking for lzo1x_1_15_compress in -llzo" >&5
+   LZO_H="1"
+fi
+
+
+
+fi
+
+
+
+   if test -n "$LZO_H"; then
+     echo "$as_me:$LINENO: checking for lzo1x_1_15_compress in -llzo" >&5
 echo $ECHO_N "checking for lzo1x_1_15_compress in -llzo... $ECHO_C" >&6
 if test "${ac_cv_lib_lzo_lzo1x_1_15_compress+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
@@ -9481,28 +9612,30 @@
 #define USE_LZO 1
 _ACEOF
 
+	 if test "$LZO_H" = "2"; then
+
+cat >>confdefs.h <<\_ACEOF
+#define LZO_HEADER_DIR 1
+_ACEOF
+
+         fi
 
 else
-  { { echo "$as_me:$LINENO: error: LZO headers were found but LZO library was not found." >&5
-echo "$as_me: error: LZO headers were found but LZO library was not found." >&2;}
+  { { echo "$as_me:$LINENO: error: LZO headers were found but LZO library was not found" >&5
+echo "$as_me: error: LZO headers were found but LZO library was not found" >&2;}
    { (exit 1); exit 1; }; }
 
 fi
 
-else
-
-          echo "$as_me:$LINENO: result: LZO library and headers not found." >&5
-echo "${ECHO_T}LZO library and headers not found." >&6
-	  echo "$as_me:$LINENO: result: LZO library available from http://www.oberhumer.com/opensource/lzo/" >&5
+   else
+     echo "$as_me:$LINENO: result: LZO headers were not found" >&5
+echo "${ECHO_T}LZO headers were not found" >&6
+     echo "$as_me:$LINENO: result: LZO library available from http://www.oberhumer.com/opensource/lzo/" >&5
 echo "${ECHO_T}LZO library available from http://www.oberhumer.com/opensource/lzo/" >&6
-	  { { echo "$as_me:$LINENO: error: Or try ./configure --disable-lzo" >&5
+     { { echo "$as_me:$LINENO: error: Or try ./configure --disable-lzo" >&5
 echo "$as_me: error: Or try ./configure --disable-lzo" >&2;}
    { (exit 1); exit 1; }; }
-
-
-fi
-
-
+   fi
 fi
 
 
diff -Naur openvpn-2.0/configure.ac openvpn-2.0.1_rc4/configure.ac
--- openvpn-2.0/configure.ac	2005-04-18 02:01:31.000000000 +0000
+++ openvpn-2.0.1_rc4/configure.ac	2005-06-15 16:30:37.000000000 +0000
@@ -461,21 +461,29 @@
 dnl
 
 if test "$LZO" = "yes"; then
+   LZO_H=""
    AC_CHECKING([for LZO Library and Header files])
-   AC_CHECK_HEADER(lzo1x.h, 
-        [AC_CHECK_LIB(lzo, lzo1x_1_15_compress,
-	    [
-	       OPENVPN_ADD_LIBS(-llzo)
-	       AC_DEFINE(USE_LZO, 1, [Use LZO compression library])
-	    ],
-	    [AC_MSG_ERROR([LZO headers were found but LZO library was not found.])]
-	)],
-	[
-          AC_MSG_RESULT([LZO library and headers not found.])
-	  AC_MSG_RESULT([LZO library available from http://www.oberhumer.com/opensource/lzo/])
-	  AC_MSG_ERROR([Or try ./configure --disable-lzo])
-	]
+   AC_CHECK_HEADER(lzo/lzo1x.h, 
+        [ LZO_H="2" ],
+	[ AC_CHECK_HEADER(lzo1x.h, [ LZO_H="1" ])]
    )
+
+   if test -n "$LZO_H"; then
+     AC_CHECK_LIB(lzo, lzo1x_1_15_compress,
+       [
+         OPENVPN_ADD_LIBS(-llzo)
+	 AC_DEFINE(USE_LZO, 1, [Use LZO compression library])
+	 if test "$LZO_H" = "2"; then
+	   AC_DEFINE(LZO_HEADER_DIR, 1, [Use lzo/ directory prefix for LZO header files (for LZO 2.0)])
+         fi
+       ],
+       [AC_MSG_ERROR([LZO headers were found but LZO library was not found])]
+     )
+   else
+     AC_MSG_RESULT([LZO headers were not found])
+     AC_MSG_RESULT([LZO library available from http://www.oberhumer.com/opensource/lzo/])
+     AC_MSG_ERROR([Or try ./configure --disable-lzo])
+   fi
 fi
 
 dnl
 	      msg (M_WARN, "Signal received from management interface, exiting");
diff -Naur openvpn-2.0/lzo.c openvpn-2.0.1_rc4/lzo.c
--- openvpn-2.0/lzo.c	2005-04-11 03:43:55.000000000 +0000
+++ openvpn-2.0.1_rc4/lzo.c	2005-06-14 15:54:24.000000000 +0000
@@ -135,7 +135,7 @@
 	      struct lzo_compress_workspace *lzowork,
 	      const struct frame* frame)
 {
-  int zlen = 0;
+  lzo_uint zlen = 0;
   int err;
   bool compressed = false;
 
@@ -191,7 +191,7 @@
 		struct lzo_compress_workspace *lzowork,
 		const struct frame* frame)
 {
-  int zlen = EXPANDED_SIZE (frame);
+  lzo_uint zlen = EXPANDED_SIZE (frame);
   uint8_t c;		/* flag indicating whether or not our peer compressed */
   int err;
 
diff -Naur openvpn-2.0/lzo.h openvpn-2.0.1_rc4/lzo.h
--- openvpn-2.0/lzo.h	2005-04-11 03:43:55.000000000 +0000
+++ openvpn-2.0.1_rc4/lzo.h	2005-06-14 09:12:37.000000000 +0000
@@ -24,8 +24,14 @@
 
 #ifdef USE_LZO
 
+#ifdef LZO_HEADER_DIR
+#include "lzo/lzoutil.h"
+#include "lzo/lzo1x.h"
+#else
 #include "lzoutil.h"
 #include "lzo1x.h"
+#endif
+
 #include "buffer.h"
 #include "mtu.h"
 #include "common.h"
@@ -44,7 +50,7 @@
 #define LZO_WORKSPACE	LZO1X_1_15_MEM_COMPRESS
 #define LZO_DECOMPRESS  lzo1x_decompress_safe
 
-#define LZO_EXTRA_BUFFER(len) ((len)/64 + 16 + 3)	/* LZO worst case size expansion. */
+#define LZO_EXTRA_BUFFER(len) ((len)/8 + 128 + 3)	/* LZO 2.0 worst case size expansion. */
 
 /*
  * Don't try to compress any packet smaller than this.
