--- gsettings-desktop-schemas-0.1.7/schemas/org.gnome.desktop.interface.gschema.xml.in.in.old	2011-02-25 22:18:10.000000000 +0100
+++ gsettings-desktop-schemas-0.1.7/schemas/org.gnome.desktop.interface.gschema.xml.in.in	2011-02-25 22:20:49.000000000 +0100
@@ -1,7 +1,14 @@
 <schemalist gettext-domain="gsettings-desktop-schemas">
-  <schema id="org.gnome.desktop.interface" path="/org/gnome/desktop/interface/">
+ <schema id="org.gnome.desktop.interface" path="/org/gnome/desktop/interface/">
+<key name="accessibility" type="b">
+      <default>false</default>
+       <summary>Enable Accessibility</summary>
+       <description>
+         Whether Applications should have accessibility support.
+       </description>
+     </key>
     <key name="toolkit-accessibility" type="b">
-      <default>true</default>
+      <default>false</default>
       <_summary>Enable Toolkit Accessibility</_summary>
       <_description>
         Whether toolkits should load accessibility related modules.
