--- gsettings-desktop-schemas-2.91.91/schemas/org.gnome.desktop.interface.gschema.xml.in.in.old	2011-03-03 13:42:00.000000000 +0100
+++ gsettings-desktop-schemas-2.91.91/schemas/org.gnome.desktop.interface.gschema.xml.in.in	2011-03-09 13:19:34.000000000 +0100
@@ -1,5 +1,12 @@
 <schemalist gettext-domain="gsettings-desktop-schemas">
   <schema id="org.gnome.desktop.interface" path="/org/gnome/desktop/interface/">
+     <key name="accessibility" type="b">
+      <default>false</default>
+       <summary>Enable Accessibility</summary>
+       <description>
+         Whether Applications should have accessibility support.
+       </description>
+     </key>
     <key name="toolkit-accessibility" type="b">
       <default>false</default>
       <_summary>Enable Toolkit Accessibility</_summary>
@@ -39,7 +46,7 @@
       </_description>
     </key>
     <key name="menus-have-icons" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Menus Have Icons</_summary>
       <_description>
         Whether menus may display an icon next to a menu entry.
@@ -97,7 +104,7 @@
       </_description>
     </key>
     <key name="icon-theme" type="s">
-      <default>'gnome'</default>
+      <default>'gnome'</default>
       <_summary>Icon Theme</_summary>
       <_description>
         Icon theme to use for the panel, nautilus etc.
@@ -192,7 +199,7 @@
       </_description>
     </key>
     <key name="cursor-theme" type="s">
-      <default>'Adwaita'</default>
+      <default>'Adwaita'</default>
       <_summary>Cursor theme</_summary>
       <_description>Cursor theme name. Used only by Xservers that support the Xcursor extension.</_description>
     </key>
