--- platform/gtk-x11/onetime.py.orig	2007-05-20 21:50:54.000000000 -0400
+++ platform/gtk-x11/onetime.py	2007-05-20 21:51:01.000000000 -0400
@@ -86,7 +86,7 @@
     # we can delete the low-level name here because these objects
     # are guaranteed to exist only once for each bus name
     def __del__(self):
-        dbus_bindings.bus_release_name(self._bus.get_connection(), self._name)
+        dbus.dbus_bindings.bus_release_name(self._bus.get_connection(), self._name)
         pass
 
     def get_bus(self):
@@ -109,7 +109,8 @@
     def __init__(self):
         bus = dbus.SessionBus()
         bus_name = BusNameFlags('org.participatoryculture.dtv.onetime', bus=bus, flags=dbus.dbus_bindings.NAME_FLAG_DO_NOT_QUEUE)
-        dbus.service.Object.__init__(self, bus_name, '/org/participatoryculture/dtv/OneTime')
+        dbus.service.Object.__init__(self, bus_name=bus_name,
+                                     object_path='/org/participatoryculture/dtv/OneTime')
 
     @dbus.service.method('org.participatoryculture.dtv.OneTimeIface')
     def HandleArgs (self, args):
