--- sharutils-4.15.2/lib/fseeko.c	2015-05-30 17:34:54.000000000 +0200
+++ sharutils-4.15.2.new/lib/fseeko.c	2019-06-20 14:34:16.517187555 +0200
@@ -100,7 +100,7 @@
 #elif defined EPLAN9                /* Plan9 */
   if (fp->rp == fp->buf
       && fp->wp == fp->buf)
-#elif FUNC_FFLUSH_STDIN < 0 && 200809 <= _POSIX_VERSION
+#elif -1 < 0 && 200809 <= _POSIX_VERSION
   /* Cross-compiling to some other system advertising conformance to
      POSIX.1-2008 or later.  Assume fseeko and fflush work as advertised.
      If this assumption is incorrect, please report the bug to
