diff -Naur lyx-1.5.2/config/common.am lyx-1.5.2-p/config/common.am
--- lyx-1.5.2/config/common.am	2006-04-20 22:44:05.000000000 +0200
+++ lyx-1.5.2-p/config/common.am	2007-11-07 23:53:12.000000000 +0100
@@ -36,10 +36,10 @@
 BOOST_IOSTREAMS = $(top_builddir)/boost/libs/iostreams/src/libboost_iostreams.la
 else
 BOOST_INCLUDES =
-BOOST_FILESYSTEM = -lboost_filesystem
-BOOST_REGEX = -lboost_regex
-BOOST_SIGNALS = -lboost_signals
-BOOST_IOSTREAMS = -lboost_iostreams
+BOOST_FILESYSTEM = -lboost_filesystem-gcc42-mt
+BOOST_REGEX = -lboost_regex-gcc42-mt
+BOOST_SIGNALS = -lboost_signals-gcc42-mt
+BOOST_IOSTREAMS = -lboost_iostreams-gcc42-mt
 endif
 
 LIBS =
