diff --color -Naur lyx-2.4.3/src/Compare.cpp lyx-2.4.3.new/src/Compare.cpp
--- lyx-2.4.3/src/Compare.cpp	2025-01-09 20:37:38.000000000 +0100
+++ lyx-2.4.3.new/src/Compare.cpp	2025-05-26 17:20:16.042824418 +0200
@@ -885,7 +885,6 @@
 		mergeParagraph(dest_buf_->params(), *dest_pars_, pit);
 }
 
+} // namespace lyx
 
 #include "moc_Compare.cpp"
-
-} // namespace lyx
diff --color -Naur lyx-2.4.3/src/support/Systemcall.cpp lyx-2.4.3.new/src/support/Systemcall.cpp
--- lyx-2.4.3/src/support/Systemcall.cpp	2025-01-09 20:37:38.000000000 +0100
+++ lyx-2.4.3.new/src/support/Systemcall.cpp	2025-05-26 17:20:16.042966760 +0200
@@ -706,10 +706,8 @@
 	}
 }
 
-
+} // namespace support
+} // namespace lyx
 
 #include "moc_SystemcallPrivate.cpp"
 #endif
-
-} // namespace support
-} // namespace lyx
