diff -Naur lyx-2.3.0/src/frontends/qt4/GuiDocument.cpp lyx-2.3.0-p/src/frontends/qt4/GuiDocument.cpp
--- lyx-2.3.0/src/frontends/qt4/GuiDocument.cpp	2018-02-25 01:11:18.000000000 +0100
+++ lyx-2.3.0-p/src/frontends/qt4/GuiDocument.cpp	2018-06-14 22:17:09.894559104 +0200
@@ -76,6 +76,7 @@
 #include <QScrollBar>
 #include <QTextBoundaryFinder>
 #include <QTextCursor>
+#include <QButtonGroup>
 
 #include <sstream>
 #include <vector>
