diff -Naur wxsvg-1.0b10/configure.in wxsvg-1.0b10-p/configure.in
--- wxsvg-1.0b10/configure.in	2008-01-15 13:59:31.000000000 +0100
+++ wxsvg-1.0b10-p/configure.in	2008-04-29 18:21:37.000000000 +0200
@@ -161,7 +161,7 @@
   AC_CHECK_LIB(avformat, av_register_all, [], [AC_MSG_ERROR([*** missing ffmpeg library: libavformat])])
   AC_CHECK_LIB(avcodec, avcodec_decode_audio2, [], [AC_MSG_ERROR([*** missing ffmpeg library: libavcodec])])
   AC_CHECK_LIB(swscale, sws_getContext)
-  AC_CHECK_HEADERS(ffmpeg/avcodec.h, [], [AC_MSG_ERROR([*** missing ffmpeg header file: ffmpeg/avcodec.h.])])
+  AC_CHECK_HEADERS(libavcodec/avcodec.h, [], [AC_MSG_ERROR([*** missing ffmpeg header file: ffmpeg/avcodec.h.])])
   CXXFLAGS="$CXXFLAGS -DUSE_FFMPEG"
 fi
 
diff -Naur wxsvg-1.0b10/src/mediadec_ffmpeg.cpp wxsvg-1.0b10-p/src/mediadec_ffmpeg.cpp
--- wxsvg-1.0b10/src/mediadec_ffmpeg.cpp	2008-03-02 19:43:54.000000000 +0100
+++ wxsvg-1.0b10-p/src/mediadec_ffmpeg.cpp	2008-04-29 18:20:09.000000000 +0200
@@ -23,8 +23,8 @@
 extern "C" {
 #define __STDC_CONSTANT_MACROS
 #define __STDC_LIMIT_MACROS
-#include <ffmpeg/avformat.h>
-#include <ffmpeg/swscale.h>
+#include <libavformat/avformat.h>
+#include <libswscale/swscale.h>
 }
 
 wxFfmpegMediaDecoder::wxFfmpegMediaDecoder():
