diff -Nur libgnomeprint-2.12.1/libgnomeprint/gnome-print-pango.c libgnomeprint-2.12.1.new/libgnomeprint/gnome-print-pango.c
--- libgnomeprint-2.12.1/libgnomeprint/gnome-print-pango.c	2005-08-12 22:09:20.000000000 +0200
+++ libgnomeprint-2.12.1.new/libgnomeprint/gnome-print-pango.c	2006-05-15 00:08:44.000000000 +0200
@@ -25,6 +25,7 @@
  */
 
 #define PANGO_ENABLE_BACKEND	/* Needed to access PangoFcFont.font_pattern */
+/* behdad: Humm, font_pattern can be accessed using "pattern" property. */
 
 #include <config.h>
 
@@ -35,6 +36,12 @@
 #include <pango/pangofc-font.h>
 #include <pango/pangoft2.h>
 
+#ifdef PANGO_GLYPH_EMPTY
+#define EMPTY_GLYPH PANGO_GLYPH_EMPTY
+#else
+#define EMPTY_GLYPH 0
+#endif
+
 /**
  * is_gnome_print_key :
  * 
@@ -298,7 +305,7 @@
 	for (i = 0; i < glyphs->num_glyphs; i++) {
 		PangoGlyphInfo *gi = &glyphs->glyphs[i];
 
-		if (gi->glyph) {
+		if (gi->glyph != EMPTY_GLYPH) {
 			int x = x_off + gi->geometry.x_offset;
 			int y = gi->geometry.y_offset;
 			
