diff -Nur libgnomeprint-2.12.1/libgnomeprint/Makefile.in libgnomeprint-2.12.1.new/libgnomeprint/Makefile.in
--- libgnomeprint-2.12.1/libgnomeprint/Makefile.in	2005-09-22 21:36:43.000000000 +0200
+++ libgnomeprint-2.12.1.new/libgnomeprint/Makefile.in	2006-07-07 16:29:29.000000000 +0200
@@ -337,7 +337,7 @@
 
 
 libgnomeprint_2_2_la_LDFLAGS = \
-	-no-undefined
+	-no-undefined -lm
 
 
 libgnomeprint_2_2_la_DEPENDENCIES = \
