diff -aur nemesi-0.6.0/decoder/libffmp3/ffmp3.c fw_nemesi-0.6.0/decoder/libffmp3/ffmp3.c
--- nemesi-0.6.0/decoder/libffmp3/ffmp3.c	2006-09-27 15:49:11.000000000 +0200
+++ fw_nemesi-0.6.0/decoder/libffmp3/ffmp3.c	2008-04-28 14:04:09.000000000 +0200
@@ -40,7 +40,7 @@
 #ifdef HAVE_AV_CONFIG_H
 #undef HAVE_AV_CONFIG_H
 #endif
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #define BUFFER 8192
 #define RESAMPLED
 int get_plugin_pt(void);
diff -aur nemesi-0.6.0/decoder/libffmpeg/ffmpeg.h fw_nemesi-0.6.0/decoder/libffmpeg/ffmpeg.h
--- nemesi-0.6.0/decoder/libffmpeg/ffmpeg.h	2006-09-26 17:32:23.000000000 +0200
+++ fw_nemesi-0.6.0/decoder/libffmpeg/ffmpeg.h	2008-04-28 14:00:19.000000000 +0200
@@ -31,7 +31,7 @@
 
 #include <stdio.h>
 
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 
 #include <nemesi/img_format.h>
 
diff -aur nemesi-0.6.0/decoder/libffmpeg4/ffmpeg4.h fw_nemesi-0.6.0/decoder/libffmpeg4/ffmpeg4.h
--- nemesi-0.6.0/decoder/libffmpeg4/ffmpeg4.h	2006-09-26 17:32:23.000000000 +0200
+++ fw_nemesi-0.6.0/decoder/libffmpeg4/ffmpeg4.h	2008-04-28 14:06:27.000000000 +0200
@@ -31,7 +31,7 @@
 
 #include <stdio.h>
 
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 
 #include <nemesi/img_format.h>
 
