diff -Naur wxxml2.orig/src/xml2.cpp wxxml2/src/xml2.cpp
--- wxxml2.orig/src/xml2.cpp	2005-08-10 10:15:08.000000000 +0000
+++ wxxml2/src/xml2.cpp	2006-04-17 02:42:18.000000000 +0000
@@ -916,7 +916,7 @@
 bool wxXml2Document::operator==(const wxXml2Document &doc) const
 {
 	// check for null pointers
-	if ((int)doc.GetObj() ^ (int)GetObj())
+	if ((long)doc.GetObj() ^ (long)GetObj())
 		return FALSE;
 	if (GetRoot() == doc.GetRoot())
 		return TRUE;
