diff -up gxine-0.5.907/src/console_output.h.glibc gxine-0.5.907/src/console_output.h
--- gxine-0.5.907/src/console_output.h.glibc	2010-04-11 16:46:54.000000000 +0200
+++ gxine-0.5.907/src/console_output.h	2012-02-23 00:52:00.158725579 +0100
@@ -21,7 +21,7 @@
 #ifndef GXINE_CONSOLE_OUTPUT_H
 #define GXINE_CONSOLE_OUTPUT_H
 
-#include <glib/gmessages.h>
+#include <glib.h>
 
 void console_output_init (void);
 
diff -up gxine-0.5.907/src/engine.c.glibc gxine-0.5.907/src/engine.c
--- gxine-0.5.907/src/engine.c.glibc	2011-10-12 00:47:52.000000000 +0200
+++ gxine-0.5.907/src/engine.c	2012-02-23 00:50:04.560170759 +0100
@@ -30,8 +30,6 @@
 #include <errno.h>
 #include <unistd.h>
 
-#include <glib/gthread.h>
-
 #include "engine.h"
 #include "ui.h"
 #include "utils.h"
diff -up gxine-0.5.907/src/main.c.glibc gxine-0.5.907/src/main.c
--- gxine-0.5.907/src/main.c.glibc	2010-12-05 12:38:44.000000000 +0100
+++ gxine-0.5.907/src/main.c	2012-02-23 00:50:23.126938644 +0100
@@ -46,7 +46,6 @@
 #include <gdk/gdk.h>
 #include <gdk/gdkkeysyms.h>
 #include <glib.h>
-#include <glib/gmessages.h>
 
 #include "desktop_integration.h"
 #include "engine.h"
