diff -ur a/Makefile b/Makefile
--- a/Makefile	2014-03-16 13:12:54.000000000 -0500
+++ b/Makefile	2014-05-15 11:09:01.692766209 -0500
@@ -8,5 +8,9 @@
 install:
 	mkdir -p $(DESTDIR)$(FIRMWAREDIR)
 	cp -r * $(DESTDIR)$(FIRMWAREDIR)
-	rm -f $(DESTDIR)/usbdux/*dux $(DESTDIR)/*/*.asm
-	rm $(DESTDIR)$(FIRMWAREDIR)/{WHENCE,LICENSE.*,LICENCE.*}
+	rm $(DESTDIR)$(FIRMWAREDIR)/{WHENCE,LICENSE.*,LICENCE.*,GPL-3}
+	find $(DESTDIR)$(FIRMWAREDIR) \( -name '*.c' -or -name '*.h' -or -name '*.asm' -or -name 'Makefile' -or -name 'configure' -or -name '*.S' -or -name "README" -or -name '*.pl' -or -name '*.pst' -or -name '*.txt' \) -delete
+	find $(DESTDIR)$(FIRMWAREDIR) \( -type d -and -empty \) -delete
+	rm -rf $(DESTDIR)$(FIRMWAREDIR)/cis/src
+	rm -rf $(DESTDIR)$(FIRMWAREDIR)/usbdux
+	rm -rf $(DESTDIR)$(FIRMWAREDIR)/carl9170fw
