diff --git a/src/doctotext_misc.cpp b/src/doctotext_misc.cpp
index 518b129..dd30a3d 100755
--- a/src/doctotext_misc.cpp
+++ b/src/doctotext_misc.cpp
@@ -30,6 +30,7 @@
 
 #include "doctotext_misc.h"
 #include <iostream>
+#include <cstdio>
 
 int prepareTable(std::vector<svector> &p_table)
 {
diff --git a/src/doctotext_rtf_parser.cpp b/src/doctotext_rtf_parser.cpp
index a9b40a9..ce1c460 100644
--- a/src/doctotext_rtf_parser.cpp
+++ b/src/doctotext_rtf_parser.cpp
@@ -31,6 +31,7 @@
 #include "doctotext_rtf_parser.h"
 #include <map>
 #include <stack>
+#include <cstdio>
 #include <stdlib.h>
 #include <string.h>
 #include <textconverter.h>
diff --git a/src/doctotext_unzip.cpp b/src/doctotext_unzip.cpp
index cd3dbfc..e4c6a0a 100644
--- a/src/doctotext_unzip.cpp
+++ b/src/doctotext_unzip.cpp
@@ -30,6 +30,7 @@
 
 #include "doctotext_unzip.h"
 
+#include <cstdio>
 #include <stdlib.h>
 #include "unzip.h"
 #include "zlib.h"
