diff --git a/Makefile b/Makefile
index 525593d..1e159bd 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,6 @@ EXEEXT=
 endif
 
 build: version.h
-	make -C 3rdparty wv2
 	make -C 3rdparty unzip
 	make -C src
 	rm -rf build
@@ -18,7 +17,6 @@ ifeq ($(WIN32),1)
 	cp 3rdparty/gettext-0.15/gettext-runtime/intl/.libs/libintl-8.dll build
 	cp 3rdparty/libxml2-2.6.26/.libs/libxml2-2.dll build
 else
-	cp 3rdparty/wv2-0.2.3/src/.libs/libwv2.so.1 build
 	echo 'LD_LIBRARY_PATH=.:$$LD_LIBRARY_PATH ./doctotext "$$@"' > build/doctotext.sh
 	chmod u+x build/doctotext.sh
 endif
diff --git a/src/Makefile b/src/Makefile
index 2c115fe..ac7faac 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -1,4 +1,4 @@
-CFLAGS = -O2 -I../3rdparty/wv2-0.2.3/src -I../3rdparty/unzip101e
+CFLAGS = -O2 -I/usr/include/wv2 -I../3rdparty/unzip101e
 LDFLAGS = ../3rdparty/unzip101e/ioapi.o ../3rdparty/unzip101e/unzip.o
 
 OBJECTS = doctotext.o doctotext_rtf_parser.o doctotext_odf_ooxml_parser.o doctotext_misc.o doctotext_xml_fixer.o doctotext_unzip.o
@@ -11,7 +11,7 @@ CC = mingw32-g++
 EXEEXT = .exe
 CFLAGS += -I../3rdparty/glib-2.14.5 -I../3rdparty/zlib-1.2.3 -I../3rdparty/libxml2-2.6.26/include
 LDFLAGS += ../3rdparty/unzip101e/iowin32.o \
-	-L../3rdparty/wv2-0.2.3/src/.libs -lwv2 \
+	-lwv2 \
 	-L../3rdparty/libgsf-1.14.1/gsf/.libs -lgsf-1 \
 	-L../3rdparty/glib-2.14.5/gobject/.libs -lgobject-2.0 \
 	-L../3rdparty/glib-2.14.5/glib/.libs -lglib-2.0 \
diff --git a/src/doctotext.cpp b/src/doctotext.cpp
index ed19e40..190e02f 100644
--- a/src/doctotext.cpp
+++ b/src/doctotext.cpp
@@ -119,14 +119,14 @@ class DocToTextHandler : public TextHandler
 		}
 };
 
-static void version()
+static void myversion()
 {
 	printf("DocToText v%s\nConverts MS Word, RTF, ODF and OOXML documents to plain text\nCopyright (c) 2006-2009 SILVERCODERS(R)\nhttp://silvercoders.com\n", VERSION);
 }
 
 static void help()
 {
-	version();
+	myversion();
 	printf("\nUsage: doctotext [OPTION]... [FILE]\n\n");
 	printf("FILE\t\tname of file to convert\n\n");
 	printf("Options:\n");
