diff -Naur konsole-20.08.0/src/Profile.cpp konsole-20.08.0-p/src/Profile.cpp
--- konsole-20.08.0/src/Profile.cpp	2020-08-05 09:42:20.000000000 +0200
+++ konsole-20.08.0-p/src/Profile.cpp	2020-09-02 19:46:37.268513943 +0200
@@ -189,7 +189,7 @@
     setProperty(HistorySize, 1000);
     setProperty(ScrollBarPosition, Enum::ScrollBarRight);
     setProperty(ScrollFullPage, false);
-    setProperty(HighlightScrolledLines, true);
+    setProperty(HighlightScrolledLines, false);
 
     setProperty(FlowControlEnabled, true);
     setProperty(UrlHintsModifiers, 0);
