--- konsole-20.12.0/src/profile/Profile.cpp	2020-12-02 23:00:50.000000000 +0100
+++ konsole-20.12.0.new/src/profile/Profile.cpp	2020-12-16 22:42:32.198471632 +0100
@@ -196,7 +196,7 @@
     setProperty(HistorySize, 1000);
     setProperty(ScrollBarPosition, Enum::ScrollBarRight);
     setProperty(ScrollFullPage, false);
-    setProperty(HighlightScrolledLines, true);
+    setProperty(HighlightScrolledLines, false);
 
     setProperty(FlowControlEnabled, true);
     setProperty(UrlHintsModifiers, 0);
