--- toml-parser-2.0.1.0/toml-parser.cabal	2001-09-09 03:46:40.000000000 +0200
+++ toml-parser-2.0.1.0.new/toml-parser.cabal	2024-12-26 19:12:25.656085868 +0100
@@ -71,7 +71,7 @@
         Toml.Syntax.ParserUtils
     build-depends:
         array           ^>= 0.5,
-        base            ^>= {4.14, 4.15, 4.16, 4.17, 4.18, 4.19, 4.20},
+        base            ^>= {4.14, 4.15, 4.16, 4.17, 4.18, 4.19, 4.20, 4.21},
         containers      ^>= {0.5, 0.6, 0.7},
         prettyprinter   ^>= 1.7,
         text            >= 0.2 && < 3,
