diff -aur transcode-1.0.5/configure.in fw_transcode-1.0.5/configure.in
--- transcode-1.0.5/configure.in	2008-01-05 15:37:14.000000000 +0100
+++ fw_transcode-1.0.5/configure.in	2008-04-29 03:13:07.000000000 +0200
@@ -611,14 +611,14 @@
   esac],
   [enable_libavcodec_static=no])
 LIBAVCODEC_EXTRA_LIBS="$LIBAVCODEC_EXTRA_LIBS $($PKG_CONFIG --libs libavcodec)"
-TC_PKG_CHECK(libavcodec, required, LIBAVCODEC, no, [ffmpeg/avcodec.h],
+TC_PKG_CHECK(libavcodec, required, LIBAVCODEC, no, [libavcodec/avcodec.h],
  avcodec, avcodec_thread_init, FFMpeg, [http://www.ffmpeg.org/])
 if test x"$have_libavcodec" = x"yes" ; then
   save_CPPFLAGS="$CPPFLAGS"
   CPPFLAGS="$CPPFLAGS $LIBAVCODEC_CFLAGS"
   AC_TRY_RUN([
 #include <stdio.h>
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 int
 main()
 {
@@ -640,8 +640,8 @@
 ],
     [LIBAVCODEC_VERSION="`./conftest$ac_exeext | sed -ne 's,VER=\(.*\),\1,p'`"
       LIBAVCODEC_BUILD="`./conftest$ac_exeext | sed -ne 's,BUILD=\(.*\),\1,p'`"],
-    [TC_PKG_ERROR(libavcodec, ffmpeg/avcodec.h, required, libavcodec, [http://www.ffmpeg.org],
-      [cannot compile ffmpeg/avcodec.h])],
+    [TC_PKG_ERROR(libavcodec, libavcodec/avcodec.h, required, libavcodec, [http://www.ffmpeg.org],
+      [cannot compile libavcodec/avcodec.h])],
     [echo $ac_n "cross compiling; assumed OK... $ac_c"
       LIBAVCODEC_VERSION=""
       LIBAVCODEC_BUILD=""])
@@ -653,7 +653,7 @@
     dnl $_w_libavcodec_l comes from TC_PKG_CHECK()
     LIBS="$LIBS $w_libavcodec_l${deflib}/libavcodec.a $LIBAVCODEC_EXTRA_LIBS"
     AC_TRY_LINK([
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 ],[
 AVCodecContext *ctx = (void *)0;
 avcodec_thread_init(ctx, 0);
@@ -820,7 +820,7 @@
 dnl
 dnl libpostproc
 dnl
-TC_PKG_CHECK(libpostproc, no, LIBPOSTPROC, no, postprocess.h,
+TC_PKG_CHECK(libpostproc, no, LIBPOSTPROC, no, libpostproc/postprocess.h,
   postproc, pp_postprocess, libpostproc, [http://www.ffmpeg.org/])
 TC_PKG_HAVE(libpostproc, LIBPOSTPROC)
 
diff -aur transcode-1.0.5/export/aud_aux.c fw_transcode-1.0.5/export/aud_aux.c
--- transcode-1.0.5/export/aud_aux.c	2007-08-06 11:56:57.000000000 +0200
+++ fw_transcode-1.0.5/export/aud_aux.c	2008-04-29 03:23:23.000000000 +0200
@@ -31,7 +31,7 @@
 #include <inttypes.h>
 #include <assert.h>
 
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 
 #include "aud_aux.h"
 #include "ac3.h"
diff -aur transcode-1.0.5/export/export_ffmpeg.c fw_transcode-1.0.5/export/export_ffmpeg.c
--- transcode-1.0.5/export/export_ffmpeg.c	2007-11-17 10:35:31.000000000 +0100
+++ fw_transcode-1.0.5/export/export_ffmpeg.c	2008-04-29 03:23:48.000000000 +0200
@@ -36,7 +36,7 @@
 #include "vid_aux.h"
 // FIXME
 #undef EMULATE_FAST_INT
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 
 #if !defined(INFINITY) && defined(HUGE_VAL)
 #define INFINITY HUGE_VAL
diff -aur transcode-1.0.5/export/ffmpeg_cfg.h fw_transcode-1.0.5/export/ffmpeg_cfg.h
--- transcode-1.0.5/export/ffmpeg_cfg.h	2004-10-18 04:04:14.000000000 +0200
+++ fw_transcode-1.0.5/export/ffmpeg_cfg.h	2008-04-29 03:24:53.000000000 +0200
@@ -2,7 +2,7 @@
 #define __FFMPEG_CFG_H
 
 #include "libioaux/configs.h"
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 
 //char *lavc_param_vcodec = "mpeg4";
 //extern int lavc_param_vbitrate;
diff -aur transcode-1.0.5/filter/filter_resample.c fw_transcode-1.0.5/filter/filter_resample.c
--- transcode-1.0.5/filter/filter_resample.c	2005-07-04 09:23:01.000000000 +0200
+++ fw_transcode-1.0.5/filter/filter_resample.c	2008-04-29 03:20:52.000000000 +0200
@@ -30,7 +30,7 @@
 #include "filter.h"
 #include "optstr.h"
 
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 
 static char * resample_buffer = NULL;
 static int bytes_per_sample;
diff -aur transcode-1.0.5/filter/pp/filter_pp.c fw_transcode-1.0.5/filter/pp/filter_pp.c
--- transcode-1.0.5/filter/pp/filter_pp.c	2005-07-04 09:23:01.000000000 +0200
+++ fw_transcode-1.0.5/filter/pp/filter_pp.c	2008-04-29 03:32:02.000000000 +0200
@@ -35,7 +35,7 @@
 #include <ctype.h>
 #include <inttypes.h>
 
-#include "postprocess.h"
+#include <libpostproc/postprocess.h>
 
 static pp_mode_t *mode[MAX_FILTER];
 static pp_context_t *context[MAX_FILTER];
diff -aur transcode-1.0.5/import/decode_lavc.c fw_transcode-1.0.5/import/decode_lavc.c
--- transcode-1.0.5/import/decode_lavc.c	2005-07-04 09:23:01.000000000 +0200
+++ fw_transcode-1.0.5/import/decode_lavc.c	2008-04-29 03:19:47.000000000 +0200
@@ -31,7 +31,7 @@
 #ifdef EMULATE_FAST_INT
 #undef EMULATE_FAST_INT
 #endif
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #include "yuv2rgb.h"
 
 #define READ_BUFFER_SIZE (10*1024*1024)
diff -aur transcode-1.0.5/import/import_ffmpeg.c fw_transcode-1.0.5/import/import_ffmpeg.c
--- transcode-1.0.5/import/import_ffmpeg.c	2007-10-09 20:32:22.000000000 +0200
+++ fw_transcode-1.0.5/import/import_ffmpeg.c	2008-04-29 03:17:47.000000000 +0200
@@ -38,7 +38,7 @@
 
 // FIXME
 #undef EMULATE_FAST_INT
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 
 #include "libvo/yuv2rgb.h"
 #include "avilib/avilib.h"
