diff -aur transcode-1.0.5/filter/pp/filter_pp.c fw_transcode-1.0.5/filter/pp/filter_pp.c
--- transcode-1.0.5/filter/pp/filter_pp.c       2005-07-04 09:23:01.000000000 +0200
+++ fw_transcode-1.0.5/filter/pp/filter_pp.c    2008-04-29 03:32:02.000000000 +0200
@@ -35,7 +35,7 @@
 #include <ctype.h>
 #include <inttypes.h>

-#include "postprocess.h"
+#include <libpostproc/postprocess.h>

 static pp_mode_t *mode[MAX_FILTER];
 static pp_context_t *context[MAX_FILTER];

