diff -Naur transcode-1.0.2/acinclude.m4 transcode-1.0.2-p/acinclude.m4
--- transcode-1.0.2/acinclude.m4	2005-07-04 08:58:50.000000000 +0200
+++ transcode-1.0.2-p/acinclude.m4	2007-05-08 16:24:05.000000000 +0200
@@ -92,7 +92,8 @@
 ],    [AC_DEFINE([HAVE_STRUCT_V4L2_BUFFER], 1,
         [define if your videodev2 header has struct v4l2_buffer])
         AC_MSG_RESULT([yes])],
-      [v4l2=no AC_MSG_RESULT([no])])
+      [v4l2=no
+       AC_MSG_RESULT([no])])
   fi
 
   if test x"$v4l" = x"yes" -o x"$v4l2" = x"yes" ; then
diff -Naur transcode-1.0.2/configure.in transcode-1.0.2-p/configure.in
--- transcode-1.0.2/configure.in	2005-10-30 05:57:45.000000000 +0100
+++ transcode-1.0.2-p/configure.in	2007-05-08 16:24:13.000000000 +0200
@@ -619,6 +619,12 @@
   AC_TRY_RUN([
 #include <stdio.h>
 #include <ffmpeg/avcodec.h>
+
+#ifndef STRINGIFY
+# define STRINGIFY(x)    STRINGIFY1(x)
+# define STRINGIFY1(x)   #x
+#endif
+
 int
 main()
 {
@@ -628,7 +634,7 @@
     printf("install ffmpeg 0.4.9-pre1 or newer, or a cvs version after 20040703");
     return(1);
   }
-  printf("VER=%s\n", FFMPEG_VERSION);
+  printf("VER=%s\n", STRINGIFY(LIBAVCODEC_VERSION));
   printf("BUILD=%d\n", LIBAVCODEC_BUILD);
   return(0);
 }
