--- transcode-1.0.5/configure.in	2008-01-05 15:37:14.000000000 +0100
+++ transcode-1.0.5-p/configure.in	2008-04-13 00:09:49.000000000 +0200
@@ -14,6 +14,7 @@
 
 # Checks for programs.
 AC_PROG_CC
+AM_PROG_CC_C_O
 AC_PROG_GCC_TRADITIONAL
 AC_PROG_CXX
 AC_PROG_AWK
@@ -27,7 +28,6 @@
 AS=$CC
 CCAS='$(CC)'
 AC_SUBST(CCAS)
-AC_PROG_CC
 AC_SUBST(CCASFLAGS)
 ASFLAGS="$CCASFLAGS"
 AC_SUBST(ASFLAGS)
@@ -1235,7 +1235,7 @@
 IMAGEMAGICK_EXTRA_LIBS="$IMAGEMAGICK_EXTRA_LIBS -lz -lm"
 dnl MagickWand libraries/headers always (?) are a superset of MagickCore ones.
 TC_PKG_CHECK(imagemagick, no, IMAGEMAGICK, Wand-config, [magick/api.h],
- Magick, InitializeMagick, Wand, [http://www.imagemagick.org/])
+ MagickCore, InitializeMagick, Wand, [http://www.imagemagick.org/])
 TC_PKG_HAVE(imagemagick, IMAGEMAGICK)
 if test x"$have_imagemagick" = x"yes" ; then
   AC_CHECK_HEADER([wand/magick-wand.h],
