diff -ur mozilla-orig/security/nss/lib/pki1/oiddata.h mozilla/security/nss/lib/pki1/oiddata.h
--- mozilla-orig/security/nss/lib/pki1/oiddata.h	2005-03-13 19:12:25.000000000 -0500
+++ mozilla/security/nss/lib/pki1/oiddata.h	2005-03-13 19:15:19.000000000 -0500
@@ -43,12 +43,6 @@
 #include "nsspki1t.h"
 #endif /* NSSPKI1T_H */
 
-extern const NSSOID nss_builtin_oids[];
-extern const PRUint32 nss_builtin_oid_count;
-
-/*extern const nssAttributeTypeAliasTable nss_attribute_type_aliases[];*/
-/*extern const PRUint32 nss_attribute_type_alias_count;*/
-
 extern const NSSOID *NSS_OID_RFC1274_UID;
 extern const NSSOID *NSS_OID_RFC1274_EMAIL;
 extern const NSSOID *NSS_OID_RFC2247_DC;
diff -ur mozilla-orig/security/nss/lib/pki1/oidgen.perl mozilla/security/nss/lib/pki1/oidgen.perl
--- mozilla-orig/security/nss/lib/pki1/oidgen.perl	2005-03-13 19:12:25.000000000 -0500
+++ mozilla/security/nss/lib/pki1/oidgen.perl	2005-03-13 19:15:19.000000000 -0500
@@ -233,12 +233,6 @@
 #include "nsspki1t.h"
 #endif /* NSSPKI1T_H */
 
-extern const NSSOID nss_builtin_oids[];
-extern const PRUint32 nss_builtin_oid_count;
-
-/*extern const nssAttributeTypeAliasTable nss_attribute_type_aliases[];*/
-/*extern const PRUint32 nss_attribute_type_alias_count;*/
-
 EOD
     ;
 
diff -ur mozilla-orig/security/nss/lib/pki1/pki1.h mozilla/security/nss/lib/pki1/pki1.h
--- mozilla-orig/security/nss/lib/pki1/pki1.h	2005-03-13 19:12:25.000000000 -0500
+++ mozilla/security/nss/lib/pki1/pki1.h	2005-03-13 19:15:19.000000000 -0500
@@ -55,7 +55,9 @@
 
 PR_BEGIN_EXTERN_C
 
-/* fgmr 19990505 moved these here from oiddata.h */
+extern const NSSOID nss_builtin_oids[];
+extern const PRUint32 nss_builtin_oid_count;
+
 extern const nssAttributeTypeAliasTable nss_attribute_type_aliases[];
 extern const PRUint32 nss_attribute_type_alias_count;
 
