diff -Naur blaze-builder-0.4.2.2/blaze-builder.cabal blaze-builder-0.4.2.2.new/blaze-builder.cabal
--- blaze-builder-0.4.2.2/blaze-builder.cabal	2001-09-09 03:46:40.000000000 +0200
+++ blaze-builder-0.4.2.2.new/blaze-builder.cabal	2022-10-14 11:26:07.608081199 +0200
@@ -33,8 +33,9 @@
 Cabal-version:       >= 1.10
 
 Tested-with:
-  GHC == 9.2.0.20210821
-  GHC == 9.0.1
+  GHC == 9.4.1
+  GHC == 9.2.2
+  GHC == 9.0.2
   GHC == 8.10.7
   GHC == 8.8.4
   GHC == 8.6.5
@@ -85,7 +86,7 @@
     , bytestring >= 0.9 && < 1.0
     , deepseq
     , ghc-prim
-    , text >= 0.10 && < 1.3
+    , text >= 0.10 && < 2.1
 
   if impl(ghc < 7.8)
      build-depends:  bytestring-builder
