--- kqemu.kmdr.orig	2006-01-29 16:08:23.000000000 +0100
+++ kqemu.kmdr	2006-02-20 22:06:19.000000000 +0100
@@ -93,7 +93,7 @@
         <property name="associations" stdset="0">
             <stringlist>
                 <string>
-ksnapshot -c || @Message.error(KSnapShot is not installed!, Knsapshot not found)</string>
+ksnapshot -c</string>
             </stringlist>
         </property>
         <property name="writeStdout">
