diff -Naur exim-4.74.orig/src/configure.default exim-4.74/src/configure.default
--- exim-4.74.orig/src/configure.default	2011-01-24 22:40:38.000000000 +0100
+++ exim-4.74/src/configure.default	2011-03-21 11:40:10.000000000 +0100
@@ -649,14 +649,14 @@
 
 # This transport is used for local delivery to user mailboxes in traditional
 # BSD mailbox format. By default it will be run under the uid and gid of the
-# local user, and requires the sticky bit to be set on the /var/mail directory.
+# local user, and requires the sticky bit to be set on the /var/spool/mail directory.
 # Some systems use the alternative approach of running mail deliveries under a
 # particular group instead of using the sticky bit. The commented options below
 # show how this can be done.
 
 local_delivery:
   driver = appendfile
-  file = /var/mail/$local_part
+  file = /var/spool/mail/$local_part
   delivery_date_add
   envelope_to_add
   return_path_add
