diff -Naur hunspell-1.0-RC2.orig/Makefile hunspell-1.0-RC2/Makefile
--- hunspell-1.0-RC2.orig/Makefile	2005-06-07 16:56:17.000000000 +0000
+++ hunspell-1.0-RC2/Makefile	2005-06-07 16:56:52.000000000 +0000
@@ -16,7 +16,7 @@
 CC=g++
 #CC=/opt/SUNWspro/bin/CC
 
-CXXFLAGS=-O2 -Wall -ansi -pedantic -I.
+CXXFLAGS+=-O2 -Wall -ansi -pedantic -I.
 
 # for debug only
 #CXXFLAGS=-g -O0 -Wall -ansi -pedantic -I.
