diff -ur qt-gstreamer-1.2.0/src/QGlib/connect.cpp qt-gstreamer-1.2.0/src/QGlib/connect.cpp
--- qt-gstreamer-1.2.0/src/QGlib/connect.cpp	2014-07-08 20:38:36.000000000 +0200
+++ qt-gstreamer-1.2.0/src/QGlib/connect.cpp	2016-01-17 22:15:13.373067347 +0100
@@ -20,10 +20,12 @@
 #include <glib-object.h>
 #include <QtCore/QHash>
 #include <QtCore/QMutex>
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 #include <boost/multi_index_container.hpp>
 #include <boost/multi_index/sequenced_index.hpp>
 #include <boost/multi_index/ordered_index.hpp>
 #include <boost/multi_index/member.hpp>
+#endif
 
 namespace QGlib {
 namespace Private {
diff -ur qt-gstreamer-1.2.0/src/QGlib/connect.h qt-gstreamer-1.2.0/src/QGlib/connect.h
--- qt-gstreamer-1.2.0/src/QGlib/connect.h	2014-07-08 20:38:36.000000000 +0200
+++ qt-gstreamer-1.2.0/src/QGlib/connect.h	2016-01-17 22:12:47.474524393 +0100
@@ -25,8 +25,10 @@
 #include <QtCore/QSharedPointer>
 #include <QtCore/QFlags>
 #include <QtCore/QHash>
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 #include <boost/type_traits.hpp>
 #include <boost/utility/enable_if.hpp>
+#endif
 
 namespace QGlib {
 
diff -ur qt-gstreamer-1.2.0/src/QGlib/connectimpl.h qt-gstreamer-1.2.0/src/QGlib/connectimpl.h
--- qt-gstreamer-1.2.0/src/QGlib/connectimpl.h	2014-07-08 20:38:36.000000000 +0200
+++ qt-gstreamer-1.2.0/src/QGlib/connectimpl.h	2016-01-17 22:14:28.910416373 +0100
@@ -26,7 +26,9 @@
 # include "refpointer.h"
 # include <QtCore/QList>
 # include <stdexcept>
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 # include <boost/type_traits.hpp>
+#endif
 
 
 namespace QGlib {
@@ -203,9 +205,11 @@
 
 # else //QGLIB_HAVE_CXX0X
 
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 #  include <boost/function.hpp>
 #  include <boost/preprocessor.hpp>
 #  include <boost/bind.hpp>
+#endif
 
 // include the second part of this file as many times as QGLIB_CONNECT_MAX_ARGS specifies
 #  define BOOST_PP_ITERATION_PARAMS_1 (3,(0, QGLIB_CONNECT_MAX_ARGS, "QGlib/connectimpl.h"))
diff -ur qt-gstreamer-1.2.0/src/QGlib/emitimpl.h qt-gstreamer-1.2.0/src/QGlib/emitimpl.h
--- qt-gstreamer-1.2.0/src/QGlib/emitimpl.h	2014-07-08 20:38:36.000000000 +0200
+++ qt-gstreamer-1.2.0/src/QGlib/emitimpl.h	2016-01-17 22:14:52.257383652 +0100
@@ -125,7 +125,9 @@
 
 # else //QGLIB_HAVE_CXX0X
 
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 #  include <boost/preprocessor.hpp>
+#endif
 
 // include the second part of this file as many times as QGLIB_SIGNAL_MAX_ARGS specifies
 #  define BOOST_PP_ITERATION_PARAMS_1 (3,(0, QGLIB_SIGNAL_MAX_ARGS, "QGlib/emitimpl.h"))
diff -ur qt-gstreamer-1.2.0/src/QGlib/gen.cpp qt-gstreamer-1.2.0/src/QGlib/gen.cpp
--- qt-gstreamer-1.2.0/src/QGlib/gen.cpp	2014-07-08 20:38:36.000000000 +0200
+++ qt-gstreamer-1.2.0/src/QGlib/gen.cpp	2016-01-17 22:11:35.846420829 +0100
@@ -19,7 +19,9 @@
 */
 
 #define INCLUDED_FROM_CODEGEN
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 #include <boost/static_assert.hpp>
+#endif
 
 #define REGISTER_TYPE_IMPLEMENTATION(T, GTYPE) \
     namespace QGlib { \
diff -ur qt-gstreamer-1.2.0/src/QGlib/global.h qt-gstreamer-1.2.0/src/QGlib/global.h
--- qt-gstreamer-1.2.0/src/QGlib/global.h	2014-07-08 20:38:36.000000000 +0200
+++ qt-gstreamer-1.2.0/src/QGlib/global.h	2016-01-17 22:16:33.849295462 +0100
@@ -23,7 +23,9 @@
 #endif
 
 #include <QtCore/QtGlobal>
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 #include <boost/config.hpp>
+#endif
 
 /* defined by cmake when building this library */
 #if defined(QtGLib_EXPORTS) || defined(Qt5GLib_EXPORTS)
@@ -84,7 +86,9 @@
 # define QGLIB_STATIC_ASSERT(expr, message) static_assert(expr, message)
 # define QGLIB_HAVE_CXX0X_STATIC_ASSERT 1
 #else
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 # include <boost/static_assert.hpp>
+#endif
 # define QGLIB_STATIC_ASSERT(expr, message) BOOST_STATIC_ASSERT(expr)
 #endif
 
diff -ur qt-gstreamer-1.2.0/src/QGlib/refpointer.h qt-gstreamer-1.2.0/src/QGlib/refpointer.h
--- qt-gstreamer-1.2.0/src/QGlib/refpointer.h	2014-07-08 20:38:36.000000000 +0200
+++ qt-gstreamer-1.2.0/src/QGlib/refpointer.h	2016-01-17 22:12:30.003303890 +0100
@@ -23,8 +23,10 @@
 #include "type.h"
 #include "wrap.h"
 #include <cstddef>
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 #include <boost/type_traits.hpp>
 #include <boost/utility/enable_if.hpp>
+#endif
 #include <QtCore/QHash>
 
 namespace QGlib {
diff -ur qt-gstreamer-1.2.0/src/QGlib/type.h qt-gstreamer-1.2.0/src/QGlib/type.h
--- qt-gstreamer-1.2.0/src/QGlib/type.h	2014-07-08 20:38:36.000000000 +0200
+++ qt-gstreamer-1.2.0/src/QGlib/type.h	2016-01-17 22:13:05.742846202 +0100
@@ -21,7 +21,9 @@
 
 #include "global.h"
 #include <QtCore/QList>
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 #include <boost/mpl/if.hpp>
+#endif
 
 /*
  * This is a re-definition of GType inside the QGlib::Private namespace.
diff -ur qt-gstreamer-1.2.0/src/QGlib/value.cpp qt-gstreamer-1.2.0/src/QGlib/value.cpp
--- qt-gstreamer-1.2.0/src/QGlib/value.cpp	2014-07-08 20:38:36.000000000 +0200
+++ qt-gstreamer-1.2.0/src/QGlib/value.cpp	2016-01-17 22:12:00.892604070 +0100
@@ -19,7 +19,9 @@
 #include "value.h"
 #include "string.h"
 #include <cstring>
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 #include <boost/type_traits.hpp>
+#endif
 #include <glib-object.h>
 #include <QtCore/QDebug>
 #include <QtCore/QReadWriteLock>
diff -ur qt-gstreamer-1.2.0/src/QGlib/value.h qt-gstreamer-1.2.0/src/QGlib/value.h
--- qt-gstreamer-1.2.0/src/QGlib/value.h	2014-07-08 20:38:36.000000000 +0200
+++ qt-gstreamer-1.2.0/src/QGlib/value.h	2016-01-17 22:15:35.077825910 +0100
@@ -23,8 +23,10 @@
 #include "type.h"
 #include "refpointer.h"
 #include "error.h"
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 #include <boost/mpl/if.hpp>
 #include <boost/type_traits.hpp>
+#endif
 #include <stdexcept>
 #include <QtCore/QString>
 #include <QtCore/QDebug>
diff -ur qt-gstreamer-1.2.0/src/QGst/bin.h qt-gstreamer-1.2.0/src/QGst/bin.h
--- qt-gstreamer-1.2.0/src/QGst/bin.h	2014-07-08 20:38:36.000000000 +0200
+++ qt-gstreamer-1.2.0/src/QGst/bin.h	2016-01-17 22:10:51.658804810 +0100
@@ -27,9 +27,11 @@
 # pragma warning(disable:4250) //Bin inherits QGst::Object::ref/unref via dominance
 #endif
 
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 #if !QGLIB_HAVE_CXX0X
 # include <boost/preprocessor.hpp>
 #endif
+#endif
 
 namespace QGst {
 
diff -ur qt-gstreamer-1.2.0/src/QGst/element.h qt-gstreamer-1.2.0/src/QGst/element.h
--- qt-gstreamer-1.2.0/src/QGst/element.h	2014-07-08 20:38:36.000000000 +0200
+++ qt-gstreamer-1.2.0/src/QGst/element.h	2016-01-17 22:10:24.171311291 +0100
@@ -21,9 +21,11 @@
 #include "caps.h"
 #include "clocktime.h"
 
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 #if !QGLIB_HAVE_CXX0X
 # include <boost/preprocessor.hpp>
 #endif
+#endif
 
 namespace QGst {
 
diff -ur qt-gstreamer-1.2.0/src/QGst/gen.cpp qt-gstreamer-1.2.0/src/QGst/gen.cpp
--- qt-gstreamer-1.2.0/src/QGst/gen.cpp	2014-07-08 20:38:36.000000000 +0200
+++ qt-gstreamer-1.2.0/src/QGst/gen.cpp	2016-01-17 22:09:33.350852272 +0100
@@ -19,7 +19,9 @@
 */
 
 #define INCLUDED_FROM_CODEGEN
+#ifndef Q_MOC_RUN  // See: https://bugreports.qt-project.org/browse/QTBUG-22829
 #include <boost/static_assert.hpp>
+#endif
 
 #define REGISTER_TYPE_IMPLEMENTATION(T, GTYPE) \
     namespace QGlib { \
