diff -Naur qt-x11-free-3.3.5/configure qt-x11-free-3.3.5-kill-off-sed/configure
--- qt-x11-free-3.3.5/configure	2005-08-22 11:31:26.000000000 +0200
+++ qt-x11-free-3.3.5-kill-off-sed/configure	2006-02-22 13:11:22.000000000 +0100
@@ -2351,7 +2351,7 @@
 	else
 	    echo "Do you accept the terms of $affix license? \c"
 	fi
-	read acceptance
+	acceptance=yes
 	echo
 	if [ "$acceptance" = "yes" ]; then
 	    break
@@ -2395,7 +2395,7 @@
 	else
 	    echo "Do you accept the terms of the license? \c"
 	fi
-	read acceptance
+	acceptance=yes
 	echo
 	if [ "$acceptance" = "yes" ]; then
 	    break
@@ -2476,7 +2476,7 @@
 	else
 	    echo "Do you accept the terms of the $Platform License? \c"
 	fi
-	read acceptance
+	acceptance=yes
 	echo
 	if [ "$acceptance" = "yes" ]; then
 	    break
diff -Naur qt-x11-free-3.3.5/Makefile qt-x11-free-3.3.5-kill-off-sed/Makefile
--- qt-x11-free-3.3.5/Makefile	2005-01-03 17:08:00.000000000 +0100
+++ qt-x11-free-3.3.5-kill-off-sed/Makefile	2006-02-22 13:11:22.000000000 +0100
@@ -11,7 +11,7 @@
 install: FORCE
 	@$(MAKE) qt.install
 
-all: symlinks src-qmake src-moc sub-src sub-tools sub-tutorial sub-examples
+all: symlinks src-qmake src-moc sub-src sub-tools 
 	@echo
 	@echo "The Qt library is now built in ./lib"
 	@echo "The Qt examples are built in the directories in ./examples"
diff -Naur qt-x11-free-3.3.5/mkspecs/linux-g++/qmake.conf qt-x11-free-3.3.5-kill-off-sed/mkspecs/linux-g++/qmake.conf
--- qt-x11-free-3.3.5/mkspecs/linux-g++/qmake.conf	2005-09-02 14:43:37.000000000 +0200
+++ qt-x11-free-3.3.5-kill-off-sed/mkspecs/linux-g++/qmake.conf	2006-02-22 13:15:35.000000000 +0100
@@ -21,7 +21,7 @@
 QMAKE_CFLAGS_DEPS	= -M
 QMAKE_CFLAGS_WARN_ON	= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	= -w
-QMAKE_CFLAGS_RELEASE	= -O2
+QMAKE_CFLAGS_RELEASE	= DUMMY -fPIC
 QMAKE_CFLAGS_DEBUG	= -g
 QMAKE_CFLAGS_SHLIB	= -fPIC
 QMAKE_CFLAGS_YACC	= -Wno-unused -Wno-parentheses
@@ -56,7 +56,7 @@
 QMAKE_LFLAGS_PLUGIN	= $$QMAKE_LFLAGS_SHLIB
 QMAKE_LFLAGS_SONAME	= -Wl,-soname,
 QMAKE_LFLAGS_THREAD	=
-QMAKE_RPATH		= -Wl,-rpath,
+QMAKE_RPATH		=

 QMAKE_LIBS		=
 QMAKE_LIBS_DYNLOAD	= -ldl
diff -Naur qt-x11-free-3.3.5/qmake/Makefile.unix qt-x11-free-3.3.5-kill-off-sed/qmake/Makefile.unix
--- qt-x11-free-3.3.5/qmake/Makefile.unix	2004-09-13 20:24:42.000000000 +0200
+++ qt-x11-free-3.3.5-kill-off-sed/qmake/Makefile.unix	2006-02-22 13:14:00.000000000 +0100
@@ -48,8 +48,7 @@
 	[ -d @QT_INSTALL_BINS@ ] || mkdir -p @QT_INSTALL_BINS@
 	-cp -f @BUILD_PATH@/bin/qmake @QT_INSTALL_BINS@
 	[ -d @QT_INSTALL_DATA@ ] || mkdir -p @QT_INSTALL_DATA@
-	-cp -R -f @SOURCE_PATH@/mkspecs @QT_INSTALL_DATA@
-	-ln -s -f @QMAKESPEC@ @QT_INSTALL_DATA@/mkspecs/default
+	
 
 clean::
 	rm -f $(OBJS) $(QOBJS)
