diff -Naur qt-x11-free-3.3.5/tools/designer/uic/form.cpp qt-x11-free-3.3.5-patched/tools/designer/uic/form.cpp
--- qt-x11-free-3.3.5/tools/designer/uic/form.cpp	2005-08-31 12:13:47.000000000 +0200
+++ qt-x11-free-3.3.5-patched/tools/designer/uic/form.cpp	2005-09-28 20:40:09.000000000 +0200
@@ -731,6 +731,13 @@
             while ( !n2.isNull() ) {
                 if ( n2.tagName() == "includehint" ) {
                     QString file = n2.firstChild().toText().data();
+                    int colons = file.find("::");
+
+                    if (colons != -1)
+                    {
+                        file = file.right(file.length() - colons - 2);
+                    }
+
                     localIncludes += file;
                 }
                 n2 = n2.nextSibling().toElement();
