diff -Naur qt-x11-free-3.3.5/include/private/qcom_p.h qt-x11-free-3.3.5-patched/include/private/qcom_p.h
--- qt-x11-free-3.3.5/include/private/qcom_p.h	2005-09-02 14:42:53.000000000 +0200
+++ qt-x11-free-3.3.5-patched/include/private/qcom_p.h	2005-10-05 15:51:28.000000000 +0200
@@ -53,6 +53,10 @@
 #include "quuid.h"
 #endif // QT_H
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
+
 #ifndef QT_NO_COMPONENT
 
 class QObject;
diff -Naur qt-x11-free-3.3.5/include/private/qimageformatinterface_p.h qt-x11-free-3.3.5-patched/include/private/qimageformatinterface_p.h
--- qt-x11-free-3.3.5/include/private/qimageformatinterface_p.h	2005-09-02 14:43:04.000000000 +0200
+++ qt-x11-free-3.3.5-patched/include/private/qimageformatinterface_p.h	2005-10-05 15:51:28.000000000 +0200
@@ -40,6 +40,9 @@
 #include <private/qcom_p.h>
 #endif // QT_H
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
 
 //
 //  W A R N I N G
diff -Naur qt-x11-free-3.3.5/include/private/qucomextra_p.h qt-x11-free-3.3.5-patched/include/private/qucomextra_p.h
--- qt-x11-free-3.3.5/include/private/qucomextra_p.h	2005-09-02 14:43:01.000000000 +0200
+++ qt-x11-free-3.3.5-patched/include/private/qucomextra_p.h	2005-10-05 15:51:28.000000000 +0200
@@ -54,6 +54,9 @@
 #include <private/qucom_p.h>
 #endif // QT_H
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
 
 class QVariant;
 
diff -Naur qt-x11-free-3.3.5/include/private/qucom_p.h qt-x11-free-3.3.5-patched/include/private/qucom_p.h
--- qt-x11-free-3.3.5/include/private/qucom_p.h	2005-09-02 14:42:54.000000000 +0200
+++ qt-x11-free-3.3.5-patched/include/private/qucom_p.h	2005-10-05 15:51:28.000000000 +0200
@@ -59,6 +59,10 @@
 #undef check
 #endif
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
+
 struct QUObject;
 struct QUInterfaceDescription;
 struct QUnknownInterface;
diff -Naur qt-x11-free-3.3.5/include/qasyncimageio.h qt-x11-free-3.3.5-patched/include/qasyncimageio.h
--- qt-x11-free-3.3.5/include/qasyncimageio.h	2005-09-02 14:43:05.000000000 +0200
+++ qt-x11-free-3.3.5-patched/include/qasyncimageio.h	2005-10-05 15:51:28.000000000 +0200
@@ -44,6 +44,9 @@
 
 #ifndef QT_NO_ASYNC_IMAGE_IO
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
 
 class Q_EXPORT QImageConsumer {
 public:
diff -Naur qt-x11-free-3.3.5/include/qfiledialog.h qt-x11-free-3.3.5-patched/include/qfiledialog.h
--- qt-x11-free-3.3.5/include/qfiledialog.h	2005-09-02 14:42:52.000000000 +0200
+++ qt-x11-free-3.3.5-patched/include/qfiledialog.h	2005-10-05 15:51:28.000000000 +0200
@@ -58,6 +58,10 @@
 #include "qurlinfo.h"
 #endif // QT_H
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
+
 #ifndef QT_NO_FILEDIALOG
 
 class Q_EXPORT QFileIconProvider : public QObject
diff -Naur qt-x11-free-3.3.5/include/qnetworkprotocol.h qt-x11-free-3.3.5-patched/include/qnetworkprotocol.h
--- qt-x11-free-3.3.5/include/qnetworkprotocol.h	2005-09-02 14:43:05.000000000 +0200
+++ qt-x11-free-3.3.5-patched/include/qnetworkprotocol.h	2005-10-05 15:51:28.000000000 +0200
@@ -47,6 +47,10 @@
 
 #ifndef QT_NO_NETWORKPROTOCOL
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
+
 class QNetworkProtocol;
 class QNetworkOperation;
 class QTimer;
diff -Naur qt-x11-free-3.3.5/include/qtooltip.h qt-x11-free-3.3.5-patched/include/qtooltip.h
--- qt-x11-free-3.3.5/include/qtooltip.h	2005-09-02 14:43:15.000000000 +0200
+++ qt-x11-free-3.3.5-patched/include/qtooltip.h	2005-10-05 15:51:28.000000000 +0200
@@ -42,6 +42,9 @@
 
 #ifndef QT_NO_TOOLTIP
 
+#if __GNUC__  - 0 > 3
+#pragma GCC system_header
+#endif
 
 class QTipManager;
 class QIconViewToolTip;
diff -Naur qt-x11-free-3.3.5/src/dialogs/qfiledialog.h qt-x11-free-3.3.5-patched/src/dialogs/qfiledialog.h
--- qt-x11-free-3.3.5/src/dialogs/qfiledialog.h	2005-09-02 14:42:52.000000000 +0200
+++ qt-x11-free-3.3.5-patched/src/dialogs/qfiledialog.h	2005-10-05 15:51:28.000000000 +0200
@@ -58,6 +58,10 @@
 #include "qurlinfo.h"
 #endif // QT_H
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
+
 #ifndef QT_NO_FILEDIALOG
 
 class Q_EXPORT QFileIconProvider : public QObject
diff -Naur qt-x11-free-3.3.5/src/kernel/qasyncimageio.h qt-x11-free-3.3.5-patched/src/kernel/qasyncimageio.h
--- qt-x11-free-3.3.5/src/kernel/qasyncimageio.h	2005-09-02 14:43:05.000000000 +0200
+++ qt-x11-free-3.3.5-patched/src/kernel/qasyncimageio.h	2005-10-05 15:51:28.000000000 +0200
@@ -44,6 +44,9 @@
 
 #ifndef QT_NO_ASYNC_IMAGE_IO
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
 
 class Q_EXPORT QImageConsumer {
 public:
diff -Naur qt-x11-free-3.3.5/src/kernel/qimageformatinterface_p.h qt-x11-free-3.3.5-patched/src/kernel/qimageformatinterface_p.h
--- qt-x11-free-3.3.5/src/kernel/qimageformatinterface_p.h	2005-09-02 14:43:04.000000000 +0200
+++ qt-x11-free-3.3.5-patched/src/kernel/qimageformatinterface_p.h	2005-10-05 15:51:28.000000000 +0200
@@ -40,6 +40,9 @@
 #include <private/qcom_p.h>
 #endif // QT_H
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
 
 //
 //  W A R N I N G
diff -Naur qt-x11-free-3.3.5/src/kernel/qnetworkprotocol.h qt-x11-free-3.3.5-patched/src/kernel/qnetworkprotocol.h
--- qt-x11-free-3.3.5/src/kernel/qnetworkprotocol.h	2005-09-02 14:43:05.000000000 +0200
+++ qt-x11-free-3.3.5-patched/src/kernel/qnetworkprotocol.h	2005-10-05 15:51:28.000000000 +0200
@@ -47,6 +47,10 @@
 
 #ifndef QT_NO_NETWORKPROTOCOL
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
+
 class QNetworkProtocol;
 class QNetworkOperation;
 class QTimer;
diff -Naur qt-x11-free-3.3.5/src/kernel/qucomextra_p.h qt-x11-free-3.3.5-patched/src/kernel/qucomextra_p.h
--- qt-x11-free-3.3.5/src/kernel/qucomextra_p.h	2005-09-02 14:43:01.000000000 +0200
+++ qt-x11-free-3.3.5-patched/src/kernel/qucomextra_p.h	2005-10-05 15:51:28.000000000 +0200
@@ -54,6 +54,9 @@
 #include <private/qucom_p.h>
 #endif // QT_H
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
 
 class QVariant;
 
diff -Naur qt-x11-free-3.3.5/src/tools/qcom_p.h qt-x11-free-3.3.5-patched/src/tools/qcom_p.h
--- qt-x11-free-3.3.5/src/tools/qcom_p.h	2005-09-02 14:42:53.000000000 +0200
+++ qt-x11-free-3.3.5-patched/src/tools/qcom_p.h	2005-10-05 15:51:28.000000000 +0200
@@ -53,6 +53,10 @@
 #include "quuid.h"
 #endif // QT_H
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
+
 #ifndef QT_NO_COMPONENT
 
 class QObject;
diff -Naur qt-x11-free-3.3.5/src/tools/qucom_p.h qt-x11-free-3.3.5-patched/src/tools/qucom_p.h
--- qt-x11-free-3.3.5/src/tools/qucom_p.h	2005-09-02 14:42:54.000000000 +0200
+++ qt-x11-free-3.3.5-patched/src/tools/qucom_p.h	2005-10-05 15:51:28.000000000 +0200
@@ -59,6 +59,10 @@
 #undef check
 #endif
 
+#if __GNUC__ - 0 > 3
+#pragma GCC system_header
+#endif
+
 struct QUObject;
 struct QUInterfaceDescription;
 struct QUnknownInterface;
diff -Naur qt-x11-free-3.3.5/src/widgets/qmainwindow.cpp qt-x11-free-3.3.5-patched/src/widgets/qmainwindow.cpp
--- qt-x11-free-3.3.5/src/widgets/qmainwindow.cpp	2005-09-02 14:43:14.000000000 +0200
+++ qt-x11-free-3.3.5-patched/src/widgets/qmainwindow.cpp	2005-10-05 15:51:28.000000000 +0200
@@ -287,8 +287,9 @@
 {
 public:
     QHideToolTip( QWidget *parent ) : QToolTip( parent ) {}
+    ~QHideToolTip() {}
 
-    void maybeTip( const QPoint &pos );
+    virtual void maybeTip( const QPoint &pos );
 };
 #endif
 
diff -Naur qt-x11-free-3.3.5/src/widgets/qtooltip.h qt-x11-free-3.3.5-patched/src/widgets/qtooltip.h
--- qt-x11-free-3.3.5/src/widgets/qtooltip.h	2005-09-02 14:43:15.000000000 +0200
+++ qt-x11-free-3.3.5-patched/src/widgets/qtooltip.h	2005-10-05 15:51:28.000000000 +0200
@@ -42,6 +42,9 @@
 
 #ifndef QT_NO_TOOLTIP
 
+#if __GNUC__  - 0 > 3
+#pragma GCC system_header
+#endif
 
 class QTipManager;
 class QIconViewToolTip;
