--- qt-x11-free-3.3.5/src/widgets/qscrollview.cpp	2005-09-02 14:43:14.000000000 +0200
+++ qt-x11-free-3.3.5-patched/src/widgets/qscrollview.cpp	2005-09-29 02:46:46.000000000 +0200
@@ -1861,7 +1861,7 @@
        the event itself.
     */
     QWheelEvent ce( viewportToContents(e->pos()),
-        e->globalPos(), e->delta(), e->state());
+        e->globalPos(), e->delta(), e->state(), e->orientation());
     contentsWheelEvent(&ce);
     if ( ce.isAccepted() )
         e->accept();
