Index: main-menu/src/main-menu-ui.c
===================================================================
--- main-menu/src/main-menu-ui.c	(revision 174)
+++ main-menu/src/main-menu-ui.c	(working copy)
@@ -442,9 +442,9 @@
 
 		gtk_box_pack_start (GTK_BOX (status_section), hd_tile, FALSE, FALSE, 0);
 	}
-
+#if 0
 	net_tile = network_status_tile_new ();
-
+#endif
 	if (net_tile)
 	{
 		g_signal_connect (G_OBJECT (net_tile), "tile-action-triggered",
Index: main-menu/src/Makefile.am
===================================================================
--- main-menu/src/Makefile.am	(revision 174)
+++ main-menu/src/Makefile.am	(working copy)
@@ -19,9 +19,6 @@
 	main-menu-engine.c						\
 	main-menu-ui.c							\
 	main-menu-conf.c		main-menu-conf.h		\
-	network-status-tile.c		network-status-tile.h		\
-	network-status-agent.c		network-status-agent.h		\
-	network-status-info.c		network-status-info.h		\
 	slab-window.c			slab-window.h			\
 	system-tile.c			system-tile.h			\
 	text-button.c			text-button.h			\
Index: configure.in
===================================================================
--- configure.in	(revision 174)
+++ configure.in	(working copy)
@@ -43,7 +43,7 @@
 AC_SUBST(LIBSLAB_CFLAGS)
 AC_SUBST(LIBSLAB_LIBS)
 
-PKG_CHECK_MODULES(MAIN_MENU, [ $COMMON_MODULES libpanelapplet-2.0 gnome-desktop-2.0 gconf-2.0 libgtop-2.0 libgnome-2.0 libgnomeui-2.0 dbus-glib-1 NetworkManager hal-storage ])
+PKG_CHECK_MODULES(MAIN_MENU, [ $COMMON_MODULES libpanelapplet-2.0 gnome-desktop-2.0 gconf-2.0 libgtop-2.0 libgnome-2.0 libgnomeui-2.0 dbus-glib-1 hal-storage ])
 
 AC_CHECK_HEADERS(iwlib.h, [ AC_CHECK_LIB(iw, iw_sockets_open, have_iwlib=yes, have_iwlib=no)], have_iwlib=no)
 
Index: libslab/application-tile.c
===================================================================
--- libslab/application-tile.c	(revision 174)
+++ libslab/application-tile.c	(working copy)
@@ -626,8 +626,6 @@
 	gboolean retval;
 
 	cmd = get_slab_gconf_string (gconf_key);
-	if (!cmd)
-		return FALSE;
 
 	args = strchr (cmd, ' ');
 
