diff -Naurp libquicktime-0.9.10/configure.ac libquicktime-0.9.10-p/configure.ac
--- libquicktime-0.9.10/configure.ac	2006-08-03 00:53:54.000000000 +0200
+++ libquicktime-0.9.10-p/configure.ac	2006-11-19 14:10:49.000000000 +0100
@@ -563,7 +563,7 @@ if test x$LQT_DEBUG != xtrue; then
 AC_DEFINE(NDEBUG)
 fi
 
-CFLAGS="$OPT_CFLAGS $COMMON_CFLAGS"
+CFLAGS="$CFLAGS $COMMON_CFLAGS"
 
 # -L is required so that linker can find libquicktime when DESTDIR is used.
 # libtool bug, documented at: http://www.geocrawler.com/mail/thread.php3?subject=install+phase+fails&list=404
