diff -Naur kdelibs-3.4.2.orig/kdecore/kglobalsettings.cpp kdelibs-3.4.2/kdecore/kglobalsettings.cpp
--- kdelibs-3.4.2.orig/kdecore/kglobalsettings.cpp	2005-05-23 14:16:21.000000000 +0200
+++ kdelibs-3.4.2/kdecore/kglobalsettings.cpp	2005-08-01 10:22:27.000000000 +0200
@@ -321,8 +321,8 @@
         return *_generalFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _generalFont = new QFont("helvetica", 12);
-    _generalFont->setPointSize(12);
+    _generalFont = new QFont("Sans Serif", 13);
+    _generalFont->setPointSize(13);
     _generalFont->setStyleHint(QFont::SansSerif);
 
     KConfigGroup g( KGlobal::config(), "General" );
@@ -337,8 +337,8 @@
         return *_fixedFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _fixedFont = new QFont("courier", 12);
-    _fixedFont->setPointSize(12);
+    _fixedFont = new QFont("Monospace", 13);
+    _fixedFont->setPointSize(13);
     _fixedFont->setStyleHint(QFont::TypeWriter);
 
     KConfigGroup g( KGlobal::config(), "General" );
@@ -353,8 +353,8 @@
         return *_toolBarFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _toolBarFont = new QFont("helvetica", 10);
-    _toolBarFont->setPointSize(10);
+    _toolBarFont = new QFont("Sans Serif", 13);
+    _toolBarFont->setPointSize(13);
     _toolBarFont->setStyleHint(QFont::SansSerif);
 
     KConfigGroup g( KGlobal::config(), "General" );
@@ -369,8 +369,8 @@
         return *_menuFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _menuFont = new QFont("helvetica", 12);
-    _menuFont->setPointSize(12);
+    _menuFont = new QFont("Sans Serif", 13);
+    _menuFont->setPointSize(13);
     _menuFont->setStyleHint(QFont::SansSerif);
 
     KConfigGroup g( KGlobal::config(), "General" );
@@ -385,8 +385,8 @@
         return *_windowTitleFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _windowTitleFont = new QFont("helvetica", 12, QFont::Bold);
-    _windowTitleFont->setPointSize(12);
+    _windowTitleFont = new QFont("Luxi Sans", 13, QFont::Bold);
+    _windowTitleFont->setPointSize(13);
     _windowTitleFont->setStyleHint(QFont::SansSerif);
 
     KConfigGroup g( KGlobal::config(), "WM" );
@@ -401,7 +401,7 @@
         return *_taskbarFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _taskbarFont = new QFont("helvetica", 11);
+    _taskbarFont = new QFont("Sans Serif", 11);
     _taskbarFont->setPointSize(11);
     _taskbarFont->setStyleHint(QFont::SansSerif);
 
diff -Naur kdelibs-3.4.2.orig/khtml/khtmldefaults.h kdelibs-3.4.2/khtml/khtmldefaults.h
--- kdelibs-3.4.2.orig/khtml/khtmldefaults.h	2005-05-23 14:16:29.000000000 +0200
+++ kdelibs-3.4.2/khtml/khtmldefaults.h	2005-08-01 10:21:34.000000000 +0200
@@ -26,12 +26,12 @@
 
 // KEEP IN SYNC WITH konqdefaults.h in kdebase/libkonq!
 // lets be modern .. -- Bernd
-#define HTML_DEFAULT_VIEW_FONT "helvetica"
-#define HTML_DEFAULT_VIEW_FIXED_FONT "courier"
+#define HTML_DEFAULT_VIEW_FONT "Sans Serif"
+#define HTML_DEFAULT_VIEW_FIXED_FONT "Monospace"
 // generic CSS fonts. Since usual X distributions don't have a good set of fonts, this
 // is quite conservative...
-#define HTML_DEFAULT_VIEW_SERIF_FONT "times"
-#define HTML_DEFAULT_VIEW_SANSSERIF_FONT "helvetica"
-#define HTML_DEFAULT_VIEW_CURSIVE_FONT "helvetica"
-#define HTML_DEFAULT_VIEW_FANTASY_FONT "helvetica"
+#define HTML_DEFAULT_VIEW_SERIF_FONT "Serif"
+#define HTML_DEFAULT_VIEW_SANSSERIF_FONT "Sans Serif"
+#define HTML_DEFAULT_VIEW_CURSIVE_FONT "Sans Serif"
+#define HTML_DEFAULT_VIEW_FANTASY_FONT "Sans Serif"
 #define HTML_DEFAULT_MIN_FONT_SIZE 7 // everything smaller is usually unreadable.
diff -Naur kdelibs-3.4.2.orig/khtml/khtmlview.cpp kdelibs-3.4.2/khtml/khtmlview.cpp
--- kdelibs-3.4.2.orig/khtml/khtmlview.cpp	2005-05-23 14:16:29.000000000 +0200
+++ kdelibs-3.4.2/khtml/khtmlview.cpp	2005-08-01 10:21:34.000000000 +0200
@@ -2358,7 +2358,7 @@
         bool printHeader = (printer->option("app-khtml-printheader") == "true");
 
         int headerHeight = 0;
-        QFont headerFont("helvetica", 8);
+        QFont headerFont("Sans Serif", 8);
 
         QString headerLeft = KGlobal::locale()->formatDate(QDate::currentDate(),true);
         QString headerMid = docname;
