diff -Naur kdelibs-3.5.5/admin/cvs.sh kdelibs-3.5.5-p/admin/cvs.sh
--- kdelibs-3.5.5/admin/cvs.sh	2006-10-02 17:01:27.000000000 +0200
+++ kdelibs-3.5.5-p/admin/cvs.sh	2006-10-26 16:01:41.000000000 +0200
@@ -68,7 +68,7 @@
     echo "*** KDE requires automake $required_automake_version"
     exit 1
     ;;
-  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9*)
+  automake*1.6.* | automake*1.7* | automake*1.8* | automake*1.9* | automake*1.10* )
     echo "*** $AUTOMAKE_STRING found."
     UNSERMAKE=no
     ;;
