diff -Naur kdelibs-4.4.0/cmake/modules/FindKDE4Internal.cmake kdelibs-4.4.0-cmake-print-flags/cmake/modules/FindKDE4Internal.cmake
--- kdelibs-4.4.0/cmake/modules/FindKDE4Internal.cmake	2010-01-20 00:15:16.000000000 +0100
+++ kdelibs-4.4.0-cmake-print-flags/cmake/modules/FindKDE4Internal.cmake	2010-02-10 15:50:35.000000000 +0100
@@ -1084,8 +1084,10 @@
 
    if (CMAKE_SYSTEM_NAME MATCHES Linux OR CMAKE_SYSTEM_NAME STREQUAL GNU)
      set ( CMAKE_C_FLAGS     "${CMAKE_C_FLAGS} -Wno-long-long -std=iso9899:1990 -Wundef -Wcast-align -Werror-implicit-function-declaration -Wchar-subscripts -Wall -W -Wpointer-arith -Wwrite-strings -Wformat-security -Wmissing-format-attribute -fno-common")
+     message(STATUS "CFLAGS: ${CMAKE_C_FLAGS}")
      # As off Qt 4.6.x we need to override the new exception macros if we want compile with -fno-exceptions
      set ( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wnon-virtual-dtor -Wno-long-long -ansi -Wundef -Wcast-align -Wchar-subscripts -Wall -W -Wpointer-arith -Wformat-security -fno-exceptions -DQT_NO_EXCEPTIONS -fno-check-new -fno-common")
+     message(STATUS " CXXFLAGS: ${CMAKE_CXX_FLAGS}")
      add_definitions (-D_BSD_SOURCE)
    endif (CMAKE_SYSTEM_NAME MATCHES Linux OR CMAKE_SYSTEM_NAME STREQUAL GNU)
 
