diff -Naur kdelibs-4.3.4/kdeui/kernel/kglobalsettings.cpp kdelibs-4.3.4-p/kdeui/kernel/kglobalsettings.cpp
--- kdelibs-4.3.4/kdeui/kernel/kglobalsettings.cpp	2009-10-30 13:26:22.000000000 +0100
+++ kdelibs-4.3.4-p/kdeui/kernel/kglobalsettings.cpp	2010-01-19 11:18:12.000000000 +0100
@@ -94,6 +94,7 @@
         WindowTitleFont,
         TaskbarFont ,
         SmallestReadableFont,
+	DesktopFont,
         FontTypesCount
     };
 
@@ -381,7 +382,7 @@
 
 // NOTE: keep in sync with kdebase/workspace/kcontrol/fonts/fonts.cpp
 static const char GeneralId[] =      "General";
-static const char DefaultFont[] =    "Sans Serif";
+static const char DefaultFont[] =    "DejaVu Sans";
 #ifdef Q_WS_MAC
 static const char DefaultMacFont[] = "Lucida Grande";
 #endif
@@ -394,14 +395,15 @@
     { GeneralId, "toolBarFont", DefaultMacFont, 11, -1, QFont::SansSerif },
     { GeneralId, "menuFont",    DefaultMacFont, 13, -1, QFont::SansSerif },
 #else
-    { GeneralId, "font",        DefaultFont, 10, -1, QFont::SansSerif },
-    { GeneralId, "fixed",       "Monospace", 10, -1, QFont::TypeWriter },
-    { GeneralId, "toolBarFont", DefaultFont,  8, -1, QFont::SansSerif },
-    { GeneralId, "menuFont",    DefaultFont, 10, -1, QFont::SansSerif },
-#endif
-    { "WM",      "activeFont",           DefaultFont,  9, QFont::Bold, QFont::SansSerif },// inconsistency
-    { GeneralId, "taskbarFont",          DefaultFont, 10, -1, QFont::SansSerif },
-    { GeneralId, "smallestReadableFont", DefaultFont,  8, -1, QFont::SansSerif }
+    { GeneralId, "font",        DefaultFont, 10, -1},
+    { GeneralId, "fixed",       "DejaVu Sans Mono", 10, -1},
+    { GeneralId, "toolBarFont", DefaultFont,  10, -1},
+    { GeneralId, "menuFont",    DefaultFont, 11, -1},
+#endif
+    { "WM",      "activeFont",           DefaultFont,  10, QFont::Bold},// inconsistency
+    { GeneralId, "taskbarFont",          DefaultFont, 10, -1},
+    { GeneralId, "smallestReadableFont", DefaultFont,  8, -1},
+    { GeneralId, "desktopFont", 	 DefaultFont,  9, -1}
 };
 
 QFont KGlobalSettingsData::font( FontTypes fontType )
@@ -462,6 +464,7 @@
     // most preferred last
     static const char* const PreferredFontNames[] =
     {
+	"DejaVu Sans",
         "Arial",
         "Sans Serif",
         "Verdana",
