diff -Naurp kdelibs-3.5.5/kdeui/kaboutdialog.cpp kdelibs-3.5.5-p/kdeui/kaboutdialog.cpp
--- kdelibs-3.5.5/kdeui/kaboutdialog.cpp	2006-01-19 18:07:18.000000000 +0100
+++ kdelibs-3.5.5-p/kdeui/kaboutdialog.cpp	2006-10-27 13:40:11.000000000 +0200
@@ -40,6 +40,7 @@
 #include <kaboutdialog.h>
 #include <kaboutdialog_private.h>
 #include <kdebug.h>
+#include <config.h>
 
 //MOC_SKIP_BEGIN
 template class QPtrList<KAboutContributor>;
@@ -702,7 +703,7 @@ void KAboutContainerBase::setProduct( co
     kdDebug(291) << "no kapp" << endl;
 
   const QString msg1 = i18n("%1 %2 (Using KDE %3)").arg(appName).arg(version).
-    arg(QString::fromLatin1(KDE_VERSION_STRING));
+    arg(QString::fromLatin1(KDE_VERSION_STRING) + " " + QString::fromLatin1(KDE_DISTRIBUTION_TEXT));
   const QString msg2 = !year.isEmpty() ? i18n("%1 %2, %3").arg('©').arg(year).
     arg(author) : QString::fromLatin1("");
 
