ndex: kdecore/configure.in.in
===================================================================
--- kdecore/configure.in.in	(Revision 624655)
+++ kdecore/configure.in.in	(Arbeitskopie)
@@ -1,6 +1,8 @@
 dnl Compile in the exec prefix to help kstddirs in finding dynamic libs
 AC_DEFINE_UNQUOTED(__KDE_EXECPREFIX, "$exec_prefix", [execprefix or NONE if not set, for libloading])
 
+AM_PROG_CC_C_O
+
 dnl Compile in kde_bindir to safely find kdesu_stub.
 if test "$exec_prefix" = "NONE"; then
     bindir_str="\"$prefix/bin\""

