diff -Naur kdelibs-3.5.1/kdecore/kglobalsettings.cpp kdelibs-3.5.1-fonts/kdecore/kglobalsettings.cpp
--- kdelibs-3.5.1/kdecore/kglobalsettings.cpp	2005-10-10 17:06:03.000000000 +0200
+++ kdelibs-3.5.1-fonts/kdecore/kglobalsettings.cpp	2006-03-10 22:02:33.000000000 +0100
@@ -322,8 +322,8 @@
         return *_generalFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _generalFont = new QFont("Sans Serif", 10);
-    _generalFont->setPointSize(10);
+    _generalFont = new QFont("DejaVu Sans", 13);
+    _generalFont->setPointSize(13);
     _generalFont->setStyleHint(QFont::SansSerif);
 
     KConfigGroup g( KGlobal::config(), "General" );
@@ -338,8 +338,8 @@
         return *_fixedFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _fixedFont = new QFont("Monospace", 10);
-    _fixedFont->setPointSize(10);
+    _fixedFont = new QFont("DejaVu Sans Mono", 13);
+    _fixedFont->setPointSize(13);
     _fixedFont->setStyleHint(QFont::TypeWriter);
 
     KConfigGroup g( KGlobal::config(), "General" );
@@ -354,8 +354,8 @@
         return *_toolBarFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _toolBarFont = new QFont("Sans Serif", 10);
-    _toolBarFont->setPointSize(10);
+    _toolBarFont = new QFont("DejaVu Sans", 13);
+    _toolBarFont->setPointSize(13);
     _toolBarFont->setStyleHint(QFont::SansSerif);
 
     KConfigGroup g( KGlobal::config(), "General" );
@@ -370,8 +370,8 @@
         return *_menuFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _menuFont = new QFont("Sans Serif", 10);
-    _menuFont->setPointSize(10);
+    _menuFont = new QFont("DejaVu Sans", 13);
+    _menuFont->setPointSize(13);
     _menuFont->setStyleHint(QFont::SansSerif);
 
     KConfigGroup g( KGlobal::config(), "General" );
@@ -386,8 +386,8 @@
         return *_windowTitleFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _windowTitleFont = new QFont("Sans Serif", 9, QFont::Bold);
-    _windowTitleFont->setPointSize(10);
+    _windowTitleFont = new QFont("DejaVu Serif Condensed", 12, QFont::Bold);
+    _windowTitleFont->setPointSize(12);
     _windowTitleFont->setStyleHint(QFont::SansSerif);
 
     KConfigGroup g( KGlobal::config(), "WM" );
@@ -402,8 +402,8 @@
         return *_taskbarFont;
 
     // Sync default with kdebase/kcontrol/fonts/fonts.cpp
-    _taskbarFont = new QFont("Sans Serif", 10);
-    _taskbarFont->setPointSize(10);
+    _taskbarFont = new QFont("DejaVu Sans", 13);
+    _taskbarFont->setPointSize(13);
     _taskbarFont->setStyleHint(QFont::SansSerif);
 
     KConfigGroup g( KGlobal::config(), "General" );
@@ -419,6 +419,8 @@
     QStringList fam = db.families();
 
     // Move a bunch of preferred fonts to the front.
+    if (fam.remove("DejaVu Sans"))
+       fam.prepend("DejaVu Sans");
     if (fam.remove("Arial"))
        fam.prepend("Arial");
     if (fam.remove("Verdana"))
diff -Naur kdelibs-3.5.1/khtml/khtmldefaults.h kdelibs-3.5.1-fonts/khtml/khtmldefaults.h
--- kdelibs-3.5.1/khtml/khtmldefaults.h	2005-10-10 17:06:13.000000000 +0200
+++ kdelibs-3.5.1-fonts/khtml/khtmldefaults.h	2006-03-10 21:53:16.000000000 +0100
@@ -24,10 +24,10 @@
 #define HTML_DEFAULT_VLNK_COLOR Qt::magenta
 #define HTML_DEFAULT_BASE_COLOR Qt::white
 
-#define HTML_DEFAULT_VIEW_FONT "Sans Serif"
-#define HTML_DEFAULT_VIEW_FIXED_FONT "Monospace"
-#define HTML_DEFAULT_VIEW_SERIF_FONT "Serif"
-#define HTML_DEFAULT_VIEW_SANSSERIF_FONT "Sans Serif"
-#define HTML_DEFAULT_VIEW_CURSIVE_FONT "Sans Serif"
-#define HTML_DEFAULT_VIEW_FANTASY_FONT "Sans Serif"
-#define HTML_DEFAULT_MIN_FONT_SIZE 7 // everything smaller is usually unreadable.
+#define HTML_DEFAULT_VIEW_FONT "DejaVu Sans"
+#define HTML_DEFAULT_VIEW_FIXED_FONT "DejaVu Sans Mono"
+#define HTML_DEFAULT_VIEW_SERIF_FONT "DejaVu Serif"
+#define HTML_DEFAULT_VIEW_SANSSERIF_FONT "DejaVu Sans"
+#define HTML_DEFAULT_VIEW_CURSIVE_FONT "DejaVu Serif"
+#define HTML_DEFAULT_VIEW_FANTASY_FONT "DejaVu Serif"
+#define HTML_DEFAULT_MIN_FONT_SIZE 8 
