diff -Naurp kdelibs-3.5.5/kded/kbuildservicefactory.cpp kdelibs-3.5.5-p/kded/kbuildservicefactory.cpp
--- kdelibs-3.5.5/kded/kbuildservicefactory.cpp	2005-10-10 17:06:29.000000000 +0200
+++ kdelibs-3.5.5-p/kded/kbuildservicefactory.cpp	2006-10-27 04:07:33.000000000 +0200
@@ -158,7 +158,7 @@ KBuildServiceFactory::saveOfferList(QDat
          KServiceType::Ptr serviceType = KServiceType::serviceType(*it);
          if (!serviceType)
          {
-           kdWarning() << "'"<< service->desktopEntryPath() << "' specifies undefined mimetype/servicetype '"<< (*it) << "'" << endl;
+           kdDebug() << "'"<< service->desktopEntryPath() << "' specifies undefined mimetype/servicetype '"<< (*it) << "'" << endl;
            continue;
          }
          serviceTypes.append(serviceType);
