diff -Naur kdelibs-3.5.1/khtml/dom/html_misc.h kdelibs-3.5.1-patched/khtml/dom/html_misc.h
--- kdelibs-3.5.1/khtml/dom/html_misc.h	2006-01-19 18:06:53.000000000 +0100
+++ kdelibs-3.5.1-patched/khtml/dom/html_misc.h	2006-02-08 22:30:38.134200250 +0100
@@ -211,7 +211,7 @@
 /**
   @internal. Not part of the public API
 */
-class HTMLMappedNameCollection : public HTMLCollection
+class KHTML_EXPORT HTMLMappedNameCollection : public HTMLCollection
 {
 public:
     HTMLMappedNameCollection(NodeImpl *base, int type, const DOMString &name );
