diff -Naur kdelibs-3.4.92/configure kdelibs-3.4.92-patched/configure
--- kdelibs-3.4.92/configure	2005-10-12 10:38:13.000000000 +0200
+++ kdelibs-3.4.92-patched/configure	2005-11-04 19:39:09.000000000 +0100
@@ -34066,9 +34066,9 @@
 
 kde_save_LIBS="$LIBS"
 if test "x$kde_use_qt_emb" != "xyes" && test "x$kde_use_qt_mac" != "xyes"; then
-LIBS="$LIBS $all_libraries $USER_LDFLAGS -lpng $LIBZ -lm -lX11 $LIBSOCKET"
+LIBS="$LIBS $all_libraries $USER_LDFLAGS $LIBZ -lpng -lm -lX11 $LIBSOCKET"
 else
-LIBS="$LIBS $all_libraries $USER_LDFLAGS -lpng $LIBZ -lm"
+LIBS="$LIBS $all_libraries $USER_LDFLAGS $LIBZ -lpng -lm"
 fi
 kde_save_CFLAGS="$CFLAGS"
 CFLAGS="$CFLAGS $all_includes $USER_INCLUDES"
@@ -34116,7 +34116,7 @@
   ac_status=$?
   echo "$as_me:$LINENO: \$? = $ac_status" >&5
   (exit $ac_status); }; }; then
-  eval "ac_cv_lib_png='-lpng $LIBZ -lm'"
+  eval "ac_cv_lib_png='$LIBZ -lpng -lm'"
 else
   echo "$as_me: failed program was:" >&5
 sed 's/^/| /' conftest.$ac_ext >&5
diff -Naur kdelibs-3.4.92/kimgio/Makefile.in kdelibs-3.4.92-patched/kimgio/Makefile.in
--- kdelibs-3.4.92/kimgio/Makefile.in	2005-10-12 10:37:33.000000000 +0200
+++ kdelibs-3.4.92-patched/kimgio/Makefile.in	2005-11-04 19:39:09.000000000 +0100
@@ -657,7 +657,7 @@
 #>- kimg_exr_la_LDFLAGS = -module $(KIMGIO_PLUGIN) -no-undefined $(all_libraries)
 #>+ 1
 kimg_exr_la_LDFLAGS = -module $(KIMGIO_PLUGIN) -no-undefined $(KDE_NO_UNDEFINED) $(all_libraries)
-kimg_exr_la_LIBADD = $(LIB_QT) $(LIB_KDECORE) $(LIB_EXR)
+kimg_exr_la_LIBADD = $(LIB_EXR) $(LIB_KDECORE) $(LIB_QT)
 kimg_dds_la_SOURCES = dds.cpp
 #>- kimg_dds_la_LDFLAGS = -module $(KIMGIO_PLUGIN) -no-undefined $(all_libraries)
 #>+ 1
