Index: kioslave/metainfo/Makefile.am
===================================================================
--- kioslave/metainfo/Makefile.am	(Revision 606834)
+++ kioslave/metainfo/Makefile.am	(Arbeitskopie)
@@ -2,7 +2,7 @@
 ## Makefile.am of kdebase/kioslave/metainfo
 
 INCLUDES = $(all_includes)
-AM_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+AM_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 METASOURCES = AUTO
 
 kde_module_LTLIBRARIES = kio_metainfo.la
Index: kioslave/gzip/Makefile.am
===================================================================
--- kioslave/gzip/Makefile.am	(Revision 606834)
+++ kioslave/gzip/Makefile.am	(Arbeitskopie)
@@ -1,5 +1,5 @@
 INCLUDES = -I$(top_srcdir)/kio $(all_includes)
-AM_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+AM_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 METASOURCES = AUTO
 
 kde_module_LTLIBRARIES = kgzipfilter.la
Index: kio/tests/Makefile.am
===================================================================
--- kio/tests/Makefile.am	(Revision 606834)
+++ kio/tests/Makefile.am	(Arbeitskopie)
@@ -18,7 +18,7 @@
 
 INCLUDES= -I$(top_srcdir) -I$(top_srcdir)/kio $(all_includes)
 LDADD = $(LIB_KIO)
-AM_LDFLAGS = $(all_libraries) $(KDE_RPATH)
+AM_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
 check_PROGRAMS = ksycocatest getalltest kruntest ktartest kziptest\
                  kioslavetest kdirwatchtest kshredtest speed kurifiltertest \
Index: dcop/tests/Makefile.am
===================================================================
--- dcop/tests/Makefile.am	(Revision 606834)
+++ dcop/tests/Makefile.am	(Arbeitskopie)
@@ -3,7 +3,6 @@
 	@echo -e "dcop tests not run.\nRun make tests in the tests directory to make them"
 
 INCLUDES = $(all_includes)
-AM_LDFLAGS   = $(all_libraries)
 
 METASOURCES = AUTO
 
@@ -17,7 +16,7 @@
 
 driver_SOURCES = driver.cpp test.stub
 driver_LDADD = $(LIB_KDECORE)
-driver_LDFLAGS = $(KDE_RPATH)
+driver_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
 CLEANFILES = batch.generated definitions.generated driver.generated shell.generated
 
Index: dcop/dcopidl2cpp/Makefile.am
===================================================================
--- dcop/dcopidl2cpp/Makefile.am	(Revision 606834)
+++ dcop/dcopidl2cpp/Makefile.am	(Arbeitskopie)
@@ -1,6 +1,5 @@
 
 INCLUDES = $(all_includes)
-AM_LDFLAGS   = $(all_libraries)
 
 ####### Files
 
@@ -10,7 +9,7 @@
 noinst_HEADERS = main.h
 
 dcopidl2cpp_LDADD = $(LIB_QT)
-dcopidl2cpp_LDFLAGS = $(KDE_RPATH)
+dcopidl2cpp_LDFLAGS = $(all_libraries) $(KDE_RPATH)
 
 dcopidl_output.kidl: ../dcopidl/dcopidl $(srcdir)/dcopidl_test.h
 	../dcopidl/dcopidl $(srcdir)/dcopidl_test.h > dcopidl_output.kidl
Index: dcop/client/Makefile.am
===================================================================
--- dcop/client/Makefile.am	(Revision 606834)
+++ dcop/client/Makefile.am	(Arbeitskopie)
@@ -1,6 +1,5 @@
 
 INCLUDES = $(all_includes)
-AM_LDFLAGS   = $(all_libraries)
 DCOP_LIB = ../libDCOP.la
 
 ####### Files
@@ -9,19 +8,19 @@
 
 dcop_SOURCES = dcop.cpp
 dcop_LDADD = $(LIB_QT) $(DCOP_LIB)
-dcop_LDFLAGS = $(KDE_RPATH)
+dcop_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
 dcopstart_SOURCES = dcopstart.cpp
 dcopstart_LDADD = $(LIB_QT) $(DCOP_LIB)
-dcopstart_LDFLAGS = $(KDE_RPATH)
+dcopstart_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
 dcopquit_SOURCES = dcopquit.cpp
 dcopquit_LDADD = $(LIB_QT) $(DCOP_LIB)
-dcopquit_LDFLAGS = $(KDE_RPATH)
+dcopquit_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
 dcopfind_SOURCES = dcopfind.cpp
 dcopfind_LDADD = $(LIB_QT) $(DCOP_LIB)
-dcopfind_LDFLAGS = $(KDE_RPATH)
+dcopfind_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
 dcopclient_SOURCES = dcopclient.c
 dcopobject_SOURCES = dcopobject.c
Index: dcop/dcopidl/Makefile.am
===================================================================
--- dcop/dcopidl/Makefile.am	(Revision 606834)
+++ dcop/dcopidl/Makefile.am	(Arbeitskopie)
@@ -1,6 +1,5 @@
 
 INCLUDES = $(all_includes)
-AM_LDFLAGS   = $(all_libraries)
 AM_CPPFLAGS = -DYYDEBUG
 
 ####### Files
@@ -11,7 +10,7 @@
 noinst_HEADERS = yacc.cc.h
 
 dcopidl_LDADD = $(LIB_QT)
-dcopidl_LDFLAGS = $(KDE_RPATH)
+dcopidl_LDFLAGS = $(KDE_RPATH) $(all_libraries)
 
 ####### Build rules
 
