diff -Naur kdelibs-3.5.1/kdecore/kaboutdata.cpp kdelibs-3.5.1-klocale/kdecore/kaboutdata.cpp
--- kdelibs-3.5.1/kdecore/kaboutdata.cpp	2006-01-19 18:06:18.000000000 +0100
+++ kdelibs-3.5.1-klocale/kdecore/kaboutdata.cpp	2006-02-22 23:48:57.000000000 +0100
@@ -275,7 +275,8 @@
   delete[] d->mTranslatedProgramName;
   d->mTranslatedProgramName = 0;
   if( KGlobal::locale())
-      d->mTranslatedProgramName = qstrdup( i18n( mProgramName ).utf8());
+       // btw no idea I'm right with that but works  
+      d->mTranslatedProgramName = qstrdup( programName().utf8());
 }
 
 QImage
