diff -up kdelibs-4.0.2/kio/kio/kprotocolmanager.cpp.branding kdelibs-4.0.2/kio/kio/kprotocolmanager.cpp
--- kdelibs-4.0.2/kio/kio/kprotocolmanager.cpp.branding	2008-01-04 18:00:15.000000000 -0600
+++ kdelibs-4.0.2/kio/kio/kprotocolmanager.cpp	2008-03-07 14:49:46.000000000 -0600
@@ -73,7 +73,7 @@ KProtocolManagerPrivate::~KProtocolManag
 
 // DEFAULT USERAGENT STRING
 #define CFG_DEFAULT_UAGENT(X) \
-QString("Mozilla/5.0 (compatible; Konqueror/%1.%2%3) KHTML/%4.%5.%6 (like Gecko)") \
+QString("Mozilla/5.0 (compatible; Konqueror/%1.%2%3) KHTML/%4.%5.%6 (like Gecko) Frugalware Linux") \
         .arg(KDE_VERSION_MAJOR).arg(KDE_VERSION_MINOR).arg(X).arg(KDE_VERSION_MAJOR).arg(KDE_VERSION_MINOR).arg(KDE_VERSION_RELEASE)
 
 #define PRIVATE_DATA \

