diff -Naur kdelibs-3.5.1/kio/kio/kfilemetainfo.cpp kdelibs-3.5.1-from-svn/kio/kio/kfilemetainfo.cpp
--- kdelibs-3.5.1/kio/kio/kfilemetainfo.cpp	2005-10-10 17:05:42.000000000 +0200
+++ kdelibs-3.5.1-from-svn/kio/kio/kfilemetainfo.cpp	2006-02-17 17:20:27.000000000 +0100
@@ -17,7 +17,7 @@
  *  the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
  *  Boston, MA 02110-1301, USA.
  *
- *  $Id: kfilemetainfo.cpp 465272 2005-09-29 09:47:40Z mueller $
+ *  $Id$
  */
 
 #include <assert.h>
@@ -1541,6 +1541,11 @@
     m_itemDict.setAutoDelete( true );
 }
 
+KFileMimeTypeInfo::GroupInfo::~GroupInfo()
+{
+    delete m_variableItemInfo;
+} 
+
 const KFileMimeTypeInfo::ItemInfo * KFileMimeTypeInfo::GroupInfo::itemInfo( const QString& key ) const
 {
     ItemInfo* item = m_itemDict.find( key );
diff -Naur kdelibs-3.5.1/kio/kio/kfilemetainfo.h kdelibs-3.5.1-from-svn/kio/kio/kfilemetainfo.h
--- kdelibs-3.5.1/kio/kio/kfilemetainfo.h	2005-11-08 23:38:43.000000000 +0100
+++ kdelibs-3.5.1-from-svn/kio/kio/kfilemetainfo.h	2006-02-17 17:20:26.000000000 +0100
@@ -219,6 +219,8 @@
             return m_variableItemInfo;
         }
 
+        /** @internal */
+        ~GroupInfo();
     private:
         /** @internal */
         GroupInfo( const QString& name, const QString& translatedName);
