--- wxPython-4.2.2/build.py	2024-09-12 01:27:32.000000000 +0200
+++ wxPython-4.2.2.new/build.py	2025-02-25 11:38:54.640381011 +0100
@@ -41,7 +41,7 @@
     from buildtools.backports.shutil_which import which
 
 
-from distutils.dep_util import newer, newer_group
+from setuptools.modified import newer, newer_group
 from buildtools.config  import Config, msg, opj, posixjoin, loadETG, etg2sip, findCmd, \
                                phoenixDir, wxDir, copyIfNewer, copyFile, \
                                macSetLoaderNames, \
