diff --git a/ext/wxWidgets/include/wx/datetime.h b/ext/wxWidgets/include/wx/datetime.h
index 6eb2f8c..8c3cf43 100644
--- a/ext/wxWidgets/include/wx/datetime.h
+++ b/ext/wxWidgets/include/wx/datetime.h
@@ -148,7 +148,7 @@ public:
         Local,

         // zones from GMT (= Greenwich Mean Time): they're guaranteed to be
-        // consequent numbers, so writing something like `GMT0 + offset' is
+        // consequent numbers, so writing something like `GMT0 + offset` is
         // safe if abs(offset) <= 12

         // underscore stands for minus
diff --git a/ext/wxWidgets/interface/wx/datetime.h b/ext/wxWidgets/interface/wx/datetime.h
index ae99947..4604b75 100644
--- a/ext/wxWidgets/interface/wx/datetime.h
+++ b/ext/wxWidgets/interface/wx/datetime.h
@@ -96,7 +96,7 @@ public:

         ///@{
         /// zones from GMT (= Greenwich Mean Time): they're guaranteed to be
-        /// consequent numbers, so writing something like `GMT0 + offset' is
+        /// consequent numbers, so writing something like `GMT0 + offset` is
         /// safe if abs(offset) <= 12

         // underscore stands for minus

