diff -Naur wxPython-src-2.8.7.1/wxPython/wxPython/_misc.py wxPython-src-2.8.7.1-typo-fiX/wxPython/wxPython/_misc.py
--- wxPython-src-2.8.7.1/wxPython/wxPython/_misc.py	2007-07-19 00:33:59.000000000 +0200
+++ wxPython-src-2.8.7.1-typo-fiX/wxPython/wxPython/_misc.py	2007-12-05 16:18:48.000000000 +0100
@@ -453,7 +453,7 @@
 wxDateTime_GetCurrentMonth = wx._misc.DateTime_GetCurrentMonth
 wxDateTime_IsLeapYear = wx._misc.DateTime_IsLeapYear
 wxDateTime_GetCentury = wx._misc.DateTime_GetCentury
-wxDateTime_GetNumberOfDaysinYear = wx._misc.DateTime_GetNumberOfDaysinYear
+wxDateTime_GetNumberOfDaysinYear = wx._misc.DateTime_GetNumberOfDaysInYear
 wxDateTime_GetNumberOfDaysInMonth = wx._misc.DateTime_GetNumberOfDaysInMonth
 wxDateTime_GetMonthName = wx._misc.DateTime_GetMonthName
 wxDateTime_GetWeekDayName = wx._misc.DateTime_GetWeekDayName
