diff -Naur medit-0.9.3.orig/medit/Makefile.am medit-0.9.3/medit/Makefile.am
--- medit-0.9.3.orig/medit/Makefile.am	2008-05-03 11:49:16.000000000 +0200
+++ medit-0.9.3/medit/Makefile.am	2008-05-03 11:50:25.000000000 +0200
@@ -58,12 +58,7 @@
 
 if MOO_ENABLE_GENERATED_FILES
 install-data-hook:
-	if echo "Updating icon cache" && $(update_icon_cache); then		\
-		echo "Done.";							\
-	else									\
-		echo "*** GTK icon cache not updated. After install, run this:";\
-		echo $(update_icon_cache);					\
-	fi
+	echo "Update icon cache skipped :p"
 uninstall-hook:
 	echo "Updating icon cache" && $(update_icon_cache) && echo "Done."
 endif
diff -Naur medit-0.9.3.orig/medit/Makefile.in medit-0.9.3/medit/Makefile.in
--- medit-0.9.3.orig/medit/Makefile.in	2008-05-03 11:49:16.000000000 +0200
+++ medit-0.9.3/medit/Makefile.in	2008-05-03 11:51:11.000000000 +0200
@@ -725,12 +725,7 @@
 @MOO_INTLTOOL_DESKTOP_RULE@
 
 @MOO_ENABLE_GENERATED_FILES_TRUE@install-data-hook:
-@MOO_ENABLE_GENERATED_FILES_TRUE@	if echo "Updating icon cache" && $(update_icon_cache); then		\
-@MOO_ENABLE_GENERATED_FILES_TRUE@		echo "Done.";							\
-@MOO_ENABLE_GENERATED_FILES_TRUE@	else									\
-@MOO_ENABLE_GENERATED_FILES_TRUE@		echo "*** GTK icon cache not updated. After install, run this:";\
-@MOO_ENABLE_GENERATED_FILES_TRUE@		echo $(update_icon_cache);					\
-@MOO_ENABLE_GENERATED_FILES_TRUE@	fi
+@MOO_ENABLE_GENERATED_FILES_TRUE@	echo "update icon cache skipped :p"
 @MOO_ENABLE_GENERATED_FILES_TRUE@uninstall-hook:
 @MOO_ENABLE_GENERATED_FILES_TRUE@	echo "Updating icon cache" && $(update_icon_cache) && echo "Done."
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
