diff -Naur medit-0.8.6.orig/medit/Makefile.am medit-0.8.6/medit/Makefile.am
--- medit-0.8.6.orig/medit/Makefile.am	2007-06-18 14:11:05.000000000 +0200
+++ medit-0.8.6/medit/Makefile.am	2007-06-18 14:12:07.000000000 +0200
@@ -59,11 +59,6 @@
 update_icon_cache = gtk-update-icon-cache -f -t $(DESTDIR)$(hicolor)
 
 install-data-hook:
-	if echo "Updating icon cache" && $(update_icon_cache); then		\
-		echo "Done.";							\
-	else									\
-		echo "*** GTK icon cache not updated. After install, run this:";\
-		echo $(update_icon_cache);					\
-	fi
+	echo "update icon cache skipped :P"
 uninstall-hook:
 	echo "Updating icon cache" && $(update_icon_cache) && echo "Done."
diff -Naur medit-0.8.6.orig/medit/Makefile.in medit-0.8.6/medit/Makefile.in
--- medit-0.8.6.orig/medit/Makefile.in	2007-06-18 14:11:05.000000000 +0200
+++ medit-0.8.6/medit/Makefile.in	2007-06-18 14:12:38.000000000 +0200
@@ -853,12 +853,7 @@
 @INTLTOOL_DESKTOP_RULE@
 
 install-data-hook:
-	if echo "Updating icon cache" && $(update_icon_cache); then		\
-		echo "Done.";							\
-	else									\
-		echo "*** GTK icon cache not updated. After install, run this:";\
-		echo $(update_icon_cache);					\
-	fi
+	echo "update icon cache skipped :P"
 uninstall-hook:
 	echo "Updating icon cache" && $(update_icon_cache) && echo "Done."
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
