--- distributive-0.6.2.1/distributive.cabal	2001-09-09 03:46:40.000000000 +0200
+++ distributive-0.6.2.1.new/distributive.cabal	2023-03-16 16:16:56.049133949 +0100
@@ -56,7 +56,7 @@
   build-depends:
     base                >= 4   && < 5,
     base-orphans        >= 0.5.2 && < 1,
-    transformers        >= 0.3 && < 0.6
+    transformers        >= 0.3 && < 0.7
 
   hs-source-dirs:  src
   exposed-modules:
