--- sg3_utils-1.20/llseek.c.llseek	2006-06-07 17:00:41.000000000 +0200
+++ sg3_utils-1.20/llseek.c	2006-06-07 17:00:53.000000000 +0200
@@ -41,14 +41,6 @@
 static int _llseek (unsigned int, unsigned long,
                    unsigned long, llse_loff_t *, unsigned int);
 
-#ifdef __NR__llseek
-
-static _syscall5(int,_llseek,unsigned int,fd,unsigned long,offset_high,
-                 unsigned long, offset_low,llse_loff_t *,result,
-                 unsigned int, origin)
-
-#else
-
 /* no __NR__llseek on compilation machine - might give it explicitly */
 static int _llseek (unsigned int fd, unsigned long oh,
                     unsigned long ol, llse_loff_t *result,
@@ -57,8 +49,6 @@
         return -1;
 }
 
-#endif
-
 static llse_loff_t my_llseek (unsigned int fd, llse_loff_t offset,
                 unsigned int origin)
 {
