diff --git plugins/Makefile.am plugins/Makefile.am
index e4f6e69..3a9ea03 100644
--- plugins/Makefile.am
+++ plugins/Makefile.am
@@ -14,7 +14,6 @@ SUBDIRS =						\
 if ENABLE_PYTHON
 SUBDIRS +=						\
 	artsearch					\
-	context						\
 	im-status					\
 	lyrics						\
 	magnatune					\
@@ -24,8 +23,12 @@ SUBDIRS +=						\
 	sendto						\
 	rb
 
+if WITH_WEBKIT
+SUBDIRS += context
 endif
 
+endif # ENABLE_PYTHON
+
 if WITH_LIRC
 SUBDIRS += lirc
 endif
-- 
1.7.12.1
