diff -ur a/thinkpad_ec.c b/thinkpad_ec.c
--- a/thinkpad_ec.c	2011-07-28 12:06:30.000000000 -0500
+++ b/thinkpad_ec.c	2014-05-03 18:48:44.747289384 -0500
@@ -97,7 +97,7 @@
 /* Kludge in case the ACPI DSDT reserves the ports we need. */
 static int force_io;    /* Willing to do IO to ports we couldn't reserve? */
 static int reserved_io; /* Successfully reserved the ports? */
-module_param_named(force_io, force_io, bool, 0600);
+module_param_named(force_io, force_io, int, 0600);
 MODULE_PARM_DESC(force_io, "Force IO even if region already reserved (0=off, 1=on)");
 
 /**
