diff -Naurp mantisbt-1.2.1-orig/core/email_api.php mantisbt-1.2.1/core/email_api.php
--- mantisbt-1.2.1-orig/core/email_api.php	2010-04-23 20:28:34.000000000 +0200
+++ mantisbt-1.2.1/core/email_api.php	2010-04-25 07:45:15.000000000 +0200
@@ -1324,14 +1324,14 @@ function email_format_bug_message( $p_vi
 	# format history
 	if( array_key_exists( 'history', $p_visible_bug_data ) ) {
 		$t_message .= lang_get( 'bug_history' ) . " \n";
-		$t_message .= utf8_str_pad( lang_get( 'date_modified' ), 17 ) . utf8_str_pad( lang_get( 'username' ), 15 ) . utf8_str_pad( lang_get( 'field' ), 25 ) . utf8_str_pad( lang_get( 'change' ), 20 ) . " \n";
+		$t_message .= utf8_str_pad( lang_get( 'date_modified' ), 17 ) .' '. utf8_str_pad( lang_get( 'username' ), 15 ) .' '. utf8_str_pad( lang_get( 'field' ), 25 ) .' '. utf8_str_pad( lang_get( 'change' ), 20 ) . " \n";
 
 		$t_message .= $t_email_separator1 . " \n";
 
 		foreach( $p_visible_bug_data['history'] as $t_raw_history_item ) {
 			$t_localized_item = history_localize_item( $t_raw_history_item['field'], $t_raw_history_item['type'], $t_raw_history_item['old_value'], $t_raw_history_item['new_value'], false );
 
-			$t_message .= utf8_str_pad( date( $t_normal_date_format, $t_raw_history_item['date'] ), 17 ) . utf8_str_pad( $t_raw_history_item['username'], 15 ) . utf8_str_pad( $t_localized_item['note'], 25 ) . utf8_str_pad( $t_localized_item['change'], 20 ) . "\n";
+			$t_message .= utf8_str_pad( date( $t_normal_date_format, $t_raw_history_item['date'] ), 17 ) .' '. utf8_str_pad( $t_raw_history_item['username'], 15 ) .' '. utf8_str_pad( $t_localized_item['note'], 25 ) .' '. utf8_str_pad( $t_localized_item['change'], 20 ) . "\n";
 		}
 		$t_message .= $t_email_separator1 . " \n\n";
 	}
