diff -Naur mantis-1.0.6-orig/core/email_api.php mantis-1.0.6/core/email_api.php
--- mantis-1.0.6-orig/core/email_api.php	2005-08-04 22:57:50.000000000 +0200
+++ mantis-1.0.6/core/email_api.php	2007-03-30 20:45:23.000000000 +0200
@@ -988,9 +988,9 @@
 		# format history
 		if ( array_key_exists( 'history', $p_visible_bug_data ) ) {
 			$t_message .=	lang_get( 'bug_history' ) . " \n";
-			$t_message .=	str_pad( lang_get( 'date_modified' ), 16 ) .
-							str_pad( lang_get( 'username' ), 15 ) .
-							str_pad( lang_get( 'field' ), 25 ) .
+			$t_message .=	str_pad( lang_get( 'date_modified' ), 16 ) . ' ' .
+							str_pad( lang_get( 'username' ), 15 ) . ' ' .
+							str_pad( lang_get( 'field' ), 25 ) . ' ' .
 							str_pad( lang_get( 'change' ), 20 ). " \n";
 
 			$t_message .= $t_email_separator1 . " \n";
@@ -1001,9 +1001,9 @@
 															$t_raw_history_item['old_value'],
 															$t_raw_history_item['new_value'] );
 
-				$t_message .=	str_pad( date( $t_normal_date_format, $t_raw_history_item['date'] ), 16 ) .
-								str_pad( $t_raw_history_item['username'], 15 ) .
-								str_pad( $t_localized_item['note'], 25 ) .
+				$t_message .=	str_pad( date( $t_normal_date_format, $t_raw_history_item['date'] ), 16 ) . ' ' .
+								str_pad( $t_raw_history_item['username'], 15 ) . ' ' .
+								str_pad( $t_localized_item['note'], 25 ) . ' ' .
 								str_pad( $t_localized_item['change'], 20 ) . "\n";
 			}
 			$t_message .= $t_email_separator1 . " \n\n";
