From 6bf7e0e1c122825adab37f2d9339029e5a076fa5 Mon Sep 17 00:00:00 2001
From: Boarders <callan.mcgill@gmail.com>
Date: Tue, 6 Oct 2020 17:50:45 -0400
Subject: [PATCH] primitive.cabal

---
 primitive.cabal | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/primitive.cabal b/primitive.cabal
index 0decb68..4b89c3f 100644
--- a/primitive.cabal
+++ b/primitive.cabal
@@ -53,7 +53,7 @@ Library
         Data.Primitive.Internal.Compat
         Data.Primitive.Internal.Operations
 
-  Build-Depends: base >= 4.5 && < 4.15
+  Build-Depends: base >= 4.5 && < 4.16
                , deepseq >= 1.1 && < 1.5
                , transformers >= 0.2 && < 0.6
   if !impl(ghc >= 8.0)
