--- internal/dcraw_common.cpp~	2018-01-18 23:57:03.000000000 -0600
+++ internal/dcraw_common.cpp	2018-01-19 11:50:19.746347270 -0600
@@ -6827,7 +6827,7 @@
 		if (table_buf[iLensData+9] &&
 			(fabs(imgdata.lens.makernotes.CurFocal) < 0.1f))
 		  imgdata.lens.makernotes.CurFocal =
-			10*(table_buf[iLensData+9]>>2) * powf64(4, (table_buf[iLensData+9] & 0x03)-2);
+			10*(table_buf[iLensData+9]>>2) * powf64((float)4, (table_buf[iLensData+9] & 0x03)-2);
 		if (table_buf[iLensData+10] & 0xf0)
 		  imgdata.lens.makernotes.MaxAp4CurFocal =
 			powf64(2.0f, (float)((table_buf[iLensData+10] & 0xf0) >>4)/4.0f);
