diff -p -up ./configure.org ./configure
--- ./configure.org     2014-06-04 21:00:03.027735142 +0200
+++ ./configure 2014-06-04 21:04:46.428267513 +0200
@@ -5650,7 +5650,8 @@ fi
 LIBPTHREAD=$LIBPTHREAD
 
 # Checks for header files.
-for ac_header in limits.h stdlib.h string.h
+for ac_header in limits.h stdlib.h string.h inttypes.h stdint.h memory.h \
+			strings.h sys/stat.h sys_time.h sys_types.h termios.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_mongrel "$LINENO" "$ac_header" "$as_ac_Header" "$ac_includes_default"
