diff -Naur sysvinit-2.85.orig/src/init.c sysvinit-2.85/src/init.c
--- sysvinit-2.85.orig/src/init.c	2003-04-15 13:16:41.000000000 +0200
+++ sysvinit-2.85/src/init.c	2004-07-28 21:24:03.000000000 +0200
@@ -2333,7 +2333,7 @@
   	/*
 	 *	Say hello to the world
 	 */
-  	log(L_CO, bootmsg, "booting");
+  	/*log(L_CO, bootmsg, "booting");*/
 
   	/*
 	 *	See if we have to start an emergency shell.
diff -Naur sysvinit-2.85.orig/src/paths.h sysvinit-2.85/src/paths.h
--- sysvinit-2.85.orig/src/paths.h	2003-04-14 13:37:01.000000000 +0200
+++ sysvinit-2.85/src/paths.h	2004-07-28 21:19:16.000000000 +0200
@@ -20,8 +20,8 @@
 #define INITTAB		"/etc/inittab"		/* Location of inittab */
 #define INIT		"/sbin/init"		/* Location of init itself. */
 #define NOLOGIN		"/etc/nologin"		/* Stop user logging in. */
-#define FASTBOOT	"/fastboot"		/* Enable fast boot. */
-#define FORCEFSCK	"/forcefsck"		/* Force fsck on boot */
+#define FASTBOOT	"/etc/fastboot"		/* Enable fast boot. */
+#define FORCEFSCK	"/etc/forcefsck"		/* Force fsck on boot */
 #define SDPID		"/var/run/shutdown.pid"	/* PID of shutdown program */
 #define SHELL		"/bin/sh"		/* Default shell */
 #define INITSCRIPT	"/etc/initscript"	/* Initscript. */
