diff -Naur ivman-0.6.13.orig/src/manager.c ivman-0.6.13/src/manager.c
--- ivman-0.6.13.orig/src/manager.c	2006-12-02 01:30:15.000000000 +0100
+++ ivman-0.6.13/src/manager.c	2007-01-03 09:31:40.000000000 +0100
@@ -1042,19 +1042,24 @@
                                               "volume.mount_point",
                                               NULL);
 
-        while (( !mount_point || (strlen(mount_point) < 2) )
+        while (( !mount_point || (strlen(mount_point) < 1) )
                && (i < 5))
         {
             sleep(1);
             mount_point = (char *)
                 libhal_device_get_property_string(hal_ctx, udi,
-                                                  " volume.mount_point ",
+                                                  "volume.mount_point ",
                                                   NULL);
+            if( dbus_error_is_set(&dbus_error) )
+            {
+               ivm_check_dbus_error(&dbus_error);
+                break;
+            }
             i++;
         }
 
 
-        if ( !mount_point || strlen(mount_point) < 2)
+        if ( !mount_point || strlen(mount_point) < 1)
         {
             DEBUG(_("Couldn't get mount point of device %s after "
                     "5 seconds\n"), device);
