diff -Naurp dbus-1.0.0/dbus/dbus-threads.h dbus-1.0.0-p/dbus/dbus-threads.h
--- dbus-1.0.0/dbus/dbus-threads.h	2006-10-27 04:17:42.000000000 +0200
+++ dbus-1.0.0-p/dbus/dbus-threads.h	2006-11-14 20:24:15.000000000 +0100
@@ -134,7 +134,7 @@ typedef enum 
   DBUS_THREAD_FUNCTIONS_RECURSIVE_MUTEX_FREE_MASK   = 1 << 11,
   DBUS_THREAD_FUNCTIONS_RECURSIVE_MUTEX_LOCK_MASK   = 1 << 12,
   DBUS_THREAD_FUNCTIONS_RECURSIVE_MUTEX_UNLOCK_MASK = 1 << 13,
-  DBUS_THREAD_FUNCTIONS_ALL_MASK     = (1 << 13) - 1
+  DBUS_THREAD_FUNCTIONS_ALL_MASK     = (1 << 14) - 1
 } DBusThreadFunctionsMask;
 
 /**
diff -Naurp dbus-1.0.0/test/name-test/test-threads-init.c dbus-1.0.0-p/test/name-test/test-threads-init.c
--- dbus-1.0.0/test/name-test/test-threads-init.c	2006-10-01 17:36:18.000000000 +0200
+++ dbus-1.0.0-p/test/name-test/test-threads-init.c	2006-11-14 20:24:49.000000000 +0100
@@ -140,7 +140,7 @@ main (int argc, char *argv[])
   check_condvar_lock (dispatch_cond1, dispatch_cond2, TRUE);
   check_condvar_lock (io_path_cond1, io_path_cond2, TRUE);
 
-  _dbus_threads_init_debug ();
+  dbus_threads_init_default ();
 
   _dbus_connection_test_get_locks (conn, &mutex1,
                                          &dispatch_mutex1,
