--- dbus-0.94/tools/dbus-launch.h.dont-crash	2006-11-12 23:44:51.000000000 -0500
+++ dbus-0.94/tools/dbus-launch.h	2006-11-12 23:45:14.000000000 -0500
@@ -49,7 +49,7 @@
 
 #ifdef DBUS_BUILD_X11
 /* defined in dbus-launch-x11.c */
-int x11_init (void);
+int x11_init (int should_init_x_atoms);
 int x11_get_address (char **paddress, pid_t *pid, long *wid);
 int x11_save_address (char *address, pid_t pid, long *wid);
 void x11_handle_event (void);
--- dbus-0.94/tools/dbus-launch-x11.c.dont-crash	2006-11-12 23:44:38.000000000 -0500
+++ dbus-0.94/tools/dbus-launch-x11.c	2006-11-12 23:47:25.000000000 -0500
@@ -247,6 +247,12 @@
 
   machine = get_machine_uuid ();
 
+  if (machine == NULL)
+    {
+      verbose ("Could not determine machine identity; aborting X11 integration.\n");
+      return FALSE;
+    }
+
   atom_name = malloc (strlen (machine) + strlen (user_name) + 2 +
                       MAX (strlen (selection_prefix),
                            MAX (strlen (address_prefix),
@@ -441,9 +447,10 @@
 }
 
 int
-x11_init (void)
+x11_init (int should_init_x_atoms)
 {
-  return open_x11 () != NULL && init_x_atoms (xdisplay);
+  return (open_x11 () != NULL) && 
+	 (!should_init_x_atoms || init_x_atoms (xdisplay));
 }
 
 void
--- dbus-0.94/tools/dbus-launch.c.dont-crash	2006-11-12 23:45:00.000000000 -0500
+++ dbus-0.94/tools/dbus-launch.c	2006-11-12 23:47:09.000000000 -0500
@@ -381,7 +381,7 @@
   sigaction (SIGTERM,  &act, NULL);
   
 #ifdef DBUS_BUILD_X11
-  x11_init();
+  x11_init(FALSE);
   if (xdisplay != NULL)
     {
       x_fd = ConnectionNumber (xdisplay);
@@ -790,7 +790,7 @@
 	  exit_with_session = TRUE;
 	}
 
-      if (!x11_init ())
+      if (!x11_init (TRUE))
 	{
 	  fprintf (stderr, "Autolaunch error: X11 initialization failed.\n");
 	  exit (1);

