diff -Naur dcron.orig/defs.h dcron/defs.h
--- dcron.orig/defs.h	2002-12-17 21:06:51.000000000 +0100
+++ dcron/defs.h	2004-08-18 12:23:41.195751048 +0200
@@ -35,7 +35,7 @@
 #define OPEN_MAX	256
 #endif
 #ifndef SENDMAIL
-#define SENDMAIL	"/usr/lib/sendmail"
+#define SENDMAIL	"/usr/sbin/sendmail"
 #endif
 #ifndef SENDMAIL_ARGS
 #define SENDMAIL_ARGS	"-t", "-oem", "-i"
diff -Naur dcron.orig/Makefile dcron/Makefile
--- dcron.orig/Makefile	2002-12-17 21:08:49.000000000 +0100
+++ dcron/Makefile	2004-08-18 12:23:41.193751352 +0200
@@ -1,6 +1,7 @@
 # Makefile for dillon's cron and crontab
 #
 
+DESTDIR =
 CC  = gcc
 CFLAGS = -O2 -Wall -Wstrict-prototypes
 LIB = 
@@ -31,18 +32,10 @@
 	rm -f *.o dcron.tgz $(PROTOS)
 
 install:
-	rm -f /usr/bin/crond /usr/bin/crontab
-	cp crond crontab /usr/bin
-	chown root.wheel /usr/bin/crond
-	chown root.cron /usr/bin/crontab
-	chmod 700 /usr/bin/crond
-	chmod 4750 /usr/bin/crontab
-	-mkdirs /usr/man/man1
-	-mkdirs /usr/man/man8
-	cp crontab.1 /usr/man/man1
-	cp crond.8 /usr/man/man8
-	chmod 444 /usr/man/man1/crontab.1
-	chmod 444 /usr/man/man8/crond.8
+	install -o root -g root -m 0755 -D crond $(DESTDIR)/usr/sbin/crond
+	install -o root -g root -m 4755 -D crontab $(DESTDIR)/usr/bin/crontab
+	install -o root -g root -m 0644 -D crontab.1 $(DESTDIR)/usr/man/man1/crontab.1
+	install -o root -g root -m 0644 -D  crond.8 $(DESTDIR)/usr/man/man8/crond.8
 
 # dillon-specific
 #
