--- rc.submount-0.1	2004-09-01 18:20:22.000000000 +0200
+++ rc.submount	2004-09-02 10:49:54.006337576 +0200
@@ -116,7 +116,7 @@
 		 umount $i 2>/dev/null
 		 rm -rf $i)
 	done
-	cat /etc/fstab |grep -v ^/dev/floppy>/etc/fstab.new
+	cat /etc/fstab |grep -v '^/dev/floppy '>/etc/fstab.new
 	mv -f /etc/fstab.new /etc/fstab
 	if [ -e /dev/floppy ]; then
 		(cd /etc; echo "/dev/floppy /mnt/floppy auto user,noauto$mountopts 0 0" >>fstab)
@@ -125,7 +125,7 @@
 
 	for i in ${devices[@]}
 	do
-		cat /etc/fstab |grep -v ^/dev/$i>/etc/fstab.new
+		cat /etc/fstab |grep -v "^/dev/$i ">/etc/fstab.new
 		mv -f /etc/fstab.new /etc/fstab
 		if grep -q $i $linkf; then
 			(cd /etc; echo "/dev/$i /mnt/$i auto ro,user,noauto$mountopts 0 0" >>fstab)
@@ -142,7 +142,7 @@
 		 umount $i 2>/dev/null
 		 rm -rf $i)
 	done
-	cat /etc/fstab |grep -v ^/dev/floppy>/etc/fstab.new
+	cat /etc/fstab |grep -v '^/dev/floppy '>/etc/fstab.new
 	mv -f /etc/fstab.new /etc/fstab
 	if [ -e /dev/floppy ]; then
 		(cd /etc; echo "/dev/floppy /mnt/floppy subfs fs=floppyfss$mountopts 0 0" >>fstab)
@@ -151,7 +151,7 @@
 
 	for i in ${devices[@]}
 	do
-		cat /etc/fstab |grep -v ^/dev/$i>/etc/fstab.new
+		cat /etc/fstab |grep -v "^/dev/$i ">/etc/fstab.new
 		mv -f /etc/fstab.new /etc/fstab
 		if grep -q $i $linkf; then
 			(cd /etc; echo "/dev/$i /mnt/$i subfs fs=auto,ro,user$mountopts 0 0" >>fstab)
