--- rc.submount-0.2	2004-10-26 22:03:50.718614600 +0200
+++ rc.submount	2004-10-26 22:04:31.000000000 +0200
@@ -4,7 +4,7 @@
 # rc.submount for Frugalware
 # distributed under GPL License
 
-# version 0.2
+# version 0.3
 
 # chkconfig: 2345 09 99
 # description:  Automatically mounts and unmounts removable media devices.
@@ -114,15 +114,15 @@
 if [ "$mode" == "dirs" ]; then
 	for i in floppy ${devices[@]}
 	do
-		(cd /mnt
+		(cd /media
 		 umount $i 2>/dev/null
 		 rm -rf $i)
 	done
 	cat /etc/fstab |grep -v '^/dev/floppy '>/etc/fstab.new
 	mv -f /etc/fstab.new /etc/fstab
 	if [ -e /dev/floppy ]; then
-		(cd /etc; echo "/dev/floppy /mnt/floppy auto user,noauto$mountopts 0 0" >>fstab)
-		mkdir /mnt/floppy
+		(cd /etc; echo "/dev/floppy /media/floppy auto user,noauto$mountopts 0 0" >>fstab)
+		mkdir -p /media/floppy
 	fi
 
 	for i in ${devices[@]}
@@ -130,8 +130,8 @@
 		cat /etc/fstab |grep -v "^/dev/$i ">/etc/fstab.new
 		mv -f /etc/fstab.new /etc/fstab
 		if grep -q $i $linkf; then
-			(cd /etc; echo "/dev/$i /mnt/$i auto ro,user,noauto$mountopts 0 0" >>fstab)
-			mkdir /mnt/$i
+			(cd /etc; echo "/dev/$i /media/$i auto ro,user,noauto$mountopts 0 0" >>fstab)
+			mkdir -p /media/$i
 		fi
 	done
 	mount -a 2>/dev/null
@@ -140,15 +140,15 @@
 if [ "$mode" == "auto" ]; then
 	for i in floppy ${devices[@]}
 	do
-		(cd /mnt
+		(cd /media
 		 umount $i 2>/dev/null
 		 rm -rf $i)
 	done
 	cat /etc/fstab |grep -v '^/dev/floppy '>/etc/fstab.new
 	mv -f /etc/fstab.new /etc/fstab
 	if [ -e /dev/floppy ]; then
-		(cd /etc; echo "/dev/floppy /mnt/floppy subfs fs=floppyfss$mountopts 0 0" >>fstab)
-		mkdir /mnt/floppy
+		(cd /etc; echo "/dev/floppy /media/floppy subfs fs=floppyfss$mountopts 0 0" >>fstab)
+		mkdir -p /media/floppy
 	fi
 
 	for i in ${devices[@]}
@@ -156,8 +156,8 @@
 		cat /etc/fstab |grep -v "^/dev/$i ">/etc/fstab.new
 		mv -f /etc/fstab.new /etc/fstab
 		if grep -q $i $linkf; then
-			(cd /etc; echo "/dev/$i /mnt/$i subfs fs=auto,ro,user$mountopts 0 0" >>fstab)
-			mkdir /mnt/$i
+			(cd /etc; echo "/dev/$i /media/$i subfs fs=auto,ro,user$mountopts 0 0" >>fstab)
+			mkdir -p /media/$i
 		fi
 	done
 	mount -a 2>/dev/null
