diff -Naur PyQt5_gpl-5.11.1/configure.py PyQt5_gpl-5.11.1-p/configure.py
--- PyQt5_gpl-5.11.1/configure.py	2018-06-23 22:22:46.000000000 +0200
+++ PyQt5_gpl-5.11.1-p/configure.py	2018-06-27 01:18:27.071932329 +0200
@@ -634,15 +634,6 @@
     def from_introspection(self, verbose, debug):
         """ Initialise the configuration by introspecting the system. """
 
-        # Check there is a private copy of the sip module already installed.
-        try:
-            from PyQt5 import sip
-        except ImportError:
-            error(
-                    "Unable to import PyQt5.sip.  Make sure you have "
-                    "configured SIP to create a private copy of the sip "
-                    "module.")
-
         # Get the details of the Python interpreter library.
         py_major = self.py_version >> 16
         py_minor = (self.py_version >> 8) & 0x0ff
@@ -2906,14 +2897,6 @@
     target_config is the target configuration.
     """
 
-    # Check there is a private copy of the sip module already installed.
-    try:
-        from PyQt5 import sip
-    except ImportError:
-        error(
-                "Unable to import PyQt5.sip.  Make sure you have configured "
-                "SIP to create a private copy of the sip module.")
-
     if target_config.sip is None:
         error(
                 "Make sure you have a working sip on your PATH or use the "
