--- PyQt5_gpl-5.11.2/configure.py	2018-06-29 13:48:34.000000000 +0200
+++ PyQt5_gpl-5.11.2.new/configure.py	2018-09-12 10:50:47.142853807 +0200
@@ -644,7 +644,7 @@
 
         # Check there is a private copy of the sip module already installed.
         try:
-            from PyQt5 import sip
+            pass
         except ImportError:
             error(
                     "Unable to import PyQt5.sip.  Make sure you have "
@@ -2916,7 +2916,7 @@
 
     # Check there is a private copy of the sip module already installed.
     try:
-        from PyQt5 import sip
+        pass
     except ImportError:
         error(
                 "Unable to import PyQt5.sip.  Make sure you have configured "
