diff -Naur asciidoc-8.2.6.orig/a2x asciidoc-8.2.6/a2x
--- asciidoc-8.2.6.orig/a2x	2007-11-25 23:59:55.000000000 +0100
+++ asciidoc-8.2.6/a2x	2008-05-03 23:28:36.000000000 +0200
@@ -502,9 +502,9 @@
 function to_pdf()
 {
     require "xsltproc"
-    require "fop.sh"
+    require "fop"
     local xsl xml fo pdf
-    xsl=$(conf_file docbook-xsl/fo.xsl)
+    xsl="/usr/share/xml/docbook/fo/docbook.xsl"
     if [ ! -r "$xsl" ]; then
         quit "file not found: $xsl"
     fi
@@ -512,9 +512,9 @@
     fo="$SRC_DIR/$SRC_NAME.fo"
     pdf="$DESTINATION_DIR/$SRC_NAME.pdf"
     to_docbook
-    execute_command_2 "xsltproc $XSLTPROC_OPTS --nonet \
+    execute_command_2 "xsltproc $XSLTPROC_OPTS --nonet --stringparam draft.mode no --stringparam sans.font.family DejaVuSans --stringparam title.font.family DejaVuSans --stringparam body.font.family DejaVuSerif --stringparam dingbat.font.family DejaVuSerif --stringparam monospace.font.family DejaVuSansMono \
         \"$xsl\" \"$xml\" >\"$fo\""
-    execute_command_2 "fop.sh \"$fo\" \"$pdf\""
+    execute_command_2 "fop -c /usr/share/fop/dejavu-ttf.xml -fo \"$fo\" -pdf \"$pdf\""
 }
 
 function to_odt()
