diff -Naur ode-0.7/Makefile.am ode-0.7-p/Makefile.am
--- ode-0.7/Makefile.am	2006-07-12 22:02:52.000000000 +0200
+++ ode-0.7-p/Makefile.am	2007-01-23 21:55:16.000000000 +0100
@@ -1,7 +1,7 @@
 LIBTOOL_DEPS = @LIBTOOL_DEPS@
 libtool: $(LIBTOOL_DEPS)
 	$(SHELL) ./config.status --recheck
-SUBDIRS =   include drawstuff ode 
+SUBDIRS =   include ode 
 
 bin_SCRIPTS = ode-config
 
diff -Naur ode-0.7/ode/Makefile.am ode-0.7-p/ode/Makefile.am
--- ode-0.7/ode/Makefile.am	2005-04-12 03:38:47.000000000 +0200
+++ ode-0.7-p/ode/Makefile.am	2007-01-23 21:55:31.000000000 +0100
@@ -1 +1 @@
-SUBDIRS = src test
\ Kein Zeilenumbruch am Dateiende.
+SUBDIRS = src
