From 6d6aeb7d11057b5df774f1425bb49837f13c80c2 Mon Sep 17 00:00:00 2001
From: Tomas Chvatal <tchvatal@suse.cz>
Date: Wed, 24 Apr 2013 11:16:23 -0400
Subject: [PATCH] Better to use AC_CONFIG_HEADERS here.

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 505c9dc..de6aaeb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -22,7 +22,7 @@ m4_define([orcus_api_version], [orcus_major_version.orcus_minor_api_version])
 # =============
 AC_INIT([liborcus],[orcus_version])
 AC_CONFIG_MACRO_DIR([m4])
-AM_CONFIG_HEADER([config.h])
+AC_CONFIG_HEADERS([config.h])
 AM_INIT_AUTOMAKE([1.11 foreign dist-bzip2 dist-xz])
 AM_SILENT_RULES([yes])
 AC_LANG([C++])
-- 
1.8.4

