--- dietlibc-0.24/include/rpc/types.h.rpc-types	2003-08-19 18:46:04.000000000 +0200
+++ dietlibc-0.24/include/rpc/types.h	2004-02-09 15:53:29.143821624 +0100
@@ -63,6 +63,16 @@ typedef unsigned long rpcport_t;
 #ifndef makedev /* ie, we haven't already included it */
 #include <sys/types.h>
 #endif
+
+# ifndef __dietlibc_u_char_defined
+typedef char* caddr_t;
+typedef uint32_t u_long;
+typedef unsigned char u_char;
+typedef unsigned short u_short;
+typedef unsigned int u_int;
+# define __dietlibc_u_char_defined
+#endif
+
 #include <sys/time.h>
 #include <sys/param.h>
 
--- dietlibc-0.24/include/sys/types.h.rpc-types	2003-11-20 12:06:10.000000000 +0100
+++ dietlibc-0.24/include/sys/types.h	2004-02-09 15:56:36.101399752 +0100
@@ -116,14 +116,6 @@ typedef unsigned short sa_family_t;
 
 #ifdef _BSD_SOURCE
 /* die, BSD, die! */
-typedef char* caddr_t __attribute_dontuse__;
-typedef unsigned char u_char __attribute_dontuse__;
-typedef unsigned short u_short __attribute_dontuse__;
-typedef unsigned int u_int __attribute_dontuse__;
-typedef unsigned char uchar __attribute_dontuse__;
-typedef unsigned short ushort __attribute_dontuse__;
-typedef unsigned int uint __attribute_dontuse__;
-typedef uint32_t u_long __attribute_dontuse__;
 typedef uint32_t n_long __attribute_dontuse__;
 typedef uint32_t n_time __attribute_dontuse__;
 typedef uint16_t n_short __attribute_dontuse__;
@@ -131,6 +123,17 @@ typedef uint32_t ulong __attribute_dontu
 typedef uint32_t nlong __attribute_dontuse__;
 typedef uint32_t ntime __attribute_dontuse__;
 typedef uint16_t nshort __attribute_dontuse__;
+#ifndef __dietlibc_u_char_defined
+typedef char* caddr_t __attribute_dontuse__;
+typedef unsigned char u_char __attribute_dontuse__;
+typedef unsigned short u_short __attribute_dontuse__;
+typedef uint32_t u_long __attribute_dontuse__;
+typedef unsigned int u_int __attribute_dontuse__;
+#define __dietlibc_u_char_defined
+#endif
+typedef unsigned char uchar __attribute_dontuse__;
+typedef unsigned short ushort __attribute_dontuse__;
+typedef unsigned int uint __attribute_dontuse__;
 
 /* never heard of these two, but dump uses them */
 typedef int64_t quad_t __attribute_dontuse__;
