--- xorgproto-2021.3/meson.build	2021-02-25 02:26:24.000000000 +0100
+++ xorgproto-2021.3.new/meson.build	2021-03-06 22:00:11.845963885 +0100
@@ -62,7 +62,7 @@
     configure_file(
         input : pc + '.pc.in',
         output : pc + '.pc',
-        install_dir : get_option('datadir') + '/pkgconfig',
+        install_dir : get_option('libdir') + '/pkgconfig',
         configuration : pc_data,
     )
 endforeach
