diff -Naur xorgproto-2018.3/meson.build xorgproto-2018.3-p/meson.build
--- xorgproto-2018.3/meson.build	2018-02-21 00:54:45.000000000 +0100
+++ xorgproto-2018.3-p/meson.build	2018-02-22 12:39:57.939562267 +0100
@@ -68,7 +68,7 @@
         filebase : pc[0],
         description : pc[0] + ' headers',
         version : pc[1],
-        install_dir : get_option('datadir') + '/pkgconfig',
+        install_dir : get_option('libdir') + '/pkgconfig',
         subdirs : subdir,
     )
 endforeach
@@ -88,7 +88,7 @@
             filebase : pc[0],
             description : pc[0] + ' headers',
             version : pc[1],
-            install_dir : get_option('datadir') + '/pkgconfig'
+            install_dir : get_option('libdir') + '/pkgconfig'
         )
     endforeach
 endif
