diff -aur plee-the-bear-0.4.1/bear-engine/core/src/engine/code/compiled_file.cpp plee-the-bear-0.4.1-2/bear-engine/core/src/engine/code/compiled_file.cpp
--- plee-the-bear-0.4.1/bear-engine/core/src/engine/code/compiled_file.cpp	2009-06-17 17:25:43.000000000 -0500
+++ plee-the-bear-0.4.1-2/bear-engine/core/src/engine/code/compiled_file.cpp	2010-05-15 01:59:36.000000000 -0500
@@ -50,7 +50,7 @@
  * \brief Read a string from the file.
  * \param s The string to read.
  */
-bear::engine::compiled_file::compiled_file&
+bear::engine::compiled_file&
 bear::engine::compiled_file::operator>>( std::string& s )
 {
   if (m_text)
@@ -66,7 +66,7 @@
  * \brief Read an long from the file.
  * \param i The long to read.
  */
-bear::engine::compiled_file::compiled_file&
+bear::engine::compiled_file&
 bear::engine::compiled_file::operator>>( long& i )
 {
   if (m_text)
@@ -82,7 +82,7 @@
  * \brief Read an unsigned long from the file.
  * \param i The long to read.
  */
-bear::engine::compiled_file::compiled_file&
+bear::engine::compiled_file&
 bear::engine::compiled_file::operator>>( unsigned long& i )
 {
   if (m_text)
@@ -98,7 +98,7 @@
  * \brief Read an integer from the file.
  * \param i The integer to read.
  */
-bear::engine::compiled_file::compiled_file&
+bear::engine::compiled_file&
 bear::engine::compiled_file::operator>>( int& i )
 {
   if (m_text)
@@ -114,7 +114,7 @@
  * \brief Read an unsigned integer from the file.
  * \param i The integer to read.
  */
-bear::engine::compiled_file::compiled_file&
+bear::engine::compiled_file&
 bear::engine::compiled_file::operator>>( unsigned int& i )
 {
   if (m_text)
@@ -130,7 +130,7 @@
  * \brief Read a real from the file.
  * \param r The value to read.
  */
-bear::engine::compiled_file::compiled_file&
+bear::engine::compiled_file&
 bear::engine::compiled_file::operator>>( double& r )
 {
   if (m_text)
@@ -146,7 +146,7 @@
  * \brief Read a boolean from the file.
  * \param b The value to read.
  */
-bear::engine::compiled_file::compiled_file&
+bear::engine::compiled_file&
 bear::engine::compiled_file::operator>>( bool& b )
 {
   if (m_text)
