diff -Naur kaffeine-2.0.9/CMakeLists.txt kaffeine-2.0.9-p/CMakeLists.txt
--- kaffeine-2.0.9/CMakeLists.txt	2017-03-08 18:54:14.000000000 +0100
+++ kaffeine-2.0.9-p/CMakeLists.txt	2017-05-15 01:21:21.628004491 +0200
@@ -22,6 +22,11 @@
   cmake_policy(SET CMP0063 NEW)
 endif(POLICY CMP0063)
 
+## workaround for ki18n 5.34.0++ changes
+if(POLICY CMP0002)
+  cmake_policy(SET CMP0002 OLD)
+endif(CMP0002)
+
 find_package(Qt5Core)
 find_package(ECM 1.0.0 REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH
