diff -Naur antlr-2.7.7/lib/cpp/antlr/CharScanner.hpp antlr-2.7.7-p/lib/cpp/antlr/CharScanner.hpp
--- antlr-2.7.7/lib/cpp/antlr/CharScanner.hpp	2006-11-01 22:37:17.000000000 +0100
+++ antlr-2.7.7-p/lib/cpp/antlr/CharScanner.hpp	2008-03-13 00:22:30.000000000 +0100
@@ -24,6 +24,7 @@
 # include <stdio.h>
 #endif
 
+#include <cstring>
 #include <antlr/TokenStream.hpp>
 #include <antlr/RecognitionException.hpp>
 #include <antlr/SemanticException.hpp>
