diff -Naur WindowMaker-0.92.0.orig/configure.ac WindowMaker-0.92.0/configure.ac
--- WindowMaker-0.92.0.orig/configure.ac	2005-07-04 14:23:20.000000000 +0200
+++ WindowMaker-0.92.0/configure.ac	2005-07-04 14:27:21.000000000 +0200
@@ -194,7 +194,19 @@
 
     # gcc-3.3 or newer complains about some of our stuff without this
     NOSTRICTALIASING="-fno-strict-aliasing"
+    
+    x86=0
+    changequote(,)dnl
+    case "$host_cpu" in
+	i[3456]86) x86=1
+	;;
+    esac
+    changequote([,])dnl
+    if test "$host_os" != "linux" -a "$host_os" != "linux-gnu" ; then
+	x86=0
+    fi
 
+    if test $x86 = 1 ; then
     AC_CACHE_CHECK(whether gcc supports x86 inline asm,
                    ac_cv_c_inline_asm,
                    [AC_TRY_LINK(,[{int x; asm volatile("movl %%eax, %%ebx\n"::
@@ -218,6 +230,7 @@
           mmx_support=yes
        fi
     fi
+    fi
 fi
 AC_SUBST(NOSTRICTALIASING)
 
