diff -Naur qownnotes-16.06.14/QOwnNotes.pro qownnotes-16.06.14-p/QOwnNotes.pro
--- qownnotes-16.06.14/QOwnNotes.pro	2016-06-30 19:08:45.000000000 +0200
+++ qownnotes-16.06.14-p/QOwnNotes.pro	2016-07-01 00:57:09.393941027 +0200
@@ -8,6 +8,66 @@
 
 greaterThan(QT_MAJOR_VERSION, 4): QT += widgets
 
+
+unix {
+
+  isEmpty(PREFIX) {
+    PREFIX = /usr
+  }
+
+  isEmpty(BINDIR) {
+    BINDIR = $$PREFIX/bin
+  }
+
+  isEmpty(DATADIR) {
+    DATADIR = $$PREFIX/share
+  }
+
+  INSTALLS += target desktop i18n icon16 icon24 icon32 icon48 icon64 icon96 icon128 icon256 icon512 svg
+
+  target.path = $$BINDIR
+  target.files += QOwnNotes
+
+  desktop.path = $$DATADIR/applications
+  desktop.files += QOwnNotes.desktop
+
+  i18n.path = $$DATADIR/QOwnNotes/languages 
+  i18n.files += languages/*.qm
+
+  ## all this not needed when the icons are right in source
+  ## eg: $size/apps/ and svg folder..  however .. -- crazy --
+
+  icon16.path = $$DATADIR/icons/hicolor/16x16/apps
+  icon16.files += images/icons/16x16/QOwnNotes.png
+
+  icon24.path = $$DATADIR/icons/hicolor/24x24/apps
+  icon24.files += images/icons/24x24/QOwnNotes.png
+
+  icon32.path = $$DATADIR/icons/hicolor/32x32/apps
+  icon32.files += images/icons/32x32/QOwnNotes.png
+
+  icon48.path = $$DATADIR/icons/hicolor/48x48/apps
+  icon48.files += images/icons/48x48/QOwnNotes.png
+
+  icon64.path = $$DATADIR/icons/hicolor/64x64/apps
+  icon64.files += images/icons/64x64/QOwnNotes.png
+
+  icon96.path = $$DATADIR/icons/hicolor/96x96/apps
+  icon96.files += images/icons/96x96/QOwnNotes.png
+
+  icon128.path = $$DATADIR/icons/hicolor/128x128/apps
+  icon128.files += images/icons/128x128/QOwnNotes.png
+
+  icon256.path = $$DATADIR/icons/hicolor/256x256/apps
+  icon256.files += images/icons/2566x256/QOwnNotes.png
+
+  icon512.path = $$DATADIR/icons/hicolor/512x512/apps
+  icon512.files += images/icons/512x512/QOwnNotes.png
+
+  svg.path = $$DATADIR/icons/hicolor/scalable/apps
+  svg.files += images/icons/scalable/QOwnNotes.svg
+}
+
 TARGET = QOwnNotes
 TEMPLATE = app
 ICON = QOwnNotes.icns
@@ -169,3 +229,5 @@
     DEFINES += QT_NO_DEBUG_OUTPUT
     message("Currently in RELEASE mode.")
 }
+
+
