diff -Naur ffmpeg/configure ffmpeg-p/configure
--- ffmpeg/configure	2007-04-22 14:34:15.000000000 +0200
+++ ffmpeg-p/configure	2007-04-22 23:06:08.000000000 +0200
@@ -1583,7 +1583,7 @@
 enabled amr_nb     && require libamrnb amrnb/interf_dec.h Speech_Decode_Frame_init -lamrnb -lm
 enabled amr_wb     && require libamrwb amrwb/dec_if.h D_IF_init -lamrwb -lm
 enabled liba52     && require liba52 a52dec/a52.h a52_init -la52
-enabled libdts     && require libdts dts.h dts_init -ldts -lm
+enabled libdts     && require libdts dts.h dca_init $(pkg-config --libs libdts) -lm
 enabled libgsm     && require libgsm gsm.h gsm_create -lgsm
 enabled libmp3lame && require LAME lame/lame.h lame_init -lmp3lame -lm
 enabled libtheora  && require libtheora theora/theora.h theora_info_init -ltheora -logg
