diff -ur a/configure b/configure
--- a/configure	2012-10-28 01:49:39.900620584 -0500
+++ b/configure	2012-10-28 01:50:17.131630614 -0500
@@ -3654,7 +3654,7 @@
 enabled libopencore_amrnb  && require libopencore_amrnb opencore-amrnb/interf_dec.h Decoder_Interface_init -lopencore-amrnb
 enabled libopencore_amrwb  && require libopencore_amrwb opencore-amrwb/dec_if.h D_IF_init -lopencore-amrwb
 enabled libopencv  && require_pkg_config opencv opencv/cxcore.h cvCreateImageHeader
-enabled libopenjpeg && require libopenjpeg openjpeg.h opj_version -lopenjpeg
+enabled libopenjpeg && require libopenjpeg openjpeg-1.5/openjpeg.h opj_version -lopenjpeg
 enabled libopus    && require_pkg_config opus opus_multistream.h opus_multistream_decoder_create
 enabled libpulse && require_pkg_config libpulse-simple pulse/simple.h pa_simple_new
 enabled librtmp    && require_pkg_config librtmp librtmp/rtmp.h RTMP_Socket
diff -ur a/libavcodec/libopenjpegdec.c b/libavcodec/libopenjpegdec.c
--- a/libavcodec/libopenjpegdec.c	2012-10-28 01:49:39.879620563 -0500
+++ b/libavcodec/libopenjpegdec.c	2012-10-28 02:01:57.172098456 -0500
@@ -25,7 +25,7 @@
  */
 
 #define  OPJ_STATIC
-#include <openjpeg.h>
+#include <openjpeg-1.5/openjpeg.h>
 
 #include "libavutil/common.h"
 #include "libavutil/intreadwrite.h"
diff -ur a/libavcodec/libopenjpegenc.c b/libavcodec/libopenjpegenc.c
--- a/libavcodec/libopenjpegenc.c	2012-10-28 01:49:39.872620557 -0500
+++ b/libavcodec/libopenjpegenc.c	2012-10-28 02:01:40.564093722 -0500
@@ -25,7 +25,7 @@
  */
 
 #define  OPJ_STATIC
-#include <openjpeg.h>
+#include <openjpeg-1.5/openjpeg.h>
 
 #include "libavutil/avassert.h"
 #include "libavutil/common.h"
