--- ffmpeg-7.1/libavcodec/libsvtav1.c	2024-09-30 01:31:48.000000000 +0200
+++ ffmpeg-7.1.new/libavcodec/libsvtav1.c	2025-02-21 10:58:11.148505296 +0100
@@ -435,7 +435,7 @@
 
     svt_enc->eos_flag = EOS_NOT_REACHED;
 
-    svt_ret = svt_av1_enc_init_handle(&svt_enc->svt_handle, svt_enc, &svt_enc->enc_params);
+    svt_ret = svt_av1_enc_init_handle(&svt_enc->svt_handle, &svt_enc->enc_params);
     if (svt_ret != EB_ErrorNone) {
         return svt_print_error(avctx, svt_ret, "Error initializing encoder handle");
     }
