diff -ur a/BaseTools/Conf/target.template b/BaseTools/Conf/target.template
--- a/BaseTools/Conf/target.template	2013-11-09 08:10:50.733723848 -0600
+++ b/BaseTools/Conf/target.template	2013-11-09 08:31:49.097677061 -0600
@@ -23,7 +23,7 @@
 #                                               build. This line is required if and only if the current
 #                                               working directory does not contain one or more description
 #                                               files.
-ACTIVE_PLATFORM       = Nt32Pkg/Nt32Pkg.dsc
+#ACTIVE_PLATFORM       = Nt32Pkg/Nt32Pkg.dsc
 
 #  TARGET                List       Optional    Zero or more of the following: DEBUG, RELEASE, NOOPT
 #                                               UserDefined; separated by a space character.
@@ -31,7 +31,7 @@
 #                                               valid targets specified in the platform description file 
 #                                               will attempt to be built. The following line will build 
 #                                               DEBUG platform target.
-TARGET                = DEBUG
+TARGET                = RELEASE
 
 #  TARGET_ARCH           List       Optional    What kind of architecture is the binary being target for.
 #                                               One, or more, of the following, IA32, IPF, X64, EBC, ARM
@@ -47,7 +47,7 @@
 #                                               If not specified, then all valid architectures specified
 #                                               in the platform file, for which tools are available, will be
 #                                               built.
-TARGET_ARCH           = IA32
+TARGET_ARCH           = X64
 
 #  TOOL_DEFINITION_FILE  Filename  Optional   Specify the name of the filename to use for specifying
 #                                             the tools to use for the build.  If not specified,
@@ -57,7 +57,7 @@
 #  TAGNAME               List      Optional   Specify the name(s) of the tools_def.txt TagName to use.
 #                                             If not specified, all applicable TagName tools will be
 #                                             used for the build.  The list uses space character separation.
-TOOL_CHAIN_TAG        = MYTOOLS
+TOOL_CHAIN_TAG        = GCC47
 
 # MAX_CONCURRENT_THREAD_NUMBER  NUMBER  Optional  The number of concurrent threads. Recommend to set this
 #                                                 value to one more than the number of your compurter
