diff -Naur irssistats-0.72.orig/Makefile irssistats-0.72/Makefile
--- irssistats-0.72.orig/Makefile	2005-02-07 14:57:05.000000000 +0100
+++ irssistats-0.72/Makefile	2006-03-31 00:12:14.000000000 +0200
@@ -3,9 +3,10 @@
 DAT = $(PRE)/share/irssistats/data
 DOC = $(PRE)/share/doc/irssistats
 MAN = $(PRE)/share/man/man1
+CFLAGS ?= -O2
 
 irssistats:irssistats.c
-	gcc -O2 -o irssistats irssistats.c
+	gcc $(CFLAGS) -o irssistats irssistats.c
 
 clean:
 	rm -f irssistats
