diff -Naur poedit-1.8.9-oss/src/tm/dump_legacy_tm.cpp poedit-1.8.9-oss-p/src/tm/dump_legacy_tm.cpp
--- poedit-1.8.9-oss/src/tm/dump_legacy_tm.cpp	2016-08-26 19:06:11.000000000 +0200
+++ poedit-1.8.9-oss-p/src/tm/dump_legacy_tm.cpp	2016-09-26 15:18:11.957811910 +0200
@@ -305,7 +305,7 @@
                             DB_RECOVER |
                             DB_CREATE;
 
-    std::unique_ptr<DbEnv> env(new DbEnv(0));
+    std::unique_ptr<DbEnv> env(new DbEnv((u_int32_t)0));
     env->open(path, flags, 0600);
 
     // This prevents the log from growing indefinitely
@@ -324,7 +324,7 @@
     env->close(0);
 
     // Clean up temporary environment files (unless they're still in use):
-    DbEnv env2(0);
+    DbEnv env2((u_int32_t)0);
     env2.remove(dbpath.c_str(), 0);
 
     env.reset();
