diff -Naur sink-0.9.0/CMakeLists.txt sink-0.9.0.new/CMakeLists.txt
--- sink-0.9.0/CMakeLists.txt	2021-02-20 15:12:47.000000000 +0100
+++ sink-0.9.0.new/CMakeLists.txt	2022-08-28 19:59:11.113888103 +0200
@@ -121,8 +121,6 @@
     add_definitions("-Werror -Wall -Weverything -Wno-unused-function -Wno-cast-align -Wno-used-but-marked-unused -Wno-shadow -Wno-weak-vtables -Wno-global-constructors -Wno-deprecated -Wno-weak-template-vtables -Wno-exit-time-destructors -Wno-covered-switch-default -Wno-shorten-64-to-32 -Wno-documentation -Wno-old-style-cast -Wno-extra-semi -Wno-unused-parameter -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-padded -Wno-missing-noreturn -Wno-missing-prototypes -Wno-documentation-unknown-command -Wno-sign-conversion -Wno-gnu-zero-variadic-macro-arguments -Wno-disabled-macro-expansion -Wno-vla-extension -Wno-vla -Wno-undefined-func-template -Wno-#warnings -Wno-unused-template -Wno-inconsistent-missing-destructor-override -Wno-zero-as-null-pointer-constant -Wno-unused-lambda-capture -Wno-switch-enum -Wno-redundant-parens -Wno-extra-semi-stmt -Wno-suggest-destructor-override")
 endif()
 
-set(CMAKE_CXX_STANDARD 20)
-set(CMAKE_CXX_STANDARD_REQUIRED ON)
 if (MSVC)
     # Workaround for older cmake versions
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /std:c++20")
@@ -152,7 +150,6 @@
 add_subdirectory(examples)
 
 # some tests
-add_subdirectory(tests)
 
 # cli
 add_subdirectory(sinksh)
diff -Naur sink-0.9.0/examples/caldavresource/CMakeLists.txt sink-0.9.0.new/examples/caldavresource/CMakeLists.txt
--- sink-0.9.0/examples/caldavresource/CMakeLists.txt	2021-02-20 15:12:47.000000000 +0100
+++ sink-0.9.0.new/examples/caldavresource/CMakeLists.txt	2022-08-28 19:59:00.801458519 +0200
@@ -12,5 +12,4 @@
 if (WIN32)
     message("Not building caldavresource resource tests on windows")
 else()
-    add_subdirectory(tests)
 endif()
diff -Naur sink-0.9.0/examples/carddavresource/CMakeLists.txt sink-0.9.0.new/examples/carddavresource/CMakeLists.txt
--- sink-0.9.0/examples/carddavresource/CMakeLists.txt	2021-02-20 15:12:47.000000000 +0100
+++ sink-0.9.0.new/examples/carddavresource/CMakeLists.txt	2022-08-28 19:59:00.801458519 +0200
@@ -12,5 +12,4 @@
 if (WIN32)
     message("Not building carddav resource tests on windows")
 else()
-    add_subdirectory(tests)
 endif()
diff -Naur sink-0.9.0/examples/imapresource/CMakeLists.txt sink-0.9.0.new/examples/imapresource/CMakeLists.txt
--- sink-0.9.0/examples/imapresource/CMakeLists.txt	2021-02-20 15:12:47.000000000 +0100
+++ sink-0.9.0.new/examples/imapresource/CMakeLists.txt	2022-08-28 19:59:00.801458519 +0200
@@ -15,5 +15,4 @@
 if (WIN32)
     message("Not building imap resource tests on windows")
 else()
-    add_subdirectory(tests)
 endif()
diff -Naur sink-0.9.0/examples/maildirresource/CMakeLists.txt sink-0.9.0.new/examples/maildirresource/CMakeLists.txt
--- sink-0.9.0/examples/maildirresource/CMakeLists.txt	2021-02-20 15:12:47.000000000 +0100
+++ sink-0.9.0.new/examples/maildirresource/CMakeLists.txt	2022-08-28 19:59:00.801458519 +0200
@@ -12,5 +12,4 @@
 if (WIN32)
     message("Not building maildir resource tests on windows")
 else()
-    add_subdirectory(tests)
 endif()
diff -Naur sink-0.9.0/examples/mailtransportresource/CMakeLists.txt sink-0.9.0.new/examples/mailtransportresource/CMakeLists.txt
--- sink-0.9.0/examples/mailtransportresource/CMakeLists.txt	2021-02-20 15:12:47.000000000 +0100
+++ sink-0.9.0.new/examples/mailtransportresource/CMakeLists.txt	2022-08-28 19:59:00.801458519 +0200
@@ -31,5 +31,4 @@
 if (WIN32)
     message("Not building mailtransport resource tests on windows")
 else()
-    add_subdirectory(tests)
 endif()
diff -Naur sink-0.9.0/mime/mimetreeparser/CMakeLists.txt sink-0.9.0.new/mime/mimetreeparser/CMakeLists.txt
--- sink-0.9.0/mime/mimetreeparser/CMakeLists.txt	2021-02-20 15:12:47.000000000 +0100
+++ sink-0.9.0.new/mime/mimetreeparser/CMakeLists.txt	2022-08-28 19:59:00.800458574 +0200
@@ -42,4 +42,3 @@
 )
 
 add_subdirectory(autotests)
-add_subdirectory(tests)
