diff -Naur sink-0.1.0/CMakeLists.txt sink-0.1.0-p/CMakeLists.txt
--- sink-0.1.0/CMakeLists.txt	2017-03-02 12:27:09.000000000 +0100
+++ sink-0.1.0-p/CMakeLists.txt	2017-03-14 00:08:38.684698633 +0100
@@ -101,10 +101,10 @@
 add_subdirectory(synchronizer)
 
 # example implementations
-add_subdirectory(examples)
+#add_subdirectory(examples)
 
 # some tests
-add_subdirectory(tests)
+#add_subdirectory(tests)
 
 # cli
 add_subdirectory(sinksh)
