diff -Naur syck-0.55/lib/emitter.c syck-0.55-p/lib/emitter.c
--- syck-0.55/lib/emitter.c	2005-05-19 08:07:42.000000000 +0200
+++ syck-0.55-p/lib/emitter.c	2007-10-29 18:01:44.000000000 +0100
@@ -378,7 +378,8 @@
 {
     SYMID oid;
     char *anchor_name = NULL;
-    int indent = 0, x = 0;
+    int indent = 0;
+    long x = 0;
     SyckLevel *lvl = syck_emitter_current_level( e );
     
     /* Add new level */
diff -Naur syck-0.55/lib/syck.h syck-0.55-p/lib/syck.h
--- syck-0.55/lib/syck.h	2005-04-13 08:27:54.000000000 +0200
+++ syck-0.55-p/lib/syck.h	2007-10-29 18:01:44.000000000 +0100
@@ -17,6 +17,8 @@
 #define YAML_DOMAIN     "yaml.org,2002"
 
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include <ctype.h>
 #ifdef HAVE_ST_H
 #include <st.h>
diff -Naur syck-0.55/lib/syck_st.c syck-0.55-p/lib/syck_st.c
--- syck-0.55/lib/syck_st.c	2003-03-04 18:10:11.000000000 +0100
+++ syck-0.55-p/lib/syck_st.c	2007-10-29 18:01:44.000000000 +0100
@@ -5,6 +5,7 @@
 #include "config.h"
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 #include "syck_st.h"
 
 #ifdef NT
@@ -52,7 +53,6 @@
     numhash,
 };
 
-extern int strcmp();
 static int strhash();
 static struct st_hash_type type_strhash = {
     strcmp,
