diff -ur dia-0.94-orig/objects/custom/shape_info.h dia-0.94/objects/custom/shape_info.h
--- dia-0.94-orig/objects/custom/shape_info.h	2005-04-15 18:45:52.000000000 -0400
+++ dia-0.94/objects/custom/shape_info.h	2005-04-15 18:36:55.000000000 -0400
@@ -166,7 +166,6 @@
 
 void shape_info_realise(ShapeInfo* info);
 void shape_info_print(ShapeInfo *info);
-void parse_path(ShapeInfo *info, const char *path_str, DiaSvgGraphicStyle *s);
 
 /*MC 11/03 handy g_new0 variant for struct with variable size */
 #define g_new0_ext(struct_type, ext_size)		\
diff -ur dia-0.94-orig/plug-ins/python/pydia-object.h dia-0.94/plug-ins/python/pydia-object.h
--- dia-0.94-orig/plug-ins/python/pydia-object.h	2005-04-15 18:45:13.000000000 -0400
+++ dia-0.94/plug-ins/python/pydia-object.h	2005-04-15 18:46:31.000000000 -0400
@@ -22,7 +22,7 @@
 
 #include <Python.h>
 
-#include "object.h"
+#include "lib/object.h"
 
 typedef struct {
     PyObject_HEAD
