Index: config/action.d/mail-buffered.conf
===================================================================
--- config/action.d/mail-buffered.conf	(revision 766)
+++ config/action.d/mail-buffered.conf	(revision 767)
@@ -81,7 +81,7 @@
 
 # Default temporary file
 #
-tmpfile = /tmp/fail2ban-mail.txt
+tmpfile = /var/run/fail2ban/tmp-mail.txt
 
 # Destination/Addressee of the mail
 #
Index: config/action.d/sendmail-buffered.conf
===================================================================
--- config/action.d/sendmail-buffered.conf	(revision 766)
+++ config/action.d/sendmail-buffered.conf	(revision 767)
@@ -101,5 +101,5 @@
 
 # Default temporary file
 #
-tmpfile = /tmp/fail2ban-mail.txt
+tmpfile = /var/run/fail2ban/tmp-mail.txt
 
Index: config/action.d/dshield.conf
===================================================================
--- config/action.d/dshield.conf	(revision 766)
+++ config/action.d/dshield.conf	(revision 767)
@@ -206,5 +206,5 @@
 # Notes.:  Base name of temporary files used for buffering
 # Values:  [ STRING ]  Default: /tmp/fail2ban-dshield
 #
-tmpfile = /tmp/fail2ban-dshield
+tmpfile = /var/run/fail2ban/tmp-dshield
 
Index: config/action.d/mynetwatchman.conf
===================================================================
--- config/action.d/mynetwatchman.conf	(revision 766)
+++ config/action.d/mynetwatchman.conf	(revision 767)
@@ -141,4 +141,4 @@
 # Notes.:  Base name of temporary files
 # Values:  [ STRING ]  Default: /tmp/fail2ban-mynetwatchman
 #
-tmpfile = /tmp/fail2ban-mynetwatchman
+tmpfile = /var/run/fail2ban/tmp-mynetwatchman
