diff -Nur gadmin-dhcpd-0.4.3.orig/desktop/gadmin-dhcpd.desktop gadmin-dhcpd-0.4.3/desktop/gadmin-dhcpd.desktop
--- gadmin-dhcpd-0.4.3.orig/desktop/gadmin-dhcpd.desktop	2008-02-19 18:56:49.000000000 +0100
+++ gadmin-dhcpd-0.4.3/desktop/gadmin-dhcpd.desktop	2008-05-02 20:24:57.000000000 +0200
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=GADMIN-DHCPD
+Name[hu]=GADMIN-DHCPD
 Name[de]=GADMIN-DHCPD
 Name[es]=GADMIN-DHCPD
 Name[fi]=GADMIN-DHCPD
@@ -15,8 +16,9 @@
 Name[zh_TW]=GADMIN-DHCPD
 GenericName=GADMIN-DHCPD
 Comment=DHCPD DHCP server administration
-Exec=gadmin-dhcpd
+Comment[hu]=Grafikus DHCPD adminisztrátor
+Exec=gksu /usr/sbin/gdhcpd
 Icon=gadmin-dhcpd.png
 Terminal=false
 Type=Application
-Categories=Application;Network;X-Red-Hat-ServerConfig;X-Red-Hat-Base;
+Categories=Application;Network;
