diff -Naur gsambad-0.1.6.orig/Makefile.in gsambad-0.1.6/Makefile.in
--- gsambad-0.1.6.orig/Makefile.in	2007-06-16 23:53:50.000000000 +0200
+++ gsambad-0.1.6/Makefile.in	2007-07-09 10:40:27.000000000 +0200
@@ -653,12 +653,12 @@
 	if test -d $(DESTDIR)$(bindir) && test ! -f $(DESTDIR)$(bindir)/gsambadpdf; then \
 	   cp scripts/gsambadpdf $(bindir)/; \
 	fi;
-	@echo Installing example logon script to static /home/netlogon.
-	if test ! -d /home/netlogon; then \
-	   mkdir -p /home/netlogon; \
+	@echo Installing example logon script to static $(DESTDIR)$(sysconfdir)/gsambad/netlogon.
+	if test ! -d $(DESTDIR)$(sysconfdir)/gsambad/netlogon; then \
+	   mkdir -p $(DESTDIR)$(sysconfdir)/gsambad/netlogon; \
 	fi;
-	if test -d /home/netlogon && test ! -f /home/netlogon/example.bat; then \
-	   cp scripts/example.bat /home/netlogon/; \
+	if test -d $(DESTDIR)$(sysconfdir)/gsambad/netlogon && test ! -f $(DESTDIR)$(sysconfdir)/gsambad/netlogon/example.bat; then \
+	   cp scripts/example.bat $(DESTDIR)$(sysconfdir)/gsambad/netlogon/; \
 	fi;
 	@echo ""
 	@echo "Start GSAMBAD by typing gsambad in a terminal"
