--- gtkglext-1.0.6/gdk/Makefile.am.old	2005-09-11 14:58:33.633063192 +0200
+++ gtkglext-1.0.6/gdk/Makefile.am	2005-09-11 14:59:56.481468320 +0200
@@ -57,7 +57,8 @@
 	$(gdkglext_win32_symbols)
 
 common_libadd = \
-	$(GDKGLEXT_DEP_LIBS)
+	$(GDKGLEXT_DEP_LIBS) \
+	-lpangox-1.0
 
 #
 # setup source file variables
--- gtkglext-1.0.6/gdk/Makefile.in.old	2005-09-11 14:58:28.765803128 +0200
+++ gtkglext-1.0.6/gdk/Makefile.in	2005-09-11 15:00:45.432026704 +0200
@@ -253,7 +253,8 @@
 
 
 common_libadd = \
-	$(GDKGLEXT_DEP_LIBS)
+	$(GDKGLEXT_DEP_LIBS) \
+	-lpangox-1.0
 
 
 
