diff -Naur kdebindings-4.3.4/csharp/plasma/CMakeLists.txt kdebindings-4.3.4-p/csharp/plasma/CMakeLists.txt
--- kdebindings-4.3.4/csharp/plasma/CMakeLists.txt	2009-07-21 17:22:11.000000000 +0200
+++ kdebindings-4.3.4-p/csharp/plasma/CMakeLists.txt	2010-01-15 18:33:41.000000000 +0100
@@ -4,7 +4,6 @@
 	FIND_PACKAGE(Qt4 REQUIRED)
 ENDIF (NOT QT4_FOUND)
 
-add_subdirectory( examples )
 
 SET(SRC_CPP src/plasma.cpp src/plasmahandlers.cpp)
 
