diff -Naur kdebindings-4.3.5/kalyptus/kalyptusCxxToSmoke.pm kdebindings-4.3.5-p/kalyptus/kalyptusCxxToSmoke.pm
--- kdebindings-4.3.5/kalyptus/kalyptusCxxToSmoke.pm	2010-01-21 21:47:43.000000000 +0100
+++ kdebindings-4.3.5-p/kalyptus/kalyptusCxxToSmoke.pm	2010-02-24 18:24:18.000000000 +0100
@@ -449,7 +449,7 @@
 			|| $className eq 'QTestKeyClicksEvent'
 			|| $className eq 'QTestKeyEvent'
 			|| $className eq 'QTestMouseEvent'
-
+			|| $className eq 'QWidget::setupUi'
 			|| $className eq 'std'
 			|| $className eq 'std::string'
 			|| $className eq 'std::wstring'
@@ -718,7 +718,7 @@
 			|| ($className eq 'Wt::Ext::ToolBar' and $name eq 'jsAfterPanelRendered')
 			|| ($className eq 'Wt::Ext::Widget' and $name eq 'bindEventHandler')
 			|| ($className eq 'Wt::Ext::Widget' and $name eq 'createMixed')
-
+			|| ($className eq 'QWidget' and $name eq 'setupUi')
 			|| ($className eq 'KTextEditor::Range' and $name =~ /operator/)
 			|| ($className eq 'KTextEditor::Range' and $name eq 'boundaryOnColumn')
 			|| ($className eq 'KTextEditor::Cursor' and $name =~ /operator/)
