--- branches/KDE/4.3/kdebindings/kalyptus/kalyptus #1060689:1060690
@@ -494,7 +494,7 @@
 		$p =~ s/#ifdef QT_GUI_LIB//;
 		$p =~ s/Q_CLASSINFO\(.*\)//;
 		$p =~ s/Q_DISABLE_COPY\s*\(\s*(\w+)\s*\)/$1(const $1 &);\n$1 &operator=(const $1 &);/;
-		$p =~ s/Q_DECLARE_INTERFACE\s*\(\s*(\w+)\s*\)//;
+                $p =~ s/Q_DECLARE_INTERFACE\s*\([^\)]*\)//;
 		$p =~ s/Q_INTERFACES\s*\(\s*([\w:]+)\s*\)//;
 		$p =~ s/KDEV_DECLARE_EXTENSION_INTERFACE_NS\s*\(\s*(\w+)\s*\)//;
 		$p =~ s/APPENDED_LISTS_STUB\s*\(\s*(\w+)\s*\)//;
--- branches/KDE/4.3/kdebindings/kalyptus/kalyptusCxxToSmoke.pm #1060689:1060690
@@ -445,6 +445,11 @@
 			|| $className eq 'QSharedData'
 			|| $className eq 'QString'
 
+			|| $className eq 'QTestEventList'
+			|| $className eq 'QTestKeyClicksEvent'
+			|| $className eq 'QTestKeyEvent'
+			|| $className eq 'QTestMouseEvent'
+
 			|| $className eq 'std'
 			|| $className eq 'std::string'
 			|| $className eq 'std::wstring'
@@ -978,7 +983,8 @@
 
 	# Hack for QAsyncIO. We don't implement the "if a class has no explicit copy ctor,
 	# then all of its member variables must be copiable, otherwise the class isn't copiable".
-	$hasPrivateCopyConstructor = 1 if ( $className eq 'QAsyncIO' );
+	$hasPrivateCopyConstructor = 1 if ( $className eq 'QAsyncIO' || $className eq 'QGraphicsLayoutItem'
+            ||$className eq 'QGraphicsSceneEvent' ||$className eq 'QPainterPathStroker' );
 
 	# Remember if this class can't be copied - it means all its descendants can't either
 	$classNode->AddProp( "CanBeCopied", !$hasPrivateCopyConstructor );
--- branches/KDE/4.3/kdebindings/smoke/qttest/header_list #1060689:1060690
@@ -354,3 +354,4 @@
 qwebpage.h
 qwebsettings.h
 qwebview.h
+qtestevent.h
--- branches/KDE/4.3/kdebindings/smoke/qtwebkit/header_list #1060689:1060690
@@ -347,3 +347,5 @@
 qsslsocket.h
 qsvggenerator.h
 qxmlstream.h
+qwebframe.h
+qwebelement.h
