diff -Naur kdebindings-4.3.4/python/pykde4/sip/kdecore/typedefs.sip kdebindings-4.3.4-p/python/pykde4/sip/kdecore/typedefs.sip
--- kdebindings-4.3.4/python/pykde4/sip/kdecore/typedefs.sip	2009-10-02 10:21:29.000000000 +0200
+++ kdebindings-4.3.4-p/python/pykde4/sip/kdecore/typedefs.sip	2010-01-15 17:52:40.000000000 +0100
@@ -466,7 +466,7 @@
 %End
 };
 
-
+/*
 template <TYPE*>
 %MappedType QSet<TYPE*>
 {
@@ -549,7 +549,7 @@
     return sipGetState(sipTransferObj);
 %End
 };
-
+*/
 
 template <TYPE>
 %MappedType QSet<TYPE>
