diff -Naur akonadi-1.4.0/server/src/storage/dbconfig.cpp akonadi-1.4.0-sqlite/server/src/storage/dbconfig.cpp
--- akonadi-1.4.0/server/src/storage/dbconfig.cpp	2010-07-31 18:12:25.000000000 +0200
+++ akonadi-1.4.0-sqlite/server/src/storage/dbconfig.cpp	2010-08-01 22:48:24.000000000 +0200
@@ -62,7 +62,7 @@
     QSettings settings( serverConfigFile, QSettings::IniFormat );
 
     // determine driver to use
-    const QString defaultDriver = QLatin1String( "QMYSQL" );
+    const QString defaultDriver = QLatin1String( "QSQLITE" );
     QString driverName = settings.value( QLatin1String( "General/Driver" ), defaultDriver ).toString();
     if ( driverName.isEmpty() )
       driverName = defaultDriver;
