diff -Naur akonadi-1.2.1/server/src/storage/dbconfig.cpp akonadi-1.2.1-sqlite/server/src/storage/dbconfig.cpp
--- akonadi-1.2.1/server/src/storage/dbconfig.cpp	2009-08-28 16:19:55.000000000 +0200
+++ akonadi-1.2.1-sqlite/server/src/storage/dbconfig.cpp	2010-02-02 20:52:55.000000000 +0100
@@ -38,7 +38,7 @@
       QSettings settings( serverConfigFile, QSettings::IniFormat );
 
       // determine driver to use
-      const QString defaultDriver = QLatin1String("QMYSQL");
+      const QString defaultDriver = QLatin1String("QSQLITE");
       mDriverName = settings.value( QLatin1String("General/Driver"), defaultDriver ).toString();
       if ( mDriverName.isEmpty() )
         mDriverName = defaultDriver;
