From e688cffa33ececfef8630c05380f93d9e38b3edf Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Tue, 28 Jul 2009 13:32:35 +1000
Subject: [PATCH] radeonhd: update for resources/RAC API removal

---
 src/rhd_driver.c |    9 ++++++---
 1 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/rhd_driver.c b/src/rhd_driver.c
index 943108c..e7dca79 100644
--- a/src/rhd_driver.c
+++ b/src/rhd_driver.c
@@ -38,7 +38,11 @@
 /* For PIO/MMIO */
 #include "compiler.h"
 
+#ifndef XSERVER_LIBPCIACCESS
 #include "xf86Resources.h"
+/* Needed by Resources Access Control (RAC) */
+#include "xf86RAC.h"
+#endif
 
 #include "xf86PciInfo.h"
 /* do we need to access PCI config space directly? */
@@ -63,9 +67,6 @@
 
 #include "fb.h"
 
-/* Needed by Resources Access Control (RAC) */
-#include "xf86RAC.h"
-
 #ifdef HAVE_XEXTPROTO_71
 #include "X11/extensions/dpmsconst.h"
 #else
@@ -562,10 +563,12 @@ RHDPreInit(ScrnInfoPtr pScrn, int flags)
 
     pScrn->chipset = (char *)xf86TokenToString(RHDChipsets, rhdPtr->ChipSet);
 
+#ifndef XSERVER_LIBPCIACCESS
     /* We will disable access to VGA legacy resources emulation and
        save/restore VGA thru MMIO when necessary */
     if (xf86RegisterResources(rhdPtr->entityIndex, NULL, ResNone))
 	goto error0;
+#endif
 
 #ifndef  ATOM_ASIC_INIT
     if (xf86LoadSubModule(pScrn, "int10")) {
-- 
1.6.3.3

