From 9e8608c4ee55ca7798c3c120938d2ecf1f7e9f00 Mon Sep 17 00:00:00 2001
From: Matthias Hopf <mhopf@suse.de>
Date: Mon, 8 Jun 2009 13:45:21 +0200
Subject: [PATCH] Adapt for nuked xf86LoaderReqSymLists() in Xserver 1.6.99.1.

---
 src/rhd_driver.c |   17 ++++++++++++++---
 1 files changed, 14 insertions(+), 3 deletions(-)

diff --git a/src/rhd_driver.c b/src/rhd_driver.c
index 8fcf32d..bc82505 100644
--- a/src/rhd_driver.c
+++ b/src/rhd_driver.c
@@ -989,8 +989,14 @@ RHDPreInit(ScrnInfoPtr pScrn, int flags)
 	    xf86DrvMsg(pScrn->scrnIndex, X_ERROR, "Failed to load XAA module."
 		       " Falling back to ShadowFB.\n");
 	    rhdPtr->AccelMethod = RHD_ACCEL_SHADOWFB;
-	} else
+	}
+	/* This is functionally void since 7.0 (old version scheme,
+	 * and removed since 1.6.99.1 (new version scheme) */
+	/* Assume anything >= 6.* is old version scheme */
+#if XORG_VERSION_CURRENT >= 6 * 10000000
+	else
 	    xf86LoaderReqSymLists(xaaSymbols, NULL);
+#endif
     }
 
 #ifdef USE_EXA
@@ -1000,9 +1006,14 @@ RHDPreInit(ScrnInfoPtr pScrn, int flags)
 	    xf86DrvMsg(pScrn->scrnIndex, X_ERROR, "Failed to load EXA module."
 		       " Falling back to ShadowFB.\n");
 	    rhdPtr->AccelMethod = RHD_ACCEL_SHADOWFB;
-	} else {
-	    xf86LoaderReqSymLists(exaSymbols, NULL);
 	}
+	/* This is functionally void since 7.0 (old version scheme,
+	 * and removed since 1.6.99.1 (new version scheme) */
+	/* Assume anything >= 6.* is old version scheme */
+#if XORG_VERSION_CURRENT >= 6 * 10000000
+	else
+	    xf86LoaderReqSymLists(exaSymbols, NULL);
+#endif
     }
 #endif /* USE_EXA */
 
-- 
1.6.3.3

