Index: kscope-1.9.4/app/app.pro
===================================================================
--- kscope-1.9.4.orig/app/app.pro	2011-11-19 23:25:37.749180214 -0500
+++ kscope-1.9.4/app/app.pro	2011-11-19 23:28:37.413174905 -0500
@@ -58,7 +58,10 @@
     -L../cscope \
     -lkscope_cscope \
     -L../editor \
-    -lkscope_editor
+    -lkscope_editor \
+    -L$${QSCI_ROOT_PATH}/lib \
+    -lqscintilla2
+
 RESOURCES = app.qrc
 target.path = $${INSTALL_PATH}/bin
 INSTALLS += target
