--- meson-0.54.2/mesonbuild/coredata.py	2020-05-15 05:45:50.000000000 +0200
+++ meson-0.54.2.new/mesonbuild/coredata.py	2020-06-02 15:31:16.368973834 +0200
@@ -1121,6 +1121,16 @@
     'sysconfdir':     {'/usr': '/etc'},
     'localstatedir':  {'/usr': '/var',     '/usr/local': '/var/local'},
     'sharedstatedir': {'/usr': '/var/lib', '/usr/local': '/var/local/lib'},
+    'libdir':         {'/usr': '/usr/lib'},
+    'bindir':         {'/usr': '/usr/bin'},
+    'sbindir':        {'/usr': '/usr/sbin'},
+    'datadir':        {'/usr': '/usr/share'},
+    'libexecdir':     {'/usr': '/usr/libexec'},
+    'datadir':        {'/usr': '/usr/share'},
+    'mandir':         {'/usr': '/usr/share/man'},
+    'localedir':      {'/usr': '/usr/share/locale'},
+    'includedir':     {'/usr': '/usr/include'},
+    'infodir':        {'/usr': '/usr/share/info'},   
 }
 
 forbidden_target_names = {'clean': None,
