diff -Naur meson-1.0.1/mesonbuild/coredata.py meson-1.0.1.new/mesonbuild/coredata.py
--- meson-1.0.1/mesonbuild/coredata.py	2023-02-23 20:23:09.000000000 +0100
+++ meson-1.0.1.new/mesonbuild/coredata.py	2023-02-27 14:44:17.755577341 +0100
@@ -1195,7 +1195,7 @@
 # Also update mesonlib._BUILTIN_NAMES. See the comment there for why this is required.
 BUILTIN_DIR_OPTIONS: 'MutableKeyedOptionDictType' = OrderedDict([
     (OptionKey('prefix'),          BuiltinOption(UserStringOption, 'Installation prefix', default_prefix())),
-    (OptionKey('bindir'),          BuiltinOption(UserStringOption, 'Executable directory', 'bin')),
+    (OptionKey('bindir'),          BuiltinOption(UserStringOption, 'Executable directory', '/usr/bin')),
     (OptionKey('datadir'),         BuiltinOption(UserStringOption, 'Data file directory', default_datadir())),
     (OptionKey('includedir'),      BuiltinOption(UserStringOption, 'Header file directory', default_includedir())),
     (OptionKey('infodir'),         BuiltinOption(UserStringOption, 'Info page directory', default_infodir())),
diff -Naur meson-1.0.1/mesonbuild/utils/universal.py meson-1.0.1.new/mesonbuild/utils/universal.py
--- meson-1.0.1/mesonbuild/utils/universal.py	2023-02-23 20:23:09.000000000 +0100
+++ meson-1.0.1.new/mesonbuild/utils/universal.py	2023-02-27 14:46:38.576580381 +0100
@@ -1015,7 +1015,7 @@
     if is_haiku():
         return 'lib'
     # There is no way to auto-detect this, so it must be set at build time
-    return 'libexec'
+    return '/usr/lib'
 
 
 def default_prefix() -> str:
@@ -1029,43 +1029,43 @@
 def default_datadir() -> str:
     if is_haiku():
         return 'data'
-    return 'share'
+    return '/usr/share'
 
 
 def default_includedir() -> str:
     if is_haiku():
         return 'develop/headers'
-    return 'include'
+    return '/usr/include'
 
 
 def default_infodir() -> str:
     if is_haiku():
         return 'documentation/info'
-    return 'share/info'
+    return '/usr/share/info'
 
 
 def default_localedir() -> str:
     if is_haiku():
         return 'data/locale'
-    return 'share/locale'
+    return '/usr/share/locale'
 
 
 def default_mandir() -> str:
     if is_haiku():
         return 'documentation/man'
-    return 'share/man'
+    return '/usr/share/man'
 
 
 def default_sbindir() -> str:
     if is_haiku():
         return 'bin'
-    return 'sbin'
+    return '/usr/sbin'
 
 
 def default_sysconfdir() -> str:
     if is_haiku():
         return 'settings'
-    return 'etc'
+    return '/usr/etc'
 
 
 def has_path_sep(name: str, sep: str = '/\\') -> bool:
