--- meson-0.57.0/mesonbuild/coredata.py	2021-02-14 20:28:44.000000000 +0100
+++ meson-0.57.0.new/mesonbuild/coredata.py	2021-02-15 09:22:37.629828796 +0100
@@ -1142,6 +1142,15 @@
     OptionKey('sysconfdir'):     {'/usr': '/etc'},
     OptionKey('localstatedir'):  {'/usr': '/var',     '/usr/local': '/var/local'},
     OptionKey('sharedstatedir'): {'/usr': '/var/lib', '/usr/local': '/var/local/lib'},
+    OptionKey('libdir'):         {'/usr': '/usr/lib'},
+    OptionKey('bindir'):         {'/usr': '/usr/bin'},
+    OptionKey('sbindir'):        {'/usr': '/usr/sbin'},
+    OptionKey('datadir'):        {'/usr': '/usr/share'},
+    OptionKey('libexecdir'):     {'/usr': '/usr/libexec'},
+    OptionKey('datadir'):        {'/usr': '/usr/share'},
+    OptionKey('mandir'):         {'/usr': '/usr/share/man'},
+    OptionKey('localedir'):      {'/usr': '/usr/share/locale'},
+    OptionKey('infodir'):        {'/usr': '/usr/share/info'},
 }
 
 FORBIDDEN_TARGET_NAMES = {'clean': None,
