diff -Naur meson-0.54.0/mesonbuild/coredata.py meson-0.54.0.new/mesonbuild/coredata.py
--- meson-0.54.0/mesonbuild/coredata.py	2020-03-29 19:37:15.000000000 +0200
+++ meson-0.54.0.new/mesonbuild/coredata.py	2020-04-14 09:41:46.910924499 +0200
@@ -1115,10 +1115,22 @@
 
 # Special prefix-dependent defaults for installation directories that reside in
 # a path outside of the prefix in FHS and common usage.
+# ROFL!!! .. so --prefix=/foo --bindir=/bar is not valid ? lol get real
+# we allow here bc I don't want to maintain large reverts an rework the parser and logic
 builtin_dir_noprefix_options = {
     'sysconfdir':     {'/usr': '/etc'},
     'localstatedir':  {'/usr': '/var',     '/usr/local': '/var/local'},
     'sharedstatedir': {'/usr': '/var/lib', '/usr/local': '/var/local/lib'},
+    'libdir':         {'/usr': '/usr/lib'},
+    'bindir':         {'/usr': '/usr/bin'},
+    'sbindir':        {'/usr': '/usr/sbin'},
+    'datadir':        {'/usr': '/usr/share'},
+    'libexecdir':     {'/usr': '/usr/libexec'},
+    'datadir':        {'/usr': '/usr/share'},
+    'mandir':         {'/usr': '/usr/share/man'},
+    'localedir':      {'/usr': '/usr/share/locale'},
+    'includedir':     {'/usr': '/usr/include'},
+    'infodir':        {'/usr': '/usr/share/info'},
 }
 
 forbidden_target_names = {'clean': None,
