--- meson-0.56.0/mesonbuild/coredata.py	2020-10-30 09:25:24.000000000 +0100
+++ meson-0.56.0.new/mesonbuild/coredata.py	2020-11-30 15:24:31.151141811 +0100
@@ -1202,6 +1202,15 @@
     'sysconfdir':     {'/usr': '/etc'},
     'localstatedir':  {'/usr': '/var',     '/usr/local': '/var/local'},
     'sharedstatedir': {'/usr': '/var/lib', '/usr/local': '/var/local/lib'},
+    'libdir':         {'/usr': '/usr/lib'},
+    'bindir':         {'/usr': '/usr/bin'},
+    'sbindir':        {'/usr': '/usr/sbin'},
+    'datadir':        {'/usr': '/usr/share'},
+    'libexecdir':     {'/usr': '/usr/libexec'},
+    'datadir':        {'/usr': '/usr/share'},
+    'mandir':         {'/usr': '/usr/share/man'},
+    'localedir':      {'/usr': '/usr/share/locale'},
+    'infodir':        {'/usr': '/usr/share/info'},   
 }
 
 FORBIDDEN_TARGET_NAMES = {'clean': None,
