--- meson-0.62.1/mesonbuild/coredata.py	2022-04-22 19:04:51.000000000 +0200
+++ meson-0.62.1.new/mesonbuild/coredata.py	2022-04-29 11:29:13.751258873 +0200
@@ -1186,18 +1186,18 @@
 # Also update mesonlib._BUILTIN_NAMES. See the comment there for why this is required.
 BUILTIN_DIR_OPTIONS: 'KeyedOptionDictType' = OrderedDict([
     (OptionKey('prefix'),          BuiltinOption(UserStringOption, 'Installation prefix', default_prefix())),
-    (OptionKey('bindir'),          BuiltinOption(UserStringOption, 'Executable directory', 'bin')),
-    (OptionKey('datadir'),         BuiltinOption(UserStringOption, 'Data file directory', 'share')),
-    (OptionKey('includedir'),      BuiltinOption(UserStringOption, 'Header file directory', 'include')),
-    (OptionKey('infodir'),         BuiltinOption(UserStringOption, 'Info page directory', 'share/info')),
+    (OptionKey('bindir'),          BuiltinOption(UserStringOption, 'Executable directory', '/usr/bin')),
+    (OptionKey('datadir'),         BuiltinOption(UserStringOption, 'Data file directory', '/usr/share')),
+    (OptionKey('includedir'),      BuiltinOption(UserStringOption, 'Header file directory', '/usr/include')),
+    (OptionKey('infodir'),         BuiltinOption(UserStringOption, 'Info page directory', '/usr/share/info')),
     (OptionKey('libdir'),          BuiltinOption(UserStringOption, 'Library directory', default_libdir())),
     (OptionKey('libexecdir'),      BuiltinOption(UserStringOption, 'Library executable directory', default_libexecdir())),
-    (OptionKey('localedir'),       BuiltinOption(UserStringOption, 'Locale data directory', 'share/locale')),
-    (OptionKey('localstatedir'),   BuiltinOption(UserStringOption, 'Localstate data directory', 'var')),
-    (OptionKey('mandir'),          BuiltinOption(UserStringOption, 'Manual page directory', 'share/man')),
-    (OptionKey('sbindir'),         BuiltinOption(UserStringOption, 'System executable directory', 'sbin')),
+    (OptionKey('localedir'),       BuiltinOption(UserStringOption, 'Locale data directory', '/usr/share/locale')),
+    (OptionKey('localstatedir'),   BuiltinOption(UserStringOption, 'Localstate data directory', '/var')),
+    (OptionKey('mandir'),          BuiltinOption(UserStringOption, 'Manual page directory', '/usr/share/man')),
+    (OptionKey('sbindir'),         BuiltinOption(UserStringOption, 'System executable directory', '/usr/sbin')),
     (OptionKey('sharedstatedir'),  BuiltinOption(UserStringOption, 'Architecture-independent data directory', 'com')),
-    (OptionKey('sysconfdir'),      BuiltinOption(UserStringOption, 'Sysconf data directory', 'etc')),
+    (OptionKey('sysconfdir'),      BuiltinOption(UserStringOption, 'Sysconf data directory', '/etc')),
 ])
 
 BUILTIN_CORE_OPTIONS: 'KeyedOptionDictType' = OrderedDict([
