diff --color -Naur fbthrift-2025.02.17.00/CMakeLists.txt fbthrift-2025.02.17.00.new/CMakeLists.txt
--- fbthrift-2025.02.17.00/CMakeLists.txt	2025-01-30 23:31:14.000000000 +0100
+++ fbthrift-2025.02.17.00.new/CMakeLists.txt	2025-02-21 09:06:20.909390679 +0100
@@ -106,7 +106,7 @@
 # Find required dependencies for thrift/lib
 if (THRIFT_LIB_ONLY OR build_all)
   find_package(Gflags REQUIRED)
-  find_package(Glog REQUIRED)
+  find_package(Glog CONFIG REQUIRED)
   find_package(folly CONFIG REQUIRED)
   find_package(fizz CONFIG REQUIRED)
   find_package(wangle CONFIG REQUIRED)
