diff -Naur vscode/resources/linux/code.appdata.xml vscode.new/resources/linux/code.appdata.xml
--- vscode/resources/linux/code.appdata.xml	2021-02-18 18:12:13.314708389 +0100
+++ vscode.new/resources/linux/code.appdata.xml	2021-02-18 18:13:04.471972447 +0100
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <component type="desktop">
-	<id>@@NAME@@.desktop</id>
-	<metadata_license>@@LICENSE@@</metadata_license>
-	<project_license>@@LICENSE@@</project_license>
-	<name>@@NAME_LONG@@</name>
+	<id>code-oss.desktop</id>
+	<metadata_license>MIT</metadata_license>
+	<project_license>MIT</project_license>
+	<name>Code - OSS</name>
 	<url type="homepage">https://code.visualstudio.com</url>
 	<summary>Visual Studio Code. Code editing. Redefined.</summary>
 	<description>
diff -Naur vscode/resources/linux/code.desktop vscode.new/resources/linux/code.desktop
--- vscode/resources/linux/code.desktop	2021-02-18 18:12:13.314708389 +0100
+++ vscode.new/resources/linux/code.desktop	2021-02-18 18:13:04.471972447 +0100
@@ -1,18 +1,18 @@
 [Desktop Entry]
-Name=@@NAME_LONG@@
+Name=Code - OSS
 Comment=Code Editing. Redefined.
 GenericName=Text Editor
-Exec=@@EXEC@@ --no-sandbox --unity-launch %F
-Icon=@@ICON@@
+Exec=/usr/bin/code-oss --no-sandbox --unity-launch %F
+Icon=com.visualstudio.code.oss
 Type=Application
 StartupNotify=false
-StartupWMClass=@@NAME_SHORT@@
+StartupWMClass=Code
 Categories=Utility;TextEditor;Development;IDE;
-MimeType=text/plain;inode/directory;application/x-@@NAME@@-workspace;
+MimeType=text/plain;application/x-code-oss-workspace;
 Actions=new-empty-window;
 Keywords=vscode;
 
 [Desktop Action new-empty-window]
 Name=New Empty Window
-Exec=@@EXEC@@ --no-sandbox --new-window %F
-Icon=@@ICON@@
+Exec=/usr/bin/code-oss --no-sandbox --new-window %F
+Icon=com.visualstudio.code.oss
diff -Naur vscode/resources/linux/code-url-handler.desktop vscode.new/resources/linux/code-url-handler.desktop
--- vscode/resources/linux/code-url-handler.desktop	2021-02-18 18:12:13.314708389 +0100
+++ vscode.new/resources/linux/code-url-handler.desktop	2021-02-18 18:13:15.652030168 +0100
@@ -1,12 +1,12 @@
 [Desktop Entry]
-Name=@@NAME_LONG@@ - URL Handler
+Name=Code - OSS - URL Handler
 Comment=Code Editing. Redefined.
 GenericName=Text Editor
-Exec=@@EXEC@@ --no-sandbox --open-url %U
-Icon=@@ICON@@
+Exec=/usr/bin/code-oss --no-sandbox --open-url %U
+Icon=com.visualstudio.code.oss
 Type=Application
 NoDisplay=true
 StartupNotify=true
 Categories=Utility;TextEditor;Development;IDE;
-MimeType=x-scheme-handler/@@URLPROTOCOL@@;
+MimeType=x-scheme-handler/code-oss;
 Keywords=vscode;
