diff -Naur gproftpd-8.2.3.orig/desktop/net-gproftpd.desktop gproftpd-8.2.3/desktop/net-gproftpd.desktop
--- gproftpd-8.2.3.orig/desktop/net-gproftpd.desktop	2006-02-07 01:00:15.000000000 +0100
+++ gproftpd-8.2.3/desktop/net-gproftpd.desktop	2006-03-06 00:24:35.000000000 +0100
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Name=GPROFTPD
+Name[hu]=GProFTPD
 Name[de]=GPROFTPD
 Name[es]=GPROFTPD
 Name[fi]=GPROFTPD
@@ -15,8 +16,9 @@
 Name[zh_TW]=GPROFTPD
 GenericName=GPROFTPD
 Comment=Proftpd FTP server administration
-Exec=gproftpd
+Comment[hu]=Grafikus Proftpd adminisztrátor
+Exec=gksu /usr/sbin/gproftpd
 Icon=gproftpd.png
 Terminal=false
 Type=Application
-Categories=Application;Network;X-Red-Hat-ServerConfig;X-Red-Hat-Base;
+Categories=Application;Network;
