diff -Naur CEGUI-0.6.0/Samples/TreeDemo/src/Sample_TreeDemo.cpp CEGUI-0.6.0.new/Samples/TreeDemo/src/Sample_TreeDemo.cpp
--- CEGUI-0.6.0/Samples/TreeDemo/src/Sample_TreeDemo.cpp	2008-02-08 15:59:54.000000000 +0530
+++ CEGUI-0.6.0.new/Samples/TreeDemo/src/Sample_TreeDemo.cpp	2008-04-02 20:31:17.000000000 +0530
@@ -13,6 +13,7 @@
 #endif
 
 #include <time.h>
+#include <stdlib.h>
 
 
 const unsigned int TreeDemoSample::TreeID    = 1;
