From 406b777137c8626c0669a330bf1410615f5ecc17 Mon Sep 17 00:00:00 2001
From: James Buren <ryuo@frugalware.org>
Date: Sat, 15 Dec 2012 13:00:38 -0600
Subject: [PATCH] drop /etc/sysconfig/language symlink

---
 Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/Makefile b/Makefile
index ea97706..3b96255 100644
--- a/Makefile
+++ b/Makefile
@@ -54,7 +54,6 @@ install:
 	done
 	$(INSTALL) -d $(DESTDIR)/etc/skel
 	$(INSTALL) -d $(DESTDIR)/etc/sysconfig
-	ln -sf ../profile.d/lang.sh $(DESTDIR)/etc/sysconfig/language
 	$(INSTALL) -d $(DESTDIR)/home
 	$(INSTALL) -d $(DESTDIR)/mnt
 	$(INSTALL) -d $(DESTDIR)/media
-- 
1.8.0.1

