From 0846bd123a98e16448b115789f5be37b39644365 Mon Sep 17 00:00:00 2001
From: DeX77 <dex77@frugalware.org>
Date: Fri, 16 Feb 2024 14:09:28 +0100
Subject: [PATCH] * remove lib64 symlinks. Confusing

---
 Makefile | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/Makefile b/Makefile
index 3df3fe1..a98113d 100644
--- a/Makefile
+++ b/Makefile
@@ -101,11 +101,6 @@ install:
 	$(INSTALL) -d $(DESTDIR)/var/spool
 	$(INSTALL) -d $(DESTDIR)/var/spool/mail
 	$(INSTALL) -d -m1777 $(DESTDIR)/var/tmp
-ifeq ($(shell uname -m),x86_64)
-	ln -sf /usr/lib $(DESTDIR)/lib64
-	ln -sf /usr/lib $(DESTDIR)/usr/lib64
-	ln -sf /usr/local/lib $(DESTDIR)/usr/local/lib64
-endif
 	ln -sf /usr/lib $(DESTDIR)/lib
 	ln -sf /usr/bin $(DESTDIR)/bin
 	ln -sf /usr/bin $(DESTDIR)/sbin
-- 
2.43.0

