diff -Naur frugalware-2.1/Makefile frugalware-2.1-p/Makefile
--- frugalware-2.1/Makefile	2016-07-29 15:49:43.000000000 +0200
+++ frugalware-2.1-p/Makefile	2019-01-16 02:40:41.366600167 +0100
@@ -88,6 +88,7 @@
 	ln -sf /var/tmp $(DESTDIR)/usr/tmp
 	$(INSTALL) -d $(DESTDIR)/var
 	ln -sf log $(DESTDIR)/var/adm
+	ln -sf ../run $(DESTDIR)/var/run
 	$(INSTALL) -d $(DESTDIR)/var/lib
 	$(INSTALL) -d $(DESTDIR)/var/lib/frugalware
 	$(INSTALL) -d $(DESTDIR)/var/lib/frugalware/system
@@ -101,7 +102,6 @@
 	for i in cat{1,2,3,4,5,6,7,8,9,n}; do \
 		$(INSTALL) -d -m700 $(DESTDIR)/var/man/$$i; \
 	done
-	$(INSTALL) -d $(DESTDIR)/var/run
 	$(INSTALL) -d $(DESTDIR)/var/spool
 	$(INSTALL) -d $(DESTDIR)/var/spool/mail
 	$(INSTALL) -d -m1777 $(DESTDIR)/var/tmp
