diff -Naur frugalware-2.1/etc/nsswitch.conf frugalware-2.1-p/etc/nsswitch.conf
--- frugalware-2.1/etc/nsswitch.conf	2016-07-29 15:49:43.000000000 +0200
+++ frugalware-2.1-p/etc/nsswitch.conf	2018-01-25 19:53:36.975777710 +0100
@@ -19,20 +19,17 @@
 #      [NOTFOUND=return]       Stop searching if not found so far
 #
 
-# passwd: 	files nis
-# shadow: 	files nis
-# group:  	files nis
+passwd:		files nis
+shadow:		files nis
+group:		files nis
 
-passwd: 	compat
-group:  	compat
-
-hosts:  	files dns
+hosts:		files dns
 networks:       files
 
 services:       files
 protocols:      files
-rpc:    	files
-ethers: 	files
+rpc:		files
+ethers:		files
 netmasks:       files
 netgroup:       files
 bootparams:     files
