commit 7091745e8d022851d75f3cfca4807b946df24821
Author: DeX77 <dex77@frugalware.org>
Date:   Wed Jan 17 13:07:36 2024 +0100

    * usr-merge

diff --git a/Makefile b/Makefile
index fe91491..3df3fe1 100644
--- a/Makefile
+++ b/Makefile
@@ -25,7 +25,6 @@ DESTDIR =
 all:
 
 install:
-	$(INSTALL) -d $(DESTDIR)/bin
 	$(INSTALL) -d $(DESTDIR)/boot
 	$(INSTALL) -d $(DESTDIR)/dev
 	$(INSTALL) -d $(DESTDIR)/etc
@@ -50,7 +49,6 @@ install:
 	$(INSTALL) -d $(DESTDIR)/opt
 	$(INSTALL) -d $(DESTDIR)/proc
 	$(INSTALL) -d -m710 $(DESTDIR)/root
-	$(INSTALL) -d $(DESTDIR)/sbin
 	$(INSTALL) -d $(DESTDIR)/sys
 	$(INSTALL) -d -m1777 $(DESTDIR)/tmp
 	$(INSTALL) -d -m1777 $(DESTDIR)/tmp/.ICE-unix
@@ -72,9 +70,7 @@ install:
 	for i in {cat,man}{1,2,3,4,5,6,7,8,9,n}; do \
 		$(INSTALL) -d $(DESTDIR)/usr/local/share/man/$$i; \
 	done
-	$(INSTALL) -d $(DESTDIR)/usr/local/sbin
 	$(INSTALL) -d $(DESTDIR)/usr/local/src
-	$(INSTALL) -d $(DESTDIR)/usr/sbin
 	$(INSTALL) -d $(DESTDIR)/usr/share
 	$(INSTALL) -d $(DESTDIR)/usr/share/man
 	for i in man{1,2,3,4,5,6,7,8,9,n}; do \
@@ -106,10 +102,14 @@ install:
 	$(INSTALL) -d $(DESTDIR)/var/spool/mail
 	$(INSTALL) -d -m1777 $(DESTDIR)/var/tmp
 ifeq ($(shell uname -m),x86_64)
-	ln -sf /lib $(DESTDIR)/lib64
+	ln -sf /usr/lib $(DESTDIR)/lib64
 	ln -sf /usr/lib $(DESTDIR)/usr/lib64
 	ln -sf /usr/local/lib $(DESTDIR)/usr/local/lib64
 endif
+	ln -sf /usr/lib $(DESTDIR)/lib
+	ln -sf /usr/bin $(DESTDIR)/bin
+	ln -sf /usr/bin $(DESTDIR)/sbin
+	ln -sf /usr/bin $(DESTDIR)/usr/sbin
 
 dist:
 	git archive --format=tar --prefix=frugalware-$(VERSION)/ HEAD > frugalware-$(VERSION).tar
