--- fox-1.7.79/lib/FXRex.cpp	2022-03-01 05:44:00.000000000 +0100
+++ fox-1.7.79.new/lib/FXRex.cpp	2022-04-26 15:47:00.485599645 +0200
@@ -503,7 +503,7 @@
 */
 
 #define TOPIC_CONSTRUCT 1000
-#define TOPIC_REXDUMP   1014          // Debugging regex code
+//#define TOPIC_REXDUMP   1014          // Debugging regex code
 
 
 // As close to infinity as we're going to get; this seems big enough.  We can not make
