diff -Naur notecase-1.1.4.old/src/LinkPropertiesDlg.h notecase-1.1.4/src/LinkPropertiesDlg.h
--- notecase-1.1.4.old/src/LinkPropertiesDlg.h	2005-10-06 13:40:00.000000000 +0200
+++ notecase-1.1.4/src/LinkPropertiesDlg.h	2006-05-10 00:48:33.000000000 +0200
@@ -28,7 +28,7 @@
 	LinkInfo m_info;
 
 protected:
-	GtkWidget* LinkPropertiesDlg::create_link_wizard ();
+	GtkWidget* create_link_wizard ();
 };
 
 #endif // LINKPROPERTIESDLG_H__
