diff -Naur svn2cl-0.8.orig/svn2cl.sh svn2cl-0.8/svn2cl.sh
--- svn2cl-0.8.orig/svn2cl.sh	2006-10-18 23:38:06.000000000 +0200
+++ svn2cl-0.8/svn2cl.sh	2006-10-19 12:29:43.000000000 +0200
@@ -243,7 +243,7 @@
 done
 dir=`dirname "$prog"`
 dir=`cd "$dir" && pwd`
-XSL="$dir/svn2${OUTSTYLE}.xsl"
+XSL="/usr/share/svn2cl/svn2${OUTSTYLE}.xsl"
 
 # check if the authors file is formatted as a legacy
 # colon separated file
