diff -Naur svn2cl-0.7.orig/svn2cl.sh svn2cl-0.7/svn2cl.sh
--- svn2cl-0.7.orig/svn2cl.sh	2006-09-05 15:19:10.000000000 +0200
+++ svn2cl-0.7/svn2cl.sh	2006-09-05 15:20:04.000000000 +0200
@@ -195,7 +195,7 @@
 done
 dir=`dirname $prog`
 dir=`cd $dir && pwd`
-XSL="$dir/svn2${OUTSTYLE}.xsl"
+XSL="/usr/share/svn2cl/svn2${OUTSTYLE}.xsl"
 
 # check if the authors file is formatted as a legacy
 # colon separated file
