--- rigs-0.2.2.20061009.orig/Ruby/GNUmakefile	2008-02-07 17:18:20.000000000 +0100
+++ rigs-0.2.2.20061009/Ruby/GNUmakefile	2008-02-07 17:34:30.000000000 +0100
@@ -28,7 +28,7 @@
 
 # Dynamically determine where the things must be installed
 # in Ruby
-RUBY_SITE_LIB_DIR := $(shell ruby -rrbconfig -e 'print Config::CONFIG["sitelibdir"]')
+RUBY_SITE_LIB_DIR := ${DESTDIR}/$(shell ruby -rrbconfig -e 'print Config::CONFIG["sitelibdir"]')
 
 
 after-install::
@@ -37,4 +37,4 @@
 	@cp -rf rigs $(RUBY_SITE_LIB_DIR)
 
 # Is there something more appropriate than tool.make here ?
-include $(GNUSTEP_MAKEFILES)/tool.make
\ Pas de fin de ligne à la fin du fichier.
+include $(GNUSTEP_MAKEFILES)/tool.make
--- rigs-0.2.2.20061009.orig/Source/GNUmakefile.postamble	2008-02-07 17:18:20.000000000 +0100
+++ rigs-0.2.2.20061009/Source/GNUmakefile.postamble	2008-02-07 17:36:21.000000000 +0100
@@ -41,7 +41,8 @@
 # Things to do after installing
 # install the librigs.so in Ruby dir.
 after-install::
-	@cp -f $(GNUSTEP_OBJ_DIR)/$(LIBRARY_NAME).so $(RUBY_SITE_ARCH_DIR)
+	@mkdir -p ${DESTDIR}/$(RUBY_SITE_ARCH_DIR)
+	@cp -f $(GNUSTEP_OBJ_DIR)/$(LIBRARY_NAME).so ${DESTDIR}/$(RUBY_SITE_ARCH_DIR)
 
 # Things to do before uninstalling
 #before-uninstall::
