diff -Naur xdvik-22.84.16/Makefile.in xdvik-22.84.16/Makefile.in
--- xdvik-22.84.16/Makefile.in	2009-03-02 22:59:36.000000000 +0100
+++ xdvik-22.84.16/Makefile.in	2010-09-05 13:21:22.776999994 +0200
@@ -48,14 +48,14 @@
 INSTDIRS = @INSTDIRS@
 
 default: all
-install: all 
+install: all
 	for dir in $(INSTDIRS); do \
 	   echo making $@ in $$dir; \
 	     (cd $$dir && PATH=$(scriptdir):$(bindir):$(PATH) \
                 $(MAKE) $@) || exit 1; \
 	done
-	if test -d $(texmf); then \
-	  $(INSTALL_DATA) $(srcdir)/release-tetex-src.txt $(texmf); \
+	if test -d $(DESTDIR)/$(texmf); then \
+	  $(INSTALL_DATA) $(srcdir)/release-tetex-src.txt $(DESTDIR)/$(texmf); \
 	fi
 @FMU@
 @FMU@install-extra:
diff -Naur xdvik-22.84.16//texk/xdvik/Makefile.in xdvik-22.84.16/texk/xdvik/Makefile.in
--- xdvik-22.84.16/texk/xdvik/Makefile.in	2009-03-02 22:59:38.000000000 +0100
+++ xdvik-22.84.16/texk/xdvik/Makefile.in	2010-09-05 13:17:21.144999993 +0200
@@ -1,5 +1,5 @@
 # @configure_input@
-# Makefile for xdvik 
+# Makefile for xdvik
 #
 
 kpse_include ../make/paths.mk
@@ -146,29 +146,29 @@
 uninstall: uninstall-exec uninstall-data
 
 install-exec: @final_exec_name@
-	$(SHELL) $(top_srcdir)/../mkinstalldirs $(bindir)
-	$(INSTALL_LIBTOOL_PROG) @final_exec_name@ $(bindir)
-	$(INSTALL_SCRIPT) xdvi-sh $(scriptdir)/@wrapper_script@
+	$(SHELL) $(top_srcdir)/../mkinstalldirs $(DESTDIR)/$(bindir)
+	$(INSTALL_LIBTOOL_PROG) @final_exec_name@ $(DESTDIR)/$(bindir)
+	$(INSTALL_SCRIPT) xdvi-sh $(DESTDIR)/$(scriptdir)/@wrapper_script@
 	rm -f $(scriptdir)/o@wrapper_script@
-	ln -s @wrapper_script@ $(scriptdir)/o@wrapper_script@
+	ln -s @wrapper_script@ $(DESTDIR)/$(scriptdir)/o@wrapper_script@
 
 uninstall-exec:
 	for p in @final_exec_name@; do rm -f $(bindir)/$$p; done
 	for p in @wrapper_script@ o@wrapper_script@; do rm -f $(scriptdir)/$$p; done
 
 install-data: $(manpage).1 pixmaps/toolbar.xpm pixmaps/toolbar2.xpm
-	$(SHELL) $(top_srcdir)/../mkinstalldirs $(man1dir) 
-	$(SHELL) $(top_srcdir)/../mkinstalldirs $(texmf)/xdvi/pixmaps
-	$(INSTALL_DATA) $(manpage).1 $(man1dir)/$(manpage).$(manext)
-	rm -f $(man1dir)/o$(manpage).$(manext)
-	ln -s $(manpage).$(manext) $(man1dir)/o$(manpage).$(manext)
-	$(INSTALL_DATA) $(srcdir)/texmf/XDvi $(texmf)/xdvi/XDvi
-	$(INSTALL_DATA) $(srcdir)/pixmaps/toolbar.xpm $(srcdir)/pixmaps/toolbar2.xpm $(texmf)/xdvi/pixmaps
+	$(SHELL) $(top_srcdir)/../mkinstalldirs $(DESTDIR)/$(man1dir)
+	$(SHELL) $(top_srcdir)/../mkinstalldirs $(DESTDIR)/$(texmf)/xdvi/pixmaps
+	$(INSTALL_DATA) $(manpage).1 $(DESTDIR)/$(man1dir)/$(manpage).$(manext)
+	rm -f $(DESTDIR)/$(man1dir)/o$(manpage).$(manext)
+	ln -s $(manpage).$(manext) $(DESTDIR)/$(man1dir)/o$(manpage).$(manext)
+	$(INSTALL_DATA) $(srcdir)/texmf/XDvi $(DESTDIR)/$(texmf)/xdvi/XDvi
+	$(INSTALL_DATA) $(srcdir)/pixmaps/toolbar.xpm $(srcdir)/pixmaps/toolbar2.xpm $(DESTDIR)/$(texmf)/xdvi/pixmaps
 	if \
-	    grep 'original xdvi.cfg --' $(texmf)/xdvi/xdvi.cfg >/dev/null 2>&1 \
-	    || test ! -r $(texmf)/xdvi/xdvi.cfg; \
+	    grep 'original xdvi.cfg --' $(DESTDIR)/$(texmf)/xdvi/xdvi.cfg >/dev/null 2>&1 \
+	    || test ! -r $(DESTDIR)/$(texmf)/xdvi/xdvi.cfg; \
 	then \
-	    $(INSTALL_DATA) $(srcdir)/texmf/xdvi.cfg $(texmf)/xdvi/xdvi.cfg; \
+	    $(INSTALL_DATA) $(srcdir)/texmf/xdvi.cfg $(DESTDIR)/$(texmf)/xdvi/xdvi.cfg; \
 	else \
 	    true; \
 	fi
