--- volume_key-0.3.12/configure.ac	2018-10-08 13:24:58.000000000 +0200
+++ volume_key-0.3.12.new/configure.ac	2022-11-04 14:00:00.834217209 +0100
@@ -183,9 +183,9 @@
 fi
 AM_CONDITIONAL([HAVE_PYTHON3], [test x$python3_found = xyes])
 
-AC_PATH_PROG([GPG], [gpg2])
+AC_PATH_PROG([GPG], [gpg])
 AC_ARG_VAR([GPG])
-AC_DEFINE_UNQUOTED([GPG_PATH], "$GPG", [Path to the gpg2 executable])
+AC_DEFINE_UNQUOTED([GPG_PATH], "$GPG", [Path to the gpg executable])
 
 # Checks for libraries.
 PKG_CHECK_MODULES(blkid, [blkid])
@@ -195,6 +195,7 @@
 PKG_CHECK_MODULES(libcryptsetup, [libcryptsetup])
 PKG_CHECK_MODULES(nss, [nss])
 
+AM_PATH_GPG_ERROR
 AM_PATH_GPGME
 
 AM_GNU_GETTEXT([external])
