diff -up kdemultimedia-4.6.2/CMakeLists.txt.no_mplayerthumbs kdemultimedia-4.6.2/CMakeLists.txt
--- kdemultimedia-4.6.2/CMakeLists.txt.no_mplayerthumbs	2011-01-19 16:23:35.000000000 -0600
+++ kdemultimedia-4.6.2/CMakeLists.txt	2011-04-17 15:52:42.161989091 -0500
@@ -42,10 +42,10 @@ configure_file (config.h.cmake ${CMAKE_C
 
 add_subdirectory(libkcddb)
 add_subdirectory(libkcompactdisc)
-macro_optional_add_subdirectory(mplayerthumbs)
+#macro_optional_add_subdirectory(mplayerthumbs)
 
 if (FFMPEG_FOUND AND SWSCALE_FOUND)
-     macro_optional_add_subdirectory(ffmpegthumbs)
+#     macro_optional_add_subdirectory(ffmpegthumbs)
 endif (FFMPEG_FOUND AND SWSCALE_FOUND)
 
 if (TAGLIB_FOUND)
