diff -Naur iproute2-2.6.11-050330.orig/include/linux/netlink.h iproute2-2.6.11-050330/include/linux/netlink.h
--- iproute2-2.6.11-050330.orig/include/linux/netlink.h	2005-06-02 15:27:33.000000000 +0000
+++ iproute2-2.6.11-050330/include/linux/netlink.h	2005-06-02 15:31:57.000000000 +0000
@@ -1,7 +1,7 @@
 #ifndef __LINUX_NETLINK_H
 #define __LINUX_NETLINK_H
 
-#include <linux/socket.h> /* for sa_family_t */
+#include <sys/socket.h> /* for sa_family_t */
 #include <linux/types.h>
 
 #define NETLINK_ROUTE		0	/* Routing/device hook				*/
