diff -urN control-center-2.16.0.orig/schemas/desktop_gnome_font_rendering.schemas.in control-center-2.16.0/schemas/desktop_gnome_font_rendering.schemas.in
--- control-center-2.16.0.orig/schemas/desktop_gnome_font_rendering.schemas.in	2002-12-17 16:24:56.000000000 +0000
+++ control-center-2.16.0/schemas/desktop_gnome_font_rendering.schemas.in	2006-09-17 14:38:22.000000000 +0100
@@ -6,7 +6,7 @@
 	    <applyto>/desktop/gnome/font_rendering/dpi</applyto>
             <owner>gnome</owner>
 	    <type>float</type>
-            <default>96</default>
+            <default>72</default>
             <locale name="C">
 	        <short>DPI</short>
                 <long>
