diff -Naur control-center-2.12.0.orig/capplets/theme-switcher/gnome-theme-installer.c control-center-2.12.0/capplets/theme-switcher/gnome-theme-installer.c
--- control-center-2.12.0.orig/capplets/theme-switcher/gnome-theme-installer.c	2005-02-08 23:42:10.000000000 +0100
+++ control-center-2.12.0/capplets/theme-switcher/gnome-theme-installer.c	2005-09-20 19:06:24.000000000 +0200
@@ -209,13 +209,13 @@
 	
 	if (path && len > 7 && ( (!strcmp (path + len - 7, ".tar.gz")) || (!strcmp (path + len - 4, ".tgz")) )) {
 		filename = g_shell_quote (path);
-		command = g_strdup_printf ("sh -c '/bin/gzip -d -c < \"%s\" | /bin/tar ft -  | head -1'",
+		command = g_strdup_printf ("sh -c '/bin/gzip -d -c < \"%s\" | /bin/tar ft -  | head -n 1'",
 					    filename);
 		theme_props->filetype=TARGZ;
 		g_free (filename);
 	} else if (path && len > 8 && !strcmp (path + len - 8, ".tar.bz2")) {
 		filename = g_shell_quote (path);
-		command = g_strdup_printf ("sh -c '/usr/bin/bzip2 -d -c < \"%s\" | /bin/tar ft - | head -1'",
+		command = g_strdup_printf ("sh -c '/usr/bin/bzip2 -d -c < \"%s\" | /bin/tar ft - | head -n 1'",
 					    filename);
 		theme_props->filetype=TARBZ;
 		g_free (filename);
