Index: xine-ui-0.99.4/src/xitk/main.c
===================================================================
--- xine-ui-0.99.4.orig/src/xitk/main.c
+++ xine-ui-0.99.4/src/xitk/main.c
@@ -375,6 +375,7 @@ static void xrm_parse(void) {
   }
   
   (void) XrmMergeDatabases(server_rmdb, &rmdb);
+  if ( server_rmdb != rmdb ) XrmDestroyDatabase(server_rmdb);
   
   if((environment = getenv("XENVIRONMENT")) == NULL) {
     int len;
@@ -758,10 +759,11 @@ static xine_video_port_t *load_video_out
 #ifdef HAVE_XINERAMA
   if (XineramaQueryExtension(gGui->video_display, &dummy_event, &dummy_error)) {
     int count = 1;
-    XineramaQueryScreens(gGui->video_display, &count);
+    XineramaScreenInfo *info = XineramaQueryScreens(gGui->video_display, &count);
     if (count > 1)
       /* multihead -> assuming square pixels */
       gGui->pixel_aspect = 1.0;
+    if ( info ) XFree(info);
   }
 #endif
 
