Search for the renamed libindicator .pc file

--- xfce4-indicator-plugin-0.2.1.orig/configure.in
+++ xfce4-indicator-plugin-0.2.1/configure.in
@@ -69,7 +69,7 @@ dnl ***********************************
 XDT_CHECK_PACKAGE([GTK], [gtk+-2.0], [2.6.0])
 XDT_CHECK_PACKAGE([LIBXFCE4UTIL], [libxfce4util-1.0], [4.3.99.2])
 XDT_CHECK_PACKAGE([LIBXFCE4PANEL], [libxfce4panel-1.0], [4.3.99.2])
-XDT_CHECK_PACKAGE([INDICATOR], [indicator], [0.3.0])
+XDT_CHECK_PACKAGE([INDICATOR], [indicator-0.4], [0.3.0])
 
 dnl ***********************************
 dnl *** Check for debugging support ***
@@ -79,8 +79,8 @@ XDT_FEATURE_DEBUG()
 dnl ***********************
 dnl *** Indicator info ***
 dnl ***********************
-INDICATORDIR=`$PKG_CONFIG --variable=indicatordir indicator`
-INDICATORICONSDIR=`$PKG_CONFIG --variable=iconsdir indicator`
+INDICATORDIR=`$PKG_CONFIG --variable=indicatordir indicator-0.4`
+INDICATORICONSDIR=`$PKG_CONFIG --variable=iconsdir indicator-0.4`
 
 AC_SUBST(INDICATORDIR)
 AC_SUBST(INDICATORICONSDIR)
--- xfce4-indicator-plugin-0.2.1.orig/configure
+++ xfce4-indicator-plugin-0.2.1/configure
@@ -13245,22 +13245,22 @@ $as_echo "$xdt_cv_PKG_CONFIG_VERSION" >&
   fi
 
 
-  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for indicator >= 0.3.0" >&5
-$as_echo_n "checking for indicator >= 0.3.0... " >&6; }
-  if $PKG_CONFIG "--atleast-version=0.3.0" "indicator" >/dev/null 2>&1; then
-    INDICATOR_VERSION=`$PKG_CONFIG --modversion "indicator"`
+  { $as_echo "$as_me:${as_lineno-$LINENO}: checking for indicator-0.4 >= 0.3.0" >&5
+$as_echo_n "checking for indicator-0.4 >= 0.3.0... " >&6; }
+  if $PKG_CONFIG "--atleast-version=0.3.0" "indicator-0.4" >/dev/null 2>&1; then
+    INDICATOR_VERSION=`$PKG_CONFIG --modversion "indicator-0.4"`
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $INDICATOR_VERSION" >&5
 $as_echo "$INDICATOR_VERSION" >&6; }
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking INDICATOR_CFLAGS" >&5
 $as_echo_n "checking INDICATOR_CFLAGS... " >&6; }
-    INDICATOR_CFLAGS=`$PKG_CONFIG --cflags "indicator"`
+    INDICATOR_CFLAGS=`$PKG_CONFIG --cflags "indicator-0.4"`
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $INDICATOR_CFLAGS" >&5
 $as_echo "$INDICATOR_CFLAGS" >&6; }
 
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking INDICATOR_LIBS" >&5
 $as_echo_n "checking INDICATOR_LIBS... " >&6; }
-    INDICATOR_LIBS=`$PKG_CONFIG --libs "indicator"`
+    INDICATOR_LIBS=`$PKG_CONFIG --libs "indicator-0.4"`
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: $INDICATOR_LIBS" >&5
 $as_echo "$INDICATOR_LIBS" >&6; }
 
@@ -13272,15 +13272,15 @@ $as_echo "$INDICATOR_LIBS" >&6; }
 
 
 
-  elif $PKG_CONFIG --exists "indicator" >/dev/null 2>&1; then
-    xdt_cv_version=`$PKG_CONFIG --modversion "indicator"`
+  elif $PKG_CONFIG --exists "indicator-0.4" >/dev/null 2>&1; then
+    xdt_cv_version=`$PKG_CONFIG --modversion "indicator-0.4"`
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: found, but $xdt_cv_version" >&5
 $as_echo "found, but $xdt_cv_version" >&6; }
 
 
-      echo "*** The required package indicator was found on your system,"
+      echo "*** The required package indicator-0.4 was found on your system,"
       echo "*** but the installed version ($xdt_cv_version) is too old."
-      echo "*** Please upgrade indicator to atleast version 0.3.0, or adjust"
+      echo "*** Please upgrade indicator-0.4 to atleast version 0.3.0, or adjust"
       echo "*** the PKG_CONFIG_PATH environment variable if you installed"
       echo "*** the new version of the package in a nonstandard prefix so"
       echo "*** pkg-config is able to find it."
@@ -13291,8 +13291,8 @@ $as_echo "found, but $xdt_cv_version" >&
 $as_echo "not found" >&6; }
 
 
-      echo "*** The required package indicator was not found on your system."
-      echo "*** Please install indicator (atleast version 0.3.0) or adjust"
+      echo "*** The required package indicator-0.4 was not found on your system."
+      echo "*** Please install indicator-0.4 (atleast version 0.3.0) or adjust"
       echo "*** the PKG_CONFIG_PATH environment variable if you"
       echo "*** installed the package in a nonstandard prefix so that"
       echo "*** pkg-config is able to find it."
@@ -13390,8 +13390,8 @@ $as_echo "minimum" >&6; }
   fi
 
 
-INDICATORDIR=`$PKG_CONFIG --variable=indicatordir indicator`
-INDICATORICONSDIR=`$PKG_CONFIG --variable=iconsdir indicator`
+INDICATORDIR=`$PKG_CONFIG --variable=indicatordir indicator-0.4`
+INDICATORICONSDIR=`$PKG_CONFIG --variable=iconsdir indicator-0.4`
 
 
 
