From 4c410464a89615547e959c0392c69c77add4eaf5 Mon Sep 17 00:00:00 2001
From: Mark Trompell <mark@foresightlinux.org>
Date: Tue, 19 Jul 2011 06:47:21 +0000
Subject: fix menu position

---
diff --git a/panel-plugin/indicator.c b/panel-plugin/indicator.c
index 1741a9c..4468a52 100644
--- a/panel-plugin/indicator.c
+++ b/panel-plugin/indicator.c
@@ -163,14 +163,12 @@ on_button_press (GtkWidget *widget, GdkEventButton *event, IndicatorPlugin *indi
   {
     if( event->button == 1) /* left click only */
     {
-      gtk_menu_popup (GTK_MENU(g_object_get_data (G_OBJECT(widget),"menu")), NULL,
-                      NULL, NULL, NULL, 1, event->time);
-      /* no approvement to the above */
-      /*
-      gtk_menu_popup (GTK_MENU(g_object_get_data (G_OBJECT(widget),"menu")), NULL, NULL,
+      GtkMenu * menu = GTK_MENU(g_object_get_data (G_OBJECT(widget),"menu"));
+      gtk_menu_attach_to_widget(menu, widget, NULL);
+      gtk_menu_popup (menu, NULL, NULL,
                       xfce_panel_plugin_position_menu,
                       indicator->plugin, 1, gtk_get_current_event_time ());
-      */
+      
       return TRUE;
     }
     /* event doesn't make it to the ebox, so I just push it. */
--
cgit 
