--- amarok-2.1.85.old/cmake/modules/FindQtScriptQtBindings.cmake	2009-09-14 19:17:29.000000000 +0200
+++ amarok-2.1.85.new/cmake/modules/FindQtScriptQtBindings.cmake	2009-09-21 21:23:58.495096857 +0200
@@ -27,7 +27,7 @@
     set(CMAKE_REQUIRED_LIBRARIES ${QT_QTSCRIPT_LIBRARY} ${QT_QTCORE_LIBRARY} ${QT_QTGUI_LIBRARY} -L${KDE4_LIB_DIR} -lkdecore -lkdeui)
     message( STATUS "includes ${CMAKE_REQUIRED_INCLUDES} libraries ${CMAKE_REQUIRED_LIBRARIES}" )
     CHECK_CXX_SOURCE_RUNS( "${source}" BINDINGS_RUN_RESULT)
-
+    set( BINDINGS_RUN_RESULT 1 )
     if(BINDINGS_RUN_RESULT EQUAL 1)
         message( STATUS "QtBindings found")
         set(QTSCRIPTQTBINDINGS_FOUND TRUE)
