--- amarok-2.3.90/src/core-impl/collections/db/sql/mysqlecollection/MySqlEmbeddedStorage.cpp	2011-01-05 11:57:08.684157296 +0100
+++ a/src/core-impl/collections/db/sql/mysqlecollection/MySqlEmbeddedStorage.cpp	2011-01-07 15:31:59.409524333 +0100
@@ -37,6 +37,7 @@
     QString storagePath = storageLocation;
     QString defaultsFile;
     QString databaseDir;
+    QString myServerOpt;
     if( storageLocation.isEmpty() )
     {
         storagePath = Amarok::saveLocation();
@@ -82,7 +83,16 @@
     }
 
     setenv( "MYSQL_HOME", storagePath.toAscii().data(), 1 );
-    if( mysql_server_init( 0 , 0, 0 ) != 0 )
+
+    myServerOpt = "--defaults-file=" + defaultsFile;
+
+    static char *server_options[] = \
+       { (char *)("Amarok"), myServerOpt.toLatin1().data(), NULL };
+    int num_elements = (sizeof(server_options) / sizeof(char *)) - 1;
+
+    static char *server_groups[] = { (char *)("embedded"), NULL };
+
+    if( mysql_library_init( num_elements , server_options, server_groups ) != 0 )
     {
         error() << "MySQL library initialization failed.";
         reportError( "init" );
