diff -up amarok-1.4.8/amarok/src/metadata/mp4/mp4file.cpp.gcc43 amarok-1.4.8/amarok/src/metadata/mp4/mp4file.cpp
--- amarok-1.4.8/amarok/src/metadata/mp4/mp4file.cpp.gcc43	2007-12-17 09:53:19.000000000 -0600
+++ amarok-1.4.8/amarok/src/metadata/mp4/mp4file.cpp	2008-02-21 21:05:18.000000000 -0600
@@ -30,6 +30,7 @@ email                : aumuell@reserv.at
 #include <audioproperties.h>
 
 #include <stdint.h>
+#include <stdlib.h>
 
 #define MP4V2_HAS_WRITE_BUG 1
 
diff -up amarok-1.4.8/amarok/src/metadata/mp4/taglib_mp4filetyperesolver.cpp.gcc43 amarok-1.4.8/amarok/src/metadata/mp4/taglib_mp4filetyperesolver.cpp
--- amarok-1.4.8/amarok/src/metadata/mp4/taglib_mp4filetyperesolver.cpp.gcc43	2007-12-17 09:53:19.000000000 -0600
+++ amarok-1.4.8/amarok/src/metadata/mp4/taglib_mp4filetyperesolver.cpp	2008-02-21 21:05:18.000000000 -0600
@@ -24,6 +24,7 @@
 
 #include "taglib_mp4filetyperesolver.h"
 #include "mp4file.h"
+#include <string.h>
 
 TagLib::File *MP4FileTypeResolver::createFile(const char *fileName,
         bool readProperties,
diff -up amarok-1.4.8/amarok/src/metadata/mp4/mp4tag.cpp.gcc43 amarok-1.4.8/amarok/src/metadata/mp4/mp4tag.cpp
--- amarok-1.4.8/amarok/src/metadata/mp4/mp4tag.cpp.gcc43	2007-12-17 09:53:19.000000000 -0600
+++ amarok-1.4.8/amarok/src/metadata/mp4/mp4tag.cpp	2008-02-21 21:05:18.000000000 -0600
@@ -24,6 +24,8 @@ email                : andrew.leadbetter
 
 #include <tag.h>
 #include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
 
 using namespace TagLib;
 
diff -up amarok-1.4.8/amarok/src/metadata/mp4/mp4properties.cpp.gcc43 amarok-1.4.8/amarok/src/metadata/mp4/mp4properties.cpp
--- amarok-1.4.8/amarok/src/metadata/mp4/mp4properties.cpp.gcc43	2007-12-17 09:53:19.000000000 -0600
+++ amarok-1.4.8/amarok/src/metadata/mp4/mp4properties.cpp	2008-02-21 21:05:18.000000000 -0600
@@ -30,6 +30,7 @@ email                : andrew.leadbetter
 #endif
 
 #include <stdint.h>
+#include <string.h>
 
 #ifndef UINT64_TO_DOUBLE
 #define UINT64_TO_DOUBLE(a) ((double)((int64_t)(a)))
diff -up amarok-1.4.8/amarok/src/engine/helix/hxplayercontrol.cpp.gcc43 amarok-1.4.8/amarok/src/engine/helix/hxplayercontrol.cpp
--- amarok-1.4.8/amarok/src/engine/helix/hxplayercontrol.cpp.gcc43	2007-12-17 09:53:18.000000000 -0600
+++ amarok-1.4.8/amarok/src/engine/helix/hxplayercontrol.cpp	2008-02-21 21:07:10.000000000 -0600
@@ -15,6 +15,8 @@
  the alsa guys
  ***************************************************************************/
 
+#include <stdlib.h>
+#include <string.h>
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/wait.h>
diff -up amarok-1.4.8/amarok/src/engine/helix/helix-sp/helix-sp.h.gcc43 amarok-1.4.8/amarok/src/engine/helix/helix-sp/helix-sp.h
--- amarok-1.4.8/amarok/src/engine/helix/helix-sp/helix-sp.h.gcc43	2007-12-17 09:53:18.000000000 -0600
+++ amarok-1.4.8/amarok/src/engine/helix/helix-sp/helix-sp.h	2008-02-21 21:07:54.000000000 -0600
@@ -23,6 +23,7 @@ class HelixSimplePlayerAudioStreamInfoRe
 #include <limits.h>
 #include <sys/param.h>
 #include <pthread.h>
+#include <string.h>
 #include <vector>
 #include <config.h>
 #include <iostream>
diff -up amarok-1.4.8/amarok/src/engine/helix/helix-engine.cpp.gcc43 amarok-1.4.8/amarok/src/engine/helix/helix-engine.cpp
--- amarok-1.4.8/amarok/src/engine/helix/helix-engine.cpp.gcc43	2007-12-17 09:53:18.000000000 -0600
+++ amarok-1.4.8/amarok/src/engine/helix/helix-engine.cpp	2008-02-21 21:09:38.000000000 -0600
@@ -15,6 +15,7 @@
 #include <climits>
 #include <cmath>
 #include <stdarg.h>
+#include <string.h>
 
 #include <config.h>
 
