diff -Naur amarok-1.4.5/amarok/src/magnatunebrowser/magnatunealbumdownloader.cpp amarok-1.4.5-p/amarok/src/magnatunebrowser/magnatunealbumdownloader.cpp
--- amarok-1.4.5/amarok/src/magnatunebrowser/magnatunealbumdownloader.cpp	2007-02-01 19:14:17.000000000 +0100
+++ amarok-1.4.5-p/amarok/src/magnatunebrowser/magnatunealbumdownloader.cpp	2007-02-16 00:09:28.000000000 +0100
@@ -89,7 +89,7 @@
 
     //ok, now we have the .zip file downloaded. All we need is to unpack it to the desired location and add it to the collection.
 
-    QString unzipString = "unzip \""+m_tempDir.name() + m_currentAlbumFileName + "\" -d \"" + m_currentAlbumUnpackLocation + "\" &";
+    QString unzipString = "unzip " + KProcess::quote( m_tempDir.name() + m_currentAlbumFileName ) + " -d " + KProcess::quote( m_currentAlbumUnpackLocation ) + " &";
 
     debug() << "unpacking: " << unzipString << endl;
 
