Only in ../../../src/textadept_2.0: LICENSE
diff -aur ../../../src/textadept_2.0/core/ext/mime_types.conf textadept/core/ext/mime_types.conf
--- ../../../src/textadept_2.0/core/ext/mime_types.conf	2009-09-27 09:26:17.000000000 -0500
+++ textadept/core/ext/mime_types.conf	2009-11-07 01:55:57.000000000 -0600
@@ -12,6 +12,7 @@
 %   #shbang_word lexer_name
 %   /lua_pattern lexer_name
 
+FrugalBuild frugalbuild
 
 % Actionscript
 as actionscript
Only in textadept/doc: LICENSE
Only in textadept/lexers: frugalbuild.lua
Only in ../../../src/textadept_2.0: src
