diff -Naur trickle-1.07/client.c trickle-1.07-p/client.c
--- trickle-1.07/client.c	2003-05-09 04:16:42.000000000 +0200
+++ trickle-1.07-p/client.c	2007-04-13 19:35:09.000000000 +0200
@@ -33,9 +33,11 @@
 #endif /* defined(HAVE_TIME_H) && defined(TIME_WITH_SYS_TIME) */
 #include <string.h>
 
+#ifndef HAVE_CONFIG_H
 #ifdef HAVE_NETINET_IN_H
 #include <netinet/in.h>
 #endif /* HAVE_NETINET_IN_H */
+#endif
 
 #include "print.h"
 #include "trickle.h"
diff -Naur trickle-1.07/trickle-overload.c trickle-1.07-p/trickle-overload.c
--- trickle-1.07/trickle-overload.c	2004-12-13 02:21:53.000000000 +0100
+++ trickle-1.07-p/trickle-overload.c	2007-04-13 19:33:03.000000000 +0200
@@ -24,7 +24,12 @@
 #include <sys/time.h>
 #endif /* HAVE_SYS_TIME_H */
 
+#ifndef HAVE_CONFIG_H
+#ifdef HAVE_NETINET_IN_H
 #include <netinet/in.h>
+#endif
+#endif
+
 
 #ifdef HAVE_ERR_H
 #include <err.h>
diff -Naur trickle-1.07/tricklectl.c trickle-1.07-p/tricklectl.c
--- trickle-1.07/tricklectl.c	2004-12-13 00:50:10.000000000 +0100
+++ trickle-1.07-p/tricklectl.c	2007-04-13 19:32:38.000000000 +0200
@@ -32,9 +32,12 @@
 #include <time.h>
 #endif /* defined(HAVE_TIME_H) && defined(TIME_WITH_SYS_TIME) */
 
+#ifndef HAVE_CONFIG_H
 #ifdef HAVE_NETINET_IN_H
 #include <netinet/in.h>
-#endif /* HAVE_NETINET_IN_H */
+#endif
+#endif
+
 
 #include "message.h"
 #include "trickledu.h"
diff -Naur trickle-1.07/trickled.c trickle-1.07-p/trickled.c
--- trickle-1.07/trickled.c	2003-12-12 23:33:23.000000000 +0100
+++ trickle-1.07-p/trickled.c	2007-04-13 19:31:45.000000000 +0200
@@ -30,7 +30,13 @@
 #include <sys/time.h>
 #endif /* HAVE_SYS_TIME_H */
 
+
+#ifndef HAVE_CONFIG_H
+#ifdef HAVE_NETINET_IN_H
 #include <netinet/in.h>
+#endif
+#endif
+
 #include <netinet/in_systm.h>
 
 #ifdef HAVE_ERR_H
diff -Naur trickle-1.07/trickledu.c trickle-1.07-p/trickledu.c
--- trickle-1.07/trickledu.c	2004-02-13 07:11:21.000000000 +0100
+++ trickle-1.07-p/trickledu.c	2007-04-13 19:32:24.000000000 +0200
@@ -29,9 +29,12 @@
 #include <time.h>
 #endif /* defined(HAVE_TIME_H) && defined(TIME_WITH_SYS_TIME) */
 
+#ifndef HAVE_CONFIG_H
 #ifdef HAVE_NETINET_IN_H
 #include <netinet/in.h>
-#endif /* HAVE_NETINET_IN_H */
+#endif
+#endif
+
 
 #include "trickle.h"
 #include "message.h"
