diff -up trickle-1.06/client.c.include_netdb trickle-1.06/client.c
--- trickle-1.06/client.c.include_netdb	2003-05-09 04:16:42.000000000 +0200
+++ trickle-1.06/client.c	2008-06-07 17:16:50.000000000 +0200
@@ -6,7 +6,7 @@
  *
  * $Id: client.c,v 1.14 2003/05/09 02:16:42 marius Exp $
  */
-
+#include <netdb.h>
 #include <sys/types.h>
 
 #ifdef HAVE_CONFIG_H
diff -up trickle-1.06/tricklectl.c.include_netdb trickle-1.06/tricklectl.c
--- trickle-1.06/tricklectl.c.include_netdb	2003-06-03 01:13:28.000000000 +0200
+++ trickle-1.06/tricklectl.c	2008-06-07 17:18:01.000000000 +0200
@@ -6,7 +6,7 @@
  *
  * $Id: tricklectl.c,v 1.4 2003/06/02 23:13:28 marius Exp $
  */
-
+#include <netdb.h>
 #include <sys/types.h>
 
 #ifdef HAVE_CONFIG_H
diff -up trickle-1.06/trickled.c.include_netdb trickle-1.06/trickled.c
--- trickle-1.06/trickled.c.include_netdb	2003-05-09 04:16:42.000000000 +0200
+++ trickle-1.06/trickled.c	2008-06-07 17:16:34.000000000 +0200
@@ -13,6 +13,7 @@
  *   initial spike
  */
 
+#include <netdb.h>
 #include <sys/types.h>
 
 #ifdef HAVE_CONFIG_H
--- trickle-1.07/trickledu.c.old        2008-08-29 04:19:47.000000000 +0300
+++ trickle-1.07/trickledu.c    2008-08-29 04:20:37.000000000 +0300
@@ -7,6 +7,7 @@
  * $Id: trickledu.c,v 1.16 2004/02/13 06:11:21 marius Exp $
  */

+#include <netdb.h>
 #include <sys/types.h>

 #ifdef HAVE_CONFIG_H
diff -up trickle-1.06/trickle-overload.c.include_netdb trickle-1.06/trickle-overload.c
--- trickle-1.06/trickle-overload.c.include_netdb	2003-06-03 01:13:28.000000000 +0200
+++ trickle-1.06/trickle-overload.c	2008-06-07 17:21:51.000000000 +0200
@@ -10,6 +10,7 @@
 /* Ick.  linux sucks. */
 #define _GNU_SOURCE
 
+#include <netdb.h>
 #include <sys/types.h>
 
 #ifdef HAVE_CONFIG_H
