--- ../../gpuPeopsSoft2/src/draw.c	2008-05-17 13:06:36.000000000 -0500
+++ draw.c	2008-07-02 07:21:23.000000000 -0500
@@ -5058,6 +5058,7 @@
 
 #ifdef USE_XF86VM
 
+#undef BOOL
 #include <X11/extensions/xf86vmode.h>
 static XF86VidModeModeInfo **modes=0;
 static int           iOldMode=0;
@@ -6374,7 +6375,13 @@
 #else
      pDstR=(unsigned long *)(pDst-(ddx+dga2Fix));
 #endif
-     for(x=0;x<ddx2;x++) *((unsigned long*)pDst)++=*pDstR++;
+     for(x=0;x<ddx2;x++)
+      {
+       *((unsigned long*)pDst) = *pDstR;
+	pDst += sizeof (unsigned long);
+	pDstR += sizeof (unsigned long);
+      }
+
     }
    else
     {
