diff -Naur gtk-sharp-2.99.3/gtk/gui-thread-check/profiler/gui-thread-check.c gtk-sharp-2.99.3.new/gtk/gui-thread-check/profiler/gui-thread-check.c
--- gtk-sharp-2.99.3/gtk/gui-thread-check/profiler/gui-thread-check.c	2014-06-05 17:06:30.000000000 +0200
+++ gtk-sharp-2.99.3.new/gtk/gui-thread-check/profiler/gui-thread-check.c	2024-04-10 10:25:38.254511449 +0200
@@ -93,6 +93,5 @@
 	
 	mono_profiler_install (NULL, NULL);
 	mono_profiler_install_enter_leave (simple_method_enter, NULL);
-	mono_profiler_set_events (MONO_PROFILE_ENTER_LEAVE);
 }
 
