diff --git a/src/ct_accel.c b/src/ct_accel.c
index a829ee3..8ec3347 100644
--- a/src/ct_accel.c
+++ b/src/ct_accel.c
@@ -52,11 +52,7 @@
 /* Our driver specific include file */
 #include "ct_driver.h"
 
-#if !defined(UNIXCPP) || defined(ANSICPP)
 #define CATNAME(prefix,subname) prefix##subname
-#else
-#define CATNAME(prefix,subname) prefix/**/subname
-#endif
 
 #ifdef CHIPS_MMIO
 #ifdef CHIPS_HIQV
diff --git a/src/ct_driver.c b/src/ct_driver.c
index 6f3a5d4..effbc60 100644
--- a/src/ct_driver.c
+++ b/src/ct_driver.c
@@ -96,9 +96,6 @@
 /* All drivers initialising the SW cursor need this */
 #include "mipointer.h"
 
-/* All drivers implementing backing store need this */
-#include "mibstore.h"
-
 /* All drivers using the mi banking wrapper need this */
 #ifdef HAVE_ISA
 #include "mibank.h"
@@ -4145,7 +4142,6 @@ CHIPSScreenInit(SCREEN_INIT_ARGS_DECL)
 	    pBankInfo = NULL;
 	    return FALSE;
 	}
-	miInitializeBackingStore(pScreen);
 	xf86SetBackingStore(pScreen);
 
 	/* Initialise cursor functions */
@@ -4296,7 +4292,6 @@ CHIPSScreenInit(SCREEN_INIT_ARGS_DECL)
 	    }
 	}
 	
-	miInitializeBackingStore(pScreen);
 	xf86SetBackingStore(pScreen);
 #ifdef ENABLE_SILKEN_MOUSE
 	xf86SetSilkenMouse(pScreen);
diff --git a/util/Makefile.am b/util/Makefile.am
index c822625..4ae900b 100644
--- a/util/Makefile.am
+++ b/util/Makefile.am
@@ -3,7 +3,7 @@ if BUILD_UTILS
 noinst_PROGRAMS = dRegs mRegs modClock memClock
 endif
 
-dRegs_SOURCES = dRegs.c AsmMacros.h
-mRegs_SOURCES = mRegs.c AsmMacros.h
-modClock_SOURCES = modClock.c AsmMacros.h
+dRegs_SOURCES = dRegs.c AsmMacros.h iopl.h
+mRegs_SOURCES = mRegs.c AsmMacros.h iopl.h
+modClock_SOURCES = modClock.c AsmMacros.h iopl.h
 memClock_SOURCES = modClock.c AsmMacros.h
