diff -Nur gnome-cups-manager-0.31/libgnomecups/gnome-cups-permission.c gnome-cups-manager-0.31.new/libgnomecups/gnome-cups-permission.c
--- gnome-cups-manager-0.31/libgnomecups/gnome-cups-permission.c	2006-05-18 12:02:58.000000000 +0200
+++ gnome-cups-manager-0.31.new/libgnomecups/gnome-cups-permission.c	2006-05-18 12:03:10.000000000 +0200
@@ -94,7 +94,8 @@
 		  gboolean force_admin, GtkWidget *toplevel)
 {
 	static char const *su_app[] = {
-		"gnomesu"
+		"gksu",
+		"--"
 	};
 	char const **args = g_new0 (char const *, argc + G_N_ELEMENTS (su_app) +
 			     1 /* app */ + 1 /* null */);
