diff -Naur gnome-cups-manager-0.31.orig/libgnomecups/snmpkit/structfill.C gnome-cups-manager-0.31/libgnomecups/snmpkit/structfill.C
--- gnome-cups-manager-0.31.orig/libgnomecups/snmpkit/structfill.C      2005-03-28 04:53:02.000000000 +0100
+++ gnome-cups-manager-0.31/libgnomecups/snmpkit/structfill.C   2008-04-17 12:53:27.000000000 +0100
@@ -28,12 +28,16 @@
 #include <string.h>
 #include <stdlib.h>
 #include <stdio.h>
-#include <iostream.h>
 #include <algorithm>

+#include <cstring>
+#include <iostream>
+
 #include "snmpkit"
 #include "oidseq.h"

+using namespace std;
+
 void SNMP_structFiller::append(const std::string &oidstr,Tags tag,
                               void (*fp)())
   throw(std::bad_alloc,BerOidBadSubOidException,BerNoOidsException){

diff -Naur gnome-cups-manager-0.31.orig/libgnomecups/snmpkit/snmpkit_interface.C gnome-cups-manager-0.31/libgnomecups/snmpkit/snmpkit_interface.C
--- gnome-cups-manager-0.31.orig/libgnomecups/snmpkit/snmpkit_interface.C       2005-06-01 15:48:36.000000000 +0100
+++ gnome-cups-manager-0.31/libgnomecups/snmpkit/snmpkit_interface.C    2008-04-17 12:54:09.000000000 +0100
@@ -30,6 +30,10 @@

 #include "errno.h"

+#include <cstring>
+
+using namespace std;
+
 SNMPERRNO sk_errno;
 char *sk_erroid;

diff -Naur gnome-cups-manager-0.31.orig/libgnomecups/snmpinter.C gnome-cups-manager-0.31/libgnomecups/snmpinter.C
--- gnome-cups-manager-0.31.orig/libgnomecups/snmpinter.C       2004-08-19 15:47:56.000000000 +0100
+++ gnome-cups-manager-0.31/libgnomecups/snmpinter.C    2008-04-17 12:56:03.000000000 +0100
@@ -9,6 +9,10 @@
 #include <iostream>
 #include <string>

+#include <cstdlib>
+
+using namespace std;
+
 static const std::string HPSTR("JETDIRECT");
 static const std::string LEXMARKSTR("Lexmark");
 static const std::string TEKTRONIXSTR("Tektronix");

