diff -Nur gnome-cups-manager-0.25.orig/gnome-cups-add/add-printer.c gnome-cups-manager-0.25/gnome-cups-add/add-printer.c
--- gnome-cups-manager-0.25.orig/gnome-cups-add/add-printer.c	2004-08-31 18:21:11.000000000 +0200
+++ gnome-cups-manager-0.25/gnome-cups-add/add-printer.c	2004-09-13 23:33:43.000000000 +0200
@@ -198,7 +198,7 @@
 
 	if (add_cups_printer (xml, uri, ppd, name)) {
 		GtkWidget *toplevel = glade_xml_get_widget (xml, "add_printer_window");
-		char const *args[] = { "-p", NULL };
+		char const *args[] = { "--properties", NULL };
 		args[1] = name;
 		gnome_cups_spawn ("gnome-cups-manager", G_N_ELEMENTS (args), args, FALSE, toplevel);
 		gtk_widget_destroy (toplevel);
diff -Nur gnome-cups-manager-0.25.orig/gnome-cups-manager/printer-properties.c gnome-cups-manager-0.25/gnome-cups-manager/printer-properties.c
--- gnome-cups-manager-0.25.orig/gnome-cups-manager/printer-properties.c	2004-09-08 20:46:49.000000000 +0200
+++ gnome-cups-manager-0.25/gnome-cups-manager/printer-properties.c	2004-09-13 23:33:19.000000000 +0200
@@ -667,7 +667,7 @@
 static void
 cb_become_admin (GnomeCupsPrinter *printer)
 {
-	char const *argv[] = { "-p", NULL };
+	char const *argv[] = { "--properties", NULL };
 	GladeXML *xml = g_object_get_data (G_OBJECT (printer), "properties-window");
 	argv[1] = gnome_cups_printer_get_name (printer);
 	if (gnome_cups_spawn ("gnome-cups-manager", G_N_ELEMENTS (argv), argv,
diff -Nur gnome-cups-manager-0.25.orig/gnome-cups-manager/tray.c gnome-cups-manager-0.25/gnome-cups-manager/tray.c
--- gnome-cups-manager-0.25.orig/gnome-cups-manager/tray.c	2004-06-24 21:10:47.000000000 +0200
+++ gnome-cups-manager-0.25/gnome-cups-manager/tray.c	2004-09-13 23:33:32.000000000 +0200
@@ -52,7 +52,7 @@
 static void 
 launch_properties (const char *printer_name)
 {
-	char *argv[] = { "gnome-cups-manager", "-p", NULL, NULL };
+	char *argv[] = { "gnome-cups-manager", "--properties", NULL, NULL };
         GError *error;
         
         error = NULL;
