diff -Naur buddi/src/org/homeunix/thecave/buddi/Const.java buddi.new/src/org/homeunix/thecave/buddi/Const.java
--- buddi/src/org/homeunix/thecave/buddi/Const.java	2012-09-09 21:21:17.000000000 +0200
+++ buddi.new/src/org/homeunix/thecave/buddi/Const.java	2012-09-10 20:56:28.446016744 +0200
@@ -20,7 +20,6 @@
 import org.homeunix.thecave.buddi.plugin.builtin.preference.NetworkPreferences;
 import org.homeunix.thecave.buddi.plugin.builtin.preference.PluginPreferences;
 import org.homeunix.thecave.buddi.plugin.builtin.preference.ViewPreferences;
-import org.homeunix.thecave.buddi.plugin.builtin.report.AccountBalance;
 import org.homeunix.thecave.buddi.plugin.builtin.report.AverageIncomeExpenseByCategory;
 import org.homeunix.thecave.buddi.plugin.builtin.report.ExpensesPieGraph;
 import org.homeunix.thecave.buddi.plugin.builtin.report.IncomeExpenseReportByCategory;
@@ -210,7 +209,6 @@
 		ExpensesPieGraph.class.getCanonicalName(),
 		NetWorthOverTime.class.getCanonicalName(),
 		Transactions.class.getCanonicalName(),
-		AccountBalance.class.getCanonicalName(),
 	};
 	
 	//Built in Imports
diff -Naur buddi/src/org/homeunix/thecave/buddi/view/dialogs/ScheduledTransactionEditorDialog.java buddi.new/src/org/homeunix/thecave/buddi/view/dialogs/ScheduledTransactionEditorDialog.java
--- buddi/src/org/homeunix/thecave/buddi/view/dialogs/ScheduledTransactionEditorDialog.java	2010-11-04 03:31:51.000000000 +0100
+++ buddi.new/src/org/homeunix/thecave/buddi/view/dialogs/ScheduledTransactionEditorDialog.java	2012-09-10 21:02:22.032484576 +0200
@@ -133,10 +133,6 @@
 		cardHolder = new JPanel(cardLayout);
 	}
 
-	protected String getType(){
-		return TextFormatter.getTranslation(BuddiKeys.ACCOUNT);
-	}
-
 	public void init() {
 		if (PrefsModel.getInstance().isShowTooltips())
 			message.setToolTipText(TextFormatter.getTranslation(BuddiKeys.TOOLTIP_SCHEDULED_MESSAGE));
