diff -aur code-browser-3.11/config/preferences.cbc code-browser-3.11-2/config/preferences.cbc
--- code-browser-3.11/config/preferences.cbc	2009-05-21 04:14:26.000000000 -0500
+++ code-browser-3.11-2/config/preferences.cbc	2009-09-12 11:20:26.000000000 -0500
@@ -2238,6 +2238,23 @@
 end
 //[cf]
 
+//[of]:FrugalBuilds
+def frugalbuild
+	prototype = configurations.default
+	patterns = FrugalBuild
+	language = sh
+	//display = default-fixed
+
+	//auto-indentation = true
+	//expand-tabulation = false
+	//tabulation-size = 8
+	//word-wrap = false
+	//elastic-tabstops = false
+
+	//layout = 0
+end
+//[cf]
+
 //[of]:Zinc
 def zinc
 	prototype = configurations.default
diff -aur code-browser-3.11/config-default/preferences.cbc code-browser-3.11-2/config-default/preferences.cbc
--- code-browser-3.11/config-default/preferences.cbc	2009-05-18 16:01:36.000000000 -0500
+++ code-browser-3.11-2/config-default/preferences.cbc	2009-09-12 11:29:44.000000000 -0500
@@ -800,6 +800,23 @@
 end
 //[cf]
 
+//[of]:FrugalBuilds
+def frugalbuild
+	//prototype = configurations.default
+	//patterns = FrugalBuild
+	//language = sh
+	//display = default-fixed
+
+	//auto-indentation = true
+	//expand-tabulation = false
+	//tabulation-size = 8
+	//word-wrap = false
+	//elastic-tabstops = false
+
+	//layout = 0
+end
+//[cf]
+
 //[of]:Zinc
 def zinc
 	//prototype = configurations.default
