diff -aur code-browser-3.14/config/preferences.cbc code-browser-3.14-2/config/preferences.cbc
--- code-browser-3.14/config/preferences.cbc	2009-12-03 15:47:17.000000000 -0600
+++ code-browser-3.14-2/config/preferences.cbc	2009-12-22 04:32:32.000000000 -0600
@@ -2300,6 +2300,23 @@
 end
 //[cf]
 
+//[of]:FrugalBuild
+def frugalbuild
+	prototype = configurations.default
+	first-line-patterns = # Compiling Time: * SBU
+	language = sh
+	//display = default-fixed
+
+	//auto-indentation = true
+	//expand-tabulation = false
+	//tabulation-size = 8
+	//word-wrap = false
+	//elastic-tabstops = false
+
+	//layout = 0
+end
+//[cf]
+
 //[of]:Zinc
 def zinc
 	prototype = configurations.default
diff -aur code-browser-3.14/config-default/preferences.cbc code-browser-3.14-2/config-default/preferences.cbc
--- code-browser-3.14/config-default/preferences.cbc	2009-10-29 12:42:54.000000000 -0500
+++ code-browser-3.14-2/config-default/preferences.cbc	2009-12-22 04:35:04.000000000 -0600
@@ -800,6 +800,23 @@
 end
 //[cf]
 
+//[of]:FrugalBuild
+def frugalbuild
+	//prototype = configurations.default
+	//first-line-patterns = # Compiling Time: * SBU
+	//language = sh
+	//display = default-fixed
+
+	//auto-indentation = true
+	//expand-tabulation = false
+	//tabulation-size = 8
+	//word-wrap = false
+	//elastic-tabstops = false
+
+	//layout = 0
+end
+//[cf]
+
 //[of]:Zinc
 def zinc
 	//prototype = configurations.default
