diff -aur code-browser-3.14/config/preferences.cbc code-browser-3.14-2/config/preferences.cbc
--- code-browser-3.14/config/preferences.cbc	2009-12-03 15:47:17.000000000 -0600
+++ code-browser-3.14-2/config/preferences.cbc	2009-12-21 14:51:36.000000000 -0600
@@ -1266,6 +1266,23 @@
 end
 //[cf]
 
+//[of]:FrugalBuilds
+def frugalbuild
+	prototype = configurations.default
+	patterns = FrugalBuild
+	language = sh
+	//display = default-fixed
+
+	//auto-indentation = true
+	//expand-tabulation = false
+	//tabulation-size = 8
+	//word-wrap = false
+	//elastic-tabstops = false
+
+	//layout = 0
+end
+//[cf]
+
 //[of]:zinc
 def zinc
 	colorizer = zinc
diff -aur code-browser-3.14/config-default/preferences.cbc code-browser-3.14-2/config-default/preferences.cbc
--- code-browser-3.14/config-default/preferences.cbc	2009-10-29 12:42:54.000000000 -0500
+++ code-browser-3.14-2/config-default/preferences.cbc	2009-12-21 14:51:43.000000000 -0600
@@ -800,6 +800,23 @@
 end
 //[cf]
 
+//[of]:FrugalBuilds
+def frugalbuild
+	//prototype = configurations.default
+	//patterns = FrugalBuild
+	//language = sh
+	//display = default-fixed
+
+	//auto-indentation = true
+	//expand-tabulation = false
+	//tabulation-size = 8
+	//word-wrap = false
+	//elastic-tabstops = false
+
+	//layout = 0
+end
+//[cf]
+
 //[of]:Zinc
 def zinc
 	//prototype = configurations.default
