diff -Naur rapidsvn-0.9.4/src/add_action.cpp rapidsvn-0.9.4-wx28/src/add_action.cpp
--- rapidsvn-0.9.4/src/add_action.cpp	2006-12-08 18:47:07.000000000 +0100
+++ rapidsvn-0.9.4-wx28/src/add_action.cpp	2007-12-07 17:26:17.000000000 +0100
@@ -23,12 +23,12 @@
  * ====================================================================
  */
 
-// svncpp
-#include "svncpp/client.hpp"
-
 // wxWidgets
 #include "wx/intl.h"
 
+// svncpp
+#include "svncpp/client.hpp"
+
 // app
 #include "ids.hpp"
 #include "add_action.hpp"
diff -Naur rapidsvn-0.9.4/src/checkout_dlg.cpp rapidsvn-0.9.4-wx28/src/checkout_dlg.cpp
--- rapidsvn-0.9.4/src/checkout_dlg.cpp	2006-12-08 18:47:07.000000000 +0100
+++ rapidsvn-0.9.4-wx28/src/checkout_dlg.cpp	2007-12-07 17:26:17.000000000 +0100
@@ -256,7 +256,6 @@
 
 BEGIN_EVENT_TABLE (CheckoutDlg, wxDialog)
   EVT_BUTTON (ID_BUTTON_BROWSE, CheckoutDlg::OnBrowse)
-  EVT_BUTTON (wxID_OK, CheckoutDlg::OnOK)
   EVT_CHECKBOX (ID_USELATEST, CheckoutDlg::OnUseLatest)
   EVT_TEXT (-1, CheckoutDlg::OnText)
 //  EVT_BUTTON (wxID_HELP, CheckoutDlg::OnHelp)
diff -Naur rapidsvn-0.9.4/src/export_dlg.cpp rapidsvn-0.9.4-wx28/src/export_dlg.cpp
--- rapidsvn-0.9.4/src/export_dlg.cpp	2006-12-08 18:47:07.000000000 +0100
+++ rapidsvn-0.9.4-wx28/src/export_dlg.cpp	2007-12-07 17:26:17.000000000 +0100
@@ -295,7 +295,6 @@
 
 BEGIN_EVENT_TABLE (ExportDlg, wxDialog)
   EVT_BUTTON (ID_BUTTON_BROWSE, ExportDlg::OnBrowse)
-  EVT_BUTTON (wxID_OK, ExportDlg::OnOK)
   EVT_CHECKBOX (ID_USE_LATEST, ExportDlg::OnUseLatest)
   EVT_CHECKBOX (ID_NOT_SPECIFIED, ExportDlg::OnNotSpecified)
   EVT_COMBOBOX (ID_NATIVE_EOL, ExportDlg::OnNativeEol)
diff -Naur rapidsvn-0.9.4/src/import_dlg.cpp rapidsvn-0.9.4-wx28/src/import_dlg.cpp
--- rapidsvn-0.9.4/src/import_dlg.cpp	2006-12-08 18:47:07.000000000 +0100
+++ rapidsvn-0.9.4-wx28/src/import_dlg.cpp	2007-12-07 17:26:17.000000000 +0100
@@ -196,7 +196,11 @@
     }
   }
 
-  wxDialog::OnOK(event);
+#if wxCHECK_VERSION (2, 7, 0)
+  wxDialog::EndModal (wxID_OK);
+#else
+  wxDialog::OnOK (event);
+#endif
 }
 
 void
diff -Naur rapidsvn-0.9.4/src/listener.cpp rapidsvn-0.9.4-wx28/src/listener.cpp
--- rapidsvn-0.9.4/src/listener.cpp	2006-12-08 18:47:07.000000000 +0100
+++ rapidsvn-0.9.4-wx28/src/listener.cpp	2007-12-07 17:26:17.000000000 +0100
@@ -253,7 +253,7 @@
   void 
   handleEvent (wxCommandEvent & event)
   {
-    switch (event.m_id)
+    switch (event.GetId ())
     {
     case SIG_GET_LOG_MSG:
       callbackGetLogMessage ();
diff -Naur rapidsvn-0.9.4/src/log_dlg.cpp rapidsvn-0.9.4-wx28/src/log_dlg.cpp
--- rapidsvn-0.9.4/src/log_dlg.cpp	2006-12-08 18:47:07.000000000 +0100
+++ rapidsvn-0.9.4-wx28/src/log_dlg.cpp	2007-12-07 17:26:17.000000000 +0100
@@ -32,6 +32,7 @@
 #include "wx/valgen.h"
 #include "wx/listctrl.h"
 #include "wx/splitter.h"
+#include "wx/notebook.h"
 
 
 // svncpp
diff -Naur rapidsvn-0.9.4/src/merge_dlg.cpp rapidsvn-0.9.4-wx28/src/merge_dlg.cpp
--- rapidsvn-0.9.4/src/merge_dlg.cpp	2006-12-08 18:47:07.000000000 +0100
+++ rapidsvn-0.9.4-wx28/src/merge_dlg.cpp	2007-12-07 17:26:17.000000000 +0100
@@ -113,7 +113,11 @@
     }
   }
 
-  wxDialog::OnOK(event);
+#if wxCHECK_VERSION (2, 7, 0)
+  wxDialog::EndModal (wxID_OK);
+#else
+  wxDialog::OnOK (event);
+#endif  
 }
 
 void
diff -Naur rapidsvn-0.9.4/src/preferences_dlg.cpp rapidsvn-0.9.4-wx28/src/preferences_dlg.cpp
--- rapidsvn-0.9.4/src/preferences_dlg.cpp	2006-12-08 18:47:07.000000000 +0100
+++ rapidsvn-0.9.4-wx28/src/preferences_dlg.cpp	2007-12-07 17:26:17.000000000 +0100
@@ -114,9 +114,14 @@
    */
   bool SelectExecutable (const wxString & title, wxTextCtrl * textCtrl)
   {
+#if wxCHECK_VERSION (2, 7, 0)
+    wxFileDialog dlg (this, title, wxEmptyString, wxEmptyString, _("*.*"), wxFD_OPEN);
+#else
     wxFileDialog dlg (this, title);
 
     dlg.SetStyle (wxHIDE_READONLY | wxOPEN);
+#endif
+
     dlg.SetWildcard (EXECUTABLE_WILDCARD);
     dlg.SetPath (textCtrl->GetValue ());
 
diff -Naur rapidsvn-0.9.4/src/rapidsvn_frame.cpp rapidsvn-0.9.4-wx28/src/rapidsvn_frame.cpp
--- rapidsvn-0.9.4/src/rapidsvn_frame.cpp	2006-12-08 18:47:07.000000000 +0100
+++ rapidsvn-0.9.4-wx28/src/rapidsvn_frame.cpp	2007-12-07 17:26:17.000000000 +0100
@@ -1150,7 +1150,7 @@
 void
 RapidSvnFrame::OnColumn (wxCommandEvent & event)
 {
-  int eventId = event.m_id;
+  int eventId = event.GetId ();
   int col = 0;
   while (col < FileListCtrl::COL_COUNT &&
          m->ColumnList[col].id != eventId)
@@ -1192,7 +1192,7 @@
 RapidSvnFrame::OnColumnSorting (wxCommandEvent & event)
 {
   // we dont want to list FileListCtrl::COL_NAME/COL_PATH/... here
-  int col = event.m_id - ID_ColumnSort_Name;
+  int col = event.GetId () - ID_ColumnSort_Name;
 
   m->listCtrl->SetSortColumn (col);
   m->listCtrl->SetSortAscending (true);
@@ -1294,7 +1294,7 @@
 void
 RapidSvnFrame::OnUpdateCommand (wxUpdateUIEvent & updateUIEvent)
 {
-  updateUIEvent.Enable (ValidateIDActionFlags (updateUIEvent.m_id, GetSelectionActionFlags ()));
+  updateUIEvent.Enable (ValidateIDActionFlags (updateUIEvent.GetId (), GetSelectionActionFlags ()));
 }
 
 void
@@ -1514,13 +1514,13 @@
 {
   Action* action = NULL;
 
-  if ((event.m_id >= ID_Verb_Min) && (event.m_id <= ID_Verb_Max))
+  if ((event.GetId () >= ID_Verb_Min) && (event.GetId () <= ID_Verb_Max))
   {
-    action = new ExternalProgramAction (this, event.m_id - ID_Verb_Min, false);
+    action = new ExternalProgramAction (this, event.GetId () - ID_Verb_Min, false);
   }
   else
   {
-    switch (event.m_id)
+    switch (event.GetId ())
     {
     case ID_Explore:
       action = new ExternalProgramAction (this, -1, true);
diff -Naur rapidsvn-0.9.4/src/resolve_action.cpp rapidsvn-0.9.4-wx28/src/resolve_action.cpp
--- rapidsvn-0.9.4/src/resolve_action.cpp	2006-12-08 18:47:07.000000000 +0100
+++ rapidsvn-0.9.4-wx28/src/resolve_action.cpp	2007-12-07 17:26:17.000000000 +0100
@@ -23,12 +23,12 @@
  * ====================================================================
  */
 
-// svncpp
-#include "svncpp/client.hpp"
-
 // wxWidgets
 #include "wx/intl.h"
 
+// svncpp
+#include "svncpp/client.hpp"
+
 // app
 #include "resolve_action.hpp"
 
