diff -urN xfce-utils-4.3.99.1.orig/scripts/xinitrc xfce-utils-4.3.99.1/scripts/xinitrc
--- xfce-utils-4.3.99.1.orig/scripts/xinitrc	2006-09-23 10:56:19.000000000 +0100
+++ xfce-utils-4.3.99.1/scripts/xinitrc	2006-09-23 11:01:29.000000000 +0100
@@ -27,11 +27,11 @@
 else
   # Those are fallback settings, use the ui plugin to change it
   # or add your overrides to ~/.Xresources
-  # Xft DPI: 96
+  # Xft DPI: 72
   # Xft.hintstyle: hintnone/hintslight/hintmedium/hintfull
   # Xft hinting: 1/0
   cat >> $XRESOURCES << EOF
-Xft.dpi: 96
+Xft.dpi: 72
 Xft.hinting: 1
 Xft.hintstyle: hintmedium
 EOF
