--- pandoc-lua-marshal-0.2.2/pandoc-lua-marshal.cabal	2001-09-09 03:46:40.000000000 +0200
+++ pandoc-lua-marshal-0.2.2.new/pandoc-lua-marshal.cabal	2023-07-22 21:22:33.448202311 +0200
@@ -46,7 +46,7 @@
 
 common common-options
   build-depends:       base                  >= 4.12     && < 5
-                     , aeson                 >= 1.5      && < 2.2
+                     , aeson                 >= 1.5      && < 2.3
                      , bytestring            >= 0.10     && < 0.12
                      , containers            >= 0.6      && < 0.7
                      , exceptions            >= 0.8      && < 0.11
