--- browsers/media.py	2006-10-09 01:34:00.000000000 +0200
+++ browsers/media.py.new	2007-01-09 21:32:43.000000000 +0100
@@ -143,12 +143,15 @@
 
     @classmethod
     def init(klass, library):
-        devices._hal.connect_to_signal(
-            'DeviceAdded', klass.__hal_device_added)
-        devices._hal.connect_to_signal(
-            'DeviceRemoved', klass.__hal_device_removed)
-        for udi in devices.discover():
-            klass.__hal_device_added(udi)
+	try:
+		devices._hal.connect_to_signal(
+				'DeviceAdded', klass.__hal_device_added)
+	        devices._hal.connect_to_signal(
+        	    'DeviceRemoved', klass.__hal_device_removed)
+	        for udi in devices.discover():
+        	    klass.__hal_device_added(udi)
+	except:
+		print "Hal not found"
 
     @classmethod
     def devices(klass):
