diff --git a/bcore/logger/logger.cpp b/bcore/logger/logger.cpp
index 25caccb..dbc8a0a 100644
--- a/bcore/logger/logger.cpp
+++ b/bcore/logger/logger.cpp
@@ -20,6 +20,8 @@
  * $Id: logger.cpp 119 2008-01-20 22:18:33Z wojci $
  */
 
+#include <stdio.h>
+
 #include "logger.h"
 #include "logif.h"
 
diff --git a/bcore/os/fileop.cpp b/bcore/os/fileop.cpp
index e0ba110..df16700 100644
--- a/bcore/os/fileop.cpp
+++ b/bcore/os/fileop.cpp
@@ -118,7 +118,7 @@ namespace btg
          {
             bool status = true;
 
-            if (::remove(_filename.c_str()) == -1)
+            if (remove(_filename.c_str()) == -1)
                {
                   status = false;
                }
diff --git a/bcore/os/socket.cpp b/bcore/os/socket.cpp
index 58ecc0c..4b20fad 100644
--- a/bcore/os/socket.cpp
+++ b/bcore/os/socket.cpp
@@ -32,6 +32,7 @@ extern "C"
 {
 #include <errno.h>
 #include <fcntl.h>
+#include <stdio.h>
 }
 
 namespace btg
