--- ./bcore/auth/hash.cpp.orig	2008-03-17 16:16:47.000000000 +0100
+++ ./bcore/auth/hash.cpp	2008-03-17 16:17:11.000000000 +0100
@@ -23,6 +23,7 @@
 #include "hash.h"
 
 #include <sstream>
+#include <cstring>
 #include <iomanip>
 
 #include <bcore/helpermacro.h>
--- ./bcore/os/socket.h.orig	2008-03-17 16:19:29.000000000 +0100
+++ ./bcore/os/socket.h	2008-03-17 16:19:37.000000000 +0100
@@ -35,6 +35,7 @@
 #include <string>
 #include <map>
 #include <vector>
+#include <cstring>
 
 extern "C"
 {
--- ./bcore/os/stdin.h.orig	2008-03-17 16:20:53.000000000 +0100
+++ ./bcore/os/stdin.h	2008-03-17 16:21:01.000000000 +0100
@@ -24,6 +24,7 @@
 #define BTG_STDIN_H
 
 #include <string>
+#include <cstring>
 
 namespace btg
 {
--- ./external/ini/cdatafile.cpp.orig	2008-03-17 15:53:07.000000000 +0100
+++ ./external/ini/cdatafile.cpp	2008-03-17 16:31:07.000000000 +0100
@@ -37,6 +37,9 @@
 
 #include <vector>
 #include <string>
+#include <cstring>
+#include <cstdlib>
+#include <climits>
 #include <ctype.h>
 #include <stdio.h>
 #include <stdarg.h>
