diff -Naur rt-8u202-ga/modules/media/src/main/native/gstreamer/3rd_party/glib/glib-2.56.1/glib/gstrfuncs.c rt-8u202-ga.new/modules/media/src/main/native/gstreamer/3rd_party/glib/glib-2.56.1/glib/gstrfuncs.c
--- rt-8u202-ga/modules/media/src/main/native/gstreamer/3rd_party/glib/glib-2.56.1/glib/gstrfuncs.c	2018-12-10 17:30:22.000000000 +0100
+++ rt-8u202-ga.new/modules/media/src/main/native/gstreamer/3rd_party/glib/glib-2.56.1/glib/gstrfuncs.c	2020-05-14 16:16:28.252461707 +0200
@@ -37,11 +37,6 @@
 #include <errno.h>
 #include <ctype.h>              /* For tolower() */
 
-#ifdef HAVE_XLOCALE_H
-/* Needed on BSD/OS X for e.g. strtod_l */
-#include <xlocale.h>
-#endif
-
 #ifdef G_OS_WIN32
 #include <windows.h>
 #endif
diff -Naur rt-8u202-ga/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.h rt-8u202-ga.new/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.h
--- rt-8u202-ga/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.h	2018-12-10 17:30:22.000000000 +0100
+++ rt-8u202-ga.new/modules/web/src/main/native/Source/ThirdParty/libxslt/src/libxslt/xsltlocale.h	2020-05-14 16:16:28.252461707 +0200
@@ -25,9 +25,6 @@
 #ifdef HAVE_LOCALE_H
 #include <locale.h>
 #endif
-#ifdef HAVE_XLOCALE_H
-#include <xlocale.h>
-#endif
 
 typedef locale_t xsltLocale;
 typedef xmlChar xsltLocaleChar;
