diff -Naur jfx-19.0.2-1/modules/javafx.web/src/main/native/Source/JavaScriptCore/offlineasm/config.rb jfx-19.0.2-1.new/modules/javafx.web/src/main/native/Source/JavaScriptCore/offlineasm/config.rb
--- jfx-19.0.2-1/modules/javafx.web/src/main/native/Source/JavaScriptCore/offlineasm/config.rb	2023-01-17 17:08:56.000000000 +0100
+++ jfx-19.0.2-1.new/modules/javafx.web/src/main/native/Source/JavaScriptCore/offlineasm/config.rb	2023-01-19 10:45:17.086069002 +0100
@@ -22,11 +22,11 @@
 # THE POSSIBILITY OF SUCH DAMAGE.
 
 buildProductsDirectory = ENV['BUILT_PRODUCTS_DIR'];
-if buildProductsDirectory and File.exists?(buildProductsDirectory)
+if buildProductsDirectory and File.exist?(buildProductsDirectory)
     $: << "#{buildProductsDirectory}/usr/local/include/WebKitAdditions/Scripts"
 end
 sdkRootDirectory = ENV['SDKROOT'];
-if sdkRootDirectory and File.exists?(sdkRootDirectory)
+if sdkRootDirectory and File.exist?(sdkRootDirectory)
     $: << "#{sdkRootDirectory}/usr/local/include/WebKitAdditions/Scripts"
 end
 
diff -Naur jfx-19.0.2-1/modules/javafx.web/src/main/native/Source/JavaScriptCore/offlineasm/parser.rb jfx-19.0.2-1.new/modules/javafx.web/src/main/native/Source/JavaScriptCore/offlineasm/parser.rb
--- jfx-19.0.2-1/modules/javafx.web/src/main/native/Source/JavaScriptCore/offlineasm/parser.rb	2023-01-17 17:08:56.000000000 +0100
+++ jfx-19.0.2-1.new/modules/javafx.web/src/main/native/Source/JavaScriptCore/offlineasm/parser.rb	2023-01-19 10:45:17.106067853 +0100
@@ -832,10 +832,10 @@
                 @idx += 1
                 additionsDirectoryName = "#{@buildProductsDirectory}/usr/local/include/WebKitAdditions/"
                 fileName = IncludeFile.new(moduleName, additionsDirectoryName).fileName
-                if not File.exists?(fileName)
+                if not File.exist?(fileName)
                     fileName = IncludeFile.new(moduleName, @tokens[@idx].codeOrigin.fileName.dirname).fileName
                 end
-                fileExists = File.exists?(fileName)
+                fileExists = File.exist?(fileName)
                 raise "File not found: #{fileName}" if not fileExists and not isOptional
                 list << parse(fileName) if fileExists
             else
@@ -864,10 +864,10 @@
                 @idx += 1
                 additionsDirectoryName = "#{@buildProductsDirectory}/usr/local/include/WebKitAdditions/"
                 fileName = IncludeFile.new(moduleName, additionsDirectoryName).fileName
-                if not File.exists?(fileName)
+                if not File.exist?(fileName)
                     fileName = IncludeFile.new(moduleName, @tokens[@idx].codeOrigin.fileName.dirname).fileName
                 end
-                fileExists = File.exists?(fileName)
+                fileExists = File.exist?(fileName)
                 raise "File not found: #{fileName}" if not fileExists and not isOptional
                 fileList << fileName if fileExists
             else
