diff -Nur evolution-2.10.0/calendar/gui/e-cal-component-memo-preview.c evolution-2.10.0.new/calendar/gui/e-cal-component-memo-preview.c
--- evolution-2.10.0/calendar/gui/e-cal-component-memo-preview.c	2007-01-03 07:34:53.000000000 -0800
+++ evolution-2.10.0.new/calendar/gui/e-cal-component-memo-preview.c	2007-03-22 16:10:33.402019507 -0700
@@ -185,7 +185,7 @@
 			}
 		}
 		
-		gtk_html_stream_printf(stream, string->str);
+		gtk_html_stream_printf(stream, "%s", string->str);
 
 		g_string_free (string, TRUE);
 	
