diff -NurdB evolution-2.3.8/widgets/misc/e-canvas-background.c evolution-2.3.8-patched/widgets/misc/e-canvas-background.c
--- evolution-2.3.8/widgets/misc/e-canvas-background.c	2005-08-22 05:52:55.000000000 -0500
+++ evolution-2.3.8-patched/widgets/misc/e-canvas-background.c	2005-09-01 23:13:54.000000000 -0500
@@ -427,6 +427,7 @@
 ecb_style_set (ECanvasBackground *ecb, GtkStyle *previous_style)
 {
 	GnomeCanvasItem *item = GNOME_CANVAS_ITEM (ecb);
+	g_return_if_fail (ecb->priv->gc != NULL);
 	gdk_gc_set_foreground (ecb->priv->gc, &GTK_WIDGET(item->canvas)->style->base[GTK_STATE_NORMAL]);
 	gnome_canvas_item_request_update (GNOME_CANVAS_ITEM (ecb));
 }
