--- evolution-2.7.2.1/shell/e-shell-window.c.quit-resize-crash	2006-05-29 17:03:30.000000000 -0400
+++ evolution-2.7.2.1/shell/e-shell-window.c	2006-05-29 17:05:48.000000000 -0400
@@ -857,6 +857,7 @@  impl_finalize (GObject *object)
 
 	g_object_unref(priv->menu);
 
+	E_SHELL_WINDOW (object)->priv = NULL;
 	g_free (priv);
 
 	(* G_OBJECT_CLASS (e_shell_window_parent_class)->finalize) (object);
@@ -906,7 +907,8 @@  store_window_size (GtkWidget* widget)
 			      widget->allocation.height, NULL);
 	g_object_unref(client);
 
-	E_SHELL_WINDOW (widget)->priv->store_window_size_timer = 0;
+	if (E_SHELL_WINDOW (widget)->priv)
+		E_SHELL_WINDOW (widget)->priv->store_window_size_timer = 0;
 	return FALSE; // remove this timeout
 }
 
