diff -Naur evolution-2.2.3.orig/configure.in evolution-2.2.3/configure.in
--- evolution-2.2.3.orig/configure.in	2005-06-17 14:12:29.000000000 +0000
+++ evolution-2.2.3/configure.in	2005-06-17 14:15:00.000000000 +0000
@@ -528,7 +528,7 @@
 AC_ARG_ENABLE(pilot-conduits, 
 [  --enable-pilot-conduits=[no/yes] Enable support for building pilot conduits.],,enable_pilot_conduits=no)
 if test "x$enable_pilot_conduits" = "xyes"; then
-	PKG_CHECK_MODULES(GNOME_PILOT, gnome-pilot-2.0)
+	PKG_CHECK_MODULES(GNOME_PILOT, gnome-pilot-2.0 glib-2.0 gnome-vfs-2.0 libgnome-2.0 ORBit-2.0 gtk+-2.0)
 
 	CFLAGS_save="$CFLAGS"
 	CFLAGS="$CFLAGS $GNOME_PILOT_CFLAGS"
