--- evolution-2.7.92/widgets/table/e-tree-memory.c.deprecated-glib	2005-06-17 11:20:34.000000000 -0400
+++ evolution-2.7.92/widgets/table/e-tree-memory.c	2006-08-23 10:41:16.000000000 -0400
@@ -38,10 +38,7 @@
 
 #include "e-tree-memory.h"
 
-#define TREEPATH_CHUNK_AREA_SIZE (30 * sizeof (ETreeMemoryPath))
-
 static ETreeModelClass *parent_class;
-static GMemChunk  *node_chunk;
 
 enum {
 	FILL_IN_CHILDREN,
@@ -358,8 +355,6 @@
 
 	parent_class                     = g_type_class_peek_parent (klass);
 	
-	node_chunk                       = g_mem_chunk_create (ETreeMemoryPath, TREEPATH_CHUNK_AREA_SIZE, G_ALLOC_AND_FREE);
-
 	signals [FILL_IN_CHILDREN] =
 		g_signal_new ("fill_in_children",
 			      E_OBJECT_CLASS_TYPE (object_class),
@@ -509,7 +504,7 @@
 	if (!tree_model->priv->frozen)
 		e_tree_model_pre_change(E_TREE_MODEL(tree_model));
 
-	new_path = g_chunk_new0 (ETreeMemoryPath, node_chunk);
+	new_path = g_slice_new0 (ETreeMemoryPath);
 
 	new_path->node_data = node_data;
 	new_path->children_computed = FALSE;
@@ -584,7 +579,7 @@
 		etree->priv->destroy_func (node->node_data, etree->priv->destroy_user_data);
 	}
 
-	g_chunk_free(node, node_chunk);
+	g_slice_free (ETreeMemoryPath, node);
 }
 
 /**
--- evolution-2.7.92/widgets/table/e-tree-sorted.c.deprecated-glib	2005-06-17 11:20:34.000000000 -0400
+++ evolution-2.7.92/widgets/table/e-tree-sorted.c	2006-08-23 10:41:16.000000000 -0400
@@ -47,12 +47,9 @@
 /* maximum insertions between an idle event that we will do without scheduling an idle sort */
 #define ETS_INSERT_MAX (4)
 
-#define TREEPATH_CHUNK_AREA_SIZE (30 * sizeof (ETreeSortedPath))
-
 #define d(x)
 
 static ETreeModel *parent_class;
-static GMemChunk  *node_chunk;
 
 enum {
 	NODE_RESORTED,
@@ -365,7 +362,7 @@
 free_path (ETreeSortedPath *path)
 {
 	free_children(path);
-	g_chunk_free(path, node_chunk);
+	g_slice_free(ETreeSortedPath, path);
 }
 
 static ETreeSortedPath *
@@ -373,7 +370,7 @@
 {
 	ETreeSortedPath *path;
 
-	path = g_chunk_new0 (ETreeSortedPath, node_chunk);
+	path = g_slice_new0 (ETreeSortedPath);
 
 	path->corresponding = corresponding;
 	path->parent = parent;
@@ -1157,8 +1154,6 @@
 
 	parent_class                     = g_type_class_peek_parent (klass);
 
-	node_chunk                       = g_mem_chunk_create (ETreeSortedPath, TREEPATH_CHUNK_AREA_SIZE, G_ALLOC_AND_FREE);
-
 	klass->node_resorted             = NULL;
 	
 	object_class->dispose            = ets_dispose;
--- evolution-2.7.92/calendar/gui/e-meeting-list-view.c.deprecated-glib	2006-04-22 00:39:20.000000000 -0400
+++ evolution-2.7.92/calendar/gui/e-meeting-list-view.c	2006-08-23 10:41:16.000000000 -0400
@@ -786,7 +786,7 @@
 					g_slist_free (*la);
 					*la = NULL;
 				} else
-					*la = g_slist_remove_link (*la, g_slist_find_custom (*la, attendee, (GCompareFunc)g_strcasecmp));
+					*la = g_slist_remove_link (*la, g_slist_find_custom (*la, attendee, (GCompareFunc)g_ascii_strcasecmp));
 			}
 		}
 
