--- evolution-2.7.92/a11y/widgets/ea-calendar-cell.c.chain-finalize	2005-06-17 11:20:24.000000000 -0400
+++ evolution-2.7.92/a11y/widgets/ea-calendar-cell.c	2006-08-30 13:43:58.000000000 -0400
@@ -203,6 +203,8 @@
 #ifdef ACC_DEBUG
 static void ea_calendar_cell_finalize (GObject *object)
 {
+        G_OBJECT_CLASS (parent_class)->finalize (object);
+
 	++n_ea_calendar_cell_destroyed;
 	g_print ("ACC_DEBUG: n_ea_calendar_cell_destroyed = %d\n",
 		n_ea_calendar_cell_destroyed);
--- evolution-2.7.92/a11y/calendar/ea-week-view-cell.c.chain-finalize	2004-09-30 03:32:36.000000000 -0400
+++ evolution-2.7.92/a11y/calendar/ea-week-view-cell.c	2006-08-30 13:43:58.000000000 -0400
@@ -190,6 +190,8 @@
 #ifdef ACC_DEBUG
 static void ea_week_view_cell_finalize (GObject *object)
 {
+        G_OBJECT_CLASS (parent_class)->finalize (object);
+
 	++n_ea_week_view_cell_destroyed;
 	printf ("ACC_DEBUG: n_ea_week_view_cell_destroyed = %d\n",
 		n_ea_week_view_cell_destroyed);
--- evolution-2.7.92/a11y/calendar/ea-day-view-cell.c.chain-finalize	2004-09-30 03:32:36.000000000 -0400
+++ evolution-2.7.92/a11y/calendar/ea-day-view-cell.c	2006-08-30 13:43:58.000000000 -0400
@@ -190,6 +190,8 @@
 #ifdef ACC_DEBUG
 static void ea_day_view_cell_finalize (GObject *object)
 {
+	G_OBJECT_CLASS (parent_class)->finalize (object);
+
 	++n_ea_day_view_cell_destroyed;
 	printf ("ACC_DEBUG: n_ea_day_view_cell_destroyed = %d\n",
 		n_ea_day_view_cell_destroyed);
--- evolution-2.7.92/a11y/calendar/ea-cal-view-event.c.chain-finalize	2006-08-19 01:49:06.000000000 -0400
+++ evolution-2.7.92/a11y/calendar/ea-cal-view-event.c	2006-08-30 13:43:58.000000000 -0400
@@ -156,6 +156,8 @@
 #ifdef ACC_DEBUG
 static void ea_cal_view_finalize (GObject *object)
 {
+	G_OBJECT_CLASS (parent_class)->finalize (object);
+
 	++n_ea_cal_view_event_destroyed;
 	printf ("ACC_DEBUG: n_ea_cal_view_event_destroyed = %d\n",
 		n_ea_cal_view_event_destroyed);
--- evolution-2.7.92/a11y/e-table/gal-a11y-e-cell-registry.c.chain-finalize	2005-04-29 10:13:45.000000000 -0400
+++ evolution-2.7.92/a11y/e-table/gal-a11y-e-cell-registry.c	2006-08-30 13:43:58.000000000 -0400
@@ -28,6 +28,8 @@
 
 	g_hash_table_destroy (registry->priv->table);
 	g_free (registry->priv);
+
+	G_OBJECT_CLASS (parent_class)->finalize (obj);
 }
 
 static void
--- evolution-2.7.92/plugins/groupwise-account-setup/camel-gw-listener.c.chain-finalize	2006-08-30 15:10:59.000000000 -0400
+++ evolution-2.7.92/plugins/groupwise-account-setup/camel-gw-listener.c	2006-08-30 15:11:06.000000000 -0400
@@ -118,6 +118,8 @@
 	}
 	
 	g_list_free (groupwise_accounts);
+
+	G_OBJECT_CLASS (parent_class)->finalize (object);
 }
 
 /*determines whehter the passed in account is groupwise or not by looking at source url */
--- evolution-2.7.92/plugins/hula-account-setup/camel-hula-listener.c.chain-finalize	2006-01-27 11:33:56.000000000 -0500
+++ evolution-2.7.92/plugins/hula-account-setup/camel-hula-listener.c	2006-08-30 13:43:58.000000000 -0400
@@ -116,6 +116,8 @@
 	}
 	
 	g_list_free (hula_accounts);
+
+	G_OBJECT_CLASS (parent_class)->finalize (object);
 }
 
 /*determines whehter the passed in account is hula or not by looking at source url */
