--- evolution-2.8.0/mail/em-folder-view.c.fix-missing-declarations	2006-09-02 02:43:04.000000000 -0400
+++ evolution-2.8.0/mail/em-folder-view.c	2006-09-05 06:48:18.000000000 -0400
@@ -69,6 +69,8 @@
 #include <gtkhtml/gtkhtml.h>
 #include <gtkhtml/gtkhtml-stream.h>
 
+#include <libedataserver/e-data-server-util.h>
+
 #include "menus/gal-view-etable.h"
 #include "menus/gal-view-factory-etable.h"
 #include "menus/gal-view-instance.h"
--- evolution-2.8.0/mail/em-format-html-display.c.fix-missing-declarations	2006-09-02 02:43:04.000000000 -0400
+++ evolution-2.8.0/mail/em-format-html-display.c	2006-09-05 06:48:18.000000000 -0400
@@ -29,6 +29,7 @@
 #include <glib.h>
 #include <glib/gstdio.h>
 #include <gdk/gdkkeysyms.h>
+#include <gtk/gtk.h>
 
 #ifdef G_OS_WIN32
 /* Work around 'DATADIR' and 'interface' lossage in <windows.h> */
@@ -42,27 +43,6 @@
 #include <gtkhtml/gtkhtml-embedded.h>
 #include <gtkhtml/gtkhtml-search.h>
 
-#include <gtk/gtkeventbox.h>
-#include <gtk/gtkvbox.h>
-#include <gtk/gtkhbox.h>
-#include <gtk/gtkbutton.h>
-#include <gtk/gtkstock.h>
-#include <gtk/gtkimage.h>
-#include <gtk/gtkarrow.h>
-#include <gtk/gtklabel.h>
-#include <gtk/gtkentry.h>
-#include <gtk/gtkscrolledwindow.h>
-#include <gtk/gtktextview.h>
-#include <gtk/gtktogglebutton.h>
-#include <gtk/gtktable.h>
-#include <gtk/gtkmenu.h>
-#include <gtk/gtkmenuitem.h>
-#include <gtk/gtkmain.h>
-#include <gtk/gtkdnd.h>
-#include <gtk/gtktoolbutton.h>
-#include <gtk/gtkframe.h>
-#include <gtk/gtkcheckbutton.h>
-
 #include <glade/glade.h>
 
 #include <libgnomevfs/gnome-vfs-mime-handlers.h>
--- evolution-2.8.0/a11y/calendar/ea-gnome-calendar.c.fix-missing-declarations	2004-10-29 05:56:35.000000000 -0400
+++ evolution-2.8.0/a11y/calendar/ea-gnome-calendar.c	2006-09-05 06:47:11.000000000 -0400
@@ -28,6 +28,7 @@
 #include <string.h>
 #include <gtk/gtknotebook.h>
 #include <libecal/e-cal-time-util.h>
+#include <libedataserver/e-data-server-util.h>
 #include <libgnome/gnome-i18n.h>
 
 static void ea_gnome_calendar_class_init (EaGnomeCalendarClass *klass);
--- evolution-2.8.0/widgets/misc/e-attachment-bar.h.fix-missing-declarations	2006-05-31 23:30:06.000000000 -0400
+++ evolution-2.8.0/widgets/misc/e-attachment-bar.h	2006-09-05 06:48:18.000000000 -0400
@@ -82,6 +82,7 @@
 GSList *e_attachment_bar_get_selected (EAttachmentBar *bar);
 void e_attachment_bar_set_width(EAttachmentBar *bar, int bar_width);
 GSList * e_attachment_bar_get_all_attachments (EAttachmentBar *bar);
+void e_attachment_bar_create_attachment_cache (EAttachment *attachment);
 
 #ifdef __cplusplus
 }
--- evolution-2.8.0/e-util/e-util.h.fix-missing-declarations	2006-09-02 02:42:18.000000000 -0400
+++ evolution-2.8.0/e-util/e-util.h	2006-09-05 06:47:11.000000000 -0400
@@ -27,6 +27,7 @@
 #include <sys/types.h>
 #include <glib-object.h>
 #include <limits.h>
+#include <cairo/cairo.h>
 #include <gconf/gconf-client.h>
 #include <cairo.h>
 
@@ -217,6 +218,8 @@
 									    ...);
 cairo_font_options_t * get_font_options ();
 
+cairo_font_options_t * get_font_options ();
+
 #ifdef __cplusplus
 }
 #endif /* __cplusplus */
--- evolution-2.8.0/addressbook/gui/widgets/e-minicard.c.fix-missing-declarations	2006-07-22 07:14:49.000000000 -0400
+++ evolution-2.8.0/addressbook/gui/widgets/e-minicard.c	2006-09-05 06:48:18.000000000 -0400
@@ -39,6 +39,7 @@
 #include "e-minicard-label.h"
 #include "e-minicard-view.h"
 #include "e-contact-editor.h"
+#include <e-util/e-html-utils.h>
 #include <e-util/e-icon-factory.h>
 #include <libebook/e-destination.h>
 #include "a11y/addressbook/ea-addressbook.h"
--- evolution-2.8.0/composer/e-msg-composer.c.fix-missing-declarations	2006-09-05 00:11:45.000000000 -0400
+++ evolution-2.8.0/composer/e-msg-composer.c	2006-09-05 06:48:18.000000000 -0400
@@ -61,6 +61,7 @@
 #include <gconf/gconf.h>
 #include <gconf/gconf-client.h>
 
+#include <libgnome/libgnome.h>
 #include <libgnome/gnome-exec.h>
 #include <libgnome/gnome-i18n.h>
 #include <libgnomeui/gnome-uidefs.h>
--- evolution-2.8.0/shell/e-shell-window.c.fix-missing-declarations	2006-09-05 00:11:45.000000000 -0400
+++ evolution-2.8.0/shell/e-shell-window.c	2006-09-05 06:48:18.000000000 -0400
@@ -58,6 +58,13 @@
 
 #include <string.h>
 
+#ifdef NM_SUPPORT_GLIB
+void e_shell_nm_glib_initialise (EShellWindow *window);
+void e_shell_nm_glib_dispose (EShellWindow *window);
+#elif NM_SUPPORT
+void e_shell_dbus_initialise (EShellWindow *window);
+void e_shell_dbus_dispose (EShellWindow *window);
+#endif
 
 /* A view for each component.  These are all created when EShellWindow is
    instantiated, but with the widget pointers to NULL and the page number set
--- evolution-2.8.0/calendar/gui/alarm-notify/alarm-notify.c.fix-missing-declarations	2006-09-02 02:41:31.000000000 -0400
+++ evolution-2.8.0/calendar/gui/alarm-notify/alarm-notify.c	2006-09-05 06:48:18.000000000 -0400
@@ -25,6 +25,7 @@
 
 #include <string.h>
 #include <bonobo/bonobo-main.h>
+#include <libedataserver/e-data-server-util.h>
 #include <libedataserver/e-url.h>
 #include <libedataserver/e-data-server-util.h>
 #include <libedataserverui/e-passwords.h>
--- evolution-2.8.0/calendar/gui/e-calendar-view.h.fix-missing-declarations	2006-08-07 00:47:42.000000000 -0400
+++ evolution-2.8.0/calendar/gui/e-calendar-view.h	2006-09-05 06:48:18.000000000 -0400
@@ -166,6 +166,18 @@
 					       gboolean new);
 
 gboolean	e_calendar_view_get_tooltips (ECalendarViewEventData *data);
+
+void           e_calendar_view_move_tip (GtkWidget *widget, int x, int y);
+
+#ifdef ENABLE_CAIRO
+void           draw_curved_rectangle (cairo_t *cr,
+                                      double x0,
+                                      double y0,
+                                      double rect_width,
+                                      double rect_height,
+                                      double radius);
+#endif
+
 G_END_DECLS
 
 #endif
--- evolution-2.8.0/calendar/gui/gnome-cal.h.fix-missing-declarations	2006-07-22 07:14:51.000000000 -0400
+++ evolution-2.8.0/calendar/gui/gnome-cal.h	2006-09-05 06:48:18.000000000 -0400
@@ -198,6 +198,8 @@
 						 const char* comp_uid,
 						 const char* comp_rid);
 
+GtkWidget * gnome_calendar_get_tag (GnomeCalendar *gcal);
+
 G_END_DECLS
 
 #endif
--- evolution-2.8.0/calendar/gui/dialogs/memo-page.h.fix-missing-declarations	2006-07-22 07:14:51.000000000 -0400
+++ evolution-2.8.0/calendar/gui/dialogs/memo-page.h	2006-09-05 06:48:18.000000000 -0400
@@ -53,6 +53,8 @@
 GtkType   memo_page_get_type  (void);
 MemoPage *memo_page_construct (MemoPage *epage);
 MemoPage *memo_page_new       (BonoboUIComponent *uic, CompEditorPageFlags flags);
+void      memo_page_set_classification (MemoPage *page, ECalComponentClassification class);
+void      memo_page_set_show_categories (MemoPage *page, gboolean state);
 
 G_END_DECLS
 
--- evolution-2.8.0/calendar/gui/cal-search-bar.c.fix-missing-declarations	2006-08-21 02:32:06.000000000 -0400
+++ evolution-2.8.0/calendar/gui/cal-search-bar.c	2006-09-05 06:53:57.000000000 -0400
@@ -31,6 +31,7 @@
 #include <gtk/gtksignal.h>
 #include <libgnome/gnome-i18n.h>
 #include <e-util/e-icon-factory.h>
+#include <libedataserver/e-categories.h>
 #include "cal-search-bar.h"
 
 typedef struct CALSearchBarItem {
--- evolution-2.8.0/calendar/gui/e-cal-model-tasks.h.fix-missing-declarations	2005-10-03 02:16:51.000000000 -0400
+++ evolution-2.8.0/calendar/gui/e-cal-model-tasks.h	2006-09-05 06:48:18.000000000 -0400
@@ -62,6 +62,7 @@
 ECalModelTasks *e_cal_model_tasks_new (void);
 
 void            e_cal_model_tasks_mark_task_complete (ECalModelTasks *model, gint model_row);
+void            e_cal_model_tasks_mark_task_incomplete (ECalModelTasks *model, gint model_row);
 void e_cal_model_tasks_update_due_tasks (ECalModelTasks *model);
 
 G_END_DECLS
