--- gnome-shell-3.3.3/data/org.gnome.shell.gschema.xml.in.old	2011-12-15 23:16:14.000000000 +0100
+++ gnome-shell-3.3.3/data/org.gnome.shell.gschema.xml.in	2012-01-07 16:24:42.000000000 +0100
@@ -97,7 +97,7 @@
       </_description>
     </key>
     <key name="show-date" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Show date in clock</_summary>
       <_description>
         If true, display date in the clock, in addition to time.
@@ -154,7 +154,7 @@
     </key>
 
     <key name="button-layout" type="s">
-      <default>":close"</default>
+      <default>":minimize,maximize,close"</default>
       <summary>Arrangement of buttons on the titlebar</summary>
       <description>
         This key overrides the key in org.gnome.desktop.wm.preferences when
