--- gnome-shell-3.1.92/data/gnome-shell.schemas.old	2011-03-17 16:45:49.000000000 +0100
+++ gnome-shell-3.1.92/data/gnome-shell.schemas	2011-09-26 10:58:01.000000000 +0200
@@ -22,7 +22,7 @@
         <applyto>/desktop/gnome/shell/windows/button_layout</applyto>
         <owner>gnome-shell</owner>
         <type>string</type>
-        <default>:close</default>
+        <default>:minimize,maximize,close</default>
         <locale name="C">
            <short>Arrangement of buttons on the titlebar</short>
            <long>
@@ -68,7 +68,7 @@
         <applyto>/desktop/gnome/shell/windows/theme</applyto>
         <owner>gnome-shell</owner>
         <type>string</type>
-        <default>Adwaita</default>
+        <default>Frugalware</default>
         <locale name="C">
           <short>Current theme</short>
           <long>
--- gnome-shell-3.1.92/data/org.gnome.shell.gschema.xml.in.old	2011-09-20 17:14:12.000000000 +0200
+++ gnome-shell-3.1.92/data/org.gnome.shell.gschema.xml.in	2011-09-26 11:33:51.000000000 +0200
@@ -88,7 +88,7 @@
       </_description>
     </key>
     <key name="show-date" type="b">
-      <default>false</default>
+      <default>true</default>
       <_summary>Show date in clock</_summary>
       <_description>
         If true, display date in the clock, in addition to time.
