--- gnome-shell-3.1.92/data/gnome-shell.schemas.old	2011-03-17 16:45:49.000000000 +0100
+++ gnome-shell-3.1.92/data/gnome-shell.schemas	2011-09-26 10:58:01.000000000 +0200
@@ -22,7 +22,7 @@
         <applyto>/desktop/gnome/shell/windows/button_layout</applyto>
         <owner>gnome-shell</owner>
         <type>string</type>
-        <default>:close</default>
+        <default>:minimize,maximize,close</default>
         <locale name="C">
            <short>Arrangement of buttons on the titlebar</short>
            <long>
@@ -68,7 +68,7 @@
         <applyto>/desktop/gnome/shell/windows/theme</applyto>
         <owner>gnome-shell</owner>
         <type>string</type>
-        <default>Adwaita</default>
+        <default>Frugalware</default>
         <locale name="C">
           <short>Current theme</short>
           <long>
