--- data/org.gnome.shell.gschema.xml.in.in
+++ data/org.gnome.shell.gschema.xml.in.in_fw
@@ -30,7 +30,7 @@
       </_description>
     </key>
     <key name="app-folder-categories" type="as">
-      <default>[ 'Utilities', 'Sundry' ]</default>
+      <default>[ 'Utilities', 'Sundry', 'Frugalware', 'Games' ]</default>
       <_summary>List of categories that should be displayed as folders</_summary>
       <_description>
         Each category name in this list will be represented as folder in the
@@ -189,7 +189,7 @@
     </key>
 
     <key name="button-layout" type="s">
-      <default>":close"</default>
+      <default>":minimize,maximize,close"</default>
       <_summary>Arrangement of buttons on the titlebar</_summary>
       <_description>
         This key overrides the key in org.gnome.desktop.wm.preferences when

