--- data/org.gnome.shell.gschema.xml.in.in
+++ data/org.gnome.shell.gschema.xml.in.in_Fw
@@ -40,7 +40,7 @@
       </_description>
     </key>
     <key name="app-folder-categories" type="as">
-      <default>[ 'Utilities', 'Sundry' ]</default>
+      <default>[ 'Utilities', 'Sundry', 'Frugalware', 'Games' ]</default>
       <_summary>List of categories that should be displayed as folders</_summary>
       <_description>
         Each category name in this list will be represented as folder in the
