From 6c3685649921790feb00616f5a840debce3e1d85 Mon Sep 17 00:00:00 2001
From: Tim Lunn <tim@feathertop.org>
Date: Tue, 05 Mar 2013 06:49:35 +0000
Subject: layout: fix errors in fallback HotCorner function

---
diff --git a/js/ui/layout.js b/js/ui/layout.js
index 63d040d..6ca5d84 100644
--- js/ui/layout.js
+++ js/ui/layout.js
@@ -1112,7 +1112,7 @@ const HotCorner = new Lang.Class({
         this._x = x;
         this._y = y;
 
-        this._setupFallbackCornerIfNeeded();
+        this._setupFallbackCornerIfNeeded(layoutManager);
 
         this._pressureBarrier = new PressureBarrier(HOT_CORNER_PRESSURE_THRESHOLD,
                                                     HOT_CORNER_PRESSURE_TIMEOUT,
@@ -1156,10 +1156,10 @@ const HotCorner = new Lang.Class({
         }
     },
 
-    _setupFallbackCornerIfNeeded: function() {
+    _setupFallbackCornerIfNeeded: function(layoutManager) {
         if (!global.display.supports_extended_barriers()) {
             this.actor = new Clutter.Actor({ name: 'hot-corner-environs',
-                                             x: x, y: y,
+                                             x: this._x, y: this._y,
                                              width: 3,
                                              height: 3,
                                              reactive: true });
--
cgit v0.9.1

