--- gnome-shell-3.2.1/js/ui/environment.js.old	2011-10-12 18:50:17.000000000 +0200
+++ gnome-shell-3.2.1/js/ui/environment.js	2011-12-01 22:39:33.274521861 +0100
@@ -71,9 +71,9 @@
     };
 
     // Work around https://bugzilla.mozilla.org/show_bug.cgi?id=508783
-    Date.prototype.toLocaleFormat = function(format) {
+   /* Date.prototype.toLocaleFormat = function(format) {
         return Shell.util_format_date(format, this.getTime());
-    };
+    };*/
 
     let slowdownEnv = GLib.getenv('GNOME_SHELL_SLOWDOWN_FACTOR');
     if (slowdownEnv) {
