diff -Naur darcs-1.0.4.orig/Repository.lhs darcs-1.0.4/Repository.lhs
--- darcs-1.0.4.orig/Repository.lhs	2005-11-13 22:28:28.000000000 +0100
+++ darcs-1.0.4/Repository.lhs	2005-11-17 23:23:50.000000000 +0100
@@ -110,7 +110,7 @@
                  setCurrentDirectory ".."
                  cd' <- getCurrentDirectory
                  if cd' /= cd
-                     then findRepository' $
+                     then amInRepository' $
                          reverse (takeWhile (/='/') $ reverse cd)///dir
                      else return (Left $
                                   "You need to be in a repository directory" ++
