diff -Naur efltk/makeinclude.in efltk-p/makeinclude.in
--- efltk/makeinclude.in	2006-09-02 22:55:56.000000000 +0200
+++ efltk-p/makeinclude.in	2006-09-03 13:20:03.735015208 +0200
@@ -55,8 +55,8 @@
 # flags for compiler:
 # WALL            = -Wcomment -Wformat -Wimplicit-int -Werror-implicit-function-declaration -Wmain -Wmultichar -Wparentheses -Wreturn-type -Wswitch -Wunused
 WALL            = -Wall
-CFLAGS		= -DHAVE_CONFIG_H -I$(EFLTK_BUILDDIR) $(WALL) @CFLAGS@ @DEBUGFLAG@
-CXXFLAGS	= -DHAVE_CONFIG_H -I$(EFLTK_BUILDDIR) $(WALL) @CXXFLAGS@ @DEBUGFLAG@
+CFLAGS		+= -DHAVE_CONFIG_H -I$(EFLTK_BUILDDIR) @CFLAGS@ 
+CXXFLAGS	+= -DHAVE_CONFIG_H -I$(EFLTK_BUILDDIR) @CXXFLAGS@
 
 # static libs:
 LIBEXT		=@LIBEXT@
