diff -Naur efltk/efltk/Fl_Text_Buffer.h efltk-p/efltk/Fl_Text_Buffer.h
--- efltk/efltk/Fl_Text_Buffer.h	2006-09-02 22:55:52.000000000 +0200
+++ efltk-p/efltk/Fl_Text_Buffer.h	2006-09-03 13:25:26.391963896 +0200
@@ -78,7 +78,7 @@
    Fl_Text_Buffer(int requestedSize = 0);
    ~Fl_Text_Buffer();
 
-   char *Fl_Text_Buffer::static_buffer();
+   char *static_buffer();
 
     // Undo stack size: (default 50)
    int undo_size();
