diff -Naur efltk/efltk/Fl_Combo_Box.h efltk-p/efltk/Fl_Combo_Box.h
--- efltk/efltk/Fl_Combo_Box.h	2006-09-02 22:55:52.000000000 +0200
+++ efltk-p/efltk/Fl_Combo_Box.h	2006-09-03 13:40:32.269249616 +0200
@@ -23,8 +23,8 @@
 
     void ctor_init();
 protected:
-    static void Fl_Combo_Box::cb_browse(Fl_Widget *w, void *data);
-    static void Fl_Combo_Box::cb_button(Fl_Widget *w, void *data);
+    static void cb_browse(Fl_Widget *w, void *data);
+    static void cb_button(Fl_Widget *w, void *data);
 public:
     static Fl_Named_Style* default_style;
 
