diff -ur efltk//src/db/odbc/Fl_ODBC_Database.cpp efltk-2//src/db/odbc/Fl_ODBC_Database.cpp
--- efltk//src/db/odbc/Fl_ODBC_Database.cpp	2009-04-22 13:51:33.000000000 +0200
+++ efltk-2//src/db/odbc/Fl_ODBC_Database.cpp	2010-05-31 11:48:01.483437976 +0200
@@ -266,9 +266,9 @@
 
 void Fl_ODBC_Database::bind_parameters(Fl_Query *query) {
     int               rc;
-    SQLINTEGER        cbNullData = SQL_NULL_DATA;
-    SQLINTEGER        cbNTS = SQL_NTS;
-    SQLINTEGER       *pcbLen;
+    SQLLEN        cbNullData = SQL_NULL_DATA;
+    SQLLEN        cbNTS = SQL_NTS;
+    SQLLEN       *pcbLen;
 
     SQLHSTMT    statement = (SQLHSTMT)query_handle(query);
     Fl_Params&  params = query->params();
@@ -389,7 +389,7 @@
  * Get the column information
  */
 void Fl_ODBC_Database::query_col_attributes(Fl_Query *query,short column,short descType,long& value) {
-    int m_retcode = SQLColAttributes(query_handle(query),column,descType,0,0,0,(SQLINTEGER *)&value);
+    int m_retcode = SQLColAttributes(query_handle(query),column,descType,0,0,0,(SQLLEN *)&value);
 
     if (!successful(m_retcode))
         fl_throw(query_error(query));
@@ -568,7 +568,7 @@
 
     Fl_Data_Fields& fields = query_fields(query);
     unsigned    fieldCount = fields.count();
-    SQLINTEGER  dataLength;
+    SQLLEN  dataLength;
 
     if (!fieldCount) return;
 
