--- alembic-1.8.0/lib/Alembic/AlembicConfig.cmake.in	2021-04-17 19:27:06.000000000 +0200
+++ alembic-1.8.0.new/lib/Alembic/AlembicConfig.cmake.in	2021-04-18 13:56:28.720416211 +0200
@@ -2,7 +2,7 @@
 
 include(CMakeFindDependencyMacro)
 # TODO whenever we loose the back-compatibility with IlmBase < 3, a REQUIRED needs to be added to find_dependency()
-find_dependency(IMath)
+find_dependency(Imath)
 
 SET(Alembic_HAS_HDF5 @USE_HDF5@)
 SET(Alembic_HAS_SHARED_LIBS @ALEMBIC_SHARED_LIBS@)
