--- Makefile	2017-12-11 21:26:37.000000000 +0100
+++ Makefile.new	2017-12-19 17:16:57.703957380 +0100
@@ -132,8 +132,8 @@
 
 $(SHARED_LIB): $(BUILD_DIR) $(OBJECTS)
 	@echo "  L: $(@F)"
-	@$(INSTALL) -m 644 libxvidcore.def $(BUILD_DIR)/libxvidcore.def
-	@$(INSTALL) -m 644 libxvidcore.ld $(BUILD_DIR)/libxvidcore.ld
+	@$(INSTALL) -m 755 libxvidcore.def $(BUILD_DIR)/libxvidcore.def
+	@$(INSTALL) -m 755 libxvidcore.ld $(BUILD_DIR)/libxvidcore.ld
 	@cd $(BUILD_DIR) && $(CC) $(LDFLAGS) $(OBJECTS) -o $(PRE_SHARED_LIB) $(SPECIFIC_LDFLAGS)
 
 #-----------------------------------------------------------------------------
@@ -144,21 +144,21 @@
 	@echo "  D: $(includedir)"
 	@$(INSTALL) -d $(DESTDIR)$(includedir)
 	@echo "  I: $(includedir)/xvid.h"
-	@$(INSTALL) -m 644 $(SRC_DIR)/xvid.h $(DESTDIR)$(includedir)/xvid.h
+	@$(INSTALL) -m 755 $(SRC_DIR)/xvid.h $(DESTDIR)$(includedir)/xvid.h
 	@echo "  D: $(libdir)"
 	@$(INSTALL) -d $(DESTDIR)$(libdir)
 	@echo "  I: $(libdir)/$(STATIC_LIB)"
-	@$(INSTALL) -m 644 $(BUILD_DIR)/$(STATIC_LIB) $(DESTDIR)$(libdir)/$(STATIC_LIB)
+	@$(INSTALL) -m 755 $(BUILD_DIR)/$(STATIC_LIB) $(DESTDIR)$(libdir)/$(STATIC_LIB)
 ifeq ($(SHARED_EXTENSION),dll)
 	@echo "  I: $(libdir)/$(SHARED_LIB).a"
-	@$(INSTALL) -m 644 $(BUILD_DIR)/$(SHARED_LIB).a $(DESTDIR)$(libdir)/$(SHARED_LIB).a
+	@$(INSTALL) -m 755 $(BUILD_DIR)/$(SHARED_LIB).a $(DESTDIR)$(libdir)/$(SHARED_LIB).a
 	@echo "  D: $(bindir)"
 	@$(INSTALL) -d $(DESTDIR)$(bindir)
 	@echo "  I: $(bindir)/$(SHARED_LIB)"
 	@$(INSTALL) -m 755 $(BUILD_DIR)/$(SHARED_LIB) $(DESTDIR)$(bindir)/$(SHARED_LIB)
 else
 	@echo "  I: $(libdir)/$(SHARED_LIB)"
-	@$(INSTALL) -m 644 $(BUILD_DIR)/$(SHARED_LIB) $(DESTDIR)$(libdir)/$(SHARED_LIB)
+	@$(INSTALL) -m 755 $(BUILD_DIR)/$(SHARED_LIB) $(DESTDIR)$(libdir)/$(SHARED_LIB)
 	@test -z "$(SO_API_MAJOR_LINK)" || \
 		$(LN_S) $(SHARED_LIB) $(DESTDIR)$(libdir)/$(SO_API_MAJOR_LINK)
 	@test -z "$(SO_LINK)" || \
