diff -Naur libepoxy-1.5.0/src/dispatch_common.c libepoxy-1.5.0-p/src/dispatch_common.c
--- libepoxy-1.5.0/src/dispatch_common.c	2018-02-24 15:09:37.000000000 +0100
+++ libepoxy-1.5.0-p/src/dispatch_common.c	2018-04-10 00:36:15.575530981 +0200
@@ -194,7 +194,7 @@
 #define EGL_LIB "libEGL.so.1"
 #define GLES1_LIB "libGLESv1_CM.so.1"
 #define GLES2_LIB "libGLESv2.so.2"
-#define OPENGL_LIB "libOpenGL.so.0"
+#define OPENGL_LIB "libOpenGL.so.1"
 #endif
 
 #ifdef __GNUC__
