--- mozilla/browser/app/mozilla.in.orig	2004-10-26 11:26:11.000000000 +0200
+++ mozilla/browser/app/mozilla.in	2004-12-29 09:50:42.737301712 +0100
@@ -136,6 +136,11 @@
     fi
 }
 
+# Force use of GTK+2 for Flash player to get better performance
+# (no impact when using mozilla gtk1 build).
+FLASH_GTK_LIBRARY=libgtk-x11-2.0.so.0
+export FLASH_GTK_LIBRARY
+
 if [ "$OSTYPE" = "beos" ]; then
   mimeset -F "$MOZILLA_BIN"
 fi
