--- mozilla/browser/app/mozilla.in.orig	2004-10-26 11:26:11.000000000 +0200
+++ mozilla/browser/app/mozilla.in	2004-12-29 09:53:00.229399736 +0100
@@ -110,6 +110,7 @@
   # Check default compile-time libdir
   if [ -x "$moz_libdir/run-mozilla.sh" ]; then
     dist_bin="$moz_libdir"
+    run_moz="$moz_libdir/run-mozilla.sh"
   else 
     echo "Cannot find mozilla runtime directory. Exiting."
     exit 1
@@ -125,7 +126,7 @@
 # This is taken almost verbatim from the Mozilla RPM package's launch script.
 MOZ_CLIENT_PROGRAM="$dist_bin/mozilla-xremote-client"
 check_running() {
-    "${run_moz}" "$MOZ_CLIENT_PROGRAM" -a "${progbase}" 'ping()' 2>/dev/null >/dev/null
+    "${run_moz}" "$MOZ_CLIENT_PROGRAM" -a firefox 'ping()' 2>/dev/null >/dev/null
     RETURN_VAL=$?
     if [ $RETURN_VAL -eq 0 ]; then
         echo 1
@@ -177,14 +178,14 @@
 		if [ ${_USE_EXIST} -eq 1 ]; then
 			# We should use an existing instance, as _USE_EXIST=$_USE_EXIST=-1
 			_remote_cmd="openURL(${_optLast})"
-			"${run_moz}" "$MOZ_CLIENT_PROGRAM" -a "${progbase}" "${_remote_cmd}"
+			"${run_moz}" "$MOZ_CLIENT_PROGRAM" -a firefox "${_remote_cmd}"
 			unset _remote_cmd
 			exit $?
 		fi
 	else
 		# No command line args. Open new window/tab
 		#exec "${run_moz}" "$MOZ_CLIENT_PROGRAM" -a "${progbase}" "xfeDoCommand(openBrowser)"
-		"${run_moz}" "$MOZ_CLIENT_PROGRAM" -a "${progbase}" "xfeDoCommand(openBrowser)"
+		"${run_moz}" "$MOZ_CLIENT_PROGRAM" -a firefox "xfeDoCommand(openBrowser)"
 		exit $?
 	fi
 fi
