--- mozilla/modules/libpref/src/init/all.js.orig	2004-10-15 14:50:11.740865580 +0200
+++ mozilla/modules/libpref/src/init/all.js	2004-10-15 14:50:34.371625152 +0200
@@ -653,7 +653,7 @@
 pref("intl.charset.detector",               "chrome://navigator/locale/navigator.properties");
 pref("intl.charset.default",                "chrome://navigator-platform/locale/navigator.properties");
 pref("intl.content.langcode",               "chrome://communicator-region/locale/region.properties");
-pref("intl.locale.matchOS",                 false);
+pref("intl.locale.matchOS",                 true);
 // fallback charset list for Unicode conversion (converting from Unicode)
 // currently used for mail send only to handle symbol characters (e.g Euro, trademark, smartquotes)
 // for ISO-8859-1
