--- mozilla/browser/app/mozilla.in.orig	2004-10-26 11:26:11.000000000 +0200
+++ mozilla/browser/app/mozilla.in	2004-12-29 09:58:27.582634480 +0100
@@ -217,14 +217,32 @@
 export MRE_HOME
 eval "set -- $moreargs"
 
+## Autodetect language
+if [ "$HOME" -a "$LANG" -a "$SET_LANG" != "0" -a -r "$moz_libdir/chrome/installed-chrome.txt" ]; then
+   MOZ_LANG_PART1=`echo $LANG | cut -f 1 --delimiter "_"`
+   MOZ_LANG_PART2=`echo $LANG | cut -f 2 --delimiter "_" | cut -f 1 --delimiter "."`
+   if [ "$MOZ_LANG_PART2" -a -r "$moz_libdir/chrome/$MOZ_LANG_PART1-$MOZ_LANG_PART2.jar" ]; then
+       MOZ_LANG="-contentLocale $MOZ_LANG_PART2 -UILocale $MOZ_LANG_PART1-$MOZ_LANG_PART2"
+   else
+       # Default locale not found. Consider the "main" locale
+       # i.e. : for fr_CA -> fr_FR, etc.
+       MOZ_LANG_PART2=`echo $MOZ_LANG_PART1 | tr a-z A-Z `;
+       if [ "$MOZ_LANG_PART2" -a -r "$moz_libdir/chrome/$MOZ_LANG_PART1-$MOZ_LANG_PART2.jar" ]; then
+           MOZ_LANG="-contentLocale $MOZ_LANG_PART2 -UILocale $MOZ_LANG_PART1-$MOZ_LANG_PART2"
+       else
+           unset MOZ_LANG
+       fi
+   fi
+fi
+
 ## Start addon scripts
 moz_pis_startstop_scripts "start"
 
 if [ $debugging = 1 ]
 then
-  echo $dist_bin/run-mozilla.sh $script_args $dist_bin/$MOZILLA_BIN "$@"
+  echo $dist_bin/run-mozilla.sh $script_args $dist_bin/$MOZILLA_BIN $MOZ_LANG "$@"
 fi
-"$dist_bin/run-mozilla.sh" $script_args "$dist_bin/$MOZILLA_BIN" "$@"
+"$dist_bin/run-mozilla.sh" $script_args "$dist_bin/$MOZILLA_BIN" $MOZ_LANG "$@"
 exitcode=$?
 
 ## Stop addon scripts
