diff -Naur mozilla-release/toolkit/xre/nsAppRunner.cpp mozilla-release-slice/toolkit/xre/nsAppRunner.cpp
--- mozilla-release/toolkit/xre/nsAppRunner.cpp	2015-09-18 00:13:31.000000000 +0200
+++ mozilla-release-slice/toolkit/xre/nsAppRunner.cpp	2015-09-25 13:56:11.831349119 +0200
@@ -4320,10 +4320,8 @@
 
 #if defined(MOZ_WIDGET_GTK)
 #if defined(MOZ_MEMORY) || defined(__FreeBSD__) || defined(__NetBSD__)
-  // Disable the slice allocator, since jemalloc already uses similar layout
-  // algorithms, and using a sub-allocator tends to increase fragmentation.
-  // This must be done before g_thread_init() is called.
-  g_slice_set_config(G_SLICE_CONFIG_ALWAYS_MALLOC, 1);
+  // enable slice allocator for child processes
+  unsetenv("G_SLICE");
 #endif
   g_thread_init(nullptr);
 #endif
