Index: mozilla/security/coreconf/location.mk
===================================================================
RCS file: /cvsroot/mozilla/security/coreconf/location.mk,v
retrieving revision 1.9
diff -u -r1.9 location.mk
--- mozilla/security/coreconf/location.mk	25 Apr 2004 15:02:17 -0000	1.9
+++ mozilla/security/coreconf/location.mk	11 Apr 2005 20:21:57 -0000
@@ -61,6 +61,8 @@
 
 DIST          = $(SOURCE_PREFIX)/$(PLATFORM)
 
+NSPR_LIBDIR   = $(DIST)/lib
+
 ifdef BUILD_DEBUG_GC
     DEFINES += -DDEBUG_GC
 endif
Index: mozilla/security/manager/Makefile.in
===================================================================
RCS file: /cvsroot/mozilla/security/manager/Makefile.in,v
retrieving revision 1.56
diff -u -r1.56 Makefile.in
--- mozilla/security/manager/Makefile.in	14 Mar 2005 10:00:58 -0000	1.56
+++ mozilla/security/manager/Makefile.in	11 Apr 2005 20:21:57 -0000
@@ -76,12 +76,23 @@
 endif
 endif
 
+ABS_DIST := $(shell cd $(DIST) && pwd)
+ifeq ($(OS_ARCH),WINNT)
+ABS_DIST := $(shell cygpath -w $(ABS_DIST) | sed -e 's|\\\\|/|g')
+endif
+NSPR_LIBDIR = $(firstword $(filter -L%,$(NSPR_LIBS)))
+ifneq (,$(strip $(NSPR_LIBDIR)))
+NSPR_LIBDIR := $(subst -L,,$(subst -L$(DIST),-L$(ABS_DIST),$(NSPR_LIBDIR)))
+else
+NSPR_LIBDIR = $(ABS_DIST)/lib
+endif
 # NSS makefiles are not safe for parallel execution.
 DEFAULT_GMAKE_FLAGS = MAKE="$(MAKE) -j1" -j1
 DEFAULT_GMAKE_FLAGS += CC="$(CC)"
-DEFAULT_GMAKE_FLAGS += MOZILLA_INCLUDES="-I$(MOZ_BUILD_ROOT)/dist/include/nspr -I$(MOZ_BUILD_ROOT)/dist/include/dbm"
-DEFAULT_GMAKE_FLAGS += SOURCE_MD_DIR=$(MOZ_BUILD_ROOT)/dist
-DEFAULT_GMAKE_FLAGS += DIST=$(MOZ_BUILD_ROOT)/dist
+DEFAULT_GMAKE_FLAGS += MOZILLA_INCLUDES="$(subst -I$(DIST),-I$(ABS_DIST),$(NSPR_CFLAGS) -I$(DIST)/include/dbm)"
+DEFAULT_GMAKE_FLAGS += SOURCE_MD_DIR=$(ABS_DIST)
+DEFAULT_GMAKE_FLAGS += DIST=$(ABS_DIST)
+DEFAULT_GMAKE_FLAGS += NSPR_LIBDIR=$(NSPR_LIBDIR)
 DEFAULT_GMAKE_FLAGS += MOZILLA_CLIENT=1
 DEFAULT_GMAKE_FLAGS += NO_MDUPDATE=1
 ABS_topsrcdir   := $(shell cd $(topsrcdir); pwd)
Index: mozilla/security/nss/lib/ckfw/builtins/Makefile
===================================================================
RCS file: /cvsroot/mozilla/security/nss/lib/ckfw/builtins/Makefile,v
retrieving revision 1.14
diff -u -r1.14 Makefile
--- mozilla/security/nss/lib/ckfw/builtins/Makefile	20 Jan 2005 02:25:46 -0000	1.14
+++ mozilla/security/nss/lib/ckfw/builtins/Makefile	11 Apr 2005 20:21:57 -0000
@@ -53,23 +53,23 @@
 # This is merely an expedient hack and not the right solution.
 ifdef NS_USE_GCC
 EXTRA_LIBS += \
-	-L$(DIST)/lib \
+	-L$(NSPR_LIBDIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
 	$(NULL)
 else
 EXTRA_LIBS += \
-	$(DIST)/lib/$(NSPR31_LIB_PREFIX)plc4_s.lib \
-	$(DIST)/lib/$(NSPR31_LIB_PREFIX)plds4_s.lib \
+	$(NSPR_LIBDIR)/$(NSPR31_LIB_PREFIX)plc4_s.lib \
+	$(NSPR_LIBDIR)/$(NSPR31_LIB_PREFIX)plds4_s.lib \
 	$(NULL)
 endif
 
 else
 
 EXTRA_LIBS += \
-	$(DIST)/lib/$(LIB_PREFIX)plc4.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)plds4.$(LIB_SUFFIX) \
+	$(NSPR_LIBDIR)/$(LIB_PREFIX)plc4.$(LIB_SUFFIX) \
+	$(NSPR_LIBDIR)/$(LIB_PREFIX)plds4.$(LIB_SUFFIX) \
 	$(NULL)
 
 endif
Index: mozilla/security/nss/lib/fortcrypt/swfort/pkcs11/Makefile
===================================================================
RCS file: /cvsroot/mozilla/security/nss/lib/fortcrypt/swfort/pkcs11/Makefile,v
retrieving revision 1.18
diff -u -r1.18 Makefile
--- mozilla/security/nss/lib/fortcrypt/swfort/pkcs11/Makefile	25 Apr 2004 15:03:08 -0000	1.18
+++ mozilla/security/nss/lib/fortcrypt/swfort/pkcs11/Makefile	11 Apr 2005 20:21:57 -0000
@@ -63,7 +63,7 @@
 	$(DIST)/lib/$(LIB_PREFIX)softokn.$(LIB_SUFFIX) \
 	$(CRYPTO_LIB) \
 	$(DIST)/lib/$(LIB_PREFIX)secutil.$(LIB_SUFFIX) \
-	-L$(DIST)/lib \
+	-L$(NSPR_LIBDIR) \
 	-lplc4 \
 	-lplds4 \
 	-lnspr4 \
@@ -76,8 +76,8 @@
 	$(DIST)/lib/softokn.lib \
 	$(CRYPTO_LIB) \
 	$(DIST)/lib/secutil.lib \
-	$(DIST)/lib/$(NSPR31_LIB_PREFIX)plc4_s.lib \
-	$(DIST)/lib/$(NSPR31_LIB_PREFIX)plds4_s.lib \
+	$(NSPR_LIBDIR)/$(NSPR31_LIB_PREFIX)plc4_s.lib \
+	$(NSPR_LIBDIR)/$(NSPR31_LIB_PREFIX)plds4_s.lib \
 	wsock32.lib \
 	winmm.lib \
 	$(NULL)
@@ -98,8 +98,8 @@
 	$(DIST)/lib/$(LIB_PREFIX)softokn.$(LIB_SUFFIX) \
 	$(CRYPTO_LIB) \
 	$(DIST)/lib/$(LIB_PREFIX)secutil.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)plc4.$(LIB_SUFFIX) \
-	$(DIST)/lib/$(LIB_PREFIX)plds4.$(LIB_SUFFIX) \
+	$(NSPR_LIBDIR)/$(LIB_PREFIX)plc4.$(LIB_SUFFIX) \
+	$(NSPR_LIBDIR)/$(LIB_PREFIX)plds4.$(LIB_SUFFIX) \
 	$(NULL)
 
 endif
