--- mozilla/toolkit/crashreporter/google-breakpad/src/client/linux/handler/exception_handler.cc.orig	2009-06-10 00:17:41.000000000 +0200
+++ mozilla/toolkit/crashreporter/google-breakpad/src/client/linux/handler/exception_handler.cc	2009-06-10 00:17:52.000000000 +0200
@@ -33,6 +33,7 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
+#include <stdio.h>
 
 #include <cassert>
 #include <cstdlib>
--- mozilla/toolkit/crashreporter/google-breakpad/src/client/linux/handler/minidump_generator.cc.orig	2009-06-10 00:18:06.000000000 +0200
+++ mozilla/toolkit/crashreporter/google-breakpad/src/client/linux/handler/minidump_generator.cc	2009-06-10 00:18:19.000000000 +0200
@@ -41,6 +41,7 @@
 #include <cstdlib>
 #include <ctime>
 #include <string.h>
+#include <stdio.h>
 
 #include "common/linux/file_id.h"
 #include "client/linux/handler/linux_thread.h"
--- mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/dump_symbols.cc.orig	2009-06-10 00:15:23.000000000 +0200
+++ mozilla/toolkit/crashreporter/google-breakpad/src/common/linux/dump_symbols.cc	2009-06-10 00:16:06.000000000 +0200
@@ -45,6 +45,7 @@
 #include <functional>
 #include <vector>
 #include <string.h>
+#include <stdio.h>
 
 #include "common/linux/dump_symbols.h"
 #include "common/linux/file_id.h"
--- mozilla/toolkit/xre/nsAppRunner.cpp.orig	2009-06-10 00:19:15.000000000 +0200
+++ mozilla/toolkit/xre/nsAppRunner.cpp	2009-06-10 00:19:27.000000000 +0200
@@ -1459,7 +1459,7 @@
   if (NS_FAILED(rv))
     return rv;
 
-#elif
+#else
 #error Oops, you need platform-specific code here
 #endif
 
