Index: firefox-32.0~b9+build1/accessible/src/base/TextRange.cpp
===================================================================
--- firefox-32.0~b9+build1.orig/accessible/src/base/TextRange.cpp	2014-08-30 00:41:16.000000000 +0100
+++ firefox-32.0~b9+build1/accessible/src/base/TextRange.cpp	2014-08-30 09:57:49.910751230 +0100
@@ -10,7 +10,8 @@
 #include "HyperTextAccessible.h"
 #include "nsAccUtils.h"
 
-using namespace mozilla::a11y;
+namespace mozilla {
+namespace a11y {
 
 ////////////////////////////////////////////////////////////////////////////////
 // TextPoint
@@ -294,3 +295,6 @@
 {
 
 }
+
+} // namespace a11y
+} // namespace mozilla
