From eb9b97f46b7d064c32ed0f086d89a70427ce1d14 Mon Sep 17 00:00:00 2001
From: Tobias Wollgam <twollgam@gmail.com>
Date: Fri, 3 May 2024 00:21:57 +0200
Subject: [PATCH] fix compile error

---
 game/scorewindow.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/game/scorewindow.cc b/game/scorewindow.cc
index 276e3fa92c..e4883f4181 100644
--- a/game/scorewindow.cc
+++ b/game/scorewindow.cc
@@ -24,7 +24,7 @@ ScoreWindow::ScoreWindow(Game& game, Instruments& instruments, Database& databas
 	}
 
 	// Instruments
-	std::remove_if(instruments.begin(), instruments.end(),[](std::unique_ptr<InstrumentGraph> const& i){ return i->getScore() < 100; }); // Dead.
+	instruments.erase(std::remove_if(instruments.begin(), instruments.end(),[](std::unique_ptr<InstrumentGraph> const& i){ return i->getScore() < 100; }), instruments.end()); // Dead.
 	for (std::unique_ptr<InstrumentGraph> const& i: instruments) {
 		input::DevType const& type = i->getGraphType();
 		std::string const& track_simple = i->getTrack();
