--- game/ffmpeg.hh~     2009-01-05 11:07:50.000000000 +0000
+++ game/ffmpeg.hh      2009-01-05 11:08:07.000000000 +0000
@@ -10,6 +10,7 @@
 #include <iostream>
 #include <vector>
 #include <limits>
+#include <stdint.h>
 
 struct AudioFrame {
        double timestamp;
--- game/color.cc~      2009-01-05 11:14:42.000000000 +0000
+++ game/color.cc       2009-01-05 11:14:47.000000000 +0000
@@ -1,4 +1,5 @@
 #include "color.hh"
+#include <cstdio>
 #include <iostream>
 
 namespace {

-- 

