diff -Naur xchat-2.4.3.orig/src/common/servlist.c xchat-2.4.3/src/common/servlist.c
--- xchat-2.4.3.orig/src/common/servlist.c	2005-03-06 01:21:20.000000000 +0100
+++ xchat-2.4.3/src/common/servlist.c	2005-06-10 13:39:13.000000000 +0200
@@ -43,6 +43,10 @@
 
 static const struct defaultserver def[] =
 {
+	{"Frugalware Servers", 0, "#frugalware"},
+	{0,                     "irc.freenode.net"},
+	{0,                     "irc.ircnet.net"},
+	
 	{"2600net",	0},
 	{0,			"irc.2600.net"},
 	{0,			"ptolomea.2600.net"},
@@ -841,7 +845,7 @@
 #endif
 			if (def[i].channel)
 				net->autojoin = strdup (def[i].channel);
-			if (!strcmp (def[i].network, "ChatJunkies"))
+			if (!strcmp (def[i].network, "Frugalware Servers"))
 				prefs.slist_select = j;
 			j++;
 		} else
