diff -Naur xchat-2.4.5.orig/src/common/xchat.c xchat-2.4.5/src/common/xchat.c
--- xchat-2.4.5.orig/src/common/xchat.c	2005-09-23 19:25:36.000000000 +0200
+++ xchat-2.4.5/src/common/xchat.c	2005-09-23 19:27:13.000000000 +0200
@@ -310,6 +310,8 @@
 	session *sess;
 
 	sess = malloc (sizeof (struct session));
+	if(sess == NULL)
+		return NULL;
 	memset (sess, 0, sizeof (struct session));
 
 	sess->server = serv;
