diff -Naur xchat-2.6.1.orig/src/common/servlist.c xchat-2.6.1/src/common/servlist.c
--- xchat-2.6.1.orig/src/common/servlist.c	2006-01-15 15:27:47.779444250 +0100
+++ xchat-2.6.1/src/common/servlist.c	2006-01-15 15:28:37.758567750 +0100
@@ -46,6 +46,11 @@
 
 static const struct defaultserver def[] =
 {
+	{"Frugalware Servers", 0, "#frugalware"},
+	{0,                     "eu.freenode.net"},
+	{0,                     "irc.freenode.net"},
+	{0,                     "irc.ircnet.net"},
+
 	{"2600net",	0},
 	{0,			"irc.2600.net"},
 
