diff -Naur xchat-2.4.5.orig/src/common/servlist.c xchat-2.4.5/src/common/servlist.c
--- xchat-2.4.5.orig/src/common/servlist.c	2005-09-17 16:10:18.000000000 +0200
+++ xchat-2.4.5/src/common/servlist.c	2005-09-17 16:11:56.000000000 +0200
@@ -45,6 +45,11 @@
 
 static const struct defaultserver def[] =
 {
+	{"Frugalware Servers", 0, "#frugalware"},
+	{0,                     "eu.freenode.net"},
+	{0,                     "irc.freenode.net"},
+	{0,                     "irc.ircnet.net"},
+
 	{"2600net",	0},
 	{0,			"irc.2600.net"},
 	{0,			"ptolomea.2600.net"},
@@ -822,7 +827,7 @@
 				free (net->encoding);
 				net->encoding = strdup (def[i].charset);
 			}
-			if (!strcmp (def[i].network, "ChatJunkies"))
+			if (!strcmp (def[i].network, "Frugalware Servers"))
 				prefs.slist_select = j;
 			j++;
 		} else
