diff -r 97d8bf4c46f9 color.c
--- a/color.c	Mon Jun 29 17:52:30 2009 +0200
+++ b/color.c	Mon Jun 29 17:54:21 2009 +0200
@@ -95,6 +95,7 @@
   { "index",		MT_COLOR_INDEX },
   { "sidebar_new",	MT_COLOR_NEW },
   { "sidebar_flagged",	MT_COLOR_FLAGGED },
+  { "sidebar",		MT_COLOR_SIDEBAR },
   { NULL,		0 }
 };
 
@@ -144,6 +145,7 @@
 
   /* set some defaults */
   ColorDefs[MT_COLOR_STATUS] = A_REVERSE;
+  ColorDefs[MT_COLOR_SIDEBAR] = A_REVERSE;
   ColorDefs[MT_COLOR_INDICATOR] = A_REVERSE;
   ColorDefs[MT_COLOR_SEARCH] = A_REVERSE;
   ColorDefs[MT_COLOR_MARKERS] = A_REVERSE;
diff -r 97d8bf4c46f9 mutt_curses.h
--- a/mutt_curses.h	Mon Jun 29 17:52:30 2009 +0200
+++ b/mutt_curses.h	Mon Jun 29 17:54:21 2009 +0200
@@ -125,6 +125,7 @@
   MT_COLOR_ATTACHMENT,
   MT_COLOR_SEARCH,
   MT_COLOR_BOLD,
+  MT_COLOR_SIDEBAR,
   MT_COLOR_UNDERLINE,
   MT_COLOR_INDEX,
   MT_COLOR_NEW,
diff -r 97d8bf4c46f9 sidebar.c
--- a/sidebar.c	Mon Jun 29 17:52:30 2009 +0200
+++ b/sidebar.c	Mon Jun 29 17:54:21 2009 +0200
@@ -85,7 +85,7 @@
 	static char *entry = 0;
 	char *c;
 	int i = 0;
-	int delim_len = strlen(SidebarDelim);
+	int delim_len = strlen(NONULL(SidebarDelim));
 
 	c = realloc(entry, SidebarWidth - delim_len + 2);
 	if ( c ) entry = c;
@@ -141,7 +141,7 @@
 #ifndef USE_SLANG_CURSES
         attr_t attrs;
 #endif
-        short delim_len = strlen(SidebarDelim);
+        short delim_len = strlen(NONULL(SidebarDelim));
         short color_pair;
 
         static bool initialized = false;
@@ -196,22 +196,26 @@
     }
 
         /* get attributes for divider */
-	SETCOLOR(MT_COLOR_STATUS);
+	SETCOLOR(MT_COLOR_SIDEBAR);
 #ifndef USE_SLANG_CURSES
         attr_get(&attrs, &color_pair, 0);
 #else
         color_pair = attr_get();
 #endif
-	SETCOLOR(MT_COLOR_NORMAL);
+	/* SETCOLOR(MT_COLOR_SIDEBAR); */
 
 	/* draw the divider */
 
 	for ( ; lines < LINES-1-(menu != MENU_PAGER || option(OPTSTATUSONTOP)); lines++ ) {
 		move(lines, SidebarWidth - delim_len);
-		addstr(NONULL(SidebarDelim));
-#ifndef USE_SLANG_CURSES
-                mvchgat(lines, SidebarWidth - delim_len, delim_len, 0, color_pair, NULL);
-#endif
+		if (option (OPTASCIICHARS))
+			addstr (NONULL (SidebarDelim));
+		else if (!option (OPTASCIICHARS) && !strcmp (NONULL(SidebarDelim), "|"))
+			addch (ACS_VLINE);
+		else if ((Charset_is_utf8) && !strcmp (NONULL(SidebarDelim), "|"))
+			addstr ("\342\224\202");
+		else
+			addstr (NONULL (SidebarDelim));
 	}
 
 	if ( Incoming == 0 ) return 0;
