--- ./polkit-gnome/polkit-gnome-action.c.old	2009-02-06 17:31:46.000000000 +0000
+++ ./polkit-gnome/polkit-gnome-action.c	2009-02-06 17:32:03.000000000 +0000
@@ -2062,6 +2062,8 @@ _button_clicked (GtkButton *button, PolK
         case POLKIT_RESULT_ONLY_VIA_SELF_AUTH_KEEP_ALWAYS:
                 /* g_debug ("blocking clicked"); */
                 g_signal_stop_emission_by_name (button, "clicked");
+                if (action->priv->polkit_action != NULL)
+                        g_signal_emit (action, signals [AUTH_START_SIGNAL], 0);
                 break;
 
         default:

