diff -up PolicyKit-gnome-0.9.2/src/polkit-gnome-manager.c.reregister PolicyKit-gnome-0.9.2/src/polkit-gnome-manager.c
--- PolicyKit-gnome-0.9.2/src/polkit-gnome-manager.c.reregister	2008-07-22 18:00:53.000000000 +0100
+++ PolicyKit-gnome-0.9.2/src/polkit-gnome-manager.c	2009-08-13 12:28:15.534034826 +0100
@@ -287,10 +287,6 @@ register_manager (PolKitGnomeManager *ma
         }
 
         dbus_g_connection_register_g_object (manager->priv->session_bus_connection, 
-                                             "/org/gnome/PolicyKit/Manager", 
-                                             G_OBJECT (manager));
-
-        dbus_g_connection_register_g_object (manager->priv->session_bus_connection, 
                                              "/", 
                                              G_OBJECT (manager));
 

