remove missing zh_CN locale

diff -ur wxMaxima-13.04.0.old/locales/Makefile.am wxMaxima-13.04.0/locales/Makefile.am
--- wxMaxima-13.04.0.old/locales/Makefile.am	2013-01-08 10:31:01.000000000 +0100
+++ wxMaxima-13.04.0/locales/Makefile.am	2013-04-22 18:42:27.026682308 +0200
@@ -2,8 +2,8 @@
 # This file is based on the corresponding file for the poedit program
 #
 
-WXMAXIMA_LINGUAS = fr es it de pt_BR ru hu uk pl zh_TW da cs el ja ca gl zh_CN
-WXWIN_LINGUAS =    fr es it de pt_BR ru hu uk pl zh_TW da cs el ja ca gl zh_CN
+WXMAXIMA_LINGUAS = fr es it de pt_BR ru hu uk pl zh_TW da cs el ja ca gl
+WXWIN_LINGUAS =    fr es it de pt_BR ru hu uk pl zh_TW da cs el ja ca gl
 
 localedir = $(datadir)/locale
 
@@ -112,8 +112,6 @@
 	$(MSGFMT) transcoded/ca.po -o transcoded/ca.mo
 	cp gl.po transcoded
 	$(MSGFMT) transcoded/gl.po -o transcoded/gl.mo
-	cp zh_CN.po transcoded
-	$(MSGFMT) transcoded/zh_CN.po -o transcoded/zh_CN.mo
 
 
 EXTRA_DIST = fr.po fr.mo wxwin/fr.mo \
diff -ur wxMaxima-13.04.0.old/locales/Makefile.in wxMaxima-13.04.0/locales/Makefile.in
--- wxMaxima-13.04.0.old/locales/Makefile.in	2013-04-04 10:56:20.000000000 +0200
+++ wxMaxima-13.04.0/locales/Makefile.in	2013-04-22 18:42:46.690681857 +0200
@@ -162,8 +162,8 @@
 target_vendor = @target_vendor@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-WXMAXIMA_LINGUAS = fr es it de pt_BR ru hu uk pl zh_TW da cs el ja ca gl zh_CN
-WXWIN_LINGUAS = fr es it de pt_BR ru hu uk pl zh_TW da cs el ja ca gl zh_CN
+WXMAXIMA_LINGUAS = fr es it de pt_BR ru hu uk pl zh_TW da cs el ja ca gl
+WXWIN_LINGUAS = fr es it de pt_BR ru hu uk pl zh_TW da cs el ja ca gl
 
 # ----------------------------------------------------------------------------
 # Logic for catalogs updating follows
@@ -445,8 +445,6 @@
 	$(MSGFMT) transcoded/ca.po -o transcoded/ca.mo
 	cp gl.po transcoded
 	$(MSGFMT) transcoded/gl.po -o transcoded/gl.mo
-	cp zh_CN.po transcoded
-	$(MSGFMT) transcoded/zh_CN.po -o transcoded/zh_CN.mo
 
 .PHONY: allpo allmo force-update stats FORCE
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
diff -ur wxMaxima-13.04.0.old/Makefile.am wxMaxima-13.04.0/Makefile.am
--- wxMaxima-13.04.0.old/Makefile.am	2013-01-08 10:31:01.000000000 +0100
+++ wxMaxima-13.04.0/Makefile.am	2013-04-22 18:41:40.304683380 +0200
@@ -23,7 +23,7 @@
 doc_DATA = README COPYING art/config/options.png art/config/maxima.png \
 	art/config/styles.png
 
-LANGS = fr es it de pt_BR ru hu uk pl zh_TW da cs el ja ca gl zh_CN
+LANGS = fr es it de pt_BR ru hu uk pl zh_TW da cs el ja ca gl
 
 wxMaxima.app: all
 	mkdir -p wxMaxima.app/Contents/MacOS
diff -ur wxMaxima-13.04.0.old/Makefile.in wxMaxima-13.04.0/Makefile.in
--- wxMaxima-13.04.0.old/Makefile.in	2013-04-04 10:56:20.000000000 +0200
+++ wxMaxima-13.04.0/Makefile.in	2013-04-22 18:41:52.945683090 +0200
@@ -219,7 +219,7 @@
 doc_DATA = README COPYING art/config/options.png art/config/maxima.png \
 	art/config/styles.png
 
-LANGS = fr es it de pt_BR ru hu uk pl zh_TW da cs el ja ca gl zh_CN
+LANGS = fr es it de pt_BR ru hu uk pl zh_TW da cs el ja ca gl
 all: all-recursive
 
 .SUFFIXES:
