diff -Naur gsoap-2.7.orig/soapcpp2/stdsoap2.cpp gsoap-2.7/soapcpp2/stdsoap2.cpp
--- gsoap-2.7.orig/soapcpp2/stdsoap2.cpp	2007-01-01 18:35:56.000000000 +0100
+++ gsoap-2.7/soapcpp2/stdsoap2.cpp	2007-01-06 20:43:30.000000000 +0100
@@ -3832,13 +3832,13 @@
               data = ext->value->data;
 #if (OPENSSL_VERSION_NUMBER > 0x00907000L)
               if (meth->it) 
-                ext_data = ASN1_item_d2i(NULL, &data, ext->value->length, ASN1_ITEM_ptr(meth->it));
+                ext_data = ASN1_item_d2i(NULL, (const unsigned char**)&data, ext->value->length, ASN1_ITEM_ptr(meth->it));
               else
 	      { /* OpenSSL not perfectly portable at this point (?):
                    Some compilers appear to prefer
                    meth->d2i(NULL, (const unsigned char**)&data, ...
 	        */
-                ext_data = meth->d2i(NULL, &data, ext->value->length);
+                ext_data = meth->d2i(NULL, (const unsigned char**)&data, ext->value->length);
 	      }
 #else
               ext_data = meth->d2i(NULL, &data, ext->value->length);
