--- gsoap-2.7/soapcpp2/stdsoap2.cpp.orig	2006-06-27 13:18:16.000000000 +0200
+++ gsoap-2.7/soapcpp2/stdsoap2.cpp	2006-06-27 13:18:35.000000000 +0200
@@ -3755,7 +3755,7 @@
             X509V3_EXT_METHOD *meth = X509V3_EXT_get(ext);
             if (!meth)
               break;
-            val = meth->i2v(meth, meth->d2i(NULL, (unsigned char**)&ext->value->data, ext->value->length), NULL);
+            val = meth->i2v(meth, meth->d2i(NULL, (unsigned const char**)&ext->value->data, ext->value->length), NULL);
             for (j = 0; j < sk_CONF_VALUE_num(val); j++)
             { CONF_VALUE *nval = sk_CONF_VALUE_value(val, j);
               if (nval && !strcmp(nval->name, "DNS") && !strcmp(nval->value, host))
