--- gsoap-2.7/soapcpp2/stdsoap2.cpp.orig	2007-02-22 18:01:35.000000000 +0100
+++ gsoap-2.7/soapcpp2/stdsoap2.cpp	2007-02-22 18:02:01.000000000 +0100
@@ -3954,7 +3954,7 @@
               data = ext->value->data;
 #if (OPENSSL_VERSION_NUMBER > 0x00907000L)
               if (meth->it) 
-                ext_data = ASN1_item_d2i(NULL, &data, ext->value->length, ASN1_ITEM_ptr(meth->it));
+                ext_data = ASN1_item_d2i(NULL, (const unsigned char**)&data, ext->value->length, ASN1_ITEM_ptr(meth->it));
               else
 	      { /* OpenSSL not perfectly portable at this point (?):
                    Some compilers appear to prefer
@@ -3962,7 +3962,7 @@
 		   or
                    meth->d2i(NULL, &data, ext->value->length);
 	        */
-                ext_data = meth->d2i(NULL, &data, ext->value->length);
+                ext_data = meth->d2i(NULL, (const unsigned char**)&data, ext->value->length);
 	      }
 #else
               ext_data = meth->d2i(NULL, &data, ext->value->length);
