diff -Naur pidgin-audacious-2.0.0/Makefile.in pidgin-audacious-2.0.0.new/Makefile.in
--- pidgin-audacious-2.0.0/Makefile.in	2007-09-02 10:20:47.000000000 +0530
+++ pidgin-audacious-2.0.0.new/Makefile.in	2008-04-30 18:10:58.000000000 +0530
@@ -1,7 +1,7 @@
 PIDGIN_CFLAGS = @PIDGIN_CFLAGS@
 AUD_CFLAGS = @AUD_CFLAGS@
 DBUS_GLIB_CFLAGS = @DBUS_GLIB_CFLAGS@
-CFLAGS = -fPIC -shared -Wall $(PIDGIN_CFLAGS) $(AUD_CFLAGS) $(DBUS_GLIB_CFLAGS)
+CFLAGS += -fPIC -shared -Wall $(PIDGIN_CFLAGS) $(AUD_CFLAGS) $(DBUS_GLIB_CFLAGS)
 
 AUD_LIBS = @AUD_LIBS@
 AUD_LIB_DIR = @AUD_LIB_DIR@
@@ -15,11 +15,12 @@
 default: pidgin-audacious.so
 
 pidgin-audacious.so: pidgin-audacious.c
-	gcc -o $@ $< $(CFLAGS) $(LDFLAGS) -g -Wl,-rpath=$(AUD_LIB_DIR)
+	gcc -o $@ $< $(CFLAGS) $(LDFLAGS) -g
 
 
 install: pidgin-audacious.so
-	install -m 755 pidgin-audacious.so $(PIDGIN_LIB_DIR)/pidgin
+	install -d $(DESTDIR)$(PIDGIN_LIB_DIR)/pidgin
+	install -m 755 pidgin-audacious.so $(DESTDIR)$(PIDGIN_LIB_DIR)/pidgin
 
 
 clean:
