--- xerces-c-src/src/xercesc/Makefile.incl.orig	2006-01-08 20:05:35.000000000 -0500
+++ xerces-c-src/src/xercesc/Makefile.incl	2006-01-08 20:08:19.000000000 -0500
@@ -348,23 +348,23 @@
 
         ifeq (${BITSTOBUILD}, 64)
             ifeq (${TRANSCODER}, ICU)
-                ALLLIBS = ${LIBS} -L/usr/lib64 -L/usr/lib -L/usr/local/lib -L/usr/ccs/lib -licuuc -licudata
+                ALLLIBS = ${LIBS} -L${XERCESCROOT}/lib -licuuc -licudata
             else
-                ALLLIBS = ${LIBS} -L/usr/lib64 -L/usr/lib -L/usr/local/lib -L/usr/ccs/lib
+                ALLLIBS = ${LIBS} -L${XERCESCROOT}/lib
             endif
 
             ifeq (${MESSAGELOADER}, ICU)
-                ALLLIBS = ${LIBS} -L/usr/lib64 -L/usr/lib -L/usr/local/lib -L/usr/ccs/lib -licuuc -licudata -lXercesMessages
+                ALLLIBS = ${LIBS} -L${XERCESCROOT}/lib -licuuc -licudata -lXercesMessages
             endif
         else
             ifeq (${TRANSCODER}, ICU)
-                ALLLIBS = ${LIBS} -L/usr/lib -L/usr/local/lib -L/usr/ccs/lib -licuuc -licudata
+                ALLLIBS = ${LIBS} -L${XERCESCROOT}/lib -licuuc -licudata
             else
-                ALLLIBS = ${LIBS} -L/usr/lib -L/usr/local/lib -L/usr/ccs/lib
+                ALLLIBS = ${LIBS} -L${XERCESCROOT}/lib
             endif
 
             ifeq (${MESSAGELOADER}, ICU)
-                ALLLIBS = ${LIBS} -L/usr/lib -L/usr/local/lib -L/usr/ccs/lib -licuuc -licudata -lXercesMessages
+                ALLLIBS = ${LIBS} -L${XERCESCROOT}/lib -licuuc -licudata -lXercesMessages
             endif
         endif
 
