From 87fcb64f23d37c0f14c091674449b43bd4b031fe Mon Sep 17 00:00:00 2001
From: Christoph Willing <chris.willing@linux.com>
Date: Mon, 26 Mar 2018 20:53:31 +1000
Subject: [PATCH] Also find libtoxav & libtoxencryptsave

Signed-off-by: Christoph Willing <chris.willing@linux.com>
---
 cmake/Findlibtox.cmake | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/cmake/Findlibtox.cmake b/cmake/Findlibtox.cmake
index 99ebd4ed4..9b9ebbf0e 100644
--- a/cmake/Findlibtox.cmake
+++ b/cmake/Findlibtox.cmake
@@ -19,6 +19,14 @@ find_library(LIBTOX_LIBRARY NAMES toxcore HINTS
     ${PKG_LIBTOX_LIBDIR}
     ${PKG_LIBTOX_LIBRARY_DIRS}
 )
+find_library(LIBTOXAV_LIBRARY NAMES toxav HINTS
+    ${PKG_LIBTOXAV_LIBDIR}
+    ${PKG_LIBTOXAV_LIBRARY_DIRS}
+)
+find_library(LIBTOXENCSAVE_LIBRARY NAMES toxencryptsave HINTS
+    ${PKG_LIBTOXENCSAVE_LIBDIR}
+    ${PKG_LIBTOXENCSAVE_LIBRARY_DIRS}
+)
 
 include(FindPackageHandleStandardArgs)
 # Handle the QUIETLY and REQUIRED arguments and set LIBTOX_FOUND to TRUE
@@ -27,10 +35,12 @@ find_package_handle_standard_args(
     libtox
     DEFAULT_MSG
     LIBTOX_LIBRARY
+    LIBTOXAV_LIBRARY
+    LIBTOXENCSAVE_LIBRARY
     LIBTOX_INCLUDE_DIR
 )
 
 mark_as_advanced(LIBTOX_INCLUDE_DIR LIBTOX_LIBRARY)
 
-set(LIBTOX_LIBRARIES ${LIBTOX_LIBRARY})
+set(LIBTOX_LIBRARIES ${LIBTOX_LIBRARY} ${LIBTOXAV_LIBRARY} ${LIBTOXENCSAVE_LIBRARY})
 set(LIBTOX_INCLUDE_DIRS ${LIBTOX_INCLUDE_DIR})
