diff -ur a/nucleo/helpers/Phone.cxx b/nucleo/helpers/Phone.cxx
--- a/nucleo/helpers/Phone.cxx	2009-06-11 03:48:06.000000000 -0500
+++ b/nucleo/helpers/Phone.cxx	2014-03-14 16:37:48.458822232 -0500
@@ -18,6 +18,7 @@
 #include <fcntl.h>
 #include <sys/ioctl.h>
 #include <errno.h>
+#include <unistd.h>
 
 #include <iostream>
 #include <sstream>
diff -ur a/nucleo/image/sink/bufferedImageSink.cxx b/nucleo/image/sink/bufferedImageSink.cxx
--- a/nucleo/image/sink/bufferedImageSink.cxx	2008-06-05 07:52:33.000000000 -0500
+++ b/nucleo/image/sink/bufferedImageSink.cxx	2014-03-14 16:36:19.212825517 -0500
@@ -10,6 +10,7 @@
  */
 
 #include <nucleo/image/sink/bufferedImageSink.H>
+#include <unistd.h>
 
 #define DEBUG_LEVEL 1
 
diff -ur a/nucleo/image/sink/novImageSink.cxx b/nucleo/image/sink/novImageSink.cxx
--- a/nucleo/image/sink/novImageSink.cxx	2008-06-05 07:52:33.000000000 -0500
+++ b/nucleo/image/sink/novImageSink.cxx	2014-03-14 16:35:47.121826698 -0500
@@ -18,6 +18,7 @@
 #include <nucleo/image/encoding/Conversion.H>
 
 #include <sys/uio.h>
+#include <unistd.h>
 
 #include <stdexcept>
 
diff -ur a/nucleo/image/sink/nserverImageSink.H b/nucleo/image/sink/nserverImageSink.H
--- a/nucleo/image/sink/nserverImageSink.H	2008-06-05 07:52:33.000000000 -0500
+++ b/nucleo/image/sink/nserverImageSink.H	2014-03-14 17:11:26.258747967 -0500
@@ -24,6 +24,7 @@
 #include <stdexcept>
 
 #include <sys/socket.h>
+#include <unistd.h>
 
 namespace nucleo {
 
diff -ur a/nucleo/image/source/v4lImageSource.H b/nucleo/image/source/v4lImageSource.H
--- a/nucleo/image/source/v4lImageSource.H	2008-06-05 07:52:33.000000000 -0500
+++ b/nucleo/image/source/v4lImageSource.H	2014-03-14 16:34:22.327829819 -0500
@@ -19,7 +19,7 @@
 #include <nucleo/core/URI.H>
 
 #include <linux/types.h>
-#include <linux/videodev.h>
+#include <libv4l1-videodev.h>
 
 namespace nucleo {
 
diff -ur a/nucleo/network/NetworkUtils.cxx b/nucleo/network/NetworkUtils.cxx
--- a/nucleo/network/NetworkUtils.cxx	2008-06-05 07:52:33.000000000 -0500
+++ b/nucleo/network/NetworkUtils.cxx	2014-03-14 16:31:15.901836680 -0500
@@ -15,6 +15,7 @@
 
 #include <arpa/inet.h>
 #include <netdb.h>
+#include <unistd.h>
 
 #include <iostream>
 #include <string>
diff -ur a/nucleo/network/udp/UdpSocket.cxx b/nucleo/network/udp/UdpSocket.cxx
--- a/nucleo/network/udp/UdpSocket.cxx	2009-06-11 03:48:06.000000000 -0500
+++ b/nucleo/network/udp/UdpSocket.cxx	2014-03-14 16:29:27.053840686 -0500
@@ -23,6 +23,7 @@
 #include <cstring>
 #include <cstdlib>
 #include <cstdio>
+#include <unistd.h>
 
 namespace nucleo {
 
diff -ur a/nucleo/utils/AppUtils.cxx b/nucleo/utils/AppUtils.cxx
--- a/nucleo/utils/AppUtils.cxx	2008-06-09 07:09:05.000000000 -0500
+++ b/nucleo/utils/AppUtils.cxx	2014-03-14 16:26:18.862847613 -0500
@@ -21,8 +21,7 @@
 #include <cstdlib>
 #include <cstring>
 
-extern char* optarg ;
-extern int optind ;
+#include <unistd.h>
 
 namespace nucleo {
 
diff -ur a/tests/test-Base64.cxx b/tests/test-Base64.cxx
--- a/tests/test-Base64.cxx	2008-06-09 08:21:20.000000000 -0500
+++ b/tests/test-Base64.cxx	2014-03-14 17:13:13.546744018 -0500
@@ -14,6 +14,7 @@
 
 #include <iostream>
 #include <cstdlib>
+#include <unistd.h>
 
 using namespace nucleo ;
 
diff -ur a/tests/test-MD5.cxx b/tests/test-MD5.cxx
--- a/tests/test-MD5.cxx	2008-06-05 07:52:33.000000000 -0500
+++ b/tests/test-MD5.cxx	2014-03-14 17:12:58.801744561 -0500
@@ -12,6 +12,7 @@
 #include <nucleo/utils/MD5.H>
 
 #include <iostream>
+#include <unistd.h>
 
 using namespace nucleo ;
 
