diff -Naur nucleo-0.7.5/demos/video/src2files.cxx nucleo-0.7.5/demos/video/src2files.cxx
--- nucleo-0.7.5/demos/video/src2files.cxx	2008-06-09 15:05:02.000000000 +0200
+++ nucleo-0.7.5/demos/video/src2files.cxx	2009-06-09 10:56:58.000000000 +0200
@@ -17,6 +17,7 @@
 
 #include <stdexcept>
 #include <cstdlib>
+#include <cstdio>
 
 #include <libgen.h>
 
diff -Naur nucleo-0.7.5/nucleo/core/TimeStamp.cxx nucleo-0.7.5/nucleo/core/TimeStamp.cxx
--- nucleo-0.7.5/nucleo/core/TimeStamp.cxx	2008-06-09 14:07:49.000000000 +0200
+++ nucleo-0.7.5/nucleo/core/TimeStamp.cxx	2009-06-09 10:48:46.000000000 +0200
@@ -22,6 +22,7 @@
 #include <sstream>
 #include <stdexcept>
 #include <cstring>
+#include <cstdio>
 
 namespace nucleo {
 
diff -Naur nucleo-0.7.5/nucleo/core/URI.cxx nucleo-0.7.5/nucleo/core/URI.cxx
--- nucleo-0.7.5/nucleo/core/URI.cxx	2008-07-21 13:47:54.000000000 +0200
+++ nucleo-0.7.5/nucleo/core/URI.cxx	2009-06-09 10:48:48.000000000 +0200
@@ -17,7 +17,7 @@
 #include <sstream>
 #include <cstdlib>
 
-// #include <cstdio>
+#include <cstdio>
 
 static void
 split(std::string &s, const std::string splitters, std::string &part, bool keepsep, bool preserve) {
diff -Naur nucleo-0.7.5/nucleo/helpers/Phone.cxx nucleo-0.7.5/nucleo/helpers/Phone.cxx
--- nucleo-0.7.5/nucleo/helpers/Phone.cxx	2008-06-05 14:52:33.000000000 +0200
+++ nucleo-0.7.5/nucleo/helpers/Phone.cxx	2009-06-09 10:50:57.000000000 +0200
@@ -21,6 +21,7 @@
 
 #include <iostream>
 #include <sstream>
+#include <cstdio>
 
 using namespace nucleo ;
 
diff -Naur nucleo-0.7.5/nucleo/network/udp/UdpSocket.cxx nucleo-0.7.5/nucleo/network/udp/UdpSocket.cxx
--- nucleo-0.7.5/nucleo/network/udp/UdpSocket.cxx	2008-06-09 14:15:38.000000000 +0200
+++ nucleo-0.7.5/nucleo/network/udp/UdpSocket.cxx	2009-06-09 10:49:43.000000000 +0200
@@ -22,6 +22,7 @@
 #include <sstream>
 #include <cstring>
 #include <cstdlib>
+#include <cstdio>
 
 namespace nucleo {
 
diff -Naur nucleo-0.7.5/nucleo/plugins/ffmpeg/ffmpegImageSink.cxx nucleo-0.7.5/nucleo/plugins/ffmpeg/ffmpegImageSink.cxx
--- nucleo-0.7.5/nucleo/plugins/ffmpeg/ffmpegImageSink.cxx	2009-05-25 14:53:15.000000000 +0200
+++ nucleo-0.7.5/nucleo/plugins/ffmpeg/ffmpegImageSink.cxx	2009-06-09 10:55:59.000000000 +0200
@@ -146,7 +146,7 @@
 
     // ----------------
 
-    format_context = avformat_alloc_context();
+    format_context = av_alloc_format_context();
     if (!format_context) {
 	 std::cerr << "ffmpegImageSink: unable to create the AVFormatContext" << std::endl ;
 	 stop() ;
