diff -Naur dnsmasq-2.33/Makefile dnsmasq-2.33-p/Makefile
--- dnsmasq-2.33/Makefile	2006-08-02 21:28:52.000000000 +0200
+++ dnsmasq-2.33-p/Makefile	2006-08-06 15:03:17.890770208 +0200
@@ -1,6 +1,6 @@
-PREFIX?=/usr/local
+PREFIX?=/usr
 BINDIR = ${PREFIX}/sbin
-MANDIR = ${PREFIX}/share/man
+MANDIR = ${PREFIX}/man
 LOCALEDIR = ${PREFIX}/share/locale
 
 SRC = src
@@ -18,7 +18,7 @@
 	rm -f *~ $(SRC)/*.mo contrib/*/*~ */*~ $(SRC)/*.pot 
 	rm -f $(SRC)/*.o $(SRC)/dnsmasq.a $(SRC)/dnsmasq core */core
 
-install : all install-common
+install : all install-common install-i18n
 
 install-common :
 	install -d $(DESTDIR)$(BINDIR) -d $(DESTDIR)$(MANDIR)/man8
