diff -Naur dnsmasq-2.41/Makefile dnsmasq-2.41-p/Makefile
--- dnsmasq-2.41/Makefile	2007-10-21 22:02:03.000000000 +0200
+++ dnsmasq-2.41-p/Makefile	2008-03-15 00:25:06.000000000 +0100
@@ -13,7 +13,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-PREFIX = /usr/local
+PREFIX ?= /usr
 BINDIR = ${PREFIX}/sbin
 MANDIR = ${PREFIX}/share/man
 LOCALEDIR = ${PREFIX}/share/locale
@@ -47,7 +47,7 @@
 	rm -f *~ $(SRC)/*.mo contrib/*/*~ */*~ $(SRC)/*.pot 
 	rm -f $(SRC)/*.o $(SRC)/dnsmasq.a $(SRC)/dnsmasq core */core
 
-install : all install-common
+install : all install-common install-i18n
 
 install-common :
 	$(INSTALL) -d $(DESTDIR)$(BINDIR) -d $(DESTDIR)$(MANDIR)/man8
