diff -urN dnsmasq-2.39.orig/Makefile dnsmasq-2.39/Makefile
--- dnsmasq-2.39.orig/Makefile	2007-05-07 17:43:09.000000000 +0100
+++ dnsmasq-2.39/Makefile	2007-05-07 17:43:22.000000000 +0100
@@ -1,4 +1,4 @@
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 BINDIR ?= ${PREFIX}/sbin
 MANDIR ?= ${PREFIX}/share/man
 LOCALEDIR ?= ${PREFIX}/share/locale
@@ -18,7 +18,7 @@
 	rm -f *~ $(SRC)/*.mo contrib/*/*~ */*~ $(SRC)/*.pot 
 	rm -f $(SRC)/*.o $(SRC)/dnsmasq.a $(SRC)/dnsmasq core */core
 
-install : all install-common
+install : all install-common install-i18n
 
 install-common :
 	install -d $(DESTDIR)$(BINDIR) -d $(DESTDIR)$(MANDIR)/man8
