--- dnsmasq-2.89/Makefile	2023-02-02 21:24:24.000000000 +0100
+++ dnsmasq-2.89.new/Makefile	2024-01-15 19:34:11.295800603 +0100
@@ -18,15 +18,15 @@
 
 # Variables you may well want to override.
 
-PREFIX        = /usr/local
-BINDIR        = $(PREFIX)/sbin
+PREFIX        = /usr
+BINDIR        = $(PREFIX)/bin
 MANDIR        = $(PREFIX)/share/man
 LOCALEDIR     = $(PREFIX)/share/locale
 BUILDDIR      = $(SRC)
 DESTDIR       = 
 CFLAGS        = -Wall -W -O2
 LDFLAGS       = 
-COPTS         = 
+COPTS         = -DHAVE_DBUS
 RPM_OPT_FLAGS = 
 LIBS          = 
 
@@ -105,7 +105,7 @@
 	rm -f core */core
 	rm -f *~ contrib/*/*~ */*~
 
-install : all install-common
+install : all install-common install-i18n
 
 install-common :
 	$(INSTALL) -d $(DESTDIR)$(BINDIR)
