diff -urN avahi-0.6.19.orig/avahi-ui-sharp/Makefile.am avahi-0.6.19/avahi-ui-sharp/Makefile.am
--- avahi-0.6.19.orig/avahi-ui-sharp/Makefile.am	2007-05-12 08:31:27.000000000 +0100
+++ avahi-0.6.19/avahi-ui-sharp/Makefile.am	2007-05-12 08:32:38.000000000 +0100
@@ -40,7 +40,7 @@
 	$(srcdir)/gencfg.sh $(top_builddir)/avahi-common/libavahi-common.la < $< > $@
 
 zssh.exe: $(srcdir)/zssh.cs
-	mcs -out:$@ $^ -r:./avahi-ui-sharp.dll -pkg:gtk-sharp-2.0 -r:Mono.Posix
+	mcs -out:$@ $^ -r:./avahi-ui-sharp.dll -r:../avahi-sharp/avahi-sharp.dll -pkg:gtk-sharp-2.0 -r:Mono.Posix
 
 if HAVE_MONO
 if HAVE_DBUS
