diff -Naur coreutils-8.29/src/dircolors.hin coreutils-8.29-p/src/dircolors.hin
--- coreutils-8.29/src/dircolors.hin	2017-01-21 17:05:58.000000000 +0100
+++ coreutils-8.29-p/src/dircolors.hin	2018-01-09 03:14:45.704698551 +0100
@@ -8,6 +8,9 @@
 # The keywords COLOR, OPTIONS, and EIGHTBIT (honored by the
 # slackware version of dircolors) are recognized but ignored.
 
+# You can copy this file to .dir_colors in your $HOME directory to override
+# the system defaults.
+
 # Below are TERM entries, which can be a glob patterns, to match
 # against the TERM environment variable to determine if it is colorizable.
 TERM Eterm
@@ -55,8 +58,8 @@
 DOOR 01;35	# door
 BLK 40;33;01	# block device driver
 CHR 40;33;01	# character device driver
-ORPHAN 40;31;01 # symlink to nonexistent file, or non-stat'able file ...
-MISSING 00      # ... and the files they point to
+ORPHAN 01;05;37;41  # symlink to nonexistent file, or non-stat'able file ...
+MISSING 01;05;37;41 # ... and the files they point to
 SETUID 37;41	# file that is setuid (u+s)
 SETGID 30;43	# file that is setgid (g+s)
 CAPABILITY 30;41	# file with capability
@@ -130,6 +133,7 @@
 .swm 01;31
 .dwm 01;31
 .esd 01;31
+.fpm 01;31
 
 # image formats
 .jpg 01;35
@@ -184,6 +188,19 @@
 .ogv 01;35
 .ogx 01;35
 
+# Text/document files
+.cfg 00;32
+.conf 00;32
+.diff 00;32
+.doc 00;32
+.ini 00;32
+.log 00;32
+.patch 00;32
+.pdf 00;32
+.ps 00;32
+.tex 00;32
+.txt 00;32
+
 # audio formats
 .aac 00;36
 .au 00;36
