diff -Naur coreutils-8.24/src/ls.c coreutils-8.24-ls/src/ls.c
--- coreutils-8.24/src/ls.c	2015-06-26 19:05:22.000000000 +0200
+++ coreutils-8.24-ls/src/ls.c	2015-10-04 17:50:44.743178376 +0200
@@ -2315,8 +2315,8 @@
   if (! term || ! *term)
     return false;
 
-  char const *line = G_line;
-  while (line - G_line < sizeof (G_line))
+  char const *line = (char const *)G_line;
+  while (line - (char const *)G_line < sizeof (G_line))
     {
       if (STRNCMP_LIT (line, "TERM ") == 0)
         {
