diff -Naur coreutils-8.24/Makefile.in coreutils-8.24-man/Makefile.in
--- coreutils-8.24/Makefile.in	2015-07-03 23:23:13.000000000 +0200
+++ coreutils-8.24-man/Makefile.in	2015-10-04 18:44:35.345124319 +0200
@@ -14156,7 +14156,6 @@
 @SINGLE_BINARY_FALSE@man/dd.1:        src/dd$(EXEEXT)
 @SINGLE_BINARY_FALSE@man/df.1:        src/df$(EXEEXT)
 @SINGLE_BINARY_FALSE@man/dir.1:       src/dir$(EXEEXT)
-@SINGLE_BINARY_FALSE@man/dircolors.1: src/dircolors$(EXEEXT)
 @SINGLE_BINARY_FALSE@man/dirname.1:   src/dirname$(EXEEXT)
 @SINGLE_BINARY_FALSE@man/du.1:        src/du$(EXEEXT)
 @SINGLE_BINARY_FALSE@man/echo.1:      src/echo$(EXEEXT)
diff -Naur coreutils-8.24/man/local.mk coreutils-8.24-man/man/local.mk
--- coreutils-8.24/man/local.mk	2015-06-26 19:04:19.000000000 +0200
+++ coreutils-8.24-man/man/local.mk	2015-10-04 18:45:24.043373074 +0200
@@ -78,7 +78,6 @@
 man/dd.1:        src/dd$(EXEEXT)
 man/df.1:        src/df$(EXEEXT)
 man/dir.1:       src/dir$(EXEEXT)
-man/dircolors.1: src/dircolors$(EXEEXT)
 man/dirname.1:   src/dirname$(EXEEXT)
 man/du.1:        src/du$(EXEEXT)
 man/echo.1:      src/echo$(EXEEXT)
