diff -Naurd qtparted-0.4.4.orig/data/Makefile.in qtparted-0.4.4/data/Makefile.in
--- qtparted-0.4.4.orig/data/Makefile.in	2004-04-25 18:19:46.000000000 +0200
+++ qtparted-0.4.4/data/Makefile.in	2005-07-23 15:10:34.000000000 +0200
@@ -213,7 +213,7 @@
 splashdir = $(prefix)/share/qtparted/pics
 
 menu_DATA = qtparted.desktop
-menudir = $(prefix)/share/applnk/System
+menudir = $(prefix)/share/applications
 
 script_DATA = run_qtparted
 scriptdir = $(prefix)/sbin
diff -Naurd qtparted-0.4.4.orig/data/qtparted.desktop qtparted-0.4.4/data/qtparted.desktop
--- qtparted-0.4.4.orig/data/qtparted.desktop	2004-04-24 18:41:27.000000000 +0200
+++ qtparted-0.4.4/data/qtparted.desktop	2005-07-23 15:20:50.000000000 +0200
@@ -8,5 +8,6 @@
 Comment[it]=Applicazione per il partizionamento di un disco
 Comment[fr]=Application pour partitionner un disque
 Comment[fi]=Kiintolevyn partitiointityökalu
+Comment[hu]=Lemezparticionáló alkalmazás
 Encoding=UTF-8
 Categories=Application;System;QT;
