diff -Naur rpm-4.4.2.orig/configure rpm-4.4.2/configure
--- rpm-4.4.2.orig/configure	2005-06-15 11:13:39.000000000 +0000
+++ rpm-4.4.2/configure	2005-06-15 11:17:47.000000000 +0000
@@ -39528,9 +39528,9 @@
 fi
 
 MARK64=
-case "${target_cpu}" in
-x86_64*|ppc64*|powerpc64*|sparc64*|s390x*)	MARK64=64 ;;
-esac
+#case "${target_cpu}" in
+#x86_64*|ppc64*|powerpc64*|sparc64*|s390x*)	MARK64=64 ;;
+#esac
 
 
 autorelocate_path='%{nil}'
diff -Naur rpm-4.4.2.orig/popt/configure rpm-4.4.2/popt/configure
--- rpm-4.4.2.orig/popt/configure	2005-06-15 11:13:40.000000000 +0000
+++ rpm-4.4.2/popt/configure	2005-06-15 11:18:59.000000000 +0000
@@ -20966,9 +20966,9 @@
 
 
 MARK64=
-case "${target_cpu}" in
-x86_64*|powerpc64*|ppc64*|sparc64*|s390x*)        MARK64=64 ;;
-esac
+#case "${target_cpu}" in
+#x86_64*|powerpc64*|ppc64*|sparc64*|s390x*)        MARK64=64 ;;
+#esac
 
 
 
diff -Naur rpm-4.4.2.orig/scripts/Makefile.in rpm-4.4.2/scripts/Makefile.in
--- rpm-4.4.2.orig/scripts/Makefile.in	2005-06-15 11:13:39.000000000 +0000
+++ rpm-4.4.2/scripts/Makefile.in	2005-06-15 11:19:32.000000000 +0000
@@ -313,7 +313,7 @@
 	macros.php* find-*.php find-php-*
 
 installprefix = $(DESTDIR)
-configdir = ${prefix}/lib/rpm
+configdir = ${libdir}/rpm
 config_SCRIPTS = \
 	brp-compress brp-python-bytecompile brp-redhat \
 	brp-strip brp-strip-comment-note \
