diff --git a/src/r8169_n.c b/src/r8169_n.c
index c0b7d0f..83e4755 100644
--- a/src/r8169_n.c
+++ b/src/r8169_n.c
@@ -1035,9 +1035,14 @@ struct rtl8169_counters {
 	u16	tx_underun;
 };
 
-static int rtl8169_get_stats_count(struct net_device *dev)
+static int rtl8169_get_sset_count(struct net_device *dev, int stringset)
 {
-	return ARRAY_SIZE(rtl8169_gstrings);
+	switch (stringset) {
+	case ETH_SS_STATS:
+		return ARRAY_SIZE(rtl8169_gstrings);
+	default:
+		return -EINVAL;
+	}
 }
 
 static void rtl8169_get_ethtool_stats(struct net_device *dev,
@@ -1148,7 +1153,7 @@ static struct ethtool_ops rtl8169_ethtool_ops = {
 	.get_wol		= rtl8169_get_wol,
 	.set_wol		= rtl8169_set_wol,
 	.get_strings		= rtl8169_get_strings,
-	.get_stats_count	= rtl8169_get_stats_count,
+	.get_sset_count	= rtl8169_get_sset_count,
 	.get_ethtool_stats	= rtl8169_get_ethtool_stats,
 #if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,23) 
 #ifdef ETHTOOL_GPERMADDR
