diff --git a/src/r8169_n.c b/src/r8169_n.c
index 1f24f50..c0b7d0f 100644
--- a/src/r8169_n.c
+++ b/src/r8169_n.c
@@ -2459,6 +2459,21 @@ rtl8169_esd_timer(unsigned long __opaque)
 	mod_timer(timer, jiffies + timeout);
 }
 
+static const struct net_device_ops rtl8169_netdev_ops = {
+	.ndo_do_ioctl           = rtl8169_ioctl,
+	.ndo_open               = rtl8169_open,
+	.ndo_start_xmit         = rtl8169_start_xmit,
+	.ndo_get_stats          = rtl8169_get_stats,
+	.ndo_stop               = rtl8169_close,
+	.ndo_tx_timeout         = rtl8169_tx_timeout,
+	.ndo_set_multicast_list = rtl8169_set_rx_mode,
+	.ndo_change_mtu         = rtl8169_change_mtu,
+	.ndo_set_mac_address    = rtl8169_set_mac_address,
+#ifdef CONFIG_NET_POLL_CONTROLLER
+	.ndo_poll_controller = rtl8169_netpoll,
+#endif
+};
+
 static int __devinit
 rtl8169_init_one(struct pci_dev *pdev, const struct pci_device_id *ent)
 {
@@ -2626,7 +2641,7 @@ rtl8169_init_one(struct pci_dev *pdev, const struct pci_device_id *ent)
 	RTL_W8(Cfg9346, Cfg9346_Unlock);
 	RTL_W8(Cfg9346, Cfg9346_Lock);
 
-	dev->do_ioctl = rtl8169_ioctl;
+	dev->netdev_ops = &rtl8169_netdev_ops;
 
 	/* Get MAC address.  FIXME: read EEPROM */
 	for (i = 0; i < MAC_ADDR_LEN; i++)
@@ -2636,18 +2651,10 @@ rtl8169_init_one(struct pci_dev *pdev, const struct pci_device_id *ent)
 #endif
 	memcpy(dev->dev_addr, dev->dev_addr, dev->addr_len);
 
-	dev->open = rtl8169_open;
-	dev->hard_start_xmit = rtl8169_start_xmit;
-	dev->get_stats = rtl8169_get_stats;
 	SET_ETHTOOL_OPS(dev, &rtl8169_ethtool_ops);
-	dev->stop = rtl8169_close;
-	dev->tx_timeout = rtl8169_tx_timeout;
-	dev->set_multicast_list = rtl8169_set_rx_mode;
 	dev->watchdog_timeo = RTL8169_TX_TIMEOUT;
 	dev->irq = pdev->irq;
 	dev->base_addr = (unsigned long) ioaddr;
-	dev->change_mtu = rtl8169_change_mtu;
-	dev->set_mac_address = rtl8169_set_mac_address;
 
 #ifdef CONFIG_R8169_NAPI
 	RTL_NAPI_CONFIG(dev, tp, rtl8169_poll, R8169_NAPI_WEIGHT);
@@ -2661,10 +2668,6 @@ rtl8169_init_one(struct pci_dev *pdev, const struct pci_device_id *ent)
 #endif //LINUX_VERSION_CODE < KERNEL_VERSION(2,6,22)
 #endif //CONFIG_R8169_VLAN
 
-#ifdef CONFIG_NET_POLL_CONTROLLER
-	dev->poll_controller = rtl8169_netpoll;
-#endif
-
 	dev->features |= NETIF_F_IP_CSUM;
 	tp->cp_cmd |= RxChkSum;
 	tp->cp_cmd |= RTL_R16(CPlusCmd);
