diff -aur r8169-6.004.00/src/r8169_n.c fw_r8169-6.004.00/src/r8169_n.c
--- r8169-6.004.00/src/r8169_n.c	2007-11-08 18:01:13.000000000 +0100
+++ fw_r8169-6.004.00/src/r8169_n.c	2007-11-29 18:10:31.000000000 +0100
@@ -1105,9 +1105,6 @@
 	.get_strings		= rtl8169_get_strings,
 	.get_stats_count	= rtl8169_get_stats_count,
 	.get_ethtool_stats	= rtl8169_get_ethtool_stats,
-#ifdef ETHTOOL_GPERMADDR
-	.get_perm_addr		= ethtool_op_get_perm_addr,
-#endif
 };
 
 static void rtl8169_get_mac_version(struct rtl8169_private *tp, void __iomem *ioaddr)
@@ -3267,7 +3264,7 @@
 		skb = dev_alloc_skb(pkt_size + align);
 		if (skb) {
 			skb_reserve(skb, (align - 1) & (u32)skb->data);
-			eth_copy_and_sum(skb, sk_buff[0]->data, pkt_size, 0);
+			skb_copy_from_linear_data(*sk_buff, skb->data, pkt_size);
 			*sk_buff = skb;
 			rtl8169_mark_to_asic(desc, rx_buf_sz);
 			ret = 0;
