diff -aur r8169-6.003.00/src/r8169_n.c fw_r8169-6.003.00/src/r8169_n.c
--- r8169-6.003.00/src/r8169_n.c	2007-08-24 13:21:31.000000000 +0200
+++ fw_r8169-6.003.00/src/r8169_n.c	2007-11-02 01:07:53.000000000 +0100
@@ -1052,9 +1052,6 @@
 	.get_strings		= rtl8169_get_strings,
 	.get_stats_count	= rtl8169_get_stats_count,
 	.get_ethtool_stats	= rtl8169_get_ethtool_stats,
-#if LINUX_VERSION_CODE > KERNEL_VERSION(2,6,13)
-	.get_perm_addr		= ethtool_op_get_perm_addr,
-#endif
 };
 
 static void rtl8169_get_mac_version(struct rtl8169_private *tp, void __iomem *ioaddr)
@@ -2605,7 +2602,8 @@
 		skb = dev_alloc_skb(pkt_size + align);
 		if (skb) {
 			skb_reserve(skb, (align - 1) & (u32)skb->data);
-			eth_copy_and_sum(skb, sk_buff[0]->data, pkt_size, 0);
+			//eth_copy_and_sum(skb, sk_buff[0]->data, pkt_size, 0);
+			skb_copy_from_linear_data(*sk_buff, skb->data, pkt_size);
 			*sk_buff = skb;
 			rtl8169_mark_to_asic(desc, rx_buf_sz);
 			ret = 0;
