--- hslua-module-version-1.1.0/hslua-module-version.cabal	2001-09-09 03:46:40.000000000 +0200
+++ hslua-module-version-1.1.0.new/hslua-module-version.cabal	2023-12-12 19:51:23.411008678 +0100
@@ -34,7 +34,7 @@
                      , hslua-core        >= 2.3    && < 2.4
                      , hslua-marshalling >= 2.3    && < 2.4
                      , hslua-packaging   >= 2.3    && < 2.4
-                     , text              >= 1.2    && < 2.1
+                     , text              >= 1.2    && < 2.2
 
   ghc-options:         -Wall
                        -Wcompat
