--- sysstat-12.7.1/configure.ac	2022-11-06 16:09:20.000000000 +0100
+++ sysstat-12.7.1.new/configure.ac	2022-12-20 09:32:23.628993363 +0100
@@ -341,18 +341,18 @@
 fi
 
 SADC_DIR=$AuxPrefix/lib
-if test -d $AuxPrefix/lib64; then
+if test -d $AuxPrefix/lib; then
     # Verify that this OS is really 64 bit
     BITS=$(getconf LONG_BIT 2>/dev/null)
     if test $? = 0; then
         if test $BITS = 64; then
-            SADC_DIR=$AuxPrefix/lib64
+            SADC_DIR=$AuxPrefix/lib
         fi
     else
         # Fallback: look for lm (long mode) flag to know if CPU is 64 bit
         grep " lm " /proc/cpuinfo >/dev/null 2>&1
         if test $? = 0; then
-            SADC_DIR=$AuxPrefix/lib64
+            SADC_DIR=$AuxPrefix/lib
         fi
     fi
 fi
