diff -Naur kube-0.8.0/CMakeLists.txt kube-0.8.0.new/CMakeLists.txt
--- kube-0.8.0/CMakeLists.txt	2019-09-15 11:58:13.000000000 +0200
+++ kube-0.8.0.new/CMakeLists.txt	2020-06-22 13:36:58.723802785 +0200
@@ -66,7 +66,6 @@
 add_subdirectory(applications)
 add_subdirectory(views)
 add_subdirectory(accounts)
-add_subdirectory(tests)
 if (${ENABLE_EXTENSIONS})
     add_subdirectory(extensions)
 endif()
diff -Naur kube-0.8.0/components/CMakeLists.txt kube-0.8.0.new/components/CMakeLists.txt
--- kube-0.8.0/components/CMakeLists.txt	2019-09-15 11:58:13.000000000 +0200
+++ kube-0.8.0.new/components/CMakeLists.txt	2020-06-22 13:36:58.722802778 +0200
@@ -8,7 +8,6 @@
 endmacro(install_qml_component)
 
 install_qml_component(kube)
-add_subdirectory(kube/tests)
 install_qml_component(accounts)
 
 
diff -Naur kube-0.8.0/components/kube/tests/CMakeLists.txt kube-0.8.0.new/components/kube/tests/CMakeLists.txt
--- kube-0.8.0/components/kube/tests/CMakeLists.txt	2019-09-15 11:58:13.000000000 +0200
+++ kube-0.8.0.new/components/kube/tests/CMakeLists.txt	2020-06-22 13:36:58.723802785 +0200
@@ -1 +0,0 @@
-add_test(NAME kubecomponenttests COMMAND kubetestrunner WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
diff -Naur kube-0.8.0/framework/CMakeLists.txt kube-0.8.0.new/framework/CMakeLists.txt
--- kube-0.8.0/framework/CMakeLists.txt	2019-09-15 11:58:13.000000000 +0200
+++ kube-0.8.0.new/framework/CMakeLists.txt	2020-06-22 13:36:58.724802793 +0200
@@ -4,4 +4,3 @@
 install(FILES qmldir DESTINATION ${FRAMEWORK_INSTALL_DIR})
 
 add_subdirectory(src)
-add_subdirectory(qml/tests)
diff -Naur kube-0.8.0/framework/qml/tests/CMakeLists.txt kube-0.8.0.new/framework/qml/tests/CMakeLists.txt
--- kube-0.8.0/framework/qml/tests/CMakeLists.txt	2019-09-15 11:58:13.000000000 +0200
+++ kube-0.8.0.new/framework/qml/tests/CMakeLists.txt	2020-06-22 13:36:58.724802793 +0200
@@ -7,5 +7,4 @@
     Qt5::QuickTest
 )
 
-add_test(NAME frameworkqmltests COMMAND testrunner WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR})
 
diff -Naur kube-0.8.0/framework/src/CMakeLists.txt kube-0.8.0.new/framework/src/CMakeLists.txt
--- kube-0.8.0/framework/src/CMakeLists.txt	2019-09-15 11:58:13.000000000 +0200
+++ kube-0.8.0.new/framework/src/CMakeLists.txt	2020-06-22 13:36:58.725802800 +0200
@@ -102,10 +102,7 @@
 
 set(BUILD_TESTING ON)
 
-add_subdirectory(tests)
 add_subdirectory(domain/mime)
-add_subdirectory(domain/mime/tests)
 add_subdirectory(domain/mime/mimetreeparser)
-add_subdirectory(domain/settings/tests)
 
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES)
diff -Naur kube-0.8.0/framework/src/domain/mime/mimetreeparser/autotests/CMakeLists.txt kube-0.8.0.new/framework/src/domain/mime/mimetreeparser/autotests/CMakeLists.txt
--- kube-0.8.0/framework/src/domain/mime/mimetreeparser/autotests/CMakeLists.txt	2019-09-15 11:58:13.000000000 +0200
+++ kube-0.8.0.new/framework/src/domain/mime/mimetreeparser/autotests/CMakeLists.txt	2020-06-22 13:36:58.725802800 +0200
@@ -10,7 +10,6 @@
 include(ECMAddTests)
 include(${CMAKE_SOURCE_DIR}/cmake/modules/add_gpg_crypto_test.cmake)
 
-# convenience macro to add qtest unit tests
 macro(add_mimetreeparser_unittest _source)
     get_filename_component(_name ${_source} NAME_WE)
     ecm_add_test(${_source} util.cpp setupenv.cpp
diff -Naur kube-0.8.0/framework/src/domain/mime/mimetreeparser/CMakeLists.txt kube-0.8.0.new/framework/src/domain/mime/mimetreeparser/CMakeLists.txt
--- kube-0.8.0/framework/src/domain/mime/mimetreeparser/CMakeLists.txt	2019-09-15 11:58:13.000000000 +0200
+++ kube-0.8.0.new/framework/src/domain/mime/mimetreeparser/CMakeLists.txt	2020-06-22 13:36:58.725802800 +0200
@@ -49,5 +49,3 @@
     Gpgme::Gpgme
 )
 
-add_subdirectory(autotests)
-add_subdirectory(tests)
diff -Naur kube-0.8.0/framework/src/tests/CMakeLists.txt kube-0.8.0.new/framework/src/tests/CMakeLists.txt
--- kube-0.8.0/framework/src/tests/CMakeLists.txt	2019-09-15 11:58:13.000000000 +0200
+++ kube-0.8.0.new/framework/src/tests/CMakeLists.txt	2020-06-22 13:36:58.724802793 +0200
@@ -5,7 +5,6 @@
 
 find_package(Qt5 REQUIRED NO_MODULE COMPONENTS Core Test Gui)
 
-macro(auto_tests)
     foreach(_testname ${ARGN})
         add_executable(${_testname} ${_testname}.cpp)
         add_test(${_testname} ${_testname})
@@ -16,9 +15,7 @@
             kubeframework
         )
     endforeach(_testname)
-endmacro(auto_tests)
 
-auto_tests(
     folderlistmodeltest
     maillistmodeltest
     eventoccurrencemodeltest
diff -Naur kube-0.8.0/tests/CMakeLists.txt kube-0.8.0.new/tests/CMakeLists.txt
--- kube-0.8.0/tests/CMakeLists.txt	2019-09-15 11:58:13.000000000 +0200
+++ kube-0.8.0.new/tests/CMakeLists.txt	2020-06-22 13:36:58.724802793 +0200
@@ -14,7 +14,6 @@
 install(DIRECTORY qml/ DESTINATION ${QML_INSTALL_DIR}/org/kube/test)
 install(FILES qmldir DESTINATION ${QML_INSTALL_DIR}/org/kube/test)
 
-add_library(testplugin SHARED testplugin.cpp teststore.cpp)
 target_link_libraries(testplugin
     kubeframework
     KF5::CalendarCore
diff -Naur kube-0.8.0/views/CMakeLists.txt kube-0.8.0.new/views/CMakeLists.txt
--- kube-0.8.0/views/CMakeLists.txt	2019-09-15 11:58:13.000000000 +0200
+++ kube-0.8.0.new/views/CMakeLists.txt	2020-06-22 13:36:58.724802793 +0200
@@ -1,7 +1,6 @@
 macro(install_view name)
     install(DIRECTORY ${name}/qml/ DESTINATION ${QML_INSTALL_DIR}/org/kube/views/${name})
     install(FILES ${name}/metadata.json DESTINATION ${QML_INSTALL_DIR}/org/kube/views/${name})
-    add_test(NAME viewtest-${name} COMMAND kubetestrunner WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/${name}/tests)
 endmacro()
 
 install_view(composer)
