diff -ur a/vboxdrv/r0drv/linux/memobj-r0drv-linux.c b/vboxdrv/r0drv/linux/memobj-r0drv-linux.c
--- a/vboxdrv/r0drv/linux/memobj-r0drv-linux.c	2012-06-20 10:32:36.000000000 -0500
+++ b/vboxdrv/r0drv/linux/memobj-r0drv-linux.c	2012-12-12 16:56:11.096567562 -0600
@@ -1448,7 +1448,7 @@
 
 #if   defined(VBOX_USE_INSERT_PAGE) && LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 22)
                     rc = vm_insert_page(vma, ulAddrCur, pMemLnxToMap->apPages[iPage]);
-                    vma->vm_flags |= VM_RESERVED; /* This flag helps making 100% sure some bad stuff wont happen (swap, core, ++). */
+                    vma->vm_flags |= (VM_DONTEXPAND | VM_DONTDUMP); /* This flag helps making 100% sure some bad stuff wont happen (swap, core, ++). */
 #elif LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 11)
                     rc = remap_pfn_range(vma, ulAddrCur, page_to_pfn(pMemLnxToMap->apPages[iPage]), PAGE_SIZE, fPg);
 #elif defined(VBOX_USE_PAE_HACK)
