diff --git a/vboxdrv/SUPDrvInternal.h b/vboxdrv/SUPDrvInternal.h
index f51717f..7034437 100644
--- a/vboxdrv/SUPDrvInternal.h
+++ b/vboxdrv/SUPDrvInternal.h
@@ -78,7 +78,7 @@
     RT_C_DECLS_END
 
 #elif defined(RT_OS_LINUX)
-#   include <linux/autoconf.h>
+#   include <generated/autoconf.h>
 #   include <linux/version.h>
 #   if defined(CONFIG_MODVERSIONS) && !defined(MODVERSIONS)
 #       define MODVERSIONS
diff --git a/vboxdrv/include/iprt/types.h b/vboxdrv/include/iprt/types.h
index 5d33962..2f5a06d 100644
--- a/vboxdrv/include/iprt/types.h
+++ b/vboxdrv/include/iprt/types.h
@@ -84,7 +84,7 @@
 #  define true linux_true
 #  define false linux_false
 #  define uintptr_t linux_uintptr_t
-#  include <linux/autoconf.h>
+#  include <generated/autoconf.h>
 #  include <linux/types.h>
 #  include <linux/stddef.h>
 #  undef uintptr_t
diff --git a/vboxdrv/r0drv/linux/the-linux-kernel.h b/vboxdrv/r0drv/linux/the-linux-kernel.h
index b40c910..ce2b4b9 100644
--- a/vboxdrv/r0drv/linux/the-linux-kernel.h
+++ b/vboxdrv/r0drv/linux/the-linux-kernel.h
@@ -38,7 +38,7 @@
 #include <iprt/types.h>
 #define bool linux_bool
 
-#include <linux/autoconf.h>
+#include <generated/autoconf.h>
 #include <linux/version.h>
 
 /* We only support 2.4 and 2.6 series kernels */
diff --git a/vboxnetadp/include/iprt/types.h b/vboxnetadp/include/iprt/types.h
index 5d33962..2f5a06d 100644
--- a/vboxnetadp/include/iprt/types.h
+++ b/vboxnetadp/include/iprt/types.h
@@ -84,7 +84,7 @@
 #  define true linux_true
 #  define false linux_false
 #  define uintptr_t linux_uintptr_t
-#  include <linux/autoconf.h>
+#  include <generated/autoconf.h>
 #  include <linux/types.h>
 #  include <linux/stddef.h>
 #  undef uintptr_t
diff --git a/vboxnetadp/r0drv/linux/the-linux-kernel.h b/vboxnetadp/r0drv/linux/the-linux-kernel.h
index b40c910..ce2b4b9 100644
--- a/vboxnetadp/r0drv/linux/the-linux-kernel.h
+++ b/vboxnetadp/r0drv/linux/the-linux-kernel.h
@@ -38,7 +38,7 @@
 #include <iprt/types.h>
 #define bool linux_bool
 
-#include <linux/autoconf.h>
+#include <generated/autoconf.h>
 #include <linux/version.h>
 
 /* We only support 2.4 and 2.6 series kernels */
diff --git a/vboxnetflt/include/iprt/types.h b/vboxnetflt/include/iprt/types.h
index 5d33962..2f5a06d 100644
--- a/vboxnetflt/include/iprt/types.h
+++ b/vboxnetflt/include/iprt/types.h
@@ -84,7 +84,7 @@
 #  define true linux_true
 #  define false linux_false
 #  define uintptr_t linux_uintptr_t
-#  include <linux/autoconf.h>
+#  include <generated/autoconf.h>
 #  include <linux/types.h>
 #  include <linux/stddef.h>
 #  undef uintptr_t
diff --git a/vboxnetflt/r0drv/linux/the-linux-kernel.h b/vboxnetflt/r0drv/linux/the-linux-kernel.h
index b40c910..ce2b4b9 100644
--- a/vboxnetflt/r0drv/linux/the-linux-kernel.h
+++ b/vboxnetflt/r0drv/linux/the-linux-kernel.h
@@ -38,7 +38,7 @@
 #include <iprt/types.h>
 #define bool linux_bool
 
-#include <linux/autoconf.h>
+#include <generated/autoconf.h>
 #include <linux/version.h>
 
 /* We only support 2.4 and 2.6 series kernels */
