diff -Naur Recode-3.6/src/argmatch.c Recode-3.6-p/src/argmatch.c
--- Recode-3.6/src/argmatch.c	2008-02-19 05:13:35.000000000 +0100
+++ Recode-3.6-p/src/argmatch.c	2019-05-28 21:19:21.455357509 +0200
@@ -38,7 +38,7 @@
 
 #include "error.h"
 #include "quotearg.h"
-
+#include <stdlib.h>
 /* When reporting an invalid argument, show nonprinting characters
    by using the quoting style ARGMATCH_QUOTING_STYLE.  Do not use
    literal_quoting_style.  */
