diff -Naur SQLObject-3.2.0/setup.py SQLObject-3.2.0-p/setup.py
--- SQLObject-3.2.0/setup.py	2017-03-11 16:20:25.000000000 +0100
+++ SQLObject-3.2.0-p/setup.py	2017-04-16 12:41:29.284782281 +0200
@@ -97,33 +97,6 @@
       package_data={"sqlobject":
                     [
                         "../LICENSE",
-                        "../docs/*.rst",
-                        "../docs/html/*",
-                        "../docs/html/_sources/*",
-                        "../docs/html/_sources/api/*",
-                        "../docs/html/_modules/*",
-                        "../docs/html/_modules/sqlobject/*",
-                        "../docs/html/_modules/sqlobject/mysql/*",
-                        "../docs/html/_modules/sqlobject/postgres/*",
-                        "../docs/html/_modules/sqlobject/manager/*",
-                        "../docs/html/_modules/sqlobject/inheritance/*",
-                        "../docs/html/_modules/sqlobject/inheritance/tests/*",
-                        "../docs/html/_modules/sqlobject/mssql/*",
-                        "../docs/html/_modules/sqlobject/tests/*",
-                        "../docs/html/_modules/sqlobject/rdbhost/*",
-                        "../docs/html/_modules/sqlobject/versioning/*",
-                        "../docs/html/_modules/sqlobject/versioning/test/*",
-                        "../docs/html/_modules/sqlobject/util/*",
-                        "../docs/html/_modules/sqlobject/maxdb/*",
-                        "../docs/html/_modules/sqlobject/firebird/*",
-                        "../docs/html/_modules/sqlobject/sybase/*",
-                        "../docs/html/_modules/sqlobject/sqlite/*",
-                        "../docs/html/_modules/sqlobject/include/*",
-                        "../docs/html/_modules/sqlobject/include/tests/*",
-                        "../docs/html/_modules/pydispatch/*",
-                        "../docs/html/_modules/_pytest/*",
-                        "../docs/html/api/*",
-                        "../docs/html/_static/*",
                     ],
                     "sqlobject.maxdb": ["readme.txt"],
                     },
@@ -148,7 +121,6 @@
 @@ CHANGES
 
 For a more complete list, please see the news:
-http://sqlobject.org/docs/News.html
 
 
 What is SQLObject
@@ -179,7 +151,6 @@
 https://pypi.python.org/pypi/SQLObject/@@
 
 News and changes:
-http://sqlobject.org/docs/News.html
 
 -- 
 Ian Bicking  /  ianb@colorstudy.com  / http://blog.ianbicking.org
