diff -Naur rlocate-0.5.1/src/rlocate-module/Makefile.am rlocate-0.5.1-p/src/rlocate-module/Makefile.am
--- rlocate-0.5.1/src/rlocate-module/Makefile.am	2006-04-21 10:01:06.000000000 +0200
+++ rlocate-0.5.1-p/src/rlocate-module/Makefile.am	2006-04-21 15:18:35.000000000 +0200
@@ -63,7 +63,6 @@
 if SANDBOXED
 else
 install-data-local: install-moduleDATA
-	-@depmod@ -a
 endif
 
 CLEANFILES = $(module_DATA) .*.cmd $(module_DATA:.o=.mod.c) $(module_DATA:.o=.@kernelext@) *~
diff -Naur rlocate-0.5.1/src/rlocate-module/Makefile.in rlocate-0.5.1-p/src/rlocate-module/Makefile.in
--- rlocate-0.5.1/src/rlocate-module/Makefile.in	2006-04-21 10:01:13.000000000 +0200
+++ rlocate-0.5.1-p/src/rlocate-module/Makefile.in	2006-04-21 15:19:09.000000000 +0200
@@ -448,7 +448,7 @@
 	-test -c $(DESTDIR)$(devdir)/rlocate && $(RM) $(DESTDIR)$(devdir)/rlocate
 
 @SANDBOXED_FALSE@install-data-local: install-moduleDATA
-@SANDBOXED_FALSE@	-@depmod@ -a
+@SANDBOXED_FALSE@
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
diff -Naur rlocate-0.5.1/src/rlocate-scripts/Makefile.am rlocate-0.5.1-p/src/rlocate-scripts/Makefile.am
--- rlocate-0.5.1/src/rlocate-scripts/Makefile.am	2006-04-21 10:01:06.000000000 +0200
+++ rlocate-0.5.1-p/src/rlocate-scripts/Makefile.am	2006-04-21 15:17:34.000000000 +0200
@@ -18,9 +18,7 @@
 	rm $(srcdir)/rlocate-checkpoint.temp
 
 install-exec-hook:
-	-/usr/sbin/groupadd $(rlocate_group)
 	chown $(root_uid):$(rlocate_group) $(DESTDIR)$(bindir)/rlocate-checkpoint
 	chmod 2755 $(DESTDIR)$(bindir)/rlocate-checkpoint
 
 uninstall-hook:
-	-/usr/sbin/groupdel $(rlocate_group)
diff -Naur rlocate-0.5.1/src/rlocate-scripts/Makefile.in rlocate-0.5.1-p/src/rlocate-scripts/Makefile.in
--- rlocate-0.5.1/src/rlocate-scripts/Makefile.in	2006-04-21 10:01:13.000000000 +0200
+++ rlocate-0.5.1-p/src/rlocate-scripts/Makefile.in	2006-04-21 15:17:21.000000000 +0200
@@ -343,12 +343,10 @@
 	rm $(srcdir)/rlocate-checkpoint.temp
 
 install-exec-hook:
-	-/usr/sbin/groupadd $(rlocate_group)
 	chown $(root_uid):$(rlocate_group) $(DESTDIR)$(bindir)/rlocate-checkpoint
 	chmod 2755 $(DESTDIR)$(bindir)/rlocate-checkpoint
 
 uninstall-hook:
-	-/usr/sbin/groupdel $(rlocate_group)
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
