diff -Naur rlocate-0.5.1.orig/m4/rlocate.m4 rlocate-0.5.1/m4/rlocate.m4
--- rlocate-0.5.1.orig/m4/rlocate.m4	2006-04-21 10:01:05.000000000 +0200
+++ rlocate-0.5.1/m4/rlocate.m4	2006-09-21 14:48:47.000000000 +0200
@@ -49,7 +49,7 @@
   #include "$kerneldir/include/linux/autoconf.h"
   #endif
 
-  #include "$kerneldir/include/linux/version.h"]],
+  #include "$kerneldir/include/linux/utsrelease.h"]],
   [[
     FILE *f;
     char kversion[128];
diff -Naur rlocate-0.5.1.orig/src/rlocate-module/rlocate.c rlocate-0.5.1/src/rlocate-module/rlocate.c
--- rlocate-0.5.1.orig/src/rlocate-module/rlocate.c	2006-04-21 10:01:06.000000000 +0200
+++ rlocate-0.5.1/src/rlocate-module/rlocate.c	2006-09-21 15:05:36.000000000 +0200
@@ -42,7 +42,9 @@
 #include <linux/rwsem.h>
 #include <linux/spinlock.h>
 #include <linux/device.h>
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,18)
 #include <linux/devfs_fs_kernel.h>
+#endif
 #include <linux/namei.h>
 #include <linux/jiffies.h>
 
