diff -Naur mcrypt-2.6.5/Makefile.am mcrypt-2.6.5-p/Makefile.am
--- mcrypt-2.6.5/Makefile.am	2003-01-19 19:09:30.000000000 +0100
+++ mcrypt-2.6.5-p/Makefile.am	2007-02-20 16:40:44.000000000 +0100
@@ -3,8 +3,8 @@
 EXTRA_DIST = config.rpath  INSTALL.generic THANKS NEWS
 SUBDIRS = m4 doc src po
 install-exec-hook:
-	-$(RM) -f $(bindir)/mdecrypt
-	$(LN_S) mcrypt $(bindir)/mdecrypt
+	-$(RM) -f $(DESTDIR)$(bindir)/mdecrypt
+	$(LN_S) mcrypt $(DESTDIR)$(bindir)/mdecrypt
 uninstall-local:
 	-$(RM) -f $(bindir)/mdecrypt
 
