diff -Naur mcrypt-2.6.5/po/Makefile.in.in mcrypt-2.6.5-p/po/Makefile.in.in
--- mcrypt-2.6.5/po/Makefile.in.in	2003-02-21 18:45:55.000000000 +0100
+++ mcrypt-2.6.5-p/po/Makefile.in.in	2007-02-20 16:34:31.000000000 +0100
@@ -26,8 +26,8 @@
 
 INSTALL = @INSTALL@
 INSTALL_DATA = @INSTALL_DATA@
-MKINSTALLDIRS = @MKINSTALLDIRS@
-mkinstalldirs = $(SHELL) `case "$(MKINSTALLDIRS)" in /*) echo "$(MKINSTALLDIRS)" ;; *) echo "$(top_builddir)/$(MKINSTALLDIRS)" ;; esac`
+MKINSTALLDIRS = ../mkinstalldirs
+mkinstalldirs = $(MKINSTALLDIRS)
 
 GMSGFMT = @GMSGFMT@
 MSGFMT = @MSGFMT@
