diff -Naur xc.orig/programs/Xserver/hw/xfree86/os-support/linux/lnx_io.c xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_io.c
--- xc.orig/programs/Xserver/hw/xfree86/os-support/linux/lnx_io.c	2004-03-04 18:48:09.000000000 +0100
+++ xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_io.c	2004-06-24 22:46:39.000000000 +0200
@@ -68,18 +68,12 @@
  * from util-linux-2.9t package */
 
 #include <linux/kd.h>
-#include <linux/version.h>
 #ifdef __sparc__
 #include <asm/param.h>
 #include <asm/kbio.h>
 #endif
 
-/* Deal with spurious kernel header change */
-#if defined(LINUX_VERSION_CODE) && defined(KERNEL_VERSION)
-# if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,42)
 #  define rate period
-# endif
-#endif
 
 static int
 KDKBDREP_ioctl_ok(int rate, int delay) {
diff -Naur xc.orig/programs/Xserver/hw/xfree86/os-support/linux/lnx_kbd.c xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_kbd.c
--- xc.orig/programs/Xserver/hw/xfree86/os-support/linux/lnx_kbd.c	2004-03-04 18:48:09.000000000 +0100
+++ xc/programs/Xserver/hw/xfree86/os-support/linux/lnx_kbd.c	2004-06-24 22:47:07.000000000 +0200
@@ -95,18 +95,12 @@
  * from util-linux-2.9t package */
 
 #include <linux/kd.h>
-#include <linux/version.h>
 #ifdef __sparc__
 #include <asm/param.h>
 #include <asm/kbio.h>
 #endif
 
-/* Deal with spurious kernel header change */
-#if defined(LINUX_VERSION_CODE) && defined(KERNEL_VERSION)
-# if LINUX_VERSION_CODE >= KERNEL_VERSION(2,5,42)
 #  define rate period
-# endif
-#endif
 
 static int
 KDKBDREP_ioctl_ok(int rate, int delay) {
