diff -Naur libreswan-3.23/programs/pluto/linux-copy/linux/xfrm.h libreswan-3.23-p/programs/pluto/linux-copy/linux/xfrm.h
--- libreswan-3.23/programs/pluto/linux-copy/linux/xfrm.h	2018-01-25 21:19:46.000000000 +0100
+++ libreswan-3.23-p/programs/pluto/linux-copy/linux/xfrm.h	2018-02-10 01:24:08.425627633 +0100
@@ -1,7 +1,7 @@
 #ifndef _LINUX_XFRM_H
 #define _LINUX_XFRM_H
 
-#include <linux/in6.h>
+#include <netinet/in.h>
 #include <linux/types.h>
 
 /* All of the structures in this file may not change size as they are
