--- libgnomedb-3.99.7/libgnomedb/Makefile.in	2008-12-31 17:30:23.000000000 +0000
+++ libgnomedb-3.99.7/libgnomedb/Makefile.in	2009-09-29 08:23:38.000000000 +0000
@@ -420,7 +420,6 @@
 	gnome-db-dsn-editor.h \
 	gnome-db-provider-auth-editor.h \
 	gnome-db-provider-spec-editor.h \
-	gnome-db-enum-types.h \
 	gnome-db-enums.h \
 	gnome-db-find-dialog.h \
 	gnome-db-form.h \
--- libgnomedb-3.99.7/libgnomedb/Makefile.am	2009-09-29 08:28:16.000000000 +0000
+++ libgnomedb-3.99.7/libgnomedb/Makefile.am	2009-09-29 08:22:43.000000000 +0000
@@ -39,7 +39,6 @@
 	gnome-db-dsn-editor.h \
 	gnome-db-provider-auth-editor.h \
 	gnome-db-provider-spec-editor.h \
-	gnome-db-enum-types.h \
 	gnome-db-enums.h \
 	gnome-db-find-dialog.h \
 	gnome-db-form.h \
