--- libgnomedbold/control-center/database-properties-4.0.desktop.in	2008-12-30 11:22:49.000000000 +0100
+++ libgnomedb/control-center/database-properties-4.0.desktop.in	2009-04-19 20:00:40.000000000 +0200
@@ -5,5 +5,5 @@
 Exec=gnome-database-properties-4.0
 Terminal=0
 Type=Application
-Categories=Settings;DesktopSettings;
+Categories=Settings;DesktopSettings;X-GNOME-PersonalSettings;
 StartupNotify=true
