diff -Naur libgnomedb-3.1.1/extra/Makefile.am libgnomedb-3.1.1-p/extra/Makefile.am
--- libgnomedb-3.1.1/extra/Makefile.am	2007-09-01 14:28:59.000000000 +0200
+++ libgnomedb-3.1.1-p/extra/Makefile.am	2007-09-27 00:23:40.000000000 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = demos
+SUBDIRS = .
 
 if PLATFORM_WIN32
 EXTRALDFLAGS=-mwindows
diff -Naur libgnomedb-3.1.1/extra/Makefile.in libgnomedb-3.1.1-p/extra/Makefile.in
--- libgnomedb-3.1.1/extra/Makefile.in	2007-09-01 18:36:05.000000000 +0200
+++ libgnomedb-3.1.1-p/extra/Makefile.in	2007-09-27 00:23:31.000000000 +0200
@@ -309,7 +309,7 @@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-SUBDIRS = demos
+SUBDIRS = .
 @PLATFORM_WIN32_FALSE@EXTRALDFLAGS = 
 @PLATFORM_WIN32_TRUE@EXTRALDFLAGS = -mwindows
 AM_CPPFLAGS = \
