diff -Naur linux-live-5.5.0.orig/cd-root/make_disk.sh linux-live-5.5.0/cd-root/make_disk.sh
--- linux-live-5.5.0.orig/cd-root/make_disk.sh	2006-05-17 18:42:03.000000000 +0200
+++ linux-live-5.5.0/cd-root/make_disk.sh	2006-11-12 16:30:06.000000000 +0100
@@ -35,7 +35,7 @@
 
 image=$TMPDIR/boot/vmlinuz
 initrd=$TMPDIR/boot/initrd.gz
-label=slax
+label=FWLive
 root=/dev/ram0
 append=\"max_loop=255 init=linuxrc load_ramdisk=1 prompt_ramdisk=0 ramdisk_size=4444\"
 read-write
diff -Naur linux-live-5.5.0.orig/cd-root/make_iso.bat linux-live-5.5.0/cd-root/make_iso.bat
--- linux-live-5.5.0.orig/cd-root/make_iso.bat	2006-02-10 17:09:28.000000000 +0100
+++ linux-live-5.5.0/cd-root/make_iso.bat	2006-11-26 01:14:14.000000000 +0100
@@ -7,12 +7,10 @@
 
 if "%1"=="" goto error1
 
-set CDLABEL=SLAX
+set CDLABEL=FWLive
 
 REM  isolinux.bin is changed during the ISO creation,
 REM  so we need to restore it from backup.
-copy /Y boot\isolinux.bi_ boot\isolinux.bin
-if not "%errorlevel%"=="0" goto error2
 
 tools\WIN\mkisofs.exe @tools\WIN\config -o "%1" -A "%CDLABEL%" -V "%CDLABEL%" .
 goto theend
@@ -22,10 +20,6 @@
 echo Example: %0 c:\target.iso
 goto theend
 
-:error2
-echo Can't recreate isolinux.bin, make sure your current directory is writable!
-goto theend
-
 :theend
 echo.
 echo New ISO should be created now.
diff -Naur linux-live-5.5.0.orig/cd-root/make_iso.sh linux-live-5.5.0/cd-root/make_iso.sh
--- linux-live-5.5.0.orig/cd-root/make_iso.sh	2006-03-22 22:38:00.000000000 +0100
+++ linux-live-5.5.0/cd-root/make_iso.sh	2006-11-26 01:11:36.000000000 +0100
@@ -12,7 +12,7 @@
   exit
 fi
 
-CDLABEL="SLAX"
+CDLABEL="server-0.2-i686 (Alderaan)"
 ISONAME="$1"
 
 if [ "$ISONAME" = "" ]; then
@@ -22,14 +22,5 @@
    if [ "$ISONAME" = "" ]; then ISONAME="$SUGGEST"; fi
 fi
 
-# isolinux.bin is changed during the ISO creation,
-# so we need to restore it from backup.
-cp -f boot/isolinux.bi_ boot/isolinux.bin
-if [ $? -ne 0 ]; then
-   echo "Can't recreate isolinux.bin, make sure your current directory is writable!"
-   exit 1
-fi
-
 mkisofs -o "$ISONAME" -v -J -R -D -A "$CDLABEL" -V "$CDLABEL" \
--no-emul-boot -boot-info-table -boot-load-size 4 \
--b boot/isolinux.bin -c boot/isolinux.boot .
+-b boot/grub/stage2_eltorito -no-emul-boot -boot-load-size 4 -boot-info-table .
diff -Naur linux-live-5.5.0.orig/cd-root/tools/WIN/config linux-live-5.5.0/cd-root/tools/WIN/config
--- linux-live-5.5.0.orig/cd-root/tools/WIN/config	2005-12-11 12:41:16.000000000 +0100
+++ linux-live-5.5.0/cd-root/tools/WIN/config	2006-11-26 01:20:15.000000000 +0100
@@ -5,5 +5,4 @@
 -no-emul-boot
 -boot-info-table
 -boot-load-size 4 
--b "boot/isolinux.bin"
--c "boot/isolinux.boot"
+-b "boot/grub/stage2_eltorito"
diff -Naur linux-live-5.5.0.orig/config linux-live-5.5.0/config
--- linux-live-5.5.0.orig/config	2006-06-25 11:35:46.000000000 +0200
+++ linux-live-5.5.0/config	2006-11-26 02:11:58.000000000 +0100
@@ -8,8 +8,8 @@
 
 # the size of RAMdisk (in KBytes)
 # if you change this, then you will need to change the size
-# also in the following files: ./cd-root/isolinux.cfg, ./cd-root/boot/DOS/config
-RAM0SIZE=4444
+# also in the following file: ./cd-root/boot/DOS/config
+RAM0SIZE=29628
 
 # kernel version. Change it to "2.6.10" for example, if you are building
 # LiveCD with a different kernel than the one you are actually running
diff -Naur linux-live-5.5.0.orig/initrd/cleanup linux-live-5.5.0/initrd/cleanup
--- linux-live-5.5.0.orig/initrd/cleanup	2006-07-20 20:19:16.000000000 +0200
+++ linux-live-5.5.0/initrd/cleanup	2006-11-12 19:55:44.000000000 +0100
@@ -4,7 +4,7 @@
 #
 # Call it this way: exec /mnt/live/cleanup
 
-export PATH=.:/mnt/live:/usr/sbin:/usr/bin:/sbin:/bin
+export PATH=.:/media/live:/usr/sbin:/usr/bin:/sbin:/bin
 . liblinuxlive
 
 header "starting Linux Live shutdown procedures..."
diff -Naur linux-live-5.5.0.orig/initrd/initrd_create linux-live-5.5.0/initrd/initrd_create
--- linux-live-5.5.0.orig/initrd/initrd_create	2006-07-20 19:00:29.000000000 +0200
+++ linux-live-5.5.0/initrd/initrd_create	2006-11-23 14:16:39.000000000 +0100
@@ -69,7 +69,7 @@
 mkdir $INITRD_TREE
 
 debug "creating directories"
-mkdir -p $INITRD_TREE/{etc,dev,bin,mnt,proc,lib,sys,tmp,var/log}
+mkdir -p $INITRD_TREE/{etc,dev,bin,media,mnt,proc,lib,sys,tmp,var/log}
 
 debug "creating some essential devices in rootdisk"
 mknod $INITRD_TREE/dev/console c 5 1
@@ -109,6 +109,28 @@
 ln -s busybox $INITRD_TREE/bin/cat
 ln -s busybox $INITRD_TREE/bin/grep
 ln -s busybox $INITRD_TREE/bin/sleep
+ln -s busybox $INITRD_TREE/bin/echo
+ln -s busybox $INITRD_TREE/bin/egrep
+ln -s busybox $INITRD_TREE/bin/sed
+ln -s busybox $INITRD_TREE/bin/cut
+ln -s busybox $INITRD_TREE/bin/basename
+ln -s busybox $INITRD_TREE/bin/mkdir
+ln -s busybox $INITRD_TREE/bin/mknod
+ln -s busybox $INITRD_TREE/bin/tr
+ln -s busybox $INITRD_TREE/bin/tail
+ln -s busybox $INITRD_TREE/bin/ln
+ln -s busybox $INITRD_TREE/bin/uname
+ln -s busybox $INITRD_TREE/bin/dirname
+ln -s busybox $INITRD_TREE/bin/cp
+ln -s busybox $INITRD_TREE/bin/rm
+ln -s busybox $INITRD_TREE/bin/gunzip
+ln -s busybox $INITRD_TREE/bin/chmod
+ln -s busybox $INITRD_TREE/bin/chown
+ln -s busybox $INITRD_TREE/bin/find
+ln -s busybox $INITRD_TREE/bin/sort
+ln -s busybox $INITRD_TREE/bin/mkswap
+ln -s busybox $INITRD_TREE/bin/mv
+ln -s busybox $INITRD_TREE/bin/pivot_root
 ln -s bin $INITRD_TREE/sbin
 
 LMK="lib/modules/$KERNEL"
@@ -129,12 +151,13 @@
 rcopy_ex /$LMK/kernel/fs/ntfs $INITRD_TREE 2>>$DEBUG
 rcopy_ex /$LMK/kernel/fs/ext3 $INITRD_TREE 2>>$DEBUG
 rcopy_ex /$LMK/kernel/fs/reiserfs $INITRD_TREE 2>>$DEBUG
+rcopy_ex /$LMK/kernel/fs/fuse $INITRD_TREE 2>>$DEBUG
 
 # add language support for filesystems
 rcopy_ex /$LMK/kernel/fs/nls/ $INITRD_TREE 2>>$DEBUG
 
 #usb modules
-rcopy_ex /$LMK/kernel/drivers/usb/storage $INITRD_TREE 2>>$DEBUG
+rcopy_ex /$LMK/kernel/drivers/usb/storage/usb-storage* $INITRD_TREE 2>>$DEBUG
 rcopy_ex /$LMK/kernel/drivers/usb/host/ehci-hcd* $INITRD_TREE 2>>$DEBUG
 rcopy_ex /$LMK/kernel/drivers/usb/host/ohci-hcd* $INITRD_TREE 2>>$DEBUG
 rcopy_ex /$LMK/kernel/drivers/usb/host/uhci-hcd* $INITRD_TREE 2>>$DEBUG
diff -Naur linux-live-5.5.0.orig/initrd/linuxrc linux-live-5.5.0/initrd/linuxrc
--- linux-live-5.5.0.orig/initrd/linuxrc	2006-08-22 08:38:18.000000000 +0200
+++ linux-live-5.5.0/initrd/linuxrc	2006-10-25 14:07:48.000000000 +0200
@@ -3,7 +3,7 @@
 export PATH=.:/:/usr/sbin:/usr/bin:/sbin:/bin
 . liblinuxlive
 
-header "starting Linux Live scripts <http://www.linux-live.org/>"
+header "FwLive setup start"
 
 echolog "mounting /proc and /sys filesystems"
 mount -t proc proc /proc
@@ -176,12 +176,9 @@
 mkdir -p $INITRAMDISK
 if [ ! -e dev/console ]; then mknod dev/console c 5 1; fi
 
-if [ -x $UNION/usr/sbin/chroot ];
-  then CHROOT=/usr/sbin/chroot
-  else CHROOT=/usr/bin/chroot
-fi
+CHROOT=/usr/sbin/chroot
 
-header "linux live end, starting the Linux distribution"
+header "FwLive setup end, starting the Linux distribution"
 pivot_root . $INITRAMDISK
 exec $CHROOT . sbin/init <dev/console >dev/console 2>&1
 
diff -Naur linux-live-5.5.0.orig/runme.sh linux-live-5.5.0/runme.sh
--- linux-live-5.5.0.orig/runme.sh	2006-06-25 11:35:50.000000000 +0200
+++ linux-live-5.5.0/runme.sh	2006-11-26 01:28:58.000000000 +0100
@@ -20,11 +20,7 @@
 
 
 # search for kernel
-VMLINUZ=$ROOT/boot/vmlinuz
-if [ -L "$VMLINUZ" ]; then VMLINUZ=`readlink -f $VMLINUZ`; fi
-echo -ne "Enter path for the kernel you wanna use [hit enter for $VMLINUZ]: "
-read NEWKERNEL
-if [ "$NEWKERNEL" != "" ]; then VMLINUZ="$NEWKERNEL"; fi
+VMLINUZ=$ROOT/boot/vmlinuz-`uname -r`
 if [ "`ls $VMLINUZ 2>>$DEBUG`" = "" ]; then echo "cannot find $VMLINUZ"; exit 1; fi
 
 header "Creating LiveCD from your Linux"
diff -Naur linux-live-5.5.0.orig/tools/liblinuxlive linux-live-5.5.0/tools/liblinuxlive
--- linux-live-5.5.0.orig/tools/liblinuxlive	2006-08-22 08:34:39.000000000 +0200
+++ linux-live-5.5.0/tools/liblinuxlive	2006-11-23 23:35:06.000000000 +0100
@@ -2,6 +2,7 @@
 
 # Functions library :: for Linux Live scripts 5.x.y
 # Author: Tomas M. <http://www.linux-live.org>
+# Modified Janny <janny@frugalware.org>
 #
 
 # ===========================================================
@@ -15,7 +16,7 @@
 # We have to set these variables very carefully
 UNION=union
 MEMORY=memory
-MOUNTDIR=mnt
+MOUNTDIR=media
 CHANGES=$MEMORY/changes
 COPY2RAM=$MEMORY/copy2ram
 IMAGES=$MEMORY/images
@@ -163,7 +164,7 @@
 
 # Mount device $1 to $2
 # $1 = /dev device to mount, eg. /dev/hda1, or loop file, or directory
-# $2 = mountpoint, eg. /mnt/hda1
+# $2 = mountpoint, eg. /$MOUNTDIR/hda1
 # $3 = optional mount options, for example "ro", or "remount,rw"
 # $4 = optional filesystem type
 #
@@ -370,7 +371,7 @@
 # Find file-path on given device
 # Mounts the device in $1 and returns path if found,
 # else unmounts and exits
-# $1 = mountpoint, eg. /mnt
+# $1 = mountpoint, eg. /$MOUNTDIR
 # $2 = device
 # $3 = path/filename
 #
@@ -386,7 +387,7 @@
 
 # Find file in computer by mounting disks or other storage devices in $1
 # and searching for $2 in the mounted directory
-# $1 = mountpoint, eg. /mnt
+# $1 = mountpoint, eg. /$MOUNTDIR
 # $2 = filename or device-path or devicepath/filename
 #
 find_file()
@@ -446,9 +447,11 @@
    modprobe_module nls_cp437
    modprobe_module nls_iso8859-1
    modprobe_module nls_iso8859-2
+   modprobe_module nls_utf8
    modprobe_module vfat
    echolog "starting ntfs support"
    modprobe_module ntfs
+   modprobe_module fuse
    create_block_devices
 }
 
@@ -503,7 +506,7 @@
 }
 
 # create correct fstab file in $1/etc/fstab and create apropriate
-# mount directories in $1/mnt. Check for iocharset boot option,
+# mount directories in $1/$MOUNTDIR. Check for iocharset boot option,
 # if present, add iocharset to all DOS/Win filesystems (vfat,ntfs)
 # $1 = root directory (union)
 #
@@ -514,16 +517,16 @@
    if [ "$AUTO" = "" ]; then AUTO="auto"; fi
 
    FSTAB="$1/etc/fstab"
-   mkdir -p $1/etc $1/mnt
+   mkdir -p $1/etc $1/$MOUNTDIR $1/mnt
    cat $FSTAB 2>/dev/null | grep -v "$FSTABLLFLAG" >$FSTAB~
 
    fstab_update_line $FSTAB~ tmpfs / tmpfs defaults
    fstab_update_line $FSTAB~ devpts /dev/pts devpts gid=5,mode=620
    fstab_update_line $FSTAB~ proc /proc proc defaults
    list_cdrom_devices | while read DEVICE; do
-      MOUNTDIR="/mnt/`basename $DEVICE`_cdrom"
-      mkdir -p $1/$MOUNTDIR
-      fstab_update_line $FSTAB~ $DEVICE $MOUNTDIR iso9660 noauto,users,exec
+      MOUNTDIR_O="/$MOUNTDIR/`basename $DEVICE`_cdrom"
+      mkdir -p $1/$MOUNTDIR_O
+      fstab_update_line $FSTAB~ $DEVICE $MOUNTDIR_O iso9660 noauto,users,exec
    done
    list_partition_devices | while read DEVICE; do
       unset REMOVABLE; unset OPT; DEV="`basename $DEVICE`"; DEV0="`echo $DEV | cut -b 1-3`"
@@ -531,13 +534,14 @@
          REMOVABLE="_removable"
       fi
 
-      MOUNTDIR="/mnt/$DEV$REMOVABLE"
+      MOUNTDIR_O="/$MOUNTDIR/$DEV$REMOVABLE"
       FS="`blkid -s TYPE $DEVICE | cut -d = -f 2 | tr -d ' \"'`"
 
       # add special options for NTFS
       if [ "$FS" = "ntfs" ]; then
          OPT=",ro"
          if [ "$IOCHARSET" != "" ]; then OPT="$OPT,nls=$IOCHARSET"; fi
+	 if [ ! "`cmdline_parameter ntfsrw`" = "" ]; then FS="ntfs-3g" OPT=",rw"; fi
       fi
 
       # add special options for VFAT
@@ -548,14 +552,14 @@
          if [ "$FS" = "swap" ]; then
             fstab_update_line $FSTAB~ $DEVICE swap swap defaults
          else
-            fstab_update_line $FSTAB~ $DEVICE $MOUNTDIR $FS $AUTO,users,suid,dev,exec$OPT
-            mkdir -p "$1/$MOUNTDIR"
+            fstab_update_line $FSTAB~ $DEVICE $MOUNTDIR_O $FS $AUTO,users,suid,dev,exec$OPT
+            mkdir -p "$1/$MOUNTDIR_O"
          fi
       fi
    done
 
-   fstab_update_line $FSTAB~ /dev/fd0 /mnt/floppy vfat,msdos noauto,users,suid,dev,exec
-   mkdir -p $1/mnt/floppy
+   fstab_update_line $FSTAB~ /dev/fd0 /$MOUNTDIR/floppy vfat,msdos noauto,users,suid,dev,exec
+   mkdir -p $1/$MOUNTDIR/floppy
    
    mv -f $FSTAB~ $FSTAB
 }
diff -Naur linux-live-5.5.0.orig/tools/uselivemod linux-live-5.5.0/tools/uselivemod
--- linux-live-5.5.0.orig/tools/uselivemod	2006-05-25 11:03:25.000000000 +0200
+++ linux-live-5.5.0/tools/uselivemod	2006-11-12 16:16:49.000000000 +0100
@@ -18,8 +18,8 @@
 . liblinuxlive || exit 1
 
 allow_only_root
-IMAGES=/mnt/live/memory/images
-MODULES=/mnt/live/memory/modules
+IMAGES=/$MOUNTDIR/live/memory/images
+MODULES=/$MOUNTDIR/live/memory/modules
 
 # are we even using union?
 unionctl --query / >/dev/null 2>&1
