diff -Naur lsof_4.74_src.orig/Configure lsof_4.74_src/Configure
--- lsof_4.74_src.orig/Configure	2005-01-17 14:24:32.000000000 +0100
+++ lsof_4.74_src/Configure	2005-04-03 00:51:06.000000000 +0200
@@ -750,7 +750,7 @@
       echo "main(){}" > ${LSOF_TMPC}.c
       echo "Getting version number of ${LSOF_CC}."
       $LSOF_CC -c ${LSOF_TMPC}.c -I${LSOF_INCLUDE} -o ${LSOF_TMPC}.o -qlist > /dev/null 2>&1
-      LSOF_CCV=`head -1 ${LSOF_TMPC}.lst | sed 's/\(.*\) ---.*/\1/'`
+      LSOF_CCV=`head -n 1 ${LSOF_TMPC}.lst | sed 's/\(.*\) ---.*/\1/'`
       rm ${LSOF_TMPC}.*
       echo "The version is \"${LSOF_CCV}\"."
       echo $LSOF_CCV | grep "Version [0-9]" > /dev/null
@@ -1178,7 +1178,7 @@
     if test "X$DU_ADVFSV" = "X"	# {
     then
       echo "Determining the ADVFS version -- this will take a while."
-      LSOF_ADVFSV=`/usr/sbin/setld -i | grep "^OSFADVFSBIN[0-9]" | sed 's/\([^ ]*\).*/\1/' | sort -u | tail -1 | sed 's/OSFADVFSBIN//'`
+      LSOF_ADVFSV=`/usr/sbin/setld -i | grep "^OSFADVFSBIN[0-9]" | sed 's/\([^ ]*\).*/\1/' | sort -u | tail -n 1 | sed 's/OSFADVFSBIN//'`
     else
       LSOF_ADVFSV=$DU_ADVFSV
     fi	# }
@@ -3861,7 +3861,7 @@
 
       if test "X$SOLARIS_23P101318" = "X"	# {
       then
-	LSOF_PL=`grep -h SUNW_PATCHID=101318 /var/sadm/pkg/SUNWcar*/pkginfo | sed 's/.*-//' | sort -u | tail -1`
+	LSOF_PL=`grep -h SUNW_PATCHID=101318 /var/sadm/pkg/SUNWcar*/pkginfo | sed 's/.*-//' | sort -u | tail -n 1`
 	if test "X$LSOF_PL" = "X"	# {
 	then
 	  LSOF_PL=0
@@ -3877,7 +3877,7 @@
     20400)
       if test "X$SOLARIS_24P101945" = "X"	# {
       then
-	LSOF_PL=`grep -h SUNW_PATCHID=101945 /var/sadm/pkg/SUNWcar*/pkginfo | sed 's/.*-//' | sort -u | tail -1`
+	LSOF_PL=`grep -h SUNW_PATCHID=101945 /var/sadm/pkg/SUNWcar*/pkginfo | sed 's/.*-//' | sort -u | tail -n 1`
 	if test "X$LSOF_PL" = "X"	# {
 	then
 	  LSOF_PL=0
@@ -3889,7 +3889,7 @@
       then
 	if test "X$SOLARIS_24P102303" = "X"	# {
 	then
-	  LSOF_PL=`grep -h SUNW_PATCHID=102303 /var/sadm/pkg/SUNWhea*/pkginfo | sed 's/.*-//' | sort -u | tail -1`
+	  LSOF_PL=`grep -h SUNW_PATCHID=102303 /var/sadm/pkg/SUNWhea*/pkginfo | sed 's/.*-//' | sort -u | tail -n 1`
 	  if test "X$LSOF_PL" = "X"	# {
 	  then
 	    LSOF_PL=0
@@ -4215,7 +4215,7 @@
 	    LSOF_TMP1=`ls /usr/vice/etc/modload/libafs* 2>/dev/null | wc -l`
 	    if test $LSOF_TMP1 -ne 0	# {
 	    then
-	      SUN_AFSAPATHDEF=`ls -t /usr/vice/etc/modload/libafs* | head -1`
+	      SUN_AFSAPATHDEF=`ls -t /usr/vice/etc/modload/libafs* | head -n 1`
 	    fi	# }
 	  fi	# }
 	fi	# }
diff -Naur lsof_4.74_src.orig/dialects/n+os/Makefile lsof_4.74_src/dialects/n+os/Makefile
--- lsof_4.74_src.orig/dialects/n+os/Makefile	2002-04-16 20:01:58.000000000 +0200
+++ lsof_4.74_src/dialects/n+os/Makefile	2005-04-03 00:52:34.000000000 +0200
@@ -107,7 +107,7 @@
 	  fi \
 	fi
 	@if [ "X${LSOF_SYSINFO}" = "X" ]; then \
-	    echo '#define	LSOF_SYSINFO	"'`hostinfo | head -2 | tail -1`'"' >> version.h; \
+	    echo '#define	LSOF_SYSINFO	"'`hostinfo | head -n 2 | tail -n 1`'"' >> version.h; \
 	else \
 	  if [ "${LSOF_SYSINFO}" = "none" ]; then \
 	    echo '#define	LSOF_SYSINFO	""' >> version.h; \
