diff -Naur lsof_4.74_src.orig/Configure lsof_4.74_src/Configure
--- lsof_4.74_src.orig/Configure	2005-01-17 14:24:32.000000000 +0100
+++ lsof_4.74_src/Configure	2005-04-03 00:51:06.000000000 +0200
@@ -750,7 +750,7 @@
       echo "main(){}" > ${LSOF_TMPC}.c
       echo "Getting version number of ${LSOF_CC}."
       $LSOF_CC -c ${LSOF_TMPC}.c -I${LSOF_INCLUDE} -o ${LSOF_TMPC}.o -qlist > /dev/null 2>&1
-      LSOF_CCV=`head -1 ${LSOF_TMPC}.lst | sed 's/\(.*\) ---.*/\1/'`
+      LSOF_CCV=`head -n 1 ${LSOF_TMPC}.lst | sed 's/\(.*\) ---.*/\1/'`
       rm ${LSOF_TMPC}.*
       echo "The version is \"${LSOF_CCV}\"."
       echo $LSOF_CCV | grep "Version [0-9]" > /dev/null
@@ -1178,7 +1178,7 @@
     if test "X$DU_ADVFSV" = "X"	# {
     then
       echo "Determining the ADVFS version -- this will take a while."
-      LSOF_ADVFSV=`/usr/sbin/setld -i | grep "^OSFADVFSBIN[0-9]" | sed 's/\([^ ]*\).*/\1/' | sort -u | tail -1 | sed 's/OSFADVFSBIN//'`
+      LSOF_ADVFSV=`/usr/sbin/setld -i | grep "^OSFADVFSBIN[0-9]" | sed 's/\([^ ]*\).*/\1/' | sort -u | tail -n 1 | sed 's/OSFADVFSBIN//'`
     else
       LSOF_ADVFSV=$DU_ADVFSV
     fi	# }
@@ -3861,7 +3861,7 @@
 
       if test "X$SOLARIS_23P101318" = "X"	# {
       then
-	LSOF_PL=`grep -h SUNW_PATCHID=101318 /var/sadm/pkg/SUNWcar*/pkginfo | sed 's/.*-//' | sort -u | tail -1`
+	LSOF_PL=`grep -h SUNW_PATCHID=101318 /var/sadm/pkg/SUNWcar*/pkginfo | sed 's/.*-//' | sort -u | tail -n 1`
 	if test "X$LSOF_PL" = "X"	# {
 	then
 	  LSOF_PL=0
@@ -3877,7 +3877,7 @@
     20400)
       if test "X$SOLARIS_24P101945" = "X"	# {
       then
-	LSOF_PL=`grep -h SUNW_PATCHID=101945 /var/sadm/pkg/SUNWcar*/pkginfo | sed 's/.*-//' | sort -u | tail -1`
+	LSOF_PL=`grep -h SUNW_PATCHID=101945 /var/sadm/pkg/SUNWcar*/pkginfo | sed 's/.*-//' | sort -u | tail -n 1`
 	if test "X$LSOF_PL" = "X"	# {
 	then
 	  LSOF_PL=0
@@ -3889,7 +3889,7 @@
       then
 	if test "X$SOLARIS_24P102303" = "X"	# {
 	then
-	  LSOF_PL=`grep -h SUNW_PATCHID=102303 /var/sadm/pkg/SUNWhea*/pkginfo | sed 's/.*-//' | sort -u | tail -1`
+	  LSOF_PL=`grep -h SUNW_PATCHID=102303 /var/sadm/pkg/SUNWhea*/pkginfo | sed 's/.*-//' | sort -u | tail -n 1`
 	  if test "X$LSOF_PL" = "X"	# {
 	  then
 	    LSOF_PL=0
@@ -4215,7 +4215,7 @@
 	    LSOF_TMP1=`ls /usr/vice/etc/modload/libafs* 2>/dev/null | wc -l`
 	    if test $LSOF_TMP1 -ne 0	# {
 	    then
-	      SUN_AFSAPATHDEF=`ls -t /usr/vice/etc/modload/libafs* | head -1`
+	      SUN_AFSAPATHDEF=`ls -t /usr/vice/etc/modload/libafs* | head -n 1`
 	    fi	# }
 	  fi	# }
 	fi	# }
diff -Naur lsof_4.74_src.orig/Customize lsof_4.74_src/Customize
--- lsof_4.74_src.orig/Customize	2002-12-09 14:51:39.000000000 +0100
+++ lsof_4.74_src/Customize	2005-04-03 00:51:47.000000000 +0200
@@ -185,7 +185,7 @@
 
 .CAT_MARK
 
-grep HASSECURITY $OLD | tail -1 | egrep "^#define" > /dev/null
+grep HASSECURITY $OLD | tail -n 1 | egrep "^#define" > /dev/null
 if test $? -eq 0
 then
   echo "HASSECURITY is enabled."
@@ -248,7 +248,7 @@
 
 .CAT_MARK
 
-  grep HASNOSOCKSECURITY $OLD | tail -1 | egrep "^#define" > /dev/null
+  grep HASNOSOCKSECURITY $OLD | tail -n 1 | egrep "^#define" > /dev/null
   if test $? -eq 0
   then
     echo "HASNOSOCKSECURITY is enabled."
@@ -309,7 +309,7 @@
 
 .CAT_MARK
 
-grep WARNINGSTATE $OLD | tail -1 | egrep "^#define" > /dev/null
+grep WARNINGSTATE $OLD | tail -n 1 | egrep "^#define" > /dev/null
 if test $? -eq 0
 then
   echo "WARNINGSTATE is disabled."
@@ -378,7 +378,7 @@
 
 .CAT_MARK
 
-  grep WARNDEVACCESS $OLD | tail -1 | egrep "^#define" > /dev/null
+  grep WARNDEVACCESS $OLD | tail -n 1 | egrep "^#define" > /dev/null
   if test $? -eq 0
   then
     echo "WARNDEVACCESS is enabled."
@@ -461,7 +461,7 @@
 
 .CAT_MARK
 
-  grep HASDCACHE $OLD | tail -1 | egrep "^#define" > /dev/null
+  grep HASDCACHE $OLD | tail -n 1 | egrep "^#define" > /dev/null
   if test $? -eq 0
   then
     echo "HASDCACHE is enabled."
@@ -526,10 +526,10 @@
   
 .CAT_MARK
 
-    grep HASENVDC $OLD | tail -1 | egrep "^#define"
-    egrep "HASPERSDC$|HASPERSDC[ 	]" $OLD | tail -1 | egrep "^#define"
-    grep HASPERSDCPATH $OLD | tail -1 | egrep "^#define"
-    grep HASSYSDC $OLD | tail -1 | egrep "^#define"
+    grep HASENVDC $OLD | tail -n 1 | egrep "^#define"
+    egrep "HASPERSDC$|HASPERSDC[ 	]" $OLD | tail -n 1 | egrep "^#define"
+    grep HASPERSDCPATH $OLD | tail -n 1 | egrep "^#define"
+    grep HASSYSDC $OLD | tail -n 1 | egrep "^#define"
     END=0
     while test $END -eq 0
     do
@@ -575,7 +575,7 @@
 .CAT_MARK
 
     echo $EO "        $EC"
-    TMP1=`grep HASENVDC $OLD | tail -1 | egrep "^#define"`
+    TMP1=`grep HASENVDC $OLD | tail -n 1 | egrep "^#define"`
     if test "X$TMP1" != "X"
     then
       TMP1=`echo "$TMP1" | sed 's/^#define[ 	]HASENVDC[ 	]"\([^"]*\)".*$/\1/'`
@@ -636,7 +636,7 @@
 .CAT_MARK
 
     echo $EO "        $EC"
-    TMP1=`egrep "HASPERSDC$|HASPERSDC[ 	]" $OLD | tail -1 | egrep "^#define"`
+    TMP1=`egrep "HASPERSDC$|HASPERSDC[ 	]" $OLD | tail -n 1 | egrep "^#define"`
     if test "X$TMP1" != "X"
     then
       TMP1=`echo "$TMP1" | sed 's/^#define[ 	]HASPERSDC[ 	]"\([^"]*\)".*$/\1/'`
@@ -700,7 +700,7 @@
 .CAT_MARK
 
     echo $EO "        $EC"
-    TMP1=`grep HASPERSDCPATH $OLD | tail -1 | egrep "^#define"`
+    TMP1=`grep HASPERSDCPATH $OLD | tail -n 1 | egrep "^#define"`
     if test "X$TMP1" != "X"
     then
       TMP1=`echo "$TMP1" | sed 's/^#define[ 	]HASPERSDCPATH[ 	]"\([^"]*\)".*$/\1/'`
@@ -753,7 +753,7 @@
 .CAT_MARK
 
     echo $EO "        $EC"
-    TMP1=`grep HASSYSDC $OLD | tail -1 | egrep "^#define"`
+    TMP1=`grep HASSYSDC $OLD | tail -n 1 | egrep "^#define"`
     if test "X$TMP1" != "X"
     then
       TMP1=`echo "$TMP1" | sed 's/^#define[ 	]HASSYSDC[ 	]"\([^"]*\)".*$/\1/'`
@@ -808,7 +808,7 @@
 # ask about changing HASXOPT_ROOT.
 
 HXRC=0
-grep HASXOPT $OLD | tail -1 | egrep "^#define" > /dev/null
+grep HASXOPT $OLD | tail -n 1 | egrep "^#define" > /dev/null
 if test $? -eq 0
 then
   grep HASXOPT_ROOT $OLD > /dev/null
@@ -834,7 +834,7 @@
 
 .CAT_MARK
 
-    grep HASXOPT_ROOT $OLD | tail -1 | egrep "^#define" > /dev/null
+    grep HASXOPT_ROOT $OLD | tail -n 1 | egrep "^#define" > /dev/null
     if test $? -eq 0
     then
       echo "HASXOPT_ROOT is defined."
@@ -898,7 +898,7 @@
 
 .CAT_MARK
 
-    grep HASKERNIDCK $OLD | tail -1 | grep "^#define" > /dev/null
+    grep HASKERNIDCK $OLD | tail -n 1 | grep "^#define" > /dev/null
     if test $? -eq 0
     then
       echo "HASKERNIDCK is enabled."
diff -Naur lsof_4.74_src.orig/dialects/n+os/Makefile lsof_4.74_src/dialects/n+os/Makefile
--- lsof_4.74_src.orig/dialects/n+os/Makefile	2002-04-16 20:01:58.000000000 +0200
+++ lsof_4.74_src/dialects/n+os/Makefile	2005-04-03 00:52:34.000000000 +0200
@@ -107,7 +107,7 @@
 	  fi \
 	fi
 	@if [ "X${LSOF_SYSINFO}" = "X" ]; then \
-	    echo '#define	LSOF_SYSINFO	"'`hostinfo | head -2 | tail -1`'"' >> version.h; \
+	    echo '#define	LSOF_SYSINFO	"'`hostinfo | head -n 2 | tail -n 1`'"' >> version.h; \
 	else \
 	  if [ "${LSOF_SYSINFO}" = "none" ]; then \
 	    echo '#define	LSOF_SYSINFO	""' >> version.h; \
