--- xplanet-1.2.1/src/Satellite.cpp.g++44	2006-03-26 09:34:21.000000000 +0900
+++ xplanet-1.2.1/src/Satellite.cpp	2009-04-23 01:45:03.000000000 +0900
@@ -1,4 +1,5 @@
 #include <clocale>
+#include <cstdio>
 #include <cstring>
 #include <iostream>
 using namespace std;
--- xplanet-1.2.1/src/Separation.cpp.g++44	2008-11-14 09:19:05.000000000 +0900
+++ xplanet-1.2.1/src/Separation.cpp	2009-04-23 01:46:37.000000000 +0900
@@ -1,5 +1,6 @@
 #include <cmath>
 #include <iostream>
+#include <cstdio>
 
 #include "Separation.h"
 #include "View.h"

