diff -Naur OpenSceneGraph-OpenSceneGraph-3.5.3/src/osgPlugins/jpeg/ReaderWriterJPEG.cpp OpenSceneGraph-OpenSceneGraph-3.5.3-p/src/osgPlugins/jpeg/ReaderWriterJPEG.cpp
--- OpenSceneGraph-OpenSceneGraph-3.5.3/src/osgPlugins/jpeg/ReaderWriterJPEG.cpp	2016-06-03 19:24:53.000000000 +0200
+++ OpenSceneGraph-OpenSceneGraph-3.5.3-p/src/osgPlugins/jpeg/ReaderWriterJPEG.cpp	2016-06-08 15:32:30.884989429 +0200
@@ -71,13 +71,16 @@
 static int jpegerror = ERR_NO_ERROR;
 
 /* Some versions of jmorecfg.h define boolean, some don't...
-   Those that do also define HAVE_BOOLEAN, so we can guard using that. */
+   Those that do also define HAVE_BOOLEAN, so we can guard using that.
 #ifndef HAVE_BOOLEAN
   typedef int boolean;
   #define FALSE 0
   #define TRUE 1
 #endif
 
+ this is for the evil OS not for linux
+*/
+
 /* CODE FOR READING/WRITING JPEG FROM STREAMS
  *  This code was taken directly from jdatasrc.c and jdatadst.c (libjpeg source)
  *  and modified to use a std::istream/ostream* instead of a FILE*
