diff -aur libixp-0.4/config.mk fw_libixp-0.4/config.mk
--- libixp-0.4/config.mk	2007-11-05 15:36:22.000000000 +0100
+++ fw_libixp-0.4/config.mk	2007-11-11 11:26:46.000000000 +0100
@@ -11,12 +11,12 @@
 TASKINC = -I${HOME}/libtask
 
 # paths
-PREFIX = /usr/local
-BIN = ${PREFIX}/bin
-MAN = ${PREFIX}/share/man
-ETC = ${PREFIX}/etc
-LIBDIR = ${PREFIX}/lib
-INCLUDE = ${PREFIX}/include
+PREFIX = /usr
+BIN = ${DESTDIR}/${PREFIX}/bin
+MAN = ${DESTDIR}/${PREFIX}/share/man
+ETC = ${DESTDIR}/etc
+LIBDIR = ${DESTDIR}/${PREFIX}/lib
+INCLUDE = ${DESTDIR}/${PREFIX}/include
 
 # Includes and libs
 INCPATH = .:${ROOT}/include:${INCLUDE}:/usr/include
