--- libproxy-0.2.3old/src/plugins/gnome.c	2008-06-05 22:56:14.000000000 +0200
+++ libproxy-0.2.3/src/plugins/gnome.c	2009-07-12 22:35:31.000000000 +0200
@@ -98,7 +98,7 @@
 												GCONF_VALUE_STRING, NULL);
 		if (ignores)
 		{
-			GSList *start = ignores;
+			//GSList *start = ignores;
 			for ( ; ignores ; ignores = g_slist_next(ignores))
 			{
 				if (ignore)
@@ -109,7 +109,7 @@
 				}
 				else
 					ignore = g_strdup(ignores->data);
-				g_free(ignores->data);
+				//g_free(ignores->data);
 			}
 			g_slist_free(start);
 		}
