diff -ruNp libnl-1.0~pre5+svn21~/include/netlink-local.h libnl-1.0~pre5+svn21/include/netlink-local.h
--- libnl-1.0~pre5+svn21~/include/netlink-local.h	2005-12-12 08:24:15.000000000 +0000
+++ libnl-1.0~pre5+svn21/include/netlink-local.h	2006-03-22 01:59:51.000000000 +0000
@@ -30,11 +30,7 @@
 #include <arpa/inet.h>
 #include <netdb.h>
 
-typedef uint8_t		__u8;
-typedef uint16_t	__u16;
-typedef uint32_t	__u32;
-typedef int32_t		__s32;
-typedef uint64_t	__u64;
+#include <linux/types.h>
 
 /* local header copies */
 #include <linux/if.h>
