diff -Naur PyKDE-3.16.0/configure.py PyKDE-3.16.0-p/configure.py
--- PyKDE-3.16.0/configure.py	2006-09-22 10:05:49.000000000 +0200
+++ PyKDE-3.16.0-p/configure.py	2006-09-26 00:35:51.340001398 +0200
@@ -542,7 +542,7 @@
     kde_sip_flags.append(sipconfig.version_to_sip_tag(kde_version, kdetags, "KDE"))
 
 
-def generate_code(mname, imports=None, extra_cflags=None, extra_cxxflags=None, extra_define=None, extra_include_dir=None, extra_lflags=None, extra_lib_dir=None, extra_lib=None, opengl=0, sip_flags=None):
+def generate_code(mname, imports=None, extra_cflags=None, extra_cxxflags="-fno-strict-aliasing -Wno-deprecated", extra_define=None, extra_include_dir=None, extra_lflags=None, extra_lib_dir=None, extra_lib=None, opengl=0, sip_flags=None):
     """Generate the code for a module.
 
     mname is the name of the module.
