Nur in PyKDE-3.16.0.py25: dcop.
diff -ru PyKDE-3.16.0/sip/kdecore/bytearray.sip PyKDE-3.16.0.py25/sip/kdecore/bytearray.sip
--- PyKDE-3.16.0/sip/kdecore/bytearray.sip	2006-09-22 07:53:14.000000000 +0200
+++ PyKDE-3.16.0.py25/sip/kdecore/bytearray.sip	2006-12-23 15:42:13.000000000 +0100
@@ -633,7 +633,7 @@
     QMap<QCString,DCOPRef> *cppmap = new QMap<QCString,DCOPRef>;
 
     PyObject *aelem, *belem;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     QCString *acpp;
     DCOPRef *bcpp;
 
@@ -717,7 +717,7 @@
     QMap<QString,DCOPRef> *cppmap = new QMap<QString,DCOPRef>;
 
     PyObject *aelem, *belem;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     QString *acpp;
     DCOPRef *bcpp;
 
diff -ru PyKDE-3.16.0/sip/kdecore/kaccel.sip PyKDE-3.16.0.py25/sip/kdecore/kaccel.sip
--- PyKDE-3.16.0/sip/kdecore/kaccel.sip	2006-09-22 05:11:16.000000000 +0200
+++ PyKDE-3.16.0.py25/sip/kdecore/kaccel.sip	2006-12-23 15:42:22.000000000 +0100
@@ -209,7 +209,7 @@
     QMap<int,QString> *map = new QMap<int,QString>;
 
         PyObject *key, *value;
-        int pos = 0;
+        SIP_SSIZE_T pos = 0;
         int iKey;
         QString *sData;
 
diff -ru PyKDE-3.16.0/sip/kdecore/kcompletion.sip PyKDE-3.16.0.py25/sip/kdecore/kcompletion.sip
--- PyKDE-3.16.0/sip/kdecore/kcompletion.sip	2006-09-22 05:11:16.000000000 +0200
+++ PyKDE-3.16.0.py25/sip/kdecore/kcompletion.sip	2006-12-23 15:42:31.000000000 +0100
@@ -201,7 +201,7 @@
     QMap<KCompletionBase::KeyBindingType,KShortcut> *kbmap = new QMap<KCompletionBase::KeyBindingType,KShortcut>;
 
     PyObject *key, *value;
-        int pos = 0;
+        SIP_SSIZE_T pos = 0;
         KShortcut *cValue;
         int iserr;
 
diff -ru PyKDE-3.16.0/sip/kdecore/kconfigdata.sip PyKDE-3.16.0.py25/sip/kdecore/kconfigdata.sip
--- PyKDE-3.16.0/sip/kdecore/kconfigdata.sip	2006-09-22 05:11:16.000000000 +0200
+++ PyKDE-3.16.0.py25/sip/kdecore/kconfigdata.sip	2006-12-23 15:42:39.000000000 +0100
@@ -146,7 +146,7 @@
     KEntryMap *cppmap = new KEntryMap;
 
     PyObject *aelem, *belem;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     KEntryKey *acpp;
     KEntry *bcpp;
 
diff -ru PyKDE-3.16.0/sip/kdecore/kconfig.sip PyKDE-3.16.0.py25/sip/kdecore/kconfig.sip
--- PyKDE-3.16.0/sip/kdecore/kconfig.sip	2006-09-22 05:11:16.000000000 +0200
+++ PyKDE-3.16.0.py25/sip/kdecore/kconfig.sip	2006-12-23 15:42:45.000000000 +0100
@@ -158,7 +158,7 @@
     QMap<QString,QString> *cppmap = new QMap<QString,QString>;
 
     PyObject *aelem, *belem;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     QString *acpp;
     QString *bcpp;
 
diff -ru PyKDE-3.16.0/sip/kdecore/kmacroexpander.sip PyKDE-3.16.0.py25/sip/kdecore/kmacroexpander.sip
--- PyKDE-3.16.0/sip/kdecore/kmacroexpander.sip	2006-09-22 05:11:16.000000000 +0200
+++ PyKDE-3.16.0.py25/sip/kdecore/kmacroexpander.sip	2006-12-23 15:42:57.000000000 +0100
@@ -169,7 +169,7 @@
     QMap<QChar,QString> *map = new QMap<QChar,QString>;
 
     PyObject *key, *value;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     QChar *cKey;
     QString *sData;
 
@@ -251,7 +251,7 @@
     QMap<QString, QStringList> *map = new QMap<QString, QStringList>;
 
     PyObject *key, *value;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     QString *sKey;
     QStringList *slData;
 
@@ -334,7 +334,7 @@
     QMap<QChar,QStringList> *map = new QMap<QChar,QStringList>;
 
     PyObject *key, *value;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     QChar *cKey;
     QStringList *slData;
 
diff -ru PyKDE-3.16.0/sip/kdecore/ktimezones.sip PyKDE-3.16.0.py25/sip/kdecore/ktimezones.sip
--- PyKDE-3.16.0/sip/kdecore/ktimezones.sip	2006-09-22 05:11:16.000000000 +0200
+++ PyKDE-3.16.0.py25/sip/kdecore/ktimezones.sip	2006-12-23 15:43:04.000000000 +0100
@@ -240,7 +240,7 @@
     QMap<QString, KTimezone*> *cppmap = new QMap<QString, KTimezone*>;
 
     PyObject *aelem, *belem;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     QString *acpp;
     KTimezone *bcpp;
 
diff -ru PyKDE-3.16.0/sip/kdecore/kurl.sip PyKDE-3.16.0.py25/sip/kdecore/kurl.sip
--- PyKDE-3.16.0/sip/kdecore/kurl.sip	2006-09-22 05:11:16.000000000 +0200
+++ PyKDE-3.16.0.py25/sip/kdecore/kurl.sip	2006-12-23 15:43:18.000000000 +0100
@@ -248,7 +248,7 @@
 %MethodCode
 //takes range | (a Python slice) | urlList | (KURL.List)
 
-    int len, start, stop, step, slicelength;
+    SIP_SSIZE_T len, start, stop, step, slicelength;
 
     len = sipCpp -> count();
 
@@ -291,7 +291,7 @@
         void                 __delitem__ (SIP_PYSLICE);
 %MethodCode
 //takes range | (a Python slice)
-    int len, start, stop, step, slicelength;
+    SIP_SSIZE_T len, start, stop, step, slicelength;
 
     len = sipCpp -> count();
     if (sipConvertFromSliceObject(a0,len,&start,&stop,&step,&slicelength) < 0)
@@ -322,7 +322,7 @@
 %MethodCode
 //returns (KURL.List)
 //takes range | (a Python slice)
-    int len, start, stop, step, slicelength;
+    SIP_SSIZE_T len, start, stop, step, slicelength;
 
     len = sipCpp -> count();
 
diff -ru PyKDE-3.16.0/sip/kdeui/kkeydialog.sip PyKDE-3.16.0.py25/sip/kdeui/kkeydialog.sip
--- PyKDE-3.16.0/sip/kdeui/kkeydialog.sip	2006-09-22 05:11:15.000000000 +0200
+++ PyKDE-3.16.0.py25/sip/kdeui/kkeydialog.sip	2006-12-23 15:43:23.000000000 +0100
@@ -231,7 +231,7 @@
     QDict<int>* qdict = new QDict<int>;
 
     PyObject *key, *value;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     QString *a0;
     int a1;
 
diff -ru PyKDE-3.16.0/sip/kio/authinfo.sip PyKDE-3.16.0.py25/sip/kio/authinfo.sip
--- PyKDE-3.16.0/sip/kio/authinfo.sip	2006-09-22 05:11:14.000000000 +0200
+++ PyKDE-3.16.0.py25/sip/kio/authinfo.sip	2006-12-23 15:43:28.000000000 +0100
@@ -176,7 +176,7 @@
     QMap<QString, QStringList> *map = new QMap<QString, QStringList>;
 
     PyObject *key, *value;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     QString *sKey;
     QStringList *sData;
 
diff -ru PyKDE-3.16.0/sip/kio/global.sip PyKDE-3.16.0.py25/sip/kio/global.sip
--- PyKDE-3.16.0/sip/kio/global.sip	2006-09-22 05:11:14.000000000 +0200
+++ PyKDE-3.16.0.py25/sip/kio/global.sip	2006-12-23 15:43:33.000000000 +0100
@@ -702,7 +702,7 @@
     QMap<QString,QString> *cppmap = new QMap<QString,QString>;
 
     PyObject *aelem, *belem;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     QString *acpp;
     QString *bcpp;
 
diff -ru PyKDE-3.16.0/sip/kio/kservicetype.sip PyKDE-3.16.0.py25/sip/kio/kservicetype.sip
--- PyKDE-3.16.0/sip/kio/kservicetype.sip	2006-09-22 05:11:14.000000000 +0200
+++ PyKDE-3.16.0.py25/sip/kio/kservicetype.sip	2006-12-23 15:43:44.000000000 +0100
@@ -183,7 +183,7 @@
     QMap<QString,QVariant::Type> *cppmap = new QMap<QString,QVariant::Type>;
 
     PyObject *aelem, *belem;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     QString *acpp;
     QVariant::Type bcpp;
 
diff -ru PyKDE-3.16.0/sip/kmdi/kmdichildfrm.sip PyKDE-3.16.0.py25/sip/kmdi/kmdichildfrm.sip
--- PyKDE-3.16.0/sip/kmdi/kmdichildfrm.sip	2006-09-22 05:11:14.000000000 +0200
+++ PyKDE-3.16.0.py25/sip/kmdi/kmdichildfrm.sip	2006-12-23 15:39:49.000000000 +0100
@@ -245,7 +245,7 @@
     QDict<QWidget::FocusPolicy>* qdict = new QDict<QWidget::FocusPolicy>;
 
     PyObject *key, *value;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     QString *a0;
     QWidget::FocusPolicy a1;
 
diff -ru PyKDE-3.16.0/sip/kparts/browserextension.sip PyKDE-3.16.0.py25/sip/kparts/browserextension.sip
--- PyKDE-3.16.0/sip/kparts/browserextension.sip	2006-09-22 05:11:15.000000000 +0200
+++ PyKDE-3.16.0.py25/sip/kparts/browserextension.sip	2006-12-23 15:33:51.000000000 +0100
@@ -419,7 +419,7 @@
     QMap<QCString,QCString> *cppmap = new QMap<QCString,QCString>;
 
     PyObject *aelem, *belem;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     QCString *acpp;
     QCString *bcpp;
 
@@ -503,7 +503,7 @@
     QMap<QCString,int> *cppmap = new QMap<QCString,int>;
 
     PyObject *aelem, *belem;
-    int pos = 0;
+    SIP_SSIZE_T pos = 0;
     QCString *acpp;
     int bcpp;
 
