diff -Naur PyKDE-3.16.0/configure.py PyKDE-3.16.0-p/configure.py
--- PyKDE-3.16.0/configure.py	2006-09-22 10:05:49.000000000 +0200
+++ PyKDE-3.16.0-p/configure.py	2006-09-26 01:41:32.630834118 +0200
@@ -766,10 +766,11 @@
         f = open ("Makefile", "r")
         buff = f.read ()
         f.close ()
-        target  = os.path.join (opt_kdelibdir, "kde3", "libkonsolepart.so")
-        symlink = os.path.join (opt_kdelibdir, "libkonsolepart.so")
-        buff = buff.replace ("install:\n", "install:\n\tln -sf %s %s\n" % (target, symlink))
-        buff = buff.replace ("\nclean", "\tldconfig\n\nclean")
+	## F****G THINK FIRST ABOUT *WHAT* YOU ARE DOING HERE ##
+        #target  = os.path.join (opt_kdelibdir, "kde3", "libkonsolepart.so")
+        #symlink = os.path.join (opt_kdelibdir, "libkonsolepart.so")
+        #buff = buff.replace ("install:\n", "install:\n\tln -sf %s %s\n" % (target, symlink))
+        #buff = buff.replace ("\nclean", "\tldconfig\n\nclean")
         f = open ("Makefile.new", "w")
         f.write (buff)
         f.close ()
