diff -Naur lgeneral-1.2beta-9.orig/src/misc.h lgeneral-1.2beta-9/src/misc.h
--- lgeneral-1.2beta-9.orig/src/misc.h	2005-11-25 22:20:53.254856500 +0100
+++ lgeneral-1.2beta-9/src/misc.h	2005-11-25 22:21:17.976401500 +0100
@@ -45,6 +45,8 @@
 /* return maximum */
 #define MAXIMUM( a, b ) ((a>b)?a:b)
 
+#define NDEBUG 1
+
 /* compile time assert */
 #ifndef NDEBUG
 #  define COMPILE_TIME_ASSERT( x ) do { switch (0) {case 0: case x:;} } while(0)
