diff -Naur gparted-0.0.9.orig/gparted.desktop gparted-0.0.9/gparted.desktop
--- gparted-0.0.9.orig/gparted.desktop	2005-11-28 14:12:47.336676750 +0100
+++ gparted-0.0.9/gparted.desktop	2005-11-28 14:13:00.537501750 +0100
@@ -54,9 +54,10 @@
 Comment[vi]=Bộ hiệu chỉnh phân vùng Gnome
 Comment[zh_CN]=Gnome 分区编辑器
 Comment[zh_TW]=Gnome 分割區編輯器
-Exec=gparted
+Exec=gksu gparted
 Icon=gparted.png
 Terminal=false
 Type=Application
 Categories=GNOME;Application;System;
 StartupNotify=true
+NotShowIn=KDE
diff -Naur gparted-0.0.9.orig/gparted.desktop.in gparted-0.0.9/gparted.desktop.in
--- gparted-0.0.9.orig/gparted.desktop.in	2005-11-28 14:12:47.336676750 +0100
+++ gparted-0.0.9/gparted.desktop.in	2005-11-28 14:13:06.169853750 +0100
@@ -2,9 +2,10 @@
 Encoding=UTF-8
 _Name=GParted
 _Comment=Gnome Partition Editor
-Exec=gparted
+Exec=gksu gparted
 Icon=gparted.png
 Terminal=false
 Type=Application
 Categories=GNOME;Application;System;
 StartupNotify=true
+NotShowIn=KDE
