diff -Naur knemo-0.4.4/src/knemod/knemod.desktop knemo-0.4.4-p/src/knemod/knemod.desktop
--- knemo-0.4.4/src/knemod/knemod.desktop	2006-08-09 00:13:44.000000000 +0200
+++ knemo-0.4.4-p/src/knemod/knemod.desktop	2006-08-10 19:31:30.084015032 +0200
@@ -5,7 +5,8 @@
 X-KDE-ModuleType=Library
 X-KDE-Library=knemod
 X-KDE-FactoryName=knemod
-X-KDE-Kded-load-on-demand=true
+X-KDE-Kded-autoload=true
+X-KDE-Kded-load-on-demand=false
 Name=KNemo
 Name[sv]=Knemo
 Name[xx]=xxKNemoxx
