--- sound-theme/gvc-sound-theme-chooser.c.orig	2013-08-25 10:19:06.754817881 +0200
+++ sound-theme/gvc-sound-theme-chooser.c	2013-08-25 10:19:19.527657904 +0200
@@ -31,7 +31,7 @@
 #include <glib/gi18n.h>
 #include <gio/gio.h>
 #include <gtk/gtk.h>
-#include <canberra-gtk.h>
+#include <canberra-gtk2.h>
 #include <libxml/tree.h>
 
 #include "gvc-sound-theme-chooser.h"
--- mate-volume-control/src/gvc-channel-bar.c.orig	2013-08-25 10:19:59.209160990 +0200
+++ mate-volume-control/src/gvc-channel-bar.c	2013-08-25 10:20:09.910024266 +0200
@@ -29,7 +29,7 @@
 #include <glib.h>
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
-#include <canberra-gtk.h>
+#include <canberra-gtk2.h>
 
 #include "gvc-channel-bar.h"
 
--- mate-volume-control/src/gvc-balance-bar.c.orig	2013-08-25 10:20:23.641855038 +0200
+++ mate-volume-control/src/gvc-balance-bar.c	2013-08-25 10:20:36.669691445 +0200
@@ -27,7 +27,7 @@
 #include <glib.h>
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
-#include <canberra-gtk.h>
+#include <canberra-gtk2.h>
 
 #include "gvc-balance-bar.h"
 
--- mate-volume-control/src/gvc-combo-box.c.orig	2013-08-25 10:20:51.784502650 +0200
+++ mate-volume-control/src/gvc-combo-box.c	2013-08-25 10:21:01.109385895 +0200
@@ -27,7 +27,7 @@
 #include <glib.h>
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
-#include <canberra-gtk.h>
+#include <canberra-gtk2.h>
 
 #include "gvc-combo-box.h"
 #include "gvc-mixer-stream.h"
--- mate-volume-control/src/gvc-speaker-test.c.orig	2013-08-25 10:21:14.944212671 +0200
+++ mate-volume-control/src/gvc-speaker-test.c	2013-08-25 10:21:26.564067269 +0200
@@ -27,8 +27,7 @@
 #include <glib.h>
 #include <glib/gi18n.h>
 #include <gtk/gtk.h>
-#include <canberra.h>
-#include <canberra-gtk.h>
+#include <canberra-gtk2.h>
 
 #include "gvc-speaker-test.h"
 #include "gvc-mixer-stream.h"
