diff -Naur xsys-2.0.8.orig/pci.c xsys-2.0.8/pci.c
--- xsys-2.0.8.orig/pci.c	2005-08-10 00:22:03.000000000 +0200
+++ xsys-2.0.8/pci.c	2005-09-22 16:46:26.000000000 +0200
@@ -25,6 +25,7 @@
 #include <ctype.h>
 #include <unistd.h>
 #include <pci/pci.h>
+#include <pci/sysdep.h>
 #include "xsys.h"
 
 struct pci_filter filter;       /* Device filter */
diff -Naur xsys-2.0.8.orig/pci.h xsys-2.0.8/pci.h
--- xsys-2.0.8.orig/pci.h	2005-06-19 16:46:12.000000000 +0200
+++ xsys-2.0.8/pci.h	2005-09-22 16:46:01.000000000 +0200
@@ -22,6 +22,7 @@
 #define _PCI_H_
 
 #include <pci/pci.h>
+#include <pci/sysdep.h>
 void pci_find_fullname(char *fullname, char *vendor, char *device);
 int  pci_find_by_class(word *class, char *vendor, char *device);
 
