diff -Naur xsys-2.0.8/Makefile xsys-2.0.8-patched/Makefile
--- xsys-2.0.8/Makefile	2005-08-05 21:52:28.000000000 +0200
+++ xsys-2.0.8-patched/Makefile	2005-11-09 20:04:46.000000000 +0100
@@ -10,15 +10,15 @@
 VER_PATCH = 8
 CC = gcc
 CFLAGS += -O2 -Wall -fPIC
-# FOR BMP # CFLAGS += -I/usr/include/beep-media-player -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -DUSE_BMP
-# FOR XMMS # CFLAGS += -I/usr/include/xmms -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -DUSE_XMMS
+CFLAGS += -I/usr/include/beep-media-player -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -DUSE_BMP
+CFLAGS += -I/usr/include/xmms -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -DUSE_XMMS
 CFLAGS += -DVER_MINOR=$(VER_MINOR) -DVER_MAJOR=$(VER_MAJOR) -DVER_PATCH=$(VER_PATCH) \
           -DVER_STRING=\"$(VER_MAJOR).$(VER_MINOR).$(VER_PATCH)\" -DPCIIDS=\"$(PCIIDS)\" $(BUTTON)
 LDFLAGS = $(CFLAGS) -shared
 LIBRARY = xsys-$(VER_MAJOR).$(VER_MINOR).$(VER_PATCH).so
 OBJECTS = xsys.o parse.o pci.o match.o
-# FOR BMP # NPLIB = -lbeep
-# FOR XMMS # NPLIB = -lxmms
+NPLIB = -lbeep
+NPLIB = -lxmms
 
 ALL : $(LIBRARY)
 
