diff -Naur xsys-2.0.5.orig/Makefile xsys-2.0.5/Makefile
--- xsys-2.0.5.orig/Makefile	2005-06-30 15:54:22.000000000 +0200
+++ xsys-2.0.5/Makefile	2005-06-30 15:54:40.000000000 +0200
@@ -1,8 +1,8 @@
 #### SET THIS VALUE TO THE LOCATION OF THE `pci.ids` file ####
-PCIIDS = /usr/share/misc/pci.ids
+PCIIDS = /usr/share/pci.ids
 
 #### UNCOMMENT THIS IF YOU WANT THE BUTTONS ####
-#BUTTON = -Dbuttonbar
+BUTTON = -Dbuttonbar
 
 #### SHOULD NOT NEED TO EDIT BELOW THIS LINE ####
 VER_MAJOR = 2
