diff -Naur xsys-2.0.9.orig/Makefile xsys-2.0.9/Makefile
--- xsys-2.0.9.orig/Makefile	2005-12-09 17:27:12.489751250 +0100
+++ xsys-2.0.9/Makefile	2005-12-09 17:28:10.341366750 +0100
@@ -2,7 +2,7 @@
 PCIIDS = /usr/share/misc/pci.ids
 
 #### UNCOMMENT THIS IF YOU WANT THE BUTTONS ####
-#BUTTON = -Dbuttonbar
+BUTTON = -Dbuttonbar
 
 #### SHOULD NOT NEED TO EDIT BELOW THIS LINE ####
 VER_MAJOR = 2
@@ -11,16 +11,16 @@
 CC = gcc
 CFLAGS += -O2 -Wall -fPIC
 # FOR AUDACIOUS # CFLAGS += -I/usr/include/audacious -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -DUSE_AUDACIOUS
-# FOR BMP # CFLAGS += -I/usr/include/beep-media-player -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -DUSE_BMP
-# FOR XMMS # CFLAGS += -I/usr/include/xmms -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -DUSE_XMMS
+CFLAGS += -I/usr/include/beep-media-player -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -DUSE_BMP
+CFLAGS += -I/usr/include/xmms -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -DUSE_XMMS
 CFLAGS += -DVER_MINOR=$(VER_MINOR) -DVER_MAJOR=$(VER_MAJOR) -DVER_PATCH=$(VER_PATCH) \
           -DVER_STRING=\"$(VER_MAJOR).$(VER_MINOR).$(VER_PATCH)\" -DPCIIDS=\"$(PCIIDS)\" $(BUTTON)
 LDFLAGS = $(CFLAGS) -shared
 LIBRARY = xsys-$(VER_MAJOR).$(VER_MINOR).$(VER_PATCH).so
 OBJECTS = xsys.o parse.o pci.o match.o hwmon.o
 # FOR AUDACIOUS # NPLIB = -laudacious
-# FOR BMP # NPLIB = -lbeep
-# FOR XMMS # NPLIB = -lxmms
+NPLIB = -lbeep
+NPLIB = -lxmms
 
 ALL : $(LIBRARY)
 
