diff -aur vamp-plugin-sdk-1.2.orig/vamp-sdk/Plugin.h vamp-plugin-sdk-1.2.new/vamp-sdk/Plugin.h
--- vamp-plugin-sdk-1.2.orig/vamp-sdk/Plugin.h	2008-02-28 17:07:44.000000000 +0300
+++ vamp-plugin-sdk-1.2.new/vamp-sdk/Plugin.h	2008-06-19 20:29:04.044144500 +0400
@@ -40,6 +40,9 @@
 #include <string>
 #include <vector>
 #include <map>
+#include <cstdlib>
+#include <cstdio>
+#include <cstring>
 
 #include "PluginBase.h"
 #include "RealTime.h"
 
