diff -Naur zapping-0.10cvs6/configure.in zapping-0.10cvs6-p/configure.in
--- zapping-0.10cvs6/configure.in	2006-06-10 18:57:08.000000000 +0200
+++ zapping-0.10cvs6-p/configure.in	2007-09-25 21:20:10.000000000 +0200
@@ -832,16 +832,13 @@
 dnl Documentation
 dnl ---------------------------------------------------------------------------
 
-SCROLLKEEPER_BUILD_REQUIRED=0.3.5
-AC_SUBST(SCROLLKEEPER_BUILD_REQUIRED)
+#RARIAN_BUILD_REQUIRED>=0.6.0
+#AC_SUBST(RARIAN_BUILD_REQUIRED)
 
-AC_PATH_PROG(SCROLLKEEPER_CONFIG, [scrollkeeper-config], [no])
-if test "x$SCROLLKEEPER_CONFIG" = "xno"; then
+PKG_CHECK_MODULES([RARIAN], [rarian >= 0.6.0], [has_rarian="yes"], [has_rarian="no"])
+if test "x$has_rarian" = "xno"; then
   AC_MSG_FAILURE([
-*** Couldn't find scrollkeeper-config. Scrollkeeper is required to
-*** install the Zapping documentation. Download the appropriate
-*** package for your distribution or get the source tarball from
-*** http://scrollkeeper.sourceforge.net
+*** Couldn't rarian. Rarian is required to install the Zapping documentation.
   ])
 fi
 
