=== modified file 'account.c'
--- account.c	2008-09-01 10:21:01 +0000
+++ account.c	2009-10-22 21:55:23 +0000
@@ -26,6 +26,7 @@
 #define BITLBEE_CORE
 #include "bitlbee.h"
 #include "account.h"
+#include "chat.h"
 
 account_t *account_add( irc_t *irc, struct prpl *prpl, char *user, char *pass )
 {

=== modified file 'bitlbee.h'
--- bitlbee.h	2009-10-17 15:43:12 +0000
+++ bitlbee.h	2009-10-22 21:55:23 +0000
@@ -128,7 +128,6 @@
 #include "commands.h"
 #include "account.h"
 #include "nick.h"
-#include "chat.h"
 #include "conf.h"
 #include "log.h"
 #include "ini.h"

=== modified file 'chat.c'
--- chat.c	2008-12-14 01:14:55 +0000
+++ chat.c	2009-10-22 21:55:23 +0000
@@ -24,6 +24,7 @@
 */
 
 #include "bitlbee.h"
+#include "chat.h"
 
 struct chat *chat_add( irc_t *irc, account_t *acc, char *handle, char *channel )
 {

=== modified file 'chat.h'
--- chat.h	2008-09-28 11:18:19 +0000
+++ chat.h	2009-10-22 21:55:23 +0000
@@ -23,6 +23,9 @@
   Suite 330, Boston, MA  02111-1307  USA
 */
 
+#ifndef _CHAT_H
+#define _CHAT_H
+
 struct chat
 {
 	account_t *acc;
@@ -44,3 +47,5 @@
 int chat_chanok( char *a );
 
 int chat_join( irc_t *irc, struct chat *c, const char *password );
+
+#endif

=== modified file 'irc_commands.c'
--- irc_commands.c	2009-10-04 19:00:53 +0000
+++ irc_commands.c	2009-10-22 21:55:23 +0000
@@ -26,6 +26,7 @@
 #define BITLBEE_CORE
 #include "bitlbee.h"
 #include "ipc.h"
+#include "chat.h"
 
 static void irc_cmd_pass( irc_t *irc, char **cmd )
 {

=== modified file 'protocols/nogaim.h'
--- protocols/nogaim.h	2009-10-10 13:39:51 +0000
+++ protocols/nogaim.h	2009-10-22 21:55:23 +0000
@@ -43,6 +43,7 @@
 #include "proxy.h"
 #include "query.h"
 #include "md5.h"
+#include "chat.h"
 
 #define BUDDY_ALIAS_MAXLEN 388   /* because MSN names can be 387 characters */
 

=== modified file 'root_commands.c'
--- root_commands.c	2009-06-24 21:48:44 +0000
+++ root_commands.c	2009-10-22 21:55:23 +0000
@@ -28,6 +28,7 @@
 #include "crypting.h"
 #include "bitlbee.h"
 #include "help.h"
+#include "chat.h"
 
 #include <string.h>
 

=== modified file 'storage_xml.c'
--- storage_xml.c	2008-09-07 14:34:50 +0000
+++ storage_xml.c	2009-10-22 21:55:23 +0000
@@ -28,6 +28,7 @@
 #include "base64.h"
 #include "arc.h"
 #include "md5.h"
+#include "chat.h"
 
 #if GLIB_CHECK_VERSION(2,8,0)
 #include <glib/gstdio.h>

