=== modified file 'lib/misc.c'
--- lib/misc.c	2008-06-30 15:37:12 +0000
+++ lib/misc.c	2009-10-23 10:43:15 +0000
@@ -45,6 +45,7 @@
 #include <resolv.h>
 #endif
 
+#include "md5.h"
 #include "ssl_client.h"
 
 void strip_linefeed(gchar *text)

=== modified file 'protocols/jabber/jabber.h'
--- protocols/jabber/jabber.h	2008-09-28 11:18:19 +0000
+++ protocols/jabber/jabber.h	2009-10-23 10:43:15 +0000
@@ -26,8 +26,9 @@
 
 #include <glib.h>
 
+#include "bitlbee.h"
+#include "md5.h"
 #include "xmltree.h"
-#include "bitlbee.h"
 
 extern GSList *jabber_connections;
 

=== modified file 'protocols/nogaim.c'
--- protocols/nogaim.c	2009-10-10 13:39:51 +0000
+++ protocols/nogaim.c	2009-10-23 10:43:15 +0000
@@ -32,8 +32,10 @@
 */
 
 #define BITLBEE_CORE
+#include <ctype.h>
+
 #include "nogaim.h"
-#include <ctype.h>
+#include "chat.h"
 
 static int remove_chat_buddy_silent( struct groupchat *b, const char *handle );
 

=== modified file 'protocols/nogaim.h'
--- protocols/nogaim.h	2009-10-22 21:55:23 +0000
+++ protocols/nogaim.h	2009-10-23 10:43:15 +0000
@@ -38,12 +38,12 @@
 #ifndef _NOGAIM_H
 #define _NOGAIM_H
 
+#include <stdint.h>
+
 #include "bitlbee.h"
 #include "account.h"
 #include "proxy.h"
 #include "query.h"
-#include "md5.h"
-#include "chat.h"
 
 #define BUDDY_ALIAS_MAXLEN 388   /* because MSN names can be 387 characters */
 

