diff -up bitlbee-3.0.2/configure.systemd bitlbee-3.0.2/configure
--- bitlbee-3.0.2/configure.systemd	2011-03-09 09:46:37.263884813 +0100
+++ bitlbee-3.0.2/configure	2011-03-09 09:47:39.983708602 +0100
@@ -15,6 +15,7 @@ datadir='$prefix/share/bitlbee/'
 config='/var/lib/bitlbee/'
 plugindir='$prefix/lib/bitlbee/'
 includedir='$prefix/include/bitlbee/'
+systemdsystemunitdir=''
 libevent='/usr/'
 pidfile='/var/run/bitlbee.pid'
 ipcsocket='/var/run/bitlbee.sock'
@@ -61,6 +62,7 @@ Option		Description				Default
 --mandir=...						$mandir
 --datadir=...						$datadir
 --plugindir=...						$plugindir
+--systemdsystemunitdir=...				$systemdsystemunitdir
 --pidfile=...						$pidfile
 --config=...						$config
 --ipcsocket=...						$ipcsocket
@@ -488,6 +490,17 @@ else
 	fi;
 fi
 
+if [ -z "$systemdsystemunitdir" ]; then
+	if $PKG_CONFIG --exists systemd; then
+		systemdsystemunitdir=`$PKG_CONFIG --variable=systemdsystemunitdir systemd`
+	fi
+fi
+if [ -n "$systemdsystemunitdir" ]; then
+	if [ "$systemdsystemunitdir" != "no" ]; then
+		echo "SYSTEMDSYSTEMUNITDIR=$systemdsystemunitdir" >> Makefile.settings
+	fi
+fi
+
 if [ "$gcov" = "1" ]; then
 	echo "CFLAGS+=--coverage" >> Makefile.settings
 	echo "EFLAGS+=--coverage" >> Makefile.settings
@@ -728,6 +741,12 @@ else
 	echo '  Off-the-Record (OTR) Messaging disabled.'
 fi
 
+if [ -n "$systemdsystemunitdir" ]; then
+	echo '  systemd enabled.'
+else
+	echo '  systemd disabled.'
+fi
+
 echo '  Using event handler: '$events
 echo '  Using SSL library: '$ssl
 #echo '  Building with these storage backends: '$STORAGES
diff -up bitlbee-3.0.2/init/bitlbee.service.in.systemd bitlbee-3.0.2/init/bitlbee.service.in
--- bitlbee-3.0.2/init/bitlbee.service.in.systemd	2011-03-09 09:46:37.266884804 +0100
+++ bitlbee-3.0.2/init/bitlbee.service.in	2011-03-09 09:46:37.266884804 +0100
@@ -0,0 +1,9 @@
+[Unit]
+Description=BitlBee IRC/IM gateway
+After=syslog.target
+
+[Service]
+ExecStart=@sbindir@bitlbee -F -n
+
+[Install]
+WantedBy=multi-user.target
diff -up bitlbee-3.0.2/init/bitlbee@.service.in.systemd bitlbee-3.0.2/init/bitlbee@.service.in
--- bitlbee-3.0.2/init/bitlbee@.service.in.systemd	2011-03-09 09:46:37.267884801 +0100
+++ bitlbee-3.0.2/init/bitlbee@.service.in	2011-03-09 09:46:37.267884801 +0100
@@ -0,0 +1,8 @@
+[Unit]
+Description=BitlBee Per-Connection Server
+After=syslog.target
+
+[Service]
+ExecStart=@sbindir@bitlbee
+StandardInput=socket
+User=bitlbee
diff -up bitlbee-3.0.2/init/bitlbee.socket.systemd bitlbee-3.0.2/init/bitlbee.socket
--- bitlbee-3.0.2/init/bitlbee.socket.systemd	2011-03-09 09:46:37.267884801 +0100
+++ bitlbee-3.0.2/init/bitlbee.socket	2011-03-09 09:46:37.267884801 +0100
@@ -0,0 +1,9 @@
+[Unit]
+Conflicts=bitlbee.service
+
+[Socket]
+ListenStream=127.0.0.1:6667
+Accept=yes
+
+[Install]
+WantedBy=sockets.target
diff -up bitlbee-3.0.2/Makefile.systemd bitlbee-3.0.2/Makefile
--- bitlbee-3.0.2/Makefile.systemd	2011-03-08 07:56:10.000000000 +0100
+++ bitlbee-3.0.2/Makefile	2011-03-09 09:46:37.266884804 +0100
@@ -26,18 +26,18 @@ endif
 # Expansion of variables
 subdirobjs = $(foreach dir,$(subdirs),$(dir)/$(dir).o)
 
-all: $(OUTFILE) $(OTR_PI)
+all: $(OUTFILE) $(OTR_PI) systemd
 	$(MAKE) -C doc
 
 uninstall: uninstall-bin uninstall-doc 
 	@echo -e '\nmake uninstall does not remove files in '$(DESTDIR)$(ETCDIR)', you can use make uninstall-etc to do that.\n'
 
-install: install-bin install-doc install-plugins
+install: install-bin install-doc install-plugins install-systemd
 	@if ! [ -d $(DESTDIR)$(CONFIG) ]; then echo -e '\nThe configuration directory $(DESTDIR)$(CONFIG) does not exist yet, don'\''t forget to create it!'; fi
 	@if ! [ -e $(DESTDIR)$(ETCDIR)/bitlbee.conf ]; then echo -e '\nNo files are installed in '$(DESTDIR)$(ETCDIR)' by make install. Run make install-etc to do that.'; fi
 	@echo
 
-.PHONY:   install   install-bin   install-etc   install-doc install-plugins \
+.PHONY:   install   install-bin   install-etc   install-doc install-plugins install-systemd \
         uninstall uninstall-bin uninstall-etc uninstall-doc \
         all clean distclean tar $(subdirs)
 
@@ -109,6 +109,20 @@ ifdef OTR_PI
 	install -m 0755 otr.so $(DESTDIR)$(PLUGINDIR)
 endif
 
+systemd:
+ifdef SYSTEMDSYSTEMUNITDIR
+	sed 's|@sbindir@|$(BINDIR)|' init/bitlbee.service.in > init/bitlbee.service
+	sed 's|@sbindir@|$(BINDIR)|' init/bitlbee@.service.in > init/bitlbee@.service
+endif
+
+install-systemd:
+ifdef SYSTEMDSYSTEMUNITDIR
+	mkdir -p $(DESTDIR)$(SYSTEMDSYSTEMUNITDIR)
+	install -m 0644 init/bitlbee.service $(DESTDIR)$(SYSTEMDSYSTEMUNITDIR)
+	install -m 0644 init/bitlbee@.service $(DESTDIR)$(SYSTEMDSYSTEMUNITDIR)
+	install -m 0644 init/bitlbee.socket $(DESTDIR)$(SYSTEMDSYSTEMUNITDIR)
+endif
+
 tar:
 	fakeroot debian/rules clean || make distclean
 	x=$$(basename $$(pwd)); \
