=== modified file 'Makefile'
--- Makefile	2012-02-11 12:55:45 +0000
+++ Makefile	2012-02-22 08:33:05 +0000
@@ -135,8 +135,8 @@
 
 systemd:
 ifdef SYSTEMDSYSTEMUNITDIR
-	sed 's|@sbindir@|$(BINDIR)|' init/bitlbee.service.in > init/bitlbee.service
-	sed 's|@sbindir@|$(BINDIR)|' init/bitlbee@.service.in > init/bitlbee@.service
+	sed 's|@sbindir@|$(SBINDIR)|' init/bitlbee.service.in > init/bitlbee.service
+	sed 's|@sbindir@|$(SBINDIR)|' init/bitlbee@.service.in > init/bitlbee@.service
 endif
 
 install-systemd:

