diff -ur a/SConstruct b/SConstruct
--- a/SConstruct	2009-08-30 14:23:30.000000000 -0500
+++ b/SConstruct	2012-11-28 22:13:31.583885454 -0600
@@ -129,7 +129,7 @@
         print "Could not find libboost_date_time or libboost_date_time-mt or boost/thread/thread.hpp"
         missing.append("libboost_date_time")
     env.Append(LIBS=[boost_date_time])
-
+    env.Append(LIBS=["boost_system"])
 
     if not conf.CheckCXXHeader("boost/shared_ptr.hpp"):
         print "Could not find boost/shared_ptr.hpp"
