--- glob2-0.9.1/SConstruct.orig	2007-09-06 18:01:34.000000000 +0200
+++ glob2-0.9.1/SConstruct	2007-09-06 18:03:00.000000000 +0200
@@ -66,8 +66,8 @@
         else:
             print "Coulf not find libz or zlib1.dll"
             Exit(1)
-    if not conf.CheckLib('boost_thread') or not conf.CheckCXXHeader('boost/thread/thread.hpp'):
-        print "Could not find libboost_thread or boost/thread/thread.hpp"
+    if not conf.CheckLib('boost_thread-mt') or not conf.CheckCXXHeader('boost/thread/thread.hpp'):
+        print "Could not find libboost_thread-mt or boost/thread/thread.hpp"
         Exit(1)
     if not conf.CheckCXXHeader('boost/shared_ptr.hpp'):
         print "Could not find boost/shared_ptr.hpp"
@@ -149,7 +149,7 @@
         env.ParseConfig("sdl-config --libs")
     if env['osx']:
         env.Append(CXXFLAGS="-framework OpenGL")
-    env.Append(LIBS=['vorbisfile', 'SDL_ttf', 'SDL_image', 'SDL_net', 'speex', 'boost_thread'])
+    env.Append(LIBS=['vorbisfile', 'SDL_ttf', 'SDL_image', 'SDL_net', 'speex', 'boost_thread-mt'])
     
     env["TARFILE"] = env.Dir("#").abspath + "/glob2-" + env["VERSION"] + ".tar.gz"
     env["TARFLAGS"] = "-c -z"
