diff -up jreen-1.2.1/src/stanzaextension.h.qt56 jreen-1.2.1/src/stanzaextension.h
--- jreen-1.2.1/src/stanzaextension.h.qt56	2015-02-07 14:10:11.000000000 -0600
+++ jreen-1.2.1/src/stanzaextension.h	2016-02-19 22:09:50.903515303 -0600
@@ -28,7 +28,11 @@
 
 #include <QObject>
 #include <QSharedPointer>
+#if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
 #include <QMetaTypeId>
+#else
+#include <QMap>
+#endif
 #include "jreen.h"
 
 class QXmlStreamWriter;
diff -up jreen-1.2.1/src/tunefactory.cpp.qt56 jreen-1.2.1/src/tunefactory.cpp
