diff --git a/driver/Makefile b/driver/Makefile
index 5e53055..3780bf9 100644
--- a/driver/Makefile
+++ b/driver/Makefile
@@ -26,7 +26,7 @@ endif
 # Kernel Makefile doesn't always know the exact kernel version, so we
 # get it from the kernel headers instead and pass it to make.
 
-VERSION_H := $(KBUILD)/include/linux/utsrelease.h
+VERSION_H := $(KBUILD)/include/generated/utsrelease.h
 ifeq (,$(wildcard $(VERSION_H)))
 VERSION_H := $(KBUILD)/include/linux/version.h
 endif
diff --git a/driver/ntoskernel.h b/driver/ntoskernel.h
index db9d209..35bf68c 100644
--- a/driver/ntoskernel.h
+++ b/driver/ntoskernel.h
@@ -875,7 +875,7 @@ static inline struct nt_slist *PopEntrySList(nt_slist_header *head,
 #define u64_low_32(x) ((u32)x)
 #define u64_high_32(x) ((u32)(x >> 32))
 
-#if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,32)
+#if LINUX_VERSION_CODE <= KERNEL_VERSION(2,6,33)
 static inline u64 cmpxchg8b(volatile u64 *ptr, u64 old, u64 new)
 {
 	u64 prev;
