diff -Naur rustc-1.69.0-src/src/bootstrap/dist.rs rustc-1.69.0-src.new/src/bootstrap/dist.rs
--- rustc-1.69.0-src/src/bootstrap/dist.rs	2023-04-16 23:39:51.000000000 +0200
+++ rustc-1.69.0-src.new/src/bootstrap/dist.rs	2023-04-25 17:27:20.552709843 +0200
@@ -414,7 +414,7 @@
                 },
                 builder.kind,
             ) {
-                builder.install(&ra_proc_macro_srv, &image.join("libexec"), 0o755);
+                builder.install(&ra_proc_macro_srv, &image.join("lib"), 0o755);
             }
 
             let libdir_relative = builder.libdir_relative(compiler);
@@ -1064,7 +1064,7 @@
         for dirent in fs::read_dir(cargo.parent().unwrap()).expect("read_dir") {
             let dirent = dirent.expect("read dir entry");
             if dirent.file_name().to_str().expect("utf8").starts_with("cargo-credential-") {
-                tarball.add_file(&dirent.path(), "libexec", 0o755);
+                tarball.add_file(&dirent.path(), "lib", 0o755);
             }
         }
 
diff -Naur rustc-1.69.0-src/src/bootstrap/tool.rs rustc-1.69.0-src.new/src/bootstrap/tool.rs
--- rustc-1.69.0-src/src/bootstrap/tool.rs	2023-04-16 23:39:51.000000000 +0200
+++ rustc-1.69.0-src.new/src/bootstrap/tool.rs	2023-04-25 17:26:39.620393177 +0200
@@ -850,7 +850,7 @@
 
         // Copy `rust-analyzer-proc-macro-srv` to `<sysroot>/libexec/`
         // so that r-a can use it.
-        let libexec_path = builder.sysroot(self.compiler).join("libexec");
+        let libexec_path = builder.sysroot(self.compiler).join("lib");
         t!(fs::create_dir_all(&libexec_path));
         builder.copy(&path, &libexec_path.join("rust-analyzer-proc-macro-srv"));
 
