--- rustc-1.56.1-src/src/tools/cargo/src/cargo/ops/registry/auth.rs	2021-11-01 08:17:47.000000000 +0100
+++ rustc-1.56.1-src.new/src/tools/cargo/src/cargo/ops/registry/auth.rs	2021-11-09 16:58:56.162332715 +0100
@@ -212,7 +212,7 @@
     Ok(token)
 }
 
-/// Gets the path to the libexec processes in the sysroot.
+/// Gets the path to the lib processes in the sysroot.
 fn sysroot_credential(
     config: &Config,
     process: &(PathBuf, Vec<String>),
@@ -223,7 +223,7 @@
         .parent()
         .and_then(|p| p.parent())
         .ok_or_else(|| format_err!("expected cargo path {}", cargo.display()))?;
-    let exe = root.join("libexec").join(format!(
+    let exe = root.join("lib").join(format!(
         "cargo-credential-{}{}",
         cred_name,
         std::env::consts::EXE_SUFFIX
