diff -Naur kate-18.08.0/kate/main.cpp kate-18.08.0-p/kate/main.cpp
--- kate-18.08.0/kate/main.cpp	2018-08-06 22:41:28.000000000 +0200
+++ kate-18.08.0-p/kate/main.cpp	2018-08-20 13:12:39.546570043 +0200
@@ -60,18 +60,7 @@
 
 int main(int argc, char **argv)
 {
-#ifndef Q_OS_WIN
-    // Prohibit using sudo or kdesu (but allow using the root user directly)
-    if (getuid() == 0) {
-        if (!qEnvironmentVariableIsEmpty("SUDO_USER")) {
-            std::cout << "Executing Kate with sudo is not possible due to unfixable security vulnerabilities." << std::endl;
-            return EXIT_FAILURE;
-        } else if (!qEnvironmentVariableIsEmpty("KDESU_USER")) {
-            std::cout << "Executing Kate with kdesu is not possible due to unfixable security vulnerabilities." << std::endl;
-            return EXIT_FAILURE;
-        }
-    }
-#endif
+
     /**
      * init resources from our static lib
      */
diff -Naur kate-18.08.0/kwrite/main.cpp kate-18.08.0-p/kwrite/main.cpp
--- kate-18.08.0/kwrite/main.cpp	2018-08-06 22:41:28.000000000 +0200
+++ kate-18.08.0-p/kwrite/main.cpp	2018-08-20 13:13:18.431739523 +0200
@@ -49,18 +49,6 @@
 
 extern "C" Q_DECL_EXPORT int main(int argc, char **argv)
 {
-#ifndef Q_OS_WIN
-    // Prohibit using sudo or kdesu (but allow using the root user directly)
-    if (getuid() == 0) {
-        if (!qEnvironmentVariableIsEmpty("SUDO_USER")) {
-            std::cout << "Executing Kate with sudo is not possible due to unfixable security vulnerabilities." << std::endl;
-            return EXIT_FAILURE;
-        } else if (!qEnvironmentVariableIsEmpty("KDESU_USER")) {
-            std::cout << "Executing Kate with kdesu is not possible due to unfixable security vulnerabilities." << std::endl;
-            return EXIT_FAILURE;
-        }
-    }
-#endif
     /**
      * Create application first
      * Enforce application name even if the executable is renamed
