diff -urN libgnomedb-1.99.0.orig/control-center/Makefile.am libgnomedb-1.99.0/control-center/Makefile.am
--- libgnomedb-1.99.0.orig/control-center/Makefile.am	2006-10-25 11:35:43.000000000 +0100
+++ libgnomedb-1.99.0/control-center/Makefile.am	2006-10-25 11:35:53.000000000 +0100
@@ -1,6 +1,6 @@
 bin_PROGRAMS=gnome-database-properties
 
-INCLUDES = \
+AM_CPPFLAGS = \
         -I$(top_srcdir) \
         -I$(top_builddir) \
         $(LIBGNOMEDB_CFLAGS) \
diff -urN libgnomedb-1.99.0.orig/extra/Makefile.am libgnomedb-1.99.0/extra/Makefile.am
--- libgnomedb-1.99.0.orig/extra/Makefile.am	2006-10-25 11:35:43.000000000 +0100
+++ libgnomedb-1.99.0/extra/Makefile.am	2006-10-25 11:36:16.000000000 +0100
@@ -1,6 +1,6 @@
 SUBDIRS = demos
 
-INCLUDES = \
+AM_CPPFLAGS = \
         -I$(top_srcdir) \
         -I$(top_builddir) \
         $(LIBGNOMEDB_CFLAGS) \
diff -urN libgnomedb-1.99.0.orig/glade/Makefile.am libgnomedb-1.99.0/glade/Makefile.am
--- libgnomedb-1.99.0.orig/glade/Makefile.am	2006-10-25 11:35:43.000000000 +0100
+++ libgnomedb-1.99.0/glade/Makefile.am	2006-10-25 11:35:53.000000000 +0100
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-INCLUDES = \
+AM_CPPFLAGS = \
 	-I$(top_srcdir)					\
 	-I$(top_builddir)				\
 	-DG_LOG_DOMAIN=\"GnomeDB\"			\
diff -urN libgnomedb-1.99.0.orig/libgnomedb/data-entries/plugins/Makefile.am libgnomedb-1.99.0/libgnomedb/data-entries/plugins/Makefile.am
--- libgnomedb-1.99.0.orig/libgnomedb/data-entries/plugins/Makefile.am	2006-10-25 11:35:43.000000000 +0100
+++ libgnomedb-1.99.0/libgnomedb/data-entries/plugins/Makefile.am	2006-10-25 11:35:53.000000000 +0100
@@ -1,7 +1,7 @@
 plugin_LTLIBRARIES = libgnomedb_entry_builtin_plugins.la
 plugindir = $(libdir)/$(PACKAGE)/plugins
 
-INCLUDES = \
+AM_CPPFLAGS = \
         -I$(top_srcdir) -I$(srcdir)/.. \
         -I$(top_builddir) \
         $(LIBGNOMEDB_CFLAGS) \
diff -urN libgnomedb-1.99.0.orig/libgnomedb/graph/Makefile.am libgnomedb-1.99.0/libgnomedb/graph/Makefile.am
--- libgnomedb-1.99.0.orig/libgnomedb/graph/Makefile.am	2006-10-25 11:35:43.000000000 +0100
+++ libgnomedb-1.99.0/libgnomedb/graph/Makefile.am	2006-10-25 11:35:53.000000000 +0100
@@ -2,7 +2,7 @@
 
 SUBDIRS = 
 
-INCLUDES = \
+AM_CPPFLAGS = \
         -I$(top_srcdir) -I$(srcdir)/.. \
         -I$(top_builddir) \
         $(LIBGNOMEDB_CFLAGS) \
diff -urN libgnomedb-1.99.0.orig/libgnomedb/Makefile.am libgnomedb-1.99.0/libgnomedb/Makefile.am
--- libgnomedb-1.99.0.orig/libgnomedb/Makefile.am	2006-10-25 11:35:43.000000000 +0100
+++ libgnomedb-1.99.0/libgnomedb/Makefile.am	2006-10-25 11:35:53.000000000 +0100
@@ -2,7 +2,7 @@
 
 SUBDIRS = . data-entries graph 
 
-INCLUDES = \
+AM_CPPFLAGS = \
 	-I$(top_srcdir) \
 	-I$(top_builddir) \
 	$(LIBGNOMEDB_CFLAGS) \
diff -urN libgnomedb-1.99.0.orig/testing/Makefile.am libgnomedb-1.99.0/testing/Makefile.am
--- libgnomedb-1.99.0.orig/testing/Makefile.am	2006-10-25 11:35:43.000000000 +0100
+++ libgnomedb-1.99.0/testing/Makefile.am	2006-10-25 11:35:53.000000000 +0100
@@ -8,7 +8,7 @@
 endif
 
 noinst_PROGRAMS = $(test_progs)
-INCLUDES = \
+AM_CPPFLAGS = \
 	-I$(top_srcdir) \
 	-I$(top_builddir) \
 	$(LIBGNOMEDB_CFLAGS) \
