diff -Naur libgnomedb-1.9.103/control-center/Makefile.am libgnomedb-1.9.103-p/control-center/Makefile.am
--- libgnomedb-1.9.103/control-center/Makefile.am	2006-08-29 15:23:39.000000000 +0200
+++ libgnomedb-1.9.103-p/control-center/Makefile.am	2006-09-04 22:23:42.584333248 +0200
@@ -1,6 +1,6 @@
 bin_PROGRAMS=gnome-database-properties
 
-INCLUDES = \
+AM_CPPFLAGS = \
         -I$(top_srcdir) \
         -I$(top_builddir) \
         $(LIBGNOMEDB_CFLAGS) \
diff -Naur libgnomedb-1.9.103/extra/Makefile.am libgnomedb-1.9.103-p/extra/Makefile.am
--- libgnomedb-1.9.103/extra/Makefile.am	2006-08-29 15:23:43.000000000 +0200
+++ libgnomedb-1.9.103-p/extra/Makefile.am	2006-09-04 22:24:07.397561064 +0200
@@ -1,4 +1,4 @@
-INCLUDES = \
+AM_CPPFLAGS = \
         -I$(top_srcdir) \
         -I$(top_builddir) \
         $(LIBGNOMEDB_CFLAGS) \
diff -Naur libgnomedb-1.9.103/glade/Makefile.am libgnomedb-1.9.103-p/glade/Makefile.am
--- libgnomedb-1.9.103/glade/Makefile.am	2006-08-29 15:23:43.000000000 +0200
+++ libgnomedb-1.9.103-p/glade/Makefile.am	2006-09-04 22:26:21.584161600 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-INCLUDES = \
+AM_CPPFLAGS = \
 	-I$(top_srcdir)					\
 	-I$(top_builddir)				\
 	-DG_LOG_DOMAIN=\"GnomeDB\"			\
diff -Naur libgnomedb-1.9.103/libgnomedb/data-entries/plugins/Makefile.am libgnomedb-1.9.103-p/libgnomedb/data-entries/plugins/Makefile.am
--- libgnomedb-1.9.103/libgnomedb/data-entries/plugins/Makefile.am	2006-08-31 20:43:25.000000000 +0200
+++ libgnomedb-1.9.103-p/libgnomedb/data-entries/plugins/Makefile.am	2006-09-04 22:33:47.258408816 +0200
@@ -1,7 +1,7 @@
 plugin_LTLIBRARIES = libgnomedb_entry_builtin_plugins.la
 plugindir = $(libdir)/$(PACKAGE)/plugins
 
-INCLUDES = \
+AM_CPPFLAGS = \
         -I$(top_srcdir) -I$(srcdir)/.. \
         -I$(top_builddir) \
         $(LIBGNOMEDB_CFLAGS) \
diff -Naur libgnomedb-1.9.103/libgnomedb/graph/Makefile.am libgnomedb-1.9.103-p/libgnomedb/graph/Makefile.am
--- libgnomedb-1.9.103/libgnomedb/graph/Makefile.am	2006-08-29 15:23:43.000000000 +0200
+++ libgnomedb-1.9.103-p/libgnomedb/graph/Makefile.am	2006-09-04 22:34:10.985801704 +0200
@@ -2,7 +2,7 @@
 
 SUBDIRS = 
 
-INCLUDES = \
+AM_CPPFLAGS = \
         -I$(top_srcdir) -I$(srcdir)/.. \
         -I$(top_builddir) \
         $(LIBGNOMEDB_CFLAGS) \
diff -Naur libgnomedb-1.9.103/libgnomedb/Makefile.am libgnomedb-1.9.103-p/libgnomedb/Makefile.am
--- libgnomedb-1.9.103/libgnomedb/Makefile.am	2006-09-01 22:37:35.000000000 +0200
+++ libgnomedb-1.9.103-p/libgnomedb/Makefile.am	2006-09-04 22:32:43.137156728 +0200
@@ -2,7 +2,7 @@
 
 SUBDIRS = . data-entries graph 
 
-INCLUDES = \
+AM_CPPFLAGS = \
 	-I$(top_srcdir) \
 	-I$(top_builddir) \
 	$(LIBGNOMEDB_CFLAGS) \
diff -Naur libgnomedb-1.9.103/testing/Makefile.am libgnomedb-1.9.103-p/testing/Makefile.am
--- libgnomedb-1.9.103/testing/Makefile.am	2006-08-29 15:23:50.000000000 +0200
+++ libgnomedb-1.9.103-p/testing/Makefile.am	2006-09-04 22:34:38.858564400 +0200
@@ -8,7 +8,7 @@
 endif
 
 noinst_PROGRAMS = $(test_progs)
-INCLUDES = \
+AM_CPPFLAGS = \
 	-I$(top_srcdir) \
 	-I$(top_builddir) \
 	$(LIBGNOMEDB_CFLAGS) \
