diff -Naurp icecast-2.3.1/src/auth_url.c icecast-2.3.1-curl/src/auth_url.c
--- icecast-2.3.1/src/auth_url.c	2005-11-30 19:16:17.000000000 +0100
+++ icecast-2.3.1-curl/src/auth_url.c	2006-11-12 01:11:51.000000000 +0100
@@ -518,7 +518,6 @@ int auth_get_url_auth (auth_t *authentic
     curl_easy_setopt (url_info->handle, CURLOPT_WRITEDATA, url_info->handle);
     curl_easy_setopt (url_info->handle, CURLOPT_NOSIGNAL, 1L);
     curl_easy_setopt (url_info->handle, CURLOPT_TIMEOUT, 15L);
-    curl_easy_setopt (url_info->handle, CURLOPT_PASSWDFUNCTION, my_getpass);
     curl_easy_setopt (url_info->handle, CURLOPT_ERRORBUFFER, &url_info->errormsg[0]);
 
     if (url_info->username && url_info->password)
