From 63fe83808e76fedfbc60fe3edf6794f8bc9573b0 Mon Sep 17 00:00:00 2001
From: Kate <kit.ty.kate@disroot.org>
Date: Mon, 24 Oct 2022 15:56:45 +0100
Subject: [PATCH] Remove an unused dependency to bytes

---
 src/core/dune | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/core/dune b/src/core/dune
index 0564fee0c..9b890d185 100644
--- a/src/core/dune
+++ b/src/core/dune
@@ -13,7 +13,6 @@ let () = Jbuild_plugin.V1.send @@ {|
  (synopsis "Monadic promises and concurrent I/O")
  (wrapped false)
  |} ^ preprocess ^ {|
- (libraries bytes)
  (flags (:standard -w +A-29)))
 
 (documentation
