diff -up postler-0.1.1/data/postler.desktop.in.orig postler-0.1.1/data/postler.desktop.in
--- postler-0.1.1/data/postler.desktop.in.orig	2011-03-21 02:06:07.000000000 +0100
+++ postler-0.1.1/data/postler.desktop.in	2011-04-05 23:19:44.670813069 +0200
@@ -8,6 +8,6 @@ _Comment=Lean mail interface
 Categories=GTK;Network;Email;
 MimeType=application/mbox;x-scheme-handler/mailto;
 Exec=postler %u
-Icon=internet-mail
+Icon=postler
 Terminal=false
 StartupNotify=true
diff -up postler-0.1.1/postler/postler-accounts.vala.orig postler-0.1.1/postler/postler-accounts.vala
--- postler-0.1.1/postler/postler-accounts.vala.orig	2011-03-21 02:06:07.000000000 +0100
+++ postler-0.1.1/postler/postler-accounts.vala	2011-04-05 23:17:04.831668868 +0200
@@ -227,7 +227,7 @@ public class Postler.Accounts : GLib.Obj
             DirUtils.create_with_parents (autostart_path, 0700);
             FileUtils.set_contents (filename,
                 "[Desktop Entry]\nVersion=1.0\nType=Application\nName=Postler\n"
-              + "Exec=postler -m service\nIcon=internet-mail");
+              + "Exec=postler -m service\nIcon=postler");
         }
         else if (infos.length () == 0)
             FileUtils.unlink (filename);
diff -up postler-0.1.1/postler/postler-app.vala.orig postler-0.1.1/postler/postler-app.vala
--- postler-0.1.1/postler/postler-app.vala.orig	2011-03-21 02:06:07.000000000 +0100
+++ postler-0.1.1/postler/postler-app.vala	2011-04-05 23:18:23.598218605 +0200
@@ -25,7 +25,7 @@ namespace Postler {
     const string STOCK_FACE_WINK = "face-wink";
     const string STOCK_FOLDER_SAVED_SEARCH = "folder-saved-search";
     const string STOCK_IMAGE = "image-x-generic";
-    const string STOCK_INTERNET_MAIL = "internet-mail";
+    const string STOCK_INTERNET_MAIL = "postler";
     const string STOCK_MAIL_ATTACHMENT = "mail-attachment";
     const string STOCK_MAIL_FORWARD = "mail-forward";
     const string STOCK_MAIL_FORWARDED = "mail-forwarded";
diff -up postler-0.1.1/wscript.orig postler-0.1.1/wscript
--- postler-0.1.1/wscript.orig	2011-03-21 02:06:07.000000000 +0100
+++ postler-0.1.1/wscript	2011-04-05 23:23:26.674175886 +0200
@@ -318,7 +318,7 @@ def build (bld):
                 Utils.pprint ('BLUE', 'File ' + desktop + ' not generated')
         finally:
             pre.close ()
-    bld.install_files ('${MDATADIR}/icons/hicolor/scalable/apps',
+    bld.install_as ('${MDATADIR}/icons/hicolor/scalable/apps/postler.svg',
                        srcdir + '/data/internet-mail.svg')
 
     if Options.commands['check']:
