diff -Naur kmobiletools-0.5.0-beta3/kmobiletools/configure.in.bot kmobiletools-0.5.0-beta3-p/kmobiletools/configure.in.bot
--- kmobiletools-0.5.0-beta3/kmobiletools/configure.in.bot	2007-06-05 16:18:07.000000000 +0200
+++ kmobiletools-0.5.0-beta3-p/kmobiletools/configure.in.bot	2007-09-23 16:18:24.000000000 +0200
@@ -10,10 +10,10 @@
 
 # kdebluetooth
 KDEBLUETOOTH_VER=$DISABLED
-if test x$have_kbluetooth = xyes; then
+if test "x$have_kbluetooth" = "xyes"; then
 	KDEBLUETOOTH_VER="1.0-beta2"
 fi
-if test x$have_new_kbluetooth = xyes; then
+if test "x$have_new_kbluetooth" = "xyes"; then
 	KDEBLUETOOTH_VER="with dbus integration"
 fi
 echo -e "KDEBluetooth Integration\t\t$KDEBLUETOOTH_VER"
@@ -44,7 +44,10 @@
 fi
 
 echo -e "\n*************************************************\n"
-if test x$have_gammu != xyes; then
+if test "x$have_gammu" != "xyes"; then
     echo -e "Please note that you need EXACTLY gammu (or libgammu) 1.10.6.\nIf your distribution has another version,\nread http://www.kmobiletools.org/gammu for more details."
 fi
 
+if test "x$have_new_kbluetooth" = "xyes" && test "x$have_obexftp" != "xyes"; then
+    echo -e "ObexFTP Kioslave support was disabled since it seems you have installed KDEBluetooth 1.0-beta3 or upper version, which does provide exactly the same obexftp kioslave included in kmobiletools."
+fi
diff -Naur kmobiletools-0.5.0-beta3/kmobiletools/configure.in.in kmobiletools-0.5.0-beta3-p/kmobiletools/configure.in.in
--- kmobiletools-0.5.0-beta3/kmobiletools/configure.in.in	2007-06-05 16:18:07.000000000 +0200
+++ kmobiletools-0.5.0-beta3-p/kmobiletools/configure.in.in	2007-09-23 16:18:24.000000000 +0200
@@ -8,11 +8,11 @@
 # BEGIN KONTACT CHECK
 ###############################################################################
 
-AC_ARG_ENABLE(kontact, AC_HELP_STRING([--disable-kontact-plugin], [disable KMobileTools Kontact plugin (default: check)]),
+AC_ARG_ENABLE(kontact, AC_HELP_STRING([--disable-kontact], [disable KMobileTools Kontact plugin (default: check)]),
 	[enable_kontact=$enableval], [enable_kontact=yes])
-if test "$enable_kontact" = "yes"; then
+if test "x$enable_kontact" = "xyes"; then
 	KDE_CHECK_HEADER(kontact/plugin.h, [have_kontact="yes"], [have_kontact="no"])
-	if test "$have_kontact" = "yes"; then
+	if test "x$have_kontact" = "xyes"; then
 		AC_DEFINE(HAVE_KONTACT, 1, [have KONTACT])
 	fi
 fi
@@ -22,9 +22,9 @@
 AC_ARG_ENABLE(libkcal, AC_HELP_STRING([--disable-libkcal], [disable KMobileTools KCalendar support (default: check)]),
 	[enable_kcal=$enableval], [enable_kcal=yes])
 
-if test "$enable_kcal" = "yes"; then
+if test "x$enable_kcal" = "xyes"; then
 	KDE_CHECK_HEADER(libkcal/event.h, [have_kcal="yes"], [have_kcal="no"])
-	if test "$have_kcal" = "yes"; then
+	if test "x$have_kcal" = "xyes"; then
 		AC_DEFINE(HAVE_KCAL, 1, [have KCAL])
 		LIB_KCAL="-lkcal"
 	else
@@ -38,14 +38,14 @@
 AC_CHECK_PROG(PKGCONFIGFOUND, pkg-config,[yes])
 
 
-if test x$PKGCONFIGFOUND != xyes; then
+if test "x$PKGCONFIGFOUND" != "xyes"; then
 	AC_ERROR(pkgconfig not found.)
 fi
 	
 
 AC_ARG_WITH(gammu, AC_HELP_STRING([--without-gammu], [Build kmobiletools without gammu engine (default: check)]),
 	[with_gammu=$withval], [with_gammu=yes])
-if test "$with_gammu" = "yes" ; then # gammu forced to disabled till it'll be completed
+if test "x$with_gammu" = "xyes" ; then # gammu forced to disabled till it'll be completed
 	PKG_CHECK_MODULES(GAMMU, gammu >= 1.10.6, have_gammu=yes, have_gammu=no)
 	if test "x$have_gammu" = "xyes" ;then
 		AC_DEFINE(WITH_GAMMU, 1, [with GAMMU])
@@ -72,59 +72,16 @@
 #AC_SUBST(LIB_GAMMU)
 #AM_CONDITIONAL(with_gammu, test $have_gammu = yes)
 
-AC_ARG_ENABLE(kioslaves, AC_HELP_STRING([--disable-kioslaves], [enable KIOSlaves for filesystem access (default: disable)]),
-    [enable_kioslaves=$enableval], [enable_kioslaves=yes])
-
-if test "x$enable_kioslaves" = "xyes"; then
-	KDE_CHECK_HEADER(p2kmoto.h, have_p2k=yes, have_p2k=no)
-	AM_CONDITIONAL(with_p2kmoto, test $have_p2k = yes)
-	KDE_CHECK_HEADER(obexftp/client.h, have_obexftp=yes, have_obexftp=no)
-	AM_CONDITIONAL(compile_obex, test $have_obexftp = yes)
-else
-	AM_CONDITIONAL(with_p2kmoto, false)
-	AM_CONDITIONAL(compile_obex, false)
-fi
-
-if test "x$have_p2k" = "xyes"; then
-	AC_DEFINE(HAVE_P2KLIB, 1, [p2k filesystem support])
-	AC_SUBST(HAVE_P2KLIB)
-fi
-
-if test "x$have_obexftp" = "xyes"; then
-	AC_DEFINE(HAVE_OBEXFTP, 1, [ObexFTP filesystem support])
-	AC_SUBST(HAVE_OBEXFTP)
-fi
-
-if test "x$have_p2k" = "xyes" || test "x$have_obexftp" = "xyes"; then
-    ENABLE_KIOSLAVES="yes";
-    else ENABLE_KIOSLAVES="no";
-fi
-
-AM_CONDITIONAL(compile_kioslaves, test "x$ENABLE_KIOSLAVES" = "xyes" )
-if test "x$ENABLE_KIOSLAVES" = "xyes"; then
-    AC_DEFINE(ENABLE_FS, 1, [filesystem support])
-    AC_SUBST(ENABLE_FS)
-fi
-
 AC_ARG_ENABLE(crashhandler, AC_HELP_STRING([--disable-crashhandler], [disable KMobileTools Crash Handler(default: enable)]),
     [enable_crashhandler=$enableval], [enable_crashhandler=yes])
-if test "$enable_crashhandler" = "yes"; then
+if test "x$enable_crashhandler" = "xyes"; then
   AC_DEFINE(USE_CRASHHANDLER, 1, [use CrashHandler])
   AC_SUBST(USE_CRASHHANDLER)
 fi
 
 AC_ARG_ENABLE(tests, AC_HELP_STRING([--enable-tests], [enable test applications (default: disable)]),
     [enable_tests=$enableval], [enable_tests=no])
-AM_CONDITIONAL(compile_tests, test x$enable_tests = xyes)
-
-KDE_CHECK_HEADER(kdebluetooth/rfcommsocketdevice.h, have_kbluetooth=yes, have_kbluetooth=no)
-AM_CONDITIONAL(with_kdebluetooth, test $have_kbluetooth = yes)
-if test "$have_kbluetooth" = "yes"; then
-	AC_DEFINE(KBLUETOOTH, 1, [use kdebluetooth rfcomm socket])
-	AC_SUBST(KBLUETOOTH)
-	LIB_KBLUETOOTH="-lkbluetooth"
-fi
-AC_SUBST(LIB_KBLUETOOTH)
+AM_CONDITIONAL(compile_tests, test "x$enable_tests" = "xyes")
 
 #KDEBluetooth-DBUS module
 PKG_CHECK_MODULES(DBUS, "dbus-1", have_dbus=yes, have_dbus=no)
@@ -135,9 +92,24 @@
 CPPFLAGS="$CPPFLAGS $DBUS_CFLAGS $all_includes -DDBUS_API_SUBJECT_TO_CHANGE"
 
 
-KDE_CHECK_HEADER(libkbluetooth/adapter.h, have_new_kbluetooth=yes, have_new_kbluetooth=no)
-AM_CONDITIONAL(with_kdebluetooth_new, test x$have_new_kbluetooth = xyes)
-if test "$have_new_kbluetooth" = "yes"; then
+AC_ARG_ENABLE(kdebluetooth, AC_HELP_STRING([--disable-kdebluetooth], [disable kdebluetooth integration (default: check)]),
+    [enable_kdebluetooth=$enableval], [enable_kdebluetooth=yes])
+if test "x$enable_kdebluetooth" = "xyes"; then
+	KDE_CHECK_HEADER(kdebluetooth/rfcommsocketdevice.h, have_kbluetooth=yes, have_kbluetooth=no)
+	KDE_CHECK_HEADER(libkbluetooth/adapter.h, have_new_kbluetooth=yes, have_new_kbluetooth=no)
+fi
+
+AM_CONDITIONAL(with_kdebluetooth, test "x$have_kbluetooth" = "xyes")
+if test "x$have_kbluetooth" = "xyes"; then
+	AC_DEFINE(KBLUETOOTH, 1, [use kdebluetooth rfcomm socket])
+	AC_SUBST(KBLUETOOTH)
+	LIB_KBLUETOOTH="-lkbluetooth"
+fi
+AC_SUBST(LIB_KBLUETOOTH)
+
+
+AM_CONDITIONAL(with_kdebluetooth_new, test "x$have_new_kbluetooth" = "xyes")
+if test "x$have_new_kbluetooth" = "xyes"; then
        PKG_CHECK_MODULES(BLUEZ, "bluez")
        AC_DEFINE(KBLUETOOTH_NEW, 1, [use new kdebluetooth])
        AC_SUBST(KBLUETOOTH_NEW)
@@ -146,6 +118,45 @@
 AC_SUBST(LIB_KBLUETOOTH)
 
 
+AC_ARG_ENABLE(kioslaves, AC_HELP_STRING([--disable-kioslaves], [disable all KIOSlaves for filesystem access (default: enable)]),
+    [enable_kioslaves=$enableval], [enable_kioslaves=yes])
+
+AC_ARG_ENABLE(p2kmoto, AC_HELP_STRING([--disable-p2kmoto], [disable p2kmoto kioslave (default: check)]),
+    [enable_p2kmoto=$enableval], [enable_p2kmoto=yes])
+#if test "x$enable_p2kmoto" = "xyes" && test "x$enable_kioslaves" = "xyes"; then
+#	KDE_CHECK_HEADER(p2kmoto.h, have_p2k=yes, have_p2k=no) 
+#fi
+AM_CONDITIONAL(with_p2kmoto, test "x$enable_p2kmoto" = "xyes")
+
+AC_ARG_ENABLE(obexftp, AC_HELP_STRING([--disable-obexftp], [disable obex kioslave (default: check)]),
+    [enable_obexftp=$enableval], [enable_obexftp=yes])
+if test "x$enable_obexftp" = "xyes" && test "x$enable_kioslaves" = "xyes" && test "x$have_new_kbluetooth" != "xyes"; then
+	KDE_CHECK_HEADER(obexftp/client.h, have_obexftp=yes, have_obexftp=no) 
+fi
+AM_CONDITIONAL(compile_obex, test "x$have_obexftp" = "xyes")
+
+if test "x$have_p2k" = "xyes"; then
+	AC_DEFINE(HAVE_P2KLIB, 1, [p2k filesystem support])
+	AC_SUBST(HAVE_P2KLIB)
+fi
+
+if test "x$have_obexftp" = "xyes"; then
+	AC_DEFINE(HAVE_OBEXFTP, 1, [ObexFTP filesystem support])
+	AC_SUBST(HAVE_OBEXFTP)
+fi
+
+if test "x$have_p2k" = "xyes" || test "x$have_obexftp" = "xyes"; then
+    ENABLE_KIOSLAVES="yes";
+    else ENABLE_KIOSLAVES="no";
+fi
+
+AM_CONDITIONAL(compile_kioslaves, test "x$ENABLE_KIOSLAVES" = "xyes" )
+if test "x$ENABLE_KIOSLAVES" = "xyes"; then
+    AC_DEFINE(ENABLE_FS, 1, [filesystem support])
+    AC_SUBST(ENABLE_FS)
+fi
+
+
 SVNREV=`svn st -v $srcdir/kmobiletools | grep -v '?' | awk '{ print $1 }' | sort -rn | head -n 1`
 AC_DEFINE_UNQUOTED( SVNREVISION, "$SVNREV", [Subversion revision] )
 
