diff -Naur kmobiletools/kmobiletools/engines/gammu_engine/device.cpp kmobiletools.new/kmobiletools/engines/gammu_engine/device.cpp
--- kmobiletools/kmobiletools/engines/gammu_engine/device.cpp	2013-03-01 15:10:17.000000000 +0100
+++ kmobiletools.new/kmobiletools/engines/gammu_engine/device.cpp	2013-03-01 15:18:16.949688907 +0100
@@ -736,18 +736,10 @@
 //                 // TODO: new enumeration value
 // 		break;
 		
-            case PBK_Number_Work:
-                addressee.insertPhoneNumber( KABC::PhoneNumber( QString::fromUtf8((char*)DecodeUnicodeString( entry->Text )), KABC::PhoneNumber::Work ) );
-                break;
-
             case PBK_Number_Fax:
                 addressee.insertPhoneNumber( KABC::PhoneNumber( QString::fromUtf8((char*)DecodeUnicodeString( entry->Text )), KABC::PhoneNumber::Fax ) );
                 break;
 
-            case PBK_Number_Home:
-                addressee.insertPhoneNumber( KABC::PhoneNumber( QString::fromUtf8((char*)DecodeUnicodeString( entry->Text )), KABC::PhoneNumber::Home ) );
-                break;
-		
 	    case PBK_Number_Messaging:
                 // TODO: new enumeration value
 		break;
@@ -830,30 +822,6 @@
                 address.setCountry( QString::fromUtf8((char*)DecodeUnicodeString( entry->Text )) );
                 break;
 		
-	    case PBK_Text_WorkStreetAddress:
-                // TODO: new enumeration value
-		break;
-
-	    case PBK_Text_WorkCity:
-                // TODO: new enumeration value
-		break;
-
-	    case PBK_Text_WorkState:
-                // TODO: new enumeration value
-		break;
-
-	    case PBK_Text_WorkZip:
-                // TODO: new enumeration value
-		break;
-
-	    case PBK_Text_WorkCountry:
-                // TODO: new enumeration value
-		break;
-
-	    case PBK_Text_WorkPostal:
-                // TODO: new enumeration value
-		break;
-
             case PBK_Text_UserID:
                 addressee.setUid( QString::fromUtf8((char*)DecodeUnicodeString( entry->Text )) );
                 break;
@@ -976,21 +944,21 @@
 
                 case KABC::PhoneNumber::Work:
                     kDebug() <<"[ !!! ] Adding KABC::PhoneNumber::Work";
-                    memoryEntry.Entries[ memoryEntry.EntriesNum ].EntryType = PBK_Number_Work;
+                    memoryEntry.Entries[ memoryEntry.EntriesNum ].EntryType = PBK_Number_General;
                     EncodeUnicode( memoryEntry.Entries[ memoryEntry.EntriesNum ].Text, (*it).number().toUtf8(), (*it).number().toUtf8().length() );
                     memoryEntry.EntriesNum++;
                 break;
 
                 case KABC::PhoneNumber::Home:
                     kDebug() <<"[ !!! ] Adding KABC::PhoneNumber::Home";
-                    memoryEntry.Entries[ memoryEntry.EntriesNum ].EntryType = PBK_Number_Home;
+                    memoryEntry.Entries[ memoryEntry.EntriesNum ].EntryType = PBK_Number_General;
                     EncodeUnicode( memoryEntry.Entries[ memoryEntry.EntriesNum ].Text, (*it).number().toUtf8(), (*it).number().toUtf8().length() );
                     memoryEntry.EntriesNum++;
                 break;
 
                 case KABC::PhoneNumber::Pref:
                     kDebug() <<"[ !!! ] Adding KABC::PhoneNumber::Pref";
-                    memoryEntry.Entries[ memoryEntry.EntriesNum ].EntryType = PBK_Number_Work;
+                    memoryEntry.Entries[ memoryEntry.EntriesNum ].EntryType = PBK_Number_General;
                     EncodeUnicode( memoryEntry.Entries[ memoryEntry.EntriesNum ].Text, (*it).number().toUtf8(), (*it).number().toUtf8().length() );
                     memoryEntry.EntriesNum++;
                 break;
