diff -Naur lilypond-2.8.1.orig/mf/GNUmakefile lilypond-2.8.1/mf/GNUmakefile
--- lilypond-2.8.1.orig/mf/GNUmakefile	2006-03-29 18:16:42.000000000 +0200
+++ lilypond-2.8.1/mf/GNUmakefile	2006-04-22 22:02:58.000000000 +0200
@@ -138,7 +138,7 @@
 	${MAKE} -C $(top-build-dir) link-mf-tree
 
 pfa_warning:
-ifneq ($(shell $(MFTRACE) --version | sed 's/ .*//'),mftrace)
+ifneq ($(shell $(MFTRACE) --version | sed 's/ .*//;q'),mftrace)
 	@echo ""
 	@echo "ERROR: mftrace not found"
 	@echo ""
