From 82e5901fb68acfead807b8e824310edda2437744 Mon Sep 17 00:00:00 2001
From: Julien Lavergne <julien.lavergne@gmail.com>
Date: Sun, 24 Jul 2011 01:14:03 +0200
Subject: [PATCH] Fix crash on start

---
 lxsession/xdg-autostart.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/lxsession/xdg-autostart.c b/lxsession/xdg-autostart.c
index 39f0d2a..b0f922a 100644
--- a/lxsession/xdg-autostart.c
+++ b/lxsession/xdg-autostart.c
@@ -283,7 +283,6 @@ static void get_autostart_files_in_dir( GHashTable* hash, const char* de_name, c
                 path = g_build_filename( dir_path, name, NULL );
                 g_hash_table_replace( hash, g_strdup(name), path );
             }
-            g_free( name );
         }
         g_dir_close( dir );
     }
-- 
1.7.0.1


