From 6c9cadd104862a6c9f7398993d5825673d4d02f7 Mon Sep 17 00:00:00 2001
From: Melko <melko@frugalware.org>
Date: Mon, 23 Sep 2013 13:21:20 +0200
Subject: [PATCH] * set wallpaper for each activity

---
 kde-frugalware/config/99-fwWallpaper.js | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/config/99-fwWallpaper.js b/kde-frugalware/config/99-fwWallpaper.js
index e305e5a..76ddf59 100644
--- a/config/99-fwWallpaper.js
+++ b/config/99-fwWallpaper.js
@@ -1,6 +1,9 @@
-// Set Frugalware wallpaper for a fresh user
+/* Copyright 2013 Frugalware Developer Team */
 
-var activity = activities()[0]
-activity.currentConfigGroup = new Array("Wallpaper", "image");
-activity.writeConfig("wallpaper", "/usr/share/wallpapers/Frugalware/");
-activity.writeConfig("userswallpaper", "/usr/share/wallpapers/Frugalware");
+// Set Frugalware wallpaper for a fresh user
+for (var i = 0; i < activities().length; i++) {
+    var activity = activities()[i]
+    activity.currentConfigGroup = new Array("Wallpaper", "image")
+    activity.writeConfig("wallpaper", "/usr/share/wallpapers/Frugalware/")
+    activity.writeConfig("userswallpaper", "/usr/share/wallpapers/Frugalware")
+}
-- 
1.8.4

