diff -Naur kde-frugalware-0.6.0/etc/xdg/menus/applications-merged/kde-not-show.menu kde-frugalware-0.6.0-p/etc/xdg/menus/applications-merged/kde-not-show.menu
--- kde-frugalware-0.6.0/etc/xdg/menus/applications-merged/kde-not-show.menu	2007-02-25 04:59:49.000000000 +0100
+++ kde-frugalware-0.6.0-p/etc/xdg/menus/applications-merged/kde-not-show.menu	2007-03-27 14:41:42.000000000 +0200
@@ -4,6 +4,7 @@
   <Filename>kde-Kfind.desktop</Filename>
   <Filename>kde-Help.desktop</Filename>
   <Filename>kde-Home.desktop</Filename>
+ </Exclude>
  <Menu>
   <Name>.hidden</Name>
   <Include>
