From e521e8240c1acc49a514db123cb67beb669dea81 Mon Sep 17 00:00:00 2001
From: Dave Airlie <airlied@redhat.com>
Date: Tue, 28 Jul 2009 03:32:28 +0000
Subject: ark: update for resources/RAC API removal

---
diff --git a/src/ark_driver.c b/src/ark_driver.c
index db5303c..ee94a14 100644
--- a/src/ark_driver.c
+++ b/src/ark_driver.c
@@ -35,7 +35,6 @@
 #include "xf86_OSproc.h"
 #include "xf86Pci.h"
 #include "xf86PciInfo.h"
-#include "xf86Resources.h"
 #include "xf86fbman.h"
 #include "xf86cmap.h"
 #include "compiler.h"
@@ -313,16 +312,7 @@ static Bool ARKPreInit(ScrnInfoPtr pScrn, int flags)
 	}
 
 	pEnt = xf86GetEntityInfo(pScrn->entityList[0]);
-	if (pEnt->resources) {
-		xfree(pEnt);
-		ARKFreeRec(pScrn);
-		return FALSE;
-	}
-
 	pARK->PciInfo = xf86GetPciInfoForEntity(pEnt->index);
-	xf86RegisterResources(pEnt->index, NULL, ResNone);
-	xf86SetOperatingState(resVgaIo, pEnt->index, ResUnusedOpr);
-	xf86SetOperatingState(resVgaMem, pEnt->index, ResDisableOpr);
 
 	if (pEnt->device->chipset && *pEnt->device->chipset) {
 		pScrn->chipset = pEnt->device->chipset;
--
cgit v0.8.2

