From ac854832326fe8e8894c24c6f564051eb12565ac Mon Sep 17 00:00:00 2001
From: Julien Schueller <schueller@phimeca.com>
Date: Tue, 21 Nov 2023 11:26:24 +0100
Subject: [PATCH] XMLToolbox: Add missing include

---
 lib/src/Base/Common/XMLToolbox.cxx | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/src/Base/Common/XMLToolbox.cxx b/lib/src/Base/Common/XMLToolbox.cxx
index bd51436df5..4462173ae8 100644
--- a/lib/src/Base/Common/XMLToolbox.cxx
+++ b/lib/src/Base/Common/XMLToolbox.cxx
@@ -30,6 +30,7 @@
 #include <cstring>
 
 #if defined OPENTURNS_HAVE_LIBXML2
+#include <libxml/parser.h>
 #include <libxml/tree.h>
 #include <libxml/xmlsave.h>
 #endif
