diff -Naur vnc_unixsrc.orig/Xvnc/config/cf/Imake.cf vnc_unixsrc/Xvnc/config/cf/Imake.cf
--- vnc_unixsrc.orig/Xvnc/config/cf/Imake.cf	2005-09-25 23:57:59.000000000 +0200
+++ vnc_unixsrc/Xvnc/config/cf/Imake.cf	2005-09-26 00:00:49.000000000 +0200
@@ -801,6 +801,10 @@
 # ifdef PpcArchitecture
 #  define PowerPCArchitecture
 # endif
+# ifdef __x86_64__
+#  define x86_64Architecture
+#  undef __x86_64__
+# endif
 #endif /* linux */
 
 #if (defined(__Lynx__) || defined(Lynx)) && (defined(i386) || defined(__i386__) || defined(__x86__) || defined(__powerpc__) || defined(__sparc__) || defined(sparc))
diff -Naur vnc_unixsrc.orig/Xvnc/config/cf/linux.cf vnc_unixsrc/Xvnc/config/cf/linux.cf
--- vnc_unixsrc.orig/Xvnc/config/cf/linux.cf	2005-09-25 23:57:59.000000000 +0200
+++ vnc_unixsrc/Xvnc/config/cf/linux.cf	2005-09-26 00:01:37.000000000 +0200
@@ -43,7 +43,7 @@
 #if defined(AlphaArchitecture) || defined(PowerPCArchitecture) || \
     defined(SparcArchitecture) || defined(ArmArchitecure) || \
     defined(i386Architecture) || defined(s390Architecture) || \
-    defined(s390xArchitecture)
+    defined(s390xArchitecture) || defined(x86_64Architecture)
 #define DefaultLinuxCLibMajorVersion	6
 #else
 #define DefaultLinuxCLibMajorVersion	5
@@ -290,6 +290,14 @@
 #define ServerExtraDefines	-DGCCUSESGAS XFree86ServerDefines
 #endif /* PowerPCArchitecture */
 
+#ifdef x86_64Architecture
+#define DefaultCCOptions        -fsigned-char
+#define OptimizedCDebugFlags   -O2
+#define LinuxMachineDefines    -D__x86_64__
+#define ServerOSDefines                XFree86ServerOSDefines -DDDXTIME -DPART_NET
+#define ServerExtraDefines     -DGCCUSESGAS XFree86ServerDefines
+#endif /* x86_64PCArchitecture */
+
 #ifdef ArmArchitecture
 #define DefaultCCOptions	-fsigned-char
 #define OptimizedCDebugFlags	-O2
