diff -Naur klibido-0.2.5/src/nntpthreadsocket.h klibido-0.2.5-p/src/nntpthreadsocket.h
--- klibido-0.2.5/src/nntpthreadsocket.h	2006-01-06 23:05:32.000000000 +0100
+++ klibido-0.2.5-p/src/nntpthreadsocket.h	2006-07-11 23:03:29.839263765 +0200
@@ -233,8 +233,8 @@
 		bool getBody(QString group, int artnum); //Unimplemented
 		bool getHead(QString group, int artnum); //Unimplemented
 		int m_getError() {return error;}
-    	const char *m_getErrorDesc() {return (const char *) errorString;}
-		char * NntpThreadSocket::m_findEndLine( char * start, char * end );
+    		const char *m_getErrorDesc() {return (const char *) errorString;}
+		char * m_findEndLine( char * start, char * end );
 		void setHost(NntpHost *nh);
 		bool m_sendCmd( QString& cmd, int response );
 		QTime prevTime, currentTime;
diff -Naur klibido-0.2.5/src/yydecoder.h klibido-0.2.5-p/src/yydecoder.h
--- klibido-0.2.5/src/yydecoder.h	2005-03-09 12:18:55.000000000 +0100
+++ klibido-0.2.5-p/src/yydecoder.h	2006-07-11 23:04:43.821016830 +0200
@@ -87,7 +87,7 @@
 	
 private:
 	
-	inline void yyDecoder::charCRC(const unsigned char *c);
+	inline void charCRC(const unsigned char *c);
 	
 	QStringList m_fileParts;
 	QStringList::Iterator fileIterator;
