diff -Naur bcc/tools/argdist.py bcc.new/tools/argdist.py
--- bcc/tools/argdist.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/argdist.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # argdist   Trace a function and display a distribution of its
 #           parameter values as a histogram or frequency count.
diff -Naur bcc/tools/bashreadline.py bcc.new/tools/bashreadline.py
--- bcc/tools/bashreadline.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/bashreadline.py	2021-02-12 14:44:02.576660867 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # bashreadline  Print entered bash commands from all running shells.
 #               For Linux, uses BCC, eBPF. Embedded C.
diff -Naur bcc/tools/bindsnoop.py bcc.new/tools/bindsnoop.py
--- bcc/tools/bindsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/bindsnoop.py	2021-02-12 14:44:02.576660867 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # bindsnoop       Trace IPv4 and IPv6 binds()s.
 #               For Linux, uses BCC, eBPF. Embedded C.
diff -Naur bcc/tools/biolatency.py bcc.new/tools/biolatency.py
--- bcc/tools/biolatency.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/biolatency.py	2021-02-12 14:44:02.576660867 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # biolatency    Summarize block device I/O latency as a histogram.
diff -Naur bcc/tools/biolatpcts.py bcc.new/tools/biolatpcts.py
--- bcc/tools/biolatpcts.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/biolatpcts.py	2021-02-12 14:44:02.576660867 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # biolatpcts.py  Monitor IO latency distribution of a block device.
 #
diff -Naur bcc/tools/biosnoop.py bcc.new/tools/biosnoop.py
--- bcc/tools/biosnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/biosnoop.py	2021-02-12 14:44:02.576660867 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # biosnoop  Trace block device I/O and print details including issuing PID.
diff -Naur bcc/tools/biotop.py bcc.new/tools/biotop.py
--- bcc/tools/biotop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/biotop.py	2021-02-12 14:44:02.576660867 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # biotop  block device (disk) I/O by process.
diff -Naur bcc/tools/bitesize.py bcc.new/tools/bitesize.py
--- bcc/tools/bitesize.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/bitesize.py	2021-02-12 14:44:02.576660867 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # bitehist.py   Block I/O size histogram.
 #               For Linux, uses BCC, eBPF. See .c file.
diff -Naur bcc/tools/bpflist.py bcc.new/tools/bpflist.py
--- bcc/tools/bpflist.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/bpflist.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # bpflist   Display processes currently using BPF programs and maps,
 #           pinned BPF programs and maps, and enabled probes.
diff -Naur bcc/tools/btrfsdist.py bcc.new/tools/btrfsdist.py
--- bcc/tools/btrfsdist.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/btrfsdist.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # btrfsdist  Summarize btrfs operation latency.
diff -Naur bcc/tools/btrfsslower.py bcc.new/tools/btrfsslower.py
--- bcc/tools/btrfsslower.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/btrfsslower.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # btrfsslower  Trace slow btrfs operations.
diff -Naur bcc/tools/cachestat.py bcc.new/tools/cachestat.py
--- bcc/tools/cachestat.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/cachestat.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # cachestat     Count cache kernel function calls.
 #               For Linux, uses BCC, eBPF. See .c file.
diff -Naur bcc/tools/cachetop.py bcc.new/tools/cachetop.py
--- bcc/tools/cachetop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/cachetop.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # cachetop      Count cache kernel function calls per processes
diff -Naur bcc/tools/capable.py bcc.new/tools/capable.py
--- bcc/tools/capable.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/capable.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # capable   Trace security capabilitiy checks (cap_capable()).
diff -Naur bcc/tools/compactsnoop.py bcc.new/tools/compactsnoop.py
--- bcc/tools/compactsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/compactsnoop.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # compactsnoop  Trace compact zone and print details including issuing PID.
diff -Naur bcc/tools/cpudist.py bcc.new/tools/cpudist.py
--- bcc/tools/cpudist.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/cpudist.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # cpudist   Summarize on- and off-CPU time per task as a histogram.
diff -Naur bcc/tools/cpuunclaimed.py bcc.new/tools/cpuunclaimed.py
--- bcc/tools/cpuunclaimed.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/cpuunclaimed.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # cpuunclaimed   Sample CPU run queues and calculate unclaimed idle CPU.
diff -Naur bcc/tools/criticalstat.py bcc.new/tools/criticalstat.py
--- bcc/tools/criticalstat.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/criticalstat.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # criticalstat  Trace long critical sections (IRQs or preemption disabled)
diff -Naur bcc/tools/dbslower.py bcc.new/tools/dbslower.py
--- bcc/tools/dbslower.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/dbslower.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # dbslower      Trace MySQL and PostgreSQL queries slower than a threshold.
 #
diff -Naur bcc/tools/dbstat.py bcc.new/tools/dbstat.py
--- bcc/tools/dbstat.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/dbstat.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # dbstat        Display a histogram of MySQL and PostgreSQL query latencies.
 #
diff -Naur bcc/tools/dcsnoop.py bcc.new/tools/dcsnoop.py
--- bcc/tools/dcsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/dcsnoop.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # dcsnoop   Trace directory entry cache (dcache) lookups.
diff -Naur bcc/tools/dcstat.py bcc.new/tools/dcstat.py
--- bcc/tools/dcstat.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/dcstat.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # dcstat   Directory entry cache (dcache) stats.
diff -Naur bcc/tools/deadlock.py bcc.new/tools/deadlock.py
--- bcc/tools/deadlock.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/deadlock.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # deadlock  Detects potential deadlocks (lock order inversions)
 #           on a running process. For Linux, uses BCC, eBPF.
diff -Naur bcc/tools/dirtop.py bcc.new/tools/dirtop.py
--- bcc/tools/dirtop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/dirtop.py	2021-02-12 14:44:02.577660871 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # dirtop  file reads and writes by directory.
diff -Naur bcc/tools/drsnoop.py bcc.new/tools/drsnoop.py
--- bcc/tools/drsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/drsnoop.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # drsnoop  Trace direct reclaim and print details including issuing PID.
diff -Naur bcc/tools/execsnoop.py bcc.new/tools/execsnoop.py
--- bcc/tools/execsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/execsnoop.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # execsnoop Trace new processes via exec() syscalls.
diff -Naur bcc/tools/exitsnoop.py bcc.new/tools/exitsnoop.py
--- bcc/tools/exitsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/exitsnoop.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 from __future__ import print_function
 
diff -Naur bcc/tools/ext4dist.py bcc.new/tools/ext4dist.py
--- bcc/tools/ext4dist.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/ext4dist.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # ext4dist  Summarize ext4 operation latency.
diff -Naur bcc/tools/ext4slower.py bcc.new/tools/ext4slower.py
--- bcc/tools/ext4slower.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/ext4slower.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # ext4slower  Trace slow ext4 operations.
diff -Naur bcc/tools/filelife.py bcc.new/tools/filelife.py
--- bcc/tools/filelife.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/filelife.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # filelife    Trace the lifespan of short-lived files.
diff -Naur bcc/tools/fileslower.py bcc.new/tools/fileslower.py
--- bcc/tools/fileslower.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/fileslower.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # fileslower  Trace slow synchronous file reads and writes.
diff -Naur bcc/tools/filetop.py bcc.new/tools/filetop.py
--- bcc/tools/filetop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/filetop.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # filetop  file reads and writes by process.
diff -Naur bcc/tools/funccount.py bcc.new/tools/funccount.py
--- bcc/tools/funccount.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/funccount.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # funccount Count functions, tracepoints, and USDT probes.
diff -Naur bcc/tools/funcinterval.py bcc.new/tools/funcinterval.py
--- bcc/tools/funcinterval.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/funcinterval.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # funcinterval   Time interval between the same function, tracepoint
diff -Naur bcc/tools/funclatency.py bcc.new/tools/funclatency.py
--- bcc/tools/funclatency.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/funclatency.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # funclatency   Time functions and print latency as a histogram.
diff -Naur bcc/tools/funcslower.py bcc.new/tools/funcslower.py
--- bcc/tools/funcslower.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/funcslower.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # funcslower  Trace slow kernel or user function calls.
diff -Naur bcc/tools/gethostlatency.py bcc.new/tools/gethostlatency.py
--- bcc/tools/gethostlatency.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/gethostlatency.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # gethostlatency  Show latency for getaddrinfo/gethostbyname[2] calls.
 #                 For Linux, uses BCC, eBPF. Embedded C.
diff -Naur bcc/tools/hardirqs.py bcc.new/tools/hardirqs.py
--- bcc/tools/hardirqs.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/hardirqs.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # hardirqs  Summarize hard IRQ (interrupt) event time.
diff -Naur bcc/tools/inject.py bcc.new/tools/inject.py
--- bcc/tools/inject.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/inject.py	2021-02-12 14:44:02.578660875 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # This script generates a BPF program with structure inspired by trace.py. The
 # generated program operates on PID-indexed stacks. Generally speaking,
diff -Naur bcc/tools/killsnoop.py bcc.new/tools/killsnoop.py
--- bcc/tools/killsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/killsnoop.py	2021-02-12 14:44:02.579660880 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # killsnoop Trace signals issued by the kill() syscall.
diff -Naur bcc/tools/klockstat.py bcc.new/tools/klockstat.py
--- bcc/tools/klockstat.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/klockstat.py	2021-02-12 14:44:02.579660880 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # klockstat traces lock events and display locks statistics.
 #
diff -Naur bcc/tools/lib/ucalls.py bcc.new/tools/lib/ucalls.py
--- bcc/tools/lib/ucalls.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/lib/ucalls.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # ucalls  Summarize method calls in high-level languages and/or system calls.
diff -Naur bcc/tools/lib/uflow.py bcc.new/tools/lib/uflow.py
--- bcc/tools/lib/uflow.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/lib/uflow.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # uflow  Trace method execution flow in high-level languages.
diff -Naur bcc/tools/lib/ugc.py bcc.new/tools/lib/ugc.py
--- bcc/tools/lib/ugc.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/lib/ugc.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # ugc  Summarize garbage collection events in high-level languages.
diff -Naur bcc/tools/lib/uobjnew.py bcc.new/tools/lib/uobjnew.py
--- bcc/tools/lib/uobjnew.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/lib/uobjnew.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # uobjnew  Summarize object allocations in high-level languages.
diff -Naur bcc/tools/lib/ustat.py bcc.new/tools/lib/ustat.py
--- bcc/tools/lib/ustat.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/lib/ustat.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # ustat  Activity stats from high-level languages, including exceptions,
diff -Naur bcc/tools/lib/uthreads.py bcc.new/tools/lib/uthreads.py
--- bcc/tools/lib/uthreads.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/lib/uthreads.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # uthreads  Trace thread creation/destruction events in high-level languages.
diff -Naur bcc/tools/llcstat.py bcc.new/tools/llcstat.py
--- bcc/tools/llcstat.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/llcstat.py	2021-02-12 14:44:02.579660880 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # llcstat.py Summarize cache references and cache misses by PID.
 #            Cache reference and cache miss are corresponding events defined in
diff -Naur bcc/tools/mdflush.py bcc.new/tools/mdflush.py
--- bcc/tools/mdflush.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/mdflush.py	2021-02-12 14:44:02.579660880 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # mdflush  Trace md flush events.
diff -Naur bcc/tools/memleak.py bcc.new/tools/memleak.py
--- bcc/tools/memleak.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/memleak.py	2021-02-12 14:44:02.579660880 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # memleak   Trace and display outstanding allocations to detect
 #           memory leaks in user-mode processes and the kernel.
diff -Naur bcc/tools/mountsnoop.py bcc.new/tools/mountsnoop.py
--- bcc/tools/mountsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/mountsnoop.py	2021-02-12 14:44:02.579660880 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # mountsnoop Trace mount() and umount syscalls.
 #            For Linux, uses BCC, eBPF. Embedded C.
diff -Naur bcc/tools/mysqld_qslower.py bcc.new/tools/mysqld_qslower.py
--- bcc/tools/mysqld_qslower.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/mysqld_qslower.py	2021-02-12 14:44:02.579660880 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # mysqld_qslower    MySQL server queries slower than a threshold.
 #                   For Linux, uses BCC, BPF. Embedded C.
diff -Naur bcc/tools/netqtop.py bcc.new/tools/netqtop.py
--- bcc/tools/netqtop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/netqtop.py	2021-02-12 14:44:02.579660880 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 
 from __future__ import print_function
 from bcc import BPF
diff -Naur bcc/tools/nfsdist.py bcc.new/tools/nfsdist.py
--- bcc/tools/nfsdist.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/nfsdist.py	2021-02-12 14:44:02.579660880 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # nfsdist   Summarize NFS operation latency
diff -Naur bcc/tools/nfsslower.py bcc.new/tools/nfsslower.py
--- bcc/tools/nfsslower.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/nfsslower.py	2021-02-12 14:44:02.579660880 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # nfsslower     Trace slow NFS operations
diff -Naur bcc/tools/offcputime.py bcc.new/tools/offcputime.py
--- bcc/tools/offcputime.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/offcputime.py	2021-02-12 14:44:02.579660880 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # offcputime    Summarize off-CPU time by stack trace
 #               For Linux, uses BCC, eBPF.
diff -Naur bcc/tools/offwaketime.py bcc.new/tools/offwaketime.py
--- bcc/tools/offwaketime.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/offwaketime.py	2021-02-12 14:44:02.579660880 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # offwaketime   Summarize blocked time by kernel off-CPU stack + waker stack
 #               For Linux, uses BCC, eBPF.
diff -Naur bcc/tools/old/bashreadline.py bcc.new/tools/old/bashreadline.py
--- bcc/tools/old/bashreadline.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/bashreadline.py	2021-02-12 14:44:02.584660901 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # bashreadline  Print entered bash commands from all running shells.
 #               For Linux, uses BCC, eBPF. Embedded C.
diff -Naur bcc/tools/old/biosnoop.py bcc.new/tools/old/biosnoop.py
--- bcc/tools/old/biosnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/biosnoop.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # biosnoop  Trace block device I/O and print details including issuing PID.
diff -Naur bcc/tools/old/compactsnoop.py bcc.new/tools/old/compactsnoop.py
--- bcc/tools/old/compactsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/compactsnoop.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # compactsnoop  Trace compact zone and print details including issuing PID.
diff -Naur bcc/tools/old/filelife.py bcc.new/tools/old/filelife.py
--- bcc/tools/old/filelife.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/filelife.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # filelife    Trace the lifespan of short-lived files.
diff -Naur bcc/tools/old/gethostlatency.py bcc.new/tools/old/gethostlatency.py
--- bcc/tools/old/gethostlatency.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/gethostlatency.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # gethostlatency  Show latency for getaddrinfo/gethostbyname[2] calls.
 #                 For Linux, uses BCC, eBPF. Embedded C.
diff -Naur bcc/tools/old/killsnoop.py bcc.new/tools/old/killsnoop.py
--- bcc/tools/old/killsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/killsnoop.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # killsnoop Trace signals issued by the kill() syscall.
diff -Naur bcc/tools/old/memleak.py bcc.new/tools/old/memleak.py
--- bcc/tools/old/memleak.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/memleak.py	2021-02-12 14:44:02.584660901 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # memleak   Trace and display outstanding allocations to detect
 #           memory leaks in user-mode processes and the kernel.
diff -Naur bcc/tools/old/offcputime.py bcc.new/tools/old/offcputime.py
--- bcc/tools/old/offcputime.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/offcputime.py	2021-02-12 14:44:02.584660901 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # offcputime    Summarize off-CPU time by kernel stack trace
 #               For Linux, uses BCC, eBPF.
diff -Naur bcc/tools/old/offwaketime.py bcc.new/tools/old/offwaketime.py
--- bcc/tools/old/offwaketime.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/offwaketime.py	2021-02-12 14:44:02.584660901 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # offwaketime   Summarize blocked time by kernel off-CPU stack + waker stack
 #               For Linux, uses BCC, eBPF.
diff -Naur bcc/tools/old/oomkill.py bcc.new/tools/old/oomkill.py
--- bcc/tools/old/oomkill.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/oomkill.py	2021-02-12 14:44:02.584660901 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # oomkill   Trace oom_kill_process(). For Linux, uses BCC, eBPF.
 #
diff -Naur bcc/tools/old/opensnoop.py bcc.new/tools/old/opensnoop.py
--- bcc/tools/old/opensnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/opensnoop.py	2021-02-12 14:44:02.584660901 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # opensnoop Trace open() syscalls.
diff -Naur bcc/tools/old/profile.py bcc.new/tools/old/profile.py
--- bcc/tools/old/profile.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/profile.py	2021-02-12 14:44:02.584660901 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # profile  Profile CPU usage by sampling stack traces at a timed interval.
diff -Naur bcc/tools/old/softirqs.py bcc.new/tools/old/softirqs.py
--- bcc/tools/old/softirqs.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/softirqs.py	2021-02-12 14:44:02.584660901 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # softirqs  Summarize soft IRQ (interrupt) event time.
diff -Naur bcc/tools/old/stackcount.py bcc.new/tools/old/stackcount.py
--- bcc/tools/old/stackcount.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/stackcount.py	2021-02-12 14:44:02.584660901 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # stackcount    Count kernel function calls and their stack traces.
 #               For Linux, uses BCC, eBPF.
diff -Naur bcc/tools/old/stacksnoop.py bcc.new/tools/old/stacksnoop.py
--- bcc/tools/old/stacksnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/stacksnoop.py	2021-02-12 14:44:02.584660901 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # stacksnoop    Trace a kernel function and print all kernel stack traces.
 #               For Linux, uses BCC, eBPF, and currently x86_64 only. Inline C.
diff -Naur bcc/tools/old/statsnoop.py bcc.new/tools/old/statsnoop.py
--- bcc/tools/old/statsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/statsnoop.py	2021-02-12 14:44:02.584660901 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # statsnoop Trace stat() syscalls.
diff -Naur bcc/tools/old/syncsnoop.py bcc.new/tools/old/syncsnoop.py
--- bcc/tools/old/syncsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/syncsnoop.py	2021-02-12 14:44:02.584660901 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # syncsnoop Trace sync() syscall.
diff -Naur bcc/tools/old/tcpaccept.py bcc.new/tools/old/tcpaccept.py
--- bcc/tools/old/tcpaccept.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/tcpaccept.py	2021-02-12 14:44:02.584660901 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # tcpaccept Trace TCP accept()s.
diff -Naur bcc/tools/old/tcpconnect.py bcc.new/tools/old/tcpconnect.py
--- bcc/tools/old/tcpconnect.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/old/tcpconnect.py	2021-02-12 14:44:02.584660901 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # tcpconnect    Trace TCP connect()s.
diff -Naur bcc/tools/oomkill.py bcc.new/tools/oomkill.py
--- bcc/tools/oomkill.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/oomkill.py	2021-02-12 14:44:02.579660880 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # oomkill   Trace oom_kill_process(). For Linux, uses BCC, eBPF.
 #
diff -Naur bcc/tools/opensnoop.py bcc.new/tools/opensnoop.py
--- bcc/tools/opensnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/opensnoop.py	2021-02-12 14:44:02.580660884 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # opensnoop Trace open() syscalls.
diff -Naur bcc/tools/pidpersec.py bcc.new/tools/pidpersec.py
--- bcc/tools/pidpersec.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/pidpersec.py	2021-02-12 14:44:02.580660884 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # pidpersec Count new processes (via fork).
diff -Naur bcc/tools/profile.py bcc.new/tools/profile.py
--- bcc/tools/profile.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/profile.py	2021-02-12 14:44:02.580660884 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # profile  Profile CPU usage by sampling stack traces at a timed interval.
diff -Naur bcc/tools/readahead.py bcc.new/tools/readahead.py
--- bcc/tools/readahead.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/readahead.py	2021-02-12 14:44:02.580660884 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # readahead     Show performance of read-ahead cache
diff -Naur bcc/tools/runqlat.py bcc.new/tools/runqlat.py
--- bcc/tools/runqlat.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/runqlat.py	2021-02-12 14:44:02.580660884 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # runqlat   Run queue (scheduler) latency as a histogram.
diff -Naur bcc/tools/runqlen.py bcc.new/tools/runqlen.py
--- bcc/tools/runqlen.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/runqlen.py	2021-02-12 14:44:02.581660888 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # runqlen    Summarize scheduler run queue length as a histogram.
diff -Naur bcc/tools/runqslower.py bcc.new/tools/runqslower.py
--- bcc/tools/runqslower.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/runqslower.py	2021-02-12 14:44:02.581660888 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # runqslower    Trace long process scheduling delays.
diff -Naur bcc/tools/shmsnoop.py bcc.new/tools/shmsnoop.py
--- bcc/tools/shmsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/shmsnoop.py	2021-02-12 14:44:02.581660888 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # shmsnoop Trace shm*() syscalls.
diff -Naur bcc/tools/slabratetop.py bcc.new/tools/slabratetop.py
--- bcc/tools/slabratetop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/slabratetop.py	2021-02-12 14:44:02.581660888 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # slabratetop  Summarize kmem_cache_alloc() calls.
diff -Naur bcc/tools/sofdsnoop.py bcc.new/tools/sofdsnoop.py
--- bcc/tools/sofdsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/sofdsnoop.py	2021-02-12 14:44:02.581660888 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # sofdsnoop traces file descriptors passed via socket
diff -Naur bcc/tools/softirqs.py bcc.new/tools/softirqs.py
--- bcc/tools/softirqs.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/softirqs.py	2021-02-12 14:44:02.581660888 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # softirqs  Summarize soft IRQ (interrupt) event time.
diff -Naur bcc/tools/solisten.py bcc.new/tools/solisten.py
--- bcc/tools/solisten.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/solisten.py	2021-02-12 14:44:02.581660888 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # solisten      Trace TCP listen events
 #               For Linux, uses BCC, eBPF. Embedded C.
diff -Naur bcc/tools/sslsniff.py bcc.new/tools/sslsniff.py
--- bcc/tools/sslsniff.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/sslsniff.py	2021-02-12 14:44:02.581660888 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # sslsniff  Captures data on read/recv or write/send functions of OpenSSL,
 #           GnuTLS and NSS
diff -Naur bcc/tools/stackcount.py bcc.new/tools/stackcount.py
--- bcc/tools/stackcount.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/stackcount.py	2021-02-12 14:44:02.581660888 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # stackcount    Count events and their stack traces.
 #               For Linux, uses BCC, eBPF.
diff -Naur bcc/tools/statsnoop.py bcc.new/tools/statsnoop.py
--- bcc/tools/statsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/statsnoop.py	2021-02-12 14:44:02.581660888 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # statsnoop Trace stat() syscalls.
diff -Naur bcc/tools/swapin.py bcc.new/tools/swapin.py
--- bcc/tools/swapin.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/swapin.py	2021-02-12 14:44:02.581660888 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # swapin        Count swapins by process.
diff -Naur bcc/tools/syncsnoop.py bcc.new/tools/syncsnoop.py
--- bcc/tools/syncsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/syncsnoop.py	2021-02-12 14:44:02.581660888 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # syncsnoop Trace sync() syscall.
diff -Naur bcc/tools/syscount.py bcc.new/tools/syscount.py
--- bcc/tools/syscount.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/syscount.py	2021-02-12 14:44:02.581660888 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # syscount   Summarize syscall counts and latencies.
 #
diff -Naur bcc/tools/tcpaccept.py bcc.new/tools/tcpaccept.py
--- bcc/tools/tcpaccept.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/tcpaccept.py	2021-02-12 14:44:02.581660888 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # tcpaccept Trace TCP accept()s.
diff -Naur bcc/tools/tcpconnect.py bcc.new/tools/tcpconnect.py
--- bcc/tools/tcpconnect.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/tcpconnect.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # tcpconnect    Trace TCP connect()s.
diff -Naur bcc/tools/tcpconnlat.py bcc.new/tools/tcpconnlat.py
--- bcc/tools/tcpconnlat.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/tcpconnlat.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # tcpconnlat    Trace TCP active connection latency (connect).
diff -Naur bcc/tools/tcpdrop.py bcc.new/tools/tcpdrop.py
--- bcc/tools/tcpdrop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/tcpdrop.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # tcpdrop   Trace TCP kernel-dropped packets/segments.
diff -Naur bcc/tools/tcplife.py bcc.new/tools/tcplife.py
--- bcc/tools/tcplife.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/tcplife.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # tcplife   Trace the lifespan of TCP sessions and summarize.
diff -Naur bcc/tools/tcpretrans.py bcc.new/tools/tcpretrans.py
--- bcc/tools/tcpretrans.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/tcpretrans.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # tcpretrans    Trace or count TCP retransmits and TLPs.
diff -Naur bcc/tools/tcprtt.py bcc.new/tools/tcprtt.py
--- bcc/tools/tcprtt.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/tcprtt.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # tcprtt    Summarize TCP RTT as a histogram. For Linux, uses BCC, eBPF.
diff -Naur bcc/tools/tcpstates.py bcc.new/tools/tcpstates.py
--- bcc/tools/tcpstates.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/tcpstates.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # -*- coding: utf-8 -*-
 # @lint-avoid-python-3-compatibility-imports
 #
diff -Naur bcc/tools/tcpsubnet.py bcc.new/tools/tcpsubnet.py
--- bcc/tools/tcpsubnet.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/tcpsubnet.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # tcpsubnet   Summarize TCP bytes sent to different subnets.
diff -Naur bcc/tools/tcpsynbl.py bcc.new/tools/tcpsynbl.py
--- bcc/tools/tcpsynbl.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/tcpsynbl.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # tcpsynbl      Show TCP SYN backlog.
diff -Naur bcc/tools/tcptop.py bcc.new/tools/tcptop.py
--- bcc/tools/tcptop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/tcptop.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # tcptop    Summarize TCP send/recv throughput by host.
diff -Naur bcc/tools/tcptracer.py bcc.new/tools/tcptracer.py
--- bcc/tools/tcptracer.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/tcptracer.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # tcpv4tracer   Trace TCP connections.
 #               For Linux, uses BCC, eBPF. Embedded C.
diff -Naur bcc/tools/threadsnoop.py bcc.new/tools/threadsnoop.py
--- bcc/tools/threadsnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/threadsnoop.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # threadsnoop   List new thread creation.
diff -Naur bcc/tools/tplist.py bcc.new/tools/tplist.py
--- bcc/tools/tplist.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/tplist.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # tplist    Display kernel tracepoints or USDT probes and their formats.
 #
diff -Naur bcc/tools/trace.py bcc.new/tools/trace.py
--- bcc/tools/trace.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/trace.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # trace         Trace a function and print a trace message based on its
 #               parameters, with an optional filter.
diff -Naur bcc/tools/ttysnoop.py bcc.new/tools/ttysnoop.py
--- bcc/tools/ttysnoop.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/ttysnoop.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # ttysnoop   Watch live output from a tty or pts device.
diff -Naur bcc/tools/vfscount.py bcc.new/tools/vfscount.py
--- bcc/tools/vfscount.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/vfscount.py	2021-02-12 14:44:02.582660893 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # vfscount  Count VFS calls ("vfs_*").
diff -Naur bcc/tools/vfsstat.py bcc.new/tools/vfsstat.py
--- bcc/tools/vfsstat.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/vfsstat.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # vfsstat   Count some VFS calls.
diff -Naur bcc/tools/wakeuptime.py bcc.new/tools/wakeuptime.py
--- bcc/tools/wakeuptime.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/wakeuptime.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # wakeuptime    Summarize sleep to wakeup time by waker kernel stack
 #               For Linux, uses BCC, eBPF.
diff -Naur bcc/tools/xfsdist.py bcc.new/tools/xfsdist.py
--- bcc/tools/xfsdist.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/xfsdist.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # xfsdist  Summarize XFS operation latency.
diff -Naur bcc/tools/xfsslower.py bcc.new/tools/xfsslower.py
--- bcc/tools/xfsslower.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/xfsslower.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # xfsslower  Trace slow XFS operations.
diff -Naur bcc/tools/zfsdist.py bcc.new/tools/zfsdist.py
--- bcc/tools/zfsdist.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/zfsdist.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # zfsdist  Summarize ZFS operation latency.
diff -Naur bcc/tools/zfsslower.py bcc.new/tools/zfsslower.py
--- bcc/tools/zfsslower.py	2021-01-05 06:28:48.000000000 +0100
+++ bcc.new/tools/zfsslower.py	2021-02-12 14:44:02.583660897 +0100
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 # @lint-avoid-python-3-compatibility-imports
 #
 # zfsslower  Trace slow ZFS operations.
