--- glest_source_2.0.0/source/shared_lib/include/graphics/particle.h.old	2006-04-14 23:10:39.000000000 +0200
+++ glest_source_2.0.0/source/shared_lib/include/graphics/particle.h	2006-04-14 23:10:51.000000000 +0200
@@ -310,7 +310,7 @@
 	void setTrajectoryFrequency(float trajectoryFrequency)	{this->trajectoryFrequency= trajectoryFrequency;}
 	void setPath(Vec3f startPos, Vec3f endPos);
 
-	static Trajectory ProjectileParticleSystem::strToTrajectory(const string &str);
+	static Trajectory strToTrajectory(const string &str);
 };
 
 // =====================================================
