From 909638221e311bcf0a815fc808884f043be71eff Mon Sep 17 00:00:00 2001
From: Signedbit <signedbit@users.noreply.github.com>
Date: Mon, 23 Dec 2019 19:43:37 -0800
Subject: [PATCH] Fix #1087

---
 src/login.coffee | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/login.coffee b/src/login.coffee
index 1a138c39..f137a638 100644
--- a/src/login.coffee
+++ b/src/login.coffee
@@ -6,6 +6,10 @@ Q = require 'q'
 # https://github.com/tdryer/hangups/issues/260#issuecomment-246578670
 LOGIN_URL = "https://accounts.google.com/o/oauth2/programmatic_auth?hl=en&scope=https%3A%2F%2Fwww.google.com%2Faccounts%2FOAuthLogin+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email&client_id=936475272427.apps.googleusercontent.com&access_type=offline&delegated_client_id=183697946088-m3jnlsqshjhh5lbvg05k46q1k4qqtrgn.apps.googleusercontent.com&top_level_cookie=1"
 
+# Hack the user agent so this works again
+# Credit to https://github.com/yakyak/yakyak/issues/1087#issuecomment-565170640
+AGENT = "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Ubuntu Chromium/80.0.3904.70 Safari/537.36"
+
 # promise for one-time oauth token
 module.exports = (mainWindow) -> Q.Promise (rs) ->
 
@@ -26,4 +30,5 @@ module.exports = (mainWindow) -> Q.Promise (rs) ->
                 rs(oauth_code) if oauth_code
 
     # redirect to google oauth
-    mainWindow.loadURL LOGIN_URL
+    options = {"userAgent": AGENT}
+    mainWindow.loadURL LOGIN_URL, options
