--- regex-posix-0.96.0.1/regex-posix.cabal	2001-09-09 03:46:40.000000000 +0200
+++ regex-posix-0.96.0.1.new/regex-posix.cabal	2024-07-10 21:19:44.361987543 +0200
@@ -1,14 +1,15 @@
 cabal-version:          1.12
 name:                   regex-posix
 version:                0.96.0.1
+x-revision:             3
 
 build-type:             Simple
 license:                BSD3
 license-file:           LICENSE
 copyright:              Copyright (c) 2007-2010, Christopher Kuklewicz
 author:                 Christopher Kuklewicz
-maintainer:             hvr@gnu.org, Andreas Abel
-bug-reports:            https://github.com/hvr/regex-posix
+maintainer:             Andreas Abel
+bug-reports:            https://github.com/haskell-hvr/regex-posix
 synopsis:               POSIX Backend for "Text.Regex" (regex-base)
 category:               Text
 description:
@@ -23,29 +24,27 @@
   cbits/myfree.h
 
 tested-with:
-  -- Haskell CI:
-  GHC == 7.0.4
-  GHC == 7.2.2
-  GHC == 7.4.2
-  GHC == 7.6.3
-  GHC == 7.8.4
-  GHC == 7.10.3
-  GHC == 8.0.2
-  GHC == 8.2.2
-  GHC == 8.4.4
-  GHC == 8.6.5
+  GHC == 9.8.0
+  GHC == 9.6.3
+  GHC == 9.4.7
+  GHC == 9.2.8
+  GHC == 9.0.2
+  GHC == 8.10.7
   GHC == 8.8.4
-  GHC == 8.10.4
-  GHC == 9.0.1
+  GHC == 8.6.5
+  GHC == 8.4.4
+  GHC == 8.2.2
+  GHC == 8.0.2
+  GHC == 7.10.3
 
 source-repository head
   type:     git
-  location: https://github.com/hvr/regex-posix.git
+  location: https://github.com/haskell-hvr/regex-posix.git
 
 source-repository this
   type:     git
-  location: https://github.com/hvr/regex-base.git
-  tag:      v0.96.0.1
+  location: https://github.com/haskell-hvr/regex-base.git
+  tag:      v0.96.0.1-r3
 
 flag _regex-posix-clib
   manual: False
@@ -85,9 +84,9 @@
       FlexibleInstances
 
   build-depends: regex-base == 0.94.*
-               , base       >= 4.3 && < 4.17
-               , containers >= 0.4 && < 0.7
-               , bytestring >= 0.9 && < 0.12
+               , base       >= 4.3 && < 5
+               , containers >= 0.4 && < 0.8
+               , bytestring >= 0.9 && < 0.13
                , array      >= 0.3 && < 0.6
 
   if impl(ghc < 8)
