diff -Naur regex-posix-0.96.0.1/regex-posix.cabal regex-posix-0.96.0.1.new/regex-posix.cabal
--- regex-posix-0.96.0.1/regex-posix.cabal	2001-09-09 03:46:40.000000000 +0200
+++ regex-posix-0.96.0.1.new/regex-posix.cabal	2022-10-14 11:06:39.526626689 +0200
@@ -1,14 +1,15 @@
 cabal-version:          1.12
 name:                   regex-posix
 version:                0.96.0.1
+x-revision:             1
 
 build-type:             Simple
 license:                BSD3
 license-file:           LICENSE
 copyright:              Copyright (c) 2007-2010, Christopher Kuklewicz
 author:                 Christopher Kuklewicz
-maintainer:             hvr@gnu.org, Andreas Abel
-bug-reports:            https://github.com/hvr/regex-posix
+maintainer:             Andreas Abel
+bug-reports:            https://github.com/haskell-hvr/regex-posix
 synopsis:               POSIX Backend for "Text.Regex" (regex-base)
 category:               Text
 description:
@@ -23,20 +24,21 @@
   cbits/myfree.h
 
 tested-with:
-  -- Haskell CI:
-  GHC == 7.0.4
-  GHC == 7.2.2
-  GHC == 7.4.2
-  GHC == 7.6.3
-  GHC == 7.8.4
-  GHC == 7.10.3
-  GHC == 8.0.2
-  GHC == 8.2.2
-  GHC == 8.4.4
-  GHC == 8.6.5
+  GHC == 9.4.1
+  GHC == 9.2.2
+  GHC == 9.0.2
+  GHC == 8.10.7
   GHC == 8.8.4
-  GHC == 8.10.4
-  GHC == 9.0.1
+  GHC == 8.6.5
+  GHC == 8.4.4
+  GHC == 8.2.2
+  GHC == 8.0.2
+  GHC == 7.10.3
+  GHC == 7.8.4
+  GHC == 7.6.3
+  GHC == 7.4.2
+  GHC == 7.2.2
+  GHC == 7.0.4
 
 source-repository head
   type:     git
@@ -45,7 +47,7 @@
 source-repository this
   type:     git
   location: https://github.com/hvr/regex-base.git
-  tag:      v0.96.0.1
+  tag:      v0.96.0.1-r1
 
 flag _regex-posix-clib
   manual: False
@@ -85,7 +87,7 @@
       FlexibleInstances
 
   build-depends: regex-base == 0.94.*
-               , base       >= 4.3 && < 4.17
+               , base       >= 4.3 && < 5
                , containers >= 0.4 && < 0.7
                , bytestring >= 0.9 && < 0.12
                , array      >= 0.3 && < 0.6
