diff --git a/regex-posix.cabal b/regex-posix.cabal
index b93c86d..3f441e9 100644
--- a/regex-posix.cabal
+++ b/regex-posix.cabal
@@ -22,6 +22,20 @@ extra-source-files:
   ChangeLog.md
   cbits/myfree.h
 
+tested-with:
+  GHC ==7.0.2
+   || ==7.2.2
+   || ==7.4.2
+   || ==7.6.3
+   || ==7.8.4
+   || ==7.10.3
+   || ==8.0.2
+   || ==8.2.2
+   || ==8.4.4
+   || ==8.6.5
+   || ==8.8.3
+   || ==8.10.1
+
 source-repository head
   type:     git
   location: https://github.com/hvr/regex-posix.git
@@ -66,7 +80,7 @@ library
       FlexibleInstances
 
   build-depends: regex-base == 0.94.*
-               , base       >= 4.3 && < 4.14
+               , base       >= 4.3 && < 4.15
                , containers >= 0.4 && < 0.7
                , bytestring >= 0.9 && < 0.11
                , array      >= 0.3 && < 0.6
