--- epgvbi/btdrv4linux.c.orig	2006-04-08 00:23:24.000000000 +0200
+++ epgvbi/btdrv4linux.c	2006-04-08 00:23:36.000000000 +0200
@@ -1939,7 +1939,7 @@
 
       zvbi_rd.sampling_rate     = vfmt2.fmt.vbi.sampling_rate;
       zvbi_rd.bytes_per_line    = vfmt2.fmt.vbi.samples_per_line;
-      zvbi_rd.offset            = vfmt2.fmt.vbi.offset;
+      zvbi_rd.offset            = 244;
       zvbi_rd.start[0]          = vfmt2.fmt.vbi.start[0];
       zvbi_rd.count[0]          = vfmt2.fmt.vbi.count[0];
       zvbi_rd.start[1]          = vfmt2.fmt.vbi.start[1];
