--- src/Classes/ActivityAreaCtrl.cpp.old	2005-11-22 03:43:38.000000000 +0800
+++ src/Classes/ActivityAreaCtrl.cpp	2005-12-06 02:03:28.000000000 +0800
@@ -204,6 +204,16 @@
 		if ( pSelectedBox == m_ActivityBox[i] )
 		{	
 			size_t k = 0;
+                        if ( wxGetApp().Prefs.eSelStyle == MUSIK_SELECTION_TYPE_NORMAL )
+                        {
+                                for(size_t j = i+1 ; j < ActivityBoxesMaxCount; j++)
+                                {
+                                        pOtherBoxes[k] = m_ActivityBox[j];
+                                        k++;
+                                }
+                        }
+                        else
+                        {
 			// assign not selected boxes to pOtherBoxes array
 			for(size_t j = 0 ; j < WXSIZEOF(pOtherBoxes);j++)
 			{
@@ -215,6 +225,7 @@
 					k = 0;
 				pOtherBoxes[j] = m_ActivityBox[k++];
 			}
+                        }
 			break;
 		}
 	}
@@ -226,12 +237,12 @@
 	//--- parent and which are children, if there	---//
 	//--- is no parent already.						---//
 	//-------------------------------------------------//
-	if ( wxGetApp().Prefs.eSelStyle == MUSIK_SELECTION_TYPE_STANDARD )
+	if ( wxGetApp().Prefs.eSelStyle == MUSIK_SELECTION_TYPE_STANDARD || wxGetApp().Prefs.eSelStyle == MUSIK_SELECTION_TYPE_NORMAL )
 	{
 		if(GetParentBox() == NULL)
 			SetParent( pSelectedBox , false );
 	}
-    if ( wxGetApp().Prefs.eSelStyle == MUSIK_SELECTION_TYPE_STANDARD || wxGetApp().Prefs.eSelStyle == MUSIK_SELECTION_TYPE_SLOPPY )
+    if ( wxGetApp().Prefs.eSelStyle == MUSIK_SELECTION_TYPE_STANDARD || wxGetApp().Prefs.eSelStyle == MUSIK_SELECTION_TYPE_SLOPPY || wxGetApp().Prefs.eSelStyle == MUSIK_SELECTION_TYPE_NORMAL )
     {
 	    //---------------------------------------------------//
 	    //--- if we're hiding unselected entries          ---//
@@ -276,6 +287,7 @@
             }
 		    wxArrayString temp_list;
 		    if ( ( wxGetApp().Prefs.eSelStyle == MUSIK_SELECTION_TYPE_STANDARD && GetParentBox() == pSelectedBox ) 
+                || wxGetApp().Prefs.eSelStyle == MUSIK_SELECTION_TYPE_NORMAL
                 || wxGetApp().Prefs.eSelStyle == MUSIK_SELECTION_TYPE_SLOPPY )
 		    {
 			    for(size_t j = 0 ; j < WXSIZEOF(pOtherBoxes);j++)
