--- src/MusikUtils.h.old	2005-11-22 03:43:40.000000000 +0800
+++ src/MusikUtils.h	2005-12-06 02:21:35.000000000 +0800
@@ -446,8 +446,8 @@
 {
     wxString sDataPath(wxT("data/"));
 #if wxCHECK_VERSION(2,5,4) && !defined(__WXMAC__)
-    
-    wxStandardPathsBase & stdpaths = wxStandardPaths::Get();
+    wxStandardPaths stdpaths;
+    stdpaths.SetInstallPrefix(wxT("/usr"));
     wxString theDataPath(stdpaths.GetDataDir()+ wxT("/") + sDataPath );
     if(!wxDirExists(theDataPath))
         theDataPath = sDataPath;
