--- src/Frames/Prefs/OptionSelectionsPanel.cpp.old	2005-11-22 03:43:38.000000000 +0800
+++ src/Frames/Prefs/OptionSelectionsPanel.cpp	2005-12-06 01:59:56.000000000 +0800
@@ -64,6 +64,7 @@
     cmbSelStyle->Append( _( "Standard" ) );
     cmbSelStyle->Append( _( "Sloppy" ) );
     cmbSelStyle->Append( _( "Highlight") );
+    cmbSelStyle->Append( _( "Normal") );
 
     vsOptions_Selections->Add( PREF_STATICTEXT(_("Selection Style:" )),		0, wxCENTER | wxRIGHT | wxALIGN_CENTER_VERTICAL, 0 );
     vsOptions_Selections->Add( cmbSelStyle,		1, wxCENTER, 0 );
