--- xalan_c-1.12/CMakeLists.txt	2020-06-07 11:17:25.000000000 +0200
+++ xalan_c-1.12.new/CMakeLists.txt	2024-05-20 19:29:47.168720068 +0200
@@ -34,7 +34,7 @@
 
 # Try C++14, then fall back to C++11 and C++98.  Used for feature tests
 # for optional features.
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 
 # Use folders (for IDE project grouping)
 set_property(GLOBAL PROPERTY USE_FOLDERS ON)
