diff -Naur memtest86+-1.70.orig/Makefile memtest86+-1.70/Makefile
--- memtest86+-1.70.orig/Makefile	2007-07-17 22:24:07.000000000 +0200
+++ memtest86+-1.70/Makefile	2007-07-17 22:24:21.000000000 +0200
@@ -12,7 +12,7 @@
 #
 # gcc compiler options, these settings should suffice
 #
-CCFLAGS=-Wall -m32 -march=i486 -Os -fomit-frame-pointer -fno-builtin -ffreestanding
+CCFLAGS=-Wall -m32 -march=i486 -Os -D__x86_64__ -fomit-frame-pointer -fno-builtin -ffreestanding
 
 AS=as -32
 
@@ -21,7 +21,7 @@
 all: memtest.bin memtest
 
 reloc.o: reloc.c
-	$(CC) -c -m32 -march=i486 -fPIC -Wall -g -O2 -fno-strict-aliasing reloc.c
+	$(CC) -c -m32 -march=i486 -D__x86_64__ -fPIC -Wall -g -O2 -fno-strict-aliasing reloc.c
 
 test.o: test.c test.h defs.h config.h
 	$(CC) -c $(CCFLAGS) test.c
