--- pangomm-2.42.1/meson.build	2020-03-21 09:00:38.000000000 +0100
+++ pangomm-2.42.1.new/meson.build	2020-11-30 16:12:24.829030099 +0100
@@ -89,7 +89,7 @@
 # The -mm libraries do not yet have pkg-config files for MSVC builds,
 # so check for them manually
 glibmm_req_minor_ver = '4'
-cairomm_req_minor_ver = '0'
+cairomm_req_minor_ver = '16'
 
 if is_msvc
   # We must have Visual Studio 2013 or later...
