diff -Naur tvtime-1.0.2.orig//src/videodev2.h tvtime-1.0.2/src/videodev2.h
--- tvtime-1.0.2.orig//src/videodev2.h	2005-01-17 03:36:40.000000000 +0100
+++ tvtime-1.0.2/src/videodev2.h	2010-06-02 19:44:08.000000000 +0200
@@ -16,7 +16,6 @@
 #ifdef __KERNEL__
 #include <linux/time.h> /* need struct timeval */
 #endif
-#include <linux/compiler.h> /* need __user */
 
 /* for kernel versions 2.4.26 and below: */
 #ifndef __user
