diff -Naur openldap-2.4.43/build/openldap.m4 openldap-2.4.43-db/build/openldap.m4
--- openldap-2.4.43/build/openldap.m4	2015-11-30 18:02:00.000000000 +0100
+++ openldap-2.4.43-db/build/openldap.m4	2016-01-06 19:18:02.981818924 +0100
@@ -554,12 +554,9 @@
 #define DB_VERSION_FULL        ((DB_VERSION_MAJOR<<16)|(DB_VERSION_MINOR<<8)|DB_VERSION_PATCH)
 
 /* require 4.4 or later, but less than 6.0.20 */
-#if DB_VERSION_FULL >= 0x040400 && DB_VERSION_FULL < 0x060014
+#if DB_VERSION_FULL >= 0x040400
 	__db_version_compat
 #endif
-#if DB_VERSION_FULL >= 0x060014
-#error "BerkeleyDB 6.0.20+ license is incompatible with LDAP"
-#endif
 	], [ol_cv_bdb_compat=yes], [ol_cv_bdb_compat=no])])
 ])
 
