diff -aur enhance-old/src/lib/enhance.c enhance/src/lib/enhance.c
--- enhance-old/src/lib/enhance.c	2007-05-14 10:49:02.000000000 +0200
+++ enhance/src/lib/enhance.c	2007-05-14 11:15:38.000000000 +0200
@@ -552,7 +552,7 @@
 	     char *tok;
 	     char *value;
 	     
-	     etk_combobox_column_add(ETK_COMBOBOX(wid->wid), ETK_COMBOBOX_LABEL, 15, ETK_TRUE, ETK_FALSE, ETK_FALSE, 0.0, 0.5);
+	     etk_combobox_column_add(ETK_COMBOBOX(wid->wid), ETK_COMBOBOX_LABEL, 15, ETK_COMBOBOX_EXPAND_FILL, 0.0);
 	     etk_combobox_build(ETK_COMBOBOX(wid->wid));
 	     value = strdup(node->value);
 	     tok = strtok(value, "\n");
@@ -582,7 +582,7 @@
 	IF_PARENT_CLASS("GtkWindow")
           {
 	     PROPERTY_INT;
-	     etk_window_resize(ETK_WINDOW(wid->wid),  value, ETK_WINDOW(wid->wid)->height);
+	     etk_window_resize(ETK_WINDOW(wid->wid),  value, value);
           }
      }
    
@@ -591,7 +591,7 @@
 	IF_PARENT_CLASS("GtkWindow")
           {
 	     PROPERTY_INT;
-	     etk_window_resize(ETK_WINDOW(wid->wid), ETK_WINDOW(wid->wid)->width, value);
+	     etk_window_resize(ETK_WINDOW(wid->wid), value, value);
           }
      }     
    
@@ -719,7 +719,7 @@
         PROPERTY_STR;
         /* TODO : edj? */
 
-        etk_image_set_from_file(ETK_IMAGE(wid->wid), value);
+        etk_image_set_from_file(ETK_IMAGE(wid->wid), value, ETK_IMAGE(wid->wid)->info.file.key);
      }
 
    else if(!strcmp(name, "active"))
diff -aur enhance-old/src/lib/enhance_widget.c enhance/src/lib/enhance_widget.c
--- enhance-old/src/lib/enhance_widget.c	2007-05-14 10:49:02.000000000 +0200
+++ enhance/src/lib/enhance_widget.c	2007-05-14 11:23:24.000000000 +0200
@@ -907,22 +907,20 @@
                                             ETK_DIALOG(parent->wid),
                                             ETK_BUTTON(child->wid),
                                             atoi(prop->value),
-                                            expand, fill, padding,
-                                            ETK_FALSE);
+                                            expand, fill, padding);
                     }
                   else
                     {
                       etk_dialog_pack_widget_in_action_area(
                                             ETK_DIALOG(parent->wid), child->wid,
-                                            expand, fill, padding, ETK_FALSE);
+                                            expand, fill, padding);
                     }
                 }
         else
           {
             etk_dialog_pack_in_main_area(ETK_DIALOG(parent->wid), 
                                          child->wid, 
-                                         expand, fill, padding, 
-                                         ETK_FALSE);
+                                         expand, fill, padding);
           }
      }
    if(!strcmp(parent_class, "GtkFrame"))
