diff -Naur mdadm-2.5.orig/Makefile mdadm-2.5/Makefile
--- mdadm-2.5.orig/Makefile	2006-05-26 06:27:55.000000000 +0200
+++ mdadm-2.5/Makefile	2006-05-28 22:07:43.000000000 +0200
@@ -39,8 +39,8 @@
 KLIBC_GCC = gcc -nostdinc -iwithprefix include -I$(KLIBC)/klibc/include -I$(KLIBC)/linux/include -I$(KLIBC)/klibc/arch/i386/include -I$(KLIBC)/klibc/include/bits32
 
 CC = $(CROSS_COMPILE)gcc
-CXFLAGS = -ggdb
-CWFLAGS = -Wall -Werror -Wstrict-prototypes
+CFLAGS ?= -ggdb -Werror
+CWFLAGS = -Wall -Wstrict-prototypes
 
 ifdef DEBIAN
 CPPFLAGS= -DDEBIAN
@@ -51,7 +51,7 @@
 SYSCONFDIR = /etc
 CONFFILE = $(SYSCONFDIR)/mdadm.conf
 MAILCMD =/usr/sbin/sendmail -t
-CFLAGS = $(CWFLAGS) $(STATIC) $(CPPFLAGS) -DCONFFILE=\"$(CONFFILE)\" $(CXFLAGS) -DSendmail=\""$(MAILCMD)"\"
+CFLAGS += $(CWFLAGS) -DCONFFILE=\"$(CONFFILE)\" -DSendmail=\""$(MAILCMD)"\"
 
 # If you want a static binary, you might uncomment these
 # LDFLAGS = -static
