diff -Naur mdadm-2.0.orig/Makefile mdadm-2.0/Makefile
--- mdadm-2.0.orig/Makefile	2005-08-25 09:30:20.000000000 +0200
+++ mdadm-2.0/Makefile	2005-08-26 12:44:48.000000000 +0200
@@ -38,12 +38,12 @@
 KLIBC_GCC = gcc -nostdinc -iwithprefix include -I$(KLIBC)/klibc/include -I$(KLIBC)/linux/include -I$(KLIBC)/klibc/arch/i386/include -I$(KLIBC)/klibc/include/bits32
 
 CC = $(CROSS_COMPILE)gcc
-CXFLAGS = -ggdb
-CWFLAGS = -Wall -Werror -Wstrict-prototypes
+CXFLAGS =
+CWFLAGS = -Wall -Wstrict-prototypes
 SYSCONFDIR = /etc
 CONFFILE = $(SYSCONFDIR)/mdadm.conf
 MAILCMD =/usr/sbin/sendmail -t
-CFLAGS = $(CWFLAGS) -DCONFFILE=\"$(CONFFILE)\" $(CXFLAGS) -DSendmail=\""$(MAILCMD)"\"
+CFLAGS += $(CWFLAGS) -DCONFFILE=\"$(CONFFILE)\" $(CXFLAGS) -DSendmail=\""$(MAILCMD)"\"
 
 # If you want a static binary, you might uncomment these
 # LDFLAGS = -static
