diff -Naur mdadm-2.5.1.orig/Makefile mdadm-2.5.1/Makefile
--- mdadm-2.5.1.orig/Makefile	2006-06-16 02:21:52.000000000 +0200
+++ mdadm-2.5.1/Makefile	2006-06-16 03:38:37.000000000 +0200
@@ -38,8 +38,8 @@
 KLIBC_GCC = gcc -nostdinc -iwithprefix include -I$(KLIBC)/klibc/include -I$(KLIBC)/linux/include -I$(KLIBC)/klibc/arch/i386/include -I$(KLIBC)/klibc/include/bits32
 
 CC = $(CROSS_COMPILE)gcc
-CXFLAGS = -ggdb
-CWFLAGS = -Wall -Werror -Wstrict-prototypes
+CFLAGS ?= -ggdb -Werror
+CWFLAGS = -Wall -Wstrict-prototypes
 
 ifdef DEBIAN
 CPPFLAGS= -DDEBIAN
@@ -52,7 +52,7 @@
 CONFFILE2 = $(SYSCONFDIR)/mdadm/mdadm.conf
 MAILCMD =/usr/sbin/sendmail -t
 CONFFILEFLAGS = -DCONFFILE=\"$(CONFFILE)\" -DCONFFILE2=\"$(CONFFILE2)\"
-CFLAGS = $(CWFLAGS) $(CXFLAGS) -DSendmail=\""$(MAILCMD)"\" $(CONFFILEFLAGS)
+CFLAGS += $(CWFLAGS) -DSendmail=\""$(MAILCMD)"\" $(CONFFILEFLAGS)
 
 # If you want a static binary, you might uncomment these
 # LDFLAGS = -static
