diff -Naur circuslinux-1.0.3/configure.in circuslinux-1.0.3-p/configure.in
--- circuslinux-1.0.3/configure.in	2000-06-11 03:50:39.000000000 +0200
+++ circuslinux-1.0.3-p/configure.in	2008-05-27 00:37:54.000000000 +0200
@@ -3,7 +3,7 @@
 
 
 dnl Setup for automake
-AM_INIT_AUTOMAKE(circuslinux,1.0.2)
+AM_INIT_AUTOMAKE(circuslinux,1.0.3)
 
 dnl Detect host info
 AC_CANONICAL_HOST
diff -Naur circuslinux-1.0.3/Makefile.am circuslinux-1.0.3-p/Makefile.am
--- circuslinux-1.0.3/Makefile.am	2001-04-29 08:53:26.000000000 +0200
+++ circuslinux-1.0.3-p/Makefile.am	2008-05-27 00:37:39.000000000 +0200
@@ -21,7 +21,7 @@
 TARGET_DEF=@TARGET_DEF@
 
 
-CFLAGS  = @CFLAGS@ -Wall -O2 -DDATA_PREFIX=\"$(DATA_PREFIX)\" \
+AM_CFLAGS  = @CFLAGS@ -Wall -O2 -DDATA_PREFIX=\"$(DATA_PREFIX)\" \
 	  -D$(TARGET_DEF) -DJOY_$(JOY)
 LIBS    = @LIBS@ 
 
@@ -29,8 +29,8 @@
 bin_PROGRAMS = circuslinux
 
 circuslinux_SOURCES = circuslinux.c
-circuslinux_CFLAGS  = CFLAGS
-circuslinux_LIBS    = LIBS
+circuslinux_CFLAGS  = $(AM_CFLAGS)
+circuslinux_LIBS    = $(LIBS)
 
 docdir = $(prefix)/share/doc/$(PACKAGE)-$(VERSION)
 
@@ -66,7 +66,7 @@
 #	$(CC) circuslinux.o $(LIBS) -o circuslinux
 
 #circuslinux.o:	circuslinux.c
-#	$(CC) -c circuslinux.c $(CFLAGS)
+#	$(CC) -c circuslinux.c $(AM_CFLAGS)
 
 clean:
 	-rm -f circuslinux
