--- a/i18n/lastfm_tr.ts
+++ b/i18n/lastfm_tr.ts
@@ -8,3 +8,3 @@
         <source>A playback error occurred.</source>
-        <translation>Bir çalma hatası meydana geldi.</translation>
+        <translation>Bir çalma hatası oluştu.</translation>
     </message>
@@ -46,3 +46,3 @@
         <source>Copyright 2007 Last.fm Ltd. (C)</source>
-        <translation>Telif hakkı 2007 Last.fm Ltd. (C)</translation>
+        <translation>Telif Hakkı 2007 Last.fm Ltd. (C)</translation>
     </message>
@@ -65,3 +65,3 @@
         <source>Please select a player and enter a path.</source>
-        <translation>Lütfen müzikçalar seçip adres yazın.</translation>
+        <translation>Lütfen bir  müzikçalar seçip adres yazın.</translation>
     </message>
@@ -80,3 +80,3 @@
         <source>Locate the executable</source>
-        <translation>Uygulama dosyasını bul</translation>
+        <translation>Çalıştırılabilir dosyayı bul</translation>
     </message>
@@ -126,3 +126,3 @@
         <comment>An input field is empty</comment>
-        <translation>Giriş alanı boş
+        <translation>Alan Boş
 </translation>
@@ -135,3 +135,3 @@
         <source>The ALSA soundsystem is either busy or not present.</source>
-        <translation>ALSA ses sistemi ya meşgul ya da mevcut değil.</translation>
+        <translation>ALSA ses sistemi meşgul ya da yok.</translation>
     </message>
@@ -143,3 +143,3 @@
         <source>Sorry, there is no more content available for this station. Please choose a different one.</source>
-        <translation>Üzgünüz, bu istasyon için daha fazla içerik yok. Lütfen başka bir tane seçin.</translation>
+        <translation>Bu istasyon için içerik bulunamadı. Lütfen başka bir istasyon seçin.</translation>
     </message>
@@ -151,3 +151,3 @@
         <source>Couldn&apos;t load radio service &apos;%1&apos;. The radio will not work.</source>
-        <translation>Radyo servisi &apos;%1&apos; yüklenemedi. Radyo çalışmayacak.</translation>
+        <translation>&apos;%1&apos; radyo servisi bulunamadı. Radyo çalışmayacak.</translation>
     </message>
@@ -194,3 +194,3 @@
         <source>Downloaded %L1kB of %L2kB.</source>
-        <translation>%L2kB içinden %L1kB indirildi.</translation>
+        <translation>%L2kB&apos;ın %L1kB&apos;ı indirildi.</translation>
     </message>
@@ -199,3 +199,3 @@
         <source>Download failed ( %1 - %2 )</source>
-        <translation>Yükleme başarısız ( %1 - %2 )</translation>
+        <translation>İndirme başarısız oldu ( %1 - %2 )</translation>
     </message>
@@ -207,3 +207,3 @@
         <source>Command string seems to be empty</source>
-        <translation>Komut dizisi boş görünüyor</translation>
+        <translation>Komut dizgesi boş görünüyor</translation>
     </message>
@@ -227,3 +227,3 @@
         <source>No arguments found in command. All commands need at least the plugin ID.</source>
-        <translation>Komutta argüman yok. Bütün komutlarda en azından eklenti tanıtıcı olmalı.</translation>
+        <translation>Komutta argüman yok. Bütün komutlarda en azından eklenti kimliği olmalı.</translation>
     </message>
@@ -247,3 +247,3 @@
         <source>Command &apos;%1&apos; isn&apos;t a valid command</source>
-        <translation>&apos;%1&apos; komutu geçerli bir komut değil</translation>
+        <translation>&apos;%1&apos; geçerli bir komut değil</translation>
     </message>
@@ -252,3 +252,3 @@
         <source>Field identifier &apos;%1&apos; occurred twice in request</source>
-        <translation>Alan tanıtıcı &apos;%1&apos; istekte iki defa geçiyor</translation>
+        <translation>İstekte, iki kere &apos;%1&apos; alan tanıtıcı oluştu</translation>
     </message>
@@ -366,3 +366,3 @@
         <source>Initialisation of Winsock failed. WSAStartup error code: %1</source>
-        <translation>Winsock kurulumu başarısız. WSAStartup hata kodu: %1</translation>
+        <translation>Winsock&apos;u başlangıç durumuna getirme işlemi başarısız. WSAStartup hata kodu: %1 </translation>
     </message>
@@ -371,3 +371,3 @@
         <source>Could not create listening socket. Socket error: %1</source>
-        <translation>Dinleme soketi yaratılamadı. Soket hatası: %1</translation>
+        <translation>Dinleme yuvası oluşturulamadı. Yuva hatası: %1</translation>
     </message>
@@ -376,3 +376,3 @@
         <source>Could not bind listening socket. Socket error: %1</source>
-        <translation>Dinleme soketine bağlantı yok. Soket hatası: %1</translation>
+        <translation>Dinleme yuvası bağlanamadı. Yuva hatası: %1</translation>
     </message>
@@ -381,3 +381,3 @@
         <source>Could not start listening on socket. Socket error: %1</source>
-        <translation>Soketten dinleme başlayamıyor. Soket hatası: %1</translation>
+        <translation>Yuva üzerinde dinleme başlatılamadı. Yuva hatası: %1</translation>
     </message>
@@ -386,3 +386,3 @@
         <source>Accept socket connection failed. Socket error: %1</source>
-        <translation>Soket bağlantısı kabul edilmedi. Soket hatası: %1</translation>
+        <translation>Alma yuvası bağlantısı başarısız. Yuva Hatası: %1</translation>
     </message>
@@ -391,3 +391,3 @@
         <source>Receiving data through socket failed. Socket error: %1</source>
-        <translation>Soket üzerinden veri alınamadı. Soket hatası: %1</translation>
+        <translation>Yuva üzerinden veri alımı başarısız. Yuva Hatası: %1</translation>
     </message>
@@ -396,3 +396,3 @@
         <source>Sending of data through socket failed. Socket error: %1</source>
-        <translation>Soket üzerinden veri gönderilemedi. Soket hatası:%1</translation>
+        <translation>Yuva üzerinden veri gönderimi başarısız. Yuva hatası: %1</translation>
     </message>
@@ -403,3 +403,3 @@
 Details: %1</source>
-        <translation>Last.fm medya çalarınızın eklentisine bağlanamadı.
+        <translation>Last.fm müzik çalar uygulanamanız için gereken eklentiye bağlanamadı.
 
@@ -412,3 +412,3 @@
 Details: %1</source>
-        <translation>Last.fm medya çalarınızın eklentisinden parçaları kabul etmeye başlayamadı.
+        <translation>Last.fm müzik çalar uygulanamanızdan parçaları kabul etme işlemini başlatamadı.
 
@@ -450,3 +450,3 @@
         <source>Sorry, there is not enough content to play this station. Please choose a different one.</source>
-        <translation>Üzgünüz, bu istasyon için yeterli içerik yok. Lütfen başka bir tane seçin.</translation>
+        <translation>Bu istasyonu çalabilmek için yeterli içerik yok. Lütfen başka bir tane seçin.</translation>
     </message>
@@ -455,3 +455,3 @@
         <source>This group does not have enough members for radio.</source>
-        <translation>Bu grupta radyo kuracak kadar üye yok.</translation>
+        <translation>Bu grubun radyo kuracak kadar üyesi yok.</translation>
     </message>
@@ -470,3 +470,3 @@
         <source>This station is available to subscribers only.&lt;p&gt;You can subscribe here: &lt;a href=&apos;http://www.last.fm/subscribe/&apos;&gt;http://www.last.fm/subscribe/&lt;/a&gt;</source>
-        <translation>Bu istasyon yalnızca abonelere açıktır.&lt;p&gt;Buradan abone olabilirsiniz: &lt;a href=&apos;http://www.last.fm/subscribe/&apos;&gt;http://www.last.fm/subscribe/&lt;/a&gt;</translation>
+        <translation>Bu istasyon sadece üyeler içindir.&lt;p&gt;Buradan üye olabilirsiniz:&lt;a href=&apos;http://www.last.fm/subscribe/&apos;&gt;http://www.last.fm/subscribe/&lt;/a&gt;</translation>
     </message>
@@ -475,3 +475,3 @@
         <source>There are not enough neighbours for this radio mode.</source>
-        <translation>Bu radyo modu için yeterli sayıda komşu yok.</translation>
+        <translation>Bu radyo kipi için yeterli komşu yok.</translation>
     </message>
@@ -480,3 +480,3 @@
         <source>The streaming system is offline for maintenance, please try again later.</source>
-        <translation>Netten yayın sistemi bakım çalışmaları için çevrimdışıdır, lütfen daha sonra tekrar deneyin.</translation>
+        <translation>İnternetten yayın sisteminde bakım çalışması var, lütfen sonra tekrar deneyin.</translation>
     </message>
@@ -493,3 +493,3 @@
         <source>Last.fm Setup</source>
-        <translation>Last.fm Kurulum</translation>
+        <translation>Last.fm Kurulumu</translation>
     </message>
@@ -500,3 +500,3 @@
 The social music revolution awaits! </source>
-        <translation>Last.fm&apos;i bilgisayarınıza yüklediniz. Şimdi sıra, kurulumda. Merak etmeyin, bu işlem uzun sürmeyecek ve yalnız bir kez yapacaksınız.
+        <translation>Last.fm&apos;i bilgisayarınıza yüklediniz. Şimdi sıra bilgisayarınıza kurmaya geldi. Merak etmeyin, bu işlem uzun sürmeyecek ve işlemi sadece bir kez yapacaksınız.
 
@@ -507,3 +507,3 @@
         <source>Click Continue to begin.</source>
-        <translation>Başlamak için Devam&apos;a tıklayın.</translation>
+        <translation>Başlamak için Devam Et düğmesine tıklayın.</translation>
     </message>
@@ -512,3 +512,3 @@
         <source>Click Next to begin.</source>
-        <translation>Başlamak için İleri&apos;ye tıklayın.</translation>
+        <translation>Başlamak için İleri düğmesine tıklayın.</translation>
     </message>
@@ -517,3 +517,3 @@
         <source>As this wizard installs file on your computer, you must be logged in as an Administrator to complete it. Please get an Administrator to run this for you.</source>
-        <translation>Sihirbaz dosyaları yüklerken bilgisayarda Yönetici olarak oturum açmış olmanız gerekiyor. Siz Yönetici değilseniz, bir Yöneticiye başvurarak yüklemeyi yapmasını isteyin.</translation>
+        <translation>Sihirbaz dosyaları bilgisayarınıza yüklerken Yönetici olarak oturum açmış olmanız gerekiyor. Eğer bu bilgisayarda yönetici değilseniz, bir yöneticiye başvurarak yüklemeyi yapmasını isteyin.</translation>
     </message>
@@ -536,3 +536,3 @@
 Click Next to continue.</source>
-        <translation>Last.fm şimdi bilgisayarınızdaki müzikçalarları inceleyecek ve skroplama için gereken eklentileri indirmeye başlayacak.
+        <translation>Last.fm şimdi bilgisayarınızdaki müzikçalarları inceleyecek ve otomatik bilgi gönderme (scrobbling) için gereken eklentileri indirmeye başlayacak.
 
@@ -555,3 +555,3 @@
         <source>Select Plugins</source>
-        <translation>Eklenti Seç</translation>
+        <translation>Eklentileri Seç</translation>
     </message>
@@ -605,3 +605,3 @@
 If you close your Last.fm window, you can easily access it from the icon in the menu bar.</source>
-        <translation>Last.fm kuruldu! Artık skroplamaya hazırsınız.
+        <translation>Last.fm kuruldu! Artık bilgi göndermeye hazırsınız.
 
@@ -614,3 +614,3 @@
 You can access Last.fm at any time by double-clicking the Last.fm user icon in the system tray.</source>
-        <translation>Last.fm kuruldu! Artık skroplamaya hazırsınız.
+        <translation>Last.fm kuruldu! Artık bilgi göndermeye hazırsınız.
 
@@ -653,3 +653,3 @@
 </source>
-        <translation>Alttaki müzikçalarlar şu anda açık görünüyor:
+        <translation>Alttaki müzikçalarlar şu anda çalışıyor durumda görünüyor:
 
@@ -669,3 +669,3 @@
         <source>Detected Running Player(s)</source>
-        <translation>Çalışır Durumda Müzikçalar Saptandı</translation>
+        <translation>Çalışan Durumdaki Müzikçalarlar</translation>
     </message>
@@ -679,3 +679,3 @@
         <source>Some of the running programs couldn&apos;t be shut down. Please close them manually.</source>
-        <translation>Çalışmakta olan programlardan bazısı kapatılamadı. Sizin kapatmanız gerekiyor.</translation>
+        <translation>Çalışmakta olan programlardan bazıları kapatılamadı. Sizin kapatmanız gerekiyor.</translation>
     </message>
@@ -684,3 +684,3 @@
         <source>Connection Problem</source>
-        <translation>Bağlantı arızası</translation>
+        <translation>Bağlantı Sorunu</translation>
     </message>
@@ -698,3 +698,3 @@
         <source>Download Error</source>
-        <translation>İndirmede arıza</translation>
+        <translation>İndirmede Hatası</translation>
     </message>
@@ -712,3 +712,3 @@
         <source>You&apos;ve connected your iPod with Last.fm running for the first time. Would you like to scrobble the tracks played on your iPod to a profile from now on?</source>
-        <translation>iPod&apos;unuzu ilk kez çalışırken Last.fm&apos;e bağladınız. iPod&apos;da çaldığınız parçalar bundan böyle profilinize skroplansın ister misiniz?</translation>
+        <translation>iPod&apos;unuz ile Last.fm&apos;e ilk defa bağlandınız. iPod&apos;unuzdaki çalınan parçalara ait bilgiler şimdi gönderilsin mi?</translation>
     </message>
@@ -724,3 +724,3 @@
         <source>Sorry, your iTunes listening history is possibly corrupted or too large to be imported into Last.fm. &lt;p&gt;Listen to music and your profile will build up quickly!</source>
-        <translation>Üzgünüz, iTunes çalma geçmişiniz muhtemelen bozulmuş veya Last.fm&apos;e alınamayacak kadar büyük. &lt;p&gt;Müzik dinlediğinizde profiliniz hemen oluşturulacaktır!</translation>
+        <translation>Üzgünüm, iTunes dinleme geçmişiniz bozuk ya da Last.fm&apos;e aktarmak için çok büyük. &lt;p&gt;Müzik dinleyin ve profiliniz çabucak oluşsun!</translation>
     </message>
@@ -729,3 +729,3 @@
         <source>Sorry, an error occurred while uploading your listening history. Please try again later.</source>
-        <translation>Üzgünüz, çalma geçmişinize yükleme yaparken bir hata oluştu. Lütfen daha sonra tekrar deneyin.</translation>
+        <translation> Üzgünüm, dinleme geçmişiniz gönderilirken bir hata oluştu. Lütfen daha sonra tekrar deneyin.</translation>
     </message>
@@ -734,3 +734,3 @@
         <source>Import your media player listening history</source>
-        <translation>Ortam çalar dinleme geçmişinizi aktarın</translation>
+        <translation>Müzik çalar dinleme geçmişini içeri aktar</translation>
     </message>
@@ -741,5 +741,5 @@
 This will add charts to your profile based on what you&apos;ve listened to in the past.</source>
-        <translation>Ortam çalar dinleme geçmişinizi aktarmak istiyor musunuz?
+        <translation>Müzik çalar dinleme geçmişinizi içeriye aktarmak istiyor musunuz?
 
-Bu işlem, geçmişte dinlediklerinize bağlı olarak profilinize listeler ekleyecektir.</translation>
+Bu işlem geçmişte dinlediklerinizi profilinize ekleyecek.</translation>
     </message>
@@ -748,3 +748,3 @@
         <source>Last.fm is now importing your listening history.</source>
-        <translation>Last.fm şu anda dinleme geçmişinizi aktarıyor.</translation>
+        <translation>Şimdi Last.fm dinleme geçmişinizi içeriye aktarıyor.</translation>
     </message>
@@ -757,3 +757,3 @@
 
-Aktarılmış ortam çalar kütüphaneniz birkaç dakika içinde profil sayfanızda belirecektir.</translation>
+İçeriye aktardığınız müzik çalar kütüphaneniz birkaç dakika içerisinde profil sayfanızda gösterilecektir.</translation>
     </message>
@@ -766,3 +766,3 @@
 
-Aktarılmış ortam çalar kütüphaneniz ortam çalar yeniden başlatıldığında aktarımı sona erdirecektir.</translation>
+Müzik çalar kütüphanesinin içeriye aktarılma işlemi müzik çalar yeniden başlatıldığında tamamlanacak.</translation>
     </message>
@@ -771,3 +771,3 @@
         <source>Media Player Restart Required</source>
-        <translation>Ortam Çaların Yeniden Başlatılması Gerekiyor</translation>
+        <translation>Müzik Çaların Yeniden Başlatılması Gerekiyor</translation>
     </message>
@@ -778,5 +778,5 @@
 Do you want Last.fm to restart it now?</source>
-        <translation>Dinleme geçmişiniz, ortam çalarınızı bir dahaki başlatışınızda aktarılacaktır.
+        <translation>Müzik çalarınızı yeniden başlattığınızda dinleme geçmişiniz içeri aktarılacak.
 
-Last.fm&apos;in şimdi yeniden başlatmasını istiyor musunuz?</translation>
+Last.fm uygulamasını yeniden başlatmak istiyor musunuz?</translation>
     </message>
@@ -785,3 +785,3 @@
         <source>Media Player Needs Starting</source>
-        <translation>Ortam Çaların Başlatılması Gerekiyor</translation>
+        <translation>Müzik Çaların Yeniden Başlatılması Gerekiyor</translation>
     </message>
@@ -792,5 +792,5 @@
 Do you want Last.fm to start it now?</source>
-        <translation>Dinleme geçmişiniz, ortam çalarınızı bir dahaki başlatışınızda aktarılacaktır.
+        <translation>Müzik çalarınızı yeniden başlattığınızda dinleme geçmişiniz içeri aktarılacak.
 
-Last.fm&apos;in şimdi yeniden başlatmasını istiyor musunuz?</translation>
+Last.fm uygulamasını yeniden başlatmak istiyor musunuz?</translation>
     </message>
@@ -802,3 +802,3 @@
         <source>Confirm</source>
-        <translation>Onay</translation>
+        <translation>Onayla</translation>
     </message>
@@ -832,3 +832,3 @@
         <source>Quit</source>
-        <translation>Kapat</translation>
+        <translation>Çık</translation>
     </message>
@@ -842,3 +842,3 @@
         <source>Are you sure you want to share &lt;b&gt;%1&lt;/b&gt; with &lt;b&gt;%2&lt;/b&gt;?</source>
-        <translation>&lt;b&gt;%1&lt;/b&gt; parçasını &lt;b&gt;%2&lt;/b&gt; ile paylaşmak istediğinizden emin misiniz?</translation>
+        <translation>&lt;b&gt;%1&lt;/b&gt; parçasını &lt;b&gt;%2&lt;/b&gt; kişisi ile paylaşmak istediğinizden emin misiniz?</translation>
     </message>
@@ -862,3 +862,3 @@
         <source>Are you sure you want to add &lt;b&gt;%1&lt;/b&gt; to your playlist?</source>
-        <translation>&lt;b&gt;%1&lt;/b&gt; parçasını müzik listenize eklemek istediğinizden emin misiniz?</translation>
+        <translation>&lt;b&gt;%1&lt;/b&gt; parçasını parça listenize eklemek istediğinizden emin misiniz?</translation>
     </message>
@@ -872,3 +872,3 @@
         <source>Are you sure you want to add &lt;b&gt;%1&lt;/b&gt; to your loved tracks?</source>
-        <translation>&lt;b&gt;%1&lt;/b&gt; parçasını sevdiğiniz parçalara eklemek istediğinizden emin misiniz?</translation>
+        <translation>&lt;b&gt;%1&lt;/b&gt; parçasını sevdiğiniz parçalar listesine eklemek istediğinizden emin misiniz?</translation>
     </message>
@@ -882,3 +882,3 @@
         <source>Are you sure you want to ban &lt;b&gt;%1&lt;/b&gt;? This means the radio will never play this track again.</source>
-        <translation>&lt;b&gt;%1&lt;/b&gt; parçasını engellemek istediğinizden emin misiniz? Bu, radyonun bu parçayı bir daha asla çalmayacağı anlamına gelir.</translation>
+        <translation>&lt;b&gt;%1&lt;/b&gt; parçasını engellemek istediğinizden emin misiniz? Bu radyonuzun bu parçayı bir daha asla çalmayacağı anlamına geliyor.</translation>
     </message>
@@ -892,3 +892,3 @@
         <source>Really quit Last.fm? Any music you listen to will not be scrobbled to your profile.</source>
-        <translation>Last.fm&apos;den kesin çıkıyor musunuz? O zaman, dinlediğiniz müzikler profilinize skroplanmaz.</translation>
+        <translation>Last.fm&apos;den gerçekten çıkıyor musunuz? Eğer çıkarsanız dinledikleriniz profilinize kaydedilmez.</translation>
     </message>
@@ -900,5 +900,5 @@
 </source>
-        <translation>Uygulama geri planda çalışmaya devam edecektir. Başlat Menüsünden başlatarak uygulamaya yeniden erişebilirsiniz.
+        <translation>Uygulama arkaplanda çalışmaya devam edecek. Uygulamaya yeniden ulaşmak için uygulamayı çalıştırdığınız menüyü kullanın.
 
-Kalıcı olarak kapatmak için Dosya &gt; Çık seçeneğini kullanın.
+Uygulamayı kalıcı olarak kapatmak için Dosya &gt; Çık yolunu kullanın.
 </translation>
@@ -911,6 +911,5 @@
 </source>
-        <translation>Uygulama geri planda çalışmaya devam edecektir. Uygulamalar klasöründen başlatarak uygulamaya yeniden erişebilirsiniz.
+        <translation>Uygulama arkaplanda çalışmaya devam edecek. Uygulamaya yeniden ulaşmak için uygulamayı çalıştırdığınız menüyü kullanın.
 
-Kalıcı olarak kapatmak için uygulama menüsünden Quit seçeneğini seçin.
-</translation>
+Uygulamayı kalıcı olarak kapatmak için Dosya &gt; Çık yolunu kullanın.</translation>
     </message>
@@ -919,3 +918,3 @@
         <source>OK</source>
-        <translation>Tamam</translation>
+        <translation>TAMAM</translation>
     </message>
@@ -924,3 +923,3 @@
         <source>Don&apos;t tell me again</source>
-        <translation>Bir daha söyleme</translation>
+        <translation>Tekrar uyarma</translation>
     </message>
@@ -1110,3 +1109,3 @@
         <source>Connection Problem</source>
-        <translation>Bağlantı Arızası</translation>
+        <translation>Bağlantı Sorunu</translation>
     </message>
@@ -1190,3 +1189,3 @@
         <source>Tagging %1 failed</source>
-        <translation>%1 etiketlenemedi</translation>
+        <translation>%1 etiketleme başarısız oldu</translation>
     </message>
@@ -1195,3 +1194,3 @@
         <source>You tagged %1 successfully</source>
-        <translation>%1 etiketlendi</translation>
+        <translation>%1 başarılı şekilde etiketlendi</translation>
     </message>
@@ -1233,3 +1232,3 @@
     <message>
-        <location filename="../src/libMoose/MooseCommon.cpp" line="139"/>
+        <location filename="../src/container.cpp" line="1173"/>
         <source>Error</source>
@@ -1300,3 +1299,3 @@
         <source>&amp;Quit</source>
-        <translation>&amp;Durdur</translation>
+        <translation>&amp;Çık</translation>
     </message>
@@ -1346,3 +1345,3 @@
         <source>Your username and password are incorrect</source>
-        <translation>Kullanıcı adı ve parola yanlış</translation>
+        <translation>Kullanıcı adınız ve parolanız yanlış</translation>
     </message>
@@ -1380,3 +1379,3 @@
         <source>This software is too old, please upgrade.</source>
-        <translation>Bu yazılım fazla eski, lütfen yükseltin.</translation>
+        <translation>Bu yazılım çok eski, lütfen güncelleyin.</translation>
     </message>
@@ -1390,3 +1389,3 @@
         <source>scrobbled</source>
-        <translation>skroplandı</translation>
+        <translation>bilgileri gönderildi</translation>
     </message>
@@ -1400,3 +1399,3 @@
         <source>Contacting radio service...</source>
-        <translation>Radyo servisine bağlanıyor...</translation>
+        <translation>Radyo servisi ile bağlantı kuruluyor...</translation>
     </message>
@@ -1405,3 +1404,3 @@
         <source>Radio service initialised</source>
-        <translation>Radyo servisi başlatıldı</translation>
+        <translation>Radyo servisi algılandı</translation>
     </message>
@@ -1415,3 +1414,3 @@
         <source>Connecting to playlist...</source>
-        <translation>Müzik listesi alınıyor...Müzik listesine bağlanıyor...</translation>
+        <translation>Parça listesine bağlanılıyor...</translation>
     </message>
@@ -1425,3 +1424,3 @@
         <source>Retrieving playlist...</source>
-        <translation>Müzik listesi alınıyor...</translation>
+        <translation>Parça listesi alınıyor...</translation>
     </message>
@@ -1430,3 +1429,3 @@
         <source>Retrieving stream...</source>
-        <translation>Netten yayın alınıyor...</translation>
+        <translation>Yayın alınıyor...</translation>
     </message>
@@ -1440,3 +1439,3 @@
         <source>Sorry</source>
-        <translation>Üzgünüz</translation>
+        <translation>Üzgünüm</translation>
     </message>
@@ -1450,3 +1449,9 @@
         <source>Couldn&apos;t load service: %1. The application won&apos;t be able to start.</source>
-        <translation>Servis yüklenemedi: %1. Bu uygulama başlayamayacak.</translation>
+        <translation>Servis yüklenemedi: %1. Uygulama başlatılamadı.</translation>
+    </message>
+    <message>
+        <location filename="../src/container.cpp" line="1129"/>
+        <source>Buffering... (%1%)</source>
+        <comment>%1 is the percentage buffering is complete</comment>
+        <translation>Tampon Belleğe Alınıyor... (%1%)</translation>
     </message>
@@ -1455,3 +1460,3 @@
         <source>Starting station %1...</source>
-        <translation>Açılan istasyon %1...</translation>
+        <translation>İstasyon başlatılıyor %1...</translation>
     </message>
@@ -1460,3 +1465,3 @@
         <source>Starting station...</source>
-        <translation>Açılan istasyon...</translation>
+        <translation>İstasyon başlatılıyor...</translation>
     </message>
@@ -1510,3 +1515,3 @@
         <source>This %1 has been shared with %2.</source>
-        <translation>%1 %2 ile paylaşıldı.</translation>
+        <translation>Bu %1 %2 kişisi ile paylaşıldı.</translation>
     </message>
@@ -1515,3 +1520,3 @@
         <source>This track has been added to your playlist.</source>
-        <translation>Bu parça müzik listenize eklendi.</translation>
+        <translation>Bu parça parça listenize eklendi.</translation>
     </message>
@@ -1535,3 +1540,3 @@
         <source>Quit Last.fm</source>
-        <translation>Last.fm&apos;den çık</translation>
+        <translation>Last.fm Uygulamasından Çık</translation>
     </message>
@@ -1541,4 +1546,4 @@
         <translation>
-            <numerusform>%n skroplanmış parça sunuyor...</numerusform>
-            <numerusform>%n skroplanmış parça sunuyor...</numerusform>
+            <numerusform>%n Bilgi gönderimi yapılıyor...</numerusform>
+            <numerusform></numerusform>
         </translation>
@@ -1549,4 +1554,4 @@
         <translation>
-            <numerusform>%n skroplanmış parça sunuldu</numerusform>
-            <numerusform>%n skroplanmış parça sunuldu</numerusform>
+            <numerusform>%n bilgi gönderimi yapıldı</numerusform>
+            <numerusform></numerusform>
         </translation>
@@ -1557,4 +1562,4 @@
         <translation>
-            <numerusform>%n skroplanmış parça daha sonra sunulacak</numerusform>
-            <numerusform>%n skroplanmış parça daha sonra sunulacak</numerusform>
+            <numerusform>%n bilgi gönderimi daha sonra yapılacak</numerusform>
+            <numerusform></numerusform>
         </translation>
@@ -1564,3 +1569,3 @@
         <source>Where is your iPod mounted?</source>
-        <translation>iPod&apos;unuz nereye takılı?</translation>
+        <translation>iPod cihazınız nereye bağlı?</translation>
     </message>
@@ -1579,3 +1584,3 @@
         <source>There was an error loading the IpodDevice plugin.</source>
-        <translation>IpodDevice eklentisini yüklerken bir hata meydana geldi.</translation>
+        <translation>IpodDevice eklentisi yüklenirken bir hata oluştu.</translation>
     </message>
@@ -1584,3 +1589,3 @@
         <source>Bad Version</source>
-        <translation>Kötü Sürüm</translation>
+        <translation>Uygunsuz Sürüm</translation>
     </message>
@@ -1594,3 +1599,3 @@
         <source>&lt;p&gt;Last.fm cannot authorise any scrobbling! :(&lt;p&gt;It appears your computer disagrees with us about what the time is.&lt;p&gt;If you are sure the time is right, check the date is correct and check your timezone is not set to something miles away, like Mars.&lt;p&gt;We&apos;re sorry about this restriction, but we impose it to help prevent scrobble spamming.</source>
-        <translation>&lt;p&gt;Last.fm skroplamaya izin vermiyor! :(&lt;p&gt;Bilgisayarınız saatin kaç olduğu konusunda bizimle aynı fikirde değil gibi görünüyor.&lt;p&gt;Saatin doğru olduğundan eminseniz, tarihin doğru olduğunu kontrol edin ve saat diliminin Mars gibi kilometrelerce uzakta bir yer olmadığından emin olun.&lt;p&gt;Bu kısıtlamadan dolayı üzgünüz, ancak bunu istenmeyen skroplamaları önlemek için yapıyoruz.</translation>
+        <translation>&lt;p&gt;Last.fm herhangi bir gönderim için yetkilendirilemedi! :( &lt;p&gt;Bilgisayarınız saatin kaç olduğu konusunda bizimle aynı fikirde değil gibi görünüyor.&lt;p&gt;Saatin doğru olduğundan eminseniz, tarihin ve zaman diliminin doğru olduğundan emin olun, Mars&apos;a göre ayarlanmış olmasın.&lt;p&gt;Bu kısıtlamadan dolayı üzgünüz ama  bunu istenmeyen (spam) gönderimleri engellemek için düşündük.</translation>
     </message>
@@ -1599,3 +1604,3 @@
         <source>Nothing to Scrobble</source>
-        <translation>Skroplanacak Bir Şey Yok</translation>
+        <translation>Gönderilecek Birşey Yok</translation>
     </message>
@@ -1604,3 +1609,3 @@
         <source>You have not played anything since you last scrobbled this iPod.</source>
-        <translation>Bu iPod&apos;u en son skroplayışınızdan bu yana hiçbir şey çalmadınız.</translation>
+        <translation>Bu iPod ile son gönderim yaptığınızdan beri birşey dinlememişsiniz.</translation>
     </message>
@@ -1860,3 +1865,3 @@
         <source>Last.fm Crash Reporter</source>
-        <translation>Last.fm Çökme Bildirimi</translation>
+        <translation>Last.fm Çökme Bildirici</translation>
     </message>
@@ -1865,3 +1870,3 @@
         <source>&lt;p&gt;&lt;b&gt;Sorry!&lt;/b&gt;&amp;nbsp;Last.fm crashed. Information about the crash is now being sent to Last.fm so that we can fix the bug.&lt;/p&gt;</source>
-        <translation>&lt;p&gt;&lt;b&gt;Üzgünüz!&lt;/b&gt;&amp;nbsp;Last.fm çöktü. Sorunu çözebilmemiz için çökme bilgileri şu anda Last.fm&apos;e gönderiliyor.&lt;/p&gt;</translation>
+        <translation>&lt;p&gt;&lt;b&gt;Üzgünüm!&lt;/b&gt;&amp;nbsp;Last.fm çöktü. Hatayı düzeltebilmemiz için gerelen bilgiler Last.fm&apos;e gönderiliyor.&lt;/p&gt;</translation>
     </message>
@@ -1870,3 +1875,3 @@
         <source>Abort</source>
-        <translation>İptal et</translation>
+        <translation>İptal</translation>
     </message>
@@ -1875,3 +1880,13 @@
         <source>You can disable sending crash reports in the configuration dialog.</source>
-        <translation>Çökme bildirimlerinin gönderilmesini yapılandırma diyaloğunda devre dışı bırakabilirsiniz.</translation>
+        <translation>Yapılandırma arayüzünden çökme bilgilerinin gönderimini kapatabilirsiniz.</translation>
+    </message>
+    <message>
+        <location filename="../src/CrashReporter/CrashReporter.cpp" line="155"/>
+        <source>Uploaded %L1 of %L2 KB.</source>
+        <translation>%L1 verinin %L2 KB kadarı gönderildi.</translation>
+    </message>
+    <message>
+        <location filename="../src/CrashReporter/CrashReporter.cpp" line="187"/>
+        <source>Close</source>
+        <translation>Kapat</translation>
     </message>
@@ -1882,2 +1897,7 @@
     </message>
+    <message>
+        <location filename="../src/CrashReporter/CrashReporter.cpp" line="188"/>
+        <source>Failed to send crash info.</source>
+        <translation>Çökme bilgileri gönderilemedi.</translation>
+    </message>
 </context>
@@ -1978,5 +1998,5 @@
 Do you want to delete user %1?</source>
-        <translation>Bu, kullanıcıyı uygulamadan çıkarır, profili Last.fm&apos;den silmez.
+        <translation>Bu işlem kullanıcıyı uygulamadan siler, Last.fm profili silinmez.
 
-%1 kullanıcısını silmek ister misiniz?</translation>
+%1 kullanıcısını silmek istiyor musunuz?</translation>
     </message>
@@ -1985,3 +2005,3 @@
         <source>Select a user to delete:</source>
-        <translation>Silinecek kullanıcıyı seçin:</translation>
+        <translation>Silmek için bir kullanıcı seç:</translation>
     </message>
@@ -2003,3 +2023,3 @@
         <source>OK</source>
-        <translation>Tamam</translation>
+        <translation>TAMAM</translation>
     </message>
@@ -2015,3 +2035,3 @@
             <numerusform>%n önbelleğe alınmış parça</numerusform>
-            <numerusform>%n önbelleğe alınmış parça</numerusform>
+            <numerusform></numerusform>
         </translation>
@@ -2026,3 +2046,3 @@
         <source>Submission Server: </source>
-        <translation>Sunum Sürücüsü: </translation>
+        <translation>Gönderme Sunucusu: </translation>
     </message>
@@ -2031,3 +2051,3 @@
         <source>Last successful connection: </source>
-        <translation>En son başarılı bağlantı: </translation>
+        <translation>Son başarılı bağlantı: </translation>
     </message>
@@ -2041,3 +2061,3 @@
         <source>Diagnostics</source>
-        <translation>Tanı</translation>
+        <translation>Uygulamayı İzle</translation>
     </message>
@@ -2046,3 +2066,3 @@
         <source>Scrobbling</source>
-        <translation>Skroplama</translation>
+        <translation>Gönderim</translation>
     </message>
@@ -2056,3 +2076,3 @@
         <source>Submission server status:</source>
-        <translation>Sunum sunucusu durumu:</translation>
+        <translation>Gönderme sunucusu durumu:</translation>
     </message>
@@ -2066,3 +2086,3 @@
         <source>Last successful connection:</source>
-        <translation>En son başarılı bağlantı:</translation>
+        <translation>Son başarılı bağlantı:</translation>
     </message>
@@ -2111,3 +2131,3 @@
         <source>Output buffer:</source>
-        <translation>Çıkış arabelleği:</translation>
+        <translation>Çıktı tamponu:</translation>
     </message>
@@ -2116,3 +2136,3 @@
         <source>Http buffer:</source>
-        <translation>Http arabelleği:</translation>
+        <translation>Http tamponu:</translation>
     </message>
@@ -2121,3 +2141,3 @@
         <source>Decoded buffer:</source>
-        <translation>Deşifre edilmiş arabellek:</translation>
+        <translation>Çözülmüş tampon:</translation>
     </message>
@@ -2131,3 +2151,3 @@
         <source>Tracks in queue:</source>
-        <translation>Sıradaki parçalar:</translation>
+        <translation>Kuyruktaki parçalar:</translation>
     </message>
@@ -2136,3 +2156,3 @@
         <source>Currently fingerprinting:</source>
-        <translation>Şu anda parmak izi alınan:</translation>
+        <translation>Şimdi parmak izi alınan:</translation>
     </message>
@@ -2141,3 +2161,3 @@
         <source>Logs</source>
-        <translation>Kayıtlar</translation>
+        <translation>Günlük Kayıtları</translation>
     </message>
@@ -2146,3 +2166,3 @@
         <source>View Log</source>
-        <translation>Kaydı Görüntüle</translation>
+        <translation>Günlük Kayıtlarını Göster</translation>
     </message>
@@ -2151,3 +2171,3 @@
         <source>View Helper Log</source>
-        <translation>Yardımcı Kaydını Görüntüle</translation>
+        <translation>Yardımcı Günlük Kayıtlarını Göster</translation>
     </message>
@@ -2171,3 +2191,3 @@
         <source>Ctrl+C</source>
-        <translation>Ctrl+C</translation>
+        <translation>Ctrl+K</translation>
     </message>
@@ -2181,3 +2201,3 @@
         <source>Fingerprinting</source>
-        <translation>Parmak izi alma</translation>
+        <translation>Parmakizi Alımı</translation>
     </message>
@@ -2186,3 +2206,3 @@
         <source>Recently fingerprinted tracks</source>
-        <translation>Son parmak izi alınmış parçalar</translation>
+        <translation>En son parmakizi alınan parçalar</translation>
     </message>
@@ -2317,3 +2337,3 @@
         <source>The track has an empty path</source>
-        <translation>Parçada boş bir yol var</translation>
+        <translation>Parçanın yolu boş</translation>
     </message>
@@ -2322,3 +2342,3 @@
         <source>The track is not an mp3</source>
-        <translation>Parça mp3 değil</translation>
+        <translation>Parça bir mp3 değil</translation>
     </message>
@@ -2327,3 +2347,3 @@
         <source>The track does not exist!</source>
-        <translation>Parça mevcut değil!</translation>
+        <translation>Parça yok!</translation>
     </message>
@@ -2337,3 +2357,3 @@
         <source>The track is too short</source>
-        <translation>Parça fazla kısa</translation>
+        <translation>Parça çok kısa</translation>
     </message>
@@ -2342,3 +2362,3 @@
         <source>Fingerprinting failed, skipping.</source>
-        <translation>Parmak izi alma başarısız, atlanıyor.</translation>
+        <translation>Parmak izi başarısız, atlanıyor.</translation>
     </message>
@@ -2347,3 +2367,3 @@
         <source>Getting bad request with this track, skipping.</source>
-        <translation>Bu parçayla kötü istek alınıyor, atlanıyor.</translation>
+        <translation>Bu parçaya uygunsuz istek, atlanıyor.</translation>
     </message>
@@ -2360,3 +2380,3 @@
         <source>The track has an empty path</source>
-        <translation>Parçada boş bir yol var</translation>
+        <translation>Parçanın yolu boş</translation>
     </message>
@@ -2365,3 +2385,3 @@
         <source>The track is not an mp3</source>
-        <translation>Parça mp3 değil</translation>
+        <translation>Parça bir mp3 değil</translation>
     </message>
@@ -2370,3 +2390,3 @@
         <source>The track does not exist!</source>
-        <translation>Parça mevcut değil!</translation>
+        <translation>Parça yok!</translation>
     </message>
@@ -2375,3 +2395,3 @@
         <source>The track is not readable</source>
-        <translation>Parça okunabilir değil</translation>
+        <translation>Parça yerel değil</translation>
     </message>
@@ -2380,3 +2400,3 @@
         <source>The track is too short</source>
-        <translation>Parça fazla kısa</translation>
+        <translation>Parça çok kısa</translation>
     </message>
@@ -2385,3 +2405,3 @@
         <source>Fingerprinting failed.</source>
-        <translation>Parmak izi alma başarısız.</translation>
+        <translation>Parmak izi başarısız.</translation>
     </message>
@@ -2393,3 +2413,3 @@
         <source>Invalid session. Please re-handshake.</source>
-        <translation>Geçersiz oturum. Lütfen tekrar tokalaşın.</translation>
+        <translation>Geçersiz oturum. Lütfen yeniden bağlantı kurun.</translation>
     </message>
@@ -2399,3 +2419,3 @@
 Please try again later.</source>
-        <translation>Üzgünüz, müzik listesi servisi yanıt vermiyor.
+        <translation>Üzgünüm, parça listesi servisi yanıt vermiyor.
 Lütfen daha sonra tekrar deneyin.</translation>
@@ -2431,3 +2451,3 @@
         <source>Could not connect to server.</source>
-        <translation>Sunucuya bağlanamadı.</translation>
+        <translation>Sunucuya bağlanılamadı.</translation>
     </message>
@@ -2436,3 +2456,3 @@
         <source>This client version is obsolete. Please update.</source>
-        <translation>Bu istemci sürümü eski. Lütfen güncelleyin.</translation>
+        <translation>İstemci sürümü çok eski. Lütfen güncelleyin.</translation>
     </message>
@@ -2449,3 +2469,3 @@
         <source>There was a problem contacting the radio streamer. Please try again later.</source>
-        <translation>Radyo yayıncısına bağlanırken sorun meydana geldi. Lütfen daha sonra tekrar deneyin.</translation>
+        <translation>Radyo yayını ile bağlantı kurulurken bir sorun oluştu. Lütfen daha sonra tekrar deneyin.</translation>
     </message>
@@ -2454,3 +2474,3 @@
         <source>Invalid authorisation.</source>
-        <translation>Geçersiz izin.</translation>
+        <translation>Geçersiz yetkilendirme.</translation>
     </message>
@@ -2459,3 +2479,3 @@
         <source>This stream is currently not available. Please try again later.</source>
-        <translation>Bu yayın şu anda mevcut değil. Lütfen daha sonra tekrar deneyin.</translation>
+        <translation>Bu yayın şu anda kullanılamıyor. Lütfen daha sonra tekrar deneyin.</translation>
     </message>
@@ -2485,3 +2505,3 @@
         <source>Nothing To Scrobble</source>
-        <translation>Skroplanacak Bir Şey Yok</translation>
+        <translation>Gönderilecek Birşey Yok</translation>
     </message>
@@ -2490,3 +2510,3 @@
         <source>Your iPod has nothing new to scrobble.</source>
-        <translation>iPod&apos;unuzda skroplanacak yeni bir şey yok.</translation>
+        <translation>iPod cihazınızda gönderilecek yeni birşey yok.</translation>
     </message>
@@ -2500,3 +2520,3 @@
         <source>Unsupported OS Version</source>
-        <translation>Desteklenmeyen OS Sürümü</translation>
+        <translation>Desteklenmeyen İşletim Sistemi Sürümü</translation>
     </message>
@@ -2505,3 +2525,3 @@
         <source>We are sorry, but Last.fm requires OS X version 10.4 (Tiger) or above.</source>
-        <translation>Üzgünüz, ancak Last.fm için OS X sürümü 10.4 (Tiger) veya yukarısı gerekiyor.</translation>
+        <translation>Üzgünüm, Last.fm OS X 10.4 (Tiger) ve daha üst sürüm gerektiriyor.</translation>
     </message>
@@ -2691,3 +2711,3 @@
         <source>Enable S&amp;crobbling</source>
-        <translation>Skr&amp;oplama Devrede</translation>
+        <translation>&amp;Gönderimi Etkinleştir</translation>
     </message>
@@ -2716,3 +2736,3 @@
         <source>Share what&apos;s currently playing with someone</source>
-        <translation>Çalmakta olan parçayı birisiyle paylaş</translation>
+        <translation>Şimdi dinlediğin parçayı birisiyle paylaş</translation>
     </message>
@@ -2721,3 +2741,3 @@
         <source>Ctrl+S</source>
-        <translation>Ctrl+S</translation>
+        <translation>CTRL+S</translation>
     </message>
@@ -2726,3 +2746,3 @@
         <source>&amp;Love</source>
-        <translation>Se&amp;vdim</translation>
+        <translation>&amp;Sevdim</translation>
     </message>
@@ -2741,3 +2761,3 @@
         <source>&amp;Ban</source>
-        <translation>E&amp;ngelle</translation>
+        <translation>&amp;Engelle</translation>
     </message>
@@ -2746,3 +2766,3 @@
         <source>Don&apos;t ever play me this again</source>
-        <translation>Bunu bir daha çalma</translation>
+        <translation>Bunu bir daha bana çalma</translation>
     </message>
@@ -2751,3 +2771,3 @@
         <source>Ctrl+B</source>
-        <translation>Ctrl+B</translation>
+        <translation>Ctrl+E</translation>
     </message>
@@ -2771,3 +2791,3 @@
         <source>Enable &amp;Discovery Mode</source>
-        <translation>&amp;Keşif Modu Devrede</translation>
+        <translation>&amp;Keşif Kipini Etkinleştir</translation>
     </message>
@@ -2786,3 +2806,3 @@
         <source>Volume &amp;Down</source>
-        <translation>Ses Düzeyi &amp;Aşağı</translation>
+        <translation>Sesi A&amp;zalt</translation>
     </message>
@@ -2796,3 +2816,3 @@
         <source>&amp;Mute</source>
-        <translation>&amp;Sessiz</translation>
+        <translation>Sesi &amp;Kapat</translation>
     </message>
@@ -2801,3 +2821,3 @@
         <source>Ctrl+M</source>
-        <translation>Ctrl+M</translation>
+        <translation>Ctrl+K</translation>
     </message>
@@ -2806,3 +2826,3 @@
         <source>Volume &amp;Up</source>
-        <translation>Ses Düzeyi &amp;Yukarı</translation>
+        <translation>Sesi Y&amp;ükselt</translation>
     </message>
@@ -2831,3 +2851,3 @@
         <source>Stop radio</source>
-        <translation>Radyoyu kapat</translation>
+        <translation>Radyoyu durdur</translation>
     </message>
@@ -2841,3 +2861,3 @@
         <source>Invite a &amp;Friend...</source>
-        <translation>Bir A&amp;rkadaş davet et...</translation>
+        <translation>Bir Arkadaş &amp;Davet Et...</translation>
     </message>
@@ -2851,3 +2871,3 @@
         <source>Resume last radio station</source>
-        <translation>Son radyo istasyonuna devam et</translation>
+        <translation>Son radyoyu yeniden yükle</translation>
     </message>
@@ -2856,3 +2876,3 @@
         <source>Ctrl+Space</source>
-        <translation>Ctrl+Boşluk</translation>
+        <translation>Ctrl+Space</translation>
     </message>
@@ -2861,3 +2881,3 @@
         <source>S&amp;kip</source>
-        <translation>G&amp;eç</translation>
+        <translation>A&amp;tla</translation>
     </message>
@@ -2866,3 +2886,3 @@
         <source>Skip this track</source>
-        <translation>Bu parçayı geç</translation>
+        <translation>Bu parçayı atla</translation>
     </message>
@@ -2876,3 +2896,3 @@
         <source>&amp;Diagnostics</source>
-        <translation>&amp;Tanı</translation>
+        <translation>&amp;Uygulamayı İzle</translation>
     </message>
@@ -2881,3 +2901,3 @@
         <source>Add To My &amp;Playlist</source>
-        <translation>&amp;Müzik Listeme Ekle</translation>
+        <translation>&amp;Parça Listeme Ekle</translation>
     </message>
@@ -2886,3 +2906,3 @@
         <source>Playlist</source>
-        <translation>Müzik listesi</translation>
+        <translation>Parça Listesi</translation>
     </message>
@@ -2891,3 +2911,3 @@
         <source>Add what&apos;s currently playing to your Last.fm Playlist</source>
-        <translation>Çalmakta olan parçayı Last.fm Müzik listenize ekleyin</translation>
+        <translation>Şimdi çalan parçayı Last.fm Parça Listesine Ekle</translation>
     </message>
@@ -2901,3 +2921,3 @@
         <source>&amp;Send to Tray</source>
-        <translation>Görev Çubuğuna &amp;Yolla</translation>
+        <translation>Sistem Çekmecesine &amp;Gönder</translation>
     </message>
@@ -2916,3 +2936,3 @@
         <source>Scrobble iPod...</source>
-        <translation>iPod&apos;u Skropla...</translation>
+        <translation>iPod&apos;dan şarkı bilgilerini Gönder...</translation>
     </message>
@@ -2944,3 +2964,3 @@
         <source>Plays</source>
-        <translation>Çalma sayısı</translation>
+        <translation>Çalınma sayısı</translation>
     </message>
@@ -2988,3 +3008,3 @@
         <source>&amp;Scrobble Selected</source>
-        <translation>&amp;Seçilenleri Skropla</translation>
+        <translation>&amp;Gönderim Seçildi</translation>
     </message>
@@ -2993,3 +3013,3 @@
         <source>Scrobble None</source>
-        <translation>Hiçbirini Skroplama</translation>
+        <translation>Gönderim Kapalı</translation>
     </message>
@@ -3006,3 +3026,3 @@
         <source>This iPod scrobbles to %1&apos;s profile!</source>
-        <translation>Bu iPod %1&apos;in profiline skroplar!</translation>
+        <translation>Bu iPod %1 kişisinin profiline gönderim yapar!</translation>
     </message>
@@ -3012,4 +3032,4 @@
         <translation>
-            <numerusform>&lt;p&gt;Last.fm, iPod&apos;unuzu son senkronize edişinizden bu yana %n adet yeni parça buldu.&lt;br/&gt;Lütfen iPod&apos;unuzdan skroplanmasını istediğiniz parçalarını seçin.</numerusform>
-            <numerusform>&lt;p&gt;Last.fm, iPod&apos;unuzu son senkronize edişinizden bu yana %n adet yeni parça buldu.&lt;br/&gt;Lütfen iPod&apos;unuzdan skroplanmasını istediğiniz parçalarını seçin.</numerusform>
+            <numerusform>&lt;p&gt;Last.fm iPod cihazınız ile yapılan son eşzamanlamadan sonra eklenen %n yeni parça buldu .&lt;br/&gt;Lütfen iPod cihazınızdan gönderimini yapmak istediğiniz parçaları seçin.</numerusform>
+            <numerusform></numerusform>
         </translation>
@@ -3037,3 +3057,3 @@
         <source>&lt;b&gt;Note&lt;/b&gt; that iTunes must be set to automatically sync the iPod for this to work.</source>
-        <translation>Bunun çalışması için iTunes&apos;un iPod&apos;u otomatikman senkronize etmek üzere ayarlanmış olması gerektiğini &lt;b&gt;unutmayın&lt;/b&gt;.</translation>
+        <translation>Bunun çalışması için, iTunes ile iPod&apos;un otomatik olarak eş zamanlı ayarlanması gerektiğini &lt;b&gt;unutmayın&lt;/b&gt;.</translation>
     </message>
@@ -3042,3 +3062,3 @@
         <source>Who is the owner of this iPod?</source>
-        <translation>Bu iPod&apos;un sahibi kim?</translation>
+        <translation>Bu iPod&apos;un sahibi kimdir?</translation>
     </message>
@@ -3300,3 +3320,3 @@
         <source>Starting station...</source>
-        <translation>İstasyon açılıyor...</translation>
+        <translation>İstasyon başlatılıyor...</translation>
     </message>
@@ -3306,4 +3326,4 @@
 or tune in to free radio</source>
-        <translation>Ortam çalarınızda dinleyin
-ya da ücretsiz radyodan bulun</translation>
+        <translation>Kendi müzik çalarınızda dinlemeye başlayın
+ya da radyomuzu dinleyin</translation>
     </message>
@@ -3312,3 +3332,3 @@
         <source>Starting station (any second now)...</source>
-        <translation>İstasyon açılıyor (her an açılabilir)...</translation>
+        <translation>İstasyon başlatılıyor (herhangi bir saniyeden)...</translation>
     </message>
@@ -3317,3 +3337,3 @@
         <source>Drag to tag/share %1</source>
-        <translation>%1 parçasını etiketlemek/paylaşmak için sürükleyin</translation>
+        <translation>%1 parçasını etiketlemek/paylaşmak için sürükle</translation>
     </message>
@@ -3322,3 +3342,3 @@
         <source>Drag to tag/share &quot;%1&quot;</source>
-        <translation>&quot;%1&quot; parçasını etiketlemek/paylaşmak için sürükleyin</translation>
+        <translation>&quot;%1&quot; parçasını etiketlemek/paylaşmak için sürükle</translation>
     </message>
@@ -3332,3 +3352,3 @@
         <source>%L1 plays scrobbled on Last.fm</source>
-        <translation>%L1, Last.fm&apos;de skroplanmış çalıyor</translation>
+        <translation>%L1 kere Last.fm&apos;e gönderilmiş</translation>
     </message>
@@ -3337,3 +3357,3 @@
         <source> on</source>
-        <translation>  açık</translation>
+        <translation> açık</translation>
     </message>
@@ -3347,3 +3367,3 @@
         <source>We don&apos;t have a description for this artist yet, care to help?</source>
-        <translation>Bu sanatçıyla ilgili açıklama yok, yardım etmek ister misiniz?</translation>
+        <translation>Bu sanatçıyla ilgili açıklama yok. Siz yazmak ister misiniz?</translation>
     </message>
@@ -3362,3 +3382,3 @@
         <source>TextLabel</source>
-        <translation>TextLabel</translation>
+        <translation>MetinEtiketi</translation>
     </message>
@@ -3372,3 +3392,3 @@
         <source>Released:</source>
-        <translation>Çıkış tarihi:</translation>
+        <translation>Yayın Tarihi:</translation>
     </message>
@@ -3382,3 +3402,3 @@
         <source>by</source>
-        <translation>Sanatçı</translation>
+        <translation>söyleyen</translation>
     </message>
@@ -3416,3 +3436,3 @@
         <source>Last.fm is importing your current media library...</source>
-        <translation>Last.fm geçerli ortam kütüphanenizi aktarıyor...</translation>
+        <translation>Last.fm geçerli müzik kütüphanenizi içeriye aktarıyor...</translation>
     </message>
@@ -3423,5 +3443,5 @@
  Click OK to continue.</source>
-        <translation>Last.fm ortam kütüphanenizi aktardı.
+        <translation>Last.fm müzik kütüphanenizi içeriye aktardı.
 
- Devam etmek için Tamam&apos;a tıklayın.</translation>
+ Devam etmek için TAMAM&apos;a tıklayın.</translation>
     </message>
@@ -3430,3 +3450,3 @@
         <source>Last.fm Library Import</source>
-        <translation>Last.fm Kütüphane Aktarımı</translation>
+        <translation>Last.fm Kütüphanesi Ekle</translation>
     </message>
@@ -3435,3 +3455,3 @@
         <source>Are you sure you want to cancel the import?</source>
-        <translation>Aktarma işlemini iptal etmek istediğinizden emin misiniz?</translation>
+        <translation>İçeriye aktarma işlemini iptal etmek istediğinizden emin misiniz?</translation>
     </message>
@@ -3442,5 +3462,5 @@
  Click OK to continue.</source>
-        <translation>Last.fm ortam kütüphanenizde çalınmış bir parça bulamadı.
+        <translation>Last.fm müzik kütüphanenizde çalınmış parça bulamadı.
 
- Devam etmek için Tamam&apos;a tıklayın.</translation>
+ Devam Etmek için TAMAM düğmesine tıklayın.</translation>
     </message>
@@ -3449,3 +3469,3 @@
         <source>Media Library Import Complete</source>
-        <translation>Ortam Kütüphanesi Aktarımı Tamamlandı</translation>
+        <translation>Müzik Kütüphanesi İçeri Aktarma İşlemi Tamamlandı</translation>
     </message>
@@ -3455,4 +3475,4 @@
 Your profile will be updated with the new tracks in a few minutes.</source>
-        <translation>Last.fm dinleme geçmişinizi sunucuya sundu.
-Profiliniz yeni parçalarla birlikte birkaç dakika içinde güncellenecek.</translation>
+        <translation>Last.fm dinleme geçmişinizi sunucuya gönderdi.
+Profiliniz yeni parçaları da gösterecek şekilde birkaç dakika içerisinde güncellenecek.</translation>
     </message>
@@ -3461,3 +3481,3 @@
         <source>Library Import Failed</source>
-        <translation>Kütüphane Aktarımı Başarısız</translation>
+        <translation>Kütüphanesi Ekleme İşlemi Başarısız Oldu</translation>
     </message>
@@ -3466,3 +3486,3 @@
         <source>Sorry, Last.fm was unable to import your listening history. This is probably because you&apos;ve already scrobbled too many tracks. Listening history can only be imported to brand new profiles.</source>
-        <translation>Üzgünüz, Last.fm dinleme geçmişinizi aktaramadı. Bunun nedeni büyük olasılıkla çok fazla parça skroplamış olmanız. Dinleme geçmişi yalnızca yeni profiller için aktarılabilir.</translation>
+        <translation>Üzgünüm, Last.fm dinleme kütüphanenizi içeriye aktaramadı. Bu yakın zamanda birçok parça bilgisi göndermenizden kaynaklanıyor olabilir. Dinleme geçmişi sadece yeni profillere aktarılabilir.</translation>
     </message>
@@ -3474,3 +3494,3 @@
         <source>No soundcard available.</source>
-        <translation>Ses kartı yok.</translation>
+        <translation>Kullanılabilir ses kartı yok.</translation>
     </message>
@@ -3479,3 +3499,3 @@
         <source>Your soundcard is either busy or not present. Try restarting the application.</source>
-        <translation>Ses kartınız ya meşgul ya da mevcut değil. Lütfen uygulamayı tekrar başlatın.</translation>
+        <translation>Ses kartınız meşgul ya da yok. Uygulamayı yeniden başlatmayı deneyin.</translation>
     </message>
@@ -3513,3 +3533,3 @@
         <source>Sorry, this track isn&apos;t in the Last.fm catalog, and thus cannot be streamed.</source>
-        <translation>Üzgünüz, bu parça Last.fm kataloğunda yer almıyor ve bu yüzden de netten yayınlanamaz.</translation>
+        <translation>Üzgünüm, bu parça Last.fm kataloğunda yok. Bu nedenle yayınlanamaz.</translation>
     </message>
@@ -3518,3 +3538,3 @@
         <source>Can&apos;t resume a station without having listened to one first.</source>
-        <translation>Önce bir istasyon dinlenmeden bir başkasına devam edilemez.</translation>
+        <translation>Bir dinlenmiş parça olmadan bir radyo istasyonu yeniden yüklenemez.</translation>
     </message>
@@ -3533,3 +3553,3 @@
 %1</source>
-        <translation>Müzik listesi okunamıyor. Hata:
+        <translation>Çalma listesi okunamadı. Hata:
 
@@ -3647,3 +3667,3 @@
         <source>The Last.fm servers are temporarily overloaded, please try again in a moment.</source>
-        <translation>Last.fm sunucuları geçici süreyle aşırı yüklenmiş durumdadır, lütfen birazdan tekrar deneyin.</translation>
+        <translation>Last.fm sunucusu geçici olarak aşırı yük altında, lütfen bir süre sonra tekrar deneyin.</translation>
     </message>
@@ -3652,3 +3672,3 @@
         <source>Cannot contact the Last.fm server. Is your Internet connection configured correctly?</source>
-        <translation>Last.fm sunucusuna bağlanamıyor. İnternet bağlantınız doğru şekilde yapılandırılmış mı?</translation>
+        <translation>Last.fm sunucusu ile bağlantı kurulamadı. İnternet bağlantınızı doğru şekilde yapılandırdınız mı?</translation>
     </message>
@@ -3662,3 +3682,3 @@
         <source>Could not connect to server. Wrong username or password.</source>
-        <translation>Sunucuya bağlanamadı. Yanlış kullanıcı adı veya parola. </translation>
+        <translation>Sunucuya bağlanılamadı. Hatalı kullanıcı adı ya da parola.</translation>
     </message>
@@ -3676,5 +3696,3 @@
         <comment>%1 is a list of plugins, eg. &apos;Foobar, Winamp, Windows Media Player&apos;</comment>
-        <translation>ya da müziğinizi %1 ya da %2 ile dinleyin.
-
-</translation>
+        <translation>ya da %1 veya %2 içerisinden müziğinizi dinleyin.</translation>
     </message>
@@ -3683,3 +3701,3 @@
         <source>or install a player plugin to scrobble music from your media player.</source>
-        <translation>ya da ortam çalarınızdan müzik skroplamak için bir müzikçalar eklentisi kurun.</translation>
+        <translation>ya da bir müzikçalar eklentisi kurarak kendi müzikçalarınızdan müzik bilgilerinizi gönderin.</translation>
     </message>
@@ -3703,3 +3721,3 @@
         <source>&lt;p&gt;Enter an &lt;b&gt;artist&lt;/b&gt; or &lt;b&gt;tag&lt;/b&gt; to start listening to Last.fm radio.&lt;/p&gt;</source>
-        <translation>&lt;p&gt;Last.fm radyosunu dinlemek için bir &lt;b&gt;sanatçı&lt;/b&gt; ya da &lt;b&gt;etiket&lt;/b&gt; girin.&lt;/p&gt;</translation>
+        <translation>&lt;p&gt;Last.fm&apos;i dinlemeye başlamak için bir &lt;b&gt;sanatçı&lt;/b&gt; ya da &lt;b&gt;etiket&lt;/b&gt; girin.&lt;/p&gt;</translation>
     </message>
@@ -3723,3 +3741,3 @@
         <source>or listen to music in your media player.</source>
-        <translation>ya da ortam çalarınızdan müzik dinleyin.</translation>
+        <translation>ya da kendi müzik çalarınızdan müzik dinleyin.</translation>
     </message>
@@ -3729,3 +3747,3 @@
         <comment>%1 is a media player</comment>
-        <translation>ya da müziğinizi %1 ile dinleyin.</translation>
+        <translation>ya da %1 içerisinden müziğinizi dinleyin.</translation>
     </message>
@@ -3734,3 +3752,3 @@
         <source>Do It!</source>
-        <translation>Yap Şunu!</translation>
+        <translation>Şimdi!</translation>
     </message>
@@ -3747,4 +3765,4 @@
 This will add charts to your profile based on what you&apos;ve listened to in the past.</source>
-        <translation>Ortam çalar dinleme geçmişinizi aktarmak istiyor musunuz?
-Bununla, geçmişte dinlediklerinize bağlı olarak profilinize listeler eklenecektir.</translation>
+        <translation>Müzik çalar dinleme geçmişinizi içeriye aktarmak istiyor musunuz?
+Bu işlem geçmişte dinlediklerinizi profilinize ekleyecek.</translation>
     </message>
@@ -3756,3 +3774,3 @@
         <source>Your soundcard is either busy or not present. Try restarting the application.</source>
-        <translation>Ses kartınız ya meşgul ya da mevcut değil. Lütfen uygulamayı tekrar başlatın.</translation>
+        <translation>Ses kartınız meşgul ya da yok. Uygulamayı yeniden başlatmayı deneyin.</translation>
     </message>
@@ -3763,3 +3781,3 @@
 %1</source>
-        <translation>Çalma başlatılamadı. Hata:
+        <translation>Çalmaya başlanamadı. Hata:
 
@@ -3773,3 +3791,3 @@
         <source>Click to enable/disable scrobbling</source>
-        <translation>Skroplama devrede/devre dışı</translation>
+        <translation>Bilgi gönderme devrede/devre dışı</translation>
     </message>
@@ -3778,3 +3796,3 @@
         <source>Scrobbling %1</source>
-        <translation>%1 skroplanıyor</translation>
+        <translation>Bilgi gönderimi %1</translation>
     </message>
@@ -3796,3 +3814,3 @@
         <source>Bad session</source>
-        <translation>Kötü oturum</translation>
+        <translation>Uygunsuz oturum</translation>
     </message>
@@ -3806,3 +3824,3 @@
         <source>Wrong username / password</source>
-        <translation>Hatalı kullanıcı adı / parola</translation>
+        <translation>Yanlış kullanıcı adı / parola</translation>
     </message>
@@ -3811,3 +3829,3 @@
         <source>Wrong timezone</source>
-        <translation>Hatalı zaman dilimi</translation>
+        <translation>Yanlış zaman dilimi</translation>
     </message>
@@ -3816,3 +3834,3 @@
         <source>Could not reach server</source>
-        <translation>Sunucuya ulaşılamadı</translation>
+        <translation>Sunucuya erişilemedi</translation>
     </message>
@@ -3829,3 +3847,3 @@
         <source>Couldn&apos;t connect to submissions server. Will cache scrobbled tracks.</source>
-        <translation>İletişim sunucusuyla bağlantı kurulamadı. Skroplanan parçalar arabelleğe alınacak.</translation>
+        <translation>İletişim sunucusuyla bağlantı kurulamadı. Gönderilen parçalar arabelleğe alınacak.</translation>
     </message>
@@ -3859,3 +3877,3 @@
         <source>%1 scrobbled.</source>
-        <translation>%1 skroplandı.</translation>
+        <translation>%1 gönerildi.</translation>
     </message>
@@ -3885,4 +3903,4 @@
         <translation>
-            <numerusform>%n parça skroplanıyor...</numerusform>
-            <numerusform>%n parça skroplanıyor...</numerusform>
+            <numerusform>%n parça gönderiliyor...</numerusform>
+            <numerusform></numerusform>
         </translation>
@@ -3892,3 +3910,3 @@
         <source>Couldn&apos;t submit, will try again later</source>
-        <translation>Giriş başarısız, daha sonra tekrar denenecek</translation>
+        <translation>Gönderilemedi, daha sonra yeniden deneyin</translation>
     </message>
@@ -3897,3 +3915,3 @@
         <source>Couldn&apos;t submit, the username or password is incorrect</source>
-        <translation>Giriş başarısız, kullanıcı adı ya da parola yanlış</translation>
+        <translation>Gönderilemedi, kullanıcı adınız ya da parolanız yanlış</translation>
     </message>
@@ -3902,3 +3920,3 @@
         <source>Couldn&apos;t contact server, will try again later</source>
-        <translation>Sunucu bağlantısı kurulamadı, daha sonra tekrar denenecek</translation>
+        <translation>Sunucuya bağlantı kurulamadı, daha sonra yeniden deneyin</translation>
     </message>
@@ -3993,3 +4011,3 @@
         <source>Select the ones you want to use for scrobbling by ticking the box next to them. </source>
-        <translation> Skroplamada kullanmak istediklerinizi seçmek için yanındaki kutuyu işaretleyin.</translation>
+        <translation>Gönderimde kullanmak istediklerinizi seçmek için yanındaki kutuyu işaretleyin.</translation>
     </message>
@@ -4016,3 +4034,3 @@
         <source>Please select the ones you want to update and click Next to have them downloaded and installed automatically.</source>
-        <translation>Güncellemek istediklerinizi seçip İleri&apos;ye tıklarsanız, hepsi indirilip otomatik olarak yüklenir.</translation>
+        <translation>Lütfen güncellemek istediklerinizi seçin ve otomatik olarak indirmek ve kurmak için Devam Et düğmesine tıklayın.</translation>
     </message>
@@ -4053,3 +4071,3 @@
 cached images and bios?</source>
-        <translation>Arabellekteki bütün resim ve biyografileri 
+        <translation>Arabellekteki bütün resim ve biyografileri
 silmek istediğinizden emin misiniz?</translation>
@@ -4091,3 +4109,3 @@
         <source>Scrobbling</source>
-        <translation>Skoplama</translation>
+        <translation>Gönderim</translation>
     </message>
@@ -4126,3 +4144,3 @@
         <source>You need to restart the application for the language change to take effect.</source>
-        <translation>Dil değişikliğinin yürürlüğe girmesi için uygulamayı yeniden başlatmanız gerekiyor.</translation>
+        <translation>Dil değişikliğini etkinleştimek için uygulamayı yeniden başlatmanız gerekir.</translation>
     </message>
@@ -4132,4 +4150,4 @@
         <translation>
-            <numerusform>Ön bellek boşaltıldı. %n dosya silindi.</numerusform>
-            <numerusform>Ön bellek boşaltıldı. %n dosya silindi.</numerusform>
+            <numerusform>Bellek boşaltıldı. %n dosya silindi.</numerusform>
+            <numerusform></numerusform>
         </translation>
@@ -4139,3 +4157,3 @@
         <source>You need to restart the application for the dock icon appearance change to take effect.</source>
-        <translation>Dock simgesi görünüm değişikliğinin yürürlüğe girmesi için uygulamayı tekrar başlatmanız gerekiyor.</translation>
+        <translation>Simge değişikliğini etkinleştimek için uygulamayı yeniden başlatmanız gerekir.</translation>
     </message>
@@ -4149,3 +4167,3 @@
         <source>Launch Last.fm with iTunes</source>
-        <translation>Last.fm&apos;i iTunes ile birlikte başlat</translation>
+        <translation>Last.fm&apos;i iTunes ile Başlat</translation>
     </message>
@@ -4154,3 +4172,3 @@
         <source>You need to restart the application for the audio device change to take effect.</source>
-        <translation>Değişikliğin yürürlüğe girmesi için uygulamayı yeniden başlatmanız gerekir.</translation>
+        <translation>Eğer ses aygıtını değiştirdiğinizde bunun etkin olabilmesi için uygulamayı yeniden başlatmanız gerekir.</translation>
     </message>
@@ -4247,3 +4265,3 @@
         <source>Show application in dock</source>
-        <translation>Uygulamayı Dock&apos;ta göster</translation>
+        <translation>Uygulamayı çerçeve içinde göster</translation>
     </message>
@@ -4262,3 +4280,3 @@
         <source>Show application icon in system tray</source>
-        <translation>Uygulama simgesini sistem çubuğunda göster</translation>
+        <translation>Uygulama simgesini sistem çekmecesinde göster</translation>
     </message>
@@ -4335,3 +4353,3 @@
         <source>Manual</source>
-        <translation>Manuel</translation>
+        <translation>El ile</translation>
     </message>
@@ -4340,3 +4358,3 @@
         <source>Host:</source>
-        <translation>Sunucu:</translation>
+        <translation>Makine:</translation>
     </message>
@@ -4350,3 +4368,3 @@
         <source>Leave fields blank to use a direct connection</source>
-        <translation>Doğrudan bağlantı kullanmak için alanları boş bırakın</translation>
+        <translation>Doğrudan bağlanmak için alanları boş bırakın</translation>
     </message>
@@ -4373,3 +4391,3 @@
         <source>Configure Growl settings</source>
-        <translation>Homurtu ayarlarını yapılandır</translation>
+        <translation>Growl ayarlarını yapılandır</translation>
     </message>
@@ -4383,3 +4401,3 @@
         <source>Send notification to Growl when a new track starts</source>
-        <translation>Yeni bir parça başladığında Homurtuya bildirim gönder</translation>
+        <translation>Yeni bir parça başladığında Growl&apos;a bildirim gönder</translation>
     </message>
@@ -4396,3 +4414,3 @@
         <source>Configure iPod Connections</source>
-        <translation>iPod Bağlantılarını yapılandır</translation>
+        <translation>iPod Bağlantılarını Yapılandır</translation>
     </message>
@@ -4401,3 +4419,3 @@
         <source>Known iPods</source>
-        <translation>Bilinen iPod&apos;lar</translation>
+        <translation>Bilinen iPodlar</translation>
     </message>
@@ -4406,3 +4424,3 @@
         <source>Device</source>
-        <translation>Cihaz</translation>
+        <translation>Aygıt</translation>
     </message>
@@ -4416,3 +4434,3 @@
         <source>Remove iPod Connection</source>
-        <translation>iPod Bağlantısını kaldır</translation>
+        <translation>iPod Bağlantısını Sil</translation>
     </message>
@@ -4421,3 +4439,3 @@
         <source>Disable this to stop the LastFMHelper from always running</source>
-        <translation>LastFMHelper&apos;ın sürekli çalışmasını durdurmak için bunu devreden çıkarın</translation>
+        <translation>LastFMHelper&apos;ın durmadan çalışmasını engellemek için bunu kapatın</translation>
     </message>
@@ -4426,3 +4444,3 @@
         <source>Enable iPod Scrobbling</source>
-        <translation>iPod Skroplaması Devrede</translation>
+        <translation>iPod Gönderimini Etkinleştir</translation>
     </message>
@@ -4431,3 +4449,3 @@
         <source>Manually Synced</source>
-        <translation>Manuel Olarak Senkronize Edilmiş</translation>
+        <translation>El ile Eşzamanlanmış</translation>
     </message>
@@ -4444,3 +4462,3 @@
         <source>Configure Messenger Settings</source>
-        <translation>Messenger Ayarlarını Yapılandır</translation>
+        <translation>Hızlı Mesajlaşma Ayarlarını Yapılandır</translation>
     </message>
@@ -4449,3 +4467,3 @@
         <source>Now Playing Notifications</source>
-        <translation>Çalınan Parça Bildirimleri</translation>
+        <translation>Şimdi Dinliyor Bildirimleri</translation>
     </message>
@@ -4454,3 +4472,3 @@
         <source>Send notification to Messenger when a new track starts</source>
-        <translation>Yeni bir parça başladığında Messenger&apos;a bildirim gönder</translation>
+        <translation>Yeni bir parça başladığında Hızlı Mesajlaşma Uygulamasına bildirim gönder</translation>
     </message>
@@ -4497,3 +4515,3 @@
         <source>Soundcard:</source>
-        <translation>Ses kartı:</translation>
+        <translation>Ses Kartı:</translation>
     </message>
@@ -4507,3 +4525,3 @@
         <source>Buffer size:</source>
-        <translation>Arabellek büyüklüğü:</translation>
+        <translation>Tampon Bellek Boyutu:</translation>
     </message>
@@ -4517,3 +4535,3 @@
         <source>Manage HTTP buffer size automatically</source>
-        <translation>HTTP arabellek büyüklüğünü otomatik olarak belirle</translation>
+        <translation>HTTP tampon boyutunu otomatik olarak yönet</translation>
     </message>
@@ -4522,3 +4540,3 @@
         <source>Stream in external player</source>
-        <translation>Harici oynatıcıda yayınla</translation>
+        <translation>Dış müzik çalarda yayın</translation>
     </message>
@@ -4527,3 +4545,3 @@
         <source>Connect to localhost on port:</source>
-        <translation>Şu porttan yerel sunucuya bağlan:</translation>
+        <translation>Localhost&apos;a bu porttan bağlan:</translation>
     </message>
@@ -4540,3 +4558,3 @@
         <source>Configure Scrobbler Settings</source>
-        <translation>Skroplama Ayarlarını Yapılandır</translation>
+        <translation>Gönderme Ayarlarını Yapılandır</translation>
     </message>
@@ -4555,3 +4573,3 @@
         <source>Scrobble directories</source>
-        <translation>Dizinleri skropla</translation>
+        <translation>Dizinleri gönder</translation>
     </message>
@@ -4560,3 +4578,3 @@
         <source>Only tracks from the selected directories will be scrobbled.</source>
-        <translation>Yalnızca seçilen dizinlerdeki parçalar skroplanacak.</translation>
+        <translation>Yalnızca seçilen dizinlerdeki parçalar gönderilecek.</translation>
     </message>
@@ -4565,3 +4583,3 @@
         <source>Scrobble at</source>
-        <translation>Skroplama oranı</translation>
+        <translation>Bilgileri parçanın yüzde</translation>
     </message>
@@ -4575,3 +4593,3 @@
         <source>percent of track length</source>
-        <translation>parça uzunluğunun yüzdesi</translation>
+        <translation>kadarı dinlendiğinde gönder</translation>
     </message>
@@ -4580,3 +4598,3 @@
         <source>Allow Last.fm to fingerprint your tracks</source>
-        <translation>Last.fm&apos;in parçalarınızın parmak izini almasına izin verin</translation>
+        <translation>Parçalarınızın parmak izini alması için Last.fm&apos;e izin ver</translation>
     </message>
@@ -4618,3 +4636,3 @@
         <source>Format:</source>
-        <translation>Format:</translation>
+        <translation>Biçim:</translation>
     </message>
@@ -4670,3 +4688,3 @@
         <comment>eg. Share &quot;Kylie Minogue&quot; with bob</comment>
-        <translation>kiminle:</translation>
+        <translation>ile:</translation>
     </message>
@@ -4694,3 +4712,10 @@
 &lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:&apos;MS Shell Dlg 2&apos;; font-size:8pt;&quot;&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</source>
-        <translation>&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;p, li { white-space: pre-wrap; }&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:&apos;Lucida Grande&apos;; font-size:13pt; font-weight:400; font-style:normal;&quot;&gt;&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:&apos;MS Shell Dlg 2&apos;; font-size:8pt;&quot;&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
+        <translation>&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
+p, li { white-space: pre-wrap; }
+&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:&apos;Lucida Grande&apos;; font-size:13pt; font-weight:400; font-style:normal;&quot;&gt;&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-family:&apos;MS Shell Dlg 2&apos;; font-size:8pt;&quot;&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</translation>
+    </message>
+    <message>
+        <location filename="../src/ShareDialog.ui" line="159"/>
+        <source>Cancel</source>
+        <translation>İptal</translation>
     </message>
@@ -4715,3 +4740,3 @@
         <source>My Recommendations</source>
-        <translation>Tavsiyelerim</translation>
+        <translation>Önerdiklerim</translation>
     </message>
@@ -4725,3 +4750,3 @@
         <source>My Loved Tracks</source>
-        <translation>Sevdiğim Parçalar </translation>
+        <translation>Sevdiğim Parçalar</translation>
     </message>
@@ -4735,3 +4760,3 @@
         <source>Recently Played</source>
-        <translation>En Son Çalınanlar</translation>
+        <translation>Son Çaldıklarım</translation>
     </message>
@@ -4740,3 +4765,3 @@
         <source>Recently Loved</source>
-        <translation>En Son Sevdiklerim</translation>
+        <translation>Son Sevdiklerim</translation>
     </message>
@@ -4745,3 +4770,3 @@
         <source>Recently Banned</source>
-        <translation>En Son Engellediklerim</translation>
+        <translation>Son Engellediklerim</translation>
     </message>
@@ -4775,3 +4800,3 @@
         <source>Change Station</source>
-        <translation>İstasyon Değiştir</translation>
+        <translation>İstasyonu Değiştir</translation>
     </message>
@@ -4780,3 +4805,3 @@
         <source>Now Playing</source>
-        <translation>Çalıyor</translation>
+        <translation>Şimdi Çalınan</translation>
     </message>
@@ -4793,3 +4818,3 @@
         <source>&amp;Play This Station</source>
-        <translation>Bu İstasyonu &amp;Çal</translation>
+        <translation>&amp;Bu İstasyonu Dinle</translation>
     </message>
@@ -4803,3 +4828,3 @@
         <source>&amp;Remove From History</source>
-        <translation>Geçmişten &amp;Kaldır</translation>
+        <translation>&amp;Geçmişten Sil</translation>
     </message>
@@ -4813,3 +4838,3 @@
         <source>Sort By Popularity</source>
-        <translation>Popülerliğe Göre Sırala</translation>
+        <translation>Beğeniye Göre Sırala</translation>
     </message>
@@ -4818,3 +4843,3 @@
         <source>Sort A-Z</source>
-        <translation>A-Z&apos;ye Göre Sırala</translation>
+        <translation>A&apos;dan Z&apos;ye Sırala</translation>
     </message>
@@ -4823,3 +4848,3 @@
         <source>Sort Z-A</source>
-        <translation>Z-A&apos;ya Göre Sırala</translation>
+        <translation>Z&apos;den A&apos;ya Sırala</translation>
     </message>
@@ -4828,3 +4853,3 @@
         <source>Play This Tag Station</source>
-        <translation>Bu Etiket İstasyonunu Çal</translation>
+        <translation>Bu Etiketi Çal</translation>
     </message>
@@ -4833,3 +4858,3 @@
         <source>Play Only Music You Tagged &quot;%1&quot;</source>
-        <translation>Yalnızca &quot;%1&quot; Olarak Etiketlediğiniz Müziği Çal</translation>
+        <translation>Sadece &quot;%1&quot; Olarak Etiketlediğim Parçaları Çal</translation>
     </message>
@@ -4843,3 +4868,3 @@
         <source>Sort Tags By Popularity</source>
-        <translation>Etiketleri Popülerliğe Göre Sırala</translation>
+        <translation>Etiketleri Beğeniye Göre Sırala</translation>
     </message>
@@ -4858,3 +4883,3 @@
         <source>Sort By Similarity</source>
-        <translation>Benzerliğe Göre Sırala</translation>
+        <translation>Benzerliğe göre Sırala</translation>
     </message>
@@ -4863,3 +4888,3 @@
         <source>Play %1&apos;s Radio Station</source>
-        <translation>%1&apos;in Radyo İstasyonunu Çal</translation>
+        <translation>%1 Kişisinin İstasyonunu Çal</translation>
     </message>
@@ -4868,3 +4893,3 @@
         <source>Play %1&apos;s Neighbourhood</source>
-        <translation>%1&apos;in Komşularını Çal</translation>
+        <translation>%1 Kişisinin Komşuluklarını Çal</translation>
     </message>
@@ -4873,3 +4898,3 @@
         <source>Play %1&apos;s Loved Tracks</source>
-        <translation>%1 Kullanıcısının Sevdiğim Parçalar Listesini Çal</translation>
+        <translation>%1 Kişisinin Sevdiği Parçaları Çal</translation>
     </message>
@@ -4878,3 +4903,3 @@
         <source>Go To %1&apos;s Profile</source>
-        <translation>%1 Kullanıcısının Profiline Git</translation>
+        <translation>%1 Kişisinin Profiline Git</translation>
     </message>
@@ -4883,3 +4908,3 @@
         <source>End Friendship</source>
-        <translation>Arkadaşlığa Son Ver</translation>
+        <translation>Arkadaşlığı Bitir</translation>
     </message>
@@ -4888,3 +4913,3 @@
         <source>Play Track</source>
-        <translation>Parça Çal</translation>
+        <translation>Parçayı Çal</translation>
     </message>
@@ -4908,3 +4933,3 @@
         <source>Tag...</source>
-        <translation>Etiketleyin...</translation>
+        <translation>Etiket...</translation>
     </message>
@@ -4918,3 +4943,3 @@
         <source>Remove From Profile</source>
-        <translation>Profilimden Kaldır</translation>
+        <translation>Profilden Sil</translation>
     </message>
@@ -4928,3 +4953,3 @@
         <source>Un-ban This Track</source>
-        <translation>Bu Parçanın Engellemesini Kaldır</translation>
+        <translation>Bu Parçanın Engelini Kaldır</translation>
     </message>
@@ -4933,3 +4958,3 @@
         <source>&amp;Empty Station History</source>
-        <translation>&amp;Boş İstasyon Geçmişi</translation>
+        <translation>&amp;İstasyon Geçmişini Temizle</translation>
     </message>
@@ -4938,3 +4963,3 @@
         <source>Confirm</source>
-        <translation>Onay</translation>
+        <translation>Onayla</translation>
     </message>
@@ -4943,3 +4968,3 @@
         <source>Do you really want to remove %1 from your friends list?</source>
-        <translation>%1 arkadaş listenizden kesin çıkarılsın mı?</translation>
+        <translation>%1 arkadaş listenizden gerçekten çıkarılsın mı?</translation>
     </message>
@@ -4953,3 +4978,3 @@
         <source>Do you really want to remove %1 from your Loved Tracks?</source>
-        <translation>%1 Sevdiğim Parçalar listenizden kesin çıkarılsın mı?</translation>
+        <translation>%1 parçasını Sevdiğim Parçalar listenizden gerçekten çıkarmak istiyor musunuz?</translation>
     </message>
@@ -4963,3 +4988,3 @@
         <source>Do you really want to un-ban %1?</source>
-        <translation>%1 üzerindeki engelleme kesin kalksın mı?</translation>
+        <translation>%1parçasının Engellemesini gerçekten kaldırmak istiyor musunuz?</translation>
     </message>
@@ -4968,3 +4993,3 @@
         <source>Do you really want to remove %1 from your profile?</source>
-        <translation>%1 profilinizden kesin çıkarılsın mı?</translation>
+        <translation>%1 parçasını gerçekten profilinizden çıkarmak istiyor musunuz?</translation>
     </message>
@@ -4978,3 +5003,3 @@
         <source>Tags %1 as %2</source>
-        <translation>%1, %2 olarak etiketlenir</translation>
+        <translation>%1, %2 olarak etiketlensin</translation>
     </message>
@@ -4983,3 +5008,3 @@
         <source>You can&apos;t add tracks to Recently Played</source>
-        <translation>Son Çalınan Parçalarım listenize parça ekleyemezsiniz</translation>
+        <translation>Son Çalınanlara parça ekleyemezsiniz</translation>
     </message>
@@ -4988,3 +5013,3 @@
         <source>Loves %1</source>
-        <translation>%1 sevilir</translation>
+        <translation>%1 Sever</translation>
     </message>
@@ -4993,3 +5018,3 @@
         <source>Bans %1</source>
-        <translation>%1 engellenir</translation>
+        <translation>%1 Engeller</translation>
     </message>
@@ -5003,3 +5028,3 @@
         <source>Tag %1 as %2</source>
-        <translation>%1, %2 olarak etiketlenir</translation>
+        <translation>%1 %2 olarak etiketlensin</translation>
     </message>
@@ -5018,3 +5043,3 @@
         <source>Shares %1 with %2</source>
-        <translation>%1 parçasını %2 ile paylaşır</translation>
+        <translation> %1 %2 ile paylaşıyor</translation>
     </message>
@@ -5023,3 +5048,3 @@
         <source>Shares %1 with yourself</source>
-        <translation>%1 parçasını kendinizle paylaşır</translation>
+        <translation> %1 sizinle paylaşıyor</translation>
     </message>
@@ -5028,3 +5053,3 @@
         <source>Add To My Loved Tracks</source>
-        <translation>Sevdiğim Parçalar Listesine Ekle</translation>
+        <translation>Sevdiğim Parçalara Ekle</translation>
     </message>
@@ -5033,3 +5058,3 @@
         <source>Add To My Banned Tracks</source>
-        <translation>Engellediğim Parçalar Listesine Ekle</translation>
+        <translation>Engellediğim Parçalara Ekle</translation>
     </message>
@@ -5038,3 +5063,3 @@
         <source>Add To My Playlist</source>
-        <translation>Müzik Listeme Ekle</translation>
+        <translation>Parça Listeme Ekle</translation>
     </message>
@@ -5043,3 +5068,3 @@
         <source>Remove From My Loved Tracks</source>
-        <translation>Sevdiğim Parçalar Listesinden Kaldır</translation>
+        <translation>Sevdiğim Parçalar Listesinden Sil</translation>
     </message>
@@ -5255,3 +5280,3 @@
         <source>Currently listening to &apos;%2&apos; by %1</source>
-        <translation>Şu anda %1 tarafından &apos;%2&apos; parçasını dinliyorsunuz</translation>
+        <translation>Şimdi %1 tarafından seslendirilen &apos;%2&apos; parçasını dinliyor</translation>
     </message>
@@ -5409,3 +5434,3 @@
         <source>Tag the artist</source>
-        <translation>Sanatçıyı etiketle</translation>
+        <translation>Sanatçıyı Etiketle</translation>
     </message>
@@ -5429,3 +5454,3 @@
         <source>Popular Tags for this Artist</source>
-        <translation>Bu Sanatçıyla ilgili Popüler Etiketler</translation>
+        <translation>Bu Sanatçı için En Çok Kullanılan Etiketler</translation>
     </message>
@@ -5434,3 +5459,3 @@
         <source>Popular Tags for this Track</source>
-        <translation>Bu Parçayla ilgili Popüler Etiketler</translation>
+        <translation>Bu Parça için En Çok Kullanılan Etiketler</translation>
     </message>
@@ -5439,3 +5464,3 @@
         <source>Popular Tags for this Album</source>
-        <translation>Bu Albümle ilgili Popüler Etiketler</translation>
+        <translation>Bu Albüm için En Çok Kullanılan Etiketler</translation>
     </message>
@@ -5449,3 +5474,3 @@
         <source>Enter comma-separated tags here</source>
-        <translation>Buraya virgülle ayrılmış etiketler girin</translation>
+        <translation>Virgül ile ayrılmış etiketleri buraya girin</translation>
     </message>
@@ -5457,3 +5482,3 @@
         <source>Sort by popularity</source>
-        <translation>Popülerliğe göre sırala</translation>
+        <translation>Beğeniye göre sırala</translation>
     </message>
@@ -5462,3 +5487,3 @@
         <source>Sort tags A-Z</source>
-        <translation>Etiketleri A-Z&apos;ye göre sırala</translation>
+        <translation>Etiketleri A&apos;dan Z&apos;ye sırala</translation>
     </message>
@@ -5467,3 +5492,3 @@
         <source>Sort tags Z-A</source>
-        <translation>Etiketleri Z-A&apos;ya göre sırala</translation>
+        <translation>Etiketleri Z&apos;den A&apos;ya sırala</translation>
     </message>
@@ -5476,3 +5501,3 @@
         <comment>Needs to be very short, displayed in track progress bar</comment>
-        <translation>skroplanmıyor</translation>
+        <translation>gönderilmeyecek</translation>
     </message>
@@ -5482,3 +5507,3 @@
         <comment>Needs to be very short, displayed in track progress bar</comment>
-        <translation>skroplandı</translation>
+        <translation>gönderildi</translation>
     </message>
@@ -5487,3 +5512,3 @@
         <source>Click to toggle between track time and scrobble time</source>
-        <translation>Parça süresi ve skroplama süresi arasında geçiş yapmak için tıklayın</translation>
+        <translation>Gönderme süresi / parça süresi arasında geçiş yapmak için tıklayın</translation>
     </message>
@@ -5502,3 +5527,3 @@
         <source>Can&apos;t scrobble: %1</source>
-        <translation>Skroplanamıyor: %1</translation>
+        <translation>Bilgi gönderilemedi: %1</translation>
     </message>
@@ -5507,3 +5532,3 @@
         <source>artist or title missing from ID3 tag</source>
-        <translation>ID3 etiketinde sanatçı veya isim eksik</translation>
+        <translation>ID3 etiketinde sanatçı ya da başlık bilgisi yok</translation>
     </message>
@@ -5512,3 +5537,3 @@
         <source>missing start time</source>
-        <translation>başlangıç zamanı eksik</translation>
+        <translation>başlatma süresi bitti</translation>
     </message>
@@ -5517,3 +5542,3 @@
         <source>track too short</source>
-        <translation>parça fazla kısa</translation>
+        <translation>parça çok kısa</translation>
     </message>
@@ -5522,3 +5547,3 @@
         <source>Won&apos;t scrobble: track is in directory set to not scrobble</source>
-        <translation>Skroplanmıyor: parça skroplanmayanlar dizininde bulunuyor</translation>
+        <translation>Bilgi gönderilmez: parçanın bulunduğu dizin, bu dizinden bilgi gönderilmesin şeklinde ayarlanmış</translation>
     </message>
@@ -5527,3 +5552,3 @@
         <source>invalid artist name</source>
-        <translation>geçersiz sanatçı adı</translation>
+        <translation>geçerisiz sanatçı adı</translation>
     </message>
@@ -5532,3 +5557,3 @@
         <source>%1 (Brought to you by %2)</source>
-        <translation>%1 (%2 tarafından size getirilmiş)</translation>
+        <translation>%1 (%2 tarafından sunuluyor)</translation>
     </message>
@@ -5540,3 +5565,3 @@
         <source>%1 is logged in</source>
-        <translation>%1, sistemde</translation>
+        <translation>%1 sistemde</translation>
     </message>
@@ -5652,3 +5677,3 @@
         <source>Detected Running Players</source>
-        <translation>Çalışan Müzikçalar Saptandı</translation>
+        <translation>Çalışan Durumdaki Müzikçalarlar</translation>
     </message>
@@ -5801,3 +5826,3 @@
         <source>Sending your listening history to Last.fm</source>
-        <translation>Çalma geçmişiniz Last.fm&apos;e gönderiliyor</translation>
+        <translation>Dinleme geçmişiniz Last.fm&apos;e gönderiliyor</translation>
     </message>
@@ -5809,3 +5834,3 @@
         <source>Import from </source>
-        <translation>Aktarma yeri</translation>
+        <translation>Buradan içeri aktar </translation>
     </message>
@@ -5814,3 +5839,3 @@
         <source>Don&apos;t import listening history</source>
-        <translation>Dinleme geçmişini aktarma</translation>
+        <translation>Dinleme geçmişini içeri aktarma</translation>
     </message>
@@ -5820,3 +5845,3 @@
     <message>
-        <location filename="../src/wizarddialog_win.ui" line="31"/>
+        <location filename="../src/wizarddialog_mac.ui" line="34"/>
         <source>Dialog</source>
@@ -5825,3 +5850,3 @@
     <message>
-        <location filename="../src/wizarddialog_win.ui" line="119"/>
+        <location filename="../src/wizarddialog_mac.ui" line="91"/>
         <source>Cancel</source>
@@ -5870,3 +5895,3 @@
         <source>TextLabel stretching over two lines and being a bit long</source>
-        <translation>TextLabel stretching over two lines and being a bit long</translation>
+        <translation type="unfinished"></translation>
     </message>
@@ -5924,3 +5949,3 @@
         <source>There is a new version of Last.fm available.</source>
-        <translation>Last.fm&apos;in yeni bir sürümü mevcut.</translation>
+        <translation>Last.fm uygulamasının yeni sürümü var.</translation>
     </message>
@@ -5929,3 +5954,8 @@
         <source>Please click Continue to have it downloaded and installed automatically.</source>
-        <translation>Otomatik olarak yükleyip kurmak için Devam butonuna tıklayın.</translation>
+        <translation>Lütfen otomatik olarak indirmek ve yüklemek için Devam Et düğmesine tıklayın.</translation>
+    </message>
+    <message>
+        <location filename="../src/wizardselectupdatespage.cpp" line="83"/>
+        <source> (version </source>
+        <translation> (sürüm </translation>
     </message>
@@ -5955,3 +5985,3 @@
         <source>Please select which player you would like to import your listening history from:</source>
-        <translation>Lütfen dinleme geçmişinizi aktarmak istediğiniz çalma programını seçin:</translation>
+        <translation>Dinleme geçmişinizi aktarmak istediğiniz müzik çaları seçin:</translation>
     </message>
