--- a/src/iconshack.cpp
+++ b/src/iconshack.cpp
@@ -73,6 +73,34 @@
 }    
 
 
+#ifdef Q_WS_X11
+QPixmap
+IconShack::GetTrayIcon( MooseEnums::UserIconColour eColour )
+{
+    if (!mbIconsLoaded)
+        LoadIcons();
+
+    if (maTrayIcons[eColour].isNull())
+        return QPixmap(22, 22);
+    else
+        return maTrayIcons[eColour];
+}
+
+
+QPixmap
+IconShack::GetDisabledTrayIcon( MooseEnums::UserIconColour eColour )
+{
+    if (!mbIconsLoaded)
+        LoadIcons();
+
+    if (maDisabledTrayIcons[eColour].isNull())
+        return QPixmap(22, 22);
+    else
+        return maDisabledTrayIcons[eColour];
+}
+#endif
+
+
 void
 IconShack::LoadIcons()
 {
@@ -106,6 +134,20 @@
     maDisabledIcons[MooseEnums::eOrange].load( MooseUtils::dataPath( "icons/user_disabled.png" ) );
     maDisabledIcons[MooseEnums::eBlack].load( MooseUtils::dataPath( "icons/user_disabled.png" ) );
 
+#ifdef Q_WS_X11
+    maTrayIcons[MooseEnums::eRed].load( MooseUtils::dataPath( "icons/user_red22.png" ) );
+    maTrayIcons[MooseEnums::eBlue].load( MooseUtils::dataPath( "icons/user_blue22.png" ) );
+    maTrayIcons[MooseEnums::eGreen].load( MooseUtils::dataPath( "icons/user_green22.png" ) );
+    maTrayIcons[MooseEnums::eOrange].load( MooseUtils::dataPath( "icons/user_orange22.png" ) );
+    maTrayIcons[MooseEnums::eBlack].load( MooseUtils::dataPath( "icons/user_black22.png" ) );
+
+    maDisabledTrayIcons[MooseEnums::eRed].load( MooseUtils::dataPath( "icons/user_disabled22.png" ) );
+    maDisabledTrayIcons[MooseEnums::eBlue].load( MooseUtils::dataPath( "icons/user_disabled22.png" ) );
+    maDisabledTrayIcons[MooseEnums::eGreen].load( MooseUtils::dataPath( "icons/user_disabled22.png" ) );
+    maDisabledTrayIcons[MooseEnums::eOrange].load( MooseUtils::dataPath( "icons/user_disabled22.png" ) );
+    maDisabledTrayIcons[MooseEnums::eBlack].load( MooseUtils::dataPath( "icons/user_disabled22.png" ) );
+#endif
+
     mbIconsLoaded = true;
 
     LOG( 4, "Icons loaded\n" );
--- a/src/iconshack.h
+++ b/src/iconshack.h
@@ -75,6 +75,14 @@
     **************************************************************************/
     QPixmap GetDisabledUserIcon( MooseEnums::UserIconColour eColour);
 
+#ifdef Q_WS_X11
+    QPixmap
+    GetTrayIcon( MooseEnums::UserIconColour eColour );
+
+    QPixmap
+    GetDisabledTrayIcon( MooseEnums::UserIconColour eColour );
+#endif
+
 private:
 
     /*********************************************************************/ /**
@@ -89,6 +97,10 @@
     QPixmap maGoodIconsExpanded[5];
     QPixmap maExclIcons[5];
     QPixmap maDisabledIcons[5];
+#ifdef Q_WS_X11
+    QPixmap maTrayIcons[5];
+    QPixmap maDisabledTrayIcons[5];
+#endif
 
     bool mbIconsLoaded;
 };
--- a/src/systray.cpp
+++ b/src/systray.cpp
@@ -46,6 +46,8 @@
     {
         #ifdef Q_WS_MAC
             icon = m_pixmap;
+        #elif defined(Q_WS_X11)
+            icon = IconShack::instance().GetTrayIcon( iconColour );
         #else
             icon = IconShack::instance().GetGoodUserIcon( iconColour );
         #endif
@@ -54,6 +56,8 @@
     {
         #ifdef Q_WS_MAC
             icon = QIcon( m_pixmap ).pixmap( 19, 12, QIcon::Disabled );
+        #elif defined Q_WS_X11
+            icon = IconShack::instance().GetDisabledTrayIcon( iconColour );
         #else
             icon = IconShack::instance().GetDisabledUserIcon( iconColour );
         #endif
