--- a/src/container.cpp
+++ b/src/container.cpp
@@ -77,7 +77,9 @@
           m_userCheck( false ),
           m_soundcardError( false ),
           m_sidebarEnabled( false ),
-          m_sidebarWidth( 190 )
+          //m_sidebarWidth( 190 )
+          m_wasShown( false ),
+          m_wasMaximized( false )
 #ifndef Q_WS_MAC
         , m_styleOverrides( 0 )
 #endif
@@ -476,19 +478,21 @@
     {
         ui.splitter->restoreState( state );
 
-        m_sidebarWidth = The::settings().sidebarWidth();
+        /*m_sidebarWidth = The::settings().sidebarWidth();
 
         if ( m_sidebarWidth > 0 )
         {
             LOGL( 3, "Restoring sidebar width: " << m_sidebarWidth );
             ui.splitter->setSizes( QList<int>() << m_sidebarWidth );
-        }
+        }*/
     }
 #ifdef WIN32
     else
         ui.splitter->setSizes( QList<int>() << 176 );
 #endif
 
+    m_savedGeometry = The::settings().containerGeometry();
+
 #ifdef Q_WS_MAC
     //HACK for osx since the splitter is undetectable for most users
     QList<int> sizes = ui.splitter->sizes();
@@ -501,13 +505,12 @@
 
     // on mac qt returns height without unified toolbar height included :(
     // so first time we have to do it like this
-    QByteArray geometry = The::settings().containerGeometry();
-    if (geometry.isEmpty())
+    if (m_savedGeometry.isEmpty())
         resize( width(), 496 );
     else
-        restoreGeometry( geometry );
+        restoreGeometry( m_savedGeometry );
 #else
-    restoreGeometry( The::settings().containerGeometry() );
+    restoreGeometry( m_savedGeometry );
 #endif
     setWindowState( The::settings().containerWindowState() );
 }
@@ -565,19 +568,27 @@
 
     // don't save if hidden as Qt is shit and this doesn't work, we save 
     // just before hiding too to ensure we always have the state saved correctly
-    if ( m_sidebarEnabled )
+
+    // We also have to make sure that the window was made visible at least once
+    // in case someone started the client with -tray
+    if ( m_sidebarEnabled && m_wasShown )
     {
-        // HACK: workaround for Qt occasionally returning width 75 when it shouldn't
+        /*// HACK: workaround for Qt occasionally returning width 75 when it shouldn't
         int widthQtThinksItIs = ui.splitter->sizes().first();
         int actualWidth = widthQtThinksItIs == 75 ? m_sidebarWidth : widthQtThinksItIs;
 
         LOGL( 3, "Saving sidebar width: " << actualWidth );
-        The::settings().setSidebarWidth( actualWidth );
+        The::settings().setSidebarWidth( actualWidth );*/
 
         The::settings().setSplitterState( ui.splitter->saveState() );
     }
 
-    The::settings().setContainerWindowState( windowState() );
+    // Here too
+    if ( m_wasShown )
+    {
+        The::settings().setContainerGeometry( m_savedGeometry );
+        The::settings().setContainerWindowState( windowState() );
+    }
     The::settings().setVolume( ui.playcontrols.volume->value() );
     The::currentUser().setSidebarEnabled( m_sidebarEnabled );
 
@@ -672,6 +683,13 @@
 
 
 void
+Container::showEvent( QShowEvent *e )
+{
+    m_wasShown = true;
+}
+
+
+void
 Container::dragMoveEvent( QDragMoveEvent* event )
 {
   #ifndef HIDE_RADIO
@@ -732,8 +750,24 @@
         // save the geometry for the non-maximised state. So instead we must save it
         // for *every* resize and move event. Yay!
 
+        // And sometimes it doesn't seem to remember the window position after it
+        // was maximized.  So only save the geometry if it's not maximized and if
+        // the previous state was unmaximized.
         if (windowState() != Qt::WindowMaximized)
-            The::settings().setContainerGeometry( saveGeometry() );
+        {
+            if ( m_wasMaximized )
+            {
+                m_wasMaximized = false;
+            }
+            else
+            {
+                m_savedGeometry = saveGeometry();
+            }
+        }
+        else
+        {
+            m_wasMaximized = true;
+        }
     }
 
     if ( e->type() == QEvent::Show )
@@ -878,7 +912,7 @@
         centralWidget()->setContentsMargins( 0, 0, 5, 0 ); //aesthetics
         #endif
         icon = IconShack::instance().GetGoodUserIconCollapsed( The::currentUser().icon() );
-        ui.splitter->setSizes( QList<int>() << m_sidebarWidth );
+        //ui.splitter->setSizes( QList<int>() << m_sidebarWidth );
     }
 
     ui.actionMyProfile->setChecked( m_sidebarEnabled );
--- a/src/container.h
+++ b/src/container.h
@@ -76,6 +76,7 @@
         void dragEnterEvent( QDragEnterEvent* );
         void dragMoveEvent( QDragMoveEvent* );
         bool event( QEvent* );
+        void showEvent( QShowEvent* );
 
     private:
         struct : Ui::MainWindow
@@ -109,7 +110,11 @@
         bool m_soundcardError;
         bool m_sidebarEnabled;
         int  m_lastVolume;
-        int  m_sidebarWidth;
+        //int  m_sidebarWidth;
+
+        bool m_wasShown;
+        bool m_wasMaximized;
+        QByteArray m_savedGeometry;
 
         #ifndef Q_WS_MAC
         QStyle* m_styleOverrides;
