--- a/src/container.cpp
+++ b/src/container.cpp
@@ -523,7 +523,7 @@
     // Hack to get it working with VS2005
     QString path = qApp->applicationDirPath();
   #else
-    QString path = qApp->applicationDirPath() + "/extensions";
+    QString path = "/usr/lib/lastfm";
   #endif
 
   #ifndef QT_NO_DEBUG
--- a/src/lastfmapplication.cpp
+++ b/src/lastfmapplication.cpp
@@ -129,7 +129,9 @@
 
     // This is needed so that relative paths will work on Windows regardless
     // of where the app is launched from.
+  #ifdef WIN32
     QDir::setCurrent( applicationDirPath() );
+  #endif
 
     // this must be set before dialogs spawn in init() or whatever
     m_user = new User( this );
--- a/src/libMoose/MooseCommon.cpp
+++ b/src/libMoose/MooseCommon.cpp
@@ -43,7 +43,7 @@
 QString
 dataPath( QString file )
 {
-    return QApplication::applicationDirPath() + "/data/" + file;
+    return "/usr/share/lastfm/" + file;
 }
 
 
@@ -108,7 +108,7 @@
         // Hack to get it working with VS2005
         dirPath = qApp->applicationDirPath();
     #else
-        dirPath = qApp->applicationDirPath() + "/services";
+        dirPath = "/usr/lib/lastfm";
     #endif
 
     #ifndef QT_NO_DEBUG
