--- a/src/libFingerprint/fplib/src/OptFFT.cpp
+++ b/src/libFingerprint/fplib/src/OptFFT.cpp
@@ -30,6 +30,7 @@
 
 #include <cmath>
 #include <cassert>
+#include <cstdlib>
 #include <iostream>
 #include <memory.h> // for memcopy
 
--- a/src/libFingerprint/fplib/src/FingerprintExtractor.cpp
+++ b/src/libFingerprint/fplib/src/FingerprintExtractor.cpp
@@ -31,6 +31,7 @@
 #include <vector>
 #include <stdexcept>
 #include <cmath>
+#include <memory.h> // for memcpy
 
 #include <samplerate.h> // libsamplerate
 
--- a/src/libFingerprint/MP3_Source_Qt.cpp
+++ b/src/libFingerprint/MP3_Source_Qt.cpp
@@ -23,6 +23,8 @@
 #include <limits>
 #include <sstream>
 #include <cassert>
+#include <climits>
+#include <cstdlib>
 #include <stdexcept>
 
 #include "MP3_Source_Qt.h"
