--- strongswan-5.9.0/src/libcharon/plugins/resolve/resolve_handler.c	2018-10-03 23:35:17.000000000 +0200
+++ strongswan-5.9.0.new/src/libcharon/plugins/resolve/resolve_handler.c	2020-10-16 15:39:38.118138474 +0200
@@ -26,7 +26,7 @@
 #include <threading/mutex.h>
 
 /* path to resolvconf executable */
-#define RESOLVCONF_EXEC "/sbin/resolvconf"
+#define RESOLVCONF_EXEC "/usr/sbin/resolvconf"
 
 /* default prefix used for resolvconf interfaces (should have high prio) */
 #define RESOLVCONF_PREFIX "lo.inet.ipsec."
