diff --git a/src/gservice-interface.c b/src/gservice-interface.c
index 362a7c5..404ad59 100644
--- a/src/gservice-interface.c
+++ b/src/gservice-interface.c
@@ -159,6 +159,7 @@ gservice_setup_treeview (void)
 				G_TYPE_STRING,		/* service name for internal use */
 				G_TYPE_BOOLEAN);	/* activation status for internal use */
 
+
 	renderer = gtk_cell_renderer_pixbuf_new ();
 	column = gtk_tree_view_column_new_with_attributes (_("Status"),
 							renderer,
@@ -269,12 +270,12 @@ gservice_populate_services (void)
 			switch (e->status)
 			{
 				case GSVC_ON:
-				gtk_list_store_set (store, &iter, 0, pixbuf_on);
+				gtk_list_store_set (store, &iter, 0, pixbuf_on, -1);
 				break;
 			
 				case GSVC_OFF:
 				case GSVC_ERR:
-				gtk_list_store_set (store, &iter, 0, pixbuf_off);
+				gtk_list_store_set (store, &iter, 0, pixbuf_off, -1);
 				break;
 			
 				default:
