diff -Naur tk8.4.6.orig/generic/tkImgPhoto.c tk8.4.6/generic/tkImgPhoto.c
--- tk8.4.6.orig/generic/tkImgPhoto.c	2004-02-23 11:49:30.000000000 +0100
+++ tk8.4.6/generic/tkImgPhoto.c	2004-06-20 11:41:38.000000000 +0200
@@ -22,7 +22,7 @@
 
 #include "tkInt.h"
 #include "tkPort.h"
-#include "tclMath.h"
+#include <tcl/tclMath.h>
 #include <ctype.h>
 
 #ifdef __WIN32__
diff -Naur tk8.4.6.orig/generic/tkMain.c tk8.4.6/generic/tkMain.c
--- tk8.4.6.orig/generic/tkMain.c	2002-12-13 17:54:35.000000000 +0100
+++ tk8.4.6/generic/tkMain.c	2004-06-20 11:41:03.000000000 +0200
@@ -20,7 +20,7 @@
 #include <stdio.h>
 #include <string.h>
 #include <tcl.h>
-#include <tclInt.h>
+#include <tcl/tclInt.h>
 #include <tk.h>
 #include "tkInt.h"
 #ifdef NO_STDLIB_H
diff -Naur tk8.4.6.orig/generic/tkScale.c tk8.4.6/generic/tkScale.c
--- tk8.4.6.orig/generic/tkScale.c	2002-08-05 06:30:40.000000000 +0200
+++ tk8.4.6/generic/tkScale.c	2004-06-20 11:41:22.000000000 +0200
@@ -24,7 +24,7 @@
 #include "tkPort.h"
 #include "default.h"
 #include "tkInt.h"
-#include "tclMath.h"
+#include <tcl/tclMath.h>
 #include "tkScale.h"
 
 /*
diff -Naur tk8.4.6.orig/unix/tkUnixPort.h tk8.4.6/unix/tkUnixPort.h
--- tk8.4.6.orig/unix/tkUnixPort.h	2002-06-14 15:35:49.000000000 +0200
+++ tk8.4.6/unix/tkUnixPort.h	2004-06-20 11:40:48.000000000 +0200
@@ -221,7 +221,7 @@
  */
 
 #ifndef _TCLINT
-#include <tclInt.h>
+#include <tcl/tclInt.h>
 #endif
 
 #endif /* _UNIXPORT */
