diff -Nur wvdial-1.55.orig/Makefile wvdial-1.55/Makefile
--- wvdial-1.55.orig/Makefile	2005-12-03 00:54:13.000000000 +0200
+++ wvdial-1.55/Makefile	2006-02-03 22:21:11.402271250 +0200
@@ -1,7 +1,7 @@
 ifeq ($(TOPDIR),)
  TOPDIR=.
 
- prefix=/usr/local
+ prefix=${DESTDIR}/usr
  WVSTREAMS_INC=
  WVSTREAMS_LIB=
  WVSTREAMS_BIN=$(prefix)/bin
@@ -24,7 +24,7 @@
 
 BINDIR=${prefix}/bin
 MANDIR=${prefix}/share/man
-PPPDIR=/etc/ppp/peers
+PPPDIR=${DESTDIR}/etc/ppp/peers
 
 include $(TOPDIR)/wvrules.mk
 
