diff -aur wvdial-1.61/config.defaults.mk fw_wvdial-1.61//config.defaults.mk
--- wvdial-1.61/config.defaults.mk	2008-10-22 21:49:20.000000000 +0200
+++ fw_wvdial-1.61//config.defaults.mk	2010-06-27 19:06:01.000000000 +0200
@@ -13,7 +13,7 @@
 LIBS_PAM=-lpam
 LIBS_TCL=
 
-prefix=/usr/local
+prefix=/usr
 datadir=${prefix}/share
 includedir=${prefix}/include
 infodir=${prefix}/share/info
diff -aur wvdial-1.61/Makefile.in fw_wvdial-1.61//Makefile.in
--- wvdial-1.61/Makefile.in	2008-10-22 22:01:09.000000000 +0200
+++ fw_wvdial-1.61//Makefile.in	2010-06-27 16:48:51.000000000 +0200
@@ -3,7 +3,7 @@
 WVPACKAGE_TARNAME=@PACKAGE_TARNAME@
 WVPACKAGE_VERSION=@PACKAGE_VERSION@
 
-prefix=/usr/local
+prefix=/usr
 WVSTREAMS_INC=
 WVSTREAMS_LIB=
 WVSTREAMS_BIN=$(prefix)/bin
@@ -21,9 +21,9 @@
 endif
 LIBS+=$(PC_LIBS)
 
-BINDIR=${prefix}/bin
-MANDIR=${prefix}/share/man
-PPPDIR=/etc/ppp/peers
+BINDIR=${DESTDIR}/${prefix}/bin
+MANDIR=${DESTDIR}/${prefix}/share/man
+PPPDIR=${DESTDIR}/etc/ppp/peers
 
 include wvrules.mk
 
