diff -Naur wvdial-1.60.orig/Makefile wvdial-1.60/Makefile
--- wvdial-1.60.orig/Makefile	2007-11-03 15:39:03.000000000 +0100
+++ wvdial-1.60/Makefile	2007-11-03 15:39:48.000000000 +0100
@@ -1,5 +1,5 @@
 
-prefix=/usr/local
+prefix=${DESTDIR}/usr
 WVSTREAMS_INC=
 WVSTREAMS_LIB=
 WVSTREAMS_BIN=$(prefix)/bin
@@ -19,7 +19,7 @@
 
 BINDIR=${prefix}/bin
 MANDIR=${prefix}/share/man
-PPPDIR=/etc/ppp/peers
+PPPDIR=${DESTDIR}/etc/ppp/peers
 
 include wvrules.mk
 
