diff -Naur qtbase-opensource-src-5.6.0/mkspecs/common/gcc-base.conf qtbase-opensource-src-5.6.0-fix/mkspecs/common/gcc-base.conf
--- qtbase-opensource-src-5.6.0/mkspecs/common/gcc-base.conf	2016-03-03 12:35:22.000000000 +0100
+++ qtbase-opensource-src-5.6.0-fix/mkspecs/common/gcc-base.conf	2016-03-23 04:24:22.933511777 +0100
@@ -31,8 +31,8 @@
 # you can use the manual test in tests/manual/mkspecs.
 #
 
-QMAKE_CFLAGS_OPTIMIZE      = -O2
-QMAKE_CFLAGS_OPTIMIZE_FULL = -O3
+QMAKE_CFLAGS_OPTIMIZE      +=
+QMAKE_CFLAGS_OPTIMIZE_FULL +=
 
 QMAKE_CFLAGS               += -pipe
 QMAKE_CFLAGS_DEPS          += -M
@@ -79,6 +79,7 @@
 QMAKE_LFLAGS_EXCEPTIONS_OFF +=
 QMAKE_LFLAGS_LTCG       = $$QMAKE_CFLAGS_LTCG -fuse-linker-plugin
 
+## we disable these with ./configure options - crazy --
 QMAKE_CFLAGS_SSE2      += -msse2
 QMAKE_CFLAGS_SSE3      += -msse3
 QMAKE_CFLAGS_SSSE3     += -mssse3
