--- node-v20.18.0/configure.py	2024-10-03 11:20:49.000000000 +0200
+++ node-v20.18.0.new/configure.py	2024-10-29 18:09:16.513729641 +0100
@@ -1829,7 +1829,7 @@
   elif with_intl == 'system-icu':
     # ICU from pkg-config.
     o['variables']['v8_enable_i18n_support'] = 1
-    pkgicu = pkg_config('icu-i18n')
+    pkgicu = pkg_config('icu-i18n icu-uc')
     if not pkgicu[0]:
       error('''Could not load pkg-config data for "icu-i18n".
        See above errors or the README.md.''')
