diff -Naur boost-jam-3.1.10.orig/Jambase boost-jam-3.1.10/Jambase
--- boost-jam-3.1.10.orig/Jambase	2005-07-11 13:22:26.000000000 +0200
+++ boost-jam-3.1.10/Jambase	2005-07-11 13:25:02.000000000 +0200
@@ -48,7 +48,7 @@
 # required by Debian Policy.
 if ! $(BOOST_BUILD_PATH) && $(UNIX)
 {
-    BOOST_BUILD_PATH = /usr/share/boost-build ;
+    BOOST_BUILD_PATH = /usr/share/boost/v2 ;
 }
 
 
diff -Naur boost-jam-3.1.10.orig/jam.h boost-jam-3.1.10/jam.h
--- boost-jam-3.1.10.orig/jam.h	2005-07-11 13:22:26.000000000 +0200
+++ boost-jam-3.1.10/jam.h	2005-07-11 13:24:04.000000000 +0200
@@ -453,6 +453,10 @@
 # define OSPLAT "OSPLAT=IA64"
 # endif
 
+# if defined( __x86_64__ )
+# define OSPLAT "OSPLAT=x86_64"
+# endif
+
 # ifdef __s390__
 # define OSPLAT "OSPLAT=390"
 # endif
diff -Naur boost-jam-3.1.10.orig/jambase.c boost-jam-3.1.10/jambase.c
--- boost-jam-3.1.10.orig/jambase.c	2005-07-11 13:22:26.000000000 +0200
+++ boost-jam-3.1.10/jambase.c	2005-07-11 13:24:27.000000000 +0200
@@ -21,7 +21,7 @@
 "BOOST_BUILD_PATH.user-value = $(BOOST_BUILD_PATH) ;\n",
 "if ! $(BOOST_BUILD_PATH) && $(UNIX)\n",
 "{\n",
-"BOOST_BUILD_PATH = /usr/share/boost-build ;\n",
+"BOOST_BUILD_PATH = /usr/share/boost/v2 ;\n",
 "}\n",
 "rule boost-build ( dir ? )\n",
 "{\n",
