diff -Naur -X dont fusedav-0.2/config.h.in fusedav-0.2-p/config.h.in
--- fusedav-0.2/config.h.in	2006-04-02 18:04:41.000000000 +0200
+++ fusedav-0.2-p/config.h.in	2008-05-07 13:52:48.000000000 +0200
@@ -113,6 +113,9 @@
 /* Define to the version of this package. */
 #undef PACKAGE_VERSION
 
+/* The size of a `off_t', as computed by sizeof. */
+#undef SIZEOF_OFF_T
+
 /* Define to 1 if you have the ANSI C header files. */
 #undef STDC_HEADERS
 
diff -Naur -X dont fusedav-0.2/configure.ac fusedav-0.2-p/configure.ac
--- fusedav-0.2/configure.ac	2006-04-02 18:04:33.000000000 +0200
+++ fusedav-0.2-p/configure.ac	2008-05-07 13:52:48.000000000 +0200
@@ -88,6 +88,8 @@
     done 
 fi
 
+AC_CHECK_SIZEOF(off_t)
+
 # 64 Bit LFS support
 AC_SYS_LARGEFILE
 
diff -Naur -X dont fusedav-0.2/src/filecache.c fusedav-0.2-p/src/filecache.c
--- fusedav-0.2/src/filecache.c	2006-04-02 16:03:08.000000000 +0200
+++ fusedav-0.2-p/src/filecache.c	2008-05-07 13:52:48.000000000 +0200
@@ -47,6 +47,11 @@
 #include "fusedav.h"
 #include "session.h"
 
+#if (SIZEOF_OFF_T >= 8)
+#  define ne_content_range64 ne_content_range
+#  define ne_get_range64 ne_get_range
+#endif
+
 struct file_info {
     char *filename;
     int fd;
diff -Naur -X dont fusedav-0.2/src/fusedav.c fusedav-0.2-p/src/fusedav.c
--- fusedav-0.2/src/fusedav.c	2006-04-02 17:42:49.000000000 +0200
+++ fusedav-0.2-p/src/fusedav.c	2008-05-07 13:52:48.000000000 +0200
@@ -1386,8 +1386,11 @@
     }
 
     if (!ne_has_support(NE_FEATURE_SSL) ||
-        !ne_has_support(NE_FEATURE_TS_SSL) ||
-        !ne_has_support(NE_FEATURE_LFS)) {
+        !ne_has_support(NE_FEATURE_TS_SSL)
+#if (SIZEOF_OFF_T < 8)
+		|| !ne_has_support(NE_FEATURE_LFS)
+#endif
+		) {
         fprintf(stderr, "fusedav requires libneon built with SSL, SSL thread safety and LFS enabled.\n");
         goto finish;
     }
