--- pcmanfm-0.5.1.orig/data/pcmanfm-find.desktop.in	2009-06-05 03:18:37.000000000 +0200
+++ pcmanfm-0.5.1/data/pcmanfm-find.desktop.in	2009-06-05 03:18:50.000000000 +0200
@@ -12,7 +12,6 @@
 StartupNotify=true
 Type=Application
 Terminal=false
-MimeType=x-directory/normal;inode/directory;
 Comment=Search for files and folders in disks
 Comment[ru]=Поиск Файлов или Папок на дисках
 Comment[zh_TW]=在磁碟中尋找檔案或資料夾

