diff -Naur pcmanfm-0.3.5.4b/Makefile.am pcmanfm-0.3.5.4b.new/Makefile.am
--- pcmanfm-0.3.5.4b/Makefile.am	2008-01-25 00:55:22.000000000 +0530
+++ pcmanfm-0.3.5.4b.new/Makefile.am	2008-02-02 13:49:57.000000000 +0530
@@ -19,10 +19,6 @@
 icondir = $(datadir)/pixmaps
 icon_DATA = pcmanfm.png
 
-install-data-hook:
-	update-mime-database $(datadir)/mime
-	update-desktop-database $(desktopdir)
-
 install-data-local:
 	@$(NORMAL_INSTALL)
 	if test -d $(srcdir)/pixmaps; then \
diff -Naur pcmanfm-0.3.5.4b/Makefile.in pcmanfm-0.3.5.4b.new/Makefile.in
--- pcmanfm-0.3.5.4b/Makefile.in	2008-01-26 01:21:14.000000000 +0530
+++ pcmanfm-0.3.5.4b.new/Makefile.in	2008-02-02 13:49:50.000000000 +0530
@@ -705,10 +705,6 @@
 	uninstall-mimeDATA
 
 
-install-data-hook:
-	update-mime-database $(datadir)/mime
-	update-desktop-database $(desktopdir)
-
 install-data-local:
 	@$(NORMAL_INSTALL)
 	if test -d $(srcdir)/pixmaps; then \
