diff -u tracker-0.6.6/src/trackerd/cache.h tracker-0.6.6new/src/trackerd/cache.h
--- tracker-0.6.6/src/trackerd/cache.h	2008-05-24 02:38:29.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/cache.h	2008-12-05 11:35:06.000000000 +0100
@@ -26,8 +26,8 @@
 
 #include <sys/types.h>
 
-#include <util/list.h>
-#include <util/memchunk.h>
+#include "list.h"
+#include "memchunk.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/charset-map.c tracker-0.6.6new/src/trackerd/charset-map.c
--- tracker-0.6.6/src/trackerd/charset-map.c	2008-05-24 02:38:26.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/charset-map.c	2008-12-05 11:48:21.000000000 +0100
@@ -108,7 +108,7 @@
 	return !memcmp (v1, v2, 256);
 }
 
-int main (int argc, char **argv)
+int main2 (int argc, char **argv)
 {
 	unsigned char *block = NULL;
 	unsigned int bit = 0x01;
Seulement dans tracker-0.6.6new/src/trackerd: .deps
diff -u tracker-0.6.6/src/trackerd/gen-table.c tracker-0.6.6new/src/trackerd/gen-table.c
--- tracker-0.6.6/src/trackerd/gen-table.c	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gen-table.c	2008-12-05 11:51:24.000000000 +0100
@@ -142,7 +142,7 @@
 	header_init_bits (IS_PSAFE, 0, FALSE, CHARS_PSPECIAL);
 }
 
-int main (int argc, char **argv)
+int main3 (int argc, char **argv)
 {
 	int i;
 	
diff -u tracker-0.6.6/src/trackerd/gmime.c tracker-0.6.6new/src/trackerd/gmime.c
--- tracker-0.6.6/src/trackerd/gmime.c	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime.c	2008-12-05 11:02:27.000000000 +0100
@@ -39,10 +39,9 @@
 
 
 GQuark gmime_error_quark;
-
-const guint gmime_major_version = GMIME_MAJOR_VERSION;
-const guint gmime_minor_version = GMIME_MINOR_VERSION;
-const guint gmime_micro_version = GMIME_MICRO_VERSION;
+const guint gmime_major_version = 2;
+const guint gmime_minor_version = 2;
+const guint gmime_micro_version = 23;
 const guint gmime_interface_age = 0;
 const guint gmime_binary_age = 0;
 
@@ -62,17 +61,8 @@
  **/
 gboolean
 g_mime_check_version (guint major, guint minor, guint micro)
-{
-	if (GMIME_MAJOR_VERSION > major)
-		return TRUE;
-	
-	if (GMIME_MAJOR_VERSION == major && GMIME_MINOR_VERSION > minor)
-		return TRUE;
-	
-	if (GMIME_MAJOR_VERSION == major && GMIME_MINOR_VERSION == minor && GMIME_MICRO_VERSION >= micro)
-		return TRUE;
-	
-	return FALSE;
+{	
+	return TRUE;
 }
 
 
diff -u tracker-0.6.6/src/trackerd/gmime-cipher-context.h tracker-0.6.6new/src/trackerd/gmime-cipher-context.h
--- tracker-0.6.6/src/trackerd/gmime-cipher-context.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-cipher-context.h	2008-12-05 10:15:30.000000000 +0100
@@ -27,8 +27,8 @@
 
 #include <time.h>
 
-#include <gmime/gmime-stream.h>
-#include <gmime/gmime-session.h>
+#include "gmime-stream.h"
+#include "gmime-session.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-content-type.h tracker-0.6.6new/src/trackerd/gmime-content-type.h
--- tracker-0.6.6/src/trackerd/gmime-content-type.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-content-type.h	2008-12-05 10:15:57.000000000 +0100
@@ -23,7 +23,7 @@
 #define __GMIME_CONTENT_TYPE_H__
 
 #include <glib.h>
-#include <gmime/gmime-param.h>
+#include "gmime-param.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-data-wrapper.h tracker-0.6.6new/src/trackerd/gmime-data-wrapper.h
--- tracker-0.6.6/src/trackerd/gmime-data-wrapper.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-data-wrapper.h	2008-12-05 10:26:20.000000000 +0100
@@ -25,9 +25,9 @@
 #include <glib.h>
 #include <glib-object.h>
 
-#include <gmime/gmime-content-type.h>
-#include <gmime/gmime-stream.h>
-#include <gmime/gmime-utils.h>
+#include "gmime-content-type.h"
+#include "gmime-stream.h"
+#include "gmime-utils.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-disposition.h tracker-0.6.6new/src/trackerd/gmime-disposition.h
--- tracker-0.6.6/src/trackerd/gmime-disposition.h	2008-05-24 02:38:26.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-disposition.h	2008-12-05 10:16:36.000000000 +0100
@@ -23,7 +23,7 @@
 #define __GMIME_DISPOSITION_H__
 
 #include <glib.h>
-#include <gmime/gmime-param.h>
+#include "gmime-param.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-filter-basic.h tracker-0.6.6new/src/trackerd/gmime-filter-basic.h
--- tracker-0.6.6/src/trackerd/gmime-filter-basic.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-filter-basic.h	2008-12-05 10:17:02.000000000 +0100
@@ -22,7 +22,7 @@
 #ifndef __GMIME_FILTER_BASIC_H__
 #define __GMIME_FILTER_BASIC_H__
 
-#include <gmime/gmime-filter.h>
+#include "gmime-filter.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-filter-best.h tracker-0.6.6new/src/trackerd/gmime-filter-best.h
--- tracker-0.6.6/src/trackerd/gmime-filter-best.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-filter-best.h	2008-12-05 10:17:29.000000000 +0100
@@ -22,9 +22,9 @@
 #ifndef __GMIME_FILTER_BEST_H__
 #define __GMIME_FILTER_BEST_H__
 
-#include <gmime/gmime-filter.h>
-#include <gmime/gmime-charset.h>
-#include <gmime/gmime-utils.h>
+#include "gmime-filter.h"
+#include "gmime-charset.h"
+#include "gmime-utils.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-filter-charset.c tracker-0.6.6new/src/trackerd/gmime-filter-charset.c
--- tracker-0.6.6/src/trackerd/gmime-filter-charset.c	2008-05-24 02:38:26.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-filter-charset.c	2008-12-05 11:29:13.000000000 +0100
@@ -54,7 +54,7 @@
 
 
 static GMimeFilterClass *parent_class = NULL;
-
+#define ICONV_CONST
 
 GType
 g_mime_filter_charset_get_type (void)
diff -u tracker-0.6.6/src/trackerd/gmime-filter-charset.h tracker-0.6.6new/src/trackerd/gmime-filter-charset.h
--- tracker-0.6.6/src/trackerd/gmime-filter-charset.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-filter-charset.h	2008-12-05 10:17:43.000000000 +0100
@@ -23,7 +23,7 @@
 #define __GMIME_FILTER_CHARSET_H__
 
 #include <iconv.h>
-#include <gmime/gmime-filter.h>
+#include "gmime-filter.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-filter-crlf.h tracker-0.6.6new/src/trackerd/gmime-filter-crlf.h
--- tracker-0.6.6/src/trackerd/gmime-filter-crlf.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-filter-crlf.h	2008-12-05 10:17:56.000000000 +0100
@@ -22,7 +22,7 @@
 #ifndef __GMIME_FILTER_CRLF_H__
 #define __GMIME_FILTER_CRLF_H__
 
-#include <gmime/gmime-filter.h>
+#include "gmime-filter.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-filter-enriched.h tracker-0.6.6new/src/trackerd/gmime-filter-enriched.h
--- tracker-0.6.6/src/trackerd/gmime-filter-enriched.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-filter-enriched.h	2008-12-05 10:18:16.000000000 +0100
@@ -22,7 +22,7 @@
 #ifndef __GMIME_FILTER_ENRICHED_H__
 #define __GMIME_FILTER_ENRICHED_H__
 
-#include <gmime/gmime-filter.h>
+#include "gmime-filter.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-filter-from.h tracker-0.6.6new/src/trackerd/gmime-filter-from.h
--- tracker-0.6.6/src/trackerd/gmime-filter-from.h	2008-05-24 02:38:26.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-filter-from.h	2008-12-05 10:18:29.000000000 +0100
@@ -22,7 +22,7 @@
 #ifndef __GMIME_FILTER_FROM_H__
 #define __GMIME_FILTER_FROM_H__
 
-#include <gmime/gmime-filter.h>
+#include "gmime-filter.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-filter-gzip.h tracker-0.6.6new/src/trackerd/gmime-filter-gzip.h
--- tracker-0.6.6/src/trackerd/gmime-filter-gzip.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-filter-gzip.h	2008-12-05 10:18:42.000000000 +0100
@@ -22,7 +22,7 @@
 #ifndef __GMIME_FILTER_GZIP_H__
 #define __GMIME_FILTER_GZIP_H__
 
-#include <gmime/gmime-filter.h>
+#include "gmime-filter.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-filter-html.h tracker-0.6.6new/src/trackerd/gmime-filter-html.h
--- tracker-0.6.6/src/trackerd/gmime-filter-html.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-filter-html.h	2008-12-05 10:18:54.000000000 +0100
@@ -22,7 +22,7 @@
 #ifndef __GMIME_FILTER_HTML_H__
 #define __GMIME_FILTER_HTML_H__
 
-#include <gmime/gmime-filter.h>
+#include "gmime-filter.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-filter-md5.h tracker-0.6.6new/src/trackerd/gmime-filter-md5.h
--- tracker-0.6.6/src/trackerd/gmime-filter-md5.h	2008-05-24 02:38:26.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-filter-md5.h	2008-12-05 10:19:43.000000000 +0100
@@ -22,7 +22,7 @@
 #ifndef __GMIME_FILTER_MD5_H__
 #define __GMIME_FILTER_MD5_H__
 
-#include <gmime/gmime-filter.h>
+#include "gmime-filter.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-filter-strip.h tracker-0.6.6new/src/trackerd/gmime-filter-strip.h
--- tracker-0.6.6/src/trackerd/gmime-filter-strip.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-filter-strip.h	2008-12-05 10:19:55.000000000 +0100
@@ -22,7 +22,7 @@
 #ifndef __GMIME_FILTER_STRIP_H__
 #define __GMIME_FILTER_STRIP_H__
 
-#include <gmime/gmime-filter.h>
+#include "gmime-filter.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-filter-windows.h tracker-0.6.6new/src/trackerd/gmime-filter-windows.h
--- tracker-0.6.6/src/trackerd/gmime-filter-windows.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-filter-windows.h	2008-12-05 10:20:06.000000000 +0100
@@ -22,7 +22,7 @@
 #ifndef __GMIME_FILTER_WINDOWS_H__
 #define __GMIME_FILTER_WINDOWS_H__
 
-#include <gmime/gmime-filter.h>
+#include "gmime-filter.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-filter-yenc.h tracker-0.6.6new/src/trackerd/gmime-filter-yenc.h
--- tracker-0.6.6/src/trackerd/gmime-filter-yenc.h	2008-05-24 02:38:26.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-filter-yenc.h	2008-12-05 10:20:19.000000000 +0100
@@ -22,7 +22,7 @@
 #ifndef __GMIME_FILTER_YENC_H__
 #define __GMIME_FILTER_YENC_H__
 
-#include <gmime/gmime-filter.h>
+#include "gmime-filter.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-gpg-context.h tracker-0.6.6new/src/trackerd/gmime-gpg-context.h
--- tracker-0.6.6/src/trackerd/gmime-gpg-context.h	2008-05-24 02:38:26.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-gpg-context.h	2008-12-05 10:20:30.000000000 +0100
@@ -22,7 +22,7 @@
 #ifndef __GMIME_GPG_CONTEXT_H__
 #define __GMIME_GPG_CONTEXT_H__
 
-#include <gmime/gmime-cipher-context.h>
+#include "gmime-cipher-context.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime.h tracker-0.6.6new/src/trackerd/gmime.h
--- tracker-0.6.6/src/trackerd/gmime.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime.h	2008-12-05 10:12:31.000000000 +0100
@@ -23,51 +23,51 @@
 #define __GMIME_H__
 
 #include <glib.h>
-#include <gmime/gmime-error.h>
-#include <gmime/gmime-charset.h>
-#include <gmime/gmime-iconv.h>
-#include <gmime/gmime-iconv-utils.h>
-#include <gmime/gmime-param.h>
-#include <gmime/gmime-content-type.h>
-#include <gmime/gmime-disposition.h>
-#include <gmime/gmime-data-wrapper.h>
-#include <gmime/gmime-object.h>
-#include <gmime/gmime-part.h>
-#include <gmime/gmime-multipart.h>
-#include <gmime/gmime-multipart-encrypted.h>
-#include <gmime/gmime-multipart-signed.h>
-#include <gmime/gmime-message.h>
-#include <gmime/gmime-message-part.h>
-#include <gmime/gmime-message-partial.h>
-#include <gmime/internet-address.h>
-#include <gmime/gmime-parser.h>
-#include <gmime/gmime-utils.h>
-#include <gmime/gmime-stream.h>
-#include <gmime/gmime-stream-buffer.h>
-#include <gmime/gmime-stream-cat.h>
-#include <gmime/gmime-stream-file.h>
-#include <gmime/gmime-stream-filter.h>
-#include <gmime/gmime-stream-fs.h>
-#include <gmime/gmime-stream-mem.h>
-#include <gmime/gmime-stream-mmap.h>
-#include <gmime/gmime-stream-null.h>
-#include <gmime/gmime-filter.h>
-#include <gmime/gmime-filter-basic.h>
-#include <gmime/gmime-filter-best.h>
-#include <gmime/gmime-filter-charset.h>
-#include <gmime/gmime-filter-crlf.h>
-#include <gmime/gmime-filter-enriched.h>
-#include <gmime/gmime-filter-from.h>
-#include <gmime/gmime-filter-gzip.h>
-#include <gmime/gmime-filter-html.h>
-#include <gmime/gmime-filter-md5.h>
-#include <gmime/gmime-filter-strip.h>
-#include <gmime/gmime-filter-windows.h>
-#include <gmime/gmime-filter-yenc.h>
-#include <gmime/gmime-session.h>
-#include <gmime/gmime-session-simple.h>
-#include <gmime/gmime-cipher-context.h>
-#include <gmime/gmime-gpg-context.h>
+#include "gmime-error.h"
+#include "gmime-charset.h"
+#include "gmime-iconv.h"
+#include "gmime-iconv-utils.h"
+#include "gmime-param.h"
+#include "gmime-content-type.h"
+#include "gmime-disposition.h"
+#include "gmime-data-wrapper.h"
+#include "gmime-object.h"
+#include "gmime-part.h"
+#include "gmime-multipart.h"
+#include "gmime-multipart-encrypted.h"
+#include "gmime-multipart-signed.h"
+#include "gmime-message.h"
+#include "gmime-message-part.h"
+#include "gmime-message-partial.h"
+#include "internet-address.h"
+#include "gmime-parser.h"
+#include "gmime-utils.h"
+#include "gmime-stream.h"
+#include "gmime-stream-buffer.h"
+#include "gmime-stream-cat.h"
+#include "gmime-stream-file.h"
+#include "gmime-stream-filter.h"
+#include "gmime-stream-fs.h"
+#include "gmime-stream-mem.h"
+#include "gmime-stream-mmap.h"
+#include "gmime-stream-null.h"
+#include "gmime-filter.h"
+#include "gmime-filter-basic.h"
+#include "gmime-filter-best.h"
+#include "gmime-filter-charset.h"
+#include "gmime-filter-crlf.h"
+#include "gmime-filter-enriched.h"
+#include "gmime-filter-from.h"
+#include "gmime-filter-gzip.h"
+#include "gmime-filter-html.h"
+#include "gmime-filter-md5.h"
+#include "gmime-filter-strip.h"
+#include "gmime-filter-windows.h"
+#include "gmime-filter-yenc.h"
+#include "gmime-session.h"
+#include "gmime-session-simple.h"
+#include "gmime-cipher-context.h"
+#include "gmime-gpg-context.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-header.h tracker-0.6.6new/src/trackerd/gmime-header.h
--- tracker-0.6.6/src/trackerd/gmime-header.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-header.h	2008-12-05 10:20:41.000000000 +0100
@@ -23,7 +23,7 @@
 #define __GMIME_HEADER_H__
 
 #include <glib.h>
-#include <gmime/gmime-stream.h>
+#include "gmime-stream.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-message.h tracker-0.6.6new/src/trackerd/gmime-message.h
--- tracker-0.6.6/src/trackerd/gmime-message.h	2008-07-05 01:05:53.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-message.h	2008-12-05 10:21:15.000000000 +0100
@@ -25,10 +25,10 @@
 #include <stdarg.h>
 #include <time.h>
 
-#include <gmime/gmime-object.h>
-#include <gmime/gmime-header.h>
-#include <gmime/gmime-stream.h>
-#include <gmime/internet-address.h>
+#include "gmime-object.h"
+#include "gmime-header.h"
+#include "gmime-stream.h"
+#include "internet-address.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-message-part.h tracker-0.6.6new/src/trackerd/gmime-message-part.h
--- tracker-0.6.6/src/trackerd/gmime-message-part.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-message-part.h	2008-12-05 10:27:02.000000000 +0100
@@ -22,8 +22,8 @@
 #ifndef __GMIME_MESSAGE_PART_H__
 #define __GMIME_MESSAGE_PART_H__
 
-#include <gmime/gmime-object.h>
-#include <gmime/gmime-message.h>
+#include "gmime-object.h"
+#include "gmime-message.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-message-partial.h tracker-0.6.6new/src/trackerd/gmime-message-partial.h
--- tracker-0.6.6/src/trackerd/gmime-message-partial.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-message-partial.h	2008-12-05 10:21:32.000000000 +0100
@@ -24,8 +24,8 @@
 
 #include <glib.h>
 
-#include <gmime/gmime-part.h>
-#include <gmime/gmime-message.h>
+#include "gmime-part.h"
+#include "gmime-message.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-multipart-encrypted.h tracker-0.6.6new/src/trackerd/gmime-multipart-encrypted.h
--- tracker-0.6.6/src/trackerd/gmime-multipart-encrypted.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-multipart-encrypted.h	2008-12-05 10:21:57.000000000 +0100
@@ -22,8 +22,8 @@
 #ifndef __GMIME_MULTIPART_ENCRYPTED_H__
 #define __GMIME_MULTIPART_ENCRYPTED_H__
 
-#include <gmime/gmime-multipart.h>
-#include <gmime/gmime-cipher-context.h>
+#include "gmime-multipart.h"
+#include "gmime-cipher-context.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-multipart.h tracker-0.6.6new/src/trackerd/gmime-multipart.h
--- tracker-0.6.6/src/trackerd/gmime-multipart.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-multipart.h	2008-12-05 10:21:42.000000000 +0100
@@ -24,7 +24,7 @@
 
 #include <glib.h>
 
-#include <gmime/gmime-object.h>
+#include "gmime-object.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-multipart-signed.h tracker-0.6.6new/src/trackerd/gmime-multipart-signed.h
--- tracker-0.6.6/src/trackerd/gmime-multipart-signed.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-multipart-signed.h	2008-12-05 10:22:12.000000000 +0100
@@ -22,8 +22,8 @@
 #ifndef __GMIME_MULTIPART_SIGNED_H__
 #define __GMIME_MULTIPART_SIGNED_H__
 
-#include <gmime/gmime-multipart.h>
-#include <gmime/gmime-cipher-context.h>
+#include "gmime-multipart.h"
+#include "gmime-cipher-context.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-object.h tracker-0.6.6new/src/trackerd/gmime-object.h
--- tracker-0.6.6/src/trackerd/gmime-object.h	2008-05-24 02:38:26.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-object.h	2008-12-05 10:22:31.000000000 +0100
@@ -25,9 +25,9 @@
 #include <glib.h>
 #include <glib-object.h>
 
-#include <gmime/gmime-content-type.h>
-#include <gmime/gmime-stream.h>
-#include <gmime/gmime-header.h>
+#include "gmime-content-type.h"
+#include "gmime-stream.h"
+#include "gmime-header.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-parser.h tracker-0.6.6new/src/trackerd/gmime-parser.h
--- tracker-0.6.6/src/trackerd/gmime-parser.h	2008-09-13 15:24:40.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-parser.h	2008-12-05 10:23:04.000000000 +0100
@@ -26,10 +26,10 @@
 #include <glib-object.h>
 #include <errno.h>
 
-#include <gmime/gmime-object.h>
-#include <gmime/gmime-message.h>
-#include <gmime/gmime-content-type.h>
-#include <gmime/gmime-stream.h>
+#include "gmime-object.h"
+#include "gmime-message.h"
+#include "gmime-content-type.h"
+#include "gmime-stream.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-part.h tracker-0.6.6new/src/trackerd/gmime-part.h
--- tracker-0.6.6/src/trackerd/gmime-part.h	2008-05-26 14:14:23.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-part.h	2008-12-05 10:23:33.000000000 +0100
@@ -25,10 +25,10 @@
 #include <glib.h>
 #include <stdio.h>
 
-#include <gmime/gmime-object.h>
-#include <gmime/gmime-param.h>
-#include <gmime/gmime-disposition.h>
-#include <gmime/gmime-data-wrapper.h>
+#include "gmime-object.h"
+#include "gmime-param.h"
+#include "gmime-disposition.h"
+#include "gmime-data-wrapper.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-session-simple.h tracker-0.6.6new/src/trackerd/gmime-session-simple.h
--- tracker-0.6.6/src/trackerd/gmime-session-simple.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-session-simple.h	2008-12-05 10:23:51.000000000 +0100
@@ -22,7 +22,7 @@
 #ifndef __GMIME_SESSION_SIMPLE_H__
 #define __GMIME_SESSION_SIMPLE_H__
 
-#include <gmime/gmime-session.h>
+#include "gmime-session.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-stream-buffer.h tracker-0.6.6new/src/trackerd/gmime-stream-buffer.h
--- tracker-0.6.6/src/trackerd/gmime-stream-buffer.h	2008-05-24 02:38:26.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-stream-buffer.h	2008-12-05 10:27:34.000000000 +0100
@@ -22,7 +22,7 @@
 #ifndef __GMIME_STREAM_BUFFER_H__
 #define __GMIME_STREAM_BUFFER_H__
 
-#include <gmime/gmime-stream.h>
+#include "gmime-stream.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-stream-cat.h tracker-0.6.6new/src/trackerd/gmime-stream-cat.h
--- tracker-0.6.6/src/trackerd/gmime-stream-cat.h	2008-05-24 02:38:26.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-stream-cat.h	2008-12-05 10:24:15.000000000 +0100
@@ -23,7 +23,7 @@
 #define __GMIME_STREAM_CAT_H__
 
 #include <glib.h>
-#include <gmime/gmime-stream.h>
+#include "gmime-stream.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-stream-file.h tracker-0.6.6new/src/trackerd/gmime-stream-file.h
--- tracker-0.6.6/src/trackerd/gmime-stream-file.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-stream-file.h	2008-12-05 10:24:27.000000000 +0100
@@ -24,7 +24,7 @@
 
 #include <glib.h>
 #include <stdio.h>
-#include <gmime/gmime-stream.h>
+#include "gmime-stream.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-stream-filter.h tracker-0.6.6new/src/trackerd/gmime-stream-filter.h
--- tracker-0.6.6/src/trackerd/gmime-stream-filter.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-stream-filter.h	2008-12-05 10:24:40.000000000 +0100
@@ -22,8 +22,8 @@
 #ifndef __GMIME_STREAM_FILTER_H__
 #define __GMIME_STREAM_FILTER_H__
 
-#include <gmime/gmime-stream.h>
-#include <gmime/gmime-filter.h>
+#include "gmime-stream.h"
+#include "gmime-filter.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-stream-fs.h tracker-0.6.6new/src/trackerd/gmime-stream-fs.h
--- tracker-0.6.6/src/trackerd/gmime-stream-fs.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-stream-fs.h	2008-12-05 10:24:51.000000000 +0100
@@ -23,7 +23,7 @@
 #define __GMIME_STREAM_FS_H__
 
 #include <unistd.h>
-#include <gmime/gmime-stream.h>
+#include "gmime-stream.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-stream-mem.h tracker-0.6.6new/src/trackerd/gmime-stream-mem.h
--- tracker-0.6.6/src/trackerd/gmime-stream-mem.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-stream-mem.h	2008-12-05 10:25:01.000000000 +0100
@@ -23,7 +23,7 @@
 #define __GMIME_STREAM_MEM_H__
 
 #include <glib.h>
-#include <gmime/gmime-stream.h>
+#include "gmime-stream.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-stream-mmap.h tracker-0.6.6new/src/trackerd/gmime-stream-mmap.h
--- tracker-0.6.6/src/trackerd/gmime-stream-mmap.h	2008-05-24 02:38:27.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-stream-mmap.h	2008-12-05 10:25:13.000000000 +0100
@@ -22,7 +22,7 @@
 #ifndef __GMIME_STREAM_MMAP_H__
 #define __GMIME_STREAM_MMAP_H__
 
-#include <gmime/gmime-stream.h>
+#include "gmime-stream.h"
 
 G_BEGIN_DECLS
 
diff -u tracker-0.6.6/src/trackerd/gmime-stream-null.h tracker-0.6.6new/src/trackerd/gmime-stream-null.h
--- tracker-0.6.6/src/trackerd/gmime-stream-null.h	2008-05-24 02:38:26.000000000 +0200
+++ tracker-0.6.6new/src/trackerd/gmime-stream-null.h	2008-12-05 10:25:25.000000000 +0100
@@ -23,7 +23,7 @@
 #define __GMIME_STREAM_NULL_H__
 
 #include <glib.h>
-#include <gmime/gmime-stream.h>
+#include "gmime-stream.h"
 
 G_BEGIN_DECLS
 

diff -u tracker-0.6.6/src/trackerd/Makefile.am tracker-0.6.6new/src/trackerd/Makefile.am
--- tracker-0.6.6/src/trackerd/Makefile.am	2008-01-15 05:05:03.000000000 +0100
+++ tracker-0.6.6new/src/trackerd/Makefile.am	2008-12-05 11:53:01.000000000 +0100
@@ -4,7 +4,6 @@
 	-DTRACKER_LOCALEDIR=\""$(localedir)"\" \
 	$(GLIB2_CFLAGS)			\
 	$(PANGO_CFLAGS)			\
-	$(GMIME_CFLAGS)			\
 	$(HAL_CFLAGS)			\
 	$(FAM_CFLAGS)			\
 	$(DBUS_CFLAGS)			\
@@ -93,7 +92,120 @@
 	tracker-ioprio.c		\
 	tracker-ioprio.h		\
 	tracker-os-dependant.h		\
-	tracker-watch.h
+	tracker-watch.h			\
+	charset-map.c	\
+	gen-table.c	\
+	gmime.c	\
+	gmime.h	\
+	gmime-charset.c	\
+	gmime-charset.h	\
+	gmime-charset-map-private.h	\
+	gmime-cipher-context.c	\
+	gmime-cipher-context.h	\
+	gmime-common.c	\
+	gmime-common.h	\
+	gmime-content-type.c	\
+	gmime-content-type.h	\
+	gmime-data-wrapper.c	\
+	gmime-data-wrapper.h	\
+	gmime-disposition.c	\
+	gmime-disposition.h	\
+	gmime-error.h	\
+	gmime-filter.c	\
+	gmime-filter.h	\
+	gmime-filter-basic.c	\
+	gmime-filter-basic.h	\
+	gmime-filter-best.c	\
+	gmime-filter-best.h	\
+	gmime-filter-charset.c	\
+	gmime-filter-charset.h	\
+	gmime-filter-crlf.c	\
+	gmime-filter-crlf.h	\
+	gmime-filter-enriched.c	\
+	gmime-filter-enriched.h	\
+	gmime-filter-from.c	\
+	gmime-filter-from.h	\
+	gmime-filter-gzip.c	\
+	gmime-filter-gzip.h	\
+	gmime-filter-html.c	\
+	gmime-filter-html.h	\
+	gmime-filter-md5.c	\
+	gmime-filter-md5.h	\
+	gmime-filter-strip.c	\
+	gmime-filter-strip.h	\
+	gmime-filter-windows.c	\
+	gmime-filter-windows.h	\
+	gmime-filter-yenc.c	\
+	gmime-filter-yenc.h	\
+	gmime-gpg-context.c	\
+	gmime-gpg-context.h	\
+	gmime-header.c	\
+	gmime-header.h	\
+	gmime-iconv.c	\
+	gmime-iconv.h	\
+	gmime-iconv-utils.c	\
+	gmime-iconv-utils.h	\
+	gmime-message.c	\
+	gmime-message.h	\
+	gmime-message-part.c	\
+	gmime-message-part.h	\
+	gmime-message-partial.c	\
+	gmime-message-partial.h	\
+	gmime-multipart.c	\
+	gmime-multipart.h	\
+	gmime-multipart-encrypted.c	\
+	gmime-multipart-encrypted.h	\
+	gmime-multipart-signed.c	\
+	gmime-multipart-signed.h	\
+	gmime-object.c	\
+	gmime-object.h	\
+	gmime-param.c	\
+	gmime-param.h	\
+	gmime-parser.c	\
+	gmime-parser.h	\
+	gmime-parse-utils.c	\
+	gmime-parse-utils.h	\
+	gmime-part.c	\
+	gmime-part.h	\
+	gmime-session.c	\
+	gmime-session.h	\
+	gmime-session-simple.c	\
+	gmime-session-simple.h	\
+	gmime-stream.c	\
+	gmime-stream.h	\
+	gmime-stream-buffer.c	\
+	gmime-stream-buffer.h	\
+	gmime-stream-cat.c	\
+	gmime-stream-cat.h	\
+	gmime-stream-file.c	\
+	gmime-stream-file.h	\
+	gmime-stream-filter.c	\
+	gmime-stream-filter.h	\
+	gmime-stream-fs.c	\
+	gmime-stream-fs.h	\
+	gmime-stream-mem.c	\
+	gmime-stream-mem.h	\
+	gmime-stream-mmap.c	\
+	gmime-stream-mmap.h	\
+	gmime-stream-null.c	\
+	gmime-stream-null.h	\
+	gmime-table-private.h	\
+	gmime-utils.c	\
+	gmime-utils.h	\
+	internet-address.c	\
+	internet-address.h \
+	cache.c \
+	cache.h \
+	gtrie.c \
+	gtrie.h \
+	list.c \
+	list.h \
+	md5-utils.c \
+	md5-utils.h \
+	memchunk.c \
+	memchunk.h \
+	url-scanner.c \
+	url-scanner.h
 
 if OS_WIN32
 trackerd_win_libs = -lws2_32 -lkernel32
@@ -104,7 +216,6 @@
 	$(FAM_LIBS)			\
 	$(DBUS_LIBS)			\
 	$(PANGO_LIBS)			\
-	$(GMIME_LIBS)			\
 	$(HAL_LIBS)			\
 	$(SQLITE3_LIBS)			\
 	$(QDBM_LIBS)			\

diff -u tracker-0.6.6/src/trackerd/tracker-email-evolution.h tracker-0.6.6new/src/trackerd/tracker-email-evolution.h
--- tracker-0.6.6/src/trackerd/tracker-email-evolution.h	2008-01-15 05:05:03.000000000 +0100
+++ tracker-0.6.6new/src/trackerd/tracker-email-evolution.h	2008-12-05 10:28:15.000000000 +0100
@@ -22,7 +22,7 @@
 #define _TRACKER_EMAIL_EVOLUTION_H_
 
 #include <glib.h>
-#include <gmime/gmime.h>
+#include "gmime.h"
 
 #include "config.h"
 
diff -u tracker-0.6.6/src/trackerd/tracker-email-kmail.h tracker-0.6.6new/src/trackerd/tracker-email-kmail.h
--- tracker-0.6.6/src/trackerd/tracker-email-kmail.h	2008-01-15 05:05:03.000000000 +0100
+++ tracker-0.6.6new/src/trackerd/tracker-email-kmail.h	2008-12-05 10:29:00.000000000 +0100
@@ -22,7 +22,7 @@
 #define _TRACKER_EMAIL_KMAIL_H_
 
 #include <glib.h>
-#include <gmime/gmime.h>
+#include "gmime.h"
 
 #include "config.h"
 
diff -u tracker-0.6.6/src/trackerd/tracker-email-thunderbird.h tracker-0.6.6new/src/trackerd/tracker-email-thunderbird.h
--- tracker-0.6.6/src/trackerd/tracker-email-thunderbird.h	2008-01-15 05:05:03.000000000 +0100
+++ tracker-0.6.6new/src/trackerd/tracker-email-thunderbird.h	2008-12-05 10:28:37.000000000 +0100
@@ -22,7 +22,7 @@
 #define _TRACKER_MBOX_THUNDERBIRD_H_
 
 #include <glib.h>
-#include <gmime/gmime.h>
+#include "gmime.h"
 
 #include "tracker-email-utils.h"
 
diff -u tracker-0.6.6/src/trackerd/tracker-email-utils.h tracker-0.6.6new/src/trackerd/tracker-email-utils.h
--- tracker-0.6.6/src/trackerd/tracker-email-utils.h	2008-01-15 05:05:03.000000000 +0100
+++ tracker-0.6.6new/src/trackerd/tracker-email-utils.h	2008-12-05 10:13:11.000000000 +0100
@@ -21,7 +21,7 @@
 #ifndef _TRACKER_EMAIL_BASE_H_
 #define _TRACKER_EMAIL_BASE_H_
 
-#include <gmime/gmime.h>
+#include "gmime.h"
 
 #include "tracker-email.h"
 
