diff -Naur kdebase-runtime-4.3.4/nepomuk/services/strigi/nepomukstrigiservice.desktop kdebase-runtime-4.3.4-p/nepomuk/services/strigi/nepomukstrigiservice.desktop
--- kdebase-runtime-4.3.4/nepomuk/services/strigi/nepomukstrigiservice.desktop	2009-10-02 10:18:57.000000000 +0200
+++ kdebase-runtime-4.3.4-p/nepomuk/services/strigi/nepomukstrigiservice.desktop	2010-01-20 10:09:30.000000000 +0100
@@ -2,7 +2,7 @@
 Type=Service
 X-KDE-ServiceTypes=NepomukService
 X-KDE-Library=nepomukstrigiservice
-X-KDE-Nepomuk-autostart=true
+X-KDE-Nepomuk-autostart=false
 X-KDE-Nepomuk-start-on-demand=false
 Name=Nepomuk Strigi Service
 Name[af]=Nepomuk Strigi diens
