diff -Naur workrave-1.8.5/backend/src/GlibIniConfigurator.cc workrave-1.8.5-p/backend/src/GlibIniConfigurator.cc
--- workrave-1.8.5/backend/src/GlibIniConfigurator.cc	2007-09-14 20:50:38.000000000 +0200
+++ workrave-1.8.5-p/backend/src/GlibIniConfigurator.cc	2008-04-06 17:38:47.000000000 +0200
@@ -29,6 +29,7 @@
 
 #include "GlibIniConfigurator.hh"
 #include <glib.h>
+#include <cstring>
 
 GlibIniConfigurator::GlibIniConfigurator()
   : config(NULL)
diff -Naur workrave-1.8.5/backend/src/Statistics.cc workrave-1.8.5-p/backend/src/Statistics.cc
--- workrave-1.8.5/backend/src/Statistics.cc	2007-09-14 20:50:38.000000000 +0200
+++ workrave-1.8.5-p/backend/src/Statistics.cc	2008-04-06 17:38:47.000000000 +0200
@@ -23,6 +23,7 @@
 
 #include <sstream>
 #include <assert.h>
+#include <cstring>
 
 #include "debug.hh"
 
diff -Naur workrave-1.8.5/backend/src/XMLConfigurator.cc workrave-1.8.5-p/backend/src/XMLConfigurator.cc
--- workrave-1.8.5/backend/src/XMLConfigurator.cc	2007-09-14 20:50:38.000000000 +0200
+++ workrave-1.8.5-p/backend/src/XMLConfigurator.cc	2008-04-06 17:38:35.000000000 +0200
@@ -23,7 +23,7 @@
 #include "debug.hh"
 #include <sstream>
 #include <iostream>
-
+#include <cstdlib>
 #include "XMLConfigurator.hh"
 
 using namespace std;
