diff -Naur breeze-icons-5.34.0/icons/index.theme breeze-icons-5.34.0-p/icons/index.theme
--- breeze-icons-5.34.0/icons/index.theme	2017-05-06 10:19:55.000000000 +0200
+++ breeze-icons-5.34.0-p/icons/index.theme	2017-06-04 00:40:21.194813344 +0200
@@ -90,7 +90,7 @@
 MainToolbarSizes=16,22,32,48
 SmallDefault=16
 SmallSizes=16,22,32,48
-PanelDefault=48
+PanelDefault=256
 PanelSizes=16,22,32,48,64,128,256
 DialogDefault=32
 DialogSizes=16,22,32,48,64,128,256
diff -Naur breeze-icons-5.34.0/icons-dark/index.theme breeze-icons-5.34.0-p/icons-dark/index.theme
--- breeze-icons-5.34.0/icons-dark/index.theme	2017-05-06 10:19:55.000000000 +0200
+++ breeze-icons-5.34.0-p/icons-dark/index.theme	2017-06-04 00:39:59.436229815 +0200
@@ -90,7 +90,7 @@
 MainToolbarSizes=16,22,32,48
 SmallDefault=16
 SmallSizes=16,22,32,48
-PanelDefault=48
+PanelDefault=256
 PanelSizes=16,22,32,48,64,128,256
 DialogDefault=32
 DialogSizes=16,22,32,48,64,128,256
