diff -Naur linux-5.1/include/acpi/platform/aclinux.h linux-5.1-p/include/acpi/platform/aclinux.h
--- linux-5.1/include/acpi/platform/aclinux.h	2019-05-06 02:42:58.000000000 +0200
+++ linux-5.1-p/include/acpi/platform/aclinux.h	2019-05-07 09:49:23.980444601 +0200
@@ -153,12 +153,12 @@
 #define ACPI_USE_ALTERNATE_PROTOTYPE_acpi_os_get_next_filename
 #define ACPI_USE_ALTERNATE_PROTOTYPE_acpi_os_close_directory
 
-#define ACPI_MSG_ERROR          KERN_ERR "ACPI Error: "
+#define ACPI_MSG_ERROR          KERN_WARNING "ACPI Error: "
 #define ACPI_MSG_EXCEPTION      KERN_ERR "ACPI Exception: "
 #define ACPI_MSG_WARNING        KERN_WARNING "ACPI Warning: "
 #define ACPI_MSG_INFO           KERN_INFO "ACPI: "
 
-#define ACPI_MSG_BIOS_ERROR     KERN_ERR "ACPI BIOS Error (bug): "
+#define ACPI_MSG_BIOS_ERROR     KERN_WARNING "ACPI BIOS Error (bug): "
 #define ACPI_MSG_BIOS_WARNING   KERN_WARNING "ACPI BIOS Warning (bug): "
 
 /*
