diff -Naur vlc-0.8.6e/modules/codec/speex.c fw_vlc-0.8.6e/modules/codec/speex.c
--- vlc-0.8.6e/modules/codec/speex.c	2007-11-21 23:23:29.000000000 +0100
+++ fw_vlc-0.8.6e/modules/codec/speex.c	2008-04-24 13:59:50.000000000 +0200
@@ -332,7 +332,7 @@
         msg_Err( p_dec, "cannot read Speex header" );
         return VLC_EGENERIC;
     }
-    if( p_header->mode >= SPEEX_NB_MODES )
+    if( p_header->mode >= SPEEX_NB_MODES || p_header->mode < 0 )
     {
         msg_Err( p_dec, "mode number %d does not (yet/any longer) exist in "
                  "this version of libspeex.", p_header->mode );
