diff -Naur vlc-0.8.6b/configure.ac vlc-0.8.6b-p/configure.ac
--- vlc-0.8.6b/configure.ac	2007-04-17 23:45:31.000000000 +0200
+++ vlc-0.8.6b-p/configure.ac	2007-05-25 20:48:05.000000000 +0200
@@ -2527,9 +2527,9 @@
    [
      if test "$with_ffmpeg_dts" = "yes"; then
              LDFLAGS="${LDFLAGS_save} ${LDFLAGS_ffmpeg}"
-             AC_CHECK_LIB(dts_pic, dts_free, 
-               [ VLC_ADD_LDFLAGS([ffmpeg],[-ldts_pic]) ],
-               [ VLC_ADD_LDFLAGS([ffmpeg],[-ldts]) ])
+             AC_CHECK_LIB(dca_pic, dca_free, 
+               [ VLC_ADD_LDFLAGS([ffmpeg],[-ldca_pic]) ],
+               [ VLC_ADD_LDFLAGS([ffmpeg],[-ldca]) ])
              LDFLAGS="${LDFLAGS_save}"
      fi])
 
@@ -2613,9 +2613,9 @@
    if fgrep -s "CONFIG_DTS=yes" "${real_ffmpeg_tree}/config.mak"; then
      if test "${with_ffmpeg_dts}" != "yes"; then
        LDFLAGS="${LDFLAGS_save} ${LDFLAGS_ffmpeg}"
-       AC_CHECK_LIB(dts_pic, dts_free, 
-         [ VLC_ADD_LDFLAGS([ffmpeg],[-ldts_pic]) ],
-         [ VLC_ADD_LDFLAGS([ffmpeg],[-ldts]) ])
+       AC_CHECK_LIB(dca_pic, dca_free, 
+         [ VLC_ADD_LDFLAGS([ffmpeg],[-ldca_pic]) ],
+         [ VLC_ADD_LDFLAGS([ffmpeg],[-ldca]) ])
        LDFLAGS="${LDFLAGS_save}"
      fi
    fi
@@ -3034,13 +3034,13 @@
       VLC_ADD_CPPFLAGS([dtstofloat32],[-I${real_dts_tree}/include])
       VLC_ADD_LDFLAGS([dtstofloat32],[-L${real_dts_tree}/libdts])
       LDFLAGS="${LDFLAGS_save} ${LDFLAGS_dtstofloat32}"
-      AC_CHECK_LIB(dts_pic, dts_free, [
+      AC_CHECK_LIB(dca_pic, dca_free, [
         VLC_ADD_PLUGINS([dtstofloat32])
-        VLC_ADD_LDFLAGS([dtstofloat32],[-ldts_pic])
+        VLC_ADD_LDFLAGS([dtstofloat32],[-ldca_pic])
         ],[
         AC_CHECK_LIB(dts, dts_free, [
           VLC_ADD_BUILTINS([dtstofloat32])
-          VLC_ADD_LDFLAGS([dtstofloat32],[-ldts])
+          VLC_ADD_LDFLAGS([dtstofloat32],[-ldca])
           ],[
           if test -f ${real_dts_tree}/libdts/libdts.a
           then
@@ -3058,13 +3058,13 @@
   else
     LDFLAGS="${LDFLAGS_save} ${LDFLAGS_dtstofloat32}"
     AC_CHECK_HEADERS(dts.h, [
-      AC_CHECK_LIB(dts_pic, dts_free, [
+      AC_CHECK_LIB(dca_pic, dca_free, [
         VLC_ADD_PLUGINS([dtstofloat32])
-        VLC_ADD_LDFLAGS([dtstofloat32],[-ldts_pic])
+        VLC_ADD_LDFLAGS([dtstofloat32],[-ldca_pic])
       ],[
-        AC_CHECK_LIB(dts, dts_free, [
+        AC_CHECK_LIB(dca, dca_free, [
           VLC_ADD_BUILTINS([dtstofloat32])
-          VLC_ADD_LDFLAGS([dtstofloat32],[-ldts])
+          VLC_ADD_LDFLAGS([dtstofloat32],[-ldca])
         ],[
           if test "${enable_dts}" = "yes"; then
             AC_MSG_ERROR([Could not find libdts on your system: you may get it from http://www.videolan.org/dtsdec.html])
