diff -Naur vlc-0.8.6e/modules/demux/mkv.cpp vlc-0.8.6e-p/modules/demux/mkv.cpp
--- vlc-0.8.6e/modules/demux/mkv.cpp	2008-02-24 20:01:53.000000000 +0100
+++ vlc-0.8.6e-p/modules/demux/mkv.cpp	2008-03-26 22:22:08.000000000 +0100
@@ -5845,7 +5845,7 @@
             binary *p_data = (*index)->GetBuffer();
             size_t i_size = *p_data++;
             // avoid reading too much from the buffer
-            i_size = min( i_size, ((*index)->GetSize() - 1) >> 3 );
+            i_size = __MIN( i_size, ((*index)->GetSize() - 1) >> 3 );
             for ( ; i_size > 0; i_size--, p_data += 8 )
             {
                 msg_Dbg( &sys.demuxer, "Matroska DVD enter command" );
@@ -5868,7 +5868,7 @@
             binary *p_data = (*index)->GetBuffer();
             size_t i_size = *p_data++;
             // avoid reading too much from the buffer
-            i_size = min( i_size, ((*index)->GetSize() - 1) >> 3 );
+            i_size = __MIN( i_size, ((*index)->GetSize() - 1) >> 3 );
             for ( ; i_size > 0; i_size--, p_data += 8 )
             {
                 msg_Dbg( &sys.demuxer, "Matroska DVD leave command" );
