--- vlc-1.1.5/modules/services_discovery/upnp_intel.cpp~	2010-05-04 21:20:21.000000000 +0200
+++ vlc-1.1.5/modules/services_discovery/upnp_intel.cpp	2010-11-14 23:28:25.885408960 +0100
@@ -813,7 +813,6 @@
         input_item_t* p_input_item = input_item_New( _p_sd,
                                                item->getResource(),
                                                item->getTitle() );
-        assert( p_input_item );
         input_item_node_AppendItem( p_input_node, p_input_item );
         item->setInputItem( p_input_item );
     }
