diff -aur vlc-0.8.6f/configure.ac fw_vlc-0.8.6f/configure.ac
--- vlc-0.8.6f/configure.ac	2008-03-23 23:41:48.000000000 +0100
+++ fw_vlc-0.8.6f/configure.ac	2008-04-28 16:45:57.000000000 +0200
@@ -2694,12 +2694,12 @@
  else
 
 dnl Trying with pkg-config
-   PKG_CHECK_MODULES(FFMPEG,[libavcodec, libavformat],
+   PKG_CHECK_MODULES(FFMPEG,[libavcodec, libavformat, libpostproc],
     [
-     AC_CHECK_HEADERS(ffmpeg/avcodec.h, [], [AC_MSG_ERROR([Missing header file ffmpeg/avcodec.h.])] )
-     AC_CHECK_HEADERS(ffmpeg/avformat.h)
-     AC_CHECK_HEADERS(ffmpeg/avutil.h)
-     AC_CHECK_HEADERS(postproc/postprocess.h, [], [AC_MSG_ERROR([Missing header file postproc/postprocess.h.])] )
+     AC_CHECK_HEADERS(libavcodec/avcodec.h, [], [AC_MSG_ERROR([Missing header file libavcodec/avcodec.h.])] )
+     AC_CHECK_HEADERS(libavformat/avformat.h)
+     AC_CHECK_HEADERS(libavutil/avutil.h)
+     AC_CHECK_HEADERS(libpostproc/postprocess.h, [], [AC_MSG_ERROR([Missing header file libpostproc/postprocess.h.])] )
      if test "${SYS}" = "darwin"; then
         VLC_ADD_BUILTINS([ffmpeg])
      else
diff -aur vlc-0.8.6f/modules/codec/ffmpeg/audio.c fw_vlc-0.8.6f/modules/codec/ffmpeg/audio.c
--- vlc-0.8.6f/modules/codec/ffmpeg/audio.c	2008-03-23 23:41:48.000000000 +0100
+++ fw_vlc-0.8.6f/modules/codec/ffmpeg/audio.c	2008-04-28 17:16:47.000000000 +0200
@@ -32,7 +32,7 @@
 #ifdef HAVE_FFMPEG_AVCODEC_H
 #   include <ffmpeg/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #include "ffmpeg.h"
diff -aur vlc-0.8.6f/modules/codec/ffmpeg/chroma.c fw_vlc-0.8.6f/modules/codec/ffmpeg/chroma.c
--- vlc-0.8.6f/modules/codec/ffmpeg/chroma.c	2008-03-23 23:41:48.000000000 +0100
+++ fw_vlc-0.8.6f/modules/codec/ffmpeg/chroma.c	2008-04-28 17:18:30.000000000 +0200
@@ -31,7 +31,7 @@
 #ifdef HAVE_FFMPEG_AVCODEC_H
 #   include <ffmpeg/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #include "ffmpeg.h"
diff -aur vlc-0.8.6f/modules/codec/ffmpeg/demux.c fw_vlc-0.8.6f/modules/codec/ffmpeg/demux.c
--- vlc-0.8.6f/modules/codec/ffmpeg/demux.c	2008-03-31 21:07:51.000000000 +0200
+++ fw_vlc-0.8.6f/modules/codec/ffmpeg/demux.c	2008-04-28 17:31:52.000000000 +0200
@@ -35,7 +35,7 @@
 #ifdef HAVE_FFMPEG_AVFORMAT_H
 #   include <ffmpeg/avformat.h>
 #else
-#   include <avformat.h>
+#   include <libavformat/avformat.h>
 #endif
 
 #include "ffmpeg.h"
diff -aur vlc-0.8.6f/modules/codec/ffmpeg/encoder.c fw_vlc-0.8.6f/modules/codec/ffmpeg/encoder.c
--- vlc-0.8.6f/modules/codec/ffmpeg/encoder.c	2008-03-23 23:41:48.000000000 +0100
+++ fw_vlc-0.8.6f/modules/codec/ffmpeg/encoder.c	2008-04-28 17:32:19.000000000 +0200
@@ -39,7 +39,7 @@
 #ifdef HAVE_FFMPEG_AVCODEC_H
 #   include <ffmpeg/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #if LIBAVCODEC_BUILD < 4704
diff -aur vlc-0.8.6f/modules/codec/ffmpeg/ffmpeg.c fw_vlc-0.8.6f/modules/codec/ffmpeg/ffmpeg.c
--- vlc-0.8.6f/modules/codec/ffmpeg/ffmpeg.c	2008-03-23 23:41:48.000000000 +0100
+++ fw_vlc-0.8.6f/modules/codec/ffmpeg/ffmpeg.c	2008-04-28 17:32:40.000000000 +0200
@@ -33,7 +33,7 @@
 #ifdef HAVE_FFMPEG_AVCODEC_H
 #   include <ffmpeg/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #if LIBAVCODEC_BUILD < 4680
diff -aur vlc-0.8.6f/modules/codec/ffmpeg/mux.c fw_vlc-0.8.6f/modules/codec/ffmpeg/mux.c
--- vlc-0.8.6f/modules/codec/ffmpeg/mux.c	2008-03-23 23:41:48.000000000 +0100
+++ fw_vlc-0.8.6f/modules/codec/ffmpeg/mux.c	2008-04-28 17:33:16.000000000 +0200
@@ -34,7 +34,7 @@
 #ifdef HAVE_FFMPEG_AVFORMAT_H
 #   include <ffmpeg/avformat.h>
 #else
-#   include <avformat.h>
+#   include <libavformat/avformat.h>
 #endif
 
 #include "ffmpeg.h"
diff -aur vlc-0.8.6f/modules/codec/ffmpeg/postprocess.c fw_vlc-0.8.6f/modules/codec/ffmpeg/postprocess.c
--- vlc-0.8.6f/modules/codec/ffmpeg/postprocess.c	2008-03-23 23:41:48.000000000 +0100
+++ fw_vlc-0.8.6f/modules/codec/ffmpeg/postprocess.c	2008-04-28 17:33:28.000000000 +0200
@@ -30,7 +30,7 @@
 #ifdef HAVE_FFMPEG_AVCODEC_H
 #   include <ffmpeg/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #include "ffmpeg.h"
diff -aur vlc-0.8.6f/modules/codec/ffmpeg/video.c fw_vlc-0.8.6f/modules/codec/ffmpeg/video.c
--- vlc-0.8.6f/modules/codec/ffmpeg/video.c	2008-03-23 23:41:48.000000000 +0100
+++ fw_vlc-0.8.6f/modules/codec/ffmpeg/video.c	2008-04-28 17:07:14.000000000 +0200
@@ -33,7 +33,7 @@
 #ifdef HAVE_FFMPEG_AVCODEC_H
 #   include <ffmpeg/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #include "ffmpeg.h"
diff -aur vlc-0.8.6f/modules/codec/ffmpeg/video_filter.c fw_vlc-0.8.6f/modules/codec/ffmpeg/video_filter.c
--- vlc-0.8.6f/modules/codec/ffmpeg/video_filter.c	2008-03-23 23:41:48.000000000 +0100
+++ fw_vlc-0.8.6f/modules/codec/ffmpeg/video_filter.c	2008-04-28 17:33:59.000000000 +0200
@@ -33,7 +33,7 @@
 #ifdef HAVE_FFMPEG_AVCODEC_H
 #   include <ffmpeg/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #include "ffmpeg.h"
diff -aur vlc-0.8.6f/modules/stream_out/switcher.c fw_vlc-0.8.6f/modules/stream_out/switcher.c
--- vlc-0.8.6f/modules/stream_out/switcher.c	2008-03-23 23:41:49.000000000 +0100
+++ fw_vlc-0.8.6f/modules/stream_out/switcher.c	2008-04-28 18:28:31.000000000 +0200
@@ -39,7 +39,7 @@
 #ifdef HAVE_FFMPEG_AVCODEC_H
 #   include <ffmpeg/avcodec.h>
 #else
-#   include <avcodec.h>
+#   include <libavcodec/avcodec.h>
 #endif
 
 #ifdef HAVE_POSTPROC_POSTPROCESS_H
