diff -Naur vlc-0.8.4.orig/configure.ac vlc-0.8.4/configure.ac
--- vlc-0.8.4.orig/configure.ac	2005-11-26 17:39:58.000000000 +0100
+++ vlc-0.8.4/configure.ac	2006-02-25 18:41:17.000000000 +0100
@@ -2204,7 +2204,7 @@
 	fi])
 
 dnl Trying with pkg-config
-   PKG_CHECK_MODULES(FFMPEG,[libavcodec, libavformat],
+   PKG_CHECK_MODULES(FFMPEG,[libavcodec, libavformat, libpostproc],
     [
      AC_CHECK_HEADERS(ffmpeg/avcodec.h)
      AC_CHECK_HEADERS(postproc/postprocess.h)
diff -Naur vlc-0.8.4.orig/modules/codec/ffmpeg/ffmpeg.c vlc-0.8.4/modules/codec/ffmpeg/ffmpeg.c
--- vlc-0.8.4.orig/modules/codec/ffmpeg/ffmpeg.c	2005-11-26 17:39:53.000000000 +0100
+++ vlc-0.8.4/modules/codec/ffmpeg/ffmpeg.c	2006-02-25 18:22:18.000000000 +0100
@@ -46,7 +46,7 @@
 #   ifdef HAVE_POSTPROC_POSTPROCESS_H
 #       include <postproc/postprocess.h>
 #   else
-#       include <libpostproc/postprocess.h>
+#       include <ffmpeg/postprocess.h>
 #   endif
 #endif
 
diff -Naur vlc-0.8.4.orig/modules/codec/ffmpeg/postprocess.c vlc-0.8.4/modules/codec/ffmpeg/postprocess.c
--- vlc-0.8.4.orig/modules/codec/ffmpeg/postprocess.c	2005-11-26 17:39:53.000000000 +0100
+++ vlc-0.8.4/modules/codec/ffmpeg/postprocess.c	2006-02-25 18:22:18.000000000 +0100
@@ -40,7 +40,7 @@
 #ifdef HAVE_POSTPROC_POSTPROCESS_H
 #   include <postproc/postprocess.h>
 #else
-#   include <libpostproc/postprocess.h>
+#   include <ffmpeg/postprocess.h>
 #endif
 
 #ifndef PP_CPU_CAPS_ALTIVEC
diff -Naur vlc-0.8.4.orig/modules/stream_out/switcher.c vlc-0.8.4/modules/stream_out/switcher.c
--- vlc-0.8.4.orig/modules/stream_out/switcher.c	2005-11-26 17:39:51.000000000 +0100
+++ vlc-0.8.4/modules/stream_out/switcher.c	2006-02-25 18:22:42.000000000 +0100
@@ -44,7 +44,7 @@
 #ifdef HAVE_POSTPROC_POSTPROCESS_H
 #   include <postproc/postprocess.h>
 #else
-#   include <libpostproc/postprocess.h>
+#   include <ffmpeg/postprocess.h>
 #endif
 
 #define SOUT_CFG_PREFIX "sout-switcher-"
