From 2deff133ff1f301edd5e4110c037d3ca65bd6ee8 Mon Sep 17 00:00:00 2001
From: Yanko Kaneti <yaneti@declera.com>
Date: Wed, 10 Nov 2010 09:49:44 +0000
Subject: background: one last old gnome_bg remnant

---
diff --git a/panels/background/bg-flickr-source.c b/panels/background/bg-flickr-source.c
index 3a0aa08..2e164e6 100644
--- a/panels/background/bg-flickr-source.c
+++ b/panels/background/bg-flickr-source.c
@@ -96,7 +96,7 @@ _view_items_added_cb (SwClientItemView *item_view,
 
       item = g_new0 (GnomeWPItem, 1);
 
-      item->options = GNOME_BG_PLACEMENT_ZOOMED;
+      item->options = G_DESKTOP_BACKGROUND_STYLE_ZOOM;
       item->name = g_strdup (sw_item_get_value (sw_item, "title"));
       item->source_url = g_strdup (sw_item_get_value (sw_item,
                                                       "x-flickr-photo-url"));
--
cgit v0.8.3.1
