diff -Naur gnome-color-manager-3.26.0/src/meson.build gnome-color-manager-3.26.0-p/src/meson.build
--- gnome-color-manager-3.26.0/src/meson.build	2017-09-11 12:39:03.000000000 +0200
+++ gnome-color-manager-3.26.0-p/src/meson.build	2018-02-17 02:21:10.159673362 +0100
@@ -29,7 +29,7 @@
     ],
     c_args : cargs,
     install : true,
-    install_dir : 'libexec',
+    install_dir : get_option('libexecdir')
   )
 endif
 
