diff -Naur inews.orig/inews.c inews/inews.c
--- inews.orig/inews.c	1996-11-04 23:01:20.000000000 +0100
+++ inews/inews.c	2004-07-20 10:48:13.000000000 +0200
@@ -233,7 +233,7 @@
 	char	*cp;
 	struct	passwd	*passwd;
 	FILE	*fp;
-	char	*index(), *getenv();
+	char	*getenv();
 	int	count = 0;
 	char	*dotdir;
 	
@@ -310,7 +310,7 @@
 	char	*full_name;
 	char	*cp;
 	struct	passwd *passwd;
-	char	*index(), *getenv();
+	char	*getenv();
 
 	passwd = getpwuid(getuid());
 
@@ -438,7 +438,6 @@
 int valid_header(h)
 register char *h;
 {
-  char *index();
   char *colon, *space;
   
   /*
