--- cassava-0.5.3.1/cassava.cabal	2001-09-09 03:46:40.000000000 +0200
+++ cassava-0.5.3.1.new/cassava.cabal	2024-07-12 00:11:22.578498404 +0200
@@ -106,7 +106,7 @@
     , bytestring   >= 0.10.4   && < 0.13
     , containers   >= 0.4.2    && < 0.8
     , deepseq      >= 1.1      && < 1.6
-    , hashable                    < 1.5
+    , hashable                    < 1.6
     , scientific   >= 0.3.4.7  && < 0.4
     , text                        < 2.2
     , text-short   == 0.1.*
