--- tagged-0.8.6/tagged.cabal	2018-07-03 00:53:28.000000000 +0200
+++ tagged-0.8.6.new/tagged.cabal	2019-11-26 14:40:30.601596732 +0100
@@ -13,7 +13,7 @@
 description:    Haskell 98 phantom types to avoid unsafely passing dummy arguments.
 build-type:     Simple
 cabal-version:  >= 1.10
-extra-source-files: .travis.yml CHANGELOG.markdown README.markdown HLint.hs
+extra-source-files: .hlint.yaml .travis.yml CHANGELOG.markdown README.markdown
 tested-with:   GHC == 7.0.4
              , GHC == 7.2.2
              , GHC == 7.4.2
@@ -22,8 +22,10 @@
              , GHC == 7.10.3
              , GHC == 8.0.2
              , GHC == 8.2.2
-             , GHC == 8.4.3
-             , GHC == 8.6.1
+             , GHC == 8.4.4
+             , GHC == 8.6.5
+             , GHC == 8.8.1
+             , GHC == 8.10.1
 
 source-repository head
   type: git
@@ -67,7 +69,7 @@
 
   if impl(ghc>=7.6)
     exposed-modules: Data.Proxy.TH
-    build-depends: template-haskell >= 2.8 && < 2.15
+    build-depends: template-haskell >= 2.8 && < 2.17
 
   if flag(deepseq)
     build-depends: deepseq >= 1.1 && < 1.5
