--- tagged-0.8.8/tagged.cabal	2001-09-09 03:46:40.000000000 +0200
+++ tagged-0.8.8.new/tagged.cabal	2024-07-09 18:58:07.288093274 +0200
@@ -14,26 +14,29 @@
 build-type:     Simple
 cabal-version:  >= 1.10
 extra-source-files: .hlint.yaml CHANGELOG.markdown README.markdown
-tested-with:   GHC == 7.0.4
-             , GHC == 7.2.2
-             , GHC == 7.4.2
-             , GHC == 7.6.3
-             , GHC == 7.8.4
-             , GHC == 7.10.3
-             , GHC == 8.0.2
-             , GHC == 8.2.2
-             , GHC == 8.4.4
-             , GHC == 8.6.5
-             , GHC == 8.8.4
-             , GHC == 8.10.7
-             , GHC == 9.0.2
-             , GHC == 9.2.8
-             , GHC == 9.4.5
-             , GHC == 9.6.2
+tested-with:
+  GHC == 7.0.4
+  GHC == 7.2.2
+  GHC == 7.4.2
+  GHC == 7.6.3
+  GHC == 7.8.4
+  GHC == 7.10.3
+  GHC == 8.0.2
+  GHC == 8.2.2
+  GHC == 8.4.4
+  GHC == 8.6.5
+  GHC == 8.8.4
+  GHC == 8.10.7
+  GHC == 9.0.2
+  GHC == 9.2.8
+  GHC == 9.4.8
+  GHC == 9.6.4
+  GHC == 9.8.2
+  GHC == 9.10.0
 
 source-repository head
   type: git
-  location: git://github.com/ekmett/tagged.git
+  location: https://github.com/ekmett/tagged.git
 
 flag deepseq
   description:
@@ -78,7 +81,7 @@
 
   if impl(ghc>=7.6)
     exposed-modules: Data.Proxy.TH
-    build-depends: template-haskell >= 2.8 && < 2.22
+    build-depends: template-haskell >= 2.8 && < 2.23
 
   if flag(deepseq)
     build-depends: deepseq >= 1.1 && < 1.6
