diff -Naur qtbase-opensource-src-5.6.0/src/gui/kernel/qguiapplication.cpp qtbase-opensource-src-5.6.0-p/src/gui/kernel/qguiapplication.cpp
--- qtbase-opensource-src-5.6.0/src/gui/kernel/qguiapplication.cpp	2016-03-03 12:35:22.000000000 +0100
+++ qtbase-opensource-src-5.6.0-p/src/gui/kernel/qguiapplication.cpp	2016-04-11 16:19:12.118425838 +0200
@@ -3294,11 +3294,6 @@
     is_saving_session = true;
 
     emit q->commitDataRequest(*session_manager);
-    if (is_fallback_session_management_enabled && session_manager->allowsInteraction()
-        && !tryCloseAllWindows()) {
-        session_manager->cancel();
-    }
-
     is_saving_session = false;
 }
 
