diff -ur a/config.in b/config.in
--- a/config.in	2013-02-19 03:28:11.000000000 -0600
+++ b/config.in	2013-04-21 15:59:45.966465443 -0500
@@ -42,7 +42,7 @@
 * course, welcome.  Answer `n' here if you have no support for
 * internationalization on your system.
 *
-bool 'Does your system support GNU gettext?' I18N n
+bool 'Does your system support GNU gettext?' I18N y
 *
 *
 *             Protocol Families.
@@ -59,7 +59,7 @@
 bool 'Econet protocol family' HAVE_AFECONET y
 bool 'DECnet protocol family' HAVE_AFDECnet n
 bool 'Ash protocol family' HAVE_AFASH y
-bool 'Bluetooth protocol family' HAVE_AFBLUETOOTH n
+bool 'Bluetooth protocol family' HAVE_AFBLUETOOTH y
 *
 *
 *            Device Hardware types.
