diff -ur a/Makefile b/Makefile
--- a/Makefile	2013-02-19 03:28:11.000000000 -0600
+++ b/Makefile	2013-04-21 16:01:51.910460760 -0500
@@ -198,12 +198,12 @@
 	install -m 0755 -d ${BASEDIR}/bin
 	install -m 0755 arp        ${BASEDIR}/sbin
 	install -m 0755 hostname   ${BASEDIR}/bin
-	install -m 0755 ifconfig   ${BASEDIR}/bin
+	install -m 0755 ifconfig   ${BASEDIR}/sbin
 	install -m 0755 nameif     ${BASEDIR}/sbin
 	install -m 0755 netstat    ${BASEDIR}/bin
 	install -m 0755 plipconfig $(BASEDIR)/sbin
 	install -m 0755 rarp       ${BASEDIR}/sbin
-	install -m 0755 route      ${BASEDIR}/bin
+	install -m 0755 route      ${BASEDIR}/sbin
 	install -m 0755 slattach   $(BASEDIR)/sbin
 ifeq ($(HAVE_IP_TOOLS),1)
 	install -m 0755 ipmaddr    $(BASEDIR)/sbin
