diff --git a/Makefile b/Makefile
index bf67bc6..1986147 100644
--- a/Makefile
+++ b/Makefile
@@ -206,17 +206,12 @@ mii-tool:	$(NET_LIB) mii-tool.o
 		$(CC) $(CFLAGS) $(LDFLAGS) -o $@ mii-tool.o $(NLIB) $(LDLIBS)
 
 installbin:
-	@echo
-	@echo "######################################################"
-	@echo "Notice: ifconfig and route are now installed into /bin"
-	@echo "######################################################"
-	@echo
 	install -m 0755 -d ${BASEDIR}${SBINDIR}
 	install -m 0755 -d ${BASEDIR}${BINDIR}
-	install -m 0755 ifconfig   ${BASEDIR}${BINDIR}
+	install -m 0755 ifconfig   ${BASEDIR}${SBINDIR}
 	install -m 0755 nameif     ${BASEDIR}${SBINDIR}
 	install -m 0755 netstat    ${BASEDIR}${BINDIR}
-	install -m 0755 route      ${BASEDIR}${BINDIR}
+	install -m 0755 route      ${BASEDIR}${SBINDIR}
 ifeq ($(HAVE_ARP_TOOLS),1)
 	install -m 0755 arp        ${BASEDIR}${SBINDIR}
 	install -m 0755 rarp       ${BASEDIR}${SBINDIR}
