diff -Naur atom-1.14.3/script/lib/verify-machine-requirements.js atom-1.14.3-p/script/lib/verify-machine-requirements.js
--- atom-1.14.3/script/lib/verify-machine-requirements.js	2017-02-17 17:53:06.000000000 +0100
+++ atom-1.14.3-p/script/lib/verify-machine-requirements.js	2017-02-22 00:26:38.032795147 +0100
@@ -17,10 +17,8 @@
 function verifyNode () {
   const fullVersion = process.versions.node
   const majorVersion = fullVersion.split('.')[0]
-  if (majorVersion >= 4 && majorVersion < 7) {
+  if (majorVersion >= 4) {
     console.log(`Node:\tv${fullVersion}`)
-  } else if (majorVersion >= 7) {
-    throw new Error(`Atom does not build properly on node v7+. node v${fullVersion} is installed.`)
   } else {
     throw new Error(`node v4+ is required to build Atom. node v${fullVersion} is installed.`)
   }
