From 8a645f78b45ae7e370c0dc2a52d0f2612aa6110b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan=20Star=C3=BD?= <hans@stare.cz>
Date: Thu, 21 Apr 2022 19:31:31 +0200
Subject: [PATCH] include utf8.h for utf8_decode()

---
 ogginfo/codec_skeleton.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ogginfo/codec_skeleton.c b/ogginfo/codec_skeleton.c
index a27f8da..2eefa39 100644
--- a/ogginfo/codec_skeleton.c
+++ b/ogginfo/codec_skeleton.c
@@ -25,7 +25,7 @@
 #include <ogg/ogg.h>
 
 #include "i18n.h"
-
+#include "utf8.h"
 #include "private.h"
 
 typedef struct {
