diff -Naur vorbis-tools-1.1.1/ogg123/speex_format.c fw_vorbis-tools-1.1.1/ogg123/speex_format.c
--- vorbis-tools-1.1.1/ogg123/speex_format.c	2005-06-20 12:01:34.000000000 +0200
+++ fw_vorbis-tools-1.1.1/ogg123/speex_format.c	2008-04-24 20:37:26.000000000 +0200
@@ -471,7 +471,7 @@
            cb->printf_error(callback_arg, ERROR, _("Cannot read header"));
      return NULL;
    }
-   if ((*header)->mode >= SPEEX_NB_MODES) {
+   if ((*header)->mode >= SPEEX_NB_MODES || (*header)->mode < 0) {
      cb->printf_error(callback_arg, ERROR, 
 		      _("Mode number %d does not (any longer) exist in this version"),
 	      (*header)->mode);
