--- bsb-http-chunked-0.0.0.4/bsb-http-chunked.cabal	2018-09-29 16:22:36.000000000 +0200
+++ bsb-http-chunked-0.0.0.4.new/bsb-http-chunked.cabal	2022-11-18 15:35:38.509428526 +0100
@@ -28,6 +28,8 @@
 Build-type:          Simple
 Cabal-version:       >= 1.8
 
+Tested-With:         GHC ==9.4.1 || ==9.2.4 || ==9.0.2 || ==8.10.7 || ==8.8.4 || ==8.6.5 || ==8.4.4 || ==8.2.2 || ==8.0.2 || ==7.10.3
+
 Extra-source-files:  CHANGELOG.md
 
 Source-repository head
@@ -36,8 +38,8 @@
 
 Library
   exposed-modules:   Data.ByteString.Builder.HTTP.Chunked
-  build-depends:     base >= 4.8 && < 4.13,
-                     bytestring >= 0.10.2 && < 0.11
+  build-depends:     base >= 4.8 && < 5,
+                     bytestring >= 0.10.2 && < 0.12
   ghc-options:       -Wall -O2
   if impl(ghc >= 8.0)
     ghc-options:     -Wcompat
@@ -73,7 +75,9 @@
                  , bsb-http-chunked
                  , bytestring
                  , deepseq
-                 , gauge
                  , semigroups
-  ghc-options: -O2 -Wall -rtsopts
+                 , tasty-bench
+  ghc-options: -O2 -Wall -rtsopts "-with-rtsopts=-A32m"
+  if impl(ghc >= 8.10)
+    ghc-options: "-with-rtsopts=-A32m --nonmoving-gc"
   type: exitcode-stdio-1.0
