diff -ur reiser4progs-1.0.4-orig/plugin/key/key_large/key_large.h reiser4progs-1.0.4/plugin/key/key_large/key_large.h
--- reiser4progs-1.0.4-orig/plugin/key/key_large/key_large.h	2005-03-12 14:12:04.000000000 -0500
+++ reiser4progs-1.0.4/plugin/key/key_large/key_large.h	2005-03-12 14:14:16.000000000 -0500
@@ -43,7 +43,7 @@
 
 typedef union key_large key_large_t;
 
-extern reiser4_plug_t key_large_plug;
+static reiser4_plug_t key_large_plug;
 
 typedef enum {
 	/* Major locality occupies higher 60 bits of the first element */
diff -ur reiser4progs-1.0.4-orig/plugin/key/key_short/key_short.h reiser4progs-1.0.4/plugin/key/key_short/key_short.h
--- reiser4progs-1.0.4-orig/plugin/key/key_short/key_short.h	2005-03-12 14:12:04.000000000 -0500
+++ reiser4progs-1.0.4/plugin/key/key_short/key_short.h	2005-03-12 14:14:04.000000000 -0500
@@ -42,7 +42,7 @@
 
 typedef union key_short key_short_t;
 
-extern reiser4_plug_t key_short_plug;
+static reiser4_plug_t key_short_plug;
 
 typedef enum {
 	/* Major locality occupies higher 60 bits of the first element */
diff -ur reiser4progs-1.0.4-orig/plugin/oid/oid40/oid40.h reiser4progs-1.0.4/plugin/oid/oid40/oid40.h
--- reiser4progs-1.0.4-orig/plugin/oid/oid40/oid40.h	2005-03-12 14:13:18.000000000 -0500
+++ reiser4progs-1.0.4/plugin/oid/oid40/oid40.h	2005-03-12 14:10:57.000000000 -0500
@@ -30,7 +30,7 @@
 
 typedef struct oid40 oid40_t;
 
-extern reiser4_plug_t oid40_plug;
+static reiser4_plug_t oid40_plug;
 
 #define oid40_get_next(area)            LE64_TO_CPU(*((d64_t *)area))
 #define oid40_set_next(area, val)       (*((d64_t *)area) = CPU_TO_LE64(val))
