diff -Naur dotconf-1.0.13.orig/dotconf.m4 dotconf-1.0.13/dotconf.m4
--- dotconf-1.0.13.orig/dotconf.m4	2002-09-17 20:08:08.000000000 +0200
+++ dotconf-1.0.13/dotconf.m4	2007-04-06 22:09:39.000000000 +0200
@@ -2,7 +2,7 @@
 dnl ## AM_PATH_DOTCONF([MINIMUM_VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT_FOUND]]])
 dnl ##
 dnl ##
-AC_DEFUN(AM_PATH_DOTCONF,
+AC_DEFUN([AM_PATH_DOTCONF],
 [dnl
 dnl Get the cflags and libs from the dotconf-config script
 dnl
