diff -ur a/shared/vmmouse_client.h b/shared/vmmouse_client.h
--- a/shared/vmmouse_client.h	2013-01-15 19:36:50.000000000 +0100
+++ b/shared/vmmouse_client.h	2015-03-21 03:01:24.312091183 +0100
@@ -39,6 +39,7 @@
 #ifndef _VMMOUSE_CLIENT_H_
 #define _VMMOUSE_CLIENT_H_
 
+#include "config.h"
 #include "xf86_OSproc.h"
 
 /*
diff -ur a/src/xf86OSmouse.h b/src/xf86OSmouse.h
--- a/src/xf86OSmouse.h	2013-01-15 19:36:50.000000000 +0100
+++ b/src/xf86OSmouse.h	2015-03-21 02:59:24.614878283 +0100
@@ -30,6 +30,7 @@
 #ifndef _XF86OSMOUSE_H_
 #define _XF86OSMOUSE_H_
 
+#include "config.h"
 #include "xf86Xinput.h"
 
 /* Mouse interface classes */
