--- qtlocation-everywhere-src-5.15.13/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro	2024-01-04 20:21:53.000000000 +0100
+++ qtlocation-everywhere-src-5.15.13.new/src/3rdparty/mapbox-gl-native/mapbox-gl-native.pro	2024-04-19 17:34:58.212788131 +0200
@@ -2,8 +2,7 @@
 
 load(qt_helper_lib)
 
-CONFIG += qt c++14 exceptions warn_off staticlib object_parallel_to_source
-CONFIG -= c++1z
+CONFIG += qt exceptions warn_off staticlib object_parallel_to_source
 
 QT += network-private \
       gui-private \
