diff -Naur systemsettings-5.11.1/app/systemsettings.kcfg systemsettings-5.11.1-p/app/systemsettings.kcfg
--- systemsettings-5.11.1/app/systemsettings.kcfg	2017-10-17 13:34:29.000000000 +0200
+++ systemsettings-5.11.1-p/app/systemsettings.kcfg	2017-10-20 12:35:27.322801065 +0200
@@ -11,7 +11,7 @@
     </entry>
     <entry name="ActiveView" type="String">
       <label>Internal name for the view used</label>
-      <default>systemsettings_sidebar_mode</default>
+      <default>icon_mode</default>
     </entry>
   </group>
 </kcfg>
