--- liboobs-2.22.0old/oobs/oobs-usersconfig.c	2008-01-02 15:06:41.000000000 +0100
+++ liboobs-2.22.0/oobs/oobs-usersconfig.c	2008-11-20 23:27:07.000000000 +0100
@@ -21,6 +21,7 @@
 #include <dbus/dbus.h>
 #include <glib-object.h>
 #include <string.h>
+#include <stdio.h>
 
 #include "oobs-object.h"
 #include "oobs-object-private.h"
@@ -359,7 +360,19 @@
 		"other-data", &other_data,
 		NULL);
 
-  g_return_val_if_fail ((login && password && homedir && shell), FALSE);
+if ( sizeof(login)<1 ) {
+	return FALSE ;
+}
+if ( sizeof(password)<1 ) {
+	return FALSE ;
+}
+if ( sizeof(homedir)<1 ) {
+	return FALSE ;
+}
+if ( sizeof(shell)<1 ) {
+	return FALSE ;
+}
+
 
   group = oobs_user_get_main_group (user);
   gid = oobs_group_get_gid (group);
