diff -aur liferea-1.2.23/src/common.c fw_liferea-1.2.23/src/common.c
--- liferea-1.2.23/src/common.c	2007-08-19 19:17:58.000000000 +0200
+++ fw_liferea-1.2.23/src/common.c	2007-11-01 11:10:14.000000000 +0100
@@ -908,6 +908,9 @@
 
 	g_free(cachePath);
 	/* lifereaUserPath reused globally */
+
+	/* ensure reasonable default umask */
+	umask (077);
 }
 
 const gchar * common_get_cache_path(void) {
diff -aur liferea-1.2.23/src/export.c fw_liferea-1.2.23/src/export.c
--- liferea-1.2.23/src/export.c	2007-06-21 00:22:10.000000000 +0200
+++ fw_liferea-1.2.23/src/export.c	2007-11-01 11:11:30.000000000 +0100
@@ -126,15 +126,15 @@
 			error = TRUE;
 		}
 		
-		if(trusted)
-			old_umask = umask(077);
+		if(!trusted)
+			old_umask = umask(022); /* give read permissions for other, per-default we wouldn't give it... */
 			
 		if(-1 == common_save_xml(doc, backupFilename)) {
 			g_warning("Could not export to OPML file!!");
 			error = TRUE;
 		}
 		
-		if(trusted)
+		if(!trusted)
 			umask(old_umask);
 			
 		xmlFreeDoc(doc);
