diff -Nur liferea-1.7.4.orig/opml/feedlist_bg.opml liferea-1.7.4/opml/feedlist_bg.opml
--- liferea-1.7.4.orig/opml/feedlist_bg.opml	2009-06-14 10:04:14.000000000 +0200
+++ liferea-1.7.4/opml/feedlist_bg.opml	2010-11-21 17:20:52.000000000 +0100
@@ -14,6 +14,9 @@
 	<!-- lets keep this default English block in sync over all feed lists! -->
 	<outline text="International">		<!-- translate this for localized feed list -->
 		<outline text="Open Source">		<!-- translate this for localized feed list -->
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 			<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
 			<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 			<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
diff -Nur liferea-1.7.4.orig/opml/feedlist_ca.opml liferea-1.7.4/opml/feedlist_ca.opml
--- liferea-1.7.4.orig/opml/feedlist_ca.opml	2009-06-14 10:04:14.000000000 +0200
+++ liferea-1.7.4/opml/feedlist_ca.opml	2010-11-21 17:21:09.000000000 +0100
@@ -19,6 +19,9 @@
 	<!-- lets keep this English block in sync over all feed lists! -->
 	<outline text="Fuentes de ejemplo (inglés)">
 		<outline text="Open Source">
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 			<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
 			<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 			<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
diff -Nur liferea-1.7.4.orig/opml/feedlist_de.opml liferea-1.7.4/opml/feedlist_de.opml
--- liferea-1.7.4.orig/opml/feedlist_de.opml	2009-06-14 10:04:14.000000000 +0200
+++ liferea-1.7.4/opml/feedlist_de.opml	2010-11-21 17:21:02.000000000 +0100
@@ -23,6 +23,9 @@
  		<!-- lets keep this default English block in sync over all feed lists! -->
 		<outline text="International">		<!-- translate this for localized feed list -->
 			<outline text="Open Source">		<!-- translate this for localized feed list -->
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 				<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
 				<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 				<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
diff -Nur liferea-1.7.4.orig/opml/feedlist_es.opml liferea-1.7.4/opml/feedlist_es.opml
--- liferea-1.7.4.orig/opml/feedlist_es.opml	2009-06-14 10:04:14.000000000 +0200
+++ liferea-1.7.4/opml/feedlist_es.opml	2010-11-21 17:21:24.000000000 +0100
@@ -15,6 +15,9 @@
 	<!-- lets keep this default English block in sync over all feed lists! -->
 	<outline text="Fuentes de ejemplo (inglés)">
 	      <outline text="Open Source">
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 			<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
 			<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 			<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
diff -Nur liferea-1.7.4.orig/opml/feedlist_eu.opml liferea-1.7.4/opml/feedlist_eu.opml
--- liferea-1.7.4.orig/opml/feedlist_eu.opml	2009-06-14 10:04:14.000000000 +0200
+++ liferea-1.7.4/opml/feedlist_eu.opml	2010-11-21 17:21:19.000000000 +0100
@@ -20,6 +20,9 @@
 	<!-- lets keep this default English block in sync over all feed lists! -->
 	<outline text="International">		<!-- translate this for localized feed list -->
 		<outline text="Open Source">		<!-- translate this for localized feed list -->
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 			<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
 			<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 			<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
diff -Nur liferea-1.7.4.orig/opml/feedlist_fr.opml liferea-1.7.4/opml/feedlist_fr.opml
--- liferea-1.7.4.orig/opml/feedlist_fr.opml	2009-06-14 10:04:14.000000000 +0200
+++ liferea-1.7.4/opml/feedlist_fr.opml	2010-11-21 17:26:35.000000000 +0100
@@ -10,6 +10,9 @@
        <outline text="Formats Ouverts" htmlUrl="http://formats-ouverts.org/blog/" xmlUrl="http://formats-ouverts.org/rss.php" />
        <outline text="Liberation.fr" htmlUrl="http://www.liberation.fr/" xmlUrl="http://www.liberation.fr/rss.php" />
        <outline text="Actualités ZDNet.fr" htmlUrl="http://www.zdnet.fr/actualites/?rss" xmlUrl="http://rss.zdnet.fr/feeds/rss/actualites/" />
+			<outline text="Frugalware FR" htmlUrl="http://frugalware.fr" xmlUrl="http://www.frugalware.fr/feed.php?rss"/>
+			<outline text="Planète Frugalware" htmlUrl="http://planet.frugalware.fr" xmlUrl="http://planet.frugalware.fr/feed.php?type=rss"/>
+			<outline text="Frugalware FR Forums" htmlUrl="http://forums.frugalware.fr" xmlUrl="http://forums.frugalware.fr/syndication.php?type=rss"/>
        <outline text="Geckozone - Articles" htmlUrl="http://www.geckozone.org/" xmlUrl="http://www.geckozone.org/articles/xml/" />
        <outline text="MozillaZine-Fr" htmlUrl="http://www.mozillazine-fr.org/" xmlUrl="http://mozillazine-fr.org/contenu.xml" />
        <outline text="LinuxFR" htmlUrl="http://linuxfr.org/" xmlUrl="http://linuxfr.org/backend/news/rss20.rss" />
@@ -22,6 +25,9 @@
 	<!-- lets keep this default English block in sync over all feed lists! -->
 	<outline text="International">		<!-- translate this for localized feed list -->
 		<outline text="Open Source">		<!-- translate this for localized feed list -->
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 			<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
 			<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 			<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
diff -Nur liferea-1.7.4.orig/opml/feedlist_it.opml liferea-1.7.4/opml/feedlist_it.opml
--- liferea-1.7.4.orig/opml/feedlist_it.opml	2009-06-14 10:04:14.000000000 +0200
+++ liferea-1.7.4/opml/feedlist_it.opml	2010-11-21 17:22:39.000000000 +0100
@@ -22,6 +22,9 @@
 	<!-- lets keep this default English block in sync over all feed lists! -->
 	<outline text="Notizie internazionali">
 		<outline text="Open Source">
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 			<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
 			<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 			<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
diff -Nur liferea-1.7.4.orig/opml/feedlist_nl.opml liferea-1.7.4/opml/feedlist_nl.opml
--- liferea-1.7.4.orig/opml/feedlist_nl.opml	2009-06-14 10:04:14.000000000 +0200
+++ liferea-1.7.4/opml/feedlist_nl.opml	2010-11-21 17:22:29.000000000 +0100
@@ -7,6 +7,9 @@
 	<!-- lets keep this default English block in sync over all feed lists! -->
 	<outline text="International">		<!-- translate this for localized feed list -->
 		<outline text="Open Source">		<!-- translate this for localized feed list -->
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 			<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
 			<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 			<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
diff -Nur liferea-1.7.4.orig/opml/feedlist.opml liferea-1.7.4/opml/feedlist.opml
--- liferea-1.7.4.orig/opml/feedlist.opml	2010-02-09 17:12:54.000000000 +0100
+++ liferea-1.7.4/opml/feedlist.opml	2010-11-21 17:20:44.000000000 +0100
@@ -13,6 +13,9 @@
 			<outline text="Science" htmlUrl="http://www.sciencenews.org" xmlUrl="http://www.sciencenews.org/view/feed/type/news/name/news.rss"/>
 		</outline>
 		<outline text="Open Source">		<!-- translate this title for localized feed list -->
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 			<outline text="Liferea Blog" htmlUrl="http://liferea.blogspot.com/" xmlUrl="http://liferea.blogspot.com/feeds/posts/default" />
 			<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 			<outline text="Planet GNOME" htmlUrl="http://planet.gnome.org/" xmlUrl="http://planet.gnome.org/atom.xml" />	
diff -Nur liferea-1.7.4.orig/opml/feedlist_pl.opml liferea-1.7.4/opml/feedlist_pl.opml
--- liferea-1.7.4.orig/opml/feedlist_pl.opml	2009-06-14 10:04:14.000000000 +0200
+++ liferea-1.7.4/opml/feedlist_pl.opml	2010-11-21 17:22:22.000000000 +0100
@@ -13,6 +13,9 @@
       <!-- lets keep this English block in sync over all feed lists! -->
       <outline text="międzynarodowe">
 	      <outline text="Open Source">
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 			<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
 			<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 			<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
diff -Nur liferea-1.7.4.orig/opml/feedlist_pt_BR.opml liferea-1.7.4/opml/feedlist_pt_BR.opml
--- liferea-1.7.4.orig/opml/feedlist_pt_BR.opml	2009-06-14 10:04:14.000000000 +0200
+++ liferea-1.7.4/opml/feedlist_pt_BR.opml	2010-11-21 17:22:10.000000000 +0100
@@ -12,6 +12,9 @@
         <!-- lets keep this English block in sync over all feed lists! -->
         <outline text="internacional">
             <outline text="Open Source">
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 			<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
 			<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 			<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
diff -Nur liferea-1.7.4.orig/opml/feedlist_pt.opml liferea-1.7.4/opml/feedlist_pt.opml
--- liferea-1.7.4.orig/opml/feedlist_pt.opml	2009-06-14 10:04:14.000000000 +0200
+++ liferea-1.7.4/opml/feedlist_pt.opml	2010-11-21 17:22:16.000000000 +0100
@@ -13,6 +13,9 @@
         <!-- lets keep this English block in sync over all feed lists! -->
         <outline text="internacional">
             <outline text="Open Source">
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 			<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
 			<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 			<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
diff -Nur liferea-1.7.4.orig/opml/feedlist_ru.opml liferea-1.7.4/opml/feedlist_ru.opml
--- liferea-1.7.4.orig/opml/feedlist_ru.opml	2009-06-14 10:04:14.000000000 +0200
+++ liferea-1.7.4/opml/feedlist_ru.opml	2010-11-21 17:22:05.000000000 +0100
@@ -17,6 +17,9 @@
     <outline text="Международные ленты новостей">
       <outline text="Свободное программное обеспечение">
 	      <outline text="Open source">
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 			<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
 			<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 			<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
diff -Nur liferea-1.7.4.orig/opml/feedlist_sk.opml liferea-1.7.4/opml/feedlist_sk.opml
--- liferea-1.7.4.orig/opml/feedlist_sk.opml	2009-06-14 10:04:14.000000000 +0200
+++ liferea-1.7.4/opml/feedlist_sk.opml	2010-11-21 17:22:54.000000000 +0100
@@ -9,6 +9,9 @@
  			<!-- lets keep this default English block in sync over all feed lists! -->
 			<outline text="International">		<!-- translate this for localized feed list -->
 				<outline text="Open Source">		<!-- translate this for localized feed list -->
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 					<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
 					<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 					<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
diff -Nur liferea-1.7.4.orig/opml/feedlist_sv.opml liferea-1.7.4/opml/feedlist_sv.opml
--- liferea-1.7.4.orig/opml/feedlist_sv.opml	2009-06-14 10:04:14.000000000 +0200
+++ liferea-1.7.4/opml/feedlist_sv.opml	2010-11-21 17:23:12.000000000 +0100
@@ -13,6 +13,9 @@
         <outline title="Nyheter fr&#xE5;n Expressen" htmlUrl="http://expressen.se/1.573280.1175529352?standAlone=true" xmlUrl="http://expressen.se/1.573280?standAlone=true" />
       </outline>
       <outline title="&#xD6;ppen k&#xE4;llkod" text="&#xD6;ppen k&#xE4;llkod" description="&#xD6;ppen k&#xE4;llkod" type="folder">
+			<outline text="Frugalware Linux News" htmlUrl="http://frugalware.org" xmlUrl="http://frugalware.org/rss/news"/>
+			<outline text="Frugalware Linux Planet" htmlUrl="http://planet.frugalware.org" xmlUrl="http://frugalware.org/rss/blogs"/>
+			<outline text="Frugalware Linux Forums" htmlUrl="http://forums.frugalware.org" xmlUrl="http://forums.frugalware.org/feed.php"/>
 			<outline text="Gnomefiles" htmlUrl="http://www.gnomefiles.org/" xmlUrl="http://www.gnomefiles.org/files/gnomefiles.rdf" />
 			<outline text="GNOME Footnotes" htmlUrl="http://www.gnomedesktop.org/" xmlUrl="http://www.gnomedesktop.org/backend.php" />
 			<outline text="GrokLaw" htmlUrl="http://www.groklaw.net/" xmlUrl="http://www.groklaw.net/backend/GrokLaw.rdf" />
