--- liferea/src/liferea.in	2007/07/23 18:15:15	3314
+++ liferea/src/liferea.in	2007/11/21 22:36:47	3569
@@ -14,8 +14,18 @@
 
 params="$@"
 
-LD_LIBRARY_PATH=@MOZILLA_LIB_ROOT@:$LD_LIBRARY_PATH
-export LD_LIBRARY_PATH
+#
+# If we run with Gecko or XulRunner we need to set 
+# LD_LIBRARY_PATH (WebKit and GtkHTML do not need this).
+#
+if [ "@MOZILLA_LIB_ROOT@" != "" ]; then
+	if [ "$LD_LIBRARY_PATH" = "" ]; then
+		LD_LIBRARY_PATH=@MOZILLA_LIB_ROOT@
+	else
+		LD_LIBRARY_PATH=@MOZILLA_LIB_ROOT@:$LD_LIBRARY_PATH
+	fi
+	export LD_LIBRARY_PATH
+fi
 
 if [ -z "$DBUS_SESSION_BUS_ADDRESS" ]; then
 	eval `dbus-launch`
