--- Makefile	2008-04-20 17:57:43.000000000 +0200
+++ Makefile	2008-04-21 21:06:03.000000000 +0200
@@ -10,14 +10,13 @@
 
 # END OF CUSTOM SETTINGS
 
-CXXFLAGS=\
-  -I$(GLPREFIX)/include \
+CXXFLAGS ?= -O2-g -Wall
+CXXFLAGS +=  -I$(GLPREFIX)/include \
   -I$(ODEPREFIX)/include \
   -I$(PLIBPREFIX)/include \
   -I../src-common \
   -I. \
   -DGAMEVERSION=$(VERSION) \
-  -O2 -g -Wall
 
 LFLAGS=\
   -L$(PLIBPREFIX)/$(LIBDIRNAME) \
