diff -Naur lxml-lxml-3.8.0/buildlibxml.py lxml-lxml-3.8.0-py27fix/buildlibxml.py
--- lxml-lxml-3.8.0/buildlibxml.py	2017-06-03 19:36:39.000000000 +0200
+++ lxml-lxml-3.8.0-py27fix/buildlibxml.py	2017-06-12 16:39:21.000000000 +0200
@@ -2,12 +2,13 @@
 import tarfile
 from distutils import log, version
 from contextlib import closing
+from ftplib import FTP
 
 try:
-    from urlparse import urljoin, unquote
+    from urlparse import urljoin, unquote, urlparse
     from urllib import urlretrieve, urlopen, urlcleanup
 except ImportError:
-    from urllib.parse import urljoin, unquote
+    from urllib.parse import urljoin, unquote, urlparse
     from urllib.request import urlretrieve, urlopen, urlcleanup
 
 multi_make_options = []
@@ -134,6 +135,27 @@
 
 def ftp_listdir(url):
     assert url.lower().startswith('ftp://')
+    try:
+        return _list_dir_urllib(url)
+    except IOError:
+        print("Requesting with urllib failed. Falling back to ftplib. Proxy argument will be ignored")
+        return _list_dir_ftplib(url)
+
+
+def _list_dir_ftplib(url):
+    parts = urlparse(url)
+    ftp = FTP(parts.netloc)
+    try:
+        ftp.login()
+        ftp.cwd(parts.path)
+        data = []
+        ftp.dir(data.append)
+    finally:
+        ftp.quit()
+    return parse_text_ftplist("\n".join(data))
+
+
+def _list_dir_urllib(url):
     with closing(urlopen(url)) as res:
         charset = _find_content_encoding(res)
         content_type = res.headers.get('Content-Type')
diff -Naur lxml-lxml-3.8.0/src/lxml/html/tests/hackers-org-data/xml-embedded-js.data lxml-lxml-3.8.0-py27fix/src/lxml/html/tests/hackers-org-data/xml-embedded-js.data
--- lxml-lxml-3.8.0/src/lxml/html/tests/hackers-org-data/xml-embedded-js.data	2017-06-03 19:36:39.000000000 +0200
+++ lxml-lxml-3.8.0-py27fix/src/lxml/html/tests/hackers-org-data/xml-embedded-js.data	2017-06-12 16:39:21.000000000 +0200
@@ -1,4 +1,4 @@
-Description: Locally hosted XML with embedded JavaScript#XSS_Local_XML that is generated using an XML data island. This is the same as above but instead referrs to a locally hosted (must be on the same server) XML file that contains your cross site scripting vector. You can see the result here <http://ha.ckers.org/xssxmltest.html>
+Description: Locally hosted XML with embedded JavaScript#XSS_Local_XML that is generated using an XML data island. This is the same as above but instead refers to a locally hosted (must be on the same server) XML file that contains your cross site scripting vector. You can see the result here <http://ha.ckers.org/xssxmltest.html>
     http://ha.ckers.org/xss.html#XSS_Local_XML
 
 <div><XML SRC="xsstest.xml" ID=I></XML>
