diff -Naur partclone-0.3.20/src/f2fs/libf2fs.c partclone-0.3.20.new/src/f2fs/libf2fs.c
--- partclone-0.3.20/src/f2fs/libf2fs.c	2022-04-27 08:56:42.000000000 +0200
+++ partclone-0.3.20.new/src/f2fs/libf2fs.c	2022-11-14 13:43:53.744682611 +0100
@@ -22,7 +22,6 @@
 #include <sys/mount.h>
 #include <sys/ioctl.h>
 #include <linux/hdreg.h>
-#include <linux/fs.h>
 
 #include "f2fs_fs.h"
 
diff -Naur partclone-0.3.20/src/partclone.c partclone-0.3.20.new/src/partclone.c
--- partclone-0.3.20/src/partclone.c	2022-04-27 08:56:42.000000000 +0200
+++ partclone-0.3.20.new/src/partclone.c	2022-11-14 13:44:01.667233133 +0100
@@ -37,7 +37,6 @@
 #include <inttypes.h>
 #include <assert.h>
 #include "gettext.h"
-#include <linux/fs.h>
 #include <sys/types.h>
 #include <dirent.h>
 #define _(STRING) gettext(STRING)
diff -Naur partclone-0.3.20/src/xfs/include/linux.h partclone-0.3.20.new/src/xfs/include/linux.h
--- partclone-0.3.20/src/xfs/include/linux.h	2022-04-27 08:56:42.000000000 +0200
+++ partclone-0.3.20.new/src/xfs/include/linux.h	2022-11-14 13:44:11.479676435 +0100
@@ -23,7 +23,6 @@
 #ifdef OVERRIDE_SYSTEM_FSXATTR
 # define fsxattr sys_fsxattr
 #endif
-#include <linux/fs.h> /* fsxattr defintion for new kernels */
 #ifdef OVERRIDE_SYSTEM_FSXATTR
 # undef fsxattr
 #endif
diff -Naur partclone-0.3.20/src/xfs/include/xfs/linux.h partclone-0.3.20.new/src/xfs/include/xfs/linux.h
--- partclone-0.3.20/src/xfs/include/xfs/linux.h	2022-04-27 08:56:42.000000000 +0200
+++ partclone-0.3.20.new/src/xfs/include/xfs/linux.h	2022-11-14 13:44:20.726151848 +0100
@@ -23,7 +23,6 @@
 #ifdef OVERRIDE_SYSTEM_FSXATTR
 # define fsxattr sys_fsxattr
 #endif
-#include <linux/fs.h> /* fsxattr defintion for new kernels */
 #ifdef OVERRIDE_SYSTEM_FSXATTR
 # undef fsxattr
 #endif
