diff -Naur fixed-0.7.5/src/Makefile fixed-0.7.5-p/src/Makefile
--- fixed-0.7.5/src/Makefile	2008-02-16 11:14:10.000000000 +0100
+++ fixed-0.7.5-p/src/Makefile	2008-04-27 15:08:59.000000000 +0200
@@ -3,31 +3,18 @@
 FIXEDVERTARGET = fixedversion.h
 FIXEDTARGET = fixed.oct 
 
-LIBPRE = lib
-LIBEXT = dll
-LIBIMPEXT = dll.a
 DLLDEF =
 ifneq (,$(findstring cygwin,$(canonical_host_type)))
-   OCT_INCLUDES_LIB = 1
    DLLDEF = -DFIXED_DLL
-   FIXEDIMPTARGET = $(LIBPRE)octave_fixed.$(LIBIMPEXT)
-   ADD_FLAGS = -Wl,--out-implib=$(FIXEDIMPTARGET)
 endif
 ifneq (,$(findstring mingw,$(canonical_host_type)))
-   OCT_INCLUDES_LIB = 1
    DLLDEF = -DFIXED_DLL
-   FIXEDIMPTARGET = $(LIBPRE)octave_fixed.$(LIBIMPEXT)
-   ADD_FLAGS = -Wl,--out-implib=$(FIXEDIMPTARGET)
 endif
 ifneq (,$(findstring msdosmsvc,$(canonical_host_type)))
-   OCT_INCLUDES_LIB = 1
-   LIBPRE =
-   LIBIMPEXT = lib
    DLLDEF = -DFIXED_DLL
 endif
 
-ifeq (1,$(OCT_INCLUDES_LIB))
-  FIXEDSOURCES = fixedColVector.cc fixedRowVector.cc \
+FIXEDSOURCES = fixedColVector.cc fixedRowVector.cc \
 	fixedMatrix.cc fixedComplex.cc fixedCColVector.cc fixedCRowVector.cc \
 	fixedCMatrix.cc Array-f.cc fixed-conv.cc ov-fixed.cc ov-fixed-mat.cc \
 	fixedNDArray.cc fixedCNDArray.cc \
@@ -37,44 +24,11 @@
 	op-fcs-fs.cc  op-fcs-fm.cc  op-fcm-fs.cc  op-fcm-fm.cc \
 	op-fcs-fcs.cc op-fcs-fcm.cc op-fcm-fcs.cc op-fcm-fcm.cc \
 	fixed-var.cc fixed.cc
-  FIXEDOBJECTS = fixed-int.o $(patsubst %.cc,%.o,$(FIXEDSOURCES))
-  FIXEDDEPENDS = fixed-int.d $(patsubst %.cc,%.d,$(FIXEDSOURCES))
-
-  TARGETS = $(FIXEDTARGET)
-  OBJECTS = $(FIXEDOBJECTS)
+FIXEDOBJECTS = fixed-int.o $(patsubst %.cc,%.o,$(FIXEDSOURCES))
+FIXEDDEPENDS = fixed-int.d $(patsubst %.cc,%.d,$(FIXEDSOURCES))
 
-  EXTRALIBS =
-else
-  FIXEDLIBTARGET = $(LIBPRE)octave_fixed.$(LIBEXT)
-  FIXEDIMPTARGET = $(FIXEDLIBTARGET:.$(LIBEXT)=.$(LIBIMPEXT))
-  FIXEDLIBSOURCES = fixedColVector.cc fixedRowVector.cc \
-	fixedMatrix.cc fixedComplex.cc fixedCColVector.cc fixedCRowVector.cc \
-	fixedCMatrix.cc Array-f.cc fixed-conv.cc ov-fixed.cc ov-fixed-mat.cc \
-	fixedNDArray.cc fixedCNDArray.cc \
-	ov-fixed-complex.cc ov-fixed-cx-mat.cc \
-	op-fs-fs.cc   op-fs-fm.cc   op-fm-fs.cc   op-fm-fm.cc \
-	op-fs-fcs.cc  op-fs-fcm.cc  op-fm-fcs.cc  op-fm-fcm.cc \
-	op-fcs-fs.cc  op-fcs-fm.cc  op-fcm-fs.cc  op-fcm-fm.cc \
-	op-fcs-fcs.cc op-fcs-fcm.cc op-fcm-fcs.cc op-fcm-fcm.cc
-  FIXEDLIBOBJECTS = fixed-int.o $(patsubst %.cc,%.o,$(FIXEDLIBSOURCES))
-  FIXEDLIBDEPENDS = fixed-int.d $(patsubst %.cc,%.d,$(FIXEDLIBSOURCES))
-
-  FIXEDSOURCES = fixed.cc fixed-var.cc
-  FIXEDOBJECTS = $(patsubst %.cc,%.o,$(FIXEDSOURCES))
-  FIXEDDEPENDS = $(FIXEDLIBDEPENDS) $(patsubst %.cc,%.d,$(FIXEDSOURCES))
-
-  TARGETS = $(FIXEDLIBTARGET) $(FIXEDTARGET)
-  OBJECTS = $(FIXEDLIBOBJECTS) $(FIXEDOBJECTS)
-
-  EXTRALIBS = -L./ -loctave_fixed
-
-  # The name of this flag changed in version 2.1.54. Use both versions
-  # so that one of them will work.
-  MY_SH_LDFLAGS = $(shell $(MKOCTFILE) -p SH_LDFLAGS) \
-	-Wl,--out-implib=$(FIXEDIMPTARGET)
-  MY_DL_LDFLAGS = $(shell $(MKOCTFILE) -p DL_LDFLAGS) \
-	-Wl,--out-implib=$(FIXEDIMPTARGET)
-endif
+TARGETS = $(FIXEDTARGET)
+OBJECTS = $(FIXEDOBJECTS)
 
 ifeq ($(MAKECMDGOALS),all)
   DEPENDS = $(FIXEDDEPENDS)
