diff -Naurp libtorrent-0.11/configure libtorrent-0.11-p/configure
--- libtorrent-0.11/configure	2006-11-08 01:38:58.000000000 +0100
+++ libtorrent-0.11-p/configure	2006-11-10 13:36:02.000000000 +0100
@@ -21693,7 +21693,7 @@ else
 fi;
 
 
-                                                  ac_config_files="$ac_config_files Makefile src/Makefile include/Makefile test/Makefile libtorrent.pc"
+                                                  ac_config_files="$ac_config_files Makefile src/Makefile include/Makefile test/Makefile rblibtorrent.pc"
 
 if [ "x$examples" == "xyes" ]; then
 	          ac_config_files="$ac_config_files examples/Makefile"
@@ -22260,7 +22260,7 @@ do
   "src/Makefile" ) CONFIG_FILES="$CONFIG_FILES src/Makefile" ;;
   "include/Makefile" ) CONFIG_FILES="$CONFIG_FILES include/Makefile" ;;
   "test/Makefile" ) CONFIG_FILES="$CONFIG_FILES test/Makefile" ;;
-  "libtorrent.pc" ) CONFIG_FILES="$CONFIG_FILES libtorrent.pc" ;;
+  "rblibtorrent.pc" ) CONFIG_FILES="$CONFIG_FILES rblibtorrent.pc" ;;
   "examples/Makefile" ) CONFIG_FILES="$CONFIG_FILES examples/Makefile" ;;
   "depfiles" ) CONFIG_COMMANDS="$CONFIG_COMMANDS depfiles" ;;
   "config.h" ) CONFIG_HEADERS="$CONFIG_HEADERS config.h" ;;
diff -Naurp libtorrent-0.11/libtorrent.pc libtorrent-0.11-p/libtorrent.pc
--- libtorrent-0.11/libtorrent.pc	2006-11-08 01:39:38.000000000 +0100
+++ libtorrent-0.11-p/libtorrent.pc	1970-01-01 01:00:00.000000000 +0100
@@ -1,14 +0,0 @@
-prefix=/usr/local
-exec_prefix=${prefix}
-bindir=${exec_prefix}/bin
-libdir=${exec_prefix}/lib
-datadir=${prefix}/share
-sysconfdir=${prefix}/etc
-includedir=${prefix}/include
-package=libtorrent
-
-Name: libtorrent
-Description: Bittorrent library.
-Version: 0.11
-Libs: -L${libdir}  -lboost_date_time -lboost_filesystem -lboost_thread $(top_builddir)/zlib/libzlib.la -ltorrent
-Cflags: -I${includedir} -I${includedir}/libtorrent 
diff -Naurp libtorrent-0.11/libtorrent.pc.in libtorrent-0.11-p/libtorrent.pc.in
--- libtorrent-0.11/libtorrent.pc.in	2006-06-25 02:12:17.000000000 +0200
+++ libtorrent-0.11-p/libtorrent.pc.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,14 +0,0 @@
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-bindir=@bindir@
-libdir=@libdir@
-datadir=@datadir@
-sysconfdir=@sysconfdir@
-includedir=@includedir@
-package=@PACKAGE@
-
-Name: libtorrent
-Description: Bittorrent library.
-Version: @VERSION@
-Libs: -L${libdir} @LIBS@ -l@BOOST_DATE_TIME_LIB@ -l@BOOST_FILESYSTEM_LIB@ -l@BOOST_THREAD_LIB@ @ZLIB@ -ltorrent
-Cflags: -I${includedir} -I${includedir}/libtorrent @CPPFLAGS@
diff -Naurp libtorrent-0.11/Makefile.am libtorrent-0.11-p/Makefile.am
--- libtorrent-0.11/Makefile.am	2006-10-13 01:51:09.000000000 +0200
+++ libtorrent-0.11-p/Makefile.am	2006-11-10 13:37:06.000000000 +0100
@@ -18,7 +18,7 @@ docs/ubuntu_build_notes.rst \
 docs/udp_tracker_protocol.html docs/client_test.rst docs/client_test.html \
 docs/unicode_support.png docs/client_test.png docs/style.css Jamfile project-root.jam \
 m4/ac_cxx_namespaces.m4 m4/acx_pthread.m4 m4/ax_boost_date-time.m4 \
-m4/ax_boost_filesystem.m4 m4/ax_boost_thread.m4 src/file_win.cpp libtorrent.pc
+m4/ax_boost_filesystem.m4 m4/ax_boost_thread.m4 src/file_win.cpp rblibtorrent.pc
 
 pkginclude_HEADER = \
 debian/changelog \
@@ -36,7 +36,7 @@ debian/rules
 
 
 pkgconfigdir   = $(libdir)/pkgconfig
-pkgconfig_DATA = libtorrent.pc
+pkgconfig_DATA = rblibtorrent.pc
 
 check: test
 	test/test_hasher \
diff -Naurp libtorrent-0.11/Makefile.in libtorrent-0.11-p/Makefile.in
--- libtorrent-0.11/Makefile.in	2006-11-08 01:38:57.000000000 +0100
+++ libtorrent-0.11-p/Makefile.in	2006-11-10 13:37:06.000000000 +0100
@@ -38,7 +38,7 @@ build_triplet = @build@
 host_triplet = @host@
 DIST_COMMON = README $(am__configure_deps) $(srcdir)/Makefile.am \
 	$(srcdir)/Makefile.in $(srcdir)/config.h.in \
-	$(srcdir)/libtorrent.pc.in $(top_srcdir)/configure AUTHORS \
+	$(srcdir)/rblibtorrent.pc.in $(top_srcdir)/configure AUTHORS \
 	COPYING ChangeLog INSTALL NEWS config.guess config.sub depcomp \
 	install-sh ltmain.sh missing
 subdir = .
@@ -56,7 +56,7 @@ am__CONFIG_DISTCLEAN_FILES = config.stat
  configure.lineno configure.status.lineno
 mkinstalldirs = $(install_sh) -d
 CONFIG_HEADER = config.h
-CONFIG_CLEAN_FILES = libtorrent.pc
+CONFIG_CLEAN_FILES = rblibtorrent.pc
 SOURCES =
 DIST_SOURCES =
 RECURSIVE_TARGETS = all-recursive check-recursive dvi-recursive \
@@ -219,7 +219,7 @@ docs/ubuntu_build_notes.rst \
 docs/udp_tracker_protocol.html docs/client_test.rst docs/client_test.html \
 docs/unicode_support.png docs/client_test.png docs/style.css Jamfile project-root.jam \
 m4/ac_cxx_namespaces.m4 m4/acx_pthread.m4 m4/ax_boost_date-time.m4 \
-m4/ax_boost_filesystem.m4 m4/ax_boost_thread.m4 src/file_win.cpp libtorrent.pc
+m4/ax_boost_filesystem.m4 m4/ax_boost_thread.m4 src/file_win.cpp rblibtorrent.pc
 
 pkginclude_HEADER = \
 debian/changelog \
@@ -236,7 +236,7 @@ debian/libtorrent0.install \
 debian/rules
 
 pkgconfigdir = $(libdir)/pkgconfig
-pkgconfig_DATA = libtorrent.pc
+pkgconfig_DATA = rblibtorrent.pc
 all: config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
 
@@ -291,7 +291,7 @@ $(srcdir)/config.h.in:  $(am__configure_
 
 distclean-hdr:
 	-rm -f config.h stamp-h1
-libtorrent.pc: $(top_builddir)/config.status $(srcdir)/libtorrent.pc.in
+rblibtorrent.pc: $(top_builddir)/config.status $(srcdir)/rblibtorrent.pc.in
 	cd $(top_builddir) && $(SHELL) ./config.status $@
 
 mostlyclean-libtool:
diff -Naurp libtorrent-0.11/rblibtorrent.pc libtorrent-0.11-p/rblibtorrent.pc
--- libtorrent-0.11/rblibtorrent.pc	1970-01-01 01:00:00.000000000 +0100
+++ libtorrent-0.11-p/rblibtorrent.pc	2006-11-08 01:39:38.000000000 +0100
@@ -0,0 +1,14 @@
+prefix=/usr/local
+exec_prefix=${prefix}
+bindir=${exec_prefix}/bin
+libdir=${exec_prefix}/lib
+datadir=${prefix}/share
+sysconfdir=${prefix}/etc
+includedir=${prefix}/include
+package=libtorrent
+
+Name: libtorrent
+Description: Bittorrent library.
+Version: 0.11
+Libs: -L${libdir}  -lboost_date_time -lboost_filesystem -lboost_thread $(top_builddir)/zlib/libzlib.la -ltorrent
+Cflags: -I${includedir} -I${includedir}/libtorrent 
diff -Naurp libtorrent-0.11/rblibtorrent.pc.in libtorrent-0.11-p/rblibtorrent.pc.in
--- libtorrent-0.11/rblibtorrent.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ libtorrent-0.11-p/rblibtorrent.pc.in	2006-06-25 02:12:17.000000000 +0200
@@ -0,0 +1,14 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+bindir=@bindir@
+libdir=@libdir@
+datadir=@datadir@
+sysconfdir=@sysconfdir@
+includedir=@includedir@
+package=@PACKAGE@
+
+Name: libtorrent
+Description: Bittorrent library.
+Version: @VERSION@
+Libs: -L${libdir} @LIBS@ -l@BOOST_DATE_TIME_LIB@ -l@BOOST_FILESYSTEM_LIB@ -l@BOOST_THREAD_LIB@ @ZLIB@ -ltorrent
+Cflags: -I${includedir} -I${includedir}/libtorrent @CPPFLAGS@
