diff -Naurp libtorrent-0.12/configure.in libtorrent-0.12-p/configure.in
--- libtorrent-0.12/configure.in	2007-03-10 21:38:36.000000000 +0100
+++ libtorrent-0.12-p/configure.in	2007-04-05 12:26:31.000000000 +0200
@@ -246,7 +246,7 @@ AC_ARG_ENABLE(
 )
 AC_SUBST(DEBUGFLAGS)
 
-AC_CONFIG_FILES(Makefile src/Makefile include/Makefile test/Makefile libtorrent.pc)
+AC_CONFIG_FILES(Makefile src/Makefile include/Makefile test/Makefile rblibtorrent.pc)
 if [[ "x$examples" == "xyes" ]]; then
 	AC_CONFIG_FILES(examples/Makefile)
 fi
diff -Naurp libtorrent-0.12/libtorrent.pc libtorrent-0.12-p/libtorrent.pc
--- libtorrent-0.12/libtorrent.pc	2007-03-18 03:18:51.000000000 +0100
+++ libtorrent-0.12-p/libtorrent.pc	1970-01-01 01:00:00.000000000 +0100
@@ -1,15 +0,0 @@
-prefix=/usr/local
-exec_prefix=${prefix}
-bindir=${exec_prefix}/bin
-libdir=${exec_prefix}/lib
-datarootdir=@datarootdir@
-datadir=${prefix}/share
-sysconfdir=${prefix}/etc
-includedir=${prefix}/include
-package=libtorrent
-
-Name: libtorrent
-Description: Bittorrent library.
-Version: 0.12
-Libs: -L${libdir}  -lboost_date_time -lboost_filesystem -lboost_thread $(top_builddir)/zlib/libzlib.la -ltorrent
-Cflags: -I${includedir} -I${includedir}/libtorrent 
diff -Naurp libtorrent-0.12/libtorrent.pc.in libtorrent-0.12-p/libtorrent.pc.in
--- libtorrent-0.12/libtorrent.pc.in	2007-03-10 21:38:38.000000000 +0100
+++ libtorrent-0.12-p/libtorrent.pc.in	1970-01-01 01:00:00.000000000 +0100
@@ -1,15 +0,0 @@
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-bindir=@bindir@
-libdir=@libdir@
-datarootdir=@datarootdir@
-datadir=@datadir@
-sysconfdir=@sysconfdir@
-includedir=@includedir@
-package=@PACKAGE@
-
-Name: libtorrent
-Description: Bittorrent library.
-Version: @VERSION@
-Libs: -L${libdir} @LIBS@ -l@BOOST_DATE_TIME_LIB@ -l@BOOST_FILESYSTEM_LIB@ -l@BOOST_THREAD_LIB@ @ZLIB@ -ltorrent
-Cflags: -I${includedir} -I${includedir}/libtorrent @CPPFLAGS@
diff -Naurp libtorrent-0.12/Makefile.am libtorrent-0.12-p/Makefile.am
--- libtorrent-0.12/Makefile.am	2007-03-12 00:56:18.000000000 +0100
+++ libtorrent-0.12-p/Makefile.am	2007-04-05 12:28:06.000000000 +0200
@@ -19,7 +19,7 @@ docs/ubuntu_build_notes.rst \
 docs/udp_tracker_protocol.html docs/client_test.rst docs/client_test.html \
 docs/unicode_support.png docs/client_test.png docs/style.css Jamfile project-root.jam \
 m4/ac_cxx_namespaces.m4 m4/acx_pthread.m4 m4/ax_boost_date-time.m4 \
-m4/ax_boost_filesystem.m4 m4/ax_boost_thread.m4 src/file_win.cpp libtorrent.pc \
+m4/ax_boost_filesystem.m4 m4/ax_boost_thread.m4 src/file_win.cpp rblibtorrent.pc \
 bindings/README.txt \
 bindings/python/Jamfile \
 bindings/python/client.py \
@@ -65,7 +65,7 @@ debian/rules
 
 
 pkgconfigdir   = $(libdir)/pkgconfig
-pkgconfig_DATA = libtorrent.pc
+pkgconfig_DATA = rblibtorrent.pc
 
 check: test
 	test/test_hasher \
diff -Naurp libtorrent-0.12/rblibtorrent.pc libtorrent-0.12-p/rblibtorrent.pc
--- libtorrent-0.12/rblibtorrent.pc	1970-01-01 01:00:00.000000000 +0100
+++ libtorrent-0.12-p/rblibtorrent.pc	2007-03-18 03:18:51.000000000 +0100
@@ -0,0 +1,15 @@
+prefix=/usr/local
+exec_prefix=${prefix}
+bindir=${exec_prefix}/bin
+libdir=${exec_prefix}/lib
+datarootdir=@datarootdir@
+datadir=${prefix}/share
+sysconfdir=${prefix}/etc
+includedir=${prefix}/include
+package=libtorrent
+
+Name: libtorrent
+Description: Bittorrent library.
+Version: 0.12
+Libs: -L${libdir}  -lboost_date_time -lboost_filesystem -lboost_thread $(top_builddir)/zlib/libzlib.la -ltorrent
+Cflags: -I${includedir} -I${includedir}/libtorrent 
diff -Naurp libtorrent-0.12/rblibtorrent.pc.in libtorrent-0.12-p/rblibtorrent.pc.in
--- libtorrent-0.12/rblibtorrent.pc.in	1970-01-01 01:00:00.000000000 +0100
+++ libtorrent-0.12-p/rblibtorrent.pc.in	2007-03-10 21:38:38.000000000 +0100
@@ -0,0 +1,15 @@
+prefix=@prefix@
+exec_prefix=@exec_prefix@
+bindir=@bindir@
+libdir=@libdir@
+datarootdir=@datarootdir@
+datadir=@datadir@
+sysconfdir=@sysconfdir@
+includedir=@includedir@
+package=@PACKAGE@
+
+Name: libtorrent
+Description: Bittorrent library.
+Version: @VERSION@
+Libs: -L${libdir} @LIBS@ -l@BOOST_DATE_TIME_LIB@ -l@BOOST_FILESYSTEM_LIB@ -l@BOOST_THREAD_LIB@ @ZLIB@ -ltorrent
+Cflags: -I${includedir} -I${includedir}/libtorrent @CPPFLAGS@
