diff -Naurp libtorrent-0.12/configure.in libtorrent-0.12-p/configure.in
--- libtorrent-0.12/configure.in        2007-03-10 21:38:36.000000000 +0100
+++ libtorrent-0.12-p/configure.in      2007-04-05 12:26:31.000000000 +0200
@@ -246,7 +246,7 @@ AC_ARG_ENABLE(
 )
 AC_SUBST(DEBUGFLAGS)

-AC_CONFIG_FILES(Makefile src/Makefile include/Makefile test/Makefile libtorrent.pc)
+AC_CONFIG_FILES(Makefile src/Makefile include/Makefile test/Makefile rblibtorrent.pc)
 if [[ "x$examples" == "xyes" ]]; then
        AC_CONFIG_FILES(examples/Makefile)
 fi
--- libtorrent-0.13/Makefile.am	2008-03-25 08:32:00.000000000 +0100
+++ libtorrent-0.13-p/Makefile.am	2008-05-07 02:15:30.000000000 +0200
@@ -1,5 +1,5 @@
 ACLOCAL_AMFLAGS = -I m4
-SUBDIRS = include @ZLIBDIR@ src @EXAMPLESDIR@ test
+SUBDIRS = include @ZLIBDIR@ src @EXAMPLESDIR@
 EXTRA_DIST = docs/manual.html docs/manual.rst docs/extension_protocol.rst \
 docs/extension_protocol.html \
 docs/projects.rst docs/projects.html \
@@ -33,7 +33,7 @@
 docs/ziptorrent_thumb.gif \
 docs/unicode_support.png docs/client_test.png docs/style.css Jamfile project-root.jam \
 m4/ac_cxx_namespaces.m4 m4/acx_pthread.m4 m4/ax_boost_date-time.m4 \
-m4/ax_boost_filesystem.m4 m4/ax_boost_thread.m4 src/file_win.cpp libtorrent.pc \
+m4/ax_boost_filesystem.m4 m4/ax_boost_thread.m4 src/file_win.cpp rblibtorrent.pc \
 bindings/README.txt \
 bindings/python/Jamfile \
 bindings/python/client.py \
@@ -80,7 +80,7 @@
 
 
 pkgconfigdir   = $(libdir)/pkgconfig
-pkgconfig_DATA = libtorrent.pc
+pkgconfig_DATA = rblibtorrent.pc
 
 check: test
 	test/test_hasher \
