--- ./include/libtorrent/buffer.hpp.orig	2008-03-17 17:43:36.000000000 +0100
+++ ./include/libtorrent/buffer.hpp	2008-03-17 17:42:20.000000000 +0100
@@ -36,6 +36,7 @@
 
 #include "libtorrent/invariant_check.hpp"
 #include <memory>
+#include <cstring>
 
 namespace libtorrent {
 
--- ./include/libtorrent/peer_id.hpp.orig	2008-03-17 17:10:28.000000000 +0100
+++ ./include/libtorrent/peer_id.hpp	2008-03-17 17:10:50.000000000 +0100
@@ -39,6 +39,7 @@
 #include <cctype>
 #include <algorithm>
 #include <string>
+#include <cstring>
 
 #include "libtorrent/config.hpp"
 
--- ./src/file.cpp.orig	2008-03-17 17:27:58.000000000 +0100
+++ ./src/file.cpp	2008-03-17 17:28:53.000000000 +0100
@@ -66,6 +66,7 @@
 #include <boost/filesystem/operations.hpp>
 #include "libtorrent/file.hpp"
 #include <sstream>
+#include <cstring>
 
 #ifndef O_BINARY
 #define O_BINARY 0
