diff -ur a/plugins/actions/findandreplace/findandreplace.cc b/plugins/actions/findandreplace/findandreplace.cc
--- a/plugins/actions/findandreplace/findandreplace.cc	2012-05-09 04:33:35.805017582 -0500
+++ b/plugins/actions/findandreplace/findandreplace.cc	2012-05-09 04:34:38.278015234 -0500
@@ -26,7 +26,7 @@
 #include <gui/dialogutility.h>
 #include <gtkmm_utility.h>
 #include <widget_config_utility.h>
-#include <glib/gregex.h>
+#include <glib.h>
 
 /*
  * FIXME:
