diff -Naur xf86-video-amdgpu-19.0.1/src/amdgpu_present.c xf86-video-amdgpu-19.0.1-revert/src/amdgpu_present.c
--- xf86-video-amdgpu-19.0.1/src/amdgpu_present.c	2019-03-19 18:49:55.000000000 +0100
+++ xf86-video-amdgpu-19.0.1-revert/src/amdgpu_present.c	2019-09-19 20:40:34.901928266 +0200
@@ -278,18 +278,12 @@
 		return FALSE;
 #endif
 
-	/* Only DC supports advanced color management features, so we can use
-	 * drmmode_cm_enabled as a proxy for "Is DC enabled?"
+	/* The kernel driver doesn't handle flipping between BOs with different
+	 * tiling parameters correctly yet
 	 */
-	if (info->dri2.pKernelDRMVersion->version_minor < 31 ||
-	    !drmmode_cm_enabled(&info->drmmode)) {
-		/* The kernel driver doesn't handle flipping between BOs with
-		 * different tiling parameters correctly
-		 */
-		if (amdgpu_pixmap_get_tiling_info(pixmap) !=
-		    amdgpu_pixmap_get_tiling_info(screen_pixmap))
-			return FALSE;
-	}
+	if (amdgpu_pixmap_get_tiling_info(pixmap) !=
+	    amdgpu_pixmap_get_tiling_info(screen_pixmap))
+		return FALSE;
 
 	for (i = 0, num_crtcs_on = 0; i < config->num_crtc; i++) {
 		if (drmmode_crtc_can_flip(config->crtc[i]))
