diff --git a/portable/libtorrent/src/disk_io_thread.cpp b/portable/libtorrent/src/disk_io_thread.cpp
index ec5b8d8..c127edf 100644
--- a/portable/libtorrent/src/disk_io_thread.cpp
+++ b/portable/libtorrent/src/disk_io_thread.cpp
@@ -31,6 +31,8 @@ POSSIBILITY OF SUCH DAMAGE.
 */
 
 #include "libtorrent/storage.hpp"
+#include <boost/thread/mutex.hpp>
+#include <boost/thread/condition.hpp>
 #include <deque>
 #include "libtorrent/disk_io_thread.hpp"
 
