diff -up Miro-1.1.2/portable/libtorrent/include/libtorrent/peer_id.hpp.fix Miro-1.1.2/portable/libtorrent/include/libtorrent/peer_id.hpp
--- Miro-1.1.2/portable/libtorrent/include/libtorrent/peer_id.hpp.gcc43	2008-02-15 03:45:47.000000000 -0700
+++ Miro-1.1.2/portable/libtorrent/include/libtorrent/peer_id.hpp	2008-02-15 03:46:00.000000000 -0700
@@ -37,7 +37,7 @@ POSSIBILITY OF SUCH DAMAGE.
 #include <iomanip>
 #include <cctype>
 #include <algorithm>
-#include <string>
+#include <cstring>
 
 #include "libtorrent/config.hpp"
 #include "libtorrent/assert.hpp"
diff -up Miro-1.1.2/portable/libtorrent/src/file.cpp.gcc43 Miro-1.1.2/portable/libtorrent/src/file.cpp
--- Miro-1.1.2/portable/libtorrent/src/file.cpp.gcc43	2008-02-15 04:12:42.000000000 -0700
+++ Miro-1.1.2/portable/libtorrent/src/file.cpp	2008-02-15 04:15:19.000000000 -0700
@@ -57,6 +57,7 @@ typedef int mode_t;
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <errno.h>
+#include <cstring>
 
 #include <boost/static_assert.hpp>
 // make sure the _FILE_OFFSET_BITS define worked

diff -up Miro-1.1.2/portable/libtorrent/src/entry.cpp.gcc43 Miro-1.1.2/portable/libtorrent/src/entry.cpp
--- Miro-1.1.2/portable/libtorrent/src/entry.cpp.gcc43	     2008-02-07 21:30:04.000000000 -0500
+++ Miro-1.1.2/portable/libtorrent/src/entry.cpp	     2008-03-02 20:32:44.000000000 -0500
@@ -32,6 +32,7 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #include "libtorrent/pch.hpp"
 
+#include <iostream>
 #include <algorithm>
 #include <iomanip>
 #include "libtorrent/entry.hpp"
@@ -287,11 +288,11 @@ namespace libtorrent
 	void entry::print(std::ostream& os, int indent) const
 	{
 		TORRENT_ASSERT(indent >= 0);
-		for (int i = 0; i < indent; ++i) os << " ";
+		for (int i = 0; i < indent; ++i) os << entry(" ");
 		switch (m_type)
 		{
 		case int_t:
-			os << integer() << "\n";
+			os << integer() << entry("\n");
 			break;
 		case string_t:
 			{
