diff -Naur build.orig/Makefile.in build/Makefile.in
--- build.orig/Makefile.in	1996-10-23 15:43:38.000000000 +0200
+++ build/Makefile.in	2007-01-26 20:31:23.000000000 +0100
@@ -78,8 +78,11 @@
 	@echo "Installing vi, ex, view: $(bindir) ..."
 	[ -d $(bindir) ] || \
 	    ($(mkdir) $(bindir) && $(chmod) $(dmode) $(bindir))
+	[ -d $(bindir)/../../bin ] || \
+	    ($(mkdir) $(bindir)/../../bin && $(chmod) $(dmode) $(bindir)/../../bin)
 	cd $(bindir) && $(rm) -f `echo vi | sed '$(transform)'`
 	$(cp) nvi $(bindir)/`echo vi | sed '$(transform)'`
+	ln -s ../usr/bin/nvi $(bindir)/../../bin/vi
 	cd $(bindir) && [ -f $(strip) ] && \
 	    $(strip) `echo vi | sed '$(transform)'`
 	cd $(bindir) && $(chmod) $(emode) `echo vi | sed '$(transform)'`
