diff -Naur build.orig/Makefile.in build/Makefile.in
--- build.orig/Makefile.in	1996-10-23 15:43:38.000000000 +0200
+++ build/Makefile.in	2007-01-25 23:26:12.000000000 +0100
@@ -139,8 +139,8 @@
 	    cd $(datadir)/vi/catalog && $(chmod) $(fmode) *)
 	@echo "Installing Perl scripts: $(datadir)/vi/perl ..."
 	$(mkdir) $(datadir)/vi/perl && $(chmod) $(dmode) $(datadir)/vi/perl
-	[ -f VI.pm ] && $(cp) VI.pm $(datadir)/vi/perl && \
-	    cd $(datadir)/vi/perl && $(chmod) $(fmode) VI.pm)
+	if [ -f VI.pm ]; then $(cp) VI.pm $(datadir)/vi/perl && \
+	    cd $(datadir)/vi/perl && $(chmod) $(fmode) VI.pm; fi
 	(cd $(srcdir)/perl_scripts && $(cp) *.pl $(datadir)/vi/perl && \
 	    cd $(datadir)/vi/perl && $(chmod) $(fmode) *.pl)
 	@echo "Installing Tcl scripts: $(datadir)/vi/tcl ..."
