diff -Naur rtmpdump-2.4/librtmp/Makefile rtmpdump-2.4-p/librtmp/Makefile
--- rtmpdump-2.4/librtmp/Makefile	2011-07-12 02:24:33.000000000 +0200
+++ rtmpdump-2.4-p/librtmp/Makefile	2016-09-19 15:59:25.000936046 +0200
@@ -5,7 +5,7 @@
 incdir=$(prefix)/include/librtmp
 bindir=$(prefix)/bin
 libdir=$(prefix)/lib
-mandir=$(prefix)/man
+mandir=$(prefix)/share/man
 BINDIR=$(DESTDIR)$(bindir)
 INCDIR=$(DESTDIR)$(incdir)
 LIBDIR=$(DESTDIR)$(libdir)
diff -Naur rtmpdump-2.4/Makefile rtmpdump-2.4-p/Makefile
--- rtmpdump-2.4/Makefile	2011-07-12 02:24:33.000000000 +0200
+++ rtmpdump-2.4-p/Makefile	2016-09-19 15:56:10.339195540 +0200
@@ -26,7 +26,7 @@
 
 bindir=$(prefix)/bin
 sbindir=$(prefix)/sbin
-mandir=$(prefix)/man
+mandir=$(prefix)/share/man
 
 BINDIR=$(DESTDIR)$(bindir)
 SBINDIR=$(DESTDIR)$(sbindir)
