diff -Naur rtmpdump/librtmp/Makefile rtmpdump.new/librtmp/Makefile
--- rtmpdump/librtmp/Makefile	2022-11-11 10:40:58.229840190 +0100
+++ rtmpdump.new/librtmp/Makefile	2022-11-11 10:41:15.536862172 +0100
@@ -16,8 +16,8 @@
 AR=$(CROSS_COMPILE)ar
 
 SYS=posix
-CRYPTO=OPENSSL
-#CRYPTO=GNUTLS
+#CRYPTO=OPENSSL
+CRYPTO=GNUTLS
 DEF_POLARSSL=-DUSE_POLARSSL
 DEF_OPENSSL=-DUSE_OPENSSL
 DEF_GNUTLS=-DUSE_GNUTLS
diff -Naur rtmpdump/Makefile rtmpdump.new/Makefile
--- rtmpdump/Makefile	2022-11-11 10:40:58.228840246 +0100
+++ rtmpdump.new/Makefile	2022-11-11 10:41:15.536862172 +0100
@@ -8,9 +8,9 @@
 SYS=posix
 #SYS=mingw
 
-CRYPTO=OPENSSL
+#CRYPTO=OPENSSL
 #CRYPTO=POLARSSL
-#CRYPTO=GNUTLS
+CRYPTO=GNUTLS
 LIBZ=-lz
 LIB_GNUTLS=-lgnutls -lhogweed -lnettle -lgmp $(LIBZ)
 LIB_OPENSSL=-lssl -lcrypto $(LIBZ)
