--- config.def.h.orig	2011-08-23 20:02:46.502074256 +0200
+++ config.def.h	2011-08-23 20:03:12.943074256 +0200
@@ -20,7 +20,7 @@
 /* browser specific settings */
 char* user_agent           = NULL;
 char* browser_name         = "jumanji/0.0";
-char* home_page            = "http://pwmt.org";
+char* home_page            = "http://frugalware.org/start";
 char* proxy                = NULL;
 gboolean private_browsing  = FALSE;
 gboolean full_content_zoom = FALSE;
