diff -ur a/src/i965_drv_video.c b/src/i965_drv_video.c
--- a/src/i965_drv_video.c	2012-09-20 03:46:40.000000000 -0500
+++ b/src/i965_drv_video.c	2012-10-26 22:52:26.848806985 -0500
@@ -2275,7 +2275,7 @@
                 unsigned int flags) /* de-interlacing flags */
 {
     struct i965_driver_data *i965 = i965_driver_data(ctx); 
-    struct dri_state *dri_state = (struct dri_state *)ctx->dri_state;
+    struct dri_state *dri_state = (struct dri_state *)ctx->drm_state;
     struct i965_render_state *render_state = &i965->render_state;
     struct dri_drawable *dri_drawable;
     union dri_buffer *buffer;
@@ -2288,7 +2288,7 @@
     int pp_flag = 0;
 
     /* Currently don't support DRI1 */
-    if (dri_state->driConnectedFlag != VA_DRI2)
+    if (dri_state->base.auth_type != VA_DRI2)
         return VA_STATUS_ERROR_UNKNOWN;
 
     /* Some broken sources such as H.264 conformance case FM2_SVA_C
diff -ur a/src/intel_driver.c b/src/intel_driver.c
--- a/src/intel_driver.c	2012-09-20 02:57:06.000000000 -0500
+++ b/src/intel_driver.c	2012-10-26 22:51:35.136784789 -0500
@@ -50,15 +50,15 @@
 intel_driver_init(VADriverContextP ctx)
 {
     struct intel_driver_data *intel = intel_driver_data(ctx);
-    struct dri_state *dri_state = (struct dri_state *)ctx->dri_state;
+    struct dri_state *dri_state = (struct dri_state *)ctx->drm_state;
     int has_exec2, has_bsd, has_blt;
 
     assert(dri_state);
-    assert(dri_state->driConnectedFlag == VA_DRI2 || 
-           dri_state->driConnectedFlag == VA_DRI1);
+    assert(dri_state->base.auth_type == VA_DRI2 || 
+           dri_state->base.auth_type == VA_DRI1);
 
-    intel->fd = dri_state->fd;
-    intel->dri2Enabled = (dri_state->driConnectedFlag == VA_DRI2);
+    intel->fd = dri_state->base.fd;
+    intel->dri2Enabled = (dri_state->base.auth_type == VA_DRI2);
 
     if (!intel->dri2Enabled) {
         return False;
