diff -Nur xfmedia-0.9.1/src/Makefile.in xfmedia-0.9.1.new/src/Makefile.in
--- xfmedia-0.9.1/src/Makefile.in	2005-09-14 12:04:31.000000000 +0300
+++ xfmedia-0.9.1.new/src/Makefile.in	2006-05-09 10:45:21.000000000 +0300
@@ -401,7 +401,7 @@
 
 xfmedia_remote_LDADD =  \
 	$(top_builddir)/xfmedia/libxfmedia-remote-client.la \
-	@GETOPT_LONG_LIBS@ @GLIB_LIBS@ @DBUS_LIBS@ $(am__append_2)
+	@LIBXFCE4UTIL_LIBS@ @GETOPT_LONG_LIBS@ @GLIB_LIBS@ @DBUS_LIBS@ $(am__append_2)
 configdir = $(sysconfdir)/xdg/xfmedia
 config_DATA = keybindings.rc
 EXTRA_DIST = $(config_DATA)
