--- smuxi-0.6.3~/src/Engine/Config/Config.cs	2008-12-28 23:01:43.000000000 +0100
+++ smuxi-0.6.3/src/Engine/Config/Config.cs	2009-03-29 18:08:03.000000000 +0200
@@ -279,6 +279,12 @@
             Get(prefix + "Hostname", "irc.freenode.net");
             Get(prefix + "Port", 6667);
             Get(prefix + "Network", "freenode");
+            Get(prefix + "OnStartupConnect", true);
+            Get(prefix + "OnConnectCommands",
+                new string[] {
+                    "/join #frugalware",
+                }
+            );
             
             prefix = "Engine/Users/";
             Get(prefix+"Users", new string[] { "local" });
