diff -Naur qtwebengine-everywhere-src-6.4.1/src/3rdparty/chromium/PRESUBMIT_test_mocks.py qtwebengine-everywhere-src-6.4.1.new/src/3rdparty/chromium/PRESUBMIT_test_mocks.py
--- qtwebengine-everywhere-src-6.4.1/src/3rdparty/chromium/PRESUBMIT_test_mocks.py	2022-11-08 22:18:48.000000000 +0100
+++ qtwebengine-everywhere-src-6.4.1.new/src/3rdparty/chromium/PRESUBMIT_test_mocks.py	2022-11-28 21:03:19.697873366 +0100
@@ -128,7 +128,7 @@
   def PresubmitLocalPath(self):
     return self.presubmit_local_path
 
-  def ReadFile(self, filename, mode='rU'):
+  def ReadFile(self, filename, mode='r'):
     if hasattr(filename, 'AbsoluteLocalPath'):
        filename = filename.AbsoluteLocalPath()
     for file_ in self.files:
diff -Naur qtwebengine-everywhere-src-6.4.1/src/3rdparty/chromium/tools/grit/grit/util.py qtwebengine-everywhere-src-6.4.1.new/src/3rdparty/chromium/tools/grit/grit/util.py
--- qtwebengine-everywhere-src-6.4.1/src/3rdparty/chromium/tools/grit/grit/util.py	2022-11-08 22:18:48.000000000 +0100
+++ qtwebengine-everywhere-src-6.4.1.new/src/3rdparty/chromium/tools/grit/grit/util.py	2022-11-28 21:03:19.697873366 +0100
@@ -209,7 +209,7 @@
     mode = 'rb'
     encoding = None
   else:
-    mode = 'rU'
+    mode = 'r'
 
   with io.open(filename, mode, encoding=encoding) as f:
     return f.read()
