--- src/Makefile	2009-12-28 20:28:44.000000000 +0000
+++ src/Makefile	2009-12-28 20:30:00.000000000 +0000
@@ -24,7 +24,7 @@
 # using '$(MAKE)' instead of just 'make' allows make to run in parallel
 # over multiple makefile.
 
-all: user kernel
+all: user
 
 user: ;
 	$(MAKE) -C utils/sysdeps
@@ -40,16 +40,6 @@
 	@echo
 	@echo "Read README file for detailed information."
 
-kernel: force
-	$(MAKE) -C kernel
-	@echo "Kernel Compilation complete          Output file"
-	@echo "-----------------------------------  ----------------"
-	@echo "Built iSCSI Open Interface module:   kernel/scsi_transport_iscsi.ko"
-	@echo "Built iSCSI library module:          kernel/libiscsi.ko"
-	@echo "Built iSCSI over TCP library module: kernel/libiscsi_tcp.ko"
-	@echo "Built iSCSI over TCP kernel module:  kernel/iscsi_tcp.ko"
-
-force: ;
 
 clean:
 	$(MAKE) -C utils/sysdeps clean
@@ -63,9 +53,9 @@
 # note that make may still execute the blocks in parallel
 .NOTPARALLEL: install_user install_programs install_initd \
 	install_initd_suse install_initd_redhat install_initd_debian \
-	install_etc install_iface install_doc install_kernel install_iname
+	install_etc install_iface install_doc install_iname
 
-install: install_kernel install_programs install_doc install_etc \
+install: install_programs install_doc install_etc \
 	install_initd install_iname install_iface
 
 install_user: install_programs install_doc install_etc \
@@ -118,8 +108,6 @@
 	$(INSTALL) -d $(DESTDIR)$(mandir)/man8
 	$(INSTALL) -m 644 $^ $(DESTDIR)$(mandir)/man8
 
-install_kernel:
-	$(MAKE) -C kernel install_kernel
 
 install_iname:
 	if [ ! -f /etc/iscsi/initiatorname.iscsi ]; then \
