Index: scilab-5.5.2/bin/scilab
===================================================================
--- scilab-5.5.2.orig/bin/scilab
+++ scilab-5.5.2/bin/scilab
@@ -785,7 +785,7 @@ SCILIB=$SCI/bin:$SCI/lib/scilab/:$SCI/li
 if test -d $SCI/lib/thirdparty; then
     SCILIB=$SCI/lib/thirdparty:$SCILIB;
 fi
-LD_LIBRARY_PATH=$SCILIB${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
+#LD_LIBRARY_PATH=$SCILIB${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}
 
 case $OS in
     *darwin* | *Darwin*) # MacOS X
@@ -829,6 +829,7 @@ case $OS in
     *)
 esac
 
+LD_LIBRARY_PATH=/usr/lib/scilab/:/usr/lib64/scilab/:.${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
 
 if test $SCIVERBOSE -ne 0; then
     echo "LD_LIBRARY_PATH : $LD_LIBRARY_PATH"
@@ -1101,6 +1102,7 @@ if test "$rest" = "yes"; then
         esac
     done
 
+LD_LIBRARY_PATH=/usr/lib/scilab/:/usr/lib64/scilab/:.${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
 
 # If the argument -a is not provided (extraarg) and we found unknown args
 # fail and provide help
