From c9fb92d12fe60e819f0af20265796b1f85424c03 Mon Sep 17 00:00:00 2001
From: Fabrice Bellet <fabrice@bellet.info>
Date: Sun, 22 Jan 2012 15:30:26 +0100
Subject: [PATCH 2/2] fix build with gcc-4.7.0

---
 utils/TerraSync/terrasync.cxx |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/utils/TerraSync/terrasync.cxx b/utils/TerraSync/terrasync.cxx
index c70fbaa..25ea117 100644
--- a/utils/TerraSync/terrasync.cxx
+++ b/utils/TerraSync/terrasync.cxx
@@ -42,6 +42,7 @@
 
 #include <stdlib.h>             // atoi() atof() abs() system()
 #include <signal.h>             // signal()
+#include <unistd.h>             // sleep() getpid() write()
 
 #include <simgear/compiler.h>
 
-- 
1.7.7.5
