diff -Naur audacity-src-1.3.2-beta/src/audacity.desktop audacity-src-1.3.2-beta-misc/src/audacity.desktop
--- audacity-src-1.3.2-beta/src/audacity.desktop	2006-10-29 01:05:29.000000000 +0200
+++ audacity-src-1.3.2-beta-misc/src/audacity.desktop	2007-02-02 20:19:56.000000000 +0100
@@ -1,22 +1,22 @@
-﻿[Desktop Entry]
-Encoding=UTF-8
-
-Name=Audacity Sound Editor
-Name[de]=Audacity Audio-Editor
-Name[ru]=Редактор звуковых файлов Audacity
-GenericName=Sound Editor
-GenericName[de]=Audio-Editor
-GenericName[ru]=Редактор звуковых файлов
-Comment=Record and edit audio files
+[Desktop Entry]
+Encoding=UTF-8
+
+Name=Audacity Sound Editor
+Name[de]=Audacity Audio-Editor
+Name[ru]=Редактор звуковых файлов Audacity
+GenericName=Sound Editor
+GenericName[de]=Audio-Editor
+GenericName[ru]=Редактор звуковых файлов
+Comment=Record and edit audio files
 Comment[de]=Audio-Dateien aufnehmen und bearbeiten
-Comment[ru]=Запись и редактирование звуковых файлов
+Comment[ru]=Запись и редактирование звуковых файлов
+
+Icon=/usr/share/audacity/audacity.xpm
+
+Type=Application
+Categories=QT;GTK;Application;AudioVideo;Audio;Multimedia;
 
-Icon=/usr/share/audacity/audacity.xpm
-
-Type=Application
-Categories=Application;AudioVideo;Audio;Multimedia;
-
-Exec=audacity
-StartupNotify=false
-Terminal=false
-MimeType=application/ogg;audio/basic;audio/mpeg;audio/x-aiff;audio/x-mp3;audio/x-wav;application/x-audacity-project;
+Exec=audacity
+StartupNotify=false
+Terminal=false
+MimeType=application/ogg;audio/basic;audio/mpeg;audio/x-aiff;audio/x-mp3;audio/x-wav;application/x-audacity-project;
diff -Naur audacity-src-1.3.2-beta/src/Audacity.h audacity-src-1.3.2-beta-misc/src/Audacity.h
--- audacity-src-1.3.2-beta/src/Audacity.h	2006-10-29 01:05:28.000000000 +0200
+++ audacity-src-1.3.2-beta-misc/src/Audacity.h	2007-02-02 20:19:21.000000000 +0100
@@ -29,12 +29,12 @@
 
 // Version string for visual display
 #define AUDACITY_VERSION_STRING "1.3.2-beta"
-// Need to figure out how to concatenate for unicode builds
-//#define AUDACITY_VERSION_STRING AUDACITY_QUOTE( AUDACITY_VERSION ) "." \
-//                                AUDACITY_QUOTE( AUDACITY_RELEASE ) "." \
-//                                AUDACITY_QUOTE( AUDACITY_REVISION ) \
-//                                AUDACITY_SUFFIX
-
+/* Need to figure out how to concatenate for unicode builds
+ *#define AUDACITY_VERSION_STRING AUDACITY_QUOTE( AUDACITY_VERSION ) "." \
+ *                                AUDACITY_QUOTE( AUDACITY_RELEASE ) "." \
+ *                                AUDACITY_QUOTE( AUDACITY_REVISION ) \
+ *                              AUDACITY_SUFFIX
+ */
 // Version string for file info (under Windows)
 #define AUDACITY_FILE_VERSION AUDACITY_QUOTE( AUDACITY_VERSION ) "," \
                               AUDACITY_QUOTE( AUDACITY_RELEASE ) "," \
