--- ./lib-src/libvamp/vamp-sdk/PluginHostAdapter.h.orig	2008-03-26 16:54:38.000000000 +0100
+++ ./lib-src/libvamp/vamp-sdk/PluginHostAdapter.h	2008-03-26 16:54:48.000000000 +0100
@@ -41,6 +41,7 @@
 #include <vamp-sdk/Plugin.h>
 
 #include <vector>
+#include <cstdlib>
 
 namespace Vamp {
 
--- ./lib-src/libvamp/vamp-sdk/PluginAdapter.h.orig	2008-03-26 16:52:08.000000000 +0100
+++ ./lib-src/libvamp/vamp-sdk/PluginAdapter.h	2008-03-26 16:53:48.000000000 +0100
@@ -42,6 +42,8 @@
 #include "Plugin.h"
 
 #include <map>
+#include <cstdlib>
+#include <cstring>
 
 namespace Vamp {
 
--- ./lib-src/libvamp/vamp-sdk/hostext/PluginLoader.h.orig	2008-03-26 16:55:16.000000000 +0100
+++ ./lib-src/libvamp/vamp-sdk/hostext/PluginLoader.h	2008-03-26 16:55:26.000000000 +0100
@@ -39,6 +39,7 @@
 
 #include <vector>
 #include <string>
+#include <cstring>
 #include <map>
 
 #include "PluginWrapper.h"
