diff -Naurp audacity-src-1.3.2-beta/locale/Makefile.in audacity-src-1.3.2-beta-locale/locale/Makefile.in
--- audacity-src-1.3.2-beta/locale/Makefile.in	2006-10-29 01:06:04.000000000 +0200
+++ audacity-src-1.3.2-beta-locale/locale/Makefile.in	2007-02-03 02:01:14.000000000 +0100
@@ -17,7 +17,7 @@ top_srcdir=@top_srcdir@
 INSTALL=@INSTALL@
 AUDACITY_NAME=@AUDACITY_NAME@
 ifdef DESTDIR
-   INSTALL_PATH=$(top_srcdir)/$(DESTDIR)
+   INSTALL_PATH=$(DESTDIR)
 else
    INSTALL_PATH=
 endif
