diff -Naur audacity-src-1.3.2-beta/acinclude.m4 audacity-src-1.3.2-beta-st/acinclude.m4
--- audacity-src-1.3.2-beta/acinclude.m4	2006-10-29 01:06:04.000000000 +0200
+++ audacity-src-1.3.2-beta-st/acinclude.m4	2007-02-02 19:43:34.000000000 +0100
@@ -88,12 +88,23 @@
 
    dnl see if soundtouch is installed on the system
 
-   dnl ... TODO: I don't know if stock, unmodified soundtouch is
-   dnl suitable for audacity, so I don't allow it
+   PKG_CHECK_MODULES(SOUNDTOUCH, soundtouch-1.0 >= 1.3.1,
+                     soundtouch_available_system="yes",
+                     soundtouch_available_system="no")
 
-   LIBSOUNDTOUCH_SYSTEM_AVAILABLE="no"
 
-   dnl see if libresample is available locally
+   if test "x$soundtouch_available_system" = "xyes" ; then
+      LIBSOUNDTOUCH_SYSTEM_AVAILABLE="yes"
+      LIBSOUNDTOUCH_SYSTEM_LIBS=$SOUNDTOUCH_LIBS
+      LIBSOUNDTOUCH_SYSTEM_CXXFLAGS=$SOUNDTOUCH_CFLAGS
+      LIBSOUNDTOUCH_SYSTEM_CPPSYMBOLS="USE_SOUNDTOUCH"
+      AC_MSG_NOTICE([Libsoundtouch libraries are available as system libraries])
+   else
+      LIBSOUNDTOUCH_SYSTEM_AVAILABLE="no"
+      AC_MSG_NOTICE([Libsoundtouch libraries are NOT available as system libraries])
+   fi
+
+   dnl see if libsoundtouch is available locally
 
    AC_CHECK_FILE(${srcdir}/lib-src/soundtouch/include/SoundTouch.h,
                  soundtouch_h_found="yes",
@@ -180,6 +191,7 @@
       LIBSAMPLERATE_SYSTEM_AVAILABLE="yes"
       LIBSAMPLERATE_SYSTEM_LIBS=$SAMPLERATE_LIBS
       LIBSAMPLERATE_SYSTEM_CXXFLAGS=$SAMPLERATE_CFLAGS
+      LIBSAMPLERATE_SYSTEM_CPPSYMBOLS="USE_LIBSAMPLERATE"
       AC_MSG_NOTICE([Libsamplerate libraries are available as system libraries])
    else
       LIBSAMPLERATE_SYSTEM_AVAILABLE="no"
