diff -ur scim-m17n-0.2.2~/src/scim_m17n_imengine.cpp scim-m17n-0.2.2/src/scim_m17n_imengine.cpp
--- scim-m17n-0.2.2~/src/scim_m17n_imengine.cpp	2006-12-01 02:45:12.000000000 -0600
+++ scim-m17n-0.2.2/src/scim_m17n_imengine.cpp	2007-12-08 16:53:49.000000000 -0600
@@ -28,6 +28,7 @@
 #define Uses_SCIM_LOOKUP_TABLE
 #define Uses_SCIM_CONFIG_BASE
 #define Uses_STL_MAP
+#define Uses_C_STRING
 
 #ifdef HAVE_CONFIG_H
   #include <config.h>
