From http://www.sweeting.org/mark/blog/2006/07/11/undefined-property-etag-in-rss_fetchinc.
diff -Naur magpierss-0.72.orig/rss_fetch.inc magpierss-0.72/rss_fetch.inc
--- magpierss-0.72.orig/rss_fetch.inc	2005-11-05 22:10:56.000000000 +0100
+++ magpierss-0.72/rss_fetch.inc	2007-10-15 00:32:23.000000000 +0200
@@ -153,7 +153,8 @@
         // setup headers
         if ( $cache_status == 'STALE' ) {
             $rss = $cache->get( $cache_key );
-            if ( $rss and $rss->etag and $rss->last_modified ) {
+            if ( $rss and array_key_exists("etag", get_object_vars($rss)) and
+                array_key_exists("last_modified", get_object_vars($rss)) ) {
                 $request_headers['If-None-Match'] = $rss->etag;
                 $request_headers['If-Last-Modified'] = $rss->last_modified;
             }
