diff -ur a/src/bin/Jmakefile b/src/bin/Jmakefile
--- a/src/bin/Jmakefile	2013-12-31 05:24:24.000000000 -0600
+++ b/src/bin/Jmakefile	2014-03-14 20:04:42.175365343 -0500
@@ -38,7 +38,7 @@
 ++COMMON_LIBS $libs
 
 LDFLAGS =
-LIBS = -L../lib -lshared $(GLIB_LDFLAGS) $(COMMON_LIBS)
+LIBS = -L../lib -shared $(GLIB_LDFLAGS) $(COMMON_LIBS)
 
 RemoteTargetDependency(sha1sum, ../lib, libshared.a)
 
diff -ur a/src/bin/Makefile.SH b/src/bin/Makefile.SH
--- a/src/bin/Makefile.SH	2013-12-31 05:24:24.000000000 -0600
+++ b/src/bin/Makefile.SH	2014-03-14 20:04:47.106365162 -0500
@@ -102,7 +102,7 @@
 	$(RM) Makefile.new
 
 LDFLAGS =
-LIBS = -L../lib -lshared $(GLIB_LDFLAGS) $(COMMON_LIBS)
+LIBS = -L../lib -shared $(GLIB_LDFLAGS) $(COMMON_LIBS)
 
 .FORCE:
 
diff -ur a/src/Jmakefile b/src/Jmakefile
--- a/src/Jmakefile	2013-12-31 05:24:24.000000000 -0600
+++ b/src/Jmakefile	2014-03-14 20:04:02.979366786 -0500
@@ -80,8 +80,8 @@
 	-Lui/gtk -lgtk-common -lgtkx -lgtk-common \
 -case
 	-Lxml -lxml \
-	-Llib -lshared \
-	-Lsdbm -lsdbm -lshared \
+	-Llib -shared \
+	-Lsdbm -lsdbm -shared \
 	$(GTK_LDFLAGS) $(DBUS_LDFLAGS) $(GNUTLS_LDFLAGS) $(SOCKER_LDFLAGS)
 
 RemoteTargetDependency(gtk-gnutella, lib, libshared.a)
diff -ur a/src/Makefile.SH b/src/Makefile.SH
--- a/src/Makefile.SH	2013-12-31 05:24:24.000000000 -0600
+++ b/src/Makefile.SH	2014-03-14 20:04:50.544365035 -0500
@@ -141,8 +141,8 @@
 esac
 $spitshell >>Makefile <<'!NO!SUBS!'
 	-Lxml -lxml \
-	-Llib -lshared \
-	-Lsdbm -lsdbm -lshared \
+	-Llib -shared \
+	-Lsdbm -lsdbm -shared \
 	$(GTK_LDFLAGS) $(DBUS_LDFLAGS) $(GNUTLS_LDFLAGS) $(SOCKER_LDFLAGS)
 
 .FORCE:
diff -ur a/src/sdbm/Jmakefile b/src/sdbm/Jmakefile
--- a/src/sdbm/Jmakefile	2013-12-31 05:24:27.000000000 -0600
+++ b/src/sdbm/Jmakefile	2014-03-14 20:04:32.644365694 -0500
@@ -92,7 +92,7 @@
 ++COMMON_LIBS $libs
 
 LDFLAGS =
-LIBS = -L. -lsdbm -L../lib -lshared $(GLIB_LDFLAGS) $(COMMON_LIBS)
+LIBS = -L. -lsdbm -L../lib -shared $(GLIB_LDFLAGS) $(COMMON_LIBS)
 
 dbu dbe dbt: libsdbm.a
 RemoteTargetDependency(dbu, ../lib, libshared.a)
diff -ur a/src/sdbm/Makefile.SH b/src/sdbm/Makefile.SH
--- a/src/sdbm/Makefile.SH	2013-12-31 05:24:27.000000000 -0600
+++ b/src/sdbm/Makefile.SH	2014-03-14 20:04:35.023365607 -0500
@@ -165,7 +165,7 @@
 	util.o 
 
 LDFLAGS =
-LIBS = -L. -lsdbm -L../lib -lshared $(GLIB_LDFLAGS) $(COMMON_LIBS)
+LIBS = -L. -lsdbm -L../lib -shared $(GLIB_LDFLAGS) $(COMMON_LIBS)
 
 dbu dbe dbt: libsdbm.a
 
