--- PyQt-x11-gpl-4.5.1/configure.py	2009-06-16 10:37:38.000000000 +0200
+++ PyQt-x11-gpl-4.5.1-p/configure.py	2009-06-26 14:23:03.000000000 +0200
@@ -1466,7 +1466,7 @@
 
     # Common checks.
     if qt_edition and ltype != "internal":
-        if qt_edition != "free" and ltype == "GPL":
+        if qt_edition != "free" and ltype == "GPL" and ltype == "Opensource":
             sipconfig.error("This version of PyQt and the %s edition of Qt have incompatible licenses." % qt_edition)
 
     # Confirm the license if not already done.
