--- ./src/common.h.orig	2002-10-29 13:27:51.000000000 -0800
+++ ./src/common.h	2003-08-28 23:16:53.000000000 -0700
@@ -55,7 +55,7 @@
 #endif				/* HAVE_SYS_SOUNDCARD_H */
 
 #ifdef HAVE_ALSA
-#include <sys/asoundlib.h>
+#include <alsa/asoundlib.h>
 #endif				/* HAVE_ALSA */
 #define MAXLEVEL 100		/* highest level permitted by OSS drivers */
 enum {
--- ./configure.orig	2002-11-24 17:35:48.000000000 -0800
+++ ./configure	2003-08-28 23:14:46.000000000 -0700
@@ -1890,16 +1890,16 @@
 fi
 
 if test $alsa_must = on; then
-    ac_safe=`echo "sys/asoundlib.h" | sed 'y%./+-%__p_%'`
-echo $ac_n "checking for sys/asoundlib.h""... $ac_c" 1>&6
-echo "configure:1896: checking for sys/asoundlib.h" >&5
+    ac_safe=`echo "alsa/asoundlib.h" | sed 'y%./+-%__p_%'`
+echo $ac_n "checking for alsa/asoundlib.h""... $ac_c" 1>&6
+echo "configure:1896: checking for alsa/asoundlib.h" >&5
 if eval "test \"`echo '$''{'ac_cv_header_$ac_safe'+set}'`\" = set"; then
   echo $ac_n "(cached) $ac_c" 1>&6
 else
   cat > conftest.$ac_ext <<EOF
 #line 1901 "configure"
 #include "confdefs.h"
-#include <sys/asoundlib.h>
+#include <alsa/asoundlib.h>
 EOF
 ac_try="$ac_cpp conftest.$ac_ext >/dev/null 2>conftest.out"
 { (eval echo configure:1906: \"$ac_try\") 1>&5; (eval $ac_try) 2>&5; }
