--- NetworkManager-0.8/src/NetworkManager.conf.old	2010-04-03 21:36:35.000000000 +0200
+++ NetworkManager-0.8/src/NetworkManager.conf	2010-04-03 21:37:15.000000000 +0200
@@ -12,7 +12,7 @@
                 <allow send_destination="org.freedesktop.NetworkManager"
                        send_interface="org.freedesktop.NetworkManager.PPP"/>
         </policy>
-        <policy at_console="true">
+        <policy group="netdev">
                 <allow send_destination="org.freedesktop.NetworkManager"/>
 
                 <allow send_destination="org.freedesktop.NetworkManager"
