--- configure.ac~	2009-10-16 06:54:58.000000000 +0200
+++ configure.ac	2009-10-26 23:42:52.000000000 +0100
@@ -336,7 +336,7 @@
 	test -n "$DHCP_CLIENT_PATH" && echo bar
 	# We fully work with upstream dhcpcd-4
 	AC_MSG_CHECKING([for dhcpcd])
-	for client in "$with_dhcp_client" /sbin/dhcpcd /usr/pkg/sbin/dhcpcd /usr/local/sbin/dhcpcd; do
+	for client in "$with_dhcp_client" /usr/sbin/dhcpcd /sbin/dhcpcd /usr/pkg/sbin/dhcpcd /usr/local/sbin/dhcpcd; do
 		test -x "$client" || continue
 		case `"$client" --version 2>/dev/null` in
 			"dhcpcd "[123]*);;

