diff -ur a/picocom.c b/picocom.c
--- a/picocom.c	2012-02-21 00:40:56.000000000 -0600
+++ b/picocom.c	2014-02-03 04:47:01.393912667 -0600
@@ -224,7 +224,14 @@
 		r = read(fd, buf, sizeof(buf)); 
 		close(fd);
 		/* if r == 4, lock file is binary (old-style) */
-		pid = (r == 4) ? *(int *)buf : strtol(buf, NULL, 10);
+		if(r == 4) {
+			union { char *a; int *b; } u;
+			u.a = buf;
+			pid = *u.b;
+		}
+		else {
+			pid = strtol(buf, NULL, 10);
+		}
 		if ( pid > 0 
 			 && kill((pid_t)pid, 0) < 0 
 			 && errno == ESRCH ) {
