diff -ur a/kernel/conftest.sh b/kernel/conftest.sh
--- a/kernel/conftest.sh	2012-04-05 23:37:05.000000000 -0500
+++ b/kernel/conftest.sh	2012-05-21 20:18:28.952140388 -0500
@@ -512,7 +512,6 @@
             # and if it as an 'event' member.
             #
             echo "$CONFTEST_PREAMBLE
-            #include <asm/system.h>
             #include <linux/pm.h>
             void conftest_pm_message_t(pm_message_t state) {
                 pm_message_t *p = &state;
diff -ur a/kernel/nv-linux.h b/kernel/nv-linux.h
--- a/kernel/nv-linux.h	2012-04-05 23:37:05.000000000 -0500
+++ b/kernel/nv-linux.h	2012-05-21 20:18:02.518141358 -0500
@@ -111,7 +111,6 @@
 #include <linux/timer.h>
 
 #include <asm/div64.h>              /* do_div()                         */
-#include <asm/system.h>             /* cli, sli, save_flags             */
 #include <asm/io.h>                 /* ioremap, virt_to_phys            */
 #include <asm/uaccess.h>            /* access_ok                        */
 #include <asm/page.h>               /* PAGE_OFFSET                      */
