diff -ur a/kernel/nv-acpi.c b/kernel/nv-acpi.c
--- a/kernel/nv-acpi.c	2012-10-15 12:05:31.639469963 -0500
+++ b/kernel/nv-acpi.c	2012-10-15 12:05:50.887469254 -0500
@@ -300,7 +300,7 @@
     if (pNvAcpiObject->notify_handler_installed)
     {
         // no status returned for this function
-        acpi_os_wait_events_complete(NULL);
+        acpi_os_wait_events_complete();
 
         // remove event notifier
         status = acpi_remove_notify_handler(device->handle, ACPI_DEVICE_NOTIFY, nv_acpi_event);
