diff -ur a/kernel/conftest.sh b/kernel/conftest.sh
--- a/kernel/conftest.sh	2012-03-23 11:21:54.788942135 -0500
+++ b/kernel/conftest.sh	2012-03-23 11:22:09.607941580 -0500
@@ -122,7 +122,7 @@
     CFLAGS="$BASE_CFLAGS $MACH_CFLAGS $OUTPUT_CFLAGS -I$HEADERS $AUTOCONF_CFLAGS"
 
     if [ "$ARCH" = "i386" -o "$ARCH" = "x86_64" ]; then
-        CFLAGS="$CFLAGS -I$SOURCES/arch/x86/include"
+        CFLAGS="$CFLAGS -I$SOURCES/arch/x86/include -I$SOURCES/arch/x86/include/generated"
     fi
     if [ -n "$BUILD_PARAMS" ]; then
         CFLAGS="$CFLAGS -D$BUILD_PARAMS"
