diff -Naur NVIDIA-Linux-x86_64-384.69-no-compat32/kernel/common/inc/nv-linux.h NVIDIA-Linux-x86_64-384.69-no-compat32.new/kernel/common/inc/nv-linux.h
--- NVIDIA-Linux-x86_64-384.69-no-compat32/kernel/common/inc/nv-linux.h	2017-08-17 06:23:25.000000000 +0200
+++ NVIDIA-Linux-x86_64-384.69-no-compat32.new/kernel/common/inc/nv-linux.h	2017-09-04 15:42:06.000000000 +0200
@@ -128,7 +128,7 @@
 #include <asm/system.h>             /* cli, sli, save_flags             */
 #endif
 #include <asm/io.h>                 /* ioremap, virt_to_phys            */
-#include <asm/uaccess.h>            /* access_ok                        */
+#include <linux/uaccess.h>            /* access_ok                        */
 #include <asm/page.h>               /* PAGE_OFFSET                      */
 #include <asm/pgtable.h>            /* pte bit definitions              */
 
diff -Naur NVIDIA-Linux-x86_64-384.69-no-compat32/kernel/nvidia-uvm/uvm_unsupported.c NVIDIA-Linux-x86_64-384.69-no-compat32.new/kernel/nvidia-uvm/uvm_unsupported.c
--- NVIDIA-Linux-x86_64-384.69-no-compat32/kernel/nvidia-uvm/uvm_unsupported.c	2017-08-17 06:23:26.000000000 +0200
+++ NVIDIA-Linux-x86_64-384.69-no-compat32.new/kernel/nvidia-uvm/uvm_unsupported.c	2017-09-04 15:42:37.000000000 +0200
@@ -42,7 +42,7 @@
 #include "conftest.h"
 
 #include <linux/module.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <linux/cdev.h>
 #include <linux/fs.h>
 
