diff -ur b/kernel/nvidia-uvm/uvm_linux.h b/kernel/nvidia-uvm/uvm_linux.h
--- kernel/nvidia-uvm/uvm_linux.h       2016-06-10 02:37:08.000000000 +0200
+++ kernel/nvidia-uvm/uvm_linux.h       2016-06-14 02:49:35.214495381 +0200
@@ -563,12 +563,13 @@
     INIT_RADIX_TREE(tree, GFP_NOWAIT);
 }
 
+/*
 static bool radix_tree_empty(struct radix_tree_root *tree)
 {
     void *dummy;
     return radix_tree_gang_lookup(tree, &dummy, 0, 1) == 0;
 }
-
+*/
 
 #if !defined(NV_USLEEP_RANGE_PRESENT)
 static void __sched usleep_range(unsigned long min, unsigned long max)
