--- NVIDIA-Linux-x86_64-381.22-no-compat32/kernel/nvidia-uvm/uvm_unsupported.c	2017-05-04 10:50:27.000000000 +0200
+++ NVIDIA-Linux-x86_64-381.22-no-compat32.new/kernel/nvidia-uvm/uvm_unsupported.c	2017-07-04 15:08:53.000000000 +0200
@@ -42,7 +42,7 @@
 #include "conftest.h"
 
 #include <linux/module.h>
-#include <asm/uaccess.h>
+#include <linux/uaccess.h>
 #include <linux/cdev.h>
 #include <linux/fs.h>
 
