--- NVIDIA-Linux-x86_64-352.41-no-compat32/kernel/nv-procfs.c	2015-08-22 06:47:24.000000000 +0200
+++ NVIDIA-Linux-x86_64-352.41-no-compat32.new/kernel/nv-procfs.c	2015-11-03 08:59:12.000000000 +0100
@@ -358,7 +358,8 @@
     registry_keys = ((nvl != NULL) ?
             nvl->registry_keys : nv_registry_keys);
 
-    return seq_printf(s, "Binary: \"%s\"\n", registry_keys);
+     seq_printf(s, "Binary: \"%s\"\n", registry_keys);
+     return 0;
 }
 
 static ssize_t
@@ -558,7 +559,8 @@
     void *v
 )
 {
-    return seq_puts(s, s->private);
+    seq_puts(s, s->private);
+    return 0;
 }
 
 NV_DEFINE_PROCFS_SINGLE_FILE(text_file);
