diff -ur cssed-0.4.0.orig/scintilla/src/LexCaml.cxx cssed-0.4.0/scintilla/src/LexCaml.cxx
--- cssed-0.4.0.orig/scintilla/src/LexCaml.cxx	2005-05-27 06:30:23.000000000 +0300
+++ cssed-0.4.0/scintilla/src/LexCaml.cxx	2009-08-06 01:58:38.000000000 +0300
@@ -273,7 +273,7 @@
 
 		case SCE_CAML_OPERATOR: {
 			// [try to] interpret as [additional] operator char
-			char* o = 0;
+			const char* o = 0;
 			if (iscaml(ch) || isspace(ch)			/* ident or whitespace */
 				|| ((o = strchr(")]};,\'\"`#", ch)) != 0)/* "termination" chars */
 				|| !strchr("!$%&*+-./:<=>?@^|~", ch)/* "operator" chars */) {
