diff -Naur qt-everywhere-opensource-src-4.8.7/config.tests/unix/alsa/alsatest.cpp qt-everywhere-opensource-src-4.8.7-alsa/config.tests/unix/alsa/alsatest.cpp
--- qt-everywhere-opensource-src-4.8.7/config.tests/unix/alsa/alsatest.cpp	2015-05-07 16:14:42.000000000 +0200
+++ qt-everywhere-opensource-src-4.8.7-alsa/config.tests/unix/alsa/alsatest.cpp	2016-03-23 11:53:28.891766299 +0100
@@ -40,7 +40,7 @@
 ****************************************************************************/
 
 #include <alsa/asoundlib.h>
-#if(!(SND_LIB_MAJOR == 1 && SND_LIB_MINOR == 0 && SND_LIB_SUBMINOR >= 10))
+#if(!(SND_LIB_MAJOR == 1 && (SND_LIB_MINOR > 0 || SND_LIB_SUBMINOR >= 10)))
 #error "Alsa version found too old, require >= 1.0.10"
 #endif
 
