diff -Naur qt-x11-opensource-src-4.2.0/mkspecs/common/g++.conf qt-x11-opensource-src-4.2.0-p/mkspecs/common/g++.conf
--- qt-x11-opensource-src-4.2.0/mkspecs/common/g++.conf	2006-09-29 16:44:55.000000000 +0200
+++ qt-x11-opensource-src-4.2.0-p/mkspecs/common/g++.conf	2006-10-05 15:38:30.000000000 +0200
@@ -11,7 +11,7 @@
 QMAKE_CFLAGS_DEPS	+= -M
 QMAKE_CFLAGS_WARN_ON	+= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	+= -w
-QMAKE_CFLAGS_RELEASE	+= -O2
+QMAKE_CFLAGS_RELEASE	+= DUMMY
 QMAKE_CFLAGS_DEBUG	+= -g
 QMAKE_CFLAGS_SHLIB	+= -fPIC
 QMAKE_CFLAGS_STATIC_LIB	+= -fPIC
diff -Naur qt-x11-opensource-src-4.2.0/mkspecs/common/linux.conf qt-x11-opensource-src-4.2.0-p/mkspecs/common/linux.conf
--- qt-x11-opensource-src-4.2.0/mkspecs/common/linux.conf	2006-09-29 16:44:55.000000000 +0200
+++ qt-x11-opensource-src-4.2.0-p/mkspecs/common/linux.conf	2006-10-05 15:53:38.000000000 +0200
@@ -3,12 +3,12 @@
 
 QMAKE_INCDIR          =
 QMAKE_LIBDIR          =
-QMAKE_INCDIR_X11      = /usr/X11R6/include
-QMAKE_LIBDIR_X11      = /usr/X11R6/lib
+QMAKE_INCDIR_X11      = /usr/X11/include
+QMAKE_LIBDIR_X11      = /usr/X11/lib
 QMAKE_INCDIR_QT       = $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT       = $$[QT_INSTALL_LIBS]
-QMAKE_INCDIR_OPENGL   = /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL   = /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL   = /usr/X11/include
+QMAKE_LIBDIR_OPENGL   = /usr/X11/lib
 
 QMAKE_LIBS            =
 QMAKE_LIBS_DYNLOAD    = -ldl
