diff -Naur a//config.tests/unix/bsymbolic_functions.test b//config.tests/unix/bsymbolic_functions.test
--- a//config.tests/unix/bsymbolic_functions.test	2010-11-06 02:55:11.000000000 +0100
+++ b//config.tests/unix/bsymbolic_functions.test	2010-12-25 22:51:25.000000000 +0100
@@ -20,3 +20,4 @@
     [ "$VERBOSE" = "yes" ] && echo "Symbolic function binding enabled."
     exit 1
 fi
+
diff -Naur a//config.tests/unix/fvisibility.test b//config.tests/unix/fvisibility.test
--- a//config.tests/unix/fvisibility.test	2010-11-06 02:55:11.000000000 +0100
+++ b//config.tests/unix/fvisibility.test	2010-12-25 22:52:57.000000000 +0100
@@ -25,15 +25,15 @@
 EOF
 
     if [ "$VERBOSE" = "yes" ] ; then
-        "$COMPILER" -c $CMDLINE fvisibility.c && FVISIBILITY_SUPPORT=yes
+        $COMPILER -c -fvisibility=hidden fvisibility.c && FVISIBILITY_SUPPORT=yes
     else
-        "$COMPILER" -c $CMDLINE fvisibility.c >/dev/null 2>&1 && FVISIBILITY_SUPPORT=yes
+        $COMPILER -c -fvisibility=hidden fvisibility.c >/dev/null 2>&1 && FVISIBILITY_SUPPORT=yes
     fi
     rm -f fvisibility.c fvisibility.o
 }
 
 
-case "$COMPILER" in
+case $COMPILER in
 *g++*|*c++*)
     CMDLINE="-fvisibility=hidden"
     RunCompileTest
@@ -71,3 +71,4 @@
     [ "$VERBOSE" = "yes" ] && echo "Symbol visibility control enabled."
     exit 1
 fi
+
diff -Naur a//config.tests/unix/precomp.test b//config.tests/unix/precomp.test
--- a//config.tests/unix/precomp.test	2010-11-06 02:55:11.000000000 +0100
+++ b//config.tests/unix/precomp.test	2010-12-25 22:53:41.000000000 +0100
@@ -4,7 +4,7 @@
 COMPILER=$1
 VERBOSE=$2
 
-case "$COMPILER" in
+case $COMPILER in
 icpc)
     cat >header.h <<EOF
 #define HEADER_H
@@ -28,7 +28,7 @@
     ;;
 
 *g++*|c++)
-    case `"$COMPILER" -dumpversion 2>/dev/null` in
+    case `$COMPILER -dumpversion 2>/dev/null` in
     3.*)
         ;;
     *)
@@ -52,3 +52,4 @@
     [ "$VERBOSE" = "yes" ] && echo "Precompiled-headers support enabled."
     exit 1
 fi
+
