diff -Naur qt-everywhere-opensource-src-4.6.1/config.tests/unix/bsymbolic_functions.test qt-everywhere-opensource-src-4.6.1-p/config.tests/unix/bsymbolic_functions.test
--- qt-everywhere-opensource-src-4.6.1/config.tests/unix/bsymbolic_functions.test	2010-01-15 18:01:36.000000000 +0100
+++ qt-everywhere-opensource-src-4.6.1-p/config.tests/unix/bsymbolic_functions.test	2010-02-01 18:46:39.000000000 +0100
@@ -8,7 +8,7 @@
 int main() { return 0; }
 EOF
 
-"$COMPILER" -o libtest.so -shared -Wl,-Bsymbolic-functions -fPIC bsymbolic_functions.c >/dev/null 2>&1 && BSYMBOLIC_FUNCTIONS_SUPPORT=yes
+$COMPILER -o libtest.so -shared -Wl,-Bsymbolic-functions -fPIC bsymbolic_functions.c >/dev/null 2>&1 && BSYMBOLIC_FUNCTIONS_SUPPORT=yes
 rm -f bsymbolic_functions.c libtest.so
 
 # done
diff -Naur qt-everywhere-opensource-src-4.6.1/config.tests/unix/fvisibility.test qt-everywhere-opensource-src-4.6.1-p/config.tests/unix/fvisibility.test
--- qt-everywhere-opensource-src-4.6.1/config.tests/unix/fvisibility.test	2010-01-15 18:01:36.000000000 +0100
+++ qt-everywhere-opensource-src-4.6.1-p/config.tests/unix/fvisibility.test	2010-02-01 18:46:39.000000000 +0100
@@ -15,14 +15,14 @@
 EOF
 
     if [ "$VERBOSE" = "yes" ] ; then
-        "$COMPILER" -c -fvisibility=hidden fvisibility.c && FVISIBILITY_SUPPORT=yes
+        $COMPILER -c -fvisibility=hidden fvisibility.c && FVISIBILITY_SUPPORT=yes
     else
-        "$COMPILER" -c -fvisibility=hidden fvisibility.c >/dev/null 2>&1 && FVISIBILITY_SUPPORT=yes
+        $COMPILER -c -fvisibility=hidden fvisibility.c >/dev/null 2>&1 && FVISIBILITY_SUPPORT=yes
     fi
     rm -f fvisibility.c fvisibility.o
 }
 
-case "$COMPILER" in
+case $COMPILER in
 aCC*)
     ;;
 
diff -Naur qt-everywhere-opensource-src-4.6.1/config.tests/unix/precomp.test qt-everywhere-opensource-src-4.6.1-p/config.tests/unix/precomp.test
--- qt-everywhere-opensource-src-4.6.1/config.tests/unix/precomp.test	2010-01-15 18:01:36.000000000 +0100
+++ qt-everywhere-opensource-src-4.6.1-p/config.tests/unix/precomp.test	2010-02-01 18:46:39.000000000 +0100
@@ -4,7 +4,7 @@
 COMPILER=$1
 VERBOSE=$2
 
-case "$COMPILER" in
+case $COMPILER in
 icpc)
     cat >header.h <<EOF
 #define HEADER_H
@@ -28,7 +28,7 @@
     ;;
 
 *g++*|c++)
-    case `"$COMPILER" -dumpversion 2>/dev/null` in
+    case `$COMPILER -dumpversion 2>/dev/null` in
     3.*)
         ;;
     *)
