diff -Naur qt-everywhere-opensource-src-4.6.1/mkspecs/common/g++.conf qt-everywhere-opensource-src-4.6.1-p/mkspecs/common/g++.conf
--- qt-everywhere-opensource-src-4.6.1/mkspecs/common/g++.conf	2010-01-15 18:01:28.000000000 +0100
+++ qt-everywhere-opensource-src-4.6.1-p/mkspecs/common/g++.conf	2010-01-31 21:12:07.000000000 +0100
@@ -7,7 +7,7 @@
 QMAKE_CFLAGS_DEPS	+= -M
 QMAKE_CFLAGS_WARN_ON	+= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	+= -w
-QMAKE_CFLAGS_RELEASE	+= -O2
+QMAKE_CFLAGS_RELEASE	+= FW_CFLAGS
 QMAKE_CFLAGS_DEBUG	+= -g
 QMAKE_CFLAGS_SHLIB	+= -fPIC
 QMAKE_CFLAGS_STATIC_LIB	+= -fPIC
@@ -35,7 +35,7 @@
 QMAKE_LINK_C		= gcc
 QMAKE_LINK_C_SHLIB	= gcc
 QMAKE_LFLAGS		+=
-QMAKE_LFLAGS_RELEASE	+= -Wl,-O1
+QMAKE_LFLAGS_RELEASE	+= FW_LDFLAGS
 QMAKE_LFLAGS_DEBUG	+=
 QMAKE_LFLAGS_APP	+=
 QMAKE_LFLAGS_SHLIB	+= -shared
diff -Naur qt-everywhere-opensource-src-4.6.1/mkspecs/common/linux.conf qt-everywhere-opensource-src-4.6.1-p/mkspecs/common/linux.conf
--- qt-everywhere-opensource-src-4.6.1/mkspecs/common/linux.conf	2010-01-15 18:01:28.000000000 +0100
+++ qt-everywhere-opensource-src-4.6.1-p/mkspecs/common/linux.conf	2010-01-31 21:10:43.000000000 +0100
@@ -7,12 +7,12 @@
 
 QMAKE_INCDIR          =
 QMAKE_LIBDIR          =
-QMAKE_INCDIR_X11      = /usr/X11R6/include
-QMAKE_LIBDIR_X11      = /usr/X11R6/lib
+QMAKE_INCDIR_X11      = /usr/include
+QMAKE_LIBDIR_X11      = /usr/lib
 QMAKE_INCDIR_QT       = $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT       = $$[QT_INSTALL_LIBS]
-QMAKE_INCDIR_OPENGL   = /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL   = /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL   = /usr/include
+QMAKE_LIBDIR_OPENGL   = /usr/lib
 QMAKE_INCDIR_OPENGL_ES1 = $$QMAKE_INCDIR_OPENGL
 QMAKE_LIBDIR_OPENGL_ES1 = $$QMAKE_LIBDIR_OPENGL
 QMAKE_INCDIR_OPENGL_ES1CL = $$QMAKE_INCDIR_OPENGL
diff -Naur qt-everywhere-opensource-src-4.6.1/mkspecs/linux-g++-32/qmake.conf qt-everywhere-opensource-src-4.6.1-p/mkspecs/linux-g++-32/qmake.conf
--- qt-everywhere-opensource-src-4.6.1/mkspecs/linux-g++-32/qmake.conf	2010-01-15 18:01:28.000000000 +0100
+++ qt-everywhere-opensource-src-4.6.1-p/mkspecs/linux-g++-32/qmake.conf	2010-01-31 21:10:43.000000000 +0100
@@ -8,8 +8,8 @@
 QT			+= core gui
 QMAKE_INCREMENTAL_STYLE = sublib
 
-QMAKE_CFLAGS		= -m32
-QMAKE_LFLAGS		= -m32
+QMAKE_CFLAGS		= 
+QMAKE_LFLAGS		= 
 
 include(../common/g++.conf)
 include(../common/linux.conf)
diff -Naur qt-everywhere-opensource-src-4.6.1/mkspecs/linux-g++-64/qmake.conf qt-everywhere-opensource-src-4.6.1-p/mkspecs/linux-g++-64/qmake.conf
--- qt-everywhere-opensource-src-4.6.1/mkspecs/linux-g++-64/qmake.conf	2010-01-15 18:01:28.000000000 +0100
+++ qt-everywhere-opensource-src-4.6.1-p/mkspecs/linux-g++-64/qmake.conf	2010-01-31 21:10:43.000000000 +0100
@@ -11,13 +11,13 @@
 QT			+= core gui
 QMAKE_INCREMENTAL_STYLE = sublib
 
-QMAKE_CFLAGS		= -m64
-QMAKE_LFLAGS		= -m64
+QMAKE_CFLAGS		=
+QMAKE_LFLAGS		= 
 
 include(../common/g++.conf)
 include(../common/linux.conf)
 
-QMAKE_LIBDIR_X11      = /usr/X11R6/lib64
-QMAKE_LIBDIR_OPENGL   = /usr/X11R6/lib64
+QMAKE_LIBDIR_X11      = /usr/lib
+QMAKE_LIBDIR_OPENGL   = /usr/lib
 
 load(qt_config)
