diff -Naur a//mkspecs/common/g++.conf b//mkspecs/common/g++.conf
--- a//mkspecs/common/g++.conf	2010-11-06 02:55:11.000000000 +0100
+++ b//mkspecs/common/g++.conf	2010-12-25 22:02:51.000000000 +0100
@@ -7,7 +7,7 @@
 QMAKE_CFLAGS_DEPS	+= -M
 QMAKE_CFLAGS_WARN_ON	+= -Wall -W
 QMAKE_CFLAGS_WARN_OFF	+= -w
-QMAKE_CFLAGS_RELEASE	+= -O2
+QMAKE_CFLAGS_RELEASE	+= FW_CFLAGS
 QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO += -O2 -g
 QMAKE_CFLAGS_DEBUG	+= -g
 QMAKE_CFLAGS_SHLIB	+= -fPIC
@@ -37,7 +37,7 @@
 QMAKE_LINK_C		= gcc
 QMAKE_LINK_C_SHLIB	= gcc
 QMAKE_LFLAGS		+=
-QMAKE_LFLAGS_RELEASE	+= -Wl,-O1
+QMAKE_LFLAGS_RELEASE	+= FW_LDFLAGS
 QMAKE_LFLAGS_DEBUG	+=
 QMAKE_LFLAGS_APP	+=
 QMAKE_LFLAGS_SHLIB	+= -shared
@@ -52,3 +52,4 @@
 # -Bsymbolic-functions (ld) support
 QMAKE_LFLAGS_BSYMBOLIC_FUNC = -Wl,-Bsymbolic-functions
 QMAKE_LFLAGS_DYNAMIC_LIST = -Wl,--dynamic-list,
+
diff -Naur a//mkspecs/common/linux.conf b//mkspecs/common/linux.conf
--- a//mkspecs/common/linux.conf	2010-11-06 02:55:11.000000000 +0100
+++ b//mkspecs/common/linux.conf	2010-12-25 22:03:45.000000000 +0100
@@ -7,19 +7,19 @@
 
 QMAKE_INCDIR          =
 QMAKE_LIBDIR          =
-QMAKE_INCDIR_X11      = /usr/X11R6/include
-QMAKE_LIBDIR_X11      = /usr/X11R6/lib
+QMAKE_INCDIR_X11      = /usr/include
+QMAKE_LIBDIR_X11      = /usr/lib
 QMAKE_INCDIR_QT       = $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT       = $$[QT_INSTALL_LIBS]
-QMAKE_INCDIR_OPENGL   = /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL   = /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL   = /usr/include
+QMAKE_LIBDIR_OPENGL   = /usr/lib
 QMAKE_INCDIR_OPENGL_ES1 = $$QMAKE_INCDIR_OPENGL
 QMAKE_LIBDIR_OPENGL_ES1 = $$QMAKE_LIBDIR_OPENGL
 QMAKE_INCDIR_OPENGL_ES2 = $$QMAKE_INCDIR_OPENGL
 QMAKE_LIBDIR_OPENGL_ES2 = $$QMAKE_LIBDIR_OPENGL
-QMAKE_INCDIR_EGL      = 
+QMAKE_INCDIR_EGL      =
 QMAKE_LIBDIR_EGL      =
-QMAKE_INCDIR_OPENVG   = 
+QMAKE_INCDIR_OPENVG   =
 QMAKE_LIBDIR_OPENVG   =
 
 QMAKE_LIBS            =
@@ -59,3 +59,4 @@
 QMAKE_INSTALL_PROGRAM = install -m 755 -p
 
 include(unix.conf)
+
diff -Naur a//mkspecs/linux-g++-32/qmake.conf b//mkspecs/linux-g++-32/qmake.conf
--- a//mkspecs/linux-g++-32/qmake.conf	2010-11-06 02:55:11.000000000 +0100
+++ b//mkspecs/linux-g++-32/qmake.conf	2010-12-25 22:04:30.000000000 +0100
@@ -9,9 +9,10 @@
 QT			+= core gui
 QMAKE_INCREMENTAL_STYLE = sublib
 
-QMAKE_CFLAGS		= -m32
-QMAKE_LFLAGS		= -m32
+QMAKE_CFLAGS		=
+QMAKE_LFLAGS		=
 
 include(../common/g++.conf)
 include(../common/linux.conf)
 load(qt_config)
+
diff -Naur a//mkspecs/linux-g++-64/qmake.conf b//mkspecs/linux-g++-64/qmake.conf
--- a//mkspecs/linux-g++-64/qmake.conf	2010-11-06 02:55:11.000000000 +0100
+++ b//mkspecs/linux-g++-64/qmake.conf	2010-12-25 22:05:22.000000000 +0100
@@ -12,13 +12,14 @@
 QT			+= core gui
 QMAKE_INCREMENTAL_STYLE = sublib
 
-QMAKE_CFLAGS		= -m64
-QMAKE_LFLAGS		= -m64
+QMAKE_CFLAGS		=
+QMAKE_LFLAGS		=
 
 include(../common/g++.conf)
 include(../common/linux.conf)
 
-QMAKE_LIBDIR_X11      = /usr/X11R6/lib64
-QMAKE_LIBDIR_OPENGL   = /usr/X11R6/lib64
+QMAKE_LIBDIR_X11      = /usr/lib
+QMAKE_LIBDIR_OPENGL   = /usr/lib
 
 load(qt_config)
+
