--- qt-everywhere-opensource-src-4.8.0/mkspecs/common/gcc-base.conf	2011-12-08 05:06:02.000000000 +0000
+++ qt-new/mkspecs/common/gcc-base.conf	2011-12-19 00:05:33.000000000 +0000
@@ -35,7 +35,7 @@
 QMAKE_CFLAGS_DEPS           += -M
 QMAKE_CFLAGS_WARN_ON        += -Wall -W
 QMAKE_CFLAGS_WARN_OFF       += -w
-QMAKE_CFLAGS_RELEASE        += -O2
+QMAKE_CFLAGS_RELEASE        += FW_CFLAGS
 QMAKE_CFLAGS_DEBUG          += -g
 QMAKE_CFLAGS_SHLIB          += -fPIC
 QMAKE_CFLAGS_STATIC_LIB     += -fPIC
@@ -56,5 +56,5 @@
 QMAKE_LFLAGS         +=
 QMAKE_LFLAGS_DEBUG   +=
 QMAKE_LFLAGS_APP     +=
-QMAKE_LFLAGS_RELEASE +=
+QMAKE_LFLAGS_RELEASE += FW_LDFLAGS
 
--- qt-everywhere-opensource-src-4.8.0/mkspecs/common/linux.conf	2011-12-08 05:06:02.000000000 +0000
+++ qt-new/mkspecs/common/linux.conf	2011-12-19 00:15:26.000000000 +0000
@@ -7,12 +7,12 @@
 
 QMAKE_INCDIR          =
 QMAKE_LIBDIR          =
-QMAKE_INCDIR_X11      = /usr/X11R6/include
-QMAKE_LIBDIR_X11      = /usr/X11R6/lib
+QMAKE_INCDIR_X11      = /usr/include
+QMAKE_LIBDIR_X11      = /usr/lib
 QMAKE_INCDIR_QT       = $$[QT_INSTALL_HEADERS]
 QMAKE_LIBDIR_QT       = $$[QT_INSTALL_LIBS]
-QMAKE_INCDIR_OPENGL   = /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL   = /usr/X11R6/lib
+QMAKE_INCDIR_OPENGL   = /usr/include
+QMAKE_LIBDIR_OPENGL   = /usr/lib
 QMAKE_INCDIR_OPENGL_ES1 = $$QMAKE_INCDIR_OPENGL
 QMAKE_LIBDIR_OPENGL_ES1 = $$QMAKE_LIBDIR_OPENGL
 QMAKE_INCDIR_OPENGL_ES2 = $$QMAKE_INCDIR_OPENGL

--- qt-everywhere-opensource-src-4.8.0/mkspecs/linux-g++-32/qmake.conf	2011-12-08 05:06:02.000000000 +0000
+++ qt-new/mkspecs/linux-g++-32/qmake.conf	2011-12-19 00:24:01.000000000 +0000
@@ -9,8 +9,8 @@
 QT			+= core gui
 QMAKE_INCREMENTAL_STYLE = sublib
 
-QMAKE_CFLAGS		= -m32
-QMAKE_LFLAGS		= -m32
+QMAKE_CFLAGS		=
+QMAKE_LFLAGS		=
 
 include(../common/linux.conf)
 include(../common/gcc-base-unix.conf)


--- qt-everywhere-opensource-src-4.8.0/mkspecs/linux-g++-64/qmake.conf	2011-12-08 05:06:02.000000000 +0000
+++ qt-new/mkspecs/linux-g++-64/qmake.conf	2011-12-19 00:28:12.000000000 +0000
@@ -12,15 +12,15 @@
 QT			+= core gui
 QMAKE_INCREMENTAL_STYLE = sublib
 
-QMAKE_CFLAGS		= -m64
-QMAKE_LFLAGS		= -m64
+QMAKE_CFLAGS		=
+QMAKE_LFLAGS		=
 
 include(../common/linux.conf)
 include(../common/gcc-base-unix.conf)
 include(../common/g++-unix.conf)
 
 
-QMAKE_LIBDIR_X11      = /usr/X11R6/lib64
-QMAKE_LIBDIR_OPENGL   = /usr/X11R6/lib64
+QMAKE_LIBDIR_X11      =
+QMAKE_LIBDIR_OPENGL   =
 
 load(qt_config)
