diff -Naur qt-everywhere-opensource-src-4.6.2/src/gui/widgets/qlineedit.cpp qt-everywhere-opensource-src-4.6.2-p/src/gui/widgets/qlineedit.cpp
--- qt-everywhere-opensource-src-4.6.2/src/gui/widgets/qlineedit.cpp	2010-02-11 16:55:22.000000000 +0100
+++ qt-everywhere-opensource-src-4.6.2-p/src/gui/widgets/qlineedit.cpp	2010-02-24 01:21:00.000000000 +0100
@@ -384,7 +384,7 @@
 }
 
 // ### Qt 4.7: remove this #if guard
-#if (QT_VERSION >= 0x407000) || defined(Q_WS_MAEMO_5)
+//#if (QT_VERSION >= 0x407000) || defined(Q_WS_MAEMO_5)
 /*!
     \since 4.7
 
@@ -414,7 +414,7 @@
             update();
     }
 }
-#endif
+//#endif
 
 /*!
     \property QLineEdit::displayText
diff -Naur qt-everywhere-opensource-src-4.6.2/src/gui/widgets/qlineedit.h qt-everywhere-opensource-src-4.6.2-p/src/gui/widgets/qlineedit.h
--- qt-everywhere-opensource-src-4.6.2/src/gui/widgets/qlineedit.h	2010-02-11 16:55:22.000000000 +0100
+++ qt-everywhere-opensource-src-4.6.2-p/src/gui/widgets/qlineedit.h	2010-02-24 00:32:30.000000000 +0100
@@ -84,9 +84,9 @@
     Q_PROPERTY(bool redoAvailable READ isRedoAvailable)
     Q_PROPERTY(bool acceptableInput READ hasAcceptableInput)
 // ### Qt 4.7: remove this #if guard
-#if (QT_VERSION >= 0x407000) || defined(Q_WS_MAEMO_5)
+//#if (QT_VERSION >= 0x407000) || defined(Q_WS_MAEMO_5)
     Q_PROPERTY(QString placeholderText READ placeholderText WRITE setPlaceholderText)
-#endif
+//#endif
 
 public:
     explicit QLineEdit(QWidget* parent=0);
@@ -103,10 +103,10 @@
     QString displayText() const;
 
 // ### Qt 4.7: remove this #if guard
-#if (QT_VERSION >= 0x407000) || defined(Q_WS_MAEMO_5)
+//#if (QT_VERSION >= 0x407000) || defined(Q_WS_MAEMO_5)
     QString placeholderText() const;
     void setPlaceholderText(const QString &);
-#endif
+//#endif
 
     int maxLength() const;
     void setMaxLength(int);
