--- qt-x11-opensource-src-4.5.2/tools/qdoc3/qdoc3.pro	2009-06-20 06:57:59.000000000 +0200
+++ qt-x11-opensource-src-4.5.2-p/tools/qdoc3/qdoc3.pro	2009-06-26 16:08:17.000000000 +0200
@@ -1,4 +1,7 @@
+TEMPLATE = app
+CONFIG  += qt warn_on
 DEFINES += QDOC2_COMPAT
+
 #DEFINES += QT_NO_CAST_TO_ASCII
 #DEFINES += QT_NO_CAST_FROM_ASCII
 
@@ -99,3 +102,12 @@
            webxmlgenerator.cpp \
 	   yyindent.cpp
 
+DESTDIR     = ../../bin
+
+DEPENDPATH += .
+INCLUDEPATH += .
+TARGET = qdoc3
+
+target.path=$$[QT_INSTALL_BINS]
+INSTALLS += target
+
