--- qt-x11-opensource-src-4.1.0/tools/designer/src/uitools/uitools.pro.orig	2005-12-16 12:52:20.000000000 +0100
+++ qt-x11-opensource-src-4.1.0/tools/designer/src/uitools/uitools.pro	2006-02-13 09:26:05.000000000 +0100
@@ -1,7 +1,7 @@
 TEMPLATE = lib
 TARGET = QtUiTools
 QT += xml
-CONFIG += qt staticlib
+CONFIG += qt
 DESTDIR = ../../../../lib
 DLLDESTDIR = ../../../../bin
 
@@ -21,6 +21,12 @@
 QMAKE_TARGET_DESCRIPTION = QUiLoader
 QMAKE_TARGET_COPYRIGHT = Copyright (c) 2003-2005 Trolltech
 
+!contains(CONFIG, static) {
+    CONFIG += dll
+} else {
+    CONFIG += staticlib
+}
+
 include(../lib/uilib/uilib.pri)
 
 HEADERS += quiloader.h
@@ -32,6 +32,9 @@
 HEADERS += quiloader.h
 SOURCES += quiloader.cpp
 
+target.path=$$[QT_INSTALL_LIBS]
+INSTALLS        += target
+
 include($$QT_SOURCE_TREE/include/QtUiTools/headers.pri)
 quitools_headers.files = $$SYNCQT.HEADER_FILES $$SYNCQT.HEADER_CLASSES
 quitools_headers.path = $$[QT_INSTALL_HEADERS]/QtUiTools
