--- makefile.orig	Sun Sep 25 03:26:37 1994
+++ makefile	Sun Sep 25 03:26:45 1994
@@ -235,7 +235,7 @@
 parse.o: zoofns.h zooio.h
 portable.o: /usr/include/stdio.h assert.h debug.h machine.h options.h
 portable.o: portable.h various.h zoo.h zoofns.h zooio.h
-prterror.o: /usr/include/stdio.h /usr/include/varargs.h options.h various.h
+prterror.o: /usr/include/stdio.h /usr/include/unistd.h options.h various.h
 prterror.o: zoofns.h zooio.h
 sysv.o: /usr/include/sys/stat.h /usr/include/sys/types.h /usr/include/time.h
 sysv.o: nixmode.i nixtime.i
--- makefile.orig	Sat Dec 31 00:23:57 1994
+++ makefile	Sat Dec 31 00:24:31 1994
@@ -23,7 +23,7 @@
 MODEL =
 EXTRA = -DBIG_MEM -DNDEBUG
 LINTFLAGS = -DLINT
-OPTIM = -O
+OPTIM = -O2
 DESTDIR = /usr/local/bin
 
 #List of all object files created for Zoo
@@ -88,7 +88,7 @@
 # A generic system -- may have less than full functionality.
 # Compile with -g, since debugging will probably be needed.
 generic:
-	$(MAKE) CFLAGS="-c -g -DGENERIC" $(TARGETS)
+	$(MAKE) CFLAGS="-c -O2 -DGENERIC" $(TARGETS)
 
 # Reasonably generic BSD 4.3
 bsd:
--- various.h.orig	Sat Dec 31 00:22:03 1994
+++ various.h	Sat Dec 31 00:22:45 1994
@@ -54,8 +54,8 @@
 int fwrite PARMS ((VOIDPTR, int, int, FILE *));
 # else
 int fputc PARMS ((char, FILE *));
-int fread PARMS ((VOIDPTR, unsigned, unsigned, FILE *));
-int fwrite PARMS ((VOIDPTR, unsigned, unsigned, FILE *));
+/* int fread PARMS ((VOIDPTR, unsigned, unsigned, FILE *));
+int fwrite PARMS ((VOIDPTR, unsigned, unsigned, FILE *)); */
 # endif /* ALWAYS_INT */
 #endif /* NO_STDIO_FN */
 
