diff -u darcs-2.5/Distribution/ShellHarness.hs\~ darcs-2.5/Distribution/ShellHarness.hs
--- darcs-2.5/Distribution/ShellHarness.hs~	2010-10-25 01:29:26.000000000 +1000
+++ darcs-2.5/Distribution/ShellHarness.hs	2010-11-26 13:24:37.000000000 +1000
@@ -2,8 +2,10 @@
 module Distribution.ShellHarness ( runTests ) where
 
 import Prelude hiding( catch )
-import System.Directory ( getCurrentDirectory, setPermissions,
-                          Permissions(..), getDirectoryContents,
+import System.Directory ( getCurrentDirectory,
+                          getPermissions, setPermissions,
+                          readable, writable, executable, searchable,
+                          getDirectoryContents,
                           findExecutable, createDirectoryIfMissing,
                           renameFile, removeFile )
 import System.Environment ( getEnv, getEnvironment )
@@ -110,8 +112,9 @@
                  system $ "hpc sum --union --output=" ++ tixdir </> "sum.tix" ++ " " ++ unwords tixfiles
                  forM tixfiles $ \f -> removeFile f
                  return ()
-             mapM_ (\x->
-                  setPermissions x (Permissions
+             mapM_ (\x-> do
+                  p <- getPermissions x
+                  setPermissions x (p
                                    {readable = True
                                    ,writable = True
                                    ,executable = False
