diff -ur a/lib/Makefile.am b/lib/Makefile.am
--- a/lib/Makefile.am	2012-11-09 17:12:36.000000000 -0600
+++ b/lib/Makefile.am	2012-12-11 18:59:42.764918561 -0600
@@ -41,13 +41,13 @@
 # Hack to avoid using ylwrap. It does not function correctly in combination
 # with --header-file=
 route/pktloc_grammar.c: route/pktloc_grammar.l
-	$(AM_V_GEN) $(LEX) --header-file=route/pktloc_grammar.h $(LFLAGS) -o $@ $^
+	$(AM_V_GEN) $(FLEX) --header-file=route/pktloc_grammar.h $(LFLAGS) -o $@ $^
 
 route/pktloc_syntax.c: route/pktloc_syntax.y
 	$(AM_V_GEN) $(YACC) -d $(YFLAGS) -o $@ $^
 
 route/cls/ematch_grammar.c: route/cls/ematch_grammar.l
-	$(AM_V_GEN) $(LEX) --header-file=route/cls/ematch_grammar.h $(LFLAGS) -o $@ $^
+	$(AM_V_GEN) $(FLEX) --header-file=route/cls/ematch_grammar.h $(LFLAGS) -o $@ $^
 
 route/cls/ematch_syntax.c: route/cls/ematch_syntax.y
 	$(AM_V_GEN) $(YACC) -d $(YFLAGS) -o $@ $^
diff -ur a/lib/Makefile.in b/lib/Makefile.in
--- a/lib/Makefile.in	2012-11-26 10:41:35.000000000 -0600
+++ b/lib/Makefile.in	2012-12-11 18:59:45.505919693 -0600
@@ -1344,13 +1344,13 @@
 # Hack to avoid using ylwrap. It does not function correctly in combination
 # with --header-file=
 route/pktloc_grammar.c: route/pktloc_grammar.l
-	$(AM_V_GEN) $(LEX) --header-file=route/pktloc_grammar.h $(LFLAGS) -o $@ $^
+	$(AM_V_GEN) $(FLEX) --header-file=route/pktloc_grammar.h $(LFLAGS) -o $@ $^
 
 route/pktloc_syntax.c: route/pktloc_syntax.y
 	$(AM_V_GEN) $(YACC) -d $(YFLAGS) -o $@ $^
 
 route/cls/ematch_grammar.c: route/cls/ematch_grammar.l
-	$(AM_V_GEN) $(LEX) --header-file=route/cls/ematch_grammar.h $(LFLAGS) -o $@ $^
+	$(AM_V_GEN) $(FLEX) --header-file=route/cls/ematch_grammar.h $(LFLAGS) -o $@ $^
 
 route/cls/ematch_syntax.c: route/cls/ematch_syntax.y
 	$(AM_V_GEN) $(YACC) -d $(YFLAGS) -o $@ $^
diff -ur a/src/lib/Makefile.am b/src/lib/Makefile.am
--- a/src/lib/Makefile.am	2012-10-29 12:16:52.000000000 -0500
+++ b/src/lib/Makefile.am	2012-12-11 18:59:50.185921628 -0600
@@ -11,13 +11,13 @@
 #cls_ematch_cmp_la_LDFLAGS = -module -version-info 2:0:0
 
 #cls/ematch_grammar.c: cls/ematch_grammar.l
-#	$(LEX) --header-file=cls/ematch_grammar.h $(LFLAGS) -o $@ $^
+#	$(FLEX) --header-file=cls/ematch_grammar.h $(LFLAGS) -o $@ $^
 
 #cls/ematch_syntax.c: cls/ematch_syntax.y
 #	$(YACC) -d $(YFLAGS) -o $@ $^
 
 #cls/pktloc_grammar.c: cls/pktloc_grammar.l
-#	$(LEX) --header-file=cls/pktloc_grammar.h $(LFLAGS) -o $@ $^
+#	$(FLEX) --header-file=cls/pktloc_grammar.h $(LFLAGS) -o $@ $^
 
 #cls/pktloc_syntax.c: cls/pktloc_syntax.y
 #	$(YACC) -d $(YFLAGS) -o $@ $^
diff -ur a/src/lib/Makefile.in b/src/lib/Makefile.in
--- a/src/lib/Makefile.in	2012-11-26 10:41:36.000000000 -0600
+++ b/src/lib/Makefile.in	2012-12-11 18:59:48.974921100 -0600
@@ -278,13 +278,13 @@
 #cls_ematch_cmp_la_LDFLAGS = -module -version-info 2:0:0
 
 #cls/ematch_grammar.c: cls/ematch_grammar.l
-#	$(LEX) --header-file=cls/ematch_grammar.h $(LFLAGS) -o $@ $^
+#	$(FLEX) --header-file=cls/ematch_grammar.h $(LFLAGS) -o $@ $^
 
 #cls/ematch_syntax.c: cls/ematch_syntax.y
 #	$(YACC) -d $(YFLAGS) -o $@ $^
 
 #cls/pktloc_grammar.c: cls/pktloc_grammar.l
-#	$(LEX) --header-file=cls/pktloc_grammar.h $(LFLAGS) -o $@ $^
+#	$(FLEX) --header-file=cls/pktloc_grammar.h $(LFLAGS) -o $@ $^
 
 #cls/pktloc_syntax.c: cls/pktloc_syntax.y
 #	$(YACC) -d $(YFLAGS) -o $@ $^
