diff -ru PokerTH-0.8.3-src.old/pokerth_game.pro PokerTH-0.8.3-src/pokerth_game.pro
--- PokerTH-0.8.3-src.old/pokerth_game.pro	2011-10-21 21:49:33.467356776 +0200
+++ PokerTH-0.8.3-src/pokerth_game.pro	2011-10-21 21:50:37.576355341 +0200
@@ -372,7 +372,6 @@
 	RC_FILE = pokerth.rc
 }
 unix:# workaround for problems with boost_filesystem exceptions
-QMAKE_LFLAGS += -no_dead_strip_inits_and_terms
 unix:!mac { 
 	# #### My release static build options
 	# QMAKE_CXXFLAGS += -ffunction-sections -fdata-sections
diff -ru PokerTH-0.8.3-src.old/pokerth_server.pro PokerTH-0.8.3-src/pokerth_server.pro
--- PokerTH-0.8.3-src.old/pokerth_server.pro	2011-10-21 21:49:33.468356776 +0200
+++ PokerTH-0.8.3-src/pokerth_server.pro	2011-10-21 21:50:57.315354851 +0200
@@ -185,7 +185,6 @@
 
 unix {
 	# workaround for problems with boost_filesystem exceptions
-	QMAKE_LFLAGS += -no_dead_strip_inits_and_terms
 }
 
 unix : !mac {
diff -ru PokerTH-0.8.3-src.old/zlib_compress.pro PokerTH-0.8.3-src/zlib_compress.pro
--- PokerTH-0.8.3-src.old/zlib_compress.pro	2011-10-21 21:49:33.467356776 +0200
+++ PokerTH-0.8.3-src/zlib_compress.pro	2011-10-21 21:50:14.891355825 +0200
@@ -43,7 +43,6 @@
 
 unix {
 	# workaround for problems with boost_filesystem exceptions
-	QMAKE_LFLAGS += -no_dead_strip_inits_and_terms
 }
 
 unix : !mac {
