diff -up PokerTH-0.8.3-src/pokerth_game.pro.orig PokerTH-0.8.3-src/pokerth_game.pro
--- PokerTH-0.8.3-src/pokerth_game.pro.orig	2011-01-16 21:37:12.000000000 +0200
+++ PokerTH-0.8.3-src/pokerth_game.pro	2011-02-21 21:57:13.348491747 +0200
@@ -19,7 +19,7 @@ QMAKE_CXXFLAGS += -std=gnu++0x
 UI_DIR = uics
 MOC_DIR = mocs
 OBJECTS_DIR = obj
-DEFINES += ENABLE_IPV6
+DEFINES += ENABLE_IPV6 BOOST_FILESYSTEM_VERSION=2
 DEFINES += PREFIX=\"$${PREFIX}\"
 TARGET = pokerth
 
diff -up PokerTH-0.8.3-src/pokerth_lib.pro.orig PokerTH-0.8.3-src/pokerth_lib.pro
--- PokerTH-0.8.3-src/pokerth_lib.pro.orig	2011-01-16 21:37:12.000000000 +0200
+++ PokerTH-0.8.3-src/pokerth_lib.pro	2011-02-21 21:57:13.350491836 +0200
@@ -12,7 +12,7 @@ UI_DIR = uics
 TARGET = lib/pokerth_lib
 MOC_DIR = mocs
 OBJECTS_DIR = obj
-DEFINES += ENABLE_IPV6
+DEFINES += ENABLE_IPV6 BOOST_FILESYSTEM_VERSION=2
 QT -= core gui
 QMAKE_CXXFLAGS += -std=gnu++0x
 #PRECOMPILED_HEADER = src/pch_lib.h
diff -up PokerTH-0.8.3-src/pokerth_server.pro.orig PokerTH-0.8.3-src/pokerth_server.pro
--- PokerTH-0.8.3-src/pokerth_server.pro.orig	2011-01-16 21:37:12.000000000 +0200
+++ PokerTH-0.8.3-src/pokerth_server.pro	2011-02-21 21:57:13.353491970 +0200
@@ -14,7 +14,7 @@ TARGET = bin/pokerth_server
 MOC_DIR = mocs
 OBJECTS_DIR = obj
 DEFINES += POKERTH_DEDICATED_SERVER
-DEFINES += ENABLE_IPV6
+DEFINES += ENABLE_IPV6 BOOST_FILESYSTEM_VERSION=2
 DEFINES += PREFIX=\"$${PREFIX}\"
 QT -= core gui
 QMAKE_CXXFLAGS += -std=gnu++0x
