diff -Naur kerry/src/configdialog.cpp kerry-p/src/configdialog.cpp
--- kerry/src/configdialog.cpp	2006-02-21 16:59:53.000000000 +0100
+++ kerry-p/src/configdialog.cpp	2006-03-11 02:43:24.000000000 +0100
@@ -322,7 +322,7 @@
 {
    QString xdgdata = getenv("XDG_DATA_DIRS");
    if (xdgdata.isEmpty())
-     xdgdata = "/usr/local/share/:/usr/share/";
+     xdgdata = "/usr/share/";
 
    QStringList xdgappsdata = QStringList::split(':', xdgdata+":"+QDir::home().absPath()+"/.kde/share/");
 
diff -Naur kerry/src/hitwidget_layout.ui kerry-p/src/hitwidget_layout.ui
--- kerry/src/hitwidget_layout.ui	2006-03-07 17:11:36.000000000 +0100
+++ kerry-p/src/hitwidget_layout.ui	2006-03-11 02:32:33.000000000 +0100
@@ -80,7 +80,7 @@
                     </property>
                     <property name="font">
                         <font>
-                            <pointsize>8</pointsize>
+                            <pointsize>11</pointsize>
                         </font>
                     </property>
                     <property name="text">
@@ -229,7 +229,7 @@
                                     </property>
                                     <property name="font">
                                         <font>
-                                            <pointsize>8</pointsize>
+                                            <pointsize>11</pointsize>
                                         </font>
                                     </property>
                                     <property name="text">
@@ -253,7 +253,7 @@
                                     </property>
                                     <property name="font">
                                         <font>
-                                            <pointsize>8</pointsize>
+                                            <pointsize>11</pointsize>
                                         </font>
                                     </property>
                                     <property name="text">
