--- winki-0.3.6/winki/install.orig	2005-03-14 11:37:39.000000000 +0100
+++ winki-0.3.6/winki/install	2005-03-26 12:41:17.000000000 +0100
@@ -2,9 +2,9 @@
 
 cd .. && ./setup.py  install --force --record installed.files
 
-mkdir -p /opt/winki/i18n/bg_BG/LC_MESSAGES
-msgfmt winki/po/bg.po -o /opt/winki/i18n/bg_BG/LC_MESSAGES/winki.mo
+mkdir -p /usr/share/winki/i18n/bg_BG/LC_MESSAGES
+msgfmt winki/po/bg.po -o /usr/share/winki/i18n/bg_BG/LC_MESSAGES/winki.mo
 
-mkdir -p /opt/winki/i18n/de_DE/LC_MESSAGES
-msgfmt winki/po/de.po -o /opt/winki/i18n/de_DE/LC_MESSAGES/winki.mo
+mkdir -p /usr/share/winki/i18n/de_DE/LC_MESSAGES
+msgfmt winki/po/de.po -o /usr/share/winki/i18n/de_DE/LC_MESSAGES/winki.mo
 
--- winki-0.3.6/winki/lib.py.orig	2005-03-14 11:37:39.000000000 +0100
+++ winki-0.3.6/winki/lib.py	2005-03-26 12:40:56.000000000 +0100
@@ -44,7 +44,7 @@
 
 # program installation path (winki_path is set here _and_ in ../setup.py)
 # (sorry for the inconvenience chris, hth)
-winki_path = os.path.join('/', 'opt', 'winki')
+winki_path = os.path.join('/', 'usr', 'share', 'winki')
 
 # enumerations for the priority of the debug messages
 info = 1
--- winki-0.3.6/setup.py.orig	2005-03-26 12:51:05.000000000 +0100
+++ winki-0.3.6/setup.py	2005-03-26 12:50:37.000000000 +0100
@@ -22,7 +22,7 @@
 src_todo = os.path.join('winki', 'TODO')
 
 # winki_path is set here _and_ in lib.py
-winki_path = os.path.join('/', 'opt', 'winki')
+winki_path = os.path.join('/', 'usr', 'share', 'winki')
 dest_glade = os.path.join(winki_path, 'glade')
 dest_doc = os.path.join(winki_path, 'doc')
 dest_etc = os.path.join('/', 'etc')
