diff -Naur OneTuple-0.3.1/OneTuple.cabal OneTuple-0.3.1.new/OneTuple.cabal
--- OneTuple-0.3.1/OneTuple.cabal	2001-09-09 03:46:40.000000000 +0200
+++ OneTuple-0.3.1.new/OneTuple.cabal	2022-10-11 16:16:40.579570830 +0200
@@ -1,6 +1,7 @@
 cabal-version:      >=1.10
 name:               OneTuple
 version:            0.3.1
+x-revision:         3
 synopsis:           Singleton Tuple
 category:           Data
 description:
@@ -32,8 +33,9 @@
    || ==8.6.5
    || ==8.8.4
    || ==8.10.4
-   || ==9.0.1
-   || ==9.2.1
+   || ==9.0.2
+   || ==9.2.4
+   || ==9.4.1
 
 extra-source-files: Changelog.md
 
@@ -50,7 +52,7 @@
 
   hs-source-dirs:   src
   build-depends:
-      base              >=4.5 && <4.17
+      base              >=4.5 && <4.18
     , template-haskell
 
   if impl(ghc >=9.0)
@@ -61,7 +63,7 @@
 
   if !impl(ghc >=8.0)
     build-depends:
-        semigroups    >=0.18.4 && <0.20
+        semigroups    >=0.18.4 && <0.21
       , transformers  >=0.3    && <0.7
 
     -- Ensure Data.Functor.Classes is always available
@@ -69,7 +71,7 @@
       build-depends: transformers >=0.4.2.0
 
     else
-      build-depends: transformers-compat >=0.5.1.0 && <0.7
+      build-depends: transformers-compat >=0.5.1.0 && <0.8
 
   if !impl(ghc >=9.2)
     build-depends: base-orphans >=0.8.6
