commit 3f51aec8647fe13f4b1e46b2f75ff635403adf91
Author: Lennart Poettering <lennart@poettering.net>
Date:   Tue Feb 16 13:18:36 2016 +0100

    core: fix assertion check
    
    Fixes: #2632

diff --git a/src/core/timer.c b/src/core/timer.c
index 6f3e6a8..3d0bae1 100644
--- a/src/core/timer.c
+++ b/src/core/timer.c
@@ -334,7 +334,7 @@ static void add_random(Timer *t, usec_t *v) {
         usec_t add;
 
         assert(t);
-        assert(*v);
+        assert(v);
 
         if (t->random_usec == 0)
                 return;
