diff -Naur systemd-231/src/basic/rm-rf.c systemd-231-p/src/basic/rm-rf.c
--- systemd-231/src/basic/rm-rf.c	2016-07-25 21:49:47.000000000 +0200
+++ systemd-231-p/src/basic/rm-rf.c	2017-04-17 12:52:55.495157108 +0200
@@ -188,6 +188,13 @@
                 return -EPERM;
         }
 
+        /* Another safe-check. Removing "/path/.." could easily remove entire root as well.
+         * It's especially easy to do using globs in tmpfiles, like "/path/.*", which the glob()
+         * function expands to both "/path/." and "/path/..".
+         * Return -EINVAL to be consistent with rmdir("/path/."). */
+        if (endswith(path, "/..") || endswith(path, "/../"))
+                return -EINVAL;
+
         if ((flags & (REMOVE_SUBVOLUME|REMOVE_ROOT|REMOVE_PHYSICAL)) == (REMOVE_SUBVOLUME|REMOVE_ROOT|REMOVE_PHYSICAL)) {
                 /* Try to remove as subvolume first */
                 r = btrfs_subvol_remove(path, BTRFS_REMOVE_RECURSIVE|BTRFS_REMOVE_QUOTA);
