--- systemd-256.4/sysusers.d/basic.conf.in	2024-07-24 14:49:39.000000000 +0200
+++ systemd-256.4.new/sysusers.d/basic.conf.in	2024-08-14 16:25:45.796939896 +0200
@@ -29,7 +29,6 @@
 g disk    {{DISK_GID   }}     -            -
 g input   {{INPUT_GID  }}     -            -
 g kmem    {{KMEM_GID   }}     -            -
-g kvm     {{KVM_GID    }}     -            -
 g lp      {{LP_GID     }}     -            -
 g render  {{RENDER_GID }}     -            -
 g sgx     {{SGX_GID    }}     -            -
