--- systemd-24/configure.ac.old	2011-04-06 21:27:48.000000000 +0200
+++ systemd-24/configure.ac	2011-04-18 21:20:26.000000000 +0200
@@ -347,6 +347,7 @@
                 SYSTEM_SYSVINIT_PATH=/etc/rc.d
                 AC_DEFINE(TARGET_FRUGALWARE, [], [Target is Frugalware])
                 M4_DISTRO_FLAG=-DTARGET_FRUGALWARE=1
+		have_plymouth=true
                 ;;
         altlinux)
                 SYSTEM_SYSVINIT_PATH=/etc/rc.d/init.d

--- systemd-25/units/frugalware/display-manager.service.old	2011-03-16 03:53:55.000000000 +0100
+++ systemd-25/units/frugalware/display-manager.service	2011-04-23 17:53:17.000000000 +0200
@@ -9,8 +9,8 @@
 Description=Display Manager
 After=syslog.target local.service systemd-user-sessions.service
 
-Conflicts=splashy-quit.service
-After=splashy-quit.service
+Conflicts=plymouth-quit.service
+After=plymouth-quit.service
 
 [Service]
 EnvironmentFile=/etc/sysconfig/desktop
