diff -Naur systemd-231/configure.ac systemd-231-p/configure.ac
--- systemd-231/configure.ac	2016-07-25 21:49:47.000000000 +0200
+++ systemd-231-p/configure.ac	2018-01-27 16:01:54.938669544 +0100
@@ -310,6 +310,7 @@
         LO_FLAGS_PARTSCAN,
         copy_file_range],
         [], [], [[
+#include <sys/mman.h>
 #include <sys/types.h>
 #include <unistd.h>
 #include <sys/mount.h>
diff -Naur systemd-231/src/basic/fileio.c systemd-231-p/src/basic/fileio.c
--- systemd-231/src/basic/fileio.c	2016-07-25 21:49:47.000000000 +0200
+++ systemd-231-p/src/basic/fileio.c	2018-01-27 15:51:25.863340242 +0100
@@ -24,6 +24,7 @@
 #include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/mman.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <unistd.h>
diff -Naur systemd-231/src/basic/parse-util.c systemd-231-p/src/basic/parse-util.c
--- systemd-231/src/basic/parse-util.c	2016-07-25 21:49:47.000000000 +0200
+++ systemd-231-p/src/basic/parse-util.c	2018-01-27 16:15:25.092072973 +0100
@@ -23,7 +23,6 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <xlocale.h>
 
 #include "alloc-util.h"
 #include "extract-word.h"
