--- systemd-221/units/getty@.service.m4	2015-04-23 16:56:35.000000000 +0200
+++ systemd-221.new/units/getty@.service.m4	2015-08-13 12:01:01.000000000 +0200
@@ -27,7 +27,7 @@
 
 [Service]
 # the VT is cleared by TTYVTDisallocate
-ExecStart=-/sbin/agetty --noclear %I $TERM
+ExecStart=-/sbin/frugalwaregetty --noclear %I $TERM
 Type=idle
 Restart=always
 RestartSec=0
@@ -42,7 +42,7 @@
 
 # Unset locale for the console getty since the console has problems
 # displaying some internationalized messages.
-Environment=LANG= LANGUAGE= LC_CTYPE= LC_NUMERIC= LC_TIME= LC_COLLATE= LC_MONETARY= LC_MESSAGES= LC_PAPER= LC_NAME= LC_ADDRESS= LC_TELEPHONE= LC_MEASUREMENT= LC_IDENTIFICATION=
+#Environment=LANG= LANGUAGE= LC_CTYPE= LC_NUMERIC= LC_TIME= LC_COLLATE= LC_MONETARY= LC_MESSAGES= LC_PAPER= LC_NAME= LC_ADDRESS= LC_TELEPHONE= LC_MEASUREMENT= LC_IDENTIFICATION=
 
 [Install]
 WantedBy=getty.target
