diff -Naur systemd-240/src/tmpfiles/tmpfiles.c systemd-240-p/src/tmpfiles/tmpfiles.c
--- systemd-240/src/tmpfiles/tmpfiles.c	2018-12-21 19:53:33.000000000 +0100
+++ systemd-240-p/src/tmpfiles/tmpfiles.c	2019-01-22 00:53:46.170063236 +0100
@@ -2969,9 +2969,7 @@
                 }
 
         if (arg_operation == 0 && !arg_cat_config)
-                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                       "You need to specify at least one of --clean, --create or --remove.");
-
+                return help();
         if (arg_replace && arg_cat_config)
                 return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
                                        "Option --replace= is not supported with --cat-config");
@@ -3169,6 +3167,8 @@
         } phase;
         int r, k;
 
+
+
         r = parse_argv(argc, argv);
         if (r <= 0)
                 return r;
@@ -3180,6 +3180,9 @@
                 if (r < 0)
                         return log_error_errno(r, "Failed to initialize configuration directory list: %m");
         } else {
+		if (geteuid() != 0) {
+			return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Oups! Root privileges needed to run this command");
+		}
                 config_dirs = strv_split_nulstr(CONF_PATHS_NULSTR("tmpfiles.d"));
                 if (!config_dirs)
                         return log_oom();
