diff -Naur systemd-231/sysusers.d/basic.conf.in systemd-231.new/sysusers.d/basic.conf.in
--- systemd-231/sysusers.d/basic.conf.in	2016-07-25 21:49:47.000000000 +0200
+++ systemd-231.new/sysusers.d/basic.conf.in	2016-10-25 13:42:46.000000000 +0200
@@ -25,11 +25,11 @@
 # Hardware access groups
 g audio   -     -            -
 g cdrom   -     -            -
-g dialout -     -            -
+g uucp -     -            -
 g disk    -     -            -
 g input   -     -            -
 g lp      -     -            -
-g tape    -     -            -
+g storage    -     -            -
 g video   -     -            -
 
 # Default group for normal users
diff -Naur systemd-231/rules/50-udev-default.rules systemd-231.new/rules/50-udev-default.rules
--- systemd-231/rules/50-udev-default.rules	2016-07-25 21:49:47.000000000 +0200
+++ systemd-231.new/rules/50-udev-default.rules	2016-10-25 13:43:07.000000000 +0200
@@ -23,7 +23,7 @@
 SUBSYSTEM=="tty", KERNEL=="ttysclp[0-9]*", GROUP="tty", MODE="0620"
 SUBSYSTEM=="tty", KERNEL=="3270/tty[0-9]*", GROUP="tty", MODE="0620"
 SUBSYSTEM=="vc", KERNEL=="vcs*|vcsa*", GROUP="tty"
-KERNEL=="tty[A-Z]*[0-9]|pppox[0-9]*|ircomm[0-9]*|noz[0-9]*|rfcomm[0-9]*", GROUP="dialout"
+KERNEL=="tty[A-Z]*[0-9]|pppox[0-9]*|ircomm[0-9]*|noz[0-9]*|rfcomm[0-9]*", GROUP="uucp"
 
 SUBSYSTEM=="mem", KERNEL=="mem|kmem|port", GROUP="kmem", MODE="0640"
 
@@ -59,7 +59,7 @@
 KERNEL=="pktcdvd[0-9]*", GROUP="cdrom"
 KERNEL=="pktcdvd", GROUP="cdrom"
 
-SUBSYSTEM=="scsi_generic|scsi_tape", SUBSYSTEMS=="scsi", ATTRS{type}=="1|8", GROUP="tape"
+SUBSYSTEM=="scsi_generic|scsi_tape", SUBSYSTEMS=="scsi", ATTRS{type}=="1|8", GROUP="storage"
 SUBSYSTEM=="scsi_generic", SUBSYSTEMS=="scsi", ATTRS{type}=="0", GROUP="disk"
 KERNEL=="qft[0-9]*|nqft[0-9]*|zqft[0-9]*|nzqft[0-9]*|rawqft[0-9]*|nrawqft[0-9]*", GROUP="disk"
 KERNEL=="loop-control", GROUP="disk", OPTIONS+="static_node=loop-control"
