diff -ur a/src/locale/localectl.c b/src/locale/localectl.c
--- a/src/locale/localectl.c	2013-01-07 15:50:49.073315546 -0600
+++ b/src/locale/localectl.c	2013-03-14 19:33:08.138800391 -0500
@@ -548,6 +548,7 @@
         nftw("/usr/share/kbd/keymaps/", nftw_cb, 20, FTW_MOUNT|FTW_PHYS);
         nftw("/usr/lib/kbd/keymaps/", nftw_cb, 20, FTW_MOUNT|FTW_PHYS);
         nftw("/lib/kbd/keymaps/", nftw_cb, 20, FTW_MOUNT|FTW_PHYS);
+        nftw("/usr/share/keymaps/", nftw_cb, 20, FTW_MOUNT|FTW_PHYS);
 
         l = set_get_strv(keymaps);
         if (!l) {
diff -ur a/src/login/logind.conf b/src/login/logind.conf
--- a/src/login/logind.conf	2012-12-23 17:29:39.965278291 -0600
+++ b/src/login/logind.conf	2013-03-14 19:33:31.778799521 -0500
@@ -8,7 +8,7 @@
 # See logind.conf(5) for details
 
 [Login]
-#NAutoVTs=6
+NAutoVTs=11
 #ReserveVT=6
 #KillUserProcesses=no
 #KillOnlyUsers=
diff -ur a/units/getty@.service.m4 b/units/getty@.service.m4
--- a/units/getty@.service.m4	2013-01-07 15:50:49.083315575 -0600
+++ b/units/getty@.service.m4	2013-03-14 19:33:08.142800391 -0500
@@ -27,7 +27,7 @@
 
 [Service]
 # the VT is cleared by TTYVTDisallocate
-ExecStart=-/sbin/agetty --noclear %I 38400 linux
+ExecStart=-/sbin/frugalwaregetty --noclear %I 38400 linux
 Type=idle
 Restart=always
 RestartSec=0
@@ -41,7 +41,7 @@
 
 # Unset locale for the console getty since the console has problems
 # displaying some internationalized messages.
-Environment=LANG= LANGUAGE= LC_CTYPE= LC_NUMERIC= LC_TIME= LC_COLLATE= LC_MONETARY= LC_MESSAGES= LC_PAPER= LC_NAME= LC_ADDRESS= LC_TELEPHONE= LC_MEASUREMENT= LC_IDENTIFICATION=
+#Environment=LANG= LANGUAGE= LC_CTYPE= LC_NUMERIC= LC_TIME= LC_COLLATE= LC_MONETARY= LC_MESSAGES= LC_PAPER= LC_NAME= LC_ADDRESS= LC_TELEPHONE= LC_MEASUREMENT= LC_IDENTIFICATION=
 
 # Some login implementations ignore SIGTERM, so we send SIGHUP
 # instead, to ensure that login terminates cleanly.
diff -ur a/units/systemd-update-utmp-runlevel.service.in b/units/systemd-update-utmp-runlevel.service.in
--- a/units/systemd-update-utmp-runlevel.service.in	2012-06-25 15:05:25.772518023 -0500
+++ b/units/systemd-update-utmp-runlevel.service.in	2013-03-14 19:33:08.146800391 -0500
@@ -10,7 +10,7 @@
 Documentation=man:systemd-update-utmp-runlevel.service(8) man:utmp(5)
 DefaultDependencies=no
 RequiresMountsFor=/var/log/wtmp
-After=systemd-remount-fs.service systemd-tmpfiles-setup.service auditd.service
+After=systemd-remount-fs.service systemd-tmpfiles-setup.service
 After=runlevel1.target runlevel2.target runlevel3.target runlevel4.target runlevel5.target
 Before=final.target
 
diff -ur a/units/systemd-update-utmp-shutdown.service.in b/units/systemd-update-utmp-shutdown.service.in
--- a/units/systemd-update-utmp-shutdown.service.in	2012-06-25 15:05:25.759517987 -0500
+++ b/units/systemd-update-utmp-shutdown.service.in	2013-03-14 19:33:08.150800390 -0500
@@ -10,7 +10,7 @@
 Documentation=man:systemd-update-utmp-runlevel.service(8) man:utmp(5)
 DefaultDependencies=no
 RequiresMountsFor=/var/log/wtmp
-After=systemd-remount-fs.service systemd-tmpfiles-setup.service auditd.service
+After=systemd-remount-fs.service systemd-tmpfiles-setup.service
 After=systemd-update-utmp-runlevel.service
 Before=final.target
 
diff -ur a/units/systemd-vconsole-setup.service.in b/units/systemd-vconsole-setup.service.in
--- a/units/systemd-vconsole-setup.service.in	2012-06-25 04:46:14.862077173 -0500
+++ b/units/systemd-vconsole-setup.service.in	2013-03-14 19:33:08.151800390 -0500
@@ -10,7 +10,8 @@
 Documentation=man:systemd-vconsole-setup.service(8) man:vconsole.conf(5)
 DefaultDependencies=no
 Conflicts=shutdown.target
-After=systemd-readahead-collect.service systemd-readahead-replay.service
+After=systemd-readahead-collect.service systemd-readahead-replay.service systemd-udev-settle.service
+Wants=systemd-udev-settle.service
 Before=sysinit.target shutdown.target
 ConditionPathExists=/dev/tty0
 
