diff -ur a/units/getty@.service.m4 b/units/getty@.service.m4
--- a/units/getty@.service.m4	2012-12-15 13:49:15.497710912 -0600
+++ b/units/getty@.service.m4	2012-12-15 13:56:35.703215524 -0600
@@ -45,7 +45,7 @@
 
 [Service]
 # the VT is cleared by TTYVTDisallocate
-ExecStart=-/sbin/agetty --noclear %I 38400 linux
+ExecStart=-/sbin/frugalwaregetty --noclear %I 38400 linux
 Type=idle
 Restart=always
 RestartSec=0
@@ -59,7 +59,7 @@
 
 # Unset locale for the console getty since the console has problems
 # displaying some internationalized messages.
-Environment=LANG= LANGUAGE= LC_CTYPE= LC_NUMERIC= LC_TIME= LC_COLLATE= LC_MONETARY= LC_MESSAGES= LC_PAPER= LC_NAME= LC_ADDRESS= LC_TELEPHONE= LC_MEASUREMENT= LC_IDENTIFICATION=
+#Environment=LANG= LANGUAGE= LC_CTYPE= LC_NUMERIC= LC_TIME= LC_COLLATE= LC_MONETARY= LC_MESSAGES= LC_PAPER= LC_NAME= LC_ADDRESS= LC_TELEPHONE= LC_MEASUREMENT= LC_IDENTIFICATION=
 
 # Some login implementations ignore SIGTERM, so we send SIGHUP
 # instead, to ensure that login terminates cleanly.
diff -ur a/units/systemd-update-utmp-runlevel.service.in b/units/systemd-update-utmp-runlevel.service.in
--- a/units/systemd-update-utmp-runlevel.service.in	2012-12-15 13:49:15.495710909 -0600
+++ b/units/systemd-update-utmp-runlevel.service.in	2012-12-15 13:57:48.935218317 -0600
@@ -10,7 +10,7 @@
 Documentation=man:systemd-update-utmp-runlevel.service(8) man:utmp(5)
 DefaultDependencies=no
 RequiresMountsFor=/var/log/wtmp
-After=systemd-remount-fs.service systemd-tmpfiles-setup.service auditd.service
+After=systemd-remount-fs.service systemd-tmpfiles-setup.service
 After=runlevel1.target runlevel2.target runlevel3.target runlevel4.target runlevel5.target
 Before=final.target
 
diff -ur a/units/systemd-update-utmp-shutdown.service.in b/units/systemd-update-utmp-shutdown.service.in
--- a/units/systemd-update-utmp-shutdown.service.in	2012-12-15 13:49:15.496710910 -0600
+++ b/units/systemd-update-utmp-shutdown.service.in	2012-12-15 13:58:06.855216211 -0600
@@ -10,7 +10,7 @@
 Documentation=man:systemd-update-utmp-runlevel.service(8) man:utmp(5)
 DefaultDependencies=no
 RequiresMountsFor=/var/log/wtmp
-After=systemd-remount-fs.service systemd-tmpfiles-setup.service auditd.service
+After=systemd-remount-fs.service systemd-tmpfiles-setup.service
 After=systemd-update-utmp-runlevel.service
 Before=final.target
 
diff -ur a/units/systemd-vconsole-setup.service.in b/units/systemd-vconsole-setup.service.in
--- a/units/systemd-vconsole-setup.service.in	2012-12-15 13:49:15.495710909 -0600
+++ b/units/systemd-vconsole-setup.service.in	2012-12-18 22:11:24.094598586 -0600
@@ -10,7 +10,8 @@
 Documentation=man:systemd-vconsole-setup.service(8) man:vconsole.conf(5)
 DefaultDependencies=no
 Conflicts=shutdown.target
-After=systemd-readahead-collect.service systemd-readahead-replay.service
+After=systemd-readahead-collect.service systemd-readahead-replay.service systemd-udev-settle.service
+Wants=systemd-udev-settle.service
 Before=sysinit.target shutdown.target
 ConditionPathExists=/dev/tty0
 
