From 4199a69aa5291d7a30032e3737a18964781141d3 Mon Sep 17 00:00:00 2001
From: crazy <crazy@frugalware.org>
Date: Tue, 15 Jan 2019 19:01:34 +0100
Subject: [PATCH 1/5] frugalware VT defaults

---
 src/login/logind.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/login/logind.conf.in b/src/login/logind.conf.in
index 1029e29bc7..13f4c46b1e 100644
--- a/src/login/logind.conf.in
+++ b/src/login/logind.conf.in
@@ -12,7 +12,7 @@
 # See logind.conf(5) for details.
 
 [Login]
-#NAutoVTs=6
+NAutoVTs=7
 #ReserveVT=6
 #KillUserProcesses=@KILL_USER_PROCESSES@
 #KillOnlyUsers=
-- 
2.20.1

