diff --git a/src/locale/localectl.c b/src/locale/localectl.c
index 8259c0a..02f9039 100644
--- a/src/locale/localectl.c
+++ b/src/locale/localectl.c
@@ -542,6 +542,7 @@ static int list_vconsole_keymaps(DBusConnection *bus, char **args, unsigned n) {
         nftw("/usr/share/kbd/keymaps/", nftw_cb, 20, FTW_MOUNT|FTW_PHYS);
         nftw("/usr/lib/kbd/keymaps/", nftw_cb, 20, FTW_MOUNT|FTW_PHYS);
         nftw("/lib/kbd/keymaps/", nftw_cb, 20, FTW_MOUNT|FTW_PHYS);
+        nftw("/usr/share/keymaps/", nftw_cb, 20, FTW_MOUNT|FTW_PHYS);
 
         l = set_get_strv(keymaps);
         if (!l) {
diff --git a/src/login/logind.conf b/src/login/logind.conf
index c0abf01..ab7aa64 100644
--- a/src/login/logind.conf
+++ b/src/login/logind.conf
@@ -8,7 +8,7 @@
 # See logind.conf(5) for details
 
 [Login]
-#NAutoVTs=6
+NAutoVTs=11
 #ReserveVT=6
 #KillUserProcesses=no
 #KillOnlyUsers=
diff --git a/units/getty@.service.m4 b/units/getty@.service.m4
index 253da85..51142f3 100644
--- a/units/getty@.service.m4
+++ b/units/getty@.service.m4
@@ -27,7 +27,7 @@ ConditionPathExists=/dev/tty0
 
 [Service]
 # the VT is cleared by TTYVTDisallocate
-ExecStart=-/sbin/agetty --noclear %I
+ExecStart=-/sbin/frugalwaregetty --noclear %I
 Type=idle
 Restart=always
 RestartSec=0
@@ -42,7 +42,7 @@ SendSIGHUP=yes
 
 # Unset locale for the console getty since the console has problems
 # displaying some internationalized messages.
-Environment=LANG= LANGUAGE= LC_CTYPE= LC_NUMERIC= LC_TIME= LC_COLLATE= LC_MONETARY= LC_MESSAGES= LC_PAPER= LC_NAME= LC_ADDRESS= LC_TELEPHONE= LC_MEASUREMENT= LC_IDENTIFICATION=
+#Environment=LANG= LANGUAGE= LC_CTYPE= LC_NUMERIC= LC_TIME= LC_COLLATE= LC_MONETARY= LC_MESSAGES= LC_PAPER= LC_NAME= LC_ADDRESS= LC_TELEPHONE= LC_MEASUREMENT= LC_IDENTIFICATION=
 
 [Install]
 WantedBy=getty.target
diff --git a/units/systemd-update-utmp.service.in b/units/systemd-update-utmp.service.in
index e7c20a5..07714cc 100644
--- a/units/systemd-update-utmp.service.in
+++ b/units/systemd-update-utmp.service.in
@@ -11,7 +11,7 @@ Documentation=man:systemd-update-utmp.service(8) man:utmp(5)
 DefaultDependencies=no
 RequiresMountsFor=/var/log/wtmp
 Conflicts=shutdown.target
-After=systemd-readahead-collect.service systemd-readahead-replay.service systemd-remount-fs.service systemd-tmpfiles-setup.service auditd.service
+After=systemd-readahead-collect.service systemd-readahead-replay.service systemd-remount-fs.service systemd-tmpfiles-setup.service
 Before=sysinit.target shutdown.target
 
 [Service]
diff --git a/units/systemd-vconsole-setup.service.in b/units/systemd-vconsole-setup.service.in
index 18faa63..8bacce7 100644
--- a/units/systemd-vconsole-setup.service.in
+++ b/units/systemd-vconsole-setup.service.in
@@ -10,7 +10,8 @@ Description=Setup Virtual Console
 Documentation=man:systemd-vconsole-setup.service(8) man:vconsole.conf(5)
 DefaultDependencies=no
 Conflicts=shutdown.target
-After=systemd-readahead-collect.service systemd-readahead-replay.service
+After=systemd-readahead-collect.service systemd-readahead-replay.service systemd-udev-settle.service
+Wants=systemd-udev-settle.service
 Before=sysinit.target shutdown.target
 ConditionPathExists=/dev/tty0
 
