From 6279e1f01442fe01fe036036081d1935e3287d52 Mon Sep 17 00:00:00 2001
From: Miklos Vajna <vmiklos@frugalware.org>
Date: Thu, 16 Dec 2010 14:00:51 +0000
Subject: Frugalware: start gettys after local.service

---
diff --git a/units/getty@.service.m4 b/units/getty@.service.m4
index 902ab6e..0b3160e 100644
--- a/units/getty@.service.m4
+++ b/units/getty@.service.m4
@@ -15,6 +15,9 @@ After=rc-local.service
 m4_ifdef(`TARGET_ARCH',
 After=rc-local.service
 )m4_dnl
+m4_ifdef(`TARGET_FRUGALWARE',
+After=local.service
+)m4_dnl
 
 # If additional gettys are spawned during boot then we should make
 # sure that this is synchronized before getty.target, even though
--
cgit v0.8.3-6-g21f6
