From 25309d888c8b7f4724b952ed51c93f3f5de14d62 Mon Sep 17 00:00:00 2001
From: Miklos Vajna <vmiklos@frugalware.org>
Date: Sun, 12 Dec 2010 02:24:38 +0100
Subject: [PATCH] Frugalware: start gettys after local.service

---
 units/getty@.service.m4 |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/units/getty@.service.m4 b/units/getty@.service.m4
index 902ab6e..0b3160e 100644
--- a/units/getty@.service.m4
+++ b/units/getty@.service.m4
@@ -15,6 +15,9 @@ After=rc-local.service
 m4_ifdef(`TARGET_ARCH',
 After=rc-local.service
 )m4_dnl
+m4_ifdef(`TARGET_FRUGALWARE',
+After=local.service
+)m4_dnl
 
 # If additional gettys are spawned during boot then we should make
 # sure that this is synchronized before getty.target, even though
-- 
1.7.3.2

