--- fgrun-1.0.1/src/advanced.cxx	2008-04-22 22:51:01.000000000 +0200
+++ fgrun-1.0.1/src/advanced.cxx	2008-06-21 16:30:50.000000000 +0200
@@ -6,6 +6,8 @@
 #endif
 #include "advanced.h"
 
+#include <cstring>
+
 void Advanced::cb_page_list_i(Fl_Browser* o, void*) {
   if (o->value() > 0)
   show_page(o->value()-1);

--- fgrun-1.0.1/src/settings.cxx	2008-04-22 22:51:01.000000000 +0200
+++ fgrun-1.0.1/src/settings.cxx	2008-06-21 16:34:40.000000000 +0200
@@ -25,6 +25,7 @@
 #endif
 
 #include <string>
+#include <cstring>
 
 #include <FL/filename.H>

