--- keyserver/gpgkeys_curl.c	2007-07-04 15:26:13.000000000 +0200
+++ keyserver/gpgkeys_curl.c.new	2008-04-16 22:14:52.000000000 +0200
@@ -300,7 +300,7 @@
       curl_easy_setopt(curl,CURLOPT_VERBOSE,1);
     }
 
-  curl_easy_setopt(curl,CURLOPT_SSL_VERIFYPEER,opt->flags.check_cert);
+  curl_easy_setopt(curl,CURLOPT_SSL_VERIFYPEER, opt->flags.check_cert != 0);
   curl_easy_setopt(curl,CURLOPT_CAINFO,opt->ca_cert_file);
 
   if(proxy)
