diff -Naur glib-1.2.10.orig/gstrfuncs.c glib-1.2.10/gstrfuncs.c
--- glib-1.2.10.orig/gstrfuncs.c	2001-02-27 07:00:22.000000000 +0100
+++ glib-1.2.10/gstrfuncs.c	2004-06-18 23:35:30.000000000 +0200
@@ -47,6 +47,8 @@
  * inteferes with g_strsignal() on some OSes
  */
 
+#define G_GNUC_PRETTY_FUNCTION
+
 typedef union  _GDoubleIEEE754  GDoubleIEEE754;
 #define G_IEEE754_DOUBLE_BIAS   (1023)
 /* multiply with base2 exponent to get base10 exponent (nomal numbers) */
