diff -Naur mpeg4ip-1.6/common/video/iso-mpeg4/src/Makefile.am mpeg4ip-1.6-p/common/video/iso-mpeg4/src/Makefile.am
--- mpeg4ip-1.6/common/video/iso-mpeg4/src/Makefile.am	2006-01-03 23:24:07.000000000 +0100
+++ mpeg4ip-1.6-p/common/video/iso-mpeg4/src/Makefile.am	2007-09-20 21:01:47.000000000 +0200
@@ -1,5 +1,5 @@
 
-AM_CXXFLAGS = -D__TRACE_AND_STATS_ -D__DOUBLE_PRECISION_ -D_REENTRANT -DNOCONTROLS -fexceptions -Wall -Werror -D_OBSS_ @BILLS_CPPWARNINGS@
+AM_CXXFLAGS = -D__TRACE_AND_STATS_ -D__DOUBLE_PRECISION_ -D_REENTRANT -DNOCONTROLS -fexceptions -Wall  -D_OBSS_ @BILLS_CPPWARNINGS@
 
 INCLUDES = -I$(srcdir)/../include
 
