diff -aur a/lib/rtp/net_udp.c b/lib/rtp/net_udp.c
--- a/lib/rtp/net_udp.c	2012-07-02 23:30:13.968751716 -0500
+++ b/lib/rtp/net_udp.c	2012-07-02 23:31:34.822748711 -0500
@@ -163,9 +163,8 @@
 	va_end(ap);
 	rtp_message(LOG_ALERT, "ERROR: %s, (%d - %s)\n", buffer, e, ws_errs[i].errname);
 #else
-	uint32_t retlen;
 	va_start(ap, msg);
-	retlen = vsnprintf(buffer, blen, msg, ap);
+	vsnprintf(buffer, blen, msg, ap);
 	va_end(ap);
 	rtp_message(LOG_ALERT, "%s:%s", buffer, strerror(errno));
 #endif
