--- player/plugin/audio/ffmpeg/ffmpeg.cpp.orig	2006-11-18 12:34:36.000000000 -0500
+++ player/plugin/audio/ffmpeg/ffmpeg.cpp	2006-11-18 12:34:50.000000000 -0500
@@ -166,7 +166,7 @@
     break;
   }
   if (userdata) {
-    ffmpeg->m_c->extradata = (void *)userdata;
+    ffmpeg->m_c->extradata = const_cast<uint8_t*>(userdata);
     ffmpeg->m_c->extradata_size = ud_size;
   }
   if (avcodec_open(ffmpeg->m_c, ffmpeg->m_codec) < 0) {
--- player/plugin/video/ffmpeg/ffmpeg.cpp.orig	2006-11-18 14:48:11.000000000 -0500
+++ player/plugin/video/ffmpeg/ffmpeg.cpp	2006-11-18 14:49:22.000000000 -0500
@@ -255,7 +255,7 @@
   }
     break;
   case CODEC_ID_SVQ3:
-    ffmpeg->m_c->extradata = (void *)userdata;
+    ffmpeg->m_c->extradata = const_cast<uint8_t*>(userdata);
     ffmpeg->m_c->extradata_size = ud_size;
     if (vinfo != NULL) {
       ffmpeg->m_c->width = vinfo->width;
