--- mpeg4ip-1.6/player/plugin/video/ffmpeg/ffmpeg.cpp	2007-07-02 19:43:14.000000000 +0200
+++ mpeg4ip-1.6/player/plugin/video/ffmpeg/ffmpeg.cpp	2009-12-22 14:29:42.000000000 +0100
@@ -585,10 +585,20 @@
 	from.data[ix] = ffmpeg->m_picture->data[ix];
 	from.linesize[ix] = ffmpeg->m_picture->linesize[ix];
       }
-      
+#ifdef HAVE_FFMPEG_INSTALLED
+      SwsContext *pSWSCtx;
+      pSWSCtx = sws_getContext(ffmpeg->m_c->width, ffmpeg->m_c->height,
+                               ffmpeg->m_c->pix_fmt,
+                               ffmpeg->m_c->width, ffmpeg->m_c->height,
+                               PIX_FMT_YUV420P, SWS_BICUBIC, 0, 0, 0);
+      sws_scale(pSWSCtx, from.data, from.linesize, 0, ffmpeg->m_c->height,
+                to.data, to.linesize);
+      sws_freeContext(pSWSCtx);
+#else
       img_convert(&to, PIX_FMT_YUV420P,
 		  &from, ffmpeg->m_c->pix_fmt,
 		  ffmpeg->m_c->width, ffmpeg->m_c->height);
+#endif
       ffmpeg->m_vft->video_filled_buffer(ffmpeg->m_ifptr,
 					 ffmpeg->have_cached_ts ?
 					 ffmpeg->cached_ts : ts);

--- mpeg4ip-1.6/player/plugin/video/ffmpeg/ffmpeg.h	2009-12-22 12:09:10.000000000 +0100
+++ mpeg4ip-1.6/player/plugin/video/ffmpeg/ffmpeg.h	2009-12-22 14:26:48.000000000 +0100
@@ -28,6 +28,7 @@
 extern "C" {
 #ifdef HAVE_FFMPEG_INSTALLED
 #include <libavcodec/avcodec.h>
+#include <libswscale/swscale.h>
 #else
 #include <avcodec.h>
 #endif
 
 --- mpeg4ip-1.6/player/plugin/audio/ffmpeg/ffmpeg.cpp	2009-12-22 12:04:12.000000000 +0100
+++ fw_mpeg4ip-1.6/player/plugin/audio/ffmpeg/ffmpeg.cpp	2009-12-22 12:04:12.000000000 +0100
@@ -220,7 +220,7 @@
   uint32_t freq_ts = pts->audio_freq_timestamp;
 
   do {
-#if HAVE_DECL_AVCODEC_DECODE_AUDIO2 != 1
+#if HAVE_DECL_AVCODEC_DECODE_AUDIO2 != 0
     used = avcodec_decode_audio(ffmpeg->m_c, (short *)ffmpeg->m_outbuf,
 				&outsize, buffer, left);
 #else
