diff -aur mpeg4ip-1.6/player/plugin/audio/ffmpeg/ffmpeg.h fw_mpeg4ip-1.6/player/plugin/audio/ffmpeg/ffmpeg.h
--- mpeg4ip-1.6/player/plugin/audio/ffmpeg/ffmpeg.h	2007-07-02 19:43:14.000000000 +0200
+++ fw_mpeg4ip-1.6/player/plugin/audio/ffmpeg/ffmpeg.h	2008-04-29 01:07:31.000000000 +0200
@@ -26,7 +26,7 @@
 #include "codec_plugin.h"
 extern "C" {
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
diff -aur mpeg4ip-1.6/player/plugin/video/ffmpeg/ffmpeg.h fw_mpeg4ip-1.6/player/plugin/video/ffmpeg/ffmpeg.h
--- mpeg4ip-1.6/player/plugin/video/ffmpeg/ffmpeg.h	2007-07-02 19:43:14.000000000 +0200
+++ fw_mpeg4ip-1.6/player/plugin/video/ffmpeg/ffmpeg.h	2008-04-29 01:08:26.000000000 +0200
@@ -27,7 +27,7 @@
 #include "codec_plugin.h"
 extern "C" {
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
diff -aur mpeg4ip-1.6/server/mp4live/audio_ffmpeg.h fw_mpeg4ip-1.6/server/mp4live/audio_ffmpeg.h
--- mpeg4ip-1.6/server/mp4live/audio_ffmpeg.h	2007-07-02 19:43:14.000000000 +0200
+++ fw_mpeg4ip-1.6/server/mp4live/audio_ffmpeg.h	2008-04-29 02:13:50.000000000 +0200
@@ -25,7 +25,7 @@
 #include "audio_encoder.h"
 extern "C" {
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
diff -aur mpeg4ip-1.6/server/mp4live/video_ffmpeg.h fw_mpeg4ip-1.6/server/mp4live/video_ffmpeg.h
--- mpeg4ip-1.6/server/mp4live/video_ffmpeg.h	2007-07-02 19:43:14.000000000 +0200
+++ fw_mpeg4ip-1.6/server/mp4live/video_ffmpeg.h	2008-04-29 02:15:13.000000000 +0200
@@ -25,7 +25,7 @@
 #include "video_encoder.h"
 extern "C" {
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
diff -aur mpeg4ip-1.6/server/mp4live/video_encoder_class.cpp fw_mpeg4ip-1.6/server/mp4live/video_encoder_class.cpp
--- mpeg4ip-1.6/server/mp4live/video_encoder_class.cpp	2007-07-02 19:43:14.000000000 +0200
+++ fw_mpeg4ip-1.6/server/mp4live/video_encoder_class.cpp	2008-04-29 02:24:15.000000000 +0200
@@ -27,7 +27,7 @@
 #ifdef HAVE_FFMPEG
 extern "C" {
 #ifdef HAVE_FFMPEG_INSTALLED
-#include <ffmpeg/avcodec.h>
+#include <libavcodec/avcodec.h>
 #else
 #include <avcodec.h>
 #endif
