diff -ur a/common/video/iso-mpeg4/configure.in b/common/video/iso-mpeg4/configure.in
--- a/common/video/iso-mpeg4/configure.in	2005-05-04 14:55:58.000000000 -0500
+++ b/common/video/iso-mpeg4/configure.in	2013-10-29 02:05:41.620852623 -0500
@@ -1,6 +1,6 @@
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT([mpeg4-2000], 1.2.3)
-AM_CONFIG_HEADER(mpeg4-2000.h)
+AC_CONFIG_HEADERS([mpeg4-2000.h])
 dnl Detect the canonical host and target build environment
 AC_CANONICAL_HOST
 AC_CANONICAL_TARGET
diff -ur a/configure.in b/configure.in
--- a/configure.in	2007-09-18 15:43:13.000000000 -0500
+++ b/configure.in	2013-10-29 02:01:06.964862835 -0500
@@ -1,7 +1,7 @@
 
 dnl Process this file with autoconf to produce a configure script.
 AC_INIT(mpeg4ip, 1.6)
-AM_CONFIG_HEADER(mpeg4ip_config.h)
+AC_CONFIG_HEADERS([mpeg4ip_config.h])
 dnl Prevent user from running just ./configure
 AC_CHECK_FILE(./bootstrapped, [did_bootstrap=yes], [did_bootstrap=no])
 if test x$did_bootstrap = xno; then
