diff -Nru mpeg4ip-1.6.1.orig/player/src/media_utils.cpp mpeg4ip-1.6.1/player/src/media_utils.cpp
--- mpeg4ip-1.6.1.orig/player/src/media_utils.cpp	2007-03-29 20:52:19.000000000 +0200
+++ mpeg4ip-1.6.1/player/src/media_utils.cpp	2009-06-13 18:44:36.000000000 +0200
@@ -610,7 +610,7 @@
   session_desc_t *sdp;
 
   name += strlen("iptv://");
-  slash = strchr(name, '/');
+  slash = strchr((char*)name, '/');
   if (slash == NULL || slash == name) {
     psptr->set_message("Invalid iptv content manager");
     return -1;


