diff -Naurp mpeg4ip-1.6/common/video/iso-mpeg4/src/type_basic.cpp mpeg4ip-1.6-p/common/video/iso-mpeg4/src/type_basic.cpp
--- mpeg4ip-1.6/common/video/iso-mpeg4/src/type_basic.cpp	2006-08-03 20:30:21.000000000 +0200
+++ mpeg4ip-1.6-p/common/video/iso-mpeg4/src/type_basic.cpp	2007-09-20 16:22:06.000000000 +0200
@@ -317,7 +317,7 @@ Void CMotionVector::computeMV ()
 	iHalfY = m_vctTrueHalfPel.y - iMVY * 2;
 }
 
-Void CMotionVector::setToZero (Void)
+Void CMotionVector::setToZero ()
 {
 	memset (this, 0, sizeof (*this));
 }
