--- src/audiere.h	2006-02-14 05:57:01.000000000 +0100
+++ audiere.h	2008-09-20 18:01:40.000000000 +0200
@@ -27,6 +27,7 @@
 
 
 #include <vector>
+#include <cstring>
 #include <string>
 
 #ifdef _MSC_VER
--- src/debug.h	2006-02-14 05:57:01.000000000 +0100
+++ debug.h	2008-09-20 18:09:08.000000000 +0200
@@ -4,6 +4,7 @@
 
 #include <assert.h>
 #include <stdio.h>
+#include <cstdlib>
 #include <string>
 
 
# This patch was created thanks to the help of the reztho's message on the AUR of Arch Linux.
# url: http://aur.archlinux.org/packages.php?ID=12005
# Thank you very much for reztho's help :-)
