--- mod_python-3.2.8-orig/conigure.in	2006-02-02 13:30:55.000000000 -0800
+++ mod_python-3.2.8/configure.in	2006-06-26 11:03:16.000000000 -0700
@@ -275,7 +275,7 @@
 # get the mod_python version
 AC_SUBST(MP_VERSION)
 MP_VERSION=`awk '/MPV_STRING/ {print $3}' src/include/mpversion.h`
-MP_VERSION=`echo $MP_VERSION | sed s/\\"//g`                                      
+MP_VERSION=`echo $MP_VERSION | sed s/\"//g`                                      
 
 AC_OUTPUT(Makefile src/Makefile Doc/Makefile test/testconf.py dist/setup.py dist/Makefile)
 

