diff -ur a/videowrapper.cpp b/videowrapper.cpp
--- a/videowrapper.cpp	2012-10-28 22:26:28.227203734 -0500
+++ b/videowrapper.cpp	2012-10-29 05:46:47.230377249 -0500
@@ -31,36 +31,25 @@
 {
     av_free_packet(&packet);
     avcodec_close(pCodecCtx);
-    av_close_input_file(pFormatCtx);
-    //avformat_close_input(&pFormatCtx);
+    //av_close_input_file(pFormatCtx);
+    avformat_close_input(&pFormatCtx);
 }
 
 int videowrapper::load_video(QString filename)
 {
     // *** OUVERTURE DU FICHIER VIDEO ***
-    #ifdef avformat_open_input
-        if (avformat_open_input(&pFormatCtx,filename.toAscii().data(),NULL,NULL)!=0)
-        {
-    #else
-        if (av_open_input_file(&pFormatCtx,filename.toAscii().data(),NULL,0,NULL)!=0)
-        {
-    #endif
+    if (avformat_open_input(&pFormatCtx,filename.toAscii().data(),NULL,NULL)!=0)
+    {
         qDebug() << "CAN'T OPEN VIDEO FILE : " << filename;
         return 1;
     }
 
     // *** RECUPERATION DES INFORMATIONS DU MEDIA ***
-    if(av_find_stream_info(pFormatCtx)<0)
-    {
+    if (avformat_find_stream_info(pFormatCtx, NULL) < 0) {
         qDebug() << "CAN'T READ INFORMATION FROM VIDEO FILE : " << filename;
         return 1;
     }
 
-    /*if (avformat_find_stream_info(pFormatCtx, NULL) < 0) {
-        qDebug() << "CAN'T READ INFORMATION FROM VIDEO FILE : " << filename;
-        return 1;
-    }*/
-
     // *** RECHERCHE STREAM VIDEO ***
 
     videoStream=-1;
