diff -aur partimage-0.6.8/src/client/netclient.h fw_partimage-0.6.8//src/client/netclient.h
--- partimage-0.6.8/src/client/netclient.h	2009-09-24 21:21:42.000000000 +0200
+++ fw_partimage-0.6.8//src/client/netclient.h	2010-06-11 23:50:06.000000000 +0200
@@ -35,7 +35,12 @@
 #ifdef HAVE_SSL
   SSL_CTX * ctx;
   X509 * server_cert;
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+  SSL_METHOD const * meth;
+#else
   SSL_METHOD * meth;
+#endif // OPENSSL_VERSION_NUMBER
+
 #endif
   bool m_bUseSSL;
   bool m_bMustLogin;
diff -aur partimage-0.6.8/src/server/netserver.h fw_partimage-0.6.8//src/server/netserver.h
--- partimage-0.6.8/src/server/netserver.h	2009-09-12 10:06:22.000000000 +0200
+++ fw_partimage-0.6.8//src/server/netserver.h	2010-06-11 23:58:30.000000000 +0200
@@ -41,7 +41,12 @@
 #ifdef HAVE_SSL
   SSL_CTX * ctx;
   X509 * client_cert;
-  SSL_METHOD * meth; 
+#if OPENSSL_VERSION_NUMBER >= 0x10000000L
+  SSL_METHOD const * meth; 
+#else
+  SSL_METHOD * meth;
+#endif // OPENSSL_VERSION_NUMBER
+ 
   int err;
 #endif
 
