--- ShellCheck-0.9.0/ShellCheck.cabal	2001-09-09 03:46:40.000000000 +0200
+++ ShellCheck-0.9.0.new/ShellCheck.cabal	2023-07-22 22:19:08.475756571 +0200
@@ -47,7 +47,7 @@
     build-depends:
       -- The lower bounds are based on GHC 7.10.3
       -- The upper bounds are based on GHC 9.4.3
-      aeson                >= 1.4.0 && < 2.2,
+      aeson                >= 1.4.0 && < 2.3,
       array                >= 0.5.1 && < 0.6,
       base                 >= 4.8.0.0 && < 5,
       bytestring           >= 0.10.6 && < 0.12,
