--- shellcheck/src/ShellCheck/Parser.hs	2018-11-09 13:52:40.638492545 +0100
+++ shellcheck.new/src/ShellCheck/Parser.hs	2018-11-09 13:53:55.194490127 +0100
@@ -1916,7 +1916,7 @@
   where
     checkBadBreak = optional $ do
                 pos <- getPosition
-                try $ lookAhead (oneOf "|&") -- |, || or &&
+                try $ lookAhead (oneOf "|&") 
                 parseProblemAt pos ErrorC 1133 "Unexpected start of line. If breaking lines, |/||/&& should be at the end of the previous one."
 readLineBreak = optional readNewlineList
 
