diff -urN galago-sharp-0.5.0.orig/tests/Makefile.am galago-sharp-0.5.0/tests/Makefile.am
--- galago-sharp-0.5.0.orig/tests/Makefile.am	2006-04-05 09:01:54.000000000 +0100
+++ galago-sharp-0.5.0/tests/Makefile.am	2006-08-21 08:06:38.000000000 +0100
@@ -23,7 +23,7 @@
 	$(CSC) /out:$@ $(REFERENCES) $(srcdir)/$<
 
 unit-tests.dll: unit-tests.cs $(ASSEMBLIES)
-	$(CSC) /out:$@ /target:library $(REFERENCES) -pkg:nunit $(srcdir)/$<
+	$(CSC) /out:$@ /target:library $(REFERENCES) -pkg:mono-nunit $(srcdir)/$<
 
 EXTRA_DIST = \
 	get-presence.cs \
diff -urN galago-sharp-0.5.0.orig/tests/Makefile.in galago-sharp-0.5.0/tests/Makefile.in
--- galago-sharp-0.5.0.orig/tests/Makefile.in	2006-04-20 21:12:09.000000000 +0100
+++ galago-sharp-0.5.0/tests/Makefile.in	2006-08-21 08:06:28.000000000 +0100
@@ -361,7 +361,7 @@
 	$(CSC) /out:$@ $(REFERENCES) $(srcdir)/$<
 
 unit-tests.dll: unit-tests.cs $(ASSEMBLIES)
-	$(CSC) /out:$@ /target:library $(REFERENCES) -pkg:nunit $(srcdir)/$<
+	$(CSC) /out:$@ /target:library $(REFERENCES) -pkg:mono-nunit $(srcdir)/$<
 # Tell versions [3.59,3.63) of GNU make to not export all variables.
 # Otherwise a system limit (for SysV at least) may be exceeded.
 .NOEXPORT:
