--- bifunctors-5.6.1/bifunctors.cabal	2001-09-09 03:46:40.000000000 +0200
+++ bifunctors-5.6.1.new/bifunctors.cabal	2023-12-11 12:15:13.145783084 +0100
@@ -45,9 +45,9 @@
     base                     >= 4.9     && < 5,
     assoc                    >= 1.1     && < 1.2,
     comonad                  >= 5.0.7   && < 6,
-    containers               >= 0.5.7.1 && < 0.7,
-    template-haskell         >= 2.11    && < 2.21,
-    th-abstraction           >= 0.4.2.0 && < 0.6,
+    containers               >= 0.5.7.1 && < 0.8,
+    template-haskell         >= 2.11    && < 2.22,
+    th-abstraction           >= 0.4.2.0 && < 0.7,
     transformers             >= 0.5     && < 0.7
 
   if !impl(ghc >= 8.2)
