--- bifunctors-5.6.2/bifunctors.cabal	2001-09-09 03:46:40.000000000 +0200
+++ bifunctors-5.6.2.new/bifunctors.cabal	2024-12-26 20:18:55.203306372 +0100
@@ -48,7 +48,7 @@
     assoc                    >= 1.1     && < 1.2,
     comonad                  >= 5.0.7   && < 6,
     containers               >= 0.5.7.1 && < 0.8,
-    template-haskell         >= 2.11    && < 2.23,
+    template-haskell         >= 2.11    && < 2.24,
     th-abstraction           >= 0.4.2.0 && < 0.8,
     transformers             >= 0.5     && < 0.7
 
