if a subdir fails, we need to abort

Index: autoconf/targets.mak
===================================================================
RCS file: /cvsroot/apcupsd/apcupsd/autoconf/targets.mak,v
retrieving revision 1.21
retrieving revision 1.22
diff -u -p -r1.21 -r1.22
--- autoconf/targets.mak	21 Sep 2006 00:07:35 -0000	1.21
+++ autoconf/targets.mak	20 Dec 2006 01:46:11 -0000	1.22
@@ -22,10 +22,11 @@ all-subdirs:
 	@if test ! x"$(subdirs)" = x; then \
 	    for file in . ${subdirs}; \
 	    do \
-		(cd $$file && if test "$$file" != "."; then $(MAKE) DESTDIR=$(DESTDIR) all; fi); \
-		  if test "$$?" != "0"; then \
-		     break; \
-		  fi; \
+		(cd $$file; \
+		 if test "$$file" != "."; then \
+		     $(MAKE) DESTDIR=$(DESTDIR) all || exit $$?; \
+		 fi; \
+		) || exit $$?; \
 	    done; \
 	fi
 
