diff -Nrup -x'*.orig' codeblocks-8.02.orig/src/sdk/configmanager.cpp codeblocks-8.02/src/sdk/configmanager.cpp
--- codeblocks-8.02.orig/src/sdk/configmanager.cpp	2008-02-27 14:15:26.000000000 +0100
+++ codeblocks-8.02/src/sdk/configmanager.cpp	2008-08-29 07:40:45.000000000 +0200
@@ -467,7 +467,7 @@ wxString ConfigManager::GetFolder(Search
             return ::wxGetCwd();
 
         case sdPluginsGlobal:
-            return ConfigManager::data_path_global + _T("/plugins");
+            return _T("@libdir@/codeblocks/plugins");
 
         case sdPluginsUser:
             return ConfigManager::data_path_user   + _T("/plugins");
