--- codeblocks/src/setup/mime/codeblocks.desktop.old	2006-09-09 18:03:05.000000000 +0200
+++ codeblocks/src/setup/mime/codeblocks.desktop	2006-09-10 10:37:08.000000000 +0200
@@ -1,14 +1,12 @@
 [Desktop Entry]
 Version=1.0
-Encoding=UTF-8
 Name=Code::Blocks IDE
 Comment=Configurable and extensible IDE
 Exec=codeblocks %F
-Icon=codeblocks.png
+Icon=codeblocks
 Terminal=false
 X-MultipleArgs=false
 Type=Application
-Categories=Application;Development;
+Categories=Development;IDE;
 StartupNotify=true
 MimeType=application/x-codeblocks;application/x-codeblocks-workspace;
-GenericName[en_GR]=
