Index: src/plugins/compilergcc/compilergcc.cpp
===================================================================
--- src/plugins/compilergcc/compilergcc.cpp	(revision 5202)
+++ src/plugins/compilergcc/compilergcc.cpp	(working copy)
@@ -1723,6 +1723,7 @@
 
     wxString cmd;
     wxString command;
+    bool has_explicit_shell = false;
     wxFileName f(out);
     f.MakeAbsolute(m_Project->GetBasePath());
 
@@ -1770,6 +1771,9 @@
                 //  and its use for such purposes should be strictly banned!"
                 //                 -- Csh Programming Considered Harmful
                 command << DEFAULT_CONSOLE_SHELL << strSPACE;
+                
+                has_explicit_shell = true;
+                command << _T("'");
             }
         }
 
@@ -1820,6 +1824,9 @@
         return -1;
     }
 
+    if (has_explicit_shell)
+	command << _T("'");
+
     wxString script = command;
 
     if (platform::macosx)
