diff -Naur codeblocks-17.12/src/plugins/contrib/wxSmith/properties/wxsproperty.h codeblocks-17.12.new/src/plugins/contrib/wxSmith/properties/wxsproperty.h
--- codeblocks-17.12/src/plugins/contrib/wxSmith/properties/wxsproperty.h	2017-11-01 00:32:40.000000000 +0100
+++ codeblocks-17.12.new/src/plugins/contrib/wxSmith/properties/wxsproperty.h	2019-12-23 14:30:24.655068933 +0100
@@ -48,7 +48,7 @@
 #if wxCHECK_VERSION(3, 0, 0) || wxCHECK_PROPGRID_VERSION(1, 4, 0)
 #define wxPGVariant                     wxVariant
 #define wxPGId                          wxPGProperty*
-#define wxParentProperty                wxPGProperty
+#define wxParentProperty                wxStringProperty
 #define wxCustomPropertyClass           wxPGProperty
 #define wxEnumPropertyClass             wxEnumProperty
 #define wxPG_VALUETYPE(T)               wxT(#T)
diff -Naur codeblocks-17.12/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxslistbox.cpp codeblocks-17.12.new/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxslistbox.cpp
--- codeblocks-17.12/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxslistbox.cpp	2017-11-01 00:32:39.000000000 +0100
+++ codeblocks-17.12.new/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxslistbox.cpp	2019-12-23 13:54:43.404240925 +0100
@@ -23,6 +23,7 @@
 #include "wxslistbox.h"
 
 #include <prep.h>
+#include <wx/listbox.h>
 
 namespace
 {
diff -Naur codeblocks-17.12/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsstatusbar.cpp codeblocks-17.12.new/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsstatusbar.cpp
--- codeblocks-17.12/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsstatusbar.cpp	2017-11-01 00:32:40.000000000 +0100
+++ codeblocks-17.12.new/src/plugins/contrib/wxSmith/wxwidgets/defitems/wxsstatusbar.cpp	2019-12-23 14:31:15.773427198 +0100
@@ -149,6 +149,7 @@
     for ( int i=0; i<m_Fields; i++ )
     {
         wxPGId ParentProp = Grid->Append(NEW_IN_WXPG14X wxParentProperty(wxString::Format(_("Field %d"),i+1),wxPG_LABEL));
+	ParentProp->ChangeFlag(wxPG_PROP_READONLY, true);
         m_WidthsIds[i] = Grid->AppendIn(ParentProp,NEW_IN_WXPG14X wxIntProperty(_("Width"),wxPG_LABEL,m_Widths[i]));
         m_VarWidthIds[i] = Grid->AppendIn(ParentProp,NEW_IN_WXPG14X wxBoolProperty(_T("Variable width"),wxPG_LABEL,m_VarWidth[i]));
         Grid->SetPropertyAttribute(m_VarWidthIds[i],wxPG_BOOL_USE_CHECKBOX,1L,wxPG_RECURSE);
@@ -195,6 +196,7 @@
             for ( int i = m_Fields; i<NewFields; i++ )
             {
                 wxPGId ParentProp = Grid->Append(NEW_IN_WXPG14X wxParentProperty(wxString::Format(_("Field %d"),i+1),wxPG_LABEL));
+		ParentProp->ChangeFlag(wxPG_PROP_READONLY, true);
                 m_WidthsIds[i] = Grid->AppendIn(ParentProp,NEW_IN_WXPG14X wxIntProperty(_("Width"),wxPG_LABEL,m_Widths[i]));
                 m_VarWidthIds[i] = Grid->AppendIn(ParentProp,NEW_IN_WXPG14X wxBoolProperty(_T("Variable width"),wxPG_LABEL,m_VarWidth[i]));
                 Grid->SetPropertyAttribute(m_VarWidthIds[i],wxPG_BOOL_USE_CHECKBOX,1L,wxPG_RECURSE);
diff -Naur codeblocks-17.12/src/src/watchesdlg.cpp codeblocks-17.12.new/src/src/watchesdlg.cpp
--- codeblocks-17.12/src/src/watchesdlg.cpp	2017-11-01 00:32:38.000000000 +0100
+++ codeblocks-17.12.new/src/src/watchesdlg.cpp	2019-12-23 13:55:28.152548031 +0100
@@ -10,6 +10,7 @@
 #include "sdk.h"
 #ifndef CB_PRECOMP
     #include <wx/app.h>
+    #include <wx/dcclient.h>
     #include <wx/dnd.h>
     #include <wx/fontutil.h>
     #include <wx/menu.h>
@@ -27,7 +28,7 @@
 #include <algorithm>
 
 #include <wx/propgrid/propgrid.h>
-
+#include <wx/propgrid/editors.h>
 #include "watchesdlg.h"
 
 #include "cbcolourmanager.h"
