--- Glob-0.10.1/Glob.cabal	2001-09-09 03:46:40.000000000 +0200
+++ Glob-0.10.1.new/Glob.cabal	2021-07-26 11:13:29.641293518 +0200
@@ -31,7 +31,7 @@
                 , dlist        >= 0.4 && < 1.1
                 , filepath     >= 1.1 && < 1.5
                 , transformers >= 0.2 && < 0.6
-                , transformers-compat >= 0.3 && < 0.7
+                , transformers-compat >= 0.3 && < 0.8
 
    if impl(ghc < 8.0)
       Build-Depends: semigroups >= 0.18 && < 0.20
