--- Glob-0.10.2/Glob.cabal	2001-09-09 03:46:40.000000000 +0200
+++ Glob-0.10.2.new/Glob.cabal	2023-03-17 12:25:11.549838845 +0100
@@ -30,7 +30,7 @@
                 , directory    <  1.4
                 , dlist        >= 0.4 && < 1.1
                 , filepath     >= 1.1 && < 1.5
-                , transformers >= 0.2 && < 0.6
+                , transformers >= 0.2 && < 0.6.3
                 , transformers-compat >= 0.3 && < 0.8
 
    if impl(ghc < 8.0)
