--- site-packages/Skype4Py/API/posix.py.orig	2007-08-20 04:22:18.000000000 +0200
+++ site-packages/Skype4Py/API/posix.py	2007-08-20 04:23:00.000000000 +0200
@@ -132,7 +132,7 @@
         self.Commands[Command.Id] = Command
         try:
             result = self.skype_out.Invoke(com)
-        except dbus.DbusException:
+        except dbus.DBusException:
             # Skype probably closed
             return False
         if result.startswith(u'#%d ' % Command.Id):
