diff --git a/data/kaccounts/ktp-haze-facebook-im.service.in b/data/kaccounts/ktp-haze-facebook-im.service.in
new file mode 100644
index 0000000..85ea473
--- /dev/null
+++ b/data/kaccounts/ktp-haze-facebook-im.service.in
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<service id="ktp-haze-facebook-im">
+  <type>IM</type>
+  <!-- Facebook Chat - Service name being displayed as a checkbox label to enable/disable this service -->
+  <_name>Chat</_name>
+  <icon>im-facebook</icon>
+  <provider>ktp-haze-facebook</provider>
+  <translations>kaccounts-providers</translations>
+
+  <template>
+      <group name="telepathy">
+          <setting name="manager">haze</setting>
+          <setting name="protocol">facebook</setting>
+      </group>
+      <group name="auth">
+          <setting name="method">password</setting>
+          <setting name="mechanism">password</setting>
+      </group>
+  </template>
+</service>
diff --git a/data/kaccounts/ktp-haze-facebook.provider.in b/data/kaccounts/ktp-haze-facebook.provider.in
new file mode 100644
index 0000000..b491030
--- /dev/null
+++ b/data/kaccounts/ktp-haze-facebook.provider.in
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<provider id="ktp-haze-facebook">
+    <name>Facebook</name>
+    <icon>im-facebook</icon>
+
+    <description>Facebook chat account</description>
+    <translations>kaccounts-providers</translations>
+    <plugin>ktpaccountskcm_plugin_kaccounts</plugin>
+</provider>
diff --git a/data/kaccounts/ktp-haze-skypeweb-im.service.in b/data/kaccounts/ktp-haze-skypeweb-im.service.in
new file mode 100644
index 0000000..000d2fc
--- /dev/null
+++ b/data/kaccounts/ktp-haze-skypeweb-im.service.in
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<service id="ktp-haze-skypeweb-im">
+  <type>IM</type>
+  <!-- Skype Chat - Service name being displayed as a checkbox label to enable/disable this service -->
+  <_name>Chat</_name>
+  <icon>im-skype</icon>
+  <provider>ktp-haze-skypeweb</provider>
+  <translations>kaccounts-providers</translations>
+
+  <template>
+      <group name="telepathy">
+          <setting name="manager">haze</setting>
+          <setting name="protocol">skypeweb</setting>
+      </group>
+      <group name="auth">
+          <setting name="method">password</setting>
+          <setting name="mechanism">password</setting>
+      </group>
+  </template>
+</service>
diff --git a/data/kaccounts/ktp-haze-skypeweb.provider.in b/data/kaccounts/ktp-haze-skypeweb.provider.in
new file mode 100644
index 0000000..27dc7b8
--- /dev/null
+++ b/data/kaccounts/ktp-haze-skypeweb.provider.in
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<provider id="ktp-haze-skypeweb">
+    <name>Skype</name>
+    <icon>im-skype</icon>
+
+    <description>Skype chat account</description>
+    <translations>kaccounts-providers</translations>
+    <plugin>ktpaccountskcm_plugin_kaccounts</plugin>
+</provider>
