diff -Naur ktp-accounts-kcm-19.08.3/plugins/gabble/CMakeLists.txt ktp-accounts-kcm-19.08.3-p/plugins/gabble/CMakeLists.txt
--- ktp-accounts-kcm-19.08.3/plugins/gabble/CMakeLists.txt	2019-11-04 09:08:02.000000000 +0100
+++ ktp-accounts-kcm-19.08.3-p/plugins/gabble/CMakeLists.txt	2019-11-14 12:11:12.664602545 +0100
@@ -37,6 +37,7 @@
                        ktpaccountskcminternal
                        Qt5::Core
                        Qt5::Widgets
+                       Qt5::Network
                        ${TELEPATHY_QT5_LIBRARIES}
 )
 
