diff -Naur ktp-accounts-kcm-19.08.3/src/KCMTelepathyAccounts/CMakeLists.txt ktp-accounts-kcm-19.08.3-p/src/KCMTelepathyAccounts/CMakeLists.txt
--- ktp-accounts-kcm-19.08.3/src/KCMTelepathyAccounts/CMakeLists.txt	2019-11-04 09:08:02.000000000 +0100
+++ ktp-accounts-kcm-19.08.3-p/src/KCMTelepathyAccounts/CMakeLists.txt	2019-11-14 11:57:35.625135877 +0100
@@ -72,6 +72,7 @@
 target_link_libraries (ktpaccountskcminternal
                        Qt5::Widgets
                        Qt5::Core
+                       Qt5::DBus
 #                        KF5::KIO
                        KF5::I18n
                        KF5::WidgetsAddons
