diff -up yafray/src/shaders/basictex.cc.gcc44 yafray/src/shaders/basictex.cc
--- yafray/src/shaders/basictex.cc.gcc44	2009-03-02 11:26:49.000000000 +0100
+++ yafray/src/shaders/basictex.cc	2009-03-02 11:27:25.000000000 +0100
@@ -261,7 +261,7 @@ textureImage_t::textureImage_t(const cha
 		intp_type = BICUBIC;
 	
 	// Load image, try to determine from extensions first
-	char *ext = strrchr(filename, '.');
+	const char *ext = strrchr(filename, '.');
 	bool jpg_tried = false;
 	bool tga_tried = false;
 	bool hdr_tried = false;
