diff -urN gnome-media-2.18.0.orig/gst-mixer/gnome-volume-control.desktop.in.in gnome-media-2.18.0/gst-mixer/gnome-volume-control.desktop.in.in
--- gnome-media-2.18.0.orig/gst-mixer/gnome-volume-control.desktop.in.in	2007-04-15 10:03:42.000000000 +0100
+++ gnome-media-2.18.0/gst-mixer/gnome-volume-control.desktop.in.in	2007-04-15 10:03:56.000000000 +0100
@@ -8,7 +8,7 @@
 X-Info=Gnome Mixer
 Terminal=false
 Type=Application
-Categories=GNOME;GTK;AudioVideo;Settings;
+Categories=GNOME;GTK;AudioVideo;Settings;HardwareSettings;
 X-GNOME-DocPath=gst-mixer/gnome-volume-control.xml
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=gnome-media
