--- project.pro	2009-08-05 20:34:41.000000000 +0200
+++ project.pro	2009-08-05 20:35:02.000000000 +0200
@@ -2,10 +2,6 @@
 CONFIG += release qt thread
 DEFINES += QT_NO_DEBUG_OUTPUT
 
-# Assume openmp-capable g++ (>=4.2)
-QMAKE_CXXFLAGS += -funroll-loops -fstrength-reduce -fschedule-insns2 -felide-constructors -frerun-loop-opt -fexceptions -fno-strict-aliasing -fexpensive-optimizations -ffast-math -pipe -fopenmp -msse2
-QMAKE_LFLAGS += -fopenmp
-
 TARGET = qtpfsgui
 
 MOC_DIR = generated_moc
@@ -295,7 +291,7 @@
 
 ############################## required by "make install" ########################################
 isEmpty(PREFIX) {
-        PREFIX = /usr/local
+        PREFIX = /usr
 }
 isEmpty(I18NDIR) {
 	mac {
