--- project.pro.orig	2007-10-01 22:17:56.000000000 +0200
+++ project.pro	2007-10-01 22:23:48.000000000 +0200
@@ -1,8 +1,6 @@
 TEMPLATE = app
 CONFIG += release qt thread
 DEFINES += QT_NO_DEBUG_OUTPUT
-QMAKE_CXXFLAGS += -O3 -funroll-loops -fstrength-reduce -fschedule-insns2 -felide-constructors -frerun-loop-opt -fexceptions -fno-strict-aliasing -fexpensive-optimizations -ffast-math -pipe
-QMAKE_CXXFLAGS_RELEASE-=-O2
 
 unix {
 ########################################### QT ###########################################
@@ -163,7 +161,7 @@
 
 ############################## required by "make install" ########################################
 isEmpty(PREFIX) {
-        PREFIX = /usr/local
+        PREFIX = /usr
 }
 isEmpty(I18NDIR) {
 	I18NDIR = $${PREFIX}/share/qtpfsgui/i18n
