--- typst-symbols-0.1.5/typst-symbols.cabal	2001-09-09 03:46:40.000000000 +0200
+++ typst-symbols-0.1.5.new/typst-symbols.cabal	2023-12-13 15:22:24.500248369 +0100
@@ -27,6 +27,6 @@
                       Typst.Shorthands
                       Typst.Emoji
     build-depends:    base >= 4.12 && < 5,
-                      text >= 1.0 && < 2.1
+                      text >= 1.0 && < 2.2
     hs-source-dirs:   src
     default-language: Haskell2010
