--- xfwm4-4.8.2/defaults/defaults	2011-09-22 21:12:37.000000000 +0200
+++ xfwm4-4.8.2new/defaults/defaults	2011-11-11 15:05:53.000000000 +0100
@@ -49,9 +49,9 @@
 snap_to_border=true
 snap_to_windows=false
 snap_width=10
-theme=Default
+theme=Frugalware
 title_alignment=center
-title_font=Sans Bold 9
+title_font=Droid Sans Bold 9
 title_horizontal_offset=0
 title_shadow_active=false
 title_shadow_inactive=false
