--- defaults/defaults.orig	2013-09-23 19:35:27.000000000 +0200
+++ defaults/defaults	2013-09-23 22:17:24.734973955 +0200
@@ -50,10 +50,10 @@
 snap_to_windows=false
 snap_width=10
 sync_to_vblank=false
-theme=Default
+theme=Frugalware
 tile_on_move=true
 title_alignment=center
-title_font=Sans Bold 9
+title_font=Droid Sans Bold 9
 title_horizontal_offset=0
 title_shadow_active=false
 title_shadow_inactive=false
