diff -Naur xfwm4-4.4.0-old/mcs-plugin/xfwm4_plugin.h xfwm4-4.4.0/mcs-plugin/xfwm4_plugin.h
--- xfwm4-4.4.0-old/mcs-plugin/xfwm4_plugin.h	2007-01-23 17:15:28.000000000 +0530
+++ xfwm4-4.4.0/mcs-plugin/xfwm4_plugin.h	2007-01-23 17:16:01.000000000 +0530
@@ -29,12 +29,12 @@
 #define CHANNEL2  "xfwm4_keys"
 #define PLUGIN_NAME "xfwm4"
 
-#define DEFAULT_THEME "Default"
+#define DEFAULT_THEME "Frugalware"
 #define DEFAULT_KEY_THEME "Default"
 #define DEFAULT_LAYOUT "OT|SHMC"
 #define DEFAULT_ACTION "maximize"
-#define DEFAULT_ALIGN "center"
-#define DEFAULT_FONT "Sans Bold 9"
+#define DEFAULT_ALIGN "left"
+#define DEFAULT_FONT "Sans Bold 12"
 
 #define MAX_ELEMENTS_BEFORE_SCROLLING 6
 
