diff -Naur xfwm4-4.3.99.1.orig/mcs-plugin/xfwm4_plugin.h xfwm4-4.3.99.1/mcs-plugin/xfwm4_plugin.h
--- xfwm4-4.3.99.1.orig/mcs-plugin/xfwm4_plugin.h	2006-09-02 21:31:35.000000000 +0530
+++ xfwm4-4.3.99.1/mcs-plugin/xfwm4_plugin.h	2006-09-06 12:33:44.000000000 +0530
@@ -29,11 +29,11 @@
 #define CHANNEL2  "xfwm4_keys"
 #define PLUGIN_NAME "xfwm4"
 
-#define DEFAULT_THEME "Default"
+#define DEFAULT_THEME "Frugalware"
 #define DEFAULT_KEY_THEME "Default"
 #define DEFAULT_LAYOUT "O|HMC"
 #define DEFAULT_ACTION "maximize"
-#define DEFAULT_ALIGN "center"
+#define DEFAULT_ALIGN "left"
 #define DEFAULT_FONT "Sans Bold 9"
 
 #define MAX_ELEMENTS_BEFORE_SCROLLING 6
