diff -urN xfwm4-4.3.99.1.orig/mcs-plugin/xfwm4_plugin.h xfwm4-4.3.99.1/mcs-plugin/xfwm4_plugin.h
--- xfwm4-4.3.99.1.orig/mcs-plugin/xfwm4_plugin.h	2006-09-23 13:43:24.000000000 +0100
+++ xfwm4-4.3.99.1/mcs-plugin/xfwm4_plugin.h	2006-09-23 13:44:01.000000000 +0100
@@ -29,12 +29,12 @@
 #define CHANNEL2  "xfwm4_keys"
 #define PLUGIN_NAME "xfwm4"
 
-#define DEFAULT_THEME "Default"
+#define DEFAULT_THEME "Frugalware"
 #define DEFAULT_KEY_THEME "Default"
 #define DEFAULT_LAYOUT "O|HMC"
 #define DEFAULT_ACTION "maximize"
-#define DEFAULT_ALIGN "center"
-#define DEFAULT_FONT "Sans Bold 9"
+#define DEFAULT_ALIGN "left"
+#define DEFAULT_FONT "Sans Bold 12"
 
 #define MAX_ELEMENTS_BEFORE_SCROLLING 6
 
