--- android-tools-35.0.2/vendor/extras/libjsonpb/parse/jsonpb.cpp	2024-08-29 19:46:57.000000000 +0200
+++ android-tools-35.0.2.new/vendor/extras/libjsonpb/parse/jsonpb.cpp	2025-03-13 11:30:35.462621724 +0100
@@ -33,7 +33,7 @@
 static constexpr char kTypeUrlPrefix[] = "type.googleapis.com";
 
 std::string GetTypeUrl(const Message& message) {
-  return std::string(kTypeUrlPrefix) + "/" + message.GetDescriptor()->full_name();
+  return std::string(kTypeUrlPrefix) + "/" + std::string(message.GetDescriptor()->full_name());
 }
 
 ErrorOr<std::string> MessageToJsonString(const Message& message) {
