--- android-tools-34.0.1.orig/vendor/CMakeLists.txt     2023-06-28 23:15:15.544237905 +0200
+++ android-tools-34.0.1/vendor/CMakeLists.txt  2023-06-28 23:45:28.672277935 +0200
@@ -73,6 +73,7 @@
 pkg_check_modules(libzstd REQUIRED IMPORTED_TARGET libzstd)
 
 find_package(Protobuf REQUIRED)
+set(PROTOBUF_LIBRARIES protobuf::libprotobuf)
 set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
 
--- android-tools-34.0.1.orig/vendor/extras/libjsonpb/parse/jsonpb.cpp  2023-06-28 23:15:15.414236253 +0200
+++ android-tools-34.0.1/vendor/extras/libjsonpb/parse/jsonpb.cpp       2023-06-28 23:34:21.672962973 +0200
@@ -51,7 +51,7 @@
 #if GOOGLE_PROTOBUF_VERSION < 3016000
     return MakeError<std::string>(status.error_message().as_string());
 #else
-    return MakeError<std::string>(status.message().as_string());
+    return MakeError<std::string>(std::string(status.message()));
 #endif
   }
   return ErrorOr<std::string>(std::move(json));
@@ -68,7 +68,7 @@
 #if GOOGLE_PROTOBUF_VERSION < 3016000
     return MakeError<std::monostate>(status.error_message().as_string());
 #else
-    return MakeError<std::monostate>(status.message().as_string());
+    return MakeError<std::monostate>(std::string(status.message()));
 #endif
   }
   if (!message->ParseFromString(binary)) {
