diff -ur a/src/logmill.cpp b/src/logmill.cpp
--- a/src/logmill.cpp	2012-11-29 07:16:09.454462733 -0600
+++ b/src/logmill.cpp	2012-11-29 07:16:32.345470708 -0600
@@ -176,7 +176,7 @@
                     logfile = repo_path.string();
                 }
             }
-        } catch(boost::filesystem3::filesystem_error& error) {
+        } catch(boost::filesystem::filesystem_error& error) {
         }
     }
 
