diff -Naur apmd.orig/apmd.c apmd/apmd.c
--- apmd.orig/apmd.c	2005-06-19 14:44:01.000000000 +0000
+++ apmd/apmd.c	2005-06-19 14:44:58.000000000 +0000
@@ -71,8 +71,8 @@
 
 
 #ifdef APM_TRACE
-#define ADEBUG(lev,args...) \
-    if (APM_TRACE>=lev) syslog(LOG_DEBUG, __FUNCTION__ ": " args)
+#define ADEBUG(lev, text, args...) \
+    if (APM_TRACE>=lev) syslog(LOG_DEBUG, "%s : " text, __FUNCTION__, args)
 #else
 #define ADEBUG(args...)
 #endif
diff -Naur apmd.orig/apmsleep.c apmd/apmsleep.c
--- apmd.orig/apmsleep.c	2005-06-19 14:44:01.000000000 +0000
+++ apmd/apmsleep.c	2005-06-19 14:45:40.000000000 +0000
@@ -42,6 +42,10 @@
  *
  */
 
+typedef unsigned char u8;
+typedef unsigned short u16;
+typedef unsigned int u32;
+
 #include <stdio.h>
 #include <linux/version.h>
 
