diff -Naur debootstrap.orig/Makefile debootstrap/Makefile
--- debootstrap.orig/Makefile	2008-04-30 23:50:28.000000000 +0200
+++ debootstrap/Makefile	2008-04-30 23:50:47.000000000 +0200
@@ -5,7 +5,7 @@
 
 MAKEDEV := $(shell if [ -e /dev/MAKEDEV ]; then echo /dev/MAKEDEV; else echo /sbin/MAKEDEV; fi)
 
-all: devices.tar.gz
+all:
 clean:
 	rm -f devices.tar.gz
 	rm -rf dev
@@ -30,14 +30,3 @@
 	sed 's/@VERSION@/$(VERSION)/g' debootstrap >$(DESTDIR)/usr/sbin/debootstrap
 	chown root:root $(DESTDIR)/usr/sbin/debootstrap
 	chmod 0755 $(DESTDIR)/usr/sbin/debootstrap
-
-	install -o root -g root -m 0644 devices.tar.gz $(DSDIR)/
-
-devices.tar.gz:
-	rm -rf dev
-	mkdir -p dev
-	chown 0:0 dev
-	chmod 755 dev
-	(cd dev && $(MAKEDEV) std ptmx fd)
-	tar cf - dev | gzip -9 >devices.tar.gz
-	rm -rf dev
