--- debootstrap/Makefile	2024-01-05 10:07:04.000000000 +0100
+++ debootstrap.new/Makefile	2024-01-16 11:10:37.880864509 +0100
@@ -8,10 +8,10 @@
 DSDIR=$(DESTDIR)/usr/share/debootstrap
 install:
 	mkdir -p $(DSDIR)/scripts
-	mkdir -p $(DESTDIR)/usr/sbin
+	mkdir -p $(DESTDIR)/usr/bin
 
 	cp -a scripts/* $(DSDIR)/scripts/
 	install -m 0644 functions $(DSDIR)/
 
-	sed 's/@VERSION@/$(VERSION)/g' debootstrap >$(DESTDIR)/usr/sbin/debootstrap
-	chmod 0755 $(DESTDIR)/usr/sbin/debootstrap
+	sed 's/@VERSION@/$(VERSION)/g' debootstrap >$(DESTDIR)/usr/bin/debootstrap
+	chmod 0755 $(DESTDIR)/usr/bin/debootstrap
