diff -Naur xmoto-0.1.11-orig/src/MotoGameL.cpp xmoto-0.1.11/src/MotoGameL.cpp
--- xmoto-0.1.11-orig/src/MotoGameL.cpp	2005-10-25 15:28:40.000000000 +0200
+++ xmoto-0.1.11/src/MotoGameL.cpp	2006-03-18 19:31:21.000000000 +0100
@@ -52,15 +52,15 @@
 
   int L_Game_PlaceInGameArrow(lua_State *pL) {
     m_pMotoGame->getArrowPointer().nArrowPointerMode = 1;
-    m_pMotoGame->getArrowPointer().ArrowPointerPos = Vector2f(luaL_check_number(pL,1),luaL_check_number(pL,2));
-    m_pMotoGame->getArrowPointer().fArrowPointerAngle = luaL_check_number(pL,3);
+    m_pMotoGame->getArrowPointer().ArrowPointerPos = Vector2f(luaL_checknumber(pL,1),luaL_checknumber(pL,2));
+    m_pMotoGame->getArrowPointer().fArrowPointerAngle = luaL_checknumber(pL,3);
     return 1;
   }
 
   int L_Game_PlaceScreenArrow(lua_State *pL) {
     m_pMotoGame->getArrowPointer().nArrowPointerMode = 2;
-    m_pMotoGame->getArrowPointer().ArrowPointerPos = Vector2f(luaL_check_number(pL,1),luaL_check_number(pL,2));
-    m_pMotoGame->getArrowPointer().fArrowPointerAngle = luaL_check_number(pL,3);
+    m_pMotoGame->getArrowPointer().ArrowPointerPos = Vector2f(luaL_checknumber(pL,1),luaL_checknumber(pL,2));
+    m_pMotoGame->getArrowPointer().fArrowPointerAngle = luaL_checknumber(pL,3);
     return 1;
   }
 
@@ -103,8 +103,8 @@
     for(int i=0;i<m_pMotoGame->getBlocks().size();i++) {
       ConvexBlock *pBlock = m_pMotoGame->getBlocks()[i];
       if(pBlock->pSrcBlock->ID == luaL_checkstring(pL,1)) {
-        pBlock->pSrcBlock->fPosX += luaL_check_number(pL,2);
-        pBlock->pSrcBlock->fPosY += luaL_check_number(pL,3);
+        pBlock->pSrcBlock->fPosX += luaL_checknumber(pL,2);
+        pBlock->pSrcBlock->fPosY += luaL_checknumber(pL,3);
         break;
       }
     }
@@ -131,8 +131,8 @@
     for(int i=0;i<m_pMotoGame->getBlocks().size();i++) {
       ConvexBlock *pBlock = m_pMotoGame->getBlocks()[i];
       if(pBlock->pSrcBlock->ID == luaL_checkstring(pL,1)) {
-        pBlock->pSrcBlock->fPosX = luaL_check_number(pL,2);
-        pBlock->pSrcBlock->fPosY = luaL_check_number(pL,3);
+        pBlock->pSrcBlock->fPosX = luaL_checknumber(pL,2);
+        pBlock->pSrcBlock->fPosY = luaL_checknumber(pL,3);
         break;
       }
     }
