diff -Naur mozjs17.0.0/js/src/config/milestone.pl mozjs17.0.0-fix/js/src/config/milestone.pl
--- mozjs17.0.0/js/src/config/milestone.pl	2013-02-11 23:33:22.000000000 +0100
+++ mozjs17.0.0-fix/js/src/config/milestone.pl	2015-12-24 04:03:33.138709959 +0100
@@ -55,7 +55,7 @@
 #
 my $milestone = Moz::Milestone::getOfficialMilestone($MILESTONE_FILE);
 
-if (defined(@TEMPLATE_FILE)) {
+if (@TEMPLATE_FILE) {
   my $TFILE;
 
   foreach $TFILE (@TEMPLATE_FILE) {
diff -Naur mozjs17.0.0/js/src/js-config.in mozjs17.0.0-fix/js/src/js-config.in
--- mozjs17.0.0/js/src/js-config.in	2013-03-25 21:34:20.000000000 +0100
+++ mozjs17.0.0-fix/js/src/js-config.in	2015-12-24 04:03:07.978075592 +0100
@@ -2,7 +2,7 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#filter substitution
+%filter substitution
 
 prefix='@prefix@'
 mozilla_version='@MOZILLA_VERSION@'
diff -Naur mozjs17.0.0/js/src/jsval.h mozjs17.0.0-fix/js/src/jsval.h
--- mozjs17.0.0/js/src/jsval.h	2013-03-25 21:34:20.000000000 +0100
+++ mozjs17.0.0-fix/js/src/jsval.h	2015-12-24 04:02:38.957343904 +0100
@@ -304,7 +304,6 @@
             int32_t        i32;
             uint32_t       u32;
             JSWhyMagic     why;
-            jsuword        word;
         } payload;
     } s;
     double asDouble;
diff -Naur mozjs17.0.0/js/src/Makefile.in mozjs17.0.0-fix/js/src/Makefile.in
--- mozjs17.0.0/js/src/Makefile.in	2013-03-25 21:34:20.000000000 +0100
+++ mozjs17.0.0-fix/js/src/Makefile.in	2015-12-24 04:05:31.782701334 +0100
@@ -778,7 +778,7 @@
 
 $(JS_CONFIG_NAME): js-config.in Makefile $(DEPTH)/config/autoconf.mk $(topsrcdir)/config/config.mk $(topsrcdir)/config/rules.mk
 	$(RM) $@.tmp
-	$(PYTHON) $(topsrcdir)/config/Preprocessor.py $(JS_CONFIG_SUBSTITUTIONS) $< > $@.tmp \
+	$(PYTHON) $(topsrcdir)/config/Preprocessor.py --marker="%" $(JS_CONFIG_SUBSTITUTIONS) $< > $@.tmp \
 	&& mv $@.tmp $@ && chmod +x $@
 
 SCRIPTS = $(JS_CONFIG_NAME)
@@ -788,7 +788,7 @@
 	$(PYTHON) $(topsrcdir)/config/Preprocessor.py $(JS_CONFIG_SUBSTITUTIONS) $< > $@
 
 install:: $(LIBRARY_NAME).pc
-		$(SYSINSTALL) $^ $(DESTDIR)$(libdir)/pkgconfig
+		$(SYSINSTALL) -m 0644 $^ $(DESTDIR)$(libdir)/pkgconfig
 
 ######################################################
 # BEGIN SpiderMonkey header installation
@@ -836,19 +836,19 @@
 #
 
 install:: $(INSTALLED_HEADERS)
-	$(SYSINSTALL) $^ $(DESTDIR)$(includedir)/$(MODULE)
+	$(SYSINSTALL) -m 0644 $^ $(DESTDIR)$(includedir)/$(MODULE)
 
 install:: $(EXPORTS_ds)
-	$(SYSINSTALL) $^ $(DESTDIR)$(includedir)/$(MODULE)/ds
+	$(SYSINSTALL) -m 0644 $^ $(DESTDIR)$(includedir)/$(MODULE)/ds
 
 install:: $(EXPORTS_gc)
-	$(SYSINSTALL) $^ $(DESTDIR)$(includedir)/$(MODULE)/gc
+	$(SYSINSTALL) -m 0644 $^ $(DESTDIR)$(includedir)/$(MODULE)/gc
 
 install:: $(EXPORTS_js)
-	$(SYSINSTALL) $^ $(DESTDIR)$(includedir)/$(MODULE)/js
+	$(SYSINSTALL) -m 0644 $^ $(DESTDIR)$(includedir)/$(MODULE)/js
 
 install:: $(EXPORTS_mozilla)
-	$(SYSINSTALL) $^ $(DESTDIR)$(includedir)/$(MODULE)/mozilla
+	$(SYSINSTALL) -m 0644 $^ $(DESTDIR)$(includedir)/$(MODULE)/mozilla
 
 #
 # END SpiderMonkey header installation
