diff -Naur guzuta-0.0.4.orig/Makefile guzuta-0.0.4/Makefile
--- guzuta-0.0.4.orig/Makefile	2005-10-08 17:25:31.000000000 +0200
+++ guzuta-0.0.4/Makefile	2006-10-16 22:12:21.000000000 +0200
@@ -3,7 +3,7 @@
 
 install: trayicon.so
 	python ./setup.py install --root=$(DESTDIR)
-	cp src/trayicon/build/lib*/trayicon.so $(DESTDIR)/usr/lib/python2.4/site-packages/libguzuta/
+	cp src/trayicon/build/lib*/trayicon.so $(DESTDIR)/usr/lib/python$(shell python -c 'from distutils import sysconfig; print sysconfig.get_python_version()')/site-packages/libguzuta/
 
 clean:
 	cd src/trayicon/ && make clean
