--- crystalhd/driver/linux/crystalhd_misc.h	2015-10-08 12:23:52.000000000 +0200
+++ crystalhd.new/driver/linux/crystalhd_misc.h	2015-10-08 12:25:25.000000000 +0200
@@ -73,7 +73,7 @@
 	uint32_t		y_done_sz;
 	uint32_t		uv_done_sz;
 	uint32_t		comp_flags;
-	bool			b422mode;
+	uint32_t		b422mode;
 };
 
 struct crystalhd_dio_req {
--- crystalhd/driver/linux/crystalhd_cmds.c	2015-10-08 12:23:52.000000000 +0200
+++ crystalhd.new/driver/linux/crystalhd_cmds.c	2015-10-08 12:31:30.000000000 +0200
@@ -85,7 +85,7 @@
 		return BC_STS_ERR_USAGE;
 	}
 
-	if ((idata->udata.u.NotifyMode.Mode && 0xFF) == DTS_MONITOR_MODE) {
+	if ((idata->udata.u.NotifyMode.Mode & 0xFF) == DTS_MONITOR_MODE) {
 		ctx->user[idata->u_id].mode = idata->udata.u.NotifyMode.Mode;
 		return BC_STS_SUCCESS;
 	}
