From 536003a1623e9bae8f2eca2d0a6fef66bea3a7fb Mon Sep 17 00:00:00 2001
From: marlemion <root@wetab.fritz.box>
Date: Tue, 1 Sep 2015 19:14:49 +0200
Subject: [PATCH] Making it compile for kernel >4.0

---
 driver/linux/crystalhd_lnx.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/driver/linux/crystalhd_lnx.c b/driver/linux/crystalhd_lnx.c
index ea5587a..ceb5c18 100644
--- a/driver/linux/crystalhd_lnx.c
+++ b/driver/linux/crystalhd_lnx.c
@@ -537,13 +537,6 @@ static int __init chd_pci_reserve_mem(struct crystalhd_adp *pinfo)
 	dev_dbg(dev, "bar0:0x%x-0x%08x  bar2:0x%x-0x%08x\n",
 	        bar0, i2o_len, bar2, mem_len);
 
-	/* bar-0 */
-	rc = check_mem_region(bar0, i2o_len);
-	if (rc) {
-		printk(KERN_ERR "No valid mem region...\n");
-		return -ENOMEM;
-	}
-
 	pinfo->i2o_addr = ioremap_nocache(bar0, i2o_len);
 	if (!pinfo->i2o_addr) {
 		printk(KERN_ERR "Failed to remap i2o region...\n");
@@ -553,13 +546,6 @@ static int __init chd_pci_reserve_mem(struct crystalhd_adp *pinfo)
 	pinfo->pci_i2o_start = bar0;
 	pinfo->pci_i2o_len   = i2o_len;
 
-	/* bar-2 */
-	rc = check_mem_region(bar2, mem_len);
-	if (rc) {
-		printk(KERN_ERR "No valid mem region...\n");
-		return -ENOMEM;
-	}
-
 	pinfo->mem_addr = ioremap_nocache(bar2, mem_len);
 	if (!pinfo->mem_addr) {
 		printk(KERN_ERR "Failed to remap mem region...\n");
