diff -ur a/driver/linux/crystalhd_hw.h b/driver/linux/crystalhd_hw.h
--- a/driver/linux/crystalhd_hw.h	2012-05-23 00:35:50.387916633 -0500
+++ b/driver/linux/crystalhd_hw.h	2012-05-23 00:42:37.071901497 -0500
@@ -30,6 +30,7 @@
 
 #include <linux/device.h>
 #include <linux/version.h>
+#include <linux/delay.h>
 #if LINUX_VERSION_CODE > KERNEL_VERSION(2, 6, 24)
 #include <linux/semaphore.h>
 #else
diff -ur a/driver/linux/crystalhd_lnx.h b/driver/linux/crystalhd_lnx.h
--- a/driver/linux/crystalhd_lnx.h	2012-05-23 00:35:50.387916633 -0500
+++ b/driver/linux/crystalhd_lnx.h	2012-05-23 00:41:17.408904475 -0500
@@ -45,7 +45,6 @@
 #include <linux/io.h>
 #include <asm/irq.h>
 #include <asm/pgtable.h>
-#include <asm/system.h>
 #include <linux/uaccess.h>
 
 #include "crystalhd_cmds.h"
diff -ur a/driver/linux/crystalhd_misc.h b/driver/linux/crystalhd_misc.h
--- a/driver/linux/crystalhd_misc.h	2012-05-23 00:35:50.387916633 -0500
+++ b/driver/linux/crystalhd_misc.h	2012-05-23 00:41:17.410904475 -0500
@@ -35,7 +35,6 @@
 #include <linux/ioctl.h>
 #include <linux/dma-mapping.h>
 #include <linux/sched.h>
-#include <asm/system.h>
 #include "bc_dts_glob_lnx.h"
 #include "crystalhd_hw.h"
 
