diff -Naur blender-3.4.1/source/blender/blenkernel/intern/writeffmpeg.c blender-3.4.1.new/source/blender/blenkernel/intern/writeffmpeg.c
--- blender-3.4.1/source/blender/blenkernel/intern/writeffmpeg.c	2022-11-08 14:50:16.000000000 +0100
+++ blender-3.4.1.new/source/blender/blenkernel/intern/writeffmpeg.c	2023-03-02 15:54:04.149681397 +0100
@@ -855,12 +855,16 @@
                                                             255);
   st->avg_frame_rate = av_inv_q(c->time_base);
 
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(59, 18, 100)
   if (codec->capabilities & AV_CODEC_CAP_AUTO_THREADS) {
     c->thread_count = 0;
   }
   else {
     c->thread_count = BLI_system_thread_count();
   }
+#else
+  c->thread_count = BLI_system_thread_count();
+#endif
 
   if (codec->capabilities & AV_CODEC_CAP_FRAME_THREADS) {
     c->thread_type = FF_THREAD_FRAME;
diff -Naur blender-3.4.1/source/blender/imbuf/intern/anim_movie.c blender-3.4.1.new/source/blender/imbuf/intern/anim_movie.c
--- blender-3.4.1/source/blender/imbuf/intern/anim_movie.c	2022-11-04 01:33:07.000000000 +0100
+++ blender-3.4.1.new/source/blender/imbuf/intern/anim_movie.c	2023-03-02 15:54:57.358107044 +0100
@@ -554,12 +554,16 @@
   avcodec_parameters_to_context(pCodecCtx, video_stream->codecpar);
   pCodecCtx->workaround_bugs = FF_BUG_AUTODETECT;
 
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(59, 18, 100)
   if (pCodec->capabilities & AV_CODEC_CAP_AUTO_THREADS) {
     pCodecCtx->thread_count = 0;
   }
   else {
     pCodecCtx->thread_count = BLI_system_thread_count();
   }
+#else
+  pCodecCtx->thread_count = BLI_system_thread_count(); 
+#endif
 
   if (pCodec->capabilities & AV_CODEC_CAP_FRAME_THREADS) {
     pCodecCtx->thread_type = FF_THREAD_FRAME;
diff -Naur blender-3.4.1/source/blender/imbuf/intern/indexer.c blender-3.4.1.new/source/blender/imbuf/intern/indexer.c
--- blender-3.4.1/source/blender/imbuf/intern/indexer.c	2022-11-04 01:33:07.000000000 +0100
+++ blender-3.4.1.new/source/blender/imbuf/intern/indexer.c	2023-03-02 15:57:28.023311195 +0100
@@ -559,12 +559,16 @@
   av_dict_set(&codec_opts, "preset", "veryfast", 0);
   av_dict_set(&codec_opts, "tune", "fastdecode", 0);
 
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(59, 18, 100)
   if (rv->codec->capabilities & AV_CODEC_CAP_AUTO_THREADS) {
     rv->c->thread_count = 0;
   }
   else {
     rv->c->thread_count = BLI_system_thread_count();
   }
+#else
+  rv->c->thread_count = BLI_system_thread_count();
+#endif
 
   if (rv->codec->capabilities & AV_CODEC_CAP_FRAME_THREADS) {
     rv->c->thread_type = FF_THREAD_FRAME;
@@ -872,12 +876,16 @@
   avcodec_parameters_to_context(context->iCodecCtx, context->iStream->codecpar);
   context->iCodecCtx->workaround_bugs = FF_BUG_AUTODETECT;
 
+#if LIBAVCODEC_VERSION_INT < AV_VERSION_INT(59, 18, 100)
   if (context->iCodec->capabilities & AV_CODEC_CAP_AUTO_THREADS) {
     context->iCodecCtx->thread_count = 0;
   }
   else {
     context->iCodecCtx->thread_count = BLI_system_thread_count();
   }
+#else
+  context->iCodecCtx->thread_count = BLI_system_thread_count();
+#endif
 
   if (context->iCodec->capabilities & AV_CODEC_CAP_FRAME_THREADS) {
     context->iCodecCtx->thread_type = FF_THREAD_FRAME;
