--- blender-4.4.3/CMakeLists.txt	2025-03-06 01:31:08.000000000 +0100
+++ blender-4.4.3.new/CMakeLists.txt	2025-06-16 12:03:00.797300657 +0200
@@ -2519,7 +2519,7 @@
     # Always use numpy bundled in precompiled libs.
   elseif((WITH_PYTHON_INSTALL AND WITH_PYTHON_INSTALL_NUMPY) OR WITH_PYTHON_NUMPY)
     if(("${PYTHON_NUMPY_PATH}" STREQUAL "") OR (${PYTHON_NUMPY_PATH} MATCHES NOTFOUND))
-      find_python_package(numpy "core/include")
+      find_python_package(numpy "_core/include")
     endif()
   endif()
 
