diff -Naur blender-2.42a/source/gameengine/Expressions/ListValue.cpp blender-2.42a-p/source/gameengine/Expressions/ListValue.cpp
--- blender-2.42a/source/gameengine/Expressions/ListValue.cpp	2004-06-07 13:03:11.000000000 +0200
+++ blender-2.42a-p/source/gameengine/Expressions/ListValue.cpp	2007-02-13 15:16:10.000000000 +0100
@@ -163,22 +163,22 @@
 
 
 static PySequenceMethods listvalue_as_sequence = {
-	(inquiry)listvalue_bufferlen,//(inquiry)buffer_length, /*sq_length*/
+	(lenfunc)listvalue_bufferlen,/*sq_length*/
 	(binaryfunc)listvalue_buffer_concat, /*sq_concat*/
-	0,//(intargfunc)buffer_repeat, /*sq_repeat*/
-	(intargfunc)listvalue_buffer_item, /*sq_item*/
-	(intintargfunc)listvalue_buffer_slice, /*sq_slice*/
-	0,//(intobjargproc)buffer_ass_item, /*sq_ass_item*/
-	0,//(intintobjargproc)buffer_ass_slice, /*sq_ass_slice*/
+	0,/*sq_repeat*/
+	(ssizeargfunc)listvalue_buffer_item, /*sq_item*/
+	(ssizessizeargfunc)listvalue_buffer_slice, /*sq_slice*/
+	0,/*sq_ass_item*/
+	0,/*sq_ass_slice*/
 };
 
 
 
 /* Is this one used ? */
 static PyMappingMethods instance_as_mapping = {
-	(inquiry)listvalue_bufferlen, /*mp_length*/
+	(lenfunc)listvalue_bufferlen, /*mp_length*/
 	(binaryfunc)listvalue_mapping_subscript, /*mp_subscript*/
-	0,//(objobjargproc)instance_ass_subscript, /*mp_ass_subscript*/
+	0,/*mp_ass_subscript*/
 };
 
 
diff -Naur blender-2.42a/source/gameengine/Expressions/Value.cpp blender-2.42a-p/source/gameengine/Expressions/Value.cpp
--- blender-2.42a/source/gameengine/Expressions/Value.cpp	2006-01-06 04:46:52.000000000 +0100
+++ blender-2.42a-p/source/gameengine/Expressions/Value.cpp	2007-02-13 15:16:10.000000000 +0100
@@ -649,8 +649,8 @@
 		CListValue* listval = new CListValue();
 		bool error = false;
 
-		int i;
-		int numitems = PyList_Size(pyobj);
+		Py_ssize_t i;
+		Py_ssize_t numitems = PyList_Size(pyobj);
 		for (i=0;i<numitems;i++)
 		{
 			PyObject* listitem = PyList_GetItem(pyobj,i); /* borrowed ref */
