diff -Naur blender-2.42a/source/gameengine/Expressions/Value.cpp blender-2.42a-p/source/gameengine/Expressions/Value.cpp
--- blender-2.42a/source/gameengine/Expressions/Value.cpp	2006-01-06 04:46:52.000000000 +0100
+++ blender-2.42a-p/source/gameengine/Expressions/Value.cpp	2007-02-13 15:16:10.000000000 +0100
@@ -649,8 +649,8 @@
 		CListValue* listval = new CListValue();
 		bool error = false;
 
-		int i;
-		int numitems = PyList_Size(pyobj);
+		Py_ssize_t i;
+		Py_ssize_t numitems = PyList_Size(pyobj);
 		for (i=0;i<numitems;i++)
 		{
 			PyObject* listitem = PyList_GetItem(pyobj,i); /* borrowed ref */
