diff -Naur vokoscreenNG-3.0.4/src/region/QvkRegionButtonArrow.cpp vokoscreenNG-3.0.4.new/src/region/QvkRegionButtonArrow.cpp
--- vokoscreenNG-3.0.4/src/region/QvkRegionButtonArrow.cpp	2020-04-29 18:11:07.000000000 +0200
+++ vokoscreenNG-3.0.4.new/src/region/QvkRegionButtonArrow.cpp	2020-06-19 12:37:02.240590905 +0200
@@ -22,6 +22,8 @@
 
 #include "QvkRegionButtonArrow.h"
 
+#include <QPainterPath>
+
 QvkRegionButtonArrow::QvkRegionButtonArrow()
 {
 }
