diff -Naur kdebluetooth-20060830/kdebluetooth/kbluetoothd/kcm_btpaired/pairedtab.cpp kdebluetooth-20060830-p/kdebluetooth/kbluetoothd/kcm_btpaired/pairedtab.cpp
--- kdebluetooth-20060830/kdebluetooth/kbluetoothd/kcm_btpaired/pairedtab.cpp	2006-04-22 04:02:57.000000000 +0200
+++ kdebluetooth-20060830-p/kdebluetooth/kbluetoothd/kcm_btpaired/pairedtab.cpp	2006-08-30 14:12:48.000000000 +0200
@@ -58,8 +58,8 @@
     KConfig* config = KApplication::kApplication()->config();
     linkKeyFilename = config->readEntry("linkKeyFile", "/etc/bluetooth/link_key");
     config->writeEntry("linkKeyFile", linkKeyFilename);
-    hcidStartCommand = config->readEntry("hcidStartCommand", "/etc/init.d/bluez-utils start");
-    hcidStopCommand = config->readEntry("hcidStopCommand", "/etc/init.d/bluez-utils stop");
+    hcidStartCommand = config->readEntry("hcidStartCommand", "/etc/rc.d/bluez-utils start");
+    hcidStopCommand = config->readEntry("hcidStopCommand", "/etc/rc.d/bluez-utils stop");
     bDirty = false;
     linkKeyFileValid = false;
     reloadList();
