diff -Naur kdebluetooth-1.0_beta8/kdebluetooth/kbluetooth/sdprecord.cpp kdebluetooth-1.0_beta8-p/kdebluetooth/kbluetooth/sdprecord.cpp
--- kdebluetooth-1.0_beta8/kdebluetooth/kbluetooth/sdprecord.cpp	2007-09-17 18:52:30.000000000 +0200
+++ kdebluetooth-1.0_beta8-p/kdebluetooth/kbluetooth/sdprecord.cpp	2008-03-15 08:50:41.000000000 +0100
@@ -16,6 +16,7 @@
 #include <bluetooth/sdp_lib.h>
 #include <kmessagebox.h>
 #include <klocale.h>
+#include <cstdlib>
 
 namespace Bluetooth
 {
diff -Naur kdebluetooth-1.0_beta8/kdebluetooth/kcm_btpaired/pairedtab.h kdebluetooth-1.0_beta8-p/kdebluetooth/kcm_btpaired/pairedtab.h
--- kdebluetooth-1.0_beta8/kdebluetooth/kcm_btpaired/pairedtab.h	2007-09-17 18:52:30.000000000 +0200
+++ kdebluetooth-1.0_beta8-p/kdebluetooth/kcm_btpaired/pairedtab.h	2008-03-15 08:50:54.000000000 +0100
@@ -29,6 +29,7 @@
 #define PAIREDTAB_H
 
 #include "pairedtabbase.h"
+#include <stdint.h>
 #include <qcstring.h>
 #include <qdatastream.h>
 #include <qstringlist.h>
diff -Naur kdebluetooth-1.0_beta8/kdebluetooth/libkbluetooth/sdpdevice.cpp kdebluetooth-1.0_beta8-p/kdebluetooth/libkbluetooth/sdpdevice.cpp
--- kdebluetooth-1.0_beta8/kdebluetooth/libkbluetooth/sdpdevice.cpp	2007-09-17 18:52:31.000000000 +0200
+++ kdebluetooth-1.0_beta8-p/kdebluetooth/libkbluetooth/sdpdevice.cpp	2008-03-15 08:50:54.000000000 +0100
@@ -16,6 +16,7 @@
 #include <bluetooth/bluetooth.h>
 #include <bluetooth/sdp.h>
 #include <bluetooth/sdp_lib.h>
+#include <cstdlib>
 
 using namespace std;
 
diff -Naur kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobexbase.cpp kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobexbase.cpp
--- kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobexbase.cpp	2007-09-17 18:52:31.000000000 +0200
+++ kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobexbase.cpp	2008-03-15 08:50:54.000000000 +0100
@@ -23,6 +23,7 @@
 
 #include "qobexbase.h"
 
+#undef myDebug
 #undef DEBUG
 // #define DEBUG
 #ifdef DEBUG
diff -Naur kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobexclient.cpp kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobexclient.cpp
--- kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobexclient.cpp	2007-09-17 18:52:31.000000000 +0200
+++ kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobexclient.cpp	2008-03-15 08:50:54.000000000 +0100
@@ -31,6 +31,7 @@
 
 #include "qobexclient.h"
 
+#undef myDebug
 //#undef DEBUG
 #define DEBUG
 #ifdef DEBUG
diff -Naur kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobexericssontransport.cpp kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobexericssontransport.cpp
--- kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobexericssontransport.cpp	2007-09-17 18:52:31.000000000 +0200
+++ kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobexericssontransport.cpp	2008-03-15 08:50:54.000000000 +0100
@@ -31,6 +31,7 @@
 
 #include "qobexericssontransport.h"
 
+#undef myDebug
 #undef DEBUG
 #define DEBUG
 #ifdef DEBUG
diff -Naur kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobexserialtransport.cpp kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobexserialtransport.cpp
--- kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobexserialtransport.cpp	2007-09-17 18:52:31.000000000 +0200
+++ kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobexserialtransport.cpp	2008-03-15 08:50:54.000000000 +0100
@@ -31,6 +31,7 @@
 
 #include "qobexserialtransport.h"
 
+#undef myDebug
 #undef DEBUG
 // #define DEBUG
 #ifdef DEBUG
diff -Naur kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobexserverconnection.cpp kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobexserverconnection.cpp
--- kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobexserverconnection.cpp	2007-09-17 18:52:31.000000000 +0200
+++ kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobexserverconnection.cpp	2008-03-15 08:50:54.000000000 +0100
@@ -42,6 +42,7 @@
 #include "qobexnullserverops.h"
 #include "qobexserverops.h"
 
+#undef myDebug
 #undef DEBUG
 #define DEBUG
 #ifdef DEBUG
diff -Naur kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobexserver.cpp kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobexserver.cpp
--- kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobexserver.cpp	2007-09-17 18:52:31.000000000 +0200
+++ kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobexserver.cpp	2008-03-15 08:50:54.000000000 +0100
@@ -31,6 +31,7 @@
 
 #include "qobexserver.h"
 
+#undef myDebug
 #undef DEBUG
 // #define DEBUG
 #ifdef DEBUG
diff -Naur kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobexserverops.cpp kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobexserverops.cpp
--- kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobexserverops.cpp	2007-09-17 18:52:31.000000000 +0200
+++ kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobexserverops.cpp	2008-03-15 08:50:54.000000000 +0100
@@ -28,6 +28,7 @@
 #include "qobexserverops.h"
 #include "qobexauth.h"
 
+#undef myDebug
 #undef DEBUG
 // #define DEBUG
 #ifdef DEBUG
diff -Naur kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobextransport.cpp kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobextransport.cpp
--- kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qobextransport.cpp	2007-09-17 18:52:31.000000000 +0200
+++ kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qobextransport.cpp	2008-03-15 08:50:54.000000000 +0100
@@ -46,6 +46,7 @@
 
 #include "qobextransport.h"
 
+#undef myDebug
 //#undef DEBUG
 #define DEBUG
 #ifdef DEBUG
diff -Naur kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qserialdevice.cpp kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qserialdevice.cpp
--- kdebluetooth-1.0_beta8/kdebluetooth/libqobex/qobex/qserialdevice.cpp	2007-09-17 18:52:31.000000000 +0200
+++ kdebluetooth-1.0_beta8-p/kdebluetooth/libqobex/qobex/qserialdevice.cpp	2008-03-15 08:50:54.000000000 +0100
@@ -43,6 +43,7 @@
 #include "qttylock.h"
 #include "qserialdevice.h"
 
+#undef myDebug
 #undef DEBUG
 // #define DEBUG
 #ifdef DEBUG
