diff -Naur kdebluetooth-1.0-beta3/kdebluetooth/kioslave/sdp/btsdp.cpp kdebluetooth-1.0-beta3-p/kdebluetooth/kioslave/sdp/btsdp.cpp
--- kdebluetooth-1.0-beta3/kdebluetooth/kioslave/sdp/btsdp.cpp	2007-06-12 12:29:42.000000000 +0200
+++ kdebluetooth-1.0-beta3-p/kdebluetooth/kioslave/sdp/btsdp.cpp	2007-06-20 01:23:16.000000000 +0200
@@ -246,7 +246,7 @@
             // save dialog. In konqueror, kbtobexclient is started.
             // In the file save dialog, kio_obex will be used.
             KURL obexPushURL;
-            obexPushURL.setProtocol("obex");
+            obexPushURL.setProtocol("obex2");
             obexPushURL.setHost(mAddress);
             if (record.rfcommChannel() > 0) {
                 obexPushURL.setPort(record.rfcommChannel());
