--- cantata-2.0.0/po/CMakeLists.txt	2016-01-24 16:31:36.000000000 +0100
+++ cantata-2.0.0.new/po/CMakeLists.txt	2016-04-04 12:23:54.000000000 +0200
@@ -35,8 +35,8 @@
         endforeach (_lang ${CANTATA_LANGUAGES})
     endif (NOT GETTEXT_MSGFMT_EXECUTABLE)
 else (ENABLE_KDE_SUPPORT)
-    find_program(LCONVERT_EXECUTABLE lconvert)
-    find_program(LRELEASE_EXECUTABLE lrelease)
+    find_program(LCONVERT_EXECUTABLE lconvert-qt5)
+    find_program(LRELEASE_EXECUTABLE lrelease-qt5)
 
     if (NOT LCONVERT_EXECUTABLE OR NOT LRELEASE_EXECUTABLE)
         message("------\n"
