--- async-2.2.5/async.cabal	2001-09-09 03:46:40.000000000 +0200
+++ async-2.2.5.new/async.cabal	2024-07-10 22:34:37.460693185 +0200
@@ -35,9 +35,10 @@
 homepage:            https://github.com/simonmar/async
 bug-reports:         https://github.com/simonmar/async/issues
 tested-with:
-    GHC == 9.8.1
-    GHC == 9.6.3
-    GHC == 9.4.7
+    GHC == 9.10.0
+    GHC == 9.8.2
+    GHC == 9.6.4
+    GHC == 9.4.8
     GHC == 9.2.8
     GHC == 9.0.2
     GHC == 8.10.7
@@ -69,8 +70,8 @@
         other-extensions: Trustworthy
     exposed-modules:     Control.Concurrent.Async
                          Control.Concurrent.Async.Internal
-    build-depends:       base     >= 4.3     && < 4.20,
-                         hashable >= 1.1.2.0 && < 1.5,
+    build-depends:       base     >= 4.3     && < 4.21,
+                         hashable >= 1.1.2.0 && < 1.6,
                          stm      >= 2.2     && < 2.6
 
 test-suite test-async
