--- async-2.2.4/async.cabal	2001-09-09 03:46:40.000000000 +0200
+++ async-2.2.4.new/async.cabal	2022-11-16 15:54:06.853241677 +0100
@@ -35,9 +35,10 @@
 homepage:            https://github.com/simonmar/async
 bug-reports:         https://github.com/simonmar/async/issues
 tested-with:
-    GHC == 9.2.0.20210821
-    GHC == 9.0.1
-    GHC == 8.10.4
+    GHC == 9.4.1
+    GHC == 9.2.4
+    GHC == 9.0.2
+    GHC == 8.10.7
     GHC == 8.8.4
     GHC == 8.6.5
     GHC == 8.4.4
@@ -64,8 +65,8 @@
     if impl(ghc>=7.1)
         other-extensions: Trustworthy
     exposed-modules:     Control.Concurrent.Async
-    build-depends:       base     >= 4.3     && < 4.17,
-                         hashable >= 1.1.2.0 && < 1.4,
+    build-depends:       base     >= 4.3     && < 4.18,
+                         hashable >= 1.1.2.0 && < 1.5,
                          stm      >= 2.2     && < 2.6
 
 test-suite test-async
@@ -73,7 +74,7 @@
     type:       exitcode-stdio-1.0
     hs-source-dirs: test
     main-is:    test-async.hs
-    build-depends: base >= 4.3 && < 4.17,
+    build-depends: base,
                    async,
                    stm,
                    test-framework,
