diff -Naur vtun-3.0.4/cfg_file.y vtun-3.0.4-p/cfg_file.y
--- vtun-3.0.4/cfg_file.y	2016-10-01 23:27:51.000000000 +0200
+++ vtun-3.0.4-p/cfg_file.y	2016-12-24 12:51:31.488590055 +0100
@@ -610,7 +610,7 @@
 }
 
 /* Clear the VTUN_NAT_HACK flag which are not relevant to the current operation mode */
-inline void clear_nat_hack_flags(int svr)
+void clear_nat_hack_flags(int svr)
 {
 	if (svr)
 		llist_trav(&host_list,clear_nat_hack_server,NULL);
diff -Naur vtun-3.0.4/generic/pty_dev.c vtun-3.0.4-p/generic/pty_dev.c
--- vtun-3.0.4/generic/pty_dev.c	2016-10-01 23:37:39.000000000 +0200
+++ vtun-3.0.4-p/generic/pty_dev.c	2016-12-24 13:12:16.607608511 +0100
@@ -22,9 +22,10 @@
 
 #include "config.h"
 
+#define _GNU_SOURCE
+#include <stdlib.h>
 #include <unistd.h>
 #include <fcntl.h>
-#include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
 #include <syslog.h>
diff -Naur vtun-3.0.4/lfd_encrypt.c vtun-3.0.4-p/lfd_encrypt.c
--- vtun-3.0.4/lfd_encrypt.c	2016-10-01 23:27:51.000000000 +0200
+++ vtun-3.0.4-p/lfd_encrypt.c	2016-12-24 12:54:21.532592575 +0100
@@ -44,6 +44,7 @@
 #include <strings.h>
 #include <string.h>
 #include <time.h>
+#include <arpa/inet.h>
 
 #include "vtun.h"
 #include "linkfd.h"
diff -Naur vtun-3.0.4/lfd_shaper.c vtun-3.0.4-p/lfd_shaper.c
--- vtun-3.0.4/lfd_shaper.c	2016-10-01 23:27:51.000000000 +0200
+++ vtun-3.0.4-p/lfd_shaper.c	2016-12-24 12:56:33.894594537 +0100
@@ -27,6 +27,7 @@
 #include <stdlib.h>
 #include <sys/time.h>
 #include <syslog.h>
+#include <time.h>
 
 #include "vtun.h"
 #include "linkfd.h"
diff -Naur vtun-3.0.4/lib.h vtun-3.0.4-p/lib.h
--- vtun-3.0.4/lib.h	2016-10-01 23:27:51.000000000 +0200
+++ vtun-3.0.4-p/lib.h	2016-12-24 13:02:02.569599409 +0100
@@ -26,6 +26,8 @@
 #include <sys/types.h>
 #include <signal.h>
 #include <errno.h>
+#include <stdio.h>
+#include <unistd.h>
 
 #ifdef HAVE_LIBUTIL_H
 #include <libutil.h>
diff -Naur vtun-3.0.4/lock.c vtun-3.0.4-p/lock.c
--- vtun-3.0.4/lock.c	2016-10-01 23:27:51.000000000 +0200
+++ vtun-3.0.4-p/lock.c	2016-12-24 12:55:56.168593978 +0100
@@ -32,6 +32,7 @@
 #include <sys/types.h>
 #include <signal.h>
 #include <errno.h>
+#include <time.h>
 
 #include "vtun.h"
 #include "linkfd.h"
diff -Naur vtun-3.0.4/vtun.h vtun-3.0.4-p/vtun.h
--- vtun-3.0.4/vtun.h	2016-10-01 23:27:51.000000000 +0200
+++ vtun-3.0.4-p/vtun.h	2016-12-24 12:55:07.426593256 +0100
@@ -232,6 +232,6 @@
 int  tunnel(struct vtun_host *host);
 int  read_config(char *file);
 struct vtun_host * find_host(char *host);
-inline void clear_nat_hack_flags(int svr);
+void clear_nat_hack_flags(int svr);
 
 #endif
