diff -Naur cyrus-sasl-2.1.26/configure.in cyrus-sasl-2.1.26-patch1/configure.in
--- cyrus-sasl-2.1.26/configure.in	2012-10-12 16:05:48.000000000 +0200
+++ cyrus-sasl-2.1.26-patch1/configure.in	2015-12-29 20:40:13.369799398 +0100
@@ -861,9 +861,9 @@
     notfound) AC_WARN([SQLite Library not found]); true;;
     *)
      if test -d ${with_sqlite}/lib; then
-         LIB_SQLITE="-L${with_sqlite}/lib -R${with_sqlite}/lib"
+         LIB_SQLITE="-L${with_sqlite}/lib"
      else
-         LIB_SQLITE="-L${with_sqlite} -R${with_sqlite}"
+         LIB_SQLITE="-L${with_sqlite}"
      fi
 
      LIB_SQLITE_DIR=$LIB_SQLITE
@@ -913,9 +913,9 @@
     notfound) AC_WARN([SQLite3 Library not found]); true;;
     *)
      if test -d ${with_sqlite3}/lib; then
-         LIB_SQLITE3="-L${with_sqlite3}/lib -R${with_sqlite3}/lib"
+         LIB_SQLITE3="-L${with_sqlite3}/lib"
      else
-         LIB_SQLITE3="-L${with_sqlite3} -R${with_sqlite3}"
+         LIB_SQLITE3="-L${with_sqlite3}"
      fi
 
      LIB_SQLITE3_DIR=$LIB_SQLITE3
@@ -1412,7 +1412,7 @@
 #endif /* CONFIG_H */
 ])
 
-AM_CONFIG_HEADER(config.h)
+AC_CONFIG_HEADERS(config.h)
 
 AC_OUTPUT(Makefile
 libsasl2.pc
