diff -up tar-1.19/lib/argp-fmtstream.h.gcc43 tar-1.19/lib/argp-fmtstream.h
--- tar-1.19/lib/argp-fmtstream.h.gcc43	2007-09-28 15:11:36.000000000 +0200
+++ tar-1.19/lib/argp-fmtstream.h	2008-02-13 16:56:58.000000000 +0100
@@ -198,7 +198,11 @@ extern int __argp_fmtstream_ensure (argp
 #endif
 
 #ifndef ARGP_FS_EI
-#define ARGP_FS_EI extern inline
+# ifdef __GNUC_STDC_INLINE__
+#  define ARGP_FS_EI extern inline __attribute__((__gnu_inline__))
+# else
+#  define ARGP_FS_EI extern inline
+# endif
 #endif
 
 ARGP_FS_EI size_t
diff -up tar-1.19/lib/argp.h.gcc43 tar-1.19/lib/argp.h
--- tar-1.19/lib/argp.h.gcc43	2007-09-28 15:11:36.000000000 +0200
+++ tar-1.19/lib/argp.h	2008-02-13 16:55:36.000000000 +0100
@@ -580,7 +580,11 @@ extern void *__argp_input (const struct 
 # endif
 
 # ifndef ARGP_EI
-#  define ARGP_EI extern __inline__
+#  if defined __GNUC_STDC_INLINE__
+#    define ARGP_EI extern __inline__ __attribute__((__gnu_inline__))
+#  else
+#    define ARGP_EI extern __inline__
+#  endif
 # endif
 
 ARGP_EI void

