--- paco-2.0.6old/lib/paco/BasePkg.cc	2008-10-03 20:57:46.000000000 +0200
+++ paco-2.0.6/lib/paco/BasePkg.cc	2009-08-18 15:54:40.000000000 +0200
@@ -196,14 +196,14 @@
 	}
 	
 	char* p;
-	char* file;
+	const char* file;
 	enum { RAW, GZ, BZ2, NSIZES };
 	long size[NSIZES], sizeInst = 0, sizeMiss = 0, filesInst = 0, filesMiss = 0;
  	
 	do {
 		if (UNLIKELY(!(file = strchr(buf.c_str(), '/'))))
 			continue;
-		else if ((p = strchr(file, '|')))
+		else if ((p = (char*) strchr(file, '|')))
 			*p++ = 0;
 		if (getSize(size[RAW], file) +
 			getSize(size[GZ], file, ".gz") +
