diff -Nuapr mpeg4ip-1.6/lib/mp4v2/Makefile.am mpeg4ip-1.6-p/lib/mp4v2/Makefile.am
--- mpeg4ip-1.6/lib/mp4v2/Makefile.am	2007-07-24 22:32:28.000000000 +0200
+++ mpeg4ip-1.6-p/lib/mp4v2/Makefile.am	2007-09-20 15:07:34.000000000 +0200
@@ -1,4 +1,4 @@
-SUBDIRS = . test util
+SUBDIRS = .
 
 INCLUDES = -I$(top_srcdir)/include
 
@@ -6,7 +6,10 @@ AM_CXXFLAGS = @BILLS_CPPWARNINGS@
 
 lib_LTLIBRARIES = libmp4v2.la 
 
-include_HEADERS = mp4.h 
+include_HEADERS = mp4.h \
+		$(top_srcdir)/include/mpeg4ip.h \
+		$(top_srcdir)/include/mpeg4ip_version.h \
+		$(top_srcdir)/mpeg4ip_config.h 
 
 libmp4v2_la_SOURCES = \
 	3gp.cpp \

