diff -Naurp mpeg4ip-1.5.0.1/lib/mp4v2/Makefile.am mpeg4ip-1.5.0.1-p/lib/mp4v2/Makefile.am
--- mpeg4ip-1.5.0.1/lib/mp4v2/Makefile.am	2006-05-11 20:11:52.000000000 +0200
+++ mpeg4ip-1.5.0.1-p/lib/mp4v2/Makefile.am	2006-11-19 12:35:11.000000000 +0100
@@ -1,4 +1,4 @@
-SUBDIRS = . test util
+SUBDIRS = . 
 
 INCLUDES = -I$(top_srcdir)/include
 
@@ -6,7 +6,7 @@ AM_CXXFLAGS = @BILLS_CPPWARNINGS@
 
 lib_LTLIBRARIES = libmp4v2.la 
 
-include_HEADERS = mp4.h
+include_HEADERS = mp4.h $(top_srcdir)/include/mpeg4ip.h $(top_srcdir)/include/mpeg4ip_version.h $(top_srcdir)/mpeg4ip_config.h 
 
 libmp4v2_la_SOURCES = \
 	3gp.cpp \
diff -Naurp mpeg4ip-1.5.0.1/lib/mp4v2/Makefile.in mpeg4ip-1.5.0.1-p/lib/mp4v2/Makefile.in
--- mpeg4ip-1.5.0.1/lib/mp4v2/Makefile.in	2006-06-08 21:17:29.000000000 +0200
+++ mpeg4ip-1.5.0.1-p/lib/mp4v2/Makefile.in	2006-11-19 12:37:33.000000000 +0100
@@ -284,11 +284,11 @@ target_alias = @target_alias@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-SUBDIRS = . test util
+SUBDIRS = .
 INCLUDES = -I$(top_srcdir)/include
 AM_CXXFLAGS = @BILLS_CPPWARNINGS@
 lib_LTLIBRARIES = libmp4v2.la 
-include_HEADERS = mp4.h
+include_HEADERS = mp4.h $(top_srcdir)/include/mpeg4ip.h $(top_srcdir)/include/mpeg4ip_version.h $(top_srcdir)/mpeg4ip_config.h 
 libmp4v2_la_SOURCES = \
 	3gp.cpp \
 	atom_amr.cpp \
