diff -Naur doxygen-1.5.0.orig/configure doxygen-1.5.0/configure
--- doxygen-1.5.0.orig/configure	2006-10-29 00:15:50.000000000 +0200
+++ doxygen-1.5.0/configure	2006-10-29 00:15:56.000000000 +0200
@@ -16,8 +16,8 @@
 # shell script to configure doxygen
 
 doxygen_version_major=1
-doxygen_version_minor=4
-doxygen_version_revision=7
+doxygen_version_minor=5
+doxygen_version_revision=0
 
 #NOTE: Setting version_mmn to "NO" will omit mmn info from the package.
 doxygen_version_mmn=NO
