diff -Naur doxygen-1.5.2.orig/src/latexgen.cpp doxygen-1.5.2/src/latexgen.cpp
--- doxygen-1.5.2.orig/src/latexgen.cpp	2007-04-07 18:35:18.000000000 +0200
+++ doxygen-1.5.2/src/latexgen.cpp	2007-04-07 18:37:27.000000000 +0200
@@ -237,16 +237,16 @@
   }
   if (Config_getBool("PDF_HYPERLINKS")) 
   {
-    t << "\\ifx\\pdfoutput\\undefined" << endl
-      << "\\usepackage[ps2pdf," << endl
+    t << "\\usepackage{ifpdf}" << endl
+      << "\\ifpdf" << endl
+      << "\\usepackage[pdftex," << endl
       << "            pagebackref=true," << endl
       << "            colorlinks=true," << endl
       << "            linkcolor=blue," << endl
       << "            unicode" << endl
       << "           ]{hyperref}" << endl
-      << "\\usepackage{pspicture}" << endl
       << "\\else" << endl
-      << "\\usepackage[pdftex," << endl
+      << "\\usepackage[ps2pdf," << endl
       << "            pagebackref=true," << endl
       << "            colorlinks=true," << endl
       << "            linkcolor=blue," << endl
