diff -Naur doxygen-1.8.14/addon/doxywizard/CMakeLists.txt doxygen-1.8.14-p/addon/doxywizard/CMakeLists.txt
--- doxygen-1.8.14/addon/doxywizard/CMakeLists.txt	2017-10-31 20:36:00.000000000 +0100
+++ doxygen-1.8.14-p/addon/doxywizard/CMakeLists.txt	2018-06-14 17:01:55.543508776 +0200
@@ -4,9 +4,6 @@
   if (Qt5Core_FOUND)
     message(STATUS "Using Qt5")
     find_package(Qt5 COMPONENTS Widgets Gui Xml)
-    macro(qt_use_modules)
-        qt5_use_modules(${ARGN})
-    endmacro()
     macro(qt_wrap_cpp)
         qt5_wrap_cpp(${ARGN})
     endmacro()
@@ -23,8 +20,6 @@
     message(STATUS "Using Qt4")
   endif()
   find_package(Qt4 REQUIRED COMPONENTS QtCore QtXml QtGui)
-  macro(qt_use_modules)
-  endmacro()
   macro(qt_wrap_cpp)
       qt4_wrap_cpp(${ARGN})
   endmacro()
@@ -104,9 +99,7 @@
 ${doxywizard_MOC}
 ${doxywizard_RESOURCES_RCC}
 )
-qt_use_modules(doxywizard Core Gui Widgets Xml)
-target_link_libraries(doxywizard
-${QT_LIBRARIES} ${QT_QTMAIN_LIBRARY}
-)
+target_link_libraries(doxywizard Qt5::Core Qt5::Gui Qt5::Widgets Qt5::Xml)
+
 
 install(TARGETS doxywizard DESTINATION bin)
