--- hotwire-0.721old/hotwire/state.py	2008-03-18 18:35:32.000000000 +0100
+++ hotwire-0.721/hotwire/state.py	2008-07-28 12:48:58.000000000 +0200
@@ -20,10 +20,9 @@
 # THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
 import os,sys,logging,time,datetime,threading
-if sys.version_info[0] < 2 or sys.version_info[1] < 5:
-    from pysqlite2 import dbapi2 as sqlite3
-else:    
-    import sqlite3
+
+from pysqlite2 import dbapi2 as sqlite3
+
 
 import gobject
 
