diff -ur a/scripts/bluetooth.rules.in b/scripts/bluetooth.rules.in
--- a/scripts/bluetooth.rules.in	2010-10-02 18:25:27.000000000 -0500
+++ b/scripts/bluetooth.rules.in	2012-04-07 00:38:30.223796262 -0500
@@ -1,4 +1,4 @@
 # Run helper every time a Bluetooth device appears
 # On remove actions, bluetoothd should go away by itself
-ACTION=="add", SUBSYSTEM=="bluetooth", RUN+="@prefix@/sbin/bluetoothd --udev"
-ACTION=="change", SUBSYSTEM=="bluetooth", RUN+="@prefix@/sbin/bluetoothd --udev"
+ACTION=="add", SUBSYSTEM=="bluetooth", RUN+="@prefix@/sbin/bluetoothd-udev-wrapper"
+ACTION=="change", SUBSYSTEM=="bluetooth", RUN+="@prefix@/sbin/bluetoothd-udev-wrapper"
