diff -Naur validator-0.8.0-orig/httpd/conf/httpd.conf validator-0.8.0/httpd/conf/httpd.conf
--- validator-0.8.0-orig/httpd/conf/httpd.conf	2007-07-25 03:26:39.000000000 +0200
+++ validator-0.8.0/httpd/conf/httpd.conf	2007-08-02 09:12:20.000000000 +0200
@@ -70,8 +70,9 @@
   RewriteRule /w3c-validator/whatsnew.atom$ http://www.w3.org/QA/Tools/validator-whatsnew.atom [P,L]
 </IfModule>
 
+<IfModule mod_proxy.c>
 <Proxy *>
   Order Deny,Allow
   Allow from all
 </Proxy>
-
+</IfModule>
