diff -Naur font-adobe-75dpi-0.99.1.orig/Makefile.am font-adobe-75dpi-0.99.1/Makefile.am
--- font-adobe-75dpi-0.99.1.orig/Makefile.am	2005-11-13 23:54:01.685337750 +0100
+++ font-adobe-75dpi-0.99.1/Makefile.am	2005-11-13 23:54:24.702776250 +0100
@@ -237,12 +237,5 @@
 
 all-local: $(BUILT_BDF_FILES) $(font_DATA)
 
-install-data-hook:
-	@rm -f $(DESTDIR)$(fontdir)/fonts.scale
-	$(MKFONTSCALE) $(DESTDIR)$(fontdir)
-	@rm -f $(DESTDIR)$(fontdir)/font.dir
-	$(MKFONTDIR) $(DESTDIR)$(fontdir)
-	$(FCCACHE) $(DESTDIR)$(fontdir)
-
 distuninstallcheck:
 	@:
