--- spice-protocol-0.14.2/meson.build	2020-05-15 16:27:33.880287200 +0200
+++ spice-protocol-0.14.2.new/meson.build	2020-06-08 15:32:28.381251541 +0200
@@ -13,8 +13,9 @@
 #
 # spice-protocol.pc generation
 #
+libdir = get_option('libdir')
 datadir = get_option('datadir')
-pkgconfig_dir = join_paths(get_option('prefix'), datadir, 'pkgconfig')
+pkgconfig_dir = join_paths(get_option('prefix'), libdir, 'pkgconfig')
 
 pkgconfig = import('pkgconfig')
 pkgconfig.generate(name : meson.project_name(),
