diff -Naur gtk-sharp-1.0.8.orig/configure gtk-sharp-1.0.8/configure
--- gtk-sharp-1.0.8.orig/configure	2005-09-23 12:59:06.000000000 +0200
+++ gtk-sharp-1.0.8/configure	2005-09-23 13:02:25.000000000 +0200
@@ -21971,20 +21971,20 @@
         echo "$as_me:$LINENO: checking for libgda >= $LIBGDA_REQUIRED_VERSION" >&5
 echo $ECHO_N "checking for libgda >= $LIBGDA_REQUIRED_VERSION... $ECHO_C" >&6
 
-        if $PKG_CONFIG --exists "libgda >= $LIBGDA_REQUIRED_VERSION" ; then
+        if $PKG_CONFIG --exists "libgda-2.0 >= $LIBGDA_REQUIRED_VERSION" ; then
             echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6
             succeeded=yes
 
             echo "$as_me:$LINENO: checking LIBGDA_DEPENDENCIES_CFLAGS" >&5
 echo $ECHO_N "checking LIBGDA_DEPENDENCIES_CFLAGS... $ECHO_C" >&6
-            LIBGDA_DEPENDENCIES_CFLAGS=`$PKG_CONFIG --cflags "libgda >= $LIBGDA_REQUIRED_VERSION"`
+            LIBGDA_DEPENDENCIES_CFLAGS=`$PKG_CONFIG --cflags "libgda-2.0 >= $LIBGDA_REQUIRED_VERSION"`
             echo "$as_me:$LINENO: result: $LIBGDA_DEPENDENCIES_CFLAGS" >&5
 echo "${ECHO_T}$LIBGDA_DEPENDENCIES_CFLAGS" >&6
 
             echo "$as_me:$LINENO: checking LIBGDA_DEPENDENCIES_LIBS" >&5
 echo $ECHO_N "checking LIBGDA_DEPENDENCIES_LIBS... $ECHO_C" >&6
-            LIBGDA_DEPENDENCIES_LIBS=`$PKG_CONFIG --libs "libgda >= $LIBGDA_REQUIRED_VERSION"`
+            LIBGDA_DEPENDENCIES_LIBS=`$PKG_CONFIG --libs "libgda-2.0 >= $LIBGDA_REQUIRED_VERSION"`
             echo "$as_me:$LINENO: result: $LIBGDA_DEPENDENCIES_LIBS" >&5
 echo "${ECHO_T}$LIBGDA_DEPENDENCIES_LIBS" >&6
         else
@@ -21992,7 +21992,7 @@
             LIBGDA_DEPENDENCIES_LIBS=""
             ## If we have a custom action on failure, don't print errors, but
             ## do set a variable so people can do so.
-            LIBGDA_DEPENDENCIES_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgda >= $LIBGDA_REQUIRED_VERSION"`
+            LIBGDA_DEPENDENCIES_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgda-2.0 >= $LIBGDA_REQUIRED_VERSION"`
 
         fi
 
@@ -22068,23 +22068,23 @@
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        echo "$as_me:$LINENO: checking for libgnomedb >= $LIBGDA_REQUIRED_VERSION" >&5
-echo $ECHO_N "checking for libgnomedb >= $LIBGDA_REQUIRED_VERSION... $ECHO_C" >&6
+        echo "$as_me:$LINENO: checking for libgnomedb-2.0 >= $LIBGDA_REQUIRED_VERSION" >&5
+echo $ECHO_N "checking for libgnomedb-2.0 >= $LIBGDA_REQUIRED_VERSION... $ECHO_C" >&6
 
-        if $PKG_CONFIG --exists "libgnomedb >= $LIBGDA_REQUIRED_VERSION" ; then
+        if $PKG_CONFIG --exists "libgnomedb-2.0 >= $LIBGDA_REQUIRED_VERSION" ; then
             echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6
             succeeded=yes
 
             echo "$as_me:$LINENO: checking LIBGNOMEDB_DEPENDENCIES_CFLAGS" >&5
 echo $ECHO_N "checking LIBGNOMEDB_DEPENDENCIES_CFLAGS... $ECHO_C" >&6
-            LIBGNOMEDB_DEPENDENCIES_CFLAGS=`$PKG_CONFIG --cflags "libgnomedb >= $LIBGDA_REQUIRED_VERSION"`
+            LIBGNOMEDB_DEPENDENCIES_CFLAGS=`$PKG_CONFIG --cflags "libgnomedb-2.0 >= $LIBGDA_REQUIRED_VERSION"`
             echo "$as_me:$LINENO: result: $LIBGNOMEDB_DEPENDENCIES_CFLAGS" >&5
 echo "${ECHO_T}$LIBGNOMEDB_DEPENDENCIES_CFLAGS" >&6
 
             echo "$as_me:$LINENO: checking LIBGNOMEDB_DEPENDENCIES_LIBS" >&5
 echo $ECHO_N "checking LIBGNOMEDB_DEPENDENCIES_LIBS... $ECHO_C" >&6
-            LIBGNOMEDB_DEPENDENCIES_LIBS=`$PKG_CONFIG --libs "libgnomedb >= $LIBGDA_REQUIRED_VERSION"`
+            LIBGNOMEDB_DEPENDENCIES_LIBS=`$PKG_CONFIG --libs "libgnomedb-2.0 >= $LIBGDA_REQUIRED_VERSION"`
             echo "$as_me:$LINENO: result: $LIBGNOMEDB_DEPENDENCIES_LIBS" >&5
 echo "${ECHO_T}$LIBGNOMEDB_DEPENDENCIES_LIBS" >&6
         else
@@ -22092,7 +22092,7 @@
             LIBGNOMEDB_DEPENDENCIES_LIBS=""
             ## If we have a custom action on failure, don't print errors, but
             ## do set a variable so people can do so.
-            LIBGNOMEDB_DEPENDENCIES_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgnomedb >= $LIBGDA_REQUIRED_VERSION"`
+            LIBGNOMEDB_DEPENDENCIES_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgnomedb-2.0 >= $LIBGDA_REQUIRED_VERSION"`
 
         fi
 
@@ -22266,23 +22266,23 @@
   else
      PKG_CONFIG_MIN_VERSION=0.9.0
      if $PKG_CONFIG --atleast-pkgconfig-version $PKG_CONFIG_MIN_VERSION; then
-        echo "$as_me:$LINENO: checking for libgtkhtml-3.6 >= $GTKHTML_REQUIRED_VERSION" >&5
-echo $ECHO_N "checking for libgtkhtml-3.6 >= $GTKHTML_REQUIRED_VERSION... $ECHO_C" >&6
+        echo "$as_me:$LINENO: checking for libgtkhtml-3.8 >= $GTKHTML_REQUIRED_VERSION" >&5
+echo $ECHO_N "checking for libgtkhtml-3.8 >= $GTKHTML_REQUIRED_VERSION... $ECHO_C" >&6
 
-        if $PKG_CONFIG --exists "libgtkhtml-3.6 >= $GTKHTML_REQUIRED_VERSION" ; then
+        if $PKG_CONFIG --exists "libgtkhtml-3.8 >= $GTKHTML_REQUIRED_VERSION" ; then
             echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6
             succeeded=yes
 
             echo "$as_me:$LINENO: checking GTKHTML_DEPENDENCIES_CFLAGS" >&5
 echo $ECHO_N "checking GTKHTML_DEPENDENCIES_CFLAGS... $ECHO_C" >&6
-            GTKHTML_DEPENDENCIES_CFLAGS=`$PKG_CONFIG --cflags "libgtkhtml-3.6 >= $GTKHTML_REQUIRED_VERSION"`
+            GTKHTML_DEPENDENCIES_CFLAGS=`$PKG_CONFIG --cflags "libgtkhtml-3.8 >= $GTKHTML_REQUIRED_VERSION"`
             echo "$as_me:$LINENO: result: $GTKHTML_DEPENDENCIES_CFLAGS" >&5
 echo "${ECHO_T}$GTKHTML_DEPENDENCIES_CFLAGS" >&6
 
             echo "$as_me:$LINENO: checking GTKHTML_DEPENDENCIES_LIBS" >&5
 echo $ECHO_N "checking GTKHTML_DEPENDENCIES_LIBS... $ECHO_C" >&6
-            GTKHTML_DEPENDENCIES_LIBS=`$PKG_CONFIG --libs "libgtkhtml-3.6 >= $GTKHTML_REQUIRED_VERSION"`
+            GTKHTML_DEPENDENCIES_LIBS=`$PKG_CONFIG --libs "libgtkhtml-3.8 >= $GTKHTML_REQUIRED_VERSION"`
             echo "$as_me:$LINENO: result: $GTKHTML_DEPENDENCIES_LIBS" >&5
 echo "${ECHO_T}$GTKHTML_DEPENDENCIES_LIBS" >&6
         else
@@ -22290,7 +22290,7 @@
             GTKHTML_DEPENDENCIES_LIBS=""
             ## If we have a custom action on failure, don't print errors, but
             ## do set a variable so people can do so.
-            GTKHTML_DEPENDENCIES_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgtkhtml-3.6 >= $GTKHTML_REQUIRED_VERSION"`
+            GTKHTML_DEPENDENCIES_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libgtkhtml-3.8 >= $GTKHTML_REQUIRED_VERSION"`
 
         fi
 
@@ -22309,8 +22309,8 @@
   fi
 
 if test "x$enable_gtkhtml" = "xyes"; then
-GTKHTML_VERSION=3.6
-GTKHTML_SOVERSION=18
+GTKHTML_VERSION=3.8
+GTKHTML_SOVERSION=15
 else
 GTKHTML_REQUIRED_VERSION=3.2.0
 
