diff -Naur cmst-cmst-2017.09.19/misc/desktop/cmst-autostart.desktop cmst-cmst-2017.09.19-p/misc/desktop/cmst-autostart.desktop
--- cmst-cmst-2017.09.19/misc/desktop/cmst-autostart.desktop	2017-09-20 01:36:48.000000000 +0200
+++ cmst-cmst-2017.09.19-p/misc/desktop/cmst-autostart.desktop	2017-09-24 16:17:01.914216808 +0200
@@ -6,11 +6,9 @@
 Comment=QT GUI frontend for connman
 Categories=Settings;System;Qt;Network;
 Icon=cmst
-Exec=cmst -w5
+Exec=cmst -w5 -m
 Terminal=false
 StartupNotify=false
-X-GNOME-Autostart-enabled=true
 Keywords=Network;Wireless;Wi-Fi;Wifi;IP;LAN;Proxy;WAN;Broadband;Bluetooth;vpn;DNS;
-
 Name[de]=Netzwerk-Konfiguration
 
