diff -Naur cmst-cmst-2017.03.18/apps/rootapp/system/distro/frugalware/org.cmst.roothelper.conf cmst-cmst-2017.03.18-fw/apps/rootapp/system/distro/frugalware/org.cmst.roothelper.conf
--- cmst-cmst-2017.03.18/apps/rootapp/system/distro/frugalware/org.cmst.roothelper.conf	1970-01-01 01:00:00.000000000 +0100
+++ cmst-cmst-2017.03.18-fw/apps/rootapp/system/distro/frugalware/org.cmst.roothelper.conf	2017-09-17 17:46:14.985711608 +0200
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?> <!-- -*- XML -*- -->
+
+<!DOCTYPE busconfig PUBLIC "-//freedesktop//DTD D-BUS Bus Configuration 1.0//EN"
+ "http://www.freedesktop.org/standards/dbus/1.0/busconfig.dtd">
+<busconfig>
+    <!-- Only root can own roothelper -->
+    <policy user="root">
+        <allow own="org.cmst.roothelper"/>
+        <allow send_destination="org.cmst.roothelper"
+               send_interface="org.cmst.roothelper"/>
+    </policy>
+    <policy group="netdev">
+        <allow send_destination="org.cmst.roothelper"
+               send_interface="org.cmst.roothelper"/>
+    </policy>
+</busconfig>
