diff -Naur imapfilter-2.6.11/src/Makefile imapfilter-2.6.11-p/src/Makefile
--- imapfilter-2.6.11/src/Makefile	2017-11-19 10:38:24.000000000 +0100
+++ imapfilter-2.6.11-p/src/Makefile	2018-02-10 00:41:20.689575295 +0100
@@ -1,8 +1,8 @@
 DESTDIR ?=
-PREFIX ?= /usr/local
+PREFIX ?= /usr
 BINDIR = $(PREFIX)/bin
 SHAREDIR = $(PREFIX)/share/imapfilter
-MANDIR = $(PREFIX)/man
+MANDIR = $(PREFIX)/share/man
 
 MYCFLAGS =
 MYLDFLAGS =
@@ -16,8 +16,8 @@
 LIBSSL = -lssl
 LIBCRYPTO = -lcrypto
 
-CFLAGS = -Wall -Wextra -O -DCONFIG_SHAREDIR='"$(SHAREDIR)"' $(INCDIRS) $(MYCFLAGS)
-LDFLAGS = $(LIBDIRS) $(MYLDFLAGS)
+CFLAGS += -Wall -Wextra -O -DCONFIG_SHAREDIR='"$(SHAREDIR)"' $(INCDIRS) $(MYCFLAGS)
+LDFLAGS += $(LIBDIRS) $(MYLDFLAGS)
 LIBS = -lm -ldl $(LIBLUA) $(LIBPCRE) $(LIBSSL) $(LIBCRYPTO) $(MYLIBS)
 
 MAN1 = imapfilter.1
