--- acinclude.m4.vala	2011-10-01 20:39:23.000000000 +0200
+++ acinclude.m4	2011-10-01 20:39:47.000000000 +0200
@@ -254,7 +254,7 @@
 # GWIBBER_PROG_VALAC([MINIMUM-VERSION])
 # --------------------------------
 AC_DEFUN([GWIBBER_PROG_VALAC],
-[AC_PATH_PROG([VALAC], [valac-0.12], [])
+[AC_PATH_PROG([VALAC], [valac], [])
  AS_IF([test -z "$VALAC"],
    [AC_MSG_WARN([No Vala compiler found.  You will not be able to compile .vala source files.])],
    [AS_IF([test -n "$1"],
