diff -Naur gmp-6.1.1/Makefile.am gmp-6.1.1-p/Makefile.am
--- gmp-6.1.1/Makefile.am	2016-06-18 22:00:00.000000000 +0200
+++ gmp-6.1.1-p/Makefile.am	2016-10-22 23:05:33.008990968 +0200
@@ -129,7 +129,7 @@
 # but anyone knowledgeable enough to be playing with exec_prefix will be able
 # to address that.
 #
-includeexecdir = $(exec_prefix)/include
+includeexecdir = @includedir@
 include_HEADERS = $(GMPXX_HEADERS_OPTION)
 nodist_includeexec_HEADERS = gmp.h
 lib_LTLIBRARIES = libgmp.la $(GMPXX_LTLIBRARIES_OPTION)
diff -Naur gmp-6.1.1/Makefile.in gmp-6.1.1-p/Makefile.in
--- gmp-6.1.1/Makefile.in	2016-06-18 22:00:06.000000000 +0200
+++ gmp-6.1.1-p/Makefile.in	2016-10-22 23:07:00.365992262 +0200
@@ -601,7 +601,7 @@
 # but anyone knowledgeable enough to be playing with exec_prefix will be able
 # to address that.
 #
-includeexecdir = $(exec_prefix)/include
+includeexecdir = @includedir@
 include_HEADERS = $(GMPXX_HEADERS_OPTION)
 nodist_includeexec_HEADERS = gmp.h
 lib_LTLIBRARIES = libgmp.la $(GMPXX_LTLIBRARIES_OPTION)
