--- xf86-video-nv-2.1.18/src/nv_driver.c.old	2010-08-18 17:55:42.000000000 +0200
+++ xf86-video-nv-2.1.18/src/nv_driver.c	2010-08-19 09:40:31.000000000 +0200
@@ -912,14 +912,6 @@
                       NVGetPCIXpressChip(dev) : dev->vendor_id << 16 | dev->device_id;
     const char *name = xf86TokenToString(NVKnownChipsets, id);
 
-    if (pci_device_has_kernel_driver(dev)) {
-        xf86DrvMsg(0, X_ERROR,
-                   NV_NAME ": The PCI device 0x%x (%s) at %2.2d@%2.2d:%2.2d:%1.1d has a kernel module claiming it.\n",
-                   id, name, dev->bus, dev->domain, dev->dev, dev->func);
-        xf86DrvMsg(0, X_ERROR,
-                   NV_NAME ": This driver cannot operate until it has been unloaded.\n");
-        return FALSE;
-    }
 
     if(dev->vendor_id == PCI_VENDOR_NVIDIA && !name &&
        !NVIsSupported(id) && !NVIsG80(id)) {
