--- requests-2.27.1/setup.py	2022-01-05 16:36:01.000000000 +0100
+++ requests-2.27.1.new/setup.py	2022-07-13 09:30:09.044793905 +0200
@@ -41,12 +41,10 @@
 packages = ['requests']
 
 requires = [
-    'charset_normalizer~=2.0.0; python_version >= "3"',
-    'chardet>=3.0.2,<5; python_version < "3"',
-    'idna>=2.5,<3; python_version < "3"',
-    'idna>=2.5,<4; python_version >= "3"',
-    'urllib3>=1.21.1,<1.27',
-    'certifi>=2017.4.17'
+    'chardet>=3.0.2',
+    'idna>=2.5',
+    'idna>=2.5',
+    'urllib3>=1.21.1',
 
 ]
 test_requirements = [
@@ -109,7 +107,7 @@
         'security': [],
         'socks': ['PySocks>=1.5.6, !=1.5.7'],
         'socks:sys_platform == "win32" and python_version == "2.7"': ['win_inet_pton'],
-        'use_chardet_on_py3': ['chardet>=3.0.2,<5']
+        'use_chardet_on_py3': ['chardet>=3.0.2']
     },
     project_urls={
         'Documentation': 'https://requests.readthedocs.io',
