diff --git a/src/software/pipeline/main_imageMasking.cpp b/src/software/pipeline/main_imageMasking.cpp
index 5acd395e7..68380d2bd 100644
--- a/src/software/pipeline/main_imageMasking.cpp
+++ b/src/software/pipeline/main_imageMasking.cpp
@@ -10,6 +10,7 @@
 #include <aliceVision/image/io.hpp>
 #include <aliceVision/imageMasking/imageMasking.hpp>
 #include <aliceVision/system/Logger.hpp>
+#include <aliceVision/system/Timer.hpp>
 #include <aliceVision/system/cmdline.hpp>
 
 #include <OpenImageIO/imagebuf.h>
diff --git a/src/software/utils/CMakeLists.txt b/src/software/utils/CMakeLists.txt
index 2657bf512..14c68c3c2 100644
--- a/src/software/utils/CMakeLists.txt
+++ b/src/software/utils/CMakeLists.txt
@@ -48,7 +48,7 @@ alicevision_add_software(aliceVision_utils_imageProcessing
 )
 
 if(ALICEVISION_HAVE_OPENCV)
-  target_link_libraries(aliceVision_utils_imageProcessing PRIVATE ${OpenCV_LIBS})
+  target_link_libraries(aliceVision_utils_imageProcessing_exe PRIVATE ${OpenCV_LIBS})
 endif()
 
 alicevision_add_software(aliceVision_utils_importMiddlebury
