diff -Naur alicevision/src/CMakeLists.txt alicevision.new/src/CMakeLists.txt
--- alicevision/src/CMakeLists.txt	2022-06-08 09:19:01.688891811 +0200
+++ alicevision.new/src/CMakeLists.txt	2022-06-08 09:21:25.910050800 +0200
@@ -41,7 +41,7 @@
 endfunction()
 
 # C++11
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # ==============================================================================
@@ -209,7 +209,7 @@
 if(NOT CXX11_COMPILER)
   message(FATAL_ERROR "The compiler does not support the CXX11 standard.")
 endif(NOT CXX11_COMPILER)
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 # ==============================================================================
diff -Naur alicevision/src/dependencies/MeshSDFilter/CMakeLists.txt alicevision.new/src/dependencies/MeshSDFilter/CMakeLists.txt
--- alicevision/src/dependencies/MeshSDFilter/CMakeLists.txt	2022-06-08 09:19:01.102887091 +0200
+++ alicevision.new/src/dependencies/MeshSDFilter/CMakeLists.txt	2022-06-08 09:22:13.890435328 +0200
@@ -2,7 +2,7 @@
 
 project(SDFilter)
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_VERBOSE_MAKEFILE ON)
diff -Naur alicevision/src/samples/aliceVisionAs3rdParty/CMakeLists.txt alicevision.new/src/samples/aliceVisionAs3rdParty/CMakeLists.txt
--- alicevision/src/samples/aliceVisionAs3rdParty/CMakeLists.txt	2022-06-08 09:18:57.820860652 +0200
+++ alicevision.new/src/samples/aliceVisionAs3rdParty/CMakeLists.txt	2022-06-08 09:22:28.738554232 +0200
@@ -2,7 +2,7 @@
 
 project(AliceVisionAs3rdParty)
 
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 find_package(AliceVision CONFIG REQUIRED)
diff --git a/src/cmake/Helpers.cmake b/src/cmake/Helpers.cmake
index 9c6f192a9..6a2084460 100644
--- a/src/cmake/Helpers.cmake
+++ b/src/cmake/Helpers.cmake
@@ -85,6 +85,7 @@ function(alicevision_add_library library_name)
     RUNTIME
         DESTINATION ${CMAKE_INSTALL_BINDIR}
   )
+    target_compile_features(${library_name} PUBLIC cxx_std_14)
 endfunction()
 
 # Add interface function
