diff --git a/src/software/export/main_exportMVE2.cpp b/src/software/export/main_exportMVE2.cpp
index c4429778ac..e2359a5238 100644
--- a/src/software/export/main_exportMVE2.cpp
+++ b/src/software/export/main_exportMVE2.cpp
@@ -19,6 +19,7 @@
 #include <cmath>
 #include <iterator>
 #include <iomanip>
+#include <fstream>
 
 // These constants define the current software version.
 // They must be updated when the command line is changed.
diff --git a/src/software/export/main_exportMatlab.cpp b/src/software/export/main_exportMatlab.cpp
index 65bee75989..9306b69bec 100644
--- a/src/software/export/main_exportMatlab.cpp
+++ b/src/software/export/main_exportMatlab.cpp
@@ -20,6 +20,7 @@
 #include <iterator>
 #include <iomanip>
 #include <map>
+#include <fstream>
 
 // These constants define the current software version.
 // They must be updated when the command line is changed.
diff --git a/src/software/export/main_exportPMVS.cpp b/src/software/export/main_exportPMVS.cpp
index c1c52b3d65..4935fac43a 100644
--- a/src/software/export/main_exportPMVS.cpp
+++ b/src/software/export/main_exportPMVS.cpp
@@ -19,6 +19,7 @@
 #include <cmath>
 #include <iterator>
 #include <iomanip>
+#include <fstream>
 
 // These constants define the current software version.
 // They must be updated when the command line is changed.
diff --git a/src/software/pipeline/main_LdrToHdrCalibration.cpp b/src/software/pipeline/main_LdrToHdrCalibration.cpp
index 0f55fd4c82..57522828de 100644
--- a/src/software/pipeline/main_LdrToHdrCalibration.cpp
+++ b/src/software/pipeline/main_LdrToHdrCalibration.cpp
@@ -32,6 +32,8 @@
 
 #include <sstream>
 
+#include <fstream>
+
 
 // These constants define the current software version.
 // They must be updated when the command line is changed.
diff --git a/src/software/pipeline/main_LdrToHdrSampling.cpp b/src/software/pipeline/main_LdrToHdrSampling.cpp
index 7abfe7656a..17b04453f7 100644
--- a/src/software/pipeline/main_LdrToHdrSampling.cpp
+++ b/src/software/pipeline/main_LdrToHdrSampling.cpp
@@ -35,6 +35,8 @@
 
 #include <sstream>
 
+#include <fstream>
+
 
 // These constants define the current software version.
 // They must be updated when the command line is changed.
diff --git a/src/software/pipeline/main_prepareDenseScene.cpp b/src/software/pipeline/main_prepareDenseScene.cpp
index 299b043774..4010104d21 100644
--- a/src/software/pipeline/main_prepareDenseScene.cpp
+++ b/src/software/pipeline/main_prepareDenseScene.cpp
@@ -24,6 +24,7 @@
 #include <set>
 #include <iterator>
 #include <iomanip>
+#include <fstream>
 
 // These constants define the current software version.
 // They must be updated when the command line is changed.
