diff -Naur alicevision/src/aliceVision/calibration/bestImages.cpp alicevision.new/src/aliceVision/calibration/bestImages.cpp
--- alicevision/src/aliceVision/calibration/bestImages.cpp	2022-02-11 12:37:07.701121813 +0100
+++ alicevision.new/src/aliceVision/calibration/bestImages.cpp	2022-02-11 12:37:30.407283297 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/camera/Pinhole.hpp alicevision.new/src/aliceVision/camera/Pinhole.hpp
--- alicevision/src/aliceVision/camera/Pinhole.hpp	2022-02-11 12:37:11.872151476 +0100
+++ alicevision.new/src/aliceVision/camera/Pinhole.hpp	2022-02-11 12:37:30.407283297 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/dataio/FeedProvider.cpp alicevision.new/src/aliceVision/dataio/FeedProvider.cpp
--- alicevision/src/aliceVision/dataio/FeedProvider.cpp	2022-02-11 12:37:07.706121849 +0100
+++ alicevision.new/src/aliceVision/dataio/FeedProvider.cpp	2022-02-11 12:37:30.407283297 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2015 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/depthMap/DepthSimMap.cpp alicevision.new/src/aliceVision/depthMap/DepthSimMap.cpp
--- alicevision/src/aliceVision/depthMap/DepthSimMap.cpp	2022-02-11 12:37:11.874151491 +0100
+++ alicevision.new/src/aliceVision/depthMap/DepthSimMap.cpp	2022-02-11 12:37:30.408283304 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2017 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/depthMap/SemiGlobalMatchingRc.cpp alicevision.new/src/aliceVision/depthMap/SemiGlobalMatchingRc.cpp
--- alicevision/src/aliceVision/depthMap/SemiGlobalMatchingRc.cpp	2022-02-11 12:37:11.875151498 +0100
+++ alicevision.new/src/aliceVision/depthMap/SemiGlobalMatchingRc.cpp	2022-02-11 12:37:30.408283304 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2017 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/feature/sift/ImageDescriber_DSPSIFT_vlfeat.cpp alicevision.new/src/aliceVision/feature/sift/ImageDescriber_DSPSIFT_vlfeat.cpp
--- alicevision/src/aliceVision/feature/sift/ImageDescriber_DSPSIFT_vlfeat.cpp	2022-02-11 12:37:11.880151533 +0100
+++ alicevision.new/src/aliceVision/feature/sift/ImageDescriber_DSPSIFT_vlfeat.cpp	2022-02-11 12:37:30.401283255 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2020 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/feature/sift/SIFT.cpp alicevision.new/src/aliceVision/feature/sift/SIFT.cpp
--- alicevision/src/aliceVision/feature/sift/SIFT.cpp	2022-02-11 12:37:07.720121948 +0100
+++ alicevision.new/src/aliceVision/feature/sift/SIFT.cpp	2022-02-11 12:37:30.402283262 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2018 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/fuseCut/DelaunayGraphCut.cpp alicevision.new/src/aliceVision/fuseCut/DelaunayGraphCut.cpp
--- alicevision/src/aliceVision/fuseCut/DelaunayGraphCut.cpp	2022-02-11 12:37:11.881151540 +0100
+++ alicevision.new/src/aliceVision/fuseCut/DelaunayGraphCut.cpp	2022-02-11 12:37:30.403283269 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2017 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/fuseCut/DelaunayGraphCut.hpp alicevision.new/src/aliceVision/fuseCut/DelaunayGraphCut.hpp
--- alicevision/src/aliceVision/fuseCut/DelaunayGraphCut.hpp	2022-02-11 12:37:11.881151540 +0100
+++ alicevision.new/src/aliceVision/fuseCut/DelaunayGraphCut.hpp	2022-02-11 12:37:30.403283269 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2017 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/fuseCut/Fuser.cpp alicevision.new/src/aliceVision/fuseCut/Fuser.cpp
--- alicevision/src/aliceVision/fuseCut/Fuser.cpp	2022-02-11 12:37:11.882151547 +0100
+++ alicevision.new/src/aliceVision/fuseCut/Fuser.cpp	2022-02-11 12:37:30.403283269 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2017 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/geometry/frustumIntersection_test.cpp alicevision.new/src/aliceVision/geometry/frustumIntersection_test.cpp
--- alicevision/src/aliceVision/geometry/frustumIntersection_test.cpp	2022-02-11 12:37:07.724121977 +0100
+++ alicevision.new/src/aliceVision/geometry/frustumIntersection_test.cpp	2022-02-11 12:37:30.403283269 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/geometry/rigidTransformation3D.cpp alicevision.new/src/aliceVision/geometry/rigidTransformation3D.cpp
--- alicevision/src/aliceVision/geometry/rigidTransformation3D.cpp	2022-02-11 12:37:07.724121977 +0100
+++ alicevision.new/src/aliceVision/geometry/rigidTransformation3D.cpp	2022-02-11 12:37:30.404283276 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/geometry/rigidTransformation3D.hpp alicevision.new/src/aliceVision/geometry/rigidTransformation3D.hpp
--- alicevision/src/aliceVision/geometry/rigidTransformation3D.hpp	2022-02-11 12:37:07.724121977 +0100
+++ alicevision.new/src/aliceVision/geometry/rigidTransformation3D.hpp	2022-02-11 12:37:30.404283276 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/graph/connectedComponent.hpp alicevision.new/src/aliceVision/graph/connectedComponent.hpp
--- alicevision/src/aliceVision/graph/connectedComponent.hpp	2022-02-11 12:37:07.725121984 +0100
+++ alicevision.new/src/aliceVision/graph/connectedComponent.hpp	2022-02-11 12:37:30.405283283 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/hdr/QuadProg++.cpp alicevision.new/src/aliceVision/hdr/QuadProg++.cpp
--- alicevision/src/aliceVision/hdr/QuadProg++.cpp	2022-02-11 12:37:07.727121998 +0100
+++ alicevision.new/src/aliceVision/hdr/QuadProg++.cpp	2022-02-11 12:37:30.406283290 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /*
 File $Id: QuadProg++.cc 232 2007-06-21 12:29:00Z digasper $
 
diff -Naur alicevision/src/aliceVision/hdr/QuadProg++.hpp alicevision.new/src/aliceVision/hdr/QuadProg++.hpp
--- alicevision/src/aliceVision/hdr/QuadProg++.hpp	2022-02-11 12:37:07.727121998 +0100
+++ alicevision.new/src/aliceVision/hdr/QuadProg++.hpp	2022-02-11 12:37:30.406283290 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /*
  File $Id: QuadProg++.hh 232 2007-06-21 12:29:00Z digasper $ 
  
diff -Naur alicevision/src/aliceVision/hdr/rgbCurve.cpp alicevision.new/src/aliceVision/hdr/rgbCurve.cpp
--- alicevision/src/aliceVision/hdr/rgbCurve.cpp	2022-02-11 12:37:11.883151555 +0100
+++ alicevision.new/src/aliceVision/hdr/rgbCurve.cpp	2022-02-11 12:37:30.406283290 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2019 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/image/Sampler.hpp alicevision.new/src/aliceVision/image/Sampler.hpp
--- alicevision/src/aliceVision/image/Sampler.hpp	2022-02-11 12:37:11.883151555 +0100
+++ alicevision.new/src/aliceVision/image/Sampler.hpp	2022-02-11 12:37:30.406283290 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/keyframe/KeyframeSelector.cpp alicevision.new/src/aliceVision/keyframe/KeyframeSelector.cpp
--- alicevision/src/aliceVision/keyframe/KeyframeSelector.cpp	2022-02-11 12:37:11.886151576 +0100
+++ alicevision.new/src/aliceVision/keyframe/KeyframeSelector.cpp	2022-02-11 12:37:30.406283290 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2017 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/localization/ILocalizer.hpp alicevision.new/src/aliceVision/localization/ILocalizer.hpp
--- alicevision/src/aliceVision/localization/ILocalizer.hpp	2022-02-11 12:37:07.743122112 +0100
+++ alicevision.new/src/aliceVision/localization/ILocalizer.hpp	2022-02-11 12:37:30.407283297 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/localization/optimization.cpp alicevision.new/src/aliceVision/localization/optimization.cpp
--- alicevision/src/aliceVision/localization/optimization.cpp	2022-02-11 12:37:11.888151590 +0100
+++ alicevision.new/src/aliceVision/localization/optimization.cpp	2022-02-11 12:37:30.407283297 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/localization/rigResection_test.cpp alicevision.new/src/aliceVision/localization/rigResection_test.cpp
--- alicevision/src/aliceVision/localization/rigResection_test.cpp	2022-02-11 12:37:11.888151590 +0100
+++ alicevision.new/src/aliceVision/localization/rigResection_test.cpp	2022-02-11 12:37:30.407283297 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/localization/VoctreeLocalizer.hpp alicevision.new/src/aliceVision/localization/VoctreeLocalizer.hpp
--- alicevision/src/aliceVision/localization/VoctreeLocalizer.hpp	2022-02-11 12:37:07.744122119 +0100
+++ alicevision.new/src/aliceVision/localization/VoctreeLocalizer.hpp	2022-02-11 12:37:30.407283297 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/matching/guidedMatching.hpp alicevision.new/src/aliceVision/matching/guidedMatching.hpp
--- alicevision/src/aliceVision/matching/guidedMatching.hpp	2022-02-11 12:37:11.888151590 +0100
+++ alicevision.new/src/aliceVision/matching/guidedMatching.hpp	2022-02-11 12:37:30.408283304 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_E_AC.hpp alicevision.new/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_E_AC.hpp
--- alicevision/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_E_AC.hpp	2022-02-11 12:37:07.749122155 +0100
+++ alicevision.new/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_E_AC.hpp	2022-02-11 12:37:30.408283304 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_F_AC.hpp alicevision.new/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_F_AC.hpp
--- alicevision/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_F_AC.hpp	2022-02-11 12:37:07.749122155 +0100
+++ alicevision.new/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_F_AC.hpp	2022-02-11 12:37:30.408283304 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_H_AC.hpp alicevision.new/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_H_AC.hpp
--- alicevision/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_H_AC.hpp	2022-02-11 12:37:07.749122155 +0100
+++ alicevision.new/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_H_AC.hpp	2022-02-11 12:37:30.408283304 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_HGrowing.hpp alicevision.new/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_HGrowing.hpp
--- alicevision/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_HGrowing.hpp	2022-02-11 12:37:11.889151597 +0100
+++ alicevision.new/src/aliceVision/matchingImageCollection/GeometricFilterMatrix_HGrowing.hpp	2022-02-11 12:37:30.408283304 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2018 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/matchingImageCollection/geometricFilterUtils.cpp alicevision.new/src/aliceVision/matchingImageCollection/geometricFilterUtils.cpp
--- alicevision/src/aliceVision/matchingImageCollection/geometricFilterUtils.cpp	2022-02-11 12:37:07.750122162 +0100
+++ alicevision.new/src/aliceVision/matchingImageCollection/geometricFilterUtils.cpp	2022-02-11 12:37:30.408283304 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2018 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/mesh/Texturing.cpp alicevision.new/src/aliceVision/mesh/Texturing.cpp
--- alicevision/src/aliceVision/mesh/Texturing.cpp	2022-02-11 12:37:11.891151611 +0100
+++ alicevision.new/src/aliceVision/mesh/Texturing.cpp	2022-02-11 12:37:30.409283311 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2017 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/mesh/UVAtlas.cpp alicevision.new/src/aliceVision/mesh/UVAtlas.cpp
--- alicevision/src/aliceVision/mesh/UVAtlas.cpp	2022-02-11 12:37:11.891151611 +0100
+++ alicevision.new/src/aliceVision/mesh/UVAtlas.cpp	2022-02-11 12:37:30.409283311 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2017 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/multiview/relativePose/Fundamental10PSolver.cpp alicevision.new/src/aliceVision/multiview/relativePose/Fundamental10PSolver.cpp
--- alicevision/src/aliceVision/multiview/relativePose/Fundamental10PSolver.cpp	2022-02-11 12:37:07.758122219 +0100
+++ alicevision.new/src/aliceVision/multiview/relativePose/Fundamental10PSolver.cpp	2022-02-11 12:37:30.409283311 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2018 AliceVision contributors.
 // Copyright (c) 2015 Jan Heller <hellej1@cmp.felk.cvut.cz>, Zuzana Kukelova <zukuke@microsoft.com>
diff -Naur alicevision/src/aliceVision/multiview/resection/EPnPSolver.cpp alicevision.new/src/aliceVision/multiview/resection/EPnPSolver.cpp
--- alicevision/src/aliceVision/multiview/resection/EPnPSolver.cpp	2022-02-11 12:37:07.760122233 +0100
+++ alicevision.new/src/aliceVision/multiview/resection/EPnPSolver.cpp	2022-02-11 12:37:30.409283311 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/multiview/resection/P4PfSolver.cpp alicevision.new/src/aliceVision/multiview/resection/P4PfSolver.cpp
--- alicevision/src/aliceVision/multiview/resection/P4PfSolver.cpp	2022-02-11 12:37:07.760122233 +0100
+++ alicevision.new/src/aliceVision/multiview/resection/P4PfSolver.cpp	2022-02-11 12:37:30.410283319 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/multiview/rotationAveraging/l1.cpp alicevision.new/src/aliceVision/multiview/rotationAveraging/l1.cpp
--- alicevision/src/aliceVision/multiview/rotationAveraging/l1.cpp	2022-02-11 12:37:07.762122247 +0100
+++ alicevision.new/src/aliceVision/multiview/rotationAveraging/l1.cpp	2022-02-11 12:37:30.409283311 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/multiview/rotationAveraging/rotationAveraging_test.cpp alicevision.new/src/aliceVision/multiview/rotationAveraging/rotationAveraging_test.cpp
--- alicevision/src/aliceVision/multiview/rotationAveraging/rotationAveraging_test.cpp	2022-02-11 12:37:07.762122247 +0100
+++ alicevision.new/src/aliceVision/multiview/rotationAveraging/rotationAveraging_test.cpp	2022-02-11 12:37:30.409283311 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/multiview/triangulation/Triangulation.cpp alicevision.new/src/aliceVision/multiview/triangulation/Triangulation.cpp
--- alicevision/src/aliceVision/multiview/triangulation/Triangulation.cpp	2022-02-11 12:37:07.764122261 +0100
+++ alicevision.new/src/aliceVision/multiview/triangulation/Triangulation.cpp	2022-02-11 12:37:30.409283311 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/panorama/coordinatesMap.cpp alicevision.new/src/aliceVision/panorama/coordinatesMap.cpp
--- alicevision/src/aliceVision/panorama/coordinatesMap.cpp	2022-02-11 12:37:11.897151654 +0100
+++ alicevision.new/src/aliceVision/panorama/coordinatesMap.cpp	2022-02-11 12:37:30.403283269 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 #include "coordinatesMap.hpp"
 
 #include "sphericalMapping.hpp"
diff -Naur alicevision/src/aliceVision/panorama/graphcut.hpp alicevision.new/src/aliceVision/panorama/graphcut.hpp
--- alicevision/src/aliceVision/panorama/graphcut.hpp	2022-02-11 12:37:07.773122325 +0100
+++ alicevision.new/src/aliceVision/panorama/graphcut.hpp	2022-02-11 12:37:30.403283269 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 #pragma once
 
 #include <boost/graph/adjacency_list.hpp>
diff -Naur alicevision/src/aliceVision/rig/Rig.cpp alicevision.new/src/aliceVision/rig/Rig.cpp
--- alicevision/src/aliceVision/rig/Rig.cpp	2022-02-11 12:37:11.898151661 +0100
+++ alicevision.new/src/aliceVision/rig/Rig.cpp	2022-02-11 12:37:30.404283276 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/robustEstimation/ACRansac.hpp alicevision.new/src/aliceVision/robustEstimation/ACRansac.hpp
--- alicevision/src/aliceVision/robustEstimation/ACRansac.hpp	2022-02-11 12:37:07.775122340 +0100
+++ alicevision.new/src/aliceVision/robustEstimation/ACRansac.hpp	2022-02-11 12:37:30.404283276 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/robustEstimation/estimators.hpp alicevision.new/src/aliceVision/robustEstimation/estimators.hpp
--- alicevision/src/aliceVision/robustEstimation/estimators.hpp	2022-02-11 12:37:07.777122354 +0100
+++ alicevision.new/src/aliceVision/robustEstimation/estimators.hpp	2022-02-11 12:37:30.404283276 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/robustEstimation/leastMedianOfSquares.hpp alicevision.new/src/aliceVision/robustEstimation/leastMedianOfSquares.hpp
--- alicevision/src/aliceVision/robustEstimation/leastMedianOfSquares.hpp	2022-02-11 12:37:07.777122354 +0100
+++ alicevision.new/src/aliceVision/robustEstimation/leastMedianOfSquares.hpp	2022-02-11 12:37:30.404283276 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/robustEstimation/leastMedianOfSquares_test.cpp alicevision.new/src/aliceVision/robustEstimation/leastMedianOfSquares_test.cpp
--- alicevision/src/aliceVision/robustEstimation/leastMedianOfSquares_test.cpp	2022-02-11 12:37:07.778122361 +0100
+++ alicevision.new/src/aliceVision/robustEstimation/leastMedianOfSquares_test.cpp	2022-02-11 12:37:30.404283276 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/robustEstimation/lineTestGenerator.hpp alicevision.new/src/aliceVision/robustEstimation/lineTestGenerator.hpp
--- alicevision/src/aliceVision/robustEstimation/lineTestGenerator.hpp	2022-02-11 12:37:07.778122361 +0100
+++ alicevision.new/src/aliceVision/robustEstimation/lineTestGenerator.hpp	2022-02-11 12:37:30.404283276 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/robustEstimation/LORansac.hpp alicevision.new/src/aliceVision/robustEstimation/LORansac.hpp
--- alicevision/src/aliceVision/robustEstimation/LORansac.hpp	2022-02-11 12:37:11.898151661 +0100
+++ alicevision.new/src/aliceVision/robustEstimation/LORansac.hpp	2022-02-11 12:37:30.404283276 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/robustEstimation/loRansac_test.cpp alicevision.new/src/aliceVision/robustEstimation/loRansac_test.cpp
--- alicevision/src/aliceVision/robustEstimation/loRansac_test.cpp	2022-02-11 12:37:07.778122361 +0100
+++ alicevision.new/src/aliceVision/robustEstimation/loRansac_test.cpp	2022-02-11 12:37:30.404283276 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/sfm/liealgebra.hpp alicevision.new/src/aliceVision/sfm/liealgebra.hpp
--- alicevision/src/aliceVision/sfm/liealgebra.hpp	2022-02-11 12:37:11.902151690 +0100
+++ alicevision.new/src/aliceVision/sfm/liealgebra.hpp	2022-02-11 12:37:30.405283283 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 #pragma once
 
 #include <Eigen/Dense>
diff -Naur alicevision/src/aliceVision/sfm/pipeline/global/GlobalSfMTranslationAveragingSolver.cpp alicevision.new/src/aliceVision/sfm/pipeline/global/GlobalSfMTranslationAveragingSolver.cpp
--- alicevision/src/aliceVision/sfm/pipeline/global/GlobalSfMTranslationAveragingSolver.cpp	2022-02-11 12:37:11.903151697 +0100
+++ alicevision.new/src/aliceVision/sfm/pipeline/global/GlobalSfMTranslationAveragingSolver.cpp	2022-02-11 12:37:30.405283283 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/sfm/pipeline/localization/SfMLocalizer.cpp alicevision.new/src/aliceVision/sfm/pipeline/localization/SfMLocalizer.cpp
--- alicevision/src/aliceVision/sfm/pipeline/localization/SfMLocalizer.cpp	2022-02-11 12:37:07.786122418 +0100
+++ alicevision.new/src/aliceVision/sfm/pipeline/localization/SfMLocalizer.cpp	2022-02-11 12:37:30.405283283 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/sfm/pipeline/localization/SfMLocalizer.hpp alicevision.new/src/aliceVision/sfm/pipeline/localization/SfMLocalizer.hpp
--- alicevision/src/aliceVision/sfm/pipeline/localization/SfMLocalizer.hpp	2022-02-11 12:37:11.903151697 +0100
+++ alicevision.new/src/aliceVision/sfm/pipeline/localization/SfMLocalizer.hpp	2022-02-11 12:37:30.405283283 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/sfm/pipeline/panorama/ReconstructionEngine_panorama.cpp alicevision.new/src/aliceVision/sfm/pipeline/panorama/ReconstructionEngine_panorama.cpp
--- alicevision/src/aliceVision/sfm/pipeline/panorama/ReconstructionEngine_panorama.cpp	2022-02-11 12:37:11.903151697 +0100
+++ alicevision.new/src/aliceVision/sfm/pipeline/panorama/ReconstructionEngine_panorama.cpp	2022-02-11 12:37:30.405283283 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/sfm/pipeline/panorama/ReconstructionEngine_panorama.hpp alicevision.new/src/aliceVision/sfm/pipeline/panorama/ReconstructionEngine_panorama.hpp
--- alicevision/src/aliceVision/sfm/pipeline/panorama/ReconstructionEngine_panorama.hpp	2022-02-11 12:37:07.787122425 +0100
+++ alicevision.new/src/aliceVision/sfm/pipeline/panorama/ReconstructionEngine_panorama.hpp	2022-02-11 12:37:30.405283283 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/sfm/pipeline/RelativePoseInfo.hpp alicevision.new/src/aliceVision/sfm/pipeline/RelativePoseInfo.hpp
--- alicevision/src/aliceVision/sfm/pipeline/RelativePoseInfo.hpp	2022-02-11 12:37:07.783122397 +0100
+++ alicevision.new/src/aliceVision/sfm/pipeline/RelativePoseInfo.hpp	2022-02-11 12:37:30.405283283 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.hpp alicevision.new/src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.hpp
--- alicevision/src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.hpp	2022-02-11 12:37:11.905151711 +0100
+++ alicevision.new/src/aliceVision/sfm/pipeline/sequential/ReconstructionEngine_sequentialSfM.hpp	2022-02-11 12:37:30.406283290 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/sfm/sfmTriangulation.cpp alicevision.new/src/aliceVision/sfm/sfmTriangulation.cpp
--- alicevision/src/aliceVision/sfm/sfmTriangulation.cpp	2022-02-11 12:37:11.906151718 +0100
+++ alicevision.new/src/aliceVision/sfm/sfmTriangulation.cpp	2022-02-11 12:37:30.405283283 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/sfm/utils/alignment.cpp alicevision.new/src/aliceVision/sfm/utils/alignment.cpp
--- alicevision/src/aliceVision/sfm/utils/alignment.cpp	2022-02-11 12:37:11.906151718 +0100
+++ alicevision.new/src/aliceVision/sfm/utils/alignment.cpp	2022-02-11 12:37:30.406283290 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/stl/DynamicBitset.hpp alicevision.new/src/aliceVision/stl/DynamicBitset.hpp
--- alicevision/src/aliceVision/stl/DynamicBitset.hpp	2022-02-11 12:37:07.809122581 +0100
+++ alicevision.new/src/aliceVision/stl/DynamicBitset.hpp	2022-02-11 12:37:30.409283311 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/aliceVision/system/MemoryInfo.cpp alicevision.new/src/aliceVision/system/MemoryInfo.cpp
--- alicevision/src/aliceVision/system/MemoryInfo.cpp	2022-02-11 12:37:07.811122596 +0100
+++ alicevision.new/src/aliceVision/system/MemoryInfo.cpp	2022-02-11 12:37:30.410283319 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/types.hpp alicevision.new/src/aliceVision/types.hpp
--- alicevision/src/aliceVision/types.hpp	2022-02-11 12:37:07.813122610 +0100
+++ alicevision.new/src/aliceVision/types.hpp	2022-02-11 12:37:30.406283290 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/voctree/kmeans_test.cpp alicevision.new/src/aliceVision/voctree/kmeans_test.cpp
--- alicevision/src/aliceVision/voctree/kmeans_test.cpp	2022-02-11 12:37:07.817122638 +0100
+++ alicevision.new/src/aliceVision/voctree/kmeans_test.cpp	2022-02-11 12:37:30.402283262 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/voctree/SimpleKmeans.hpp alicevision.new/src/aliceVision/voctree/SimpleKmeans.hpp
--- alicevision/src/aliceVision/voctree/SimpleKmeans.hpp	2022-02-11 12:37:07.816122631 +0100
+++ alicevision.new/src/aliceVision/voctree/SimpleKmeans.hpp	2022-02-11 12:37:30.402283262 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/aliceVision/voctree/VocabularyTree.hpp alicevision.new/src/aliceVision/voctree/VocabularyTree.hpp
--- alicevision/src/aliceVision/voctree/VocabularyTree.hpp	2022-02-11 12:37:07.816122631 +0100
+++ alicevision.new/src/aliceVision/voctree/VocabularyTree.hpp	2022-02-11 12:37:30.402283262 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2016 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/dependencies/flann/src/cpp/flann/algorithms/kdtree_cuda_3d_index.cu alicevision.new/src/dependencies/flann/src/cpp/flann/algorithms/kdtree_cuda_3d_index.cu
--- alicevision/src/dependencies/flann/src/cpp/flann/algorithms/kdtree_cuda_3d_index.cu	2022-02-11 12:37:07.823122681 +0100
+++ alicevision.new/src/dependencies/flann/src/cpp/flann/algorithms/kdtree_cuda_3d_index.cu	2022-02-11 12:37:30.389283169 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /**********************************************************************
  * Software License Agreement (BSD License)
  *
diff -Naur alicevision/src/dependencies/flann/src/cpp/flann/algorithms/kmeans_index.h alicevision.new/src/dependencies/flann/src/cpp/flann/algorithms/kmeans_index.h
--- alicevision/src/dependencies/flann/src/cpp/flann/algorithms/kmeans_index.h	2022-02-11 12:37:07.824122688 +0100
+++ alicevision.new/src/dependencies/flann/src/cpp/flann/algorithms/kmeans_index.h	2022-02-11 12:37:30.389283169 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /***********************************************************************
  * Software License Agreement (BSD License)
  *
diff -Naur alicevision/src/dependencies/flann/src/cpp/flann/algorithms/lsh_index.h alicevision.new/src/dependencies/flann/src/cpp/flann/algorithms/lsh_index.h
--- alicevision/src/dependencies/flann/src/cpp/flann/algorithms/lsh_index.h	2022-02-11 12:37:07.825122695 +0100
+++ alicevision.new/src/dependencies/flann/src/cpp/flann/algorithms/lsh_index.h	2022-02-11 12:37:30.389283169 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /***********************************************************************
  * Software License Agreement (BSD License)
  *
diff -Naur alicevision/src/dependencies/flann/src/cpp/flann/algorithms/nn_index.h alicevision.new/src/dependencies/flann/src/cpp/flann/algorithms/nn_index.h
--- alicevision/src/dependencies/flann/src/cpp/flann/algorithms/nn_index.h	2022-02-11 12:37:07.825122695 +0100
+++ alicevision.new/src/dependencies/flann/src/cpp/flann/algorithms/nn_index.h	2022-02-11 12:37:30.389283169 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /***********************************************************************
  * Software License Agreement (BSD License)
  *
diff -Naur alicevision/src/dependencies/flann/src/cpp/flann/util/result_set.h alicevision.new/src/dependencies/flann/src/cpp/flann/util/result_set.h
--- alicevision/src/dependencies/flann/src/cpp/flann/util/result_set.h	2022-02-11 12:37:07.828122717 +0100
+++ alicevision.new/src/dependencies/flann/src/cpp/flann/util/result_set.h	2022-02-11 12:37:30.390283176 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /***********************************************************************
  * Software License Agreement (BSD License)
  *
diff -Naur alicevision/src/dependencies/lemon/lemon/bellman_ford.h alicevision.new/src/dependencies/lemon/lemon/bellman_ford.h
--- alicevision/src/dependencies/lemon/lemon/bellman_ford.h	2022-02-11 12:37:07.831122738 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/bellman_ford.h	2022-02-11 12:37:30.390283176 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/capacity_scaling.h alicevision.new/src/dependencies/lemon/lemon/capacity_scaling.h
--- alicevision/src/dependencies/lemon/lemon/capacity_scaling.h	2022-02-11 12:37:07.834122759 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/capacity_scaling.h	2022-02-11 12:37:30.390283176 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/circulation.h alicevision.new/src/dependencies/lemon/lemon/circulation.h
--- alicevision/src/dependencies/lemon/lemon/circulation.h	2022-02-11 12:37:07.834122759 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/circulation.h	2022-02-11 12:37:30.390283176 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/cost_scaling.h alicevision.new/src/dependencies/lemon/lemon/cost_scaling.h
--- alicevision/src/dependencies/lemon/lemon/cost_scaling.h	2022-02-11 12:37:07.837122780 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/cost_scaling.h	2022-02-11 12:37:30.391283183 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/cycle_canceling.h alicevision.new/src/dependencies/lemon/lemon/cycle_canceling.h
--- alicevision/src/dependencies/lemon/lemon/cycle_canceling.h	2022-02-11 12:37:07.837122780 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/cycle_canceling.h	2022-02-11 12:37:30.391283183 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/dimacs.h alicevision.new/src/dependencies/lemon/lemon/dimacs.h
--- alicevision/src/dependencies/lemon/lemon/dimacs.h	2022-02-11 12:37:07.838122788 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/dimacs.h	2022-02-11 12:37:30.391283183 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/fractional_matching.h alicevision.new/src/dependencies/lemon/lemon/fractional_matching.h
--- alicevision/src/dependencies/lemon/lemon/fractional_matching.h	2022-02-11 12:37:07.839122795 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/fractional_matching.h	2022-02-11 12:37:30.391283183 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/gomory_hu.h alicevision.new/src/dependencies/lemon/lemon/gomory_hu.h
--- alicevision/src/dependencies/lemon/lemon/gomory_hu.h	2022-02-11 12:37:07.840122802 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/gomory_hu.h	2022-02-11 12:37:30.391283183 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/grosso_locatelli_pullan_mc.h alicevision.new/src/dependencies/lemon/lemon/grosso_locatelli_pullan_mc.h
--- alicevision/src/dependencies/lemon/lemon/grosso_locatelli_pullan_mc.h	2022-02-11 12:37:07.840122802 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/grosso_locatelli_pullan_mc.h	2022-02-11 12:37:30.392283190 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/hao_orlin.h alicevision.new/src/dependencies/lemon/lemon/hao_orlin.h
--- alicevision/src/dependencies/lemon/lemon/hao_orlin.h	2022-02-11 12:37:07.841122809 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/hao_orlin.h	2022-02-11 12:37:30.392283190 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/hartmann_orlin_mmc.h alicevision.new/src/dependencies/lemon/lemon/hartmann_orlin_mmc.h
--- alicevision/src/dependencies/lemon/lemon/hartmann_orlin_mmc.h	2022-02-11 12:37:07.841122809 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/hartmann_orlin_mmc.h	2022-02-11 12:37:30.392283190 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/howard_mmc.h alicevision.new/src/dependencies/lemon/lemon/howard_mmc.h
--- alicevision/src/dependencies/lemon/lemon/howard_mmc.h	2022-02-11 12:37:07.841122809 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/howard_mmc.h	2022-02-11 12:37:30.392283190 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/karp_mmc.h alicevision.new/src/dependencies/lemon/lemon/karp_mmc.h
--- alicevision/src/dependencies/lemon/lemon/karp_mmc.h	2022-02-11 12:37:07.841122809 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/karp_mmc.h	2022-02-11 12:37:30.393283198 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/lp_base.cc alicevision.new/src/dependencies/lemon/lemon/lp_base.cc
--- alicevision/src/dependencies/lemon/lemon/lp_base.cc	2022-02-11 12:37:07.843122823 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/lp_base.cc	2022-02-11 12:37:30.393283198 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/matching.h alicevision.new/src/dependencies/lemon/lemon/matching.h
--- alicevision/src/dependencies/lemon/lemon/matching.h	2022-02-11 12:37:07.844122830 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/matching.h	2022-02-11 12:37:30.393283198 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/nagamochi_ibaraki.h alicevision.new/src/dependencies/lemon/lemon/nagamochi_ibaraki.h
--- alicevision/src/dependencies/lemon/lemon/nagamochi_ibaraki.h	2022-02-11 12:37:07.844122830 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/nagamochi_ibaraki.h	2022-02-11 12:37:30.393283198 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/network_simplex.h alicevision.new/src/dependencies/lemon/lemon/network_simplex.h
--- alicevision/src/dependencies/lemon/lemon/network_simplex.h	2022-02-11 12:37:07.845122837 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/network_simplex.h	2022-02-11 12:37:30.394283205 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/radix_sort.h alicevision.new/src/dependencies/lemon/lemon/radix_sort.h
--- alicevision/src/dependencies/lemon/lemon/radix_sort.h	2022-02-11 12:37:07.846122844 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/radix_sort.h	2022-02-11 12:37:30.394283205 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/lemon/random.h alicevision.new/src/dependencies/lemon/lemon/random.h
--- alicevision/src/dependencies/lemon/lemon/random.h	2022-02-11 12:37:07.846122844 +0100
+++ alicevision.new/src/dependencies/lemon/lemon/random.h	2022-02-11 12:37:30.394283205 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* -*- mode: C++; indent-tabs-mode: nil; -*-
  *
  * This file is a part of LEMON, a generic C++ optimization library.
diff -Naur alicevision/src/dependencies/lemon/NEWS alicevision.new/src/dependencies/lemon/NEWS
--- alicevision/src/dependencies/lemon/NEWS	2022-02-11 12:37:07.829122724 +0100
+++ alicevision.new/src/dependencies/lemon/NEWS	2022-02-11 12:37:30.390283176 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 2013-08-10 Version 1.3 released
 
         This is major feature release
diff -Naur alicevision/src/dependencies/MeshSDFilter/external/OpenMesh/Core/IO/OMFormat.hh alicevision.new/src/dependencies/MeshSDFilter/external/OpenMesh/Core/IO/OMFormat.hh
--- alicevision/src/dependencies/MeshSDFilter/external/OpenMesh/Core/IO/OMFormat.hh	2022-02-11 12:37:11.304147437 +0100
+++ alicevision.new/src/dependencies/MeshSDFilter/external/OpenMesh/Core/IO/OMFormat.hh	2022-02-11 12:37:30.388283162 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* ========================================================================= *
  *                                                                           *
  *                               OpenMesh                                    *
diff -Naur alicevision/src/dependencies/MeshSDFilter/external/OpenMesh/Core/IO/SR_binary_spec.hh alicevision.new/src/dependencies/MeshSDFilter/external/OpenMesh/Core/IO/SR_binary_spec.hh
--- alicevision/src/dependencies/MeshSDFilter/external/OpenMesh/Core/IO/SR_binary_spec.hh	2022-02-11 12:37:11.305147444 +0100
+++ alicevision.new/src/dependencies/MeshSDFilter/external/OpenMesh/Core/IO/SR_binary_spec.hh	2022-02-11 12:37:30.388283162 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* ========================================================================= *
  *                                                                           *
  *                               OpenMesh                                    *
diff -Naur alicevision/src/dependencies/MeshSDFilter/external/OpenMesh/Core/IO/writer/OBJWriter.cc alicevision.new/src/dependencies/MeshSDFilter/external/OpenMesh/Core/IO/writer/OBJWriter.cc
--- alicevision/src/dependencies/MeshSDFilter/external/OpenMesh/Core/IO/writer/OBJWriter.cc	2022-02-11 12:37:11.307147458 +0100
+++ alicevision.new/src/dependencies/MeshSDFilter/external/OpenMesh/Core/IO/writer/OBJWriter.cc	2022-02-11 12:37:30.388283162 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* ========================================================================= *
  *                                                                           *
  *                               OpenMesh                                    *
diff -Naur alicevision/src/dependencies/MeshSDFilter/external/OpenMesh/Core/Utils/PropertyManager.hh alicevision.new/src/dependencies/MeshSDFilter/external/OpenMesh/Core/Utils/PropertyManager.hh
--- alicevision/src/dependencies/MeshSDFilter/external/OpenMesh/Core/Utils/PropertyManager.hh	2022-02-11 12:37:11.313147501 +0100
+++ alicevision.new/src/dependencies/MeshSDFilter/external/OpenMesh/Core/Utils/PropertyManager.hh	2022-02-11 12:37:30.389283169 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* ========================================================================= *
  *                                                                           *
  *                               OpenMesh                                    *
diff -Naur alicevision/src/dependencies/MeshSDFilter/external/OpenMesh/Tools/Utils/NumLimitsT.hh alicevision.new/src/dependencies/MeshSDFilter/external/OpenMesh/Tools/Utils/NumLimitsT.hh
--- alicevision/src/dependencies/MeshSDFilter/external/OpenMesh/Tools/Utils/NumLimitsT.hh	2022-02-11 12:37:11.319147543 +0100
+++ alicevision.new/src/dependencies/MeshSDFilter/external/OpenMesh/Tools/Utils/NumLimitsT.hh	2022-02-11 12:37:30.388283162 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* ========================================================================= *
  *                                                                           *
  *                               OpenMesh                                    *
diff -Naur alicevision/src/dependencies/nanoflann/include/nanoflann.hpp alicevision.new/src/dependencies/nanoflann/include/nanoflann.hpp
--- alicevision/src/dependencies/nanoflann/include/nanoflann.hpp	2022-02-11 12:37:11.417148241 +0100
+++ alicevision.new/src/dependencies/nanoflann/include/nanoflann.hpp	2022-02-11 12:37:30.401283255 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /***********************************************************************
  * Software License Agreement (BSD License)
  *
diff -Naur alicevision/src/dependencies/nanoflann/tests/gtest-1.7.0/fused-src/gtest/gtest-all.cc alicevision.new/src/dependencies/nanoflann/tests/gtest-1.7.0/fused-src/gtest/gtest-all.cc
--- alicevision/src/dependencies/nanoflann/tests/gtest-1.7.0/fused-src/gtest/gtest-all.cc	2022-02-11 12:37:11.428148319 +0100
+++ alicevision.new/src/dependencies/nanoflann/tests/gtest-1.7.0/fused-src/gtest/gtest-all.cc	2022-02-11 12:37:30.396283219 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // Copyright 2008, Google Inc.
 // All rights reserved.
 //
diff -Naur alicevision/src/dependencies/nanoflann/tests/gtest-1.7.0/fused-src/gtest/gtest.h alicevision.new/src/dependencies/nanoflann/tests/gtest-1.7.0/fused-src/gtest/gtest.h
--- alicevision/src/dependencies/nanoflann/tests/gtest-1.7.0/fused-src/gtest/gtest.h	2022-02-11 12:37:11.431148340 +0100
+++ alicevision.new/src/dependencies/nanoflann/tests/gtest-1.7.0/fused-src/gtest/gtest.h	2022-02-11 12:37:30.399283240 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // Copyright 2005, Google Inc.
 // All rights reserved.
 //
diff -Naur alicevision/src/dependencies/nanoflann/tests/gtest-1.7.0/include/gtest/gtest.h alicevision.new/src/dependencies/nanoflann/tests/gtest-1.7.0/include/gtest/gtest.h
--- alicevision/src/dependencies/nanoflann/tests/gtest-1.7.0/include/gtest/gtest.h	2022-02-11 12:37:11.435148369 +0100
+++ alicevision.new/src/dependencies/nanoflann/tests/gtest-1.7.0/include/gtest/gtest.h	2022-02-11 12:37:30.399283240 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // Copyright 2005, Google Inc.
 // All rights reserved.
 //
diff -Naur alicevision/src/dependencies/nanoflann/tests/gtest-1.7.0/include/gtest/internal/gtest-internal.h alicevision.new/src/dependencies/nanoflann/tests/gtest-1.7.0/include/gtest/internal/gtest-internal.h
--- alicevision/src/dependencies/nanoflann/tests/gtest-1.7.0/include/gtest/internal/gtest-internal.h	2022-02-11 12:37:11.438148390 +0100
+++ alicevision.new/src/dependencies/nanoflann/tests/gtest-1.7.0/include/gtest/internal/gtest-internal.h	2022-02-11 12:37:30.399283240 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // Copyright 2005, Google Inc.
 // All rights reserved.
 //
diff -Naur alicevision/src/dependencies/nanoflann/tests/gtest-1.7.0/src/gtest.cc alicevision.new/src/dependencies/nanoflann/tests/gtest-1.7.0/src/gtest.cc
--- alicevision/src/dependencies/nanoflann/tests/gtest-1.7.0/src/gtest.cc	2022-02-11 12:37:11.447148454 +0100
+++ alicevision.new/src/dependencies/nanoflann/tests/gtest-1.7.0/src/gtest.cc	2022-02-11 12:37:30.400283247 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // Copyright 2005, Google Inc.
 // All rights reserved.
 //
diff -Naur alicevision/src/dependencies/nanoflann/tests/gtest-1.7.0/test/gtest-death-test_test.cc alicevision.new/src/dependencies/nanoflann/tests/gtest-1.7.0/test/gtest-death-test_test.cc
--- alicevision/src/dependencies/nanoflann/tests/gtest-1.7.0/test/gtest-death-test_test.cc	2022-02-11 12:37:11.448148461 +0100
+++ alicevision.new/src/dependencies/nanoflann/tests/gtest-1.7.0/test/gtest-death-test_test.cc	2022-02-11 12:37:30.401283255 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // Copyright 2005, Google Inc.
 // All rights reserved.
 //
diff -Naur alicevision/src/dependencies/nanoflann/tests/test_main.cpp alicevision.new/src/dependencies/nanoflann/tests/test_main.cpp
--- alicevision/src/dependencies/nanoflann/tests/test_main.cpp	2022-02-11 12:37:11.457148525 +0100
+++ alicevision.new/src/dependencies/nanoflann/tests/test_main.cpp	2022-02-11 12:37:30.394283205 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /***********************************************************************
  * Software License Agreement (BSD License)
  *
diff -Naur alicevision/src/dependencies/osi_clp/CoinUtils/src/CoinFinite.hpp alicevision.new/src/dependencies/osi_clp/CoinUtils/src/CoinFinite.hpp
--- alicevision/src/dependencies/osi_clp/CoinUtils/src/CoinFinite.hpp	2022-02-11 12:37:11.600149542 +0100
+++ alicevision.new/src/dependencies/osi_clp/CoinUtils/src/CoinFinite.hpp	2022-02-11 12:37:30.388283162 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 /* $Id: CoinFinite.hpp 1423 2011-04-30 10:17:48Z stefan $ */
 // Copyright (C) 2002, International Business Machines
 // Corporation and others.  All Rights Reserved.
diff -Naur alicevision/src/samples/accv12Demo/main_openCVFitting.cpp alicevision.new/src/samples/accv12Demo/main_openCVFitting.cpp
--- alicevision/src/samples/accv12Demo/main_openCVFitting.cpp	2022-02-11 12:37:07.853122894 +0100
+++ alicevision.new/src/samples/accv12Demo/main_openCVFitting.cpp	2022-02-11 12:37:30.401283255 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2017 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/samples/robustEssentialSpherical/main_robustEssentialSpherical.cpp alicevision.new/src/samples/robustEssentialSpherical/main_robustEssentialSpherical.cpp
--- alicevision/src/samples/robustEssentialSpherical/main_robustEssentialSpherical.cpp	2022-02-11 12:37:07.866122987 +0100
+++ alicevision.new/src/samples/robustEssentialSpherical/main_robustEssentialSpherical.cpp	2022-02-11 12:37:30.401283255 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2017 AliceVision contributors.
 // Copyright (c) 2014 openMVG contributors.
diff -Naur alicevision/src/samples/robustHomography/main_robustHomography.cpp alicevision.new/src/samples/robustHomography/main_robustHomography.cpp
--- alicevision/src/samples/robustHomography/main_robustHomography.cpp	2022-02-11 12:37:07.867122994 +0100
+++ alicevision.new/src/samples/robustHomography/main_robustHomography.cpp	2022-02-11 12:37:30.401283255 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2017 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/samples/robustHomographyGuided/main_robustHomographyGuided.cpp alicevision.new/src/samples/robustHomographyGuided/main_robustHomographyGuided.cpp
--- alicevision/src/samples/robustHomographyGuided/main_robustHomographyGuided.cpp	2022-02-11 12:37:07.867122994 +0100
+++ alicevision.new/src/samples/robustHomographyGuided/main_robustHomographyGuided.cpp	2022-02-11 12:37:30.401283255 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2017 AliceVision contributors.
 // Copyright (c) 2012 openMVG contributors.
diff -Naur alicevision/src/software/pipeline/main_panoramaCompositing.cpp alicevision.new/src/software/pipeline/main_panoramaCompositing.cpp
--- alicevision/src/software/pipeline/main_panoramaCompositing.cpp	2022-02-11 12:37:11.916151789 +0100
+++ alicevision.new/src/software/pipeline/main_panoramaCompositing.cpp	2022-02-11 12:37:30.387283155 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2020 AliceVision contributors.
 // This Source Code Form is subject to the terms of the Mozilla Public License,
diff -Naur alicevision/src/software/pipeline/main_panoramaInit.cpp alicevision.new/src/software/pipeline/main_panoramaInit.cpp
--- alicevision/src/software/pipeline/main_panoramaInit.cpp	2022-02-11 12:37:11.917151796 +0100
+++ alicevision.new/src/software/pipeline/main_panoramaInit.cpp	2022-02-11 12:37:30.388283162 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 #include <aliceVision/image/all.hpp>
 #include <aliceVision/system/cmdline.hpp>
 #include <aliceVision/system/Logger.hpp>
diff -Naur alicevision/src/software/utils/main_sfmLocalization.cpp alicevision.new/src/software/utils/main_sfmLocalization.cpp
--- alicevision/src/software/utils/main_sfmLocalization.cpp	2022-02-11 12:37:07.879123079 +0100
+++ alicevision.new/src/software/utils/main_sfmLocalization.cpp	2022-02-11 12:37:30.388283162 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2017 AliceVision contributors.
 // Copyright (c) 2015 openMVG contributors.
diff -Naur alicevision/src/software/utils/sfmColorHarmonize/colorHarmonizeEngineGlobal.cpp alicevision.new/src/software/utils/sfmColorHarmonize/colorHarmonizeEngineGlobal.cpp
--- alicevision/src/software/utils/sfmColorHarmonize/colorHarmonizeEngineGlobal.cpp	2022-02-11 12:37:11.919151811 +0100
+++ alicevision.new/src/software/utils/sfmColorHarmonize/colorHarmonizeEngineGlobal.cpp	2022-02-11 12:37:30.388283162 +0100
@@ -1,3 +1,4 @@
+#include <limits>
 // This file is part of the AliceVision project.
 // Copyright (c) 2017 AliceVision contributors.
 // Copyright (c) 2013 openMVG contributors.
