diff -Naur nufraw-0.43-3/nufraw.desktop nufraw-0.43-3.new/nufraw.desktop
--- nufraw-0.43-3/nufraw.desktop	1970-01-01 01:00:00.000000000 +0100
+++ nufraw-0.43-3.new/nufraw.desktop	2022-12-08 21:40:16.770622244 +0100
@@ -0,0 +1,20 @@
+[Desktop Entry]
+Name=nUFRaw
+GenericName=Raw image converter
+
+Comment=A utility to read and manipulate raw images from digital cameras
+
+Version=1.0
+
+Type=Application
+Categories=Graphics;Photography;GTK;
+
+Exec=nufraw %U
+TryExec=nufraw
+Terminal=false
+StartupNotify=true
+
+MimeType=application/x-nufraw;image/x-dcraw;
+
+Icon=nufraw
+Name[en_US]=nufraw.desktop
diff -Naur nufraw-0.43-3/nufraw.desktop.desktop nufraw-0.43-3.new/nufraw.desktop.desktop
--- nufraw-0.43-3/nufraw.desktop.desktop	2016-09-05 16:24:50.000000000 +0200
+++ nufraw-0.43-3.new/nufraw.desktop.desktop	1970-01-01 01:00:00.000000000 +0100
@@ -1,20 +0,0 @@
-[Desktop Entry]
-Name=nUFRaw
-GenericName=Raw image converter
-
-Comment=A utility to read and manipulate raw images from digital cameras
-
-Version=1.0
-
-Type=Application
-Categories=Graphics;Photography;GTK;
-
-Exec=nufraw %U
-TryExec=nufraw
-Terminal=false
-StartupNotify=true
-
-MimeType=application/x-nufraw;image/x-dcraw;
-
-Icon=nufraw
-Name[en_US]=nufraw.desktop
