diff -u nufraw-0.43-3/nufraw_exiv2.cc nufraw-0.43-3-r1/nufraw_exiv2.cc
--- nufraw-0.43-3/nufraw_exiv2.cc	2020-02-29 16:45:02.000000000 +0100
+++ nufraw-0.43-3-r1/nufraw_exiv2.cc	2023-09-06 02:01:48.655571418 +0200
@@ -54,7 +54,7 @@
         uf->inputExifBuf = NULL;
         uf->inputExifBufLen = 0;
 
-        Exiv2::Image::AutoPtr image;
+        Exiv2::Image::UniquePtr image;
         if (uf->unzippedBuf != NULL) {
             image = Exiv2::ImageFactory::open(
                         (const Exiv2::byte*)uf->unzippedBuf, uf->unzippedBufLen);
@@ -135,7 +135,7 @@
         nufraw_message(NUFRAW_SET_LOG, "%s\n", stderror.str().c_str());
 
         return NUFRAW_SUCCESS;
-    } catch (Exiv2::AnyError& e) {
+    } catch (Exiv2::Error& e) {
         std::cerr.rdbuf(savecerr);
         std::string s(e.what());
         nufraw_message(NUFRAW_SET_WARNING, "%s\n", s.c_str());
@@ -156,7 +156,7 @@
         if ((pos = exifData.findKey(Exiv2::ExifKey("Exif.Image.Orientation")))
                 != exifData.end()) {
             nufraw_message(NUFRAW_SET_LOG, "Resetting %s from '%d' to '1'\n",
-                          pos->key().c_str(), pos->value().toLong());
+                          pos->key().c_str(), pos->value().toInt64());
             pos->setValue("1"); /* 1 = Normal orientation */
         }
     }
@@ -327,7 +327,7 @@
         nufraw_message(NUFRAW_SET_LOG, "%s\n", stderror.str().c_str());
 
         return NUFRAW_SUCCESS;
-    } catch (Exiv2::AnyError& e) {
+    } catch (Exiv2::Error& e) {
         std::cerr.rdbuf(savecerr);
         std::string s(e.what());
         nufraw_message(NUFRAW_SET_WARNING, "%s\n", s.c_str());
@@ -347,7 +347,7 @@
 
         char *filename =
             uf_win32_locale_filename_from_utf8(uf->conf->outputFilename);
-        Exiv2::Image::AutoPtr image = Exiv2::ImageFactory::open(filename);
+        Exiv2::Image::UniquePtr image = Exiv2::ImageFactory::open(filename);
         uf_win32_locale_filename_free(filename);
         assert(image.get() != 0);
 
@@ -367,7 +367,7 @@
         nufraw_message(NUFRAW_SET_LOG, "%s\n", stderror.str().c_str());
 
         return NUFRAW_SUCCESS;
-    } catch (Exiv2::AnyError& e) {
+    } catch (Exiv2::Error& e) {
         std::cerr.rdbuf(savecerr);
         std::string s(e.what());
         nufraw_message(NUFRAW_SET_WARNING, "%s\n", s.c_str());
