--- psqueues-0.2.8.0/psqueues.cabal	2023-10-27 17:49:07.000000000 +0200
+++ psqueues-0.2.8.0.new/psqueues.cabal	2024-07-11 16:23:57.391218976 +0200
@@ -68,7 +68,7 @@
     Build-depends:
           base     >= 4.2     && < 5
         , deepseq  >= 1.2     && < 1.6
-        , hashable >= 1.1.2.3 && < 1.5
+        , hashable >= 1.1.2.3 && < 1.6
 
     if impl(ghc>=6.10)
         Build-depends: ghc-prim
