diff -Naur plasma-sdk-5.9.2/plasmate/app/main.cpp plasma-sdk-5.9.2-p/plasmate/app/main.cpp
--- plasma-sdk-5.9.2/plasmate/app/main.cpp	2017-02-14 14:28:25.000000000 +0100
+++ plasma-sdk-5.9.2-p/plasmate/app/main.cpp	2017-02-22 01:30:34.216982227 +0100
@@ -52,7 +52,7 @@
     aboutData.addAuthor(i18n("Giorgos Tsiapaliokas"), i18n("Author"), "terietor@gmail.com");
     aboutData.addAuthor(i18n("Antonis Tsiapaliokas"), i18n("Author"), "kok3rs@gmail.com");
 
-    app.setWindowIcon(QIcon::fromTheme("plasmagik"));
+    app.setWindowIcon(QIcon::fromTheme("plasma"));
 
     QCommandLineParser parser;
     parser.addVersionOption();
diff -Naur plasma-sdk-5.9.2/plasmate/app/plasmate.desktop plasma-sdk-5.9.2-p/plasmate/app/plasmate.desktop
--- plasma-sdk-5.9.2/plasmate/app/plasmate.desktop	2017-02-14 14:28:25.000000000 +0100
+++ plasma-sdk-5.9.2-p/plasmate/app/plasmate.desktop	2017-02-22 01:23:08.719304867 +0100
@@ -66,7 +66,7 @@
 GenericName[zh_CN]=Plasma 附加组件创建器
 GenericName[zh_TW]=Plasma 附加元件建立器
 Exec=plasmate
-Icon=plasmagik
+Icon=plasma
 Type=Application
 InitialPreference=7
 Categories=Qt;KDE;Development;IDE;
diff -Naur plasma-sdk-5.9.2/plasmate/docbrowser/docbrowser.cpp plasma-sdk-5.9.2-p/plasmate/docbrowser/docbrowser.cpp
--- plasma-sdk-5.9.2/plasmate/docbrowser/docbrowser.cpp	2017-02-14 14:28:25.000000000 +0100
+++ plasma-sdk-5.9.2-p/plasmate/docbrowser/docbrowser.cpp	2017-02-22 01:29:12.080009292 +0100
@@ -51,7 +51,7 @@
     connect(linkButton, SIGNAL(clicked()), this, SLOT(showApi()));
     topbar->addWidget(linkButton);
 
-    linkButton = new QPushButton(QIcon::fromTheme("plasmagik"), i18n("Plasmate"), this);
+    linkButton = new QPushButton(QIcon::fromTheme("plasma"), i18n("Plasmate"), this);
     connect(linkButton, SIGNAL(clicked()), this, SLOT(showHelp()));
     topbar->addWidget(linkButton);
 
diff -Naur plasma-sdk-5.9.2/plasmate/plugins/plasmaprojectmanager/kdevplasmamanager.json plasma-sdk-5.9.2-p/plasmate/plugins/plasmaprojectmanager/kdevplasmamanager.json
--- plasma-sdk-5.9.2/plasmate/plugins/plasmaprojectmanager/kdevplasmamanager.json	2017-02-14 14:28:25.000000000 +0100
+++ plasma-sdk-5.9.2-p/plasmate/plugins/plasmaprojectmanager/kdevplasmamanager.json	2017-02-22 01:29:44.216780993 +0100
@@ -67,7 +67,7 @@
         "Description[x-test]": "xxAllow KDevPlatform to manage Plasma projectsxx", 
         "Description[zh_CN]": "允许 KDevPlatform 来管理 Plasma 工程", 
         "Description[zh_TW]": "允許 KDevPlatform 管理 Plasma 專案", 
-        "Icon": "plasmagik", 
+        "Icon": "plasma", 
         "Id": "KDevPlasmaManager", 
         "Name": "Plasma Project Manager", 
         "Name[ca@valencia]": "Gestor de projectes del Plasma", 
diff -Naur plasma-sdk-5.9.2/plasmate/plugins/savesystemview/kdevsavesystemview.desktop.cmake plasma-sdk-5.9.2-p/plasmate/plugins/savesystemview/kdevsavesystemview.desktop.cmake
--- plasma-sdk-5.9.2/plasmate/plugins/savesystemview/kdevsavesystemview.desktop.cmake	2017-02-14 14:28:25.000000000 +0100
+++ plasma-sdk-5.9.2-p/plasmate/plugins/savesystemview/kdevsavesystemview.desktop.cmake	2017-02-22 01:29:29.586429637 +0100
@@ -89,7 +89,7 @@
 GenericName[x-test]=xxSave System Viewxx
 GenericName[zh_CN]=保存系统视图
 GenericName[zh_TW]=儲存系統檢視
-Icon=plasmagik
+Icon=plasma
 ServiceTypes=KDevelop/Plugin
 X-KDE-Library=kdevsavesystemview
 X-KDE-PluginInfo-Name=KDevSaveSystemView
diff -Naur plasma-sdk-5.9.2/plasmate/tests/metadatahandlertest.cpp plasma-sdk-5.9.2-p/plasmate/tests/metadatahandlertest.cpp
--- plasma-sdk-5.9.2/plasmate/tests/metadatahandlertest.cpp	2017-02-14 14:28:25.000000000 +0100
+++ plasma-sdk-5.9.2-p/plasmate/tests/metadatahandlertest.cpp	2017-02-22 01:30:20.027641263 +0100
@@ -79,7 +79,7 @@
 
     const QString name(QStringLiteral("Package handler test"));
     const QString description(QStringLiteral("A Dummy package"));
-    const QString icon(QStringLiteral("plasmagik"));
+    const QString icon(QStringLiteral("plasma"));
     const QString author(QStringLiteral("plasma"));
     const QString category(QStringLiteral("System Information"));
     const QString email(QStringLiteral("plasma@kde.org"));
