diff -aur jargoninformatique-1.3.6/src/gstr.c jargoninformatique-1.3.6-2/src/gstr.c
--- jargoninformatique-1.3.6/src/gstr.c	2006-04-17 09:30:35.000000000 -0500
+++ jargoninformatique-1.3.6-2/src/gstr.c	2010-05-16 01:48:11.000000000 -0500
@@ -316,7 +316,7 @@
  * 'delim' dans 'chaine'
  ************************************************************/
 //OK
-GS_LIST *gs_split_ex(const char *string, const char *delim, char *(*_strstr)(const char *,const char *), size_t delim_len)
+GS_LIST *gs_split_ex(const char *string, const char *delim, const char *(*_strstr)(const char *,const char *), size_t delim_len)
 {
 	static const char *suivant, *precedent, *cp /*const p*/;
        	char *p;
