From effafca77eacbb4988006b1f3f4d9154df6c33f8 Mon Sep 17 00:00:00 2001
From: Markus Mittendrein <git@maxmitti.tk>
Date: Tue, 1 Oct 2019 23:06:02 +0200
Subject: [PATCH] FindGTK2: Add harfbuzz to GTK2_INCLUDE_DIRS

Fixes: #19531
---
 Modules/FindGTK2.cmake | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Modules/FindGTK2.cmake b/Modules/FindGTK2.cmake
index e3af6766f5..83091f3d83 100644
--- a/Modules/FindGTK2.cmake
+++ b/Modules/FindGTK2.cmake
@@ -259,6 +259,7 @@ function(_GTK2_FIND_INCLUDE_DIR _var _hdr)
         gtkmm-2.4
         libglade-2.0
         libglademm-2.4
+        harfbuzz
         pango-1.0
         pangomm-1.4
         sigc++-2.0
@@ -711,6 +712,8 @@ foreach(_GTK2_component ${GTK2_FIND_COMPONENTS})
         _GTK2_FIND_LIBRARY    (PANGO pango false true)
         _GTK2_ADD_TARGET      (PANGO GTK2_DEPENDS gobject glib)
 
+        _GTK2_FIND_INCLUDE_DIR(HARFBUZZ hb.h)
+
         _GTK2_FIND_LIBRARY    (PANGOCAIRO pangocairo false true)
         _GTK2_ADD_TARGET      (PANGOCAIRO GTK2_DEPENDS pango cairo gobject glib)
 
-- 
2.22.0

