diff -Naur tango-icon-theme-0.8.1/configure tango-icon-theme-0.8.1.new/configure
--- tango-icon-theme-0.8.1/configure	2007-08-14 07:30:52.000000000 +0530
+++ tango-icon-theme-0.8.1.new/configure	2007-08-15 09:13:49.000000000 +0530
@@ -6293,7 +6293,7 @@
 fi
 if test "x$have_utils" = "xyes"; then
    UTILS_PATH="`$PKG_CONFIG --variable=program_path icon-naming-utils`"
-   ICONMAP="$UTILS_PATH/icon-name-mapping"
+   ICONMAP="/usr/libexec/icon-name-mapping"
 
    { echo "$as_me:$LINENO: result: yes" >&5
 echo "${ECHO_T}yes" >&6; }
