diff --git a/doc/Makefile.am b/doc/Makefile.am
index de1c679..2d8928a 100644
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -44,7 +44,7 @@ DISTCLEANFILES    =      siegerc $(man_MANS)
 
 SIEGERC           =      $(HOME)/.siegerc
 
-URLSTXT           =      $(sysconfdir)/urls.txt
+URLSTXT           =      $(DESTDIR)$(sysconfdir)/urls.txt
 
 install-exec-hook: 
 	@if test -f $(SIEGERC); then \
@@ -70,7 +70,7 @@ install-exec-hook:
 	    echo "         It was NOT replaced with this installation."; \
 	  fi; \
 	else \
-	  $(mkinstalldirs) $(sysconfdir); \
+	  $(mkinstalldirs) $(DESTDIR)$(sysconfdir); \
 	  $(INSTALL_DATA) $(srcdir)/urls.txt $(URLSTXT); \
 	fi 
 
