diff -Naur siege-2.68-orig/doc/Makefile.am siege-2.68/doc/Makefile.am
--- siege-2.68-orig/doc/Makefile.am	2009-04-02 14:12:21.000000000 +0200
+++ siege-2.68/doc/Makefile.am	2009-04-25 10:14:31.000000000 +0200
@@ -42,9 +42,9 @@
 
 DISTCLEANFILES    =      siegerc $(man_MANS)
 
-SIEGERC           =      $(sysconfdir)/siegerc
+SIEGERC           =      $(DESTDIR)$(sysconfdir)/siegerc
 
-URLSTXT           =      $(sysconfdir)/urls.txt
+URLSTXT           =      $(DESTDIR)$(sysconfdir)/urls.txt
 
 install-exec-hook: 
 	@if test -f $(SIEGERC); then \
@@ -61,6 +61,7 @@
 	    echo "#####################################################"; \
 	  fi; \
 	else \
+	  $(mkinstalldirs) $(DESTDIR)$(sysconfdir); \
 	  $(INSTALL_DATA) $(srcdir)/siegerc $(SIEGERC); \
 	fi 
 	@if test -f $(URLSTXT); then \
@@ -70,7 +71,7 @@
 	    echo "         It was NOT replaced with this installation."; \
 	  fi; \
 	else \
-	  $(mkinstalldirs) $(sysconfdir); \
+	  $(mkinstalldirs) $(DESTDIR)$(sysconfdir); \
 	  $(INSTALL_DATA) $(srcdir)/urls.txt $(URLSTXT); \
 	fi 
 
