diff --git a/configure.ac b/configure.ac
index 487e9b5..c28531d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -20,7 +20,7 @@ AC_DEFINE_UNQUOTED(GETTEXT_PACKAGE, "$GETTEXT_PACKAGE", [Gettext package])
 AM_GLIB_GNU_GETTEXT
 IT_PROG_INTLTOOL([0.40.0])
 
-AM_PATH_PYTHON_VERSION(2.6, 2.6.0, 2.5, 2.5.0, 2.4, 2.4.0)
+AM_PATH_PYTHON_VERSION(2.7, 2.7.0, 2.6, 2.6.0, 2.5, 2.5.0, 2.4, 2.4.0)
 
 PKG_CHECK_MODULES(ALACARTE, libgnome-menu >= 2.27.92)
 
