--- kvirc/src/kvirc/kernel/KviOptions.cpp	2021-07-13 12:52:56.928935855 +0200
+++ kvirc.new/src/kvirc/kernel/KviOptions.cpp	2021-07-13 12:54:05.452357181 +0200
@@ -424,7 +424,7 @@
 KviStringListOption g_stringlistOptionsTable[KVI_NUM_STRINGLIST_OPTIONS] = {
 	STRINGLIST_OPTION("HighlightWords", KviOption_sectFlagIrcView),
 	STRINGLIST_OPTION("SpamWords", KviOption_sectFlagAntiSpam),
-	STRINGLIST_OPTION_WITHDEFAULT("RecentChannels", KviOption_sectFlagRecent, "#kvirc" KVI_RECENT_CHANNELS_SEPARATOR "LiberaChat"),
+	STRINGLIST_OPTION_WITHDEFAULT("RecentChannels", KviOption_sectFlagRecent, "#frugalware" KVI_RECENT_CHANNELS_SEPARATOR "LiberaChat"),
 	STRINGLIST_OPTION("RecentServers", KviOption_sectFlagRecent),
 	STRINGLIST_OPTION("RecentNicknames", KviOption_sectFlagRecent),
 	STRINGLIST_OPTION("ModuleExtensionToolbars", KviOption_sectFlagFrame),
