diff -Naur djvulibre-3.5.20/configure.ac djvulibre-3.5.20-p/configure.ac
--- djvulibre-3.5.20/configure.ac	2007-11-29 00:46:44.000000000 +0100
+++ djvulibre-3.5.20-p/configure.ac	2007-12-06 21:17:12.000000000 +0100
@@ -387,7 +387,7 @@
 else
   AC_MSG_CHECKING([g++ version]) 
 changequote(<<, >>)dnl
-  cxxver="`$CXX --version | head -1 | 
+  cxxver="`$CXX --version | head -n 1 | 
              sed -e 's/^.*[^0-9.]\([2-4]\.[0-9.]*\).*$/\1/'`"
 changequote([, ])dnl 
   AC_MSG_RESULT($cxxver)
diff -Naur djvulibre-3.5.20/gui/bin2cpp/Makefile.in djvulibre-3.5.20-p/gui/bin2cpp/Makefile.in
--- djvulibre-3.5.20/gui/bin2cpp/Makefile.in	2007-04-20 18:44:23.000000000 +0200
+++ djvulibre-3.5.20-p/gui/bin2cpp/Makefile.in	2007-12-06 21:17:05.000000000 +0100
@@ -47,10 +47,10 @@
 LIBDJVU = ${top_builddir}/libdjvu/libdjvulibre.la
 
 INCS = -I${top_builddir} -I${top_srcdir}/libdjvu -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${JPEG_LIBS} ${THREAD_LIBS} @LIBS@
-CFLAGS = @CPPFLAGS@ @CFLAGS@ ${FLAGS}
-CXXFLAGS = @CPPFLAGS@ @CXXFLAGS@ ${FLAGS}
+CFLAGS += @CPPFLAGS@ @CFLAGS@ ${FLAGS}
+CXXFLAGS += @CPPFLAGS@ @CXXFLAGS@ ${FLAGS}
 
 PROGRAMS = bin2cpp
 
diff -Naur djvulibre-3.5.20/gui/nsdejavu/Makefile.in djvulibre-3.5.20-p/gui/nsdejavu/Makefile.in
--- djvulibre-3.5.20/gui/nsdejavu/Makefile.in	2007-04-20 18:44:23.000000000 +0200
+++ djvulibre-3.5.20-p/gui/nsdejavu/Makefile.in	2007-12-06 21:17:05.000000000 +0100
@@ -51,7 +51,7 @@
  -I${top_srcdir}/gui/shared/res \
  -I${top_srcdir}/gui/npsdk 
 
-CFLAGS = @CPPFLAGS@ ${DEFS} ${OPTS} ${INCS} ${NSDEJAVU_CFLAGS} 
+CFLAGS += @CPPFLAGS@ ${DEFS} ${INCS} ${NSDEJAVU_CFLAGS} 
 
 LIBS = @LDFLAGS@ ${NSDEJAVU_LIBS} 
 
diff -Naur djvulibre-3.5.20/libdjvu/Makefile.in djvulibre-3.5.20-p/libdjvu/Makefile.in
--- djvulibre-3.5.20/libdjvu/Makefile.in	2007-05-19 05:07:34.000000000 +0200
+++ djvulibre-3.5.20-p/libdjvu/Makefile.in	2007-12-06 21:16:34.000000000 +0100
@@ -53,7 +53,7 @@
 LIBDJVU_VER = @LIBDJVU_VER@
 
 INCS = -I${top_builddir} -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${JPEG_LIBS} ${THREAD_LIBS} @LIBS@
 CFLAGS = ${FLAGS} @CPPFLAGS@ @CFLAGS@ 
 CXXFLAGS =  ${FLAGS} ${CXXPICFLAGS} ${CXXRPOFLAGS} @CPPFLAGS@ @CXXFLAGS@ 
diff -Naur djvulibre-3.5.20/tools/jb2cmp/Makefile.in djvulibre-3.5.20-p/tools/jb2cmp/Makefile.in
--- djvulibre-3.5.20/tools/jb2cmp/Makefile.in	2007-11-29 00:55:29.000000000 +0100
+++ djvulibre-3.5.20-p/tools/jb2cmp/Makefile.in	2007-12-06 21:17:05.000000000 +0100
@@ -47,10 +47,10 @@
 THREAD_CFLAGS = @THREAD_CFLAGS@
 
 INCS = -I${top_builddir} -I${top_srcdir} -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${THREAD_LIBS} @LIBS@
-CFLAGS = @CPPFLAGS@ @CFLAGS@ ${FLAGS}
-CXXFLAGS = @CPPFLAGS@ @CXXFLAGS@ ${CXXRPOFLAGS} ${FLAGS}
+CFLAGS += @CPPFLAGS@ @CFLAGS@ ${FLAGS}
+CXXFLAGS += @CPPFLAGS@ @CXXFLAGS@ ${CXXRPOFLAGS} ${FLAGS}
 
 
 OBJS = classify.o cuts.o frames.o patterns.o
diff -Naur djvulibre-3.5.20/tools/Makefile.in djvulibre-3.5.20-p/tools/Makefile.in
--- djvulibre-3.5.20/tools/Makefile.in	2007-04-20 18:44:24.000000000 +0200
+++ djvulibre-3.5.20-p/tools/Makefile.in	2007-12-06 21:17:05.000000000 +0100
@@ -52,11 +52,11 @@
 LIBDJVU = ${top_builddir}/libdjvu/libdjvulibre.la
 
 INCS = -I${top_builddir} -I${top_srcdir} -I${top_srcdir}/libdjvu -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${JPEG_CFLAGS} ${TIFF_CFLAGS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS} ${JPEG_CFLAGS} ${TIFF_CFLAGS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${JPEG_LIBS} ${THREAD_LIBS} @LIBS@
 MORELIBS= @LDFLAGS@ ${JPEG_LIBS} ${TIFF_LIBS} ${THREAD_LIBS} @LIBS@
-CFLAGS = ${FLAGS} @CPPFLAGS@ @CFLAGS@ 
-CXXFLAGS = ${FLAGS} ${CXXRPOFLAGS} @CPPFLAGS@ @CXXFLAGS@ 
+CFLAGS += ${FLAGS} @CPPFLAGS@ @CFLAGS@ 
+CXXFLAGS += ${FLAGS} ${CXXRPOFLAGS} @CPPFLAGS@ @CXXFLAGS@ 
 
 SUBDIRS = jb2cmp
 
diff -Naur djvulibre-3.5.20/xmltools/Makefile.in djvulibre-3.5.20-p/xmltools/Makefile.in
--- djvulibre-3.5.20/xmltools/Makefile.in	2007-09-17 08:23:41.000000000 +0200
+++ djvulibre-3.5.20-p/xmltools/Makefile.in	2007-12-06 21:17:05.000000000 +0100
@@ -50,10 +50,10 @@
 LIBDJVU = ${top_builddir}/libdjvu/libdjvulibre.la
 
 INCS = -I${top_builddir} -I${top_srcdir}/libdjvu -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${JPEG_LIBS} ${THREAD_LIBS} @LIBS@
-CFLAGS = ${FLAGS} @CPPFLAGS@ @CFLAGS@ 
-CXXFLAGS = ${FLAGS} ${CXXRPOFLAGS} @CPPFLAGS@ @CXXFLAGS@ 
+CFLAGS += ${FLAGS} @CPPFLAGS@ @CFLAGS@ 
+CXXFLAGS += ${FLAGS} ${CXXRPOFLAGS} @CPPFLAGS@ @CXXFLAGS@ 
 
 PROGRAMS = djvutoxml djvuxmlparser 
 
