diff -Naur djvulibre-3.5.19/gui/bin2cpp/Makefile.in djvulibre-3.5.19-p/gui/bin2cpp/Makefile.in
--- djvulibre-3.5.19/gui/bin2cpp/Makefile.in	2007-03-25 22:48:22.000000000 +0200
+++ djvulibre-3.5.19-p/gui/bin2cpp/Makefile.in	2007-04-20 14:57:06.000000000 +0200
@@ -46,10 +46,10 @@
 LIBDJVU = ${top_builddir}/libdjvu/libdjvulibre.la
 
 INCS = -I${top_builddir} -I${top_srcdir}/libdjvu -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${JPEG_LIBS} ${THREAD_LIBS} @LIBS@
-CFLAGS = @CPPFLAGS@ @CFLAGS@ ${FLAGS}
-CXXFLAGS = @CPPFLAGS@ @CXXFLAGS@ ${FLAGS}
+CFLAGS += @CPPFLAGS@ @CFLAGS@ ${FLAGS}
+CXXFLAGS += @CPPFLAGS@ @CXXFLAGS@ ${FLAGS}
 
 PROGRAMS = bin2cpp
 
diff -Naur djvulibre-3.5.19/gui/nsdejavu/Makefile.in djvulibre-3.5.19-p/gui/nsdejavu/Makefile.in
--- djvulibre-3.5.19/gui/nsdejavu/Makefile.in	2007-03-25 22:48:24.000000000 +0200
+++ djvulibre-3.5.19-p/gui/nsdejavu/Makefile.in	2007-04-20 14:57:36.000000000 +0200
@@ -50,7 +50,7 @@
  -I${top_srcdir}/gui/shared/res \
  -I${top_srcdir}/gui/npsdk 
 
-CFLAGS = @CPPFLAGS@ ${DEFS} ${OPTS} ${INCS} ${NSDEJAVU_CFLAGS} 
+CFLAGS += @CPPFLAGS@ ${DEFS} ${INCS} ${NSDEJAVU_CFLAGS} 
 
 LIBS = @LDFLAGS@ ${NSDEJAVU_LIBS} 
 
diff -Naur djvulibre-3.5.19/libdjvu/Makefile.in djvulibre-3.5.19-p/libdjvu/Makefile.in
--- djvulibre-3.5.19/libdjvu/Makefile.in	2007-03-25 22:48:34.000000000 +0200
+++ djvulibre-3.5.19-p/libdjvu/Makefile.in	2007-04-20 14:58:07.000000000 +0200
@@ -52,10 +52,10 @@
 LIBDJVU_VER = @LIBDJVU_VER@
 
 INCS = -I${top_builddir} -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${JPEG_LIBS} ${THREAD_LIBS} @LIBS@
-CFLAGS = ${FLAGS} @CPPFLAGS@ @CFLAGS@ 
-CXXFLAGS =  ${FLAGS} ${CXXPICFLAGS} ${CXXRPOFLAGS} @CPPFLAGS@ @CXXFLAGS@ 
+CFLAGS += ${FLAGS} @CPPFLAGS@ @CFLAGS@ 
+CXXFLAGS +=  ${FLAGS} ${CXXPICFLAGS} ${CXXRPOFLAGS} @CPPFLAGS@ @CXXFLAGS@ 
 
 all: ${LIBDJVU_LA} 
 
diff -Naur djvulibre-3.5.19/tools/jb2cmp/Makefile.in djvulibre-3.5.19-p/tools/jb2cmp/Makefile.in
--- djvulibre-3.5.19/tools/jb2cmp/Makefile.in	2007-03-25 22:48:36.000000000 +0200
+++ djvulibre-3.5.19-p/tools/jb2cmp/Makefile.in	2007-04-20 14:58:30.000000000 +0200
@@ -45,10 +45,10 @@
 THREAD_CFLAGS = @THREAD_CFLAGS@
 
 INCS = -I${top_builddir} -I${top_srcdir} -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${THREAD_LIBS} @LIBS@
-CFLAGS = @CPPFLAGS@ @CFLAGS@ ${FLAGS}
-CXXFLAGS = @CPPFLAGS@ @CXXFLAGS@ ${CXXRPOFLAGS} ${FLAGS}
+CFLAGS += @CPPFLAGS@ @CFLAGS@ ${FLAGS}
+CXXFLAGS += @CPPFLAGS@ @CXXFLAGS@ ${CXXRPOFLAGS} ${FLAGS}
 
 
 OBJS = classify.o cuts.o frames.o patterns.o
diff -Naur djvulibre-3.5.19/tools/Makefile.in djvulibre-3.5.19-p/tools/Makefile.in
--- djvulibre-3.5.19/tools/Makefile.in	2007-03-25 22:48:35.000000000 +0200
+++ djvulibre-3.5.19-p/tools/Makefile.in	2007-04-20 14:58:55.000000000 +0200
@@ -51,11 +51,11 @@
 LIBDJVU = ${top_builddir}/libdjvu/libdjvulibre.la
 
 INCS = -I${top_builddir} -I${top_srcdir} -I${top_srcdir}/libdjvu -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${JPEG_CFLAGS} ${TIFF_CFLAGS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS} ${JPEG_CFLAGS} ${TIFF_CFLAGS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${JPEG_LIBS} ${THREAD_LIBS} @LIBS@
 MORELIBS= @LDFLAGS@ ${JPEG_LIBS} ${TIFF_LIBS} ${THREAD_LIBS} @LIBS@
-CFLAGS = ${FLAGS} @CPPFLAGS@ @CFLAGS@ 
-CXXFLAGS = ${FLAGS} ${CXXRPOFLAGS} @CPPFLAGS@ @CXXFLAGS@ 
+CFLAGS += ${FLAGS} @CPPFLAGS@ @CFLAGS@ 
+CXXFLAGS += ${FLAGS} ${CXXRPOFLAGS} @CPPFLAGS@ @CXXFLAGS@ 
 
 SUBDIRS = jb2cmp
 
diff -Naur djvulibre-3.5.19/xmltools/Makefile.in djvulibre-3.5.19-p/xmltools/Makefile.in
--- djvulibre-3.5.19/xmltools/Makefile.in	2007-03-25 22:48:36.000000000 +0200
+++ djvulibre-3.5.19-p/xmltools/Makefile.in	2007-04-20 14:59:16.000000000 +0200
@@ -49,10 +49,10 @@
 LIBDJVU = ${top_builddir}/libdjvu/libdjvulibre.la
 
 INCS = -I${top_builddir} -I${top_srcdir}/libdjvu -I${srcdir}
-FLAGS = ${DEFS} ${INCS} ${OPTS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
+FLAGS = ${DEFS} ${INCS} ${JPEG_CFLAGS} ${THREAD_CFLAGS}
 LIBS= @LDFLAGS@ ${JPEG_LIBS} ${THREAD_LIBS} @LIBS@
-CFLAGS = ${FLAGS} @CPPFLAGS@ @CFLAGS@ 
-CXXFLAGS = ${FLAGS} ${CXXRPOFLAGS} @CPPFLAGS@ @CXXFLAGS@ 
+CFLAGS += ${FLAGS} @CPPFLAGS@ @CFLAGS@ 
+CXXFLAGS += ${FLAGS} ${CXXRPOFLAGS} @CPPFLAGS@ @CXXFLAGS@ 
 
 PROGRAMS = djvutoxml djvuxmlparser 
 
