diff -Naur chkconfig-1.3.30c/chkconfig.c chkconfig-1.3.30c-p/chkconfig.c
--- chkconfig-1.3.30c/chkconfig.c	2007-04-01 23:36:25.000000000 +0200
+++ chkconfig-1.3.30c-p/chkconfig.c	2008-03-10 15:14:15.000000000 +0100
@@ -202,13 +202,14 @@
 
 		/* Skip files with known bad extensions */
 		if ((dn = strrchr(ent->d_name, '.')) != NULL &&
-		    (!strcmp(dn, ".rpmsave") || !strcmp(dn, ".rpmnew") || !strcmp(dn, ".rpmorig") || !strcmp(dn, ".swp")))
+		    (!strcmp(dn, ".rpmsave") || !strcmp(dn, ".rpmnew") || !strcmp(dn, ".rpmorig") || !strcmp(dn, ".swp") 
+			|| !strcmp(dn, ".pacnew") || !strcmp(dn, ".pacsave") || !strcmp(dn, ".pacorig")))
 			continue;
-		
+
 		dn = ent->d_name + strlen(ent->d_name) - 1;
 		if (*dn == '~' || *dn == ',')
 			continue;
-	
+
 		sprintf(fn, "%s/%s", conf.initdir, ent->d_name);
 		if (stat(fn, &sb)) {
 			continue;
@@ -224,7 +225,6 @@
 	do {
 	  	nResolved = 0;
 		int i;
-		
 		for (i = 0; i < numservs ; i++) {
 			if ((servs+i)->isLSB)
 				nResolved += frobOneDependencies(servs+i, servs, numservs, 0);
@@ -244,9 +244,13 @@
 	return 1;
     }
 
-    printf("start = %d\n", s.sPriority);
-    printf("end = %d\n", s.kPriority);
-	
+    #if 0
+
+	printf("start = %d\n", s.sPriority);
+    	printf("end = %d\n", s.kPriority);
+
+    #endif
+
     if (s.type == TYPE_XINETD) return 0;
     if (s.isLSB)
 		frobDependencies(&s);
@@ -343,7 +347,7 @@
 
     numServicesAlloced = 10;
     services = malloc(sizeof(*services) * numServicesAlloced);
-    
+
     if (!(dir = opendir(conf.initdir))) {
 	fprintf(stderr, _("failed to open %s: %s\n"), conf.initdir,
 		strerror(errno));
@@ -361,13 +365,14 @@
     (!strcmp(dn, ".rpmsave") || !strcmp(dn, ".rpmnew")
      || !strcmp(dn, ".rpmorig") || !strcmp(dn, ".swp")
      || !strcmp(dn, ".dpkg-dist") || !strcmp(dn, ".dpkg-old")
-     || !strcmp(dn, ".dpkg-new")))
+     || !strcmp(dn, ".dpkg-new") || !strcmp(dn, ".pacsave")
+     || !strcmp(dn, ".pacnew") || !strcmp(dn, ".pacorig")))
 	    continue;
 
 	dn = ent->d_name + strlen(ent->d_name) - 1;
 	if (*dn == '~' || *dn == ',')
 	    continue;
-	
+
 	sprintf(fn, "%s/%s", conf.initdir, ent->d_name);
 	if (stat(fn, &sb)) {
 	    fprintf(stderr, _("error reading info for service %s: %s\n"), 
@@ -421,7 +426,8 @@
 
 		    /* Skip files with known bad extensions */
 		    if ((dn = strrchr(ent->d_name, '.')) != NULL &&
-			(!strcmp(dn, ".rpmsave") || !strcmp(dn, ".rpmnew") || !strcmp(dn, ".rpmorig") || !strcmp(dn, ".swp")))
+			(!strcmp(dn, ".rpmsave") || !strcmp(dn, ".rpmnew") || !strcmp(dn, ".rpmorig") || !strcmp(dn, ".swp")
+				||!strcmp(dn, ".pacsave") || !strcmp(dn, ".pacnew") || !strcmp(dn, ".pacorig")))
 		      continue;
 
 		    dn = ent->d_name + strlen(ent->d_name) - 1;
