--- gnome-terminal-2.24.0old/src/gnome-terminal.schemas.in	2008-06-17 22:28:21.000000000 +0200
+++ gnome-terminal-2.24.0/src/gnome-terminal.schemas.in	2008-09-30 09:41:51.000000000 +0200
@@ -60,7 +60,7 @@
       <applyto>/apps/gnome-terminal/global/use_menu_accelerators</applyto>
       <owner>gnome-terminal</owner>
       <type>bool</type>
-      <default>true</default>
+      <default>false</default>
       <locale name="C">
          <short>Whether the standard GTK shortcut for menubar access is enabled</short>
          <long>

