diff -Naur gnome-terminal-2.10.0.orig/src/gnome-terminal.schemas.in gnome-terminal-2.10.0/src/gnome-terminal.schemas.in
--- gnome-terminal-2.10.0.orig/src/gnome-terminal.schemas.in	2004-08-17 16:41:35.000000000 +0200
+++ gnome-terminal-2.10.0/src/gnome-terminal.schemas.in	2005-07-28 19:17:36.000000000 +0200
@@ -341,7 +341,7 @@
       <applyto>/apps/gnome-terminal/profiles/Default/login_shell</applyto>
       <owner>gnome-terminal</owner>
       <type>bool</type>
-      <default>false</default>
+      <default>true</default>
       <locale name="C">
          <short>Whether to launch the command in the terminal as a login shell</short>
          <long>
