diff -Naur gnome-terminal-2.12.0.orig/src/gnome-terminal.schemas.in gnome-terminal-2.12.0/src/gnome-terminal.schemas.in
--- gnome-terminal-2.12.0.orig/src/gnome-terminal.schemas.in	2005-07-02 01:35:09.000000000 +0200
+++ gnome-terminal-2.12.0/src/gnome-terminal.schemas.in	2005-11-19 18:35:09.000000000 +0100
@@ -60,7 +60,7 @@
       <applyto>/apps/gnome-terminal/global/use_menu_accelerators</applyto>
       <owner>gnome-terminal</owner>
       <type>bool</type>
-      <default>true</default>
+      <default>false</default>
       <locale name="C">
          <short>Whether the standard GTK shortcut for menubar access is enabled</short>
          <long>
@@ -871,7 +871,7 @@
       <applyto>/apps/gnome-terminal/keybindings/switch_to_tab_1</applyto>
       <owner>gnome-terminal</owner>
       <type>string</type>
-      <default>&lt;Alt&gt;1</default>
+<!-- no default -->
       <locale name="C">
          <short>Keyboard shortcut to switch to tab 1</short>
          <long>
@@ -888,7 +888,7 @@
       <applyto>/apps/gnome-terminal/keybindings/switch_to_tab_2</applyto>
       <owner>gnome-terminal</owner>
       <type>string</type>
-      <default>&lt;Alt&gt;2</default>
+<!-- no default -->
       <locale name="C">
          <short>Keyboard shortcut to switch to tab 2</short>
          <long>
@@ -905,7 +905,7 @@
       <applyto>/apps/gnome-terminal/keybindings/switch_to_tab_3</applyto>
       <owner>gnome-terminal</owner>
       <type>string</type>
-      <default>&lt;Alt&gt;3</default>
+<!-- no default -->
       <locale name="C">
          <short>Keyboard shortcut to switch to tab 3</short>
          <long>
@@ -922,7 +922,7 @@
       <applyto>/apps/gnome-terminal/keybindings/switch_to_tab_4</applyto>
       <owner>gnome-terminal</owner>
       <type>string</type>
-      <default>&lt;Alt&gt;4</default>
+<!-- no default -->
       <locale name="C">
          <short>Keyboard shortcut to switch to tab 4</short>
          <long>
@@ -939,7 +939,7 @@
       <applyto>/apps/gnome-terminal/keybindings/switch_to_tab_5</applyto>
       <owner>gnome-terminal</owner>
       <type>string</type>
-      <default>&lt;Alt&gt;5</default>
+<!-- no default -->
       <locale name="C">
          <short>Keyboard shortcut to switch to tab 5</short>
          <long>
@@ -956,7 +956,7 @@
       <applyto>/apps/gnome-terminal/keybindings/switch_to_tab_6</applyto>
       <owner>gnome-terminal</owner>
       <type>string</type>
-      <default>&lt;Alt&gt;6</default>
+<!-- no default -->
       <locale name="C">
          <short>Keyboard shortcut to switch to tab 6</short>
          <long>
@@ -973,7 +973,7 @@
       <applyto>/apps/gnome-terminal/keybindings/switch_to_tab_7</applyto>
       <owner>gnome-terminal</owner>
       <type>string</type>
-      <default>&lt;Alt&gt;7</default>
+<!-- no default -->
       <locale name="C">
          <short>Keyboard shortcut to switch to tab 7</short>
          <long>
@@ -990,7 +990,7 @@
       <applyto>/apps/gnome-terminal/keybindings/switch_to_tab_8</applyto>
       <owner>gnome-terminal</owner>
       <type>string</type>
-      <default>&lt;Alt&gt;8</default>
+<!-- no default -->
       <locale name="C">
          <short>Keyboard shortcut to switch to tab 8</short>
          <long>
@@ -1007,7 +1007,7 @@
       <applyto>/apps/gnome-terminal/keybindings/switch_to_tab_9</applyto>
       <owner>gnome-terminal</owner>
       <type>string</type>
-      <default>&lt;Alt&gt;9</default>
+<!-- no default -->
       <locale name="C">
          <short>Keyboard shortcut to switch to tab 9</short>
          <long>
