From afbdd5f632db00d9065ea8eeda3c2d3bbc646627 Mon Sep 17 00:00:00 2001
From: Yanko Kaneti <yaneti@declera.com>
Date: Fri, 25 Jan 2013 14:56:45 +0200
Subject: [PATCH] Don't assume enum size

---
 src/terminal-screen-container.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/terminal-screen-container.c b/src/terminal-screen-container.c
index 843f871..c7ec914 100644
--- a/src/terminal-screen-container.c
+++ b/src/terminal-screen-container.c
@@ -73,7 +73,7 @@ terminal_screen_container_style_updated (GtkWidget *widget)
 {
   TerminalScreenContainer *container = TERMINAL_SCREEN_CONTAINER (widget);
   TerminalScreenContainerPrivate *priv = container->priv;
-  glong corner;
+  GtkCornerType corner;
   gboolean set;  
 
   GTK_WIDGET_CLASS (terminal_screen_container_parent_class)->style_updated (widget);
-- 
1.8.1


