--- alarm-clock-1.4.3/src/alarm_runner.c.old	2010-11-04 18:24:49.000000000 +0100
+++ alarm-clock-1.4.3/src/alarm_runner.c	2011-04-20 15:55:48.000000000 +0200
@@ -212,12 +212,7 @@
 	text = g_key_file_get_string(loaded_alarms, name, "PassivePopupText", NULL);
 	
 	if (notify != NULL) notify_notification_close(notify, NULL);
-#ifndef APPINDICATOR
-	notify = notify_notification_new_with_status_icon(title, text, NULL, status_icon);
-#endif
-#ifdef APPINDICATOR
-	notify = notify_notification_new(title, text, NULL, NULL);
-#endif
+	notify = notify_notification_new(title, text, NULL);
 	if (notify == NULL) g_print("BAD");
 	notify_notification_set_urgency(notify, type);
 	notify_notification_set_timeout(notify, timeout * 1000);
