diff -urp pcmcia-cs-3.2.8/debug-tools/lspnp.c pcmcia-cs-3.2.8-gcc4/debug-tools/lspnp.c
--- pcmcia-cs-3.2.8/debug-tools/lspnp.c	2002-02-13 06:45:01.000000000 +0100
+++ pcmcia-cs-3.2.8-gcc4/debug-tools/lspnp.c	2005-05-08 10:08:07.000000000 +0200
@@ -496,7 +496,7 @@ static char *dump_chain(u_char *buf, int
 		dump_io_fixed(r); break;
 	    }
 	}
-	(u_char *)p += sz + 1;
+	p = (union pnp_resource *) ((u_char *)p + sz + 1);
     }
     return (u_char *)p;
 }
diff -urp pcmcia-cs-3.2.8/debug-tools/setpnp.c pcmcia-cs-3.2.8-gcc4/debug-tools/setpnp.c
--- pcmcia-cs-3.2.8/debug-tools/setpnp.c	2001-10-10 04:58:12.000000000 +0200
+++ pcmcia-cs-3.2.8-gcc4/debug-tools/setpnp.c	2005-05-08 10:08:07.000000000 +0200
@@ -163,7 +163,7 @@ static char *update_chain(u_char *buf, i
 		break;
 	    }
 	}
-	(u_char *)p += sz + 1;
+	p = (union pnp_resource *) ((u_char *)p + sz + 1);
     }
     return (u_char *)p;
 }
