diff -Naur openvas-scanner-6.0.0/CMakeLists.txt openvas-scanner-6.0.0-p/CMakeLists.txt
--- openvas-scanner-6.0.0/CMakeLists.txt	2019-04-05 11:11:38.000000000 +0200
+++ openvas-scanner-6.0.0-p/CMakeLists.txt	2019-04-09 23:39:16.305783269 +0200
@@ -39,15 +39,6 @@
 ## Retrieve git revision (at configure time)
 include (GetGit)
 
-if (NOT CMAKE_BUILD_TYPE MATCHES "Release")
-  if (EXISTS "${CMAKE_SOURCE_DIR}/.git/")
-    if (GIT_FOUND)
-      Git_GET_REVISION(${CMAKE_SOURCE_DIR} ProjectRevision)
-      set (GIT_REVISION "~git-${ProjectRevision}")
-    endif (GIT_FOUND)
-  endif (EXISTS "${CMAKE_SOURCE_DIR}/.git/")
-endif (NOT CMAKE_BUILD_TYPE MATCHES "Release")
-
 ## make format
 message (STATUS "Looking for clang-format...")
 find_program (CLANG_FORMAT clang-format)
@@ -64,12 +55,6 @@
 # unset if this is a stable release series.
 set (PROJECT_BETA_RELEASE 0)
 
-if (GIT_REVISION)
-  set (PROJECT_VERSION_GIT "${GIT_REVISION}")
-else (GIT_REVISION)
-  set (PROJECT_VERSION_GIT "~git")
-endif (GIT_REVISION)
-
 # If PROJECT_BETA_RELEASE is set, the version string will be set to:
 #   "major.minor+beta${PROJECT_BETA_RELEASE}"
 # If PROJECT_BETA_RELEASE is NOT set, the version string will be set to:
