diff -Naur pypatchelf-0.9/pypatchelf/__init__.py pypatchelf-0.9.new/pypatchelf/__init__.py
--- pypatchelf-0.9/pypatchelf/__init__.py	2016-01-15 10:45:21.000000000 +0100
+++ pypatchelf-0.9.new/pypatchelf/__init__.py	2022-11-24 11:12:05.611211104 +0100
@@ -2,7 +2,7 @@
 import os.path
 import subprocess
 
-PATCHELF = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'patchelf')
+PATCHELF = "/usr/bin/patchelf"
 
 
 def main():
diff -Naur pypatchelf-0.9/setup.py pypatchelf-0.9.new/setup.py
--- pypatchelf-0.9/setup.py	2016-03-31 06:31:22.000000000 +0200
+++ pypatchelf-0.9.new/setup.py	2022-11-24 11:12:00.003525432 +0100
@@ -51,9 +51,8 @@
       license="GPL",
       entry_points={
           'console_scripts': [
-              'patchelf = pypatchelf:main',
+              
               ]},
       packages=find_packages(),
       zip_safe=False,
-      ext_modules=[Extension('mock', ['mock'])],
-      cmdclass={'build_ext': build_ext})
+      cmdclass={})
