--- a/texk/dvisvgm/dvisvgm-src/src/Ghostscript.cpp	2022-01-21 18:25:16.000000000 +0100
+++ b/texk/dvisvgm/dvisvgm-src/src/Ghostscript.cpp	2023-01-18 19:32:46.948841498 +0100
@@ -377,17 +377,6 @@
 	const char *error_names[] = { ERROR_NAMES };
 	if (code == 0 || (size_t)code > sizeof(error_names)/sizeof(error_names[0]))
 		return nullptr;
-#if defined(HAVE_LIBGS)
-	// use array defined in libgs to avoid linking the error strings into the binary
-	return gs_error_names[code-1];
-#elif defined(_WIN32)
-	// gs_error_names is private in the Ghostscript DLL so we can't access it here
-	return error_names[code-1];
-#else
-	if (auto error_names = loadSymbol<const char**>("gs_error_names"))
-		return error_names[code-1];
-	return nullptr;
-#endif
 }
 
 #endif  // !DISABLE_GS
