diff -Naur texlive-20091107-source/texk/web2c/luatexdir/image/pdftoepdf.cc texlive-20091107-sourcenew//texk/web2c/luatexdir/image/pdftoepdf.cc
--- texlive-20091107-source/texk/web2c/luatexdir/image/pdftoepdf.cc	2009-06-25 15:05:10.000000000 +0200
+++ texlive-20091107-sourcenew//texk/web2c/luatexdir/image/pdftoepdf.cc	2010-02-27 21:59:34.328653858 +0100
@@ -666,7 +666,7 @@
     }
     for (r = encodingList; r != NULL; r = n) {
         n = r->next;
-        delete r->font;
+        //delete r->font;
         delete r;
     }
 }
@@ -720,7 +720,7 @@
     // this works only for PDF 1.x -- but since any versions of PDF newer
     // than 1.x will not be backwards compatible to PDF 1.x, pdfTeX will
     // then have to changed drastically anyway.
-    pdf_version_found = pdf_doc->doc->getPDFVersion();
+    pdf_version_found = pdf_doc->doc->getPDFMajorVersion() + 0.1 * pdf_doc->doc->getPDFMinorVersion();
     pdf_version_wanted = 1 + (minor_pdf_version_wanted * 0.1);
     if (pdf_version_found > pdf_version_wanted + 0.01) {
         char msg[] =
diff -Naur texlive-20091107-source/texk/web2c/pdftexdir/pdftoepdf.cc texlive-20091107-sourcenew//texk/web2c/pdftexdir/pdftoepdf.cc
--- texlive-20091107-source/texk/web2c/pdftexdir/pdftoepdf.cc	2009-05-06 10:19:56.000000000 +0200
+++ texlive-20091107-sourcenew//texk/web2c/pdftexdir/pdftoepdf.cc	2010-02-27 21:59:01.546653738 +0100
@@ -650,7 +650,7 @@
     }
     for (r = encodingList; r != 0; r = n) {
         n = r->next;
-        delete r->font;
+        //delete r->font;
         delete r;
     }
 }
@@ -705,7 +705,7 @@
     // this works only for PDF 1.x -- but since any versions of PDF newer
     // than 1.x will not be backwards compatible to PDF 1.x, pdfTeX will
     // then have to changed drastically anyway.
-    pdf_version_found = pdf_doc->doc->getPDFVersion();
+    pdf_doc->doc->getPDFMajorVersion() + 0.1 * pdf_doc->doc->getPDFMinorVersion();
     pdf_version_wanted = 1 + (minor_pdf_version_wanted * 0.1);
     if (pdf_version_found > pdf_version_wanted) {
         char msg[] =
