diff -up a/texk/web2c/am/texmf.am.dt a/texk/web2c/am/texmf.am
--- a/texk/web2c/am/texmf.am.dt	2015-07-10 05:42:52.000000000 -0400
+++ a/texk/web2c/am/texmf.am	2019-05-23 15:07:41.663729872 -0400
@@ -73,7 +73,7 @@ DISTCLEANFILES += $(nodist_tex_SOURCES)
 
 # TeX tests
 #
-tex_tests = triptest.test tests/write18-quote-test.pl
+tex_tests = tests/write18-quote-test.pl
 triptest.log: tex$(EXEEXT) dvitype$(EXEEXT) pltotf$(EXEEXT) tftopl$(EXEEXT)
 tests/write18-quote-test.log: tex$(EXEEXT)
 EXTRA_DIST += $(tex_tests)
@@ -201,7 +201,7 @@ DISTCLEANFILES += $(nodist_libmf_a_SOURC
 
 # Metafont tests
 #
-mf_tests = mftraptest.test
+mf_tests =
 mftraptest.log: mf$(EXEEXT) gftype$(EXEEXT) tftopl$(EXEEXT)
 EXTRA_DIST += $(mf_tests)
 if MF
diff -up a/texk/web2c/eptexdir/am/eptex.am.dt a/texk/web2c/eptexdir/am/eptex.am
--- a/texk/web2c/eptexdir/am/eptex.am.dt	2015-08-06 04:49:49.000000000 -0400
+++ a/texk/web2c/eptexdir/am/eptex.am	2019-05-23 15:07:41.663729872 -0400
@@ -81,7 +81,7 @@ EXTRA_DIST += \
 
 # e-pTeX Tests
 #
-eptex_tests = eptexdir/eptriptest.test eptexdir/pdfprimitive.test
+eptex_tests =
 eptexdir/eptriptest.log: eptex$(EXEEXT) dvitype$(EXEEXT) pltotf$(EXEEXT) tftopl$(EXEEXT)
 eptexdir/pdfprimitive.log: eptex$(EXEEXT)
 
diff -up a/texk/web2c/euptexdir/am/euptex.am.dt a/texk/web2c/euptexdir/am/euptex.am
--- a/texk/web2c/euptexdir/am/euptex.am.dt	2018-01-20 22:48:06.000000000 -0500
+++ a/texk/web2c/euptexdir/am/euptex.am	2019-05-23 15:07:41.663729872 -0400
@@ -82,7 +82,7 @@ EXTRA_DIST += \
 
 # e-upTeX Tests
 #
-euptex_tests = euptexdir/euptriptest.test euptexdir/pdfprimitive.test
+euptex_tests =
 euptexdir/euptriptest.log: euptex$(EXEEXT) dvitype$(EXEEXT) pltotf$(EXEEXT) tftopl$(EXEEXT)
 euptexdir/pdfprimitive.log: euptex$(EXEEXT)
 
diff -up a/texk/web2c/mfluadir/am/mflua.am.dt a/texk/web2c/mfluadir/am/mflua.am
--- a/texk/web2c/mfluadir/am/mflua.am.dt	2017-04-16 06:11:22.000000000 -0400
+++ a/texk/web2c/mfluadir/am/mflua.am	2019-05-23 15:07:41.663729872 -0400
@@ -110,7 +110,7 @@ EXTRA_DIST += \
 
 # MFLua tests
 #
-mflua_tests = mfluadir/mfluatraptest.test
+mflua_tests =
 mfluadir/mfluatraptest.log: mflua$(EXEEXT) gftype$(EXEEXT) tftopl$(EXEEXT)
 EXTRA_DIST += $(mflua_tests)
 
diff -up a/texk/web2c/mfluajitdir/am/mfluajit.am.dt a/texk/web2c/mfluajitdir/am/mfluajit.am
--- a/texk/web2c/mfluajitdir/am/mfluajit.am.dt	2017-04-16 06:11:22.000000000 -0400
+++ a/texk/web2c/mfluajitdir/am/mfluajit.am	2019-05-23 15:07:41.663729872 -0400
@@ -106,7 +106,7 @@ EXTRA_DIST += \
 
 # MFLuaJIT tests
 #
-mfluajit_tests = mfluajitdir/mfluajittraptest.test
+mfluajit_tests =
 mfluajitdir/mfluajittraptest.log: mfluajit$(EXEEXT) gftype$(EXEEXT) tftopl$(EXEEXT)
 EXTRA_DIST += $(mfluajit_tests)
 
diff -up a/texk/web2c/pdftexdir/am/pdftex.am.dt a/texk/web2c/pdftexdir/am/pdftex.am
--- a/texk/web2c/pdftexdir/am/pdftex.am.dt	2019-05-23 15:07:41.663729872 -0400
+++ a/texk/web2c/pdftexdir/am/pdftex.am	2019-05-23 15:09:34.941250514 -0400
@@ -94,8 +94,8 @@ EXTRA_DIST += \
 
 # pdfTeX tests
 #
-pdftex_tests = pdftexdir/wprob.test pdftexdir/pdftex.test \
-  pdftexdir/pdfimage.test pdftexdir/expanded.test
+pdftex_tests = pdftexdir/pdftex.test \
+  pdftexdir/expanded.test
 
 pdftexdir/wprob.log pdftexdir/pdftex.log \
   pdftexdir/pdfimage.log pdftexdir/expanded.log: pdftex$(EXEEXT)
diff -up a/texk/web2c/ptexdir/am/ptex.am.dt a/texk/web2c/ptexdir/am/ptex.am
--- a/texk/web2c/ptexdir/am/ptex.am.dt	2019-02-06 05:58:23.000000000 -0500
+++ a/texk/web2c/ptexdir/am/ptex.am	2019-05-23 15:07:41.664729850 -0400
@@ -167,7 +167,7 @@ EXTRA_DIST += \
 
 # pTeX Tests
 #
-ptex_tests = ptexdir/ptriptest.test
+ptex_tests =
 ptexdir/ptriptest.log: ptex$(EXEEXT) dvitype$(EXEEXT) pltotf$(EXEEXT) tftopl$(EXEEXT)
 pweb_tests = \
 	ptexdir/pbibtex.test \
diff -up a/texk/web2c/uptexdir/am/uptex.am.dt a/texk/web2c/uptexdir/am/uptex.am
--- a/texk/web2c/uptexdir/am/uptex.am.dt	2018-08-18 07:45:50.000000000 -0400
+++ a/texk/web2c/uptexdir/am/uptex.am	2019-05-23 15:07:41.664729850 -0400
@@ -157,7 +157,7 @@ EXTRA_DIST += \
 	uptexdir/ChangeLog
 
 # upTeX Tests
-uptex_tests = uptexdir/uptriptest.test
+uptex_tests =
 uptexdir/uptriptest.log: uptex$(EXEEXT) dvitype$(EXEEXT) pltotf$(EXEEXT) tftopl$(EXEEXT)
 upweb_tests = \
 	uptexdir/upbibtex.test \
diff -up a/texk/web2c/xetexdir/am/xetex.am.dt a/texk/web2c/xetexdir/am/xetex.am
--- a/texk/web2c/xetexdir/am/xetex.am.dt	2018-01-17 17:54:13.000000000 -0500
+++ a/texk/web2c/xetexdir/am/xetex.am	2019-05-23 15:07:41.664729850 -0400
@@ -200,7 +200,6 @@ EXTRA_DIST += \
 # XeTeX Tests
 #
 xetex_tests = \
-	xetexdir/xetex-bug73.test \
 	xetexdir/xetex.test
 xetexdir/xetex-bug73.log xetexdir/xetex.log: xetex$(EXEEXT)
 
