diff -Naur texlive-20160523-texmf/texmf-dist/scripts/texlive/fmtutil.pl texlive-20160523-texmf-p/texmf-dist/scripts/texlive/fmtutil.pl
--- texlive-20160523-texmf/texmf-dist/scripts/texlive/fmtutil.pl	2016-04-22 15:16:07.000000000 +0200
+++ texlive-20160523-texmf-p/texmf-dist/scripts/texlive/fmtutil.pl	2016-07-17 18:57:20.671268023 +0200
@@ -65,7 +65,7 @@
 # sudo sometimes does not reset the home dir of root, check on that
 # see more comments at the definition of the function itself
 # this function checks by itself whether it is running on windows or not
-reset_root_home();
+#reset_root_home();
 
 chomp(our $TEXMFDIST = `kpsewhich --var-value=TEXMFDIST`);
 chomp(our $TEXMFVAR = `kpsewhich -var-value=TEXMFVAR`);
