diff -Naur texlive-20091107-source/utils/xindy/configure texlive-20091107-source/utils/xindy/configure
--- texlive-20091107-source/utils/xindy/configure	2009-06-15 17:14:49.000000000 +0200
+++ texlive-20091107-source/utils/xindy/configure	2010-08-25 00:24:46.462000037 +0200
@@ -4238,7 +4238,7 @@
   CLISP_LIB=`$CLISP -q -norc -x '(progn (princ *lib-directory*) (values))' | sed -e 's,\\/$,,'`
   { $as_echo "$as_me:$LINENO: result: $CLISP_LIB" >&5
 $as_echo "$CLISP_LIB" >&6; }
-  test -f "$CLISP_LIB/clisp-link" || if test "x$enable_build" = xno; then
+  test -f "/usr/bin/clisp-link" || if test "x$enable_build" = xno; then
   { $as_echo "$as_me:$LINENO: WARNING: building xindy has been disabled and would fail because
 clisp is not usable: expected file is not found: $CLISP_LIB/clisp-link" >&5
 $as_echo "$as_me: WARNING: building xindy has been disabled and would fail because
@@ -4250,7 +4250,7 @@
 fi
 
 
-  test -d "$CLISP_LIB/full" || if test "x$enable_build" = xno; then
+  test -d "$CLISP_LIB/base" || if test "x$enable_build" = xno; then
   { $as_echo "$as_me:$LINENO: WARNING: building xindy has been disabled and would fail because
 clisp is not usable: expected directory is not found: $CLISP_LIB/full" >&5
 $as_echo "$as_me: WARNING: building xindy has been disabled and would fail because

diff -Naur texlive-20091107-source/utils/xindy/rte/Makefile.in texlive-20091107-source/utils/xindy/rte/Makefile.in
--- texlive-20091107-source/utils/xindy/rte/Makefile.in	2009-08-05 13:36:50.000000000 +0200
+++ texlive-20091107-source/utils/xindy/rte/Makefile.in	2010-08-25 00:25:43.954000035 +0200
@@ -548,8 +548,8 @@
 	test -f ordrules/link.sh || for i in $(ORDRULES_FILES); do cp -p $(top_srcdir)/rte/ordrules/$$i ordrules/$$i; done
 	$(MAKE) -C ordrules ordrulei.c
 	env CLISP_LINKKIT='$(CLISP_LIB)/linkkit' MAKE="$(MAKE)" \
-	sh '$(CLISP_LIB)/clisp-link' add-module-set ordrules '$(CLISP_LIB)/full' xindy-build-dir || \
-	sh '$(CLISP_LIB)/clisp-link' add '$(CLISP_LIB)/full' xindy-build-dir ordrules
+	sh '/usr/bin/clisp-link' add-module-set ordrules '$(CLISP_LIB)/base' xindy-build-dir || \
+	sh '/usr/bin/clisp-link' add '$(CLISP_LIB)/base' xindy-build-dir ordrules
 	$(INSTALL_STRIP_PROGRAM) xindy-build-dir/$(LISP_RUN) $(binariesdir)/$(XINDY_RUN)
 
 clean-local:
