diff -Naur archmage-0.0.6.orig/chmlib/chm_lib.h archmage-0.0.6/chmlib/chm_lib.h
--- archmage-0.0.6.orig/chmlib/chm_lib.h	2005-09-05 17:28:20.000000000 +0200
+++ archmage-0.0.6/chmlib/chm_lib.h	2005-09-05 17:29:53.000000000 +0200
@@ -53,6 +53,7 @@
 
 #include "lzx.h"
 
+#include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
 
@@ -139,7 +140,7 @@
 /* I386, 32-bit, non-Windows */
 /* Sparc        */
 /* MIPS         */
-#elif __i386__ || __sun || __sgi
+#elif __GNUC__
 typedef unsigned char           UChar;
 typedef short                   Int16;
 typedef unsigned short          UInt16;
