--- gnumeric-1.4.3.orig/plugins/excel/boot.c	2005/01/17 04:37:25	1.122
+++ gnumeric-1.4.3/plugins/excel/boot.c	2005/07/04 20:00:58	1.122.2.1
@@ -117,22 +117,19 @@
 }
 
 static void
-excel_read_metadata (Workbook  *wb, GsfInfile *ole, char const *name,
+excel_read_metadata (GsfDocMetaData *meta_data, GsfInfile *ole, char const *name,
 		     IOContext *context)
 {
-	GError   *err = NULL;
 	GsfInput *stream = gsf_infile_child_by_name (ole, name);
 
 	if (stream != NULL) {
-		gsf_msole_metadata_read (stream, &err);
+		GError *err = gsf_msole_metadata_read (stream, meta_data);
 		if (err != NULL) {
 			gnm_io_warning (context, err->message);
 			g_error_free (err);
 		}
-		gsf_input_seek (stream, 0, G_SEEK_SET);
-		g_object_set_data_full (G_OBJECT (wb), name,
-			gsf_structured_blob_read (stream), g_object_unref);
-		g_object_unref (G_OBJECT (stream));
+
+		g_object_unref (stream);
 	}
 }
 
@@ -154,6 +151,7 @@
 	GsfInfile *ole = gsf_infile_msole_new (input, &err);
 	Workbook  *wb = wb_view_workbook (wbv);
 	gboolean   is_double_stream_file, is_97;
+	GsfDocMetaData *meta_data;
 
 	if (ole == NULL) {
 		guint8 const *data;
@@ -188,8 +186,11 @@
 	excel_read_workbook (context, wbv, stream, &is_double_stream_file);
 	g_object_unref (G_OBJECT (stream));
 
-	excel_read_metadata (wb, ole, "\05SummaryInformation", context);
-	excel_read_metadata (wb, ole, "\05DocumentSummaryInformation", context);
+	meta_data = gsf_doc_meta_data_new ();
+	excel_read_metadata (meta_data, ole, "\05DocumentSummaryInformation", context);
+	excel_read_metadata (meta_data, ole, "\05SummaryInformation", context);
+	g_object_set_data_full (G_OBJECT (wb), "GsfDocMetaData",
+		meta_data, g_object_unref);
 
 	/* See if there are any macros to keep around */
 	stream = gsf_infile_child_by_name (ole, "\01CompObj");
@@ -228,6 +229,7 @@
 	GsfOutfile *outfile;
 	ExcelWriteState *ewb = NULL;
 	GsfStructuredBlob *blob;
+	GsfDocMetaData *meta_data;
 
 	io_progress_message (context, _("Preparing to save..."));
 	io_progress_range_push (context, 0.0, 0.1);
@@ -250,25 +252,20 @@
 	excel_write_state_free (ewb);
 	io_progress_range_pop (context);
 
-	blob = g_object_get_data (G_OBJECT (wb), "\05DocumentSummaryInformation");
-	if (blob == NULL) {
+	meta_data = g_object_get_data (G_OBJECT (wb), "GsfDocMetaData");
+	if (meta_data != NULL) {
 		content = gsf_outfile_new_child (outfile,
 			"\05DocumentSummaryInformation", FALSE);
-		gsf_msole_metadata_write (content, TRUE, NULL);
+		gsf_msole_metadata_write (content, meta_data, TRUE);
 		gsf_output_close (content);
 		g_object_unref (G_OBJECT (content));
-	} else
-		gsf_structured_blob_write (blob, outfile);
 
-	blob = g_object_get_data (G_OBJECT (wb), "\05SummaryInformation");
-	if (blob == NULL) {
 		content = gsf_outfile_new_child (outfile,
 			"\05SummaryInformation", FALSE);
-		gsf_msole_metadata_write (content, FALSE, NULL);
+		gsf_msole_metadata_write (content, meta_data, FALSE);
 		gsf_output_close (content);
 		g_object_unref (G_OBJECT (content));
-	} else
-		gsf_structured_blob_write (blob, outfile);
+	}
 
 	/* restore the macros we loaded */
 	blob = g_object_get_data (G_OBJECT (wb), "MS_EXCEL_COMPOBJ");
diff -urN gnumeric-1.4.3/plugins/corba/corba-workbook.c gnumeric-1.4.3-work/plugins/corba/corba-workbook.c
--- gnumeric-1.4.3/plugins/corba/corba-workbook.c	2004-09-24 22:35:15.000000000 +0200
+++ gnumeric-1.4.3-work/plugins/corba/corba-workbook.c	2005-09-11 21:42:35.000000000 +0200
@@ -313,8 +313,8 @@
 }
 
 GSF_CLASS_FULL (WorkbookControlCORBA, workbook_control_corba,
-		wbcc_class_init, wbcc_init, 
-		WORKBOOK_CONTROL_TYPE, 0,
+		NULL, NULL, wbcc_class_init, NULL,
+		wbcc_init, WORKBOOK_CONTROL_TYPE, 0,
 		GSF_INTERFACE (wbcc_gnm_cmd_context_init,
 			       GNM_CMD_CONTEXT_TYPE))
 
diff -urN gnumeric-1.4.3/src/command-context-stderr.c gnumeric-1.4.3-work/src/command-context-stderr.c
--- gnumeric-1.4.3/src/command-context-stderr.c	2004-06-13 20:48:34.000000000 +0200
+++ gnumeric-1.4.3-work/src/command-context-stderr.c	2005-09-11 21:16:51.000000000 +0200
@@ -107,6 +107,8 @@
 }
 
 GSF_CLASS_FULL (CmdContextStderr, cmd_context_stderr,
-		NULL, ccs_init,
-		G_TYPE_OBJECT, 0,
+		NULL, NULL, NULL, NULL,
+		ccs_init, G_TYPE_OBJECT, 0,
 		GSF_INTERFACE (ccs_gnm_cmd_context_init, GNM_CMD_CONTEXT_TYPE))
+
+/* EOF */
\ No newline at end of file
diff -urN gnumeric-1.4.3/src/cut-n-paste-code/goffice/graph/gog-axis.c gnumeric-1.4.3-work/src/cut-n-paste-code/goffice/graph/gog-axis.c
--- gnumeric-1.4.3/src/cut-n-paste-code/goffice/graph/gog-axis.c	2005-02-09 22:54:29.000000000 +0100
+++ gnumeric-1.4.3-work/src/cut-n-paste-code/goffice/graph/gog-axis.c	2005-09-11 17:25:53.000000000 +0200
@@ -1678,8 +1678,8 @@
 }
 
 GSF_CLASS_FULL (GogAxis, gog_axis,
-		gog_axis_class_init, gog_axis_init,
-		GOG_STYLED_OBJECT_TYPE, 0,
+		NULL, NULL, gog_axis_class_init, NULL,
+		gog_axis_init, GOG_STYLED_OBJECT_TYPE, 0,
 		GSF_INTERFACE (gog_axis_dataset_init, GOG_DATASET_TYPE))
 
 
diff -urN gnumeric-1.4.3/src/cut-n-paste-code/goffice/graph/gog-error-bar.c gnumeric-1.4.3-work/src/cut-n-paste-code/goffice/graph/gog-error-bar.c
--- gnumeric-1.4.3/src/cut-n-paste-code/goffice/graph/gog-error-bar.c	2004-12-02 14:57:26.000000000 +0100
+++ gnumeric-1.4.3-work/src/cut-n-paste-code/goffice/graph/gog-error-bar.c	2005-09-11 17:25:57.000000000 +0200
@@ -488,8 +488,8 @@
 }
 
 GSF_CLASS_FULL (GogErrorBar, gog_error_bar,
-		gog_error_bar_class_init, gog_error_bar_init,
-		G_TYPE_OBJECT, 0,
+		NULL, NULL, gog_error_bar_class_init, NULL,
+		gog_error_bar_init, GOG_STYLED_OBJECT_TYPE, 0,
 		GSF_INTERFACE (gog_error_bar_persist_init, GOG_PERSIST_TYPE))
 
 
diff -urN gnumeric-1.4.3/src/cut-n-paste-code/goffice/graph/gog-label.c gnumeric-1.4.3-work/src/cut-n-paste-code/goffice/graph/gog-label.c
--- gnumeric-1.4.3/src/cut-n-paste-code/goffice/graph/gog-label.c	2004-10-25 22:36:05.000000000 +0200
+++ gnumeric-1.4.3-work/src/cut-n-paste-code/goffice/graph/gog-label.c	2005-09-11 17:27:59.000000000 +0200
@@ -176,8 +176,8 @@
 }
 
 GSF_CLASS_FULL (GogLabel, gog_label,
-		gog_label_class_init, NULL,
-		GOG_OUTLINED_OBJECT_TYPE, 0,
+		NULL, NULL, gog_label_class_init, NULL,
+		NULL, GOG_STYLED_OBJECT_TYPE, 0,
 		GSF_INTERFACE (gog_label_dataset_init, GOG_DATASET_TYPE))
 
 /************************************************************************/
diff -urN gnumeric-1.4.3/src/cut-n-paste-code/goffice/graph/gog-series.c gnumeric-1.4.3-work/src/cut-n-paste-code/goffice/graph/gog-series.c
--- gnumeric-1.4.3/src/cut-n-paste-code/goffice/graph/gog-series.c	2005-01-05 19:05:53.000000000 +0100
+++ gnumeric-1.4.3-work/src/cut-n-paste-code/goffice/graph/gog-series.c	2005-09-11 17:32:55.000000000 +0200
@@ -589,8 +589,8 @@
 }
 
 GSF_CLASS_FULL (GogSeries, gog_series,
-		gog_series_class_init, gog_series_init,
-		GOG_STYLED_OBJECT_TYPE, 0,
+		NULL, NULL, gog_series_class_init, NULL,
+		gog_series_init, GOG_STYLED_OBJECT_TYPE, 0,
 		GSF_INTERFACE (gog_series_dataset_init, GOG_DATASET_TYPE))
 
 /**
diff -urN gnumeric-1.4.3/src/cut-n-paste-code/goffice/graph/gog-style.c gnumeric-1.4.3-work/src/cut-n-paste-code/goffice/graph/gog-style.c
--- gnumeric-1.4.3/src/cut-n-paste-code/goffice/graph/gog-style.c	2004-11-19 16:49:12.000000000 +0100
+++ gnumeric-1.4.3-work/src/cut-n-paste-code/goffice/graph/gog-style.c	2005-09-11 17:38:08.000000000 +0200
@@ -1687,8 +1687,8 @@
 }
 
 GSF_CLASS_FULL (GogStyle, gog_style,
-		gog_style_class_init, gog_style_init,
-		G_TYPE_OBJECT, 0,
+		NULL, NULL, gog_style_class_init, NULL,
+		gog_style_init, GOG_STYLED_OBJECT_TYPE, 0,
 		GSF_INTERFACE (gog_style_persist_init, GOG_PERSIST_TYPE))
 
 gboolean
diff -urN gnumeric-1.4.3/src/gnm-so-filled.c gnumeric-1.4.3-work/src/gnm-so-filled.c
--- gnumeric-1.4.3/src/gnm-so-filled.c	2004-10-17 04:36:05.000000000 +0200
+++ gnumeric-1.4.3-work/src/gnm-so-filled.c	2005-09-11 21:31:05.000000000 +0200
@@ -113,8 +113,8 @@
 typedef FooCanvasGroup		FilledFooView;
 typedef FooCanvasGroupClass	FilledFooViewClass;
 static GSF_CLASS_FULL (FilledFooView, so_filled_foo_view,
-	NULL, NULL,
-	FOO_TYPE_CANVAS_GROUP, 0,
+		NULL, NULL, NULL, NULL,
+		NULL, FOO_TYPE_CANVAS_GROUP, 0,
 	GSF_INTERFACE (so_filled_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 #endif /* WITH_GTK */
 
@@ -632,8 +632,8 @@
 typedef FooCanvasPolygon	PolygonFooView;
 typedef FooCanvasPolygonClass	PolygonFooViewClass;
 static GSF_CLASS_FULL (PolygonFooView, so_polygon_foo_view,
-	NULL, NULL,
-	FOO_TYPE_CANVAS_POLYGON, 0,
+	NULL, NULL, NULL, NULL,
+	NULL, FOO_TYPE_CANVAS_POLYGON, 0,
 	GSF_INTERFACE (so_polygon_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 #endif /* WITH_GTK */
 
diff -urN gnumeric-1.4.3/src/gnm-so-line.c gnumeric-1.4.3-work/src/gnm-so-line.c
--- gnumeric-1.4.3/src/gnm-so-line.c	2004-11-30 19:37:07.000000000 +0100
+++ gnumeric-1.4.3-work/src/gnm-so-line.c	2005-09-11 17:55:44.000000000 +0200
@@ -111,8 +111,8 @@
 typedef FooCanvasLine		LineFooView;
 typedef FooCanvasLineClass	LineFooViewClass;
 static GSF_CLASS_FULL (LineFooView, so_line_foo_view,
-	NULL, NULL,
-	FOO_TYPE_CANVAS_LINE, 0,
+		NULL, NULL, NULL, NULL,
+		NULL, FOO_TYPE_CANVAS_LINE, 0,
 	GSF_INTERFACE (so_line_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 #endif /* WITH_GTK */
 enum {
diff -urN gnumeric-1.4.3/src/io-context-gtk.c gnumeric-1.4.3-work/src/io-context-gtk.c
--- gnumeric-1.4.3/src/io-context-gtk.c	2004-10-30 05:51:35.000000000 +0200
+++ gnumeric-1.4.3-work/src/io-context-gtk.c	2005-09-11 17:58:49.000000000 +0200
@@ -367,8 +367,8 @@
 }
 
 GSF_CLASS_FULL (IOContextGtk, io_context_gtk,
-		icg_class_init, icg_init,
-		TYPE_IO_CONTEXT, 0,
+		NULL, NULL, icg_class_init, NULL,
+		icg_init, TYPE_IO_CONTEXT, 0,
 		GSF_INTERFACE (icg_gnm_cmd_context_init, GNM_CMD_CONTEXT_TYPE))
 
 void
diff -urN gnumeric-1.4.3/src/io-context.c gnumeric-1.4.3-work/src/io-context.c
--- gnumeric-1.4.3/src/io-context.c	2004-07-20 02:37:13.000000000 +0200
+++ gnumeric-1.4.3-work/src/io-context.c	2005-09-11 18:01:07.000000000 +0200
@@ -117,8 +117,8 @@
 }
 
 GSF_CLASS_FULL (IOContext, io_context,
-		io_context_class_init, io_context_init,
-		G_TYPE_OBJECT, 0,
+		NULL, NULL, io_context_class_init, NULL,
+		io_context_init, G_TYPE_OBJECT, 0,
 		GSF_INTERFACE (io_context_gnm_cmd_context_init, GNM_CMD_CONTEXT_TYPE))
 
 IOContext *
diff -urN gnumeric-1.4.3/src/sheet-filter.c gnumeric-1.4.3-work/src/sheet-filter.c
--- gnumeric-1.4.3/src/sheet-filter.c	2004-10-17 04:36:05.000000000 +0200
+++ gnumeric-1.4.3-work/src/sheet-filter.c	2005-09-11 18:03:13.000000000 +0200
@@ -563,8 +563,8 @@
 typedef FooCanvasWidget		FilterFooView;
 typedef FooCanvasWidgetClass	FilterFooViewClass;
 static GSF_CLASS_FULL (FilterFooView, filter_foo_view,
-	NULL, NULL,
-	FOO_TYPE_CANVAS_WIDGET, 0,
+		NULL, NULL, NULL, NULL,
+		NULL, FOO_TYPE_CANVAS_WIDGET, 0,
 	GSF_INTERFACE (filter_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 
 static SheetObjectView *
diff -urN gnumeric-1.4.3/src/sheet-object-cell-comment.c gnumeric-1.4.3-work/src/sheet-object-cell-comment.c
--- gnumeric-1.4.3/src/sheet-object-cell-comment.c	2005-01-14 06:48:29.000000000 +0100
+++ gnumeric-1.4.3-work/src/sheet-object-cell-comment.c	2005-09-11 18:04:54.000000000 +0200
@@ -123,8 +123,8 @@
 typedef FooCanvasPolygon	CommentFooView;
 typedef FooCanvasPolygonClass	CommentFooViewClass;
 static GSF_CLASS_FULL (CommentFooView, comment_foo_view,
-	NULL, NULL,
-	FOO_TYPE_CANVAS_POLYGON, 0,
+	NULL, NULL, NULL, NULL,
+	NULL, FOO_TYPE_CANVAS_POLYGON, 0,
 	GSF_INTERFACE (comment_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 
 static void
diff -urN gnumeric-1.4.3/src/sheet-object-graph.c gnumeric-1.4.3-work/src/sheet-object-graph.c
--- gnumeric-1.4.3/src/sheet-object-graph.c	2004-10-29 06:13:13.000000000 +0200
+++ gnumeric-1.4.3-work/src/sheet-object-graph.c	2005-09-11 18:06:20.000000000 +0200
@@ -94,8 +94,8 @@
 typedef GogControlFooCanvas		SOGraphFooView;
 typedef GogControlFooCanvasClass	SOGraphFooViewClass;
 static GSF_CLASS_FULL (SOGraphFooView, so_graph_foo_view,
-	NULL, NULL,
-	GOG_CONTROL_FOOCANVAS_TYPE, 0,
+	NULL, NULL,NULL, NULL,
+	NULL, GOG_CONTROL_FOOCANVAS_TYPE, 0,
 	GSF_INTERFACE (so_graph_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 
 /****************************************************************************/
diff -urN gnumeric-1.4.3/src/sheet-object-image.c gnumeric-1.4.3-work/src/sheet-object-image.c
--- gnumeric-1.4.3/src/sheet-object-image.c	2004-12-18 05:48:37.000000000 +0100
+++ gnumeric-1.4.3-work/src/sheet-object-image.c	2005-09-11 18:07:26.000000000 +0200
@@ -90,8 +90,8 @@
 typedef FooCanvasPixbuf		SOImageFooView;
 typedef FooCanvasPixbufClass	SOImageFooViewClass;
 static GSF_CLASS_FULL (SOImageFooView, so_image_foo_view,
-	NULL, NULL,
-	FOO_TYPE_CANVAS_PIXBUF, 0,
+	NULL, NULL, NULL, NULL,
+	NULL, FOO_TYPE_CANVAS_PIXBUF, 0,
 	GSF_INTERFACE (so_image_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 
 /****************************************************************************/
diff -urN gnumeric-1.4.3/src/sheet-object-widget.c gnumeric-1.4.3-work/src/sheet-object-widget.c
--- gnumeric-1.4.3/src/sheet-object-widget.c	2005-02-09 22:54:29.000000000 +0100
+++ gnumeric-1.4.3-work/src/sheet-object-widget.c	2005-09-11 18:08:09.000000000 +0200
@@ -99,8 +99,8 @@
 typedef FooCanvasWidget		SOWidgetFooView;
 typedef FooCanvasWidgetClass	SOWidgetFooViewClass;
 static GSF_CLASS_FULL (SOWidgetFooView, so_widget_foo_view,
-	NULL, NULL,
-	FOO_TYPE_CANVAS_WIDGET, 0,
+	NULL, NULL, NULL, NULL,
+	NULL, FOO_TYPE_CANVAS_WIDGET, 0,
 	GSF_INTERFACE (so_widget_foo_view_init, SHEET_OBJECT_VIEW_TYPE))
 
 /****************************************************************************/
diff -urN gnumeric-1.4.3/src/widgets/gnumeric-expr-entry.c gnumeric-1.4.3-work/src/widgets/gnumeric-expr-entry.c
--- gnumeric-1.4.3/src/widgets/gnumeric-expr-entry.c	2004-09-07 21:28:41.000000000 +0200
+++ gnumeric-1.4.3-work/src/widgets/gnumeric-expr-entry.c	2005-09-11 17:48:56.000000000 +0200
@@ -503,8 +503,8 @@
 }
 
 GSF_CLASS_FULL (GnmExprEntry, gnm_expr_entry,
-		gee_class_init, gee_init,
-		GTK_TYPE_HBOX, 0,
+		NULL, NULL, gee_class_init, NULL,
+		gee_init, GTK_TYPE_HBOX, 0,
 		GSF_INTERFACE (gee_cell_editable_init, GTK_TYPE_CELL_EDITABLE))
 
 /**
diff -urN gnumeric-1.4.3/src/workbook-control-gui.c gnumeric-1.4.3-work/src/workbook-control-gui.c
--- gnumeric-1.4.3/src/workbook-control-gui.c	2005-03-13 01:05:45.000000000 +0100
+++ gnumeric-1.4.3-work/src/workbook-control-gui.c	2005-09-11 18:10:24.000000000 +0200
@@ -2545,8 +2545,8 @@
 }
 
 GSF_CLASS_FULL (WorkbookControlGUI, workbook_control_gui,
-		workbook_control_gui_class_init, workbook_control_gui_init,
-		WORKBOOK_CONTROL_TYPE, G_TYPE_FLAG_ABSTRACT,
+		NULL, NULL, workbook_control_gui_class_init, NULL,
+		workbook_control_gui_init, WORKBOOK_CONTROL_TYPE, G_TYPE_FLAG_ABSTRACT,
 		GSF_INTERFACE (wbcg_go_plot_data_allocator_init, GOG_DATA_ALLOCATOR_TYPE);
 		GSF_INTERFACE (wbcg_gnm_cmd_context_init, GNM_CMD_CONTEXT_TYPE))
 
