diff -Naur loudmouth-1.4.3/configure.ac loudmouth-1.4.3-fix/configure.ac
--- loudmouth-1.4.3/configure.ac	2008-10-29 21:23:52.000000000 +0100
+++ loudmouth-1.4.3-fix/configure.ac	2016-01-10 03:37:38.019693080 +0100
@@ -263,9 +263,6 @@
 docs/Makefile
 docs/reference/Makefile
 loudmouth/Makefile
-examples/Makefile
-tests/Makefile
-tests/parser-tests/Makefile
 loudmouth-1.0.pc
 loudmouth.spec])
 
diff -Naur loudmouth-1.4.3/Makefile.am loudmouth-1.4.3-fix/Makefile.am
--- loudmouth-1.4.3/Makefile.am	2008-10-29 14:19:24.000000000 +0100
+++ loudmouth-1.4.3-fix/Makefile.am	2016-01-10 03:35:15.109967559 +0100
@@ -1,4 +1,4 @@
-SUBDIRS = loudmouth docs examples tests
+SUBDIRS = loudmouth docs
 
 EXTRA_DIST =			\
 	loudmouth-1.0.pc.in     \
diff -Naur loudmouth-1.4.3/Makefile.in loudmouth-1.4.3-fix/Makefile.in
--- loudmouth-1.4.3/Makefile.in	2008-10-29 21:56:10.000000000 +0100
+++ loudmouth-1.4.3-fix/Makefile.in	2016-01-10 03:37:06.108861280 +0100
@@ -220,7 +220,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = loudmouth docs examples tests
+SUBDIRS = loudmouth docs
 EXTRA_DIST = \
 	loudmouth-1.0.pc.in     \
         loudmouth.spec          \
