diff -Naur audience-20120708/Audience/Widgets/TagView.vala audience-20120708-b/Audience/Widgets/TagView.vala
--- audience-20120708/Audience/Widgets/TagView.vala	2012-07-08 11:58:07.000000000 +0200
+++ audience-20120708-b/Audience/Widgets/TagView.vala	2012-07-08 12:10:57.000000000 +0200
@@ -38,7 +38,7 @@
             this.reactive = true;
             this.buffer   = new Granite.Drawing.BufferSurface (100, 100);
             
-            var notebook = new Granite.Widgets.StaticNotebook (false);
+            var notebook = new Granite.Widgets.StaticNotebook ();
             
             /*tags*/
             var tagview = new Gtk.ScrolledWindow (null, null);