From eca00c65ca79d4a55b34efe43604aaf7a876a399 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Se=C3=A1n=20de=20B=C3=BArca?= <leftmostcat@gmail.com>
Date: Tue, 10 Sep 2013 21:29:25 -0600
Subject: [PATCH] Allow use of tracker-sparql-0.16 as well

---
 configure.ac | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/configure.ac b/configure.ac
index 508b612..8af7ca9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -358,11 +358,14 @@ AC_ARG_ENABLE([webremote],
               [],
               [enable_webremote=yes])
 
-PKG_CHECK_EXISTS([tracker-sparql-0.14],
-		 [tracker_dep="tracker-sparql-0.14"],
-		 [PKG_CHECK_EXISTS([tracker-sparql-0.12],
-			[tracker_dep="tracker-sparql-0.12"],
-			[tracker_dep="tracker-sparql-0.10"])
+PKG_CHECK_EXISTS([tracker-sparql-0.16],
+		 [tracker_dep="tracker-sparql-0.16"],
+		 [PKG_CHECK_EXISTS([tracker-sparql-0.14],
+			[tracker_dep="tracker-sparql-0.14"],
+			[PKG_CHECK_EXISTS([tracker-sparql-0.12],
+				[tracker_dep="tracker-sparql-0.12"],
+				[tracker_dep="tracker-sparql-0.10"])
+			 ])
 		 ])
 
 AS_IF([test "x$enable_webremote" = "xyes"],
-- 
1.8.5.5

