diff -Naur iptables-1.4.0.orig/include/linux/netfilter_ipv4/ip_tables.h iptables-1.4.0/include/linux/netfilter_ipv4/ip_tables.h
--- iptables-1.4.0.orig/include/linux/netfilter_ipv4/ip_tables.h	2007-12-22 13:44:17.000000000 +0100
+++ iptables-1.4.0/include/linux/netfilter_ipv4/ip_tables.h	2008-01-03 23:39:55.000000000 +0100
@@ -15,7 +15,6 @@
 #ifndef _IPTABLES_H
 #define _IPTABLES_H
 
-#include <linux/compiler.h>
 #include <linux/netfilter_ipv4.h>
 
 #include <linux/netfilter/x_tables.h>
@@ -187,7 +186,7 @@
 	/* Number of counters (must be equal to current number of entries). */
 	unsigned int num_counters;
 	/* The old entries' counters. */
-	struct xt_counters __user *counters;
+	struct xt_counters *counters;
 
 	/* The entries (hang off end: not really an array). */
 	struct ipt_entry entries[0];
diff -Naur iptables-1.4.0.orig/include/linux/netfilter_ipv6/ip6_tables.h iptables-1.4.0/include/linux/netfilter_ipv6/ip6_tables.h
--- iptables-1.4.0.orig/include/linux/netfilter_ipv6/ip6_tables.h	2007-12-22 13:44:17.000000000 +0100
+++ iptables-1.4.0/include/linux/netfilter_ipv6/ip6_tables.h	2008-01-03 23:39:46.000000000 +0100
@@ -15,7 +15,6 @@
 #ifndef _IP6_TABLES_H
 #define _IP6_TABLES_H
 
-#include <linux/compiler.h>
 #include <linux/netfilter_ipv6.h>
 
 #include <linux/netfilter/x_tables.h>
@@ -247,7 +246,7 @@
 	/* Number of counters (must be equal to current number of entries). */
 	unsigned int num_counters;
 	/* The old entries' counters. */
-	struct xt_counters __user *counters;
+	struct xt_counters *counters;
 
 	/* The entries (hang off end: not really an array). */
 	struct ip6t_entry entries[0];
