From 4d3757b1b935f2cbb4b5061d2a075f9383fb2998 Mon Sep 17 00:00:00 2001
From: Sandro Stiller <sandro.stiller@elfin.de>
Date: Tue, 7 May 2019 14:17:37 +0200
Subject: [PATCH] Fix pthreads linking error with gcc-7.3.0

---
 tools/Makefile.am | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/tools/Makefile.am b/tools/Makefile.am
index 85df1f5..dc811c9 100644
--- a/tools/Makefile.am
+++ b/tools/Makefile.am
@@ -5,7 +5,8 @@ AM_CPPFLAGS = \
 	$(xml_CFLAGS) \
 	-I$(top_srcdir)/include \
 	-I$(top_builddir)/include \
-	-Wall
+	-Wall \
+  $(PTHREAD_CFLAGS)
 
 if CROSS_COMPILING
 libdbus_cxx_la = $(BUILD_LIBDBUS_CXX_DIR)/src/libdbus-c++-1.la
@@ -21,7 +22,7 @@ dbusxx_xml2cpp_SOURCES = xml.h xml.cpp xml2cpp.h xml2cpp.cpp \
 dbusxx_xml2cpp_LDADD = $(xml_LIBS)
 
 dbusxx_introspect_SOURCES = introspect.h introspect.cpp
-dbusxx_introspect_LDADD = $(libdbus_cxx_la)
+dbusxx_introspect_LDADD = $(libdbus_cxx_la) $(PTHREAD_LIBS)
 
 MAINTAINERCLEANFILES = \
 	Makefile.in
