diff -Naur knx-0.1.orig/knx/src/knxconnection.cpp knx-0.1/knx/src/knxconnection.cpp
--- knx-0.1.orig/knx/src/knxconnection.cpp	2004-09-24 12:52:17.000000000 +0200
+++ knx-0.1/knx/src/knxconnection.cpp	2006-06-23 23:41:39.000000000 +0200
@@ -135,7 +135,7 @@
 					if (findLineStartingWith("HELLO NXSERVER")) {
 						if (waitsForInput()) {
 							if (m_connecting) m_connecting->setActivity(i18n("Setting initial connection parameters"));
-							sendString("hello NXCLIENT - Version 1.4.0\n");
+							sendString("hello NXCLIENT - Version 1.3.2\n");
 							m_state=PREAMBLE_SHELL;
 							m_connectionTimeout->changeInterval(5000);
 						}
