--- src/gtk/glx_3Demu.cpp.orig	2014-03-11 20:20:39.389282149 +0100
+++ src/gtk/glx_3Demu.cpp	2014-03-11 20:21:18.555607566 +0100
@@ -31,7 +31,7 @@
 static GLXContext ctx;
 static GLXPbuffer pbuf;
 
-void deinit_glx_3Demu(void)
+int deinit_glx_3Demu(void)
 {
     Display *dpy = glXGetCurrentDisplay();
 
@@ -42,10 +42,8 @@
 
         XCloseDisplay(dpy);
 
-        return true;
     }
 
-    return false;
 }
 
 int init_glx_3Demu(void) 
