diff -Naur albumart-1.6.0.old/lib/albumart/albumart_images.py albumart-1.6.0/lib/albumart/albumart_images.py
--- albumart-1.6.0.old/lib/albumart/albumart_images.py	2005-02-26 20:27:07.000000000 +0530
+++ albumart-1.6.0/lib/albumart/albumart_images.py	2007-03-03 22:52:27.000000000 +0530
@@ -3,7 +3,7 @@
 # Generated from reading image files: 
 #      ../../share/albumart/nocover.png
 #
-# Created: la helmi  26 16:52:36 2005
+# Created: la helmi   26 16:52:36 2005
 #      by: The PyQt User Interface Compiler (pyuic) 3.13
 #
 # WARNING! All changes made in this file will be lost!
