diff -Naur k3d-0.5.15.0/ngui/file_chooser_dialog.cpp k3d-0.5.15.0-compile-fix/ngui/file_chooser_dialog.cpp
--- k3d-0.5.15.0/ngui/file_chooser_dialog.cpp	2006-06-21 04:55:55.000000000 +0200
+++ k3d-0.5.15.0-compile-fix/ngui/file_chooser_dialog.cpp	2006-07-16 18:48:32.173155096 +0200
@@ -198,7 +198,7 @@
 	add_shortcut_folder(k3d::share_path().native_file_string());
 
 	add_action_widget(cancel_widget, Gtk::RESPONSE_CANCEL);
-	switch(get_action())
+	switch(Gtk::FileChooser::get_action())
 	{
 		case Gtk::FILE_CHOOSER_ACTION_OPEN:
 		case Gtk::FILE_CHOOSER_ACTION_SELECT_FOLDER:
@@ -254,7 +254,7 @@
 	// Record the path for posterity ...
 	k3d::options::set_path(path_type, Result);
 
-	switch(get_action())
+	switch(Gtk::FileChooser::get_action())
 	{
 		case Gtk::FILE_CHOOSER_ACTION_OPEN:
 		case Gtk::FILE_CHOOSER_ACTION_SELECT_FOLDER:
@@ -272,7 +272,7 @@
 
 bool file_chooser_dialog::prompt_file_overwrite(const boost::filesystem::path& File)
 {
-	switch(get_action())
+	switch(Gtk::FileChooser::get_action())
 	{
 		case Gtk::FILE_CHOOSER_ACTION_OPEN:
 		case Gtk::FILE_CHOOSER_ACTION_SELECT_FOLDER:
