--- network-manager-applet-0.8/nm-applet-dbus-126.conf.old	2009-12-23 19:30:42.000000000 +0100
+++ network-manager-applet-0.8/nm-applet-dbus-126.conf	2010-04-04 21:32:40.000000000 +0200
@@ -23,7 +23,7 @@
                 <allow send_destination="org.freedesktop.NetworkManagerUserSettings"
                        send_interface="org.freedesktop.NetworkManagerSettings.Connection.Secrets"/>
         </policy>
-        <policy at_console="true">
+        <policy group="netdev">
                 <allow own="org.freedesktop.NetworkManagerUserSettings"/>
 
                 <allow send_destination="org.freedesktop.NetworkManagerUserSettings"
