From fa9fc090e59c3b1745da05e066a335d516b756d4 Mon Sep 17 00:00:00 2001
From: Vincent Untz <vuntz@gnome.org>
Date: Mon, 09 Aug 2010 15:34:11 +0000
Subject: build: fix compilation with glib >= 2.25.12 (bgo #626424)

Work around API breakage in glib.
---
diff --git a/src/utils/tests/test-utils.c b/src/utils/tests/test-utils.c
index 7eb1ffa..83b8ffe 100644
--- a/src/utils/tests/test-utils.c
+++ b/src/utils/tests/test-utils.c
@@ -413,7 +413,11 @@ test_ap_hash_foobar_asdf11_adhoc_wpa_rsn (void *f, TestData *d)
 	g_assert (strcmp (d->foobar_adhoc_wpa_rsn, d->asdf11_adhoc_wpa_rsn));
 }
 
+#if GLIB_CHECK_VERSION(2,25,12)
+typedef GTestFixtureFunc TCFunc;
+#else
 typedef void (*TCFunc)(void);
+#endif
 
 #define TESTCASE(t, d) g_test_create_case (#t, 0, d, NULL, (TCFunc) t, NULL)
 
--
cgit v0.8.3.1
