diff -ur mplayerplug-in-orig/Source/plugin-ui.cpp mplayerplug-in/Source/plugin-ui.cpp
--- mplayerplug-in-orig/Source/plugin-ui.cpp	2006-04-03 16:33:48.362321612 +0000
+++ mplayerplug-in/Source/plugin-ui.cpp	2006-04-03 16:34:34.376441250 +0000
@@ -280,11 +280,6 @@
     int term, chop;
     int PercentRedraw;
 
-    if (0) {
-	printf("Widget: %i\n", (int) w);
-	printf("Message: %s\n", message);
-    }
-
     if (DEBUG > 1)
 	printf("state = %i, height = %i width = %i\n", instance->state,
 	       instance->movie_height, instance->movie_width);

