--- mpage-2.5.4/mpage.h.gcc4	2005-02-17 16:56:35.385185734 +0000
+++ mpage-2.5.4/mpage.h	2005-02-17 16:57:20.440559802 +0000
@@ -172,6 +172,16 @@
 extern int ps_width;	/* number of points in the X direction (8.5 inches) */
 extern int ps_height;	/* number of points in the Y direction (11 inches) */
 extern char * media;	/* name of output page media */
+
+/*
+ * Structure to describe a physical piece of paper, e.g. A4 or Letter
+ */
+struct page_desc {
+    char *media;
+    int width;
+    int height;
+};
+
 extern struct page_desc paper[];
 
 /*
@@ -213,16 +223,6 @@
 };
 
 
-/*
- * Structure to describe a physical piece of paper, e.g. A4 or Letter
- */
-struct page_desc {
-    char *media;
-    int width;
-    int height;
-};
-    
-
 /* array of sheets where pages are ordered for coli ??? */
 extern struct sheet coli[];
 
