--- KTrafficAnalyzer_0.3.3/Makefile	2006-05-16 14:57:21.000000000 +0200
+++ KTrafficAnalyzer_0.3.3-p/Makefile	2006-05-17 13:51:34.000000000 +0200
@@ -5,20 +5,24 @@
 SRC_ROOT	= `pwd`
 LFLAGS		= $(ADD_LFLAGS) -L$(QTDIR)/lib/ -L$(KDEDIR)/lib/ -lqt-mt -lkdeui -lkio
 CFLAGS		= $(ADD_CFLAGS) -I$(QTDIR)/include/ -I$(KDEDIR)/include/ -I$(SRC_ROOT)
+MYFLAGS	       += $(CXXFLAGS)
 ECHO		= echo
-KDEDIR		= `kde-config --prefix`
+KKDEDIR		= `kde-config --prefix`
+KDEDIR		= $(KKDEDIR)
+CP		= cp
+MKDIR		= mkdir -p
 
 do-it-always:
 	if test "$(KDEDIR)" = ""; then $(ECHO) KDEDIR not set!; exit 1; fi;
 	if test "$(QTDIR)" = ""; then $(ECHO) QTDIR not set!; exit 1; fi;
-	for i in $(OBJ); do $(MAKE) -C $$i SRC_ROOT=$(SRC_ROOT) "CFLAGS=$(CFLAGS)"; done
-	gcc -c KTrafficAnalyzer.cpp $(CFLAGS)
+	for i in $(OBJ); do $(MAKE) -C $$i SRC_ROOT=$(SRC_ROOT) "CFLAGS=$(CFLAGS) $(MYFLAGS)"; done
+	gcc -c KTrafficAnalyzer.cpp $(CFLAGS) $(MYFLAGS)
 	gcc -o $(TARGET) *.o $(addsuffix /*.o,$(OBJ)) $(LFLAGS)
 
 install:
-	if test "$(KDEDIR)" = ""; then $(ECHO) KDEDIR not set!; exit 1; fi;
-	cp KTrafficAnalyzer $(KDEDIR)/bin/
-	ln -s $(KDEDIR)/bin/KTrafficAnalyzer $(KDEDIR)/bin/ktrafficanalyzer
+	$(MKDIR) $(DESTDIR)$(KDEDIR)/bin/
+	$(CP) KTrafficAnalyzer $(DESTDIR)$(KDEDIR)/bin/
+	
 
 uninstall:
 	if test "$(KDEDIR)" = ""; then $(ECHO) KDEDIR not set!; exit 1; fi;
