diff -Naur blam-1.8.2.orig/libblam/blam-gecko-utils.cpp blam-1.8.2/libblam/blam-gecko-utils.cpp
--- blam-1.8.2.orig/libblam/blam-gecko-utils.cpp	2005-05-06 10:07:49.000000000 +0200
+++ blam-1.8.2/libblam/blam-gecko-utils.cpp	2005-12-07 23:33:16.000000000 +0100
@@ -29,6 +29,8 @@
 #include <nsEmbedString.h>
 #include <nsIPrefService.h>
 #include <nsIServiceManager.h>
+#include <nsCOMPtr.h>
+#include <nsServiceManagerUtils.h>
 
 #if defined (HAVE_CHROME_NSICHROMEREGISTRYSEA_H)
 #include <chrome/nsIChromeRegistrySea.h>
