--- torcs-1.3.0/configure.in.orig	2006-11-09 01:21:57.000000000 +0100
+++ torcs-1.3.0/configure.in	2006-11-09 01:23:24.000000000 +0100
@@ -97,7 +97,9 @@
 dnl Replace `main' with a function in -lm:
 AC_CHECK_LIB(m, sin,,AC_MSG_ERROR([Can't find libm. Please check config.log and if you can't solve the problem send the file to torcs-users@lists.sourceforge.net with the subject \"torcs compilation problem\"]))
 
-LDFLAGS="$LDFLAGS -L$x_libraries"
+if test "x$x_libraries" != "x"; then
+	LDFLAGS="$LDFLAGS -L$x_libraries"
+fi
 
 dnl Replace `main' with a function in -lX11:
 AC_CHECK_LIB(X11, XOpenDisplay,,AC_MSG_ERROR([Can't find libX11. Please check config.log and if you can't solve the problem send the file to torcs-users@lists.sourceforge.net with the subject \"torcs compilation problem\"]))
