diff -u mlt-0.4.10/src/modules/qimageconfigure mlt-0.4.10/src/modules/qimage/configure
--- mlt-0.4.10/src/modules/qimage/configure	2009-12-13 02:21:41.000000000 +0100
+++ mlt-0.4.10/src/modules/qimage/configure	2009-12-13 02:21:29.000000000 +0100
@@ -29,8 +29,8 @@
 		;;
 	esac
 
-	qimage_includedir=/usr/include/qt4
-	qimage_libdir=/usr/lib/qt4
+	qimage_includedir=/usr/include/qt
+	qimage_libdir=/usr/lib/qt
 
 	if [ ! -d "$qimage_libdir" -o ! -d "$qimage_includedir" ]
 	then
@@ -48,7 +48,7 @@
 	if [ "$QTDIR" != "" ]
 	then
 		qimage_includedir="$QTDIR/include"
-		qimage_libdir="$QTDIR/lib"
+		qimage_libdir="$QTDIR"
 	fi
 
 	export force_qt3=
