diff -aur mlt-0.2.4/src/modules/avformat/configure fw_mlt-0.2.4/src/modules/avformat/configure
--- mlt-0.2.4/src/modules/avformat/configure	2007-07-15 09:43:48.000000000 +0200
+++ fw_mlt-0.2.4/src/modules/avformat/configure	2008-04-28 14:36:45.000000000 +0200
@@ -32,7 +32,7 @@
 	bits=$(uname -m)
 	case $bits in
 	x86_64)
-		export LIBDIR=lib64
+		export LIBDIR=lib
 		;;
 	*)
 		export LIBDIR=lib
@@ -125,9 +125,9 @@
 			echo 0
 		fi
 	else 
-		if [ -d "$shared_ffmpeg/include/ffmpeg" -a -f "$shared_ffmpeg/$LIBDIR/libavformat$avformat_suffix$LIBSUF" ]
+		if [ -d "$shared_ffmpeg/include" -a -f "$shared_ffmpeg/$LIBDIR/libavformat$avformat_suffix$LIBSUF" ]
 		then
-			echo "CFLAGS+=-I$shared_ffmpeg/include/ffmpeg " >> config.mak
+			echo "CFLAGS+=-I$shared_ffmpeg/include " >> config.mak
 			echo "LDFLAGS+=-L$shared_ffmpeg/$LIBDIR" >> config.mak
 			[ "$swscale" != "" ] && echo "SWSCALE=1" >> config.mak
 		else
