--- /src/modules/avformat/producer_avformat.c	2009-12-09 05:38:48.000000000 +0100
+++ /src/modules/avformat/producer_avformat.c	2009-12-12 21:28:54.000000000 +0100
@@ -26,10 +26,10 @@
 #include <framework/mlt_log.h>
 
 // ffmpeg Header files
-#include <avformat.h>
+#include <libavformat/avformat.h>
 #include <opt.h>
 #ifdef SWSCALE
-#  include <swscale.h>
+#include <libswscale/swscale.h>
 #endif
 #if (LIBAVCODEC_VERSION_INT >= ((51<<16)+(71<<8)+0))
 #  include "audioconvert.h"
