diff -aur mlt-0.2.4/src/modules/avformat/consumer_avformat.c fw_mlt-0.2.4/src/modules/avformat/consumer_avformat.c
--- mlt-0.2.4/src/modules/avformat/consumer_avformat.c	2007-07-30 21:11:49.000000000 +0200
+++ fw_mlt-0.2.4/src/modules/avformat/consumer_avformat.c	2008-04-28 14:51:13.000000000 +0200
@@ -34,9 +34,9 @@
 #include <math.h>
 
 // avformat header files
-#include <avformat.h>
+#include <libavformat/avformat.h>
 #ifdef SWSCALE
-#include <swscale.h>
+#include <libswscale/swscale.h>
 #endif
 
 //
diff -aur mlt-0.2.4/src/modules/avformat/factory.c fw_mlt-0.2.4/src/modules/avformat/factory.c
--- mlt-0.2.4/src/modules/avformat/factory.c	2007-03-30 08:55:55.000000000 +0200
+++ fw_mlt-0.2.4/src/modules/avformat/factory.c	2008-04-28 14:48:15.000000000 +0200
@@ -29,7 +29,7 @@
 #include "filter_avresample.h"
 
 // ffmpeg Header files
-#include <avformat.h>
+#include <libavformat/avformat.h>
 
 // A static flag used to determine if avformat has been initialised
 static int avformat_initialised = 0;
diff -aur mlt-0.2.4/src/modules/avformat/filter_avcolour_space.c fw_mlt-0.2.4/src/modules/avformat/filter_avcolour_space.c
--- mlt-0.2.4/src/modules/avformat/filter_avcolour_space.c	2007-03-30 08:55:55.000000000 +0200
+++ fw_mlt-0.2.4/src/modules/avformat/filter_avcolour_space.c	2008-04-28 15:04:39.000000000 +0200
@@ -23,9 +23,9 @@
 #include <framework/mlt_frame.h>
 
 // ffmpeg Header files
-#include <avformat.h>
+#include <libavformat/avformat.h>
 #ifdef SWSCALE
-#include <swscale.h>
+#include <libswscale/swscale.h>
 #endif
 
 #include <stdio.h>
diff -aur mlt-0.2.4/src/modules/avformat/filter_avdeinterlace.c fw_mlt-0.2.4/src/modules/avformat/filter_avdeinterlace.c
--- mlt-0.2.4/src/modules/avformat/filter_avdeinterlace.c	2007-03-30 08:55:55.000000000 +0200
+++ fw_mlt-0.2.4/src/modules/avformat/filter_avdeinterlace.c	2008-04-28 15:06:58.000000000 +0200
@@ -26,7 +26,7 @@
 #include <stdlib.h>
 
 // ffmpeg Header files
-#include <avformat.h>
+#include <libavformat/avformat.h>
 
 #ifdef USE_MMX
 #include "mmx.h"
diff -aur mlt-0.2.4/src/modules/avformat/filter_avresample.c fw_mlt-0.2.4/src/modules/avformat/filter_avresample.c
--- mlt-0.2.4/src/modules/avformat/filter_avresample.c	2007-03-30 08:55:55.000000000 +0200
+++ fw_mlt-0.2.4/src/modules/avformat/filter_avresample.c	2008-04-28 15:06:08.000000000 +0200
@@ -27,7 +27,7 @@
 #include <string.h>
 
 // ffmpeg Header files
-#include <avformat.h>
+#include <libavformat/avformat.h>
 
 /** Get the audio.
 */
diff -aur mlt-0.2.4/src/modules/avformat/producer_avformat.c fw_mlt-0.2.4/src/modules/avformat/producer_avformat.c
--- mlt-0.2.4/src/modules/avformat/producer_avformat.c	2007-07-15 00:34:45.000000000 +0200
+++ fw_mlt-0.2.4/src/modules/avformat/producer_avformat.c	2008-04-28 15:00:03.000000000 +0200
@@ -25,9 +25,9 @@
 #include <framework/mlt_frame.h>
 
 // ffmpeg Header files
-#include <avformat.h>
+#include <libavformat/avformat.h>
 #ifdef SWSCALE
-#include <swscale.h>
+#include <libswscale/swscale.h>
 #endif
 
 // System header files
@@ -203,7 +203,7 @@
 			params->width = 640;
 			params->height = 480;
 			params->time_base= (AVRational){1,25};
-			params->device = file;
+			// params->device = file;
 			params->channels = 2;
 			params->sample_rate = 48000;
 		}
