diff -ur a/src/modules/avformat/consumer_avformat.c b/src/modules/avformat/consumer_avformat.c
--- a/src/modules/avformat/consumer_avformat.c	2012-10-30 04:53:45.958351499 -0500
+++ b/src/modules/avformat/consumer_avformat.c	2012-10-30 05:02:34.059717108 -0500
@@ -238,9 +238,9 @@
 		mlt_properties_set_data( doc, "audio_codecs", codecs, 0, NULL, NULL );
 		while ( ( codec = av_codec_next( codec ) ) )
 #if LIBAVCODEC_VERSION_INT >= ((54<<16)+(0<<8)+0)
-			if ( ( codec->encode || codec->encode2 ) && codec->type == CODEC_TYPE_AUDIO )
+			if ( codec->encode2  && codec->type == CODEC_TYPE_AUDIO )
 #else
-			if ( codec->encode && codec->type == CODEC_TYPE_AUDIO )
+			if ( codec->type == CODEC_TYPE_AUDIO )
 #endif
 			{
 				snprintf( key, sizeof(key), "%d", mlt_properties_count( codecs ) );
@@ -262,9 +262,9 @@
 		mlt_properties_set_data( doc, "video_codecs", codecs, 0, NULL, NULL );
 		while ( ( codec = av_codec_next( codec ) ) )
 #if LIBAVCODEC_VERSION_INT >= ((54<<16)+(0<<8)+0)
-			if ( (codec->encode || codec->encode2) && codec->type == CODEC_TYPE_VIDEO )
+			if ( codec->encode2 && codec->type == CODEC_TYPE_VIDEO )
 #else
-			if ( codec->encode && codec->type == CODEC_TYPE_VIDEO )
+			if ( codec->type == CODEC_TYPE_VIDEO )
 #endif
 			{
 				snprintf( key, sizeof(key), "%d", mlt_properties_count( codecs ) );
