--- src/modules/kino/avi.cc
+++ src/modules/kino/avi.cc
@@ -175,6 +175,7 @@
 // C++ includes
 
 #include <string>
+#include <cstring>
 #include <iostream>
 #include <iomanip>
 
--- src/modules/kino/filehandler.cc
+++ src/modules/kino/filehandler.cc
@@ -24,6 +24,7 @@
 }
 
 #include <string>
+#include <cstring>
 #include <iostream>
 #include <sstream>
 #include <iomanip>
--- src/modules/kino/kino_wrapper.cc
+++ src/modules/kino/kino_wrapper.cc
@@ -19,7 +19,8 @@
  */
 
 #include "config.h"
-#include <strings.h>
+#include <cstring>
+#include <cstdlib>
 #include "kino_wrapper.h"
 #include "filehandler.h"
 
