diff -Naurp mlt-0.2.2/configure mlt-0.2.2-p/configure
--- mlt-0.2.2/configure	2006-05-27 21:58:15.000000000 +0200
+++ mlt-0.2.2-p/configure	2006-12-28 18:14:39.000000000 +0100
@@ -54,9 +54,8 @@ function build_config
 		[ "$cpu" != "" ] &&
 		echo "TARGETARCH=-march=$cpu" &&
 		echo "TARGETCPU=-mcpu=$cpu"
-		echo "OPTIMISATIONS=-O4 -pipe -fomit-frame-pointer"
 
-		echo "CFLAGS+=-Wall -fPIC -DPIC \$(TARGETARCH) \$(TARGETCPU) \$(OPTIMISATIONS) \$(MMX_FLAGS) \$(DEBUG_FLAGS) \$(LARGE_FILE)"
+		echo "CFLAGS+=-Wall -fPIC -DPIC \$(TARGETARCH) \$(TARGETCPU)  \$(MMX_FLAGS) \$(DEBUG_FLAGS) \$(LARGE_FILE)"
 
 		case $targetos in
 		Darwin)
@@ -65,7 +64,6 @@ function build_config
 		echo "LDFLAGS+=`sdl-config --libs`"
 		;;
 		Linux)
-		echo "OPTIMISATIONS+=-ffast-math"
 		echo "CFLAGS+=-pthread"
 		echo "SHFLAGS=-shared"
 		echo "LIBDL=-ldl"
