diff -Naur libexif-0.6.12.orig/libexif/exif-data.c libexif-0.6.12/libexif/exif-data.c
--- libexif-0.6.12.orig/libexif/exif-data.c	2005-08-16 12:41:35.000000000 +0200
+++ libexif-0.6.12/libexif/exif-data.c	2005-08-16 12:42:11.000000000 +0200
@@ -231,7 +231,7 @@
 		 * the offset must be an even number. If we need to introduce
 		 * a padding byte, we set it to 0.
 		 */
-		if (s & 1) *ds++;
+		if (s & 1) (*ds)++;
 		*d = exif_mem_realloc (data->priv->mem, *d, *ds);
 		if (!*d) {
 			EXIF_LOG_NO_MEMORY (data->priv->log, "ExifData", *ds);
