diff -Naur QScintilla-1.73-gpl-2.1/designer-Qt3/designer.pro QScintilla-1.73-gpl-2.1-p/designer-Qt3/designer.pro
--- QScintilla-1.73-gpl-2.1/designer-Qt3/designer.pro	2006-09-16 18:07:21.000000000 +0200
+++ QScintilla-1.73-gpl-2.1-p/designer-Qt3/designer.pro	2007-06-04 02:03:46.000000000 +0200
@@ -3,10 +3,10 @@
 
 TEMPLATE = lib
 TARGET = qscintillaplugin
-DESTDIR = $(QTDIR)/plugins/designer
+DESTDIR = $(INSTALL_ROOT)$(QTDIR)/plugins/designer
 
 CONFIG += qt warn_on release plugin
 
 SOURCES += qscintillaplugin.cpp
 
-LIBS += -lqscintilla2
+LIBS +=  -lqscintilla2

