diff -Naur QScintilla-gpl-2.4.5//Python/sip/qsciscintillabase3.sip QScintilla-gpl-2.4.5-p//Python/sip/qsciscintillabase3.sip
--- QScintilla-gpl-2.4.5//Python/sip/qsciscintillabase3.sip	2010-08-31 09:49:37.000000000 +0200
+++ QScintilla-gpl-2.4.5-p//Python/sip/qsciscintillabase3.sip	2010-09-04 15:05:21.169000031 +0200
@@ -1025,17 +1025,7 @@
             long lParam = 0) const;
     long SendScintilla(unsigned int msg, unsigned long wParam,
             void *lParam) const;
-    long SendScintilla(unsigned int msg, unsigned long wParam,
-            const char *lParam /DocType="Py_v3:bytes;str", Encoding="None"/) const;
-    long SendScintilla(unsigned int msg,
-            const char *lParam /DocType="Py_v3:bytes;str", Encoding="None"/); const
-    long SendScintilla(unsigned int msg,
-            const char *wParam /DocType="Py_v3:bytes;str", Encoding="None"/,
-            const char *lParam /DocType="Py_v3:bytes;str", Encoding="None"/) const;
     long SendScintilla(unsigned int msg, long wParam) const;
-    //long SendScintilla(unsigned int msg, int wParam) const;
-    long SendScintilla(unsigned int msg, long cpMin, long cpMax,
-            char *lpstrText /DocType="Py_v3:bytes;str", Encoding="None"/) const;
     long SendScintilla(unsigned int msg, unsigned long wParam,
             const QColor &col) const;
     long SendScintilla(unsigned int msg, const QColor &col) const;
