diff -Naur xdemos.orig/Makefile xdemos/Makefile
--- xdemos.orig/Makefile	2005-11-10 17:35:07.000000000 +0100
+++ xdemos/Makefile	2006-01-26 21:55:29.000000000 +0100
@@ -1,12 +1,6 @@
 # progs/xdemos/Makefile
 
-TOP = ../..
-include $(TOP)/configs/current
-
-
-INCDIR = $(TOP)/include
-
-LIB_DEP = $(LIB_DIR)/$(GL_LIB_NAME) $(LIB_DIR)/$(GLU_LIB_NAME)
+APP_LIB_DEPS = -lGL -lGLU -lglut -lm -lXi -lXmu
 
 PROGS = glthreads \
 	glxdemo \
@@ -36,8 +30,8 @@
 .SUFFIXES:
 .SUFFIXES: .c
 
-.c: $(LIB_DEP)
-	$(CC) -I$(INCDIR) $(CFLAGS) $< $(APP_LIB_DEPS) -o $@
+.c:
+	$(CC) $(CFLAGS) $< $(APP_LIB_DEPS) -o $@
 
 
 ##### TARGETS #####
@@ -58,26 +52,25 @@
 	$(CC) $(CFLAGS) pbdemo.o pbutil.o $(APP_LIB_DEPS) -o $@
 
 pbinfo.o: pbinfo.c pbutil.h
-	$(CC) -c -I. -I$(INCDIR) $(CFLAGS) pbinfo.c
+	$(CC) -c -I. $(CFLAGS) pbinfo.c
 
 pbdemo.o: pbdemo.c pbutil.h
-	$(CC) -c -I. -I$(INCDIR) $(CFLAGS) pbdemo.c
+	$(CC) -c -I. $(CFLAGS) pbdemo.c
 
 pbutil.o: pbutil.c pbutil.h
-	$(CC) -c -I. -I$(INCDIR) $(CFLAGS) pbutil.c
+	$(CC) -c -I. $(CFLAGS) pbutil.c
 
 glxgears_fbconfig: glxgears_fbconfig.o pbutil.o
 	$(CC) $(CFLAGS) glxgears_fbconfig.o pbutil.o $(APP_LIB_DEPS) -o $@
 
 glxgears_fbconfig.o: glxgears_fbconfig.c pbutil.h
-	$(CC) $(CFLAGS) -c -I. -I$(INCDIR) $(CFLAGS) glxgears_fbconfig.c
+	$(CC) $(CFLAGS) -c -I. $(CFLAGS) glxgears_fbconfig.c
 
 xrotfontdemo: xrotfontdemo.o xuserotfont.o
 	$(CC) $(CFLAGS) xrotfontdemo.o xuserotfont.o $(APP_LIB_DEPS) -o $@
 
 xuserotfont.o: xuserotfont.c xuserotfont.h
-	$(CC) -c -I. -I$(INCDIR) $(CFLAGS) xuserotfont.c
+	$(CC) -c -I. $(CFLAGS) xuserotfont.c
 
 xrotfontdemo.o: xrotfontdemo.c xuserotfont.h
-	$(CC) -c -I. -I$(INCDIR) $(CFLAGS) xrotfontdemo.c
-
+	$(CC) -c -I. $(CFLAGS) xrotfontdemo.c
