diff -aur motion-3.2.9/configure.in fw_motion-3.2.9/configure.in
--- motion-3.2.9/configure.in	2007-11-06 04:19:30.000000000 +0100
+++ fw_motion-3.2.9/configure.in	2008-04-29 10:52:40.000000000 +0200
@@ -383,16 +383,16 @@
 	if test "${FFMPEG_CFLAGS_DEB}" != "" ; then
 		FFMPEG_CFLAGS="${FFMPEG_CFLAGS_DEB}"
 		AC_MSG_RESULT(found for debian)
-	elif test -f ${FFMPEG}/include/avformat.h; then
+	elif test -f ${FFMPEG}/include/libavformat/avformat.h; then
 		AC_MSG_RESULT(found)
 		FFMPEG_CFLAGS="-I${FFMPEG}/include"
-	elif test -f ${FFMPEG}/avformat.h; then
+	elif test -f ${FFMPEG}/libavformat/avformat.h; then
 		AC_MSG_RESULT(found)
 		FFMPEG_CFLAGS="-I${FFMPEG}"
-	elif test -f `AS_DIRNAME([${FFMPEG}])`/include/avformat.h; then
+	elif test -f `AS_DIRNAME([${FFMPEG}])`/include/libavformat/avformat.h; then
 		AC_MSG_RESULT(found)
 		FFMPEG_CFLAGS="-I`AS_DIRNAME([${FFMPEG}])`/include"
-	elif test -f `AS_DIRNAME([${FFMPEG}])`/include/ffmpeg/avformat.h; then
+	elif test -f `AS_DIRNAME([${FFMPEG}])`/include/libavformat/avformat.h; then
 		AC_MSG_RESULT(found)
 		FFMPEG_CFLAGS="-I`AS_DIRNAME([${FFMPEG}])`/include/ffmpeg"
 	else
@@ -432,7 +432,7 @@
 
                 AC_COMPILE_IFELSE(
                         [
-                        #include <avformat.h>
+                        #include <libavformat/avformat.h>
                         URLProtocol test_file_protocol;
                         int main(void){
                                 test_file_protocol.url_read  = file_protocol.url_read;
diff -aur motion-3.2.9/ffmpeg.c fw_motion-3.2.9/ffmpeg.c
--- motion-3.2.9/ffmpeg.c	2007-11-06 04:50:07.000000000 +0100
+++ fw_motion-3.2.9/ffmpeg.c	2008-04-29 10:56:35.000000000 +0200
@@ -121,7 +121,7 @@
  * 
  */
 
-#include "avstring.h"
+#include "libavutil/avstring.h"
 
 static int file_open(URLContext *h, const char *filename, int flags)
 {
diff -aur motion-3.2.9/ffmpeg.h fw_motion-3.2.9/ffmpeg.h
--- motion-3.2.9/ffmpeg.h	2007-07-18 09:18:05.000000000 +0200
+++ fw_motion-3.2.9/ffmpeg.h	2008-04-29 10:54:13.000000000 +0200
@@ -2,7 +2,7 @@
 #define _INCLUDE_FFMPEG_H_
 
 #ifdef HAVE_FFMPEG
-#include <avformat.h>
+#include <libavformat/avformat.h>
 #endif
 
 #include <stdio.h>
