diff -Naur audacity-src-1.3.0b-beta/lib-src/soundtouch/include/SoundTouch.h audacity-src-1.3.0b-beta.gcc4/lib-src/soundtouch/include/SoundTouch.h
--- audacity-src-1.3.0b-beta/lib-src/soundtouch/include/SoundTouch.h	2005-11-27 17:47:29.000000000 +0100
+++ audacity-src-1.3.0b-beta.gcc41/lib-src/soundtouch/include/SoundTouch.h	2006-05-02 13:19:35.000000000 +0200
@@ -146,7 +146,7 @@
     static const char *getVersionString();
 
     /// Get SoundTouch library version Id
-    static uint SoundTouch::getVersionId();
+    static uint getVersionId();
 
     /// Sets new rate control value. Normal rate = 1.0, smaller values
     /// represent slower rate, larger faster rates.
