diff -Naur easytag-1.99.3.orig/easytag.desktop easytag-1.99.3/easytag.desktop
--- easytag-1.99.3.orig/easytag.desktop	2005-01-11 21:27:26.000000000 +0100
+++ easytag-1.99.3/easytag.desktop	2005-01-22 10:31:19.156892632 +0100
@@ -6,7 +6,7 @@
 Comment[de]=EasyTAG ist ein Werkzeug zum Anschauen, Editieren und Schreiben von Tags Ihrer MP3-, MP2-, FLAC-, Ogg Vorbis-, MusePack- und  Monkey's Audio-Dateien.
 Comment[es]=EasyTAG es una utilidad para ver, editar y escribir etiquetas a tus archivos MP3, MP2, FLAC y Ogg Vorbis.
 Comment[fr]=EasyTAG est un utilitaire pour visualiser et éditer les tags des fichiers MP3, MP2, FLAC, Ogg Vorbis, MusePack et Monkey's Audio avec une interface GTK+.
-Comment[hu]=Az EasyTag egy GTK+-on alapuló ID3 tag nézö/szerkesztö, valamint fájl átnevezö eszköz MP3, MP2, FLAC és Ogg Vorbis fájlokhoz.
+Comment[hu]=Az EasyTAG egy GTK+-on alapuló tagnéző és -szerkesztő segédeszköz MP3, MP2, FLAC, Ogg Vorbis, MusePack és Monkey's Audio fájlokhoz.
 Comment[it]=EasyTAG è programma per vedere, modificare e scrivere i tag dei file MP3, MP2, FLAC, Ogg Vorbis, MusePack e Monkey's Audio.
 Comment[ja]=EasyTAG は MP3, MP2, FLAC, Ogg Vorbis, MusePack と Monkey's オーディオ・ファイルの ID3 タグを表示したり、編集するためのツールです。シンプルでクールな GTK+ インタフェースを採用しているので、GNU/Linux で簡単にタグの編集ができます。
 Comment[nl]=EasyTAG is een programma voor het bewerken van tags van MP2-, MP3-, FLAC-, Ogg Vorbis, MusePack en Monkeys Audio-bestanden. Dit gebeurt in een verzorgde GTK+-omgeving.
