diff -Naur Mesa-6.5.1.orig/src/glx/x11/dri_glx.c Mesa-6.5.1/src/glx/x11/dri_glx.c
--- Mesa-6.5.1.orig/src/glx/x11/dri_glx.c	2006-09-16 11:50:59.000000000 +0200
+++ Mesa-6.5.1/src/glx/x11/dri_glx.c	2006-09-16 11:51:29.000000000 +0200
@@ -58,7 +58,7 @@
 
 #ifndef DEFAULT_DRIVER_DIR
 /* this is normally defined in Mesa/configs/default with DRI_DRIVER_SEARCH_PATH */
-#define DEFAULT_DRIVER_DIR "/usr/X11R6/lib/modules/dri"
+#define DEFAULT_DRIVER_DIR "/usr/lib/dri"
 #endif
 
 static __DRIdriver *Drivers = NULL;
