diff -Naur xorg-server-1.20.5/configure.ac xorg-server-1.20.5.new/configure.ac
--- xorg-server-1.20.5/configure.ac	2019-05-30 20:27:34.000000000 +0200
+++ xorg-server-1.20.5.new/configure.ac	2019-10-25 12:24:40.328882752 +0200
@@ -765,7 +765,7 @@
 LIBDRM="libdrm >= 2.4.89"
 LIBEGL="egl"
 LIBGBM="gbm >= 10.2.0"
-LIBGL="gl >= 7.1.0"
+LIBGL="gl >= 1.2"
 LIBXEXT="xext >= 1.0.99.4"
 LIBXFONT="xfont2 >= 2.0.0"
 LIBXI="xi >= 1.2.99.1"
@@ -1116,7 +1116,7 @@
 	yes,yes | auto,yes)
 		AC_DEFINE(DRI2, 1, [Build DRI2 extension])
 		DRI2=yes
-		LIBGL="gl >= 9.2.0"
+		LIBGL="gl >= 1.2"
 		SDK_REQUIRED_MODULES="$SDK_REQUIRED_MODULES $DRI2PROTO"
 		;;
 esac
diff -Naur xorg-server-1.20.5/glamor/glamor_egl.h xorg-server-1.20.5.new/glamor/glamor_egl.h
--- xorg-server-1.20.5/glamor/glamor_egl.h	2019-05-30 20:27:34.000000000 +0200
+++ xorg-server-1.20.5.new/glamor/glamor_egl.h	2019-10-25 12:24:40.329882759 +0200
@@ -27,7 +27,7 @@
 #ifndef GLAMOR_EGL_H
 #define GLAMOR_EGL_H
 
-#define MESA_EGL_NO_X11_HEADERS
+#define EGL_NO_X11
 #include <epoxy/gl.h>
 #include <epoxy/egl.h>
 #include <glamor_egl_ext.h>
diff -Naur xorg-server-1.20.5/glamor/glamor_priv.h xorg-server-1.20.5.new/glamor/glamor_priv.h
--- xorg-server-1.20.5/glamor/glamor_priv.h	2019-05-30 20:27:34.000000000 +0200
+++ xorg-server-1.20.5.new/glamor/glamor_priv.h	2019-10-25 12:24:40.329882759 +0200
@@ -39,7 +39,7 @@
 
 #include <epoxy/gl.h>
 #ifdef GLAMOR_HAS_GBM
-#define MESA_EGL_NO_X11_HEADERS
+#define EGL_NO_X11
 #include <epoxy/egl.h>
 #endif
 
diff -Naur xorg-server-1.20.5/glx/meson.build xorg-server-1.20.5.new/glx/meson.build
--- xorg-server-1.20.5/glx/meson.build	2019-05-30 20:27:34.000000000 +0200
+++ xorg-server-1.20.5.new/glx/meson.build	2019-10-25 12:24:40.329882759 +0200
@@ -39,7 +39,7 @@
             common_dep,
             dl_dep,
             dependency('glproto', version: '>= 1.4.17'),
-            dependency('gl', version: '>= 9.2.0'),
+            dependency('gl', version: '>= 1.2'),
         ],
         c_args: [
             glx_align64,
@@ -74,7 +74,7 @@
             common_dep,
             dl_dep,
             dependency('glproto', version: '>= 1.4.17'),
-            dependency('gl', version: '>= 9.2.0'),
+            dependency('gl', version: '>= 1.2'),
         ],
     )
 
diff -Naur xorg-server-1.20.5/hw/xwayland/xwayland-glamor.c xorg-server-1.20.5.new/hw/xwayland/xwayland-glamor.c
--- xorg-server-1.20.5/hw/xwayland/xwayland-glamor.c	2019-05-30 20:27:34.000000000 +0200
+++ xorg-server-1.20.5.new/hw/xwayland/xwayland-glamor.c	2019-10-25 12:24:40.330882766 +0200
@@ -25,7 +25,7 @@
 
 #include "xwayland.h"
 
-#define MESA_EGL_NO_X11_HEADERS
+#define EGL_NO_X11
 #include <glamor_egl.h>
 
 #include <glamor.h>
diff -Naur xorg-server-1.20.5/hw/xwayland/xwayland-glamor-eglstream.c xorg-server-1.20.5.new/hw/xwayland/xwayland-glamor-eglstream.c
--- xorg-server-1.20.5/hw/xwayland/xwayland-glamor-eglstream.c	2019-05-30 20:27:34.000000000 +0200
+++ xorg-server-1.20.5.new/hw/xwayland/xwayland-glamor-eglstream.c	2019-10-25 12:24:40.330882766 +0200
@@ -32,7 +32,7 @@
 #include "wayland-eglstream-client-protocol.h"
 #include "wayland-eglstream-controller-client-protocol.h"
 
-#define MESA_EGL_NO_X11_HEADERS
+#define EGL_NO_X11
 #include <glamor_egl.h>
 #include <glamor.h>
 #include <glamor_transform.h>
diff -Naur xorg-server-1.20.5/hw/xwayland/xwayland-glamor-gbm.c xorg-server-1.20.5.new/hw/xwayland/xwayland-glamor-gbm.c
--- xorg-server-1.20.5/hw/xwayland/xwayland-glamor-gbm.c	2019-05-30 20:27:34.000000000 +0200
+++ xorg-server-1.20.5.new/hw/xwayland/xwayland-glamor-gbm.c	2019-10-25 12:24:40.330882766 +0200
@@ -35,7 +35,7 @@
 #include <xf86drm.h>
 #include <drm_fourcc.h>
 
-#define MESA_EGL_NO_X11_HEADERS
+#define EGL_NO_X11
 #include <gbm.h>
 #include <glamor_egl.h>
 
