diff -Nur GL.orig/glx/glxext.c GL/glx/glxext.c
--- GL.orig/glx/glxext.c	2006-12-01 03:40:10.000000000 +0200
+++ GL/glx/glxext.c	2007-04-23 01:23:39.900111785 +0300
@@ -275,7 +275,7 @@
 /*
 ** The GL was will call this routine if an error occurs.
 */
-void __glXErrorCallBack(__GLinterface *gc, GLenum code)
+void __glXErrorCallBack(GLenum code)
 {
     errorOccured = GL_TRUE;
 }
diff -Nur GL.orig/glx/glxext.h GL/glx/glxext.h
--- GL.orig/glx/glxext.h	2006-09-18 09:04:17.000000000 +0300
+++ GL/glx/glxext.h	2007-04-23 01:23:39.900111785 +0300
@@ -66,7 +66,7 @@
 extern GLboolean __glXFreeContext(__GLXcontext *glxc);
 extern void __glXFlushContextCache(void);
 
-extern void __glXErrorCallBack(__GLinterface *gc, GLenum code);
+extern void __glXErrorCallBack(GLenum code);
 extern void __glXClearErrorOccured(void);
 extern GLboolean __glXErrorOccured(void);
 extern void __glXResetLargeCommandStatus(__GLXclientState*);
diff -Nur hw.orig/dmx/glxProxy/glxext.h hw/dmx/glxProxy/glxext.h
--- hw.orig/dmx/glxProxy/glxext.h	2006-09-18 09:04:17.000000000 +0300
+++ hw/dmx/glxProxy/glxext.h	2006-09-18 09:04:17.000000000 +0300
@@ -67,7 +67,7 @@
 
 extern void __glXNoSuchRenderOpcode(GLbyte*);
 extern int __glXNoSuchSingleOpcode(__GLXclientState*, GLbyte*);
-extern void __glXErrorCallBack(__GLinterface *gc, GLenum code);
+extern void __glXErrorCallBack(GLenum code);
 extern void __glXClearErrorOccured(void);
 extern GLboolean __glXErrorOccured(void);
 extern void __glXResetLargeCommandStatus(__GLXclientState*);
