diff -Naur xorg-server-1.1.1.orig/GL/mesa/main/Makefile.am xorg-server-1.1.1/GL/mesa/main/Makefile.am
--- xorg-server-1.1.1.orig/GL/mesa/main/Makefile.am	2006-09-25 10:49:54.000000000 +0200
+++ xorg-server-1.1.1/GL/mesa/main/Makefile.am	2006-09-25 10:50:44.000000000 +0200
@@ -29,6 +29,7 @@
                     api_loopback.c \
                     api_noop.c \
                     api_validate.c \
+                    arrayobj.c \
                     attrib.c \
                     blend.c \
                     bufferobj.c \
@@ -67,6 +68,7 @@
                     points.c \
                     polygon.c \
                     rastpos.c \
+                    rbadaptors.c \
                     renderbuffer.c \
                     state.c \
                     stencil.c \
diff -Naur xorg-server-1.1.1.orig/GL/mesa/main/Makefile.in xorg-server-1.1.1/GL/mesa/main/Makefile.in
--- xorg-server-1.1.1.orig/GL/mesa/main/Makefile.in	2006-09-25 10:49:54.000000000 +0200
+++ xorg-server-1.1.1/GL/mesa/main/Makefile.in	2006-09-25 10:52:17.000000000 +0200
@@ -57,14 +57,14 @@
 libmain_la_LIBADD =
 @NEED_VSNPRINTF_TRUE@am__objects_1 = vsnprintf.lo
 nodist_libmain_la_OBJECTS = accum.lo api_arrayelt.lo api_loopback.lo \
-	api_noop.lo api_validate.lo attrib.lo blend.lo bufferobj.lo \
+	api_noop.lo api_validate.lo arrayobj.lo attrib.lo blend.lo bufferobj.lo \
 	buffers.lo clip.lo colortab.lo context.lo convolve.lo debug.lo \
 	depth.lo depthstencil.lo dlist.lo drawpix.lo enable.lo \
 	enums.lo eval.lo execmem.lo extensions.lo fbobject.lo \
 	feedback.lo fog.lo framebuffer.lo get.lo getstring.lo hash.lo \
 	hint.lo histogram.lo image.lo imports.lo light.lo lines.lo \
 	matrix.lo mm.lo occlude.lo pixel.lo points.lo polygon.lo \
-	rastpos.lo renderbuffer.lo state.lo stencil.lo texcompress.lo \
+	rastpos.lo rbadaptors.c renderbuffer.lo state.lo stencil.lo texcompress.lo \
 	texcompress_fxt1.lo texcompress_s3tc.lo texenvprogram.lo \
 	texformat.lo teximage.lo texobj.lo texrender.lo texstate.lo \
 	texstore.lo varray.lo $(am__objects_1) vtxfmt.lo
@@ -563,6 +563,7 @@
                     api_loopback.c \
                     api_noop.c \
                     api_validate.c \
+				arrayobj.c \
                     attrib.c \
                     blend.c \
                     bufferobj.c \
@@ -601,6 +602,7 @@
                     points.c \
                     polygon.c \
                     rastpos.c \
+				rbadaptors.c \
                     renderbuffer.c \
                     state.c \
                     stencil.c \
@@ -674,6 +676,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/api_loopback.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/api_noop.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/api_validate.Plo@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/arrayobj.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/attrib.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/blend.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/bufferobj.Plo@am__quote@
@@ -712,6 +715,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/points.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/polygon.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/rastpos.Plo@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/rbadaptors.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/renderbuffer.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/state.Plo@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/stencil.Plo@am__quote@
diff -Naur xorg-server-1.1.1.orig/GL/symlink-mesa.sh xorg-server-1.1.1/GL/symlink-mesa.sh
--- xorg-server-1.1.1.orig/GL/symlink-mesa.sh	2006-09-25 10:49:54.000000000 +0200
+++ xorg-server-1.1.1/GL/symlink-mesa.sh	2006-09-25 10:50:44.000000000 +0200
@@ -88,8 +88,11 @@
     action api_noop.h
     action api_validate.c
     action api_validate.h
+    action arrayobj.c
+    action arrayobj.h
     action attrib.c
     action attrib.h
+    action bitset.h
     action blend.c
     action blend.h
     action bufferobj.c
@@ -168,6 +171,8 @@
     action polygon.h
     action rastpos.c
     action rastpos.h
+    action rbadaptors.c
+    action rbadaptors.h
     action renderbuffer.c
     action renderbuffer.h
     action simple_list.h
@@ -463,7 +468,7 @@
     action slang_core_gc.h
     action slang_fragment_builtin_gc.h
     action slang_shader_syn.h
-    action slang_version_syn.h
+    action slang_pp_version_syn.h
     action slang_vertex_builtin_gc.h
 }        
 
