diff -Naur xorg-server-1.1.99.901.orig/hw/xprint/etc/init.d/xprint.cpp xorg-server-1.1.99.901/hw/xprint/etc/init.d/xprint.cpp
--- xorg-server-1.1.99.901.orig/hw/xprint/etc/init.d/xprint.cpp	2006-10-18 18:48:22.000000000 +0200
+++ xorg-server-1.1.99.901/hw/xprint/etc/init.d/xprint.cpp	2006-10-18 18:50:28.000000000 +0200
@@ -42,21 +42,21 @@
 XCOMM
 XCOMM Usage:
 XCOMM - Start Xprint server(s) manually:
-XCOMM   % /etc/init.d/xprint start
+XCOMM   % /etc/rc.d/rc.xprint start
 XCOMM
 XCOMM - Stop Xprint server(s) manually:
-XCOMM   % /etc/init.d/xprint stop
+XCOMM   % /etc/rc.d/rc.xprint stop
 XCOMM
 XCOMM - Populate $XPSERVERLIST env var (for example as part of a global
 XCOMM   login script like /etc/profile or per-user login scripts like
 XCOMM   $HOME/.profile (sh/ksh/bash))
-XCOMM   % XPSERVERLIST="`/etc/init.d/xprint get_xpserverlist`"
+XCOMM   % XPSERVERLIST="`/etc/rc.d/rc.xprint get_xpserverlist`"
 XCOMM   % export XPSERVERLIST
 XCOMM
 XCOMM Installation:
-XCOMM   Copy this script to /etc/init.d/xprint and make sure that it is
+XCOMM   Copy this script to /etc/rc.d/rc.xprint and make sure that it is
 XCOMM   executable. If your installation is LSB-compliant, then run
-XCOMM   % /usr/lib/lsb/install_initd /etc/init.d/xprint
+XCOMM   % /usr/lib/lsb/install_initd /etc/rc.d/rc.xprint
 XCOMM   to start the service on startup. Otherwise, manually create links
 XCOMM   to the matching run levels.
 XCOMM   Examples:
@@ -683,7 +683,7 @@
                   echo "Xprint server pid=${server_pid} done, exitcode=$?."
                   
                   # Remove the dead server from the registry
-                  # (only if the registry still exists - if /etc/init.d/xprint stop" gets called the registry
+                  # (only if the registry still exists - if /etc/rc.d/rc.xprint stop" gets called the registry
                   # will be removed - and we should not re-create it afterwards...)
                   lock_server_registry
                   if [ -f "${XP_SERVERS}" ] ; then
@@ -748,7 +748,7 @@
     else 
         # Continue in the background if there is no spooler running yet (that
         # we don't hold off the boot process nor run in a possible race-condition
-        # when /etc/init.d/lpd was not called yet but the runlevel script waits
+        # when /etc/rc.d/rc.lpd was not called yet but the runlevel script waits
         # for us to finish first ...
         if is_spooler_running ; then
             msg "Starting Xprint servers: Xprt."
@@ -1098,7 +1098,7 @@
         # done with setting the config for per-user Xprt instances
         return 0;
     else
-        # Use /etc/init.d/xprint's builtin config
+        # Use /etc/rc.d/rc.xprint's builtin config
         # Each entry should start with |setup_config_defaults| to pull the
         # platform defaults and finish with |num_xpstart=$(($num_xpstart + 1))|
         # to end the entry
diff -Naur xorg-server-1.1.99.901.orig/hw/xprint/etc/profile.d/xprint.csh xorg-server-1.1.99.901/hw/xprint/etc/profile.d/xprint.csh
--- xorg-server-1.1.99.901.orig/hw/xprint/etc/profile.d/xprint.csh	2006-10-18 18:48:22.000000000 +0200
+++ xorg-server-1.1.99.901/hw/xprint/etc/profile.d/xprint.csh	2006-10-18 18:48:49.000000000 +0200
@@ -9,8 +9,8 @@
 # Obtain list of Xprint servers
 #
 
-if ( -f /etc/init.d/xprint ) then
-  setenv XPSERVERLIST "`/bin/sh /etc/init.d/xprint get_xpserverlist`"
+if ( -f /etc/rc.d/rc.xprint ) then
+  setenv XPSERVERLIST "`/bin/sh /etc/rc.d/xprint get_xpserverlist`"
 endif
 
 # /etc/profile.d/xprint.csh ends here.
diff -Naur xorg-server-1.1.99.901.orig/hw/xprint/etc/profile.d/xprint.sh xorg-server-1.1.99.901/hw/xprint/etc/profile.d/xprint.sh
--- xorg-server-1.1.99.901.orig/hw/xprint/etc/profile.d/xprint.sh	2006-10-18 18:48:22.000000000 +0200
+++ xorg-server-1.1.99.901/hw/xprint/etc/profile.d/xprint.sh	2006-10-18 18:49:09.000000000 +0200
@@ -8,8 +8,8 @@
 # Obtain list of Xprint servers
 #
 
-if [ -f "/etc/init.d/xprint" ] ; then
-  XPSERVERLIST="`/bin/sh /etc/init.d/xprint get_xpserverlist`"
+if [ -f "/etc/rc.d/rc.xprint" ] ; then
+  XPSERVERLIST="`/bin/sh /etc/rc.d/rc.xprint get_xpserverlist`"
   export XPSERVERLIST
 fi
 
diff -Naur xorg-server-1.1.99.901.orig/hw/xprint/etc/Xsession.d/92xprint-xpserverlist xorg-server-1.1.99.901/hw/xprint/etc/Xsession.d/92xprint-xpserverlist
--- xorg-server-1.1.99.901.orig/hw/xprint/etc/Xsession.d/92xprint-xpserverlist	2006-10-18 18:48:22.000000000 +0200
+++ xorg-server-1.1.99.901/hw/xprint/etc/Xsession.d/92xprint-xpserverlist	2006-10-18 18:50:47.000000000 +0200
@@ -22,8 +22,8 @@
 # Obtain list of Xprint servers
 #
 
-if [ -x "/etc/init.d/xprint" ] ; then
-  XPSERVERLIST="`/etc/init.d/xprint get_xpserverlist`"
+if [ -x "/etc/rc.d/rc.xprint" ] ; then
+  XPSERVERLIST="`/etc/rc.d/rc.xprint get_xpserverlist`"
   export XPSERVERLIST
 fi
 
diff -Naur xorg-server-1.1.99.901.orig/hw/xprint/etc/Xsession.d/Makefile.am xorg-server-1.1.99.901/hw/xprint/etc/Xsession.d/Makefile.am
--- xorg-server-1.1.99.901.orig/hw/xprint/etc/Xsession.d/Makefile.am	2006-10-18 18:48:22.000000000 +0200
+++ xorg-server-1.1.99.901/hw/xprint/etc/Xsession.d/Makefile.am	2006-10-18 18:50:56.000000000 +0200
@@ -1,2 +1,2 @@
-xpcdir = $(sysconfdir)/X11/Xsession.d
+xpcdir = $(sysconfdir)/Xsession.d
 dist_xpc_DATA = 92xprint-xpserverlist
diff -Naur xorg-server-1.1.99.901.orig/hw/xprint/etc/Xsession.d/Makefile.in xorg-server-1.1.99.901/hw/xprint/etc/Xsession.d/Makefile.in
--- xorg-server-1.1.99.901.orig/hw/xprint/etc/Xsession.d/Makefile.in	2006-10-18 18:48:22.000000000 +0200
+++ xorg-server-1.1.99.901/hw/xprint/etc/Xsession.d/Makefile.in	2006-10-18 18:51:30.000000000 +0200
@@ -518,7 +518,7 @@
 target_alias = @target_alias@
 xglmoduledir = @xglmoduledir@
 xpconfigdir = @xpconfigdir@
-xpcdir = $(sysconfdir)/X11/Xsession.d
+xpcdir = $(sysconfdir)/Xsession.d
 dist_xpc_DATA = 92xprint-xpserverlist
 all: all-am
 
