diff -Naur geoclue-2.4.7/Makefile.am geoclue-2.4.7-p/Makefile.am
--- geoclue-2.4.7/Makefile.am	2016-09-02 11:22:25.000000000 +0200
+++ geoclue-2.4.7-p/Makefile.am	2018-02-17 00:22:16.877241295 +0100
@@ -13,7 +13,14 @@
 BACKEND_DIR =
 endif
 
-SUBDIRS = build-aux public-api $(BACKEND_DIR) data $(LIBGEOCLUE_DIR) demo po docs
+if BUILD_DEMO_AGENT
+DEMO_DIR = demo
+else
+DEMO_DIR =
+endif
+
+
+SUBDIRS = build-aux public-api $(BACKEND_DIR) data $(LIBGEOCLUE_DIR) $(DEMO_DIR) po docs
 
 INTLTOOL_FILES =				\
 	intltool-extract.in			\
