--- openssh-9.1p1/contrib/sshd.pam.generic	2022-10-03 16:51:42.000000000 +0200
+++ openssh-9.1p1.new/contrib/sshd.pam.generic	2022-12-26 21:58:47.750259873 +0100
@@ -2,7 +2,6 @@
 auth       required     /lib/security/pam_unix.so shadow nodelay
 account    required     /lib/security/pam_nologin.so
 account    required     /lib/security/pam_unix.so
-password   required     /lib/security/pam_cracklib.so
 password   required     /lib/security/pam_unix.so shadow nullok use_authtok
 session    required     /lib/security/pam_unix.so
 session    required     /lib/security/pam_limits.so
