--- v4l-utils-1.26.1/utils/v4l2-dbg/meson.build	2023-12-12 22:40:38.000000000 +0100
+++ v4l-utils-1.26.1.new/utils/v4l2-dbg/meson.build	2024-01-05 12:13:32.426113287 +0100
@@ -13,7 +13,7 @@
 v4l2_dbg = executable('v4l2-dbg',
                       v4l2_dbg_sources,
                       install : true,
-                      install_dir : 'sbin',
+                      install_dir : 'bin',
                       include_directories : [
                           v4l2_utils_incdir,
                           utils_common_incdir,
