Index: gtkpod-2.1.0/plugins/cover_display/display_coverart.c
===================================================================
--- gtkpod-2.1.0.orig/plugins/cover_display/display_coverart.c
+++ gtkpod-2.1.0/plugins/cover_display/display_coverart.c
@@ -174,7 +174,7 @@ void coverart_init_display(GtkWidget *pa
     cdwidget->controlbox = gtkpod_builder_xml_get_widget(xml, "cover_display_controlbox");
     cdwidget->leftbutton = GTK_BUTTON (gtkpod_builder_xml_get_widget (xml, "cover_display_leftbutton"));
     cdwidget->rightbutton = GTK_BUTTON (gtkpod_builder_xml_get_widget (xml, "cover_display_rightbutton"));
-    cdwidget->cdslider = GTK_HSCALE (gtkpod_builder_xml_get_widget (xml, "cover_display_scaler"));
+    cdwidget->cdslider = GTK_SCALE (gtkpod_builder_xml_get_widget (xml, "cover_display_scaler"));
     /* create a new drawing area */
     cdwidget->draw_area = gtk_drawing_area_new();
     cdwidget->cdcovers = g_ptr_array_sized_new(IMG_TOTAL);
Index: gtkpod-2.1.0/plugins/cover_display/display_coverart.h
===================================================================
--- gtkpod-2.1.0.orig/plugins/cover_display/display_coverart.h
+++ gtkpod-2.1.0/plugins/cover_display/display_coverart.h
@@ -68,7 +68,7 @@ typedef struct {
 	GtkWidget *canvasbox;
 	GtkWidget *controlbox;
 	GtkButton *leftbutton;
-	GtkHScale *cdslider;
+	GtkScale  *cdslider;
 	GtkButton *rightbutton;
 
 	/* Utility data */
Index: gtkpod-2.1.0/plugins/sorttab_display/display_sorttabs.c
===================================================================
--- gtkpod-2.1.0.orig/plugins/sorttab_display/display_sorttabs.c
+++ gtkpod-2.1.0/plugins/sorttab_display/display_sorttabs.c
@@ -2300,7 +2300,7 @@ static void st_create_paned() {
     for (i = 0; i < PANED_NUM_ST; ++i) {
         GtkWidget *paned;
 
-        paned = gtk_hpaned_new();
+        paned = gtk_paned_new(GTK_ORIENTATION_HORIZONTAL);
         gtk_widget_show(paned);
 
         if (!i) {
Index: gtkpod-2.1.0/plugins/sorttab_display/plugin.c
===================================================================
--- gtkpod-2.1.0.orig/plugins/sorttab_display/plugin.c
+++ gtkpod-2.1.0/plugins/sorttab_display/plugin.c
@@ -176,7 +176,7 @@ static gboolean activate_sorttab_display
     set_default_preferences();
 
     /* Add widget in Shell. Any number of widgets can be added */
-    sorttab_display_plugin->st_paned = gtk_hpaned_new();
+    sorttab_display_plugin->st_paned = gtk_paned_new(GTK_ORIENTATION_HORIZONTAL);
     gchar *glade_path = g_build_filename(get_glade_dir(), "sorttab_display.xml", NULL);
     st_create_tabs(GTK_PANED(sorttab_display_plugin->st_paned), glade_path);
     gtk_widget_show(sorttab_display_plugin->st_paned);
