diff -Naur lua-5.1.4/etc/lua.pc lua-5.1.4-2/etc/lua.pc
--- lua-5.1.4/etc/lua.pc	2008-08-08 07:46:11.000000000 -0500
+++ lua-5.1.4-2/etc/lua.pc	2010-10-05 02:27:30.379001571 -0500
@@ -8,7 +8,7 @@
 R= 5.1.4
 
 # grep '^INSTALL_.*=' ../Makefile | sed 's/INSTALL_TOP/prefix/'
-prefix= /usr/local
+prefix= /usr
 INSTALL_BIN= ${prefix}/bin
 INSTALL_INC= ${prefix}/include
 INSTALL_LIB= ${prefix}/lib
diff -Naur lua-5.1.4/Makefile lua-5.1.4-2/Makefile
--- lua-5.1.4/Makefile	2008-08-11 19:40:48.000000000 -0500
+++ lua-5.1.4-2/Makefile	2010-10-05 02:27:21.284001613 -0500
@@ -9,7 +9,7 @@
 
 # Where to install. The installation starts in the src and doc directories,
 # so take care if INSTALL_TOP is not an absolute path.
-INSTALL_TOP= /usr/local
+INSTALL_TOP= /usr
 INSTALL_BIN= $(INSTALL_TOP)/bin
 INSTALL_INC= $(INSTALL_TOP)/include
 INSTALL_LIB= $(INSTALL_TOP)/lib
diff -Naur lua-5.1.4/src/luaconf.h lua-5.1.4-2/src/luaconf.h
--- lua-5.1.4/src/luaconf.h	2008-02-11 10:25:08.000000000 -0600
+++ lua-5.1.4-2/src/luaconf.h	2010-10-05 02:27:44.490001525 -0500
@@ -94,7 +94,7 @@
 	".\\?.dll;"  LUA_CDIR"?.dll;" LUA_CDIR"loadall.dll"
 
 #else
-#define LUA_ROOT	"/usr/local/"
+#define LUA_ROOT	"/usr/"
 #define LUA_LDIR	LUA_ROOT "share/lua/5.1/"
 #define LUA_CDIR	LUA_ROOT "lib/lua/5.1/"
 #define LUA_PATH_DEFAULT  \
