--- Linux/Kernel/Dm-target.c	2007-04-24 19:32:06.000000000 +0300
+++ Linux/Kernel/Dm-target.c	2007-10-10 23:18:24.000000000 +0200
@@ -656,7 +656,11 @@ int __init dm_truecrypt_init(void)
 		goto err;
 	}
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,23)
 	bio_ctx_cache = kmem_cache_create ("truecrypt-bioctx", sizeof (struct bio_ctx), 0, 0, NULL, NULL);
+#else
+	bio_ctx_cache = kmem_cache_create ("truecrypt-bioctx", sizeof (struct bio_ctx), 0, 0, NULL);
+#endif
 	if (!bio_ctx_cache)
 	{
 		error ("kmem_cache_create failed");
