--- Linux/Kernel/Dm-target.c.orig	2006-10-23 00:07:22.000000000 +0200
+++ Linux/Kernel/Dm-target.c	2006-12-14 18:01:24.557572653 +0100
@@ -379,16 +379,16 @@
 }
 
 
-static void work_process (void *qdata)
+static void work_process (struct work_struct *work) // void *qdata
 {
-	struct bio_ctx *bc = (struct bio_ctx *) qdata;
+	struct bio_ctx *bc = container_of(work, struct bio_ctx, work); // qdata;
 	struct target_ctx *tc = (struct target_ctx *) bc->target->private;
 	struct bio_vec *bv;
 	u64 sec_no = bc->crypto_sector;
 	int seg_no;
 	unsigned long flags;
 
-	trace (3, "work_process (%p)\n", qdata);
+	trace (3, "work_process (%p)\n", bc);
 
 	// Decrypt queued data
 	bio_for_each_segment (bv, bc->orig_bio, seg_no)
@@ -437,7 +437,7 @@
 		bio_put (bio);
 
 		// Queue decryption to leave completion interrupt ASAP
-		INIT_WORK (&bc->work, work_process, bc);
+		INIT_WORK (&bc->work, work_process); // bc
 		trace (3, "queue_work (%p)\n", work_queue);
 		queue_work (work_queue, &bc->work);
 		return error;
