diff -Naur truecrypt-4.3a-source-code.orig/Linux/Kernel/Dm-target.c truecrypt-4.3a-source-code/Linux/Kernel/Dm-target.c
--- truecrypt-4.3a-source-code.orig/Linux/Kernel/Dm-target.c	2007-04-24 18:32:06.000000000 +0200
+++ truecrypt-4.3a-source-code/Linux/Kernel/Dm-target.c	2008-01-28 00:51:49.000000000 +0100
@@ -375,7 +375,7 @@
 	if (!atomic_dec_and_test (&bc->ref_count))
 		return;
 
-	bio_endio (bc->orig_bio, bc->orig_bio->bi_size, bc->error);
+	bio_endio (bc->orig_bio, bc->error);
 	mempool_free (bc, tc->bio_ctx_pool);
 }
 
@@ -656,7 +656,11 @@
 		goto err;
 	}
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(2,6,23)
 	bio_ctx_cache = kmem_cache_create ("truecrypt-bioctx", sizeof (struct bio_ctx), 0, 0, NULL, NULL);
+#else
+	bio_ctx_cache = kmem_cache_create ("truecrypt-bioctx", sizeof (struct bio_ctx), 0, 0, NULL);
+#endif
 	if (!bio_ctx_cache)
 	{
 		error ("kmem_cache_create failed");
