diff -Naur truecrypt-7.1a-source/Main/Application.cpp truecrypt-7.1a-source.new/Main/Application.cpp
--- truecrypt-7.1a-source/Main/Application.cpp	2012-02-07 11:36:48.000000000 +0100
+++ truecrypt-7.1a-source.new/Main/Application.cpp	2014-02-11 10:31:16.621290725 +0100
@@ -36,7 +36,6 @@
 
 	FilePath Application::GetConfigFilePath (const wxString &configFileName, bool createConfigDir)
 	{
-		wxStandardPaths stdPaths;
 		DirectoryPath configDir;
 		
 		if (!Core->IsInPortableMode())
@@ -46,7 +45,7 @@
 			configPath.Normalize();
 			configDir = wstring (configPath.GetFullPath());
 #else
-			configDir = wstring (stdPaths.GetUserDataDir());
+			configDir = wstring (wxStandardPaths::Get().GetUserDataDir());
 #endif
 		}
 		else
@@ -61,12 +60,12 @@
 
 	DirectoryPath Application::GetExecutableDirectory ()
 	{
-		return wstring (wxFileName (wxStandardPaths().GetExecutablePath()).GetPath());
+		return wstring (wxFileName (wxStandardPaths::Get().GetExecutablePath()).GetPath());
 	}
 
 	FilePath Application::GetExecutablePath ()
 	{
-		return wstring (wxStandardPaths().GetExecutablePath());
+		return wstring (wxStandardPaths::Get().GetExecutablePath());
 	}
 
 	void Application::Initialize (UserInterfaceType::Enum type)
diff -Naur truecrypt-7.1a-source/Main/CommandLineInterface.cpp truecrypt-7.1a-source.new/Main/CommandLineInterface.cpp
--- truecrypt-7.1a-source/Main/CommandLineInterface.cpp	2012-02-07 11:36:48.000000000 +0100
+++ truecrypt-7.1a-source.new/Main/CommandLineInterface.cpp	2014-02-11 09:52:51.693189622 +0100
@@ -380,7 +380,7 @@
 		ArgQuick = parser.Found (L"quick");
 
 		if (parser.Found (L"random-source", &str))
-			ArgRandomSourcePath = FilesystemPath (str);
+			ArgRandomSourcePath = FilesystemPath (str.wc_str());
 
 		if (parser.Found (L"restore-headers"))
 		{
@@ -471,7 +471,7 @@
 
 			if (param1IsFile)
 			{
-				ArgFilePath.reset (new FilePath (parser.GetParam (0)));
+				ArgFilePath.reset (new FilePath (parser.GetParam (0).wc_str()));
 			}
 		}
 
@@ -524,7 +524,7 @@
 					arr.Add (L"");
 					continue;
 				}
-				arr.Last() += token.empty() ? L',' : token;
+				arr.Last() += token.empty() ? wxString(L',') : token;
 			}
 			else
 				arr.Add (token);
@@ -562,12 +562,12 @@
 			{
 				filteredVolumes.push_back (volume);
 			}
-			else if (wxString (volume->Path) == pathFilter.GetFullPath())
+			else if (wxString (wstring(volume->Path)) == pathFilter.GetFullPath())
 			{
 				filteredVolumes.push_back (volume);
 			}
-			else if (wxString (volume->MountPoint) == pathFilter.GetFullPath()
-				|| (wxString (volume->MountPoint) + wxFileName::GetPathSeparator()) == pathFilter.GetFullPath())
+			else if (wxString (wstring(volume->MountPoint)) == pathFilter.GetFullPath()
+				|| (wxString (wstring(volume->MountPoint)) + wxFileName::GetPathSeparator()) == pathFilter.GetFullPath())
 			{
 				filteredVolumes.push_back (volume);
 			}
diff -Naur truecrypt-7.1a-source/Main/Forms/MainFrame.cpp truecrypt-7.1a-source.new/Main/Forms/MainFrame.cpp
--- truecrypt-7.1a-source/Main/Forms/MainFrame.cpp	2012-02-07 11:36:48.000000000 +0100
+++ truecrypt-7.1a-source.new/Main/Forms/MainFrame.cpp	2014-02-11 09:58:09.782203575 +0100
@@ -828,7 +828,7 @@
 			// File-hosted volumes
 			if (!volume->Path.IsDevice() && !mountPoint.IsEmpty())
 			{
-				if (wxString (volume->Path).Upper().StartsWith (wstring (mountPoint).c_str()))
+				if (wxString (wstring(volume->Path)).Upper().StartsWith (wxString (wstring(mountPoint))))
 				{
 					removedVolumes.push_back (volume);
 					continue;
@@ -1020,7 +1020,7 @@
 		if (!ListItemRightClickEventPending)
 		{
 			ListItemRightClickEventPending = true;
-			SlotListCtrl->AddPendingEvent (event);
+			SlotListCtrl->GetEventHandler()->AddPendingEvent (event);
 			return;
 		}
 
diff -Naur truecrypt-7.1a-source/Main/Forms/SelectDirectoryWizardPage.cpp truecrypt-7.1a-source.new/Main/Forms/SelectDirectoryWizardPage.cpp
--- truecrypt-7.1a-source/Main/Forms/SelectDirectoryWizardPage.cpp	2012-02-07 11:36:48.000000000 +0100
+++ truecrypt-7.1a-source.new/Main/Forms/SelectDirectoryWizardPage.cpp	2014-02-11 09:58:58.784205724 +0100
@@ -16,7 +16,7 @@
 	{
 		if (!DirectoryTextCtrl->IsEmpty())
 		{
-			return FilesystemPath (DirectoryTextCtrl->GetValue()).IsDirectory();
+			return FilesystemPath (wstring(DirectoryTextCtrl->GetValue())).IsDirectory();
 		}
 
 		return false;
diff -Naur truecrypt-7.1a-source/Main/Forms/SelectDirectoryWizardPage.h truecrypt-7.1a-source.new/Main/Forms/SelectDirectoryWizardPage.h
--- truecrypt-7.1a-source/Main/Forms/SelectDirectoryWizardPage.h	2012-02-07 11:36:48.000000000 +0100
+++ truecrypt-7.1a-source.new/Main/Forms/SelectDirectoryWizardPage.h	2014-02-11 09:59:29.475207071 +0100
@@ -18,7 +18,7 @@
 	public:
 		SelectDirectoryWizardPage (wxPanel* parent) : SelectDirectoryWizardPageBase (parent) { }
 
-		DirectoryPath GetDirectory () const { return DirectoryPath (DirectoryTextCtrl->GetValue()); }
+		DirectoryPath GetDirectory () const { return DirectoryPath (wstring(DirectoryTextCtrl->GetValue())); }
 		bool IsValid ();
 		void SetDirectory (const DirectoryPath &path) { DirectoryTextCtrl->SetValue (wstring (path)); }
 		void SetMaxStaticTextWidth (int width) { InfoStaticText->Wrap (width); }
diff -Naur truecrypt-7.1a-source/Main/GraphicUserInterface.cpp truecrypt-7.1a-source.new/Main/GraphicUserInterface.cpp
--- truecrypt-7.1a-source/Main/GraphicUserInterface.cpp	2012-02-07 11:36:48.000000000 +0100
+++ truecrypt-7.1a-source.new/Main/GraphicUserInterface.cpp	2014-02-11 10:36:13.104303730 +0100
@@ -1384,7 +1384,7 @@
 #else
 			L"",
 #endif
-			L"", wxDD_DEFAULT_STYLE | (existingOnly ? wxDD_DIR_MUST_EXIST : 0), wxDefaultPosition, parent));
+			L"", wxDD_DEFAULT_STYLE | (existingOnly ? wxDD_DIR_MUST_EXIST : 0), wxDefaultPosition, parent).wc_str());
 	}
 
 	FilePathList GraphicUserInterface::SelectFiles (wxWindow *parent, const wxString &caption, bool saveMode, bool allowMultiple, const list < pair <wstring, wstring> > &fileExtensions, const DirectoryPath &directory) const
@@ -1428,14 +1428,14 @@
 		if (dialog.ShowModal() == wxID_OK)
 		{
 			if (!allowMultiple)
-				files.push_back (make_shared <FilePath> (dialog.GetPath()));
+				files.push_back (make_shared <FilePath> (dialog.GetPath().wc_str()));
 			else
 			{
 				wxArrayString paths;
 				dialog.GetPaths (paths);
 
 				foreach (const wxString &path, paths)
-					files.push_back (make_shared <FilePath> (path));
+					files.push_back (make_shared <FilePath> (path.wc_str()));
 			}
 		}
 
diff -Naur truecrypt-7.1a-source/Main/TextUserInterface.cpp truecrypt-7.1a-source.new/Main/TextUserInterface.cpp
--- truecrypt-7.1a-source/Main/TextUserInterface.cpp	2012-02-07 11:36:48.000000000 +0100
+++ truecrypt-7.1a-source.new/Main/TextUserInterface.cpp	2014-02-11 10:03:41.528218127 +0100
@@ -116,7 +116,7 @@
 			for (size_t i = 0; i < length && i < VolumePassword::MaxSize; ++i)
 			{
 				passwordBuf[i] = (wchar_t) passwordStr[i];
-				const_cast <wchar_t *> (passwordStr.c_str())[i] = L'X';
+				const_cast <wchar_t *> (passwordStr.wc_str())[i] = L'X';
 			}
 
 			if (verify && verPhase)
@@ -763,8 +763,8 @@
 
 				ShowString (wxString::Format (L"\rDone: %7.3f%%  Speed: %9s  Left: %s         ",
 					100.0 - double (options->Size - progress.SizeDone) / (double (options->Size) / 100.0),
-					speed > 0 ? SpeedToString (speed).c_str() : L" ",
-					speed > 0 ? TimeSpanToString ((options->Size - progress.SizeDone) / speed).c_str() : L""));
+					speed > 0 ? SpeedToString (speed).wc_str() : L" ",
+					speed > 0 ? TimeSpanToString ((options->Size - progress.SizeDone) / speed).wc_str() : L""));
 			}
 
 			Thread::Sleep (100);
diff -Naur truecrypt-7.1a-source/Main/UserPreferences.cpp truecrypt-7.1a-source.new/Main/UserPreferences.cpp
--- truecrypt-7.1a-source/Main/UserPreferences.cpp	2012-02-07 11:36:48.000000000 +0100
+++ truecrypt-7.1a-source.new/Main/UserPreferences.cpp	2014-02-11 10:04:26.222220087 +0100
@@ -219,7 +219,7 @@
 
 			foreach_ref (const Keyfile &keyfile, DefaultKeyfiles)
 			{
-				keyfilesXml.InnerNodes.push_back (XmlNode (L"keyfile", wxString (FilesystemPath (keyfile))));
+				keyfilesXml.InnerNodes.push_back (XmlNode (L"keyfile", wxString (wstring(FilesystemPath (keyfile)))));
 			}
 
 			XmlWriter keyfileWriter (keyfilesCfgPath);
diff -Naur truecrypt-7.1a-source/Platform/Unix/Process.cpp truecrypt-7.1a-source.new/Platform/Unix/Process.cpp
--- truecrypt-7.1a-source/Platform/Unix/Process.cpp	2012-02-07 11:36:48.000000000 +0100
+++ truecrypt-7.1a-source.new/Platform/Unix/Process.cpp	2014-02-11 10:06:08.925224592 +0100
@@ -52,9 +52,9 @@
 					if (!execFunctor)
 						args[argIndex++] = const_cast <char*> (processName.c_str());
 
-					foreach (const string &arg, arguments)
+					for (list<string>::const_iterator arg=arguments.begin(); arg != arguments.end(); ++arg)
 					{
-						args[argIndex++] = const_cast <char*> (arg.c_str());
+						args[argIndex++] = const_cast <char*> ((*arg).c_str());
 					}
 					args[argIndex] = nullptr;
 
