diff -ur a/phc-k8.c b/phc-k8.c
--- a/phc-k8.c	2011-06-26 15:42:16.000000000 -0500
+++ b/phc-k8.c	2012-07-29 06:43:46.286246524 -0500
@@ -57,10 +57,10 @@
 #include "mperf.h"
 
 /* Module parameters */
-static int direct_transitions = 0;
+static bool direct_transitions = 0;
 module_param(direct_transitions, bool, 0444);
 MODULE_PARM_DESC(direct_transitions, "Use direct P-States transitions");
-static int min_800MHz = 0;
+static bool min_800MHz = 0;
 module_param(min_800MHz, bool, 0444);
 MODULE_PARM_DESC(min_800MHz, "Allow 800MHz as lowest frequency with direct transitions enabled");
 static int be_maxfid = -1;
