diff '--color=auto' -Naur mininet-2.3.1b4/Makefile mininet-2.3.1b4.new/Makefile
--- mininet-2.3.1b4/Makefile	2023-05-03 03:29:26.000000000 +0200
+++ mininet-2.3.1b4.new/Makefile	2025-06-18 17:10:45.532147194 +0200
@@ -9,7 +9,7 @@
 MNEXEC = mnexec
 MANPAGES = mn.1 mnexec.1
 P8IGN = E251,E201,E302,E202,E126,E127,E203,E226,E402,W504,W503,E731
-PREFIX ?= /usr
+PREFIX ?= "$(DESTDIR)"/usr
 BINDIR ?= $(PREFIX)/bin
 MANDIR ?= $(PREFIX)/share/man/man1
 DOCDIRS = doc/html doc/latex
@@ -58,15 +58,11 @@
 
 install: install-mnexec install-manpages
 #	This seems to work on all pip versions
-	$(PYTHON) -m pip uninstall -y mininet || true
-	$(PYTHON) -m pip install .
 
 develop: $(MNEXEC) $(MANPAGES)
 # 	Perhaps we should link these as well
 	install $(MNEXEC) $(BINDIR)
 	install $(MANPAGES) $(MANDIR)
-	$(PYTHON) -m pip uninstall -y mininet || true
-	$(PYTHON) -m pip install -e . --no-binary :all:
 
 man: $(MANPAGES)
 
