--- m17n-lib-1.8.0/configure.ac	2017-12-08 13:13:45.000000000 +0100
+++ m17n-lib-1.8.0.new/configure.ac	2022-05-17 18:16:31.155472511 +0200
@@ -204,21 +204,21 @@
 AC_SUBST(OTF_LD_FLAGS)
 
 dnl Check for Freetype2 usability.
-AC_CHECK_PROG(HAVE_FREETYPE_CONFIG, freetype-config, yes)
+AC_CHECK_PROG(HAVE_FREETYPE_CONFIG, pkg-config freetype2, yes)
 if test "x$HAVE_FREETYPE_CONFIG" = "xyes"; then
-  FREETYPE_INC=`freetype-config --cflags`
+  FREETYPE_INC=`pkg-config freetype2 --cflags`
   save_CPPFLAGS="$CPPFLAGS"
   CPPFLAGS="$CPPFLAGS $FREETYPE_INC"
   AC_CHECK_HEADER(ft2build.h, HAVE_FREETYPE=yes,
   			      HAVE_FREETYPE=no CPPFLAGS="$save_CPPFLAGS")
   if test "x$HAVE_FREETYPE" = "xyes" ; then
     save_LIBS="$LIBS"
-    LIBS="$LIBS `freetype-config --libs`"
+    LIBS="$LIBS `pkg-config freetype2 --libs`"
     AC_CHECK_LIB(freetype, FT_Init_FreeType, HAVE_FREETYPE=yes,
     			   		     HAVE_FREETYPE=no)
     LIBS="$save_LIBS"
     if test "x$HAVE_FREETYPE" = "xyes"; then
-      FREETYPE_LD_FLAGS=`freetype-config --libs`
+      FREETYPE_LD_FLAGS=`pkg-config freetype2 --libs`
       AC_DEFINE(HAVE_FREETYPE, 1, 
 		[Define to 1 if you have FreeType library and header file.])
       M17N_EXT_LIBS="$M17N_EXT_LIBS freetype"
@@ -356,10 +356,10 @@
 AC_SUBST(ANTHY_LD_FLAGS)
 
 dnl Check for Ispell usability.
-AC_CHECK_PROG(HAVE_ISPELL, ispell, yes)
+AC_CHECK_PROG(HAVE_ISPELL, aspell, yes)
 if test "x$HAVE_ISPELL" = "xyes"; then
-  AC_DEFINE(HAVE_ISPELL, 1, [Define if ispell is available.])
-  M17N_EXT_LIBS="$M17N_EXT_LIBS ispell"
+  AC_DEFINE(HAVE_ISPELL, 1, [Define if aspell is available.])
+  M17N_EXT_LIBS="$M17N_EXT_LIBS aspell"
   CONFIG_FLAGS="$CONFIG_FLAGS -DHAVE_ISPELL"
 fi
 
