--- blueproximity-1.3.3/proximity.py	2020-08-25 13:17:29.000000000 +0200
+++ blueproximity-1.3.3.new/proximity.py	2024-08-26 16:06:14.871006347 +0200
@@ -26,7 +26,7 @@
 # This value gives us the base directory for language files and icons.
 # Set this value to './' for local folder version
 # or, for instance, to '/usr/share/blueproximity/' for packaged version
-dist_path = './'
+dist_path = '/usr/share/blueproximity/'
 
 # Translation stuff
 import gettext
