diff -x '*~' -ur xine-lib-1.0.1/configure.ac xine-lib-1.0.1-vidix64/configure.ac
--- xine-lib-1.0.1/configure.ac	2005-04-26 10:09:18.000000000 +0200
+++ xine-lib-1.0.1-vidix64/configure.ac	2005-05-01 17:16:05.289166056 +0200
@@ -940,7 +940,7 @@
 if test x"$check_vidix" = "xyes" -a x"$ac_cv_prog_AWK" != "xno"; then
   if test x"$no_x" != "xyes" -o x"$have_fb" = "xyes"; then
     case "$host_or_hostalias" in
-      i?86-*-linux* | k?-*-linux* | athlon-*-linux*)
+      i?86-*-linux* | k?-*-linux* | athlon-*-linux* | x86_64-*-linux*)
         enable_vidix="yes"
         enable_linux="yes"
         ;;
diff -x '*~' -ur xine-lib-1.0.1/src/video_out/libdha/AsmMacros.h xine-lib-1.0.1-vidix64/src/video_out/libdha/AsmMacros.h
--- xine-lib-1.0.1/src/video_out/libdha/AsmMacros.h	2005-04-26 10:09:11.000000000 +0200
+++ xine-lib-1.0.1-vidix64/src/video_out/libdha/AsmMacros.h	2005-05-01 18:22:18.533141160 +0200
@@ -64,6 +64,10 @@
 */
 
 #if defined(__GNUC__) || defined(__ICC)
+
+#ifdef __linux__
+#include <sys/io.h>
+#else
 #if defined(__alpha__)
 #include "sysdep/AsmMacros_alpha.h"
 #elif defined(__ia64__)
@@ -74,11 +78,12 @@
 #include "sysdep/AsmMacros_arm32.h"
 #elif defined(__powerpc__)
 #include "sysdep/AsmMacros_powerpc.h"
-#elif defined (__i386__)
+#elif defined (__i386__) || defined (__amd64__)
 #include "sysdep/AsmMacros_x86.h"
 #else
 #include "sysdep/AsmMacros_generic.h"
 #endif
+#endif
 
 #else /* __GNUC__ */
 
diff -x '*~' -ur xine-lib-1.0.1/src/video_out/libdha/pci.c xine-lib-1.0.1-vidix64/src/video_out/libdha/pci.c
--- xine-lib-1.0.1/src/video_out/libdha/pci.c	2005-04-26 10:09:11.000000000 +0200
+++ xine-lib-1.0.1-vidix64/src/video_out/libdha/pci.c	2005-05-01 17:31:09.042774624 +0200
@@ -1,6 +1,6 @@
 /*
    (C) 2002 - library implementation by Nick Kyrshev
-   XFree86 3.3.3 scanpci.c, modified for GATOS/win/gfxdump by Øyvind Aabling.
+   XFree86 3.3.3 scanpci.c, modified for GATOS/win/gfxdump by ï¿½vind Aabling.
  */
 /* $XConsortium: scanpci.c /main/25 1996/10/27 11:48:40 kaleb $ */
 /*
@@ -464,7 +464,7 @@
 #include "sysdep/pci_arm32.c"
 #elif defined(__powerpc__)
 #include "sysdep/pci_powerpc.c"
-#elif defined( __i386__ )
+#elif defined( __i386__ ) || defined(__amd64__)
 #include "sysdep/pci_x86.c"
 #else
 #include "sysdep/pci_generic_cpu.c"
diff -x '*~' -ur xine-lib-1.0.1/src/video_out/libdha/sysdep/AsmMacros_x86.h xine-lib-1.0.1-vidix64/src/video_out/libdha/sysdep/AsmMacros_x86.h
--- xine-lib-1.0.1/src/video_out/libdha/sysdep/AsmMacros_x86.h	2005-04-26 10:09:11.000000000 +0200
+++ xine-lib-1.0.1-vidix64/src/video_out/libdha/sysdep/AsmMacros_x86.h	2005-05-01 18:00:40.057539456 +0200
@@ -13,25 +13,25 @@
 
 #include "config.h"
 
-static __inline__ void outb(short port,char val)
+static __inline__ void outb(unsigned short port,char val)
 {
    __asm__ __volatile__("outb %0,%1" : :"a" (val), "d" (port));
     return;
 }
 
-static __inline__ void outw(short port,short val)
+static __inline__ void outw(unsigned short port,short val)
 {
    __asm__ __volatile__("outw %0,%1" : :"a" (val), "d" (port));
     return;
 }
 
-static __inline__ void outl(short port,unsigned int val)
+static __inline__ void outl(unsigned short port,unsigned int val)
 {
    __asm__ __volatile__("outl %0,%1" : :"a" (val), "d" (port));
     return;
 }
 
-static __inline__ unsigned int inb(short port)
+static __inline__ unsigned int inb(unsigned short port)
 {
    unsigned char ret;
    __asm__ __volatile__("inb %1,%0" :
@@ -40,7 +40,7 @@
    return ret;
 }
 
-static __inline__ unsigned int inw(short port)
+static __inline__ unsigned int inw(unsigned short port)
 {
    unsigned short ret;
    __asm__ __volatile__("inw %1,%0" :
@@ -49,7 +49,7 @@
    return ret;
 }
 
-static __inline__ unsigned int inl(short port)
+static __inline__ unsigned int inl(unsigned short port)
 {
    unsigned int ret;
    __asm__ __volatile__("inl %1,%0" :
