--- xine-lib-1.2.12/src/video_dec/dav1d.c	2022-03-06 17:31:24.000000000 +0100
+++ xine-lib-1.2.12.new/src/video_dec/dav1d.c	2022-08-25 19:02:59.832190339 +0200
@@ -544,11 +544,17 @@
 
   /* multithreading */
   ncpu = xine_cpu_count();
+#if DAV1D_API_VERSION_MAJOR > 5
+  settings.n_threads = ncpu + 1;
+  xprintf(stream->xine, XINE_VERBOSITY_DEBUG, LOG_MODULE ": "
+            "Using %d threads\n", settings.n_threads);
+#else
   settings.n_frame_threads = (ncpu > 8) ? 4 : (ncpu < 2) ? 1 : ncpu/2;
   settings.n_tile_threads = MAX(1, ncpu - settings.n_frame_threads + 1);
   xprintf(stream->xine, XINE_VERBOSITY_DEBUG, LOG_MODULE ": "
             "Using %d frame threads, %d tile threads\n",
             settings.n_frame_threads, settings.n_tile_threads);
+#endif
 
   /* dri frame allocator */
   settings.allocator.cookie = this;
