diff -uNrp a/src/combined/ffmpeg/ffmpeg_compat.h b/src/combined/ffmpeg/ffmpeg_compat.h
--- a/src/combined/ffmpeg/ffmpeg_compat.h	2022-03-06 16:31:23.000000000 +0000
+++ b/src/combined/ffmpeg/ffmpeg_compat.h	2022-08-09 12:09:50.709059577 +0100
@@ -19,6 +19,8 @@
  *
  * Compability macros for various ffmpeg versions
  */
+#include <libavcodec/version.h>
+#include <libavutil/version.h>
 
 #ifndef XINE_AVCODEC_COMPAT_H
 #define XINE_AVCODEC_COMPAT_H
diff -uNrp a/src/combined/ffmpeg/ffmpeg_decoder.h b/src/combined/ffmpeg/ffmpeg_decoder.h
--- a/src/combined/ffmpeg/ffmpeg_decoder.h	2022-03-06 16:31:23.000000000 +0000
+++ b/src/combined/ffmpeg/ffmpeg_decoder.h	2022-08-09 12:42:16.081772933 +0100
@@ -17,6 +17,8 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110, USA
  */
+#include <libavcodec/version.h>
+#include <libavformat/avformat.h>
 
 #ifndef HAVE_XINE_DECODER_H
 #define HAVE_XINE_DECODER_H
diff -uNrp a/src/combined/ffmpeg/input_avio.c b/src/combined/ffmpeg/input_avio.c
--- a/src/combined/ffmpeg/input_avio.c	2022-03-06 16:31:23.000000000 +0000
+++ b/src/combined/ffmpeg/input_avio.c	2022-08-09 11:33:54.527895581 +0100
@@ -28,6 +28,7 @@
 #include <string.h>
 #include <pthread.h>
 
+#include <libavutil/avutil.h>
 #include <libavformat/avio.h>
 
 #define LOG_MODULE "libavio"
