--- xine-lib-1.1.4.orig/src/libw32dll/dmo/DMO_VideoDecoder.c
+++ xine-lib-1.1.4/src/libw32dll/dmo/DMO_VideoDecoder.c
@@ -118,6 +118,7 @@
      
         this->iv.m_bh = (BITMAPINFOHEADER*)malloc(bihs);
         memcpy(this->iv.m_bh, format, bihs);
+        this->iv.m_bh->biSize = bihs;
 
         this->iv.m_State = STOP;
         //this->iv.m_pFrame = 0;
--- xine-lib-1.1.4.orig/src/libw32dll/DirectShow/DS_VideoDecoder.c
+++ xine-lib-1.1.4/src/libw32dll/DirectShow/DS_VideoDecoder.c
@@ -110,6 +110,7 @@
      
         this->iv.m_bh = (BITMAPINFOHEADER*)malloc(bihs);
         memcpy(this->iv.m_bh, format, bihs);
+        this->iv.m_bh->biSize = bihs;
 
         this->iv.m_State = STOP;
         //this->iv.m_pFrame = 0;
