--- yeahlaunch-0.2/yeahlaunch.c	2004-12-18 04:46:10.000000000 -0600
+++ yeahlaunch-0.2-2/yeahlaunch.c	2008-07-24 01:01:04.000000000 -0500
@@ -120,7 +120,7 @@
         if (!strncmp(argv[i], "-", 1) && i + 1 < argc)
             i++;
         else if (i + 1 < argc)
-            make_new_tab(argv[++i], argv[i]);
+            make_new_tab(argv[i], argv[i++]);
     }
     for (t = head_tab; t; t = t->next)
         width += t->width;
