From 8484af61bb1f6d383d115364566575fd1ed24bd8 Mon Sep 17 00:00:00 2001
From: Benson Muite <bkmgit@users.noreply.github.com>
Date: Wed, 22 Feb 2023 11:12:09 +0300
Subject: [PATCH] Remove unneeded check

---
 configure.ac | 1 -
 1 file changed, 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 0aaa6d6c1..02b8be000 100644
--- a/configure.ac
+++ b/configure.ac
@@ -37,7 +37,6 @@ case "$ac_cv_search_dlopen" in
 esac
 
 dnl Checks for header files.
-AC_HEADER_STDC
 AC_CHECK_HEADERS(asm/ptrace_offsets.h asm/ptrace.h asm/vsyscall.h endian.h sys/endian.h \
 		sys/param.h execinfo.h ia64intrin.h sys/uc_access.h unistd.h signal.h \
 		sys/types.h sys/procfs.h sys/ptrace.h sys/syscall.h byteswap.h elf.h \
