diff -Naur pwmanager-1.2.4-orig/pwmanager/pwmanager.desktop pwmanager-1.2.4/pwmanager/pwmanager.desktop
--- pwmanager-1.2.4-orig/pwmanager/pwmanager.desktop	2005-11-06 02:16:50.000000000 +0100
+++ pwmanager-1.2.4/pwmanager/pwmanager.desktop	2007-06-13 21:31:55.000000000 +0200
@@ -1,6 +1,7 @@
 [Desktop Entry]
 Encoding=UTF-8
 Type=Application
+Categories=Application;KDE;Utility;
 Exec=pwmanager -caption "%c" %i %m
 Icon=pwmanager
 Comment=The convenient way of managing passwords
