diff -Naur xkbcomp-1.0.3.orig/Makefile.am xkbcomp-1.0.3/Makefile.am
--- xkbcomp-1.0.3.orig/Makefile.am	2006-12-06 01:38:58.000000000 +0100
+++ xkbcomp-1.0.3/Makefile.am	2006-12-06 01:40:05.000000000 +0100
@@ -21,7 +21,7 @@
 
 bin_PROGRAMS = xkbcomp
 
-AM_CFLAGS = $(XKBCOMP_CFLAGS) -DDFLT_XKB_CONFIG_ROOT='"$(datadir)/X11/xkb"'
+AM_CFLAGS = $(XKBCOMP_CFLAGS) -DDFLT_XKB_CONFIG_ROOT='"$(libdir)/X11/xkb"'
 xkbcomp_LDADD = $(XKBCOMP_LIBS)
 
 xkbcomp_SOURCES = \
