--- DiffWindow.cxx.orig	2011-09-21 09:22:19.921002567 +0200
+++ DiffWindow.cxx	2010-12-21 20:41:45.000000000 +0100
@@ -1,5 +1,5 @@
 //
-// "$Id: DiffWindow.cxx 407 2006-11-13 18:54:02Z mike $"
+// "$Id$"
 //
 // DiffWindow widget code.
 //
@@ -58,7 +58,7 @@
 #include <string.h>
 #include <errno.h>
 
-#if defined(WIN32)
+#if defined(WIN32)  && !defined(__CYGWIN__)
 #include <io.h>
 #define snprintf _snprintf
 #define access _access
@@ -1126,11 +1126,11 @@
 DiffWindow::save_prefs()
 {
   // Save the window prefs for the next run...
-  prefs_.set("color", color());
-  prefs_.set("selection_color", selection_color());
+  prefs_.set("color", (int)color());
+  prefs_.set("selection_color", (int)selection_color());
   prefs_.set("showlinenum", showlinenum());
   prefs_.set("tabwidth", tabwidth());
-  prefs_.set("textcolor", textcolor());
+  prefs_.set("textcolor", (int)textcolor());
   prefs_.set("textsize", textsize());
   prefs_.set("ignoreblanks", ignoreblanks());
 }
@@ -1301,5 +1301,5 @@
 
 
 //
-// End of "$Id: DiffWindow.cxx 407 2006-11-13 18:54:02Z mike $".
+// End of "$Id$".
 //
