--- lxmusic-0.4.4.orig/src/lxmusic-notify.c	2010-03-01 20:09:00.000000000 +0100
+++ lxmusic-0.4.4/src/lxmusic-notify.c	2010-11-18 00:47:51.461257411 +0100
@@ -57,11 +57,9 @@
 
 	lxmusic_clear_notify (n);
 
-	notify = notify_notification_new (summary, message,
-	                                  "lxmusic", NULL);
+	notify = notify_notification_new (summary, message, "lxmusic");
 	n->notification = notify;
 
-	notify_notification_attach_to_status_icon (notify, n->status_icon);
 	notify_notification_set_urgency (notify, NOTIFY_URGENCY_NORMAL);
 	notify_notification_set_timeout (notify, NOTIFY_EXPIRES_DEFAULT);
 
