diff -ur a/Makefile b/Makefile
--- a/Makefile	2012-10-24 11:28:55.477838649 -0500
+++ b/Makefile	2012-10-24 11:29:31.606854167 -0500
@@ -15,8 +15,7 @@
 DEPFLAGS = -Wp,-MMD,$(@D)/.$(@F).d,-MT,$@
 
 INSTALL = install
-prefix ?= /usr/local
-bindir = $(prefix)/bin
+bindir = /sbin
 LIBS=-luuid -lm
 RESTORE_LIBS=-lz
 
diff -ur a/man/Makefile b/man/Makefile
--- a/man/Makefile	2012-10-24 11:28:55.476838648 -0500
+++ b/man/Makefile	2012-10-24 11:35:31.863031553 -0500
@@ -1,10 +1,9 @@
 GZIP=gzip
 INSTALL= install
 
-prefix ?= /usr/local
-bindir = $(prefix)/bin
-mandir = $(prefix)/man
-man8dir = $(mandir)/man8
+bindir = /sbin
+mandir = /usr/share/man
+man8dir = /usr/share/man/man8
 
 MANPAGES = mkfs.btrfs.8.gz btrfsctl.8.gz btrfsck.8.gz btrfs-image.8.gz \
 	   btrfs-show.8.gz btrfs.8.gz
