diff -Naur btrfs-progs-v4.10/ctree.h btrfs-progs-v4.10-p/ctree.h
--- btrfs-progs-v4.10/ctree.h	2017-03-08 14:23:24.000000000 +0100
+++ btrfs-progs-v4.10-p/ctree.h	2017-03-15 15:13:26.765989563 +0100
@@ -34,7 +34,7 @@
 #include <btrfs/extent-cache.h>
 #include <btrfs/extent_io.h>
 #include <btrfs/ioctl.h>
-#include <linux/sizes.h>
+#include <btrfs/sizes.h>
 #endif /* BTRFS_FLAT_INCLUDES */
 
 struct btrfs_root;
diff -Naur btrfs-progs-v4.10/Makefile btrfs-progs-v4.10-p/Makefile
--- btrfs-progs-v4.10/Makefile	2017-03-08 14:23:24.000000000 +0100
+++ btrfs-progs-v4.10-p/Makefile	2017-03-15 15:12:43.734828807 +0100
@@ -108,7 +108,7 @@
 		   uuid-tree.o utils-lib.o rbtree-utils.o
 libbtrfs_headers = send-stream.h send-utils.h send.h kernel-lib/rbtree.h btrfs-list.h \
 	       kernel-lib/crc32c.h kernel-lib/list.h kerncompat.h \
-	       kernel-lib/radix-tree.h extent-cache.h \
+	       kernel-lib/radix-tree.h extent-cache.h kernel-lib/sizes.h \
 	       extent_io.h ioctl.h ctree.h btrfsck.h version.h
 convert_objects = convert/main.o convert/common.o convert/source-fs.o \
 		  convert/source-ext2.o
