diff -Naur sqlitebrowser-3.10.1/CMakeLists.txt sqlitebrowser-3.10.1-p/CMakeLists.txt
--- sqlitebrowser-3.10.1/CMakeLists.txt	2017-09-20 15:59:51.000000000 +0200
+++ sqlitebrowser-3.10.1-p/CMakeLists.txt	2018-06-16 01:44:57.863076518 +0200
@@ -46,6 +46,7 @@
 	set(VSREDIST_DIR "C:/dev/dependencies")
 endif()
 
+
 if(NOT FORCE_INTERNAL_ANTLR)
     find_package(Antlr2)
 endif()
@@ -70,6 +71,9 @@
 find_package(Qt5Widgets REQUIRED)
 find_package(Qt5LinguistTools REQUIRED)
 find_package(Qt5Network REQUIRED)
+find_package(Qt5PrintSupport REQUIRED)
+find_package(Qt5Test REQUIRED)
+
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 
@@ -291,7 +295,7 @@
 		${SQLB_RESOURCES_RCC}
 		${SQLB_MISC})
 
-qt5_use_modules(${PROJECT_NAME} Gui Widgets Network Test PrintSupport)
+target_link_libraries(${PROJECT_NAME} Qt5::Gui Qt5::Widgets Qt5::Network Qt5::Test Qt5::PrintSupport)
 set(QT_LIBRARIES "")
 
 add_dependencies(${PROJECT_NAME} qhexedit qcustomplot)
diff -Naur sqlitebrowser-3.10.1/libs/qcustomplot-source/CMakeLists.txt sqlitebrowser-3.10.1-p/libs/qcustomplot-source/CMakeLists.txt
--- sqlitebrowser-3.10.1/libs/qcustomplot-source/CMakeLists.txt	2017-09-20 15:59:51.000000000 +0200
+++ sqlitebrowser-3.10.1-p/libs/qcustomplot-source/CMakeLists.txt	2018-06-16 01:43:54.559713223 +0200
@@ -3,6 +3,7 @@
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 find_package(Qt5Widgets REQUIRED)
+find_package(Qt5PrintSupport REQUIRED)
 
 set(QCUSTOMPLOT_SRC
 	qcustomplot.cpp
@@ -17,4 +18,4 @@
 
 add_library(qcustomplot ${QCUSTOMPLOT_SRC} ${QCUSTOMPLOT_HDR} ${QCUSTOMPLOT_MOC})
 
-qt5_use_modules(qcustomplot Widgets PrintSupport)
+target_link_libraries(qcustomplot Qt5::Widgets Qt5::PrintSupport)
diff -Naur sqlitebrowser-3.10.1/libs/qhexedit/CMakeLists.txt sqlitebrowser-3.10.1-p/libs/qhexedit/CMakeLists.txt
--- sqlitebrowser-3.10.1/libs/qhexedit/CMakeLists.txt	2017-09-20 15:59:51.000000000 +0200
+++ sqlitebrowser-3.10.1-p/libs/qhexedit/CMakeLists.txt	2018-06-16 01:43:25.090545101 +0200
@@ -22,4 +22,4 @@
 
 add_library(qhexedit ${QHEXEDIT_SRC} ${QHEXEDIT_HDR} ${QHEXEDIT_MOC})
 
-qt5_use_modules(qhexedit Widgets)
+target_link_libraries(qhexedit Qt5::Widgets)
diff -Naur sqlitebrowser-3.10.1/libs/qscintilla/Qt4Qt5/CMakeLists.txt sqlitebrowser-3.10.1-p/libs/qscintilla/Qt4Qt5/CMakeLists.txt
--- sqlitebrowser-3.10.1/libs/qscintilla/Qt4Qt5/CMakeLists.txt	2017-09-20 15:59:51.000000000 +0200
+++ sqlitebrowser-3.10.1-p/libs/qscintilla/Qt4Qt5/CMakeLists.txt	2018-06-16 01:43:25.090545101 +0200
@@ -3,6 +3,7 @@
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 find_package(Qt5Widgets REQUIRED)
+find_package(Qt5PrintSupport REQUIRED)
 
 if(APPLE)
     find_package(Qt5MacExtras)
@@ -153,7 +154,7 @@
 add_library(qscintilla2 ${QSCINTILLA_SRC} ${QSCINTILLA_HDR} ${QSCINTILLA_MOC_HDR} ${QSCINTILLA_MOC})
 
 if (APPLE)
-	qt5_use_modules(qscintilla2 Widgets PrintSupport MacExtras)
+	target_link_libraries(qscintilla2 Qt5::Widgets Qt5::PrintSupport Qt5::MacExtras)
 else()
-	qt5_use_modules(qscintilla2 Widgets PrintSupport)
+	target_link_libraries(qscintilla2 Qt5::Widgets Qt5::PrintSupport)
 endif()
diff -Naur sqlitebrowser-3.10.1/src/tests/CMakeLists.txt sqlitebrowser-3.10.1-p/src/tests/CMakeLists.txt
--- sqlitebrowser-3.10.1/src/tests/CMakeLists.txt	2017-09-20 15:59:51.000000000 +0200
+++ sqlitebrowser-3.10.1-p/src/tests/CMakeLists.txt	2018-06-16 01:43:25.090545101 +0200
@@ -37,7 +37,7 @@
 
 add_executable(test-sqlobjects ${TESTSQLOBJECTS_MOC} ${TESTSQLOBJECTS_HDR} ${TESTSQLOBJECTS_SRC} ${TESTSQLOBJECTS_FORM_HDR})
 
-qt5_use_modules(test-sqlobjects Test Widgets Gui)
+target_link_libraries(test-sqlobjects Qt5::Test Qt5::Widgets Qt5::Gui)
 set(QT_LIBRARIES "")
 
 if(NOT ANTLR2_FOUND)
@@ -67,7 +67,7 @@
 
 add_executable(test-import ${TESTIMPORT_MOC} ${TESTIMPORT_SRC})
 
-qt5_use_modules(test-import Test Core)
+target_link_libraries(test-import Qt5::Test Qt5::Core)
 set(QT_LIBRARIES "")
 
 target_link_libraries(test-import ${QT_LIBRARIES})
@@ -106,7 +106,7 @@
 
 add_executable(test-regex ${TESTREGEX_MOC} ${TESTREGEX_HDR} ${TESTREGEX_SRC})
 
-qt5_use_modules(test-regex Test Core Gui Widgets)
+target_link_libraries(test-regex Qt5::Test Qt5::Core Qt5::Gui Qt5::Widgets)
 set(QT_LIBRARIES "")
 
 if(NOT ANTLR2_FOUND)
