--- sqlitebrowser-3.3.1/CMakeLists.txt	2014-08-31 17:23:49.000000000 +0300
+++ sqlitebrowser-3.3.1.new/CMakeLists.txt	2014-10-03 00:18:43.000000000 +0300
@@ -202,6 +202,14 @@
 install(TARGETS ${PROJECT_NAME}
 	RUNTIME DESTINATION bin
 	LIBRARY DESTINATION lib)
+	
+if(UNIX AND NOT APPLE)
+install(FILES src/icons/${PROJECT_NAME}.png
+	DESTINATION share/pixmaps/)
+	
+install(FILES distri/${PROJECT_NAME}.desktop
+	DESTINATION share/applications/)
+endif(UNIX AND NOT APPLE)
 
 #cpack
 set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "Database Browser for SQLite")
--- sqlitebrowser-3.3.1/distri/sqlitebrowser.desktop	2014-08-31 17:23:49.000000000 +0300
+++ sqlitebrowser-3.3.1.new/distri/sqlitebrowser.desktop	2014-10-03 00:18:54.000000000 +0300
@@ -3,7 +3,7 @@
 Comment=SQLiteBrowser is a light GUI editor for SQLite databases
 Comment[de]=SQLiteBrowser ist ein GUI-Editor für SQLite-Datenbanken
 Exec=sqlitebrowser %f
-Icon=/usr/share/sqlitebrowser/sqlitebrowser.svg
+Icon=sqlitebrowser
 Terminal=false
 X-MultipleArgs=false
 Type=Application
