diff -Naur sqlitebrowser-3.10.1/cmake/FindQScintilla.cmake sqlitebrowser-3.10.1-pp/cmake/FindQScintilla.cmake
--- sqlitebrowser-3.10.1/cmake/FindQScintilla.cmake	2017-09-20 15:59:51.000000000 +0200
+++ sqlitebrowser-3.10.1-pp/cmake/FindQScintilla.cmake	2018-06-16 01:37:23.337500050 +0200
@@ -42,10 +42,12 @@
 # either expressed or implied, of the FreeBSD Project.
 #=============================================================================
 
+SET (QT_MIN_VERSION "5.11.0")
+FIND_PACKAGE( Qt5 ${QT_MIN_VERSION} CONFIG REQUIRED COMPONENTS Network Widgets )
 
 find_path ( QSCINTILLA_INCLUDE_DIR
   NAMES qsciscintilla.h
-  HINTS ${QT_INCLUDE_DIR}
+  HINTS ${Qt5Core_INCLUDE_DIRS}
   PATH_SUFFIXES Qsci
 )
 
@@ -80,8 +82,8 @@
 
 
 find_library ( QSCINTILLA_LIBRARY
-  NAMES qscintilla qscintilla2 libqscintilla2
-  HINTS ${QT_LIBRARY_DIR}
+  NAMES qscintilla2 libqscintilla2 qscintilla2_qt5 libqscintilla2_qt5
+  HINTS ${Qt5Core_LIBRARIES}
 )
 
 set ( QSCINTILLA_LIBRARIES ${QSCINTILLA_LIBRARY} )
