diff -Naur mysql-5.5.9/sql-common/client_plugin.c mysql-5.5.9/sql-common/client_plugin.c
--- mysql-5.5.9/sql-common/client_plugin.c	2011-01-19 23:37:09.000000000 +0100
+++ mysql-5.5.9/sql-common/client_plugin.c	2011-03-16 15:10:21.076223240 +0100
@@ -227,6 +227,7 @@
 {
   MYSQL mysql;
   struct st_mysql_client_plugin **builtin;
+  va_list unused;
 
   if (initialized)
     return 0;
@@ -243,7 +244,7 @@
   pthread_mutex_lock(&LOCK_load_client_plugin);
 
   for (builtin= mysql_client_builtins; *builtin; builtin++)
-    add_plugin(&mysql, *builtin, 0, 0, 0);
+    add_plugin(&mysql, *builtin, 0, 0, unused);
 
   pthread_mutex_unlock(&LOCK_load_client_plugin);
 
@@ -287,6 +288,8 @@
 mysql_client_register_plugin(MYSQL *mysql,
                              struct st_mysql_client_plugin *plugin)
 {
+  va_list unused;
+
   if (is_not_initialized(mysql, plugin->name))
     return NULL;
 
@@ -301,7 +304,7 @@
     plugin= NULL;
   }
   else
-    plugin= add_plugin(mysql, plugin, 0, 0, 0);
+    plugin= add_plugin(mysql, plugin, 0, 0, unused);
 
   pthread_mutex_unlock(&LOCK_load_client_plugin);
   return plugin;
