--- cifs-utils-6.10/Makefile.am	2019-12-18 15:07:16.696530170 +0100
+++ cifs-utils-6.10.new/Makefile.am	2019-12-18 15:07:41.136702655 +0100
@@ -117,13 +117,3 @@
 endif
 
 SUBDIRS = contrib
-
-install-exec-hook:
-	(cd $(ROOTSBINDIR) && ln -sf mount.cifs mount.smb3)
-
-install-data-hook:
-	(cd $(man8dir) && ln -sf mount.cifs.8 mount.smb3.8)
-
-uninstall-hook:
-	(cd $(ROOTSBINDIR) && rm -f $(ROOTSBINDIR)/mount.smb3)
-	(cd $(man8dir) && rm -f $(man8dir)/mount.smb3.8)
