diff -Naur fontconfig-2.2.3.orig/fonts.conf.in fontconfig-2.2.3/fonts.conf.in
--- fontconfig-2.2.3.orig/fonts.conf.in	2003-04-17 00:04:42.000000000 +0200
+++ fontconfig-2.2.3/fonts.conf.in	2005-01-27 16:14:10.288628752 +0100
@@ -72,10 +72,10 @@
   Serif faces
  -->
 	<alias>
+		<family>Nimbus Roman No9 L</family>
 		<family>Bitstream Vera Serif</family>
 		<family>Times</family>
 		<family>Times New Roman</family>
-		<family>Nimbus Roman No9 L</family>
 		<family>Luxi Serif</family>
 		<family>Kochi Mincho</family>
 		<family>AR PL SungtiL GB</family>
@@ -87,11 +87,11 @@
   Sans-serif faces
  -->
 	<alias>
+		<family>Nimbus Sans L</family>
 		<family>Bitstream Vera Sans</family>
 		<family>Helvetica</family>
 		<family>Arial</family>
 		<family>Verdana</family>
-		<family>Nimbus Sans L</family>
 		<family>Luxi Sans</family>
 		<family>Kochi Gothic</family>
 		<family>AR PL KaitiM GB</family>
@@ -104,11 +104,11 @@
   Monospace faces
  -->
  	<alias>
+		<family>Luxi Mono</family>
 		<family>Bitstream Vera Sans Mono</family>
 		<family>Courier</family>
 		<family>Courier New</family>
 		<family>Andale Mono</family>
-		<family>Luxi Mono</family>
 		<family>Nimbus Mono L</family>
 		<family>NSimSun</family>
 		<default><family>monospace</family></default>
@@ -264,9 +264,9 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
+			<family>Nimbus Roman No9 L</family>
 			<family>Bitstream Vera Serif</family>
 			<family>Times New Roman</family>
-			<family>Nimbus Roman No9 L</family>
 			<family>Luxi Serif</family>
 			<family>Times</family>
 			<family>Kochi Mincho</family>
@@ -278,9 +278,9 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
+			<family>Nimbus Sans L</family>
 			<family>Bitstream Vera Sans</family>
 			<family>Verdana</family>
-			<family>Nimbus Sans L</family>
 			<family>Luxi Sans</family>
 			<family>Arial</family>
 			<family>Helvetica</family>
@@ -294,10 +294,10 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
+			<family>Luxi Mono</family>
 			<family>Bitstream Vera Sans Mono</family>
 			<family>Andale Mono</family>
 			<family>Courier New</family>
-			<family>Luxi Mono</family>
 			<family>Nimbus Mono L</family>
 			<family>Kochi Gothic</family>
 			<family>AR PL KaitiM GB</family>
