diff -Naur fontconfig-2.3.2.orig/fonts.conf.in fontconfig-2.3.2/fonts.conf.in
--- fontconfig-2.3.2.orig/fonts.conf.in	2005-04-03 23:07:50.000000000 +0200
+++ fontconfig-2.3.2/fonts.conf.in	2005-11-05 18:17:33.000000000 +0100
@@ -152,7 +152,7 @@
 	</alias>
         <alias>
 	  <family>Courier</family>
-	  <accept><family>Nimbus Mono L</family></accept>
+	  <accept><family>Luxi Mono</family></accept>
 	</alias>
 
 <!--
@@ -161,15 +161,15 @@
  -->
  	<alias>
 		<family>Times New Roman</family>
-		<accept><family>Thorndale AMT</family></accept>
+		<accept><family>Nimbus Roman No9 L</family></accept>
 	</alias>
  	<alias>
 		<family>Arial</family>
-		<accept><family>Albany AMT</family></accept>
+		<accept><family>Nimbus Sans L</family></accept>
 	</alias>
  	<alias>
 		<family>Courier New</family>
-		<accept><family>Cumberland AMT</family></accept>
+		<accept><family>Luxi Mono</family></accept>
 	</alias>
  
 <!--
@@ -258,11 +258,11 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
+			<family>Nimbus Roman No9 L</family>
 			<family>Bitstream Vera Serif</family>
 			<family>Times New Roman</family>
 			<family>Thorndale AMT</family>
 			<family>Luxi Serif</family>
-			<family>Nimbus Roman No9 L</family>
 			<family>Times</family>
 			<family>Frank Ruehl</family>
 			<family>Kochi Mincho</family>
@@ -276,11 +276,11 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
+			<family>Luxi Sans</family>
 			<family>Bitstream Vera Sans</family>
 			<family>Verdana</family>
 			<family>Arial</family>
 			<family>Albany AMT</family>
-			<family>Luxi Sans</family>
 			<family>Nimbus Sans L</family>
 			<family>Helvetica</family>
 			<family>Nachlieli</family>
@@ -296,11 +296,11 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
+			<family>Luxi Mono</family>
 			<family>Bitstream Vera Sans Mono</family>
 			<family>Andale Mono</family>
 			<family>Courier New</family>
 			<family>Cumberland AMT</family>
-			<family>Luxi Mono</family>
 			<family>Nimbus Mono L</family>
 			<family>Courier</family>
 			<family>Miriam Mono</family>
