diff -Naur fontconfig-2.3.2.orig/fonts.conf.in fontconfig-2.3.2/fonts.conf.in
--- fontconfig-2.3.2.orig/fonts.conf.in	2005-04-03 23:07:50.000000000 +0200
+++ fontconfig-2.3.2/fonts.conf.in	2005-07-31 17:53:27.000000000 +0200
@@ -72,11 +72,11 @@
   Serif faces
  -->
 	<alias>
+		<family>Nimbus Roman No9 L</family>
 		<family>Bitstream Vera Serif</family>
 		<family>Times New Roman</family>
 		<family>Thorndale AMT</family>
 		<family>Times</family>
-		<family>Nimbus Roman No9 L</family>
 		<family>Luxi Serif</family>
 		<family>Kochi Mincho</family>
 		<family>AR PL SungtiL GB</family>
@@ -90,12 +90,12 @@
   Sans-serif faces
  -->
 	<alias>
+		<family>Nimbus Sans L</family>
 		<family>Bitstream Vera Sans</family>
 		<family>Helvetica</family>
 		<family>Arial</family>
 		<family>Verdana</family>
 		<family>Albany AMT</family>
-		<family>Nimbus Sans L</family>
 		<family>Luxi Sans</family>
 		<family>Kochi Gothic</family>
 		<family>AR PL KaitiM GB</family>
@@ -110,11 +110,11 @@
   Monospace faces
  -->
  	<alias>
+		<family>Luxi Mono</family>
 		<family>Bitstream Vera Sans Mono</family>
 		<family>Courier</family>
 		<family>Courier New</family>
 		<family>Andale Mono</family>
-		<family>Luxi Mono</family>
 		<family>Cumberland AMT</family>
 		<family>Nimbus Mono L</family>
 		<family>NSimSun</family>
@@ -258,11 +258,11 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
+			<family>Nimbus Roman No9 L</family>
 			<family>Bitstream Vera Serif</family>
 			<family>Times New Roman</family>
 			<family>Thorndale AMT</family>
 			<family>Luxi Serif</family>
-			<family>Nimbus Roman No9 L</family>
 			<family>Times</family>
 			<family>Frank Ruehl</family>
 			<family>Kochi Mincho</family>
@@ -276,12 +276,12 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
+			<family>Nimbus Sans L</family>
 			<family>Bitstream Vera Sans</family>
 			<family>Verdana</family>
 			<family>Arial</family>
 			<family>Albany AMT</family>
 			<family>Luxi Sans</family>
-			<family>Nimbus Sans L</family>
 			<family>Helvetica</family>
 			<family>Nachlieli</family>
 			<family>Kochi Gothic</family>
@@ -296,11 +296,11 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
+			<family>Luxi Mono</family>
 			<family>Bitstream Vera Sans Mono</family>
 			<family>Andale Mono</family>
 			<family>Courier New</family>
 			<family>Cumberland AMT</family>
-			<family>Luxi Mono</family>
 			<family>Nimbus Mono L</family>
 			<family>Courier</family>
 			<family>Miriam Mono</family>
