diff -Naur fontconfig-2.4.0/conf.d/40-generic.conf fontconfig-2.4.0-p/conf.d/40-generic.conf
--- fontconfig-2.4.0/conf.d/40-generic.conf	2006-09-10 05:33:43.000000000 +0200
+++ fontconfig-2.4.0-p/conf.d/40-generic.conf	2006-09-17 13:27:24.000000000 +0200
@@ -10,8 +10,9 @@
   Serif faces
  -->
 	<alias>
-		<family>Bitstream Vera Serif</family>
 		<family>DejaVu Serif</family>
+		<family>Luxi Serif</family>
+		<family>Bitstream Vera Serif</family>
 		<family>Times New Roman</family>
 		<family>Times</family>
 		<family>Nimbus Roman No9 L</family>
@@ -29,8 +30,9 @@
   Sans-serif faces
  -->
 	<alias>
-		<family>Bitstream Vera Sans</family>
 		<family>DejaVu Sans</family>
+		<family>Luxi Sans</family>
+		<family>Bitstream Vera Sans</family>
 		<family>Helvetica</family>
 		<family>Arial</family>
 		<family>Verdana</family>
@@ -51,12 +53,12 @@
   Monospace faces
  -->
  	<alias>
-		<family>Bitstream Vera Sans Mono</family>
 		<family>DejaVu Sans Mono</family>
+		<family>Luxi Mono</family>
+		<family>Bitstream Vera Sans Mono</family>
 		<family>Courier</family>
 		<family>Courier New</family>
 		<family>Andale Mono</family>
-		<family>Luxi Mono</family>
 		<family>Cumberland AMT</family>
 		<family>Nimbus Mono L</family>
 		<family>NSimSun</family>
