diff -Naur fontconfig-2.4.0/conf.d/60-latin.conf fontconfig-2.4.0-p/conf.d/60-latin.conf
--- fontconfig-2.4.0/conf.d/60-latin.conf	2006-09-10 05:30:00.000000000 +0200
+++ fontconfig-2.4.0-p/conf.d/60-latin.conf	2006-09-17 16:48:30.000000000 +0200
@@ -4,11 +4,11 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
-			<family>Bitstream Vera Serif</family>
 			<family>DejaVu Serif</family>
+			<family>Luxi Serif</family>
+			<family>Bitstream Vera Serif</family>
 			<family>Times New Roman</family>
 			<family>Thorndale AMT</family>
-			<family>Luxi Serif</family>
 			<family>Nimbus Roman No9 L</family>
 			<family>Times</family>
 		</prefer>
@@ -16,12 +16,12 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
-			<family>Bitstream Vera Sans</family>
 			<family>DejaVu Sans</family>
+			<family>Luxi Sans</family>
+			<family>Bitstream Vera Sans</family>
 			<family>Verdana</family>
 			<family>Arial</family>
 			<family>Albany AMT</family>
-			<family>Luxi Sans</family>
 			<family>Nimbus Sans L</family>
 			<family>Helvetica</family>
 		</prefer>
@@ -29,12 +29,12 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
-			<family>Bitstream Vera Sans Mono</family>
 			<family>DejaVu Sans Mono</family>
+			<family>Luxi Mono</family>
+			<family>Bitstream Vera Sans Mono</family>
 			<family>Andale Mono</family>
 			<family>Courier New</family>
 			<family>Cumberland AMT</family>
-			<family>Luxi Mono</family>
 			<family>Nimbus Mono L</family>
 			<family>Courier</family>
 		</prefer>
