diff -aur fontconfig-2.8.0/conf.d/60-latin.conf fontconfig-2.8.0-2/conf.d/60-latin.conf
--- fontconfig-2.8.0/conf.d/60-latin.conf	2009-11-16 11:24:52.000000000 -0600
+++ fontconfig-2.8.0-2/conf.d/60-latin.conf	2010-11-12 00:53:55.675993922 -0600
@@ -4,11 +4,12 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
-			<family>Bitstream Vera Serif</family>
 			<family>DejaVu Serif</family>
+			<family>Luxi Serif</family>
+			<family>Bitstream Vera Serif</family>
+			<family>Droid Serif</family>
 			<family>Times New Roman</family>
 			<family>Thorndale AMT</family>
-			<family>Luxi Serif</family>
 			<family>Nimbus Roman No9 L</family>
 			<family>Times</family>
 		</prefer>
@@ -16,12 +17,13 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
-			<family>Bitstream Vera Sans</family>
 			<family>DejaVu Sans</family>
+			<family>Luxi Sans</family>
+			<family>Bitstream Vera Sans</family>
+			<family>Droid Sans</family>
 			<family>Verdana</family>
 			<family>Arial</family>
 			<family>Albany AMT</family>
-			<family>Luxi Sans</family>
 			<family>Nimbus Sans L</family>
 			<family>Helvetica</family>
 			<family>Lucida Sans Unicode</family>
@@ -32,13 +34,14 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
-			<family>Bitstream Vera Sans Mono</family>
 			<family>DejaVu Sans Mono</family>
+			<family>Luxi Mono</family>
+			<family>Bitstream Vera Sans Mono</family>
+			<family>Droid Sans Mono</family>
 			<family>Inconsolata</family>
 			<family>Andale Mono</family>
 			<family>Courier New</family>
 			<family>Cumberland AMT</family>
-			<family>Luxi Mono</family>
 			<family>Nimbus Mono L</family>
 			<family>Courier</family>
 		</prefer>
