diff -Naur fontconfig-2.3.96/src/fcxml.c fontconfig-2.3.96-p/src/fcxml.c
--- fontconfig-2.3.96/src/fcxml.c	2006-09-03 08:12:01.000000000 +0200
+++ fontconfig-2.3.96-p/src/fcxml.c	2006-09-04 05:42:33.971217112 +0200
@@ -545,8 +545,9 @@
 	if ((value == FcTypeLangSet && type == FcTypeString) ||
 	    (value == FcTypeString && type == FcTypeLangSet))
 	    return;
-	FcConfigMessage (parse, FcSevereWarning, "saw %s, expected %s",
+	/* FcConfigMessage (parse, FcSevereWarning, "saw %s, expected %s",
 			 FcTypeName (value), FcTypeName (type));
+	*/
     }
 }
 
