diff -Naur efivar-devel-0.22/Make.defaults efivar-devel-0.22-fix/Make.defaults
--- efivar-devel-0.22/Make.defaults	2016-01-20 19:29:21.000000000 +0100
+++ efivar-devel-0.22-fix/Make.defaults	2016-02-21 13:19:22.987211845 +0100
@@ -25,6 +25,7 @@
 clang_cflags =
 gcc_cflags = -specs=$(TOPDIR)/gcc.specs
 cflags	= $(CFLAGS) -I${TOPDIR}/src/include/efivar/ \
+	-L${TOPDIR}/src/ \
 	$(if $(filter $(CC),clang),$(clang_cflags),) \
 	$(if $(filter $(CC),gcc),$(gcc_cflags),) \
 	$(call pkg-config-cflags)
