diff -Naur digikam-8.3.0/CMakeLists.txt digikam-8.3.0.new/CMakeLists.txt
--- digikam-8.3.0/CMakeLists.txt	2024-03-16 05:40:45.000000000 +0100
+++ digikam-8.3.0.new/CMakeLists.txt	2024-04-18 13:37:43.150498238 +0200
@@ -55,7 +55,7 @@
         LANGUAGES CXX C)
 
 # Settings from ECM
-set(CMAKE_CXX_STANDARD          17)
+set(CMAKE_CXX_STANDARD          20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS        OFF)
 
diff -Naur digikam-8.3.0/core/dplugins/generic/tools/mediaserver/CMakeLists.txt digikam-8.3.0.new/core/dplugins/generic/tools/mediaserver/CMakeLists.txt
--- digikam-8.3.0/core/dplugins/generic/tools/mediaserver/CMakeLists.txt	2024-03-16 05:40:45.000000000 +0100
+++ digikam-8.3.0.new/core/dplugins/generic/tools/mediaserver/CMakeLists.txt	2024-04-18 13:37:50.318551788 +0200
@@ -160,7 +160,9 @@
 file(GLOB_RECURSE upnpsdk_headers ${CMAKE_CURRENT_SOURCE_DIR}/upnpsdk/*.h)
 
 foreach(_file ${upnpsdk_headers})
+
     set_property(SOURCE ${_file} PROPERTY SKIP_AUTOMOC ON)
+
 endforeach()
 
 set(libmediaserver_SRCS
@@ -173,6 +175,7 @@
 
 # For Unit-tests and plugin
 add_library(mediaserverbackend STATIC ${libmediaserver_SRCS})
+set_property(TARGET mediaserverbackend PROPERTY CXX_STANDARD 17)
 
 if(MSVC)
 
@@ -191,12 +194,14 @@
 )
 
 if(WIN32)
+
     target_link_libraries(mediaserverbackend
                           PRIVATE
                           # Defined in RulesWindows.cmake
                           ${WSOCK32_LIBRARY}
                           ${WS2_32_LIBRARY}
     )
+
 endif()
 
 # ---------------------------------------------------------------------------------------------------
diff -Naur digikam-8.3.0/core/tests/mediaserver/CMakeLists.txt digikam-8.3.0.new/core/tests/mediaserver/CMakeLists.txt
--- digikam-8.3.0/core/tests/mediaserver/CMakeLists.txt	2024-03-16 05:40:47.000000000 +0100
+++ digikam-8.3.0.new/core/tests/mediaserver/CMakeLists.txt	2024-04-18 13:37:55.004586794 +0200
@@ -30,6 +30,8 @@
                ${CMAKE_SOURCE_DIR}/core/dplugins/generic/tools/mediaserver/upnpsdk/Platinum/Source/Tests/FileMediaServer/FileMediaServerTest.cpp
 )
 
+set_property(TARGET filemediaserver_cli PROPERTY CXX_STANDARD 17)
+
 target_link_libraries(filemediaserver_cli
 
                       digikamcore
