A quick & dirty hack to access GLists from python.
diff -Naur glib-2.12.3.orig/Makefile.am glib-2.12.3/Makefile.am
--- glib-2.12.3.orig/Makefile.am	2006-03-09 06:31:52.000000000 +0100
+++ glib-2.12.3/Makefile.am	2006-09-28 00:27:13.000000000 +0200
@@ -2,7 +2,8 @@
 
 AUTOMAKE_OPTIONS = 1.7
 
-SUBDIRS = . m4macros glib gobject gmodule gthread tests build po docs
+SUBDIRS = . m4macros glib gobject gmodule gthread tests build po docs \
+	  bindings/python
 
 bin_SCRIPTS = glib-gettextize
 
diff -Naur glib-2.12.3.orig/bindings/glist.i glib-2.12.3/bindings/glist.i
--- glib-2.12.3.orig/bindings/glist.i	1970-01-01 01:00:00.000000000 +0100
+++ glib-2.12.3/bindings/glist.i	2006-09-28 00:19:06.000000000 +0200
@@ -0,0 +1,12 @@
+%module glist
+%nodefaultctor;        /* Don't create default constructors */
+%include "cpointer.i"
+
+/* Create a casting function */
+%pointer_cast(gpointer, char *, gpointer_to_char);
+%pointer_cast(char *, gpointer, char_to_gpointer);
+
+%pointer_cast(int,    GList*, int_to_glist);
+%pointer_cast(int,    guint, int_to_guint);
+%pointer_cast(guint,    int, guint_to_int);
+%include "glist.h"
diff -Naur glib-2.12.3.orig/bindings/python/Makefile glib-2.12.3/bindings/python/Makefile
--- glib-2.12.3.orig/bindings/python/Makefile	1970-01-01 01:00:00.000000000 +0100
+++ glib-2.12.3/bindings/python/Makefile	2006-09-28 00:20:08.000000000 +0200
@@ -0,0 +1,47 @@
+CFLAGS += $(shell python -c 'from distutils import sysconfig; print "-I" + sysconfig.get_python_inc()')
+CFLAGS += -I../../ -I../../glib
+CFLAGS += -DSWIG_PYTHON_SILENT_MEMLEAK
+ifeq ($(shell arch),x86_64)
+CFLAGS += -fPIC
+endif
+LDFLAGS += -L../../glib/.libs -lglib-2.0
+LIBDIR += $(shell python -c 'from distutils import sysconfig; print sysconfig.get_python_lib()')
+
+all: _glist.so glist.pyc
+
+_glist.so: glist_wrap.o
+	$(CC) -shared -Wl,-soname,$@ -o $@ $^ $(LDFLAGS)
+
+glist_wrap.o: glist_wrap.c
+	$(CC) $(CFLAGS) -c -o $@ -include glist.h $^
+
+glist_wrap.c:
+	cp ../../glib/glist.h ./
+	cp ../glist.i ./
+	# no GCrap thx
+	sed -i '/^\(^[^#]*\| \)G_[A-Z]/s/G_[A-Z_]\+//' glist.h
+	swig -python glist.i
+	# strip the unnecessary prefixes
+	sed -i 's/^g_list_//' glist.py
+
+glist.pyc: glist.py
+	python -c "import compileall; compileall.compile_dir('.',1,'.')"
+
+glist.py: glist_wrap.c
+
+install: install-so install-py
+
+install-so: _glist.so
+	mkdir -p $(DESTDIR)$(LIBDIR)
+	install $^ $(DESTDIR)$(LIBDIR)
+
+install-py: glist.py glist.pyc
+	install -m644 $^ $(DESTDIR)$(LIBDIR)
+
+clean:
+	rm -f _glist* glist{.h,.i,.py*,_wrap*}
+
+distclean: clean
+	rm -f Makefile
+
+check:
diff -Naur glib-2.12.3.orig/bindings/python/test.py glib-2.12.3/bindings/python/test.py
--- glib-2.12.3.orig/bindings/python/test.py	1970-01-01 01:00:00.000000000 +0100
+++ glib-2.12.3/bindings/python/test.py	2006-09-28 00:14:58.000000000 +0200
@@ -0,0 +1,10 @@
+#!/usr/bin/env python
+import glist
+
+a = "foo"
+b = "bar"
+l = glist.append(glist.int_to_glist(0), glist.char_to_gpointer(a))
+l = glist.append(l, glist.char_to_gpointer(b))
+print glist.gpointer_to_char(glist.nth_data(l, glist.int_to_guint(0)))
+print glist.guint_to_int(glist.length(l))
+glist.free(l)
