Index: search.c
===================================================================
--- cgi-bin/search.c	(revision 7354)
+++ search.c	(working copy)
@@ -167,7 +167,9 @@
       * string + RE overhead...
       */
 
-      wlen = (sptr - s) + 4 * wlen + 2 * strlen(prefix) + 4;
+      wlen = (sptr - s) + 2 * 4 * wlen + 2 * strlen(prefix) + 11;
+      if (lword)
+        wlen += strlen(lword);
 
       if (wlen > slen)
       {
