--- fglrx.orig/build_mod/firegl_public.c 2005-12-05 15:47:41.000000000 +0000
+++ lib/modules/fglrx/build_mod/firegl_public.c 2005-12-05 17:18:12.000000000 +0000
@@ -2586,7 +2586,7 @@ static __inline__ vm_nopage_ret_t do_vm_

     pMmPage = virt_to_page(kaddr);

-#if 0
+#if LINUX_VERSION_CODE >= 0x02060f
     // WARNING WARNINIG WARNNING WARNNING WARNNING WARNNING WARNNING WARNNING
     // Don't increment page usage count, cause ctx pages are allocated
     // with drm_alloc_pages, which marks all pages as reserved. Reserved
--- lib/modules/fglrx/build_mod/firegl_agpgart/firegl_wrap.c	2005-12-07 18:03:53.000000000 +0100
+++ lib/modules/fglrx/build_mod/firegl_agpgart/firegl_wrap.c.new	2006-01-09 17:20:01.000000000 +0100
@@ -21,6 +21,8 @@
 !!! since it requires changes to linux/init/main.c.
 #endif /* !MODULE */
 
+#define _LINUX_UNISTD_H_
+
 // ============================================================
 #include <linux/version.h>
 #include <linux/autoconf.h>
@@ -1120,7 +1122,7 @@
 
 int ATI_API_CALL __ke_verify_area(int type, const void * addr, unsigned long size)
 {
-    return verify_area(type, addr, size);
+    return access_ok(type,addr,size) ? 0 : -EFAULT;
 }
 
 int ATI_API_CALL __ke_get_pci_device_info(__ke_pci_dev_t* dev, __ke_pci_device_info_t *pinfo)
