--- lib/modules/fglrx/build_mod/firegl_public.c	2005-10-29 12:21:07.000000000 +0200
+++ lib/modules/fglrx/build_mod/firegl_public.c	2005-10-29 12:21:38.000000000 +0200
@@ -1475,7 +1475,7 @@
 
 int ATI_API_CALL __ke_verify_area(int type, const void * addr, unsigned long size)
 {
-    return verify_area(type, addr, size);
+    return access_ok(type, addr, size);
 }
 
 int ATI_API_CALL __ke_get_pci_device_info(__ke_pci_dev_t* dev, __ke_pci_device_info_t *pinfo)
