diff -Naur gwenview-1.3.92/src/gvcore/filterbar.ui gwenview-1.3.92-unknow-headers/src/gvcore/filterbar.ui
--- gwenview-1.3.92/src/gvcore/filterbar.ui	2006-07-14 17:51:21.000000000 +0200
+++ gwenview-1.3.92-unknow-headers/src/gvcore/filterbar.ui	2006-07-14 22:21:50.725072440 +0200
@@ -255,7 +255,4 @@
     <tabstop>mFilterButton</tabstop>
 </tabstops>
 <layoutdefaults spacing="6" margin="11"/>
-<includehints>
-    <includehint>doublespinbox.h</includehint>
-</includehints>
 </UI>
