diff -Naur doxygen-1.4.3.orig/tmake/lib/linux-g++/tmake.conf doxygen-1.4.3/tmake/lib/linux-g++/tmake.conf
--- doxygen-1.4.3.orig/tmake/lib/linux-g++/tmake.conf	2005-07-09 00:45:21.000000000 +0200
+++ doxygen-1.4.3/tmake/lib/linux-g++/tmake.conf	2005-07-09 00:46:46.000000000 +0200
@@ -50,7 +50,7 @@
 TMAKE_LIBS_QT_OPENGL	= -lqgl
 TMAKE_LIBS_OPENGL	= -lMesaGL -lMesaGLU -lXmu
 
-TMAKE_MOC		= moc
+TMAKE_MOC		= $(QTDIR)/bin/moc
 
 TMAKE_AR		= ar cqs
 TMAKE_RANLIB		=
