diff -Naur doxygen-1.4.6.orig/src/latexgen.cpp doxygen-1.4.6/src/latexgen.cpp
--- doxygen-1.4.6.orig/src/latexgen.cpp	2006-01-17 23:47:28.785798750 +0100
+++ doxygen-1.4.6/src/latexgen.cpp	2006-01-17 23:49:08.492030000 +0100
@@ -235,15 +235,15 @@
   }
   if (Config_getBool("PDF_HYPERLINKS")) 
   {
-    t << "\\ifx\\pdfoutput\\undefined" << endl
-      << "\\usepackage[ps2pdf," << endl
+    t << "\\usepackage{ifpdf}" << endl
+      << "\\ifpdf" << endl
+      << "\\usepackage[pdftex," << endl
       << "            pagebackref=true," << endl
       << "            colorlinks=true," << endl
       << "            linkcolor=blue" << endl
       << "           ]{hyperref}" << endl
-      << "\\usepackage{pspicture}" << endl
       << "\\else" << endl
-      << "\\usepackage[pdftex," << endl
+      << "\\usepackage[ps2pdf," << endl
       << "            pagebackref=true," << endl
       << "            colorlinks=true," << endl
       << "            linkcolor=blue" << endl
