diff -Naur doxygen-1.4.3.orig/Makefile.in doxygen-1.4.3/Makefile.in
--- doxygen-1.4.3.orig/Makefile.in	2005-07-09 00:45:21.000000000 +0200
+++ doxygen-1.4.3/Makefile.in	2005-07-09 00:47:31.000000000 +0200
@@ -53,9 +53,9 @@
 	$(INSTTOOL) -m 755 bin/doxygen    $(INSTALL)/bin
 	$(INSTTOOL) -m 755 bin/doxytag    $(INSTALL)/bin
 	$(INSTTOOL) -d $(INSTALL)/$(MAN1DIR)
-	cat doc/doxygen.1    | sed -e "s/DATE/\$(DATE)/g" -e "s/VERSION/$(VERSION)/g" > $(INSTALL)/$(MAN1DIR)/doxygen.1 ; \
-	cat doc/doxytag.1    | sed -e "s/DATE/\$(DATE)/g" -e "s/VERSION/$(VERSION)/g" > $(INSTALL)/$(MAN1DIR)/doxytag.1 ; \
-	cat doc/doxywizard.1 | sed -e "s/DATE/\$(DATE)/g" -e "s/VERSION/$(VERSION)/g" > $(INSTALL)/$(MAN1DIR)/doxywizard.1 ;
+	cat doc/doxygen.1    | sed -e "s/DATE/$(DATE)/g" -e "s/VERSION/$(VERSION)/g" > $(INSTALL)/$(MAN1DIR)/doxygen.1 ; \
+	cat doc/doxytag.1    | sed -e "s/DATE/$(DATE)/g" -e "s/VERSION/$(VERSION)/g" > $(INSTALL)/$(MAN1DIR)/doxytag.1 ; \
+	cat doc/doxywizard.1 | sed -e "s/DATE/$(DATE)/g" -e "s/VERSION/$(VERSION)/g" > $(INSTALL)/$(MAN1DIR)/doxywizard.1 ;
 
 install_docs: 
 	$(INSTTOOL) -d $(DOCDIR)
diff -Naur doxygen-1.4.3.orig/tmake/lib/linux-g++/tmake.conf doxygen-1.4.3/tmake/lib/linux-g++/tmake.conf
--- doxygen-1.4.3.orig/tmake/lib/linux-g++/tmake.conf	2005-07-09 00:45:21.000000000 +0200
+++ doxygen-1.4.3/tmake/lib/linux-g++/tmake.conf	2005-07-09 00:46:46.000000000 +0200
@@ -50,7 +50,7 @@
 TMAKE_LIBS_QT_OPENGL	= -lqgl
 TMAKE_LIBS_OPENGL	= -lMesaGL -lMesaGLU -lXmu
 
-TMAKE_MOC		= moc
+TMAKE_MOC		= $(QTDIR)/bin/moc
 
 TMAKE_AR		= ar cqs
 TMAKE_RANLIB		=
