diff -Naur doxygen-1.4.3.orig/configure doxygen-1.4.3/configure
--- doxygen-1.4.3.orig/configure	2005-07-09 00:42:42.000000000 +0200
+++ doxygen-1.4.3/configure	2005-07-09 00:43:29.000000000 +0200
@@ -466,7 +466,7 @@
 EOF
 fi
 
-f_inmakefiles="Makefile.in qtools/Makefile.in src/Makefile.in examples/Makefile.in doc/Makefile.in addon/doxywizard/Makefile.in addon/doxmlparser/src/Makefile.in addon/doxmlparser/test/Makefile.in addon/doxmlparser/examples/metrics/Makefile.in libpng/Makefile.in libmd5/Makefile.in"
+f_inmakefiles="Makefile.in qtools/Makefile.in src/Makefile.in examples/Makefile.in doc/Makefile.in addon/doxywizard/Makefile.in addon/doxmlparser/src/Makefile.in addon/doxmlparser/test/Makefile.in addon/doxmlparser/examples/metrics/Makefile.in libmd5/Makefile.in"
 
 for i in $f_inmakefiles ; do
      SRC=$i
@@ -483,7 +483,6 @@
        echo "" >> $DST
        echo "all: src/version.cpp " >> $DST
        echo "	\$(MAKE) -C qtools" >> $DST
-       echo "	\$(MAKE) -C libpng" >> $DST
        echo "	\$(MAKE) -C libmd5" >> $DST
        echo "	\$(MAKE) -C src" >> $DST
        if test $f_wizard = YES; then
@@ -500,7 +499,7 @@
      echo "  Created $DST from $SRC..."
 done
 
-f_inprofiles="qtools/qtools.pro.in src/libdoxygen.pro.in src/libdoxycfg.pro.in src/doxygen.pro.in src/doxytag.pro.in addon/doxywizard/doxywizard.pro.in addon/doxmlparser/src/doxmlparser.pro.in addon/doxmlparser/test/xmlparse.pro.in addon/doxmlparser/examples/metrics/metrics.pro.in libpng/libpng.pro.in libmd5/libmd5.pro.in" 
+f_inprofiles="qtools/qtools.pro.in src/libdoxygen.pro.in src/libdoxycfg.pro.in src/doxygen.pro.in src/doxytag.pro.in addon/doxywizard/doxywizard.pro.in addon/doxmlparser/src/doxmlparser.pro.in addon/doxmlparser/test/xmlparse.pro.in addon/doxmlparser/examples/metrics/metrics.pro.in libmd5/libmd5.pro.in" 
 
 for i in $f_inprofiles ; do
      SRC=$i
diff -Naur doxygen-1.4.3.orig/src/doxygen.pro.in doxygen-1.4.3/src/doxygen.pro.in
--- doxygen-1.4.3.orig/src/doxygen.pro.in	2005-07-09 00:42:42.000000000 +0200
+++ doxygen-1.4.3/src/doxygen.pro.in	2005-07-09 00:43:53.000000000 +0200
@@ -28,7 +28,7 @@
 win32:TMAKE_CXXFLAGS       += -DQT_NODLL
 win32-g++:LIBS             = -L../lib -ldoxygen -ldoxycfg -lqtools -lpng -lmd5
 win32-g++:TMAKE_CXXFLAGS   += -fno-exceptions -fno-rtti
-INCLUDEPATH                += ../qtools ../libpng ../libmd5 .
+INCLUDEPATH                += ../qtools ../libmd5 .
 #win32-g++:INCLUDEPATH      -= ../libpng
 DESTDIR                    =  ../bin
 TARGET                     =  doxygen
diff -Naur doxygen-1.4.3.orig/src/libdoxygen.pro.in doxygen-1.4.3/src/libdoxygen.pro.in
--- doxygen-1.4.3.orig/src/libdoxygen.pro.in	2005-07-09 00:42:42.000000000 +0200
+++ doxygen-1.4.3/src/libdoxygen.pro.in	2005-07-09 00:44:31.000000000 +0200
@@ -194,7 +194,6 @@
 win32-msvc:TMAKE_CXXFLAGS  += -Zm200
 win32-g++:TMAKE_CXXFLAGS   += -fno-exceptions -fno-rtti
 INCLUDEPATH                += ../qtools
-INCLUDEPATH                += ../libpng
 INCLUDEPATH                += ../libmd5
 win32:INCLUDEPATH          += .
 win32-g++:INCLUDEPATH      = ../qtools /usr/include/libpng12 ../libmd5
diff -Naur doxygen-1.4.3.orig/src/pngenc.cpp doxygen-1.4.3/src/pngenc.cpp
--- doxygen-1.4.3.orig/src/pngenc.cpp	2005-07-09 00:42:42.000000000 +0200
+++ doxygen-1.4.3/src/pngenc.cpp	2005-07-09 00:44:52.000000000 +0200
@@ -25,7 +25,7 @@
 #endif
 
 #define ALL_STATIC
-#include <../libpng/png.h>
+#include <png.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
