diff -ur a/src/AudioDecoder/ffmpegdecoder.cpp b/src/AudioDecoder/ffmpegdecoder.cpp
--- a/src/AudioDecoder/ffmpegdecoder.cpp	2012-10-28 16:39:15.414128154 -0500
+++ b/src/AudioDecoder/ffmpegdecoder.cpp	2012-10-29 02:52:01.827024300 -0500
@@ -71,7 +71,7 @@
 
     if (m_pFormatContext)
     {
-        av_close_input_file(m_pFormatContext);
+        avformat_close_input(&m_pFormatContext);
         m_pFormatContext = NULL;
     }
 
@@ -83,12 +83,12 @@
     avcodec_register_all();
     av_register_all();
 
-    if (av_open_input_file(&m_pFormatContext, m_Filepath.c_str(), NULL, 0, NULL) != 0)
+    if (avformat_open_input(&m_pFormatContext, m_Filepath.c_str(), NULL, NULL) != 0)
     {
         throw logic_error("Could not open input file: " + m_Filepath);
     }
 
-    if (av_find_stream_info(m_pFormatContext) < 0)
+    if (avformat_find_stream_info(m_pFormatContext,NULL) < 0)
     {
         throw logic_error("Could not find stream information in " + m_Filepath);
     }
@@ -104,7 +104,7 @@
 {
     for(uint32_t i = 0; i < m_pFormatContext->nb_streams; ++i)
     {
-        if (m_pFormatContext->streams[i]->codec->codec_type == CODEC_TYPE_AUDIO)
+        if (m_pFormatContext->streams[i]->codec->codec_type == AVMEDIA_TYPE_AUDIO)
         {
             m_pAudioStream = m_pFormatContext->streams[i];
             m_AudioStream = i;
@@ -309,13 +309,13 @@
 
     switch(m_pAudioCodecContext->sample_fmt)
     {
-    case SAMPLE_FMT_U8:
+    case AV_SAMPLE_FMT_U8:
         format.bits = 8;
         break;
-    case SAMPLE_FMT_S16:
+    case AV_SAMPLE_FMT_S16:
         format.bits = 16;
         break;
-    case SAMPLE_FMT_S32:
+    case AV_SAMPLE_FMT_S32:
         format.bits = 32;
         break;
     default:
