diff -Naur gejengel-0.1.4/src/AudioDecoder/ffmpegdecoder.cpp gejengel-0.1.4.new/src/AudioDecoder/ffmpegdecoder.cpp
--- gejengel-0.1.4/src/AudioDecoder/ffmpegdecoder.cpp	2010-03-19 15:31:43.000000000 +0100
+++ gejengel-0.1.4.new/src/AudioDecoder/ffmpegdecoder.cpp	2015-01-03 10:00:05.062571381 +0100
@@ -71,7 +71,7 @@
 
     if (m_pFormatContext)
     {
-        av_close_input_file(m_pFormatContext);
+        avformat_close_input(&m_pFormatContext);
         m_pFormatContext = NULL;
     }
 
@@ -83,12 +83,12 @@
     avcodec_register_all();
     av_register_all();
 
-    if (av_open_input_file(&m_pFormatContext, m_Filepath.c_str(), NULL, 0, NULL) != 0)
+    if (avformat_open_input(&m_pFormatContext, m_Filepath.c_str(), NULL, NULL) != 0)
     {
         throw logic_error("Could not open input file: " + m_Filepath);
     }
 
-    if (av_find_stream_info(m_pFormatContext) < 0)
+    if (avformat_find_stream_info(m_pFormatContext,NULL) < 0)
     {
         throw logic_error("Could not find stream information in " + m_Filepath);
     }
@@ -104,7 +104,7 @@
 {
     for(uint32_t i = 0; i < m_pFormatContext->nb_streams; ++i)
     {
-        if (m_pFormatContext->streams[i]->codec->codec_type == CODEC_TYPE_AUDIO)
+        if (m_pFormatContext->streams[i]->codec->codec_type == AVMEDIA_TYPE_AUDIO)
         {
             m_pAudioStream = m_pFormatContext->streams[i];
             m_AudioStream = i;
@@ -134,7 +134,7 @@
     m_pFormatContext->flags |= AVFMT_FLAG_GENPTS;
     m_pFormatContext->streams[m_AudioStream]->discard = AVDISCARD_DEFAULT;
 
-    if (avcodec_open(m_pAudioCodecContext, m_pAudioCodec) < 0)
+    if (avcodec_open2(m_pAudioCodecContext, m_pAudioCodec, NULL) < 0)
     {
         throw logic_error("Could not open audio codec for " + m_Filepath);
     }
@@ -309,13 +309,13 @@
 
     switch(m_pAudioCodecContext->sample_fmt)
     {
-    case SAMPLE_FMT_U8:
+    case AV_SAMPLE_FMT_U8:
         format.bits = 8;
         break;
-    case SAMPLE_FMT_S16:
+    case AV_SAMPLE_FMT_S16:
         format.bits = 16;
         break;
-    case SAMPLE_FMT_S32:
+    case AV_SAMPLE_FMT_S32:
         format.bits = 32;
         break;
     default:
