--- criu-3.19/crit/Makefile	2023-11-28 01:47:16.000000000 +0100
+++ criu-3.19.new/crit/Makefile	2023-12-04 10:18:39.751972173 +0100
@@ -10,31 +10,7 @@
 	$(Q) echo "__version__ = '${CRIU_VERSION}'" > $@
 
 install: ${VERSION_FILE}
-ifeq ($(PYTHON_EXTERNALLY_MANAGED),1)
-ifeq ($(PIP_BREAK_SYSTEM_PACKAGES),0)
-	$(E) "  SKIP INSTALL crit: Externally managed python environment (See PEP 668 for more information)"
-	$(E) "  Consider using PIP_BREAK_SYSTEM_PACKAGES=1 make install"
-else
-	$(E) "  INSTALL " crit
-	$(Q) $(PYTHON) -m pip install --upgrade --ignore-installed --prefix=$(DESTDIR)$(PREFIX) ./crit
-endif
-else
-	$(E) "  INSTALL " crit
-	$(Q) $(PYTHON) -m pip install --upgrade --ignore-installed --prefix=$(DESTDIR)$(PREFIX) ./crit
-endif
 .PHONY: install
 
 uninstall:
-ifeq ($(PYTHON_EXTERNALLY_MANAGED),1)
-ifeq ($(PIP_BREAK_SYSTEM_PACKAGES),0)
-	$(E) " SKIP UNINSTALL crit: Externally managed python environment (See PEP 668 for more information)"
-	$(E) " Consider using PIP_BREAK_SYSTEM_PACKAGES=1 make uninstall"
-else
-	$(E) " UNINSTALL" crit
-	$(Q) $(PYTHON) ./scripts/uninstall_module.py --prefix=$(DESTDIR)$(PREFIX) crit
-endif
-else
-	$(E) " UNINSTALL" crit
-	$(Q) $(PYTHON) ./scripts/uninstall_module.py --prefix=$(DESTDIR)$(PREFIX) crit
-endif
 .PHONY: uninstall
--- criu-3.19/lib/Makefile	2023-11-28 01:47:16.000000000 +0100
+++ criu-3.19.new/lib/Makefile	2023-12-04 10:15:56.784703410 +0100
@@ -57,18 +57,6 @@
 	$(Q) mkdir -p $(DESTDIR)$(LIBDIR)/pkgconfig
 	$(Q) sed -e 's,@version@,$(CRIU_VERSION),' -e 's,@libdir@,$(LIBDIR),' -e 's,@includedir@,$(dir $(INCLUDEDIR)/criu/),' lib/c/criu.pc.in > lib/c/criu.pc
 	$(Q) install -m 644 lib/c/criu.pc $(DESTDIR)$(LIBDIR)/pkgconfig
-ifeq ($(PYTHON_EXTERNALLY_MANAGED),1)
-ifeq ($(PIP_BREAK_SYSTEM_PACKAGES),0)
-	$(E) "  SKIP INSTALL pycriu: Externally managed python environment (See PEP 668 for more information)"
-	$(E) "  Consider using PIP_BREAK_SYSTEM_PACKAGES=1 make install"
-else
-	$(E) "  INSTALL " pycriu
-	$(Q) $(PYTHON) -m pip install --upgrade --ignore-installed --prefix=$(DESTDIR)$(PREFIX) ./lib
-endif
-else
-	$(E) "  INSTALL " pycriu
-	$(Q) $(PYTHON) -m pip install --upgrade --ignore-installed --prefix=$(DESTDIR)$(PREFIX) ./lib
-endif
 .PHONY: install
 
 uninstall:
@@ -80,16 +68,4 @@
 	$(Q) $(RM) $(addprefix $(DESTDIR)$(INCLUDEDIR)/criu/,$(notdir $(UAPI_HEADERS)))
 	$(E) " UNINSTALL" pkgconfig/criu.pc
 	$(Q) $(RM) $(addprefix $(DESTDIR)$(LIBDIR)/pkgconfig/,criu.pc)
-ifeq ($(PYTHON_EXTERNALLY_MANAGED),1)
-ifeq ($(PIP_BREAK_SYSTEM_PACKAGES),0)
-	$(E) " SKIP UNINSTALL pycriu: Externally managed python environment (See PEP 668 for more information)"
-	$(E) " Consider using PIP_BREAK_SYSTEM_PACKAGES=1 make uninstall"
-else
-	$(E) " UNINSTALL" pycriu
-	$(Q) $(PYTHON) ./scripts/uninstall_module.py --prefix=$(DESTDIR)$(PREFIX) pycriu
-endif
-else
-	$(E) " UNINSTALL" pycriu
-	$(Q) $(PYTHON) ./scripts/uninstall_module.py --prefix=$(DESTDIR)$(PREFIX) pycriu
-endif
 .PHONY: uninstall
