diff -up libofa-0.9.3/examples/example.cpp.gcc43 libofa-0.9.3/examples/example.cpp
--- libofa-0.9.3/examples/example.cpp.gcc43	2008-02-12 16:32:04.000000000 -0600
+++ libofa-0.9.3/examples/example.cpp	2008-02-12 16:36:52.000000000 -0600
@@ -7,6 +7,7 @@
 
 -------------------------------------------------------------------*/
 
+#include <string.h>
 #include "protocol.h"
 
 AudioData* loadWaveFile(char *file);
diff -up libofa-0.9.3/examples/protocol.cpp.gcc43 libofa-0.9.3/examples/protocol.cpp
--- libofa-0.9.3/examples/protocol.cpp.gcc43	2008-02-12 16:32:04.000000000 -0600
+++ libofa-0.9.3/examples/protocol.cpp	2008-02-12 16:45:35.000000000 -0600
@@ -8,7 +8,7 @@
 -------------------------------------------------------------------*/
 #include <stdio.h>
 #include <stdlib.h>
-#include <string>
+#include <string.h>
 #include <map>
 #include <expat.h>
 #include <curl/curl.h>
diff -up libofa-0.9.3/lib/signal_op.h.gcc43 libofa-0.9.3/lib/signal_op.h
diff -up libofa-0.9.3/lib/signal_op.cpp.gcc43 libofa-0.9.3/lib/signal_op.cpp
--- libofa-0.9.3/lib/signal_op.cpp.gcc43	2008-02-12 16:32:04.000000000 -0600
+++ libofa-0.9.3/lib/signal_op.cpp	2008-02-12 16:33:22.000000000 -0600
@@ -13,6 +13,7 @@
 
 
 #include <math.h>
+#include <stdlib.h>
 #include "signal_op.h"
 #include "AFLIB/aflibConverter.h"
 #include "error_op.h"
