diff -ur a/libmultipath/blacklist.c b/libmultipath/blacklist.c
--- a/libmultipath/blacklist.c	2014-02-14 08:36:00.561065505 -0600
+++ b/libmultipath/blacklist.c	2014-02-14 09:21:53.541963148 -0600
@@ -182,6 +182,13 @@
 	if (store_ble(conf->elist_property, str, ORIGIN_DEFAULT))
 		return 1;
 
+	str = STRDUP("^cciss!c[0-9]d[0-9]*");
+		if (!str)
+			return 1;
+		if (store_ble(conf->blist_devnode, str, ORIGIN_DEFAULT))
+			return 1;
+
+
 	vector_foreach_slot (conf->hwtable, hwe, i) {
 		if (hwe->bl_product) {
 			if (_blacklist_device(conf->blist_device, hwe->vendor,
diff -ur a/libmultipath/prioritizers/hds.c b/libmultipath/prioritizers/hds.c
--- a/libmultipath/prioritizers/hds.c	2014-02-14 08:36:00.563065505 -0600
+++ b/libmultipath/prioritizers/hds.c	2014-02-14 09:21:44.228963495 -0600
@@ -125,7 +125,7 @@
 		return -1;
 	}
 
-	snprintf (vendor, 9, "%.8s", inqBuffp + 8);
+	snprintf (vendor, 8, "%.8s", inqBuffp + 8);
 	snprintf (product, 17, "%.16s", inqBuffp + 16);
 	snprintf (serial, 5, "%.4s", inqBuffp + 40);
 	snprintf (ldev, 5, "%.4s", inqBuffp + 44);
