--- multipath-tools/libdmmp/libdmmp_private.h	2020-04-19 13:59:04.000015801 +0200
+++ multipath-tools.new/libdmmp/libdmmp_private.h	2020-04-19 14:02:16.498324978 +0200
@@ -38,6 +38,20 @@
 extern "C" {
 #endif
 
+/* json-c 0.13.99 does not define TRUE/FALSE anymore
+* the json-c maintainers replaced them with pure 1/0
+* https://github.com/json-c/json-c/commit/0992aac61f8b
+*/
+#if defined JSON_C_VERSION_NUM && JSON_C_VERSION_NUM >= ((13 << 8) | 99)
+#ifndef FALSE
+#define FALSE 0
+#endif
+#ifndef TRUE
+#define TRUE  1
+#endif
+#endif
+
+
 #define _good(rc, rc_val, out) \
 	do { \
 		rc_val = rc; \
