diff -ur a/kpartx/Makefile b/kpartx/Makefile
--- a/kpartx/Makefile	2014-02-14 08:36:00.560065505 -0600
+++ b/kpartx/Makefile	2014-02-15 05:48:25.746014932 -0600
@@ -28,8 +28,8 @@
 	$(INSTALL_PROGRAM) -m 755 $(EXEC) $(DESTDIR)$(bindir)
 	$(INSTALL_PROGRAM) -d $(DESTDIR)$(libudevdir)
 	$(INSTALL_PROGRAM) -m 755 kpartx_id $(DESTDIR)$(libudevdir)
-	$(INSTALL_PROGRAM) -d $(DESTDIR)/etc/udev/rules.d
-	$(INSTALL_PROGRAM) -m 644 kpartx.rules $(DESTDIR)/etc/udev/rules.d/
+	$(INSTALL_PROGRAM) -d $(DESTDIR)/lib/udev/rules.d
+	$(INSTALL_PROGRAM) -m 644 kpartx.rules $(DESTDIR)/lib/udev/rules.d/66-kpartx.rules
 	$(INSTALL_PROGRAM) -d $(DESTDIR)$(mandir)
 	$(INSTALL_PROGRAM) -m 644 $(EXEC).8.gz $(DESTDIR)$(mandir)
 
diff -ur a/libmpathpersist/Makefile b/libmpathpersist/Makefile
--- a/libmpathpersist/Makefile	2014-02-14 08:36:00.558065506 -0600
+++ b/libmpathpersist/Makefile	2014-02-15 05:48:06.435015650 -0600
@@ -31,8 +31,7 @@
 	$(INSTALL_PROGRAM) -m 755 -d $(DESTDIR)$(syslibdir)
 	$(INSTALL_PROGRAM) -m 755 -d $(DESTDIR)$(man3dir)
 	$(INSTALL_PROGRAM) -m 755 -d $(DESTDIR)/usr/include/
-	$(INSTALL_PROGRAM) -m 755 -d $(DESTDIR)/usr/share/doc/mpathpersist/
-	ln -sf $(DESTDIR)$(syslibdir)/$(LIBS) $(DESTDIR)$(syslibdir)/$(DEVLIB)
+	ln -sf $(LIBS) $(DESTDIR)$(syslibdir)/$(DEVLIB)
 	install -m 644 mpath_persistent_reserve_in.3.gz $(DESTDIR)$(man3dir)	
 	install -m 644 mpath_persistent_reserve_out.3.gz $(DESTDIR)$(man3dir)	
 	install -m 644 mpath_persist.h $(DESTDIR)/usr/include/
diff -ur a/libmultipath/Makefile b/libmultipath/Makefile
--- a/libmultipath/Makefile	2014-02-14 08:36:00.562065505 -0600
+++ b/libmultipath/Makefile	2014-02-15 05:48:06.435015650 -0600
@@ -52,6 +52,7 @@
 	$(INSTALL_PROGRAM) -d $(DESTDIR)$(syslibdir)
 	$(INSTALL_PROGRAM) -m 755 $(LIBS) $(DESTDIR)$(syslibdir)/$(LIBS)
 	$(INSTALL_PROGRAM) -m 755 -d $(DESTDIR)$(libdir)
+	ln -sf $(LIBS) $(DESTDIR)$(syslibdir)/$(DEVLIB)
 
 uninstall:
 	rm -f $(DESTDIR)$(syslibdir)/$(LIBS)
diff -ur a/Makefile.inc b/Makefile.inc
--- a/Makefile.inc	2014-02-14 08:36:00.563065505 -0600
+++ b/Makefile.inc	2014-02-15 05:48:06.435015650 -0600
@@ -14,11 +14,7 @@
 endif
 
 ifndef LIB
-	ifeq ($(shell test -d /lib64 && echo 1),1)
-		LIB=lib64
-	else
-		LIB=lib
-	endif
+	LIB=lib
 endif
 
 ifndef SYSTEMD
@@ -38,7 +34,7 @@
 rcdir	    = $(prefix)/etc/init.d
 syslibdir   = $(prefix)/$(LIB)
 libdir	    = $(prefix)/$(LIB)/multipath
-unitdir     = $(prefix)/usr/lib/systemd/system
+unitdir     = $(prefix)/lib/systemd/system
 mpathpersistdir = $(TOPDIR)/libmpathpersist
 
 GZIP        = gzip -9 -c
