diff -Naur xmoto-0.4.2-orig/src/xmscene/BikePlayer.cpp xmoto-0.4.2/src/xmscene/BikePlayer.cpp
--- xmoto-0.4.2-orig/src/xmscene/BikePlayer.cpp	2008-03-14 08:36:14.000000000 +0100
+++ xmoto-0.4.2/src/xmscene/BikePlayer.cpp	2008-09-26 18:17:26.000000000 +0200
@@ -146,6 +146,7 @@
   m_bFirstPhysicsUpdate = true;
 
   /* Setup ODE */
+  dInitODE();
   m_WorldID = dWorldCreate();
   dWorldSetERP(m_WorldID,PHYS_WORLD_ERP);
   dWorldSetCFM(m_WorldID,PHYS_WORLD_CFM);
