diff -urp xmoto-0.5.1.orig/src/XMArgs.cpp xmoto-0.5.1/src/XMArgs.cpp
--- xmoto-0.5.1.orig/src/XMArgs.cpp	2009-05-31 02:43:52.000000000 +0400
+++ xmoto-0.5.1/src/XMArgs.cpp	2009-05-31 02:45:14.000000000 +0400
@@ -24,6 +24,7 @@ Foundation, Inc., 59 Temple Place, Suite
 #include "VFileIO.h"
 #include <sstream>
 #include <stdlib.h>
+#include <stdio.h>
 
 XMArguments::XMArguments() {
   m_opt_pack   	       = false;
diff -urp xmoto-0.5.1.orig/src/helpers/Log.cpp xmoto-0.5.1/src/helpers/Log.cpp
--- xmoto-0.5.1.orig/src/helpers/Log.cpp	2009-05-31 02:43:52.000000000 +0400
+++ xmoto-0.5.1/src/helpers/Log.cpp	2009-05-31 02:47:51.000000000 +0400
@@ -21,8 +21,9 @@ Foundation, Inc., 59 Temple Place, Suite
 #include "Log.h"
 #include "VExcept.h"
 #include "../VFileIO.h"
-#include "assert.h"
+#include <assert.h>
 #include <stdarg.h>
+#include <stdio.h>
 
 bool  Logger::m_isInitialized = false;
 bool  Logger::m_activ         = true;
diff -urp xmoto-0.5.1.orig/src/helpers/Text.cpp xmoto-0.5.1/src/helpers/Text.cpp
--- xmoto-0.5.1.orig/src/helpers/Text.cpp	2009-05-31 02:43:52.000000000 +0400
+++ xmoto-0.5.1/src/helpers/Text.cpp	2009-05-31 02:46:21.000000000 +0400
@@ -19,6 +19,7 @@ Foundation, Inc., 59 Temple Place, Suite
 =============================================================================*/
 
 #include "Text.h"
+#include <stdio.h>
 
 std::string txtToLower(const std::string& i_str) {
     std::string v_res;
