diff -Naur tar-1.16.orig/src/buffer.c tar-1.16/src/buffer.c
--- tar-1.16.orig/src/buffer.c	2006-10-16 08:33:55.000000000 +0200
+++ tar-1.16/src/buffer.c	2006-10-22 00:52:49.000000000 +0200
@@ -196,7 +196,8 @@
   ct_none,
   ct_compress,
   ct_gzip,
-  ct_bzip2
+  ct_bzip2,
+  ct_lzma
 };
 
 struct zip_magic
@@ -213,6 +214,7 @@
   { ct_compress, 2, "\037\235", "compress", "-Z" },
   { ct_gzip,     2, "\037\213", "gzip", "-z"  },
   { ct_bzip2,    3, "BZh",      "bzip2", "-j" },
+  { ct_lzma,     6, "\xFFLZMA\x00", "lzma", "-Y" },
 };
 
 #define NMAGIC (sizeof(magic)/sizeof(magic[0]))
diff -Naur tar-1.16.orig/src/tar.c tar-1.16/src/tar.c
--- tar-1.16.orig/src/tar.c	2006-10-17 10:11:31.000000000 +0200
+++ tar-1.16/src/tar.c	2006-10-22 00:54:13.000000000 +0200
@@ -568,6 +568,8 @@
    N_("control pax keywords"), GRID+8 },
   {"label", 'V', N_("TEXT"), 0,
    N_("create archive with volume name TEXT; at list/extract time, use TEXT as a globbing pattern for volume name"), GRID+8 },
+  {"lzma", 'Y', 0, 0,
+   N_("filter the archive through lzma using 'lzmash' wrapper script"), 68 },
   {"bzip2", 'j', 0, 0,
    N_("filter the archive through bzip2"), GRID+8 },
   {"gzip", 'z', 0, 0,
@@ -1423,6 +1425,10 @@
 	}
       break;
 
+    case 'Y':
+      set_use_compress_program_option ("lzma");
+      break;
+
     case 'z':
       set_use_compress_program_option ("gzip");
       break;
