diff -Naur grub-2.02/util/grub-mkimagexx.c grub-2.02-p/util/grub-mkimagexx.c
--- grub-2.02/util/grub-mkimagexx.c	2017-04-24 11:16:00.000000000 +0200
+++ grub-2.02-p/util/grub-mkimagexx.c	2019-05-25 13:46:46.002828369 +0200
@@ -832,6 +832,7 @@
 		  break;
 
 		case R_X86_64_PC32:
+		case R_X86_64_PLT32:
 		  {
 		    grub_uint32_t *t32 = (grub_uint32_t *) target;
 		    *t32 = grub_host_to_target64 (grub_target_to_host32 (*t32)
diff -Naur grub-2.02/util/grub-module-verifier.c grub-2.02-p/util/grub-module-verifier.c
--- grub-2.02/util/grub-module-verifier.c	2017-04-24 11:16:00.000000000 +0200
+++ grub-2.02-p/util/grub-module-verifier.c	2019-05-25 13:48:09.933372320 +0200
@@ -19,6 +19,7 @@
       -1
     }, (int[]){
       R_X86_64_PC32,
+      R_X86_64_PLT32,
       -1
     }
   },
