diff -ur a/conf/Makefile.common b/conf/Makefile.common
--- a/conf/Makefile.common	2011-04-08 06:45:13.000000000 -0500
+++ b/conf/Makefile.common	2012-07-18 16:15:20.039713751 -0500
@@ -11,10 +11,10 @@
   CFLAGS_PLATFORM += -mrtd -mregparm=3
 endif
 if COND_i386_efi
-  LDFLAGS_PLATFORM = -melf_i386
+  LDFLAGS_PLATFORM = -Wl,-m,elf_i386
 endif
 if COND_x86_64_efi
-  LDFLAGS_PLATFORM = -melf_x86_64
+  LDFLAGS_PLATFORM = -Wl,-m,elf_x86_64
 endif
 if COND_i386_qemu
   CFLAGS_PLATFORM += -mrtd -mregparm=3
diff -ur a/configure b/configure
--- a/configure	2011-05-14 15:57:49.000000000 -0500
+++ b/configure	2012-07-18 15:05:43.124717487 -0500
@@ -3806,7 +3806,7 @@
 
 # Optimization flag.  Allow user to override.
 if test "x$TARGET_CFLAGS" = x; then
-  TARGET_CFLAGS="$TARGET_CFLAGS -Os"
+  TARGET_CFLAGS="$TARGET_CFLAGS -Os -fno-strict-aliasing"
 fi
 
 # Default HOST_CPPFLAGS
diff -ur a/grub-core/commands/usbtest.c b/grub-core/commands/usbtest.c
--- a/grub-core/commands/usbtest.c	2011-04-11 14:51:24.000000000 -0500
+++ b/grub-core/commands/usbtest.c	2012-07-18 14:10:58.998993357 -0500
@@ -111,7 +111,7 @@
 static void
 usb_print_str (const char *description, grub_usb_device_t dev, int idx)
 {
-  char *name;
+  char *name = 0;
   grub_usb_err_t err;
   /* XXX: LANGID  */
 
diff -ur a/grub-core/Makefile.in b/grub-core/Makefile.in
--- a/grub-core/Makefile.in	2011-05-14 15:58:05.000000000 -0500
+++ b/grub-core/Makefile.in	2012-07-18 16:15:33.139398613 -0500
@@ -23961,9 +23961,9 @@
 CFLAGS_PLATFORM = $(am__append_1) $(am__append_2) $(am__append_3) \
 	$(am__append_4) $(am__append_5) $(am__append_6) \
 	$(am__append_7)
-@COND_i386_efi_TRUE@LDFLAGS_PLATFORM = -melf_i386
+@COND_i386_efi_TRUE@LDFLAGS_PLATFORM = -Wl,-m,elf_i386
 @COND_sparc64_ieee1275_TRUE@LDFLAGS_PLATFORM = -melf64_sparc -mno-relax
-@COND_x86_64_efi_TRUE@LDFLAGS_PLATFORM = -melf_x86_64
+@COND_x86_64_efi_TRUE@LDFLAGS_PLATFORM = -Wl,-m,elf_x86_64
 @COND_mips_yeeloong_TRUE@CPPFLAGS_PLATFORM = -DUSE_ASCII_FAILBACK
 @COND_mips_yeeloong_TRUE@CCASFLAGS_PLATFORM = -march=mips3
 
diff -ur a/Makefile.in b/Makefile.in
--- a/Makefile.in	2011-05-14 15:58:08.000000000 -0500
+++ b/Makefile.in	2012-07-18 16:15:28.386512952 -0500
@@ -7638,9 +7638,9 @@
 CFLAGS_PLATFORM = $(am__append_1) $(am__append_2) $(am__append_3) \
 	$(am__append_4) $(am__append_5) $(am__append_6) \
 	$(am__append_7)
-@COND_i386_efi_TRUE@LDFLAGS_PLATFORM = -melf_i386
+@COND_i386_efi_TRUE@LDFLAGS_PLATFORM = -Wl,-m,elf_i386
 @COND_sparc64_ieee1275_TRUE@LDFLAGS_PLATFORM = -melf64_sparc -mno-relax
-@COND_x86_64_efi_TRUE@LDFLAGS_PLATFORM = -melf_x86_64
+@COND_x86_64_efi_TRUE@LDFLAGS_PLATFORM = -Wl,-m,elf_x86_64
 @COND_mips_yeeloong_TRUE@CPPFLAGS_PLATFORM = -DUSE_ASCII_FAILBACK
 @COND_mips_yeeloong_TRUE@CCASFLAGS_PLATFORM = -march=mips3
 
diff -ur a/util/bash-completion.d/grub-completion.bash.in b/util/bash-completion.d/grub-completion.bash.in
--- a/util/bash-completion.d/grub-completion.bash.in	2010-12-01 08:45:43.000000000 -0600
+++ b/util/bash-completion.d/grub-completion.bash.in	2012-07-18 18:53:21.252635046 -0500
@@ -402,7 +402,7 @@
 #
 # grub-mkpasswd-pbkdf2
 #
-_grub_mkpasswd-pbkdf2 () {
+_grub_mkpasswd_pbkdf2 () {
     local cur
 
     COMPREPLY=()
@@ -417,7 +417,7 @@
 }
 __grub_mkpasswd_pbkdf2_program=$( echo grub-mkpasswd-pbkdf2 | sed "@program_transform_name@" )
 have ${__grub_mkpasswd_pbkdf2_program} && \
- complete -F _grub_mkpasswd-pbkdf2 -o filenames ${__grub_mkpasswd_pbkdf2_program}
+ complete -F _grub_mkpasswd_pbkdf2 -o filenames ${__grub_mkpasswd_pbkdf2_program}
 unset __grub_mkpasswd_pbkdf2_program
 
 
@@ -462,7 +462,7 @@
 #
 # grub-script-check
 #
-_grub_script-check () {
+_grub_script_check () {
     local cur
 
     COMPREPLY=()
@@ -477,7 +477,7 @@
 }
 __grub_script_check_program=$( echo grub-script-check | sed "@program_transform_name@" )
 have ${__grub_script_check_program} && \
- complete -F _grub_script-check -o filenames ${__grub_script_check_program}
+ complete -F _grub_script_check -o filenames ${__grub_script_check_program}
 
 
 # Local variables:
