diff -Naur grub-2.02/util/grub.d/10_linux.in grub-2.02-p/util/grub.d/10_linux.in
--- grub-2.02/util/grub.d/10_linux.in	2018-10-16 19:38:34.846910007 +0200
+++ grub-2.02-p/util/grub.d/10_linux.in	2018-10-16 19:42:38.390426437 +0200
@@ -63,6 +63,9 @@
 	if [ "x${rootsubvol}" != x ]; then
 	    GRUB_CMDLINE_LINUX="rootflags=subvol=${rootsubvol} ${GRUB_CMDLINE_LINUX}"
 	fi;;
+    xf2fs)
+	GRUB_SAVEDEFAULT=false
+	;;
     xzfs)
 	rpool=`${grub_probe} --device ${GRUB_DEVICE} --target=fs_label 2>/dev/null || true`
 	bootfs="`make_system_path_relative_to_its_root / | sed -e "s,@$,,"`"
