diff -ur a/util/grub-mkrescue.in b/util/grub-mkrescue.in
--- a/util/grub-mkrescue.in	2012-06-14 12:25:07.000000000 -0500
+++ b/util/grub-mkrescue.in	2013-03-06 17:53:03.714570600 -0600
@@ -332,7 +332,7 @@
     mformat -C -f 2880 -L 16 -i "${iso9660_dir}"/efi.img ::
     mcopy -s -i "${iso9660_dir}"/efi.img ${efi_dir}/efi ::/
     rm -rf ${efi_dir}
-    grub_mkisofs_arguments="${grub_mkisofs_arguments} --efi-boot efi.img"
+    grub_mkisofs_arguments="${grub_mkisofs_arguments} --efi-boot efi.img -append_partition 2 0xEF ${iso9660_dir}/efi.img"
 fi
 
 make_image "${mipsel_qemu_dir}" mipsel-qemu_mips-elf "${iso9660_dir}/boot/mipsel-qemu_mips.elf" "pata"
