diff -Naur grub-2.02/util/grub.d/10_linux.in grub-2.02-p/util/grub.d/10_linux.in
--- grub-2.02/util/grub.d/10_linux.in	2018-04-10 19:42:11.539930309 +0200
+++ grub-2.02-p/util/grub.d/10_linux.in	2018-04-10 19:40:41.758567549 +0200
@@ -56,6 +56,8 @@
 
 case x"$GRUB_FS" in
     xbtrfs)
+	## does not work with this FS
+	GRUB_SAVEDEFAULT=false
 	rootsubvol="`make_system_path_relative_to_its_root /`"
 	rootsubvol="${rootsubvol#/}"
 	if [ "x${rootsubvol}" != x ]; then
@@ -79,6 +81,19 @@
 
 title_correction_code=
 
+_resume=""
+
+while read line; do
+	_device=$(echo $line | cut -f 1 -d ' ')
+	_type=$(echo $line | cut -f 2 -d ' ')
+	[ "$_type" != "partition" ] && continue
+	_uuid=$(/sbin/blkid -s UUID -o value $_device)
+	_resume="resume=UUID=$_uuid"
+	break
+done << EOF
+$(tail -n +2 /proc/swaps)
+EOF
+
 linux_entry ()
 {
   os="$1"
@@ -104,22 +119,31 @@
   if [ -z "$boot_device_id" ]; then
       boot_device_id="$(grub_get_device_id "${GRUB_DEVICE}")"
   fi
-  if [ x$type != xsimple ] ; then
-      case $type in
-	  recovery)
-	      title="$(printf "$(mktitle) (recovery mode)" "${version}")" ;;
+
+  case $type in
+	simple)
+	     title="$(gettext_printf "%s, with Linux %s" "${os}" "${version}")" ;;
+	 console)
+	      title_c="(console mode)"
+	      title_="$(gettext_printf "%s, with Linux %s" "${os}" "${version}")"
+	      title="${title_} ${title_c}" ;;
+	 recovery)
+   	      title_r="(recovery mode)"
+              title_="$(gettext_printf "%s, with Linux %s" "${os}" "${version}")"
+              title="${title_} ${title_r}" ;;
 	  *)
 	      title="$(printf "$(mktitle)" "${version}")" ;;
-      esac
+   esac
       if [ x"$title" = x"$GRUB_ACTUAL_DEFAULT" ] || [ x"Previous Linux versions>$title" = x"$GRUB_ACTUAL_DEFAULT" ]; then
 	  replacement_title="$(echo "Advanced options for ${OS}" | sed 's,>,>>,g')>$(echo "$title" | sed 's,>,>>,g')"
 	  quoted="$(echo "$GRUB_ACTUAL_DEFAULT" | grub_quote)"
 	  title_correction_code="${title_correction_code}if [ \"x\$default\" = '$quoted' ]; then default='$(echo "$replacement_title" | grub_quote)'; fi;"
       fi
-      echo "menuentry '$(echo "$title" | grub_quote)' ${CLASS} \$menuentry_id_option 'gnulinux-$version-$type-$boot_device_id' {" | sed "s/^/$submenu_indentation/"
+  if [ x$type != xsimple ] ; then
+ 	echo "menuentry '$(echo "$title" | grub_quote)' ${CLASS} \$menuentry_id_option 'gnulinux-$version-$type-$boot_device_id' {" | sed "s/^/$submenu_indentation/"
   else
-      echo "menuentry '$(echo "$os" | grub_quote)' ${CLASS} \$menuentry_id_option 'gnulinux-simple-$boot_device_id' {" | sed "s/^/$submenu_indentation/"
-  fi      
+      echo "menuentry '$(echo "$title" | grub_quote)' ${CLASS} \$menuentry_id_option 'gnulinux-simple-$boot_device_id' {" | sed "s/^/$submenu_indentation/"
+  fi
   if [ x$type != xrecovery ] ; then
       save_default_entry | grub_add_tab
   fi
@@ -154,11 +178,11 @@
   fi
   if [ -d /sys/firmware/efi ]; then
     sed "s/^/$submenu_indentation/" << EOF
-	${linuxefi} ${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args}
+	${linuxefi} ${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args} ${_resume}
 EOF
   else
     sed "s/^/$submenu_indentation/" << EOF
-	linux${sixteenbit} ${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args}
+	linux${sixteenbit} ${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args} ${_resume}
 EOF
   fi
   if test -n "${initrd}" ; then
@@ -225,8 +249,8 @@
   linux_root_device_thisversion="${LINUX_ROOT_DEVICE}"
 
   initrd=
-  for i in "initrd.img-${version}" "initrd-${version}.img" "initrd-${version}.gz" \
-	   "initrd-${version}" "initramfs-${version}.img" \
+  for i in "initramfs-${version}.img" "initrd.img-${version}" \
+	   "initrd-${version}.img" "initrd-${version}.gz" "initrd-${version}"  \
 	   "initrd.img-${alt_version}" "initrd-${alt_version}.img" \
 	   "initrd-${alt_version}" "initramfs-${alt_version}.img" \
 	   "initramfs-genkernel-${version}" \
@@ -272,27 +296,40 @@
     GRUB_DISABLE_SUBMENU="true"
   fi
 
+  ## ok this whole submenu logic is broken we use just menu like
+  ## menu " Frugalware with Linux some_version
+  ## menu " Frugalware with Linux some_other_version and so on
   if [ "x$is_top_level" = xtrue ] && [ "x${GRUB_DISABLE_SUBMENU}" != xtrue ]; then
     linux_entry "${OS}" "${version}" simple \
-    "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
+    	"${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
+
+    linux_entry "${OS}" "${version}" console \
+        "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} systemd.unit=multi-user.target"
+
+   if [ "x${GRUB_DISABLE_RECOVERY}" != "xtrue" ]; then
+        linux_entry "${OS}" "${version}" recovery \
+                "${GRUB_CMDLINE_LINUX} systemd.unit=rescue.target"
+   fi
+
 
-    submenu_indentation="$grub_tab"
-    
     if [ -z "$boot_device_id" ]; then
 	boot_device_id="$(grub_get_device_id "${GRUB_DEVICE}")"
     fi
+
+    else ## we don' use this for now
+    submenu_indentation="$grub_tab"
     # TRANSLATORS: %s is replaced with an OS name
     echo "submenu '$(gettext_printf "Advanced options for %s" "${OS}" | grub_quote)' \$menuentry_id_option 'gnulinux-advanced-$boot_device_id' {"
     is_top_level=false
-  fi
 
-  linux_entry "${OS}" "${version}" advanced \
-              "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
-  if [ "x${GRUB_DISABLE_RECOVERY}" != "xtrue" ]; then
-    linux_entry "${OS}" "${version}" recovery \
-                "single ${GRUB_CMDLINE_LINUX}"
-  fi
+    linux_entry "${OS}" "${version}" console \
+              "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} systemd.unit=multi-user.target"
 
+   if [ "x${GRUB_DISABLE_RECOVERY}" != "xtrue" ]; then
+     linux_entry "${OS}" "${version}" recovery \
+                "${GRUB_CMDLINE_LINUX} systemd.unit=rescue.target"
+   fi
+fi
   list=`echo $list | tr ' ' '\n' | fgrep -vx "$linux" | tr '\n' ' '`
 done
 
