--- grub-2.02/util/grub.d/10_linux.in	2017-07-13 13:56:53.000000000 +0200
+++ grub-2.02.new/util/grub.d/10_linux.in	2017-07-13 13:55:50.000000000 +0200
@@ -79,6 +79,19 @@
 
 title_correction_code=
 
+_resume=""
+
+while read line; do
+	_device=$(echo $line | cut -f 1 -d ' ')
+	_type=$(echo $line | cut -f 2 -d ' ')
+	[ "$_type" != "partition" ] && continue
+	_uuid=$(/sbin/blkid -s UUID -o value $_device)
+	_resume="resume=UUID=$_uuid"
+	break
+done << EOF
+$(tail -n +2 /proc/swaps)
+EOF
+
 linux_entry ()
 {
   os="$1"
@@ -106,6 +119,8 @@
   fi
   if [ x$type != xsimple ] ; then
       case $type in
+	  console)
+	      title="$(gettext_printf "%s, with Linux %s (console mode)" "${os}" "${version}")" ;;
 	  recovery)
 	      title="$(printf "$(mktitle) (recovery mode)" "${version}")" ;;
 	  *)
@@ -154,11 +169,11 @@
   fi
   if [ -d /sys/firmware/efi ]; then
     sed "s/^/$submenu_indentation/" << EOF
-	${linuxefi} ${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args}
+	${linuxefi} ${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args} ${_resume}
 EOF
   else
     sed "s/^/$submenu_indentation/" << EOF
-	linux${sixteenbit} ${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args}
+	linux${sixteenbit} ${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args} ${_resume}
 EOF
   fi
   if test -n "${initrd}" ; then
@@ -286,11 +301,12 @@
     is_top_level=false
   fi
 
-  linux_entry "${OS}" "${version}" advanced \
-              "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT}"
+  linux_entry "${OS}" "${version}" console \
+              "${GRUB_CMDLINE_LINUX} ${GRUB_CMDLINE_LINUX_DEFAULT} systemd.unit=multi-user.target"
+
   if [ "x${GRUB_DISABLE_RECOVERY}" != "xtrue" ]; then
     linux_entry "${OS}" "${version}" recovery \
-                "single ${GRUB_CMDLINE_LINUX}"
+                "${GRUB_CMDLINE_LINUX} systemd.unit=rescue.target"
   fi
 
   list=`echo $list | tr ' ' '\n' | fgrep -vx "$linux" | tr '\n' ' '`
