diff -ur a/util/grub.d/10_linux.in b/util/grub.d/10_linux.in
--- a/util/grub.d/10_linux.in	2012-02-18 01:48:48.930000001 -0600
+++ b/util/grub.d/10_linux.in	2012-02-21 22:56:19.719998017 -0600
@@ -59,6 +59,12 @@
   fi
 fi
 
+modeset=""
+
+for i in `find /lib/modules -regex '^.*/\(nvidia\|fglrx\)\.ko.*$'`; do
+  modeset="nomodeset"
+done
+
 linux_entry ()
 {
   os="$1"
@@ -116,7 +122,7 @@
   message="$(gettext_printf "Loading Linux %s ..." ${version})"
   cat << EOF
 	echo	'$message'
-	linux	${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args}
+	linux	${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args} ${modeset}
 EOF
   if test -n "${initrd}" ; then
     message="$(gettext_printf "Loading initial ramdisk ...")"
