diff -Naur light-monitor-v1.6.orig/makefile light-monitor-v1.6/makefile
--- light-monitor-v1.6.orig/makefile	2007-07-03 17:08:26.000000000 +0200
+++ light-monitor-v1.6/makefile	2007-07-03 17:08:54.000000000 +0200
@@ -1,7 +1,7 @@
 DEST=light-monitor-v1.6
 all : light-monitor light-calendar 
 
-OPTS=-O2 -Wall 
+OPTS=${CFLAGS}
 #OPTS=-g -Wall #-DDEBUG
 
 XFTCFLAGS=`xft-config --cflags`
@@ -57,6 +57,8 @@
 
 
 install:
-	install -m 0755 -s light-monitor /usr/bin/
-	install -m 0755 -s light-calendar /usr/bin/
-	install -m 0644    light-monitor.conf /etc/
+	install -d ${DESTDIR}/usr/bin/
+	install -d ${DESTDIR}/etc/
+	install -m 0755 light-monitor ${DESTDIR}/usr/bin/
+	install -m 0755 light-calendar ${DESTDIR}/usr/bin/
+	install -m 0644    light-monitor.conf ${DESTDIR}/etc/
