diff -Naur dbus-0.33.orig/configure dbus-0.33/configure
--- dbus-0.33.orig/configure	2005-06-08 20:48:17.000000000 +0000
+++ dbus-0.33/configure	2005-06-08 20:49:08.000000000 +0000
@@ -26688,6 +26688,7 @@
         if test -f "$dir/qglobal.h"; then
             have_qt=yes
             DBUS_QT_CXXFLAGS="-I$dir"
+			DBUS_QT_LIBS="-I$QTDIR/lib/libqt-mt.la"
        fi
     done
 fi
@@ -26699,17 +26700,6 @@
 echo "${ECHO_T}not found" >&6
 fi
 
-if (! kde-config >& /dev/null); then
-    have_qt=no
-else
-    kdelibs=`kde-config --install lib --expandvars 2>/dev/null`
-    if test -z $kdelibs -o ! -f $kdelibs/libkdecore.la; then
-        have_qt=no
-    else
-        DBUS_QT_LIBS="$kdelibs/libkdecore.la"
-    fi
-fi
-
 if test x$have_qt = xno ; then
     { echo "$as_me:$LINENO: WARNING: Qt development libraries not found" >&5
 echo "$as_me: WARNING: Qt development libraries not found" >&2;}
diff -Naur dbus-0.33.orig/python/Makefile.am dbus-0.33/python/Makefile.am
--- dbus-0.33.orig/python/Makefile.am	2005-06-08 20:48:17.000000000 +0000
+++ dbus-0.33/python/Makefile.am	2005-06-08 20:48:53.000000000 +0000
@@ -5,7 +5,7 @@
 dbusdir = $(pythondir)
 dbus_PYTHON = dbus.py
 
-dbusbindingsdir = $(pythondir)
+dbusbindingsdir = $(pyexecdir)
 dbusbindings_LTLIBRARIES = dbus_bindings.la
 
 dbus_bindings_la_LDFLAGS = -module -avoid-version -fPIC -export-symbols-regex initdbus_bindings
diff -Naur dbus-0.33.orig/python/Makefile.in dbus-0.33/python/Makefile.in
--- dbus-0.33.orig/python/Makefile.in	2005-06-08 20:48:17.000000000 +0000
+++ dbus-0.33/python/Makefile.in	2005-06-08 20:48:53.000000000 +0000
@@ -312,7 +312,7 @@
 INCLUDES = -I$(top_builddir) -I$(top_builddir)/dbus $(DBUS_CLIENT_CFLAGS) $(DBUS_GLIB_CFLAGS) $(DBUS_GLIB_TOOL_CFLAGS) $(PYTHON_INCLUDES)
 dbusdir = $(pythondir)
 dbus_PYTHON = dbus.py
-dbusbindingsdir = $(pythondir)
+dbusbindingsdir = $(pyexecdir)
 dbusbindings_LTLIBRARIES = dbus_bindings.la
 dbus_bindings_la_LDFLAGS = -module -avoid-version -fPIC -export-symbols-regex initdbus_bindings
 dbus_bindings_la_LIBADD = $(top_builddir)/dbus/libdbus-1.la $(top_builddir)/glib/libdbus-glib-1.la
diff -Naur dbus-0.33.orig/tools/dbus-launch.c dbus-0.33/tools/dbus-launch.c
--- dbus-0.33.orig/tools/dbus-launch.c	2005-06-08 20:48:17.000000000 +0000
+++ dbus-0.33/tools/dbus-launch.c	2005-06-08 20:48:46.000000000 +0000
@@ -26,6 +26,7 @@
 #include <fcntl.h>
 #include <signal.h>
 #include <sys/wait.h>
+#include <sys/select.h>
 #include <errno.h>
 #include <stdio.h>
 #include <string.h>
