diff -Naur libarchive-2.1.10/libarchive/archive_entry.c libarchive-2.1.10-p/libarchive/archive_entry.c
--- libarchive-2.1.10/libarchive/archive_entry.c	2007-04-28 04:34:46.000000000 +0200
+++ libarchive-2.1.10-p/libarchive/archive_entry.c	2007-05-03 17:24:57.000000000 +0200
@@ -99,6 +99,18 @@
 #undef max
 #define	max(a, b)	((a)>(b)?(a):(b))
 
+static void
+dev_set_major(dev_t *d, dev_t m)
+{
+      *d = makedev(m, minor(*d));
+}
+
+static void
+dev_set_minor(dev_t *d, dev_t m)
+{
+      *d = makedev(major(*d), m);
+}
+
 /*
  * Handle wide character (i.e., Unicode) and non-wide character
  * strings transparently.
@@ -780,19 +792,13 @@
 void
 archive_entry_set_devmajor(struct archive_entry *entry, dev_t m)
 {
-	dev_t d;
-
-	d = entry->ae_stat.st_dev;
-	entry->ae_stat.st_dev = makedev(major(m), minor(d));
+      dev_set_major(&entry->ae_stat.st_dev, m);
 }
 
 void
 archive_entry_set_devminor(struct archive_entry *entry, dev_t m)
 {
-	dev_t d;
-
-	d = entry->ae_stat.st_dev;
-	entry->ae_stat.st_dev = makedev(major(d), minor(m));
+      dev_set_minor(entry->ae_stat.st_dev, m);
 }
 
 /* Set symlink if symlink is already set, else set hardlink. */
@@ -858,19 +864,13 @@
 void
 archive_entry_set_rdevmajor(struct archive_entry *entry, dev_t m)
 {
-	dev_t d;
-
-	d = entry->ae_stat.st_rdev;
-	entry->ae_stat.st_rdev = makedev(major(m), minor(d));
+      dev_set_major(&entry->ae_stat.st_rdev, m);
 }
 
 void
 archive_entry_set_rdevminor(struct archive_entry *entry, dev_t m)
 {
-	dev_t d;
-
-	d = entry->ae_stat.st_rdev;
-	entry->ae_stat.st_rdev = makedev(major(d), minor(m));
+      dev_set_minor(&entry->ae_stat.st_rdev, m);
 }
 
 void
