diff --git a/src/tracker-extract/tracker-main.h b/src/tracker-extract/tracker-main.h
index 77d798e..4af2e3e 100644
--- a/src/tracker-extract/tracker-main.h
+++ b/src/tracker-extract/tracker-main.h
@@ -25,11 +25,6 @@
 
 G_BEGIN_DECLS
 
-/* This is used to not shutdown after the default of 30 seconds if we
- * get more work to do.
- */
-void              tracker_main_quit_timeout_reset (void);
-
 /* Enables getting the config object from extractors */
 TrackerConfig    *tracker_main_get_config         (void);
 