diff --git a/src/tracker-extract/tracker-extract-gif.c b/src/tracker-extract/tracker-extract-gif.c
index f260bd5..ecff110 100644
--- a/src/tracker-extract/tracker-extract-gif.c
+++ b/src/tracker-extract/tracker-extract-gif.c
@@ -318,7 +318,7 @@ read_metadata (GifFileType          *gifFile,
 		TrackerResource *white_balance;
 
 		white_balance = tracker_resource_new (xd->white_balance);
-		tracker_resource_set_relation (metadata, "nmm:meteringMode", white_balance);
+		tracker_resource_set_relation (metadata, "nmm:whiteBalance", white_balance);
 		g_object_unref (white_balance);
 	}
 
diff --git a/src/tracker-extract/tracker-extract-jpeg.c b/src/tracker-extract/tracker-extract-jpeg.c
index c6b07bf..00b3764 100644
--- a/src/tracker-extract/tracker-extract-jpeg.c
+++ b/src/tracker-extract/tracker-extract-jpeg.c
@@ -417,7 +417,7 @@ tracker_extract_get_metadata (TrackerExtractInfo *info)
 		TrackerResource *white_balance;
 
 		white_balance = tracker_resource_new (md.white_balance);
-		tracker_resource_set_relation (metadata, "nmm:meteringMode", white_balance);
+		tracker_resource_set_relation (metadata, "nmm:whiteBalance", white_balance);
 		g_object_unref (white_balance);
 	}
 
diff --git a/src/tracker-extract/tracker-extract-pdf.c b/src/tracker-extract/tracker-extract-pdf.c
index acce7c8..a89b94a 100644
--- a/src/tracker-extract/tracker-extract-pdf.c
+++ b/src/tracker-extract/tracker-extract-pdf.c
@@ -481,7 +481,7 @@ tracker_extract_get_metadata (TrackerExtractInfo *info)
 			TrackerResource *white_balance;
 
 			white_balance = tracker_resource_new (xd->white_balance);
-			tracker_resource_set_relation (metadata, "nmm:meteringMode", white_balance);
+			tracker_resource_set_relation (metadata, "nmm:whiteBalance", white_balance);
 			g_object_unref (white_balance);
 		}
 
diff --git a/src/tracker-extract/tracker-extract-png.c b/src/tracker-extract/tracker-extract-png.c
index 0e19765..54d54ac 100644
--- a/src/tracker-extract/tracker-extract-png.c
+++ b/src/tracker-extract/tracker-extract-png.c
@@ -433,7 +433,7 @@ read_metadata (TrackerResource      *metadata,
 		TrackerResource *white_balance;
 
 		white_balance = tracker_resource_new (md.white_balance);
-		tracker_resource_set_relation (metadata, "nmm:meteringMode", white_balance);
+		tracker_resource_set_relation (metadata, "nmm:whiteBalance", white_balance);
 		g_object_unref (white_balance);
 	}
 
diff --git a/src/tracker-extract/tracker-extract-tiff.c b/src/tracker-extract/tracker-extract-tiff.c
index 64ff894..3c411c3 100644
--- a/src/tracker-extract/tracker-extract-tiff.c
+++ b/src/tracker-extract/tracker-extract-tiff.c
@@ -545,7 +545,7 @@ tracker_extract_get_metadata (TrackerExtractInfo *info)
 		TrackerResource *white_balance;
 
 		white_balance = tracker_resource_new (md.white_balance);
-		tracker_resource_set_relation (metadata, "nmm:meteringMode", white_balance);
+		tracker_resource_set_relation (metadata, "nmm:whiteBalance", white_balance);
 		g_object_unref (white_balance);
 	}
