--- src/src/java/org/jdesktop/swingx/auth/LoginService.java	2008-06-13 21:43:03.000000000 +0200
+++ src/src/java/org/jdesktop/swingx/auth/LoginService.java	2008-06-13 21:43:18.000000000 +0200
@@ -27,7 +27,7 @@
 import javax.swing.event.EventListenerList;
 
 import org.jdesktop.beans.AbstractBean;
-import org.jdesktop.swingworker.SwingWorker;
+import javax.swing.SwingWorker;
 
 /**
  * <b>LoginService</b> is the abstract base class for all classes implementing
